<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.SearchIO.BlatIO</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.SearchIO-module.html">Package&nbsp;SearchIO</a> ::
        Module&nbsp;BlatIO
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.SearchIO.BlatIO-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.SearchIO.BlatIO-module.html">Module Bio.SearchIO.BlatIO</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2012 by Wibowo Arindrarto.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-docstring">"""Bio.SearchIO parser for BLAT output formats.</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-docstring">This module adds support for parsing BLAT outputs. BLAT (BLAST-Like Alignment</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-docstring">Tool) is a sequence similarity search program initially built for annotating</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-docstring">the human genome.</tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-docstring">Bio.SearchIO.BlastIO was tested using standalone BLAT version 34, psLayout</tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-docstring">version 3. It should be able to parse psLayout version 4 without problems.</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-docstring">More information on BLAT is available from these sites:</tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-docstring">    - Publication: http://genome.cshlp.org/content/12/4/656</tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-docstring">    - User guide: http://genome.ucsc.edu/goldenPath/help/blatSpec.html</tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-docstring">    - Source download: http://www.soe.ucsc.edu/~kent/src</tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-docstring">    - Executable download: http://hgdownload.cse.ucsc.edu/admin/exe/</tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-docstring">    - Blat score calculation: http://genome.ucsc.edu/FAQ/FAQblat.html#blat4</tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-docstring">Supported Formats</tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-docstring">=================</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"><tt class="py-docstring">BlatIO supports parsing, indexing, and writing for both PSL and PSLX output</tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"><tt class="py-docstring">formats, with or without header. To parse, index, or write PSLX files, use the</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"><tt class="py-docstring">'pslx' keyword argument and set it to True.</tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"><tt class="py-docstring">    # blat-psl defaults to PSL files</tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SearchIO</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; psl = 'Blat/psl_34_004.psl'</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; qresult = SearchIO.read(psl, 'blat-psl')</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; qresult</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"><tt class="py-docstring">    QueryResult(id='hg19_dna', 10 hits)</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"><tt class="py-docstring">    # set the pslx flag to parse PSLX files</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; pslx = 'Blat/pslx_34_004.pslx'</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; qresult = SearchIO.read(pslx, 'blat-psl', pslx=True)</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; qresult</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line"><tt class="py-docstring">    QueryResult(id='hg19_dna', 10 hits)</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line"><tt class="py-docstring">For parsing and indexing, you do not need to specify whether the file has a</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line"><tt class="py-docstring">header or not. For writing, if you want to write a header, you can set the</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line"><tt class="py-docstring">'header' keyword argument to True. This will write a 'psLayout version 3' header</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line"><tt class="py-docstring">to your output file.</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line"><tt class="py-docstring">    from Bio import SearchIO</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line"><tt class="py-docstring">    qresult = SearchIO.read(psl, 'blat-psl')</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line"><tt class="py-docstring">    SearchIO.write(qresult, 'header.psl', header=True)</tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line"><tt class="py-docstring">    &lt;stdout&gt; (1, 10, 19, 23)</tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line"><tt class="py-docstring">Note that the number of HSPFragments written may exceed the number of HSP</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line"><tt class="py-docstring">objects. This is because in PSL files, it is possible to have single matches</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line"><tt class="py-docstring">consisting of noncontiguous sequence fragments. This is where the HSPFragment</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"><tt class="py-docstring">object comes into play. These fragments are grouped into a single HSP because</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line"><tt class="py-docstring">they share the same statistics (e.g. match numbers, BLAT score, etc.). However,</tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line"><tt class="py-docstring">they do not share the same sequence attributes, such as the start and end</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line"><tt class="py-docstring">coordinates, making them distinct objects.</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line"><tt class="py-docstring">In addition to parsing PSL(X) files, BlatIO also computes the percent identities</tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line"><tt class="py-docstring">and scores of your search results. This is done using the calculation formula</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line"><tt class="py-docstring">posted here: http://genome.ucsc.edu/FAQ/FAQblat.html#blat4. It mimics the score</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line"><tt class="py-docstring">and percent identity calculation done by UCSC's web BLAT service.</tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line"><tt class="py-docstring">Since BlatIO parses the file in a single pass, it expects all results from</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line"><tt class="py-docstring">the same query to be in consecutive rows. If the results from one query are</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line"><tt class="py-docstring">spread in nonconsecutive rows, BlatIO will consider them to be separate</tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line"><tt class="py-docstring">QueryResult objects.</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line"><tt class="py-docstring">In most cases, the PSL(X) format uses the same coordinate system as Python</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line"><tt class="py-docstring">(zero-based, half open). These coordinates are anchored on the plus strand.</tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line"><tt class="py-docstring">However, if the query aligns on the minus strand, BLAT will anchor the qStarts</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line"><tt class="py-docstring">coordinates on the minus strand instead. BlatIO is aware of this, and will</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line"><tt class="py-docstring">re-anchor the qStarts coordinates to the plus strand whenever it sees a minus</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line"><tt class="py-docstring">strand query match. Conversely, when you write out to a PSL(X) file, BlatIO will</tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line"><tt class="py-docstring">reanchor qStarts to the minus strand again.</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line"><tt class="py-docstring">BlatIO provides the following attribute-column mapping:</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line"><tt class="py-docstring">+----------------+-------------------------+-----------------------------------+</tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line"><tt class="py-docstring">| Object         | Attribute               | Column Name, Value                |</tt> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line"><tt class="py-docstring">+================+=========================+===================================+</tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line"><tt class="py-docstring">| QueryResutl    | id                      | Q name, query sequence ID         |</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line"><tt class="py-docstring">|                +-------------------------+-----------------------------------+</tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line"><tt class="py-docstring">|                | seq_len                 | Q size, query sequence full       |</tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line"><tt class="py-docstring">|                |                         | length                            |</tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line"><tt class="py-docstring">+----------------+-------------------------+-----------------------------------+</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line"><tt class="py-docstring">| Hit            | id                      | T name, hit sequence ID           |</tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line"><tt class="py-docstring">|                +-------------------------+-----------------------------------+</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line"><tt class="py-docstring">|                | seq_len                 | T size, hit sequence full length  |</tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line"><tt class="py-docstring">+----------------+-------------------------+-----------------------------------+</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line"><tt class="py-docstring">| HSP            | hit_end                 | T end, end coordinate of the last |</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line"><tt class="py-docstring">|                |                         | hit fragment                      |</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line"><tt class="py-docstring">|                +-------------------------+-----------------------------------+</tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line"><tt class="py-docstring">|                | hit_gap_num             | T gap bases, number of bases      |</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line"><tt class="py-docstring">|                |                         | inserted in hit                   |</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line"><tt class="py-docstring">|                +-------------------------+-----------------------------------+</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line"><tt class="py-docstring">|                | hit_gapopen_num         | T gap count, number of hit gap    |</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line"><tt class="py-docstring">|                |                         | inserts                           |</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line"><tt class="py-docstring">|                +-------------------------+-----------------------------------+</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line"><tt class="py-docstring">|                | hit_span_all            | blockSizes, sizes of each         |</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line"><tt class="py-docstring">|                |                         | fragment                          |</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line"><tt class="py-docstring">|                +-------------------------+-----------------------------------+</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line"><tt class="py-docstring">|                | hit_start               | T start, start coordinate of the  |</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line"><tt class="py-docstring">|                |                         | first hit fragment                |</tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line"><tt class="py-docstring">|                +-------------------------+-----------------------------------+</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line"><tt class="py-docstring">|                | hit_start_all           | tStarts, start coordinate of each |</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line"><tt class="py-docstring">|                |                         | hit fragment                      |</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line"><tt class="py-docstring">|                +-------------------------+-----------------------------------+</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line"><tt class="py-docstring">|                | match_num               | match, number of non-repeat       |</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line"><tt class="py-docstring">|                |                         | matches                           |</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line"><tt class="py-docstring">|                +-------------------------+-----------------------------------+</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line"><tt class="py-docstring">|                | mismatch_num            | mismatch, number of mismatches    |</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line"><tt class="py-docstring">|                +-------------------------+-----------------------------------+</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line"><tt class="py-docstring">|                | match_rep_num           | rep. match, number of matches     |</tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line"><tt class="py-docstring">|                |                         | that are part of repeats          |</tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line"><tt class="py-docstring">|                +-------------------------+-----------------------------------+</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line"><tt class="py-docstring">|                | n_num                   | N's, number of N bases            |</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line"><tt class="py-docstring">|                +-------------------------+-----------------------------------+</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line"><tt class="py-docstring">|                | query_end               | Q end, end coordinate of the last |</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line"><tt class="py-docstring">|                +-------------------------+-----------------------------------+</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line"><tt class="py-docstring">|                |                         | query fragment                    |</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line"><tt class="py-docstring">|                | query_gap_num           | Q gap bases, number of bases      |</tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line"><tt class="py-docstring">|                |                         | inserted in query                 |</tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line"><tt class="py-docstring">|                +-------------------------+-----------------------------------+</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line"><tt class="py-docstring">|                | query_gapopen_num       | Q gap count, number of query gap  |</tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line"><tt class="py-docstring">|                |                         | inserts                           |</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line"><tt class="py-docstring">|                +-------------------------+-----------------------------------+</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line"><tt class="py-docstring">|                | query_span_all          | blockSizes, sizes of each         |</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line"><tt class="py-docstring">|                |                         | fragment                          |</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line"><tt class="py-docstring">|                +-------------------------+-----------------------------------+</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line"><tt class="py-docstring">|                | query_start             | Q start, start coordinate of the  |</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line"><tt class="py-docstring">|                |                         | first query block                 |</tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line"><tt class="py-docstring">|                +-------------------------+-----------------------------------+</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line"><tt class="py-docstring">|                | query_start_all         | qStarts, start coordinate of each |</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line"><tt class="py-docstring">|                |                         | query fragment                    |</tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line"><tt class="py-docstring">|                +-------------------------+-----------------------------------+</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line"><tt class="py-docstring">|                | len [*]_                | block count, the number of blocks |</tt> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line"><tt class="py-docstring">|                |                         | in the alignment                  |</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line"><tt class="py-docstring">+----------------+-------------------------+-----------------------------------+</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line"><tt class="py-docstring">| HSPFragment    | hit                     | hit sequence, if present          |</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line"><tt class="py-docstring">|                +-------------------------+-----------------------------------+</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line"><tt class="py-docstring">|                | hit_strand              | strand, hit sequence strand       |</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line"><tt class="py-docstring">|                +-------------------------+-----------------------------------+</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line"><tt class="py-docstring">|                | query                   | query sequence, if present        |</tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line"><tt class="py-docstring">|                +-------------------------+-----------------------------------+</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line"><tt class="py-docstring">|                | query_strand            | strand, query sequence strand     |</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line"><tt class="py-docstring">+----------------+-------------------------+-----------------------------------+</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line"><tt class="py-docstring">In addition to the column mappings above, BlatIO also provides the following</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line"><tt class="py-docstring">object attributes:</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line"><tt class="py-docstring">+----------------+-------------------------+-----------------------------------+</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line"><tt class="py-docstring">| Object         | Attribute               | Value                             |</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line"><tt class="py-docstring">+================+=========================+===================================+</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line"><tt class="py-docstring">| HSP            | gapopen_num             | Q gap count + T gap count, total  |</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line"><tt class="py-docstring">|                |                         |  number of gap openings           |</tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line"><tt class="py-docstring">|                +-------------------------+-----------------------------------+</tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line"><tt class="py-docstring">|                | ident_num               | matches + repmatches, total       |</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line"><tt class="py-docstring">|                |                         | number of identical residues      |</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line"><tt class="py-docstring">|                +-------------------------+-----------------------------------+</tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line"><tt class="py-docstring">|                | ident_pct               | percent identity, calculated      |</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line"><tt class="py-docstring">|                |                         | using UCSC's formula              |</tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line"><tt class="py-docstring">|                +-------------------------+-----------------------------------+</tt> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line"><tt class="py-docstring">|                | query_is_protein        | boolean, whether the query        |</tt> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line"><tt class="py-docstring">|                |                         | sequence is a protein             |</tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line"><tt class="py-docstring">|                +-------------------------+-----------------------------------+</tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line"><tt class="py-docstring">|                | score                   | HSP score, calculated using       |</tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line"><tt class="py-docstring">|                |                         | UCSC's formula                    |</tt> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line"><tt class="py-docstring">+----------------+-------------------------+-----------------------------------+</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line"><tt class="py-docstring">Finally, the default HSP and HSPFragment properties are also provided. See the</tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line"><tt class="py-docstring">HSP and HSPFragment documentation for more details on these properties.</tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line"><tt class="py-docstring">.. [*] You can obtain the number of blocks / fragments in the HSP by invoking</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line"><tt class="py-docstring">   ``len`` on the HSP</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">re</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">math</tt> <tt class="py-keyword">import</tt> <tt class="py-name">log</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line"> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-1', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt id="link-2" class="py-name" targets="Function Bio._py3k._as_bytes()=Bio._py3k-module.html#_as_bytes"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-2', '_as_bytes', 'link-2');">_as_bytes</a></tt><tt class="py-op">,</tt> <tt id="link-3" class="py-name" targets="Function Bio._py3k._bytes_to_string()=Bio._py3k-module.html#_bytes_to_string"><a title="Bio._py3k._bytes_to_string" class="py-name" href="#" onclick="return doclink('link-3', '_bytes_to_string', 'link-3');">_bytes_to_string</a></tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-4" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-4', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-5" class="py-name"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-5', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">zip</tt> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line"> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-6" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-6', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-7" class="py-name" targets="Package Bio.Alphabet=Bio.Alphabet-module.html,Class Bio.Alphabet.Alphabet=Bio.Alphabet.Alphabet-class.html"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-7', 'Alphabet', 'link-7');">Alphabet</a></tt> <tt class="py-keyword">import</tt> <tt id="link-8" class="py-name" targets="Variable Bio.Alphabet.generic_dna=Bio.Alphabet-module.html#generic_dna"><a title="Bio.Alphabet.generic_dna" class="py-name" href="#" onclick="return doclink('link-8', 'generic_dna', 'link-8');">generic_dna</a></tt> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-9" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-9', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-10" class="py-name" targets="Package Bio.SearchIO=Bio.SearchIO-module.html"><a title="Bio.SearchIO" class="py-name" href="#" onclick="return doclink('link-10', 'SearchIO', 'link-10');">SearchIO</a></tt><tt class="py-op">.</tt><tt id="link-11" class="py-name" targets="Module Bio.SearchIO._index=Bio.SearchIO._index-module.html,Module Bio.SeqIO._index=Bio.SeqIO._index-module.html"><a title="Bio.SearchIO._index
Bio.SeqIO._index" class="py-name" href="#" onclick="return doclink('link-11', '_index', 'link-11');">_index</a></tt> <tt class="py-keyword">import</tt> <tt id="link-12" class="py-name" targets="Class Bio.SearchIO._index.SearchIndexer=Bio.SearchIO._index.SearchIndexer-class.html"><a title="Bio.SearchIO._index.SearchIndexer" class="py-name" href="#" onclick="return doclink('link-12', 'SearchIndexer', 'link-12');">SearchIndexer</a></tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-13" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-13', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-14" class="py-name"><a title="Bio.SearchIO" class="py-name" href="#" onclick="return doclink('link-14', 'SearchIO', 'link-10');">SearchIO</a></tt><tt class="py-op">.</tt><tt id="link-15" class="py-name" targets="Package Bio.SearchIO._model=Bio.SearchIO._model-module.html"><a title="Bio.SearchIO._model" class="py-name" href="#" onclick="return doclink('link-15', '_model', 'link-15');">_model</a></tt> <tt class="py-keyword">import</tt> <tt id="link-16" class="py-name" targets="Class Bio.SearchIO._model.query.QueryResult=Bio.SearchIO._model.query.QueryResult-class.html"><a title="Bio.SearchIO._model.query.QueryResult" class="py-name" href="#" onclick="return doclink('link-16', 'QueryResult', 'link-16');">QueryResult</a></tt><tt class="py-op">,</tt> <tt id="link-17" class="py-name" targets="Class Bio.SearchIO._model.hit.Hit=Bio.SearchIO._model.hit.Hit-class.html"><a title="Bio.SearchIO._model.hit.Hit" class="py-name" href="#" onclick="return doclink('link-17', 'Hit', 'link-17');">Hit</a></tt><tt class="py-op">,</tt> <tt id="link-18" class="py-name" targets="Class Bio.Blast.Record.HSP=Bio.Blast.Record.HSP-class.html,Class Bio.SearchIO._model.hsp.HSP=Bio.SearchIO._model.hsp.HSP-class.html"><a title="Bio.Blast.Record.HSP
Bio.SearchIO._model.hsp.HSP" class="py-name" href="#" onclick="return doclink('link-18', 'HSP', 'link-18');">HSP</a></tt><tt class="py-op">,</tt> <tt id="link-19" class="py-name" targets="Class Bio.SearchIO._model.hsp.HSPFragment=Bio.SearchIO._model.hsp.HSPFragment-class.html"><a title="Bio.SearchIO._model.hsp.HSPFragment" class="py-name" href="#" onclick="return doclink('link-19', 'HSPFragment', 'link-19');">HSPFragment</a></tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line"> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line"> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line"><tt class="py-name">__all__</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-string">'BlatPslParser'</tt><tt class="py-op">,</tt> <tt class="py-string">'BlatPslIndexer'</tt><tt class="py-op">,</tt> <tt class="py-string">'BlatPslWriter'</tt><tt class="py-op">)</tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line"> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line"> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line"><tt class="py-comment"># precompile regex patterns</tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line"><tt id="link-20" class="py-name" targets="Variable Bio.SearchIO.BlatIO._PTR_ROW_CHECK=Bio.SearchIO.BlatIO-module.html#_PTR_ROW_CHECK"><a title="Bio.SearchIO.BlatIO._PTR_ROW_CHECK" class="py-name" href="#" onclick="return doclink('link-20', '_PTR_ROW_CHECK', 'link-20');">_PTR_ROW_CHECK</a></tt> <tt class="py-op">=</tt> <tt class="py-string">r'^\d+\s+\d+\s+\d+\s+\d+'</tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line"><tt id="link-21" class="py-name" targets="Variable Bio.SearchIO.BlatIO._RE_ROW_CHECK=Bio.SearchIO.BlatIO-module.html#_RE_ROW_CHECK"><a title="Bio.SearchIO.BlatIO._RE_ROW_CHECK" class="py-name" href="#" onclick="return doclink('link-21', '_RE_ROW_CHECK', 'link-21');">_RE_ROW_CHECK</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt id="link-22" class="py-name"><a title="Bio.SearchIO.BlatIO._PTR_ROW_CHECK" class="py-name" href="#" onclick="return doclink('link-22', '_PTR_ROW_CHECK', 'link-20');">_PTR_ROW_CHECK</a></tt><tt class="py-op">)</tt> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line"><tt id="link-23" class="py-name" targets="Variable Bio.SearchIO.BlatIO._RE_ROW_CHECK_IDX=Bio.SearchIO.BlatIO-module.html#_RE_ROW_CHECK_IDX"><a title="Bio.SearchIO.BlatIO._RE_ROW_CHECK_IDX" class="py-name" href="#" onclick="return doclink('link-23', '_RE_ROW_CHECK_IDX', 'link-23');">_RE_ROW_CHECK_IDX</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt id="link-24" class="py-name"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-24', '_as_bytes', 'link-2');">_as_bytes</a></tt><tt class="py-op">(</tt><tt id="link-25" class="py-name"><a title="Bio.SearchIO.BlatIO._PTR_ROW_CHECK" class="py-name" href="#" onclick="return doclink('link-25', '_PTR_ROW_CHECK', 'link-20');">_PTR_ROW_CHECK</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line"> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line"> </tt>
<a name="_list_from_csv"></a><div id="_list_from_csv-def"><a name="L202"></a><tt class="py-lineno">202</tt> <a class="py-toggle" href="#" id="_list_from_csv-toggle" onclick="return toggle('_list_from_csv');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlatIO-module.html#_list_from_csv">_list_from_csv</a><tt class="py-op">(</tt><tt class="py-param">csv_string</tt><tt class="py-op">,</tt> <tt class="py-param">caster</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_list_from_csv-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_list_from_csv-expanded"><a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line">    <tt class="py-docstring">"""Transform the given comma-separated string into a list (PRIVATE).</tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line"><tt class="py-docstring">    :param csv_string: comma-separated input string</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line"><tt class="py-docstring">    :type csv_string: string</tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line"><tt class="py-docstring">    :param caster: function used to cast each item in the input string</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line"><tt class="py-docstring">                   to its intended type</tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line"><tt class="py-docstring">    :type caster: callable, accepts string, returns object</tt> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">caster</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt id="link-26" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.x=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#x,Variable Bio.Statistics.lowess.x=Bio.Statistics.lowess-module.html#x"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-26', 'x', 'link-26');">x</a></tt> <tt class="py-keyword">for</tt> <tt id="link-27" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-27', 'x', 'link-26');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">csv_string</tt><tt class="py-op">.</tt><tt id="link-28" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-28', 'split', 'link-28');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">','</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt id="link-29" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-29', 'x', 'link-26');">x</a></tt><tt class="py-op">]</tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">caster</tt><tt class="py-op">(</tt><tt id="link-30" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-30', 'x', 'link-26');">x</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-31" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-31', 'x', 'link-26');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">csv_string</tt><tt class="py-op">.</tt><tt id="link-32" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-32', 'split', 'link-28');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">','</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt id="link-33" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-33', 'x', 'link-26');">x</a></tt><tt class="py-op">]</tt> </tt>
</div><a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line"> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line"> </tt>
<a name="_reorient_starts"></a><div id="_reorient_starts-def"><a name="L218"></a><tt class="py-lineno">218</tt> <a class="py-toggle" href="#" id="_reorient_starts-toggle" onclick="return toggle('_reorient_starts');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlatIO-module.html#_reorient_starts">_reorient_starts</a><tt class="py-op">(</tt><tt class="py-param">starts</tt><tt class="py-op">,</tt> <tt class="py-param">blksizes</tt><tt class="py-op">,</tt> <tt class="py-param">seqlen</tt><tt class="py-op">,</tt> <tt class="py-param">strand</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_reorient_starts-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_reorient_starts-expanded"><a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line">    <tt class="py-docstring">"""Reorients block starts into the opposite strand's coordinates (PRIVATE).</tt> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line"><tt class="py-docstring">    :param starts: start coordinates</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line"><tt class="py-docstring">    :type starts: list [int]</tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line"><tt class="py-docstring">    :param blksizes: block sizes</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line"><tt class="py-docstring">    :type blksizes: list [int]</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line"><tt class="py-docstring">    :param seqlen: sequence length</tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line"><tt class="py-docstring">    :type seqlen: int</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line"><tt class="py-docstring">    :param strand: sequence strand</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line"><tt class="py-docstring">    :type strand: int, choice of -1, 0, or 1</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">starts</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">blksizes</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> \ </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line">            <tt class="py-string">"Unequal start coordinates and block sizes list (%r vs %r)"</tt> \ </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">            <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">starts</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">blksizes</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line">    <tt class="py-comment"># see: http://genome.ucsc.edu/goldenPath/help/blatSpec.html</tt> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line">    <tt class="py-comment"># no need to reorient if it's already the positive strand</tt> </tt>
<a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt id="link-34" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HSPConsumer.strand()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#strand,Variable Bio.SeqFeature.CompoundLocation.strand=Bio.SeqFeature.CompoundLocation-class.html#strand,Variable Bio.SeqFeature.FeatureLocation.strand=Bio.SeqFeature.FeatureLocation-class.html#strand,Variable Bio.SeqFeature.SeqFeature.strand=Bio.SeqFeature.SeqFeature-class.html#strand"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-34', 'strand', 'link-34');">strand</a></tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">starts</tt> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line">        <tt class="py-comment"># the plus-oriented coordinate is calculated by this:</tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line">        <tt class="py-comment"># plus_coord = length - minus_coord - block_size</tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">seqlen</tt> <tt class="py-op">-</tt> <tt id="link-35" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.start=Bio.SeqFeature.CompoundLocation-class.html#start,Variable Bio.SeqFeature.FeatureLocation.start=Bio.SeqFeature.FeatureLocation-class.html#start"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-35', 'start', 'link-35');">start</a></tt> <tt class="py-op">-</tt> <tt class="py-name">blksize</tt> <tt class="py-keyword">for</tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line">                <tt id="link-36" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-36', 'start', 'link-35');">start</a></tt><tt class="py-op">,</tt> <tt class="py-name">blksize</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">starts</tt><tt class="py-op">,</tt> <tt class="py-name">blksizes</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
</div><a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line"> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line"> </tt>
<a name="_is_protein"></a><div id="_is_protein-def"><a name="L245"></a><tt class="py-lineno">245</tt> <a class="py-toggle" href="#" id="_is_protein-toggle" onclick="return toggle('_is_protein');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlatIO-module.html#_is_protein">_is_protein</a><tt class="py-op">(</tt><tt class="py-param">psl</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_is_protein-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_is_protein-expanded"><a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line">    <tt class="py-comment"># check if query is protein or not</tt> </tt>
<a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line">    <tt class="py-comment"># adapted from http://genome.ucsc.edu/FAQ/FAQblat.html#blat4</tt> </tt>
<a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'strand'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> </tt>
<a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'strand'</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'+'</tt><tt class="py-op">:</tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'tend'</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'tstarts'</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line">                    <tt class="py-number">3</tt> <tt class="py-op">*</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'blocksizes'</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'strand'</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'-'</tt><tt class="py-op">:</tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'tstart'</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'tsize'</tt><tt class="py-op">]</tt> <tt class="py-op">-</tt> \ </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line">                    <tt class="py-op">(</tt><tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'tstarts'</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-number">3</tt> <tt class="py-op">*</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'blocksizes'</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line"> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">False</tt> </tt>
</div><a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line"> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line"> </tt>
<a name="_calc_millibad"></a><div id="_calc_millibad-def"><a name="L259"></a><tt class="py-lineno">259</tt> <a class="py-toggle" href="#" id="_calc_millibad-toggle" onclick="return toggle('_calc_millibad');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlatIO-module.html#_calc_millibad">_calc_millibad</a><tt class="py-op">(</tt><tt class="py-param">psl</tt><tt class="py-op">,</tt> <tt class="py-param">is_protein</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_calc_millibad-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_calc_millibad-expanded"><a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line">    <tt class="py-comment"># calculates millibad</tt> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line">    <tt class="py-comment"># adapted from http://genome.ucsc.edu/FAQ/FAQblat.html#blat4</tt> </tt>
<a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line">    <tt class="py-name">size_mul</tt> <tt class="py-op">=</tt> <tt class="py-number">3</tt> <tt class="py-keyword">if</tt> <tt class="py-name">is_protein</tt> <tt class="py-keyword">else</tt> <tt class="py-number">1</tt> </tt>
<a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line">    <tt class="py-name">millibad</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line"> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line">    <tt class="py-name">qali_size</tt> <tt class="py-op">=</tt> <tt class="py-name">size_mul</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'qend'</tt><tt class="py-op">]</tt> <tt class="py-op">-</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'qstart'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line">    <tt class="py-name">tali_size</tt> <tt class="py-op">=</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'tend'</tt><tt class="py-op">]</tt> <tt class="py-op">-</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'tstart'</tt><tt class="py-op">]</tt> </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line">    <tt class="py-name">ali_size</tt> <tt class="py-op">=</tt> <tt id="link-37" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.min=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#min"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-37', 'min', 'link-37');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">qali_size</tt><tt class="py-op">,</tt> <tt class="py-name">tali_size</tt><tt class="py-op">)</tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">ali_size</tt> <tt class="py-op">&lt;=</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-number">0</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line"> </tt>
<a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line">    <tt class="py-name">size_dif</tt> <tt class="py-op">=</tt> <tt class="py-name">qali_size</tt> <tt class="py-op">-</tt> <tt class="py-name">tali_size</tt> </tt>
<a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line">    <tt class="py-name">size_dif</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> <tt class="py-keyword">if</tt> <tt class="py-name">size_dif</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt> <tt class="py-keyword">else</tt> <tt class="py-name">size_dif</tt> </tt>
<a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line"> </tt>
<a name="L274"></a><tt class="py-lineno">274</tt>  <tt class="py-line">    <tt class="py-name">total</tt> <tt class="py-op">=</tt> <tt class="py-name">size_mul</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'matches'</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'repmatches'</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'mismatches'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">total</tt> <tt class="py-op">!=</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line">        <tt class="py-name">millibad</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-number">1000</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'mismatches'</tt><tt class="py-op">]</tt> <tt class="py-op">*</tt> <tt class="py-name">size_mul</tt> <tt class="py-op">+</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'qnuminsert'</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> </tt>
<a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line">                <tt id="link-38" class="py-name" targets="Method Bio.Blast.NCBIStandalone._BlastConsumer.round()=Bio.Blast.NCBIStandalone._BlastConsumer-class.html#round,Method Bio.Blast.NCBIStandalone._DescriptionConsumer.round()=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#round"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.round
Bio.Blast.NCBIStandalone._DescriptionConsumer.round" class="py-name" href="#" onclick="return doclink('link-38', 'round', 'link-38');">round</a></tt><tt class="py-op">(</tt><tt class="py-number">3</tt> <tt class="py-op">*</tt> <tt class="py-name">log</tt><tt class="py-op">(</tt><tt class="py-number">1</tt> <tt class="py-op">+</tt> <tt class="py-name">size_dif</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-name">total</tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line"> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">millibad</tt> </tt>
</div><a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line"> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line"> </tt>
<a name="_calc_score"></a><div id="_calc_score-def"><a name="L282"></a><tt class="py-lineno">282</tt> <a class="py-toggle" href="#" id="_calc_score-toggle" onclick="return toggle('_calc_score');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlatIO-module.html#_calc_score">_calc_score</a><tt class="py-op">(</tt><tt class="py-param">psl</tt><tt class="py-op">,</tt> <tt class="py-param">is_protein</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_calc_score-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_calc_score-expanded"><a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line">    <tt class="py-comment"># calculates score</tt> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line">    <tt class="py-comment"># adapted from http://genome.ucsc.edu/FAQ/FAQblat.html#blat4</tt> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line">    <tt class="py-name">size_mul</tt> <tt class="py-op">=</tt> <tt class="py-number">3</tt> <tt class="py-keyword">if</tt> <tt class="py-name">is_protein</tt> <tt class="py-keyword">else</tt> <tt class="py-number">1</tt> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-name">size_mul</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'matches'</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'repmatches'</tt><tt class="py-op">]</tt> <tt class="py-op">&gt;&gt;</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line">            <tt class="py-op">-</tt> <tt class="py-name">size_mul</tt> <tt class="py-op">*</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'mismatches'</tt><tt class="py-op">]</tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">            <tt class="py-op">-</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'qnuminsert'</tt><tt class="py-op">]</tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line">            <tt class="py-op">-</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'tnuminsert'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div><a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line"> </tt>
<a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line"> </tt>
<a name="_create_hsp"></a><div id="_create_hsp-def"><a name="L292"></a><tt class="py-lineno">292</tt> <a class="py-toggle" href="#" id="_create_hsp-toggle" onclick="return toggle('_create_hsp');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlatIO-module.html#_create_hsp">_create_hsp</a><tt class="py-op">(</tt><tt class="py-param">hid</tt><tt class="py-op">,</tt> <tt class="py-param">qid</tt><tt class="py-op">,</tt> <tt class="py-param">psl</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_create_hsp-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_create_hsp-expanded"><a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line">    <tt class="py-comment"># protein flag</tt> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line">    <tt class="py-name">is_protein</tt> <tt class="py-op">=</tt> <tt id="link-39" class="py-name" targets="Function Bio.SearchIO.BlatIO._is_protein()=Bio.SearchIO.BlatIO-module.html#_is_protein"><a title="Bio.SearchIO.BlatIO._is_protein" class="py-name" href="#" onclick="return doclink('link-39', '_is_protein', 'link-39');">_is_protein</a></tt><tt class="py-op">(</tt><tt class="py-name">psl</tt><tt class="py-op">)</tt> </tt>
<a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line">    <tt class="py-comment"># strand</tt> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line">    <tt class="py-comment"># if query is protein, strand is 0</tt> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">is_protein</tt><tt class="py-op">:</tt> </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line">        <tt class="py-name">qstrand</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line">        <tt class="py-name">qstrand</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> <tt class="py-keyword">if</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'strand'</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'+'</tt> <tt class="py-keyword">else</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line">    <tt class="py-comment"># try to get hit strand, if it exists</tt> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line">        <tt class="py-name">hstrand</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> <tt class="py-keyword">if</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'strand'</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'+'</tt> <tt class="py-keyword">else</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt> </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">IndexError</tt><tt class="py-op">:</tt> </tt>
<a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line">        <tt class="py-name">hstrand</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt>  <tt class="py-comment"># hit strand defaults to plus</tt> </tt>
<a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line"> </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line">    <tt class="py-name">blocksize_multiplier</tt> <tt class="py-op">=</tt> <tt class="py-number">3</tt> <tt class="py-keyword">if</tt> <tt class="py-name">is_protein</tt> <tt class="py-keyword">else</tt> <tt class="py-number">1</tt> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line">    <tt class="py-comment"># query block starts</tt> </tt>
<a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line">    <tt class="py-name">qstarts</tt> <tt class="py-op">=</tt> <tt id="link-40" class="py-name" targets="Function Bio.SearchIO.BlatIO._reorient_starts()=Bio.SearchIO.BlatIO-module.html#_reorient_starts"><a title="Bio.SearchIO.BlatIO._reorient_starts" class="py-name" href="#" onclick="return doclink('link-40', '_reorient_starts', 'link-40');">_reorient_starts</a></tt><tt class="py-op">(</tt><tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'qstarts'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line">            <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'blocksizes'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'qsize'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">qstrand</tt><tt class="py-op">)</tt> </tt>
<a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line">    <tt class="py-comment"># hit block starts</tt> </tt>
<a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'strand'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> </tt>
<a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line">        <tt class="py-name">hstarts</tt> <tt class="py-op">=</tt> <tt id="link-41" class="py-name"><a title="Bio.SearchIO.BlatIO._reorient_starts" class="py-name" href="#" onclick="return doclink('link-41', '_reorient_starts', 'link-40');">_reorient_starts</a></tt><tt class="py-op">(</tt><tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'tstarts'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line">                <tt class="py-op">[</tt><tt class="py-name">blocksize_multiplier</tt> <tt class="py-op">*</tt> <tt id="link-42" class="py-name" targets="Variable Bio.PDB.Polypeptide.i=Bio.PDB.Polypeptide-module.html#i,Variable Bio.SeqIO.SffIO.i=Bio.SeqIO.SffIO-module.html#i"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-42', 'i', 'link-42');">i</a></tt> <tt class="py-keyword">for</tt> <tt id="link-43" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-43', 'i', 'link-42');">i</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'blocksizes'</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line">                <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'tsize'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">hstrand</tt><tt class="py-op">)</tt> </tt>
<a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line">        <tt class="py-name">hstarts</tt> <tt class="py-op">=</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'tstarts'</tt><tt class="py-op">]</tt> </tt>
<a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line">    <tt class="py-comment"># set query and hit coords</tt> </tt>
<a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line">    <tt class="py-comment"># this assumes each block has no gaps (which seems to be the case)</tt> </tt>
<a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">qstarts</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">hstarts</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'blocksizes'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line">    <tt id="link-44" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.query_range_all=Bio.SearchIO._model.hsp.HSP-class.html#query_range_all"><a title="Bio.SearchIO._model.hsp.HSP.query_range_all" class="py-name" href="#" onclick="return doclink('link-44', 'query_range_all', 'link-44');">query_range_all</a></tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">qstarts</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt id="link-45" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-45', 'x', 'link-26');">x</a></tt> <tt class="py-op">+</tt> <tt id="link-46" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.y=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#y"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-46', 'y', 'link-46');">y</a></tt> <tt class="py-keyword">for</tt> <tt id="link-47" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-47', 'x', 'link-26');">x</a></tt><tt class="py-op">,</tt> <tt id="link-48" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-48', 'y', 'link-46');">y</a></tt> <tt class="py-keyword">in</tt> </tt>
<a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line">                                         <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">qstarts</tt><tt class="py-op">,</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'blocksizes'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line">    <tt id="link-49" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.hit_range_all=Bio.SearchIO._model.hsp.HSP-class.html#hit_range_all"><a title="Bio.SearchIO._model.hsp.HSP.hit_range_all" class="py-name" href="#" onclick="return doclink('link-49', 'hit_range_all', 'link-49');">hit_range_all</a></tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">hstarts</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt id="link-50" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-50', 'x', 'link-26');">x</a></tt> <tt class="py-op">+</tt> <tt id="link-51" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-51', 'y', 'link-46');">y</a></tt> <tt class="py-op">*</tt> <tt class="py-name">blocksize_multiplier</tt> <tt class="py-keyword">for</tt> <tt id="link-52" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-52', 'x', 'link-26');">x</a></tt><tt class="py-op">,</tt> <tt id="link-53" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-53', 'y', 'link-46');">y</a></tt> <tt class="py-keyword">in</tt> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line">                                       <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">hstarts</tt><tt class="py-op">,</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'blocksizes'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L325"></a><tt class="py-lineno">325</tt>  <tt class="py-line">    <tt class="py-comment"># check length of sequences and coordinates, all must match</tt> </tt>
<a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-string">'tseqs'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">psl</tt> <tt class="py-keyword">and</tt> <tt class="py-string">'qseqs'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">psl</tt><tt class="py-op">:</tt> </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'tseqs'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'qseqs'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> \ </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line">                <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-54" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_range_all" class="py-name" href="#" onclick="return doclink('link-54', 'query_range_all', 'link-44');">query_range_all</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-55" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.hit_range_all" class="py-name" href="#" onclick="return doclink('link-55', 'hit_range_all', 'link-49');">hit_range_all</a></tt><tt class="py-op">)</tt> </tt>
<a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-56" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_range_all" class="py-name" href="#" onclick="return doclink('link-56', 'query_range_all', 'link-44');">query_range_all</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-57" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.hit_range_all" class="py-name" href="#" onclick="return doclink('link-57', 'hit_range_all', 'link-49');">hit_range_all</a></tt><tt class="py-op">)</tt> </tt>
<a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line"> </tt>
<a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line">    <tt class="py-name">frags</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L333"></a><tt class="py-lineno">333</tt>  <tt class="py-line">    <tt class="py-comment"># iterating over query_range_all, but hit_range_all works just as well</tt> </tt>
<a name="L334"></a><tt class="py-lineno">334</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">idx</tt><tt class="py-op">,</tt> <tt class="py-name">qcoords</tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt id="link-58" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_range_all" class="py-name" href="#" onclick="return doclink('link-58', 'query_range_all', 'link-44');">query_range_all</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L335"></a><tt class="py-lineno">335</tt>  <tt class="py-line">        <tt class="py-name">hseqlist</tt> <tt class="py-op">=</tt> <tt class="py-name">psl</tt><tt class="py-op">.</tt><tt id="link-59" class="py-name" targets="Method Bio.Crystal.Crystal.get()=Bio.Crystal.Crystal-class.html#get,Method Bio.Data.CodonTable.AmbiguousForwardTable.get()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#get,Method Bio.File._IndexedSeqFileDict.get()=Bio.File._IndexedSeqFileDict-class.html#get,Method Bio.File._IndexedSeqFileProxy.get()=Bio.File._IndexedSeqFileProxy-class.html#get,Method Bio.File._SQLiteManySeqFilesDict.get()=Bio.File._SQLiteManySeqFilesDict-class.html#get,Method Bio.Restriction.Restriction.RestrictionBatch.get()=Bio.Restriction.Restriction.RestrictionBatch-class.html#get,Method Bio.SearchIO._index.SearchIndexer.get()=Bio.SearchIO._index.SearchIndexer-class.html#get,Method Bio.SeqIO._index.SeqFileRandomAccess.get()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#get,Method Bio.SeqIO._index.SffRandomAccess.get()=Bio.SeqIO._index.SffRandomAccess-class.html#get,Method Bio.SeqIO._index.SffTrimedRandomAccess.get()=Bio.SeqIO._index.SffTrimedRandomAccess-class.html#get,Method Bio.SeqIO._index.UniprotRandomAccess.get()=Bio.SeqIO._index.UniprotRandomAccess-class.html#get"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-59', 'get', 'link-59');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'tseqs'</tt><tt class="py-op">)</tt> </tt>
<a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line">        <tt class="py-name">hseq</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">hseqlist</tt> <tt class="py-keyword">else</tt> <tt class="py-name">hseqlist</tt><tt class="py-op">[</tt><tt class="py-name">idx</tt><tt class="py-op">]</tt> </tt>
<a name="L337"></a><tt class="py-lineno">337</tt>  <tt class="py-line">        <tt class="py-name">qseqlist</tt> <tt class="py-op">=</tt> <tt class="py-name">psl</tt><tt class="py-op">.</tt><tt id="link-60" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-60', 'get', 'link-59');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'qseqs'</tt><tt class="py-op">)</tt> </tt>
<a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line">        <tt class="py-name">qseq</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">qseqlist</tt> <tt class="py-keyword">else</tt> <tt class="py-name">qseqlist</tt><tt class="py-op">[</tt><tt class="py-name">idx</tt><tt class="py-op">]</tt> </tt>
<a name="L339"></a><tt class="py-lineno">339</tt>  <tt class="py-line">        <tt class="py-name">frag</tt> <tt class="py-op">=</tt> <tt id="link-61" class="py-name"><a title="Bio.SearchIO._model.hsp.HSPFragment" class="py-name" href="#" onclick="return doclink('link-61', 'HSPFragment', 'link-19');">HSPFragment</a></tt><tt class="py-op">(</tt><tt class="py-name">hid</tt><tt class="py-op">,</tt> <tt class="py-name">qid</tt><tt class="py-op">,</tt> <tt id="link-62" class="py-name" targets="Module Bio.SearchIO._model.hit=Bio.SearchIO._model.hit-module.html,Variable Bio.SearchIO._model.hsp.HSP.hit=Bio.SearchIO._model.hsp.HSP-class.html#hit,Variable Bio.SearchIO._model.hsp.HSPFragment.hit=Bio.SearchIO._model.hsp.HSPFragment-class.html#hit"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-62', 'hit', 'link-62');">hit</a></tt><tt class="py-op">=</tt><tt class="py-name">hseq</tt><tt class="py-op">,</tt> <tt id="link-63" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HSPConsumer.query()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#query,Variable Bio.SearchIO._model.hsp.HSP.query=Bio.SearchIO._model.hsp.HSP-class.html#query,Variable Bio.SearchIO._model.hsp.HSPFragment.query=Bio.SearchIO._model.hsp.HSPFragment-class.html#query,Module Bio.SearchIO._model.query=Bio.SearchIO._model.query-module.html"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.query
Bio.SearchIO._model.hsp.HSP.query
Bio.SearchIO._model.hsp.HSPFragment.query
Bio.SearchIO._model.query" class="py-name" href="#" onclick="return doclink('link-63', 'query', 'link-63');">query</a></tt><tt class="py-op">=</tt><tt class="py-name">qseq</tt><tt class="py-op">)</tt> </tt>
<a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line">        <tt class="py-comment"># set alphabet</tt> </tt>
<a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line">        <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-64" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.alphabet=Bio.SearchIO._model.hsp.HSP-class.html#alphabet,Variable Bio.SearchIO._model.hsp.HSPFragment.alphabet=Bio.SearchIO._model.hsp.HSPFragment-class.html#alphabet"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-64', 'alphabet', 'link-64');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-65" class="py-name"><a title="Bio.Alphabet.generic_dna" class="py-name" href="#" onclick="return doclink('link-65', 'generic_dna', 'link-8');">generic_dna</a></tt> </tt>
<a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line">        <tt class="py-comment"># set coordinates</tt> </tt>
<a name="L343"></a><tt class="py-lineno">343</tt>  <tt class="py-line">        <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-66" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.query_start=Bio.SearchIO._model.hsp.HSP-class.html#query_start,Variable Bio.SearchIO._model.hsp.HSPFragment.query_start=Bio.SearchIO._model.hsp.HSPFragment-class.html#query_start"><a title="Bio.SearchIO._model.hsp.HSP.query_start
Bio.SearchIO._model.hsp.HSPFragment.query_start" class="py-name" href="#" onclick="return doclink('link-66', 'query_start', 'link-66');">query_start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">qcoords</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L344"></a><tt class="py-lineno">344</tt>  <tt class="py-line">        <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-67" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.query_end=Bio.SearchIO._model.hsp.HSP-class.html#query_end,Variable Bio.SearchIO._model.hsp.HSPFragment.query_end=Bio.SearchIO._model.hsp.HSPFragment-class.html#query_end"><a title="Bio.SearchIO._model.hsp.HSP.query_end
Bio.SearchIO._model.hsp.HSPFragment.query_end" class="py-name" href="#" onclick="return doclink('link-67', 'query_end', 'link-67');">query_end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">qcoords</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L345"></a><tt class="py-lineno">345</tt>  <tt class="py-line">        <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-68" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.hit_start=Bio.SearchIO._model.hsp.HSP-class.html#hit_start,Variable Bio.SearchIO._model.hsp.HSPFragment.hit_start=Bio.SearchIO._model.hsp.HSPFragment-class.html#hit_start"><a title="Bio.SearchIO._model.hsp.HSP.hit_start
Bio.SearchIO._model.hsp.HSPFragment.hit_start" class="py-name" href="#" onclick="return doclink('link-68', 'hit_start', 'link-68');">hit_start</a></tt> <tt class="py-op">=</tt> <tt id="link-69" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.hit_range_all" class="py-name" href="#" onclick="return doclink('link-69', 'hit_range_all', 'link-49');">hit_range_all</a></tt><tt class="py-op">[</tt><tt class="py-name">idx</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L346"></a><tt class="py-lineno">346</tt>  <tt class="py-line">        <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-70" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.hit_end=Bio.SearchIO._model.hsp.HSP-class.html#hit_end,Variable Bio.SearchIO._model.hsp.HSPFragment.hit_end=Bio.SearchIO._model.hsp.HSPFragment-class.html#hit_end"><a title="Bio.SearchIO._model.hsp.HSP.hit_end
Bio.SearchIO._model.hsp.HSPFragment.hit_end" class="py-name" href="#" onclick="return doclink('link-70', 'hit_end', 'link-70');">hit_end</a></tt> <tt class="py-op">=</tt> <tt id="link-71" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.hit_range_all" class="py-name" href="#" onclick="return doclink('link-71', 'hit_range_all', 'link-49');">hit_range_all</a></tt><tt class="py-op">[</tt><tt class="py-name">idx</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L347"></a><tt class="py-lineno">347</tt>  <tt class="py-line">        <tt class="py-comment"># and strands</tt> </tt>
<a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line">        <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-72" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.query_strand=Bio.SearchIO._model.hsp.HSP-class.html#query_strand,Variable Bio.SearchIO._model.hsp.HSPFragment.query_strand=Bio.SearchIO._model.hsp.HSPFragment-class.html#query_strand"><a title="Bio.SearchIO._model.hsp.HSP.query_strand
Bio.SearchIO._model.hsp.HSPFragment.query_strand" class="py-name" href="#" onclick="return doclink('link-72', 'query_strand', 'link-72');">query_strand</a></tt> <tt class="py-op">=</tt> <tt class="py-name">qstrand</tt> </tt>
<a name="L349"></a><tt class="py-lineno">349</tt>  <tt class="py-line">        <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-73" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.hit_strand=Bio.SearchIO._model.hsp.HSP-class.html#hit_strand,Variable Bio.SearchIO._model.hsp.HSPFragment.hit_strand=Bio.SearchIO._model.hsp.HSPFragment-class.html#hit_strand"><a title="Bio.SearchIO._model.hsp.HSP.hit_strand
Bio.SearchIO._model.hsp.HSPFragment.hit_strand" class="py-name" href="#" onclick="return doclink('link-73', 'hit_strand', 'link-73');">hit_strand</a></tt> <tt class="py-op">=</tt> <tt class="py-name">hstrand</tt> </tt>
<a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line">        <tt class="py-name">frags</tt><tt class="py-op">.</tt><tt id="link-74" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-74', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">)</tt> </tt>
<a name="L351"></a><tt class="py-lineno">351</tt>  <tt class="py-line"> </tt>
<a name="L352"></a><tt class="py-lineno">352</tt>  <tt class="py-line">    <tt class="py-comment"># create hsp object</tt> </tt>
<a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line">    <tt id="link-75" class="py-name" targets="Module Bio.SearchIO._model.hsp=Bio.SearchIO._model.hsp-module.html"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-75', 'hsp', 'link-75');">hsp</a></tt> <tt class="py-op">=</tt> <tt id="link-76" class="py-name"><a title="Bio.Blast.Record.HSP
Bio.SearchIO._model.hsp.HSP" class="py-name" href="#" onclick="return doclink('link-76', 'HSP', 'link-18');">HSP</a></tt><tt class="py-op">(</tt><tt class="py-name">frags</tt><tt class="py-op">)</tt> </tt>
<a name="L354"></a><tt class="py-lineno">354</tt>  <tt class="py-line">    <tt class="py-comment"># check if start and end are set correctly</tt> </tt>
<a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt id="link-77" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-77', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-78" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_start
Bio.SearchIO._model.hsp.HSPFragment.query_start" class="py-name" href="#" onclick="return doclink('link-78', 'query_start', 'link-66');">query_start</a></tt> <tt class="py-op">==</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'qstart'</tt><tt class="py-op">]</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt id="link-79" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-79', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-80" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_end
Bio.SearchIO._model.hsp.HSPFragment.query_end" class="py-name" href="#" onclick="return doclink('link-80', 'query_end', 'link-67');">query_end</a></tt> <tt class="py-op">==</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'qend'</tt><tt class="py-op">]</tt> </tt>
<a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt id="link-81" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-81', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-82" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.hit_start
Bio.SearchIO._model.hsp.HSPFragment.hit_start" class="py-name" href="#" onclick="return doclink('link-82', 'hit_start', 'link-68');">hit_start</a></tt> <tt class="py-op">==</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'tstart'</tt><tt class="py-op">]</tt> </tt>
<a name="L358"></a><tt class="py-lineno">358</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt id="link-83" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-83', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-84" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.hit_end
Bio.SearchIO._model.hsp.HSPFragment.hit_end" class="py-name" href="#" onclick="return doclink('link-84', 'hit_end', 'link-70');">hit_end</a></tt> <tt class="py-op">==</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'tend'</tt><tt class="py-op">]</tt> </tt>
<a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line">    <tt class="py-comment"># and check block spans as well</tt> </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line">    <tt class="py-name">hit_spans</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">span</tt> <tt class="py-op">/</tt> <tt class="py-name">blocksize_multiplier</tt> <tt class="py-keyword">for</tt> <tt class="py-name">span</tt> <tt class="py-keyword">in</tt> <tt id="link-85" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-85', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-86" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.hit_span_all=Bio.SearchIO._model.hsp.HSP-class.html#hit_span_all"><a title="Bio.SearchIO._model.hsp.HSP.hit_span_all" class="py-name" href="#" onclick="return doclink('link-86', 'hit_span_all', 'link-86');">hit_span_all</a></tt><tt class="py-op">]</tt> </tt>
<a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">hit_spans</tt> <tt class="py-op">==</tt> <tt id="link-87" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-87', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-88" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.query_span_all=Bio.SearchIO._model.hsp.HSP-class.html#query_span_all"><a title="Bio.SearchIO._model.hsp.HSP.query_span_all" class="py-name" href="#" onclick="return doclink('link-88', 'query_span_all', 'link-88');">query_span_all</a></tt> <tt class="py-op">==</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'blocksizes'</tt><tt class="py-op">]</tt> </tt>
<a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line">    <tt class="py-comment"># set its attributes</tt> </tt>
<a name="L363"></a><tt class="py-lineno">363</tt>  <tt class="py-line">    <tt id="link-89" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-89', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">match_num</tt> <tt class="py-op">=</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'matches'</tt><tt class="py-op">]</tt> </tt>
<a name="L364"></a><tt class="py-lineno">364</tt>  <tt class="py-line">    <tt id="link-90" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-90', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">mismatch_num</tt> <tt class="py-op">=</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'mismatches'</tt><tt class="py-op">]</tt> </tt>
<a name="L365"></a><tt class="py-lineno">365</tt>  <tt class="py-line">    <tt id="link-91" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-91', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">match_rep_num</tt> <tt class="py-op">=</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'repmatches'</tt><tt class="py-op">]</tt> </tt>
<a name="L366"></a><tt class="py-lineno">366</tt>  <tt class="py-line">    <tt id="link-92" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-92', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">n_num</tt> <tt class="py-op">=</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'ncount'</tt><tt class="py-op">]</tt> </tt>
<a name="L367"></a><tt class="py-lineno">367</tt>  <tt class="py-line">    <tt id="link-93" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-93', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">query_gapopen_num</tt> <tt class="py-op">=</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'qnuminsert'</tt><tt class="py-op">]</tt> </tt>
<a name="L368"></a><tt class="py-lineno">368</tt>  <tt class="py-line">    <tt id="link-94" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-94', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">query_gap_num</tt> <tt class="py-op">=</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'qbaseinsert'</tt><tt class="py-op">]</tt> </tt>
<a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line">    <tt id="link-95" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-95', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">hit_gapopen_num</tt> <tt class="py-op">=</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'tnuminsert'</tt><tt class="py-op">]</tt> </tt>
<a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line">    <tt id="link-96" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-96', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">hit_gap_num</tt> <tt class="py-op">=</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'tbaseinsert'</tt><tt class="py-op">]</tt> </tt>
<a name="L371"></a><tt class="py-lineno">371</tt>  <tt class="py-line"> </tt>
<a name="L372"></a><tt class="py-lineno">372</tt>  <tt class="py-line">    <tt id="link-97" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-97', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">ident_num</tt> <tt class="py-op">=</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'matches'</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'repmatches'</tt><tt class="py-op">]</tt> </tt>
<a name="L373"></a><tt class="py-lineno">373</tt>  <tt class="py-line">    <tt id="link-98" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-98', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">gapopen_num</tt> <tt class="py-op">=</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'qnuminsert'</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'tnuminsert'</tt><tt class="py-op">]</tt> </tt>
<a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line">    <tt id="link-99" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-99', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">gap_num</tt> <tt class="py-op">=</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'qbaseinsert'</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'tbaseinsert'</tt><tt class="py-op">]</tt> </tt>
<a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line">    <tt id="link-100" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-100', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">query_is_protein</tt> <tt class="py-op">=</tt> <tt class="py-name">is_protein</tt> </tt>
<a name="L376"></a><tt class="py-lineno">376</tt>  <tt class="py-line">    <tt id="link-101" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-101', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">ident_pct</tt> <tt class="py-op">=</tt> <tt class="py-number">100.0</tt> <tt class="py-op">-</tt> <tt id="link-102" class="py-name" targets="Function Bio.SearchIO.BlatIO._calc_millibad()=Bio.SearchIO.BlatIO-module.html#_calc_millibad"><a title="Bio.SearchIO.BlatIO._calc_millibad" class="py-name" href="#" onclick="return doclink('link-102', '_calc_millibad', 'link-102');">_calc_millibad</a></tt><tt class="py-op">(</tt><tt class="py-name">psl</tt><tt class="py-op">,</tt> <tt class="py-name">is_protein</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt class="py-number">0.1</tt> </tt>
<a name="L377"></a><tt class="py-lineno">377</tt>  <tt class="py-line">    <tt id="link-103" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-103', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-104" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HSPConsumer.score()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#score"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-104', 'score', 'link-104');">score</a></tt> <tt class="py-op">=</tt> <tt id="link-105" class="py-name" targets="Function Bio.SearchIO.BlatIO._calc_score()=Bio.SearchIO.BlatIO-module.html#_calc_score"><a title="Bio.SearchIO.BlatIO._calc_score" class="py-name" href="#" onclick="return doclink('link-105', '_calc_score', 'link-105');">_calc_score</a></tt><tt class="py-op">(</tt><tt class="py-name">psl</tt><tt class="py-op">,</tt> <tt class="py-name">is_protein</tt><tt class="py-op">)</tt> </tt>
<a name="L378"></a><tt class="py-lineno">378</tt>  <tt class="py-line">    <tt class="py-comment"># helper flag, for writing</tt> </tt>
<a name="L379"></a><tt class="py-lineno">379</tt>  <tt class="py-line">    <tt id="link-106" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-106', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">_has_hit_strand</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'strand'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">2</tt> </tt>
<a name="L380"></a><tt class="py-lineno">380</tt>  <tt class="py-line"> </tt>
<a name="L381"></a><tt class="py-lineno">381</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-107" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-107', 'hsp', 'link-75');">hsp</a></tt> </tt>
</div><a name="L382"></a><tt class="py-lineno">382</tt>  <tt class="py-line"> </tt>
<a name="L383"></a><tt class="py-lineno">383</tt>  <tt class="py-line"> </tt>
<a name="BlatPslParser"></a><div id="BlatPslParser-def"><a name="L384"></a><tt class="py-lineno">384</tt> <a class="py-toggle" href="#" id="BlatPslParser-toggle" onclick="return toggle('BlatPslParser');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.SearchIO.BlatIO.BlatPslParser-class.html">BlatPslParser</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlatPslParser-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="BlatPslParser-expanded"><a name="L385"></a><tt class="py-lineno">385</tt>  <tt class="py-line">    <tt class="py-docstring">"""Parser for the BLAT PSL format."""</tt> </tt>
<a name="L386"></a><tt class="py-lineno">386</tt>  <tt class="py-line"> </tt>
<a name="BlatPslParser.__init__"></a><div id="BlatPslParser.__init__-def"><a name="L387"></a><tt class="py-lineno">387</tt> <a class="py-toggle" href="#" id="BlatPslParser.__init__-toggle" onclick="return toggle('BlatPslParser.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlatIO.BlatPslParser-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">pslx</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlatPslParser.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlatPslParser.__init__-expanded"><a name="L388"></a><tt class="py-lineno">388</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the class."""</tt> </tt>
<a name="L389"></a><tt class="py-lineno">389</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt> </tt>
<a name="L390"></a><tt class="py-lineno">390</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-108" class="py-name" targets="Method Bio.File.UndoHandle.readline()=Bio.File.UndoHandle-class.html#readline,Method Bio.bgzf.BgzfReader.readline()=Bio.bgzf.BgzfReader-class.html#readline"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-108', 'readline', 'link-108');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L391"></a><tt class="py-lineno">391</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">pslx</tt> <tt class="py-op">=</tt> <tt class="py-name">pslx</tt> </tt>
</div><a name="L392"></a><tt class="py-lineno">392</tt>  <tt class="py-line"> </tt>
<a name="BlatPslParser.__iter__"></a><div id="BlatPslParser.__iter__-def"><a name="L393"></a><tt class="py-lineno">393</tt> <a class="py-toggle" href="#" id="BlatPslParser.__iter__-toggle" onclick="return toggle('BlatPslParser.__iter__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlatIO.BlatPslParser-class.html#__iter__">__iter__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlatPslParser.__iter__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlatPslParser.__iter__-expanded"><a name="L394"></a><tt class="py-lineno">394</tt>  <tt class="py-line">        <tt class="py-comment"># break out if it's an empty file</tt> </tt>
<a name="L395"></a><tt class="py-lineno">395</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L396"></a><tt class="py-lineno">396</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L397"></a><tt class="py-lineno">397</tt>  <tt class="py-line"> </tt>
<a name="L398"></a><tt class="py-lineno">398</tt>  <tt class="py-line">        <tt class="py-comment"># read through header</tt> </tt>
<a name="L399"></a><tt class="py-lineno">399</tt>  <tt class="py-line">        <tt class="py-comment"># this assumes that the result row match the regex</tt> </tt>
<a name="L400"></a><tt class="py-lineno">400</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-keyword">not</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-109" class="py-name" targets="Method Bio.AlignIO.MafIO.MafIndex.search()=Bio.AlignIO.MafIO.MafIndex-class.html#search,Method Bio.KDTree.KDTree'.KDTree.search()=Bio.KDTree.KDTree%27.KDTree-class.html#search,Method Bio.PDB.NeighborSearch'.NeighborSearch.search()=Bio.PDB.NeighborSearch%27.NeighborSearch-class.html#search,Method Bio.Phylo.TreeConstruction.NNITreeSearcher.search()=Bio.Phylo.TreeConstruction.NNITreeSearcher-class.html#search,Method Bio.Phylo.TreeConstruction.TreeSearcher.search()=Bio.Phylo.TreeConstruction.TreeSearcher-class.html#search,Class Method Bio.Restriction.Restriction.AbstractCut.search()=Bio.Restriction.Restriction.AbstractCut-class.html#search,Method Bio.Restriction.Restriction.RestrictionBatch.search()=Bio.Restriction.Restriction.RestrictionBatch-class.html#search,Function Bio.SCOP.search()=Bio.SCOP-module.html#search,Function Bio.TogoWS.search()=Bio.TogoWS-module.html#search,Method Bio.motifs.Instances.search()=Bio.motifs.Instances-class.html#search,Method Bio.motifs.matrix.PositionSpecificScoringMatrix.search()=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#search"><a title="Bio.AlignIO.MafIO.MafIndex.search
Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-109', 'search', 'link-109');">search</a></tt><tt class="py-op">(</tt><tt id="link-110" class="py-name"><a title="Bio.SearchIO.BlatIO._RE_ROW_CHECK" class="py-name" href="#" onclick="return doclink('link-110', '_RE_ROW_CHECK', 'link-21');">_RE_ROW_CHECK</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-111" class="py-name" targets="Method Bio.Seq.Seq.strip()=Bio.Seq.Seq-class.html#strip"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-111', 'strip', 'link-111');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L401"></a><tt class="py-lineno">401</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-112" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-112', 'readline', 'link-108');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L402"></a><tt class="py-lineno">402</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L403"></a><tt class="py-lineno">403</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> </tt>
<a name="L404"></a><tt class="py-lineno">404</tt>  <tt class="py-line"> </tt>
<a name="L405"></a><tt class="py-lineno">405</tt>  <tt class="py-line">        <tt class="py-comment"># parse into query results</tt> </tt>
<a name="L406"></a><tt class="py-lineno">406</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">qresult</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-113" class="py-name" targets="Method Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._parse_qresult()=Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#_parse_qresult,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser._parse_qresult()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser-class.html#_parse_qresult,Method Bio.SearchIO.BlatIO.BlatPslParser._parse_qresult()=Bio.SearchIO.BlatIO.BlatPslParser-class.html#_parse_qresult,Method Bio.SearchIO.ExonerateIO._base._BaseExonerateParser._parse_qresult()=Bio.SearchIO.ExonerateIO._base._BaseExonerateParser-class.html#_parse_qresult,Method Bio.SearchIO.FastaIO.FastaM10Parser._parse_qresult()=Bio.SearchIO.FastaIO.FastaM10Parser-class.html#_parse_qresult,Method Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabParser._parse_qresult()=Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabParser-class.html#_parse_qresult,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser._parse_qresult()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser-class.html#_parse_qresult,Method Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser._parse_qresult()=Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html#_parse_qresult,Method Bio.SearchIO.InterproscanIO.interproscan_xml.InterproscanXmlParser._parse_qresult()=Bio.SearchIO.InterproscanIO.interproscan_xml.InterproscanXmlParser-class.html#_parse_qresult"><a title="Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._parse_qresult
Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser._parse_qresult
Bio.SearchIO.BlatIO.BlatPslParser._parse_qresult
Bio.SearchIO.ExonerateIO._base._BaseExonerateParser._parse_qresult
Bio.SearchIO.FastaIO.FastaM10Parser._parse_qresult
Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabParser._parse_qresult
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser._parse_qresult
Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser._parse_qresult
Bio.SearchIO.InterproscanIO.interproscan_xml.InterproscanXmlParser._parse_qresult" class="py-name" href="#" onclick="return doclink('link-113', '_parse_qresult', 'link-113');">_parse_qresult</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L407"></a><tt class="py-lineno">407</tt>  <tt class="py-line">            <tt class="py-name">qresult</tt><tt class="py-op">.</tt><tt class="py-name">program</tt> <tt class="py-op">=</tt> <tt class="py-string">'blat'</tt> </tt>
<a name="L408"></a><tt class="py-lineno">408</tt>  <tt class="py-line">            <tt class="py-keyword">yield</tt> <tt class="py-name">qresult</tt> </tt>
</div><a name="L409"></a><tt class="py-lineno">409</tt>  <tt class="py-line"> </tt>
<a name="BlatPslParser._parse_row"></a><div id="BlatPslParser._parse_row-def"><a name="L410"></a><tt class="py-lineno">410</tt> <a class="py-toggle" href="#" id="BlatPslParser._parse_row-toggle" onclick="return toggle('BlatPslParser._parse_row');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlatIO.BlatPslParser-class.html#_parse_row">_parse_row</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlatPslParser._parse_row-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlatPslParser._parse_row-expanded"><a name="L411"></a><tt class="py-lineno">411</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a dictionary of parsed column values (PRIVATE)."""</tt> </tt>
<a name="L412"></a><tt class="py-lineno">412</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> </tt>
<a name="L413"></a><tt class="py-lineno">413</tt>  <tt class="py-line">        <tt class="py-name">cols</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-114" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-114', 'x', 'link-26');">x</a></tt> <tt class="py-keyword">for</tt> <tt id="link-115" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-115', 'x', 'link-26');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-116" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-116', 'strip', 'link-111');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-117" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-117', 'split', 'link-28');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">'\t'</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt id="link-118" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-118', 'x', 'link-26');">x</a></tt><tt class="py-op">]</tt> </tt>
<a name="L414"></a><tt class="py-lineno">414</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-119" class="py-name" targets="Method Bio.SearchIO.BlatIO.BlatPslParser._validate_cols()=Bio.SearchIO.BlatIO.BlatPslParser-class.html#_validate_cols"><a title="Bio.SearchIO.BlatIO.BlatPslParser._validate_cols" class="py-name" href="#" onclick="return doclink('link-119', '_validate_cols', 'link-119');">_validate_cols</a></tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">)</tt> </tt>
<a name="L415"></a><tt class="py-lineno">415</tt>  <tt class="py-line"> </tt>
<a name="L416"></a><tt class="py-lineno">416</tt>  <tt class="py-line">        <tt class="py-name">psl</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L417"></a><tt class="py-lineno">417</tt>  <tt class="py-line">        <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'qname'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">]</tt>                             <tt class="py-comment"># qName</tt> </tt>
<a name="L418"></a><tt class="py-lineno">418</tt>  <tt class="py-line">        <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'qsize'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-number">10</tt><tt class="py-op">]</tt><tt class="py-op">)</tt>                       <tt class="py-comment"># qSize</tt> </tt>
<a name="L419"></a><tt class="py-lineno">419</tt>  <tt class="py-line">        <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'tname'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-number">13</tt><tt class="py-op">]</tt>                            <tt class="py-comment"># tName</tt> </tt>
<a name="L420"></a><tt class="py-lineno">420</tt>  <tt class="py-line">        <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'tsize'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-number">14</tt><tt class="py-op">]</tt><tt class="py-op">)</tt>                       <tt class="py-comment"># tSize</tt> </tt>
<a name="L421"></a><tt class="py-lineno">421</tt>  <tt class="py-line">        <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'matches'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt>                      <tt class="py-comment"># matches</tt> </tt>
<a name="L422"></a><tt class="py-lineno">422</tt>  <tt class="py-line">        <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'mismatches'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt>                   <tt class="py-comment"># misMatches</tt> </tt>
<a name="L423"></a><tt class="py-lineno">423</tt>  <tt class="py-line">        <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'repmatches'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt>                   <tt class="py-comment"># repMatches</tt> </tt>
<a name="L424"></a><tt class="py-lineno">424</tt>  <tt class="py-line">        <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'ncount'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">)</tt>                       <tt class="py-comment"># nCount</tt> </tt>
<a name="L425"></a><tt class="py-lineno">425</tt>  <tt class="py-line">        <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'qnuminsert'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt><tt class="py-op">)</tt>                   <tt class="py-comment"># qNumInsert</tt> </tt>
<a name="L426"></a><tt class="py-lineno">426</tt>  <tt class="py-line">        <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'qbaseinsert'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-number">5</tt><tt class="py-op">]</tt><tt class="py-op">)</tt>                  <tt class="py-comment"># qBaseInsert</tt> </tt>
<a name="L427"></a><tt class="py-lineno">427</tt>  <tt class="py-line">        <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'tnuminsert'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">]</tt><tt class="py-op">)</tt>                   <tt class="py-comment"># tNumInsert</tt> </tt>
<a name="L428"></a><tt class="py-lineno">428</tt>  <tt class="py-line">        <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'tbaseinsert'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">]</tt><tt class="py-op">)</tt>                  <tt class="py-comment"># tBaseInsert</tt> </tt>
<a name="L429"></a><tt class="py-lineno">429</tt>  <tt class="py-line">        <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'strand'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-number">8</tt><tt class="py-op">]</tt>                            <tt class="py-comment"># strand</tt> </tt>
<a name="L430"></a><tt class="py-lineno">430</tt>  <tt class="py-line">        <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'qstart'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-number">11</tt><tt class="py-op">]</tt><tt class="py-op">)</tt>                      <tt class="py-comment"># qStart</tt> </tt>
<a name="L431"></a><tt class="py-lineno">431</tt>  <tt class="py-line">        <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'qend'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-number">12</tt><tt class="py-op">]</tt><tt class="py-op">)</tt>                        <tt class="py-comment"># qEnd</tt> </tt>
<a name="L432"></a><tt class="py-lineno">432</tt>  <tt class="py-line">        <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'tstart'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-number">15</tt><tt class="py-op">]</tt><tt class="py-op">)</tt>                      <tt class="py-comment"># tStart</tt> </tt>
<a name="L433"></a><tt class="py-lineno">433</tt>  <tt class="py-line">        <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'tend'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-number">16</tt><tt class="py-op">]</tt><tt class="py-op">)</tt>                        <tt class="py-comment"># tEnd</tt> </tt>
<a name="L434"></a><tt class="py-lineno">434</tt>  <tt class="py-line">        <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'blockcount'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-number">17</tt><tt class="py-op">]</tt><tt class="py-op">)</tt>                  <tt class="py-comment"># blockCount</tt> </tt>
<a name="L435"></a><tt class="py-lineno">435</tt>  <tt class="py-line">        <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'blocksizes'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-120" class="py-name" targets="Function Bio.SearchIO.BlatIO._list_from_csv()=Bio.SearchIO.BlatIO-module.html#_list_from_csv"><a title="Bio.SearchIO.BlatIO._list_from_csv" class="py-name" href="#" onclick="return doclink('link-120', '_list_from_csv', 'link-120');">_list_from_csv</a></tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-number">18</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt>  <tt class="py-comment"># blockSizes</tt> </tt>
<a name="L436"></a><tt class="py-lineno">436</tt>  <tt class="py-line">        <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'qstarts'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-121" class="py-name"><a title="Bio.SearchIO.BlatIO._list_from_csv" class="py-name" href="#" onclick="return doclink('link-121', '_list_from_csv', 'link-120');">_list_from_csv</a></tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-number">19</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt>     <tt class="py-comment"># qStarts</tt> </tt>
<a name="L437"></a><tt class="py-lineno">437</tt>  <tt class="py-line">        <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'tstarts'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-122" class="py-name"><a title="Bio.SearchIO.BlatIO._list_from_csv" class="py-name" href="#" onclick="return doclink('link-122', '_list_from_csv', 'link-120');">_list_from_csv</a></tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-number">20</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt>     <tt class="py-comment"># tStarts</tt> </tt>
<a name="L438"></a><tt class="py-lineno">438</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">pslx</tt><tt class="py-op">:</tt> </tt>
<a name="L439"></a><tt class="py-lineno">439</tt>  <tt class="py-line">            <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'qseqs'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-123" class="py-name"><a title="Bio.SearchIO.BlatIO._list_from_csv" class="py-name" href="#" onclick="return doclink('link-123', '_list_from_csv', 'link-120');">_list_from_csv</a></tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-number">21</tt><tt class="py-op">]</tt><tt class="py-op">)</tt>       <tt class="py-comment"># query sequence</tt> </tt>
<a name="L440"></a><tt class="py-lineno">440</tt>  <tt class="py-line">            <tt class="py-name">psl</tt><tt class="py-op">[</tt><tt class="py-string">'tseqs'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-124" class="py-name"><a title="Bio.SearchIO.BlatIO._list_from_csv" class="py-name" href="#" onclick="return doclink('link-124', '_list_from_csv', 'link-120');">_list_from_csv</a></tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-number">22</tt><tt class="py-op">]</tt><tt class="py-op">)</tt>       <tt class="py-comment"># hit sequence</tt> </tt>
<a name="L441"></a><tt class="py-lineno">441</tt>  <tt class="py-line"> </tt>
<a name="L442"></a><tt class="py-lineno">442</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">psl</tt> </tt>
</div><a name="L443"></a><tt class="py-lineno">443</tt>  <tt class="py-line"> </tt>
<a name="BlatPslParser._validate_cols"></a><div id="BlatPslParser._validate_cols-def"><a name="L444"></a><tt class="py-lineno">444</tt> <a class="py-toggle" href="#" id="BlatPslParser._validate_cols-toggle" onclick="return toggle('BlatPslParser._validate_cols');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlatIO.BlatPslParser-class.html#_validate_cols">_validate_cols</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">cols</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlatPslParser._validate_cols-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlatPslParser._validate_cols-expanded"><a name="L445"></a><tt class="py-lineno">445</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">pslx</tt><tt class="py-op">:</tt> </tt>
<a name="L446"></a><tt class="py-lineno">446</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">21</tt><tt class="py-op">,</tt> <tt class="py-string">"Invalid PSL line: %r. "</tt> \ </tt>
<a name="L447"></a><tt class="py-lineno">447</tt>  <tt class="py-line">            <tt class="py-string">"Expected 21 tab-separated columns, found %i"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L448"></a><tt class="py-lineno">448</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L449"></a><tt class="py-lineno">449</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">23</tt><tt class="py-op">,</tt> <tt class="py-string">"Invalid PSLX line: %r. "</tt> \ </tt>
<a name="L450"></a><tt class="py-lineno">450</tt>  <tt class="py-line">            <tt class="py-string">"Expected 23 tab-separated columns, found %i"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L451"></a><tt class="py-lineno">451</tt>  <tt class="py-line"> </tt>
<a name="BlatPslParser._parse_qresult"></a><div id="BlatPslParser._parse_qresult-def"><a name="L452"></a><tt class="py-lineno">452</tt> <a class="py-toggle" href="#" id="BlatPslParser._parse_qresult-toggle" onclick="return toggle('BlatPslParser._parse_qresult');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlatIO.BlatPslParser-class.html#_parse_qresult">_parse_qresult</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlatPslParser._parse_qresult-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlatPslParser._parse_qresult-expanded"><a name="L453"></a><tt class="py-lineno">453</tt>  <tt class="py-line">        <tt class="py-docstring">"""Yield QueryResult objects (PRIVATE)."""</tt> </tt>
<a name="L454"></a><tt class="py-lineno">454</tt>  <tt class="py-line">        <tt class="py-comment"># state values, determines what to do for each line</tt> </tt>
<a name="L455"></a><tt class="py-lineno">455</tt>  <tt class="py-line">        <tt class="py-name">state_EOF</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L456"></a><tt class="py-lineno">456</tt>  <tt class="py-line">        <tt class="py-name">state_QRES_NEW</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L457"></a><tt class="py-lineno">457</tt>  <tt class="py-line">        <tt class="py-name">state_QRES_SAME</tt> <tt class="py-op">=</tt> <tt class="py-number">3</tt> </tt>
<a name="L458"></a><tt class="py-lineno">458</tt>  <tt class="py-line">        <tt class="py-name">state_HIT_NEW</tt> <tt class="py-op">=</tt> <tt class="py-number">2</tt> </tt>
<a name="L459"></a><tt class="py-lineno">459</tt>  <tt class="py-line">        <tt class="py-name">state_HIT_SAME</tt> <tt class="py-op">=</tt> <tt class="py-number">4</tt> </tt>
<a name="L460"></a><tt class="py-lineno">460</tt>  <tt class="py-line">        <tt class="py-comment"># initial dummy values</tt> </tt>
<a name="L461"></a><tt class="py-lineno">461</tt>  <tt class="py-line">        <tt class="py-name">qres_state</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L462"></a><tt class="py-lineno">462</tt>  <tt class="py-line">        <tt class="py-name">file_state</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L463"></a><tt class="py-lineno">463</tt>  <tt class="py-line">        <tt class="py-name">cur_qid</tt><tt class="py-op">,</tt> <tt class="py-name">cur_hid</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt> </tt>
<a name="L464"></a><tt class="py-lineno">464</tt>  <tt class="py-line">        <tt class="py-name">prev_qid</tt><tt class="py-op">,</tt> <tt class="py-name">prev_hid</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt> </tt>
<a name="L465"></a><tt class="py-lineno">465</tt>  <tt class="py-line">        <tt class="py-name">cur</tt><tt class="py-op">,</tt> <tt class="py-name">prev</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt> </tt>
<a name="L466"></a><tt class="py-lineno">466</tt>  <tt class="py-line">        <tt class="py-name">hit_list</tt><tt class="py-op">,</tt> <tt class="py-name">hsp_list</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L467"></a><tt class="py-lineno">467</tt>  <tt class="py-line"> </tt>
<a name="L468"></a><tt class="py-lineno">468</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L469"></a><tt class="py-lineno">469</tt>  <tt class="py-line">            <tt class="py-comment"># store previous line's parsed values for all lines after the first</tt> </tt>
<a name="L470"></a><tt class="py-lineno">470</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">cur</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L471"></a><tt class="py-lineno">471</tt>  <tt class="py-line">                <tt class="py-name">prev</tt> <tt class="py-op">=</tt> <tt class="py-name">cur</tt> </tt>
<a name="L472"></a><tt class="py-lineno">472</tt>  <tt class="py-line">                <tt class="py-name">prev_qid</tt> <tt class="py-op">=</tt> <tt class="py-name">cur_qid</tt> </tt>
<a name="L473"></a><tt class="py-lineno">473</tt>  <tt class="py-line">                <tt class="py-name">prev_hid</tt> <tt class="py-op">=</tt> <tt class="py-name">cur_hid</tt> </tt>
<a name="L474"></a><tt class="py-lineno">474</tt>  <tt class="py-line">            <tt class="py-comment"># only parse the result row if it's not EOF</tt> </tt>
<a name="L475"></a><tt class="py-lineno">475</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L476"></a><tt class="py-lineno">476</tt>  <tt class="py-line">                <tt class="py-name">cur</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-125" class="py-name" targets="Method Bio.SearchIO.BlatIO.BlatPslParser._parse_row()=Bio.SearchIO.BlatIO.BlatPslParser-class.html#_parse_row,Method Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabParser._parse_row()=Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabParser-class.html#_parse_row,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser._parse_row()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser-class.html#_parse_row"><a title="Bio.SearchIO.BlatIO.BlatPslParser._parse_row
Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabParser._parse_row
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser._parse_row" class="py-name" href="#" onclick="return doclink('link-125', '_parse_row', 'link-125');">_parse_row</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L477"></a><tt class="py-lineno">477</tt>  <tt class="py-line">                <tt class="py-name">cur_qid</tt> <tt class="py-op">=</tt> <tt class="py-name">cur</tt><tt class="py-op">[</tt><tt class="py-string">'qname'</tt><tt class="py-op">]</tt> </tt>
<a name="L478"></a><tt class="py-lineno">478</tt>  <tt class="py-line">                <tt class="py-name">cur_hid</tt> <tt class="py-op">=</tt> <tt class="py-name">cur</tt><tt class="py-op">[</tt><tt class="py-string">'tname'</tt><tt class="py-op">]</tt> </tt>
<a name="L479"></a><tt class="py-lineno">479</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L480"></a><tt class="py-lineno">480</tt>  <tt class="py-line">                <tt class="py-name">file_state</tt> <tt class="py-op">=</tt> <tt class="py-name">state_EOF</tt> </tt>
<a name="L481"></a><tt class="py-lineno">481</tt>  <tt class="py-line">                <tt class="py-comment"># mock values, since we have nothing to parse</tt> </tt>
<a name="L482"></a><tt class="py-lineno">482</tt>  <tt class="py-line">                <tt class="py-name">cur_qid</tt><tt class="py-op">,</tt> <tt class="py-name">cur_hid</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt> </tt>
<a name="L483"></a><tt class="py-lineno">483</tt>  <tt class="py-line"> </tt>
<a name="L484"></a><tt class="py-lineno">484</tt>  <tt class="py-line">            <tt class="py-comment"># get the state of hit and qresult</tt> </tt>
<a name="L485"></a><tt class="py-lineno">485</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">prev_qid</tt> <tt class="py-op">!=</tt> <tt class="py-name">cur_qid</tt><tt class="py-op">:</tt> </tt>
<a name="L486"></a><tt class="py-lineno">486</tt>  <tt class="py-line">                <tt class="py-name">qres_state</tt> <tt class="py-op">=</tt> <tt class="py-name">state_QRES_NEW</tt> </tt>
<a name="L487"></a><tt class="py-lineno">487</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L488"></a><tt class="py-lineno">488</tt>  <tt class="py-line">                <tt class="py-name">qres_state</tt> <tt class="py-op">=</tt> <tt class="py-name">state_QRES_SAME</tt> </tt>
<a name="L489"></a><tt class="py-lineno">489</tt>  <tt class="py-line">            <tt class="py-comment"># new hits are hits with different ids or hits in a new qresult</tt> </tt>
<a name="L490"></a><tt class="py-lineno">490</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">prev_hid</tt> <tt class="py-op">!=</tt> <tt class="py-name">cur_hid</tt> <tt class="py-keyword">or</tt> <tt class="py-name">qres_state</tt> <tt class="py-op">==</tt> <tt class="py-name">state_QRES_NEW</tt><tt class="py-op">:</tt> </tt>
<a name="L491"></a><tt class="py-lineno">491</tt>  <tt class="py-line">                <tt class="py-name">hit_state</tt> <tt class="py-op">=</tt> <tt class="py-name">state_HIT_NEW</tt> </tt>
<a name="L492"></a><tt class="py-lineno">492</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L493"></a><tt class="py-lineno">493</tt>  <tt class="py-line">                <tt class="py-name">hit_state</tt> <tt class="py-op">=</tt> <tt class="py-name">state_HIT_SAME</tt> </tt>
<a name="L494"></a><tt class="py-lineno">494</tt>  <tt class="py-line"> </tt>
<a name="L495"></a><tt class="py-lineno">495</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">prev</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L496"></a><tt class="py-lineno">496</tt>  <tt class="py-line">                <tt class="py-comment"># create fragment and HSP and set their attributes</tt> </tt>
<a name="L497"></a><tt class="py-lineno">497</tt>  <tt class="py-line">                <tt id="link-126" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-126', 'hsp', 'link-75');">hsp</a></tt> <tt class="py-op">=</tt> <tt id="link-127" class="py-name" targets="Function Bio.SearchIO.BlatIO._create_hsp()=Bio.SearchIO.BlatIO-module.html#_create_hsp,Function Bio.SearchIO.ExonerateIO._base._create_hsp()=Bio.SearchIO.ExonerateIO._base-module.html#_create_hsp"><a title="Bio.SearchIO.BlatIO._create_hsp
Bio.SearchIO.ExonerateIO._base._create_hsp" class="py-name" href="#" onclick="return doclink('link-127', '_create_hsp', 'link-127');">_create_hsp</a></tt><tt class="py-op">(</tt><tt class="py-name">prev_hid</tt><tt class="py-op">,</tt> <tt class="py-name">prev_qid</tt><tt class="py-op">,</tt> <tt class="py-name">prev</tt><tt class="py-op">)</tt> </tt>
<a name="L498"></a><tt class="py-lineno">498</tt>  <tt class="py-line">                <tt class="py-name">hsp_list</tt><tt class="py-op">.</tt><tt id="link-128" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-128', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt id="link-129" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-129', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">)</tt> </tt>
<a name="L499"></a><tt class="py-lineno">499</tt>  <tt class="py-line"> </tt>
<a name="L500"></a><tt class="py-lineno">500</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">hit_state</tt> <tt class="py-op">==</tt> <tt class="py-name">state_HIT_NEW</tt><tt class="py-op">:</tt> </tt>
<a name="L501"></a><tt class="py-lineno">501</tt>  <tt class="py-line">                    <tt class="py-comment"># create Hit and set its attributes</tt> </tt>
<a name="L502"></a><tt class="py-lineno">502</tt>  <tt class="py-line">                    <tt id="link-130" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-130', 'hit', 'link-62');">hit</a></tt> <tt class="py-op">=</tt> <tt id="link-131" class="py-name"><a title="Bio.SearchIO._model.hit.Hit" class="py-name" href="#" onclick="return doclink('link-131', 'Hit', 'link-17');">Hit</a></tt><tt class="py-op">(</tt><tt class="py-name">hsp_list</tt><tt class="py-op">)</tt> </tt>
<a name="L503"></a><tt class="py-lineno">503</tt>  <tt class="py-line">                    <tt id="link-132" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-132', 'hit', 'link-62');">hit</a></tt><tt class="py-op">.</tt><tt class="py-name">seq_len</tt> <tt class="py-op">=</tt> <tt class="py-name">prev</tt><tt class="py-op">[</tt><tt class="py-string">'tsize'</tt><tt class="py-op">]</tt> </tt>
<a name="L504"></a><tt class="py-lineno">504</tt>  <tt class="py-line">                    <tt class="py-name">hit_list</tt><tt class="py-op">.</tt><tt id="link-133" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-133', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt id="link-134" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-134', 'hit', 'link-62');">hit</a></tt><tt class="py-op">)</tt> </tt>
<a name="L505"></a><tt class="py-lineno">505</tt>  <tt class="py-line">                    <tt class="py-name">hsp_list</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L506"></a><tt class="py-lineno">506</tt>  <tt class="py-line"> </tt>
<a name="L507"></a><tt class="py-lineno">507</tt>  <tt class="py-line">                <tt class="py-comment"># create qresult and yield if we're at a new qresult or at EOF</tt> </tt>
<a name="L508"></a><tt class="py-lineno">508</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">qres_state</tt> <tt class="py-op">==</tt> <tt class="py-name">state_QRES_NEW</tt> <tt class="py-keyword">or</tt> <tt class="py-name">file_state</tt> <tt class="py-op">==</tt> <tt class="py-name">state_EOF</tt><tt class="py-op">:</tt> </tt>
<a name="L509"></a><tt class="py-lineno">509</tt>  <tt class="py-line">                    <tt class="py-name">qresult</tt> <tt class="py-op">=</tt> <tt id="link-135" class="py-name"><a title="Bio.SearchIO._model.query.QueryResult" class="py-name" href="#" onclick="return doclink('link-135', 'QueryResult', 'link-16');">QueryResult</a></tt><tt class="py-op">(</tt><tt id="link-136" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.id=Bio.KEGG.KGML.KGML_pathway.Component-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Entry.id=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.id=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#id,Variable Bio.PDB.Entity.Entity.id=Bio.PDB.Entity.Entity-class.html#id,Method Bio.Phylo.PhyloXMLIO.Parser.id()=Bio.Phylo.PhyloXMLIO.Parser-class.html#id,Method Bio.Phylo.PhyloXMLIO.Writer.id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#id,Variable Bio.SearchIO._model.hit.Hit.id=Bio.SearchIO._model.hit.Hit-class.html#id,Variable Bio.SearchIO._model.query.QueryResult.id=Bio.SearchIO._model.query.QueryResult-class.html#id"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-136', 'id', 'link-136');">id</a></tt><tt class="py-op">=</tt><tt class="py-name">prev_qid</tt><tt class="py-op">)</tt> </tt>
<a name="L510"></a><tt class="py-lineno">510</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt id="link-137" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-137', 'hit', 'link-62');">hit</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">hit_list</tt><tt class="py-op">:</tt> </tt>
<a name="L511"></a><tt class="py-lineno">511</tt>  <tt class="py-line">                        <tt class="py-name">qresult</tt><tt class="py-op">.</tt><tt id="link-138" class="py-name" targets="Method Bio.SearchIO._model.query.QueryResult.absorb()=Bio.SearchIO._model.query.QueryResult-class.html#absorb"><a title="Bio.SearchIO._model.query.QueryResult.absorb" class="py-name" href="#" onclick="return doclink('link-138', 'absorb', 'link-138');">absorb</a></tt><tt class="py-op">(</tt><tt id="link-139" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-139', 'hit', 'link-62');">hit</a></tt><tt class="py-op">)</tt> </tt>
<a name="L512"></a><tt class="py-lineno">512</tt>  <tt class="py-line">                    <tt class="py-name">qresult</tt><tt class="py-op">.</tt><tt class="py-name">seq_len</tt> <tt class="py-op">=</tt> <tt class="py-name">prev</tt><tt class="py-op">[</tt><tt class="py-string">'qsize'</tt><tt class="py-op">]</tt> </tt>
<a name="L513"></a><tt class="py-lineno">513</tt>  <tt class="py-line">                    <tt class="py-keyword">yield</tt> <tt class="py-name">qresult</tt> </tt>
<a name="L514"></a><tt class="py-lineno">514</tt>  <tt class="py-line">                    <tt class="py-comment"># if we're at EOF, break</tt> </tt>
<a name="L515"></a><tt class="py-lineno">515</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">file_state</tt> <tt class="py-op">==</tt> <tt class="py-name">state_EOF</tt><tt class="py-op">:</tt> </tt>
<a name="L516"></a><tt class="py-lineno">516</tt>  <tt class="py-line">                        <tt class="py-keyword">break</tt> </tt>
<a name="L517"></a><tt class="py-lineno">517</tt>  <tt class="py-line">                    <tt class="py-name">hit_list</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L518"></a><tt class="py-lineno">518</tt>  <tt class="py-line"> </tt>
<a name="L519"></a><tt class="py-lineno">519</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-140" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-140', 'readline', 'link-108');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L520"></a><tt class="py-lineno">520</tt>  <tt class="py-line"> </tt>
<a name="L521"></a><tt class="py-lineno">521</tt>  <tt class="py-line"> </tt>
<a name="BlatPslIndexer"></a><div id="BlatPslIndexer-def"><a name="L522"></a><tt class="py-lineno">522</tt> <a class="py-toggle" href="#" id="BlatPslIndexer-toggle" onclick="return toggle('BlatPslIndexer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.SearchIO.BlatIO.BlatPslIndexer-class.html">BlatPslIndexer</a><tt class="py-op">(</tt><tt class="py-base-class">SearchIndexer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlatPslIndexer-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="BlatPslIndexer-expanded"><a name="L523"></a><tt class="py-lineno">523</tt>  <tt class="py-line">    <tt class="py-docstring">"""Indexer class for BLAT PSL output."""</tt> </tt>
<a name="L524"></a><tt class="py-lineno">524</tt>  <tt class="py-line"> </tt>
<a name="L525"></a><tt class="py-lineno">525</tt>  <tt class="py-line">    <tt id="link-141" class="py-name" targets="Variable Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer._parser=Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer-class.html#_parser"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer._parser" class="py-name" href="#" onclick="return doclink('link-141', '_parser', 'link-141');">_parser</a></tt> <tt class="py-op">=</tt> <tt id="link-142" class="py-name" targets="Class Bio.SearchIO.BlatIO.BlatPslParser=Bio.SearchIO.BlatIO.BlatPslParser-class.html"><a title="Bio.SearchIO.BlatIO.BlatPslParser" class="py-name" href="#" onclick="return doclink('link-142', 'BlatPslParser', 'link-142');">BlatPslParser</a></tt> </tt>
<a name="L526"></a><tt class="py-lineno">526</tt>  <tt class="py-line"> </tt>
<a name="BlatPslIndexer.__init__"></a><div id="BlatPslIndexer.__init__-def"><a name="L527"></a><tt class="py-lineno">527</tt> <a class="py-toggle" href="#" id="BlatPslIndexer.__init__-toggle" onclick="return toggle('BlatPslIndexer.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlatIO.BlatPslIndexer-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">filename</tt><tt class="py-op">,</tt> <tt class="py-param">pslx</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlatPslIndexer.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlatPslIndexer.__init__-expanded"><a name="L528"></a><tt class="py-lineno">528</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the class."""</tt> </tt>
<a name="L529"></a><tt class="py-lineno">529</tt>  <tt class="py-line">        <tt id="link-143" class="py-name"><a title="Bio.SearchIO._index.SearchIndexer" class="py-name" href="#" onclick="return doclink('link-143', 'SearchIndexer', 'link-12');">SearchIndexer</a></tt><tt class="py-op">.</tt><tt id="link-144" class="py-name" targets="Method Bio.Affy.CelFile.ParserError.__init__()=Bio.Affy.CelFile.ParserError-class.html#__init__,Method Bio.Affy.CelFile.Record.__init__()=Bio.Affy.CelFile.Record-class.html#__init__,Method Bio.Align.AlignInfo.PSSM.__init__()=Bio.Align.AlignInfo.PSSM-class.html#__init__,Method Bio.Align.AlignInfo.SummaryInfo.__init__()=Bio.Align.AlignInfo.SummaryInfo-class.html#__init__,Method Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline.__init__()=Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline-class.html#__init__,Method Bio.Align.Applications._Clustalw.ClustalwCommandline.__init__()=Bio.Align.Applications._Clustalw.ClustalwCommandline-class.html#__init__,Method Bio.Align.Applications._Dialign.DialignCommandline.__init__()=Bio.Align.Applications._Dialign.DialignCommandline-class.html#__init__,Method Bio.Align.Applications._MSAProbs.MSAProbsCommandline.__init__()=Bio.Align.Applications._MSAProbs.MSAProbsCommandline-class.html#__init__,Method Bio.Align.Applications._Mafft.MafftCommandline.__init__()=Bio.Align.Applications._Mafft.MafftCommandline-class.html#__init__,Method Bio.Align.Applications._Muscle.MuscleCommandline.__init__()=Bio.Align.Applications._Muscle.MuscleCommandline-class.html#__init__,Method Bio.Align.Applications._Prank.PrankCommandline.__init__()=Bio.Align.Applications._Prank.PrankCommandline-class.html#__init__,Method Bio.Align.Applications._Probcons.ProbconsCommandline.__init__()=Bio.Align.Applications._Probcons.ProbconsCommandline-class.html#__init__,Method Bio.Align.Applications._TCoffee.TCoffeeCommandline.__init__()=Bio.Align.Applications._TCoffee.TCoffeeCommandline-class.html#__init__,Method Bio.Align.MultipleSeqAlignment.__init__()=Bio.Align.MultipleSeqAlignment-class.html#__init__,Method Bio.AlignIO.Interfaces.AlignmentIterator.__init__()=Bio.AlignIO.Interfaces.AlignmentIterator-class.html#__init__,Method Bio.AlignIO.Interfaces.AlignmentWriter.__init__()=Bio.AlignIO.Interfaces.AlignmentWriter-class.html#__init__,Method Bio.AlignIO.Interfaces.SequentialAlignmentWriter.__init__()=Bio.AlignIO.Interfaces.SequentialAlignmentWriter-class.html#__init__,Method Bio.AlignIO.MafIO.MafIndex.__init__()=Bio.AlignIO.MafIO.MafIndex-class.html#__init__,Method Bio.AlignIO.MauveIO.MauveWriter.__init__()=Bio.AlignIO.MauveIO.MauveWriter-class.html#__init__,Method Bio.Alphabet.AlphabetEncoder.__init__()=Bio.Alphabet.AlphabetEncoder-class.html#__init__,Method Bio.Alphabet.Gapped.__init__()=Bio.Alphabet.Gapped-class.html#__init__,Method Bio.Alphabet.HasStopCodon.__init__()=Bio.Alphabet.HasStopCodon-class.html#__init__,Method Bio.Application.AbstractCommandline.__init__()=Bio.Application.AbstractCommandline-class.html#__init__,Method Bio.Application.ApplicationError.__init__()=Bio.Application.ApplicationError-class.html#__init__,Method Bio.Application._AbstractParameter.__init__()=Bio.Application._AbstractParameter-class.html#__init__,Method Bio.Application._Argument.__init__()=Bio.Application._Argument-class.html#__init__,Method Bio.Application._Option.__init__()=Bio.Application._Option-class.html#__init__,Method Bio.Application._StaticArgument.__init__()=Bio.Application._StaticArgument-class.html#__init__,Method Bio.Application._Switch.__init__()=Bio.Application._Switch-class.html#__init__,Method Bio.Blast.Applications.NcbiblastformatterCommandline.__init__()=Bio.Blast.Applications.NcbiblastformatterCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastnCommandline.__init__()=Bio.Blast.Applications.NcbiblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastpCommandline.__init__()=Bio.Blast.Applications.NcbiblastpCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastxCommandline.__init__()=Bio.Blast.Applications.NcbiblastxCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbideltablastCommandline.__init__()=Bio.Blast.Applications.NcbideltablastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbipsiblastCommandline.__init__()=Bio.Blast.Applications.NcbipsiblastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbirpsblastCommandline.__init__()=Bio.Blast.Applications.NcbirpsblastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbirpstblastnCommandline.__init__()=Bio.Blast.Applications.NcbirpstblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbitblastnCommandline.__init__()=Bio.Blast.Applications.NcbitblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbitblastxCommandline.__init__()=Bio.Blast.Applications.NcbitblastxCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbibaseblastCommandline.__init__()=Bio.Blast.Applications._NcbibaseblastCommandline-class.html#__init__,Method Bio.Blast.Applications._Ncbiblast2SeqCommandline.__init__()=Bio.Blast.Applications._Ncbiblast2SeqCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbiblastCommandline.__init__()=Bio.Blast.Applications._NcbiblastCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbiblastMain2SeqCommandline.__init__()=Bio.Blast.Applications._NcbiblastMain2SeqCommandline-class.html#__init__,Method Bio.Blast.NCBIStandalone.BlastErrorParser.__init__()=Bio.Blast.NCBIStandalone.BlastErrorParser-class.html#__init__,Method Bio.Blast.NCBIStandalone.BlastParser.__init__()=Bio.Blast.NCBIStandalone.BlastParser-class.html#__init__,Method Bio.Blast.NCBIStandalone.Iterator.__init__()=Bio.Blast.NCBIStandalone.Iterator-class.html#__init__,Method Bio.Blast.NCBIStandalone.PSIBlastParser.__init__()=Bio.Blast.NCBIStandalone.PSIBlastParser-class.html#__init__,Method Bio.Blast.NCBIStandalone._BlastConsumer.__init__()=Bio.Blast.NCBIStandalone._BlastConsumer-class.html#__init__,Method Bio.Blast.NCBIStandalone._BlastErrorConsumer.__init__()=Bio.Blast.NCBIStandalone._BlastErrorConsumer-class.html#__init__,Method Bio.Blast.NCBIStandalone._PSIBlastConsumer.__init__()=Bio.Blast.NCBIStandalone._PSIBlastConsumer-class.html#__init__,Method Bio.Blast.NCBIXML.BlastParser.__init__()=Bio.Blast.NCBIXML.BlastParser-class.html#__init__,Method Bio.Blast.NCBIXML._XMLparser.__init__()=Bio.Blast.NCBIXML._XMLparser-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableEntry.__init__()=Bio.Blast.ParseBlastTable.BlastTableEntry-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableReader.__init__()=Bio.Blast.ParseBlastTable.BlastTableReader-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableRec.__init__()=Bio.Blast.ParseBlastTable.BlastTableRec-class.html#__init__,Method Bio.Blast.Record.Alignment.__init__()=Bio.Blast.Record.Alignment-class.html#__init__,Method Bio.Blast.Record.Blast.__init__()=Bio.Blast.Record.Blast-class.html#__init__,Method Bio.Blast.Record.DatabaseReport.__init__()=Bio.Blast.Record.DatabaseReport-class.html#__init__,Method Bio.Blast.Record.Description.__init__()=Bio.Blast.Record.Description-class.html#__init__,Method Bio.Blast.Record.HSP.__init__()=Bio.Blast.Record.HSP-class.html#__init__,Method Bio.Blast.Record.Header.__init__()=Bio.Blast.Record.Header-class.html#__init__,Method Bio.Blast.Record.MultipleAlignment.__init__()=Bio.Blast.Record.MultipleAlignment-class.html#__init__,Method Bio.Blast.Record.PSIBlast.__init__()=Bio.Blast.Record.PSIBlast-class.html#__init__,Method Bio.Blast.Record.Parameters.__init__()=Bio.Blast.Record.Parameters-class.html#__init__,Method Bio.Blast.Record.Round.__init__()=Bio.Blast.Record.Round-class.html#__init__,Method Bio.CAPS.CAPSMap.__init__()=Bio.CAPS.CAPSMap-class.html#__init__,Method Bio.CAPS.DifferentialCutsite.__init__()=Bio.CAPS.DifferentialCutsite-class.html#__init__,Method Bio.Cluster.Record.__init__()=Bio.Cluster.Record-class.html#__init__,Method Bio.Compass.Record.__init__()=Bio.Compass.Record-class.html#__init__,Method Bio.Crystal.Chain.__init__()=Bio.Crystal.Chain-class.html#__init__,Method Bio.Crystal.Crystal.__init__()=Bio.Crystal.Crystal-class.html#__init__,Method Bio.Crystal.Hetero.__init__()=Bio.Crystal.Hetero-class.html#__init__,Method Bio.Data.CodonTable.AmbiguousCodonTable.__init__()=Bio.Data.CodonTable.AmbiguousCodonTable-class.html#__init__,Method Bio.Data.CodonTable.AmbiguousForwardTable.__init__()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#__init__,Method Bio.Data.CodonTable.CodonTable.__init__()=Bio.Data.CodonTable.CodonTable-class.html#__init__,Method Bio.Data.CodonTable.NCBICodonTable.__init__()=Bio.Data.CodonTable.NCBICodonTable-class.html#__init__,Method Bio.DocSQL.Create.__init__()=Bio.DocSQL.Create-class.html#__init__,Method Bio.DocSQL.Insert.__init__()=Bio.DocSQL.Insert-class.html#__init__,Method Bio.DocSQL.IterationCursor.__init__()=Bio.DocSQL.IterationCursor-class.html#__init__,Method Bio.DocSQL.Query.__init__()=Bio.DocSQL.Query-class.html#__init__,Method Bio.DocSQL.QueryAll.__init__()=Bio.DocSQL.QueryAll-class.html#__init__,Method Bio.DocSQL.QueryGeneric.__init__()=Bio.DocSQL.QueryGeneric-class.html#__init__,Method Bio.DocSQL.QueryRow.__init__()=Bio.DocSQL.QueryRow-class.html#__init__,Method Bio.DocSQL.QuerySingle.__init__()=Bio.DocSQL.QuerySingle-class.html#__init__,Method Bio.Emboss.Applications.DiffseqCommandline.__init__()=Bio.Emboss.Applications.DiffseqCommandline-class.html#__init__,Method Bio.Emboss.Applications.EInvertedCommandline.__init__()=Bio.Emboss.Applications.EInvertedCommandline-class.html#__init__,Method Bio.Emboss.Applications.ETandemCommandline.__init__()=Bio.Emboss.Applications.ETandemCommandline-class.html#__init__,Method Bio.Emboss.Applications.Est2GenomeCommandline.__init__()=Bio.Emboss.Applications.Est2GenomeCommandline-class.html#__init__,Method Bio.Emboss.Applications.FConsenseCommandline.__init__()=Bio.Emboss.Applications.FConsenseCommandline-class.html#__init__,Method Bio.Emboss.Applications.FDNADistCommandline.__init__()=Bio.Emboss.Applications.FDNADistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FDNAParsCommandline.__init__()=Bio.Emboss.Applications.FDNAParsCommandline-class.html#__init__,Method Bio.Emboss.Applications.FNeighborCommandline.__init__()=Bio.Emboss.Applications.FNeighborCommandline-class.html#__init__,Method Bio.Emboss.Applications.FProtDistCommandline.__init__()=Bio.Emboss.Applications.FProtDistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FProtParsCommandline.__init__()=Bio.Emboss.Applications.FProtParsCommandline-class.html#__init__,Method Bio.Emboss.Applications.FSeqBootCommandline.__init__()=Bio.Emboss.Applications.FSeqBootCommandline-class.html#__init__,Method Bio.Emboss.Applications.FTreeDistCommandline.__init__()=Bio.Emboss.Applications.FTreeDistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FuzznucCommandline.__init__()=Bio.Emboss.Applications.FuzznucCommandline-class.html#__init__,Method Bio.Emboss.Applications.FuzzproCommandline.__init__()=Bio.Emboss.Applications.FuzzproCommandline-class.html#__init__,Method Bio.Emboss.Applications.IepCommandline.__init__()=Bio.Emboss.Applications.IepCommandline-class.html#__init__,Method Bio.Emboss.Applications.NeedleCommandline.__init__()=Bio.Emboss.Applications.NeedleCommandline-class.html#__init__,Method Bio.Emboss.Applications.NeedleallCommandline.__init__()=Bio.Emboss.Applications.NeedleallCommandline-class.html#__init__,Method Bio.Emboss.Applications.PalindromeCommandline.__init__()=Bio.Emboss.Applications.PalindromeCommandline-class.html#__init__,Method Bio.Emboss.Applications.Primer3Commandline.__init__()=Bio.Emboss.Applications.Primer3Commandline-class.html#__init__,Method Bio.Emboss.Applications.PrimerSearchCommandline.__init__()=Bio.Emboss.Applications.PrimerSearchCommandline-class.html#__init__,Method Bio.Emboss.Applications.SeqmatchallCommandline.__init__()=Bio.Emboss.Applications.SeqmatchallCommandline-class.html#__init__,Method Bio.Emboss.Applications.SeqretCommandline.__init__()=Bio.Emboss.Applications.SeqretCommandline-class.html#__init__,Method Bio.Emboss.Applications.StretcherCommandline.__init__()=Bio.Emboss.Applications.StretcherCommandline-class.html#__init__,Method Bio.Emboss.Applications.TranalignCommandline.__init__()=Bio.Emboss.Applications.TranalignCommandline-class.html#__init__,Method Bio.Emboss.Applications.WaterCommandline.__init__()=Bio.Emboss.Applications.WaterCommandline-class.html#__init__,Method Bio.Emboss.Applications._EmbossCommandLine.__init__()=Bio.Emboss.Applications._EmbossCommandLine-class.html#__init__,Method Bio.Emboss.Applications._EmbossMinimalCommandLine.__init__()=Bio.Emboss.Applications._EmbossMinimalCommandLine-class.html#__init__,Method Bio.Emboss.Primer3.Primers.__init__()=Bio.Emboss.Primer3.Primers-class.html#__init__,Method Bio.Emboss.Primer3.Record.__init__()=Bio.Emboss.Primer3.Record-class.html#__init__,Method Bio.Emboss.PrimerSearch.Amplifier.__init__()=Bio.Emboss.PrimerSearch.Amplifier-class.html#__init__,Method Bio.Emboss.PrimerSearch.InputRecord.__init__()=Bio.Emboss.PrimerSearch.InputRecord-class.html#__init__,Method Bio.Emboss.PrimerSearch.OutputRecord.__init__()=Bio.Emboss.PrimerSearch.OutputRecord-class.html#__init__,Method Bio.Entrez.Parser.CorruptedXMLError.__init__()=Bio.Entrez.Parser.CorruptedXMLError-class.html#__init__,Method Bio.Entrez.Parser.DataHandler.__init__()=Bio.Entrez.Parser.DataHandler-class.html#__init__,Method Bio.Entrez.Parser.NotXMLError.__init__()=Bio.Entrez.Parser.NotXMLError-class.html#__init__,Method Bio.Entrez.Parser.StructureElement.__init__()=Bio.Entrez.Parser.StructureElement-class.html#__init__,Method Bio.Entrez.Parser.ValidationError.__init__()=Bio.Entrez.Parser.ValidationError-class.html#__init__,Method Bio.ExPASy.Enzyme.Record.__init__()=Bio.ExPASy.Enzyme.Record-class.html#__init__,Method Bio.ExPASy.Prodoc.Record.__init__()=Bio.ExPASy.Prodoc.Record-class.html#__init__,Method Bio.ExPASy.Prodoc.Reference.__init__()=Bio.ExPASy.Prodoc.Reference-class.html#__init__,Method Bio.ExPASy.Prosite.Record.__init__()=Bio.ExPASy.Prosite.Record-class.html#__init__,Method Bio.ExPASy.ScanProsite.ContentHandler.__init__()=Bio.ExPASy.ScanProsite.ContentHandler-class.html#__init__,Method Bio.ExPASy.ScanProsite.Parser.__init__()=Bio.ExPASy.ScanProsite.Parser-class.html#__init__,Method Bio.ExPASy.ScanProsite.Record.__init__()=Bio.ExPASy.ScanProsite.Record-class.html#__init__,Method Bio.ExPASy.cellosaurus.Record.__init__()=Bio.ExPASy.cellosaurus.Record-class.html#__init__,Method Bio.FSSP.FSSPAlignDict.__init__()=Bio.FSSP.FSSPAlignDict-class.html#__init__,Method Bio.FSSP.FSSPAlignRec.__init__()=Bio.FSSP.FSSPAlignRec-class.html#__init__,Method Bio.FSSP.FSSPHeader.__init__()=Bio.FSSP.FSSPHeader-class.html#__init__,Method Bio.FSSP.FSSPSumRec.__init__()=Bio.FSSP.FSSPSumRec-class.html#__init__,Method Bio.FSSP.FSSPTools.FSSPMultAlign.__init__()=Bio.FSSP.FSSPTools.FSSPMultAlign-class.html#__init__,Method Bio.FSSP.PosAlign.__init__()=Bio.FSSP.PosAlign-class.html#__init__,Method Bio.FSSP.fssp_rec.fff_rec.__init__()=Bio.FSSP.fssp_rec.fff_rec-class.html#__init__,Method Bio.File.UndoHandle.__init__()=Bio.File.UndoHandle-class.html#__init__,Method Bio.File._IndexedSeqFileDict.__init__()=Bio.File._IndexedSeqFileDict-class.html#__init__,Method Bio.File._SQLiteManySeqFilesDict.__init__()=Bio.File._SQLiteManySeqFilesDict-class.html#__init__,Method Bio.GA.Crossover.General.SafeFitnessCrossover.__init__()=Bio.GA.Crossover.General.SafeFitnessCrossover-class.html#__init__,Method Bio.GA.Crossover.GeneralPoint.GeneralPointCrossover.__init__()=Bio.GA.Crossover.GeneralPoint.GeneralPointCrossover-class.html#__init__,Method Bio.GA.Crossover.GeneralPoint.InterleaveCrossover.__init__()=Bio.GA.Crossover.GeneralPoint.InterleaveCrossover-class.html#__init__,Method Bio.GA.Crossover.Point.SinglePointCrossover.__init__()=Bio.GA.Crossover.Point.SinglePointCrossover-class.html#__init__,Method Bio.GA.Crossover.TwoPoint.TwoPointCrossover.__init__()=Bio.GA.Crossover.TwoPoint.TwoPointCrossover-class.html#__init__,Method Bio.GA.Crossover.Uniform.UniformCrossover.__init__()=Bio.GA.Crossover.Uniform.UniformCrossover-class.html#__init__,Method Bio.GA.Evolver.GenerationEvolver.__init__()=Bio.GA.Evolver.GenerationEvolver-class.html#__init__,Method Bio.GA.Evolver.SteadyStateEvolver.__init__()=Bio.GA.Evolver.SteadyStateEvolver-class.html#__init__,Method Bio.GA.Mutation.General.SafeFitnessMutation.__init__()=Bio.GA.Mutation.General.SafeFitnessMutation-class.html#__init__,Method Bio.GA.Mutation.Simple.ConversionMutation.__init__()=Bio.GA.Mutation.Simple.ConversionMutation-class.html#__init__,Method Bio.GA.Mutation.Simple.SinglePositionMutation.__init__()=Bio.GA.Mutation.Simple.SinglePositionMutation-class.html#__init__,Method Bio.GA.Organism.Organism.__init__()=Bio.GA.Organism.Organism-class.html#__init__,Method Bio.GA.Repair.Stabilizing.AmbiguousRepair.__init__()=Bio.GA.Repair.Stabilizing.AmbiguousRepair-class.html#__init__,Method Bio.GA.Selection.Abstract.AbstractSelection.__init__()=Bio.GA.Selection.Abstract.AbstractSelection-class.html#__init__,Method Bio.GA.Selection.Diversity.DiversitySelection.__init__()=Bio.GA.Selection.Diversity.DiversitySelection-class.html#__init__,Method Bio.GA.Selection.RouletteWheel.RouletteWheelSelection.__init__()=Bio.GA.Selection.RouletteWheel.RouletteWheelSelection-class.html#__init__,Method Bio.GA.Selection.Tournament.TournamentSelection.__init__()=Bio.GA.Selection.Tournament.TournamentSelection-class.html#__init__,Method Bio.GenBank.FeatureParser.__init__()=Bio.GenBank.FeatureParser-class.html#__init__,Method Bio.GenBank.Iterator.__init__()=Bio.GenBank.Iterator-class.html#__init__,Method Bio.GenBank.Record.Feature.__init__()=Bio.GenBank.Record.Feature-class.html#__init__,Method Bio.GenBank.Record.Qualifier.__init__()=Bio.GenBank.Record.Qualifier-class.html#__init__,Method Bio.GenBank.Record.Record.__init__()=Bio.GenBank.Record.Record-class.html#__init__,Method Bio.GenBank.Record.Reference.__init__()=Bio.GenBank.Record.Reference-class.html#__init__,Method Bio.GenBank.RecordParser.__init__()=Bio.GenBank.RecordParser-class.html#__init__,Method Bio.GenBank.Scanner.InsdcScanner.__init__()=Bio.GenBank.Scanner.InsdcScanner-class.html#__init__,Method Bio.GenBank._BaseGenBankConsumer.__init__()=Bio.GenBank._BaseGenBankConsumer-class.html#__init__,Method Bio.GenBank._FeatureConsumer.__init__()=Bio.GenBank._FeatureConsumer-class.html#__init__,Method Bio.GenBank._RecordConsumer.__init__()=Bio.GenBank._RecordConsumer-class.html#__init__,Method Bio.GenBank.utils.FeatureValueCleaner.__init__()=Bio.GenBank.utils.FeatureValueCleaner-class.html#__init__,Method Bio.Geo.Record.Record.__init__()=Bio.Geo.Record.Record-class.html#__init__,Method Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment.__init__()=Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome.Chromosome.__init__()=Bio.Graphics.BasicChromosome.Chromosome-class.html#__init__,Method Bio.Graphics.BasicChromosome.ChromosomeSegment.__init__()=Bio.Graphics.BasicChromosome.ChromosomeSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome.Organism.__init__()=Bio.Graphics.BasicChromosome.Organism-class.html#__init__,Method Bio.Graphics.BasicChromosome.TelomereSegment.__init__()=Bio.Graphics.BasicChromosome.TelomereSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome._ChromosomeComponent.__init__()=Bio.Graphics.BasicChromosome._ChromosomeComponent-class.html#__init__,Method Bio.Graphics.ColorSpiral.ColorSpiral.__init__()=Bio.Graphics.ColorSpiral.ColorSpiral-class.html#__init__,Method Bio.Graphics.Comparative.ComparativeScatterPlot.__init__()=Bio.Graphics.Comparative.ComparativeScatterPlot-class.html#__init__,Method Bio.Graphics.DisplayRepresentation.ChromosomeCounts.__init__()=Bio.Graphics.DisplayRepresentation.ChromosomeCounts-class.html#__init__,Method Bio.Graphics.Distribution.BarChartDistribution.__init__()=Bio.Graphics.Distribution.BarChartDistribution-class.html#__init__,Method Bio.Graphics.Distribution.DistributionPage.__init__()=Bio.Graphics.Distribution.DistributionPage-class.html#__init__,Method Bio.Graphics.Distribution.LineDistribution.__init__()=Bio.Graphics.Distribution.LineDistribution-class.html#__init__,Method Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.__init__()=Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.__init__()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.__init__()=Bio.Graphics.GenomeDiagram._Colors.ColorTranslator-class.html#__init__,Method Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.__init__()=Bio.Graphics.GenomeDiagram._CrossLink.CrossLink-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.__init__()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Feature.Feature.__init__()=Bio.Graphics.GenomeDiagram._Feature.Feature-class.html#__init__,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.__init__()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.__init__()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#__init__,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.__init__()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#__init__,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.__init__()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Track.Track.__init__()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#__init__,Method Bio.Graphics.KGML_vis.KGMLCanvas.__init__()=Bio.Graphics.KGML_vis.KGMLCanvas-class.html#__init__,Method Bio.HMM.DynamicProgramming.AbstractDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.AbstractDPAlgorithms-class.html#__init__,Method Bio.HMM.DynamicProgramming.LogDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.LogDPAlgorithms-class.html#__init__,Method Bio.HMM.DynamicProgramming.ScaledDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.ScaledDPAlgorithms-class.html#__init__,Method Bio.HMM.MarkovModel.HiddenMarkovModel.__init__()=Bio.HMM.MarkovModel.HiddenMarkovModel-class.html#__init__,Method Bio.HMM.MarkovModel.MarkovModelBuilder.__init__()=Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#__init__,Method Bio.HMM.Trainer.AbstractTrainer.__init__()=Bio.HMM.Trainer.AbstractTrainer-class.html#__init__,Method Bio.HMM.Trainer.BaumWelchTrainer.__init__()=Bio.HMM.Trainer.BaumWelchTrainer-class.html#__init__,Method Bio.HMM.Trainer.KnownStateTrainer.__init__()=Bio.HMM.Trainer.KnownStateTrainer-class.html#__init__,Method Bio.HMM.Trainer.TrainingSequence.__init__()=Bio.HMM.Trainer.TrainingSequence-class.html#__init__,Method Bio.Index._InMemoryIndex.__init__()=Bio.Index._InMemoryIndex-class.html#__init__,Method Bio.Index._ShelveIndex.__init__()=Bio.Index._ShelveIndex-class.html#__init__,Method Bio.KDTree.KDTree'.KDTree.__init__()=Bio.KDTree.KDTree%27.KDTree-class.html#__init__,Method Bio.KEGG.Compound.Record.__init__()=Bio.KEGG.Compound.Record-class.html#__init__,Method Bio.KEGG.Enzyme.Record.__init__()=Bio.KEGG.Enzyme.Record-class.html#__init__,Method Bio.KEGG.Gene.Record.__init__()=Bio.KEGG.Gene.Record-class.html#__init__,Method Bio.KEGG.KGML.KGML_parser.KGMLParser.__init__()=Bio.KEGG.KGML.KGML_parser.KGMLParser-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Component.__init__()=Bio.KEGG.KGML.KGML_pathway.Component-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Entry.__init__()=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Graphics.__init__()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Pathway.__init__()=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Reaction.__init__()=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Relation.__init__()=Bio.KEGG.KGML.KGML_pathway.Relation-class.html#__init__,Method Bio.LogisticRegression.LogisticRegression.__init__()=Bio.LogisticRegression.LogisticRegression-class.html#__init__,Method Bio.MarkovModel.MarkovModel.__init__()=Bio.MarkovModel.MarkovModel-class.html#__init__,Method Bio.MaxEntropy.MaxEntropy.__init__()=Bio.MaxEntropy.MaxEntropy-class.html#__init__,Method Bio.NMR.xpktools.Peaklist.__init__()=Bio.NMR.xpktools.Peaklist-class.html#__init__,Method Bio.NMR.xpktools.XpkEntry.__init__()=Bio.NMR.xpktools.XpkEntry-class.html#__init__,Method Bio.NaiveBayes.NaiveBayes.__init__()=Bio.NaiveBayes.NaiveBayes-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.AbstractLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.AbstractLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.InputLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Network.BasicNetwork.__init__()=Bio.NeuralNetwork.BackPropagation.Network.BasicNetwork-class.html#__init__,Method Bio.NeuralNetwork.Gene.Motif.MotifCoder.__init__()=Bio.NeuralNetwork.Gene.Motif.MotifCoder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Motif.MotifFinder.__init__()=Bio.NeuralNetwork.Gene.Motif.MotifFinder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Pattern.PatternIO.__init__()=Bio.NeuralNetwork.Gene.Pattern.PatternIO-class.html#__init__,Method Bio.NeuralNetwork.Gene.Pattern.PatternRepository.__init__()=Bio.NeuralNetwork.Gene.Pattern.PatternRepository-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness.__init__()=Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder.__init__()=Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness.__init__()=Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator.__init__()=Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.Schema.__init__()=Bio.NeuralNetwork.Gene.Schema.Schema-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SchemaCoder.__init__()=Bio.NeuralNetwork.Gene.Schema.SchemaCoder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SchemaFactory.__init__()=Bio.NeuralNetwork.Gene.Schema.SchemaFactory-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SchemaFinder.__init__()=Bio.NeuralNetwork.Gene.Schema.SchemaFinder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SimpleFinisher.__init__()=Bio.NeuralNetwork.Gene.Schema.SimpleFinisher-class.html#__init__,Method Bio.NeuralNetwork.Gene.Signature.SignatureCoder.__init__()=Bio.NeuralNetwork.Gene.Signature.SignatureCoder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Signature.SignatureFinder.__init__()=Bio.NeuralNetwork.Gene.Signature.SignatureFinder-class.html#__init__,Method Bio.NeuralNetwork.StopTraining.ValidationIncreaseStop.__init__()=Bio.NeuralNetwork.StopTraining.ValidationIncreaseStop-class.html#__init__,Method Bio.NeuralNetwork.Training.ExampleManager.__init__()=Bio.NeuralNetwork.Training.ExampleManager-class.html#__init__,Method Bio.NeuralNetwork.Training.TrainingExample.__init__()=Bio.NeuralNetwork.Training.TrainingExample-class.html#__init__,Method Bio.Nexus.Nexus.Block.__init__()=Bio.Nexus.Nexus.Block-class.html#__init__,Method Bio.Nexus.Nexus.CharBuffer.__init__()=Bio.Nexus.Nexus.CharBuffer-class.html#__init__,Method Bio.Nexus.Nexus.Commandline.__init__()=Bio.Nexus.Nexus.Commandline-class.html#__init__,Method Bio.Nexus.Nexus.Nexus.__init__()=Bio.Nexus.Nexus.Nexus-class.html#__init__,Method Bio.Nexus.Nexus.StepMatrix.__init__()=Bio.Nexus.Nexus.StepMatrix-class.html#__init__,Method Bio.Nexus.Nodes.Chain.__init__()=Bio.Nexus.Nodes.Chain-class.html#__init__,Method Bio.Nexus.Nodes.Node.__init__()=Bio.Nexus.Nodes.Node-class.html#__init__,Method Bio.Nexus.StandardData.StandardData.__init__()=Bio.Nexus.StandardData.StandardData-class.html#__init__,Method Bio.Nexus.Trees.NodeData.__init__()=Bio.Nexus.Trees.NodeData-class.html#__init__,Method Bio.Nexus.Trees.Tree.__init__()=Bio.Nexus.Trees.Tree-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractPropertyMap-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap-class.html#__init__,Method Bio.PDB.Atom.Atom.__init__()=Bio.PDB.Atom.Atom-class.html#__init__,Method Bio.PDB.Atom.DisorderedAtom.__init__()=Bio.PDB.Atom.DisorderedAtom-class.html#__init__,Method Bio.PDB.Chain.Chain.__init__()=Bio.PDB.Chain.Chain-class.html#__init__,Method Bio.PDB.DSSP'.DSSP.__init__()=Bio.PDB.DSSP%27.DSSP-class.html#__init__,Method Bio.PDB.Dice.ChainSelector.__init__()=Bio.PDB.Dice.ChainSelector-class.html#__init__,Method Bio.PDB.Entity.DisorderedEntityWrapper.__init__()=Bio.PDB.Entity.DisorderedEntityWrapper-class.html#__init__,Method Bio.PDB.Entity.Entity.__init__()=Bio.PDB.Entity.Entity-class.html#__init__,Method Bio.PDB.FragmentMapper'.Fragment.__init__()=Bio.PDB.FragmentMapper%27.Fragment-class.html#__init__,Method Bio.PDB.FragmentMapper'.FragmentMapper.__init__()=Bio.PDB.FragmentMapper%27.FragmentMapper-class.html#__init__,Method Bio.PDB.HSExposure.ExposureCN.__init__()=Bio.PDB.HSExposure.ExposureCN-class.html#__init__,Method Bio.PDB.HSExposure.HSExposureCA.__init__()=Bio.PDB.HSExposure.HSExposureCA-class.html#__init__,Method Bio.PDB.HSExposure.HSExposureCB.__init__()=Bio.PDB.HSExposure.HSExposureCB-class.html#__init__,Method Bio.PDB.HSExposure._AbstractHSExposure.__init__()=Bio.PDB.HSExposure._AbstractHSExposure-class.html#__init__,Method Bio.PDB.MMCIF2Dict.MMCIF2Dict.__init__()=Bio.PDB.MMCIF2Dict.MMCIF2Dict-class.html#__init__,Method Bio.PDB.MMCIFParser'.FastMMCIFParser.__init__()=Bio.PDB.MMCIFParser%27.FastMMCIFParser-class.html#__init__,Method Bio.PDB.MMCIFParser'.MMCIFParser.__init__()=Bio.PDB.MMCIFParser%27.MMCIFParser-class.html#__init__,Method Bio.PDB.Model.Model.__init__()=Bio.PDB.Model.Model-class.html#__init__,Method Bio.PDB.NACCESS.NACCESS.__init__()=Bio.PDB.NACCESS.NACCESS-class.html#__init__,Method Bio.PDB.NACCESS.NACCESS_atomic.__init__()=Bio.PDB.NACCESS.NACCESS_atomic-class.html#__init__,Method Bio.PDB.NeighborSearch'.NeighborSearch.__init__()=Bio.PDB.NeighborSearch%27.NeighborSearch-class.html#__init__,Method Bio.PDB.PDBIO'.PDBIO.__init__()=Bio.PDB.PDBIO%27.PDBIO-class.html#__init__,Method Bio.PDB.PDBList'.PDBList.__init__()=Bio.PDB.PDBList%27.PDBList-class.html#__init__,Method Bio.PDB.PDBParser'.PDBParser.__init__()=Bio.PDB.PDBParser%27.PDBParser-class.html#__init__,Method Bio.PDB.PSEA.PSEA.__init__()=Bio.PDB.PSEA.PSEA-class.html#__init__,Method Bio.PDB.Polypeptide.CaPPBuilder.__init__()=Bio.PDB.Polypeptide.CaPPBuilder-class.html#__init__,Method Bio.PDB.Polypeptide.PPBuilder.__init__()=Bio.PDB.Polypeptide.PPBuilder-class.html#__init__,Method Bio.PDB.Polypeptide._PPBuilder.__init__()=Bio.PDB.Polypeptide._PPBuilder-class.html#__init__,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.__init__()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#__init__,Method Bio.PDB.Residue.DisorderedResidue.__init__()=Bio.PDB.Residue.DisorderedResidue-class.html#__init__,Method Bio.PDB.Residue.Residue.__init__()=Bio.PDB.Residue.Residue-class.html#__init__,Method Bio.PDB.ResidueDepth'.ResidueDepth.__init__()=Bio.PDB.ResidueDepth%27.ResidueDepth-class.html#__init__,Method Bio.PDB.Structure.Structure.__init__()=Bio.PDB.Structure.Structure-class.html#__init__,Method Bio.PDB.StructureAlignment'.StructureAlignment.__init__()=Bio.PDB.StructureAlignment%27.StructureAlignment-class.html#__init__,Method Bio.PDB.StructureBuilder.StructureBuilder.__init__()=Bio.PDB.StructureBuilder.StructureBuilder-class.html#__init__,Method Bio.PDB.Superimposer'.Superimposer.__init__()=Bio.PDB.Superimposer%27.Superimposer-class.html#__init__,Method Bio.PDB.mmcifio.MMCIFIO.__init__()=Bio.PDB.mmcifio.MMCIFIO-class.html#__init__,Method Bio.PDB.mmtf.DefaultParser.StructureDecoder.__init__()=Bio.PDB.mmtf.DefaultParser.StructureDecoder-class.html#__init__,Method Bio.PDB.vectors.Vector.__init__()=Bio.PDB.vectors.Vector-class.html#__init__,Method Bio.ParserSupport.TaggingConsumer.__init__()=Bio.ParserSupport.TaggingConsumer-class.html#__init__,Method Bio.Pathway.Network.__init__()=Bio.Pathway.Network-class.html#__init__,Method Bio.Pathway.Reaction.__init__()=Bio.Pathway.Reaction-class.html#__init__,Method Bio.Pathway.Rep.Graph.Graph.__init__()=Bio.Pathway.Rep.Graph.Graph-class.html#__init__,Method Bio.Pathway.Rep.MultiGraph.MultiGraph.__init__()=Bio.Pathway.Rep.MultiGraph.MultiGraph-class.html#__init__,Method Bio.Pathway.System.__init__()=Bio.Pathway.System-class.html#__init__,Method Bio.Phylo.Applications._Fasttree.FastTreeCommandline.__init__()=Bio.Phylo.Applications._Fasttree.FastTreeCommandline-class.html#__init__,Method Bio.Phylo.Applications._Phyml.PhymlCommandline.__init__()=Bio.Phylo.Applications._Phyml.PhymlCommandline-class.html#__init__,Method Bio.Phylo.Applications._Raxml.RaxmlCommandline.__init__()=Bio.Phylo.Applications._Raxml.RaxmlCommandline-class.html#__init__,Method Bio.Phylo.BaseTree.BranchColor.__init__()=Bio.Phylo.BaseTree.BranchColor-class.html#__init__,Method Bio.Phylo.BaseTree.Clade.__init__()=Bio.Phylo.BaseTree.Clade-class.html#__init__,Method Bio.Phylo.BaseTree.Tree.__init__()=Bio.Phylo.BaseTree.Tree-class.html#__init__,Method Bio.Phylo.CDAO.Clade.__init__()=Bio.Phylo.CDAO.Clade-class.html#__init__,Method Bio.Phylo.CDAO.Tree.__init__()=Bio.Phylo.CDAO.Tree-class.html#__init__,Method Bio.Phylo.CDAOIO.Parser.__init__()=Bio.Phylo.CDAOIO.Parser-class.html#__init__,Method Bio.Phylo.CDAOIO.Writer.__init__()=Bio.Phylo.CDAOIO.Writer-class.html#__init__,Method Bio.Phylo.NeXML.Clade.__init__()=Bio.Phylo.NeXML.Clade-class.html#__init__,Method Bio.Phylo.NeXML.Tree.__init__()=Bio.Phylo.NeXML.Tree-class.html#__init__,Method Bio.Phylo.NeXMLIO.Parser.__init__()=Bio.Phylo.NeXMLIO.Parser-class.html#__init__,Method Bio.Phylo.NeXMLIO.Writer.__init__()=Bio.Phylo.NeXMLIO.Writer-class.html#__init__,Method Bio.Phylo.Newick.Clade.__init__()=Bio.Phylo.Newick.Clade-class.html#__init__,Method Bio.Phylo.Newick.Tree.__init__()=Bio.Phylo.Newick.Tree-class.html#__init__,Method Bio.Phylo.NewickIO.Parser.__init__()=Bio.Phylo.NewickIO.Parser-class.html#__init__,Method Bio.Phylo.NewickIO.Writer.__init__()=Bio.Phylo.NewickIO.Writer-class.html#__init__,Method Bio.Phylo.PAML._paml.Paml.__init__()=Bio.Phylo.PAML._paml.Paml-class.html#__init__,Method Bio.Phylo.PAML.baseml.Baseml.__init__()=Bio.Phylo.PAML.baseml.Baseml-class.html#__init__,Method Bio.Phylo.PAML.codeml.Codeml.__init__()=Bio.Phylo.PAML.codeml.Codeml-class.html#__init__,Method Bio.Phylo.PAML.yn00.Yn00.__init__()=Bio.Phylo.PAML.yn00.Yn00-class.html#__init__,Method Bio.Phylo.PhyloXML.Accession.__init__()=Bio.Phylo.PhyloXML.Accession-class.html#__init__,Method Bio.Phylo.PhyloXML.Annotation.__init__()=Bio.Phylo.PhyloXML.Annotation-class.html#__init__,Method Bio.Phylo.PhyloXML.BinaryCharacters.__init__()=Bio.Phylo.PhyloXML.BinaryCharacters-class.html#__init__,Method Bio.Phylo.PhyloXML.BranchColor.__init__()=Bio.Phylo.PhyloXML.BranchColor-class.html#__init__,Method Bio.Phylo.PhyloXML.Clade.__init__()=Bio.Phylo.PhyloXML.Clade-class.html#__init__,Method Bio.Phylo.PhyloXML.CladeRelation.__init__()=Bio.Phylo.PhyloXML.CladeRelation-class.html#__init__,Method Bio.Phylo.PhyloXML.Confidence.__init__()=Bio.Phylo.PhyloXML.Confidence-class.html#__init__,Method Bio.Phylo.PhyloXML.Date.__init__()=Bio.Phylo.PhyloXML.Date-class.html#__init__,Method Bio.Phylo.PhyloXML.Distribution.__init__()=Bio.Phylo.PhyloXML.Distribution-class.html#__init__,Method Bio.Phylo.PhyloXML.DomainArchitecture.__init__()=Bio.Phylo.PhyloXML.DomainArchitecture-class.html#__init__,Method Bio.Phylo.PhyloXML.Events.__init__()=Bio.Phylo.PhyloXML.Events-class.html#__init__,Method Bio.Phylo.PhyloXML.Id.__init__()=Bio.Phylo.PhyloXML.Id-class.html#__init__,Method Bio.Phylo.PhyloXML.MolSeq.__init__()=Bio.Phylo.PhyloXML.MolSeq-class.html#__init__,Method Bio.Phylo.PhyloXML.Other.__init__()=Bio.Phylo.PhyloXML.Other-class.html#__init__,Method Bio.Phylo.PhyloXML.Phylogeny.__init__()=Bio.Phylo.PhyloXML.Phylogeny-class.html#__init__,Method Bio.Phylo.PhyloXML.Phyloxml.__init__()=Bio.Phylo.PhyloXML.Phyloxml-class.html#__init__,Method Bio.Phylo.PhyloXML.Point.__init__()=Bio.Phylo.PhyloXML.Point-class.html#__init__,Method Bio.Phylo.PhyloXML.Polygon.__init__()=Bio.Phylo.PhyloXML.Polygon-class.html#__init__,Method Bio.Phylo.PhyloXML.Property.__init__()=Bio.Phylo.PhyloXML.Property-class.html#__init__,Method Bio.Phylo.PhyloXML.ProteinDomain.__init__()=Bio.Phylo.PhyloXML.ProteinDomain-class.html#__init__,Method Bio.Phylo.PhyloXML.Reference.__init__()=Bio.Phylo.PhyloXML.Reference-class.html#__init__,Method Bio.Phylo.PhyloXML.Sequence.__init__()=Bio.Phylo.PhyloXML.Sequence-class.html#__init__,Method Bio.Phylo.PhyloXML.SequenceRelation.__init__()=Bio.Phylo.PhyloXML.SequenceRelation-class.html#__init__,Method Bio.Phylo.PhyloXML.Taxonomy.__init__()=Bio.Phylo.PhyloXML.Taxonomy-class.html#__init__,Method Bio.Phylo.PhyloXML.Uri.__init__()=Bio.Phylo.PhyloXML.Uri-class.html#__init__,Method Bio.Phylo.PhyloXMLIO.Parser.__init__()=Bio.Phylo.PhyloXMLIO.Parser-class.html#__init__,Method Bio.Phylo.PhyloXMLIO.Writer.__init__()=Bio.Phylo.PhyloXMLIO.Writer-class.html#__init__,Method Bio.Phylo.TreeConstruction.DistanceCalculator.__init__()=Bio.Phylo.TreeConstruction.DistanceCalculator-class.html#__init__,Method Bio.Phylo.TreeConstruction.DistanceMatrix.__init__()=Bio.Phylo.TreeConstruction.DistanceMatrix-class.html#__init__,Method Bio.Phylo.TreeConstruction.DistanceTreeConstructor.__init__()=Bio.Phylo.TreeConstruction.DistanceTreeConstructor-class.html#__init__,Method Bio.Phylo.TreeConstruction.NNITreeSearcher.__init__()=Bio.Phylo.TreeConstruction.NNITreeSearcher-class.html#__init__,Method Bio.Phylo.TreeConstruction.ParsimonyScorer.__init__()=Bio.Phylo.TreeConstruction.ParsimonyScorer-class.html#__init__,Method Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor.__init__()=Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor-class.html#__init__,Method Bio.Phylo.TreeConstruction._Matrix.__init__()=Bio.Phylo.TreeConstruction._Matrix-class.html#__init__,Method Bio.PopGen.GenePop.Controller.GenePopController.__init__()=Bio.PopGen.GenePop.Controller.GenePopController-class.html#__init__,Method Bio.PopGen.GenePop.Controller._FileIterator.__init__()=Bio.PopGen.GenePop.Controller._FileIterator-class.html#__init__,Method Bio.PopGen.GenePop.Controller._GenePopCommandline.__init__()=Bio.PopGen.GenePop.Controller._GenePopCommandline-class.html#__init__,Method Bio.PopGen.GenePop.EasyController.EasyController.__init__()=Bio.PopGen.GenePop.EasyController.EasyController-class.html#__init__,Method Bio.PopGen.GenePop.FileParser.FileRecord.__init__()=Bio.PopGen.GenePop.FileParser.FileRecord-class.html#__init__,Method Bio.PopGen.GenePop.LargeFileParser.Record.__init__()=Bio.PopGen.GenePop.LargeFileParser.Record-class.html#__init__,Method Bio.PopGen.GenePop.Record.__init__()=Bio.PopGen.GenePop.Record-class.html#__init__,Method Bio.Restriction.PrintFormat.PrintFormat.__init__()=Bio.Restriction.PrintFormat.PrintFormat-class.html#__init__,Method Bio.Restriction.Restriction.Analysis.__init__()=Bio.Restriction.Restriction.Analysis-class.html#__init__,Method Bio.Restriction.Restriction.FormattedSeq.__init__()=Bio.Restriction.Restriction.FormattedSeq-class.html#__init__,Method Bio.Restriction.Restriction.RestrictionBatch.__init__()=Bio.Restriction.Restriction.RestrictionBatch-class.html#__init__,Method Bio.Restriction.Restriction.RestrictionType.__init__()=Bio.Restriction.Restriction.RestrictionType-class.html#__init__,Method Bio.SCOP.Astral.__init__()=Bio.SCOP.Astral-class.html#__init__,Method Bio.SCOP.Cla.Index.__init__()=Bio.SCOP.Cla.Index-class.html#__init__,Method Bio.SCOP.Cla.Record.__init__()=Bio.SCOP.Cla.Record-class.html#__init__,Method Bio.SCOP.Des.Record.__init__()=Bio.SCOP.Des.Record-class.html#__init__,Method Bio.SCOP.Dom.Record.__init__()=Bio.SCOP.Dom.Record-class.html#__init__,Method Bio.SCOP.Domain.__init__()=Bio.SCOP.Domain-class.html#__init__,Method Bio.SCOP.Hie.Record.__init__()=Bio.SCOP.Hie.Record-class.html#__init__,Method Bio.SCOP.Node.__init__()=Bio.SCOP.Node-class.html#__init__,Method Bio.SCOP.Raf.Res.__init__()=Bio.SCOP.Raf.Res-class.html#__init__,Method Bio.SCOP.Raf.SeqMap.__init__()=Bio.SCOP.Raf.SeqMap-class.html#__init__,Method Bio.SCOP.Raf.SeqMapIndex.__init__()=Bio.SCOP.Raf.SeqMapIndex-class.html#__init__,Method Bio.SCOP.Residues.Residues.__init__()=Bio.SCOP.Residues.Residues-class.html#__init__,Method Bio.SCOP.Scop.__init__()=Bio.SCOP.Scop-class.html#__init__,Method Bio.SVDSuperimposer.SVDSuperimposer.__init__()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabParser.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_text.BlastTextParser.__init__()=Bio.SearchIO.BlastIO.blast_text.BlastTextParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator.__init__()=Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslIndexer.__init__()=Bio.SearchIO.BlatIO.BlatPslIndexer-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslParser.__init__()=Bio.SearchIO.BlatIO.BlatPslParser-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslWriter.__init__()=Bio.SearchIO.BlatIO.BlatPslWriter-class.html#__init__,Method Bio.SearchIO.ExonerateIO._base._BaseExonerateParser.__init__()=Bio.SearchIO.ExonerateIO._base._BaseExonerateParser-class.html#__init__,Method Bio.SearchIO.FastaIO.FastaM10Indexer.__init__()=Bio.SearchIO.FastaIO.FastaM10Indexer-class.html#__init__,Method Bio.SearchIO.FastaIO.FastaM10Parser.__init__()=Bio.SearchIO.FastaIO.FastaM10Parser-class.html#__init__,Method Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer.__init__()=Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser.__init__()=Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.__init__()=Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser.__init__()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.__init__()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser.__init__()=Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html#__init__,Method Bio.SearchIO.InterproscanIO.interproscan_xml.InterproscanXmlParser.__init__()=Bio.SearchIO.InterproscanIO.interproscan_xml.InterproscanXmlParser-class.html#__init__,Method Bio.SearchIO._index.SearchIndexer.__init__()=Bio.SearchIO._index.SearchIndexer-class.html#__init__,Method Bio.SearchIO._model.hit.Hit.__init__()=Bio.SearchIO._model.hit.Hit-class.html#__init__,Method Bio.SearchIO._model.hsp.HSP.__init__()=Bio.SearchIO._model.hsp.HSP-class.html#__init__,Method Bio.SearchIO._model.hsp.HSPFragment.__init__()=Bio.SearchIO._model.hsp.HSPFragment-class.html#__init__,Method Bio.SearchIO._model.query.QueryResult.__init__()=Bio.SearchIO._model.query.QueryResult-class.html#__init__,Method Bio.Seq.MutableSeq.__init__()=Bio.Seq.MutableSeq-class.html#__init__,Method Bio.Seq.Seq.__init__()=Bio.Seq.Seq-class.html#__init__,Method Bio.Seq.UnknownSeq.__init__()=Bio.Seq.UnknownSeq-class.html#__init__,Method Bio.SeqFeature.CompoundLocation.__init__()=Bio.SeqFeature.CompoundLocation-class.html#__init__,Method Bio.SeqFeature.FeatureLocation.__init__()=Bio.SeqFeature.FeatureLocation-class.html#__init__,Method Bio.SeqFeature.PositionGap.__init__()=Bio.SeqFeature.PositionGap-class.html#__init__,Method Bio.SeqFeature.Reference.__init__()=Bio.SeqFeature.Reference-class.html#__init__,Method Bio.SeqFeature.SeqFeature.__init__()=Bio.SeqFeature.SeqFeature-class.html#__init__,Method Bio.SeqIO.FastaIO.FastaTwoLineWriter.__init__()=Bio.SeqIO.FastaIO.FastaTwoLineWriter-class.html#__init__,Method Bio.SeqIO.FastaIO.FastaWriter.__init__()=Bio.SeqIO.FastaIO.FastaWriter-class.html#__init__,Method Bio.SeqIO.Interfaces.SequenceIterator.__init__()=Bio.SeqIO.Interfaces.SequenceIterator-class.html#__init__,Method Bio.SeqIO.Interfaces.SequenceWriter.__init__()=Bio.SeqIO.Interfaces.SequenceWriter-class.html#__init__,Method Bio.SeqIO.Interfaces.SequentialSequenceWriter.__init__()=Bio.SeqIO.Interfaces.SequentialSequenceWriter-class.html#__init__,Method Bio.SeqIO.PhdIO.PhdWriter.__init__()=Bio.SeqIO.PhdIO.PhdWriter-class.html#__init__,Method Bio.SeqIO.PirIO.PirWriter.__init__()=Bio.SeqIO.PirIO.PirWriter-class.html#__init__,Method Bio.SeqIO.QualityIO.QualPhredWriter.__init__()=Bio.SeqIO.QualityIO.QualPhredWriter-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.SeqXmlIterator.__init__()=Bio.SeqIO.SeqXmlIO.SeqXmlIterator-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.SeqXmlWriter.__init__()=Bio.SeqIO.SeqXmlIO.SeqXmlWriter-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.XMLRecordIterator.__init__()=Bio.SeqIO.SeqXmlIO.XMLRecordIterator-class.html#__init__,Method Bio.SeqIO.SffIO.SffWriter.__init__()=Bio.SeqIO.SffIO.SffWriter-class.html#__init__,Method Bio.SeqIO.SffIO._AddTellHandle.__init__()=Bio.SeqIO.SffIO._AddTellHandle-class.html#__init__,Method Bio.SeqIO.UniprotIO.Parser.__init__()=Bio.SeqIO.UniprotIO.Parser-class.html#__init__,Method Bio.SeqIO._index.IntelliGeneticsRandomAccess.__init__()=Bio.SeqIO._index.IntelliGeneticsRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SeqFileRandomAccess.__init__()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SequentialSeqFileRandomAccess.__init__()=Bio.SeqIO._index.SequentialSeqFileRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SffRandomAccess.__init__()=Bio.SeqIO._index.SffRandomAccess-class.html#__init__,Method Bio.SeqRecord.SeqRecord.__init__()=Bio.SeqRecord.SeqRecord-class.html#__init__,Method Bio.SeqRecord._RestrictedDict.__init__()=Bio.SeqRecord._RestrictedDict-class.html#__init__,Method Bio.SeqUtils.CodonUsage.CodonAdaptationIndex.__init__()=Bio.SeqUtils.CodonUsage.CodonAdaptationIndex-class.html#__init__,Method Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.__init__()=Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint-class.html#__init__,Method Bio.SeqUtils.ProtParam.ProteinAnalysis.__init__()=Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#__init__,Method Bio.Sequencing.Ace.ACEFileRecord.__init__()=Bio.Sequencing.Ace.ACEFileRecord-class.html#__init__,Method Bio.Sequencing.Ace.Contig.__init__()=Bio.Sequencing.Ace.Contig-class.html#__init__,Method Bio.Sequencing.Ace.Reads.__init__()=Bio.Sequencing.Ace.Reads-class.html#__init__,Method Bio.Sequencing.Ace.af.__init__()=Bio.Sequencing.Ace.af-class.html#__init__,Method Bio.Sequencing.Ace.bs.__init__()=Bio.Sequencing.Ace.bs-class.html#__init__,Method Bio.Sequencing.Ace.ct.__init__()=Bio.Sequencing.Ace.ct-class.html#__init__,Method Bio.Sequencing.Ace.ds.__init__()=Bio.Sequencing.Ace.ds-class.html#__init__,Method Bio.Sequencing.Ace.qa.__init__()=Bio.Sequencing.Ace.qa-class.html#__init__,Method Bio.Sequencing.Ace.rd.__init__()=Bio.Sequencing.Ace.rd-class.html#__init__,Method Bio.Sequencing.Ace.rt.__init__()=Bio.Sequencing.Ace.rt-class.html#__init__,Method Bio.Sequencing.Ace.wa.__init__()=Bio.Sequencing.Ace.wa-class.html#__init__,Method Bio.Sequencing.Ace.wr.__init__()=Bio.Sequencing.Ace.wr-class.html#__init__,Method Bio.Sequencing.Applications._Novoalign.NovoalignCommandline.__init__()=Bio.Sequencing.Applications._Novoalign.NovoalignCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaAlignCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaAlignCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaBwaswCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaBwaswCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaIndexCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaIndexCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaSampeCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaSampeCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaSamseCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaSamseCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsVersion0xSortCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsVersion0xSortCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsVersion1xSortCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsVersion1xSortCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline-class.html#__init__,Method Bio.Sequencing.Phd.Record.__init__()=Bio.Sequencing.Phd.Record-class.html#__init__,Method Bio.SubsMat.FreqTable.FreqTable.__init__()=Bio.SubsMat.FreqTable.FreqTable-class.html#__init__,Method Bio.SubsMat.SeqMat.__init__()=Bio.SubsMat.SeqMat-class.html#__init__,Method Bio.SwissProt.KeyWList.Record.__init__()=Bio.SwissProt.KeyWList.Record-class.html#__init__,Method Bio.SwissProt.Record.__init__()=Bio.SwissProt.Record-class.html#__init__,Method Bio.SwissProt.Reference.__init__()=Bio.SwissProt.Reference-class.html#__init__,Method Bio.UniGene.ProtsimLine.__init__()=Bio.UniGene.ProtsimLine-class.html#__init__,Method Bio.UniGene.Record.__init__()=Bio.UniGene.Record-class.html#__init__,Method Bio.UniGene.STSLine.__init__()=Bio.UniGene.STSLine-class.html#__init__,Method Bio.UniGene.SequenceLine.__init__()=Bio.UniGene.SequenceLine-class.html#__init__,Method Bio.Wise.dnal.Statistics.__init__()=Bio.Wise.dnal.Statistics-class.html#__init__,Method Bio.Wise.psw.AlignmentColumn.__init__()=Bio.Wise.psw.AlignmentColumn-class.html#__init__,Method Bio.Wise.psw.ColumnUnit.__init__()=Bio.Wise.psw.ColumnUnit-class.html#__init__,Method Bio.bgzf.BgzfReader.__init__()=Bio.bgzf.BgzfReader-class.html#__init__,Method Bio.bgzf.BgzfWriter.__init__()=Bio.bgzf.BgzfWriter-class.html#__init__,Method Bio.codonalign.codonalignment.CodonAlignment.__init__()=Bio.codonalign.codonalignment.CodonAlignment-class.html#__init__,Method Bio.codonalign.codonseq.CodonSeq.__init__()=Bio.codonalign.codonseq.CodonSeq-class.html#__init__,Method Bio.kNN.kNN.__init__()=Bio.kNN.kNN-class.html#__init__,Method Bio.motifs.Instances.__init__()=Bio.motifs.Instances-class.html#__init__,Method Bio.motifs.Motif.__init__()=Bio.motifs.Motif-class.html#__init__,Method Bio.motifs.alignace.Record.__init__()=Bio.motifs.alignace.Record-class.html#__init__,Method Bio.motifs.applications._xxmotif.XXmotifCommandline.__init__()=Bio.motifs.applications._xxmotif.XXmotifCommandline-class.html#__init__,Method Bio.motifs.jaspar.Motif.__init__()=Bio.motifs.jaspar.Motif-class.html#__init__,Method Bio.motifs.jaspar.Record.__init__()=Bio.motifs.jaspar.Record-class.html#__init__,Method Bio.motifs.jaspar.db.JASPAR5.__init__()=Bio.motifs.jaspar.db.JASPAR5-class.html#__init__,Method Bio.motifs.mast.Record.__init__()=Bio.motifs.mast.Record-class.html#__init__,Method Bio.motifs.matrix.GenericPositionMatrix.__init__()=Bio.motifs.matrix.GenericPositionMatrix-class.html#__init__,Method Bio.motifs.matrix.PositionWeightMatrix.__init__()=Bio.motifs.matrix.PositionWeightMatrix-class.html#__init__,Method Bio.motifs.meme.Instance.__init__()=Bio.motifs.meme.Instance-class.html#__init__,Method Bio.motifs.meme.Motif.__init__()=Bio.motifs.meme.Motif-class.html#__init__,Method Bio.motifs.meme.Record.__init__()=Bio.motifs.meme.Record-class.html#__init__,Method Bio.motifs.thresholds.ScoreDistribution.__init__()=Bio.motifs.thresholds.ScoreDistribution-class.html#__init__,Method Bio.motifs.transfac.Record.__init__()=Bio.motifs.transfac.Record-class.html#__init__,Method Bio.pairwise2.affine_penalty.__init__()=Bio.pairwise2.affine_penalty-class.html#__init__,Method Bio.pairwise2.dictionary_match.__init__()=Bio.pairwise2.dictionary_match-class.html#__init__,Method Bio.pairwise2.identity_match.__init__()=Bio.pairwise2.identity_match-class.html#__init__,Method Bio.phenotype.phen_micro.JsonWriter.__init__()=Bio.phenotype.phen_micro.JsonWriter-class.html#__init__,Method Bio.phenotype.phen_micro.PlateRecord.__init__()=Bio.phenotype.phen_micro.PlateRecord-class.html#__init__,Method Bio.phenotype.phen_micro.WellRecord.__init__()=Bio.phenotype.phen_micro.WellRecord-class.html#__init__,Method BioSQL.BioSeq.DBSeq.__init__()=BioSQL.BioSeq.DBSeq-class.html#__init__,Method BioSQL.BioSeq.DBSeqRecord.__init__()=BioSQL.BioSeq.DBSeqRecord-class.html#__init__,Method BioSQL.BioSeqDatabase.Adaptor.__init__()=BioSQL.BioSeqDatabase.Adaptor-class.html#__init__,Method BioSQL.BioSeqDatabase.BioSeqDatabase.__init__()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#__init__,Method BioSQL.BioSeqDatabase.DBServer.__init__()=BioSQL.BioSeqDatabase.DBServer-class.html#__init__,Method BioSQL.BioSeqDatabase._CursorWrapper.__init__()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#__init__,Method BioSQL.DBUtils.Generic_dbutils.__init__()=BioSQL.DBUtils.Generic_dbutils-class.html#__init__,Method BioSQL.Loader.DatabaseLoader.__init__()=BioSQL.Loader.DatabaseLoader-class.html#__init__,Method BioSQL.Loader.DatabaseRemover.__init__()=BioSQL.Loader.DatabaseRemover-class.html#__init__"><a title="Bio.Affy.CelFile.ParserError.__init__
Bio.Affy.CelFile.Record.__init__
Bio.Align.AlignInfo.PSSM.__init__
Bio.Align.AlignInfo.SummaryInfo.__init__
Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline.__init__
Bio.Align.Applications._Clustalw.ClustalwCommandline.__init__
Bio.Align.Applications._Dialign.DialignCommandline.__init__
Bio.Align.Applications._MSAProbs.MSAProbsCommandline.__init__
Bio.Align.Applications._Mafft.MafftCommandline.__init__
Bio.Align.Applications._Muscle.MuscleCommandline.__init__
Bio.Align.Applications._Prank.PrankCommandline.__init__
Bio.Align.Applications._Probcons.ProbconsCommandline.__init__
Bio.Align.Applications._TCoffee.TCoffeeCommandline.__init__
Bio.Align.MultipleSeqAlignment.__init__
Bio.AlignIO.Interfaces.AlignmentIterator.__init__
Bio.AlignIO.Interfaces.AlignmentWriter.__init__
Bio.AlignIO.Interfaces.SequentialAlignmentWriter.__init__
Bio.AlignIO.MafIO.MafIndex.__init__
Bio.AlignIO.MauveIO.MauveWriter.__init__
Bio.Alphabet.AlphabetEncoder.__init__
Bio.Alphabet.Gapped.__init__
Bio.Alphabet.HasStopCodon.__init__
Bio.Application.AbstractCommandline.__init__
Bio.Application.ApplicationError.__init__
Bio.Application._AbstractParameter.__init__
Bio.Application._Argument.__init__
Bio.Application._Option.__init__
Bio.Application._StaticArgument.__init__
Bio.Application._Switch.__init__
Bio.Blast.Applications.NcbiblastformatterCommandline.__init__
Bio.Blast.Applications.NcbiblastnCommandline.__init__
Bio.Blast.Applications.NcbiblastpCommandline.__init__
Bio.Blast.Applications.NcbiblastxCommandline.__init__
Bio.Blast.Applications.NcbideltablastCommandline.__init__
Bio.Blast.Applications.NcbipsiblastCommandline.__init__
Bio.Blast.Applications.NcbirpsblastCommandline.__init__
Bio.Blast.Applications.NcbirpstblastnCommandline.__init__
Bio.Blast.Applications.NcbitblastnCommandline.__init__
Bio.Blast.Applications.NcbitblastxCommandline.__init__
Bio.Blast.Applications._NcbibaseblastCommandline.__init__
Bio.Blast.Applications._Ncbiblast2SeqCommandline.__init__
Bio.Blast.Applications._NcbiblastCommandline.__init__
Bio.Blast.Applications._NcbiblastMain2SeqCommandline.__init__
Bio.Blast.NCBIStandalone.BlastErrorParser.__init__
Bio.Blast.NCBIStandalone.BlastParser.__init__
Bio.Blast.NCBIStandalone.Iterator.__init__
Bio.Blast.NCBIStandalone.PSIBlastParser.__init__
Bio.Blast.NCBIStandalone._BlastConsumer.__init__
Bio.Blast.NCBIStandalone._BlastErrorConsumer.__init__
Bio.Blast.NCBIStandalone._PSIBlastConsumer.__init__
Bio.Blast.NCBIXML.BlastParser.__init__
Bio.Blast.NCBIXML._XMLparser.__init__
Bio.Blast.ParseBlastTable.BlastTableEntry.__init__
Bio.Blast.ParseBlastTable.BlastTableReader.__init__
Bio.Blast.ParseBlastTable.BlastTableRec.__init__
Bio.Blast.Record.Alignment.__init__
Bio.Blast.Record.Blast.__init__
Bio.Blast.Record.DatabaseReport.__init__
Bio.Blast.Record.Description.__init__
Bio.Blast.Record.HSP.__init__
Bio.Blast.Record.Header.__init__
Bio.Blast.Record.MultipleAlignment.__init__
Bio.Blast.Record.PSIBlast.__init__
Bio.Blast.Record.Parameters.__init__
Bio.Blast.Record.Round.__init__
Bio.CAPS.CAPSMap.__init__
Bio.CAPS.DifferentialCutsite.__init__
Bio.Cluster.Record.__init__
Bio.Compass.Record.__init__
Bio.Crystal.Chain.__init__
Bio.Crystal.Crystal.__init__
Bio.Crystal.Hetero.__init__
Bio.Data.CodonTable.AmbiguousCodonTable.__init__
Bio.Data.CodonTable.AmbiguousForwardTable.__init__
Bio.Data.CodonTable.CodonTable.__init__
Bio.Data.CodonTable.NCBICodonTable.__init__
Bio.DocSQL.Create.__init__
Bio.DocSQL.Insert.__init__
Bio.DocSQL.IterationCursor.__init__
Bio.DocSQL.Query.__init__
Bio.DocSQL.QueryAll.__init__
Bio.DocSQL.QueryGeneric.__init__
Bio.DocSQL.QueryRow.__init__
Bio.DocSQL.QuerySingle.__init__
Bio.Emboss.Applications.DiffseqCommandline.__init__
Bio.Emboss.Applications.EInvertedCommandline.__init__
Bio.Emboss.Applications.ETandemCommandline.__init__
Bio.Emboss.Applications.Est2GenomeCommandline.__init__
Bio.Emboss.Applications.FConsenseCommandline.__init__
Bio.Emboss.Applications.FDNADistCommandline.__init__
Bio.Emboss.Applications.FDNAParsCommandline.__init__
Bio.Emboss.Applications.FNeighborCommandline.__init__
Bio.Emboss.Applications.FProtDistCommandline.__init__
Bio.Emboss.Applications.FProtParsCommandline.__init__
Bio.Emboss.Applications.FSeqBootCommandline.__init__
Bio.Emboss.Applications.FTreeDistCommandline.__init__
Bio.Emboss.Applications.FuzznucCommandline.__init__
Bio.Emboss.Applications.FuzzproCommandline.__init__
Bio.Emboss.Applications.IepCommandline.__init__
Bio.Emboss.Applications.NeedleCommandline.__init__
Bio.Emboss.Applications.NeedleallCommandline.__init__
Bio.Emboss.Applications.PalindromeCommandline.__init__
Bio.Emboss.Applications.Primer3Commandline.__init__
Bio.Emboss.Applications.PrimerSearchCommandline.__init__
Bio.Emboss.Applications.SeqmatchallCommandline.__init__
Bio.Emboss.Applications.SeqretCommandline.__init__
Bio.Emboss.Applications.StretcherCommandline.__init__
Bio.Emboss.Applications.TranalignCommandline.__init__
Bio.Emboss.Applications.WaterCommandline.__init__
Bio.Emboss.Applications._EmbossCommandLine.__init__
Bio.Emboss.Applications._EmbossMinimalCommandLine.__init__
Bio.Emboss.Primer3.Primers.__init__
Bio.Emboss.Primer3.Record.__init__
Bio.Emboss.PrimerSearch.Amplifier.__init__
Bio.Emboss.PrimerSearch.InputRecord.__init__
Bio.Emboss.PrimerSearch.OutputRecord.__init__
Bio.Entrez.Parser.CorruptedXMLError.__init__
Bio.Entrez.Parser.DataHandler.__init__
Bio.Entrez.Parser.NotXMLError.__init__
Bio.Entrez.Parser.StructureElement.__init__
Bio.Entrez.Parser.ValidationError.__init__
Bio.ExPASy.Enzyme.Record.__init__
Bio.ExPASy.Prodoc.Record.__init__
Bio.ExPASy.Prodoc.Reference.__init__
Bio.ExPASy.Prosite.Record.__init__
Bio.ExPASy.ScanProsite.ContentHandler.__init__
Bio.ExPASy.ScanProsite.Parser.__init__
Bio.ExPASy.ScanProsite.Record.__init__
Bio.ExPASy.cellosaurus.Record.__init__
Bio.FSSP.FSSPAlignDict.__init__
Bio.FSSP.FSSPAlignRec.__init__
Bio.FSSP.FSSPHeader.__init__
Bio.FSSP.FSSPSumRec.__init__
Bio.FSSP.FSSPTools.FSSPMultAlign.__init__
Bio.FSSP.PosAlign.__init__
Bio.FSSP.fssp_rec.fff_rec.__init__
Bio.File.UndoHandle.__init__
Bio.File._IndexedSeqFileDict.__init__
Bio.File._SQLiteManySeqFilesDict.__init__
Bio.GA.Crossover.General.SafeFitnessCrossover.__init__
Bio.GA.Crossover.GeneralPoint.GeneralPointCrossover.__init__
Bio.GA.Crossover.GeneralPoint.InterleaveCrossover.__init__
Bio.GA.Crossover.Point.SinglePointCrossover.__init__
Bio.GA.Crossover.TwoPoint.TwoPointCrossover.__init__
Bio.GA.Crossover.Uniform.UniformCrossover.__init__
Bio.GA.Evolver.GenerationEvolver.__init__
Bio.GA.Evolver.SteadyStateEvolver.__init__
Bio.GA.Mutation.General.SafeFitnessMutation.__init__
Bio.GA.Mutation.Simple.ConversionMutation.__init__
Bio.GA.Mutation.Simple.SinglePositionMutation.__init__
Bio.GA.Organism.Organism.__init__
Bio.GA.Repair.Stabilizing.AmbiguousRepair.__init__
Bio.GA.Selection.Abstract.AbstractSelection.__init__
Bio.GA.Selection.Diversity.DiversitySelection.__init__
Bio.GA.Selection.RouletteWheel.RouletteWheelSelection.__init__
Bio.GA.Selection.Tournament.TournamentSelection.__init__
Bio.GenBank.FeatureParser.__init__
Bio.GenBank.Iterator.__init__
Bio.GenBank.Record.Feature.__init__
Bio.GenBank.Record.Qualifier.__init__
Bio.GenBank.Record.Record.__init__
Bio.GenBank.Record.Reference.__init__
Bio.GenBank.RecordParser.__init__
Bio.GenBank.Scanner.InsdcScanner.__init__
Bio.GenBank._BaseGenBankConsumer.__init__
Bio.GenBank._FeatureConsumer.__init__
Bio.GenBank._RecordConsumer.__init__
Bio.GenBank.utils.FeatureValueCleaner.__init__
Bio.Geo.Record.Record.__init__
Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment.__init__
Bio.Graphics.BasicChromosome.Chromosome.__init__
Bio.Graphics.BasicChromosome.ChromosomeSegment.__init__
Bio.Graphics.BasicChromosome.Organism.__init__
Bio.Graphics.BasicChromosome.TelomereSegment.__init__
Bio.Graphics.BasicChromosome._ChromosomeComponent.__init__
Bio.Graphics.ColorSpiral.ColorSpiral.__init__
Bio.Graphics.Comparative.ComparativeScatterPlot.__init__
Bio.Graphics.DisplayRepresentation.ChromosomeCounts.__init__
Bio.Graphics.Distribution.BarChartDistribution.__init__
Bio.Graphics.Distribution.DistributionPage.__init__
Bio.Graphics.Distribution.LineDistribution.__init__
Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.__init__
Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.__init__
Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.__init__
Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.__init__
Bio.Graphics.GenomeDiagram._Diagram.Diagram.__init__
Bio.Graphics.GenomeDiagram._Feature.Feature.__init__
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.__init__
Bio.Graphics.GenomeDiagram._Graph.GraphData.__init__
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.__init__
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.__init__
Bio.Graphics.GenomeDiagram._Track.Track.__init__
Bio.Graphics.KGML_vis.KGMLCanvas.__init__
Bio.HMM.DynamicProgramming.AbstractDPAlgorithms.__init__
Bio.HMM.DynamicProgramming.LogDPAlgorithms.__init__
Bio.HMM.DynamicProgramming.ScaledDPAlgorithms.__init__
Bio.HMM.MarkovModel.HiddenMarkovModel.__init__
Bio.HMM.MarkovModel.MarkovModelBuilder.__init__
Bio.HMM.Trainer.AbstractTrainer.__init__
Bio.HMM.Trainer.BaumWelchTrainer.__init__
Bio.HMM.Trainer.KnownStateTrainer.__init__
Bio.HMM.Trainer.TrainingSequence.__init__
Bio.Index._InMemoryIndex.__init__
Bio.Index._ShelveIndex.__init__
Bio.KDTree.KDTree'.KDTree.__init__
Bio.KEGG.Compound.Record.__init__
Bio.KEGG.Enzyme.Record.__init__
Bio.KEGG.Gene.Record.__init__
Bio.KEGG.KGML.KGML_parser.KGMLParser.__init__
Bio.KEGG.KGML.KGML_pathway.Component.__init__
Bio.KEGG.KGML.KGML_pathway.Entry.__init__
Bio.KEGG.KGML.KGML_pathway.Graphics.__init__
Bio.KEGG.KGML.KGML_pathway.Pathway.__init__
Bio.KEGG.KGML.KGML_pathway.Reaction.__init__
Bio.KEGG.KGML.KGML_pathway.Relation.__init__
Bio.LogisticRegression.LogisticRegression.__init__
Bio.MarkovModel.MarkovModel.__init__
Bio.MaxEntropy.MaxEntropy.__init__
Bio.NMR.xpktools.Peaklist.__init__
Bio.NMR.xpktools.XpkEntry.__init__
Bio.NaiveBayes.NaiveBayes.__init__
Bio.NeuralNetwork.BackPropagation.Layer.AbstractLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.__init__
Bio.NeuralNetwork.BackPropagation.Network.BasicNetwork.__init__
Bio.NeuralNetwork.Gene.Motif.MotifCoder.__init__
Bio.NeuralNetwork.Gene.Motif.MotifFinder.__init__
Bio.NeuralNetwork.Gene.Pattern.PatternIO.__init__
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.__init__
Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness.__init__
Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder.__init__
Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness.__init__
Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator.__init__
Bio.NeuralNetwork.Gene.Schema.Schema.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaCoder.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaFactory.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaFinder.__init__
Bio.NeuralNetwork.Gene.Schema.SimpleFinisher.__init__
Bio.NeuralNetwork.Gene.Signature.SignatureCoder.__init__
Bio.NeuralNetwork.Gene.Signature.SignatureFinder.__init__
Bio.NeuralNetwork.StopTraining.ValidationIncreaseStop.__init__
Bio.NeuralNetwork.Training.ExampleManager.__init__
Bio.NeuralNetwork.Training.TrainingExample.__init__
Bio.Nexus.Nexus.Block.__init__
Bio.Nexus.Nexus.CharBuffer.__init__
Bio.Nexus.Nexus.Commandline.__init__
Bio.Nexus.Nexus.Nexus.__init__
Bio.Nexus.Nexus.StepMatrix.__init__
Bio.Nexus.Nodes.Chain.__init__
Bio.Nexus.Nodes.Node.__init__
Bio.Nexus.StandardData.StandardData.__init__
Bio.Nexus.Trees.NodeData.__init__
Bio.Nexus.Trees.Tree.__init__
Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap.__init__
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__init__
Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap.__init__
Bio.PDB.Atom.Atom.__init__
Bio.PDB.Atom.DisorderedAtom.__init__
Bio.PDB.Chain.Chain.__init__
Bio.PDB.DSSP'.DSSP.__init__
Bio.PDB.Dice.ChainSelector.__init__
Bio.PDB.Entity.DisorderedEntityWrapper.__init__
Bio.PDB.Entity.Entity.__init__
Bio.PDB.FragmentMapper'.Fragment.__init__
Bio.PDB.FragmentMapper'.FragmentMapper.__init__
Bio.PDB.HSExposure.ExposureCN.__init__
Bio.PDB.HSExposure.HSExposureCA.__init__
Bio.PDB.HSExposure.HSExposureCB.__init__
Bio.PDB.HSExposure._AbstractHSExposure.__init__
Bio.PDB.MMCIF2Dict.MMCIF2Dict.__init__
Bio.PDB.MMCIFParser'.FastMMCIFParser.__init__
Bio.PDB.MMCIFParser'.MMCIFParser.__init__
Bio.PDB.Model.Model.__init__
Bio.PDB.NACCESS.NACCESS.__init__
Bio.PDB.NACCESS.NACCESS_atomic.__init__
Bio.PDB.NeighborSearch'.NeighborSearch.__init__
Bio.PDB.PDBIO'.PDBIO.__init__
Bio.PDB.PDBList'.PDBList.__init__
Bio.PDB.PDBParser'.PDBParser.__init__
Bio.PDB.PSEA.PSEA.__init__
Bio.PDB.Polypeptide.CaPPBuilder.__init__
Bio.PDB.Polypeptide.PPBuilder.__init__
Bio.PDB.Polypeptide._PPBuilder.__init__
Bio.PDB.QCPSuperimposer.QCPSuperimposer.__init__
Bio.PDB.Residue.DisorderedResidue.__init__
Bio.PDB.Residue.Residue.__init__
Bio.PDB.ResidueDepth'.ResidueDepth.__init__
Bio.PDB.Structure.Structure.__init__
Bio.PDB.StructureAlignment'.StructureAlignment.__init__
Bio.PDB.StructureBuilder.StructureBuilder.__init__
Bio.PDB.Superimposer'.Superimposer.__init__
Bio.PDB.mmcifio.MMCIFIO.__init__
Bio.PDB.mmtf.DefaultParser.StructureDecoder.__init__
Bio.PDB.vectors.Vector.__init__
Bio.ParserSupport.TaggingConsumer.__init__
Bio.Pathway.Network.__init__
Bio.Pathway.Reaction.__init__
Bio.Pathway.Rep.Graph.Graph.__init__
Bio.Pathway.Rep.MultiGraph.MultiGraph.__init__
Bio.Pathway.System.__init__
Bio.Phylo.Applications._Fasttree.FastTreeCommandline.__init__
Bio.Phylo.Applications._Phyml.PhymlCommandline.__init__
Bio.Phylo.Applications._Raxml.RaxmlCommandline.__init__
Bio.Phylo.BaseTree.BranchColor.__init__
Bio.Phylo.BaseTree.Clade.__init__
Bio.Phylo.BaseTree.Tree.__init__
Bio.Phylo.CDAO.Clade.__init__
Bio.Phylo.CDAO.Tree.__init__
Bio.Phylo.CDAOIO.Parser.__init__
Bio.Phylo.CDAOIO.Writer.__init__
Bio.Phylo.NeXML.Clade.__init__
Bio.Phylo.NeXML.Tree.__init__
Bio.Phylo.NeXMLIO.Parser.__init__
Bio.Phylo.NeXMLIO.Writer.__init__
Bio.Phylo.Newick.Clade.__init__
Bio.Phylo.Newick.Tree.__init__
Bio.Phylo.NewickIO.Parser.__init__
Bio.Phylo.NewickIO.Writer.__init__
Bio.Phylo.PAML._paml.Paml.__init__
Bio.Phylo.PAML.baseml.Baseml.__init__
Bio.Phylo.PAML.codeml.Codeml.__init__
Bio.Phylo.PAML.yn00.Yn00.__init__
Bio.Phylo.PhyloXML.Accession.__init__
Bio.Phylo.PhyloXML.Annotation.__init__
Bio.Phylo.PhyloXML.BinaryCharacters.__init__
Bio.Phylo.PhyloXML.BranchColor.__init__
Bio.Phylo.PhyloXML.Clade.__init__
Bio.Phylo.PhyloXML.CladeRelation.__init__
Bio.Phylo.PhyloXML.Confidence.__init__
Bio.Phylo.PhyloXML.Date.__init__
Bio.Phylo.PhyloXML.Distribution.__init__
Bio.Phylo.PhyloXML.DomainArchitecture.__init__
Bio.Phylo.PhyloXML.Events.__init__
Bio.Phylo.PhyloXML.Id.__init__
Bio.Phylo.PhyloXML.MolSeq.__init__
Bio.Phylo.PhyloXML.Other.__init__
Bio.Phylo.PhyloXML.Phylogeny.__init__
Bio.Phylo.PhyloXML.Phyloxml.__init__
Bio.Phylo.PhyloXML.Point.__init__
Bio.Phylo.PhyloXML.Polygon.__init__
Bio.Phylo.PhyloXML.Property.__init__
Bio.Phylo.PhyloXML.ProteinDomain.__init__
Bio.Phylo.PhyloXML.Reference.__init__
Bio.Phylo.PhyloXML.Sequence.__init__
Bio.Phylo.PhyloXML.SequenceRelation.__init__
Bio.Phylo.PhyloXML.Taxonomy.__init__
Bio.Phylo.PhyloXML.Uri.__init__
Bio.Phylo.PhyloXMLIO.Parser.__init__
Bio.Phylo.PhyloXMLIO.Writer.__init__
Bio.Phylo.TreeConstruction.DistanceCalculator.__init__
Bio.Phylo.TreeConstruction.DistanceMatrix.__init__
Bio.Phylo.TreeConstruction.DistanceTreeConstructor.__init__
Bio.Phylo.TreeConstruction.NNITreeSearcher.__init__
Bio.Phylo.TreeConstruction.ParsimonyScorer.__init__
Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor.__init__
Bio.Phylo.TreeConstruction._Matrix.__init__
Bio.PopGen.GenePop.Controller.GenePopController.__init__
Bio.PopGen.GenePop.Controller._FileIterator.__init__
Bio.PopGen.GenePop.Controller._GenePopCommandline.__init__
Bio.PopGen.GenePop.EasyController.EasyController.__init__
Bio.PopGen.GenePop.FileParser.FileRecord.__init__
Bio.PopGen.GenePop.LargeFileParser.Record.__init__
Bio.PopGen.GenePop.Record.__init__
Bio.Restriction.PrintFormat.PrintFormat.__init__
Bio.Restriction.Restriction.Analysis.__init__
Bio.Restriction.Restriction.FormattedSeq.__init__
Bio.Restriction.Restriction.RestrictionBatch.__init__
Bio.Restriction.Restriction.RestrictionType.__init__
Bio.SCOP.Astral.__init__
Bio.SCOP.Cla.Index.__init__
Bio.SCOP.Cla.Record.__init__
Bio.SCOP.Des.Record.__init__
Bio.SCOP.Dom.Record.__init__
Bio.SCOP.Domain.__init__
Bio.SCOP.Hie.Record.__init__
Bio.SCOP.Node.__init__
Bio.SCOP.Raf.Res.__init__
Bio.SCOP.Raf.SeqMap.__init__
Bio.SCOP.Raf.SeqMapIndex.__init__
Bio.SCOP.Residues.Residues.__init__
Bio.SCOP.Scop.__init__
Bio.SVDSuperimposer.SVDSuperimposer.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabParser.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.__init__
Bio.SearchIO.BlastIO.blast_text.BlastTextParser.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.__init__
Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator.__init__
Bio.SearchIO.BlatIO.BlatPslIndexer.__init__
Bio.SearchIO.BlatIO.BlatPslParser.__init__
Bio.SearchIO.BlatIO.BlatPslWriter.__init__
Bio.SearchIO.ExonerateIO._base._BaseExonerateParser.__init__
Bio.SearchIO.FastaIO.FastaM10Indexer.__init__
Bio.SearchIO.FastaIO.FastaM10Parser.__init__
Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer.__init__
Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser.__init__
Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.__init__
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser.__init__
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.__init__
Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser.__init__
Bio.SearchIO.InterproscanIO.interproscan_xml.InterproscanXmlParser.__init__
Bio.SearchIO._index.SearchIndexer.__init__
Bio.SearchIO._model.hit.Hit.__init__
Bio.SearchIO._model.hsp.HSP.__init__
Bio.SearchIO._model.hsp.HSPFragment.__init__
Bio.SearchIO._model.query.QueryResult.__init__
Bio.Seq.MutableSeq.__init__
Bio.Seq.Seq.__init__
Bio.Seq.UnknownSeq.__init__
Bio.SeqFeature.CompoundLocation.__init__
Bio.SeqFeature.FeatureLocation.__init__
Bio.SeqFeature.PositionGap.__init__
Bio.SeqFeature.Reference.__init__
Bio.SeqFeature.SeqFeature.__init__
Bio.SeqIO.FastaIO.FastaTwoLineWriter.__init__
Bio.SeqIO.FastaIO.FastaWriter.__init__
Bio.SeqIO.Interfaces.SequenceIterator.__init__
Bio.SeqIO.Interfaces.SequenceWriter.__init__
Bio.SeqIO.Interfaces.SequentialSequenceWriter.__init__
Bio.SeqIO.PhdIO.PhdWriter.__init__
Bio.SeqIO.PirIO.PirWriter.__init__
Bio.SeqIO.QualityIO.QualPhredWriter.__init__
Bio.SeqIO.SeqXmlIO.SeqXmlIterator.__init__
Bio.SeqIO.SeqXmlIO.SeqXmlWriter.__init__
Bio.SeqIO.SeqXmlIO.XMLRecordIterator.__init__
Bio.SeqIO.SffIO.SffWriter.__init__
Bio.SeqIO.SffIO._AddTellHandle.__init__
Bio.SeqIO.UniprotIO.Parser.__init__
Bio.SeqIO._index.IntelliGeneticsRandomAccess.__init__
Bio.SeqIO._index.SeqFileRandomAccess.__init__
Bio.SeqIO._index.SequentialSeqFileRandomAccess.__init__
Bio.SeqIO._index.SffRandomAccess.__init__
Bio.SeqRecord.SeqRecord.__init__
Bio.SeqRecord._RestrictedDict.__init__
Bio.SeqUtils.CodonUsage.CodonAdaptationIndex.__init__
Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.__init__
Bio.SeqUtils.ProtParam.ProteinAnalysis.__init__
Bio.Sequencing.Ace.ACEFileRecord.__init__
Bio.Sequencing.Ace.Contig.__init__
Bio.Sequencing.Ace.Reads.__init__
Bio.Sequencing.Ace.af.__init__
Bio.Sequencing.Ace.bs.__init__
Bio.Sequencing.Ace.ct.__init__
Bio.Sequencing.Ace.ds.__init__
Bio.Sequencing.Ace.qa.__init__
Bio.Sequencing.Ace.rd.__init__
Bio.Sequencing.Ace.rt.__init__
Bio.Sequencing.Ace.wa.__init__
Bio.Sequencing.Ace.wr.__init__
Bio.Sequencing.Applications._Novoalign.NovoalignCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaAlignCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaBwaswCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaIndexCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaSampeCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaSamseCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsVersion0xSortCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsVersion1xSortCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline.__init__
Bio.Sequencing.Phd.Record.__init__
Bio.SubsMat.FreqTable.FreqTable.__init__
Bio.SubsMat.SeqMat.__init__
Bio.SwissProt.KeyWList.Record.__init__
Bio.SwissProt.Record.__init__
Bio.SwissProt.Reference.__init__
Bio.UniGene.ProtsimLine.__init__
Bio.UniGene.Record.__init__
Bio.UniGene.STSLine.__init__
Bio.UniGene.SequenceLine.__init__
Bio.Wise.dnal.Statistics.__init__
Bio.Wise.psw.AlignmentColumn.__init__
Bio.Wise.psw.ColumnUnit.__init__
Bio.bgzf.BgzfReader.__init__
Bio.bgzf.BgzfWriter.__init__
Bio.codonalign.codonalignment.CodonAlignment.__init__
Bio.codonalign.codonseq.CodonSeq.__init__
Bio.kNN.kNN.__init__
Bio.motifs.Instances.__init__
Bio.motifs.Motif.__init__
Bio.motifs.alignace.Record.__init__
Bio.motifs.applications._xxmotif.XXmotifCommandline.__init__
Bio.motifs.jaspar.Motif.__init__
Bio.motifs.jaspar.Record.__init__
Bio.motifs.jaspar.db.JASPAR5.__init__
Bio.motifs.mast.Record.__init__
Bio.motifs.matrix.GenericPositionMatrix.__init__
Bio.motifs.matrix.PositionWeightMatrix.__init__
Bio.motifs.meme.Instance.__init__
Bio.motifs.meme.Motif.__init__
Bio.motifs.meme.Record.__init__
Bio.motifs.thresholds.ScoreDistribution.__init__
Bio.motifs.transfac.Record.__init__
Bio.pairwise2.affine_penalty.__init__
Bio.pairwise2.dictionary_match.__init__
Bio.pairwise2.identity_match.__init__
Bio.phenotype.phen_micro.JsonWriter.__init__
Bio.phenotype.phen_micro.PlateRecord.__init__
Bio.phenotype.phen_micro.WellRecord.__init__
BioSQL.BioSeq.DBSeq.__init__
BioSQL.BioSeq.DBSeqRecord.__init__
BioSQL.BioSeqDatabase.Adaptor.__init__
BioSQL.BioSeqDatabase.BioSeqDatabase.__init__
BioSQL.BioSeqDatabase.DBServer.__init__
BioSQL.BioSeqDatabase._CursorWrapper.__init__
BioSQL.DBUtils.Generic_dbutils.__init__
BioSQL.Loader.DatabaseLoader.__init__
BioSQL.Loader.DatabaseRemover.__init__" class="py-name" href="#" onclick="return doclink('link-144', '__init__', 'link-144');">__init__</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-name">pslx</tt><tt class="py-op">=</tt><tt class="py-name">pslx</tt><tt class="py-op">)</tt> </tt>
</div><a name="L530"></a><tt class="py-lineno">530</tt>  <tt class="py-line"> </tt>
<a name="BlatPslIndexer.__iter__"></a><div id="BlatPslIndexer.__iter__-def"><a name="L531"></a><tt class="py-lineno">531</tt> <a class="py-toggle" href="#" id="BlatPslIndexer.__iter__-toggle" onclick="return toggle('BlatPslIndexer.__iter__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlatIO.BlatPslIndexer-class.html#__iter__">__iter__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlatPslIndexer.__iter__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlatPslIndexer.__iter__-expanded"><a name="L532"></a><tt class="py-lineno">532</tt>  <tt class="py-line">        <tt class="py-docstring">"""Iterate over the file handle; yields key, start offset, and length."""</tt> </tt>
<a name="L533"></a><tt class="py-lineno">533</tt>  <tt class="py-line">        <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_handle</tt> </tt>
<a name="L534"></a><tt class="py-lineno">534</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-145" class="py-name" targets="Method Bio.File.UndoHandle.seek()=Bio.File.UndoHandle-class.html#seek,Method Bio.SeqIO.SffIO._AddTellHandle.seek()=Bio.SeqIO.SffIO._AddTellHandle-class.html#seek,Method Bio.bgzf.BgzfReader.seek()=Bio.bgzf.BgzfReader-class.html#seek"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-145', 'seek', 'link-145');">seek</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L535"></a><tt class="py-lineno">535</tt>  <tt class="py-line">        <tt class="py-comment"># denotes column location for query identifier</tt> </tt>
<a name="L536"></a><tt class="py-lineno">536</tt>  <tt class="py-line">        <tt class="py-name">query_id_idx</tt> <tt class="py-op">=</tt> <tt class="py-number">9</tt> </tt>
<a name="L537"></a><tt class="py-lineno">537</tt>  <tt class="py-line">        <tt class="py-name">qresult_key</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L538"></a><tt class="py-lineno">538</tt>  <tt class="py-line">        <tt class="py-name">tab_char</tt> <tt class="py-op">=</tt> <tt class="py-string">b"\t"</tt> </tt>
<a name="L539"></a><tt class="py-lineno">539</tt>  <tt class="py-line"> </tt>
<a name="L540"></a><tt class="py-lineno">540</tt>  <tt class="py-line">        <tt class="py-name">start_offset</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-146" class="py-name" targets="Method Bio.File.UndoHandle.tell()=Bio.File.UndoHandle-class.html#tell,Method Bio.SeqIO.SffIO._AddTellHandle.tell()=Bio.SeqIO.SffIO._AddTellHandle-class.html#tell,Method Bio.bgzf.BgzfReader.tell()=Bio.bgzf.BgzfReader-class.html#tell,Method Bio.bgzf.BgzfWriter.tell()=Bio.bgzf.BgzfWriter-class.html#tell"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-146', 'tell', 'link-146');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L541"></a><tt class="py-lineno">541</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-147" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-147', 'readline', 'link-108');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L542"></a><tt class="py-lineno">542</tt>  <tt class="py-line">        <tt class="py-comment"># read through header</tt> </tt>
<a name="L543"></a><tt class="py-lineno">543</tt>  <tt class="py-line">        <tt class="py-comment"># this assumes that the result row match the regex</tt> </tt>
<a name="L544"></a><tt class="py-lineno">544</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-keyword">not</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-148" class="py-name"><a title="Bio.AlignIO.MafIO.MafIndex.search
Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-148', 'search', 'link-109');">search</a></tt><tt class="py-op">(</tt><tt id="link-149" class="py-name"><a title="Bio.SearchIO.BlatIO._RE_ROW_CHECK_IDX" class="py-name" href="#" onclick="return doclink('link-149', '_RE_ROW_CHECK_IDX', 'link-23');">_RE_ROW_CHECK_IDX</a></tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-150" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-150', 'strip', 'link-111');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L545"></a><tt class="py-lineno">545</tt>  <tt class="py-line">            <tt class="py-name">start_offset</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-151" class="py-name"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-151', 'tell', 'link-146');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L546"></a><tt class="py-lineno">546</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-152" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-152', 'readline', 'link-108');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L547"></a><tt class="py-lineno">547</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L548"></a><tt class="py-lineno">548</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> </tt>
<a name="L549"></a><tt class="py-lineno">549</tt>  <tt class="py-line"> </tt>
<a name="L550"></a><tt class="py-lineno">550</tt>  <tt class="py-line">        <tt class="py-comment"># and index the qresults</tt> </tt>
<a name="L551"></a><tt class="py-lineno">551</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L552"></a><tt class="py-lineno">552</tt>  <tt class="py-line">            <tt class="py-name">end_offset</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-153" class="py-name"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-153', 'tell', 'link-146');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L553"></a><tt class="py-lineno">553</tt>  <tt class="py-line"> </tt>
<a name="L554"></a><tt class="py-lineno">554</tt>  <tt class="py-line">            <tt class="py-name">cols</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-154" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-154', 'x', 'link-26');">x</a></tt> <tt class="py-keyword">for</tt> <tt id="link-155" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-155', 'x', 'link-26');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-156" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-156', 'strip', 'link-111');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-157" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-157', 'split', 'link-28');">split</a></tt><tt class="py-op">(</tt><tt class="py-name">tab_char</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt id="link-158" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-158', 'x', 'link-26');">x</a></tt><tt class="py-op">]</tt> </tt>
<a name="L555"></a><tt class="py-lineno">555</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">qresult_key</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L556"></a><tt class="py-lineno">556</tt>  <tt class="py-line">                <tt class="py-name">qresult_key</tt> <tt class="py-op">=</tt> <tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-name">query_id_idx</tt><tt class="py-op">]</tt> </tt>
<a name="L557"></a><tt class="py-lineno">557</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L558"></a><tt class="py-lineno">558</tt>  <tt class="py-line">                <tt class="py-name">curr_key</tt> <tt class="py-op">=</tt> <tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-name">query_id_idx</tt><tt class="py-op">]</tt> </tt>
<a name="L559"></a><tt class="py-lineno">559</tt>  <tt class="py-line"> </tt>
<a name="L560"></a><tt class="py-lineno">560</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">curr_key</tt> <tt class="py-op">!=</tt> <tt class="py-name">qresult_key</tt><tt class="py-op">:</tt> </tt>
<a name="L561"></a><tt class="py-lineno">561</tt>  <tt class="py-line">                    <tt class="py-keyword">yield</tt> <tt id="link-159" class="py-name"><a title="Bio._py3k._bytes_to_string" class="py-name" href="#" onclick="return doclink('link-159', '_bytes_to_string', 'link-3');">_bytes_to_string</a></tt><tt class="py-op">(</tt><tt class="py-name">qresult_key</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">start_offset</tt><tt class="py-op">,</tt> \ </tt>
<a name="L562"></a><tt class="py-lineno">562</tt>  <tt class="py-line">                            <tt class="py-name">end_offset</tt> <tt class="py-op">-</tt> <tt class="py-name">start_offset</tt> </tt>
<a name="L563"></a><tt class="py-lineno">563</tt>  <tt class="py-line">                    <tt class="py-name">qresult_key</tt> <tt class="py-op">=</tt> <tt class="py-name">curr_key</tt> </tt>
<a name="L564"></a><tt class="py-lineno">564</tt>  <tt class="py-line">                    <tt class="py-name">start_offset</tt> <tt class="py-op">=</tt> <tt class="py-name">end_offset</tt> <tt class="py-op">-</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L565"></a><tt class="py-lineno">565</tt>  <tt class="py-line"> </tt>
<a name="L566"></a><tt class="py-lineno">566</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-160" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-160', 'readline', 'link-108');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L567"></a><tt class="py-lineno">567</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L568"></a><tt class="py-lineno">568</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt id="link-161" class="py-name"><a title="Bio._py3k._bytes_to_string" class="py-name" href="#" onclick="return doclink('link-161', '_bytes_to_string', 'link-3');">_bytes_to_string</a></tt><tt class="py-op">(</tt><tt class="py-name">qresult_key</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">start_offset</tt><tt class="py-op">,</tt> \ </tt>
<a name="L569"></a><tt class="py-lineno">569</tt>  <tt class="py-line">                        <tt class="py-name">end_offset</tt> <tt class="py-op">-</tt> <tt class="py-name">start_offset</tt> </tt>
<a name="L570"></a><tt class="py-lineno">570</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
</div><a name="L571"></a><tt class="py-lineno">571</tt>  <tt class="py-line"> </tt>
<a name="BlatPslIndexer.get_raw"></a><div id="BlatPslIndexer.get_raw-def"><a name="L572"></a><tt class="py-lineno">572</tt> <a class="py-toggle" href="#" id="BlatPslIndexer.get_raw-toggle" onclick="return toggle('BlatPslIndexer.get_raw');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlatIO.BlatPslIndexer-class.html#get_raw">get_raw</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">offset</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlatPslIndexer.get_raw-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlatPslIndexer.get_raw-expanded"><a name="L573"></a><tt class="py-lineno">573</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return raw bytes string of a QueryResult object from the given offset."""</tt> </tt>
<a name="L574"></a><tt class="py-lineno">574</tt>  <tt class="py-line">        <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_handle</tt> </tt>
<a name="L575"></a><tt class="py-lineno">575</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-162" class="py-name"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-162', 'seek', 'link-145');">seek</a></tt><tt class="py-op">(</tt><tt class="py-name">offset</tt><tt class="py-op">)</tt> </tt>
<a name="L576"></a><tt class="py-lineno">576</tt>  <tt class="py-line">        <tt class="py-name">query_id_idx</tt> <tt class="py-op">=</tt> <tt class="py-number">9</tt> </tt>
<a name="L577"></a><tt class="py-lineno">577</tt>  <tt class="py-line">        <tt class="py-name">qresult_key</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L578"></a><tt class="py-lineno">578</tt>  <tt class="py-line">        <tt class="py-name">qresult_raw</tt> <tt class="py-op">=</tt> <tt class="py-string">b""</tt> </tt>
<a name="L579"></a><tt class="py-lineno">579</tt>  <tt class="py-line">        <tt class="py-name">tab_char</tt> <tt class="py-op">=</tt> <tt class="py-string">b"\t"</tt> </tt>
<a name="L580"></a><tt class="py-lineno">580</tt>  <tt class="py-line"> </tt>
<a name="L581"></a><tt class="py-lineno">581</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L582"></a><tt class="py-lineno">582</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-163" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-163', 'readline', 'link-108');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L583"></a><tt class="py-lineno">583</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L584"></a><tt class="py-lineno">584</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L585"></a><tt class="py-lineno">585</tt>  <tt class="py-line">            <tt class="py-name">cols</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-164" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-164', 'x', 'link-26');">x</a></tt> <tt class="py-keyword">for</tt> <tt id="link-165" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-165', 'x', 'link-26');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-166" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-166', 'strip', 'link-111');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-167" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-167', 'split', 'link-28');">split</a></tt><tt class="py-op">(</tt><tt class="py-name">tab_char</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt id="link-168" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-168', 'x', 'link-26');">x</a></tt><tt class="py-op">]</tt> </tt>
<a name="L586"></a><tt class="py-lineno">586</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">qresult_key</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L587"></a><tt class="py-lineno">587</tt>  <tt class="py-line">                <tt class="py-name">qresult_key</tt> <tt class="py-op">=</tt> <tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-name">query_id_idx</tt><tt class="py-op">]</tt> </tt>
<a name="L588"></a><tt class="py-lineno">588</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L589"></a><tt class="py-lineno">589</tt>  <tt class="py-line">                <tt class="py-name">curr_key</tt> <tt class="py-op">=</tt> <tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-name">query_id_idx</tt><tt class="py-op">]</tt> </tt>
<a name="L590"></a><tt class="py-lineno">590</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">curr_key</tt> <tt class="py-op">!=</tt> <tt class="py-name">qresult_key</tt><tt class="py-op">:</tt> </tt>
<a name="L591"></a><tt class="py-lineno">591</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L592"></a><tt class="py-lineno">592</tt>  <tt class="py-line">            <tt class="py-name">qresult_raw</tt> <tt class="py-op">+=</tt> <tt class="py-name">line</tt> </tt>
<a name="L593"></a><tt class="py-lineno">593</tt>  <tt class="py-line"> </tt>
<a name="L594"></a><tt class="py-lineno">594</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">qresult_raw</tt> </tt>
</div></div><a name="L595"></a><tt class="py-lineno">595</tt>  <tt class="py-line"> </tt>
<a name="L596"></a><tt class="py-lineno">596</tt>  <tt class="py-line"> </tt>
<a name="BlatPslWriter"></a><div id="BlatPslWriter-def"><a name="L597"></a><tt class="py-lineno">597</tt> <a class="py-toggle" href="#" id="BlatPslWriter-toggle" onclick="return toggle('BlatPslWriter');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.SearchIO.BlatIO.BlatPslWriter-class.html">BlatPslWriter</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlatPslWriter-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="BlatPslWriter-expanded"><a name="L598"></a><tt class="py-lineno">598</tt>  <tt class="py-line">    <tt class="py-docstring">"""Writer for the blat-psl format."""</tt> </tt>
<a name="L599"></a><tt class="py-lineno">599</tt>  <tt class="py-line"> </tt>
<a name="BlatPslWriter.__init__"></a><div id="BlatPslWriter.__init__-def"><a name="L600"></a><tt class="py-lineno">600</tt> <a class="py-toggle" href="#" id="BlatPslWriter.__init__-toggle" onclick="return toggle('BlatPslWriter.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlatIO.BlatPslWriter-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">header</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt> <tt class="py-param">pslx</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlatPslWriter.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlatPslWriter.__init__-expanded"><a name="L601"></a><tt class="py-lineno">601</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the class."""</tt> </tt>
<a name="L602"></a><tt class="py-lineno">602</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt> </tt>
<a name="L603"></a><tt class="py-lineno">603</tt>  <tt class="py-line">        <tt class="py-comment"># flag for writing header or not</tt> </tt>
<a name="L604"></a><tt class="py-lineno">604</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-169" class="py-name" targets="Variable Bio.Wise.dnal.Statistics.header=Bio.Wise.dnal.Statistics-class.html#header"><a title="Bio.Wise.dnal.Statistics.header" class="py-name" href="#" onclick="return doclink('link-169', 'header', 'link-169');">header</a></tt> <tt class="py-op">=</tt> <tt id="link-170" class="py-name"><a title="Bio.Wise.dnal.Statistics.header" class="py-name" href="#" onclick="return doclink('link-170', 'header', 'link-169');">header</a></tt> </tt>
<a name="L605"></a><tt class="py-lineno">605</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">pslx</tt> <tt class="py-op">=</tt> <tt class="py-name">pslx</tt> </tt>
</div><a name="L606"></a><tt class="py-lineno">606</tt>  <tt class="py-line"> </tt>
<a name="BlatPslWriter.write_file"></a><div id="BlatPslWriter.write_file-def"><a name="L607"></a><tt class="py-lineno">607</tt> <a class="py-toggle" href="#" id="BlatPslWriter.write_file-toggle" onclick="return toggle('BlatPslWriter.write_file');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlatIO.BlatPslWriter-class.html#write_file">write_file</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">qresults</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlatPslWriter.write_file-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlatPslWriter.write_file-expanded"><a name="L608"></a><tt class="py-lineno">608</tt>  <tt class="py-line">        <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt> </tt>
<a name="L609"></a><tt class="py-lineno">609</tt>  <tt class="py-line">        <tt class="py-name">qresult_counter</tt><tt class="py-op">,</tt> <tt class="py-name">hit_counter</tt><tt class="py-op">,</tt> <tt class="py-name">hsp_counter</tt><tt class="py-op">,</tt> <tt class="py-name">frag_counter</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt> </tt>
<a name="L610"></a><tt class="py-lineno">610</tt>  <tt class="py-line"> </tt>
<a name="L611"></a><tt class="py-lineno">611</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-171" class="py-name"><a title="Bio.Wise.dnal.Statistics.header" class="py-name" href="#" onclick="return doclink('link-171', 'header', 'link-169');">header</a></tt><tt class="py-op">:</tt> </tt>
<a name="L612"></a><tt class="py-lineno">612</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-172" class="py-name" targets="Function Bio.AlignIO.write()=Bio.AlignIO-module.html#write,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.write()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#write,Method Bio.NeuralNetwork.Gene.Pattern.PatternIO.write()=Bio.NeuralNetwork.Gene.Pattern.PatternIO-class.html#write,Method Bio.Phylo.CDAOIO.Writer.write()=Bio.Phylo.CDAOIO.Writer-class.html#write,Function Bio.Phylo.CDAOIO.write()=Bio.Phylo.CDAOIO-module.html#write,Method Bio.Phylo.NeXMLIO.Writer.write()=Bio.Phylo.NeXMLIO.Writer-class.html#write,Function Bio.Phylo.NeXMLIO.write()=Bio.Phylo.NeXMLIO-module.html#write,Method Bio.Phylo.NewickIO.Writer.write()=Bio.Phylo.NewickIO.Writer-class.html#write,Function Bio.Phylo.NewickIO.write()=Bio.Phylo.NewickIO-module.html#write,Function Bio.Phylo.NexusIO.write()=Bio.Phylo.NexusIO-module.html#write,Method Bio.Phylo.PhyloXMLIO.Writer.write()=Bio.Phylo.PhyloXMLIO.Writer-class.html#write,Function Bio.Phylo.PhyloXMLIO.write()=Bio.Phylo.PhyloXMLIO-module.html#write,Function Bio.Phylo._io.write()=Bio.Phylo._io-module.html#write,Function Bio.SearchIO.write()=Bio.SearchIO-module.html#write,Function Bio.SeqIO.write()=Bio.SeqIO-module.html#write,Method Bio.bgzf.BgzfWriter.write()=Bio.bgzf.BgzfWriter-class.html#write,Function Bio.motifs.jaspar.write()=Bio.motifs.jaspar-module.html#write,Function Bio.motifs.transfac.write()=Bio.motifs.transfac-module.html#write,Function Bio.motifs.write()=Bio.motifs-module.html#write,Method Bio.phenotype.phen_micro.JsonWriter.write()=Bio.phenotype.phen_micro.JsonWriter-class.html#write,Function Bio.phenotype.write()=Bio.phenotype-module.html#write"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-172', 'write', 'link-172');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-173" class="py-name" targets="Method Bio.SearchIO.BlatIO.BlatPslWriter._build_header()=Bio.SearchIO.BlatIO.BlatPslWriter-class.html#_build_header,Method Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter._build_header()=Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter-class.html#_build_header,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter._build_header()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter-class.html#_build_header"><a title="Bio.SearchIO.BlatIO.BlatPslWriter._build_header
Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter._build_header
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter._build_header" class="py-name" href="#" onclick="return doclink('link-173', '_build_header', 'link-173');">_build_header</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L613"></a><tt class="py-lineno">613</tt>  <tt class="py-line"> </tt>
<a name="L614"></a><tt class="py-lineno">614</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">qresult</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qresults</tt><tt class="py-op">:</tt> </tt>
<a name="L615"></a><tt class="py-lineno">615</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">qresult</tt><tt class="py-op">:</tt> </tt>
<a name="L616"></a><tt class="py-lineno">616</tt>  <tt class="py-line">                <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-174" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-174', 'write', 'link-172');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-175" class="py-name" targets="Method Bio.SearchIO.BlatIO.BlatPslWriter._build_row()=Bio.SearchIO.BlatIO.BlatPslWriter-class.html#_build_row,Method Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter._build_row()=Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter-class.html#_build_row,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter._build_row()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter-class.html#_build_row"><a title="Bio.SearchIO.BlatIO.BlatPslWriter._build_row
Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter._build_row
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter._build_row" class="py-name" href="#" onclick="return doclink('link-175', '_build_row', 'link-175');">_build_row</a></tt><tt class="py-op">(</tt><tt class="py-name">qresult</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L617"></a><tt class="py-lineno">617</tt>  <tt class="py-line">                <tt class="py-name">qresult_counter</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L618"></a><tt class="py-lineno">618</tt>  <tt class="py-line">                <tt class="py-name">hit_counter</tt> <tt class="py-op">+=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">qresult</tt><tt class="py-op">)</tt> </tt>
<a name="L619"></a><tt class="py-lineno">619</tt>  <tt class="py-line">                <tt class="py-name">hsp_counter</tt> <tt class="py-op">+=</tt> <tt id="link-176" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.sum()=Bio.Nexus.Nexus.StepMatrix-class.html#sum,Method Bio.SubsMat.SeqMat.sum()=Bio.SubsMat.SeqMat-class.html#sum"><a title="Bio.Nexus.Nexus.StepMatrix.sum
Bio.SubsMat.SeqMat.sum" class="py-name" href="#" onclick="return doclink('link-176', 'sum', 'link-176');">sum</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-177" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-177', 'hit', 'link-62');">hit</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-178" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-178', 'hit', 'link-62');">hit</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">qresult</tt><tt class="py-op">)</tt> </tt>
<a name="L620"></a><tt class="py-lineno">620</tt>  <tt class="py-line">                <tt class="py-name">frag_counter</tt> <tt class="py-op">+=</tt> <tt id="link-179" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.sum
Bio.SubsMat.SeqMat.sum" class="py-name" href="#" onclick="return doclink('link-179', 'sum', 'link-176');">sum</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-180" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-180', 'hit', 'link-62');">hit</a></tt><tt class="py-op">.</tt><tt id="link-181" class="py-name" targets="Variable Bio.SearchIO._model.hit.Hit.fragments=Bio.SearchIO._model.hit.Hit-class.html#fragments,Variable Bio.SearchIO._model.hsp.HSP.fragments=Bio.SearchIO._model.hsp.HSP-class.html#fragments,Variable Bio.SearchIO._model.query.QueryResult.fragments=Bio.SearchIO._model.query.QueryResult-class.html#fragments"><a title="Bio.SearchIO._model.hit.Hit.fragments
Bio.SearchIO._model.hsp.HSP.fragments
Bio.SearchIO._model.query.QueryResult.fragments" class="py-name" href="#" onclick="return doclink('link-181', 'fragments', 'link-181');">fragments</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-182" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-182', 'hit', 'link-62');">hit</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">qresult</tt><tt class="py-op">)</tt> </tt>
<a name="L621"></a><tt class="py-lineno">621</tt>  <tt class="py-line"> </tt>
<a name="L622"></a><tt class="py-lineno">622</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">qresult_counter</tt><tt class="py-op">,</tt> <tt class="py-name">hit_counter</tt><tt class="py-op">,</tt> <tt class="py-name">hsp_counter</tt><tt class="py-op">,</tt> <tt class="py-name">frag_counter</tt> </tt>
</div><a name="L623"></a><tt class="py-lineno">623</tt>  <tt class="py-line"> </tt>
<a name="BlatPslWriter._build_header"></a><div id="BlatPslWriter._build_header-def"><a name="L624"></a><tt class="py-lineno">624</tt> <a class="py-toggle" href="#" id="BlatPslWriter._build_header-toggle" onclick="return toggle('BlatPslWriter._build_header');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlatIO.BlatPslWriter-class.html#_build_header">_build_header</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlatPslWriter._build_header-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlatPslWriter._build_header-expanded"><a name="L625"></a><tt class="py-lineno">625</tt>  <tt class="py-line">        <tt class="py-comment"># for now, always use the psLayout version 3</tt> </tt>
<a name="L626"></a><tt class="py-lineno">626</tt>  <tt class="py-line">        <tt id="link-183" class="py-name"><a title="Bio.Wise.dnal.Statistics.header" class="py-name" href="#" onclick="return doclink('link-183', 'header', 'link-169');">header</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'psLayout version 3\n'</tt> </tt>
<a name="L627"></a><tt class="py-lineno">627</tt>  <tt class="py-line"> </tt>
<a name="L628"></a><tt class="py-lineno">628</tt>  <tt class="py-line">        <tt class="py-comment"># adapted from BLAT's source: lib/psl.c#L496</tt> </tt>
<a name="L629"></a><tt class="py-lineno">629</tt>  <tt class="py-line">        <tt id="link-184" class="py-name"><a title="Bio.Wise.dnal.Statistics.header" class="py-name" href="#" onclick="return doclink('link-184', 'header', 'link-169');">header</a></tt> <tt class="py-op">+=</tt> <tt class="py-string">"\nmatch\tmis- \trep. \tN's\tQ gap\tQ gap\tT gap\tT "</tt> </tt>
<a name="L630"></a><tt class="py-lineno">630</tt>  <tt class="py-line">        <tt class="py-string">"gap\tstrand\tQ        \tQ   \tQ    \tQ  \tT        \tT   \tT    "</tt> </tt>
<a name="L631"></a><tt class="py-lineno">631</tt>  <tt class="py-line">        <tt class="py-string">"\tT  \tblock\tblockSizes \tqStarts\t tStarts\n     "</tt> \ </tt>
<a name="L632"></a><tt class="py-lineno">632</tt>  <tt class="py-line">        <tt class="py-string">"\tmatch\tmatch\t   \tcount\tbases\tcount\tbases\t      \tname     "</tt> </tt>
<a name="L633"></a><tt class="py-lineno">633</tt>  <tt class="py-line">        <tt class="py-string">"\tsize\tstart\tend\tname     \tsize\tstart\tend\tcount"</tt> </tt>
<a name="L634"></a><tt class="py-lineno">634</tt>  <tt class="py-line">        <tt class="py-string">"\n%s\n"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-string">'-'</tt> <tt class="py-op">*</tt> <tt class="py-number">159</tt><tt class="py-op">)</tt> </tt>
<a name="L635"></a><tt class="py-lineno">635</tt>  <tt class="py-line"> </tt>
<a name="L636"></a><tt class="py-lineno">636</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-185" class="py-name"><a title="Bio.Wise.dnal.Statistics.header" class="py-name" href="#" onclick="return doclink('link-185', 'header', 'link-169');">header</a></tt> </tt>
</div><a name="L637"></a><tt class="py-lineno">637</tt>  <tt class="py-line"> </tt>
<a name="BlatPslWriter._build_row"></a><div id="BlatPslWriter._build_row-def"><a name="L638"></a><tt class="py-lineno">638</tt> <a class="py-toggle" href="#" id="BlatPslWriter._build_row-toggle" onclick="return toggle('BlatPslWriter._build_row');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.BlatIO.BlatPslWriter-class.html#_build_row">_build_row</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">qresult</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlatPslWriter._build_row-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BlatPslWriter._build_row-expanded"><a name="L639"></a><tt class="py-lineno">639</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a string or one row or more of the QueryResult object (PRIVATE)."""</tt> </tt>
<a name="L640"></a><tt class="py-lineno">640</tt>  <tt class="py-line">        <tt class="py-comment"># For now, our writer writes the row according to the order in</tt> </tt>
<a name="L641"></a><tt class="py-lineno">641</tt>  <tt class="py-line">        <tt class="py-comment"># the QueryResult and Hit objects.</tt> </tt>
<a name="L642"></a><tt class="py-lineno">642</tt>  <tt class="py-line">        <tt class="py-comment"># This is different from BLAT's native output, where the rows are</tt> </tt>
<a name="L643"></a><tt class="py-lineno">643</tt>  <tt class="py-line">        <tt class="py-comment"># grouped by strand.</tt> </tt>
<a name="L644"></a><tt class="py-lineno">644</tt>  <tt class="py-line">        <tt class="py-comment"># Should we tweak the behavior to better mimic the native output?</tt> </tt>
<a name="L645"></a><tt class="py-lineno">645</tt>  <tt class="py-line">        <tt class="py-name">qresult_lines</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L646"></a><tt class="py-lineno">646</tt>  <tt class="py-line"> </tt>
<a name="L647"></a><tt class="py-lineno">647</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-186" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-186', 'hit', 'link-62');">hit</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">qresult</tt><tt class="py-op">:</tt> </tt>
<a name="L648"></a><tt class="py-lineno">648</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt id="link-187" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-187', 'hsp', 'link-75');">hsp</a></tt> <tt class="py-keyword">in</tt> <tt id="link-188" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-188', 'hit', 'link-62');">hit</a></tt><tt class="py-op">.</tt><tt id="link-189" class="py-name" targets="Variable Bio.SearchIO._model.hit.Hit.hsps=Bio.SearchIO._model.hit.Hit-class.html#hsps,Variable Bio.SearchIO._model.query.QueryResult.hsps=Bio.SearchIO._model.query.QueryResult-class.html#hsps"><a title="Bio.SearchIO._model.hit.Hit.hsps
Bio.SearchIO._model.query.QueryResult.hsps" class="py-name" href="#" onclick="return doclink('link-189', 'hsps', 'link-189');">hsps</a></tt><tt class="py-op">:</tt> </tt>
<a name="L649"></a><tt class="py-lineno">649</tt>  <tt class="py-line"> </tt>
<a name="L650"></a><tt class="py-lineno">650</tt>  <tt class="py-line">                <tt class="py-name">query_is_protein</tt> <tt class="py-op">=</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt id="link-190" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-190', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">,</tt> <tt class="py-string">"query_is_protein"</tt><tt class="py-op">,</tt> <tt class="py-name">False</tt><tt class="py-op">)</tt> </tt>
<a name="L651"></a><tt class="py-lineno">651</tt>  <tt class="py-line">                <tt class="py-name">blocksize_multiplier</tt> <tt class="py-op">=</tt> <tt class="py-number">3</tt> <tt class="py-keyword">if</tt> <tt class="py-name">query_is_protein</tt> <tt class="py-keyword">else</tt> <tt class="py-number">1</tt> </tt>
<a name="L652"></a><tt class="py-lineno">652</tt>  <tt class="py-line"> </tt>
<a name="L653"></a><tt class="py-lineno">653</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L654"></a><tt class="py-lineno">654</tt>  <tt class="py-line">                <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-191" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-191', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt id="link-192" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-192', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">match_num</tt><tt class="py-op">)</tt> </tt>
<a name="L655"></a><tt class="py-lineno">655</tt>  <tt class="py-line">                <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-193" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-193', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt id="link-194" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-194', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">mismatch_num</tt><tt class="py-op">)</tt> </tt>
<a name="L656"></a><tt class="py-lineno">656</tt>  <tt class="py-line">                <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-195" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-195', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt id="link-196" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-196', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">match_rep_num</tt><tt class="py-op">)</tt> </tt>
<a name="L657"></a><tt class="py-lineno">657</tt>  <tt class="py-line">                <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-197" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-197', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt id="link-198" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-198', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">n_num</tt><tt class="py-op">)</tt> </tt>
<a name="L658"></a><tt class="py-lineno">658</tt>  <tt class="py-line">                <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-199" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-199', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt id="link-200" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-200', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">query_gapopen_num</tt><tt class="py-op">)</tt> </tt>
<a name="L659"></a><tt class="py-lineno">659</tt>  <tt class="py-line">                <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-201" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-201', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt id="link-202" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-202', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">query_gap_num</tt><tt class="py-op">)</tt> </tt>
<a name="L660"></a><tt class="py-lineno">660</tt>  <tt class="py-line">                <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-203" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-203', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt id="link-204" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-204', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">hit_gapopen_num</tt><tt class="py-op">)</tt> </tt>
<a name="L661"></a><tt class="py-lineno">661</tt>  <tt class="py-line">                <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-205" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-205', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt id="link-206" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-206', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">hit_gap_num</tt><tt class="py-op">)</tt> </tt>
<a name="L662"></a><tt class="py-lineno">662</tt>  <tt class="py-line"> </tt>
<a name="L663"></a><tt class="py-lineno">663</tt>  <tt class="py-line">                <tt class="py-comment"># check spans</tt> </tt>
<a name="L664"></a><tt class="py-lineno">664</tt>  <tt class="py-line">                <tt class="py-name">eff_query_spans</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">blocksize_multiplier</tt> <tt class="py-op">*</tt> <tt class="py-name">s</tt> <tt class="py-keyword">for</tt> <tt class="py-name">s</tt> <tt class="py-keyword">in</tt> <tt id="link-207" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-207', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-208" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_span_all" class="py-name" href="#" onclick="return doclink('link-208', 'query_span_all', 'link-88');">query_span_all</a></tt><tt class="py-op">]</tt> </tt>
<a name="L665"></a><tt class="py-lineno">665</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-209" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-209', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-210" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.hit_span_all" class="py-name" href="#" onclick="return doclink('link-210', 'hit_span_all', 'link-86');">hit_span_all</a></tt> <tt class="py-op">!=</tt> <tt class="py-name">eff_query_spans</tt><tt class="py-op">:</tt> </tt>
<a name="L666"></a><tt class="py-lineno">666</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"HSP hit span and query span values do not match."</tt><tt class="py-op">)</tt> </tt>
<a name="L667"></a><tt class="py-lineno">667</tt>  <tt class="py-line">                <tt class="py-name">block_sizes</tt> <tt class="py-op">=</tt> <tt id="link-211" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-211', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-212" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_span_all" class="py-name" href="#" onclick="return doclink('link-212', 'query_span_all', 'link-88');">query_span_all</a></tt> </tt>
<a name="L668"></a><tt class="py-lineno">668</tt>  <tt class="py-line"> </tt>
<a name="L669"></a><tt class="py-lineno">669</tt>  <tt class="py-line">                <tt class="py-comment"># set strand and starts</tt> </tt>
<a name="L670"></a><tt class="py-lineno">670</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-213" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-213', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-214" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_strand
Bio.SearchIO._model.hsp.HSPFragment.query_strand" class="py-name" href="#" onclick="return doclink('link-214', 'query_strand', 'link-72');">query_strand</a></tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt>  <tt class="py-comment"># since it may be a protein seq</tt> </tt>
<a name="L671"></a><tt class="py-lineno">671</tt>  <tt class="py-line">                    <tt id="link-215" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-215', 'strand', 'link-34');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'+'</tt> </tt>
<a name="L672"></a><tt class="py-lineno">672</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L673"></a><tt class="py-lineno">673</tt>  <tt class="py-line">                    <tt id="link-216" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-216', 'strand', 'link-34');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'-'</tt> </tt>
<a name="L674"></a><tt class="py-lineno">674</tt>  <tt class="py-line">                <tt class="py-name">qstarts</tt> <tt class="py-op">=</tt> <tt id="link-217" class="py-name"><a title="Bio.SearchIO.BlatIO._reorient_starts" class="py-name" href="#" onclick="return doclink('link-217', '_reorient_starts', 'link-40');">_reorient_starts</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt id="link-218" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-218', 'x', 'link-26');">x</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt id="link-219" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-219', 'x', 'link-26');">x</a></tt> <tt class="py-keyword">in</tt> <tt id="link-220" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-220', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-221" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_range_all" class="py-name" href="#" onclick="return doclink('link-221', 'query_range_all', 'link-44');">query_range_all</a></tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L675"></a><tt class="py-lineno">675</tt>  <tt class="py-line">                        <tt id="link-222" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-222', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-223" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_span_all" class="py-name" href="#" onclick="return doclink('link-223', 'query_span_all', 'link-88');">query_span_all</a></tt><tt class="py-op">,</tt> <tt class="py-name">qresult</tt><tt class="py-op">.</tt><tt class="py-name">seq_len</tt><tt class="py-op">,</tt> <tt id="link-224" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-224', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-225" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_strand
Bio.SearchIO._model.hsp.HSPFragment.query_strand" class="py-name" href="#" onclick="return doclink('link-225', 'query_strand', 'link-72');">query_strand</a></tt><tt class="py-op">)</tt> </tt>
<a name="L676"></a><tt class="py-lineno">676</tt>  <tt class="py-line"> </tt>
<a name="L677"></a><tt class="py-lineno">677</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-226" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-226', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-227" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.hit_strand
Bio.SearchIO._model.hsp.HSPFragment.hit_strand" class="py-name" href="#" onclick="return doclink('link-227', 'hit_strand', 'link-73');">hit_strand</a></tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L678"></a><tt class="py-lineno">678</tt>  <tt class="py-line">                    <tt class="py-name">hstrand</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L679"></a><tt class="py-lineno">679</tt>  <tt class="py-line">                    <tt class="py-comment"># only write hit strand if it was present in the source file</tt> </tt>
<a name="L680"></a><tt class="py-lineno">680</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt id="link-228" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-228', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">_has_hit_strand</tt><tt class="py-op">:</tt> </tt>
<a name="L681"></a><tt class="py-lineno">681</tt>  <tt class="py-line">                        <tt id="link-229" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-229', 'strand', 'link-34');">strand</a></tt> <tt class="py-op">+=</tt> <tt class="py-string">'+'</tt> </tt>
<a name="L682"></a><tt class="py-lineno">682</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L683"></a><tt class="py-lineno">683</tt>  <tt class="py-line">                    <tt class="py-name">hstrand</tt> <tt class="py-op">=</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt> </tt>
<a name="L684"></a><tt class="py-lineno">684</tt>  <tt class="py-line">                    <tt id="link-230" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-230', 'strand', 'link-34');">strand</a></tt> <tt class="py-op">+=</tt> <tt class="py-string">'-'</tt> </tt>
<a name="L685"></a><tt class="py-lineno">685</tt>  <tt class="py-line">                <tt class="py-name">hstarts</tt> <tt class="py-op">=</tt> <tt id="link-231" class="py-name"><a title="Bio.SearchIO.BlatIO._reorient_starts" class="py-name" href="#" onclick="return doclink('link-231', '_reorient_starts', 'link-40');">_reorient_starts</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt id="link-232" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-232', 'x', 'link-26');">x</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt id="link-233" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-233', 'x', 'link-26');">x</a></tt> <tt class="py-keyword">in</tt> <tt id="link-234" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-234', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-235" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.hit_range_all" class="py-name" href="#" onclick="return doclink('link-235', 'hit_range_all', 'link-49');">hit_range_all</a></tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L686"></a><tt class="py-lineno">686</tt>  <tt class="py-line">                        <tt id="link-236" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-236', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-237" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.hit_span_all" class="py-name" href="#" onclick="return doclink('link-237', 'hit_span_all', 'link-86');">hit_span_all</a></tt><tt class="py-op">,</tt> <tt id="link-238" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-238', 'hit', 'link-62');">hit</a></tt><tt class="py-op">.</tt><tt class="py-name">seq_len</tt><tt class="py-op">,</tt> <tt class="py-name">hstrand</tt><tt class="py-op">)</tt> </tt>
<a name="L687"></a><tt class="py-lineno">687</tt>  <tt class="py-line"> </tt>
<a name="L688"></a><tt class="py-lineno">688</tt>  <tt class="py-line">                <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-239" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-239', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt id="link-240" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-240', 'strand', 'link-34');">strand</a></tt><tt class="py-op">)</tt> </tt>
<a name="L689"></a><tt class="py-lineno">689</tt>  <tt class="py-line">                <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-241" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-241', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">qresult</tt><tt class="py-op">.</tt><tt id="link-242" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-242', 'id', 'link-136');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L690"></a><tt class="py-lineno">690</tt>  <tt class="py-line">                <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-243" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-243', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">qresult</tt><tt class="py-op">.</tt><tt class="py-name">seq_len</tt><tt class="py-op">)</tt> </tt>
<a name="L691"></a><tt class="py-lineno">691</tt>  <tt class="py-line">                <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-244" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-244', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt id="link-245" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-245', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-246" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_start
Bio.SearchIO._model.hsp.HSPFragment.query_start" class="py-name" href="#" onclick="return doclink('link-246', 'query_start', 'link-66');">query_start</a></tt><tt class="py-op">)</tt> </tt>
<a name="L692"></a><tt class="py-lineno">692</tt>  <tt class="py-line">                <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-247" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-247', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt id="link-248" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-248', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-249" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_end
Bio.SearchIO._model.hsp.HSPFragment.query_end" class="py-name" href="#" onclick="return doclink('link-249', 'query_end', 'link-67');">query_end</a></tt><tt class="py-op">)</tt> </tt>
<a name="L693"></a><tt class="py-lineno">693</tt>  <tt class="py-line">                <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-250" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-250', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt id="link-251" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-251', 'hit', 'link-62');">hit</a></tt><tt class="py-op">.</tt><tt id="link-252" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-252', 'id', 'link-136');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L694"></a><tt class="py-lineno">694</tt>  <tt class="py-line">                <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-253" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-253', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt id="link-254" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-254', 'hit', 'link-62');">hit</a></tt><tt class="py-op">.</tt><tt class="py-name">seq_len</tt><tt class="py-op">)</tt> </tt>
<a name="L695"></a><tt class="py-lineno">695</tt>  <tt class="py-line">                <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-255" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-255', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt id="link-256" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-256', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-257" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.hit_start
Bio.SearchIO._model.hsp.HSPFragment.hit_start" class="py-name" href="#" onclick="return doclink('link-257', 'hit_start', 'link-68');">hit_start</a></tt><tt class="py-op">)</tt> </tt>
<a name="L696"></a><tt class="py-lineno">696</tt>  <tt class="py-line">                <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-258" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-258', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt id="link-259" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-259', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-260" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.hit_end
Bio.SearchIO._model.hsp.HSPFragment.hit_end" class="py-name" href="#" onclick="return doclink('link-260', 'hit_end', 'link-70');">hit_end</a></tt><tt class="py-op">)</tt> </tt>
<a name="L697"></a><tt class="py-lineno">697</tt>  <tt class="py-line">                <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-261" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-261', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-262" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-262', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L698"></a><tt class="py-lineno">698</tt>  <tt class="py-line">                <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-263" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-263', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">','</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-264" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-264', 'x', 'link-26');">x</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-265" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-265', 'x', 'link-26');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">block_sizes</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">','</tt><tt class="py-op">)</tt> </tt>
<a name="L699"></a><tt class="py-lineno">699</tt>  <tt class="py-line">                <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-266" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-266', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">','</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-267" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-267', 'x', 'link-26');">x</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-268" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-268', 'x', 'link-26');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">qstarts</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">','</tt><tt class="py-op">)</tt> </tt>
<a name="L700"></a><tt class="py-lineno">700</tt>  <tt class="py-line">                <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-269" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-269', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">','</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-270" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-270', 'x', 'link-26');">x</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-271" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-271', 'x', 'link-26');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">hstarts</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">','</tt><tt class="py-op">)</tt> </tt>
<a name="L701"></a><tt class="py-lineno">701</tt>  <tt class="py-line"> </tt>
<a name="L702"></a><tt class="py-lineno">702</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">pslx</tt><tt class="py-op">:</tt> </tt>
<a name="L703"></a><tt class="py-lineno">703</tt>  <tt class="py-line">                    <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-272" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-272', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">','</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-273" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-273', 'x', 'link-26');">x</a></tt><tt class="py-op">.</tt><tt id="link-274" class="py-name" targets="Variable Bio.SeqRecord.SeqRecord.seq=Bio.SeqRecord.SeqRecord-class.html#seq,Variable BioSQL.BioSeq.DBSeqRecord.seq=BioSQL.BioSeq.DBSeqRecord-class.html#seq"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-274', 'seq', 'link-274');">seq</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-275" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-275', 'x', 'link-26');">x</a></tt> <tt class="py-keyword">in</tt> <tt id="link-276" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-276', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-277" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.query_all=Bio.SearchIO._model.hsp.HSP-class.html#query_all"><a title="Bio.SearchIO._model.hsp.HSP.query_all" class="py-name" href="#" onclick="return doclink('link-277', 'query_all', 'link-277');">query_all</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">','</tt><tt class="py-op">)</tt> </tt>
<a name="L704"></a><tt class="py-lineno">704</tt>  <tt class="py-line">                    <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-278" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-278', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">','</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-279" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-279', 'x', 'link-26');">x</a></tt><tt class="py-op">.</tt><tt id="link-280" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-280', 'seq', 'link-274');">seq</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-281" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-281', 'x', 'link-26');">x</a></tt> <tt class="py-keyword">in</tt> <tt id="link-282" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-282', 'hsp', 'link-75');">hsp</a></tt><tt class="py-op">.</tt><tt id="link-283" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.hit_all=Bio.SearchIO._model.hsp.HSP-class.html#hit_all"><a title="Bio.SearchIO._model.hsp.HSP.hit_all" class="py-name" href="#" onclick="return doclink('link-283', 'hit_all', 'link-283');">hit_all</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">','</tt><tt class="py-op">)</tt> </tt>
<a name="L705"></a><tt class="py-lineno">705</tt>  <tt class="py-line"> </tt>
<a name="L706"></a><tt class="py-lineno">706</tt>  <tt class="py-line">                <tt class="py-name">qresult_lines</tt><tt class="py-op">.</tt><tt id="link-284" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-284', 'append', 'link-74');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">'\t'</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-285" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-285', 'x', 'link-26');">x</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-286" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-286', 'x', 'link-26');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L707"></a><tt class="py-lineno">707</tt>  <tt class="py-line"> </tt>
<a name="L708"></a><tt class="py-lineno">708</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">'\n'</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">qresult_lines</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">'\n'</tt> </tt>
</div></div><a name="L709"></a><tt class="py-lineno">709</tt>  <tt class="py-line"> </tt>
<a name="L710"></a><tt class="py-lineno">710</tt>  <tt class="py-line"> </tt>
<a name="L711"></a><tt class="py-lineno">711</tt>  <tt class="py-line"><tt class="py-comment"># if not used as a module, run the doctest</tt> </tt>
<a name="L712"></a><tt class="py-lineno">712</tt>  <tt class="py-line"><tt class="py-keyword">if</tt> <tt class="py-name">__name__</tt> <tt class="py-op">==</tt> <tt class="py-string">"__main__"</tt><tt class="py-op">:</tt> </tt>
<a name="L713"></a><tt class="py-lineno">713</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-287" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-287', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-288" class="py-name" targets="Module Bio.Phylo._utils=Bio.Phylo._utils-module.html,Module Bio.SearchIO._utils=Bio.SearchIO._utils-module.html,Module Bio._utils=Bio._utils-module.html"><a title="Bio.Phylo._utils
Bio.SearchIO._utils
Bio._utils" class="py-name" href="#" onclick="return doclink('link-288', '_utils', 'link-288');">_utils</a></tt> <tt class="py-keyword">import</tt> <tt id="link-289" class="py-name" targets="Function Bio._utils.run_doctest()=Bio._utils-module.html#run_doctest"><a title="Bio._utils.run_doctest" class="py-name" href="#" onclick="return doclink('link-289', 'run_doctest', 'link-289');">run_doctest</a></tt> </tt>
<a name="L714"></a><tt class="py-lineno">714</tt>  <tt class="py-line">    <tt id="link-290" class="py-name"><a title="Bio._utils.run_doctest" class="py-name" href="#" onclick="return doclink('link-290', 'run_doctest', 'link-289');">run_doctest</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L715"></a><tt class="py-lineno">715</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Apr  4 10:49:59 2018
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
