<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.PopGen.GenePop.Controller.GenePopController</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.PopGen-module.html">Package&nbsp;PopGen</a> ::
        <a href="Bio.PopGen.GenePop-module.html">Package&nbsp;GenePop</a> ::
        <a href="Bio.PopGen.GenePop.Controller-module.html">Module&nbsp;Controller</a> ::
        Class&nbsp;GenePopController
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.PopGen.GenePop.Controller.GenePopController-class.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== CLASS DESCRIPTION ==================== -->
<h1 class="epydoc">Class GenePopController</h1><p class="nomargin-top"><span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController">source&nbsp;code</a></span></p>
<pre class="base-tree">
object --+
         |
        <strong class="uidshort">GenePopController</strong>
</pre>

<hr />
<!-- ==================== INSTANCE METHODS ==================== -->
<a name="section-InstanceMethods"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Instance Methods</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-InstanceMethods"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.PopGen.GenePop.Controller.GenePopController-class.html#__init__" class="summary-sig-name">__init__</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">genepop_dir</span>=<span class="summary-sig-default">None</span>)</span><br />
      Initialize the controller.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.__init__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_get_opts"></a><span class="summary-sig-name">_get_opts</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">dememorization</span>,
        <span class="summary-sig-arg">batches</span>,
        <span class="summary-sig-arg">iterations</span>,
        <span class="summary-sig-arg">enum_test</span>=<span class="summary-sig-default">None</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController._get_opts">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_run_genepop"></a><span class="summary-sig-name">_run_genepop</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">extensions</span>,
        <span class="summary-sig-arg">option</span>,
        <span class="summary-sig-arg">fname</span>,
        <span class="summary-sig-arg">opts</span>=<span class="summary-sig-default">None</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController._run_genepop">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.PopGen.GenePop.Controller.GenePopController-class.html#_test_pop_hz_both" class="summary-sig-name" onclick="show_private();">_test_pop_hz_both</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">fname</span>,
        <span class="summary-sig-arg">type</span>,
        <span class="summary-sig-arg">ext</span>,
        <span class="summary-sig-arg">enum_test</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">dememorization</span>=<span class="summary-sig-default">10000</span>,
        <span class="summary-sig-arg">batches</span>=<span class="summary-sig-default">20</span>,
        <span class="summary-sig-arg">iterations</span>=<span class="summary-sig-default">5000</span>)</span><br />
      Use Hardy-Weinberg test for heterozygote deficiency/excess.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController._test_pop_hz_both">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.PopGen.GenePop.Controller.GenePopController-class.html#_test_global_hz_both" class="summary-sig-name" onclick="show_private();">_test_global_hz_both</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">fname</span>,
        <span class="summary-sig-arg">type</span>,
        <span class="summary-sig-arg">ext</span>,
        <span class="summary-sig-arg">enum_test</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">dememorization</span>=<span class="summary-sig-default">10000</span>,
        <span class="summary-sig-arg">batches</span>=<span class="summary-sig-default">20</span>,
        <span class="summary-sig-arg">iterations</span>=<span class="summary-sig-default">5000</span>)</span><br />
      Use Global Hardy-Weinberg test for heterozygote deficiency/excess.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController._test_global_hz_both">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.PopGen.GenePop.Controller.GenePopController-class.html#test_pop_hz_deficiency" class="summary-sig-name">test_pop_hz_deficiency</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">fname</span>,
        <span class="summary-sig-arg">enum_test</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">dememorization</span>=<span class="summary-sig-default">10000</span>,
        <span class="summary-sig-arg">batches</span>=<span class="summary-sig-default">20</span>,
        <span class="summary-sig-arg">iterations</span>=<span class="summary-sig-default">5000</span>)</span><br />
      Use Hardy-Weinberg test for heterozygote deficiency.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.test_pop_hz_deficiency">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.PopGen.GenePop.Controller.GenePopController-class.html#test_pop_hz_excess" class="summary-sig-name">test_pop_hz_excess</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">fname</span>,
        <span class="summary-sig-arg">enum_test</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">dememorization</span>=<span class="summary-sig-default">10000</span>,
        <span class="summary-sig-arg">batches</span>=<span class="summary-sig-default">20</span>,
        <span class="summary-sig-arg">iterations</span>=<span class="summary-sig-default">5000</span>)</span><br />
      Use Hardy-Weinberg test for heterozygote deficiency.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.test_pop_hz_excess">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.PopGen.GenePop.Controller.GenePopController-class.html#test_pop_hz_prob" class="summary-sig-name">test_pop_hz_prob</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">fname</span>,
        <span class="summary-sig-arg">ext</span>,
        <span class="summary-sig-arg">enum_test</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">dememorization</span>=<span class="summary-sig-default">10000</span>,
        <span class="summary-sig-arg">batches</span>=<span class="summary-sig-default">20</span>,
        <span class="summary-sig-arg">iterations</span>=<span class="summary-sig-default">5000</span>)</span><br />
      Use Hardy-Weinberg test based on probability.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.test_pop_hz_prob">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.PopGen.GenePop.Controller.GenePopController-class.html#test_global_hz_deficiency" class="summary-sig-name">test_global_hz_deficiency</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">fname</span>,
        <span class="summary-sig-arg">enum_test</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">dememorization</span>=<span class="summary-sig-default">10000</span>,
        <span class="summary-sig-arg">batches</span>=<span class="summary-sig-default">20</span>,
        <span class="summary-sig-arg">iterations</span>=<span class="summary-sig-default">5000</span>)</span><br />
      Use Global Hardy-Weinberg test for heterozygote deficiency.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.test_global_hz_deficiency">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.PopGen.GenePop.Controller.GenePopController-class.html#test_global_hz_excess" class="summary-sig-name">test_global_hz_excess</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">fname</span>,
        <span class="summary-sig-arg">enum_test</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">dememorization</span>=<span class="summary-sig-default">10000</span>,
        <span class="summary-sig-arg">batches</span>=<span class="summary-sig-default">20</span>,
        <span class="summary-sig-arg">iterations</span>=<span class="summary-sig-default">5000</span>)</span><br />
      Use Global Hardy-Weinberg test for heterozygote excess.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.test_global_hz_excess">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="test_ld"></a><span class="summary-sig-name">test_ld</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">fname</span>,
        <span class="summary-sig-arg">dememorization</span>=<span class="summary-sig-default">10000</span>,
        <span class="summary-sig-arg">batches</span>=<span class="summary-sig-default">20</span>,
        <span class="summary-sig-arg">iterations</span>=<span class="summary-sig-default">5000</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.test_ld">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="create_contingency_tables"></a><span class="summary-sig-name">create_contingency_tables</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">fname</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.create_contingency_tables">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="test_genic_diff_all"></a><span class="summary-sig-name">test_genic_diff_all</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">fname</span>,
        <span class="summary-sig-arg">dememorization</span>=<span class="summary-sig-default">10000</span>,
        <span class="summary-sig-arg">batches</span>=<span class="summary-sig-default">20</span>,
        <span class="summary-sig-arg">iterations</span>=<span class="summary-sig-default">5000</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.test_genic_diff_all">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="test_genic_diff_pair"></a><span class="summary-sig-name">test_genic_diff_pair</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">fname</span>,
        <span class="summary-sig-arg">dememorization</span>=<span class="summary-sig-default">10000</span>,
        <span class="summary-sig-arg">batches</span>=<span class="summary-sig-default">20</span>,
        <span class="summary-sig-arg">iterations</span>=<span class="summary-sig-default">5000</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.test_genic_diff_pair">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="test_genotypic_diff_all"></a><span class="summary-sig-name">test_genotypic_diff_all</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">fname</span>,
        <span class="summary-sig-arg">dememorization</span>=<span class="summary-sig-default">10000</span>,
        <span class="summary-sig-arg">batches</span>=<span class="summary-sig-default">20</span>,
        <span class="summary-sig-arg">iterations</span>=<span class="summary-sig-default">5000</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.test_genotypic_diff_all">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="test_genotypic_diff_pair"></a><span class="summary-sig-name">test_genotypic_diff_pair</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">fname</span>,
        <span class="summary-sig-arg">dememorization</span>=<span class="summary-sig-default">10000</span>,
        <span class="summary-sig-arg">batches</span>=<span class="summary-sig-default">20</span>,
        <span class="summary-sig-arg">iterations</span>=<span class="summary-sig-default">5000</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.test_genotypic_diff_pair">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="estimate_nm"></a><span class="summary-sig-name">estimate_nm</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">fname</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.estimate_nm">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.PopGen.GenePop.Controller.GenePopController-class.html#calc_allele_genotype_freqs" class="summary-sig-name">calc_allele_genotype_freqs</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">fname</span>)</span><br />
      Calculate allele and genotype frequencies per locus and per sample.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.calc_allele_genotype_freqs">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_calc_diversities_fis"></a><span class="summary-sig-name">_calc_diversities_fis</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">fname</span>,
        <span class="summary-sig-arg">ext</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController._calc_diversities_fis">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="calc_diversities_fis_with_identity"></a><span class="summary-sig-name">calc_diversities_fis_with_identity</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">fname</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.calc_diversities_fis_with_identity">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="calc_diversities_fis_with_size"></a><span class="summary-sig-name">calc_diversities_fis_with_size</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">fname</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.calc_diversities_fis_with_size">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.PopGen.GenePop.Controller.GenePopController-class.html#calc_fst_all" class="summary-sig-name">calc_fst_all</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">fname</span>)</span><br />
      Execute GenePop and gets Fst/Fis/Fit (all populations).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.calc_fst_all">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="calc_fst_pair"></a><span class="summary-sig-name">calc_fst_pair</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">fname</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.calc_fst_pair">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="calc_rho_all"></a><span class="summary-sig-name">calc_rho_all</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">fname</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.calc_rho_all">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="calc_rho_pair"></a><span class="summary-sig-name">calc_rho_pair</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">fname</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.calc_rho_pair">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_calc_ibd"></a><span class="summary-sig-name">_calc_ibd</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">fname</span>,
        <span class="summary-sig-arg">sub</span>,
        <span class="summary-sig-arg">stat</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">a</code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">scale</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">Log</code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">min_dist</span>=<span class="summary-sig-default">1e-05</span>)</span><br />
      Calculate isolation by distance statistics (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController._calc_ibd">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.PopGen.GenePop.Controller.GenePopController-class.html#calc_ibd_diplo" class="summary-sig-name">calc_ibd_diplo</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">fname</span>,
        <span class="summary-sig-arg">stat</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">a</code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">scale</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">Log</code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">min_dist</span>=<span class="summary-sig-default">1e-05</span>)</span><br />
      Calculate isolation by distance statistics for diploid data.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.calc_ibd_diplo">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.PopGen.GenePop.Controller.GenePopController-class.html#calc_ibd_haplo" class="summary-sig-name">calc_ibd_haplo</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">fname</span>,
        <span class="summary-sig-arg">stat</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">a</code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">scale</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">Log</code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">min_dist</span>=<span class="summary-sig-default">1e-05</span>)</span><br />
      Calculate isolation by distance statistics for haploid data.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.calc_ibd_haplo">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code>object</code></b>:
      <code>__delattr__</code>,
      <code>__format__</code>,
      <code>__getattribute__</code>,
      <code>__hash__</code>,
      <code>__new__</code>,
      <code>__reduce__</code>,
      <code>__reduce_ex__</code>,
      <code>__repr__</code>,
      <code>__setattr__</code>,
      <code>__sizeof__</code>,
      <code>__str__</code>,
      <code>__subclasshook__</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== PROPERTIES ==================== -->
<a name="section-Properties"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Properties</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Properties"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code>object</code></b>:
      <code>__class__</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== METHOD DETAILS ==================== -->
<a name="section-MethodDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Method Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-MethodDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="__init__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__init__</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">genepop_dir</span>=<span class="sig-default">None</span>)</span>
    <br /><em class="fname">(Constructor)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.__init__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Initialize the controller.</p>
<p>genepop_dir is the directory where GenePop is.</p>
<p>The binary should be called Genepop (capital G)</p>
  <dl class="fields">
    <dt>Overrides:
        object.__init__
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="_test_pop_hz_both"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_test_pop_hz_both</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">fname</span>,
        <span class="sig-arg">type</span>,
        <span class="sig-arg">ext</span>,
        <span class="sig-arg">enum_test</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">dememorization</span>=<span class="sig-default">10000</span>,
        <span class="sig-arg">batches</span>=<span class="sig-default">20</span>,
        <span class="sig-arg">iterations</span>=<span class="sig-default">5000</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController._test_pop_hz_both">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Use Hardy-Weinberg test for heterozygote deficiency/excess.</p>
<p>Returns a population iterator containing a dictionary where
dictionary[locus]=(P-val, SE, Fis-WC, Fis-RH, steps).</p>
<p>Some loci have a None if the info is not available.
SE might be none (for enumerations).</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_test_global_hz_both"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_test_global_hz_both</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">fname</span>,
        <span class="sig-arg">type</span>,
        <span class="sig-arg">ext</span>,
        <span class="sig-arg">enum_test</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">dememorization</span>=<span class="sig-default">10000</span>,
        <span class="sig-arg">batches</span>=<span class="sig-default">20</span>,
        <span class="sig-arg">iterations</span>=<span class="sig-default">5000</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController._test_global_hz_both">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Use Global Hardy-Weinberg test for heterozygote deficiency/excess.</p>
<dl class="rst-docutils">
<dt>Returns a triple with:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>A list per population containing (pop_name, P-val, SE, switches).
Some pops have a None if the info is not available.
SE might be none (for enumerations).</li>
<li>A list per loci containing (locus_name, P-val, SE, switches).
Some loci have a None if the info is not available.
SE might be none (for enumerations).</li>
<li>Overall results (P-val, SE, switches).</li>
</ul>
</dd>
</dl>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="test_pop_hz_deficiency"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">test_pop_hz_deficiency</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">fname</span>,
        <span class="sig-arg">enum_test</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">dememorization</span>=<span class="sig-default">10000</span>,
        <span class="sig-arg">batches</span>=<span class="sig-default">20</span>,
        <span class="sig-arg">iterations</span>=<span class="sig-default">5000</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.test_pop_hz_deficiency">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Use Hardy-Weinberg test for heterozygote deficiency.</p>
<p>Returns a population iterator containing a dictionary wehre
dictionary[locus]=(P-val, SE, Fis-WC, Fis-RH, steps).</p>
<p>Some loci have a None if the info is not available.
SE might be none (for enumerations).</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="test_pop_hz_excess"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">test_pop_hz_excess</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">fname</span>,
        <span class="sig-arg">enum_test</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">dememorization</span>=<span class="sig-default">10000</span>,
        <span class="sig-arg">batches</span>=<span class="sig-default">20</span>,
        <span class="sig-arg">iterations</span>=<span class="sig-default">5000</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.test_pop_hz_excess">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Use Hardy-Weinberg test for heterozygote deficiency.</p>
<p>Returns a population iterator containing a dictionary where
dictionary[locus]=(P-val, SE, Fis-WC, Fis-RH, steps).</p>
<p>Some loci have a None if the info is not available.
SE might be none (for enumerations).</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="test_pop_hz_prob"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">test_pop_hz_prob</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">fname</span>,
        <span class="sig-arg">ext</span>,
        <span class="sig-arg">enum_test</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">dememorization</span>=<span class="sig-default">10000</span>,
        <span class="sig-arg">batches</span>=<span class="sig-default">20</span>,
        <span class="sig-arg">iterations</span>=<span class="sig-default">5000</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.test_pop_hz_prob">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Use Hardy-Weinberg test based on probability.</p>
<p>Returns 2 iterators and a final tuple:</p>
<blockquote>
<ol class="rst-arabic">
<li><dl class="rst-first rst-docutils">
<dt>Returns a loci iterator containing:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>A dictionary[pop_pos]=(P-val, SE, Fis-WC, Fis-RH, steps).
Some pops have a None if the info is not available.
SE might be none (for enumerations).</li>
<li>Result of Fisher's test (Chi2, deg freedom, prob).</li>
</ul>
</dd>
</dl>
</li>
<li><dl class="rst-first rst-docutils">
<dt>Returns a population iterator containing:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>A dictionary[locus]=(P-val, SE, Fis-WC, Fis-RH, steps).
Some loci have a None if the info is not available.
SE might be none (for enumerations).</li>
<li>Result of Fisher's test (Chi2, deg freedom, prob).</li>
</ul>
</dd>
</dl>
</li>
<li><p class="rst-first">Final tuple (Chi2, deg freedom, prob).</p>
</li>
</ol>
</blockquote>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="test_global_hz_deficiency"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">test_global_hz_deficiency</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">fname</span>,
        <span class="sig-arg">enum_test</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">dememorization</span>=<span class="sig-default">10000</span>,
        <span class="sig-arg">batches</span>=<span class="sig-default">20</span>,
        <span class="sig-arg">iterations</span>=<span class="sig-default">5000</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.test_global_hz_deficiency">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Use Global Hardy-Weinberg test for heterozygote deficiency.</p>
<dl class="rst-docutils">
<dt>Returns a triple with:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>An list per population containing (pop_name, P-val, SE, switches).
Some pops have a None if the info is not available.
SE might be none (for enumerations).</li>
<li>An list per loci containing (locus_name, P-val, SE, switches).
Some loci have a None if the info is not available.
SE might be none (for enumerations).</li>
<li>Overall results (P-val, SE, switches).</li>
</ul>
</dd>
</dl>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="test_global_hz_excess"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">test_global_hz_excess</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">fname</span>,
        <span class="sig-arg">enum_test</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">dememorization</span>=<span class="sig-default">10000</span>,
        <span class="sig-arg">batches</span>=<span class="sig-default">20</span>,
        <span class="sig-arg">iterations</span>=<span class="sig-default">5000</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.test_global_hz_excess">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Use Global Hardy-Weinberg test for heterozygote excess.</p>
<dl class="rst-docutils">
<dt>Returns a triple with:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>A list per population containing (pop_name, P-val, SE, switches).
Some pops have a None if the info is not available.
SE might be none (for enumerations).</li>
<li>A list per loci containing (locus_name, P-val, SE, switches).
Some loci have a None if the info is not available.
SE might be none (for enumerations).</li>
<li>Overall results (P-val, SE, switches)</li>
</ul>
</dd>
</dl>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="calc_allele_genotype_freqs"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">calc_allele_genotype_freqs</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">fname</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.calc_allele_genotype_freqs">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculate allele and genotype frequencies per locus and per sample.</p>
<dl class="rst-docutils">
<dt>Parameters:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>fname - file name</li>
</ul>
</dd>
<dt>Returns tuple with 2 elements:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>Population iterator with<ul>
<li>population name</li>
<li>Locus dictionary with key = locus name and content tuple as
Genotype List with
(Allele1, Allele2, observed, expected)
(expected homozygotes, observed hm,
expected heterozygotes, observed ht)
Allele frequency/Fis dictionary with allele as key and
(count, frequency, Fis Weir &amp; Cockerham)</li>
<li>Totals as a pair</li>
<li>count</li>
<li>Fis Weir &amp; Cockerham,</li>
<li>Fis Robertson &amp; Hill</li>
</ul>
</li>
<li>Locus iterator with<ul>
<li>Locus name</li>
<li>allele list</li>
<li>Population list with a triple<ul>
<li>population name</li>
<li>list of allele frequencies in the same order as allele list above</li>
<li>number of genes</li>
</ul>
</li>
</ul>
</li>
</ul>
</dd>
</dl>
<p>Will create a file called fname.INF</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="calc_fst_all"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">calc_fst_all</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">fname</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.calc_fst_all">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Execute GenePop and gets Fst/Fis/Fit (all populations).</p>
<dl class="rst-docutils">
<dt>Parameters:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>fname - file name</li>
</ul>
</dd>
<dt>Returns:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>(multiLocusFis, multiLocusFst, multiLocus Fit),</li>
<li>Iterator of tuples
(Locus name, Fis, Fst, Fit, Qintra, Qinter)</li>
</ul>
</dd>
</dl>
<p>Will create a file called <code class="link">fname.FST</code>.</p>
<p>This does not return the genotype frequencies.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="calc_ibd_diplo"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">calc_ibd_diplo</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">fname</span>,
        <span class="sig-arg">stat</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">a</code><code class="variable-quote">'</code></span>,
        <span class="sig-arg">scale</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">Log</code><code class="variable-quote">'</code></span>,
        <span class="sig-arg">min_dist</span>=<span class="sig-default">1e-05</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.calc_ibd_diplo">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculate isolation by distance statistics for diploid data.</p>
<p>See _calc_ibd for parameter details.</p>
<p>Note that each pop can only have a single individual and
the individual name has to be the sample coordinates.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="calc_ibd_haplo"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">calc_ibd_haplo</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">fname</span>,
        <span class="sig-arg">stat</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">a</code><code class="variable-quote">'</code></span>,
        <span class="sig-arg">scale</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">Log</code><code class="variable-quote">'</code></span>,
        <span class="sig-arg">min_dist</span>=<span class="sig-default">1e-05</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.PopGen.GenePop.Controller-pysrc.html#GenePopController.calc_ibd_haplo">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculate isolation by distance statistics for haploid data.</p>
<p>See _calc_ibd for parameter details.</p>
<p>Note that each pop can only have a single individual and
the individual name has to be the sample coordinates.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Fri Jun 22 16:34:42 2018
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
