<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.Graphics.Distribution</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.Graphics-module.html">Package&nbsp;Graphics</a> ::
        Module&nbsp;Distribution
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.Graphics.Distribution-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.Graphics.Distribution-module.html">Module Bio.Graphics.Distribution</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-docstring">"""Display information distributed across a Chromosome-like object.</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-docstring">These classes are meant to show the distribution of some kind of information</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-docstring">as it changes across any kind of segment. It was designed with chromosome</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-docstring">distributions in mind, but could also work for chromosome regions, BAC clones</tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-docstring">or anything similar.</tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-docstring">Reportlab is used for producing the graphical output.</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-comment"># standard library</tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">math</tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-comment"># reportlab</tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">reportlab</tt><tt class="py-op">.</tt><tt class="py-name">lib</tt><tt class="py-op">.</tt><tt class="py-name">pagesizes</tt> <tt class="py-keyword">import</tt> <tt class="py-name">letter</tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">reportlab</tt><tt class="py-op">.</tt><tt class="py-name">lib</tt><tt class="py-op">.</tt><tt class="py-name">units</tt> <tt class="py-keyword">import</tt> <tt class="py-name">inch</tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">reportlab</tt><tt class="py-op">.</tt><tt class="py-name">lib</tt> <tt class="py-keyword">import</tt> <tt class="py-name">colors</tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">reportlab</tt><tt class="py-op">.</tt><tt class="py-name">graphics</tt><tt class="py-op">.</tt><tt class="py-name">shapes</tt> <tt class="py-keyword">import</tt> <tt class="py-name">Drawing</tt><tt class="py-op">,</tt> <tt class="py-name">String</tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">reportlab</tt><tt class="py-op">.</tt><tt class="py-name">graphics</tt><tt class="py-op">.</tt><tt class="py-name">charts</tt><tt class="py-op">.</tt><tt class="py-name">barcharts</tt> <tt class="py-keyword">import</tt> <tt class="py-name">VerticalBarChart</tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">reportlab</tt><tt class="py-op">.</tt><tt class="py-name">graphics</tt><tt class="py-op">.</tt><tt class="py-name">charts</tt><tt class="py-op">.</tt><tt class="py-name">barcharts</tt> <tt class="py-keyword">import</tt> <tt class="py-name">BarChartProperties</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">reportlab</tt><tt class="py-op">.</tt><tt class="py-name">graphics</tt><tt class="py-op">.</tt><tt class="py-name">widgetbase</tt> <tt class="py-keyword">import</tt> <tt class="py-name">TypedPropertyCollection</tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio.Graphics=Bio.Graphics-module.html,Class Bio.KEGG.KGML.KGML_pathway.Graphics=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html"><a title="Bio.Graphics
Bio.KEGG.KGML.KGML_pathway.Graphics" class="py-name" href="#" onclick="return doclink('link-1', 'Graphics', 'link-1');">Graphics</a></tt> <tt class="py-keyword">import</tt> <tt id="link-2" class="py-name" targets="Function Bio.Graphics._write()=Bio.Graphics-module.html#_write"><a title="Bio.Graphics._write" class="py-name" href="#" onclick="return doclink('link-2', '_write', 'link-2');">_write</a></tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"> </tt>
<a name="DistributionPage"></a><div id="DistributionPage-def"><a name="L31"></a><tt class="py-lineno"> 31</tt> <a class="py-toggle" href="#" id="DistributionPage-toggle" onclick="return toggle('DistributionPage');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Graphics.Distribution.DistributionPage-class.html">DistributionPage</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DistributionPage-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="DistributionPage-expanded"><a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line">    <tt class="py-docstring">"""Display a grouping of distributions on a page.</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-docstring">    This organizes Distributions, and will display them nicely</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-docstring">    on a single page.</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"> </tt>
<a name="DistributionPage.__init__"></a><div id="DistributionPage.__init__-def"><a name="L38"></a><tt class="py-lineno"> 38</tt> <a class="py-toggle" href="#" id="DistributionPage.__init__-toggle" onclick="return toggle('DistributionPage.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.Distribution.DistributionPage-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">output_format</tt><tt class="py-op">=</tt><tt class="py-string">'pdf'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DistributionPage.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DistributionPage.__init__-expanded"><a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize."""</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">distributions</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line">        <tt class="py-comment"># customizable attributes</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">number_of_columns</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">page_size</tt> <tt class="py-op">=</tt> <tt class="py-name">letter</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">title_size</tt> <tt class="py-op">=</tt> <tt class="py-number">20</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line"> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">output_format</tt> <tt class="py-op">=</tt> <tt class="py-name">output_format</tt> </tt>
</div><a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line"> </tt>
<a name="DistributionPage.draw"></a><div id="DistributionPage.draw-def"><a name="L49"></a><tt class="py-lineno"> 49</tt> <a class="py-toggle" href="#" id="DistributionPage.draw-toggle" onclick="return toggle('DistributionPage.draw');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.Distribution.DistributionPage-class.html#draw">draw</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">output_file</tt><tt class="py-op">,</tt> <tt class="py-param">title</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DistributionPage.draw-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DistributionPage.draw-expanded"><a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line">        <tt class="py-docstring">"""Draw out the distribution information.</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line"><tt class="py-docstring">         - output_file - The name of the file to output the information to,</tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line"><tt class="py-docstring">           or a handle to write to.</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line"><tt class="py-docstring">         - title - A title to display on the graphic.</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line">        <tt id="link-3" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.width=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#width,Method Bio.Phylo.PhyloXMLIO.Writer.width()=Bio.Phylo.PhyloXMLIO.Writer-class.html#width"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.width
Bio.Phylo.PhyloXMLIO.Writer.width" class="py-name" href="#" onclick="return doclink('link-3', 'width', 'link-3');">width</a></tt><tt class="py-op">,</tt> <tt id="link-4" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.height=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#height"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.height" class="py-name" href="#" onclick="return doclink('link-4', 'height', 'link-4');">height</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">page_size</tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line">        <tt class="py-name">cur_drawing</tt> <tt class="py-op">=</tt> <tt class="py-name">Drawing</tt><tt class="py-op">(</tt><tt id="link-5" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.width
Bio.Phylo.PhyloXMLIO.Writer.width" class="py-name" href="#" onclick="return doclink('link-5', 'width', 'link-3');">width</a></tt><tt class="py-op">,</tt> <tt id="link-6" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.height" class="py-name" href="#" onclick="return doclink('link-6', 'height', 'link-4');">height</a></tt><tt class="py-op">)</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line"> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-7" class="py-name" targets="Method Bio.Graphics.BasicChromosome.Organism._draw_title()=Bio.Graphics.BasicChromosome.Organism-class.html#_draw_title,Method Bio.Graphics.Comparative.ComparativeScatterPlot._draw_title()=Bio.Graphics.Comparative.ComparativeScatterPlot-class.html#_draw_title,Method Bio.Graphics.Distribution.BarChartDistribution._draw_title()=Bio.Graphics.Distribution.BarChartDistribution-class.html#_draw_title,Method Bio.Graphics.Distribution.DistributionPage._draw_title()=Bio.Graphics.Distribution.DistributionPage-class.html#_draw_title"><a title="Bio.Graphics.BasicChromosome.Organism._draw_title
Bio.Graphics.Comparative.ComparativeScatterPlot._draw_title
Bio.Graphics.Distribution.BarChartDistribution._draw_title
Bio.Graphics.Distribution.DistributionPage._draw_title" class="py-name" href="#" onclick="return doclink('link-7', '_draw_title', 'link-7');">_draw_title</a></tt><tt class="py-op">(</tt><tt class="py-name">cur_drawing</tt><tt class="py-op">,</tt> <tt id="link-8" class="py-name" targets="Method Bio.Blast.NCBIStandalone._AlignmentConsumer.title()=Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#title,Method Bio.GenBank._FeatureConsumer.title()=Bio.GenBank._FeatureConsumer-class.html#title,Method Bio.GenBank._RecordConsumer.title()=Bio.GenBank._RecordConsumer-class.html#title"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-8', 'title', 'link-8');">title</a></tt><tt class="py-op">,</tt> <tt id="link-9" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.width
Bio.Phylo.PhyloXMLIO.Writer.width" class="py-name" href="#" onclick="return doclink('link-9', 'width', 'link-3');">width</a></tt><tt class="py-op">,</tt> <tt id="link-10" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.height" class="py-name" href="#" onclick="return doclink('link-10', 'height', 'link-4');">height</a></tt><tt class="py-op">)</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line"> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line">        <tt class="py-comment"># calculate the x and y position changes for each distribution</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line">        <tt class="py-name">cur_x_pos</tt> <tt class="py-op">=</tt> <tt class="py-name">inch</tt> <tt class="py-op">*</tt> <tt class="py-number">.5</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line">        <tt class="py-name">end_x_pos</tt> <tt class="py-op">=</tt> <tt id="link-11" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.width
Bio.Phylo.PhyloXMLIO.Writer.width" class="py-name" href="#" onclick="return doclink('link-11', 'width', 'link-3');">width</a></tt> <tt class="py-op">-</tt> <tt class="py-name">inch</tt> <tt class="py-op">*</tt> <tt class="py-number">.5</tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line">        <tt class="py-name">cur_y_pos</tt> <tt class="py-op">=</tt> <tt id="link-12" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.height" class="py-name" href="#" onclick="return doclink('link-12', 'height', 'link-4');">height</a></tt> <tt class="py-op">-</tt> <tt class="py-number">1.5</tt> <tt class="py-op">*</tt> <tt class="py-name">inch</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line">        <tt class="py-name">end_y_pos</tt> <tt class="py-op">=</tt> <tt class="py-number">.5</tt> <tt class="py-op">*</tt> <tt class="py-name">inch</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line">        <tt class="py-name">x_pos_change</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">end_x_pos</tt> <tt class="py-op">-</tt> <tt class="py-name">cur_x_pos</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line">                        <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">number_of_columns</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line">        <tt class="py-name">num_y_rows</tt> <tt class="py-op">=</tt> <tt class="py-name">math</tt><tt class="py-op">.</tt><tt class="py-name">ceil</tt><tt class="py-op">(</tt><tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">distributions</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line">                               <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">number_of_columns</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line">        <tt class="py-name">y_pos_change</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">cur_y_pos</tt> <tt class="py-op">-</tt> <tt class="py-name">end_y_pos</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-name">num_y_rows</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line"> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-13" class="py-name" targets="Method Bio.Graphics.Distribution.DistributionPage._draw_distributions()=Bio.Graphics.Distribution.DistributionPage-class.html#_draw_distributions"><a title="Bio.Graphics.Distribution.DistributionPage._draw_distributions" class="py-name" href="#" onclick="return doclink('link-13', '_draw_distributions', 'link-13');">_draw_distributions</a></tt><tt class="py-op">(</tt><tt class="py-name">cur_drawing</tt><tt class="py-op">,</tt> <tt class="py-name">cur_x_pos</tt><tt class="py-op">,</tt> <tt class="py-name">x_pos_change</tt><tt class="py-op">,</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line">                                 <tt class="py-name">cur_y_pos</tt><tt class="py-op">,</tt> <tt class="py-name">y_pos_change</tt><tt class="py-op">,</tt> <tt class="py-name">num_y_rows</tt><tt class="py-op">)</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-14" class="py-name" targets="Method Bio.Graphics.BasicChromosome.Organism._draw_legend()=Bio.Graphics.BasicChromosome.Organism-class.html#_draw_legend,Method Bio.Graphics.Distribution.DistributionPage._draw_legend()=Bio.Graphics.Distribution.DistributionPage-class.html#_draw_legend"><a title="Bio.Graphics.BasicChromosome.Organism._draw_legend
Bio.Graphics.Distribution.DistributionPage._draw_legend" class="py-name" href="#" onclick="return doclink('link-14', '_draw_legend', 'link-14');">_draw_legend</a></tt><tt class="py-op">(</tt><tt class="py-name">cur_drawing</tt><tt class="py-op">,</tt> <tt class="py-number">2.5</tt> <tt class="py-op">*</tt> <tt class="py-name">inch</tt><tt class="py-op">,</tt> <tt id="link-15" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.width
Bio.Phylo.PhyloXMLIO.Writer.width" class="py-name" href="#" onclick="return doclink('link-15', 'width', 'link-3');">width</a></tt><tt class="py-op">)</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line"> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-16" class="py-name"><a title="Bio.Graphics._write" class="py-name" href="#" onclick="return doclink('link-16', '_write', 'link-2');">_write</a></tt><tt class="py-op">(</tt><tt class="py-name">cur_drawing</tt><tt class="py-op">,</tt> <tt class="py-name">output_file</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">output_format</tt><tt class="py-op">)</tt> </tt>
</div><a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line"> </tt>
<a name="DistributionPage._draw_title"></a><div id="DistributionPage._draw_title-def"><a name="L80"></a><tt class="py-lineno"> 80</tt> <a class="py-toggle" href="#" id="DistributionPage._draw_title-toggle" onclick="return toggle('DistributionPage._draw_title');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.Distribution.DistributionPage-class.html#_draw_title">_draw_title</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">cur_drawing</tt><tt class="py-op">,</tt> <tt class="py-param">title</tt><tt class="py-op">,</tt> <tt class="py-param">width</tt><tt class="py-op">,</tt> <tt class="py-param">height</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DistributionPage._draw_title-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DistributionPage._draw_title-expanded"><a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add the title of the figure to the drawing (PRIVATE)."""</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line">        <tt class="py-name">title_string</tt> <tt class="py-op">=</tt> <tt class="py-name">String</tt><tt class="py-op">(</tt><tt id="link-17" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.width
Bio.Phylo.PhyloXMLIO.Writer.width" class="py-name" href="#" onclick="return doclink('link-17', 'width', 'link-3');">width</a></tt> <tt class="py-op">/</tt> <tt class="py-number">2</tt><tt class="py-op">,</tt> <tt id="link-18" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.height" class="py-name" href="#" onclick="return doclink('link-18', 'height', 'link-4');">height</a></tt> <tt class="py-op">-</tt> <tt class="py-name">inch</tt><tt class="py-op">,</tt> <tt id="link-19" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-19', 'title', 'link-8');">title</a></tt><tt class="py-op">)</tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line">        <tt class="py-name">title_string</tt><tt class="py-op">.</tt><tt class="py-name">fontName</tt> <tt class="py-op">=</tt> <tt class="py-string">'Helvetica-Bold'</tt> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line">        <tt class="py-name">title_string</tt><tt class="py-op">.</tt><tt class="py-name">fontSize</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">title_size</tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line">        <tt class="py-name">title_string</tt><tt class="py-op">.</tt><tt class="py-name">textAnchor</tt> <tt class="py-op">=</tt> <tt class="py-string">"middle"</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line"> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line">        <tt class="py-name">cur_drawing</tt><tt class="py-op">.</tt><tt id="link-20" class="py-name" targets="Method Bio.Graphics.BasicChromosome._ChromosomeComponent.add()=Bio.Graphics.BasicChromosome._ChromosomeComponent-class.html#add,Method Bio.Nexus.Nexus.StepMatrix.add()=Bio.Nexus.Nexus.StepMatrix-class.html#add,Method Bio.Nexus.Nodes.Chain.add()=Bio.Nexus.Nodes.Chain-class.html#add,Method Bio.PDB.Entity.Entity.add()=Bio.PDB.Entity.Entity-class.html#add,Method Bio.PDB.Residue.DisorderedResidue.add()=Bio.PDB.Residue.DisorderedResidue-class.html#add,Method Bio.PDB.Residue.Residue.add()=Bio.PDB.Residue.Residue-class.html#add,Method Bio.Restriction.Restriction.RestrictionBatch.add()=Bio.Restriction.Restriction.RestrictionBatch-class.html#add"><a title="Bio.Graphics.BasicChromosome._ChromosomeComponent.add
Bio.Nexus.Nexus.StepMatrix.add
Bio.Nexus.Nodes.Chain.add
Bio.PDB.Entity.Entity.add
Bio.PDB.Residue.DisorderedResidue.add
Bio.PDB.Residue.Residue.add
Bio.Restriction.Restriction.RestrictionBatch.add" class="py-name" href="#" onclick="return doclink('link-20', 'add', 'link-20');">add</a></tt><tt class="py-op">(</tt><tt class="py-name">title_string</tt><tt class="py-op">)</tt> </tt>
</div><a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line"> </tt>
<a name="DistributionPage._draw_distributions"></a><div id="DistributionPage._draw_distributions-def"><a name="L89"></a><tt class="py-lineno"> 89</tt> <a class="py-toggle" href="#" id="DistributionPage._draw_distributions-toggle" onclick="return toggle('DistributionPage._draw_distributions');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.Distribution.DistributionPage-class.html#_draw_distributions">_draw_distributions</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">cur_drawing</tt><tt class="py-op">,</tt> <tt class="py-param">start_x_pos</tt><tt class="py-op">,</tt> <tt class="py-param">x_pos_change</tt><tt class="py-op">,</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line">                            <tt class="py-param">start_y_pos</tt><tt class="py-op">,</tt> <tt class="py-param">y_pos_change</tt><tt class="py-op">,</tt> <tt class="py-param">num_y_drawings</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DistributionPage._draw_distributions-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DistributionPage._draw_distributions-expanded"><a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line">        <tt class="py-docstring">"""Draw all of the distributions on the page.</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line"><tt class="py-docstring">         - cur_drawing - The drawing we are working with.</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line"><tt class="py-docstring">         - start_x_pos - The x position on the page to start drawing at.</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line"><tt class="py-docstring">         - x_pos_change - The change in x position between each figure.</tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line"><tt class="py-docstring">         - start_y_pos - The y position on the page to start drawing at.</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line"><tt class="py-docstring">         - y_pos_change - The change in y position between each figure.</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line"><tt class="py-docstring">         - num_y_drawings - The number of drawings we'll have in the y</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line"><tt class="py-docstring">           (up/down) direction.</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">y_drawing</tt> <tt class="py-keyword">in</tt> <tt id="link-21" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.range()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#range,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.range()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#range,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Track.Track.range()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#range"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-21', 'range', 'link-21');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">num_y_drawings</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line">            <tt class="py-comment"># if we are on the last y position, we may not be able</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">            <tt class="py-comment"># to fill all of the x columns</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-op">(</tt><tt class="py-name">y_drawing</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">number_of_columns</tt> <tt class="py-op">&gt;</tt> \ </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line">               <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">distributions</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line">                <tt class="py-name">num_x_drawings</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">distributions</tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> \ </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line">                                 <tt class="py-name">y_drawing</tt> <tt class="py-op">*</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">number_of_columns</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line">                <tt class="py-name">num_x_drawings</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">number_of_columns</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">x_drawing</tt> <tt class="py-keyword">in</tt> <tt id="link-22" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-22', 'range', 'link-21');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">num_x_drawings</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">                <tt class="py-name">dist_num</tt> <tt class="py-op">=</tt> <tt class="py-name">y_drawing</tt> <tt class="py-op">*</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">number_of_columns</tt> <tt class="py-op">+</tt> <tt class="py-name">x_drawing</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">                <tt class="py-name">cur_distribution</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">distributions</tt><tt class="py-op">[</tt><tt class="py-name">dist_num</tt><tt class="py-op">]</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line"> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">                <tt class="py-comment"># find the x and y boundaries of the distribution</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line">                <tt class="py-name">x_pos</tt> <tt class="py-op">=</tt> <tt class="py-name">start_x_pos</tt> <tt class="py-op">+</tt> <tt class="py-name">x_drawing</tt> <tt class="py-op">*</tt> <tt class="py-name">x_pos_change</tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line">                <tt class="py-name">end_x_pos</tt> <tt class="py-op">=</tt> <tt class="py-name">x_pos</tt> <tt class="py-op">+</tt> <tt class="py-name">x_pos_change</tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line">                <tt class="py-name">end_y_pos</tt> <tt class="py-op">=</tt> <tt class="py-name">start_y_pos</tt> <tt class="py-op">-</tt> <tt class="py-name">y_drawing</tt> <tt class="py-op">*</tt> <tt class="py-name">y_pos_change</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">                <tt class="py-name">y_pos</tt> <tt class="py-op">=</tt> <tt class="py-name">end_y_pos</tt> <tt class="py-op">-</tt> <tt class="py-name">y_pos_change</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line"> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line">                <tt class="py-comment"># draw the distribution</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line">                <tt class="py-name">cur_distribution</tt><tt class="py-op">.</tt><tt id="link-23" class="py-name" targets="Method Bio.Graphics.BasicChromosome.Chromosome.draw()=Bio.Graphics.BasicChromosome.Chromosome-class.html#draw,Method Bio.Graphics.BasicChromosome.ChromosomeSegment.draw()=Bio.Graphics.BasicChromosome.ChromosomeSegment-class.html#draw,Method Bio.Graphics.BasicChromosome.Organism.draw()=Bio.Graphics.BasicChromosome.Organism-class.html#draw,Method Bio.Graphics.BasicChromosome.SpacerSegment.draw()=Bio.Graphics.BasicChromosome.SpacerSegment-class.html#draw,Method Bio.Graphics.BasicChromosome._ChromosomeComponent.draw()=Bio.Graphics.BasicChromosome._ChromosomeComponent-class.html#draw,Method Bio.Graphics.Distribution.BarChartDistribution.draw()=Bio.Graphics.Distribution.BarChartDistribution-class.html#draw,Method Bio.Graphics.Distribution.DistributionPage.draw()=Bio.Graphics.Distribution.DistributionPage-class.html#draw,Method Bio.Graphics.Distribution.LineDistribution.draw()=Bio.Graphics.Distribution.LineDistribution-class.html#draw,Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.draw()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#draw,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#draw,Method Bio.Graphics.KGML_vis.KGMLCanvas.draw()=Bio.Graphics.KGML_vis.KGMLCanvas-class.html#draw,Function Bio.Phylo._utils.draw()=Bio.Phylo._utils-module.html#draw"><a title="Bio.Graphics.BasicChromosome.Chromosome.draw
Bio.Graphics.BasicChromosome.ChromosomeSegment.draw
Bio.Graphics.BasicChromosome.Organism.draw
Bio.Graphics.BasicChromosome.SpacerSegment.draw
Bio.Graphics.BasicChromosome._ChromosomeComponent.draw
Bio.Graphics.Distribution.BarChartDistribution.draw
Bio.Graphics.Distribution.DistributionPage.draw
Bio.Graphics.Distribution.LineDistribution.draw
Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw
Bio.Graphics.GenomeDiagram._Diagram.Diagram.draw
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw
Bio.Graphics.KGML_vis.KGMLCanvas.draw
Bio.Phylo._utils.draw" class="py-name" href="#" onclick="return doclink('link-23', 'draw', 'link-23');">draw</a></tt><tt class="py-op">(</tt><tt class="py-name">cur_drawing</tt><tt class="py-op">,</tt> <tt class="py-name">x_pos</tt><tt class="py-op">,</tt> <tt class="py-name">y_pos</tt><tt class="py-op">,</tt> <tt class="py-name">end_x_pos</tt><tt class="py-op">,</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line">                                      <tt class="py-name">end_y_pos</tt><tt class="py-op">)</tt> </tt>
</div><a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line"> </tt>
<a name="DistributionPage._draw_legend"></a><div id="DistributionPage._draw_legend-def"><a name="L126"></a><tt class="py-lineno">126</tt> <a class="py-toggle" href="#" id="DistributionPage._draw_legend-toggle" onclick="return toggle('DistributionPage._draw_legend');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.Distribution.DistributionPage-class.html#_draw_legend">_draw_legend</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">cur_drawing</tt><tt class="py-op">,</tt> <tt class="py-param">start_y</tt><tt class="py-op">,</tt> <tt class="py-param">width</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DistributionPage._draw_legend-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DistributionPage._draw_legend-expanded"><a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add a legend to the figure (PRIVATE).</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line"><tt class="py-docstring">        Subclasses can implement to provide a specialized legend.</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div></div><a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line"> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line"> </tt>
<a name="BarChartDistribution"></a><div id="BarChartDistribution-def"><a name="L134"></a><tt class="py-lineno">134</tt> <a class="py-toggle" href="#" id="BarChartDistribution-toggle" onclick="return toggle('BarChartDistribution');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Graphics.Distribution.BarChartDistribution-class.html">BarChartDistribution</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BarChartDistribution-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="BarChartDistribution-expanded"><a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line">    <tt class="py-docstring">"""Display the distribution of values as a bunch of bars."""</tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line"> </tt>
<a name="BarChartDistribution.__init__"></a><div id="BarChartDistribution.__init__-def"><a name="L137"></a><tt class="py-lineno">137</tt> <a class="py-toggle" href="#" id="BarChartDistribution.__init__-toggle" onclick="return toggle('BarChartDistribution.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.Distribution.BarChartDistribution-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">display_info</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BarChartDistribution.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BarChartDistribution.__init__-expanded"><a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize a Bar Chart display of distribution info.</tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line"><tt class="py-docstring">        Attributes:</tt> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line"><tt class="py-docstring">         - display_info - the information to be displayed in the distribution.</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line"><tt class="py-docstring">           This should be ordered as a list of lists, where each internal list</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line"><tt class="py-docstring">           is a data set to display in the bar chart.</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">display_info</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line">            <tt class="py-name">display_info</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">display_info</tt> <tt class="py-op">=</tt> <tt class="py-name">display_info</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line"> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">x_axis_title</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">y_axis_title</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chart_title</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chart_title_size</tt> <tt class="py-op">=</tt> <tt class="py-number">10</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line"> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">padding_percent</tt> <tt class="py-op">=</tt> <tt class="py-number">0.15</tt> </tt>
</div><a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line"> </tt>
<a name="BarChartDistribution.draw"></a><div id="BarChartDistribution.draw-def"><a name="L157"></a><tt class="py-lineno">157</tt> <a class="py-toggle" href="#" id="BarChartDistribution.draw-toggle" onclick="return toggle('BarChartDistribution.draw');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.Distribution.BarChartDistribution-class.html#draw">draw</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">cur_drawing</tt><tt class="py-op">,</tt> <tt class="py-param">start_x</tt><tt class="py-op">,</tt> <tt class="py-param">start_y</tt><tt class="py-op">,</tt> <tt class="py-param">end_x</tt><tt class="py-op">,</tt> <tt class="py-param">end_y</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BarChartDistribution.draw-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BarChartDistribution.draw-expanded"><a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">        <tt class="py-docstring">"""Draw a bar chart with the info in the specified range."""</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">        <tt class="py-name">bar_chart</tt> <tt class="py-op">=</tt> <tt class="py-name">VerticalBarChart</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chart_title</tt><tt class="py-op">:</tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-24" class="py-name"><a title="Bio.Graphics.BasicChromosome.Organism._draw_title
Bio.Graphics.Comparative.ComparativeScatterPlot._draw_title
Bio.Graphics.Distribution.BarChartDistribution._draw_title
Bio.Graphics.Distribution.DistributionPage._draw_title" class="py-name" href="#" onclick="return doclink('link-24', '_draw_title', 'link-7');">_draw_title</a></tt><tt class="py-op">(</tt><tt class="py-name">cur_drawing</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chart_title</tt><tt class="py-op">,</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">                             <tt class="py-name">start_x</tt><tt class="py-op">,</tt> <tt class="py-name">start_y</tt><tt class="py-op">,</tt> <tt class="py-name">end_x</tt><tt class="py-op">,</tt> <tt class="py-name">end_y</tt><tt class="py-op">)</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line">        <tt class="py-comment"># set the position of the bar chart</tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">        <tt class="py-name">x_start</tt><tt class="py-op">,</tt> <tt class="py-name">x_end</tt><tt class="py-op">,</tt> <tt class="py-name">y_start</tt><tt class="py-op">,</tt> <tt class="py-name">y_end</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-25" class="py-name" targets="Method Bio.Graphics.Distribution.BarChartDistribution._determine_position()=Bio.Graphics.Distribution.BarChartDistribution-class.html#_determine_position"><a title="Bio.Graphics.Distribution.BarChartDistribution._determine_position" class="py-name" href="#" onclick="return doclink('link-25', '_determine_position', 'link-25');">_determine_position</a></tt><tt class="py-op">(</tt><tt class="py-name">start_x</tt><tt class="py-op">,</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line">                                                                  <tt class="py-name">start_y</tt><tt class="py-op">,</tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line">                                                                  <tt class="py-name">end_x</tt><tt class="py-op">,</tt> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">                                                                  <tt class="py-name">end_y</tt><tt class="py-op">)</tt> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line"> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line">        <tt class="py-name">bar_chart</tt><tt class="py-op">.</tt><tt id="link-26" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.x=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#x,Variable Bio.Statistics.lowess.x=Bio.Statistics.lowess-module.html#x"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-26', 'x', 'link-26');">x</a></tt> <tt class="py-op">=</tt> <tt class="py-name">x_start</tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line">        <tt class="py-name">bar_chart</tt><tt class="py-op">.</tt><tt id="link-27" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.y=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#y"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-27', 'y', 'link-27');">y</a></tt> <tt class="py-op">=</tt> <tt class="py-name">y_start</tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line">        <tt class="py-name">bar_chart</tt><tt class="py-op">.</tt><tt id="link-28" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.width
Bio.Phylo.PhyloXMLIO.Writer.width" class="py-name" href="#" onclick="return doclink('link-28', 'width', 'link-3');">width</a></tt> <tt class="py-op">=</tt> <tt id="link-29" class="py-name" targets="Method Bio.FSSP.FSSPAlignDict.abs()=Bio.FSSP.FSSPAlignDict-class.html#abs"><a title="Bio.FSSP.FSSPAlignDict.abs" class="py-name" href="#" onclick="return doclink('link-29', 'abs', 'link-29');">abs</a></tt><tt class="py-op">(</tt><tt class="py-name">x_start</tt> <tt class="py-op">-</tt> <tt class="py-name">x_end</tt><tt class="py-op">)</tt> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line">        <tt class="py-name">bar_chart</tt><tt class="py-op">.</tt><tt id="link-30" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.height" class="py-name" href="#" onclick="return doclink('link-30', 'height', 'link-4');">height</a></tt> <tt class="py-op">=</tt> <tt id="link-31" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.abs" class="py-name" href="#" onclick="return doclink('link-31', 'abs', 'link-29');">abs</a></tt><tt class="py-op">(</tt><tt class="py-name">y_start</tt> <tt class="py-op">-</tt> <tt class="py-name">y_end</tt><tt class="py-op">)</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line"> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line">        <tt class="py-comment"># set the information in the bar chart</tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line">        <tt class="py-name">bar_chart</tt><tt class="py-op">.</tt><tt id="link-32" class="py-name" targets="Variable BioSQL.BioSeq.DBSeq.data=BioSQL.BioSeq.DBSeq-class.html#data"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-32', 'data', 'link-32');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">display_info</tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">        <tt class="py-name">bar_chart</tt><tt class="py-op">.</tt><tt class="py-name">valueAxis</tt><tt class="py-op">.</tt><tt class="py-name">valueMin</tt> <tt class="py-op">=</tt> <tt id="link-33" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.min=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#min"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-33', 'min', 'link-33');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">display_info</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line">        <tt class="py-name">bar_chart</tt><tt class="py-op">.</tt><tt class="py-name">valueAxis</tt><tt class="py-op">.</tt><tt class="py-name">valueMax</tt> <tt class="py-op">=</tt> <tt id="link-34" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.max=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#max"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-34', 'max', 'link-34');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">display_info</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">data_set</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">display_info</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-35" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-35', 'min', 'link-33');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">data_set</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">bar_chart</tt><tt class="py-op">.</tt><tt class="py-name">valueAxis</tt><tt class="py-op">.</tt><tt class="py-name">valueMin</tt><tt class="py-op">:</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line">                <tt class="py-name">bar_chart</tt><tt class="py-op">.</tt><tt class="py-name">valueAxis</tt><tt class="py-op">.</tt><tt class="py-name">valueMin</tt> <tt class="py-op">=</tt> <tt id="link-36" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-36', 'min', 'link-33');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">data_set</tt><tt class="py-op">)</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-37" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-37', 'max', 'link-34');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">data_set</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-name">bar_chart</tt><tt class="py-op">.</tt><tt class="py-name">valueAxis</tt><tt class="py-op">.</tt><tt class="py-name">valueMax</tt><tt class="py-op">:</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">                <tt class="py-name">bar_chart</tt><tt class="py-op">.</tt><tt class="py-name">valueAxis</tt><tt class="py-op">.</tt><tt class="py-name">valueMax</tt> <tt class="py-op">=</tt> <tt id="link-38" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-38', 'max', 'link-34');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">data_set</tt><tt class="py-op">)</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line"> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">        <tt class="py-comment"># set other formatting options</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">display_info</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line">            <tt class="py-name">bar_chart</tt><tt class="py-op">.</tt><tt class="py-name">groupSpacing</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">            <tt class="py-name">style</tt> <tt class="py-op">=</tt> <tt class="py-name">TypedPropertyCollection</tt><tt class="py-op">(</tt><tt class="py-name">BarChartProperties</tt><tt class="py-op">)</tt> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line">            <tt class="py-name">style</tt><tt class="py-op">.</tt><tt class="py-name">strokeWidth</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line">            <tt class="py-name">style</tt><tt class="py-op">.</tt><tt class="py-name">strokeColor</tt> <tt class="py-op">=</tt> <tt class="py-name">colors</tt><tt class="py-op">.</tt><tt id="link-39" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.green()=Bio.Phylo.PhyloXMLIO.Writer-class.html#green"><a title="Bio.Phylo.PhyloXMLIO.Writer.green" class="py-name" href="#" onclick="return doclink('link-39', 'green', 'link-39');">green</a></tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line">            <tt class="py-name">style</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">fillColor</tt> <tt class="py-op">=</tt> <tt class="py-name">colors</tt><tt class="py-op">.</tt><tt id="link-40" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.green" class="py-name" href="#" onclick="return doclink('link-40', 'green', 'link-39');">green</a></tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line"> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">            <tt class="py-name">bar_chart</tt><tt class="py-op">.</tt><tt class="py-name">bars</tt> <tt class="py-op">=</tt> <tt class="py-name">style</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line"> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">        <tt class="py-comment"># set the labels</tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line">        <tt class="py-comment"># XXX labels don't work yet</tt> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">        <tt class="py-comment"># bar_chart.valueAxis.title = self.x_axis_title</tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line">        <tt class="py-comment"># bar_chart.categoryAxis.title = self.y_axis_title</tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line"> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line">        <tt class="py-name">cur_drawing</tt><tt class="py-op">.</tt><tt id="link-41" class="py-name"><a title="Bio.Graphics.BasicChromosome._ChromosomeComponent.add
Bio.Nexus.Nexus.StepMatrix.add
Bio.Nexus.Nodes.Chain.add
Bio.PDB.Entity.Entity.add
Bio.PDB.Residue.DisorderedResidue.add
Bio.PDB.Residue.Residue.add
Bio.Restriction.Restriction.RestrictionBatch.add" class="py-name" href="#" onclick="return doclink('link-41', 'add', 'link-20');">add</a></tt><tt class="py-op">(</tt><tt class="py-name">bar_chart</tt><tt class="py-op">)</tt> </tt>
</div><a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line"> </tt>
<a name="BarChartDistribution._draw_title"></a><div id="BarChartDistribution._draw_title-def"><a name="L201"></a><tt class="py-lineno">201</tt> <a class="py-toggle" href="#" id="BarChartDistribution._draw_title-toggle" onclick="return toggle('BarChartDistribution._draw_title');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.Distribution.BarChartDistribution-class.html#_draw_title">_draw_title</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">cur_drawing</tt><tt class="py-op">,</tt> <tt class="py-param">title</tt><tt class="py-op">,</tt> <tt class="py-param">start_x</tt><tt class="py-op">,</tt> <tt class="py-param">start_y</tt><tt class="py-op">,</tt> <tt class="py-param">end_x</tt><tt class="py-op">,</tt> <tt class="py-param">end_y</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BarChartDistribution._draw_title-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BarChartDistribution._draw_title-expanded"><a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add the title of the figure to the drawing (PRIVATE)."""</tt> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line">        <tt class="py-name">x_center</tt> <tt class="py-op">=</tt> <tt class="py-name">start_x</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt class="py-name">end_x</tt> <tt class="py-op">-</tt> <tt class="py-name">start_x</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-number">2</tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line">        <tt class="py-name">y_pos</tt> <tt class="py-op">=</tt> <tt class="py-name">end_y</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">padding_percent</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-name">start_y</tt> <tt class="py-op">-</tt> <tt class="py-name">end_y</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-number">2</tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line">        <tt class="py-name">title_string</tt> <tt class="py-op">=</tt> <tt class="py-name">String</tt><tt class="py-op">(</tt><tt class="py-name">x_center</tt><tt class="py-op">,</tt> <tt class="py-name">y_pos</tt><tt class="py-op">,</tt> <tt id="link-42" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-42', 'title', 'link-8');">title</a></tt><tt class="py-op">)</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line">        <tt class="py-name">title_string</tt><tt class="py-op">.</tt><tt class="py-name">fontName</tt> <tt class="py-op">=</tt> <tt class="py-string">'Helvetica-Bold'</tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line">        <tt class="py-name">title_string</tt><tt class="py-op">.</tt><tt class="py-name">fontSize</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chart_title_size</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line">        <tt class="py-name">title_string</tt><tt class="py-op">.</tt><tt class="py-name">textAnchor</tt> <tt class="py-op">=</tt> <tt class="py-string">"middle"</tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line"> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line">        <tt class="py-name">cur_drawing</tt><tt class="py-op">.</tt><tt id="link-43" class="py-name"><a title="Bio.Graphics.BasicChromosome._ChromosomeComponent.add
Bio.Nexus.Nexus.StepMatrix.add
Bio.Nexus.Nodes.Chain.add
Bio.PDB.Entity.Entity.add
Bio.PDB.Residue.DisorderedResidue.add
Bio.PDB.Residue.Residue.add
Bio.Restriction.Restriction.RestrictionBatch.add" class="py-name" href="#" onclick="return doclink('link-43', 'add', 'link-20');">add</a></tt><tt class="py-op">(</tt><tt class="py-name">title_string</tt><tt class="py-op">)</tt> </tt>
</div><a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line"> </tt>
<a name="BarChartDistribution._determine_position"></a><div id="BarChartDistribution._determine_position-def"><a name="L212"></a><tt class="py-lineno">212</tt> <a class="py-toggle" href="#" id="BarChartDistribution._determine_position-toggle" onclick="return toggle('BarChartDistribution._determine_position');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.Distribution.BarChartDistribution-class.html#_determine_position">_determine_position</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">start_x</tt><tt class="py-op">,</tt> <tt class="py-param">start_y</tt><tt class="py-op">,</tt> <tt class="py-param">end_x</tt><tt class="py-op">,</tt> <tt class="py-param">end_y</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BarChartDistribution._determine_position-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BarChartDistribution._determine_position-expanded"><a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line">        <tt class="py-docstring">"""Calculate the position of the chart with blank space (PRIVATE).</tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line"><tt class="py-docstring">        This uses some padding around the chart, and takes into account</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line"><tt class="py-docstring">        whether the chart has a title. It returns 4 values, which are,</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line"><tt class="py-docstring">        in order, the x_start, x_end, y_start and y_end of the chart</tt> </tt>
<a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line"><tt class="py-docstring">        itself.</tt> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line">        <tt class="py-name">x_padding</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">padding_percent</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-name">end_x</tt> <tt class="py-op">-</tt> <tt class="py-name">start_x</tt><tt class="py-op">)</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line">        <tt class="py-name">y_padding</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">padding_percent</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-name">start_y</tt> <tt class="py-op">-</tt> <tt class="py-name">end_y</tt><tt class="py-op">)</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line"> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line">        <tt class="py-name">new_x_start</tt> <tt class="py-op">=</tt> <tt class="py-name">start_x</tt> <tt class="py-op">+</tt> <tt class="py-name">x_padding</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line">        <tt class="py-name">new_x_end</tt> <tt class="py-op">=</tt> <tt class="py-name">end_x</tt> <tt class="py-op">-</tt> <tt class="py-name">x_padding</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line"> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chart_title</tt><tt class="py-op">:</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line">            <tt class="py-name">new_y_start</tt> <tt class="py-op">=</tt> <tt class="py-name">start_y</tt> <tt class="py-op">-</tt> <tt class="py-name">y_padding</tt> <tt class="py-op">-</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chart_title_size</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line">            <tt class="py-name">new_y_start</tt> <tt class="py-op">=</tt> <tt class="py-name">start_y</tt> <tt class="py-op">-</tt> <tt class="py-name">y_padding</tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line"> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line">        <tt class="py-name">new_y_end</tt> <tt class="py-op">=</tt> <tt class="py-name">end_y</tt> <tt class="py-op">+</tt> <tt class="py-name">y_padding</tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line"> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">new_x_start</tt><tt class="py-op">,</tt> <tt class="py-name">new_x_end</tt><tt class="py-op">,</tt> <tt class="py-name">new_y_start</tt><tt class="py-op">,</tt> <tt class="py-name">new_y_end</tt> </tt>
</div></div><a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line"> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line"> </tt>
<a name="LineDistribution"></a><div id="LineDistribution-def"><a name="L236"></a><tt class="py-lineno">236</tt> <a class="py-toggle" href="#" id="LineDistribution-toggle" onclick="return toggle('LineDistribution');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Graphics.Distribution.LineDistribution-class.html">LineDistribution</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="LineDistribution-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="LineDistribution-expanded"><a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line">    <tt class="py-docstring">"""Display the distribution of values as connected lines.</tt> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line"><tt class="py-docstring">    This distribution displays the change in values across the object as</tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line"><tt class="py-docstring">    lines. This also allows multiple distributions to be displayed on a</tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line"><tt class="py-docstring">    single graph.</tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line"> </tt>
<a name="LineDistribution.__init__"></a><div id="LineDistribution.__init__-def"><a name="L244"></a><tt class="py-lineno">244</tt> <a class="py-toggle" href="#" id="LineDistribution.__init__-toggle" onclick="return toggle('LineDistribution.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.Distribution.LineDistribution-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="LineDistribution.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="LineDistribution.__init__-expanded"><a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize."""</tt> </tt>
<a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line"> </tt>
<a name="LineDistribution.draw"></a><div id="LineDistribution.draw-def"><a name="L248"></a><tt class="py-lineno">248</tt> <a class="py-toggle" href="#" id="LineDistribution.draw-toggle" onclick="return toggle('LineDistribution.draw');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.Distribution.LineDistribution-class.html#draw">draw</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">cur_drawing</tt><tt class="py-op">,</tt> <tt class="py-param">start_x</tt><tt class="py-op">,</tt> <tt class="py-param">start_y</tt><tt class="py-op">,</tt> <tt class="py-param">end_x</tt><tt class="py-op">,</tt> <tt class="py-param">end_y</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="LineDistribution.draw-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="LineDistribution.draw-expanded"><a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line">        <tt class="py-docstring">"""Draw a line distribution into the current drawing."""</tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div></div><a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Apr  4 10:49:55 2018
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
