<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.UniGene</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        Package&nbsp;UniGene
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.UniGene-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.UniGene-module.html">Package Bio.UniGene</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2006 by Sean Davis.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-comment"># $Id: __init__.py,v 1.12 2009-04-24 12:03:45 mdehoon Exp $</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-comment"># Sean Davis &lt;sdavis2 at mail dot nih dot gov&gt;</tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-comment"># National Cancer Institute</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-comment"># National Institutes of Health</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-comment"># Bethesda, MD, USA</tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-docstring">"""Parse Unigene flat file format files such as the Hs.data file.</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-docstring">Here is an overview of the flat file format that this parser deals with:</tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-docstring">   Line types/qualifiers::</tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-docstring">       ID           UniGene cluster ID</tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-docstring">       TITLE        Title for the cluster</tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-docstring">       GENE         Gene symbol</tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-docstring">       CYTOBAND     Cytological band</tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-docstring">       EXPRESS      Tissues of origin for ESTs in cluster</tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-docstring">       RESTR_EXPR   Single tissue or development stage contributes</tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-docstring">                    more than half the total EST frequency for this gene.</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"><tt class="py-docstring">       GNM_TERMINUS genomic confirmation of presence of a 3' terminus;</tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"><tt class="py-docstring">                    T if a non-templated polyA tail is found among</tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"><tt class="py-docstring">                    a cluster's sequences; else</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"><tt class="py-docstring">                    I if templated As are found in genomic sequence or</tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"><tt class="py-docstring">                    S if a canonical polyA signal is found on</tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"><tt class="py-docstring">                      the genomic sequence</tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"><tt class="py-docstring">       GENE_ID      Entrez gene identifier associated with at least one</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-docstring">                    sequence in this cluster;</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-docstring">                    to be used instead of LocusLink.</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-docstring">       LOCUSLINK    LocusLink identifier associated with at least one</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"><tt class="py-docstring">                    sequence in this cluster;</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"><tt class="py-docstring">                    deprecated in favor of GENE_ID</tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"><tt class="py-docstring">       HOMOL        Homology;</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line"><tt class="py-docstring">       CHROMOSOME   Chromosome.  For plants, CHROMOSOME refers to mapping</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"><tt class="py-docstring">                    on the arabidopsis genome.</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"><tt class="py-docstring">       STS          STS</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line"><tt class="py-docstring">            ACC=         GenBank/EMBL/DDBJ accession number of STS</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line"><tt class="py-docstring">                         [optional field]</tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line"><tt class="py-docstring">            UNISTS=      identifier in NCBI's UNISTS database</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line"><tt class="py-docstring">       TXMAP        Transcript map interval</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line"><tt class="py-docstring">            MARKER=      Marker found on at least one sequence in this</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line"><tt class="py-docstring">                         cluster</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line"><tt class="py-docstring">            RHPANEL=     Radiation Hybrid panel used to place marker</tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line"><tt class="py-docstring">       PROTSIM      Protein Similarity data for the sequence with</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line"><tt class="py-docstring">                    highest-scoring protein similarity in this cluster</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line"><tt class="py-docstring">            ORG=         Organism</tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line"><tt class="py-docstring">            PROTGI=      Sequence GI of protein</tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line"><tt class="py-docstring">            PROTID=      Sequence ID of protein</tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line"><tt class="py-docstring">            PCT=         Percent alignment</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line"><tt class="py-docstring">            ALN=         length of aligned region (aa)</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line"><tt class="py-docstring">       SCOUNT       Number of sequences in the cluster</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"><tt class="py-docstring">       SEQUENCE     Sequence</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line"><tt class="py-docstring">            ACC=         GenBank/EMBL/DDBJ accession number of sequence</tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line"><tt class="py-docstring">            NID=         Unique nucleotide sequence identifier (gi)</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line"><tt class="py-docstring">            PID=         Unique protein sequence identifier (used for</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line"><tt class="py-docstring">                         non-ESTs)</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line"><tt class="py-docstring">            CLONE=       Clone identifier (used for ESTs only)</tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line"><tt class="py-docstring">            END=         End (5'/3') of clone insert read (used for</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line"><tt class="py-docstring">                         ESTs only)</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line"><tt class="py-docstring">            LID=         Library ID; see Hs.lib.info for library name</tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line"><tt class="py-docstring">                         and tissue</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line"><tt class="py-docstring">            MGC=         5' CDS-completeness indicator; if present, the</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line"><tt class="py-docstring">                         clone associated with this sequence is believed</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line"><tt class="py-docstring">                         CDS-complete. A value greater than 511 is the gi</tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line"><tt class="py-docstring">                         of the CDS-complete mRNA matched by the EST,</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line"><tt class="py-docstring">                         otherwise the value is an indicator of the</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line"><tt class="py-docstring">                         reliability of the test indicating CDS</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line"><tt class="py-docstring">                         completeness; higher values indicate more</tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line"><tt class="py-docstring">                         reliable CDS-completeness predictions.</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line"><tt class="py-docstring">           SEQTYPE=      Description of the nucleotide sequence.</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line"><tt class="py-docstring">                         Possible values are mRNA, EST and HTC.</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line"><tt class="py-docstring">           TRACE=        The Trace ID of the EST sequence, as provided by</tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line"><tt class="py-docstring">                         NCBI Trace Archive</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line"> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line"><tt class="py-name">__docformat__</tt> <tt class="py-op">=</tt> <tt class="py-string">"restructuredtext en"</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line"> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line"> </tt>
<a name="SequenceLine"></a><div id="SequenceLine-def"><a name="L84"></a><tt class="py-lineno"> 84</tt> <a class="py-toggle" href="#" id="SequenceLine-toggle" onclick="return toggle('SequenceLine');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.UniGene.SequenceLine-class.html">SequenceLine</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceLine-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="SequenceLine-expanded"><a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line">    <tt class="py-docstring">"""Store the information for one SEQUENCE line from a Unigene file</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line"><tt class="py-docstring">    Initialize with the text part of the SEQUENCE line, or nothing.</tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line"><tt class="py-docstring">    Attributes and descriptions (access as LOWER CASE):</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line"><tt class="py-docstring">    </tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line"><tt class="py-docstring">        - ACC=         GenBank/EMBL/DDBJ accession number of sequence</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line"><tt class="py-docstring">        - NID=         Unique nucleotide sequence identifier (gi)</tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line"><tt class="py-docstring">        - PID=         Unique protein sequence identifier (used for non-ESTs)</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line"><tt class="py-docstring">        - CLONE=       Clone identifier (used for ESTs only)</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line"><tt class="py-docstring">        - END=         End (5'/3') of clone insert read (used for ESTs only)</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line"><tt class="py-docstring">        - LID=         Library ID; see Hs.lib.info for library name and tissue</tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line"><tt class="py-docstring">        - MGC=         5' CDS-completeness indicator; if present,</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line"><tt class="py-docstring">          the clone associated with this sequence</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line"><tt class="py-docstring">          is believed CDS-complete. A value greater than 511</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line"><tt class="py-docstring">          is the gi of the CDS-complete mRNA matched by the EST,</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line"><tt class="py-docstring">          otherwise the value is an indicator of the reliability</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line"><tt class="py-docstring">          of the test indicating CDS completeness;</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line"><tt class="py-docstring">          higher values indicate more reliable CDS-completeness</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line"><tt class="py-docstring">          predictions.</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line"><tt class="py-docstring">        - SEQTYPE=     Description of the nucleotide sequence. Possible values</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line"><tt class="py-docstring">          are mRNA, EST and HTC.</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line"><tt class="py-docstring">        - TRACE=       The Trace ID of the EST sequence, as provided by NCBI</tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line"><tt class="py-docstring">          Trace Archive</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line"> </tt>
<a name="SequenceLine.__init__"></a><div id="SequenceLine.__init__-def"><a name="L111"></a><tt class="py-lineno">111</tt> <a class="py-toggle" href="#" id="SequenceLine.__init__-toggle" onclick="return toggle('SequenceLine.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.UniGene.SequenceLine-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">text</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceLine.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SequenceLine.__init__-expanded"><a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-0" class="py-name" targets="Variable Bio.FSSP.fssp_rec.align.acc=Bio.FSSP.fssp_rec.align-class.html#acc"><a title="Bio.FSSP.fssp_rec.align.acc" class="py-name" href="#" onclick="return doclink('link-0', 'acc', 'link-0');">acc</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.nid()=Bio.GenBank._FeatureConsumer-class.html#nid,Method Bio.GenBank._RecordConsumer.nid()=Bio.GenBank._RecordConsumer-class.html#nid"><a title="Bio.GenBank._FeatureConsumer.nid
Bio.GenBank._RecordConsumer.nid" class="py-name" href="#" onclick="return doclink('link-1', 'nid', 'link-1');">nid</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">lid</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-2" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.pid()=Bio.GenBank._FeatureConsumer-class.html#pid,Method Bio.GenBank._RecordConsumer.pid()=Bio.GenBank._RecordConsumer-class.html#pid"><a title="Bio.GenBank._FeatureConsumer.pid
Bio.GenBank._RecordConsumer.pid" class="py-name" href="#" onclick="return doclink('link-2', 'pid', 'link-2');">pid</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">clone</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">image</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">is_image</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-3" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.end=Bio.SeqFeature.CompoundLocation-class.html#end,Variable Bio.SeqFeature.FeatureLocation.end=Bio.SeqFeature.FeatureLocation-class.html#end"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-3', 'end', 'link-3');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">mgc</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">seqtype</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-4" class="py-name" targets="Method Bio.Nexus.Nodes.Chain.trace()=Bio.Nexus.Nodes.Chain-class.html#trace,Method Bio.Phylo.BaseTree.TreeMixin.trace()=Bio.Phylo.BaseTree.TreeMixin-class.html#trace"><a title="Bio.Nexus.Nodes.Chain.trace
Bio.Phylo.BaseTree.TreeMixin.trace" class="py-name" href="#" onclick="return doclink('link-4', 'trace', 'link-4');">trace</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">text</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">text</tt> <tt class="py-op">=</tt> <tt class="py-name">text</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-5" class="py-name" targets="Method Bio.UniGene.ProtsimLine._init_from_text()=Bio.UniGene.ProtsimLine-class.html#_init_from_text,Method Bio.UniGene.STSLine._init_from_text()=Bio.UniGene.STSLine-class.html#_init_from_text,Method Bio.UniGene.SequenceLine._init_from_text()=Bio.UniGene.SequenceLine-class.html#_init_from_text"><a title="Bio.UniGene.ProtsimLine._init_from_text
Bio.UniGene.STSLine._init_from_text
Bio.UniGene.SequenceLine._init_from_text" class="py-name" href="#" onclick="return doclink('link-5', '_init_from_text', 'link-5');">_init_from_text</a></tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
</div><a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line"> </tt>
<a name="SequenceLine._init_from_text"></a><div id="SequenceLine._init_from_text-def"><a name="L127"></a><tt class="py-lineno">127</tt> <a class="py-toggle" href="#" id="SequenceLine._init_from_text-toggle" onclick="return toggle('SequenceLine._init_from_text');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.UniGene.SequenceLine-class.html#_init_from_text">_init_from_text</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceLine._init_from_text-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SequenceLine._init_from_text-expanded"><a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line">        <tt id="link-6" class="py-name" targets="Variable Bio.SeqFeature.FeatureLocation.parts=Bio.SeqFeature.FeatureLocation-class.html#parts"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-6', 'parts', 'link-6');">parts</a></tt> <tt class="py-op">=</tt> <tt class="py-name">text</tt><tt class="py-op">.</tt><tt id="link-7" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-7', 'split', 'link-7');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">'; '</tt><tt class="py-op">)</tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">part</tt> <tt class="py-keyword">in</tt> <tt id="link-8" class="py-name"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-8', 'parts', 'link-6');">parts</a></tt><tt class="py-op">:</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line">            <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">val</tt> <tt class="py-op">=</tt> <tt class="py-name">part</tt><tt class="py-op">.</tt><tt id="link-9" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-9', 'split', 'link-7');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">"="</tt><tt class="py-op">)</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">key</tt> <tt class="py-op">==</tt> <tt class="py-string">'CLONE'</tt><tt class="py-op">:</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">val</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">5</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'IMAGE'</tt><tt class="py-op">:</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">is_image</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">image</tt> <tt class="py-op">=</tt> <tt class="py-name">val</tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line">            <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-name">key</tt><tt class="py-op">.</tt><tt id="link-10" class="py-name" targets="Method Bio.Seq.Seq.lower()=Bio.Seq.Seq-class.html#lower,Method Bio.Seq.UnknownSeq.lower()=Bio.Seq.UnknownSeq-class.html#lower,Method Bio.SeqRecord.SeqRecord.lower()=Bio.SeqRecord.SeqRecord-class.html#lower"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-10', 'lower', 'link-10');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">val</tt><tt class="py-op">)</tt> </tt>
</div><a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line"> </tt>
<a name="SequenceLine.__repr__"></a><div id="SequenceLine.__repr__-def"><a name="L137"></a><tt class="py-lineno">137</tt> <a class="py-toggle" href="#" id="SequenceLine.__repr__-toggle" onclick="return toggle('SequenceLine.__repr__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.UniGene.SequenceLine-class.html#__repr__">__repr__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="SequenceLine.__repr__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="SequenceLine.__repr__-expanded"><a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">text</tt> </tt>
</div></div><a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line"> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line"> </tt>
<a name="ProtsimLine"></a><div id="ProtsimLine-def"><a name="L141"></a><tt class="py-lineno">141</tt> <a class="py-toggle" href="#" id="ProtsimLine-toggle" onclick="return toggle('ProtsimLine');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.UniGene.ProtsimLine-class.html">ProtsimLine</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ProtsimLine-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="ProtsimLine-expanded"><a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line">    <tt class="py-docstring">"""Store the information for one PROTSIM line from a Unigene file</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line"><tt class="py-docstring">    Initialize with the text part of the PROTSIM line, or nothing.</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line"><tt class="py-docstring">    Attributes and descriptions (access as LOWER CASE)</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line"><tt class="py-docstring">    ORG=         Organism</tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line"><tt class="py-docstring">    PROTGI=      Sequence GI of protein</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line"><tt class="py-docstring">    PROTID=      Sequence ID of protein</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line"><tt class="py-docstring">    PCT=         Percent alignment</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line"><tt class="py-docstring">    ALN=         length of aligned region (aa)</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line"> </tt>
<a name="ProtsimLine.__init__"></a><div id="ProtsimLine.__init__-def"><a name="L154"></a><tt class="py-lineno">154</tt> <a class="py-toggle" href="#" id="ProtsimLine.__init__-toggle" onclick="return toggle('ProtsimLine.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.UniGene.ProtsimLine-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">text</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ProtsimLine.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ProtsimLine.__init__-expanded"><a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">org</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">protgi</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">protid</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">pct</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-11" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.aln=Bio.SearchIO._model.hsp.HSP-class.html#aln,Variable Bio.SearchIO._model.hsp.HSPFragment.aln=Bio.SearchIO._model.hsp.HSPFragment-class.html#aln"><a title="Bio.SearchIO._model.hsp.HSP.aln
Bio.SearchIO._model.hsp.HSPFragment.aln" class="py-name" href="#" onclick="return doclink('link-11', 'aln', 'link-11');">aln</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">text</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">text</tt> <tt class="py-op">=</tt> <tt class="py-name">text</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-12" class="py-name"><a title="Bio.UniGene.ProtsimLine._init_from_text
Bio.UniGene.STSLine._init_from_text
Bio.UniGene.SequenceLine._init_from_text" class="py-name" href="#" onclick="return doclink('link-12', '_init_from_text', 'link-5');">_init_from_text</a></tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
</div><a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line"> </tt>
<a name="ProtsimLine._init_from_text"></a><div id="ProtsimLine._init_from_text-def"><a name="L164"></a><tt class="py-lineno">164</tt> <a class="py-toggle" href="#" id="ProtsimLine._init_from_text-toggle" onclick="return toggle('ProtsimLine._init_from_text');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.UniGene.ProtsimLine-class.html#_init_from_text">_init_from_text</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ProtsimLine._init_from_text-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ProtsimLine._init_from_text-expanded"><a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line">        <tt id="link-13" class="py-name"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-13', 'parts', 'link-6');">parts</a></tt> <tt class="py-op">=</tt> <tt class="py-name">text</tt><tt class="py-op">.</tt><tt id="link-14" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-14', 'split', 'link-7');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">'; '</tt><tt class="py-op">)</tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line"> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">part</tt> <tt class="py-keyword">in</tt> <tt id="link-15" class="py-name"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-15', 'parts', 'link-6');">parts</a></tt><tt class="py-op">:</tt> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">            <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">val</tt> <tt class="py-op">=</tt> <tt class="py-name">part</tt><tt class="py-op">.</tt><tt id="link-16" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-16', 'split', 'link-7');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">"="</tt><tt class="py-op">)</tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line">            <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-name">key</tt><tt class="py-op">.</tt><tt id="link-17" class="py-name"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-17', 'lower', 'link-10');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">val</tt><tt class="py-op">)</tt> </tt>
</div><a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line"> </tt>
<a name="ProtsimLine.__repr__"></a><div id="ProtsimLine.__repr__-def"><a name="L171"></a><tt class="py-lineno">171</tt> <a class="py-toggle" href="#" id="ProtsimLine.__repr__-toggle" onclick="return toggle('ProtsimLine.__repr__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.UniGene.ProtsimLine-class.html#__repr__">__repr__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ProtsimLine.__repr__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ProtsimLine.__repr__-expanded"><a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">text</tt> </tt>
</div></div><a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line"> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line"> </tt>
<a name="STSLine"></a><div id="STSLine-def"><a name="L175"></a><tt class="py-lineno">175</tt> <a class="py-toggle" href="#" id="STSLine-toggle" onclick="return toggle('STSLine');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.UniGene.STSLine-class.html">STSLine</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="STSLine-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="STSLine-expanded"><a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">    <tt class="py-docstring">"""Store the information for one STS line from a Unigene file</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line"><tt class="py-docstring">    Initialize with the text part of the STS line, or nothing.</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line"><tt class="py-docstring">    Attributes and descriptions (access as LOWER CASE)</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line"><tt class="py-docstring">    ACC=         GenBank/EMBL/DDBJ accession number of STS [optional field]</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line"><tt class="py-docstring">    UNISTS=      identifier in NCBI's UNISTS database</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line"> </tt>
<a name="STSLine.__init__"></a><div id="STSLine.__init__-def"><a name="L186"></a><tt class="py-lineno">186</tt> <a class="py-toggle" href="#" id="STSLine.__init__-toggle" onclick="return toggle('STSLine.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.UniGene.STSLine-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">text</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="STSLine.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="STSLine.__init__-expanded"><a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-18" class="py-name"><a title="Bio.FSSP.fssp_rec.align.acc" class="py-name" href="#" onclick="return doclink('link-18', 'acc', 'link-0');">acc</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">unists</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">text</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">text</tt> <tt class="py-op">=</tt> <tt class="py-name">text</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-19" class="py-name"><a title="Bio.UniGene.ProtsimLine._init_from_text
Bio.UniGene.STSLine._init_from_text
Bio.UniGene.SequenceLine._init_from_text" class="py-name" href="#" onclick="return doclink('link-19', '_init_from_text', 'link-5');">_init_from_text</a></tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
</div><a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line"> </tt>
<a name="STSLine._init_from_text"></a><div id="STSLine._init_from_text-def"><a name="L193"></a><tt class="py-lineno">193</tt> <a class="py-toggle" href="#" id="STSLine._init_from_text-toggle" onclick="return toggle('STSLine._init_from_text');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.UniGene.STSLine-class.html#_init_from_text">_init_from_text</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="STSLine._init_from_text-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="STSLine._init_from_text-expanded"><a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">        <tt id="link-20" class="py-name"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-20', 'parts', 'link-6');">parts</a></tt> <tt class="py-op">=</tt> <tt class="py-name">text</tt><tt class="py-op">.</tt><tt id="link-21" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-21', 'split', 'link-7');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">' '</tt><tt class="py-op">)</tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line"> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">part</tt> <tt class="py-keyword">in</tt> <tt id="link-22" class="py-name"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-22', 'parts', 'link-6');">parts</a></tt><tt class="py-op">:</tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line">            <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">val</tt> <tt class="py-op">=</tt> <tt class="py-name">part</tt><tt class="py-op">.</tt><tt id="link-23" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-23', 'split', 'link-7');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">"="</tt><tt class="py-op">)</tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line">            <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-name">key</tt><tt class="py-op">.</tt><tt id="link-24" class="py-name"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-24', 'lower', 'link-10');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">val</tt><tt class="py-op">)</tt> </tt>
</div><a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line"> </tt>
<a name="STSLine.__repr__"></a><div id="STSLine.__repr__-def"><a name="L200"></a><tt class="py-lineno">200</tt> <a class="py-toggle" href="#" id="STSLine.__repr__-toggle" onclick="return toggle('STSLine.__repr__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.UniGene.STSLine-class.html#__repr__">__repr__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="STSLine.__repr__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="STSLine.__repr__-expanded"><a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">text</tt> </tt>
</div></div><a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line"> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line"> </tt>
<a name="Record"></a><div id="Record-def"><a name="L204"></a><tt class="py-lineno">204</tt> <a class="py-toggle" href="#" id="Record-toggle" onclick="return toggle('Record');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.UniGene.Record-class.html">Record</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Record-expanded"><a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line">    <tt class="py-docstring">"""Store a Unigene record</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line"><tt class="py-docstring">    Here is what is stored::</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line"><tt class="py-docstring">        self.ID           = ''  # ID line</tt> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line"><tt class="py-docstring">        self.species      = ''  # Hs, Bt, etc.</tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line"><tt class="py-docstring">        self.title        = ''  # TITLE line</tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line"><tt class="py-docstring">        self.symbol       = ''  # GENE line</tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line"><tt class="py-docstring">        self.cytoband     = ''  # CYTOBAND line</tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line"><tt class="py-docstring">        self.express      = []  # EXPRESS line, parsed on ';'</tt> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line"><tt class="py-docstring">                                # Will be an array of strings</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line"><tt class="py-docstring">        self.restr_expr   = ''  # RESTR_EXPR line</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line"><tt class="py-docstring">        self.gnm_terminus = ''  # GNM_TERMINUS line</tt> </tt>
<a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line"><tt class="py-docstring">        self.gene_id      = ''  # GENE_ID line</tt> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line"><tt class="py-docstring">        self.locuslink    = ''  # LOCUSLINK line</tt> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line"><tt class="py-docstring">        self.homol        = ''  # HOMOL line</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line"><tt class="py-docstring">        self.chromosome   = ''  # CHROMOSOME line</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line"><tt class="py-docstring">        self.protsim      = []  # PROTSIM entries, array of Protsims</tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line"><tt class="py-docstring">                                # Type ProtsimLine</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line"><tt class="py-docstring">        self.sequence     = []  # SEQUENCE entries, array of Sequence entries</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line"><tt class="py-docstring">                                # Type SequenceLine</tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line"><tt class="py-docstring">        self.sts          = []  # STS entries, array of STS entries</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line"><tt class="py-docstring">                                # Type STSLine</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line"><tt class="py-docstring">        self.txmap        = []  # TXMAP entries, array of TXMap entries</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line"> </tt>
<a name="Record.__init__"></a><div id="Record.__init__-def"><a name="L231"></a><tt class="py-lineno">231</tt> <a class="py-toggle" href="#" id="Record.__init__-toggle" onclick="return toggle('Record.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.UniGene.Record-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record.__init__-expanded"><a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ID</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt>  <tt class="py-comment"># ID line</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-25" class="py-name" targets="Method Bio.Pathway.Network.species()=Bio.Pathway.Network-class.html#species,Method Bio.Pathway.Reaction.species()=Bio.Pathway.Reaction-class.html#species,Method Bio.Pathway.System.species()=Bio.Pathway.System-class.html#species"><a title="Bio.Pathway.Network.species
Bio.Pathway.Reaction.species
Bio.Pathway.System.species" class="py-name" href="#" onclick="return doclink('link-25', 'species', 'link-25');">species</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt>  <tt class="py-comment"># Hs, Bt, etc.</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-26" class="py-name" targets="Method Bio.Blast.NCBIStandalone._AlignmentConsumer.title()=Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#title,Method Bio.GenBank._FeatureConsumer.title()=Bio.GenBank._FeatureConsumer-class.html#title,Method Bio.GenBank._RecordConsumer.title()=Bio.GenBank._RecordConsumer-class.html#title"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-26', 'title', 'link-26');">title</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt>  <tt class="py-comment"># TITLE line</tt> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-27" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.symbol()=Bio.Phylo.PhyloXMLIO.Writer-class.html#symbol"><a title="Bio.Phylo.PhyloXMLIO.Writer.symbol" class="py-name" href="#" onclick="return doclink('link-27', 'symbol', 'link-27');">symbol</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt>  <tt class="py-comment"># GENE line</tt> </tt>
<a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">cytoband</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt>  <tt class="py-comment"># CYTOBAND line</tt> </tt>
<a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">express</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>  <tt class="py-comment"># EXPRESS line, parsed on ';'</tt> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">restr_expr</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt>  <tt class="py-comment"># RESTR_EXPR line</tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">gnm_terminus</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt>  <tt class="py-comment"># GNM_TERMINUS line</tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">gene_id</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt>  <tt class="py-comment"># GENE_ID line</tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">locuslink</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt>  <tt class="py-comment"># LOCUSLINK line</tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">homol</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt>  <tt class="py-comment"># HOMOL line</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">chromosome</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt>  <tt class="py-comment"># CHROMOSOME line</tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">protsim</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>  <tt class="py-comment"># PROTSIM entries, array of Protsims</tt> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-28" class="py-name" targets="Method Bio.FSSP.FSSPAlignDict.sequence()=Bio.FSSP.FSSPAlignDict-class.html#sequence,Method Bio.GenBank._FeatureConsumer.sequence()=Bio.GenBank._FeatureConsumer-class.html#sequence,Method Bio.GenBank._RecordConsumer.sequence()=Bio.GenBank._RecordConsumer-class.html#sequence,Method Bio.Phylo.PhyloXMLIO.Writer.sequence()=Bio.Phylo.PhyloXMLIO.Writer-class.html#sequence"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-28', 'sequence', 'link-28');">sequence</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>  <tt class="py-comment"># SEQUENCE entries, array of Sequence entries</tt> </tt>
<a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">sts</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>  <tt class="py-comment"># STS entries, array of STS entries</tt> </tt>
<a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">txmap</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>  <tt class="py-comment"># TXMAP entries, array of TXMap entries</tt> </tt>
</div><a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line"> </tt>
<a name="Record.__repr__"></a><div id="Record.__repr__-def"><a name="L249"></a><tt class="py-lineno">249</tt> <a class="py-toggle" href="#" id="Record.__repr__-toggle" onclick="return toggle('Record.__repr__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.UniGene.Record-class.html#__repr__">__repr__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record.__repr__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record.__repr__-expanded"><a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">"&lt;%s&gt; %s %s\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">,</tt> </tt>
<a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line">                          <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ID</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-29" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.symbol" class="py-name" href="#" onclick="return doclink('link-29', 'symbol', 'link-27');">symbol</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-30" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-30', 'title', 'link-26');">title</a></tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line"> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line"> </tt>
<a name="parse"></a><div id="parse-def"><a name="L254"></a><tt class="py-lineno">254</tt> <a class="py-toggle" href="#" id="parse-toggle" onclick="return toggle('parse');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.UniGene-module.html#parse">parse</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="parse-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="parse-expanded"><a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line">    <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line">        <tt class="py-name">record</tt> <tt class="py-op">=</tt> <tt id="link-31" class="py-name" targets="Method Bio.Motif._Motif.Motif._read()=Bio.Motif._Motif.Motif-class.html#_read,Function Bio.SwissProt._read()=Bio.SwissProt-module.html#_read,Function Bio.UniGene._read()=Bio.UniGene-module.html#_read"><a title="Bio.Motif._Motif.Motif._read
Bio.SwissProt._read
Bio.UniGene._read" class="py-name" href="#" onclick="return doclink('link-31', '_read', 'link-31');">_read</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">record</tt><tt class="py-op">:</tt> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line">        <tt class="py-keyword">yield</tt> <tt class="py-name">record</tt> </tt>
</div><a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line"> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line"> </tt>
<a name="read"></a><div id="read-def"><a name="L262"></a><tt class="py-lineno">262</tt> <a class="py-toggle" href="#" id="read-toggle" onclick="return toggle('read');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.UniGene-module.html#read">read</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="read-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="read-expanded"><a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line">    <tt class="py-name">record</tt> <tt class="py-op">=</tt> <tt id="link-32" class="py-name"><a title="Bio.Motif._Motif.Motif._read
Bio.SwissProt._read
Bio.UniGene._read" class="py-name" href="#" onclick="return doclink('link-32', '_read', 'link-31');">_read</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">record</tt><tt class="py-op">:</tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"No SwissProt record found"</tt><tt class="py-op">)</tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line">    <tt class="py-comment"># We should have reached the end of the record by now</tt> </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line">    <tt class="py-name">remainder</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-33" class="py-name" targets="Function Bio.Affy.CelFile.read()=Bio.Affy.CelFile-module.html#read,Function Bio.AlignIO.read()=Bio.AlignIO-module.html#read,Function Bio.Blast.NCBIXML.read()=Bio.Blast.NCBIXML-module.html#read,Function Bio.Cluster.read()=Bio.Cluster-module.html#read,Function Bio.Compass.read()=Bio.Compass-module.html#read,Function Bio.Emboss.Primer3.read()=Bio.Emboss.Primer3-module.html#read,Function Bio.Emboss.PrimerSearch.read()=Bio.Emboss.PrimerSearch-module.html#read,Method Bio.Entrez.Parser.DataHandler.read()=Bio.Entrez.Parser.DataHandler-class.html#read,Function Bio.Entrez.read()=Bio.Entrez-module.html#read,Function Bio.ExPASy.Enzyme.read()=Bio.ExPASy.Enzyme-module.html#read,Function Bio.ExPASy.Prodoc.read()=Bio.ExPASy.Prodoc-module.html#read,Function Bio.ExPASy.Prosite.read()=Bio.ExPASy.Prosite-module.html#read,Function Bio.ExPASy.ScanProsite.read()=Bio.ExPASy.ScanProsite-module.html#read,Method Bio.File.UndoHandle.read()=Bio.File.UndoHandle-class.html#read,Function Bio.KEGG.KGML.KGML_parser.read()=Bio.KEGG.KGML.KGML_parser-module.html#read,Function Bio.Medline.read()=Bio.Medline-module.html#read,Function Bio.Motif.Parsers.AlignAce.read()=Bio.Motif.Parsers.AlignAce-module.html#read,Function Bio.Motif.Parsers.MAST.read()=Bio.Motif.Parsers.MAST-module.html#read,Function Bio.Motif.Parsers.MEME.read()=Bio.Motif.Parsers.MEME-module.html#read,Function Bio.Motif.read()=Bio.Motif-module.html#read,Method Bio.NeuralNetwork.Gene.Pattern.PatternIO.read()=Bio.NeuralNetwork.Gene.Pattern.PatternIO-class.html#read,Method Bio.Nexus.Nexus.Nexus.read()=Bio.Nexus.Nexus.Nexus-class.html#read,Function Bio.Phylo.PAML.baseml.read()=Bio.Phylo.PAML.baseml-module.html#read,Function Bio.Phylo.PAML.codeml.read()=Bio.Phylo.PAML.codeml-module.html#read,Function Bio.Phylo.PAML.yn00.read()=Bio.Phylo.PAML.yn00-module.html#read,Method Bio.Phylo.PhyloXMLIO.Parser.read()=Bio.Phylo.PhyloXMLIO.Parser-class.html#read,Function Bio.Phylo.PhyloXMLIO.read()=Bio.Phylo.PhyloXMLIO-module.html#read,Function Bio.Phylo._io.read()=Bio.Phylo._io-module.html#read,Function Bio.PopGen.GenePop.FileParser.read()=Bio.PopGen.GenePop.FileParser-module.html#read,Function Bio.PopGen.GenePop.LargeFileParser.read()=Bio.PopGen.GenePop.LargeFileParser-module.html#read,Function Bio.PopGen.GenePop.read()=Bio.PopGen.GenePop-module.html#read,Function Bio.SearchIO.read()=Bio.SearchIO-module.html#read,Method Bio.SeqIO.SffIO._AddTellHandle.read()=Bio.SeqIO.SffIO._AddTellHandle-class.html#read,Function Bio.SeqIO.read()=Bio.SeqIO-module.html#read,Function Bio.Sequencing.Ace.read()=Bio.Sequencing.Ace-module.html#read,Function Bio.Sequencing.Phd.read()=Bio.Sequencing.Phd-module.html#read,Function Bio.SwissProt.read()=Bio.SwissProt-module.html#read,Function Bio.UniGene.read()=Bio.UniGene-module.html#read,Method Bio.bgzf.BgzfReader.read()=Bio.bgzf.BgzfReader-class.html#read,Function Bio.motifs.alignace.read()=Bio.motifs.alignace-module.html#read,Function Bio.motifs.jaspar.read()=Bio.motifs.jaspar-module.html#read,Function Bio.motifs.mast.read()=Bio.motifs.mast-module.html#read,Function Bio.motifs.meme.read()=Bio.motifs.meme-module.html#read,Function Bio.motifs.read()=Bio.motifs-module.html#read,Function Bio.motifs.transfac.read()=Bio.motifs.transfac-module.html#read"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-33', 'read', 'link-33');">read</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">remainder</tt><tt class="py-op">:</tt> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"More than one SwissProt record found"</tt><tt class="py-op">)</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">record</tt> </tt>
</div><a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line"> </tt>
<a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line"> </tt>
<a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line"><tt class="py-comment"># Everything below is private</tt> </tt>
<a name="L274"></a><tt class="py-lineno">274</tt>  <tt class="py-line"> </tt>
<a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line"> </tt>
<a name="_read"></a><div id="_read-def"><a name="L276"></a><tt class="py-lineno">276</tt> <a class="py-toggle" href="#" id="_read-toggle" onclick="return toggle('_read');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.UniGene-module.html#_read">_read</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_read-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_read-expanded"><a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line">    <tt class="py-name">UG_INDENT</tt> <tt class="py-op">=</tt> <tt class="py-number">12</tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line">    <tt class="py-name">record</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line">        <tt id="link-34" class="py-name" targets="Variable Bio.SeqIO.AbiIO.tag=Bio.SeqIO.AbiIO-module.html#tag"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-34', 'tag', 'link-34');">tag</a></tt><tt class="py-op">,</tt> <tt id="link-35" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.value()=Bio.Phylo.PhyloXMLIO.Writer-class.html#value"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-35', 'value', 'link-35');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">UG_INDENT</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-36" class="py-name" targets="Method Bio.Seq.Seq.rstrip()=Bio.Seq.Seq-class.html#rstrip"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-36', 'rstrip', 'link-36');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-name">UG_INDENT</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-37" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-37', 'rstrip', 'link-36');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-38" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-38', 'rstrip', 'link-36');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-39" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-39', 'tag', 'link-34');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"ID"</tt><tt class="py-op">:</tt> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line">            <tt class="py-name">record</tt> <tt class="py-op">=</tt> <tt id="link-40" class="py-name" targets="Class Bio.Affy.CelFile.Record=Bio.Affy.CelFile.Record-class.html,Module Bio.Blast.Record=Bio.Blast.Record-module.html,Class Bio.Cluster.Record=Bio.Cluster.Record-class.html,Class Bio.Compass.Record=Bio.Compass.Record-class.html,Class Bio.Emboss.Primer3.Record=Bio.Emboss.Primer3.Record-class.html,Class Bio.ExPASy.Enzyme.Record=Bio.ExPASy.Enzyme.Record-class.html,Class Bio.ExPASy.Prodoc.Record=Bio.ExPASy.Prodoc.Record-class.html,Class Bio.ExPASy.Prosite.Record=Bio.ExPASy.Prosite.Record-class.html,Class Bio.ExPASy.ScanProsite.Record=Bio.ExPASy.ScanProsite.Record-class.html,Module Bio.GenBank.Record=Bio.GenBank.Record-module.html,Class Bio.GenBank.Record.Record=Bio.GenBank.Record.Record-class.html,Module Bio.Geo.Record=Bio.Geo.Record-module.html,Class Bio.Geo.Record.Record=Bio.Geo.Record.Record-class.html,Class Bio.KEGG.Compound.Record=Bio.KEGG.Compound.Record-class.html,Class Bio.KEGG.Enzyme.Record=Bio.KEGG.Enzyme.Record-class.html,Class Bio.Medline.Record=Bio.Medline.Record-class.html,Class Bio.Motif.Parsers.AlignAce.Record=Bio.Motif.Parsers.AlignAce.Record-class.html,Class Bio.Motif.Parsers.MAST.Record=Bio.Motif.Parsers.MAST.Record-class.html,Class Bio.PopGen.FDist.Record=Bio.PopGen.FDist.Record-class.html,Class Bio.PopGen.GenePop.LargeFileParser.Record=Bio.PopGen.GenePop.LargeFileParser.Record-class.html,Class Bio.PopGen.GenePop.Record=Bio.PopGen.GenePop.Record-class.html,Class Bio.SCOP.Cla.Record=Bio.SCOP.Cla.Record-class.html,Class Bio.SCOP.Des.Record=Bio.SCOP.Des.Record-class.html,Class Bio.SCOP.Dom.Record=Bio.SCOP.Dom.Record-class.html,Class Bio.SCOP.Hie.Record=Bio.SCOP.Hie.Record-class.html,Class Bio.Sequencing.Phd.Record=Bio.Sequencing.Phd.Record-class.html,Class Bio.SwissProt.KeyWList.Record=Bio.SwissProt.KeyWList.Record-class.html,Class Bio.SwissProt.Record=Bio.SwissProt.Record-class.html,Class Bio.UniGene.Record=Bio.UniGene.Record-class.html,Class Bio.motifs.alignace.Record=Bio.motifs.alignace.Record-class.html,Class Bio.motifs.jaspar.Record=Bio.motifs.jaspar.Record-class.html,Class Bio.motifs.mast.Record=Bio.motifs.mast.Record-class.html,Class Bio.motifs.meme.Record=Bio.motifs.meme.Record-class.html,Class Bio.motifs.transfac.Record=Bio.motifs.transfac.Record-class.html"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-40', 'Record', 'link-40');">Record</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">ID</tt> <tt class="py-op">=</tt> <tt id="link-41" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-41', 'value', 'link-35');">value</a></tt> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-42" class="py-name"><a title="Bio.Pathway.Network.species
Bio.Pathway.Reaction.species
Bio.Pathway.System.species" class="py-name" href="#" onclick="return doclink('link-42', 'species', 'link-25');">species</a></tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">ID</tt><tt class="py-op">.</tt><tt id="link-43" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-43', 'split', 'link-7');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">'.'</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-44" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-44', 'tag', 'link-34');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"TITLE"</tt><tt class="py-op">:</tt> </tt>
<a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-45" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-45', 'title', 'link-26');">title</a></tt> <tt class="py-op">=</tt> <tt id="link-46" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-46', 'value', 'link-35');">value</a></tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-47" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-47', 'tag', 'link-34');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"GENE"</tt><tt class="py-op">:</tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-48" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.symbol" class="py-name" href="#" onclick="return doclink('link-48', 'symbol', 'link-27');">symbol</a></tt> <tt class="py-op">=</tt> <tt id="link-49" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-49', 'value', 'link-35');">value</a></tt> </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-50" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-50', 'tag', 'link-34');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"GENE_ID"</tt><tt class="py-op">:</tt> </tt>
<a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">gene_id</tt> <tt class="py-op">=</tt> <tt id="link-51" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-51', 'value', 'link-35');">value</a></tt> </tt>
<a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-52" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-52', 'tag', 'link-34');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"LOCUSLINK"</tt><tt class="py-op">:</tt> </tt>
<a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">locuslink</tt> <tt class="py-op">=</tt> <tt id="link-53" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-53', 'value', 'link-35');">value</a></tt> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-54" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-54', 'tag', 'link-34');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"HOMOL"</tt><tt class="py-op">:</tt> </tt>
<a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-55" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-55', 'value', 'link-35');">value</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"YES"</tt><tt class="py-op">:</tt> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line">                <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">homol</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-56" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-56', 'value', 'link-35');">value</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"NO"</tt><tt class="py-op">:</tt> </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line">                <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">homol</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Cannot parse HOMOL line %s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-57" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-57', 'tag', 'link-34');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"EXPRESS"</tt><tt class="py-op">:</tt> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">express</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">word</tt><tt class="py-op">.</tt><tt id="link-58" class="py-name" targets="Method Bio.Seq.Seq.strip()=Bio.Seq.Seq-class.html#strip"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-58', 'strip', 'link-58');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">word</tt> <tt class="py-keyword">in</tt> <tt id="link-59" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-59', 'value', 'link-35');">value</a></tt><tt class="py-op">.</tt><tt id="link-60" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-60', 'split', 'link-7');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">"|"</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-61" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-61', 'tag', 'link-34');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"RESTR_EXPR"</tt><tt class="py-op">:</tt> </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">restr_expr</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">word</tt><tt class="py-op">.</tt><tt id="link-62" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-62', 'strip', 'link-58');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">word</tt> <tt class="py-keyword">in</tt> <tt id="link-63" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-63', 'value', 'link-35');">value</a></tt><tt class="py-op">.</tt><tt id="link-64" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-64', 'split', 'link-7');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">"|"</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-65" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-65', 'tag', 'link-34');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"CHROMOSOME"</tt><tt class="py-op">:</tt> </tt>
<a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">chromosome</tt> <tt class="py-op">=</tt> <tt id="link-66" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-66', 'value', 'link-35');">value</a></tt> </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-67" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-67', 'tag', 'link-34');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"CYTOBAND"</tt><tt class="py-op">:</tt> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">cytoband</tt> <tt class="py-op">=</tt> <tt id="link-68" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-68', 'value', 'link-35');">value</a></tt> </tt>
<a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-69" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-69', 'tag', 'link-34');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"PROTSIM"</tt><tt class="py-op">:</tt> </tt>
<a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line">            <tt class="py-name">protsim</tt> <tt class="py-op">=</tt> <tt id="link-70" class="py-name" targets="Class Bio.UniGene.ProtsimLine=Bio.UniGene.ProtsimLine-class.html"><a title="Bio.UniGene.ProtsimLine" class="py-name" href="#" onclick="return doclink('link-70', 'ProtsimLine', 'link-70');">ProtsimLine</a></tt><tt class="py-op">(</tt><tt id="link-71" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-71', 'value', 'link-35');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">protsim</tt><tt class="py-op">.</tt><tt id="link-72" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-72', 'append', 'link-72');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">protsim</tt><tt class="py-op">)</tt> </tt>
<a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-73" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-73', 'tag', 'link-34');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"SCOUNT"</tt><tt class="py-op">:</tt> </tt>
<a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line">            <tt class="py-name">scount</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-74" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-74', 'value', 'link-35');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-75" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-75', 'tag', 'link-34');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"SEQUENCE"</tt><tt class="py-op">:</tt> </tt>
<a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line">            <tt id="link-76" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-76', 'sequence', 'link-28');">sequence</a></tt> <tt class="py-op">=</tt> <tt id="link-77" class="py-name" targets="Class Bio.UniGene.SequenceLine=Bio.UniGene.SequenceLine-class.html"><a title="Bio.UniGene.SequenceLine" class="py-name" href="#" onclick="return doclink('link-77', 'SequenceLine', 'link-77');">SequenceLine</a></tt><tt class="py-op">(</tt><tt id="link-78" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-78', 'value', 'link-35');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-79" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-79', 'sequence', 'link-28');">sequence</a></tt><tt class="py-op">.</tt><tt id="link-80" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-80', 'append', 'link-72');">append</a></tt><tt class="py-op">(</tt><tt id="link-81" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-81', 'sequence', 'link-28');">sequence</a></tt><tt class="py-op">)</tt> </tt>
<a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-82" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-82', 'tag', 'link-34');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"STS"</tt><tt class="py-op">:</tt> </tt>
<a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line">            <tt class="py-name">sts</tt> <tt class="py-op">=</tt> <tt id="link-83" class="py-name" targets="Class Bio.UniGene.STSLine=Bio.UniGene.STSLine-class.html"><a title="Bio.UniGene.STSLine" class="py-name" href="#" onclick="return doclink('link-83', 'STSLine', 'link-83');">STSLine</a></tt><tt class="py-op">(</tt><tt id="link-84" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-84', 'value', 'link-35');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">sts</tt><tt class="py-op">.</tt><tt id="link-85" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-85', 'append', 'link-72');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">sts</tt><tt class="py-op">)</tt> </tt>
<a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-86" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-86', 'tag', 'link-34');">tag</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'//'</tt><tt class="py-op">:</tt> </tt>
<a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-87" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-87', 'sequence', 'link-28');">sequence</a></tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">scount</tt><tt class="py-op">:</tt> </tt>
<a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"The number of sequences specified in the record"</tt> </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line">                                 <tt class="py-string">" (%d) does not agree with the number of sequences found (%d)"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">scount</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-88" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-88', 'sequence', 'link-28');">sequence</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">record</tt> </tt>
<a name="L325"></a><tt class="py-lineno">325</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unknown tag %s"</tt> <tt class="py-op">%</tt> <tt id="link-89" class="py-name"><a title="Bio.SeqIO.AbiIO.tag" class="py-name" href="#" onclick="return doclink('link-89', 'tag', 'link-34');">tag</a></tt><tt class="py-op">)</tt> </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">record</tt><tt class="py-op">:</tt> </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unexpected end of stream."</tt><tt class="py-op">)</tt> </tt>
</div><a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Oct 21 18:04:59 2015
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
