<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.GenBank._FeatureConsumer</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        Bio ::
        GenBank ::
        _FeatureConsumer ::
        Class&nbsp;_FeatureConsumer
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.GenBank._FeatureConsumer-class.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== CLASS DESCRIPTION ==================== -->
<h1 class="epydoc">Class _FeatureConsumer</h1><p class="nomargin-top"><span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer">source&nbsp;code</a></span></p>
<pre class="base-tree">
          object --+    
                   |    
<a href="Bio.GenBank._BaseGenBankConsumer-class.html">_BaseGenBankConsumer</a> --+
                       |
                      <strong class="uidshort">_FeatureConsumer</strong>
</pre>

<hr />
<p>Create a SeqRecord object with Features to return (PRIVATE).</p>
<dl class="rst-docutils">
<dt>Attributes:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>use_fuzziness - specify whether or not to parse with fuzziness in
feature locations.</li>
<li>feature_cleaner - a class that will be used to provide specialized
cleaning-up of feature values.</li>
</ul>
</dd>
</dl>

<!-- ==================== INSTANCE METHODS ==================== -->
<a name="section-InstanceMethods"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Instance Methods</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-InstanceMethods"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.GenBank._FeatureConsumer-class.html#__init__" class="summary-sig-name">__init__</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">use_fuzziness</span>,
        <span class="summary-sig-arg">feature_cleaner</span>=<span class="summary-sig-default">None</span>)</span><br />
      x.__init__(...) initializes x; see help(type(x)) for signature</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.__init__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="locus"></a><span class="summary-sig-name">locus</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">locus_name</span>)</span><br />
      Set the locus name is set as the name of the Sequence.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.locus">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="size"></a><span class="summary-sig-name">size</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span><br />
      Record the sequence length.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.size">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.GenBank._FeatureConsumer-class.html#residue_type" class="summary-sig-name">residue_type</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">type</span>)</span><br />
      Record the sequence type (SEMI-OBSOLETE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.residue_type">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="topology"></a><span class="summary-sig-name">topology</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">topology</span>)</span><br />
      Validate and record sequence topology (linear or circular as strings).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.topology">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="molecule_type"></a><span class="summary-sig-name">molecule_type</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">mol_type</span>)</span><br />
      Validate and record the molecule type (for round-trip etc).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.molecule_type">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="data_file_division"></a><span class="summary-sig-name">data_file_division</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">division</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.data_file_division">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="date"></a><span class="summary-sig-name">date</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">submit_date</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.date">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="definition"></a><span class="summary-sig-name">definition</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">definition</span>)</span><br />
      Set the definition as the description of the sequence.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.definition">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.GenBank._FeatureConsumer-class.html#accession" class="summary-sig-name">accession</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">acc_num</span>)</span><br />
      Set the accession number as the id of the sequence.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.accession">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="wgs"></a><span class="summary-sig-name">wgs</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.wgs">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="add_wgs_scafld"></a><span class="summary-sig-name">add_wgs_scafld</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.add_wgs_scafld">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="nid"></a><span class="summary-sig-name">nid</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.nid">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="pid"></a><span class="summary-sig-name">pid</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.pid">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="version"></a><span class="summary-sig-name">version</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">version_id</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.version">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.GenBank._FeatureConsumer-class.html#project" class="summary-sig-name">project</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span><br />
      Handle the information from the PROJECT line as a list of projects.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.project">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.GenBank._FeatureConsumer-class.html#dblink" class="summary-sig-name">dblink</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span><br />
      Store DBLINK cross references as dbxrefs in our record object.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.dblink">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.GenBank._FeatureConsumer-class.html#version_suffix" class="summary-sig-name">version_suffix</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">version</span>)</span><br />
      Set the version to overwrite the id.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.version_suffix">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="db_source"></a><span class="summary-sig-name">db_source</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.db_source">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="gi"></a><span class="summary-sig-name">gi</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.gi">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="keywords"></a><span class="summary-sig-name">keywords</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.keywords">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="segment"></a><span class="summary-sig-name">segment</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.segment">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="source"></a><span class="summary-sig-name">source</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.source">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="organism"></a><span class="summary-sig-name">organism</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.organism">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="taxonomy"></a><span class="summary-sig-name">taxonomy</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span><br />
      Record (another line of) the taxonomy lineage.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.taxonomy">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="reference_num"></a><span class="summary-sig-name">reference_num</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span><br />
      Signal the beginning of a new reference object.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.reference_num">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.GenBank._FeatureConsumer-class.html#reference_bases" class="summary-sig-name">reference_bases</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span><br />
      Attempt to determine the sequence region the reference entails.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.reference_bases">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.GenBank._FeatureConsumer-class.html#_split_reference_locations" class="summary-sig-name" onclick="show_private();">_split_reference_locations</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">location_string</span>)</span><br />
      Get reference locations out of a string of reference information (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer._split_reference_locations">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="authors"></a><span class="summary-sig-name">authors</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.authors">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="consrtm"></a><span class="summary-sig-name">consrtm</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.consrtm">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="title"></a><span class="summary-sig-name">title</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.title">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="journal"></a><span class="summary-sig-name">journal</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.journal">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="medline_id"></a><span class="summary-sig-name">medline_id</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.medline_id">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="pubmed_id"></a><span class="summary-sig-name">pubmed_id</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.pubmed_id">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="remark"></a><span class="summary-sig-name">remark</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span><br />
      Deal with a reference comment.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.remark">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="comment"></a><span class="summary-sig-name">comment</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.comment">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="structured_comment"></a><span class="summary-sig-name">structured_comment</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.structured_comment">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="features_line"></a><span class="summary-sig-name">features_line</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span><br />
      Get ready for the feature table when we reach the FEATURE line.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.features_line">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="start_feature_table"></a><span class="summary-sig-name">start_feature_table</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Indicate we've got to the start of the feature table.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.start_feature_table">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="feature_key"></a><span class="summary-sig-name">feature_key</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.feature_key">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.GenBank._FeatureConsumer-class.html#location" class="summary-sig-name">location</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span><br />
      Parse out location information from the location string.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.location">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.GenBank._FeatureConsumer-class.html#feature_qualifier" class="summary-sig-name">feature_qualifier</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">key</span>,
        <span class="summary-sig-arg">value</span>)</span><br />
      When we get a qualifier key and its value.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.feature_qualifier">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="feature_qualifier_name"></a><span class="summary-sig-name">feature_qualifier_name</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content_list</span>)</span><br />
      Use feature_qualifier instead (OBSOLETE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.feature_qualifier_name">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="feature_qualifier_description"></a><span class="summary-sig-name">feature_qualifier_description</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span><br />
      Use feature_qualifier instead (OBSOLETE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.feature_qualifier_description">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="contig_location"></a><span class="summary-sig-name">contig_location</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span><br />
      Deal with CONTIG information.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.contig_location">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="origin_name"></a><span class="summary-sig-name">origin_name</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.origin_name">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="base_count"></a><span class="summary-sig-name">base_count</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.base_count">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="base_number"></a><span class="summary-sig-name">base_number</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.base_number">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.GenBank._FeatureConsumer-class.html#sequence" class="summary-sig-name">sequence</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span><br />
      Add up sequence information as we get it.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.sequence">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="record_end"></a><span class="summary-sig-name">record_end</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">content</span>)</span><br />
      Clean up when we've finished the record.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.record_end">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code><a href="Bio.GenBank._BaseGenBankConsumer-class.html">_BaseGenBankConsumer</a></code></b>:
      <code><a href="Bio.GenBank._BaseGenBankConsumer-class.html#__getattr__">__getattr__</a></code>
      </p>
    <div class="private">    <p class="indent-wrapped-lines"><b>Inherited from <code><a href="Bio.GenBank._BaseGenBankConsumer-class.html">_BaseGenBankConsumer</a></code></b> (private):
      <code><a href="Bio.GenBank._BaseGenBankConsumer-class.html#_unhandled" onclick="show_private();">_unhandled</a></code>
      </p></div>
    <p class="indent-wrapped-lines"><b>Inherited from <code>object</code></b>:
      <code>__delattr__</code>,
      <code>__format__</code>,
      <code>__getattribute__</code>,
      <code>__hash__</code>,
      <code>__new__</code>,
      <code>__reduce__</code>,
      <code>__reduce_ex__</code>,
      <code>__repr__</code>,
      <code>__setattr__</code>,
      <code>__sizeof__</code>,
      <code>__str__</code>,
      <code>__subclasshook__</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== STATIC METHODS ==================== -->
<a name="section-StaticMethods"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Static Methods</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-StaticMethods"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
  <tr>
    <td colspan="2" class="summary">
    <div class="private">    <p class="indent-wrapped-lines"><b>Inherited from <code><a href="Bio.GenBank._BaseGenBankConsumer-class.html">_BaseGenBankConsumer</a></code></b> (private):
      <code><a href="Bio.GenBank._BaseGenBankConsumer-class.html#_clean_location" onclick="show_private();">_clean_location</a></code>,
      <code><a href="Bio.GenBank._BaseGenBankConsumer-class.html#_convert_to_python_numbers" onclick="show_private();">_convert_to_python_numbers</a></code>,
      <code><a href="Bio.GenBank._BaseGenBankConsumer-class.html#_normalize_spaces" onclick="show_private();">_normalize_spaces</a></code>,
      <code><a href="Bio.GenBank._BaseGenBankConsumer-class.html#_remove_newlines" onclick="show_private();">_remove_newlines</a></code>,
      <code><a href="Bio.GenBank._BaseGenBankConsumer-class.html#_remove_spaces" onclick="show_private();">_remove_spaces</a></code>,
      <code><a href="Bio.GenBank._BaseGenBankConsumer-class.html#_split_accessions" onclick="show_private();">_split_accessions</a></code>,
      <code><a href="Bio.GenBank._BaseGenBankConsumer-class.html#_split_keywords" onclick="show_private();">_split_keywords</a></code>,
      <code><a href="Bio.GenBank._BaseGenBankConsumer-class.html#_split_taxonomy" onclick="show_private();">_split_taxonomy</a></code>
      </p></div>
    </td>
  </tr>
</table>
<!-- ==================== CLASS VARIABLES ==================== -->
<a name="section-ClassVariables"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Class Variables</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-ClassVariables"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code><a href="Bio.GenBank._BaseGenBankConsumer-class.html">_BaseGenBankConsumer</a></code></b>:
      <code><a href="Bio.GenBank._BaseGenBankConsumer-class.html#remove_space_keys">remove_space_keys</a></code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== PROPERTIES ==================== -->
<a name="section-Properties"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Properties</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Properties"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code>object</code></b>:
      <code>__class__</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== METHOD DETAILS ==================== -->
<a name="section-MethodDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Method Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-MethodDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="__init__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__init__</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">use_fuzziness</span>,
        <span class="sig-arg">feature_cleaner</span>=<span class="sig-default">None</span>)</span>
    <br /><em class="fname">(Constructor)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.__init__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  x.__init__(...) initializes x; see help(type(x)) for signature
  <dl class="fields">
    <dt>Overrides:
        object.__init__
        <dd><em class="note">(inherited documentation)</em></dd>
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="residue_type"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">residue_type</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">type</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.residue_type">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Record the sequence type (SEMI-OBSOLETE).</p>
<p>This reflects the fact that the topology (linear/circular) and
molecule type (e.g. DNA vs RNA) were a single field in early
files. Current GenBank/EMBL files have two fields.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="accession"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">accession</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">acc_num</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.accession">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Set the accession number as the id of the sequence.</p>
<p>If we have multiple accession numbers, the first one passed is
used.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="project"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">project</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">content</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.project">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Handle the information from the PROJECT line as a list of projects.</p>
<p>e.g.:</p>
<pre class="rst-literal-block">
PROJECT     GenomeProject:28471
</pre>
<p>or:</p>
<pre class="rst-literal-block">
PROJECT     GenomeProject:13543  GenomeProject:99999
</pre>
<p>This is stored as dbxrefs in the SeqRecord to be consistent with the
projected switch of this line to DBLINK in future GenBank versions.
Note the NCBI plan to replace &quot;GenomeProject:28471&quot; with the shorter
&quot;Project:28471&quot; as part of this transition.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="dblink"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">dblink</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">content</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.dblink">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Store DBLINK cross references as dbxrefs in our record object.</p>
<p>This line type is expected to replace the PROJECT line in 2009. e.g.</p>
<p>During transition:</p>
<pre class="rst-literal-block">
PROJECT     GenomeProject:28471
DBLINK      Project:28471
            Trace Assembly Archive:123456
</pre>
<p>Once the project line is dropped:</p>
<pre class="rst-literal-block">
DBLINK      Project:28471
            Trace Assembly Archive:123456
</pre>
<p>Note GenomeProject -&gt; Project.</p>
<p>We'll have to see some real examples to be sure, but based on the
above example we can expect one reference per line.</p>
<p>Note that at some point the NCBI have included an extra space, e.g.:</p>
<pre class="rst-literal-block">
DBLINK      Project: 28471
</pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="version_suffix"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">version_suffix</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">version</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.version_suffix">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Set the version to overwrite the id.</p>
<p>Since the version provides the same information as the accession
number, plus some extra info, we set this as the id if we have
a version.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="reference_bases"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">reference_bases</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">content</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.reference_bases">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Attempt to determine the sequence region the reference entails.</p>
<p>Possible types of information we may have to deal with:</p>
<p>(bases 1 to 86436)
(sites)
(bases 1 to 105654; 110423 to 111122)
1  (residues 1 to 182)</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_split_reference_locations"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_split_reference_locations</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">location_string</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer._split_reference_locations">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Get reference locations out of a string of reference information (PRIVATE).</p>
<p>The passed string should be of the form:</p>
<pre class="rst-literal-block">
1 to 20; 20 to 100
</pre>
<p>This splits the information out and returns a list of location objects
based on the reference locations.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="location"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">location</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">content</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.location">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Parse out location information from the location string.</p>
<p>This uses simple Python code with some regular expressions to do the
parsing, and then translates the results into appropriate objects.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="feature_qualifier"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">feature_qualifier</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">key</span>,
        <span class="sig-arg">value</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.feature_qualifier">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>When we get a qualifier key and its value.</p>
<p>Can receive None, since you can have valueless keys such as /pseudo</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="sequence"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">sequence</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">content</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.GenBank-pysrc.html#_FeatureConsumer.sequence">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Add up sequence information as we get it.</p>
<p>To try and make things speedier, this puts all of the strings
into a list of strings, and then uses string.join later to put
them together. Supposedly, this is a big time savings</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Fri Jun 22 16:34:33 2018
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
