<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.Motif.Parsers.AlignAce</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.Motif-module.html">Package&nbsp;Motif</a> ::
        <a href="Bio.Motif.Parsers-module.html">Package&nbsp;Parsers</a> ::
        Module&nbsp;AlignAce
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.Motif.Parsers.AlignAce-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.Motif.Parsers.AlignAce-module.html">Module Bio.Motif.Parsers.AlignAce</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno"> 1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2003 by Bartek Wilczynski.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno"> 2</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L3"></a><tt class="py-lineno"> 3</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L4"></a><tt class="py-lineno"> 4</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L5"></a><tt class="py-lineno"> 5</tt>  <tt class="py-line"> </tt>
<a name="L6"></a><tt class="py-lineno"> 6</tt>  <tt class="py-line"><tt class="py-docstring">"""Parsing AlignACE files: AlignAceParser</tt> </tt>
<a name="L7"></a><tt class="py-lineno"> 7</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L8"></a><tt class="py-lineno"> 8</tt>  <tt class="py-line"> </tt>
<a name="L9"></a><tt class="py-lineno"> 9</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio.Motif=Bio.Motif-module.html,Class Bio.Motif._Motif.Motif=Bio.Motif._Motif.Motif-class.html,Module Bio.NeuralNetwork.Gene.Motif=Bio.NeuralNetwork.Gene.Motif-module.html,Class Bio.motifs.Motif=Bio.motifs.Motif-class.html,Class Bio.motifs.jaspar.Motif=Bio.motifs.jaspar.Motif-class.html,Class Bio.motifs.meme.Motif=Bio.motifs.meme.Motif-class.html,Class Bio.motifs.transfac.Motif=Bio.motifs.transfac.Motif-class.html"><a title="Bio.Motif
Bio.Motif._Motif.Motif
Bio.NeuralNetwork.Gene.Motif
Bio.motifs.Motif
Bio.motifs.jaspar.Motif
Bio.motifs.meme.Motif
Bio.motifs.transfac.Motif" class="py-name" href="#" onclick="return doclink('link-1', 'Motif', 'link-1');">Motif</a></tt> <tt class="py-keyword">import</tt> <tt id="link-2" class="py-name"><a title="Bio.Motif
Bio.Motif._Motif.Motif
Bio.NeuralNetwork.Gene.Motif
Bio.motifs.Motif
Bio.motifs.jaspar.Motif
Bio.motifs.meme.Motif
Bio.motifs.transfac.Motif" class="py-name" href="#" onclick="return doclink('link-2', 'Motif', 'link-1');">Motif</a></tt> </tt>
<a name="L10"></a><tt class="py-lineno">10</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-3" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-3', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-4" class="py-name" targets="Package Bio.Alphabet=Bio.Alphabet-module.html,Class Bio.Alphabet.Alphabet=Bio.Alphabet.Alphabet-class.html"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-4', 'Alphabet', 'link-4');">Alphabet</a></tt> <tt class="py-keyword">import</tt> <tt id="link-5" class="py-name" targets="Module Bio.Alphabet.IUPAC=Bio.Alphabet.IUPAC-module.html"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-5', 'IUPAC', 'link-5');">IUPAC</a></tt> </tt>
<a name="L11"></a><tt class="py-lineno">11</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-6" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-6', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-7" class="py-name" targets="Module Bio.Seq=Bio.Seq-module.html,Class Bio.Seq.Seq=Bio.Seq.Seq-class.html"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-7', 'Seq', 'link-7');">Seq</a></tt> <tt class="py-keyword">import</tt> <tt id="link-8" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-8', 'Seq', 'link-7');">Seq</a></tt> </tt>
<a name="L12"></a><tt class="py-lineno">12</tt>  <tt class="py-line"> </tt>
<a name="L13"></a><tt class="py-lineno">13</tt>  <tt class="py-line"> </tt>
<a name="Record"></a><div id="Record-def"><a name="L14"></a><tt class="py-lineno">14</tt> <a class="py-toggle" href="#" id="Record-toggle" onclick="return toggle('Record');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Motif.Parsers.AlignAce.Record-class.html">Record</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record-collapsed" style="display:none;" pad="++" indent="++++"></div><div id="Record-expanded"><a name="Record.__init__"></a><div id="Record.__init__-def"><a name="L15"></a><tt class="py-lineno">15</tt> <a class="py-toggle" href="#" id="Record.__init__-toggle" onclick="return toggle('Record.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Motif.Parsers.AlignAce.Record-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record.__init__-collapsed" style="display:none;" pad="++" indent="++++++++"></div><div id="Record.__init__-expanded"><a name="L16"></a><tt class="py-lineno">16</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-9" class="py-name" targets="Package Bio.motifs=Bio.motifs-module.html"><a title="Bio.motifs" class="py-name" href="#" onclick="return doclink('link-9', 'motifs', 'link-9');">motifs</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L17"></a><tt class="py-lineno">17</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">current_motif</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L18"></a><tt class="py-lineno">18</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">param_dict</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
</div></div><a name="L19"></a><tt class="py-lineno">19</tt>  <tt class="py-line"> </tt>
<a name="L20"></a><tt class="py-lineno">20</tt>  <tt class="py-line"> </tt>
<a name="read"></a><div id="read-def"><a name="L21"></a><tt class="py-lineno">21</tt> <a class="py-toggle" href="#" id="read-toggle" onclick="return toggle('read');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Motif.Parsers.AlignAce-module.html#read">read</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="read-collapsed" style="display:none;" pad="++" indent="++++"></div><div id="read-expanded"><a name="L22"></a><tt class="py-lineno">22</tt>  <tt class="py-line">    <tt class="py-docstring">"""read(handle)"""</tt> </tt>
<a name="L23"></a><tt class="py-lineno">23</tt>  <tt class="py-line">    <tt class="py-name">record</tt> <tt class="py-op">=</tt> <tt id="link-10" class="py-name" targets="Class Bio.Affy.CelFile.Record=Bio.Affy.CelFile.Record-class.html,Module Bio.Blast.Record=Bio.Blast.Record-module.html,Class Bio.Cluster.Record=Bio.Cluster.Record-class.html,Class Bio.Compass.Record=Bio.Compass.Record-class.html,Class Bio.Emboss.Primer3.Record=Bio.Emboss.Primer3.Record-class.html,Class Bio.ExPASy.Enzyme.Record=Bio.ExPASy.Enzyme.Record-class.html,Class Bio.ExPASy.Prodoc.Record=Bio.ExPASy.Prodoc.Record-class.html,Class Bio.ExPASy.Prosite.Record=Bio.ExPASy.Prosite.Record-class.html,Class Bio.ExPASy.ScanProsite.Record=Bio.ExPASy.ScanProsite.Record-class.html,Module Bio.GenBank.Record=Bio.GenBank.Record-module.html,Class Bio.GenBank.Record.Record=Bio.GenBank.Record.Record-class.html,Module Bio.Geo.Record=Bio.Geo.Record-module.html,Class Bio.Geo.Record.Record=Bio.Geo.Record.Record-class.html,Class Bio.KEGG.Compound.Record=Bio.KEGG.Compound.Record-class.html,Class Bio.KEGG.Enzyme.Record=Bio.KEGG.Enzyme.Record-class.html,Class Bio.Medline.Record=Bio.Medline.Record-class.html,Class Bio.Motif.Parsers.AlignAce.Record=Bio.Motif.Parsers.AlignAce.Record-class.html,Class Bio.Motif.Parsers.MAST.Record=Bio.Motif.Parsers.MAST.Record-class.html,Class Bio.PopGen.FDist.Record=Bio.PopGen.FDist.Record-class.html,Class Bio.PopGen.GenePop.LargeFileParser.Record=Bio.PopGen.GenePop.LargeFileParser.Record-class.html,Class Bio.PopGen.GenePop.Record=Bio.PopGen.GenePop.Record-class.html,Class Bio.SCOP.Cla.Record=Bio.SCOP.Cla.Record-class.html,Class Bio.SCOP.Des.Record=Bio.SCOP.Des.Record-class.html,Class Bio.SCOP.Dom.Record=Bio.SCOP.Dom.Record-class.html,Class Bio.SCOP.Hie.Record=Bio.SCOP.Hie.Record-class.html,Class Bio.Sequencing.Phd.Record=Bio.Sequencing.Phd.Record-class.html,Class Bio.SwissProt.KeyWList.Record=Bio.SwissProt.KeyWList.Record-class.html,Class Bio.SwissProt.Record=Bio.SwissProt.Record-class.html,Class Bio.UniGene.Record=Bio.UniGene.Record-class.html,Class Bio.motifs.alignace.Record=Bio.motifs.alignace.Record-class.html,Class Bio.motifs.jaspar.Record=Bio.motifs.jaspar.Record-class.html,Class Bio.motifs.mast.Record=Bio.motifs.mast.Record-class.html,Class Bio.motifs.meme.Record=Bio.motifs.meme.Record-class.html,Class Bio.motifs.transfac.Record=Bio.motifs.transfac.Record-class.html"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-10', 'Record', 'link-10');">Record</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L24"></a><tt class="py-lineno">24</tt>  <tt class="py-line">    <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">ver</tt> <tt class="py-op">=</tt> <tt id="link-11" class="py-name" targets="Method Bio.AlignIO.Interfaces.AlignmentIterator.next()=Bio.AlignIO.Interfaces.AlignmentIterator-class.html#next,Method Bio.Blast.NCBIStandalone.Iterator.next()=Bio.Blast.NCBIStandalone.Iterator-class.html#next,Method Bio.Blast.ParseBlastTable.BlastTableReader.next()=Bio.Blast.ParseBlastTable.BlastTableReader-class.html#next,Method Bio.DocSQL.IterationCursor.next()=Bio.DocSQL.IterationCursor-class.html#next,Method Bio.File.UndoHandle.next()=Bio.File.UndoHandle-class.html#next,Method Bio.GenBank.Iterator.next()=Bio.GenBank.Iterator-class.html#next,Method Bio.Nexus.Nexus.CharBuffer.next()=Bio.Nexus.Nexus.CharBuffer-class.html#next,Method Bio.PopGen.GenePop.Controller._FileIterator.next()=Bio.PopGen.GenePop.Controller._FileIterator-class.html#next,Method Bio.SeqIO.Interfaces.SequenceIterator.next()=Bio.SeqIO.Interfaces.SequenceIterator-class.html#next,Method Bio.bgzf.BgzfReader.next()=Bio.bgzf.BgzfReader-class.html#next"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-11', 'next', 'link-11');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L25"></a><tt class="py-lineno">25</tt>  <tt class="py-line">    <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">cmd_line</tt> <tt class="py-op">=</tt> <tt id="link-12" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-12', 'next', 'link-11');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L26"></a><tt class="py-lineno">26</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">handle</tt><tt class="py-op">:</tt> </tt>
<a name="L27"></a><tt class="py-lineno">27</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-13" class="py-name" targets="Method Bio.Seq.Seq.strip()=Bio.Seq.Seq-class.html#strip"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-13', 'strip', 'link-13');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-string">""</tt><tt class="py-op">:</tt> </tt>
<a name="L28"></a><tt class="py-lineno">28</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
<a name="L29"></a><tt class="py-lineno">29</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">4</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"Para"</tt><tt class="py-op">:</tt> </tt>
<a name="L30"></a><tt class="py-lineno">30</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">param_dict</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L31"></a><tt class="py-lineno">31</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"#"</tt><tt class="py-op">:</tt> </tt>
<a name="L32"></a><tt class="py-lineno">32</tt>  <tt class="py-line">            <tt class="py-name">seq_name</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-14" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-14', 'split', 'link-14');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">"\t"</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L33"></a><tt class="py-lineno">33</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">seq_dict</tt><tt class="py-op">.</tt><tt id="link-15" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-15', 'append', 'link-15');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">seq_name</tt><tt class="py-op">)</tt> </tt>
<a name="L34"></a><tt class="py-lineno">34</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-string">"="</tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L35"></a><tt class="py-lineno">35</tt>  <tt class="py-line">            <tt class="py-name">par_name</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-16" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-16', 'split', 'link-14');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">"="</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-17" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-17', 'strip', 'link-13');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L36"></a><tt class="py-lineno">36</tt>  <tt class="py-line">            <tt class="py-name">par_value</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-18" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-18', 'split', 'link-14');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">"="</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-19" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-19', 'strip', 'link-13');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L37"></a><tt class="py-lineno">37</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">param_dict</tt><tt class="py-op">[</tt><tt class="py-name">par_name</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">par_value</tt> </tt>
<a name="L38"></a><tt class="py-lineno">38</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">5</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"Input"</tt><tt class="py-op">:</tt> </tt>
<a name="L39"></a><tt class="py-lineno">39</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">seq_dict</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L40"></a><tt class="py-lineno">40</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">5</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"Motif"</tt><tt class="py-op">:</tt> </tt>
<a name="L41"></a><tt class="py-lineno">41</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">current_motif</tt> <tt class="py-op">=</tt> <tt id="link-20" class="py-name"><a title="Bio.Motif
Bio.Motif._Motif.Motif
Bio.NeuralNetwork.Gene.Motif
Bio.motifs.Motif
Bio.motifs.jaspar.Motif
Bio.motifs.meme.Motif
Bio.motifs.transfac.Motif" class="py-name" href="#" onclick="return doclink('link-20', 'Motif', 'link-1');">Motif</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L42"></a><tt class="py-lineno">42</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-21" class="py-name"><a title="Bio.motifs" class="py-name" href="#" onclick="return doclink('link-21', 'motifs', 'link-9');">motifs</a></tt><tt class="py-op">.</tt><tt id="link-22" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-22', 'append', 'link-15');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">current_motif</tt><tt class="py-op">)</tt> </tt>
<a name="L43"></a><tt class="py-lineno">43</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">current_motif</tt><tt class="py-op">.</tt><tt id="link-23" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.alphabet=Bio.SearchIO._model.hsp.HSP-class.html#alphabet,Variable Bio.SearchIO._model.hsp.HSPFragment.alphabet=Bio.SearchIO._model.hsp.HSPFragment-class.html#alphabet"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-23', 'alphabet', 'link-23');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-24" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-24', 'IUPAC', 'link-5');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-25" class="py-name" targets="Variable Bio.Alphabet.IUPAC.unambiguous_dna=Bio.Alphabet.IUPAC-module.html#unambiguous_dna"><a title="Bio.Alphabet.IUPAC.unambiguous_dna" class="py-name" href="#" onclick="return doclink('link-25', 'unambiguous_dna', 'link-25');">unambiguous_dna</a></tt> </tt>
<a name="L44"></a><tt class="py-lineno">44</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">3</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"MAP"</tt><tt class="py-op">:</tt> </tt>
<a name="L45"></a><tt class="py-lineno">45</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">current_motif</tt><tt class="py-op">.</tt><tt id="link-26" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HSPConsumer.score()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#score"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-26', 'score', 'link-26');">score</a></tt> <tt class="py-op">=</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-27" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-27', 'split', 'link-14');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L46"></a><tt class="py-lineno">46</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-28" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-28', 'split', 'link-14');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">"\t"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">4</tt><tt class="py-op">:</tt> </tt>
<a name="L47"></a><tt class="py-lineno">47</tt>  <tt class="py-line">            <tt id="link-29" class="py-name" targets="Variable Bio.SeqRecord.SeqRecord.seq=Bio.SeqRecord.SeqRecord-class.html#seq,Variable BioSQL.BioSeq.DBSeqRecord.seq=BioSQL.BioSeq.DBSeqRecord-class.html#seq"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-29', 'seq', 'link-29');">seq</a></tt> <tt class="py-op">=</tt> <tt id="link-30" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-30', 'Seq', 'link-7');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-31" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-31', 'split', 'link-14');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">"\t"</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt id="link-32" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-32', 'IUPAC', 'link-5');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-33" class="py-name"><a title="Bio.Alphabet.IUPAC.unambiguous_dna" class="py-name" href="#" onclick="return doclink('link-33', 'unambiguous_dna', 'link-25');">unambiguous_dna</a></tt><tt class="py-op">)</tt> </tt>
<a name="L48"></a><tt class="py-lineno">48</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">current_motif</tt><tt class="py-op">.</tt><tt id="link-34" class="py-name" targets="Method Bio.Motif._Motif.Motif.add_instance()=Bio.Motif._Motif.Motif-class.html#add_instance"><a title="Bio.Motif._Motif.Motif.add_instance" class="py-name" href="#" onclick="return doclink('link-34', 'add_instance', 'link-34');">add_instance</a></tt><tt class="py-op">(</tt><tt id="link-35" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-35', 'seq', 'link-29');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L49"></a><tt class="py-lineno">49</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-string">"*"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L50"></a><tt class="py-lineno">50</tt>  <tt class="py-line">            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">current_motif</tt><tt class="py-op">.</tt><tt id="link-36" class="py-name" targets="Method Bio.Motif._Motif.Motif.set_mask()=Bio.Motif._Motif.Motif-class.html#set_mask"><a title="Bio.Motif._Motif.Motif.set_mask" class="py-name" href="#" onclick="return doclink('link-36', 'set_mask', 'link-36');">set_mask</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-37" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-37', 'strip', 'link-13');">strip</a></tt><tt class="py-op">(</tt><tt class="py-string">"\n\c"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L51"></a><tt class="py-lineno">51</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L52"></a><tt class="py-lineno">52</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L53"></a><tt class="py-lineno">53</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">record</tt> </tt>
</div><a name="L54"></a><tt class="py-lineno">54</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Oct 21 18:04:42 2015
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
