<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.GenBank</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        Package&nbsp;GenBank
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.GenBank-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.GenBank-module.html">Package Bio.GenBank</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">   1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2000 by Jeffrey Chang, Brad Chapman.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">   2</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2006-2013 by Peter Cock.  All rights reserved.</tt> </tt>
<a name="L3"></a><tt class="py-lineno">   3</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L4"></a><tt class="py-lineno">   4</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L5"></a><tt class="py-lineno">   5</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L6"></a><tt class="py-lineno">   6</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L7"></a><tt class="py-lineno">   7</tt>  <tt class="py-line"> </tt>
<a name="L8"></a><tt class="py-lineno">   8</tt>  <tt class="py-line"><tt class="py-docstring">"""Code to work with GenBank formatted files.</tt> </tt>
<a name="L9"></a><tt class="py-lineno">   9</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L10"></a><tt class="py-lineno">  10</tt>  <tt class="py-line"><tt class="py-docstring">Rather than using Bio.GenBank, you are now encouraged to use Bio.SeqIO with</tt> </tt>
<a name="L11"></a><tt class="py-lineno">  11</tt>  <tt class="py-line"><tt class="py-docstring">the "genbank" or "embl" format names to parse GenBank or EMBL files into</tt> </tt>
<a name="L12"></a><tt class="py-lineno">  12</tt>  <tt class="py-line"><tt class="py-docstring">SeqRecord and SeqFeature objects (see the Biopython tutorial for details).</tt> </tt>
<a name="L13"></a><tt class="py-lineno">  13</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L14"></a><tt class="py-lineno">  14</tt>  <tt class="py-line"><tt class="py-docstring">Using Bio.GenBank directly to parse GenBank files is only useful if you want</tt> </tt>
<a name="L15"></a><tt class="py-lineno">  15</tt>  <tt class="py-line"><tt class="py-docstring">to obtain GenBank-specific Record objects, which is a much closer</tt> </tt>
<a name="L16"></a><tt class="py-lineno">  16</tt>  <tt class="py-line"><tt class="py-docstring">representation to the raw file contents that the SeqRecord alternative from</tt> </tt>
<a name="L17"></a><tt class="py-lineno">  17</tt>  <tt class="py-line"><tt class="py-docstring">the FeatureParser (used in Bio.SeqIO).</tt> </tt>
<a name="L18"></a><tt class="py-lineno">  18</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L19"></a><tt class="py-lineno">  19</tt>  <tt class="py-line"><tt class="py-docstring">To use the Bio.GenBank parser, there are two helper functions:</tt> </tt>
<a name="L20"></a><tt class="py-lineno">  20</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L21"></a><tt class="py-lineno">  21</tt>  <tt class="py-line"><tt class="py-docstring">    - read                  Parse a handle containing a single GenBank record</tt> </tt>
<a name="L22"></a><tt class="py-lineno">  22</tt>  <tt class="py-line"><tt class="py-docstring">      as Bio.GenBank specific Record objects.</tt> </tt>
<a name="L23"></a><tt class="py-lineno">  23</tt>  <tt class="py-line"><tt class="py-docstring">    - parse                 Iterate over a handle containing multiple GenBank</tt> </tt>
<a name="L24"></a><tt class="py-lineno">  24</tt>  <tt class="py-line"><tt class="py-docstring">      records as Bio.GenBank specific Record objects.</tt> </tt>
<a name="L25"></a><tt class="py-lineno">  25</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L26"></a><tt class="py-lineno">  26</tt>  <tt class="py-line"><tt class="py-docstring">The following internal classes are not intended for direct use and may</tt> </tt>
<a name="L27"></a><tt class="py-lineno">  27</tt>  <tt class="py-line"><tt class="py-docstring">be deprecated in a future release.</tt> </tt>
<a name="L28"></a><tt class="py-lineno">  28</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L29"></a><tt class="py-lineno">  29</tt>  <tt class="py-line"><tt class="py-docstring">Classes:</tt> </tt>
<a name="L30"></a><tt class="py-lineno">  30</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L31"></a><tt class="py-lineno">  31</tt>  <tt class="py-line"><tt class="py-docstring">    - Iterator              Iterate through a file of GenBank entries</tt> </tt>
<a name="L32"></a><tt class="py-lineno">  32</tt>  <tt class="py-line"><tt class="py-docstring">    - ErrorFeatureParser    Catch errors caused during parsing.</tt> </tt>
<a name="L33"></a><tt class="py-lineno">  33</tt>  <tt class="py-line"><tt class="py-docstring">    - FeatureParser         Parse GenBank data in SeqRecord and SeqFeature objects.</tt> </tt>
<a name="L34"></a><tt class="py-lineno">  34</tt>  <tt class="py-line"><tt class="py-docstring">    - RecordParser          Parse GenBank data into a Record object.</tt> </tt>
<a name="L35"></a><tt class="py-lineno">  35</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L36"></a><tt class="py-lineno">  36</tt>  <tt class="py-line"><tt class="py-docstring">Exceptions:</tt> </tt>
<a name="L37"></a><tt class="py-lineno">  37</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L38"></a><tt class="py-lineno">  38</tt>  <tt class="py-line"><tt class="py-docstring">    - ParserFailureError    Exception indicating a failure in the parser (ie.</tt> </tt>
<a name="L39"></a><tt class="py-lineno">  39</tt>  <tt class="py-line"><tt class="py-docstring">      scanner or consumer)</tt> </tt>
<a name="L40"></a><tt class="py-lineno">  40</tt>  <tt class="py-line"><tt class="py-docstring">    - LocationParserError   Exception indicating a problem with the spark based</tt> </tt>
<a name="L41"></a><tt class="py-lineno">  41</tt>  <tt class="py-line"><tt class="py-docstring">      location parser.</tt> </tt>
<a name="L42"></a><tt class="py-lineno">  42</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L43"></a><tt class="py-lineno">  43</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L44"></a><tt class="py-lineno">  44</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">__future__</tt> <tt class="py-keyword">import</tt> <tt class="py-name">print_function</tt> </tt>
<a name="L45"></a><tt class="py-lineno">  45</tt>  <tt class="py-line"> </tt>
<a name="L46"></a><tt class="py-lineno">  46</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">re</tt> </tt>
<a name="L47"></a><tt class="py-lineno">  47</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">sys</tt>  <tt class="py-comment"># for checking if Python 2</tt> </tt>
<a name="L48"></a><tt class="py-lineno">  48</tt>  <tt class="py-line"> </tt>
<a name="L49"></a><tt class="py-lineno">  49</tt>  <tt class="py-line"><tt class="py-comment"># other Biopython stuff</tt> </tt>
<a name="L50"></a><tt class="py-lineno">  50</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-1" class="py-name" targets="Module Bio.SeqFeature=Bio.SeqFeature-module.html,Class Bio.SeqFeature.SeqFeature=Bio.SeqFeature.SeqFeature-class.html"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-1', 'SeqFeature', 'link-1');">SeqFeature</a></tt> </tt>
<a name="L51"></a><tt class="py-lineno">  51</tt>  <tt class="py-line"> </tt>
<a name="L52"></a><tt class="py-lineno">  52</tt>  <tt class="py-line"><tt class="py-comment"># other Bio.GenBank stuff</tt> </tt>
<a name="L53"></a><tt class="py-lineno">  53</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt><tt id="link-2" class="py-name" targets="Module Bio.GenBank.utils=Bio.GenBank.utils-module.html"><a title="Bio.GenBank.utils" class="py-name" href="#" onclick="return doclink('link-2', 'utils', 'link-2');">utils</a></tt> <tt class="py-keyword">import</tt> <tt id="link-3" class="py-name" targets="Class Bio.GenBank.utils.FeatureValueCleaner=Bio.GenBank.utils.FeatureValueCleaner-class.html"><a title="Bio.GenBank.utils.FeatureValueCleaner" class="py-name" href="#" onclick="return doclink('link-3', 'FeatureValueCleaner', 'link-3');">FeatureValueCleaner</a></tt> </tt>
<a name="L54"></a><tt class="py-lineno">  54</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt><tt id="link-4" class="py-name" targets="Module Bio.GenBank.Scanner=Bio.GenBank.Scanner-module.html"><a title="Bio.GenBank.Scanner" class="py-name" href="#" onclick="return doclink('link-4', 'Scanner', 'link-4');">Scanner</a></tt> <tt class="py-keyword">import</tt> <tt id="link-5" class="py-name" targets="Class Bio.GenBank.Scanner.GenBankScanner=Bio.GenBank.Scanner.GenBankScanner-class.html"><a title="Bio.GenBank.Scanner.GenBankScanner" class="py-name" href="#" onclick="return doclink('link-5', 'GenBankScanner', 'link-5');">GenBankScanner</a></tt> </tt>
<a name="L55"></a><tt class="py-lineno">  55</tt>  <tt class="py-line"> </tt>
<a name="L56"></a><tt class="py-lineno">  56</tt>  <tt class="py-line"><tt class="py-name">__docformat__</tt> <tt class="py-op">=</tt> <tt class="py-string">"restructuredtext en"</tt> </tt>
<a name="L57"></a><tt class="py-lineno">  57</tt>  <tt class="py-line"> </tt>
<a name="L58"></a><tt class="py-lineno">  58</tt>  <tt class="py-line"><tt class="py-comment"># Constants used to parse GenBank header lines</tt> </tt>
<a name="L59"></a><tt class="py-lineno">  59</tt>  <tt class="py-line"><tt id="link-6" class="py-name" targets="Variable Bio.GenBank.GENBANK_INDENT=Bio.GenBank-module.html#GENBANK_INDENT"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-6', 'GENBANK_INDENT', 'link-6');">GENBANK_INDENT</a></tt> <tt class="py-op">=</tt> <tt class="py-number">12</tt> </tt>
<a name="L60"></a><tt class="py-lineno">  60</tt>  <tt class="py-line"><tt id="link-7" class="py-name" targets="Variable Bio.GenBank.GENBANK_SPACER=Bio.GenBank-module.html#GENBANK_SPACER"><a title="Bio.GenBank.GENBANK_SPACER" class="py-name" href="#" onclick="return doclink('link-7', 'GENBANK_SPACER', 'link-7');">GENBANK_SPACER</a></tt> <tt class="py-op">=</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt id="link-8" class="py-name"><a title="Bio.GenBank.GENBANK_INDENT" class="py-name" href="#" onclick="return doclink('link-8', 'GENBANK_INDENT', 'link-6');">GENBANK_INDENT</a></tt> </tt>
<a name="L61"></a><tt class="py-lineno">  61</tt>  <tt class="py-line"> </tt>
<a name="L62"></a><tt class="py-lineno">  62</tt>  <tt class="py-line"><tt class="py-comment"># Constants for parsing GenBank feature lines</tt> </tt>
<a name="L63"></a><tt class="py-lineno">  63</tt>  <tt class="py-line"><tt id="link-9" class="py-name" targets="Variable Bio.GenBank.FEATURE_KEY_INDENT=Bio.GenBank-module.html#FEATURE_KEY_INDENT"><a title="Bio.GenBank.FEATURE_KEY_INDENT" class="py-name" href="#" onclick="return doclink('link-9', 'FEATURE_KEY_INDENT', 'link-9');">FEATURE_KEY_INDENT</a></tt> <tt class="py-op">=</tt> <tt class="py-number">5</tt> </tt>
<a name="L64"></a><tt class="py-lineno">  64</tt>  <tt class="py-line"><tt id="link-10" class="py-name" targets="Variable Bio.GenBank.FEATURE_QUALIFIER_INDENT=Bio.GenBank-module.html#FEATURE_QUALIFIER_INDENT,Variable Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_INDENT=Bio.GenBank.Scanner.EmblScanner-class.html#FEATURE_QUALIFIER_INDENT,Variable Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_INDENT=Bio.GenBank.Scanner.GenBankScanner-class.html#FEATURE_QUALIFIER_INDENT,Variable Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_INDENT=Bio.GenBank.Scanner.InsdcScanner-class.html#FEATURE_QUALIFIER_INDENT"><a title="Bio.GenBank.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-10', 'FEATURE_QUALIFIER_INDENT', 'link-10');">FEATURE_QUALIFIER_INDENT</a></tt> <tt class="py-op">=</tt> <tt class="py-number">21</tt> </tt>
<a name="L65"></a><tt class="py-lineno">  65</tt>  <tt class="py-line"><tt id="link-11" class="py-name" targets="Variable Bio.GenBank.FEATURE_KEY_SPACER=Bio.GenBank-module.html#FEATURE_KEY_SPACER"><a title="Bio.GenBank.FEATURE_KEY_SPACER" class="py-name" href="#" onclick="return doclink('link-11', 'FEATURE_KEY_SPACER', 'link-11');">FEATURE_KEY_SPACER</a></tt> <tt class="py-op">=</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt id="link-12" class="py-name"><a title="Bio.GenBank.FEATURE_KEY_INDENT" class="py-name" href="#" onclick="return doclink('link-12', 'FEATURE_KEY_INDENT', 'link-9');">FEATURE_KEY_INDENT</a></tt> </tt>
<a name="L66"></a><tt class="py-lineno">  66</tt>  <tt class="py-line"><tt id="link-13" class="py-name" targets="Variable Bio.GenBank.FEATURE_QUALIFIER_SPACER=Bio.GenBank-module.html#FEATURE_QUALIFIER_SPACER,Variable Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_SPACER=Bio.GenBank.Scanner.EmblScanner-class.html#FEATURE_QUALIFIER_SPACER,Variable Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_SPACER=Bio.GenBank.Scanner.GenBankScanner-class.html#FEATURE_QUALIFIER_SPACER,Variable Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_SPACER=Bio.GenBank.Scanner.InsdcScanner-class.html#FEATURE_QUALIFIER_SPACER"><a title="Bio.GenBank.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_SPACER
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_SPACER" class="py-name" href="#" onclick="return doclink('link-13', 'FEATURE_QUALIFIER_SPACER', 'link-13');">FEATURE_QUALIFIER_SPACER</a></tt> <tt class="py-op">=</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt id="link-14" class="py-name"><a title="Bio.GenBank.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.EmblScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.GenBankScanner.FEATURE_QUALIFIER_INDENT
Bio.GenBank.Scanner.InsdcScanner.FEATURE_QUALIFIER_INDENT" class="py-name" href="#" onclick="return doclink('link-14', 'FEATURE_QUALIFIER_INDENT', 'link-10');">FEATURE_QUALIFIER_INDENT</a></tt> </tt>
<a name="L67"></a><tt class="py-lineno">  67</tt>  <tt class="py-line"> </tt>
<a name="L68"></a><tt class="py-lineno">  68</tt>  <tt class="py-line"><tt class="py-comment"># Regular expressions for location parsing</tt> </tt>
<a name="L69"></a><tt class="py-lineno">  69</tt>  <tt class="py-line"><tt id="link-15" class="py-name" targets="Variable Bio.GenBank._solo_location=Bio.GenBank-module.html#_solo_location"><a title="Bio.GenBank._solo_location" class="py-name" href="#" onclick="return doclink('link-15', '_solo_location', 'link-15');">_solo_location</a></tt> <tt class="py-op">=</tt> <tt class="py-string">r"[&lt;&gt;]?\d+"</tt> </tt>
<a name="L70"></a><tt class="py-lineno">  70</tt>  <tt class="py-line"><tt id="link-16" class="py-name" targets="Variable Bio.GenBank._pair_location=Bio.GenBank-module.html#_pair_location"><a title="Bio.GenBank._pair_location" class="py-name" href="#" onclick="return doclink('link-16', '_pair_location', 'link-16');">_pair_location</a></tt> <tt class="py-op">=</tt> <tt class="py-string">r"[&lt;&gt;]?\d+\.\.[&lt;&gt;]?\d+"</tt> </tt>
<a name="L71"></a><tt class="py-lineno">  71</tt>  <tt class="py-line"><tt id="link-17" class="py-name" targets="Variable Bio.GenBank._between_location=Bio.GenBank-module.html#_between_location"><a title="Bio.GenBank._between_location" class="py-name" href="#" onclick="return doclink('link-17', '_between_location', 'link-17');">_between_location</a></tt> <tt class="py-op">=</tt> <tt class="py-string">r"\d+\^\d+"</tt> </tt>
<a name="L72"></a><tt class="py-lineno">  72</tt>  <tt class="py-line"> </tt>
<a name="L73"></a><tt class="py-lineno">  73</tt>  <tt class="py-line"><tt id="link-18" class="py-name" targets="Variable Bio.GenBank._within_position=Bio.GenBank-module.html#_within_position"><a title="Bio.GenBank._within_position" class="py-name" href="#" onclick="return doclink('link-18', '_within_position', 'link-18');">_within_position</a></tt> <tt class="py-op">=</tt> <tt class="py-string">r"\(\d+\.\d+\)"</tt> </tt>
<a name="L74"></a><tt class="py-lineno">  74</tt>  <tt class="py-line"><tt id="link-19" class="py-name" targets="Variable Bio.GenBank._re_within_position=Bio.GenBank-module.html#_re_within_position"><a title="Bio.GenBank._re_within_position" class="py-name" href="#" onclick="return doclink('link-19', '_re_within_position', 'link-19');">_re_within_position</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt id="link-20" class="py-name"><a title="Bio.GenBank._within_position" class="py-name" href="#" onclick="return doclink('link-20', '_within_position', 'link-18');">_within_position</a></tt><tt class="py-op">)</tt> </tt>
<a name="L75"></a><tt class="py-lineno">  75</tt>  <tt class="py-line"><tt id="link-21" class="py-name" targets="Variable Bio.GenBank._within_location=Bio.GenBank-module.html#_within_location"><a title="Bio.GenBank._within_location" class="py-name" href="#" onclick="return doclink('link-21', '_within_location', 'link-21');">_within_location</a></tt> <tt class="py-op">=</tt> <tt class="py-string">r"([&lt;&gt;]?\d+|%s)\.\.([&lt;&gt;]?\d+|%s)"</tt> \ </tt>
<a name="L76"></a><tt class="py-lineno">  76</tt>  <tt class="py-line">                   <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-22" class="py-name"><a title="Bio.GenBank._within_position" class="py-name" href="#" onclick="return doclink('link-22', '_within_position', 'link-18');">_within_position</a></tt><tt class="py-op">,</tt> <tt id="link-23" class="py-name"><a title="Bio.GenBank._within_position" class="py-name" href="#" onclick="return doclink('link-23', '_within_position', 'link-18');">_within_position</a></tt><tt class="py-op">)</tt> </tt>
<a name="L77"></a><tt class="py-lineno">  77</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt id="link-24" class="py-name"><a title="Bio.GenBank._re_within_position" class="py-name" href="#" onclick="return doclink('link-24', '_re_within_position', 'link-19');">_re_within_position</a></tt><tt class="py-op">.</tt><tt id="link-25" class="py-name" targets="Function Bio.triefind.match()=Bio.triefind-module.html#match"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-25', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"(3.9)"</tt><tt class="py-op">)</tt> </tt>
<a name="L78"></a><tt class="py-lineno">  78</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt id="link-26" class="py-name"><a title="Bio.GenBank._within_location" class="py-name" href="#" onclick="return doclink('link-26', '_within_location', 'link-21');">_within_location</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-27" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-27', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"(3.9)..10"</tt><tt class="py-op">)</tt> </tt>
<a name="L79"></a><tt class="py-lineno">  79</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt id="link-28" class="py-name"><a title="Bio.GenBank._within_location" class="py-name" href="#" onclick="return doclink('link-28', '_within_location', 'link-21');">_within_location</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-29" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-29', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"26..(30.33)"</tt><tt class="py-op">)</tt> </tt>
<a name="L80"></a><tt class="py-lineno">  80</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt id="link-30" class="py-name"><a title="Bio.GenBank._within_location" class="py-name" href="#" onclick="return doclink('link-30', '_within_location', 'link-21');">_within_location</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-31" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-31', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"(13.19)..(20.28)"</tt><tt class="py-op">)</tt> </tt>
<a name="L81"></a><tt class="py-lineno">  81</tt>  <tt class="py-line"> </tt>
<a name="L82"></a><tt class="py-lineno">  82</tt>  <tt class="py-line"><tt id="link-32" class="py-name" targets="Variable Bio.GenBank._oneof_position=Bio.GenBank-module.html#_oneof_position"><a title="Bio.GenBank._oneof_position" class="py-name" href="#" onclick="return doclink('link-32', '_oneof_position', 'link-32');">_oneof_position</a></tt> <tt class="py-op">=</tt> <tt class="py-string">r"one\-of\(\d+(,\d+)+\)"</tt> </tt>
<a name="L83"></a><tt class="py-lineno">  83</tt>  <tt class="py-line"><tt id="link-33" class="py-name" targets="Variable Bio.GenBank._re_oneof_position=Bio.GenBank-module.html#_re_oneof_position"><a title="Bio.GenBank._re_oneof_position" class="py-name" href="#" onclick="return doclink('link-33', '_re_oneof_position', 'link-33');">_re_oneof_position</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt id="link-34" class="py-name"><a title="Bio.GenBank._oneof_position" class="py-name" href="#" onclick="return doclink('link-34', '_oneof_position', 'link-32');">_oneof_position</a></tt><tt class="py-op">)</tt> </tt>
<a name="L84"></a><tt class="py-lineno">  84</tt>  <tt class="py-line"><tt id="link-35" class="py-name" targets="Variable Bio.GenBank._oneof_location=Bio.GenBank-module.html#_oneof_location"><a title="Bio.GenBank._oneof_location" class="py-name" href="#" onclick="return doclink('link-35', '_oneof_location', 'link-35');">_oneof_location</a></tt> <tt class="py-op">=</tt> <tt class="py-string">r"([&lt;&gt;]?\d+|%s)\.\.([&lt;&gt;]?\d+|%s)"</tt> \ </tt>
<a name="L85"></a><tt class="py-lineno">  85</tt>  <tt class="py-line">                   <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-36" class="py-name"><a title="Bio.GenBank._oneof_position" class="py-name" href="#" onclick="return doclink('link-36', '_oneof_position', 'link-32');">_oneof_position</a></tt><tt class="py-op">,</tt> <tt id="link-37" class="py-name"><a title="Bio.GenBank._oneof_position" class="py-name" href="#" onclick="return doclink('link-37', '_oneof_position', 'link-32');">_oneof_position</a></tt><tt class="py-op">)</tt> </tt>
<a name="L86"></a><tt class="py-lineno">  86</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt id="link-38" class="py-name"><a title="Bio.GenBank._re_oneof_position" class="py-name" href="#" onclick="return doclink('link-38', '_re_oneof_position', 'link-33');">_re_oneof_position</a></tt><tt class="py-op">.</tt><tt id="link-39" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-39', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"one-of(6,9)"</tt><tt class="py-op">)</tt> </tt>
<a name="L87"></a><tt class="py-lineno">  87</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt id="link-40" class="py-name"><a title="Bio.GenBank._oneof_location" class="py-name" href="#" onclick="return doclink('link-40', '_oneof_location', 'link-35');">_oneof_location</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-41" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-41', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"one-of(6,9)..101"</tt><tt class="py-op">)</tt> </tt>
<a name="L88"></a><tt class="py-lineno">  88</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt id="link-42" class="py-name"><a title="Bio.GenBank._oneof_location" class="py-name" href="#" onclick="return doclink('link-42', '_oneof_location', 'link-35');">_oneof_location</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-43" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-43', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"one-of(6,9)..one-of(101,104)"</tt><tt class="py-op">)</tt> </tt>
<a name="L89"></a><tt class="py-lineno">  89</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt id="link-44" class="py-name"><a title="Bio.GenBank._oneof_location" class="py-name" href="#" onclick="return doclink('link-44', '_oneof_location', 'link-35');">_oneof_location</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-45" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-45', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"6..one-of(101,104)"</tt><tt class="py-op">)</tt> </tt>
<a name="L90"></a><tt class="py-lineno">  90</tt>  <tt class="py-line"> </tt>
<a name="L91"></a><tt class="py-lineno">  91</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt id="link-46" class="py-name"><a title="Bio.GenBank._re_oneof_position" class="py-name" href="#" onclick="return doclink('link-46', '_re_oneof_position', 'link-33');">_re_oneof_position</a></tt><tt class="py-op">.</tt><tt id="link-47" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-47', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"one-of(3)"</tt><tt class="py-op">)</tt> </tt>
<a name="L92"></a><tt class="py-lineno">  92</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt id="link-48" class="py-name"><a title="Bio.GenBank._re_oneof_position" class="py-name" href="#" onclick="return doclink('link-48', '_re_oneof_position', 'link-33');">_re_oneof_position</a></tt><tt class="py-op">.</tt><tt id="link-49" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-49', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"one-of(3,6)"</tt><tt class="py-op">)</tt> </tt>
<a name="L93"></a><tt class="py-lineno">  93</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt id="link-50" class="py-name"><a title="Bio.GenBank._re_oneof_position" class="py-name" href="#" onclick="return doclink('link-50', '_re_oneof_position', 'link-33');">_re_oneof_position</a></tt><tt class="py-op">.</tt><tt id="link-51" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-51', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"one-of(3,6,9)"</tt><tt class="py-op">)</tt> </tt>
<a name="L94"></a><tt class="py-lineno">  94</tt>  <tt class="py-line"> </tt>
<a name="L95"></a><tt class="py-lineno">  95</tt>  <tt class="py-line"> </tt>
<a name="L96"></a><tt class="py-lineno">  96</tt>  <tt class="py-line"><tt id="link-52" class="py-name" targets="Variable Bio.GenBank._simple_location=Bio.GenBank-module.html#_simple_location"><a title="Bio.GenBank._simple_location" class="py-name" href="#" onclick="return doclink('link-52', '_simple_location', 'link-52');">_simple_location</a></tt> <tt class="py-op">=</tt> <tt class="py-string">r"\d+\.\.\d+"</tt> </tt>
<a name="L97"></a><tt class="py-lineno">  97</tt>  <tt class="py-line"><tt id="link-53" class="py-name" targets="Variable Bio.GenBank._re_simple_location=Bio.GenBank-module.html#_re_simple_location"><a title="Bio.GenBank._re_simple_location" class="py-name" href="#" onclick="return doclink('link-53', '_re_simple_location', 'link-53');">_re_simple_location</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r"^%s$"</tt> <tt class="py-op">%</tt> <tt id="link-54" class="py-name"><a title="Bio.GenBank._simple_location" class="py-name" href="#" onclick="return doclink('link-54', '_simple_location', 'link-52');">_simple_location</a></tt><tt class="py-op">)</tt> </tt>
<a name="L98"></a><tt class="py-lineno">  98</tt>  <tt class="py-line"><tt id="link-55" class="py-name" targets="Variable Bio.GenBank._re_simple_compound=Bio.GenBank-module.html#_re_simple_compound"><a title="Bio.GenBank._re_simple_compound" class="py-name" href="#" onclick="return doclink('link-55', '_re_simple_compound', 'link-55');">_re_simple_compound</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r"^(join|order|bond)\(%s(,%s)*\)$"</tt> </tt>
<a name="L99"></a><tt class="py-lineno">  99</tt>  <tt class="py-line">                                 <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-56" class="py-name"><a title="Bio.GenBank._simple_location" class="py-name" href="#" onclick="return doclink('link-56', '_simple_location', 'link-52');">_simple_location</a></tt><tt class="py-op">,</tt> <tt id="link-57" class="py-name"><a title="Bio.GenBank._simple_location" class="py-name" href="#" onclick="return doclink('link-57', '_simple_location', 'link-52');">_simple_location</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L100"></a><tt class="py-lineno"> 100</tt>  <tt class="py-line"><tt id="link-58" class="py-name" targets="Variable Bio.GenBank._complex_location=Bio.GenBank-module.html#_complex_location"><a title="Bio.GenBank._complex_location" class="py-name" href="#" onclick="return doclink('link-58', '_complex_location', 'link-58');">_complex_location</a></tt> <tt class="py-op">=</tt> <tt class="py-string">r"([a-zA-z][a-zA-Z0-9_]*(\.[a-zA-Z0-9]+)?\:)?(%s|%s|%s|%s|%s)"</tt> \ </tt>
<a name="L101"></a><tt class="py-lineno"> 101</tt>  <tt class="py-line">                    <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-59" class="py-name"><a title="Bio.GenBank._pair_location" class="py-name" href="#" onclick="return doclink('link-59', '_pair_location', 'link-16');">_pair_location</a></tt><tt class="py-op">,</tt> <tt id="link-60" class="py-name"><a title="Bio.GenBank._solo_location" class="py-name" href="#" onclick="return doclink('link-60', '_solo_location', 'link-15');">_solo_location</a></tt><tt class="py-op">,</tt> <tt id="link-61" class="py-name"><a title="Bio.GenBank._between_location" class="py-name" href="#" onclick="return doclink('link-61', '_between_location', 'link-17');">_between_location</a></tt><tt class="py-op">,</tt> </tt>
<a name="L102"></a><tt class="py-lineno"> 102</tt>  <tt class="py-line">                       <tt id="link-62" class="py-name"><a title="Bio.GenBank._within_location" class="py-name" href="#" onclick="return doclink('link-62', '_within_location', 'link-21');">_within_location</a></tt><tt class="py-op">,</tt> <tt id="link-63" class="py-name"><a title="Bio.GenBank._oneof_location" class="py-name" href="#" onclick="return doclink('link-63', '_oneof_location', 'link-35');">_oneof_location</a></tt><tt class="py-op">)</tt> </tt>
<a name="L103"></a><tt class="py-lineno"> 103</tt>  <tt class="py-line"><tt id="link-64" class="py-name" targets="Variable Bio.GenBank._re_complex_location=Bio.GenBank-module.html#_re_complex_location"><a title="Bio.GenBank._re_complex_location" class="py-name" href="#" onclick="return doclink('link-64', '_re_complex_location', 'link-64');">_re_complex_location</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r"^%s$"</tt> <tt class="py-op">%</tt> <tt id="link-65" class="py-name"><a title="Bio.GenBank._complex_location" class="py-name" href="#" onclick="return doclink('link-65', '_complex_location', 'link-58');">_complex_location</a></tt><tt class="py-op">)</tt> </tt>
<a name="L104"></a><tt class="py-lineno"> 104</tt>  <tt class="py-line"><tt id="link-66" class="py-name" targets="Variable Bio.GenBank._possibly_complemented_complex_location=Bio.GenBank-module.html#_possibly_complemented_complex_location"><a title="Bio.GenBank._possibly_complemented_complex_location" class="py-name" href="#" onclick="return doclink('link-66', '_possibly_complemented_complex_location', 'link-66');">_possibly_complemented_complex_location</a></tt> <tt class="py-op">=</tt> <tt class="py-string">r"(%s|complement\(%s\))"</tt> \ </tt>
<a name="L105"></a><tt class="py-lineno"> 105</tt>  <tt class="py-line">                                          <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-67" class="py-name"><a title="Bio.GenBank._complex_location" class="py-name" href="#" onclick="return doclink('link-67', '_complex_location', 'link-58');">_complex_location</a></tt><tt class="py-op">,</tt> <tt id="link-68" class="py-name"><a title="Bio.GenBank._complex_location" class="py-name" href="#" onclick="return doclink('link-68', '_complex_location', 'link-58');">_complex_location</a></tt><tt class="py-op">)</tt> </tt>
<a name="L106"></a><tt class="py-lineno"> 106</tt>  <tt class="py-line"><tt id="link-69" class="py-name" targets="Variable Bio.GenBank._re_complex_compound=Bio.GenBank-module.html#_re_complex_compound"><a title="Bio.GenBank._re_complex_compound" class="py-name" href="#" onclick="return doclink('link-69', '_re_complex_compound', 'link-69');">_re_complex_compound</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r"^(join|order|bond)\(%s(,%s)*\)$"</tt> </tt>
<a name="L107"></a><tt class="py-lineno"> 107</tt>  <tt class="py-line">                                 <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-70" class="py-name"><a title="Bio.GenBank._possibly_complemented_complex_location" class="py-name" href="#" onclick="return doclink('link-70', '_possibly_complemented_complex_location', 'link-66');">_possibly_complemented_complex_location</a></tt><tt class="py-op">,</tt> </tt>
<a name="L108"></a><tt class="py-lineno"> 108</tt>  <tt class="py-line">                                    <tt id="link-71" class="py-name"><a title="Bio.GenBank._possibly_complemented_complex_location" class="py-name" href="#" onclick="return doclink('link-71', '_possibly_complemented_complex_location', 'link-66');">_possibly_complemented_complex_location</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L109"></a><tt class="py-lineno"> 109</tt>  <tt class="py-line"> </tt>
<a name="L110"></a><tt class="py-lineno"> 110</tt>  <tt class="py-line"> </tt>
<a name="L111"></a><tt class="py-lineno"> 111</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt id="link-72" class="py-name"><a title="Bio.GenBank._re_simple_location" class="py-name" href="#" onclick="return doclink('link-72', '_re_simple_location', 'link-53');">_re_simple_location</a></tt><tt class="py-op">.</tt><tt id="link-73" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-73', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"104..160"</tt><tt class="py-op">)</tt> </tt>
<a name="L112"></a><tt class="py-lineno"> 112</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt id="link-74" class="py-name"><a title="Bio.GenBank._re_simple_location" class="py-name" href="#" onclick="return doclink('link-74', '_re_simple_location', 'link-53');">_re_simple_location</a></tt><tt class="py-op">.</tt><tt id="link-75" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-75', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"68451760..68452073^68452074"</tt><tt class="py-op">)</tt> </tt>
<a name="L113"></a><tt class="py-lineno"> 113</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt id="link-76" class="py-name"><a title="Bio.GenBank._re_simple_location" class="py-name" href="#" onclick="return doclink('link-76', '_re_simple_location', 'link-53');">_re_simple_location</a></tt><tt class="py-op">.</tt><tt id="link-77" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-77', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"&lt;104..&gt;160"</tt><tt class="py-op">)</tt> </tt>
<a name="L114"></a><tt class="py-lineno"> 114</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt id="link-78" class="py-name"><a title="Bio.GenBank._re_simple_location" class="py-name" href="#" onclick="return doclink('link-78', '_re_simple_location', 'link-53');">_re_simple_location</a></tt><tt class="py-op">.</tt><tt id="link-79" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-79', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"104"</tt><tt class="py-op">)</tt> </tt>
<a name="L115"></a><tt class="py-lineno"> 115</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt id="link-80" class="py-name"><a title="Bio.GenBank._re_simple_location" class="py-name" href="#" onclick="return doclink('link-80', '_re_simple_location', 'link-53');">_re_simple_location</a></tt><tt class="py-op">.</tt><tt id="link-81" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-81', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"&lt;1"</tt><tt class="py-op">)</tt> </tt>
<a name="L116"></a><tt class="py-lineno"> 116</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt id="link-82" class="py-name"><a title="Bio.GenBank._re_simple_location" class="py-name" href="#" onclick="return doclink('link-82', '_re_simple_location', 'link-53');">_re_simple_location</a></tt><tt class="py-op">.</tt><tt id="link-83" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-83', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"&gt;99999"</tt><tt class="py-op">)</tt> </tt>
<a name="L117"></a><tt class="py-lineno"> 117</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt id="link-84" class="py-name"><a title="Bio.GenBank._re_simple_location" class="py-name" href="#" onclick="return doclink('link-84', '_re_simple_location', 'link-53');">_re_simple_location</a></tt><tt class="py-op">.</tt><tt id="link-85" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-85', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"join(104..160,320..390,504..579)"</tt><tt class="py-op">)</tt> </tt>
<a name="L118"></a><tt class="py-lineno"> 118</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt id="link-86" class="py-name"><a title="Bio.GenBank._re_simple_compound" class="py-name" href="#" onclick="return doclink('link-86', '_re_simple_compound', 'link-55');">_re_simple_compound</a></tt><tt class="py-op">.</tt><tt id="link-87" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-87', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"bond(12,63)"</tt><tt class="py-op">)</tt> </tt>
<a name="L119"></a><tt class="py-lineno"> 119</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt id="link-88" class="py-name"><a title="Bio.GenBank._re_simple_compound" class="py-name" href="#" onclick="return doclink('link-88', '_re_simple_compound', 'link-55');">_re_simple_compound</a></tt><tt class="py-op">.</tt><tt id="link-89" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-89', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"join(104..160,320..390,504..579)"</tt><tt class="py-op">)</tt> </tt>
<a name="L120"></a><tt class="py-lineno"> 120</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt id="link-90" class="py-name"><a title="Bio.GenBank._re_simple_compound" class="py-name" href="#" onclick="return doclink('link-90', '_re_simple_compound', 'link-55');">_re_simple_compound</a></tt><tt class="py-op">.</tt><tt id="link-91" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-91', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"order(1..69,1308..1465)"</tt><tt class="py-op">)</tt> </tt>
<a name="L121"></a><tt class="py-lineno"> 121</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt id="link-92" class="py-name"><a title="Bio.GenBank._re_simple_compound" class="py-name" href="#" onclick="return doclink('link-92', '_re_simple_compound', 'link-55');">_re_simple_compound</a></tt><tt class="py-op">.</tt><tt id="link-93" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-93', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"order(1..69,1308..1465,1524)"</tt><tt class="py-op">)</tt> </tt>
<a name="L122"></a><tt class="py-lineno"> 122</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt id="link-94" class="py-name"><a title="Bio.GenBank._re_simple_compound" class="py-name" href="#" onclick="return doclink('link-94', '_re_simple_compound', 'link-55');">_re_simple_compound</a></tt><tt class="py-op">.</tt><tt id="link-95" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-95', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"join(&lt;1..442,992..1228,1524..&gt;1983)"</tt><tt class="py-op">)</tt> </tt>
<a name="L123"></a><tt class="py-lineno"> 123</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt id="link-96" class="py-name"><a title="Bio.GenBank._re_simple_compound" class="py-name" href="#" onclick="return doclink('link-96', '_re_simple_compound', 'link-55');">_re_simple_compound</a></tt><tt class="py-op">.</tt><tt id="link-97" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-97', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"join(&lt;1..181,254..336,422..497,574..&gt;590)"</tt><tt class="py-op">)</tt> </tt>
<a name="L124"></a><tt class="py-lineno"> 124</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt id="link-98" class="py-name"><a title="Bio.GenBank._re_simple_compound" class="py-name" href="#" onclick="return doclink('link-98', '_re_simple_compound', 'link-55');">_re_simple_compound</a></tt><tt class="py-op">.</tt><tt id="link-99" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-99', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"join(1475..1577,2841..2986,3074..3193,3314..3481,4126..&gt;4215)"</tt><tt class="py-op">)</tt> </tt>
<a name="L125"></a><tt class="py-lineno"> 125</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt id="link-100" class="py-name"><a title="Bio.GenBank._re_simple_compound" class="py-name" href="#" onclick="return doclink('link-100', '_re_simple_compound', 'link-55');">_re_simple_compound</a></tt><tt class="py-op">.</tt><tt id="link-101" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-101', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"test(1..69,1308..1465)"</tt><tt class="py-op">)</tt> </tt>
<a name="L126"></a><tt class="py-lineno"> 126</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt id="link-102" class="py-name"><a title="Bio.GenBank._re_simple_compound" class="py-name" href="#" onclick="return doclink('link-102', '_re_simple_compound', 'link-55');">_re_simple_compound</a></tt><tt class="py-op">.</tt><tt id="link-103" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-103', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"complement(1..69)"</tt><tt class="py-op">)</tt> </tt>
<a name="L127"></a><tt class="py-lineno"> 127</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt id="link-104" class="py-name"><a title="Bio.GenBank._re_simple_compound" class="py-name" href="#" onclick="return doclink('link-104', '_re_simple_compound', 'link-55');">_re_simple_compound</a></tt><tt class="py-op">.</tt><tt id="link-105" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-105', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"(1..69)"</tt><tt class="py-op">)</tt> </tt>
<a name="L128"></a><tt class="py-lineno"> 128</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt id="link-106" class="py-name"><a title="Bio.GenBank._re_complex_location" class="py-name" href="#" onclick="return doclink('link-106', '_re_complex_location', 'link-64');">_re_complex_location</a></tt><tt class="py-op">.</tt><tt id="link-107" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-107', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"(3.9)..10"</tt><tt class="py-op">)</tt> </tt>
<a name="L129"></a><tt class="py-lineno"> 129</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt id="link-108" class="py-name"><a title="Bio.GenBank._re_complex_location" class="py-name" href="#" onclick="return doclink('link-108', '_re_complex_location', 'link-64');">_re_complex_location</a></tt><tt class="py-op">.</tt><tt id="link-109" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-109', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"26..(30.33)"</tt><tt class="py-op">)</tt> </tt>
<a name="L130"></a><tt class="py-lineno"> 130</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt id="link-110" class="py-name"><a title="Bio.GenBank._re_complex_location" class="py-name" href="#" onclick="return doclink('link-110', '_re_complex_location', 'link-64');">_re_complex_location</a></tt><tt class="py-op">.</tt><tt id="link-111" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-111', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"(13.19)..(20.28)"</tt><tt class="py-op">)</tt> </tt>
<a name="L131"></a><tt class="py-lineno"> 131</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt id="link-112" class="py-name"><a title="Bio.GenBank._re_complex_location" class="py-name" href="#" onclick="return doclink('link-112', '_re_complex_location', 'link-64');">_re_complex_location</a></tt><tt class="py-op">.</tt><tt id="link-113" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-113', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"41^42"</tt><tt class="py-op">)</tt>  <tt class="py-comment"># between</tt> </tt>
<a name="L132"></a><tt class="py-lineno"> 132</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt id="link-114" class="py-name"><a title="Bio.GenBank._re_complex_location" class="py-name" href="#" onclick="return doclink('link-114', '_re_complex_location', 'link-64');">_re_complex_location</a></tt><tt class="py-op">.</tt><tt id="link-115" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-115', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"AL121804:41^42"</tt><tt class="py-op">)</tt> </tt>
<a name="L133"></a><tt class="py-lineno"> 133</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt id="link-116" class="py-name"><a title="Bio.GenBank._re_complex_location" class="py-name" href="#" onclick="return doclink('link-116', '_re_complex_location', 'link-64');">_re_complex_location</a></tt><tt class="py-op">.</tt><tt id="link-117" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-117', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"AL121804:41..610"</tt><tt class="py-op">)</tt> </tt>
<a name="L134"></a><tt class="py-lineno"> 134</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt id="link-118" class="py-name"><a title="Bio.GenBank._re_complex_location" class="py-name" href="#" onclick="return doclink('link-118', '_re_complex_location', 'link-64');">_re_complex_location</a></tt><tt class="py-op">.</tt><tt id="link-119" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-119', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"AL121804.2:41..610"</tt><tt class="py-op">)</tt> </tt>
<a name="L135"></a><tt class="py-lineno"> 135</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt id="link-120" class="py-name"><a title="Bio.GenBank._re_complex_location" class="py-name" href="#" onclick="return doclink('link-120', '_re_complex_location', 'link-64');">_re_complex_location</a></tt><tt class="py-op">.</tt><tt id="link-121" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-121', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"one-of(3,6)..101"</tt><tt class="py-op">)</tt> </tt>
<a name="L136"></a><tt class="py-lineno"> 136</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt id="link-122" class="py-name"><a title="Bio.GenBank._re_complex_compound" class="py-name" href="#" onclick="return doclink('link-122', '_re_complex_compound', 'link-69');">_re_complex_compound</a></tt><tt class="py-op">.</tt><tt id="link-123" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-123', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"join(153490..154269,AL121804.2:41..610,AL121804.2:672..1487)"</tt><tt class="py-op">)</tt> </tt>
<a name="L137"></a><tt class="py-lineno"> 137</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt id="link-124" class="py-name"><a title="Bio.GenBank._re_simple_compound" class="py-name" href="#" onclick="return doclink('link-124', '_re_simple_compound', 'link-55');">_re_simple_compound</a></tt><tt class="py-op">.</tt><tt id="link-125" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-125', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"join(153490..154269,AL121804.2:41..610,AL121804.2:672..1487)"</tt><tt class="py-op">)</tt> </tt>
<a name="L138"></a><tt class="py-lineno"> 138</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt id="link-126" class="py-name"><a title="Bio.GenBank._re_complex_compound" class="py-name" href="#" onclick="return doclink('link-126', '_re_complex_compound', 'link-69');">_re_complex_compound</a></tt><tt class="py-op">.</tt><tt id="link-127" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-127', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"join(complement(69611..69724),139856..140650)"</tt><tt class="py-op">)</tt> </tt>
<a name="L139"></a><tt class="py-lineno"> 139</tt>  <tt class="py-line"> </tt>
<a name="L140"></a><tt class="py-lineno"> 140</tt>  <tt class="py-line"><tt class="py-comment"># Trans-spliced example from NC_016406, note underscore in reference name:</tt> </tt>
<a name="L141"></a><tt class="py-lineno"> 141</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt id="link-128" class="py-name"><a title="Bio.GenBank._re_complex_location" class="py-name" href="#" onclick="return doclink('link-128', '_re_complex_location', 'link-64');">_re_complex_location</a></tt><tt class="py-op">.</tt><tt id="link-129" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-129', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"NC_016402.1:6618..6676"</tt><tt class="py-op">)</tt> </tt>
<a name="L142"></a><tt class="py-lineno"> 142</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt id="link-130" class="py-name"><a title="Bio.GenBank._re_complex_location" class="py-name" href="#" onclick="return doclink('link-130', '_re_complex_location', 'link-64');">_re_complex_location</a></tt><tt class="py-op">.</tt><tt id="link-131" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-131', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"181647..181905"</tt><tt class="py-op">)</tt> </tt>
<a name="L143"></a><tt class="py-lineno"> 143</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt id="link-132" class="py-name"><a title="Bio.GenBank._re_complex_compound" class="py-name" href="#" onclick="return doclink('link-132', '_re_complex_compound', 'link-69');">_re_complex_compound</a></tt><tt class="py-op">.</tt><tt id="link-133" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-133', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"join(complement(149815..150200),complement(293787..295573),NC_016402.1:6618..6676,181647..181905)"</tt><tt class="py-op">)</tt> </tt>
<a name="L144"></a><tt class="py-lineno"> 144</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt id="link-134" class="py-name"><a title="Bio.GenBank._re_complex_location" class="py-name" href="#" onclick="return doclink('link-134', '_re_complex_location', 'link-64');">_re_complex_location</a></tt><tt class="py-op">.</tt><tt id="link-135" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-135', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"join(complement(149815..150200),complement(293787..295573),NC_016402.1:6618..6676,181647..181905)"</tt><tt class="py-op">)</tt> </tt>
<a name="L145"></a><tt class="py-lineno"> 145</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt id="link-136" class="py-name"><a title="Bio.GenBank._re_simple_compound" class="py-name" href="#" onclick="return doclink('link-136', '_re_simple_compound', 'link-55');">_re_simple_compound</a></tt><tt class="py-op">.</tt><tt id="link-137" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-137', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"join(complement(149815..150200),complement(293787..295573),NC_016402.1:6618..6676,181647..181905)"</tt><tt class="py-op">)</tt> </tt>
<a name="L146"></a><tt class="py-lineno"> 146</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt id="link-138" class="py-name"><a title="Bio.GenBank._re_complex_location" class="py-name" href="#" onclick="return doclink('link-138', '_re_complex_location', 'link-64');">_re_complex_location</a></tt><tt class="py-op">.</tt><tt id="link-139" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-139', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"join(complement(149815..150200),complement(293787..295573),NC_016402.1:6618..6676,181647..181905)"</tt><tt class="py-op">)</tt> </tt>
<a name="L147"></a><tt class="py-lineno"> 147</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt id="link-140" class="py-name"><a title="Bio.GenBank._re_simple_location" class="py-name" href="#" onclick="return doclink('link-140', '_re_simple_location', 'link-53');">_re_simple_location</a></tt><tt class="py-op">.</tt><tt id="link-141" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-141', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"join(complement(149815..150200),complement(293787..295573),NC_016402.1:6618..6676,181647..181905)"</tt><tt class="py-op">)</tt> </tt>
<a name="L148"></a><tt class="py-lineno"> 148</tt>  <tt class="py-line"> </tt>
<a name="L149"></a><tt class="py-lineno"> 149</tt>  <tt class="py-line"><tt id="link-142" class="py-name" targets="Variable Bio.GenBank._solo_bond=Bio.GenBank-module.html#_solo_bond"><a title="Bio.GenBank._solo_bond" class="py-name" href="#" onclick="return doclink('link-142', '_solo_bond', 'link-142');">_solo_bond</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">"bond\(%s\)"</tt> <tt class="py-op">%</tt> <tt id="link-143" class="py-name"><a title="Bio.GenBank._solo_location" class="py-name" href="#" onclick="return doclink('link-143', '_solo_location', 'link-15');">_solo_location</a></tt><tt class="py-op">)</tt> </tt>
<a name="L150"></a><tt class="py-lineno"> 150</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt id="link-144" class="py-name"><a title="Bio.GenBank._solo_bond" class="py-name" href="#" onclick="return doclink('link-144', '_solo_bond', 'link-142');">_solo_bond</a></tt><tt class="py-op">.</tt><tt id="link-145" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-145', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-string">"bond(196)"</tt><tt class="py-op">)</tt> </tt>
<a name="L151"></a><tt class="py-lineno"> 151</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt id="link-146" class="py-name"><a title="Bio.GenBank._solo_bond" class="py-name" href="#" onclick="return doclink('link-146', '_solo_bond', 'link-142');">_solo_bond</a></tt><tt class="py-op">.</tt><tt id="link-147" class="py-name" targets="Method Bio.KDTree.KDTree'.KDTree.search()=Bio.KDTree.KDTree%27.KDTree-class.html#search,Method Bio.PDB.NeighborSearch'.NeighborSearch.search()=Bio.PDB.NeighborSearch%27.NeighborSearch-class.html#search,Method Bio.Phylo.TreeConstruction.NNITreeSearcher.search()=Bio.Phylo.TreeConstruction.NNITreeSearcher-class.html#search,Method Bio.Phylo.TreeConstruction.TreeSearcher.search()=Bio.Phylo.TreeConstruction.TreeSearcher-class.html#search,Class Method Bio.Restriction.Restriction.AbstractCut.search()=Bio.Restriction.Restriction.AbstractCut-class.html#search,Method Bio.Restriction.Restriction.RestrictionBatch.search()=Bio.Restriction.Restriction.RestrictionBatch-class.html#search,Function Bio.SCOP.search()=Bio.SCOP-module.html#search,Function Bio.TogoWS.search()=Bio.TogoWS-module.html#search,Method Bio.motifs.Instances.search()=Bio.motifs.Instances-class.html#search,Method Bio.motifs.matrix.PositionSpecificScoringMatrix.search()=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#search"><a title="Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-147', 'search', 'link-147');">search</a></tt><tt class="py-op">(</tt><tt class="py-string">"bond(196)"</tt><tt class="py-op">)</tt> </tt>
<a name="L152"></a><tt class="py-lineno"> 152</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt id="link-148" class="py-name"><a title="Bio.GenBank._solo_bond" class="py-name" href="#" onclick="return doclink('link-148', '_solo_bond', 'link-142');">_solo_bond</a></tt><tt class="py-op">.</tt><tt id="link-149" class="py-name"><a title="Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-149', 'search', 'link-147');">search</a></tt><tt class="py-op">(</tt><tt class="py-string">"join(bond(284),bond(305),bond(309),bond(305))"</tt><tt class="py-op">)</tt> </tt>
<a name="L153"></a><tt class="py-lineno"> 153</tt>  <tt class="py-line"> </tt>
<a name="L154"></a><tt class="py-lineno"> 154</tt>  <tt class="py-line"> </tt>
<a name="_pos"></a><div id="_pos-def"><a name="L155"></a><tt class="py-lineno"> 155</tt> <a class="py-toggle" href="#" id="_pos-toggle" onclick="return toggle('_pos');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank-module.html#_pos">_pos</a><tt class="py-op">(</tt><tt class="py-param">pos_str</tt><tt class="py-op">,</tt> <tt class="py-param">offset</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_pos-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_pos-expanded"><a name="L156"></a><tt class="py-lineno"> 156</tt>  <tt class="py-line">    <tt class="py-docstring">"""Build a Position object (PRIVATE).</tt> </tt>
<a name="L157"></a><tt class="py-lineno"> 157</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L158"></a><tt class="py-lineno"> 158</tt>  <tt class="py-line"><tt class="py-docstring">    For an end position, leave offset as zero (default):</tt> </tt>
<a name="L159"></a><tt class="py-lineno"> 159</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L160"></a><tt class="py-lineno"> 160</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _pos("5")</tt> </tt>
<a name="L161"></a><tt class="py-lineno"> 161</tt>  <tt class="py-line"><tt class="py-docstring">    ExactPosition(5)</tt> </tt>
<a name="L162"></a><tt class="py-lineno"> 162</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L163"></a><tt class="py-lineno"> 163</tt>  <tt class="py-line"><tt class="py-docstring">    For a start position, set offset to minus one (for Python counting):</tt> </tt>
<a name="L164"></a><tt class="py-lineno"> 164</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L165"></a><tt class="py-lineno"> 165</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _pos("5", -1)</tt> </tt>
<a name="L166"></a><tt class="py-lineno"> 166</tt>  <tt class="py-line"><tt class="py-docstring">    ExactPosition(4)</tt> </tt>
<a name="L167"></a><tt class="py-lineno"> 167</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L168"></a><tt class="py-lineno"> 168</tt>  <tt class="py-line"><tt class="py-docstring">    This also covers fuzzy positions:</tt> </tt>
<a name="L169"></a><tt class="py-lineno"> 169</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L170"></a><tt class="py-lineno"> 170</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; p = _pos("&lt;5")</tt> </tt>
<a name="L171"></a><tt class="py-lineno"> 171</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; p</tt> </tt>
<a name="L172"></a><tt class="py-lineno"> 172</tt>  <tt class="py-line"><tt class="py-docstring">    BeforePosition(5)</tt> </tt>
<a name="L173"></a><tt class="py-lineno"> 173</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(p)</tt> </tt>
<a name="L174"></a><tt class="py-lineno"> 174</tt>  <tt class="py-line"><tt class="py-docstring">    &lt;5</tt> </tt>
<a name="L175"></a><tt class="py-lineno"> 175</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; int(p)</tt> </tt>
<a name="L176"></a><tt class="py-lineno"> 176</tt>  <tt class="py-line"><tt class="py-docstring">    5</tt> </tt>
<a name="L177"></a><tt class="py-lineno"> 177</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L178"></a><tt class="py-lineno"> 178</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _pos("&gt;5")</tt> </tt>
<a name="L179"></a><tt class="py-lineno"> 179</tt>  <tt class="py-line"><tt class="py-docstring">    AfterPosition(5)</tt> </tt>
<a name="L180"></a><tt class="py-lineno"> 180</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L181"></a><tt class="py-lineno"> 181</tt>  <tt class="py-line"><tt class="py-docstring">    By default assumes an end position, so note the integer behaviour:</tt> </tt>
<a name="L182"></a><tt class="py-lineno"> 182</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L183"></a><tt class="py-lineno"> 183</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; p = _pos("one-of(5,8,11)")</tt> </tt>
<a name="L184"></a><tt class="py-lineno"> 184</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; p</tt> </tt>
<a name="L185"></a><tt class="py-lineno"> 185</tt>  <tt class="py-line"><tt class="py-docstring">    OneOfPosition(11, choices=[ExactPosition(5), ExactPosition(8), ExactPosition(11)])</tt> </tt>
<a name="L186"></a><tt class="py-lineno"> 186</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(p)</tt> </tt>
<a name="L187"></a><tt class="py-lineno"> 187</tt>  <tt class="py-line"><tt class="py-docstring">    one-of(5,8,11)</tt> </tt>
<a name="L188"></a><tt class="py-lineno"> 188</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; int(p)</tt> </tt>
<a name="L189"></a><tt class="py-lineno"> 189</tt>  <tt class="py-line"><tt class="py-docstring">    11</tt> </tt>
<a name="L190"></a><tt class="py-lineno"> 190</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L191"></a><tt class="py-lineno"> 191</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _pos("(8.10)")</tt> </tt>
<a name="L192"></a><tt class="py-lineno"> 192</tt>  <tt class="py-line"><tt class="py-docstring">    WithinPosition(10, left=8, right=10)</tt> </tt>
<a name="L193"></a><tt class="py-lineno"> 193</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L194"></a><tt class="py-lineno"> 194</tt>  <tt class="py-line"><tt class="py-docstring">    Fuzzy start positions:</tt> </tt>
<a name="L195"></a><tt class="py-lineno"> 195</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L196"></a><tt class="py-lineno"> 196</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; p = _pos("&lt;5", -1)</tt> </tt>
<a name="L197"></a><tt class="py-lineno"> 197</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; p</tt> </tt>
<a name="L198"></a><tt class="py-lineno"> 198</tt>  <tt class="py-line"><tt class="py-docstring">    BeforePosition(4)</tt> </tt>
<a name="L199"></a><tt class="py-lineno"> 199</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(p)</tt> </tt>
<a name="L200"></a><tt class="py-lineno"> 200</tt>  <tt class="py-line"><tt class="py-docstring">    &lt;4</tt> </tt>
<a name="L201"></a><tt class="py-lineno"> 201</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; int(p)</tt> </tt>
<a name="L202"></a><tt class="py-lineno"> 202</tt>  <tt class="py-line"><tt class="py-docstring">    4</tt> </tt>
<a name="L203"></a><tt class="py-lineno"> 203</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L204"></a><tt class="py-lineno"> 204</tt>  <tt class="py-line"><tt class="py-docstring">    Notice how the integer behaviour changes too!</tt> </tt>
<a name="L205"></a><tt class="py-lineno"> 205</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L206"></a><tt class="py-lineno"> 206</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; p = _pos("one-of(5,8,11)", -1)</tt> </tt>
<a name="L207"></a><tt class="py-lineno"> 207</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; p</tt> </tt>
<a name="L208"></a><tt class="py-lineno"> 208</tt>  <tt class="py-line"><tt class="py-docstring">    OneOfPosition(4, choices=[ExactPosition(4), ExactPosition(7), ExactPosition(10)])</tt> </tt>
<a name="L209"></a><tt class="py-lineno"> 209</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(p)</tt> </tt>
<a name="L210"></a><tt class="py-lineno"> 210</tt>  <tt class="py-line"><tt class="py-docstring">    one-of(4,7,10)</tt> </tt>
<a name="L211"></a><tt class="py-lineno"> 211</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; int(p)</tt> </tt>
<a name="L212"></a><tt class="py-lineno"> 212</tt>  <tt class="py-line"><tt class="py-docstring">    4</tt> </tt>
<a name="L213"></a><tt class="py-lineno"> 213</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L214"></a><tt class="py-lineno"> 214</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L215"></a><tt class="py-lineno"> 215</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">pos_str</tt><tt class="py-op">.</tt><tt id="link-150" class="py-name" targets="Method Bio.Seq.Seq.startswith()=Bio.Seq.Seq-class.html#startswith"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-150', 'startswith', 'link-150');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"&lt;"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L216"></a><tt class="py-lineno"> 216</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-151" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-151', 'SeqFeature', 'link-1');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-152" class="py-name" targets="Class Bio.SeqFeature.BeforePosition=Bio.SeqFeature.BeforePosition-class.html"><a title="Bio.SeqFeature.BeforePosition" class="py-name" href="#" onclick="return doclink('link-152', 'BeforePosition', 'link-152');">BeforePosition</a></tt><tt class="py-op">(</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">pos_str</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-name">offset</tt><tt class="py-op">)</tt> </tt>
<a name="L217"></a><tt class="py-lineno"> 217</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">pos_str</tt><tt class="py-op">.</tt><tt id="link-153" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-153', 'startswith', 'link-150');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"&gt;"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L218"></a><tt class="py-lineno"> 218</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-154" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-154', 'SeqFeature', 'link-1');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-155" class="py-name" targets="Class Bio.SeqFeature.AfterPosition=Bio.SeqFeature.AfterPosition-class.html"><a title="Bio.SeqFeature.AfterPosition" class="py-name" href="#" onclick="return doclink('link-155', 'AfterPosition', 'link-155');">AfterPosition</a></tt><tt class="py-op">(</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">pos_str</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-name">offset</tt><tt class="py-op">)</tt> </tt>
<a name="L219"></a><tt class="py-lineno"> 219</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt id="link-156" class="py-name"><a title="Bio.GenBank._re_within_position" class="py-name" href="#" onclick="return doclink('link-156', '_re_within_position', 'link-19');">_re_within_position</a></tt><tt class="py-op">.</tt><tt id="link-157" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-157', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-name">pos_str</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L220"></a><tt class="py-lineno"> 220</tt>  <tt class="py-line">        <tt class="py-name">s</tt><tt class="py-op">,</tt> <tt class="py-name">e</tt> <tt class="py-op">=</tt> <tt class="py-name">pos_str</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-158" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-158', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">"."</tt><tt class="py-op">)</tt> </tt>
<a name="L221"></a><tt class="py-lineno"> 221</tt>  <tt class="py-line">        <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-name">offset</tt> </tt>
<a name="L222"></a><tt class="py-lineno"> 222</tt>  <tt class="py-line">        <tt class="py-name">e</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">e</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-name">offset</tt> </tt>
<a name="L223"></a><tt class="py-lineno"> 223</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">offset</tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L224"></a><tt class="py-lineno"> 224</tt>  <tt class="py-line">            <tt class="py-name">default</tt> <tt class="py-op">=</tt> <tt class="py-name">s</tt> </tt>
<a name="L225"></a><tt class="py-lineno"> 225</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L226"></a><tt class="py-lineno"> 226</tt>  <tt class="py-line">            <tt class="py-name">default</tt> <tt class="py-op">=</tt> <tt class="py-name">e</tt> </tt>
<a name="L227"></a><tt class="py-lineno"> 227</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-159" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-159', 'SeqFeature', 'link-1');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-160" class="py-name" targets="Class Bio.SeqFeature.WithinPosition=Bio.SeqFeature.WithinPosition-class.html"><a title="Bio.SeqFeature.WithinPosition" class="py-name" href="#" onclick="return doclink('link-160', 'WithinPosition', 'link-160');">WithinPosition</a></tt><tt class="py-op">(</tt><tt class="py-name">default</tt><tt class="py-op">,</tt> <tt class="py-name">left</tt><tt class="py-op">=</tt><tt class="py-name">s</tt><tt class="py-op">,</tt> <tt class="py-name">right</tt><tt class="py-op">=</tt><tt class="py-name">e</tt><tt class="py-op">)</tt> </tt>
<a name="L228"></a><tt class="py-lineno"> 228</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt id="link-161" class="py-name"><a title="Bio.GenBank._re_oneof_position" class="py-name" href="#" onclick="return doclink('link-161', '_re_oneof_position', 'link-33');">_re_oneof_position</a></tt><tt class="py-op">.</tt><tt id="link-162" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-162', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-name">pos_str</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L229"></a><tt class="py-lineno"> 229</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">pos_str</tt><tt class="py-op">.</tt><tt id="link-163" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-163', 'startswith', 'link-150');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"one-of("</tt><tt class="py-op">)</tt> </tt>
<a name="L230"></a><tt class="py-lineno"> 230</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">pos_str</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">")"</tt> </tt>
<a name="L231"></a><tt class="py-lineno"> 231</tt>  <tt class="py-line">        <tt id="link-164" class="py-name" targets="Variable Bio.SeqFeature.FeatureLocation.parts=Bio.SeqFeature.FeatureLocation-class.html#parts"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-164', 'parts', 'link-164');">parts</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-165" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-165', 'SeqFeature', 'link-1');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-166" class="py-name" targets="Class Bio.SeqFeature.ExactPosition=Bio.SeqFeature.ExactPosition-class.html"><a title="Bio.SeqFeature.ExactPosition" class="py-name" href="#" onclick="return doclink('link-166', 'ExactPosition', 'link-166');">ExactPosition</a></tt><tt class="py-op">(</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">pos</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-name">offset</tt><tt class="py-op">)</tt> </tt>
<a name="L232"></a><tt class="py-lineno"> 232</tt>  <tt class="py-line">                 <tt class="py-keyword">for</tt> <tt class="py-name">pos</tt> <tt class="py-keyword">in</tt> <tt class="py-name">pos_str</tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-167" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-167', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">","</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L233"></a><tt class="py-lineno"> 233</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">offset</tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L234"></a><tt class="py-lineno"> 234</tt>  <tt class="py-line">            <tt class="py-name">default</tt> <tt class="py-op">=</tt> <tt id="link-168" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.min=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#min"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-168', 'min', 'link-168');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">pos</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">pos</tt> <tt class="py-keyword">in</tt> <tt id="link-169" class="py-name"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-169', 'parts', 'link-164');">parts</a></tt><tt class="py-op">)</tt> </tt>
<a name="L235"></a><tt class="py-lineno"> 235</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L236"></a><tt class="py-lineno"> 236</tt>  <tt class="py-line">            <tt class="py-name">default</tt> <tt class="py-op">=</tt> <tt id="link-170" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.max=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#max"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-170', 'max', 'link-170');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">pos</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">pos</tt> <tt class="py-keyword">in</tt> <tt id="link-171" class="py-name"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-171', 'parts', 'link-164');">parts</a></tt><tt class="py-op">)</tt> </tt>
<a name="L237"></a><tt class="py-lineno"> 237</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-172" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-172', 'SeqFeature', 'link-1');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-173" class="py-name" targets="Class Bio.SeqFeature.OneOfPosition=Bio.SeqFeature.OneOfPosition-class.html"><a title="Bio.SeqFeature.OneOfPosition" class="py-name" href="#" onclick="return doclink('link-173', 'OneOfPosition', 'link-173');">OneOfPosition</a></tt><tt class="py-op">(</tt><tt class="py-name">default</tt><tt class="py-op">,</tt> <tt class="py-name">choices</tt><tt class="py-op">=</tt><tt id="link-174" class="py-name"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-174', 'parts', 'link-164');">parts</a></tt><tt class="py-op">)</tt> </tt>
<a name="L238"></a><tt class="py-lineno"> 238</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L239"></a><tt class="py-lineno"> 239</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-175" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-175', 'SeqFeature', 'link-1');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-176" class="py-name"><a title="Bio.SeqFeature.ExactPosition" class="py-name" href="#" onclick="return doclink('link-176', 'ExactPosition', 'link-166');">ExactPosition</a></tt><tt class="py-op">(</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">pos_str</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-name">offset</tt><tt class="py-op">)</tt> </tt>
</div><a name="L240"></a><tt class="py-lineno"> 240</tt>  <tt class="py-line"> </tt>
<a name="L241"></a><tt class="py-lineno"> 241</tt>  <tt class="py-line"> </tt>
<a name="_loc"></a><div id="_loc-def"><a name="L242"></a><tt class="py-lineno"> 242</tt> <a class="py-toggle" href="#" id="_loc-toggle" onclick="return toggle('_loc');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank-module.html#_loc">_loc</a><tt class="py-op">(</tt><tt class="py-param">loc_str</tt><tt class="py-op">,</tt> <tt class="py-param">expected_seq_length</tt><tt class="py-op">,</tt> <tt class="py-param">strand</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_loc-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_loc-expanded"><a name="L243"></a><tt class="py-lineno"> 243</tt>  <tt class="py-line">    <tt class="py-docstring">"""FeatureLocation from non-compound non-complement location (PRIVATE).</tt> </tt>
<a name="L244"></a><tt class="py-lineno"> 244</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L245"></a><tt class="py-lineno"> 245</tt>  <tt class="py-line"><tt class="py-docstring">    Simple examples,</tt> </tt>
<a name="L246"></a><tt class="py-lineno"> 246</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L247"></a><tt class="py-lineno"> 247</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _loc("123..456", 1000, +1)</tt> </tt>
<a name="L248"></a><tt class="py-lineno"> 248</tt>  <tt class="py-line"><tt class="py-docstring">    FeatureLocation(ExactPosition(122), ExactPosition(456), strand=1)</tt> </tt>
<a name="L249"></a><tt class="py-lineno"> 249</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _loc("&lt;123..&gt;456", 1000, strand = -1)</tt> </tt>
<a name="L250"></a><tt class="py-lineno"> 250</tt>  <tt class="py-line"><tt class="py-docstring">    FeatureLocation(BeforePosition(122), AfterPosition(456), strand=-1)</tt> </tt>
<a name="L251"></a><tt class="py-lineno"> 251</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L252"></a><tt class="py-lineno"> 252</tt>  <tt class="py-line"><tt class="py-docstring">    A more complex location using within positions,</tt> </tt>
<a name="L253"></a><tt class="py-lineno"> 253</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L254"></a><tt class="py-lineno"> 254</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _loc("(9.10)..(20.25)", 1000, 1)</tt> </tt>
<a name="L255"></a><tt class="py-lineno"> 255</tt>  <tt class="py-line"><tt class="py-docstring">    FeatureLocation(WithinPosition(8, left=8, right=9), WithinPosition(25, left=20, right=25), strand=1)</tt> </tt>
<a name="L256"></a><tt class="py-lineno"> 256</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L257"></a><tt class="py-lineno"> 257</tt>  <tt class="py-line"><tt class="py-docstring">    Notice how that will act as though it has overall start 8 and end 25.</tt> </tt>
<a name="L258"></a><tt class="py-lineno"> 258</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L259"></a><tt class="py-lineno"> 259</tt>  <tt class="py-line"><tt class="py-docstring">    Zero length between feature,</tt> </tt>
<a name="L260"></a><tt class="py-lineno"> 260</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L261"></a><tt class="py-lineno"> 261</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _loc("123^124", 1000, 0)</tt> </tt>
<a name="L262"></a><tt class="py-lineno"> 262</tt>  <tt class="py-line"><tt class="py-docstring">    FeatureLocation(ExactPosition(123), ExactPosition(123), strand=0)</tt> </tt>
<a name="L263"></a><tt class="py-lineno"> 263</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L264"></a><tt class="py-lineno"> 264</tt>  <tt class="py-line"><tt class="py-docstring">    The expected sequence length is needed for a special case, a between</tt> </tt>
<a name="L265"></a><tt class="py-lineno"> 265</tt>  <tt class="py-line"><tt class="py-docstring">    position at the start/end of a circular genome:</tt> </tt>
<a name="L266"></a><tt class="py-lineno"> 266</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L267"></a><tt class="py-lineno"> 267</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _loc("1000^1", 1000, 1)</tt> </tt>
<a name="L268"></a><tt class="py-lineno"> 268</tt>  <tt class="py-line"><tt class="py-docstring">    FeatureLocation(ExactPosition(1000), ExactPosition(1000), strand=1)</tt> </tt>
<a name="L269"></a><tt class="py-lineno"> 269</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L270"></a><tt class="py-lineno"> 270</tt>  <tt class="py-line"><tt class="py-docstring">    Apart from this special case, between positions P^Q must have P+1==Q,</tt> </tt>
<a name="L271"></a><tt class="py-lineno"> 271</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L272"></a><tt class="py-lineno"> 272</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _loc("123^456", 1000, 1)</tt> </tt>
<a name="L273"></a><tt class="py-lineno"> 273</tt>  <tt class="py-line"><tt class="py-docstring">    Traceback (most recent call last):</tt> </tt>
<a name="L274"></a><tt class="py-lineno"> 274</tt>  <tt class="py-line"><tt class="py-docstring">       ...</tt> </tt>
<a name="L275"></a><tt class="py-lineno"> 275</tt>  <tt class="py-line"><tt class="py-docstring">    ValueError: Invalid between location '123^456'</tt> </tt>
<a name="L276"></a><tt class="py-lineno"> 276</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L277"></a><tt class="py-lineno"> 277</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L278"></a><tt class="py-lineno"> 278</tt>  <tt class="py-line">        <tt class="py-name">s</tt><tt class="py-op">,</tt> <tt class="py-name">e</tt> <tt class="py-op">=</tt> <tt class="py-name">loc_str</tt><tt class="py-op">.</tt><tt id="link-177" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-177', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">".."</tt><tt class="py-op">)</tt> </tt>
<a name="L279"></a><tt class="py-lineno"> 279</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L280"></a><tt class="py-lineno"> 280</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-string">".."</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">loc_str</tt> </tt>
<a name="L281"></a><tt class="py-lineno"> 281</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"^"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">loc_str</tt><tt class="py-op">:</tt> </tt>
<a name="L282"></a><tt class="py-lineno"> 282</tt>  <tt class="py-line">            <tt class="py-comment"># A between location like "67^68" (one based counting) is a</tt> </tt>
<a name="L283"></a><tt class="py-lineno"> 283</tt>  <tt class="py-line">            <tt class="py-comment"># special case (note it has zero length). In python slice</tt> </tt>
<a name="L284"></a><tt class="py-lineno"> 284</tt>  <tt class="py-line">            <tt class="py-comment"># notation this is 67:67, a zero length slice.  See Bug 2622</tt> </tt>
<a name="L285"></a><tt class="py-lineno"> 285</tt>  <tt class="py-line">            <tt class="py-comment"># Further more, on a circular genome of length N you can have</tt> </tt>
<a name="L286"></a><tt class="py-lineno"> 286</tt>  <tt class="py-line">            <tt class="py-comment"># a location N^1 meaning the junction at the origin. See Bug 3098.</tt> </tt>
<a name="L287"></a><tt class="py-lineno"> 287</tt>  <tt class="py-line">            <tt class="py-comment"># NOTE - We can imagine between locations like "2^4", but this</tt> </tt>
<a name="L288"></a><tt class="py-lineno"> 288</tt>  <tt class="py-line">            <tt class="py-comment"># is just "3".  Similarly, "2^5" is just "3..4"</tt> </tt>
<a name="L289"></a><tt class="py-lineno"> 289</tt>  <tt class="py-line">            <tt class="py-name">s</tt><tt class="py-op">,</tt> <tt class="py-name">e</tt> <tt class="py-op">=</tt> <tt class="py-name">loc_str</tt><tt class="py-op">.</tt><tt id="link-178" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-178', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">"^"</tt><tt class="py-op">)</tt> </tt>
<a name="L290"></a><tt class="py-lineno"> 290</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt> <tt class="py-op">==</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">e</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L291"></a><tt class="py-lineno"> 291</tt>  <tt class="py-line">                <tt class="py-name">pos</tt> <tt class="py-op">=</tt> <tt class="py-name">_pos</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">)</tt> </tt>
<a name="L292"></a><tt class="py-lineno"> 292</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">expected_seq_length</tt> <tt class="py-keyword">and</tt> <tt class="py-name">e</tt> <tt class="py-op">==</tt> <tt class="py-string">"1"</tt><tt class="py-op">:</tt> </tt>
<a name="L293"></a><tt class="py-lineno"> 293</tt>  <tt class="py-line">                <tt class="py-name">pos</tt> <tt class="py-op">=</tt> <tt class="py-name">_pos</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">)</tt> </tt>
<a name="L294"></a><tt class="py-lineno"> 294</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L295"></a><tt class="py-lineno"> 295</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid between location %s"</tt> <tt class="py-op">%</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">loc_str</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L296"></a><tt class="py-lineno"> 296</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-179" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-179', 'SeqFeature', 'link-1');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-180" class="py-name" targets="Class Bio.SeqFeature.FeatureLocation=Bio.SeqFeature.FeatureLocation-class.html"><a title="Bio.SeqFeature.FeatureLocation" class="py-name" href="#" onclick="return doclink('link-180', 'FeatureLocation', 'link-180');">FeatureLocation</a></tt><tt class="py-op">(</tt><tt class="py-name">pos</tt><tt class="py-op">,</tt> <tt class="py-name">pos</tt><tt class="py-op">,</tt> <tt id="link-181" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HSPConsumer.strand()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#strand,Variable Bio.SeqFeature.CompoundLocation.strand=Bio.SeqFeature.CompoundLocation-class.html#strand,Variable Bio.SeqFeature.FeatureLocation.strand=Bio.SeqFeature.FeatureLocation-class.html#strand,Variable Bio.SeqFeature.SeqFeature.strand=Bio.SeqFeature.SeqFeature-class.html#strand"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-181', 'strand', 'link-181');">strand</a></tt><tt class="py-op">)</tt> </tt>
<a name="L297"></a><tt class="py-lineno"> 297</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L298"></a><tt class="py-lineno"> 298</tt>  <tt class="py-line">            <tt class="py-comment"># e.g. "123"</tt> </tt>
<a name="L299"></a><tt class="py-lineno"> 299</tt>  <tt class="py-line">            <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt class="py-name">loc_str</tt> </tt>
<a name="L300"></a><tt class="py-lineno"> 300</tt>  <tt class="py-line">            <tt class="py-name">e</tt> <tt class="py-op">=</tt> <tt class="py-name">loc_str</tt> </tt>
<a name="L301"></a><tt class="py-lineno"> 301</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-182" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-182', 'SeqFeature', 'link-1');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-183" class="py-name"><a title="Bio.SeqFeature.FeatureLocation" class="py-name" href="#" onclick="return doclink('link-183', 'FeatureLocation', 'link-180');">FeatureLocation</a></tt><tt class="py-op">(</tt><tt class="py-name">_pos</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">,</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">_pos</tt><tt class="py-op">(</tt><tt class="py-name">e</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-184" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-184', 'strand', 'link-181');">strand</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L302"></a><tt class="py-lineno"> 302</tt>  <tt class="py-line"> </tt>
<a name="L303"></a><tt class="py-lineno"> 303</tt>  <tt class="py-line"> </tt>
<a name="_split_compound_loc"></a><div id="_split_compound_loc-def"><a name="L304"></a><tt class="py-lineno"> 304</tt> <a class="py-toggle" href="#" id="_split_compound_loc-toggle" onclick="return toggle('_split_compound_loc');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank-module.html#_split_compound_loc">_split_compound_loc</a><tt class="py-op">(</tt><tt class="py-param">compound_loc</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_split_compound_loc-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_split_compound_loc-expanded"><a name="L305"></a><tt class="py-lineno"> 305</tt>  <tt class="py-line">    <tt class="py-docstring">"""Split a tricky compound location string (PRIVATE).</tt> </tt>
<a name="L306"></a><tt class="py-lineno"> 306</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L307"></a><tt class="py-lineno"> 307</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; list(_split_compound_loc("123..145"))</tt> </tt>
<a name="L308"></a><tt class="py-lineno"> 308</tt>  <tt class="py-line"><tt class="py-docstring">    ['123..145']</tt> </tt>
<a name="L309"></a><tt class="py-lineno"> 309</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; list(_split_compound_loc("123..145,200..209"))</tt> </tt>
<a name="L310"></a><tt class="py-lineno"> 310</tt>  <tt class="py-line"><tt class="py-docstring">    ['123..145', '200..209']</tt> </tt>
<a name="L311"></a><tt class="py-lineno"> 311</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; list(_split_compound_loc("one-of(200,203)..300"))</tt> </tt>
<a name="L312"></a><tt class="py-lineno"> 312</tt>  <tt class="py-line"><tt class="py-docstring">    ['one-of(200,203)..300']</tt> </tt>
<a name="L313"></a><tt class="py-lineno"> 313</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; list(_split_compound_loc("complement(123..145),200..209"))</tt> </tt>
<a name="L314"></a><tt class="py-lineno"> 314</tt>  <tt class="py-line"><tt class="py-docstring">    ['complement(123..145)', '200..209']</tt> </tt>
<a name="L315"></a><tt class="py-lineno"> 315</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; list(_split_compound_loc("123..145,one-of(200,203)..209"))</tt> </tt>
<a name="L316"></a><tt class="py-lineno"> 316</tt>  <tt class="py-line"><tt class="py-docstring">    ['123..145', 'one-of(200,203)..209']</tt> </tt>
<a name="L317"></a><tt class="py-lineno"> 317</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; list(_split_compound_loc("123..145,one-of(200,203)..one-of(209,211),300"))</tt> </tt>
<a name="L318"></a><tt class="py-lineno"> 318</tt>  <tt class="py-line"><tt class="py-docstring">    ['123..145', 'one-of(200,203)..one-of(209,211)', '300']</tt> </tt>
<a name="L319"></a><tt class="py-lineno"> 319</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; list(_split_compound_loc("123..145,complement(one-of(200,203)..one-of(209,211)),300"))</tt> </tt>
<a name="L320"></a><tt class="py-lineno"> 320</tt>  <tt class="py-line"><tt class="py-docstring">    ['123..145', 'complement(one-of(200,203)..one-of(209,211))', '300']</tt> </tt>
<a name="L321"></a><tt class="py-lineno"> 321</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; list(_split_compound_loc("123..145,200..one-of(209,211),300"))</tt> </tt>
<a name="L322"></a><tt class="py-lineno"> 322</tt>  <tt class="py-line"><tt class="py-docstring">    ['123..145', '200..one-of(209,211)', '300']</tt> </tt>
<a name="L323"></a><tt class="py-lineno"> 323</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; list(_split_compound_loc("123..145,200..one-of(209,211)"))</tt> </tt>
<a name="L324"></a><tt class="py-lineno"> 324</tt>  <tt class="py-line"><tt class="py-docstring">    ['123..145', '200..one-of(209,211)']</tt> </tt>
<a name="L325"></a><tt class="py-lineno"> 325</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; list(_split_compound_loc("complement(149815..150200),complement(293787..295573),NC_016402.1:6618..6676,181647..181905"))</tt> </tt>
<a name="L326"></a><tt class="py-lineno"> 326</tt>  <tt class="py-line"><tt class="py-docstring">    ['complement(149815..150200)', 'complement(293787..295573)', 'NC_016402.1:6618..6676', '181647..181905']</tt> </tt>
<a name="L327"></a><tt class="py-lineno"> 327</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L328"></a><tt class="py-lineno"> 328</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-string">"one-of("</tt> <tt class="py-keyword">in</tt> <tt class="py-name">compound_loc</tt><tt class="py-op">:</tt> </tt>
<a name="L329"></a><tt class="py-lineno"> 329</tt>  <tt class="py-line">        <tt class="py-comment"># Hard case</tt> </tt>
<a name="L330"></a><tt class="py-lineno"> 330</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-string">","</tt> <tt class="py-keyword">in</tt> <tt class="py-name">compound_loc</tt><tt class="py-op">:</tt> </tt>
<a name="L331"></a><tt class="py-lineno"> 331</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">compound_loc</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-string">","</tt> </tt>
<a name="L332"></a><tt class="py-lineno"> 332</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">compound_loc</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">:</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-string">".."</tt> </tt>
<a name="L333"></a><tt class="py-lineno"> 333</tt>  <tt class="py-line">            <tt id="link-185" class="py-name" targets="Variable Bio.PDB.Polypeptide.i=Bio.PDB.Polypeptide-module.html#i,Variable Bio.SeqIO.SffIO.i=Bio.SeqIO.SffIO-module.html#i"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-185', 'i', 'link-185');">i</a></tt> <tt class="py-op">=</tt> <tt class="py-name">compound_loc</tt><tt class="py-op">.</tt><tt id="link-186" class="py-name" targets="Method Bio.NeuralNetwork.Gene.Motif.MotifFinder.find()=Bio.NeuralNetwork.Gene.Motif.MotifFinder-class.html#find,Method Bio.NeuralNetwork.Gene.Schema.SchemaFinder.find()=Bio.NeuralNetwork.Gene.Schema.SchemaFinder-class.html#find,Method Bio.NeuralNetwork.Gene.Signature.SignatureFinder.find()=Bio.NeuralNetwork.Gene.Signature.SignatureFinder-class.html#find,Method Bio.Seq.Seq.find()=Bio.Seq.Seq-class.html#find,Function Bio.triefind.find()=Bio.triefind-module.html#find"><a title="Bio.NeuralNetwork.Gene.Motif.MotifFinder.find
Bio.NeuralNetwork.Gene.Schema.SchemaFinder.find
Bio.NeuralNetwork.Gene.Signature.SignatureFinder.find
Bio.Seq.Seq.find
Bio.triefind.find" class="py-name" href="#" onclick="return doclink('link-186', 'find', 'link-186');">find</a></tt><tt class="py-op">(</tt><tt class="py-string">","</tt><tt class="py-op">)</tt> </tt>
<a name="L334"></a><tt class="py-lineno"> 334</tt>  <tt class="py-line">            <tt class="py-name">part</tt> <tt class="py-op">=</tt> <tt class="py-name">compound_loc</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt id="link-187" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-187', 'i', 'link-185');">i</a></tt><tt class="py-op">]</tt> </tt>
<a name="L335"></a><tt class="py-lineno"> 335</tt>  <tt class="py-line">            <tt class="py-name">compound_loc</tt> <tt class="py-op">=</tt> <tt class="py-name">compound_loc</tt><tt class="py-op">[</tt><tt id="link-188" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-188', 'i', 'link-185');">i</a></tt><tt class="py-op">:</tt><tt class="py-op">]</tt>  <tt class="py-comment"># includes the comma</tt> </tt>
<a name="L336"></a><tt class="py-lineno"> 336</tt>  <tt class="py-line">            <tt class="py-keyword">while</tt> <tt class="py-name">part</tt><tt class="py-op">.</tt><tt id="link-189" class="py-name" targets="Method Bio.Crystal.Chain.count()=Bio.Crystal.Chain-class.html#count,Method Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count()=Bio.NeuralNetwork.Gene.Pattern.PatternRepository-class.html#count,Method Bio.Seq.MutableSeq.count()=Bio.Seq.MutableSeq-class.html#count,Method Bio.Seq.Seq.count()=Bio.Seq.Seq-class.html#count,Method Bio.Seq.UnknownSeq.count()=Bio.Seq.UnknownSeq-class.html#count,Method Bio.motifs.Instances.count()=Bio.motifs.Instances-class.html#count"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-189', 'count', 'link-189');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">"("</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-name">part</tt><tt class="py-op">.</tt><tt id="link-190" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-190', 'count', 'link-189');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">")"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L337"></a><tt class="py-lineno"> 337</tt>  <tt class="py-line">                <tt class="py-keyword">assert</tt> <tt class="py-string">"one-of("</tt> <tt class="py-keyword">in</tt> <tt class="py-name">part</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">part</tt><tt class="py-op">,</tt> <tt class="py-name">compound_loc</tt><tt class="py-op">)</tt> </tt>
<a name="L338"></a><tt class="py-lineno"> 338</tt>  <tt class="py-line">                <tt id="link-191" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-191', 'i', 'link-185');">i</a></tt> <tt class="py-op">=</tt> <tt class="py-name">compound_loc</tt><tt class="py-op">.</tt><tt id="link-192" class="py-name"><a title="Bio.NeuralNetwork.Gene.Motif.MotifFinder.find
Bio.NeuralNetwork.Gene.Schema.SchemaFinder.find
Bio.NeuralNetwork.Gene.Signature.SignatureFinder.find
Bio.Seq.Seq.find
Bio.triefind.find" class="py-name" href="#" onclick="return doclink('link-192', 'find', 'link-186');">find</a></tt><tt class="py-op">(</tt><tt class="py-string">")"</tt><tt class="py-op">)</tt> </tt>
<a name="L339"></a><tt class="py-lineno"> 339</tt>  <tt class="py-line">                <tt class="py-name">part</tt> <tt class="py-op">+=</tt> <tt class="py-name">compound_loc</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt id="link-193" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-193', 'i', 'link-185');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L340"></a><tt class="py-lineno"> 340</tt>  <tt class="py-line">                <tt class="py-name">compound_loc</tt> <tt class="py-op">=</tt> <tt class="py-name">compound_loc</tt><tt class="py-op">[</tt><tt id="link-194" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-194', 'i', 'link-185');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L341"></a><tt class="py-lineno"> 341</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">compound_loc</tt><tt class="py-op">.</tt><tt id="link-195" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-195', 'startswith', 'link-150');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">".."</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L342"></a><tt class="py-lineno"> 342</tt>  <tt class="py-line">                <tt id="link-196" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-196', 'i', 'link-185');">i</a></tt> <tt class="py-op">=</tt> <tt class="py-name">compound_loc</tt><tt class="py-op">.</tt><tt id="link-197" class="py-name"><a title="Bio.NeuralNetwork.Gene.Motif.MotifFinder.find
Bio.NeuralNetwork.Gene.Schema.SchemaFinder.find
Bio.NeuralNetwork.Gene.Signature.SignatureFinder.find
Bio.Seq.Seq.find
Bio.triefind.find" class="py-name" href="#" onclick="return doclink('link-197', 'find', 'link-186');">find</a></tt><tt class="py-op">(</tt><tt class="py-string">","</tt><tt class="py-op">)</tt> </tt>
<a name="L343"></a><tt class="py-lineno"> 343</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-198" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-198', 'i', 'link-185');">i</a></tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L344"></a><tt class="py-lineno"> 344</tt>  <tt class="py-line">                    <tt class="py-name">part</tt> <tt class="py-op">+=</tt> <tt class="py-name">compound_loc</tt> </tt>
<a name="L345"></a><tt class="py-lineno"> 345</tt>  <tt class="py-line">                    <tt class="py-name">compound_loc</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L346"></a><tt class="py-lineno"> 346</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L347"></a><tt class="py-lineno"> 347</tt>  <tt class="py-line">                    <tt class="py-name">part</tt> <tt class="py-op">+=</tt> <tt class="py-name">compound_loc</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt id="link-199" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-199', 'i', 'link-185');">i</a></tt><tt class="py-op">]</tt> </tt>
<a name="L348"></a><tt class="py-lineno"> 348</tt>  <tt class="py-line">                    <tt class="py-name">compound_loc</tt> <tt class="py-op">=</tt> <tt class="py-name">compound_loc</tt><tt class="py-op">[</tt><tt id="link-200" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-200', 'i', 'link-185');">i</a></tt><tt class="py-op">:</tt><tt class="py-op">]</tt>  <tt class="py-comment"># includes the comma</tt> </tt>
<a name="L349"></a><tt class="py-lineno"> 349</tt>  <tt class="py-line">            <tt class="py-keyword">while</tt> <tt class="py-name">part</tt><tt class="py-op">.</tt><tt id="link-201" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-201', 'count', 'link-189');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">"("</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-name">part</tt><tt class="py-op">.</tt><tt id="link-202" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-202', 'count', 'link-189');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">")"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L350"></a><tt class="py-lineno"> 350</tt>  <tt class="py-line">                <tt class="py-keyword">assert</tt> <tt class="py-name">part</tt><tt class="py-op">.</tt><tt id="link-203" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-203', 'count', 'link-189');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">"one-of("</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">2</tt> </tt>
<a name="L351"></a><tt class="py-lineno"> 351</tt>  <tt class="py-line">                <tt id="link-204" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-204', 'i', 'link-185');">i</a></tt> <tt class="py-op">=</tt> <tt class="py-name">compound_loc</tt><tt class="py-op">.</tt><tt id="link-205" class="py-name"><a title="Bio.NeuralNetwork.Gene.Motif.MotifFinder.find
Bio.NeuralNetwork.Gene.Schema.SchemaFinder.find
Bio.NeuralNetwork.Gene.Signature.SignatureFinder.find
Bio.Seq.Seq.find
Bio.triefind.find" class="py-name" href="#" onclick="return doclink('link-205', 'find', 'link-186');">find</a></tt><tt class="py-op">(</tt><tt class="py-string">")"</tt><tt class="py-op">)</tt> </tt>
<a name="L352"></a><tt class="py-lineno"> 352</tt>  <tt class="py-line">                <tt class="py-name">part</tt> <tt class="py-op">+=</tt> <tt class="py-name">compound_loc</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt id="link-206" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-206', 'i', 'link-185');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L353"></a><tt class="py-lineno"> 353</tt>  <tt class="py-line">                <tt class="py-name">compound_loc</tt> <tt class="py-op">=</tt> <tt class="py-name">compound_loc</tt><tt class="py-op">[</tt><tt id="link-207" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-207', 'i', 'link-185');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L354"></a><tt class="py-lineno"> 354</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">compound_loc</tt><tt class="py-op">.</tt><tt id="link-208" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-208', 'startswith', 'link-150');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">","</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L355"></a><tt class="py-lineno"> 355</tt>  <tt class="py-line">                <tt class="py-name">compound_loc</tt> <tt class="py-op">=</tt> <tt class="py-name">compound_loc</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L356"></a><tt class="py-lineno"> 356</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">part</tt> </tt>
<a name="L357"></a><tt class="py-lineno"> 357</tt>  <tt class="py-line">            <tt class="py-keyword">yield</tt> <tt class="py-name">part</tt> </tt>
<a name="L358"></a><tt class="py-lineno"> 358</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">compound_loc</tt><tt class="py-op">:</tt> </tt>
<a name="L359"></a><tt class="py-lineno"> 359</tt>  <tt class="py-line">            <tt class="py-keyword">yield</tt> <tt class="py-name">compound_loc</tt> </tt>
<a name="L360"></a><tt class="py-lineno"> 360</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L361"></a><tt class="py-lineno"> 361</tt>  <tt class="py-line">        <tt class="py-comment"># Easy case</tt> </tt>
<a name="L362"></a><tt class="py-lineno"> 362</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">part</tt> <tt class="py-keyword">in</tt> <tt class="py-name">compound_loc</tt><tt class="py-op">.</tt><tt id="link-209" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-209', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">","</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L363"></a><tt class="py-lineno"> 363</tt>  <tt class="py-line">            <tt class="py-keyword">yield</tt> <tt class="py-name">part</tt> </tt>
</div><a name="L364"></a><tt class="py-lineno"> 364</tt>  <tt class="py-line"> </tt>
<a name="L365"></a><tt class="py-lineno"> 365</tt>  <tt class="py-line"> </tt>
<a name="Iterator"></a><div id="Iterator-def"><a name="L366"></a><tt class="py-lineno"> 366</tt> <a class="py-toggle" href="#" id="Iterator-toggle" onclick="return toggle('Iterator');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.GenBank.Iterator-class.html">Iterator</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Iterator-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Iterator-expanded"><a name="L367"></a><tt class="py-lineno"> 367</tt>  <tt class="py-line">    <tt class="py-docstring">"""Iterator interface to move over a file of GenBank entries one at a time (OBSOLETE).</tt> </tt>
<a name="L368"></a><tt class="py-lineno"> 368</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L369"></a><tt class="py-lineno"> 369</tt>  <tt class="py-line"><tt class="py-docstring">    This class is likely to be deprecated in a future release of Biopython.</tt> </tt>
<a name="L370"></a><tt class="py-lineno"> 370</tt>  <tt class="py-line"><tt class="py-docstring">    Please use Bio.SeqIO.parse(..., format="gb") or Bio.GenBank.parse(...)</tt> </tt>
<a name="L371"></a><tt class="py-lineno"> 371</tt>  <tt class="py-line"><tt class="py-docstring">    for SeqRecord and GenBank specific Record objects respectively instead.</tt> </tt>
<a name="L372"></a><tt class="py-lineno"> 372</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="Iterator.__init__"></a><div id="Iterator.__init__-def"><a name="L373"></a><tt class="py-lineno"> 373</tt> <a class="py-toggle" href="#" id="Iterator.__init__-toggle" onclick="return toggle('Iterator.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Iterator-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">parser</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Iterator.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Iterator.__init__-expanded"><a name="L374"></a><tt class="py-lineno"> 374</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the iterator.</tt> </tt>
<a name="L375"></a><tt class="py-lineno"> 375</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L376"></a><tt class="py-lineno"> 376</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L377"></a><tt class="py-lineno"> 377</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L378"></a><tt class="py-lineno"> 378</tt>  <tt class="py-line"><tt class="py-docstring">            - handle - A handle with GenBank entries to iterate through.</tt> </tt>
<a name="L379"></a><tt class="py-lineno"> 379</tt>  <tt class="py-line"><tt class="py-docstring">            - parser - An optional parser to pass the entries through before</tt> </tt>
<a name="L380"></a><tt class="py-lineno"> 380</tt>  <tt class="py-line"><tt class="py-docstring">              returning them. If None, then the raw entry will be returned.</tt> </tt>
<a name="L381"></a><tt class="py-lineno"> 381</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L382"></a><tt class="py-lineno"> 382</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt> </tt>
<a name="L383"></a><tt class="py-lineno"> 383</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-210" class="py-name" targets="Variable Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer._parser=Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer-class.html#_parser"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer._parser" class="py-name" href="#" onclick="return doclink('link-210', '_parser', 'link-210');">_parser</a></tt> <tt class="py-op">=</tt> <tt class="py-name">parser</tt> </tt>
</div><a name="L384"></a><tt class="py-lineno"> 384</tt>  <tt class="py-line"> </tt>
<a name="Iterator.__next__"></a><div id="Iterator.__next__-def"><a name="L385"></a><tt class="py-lineno"> 385</tt> <a class="py-toggle" href="#" id="Iterator.__next__-toggle" onclick="return toggle('Iterator.__next__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Iterator-class.html#__next__">__next__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Iterator.__next__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Iterator.__next__-expanded"><a name="L386"></a><tt class="py-lineno"> 386</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the next GenBank record from the handle.</tt> </tt>
<a name="L387"></a><tt class="py-lineno"> 387</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L388"></a><tt class="py-lineno"> 388</tt>  <tt class="py-line"><tt class="py-docstring">        Will return None if we ran out of records.</tt> </tt>
<a name="L389"></a><tt class="py-lineno"> 389</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L390"></a><tt class="py-lineno"> 390</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-211" class="py-name"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer._parser" class="py-name" href="#" onclick="return doclink('link-211', '_parser', 'link-210');">_parser</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L391"></a><tt class="py-lineno"> 391</tt>  <tt class="py-line">            <tt class="py-name">lines</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L392"></a><tt class="py-lineno"> 392</tt>  <tt class="py-line">            <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L393"></a><tt class="py-lineno"> 393</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-212" class="py-name" targets="Method Bio.File.UndoHandle.readline()=Bio.File.UndoHandle-class.html#readline,Method Bio.bgzf.BgzfReader.readline()=Bio.bgzf.BgzfReader-class.html#readline"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-212', 'readline', 'link-212');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L394"></a><tt class="py-lineno"> 394</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L395"></a><tt class="py-lineno"> 395</tt>  <tt class="py-line">                    <tt class="py-keyword">return</tt> <tt class="py-name">None</tt>  <tt class="py-comment"># Premature end of file?</tt> </tt>
<a name="L396"></a><tt class="py-lineno"> 396</tt>  <tt class="py-line">                <tt class="py-name">lines</tt><tt class="py-op">.</tt><tt id="link-213" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-213', 'append', 'link-213');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L397"></a><tt class="py-lineno"> 397</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-214" class="py-name" targets="Method Bio.Seq.Seq.rstrip()=Bio.Seq.Seq-class.html#rstrip"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-214', 'rstrip', 'link-214');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-string">"//"</tt><tt class="py-op">:</tt> </tt>
<a name="L398"></a><tt class="py-lineno"> 398</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L399"></a><tt class="py-lineno"> 399</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">lines</tt><tt class="py-op">)</tt> </tt>
<a name="L400"></a><tt class="py-lineno"> 400</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L401"></a><tt class="py-lineno"> 401</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-215" class="py-name"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer._parser" class="py-name" href="#" onclick="return doclink('link-215', '_parser', 'link-210');">_parser</a></tt><tt class="py-op">.</tt><tt id="link-216" class="py-name" targets="Function Bio.AlignIO.parse()=Bio.AlignIO-module.html#parse,Method Bio.Blast.NCBIStandalone.BlastErrorParser.parse()=Bio.Blast.NCBIStandalone.BlastErrorParser-class.html#parse,Method Bio.Blast.NCBIStandalone.BlastParser.parse()=Bio.Blast.NCBIStandalone.BlastParser-class.html#parse,Method Bio.Blast.NCBIStandalone.PSIBlastParser.parse()=Bio.Blast.NCBIStandalone.PSIBlastParser-class.html#parse,Function Bio.Blast.NCBIXML.parse()=Bio.Blast.NCBIXML-module.html#parse,Function Bio.Compass.parse()=Bio.Compass-module.html#parse,Function Bio.Emboss.Primer3.parse()=Bio.Emboss.Primer3-module.html#parse,Method Bio.Entrez.Parser.DataHandler.parse()=Bio.Entrez.Parser.DataHandler-class.html#parse,Function Bio.Entrez.parse()=Bio.Entrez-module.html#parse,Function Bio.ExPASy.Enzyme.parse()=Bio.ExPASy.Enzyme-module.html#parse,Function Bio.ExPASy.Prodoc.parse()=Bio.ExPASy.Prodoc-module.html#parse,Function Bio.ExPASy.Prosite.parse()=Bio.ExPASy.Prosite-module.html#parse,Method Bio.GenBank.FeatureParser.parse()=Bio.GenBank.FeatureParser-class.html#parse,Method Bio.GenBank.RecordParser.parse()=Bio.GenBank.RecordParser-class.html#parse,Method Bio.GenBank.Scanner.InsdcScanner.parse()=Bio.GenBank.Scanner.InsdcScanner-class.html#parse,Function Bio.Geo.parse()=Bio.Geo-module.html#parse,Function Bio.KEGG.Compound.parse()=Bio.KEGG.Compound-module.html#parse,Function Bio.KEGG.Enzyme.parse()=Bio.KEGG.Enzyme-module.html#parse,Method Bio.KEGG.KGML.KGML_parser.KGMLParser.parse()=Bio.KEGG.KGML.KGML_parser.KGMLParser-class.html#parse,Function Bio.KEGG.KGML.KGML_parser.parse()=Bio.KEGG.KGML.KGML_parser-module.html#parse,Function Bio.KEGG.Map.parse()=Bio.KEGG.Map-module.html#parse,Function Bio.Medline.parse()=Bio.Medline-module.html#parse,Function Bio.Motif.parse()=Bio.Motif-module.html#parse,Method Bio.ParserSupport.AbstractParser.parse()=Bio.ParserSupport.AbstractParser-class.html#parse,Method Bio.Phylo.CDAOIO.Parser.parse()=Bio.Phylo.CDAOIO.Parser-class.html#parse,Function Bio.Phylo.CDAOIO.parse()=Bio.Phylo.CDAOIO-module.html#parse,Method Bio.Phylo.NeXMLIO.Parser.parse()=Bio.Phylo.NeXMLIO.Parser-class.html#parse,Function Bio.Phylo.NeXMLIO.parse()=Bio.Phylo.NeXMLIO-module.html#parse,Method Bio.Phylo.NewickIO.Parser.parse()=Bio.Phylo.NewickIO.Parser-class.html#parse,Function Bio.Phylo.NewickIO.parse()=Bio.Phylo.NewickIO-module.html#parse,Function Bio.Phylo.NexusIO.parse()=Bio.Phylo.NexusIO-module.html#parse,Method Bio.Phylo.PhyloXMLIO.Parser.parse()=Bio.Phylo.PhyloXMLIO.Parser-class.html#parse,Function Bio.Phylo.PhyloXMLIO.parse()=Bio.Phylo.PhyloXMLIO-module.html#parse,Function Bio.Phylo._io.parse()=Bio.Phylo._io-module.html#parse,Function Bio.SCOP.Cla.parse()=Bio.SCOP.Cla-module.html#parse,Function Bio.SCOP.Des.parse()=Bio.SCOP.Des-module.html#parse,Function Bio.SCOP.Dom.parse()=Bio.SCOP.Dom-module.html#parse,Function Bio.SCOP.Hie.parse()=Bio.SCOP.Hie-module.html#parse,Function Bio.SCOP.Raf.parse()=Bio.SCOP.Raf-module.html#parse,Function Bio.SearchIO.parse()=Bio.SearchIO-module.html#parse,Method Bio.SeqIO.UniprotIO.Parser.parse()=Bio.SeqIO.UniprotIO.Parser-class.html#parse,Function Bio.SeqIO.parse()=Bio.SeqIO-module.html#parse,Function Bio.Sequencing.Ace.parse()=Bio.Sequencing.Ace-module.html#parse,Function Bio.Sequencing.Phd.parse()=Bio.Sequencing.Phd-module.html#parse,Function Bio.SwissProt.KeyWList.parse()=Bio.SwissProt.KeyWList-module.html#parse,Function Bio.SwissProt.parse()=Bio.SwissProt-module.html#parse,Function Bio.UniGene.parse()=Bio.UniGene-module.html#parse,Function Bio.Wise.psw.parse()=Bio.Wise.psw-module.html#parse,Function Bio.motifs.parse()=Bio.motifs-module.html#parse"><a title="Bio.AlignIO.parse
Bio.Blast.NCBIStandalone.BlastErrorParser.parse
Bio.Blast.NCBIStandalone.BlastParser.parse
Bio.Blast.NCBIStandalone.PSIBlastParser.parse
Bio.Blast.NCBIXML.parse
Bio.Compass.parse
Bio.Emboss.Primer3.parse
Bio.Entrez.Parser.DataHandler.parse
Bio.Entrez.parse
Bio.ExPASy.Enzyme.parse
Bio.ExPASy.Prodoc.parse
Bio.ExPASy.Prosite.parse
Bio.GenBank.FeatureParser.parse
Bio.GenBank.RecordParser.parse
Bio.GenBank.Scanner.InsdcScanner.parse
Bio.Geo.parse
Bio.KEGG.Compound.parse
Bio.KEGG.Enzyme.parse
Bio.KEGG.KGML.KGML_parser.KGMLParser.parse
Bio.KEGG.KGML.KGML_parser.parse
Bio.KEGG.Map.parse
Bio.Medline.parse
Bio.Motif.parse
Bio.ParserSupport.AbstractParser.parse
Bio.Phylo.CDAOIO.Parser.parse
Bio.Phylo.CDAOIO.parse
Bio.Phylo.NeXMLIO.Parser.parse
Bio.Phylo.NeXMLIO.parse
Bio.Phylo.NewickIO.Parser.parse
Bio.Phylo.NewickIO.parse
Bio.Phylo.NexusIO.parse
Bio.Phylo.PhyloXMLIO.Parser.parse
Bio.Phylo.PhyloXMLIO.parse
Bio.Phylo._io.parse
Bio.SCOP.Cla.parse
Bio.SCOP.Des.parse
Bio.SCOP.Dom.parse
Bio.SCOP.Hie.parse
Bio.SCOP.Raf.parse
Bio.SearchIO.parse
Bio.SeqIO.UniprotIO.Parser.parse
Bio.SeqIO.parse
Bio.Sequencing.Ace.parse
Bio.Sequencing.Phd.parse
Bio.SwissProt.KeyWList.parse
Bio.SwissProt.parse
Bio.UniGene.parse
Bio.Wise.psw.parse
Bio.motifs.parse" class="py-name" href="#" onclick="return doclink('link-216', 'parse', 'link-216');">parse</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L402"></a><tt class="py-lineno"> 402</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L403"></a><tt class="py-lineno"> 403</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L404"></a><tt class="py-lineno"> 404</tt>  <tt class="py-line"> </tt>
<a name="L405"></a><tt class="py-lineno"> 405</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">version_info</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">3</tt><tt class="py-op">:</tt> </tt>
<a name="Iterator.next"></a><div id="Iterator.next-def"><a name="L406"></a><tt class="py-lineno"> 406</tt> <a class="py-toggle" href="#" id="Iterator.next-toggle" onclick="return toggle('Iterator.next');">-</a><tt class="py-line">        <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Iterator-class.html#next">next</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Iterator.next-collapsed" style="display:none;" pad="++++" indent="++++++++++++++++"></div><div id="Iterator.next-expanded"><a name="L407"></a><tt class="py-lineno"> 407</tt>  <tt class="py-line">            <tt class="py-docstring">"""Python 2 style alias for Python 3 style __next__ method."""</tt> </tt>
<a name="L408"></a><tt class="py-lineno"> 408</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-217" class="py-name" targets="Method Bio.AlignIO.ClustalIO.ClustalIterator.__next__()=Bio.AlignIO.ClustalIO.ClustalIterator-class.html#__next__,Method Bio.AlignIO.EmbossIO.EmbossIterator.__next__()=Bio.AlignIO.EmbossIO.EmbossIterator-class.html#__next__,Method Bio.AlignIO.Interfaces.AlignmentIterator.__next__()=Bio.AlignIO.Interfaces.AlignmentIterator-class.html#__next__,Method Bio.AlignIO.PhylipIO.PhylipIterator.__next__()=Bio.AlignIO.PhylipIO.PhylipIterator-class.html#__next__,Method Bio.AlignIO.PhylipIO.SequentialPhylipIterator.__next__()=Bio.AlignIO.PhylipIO.SequentialPhylipIterator-class.html#__next__,Method Bio.AlignIO.StockholmIO.StockholmIterator.__next__()=Bio.AlignIO.StockholmIO.StockholmIterator-class.html#__next__,Method Bio.Blast.NCBIStandalone.Iterator.__next__()=Bio.Blast.NCBIStandalone.Iterator-class.html#__next__,Method Bio.Blast.ParseBlastTable.BlastTableReader.__next__()=Bio.Blast.ParseBlastTable.BlastTableReader-class.html#__next__,Method Bio.DocSQL.IterationCursor.__next__()=Bio.DocSQL.IterationCursor-class.html#__next__,Method Bio.File.UndoHandle.__next__()=Bio.File.UndoHandle-class.html#__next__,Method Bio.GenBank.Iterator.__next__()=Bio.GenBank.Iterator-class.html#__next__,Method Bio.Nexus.Nexus.CharBuffer.__next__()=Bio.Nexus.Nexus.CharBuffer-class.html#__next__,Method Bio.PopGen.GenePop.Controller._FileIterator.__next__()=Bio.PopGen.GenePop.Controller._FileIterator-class.html#__next__,Method Bio.SeqIO.Interfaces.SequenceIterator.__next__()=Bio.SeqIO.Interfaces.SequenceIterator-class.html#__next__,Method Bio.bgzf.BgzfReader.__next__()=Bio.bgzf.BgzfReader-class.html#__next__"><a title="Bio.AlignIO.ClustalIO.ClustalIterator.__next__
Bio.AlignIO.EmbossIO.EmbossIterator.__next__
Bio.AlignIO.Interfaces.AlignmentIterator.__next__
Bio.AlignIO.PhylipIO.PhylipIterator.__next__
Bio.AlignIO.PhylipIO.SequentialPhylipIterator.__next__
Bio.AlignIO.StockholmIO.StockholmIterator.__next__
Bio.Blast.NCBIStandalone.Iterator.__next__
Bio.Blast.ParseBlastTable.BlastTableReader.__next__
Bio.DocSQL.IterationCursor.__next__
Bio.File.UndoHandle.__next__
Bio.GenBank.Iterator.__next__
Bio.Nexus.Nexus.CharBuffer.__next__
Bio.PopGen.GenePop.Controller._FileIterator.__next__
Bio.SeqIO.Interfaces.SequenceIterator.__next__
Bio.bgzf.BgzfReader.__next__" class="py-name" href="#" onclick="return doclink('link-217', '__next__', 'link-217');">__next__</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L409"></a><tt class="py-lineno"> 409</tt>  <tt class="py-line"> </tt>
<a name="Iterator.__iter__"></a><div id="Iterator.__iter__-def"><a name="L410"></a><tt class="py-lineno"> 410</tt> <a class="py-toggle" href="#" id="Iterator.__iter__-toggle" onclick="return toggle('Iterator.__iter__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Iterator-class.html#__iter__">__iter__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Iterator.__iter__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Iterator.__iter__-expanded"><a name="L411"></a><tt class="py-lineno"> 411</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">iter</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-218" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator.__next__
Bio.AlignIO.EmbossIO.EmbossIterator.__next__
Bio.AlignIO.Interfaces.AlignmentIterator.__next__
Bio.AlignIO.PhylipIO.PhylipIterator.__next__
Bio.AlignIO.PhylipIO.SequentialPhylipIterator.__next__
Bio.AlignIO.StockholmIO.StockholmIterator.__next__
Bio.Blast.NCBIStandalone.Iterator.__next__
Bio.Blast.ParseBlastTable.BlastTableReader.__next__
Bio.DocSQL.IterationCursor.__next__
Bio.File.UndoHandle.__next__
Bio.GenBank.Iterator.__next__
Bio.Nexus.Nexus.CharBuffer.__next__
Bio.PopGen.GenePop.Controller._FileIterator.__next__
Bio.SeqIO.Interfaces.SequenceIterator.__next__
Bio.bgzf.BgzfReader.__next__" class="py-name" href="#" onclick="return doclink('link-218', '__next__', 'link-217');">__next__</a></tt><tt class="py-op">,</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L412"></a><tt class="py-lineno"> 412</tt>  <tt class="py-line"> </tt>
<a name="L413"></a><tt class="py-lineno"> 413</tt>  <tt class="py-line"> </tt>
<a name="ParserFailureError"></a><div id="ParserFailureError-def"><a name="L414"></a><tt class="py-lineno"> 414</tt> <a class="py-toggle" href="#" id="ParserFailureError-toggle" onclick="return toggle('ParserFailureError');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.GenBank.ParserFailureError-class.html">ParserFailureError</a><tt class="py-op">(</tt><tt class="py-base-class">Exception</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ParserFailureError-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="ParserFailureError-expanded"><a name="L415"></a><tt class="py-lineno"> 415</tt>  <tt class="py-line">    <tt class="py-docstring">"""Failure caused by some kind of problem in the parser.</tt> </tt>
<a name="L416"></a><tt class="py-lineno"> 416</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L417"></a><tt class="py-lineno"> 417</tt>  <tt class="py-line">    <tt class="py-keyword">pass</tt> </tt>
</div><a name="L418"></a><tt class="py-lineno"> 418</tt>  <tt class="py-line"> </tt>
<a name="L419"></a><tt class="py-lineno"> 419</tt>  <tt class="py-line"> </tt>
<a name="LocationParserError"></a><div id="LocationParserError-def"><a name="L420"></a><tt class="py-lineno"> 420</tt> <a class="py-toggle" href="#" id="LocationParserError-toggle" onclick="return toggle('LocationParserError');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.GenBank.LocationParserError-class.html">LocationParserError</a><tt class="py-op">(</tt><tt class="py-base-class">Exception</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="LocationParserError-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="LocationParserError-expanded"><a name="L421"></a><tt class="py-lineno"> 421</tt>  <tt class="py-line">    <tt class="py-docstring">"""Could not Properly parse out a location from a GenBank file.</tt> </tt>
<a name="L422"></a><tt class="py-lineno"> 422</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L423"></a><tt class="py-lineno"> 423</tt>  <tt class="py-line">    <tt class="py-keyword">pass</tt> </tt>
</div><a name="L424"></a><tt class="py-lineno"> 424</tt>  <tt class="py-line"> </tt>
<a name="L425"></a><tt class="py-lineno"> 425</tt>  <tt class="py-line"> </tt>
<a name="FeatureParser"></a><div id="FeatureParser-def"><a name="L426"></a><tt class="py-lineno"> 426</tt> <a class="py-toggle" href="#" id="FeatureParser-toggle" onclick="return toggle('FeatureParser');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.GenBank.FeatureParser-class.html">FeatureParser</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FeatureParser-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="FeatureParser-expanded"><a name="L427"></a><tt class="py-lineno"> 427</tt>  <tt class="py-line">    <tt class="py-docstring">"""Parse GenBank files into Seq + Feature objects (OBSOLETE).</tt> </tt>
<a name="L428"></a><tt class="py-lineno"> 428</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L429"></a><tt class="py-lineno"> 429</tt>  <tt class="py-line"><tt class="py-docstring">    Direct use of this class is discouraged, and may be deprecated in</tt> </tt>
<a name="L430"></a><tt class="py-lineno"> 430</tt>  <tt class="py-line"><tt class="py-docstring">    a future release of Biopython.</tt> </tt>
<a name="L431"></a><tt class="py-lineno"> 431</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L432"></a><tt class="py-lineno"> 432</tt>  <tt class="py-line"><tt class="py-docstring">    Please use Bio.SeqIO.parse(...) or Bio.SeqIO.read(...) instead.</tt> </tt>
<a name="L433"></a><tt class="py-lineno"> 433</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="FeatureParser.__init__"></a><div id="FeatureParser.__init__-def"><a name="L434"></a><tt class="py-lineno"> 434</tt> <a class="py-toggle" href="#" id="FeatureParser.__init__-toggle" onclick="return toggle('FeatureParser.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.FeatureParser-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">debug_level</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-param">use_fuzziness</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> </tt>
<a name="L435"></a><tt class="py-lineno"> 435</tt>  <tt class="py-line">                 <tt class="py-param">feature_cleaner</tt><tt class="py-op">=</tt><tt id="link-219" class="py-name"><a title="Bio.GenBank.utils.FeatureValueCleaner" class="py-name" href="#" onclick="return doclink('link-219', 'FeatureValueCleaner', 'link-3');">FeatureValueCleaner</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FeatureParser.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="FeatureParser.__init__-expanded"><a name="L436"></a><tt class="py-lineno"> 436</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize a GenBank parser and Feature consumer.</tt> </tt>
<a name="L437"></a><tt class="py-lineno"> 437</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L438"></a><tt class="py-lineno"> 438</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L439"></a><tt class="py-lineno"> 439</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L440"></a><tt class="py-lineno"> 440</tt>  <tt class="py-line"><tt class="py-docstring">            - debug_level - An optional argument that species the amount of</tt> </tt>
<a name="L441"></a><tt class="py-lineno"> 441</tt>  <tt class="py-line"><tt class="py-docstring">              debugging information the parser should spit out. By default we have</tt> </tt>
<a name="L442"></a><tt class="py-lineno"> 442</tt>  <tt class="py-line"><tt class="py-docstring">              no debugging info (the fastest way to do things), but if you want</tt> </tt>
<a name="L443"></a><tt class="py-lineno"> 443</tt>  <tt class="py-line"><tt class="py-docstring">              you can set this as high as two and see exactly where a parse fails.</tt> </tt>
<a name="L444"></a><tt class="py-lineno"> 444</tt>  <tt class="py-line"><tt class="py-docstring">            - use_fuzziness - Specify whether or not to use fuzzy representations.</tt> </tt>
<a name="L445"></a><tt class="py-lineno"> 445</tt>  <tt class="py-line"><tt class="py-docstring">              The default is 1 (use fuzziness).</tt> </tt>
<a name="L446"></a><tt class="py-lineno"> 446</tt>  <tt class="py-line"><tt class="py-docstring">            - feature_cleaner - A class which will be used to clean out the</tt> </tt>
<a name="L447"></a><tt class="py-lineno"> 447</tt>  <tt class="py-line"><tt class="py-docstring">              values of features. This class must implement the function</tt> </tt>
<a name="L448"></a><tt class="py-lineno"> 448</tt>  <tt class="py-line"><tt class="py-docstring">              clean_value. GenBank.utils has a "standard" cleaner class, which</tt> </tt>
<a name="L449"></a><tt class="py-lineno"> 449</tt>  <tt class="py-line"><tt class="py-docstring">              is used by default.</tt> </tt>
<a name="L450"></a><tt class="py-lineno"> 450</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L451"></a><tt class="py-lineno"> 451</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_scanner</tt> <tt class="py-op">=</tt> <tt id="link-220" class="py-name"><a title="Bio.GenBank.Scanner.GenBankScanner" class="py-name" href="#" onclick="return doclink('link-220', 'GenBankScanner', 'link-5');">GenBankScanner</a></tt><tt class="py-op">(</tt><tt class="py-name">debug_level</tt><tt class="py-op">)</tt> </tt>
<a name="L452"></a><tt class="py-lineno"> 452</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">use_fuzziness</tt> <tt class="py-op">=</tt> <tt class="py-name">use_fuzziness</tt> </tt>
<a name="L453"></a><tt class="py-lineno"> 453</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cleaner</tt> <tt class="py-op">=</tt> <tt class="py-name">feature_cleaner</tt> </tt>
</div><a name="L454"></a><tt class="py-lineno"> 454</tt>  <tt class="py-line"> </tt>
<a name="FeatureParser.parse"></a><div id="FeatureParser.parse-def"><a name="L455"></a><tt class="py-lineno"> 455</tt> <a class="py-toggle" href="#" id="FeatureParser.parse-toggle" onclick="return toggle('FeatureParser.parse');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.FeatureParser-class.html#parse">parse</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FeatureParser.parse-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="FeatureParser.parse-expanded"><a name="L456"></a><tt class="py-lineno"> 456</tt>  <tt class="py-line">        <tt class="py-docstring">"""Parse the specified handle.</tt> </tt>
<a name="L457"></a><tt class="py-lineno"> 457</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L458"></a><tt class="py-lineno"> 458</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_consumer</tt> <tt class="py-op">=</tt> <tt id="link-221" class="py-name" targets="Class Bio.GenBank._FeatureConsumer=Bio.GenBank._FeatureConsumer-class.html"><a title="Bio.GenBank._FeatureConsumer" class="py-name" href="#" onclick="return doclink('link-221', '_FeatureConsumer', 'link-221');">_FeatureConsumer</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">use_fuzziness</tt><tt class="py-op">,</tt> </tt>
<a name="L459"></a><tt class="py-lineno"> 459</tt>  <tt class="py-line">                                          <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cleaner</tt><tt class="py-op">)</tt> </tt>
<a name="L460"></a><tt class="py-lineno"> 460</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_scanner</tt><tt class="py-op">.</tt><tt id="link-222" class="py-name" targets="Method Bio.Blast.NCBIStandalone._Scanner.feed()=Bio.Blast.NCBIStandalone._Scanner-class.html#feed,Method Bio.ExPASy.ScanProsite.Parser.feed()=Bio.ExPASy.ScanProsite.Parser-class.html#feed,Method Bio.GenBank.Scanner.InsdcScanner.feed()=Bio.GenBank.Scanner.InsdcScanner-class.html#feed"><a title="Bio.Blast.NCBIStandalone._Scanner.feed
Bio.ExPASy.ScanProsite.Parser.feed
Bio.GenBank.Scanner.InsdcScanner.feed" class="py-name" href="#" onclick="return doclink('link-222', 'feed', 'link-222');">feed</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_consumer</tt><tt class="py-op">)</tt> </tt>
<a name="L461"></a><tt class="py-lineno"> 461</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_consumer</tt><tt class="py-op">.</tt><tt id="link-223" class="py-name" targets="Variable BioSQL.BioSeq.DBSeq.data=BioSQL.BioSeq.DBSeq-class.html#data"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-223', 'data', 'link-223');">data</a></tt> </tt>
</div></div><a name="L462"></a><tt class="py-lineno"> 462</tt>  <tt class="py-line"> </tt>
<a name="L463"></a><tt class="py-lineno"> 463</tt>  <tt class="py-line"> </tt>
<a name="RecordParser"></a><div id="RecordParser-def"><a name="L464"></a><tt class="py-lineno"> 464</tt> <a class="py-toggle" href="#" id="RecordParser-toggle" onclick="return toggle('RecordParser');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.GenBank.RecordParser-class.html">RecordParser</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RecordParser-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="RecordParser-expanded"><a name="L465"></a><tt class="py-lineno"> 465</tt>  <tt class="py-line">    <tt class="py-docstring">"""Parse GenBank files into Record objects (OBSOLETE).</tt> </tt>
<a name="L466"></a><tt class="py-lineno"> 466</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L467"></a><tt class="py-lineno"> 467</tt>  <tt class="py-line"><tt class="py-docstring">    Direct use of this class is discouraged, and may be deprecated in</tt> </tt>
<a name="L468"></a><tt class="py-lineno"> 468</tt>  <tt class="py-line"><tt class="py-docstring">    a future release of Biopython.</tt> </tt>
<a name="L469"></a><tt class="py-lineno"> 469</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L470"></a><tt class="py-lineno"> 470</tt>  <tt class="py-line"><tt class="py-docstring">    Please use the Bio.GenBank.parse(...) or Bio.GenBank.read(...) functions</tt> </tt>
<a name="L471"></a><tt class="py-lineno"> 471</tt>  <tt class="py-line"><tt class="py-docstring">    instead.</tt> </tt>
<a name="L472"></a><tt class="py-lineno"> 472</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="RecordParser.__init__"></a><div id="RecordParser.__init__-def"><a name="L473"></a><tt class="py-lineno"> 473</tt> <a class="py-toggle" href="#" id="RecordParser.__init__-toggle" onclick="return toggle('RecordParser.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.RecordParser-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">debug_level</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RecordParser.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="RecordParser.__init__-expanded"><a name="L474"></a><tt class="py-lineno"> 474</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize the parser.</tt> </tt>
<a name="L475"></a><tt class="py-lineno"> 475</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L476"></a><tt class="py-lineno"> 476</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L477"></a><tt class="py-lineno"> 477</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L478"></a><tt class="py-lineno"> 478</tt>  <tt class="py-line"><tt class="py-docstring">            - debug_level - An optional argument that species the amount of</tt> </tt>
<a name="L479"></a><tt class="py-lineno"> 479</tt>  <tt class="py-line"><tt class="py-docstring">              debugging information the parser should spit out. By default we have</tt> </tt>
<a name="L480"></a><tt class="py-lineno"> 480</tt>  <tt class="py-line"><tt class="py-docstring">              no debugging info (the fastest way to do things), but if you want</tt> </tt>
<a name="L481"></a><tt class="py-lineno"> 481</tt>  <tt class="py-line"><tt class="py-docstring">              you can set this as high as two and see exactly where a parse fails.</tt> </tt>
<a name="L482"></a><tt class="py-lineno"> 482</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L483"></a><tt class="py-lineno"> 483</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_scanner</tt> <tt class="py-op">=</tt> <tt id="link-224" class="py-name"><a title="Bio.GenBank.Scanner.GenBankScanner" class="py-name" href="#" onclick="return doclink('link-224', 'GenBankScanner', 'link-5');">GenBankScanner</a></tt><tt class="py-op">(</tt><tt class="py-name">debug_level</tt><tt class="py-op">)</tt> </tt>
</div><a name="L484"></a><tt class="py-lineno"> 484</tt>  <tt class="py-line"> </tt>
<a name="RecordParser.parse"></a><div id="RecordParser.parse-def"><a name="L485"></a><tt class="py-lineno"> 485</tt> <a class="py-toggle" href="#" id="RecordParser.parse-toggle" onclick="return toggle('RecordParser.parse');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.RecordParser-class.html#parse">parse</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="RecordParser.parse-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="RecordParser.parse-expanded"><a name="L486"></a><tt class="py-lineno"> 486</tt>  <tt class="py-line">        <tt class="py-docstring">"""Parse the specified handle into a GenBank record.</tt> </tt>
<a name="L487"></a><tt class="py-lineno"> 487</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L488"></a><tt class="py-lineno"> 488</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_consumer</tt> <tt class="py-op">=</tt> <tt id="link-225" class="py-name" targets="Class Bio.GenBank._RecordConsumer=Bio.GenBank._RecordConsumer-class.html"><a title="Bio.GenBank._RecordConsumer" class="py-name" href="#" onclick="return doclink('link-225', '_RecordConsumer', 'link-225');">_RecordConsumer</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L489"></a><tt class="py-lineno"> 489</tt>  <tt class="py-line"> </tt>
<a name="L490"></a><tt class="py-lineno"> 490</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_scanner</tt><tt class="py-op">.</tt><tt id="link-226" class="py-name"><a title="Bio.Blast.NCBIStandalone._Scanner.feed
Bio.ExPASy.ScanProsite.Parser.feed
Bio.GenBank.Scanner.InsdcScanner.feed" class="py-name" href="#" onclick="return doclink('link-226', 'feed', 'link-222');">feed</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_consumer</tt><tt class="py-op">)</tt> </tt>
<a name="L491"></a><tt class="py-lineno"> 491</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_consumer</tt><tt class="py-op">.</tt><tt id="link-227" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-227', 'data', 'link-223');">data</a></tt> </tt>
</div></div><a name="L492"></a><tt class="py-lineno"> 492</tt>  <tt class="py-line"> </tt>
<a name="L493"></a><tt class="py-lineno"> 493</tt>  <tt class="py-line"> </tt>
<a name="_BaseGenBankConsumer"></a><div id="_BaseGenBankConsumer-def"><a name="L494"></a><tt class="py-lineno"> 494</tt> <a class="py-toggle" href="#" id="_BaseGenBankConsumer-toggle" onclick="return toggle('_BaseGenBankConsumer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.GenBank._BaseGenBankConsumer-class.html">_BaseGenBankConsumer</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BaseGenBankConsumer-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_BaseGenBankConsumer-expanded"><a name="L495"></a><tt class="py-lineno"> 495</tt>  <tt class="py-line">    <tt class="py-docstring">"""Abstract GenBank consumer providing useful general functions (PRIVATE).</tt> </tt>
<a name="L496"></a><tt class="py-lineno"> 496</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L497"></a><tt class="py-lineno"> 497</tt>  <tt class="py-line"><tt class="py-docstring">    This just helps to eliminate some duplication in things that most</tt> </tt>
<a name="L498"></a><tt class="py-lineno"> 498</tt>  <tt class="py-line"><tt class="py-docstring">    GenBank consumers want to do.</tt> </tt>
<a name="L499"></a><tt class="py-lineno"> 499</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L500"></a><tt class="py-lineno"> 500</tt>  <tt class="py-line">    <tt class="py-comment"># Special keys in GenBank records that we should remove spaces from</tt> </tt>
<a name="L501"></a><tt class="py-lineno"> 501</tt>  <tt class="py-line">    <tt class="py-comment"># For instance, \translation keys have values which are proteins and</tt> </tt>
<a name="L502"></a><tt class="py-lineno"> 502</tt>  <tt class="py-line">    <tt class="py-comment"># should have spaces and newlines removed from them. This class</tt> </tt>
<a name="L503"></a><tt class="py-lineno"> 503</tt>  <tt class="py-line">    <tt class="py-comment"># attribute gives us more control over specific formatting problems.</tt> </tt>
<a name="L504"></a><tt class="py-lineno"> 504</tt>  <tt class="py-line">    <tt id="link-228" class="py-name" targets="Variable Bio.GenBank._BaseGenBankConsumer.remove_space_keys=Bio.GenBank._BaseGenBankConsumer-class.html#remove_space_keys"><a title="Bio.GenBank._BaseGenBankConsumer.remove_space_keys" class="py-name" href="#" onclick="return doclink('link-228', 'remove_space_keys', 'link-228');">remove_space_keys</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">"translation"</tt><tt class="py-op">]</tt> </tt>
<a name="L505"></a><tt class="py-lineno"> 505</tt>  <tt class="py-line"> </tt>
<a name="_BaseGenBankConsumer.__init__"></a><div id="_BaseGenBankConsumer.__init__-def"><a name="L506"></a><tt class="py-lineno"> 506</tt> <a class="py-toggle" href="#" id="_BaseGenBankConsumer.__init__-toggle" onclick="return toggle('_BaseGenBankConsumer.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._BaseGenBankConsumer-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BaseGenBankConsumer.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_BaseGenBankConsumer.__init__-expanded"><a name="L507"></a><tt class="py-lineno"> 507</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L508"></a><tt class="py-lineno"> 508</tt>  <tt class="py-line"> </tt>
<a name="_BaseGenBankConsumer._unhandled"></a><div id="_BaseGenBankConsumer._unhandled-def"><a name="L509"></a><tt class="py-lineno"> 509</tt> <a class="py-toggle" href="#" id="_BaseGenBankConsumer._unhandled-toggle" onclick="return toggle('_BaseGenBankConsumer._unhandled');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._BaseGenBankConsumer-class.html#_unhandled">_unhandled</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">data</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BaseGenBankConsumer._unhandled-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_BaseGenBankConsumer._unhandled-expanded"><a name="L510"></a><tt class="py-lineno"> 510</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L511"></a><tt class="py-lineno"> 511</tt>  <tt class="py-line"> </tt>
<a name="_BaseGenBankConsumer.__getattr__"></a><div id="_BaseGenBankConsumer.__getattr__-def"><a name="L512"></a><tt class="py-lineno"> 512</tt> <a class="py-toggle" href="#" id="_BaseGenBankConsumer.__getattr__-toggle" onclick="return toggle('_BaseGenBankConsumer.__getattr__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._BaseGenBankConsumer-class.html#__getattr__">__getattr__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">attr</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BaseGenBankConsumer.__getattr__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_BaseGenBankConsumer.__getattr__-expanded"><a name="L513"></a><tt class="py-lineno"> 513</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-229" class="py-name" targets="Method Bio.GenBank._BaseGenBankConsumer._unhandled()=Bio.GenBank._BaseGenBankConsumer-class.html#_unhandled,Method Bio.ParserSupport.AbstractConsumer._unhandled()=Bio.ParserSupport.AbstractConsumer-class.html#_unhandled"><a title="Bio.GenBank._BaseGenBankConsumer._unhandled
Bio.ParserSupport.AbstractConsumer._unhandled" class="py-name" href="#" onclick="return doclink('link-229', '_unhandled', 'link-229');">_unhandled</a></tt> </tt>
</div><a name="L514"></a><tt class="py-lineno"> 514</tt>  <tt class="py-line"> </tt>
<a name="_BaseGenBankConsumer._split_keywords"></a><div id="_BaseGenBankConsumer._split_keywords-def"><a name="L515"></a><tt class="py-lineno"> 515</tt> <a class="py-toggle" href="#" id="_BaseGenBankConsumer._split_keywords-toggle" onclick="return toggle('_BaseGenBankConsumer._split_keywords');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._BaseGenBankConsumer-class.html#_split_keywords">_split_keywords</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">keyword_string</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BaseGenBankConsumer._split_keywords-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_BaseGenBankConsumer._split_keywords-expanded"><a name="L516"></a><tt class="py-lineno"> 516</tt>  <tt class="py-line">        <tt class="py-docstring">"""Split a string of keywords into a nice clean list.</tt> </tt>
<a name="L517"></a><tt class="py-lineno"> 517</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L518"></a><tt class="py-lineno"> 518</tt>  <tt class="py-line">        <tt class="py-comment"># process the keywords into a python list</tt> </tt>
<a name="L519"></a><tt class="py-lineno"> 519</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">keyword_string</tt> <tt class="py-op">==</tt> <tt class="py-string">""</tt> <tt class="py-keyword">or</tt> <tt class="py-name">keyword_string</tt> <tt class="py-op">==</tt> <tt class="py-string">"."</tt><tt class="py-op">:</tt> </tt>
<a name="L520"></a><tt class="py-lineno"> 520</tt>  <tt class="py-line">            <tt id="link-230" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.keywords()=Bio.GenBank._FeatureConsumer-class.html#keywords,Method Bio.GenBank._RecordConsumer.keywords()=Bio.GenBank._RecordConsumer-class.html#keywords"><a title="Bio.GenBank._FeatureConsumer.keywords
Bio.GenBank._RecordConsumer.keywords" class="py-name" href="#" onclick="return doclink('link-230', 'keywords', 'link-230');">keywords</a></tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L521"></a><tt class="py-lineno"> 521</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">keyword_string</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'.'</tt><tt class="py-op">:</tt> </tt>
<a name="L522"></a><tt class="py-lineno"> 522</tt>  <tt class="py-line">            <tt id="link-231" class="py-name"><a title="Bio.GenBank._FeatureConsumer.keywords
Bio.GenBank._RecordConsumer.keywords" class="py-name" href="#" onclick="return doclink('link-231', 'keywords', 'link-230');">keywords</a></tt> <tt class="py-op">=</tt> <tt class="py-name">keyword_string</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L523"></a><tt class="py-lineno"> 523</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L524"></a><tt class="py-lineno"> 524</tt>  <tt class="py-line">            <tt id="link-232" class="py-name"><a title="Bio.GenBank._FeatureConsumer.keywords
Bio.GenBank._RecordConsumer.keywords" class="py-name" href="#" onclick="return doclink('link-232', 'keywords', 'link-230');">keywords</a></tt> <tt class="py-op">=</tt> <tt class="py-name">keyword_string</tt> </tt>
<a name="L525"></a><tt class="py-lineno"> 525</tt>  <tt class="py-line">        <tt class="py-name">keyword_list</tt> <tt class="py-op">=</tt> <tt id="link-233" class="py-name"><a title="Bio.GenBank._FeatureConsumer.keywords
Bio.GenBank._RecordConsumer.keywords" class="py-name" href="#" onclick="return doclink('link-233', 'keywords', 'link-230');">keywords</a></tt><tt class="py-op">.</tt><tt id="link-234" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-234', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">';'</tt><tt class="py-op">)</tt> </tt>
<a name="L526"></a><tt class="py-lineno"> 526</tt>  <tt class="py-line">        <tt class="py-name">clean_keyword_list</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-235" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.x=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#x"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-235', 'x', 'link-235');">x</a></tt><tt class="py-op">.</tt><tt id="link-236" class="py-name" targets="Method Bio.Seq.Seq.strip()=Bio.Seq.Seq-class.html#strip"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-236', 'strip', 'link-236');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-237" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-237', 'x', 'link-235');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">keyword_list</tt><tt class="py-op">]</tt> </tt>
<a name="L527"></a><tt class="py-lineno"> 527</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">clean_keyword_list</tt> </tt>
</div><a name="L528"></a><tt class="py-lineno"> 528</tt>  <tt class="py-line"> </tt>
<a name="_BaseGenBankConsumer._split_accessions"></a><div id="_BaseGenBankConsumer._split_accessions-def"><a name="L529"></a><tt class="py-lineno"> 529</tt> <a class="py-toggle" href="#" id="_BaseGenBankConsumer._split_accessions-toggle" onclick="return toggle('_BaseGenBankConsumer._split_accessions');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._BaseGenBankConsumer-class.html#_split_accessions">_split_accessions</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">accession_string</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BaseGenBankConsumer._split_accessions-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_BaseGenBankConsumer._split_accessions-expanded"><a name="L530"></a><tt class="py-lineno"> 530</tt>  <tt class="py-line">        <tt class="py-docstring">"""Split a string of accession numbers into a list.</tt> </tt>
<a name="L531"></a><tt class="py-lineno"> 531</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L532"></a><tt class="py-lineno"> 532</tt>  <tt class="py-line">        <tt class="py-comment"># first replace all line feeds with spaces</tt> </tt>
<a name="L533"></a><tt class="py-lineno"> 533</tt>  <tt class="py-line">        <tt class="py-comment"># Also, EMBL style accessions are split with ';'</tt> </tt>
<a name="L534"></a><tt class="py-lineno"> 534</tt>  <tt class="py-line">        <tt id="link-238" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.accession()=Bio.GenBank._FeatureConsumer-class.html#accession,Method Bio.GenBank._RecordConsumer.accession()=Bio.GenBank._RecordConsumer-class.html#accession,Method Bio.Phylo.PhyloXMLIO.Parser.accession()=Bio.Phylo.PhyloXMLIO.Parser-class.html#accession,Method Bio.Phylo.PhyloXMLIO.Writer.accession()=Bio.Phylo.PhyloXMLIO.Writer-class.html#accession"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-238', 'accession', 'link-238');">accession</a></tt> <tt class="py-op">=</tt> <tt class="py-name">accession_string</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"\n"</tt><tt class="py-op">,</tt> <tt class="py-string">" "</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">";"</tt><tt class="py-op">,</tt> <tt class="py-string">" "</tt><tt class="py-op">)</tt> </tt>
<a name="L535"></a><tt class="py-lineno"> 535</tt>  <tt class="py-line"> </tt>
<a name="L536"></a><tt class="py-lineno"> 536</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt id="link-239" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-239', 'x', 'link-235');">x</a></tt><tt class="py-op">.</tt><tt id="link-240" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-240', 'strip', 'link-236');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-241" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-241', 'x', 'link-235');">x</a></tt> <tt class="py-keyword">in</tt> <tt id="link-242" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-242', 'accession', 'link-238');">accession</a></tt><tt class="py-op">.</tt><tt id="link-243" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-243', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt id="link-244" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-244', 'x', 'link-235');">x</a></tt><tt class="py-op">.</tt><tt id="link-245" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-245', 'strip', 'link-236');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
</div><a name="L537"></a><tt class="py-lineno"> 537</tt>  <tt class="py-line"> </tt>
<a name="_BaseGenBankConsumer._split_taxonomy"></a><div id="_BaseGenBankConsumer._split_taxonomy-def"><a name="L538"></a><tt class="py-lineno"> 538</tt> <a class="py-toggle" href="#" id="_BaseGenBankConsumer._split_taxonomy-toggle" onclick="return toggle('_BaseGenBankConsumer._split_taxonomy');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._BaseGenBankConsumer-class.html#_split_taxonomy">_split_taxonomy</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">taxonomy_string</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BaseGenBankConsumer._split_taxonomy-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_BaseGenBankConsumer._split_taxonomy-expanded"><a name="L539"></a><tt class="py-lineno"> 539</tt>  <tt class="py-line">        <tt class="py-docstring">"""Split a string with taxonomy info into a list.</tt> </tt>
<a name="L540"></a><tt class="py-lineno"> 540</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L541"></a><tt class="py-lineno"> 541</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">taxonomy_string</tt> <tt class="py-keyword">or</tt> <tt class="py-name">taxonomy_string</tt> <tt class="py-op">==</tt> <tt class="py-string">"."</tt><tt class="py-op">:</tt> </tt>
<a name="L542"></a><tt class="py-lineno"> 542</tt>  <tt class="py-line">            <tt class="py-comment"># Missing data, no taxonomy</tt> </tt>
<a name="L543"></a><tt class="py-lineno"> 543</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L544"></a><tt class="py-lineno"> 544</tt>  <tt class="py-line"> </tt>
<a name="L545"></a><tt class="py-lineno"> 545</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">taxonomy_string</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'.'</tt><tt class="py-op">:</tt> </tt>
<a name="L546"></a><tt class="py-lineno"> 546</tt>  <tt class="py-line">            <tt class="py-name">tax_info</tt> <tt class="py-op">=</tt> <tt class="py-name">taxonomy_string</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L547"></a><tt class="py-lineno"> 547</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L548"></a><tt class="py-lineno"> 548</tt>  <tt class="py-line">            <tt class="py-name">tax_info</tt> <tt class="py-op">=</tt> <tt class="py-name">taxonomy_string</tt> </tt>
<a name="L549"></a><tt class="py-lineno"> 549</tt>  <tt class="py-line">        <tt class="py-name">tax_list</tt> <tt class="py-op">=</tt> <tt class="py-name">tax_info</tt><tt class="py-op">.</tt><tt id="link-246" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-246', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">';'</tt><tt class="py-op">)</tt> </tt>
<a name="L550"></a><tt class="py-lineno"> 550</tt>  <tt class="py-line">        <tt class="py-name">new_tax_list</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L551"></a><tt class="py-lineno"> 551</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">tax_item</tt> <tt class="py-keyword">in</tt> <tt class="py-name">tax_list</tt><tt class="py-op">:</tt> </tt>
<a name="L552"></a><tt class="py-lineno"> 552</tt>  <tt class="py-line">            <tt class="py-name">new_items</tt> <tt class="py-op">=</tt> <tt class="py-name">tax_item</tt><tt class="py-op">.</tt><tt id="link-247" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-247', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">"\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L553"></a><tt class="py-lineno"> 553</tt>  <tt class="py-line">            <tt class="py-name">new_tax_list</tt><tt class="py-op">.</tt><tt id="link-248" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.extend()=Bio.Align.MultipleSeqAlignment-class.html#extend,Method Bio.SCOP.Raf.SeqMap.extend()=Bio.SCOP.Raf.SeqMap-class.html#extend,Method Bio.Seq.MutableSeq.extend()=Bio.Seq.MutableSeq-class.html#extend"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-248', 'extend', 'link-248');">extend</a></tt><tt class="py-op">(</tt><tt class="py-name">new_items</tt><tt class="py-op">)</tt> </tt>
<a name="L554"></a><tt class="py-lineno"> 554</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-string">''</tt> <tt class="py-keyword">in</tt> <tt class="py-name">new_tax_list</tt><tt class="py-op">:</tt> </tt>
<a name="L555"></a><tt class="py-lineno"> 555</tt>  <tt class="py-line">            <tt class="py-name">new_tax_list</tt><tt class="py-op">.</tt><tt id="link-249" class="py-name" targets="Method Bio.Crystal.Chain.remove()=Bio.Crystal.Chain-class.html#remove,Method Bio.Graphics.BasicChromosome._ChromosomeComponent.remove()=Bio.Graphics.BasicChromosome._ChromosomeComponent-class.html#remove,Method Bio.Restriction.Restriction.RestrictionBatch.remove()=Bio.Restriction.Restriction.RestrictionBatch-class.html#remove,Method Bio.Seq.MutableSeq.remove()=Bio.Seq.MutableSeq-class.html#remove,Method BioSQL.Loader.DatabaseRemover.remove()=BioSQL.Loader.DatabaseRemover-class.html#remove"><a title="Bio.Crystal.Chain.remove
Bio.Graphics.BasicChromosome._ChromosomeComponent.remove
Bio.Restriction.Restriction.RestrictionBatch.remove
Bio.Seq.MutableSeq.remove
BioSQL.Loader.DatabaseRemover.remove" class="py-name" href="#" onclick="return doclink('link-249', 'remove', 'link-249');">remove</a></tt><tt class="py-op">(</tt><tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L556"></a><tt class="py-lineno"> 556</tt>  <tt class="py-line">        <tt class="py-name">clean_tax_list</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-250" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-250', 'x', 'link-235');">x</a></tt><tt class="py-op">.</tt><tt id="link-251" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-251', 'strip', 'link-236');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-252" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-252', 'x', 'link-235');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">new_tax_list</tt><tt class="py-op">]</tt> </tt>
<a name="L557"></a><tt class="py-lineno"> 557</tt>  <tt class="py-line"> </tt>
<a name="L558"></a><tt class="py-lineno"> 558</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">clean_tax_list</tt> </tt>
</div><a name="L559"></a><tt class="py-lineno"> 559</tt>  <tt class="py-line"> </tt>
<a name="_BaseGenBankConsumer._clean_location"></a><div id="_BaseGenBankConsumer._clean_location-def"><a name="L560"></a><tt class="py-lineno"> 560</tt> <a class="py-toggle" href="#" id="_BaseGenBankConsumer._clean_location-toggle" onclick="return toggle('_BaseGenBankConsumer._clean_location');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._BaseGenBankConsumer-class.html#_clean_location">_clean_location</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">location_string</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BaseGenBankConsumer._clean_location-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_BaseGenBankConsumer._clean_location-expanded"><a name="L561"></a><tt class="py-lineno"> 561</tt>  <tt class="py-line">        <tt class="py-docstring">"""Clean whitespace out of a location string.</tt> </tt>
<a name="L562"></a><tt class="py-lineno"> 562</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L563"></a><tt class="py-lineno"> 563</tt>  <tt class="py-line"><tt class="py-docstring">        The location parser isn't a fan of whitespace, so we clean it out</tt> </tt>
<a name="L564"></a><tt class="py-lineno"> 564</tt>  <tt class="py-line"><tt class="py-docstring">        before feeding it into the parser.</tt> </tt>
<a name="L565"></a><tt class="py-lineno"> 565</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L566"></a><tt class="py-lineno"> 566</tt>  <tt class="py-line">        <tt class="py-comment"># Originally this imported string.whitespace and did a replace</tt> </tt>
<a name="L567"></a><tt class="py-lineno"> 567</tt>  <tt class="py-line">        <tt class="py-comment"># via a loop.  It's simpler to just split on whitespace and rejoin</tt> </tt>
<a name="L568"></a><tt class="py-lineno"> 568</tt>  <tt class="py-line">        <tt class="py-comment"># the string - and this avoids importing string too.  See Bug 2684.</tt> </tt>
<a name="L569"></a><tt class="py-lineno"> 569</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">''</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">location_string</tt><tt class="py-op">.</tt><tt id="link-253" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-253', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L570"></a><tt class="py-lineno"> 570</tt>  <tt class="py-line"> </tt>
<a name="_BaseGenBankConsumer._remove_newlines"></a><div id="_BaseGenBankConsumer._remove_newlines-def"><a name="L571"></a><tt class="py-lineno"> 571</tt> <a class="py-toggle" href="#" id="_BaseGenBankConsumer._remove_newlines-toggle" onclick="return toggle('_BaseGenBankConsumer._remove_newlines');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._BaseGenBankConsumer-class.html#_remove_newlines">_remove_newlines</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BaseGenBankConsumer._remove_newlines-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_BaseGenBankConsumer._remove_newlines-expanded"><a name="L572"></a><tt class="py-lineno"> 572</tt>  <tt class="py-line">        <tt class="py-docstring">"""Remove any newlines in the passed text, returning the new string.</tt> </tt>
<a name="L573"></a><tt class="py-lineno"> 573</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L574"></a><tt class="py-lineno"> 574</tt>  <tt class="py-line">        <tt class="py-comment"># get rid of newlines in the qualifier value</tt> </tt>
<a name="L575"></a><tt class="py-lineno"> 575</tt>  <tt class="py-line">        <tt class="py-name">newlines</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">"\n"</tt><tt class="py-op">,</tt> <tt class="py-string">"\r"</tt><tt class="py-op">]</tt> </tt>
<a name="L576"></a><tt class="py-lineno"> 576</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">ws</tt> <tt class="py-keyword">in</tt> <tt class="py-name">newlines</tt><tt class="py-op">:</tt> </tt>
<a name="L577"></a><tt class="py-lineno"> 577</tt>  <tt class="py-line">            <tt class="py-name">text</tt> <tt class="py-op">=</tt> <tt class="py-name">text</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-name">ws</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">)</tt> </tt>
<a name="L578"></a><tt class="py-lineno"> 578</tt>  <tt class="py-line"> </tt>
<a name="L579"></a><tt class="py-lineno"> 579</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">text</tt> </tt>
</div><a name="L580"></a><tt class="py-lineno"> 580</tt>  <tt class="py-line"> </tt>
<a name="_BaseGenBankConsumer._normalize_spaces"></a><div id="_BaseGenBankConsumer._normalize_spaces-def"><a name="L581"></a><tt class="py-lineno"> 581</tt> <a class="py-toggle" href="#" id="_BaseGenBankConsumer._normalize_spaces-toggle" onclick="return toggle('_BaseGenBankConsumer._normalize_spaces');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._BaseGenBankConsumer-class.html#_normalize_spaces">_normalize_spaces</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BaseGenBankConsumer._normalize_spaces-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_BaseGenBankConsumer._normalize_spaces-expanded"><a name="L582"></a><tt class="py-lineno"> 582</tt>  <tt class="py-line">        <tt class="py-docstring">"""Replace multiple spaces in the passed text with single spaces.</tt> </tt>
<a name="L583"></a><tt class="py-lineno"> 583</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L584"></a><tt class="py-lineno"> 584</tt>  <tt class="py-line">        <tt class="py-comment"># get rid of excessive spaces</tt> </tt>
<a name="L585"></a><tt class="py-lineno"> 585</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">' '</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt id="link-254" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-254', 'x', 'link-235');">x</a></tt> <tt class="py-keyword">for</tt> <tt id="link-255" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-255', 'x', 'link-235');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">text</tt><tt class="py-op">.</tt><tt id="link-256" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-256', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">" "</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt id="link-257" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-257', 'x', 'link-235');">x</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L586"></a><tt class="py-lineno"> 586</tt>  <tt class="py-line"> </tt>
<a name="_BaseGenBankConsumer._remove_spaces"></a><div id="_BaseGenBankConsumer._remove_spaces-def"><a name="L587"></a><tt class="py-lineno"> 587</tt> <a class="py-toggle" href="#" id="_BaseGenBankConsumer._remove_spaces-toggle" onclick="return toggle('_BaseGenBankConsumer._remove_spaces');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._BaseGenBankConsumer-class.html#_remove_spaces">_remove_spaces</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BaseGenBankConsumer._remove_spaces-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_BaseGenBankConsumer._remove_spaces-expanded"><a name="L588"></a><tt class="py-lineno"> 588</tt>  <tt class="py-line">        <tt class="py-docstring">"""Remove all spaces from the passed text.</tt> </tt>
<a name="L589"></a><tt class="py-lineno"> 589</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L590"></a><tt class="py-lineno"> 590</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">text</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">" "</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">)</tt> </tt>
</div><a name="L591"></a><tt class="py-lineno"> 591</tt>  <tt class="py-line"> </tt>
<a name="_BaseGenBankConsumer._convert_to_python_numbers"></a><div id="_BaseGenBankConsumer._convert_to_python_numbers-def"><a name="L592"></a><tt class="py-lineno"> 592</tt> <a class="py-toggle" href="#" id="_BaseGenBankConsumer._convert_to_python_numbers-toggle" onclick="return toggle('_BaseGenBankConsumer._convert_to_python_numbers');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._BaseGenBankConsumer-class.html#_convert_to_python_numbers">_convert_to_python_numbers</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BaseGenBankConsumer._convert_to_python_numbers-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_BaseGenBankConsumer._convert_to_python_numbers-expanded"><a name="L593"></a><tt class="py-lineno"> 593</tt>  <tt class="py-line">        <tt class="py-docstring">"""Convert a start and end range to python notation.</tt> </tt>
<a name="L594"></a><tt class="py-lineno"> 594</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L595"></a><tt class="py-lineno"> 595</tt>  <tt class="py-line"><tt class="py-docstring">        In GenBank, starts and ends are defined in "biological" coordinates,</tt> </tt>
<a name="L596"></a><tt class="py-lineno"> 596</tt>  <tt class="py-line"><tt class="py-docstring">        where 1 is the first base and [i, j] means to include both i and j.</tt> </tt>
<a name="L597"></a><tt class="py-lineno"> 597</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L598"></a><tt class="py-lineno"> 598</tt>  <tt class="py-line"><tt class="py-docstring">        In python, 0 is the first base and [i, j] means to include i, but</tt> </tt>
<a name="L599"></a><tt class="py-lineno"> 599</tt>  <tt class="py-line"><tt class="py-docstring">        not j.</tt> </tt>
<a name="L600"></a><tt class="py-lineno"> 600</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L601"></a><tt class="py-lineno"> 601</tt>  <tt class="py-line"><tt class="py-docstring">        So, to convert "biological" to python coordinates, we need to</tt> </tt>
<a name="L602"></a><tt class="py-lineno"> 602</tt>  <tt class="py-line"><tt class="py-docstring">        subtract 1 from the start, and leave the end and things should</tt> </tt>
<a name="L603"></a><tt class="py-lineno"> 603</tt>  <tt class="py-line"><tt class="py-docstring">        be converted happily.</tt> </tt>
<a name="L604"></a><tt class="py-lineno"> 604</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L605"></a><tt class="py-lineno"> 605</tt>  <tt class="py-line">        <tt class="py-name">new_start</tt> <tt class="py-op">=</tt> <tt id="link-258" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.start=Bio.SeqFeature.CompoundLocation-class.html#start,Variable Bio.SeqFeature.FeatureLocation.start=Bio.SeqFeature.FeatureLocation-class.html#start"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-258', 'start', 'link-258');">start</a></tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt> </tt>
<a name="L606"></a><tt class="py-lineno"> 606</tt>  <tt class="py-line">        <tt class="py-name">new_end</tt> <tt class="py-op">=</tt> <tt id="link-259" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.end=Bio.SeqFeature.CompoundLocation-class.html#end,Variable Bio.SeqFeature.FeatureLocation.end=Bio.SeqFeature.FeatureLocation-class.html#end"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-259', 'end', 'link-259');">end</a></tt> </tt>
<a name="L607"></a><tt class="py-lineno"> 607</tt>  <tt class="py-line"> </tt>
<a name="L608"></a><tt class="py-lineno"> 608</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">new_start</tt><tt class="py-op">,</tt> <tt class="py-name">new_end</tt> </tt>
</div></div><a name="L609"></a><tt class="py-lineno"> 609</tt>  <tt class="py-line"> </tt>
<a name="L610"></a><tt class="py-lineno"> 610</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer"></a><div id="_FeatureConsumer-def"><a name="L611"></a><tt class="py-lineno"> 611</tt> <a class="py-toggle" href="#" id="_FeatureConsumer-toggle" onclick="return toggle('_FeatureConsumer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html">_FeatureConsumer</a><tt class="py-op">(</tt><tt class="py-base-class">_BaseGenBankConsumer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_FeatureConsumer-expanded"><a name="L612"></a><tt class="py-lineno"> 612</tt>  <tt class="py-line">    <tt class="py-docstring">"""Create a SeqRecord object with Features to return (PRIVATE).</tt> </tt>
<a name="L613"></a><tt class="py-lineno"> 613</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L614"></a><tt class="py-lineno"> 614</tt>  <tt class="py-line"><tt class="py-docstring">    Attributes:</tt> </tt>
<a name="L615"></a><tt class="py-lineno"> 615</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L616"></a><tt class="py-lineno"> 616</tt>  <tt class="py-line"><tt class="py-docstring">     - use_fuzziness - specify whether or not to parse with fuzziness in</tt> </tt>
<a name="L617"></a><tt class="py-lineno"> 617</tt>  <tt class="py-line"><tt class="py-docstring">       feature locations.</tt> </tt>
<a name="L618"></a><tt class="py-lineno"> 618</tt>  <tt class="py-line"><tt class="py-docstring">     - feature_cleaner - a class that will be used to provide specialized</tt> </tt>
<a name="L619"></a><tt class="py-lineno"> 619</tt>  <tt class="py-line"><tt class="py-docstring">       cleaning-up of feature values.</tt> </tt>
<a name="L620"></a><tt class="py-lineno"> 620</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="_FeatureConsumer.__init__"></a><div id="_FeatureConsumer.__init__-def"><a name="L621"></a><tt class="py-lineno"> 621</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.__init__-toggle" onclick="return toggle('_FeatureConsumer.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">use_fuzziness</tt><tt class="py-op">,</tt> <tt class="py-param">feature_cleaner</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.__init__-expanded"><a name="L622"></a><tt class="py-lineno"> 622</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt id="link-260" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-260', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-261" class="py-name" targets="Module Bio.SeqRecord=Bio.SeqRecord-module.html,Class Bio.SeqRecord.SeqRecord=Bio.SeqRecord.SeqRecord-class.html"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-261', 'SeqRecord', 'link-261');">SeqRecord</a></tt> <tt class="py-keyword">import</tt> <tt id="link-262" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-262', 'SeqRecord', 'link-261');">SeqRecord</a></tt> </tt>
<a name="L623"></a><tt class="py-lineno"> 623</tt>  <tt class="py-line">        <tt id="link-263" class="py-name" targets="Class Bio.GenBank._BaseGenBankConsumer=Bio.GenBank._BaseGenBankConsumer-class.html"><a title="Bio.GenBank._BaseGenBankConsumer" class="py-name" href="#" onclick="return doclink('link-263', '_BaseGenBankConsumer', 'link-263');">_BaseGenBankConsumer</a></tt><tt class="py-op">.</tt><tt id="link-264" class="py-name" targets="Method Bio.Affy.CelFile.Record.__init__()=Bio.Affy.CelFile.Record-class.html#__init__,Method Bio.Align.AlignInfo.PSSM.__init__()=Bio.Align.AlignInfo.PSSM-class.html#__init__,Method Bio.Align.AlignInfo.SummaryInfo.__init__()=Bio.Align.AlignInfo.SummaryInfo-class.html#__init__,Method Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline.__init__()=Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline-class.html#__init__,Method Bio.Align.Applications._Clustalw.ClustalwCommandline.__init__()=Bio.Align.Applications._Clustalw.ClustalwCommandline-class.html#__init__,Method Bio.Align.Applications._Dialign.DialignCommandline.__init__()=Bio.Align.Applications._Dialign.DialignCommandline-class.html#__init__,Method Bio.Align.Applications._MSAProbs.MSAProbsCommandline.__init__()=Bio.Align.Applications._MSAProbs.MSAProbsCommandline-class.html#__init__,Method Bio.Align.Applications._Mafft.MafftCommandline.__init__()=Bio.Align.Applications._Mafft.MafftCommandline-class.html#__init__,Method Bio.Align.Applications._Muscle.MuscleCommandline.__init__()=Bio.Align.Applications._Muscle.MuscleCommandline-class.html#__init__,Method Bio.Align.Applications._Prank.PrankCommandline.__init__()=Bio.Align.Applications._Prank.PrankCommandline-class.html#__init__,Method Bio.Align.Applications._Probcons.ProbconsCommandline.__init__()=Bio.Align.Applications._Probcons.ProbconsCommandline-class.html#__init__,Method Bio.Align.Applications._TCoffee.TCoffeeCommandline.__init__()=Bio.Align.Applications._TCoffee.TCoffeeCommandline-class.html#__init__,Method Bio.Align.Generic.Alignment.__init__()=Bio.Align.Generic.Alignment-class.html#__init__,Method Bio.Align.MultipleSeqAlignment.__init__()=Bio.Align.MultipleSeqAlignment-class.html#__init__,Method Bio.AlignIO.Interfaces.AlignmentIterator.__init__()=Bio.AlignIO.Interfaces.AlignmentIterator-class.html#__init__,Method Bio.AlignIO.Interfaces.AlignmentWriter.__init__()=Bio.AlignIO.Interfaces.AlignmentWriter-class.html#__init__,Method Bio.AlignIO.Interfaces.SequentialAlignmentWriter.__init__()=Bio.AlignIO.Interfaces.SequentialAlignmentWriter-class.html#__init__,Method Bio.Alphabet.AlphabetEncoder.__init__()=Bio.Alphabet.AlphabetEncoder-class.html#__init__,Method Bio.Alphabet.Gapped.__init__()=Bio.Alphabet.Gapped-class.html#__init__,Method Bio.Alphabet.HasStopCodon.__init__()=Bio.Alphabet.HasStopCodon-class.html#__init__,Method Bio.Application.AbstractCommandline.__init__()=Bio.Application.AbstractCommandline-class.html#__init__,Method Bio.Application.ApplicationError.__init__()=Bio.Application.ApplicationError-class.html#__init__,Method Bio.Application._AbstractParameter.__init__()=Bio.Application._AbstractParameter-class.html#__init__,Method Bio.Application._Argument.__init__()=Bio.Application._Argument-class.html#__init__,Method Bio.Application._Option.__init__()=Bio.Application._Option-class.html#__init__,Method Bio.Application._StaticArgument.__init__()=Bio.Application._StaticArgument-class.html#__init__,Method Bio.Application._Switch.__init__()=Bio.Application._Switch-class.html#__init__,Method Bio.Blast.Applications.NcbiblastformatterCommandline.__init__()=Bio.Blast.Applications.NcbiblastformatterCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastnCommandline.__init__()=Bio.Blast.Applications.NcbiblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastpCommandline.__init__()=Bio.Blast.Applications.NcbiblastpCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastxCommandline.__init__()=Bio.Blast.Applications.NcbiblastxCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbideltablastCommandline.__init__()=Bio.Blast.Applications.NcbideltablastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbipsiblastCommandline.__init__()=Bio.Blast.Applications.NcbipsiblastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbirpsblastCommandline.__init__()=Bio.Blast.Applications.NcbirpsblastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbirpstblastnCommandline.__init__()=Bio.Blast.Applications.NcbirpstblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbitblastnCommandline.__init__()=Bio.Blast.Applications.NcbitblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbitblastxCommandline.__init__()=Bio.Blast.Applications.NcbitblastxCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbibaseblastCommandline.__init__()=Bio.Blast.Applications._NcbibaseblastCommandline-class.html#__init__,Method Bio.Blast.Applications._Ncbiblast2SeqCommandline.__init__()=Bio.Blast.Applications._Ncbiblast2SeqCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbiblastCommandline.__init__()=Bio.Blast.Applications._NcbiblastCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbiblastMain2SeqCommandline.__init__()=Bio.Blast.Applications._NcbiblastMain2SeqCommandline-class.html#__init__,Method Bio.Blast.NCBIStandalone.BlastErrorParser.__init__()=Bio.Blast.NCBIStandalone.BlastErrorParser-class.html#__init__,Method Bio.Blast.NCBIStandalone.BlastParser.__init__()=Bio.Blast.NCBIStandalone.BlastParser-class.html#__init__,Method Bio.Blast.NCBIStandalone.Iterator.__init__()=Bio.Blast.NCBIStandalone.Iterator-class.html#__init__,Method Bio.Blast.NCBIStandalone.PSIBlastParser.__init__()=Bio.Blast.NCBIStandalone.PSIBlastParser-class.html#__init__,Method Bio.Blast.NCBIStandalone._BlastConsumer.__init__()=Bio.Blast.NCBIStandalone._BlastConsumer-class.html#__init__,Method Bio.Blast.NCBIStandalone._BlastErrorConsumer.__init__()=Bio.Blast.NCBIStandalone._BlastErrorConsumer-class.html#__init__,Method Bio.Blast.NCBIStandalone._PSIBlastConsumer.__init__()=Bio.Blast.NCBIStandalone._PSIBlastConsumer-class.html#__init__,Method Bio.Blast.NCBIXML.BlastParser.__init__()=Bio.Blast.NCBIXML.BlastParser-class.html#__init__,Method Bio.Blast.NCBIXML._XMLparser.__init__()=Bio.Blast.NCBIXML._XMLparser-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableEntry.__init__()=Bio.Blast.ParseBlastTable.BlastTableEntry-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableReader.__init__()=Bio.Blast.ParseBlastTable.BlastTableReader-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableRec.__init__()=Bio.Blast.ParseBlastTable.BlastTableRec-class.html#__init__,Method Bio.Blast.Record.Alignment.__init__()=Bio.Blast.Record.Alignment-class.html#__init__,Method Bio.Blast.Record.Blast.__init__()=Bio.Blast.Record.Blast-class.html#__init__,Method Bio.Blast.Record.DatabaseReport.__init__()=Bio.Blast.Record.DatabaseReport-class.html#__init__,Method Bio.Blast.Record.Description.__init__()=Bio.Blast.Record.Description-class.html#__init__,Method Bio.Blast.Record.HSP.__init__()=Bio.Blast.Record.HSP-class.html#__init__,Method Bio.Blast.Record.Header.__init__()=Bio.Blast.Record.Header-class.html#__init__,Method Bio.Blast.Record.MultipleAlignment.__init__()=Bio.Blast.Record.MultipleAlignment-class.html#__init__,Method Bio.Blast.Record.PSIBlast.__init__()=Bio.Blast.Record.PSIBlast-class.html#__init__,Method Bio.Blast.Record.Parameters.__init__()=Bio.Blast.Record.Parameters-class.html#__init__,Method Bio.Blast.Record.Round.__init__()=Bio.Blast.Record.Round-class.html#__init__,Method Bio.CAPS.CAPSMap.__init__()=Bio.CAPS.CAPSMap-class.html#__init__,Method Bio.CAPS.DifferentialCutsite.__init__()=Bio.CAPS.DifferentialCutsite-class.html#__init__,Method Bio.Cluster.Record.__init__()=Bio.Cluster.Record-class.html#__init__,Method Bio.Compass.Record.__init__()=Bio.Compass.Record-class.html#__init__,Method Bio.Crystal.Chain.__init__()=Bio.Crystal.Chain-class.html#__init__,Method Bio.Crystal.Crystal.__init__()=Bio.Crystal.Crystal-class.html#__init__,Method Bio.Crystal.Hetero.__init__()=Bio.Crystal.Hetero-class.html#__init__,Method Bio.Data.CodonTable.AmbiguousCodonTable.__init__()=Bio.Data.CodonTable.AmbiguousCodonTable-class.html#__init__,Method Bio.Data.CodonTable.AmbiguousForwardTable.__init__()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#__init__,Method Bio.Data.CodonTable.CodonTable.__init__()=Bio.Data.CodonTable.CodonTable-class.html#__init__,Method Bio.Data.CodonTable.NCBICodonTable.__init__()=Bio.Data.CodonTable.NCBICodonTable-class.html#__init__,Method Bio.DocSQL.Create.__init__()=Bio.DocSQL.Create-class.html#__init__,Method Bio.DocSQL.Insert.__init__()=Bio.DocSQL.Insert-class.html#__init__,Method Bio.DocSQL.IterationCursor.__init__()=Bio.DocSQL.IterationCursor-class.html#__init__,Method Bio.DocSQL.Query.__init__()=Bio.DocSQL.Query-class.html#__init__,Method Bio.DocSQL.QueryAll.__init__()=Bio.DocSQL.QueryAll-class.html#__init__,Method Bio.DocSQL.QueryGeneric.__init__()=Bio.DocSQL.QueryGeneric-class.html#__init__,Method Bio.DocSQL.QueryRow.__init__()=Bio.DocSQL.QueryRow-class.html#__init__,Method Bio.DocSQL.QuerySingle.__init__()=Bio.DocSQL.QuerySingle-class.html#__init__,Method Bio.Emboss.Applications.DiffseqCommandline.__init__()=Bio.Emboss.Applications.DiffseqCommandline-class.html#__init__,Method Bio.Emboss.Applications.EInvertedCommandline.__init__()=Bio.Emboss.Applications.EInvertedCommandline-class.html#__init__,Method Bio.Emboss.Applications.ETandemCommandline.__init__()=Bio.Emboss.Applications.ETandemCommandline-class.html#__init__,Method Bio.Emboss.Applications.Est2GenomeCommandline.__init__()=Bio.Emboss.Applications.Est2GenomeCommandline-class.html#__init__,Method Bio.Emboss.Applications.FConsenseCommandline.__init__()=Bio.Emboss.Applications.FConsenseCommandline-class.html#__init__,Method Bio.Emboss.Applications.FDNADistCommandline.__init__()=Bio.Emboss.Applications.FDNADistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FDNAParsCommandline.__init__()=Bio.Emboss.Applications.FDNAParsCommandline-class.html#__init__,Method Bio.Emboss.Applications.FNeighborCommandline.__init__()=Bio.Emboss.Applications.FNeighborCommandline-class.html#__init__,Method Bio.Emboss.Applications.FProtDistCommandline.__init__()=Bio.Emboss.Applications.FProtDistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FProtParsCommandline.__init__()=Bio.Emboss.Applications.FProtParsCommandline-class.html#__init__,Method Bio.Emboss.Applications.FSeqBootCommandline.__init__()=Bio.Emboss.Applications.FSeqBootCommandline-class.html#__init__,Method Bio.Emboss.Applications.FTreeDistCommandline.__init__()=Bio.Emboss.Applications.FTreeDistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FuzznucCommandline.__init__()=Bio.Emboss.Applications.FuzznucCommandline-class.html#__init__,Method Bio.Emboss.Applications.IepCommandline.__init__()=Bio.Emboss.Applications.IepCommandline-class.html#__init__,Method Bio.Emboss.Applications.NeedleCommandline.__init__()=Bio.Emboss.Applications.NeedleCommandline-class.html#__init__,Method Bio.Emboss.Applications.NeedleallCommandline.__init__()=Bio.Emboss.Applications.NeedleallCommandline-class.html#__init__,Method Bio.Emboss.Applications.PalindromeCommandline.__init__()=Bio.Emboss.Applications.PalindromeCommandline-class.html#__init__,Method Bio.Emboss.Applications.Primer3Commandline.__init__()=Bio.Emboss.Applications.Primer3Commandline-class.html#__init__,Method Bio.Emboss.Applications.PrimerSearchCommandline.__init__()=Bio.Emboss.Applications.PrimerSearchCommandline-class.html#__init__,Method Bio.Emboss.Applications.SeqmatchallCommandline.__init__()=Bio.Emboss.Applications.SeqmatchallCommandline-class.html#__init__,Method Bio.Emboss.Applications.SeqretCommandline.__init__()=Bio.Emboss.Applications.SeqretCommandline-class.html#__init__,Method Bio.Emboss.Applications.StretcherCommandline.__init__()=Bio.Emboss.Applications.StretcherCommandline-class.html#__init__,Method Bio.Emboss.Applications.TranalignCommandline.__init__()=Bio.Emboss.Applications.TranalignCommandline-class.html#__init__,Method Bio.Emboss.Applications.WaterCommandline.__init__()=Bio.Emboss.Applications.WaterCommandline-class.html#__init__,Method Bio.Emboss.Applications._EmbossCommandLine.__init__()=Bio.Emboss.Applications._EmbossCommandLine-class.html#__init__,Method Bio.Emboss.Applications._EmbossMinimalCommandLine.__init__()=Bio.Emboss.Applications._EmbossMinimalCommandLine-class.html#__init__,Method Bio.Emboss.Primer3.Primers.__init__()=Bio.Emboss.Primer3.Primers-class.html#__init__,Method Bio.Emboss.Primer3.Record.__init__()=Bio.Emboss.Primer3.Record-class.html#__init__,Method Bio.Emboss.PrimerSearch.Amplifier.__init__()=Bio.Emboss.PrimerSearch.Amplifier-class.html#__init__,Method Bio.Emboss.PrimerSearch.InputRecord.__init__()=Bio.Emboss.PrimerSearch.InputRecord-class.html#__init__,Method Bio.Emboss.PrimerSearch.OutputRecord.__init__()=Bio.Emboss.PrimerSearch.OutputRecord-class.html#__init__,Method Bio.Entrez.Parser.CorruptedXMLError.__init__()=Bio.Entrez.Parser.CorruptedXMLError-class.html#__init__,Method Bio.Entrez.Parser.DataHandler.__init__()=Bio.Entrez.Parser.DataHandler-class.html#__init__,Method Bio.Entrez.Parser.NotXMLError.__init__()=Bio.Entrez.Parser.NotXMLError-class.html#__init__,Method Bio.Entrez.Parser.StructureElement.__init__()=Bio.Entrez.Parser.StructureElement-class.html#__init__,Method Bio.Entrez.Parser.ValidationError.__init__()=Bio.Entrez.Parser.ValidationError-class.html#__init__,Method Bio.ExPASy.Enzyme.Record.__init__()=Bio.ExPASy.Enzyme.Record-class.html#__init__,Method Bio.ExPASy.Prodoc.Record.__init__()=Bio.ExPASy.Prodoc.Record-class.html#__init__,Method Bio.ExPASy.Prodoc.Reference.__init__()=Bio.ExPASy.Prodoc.Reference-class.html#__init__,Method Bio.ExPASy.Prosite.Record.__init__()=Bio.ExPASy.Prosite.Record-class.html#__init__,Method Bio.ExPASy.ScanProsite.ContentHandler.__init__()=Bio.ExPASy.ScanProsite.ContentHandler-class.html#__init__,Method Bio.ExPASy.ScanProsite.Parser.__init__()=Bio.ExPASy.ScanProsite.Parser-class.html#__init__,Method Bio.ExPASy.ScanProsite.Record.__init__()=Bio.ExPASy.ScanProsite.Record-class.html#__init__,Method Bio.FSSP.FSSPAlignDict.__init__()=Bio.FSSP.FSSPAlignDict-class.html#__init__,Method Bio.FSSP.FSSPAlignRec.__init__()=Bio.FSSP.FSSPAlignRec-class.html#__init__,Method Bio.FSSP.FSSPHeader.__init__()=Bio.FSSP.FSSPHeader-class.html#__init__,Method Bio.FSSP.FSSPSumRec.__init__()=Bio.FSSP.FSSPSumRec-class.html#__init__,Method Bio.FSSP.FSSPTools.FSSPMultAlign.__init__()=Bio.FSSP.FSSPTools.FSSPMultAlign-class.html#__init__,Method Bio.FSSP.PosAlign.__init__()=Bio.FSSP.PosAlign-class.html#__init__,Method Bio.FSSP.fssp_rec.fff_rec.__init__()=Bio.FSSP.fssp_rec.fff_rec-class.html#__init__,Method Bio.File.UndoHandle.__init__()=Bio.File.UndoHandle-class.html#__init__,Method Bio.File._IndexedSeqFileDict.__init__()=Bio.File._IndexedSeqFileDict-class.html#__init__,Method Bio.File._SQLiteManySeqFilesDict.__init__()=Bio.File._SQLiteManySeqFilesDict-class.html#__init__,Method Bio.GA.Crossover.General.SafeFitnessCrossover.__init__()=Bio.GA.Crossover.General.SafeFitnessCrossover-class.html#__init__,Method Bio.GA.Crossover.GeneralPoint.GeneralPointCrossover.__init__()=Bio.GA.Crossover.GeneralPoint.GeneralPointCrossover-class.html#__init__,Method Bio.GA.Crossover.GeneralPoint.InterleaveCrossover.__init__()=Bio.GA.Crossover.GeneralPoint.InterleaveCrossover-class.html#__init__,Method Bio.GA.Crossover.Point.SinglePointCrossover.__init__()=Bio.GA.Crossover.Point.SinglePointCrossover-class.html#__init__,Method Bio.GA.Crossover.TwoPoint.TwoPointCrossover.__init__()=Bio.GA.Crossover.TwoPoint.TwoPointCrossover-class.html#__init__,Method Bio.GA.Crossover.Uniform.UniformCrossover.__init__()=Bio.GA.Crossover.Uniform.UniformCrossover-class.html#__init__,Method Bio.GA.Evolver.GenerationEvolver.__init__()=Bio.GA.Evolver.GenerationEvolver-class.html#__init__,Method Bio.GA.Evolver.SteadyStateEvolver.__init__()=Bio.GA.Evolver.SteadyStateEvolver-class.html#__init__,Method Bio.GA.Mutation.General.SafeFitnessMutation.__init__()=Bio.GA.Mutation.General.SafeFitnessMutation-class.html#__init__,Method Bio.GA.Mutation.Simple.ConversionMutation.__init__()=Bio.GA.Mutation.Simple.ConversionMutation-class.html#__init__,Method Bio.GA.Mutation.Simple.SinglePositionMutation.__init__()=Bio.GA.Mutation.Simple.SinglePositionMutation-class.html#__init__,Method Bio.GA.Organism.Organism.__init__()=Bio.GA.Organism.Organism-class.html#__init__,Method Bio.GA.Repair.Stabilizing.AmbiguousRepair.__init__()=Bio.GA.Repair.Stabilizing.AmbiguousRepair-class.html#__init__,Method Bio.GA.Selection.Abstract.AbstractSelection.__init__()=Bio.GA.Selection.Abstract.AbstractSelection-class.html#__init__,Method Bio.GA.Selection.Diversity.DiversitySelection.__init__()=Bio.GA.Selection.Diversity.DiversitySelection-class.html#__init__,Method Bio.GA.Selection.RouletteWheel.RouletteWheelSelection.__init__()=Bio.GA.Selection.RouletteWheel.RouletteWheelSelection-class.html#__init__,Method Bio.GA.Selection.Tournament.TournamentSelection.__init__()=Bio.GA.Selection.Tournament.TournamentSelection-class.html#__init__,Method Bio.GenBank.FeatureParser.__init__()=Bio.GenBank.FeatureParser-class.html#__init__,Method Bio.GenBank.Iterator.__init__()=Bio.GenBank.Iterator-class.html#__init__,Method Bio.GenBank.Record.Feature.__init__()=Bio.GenBank.Record.Feature-class.html#__init__,Method Bio.GenBank.Record.Qualifier.__init__()=Bio.GenBank.Record.Qualifier-class.html#__init__,Method Bio.GenBank.Record.Record.__init__()=Bio.GenBank.Record.Record-class.html#__init__,Method Bio.GenBank.Record.Reference.__init__()=Bio.GenBank.Record.Reference-class.html#__init__,Method Bio.GenBank.RecordParser.__init__()=Bio.GenBank.RecordParser-class.html#__init__,Method Bio.GenBank.Scanner.InsdcScanner.__init__()=Bio.GenBank.Scanner.InsdcScanner-class.html#__init__,Method Bio.GenBank._BaseGenBankConsumer.__init__()=Bio.GenBank._BaseGenBankConsumer-class.html#__init__,Method Bio.GenBank._FeatureConsumer.__init__()=Bio.GenBank._FeatureConsumer-class.html#__init__,Method Bio.GenBank._RecordConsumer.__init__()=Bio.GenBank._RecordConsumer-class.html#__init__,Method Bio.GenBank.utils.FeatureValueCleaner.__init__()=Bio.GenBank.utils.FeatureValueCleaner-class.html#__init__,Method Bio.Geo.Record.Record.__init__()=Bio.Geo.Record.Record-class.html#__init__,Method Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment.__init__()=Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome.Chromosome.__init__()=Bio.Graphics.BasicChromosome.Chromosome-class.html#__init__,Method Bio.Graphics.BasicChromosome.ChromosomeSegment.__init__()=Bio.Graphics.BasicChromosome.ChromosomeSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome.Organism.__init__()=Bio.Graphics.BasicChromosome.Organism-class.html#__init__,Method Bio.Graphics.BasicChromosome.TelomereSegment.__init__()=Bio.Graphics.BasicChromosome.TelomereSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome._ChromosomeComponent.__init__()=Bio.Graphics.BasicChromosome._ChromosomeComponent-class.html#__init__,Method Bio.Graphics.ColorSpiral.ColorSpiral.__init__()=Bio.Graphics.ColorSpiral.ColorSpiral-class.html#__init__,Method Bio.Graphics.Comparative.ComparativeScatterPlot.__init__()=Bio.Graphics.Comparative.ComparativeScatterPlot-class.html#__init__,Method Bio.Graphics.DisplayRepresentation.ChromosomeCounts.__init__()=Bio.Graphics.DisplayRepresentation.ChromosomeCounts-class.html#__init__,Method Bio.Graphics.Distribution.BarChartDistribution.__init__()=Bio.Graphics.Distribution.BarChartDistribution-class.html#__init__,Method Bio.Graphics.Distribution.DistributionPage.__init__()=Bio.Graphics.Distribution.DistributionPage-class.html#__init__,Method Bio.Graphics.Distribution.LineDistribution.__init__()=Bio.Graphics.Distribution.LineDistribution-class.html#__init__,Method Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.__init__()=Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.__init__()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.__init__()=Bio.Graphics.GenomeDiagram._Colors.ColorTranslator-class.html#__init__,Method Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.__init__()=Bio.Graphics.GenomeDiagram._CrossLink.CrossLink-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.__init__()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Feature.Feature.__init__()=Bio.Graphics.GenomeDiagram._Feature.Feature-class.html#__init__,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.__init__()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.__init__()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#__init__,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.__init__()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#__init__,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.__init__()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Track.Track.__init__()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#__init__,Method Bio.Graphics.KGML_vis.KGMLCanvas.__init__()=Bio.Graphics.KGML_vis.KGMLCanvas-class.html#__init__,Method Bio.HMM.DynamicProgramming.AbstractDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.AbstractDPAlgorithms-class.html#__init__,Method Bio.HMM.DynamicProgramming.LogDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.LogDPAlgorithms-class.html#__init__,Method Bio.HMM.DynamicProgramming.ScaledDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.ScaledDPAlgorithms-class.html#__init__,Method Bio.HMM.MarkovModel.HiddenMarkovModel.__init__()=Bio.HMM.MarkovModel.HiddenMarkovModel-class.html#__init__,Method Bio.HMM.MarkovModel.MarkovModelBuilder.__init__()=Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#__init__,Method Bio.HMM.Trainer.AbstractTrainer.__init__()=Bio.HMM.Trainer.AbstractTrainer-class.html#__init__,Method Bio.HMM.Trainer.BaumWelchTrainer.__init__()=Bio.HMM.Trainer.BaumWelchTrainer-class.html#__init__,Method Bio.HMM.Trainer.KnownStateTrainer.__init__()=Bio.HMM.Trainer.KnownStateTrainer-class.html#__init__,Method Bio.HMM.Trainer.TrainingSequence.__init__()=Bio.HMM.Trainer.TrainingSequence-class.html#__init__,Method Bio.Index._InMemoryIndex.__init__()=Bio.Index._InMemoryIndex-class.html#__init__,Method Bio.Index._ShelveIndex.__init__()=Bio.Index._ShelveIndex-class.html#__init__,Method Bio.KDTree.KDTree'.KDTree.__init__()=Bio.KDTree.KDTree%27.KDTree-class.html#__init__,Method Bio.KEGG.Compound.Record.__init__()=Bio.KEGG.Compound.Record-class.html#__init__,Method Bio.KEGG.Enzyme.Record.__init__()=Bio.KEGG.Enzyme.Record-class.html#__init__,Method Bio.KEGG.KGML.KGML_parser.KGMLParser.__init__()=Bio.KEGG.KGML.KGML_parser.KGMLParser-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Component.__init__()=Bio.KEGG.KGML.KGML_pathway.Component-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Entry.__init__()=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Graphics.__init__()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Pathway.__init__()=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Reaction.__init__()=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Relation.__init__()=Bio.KEGG.KGML.KGML_pathway.Relation-class.html#__init__,Method Bio.LogisticRegression.LogisticRegression.__init__()=Bio.LogisticRegression.LogisticRegression-class.html#__init__,Method Bio.MarkovModel.MarkovModel.__init__()=Bio.MarkovModel.MarkovModel-class.html#__init__,Method Bio.MaxEntropy.MaxEntropy.__init__()=Bio.MaxEntropy.MaxEntropy-class.html#__init__,Method Bio.Motif.Parsers.AlignAce.Record.__init__()=Bio.Motif.Parsers.AlignAce.Record-class.html#__init__,Method Bio.Motif.Parsers.MAST.Record.__init__()=Bio.Motif.Parsers.MAST.Record-class.html#__init__,Method Bio.Motif.Parsers.MEME.MEMEInstance.__init__()=Bio.Motif.Parsers.MEME.MEMEInstance-class.html#__init__,Method Bio.Motif.Parsers.MEME.MEMEMotif.__init__()=Bio.Motif.Parsers.MEME.MEMEMotif-class.html#__init__,Method Bio.Motif.Parsers.MEME.MEMERecord.__init__()=Bio.Motif.Parsers.MEME.MEMERecord-class.html#__init__,Method Bio.Motif.Thresholds.ScoreDistribution.__init__()=Bio.Motif.Thresholds.ScoreDistribution-class.html#__init__,Method Bio.Motif._Motif.Motif.__init__()=Bio.Motif._Motif.Motif-class.html#__init__,Method Bio.NMR.xpktools.Peaklist.__init__()=Bio.NMR.xpktools.Peaklist-class.html#__init__,Method Bio.NMR.xpktools.XpkEntry.__init__()=Bio.NMR.xpktools.XpkEntry-class.html#__init__,Method Bio.NaiveBayes.NaiveBayes.__init__()=Bio.NaiveBayes.NaiveBayes-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.AbstractLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.AbstractLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.InputLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Network.BasicNetwork.__init__()=Bio.NeuralNetwork.BackPropagation.Network.BasicNetwork-class.html#__init__,Method Bio.NeuralNetwork.Gene.Motif.MotifCoder.__init__()=Bio.NeuralNetwork.Gene.Motif.MotifCoder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Motif.MotifFinder.__init__()=Bio.NeuralNetwork.Gene.Motif.MotifFinder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Pattern.PatternIO.__init__()=Bio.NeuralNetwork.Gene.Pattern.PatternIO-class.html#__init__,Method Bio.NeuralNetwork.Gene.Pattern.PatternRepository.__init__()=Bio.NeuralNetwork.Gene.Pattern.PatternRepository-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness.__init__()=Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder.__init__()=Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness.__init__()=Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator.__init__()=Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.Schema.__init__()=Bio.NeuralNetwork.Gene.Schema.Schema-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SchemaCoder.__init__()=Bio.NeuralNetwork.Gene.Schema.SchemaCoder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SchemaFactory.__init__()=Bio.NeuralNetwork.Gene.Schema.SchemaFactory-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SchemaFinder.__init__()=Bio.NeuralNetwork.Gene.Schema.SchemaFinder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SimpleFinisher.__init__()=Bio.NeuralNetwork.Gene.Schema.SimpleFinisher-class.html#__init__,Method Bio.NeuralNetwork.Gene.Signature.SignatureCoder.__init__()=Bio.NeuralNetwork.Gene.Signature.SignatureCoder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Signature.SignatureFinder.__init__()=Bio.NeuralNetwork.Gene.Signature.SignatureFinder-class.html#__init__,Method Bio.NeuralNetwork.StopTraining.ValidationIncreaseStop.__init__()=Bio.NeuralNetwork.StopTraining.ValidationIncreaseStop-class.html#__init__,Method Bio.NeuralNetwork.Training.ExampleManager.__init__()=Bio.NeuralNetwork.Training.ExampleManager-class.html#__init__,Method Bio.NeuralNetwork.Training.TrainingExample.__init__()=Bio.NeuralNetwork.Training.TrainingExample-class.html#__init__,Method Bio.Nexus.Nexus.Block.__init__()=Bio.Nexus.Nexus.Block-class.html#__init__,Method Bio.Nexus.Nexus.CharBuffer.__init__()=Bio.Nexus.Nexus.CharBuffer-class.html#__init__,Method Bio.Nexus.Nexus.Commandline.__init__()=Bio.Nexus.Nexus.Commandline-class.html#__init__,Method Bio.Nexus.Nexus.Nexus.__init__()=Bio.Nexus.Nexus.Nexus-class.html#__init__,Method Bio.Nexus.Nexus.StepMatrix.__init__()=Bio.Nexus.Nexus.StepMatrix-class.html#__init__,Method Bio.Nexus.Nodes.Chain.__init__()=Bio.Nexus.Nodes.Chain-class.html#__init__,Method Bio.Nexus.Nodes.Node.__init__()=Bio.Nexus.Nodes.Node-class.html#__init__,Method Bio.Nexus.Trees.NodeData.__init__()=Bio.Nexus.Trees.NodeData-class.html#__init__,Method Bio.Nexus.Trees.Tree.__init__()=Bio.Nexus.Trees.Tree-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractPropertyMap-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap-class.html#__init__,Method Bio.PDB.Atom.Atom.__init__()=Bio.PDB.Atom.Atom-class.html#__init__,Method Bio.PDB.Atom.DisorderedAtom.__init__()=Bio.PDB.Atom.DisorderedAtom-class.html#__init__,Method Bio.PDB.Chain.Chain.__init__()=Bio.PDB.Chain.Chain-class.html#__init__,Method Bio.PDB.DSSP'.DSSP.__init__()=Bio.PDB.DSSP%27.DSSP-class.html#__init__,Method Bio.PDB.Dice.ChainSelector.__init__()=Bio.PDB.Dice.ChainSelector-class.html#__init__,Method Bio.PDB.Entity.DisorderedEntityWrapper.__init__()=Bio.PDB.Entity.DisorderedEntityWrapper-class.html#__init__,Method Bio.PDB.Entity.Entity.__init__()=Bio.PDB.Entity.Entity-class.html#__init__,Method Bio.PDB.FragmentMapper'.Fragment.__init__()=Bio.PDB.FragmentMapper%27.Fragment-class.html#__init__,Method Bio.PDB.FragmentMapper'.FragmentMapper.__init__()=Bio.PDB.FragmentMapper%27.FragmentMapper-class.html#__init__,Method Bio.PDB.HSExposure.ExposureCN.__init__()=Bio.PDB.HSExposure.ExposureCN-class.html#__init__,Method Bio.PDB.HSExposure.HSExposureCA.__init__()=Bio.PDB.HSExposure.HSExposureCA-class.html#__init__,Method Bio.PDB.HSExposure.HSExposureCB.__init__()=Bio.PDB.HSExposure.HSExposureCB-class.html#__init__,Method Bio.PDB.HSExposure._AbstractHSExposure.__init__()=Bio.PDB.HSExposure._AbstractHSExposure-class.html#__init__,Method Bio.PDB.MMCIF2Dict.MMCIF2Dict.__init__()=Bio.PDB.MMCIF2Dict.MMCIF2Dict-class.html#__init__,Method Bio.PDB.MMCIFParser'.MMCIFParser.__init__()=Bio.PDB.MMCIFParser%27.MMCIFParser-class.html#__init__,Method Bio.PDB.Model.Model.__init__()=Bio.PDB.Model.Model-class.html#__init__,Method Bio.PDB.NACCESS.NACCESS.__init__()=Bio.PDB.NACCESS.NACCESS-class.html#__init__,Method Bio.PDB.NACCESS.NACCESS_atomic.__init__()=Bio.PDB.NACCESS.NACCESS_atomic-class.html#__init__,Method Bio.PDB.NeighborSearch'.NeighborSearch.__init__()=Bio.PDB.NeighborSearch%27.NeighborSearch-class.html#__init__,Method Bio.PDB.PDBIO'.PDBIO.__init__()=Bio.PDB.PDBIO%27.PDBIO-class.html#__init__,Method Bio.PDB.PDBList'.PDBList.__init__()=Bio.PDB.PDBList%27.PDBList-class.html#__init__,Method Bio.PDB.PDBParser'.PDBParser.__init__()=Bio.PDB.PDBParser%27.PDBParser-class.html#__init__,Method Bio.PDB.PSEA.PSEA.__init__()=Bio.PDB.PSEA.PSEA-class.html#__init__,Method Bio.PDB.Polypeptide.CaPPBuilder.__init__()=Bio.PDB.Polypeptide.CaPPBuilder-class.html#__init__,Method Bio.PDB.Polypeptide.PPBuilder.__init__()=Bio.PDB.Polypeptide.PPBuilder-class.html#__init__,Method Bio.PDB.Polypeptide._PPBuilder.__init__()=Bio.PDB.Polypeptide._PPBuilder-class.html#__init__,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.__init__()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#__init__,Method Bio.PDB.Residue.DisorderedResidue.__init__()=Bio.PDB.Residue.DisorderedResidue-class.html#__init__,Method Bio.PDB.Residue.Residue.__init__()=Bio.PDB.Residue.Residue-class.html#__init__,Method Bio.PDB.ResidueDepth'.ResidueDepth.__init__()=Bio.PDB.ResidueDepth%27.ResidueDepth-class.html#__init__,Method Bio.PDB.Structure.Structure.__init__()=Bio.PDB.Structure.Structure-class.html#__init__,Method Bio.PDB.StructureAlignment'.StructureAlignment.__init__()=Bio.PDB.StructureAlignment%27.StructureAlignment-class.html#__init__,Method Bio.PDB.StructureBuilder.StructureBuilder.__init__()=Bio.PDB.StructureBuilder.StructureBuilder-class.html#__init__,Method Bio.PDB.Superimposer'.Superimposer.__init__()=Bio.PDB.Superimposer%27.Superimposer-class.html#__init__,Method Bio.PDB.Vector'.Vector.__init__()=Bio.PDB.Vector%27.Vector-class.html#__init__,Method Bio.ParserSupport.EventGenerator.__init__()=Bio.ParserSupport.EventGenerator-class.html#__init__,Method Bio.ParserSupport.TaggingConsumer.__init__()=Bio.ParserSupport.TaggingConsumer-class.html#__init__,Method Bio.Pathway.Network.__init__()=Bio.Pathway.Network-class.html#__init__,Method Bio.Pathway.Reaction.__init__()=Bio.Pathway.Reaction-class.html#__init__,Method Bio.Pathway.Rep.Graph.Graph.__init__()=Bio.Pathway.Rep.Graph.Graph-class.html#__init__,Method Bio.Pathway.Rep.MultiGraph.MultiGraph.__init__()=Bio.Pathway.Rep.MultiGraph.MultiGraph-class.html#__init__,Method Bio.Pathway.System.__init__()=Bio.Pathway.System-class.html#__init__,Method Bio.Phylo.Applications._Fasttree.FastTreeCommandline.__init__()=Bio.Phylo.Applications._Fasttree.FastTreeCommandline-class.html#__init__,Method Bio.Phylo.Applications._Phyml.PhymlCommandline.__init__()=Bio.Phylo.Applications._Phyml.PhymlCommandline-class.html#__init__,Method Bio.Phylo.Applications._Raxml.RaxmlCommandline.__init__()=Bio.Phylo.Applications._Raxml.RaxmlCommandline-class.html#__init__,Method Bio.Phylo.BaseTree.BranchColor.__init__()=Bio.Phylo.BaseTree.BranchColor-class.html#__init__,Method Bio.Phylo.BaseTree.Clade.__init__()=Bio.Phylo.BaseTree.Clade-class.html#__init__,Method Bio.Phylo.BaseTree.Tree.__init__()=Bio.Phylo.BaseTree.Tree-class.html#__init__,Method Bio.Phylo.CDAO.Clade.__init__()=Bio.Phylo.CDAO.Clade-class.html#__init__,Method Bio.Phylo.CDAO.Tree.__init__()=Bio.Phylo.CDAO.Tree-class.html#__init__,Method Bio.Phylo.CDAOIO.Parser.__init__()=Bio.Phylo.CDAOIO.Parser-class.html#__init__,Method Bio.Phylo.CDAOIO.Writer.__init__()=Bio.Phylo.CDAOIO.Writer-class.html#__init__,Method Bio.Phylo.NeXML.Clade.__init__()=Bio.Phylo.NeXML.Clade-class.html#__init__,Method Bio.Phylo.NeXML.Tree.__init__()=Bio.Phylo.NeXML.Tree-class.html#__init__,Method Bio.Phylo.NeXMLIO.Parser.__init__()=Bio.Phylo.NeXMLIO.Parser-class.html#__init__,Method Bio.Phylo.NeXMLIO.Writer.__init__()=Bio.Phylo.NeXMLIO.Writer-class.html#__init__,Method Bio.Phylo.Newick.Clade.__init__()=Bio.Phylo.Newick.Clade-class.html#__init__,Method Bio.Phylo.Newick.Tree.__init__()=Bio.Phylo.Newick.Tree-class.html#__init__,Method Bio.Phylo.NewickIO.Parser.__init__()=Bio.Phylo.NewickIO.Parser-class.html#__init__,Method Bio.Phylo.NewickIO.Writer.__init__()=Bio.Phylo.NewickIO.Writer-class.html#__init__,Method Bio.Phylo.PAML._paml.Paml.__init__()=Bio.Phylo.PAML._paml.Paml-class.html#__init__,Method Bio.Phylo.PAML.baseml.Baseml.__init__()=Bio.Phylo.PAML.baseml.Baseml-class.html#__init__,Method Bio.Phylo.PAML.codeml.Codeml.__init__()=Bio.Phylo.PAML.codeml.Codeml-class.html#__init__,Method Bio.Phylo.PAML.yn00.Yn00.__init__()=Bio.Phylo.PAML.yn00.Yn00-class.html#__init__,Method Bio.Phylo.PhyloXML.Accession.__init__()=Bio.Phylo.PhyloXML.Accession-class.html#__init__,Method Bio.Phylo.PhyloXML.Annotation.__init__()=Bio.Phylo.PhyloXML.Annotation-class.html#__init__,Method Bio.Phylo.PhyloXML.BinaryCharacters.__init__()=Bio.Phylo.PhyloXML.BinaryCharacters-class.html#__init__,Method Bio.Phylo.PhyloXML.BranchColor.__init__()=Bio.Phylo.PhyloXML.BranchColor-class.html#__init__,Method Bio.Phylo.PhyloXML.Clade.__init__()=Bio.Phylo.PhyloXML.Clade-class.html#__init__,Method Bio.Phylo.PhyloXML.CladeRelation.__init__()=Bio.Phylo.PhyloXML.CladeRelation-class.html#__init__,Method Bio.Phylo.PhyloXML.Confidence.__init__()=Bio.Phylo.PhyloXML.Confidence-class.html#__init__,Method Bio.Phylo.PhyloXML.Date.__init__()=Bio.Phylo.PhyloXML.Date-class.html#__init__,Method Bio.Phylo.PhyloXML.Distribution.__init__()=Bio.Phylo.PhyloXML.Distribution-class.html#__init__,Method Bio.Phylo.PhyloXML.DomainArchitecture.__init__()=Bio.Phylo.PhyloXML.DomainArchitecture-class.html#__init__,Method Bio.Phylo.PhyloXML.Events.__init__()=Bio.Phylo.PhyloXML.Events-class.html#__init__,Method Bio.Phylo.PhyloXML.Id.__init__()=Bio.Phylo.PhyloXML.Id-class.html#__init__,Method Bio.Phylo.PhyloXML.MolSeq.__init__()=Bio.Phylo.PhyloXML.MolSeq-class.html#__init__,Method Bio.Phylo.PhyloXML.Other.__init__()=Bio.Phylo.PhyloXML.Other-class.html#__init__,Method Bio.Phylo.PhyloXML.Phylogeny.__init__()=Bio.Phylo.PhyloXML.Phylogeny-class.html#__init__,Method Bio.Phylo.PhyloXML.Phyloxml.__init__()=Bio.Phylo.PhyloXML.Phyloxml-class.html#__init__,Method Bio.Phylo.PhyloXML.Point.__init__()=Bio.Phylo.PhyloXML.Point-class.html#__init__,Method Bio.Phylo.PhyloXML.Polygon.__init__()=Bio.Phylo.PhyloXML.Polygon-class.html#__init__,Method Bio.Phylo.PhyloXML.Property.__init__()=Bio.Phylo.PhyloXML.Property-class.html#__init__,Method Bio.Phylo.PhyloXML.ProteinDomain.__init__()=Bio.Phylo.PhyloXML.ProteinDomain-class.html#__init__,Method Bio.Phylo.PhyloXML.Reference.__init__()=Bio.Phylo.PhyloXML.Reference-class.html#__init__,Method Bio.Phylo.PhyloXML.Sequence.__init__()=Bio.Phylo.PhyloXML.Sequence-class.html#__init__,Method Bio.Phylo.PhyloXML.SequenceRelation.__init__()=Bio.Phylo.PhyloXML.SequenceRelation-class.html#__init__,Method Bio.Phylo.PhyloXML.Taxonomy.__init__()=Bio.Phylo.PhyloXML.Taxonomy-class.html#__init__,Method Bio.Phylo.PhyloXML.Uri.__init__()=Bio.Phylo.PhyloXML.Uri-class.html#__init__,Method Bio.Phylo.PhyloXMLIO.Parser.__init__()=Bio.Phylo.PhyloXMLIO.Parser-class.html#__init__,Method Bio.Phylo.PhyloXMLIO.Writer.__init__()=Bio.Phylo.PhyloXMLIO.Writer-class.html#__init__,Method Bio.Phylo.TreeConstruction.DistanceCalculator.__init__()=Bio.Phylo.TreeConstruction.DistanceCalculator-class.html#__init__,Method Bio.Phylo.TreeConstruction.DistanceTreeConstructor.__init__()=Bio.Phylo.TreeConstruction.DistanceTreeConstructor-class.html#__init__,Method Bio.Phylo.TreeConstruction.NNITreeSearcher.__init__()=Bio.Phylo.TreeConstruction.NNITreeSearcher-class.html#__init__,Method Bio.Phylo.TreeConstruction.ParsimonyScorer.__init__()=Bio.Phylo.TreeConstruction.ParsimonyScorer-class.html#__init__,Method Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor.__init__()=Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor-class.html#__init__,Method Bio.Phylo.TreeConstruction._DistanceMatrix.__init__()=Bio.Phylo.TreeConstruction._DistanceMatrix-class.html#__init__,Method Bio.Phylo.TreeConstruction._Matrix.__init__()=Bio.Phylo.TreeConstruction._Matrix-class.html#__init__,Method Bio.PopGen.Async.Async.__init__()=Bio.PopGen.Async.Async-class.html#__init__,Method Bio.PopGen.Async.DirectoryRetriever.__init__()=Bio.PopGen.Async.DirectoryRetriever-class.html#__init__,Method Bio.PopGen.Async.FileRetriever.__init__()=Bio.PopGen.Async.FileRetriever-class.html#__init__,Method Bio.PopGen.Async.Local.Local.__init__()=Bio.PopGen.Async.Local.Local-class.html#__init__,Method Bio.PopGen.FDist.Async.FDistAsync.__init__()=Bio.PopGen.FDist.Async.FDistAsync-class.html#__init__,Method Bio.PopGen.FDist.Async.SplitFDist.__init__()=Bio.PopGen.FDist.Async.SplitFDist-class.html#__init__,Method Bio.PopGen.FDist.Controller.FDistController.__init__()=Bio.PopGen.FDist.Controller.FDistController-class.html#__init__,Method Bio.PopGen.FDist.Record.__init__()=Bio.PopGen.FDist.Record-class.html#__init__,Method Bio.PopGen.GenePop.Controller.GenePopController.__init__()=Bio.PopGen.GenePop.Controller.GenePopController-class.html#__init__,Method Bio.PopGen.GenePop.Controller._FileIterator.__init__()=Bio.PopGen.GenePop.Controller._FileIterator-class.html#__init__,Method Bio.PopGen.GenePop.Controller._GenePopCommandline.__init__()=Bio.PopGen.GenePop.Controller._GenePopCommandline-class.html#__init__,Method Bio.PopGen.GenePop.EasyController.EasyController.__init__()=Bio.PopGen.GenePop.EasyController.EasyController-class.html#__init__,Method Bio.PopGen.GenePop.FileParser.FileRecord.__init__()=Bio.PopGen.GenePop.FileParser.FileRecord-class.html#__init__,Method Bio.PopGen.GenePop.LargeFileParser.Record.__init__()=Bio.PopGen.GenePop.LargeFileParser.Record-class.html#__init__,Method Bio.PopGen.GenePop.Record.__init__()=Bio.PopGen.GenePop.Record-class.html#__init__,Method Bio.PopGen.SimCoal.Async.SimCoalCache.__init__()=Bio.PopGen.SimCoal.Async.SimCoalCache-class.html#__init__,Method Bio.PopGen.SimCoal.Cache.SimCoalCache.__init__()=Bio.PopGen.SimCoal.Cache.SimCoalCache-class.html#__init__,Method Bio.PopGen.SimCoal.Controller.FastSimCoalController.__init__()=Bio.PopGen.SimCoal.Controller.FastSimCoalController-class.html#__init__,Method Bio.PopGen.SimCoal.Controller.SimCoalController.__init__()=Bio.PopGen.SimCoal.Controller.SimCoalController-class.html#__init__,Method Bio.PopGen.SimCoal.Controller._FastSimCoalCommandLine.__init__()=Bio.PopGen.SimCoal.Controller._FastSimCoalCommandLine-class.html#__init__,Method Bio.Restriction.PrintFormat.PrintFormat.__init__()=Bio.Restriction.PrintFormat.PrintFormat-class.html#__init__,Method Bio.Restriction.Restriction.Analysis.__init__()=Bio.Restriction.Restriction.Analysis-class.html#__init__,Method Bio.Restriction.Restriction.FormattedSeq.__init__()=Bio.Restriction.Restriction.FormattedSeq-class.html#__init__,Method Bio.Restriction.Restriction.RestrictionBatch.__init__()=Bio.Restriction.Restriction.RestrictionBatch-class.html#__init__,Method Bio.Restriction.Restriction.RestrictionType.__init__()=Bio.Restriction.Restriction.RestrictionType-class.html#__init__,Method Bio.SCOP.Astral.__init__()=Bio.SCOP.Astral-class.html#__init__,Method Bio.SCOP.Cla.Index.__init__()=Bio.SCOP.Cla.Index-class.html#__init__,Method Bio.SCOP.Cla.Record.__init__()=Bio.SCOP.Cla.Record-class.html#__init__,Method Bio.SCOP.Des.Record.__init__()=Bio.SCOP.Des.Record-class.html#__init__,Method Bio.SCOP.Dom.Record.__init__()=Bio.SCOP.Dom.Record-class.html#__init__,Method Bio.SCOP.Domain.__init__()=Bio.SCOP.Domain-class.html#__init__,Method Bio.SCOP.Hie.Record.__init__()=Bio.SCOP.Hie.Record-class.html#__init__,Method Bio.SCOP.Node.__init__()=Bio.SCOP.Node-class.html#__init__,Method Bio.SCOP.Raf.Res.__init__()=Bio.SCOP.Raf.Res-class.html#__init__,Method Bio.SCOP.Raf.SeqMap.__init__()=Bio.SCOP.Raf.SeqMap-class.html#__init__,Method Bio.SCOP.Raf.SeqMapIndex.__init__()=Bio.SCOP.Raf.SeqMapIndex-class.html#__init__,Method Bio.SCOP.Residues.Residues.__init__()=Bio.SCOP.Residues.Residues-class.html#__init__,Method Bio.SCOP.Scop.__init__()=Bio.SCOP.Scop-class.html#__init__,Method Bio.SVDSuperimposer.SVDSuperimposer.__init__()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabParser.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_text.BlastTextParser.__init__()=Bio.SearchIO.BlastIO.blast_text.BlastTextParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator.__init__()=Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslIndexer.__init__()=Bio.SearchIO.BlatIO.BlatPslIndexer-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslParser.__init__()=Bio.SearchIO.BlatIO.BlatPslParser-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslWriter.__init__()=Bio.SearchIO.BlatIO.BlatPslWriter-class.html#__init__,Method Bio.SearchIO.ExonerateIO._base._BaseExonerateParser.__init__()=Bio.SearchIO.ExonerateIO._base._BaseExonerateParser-class.html#__init__,Method Bio.SearchIO.FastaIO.FastaM10Indexer.__init__()=Bio.SearchIO.FastaIO.FastaM10Indexer-class.html#__init__,Method Bio.SearchIO.FastaIO.FastaM10Parser.__init__()=Bio.SearchIO.FastaIO.FastaM10Parser-class.html#__init__,Method Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer.__init__()=Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser.__init__()=Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.__init__()=Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser.__init__()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.__init__()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser.__init__()=Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html#__init__,Method Bio.SearchIO._index.SearchIndexer.__init__()=Bio.SearchIO._index.SearchIndexer-class.html#__init__,Method Bio.SearchIO._model.hit.Hit.__init__()=Bio.SearchIO._model.hit.Hit-class.html#__init__,Method Bio.SearchIO._model.hsp.HSP.__init__()=Bio.SearchIO._model.hsp.HSP-class.html#__init__,Method Bio.SearchIO._model.hsp.HSPFragment.__init__()=Bio.SearchIO._model.hsp.HSPFragment-class.html#__init__,Method Bio.SearchIO._model.query.QueryResult.__init__()=Bio.SearchIO._model.query.QueryResult-class.html#__init__,Method Bio.Seq.MutableSeq.__init__()=Bio.Seq.MutableSeq-class.html#__init__,Method Bio.Seq.Seq.__init__()=Bio.Seq.Seq-class.html#__init__,Method Bio.Seq.UnknownSeq.__init__()=Bio.Seq.UnknownSeq-class.html#__init__,Method Bio.SeqFeature.CompoundLocation.__init__()=Bio.SeqFeature.CompoundLocation-class.html#__init__,Method Bio.SeqFeature.FeatureLocation.__init__()=Bio.SeqFeature.FeatureLocation-class.html#__init__,Method Bio.SeqFeature.PositionGap.__init__()=Bio.SeqFeature.PositionGap-class.html#__init__,Method Bio.SeqFeature.Reference.__init__()=Bio.SeqFeature.Reference-class.html#__init__,Method Bio.SeqFeature.SeqFeature.__init__()=Bio.SeqFeature.SeqFeature-class.html#__init__,Method Bio.SeqIO.FastaIO.FastaWriter.__init__()=Bio.SeqIO.FastaIO.FastaWriter-class.html#__init__,Method Bio.SeqIO.Interfaces.SequenceIterator.__init__()=Bio.SeqIO.Interfaces.SequenceIterator-class.html#__init__,Method Bio.SeqIO.Interfaces.SequenceWriter.__init__()=Bio.SeqIO.Interfaces.SequenceWriter-class.html#__init__,Method Bio.SeqIO.Interfaces.SequentialSequenceWriter.__init__()=Bio.SeqIO.Interfaces.SequentialSequenceWriter-class.html#__init__,Method Bio.SeqIO.PhdIO.PhdWriter.__init__()=Bio.SeqIO.PhdIO.PhdWriter-class.html#__init__,Method Bio.SeqIO.QualityIO.QualPhredWriter.__init__()=Bio.SeqIO.QualityIO.QualPhredWriter-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.SeqXmlIterator.__init__()=Bio.SeqIO.SeqXmlIO.SeqXmlIterator-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.SeqXmlWriter.__init__()=Bio.SeqIO.SeqXmlIO.SeqXmlWriter-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.XMLRecordIterator.__init__()=Bio.SeqIO.SeqXmlIO.XMLRecordIterator-class.html#__init__,Method Bio.SeqIO.SffIO.SffWriter.__init__()=Bio.SeqIO.SffIO.SffWriter-class.html#__init__,Method Bio.SeqIO.SffIO._AddTellHandle.__init__()=Bio.SeqIO.SffIO._AddTellHandle-class.html#__init__,Method Bio.SeqIO.UniprotIO.Parser.__init__()=Bio.SeqIO.UniprotIO.Parser-class.html#__init__,Method Bio.SeqIO._index.IntelliGeneticsRandomAccess.__init__()=Bio.SeqIO._index.IntelliGeneticsRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SeqFileRandomAccess.__init__()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SequentialSeqFileRandomAccess.__init__()=Bio.SeqIO._index.SequentialSeqFileRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SffRandomAccess.__init__()=Bio.SeqIO._index.SffRandomAccess-class.html#__init__,Method Bio.SeqRecord.SeqRecord.__init__()=Bio.SeqRecord.SeqRecord-class.html#__init__,Method Bio.SeqRecord._RestrictedDict.__init__()=Bio.SeqRecord._RestrictedDict-class.html#__init__,Method Bio.SeqUtils.CodonUsage.CodonAdaptationIndex.__init__()=Bio.SeqUtils.CodonUsage.CodonAdaptationIndex-class.html#__init__,Method Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.__init__()=Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint-class.html#__init__,Method Bio.SeqUtils.ProtParam.ProteinAnalysis.__init__()=Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#__init__,Method Bio.Sequencing.Ace.ACEFileRecord.__init__()=Bio.Sequencing.Ace.ACEFileRecord-class.html#__init__,Method Bio.Sequencing.Ace.Contig.__init__()=Bio.Sequencing.Ace.Contig-class.html#__init__,Method Bio.Sequencing.Ace.Reads.__init__()=Bio.Sequencing.Ace.Reads-class.html#__init__,Method Bio.Sequencing.Ace.af.__init__()=Bio.Sequencing.Ace.af-class.html#__init__,Method Bio.Sequencing.Ace.bs.__init__()=Bio.Sequencing.Ace.bs-class.html#__init__,Method Bio.Sequencing.Ace.ct.__init__()=Bio.Sequencing.Ace.ct-class.html#__init__,Method Bio.Sequencing.Ace.ds.__init__()=Bio.Sequencing.Ace.ds-class.html#__init__,Method Bio.Sequencing.Ace.qa.__init__()=Bio.Sequencing.Ace.qa-class.html#__init__,Method Bio.Sequencing.Ace.rd.__init__()=Bio.Sequencing.Ace.rd-class.html#__init__,Method Bio.Sequencing.Ace.rt.__init__()=Bio.Sequencing.Ace.rt-class.html#__init__,Method Bio.Sequencing.Ace.wa.__init__()=Bio.Sequencing.Ace.wa-class.html#__init__,Method Bio.Sequencing.Ace.wr.__init__()=Bio.Sequencing.Ace.wr-class.html#__init__,Method Bio.Sequencing.Applications._Novoalign.NovoalignCommandline.__init__()=Bio.Sequencing.Applications._Novoalign.NovoalignCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaAlignCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaAlignCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaBwaswCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaBwaswCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaIndexCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaIndexCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaSampeCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaSampeCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaSamseCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaSamseCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsSortCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsSortCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline-class.html#__init__,Method Bio.Sequencing.Phd.Record.__init__()=Bio.Sequencing.Phd.Record-class.html#__init__,Method Bio.SubsMat.FreqTable.FreqTable.__init__()=Bio.SubsMat.FreqTable.FreqTable-class.html#__init__,Method Bio.SubsMat.SeqMat.__init__()=Bio.SubsMat.SeqMat-class.html#__init__,Method Bio.SwissProt.KeyWList.Record.__init__()=Bio.SwissProt.KeyWList.Record-class.html#__init__,Method Bio.SwissProt.Record.__init__()=Bio.SwissProt.Record-class.html#__init__,Method Bio.SwissProt.Reference.__init__()=Bio.SwissProt.Reference-class.html#__init__,Method Bio.UniGene.ProtsimLine.__init__()=Bio.UniGene.ProtsimLine-class.html#__init__,Method Bio.UniGene.Record.__init__()=Bio.UniGene.Record-class.html#__init__,Method Bio.UniGene.STSLine.__init__()=Bio.UniGene.STSLine-class.html#__init__,Method Bio.UniGene.SequenceLine.__init__()=Bio.UniGene.SequenceLine-class.html#__init__,Method Bio.Wise.dnal.Statistics.__init__()=Bio.Wise.dnal.Statistics-class.html#__init__,Method Bio.Wise.psw.AlignmentColumn.__init__()=Bio.Wise.psw.AlignmentColumn-class.html#__init__,Method Bio.Wise.psw.ColumnUnit.__init__()=Bio.Wise.psw.ColumnUnit-class.html#__init__,Method Bio._py3k._ordereddict.OrderedDict.__init__()=Bio._py3k._ordereddict.OrderedDict-class.html#__init__,Method Bio.bgzf.BgzfReader.__init__()=Bio.bgzf.BgzfReader-class.html#__init__,Method Bio.bgzf.BgzfWriter.__init__()=Bio.bgzf.BgzfWriter-class.html#__init__,Method Bio.codonalign.codonalignment.CodonAlignment.__init__()=Bio.codonalign.codonalignment.CodonAlignment-class.html#__init__,Method Bio.codonalign.codonseq.CodonSeq.__init__()=Bio.codonalign.codonseq.CodonSeq-class.html#__init__,Method Bio.kNN.kNN.__init__()=Bio.kNN.kNN-class.html#__init__,Method Bio.motifs.Instances.__init__()=Bio.motifs.Instances-class.html#__init__,Method Bio.motifs.Motif.__init__()=Bio.motifs.Motif-class.html#__init__,Method Bio.motifs.alignace.Record.__init__()=Bio.motifs.alignace.Record-class.html#__init__,Method Bio.motifs.applications._alignace.AlignAceCommandline.__init__()=Bio.motifs.applications._alignace.AlignAceCommandline-class.html#__init__,Method Bio.motifs.applications._alignace.CompareAceCommandline.__init__()=Bio.motifs.applications._alignace.CompareAceCommandline-class.html#__init__,Method Bio.motifs.applications._xxmotif.XXmotifCommandline.__init__()=Bio.motifs.applications._xxmotif.XXmotifCommandline-class.html#__init__,Method Bio.motifs.jaspar.Motif.__init__()=Bio.motifs.jaspar.Motif-class.html#__init__,Method Bio.motifs.jaspar.Record.__init__()=Bio.motifs.jaspar.Record-class.html#__init__,Method Bio.motifs.jaspar.db.JASPAR5.__init__()=Bio.motifs.jaspar.db.JASPAR5-class.html#__init__,Method Bio.motifs.mast.Record.__init__()=Bio.motifs.mast.Record-class.html#__init__,Method Bio.motifs.matrix.GenericPositionMatrix.__init__()=Bio.motifs.matrix.GenericPositionMatrix-class.html#__init__,Method Bio.motifs.matrix.PositionWeightMatrix.__init__()=Bio.motifs.matrix.PositionWeightMatrix-class.html#__init__,Method Bio.motifs.meme.Instance.__init__()=Bio.motifs.meme.Instance-class.html#__init__,Method Bio.motifs.meme.Motif.__init__()=Bio.motifs.meme.Motif-class.html#__init__,Method Bio.motifs.meme.Record.__init__()=Bio.motifs.meme.Record-class.html#__init__,Method Bio.motifs.thresholds.ScoreDistribution.__init__()=Bio.motifs.thresholds.ScoreDistribution-class.html#__init__,Method Bio.motifs.transfac.Record.__init__()=Bio.motifs.transfac.Record-class.html#__init__,Method Bio.pairwise2.affine_penalty.__init__()=Bio.pairwise2.affine_penalty-class.html#__init__,Method Bio.pairwise2.dictionary_match.__init__()=Bio.pairwise2.dictionary_match-class.html#__init__,Method Bio.pairwise2.identity_match.__init__()=Bio.pairwise2.identity_match-class.html#__init__,Method BioSQL.BioSeq.DBSeq.__init__()=BioSQL.BioSeq.DBSeq-class.html#__init__,Method BioSQL.BioSeq.DBSeqRecord.__init__()=BioSQL.BioSeq.DBSeqRecord-class.html#__init__,Method BioSQL.BioSeqDatabase.Adaptor.__init__()=BioSQL.BioSeqDatabase.Adaptor-class.html#__init__,Method BioSQL.BioSeqDatabase.BioSeqDatabase.__init__()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#__init__,Method BioSQL.BioSeqDatabase.DBServer.__init__()=BioSQL.BioSeqDatabase.DBServer-class.html#__init__,Method BioSQL.BioSeqDatabase._CursorWrapper.__init__()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#__init__,Method BioSQL.DBUtils.Generic_dbutils.__init__()=BioSQL.DBUtils.Generic_dbutils-class.html#__init__,Method BioSQL.Loader.DatabaseLoader.__init__()=BioSQL.Loader.DatabaseLoader-class.html#__init__,Method BioSQL.Loader.DatabaseRemover.__init__()=BioSQL.Loader.DatabaseRemover-class.html#__init__"><a title="Bio.Affy.CelFile.Record.__init__
Bio.Align.AlignInfo.PSSM.__init__
Bio.Align.AlignInfo.SummaryInfo.__init__
Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline.__init__
Bio.Align.Applications._Clustalw.ClustalwCommandline.__init__
Bio.Align.Applications._Dialign.DialignCommandline.__init__
Bio.Align.Applications._MSAProbs.MSAProbsCommandline.__init__
Bio.Align.Applications._Mafft.MafftCommandline.__init__
Bio.Align.Applications._Muscle.MuscleCommandline.__init__
Bio.Align.Applications._Prank.PrankCommandline.__init__
Bio.Align.Applications._Probcons.ProbconsCommandline.__init__
Bio.Align.Applications._TCoffee.TCoffeeCommandline.__init__
Bio.Align.Generic.Alignment.__init__
Bio.Align.MultipleSeqAlignment.__init__
Bio.AlignIO.Interfaces.AlignmentIterator.__init__
Bio.AlignIO.Interfaces.AlignmentWriter.__init__
Bio.AlignIO.Interfaces.SequentialAlignmentWriter.__init__
Bio.Alphabet.AlphabetEncoder.__init__
Bio.Alphabet.Gapped.__init__
Bio.Alphabet.HasStopCodon.__init__
Bio.Application.AbstractCommandline.__init__
Bio.Application.ApplicationError.__init__
Bio.Application._AbstractParameter.__init__
Bio.Application._Argument.__init__
Bio.Application._Option.__init__
Bio.Application._StaticArgument.__init__
Bio.Application._Switch.__init__
Bio.Blast.Applications.NcbiblastformatterCommandline.__init__
Bio.Blast.Applications.NcbiblastnCommandline.__init__
Bio.Blast.Applications.NcbiblastpCommandline.__init__
Bio.Blast.Applications.NcbiblastxCommandline.__init__
Bio.Blast.Applications.NcbideltablastCommandline.__init__
Bio.Blast.Applications.NcbipsiblastCommandline.__init__
Bio.Blast.Applications.NcbirpsblastCommandline.__init__
Bio.Blast.Applications.NcbirpstblastnCommandline.__init__
Bio.Blast.Applications.NcbitblastnCommandline.__init__
Bio.Blast.Applications.NcbitblastxCommandline.__init__
Bio.Blast.Applications._NcbibaseblastCommandline.__init__
Bio.Blast.Applications._Ncbiblast2SeqCommandline.__init__
Bio.Blast.Applications._NcbiblastCommandline.__init__
Bio.Blast.Applications._NcbiblastMain2SeqCommandline.__init__
Bio.Blast.NCBIStandalone.BlastErrorParser.__init__
Bio.Blast.NCBIStandalone.BlastParser.__init__
Bio.Blast.NCBIStandalone.Iterator.__init__
Bio.Blast.NCBIStandalone.PSIBlastParser.__init__
Bio.Blast.NCBIStandalone._BlastConsumer.__init__
Bio.Blast.NCBIStandalone._BlastErrorConsumer.__init__
Bio.Blast.NCBIStandalone._PSIBlastConsumer.__init__
Bio.Blast.NCBIXML.BlastParser.__init__
Bio.Blast.NCBIXML._XMLparser.__init__
Bio.Blast.ParseBlastTable.BlastTableEntry.__init__
Bio.Blast.ParseBlastTable.BlastTableReader.__init__
Bio.Blast.ParseBlastTable.BlastTableRec.__init__
Bio.Blast.Record.Alignment.__init__
Bio.Blast.Record.Blast.__init__
Bio.Blast.Record.DatabaseReport.__init__
Bio.Blast.Record.Description.__init__
Bio.Blast.Record.HSP.__init__
Bio.Blast.Record.Header.__init__
Bio.Blast.Record.MultipleAlignment.__init__
Bio.Blast.Record.PSIBlast.__init__
Bio.Blast.Record.Parameters.__init__
Bio.Blast.Record.Round.__init__
Bio.CAPS.CAPSMap.__init__
Bio.CAPS.DifferentialCutsite.__init__
Bio.Cluster.Record.__init__
Bio.Compass.Record.__init__
Bio.Crystal.Chain.__init__
Bio.Crystal.Crystal.__init__
Bio.Crystal.Hetero.__init__
Bio.Data.CodonTable.AmbiguousCodonTable.__init__
Bio.Data.CodonTable.AmbiguousForwardTable.__init__
Bio.Data.CodonTable.CodonTable.__init__
Bio.Data.CodonTable.NCBICodonTable.__init__
Bio.DocSQL.Create.__init__
Bio.DocSQL.Insert.__init__
Bio.DocSQL.IterationCursor.__init__
Bio.DocSQL.Query.__init__
Bio.DocSQL.QueryAll.__init__
Bio.DocSQL.QueryGeneric.__init__
Bio.DocSQL.QueryRow.__init__
Bio.DocSQL.QuerySingle.__init__
Bio.Emboss.Applications.DiffseqCommandline.__init__
Bio.Emboss.Applications.EInvertedCommandline.__init__
Bio.Emboss.Applications.ETandemCommandline.__init__
Bio.Emboss.Applications.Est2GenomeCommandline.__init__
Bio.Emboss.Applications.FConsenseCommandline.__init__
Bio.Emboss.Applications.FDNADistCommandline.__init__
Bio.Emboss.Applications.FDNAParsCommandline.__init__
Bio.Emboss.Applications.FNeighborCommandline.__init__
Bio.Emboss.Applications.FProtDistCommandline.__init__
Bio.Emboss.Applications.FProtParsCommandline.__init__
Bio.Emboss.Applications.FSeqBootCommandline.__init__
Bio.Emboss.Applications.FTreeDistCommandline.__init__
Bio.Emboss.Applications.FuzznucCommandline.__init__
Bio.Emboss.Applications.IepCommandline.__init__
Bio.Emboss.Applications.NeedleCommandline.__init__
Bio.Emboss.Applications.NeedleallCommandline.__init__
Bio.Emboss.Applications.PalindromeCommandline.__init__
Bio.Emboss.Applications.Primer3Commandline.__init__
Bio.Emboss.Applications.PrimerSearchCommandline.__init__
Bio.Emboss.Applications.SeqmatchallCommandline.__init__
Bio.Emboss.Applications.SeqretCommandline.__init__
Bio.Emboss.Applications.StretcherCommandline.__init__
Bio.Emboss.Applications.TranalignCommandline.__init__
Bio.Emboss.Applications.WaterCommandline.__init__
Bio.Emboss.Applications._EmbossCommandLine.__init__
Bio.Emboss.Applications._EmbossMinimalCommandLine.__init__
Bio.Emboss.Primer3.Primers.__init__
Bio.Emboss.Primer3.Record.__init__
Bio.Emboss.PrimerSearch.Amplifier.__init__
Bio.Emboss.PrimerSearch.InputRecord.__init__
Bio.Emboss.PrimerSearch.OutputRecord.__init__
Bio.Entrez.Parser.CorruptedXMLError.__init__
Bio.Entrez.Parser.DataHandler.__init__
Bio.Entrez.Parser.NotXMLError.__init__
Bio.Entrez.Parser.StructureElement.__init__
Bio.Entrez.Parser.ValidationError.__init__
Bio.ExPASy.Enzyme.Record.__init__
Bio.ExPASy.Prodoc.Record.__init__
Bio.ExPASy.Prodoc.Reference.__init__
Bio.ExPASy.Prosite.Record.__init__
Bio.ExPASy.ScanProsite.ContentHandler.__init__
Bio.ExPASy.ScanProsite.Parser.__init__
Bio.ExPASy.ScanProsite.Record.__init__
Bio.FSSP.FSSPAlignDict.__init__
Bio.FSSP.FSSPAlignRec.__init__
Bio.FSSP.FSSPHeader.__init__
Bio.FSSP.FSSPSumRec.__init__
Bio.FSSP.FSSPTools.FSSPMultAlign.__init__
Bio.FSSP.PosAlign.__init__
Bio.FSSP.fssp_rec.fff_rec.__init__
Bio.File.UndoHandle.__init__
Bio.File._IndexedSeqFileDict.__init__
Bio.File._SQLiteManySeqFilesDict.__init__
Bio.GA.Crossover.General.SafeFitnessCrossover.__init__
Bio.GA.Crossover.GeneralPoint.GeneralPointCrossover.__init__
Bio.GA.Crossover.GeneralPoint.InterleaveCrossover.__init__
Bio.GA.Crossover.Point.SinglePointCrossover.__init__
Bio.GA.Crossover.TwoPoint.TwoPointCrossover.__init__
Bio.GA.Crossover.Uniform.UniformCrossover.__init__
Bio.GA.Evolver.GenerationEvolver.__init__
Bio.GA.Evolver.SteadyStateEvolver.__init__
Bio.GA.Mutation.General.SafeFitnessMutation.__init__
Bio.GA.Mutation.Simple.ConversionMutation.__init__
Bio.GA.Mutation.Simple.SinglePositionMutation.__init__
Bio.GA.Organism.Organism.__init__
Bio.GA.Repair.Stabilizing.AmbiguousRepair.__init__
Bio.GA.Selection.Abstract.AbstractSelection.__init__
Bio.GA.Selection.Diversity.DiversitySelection.__init__
Bio.GA.Selection.RouletteWheel.RouletteWheelSelection.__init__
Bio.GA.Selection.Tournament.TournamentSelection.__init__
Bio.GenBank.FeatureParser.__init__
Bio.GenBank.Iterator.__init__
Bio.GenBank.Record.Feature.__init__
Bio.GenBank.Record.Qualifier.__init__
Bio.GenBank.Record.Record.__init__
Bio.GenBank.Record.Reference.__init__
Bio.GenBank.RecordParser.__init__
Bio.GenBank.Scanner.InsdcScanner.__init__
Bio.GenBank._BaseGenBankConsumer.__init__
Bio.GenBank._FeatureConsumer.__init__
Bio.GenBank._RecordConsumer.__init__
Bio.GenBank.utils.FeatureValueCleaner.__init__
Bio.Geo.Record.Record.__init__
Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment.__init__
Bio.Graphics.BasicChromosome.Chromosome.__init__
Bio.Graphics.BasicChromosome.ChromosomeSegment.__init__
Bio.Graphics.BasicChromosome.Organism.__init__
Bio.Graphics.BasicChromosome.TelomereSegment.__init__
Bio.Graphics.BasicChromosome._ChromosomeComponent.__init__
Bio.Graphics.ColorSpiral.ColorSpiral.__init__
Bio.Graphics.Comparative.ComparativeScatterPlot.__init__
Bio.Graphics.DisplayRepresentation.ChromosomeCounts.__init__
Bio.Graphics.Distribution.BarChartDistribution.__init__
Bio.Graphics.Distribution.DistributionPage.__init__
Bio.Graphics.Distribution.LineDistribution.__init__
Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.__init__
Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.__init__
Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.__init__
Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.__init__
Bio.Graphics.GenomeDiagram._Diagram.Diagram.__init__
Bio.Graphics.GenomeDiagram._Feature.Feature.__init__
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.__init__
Bio.Graphics.GenomeDiagram._Graph.GraphData.__init__
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.__init__
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.__init__
Bio.Graphics.GenomeDiagram._Track.Track.__init__
Bio.Graphics.KGML_vis.KGMLCanvas.__init__
Bio.HMM.DynamicProgramming.AbstractDPAlgorithms.__init__
Bio.HMM.DynamicProgramming.LogDPAlgorithms.__init__
Bio.HMM.DynamicProgramming.ScaledDPAlgorithms.__init__
Bio.HMM.MarkovModel.HiddenMarkovModel.__init__
Bio.HMM.MarkovModel.MarkovModelBuilder.__init__
Bio.HMM.Trainer.AbstractTrainer.__init__
Bio.HMM.Trainer.BaumWelchTrainer.__init__
Bio.HMM.Trainer.KnownStateTrainer.__init__
Bio.HMM.Trainer.TrainingSequence.__init__
Bio.Index._InMemoryIndex.__init__
Bio.Index._ShelveIndex.__init__
Bio.KDTree.KDTree'.KDTree.__init__
Bio.KEGG.Compound.Record.__init__
Bio.KEGG.Enzyme.Record.__init__
Bio.KEGG.KGML.KGML_parser.KGMLParser.__init__
Bio.KEGG.KGML.KGML_pathway.Component.__init__
Bio.KEGG.KGML.KGML_pathway.Entry.__init__
Bio.KEGG.KGML.KGML_pathway.Graphics.__init__
Bio.KEGG.KGML.KGML_pathway.Pathway.__init__
Bio.KEGG.KGML.KGML_pathway.Reaction.__init__
Bio.KEGG.KGML.KGML_pathway.Relation.__init__
Bio.LogisticRegression.LogisticRegression.__init__
Bio.MarkovModel.MarkovModel.__init__
Bio.MaxEntropy.MaxEntropy.__init__
Bio.Motif.Parsers.AlignAce.Record.__init__
Bio.Motif.Parsers.MAST.Record.__init__
Bio.Motif.Parsers.MEME.MEMEInstance.__init__
Bio.Motif.Parsers.MEME.MEMEMotif.__init__
Bio.Motif.Parsers.MEME.MEMERecord.__init__
Bio.Motif.Thresholds.ScoreDistribution.__init__
Bio.Motif._Motif.Motif.__init__
Bio.NMR.xpktools.Peaklist.__init__
Bio.NMR.xpktools.XpkEntry.__init__
Bio.NaiveBayes.NaiveBayes.__init__
Bio.NeuralNetwork.BackPropagation.Layer.AbstractLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.__init__
Bio.NeuralNetwork.BackPropagation.Network.BasicNetwork.__init__
Bio.NeuralNetwork.Gene.Motif.MotifCoder.__init__
Bio.NeuralNetwork.Gene.Motif.MotifFinder.__init__
Bio.NeuralNetwork.Gene.Pattern.PatternIO.__init__
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.__init__
Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness.__init__
Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder.__init__
Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness.__init__
Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator.__init__
Bio.NeuralNetwork.Gene.Schema.Schema.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaCoder.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaFactory.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaFinder.__init__
Bio.NeuralNetwork.Gene.Schema.SimpleFinisher.__init__
Bio.NeuralNetwork.Gene.Signature.SignatureCoder.__init__
Bio.NeuralNetwork.Gene.Signature.SignatureFinder.__init__
Bio.NeuralNetwork.StopTraining.ValidationIncreaseStop.__init__
Bio.NeuralNetwork.Training.ExampleManager.__init__
Bio.NeuralNetwork.Training.TrainingExample.__init__
Bio.Nexus.Nexus.Block.__init__
Bio.Nexus.Nexus.CharBuffer.__init__
Bio.Nexus.Nexus.Commandline.__init__
Bio.Nexus.Nexus.Nexus.__init__
Bio.Nexus.Nexus.StepMatrix.__init__
Bio.Nexus.Nodes.Chain.__init__
Bio.Nexus.Nodes.Node.__init__
Bio.Nexus.Trees.NodeData.__init__
Bio.Nexus.Trees.Tree.__init__
Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap.__init__
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__init__
Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap.__init__
Bio.PDB.Atom.Atom.__init__
Bio.PDB.Atom.DisorderedAtom.__init__
Bio.PDB.Chain.Chain.__init__
Bio.PDB.DSSP'.DSSP.__init__
Bio.PDB.Dice.ChainSelector.__init__
Bio.PDB.Entity.DisorderedEntityWrapper.__init__
Bio.PDB.Entity.Entity.__init__
Bio.PDB.FragmentMapper'.Fragment.__init__
Bio.PDB.FragmentMapper'.FragmentMapper.__init__
Bio.PDB.HSExposure.ExposureCN.__init__
Bio.PDB.HSExposure.HSExposureCA.__init__
Bio.PDB.HSExposure.HSExposureCB.__init__
Bio.PDB.HSExposure._AbstractHSExposure.__init__
Bio.PDB.MMCIF2Dict.MMCIF2Dict.__init__
Bio.PDB.MMCIFParser'.MMCIFParser.__init__
Bio.PDB.Model.Model.__init__
Bio.PDB.NACCESS.NACCESS.__init__
Bio.PDB.NACCESS.NACCESS_atomic.__init__
Bio.PDB.NeighborSearch'.NeighborSearch.__init__
Bio.PDB.PDBIO'.PDBIO.__init__
Bio.PDB.PDBList'.PDBList.__init__
Bio.PDB.PDBParser'.PDBParser.__init__
Bio.PDB.PSEA.PSEA.__init__
Bio.PDB.Polypeptide.CaPPBuilder.__init__
Bio.PDB.Polypeptide.PPBuilder.__init__
Bio.PDB.Polypeptide._PPBuilder.__init__
Bio.PDB.QCPSuperimposer.QCPSuperimposer.__init__
Bio.PDB.Residue.DisorderedResidue.__init__
Bio.PDB.Residue.Residue.__init__
Bio.PDB.ResidueDepth'.ResidueDepth.__init__
Bio.PDB.Structure.Structure.__init__
Bio.PDB.StructureAlignment'.StructureAlignment.__init__
Bio.PDB.StructureBuilder.StructureBuilder.__init__
Bio.PDB.Superimposer'.Superimposer.__init__
Bio.PDB.Vector'.Vector.__init__
Bio.ParserSupport.EventGenerator.__init__
Bio.ParserSupport.TaggingConsumer.__init__
Bio.Pathway.Network.__init__
Bio.Pathway.Reaction.__init__
Bio.Pathway.Rep.Graph.Graph.__init__
Bio.Pathway.Rep.MultiGraph.MultiGraph.__init__
Bio.Pathway.System.__init__
Bio.Phylo.Applications._Fasttree.FastTreeCommandline.__init__
Bio.Phylo.Applications._Phyml.PhymlCommandline.__init__
Bio.Phylo.Applications._Raxml.RaxmlCommandline.__init__
Bio.Phylo.BaseTree.BranchColor.__init__
Bio.Phylo.BaseTree.Clade.__init__
Bio.Phylo.BaseTree.Tree.__init__
Bio.Phylo.CDAO.Clade.__init__
Bio.Phylo.CDAO.Tree.__init__
Bio.Phylo.CDAOIO.Parser.__init__
Bio.Phylo.CDAOIO.Writer.__init__
Bio.Phylo.NeXML.Clade.__init__
Bio.Phylo.NeXML.Tree.__init__
Bio.Phylo.NeXMLIO.Parser.__init__
Bio.Phylo.NeXMLIO.Writer.__init__
Bio.Phylo.Newick.Clade.__init__
Bio.Phylo.Newick.Tree.__init__
Bio.Phylo.NewickIO.Parser.__init__
Bio.Phylo.NewickIO.Writer.__init__
Bio.Phylo.PAML._paml.Paml.__init__
Bio.Phylo.PAML.baseml.Baseml.__init__
Bio.Phylo.PAML.codeml.Codeml.__init__
Bio.Phylo.PAML.yn00.Yn00.__init__
Bio.Phylo.PhyloXML.Accession.__init__
Bio.Phylo.PhyloXML.Annotation.__init__
Bio.Phylo.PhyloXML.BinaryCharacters.__init__
Bio.Phylo.PhyloXML.BranchColor.__init__
Bio.Phylo.PhyloXML.Clade.__init__
Bio.Phylo.PhyloXML.CladeRelation.__init__
Bio.Phylo.PhyloXML.Confidence.__init__
Bio.Phylo.PhyloXML.Date.__init__
Bio.Phylo.PhyloXML.Distribution.__init__
Bio.Phylo.PhyloXML.DomainArchitecture.__init__
Bio.Phylo.PhyloXML.Events.__init__
Bio.Phylo.PhyloXML.Id.__init__
Bio.Phylo.PhyloXML.MolSeq.__init__
Bio.Phylo.PhyloXML.Other.__init__
Bio.Phylo.PhyloXML.Phylogeny.__init__
Bio.Phylo.PhyloXML.Phyloxml.__init__
Bio.Phylo.PhyloXML.Point.__init__
Bio.Phylo.PhyloXML.Polygon.__init__
Bio.Phylo.PhyloXML.Property.__init__
Bio.Phylo.PhyloXML.ProteinDomain.__init__
Bio.Phylo.PhyloXML.Reference.__init__
Bio.Phylo.PhyloXML.Sequence.__init__
Bio.Phylo.PhyloXML.SequenceRelation.__init__
Bio.Phylo.PhyloXML.Taxonomy.__init__
Bio.Phylo.PhyloXML.Uri.__init__
Bio.Phylo.PhyloXMLIO.Parser.__init__
Bio.Phylo.PhyloXMLIO.Writer.__init__
Bio.Phylo.TreeConstruction.DistanceCalculator.__init__
Bio.Phylo.TreeConstruction.DistanceTreeConstructor.__init__
Bio.Phylo.TreeConstruction.NNITreeSearcher.__init__
Bio.Phylo.TreeConstruction.ParsimonyScorer.__init__
Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor.__init__
Bio.Phylo.TreeConstruction._DistanceMatrix.__init__
Bio.Phylo.TreeConstruction._Matrix.__init__
Bio.PopGen.Async.Async.__init__
Bio.PopGen.Async.DirectoryRetriever.__init__
Bio.PopGen.Async.FileRetriever.__init__
Bio.PopGen.Async.Local.Local.__init__
Bio.PopGen.FDist.Async.FDistAsync.__init__
Bio.PopGen.FDist.Async.SplitFDist.__init__
Bio.PopGen.FDist.Controller.FDistController.__init__
Bio.PopGen.FDist.Record.__init__
Bio.PopGen.GenePop.Controller.GenePopController.__init__
Bio.PopGen.GenePop.Controller._FileIterator.__init__
Bio.PopGen.GenePop.Controller._GenePopCommandline.__init__
Bio.PopGen.GenePop.EasyController.EasyController.__init__
Bio.PopGen.GenePop.FileParser.FileRecord.__init__
Bio.PopGen.GenePop.LargeFileParser.Record.__init__
Bio.PopGen.GenePop.Record.__init__
Bio.PopGen.SimCoal.Async.SimCoalCache.__init__
Bio.PopGen.SimCoal.Cache.SimCoalCache.__init__
Bio.PopGen.SimCoal.Controller.FastSimCoalController.__init__
Bio.PopGen.SimCoal.Controller.SimCoalController.__init__
Bio.PopGen.SimCoal.Controller._FastSimCoalCommandLine.__init__
Bio.Restriction.PrintFormat.PrintFormat.__init__
Bio.Restriction.Restriction.Analysis.__init__
Bio.Restriction.Restriction.FormattedSeq.__init__
Bio.Restriction.Restriction.RestrictionBatch.__init__
Bio.Restriction.Restriction.RestrictionType.__init__
Bio.SCOP.Astral.__init__
Bio.SCOP.Cla.Index.__init__
Bio.SCOP.Cla.Record.__init__
Bio.SCOP.Des.Record.__init__
Bio.SCOP.Dom.Record.__init__
Bio.SCOP.Domain.__init__
Bio.SCOP.Hie.Record.__init__
Bio.SCOP.Node.__init__
Bio.SCOP.Raf.Res.__init__
Bio.SCOP.Raf.SeqMap.__init__
Bio.SCOP.Raf.SeqMapIndex.__init__
Bio.SCOP.Residues.Residues.__init__
Bio.SCOP.Scop.__init__
Bio.SVDSuperimposer.SVDSuperimposer.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabParser.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.__init__
Bio.SearchIO.BlastIO.blast_text.BlastTextParser.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.__init__
Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator.__init__
Bio.SearchIO.BlatIO.BlatPslIndexer.__init__
Bio.SearchIO.BlatIO.BlatPslParser.__init__
Bio.SearchIO.BlatIO.BlatPslWriter.__init__
Bio.SearchIO.ExonerateIO._base._BaseExonerateParser.__init__
Bio.SearchIO.FastaIO.FastaM10Indexer.__init__
Bio.SearchIO.FastaIO.FastaM10Parser.__init__
Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer.__init__
Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser.__init__
Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.__init__
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser.__init__
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.__init__
Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser.__init__
Bio.SearchIO._index.SearchIndexer.__init__
Bio.SearchIO._model.hit.Hit.__init__
Bio.SearchIO._model.hsp.HSP.__init__
Bio.SearchIO._model.hsp.HSPFragment.__init__
Bio.SearchIO._model.query.QueryResult.__init__
Bio.Seq.MutableSeq.__init__
Bio.Seq.Seq.__init__
Bio.Seq.UnknownSeq.__init__
Bio.SeqFeature.CompoundLocation.__init__
Bio.SeqFeature.FeatureLocation.__init__
Bio.SeqFeature.PositionGap.__init__
Bio.SeqFeature.Reference.__init__
Bio.SeqFeature.SeqFeature.__init__
Bio.SeqIO.FastaIO.FastaWriter.__init__
Bio.SeqIO.Interfaces.SequenceIterator.__init__
Bio.SeqIO.Interfaces.SequenceWriter.__init__
Bio.SeqIO.Interfaces.SequentialSequenceWriter.__init__
Bio.SeqIO.PhdIO.PhdWriter.__init__
Bio.SeqIO.QualityIO.QualPhredWriter.__init__
Bio.SeqIO.SeqXmlIO.SeqXmlIterator.__init__
Bio.SeqIO.SeqXmlIO.SeqXmlWriter.__init__
Bio.SeqIO.SeqXmlIO.XMLRecordIterator.__init__
Bio.SeqIO.SffIO.SffWriter.__init__
Bio.SeqIO.SffIO._AddTellHandle.__init__
Bio.SeqIO.UniprotIO.Parser.__init__
Bio.SeqIO._index.IntelliGeneticsRandomAccess.__init__
Bio.SeqIO._index.SeqFileRandomAccess.__init__
Bio.SeqIO._index.SequentialSeqFileRandomAccess.__init__
Bio.SeqIO._index.SffRandomAccess.__init__
Bio.SeqRecord.SeqRecord.__init__
Bio.SeqRecord._RestrictedDict.__init__
Bio.SeqUtils.CodonUsage.CodonAdaptationIndex.__init__
Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.__init__
Bio.SeqUtils.ProtParam.ProteinAnalysis.__init__
Bio.Sequencing.Ace.ACEFileRecord.__init__
Bio.Sequencing.Ace.Contig.__init__
Bio.Sequencing.Ace.Reads.__init__
Bio.Sequencing.Ace.af.__init__
Bio.Sequencing.Ace.bs.__init__
Bio.Sequencing.Ace.ct.__init__
Bio.Sequencing.Ace.ds.__init__
Bio.Sequencing.Ace.qa.__init__
Bio.Sequencing.Ace.rd.__init__
Bio.Sequencing.Ace.rt.__init__
Bio.Sequencing.Ace.wa.__init__
Bio.Sequencing.Ace.wr.__init__
Bio.Sequencing.Applications._Novoalign.NovoalignCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaAlignCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaBwaswCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaIndexCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaSampeCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaSamseCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsSortCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline.__init__
Bio.Sequencing.Phd.Record.__init__
Bio.SubsMat.FreqTable.FreqTable.__init__
Bio.SubsMat.SeqMat.__init__
Bio.SwissProt.KeyWList.Record.__init__
Bio.SwissProt.Record.__init__
Bio.SwissProt.Reference.__init__
Bio.UniGene.ProtsimLine.__init__
Bio.UniGene.Record.__init__
Bio.UniGene.STSLine.__init__
Bio.UniGene.SequenceLine.__init__
Bio.Wise.dnal.Statistics.__init__
Bio.Wise.psw.AlignmentColumn.__init__
Bio.Wise.psw.ColumnUnit.__init__
Bio._py3k._ordereddict.OrderedDict.__init__
Bio.bgzf.BgzfReader.__init__
Bio.bgzf.BgzfWriter.__init__
Bio.codonalign.codonalignment.CodonAlignment.__init__
Bio.codonalign.codonseq.CodonSeq.__init__
Bio.kNN.kNN.__init__
Bio.motifs.Instances.__init__
Bio.motifs.Motif.__init__
Bio.motifs.alignace.Record.__init__
Bio.motifs.applications._alignace.AlignAceCommandline.__init__
Bio.motifs.applications._alignace.CompareAceCommandline.__init__
Bio.motifs.applications._xxmotif.XXmotifCommandline.__init__
Bio.motifs.jaspar.Motif.__init__
Bio.motifs.jaspar.Record.__init__
Bio.motifs.jaspar.db.JASPAR5.__init__
Bio.motifs.mast.Record.__init__
Bio.motifs.matrix.GenericPositionMatrix.__init__
Bio.motifs.matrix.PositionWeightMatrix.__init__
Bio.motifs.meme.Instance.__init__
Bio.motifs.meme.Motif.__init__
Bio.motifs.meme.Record.__init__
Bio.motifs.thresholds.ScoreDistribution.__init__
Bio.motifs.transfac.Record.__init__
Bio.pairwise2.affine_penalty.__init__
Bio.pairwise2.dictionary_match.__init__
Bio.pairwise2.identity_match.__init__
BioSQL.BioSeq.DBSeq.__init__
BioSQL.BioSeq.DBSeqRecord.__init__
BioSQL.BioSeqDatabase.Adaptor.__init__
BioSQL.BioSeqDatabase.BioSeqDatabase.__init__
BioSQL.BioSeqDatabase.DBServer.__init__
BioSQL.BioSeqDatabase._CursorWrapper.__init__
BioSQL.DBUtils.Generic_dbutils.__init__
BioSQL.Loader.DatabaseLoader.__init__
BioSQL.Loader.DatabaseRemover.__init__" class="py-name" href="#" onclick="return doclink('link-264', '__init__', 'link-264');">__init__</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
<a name="L624"></a><tt class="py-lineno"> 624</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-265" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-265', 'data', 'link-223');">data</a></tt> <tt class="py-op">=</tt> <tt id="link-266" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-266', 'SeqRecord', 'link-261');">SeqRecord</a></tt><tt class="py-op">(</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt id="link-267" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.id=Bio.KEGG.KGML.KGML_pathway.Component-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Entry.id=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.id=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#id,Method Bio.Phylo.PhyloXMLIO.Parser.id()=Bio.Phylo.PhyloXMLIO.Parser-class.html#id,Method Bio.Phylo.PhyloXMLIO.Writer.id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#id,Variable Bio.SearchIO._model.hit.Hit.id=Bio.SearchIO._model.hit.Hit-class.html#id,Variable Bio.SearchIO._model.query.QueryResult.id=Bio.SearchIO._model.query.QueryResult-class.html#id"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-267', 'id', 'link-267');">id</a></tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
<a name="L625"></a><tt class="py-lineno"> 625</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-268" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-268', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-269" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-269', 'id', 'link-267');">id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L626"></a><tt class="py-lineno"> 626</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-270" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-270', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-271" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DescriptionConsumer.description()=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#description,Method Bio.Phylo.PhyloXMLIO.Writer.description()=Bio.Phylo.PhyloXMLIO.Writer-class.html#description,Variable Bio.SearchIO._model.hit.Hit.description=Bio.SearchIO._model.hit.Hit-class.html#description,Variable Bio.SearchIO._model.query.QueryResult.description=Bio.SearchIO._model.query.QueryResult-class.html#description"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-271', 'description', 'link-271');">description</a></tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L627"></a><tt class="py-lineno"> 627</tt>  <tt class="py-line"> </tt>
<a name="L628"></a><tt class="py-lineno"> 628</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_use_fuzziness</tt> <tt class="py-op">=</tt> <tt class="py-name">use_fuzziness</tt> </tt>
<a name="L629"></a><tt class="py-lineno"> 629</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_feature_cleaner</tt> <tt class="py-op">=</tt> <tt class="py-name">feature_cleaner</tt> </tt>
<a name="L630"></a><tt class="py-lineno"> 630</tt>  <tt class="py-line"> </tt>
<a name="L631"></a><tt class="py-lineno"> 631</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_type</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L632"></a><tt class="py-lineno"> 632</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_data</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L633"></a><tt class="py-lineno"> 633</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L634"></a><tt class="py-lineno"> 634</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_feature</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L635"></a><tt class="py-lineno"> 635</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_expected_size</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L636"></a><tt class="py-lineno"> 636</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.locus"></a><div id="_FeatureConsumer.locus-def"><a name="L637"></a><tt class="py-lineno"> 637</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.locus-toggle" onclick="return toggle('_FeatureConsumer.locus');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#locus">locus</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">locus_name</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.locus-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.locus-expanded"><a name="L638"></a><tt class="py-lineno"> 638</tt>  <tt class="py-line">        <tt class="py-docstring">"""Set the locus name is set as the name of the Sequence.</tt> </tt>
<a name="L639"></a><tt class="py-lineno"> 639</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L640"></a><tt class="py-lineno"> 640</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-272" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-272', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-273" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.name=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.name=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.name=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#name,Method Bio.Phylo.PhyloXMLIO.Writer.name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#name,Variable Bio.codonalign.codonalphabet.CodonAlphabet.name=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-273', 'name', 'link-273');">name</a></tt> <tt class="py-op">=</tt> <tt class="py-name">locus_name</tt> </tt>
</div><a name="L641"></a><tt class="py-lineno"> 641</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.size"></a><div id="_FeatureConsumer.size-def"><a name="L642"></a><tt class="py-lineno"> 642</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.size-toggle" onclick="return toggle('_FeatureConsumer.size');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#size">size</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.size-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.size-expanded"><a name="L643"></a><tt class="py-lineno"> 643</tt>  <tt class="py-line">        <tt class="py-docstring">"""Record the sequence length."""</tt> </tt>
<a name="L644"></a><tt class="py-lineno"> 644</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_expected_size</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">)</tt> </tt>
</div><a name="L645"></a><tt class="py-lineno"> 645</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.residue_type"></a><div id="_FeatureConsumer.residue_type-def"><a name="L646"></a><tt class="py-lineno"> 646</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.residue_type-toggle" onclick="return toggle('_FeatureConsumer.residue_type');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#residue_type">residue_type</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">type</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.residue_type-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.residue_type-expanded"><a name="L647"></a><tt class="py-lineno"> 647</tt>  <tt class="py-line">        <tt class="py-docstring">"""Record the sequence type so we can choose an appropriate alphabet.</tt> </tt>
<a name="L648"></a><tt class="py-lineno"> 648</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L649"></a><tt class="py-lineno"> 649</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_type</tt> <tt class="py-op">=</tt> <tt id="link-274" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.type()=Bio.Phylo.PhyloXMLIO.Writer-class.html#type"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-274', 'type', 'link-274');">type</a></tt><tt class="py-op">.</tt><tt id="link-275" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-275', 'strip', 'link-236');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L650"></a><tt class="py-lineno"> 650</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.data_file_division"></a><div id="_FeatureConsumer.data_file_division-def"><a name="L651"></a><tt class="py-lineno"> 651</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.data_file_division-toggle" onclick="return toggle('_FeatureConsumer.data_file_division');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#data_file_division">data_file_division</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">division</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.data_file_division-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.data_file_division-expanded"><a name="L652"></a><tt class="py-lineno"> 652</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-276" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-276', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-277" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.annotations=BioSQL.BioSeq.DBSeqRecord-class.html#annotations"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-277', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'data_file_division'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">division</tt> </tt>
</div><a name="L653"></a><tt class="py-lineno"> 653</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.date"></a><div id="_FeatureConsumer.date-def"><a name="L654"></a><tt class="py-lineno"> 654</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.date-toggle" onclick="return toggle('_FeatureConsumer.date');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#date">date</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">submit_date</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.date-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.date-expanded"><a name="L655"></a><tt class="py-lineno"> 655</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-278" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-278', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-279" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-279', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'date'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">submit_date</tt> </tt>
</div><a name="L656"></a><tt class="py-lineno"> 656</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.definition"></a><div id="_FeatureConsumer.definition-def"><a name="L657"></a><tt class="py-lineno"> 657</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.definition-toggle" onclick="return toggle('_FeatureConsumer.definition');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#definition">definition</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">definition</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.definition-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.definition-expanded"><a name="L658"></a><tt class="py-lineno"> 658</tt>  <tt class="py-line">        <tt class="py-docstring">"""Set the definition as the description of the sequence.</tt> </tt>
<a name="L659"></a><tt class="py-lineno"> 659</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L660"></a><tt class="py-lineno"> 660</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-280" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-280', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-281" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-281', 'description', 'link-271');">description</a></tt><tt class="py-op">:</tt> </tt>
<a name="L661"></a><tt class="py-lineno"> 661</tt>  <tt class="py-line">            <tt class="py-comment"># Append to any existing description</tt> </tt>
<a name="L662"></a><tt class="py-lineno"> 662</tt>  <tt class="py-line">            <tt class="py-comment"># e.g. EMBL files with two DE lines.</tt> </tt>
<a name="L663"></a><tt class="py-lineno"> 663</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-282" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-282', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-283" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-283', 'description', 'link-271');">description</a></tt> <tt class="py-op">+=</tt> <tt class="py-string">" "</tt> <tt class="py-op">+</tt> <tt id="link-284" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.definition()=Bio.GenBank._FeatureConsumer-class.html#definition,Method Bio.GenBank._RecordConsumer.definition()=Bio.GenBank._RecordConsumer-class.html#definition"><a title="Bio.GenBank._FeatureConsumer.definition
Bio.GenBank._RecordConsumer.definition" class="py-name" href="#" onclick="return doclink('link-284', 'definition', 'link-284');">definition</a></tt> </tt>
<a name="L664"></a><tt class="py-lineno"> 664</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L665"></a><tt class="py-lineno"> 665</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-285" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-285', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-286" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-286', 'description', 'link-271');">description</a></tt> <tt class="py-op">=</tt> <tt id="link-287" class="py-name"><a title="Bio.GenBank._FeatureConsumer.definition
Bio.GenBank._RecordConsumer.definition" class="py-name" href="#" onclick="return doclink('link-287', 'definition', 'link-284');">definition</a></tt> </tt>
</div><a name="L666"></a><tt class="py-lineno"> 666</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.accession"></a><div id="_FeatureConsumer.accession-def"><a name="L667"></a><tt class="py-lineno"> 667</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.accession-toggle" onclick="return toggle('_FeatureConsumer.accession');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#accession">accession</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">acc_num</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.accession-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.accession-expanded"><a name="L668"></a><tt class="py-lineno"> 668</tt>  <tt class="py-line">        <tt class="py-docstring">"""Set the accession number as the id of the sequence.</tt> </tt>
<a name="L669"></a><tt class="py-lineno"> 669</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L670"></a><tt class="py-lineno"> 670</tt>  <tt class="py-line"><tt class="py-docstring">        If we have multiple accession numbers, the first one passed is</tt> </tt>
<a name="L671"></a><tt class="py-lineno"> 671</tt>  <tt class="py-line"><tt class="py-docstring">        used.</tt> </tt>
<a name="L672"></a><tt class="py-lineno"> 672</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L673"></a><tt class="py-lineno"> 673</tt>  <tt class="py-line">        <tt class="py-name">new_acc_nums</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-288" class="py-name" targets="Method Bio.GenBank._BaseGenBankConsumer._split_accessions()=Bio.GenBank._BaseGenBankConsumer-class.html#_split_accessions"><a title="Bio.GenBank._BaseGenBankConsumer._split_accessions" class="py-name" href="#" onclick="return doclink('link-288', '_split_accessions', 'link-288');">_split_accessions</a></tt><tt class="py-op">(</tt><tt class="py-name">acc_num</tt><tt class="py-op">)</tt> </tt>
<a name="L674"></a><tt class="py-lineno"> 674</tt>  <tt class="py-line"> </tt>
<a name="L675"></a><tt class="py-lineno"> 675</tt>  <tt class="py-line">        <tt class="py-comment"># Also record them ALL in the annotations</tt> </tt>
<a name="L676"></a><tt class="py-lineno"> 676</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L677"></a><tt class="py-lineno"> 677</tt>  <tt class="py-line">            <tt class="py-comment"># On the off chance there was more than one accession line:</tt> </tt>
<a name="L678"></a><tt class="py-lineno"> 678</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt id="link-289" class="py-name" targets="Variable Bio.FSSP.fssp_rec.align.acc=Bio.FSSP.fssp_rec.align-class.html#acc"><a title="Bio.FSSP.fssp_rec.align.acc" class="py-name" href="#" onclick="return doclink('link-289', 'acc', 'link-289');">acc</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">new_acc_nums</tt><tt class="py-op">:</tt> </tt>
<a name="L679"></a><tt class="py-lineno"> 679</tt>  <tt class="py-line">                <tt class="py-comment"># Prevent repeat entries</tt> </tt>
<a name="L680"></a><tt class="py-lineno"> 680</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-290" class="py-name"><a title="Bio.FSSP.fssp_rec.align.acc" class="py-name" href="#" onclick="return doclink('link-290', 'acc', 'link-289');">acc</a></tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-291" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-291', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-292" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-292', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'accessions'</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L681"></a><tt class="py-lineno"> 681</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-293" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-293', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-294" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-294', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'accessions'</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-295" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-295', 'append', 'link-213');">append</a></tt><tt class="py-op">(</tt><tt id="link-296" class="py-name"><a title="Bio.FSSP.fssp_rec.align.acc" class="py-name" href="#" onclick="return doclink('link-296', 'acc', 'link-289');">acc</a></tt><tt class="py-op">)</tt> </tt>
<a name="L682"></a><tt class="py-lineno"> 682</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L683"></a><tt class="py-lineno"> 683</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-297" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-297', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-298" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-298', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'accessions'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">new_acc_nums</tt> </tt>
<a name="L684"></a><tt class="py-lineno"> 684</tt>  <tt class="py-line"> </tt>
<a name="L685"></a><tt class="py-lineno"> 685</tt>  <tt class="py-line">        <tt class="py-comment"># if we haven't set the id information yet, add the first acc num</tt> </tt>
<a name="L686"></a><tt class="py-lineno"> 686</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-299" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-299', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-300" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-300', 'id', 'link-267');">id</a></tt><tt class="py-op">:</tt> </tt>
<a name="L687"></a><tt class="py-lineno"> 687</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">new_acc_nums</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L688"></a><tt class="py-lineno"> 688</tt>  <tt class="py-line">                <tt class="py-comment"># self.data.id = new_acc_nums[0]</tt> </tt>
<a name="L689"></a><tt class="py-lineno"> 689</tt>  <tt class="py-line">                <tt class="py-comment"># Use the FIRST accession as the ID, not the first on this line!</tt> </tt>
<a name="L690"></a><tt class="py-lineno"> 690</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-301" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-301', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-302" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-302', 'id', 'link-267');">id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-303" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-303', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-304" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-304', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'accessions'</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
</div><a name="L691"></a><tt class="py-lineno"> 691</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.wgs"></a><div id="_FeatureConsumer.wgs-def"><a name="L692"></a><tt class="py-lineno"> 692</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.wgs-toggle" onclick="return toggle('_FeatureConsumer.wgs');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#wgs">wgs</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.wgs-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.wgs-expanded"><a name="L693"></a><tt class="py-lineno"> 693</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-305" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-305', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-306" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-306', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'wgs'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt><tt class="py-op">.</tt><tt id="link-307" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-307', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">'-'</tt><tt class="py-op">)</tt> </tt>
</div><a name="L694"></a><tt class="py-lineno"> 694</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.add_wgs_scafld"></a><div id="_FeatureConsumer.add_wgs_scafld-def"><a name="L695"></a><tt class="py-lineno"> 695</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.add_wgs_scafld-toggle" onclick="return toggle('_FeatureConsumer.add_wgs_scafld');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#add_wgs_scafld">add_wgs_scafld</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.add_wgs_scafld-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.add_wgs_scafld-expanded"><a name="L696"></a><tt class="py-lineno"> 696</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-308" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-308', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-309" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-309', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-310" class="py-name" targets="Method Bio.Crystal.Crystal.setdefault()=Bio.Crystal.Crystal-class.html#setdefault,Method Bio._py3k._ordereddict.OrderedDict.setdefault()=Bio._py3k._ordereddict.OrderedDict-class.html#setdefault"><a title="Bio.Crystal.Crystal.setdefault
Bio._py3k._ordereddict.OrderedDict.setdefault" class="py-name" href="#" onclick="return doclink('link-310', 'setdefault', 'link-310');">setdefault</a></tt><tt class="py-op">(</tt><tt class="py-string">'wgs_scafld'</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-311" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-311', 'append', 'link-213');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">.</tt><tt id="link-312" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-312', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">'-'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L697"></a><tt class="py-lineno"> 697</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.nid"></a><div id="_FeatureConsumer.nid-def"><a name="L698"></a><tt class="py-lineno"> 698</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.nid-toggle" onclick="return toggle('_FeatureConsumer.nid');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#nid">nid</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.nid-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.nid-expanded"><a name="L699"></a><tt class="py-lineno"> 699</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-313" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-313', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-314" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-314', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'nid'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L700"></a><tt class="py-lineno"> 700</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.pid"></a><div id="_FeatureConsumer.pid-def"><a name="L701"></a><tt class="py-lineno"> 701</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.pid-toggle" onclick="return toggle('_FeatureConsumer.pid');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#pid">pid</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.pid-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.pid-expanded"><a name="L702"></a><tt class="py-lineno"> 702</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-315" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-315', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-316" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-316', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'pid'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L703"></a><tt class="py-lineno"> 703</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.version"></a><div id="_FeatureConsumer.version-def"><a name="L704"></a><tt class="py-lineno"> 704</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.version-toggle" onclick="return toggle('_FeatureConsumer.version');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#version">version</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">version_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.version-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.version-expanded"><a name="L705"></a><tt class="py-lineno"> 705</tt>  <tt class="py-line">        <tt class="py-comment"># Want to use the versioned accession as the record.id</tt> </tt>
<a name="L706"></a><tt class="py-lineno"> 706</tt>  <tt class="py-line">        <tt class="py-comment"># This comes from the VERSION line in GenBank files, or the</tt> </tt>
<a name="L707"></a><tt class="py-lineno"> 707</tt>  <tt class="py-line">        <tt class="py-comment"># obsolete SV line in EMBL.  For the new EMBL files we need</tt> </tt>
<a name="L708"></a><tt class="py-lineno"> 708</tt>  <tt class="py-line">        <tt class="py-comment"># both the version suffix from the ID line and the accession</tt> </tt>
<a name="L709"></a><tt class="py-lineno"> 709</tt>  <tt class="py-line">        <tt class="py-comment"># from the AC line.</tt> </tt>
<a name="L710"></a><tt class="py-lineno"> 710</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">version_id</tt><tt class="py-op">.</tt><tt id="link-317" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-317', 'count', 'link-189');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">"."</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt> <tt class="py-keyword">and</tt> <tt class="py-name">version_id</tt><tt class="py-op">.</tt><tt id="link-318" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-318', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">"."</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">isdigit</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L711"></a><tt class="py-lineno"> 711</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-319" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-319', 'accession', 'link-238');">accession</a></tt><tt class="py-op">(</tt><tt class="py-name">version_id</tt><tt class="py-op">.</tt><tt id="link-320" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-320', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">"."</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L712"></a><tt class="py-lineno"> 712</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-321" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.version_suffix()=Bio.GenBank._FeatureConsumer-class.html#version_suffix"><a title="Bio.GenBank._FeatureConsumer.version_suffix" class="py-name" href="#" onclick="return doclink('link-321', 'version_suffix', 'link-321');">version_suffix</a></tt><tt class="py-op">(</tt><tt class="py-name">version_id</tt><tt class="py-op">.</tt><tt id="link-322" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-322', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">"."</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L713"></a><tt class="py-lineno"> 713</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">version_id</tt><tt class="py-op">:</tt> </tt>
<a name="L714"></a><tt class="py-lineno"> 714</tt>  <tt class="py-line">            <tt class="py-comment"># For backwards compatibility...</tt> </tt>
<a name="L715"></a><tt class="py-lineno"> 715</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-323" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-323', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-324" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-324', 'id', 'link-267');">id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">version_id</tt> </tt>
</div><a name="L716"></a><tt class="py-lineno"> 716</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.project"></a><div id="_FeatureConsumer.project-def"><a name="L717"></a><tt class="py-lineno"> 717</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.project-toggle" onclick="return toggle('_FeatureConsumer.project');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#project">project</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.project-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.project-expanded"><a name="L718"></a><tt class="py-lineno"> 718</tt>  <tt class="py-line">        <tt class="py-docstring">"""Handle the information from the PROJECT line as a list of projects.</tt> </tt>
<a name="L719"></a><tt class="py-lineno"> 719</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L720"></a><tt class="py-lineno"> 720</tt>  <tt class="py-line"><tt class="py-docstring">        e.g.::</tt> </tt>
<a name="L721"></a><tt class="py-lineno"> 721</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L722"></a><tt class="py-lineno"> 722</tt>  <tt class="py-line"><tt class="py-docstring">            PROJECT     GenomeProject:28471</tt> </tt>
<a name="L723"></a><tt class="py-lineno"> 723</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L724"></a><tt class="py-lineno"> 724</tt>  <tt class="py-line"><tt class="py-docstring">        or::</tt> </tt>
<a name="L725"></a><tt class="py-lineno"> 725</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L726"></a><tt class="py-lineno"> 726</tt>  <tt class="py-line"><tt class="py-docstring">            PROJECT     GenomeProject:13543  GenomeProject:99999</tt> </tt>
<a name="L727"></a><tt class="py-lineno"> 727</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L728"></a><tt class="py-lineno"> 728</tt>  <tt class="py-line"><tt class="py-docstring">        This is stored as dbxrefs in the SeqRecord to be consistent with the</tt> </tt>
<a name="L729"></a><tt class="py-lineno"> 729</tt>  <tt class="py-line"><tt class="py-docstring">        projected switch of this line to DBLINK in future GenBank versions.</tt> </tt>
<a name="L730"></a><tt class="py-lineno"> 730</tt>  <tt class="py-line"><tt class="py-docstring">        Note the NCBI plan to replace "GenomeProject:28471" with the shorter</tt> </tt>
<a name="L731"></a><tt class="py-lineno"> 731</tt>  <tt class="py-line"><tt class="py-docstring">        "Project:28471" as part of this transition.</tt> </tt>
<a name="L732"></a><tt class="py-lineno"> 732</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L733"></a><tt class="py-lineno"> 733</tt>  <tt class="py-line">        <tt class="py-name">content</tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"GenomeProject:"</tt><tt class="py-op">,</tt> <tt class="py-string">"Project:"</tt><tt class="py-op">)</tt> </tt>
<a name="L734"></a><tt class="py-lineno"> 734</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-325" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-325', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-326" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.dbxrefs=BioSQL.BioSeq.DBSeqRecord-class.html#dbxrefs"><a title="BioSQL.BioSeq.DBSeqRecord.dbxrefs" class="py-name" href="#" onclick="return doclink('link-326', 'dbxrefs', 'link-326');">dbxrefs</a></tt><tt class="py-op">.</tt><tt id="link-327" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-327', 'extend', 'link-248');">extend</a></tt><tt class="py-op">(</tt><tt class="py-name">p</tt> <tt class="py-keyword">for</tt> <tt class="py-name">p</tt> <tt class="py-keyword">in</tt> <tt class="py-name">content</tt><tt class="py-op">.</tt><tt id="link-328" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-328', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt class="py-name">p</tt><tt class="py-op">)</tt> </tt>
</div><a name="L735"></a><tt class="py-lineno"> 735</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.dblink"></a><div id="_FeatureConsumer.dblink-def"><a name="L736"></a><tt class="py-lineno"> 736</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.dblink-toggle" onclick="return toggle('_FeatureConsumer.dblink');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#dblink">dblink</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.dblink-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.dblink-expanded"><a name="L737"></a><tt class="py-lineno"> 737</tt>  <tt class="py-line">        <tt class="py-docstring">"""Store DBLINK cross references as dbxrefs in our record object.</tt> </tt>
<a name="L738"></a><tt class="py-lineno"> 738</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L739"></a><tt class="py-lineno"> 739</tt>  <tt class="py-line"><tt class="py-docstring">        This line type is expected to replace the PROJECT line in 2009. e.g.</tt> </tt>
<a name="L740"></a><tt class="py-lineno"> 740</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L741"></a><tt class="py-lineno"> 741</tt>  <tt class="py-line"><tt class="py-docstring">        During transition::</tt> </tt>
<a name="L742"></a><tt class="py-lineno"> 742</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L743"></a><tt class="py-lineno"> 743</tt>  <tt class="py-line"><tt class="py-docstring">            PROJECT     GenomeProject:28471</tt> </tt>
<a name="L744"></a><tt class="py-lineno"> 744</tt>  <tt class="py-line"><tt class="py-docstring">            DBLINK      Project:28471</tt> </tt>
<a name="L745"></a><tt class="py-lineno"> 745</tt>  <tt class="py-line"><tt class="py-docstring">                        Trace Assembly Archive:123456</tt> </tt>
<a name="L746"></a><tt class="py-lineno"> 746</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L747"></a><tt class="py-lineno"> 747</tt>  <tt class="py-line"><tt class="py-docstring">        Once the project line is dropped::</tt> </tt>
<a name="L748"></a><tt class="py-lineno"> 748</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L749"></a><tt class="py-lineno"> 749</tt>  <tt class="py-line"><tt class="py-docstring">            DBLINK      Project:28471</tt> </tt>
<a name="L750"></a><tt class="py-lineno"> 750</tt>  <tt class="py-line"><tt class="py-docstring">                        Trace Assembly Archive:123456</tt> </tt>
<a name="L751"></a><tt class="py-lineno"> 751</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L752"></a><tt class="py-lineno"> 752</tt>  <tt class="py-line"><tt class="py-docstring">        Note GenomeProject -&gt; Project.</tt> </tt>
<a name="L753"></a><tt class="py-lineno"> 753</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L754"></a><tt class="py-lineno"> 754</tt>  <tt class="py-line"><tt class="py-docstring">        We'll have to see some real examples to be sure, but based on the</tt> </tt>
<a name="L755"></a><tt class="py-lineno"> 755</tt>  <tt class="py-line"><tt class="py-docstring">        above example we can expect one reference per line.</tt> </tt>
<a name="L756"></a><tt class="py-lineno"> 756</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L757"></a><tt class="py-lineno"> 757</tt>  <tt class="py-line"><tt class="py-docstring">        Note that at some point the NCBI have included an extra space, e.g.::</tt> </tt>
<a name="L758"></a><tt class="py-lineno"> 758</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L759"></a><tt class="py-lineno"> 759</tt>  <tt class="py-line"><tt class="py-docstring">            DBLINK      Project: 28471</tt> </tt>
<a name="L760"></a><tt class="py-lineno"> 760</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L761"></a><tt class="py-lineno"> 761</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L762"></a><tt class="py-lineno"> 762</tt>  <tt class="py-line">        <tt class="py-comment"># During the transition period with both PROJECT and DBLINK lines,</tt> </tt>
<a name="L763"></a><tt class="py-lineno"> 763</tt>  <tt class="py-line">        <tt class="py-comment"># we don't want to add the same cross reference twice.</tt> </tt>
<a name="L764"></a><tt class="py-lineno"> 764</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-string">": "</tt> <tt class="py-keyword">in</tt> <tt class="py-name">content</tt><tt class="py-op">:</tt> </tt>
<a name="L765"></a><tt class="py-lineno"> 765</tt>  <tt class="py-line">            <tt class="py-name">content</tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">": "</tt><tt class="py-op">,</tt> <tt class="py-string">":"</tt><tt class="py-op">)</tt> </tt>
<a name="L766"></a><tt class="py-lineno"> 766</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">content</tt><tt class="py-op">.</tt><tt id="link-329" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-329', 'strip', 'link-236');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-330" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-330', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-331" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.dbxrefs" class="py-name" href="#" onclick="return doclink('link-331', 'dbxrefs', 'link-326');">dbxrefs</a></tt><tt class="py-op">:</tt> </tt>
<a name="L767"></a><tt class="py-lineno"> 767</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-332" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-332', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-333" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.dbxrefs" class="py-name" href="#" onclick="return doclink('link-333', 'dbxrefs', 'link-326');">dbxrefs</a></tt><tt class="py-op">.</tt><tt id="link-334" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-334', 'append', 'link-213');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">.</tt><tt id="link-335" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-335', 'strip', 'link-236');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L768"></a><tt class="py-lineno"> 768</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.version_suffix"></a><div id="_FeatureConsumer.version_suffix-def"><a name="L769"></a><tt class="py-lineno"> 769</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.version_suffix-toggle" onclick="return toggle('_FeatureConsumer.version_suffix');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#version_suffix">version_suffix</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">version</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.version_suffix-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.version_suffix-expanded"><a name="L770"></a><tt class="py-lineno"> 770</tt>  <tt class="py-line">        <tt class="py-docstring">"""Set the version to overwrite the id.</tt> </tt>
<a name="L771"></a><tt class="py-lineno"> 771</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L772"></a><tt class="py-lineno"> 772</tt>  <tt class="py-line"><tt class="py-docstring">        Since the version provides the same information as the accession</tt> </tt>
<a name="L773"></a><tt class="py-lineno"> 773</tt>  <tt class="py-line"><tt class="py-docstring">        number, plus some extra info, we set this as the id if we have</tt> </tt>
<a name="L774"></a><tt class="py-lineno"> 774</tt>  <tt class="py-line"><tt class="py-docstring">        a version.</tt> </tt>
<a name="L775"></a><tt class="py-lineno"> 775</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L776"></a><tt class="py-lineno"> 776</tt>  <tt class="py-line">        <tt class="py-comment"># e.g. GenBank line:</tt> </tt>
<a name="L777"></a><tt class="py-lineno"> 777</tt>  <tt class="py-line">        <tt class="py-comment"># VERSION     U49845.1  GI:1293613</tt> </tt>
<a name="L778"></a><tt class="py-lineno"> 778</tt>  <tt class="py-line">        <tt class="py-comment"># or the obsolete EMBL line:</tt> </tt>
<a name="L779"></a><tt class="py-lineno"> 779</tt>  <tt class="py-line">        <tt class="py-comment"># SV   U49845.1</tt> </tt>
<a name="L780"></a><tt class="py-lineno"> 780</tt>  <tt class="py-line">        <tt class="py-comment"># Scanner calls consumer.version("U49845.1")</tt> </tt>
<a name="L781"></a><tt class="py-lineno"> 781</tt>  <tt class="py-line">        <tt class="py-comment"># which then calls consumer.version_suffix(1)</tt> </tt>
<a name="L782"></a><tt class="py-lineno"> 782</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L783"></a><tt class="py-lineno"> 783</tt>  <tt class="py-line">        <tt class="py-comment"># e.g. EMBL new line:</tt> </tt>
<a name="L784"></a><tt class="py-lineno"> 784</tt>  <tt class="py-line">        <tt class="py-comment"># ID   X56734; SV 1; linear; mRNA; STD; PLN; 1859 BP.</tt> </tt>
<a name="L785"></a><tt class="py-lineno"> 785</tt>  <tt class="py-line">        <tt class="py-comment"># Scanner calls consumer.version_suffix(1)</tt> </tt>
<a name="L786"></a><tt class="py-lineno"> 786</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt id="link-336" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HeaderConsumer.version()=Bio.Blast.NCBIStandalone._HeaderConsumer-class.html#version,Function Bio.Cluster.cluster.version()=Bio.Cluster.cluster-module.html#version,Method Bio.GenBank._FeatureConsumer.version()=Bio.GenBank._FeatureConsumer-class.html#version,Method Bio.GenBank._RecordConsumer.version()=Bio.GenBank._RecordConsumer-class.html#version,Variable Bio.motifs.jaspar.Motif.version=Bio.motifs.jaspar.Motif-class.html#version"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-336', 'version', 'link-336');">version</a></tt><tt class="py-op">.</tt><tt class="py-name">isdigit</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L787"></a><tt class="py-lineno"> 787</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-337" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-337', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-338" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-338', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'sequence_version'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-339" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-339', 'version', 'link-336');">version</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L788"></a><tt class="py-lineno"> 788</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.db_source"></a><div id="_FeatureConsumer.db_source-def"><a name="L789"></a><tt class="py-lineno"> 789</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.db_source-toggle" onclick="return toggle('_FeatureConsumer.db_source');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#db_source">db_source</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.db_source-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.db_source-expanded"><a name="L790"></a><tt class="py-lineno"> 790</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-340" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-340', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-341" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-341', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'db_source'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt><tt class="py-op">.</tt><tt id="link-342" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-342', 'rstrip', 'link-214');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L791"></a><tt class="py-lineno"> 791</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.gi"></a><div id="_FeatureConsumer.gi-def"><a name="L792"></a><tt class="py-lineno"> 792</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.gi-toggle" onclick="return toggle('_FeatureConsumer.gi');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#gi">gi</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.gi-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.gi-expanded"><a name="L793"></a><tt class="py-lineno"> 793</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-343" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-343', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-344" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-344', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'gi'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L794"></a><tt class="py-lineno"> 794</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.keywords"></a><div id="_FeatureConsumer.keywords-def"><a name="L795"></a><tt class="py-lineno"> 795</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.keywords-toggle" onclick="return toggle('_FeatureConsumer.keywords');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#keywords">keywords</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.keywords-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.keywords-expanded"><a name="L796"></a><tt class="py-lineno"> 796</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">'keywords'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-345" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-345', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-346" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-346', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">:</tt> </tt>
<a name="L797"></a><tt class="py-lineno"> 797</tt>  <tt class="py-line">            <tt class="py-comment"># Multi-line keywords, append to list</tt> </tt>
<a name="L798"></a><tt class="py-lineno"> 798</tt>  <tt class="py-line">            <tt class="py-comment"># Note EMBL states "A keyword is never split between lines."</tt> </tt>
<a name="L799"></a><tt class="py-lineno"> 799</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-347" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-347', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-348" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-348', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'keywords'</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-349" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-349', 'extend', 'link-248');">extend</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-350" class="py-name" targets="Method Bio.GenBank._BaseGenBankConsumer._split_keywords()=Bio.GenBank._BaseGenBankConsumer-class.html#_split_keywords"><a title="Bio.GenBank._BaseGenBankConsumer._split_keywords" class="py-name" href="#" onclick="return doclink('link-350', '_split_keywords', 'link-350');">_split_keywords</a></tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L800"></a><tt class="py-lineno"> 800</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L801"></a><tt class="py-lineno"> 801</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-351" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-351', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-352" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-352', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'keywords'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-353" class="py-name"><a title="Bio.GenBank._BaseGenBankConsumer._split_keywords" class="py-name" href="#" onclick="return doclink('link-353', '_split_keywords', 'link-350');">_split_keywords</a></tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">)</tt> </tt>
</div><a name="L802"></a><tt class="py-lineno"> 802</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.segment"></a><div id="_FeatureConsumer.segment-def"><a name="L803"></a><tt class="py-lineno"> 803</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.segment-toggle" onclick="return toggle('_FeatureConsumer.segment');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#segment">segment</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.segment-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.segment-expanded"><a name="L804"></a><tt class="py-lineno"> 804</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-354" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-354', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-355" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-355', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'segment'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L805"></a><tt class="py-lineno"> 805</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.source"></a><div id="_FeatureConsumer.source-def"><a name="L806"></a><tt class="py-lineno"> 806</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.source-toggle" onclick="return toggle('_FeatureConsumer.source');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#source">source</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.source-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.source-expanded"><a name="L807"></a><tt class="py-lineno"> 807</tt>  <tt class="py-line">        <tt class="py-comment"># Note that some software (e.g. VectorNTI) may produce an empty</tt> </tt>
<a name="L808"></a><tt class="py-lineno"> 808</tt>  <tt class="py-line">        <tt class="py-comment"># source (rather than using a dot/period as might be expected).</tt> </tt>
<a name="L809"></a><tt class="py-lineno"> 809</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">content</tt> <tt class="py-op">==</tt> <tt class="py-string">""</tt><tt class="py-op">:</tt> </tt>
<a name="L810"></a><tt class="py-lineno"> 810</tt>  <tt class="py-line">            <tt class="py-name">source_info</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L811"></a><tt class="py-lineno"> 811</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">content</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'.'</tt><tt class="py-op">:</tt> </tt>
<a name="L812"></a><tt class="py-lineno"> 812</tt>  <tt class="py-line">            <tt class="py-name">source_info</tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L813"></a><tt class="py-lineno"> 813</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L814"></a><tt class="py-lineno"> 814</tt>  <tt class="py-line">            <tt class="py-name">source_info</tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
<a name="L815"></a><tt class="py-lineno"> 815</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-356" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-356', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-357" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-357', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'source'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">source_info</tt> </tt>
</div><a name="L816"></a><tt class="py-lineno"> 816</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.organism"></a><div id="_FeatureConsumer.organism-def"><a name="L817"></a><tt class="py-lineno"> 817</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.organism-toggle" onclick="return toggle('_FeatureConsumer.organism');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#organism">organism</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.organism-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.organism-expanded"><a name="L818"></a><tt class="py-lineno"> 818</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-358" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-358', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-359" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-359', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'organism'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L819"></a><tt class="py-lineno"> 819</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.taxonomy"></a><div id="_FeatureConsumer.taxonomy-def"><a name="L820"></a><tt class="py-lineno"> 820</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.taxonomy-toggle" onclick="return toggle('_FeatureConsumer.taxonomy');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#taxonomy">taxonomy</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.taxonomy-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.taxonomy-expanded"><a name="L821"></a><tt class="py-lineno"> 821</tt>  <tt class="py-line">        <tt class="py-docstring">"""Records (another line of) the taxonomy lineage.</tt> </tt>
<a name="L822"></a><tt class="py-lineno"> 822</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L823"></a><tt class="py-lineno"> 823</tt>  <tt class="py-line">        <tt class="py-name">lineage</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-360" class="py-name" targets="Method Bio.GenBank._BaseGenBankConsumer._split_taxonomy()=Bio.GenBank._BaseGenBankConsumer-class.html#_split_taxonomy"><a title="Bio.GenBank._BaseGenBankConsumer._split_taxonomy" class="py-name" href="#" onclick="return doclink('link-360', '_split_taxonomy', 'link-360');">_split_taxonomy</a></tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">)</tt> </tt>
<a name="L824"></a><tt class="py-lineno"> 824</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L825"></a><tt class="py-lineno"> 825</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-361" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-361', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-362" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-362', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'taxonomy'</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-363" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-363', 'extend', 'link-248');">extend</a></tt><tt class="py-op">(</tt><tt class="py-name">lineage</tt><tt class="py-op">)</tt> </tt>
<a name="L826"></a><tt class="py-lineno"> 826</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L827"></a><tt class="py-lineno"> 827</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-364" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-364', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-365" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-365', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'taxonomy'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">lineage</tt> </tt>
</div><a name="L828"></a><tt class="py-lineno"> 828</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.reference_num"></a><div id="_FeatureConsumer.reference_num-def"><a name="L829"></a><tt class="py-lineno"> 829</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.reference_num-toggle" onclick="return toggle('_FeatureConsumer.reference_num');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#reference_num">reference_num</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.reference_num-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.reference_num-expanded"><a name="L830"></a><tt class="py-lineno"> 830</tt>  <tt class="py-line">        <tt class="py-docstring">"""Signal the beginning of a new reference object.</tt> </tt>
<a name="L831"></a><tt class="py-lineno"> 831</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L832"></a><tt class="py-lineno"> 832</tt>  <tt class="py-line">        <tt class="py-comment"># if we have a current reference that hasn't been added to</tt> </tt>
<a name="L833"></a><tt class="py-lineno"> 833</tt>  <tt class="py-line">        <tt class="py-comment"># the list of references, add it.</tt> </tt>
<a name="L834"></a><tt class="py-lineno"> 834</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L835"></a><tt class="py-lineno"> 835</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-366" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-366', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-367" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-367', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'references'</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-368" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-368', 'append', 'link-213');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">)</tt> </tt>
<a name="L836"></a><tt class="py-lineno"> 836</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L837"></a><tt class="py-lineno"> 837</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-369" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-369', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-370" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-370', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'references'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L838"></a><tt class="py-lineno"> 838</tt>  <tt class="py-line"> </tt>
<a name="L839"></a><tt class="py-lineno"> 839</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt> <tt class="py-op">=</tt> <tt id="link-371" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-371', 'SeqFeature', 'link-1');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-372" class="py-name" targets="Class Bio.ExPASy.Prodoc.Reference=Bio.ExPASy.Prodoc.Reference-class.html,Class Bio.GenBank.Record.Reference=Bio.GenBank.Record.Reference-class.html,Class Bio.Phylo.PhyloXML.Reference=Bio.Phylo.PhyloXML.Reference-class.html,Class Bio.SeqFeature.Reference=Bio.SeqFeature.Reference-class.html,Class Bio.SwissProt.Reference=Bio.SwissProt.Reference-class.html"><a title="Bio.ExPASy.Prodoc.Reference
Bio.GenBank.Record.Reference
Bio.Phylo.PhyloXML.Reference
Bio.SeqFeature.Reference
Bio.SwissProt.Reference" class="py-name" href="#" onclick="return doclink('link-372', 'Reference', 'link-372');">Reference</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L840"></a><tt class="py-lineno"> 840</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.reference_bases"></a><div id="_FeatureConsumer.reference_bases-def"><a name="L841"></a><tt class="py-lineno"> 841</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.reference_bases-toggle" onclick="return toggle('_FeatureConsumer.reference_bases');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#reference_bases">reference_bases</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.reference_bases-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.reference_bases-expanded"><a name="L842"></a><tt class="py-lineno"> 842</tt>  <tt class="py-line">        <tt class="py-docstring">"""Attempt to determine the sequence region the reference entails.</tt> </tt>
<a name="L843"></a><tt class="py-lineno"> 843</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L844"></a><tt class="py-lineno"> 844</tt>  <tt class="py-line"><tt class="py-docstring">        Possible types of information we may have to deal with:</tt> </tt>
<a name="L845"></a><tt class="py-lineno"> 845</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L846"></a><tt class="py-lineno"> 846</tt>  <tt class="py-line"><tt class="py-docstring">        (bases 1 to 86436)</tt> </tt>
<a name="L847"></a><tt class="py-lineno"> 847</tt>  <tt class="py-line"><tt class="py-docstring">        (sites)</tt> </tt>
<a name="L848"></a><tt class="py-lineno"> 848</tt>  <tt class="py-line"><tt class="py-docstring">        (bases 1 to 105654; 110423 to 111122)</tt> </tt>
<a name="L849"></a><tt class="py-lineno"> 849</tt>  <tt class="py-line"><tt class="py-docstring">        1  (residues 1 to 182)</tt> </tt>
<a name="L850"></a><tt class="py-lineno"> 850</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L851"></a><tt class="py-lineno"> 851</tt>  <tt class="py-line">        <tt class="py-comment"># first remove the parentheses or other junk</tt> </tt>
<a name="L852"></a><tt class="py-lineno"> 852</tt>  <tt class="py-line">        <tt class="py-name">ref_base_info</tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L853"></a><tt class="py-lineno"> 853</tt>  <tt class="py-line"> </tt>
<a name="L854"></a><tt class="py-lineno"> 854</tt>  <tt class="py-line">        <tt class="py-name">all_locations</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L855"></a><tt class="py-lineno"> 855</tt>  <tt class="py-line">        <tt class="py-comment"># parse if we've got 'bases' and 'to'</tt> </tt>
<a name="L856"></a><tt class="py-lineno"> 856</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">'bases'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">ref_base_info</tt> <tt class="py-keyword">and</tt> <tt class="py-string">'to'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">ref_base_info</tt><tt class="py-op">:</tt> </tt>
<a name="L857"></a><tt class="py-lineno"> 857</tt>  <tt class="py-line">            <tt class="py-comment"># get rid of the beginning 'bases'</tt> </tt>
<a name="L858"></a><tt class="py-lineno"> 858</tt>  <tt class="py-line">            <tt class="py-name">ref_base_info</tt> <tt class="py-op">=</tt> <tt class="py-name">ref_base_info</tt><tt class="py-op">[</tt><tt class="py-number">5</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L859"></a><tt class="py-lineno"> 859</tt>  <tt class="py-line">            <tt class="py-name">locations</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-373" class="py-name" targets="Method Bio.GenBank._FeatureConsumer._split_reference_locations()=Bio.GenBank._FeatureConsumer-class.html#_split_reference_locations"><a title="Bio.GenBank._FeatureConsumer._split_reference_locations" class="py-name" href="#" onclick="return doclink('link-373', '_split_reference_locations', 'link-373');">_split_reference_locations</a></tt><tt class="py-op">(</tt><tt class="py-name">ref_base_info</tt><tt class="py-op">)</tt> </tt>
<a name="L860"></a><tt class="py-lineno"> 860</tt>  <tt class="py-line">            <tt class="py-name">all_locations</tt><tt class="py-op">.</tt><tt id="link-374" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-374', 'extend', 'link-248');">extend</a></tt><tt class="py-op">(</tt><tt class="py-name">locations</tt><tt class="py-op">)</tt> </tt>
<a name="L861"></a><tt class="py-lineno"> 861</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-string">'residues'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">ref_base_info</tt> <tt class="py-keyword">and</tt> <tt class="py-string">'to'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">ref_base_info</tt><tt class="py-op">:</tt> </tt>
<a name="L862"></a><tt class="py-lineno"> 862</tt>  <tt class="py-line">            <tt class="py-name">residues_start</tt> <tt class="py-op">=</tt> <tt class="py-name">ref_base_info</tt><tt class="py-op">.</tt><tt id="link-375" class="py-name"><a title="Bio.NeuralNetwork.Gene.Motif.MotifFinder.find
Bio.NeuralNetwork.Gene.Schema.SchemaFinder.find
Bio.NeuralNetwork.Gene.Signature.SignatureFinder.find
Bio.Seq.Seq.find
Bio.triefind.find" class="py-name" href="#" onclick="return doclink('link-375', 'find', 'link-186');">find</a></tt><tt class="py-op">(</tt><tt class="py-string">"residues"</tt><tt class="py-op">)</tt> </tt>
<a name="L863"></a><tt class="py-lineno"> 863</tt>  <tt class="py-line">            <tt class="py-comment"># get only the information after "residues"</tt> </tt>
<a name="L864"></a><tt class="py-lineno"> 864</tt>  <tt class="py-line">            <tt class="py-name">ref_base_info</tt> <tt class="py-op">=</tt> <tt class="py-name">ref_base_info</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">residues_start</tt> <tt class="py-op">+</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-string">"residues "</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L865"></a><tt class="py-lineno"> 865</tt>  <tt class="py-line">            <tt class="py-name">locations</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-376" class="py-name"><a title="Bio.GenBank._FeatureConsumer._split_reference_locations" class="py-name" href="#" onclick="return doclink('link-376', '_split_reference_locations', 'link-373');">_split_reference_locations</a></tt><tt class="py-op">(</tt><tt class="py-name">ref_base_info</tt><tt class="py-op">)</tt> </tt>
<a name="L866"></a><tt class="py-lineno"> 866</tt>  <tt class="py-line">            <tt class="py-name">all_locations</tt><tt class="py-op">.</tt><tt id="link-377" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-377', 'extend', 'link-248');">extend</a></tt><tt class="py-op">(</tt><tt class="py-name">locations</tt><tt class="py-op">)</tt> </tt>
<a name="L867"></a><tt class="py-lineno"> 867</tt>  <tt class="py-line"> </tt>
<a name="L868"></a><tt class="py-lineno"> 868</tt>  <tt class="py-line">        <tt class="py-comment"># make sure if we are not finding information then we have</tt> </tt>
<a name="L869"></a><tt class="py-lineno"> 869</tt>  <tt class="py-line">        <tt class="py-comment"># the string 'sites' or the string 'bases'</tt> </tt>
<a name="L870"></a><tt class="py-lineno"> 870</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-op">(</tt><tt class="py-name">ref_base_info</tt> <tt class="py-op">==</tt> <tt class="py-string">'sites'</tt> <tt class="py-keyword">or</tt> </tt>
<a name="L871"></a><tt class="py-lineno"> 871</tt>  <tt class="py-line">              <tt class="py-name">ref_base_info</tt><tt class="py-op">.</tt><tt id="link-378" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-378', 'strip', 'link-236');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-string">'bases'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L872"></a><tt class="py-lineno"> 872</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
<a name="L873"></a><tt class="py-lineno"> 873</tt>  <tt class="py-line">        <tt class="py-comment"># otherwise raise an error</tt> </tt>
<a name="L874"></a><tt class="py-lineno"> 874</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L875"></a><tt class="py-lineno"> 875</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Could not parse base info %s in record %s"</tt> <tt class="py-op">%</tt> </tt>
<a name="L876"></a><tt class="py-lineno"> 876</tt>  <tt class="py-line">                             <tt class="py-op">(</tt><tt class="py-name">ref_base_info</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-379" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-379', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-380" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-380', 'id', 'link-267');">id</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L877"></a><tt class="py-lineno"> 877</tt>  <tt class="py-line"> </tt>
<a name="L878"></a><tt class="py-lineno"> 878</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt id="link-381" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.location()=Bio.GenBank._FeatureConsumer-class.html#location,Method Bio.GenBank._RecordConsumer.location()=Bio.GenBank._RecordConsumer-class.html#location,Method Bio.Phylo.PhyloXMLIO.Writer.location()=Bio.Phylo.PhyloXMLIO.Writer-class.html#location"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-381', 'location', 'link-381');">location</a></tt> <tt class="py-op">=</tt> <tt class="py-name">all_locations</tt> </tt>
</div><a name="L879"></a><tt class="py-lineno"> 879</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer._split_reference_locations"></a><div id="_FeatureConsumer._split_reference_locations-def"><a name="L880"></a><tt class="py-lineno"> 880</tt> <a class="py-toggle" href="#" id="_FeatureConsumer._split_reference_locations-toggle" onclick="return toggle('_FeatureConsumer._split_reference_locations');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#_split_reference_locations">_split_reference_locations</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">location_string</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer._split_reference_locations-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer._split_reference_locations-expanded"><a name="L881"></a><tt class="py-lineno"> 881</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get reference locations out of a string of reference information</tt> </tt>
<a name="L882"></a><tt class="py-lineno"> 882</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L883"></a><tt class="py-lineno"> 883</tt>  <tt class="py-line"><tt class="py-docstring">        The passed string should be of the form::</tt> </tt>
<a name="L884"></a><tt class="py-lineno"> 884</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L885"></a><tt class="py-lineno"> 885</tt>  <tt class="py-line"><tt class="py-docstring">            1 to 20; 20 to 100</tt> </tt>
<a name="L886"></a><tt class="py-lineno"> 886</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L887"></a><tt class="py-lineno"> 887</tt>  <tt class="py-line"><tt class="py-docstring">        This splits the information out and returns a list of location objects</tt> </tt>
<a name="L888"></a><tt class="py-lineno"> 888</tt>  <tt class="py-line"><tt class="py-docstring">        based on the reference locations.</tt> </tt>
<a name="L889"></a><tt class="py-lineno"> 889</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L890"></a><tt class="py-lineno"> 890</tt>  <tt class="py-line">        <tt class="py-comment"># split possibly multiple locations using the ';'</tt> </tt>
<a name="L891"></a><tt class="py-lineno"> 891</tt>  <tt class="py-line">        <tt class="py-name">all_base_info</tt> <tt class="py-op">=</tt> <tt class="py-name">location_string</tt><tt class="py-op">.</tt><tt id="link-382" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-382', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">';'</tt><tt class="py-op">)</tt> </tt>
<a name="L892"></a><tt class="py-lineno"> 892</tt>  <tt class="py-line"> </tt>
<a name="L893"></a><tt class="py-lineno"> 893</tt>  <tt class="py-line">        <tt class="py-name">new_locations</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L894"></a><tt class="py-lineno"> 894</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">base_info</tt> <tt class="py-keyword">in</tt> <tt class="py-name">all_base_info</tt><tt class="py-op">:</tt> </tt>
<a name="L895"></a><tt class="py-lineno"> 895</tt>  <tt class="py-line">            <tt id="link-383" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-383', 'start', 'link-258');">start</a></tt><tt class="py-op">,</tt> <tt id="link-384" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-384', 'end', 'link-259');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">base_info</tt><tt class="py-op">.</tt><tt id="link-385" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-385', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">'to'</tt><tt class="py-op">)</tt> </tt>
<a name="L896"></a><tt class="py-lineno"> 896</tt>  <tt class="py-line">            <tt class="py-name">new_start</tt><tt class="py-op">,</tt> <tt class="py-name">new_end</tt> <tt class="py-op">=</tt> \ </tt>
<a name="L897"></a><tt class="py-lineno"> 897</tt>  <tt class="py-line">              <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-386" class="py-name" targets="Method Bio.GenBank._BaseGenBankConsumer._convert_to_python_numbers()=Bio.GenBank._BaseGenBankConsumer-class.html#_convert_to_python_numbers"><a title="Bio.GenBank._BaseGenBankConsumer._convert_to_python_numbers" class="py-name" href="#" onclick="return doclink('link-386', '_convert_to_python_numbers', 'link-386');">_convert_to_python_numbers</a></tt><tt class="py-op">(</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-387" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-387', 'start', 'link-258');">start</a></tt><tt class="py-op">.</tt><tt id="link-388" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-388', 'strip', 'link-236');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L898"></a><tt class="py-lineno"> 898</tt>  <tt class="py-line">                                              <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-389" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-389', 'end', 'link-259');">end</a></tt><tt class="py-op">.</tt><tt id="link-390" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-390', 'strip', 'link-236');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L899"></a><tt class="py-lineno"> 899</tt>  <tt class="py-line">            <tt class="py-name">this_location</tt> <tt class="py-op">=</tt> <tt id="link-391" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-391', 'SeqFeature', 'link-1');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-392" class="py-name"><a title="Bio.SeqFeature.FeatureLocation" class="py-name" href="#" onclick="return doclink('link-392', 'FeatureLocation', 'link-180');">FeatureLocation</a></tt><tt class="py-op">(</tt><tt class="py-name">new_start</tt><tt class="py-op">,</tt> <tt class="py-name">new_end</tt><tt class="py-op">)</tt> </tt>
<a name="L900"></a><tt class="py-lineno"> 900</tt>  <tt class="py-line">            <tt class="py-name">new_locations</tt><tt class="py-op">.</tt><tt id="link-393" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-393', 'append', 'link-213');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">this_location</tt><tt class="py-op">)</tt> </tt>
<a name="L901"></a><tt class="py-lineno"> 901</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">new_locations</tt> </tt>
</div><a name="L902"></a><tt class="py-lineno"> 902</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.authors"></a><div id="_FeatureConsumer.authors-def"><a name="L903"></a><tt class="py-lineno"> 903</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.authors-toggle" onclick="return toggle('_FeatureConsumer.authors');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#authors">authors</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.authors-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.authors-expanded"><a name="L904"></a><tt class="py-lineno"> 904</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt id="link-394" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.authors()=Bio.GenBank._FeatureConsumer-class.html#authors,Method Bio.GenBank._RecordConsumer.authors()=Bio.GenBank._RecordConsumer-class.html#authors"><a title="Bio.GenBank._FeatureConsumer.authors
Bio.GenBank._RecordConsumer.authors" class="py-name" href="#" onclick="return doclink('link-394', 'authors', 'link-394');">authors</a></tt><tt class="py-op">:</tt> </tt>
<a name="L905"></a><tt class="py-lineno"> 905</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt id="link-395" class="py-name"><a title="Bio.GenBank._FeatureConsumer.authors
Bio.GenBank._RecordConsumer.authors" class="py-name" href="#" onclick="return doclink('link-395', 'authors', 'link-394');">authors</a></tt> <tt class="py-op">+=</tt> <tt class="py-string">' '</tt> <tt class="py-op">+</tt> <tt class="py-name">content</tt> </tt>
<a name="L906"></a><tt class="py-lineno"> 906</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L907"></a><tt class="py-lineno"> 907</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt id="link-396" class="py-name"><a title="Bio.GenBank._FeatureConsumer.authors
Bio.GenBank._RecordConsumer.authors" class="py-name" href="#" onclick="return doclink('link-396', 'authors', 'link-394');">authors</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L908"></a><tt class="py-lineno"> 908</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.consrtm"></a><div id="_FeatureConsumer.consrtm-def"><a name="L909"></a><tt class="py-lineno"> 909</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.consrtm-toggle" onclick="return toggle('_FeatureConsumer.consrtm');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#consrtm">consrtm</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.consrtm-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.consrtm-expanded"><a name="L910"></a><tt class="py-lineno"> 910</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt id="link-397" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.consrtm()=Bio.GenBank._FeatureConsumer-class.html#consrtm,Method Bio.GenBank._RecordConsumer.consrtm()=Bio.GenBank._RecordConsumer-class.html#consrtm"><a title="Bio.GenBank._FeatureConsumer.consrtm
Bio.GenBank._RecordConsumer.consrtm" class="py-name" href="#" onclick="return doclink('link-397', 'consrtm', 'link-397');">consrtm</a></tt><tt class="py-op">:</tt> </tt>
<a name="L911"></a><tt class="py-lineno"> 911</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt id="link-398" class="py-name"><a title="Bio.GenBank._FeatureConsumer.consrtm
Bio.GenBank._RecordConsumer.consrtm" class="py-name" href="#" onclick="return doclink('link-398', 'consrtm', 'link-397');">consrtm</a></tt> <tt class="py-op">+=</tt> <tt class="py-string">' '</tt> <tt class="py-op">+</tt> <tt class="py-name">content</tt> </tt>
<a name="L912"></a><tt class="py-lineno"> 912</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L913"></a><tt class="py-lineno"> 913</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt id="link-399" class="py-name"><a title="Bio.GenBank._FeatureConsumer.consrtm
Bio.GenBank._RecordConsumer.consrtm" class="py-name" href="#" onclick="return doclink('link-399', 'consrtm', 'link-397');">consrtm</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L914"></a><tt class="py-lineno"> 914</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.title"></a><div id="_FeatureConsumer.title-def"><a name="L915"></a><tt class="py-lineno"> 915</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.title-toggle" onclick="return toggle('_FeatureConsumer.title');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#title">title</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.title-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.title-expanded"><a name="L916"></a><tt class="py-lineno"> 916</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L917"></a><tt class="py-lineno"> 917</tt>  <tt class="py-line">            <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L918"></a><tt class="py-lineno"> 918</tt>  <tt class="py-line">            <tt class="py-keyword">from</tt> <tt id="link-400" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-400', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-401" class="py-name" targets="Class Bio.BiopythonParserWarning=Bio.BiopythonParserWarning-class.html"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-401', 'BiopythonParserWarning', 'link-401');">BiopythonParserWarning</a></tt> </tt>
<a name="L919"></a><tt class="py-lineno"> 919</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"GenBank TITLE line without REFERENCE line."</tt><tt class="py-op">,</tt> </tt>
<a name="L920"></a><tt class="py-lineno"> 920</tt>  <tt class="py-line">                          <tt id="link-402" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-402', 'BiopythonParserWarning', 'link-401');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L921"></a><tt class="py-lineno"> 921</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt id="link-403" class="py-name" targets="Method Bio.Blast.NCBIStandalone._AlignmentConsumer.title()=Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#title,Method Bio.GenBank._FeatureConsumer.title()=Bio.GenBank._FeatureConsumer-class.html#title,Method Bio.GenBank._RecordConsumer.title()=Bio.GenBank._RecordConsumer-class.html#title"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-403', 'title', 'link-403');">title</a></tt><tt class="py-op">:</tt> </tt>
<a name="L922"></a><tt class="py-lineno"> 922</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt id="link-404" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-404', 'title', 'link-403');">title</a></tt> <tt class="py-op">+=</tt> <tt class="py-string">' '</tt> <tt class="py-op">+</tt> <tt class="py-name">content</tt> </tt>
<a name="L923"></a><tt class="py-lineno"> 923</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L924"></a><tt class="py-lineno"> 924</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt id="link-405" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-405', 'title', 'link-403');">title</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L925"></a><tt class="py-lineno"> 925</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.journal"></a><div id="_FeatureConsumer.journal-def"><a name="L926"></a><tt class="py-lineno"> 926</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.journal-toggle" onclick="return toggle('_FeatureConsumer.journal');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#journal">journal</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.journal-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.journal-expanded"><a name="L927"></a><tt class="py-lineno"> 927</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt id="link-406" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.journal()=Bio.GenBank._FeatureConsumer-class.html#journal,Method Bio.GenBank._RecordConsumer.journal()=Bio.GenBank._RecordConsumer-class.html#journal"><a title="Bio.GenBank._FeatureConsumer.journal
Bio.GenBank._RecordConsumer.journal" class="py-name" href="#" onclick="return doclink('link-406', 'journal', 'link-406');">journal</a></tt><tt class="py-op">:</tt> </tt>
<a name="L928"></a><tt class="py-lineno"> 928</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt id="link-407" class="py-name"><a title="Bio.GenBank._FeatureConsumer.journal
Bio.GenBank._RecordConsumer.journal" class="py-name" href="#" onclick="return doclink('link-407', 'journal', 'link-406');">journal</a></tt> <tt class="py-op">+=</tt> <tt class="py-string">' '</tt> <tt class="py-op">+</tt> <tt class="py-name">content</tt> </tt>
<a name="L929"></a><tt class="py-lineno"> 929</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L930"></a><tt class="py-lineno"> 930</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt id="link-408" class="py-name"><a title="Bio.GenBank._FeatureConsumer.journal
Bio.GenBank._RecordConsumer.journal" class="py-name" href="#" onclick="return doclink('link-408', 'journal', 'link-406');">journal</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L931"></a><tt class="py-lineno"> 931</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.medline_id"></a><div id="_FeatureConsumer.medline_id-def"><a name="L932"></a><tt class="py-lineno"> 932</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.medline_id-toggle" onclick="return toggle('_FeatureConsumer.medline_id');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#medline_id">medline_id</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.medline_id-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.medline_id-expanded"><a name="L933"></a><tt class="py-lineno"> 933</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt id="link-409" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.medline_id()=Bio.GenBank._FeatureConsumer-class.html#medline_id,Method Bio.GenBank._RecordConsumer.medline_id()=Bio.GenBank._RecordConsumer-class.html#medline_id"><a title="Bio.GenBank._FeatureConsumer.medline_id
Bio.GenBank._RecordConsumer.medline_id" class="py-name" href="#" onclick="return doclink('link-409', 'medline_id', 'link-409');">medline_id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L934"></a><tt class="py-lineno"> 934</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.pubmed_id"></a><div id="_FeatureConsumer.pubmed_id-def"><a name="L935"></a><tt class="py-lineno"> 935</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.pubmed_id-toggle" onclick="return toggle('_FeatureConsumer.pubmed_id');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#pubmed_id">pubmed_id</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.pubmed_id-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.pubmed_id-expanded"><a name="L936"></a><tt class="py-lineno"> 936</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt id="link-410" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.pubmed_id()=Bio.GenBank._FeatureConsumer-class.html#pubmed_id,Method Bio.GenBank._RecordConsumer.pubmed_id()=Bio.GenBank._RecordConsumer-class.html#pubmed_id"><a title="Bio.GenBank._FeatureConsumer.pubmed_id
Bio.GenBank._RecordConsumer.pubmed_id" class="py-name" href="#" onclick="return doclink('link-410', 'pubmed_id', 'link-410');">pubmed_id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L937"></a><tt class="py-lineno"> 937</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.remark"></a><div id="_FeatureConsumer.remark-def"><a name="L938"></a><tt class="py-lineno"> 938</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.remark-toggle" onclick="return toggle('_FeatureConsumer.remark');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#remark">remark</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.remark-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.remark-expanded"><a name="L939"></a><tt class="py-lineno"> 939</tt>  <tt class="py-line">        <tt class="py-docstring">"""Deal with a reference comment."""</tt> </tt>
<a name="L940"></a><tt class="py-lineno"> 940</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt id="link-411" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.comment()=Bio.GenBank._FeatureConsumer-class.html#comment,Method Bio.GenBank._RecordConsumer.comment()=Bio.GenBank._RecordConsumer-class.html#comment"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-411', 'comment', 'link-411');">comment</a></tt><tt class="py-op">:</tt> </tt>
<a name="L941"></a><tt class="py-lineno"> 941</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt id="link-412" class="py-name"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-412', 'comment', 'link-411');">comment</a></tt> <tt class="py-op">+=</tt> <tt class="py-string">' '</tt> <tt class="py-op">+</tt> <tt class="py-name">content</tt> </tt>
<a name="L942"></a><tt class="py-lineno"> 942</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L943"></a><tt class="py-lineno"> 943</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt id="link-413" class="py-name"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-413', 'comment', 'link-411');">comment</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L944"></a><tt class="py-lineno"> 944</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.comment"></a><div id="_FeatureConsumer.comment-def"><a name="L945"></a><tt class="py-lineno"> 945</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.comment-toggle" onclick="return toggle('_FeatureConsumer.comment');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#comment">comment</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.comment-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.comment-expanded"><a name="L946"></a><tt class="py-lineno"> 946</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L947"></a><tt class="py-lineno"> 947</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-414" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-414', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-415" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-415', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'comment'</tt><tt class="py-op">]</tt> <tt class="py-op">+=</tt> <tt class="py-string">"\n"</tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">)</tt> </tt>
<a name="L948"></a><tt class="py-lineno"> 948</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L949"></a><tt class="py-lineno"> 949</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-416" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-416', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-417" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-417', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'comment'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">"\n"</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">)</tt> </tt>
</div><a name="L950"></a><tt class="py-lineno"> 950</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.features_line"></a><div id="_FeatureConsumer.features_line-def"><a name="L951"></a><tt class="py-lineno"> 951</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.features_line-toggle" onclick="return toggle('_FeatureConsumer.features_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#features_line">features_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.features_line-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.features_line-expanded"><a name="L952"></a><tt class="py-lineno"> 952</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get ready for the feature table when we reach the FEATURE line.</tt> </tt>
<a name="L953"></a><tt class="py-lineno"> 953</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L954"></a><tt class="py-lineno"> 954</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-418" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.start_feature_table()=Bio.GenBank._FeatureConsumer-class.html#start_feature_table,Method Bio.GenBank._RecordConsumer.start_feature_table()=Bio.GenBank._RecordConsumer-class.html#start_feature_table"><a title="Bio.GenBank._FeatureConsumer.start_feature_table
Bio.GenBank._RecordConsumer.start_feature_table" class="py-name" href="#" onclick="return doclink('link-418', 'start_feature_table', 'link-418');">start_feature_table</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L955"></a><tt class="py-lineno"> 955</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.start_feature_table"></a><div id="_FeatureConsumer.start_feature_table-def"><a name="L956"></a><tt class="py-lineno"> 956</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.start_feature_table-toggle" onclick="return toggle('_FeatureConsumer.start_feature_table');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#start_feature_table">start_feature_table</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.start_feature_table-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.start_feature_table-expanded"><a name="L957"></a><tt class="py-lineno"> 957</tt>  <tt class="py-line">        <tt class="py-docstring">"""Indicate we've got to the start of the feature table.</tt> </tt>
<a name="L958"></a><tt class="py-lineno"> 958</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L959"></a><tt class="py-lineno"> 959</tt>  <tt class="py-line">        <tt class="py-comment"># make sure we've added on our last reference object</tt> </tt>
<a name="L960"></a><tt class="py-lineno"> 960</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L961"></a><tt class="py-lineno"> 961</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-419" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-419', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-420" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-420', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'references'</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-421" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-421', 'append', 'link-213');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">)</tt> </tt>
<a name="L962"></a><tt class="py-lineno"> 962</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L963"></a><tt class="py-lineno"> 963</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.feature_key"></a><div id="_FeatureConsumer.feature_key-def"><a name="L964"></a><tt class="py-lineno"> 964</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.feature_key-toggle" onclick="return toggle('_FeatureConsumer.feature_key');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#feature_key">feature_key</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.feature_key-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.feature_key-expanded"><a name="L965"></a><tt class="py-lineno"> 965</tt>  <tt class="py-line">        <tt class="py-comment"># start a new feature</tt> </tt>
<a name="L966"></a><tt class="py-lineno"> 966</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_feature</tt> <tt class="py-op">=</tt> <tt id="link-422" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-422', 'SeqFeature', 'link-1');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-423" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-423', 'SeqFeature', 'link-1');">SeqFeature</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L967"></a><tt class="py-lineno"> 967</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_feature</tt><tt class="py-op">.</tt><tt id="link-424" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-424', 'type', 'link-274');">type</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
<a name="L968"></a><tt class="py-lineno"> 968</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-425" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-425', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-426" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.features=BioSQL.BioSeq.DBSeqRecord-class.html#features"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-426', 'features', 'link-426');">features</a></tt><tt class="py-op">.</tt><tt id="link-427" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-427', 'append', 'link-213');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_feature</tt><tt class="py-op">)</tt> </tt>
</div><a name="L969"></a><tt class="py-lineno"> 969</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.location"></a><div id="_FeatureConsumer.location-def"><a name="L970"></a><tt class="py-lineno"> 970</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.location-toggle" onclick="return toggle('_FeatureConsumer.location');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#location">location</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.location-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.location-expanded"><a name="L971"></a><tt class="py-lineno"> 971</tt>  <tt class="py-line">        <tt class="py-docstring">"""Parse out location information from the location string.</tt> </tt>
<a name="L972"></a><tt class="py-lineno"> 972</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L973"></a><tt class="py-lineno"> 973</tt>  <tt class="py-line"><tt class="py-docstring">        This uses simple Python code with some regular expressions to do the</tt> </tt>
<a name="L974"></a><tt class="py-lineno"> 974</tt>  <tt class="py-line"><tt class="py-docstring">        parsing, and then translates the results into appropriate objects.</tt> </tt>
<a name="L975"></a><tt class="py-lineno"> 975</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L976"></a><tt class="py-lineno"> 976</tt>  <tt class="py-line">        <tt class="py-comment"># clean up newlines and other whitespace inside the location before</tt> </tt>
<a name="L977"></a><tt class="py-lineno"> 977</tt>  <tt class="py-line">        <tt class="py-comment"># parsing - locations should have no whitespace whatsoever</tt> </tt>
<a name="L978"></a><tt class="py-lineno"> 978</tt>  <tt class="py-line">        <tt class="py-name">location_line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-428" class="py-name" targets="Method Bio.GenBank._BaseGenBankConsumer._clean_location()=Bio.GenBank._BaseGenBankConsumer-class.html#_clean_location"><a title="Bio.GenBank._BaseGenBankConsumer._clean_location" class="py-name" href="#" onclick="return doclink('link-428', '_clean_location', 'link-428');">_clean_location</a></tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">)</tt> </tt>
<a name="L979"></a><tt class="py-lineno"> 979</tt>  <tt class="py-line"> </tt>
<a name="L980"></a><tt class="py-lineno"> 980</tt>  <tt class="py-line">        <tt class="py-comment"># Older records have junk like replace(266,"c") in the</tt> </tt>
<a name="L981"></a><tt class="py-lineno"> 981</tt>  <tt class="py-line">        <tt class="py-comment"># location line. Newer records just replace this with</tt> </tt>
<a name="L982"></a><tt class="py-lineno"> 982</tt>  <tt class="py-line">        <tt class="py-comment"># the number 266 and have the information in a more reasonable</tt> </tt>
<a name="L983"></a><tt class="py-lineno"> 983</tt>  <tt class="py-line">        <tt class="py-comment"># place. So we'll just grab out the number and feed this to the</tt> </tt>
<a name="L984"></a><tt class="py-lineno"> 984</tt>  <tt class="py-line">        <tt class="py-comment"># parser. We shouldn't really be losing any info this way.</tt> </tt>
<a name="L985"></a><tt class="py-lineno"> 985</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">'replace'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">location_line</tt><tt class="py-op">:</tt> </tt>
<a name="L986"></a><tt class="py-lineno"> 986</tt>  <tt class="py-line">            <tt class="py-name">comma_pos</tt> <tt class="py-op">=</tt> <tt class="py-name">location_line</tt><tt class="py-op">.</tt><tt id="link-429" class="py-name"><a title="Bio.NeuralNetwork.Gene.Motif.MotifFinder.find
Bio.NeuralNetwork.Gene.Schema.SchemaFinder.find
Bio.NeuralNetwork.Gene.Signature.SignatureFinder.find
Bio.Seq.Seq.find
Bio.triefind.find" class="py-name" href="#" onclick="return doclink('link-429', 'find', 'link-186');">find</a></tt><tt class="py-op">(</tt><tt class="py-string">','</tt><tt class="py-op">)</tt> </tt>
<a name="L987"></a><tt class="py-lineno"> 987</tt>  <tt class="py-line">            <tt class="py-name">location_line</tt> <tt class="py-op">=</tt> <tt class="py-name">location_line</tt><tt class="py-op">[</tt><tt class="py-number">8</tt><tt class="py-op">:</tt><tt class="py-name">comma_pos</tt><tt class="py-op">]</tt> </tt>
<a name="L988"></a><tt class="py-lineno"> 988</tt>  <tt class="py-line"> </tt>
<a name="L989"></a><tt class="py-lineno"> 989</tt>  <tt class="py-line">        <tt class="py-name">cur_feature</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_feature</tt> </tt>
<a name="L990"></a><tt class="py-lineno"> 990</tt>  <tt class="py-line"> </tt>
<a name="L991"></a><tt class="py-lineno"> 991</tt>  <tt class="py-line">        <tt class="py-comment"># Handle top level complement here for speed</tt> </tt>
<a name="L992"></a><tt class="py-lineno"> 992</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">location_line</tt><tt class="py-op">.</tt><tt id="link-430" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-430', 'startswith', 'link-150');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"complement("</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L993"></a><tt class="py-lineno"> 993</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">location_line</tt><tt class="py-op">.</tt><tt id="link-431" class="py-name" targets="Method Bio.Seq.Seq.endswith()=Bio.Seq.Seq-class.html#endswith"><a title="Bio.Seq.Seq.endswith" class="py-name" href="#" onclick="return doclink('link-431', 'endswith', 'link-431');">endswith</a></tt><tt class="py-op">(</tt><tt class="py-string">")"</tt><tt class="py-op">)</tt> </tt>
<a name="L994"></a><tt class="py-lineno"> 994</tt>  <tt class="py-line">            <tt class="py-name">location_line</tt> <tt class="py-op">=</tt> <tt class="py-name">location_line</tt><tt class="py-op">[</tt><tt class="py-number">11</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L995"></a><tt class="py-lineno"> 995</tt>  <tt class="py-line">            <tt id="link-432" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-432', 'strand', 'link-181');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt> </tt>
<a name="L996"></a><tt class="py-lineno"> 996</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-string">"PROTEIN"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_type</tt><tt class="py-op">.</tt><tt id="link-433" class="py-name" targets="Method Bio.Seq.Seq.upper()=Bio.Seq.Seq-class.html#upper,Method Bio.Seq.UnknownSeq.upper()=Bio.Seq.UnknownSeq-class.html#upper,Method Bio.SeqRecord.SeqRecord.upper()=Bio.SeqRecord.SeqRecord-class.html#upper"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-433', 'upper', 'link-433');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L997"></a><tt class="py-lineno"> 997</tt>  <tt class="py-line">            <tt id="link-434" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-434', 'strand', 'link-181');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L998"></a><tt class="py-lineno"> 998</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L999"></a><tt class="py-lineno"> 999</tt>  <tt class="py-line">            <tt class="py-comment"># Assume nucleotide otherwise feature strand for</tt> </tt>
<a name="L1000"></a><tt class="py-lineno">1000</tt>  <tt class="py-line">            <tt class="py-comment"># GenBank files with bad LOCUS lines set to None</tt> </tt>
<a name="L1001"></a><tt class="py-lineno">1001</tt>  <tt class="py-line">            <tt id="link-435" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-435', 'strand', 'link-181');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L1002"></a><tt class="py-lineno">1002</tt>  <tt class="py-line"> </tt>
<a name="L1003"></a><tt class="py-lineno">1003</tt>  <tt class="py-line">        <tt class="py-comment"># Special case handling of the most common cases for speed</tt> </tt>
<a name="L1004"></a><tt class="py-lineno">1004</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-436" class="py-name"><a title="Bio.GenBank._re_simple_location" class="py-name" href="#" onclick="return doclink('link-436', '_re_simple_location', 'link-53');">_re_simple_location</a></tt><tt class="py-op">.</tt><tt id="link-437" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-437', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-name">location_line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1005"></a><tt class="py-lineno">1005</tt>  <tt class="py-line">            <tt class="py-comment"># e.g. "123..456"</tt> </tt>
<a name="L1006"></a><tt class="py-lineno">1006</tt>  <tt class="py-line">            <tt class="py-name">s</tt><tt class="py-op">,</tt> <tt class="py-name">e</tt> <tt class="py-op">=</tt> <tt class="py-name">location_line</tt><tt class="py-op">.</tt><tt id="link-438" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-438', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">".."</tt><tt class="py-op">)</tt> </tt>
<a name="L1007"></a><tt class="py-lineno">1007</tt>  <tt class="py-line">            <tt class="py-name">cur_feature</tt><tt class="py-op">.</tt><tt id="link-439" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-439', 'location', 'link-381');">location</a></tt> <tt class="py-op">=</tt> <tt id="link-440" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-440', 'SeqFeature', 'link-1');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-441" class="py-name"><a title="Bio.SeqFeature.FeatureLocation" class="py-name" href="#" onclick="return doclink('link-441', 'FeatureLocation', 'link-180');">FeatureLocation</a></tt><tt class="py-op">(</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> </tt>
<a name="L1008"></a><tt class="py-lineno">1008</tt>  <tt class="py-line">                                                              <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">e</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1009"></a><tt class="py-lineno">1009</tt>  <tt class="py-line">                                                              <tt id="link-442" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-442', 'strand', 'link-181');">strand</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1010"></a><tt class="py-lineno">1010</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L1011"></a><tt class="py-lineno">1011</tt>  <tt class="py-line"> </tt>
<a name="L1012"></a><tt class="py-lineno">1012</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-443" class="py-name"><a title="Bio.GenBank._solo_bond" class="py-name" href="#" onclick="return doclink('link-443', '_solo_bond', 'link-142');">_solo_bond</a></tt><tt class="py-op">.</tt><tt id="link-444" class="py-name"><a title="Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-444', 'search', 'link-147');">search</a></tt><tt class="py-op">(</tt><tt class="py-name">location_line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1013"></a><tt class="py-lineno">1013</tt>  <tt class="py-line">            <tt class="py-comment"># e.g. bond(196)</tt> </tt>
<a name="L1014"></a><tt class="py-lineno">1014</tt>  <tt class="py-line">            <tt class="py-comment"># e.g. join(bond(284),bond(305),bond(309),bond(305))</tt> </tt>
<a name="L1015"></a><tt class="py-lineno">1015</tt>  <tt class="py-line">            <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L1016"></a><tt class="py-lineno">1016</tt>  <tt class="py-line">            <tt class="py-keyword">from</tt> <tt id="link-445" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-445', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-446" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-446', 'BiopythonParserWarning', 'link-401');">BiopythonParserWarning</a></tt> </tt>
<a name="L1017"></a><tt class="py-lineno">1017</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Dropping bond qualifier in feature location"</tt><tt class="py-op">,</tt> <tt id="link-447" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-447', 'BiopythonParserWarning', 'link-401');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1018"></a><tt class="py-lineno">1018</tt>  <tt class="py-line">            <tt class="py-comment"># There ought to be a better way to do this...</tt> </tt>
<a name="L1019"></a><tt class="py-lineno">1019</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt id="link-448" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-448', 'x', 'link-235');">x</a></tt> <tt class="py-keyword">in</tt> <tt id="link-449" class="py-name"><a title="Bio.GenBank._solo_bond" class="py-name" href="#" onclick="return doclink('link-449', '_solo_bond', 'link-142');">_solo_bond</a></tt><tt class="py-op">.</tt><tt id="link-450" class="py-name" targets="Method Bio.Restriction.Restriction.FormattedSeq.finditer()=Bio.Restriction.Restriction.FormattedSeq-class.html#finditer"><a title="Bio.Restriction.Restriction.FormattedSeq.finditer" class="py-name" href="#" onclick="return doclink('link-450', 'finditer', 'link-450');">finditer</a></tt><tt class="py-op">(</tt><tt class="py-name">location_line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1020"></a><tt class="py-lineno">1020</tt>  <tt class="py-line">                <tt id="link-451" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-451', 'x', 'link-235');">x</a></tt> <tt class="py-op">=</tt> <tt id="link-452" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-452', 'x', 'link-235');">x</a></tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1021"></a><tt class="py-lineno">1021</tt>  <tt class="py-line">                <tt class="py-name">location_line</tt> <tt class="py-op">=</tt> <tt class="py-name">location_line</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt id="link-453" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-453', 'x', 'link-235');">x</a></tt><tt class="py-op">,</tt> <tt id="link-454" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-454', 'x', 'link-235');">x</a></tt><tt class="py-op">[</tt><tt class="py-number">5</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1022"></a><tt class="py-lineno">1022</tt>  <tt class="py-line"> </tt>
<a name="L1023"></a><tt class="py-lineno">1023</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-455" class="py-name"><a title="Bio.GenBank._re_simple_compound" class="py-name" href="#" onclick="return doclink('link-455', '_re_simple_compound', 'link-55');">_re_simple_compound</a></tt><tt class="py-op">.</tt><tt id="link-456" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-456', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-name">location_line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1024"></a><tt class="py-lineno">1024</tt>  <tt class="py-line">            <tt class="py-comment"># e.g. join(&lt;123..456,480..&gt;500)</tt> </tt>
<a name="L1025"></a><tt class="py-lineno">1025</tt>  <tt class="py-line">            <tt id="link-457" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-457', 'i', 'link-185');">i</a></tt> <tt class="py-op">=</tt> <tt class="py-name">location_line</tt><tt class="py-op">.</tt><tt id="link-458" class="py-name"><a title="Bio.NeuralNetwork.Gene.Motif.MotifFinder.find
Bio.NeuralNetwork.Gene.Schema.SchemaFinder.find
Bio.NeuralNetwork.Gene.Signature.SignatureFinder.find
Bio.Seq.Seq.find
Bio.triefind.find" class="py-name" href="#" onclick="return doclink('link-458', 'find', 'link-186');">find</a></tt><tt class="py-op">(</tt><tt class="py-string">"("</tt><tt class="py-op">)</tt> </tt>
<a name="L1026"></a><tt class="py-lineno">1026</tt>  <tt class="py-line">            <tt class="py-comment"># cur_feature.location_operator = location_line[:i]</tt> </tt>
<a name="L1027"></a><tt class="py-lineno">1027</tt>  <tt class="py-line">            <tt class="py-comment"># we can split on the comma because these are simple locations</tt> </tt>
<a name="L1028"></a><tt class="py-lineno">1028</tt>  <tt class="py-line">            <tt id="link-459" class="py-name" targets="Variable Bio.SeqFeature.SeqFeature.sub_features=Bio.SeqFeature.SeqFeature-class.html#sub_features"><a title="Bio.SeqFeature.SeqFeature.sub_features" class="py-name" href="#" onclick="return doclink('link-459', 'sub_features', 'link-459');">sub_features</a></tt> <tt class="py-op">=</tt> <tt class="py-name">cur_feature</tt><tt class="py-op">.</tt><tt id="link-460" class="py-name"><a title="Bio.SeqFeature.SeqFeature.sub_features" class="py-name" href="#" onclick="return doclink('link-460', 'sub_features', 'link-459');">sub_features</a></tt> </tt>
<a name="L1029"></a><tt class="py-lineno">1029</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">part</tt> <tt class="py-keyword">in</tt> <tt class="py-name">location_line</tt><tt class="py-op">[</tt><tt id="link-461" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-461', 'i', 'link-185');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-462" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-462', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">","</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1030"></a><tt class="py-lineno">1030</tt>  <tt class="py-line">                <tt class="py-name">s</tt><tt class="py-op">,</tt> <tt class="py-name">e</tt> <tt class="py-op">=</tt> <tt class="py-name">part</tt><tt class="py-op">.</tt><tt id="link-463" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-463', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">".."</tt><tt class="py-op">)</tt> </tt>
<a name="L1031"></a><tt class="py-lineno">1031</tt>  <tt class="py-line">                <tt class="py-name">f</tt> <tt class="py-op">=</tt> <tt id="link-464" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-464', 'SeqFeature', 'link-1');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-465" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-465', 'SeqFeature', 'link-1');">SeqFeature</a></tt><tt class="py-op">(</tt><tt id="link-466" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-466', 'SeqFeature', 'link-1');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-467" class="py-name"><a title="Bio.SeqFeature.FeatureLocation" class="py-name" href="#" onclick="return doclink('link-467', 'FeatureLocation', 'link-180');">FeatureLocation</a></tt><tt class="py-op">(</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> </tt>
<a name="L1032"></a><tt class="py-lineno">1032</tt>  <tt class="py-line">                                                                     <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">e</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1033"></a><tt class="py-lineno">1033</tt>  <tt class="py-line">                                                                     <tt id="link-468" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-468', 'strand', 'link-181');">strand</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1034"></a><tt class="py-lineno">1034</tt>  <tt class="py-line">                        <tt id="link-469" class="py-name" targets="Variable Bio.SeqFeature.SeqFeature.location_operator=Bio.SeqFeature.SeqFeature-class.html#location_operator"><a title="Bio.SeqFeature.SeqFeature.location_operator" class="py-name" href="#" onclick="return doclink('link-469', 'location_operator', 'link-469');">location_operator</a></tt><tt class="py-op">=</tt><tt class="py-name">cur_feature</tt><tt class="py-op">.</tt><tt id="link-470" class="py-name"><a title="Bio.SeqFeature.SeqFeature.location_operator" class="py-name" href="#" onclick="return doclink('link-470', 'location_operator', 'link-469');">location_operator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1035"></a><tt class="py-lineno">1035</tt>  <tt class="py-line">                        <tt id="link-471" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-471', 'type', 'link-274');">type</a></tt><tt class="py-op">=</tt><tt class="py-name">cur_feature</tt><tt class="py-op">.</tt><tt id="link-472" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-472', 'type', 'link-274');">type</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1036"></a><tt class="py-lineno">1036</tt>  <tt class="py-line">                <tt id="link-473" class="py-name"><a title="Bio.SeqFeature.SeqFeature.sub_features" class="py-name" href="#" onclick="return doclink('link-473', 'sub_features', 'link-459');">sub_features</a></tt><tt class="py-op">.</tt><tt id="link-474" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-474', 'append', 'link-213');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">f</tt><tt class="py-op">)</tt> </tt>
<a name="L1037"></a><tt class="py-lineno">1037</tt>  <tt class="py-line">            <tt class="py-comment"># s = cur_feature.sub_features[0].location.start</tt> </tt>
<a name="L1038"></a><tt class="py-lineno">1038</tt>  <tt class="py-line">            <tt class="py-comment"># e = cur_feature.sub_features[-1].location.end</tt> </tt>
<a name="L1039"></a><tt class="py-lineno">1039</tt>  <tt class="py-line">            <tt class="py-comment"># cur_feature.location = SeqFeature.FeatureLocation(s,e, strand)</tt> </tt>
<a name="L1040"></a><tt class="py-lineno">1040</tt>  <tt class="py-line">            <tt class="py-comment"># TODO - Remove use of sub_features</tt> </tt>
<a name="L1041"></a><tt class="py-lineno">1041</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-475" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-475', 'strand', 'link-181');">strand</a></tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L1042"></a><tt class="py-lineno">1042</tt>  <tt class="py-line">                <tt class="py-name">cur_feature</tt><tt class="py-op">.</tt><tt id="link-476" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-476', 'location', 'link-381');">location</a></tt> <tt class="py-op">=</tt> <tt id="link-477" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-477', 'SeqFeature', 'link-1');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-478" class="py-name" targets="Class Bio.SeqFeature.CompoundLocation=Bio.SeqFeature.CompoundLocation-class.html"><a title="Bio.SeqFeature.CompoundLocation" class="py-name" href="#" onclick="return doclink('link-478', 'CompoundLocation', 'link-478');">CompoundLocation</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">f</tt><tt class="py-op">.</tt><tt id="link-479" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-479', 'location', 'link-381');">location</a></tt> <tt class="py-keyword">for</tt> <tt class="py-name">f</tt> <tt class="py-keyword">in</tt> <tt id="link-480" class="py-name"><a title="Bio.SeqFeature.SeqFeature.sub_features" class="py-name" href="#" onclick="return doclink('link-480', 'sub_features', 'link-459');">sub_features</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L1043"></a><tt class="py-lineno">1043</tt>  <tt class="py-line">                                                                   <tt class="py-name">operator</tt><tt class="py-op">=</tt><tt class="py-name">location_line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt id="link-481" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-481', 'i', 'link-185');">i</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1044"></a><tt class="py-lineno">1044</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1045"></a><tt class="py-lineno">1045</tt>  <tt class="py-line">                <tt class="py-name">cur_feature</tt><tt class="py-op">.</tt><tt id="link-482" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-482', 'location', 'link-381');">location</a></tt> <tt class="py-op">=</tt> <tt id="link-483" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-483', 'SeqFeature', 'link-1');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-484" class="py-name"><a title="Bio.SeqFeature.CompoundLocation" class="py-name" href="#" onclick="return doclink('link-484', 'CompoundLocation', 'link-478');">CompoundLocation</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">f</tt><tt class="py-op">.</tt><tt id="link-485" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-485', 'location', 'link-381');">location</a></tt> <tt class="py-keyword">for</tt> <tt class="py-name">f</tt> <tt class="py-keyword">in</tt> <tt id="link-486" class="py-name"><a title="Bio.SeqFeature.SeqFeature.sub_features" class="py-name" href="#" onclick="return doclink('link-486', 'sub_features', 'link-459');">sub_features</a></tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L1046"></a><tt class="py-lineno">1046</tt>  <tt class="py-line">                                                                   <tt class="py-name">operator</tt><tt class="py-op">=</tt><tt class="py-name">location_line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt id="link-487" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-487', 'i', 'link-185');">i</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1047"></a><tt class="py-lineno">1047</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L1048"></a><tt class="py-lineno">1048</tt>  <tt class="py-line"> </tt>
<a name="L1049"></a><tt class="py-lineno">1049</tt>  <tt class="py-line">        <tt class="py-comment"># Handle the general case with more complex regular expressions</tt> </tt>
<a name="L1050"></a><tt class="py-lineno">1050</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-488" class="py-name"><a title="Bio.GenBank._re_complex_location" class="py-name" href="#" onclick="return doclink('link-488', '_re_complex_location', 'link-64');">_re_complex_location</a></tt><tt class="py-op">.</tt><tt id="link-489" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-489', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-name">location_line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1051"></a><tt class="py-lineno">1051</tt>  <tt class="py-line">            <tt class="py-comment"># e.g. "AL121804.2:41..610"</tt> </tt>
<a name="L1052"></a><tt class="py-lineno">1052</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-string">":"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">location_line</tt><tt class="py-op">:</tt> </tt>
<a name="L1053"></a><tt class="py-lineno">1053</tt>  <tt class="py-line">                <tt class="py-name">location_ref</tt><tt class="py-op">,</tt> <tt class="py-name">location_line</tt> <tt class="py-op">=</tt> <tt class="py-name">location_line</tt><tt class="py-op">.</tt><tt id="link-490" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-490', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">":"</tt><tt class="py-op">)</tt> </tt>
<a name="L1054"></a><tt class="py-lineno">1054</tt>  <tt class="py-line">                <tt class="py-name">cur_feature</tt><tt class="py-op">.</tt><tt id="link-491" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-491', 'location', 'link-381');">location</a></tt> <tt class="py-op">=</tt> <tt class="py-name">_loc</tt><tt class="py-op">(</tt><tt class="py-name">location_line</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_expected_size</tt><tt class="py-op">,</tt> <tt id="link-492" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-492', 'strand', 'link-181');">strand</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1055"></a><tt class="py-lineno">1055</tt>  <tt class="py-line">                <tt class="py-name">cur_feature</tt><tt class="py-op">.</tt><tt id="link-493" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-493', 'location', 'link-381');">location</a></tt><tt class="py-op">.</tt><tt id="link-494" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.ref=Bio.SeqFeature.CompoundLocation-class.html#ref,Variable Bio.SeqFeature.SeqFeature.ref=Bio.SeqFeature.SeqFeature-class.html#ref"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-494', 'ref', 'link-494');">ref</a></tt> <tt class="py-op">=</tt> <tt class="py-name">location_ref</tt> </tt>
<a name="L1056"></a><tt class="py-lineno">1056</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1057"></a><tt class="py-lineno">1057</tt>  <tt class="py-line">                <tt class="py-name">cur_feature</tt><tt class="py-op">.</tt><tt id="link-495" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-495', 'location', 'link-381');">location</a></tt> <tt class="py-op">=</tt> <tt class="py-name">_loc</tt><tt class="py-op">(</tt><tt class="py-name">location_line</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_expected_size</tt><tt class="py-op">,</tt> <tt id="link-496" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-496', 'strand', 'link-181');">strand</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1058"></a><tt class="py-lineno">1058</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L1059"></a><tt class="py-lineno">1059</tt>  <tt class="py-line"> </tt>
<a name="L1060"></a><tt class="py-lineno">1060</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-497" class="py-name"><a title="Bio.GenBank._re_complex_compound" class="py-name" href="#" onclick="return doclink('link-497', '_re_complex_compound', 'link-69');">_re_complex_compound</a></tt><tt class="py-op">.</tt><tt id="link-498" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-498', 'match', 'link-25');">match</a></tt><tt class="py-op">(</tt><tt class="py-name">location_line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1061"></a><tt class="py-lineno">1061</tt>  <tt class="py-line">            <tt id="link-499" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-499', 'i', 'link-185');">i</a></tt> <tt class="py-op">=</tt> <tt class="py-name">location_line</tt><tt class="py-op">.</tt><tt id="link-500" class="py-name"><a title="Bio.NeuralNetwork.Gene.Motif.MotifFinder.find
Bio.NeuralNetwork.Gene.Schema.SchemaFinder.find
Bio.NeuralNetwork.Gene.Signature.SignatureFinder.find
Bio.Seq.Seq.find
Bio.triefind.find" class="py-name" href="#" onclick="return doclink('link-500', 'find', 'link-186');">find</a></tt><tt class="py-op">(</tt><tt class="py-string">"("</tt><tt class="py-op">)</tt> </tt>
<a name="L1062"></a><tt class="py-lineno">1062</tt>  <tt class="py-line">            <tt class="py-comment"># cur_feature.location_operator = location_line[:i]</tt> </tt>
<a name="L1063"></a><tt class="py-lineno">1063</tt>  <tt class="py-line">            <tt class="py-comment"># Can't split on the comma because of positions like one-of(1,2,3)</tt> </tt>
<a name="L1064"></a><tt class="py-lineno">1064</tt>  <tt class="py-line">            <tt id="link-501" class="py-name"><a title="Bio.SeqFeature.SeqFeature.sub_features" class="py-name" href="#" onclick="return doclink('link-501', 'sub_features', 'link-459');">sub_features</a></tt> <tt class="py-op">=</tt> <tt class="py-name">cur_feature</tt><tt class="py-op">.</tt><tt id="link-502" class="py-name"><a title="Bio.SeqFeature.SeqFeature.sub_features" class="py-name" href="#" onclick="return doclink('link-502', 'sub_features', 'link-459');">sub_features</a></tt> </tt>
<a name="L1065"></a><tt class="py-lineno">1065</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">part</tt> <tt class="py-keyword">in</tt> <tt class="py-name">_split_compound_loc</tt><tt class="py-op">(</tt><tt class="py-name">location_line</tt><tt class="py-op">[</tt><tt id="link-503" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-503', 'i', 'link-185');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1066"></a><tt class="py-lineno">1066</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">part</tt><tt class="py-op">.</tt><tt id="link-504" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-504', 'startswith', 'link-150');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"complement("</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1067"></a><tt class="py-lineno">1067</tt>  <tt class="py-line">                    <tt class="py-keyword">assert</tt> <tt class="py-name">part</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">")"</tt> </tt>
<a name="L1068"></a><tt class="py-lineno">1068</tt>  <tt class="py-line">                    <tt class="py-name">part</tt> <tt class="py-op">=</tt> <tt class="py-name">part</tt><tt class="py-op">[</tt><tt class="py-number">11</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L1069"></a><tt class="py-lineno">1069</tt>  <tt class="py-line">                    <tt class="py-keyword">assert</tt> <tt id="link-505" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-505', 'strand', 'link-181');">strand</a></tt> <tt class="py-op">!=</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-string">"Double complement?"</tt> </tt>
<a name="L1070"></a><tt class="py-lineno">1070</tt>  <tt class="py-line">                    <tt class="py-name">part_strand</tt> <tt class="py-op">=</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt> </tt>
<a name="L1071"></a><tt class="py-lineno">1071</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1072"></a><tt class="py-lineno">1072</tt>  <tt class="py-line">                    <tt class="py-name">part_strand</tt> <tt class="py-op">=</tt> <tt id="link-506" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-506', 'strand', 'link-181');">strand</a></tt> </tt>
<a name="L1073"></a><tt class="py-lineno">1073</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-string">":"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">part</tt><tt class="py-op">:</tt> </tt>
<a name="L1074"></a><tt class="py-lineno">1074</tt>  <tt class="py-line">                    <tt id="link-507" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-507', 'ref', 'link-494');">ref</a></tt><tt class="py-op">,</tt> <tt class="py-name">part</tt> <tt class="py-op">=</tt> <tt class="py-name">part</tt><tt class="py-op">.</tt><tt id="link-508" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-508', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">":"</tt><tt class="py-op">)</tt> </tt>
<a name="L1075"></a><tt class="py-lineno">1075</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1076"></a><tt class="py-lineno">1076</tt>  <tt class="py-line">                    <tt id="link-509" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-509', 'ref', 'link-494');">ref</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L1077"></a><tt class="py-lineno">1077</tt>  <tt class="py-line">                <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1078"></a><tt class="py-lineno">1078</tt>  <tt class="py-line">                    <tt class="py-name">loc</tt> <tt class="py-op">=</tt> <tt class="py-name">_loc</tt><tt class="py-op">(</tt><tt class="py-name">part</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_expected_size</tt><tt class="py-op">,</tt> <tt class="py-name">part_strand</tt><tt class="py-op">)</tt> </tt>
<a name="L1079"></a><tt class="py-lineno">1079</tt>  <tt class="py-line">                <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt> <tt class="py-keyword">as</tt> <tt class="py-name">err</tt><tt class="py-op">:</tt> </tt>
<a name="L1080"></a><tt class="py-lineno">1080</tt>  <tt class="py-line">                    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">location_line</tt><tt class="py-op">)</tt> </tt>
<a name="L1081"></a><tt class="py-lineno">1081</tt>  <tt class="py-line">                    <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-name">part</tt><tt class="py-op">)</tt> </tt>
<a name="L1082"></a><tt class="py-lineno">1082</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt class="py-name">err</tt> </tt>
<a name="L1083"></a><tt class="py-lineno">1083</tt>  <tt class="py-line">                <tt class="py-name">f</tt> <tt class="py-op">=</tt> <tt id="link-510" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-510', 'SeqFeature', 'link-1');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-511" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-511', 'SeqFeature', 'link-1');">SeqFeature</a></tt><tt class="py-op">(</tt><tt id="link-512" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-512', 'location', 'link-381');">location</a></tt><tt class="py-op">=</tt><tt class="py-name">loc</tt><tt class="py-op">,</tt> <tt id="link-513" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-513', 'ref', 'link-494');">ref</a></tt><tt class="py-op">=</tt><tt id="link-514" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-514', 'ref', 'link-494');">ref</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1084"></a><tt class="py-lineno">1084</tt>  <tt class="py-line">                        <tt id="link-515" class="py-name"><a title="Bio.SeqFeature.SeqFeature.location_operator" class="py-name" href="#" onclick="return doclink('link-515', 'location_operator', 'link-469');">location_operator</a></tt><tt class="py-op">=</tt><tt class="py-name">cur_feature</tt><tt class="py-op">.</tt><tt id="link-516" class="py-name"><a title="Bio.SeqFeature.SeqFeature.location_operator" class="py-name" href="#" onclick="return doclink('link-516', 'location_operator', 'link-469');">location_operator</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1085"></a><tt class="py-lineno">1085</tt>  <tt class="py-line">                        <tt id="link-517" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-517', 'type', 'link-274');">type</a></tt><tt class="py-op">=</tt><tt class="py-name">cur_feature</tt><tt class="py-op">.</tt><tt id="link-518" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-518', 'type', 'link-274');">type</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1086"></a><tt class="py-lineno">1086</tt>  <tt class="py-line">                <tt id="link-519" class="py-name"><a title="Bio.SeqFeature.SeqFeature.sub_features" class="py-name" href="#" onclick="return doclink('link-519', 'sub_features', 'link-459');">sub_features</a></tt><tt class="py-op">.</tt><tt id="link-520" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-520', 'append', 'link-213');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">f</tt><tt class="py-op">)</tt> </tt>
<a name="L1087"></a><tt class="py-lineno">1087</tt>  <tt class="py-line">            <tt class="py-comment"># Historically a join on the reverse strand has been represented</tt> </tt>
<a name="L1088"></a><tt class="py-lineno">1088</tt>  <tt class="py-line">            <tt class="py-comment"># in Biopython with both the parent SeqFeature and its children</tt> </tt>
<a name="L1089"></a><tt class="py-lineno">1089</tt>  <tt class="py-line">            <tt class="py-comment"># (the exons for a CDS) all given a strand of -1.  Likewise, for</tt> </tt>
<a name="L1090"></a><tt class="py-lineno">1090</tt>  <tt class="py-line">            <tt class="py-comment"># a join feature on the forward strand they all have strand +1.</tt> </tt>
<a name="L1091"></a><tt class="py-lineno">1091</tt>  <tt class="py-line">            <tt class="py-comment"># However, we must also consider evil mixed strand examples like</tt> </tt>
<a name="L1092"></a><tt class="py-lineno">1092</tt>  <tt class="py-line">            <tt class="py-comment"># this, join(complement(69611..69724),139856..140087,140625..140650)</tt> </tt>
<a name="L1093"></a><tt class="py-lineno">1093</tt>  <tt class="py-line">            <tt class="py-comment">#</tt> </tt>
<a name="L1094"></a><tt class="py-lineno">1094</tt>  <tt class="py-line">            <tt class="py-comment"># TODO - Remove use of sub_features</tt> </tt>
<a name="L1095"></a><tt class="py-lineno">1095</tt>  <tt class="py-line">            <tt class="py-name">strands</tt> <tt class="py-op">=</tt> <tt id="link-521" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.set()=Bio.Nexus.Nexus.StepMatrix-class.html#set,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.set()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#set,Method Bio.SVDSuperimposer.SVDSuperimposer.set()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#set"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-521', 'set', 'link-521');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">sf</tt><tt class="py-op">.</tt><tt id="link-522" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-522', 'strand', 'link-181');">strand</a></tt> <tt class="py-keyword">for</tt> <tt class="py-name">sf</tt> <tt class="py-keyword">in</tt> <tt id="link-523" class="py-name"><a title="Bio.SeqFeature.SeqFeature.sub_features" class="py-name" href="#" onclick="return doclink('link-523', 'sub_features', 'link-459');">sub_features</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1096"></a><tt class="py-lineno">1096</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-524" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-524', 'strand', 'link-181');">strand</a></tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L1097"></a><tt class="py-lineno">1097</tt>  <tt class="py-line">                <tt class="py-comment"># Whole thing was wrapped in complement(...)</tt> </tt>
<a name="L1098"></a><tt class="py-lineno">1098</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">sf</tt> <tt class="py-keyword">in</tt> <tt id="link-525" class="py-name"><a title="Bio.SeqFeature.SeqFeature.sub_features" class="py-name" href="#" onclick="return doclink('link-525', 'sub_features', 'link-459');">sub_features</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1099"></a><tt class="py-lineno">1099</tt>  <tt class="py-line">                    <tt class="py-keyword">assert</tt> <tt class="py-name">sf</tt><tt class="py-op">.</tt><tt id="link-526" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-526', 'strand', 'link-181');">strand</a></tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt> </tt>
<a name="L1100"></a><tt class="py-lineno">1100</tt>  <tt class="py-line">                <tt class="py-comment"># Reverse the backwards order used in GenBank files</tt> </tt>
<a name="L1101"></a><tt class="py-lineno">1101</tt>  <tt class="py-line">                <tt class="py-comment"># with complement(join(...))</tt> </tt>
<a name="L1102"></a><tt class="py-lineno">1102</tt>  <tt class="py-line">                <tt class="py-name">cur_feature</tt><tt class="py-op">.</tt><tt id="link-527" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-527', 'location', 'link-381');">location</a></tt> <tt class="py-op">=</tt> <tt id="link-528" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-528', 'SeqFeature', 'link-1');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-529" class="py-name"><a title="Bio.SeqFeature.CompoundLocation" class="py-name" href="#" onclick="return doclink('link-529', 'CompoundLocation', 'link-478');">CompoundLocation</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">f</tt><tt class="py-op">.</tt><tt id="link-530" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-530', 'location', 'link-381');">location</a></tt> <tt class="py-keyword">for</tt> <tt class="py-name">f</tt> <tt class="py-keyword">in</tt> <tt id="link-531" class="py-name"><a title="Bio.SeqFeature.SeqFeature.sub_features" class="py-name" href="#" onclick="return doclink('link-531', 'sub_features', 'link-459');">sub_features</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L1103"></a><tt class="py-lineno">1103</tt>  <tt class="py-line">                                                                   <tt class="py-name">operator</tt><tt class="py-op">=</tt><tt class="py-name">location_line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt id="link-532" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-532', 'i', 'link-185');">i</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1104"></a><tt class="py-lineno">1104</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1105"></a><tt class="py-lineno">1105</tt>  <tt class="py-line">                <tt class="py-name">cur_feature</tt><tt class="py-op">.</tt><tt id="link-533" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-533', 'location', 'link-381');">location</a></tt> <tt class="py-op">=</tt> <tt id="link-534" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-534', 'SeqFeature', 'link-1');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-535" class="py-name"><a title="Bio.SeqFeature.CompoundLocation" class="py-name" href="#" onclick="return doclink('link-535', 'CompoundLocation', 'link-478');">CompoundLocation</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">f</tt><tt class="py-op">.</tt><tt id="link-536" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-536', 'location', 'link-381');">location</a></tt> <tt class="py-keyword">for</tt> <tt class="py-name">f</tt> <tt class="py-keyword">in</tt> <tt id="link-537" class="py-name"><a title="Bio.SeqFeature.SeqFeature.sub_features" class="py-name" href="#" onclick="return doclink('link-537', 'sub_features', 'link-459');">sub_features</a></tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L1106"></a><tt class="py-lineno">1106</tt>  <tt class="py-line">                                                                   <tt class="py-name">operator</tt><tt class="py-op">=</tt><tt class="py-name">location_line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt id="link-538" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-538', 'i', 'link-185');">i</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1107"></a><tt class="py-lineno">1107</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L1108"></a><tt class="py-lineno">1108</tt>  <tt class="py-line">        <tt class="py-comment"># Not recognised</tt> </tt>
<a name="L1109"></a><tt class="py-lineno">1109</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"order"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">location_line</tt> <tt class="py-keyword">and</tt> <tt class="py-string">"join"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">location_line</tt><tt class="py-op">:</tt> </tt>
<a name="L1110"></a><tt class="py-lineno">1110</tt>  <tt class="py-line">            <tt class="py-comment"># See Bug 3197</tt> </tt>
<a name="L1111"></a><tt class="py-lineno">1111</tt>  <tt class="py-line">            <tt class="py-name">msg</tt> <tt class="py-op">=</tt> <tt class="py-string">'Combinations of "join" and "order" within the same '</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L1112"></a><tt class="py-lineno">1112</tt>  <tt class="py-line">                  <tt class="py-string">'location (nested operators) are illegal:\n'</tt> <tt class="py-op">+</tt> <tt class="py-name">location_line</tt> </tt>
<a name="L1113"></a><tt class="py-lineno">1113</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-539" class="py-name" targets="Class Bio.GenBank.LocationParserError=Bio.GenBank.LocationParserError-class.html"><a title="Bio.GenBank.LocationParserError" class="py-name" href="#" onclick="return doclink('link-539', 'LocationParserError', 'link-539');">LocationParserError</a></tt><tt class="py-op">(</tt><tt class="py-name">msg</tt><tt class="py-op">)</tt> </tt>
<a name="L1114"></a><tt class="py-lineno">1114</tt>  <tt class="py-line">        <tt class="py-comment"># This used to be an error....</tt> </tt>
<a name="L1115"></a><tt class="py-lineno">1115</tt>  <tt class="py-line">        <tt class="py-name">cur_feature</tt><tt class="py-op">.</tt><tt id="link-540" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-540', 'location', 'link-381');">location</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L1116"></a><tt class="py-lineno">1116</tt>  <tt class="py-line">        <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L1117"></a><tt class="py-lineno">1117</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt id="link-541" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-541', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-542" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-542', 'BiopythonParserWarning', 'link-401');">BiopythonParserWarning</a></tt> </tt>
<a name="L1118"></a><tt class="py-lineno">1118</tt>  <tt class="py-line">        <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt id="link-543" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-543', 'BiopythonParserWarning', 'link-401');">BiopythonParserWarning</a></tt><tt class="py-op">(</tt><tt class="py-string">"Couldn't parse feature location: %r"</tt> </tt>
<a name="L1119"></a><tt class="py-lineno">1119</tt>  <tt class="py-line">                                             <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">location_line</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1120"></a><tt class="py-lineno">1120</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.feature_qualifier"></a><div id="_FeatureConsumer.feature_qualifier-def"><a name="L1121"></a><tt class="py-lineno">1121</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.feature_qualifier-toggle" onclick="return toggle('_FeatureConsumer.feature_qualifier');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#feature_qualifier">feature_qualifier</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">key</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.feature_qualifier-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.feature_qualifier-expanded"><a name="L1122"></a><tt class="py-lineno">1122</tt>  <tt class="py-line">        <tt class="py-docstring">"""When we get a qualifier key and its value.</tt> </tt>
<a name="L1123"></a><tt class="py-lineno">1123</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1124"></a><tt class="py-lineno">1124</tt>  <tt class="py-line"><tt class="py-docstring">        Can receive None, since you can have valueless keys such as /pseudo</tt> </tt>
<a name="L1125"></a><tt class="py-lineno">1125</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1126"></a><tt class="py-lineno">1126</tt>  <tt class="py-line">        <tt class="py-comment"># Hack to try to preserve historical behaviour of /pseudo etc</tt> </tt>
<a name="L1127"></a><tt class="py-lineno">1127</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-544" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.value()=Bio.Phylo.PhyloXMLIO.Writer-class.html#value"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-544', 'value', 'link-544');">value</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1128"></a><tt class="py-lineno">1128</tt>  <tt class="py-line">            <tt class="py-comment"># if the key doesn't exist yet, add an empty string</tt> </tt>
<a name="L1129"></a><tt class="py-lineno">1129</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">key</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_feature</tt><tt class="py-op">.</tt><tt class="py-name">qualifiers</tt><tt class="py-op">:</tt> </tt>
<a name="L1130"></a><tt class="py-lineno">1130</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_feature</tt><tt class="py-op">.</tt><tt class="py-name">qualifiers</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">""</tt><tt class="py-op">]</tt> </tt>
<a name="L1131"></a><tt class="py-lineno">1131</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> </tt>
<a name="L1132"></a><tt class="py-lineno">1132</tt>  <tt class="py-line">            <tt class="py-comment"># otherwise just skip this key</tt> </tt>
<a name="L1133"></a><tt class="py-lineno">1133</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L1134"></a><tt class="py-lineno">1134</tt>  <tt class="py-line"> </tt>
<a name="L1135"></a><tt class="py-lineno">1135</tt>  <tt class="py-line">        <tt id="link-545" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-545', 'value', 'link-544');">value</a></tt> <tt class="py-op">=</tt> <tt id="link-546" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-546', 'value', 'link-544');">value</a></tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'"'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L1136"></a><tt class="py-lineno">1136</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_feature_cleaner</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1137"></a><tt class="py-lineno">1137</tt>  <tt class="py-line">            <tt id="link-547" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-547', 'value', 'link-544');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_feature_cleaner</tt><tt class="py-op">.</tt><tt id="link-548" class="py-name" targets="Method Bio.GenBank.utils.FeatureValueCleaner.clean_value()=Bio.GenBank.utils.FeatureValueCleaner-class.html#clean_value"><a title="Bio.GenBank.utils.FeatureValueCleaner.clean_value" class="py-name" href="#" onclick="return doclink('link-548', 'clean_value', 'link-548');">clean_value</a></tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">,</tt> <tt id="link-549" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-549', 'value', 'link-544');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1138"></a><tt class="py-lineno">1138</tt>  <tt class="py-line"> </tt>
<a name="L1139"></a><tt class="py-lineno">1139</tt>  <tt class="py-line">        <tt class="py-comment"># if the qualifier name exists, append the value</tt> </tt>
<a name="L1140"></a><tt class="py-lineno">1140</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_feature</tt><tt class="py-op">.</tt><tt class="py-name">qualifiers</tt><tt class="py-op">:</tt> </tt>
<a name="L1141"></a><tt class="py-lineno">1141</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_feature</tt><tt class="py-op">.</tt><tt class="py-name">qualifiers</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-550" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-550', 'append', 'link-213');">append</a></tt><tt class="py-op">(</tt><tt id="link-551" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-551', 'value', 'link-544');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1142"></a><tt class="py-lineno">1142</tt>  <tt class="py-line">        <tt class="py-comment"># otherwise start a new list of the key with its values</tt> </tt>
<a name="L1143"></a><tt class="py-lineno">1143</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1144"></a><tt class="py-lineno">1144</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_feature</tt><tt class="py-op">.</tt><tt class="py-name">qualifiers</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-552" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-552', 'value', 'link-544');">value</a></tt><tt class="py-op">]</tt> </tt>
</div><a name="L1145"></a><tt class="py-lineno">1145</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.feature_qualifier_name"></a><div id="_FeatureConsumer.feature_qualifier_name-def"><a name="L1146"></a><tt class="py-lineno">1146</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.feature_qualifier_name-toggle" onclick="return toggle('_FeatureConsumer.feature_qualifier_name');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#feature_qualifier_name">feature_qualifier_name</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content_list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.feature_qualifier_name-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.feature_qualifier_name-expanded"><a name="L1147"></a><tt class="py-lineno">1147</tt>  <tt class="py-line">        <tt class="py-docstring">"""Use feature_qualifier instead (OBSOLETE)."""</tt> </tt>
<a name="L1148"></a><tt class="py-lineno">1148</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">NotImplementedError</tt><tt class="py-op">(</tt><tt class="py-string">"Use the feature_qualifier method instead."</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1149"></a><tt class="py-lineno">1149</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.feature_qualifier_description"></a><div id="_FeatureConsumer.feature_qualifier_description-def"><a name="L1150"></a><tt class="py-lineno">1150</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.feature_qualifier_description-toggle" onclick="return toggle('_FeatureConsumer.feature_qualifier_description');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#feature_qualifier_description">feature_qualifier_description</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.feature_qualifier_description-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.feature_qualifier_description-expanded"><a name="L1151"></a><tt class="py-lineno">1151</tt>  <tt class="py-line">        <tt class="py-docstring">"""Use feature_qualifier instead (OBSOLETE)."""</tt> </tt>
<a name="L1152"></a><tt class="py-lineno">1152</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">NotImplementedError</tt><tt class="py-op">(</tt><tt class="py-string">"Use the feature_qualifier method instead."</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1153"></a><tt class="py-lineno">1153</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.contig_location"></a><div id="_FeatureConsumer.contig_location-def"><a name="L1154"></a><tt class="py-lineno">1154</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.contig_location-toggle" onclick="return toggle('_FeatureConsumer.contig_location');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#contig_location">contig_location</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.contig_location-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.contig_location-expanded"><a name="L1155"></a><tt class="py-lineno">1155</tt>  <tt class="py-line">        <tt class="py-docstring">"""Deal with CONTIG information."""</tt> </tt>
<a name="L1156"></a><tt class="py-lineno">1156</tt>  <tt class="py-line">        <tt class="py-comment"># Historically this was stored as a SeqFeature object, but it was</tt> </tt>
<a name="L1157"></a><tt class="py-lineno">1157</tt>  <tt class="py-line">        <tt class="py-comment"># stored under record.annotations["contig"] and not under</tt> </tt>
<a name="L1158"></a><tt class="py-lineno">1158</tt>  <tt class="py-line">        <tt class="py-comment"># record.features with the other SeqFeature objects.</tt> </tt>
<a name="L1159"></a><tt class="py-lineno">1159</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L1160"></a><tt class="py-lineno">1160</tt>  <tt class="py-line">        <tt class="py-comment"># The CONTIG location line can include additional tokens like</tt> </tt>
<a name="L1161"></a><tt class="py-lineno">1161</tt>  <tt class="py-line">        <tt class="py-comment"># Gap(), Gap(100) or Gap(unk100) which are not used in the feature</tt> </tt>
<a name="L1162"></a><tt class="py-lineno">1162</tt>  <tt class="py-line">        <tt class="py-comment"># location lines, so storing it using SeqFeature based location</tt> </tt>
<a name="L1163"></a><tt class="py-lineno">1163</tt>  <tt class="py-line">        <tt class="py-comment"># objects is difficult.</tt> </tt>
<a name="L1164"></a><tt class="py-lineno">1164</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L1165"></a><tt class="py-lineno">1165</tt>  <tt class="py-line">        <tt class="py-comment"># We now store this a string, which means for BioSQL we are now in</tt> </tt>
<a name="L1166"></a><tt class="py-lineno">1166</tt>  <tt class="py-line">        <tt class="py-comment"># much better agreement with how BioPerl records the CONTIG line</tt> </tt>
<a name="L1167"></a><tt class="py-lineno">1167</tt>  <tt class="py-line">        <tt class="py-comment"># in the database.</tt> </tt>
<a name="L1168"></a><tt class="py-lineno">1168</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L1169"></a><tt class="py-lineno">1169</tt>  <tt class="py-line">        <tt class="py-comment"># NOTE - This code assumes the scanner will return all the CONTIG</tt> </tt>
<a name="L1170"></a><tt class="py-lineno">1170</tt>  <tt class="py-line">        <tt class="py-comment"># lines already combined into one long string!</tt> </tt>
<a name="L1171"></a><tt class="py-lineno">1171</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-553" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-553', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-554" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-554', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"contig"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1172"></a><tt class="py-lineno">1172</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.origin_name"></a><div id="_FeatureConsumer.origin_name-def"><a name="L1173"></a><tt class="py-lineno">1173</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.origin_name-toggle" onclick="return toggle('_FeatureConsumer.origin_name');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#origin_name">origin_name</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.origin_name-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.origin_name-expanded"><a name="L1174"></a><tt class="py-lineno">1174</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L1175"></a><tt class="py-lineno">1175</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.base_count"></a><div id="_FeatureConsumer.base_count-def"><a name="L1176"></a><tt class="py-lineno">1176</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.base_count-toggle" onclick="return toggle('_FeatureConsumer.base_count');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#base_count">base_count</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.base_count-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.base_count-expanded"><a name="L1177"></a><tt class="py-lineno">1177</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L1178"></a><tt class="py-lineno">1178</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.base_number"></a><div id="_FeatureConsumer.base_number-def"><a name="L1179"></a><tt class="py-lineno">1179</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.base_number-toggle" onclick="return toggle('_FeatureConsumer.base_number');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#base_number">base_number</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.base_number-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.base_number-expanded"><a name="L1180"></a><tt class="py-lineno">1180</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L1181"></a><tt class="py-lineno">1181</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.sequence"></a><div id="_FeatureConsumer.sequence-def"><a name="L1182"></a><tt class="py-lineno">1182</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.sequence-toggle" onclick="return toggle('_FeatureConsumer.sequence');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#sequence">sequence</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.sequence-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.sequence-expanded"><a name="L1183"></a><tt class="py-lineno">1183</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add up sequence information as we get it.</tt> </tt>
<a name="L1184"></a><tt class="py-lineno">1184</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1185"></a><tt class="py-lineno">1185</tt>  <tt class="py-line"><tt class="py-docstring">        To try and make things speedier, this puts all of the strings</tt> </tt>
<a name="L1186"></a><tt class="py-lineno">1186</tt>  <tt class="py-line"><tt class="py-docstring">        into a list of strings, and then uses string.join later to put</tt> </tt>
<a name="L1187"></a><tt class="py-lineno">1187</tt>  <tt class="py-line"><tt class="py-docstring">        them together. Supposedly, this is a big time savings</tt> </tt>
<a name="L1188"></a><tt class="py-lineno">1188</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1189"></a><tt class="py-lineno">1189</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-string">' '</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">content</tt> </tt>
<a name="L1190"></a><tt class="py-lineno">1190</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_data</tt><tt class="py-op">.</tt><tt id="link-555" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-555', 'append', 'link-213');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">.</tt><tt id="link-556" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-556', 'upper', 'link-433');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1191"></a><tt class="py-lineno">1191</tt>  <tt class="py-line"> </tt>
<a name="_FeatureConsumer.record_end"></a><div id="_FeatureConsumer.record_end-def"><a name="L1192"></a><tt class="py-lineno">1192</tt> <a class="py-toggle" href="#" id="_FeatureConsumer.record_end-toggle" onclick="return toggle('_FeatureConsumer.record_end');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._FeatureConsumer-class.html#record_end">record_end</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_FeatureConsumer.record_end-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_FeatureConsumer.record_end-expanded"><a name="L1193"></a><tt class="py-lineno">1193</tt>  <tt class="py-line">        <tt class="py-docstring">"""Clean up when we've finished the record.</tt> </tt>
<a name="L1194"></a><tt class="py-lineno">1194</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1195"></a><tt class="py-lineno">1195</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt id="link-557" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-557', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-558" class="py-name" targets="Package Bio.Alphabet=Bio.Alphabet-module.html,Class Bio.Alphabet.Alphabet=Bio.Alphabet.Alphabet-class.html"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-558', 'Alphabet', 'link-558');">Alphabet</a></tt> </tt>
<a name="L1196"></a><tt class="py-lineno">1196</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt id="link-559" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-559', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-560" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-560', 'Alphabet', 'link-558');">Alphabet</a></tt> <tt class="py-keyword">import</tt> <tt id="link-561" class="py-name" targets="Module Bio.Alphabet.IUPAC=Bio.Alphabet.IUPAC-module.html"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-561', 'IUPAC', 'link-561');">IUPAC</a></tt> </tt>
<a name="L1197"></a><tt class="py-lineno">1197</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt id="link-562" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-562', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-563" class="py-name" targets="Module Bio.Seq=Bio.Seq-module.html,Class Bio.Seq.Seq=Bio.Seq.Seq-class.html"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-563', 'Seq', 'link-563');">Seq</a></tt> <tt class="py-keyword">import</tt> <tt id="link-564" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-564', 'Seq', 'link-563');">Seq</a></tt><tt class="py-op">,</tt> <tt id="link-565" class="py-name" targets="Class Bio.Seq.UnknownSeq=Bio.Seq.UnknownSeq-class.html"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-565', 'UnknownSeq', 'link-565');">UnknownSeq</a></tt> </tt>
<a name="L1198"></a><tt class="py-lineno">1198</tt>  <tt class="py-line"> </tt>
<a name="L1199"></a><tt class="py-lineno">1199</tt>  <tt class="py-line">        <tt class="py-comment"># Try and append the version number to the accession for the full id</tt> </tt>
<a name="L1200"></a><tt class="py-lineno">1200</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-566" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-566', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-567" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-567', 'id', 'link-267');">id</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1201"></a><tt class="py-lineno">1201</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-string">'accessions'</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-568" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-568', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-569" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-569', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">,</tt> \ </tt>
<a name="L1202"></a><tt class="py-lineno">1202</tt>  <tt class="py-line">                   <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-570" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-570', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-571" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-571', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'accessions'</tt><tt class="py-op">]</tt> </tt>
<a name="L1203"></a><tt class="py-lineno">1203</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-572" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-572', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-573" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-573', 'id', 'link-267');">id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-574" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-574', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-575" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-575', 'name', 'link-273');">name</a></tt>  <tt class="py-comment"># Good fall back?</tt> </tt>
<a name="L1204"></a><tt class="py-lineno">1204</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-576" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-576', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-577" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-577', 'id', 'link-267');">id</a></tt><tt class="py-op">.</tt><tt id="link-578" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-578', 'count', 'link-189');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">'.'</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1205"></a><tt class="py-lineno">1205</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1206"></a><tt class="py-lineno">1206</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-579" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-579', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-580" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-580', 'id', 'link-267');">id</a></tt> <tt class="py-op">+=</tt> <tt class="py-string">'.%i'</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-581" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-581', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-582" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-582', 'annotations', 'link-277');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">'sequence_version'</tt><tt class="py-op">]</tt> </tt>
<a name="L1207"></a><tt class="py-lineno">1207</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L1208"></a><tt class="py-lineno">1208</tt>  <tt class="py-line">                <tt class="py-keyword">pass</tt> </tt>
<a name="L1209"></a><tt class="py-lineno">1209</tt>  <tt class="py-line"> </tt>
<a name="L1210"></a><tt class="py-lineno">1210</tt>  <tt class="py-line">        <tt class="py-comment"># add the sequence information</tt> </tt>
<a name="L1211"></a><tt class="py-lineno">1211</tt>  <tt class="py-line">        <tt class="py-comment"># first, determine the alphabet</tt> </tt>
<a name="L1212"></a><tt class="py-lineno">1212</tt>  <tt class="py-line">        <tt class="py-comment"># we default to an generic alphabet if we don't have a</tt> </tt>
<a name="L1213"></a><tt class="py-lineno">1213</tt>  <tt class="py-line">        <tt class="py-comment"># seq type or have strange sequence information.</tt> </tt>
<a name="L1214"></a><tt class="py-lineno">1214</tt>  <tt class="py-line">        <tt class="py-name">seq_alphabet</tt> <tt class="py-op">=</tt> <tt id="link-583" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-583', 'Alphabet', 'link-558');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-584" class="py-name" targets="Variable Bio.Alphabet.generic_alphabet=Bio.Alphabet-module.html#generic_alphabet"><a title="Bio.Alphabet.generic_alphabet" class="py-name" href="#" onclick="return doclink('link-584', 'generic_alphabet', 'link-584');">generic_alphabet</a></tt> </tt>
<a name="L1215"></a><tt class="py-lineno">1215</tt>  <tt class="py-line"> </tt>
<a name="L1216"></a><tt class="py-lineno">1216</tt>  <tt class="py-line">        <tt class="py-comment"># now set the sequence</tt> </tt>
<a name="L1217"></a><tt class="py-lineno">1217</tt>  <tt class="py-line">        <tt id="link-585" class="py-name" targets="Method Bio.FSSP.FSSPAlignDict.sequence()=Bio.FSSP.FSSPAlignDict-class.html#sequence,Method Bio.GenBank._FeatureConsumer.sequence()=Bio.GenBank._FeatureConsumer-class.html#sequence,Method Bio.GenBank._RecordConsumer.sequence()=Bio.GenBank._RecordConsumer-class.html#sequence,Method Bio.Phylo.PhyloXMLIO.Writer.sequence()=Bio.Phylo.PhyloXMLIO.Writer-class.html#sequence"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-585', 'sequence', 'link-585');">sequence</a></tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_data</tt><tt class="py-op">)</tt> </tt>
<a name="L1218"></a><tt class="py-lineno">1218</tt>  <tt class="py-line"> </tt>
<a name="L1219"></a><tt class="py-lineno">1219</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_expected_size</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> \ </tt>
<a name="L1220"></a><tt class="py-lineno">1220</tt>  <tt class="py-line">        <tt class="py-keyword">and</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-586" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-586', 'sequence', 'link-585');">sequence</a></tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-number">0</tt> \ </tt>
<a name="L1221"></a><tt class="py-lineno">1221</tt>  <tt class="py-line">        <tt class="py-keyword">and</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_expected_size</tt> <tt class="py-op">!=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-587" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-587', 'sequence', 'link-585');">sequence</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1222"></a><tt class="py-lineno">1222</tt>  <tt class="py-line">            <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L1223"></a><tt class="py-lineno">1223</tt>  <tt class="py-line">            <tt class="py-keyword">from</tt> <tt id="link-588" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-588', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-589" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-589', 'BiopythonParserWarning', 'link-401');">BiopythonParserWarning</a></tt> </tt>
<a name="L1224"></a><tt class="py-lineno">1224</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Expected sequence length %i, found %i (%s)."</tt> </tt>
<a name="L1225"></a><tt class="py-lineno">1225</tt>  <tt class="py-line">                          <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_expected_size</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-590" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-590', 'sequence', 'link-585');">sequence</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-591" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-591', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-592" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-592', 'id', 'link-267');">id</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1226"></a><tt class="py-lineno">1226</tt>  <tt class="py-line">                          <tt id="link-593" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-593', 'BiopythonParserWarning', 'link-401');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1227"></a><tt class="py-lineno">1227</tt>  <tt class="py-line"> </tt>
<a name="L1228"></a><tt class="py-lineno">1228</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_type</tt><tt class="py-op">:</tt> </tt>
<a name="L1229"></a><tt class="py-lineno">1229</tt>  <tt class="py-line">            <tt class="py-comment"># mRNA is really also DNA, since it is actually cDNA</tt> </tt>
<a name="L1230"></a><tt class="py-lineno">1230</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-string">'DNA'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_type</tt><tt class="py-op">.</tt><tt id="link-594" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-594', 'upper', 'link-433');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-string">'MRNA'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_type</tt><tt class="py-op">.</tt><tt id="link-595" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-595', 'upper', 'link-433');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1231"></a><tt class="py-lineno">1231</tt>  <tt class="py-line">                <tt class="py-name">seq_alphabet</tt> <tt class="py-op">=</tt> <tt id="link-596" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-596', 'IUPAC', 'link-561');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-597" class="py-name" targets="Variable Bio.Alphabet.IUPAC.ambiguous_dna=Bio.Alphabet.IUPAC-module.html#ambiguous_dna"><a title="Bio.Alphabet.IUPAC.ambiguous_dna" class="py-name" href="#" onclick="return doclink('link-597', 'ambiguous_dna', 'link-597');">ambiguous_dna</a></tt> </tt>
<a name="L1232"></a><tt class="py-lineno">1232</tt>  <tt class="py-line">            <tt class="py-comment"># are there ever really RNA sequences in GenBank?</tt> </tt>
<a name="L1233"></a><tt class="py-lineno">1233</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-string">'RNA'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_type</tt><tt class="py-op">.</tt><tt id="link-598" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-598', 'upper', 'link-433');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1234"></a><tt class="py-lineno">1234</tt>  <tt class="py-line">                <tt class="py-comment"># Even for data which was from RNA, the sequence string</tt> </tt>
<a name="L1235"></a><tt class="py-lineno">1235</tt>  <tt class="py-line">                <tt class="py-comment"># is usually given as DNA (T not U).  Bug 2408</tt> </tt>
<a name="L1236"></a><tt class="py-lineno">1236</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-string">"T"</tt> <tt class="py-keyword">in</tt> <tt id="link-599" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-599', 'sequence', 'link-585');">sequence</a></tt> <tt class="py-keyword">and</tt> <tt class="py-string">"U"</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt id="link-600" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-600', 'sequence', 'link-585');">sequence</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1237"></a><tt class="py-lineno">1237</tt>  <tt class="py-line">                    <tt class="py-name">seq_alphabet</tt> <tt class="py-op">=</tt> <tt id="link-601" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-601', 'IUPAC', 'link-561');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-602" class="py-name"><a title="Bio.Alphabet.IUPAC.ambiguous_dna" class="py-name" href="#" onclick="return doclink('link-602', 'ambiguous_dna', 'link-597');">ambiguous_dna</a></tt> </tt>
<a name="L1238"></a><tt class="py-lineno">1238</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1239"></a><tt class="py-lineno">1239</tt>  <tt class="py-line">                    <tt class="py-name">seq_alphabet</tt> <tt class="py-op">=</tt> <tt id="link-603" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-603', 'IUPAC', 'link-561');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-604" class="py-name" targets="Variable Bio.Alphabet.IUPAC.ambiguous_rna=Bio.Alphabet.IUPAC-module.html#ambiguous_rna"><a title="Bio.Alphabet.IUPAC.ambiguous_rna" class="py-name" href="#" onclick="return doclink('link-604', 'ambiguous_rna', 'link-604');">ambiguous_rna</a></tt> </tt>
<a name="L1240"></a><tt class="py-lineno">1240</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-string">'PROTEIN'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_type</tt><tt class="py-op">.</tt><tt id="link-605" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-605', 'upper', 'link-433');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> \ </tt>
<a name="L1241"></a><tt class="py-lineno">1241</tt>  <tt class="py-line">            <tt class="py-keyword">or</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_type</tt> <tt class="py-op">==</tt> <tt class="py-string">"PRT"</tt><tt class="py-op">:</tt>  <tt class="py-comment"># PRT is used in EMBL-bank for patents</tt> </tt>
<a name="L1242"></a><tt class="py-lineno">1242</tt>  <tt class="py-line">                <tt class="py-name">seq_alphabet</tt> <tt class="py-op">=</tt> <tt id="link-606" class="py-name"><a title="Bio.Alphabet.IUPAC" class="py-name" href="#" onclick="return doclink('link-606', 'IUPAC', 'link-561');">IUPAC</a></tt><tt class="py-op">.</tt><tt id="link-607" class="py-name" targets="Variable Bio.Alphabet.IUPAC.protein=Bio.Alphabet.IUPAC-module.html#protein"><a title="Bio.Alphabet.IUPAC.protein" class="py-name" href="#" onclick="return doclink('link-607', 'protein', 'link-607');">protein</a></tt>  <tt class="py-comment"># or extended protein?</tt> </tt>
<a name="L1243"></a><tt class="py-lineno">1243</tt>  <tt class="py-line">            <tt class="py-comment"># work around ugly GenBank records which have circular or</tt> </tt>
<a name="L1244"></a><tt class="py-lineno">1244</tt>  <tt class="py-line">            <tt class="py-comment"># linear but no indication of sequence type</tt> </tt>
<a name="L1245"></a><tt class="py-lineno">1245</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_type</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">"circular"</tt><tt class="py-op">,</tt> <tt class="py-string">"linear"</tt><tt class="py-op">,</tt> <tt class="py-string">"unspecified"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L1246"></a><tt class="py-lineno">1246</tt>  <tt class="py-line">                <tt class="py-keyword">pass</tt> </tt>
<a name="L1247"></a><tt class="py-lineno">1247</tt>  <tt class="py-line">            <tt class="py-comment"># we have a bug if we get here</tt> </tt>
<a name="L1248"></a><tt class="py-lineno">1248</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1249"></a><tt class="py-lineno">1249</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Could not determine alphabet for seq_type %s"</tt> </tt>
<a name="L1250"></a><tt class="py-lineno">1250</tt>  <tt class="py-line">                                 <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_type</tt><tt class="py-op">)</tt> </tt>
<a name="L1251"></a><tt class="py-lineno">1251</tt>  <tt class="py-line"> </tt>
<a name="L1252"></a><tt class="py-lineno">1252</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-608" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-608', 'sequence', 'link-585');">sequence</a></tt> <tt class="py-keyword">and</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__expected_size</tt><tt class="py-op">:</tt> </tt>
<a name="L1253"></a><tt class="py-lineno">1253</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-609" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-609', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-610" class="py-name" targets="Variable Bio.SeqRecord.SeqRecord.seq=Bio.SeqRecord.SeqRecord-class.html#seq,Variable BioSQL.BioSeq.DBSeqRecord.seq=BioSQL.BioSeq.DBSeqRecord-class.html#seq"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-610', 'seq', 'link-610');">seq</a></tt> <tt class="py-op">=</tt> <tt id="link-611" class="py-name"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-611', 'UnknownSeq', 'link-565');">UnknownSeq</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_expected_size</tt><tt class="py-op">,</tt> <tt class="py-name">seq_alphabet</tt><tt class="py-op">)</tt> </tt>
<a name="L1254"></a><tt class="py-lineno">1254</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1255"></a><tt class="py-lineno">1255</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-612" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-612', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-613" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-613', 'seq', 'link-610');">seq</a></tt> <tt class="py-op">=</tt> <tt id="link-614" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-614', 'Seq', 'link-563');">Seq</a></tt><tt class="py-op">(</tt><tt id="link-615" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-615', 'sequence', 'link-585');">sequence</a></tt><tt class="py-op">,</tt> <tt class="py-name">seq_alphabet</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1256"></a><tt class="py-lineno">1256</tt>  <tt class="py-line"> </tt>
<a name="L1257"></a><tt class="py-lineno">1257</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer"></a><div id="_RecordConsumer-def"><a name="L1258"></a><tt class="py-lineno">1258</tt> <a class="py-toggle" href="#" id="_RecordConsumer-toggle" onclick="return toggle('_RecordConsumer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html">_RecordConsumer</a><tt class="py-op">(</tt><tt class="py-base-class">_BaseGenBankConsumer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_RecordConsumer-expanded"><a name="L1259"></a><tt class="py-lineno">1259</tt>  <tt class="py-line">    <tt class="py-docstring">"""Create a GenBank Record object from scanner generated information (PRIVATE).</tt> </tt>
<a name="L1260"></a><tt class="py-lineno">1260</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="_RecordConsumer.__init__"></a><div id="_RecordConsumer.__init__-def"><a name="L1261"></a><tt class="py-lineno">1261</tt> <a class="py-toggle" href="#" id="_RecordConsumer.__init__-toggle" onclick="return toggle('_RecordConsumer.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.__init__-expanded"><a name="L1262"></a><tt class="py-lineno">1262</tt>  <tt class="py-line">        <tt id="link-616" class="py-name"><a title="Bio.GenBank._BaseGenBankConsumer" class="py-name" href="#" onclick="return doclink('link-616', '_BaseGenBankConsumer', 'link-263');">_BaseGenBankConsumer</a></tt><tt class="py-op">.</tt><tt id="link-617" class="py-name"><a title="Bio.Affy.CelFile.Record.__init__
Bio.Align.AlignInfo.PSSM.__init__
Bio.Align.AlignInfo.SummaryInfo.__init__
Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline.__init__
Bio.Align.Applications._Clustalw.ClustalwCommandline.__init__
Bio.Align.Applications._Dialign.DialignCommandline.__init__
Bio.Align.Applications._MSAProbs.MSAProbsCommandline.__init__
Bio.Align.Applications._Mafft.MafftCommandline.__init__
Bio.Align.Applications._Muscle.MuscleCommandline.__init__
Bio.Align.Applications._Prank.PrankCommandline.__init__
Bio.Align.Applications._Probcons.ProbconsCommandline.__init__
Bio.Align.Applications._TCoffee.TCoffeeCommandline.__init__
Bio.Align.Generic.Alignment.__init__
Bio.Align.MultipleSeqAlignment.__init__
Bio.AlignIO.Interfaces.AlignmentIterator.__init__
Bio.AlignIO.Interfaces.AlignmentWriter.__init__
Bio.AlignIO.Interfaces.SequentialAlignmentWriter.__init__
Bio.Alphabet.AlphabetEncoder.__init__
Bio.Alphabet.Gapped.__init__
Bio.Alphabet.HasStopCodon.__init__
Bio.Application.AbstractCommandline.__init__
Bio.Application.ApplicationError.__init__
Bio.Application._AbstractParameter.__init__
Bio.Application._Argument.__init__
Bio.Application._Option.__init__
Bio.Application._StaticArgument.__init__
Bio.Application._Switch.__init__
Bio.Blast.Applications.NcbiblastformatterCommandline.__init__
Bio.Blast.Applications.NcbiblastnCommandline.__init__
Bio.Blast.Applications.NcbiblastpCommandline.__init__
Bio.Blast.Applications.NcbiblastxCommandline.__init__
Bio.Blast.Applications.NcbideltablastCommandline.__init__
Bio.Blast.Applications.NcbipsiblastCommandline.__init__
Bio.Blast.Applications.NcbirpsblastCommandline.__init__
Bio.Blast.Applications.NcbirpstblastnCommandline.__init__
Bio.Blast.Applications.NcbitblastnCommandline.__init__
Bio.Blast.Applications.NcbitblastxCommandline.__init__
Bio.Blast.Applications._NcbibaseblastCommandline.__init__
Bio.Blast.Applications._Ncbiblast2SeqCommandline.__init__
Bio.Blast.Applications._NcbiblastCommandline.__init__
Bio.Blast.Applications._NcbiblastMain2SeqCommandline.__init__
Bio.Blast.NCBIStandalone.BlastErrorParser.__init__
Bio.Blast.NCBIStandalone.BlastParser.__init__
Bio.Blast.NCBIStandalone.Iterator.__init__
Bio.Blast.NCBIStandalone.PSIBlastParser.__init__
Bio.Blast.NCBIStandalone._BlastConsumer.__init__
Bio.Blast.NCBIStandalone._BlastErrorConsumer.__init__
Bio.Blast.NCBIStandalone._PSIBlastConsumer.__init__
Bio.Blast.NCBIXML.BlastParser.__init__
Bio.Blast.NCBIXML._XMLparser.__init__
Bio.Blast.ParseBlastTable.BlastTableEntry.__init__
Bio.Blast.ParseBlastTable.BlastTableReader.__init__
Bio.Blast.ParseBlastTable.BlastTableRec.__init__
Bio.Blast.Record.Alignment.__init__
Bio.Blast.Record.Blast.__init__
Bio.Blast.Record.DatabaseReport.__init__
Bio.Blast.Record.Description.__init__
Bio.Blast.Record.HSP.__init__
Bio.Blast.Record.Header.__init__
Bio.Blast.Record.MultipleAlignment.__init__
Bio.Blast.Record.PSIBlast.__init__
Bio.Blast.Record.Parameters.__init__
Bio.Blast.Record.Round.__init__
Bio.CAPS.CAPSMap.__init__
Bio.CAPS.DifferentialCutsite.__init__
Bio.Cluster.Record.__init__
Bio.Compass.Record.__init__
Bio.Crystal.Chain.__init__
Bio.Crystal.Crystal.__init__
Bio.Crystal.Hetero.__init__
Bio.Data.CodonTable.AmbiguousCodonTable.__init__
Bio.Data.CodonTable.AmbiguousForwardTable.__init__
Bio.Data.CodonTable.CodonTable.__init__
Bio.Data.CodonTable.NCBICodonTable.__init__
Bio.DocSQL.Create.__init__
Bio.DocSQL.Insert.__init__
Bio.DocSQL.IterationCursor.__init__
Bio.DocSQL.Query.__init__
Bio.DocSQL.QueryAll.__init__
Bio.DocSQL.QueryGeneric.__init__
Bio.DocSQL.QueryRow.__init__
Bio.DocSQL.QuerySingle.__init__
Bio.Emboss.Applications.DiffseqCommandline.__init__
Bio.Emboss.Applications.EInvertedCommandline.__init__
Bio.Emboss.Applications.ETandemCommandline.__init__
Bio.Emboss.Applications.Est2GenomeCommandline.__init__
Bio.Emboss.Applications.FConsenseCommandline.__init__
Bio.Emboss.Applications.FDNADistCommandline.__init__
Bio.Emboss.Applications.FDNAParsCommandline.__init__
Bio.Emboss.Applications.FNeighborCommandline.__init__
Bio.Emboss.Applications.FProtDistCommandline.__init__
Bio.Emboss.Applications.FProtParsCommandline.__init__
Bio.Emboss.Applications.FSeqBootCommandline.__init__
Bio.Emboss.Applications.FTreeDistCommandline.__init__
Bio.Emboss.Applications.FuzznucCommandline.__init__
Bio.Emboss.Applications.IepCommandline.__init__
Bio.Emboss.Applications.NeedleCommandline.__init__
Bio.Emboss.Applications.NeedleallCommandline.__init__
Bio.Emboss.Applications.PalindromeCommandline.__init__
Bio.Emboss.Applications.Primer3Commandline.__init__
Bio.Emboss.Applications.PrimerSearchCommandline.__init__
Bio.Emboss.Applications.SeqmatchallCommandline.__init__
Bio.Emboss.Applications.SeqretCommandline.__init__
Bio.Emboss.Applications.StretcherCommandline.__init__
Bio.Emboss.Applications.TranalignCommandline.__init__
Bio.Emboss.Applications.WaterCommandline.__init__
Bio.Emboss.Applications._EmbossCommandLine.__init__
Bio.Emboss.Applications._EmbossMinimalCommandLine.__init__
Bio.Emboss.Primer3.Primers.__init__
Bio.Emboss.Primer3.Record.__init__
Bio.Emboss.PrimerSearch.Amplifier.__init__
Bio.Emboss.PrimerSearch.InputRecord.__init__
Bio.Emboss.PrimerSearch.OutputRecord.__init__
Bio.Entrez.Parser.CorruptedXMLError.__init__
Bio.Entrez.Parser.DataHandler.__init__
Bio.Entrez.Parser.NotXMLError.__init__
Bio.Entrez.Parser.StructureElement.__init__
Bio.Entrez.Parser.ValidationError.__init__
Bio.ExPASy.Enzyme.Record.__init__
Bio.ExPASy.Prodoc.Record.__init__
Bio.ExPASy.Prodoc.Reference.__init__
Bio.ExPASy.Prosite.Record.__init__
Bio.ExPASy.ScanProsite.ContentHandler.__init__
Bio.ExPASy.ScanProsite.Parser.__init__
Bio.ExPASy.ScanProsite.Record.__init__
Bio.FSSP.FSSPAlignDict.__init__
Bio.FSSP.FSSPAlignRec.__init__
Bio.FSSP.FSSPHeader.__init__
Bio.FSSP.FSSPSumRec.__init__
Bio.FSSP.FSSPTools.FSSPMultAlign.__init__
Bio.FSSP.PosAlign.__init__
Bio.FSSP.fssp_rec.fff_rec.__init__
Bio.File.UndoHandle.__init__
Bio.File._IndexedSeqFileDict.__init__
Bio.File._SQLiteManySeqFilesDict.__init__
Bio.GA.Crossover.General.SafeFitnessCrossover.__init__
Bio.GA.Crossover.GeneralPoint.GeneralPointCrossover.__init__
Bio.GA.Crossover.GeneralPoint.InterleaveCrossover.__init__
Bio.GA.Crossover.Point.SinglePointCrossover.__init__
Bio.GA.Crossover.TwoPoint.TwoPointCrossover.__init__
Bio.GA.Crossover.Uniform.UniformCrossover.__init__
Bio.GA.Evolver.GenerationEvolver.__init__
Bio.GA.Evolver.SteadyStateEvolver.__init__
Bio.GA.Mutation.General.SafeFitnessMutation.__init__
Bio.GA.Mutation.Simple.ConversionMutation.__init__
Bio.GA.Mutation.Simple.SinglePositionMutation.__init__
Bio.GA.Organism.Organism.__init__
Bio.GA.Repair.Stabilizing.AmbiguousRepair.__init__
Bio.GA.Selection.Abstract.AbstractSelection.__init__
Bio.GA.Selection.Diversity.DiversitySelection.__init__
Bio.GA.Selection.RouletteWheel.RouletteWheelSelection.__init__
Bio.GA.Selection.Tournament.TournamentSelection.__init__
Bio.GenBank.FeatureParser.__init__
Bio.GenBank.Iterator.__init__
Bio.GenBank.Record.Feature.__init__
Bio.GenBank.Record.Qualifier.__init__
Bio.GenBank.Record.Record.__init__
Bio.GenBank.Record.Reference.__init__
Bio.GenBank.RecordParser.__init__
Bio.GenBank.Scanner.InsdcScanner.__init__
Bio.GenBank._BaseGenBankConsumer.__init__
Bio.GenBank._FeatureConsumer.__init__
Bio.GenBank._RecordConsumer.__init__
Bio.GenBank.utils.FeatureValueCleaner.__init__
Bio.Geo.Record.Record.__init__
Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment.__init__
Bio.Graphics.BasicChromosome.Chromosome.__init__
Bio.Graphics.BasicChromosome.ChromosomeSegment.__init__
Bio.Graphics.BasicChromosome.Organism.__init__
Bio.Graphics.BasicChromosome.TelomereSegment.__init__
Bio.Graphics.BasicChromosome._ChromosomeComponent.__init__
Bio.Graphics.ColorSpiral.ColorSpiral.__init__
Bio.Graphics.Comparative.ComparativeScatterPlot.__init__
Bio.Graphics.DisplayRepresentation.ChromosomeCounts.__init__
Bio.Graphics.Distribution.BarChartDistribution.__init__
Bio.Graphics.Distribution.DistributionPage.__init__
Bio.Graphics.Distribution.LineDistribution.__init__
Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.__init__
Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.__init__
Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.__init__
Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.__init__
Bio.Graphics.GenomeDiagram._Diagram.Diagram.__init__
Bio.Graphics.GenomeDiagram._Feature.Feature.__init__
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.__init__
Bio.Graphics.GenomeDiagram._Graph.GraphData.__init__
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.__init__
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.__init__
Bio.Graphics.GenomeDiagram._Track.Track.__init__
Bio.Graphics.KGML_vis.KGMLCanvas.__init__
Bio.HMM.DynamicProgramming.AbstractDPAlgorithms.__init__
Bio.HMM.DynamicProgramming.LogDPAlgorithms.__init__
Bio.HMM.DynamicProgramming.ScaledDPAlgorithms.__init__
Bio.HMM.MarkovModel.HiddenMarkovModel.__init__
Bio.HMM.MarkovModel.MarkovModelBuilder.__init__
Bio.HMM.Trainer.AbstractTrainer.__init__
Bio.HMM.Trainer.BaumWelchTrainer.__init__
Bio.HMM.Trainer.KnownStateTrainer.__init__
Bio.HMM.Trainer.TrainingSequence.__init__
Bio.Index._InMemoryIndex.__init__
Bio.Index._ShelveIndex.__init__
Bio.KDTree.KDTree'.KDTree.__init__
Bio.KEGG.Compound.Record.__init__
Bio.KEGG.Enzyme.Record.__init__
Bio.KEGG.KGML.KGML_parser.KGMLParser.__init__
Bio.KEGG.KGML.KGML_pathway.Component.__init__
Bio.KEGG.KGML.KGML_pathway.Entry.__init__
Bio.KEGG.KGML.KGML_pathway.Graphics.__init__
Bio.KEGG.KGML.KGML_pathway.Pathway.__init__
Bio.KEGG.KGML.KGML_pathway.Reaction.__init__
Bio.KEGG.KGML.KGML_pathway.Relation.__init__
Bio.LogisticRegression.LogisticRegression.__init__
Bio.MarkovModel.MarkovModel.__init__
Bio.MaxEntropy.MaxEntropy.__init__
Bio.Motif.Parsers.AlignAce.Record.__init__
Bio.Motif.Parsers.MAST.Record.__init__
Bio.Motif.Parsers.MEME.MEMEInstance.__init__
Bio.Motif.Parsers.MEME.MEMEMotif.__init__
Bio.Motif.Parsers.MEME.MEMERecord.__init__
Bio.Motif.Thresholds.ScoreDistribution.__init__
Bio.Motif._Motif.Motif.__init__
Bio.NMR.xpktools.Peaklist.__init__
Bio.NMR.xpktools.XpkEntry.__init__
Bio.NaiveBayes.NaiveBayes.__init__
Bio.NeuralNetwork.BackPropagation.Layer.AbstractLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.__init__
Bio.NeuralNetwork.BackPropagation.Network.BasicNetwork.__init__
Bio.NeuralNetwork.Gene.Motif.MotifCoder.__init__
Bio.NeuralNetwork.Gene.Motif.MotifFinder.__init__
Bio.NeuralNetwork.Gene.Pattern.PatternIO.__init__
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.__init__
Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness.__init__
Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder.__init__
Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness.__init__
Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator.__init__
Bio.NeuralNetwork.Gene.Schema.Schema.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaCoder.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaFactory.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaFinder.__init__
Bio.NeuralNetwork.Gene.Schema.SimpleFinisher.__init__
Bio.NeuralNetwork.Gene.Signature.SignatureCoder.__init__
Bio.NeuralNetwork.Gene.Signature.SignatureFinder.__init__
Bio.NeuralNetwork.StopTraining.ValidationIncreaseStop.__init__
Bio.NeuralNetwork.Training.ExampleManager.__init__
Bio.NeuralNetwork.Training.TrainingExample.__init__
Bio.Nexus.Nexus.Block.__init__
Bio.Nexus.Nexus.CharBuffer.__init__
Bio.Nexus.Nexus.Commandline.__init__
Bio.Nexus.Nexus.Nexus.__init__
Bio.Nexus.Nexus.StepMatrix.__init__
Bio.Nexus.Nodes.Chain.__init__
Bio.Nexus.Nodes.Node.__init__
Bio.Nexus.Trees.NodeData.__init__
Bio.Nexus.Trees.Tree.__init__
Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap.__init__
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__init__
Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap.__init__
Bio.PDB.Atom.Atom.__init__
Bio.PDB.Atom.DisorderedAtom.__init__
Bio.PDB.Chain.Chain.__init__
Bio.PDB.DSSP'.DSSP.__init__
Bio.PDB.Dice.ChainSelector.__init__
Bio.PDB.Entity.DisorderedEntityWrapper.__init__
Bio.PDB.Entity.Entity.__init__
Bio.PDB.FragmentMapper'.Fragment.__init__
Bio.PDB.FragmentMapper'.FragmentMapper.__init__
Bio.PDB.HSExposure.ExposureCN.__init__
Bio.PDB.HSExposure.HSExposureCA.__init__
Bio.PDB.HSExposure.HSExposureCB.__init__
Bio.PDB.HSExposure._AbstractHSExposure.__init__
Bio.PDB.MMCIF2Dict.MMCIF2Dict.__init__
Bio.PDB.MMCIFParser'.MMCIFParser.__init__
Bio.PDB.Model.Model.__init__
Bio.PDB.NACCESS.NACCESS.__init__
Bio.PDB.NACCESS.NACCESS_atomic.__init__
Bio.PDB.NeighborSearch'.NeighborSearch.__init__
Bio.PDB.PDBIO'.PDBIO.__init__
Bio.PDB.PDBList'.PDBList.__init__
Bio.PDB.PDBParser'.PDBParser.__init__
Bio.PDB.PSEA.PSEA.__init__
Bio.PDB.Polypeptide.CaPPBuilder.__init__
Bio.PDB.Polypeptide.PPBuilder.__init__
Bio.PDB.Polypeptide._PPBuilder.__init__
Bio.PDB.QCPSuperimposer.QCPSuperimposer.__init__
Bio.PDB.Residue.DisorderedResidue.__init__
Bio.PDB.Residue.Residue.__init__
Bio.PDB.ResidueDepth'.ResidueDepth.__init__
Bio.PDB.Structure.Structure.__init__
Bio.PDB.StructureAlignment'.StructureAlignment.__init__
Bio.PDB.StructureBuilder.StructureBuilder.__init__
Bio.PDB.Superimposer'.Superimposer.__init__
Bio.PDB.Vector'.Vector.__init__
Bio.ParserSupport.EventGenerator.__init__
Bio.ParserSupport.TaggingConsumer.__init__
Bio.Pathway.Network.__init__
Bio.Pathway.Reaction.__init__
Bio.Pathway.Rep.Graph.Graph.__init__
Bio.Pathway.Rep.MultiGraph.MultiGraph.__init__
Bio.Pathway.System.__init__
Bio.Phylo.Applications._Fasttree.FastTreeCommandline.__init__
Bio.Phylo.Applications._Phyml.PhymlCommandline.__init__
Bio.Phylo.Applications._Raxml.RaxmlCommandline.__init__
Bio.Phylo.BaseTree.BranchColor.__init__
Bio.Phylo.BaseTree.Clade.__init__
Bio.Phylo.BaseTree.Tree.__init__
Bio.Phylo.CDAO.Clade.__init__
Bio.Phylo.CDAO.Tree.__init__
Bio.Phylo.CDAOIO.Parser.__init__
Bio.Phylo.CDAOIO.Writer.__init__
Bio.Phylo.NeXML.Clade.__init__
Bio.Phylo.NeXML.Tree.__init__
Bio.Phylo.NeXMLIO.Parser.__init__
Bio.Phylo.NeXMLIO.Writer.__init__
Bio.Phylo.Newick.Clade.__init__
Bio.Phylo.Newick.Tree.__init__
Bio.Phylo.NewickIO.Parser.__init__
Bio.Phylo.NewickIO.Writer.__init__
Bio.Phylo.PAML._paml.Paml.__init__
Bio.Phylo.PAML.baseml.Baseml.__init__
Bio.Phylo.PAML.codeml.Codeml.__init__
Bio.Phylo.PAML.yn00.Yn00.__init__
Bio.Phylo.PhyloXML.Accession.__init__
Bio.Phylo.PhyloXML.Annotation.__init__
Bio.Phylo.PhyloXML.BinaryCharacters.__init__
Bio.Phylo.PhyloXML.BranchColor.__init__
Bio.Phylo.PhyloXML.Clade.__init__
Bio.Phylo.PhyloXML.CladeRelation.__init__
Bio.Phylo.PhyloXML.Confidence.__init__
Bio.Phylo.PhyloXML.Date.__init__
Bio.Phylo.PhyloXML.Distribution.__init__
Bio.Phylo.PhyloXML.DomainArchitecture.__init__
Bio.Phylo.PhyloXML.Events.__init__
Bio.Phylo.PhyloXML.Id.__init__
Bio.Phylo.PhyloXML.MolSeq.__init__
Bio.Phylo.PhyloXML.Other.__init__
Bio.Phylo.PhyloXML.Phylogeny.__init__
Bio.Phylo.PhyloXML.Phyloxml.__init__
Bio.Phylo.PhyloXML.Point.__init__
Bio.Phylo.PhyloXML.Polygon.__init__
Bio.Phylo.PhyloXML.Property.__init__
Bio.Phylo.PhyloXML.ProteinDomain.__init__
Bio.Phylo.PhyloXML.Reference.__init__
Bio.Phylo.PhyloXML.Sequence.__init__
Bio.Phylo.PhyloXML.SequenceRelation.__init__
Bio.Phylo.PhyloXML.Taxonomy.__init__
Bio.Phylo.PhyloXML.Uri.__init__
Bio.Phylo.PhyloXMLIO.Parser.__init__
Bio.Phylo.PhyloXMLIO.Writer.__init__
Bio.Phylo.TreeConstruction.DistanceCalculator.__init__
Bio.Phylo.TreeConstruction.DistanceTreeConstructor.__init__
Bio.Phylo.TreeConstruction.NNITreeSearcher.__init__
Bio.Phylo.TreeConstruction.ParsimonyScorer.__init__
Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor.__init__
Bio.Phylo.TreeConstruction._DistanceMatrix.__init__
Bio.Phylo.TreeConstruction._Matrix.__init__
Bio.PopGen.Async.Async.__init__
Bio.PopGen.Async.DirectoryRetriever.__init__
Bio.PopGen.Async.FileRetriever.__init__
Bio.PopGen.Async.Local.Local.__init__
Bio.PopGen.FDist.Async.FDistAsync.__init__
Bio.PopGen.FDist.Async.SplitFDist.__init__
Bio.PopGen.FDist.Controller.FDistController.__init__
Bio.PopGen.FDist.Record.__init__
Bio.PopGen.GenePop.Controller.GenePopController.__init__
Bio.PopGen.GenePop.Controller._FileIterator.__init__
Bio.PopGen.GenePop.Controller._GenePopCommandline.__init__
Bio.PopGen.GenePop.EasyController.EasyController.__init__
Bio.PopGen.GenePop.FileParser.FileRecord.__init__
Bio.PopGen.GenePop.LargeFileParser.Record.__init__
Bio.PopGen.GenePop.Record.__init__
Bio.PopGen.SimCoal.Async.SimCoalCache.__init__
Bio.PopGen.SimCoal.Cache.SimCoalCache.__init__
Bio.PopGen.SimCoal.Controller.FastSimCoalController.__init__
Bio.PopGen.SimCoal.Controller.SimCoalController.__init__
Bio.PopGen.SimCoal.Controller._FastSimCoalCommandLine.__init__
Bio.Restriction.PrintFormat.PrintFormat.__init__
Bio.Restriction.Restriction.Analysis.__init__
Bio.Restriction.Restriction.FormattedSeq.__init__
Bio.Restriction.Restriction.RestrictionBatch.__init__
Bio.Restriction.Restriction.RestrictionType.__init__
Bio.SCOP.Astral.__init__
Bio.SCOP.Cla.Index.__init__
Bio.SCOP.Cla.Record.__init__
Bio.SCOP.Des.Record.__init__
Bio.SCOP.Dom.Record.__init__
Bio.SCOP.Domain.__init__
Bio.SCOP.Hie.Record.__init__
Bio.SCOP.Node.__init__
Bio.SCOP.Raf.Res.__init__
Bio.SCOP.Raf.SeqMap.__init__
Bio.SCOP.Raf.SeqMapIndex.__init__
Bio.SCOP.Residues.Residues.__init__
Bio.SCOP.Scop.__init__
Bio.SVDSuperimposer.SVDSuperimposer.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabParser.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.__init__
Bio.SearchIO.BlastIO.blast_text.BlastTextParser.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.__init__
Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator.__init__
Bio.SearchIO.BlatIO.BlatPslIndexer.__init__
Bio.SearchIO.BlatIO.BlatPslParser.__init__
Bio.SearchIO.BlatIO.BlatPslWriter.__init__
Bio.SearchIO.ExonerateIO._base._BaseExonerateParser.__init__
Bio.SearchIO.FastaIO.FastaM10Indexer.__init__
Bio.SearchIO.FastaIO.FastaM10Parser.__init__
Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer.__init__
Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser.__init__
Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.__init__
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser.__init__
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.__init__
Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser.__init__
Bio.SearchIO._index.SearchIndexer.__init__
Bio.SearchIO._model.hit.Hit.__init__
Bio.SearchIO._model.hsp.HSP.__init__
Bio.SearchIO._model.hsp.HSPFragment.__init__
Bio.SearchIO._model.query.QueryResult.__init__
Bio.Seq.MutableSeq.__init__
Bio.Seq.Seq.__init__
Bio.Seq.UnknownSeq.__init__
Bio.SeqFeature.CompoundLocation.__init__
Bio.SeqFeature.FeatureLocation.__init__
Bio.SeqFeature.PositionGap.__init__
Bio.SeqFeature.Reference.__init__
Bio.SeqFeature.SeqFeature.__init__
Bio.SeqIO.FastaIO.FastaWriter.__init__
Bio.SeqIO.Interfaces.SequenceIterator.__init__
Bio.SeqIO.Interfaces.SequenceWriter.__init__
Bio.SeqIO.Interfaces.SequentialSequenceWriter.__init__
Bio.SeqIO.PhdIO.PhdWriter.__init__
Bio.SeqIO.QualityIO.QualPhredWriter.__init__
Bio.SeqIO.SeqXmlIO.SeqXmlIterator.__init__
Bio.SeqIO.SeqXmlIO.SeqXmlWriter.__init__
Bio.SeqIO.SeqXmlIO.XMLRecordIterator.__init__
Bio.SeqIO.SffIO.SffWriter.__init__
Bio.SeqIO.SffIO._AddTellHandle.__init__
Bio.SeqIO.UniprotIO.Parser.__init__
Bio.SeqIO._index.IntelliGeneticsRandomAccess.__init__
Bio.SeqIO._index.SeqFileRandomAccess.__init__
Bio.SeqIO._index.SequentialSeqFileRandomAccess.__init__
Bio.SeqIO._index.SffRandomAccess.__init__
Bio.SeqRecord.SeqRecord.__init__
Bio.SeqRecord._RestrictedDict.__init__
Bio.SeqUtils.CodonUsage.CodonAdaptationIndex.__init__
Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.__init__
Bio.SeqUtils.ProtParam.ProteinAnalysis.__init__
Bio.Sequencing.Ace.ACEFileRecord.__init__
Bio.Sequencing.Ace.Contig.__init__
Bio.Sequencing.Ace.Reads.__init__
Bio.Sequencing.Ace.af.__init__
Bio.Sequencing.Ace.bs.__init__
Bio.Sequencing.Ace.ct.__init__
Bio.Sequencing.Ace.ds.__init__
Bio.Sequencing.Ace.qa.__init__
Bio.Sequencing.Ace.rd.__init__
Bio.Sequencing.Ace.rt.__init__
Bio.Sequencing.Ace.wa.__init__
Bio.Sequencing.Ace.wr.__init__
Bio.Sequencing.Applications._Novoalign.NovoalignCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaAlignCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaBwaswCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaIndexCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaSampeCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaSamseCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsSortCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline.__init__
Bio.Sequencing.Phd.Record.__init__
Bio.SubsMat.FreqTable.FreqTable.__init__
Bio.SubsMat.SeqMat.__init__
Bio.SwissProt.KeyWList.Record.__init__
Bio.SwissProt.Record.__init__
Bio.SwissProt.Reference.__init__
Bio.UniGene.ProtsimLine.__init__
Bio.UniGene.Record.__init__
Bio.UniGene.STSLine.__init__
Bio.UniGene.SequenceLine.__init__
Bio.Wise.dnal.Statistics.__init__
Bio.Wise.psw.AlignmentColumn.__init__
Bio.Wise.psw.ColumnUnit.__init__
Bio._py3k._ordereddict.OrderedDict.__init__
Bio.bgzf.BgzfReader.__init__
Bio.bgzf.BgzfWriter.__init__
Bio.codonalign.codonalignment.CodonAlignment.__init__
Bio.codonalign.codonseq.CodonSeq.__init__
Bio.kNN.kNN.__init__
Bio.motifs.Instances.__init__
Bio.motifs.Motif.__init__
Bio.motifs.alignace.Record.__init__
Bio.motifs.applications._alignace.AlignAceCommandline.__init__
Bio.motifs.applications._alignace.CompareAceCommandline.__init__
Bio.motifs.applications._xxmotif.XXmotifCommandline.__init__
Bio.motifs.jaspar.Motif.__init__
Bio.motifs.jaspar.Record.__init__
Bio.motifs.jaspar.db.JASPAR5.__init__
Bio.motifs.mast.Record.__init__
Bio.motifs.matrix.GenericPositionMatrix.__init__
Bio.motifs.matrix.PositionWeightMatrix.__init__
Bio.motifs.meme.Instance.__init__
Bio.motifs.meme.Motif.__init__
Bio.motifs.meme.Record.__init__
Bio.motifs.thresholds.ScoreDistribution.__init__
Bio.motifs.transfac.Record.__init__
Bio.pairwise2.affine_penalty.__init__
Bio.pairwise2.dictionary_match.__init__
Bio.pairwise2.identity_match.__init__
BioSQL.BioSeq.DBSeq.__init__
BioSQL.BioSeq.DBSeqRecord.__init__
BioSQL.BioSeqDatabase.Adaptor.__init__
BioSQL.BioSeqDatabase.BioSeqDatabase.__init__
BioSQL.BioSeqDatabase.DBServer.__init__
BioSQL.BioSeqDatabase._CursorWrapper.__init__
BioSQL.DBUtils.Generic_dbutils.__init__
BioSQL.Loader.DatabaseLoader.__init__
BioSQL.Loader.DatabaseRemover.__init__" class="py-name" href="#" onclick="return doclink('link-617', '__init__', 'link-264');">__init__</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
<a name="L1263"></a><tt class="py-lineno">1263</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt class="py-op">.</tt> <tt class="py-keyword">import</tt> <tt id="link-618" class="py-name" targets="Class Bio.Affy.CelFile.Record=Bio.Affy.CelFile.Record-class.html,Module Bio.Blast.Record=Bio.Blast.Record-module.html,Class Bio.Cluster.Record=Bio.Cluster.Record-class.html,Class Bio.Compass.Record=Bio.Compass.Record-class.html,Class Bio.Emboss.Primer3.Record=Bio.Emboss.Primer3.Record-class.html,Class Bio.ExPASy.Enzyme.Record=Bio.ExPASy.Enzyme.Record-class.html,Class Bio.ExPASy.Prodoc.Record=Bio.ExPASy.Prodoc.Record-class.html,Class Bio.ExPASy.Prosite.Record=Bio.ExPASy.Prosite.Record-class.html,Class Bio.ExPASy.ScanProsite.Record=Bio.ExPASy.ScanProsite.Record-class.html,Module Bio.GenBank.Record=Bio.GenBank.Record-module.html,Class Bio.GenBank.Record.Record=Bio.GenBank.Record.Record-class.html,Module Bio.Geo.Record=Bio.Geo.Record-module.html,Class Bio.Geo.Record.Record=Bio.Geo.Record.Record-class.html,Class Bio.KEGG.Compound.Record=Bio.KEGG.Compound.Record-class.html,Class Bio.KEGG.Enzyme.Record=Bio.KEGG.Enzyme.Record-class.html,Class Bio.Medline.Record=Bio.Medline.Record-class.html,Class Bio.Motif.Parsers.AlignAce.Record=Bio.Motif.Parsers.AlignAce.Record-class.html,Class Bio.Motif.Parsers.MAST.Record=Bio.Motif.Parsers.MAST.Record-class.html,Class Bio.PopGen.FDist.Record=Bio.PopGen.FDist.Record-class.html,Class Bio.PopGen.GenePop.LargeFileParser.Record=Bio.PopGen.GenePop.LargeFileParser.Record-class.html,Class Bio.PopGen.GenePop.Record=Bio.PopGen.GenePop.Record-class.html,Class Bio.SCOP.Cla.Record=Bio.SCOP.Cla.Record-class.html,Class Bio.SCOP.Des.Record=Bio.SCOP.Des.Record-class.html,Class Bio.SCOP.Dom.Record=Bio.SCOP.Dom.Record-class.html,Class Bio.SCOP.Hie.Record=Bio.SCOP.Hie.Record-class.html,Class Bio.Sequencing.Phd.Record=Bio.Sequencing.Phd.Record-class.html,Class Bio.SwissProt.KeyWList.Record=Bio.SwissProt.KeyWList.Record-class.html,Class Bio.SwissProt.Record=Bio.SwissProt.Record-class.html,Class Bio.UniGene.Record=Bio.UniGene.Record-class.html,Class Bio.motifs.alignace.Record=Bio.motifs.alignace.Record-class.html,Class Bio.motifs.jaspar.Record=Bio.motifs.jaspar.Record-class.html,Class Bio.motifs.mast.Record=Bio.motifs.mast.Record-class.html,Class Bio.motifs.meme.Record=Bio.motifs.meme.Record-class.html,Class Bio.motifs.transfac.Record=Bio.motifs.transfac.Record-class.html"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-618', 'Record', 'link-618');">Record</a></tt> </tt>
<a name="L1264"></a><tt class="py-lineno">1264</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-619" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-619', 'data', 'link-223');">data</a></tt> <tt class="py-op">=</tt> <tt id="link-620" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-620', 'Record', 'link-618');">Record</a></tt><tt class="py-op">.</tt><tt id="link-621" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-621', 'Record', 'link-618');">Record</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1265"></a><tt class="py-lineno">1265</tt>  <tt class="py-line"> </tt>
<a name="L1266"></a><tt class="py-lineno">1266</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_data</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L1267"></a><tt class="py-lineno">1267</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L1268"></a><tt class="py-lineno">1268</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_feature</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L1269"></a><tt class="py-lineno">1269</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_qualifier</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L1270"></a><tt class="py-lineno">1270</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.wgs"></a><div id="_RecordConsumer.wgs-def"><a name="L1271"></a><tt class="py-lineno">1271</tt> <a class="py-toggle" href="#" id="_RecordConsumer.wgs-toggle" onclick="return toggle('_RecordConsumer.wgs');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#wgs">wgs</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.wgs-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.wgs-expanded"><a name="L1272"></a><tt class="py-lineno">1272</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-622" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-622', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-623" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.wgs()=Bio.GenBank._FeatureConsumer-class.html#wgs,Method Bio.GenBank._RecordConsumer.wgs()=Bio.GenBank._RecordConsumer-class.html#wgs"><a title="Bio.GenBank._FeatureConsumer.wgs
Bio.GenBank._RecordConsumer.wgs" class="py-name" href="#" onclick="return doclink('link-623', 'wgs', 'link-623');">wgs</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt><tt class="py-op">.</tt><tt id="link-624" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-624', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">'-'</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1273"></a><tt class="py-lineno">1273</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.add_wgs_scafld"></a><div id="_RecordConsumer.add_wgs_scafld-def"><a name="L1274"></a><tt class="py-lineno">1274</tt> <a class="py-toggle" href="#" id="_RecordConsumer.add_wgs_scafld-toggle" onclick="return toggle('_RecordConsumer.add_wgs_scafld');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#add_wgs_scafld">add_wgs_scafld</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.add_wgs_scafld-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.add_wgs_scafld-expanded"><a name="L1275"></a><tt class="py-lineno">1275</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-625" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-625', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">wgs_scafld</tt><tt class="py-op">.</tt><tt id="link-626" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-626', 'append', 'link-213');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">.</tt><tt id="link-627" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-627', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">'-'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1276"></a><tt class="py-lineno">1276</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.locus"></a><div id="_RecordConsumer.locus-def"><a name="L1277"></a><tt class="py-lineno">1277</tt> <a class="py-toggle" href="#" id="_RecordConsumer.locus-toggle" onclick="return toggle('_RecordConsumer.locus');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#locus">locus</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.locus-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.locus-expanded"><a name="L1278"></a><tt class="py-lineno">1278</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-628" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-628', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-629" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.locus()=Bio.GenBank._FeatureConsumer-class.html#locus,Method Bio.GenBank._RecordConsumer.locus()=Bio.GenBank._RecordConsumer-class.html#locus"><a title="Bio.GenBank._FeatureConsumer.locus
Bio.GenBank._RecordConsumer.locus" class="py-name" href="#" onclick="return doclink('link-629', 'locus', 'link-629');">locus</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1279"></a><tt class="py-lineno">1279</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.size"></a><div id="_RecordConsumer.size-def"><a name="L1280"></a><tt class="py-lineno">1280</tt> <a class="py-toggle" href="#" id="_RecordConsumer.size-toggle" onclick="return toggle('_RecordConsumer.size');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#size">size</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.size-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.size-expanded"><a name="L1281"></a><tt class="py-lineno">1281</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-630" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-630', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-631" class="py-name" targets="Variable Bio.Alphabet.Alphabet.size=Bio.Alphabet.Alphabet-class.html#size,Variable Bio.Alphabet.Reduced.HPModel.size=Bio.Alphabet.Reduced.HPModel-class.html#size,Variable Bio.Alphabet.Reduced.Murphy10.size=Bio.Alphabet.Reduced.Murphy10-class.html#size,Variable Bio.Alphabet.Reduced.Murphy15.size=Bio.Alphabet.Reduced.Murphy15-class.html#size,Variable Bio.Alphabet.Reduced.Murphy4.size=Bio.Alphabet.Reduced.Murphy4-class.html#size,Variable Bio.Alphabet.Reduced.Murphy8.size=Bio.Alphabet.Reduced.Murphy8-class.html#size,Variable Bio.Alphabet.Reduced.PC5.size=Bio.Alphabet.Reduced.PC5-class.html#size,Variable Bio.Alphabet.SingleLetterAlphabet.size=Bio.Alphabet.SingleLetterAlphabet-class.html#size,Variable Bio.Alphabet.ThreeLetterProtein.size=Bio.Alphabet.ThreeLetterProtein-class.html#size,Method Bio.GenBank._FeatureConsumer.size()=Bio.GenBank._FeatureConsumer-class.html#size,Method Bio.GenBank._RecordConsumer.size()=Bio.GenBank._RecordConsumer-class.html#size,Variable Bio.Restriction.Restriction.AanI.size=Bio.Restriction.Restriction.AanI-class.html#size,Variable Bio.Restriction.Restriction.AarI.size=Bio.Restriction.Restriction.AarI-class.html#size,Variable Bio.Restriction.Restriction.AasI.size=Bio.Restriction.Restriction.AasI-class.html#size,Variable Bio.Restriction.Restriction.AatII.size=Bio.Restriction.Restriction.AatII-class.html#size,Variable Bio.Restriction.Restriction.AbaSI.size=Bio.Restriction.Restriction.AbaSI-class.html#size,Variable Bio.Restriction.Restriction.AbsI.size=Bio.Restriction.Restriction.AbsI-class.html#size,Variable Bio.Restriction.Restriction.Acc16I.size=Bio.Restriction.Restriction.Acc16I-class.html#size,Variable Bio.Restriction.Restriction.Acc36I.size=Bio.Restriction.Restriction.Acc36I-class.html#size,Variable Bio.Restriction.Restriction.Acc65I.size=Bio.Restriction.Restriction.Acc65I-class.html#size,Variable Bio.Restriction.Restriction.AccB1I.size=Bio.Restriction.Restriction.AccB1I-class.html#size,Variable Bio.Restriction.Restriction.AccB7I.size=Bio.Restriction.Restriction.AccB7I-class.html#size,Variable Bio.Restriction.Restriction.AccBSI.size=Bio.Restriction.Restriction.AccBSI-class.html#size,Variable Bio.Restriction.Restriction.AccI.size=Bio.Restriction.Restriction.AccI-class.html#size,Variable Bio.Restriction.Restriction.AccII.size=Bio.Restriction.Restriction.AccII-class.html#size,Variable Bio.Restriction.Restriction.AccIII.size=Bio.Restriction.Restriction.AccIII-class.html#size,Variable Bio.Restriction.Restriction.AceIII.size=Bio.Restriction.Restriction.AceIII-class.html#size,Variable Bio.Restriction.Restriction.AciI.size=Bio.Restriction.Restriction.AciI-class.html#size,Variable Bio.Restriction.Restriction.AclI.size=Bio.Restriction.Restriction.AclI-class.html#size,Variable Bio.Restriction.Restriction.AclWI.size=Bio.Restriction.Restriction.AclWI-class.html#size,Variable Bio.Restriction.Restriction.AcoI.size=Bio.Restriction.Restriction.AcoI-class.html#size,Variable Bio.Restriction.Restriction.AcsI.size=Bio.Restriction.Restriction.AcsI-class.html#size,Variable Bio.Restriction.Restriction.AcuI.size=Bio.Restriction.Restriction.AcuI-class.html#size,Variable Bio.Restriction.Restriction.AcvI.size=Bio.Restriction.Restriction.AcvI-class.html#size,Variable Bio.Restriction.Restriction.AcyI.size=Bio.Restriction.Restriction.AcyI-class.html#size,Variable Bio.Restriction.Restriction.AdeI.size=Bio.Restriction.Restriction.AdeI-class.html#size,Variable Bio.Restriction.Restriction.AfaI.size=Bio.Restriction.Restriction.AfaI-class.html#size,Variable Bio.Restriction.Restriction.AfeI.size=Bio.Restriction.Restriction.AfeI-class.html#size,Variable Bio.Restriction.Restriction.AfiI.size=Bio.Restriction.Restriction.AfiI-class.html#size,Variable Bio.Restriction.Restriction.AflII.size=Bio.Restriction.Restriction.AflII-class.html#size,Variable Bio.Restriction.Restriction.AflIII.size=Bio.Restriction.Restriction.AflIII-class.html#size,Variable Bio.Restriction.Restriction.AgeI.size=Bio.Restriction.Restriction.AgeI-class.html#size,Variable Bio.Restriction.Restriction.AgsI.size=Bio.Restriction.Restriction.AgsI-class.html#size,Variable Bio.Restriction.Restriction.AhaIII.size=Bio.Restriction.Restriction.AhaIII-class.html#size,Variable Bio.Restriction.Restriction.AhdI.size=Bio.Restriction.Restriction.AhdI-class.html#size,Variable Bio.Restriction.Restriction.AhlI.size=Bio.Restriction.Restriction.AhlI-class.html#size,Variable Bio.Restriction.Restriction.AjiI.size=Bio.Restriction.Restriction.AjiI-class.html#size,Variable Bio.Restriction.Restriction.AjnI.size=Bio.Restriction.Restriction.AjnI-class.html#size,Variable Bio.Restriction.Restriction.AjuI.size=Bio.Restriction.Restriction.AjuI-class.html#size,Variable Bio.Restriction.Restriction.AleI.size=Bio.Restriction.Restriction.AleI-class.html#size,Variable Bio.Restriction.Restriction.AlfI.size=Bio.Restriction.Restriction.AlfI-class.html#size,Variable Bio.Restriction.Restriction.AloI.size=Bio.Restriction.Restriction.AloI-class.html#size,Variable Bio.Restriction.Restriction.AluBI.size=Bio.Restriction.Restriction.AluBI-class.html#size,Variable Bio.Restriction.Restriction.AluI.size=Bio.Restriction.Restriction.AluI-class.html#size,Variable Bio.Restriction.Restriction.Alw21I.size=Bio.Restriction.Restriction.Alw21I-class.html#size,Variable Bio.Restriction.Restriction.Alw26I.size=Bio.Restriction.Restriction.Alw26I-class.html#size,Variable Bio.Restriction.Restriction.Alw44I.size=Bio.Restriction.Restriction.Alw44I-class.html#size,Variable Bio.Restriction.Restriction.AlwFI.size=Bio.Restriction.Restriction.AlwFI-class.html#size,Variable Bio.Restriction.Restriction.AlwI.size=Bio.Restriction.Restriction.AlwI-class.html#size,Variable Bio.Restriction.Restriction.AlwNI.size=Bio.Restriction.Restriction.AlwNI-class.html#size,Variable Bio.Restriction.Restriction.Ama87I.size=Bio.Restriction.Restriction.Ama87I-class.html#size,Variable Bio.Restriction.Restriction.Aor13HI.size=Bio.Restriction.Restriction.Aor13HI-class.html#size,Variable Bio.Restriction.Restriction.Aor51HI.size=Bio.Restriction.Restriction.Aor51HI-class.html#size,Variable Bio.Restriction.Restriction.AoxI.size=Bio.Restriction.Restriction.AoxI-class.html#size,Variable Bio.Restriction.Restriction.ApaBI.size=Bio.Restriction.Restriction.ApaBI-class.html#size,Variable Bio.Restriction.Restriction.ApaI.size=Bio.Restriction.Restriction.ApaI-class.html#size,Variable Bio.Restriction.Restriction.ApaLI.size=Bio.Restriction.Restriction.ApaLI-class.html#size,Variable Bio.Restriction.Restriction.ApeKI.size=Bio.Restriction.Restriction.ApeKI-class.html#size,Variable Bio.Restriction.Restriction.ApoI.size=Bio.Restriction.Restriction.ApoI-class.html#size,Variable Bio.Restriction.Restriction.ApyPI.size=Bio.Restriction.Restriction.ApyPI-class.html#size,Variable Bio.Restriction.Restriction.AquII.size=Bio.Restriction.Restriction.AquII-class.html#size,Variable Bio.Restriction.Restriction.AquIII.size=Bio.Restriction.Restriction.AquIII-class.html#size,Variable Bio.Restriction.Restriction.AquIV.size=Bio.Restriction.Restriction.AquIV-class.html#size,Variable Bio.Restriction.Restriction.ArsI.size=Bio.Restriction.Restriction.ArsI-class.html#size,Variable Bio.Restriction.Restriction.AscI.size=Bio.Restriction.Restriction.AscI-class.html#size,Variable Bio.Restriction.Restriction.AseI.size=Bio.Restriction.Restriction.AseI-class.html#size,Variable Bio.Restriction.Restriction.Asi256I.size=Bio.Restriction.Restriction.Asi256I-class.html#size,Variable Bio.Restriction.Restriction.AsiGI.size=Bio.Restriction.Restriction.AsiGI-class.html#size,Variable Bio.Restriction.Restriction.AsiSI.size=Bio.Restriction.Restriction.AsiSI-class.html#size,Variable Bio.Restriction.Restriction.Asp700I.size=Bio.Restriction.Restriction.Asp700I-class.html#size,Variable Bio.Restriction.Restriction.Asp718I.size=Bio.Restriction.Restriction.Asp718I-class.html#size,Variable Bio.Restriction.Restriction.AspA2I.size=Bio.Restriction.Restriction.AspA2I-class.html#size,Variable Bio.Restriction.Restriction.AspBHI.size=Bio.Restriction.Restriction.AspBHI-class.html#size,Variable Bio.Restriction.Restriction.AspLEI.size=Bio.Restriction.Restriction.AspLEI-class.html#size,Variable Bio.Restriction.Restriction.AspS9I.size=Bio.Restriction.Restriction.AspS9I-class.html#size,Variable Bio.Restriction.Restriction.AssI.size=Bio.Restriction.Restriction.AssI-class.html#size,Variable Bio.Restriction.Restriction.AsuC2I.size=Bio.Restriction.Restriction.AsuC2I-class.html#size,Variable Bio.Restriction.Restriction.AsuHPI.size=Bio.Restriction.Restriction.AsuHPI-class.html#size,Variable Bio.Restriction.Restriction.AsuI.size=Bio.Restriction.Restriction.AsuI-class.html#size,Variable Bio.Restriction.Restriction.AsuII.size=Bio.Restriction.Restriction.AsuII-class.html#size,Variable Bio.Restriction.Restriction.AsuNHI.size=Bio.Restriction.Restriction.AsuNHI-class.html#size,Variable Bio.Restriction.Restriction.AvaI.size=Bio.Restriction.Restriction.AvaI-class.html#size,Variable Bio.Restriction.Restriction.AvaII.size=Bio.Restriction.Restriction.AvaII-class.html#size,Variable Bio.Restriction.Restriction.AvaIII.size=Bio.Restriction.Restriction.AvaIII-class.html#size,Variable Bio.Restriction.Restriction.AvrII.size=Bio.Restriction.Restriction.AvrII-class.html#size,Variable Bio.Restriction.Restriction.AxyI.size=Bio.Restriction.Restriction.AxyI-class.html#size,Variable Bio.Restriction.Restriction.BaeGI.size=Bio.Restriction.Restriction.BaeGI-class.html#size,Variable Bio.Restriction.Restriction.BaeI.size=Bio.Restriction.Restriction.BaeI-class.html#size,Variable Bio.Restriction.Restriction.BalI.size=Bio.Restriction.Restriction.BalI-class.html#size,Variable Bio.Restriction.Restriction.BamHI.size=Bio.Restriction.Restriction.BamHI-class.html#size,Variable Bio.Restriction.Restriction.BanI.size=Bio.Restriction.Restriction.BanI-class.html#size,Variable Bio.Restriction.Restriction.BanII.size=Bio.Restriction.Restriction.BanII-class.html#size,Variable Bio.Restriction.Restriction.BarI.size=Bio.Restriction.Restriction.BarI-class.html#size,Variable Bio.Restriction.Restriction.BasI.size=Bio.Restriction.Restriction.BasI-class.html#size,Variable Bio.Restriction.Restriction.BauI.size=Bio.Restriction.Restriction.BauI-class.html#size,Variable Bio.Restriction.Restriction.Bbr7I.size=Bio.Restriction.Restriction.Bbr7I-class.html#size,Variable Bio.Restriction.Restriction.BbrPI.size=Bio.Restriction.Restriction.BbrPI-class.html#size,Variable Bio.Restriction.Restriction.BbsI.size=Bio.Restriction.Restriction.BbsI-class.html#size,Variable Bio.Restriction.Restriction.Bbv12I.size=Bio.Restriction.Restriction.Bbv12I-class.html#size,Variable Bio.Restriction.Restriction.BbvCI.size=Bio.Restriction.Restriction.BbvCI-class.html#size,Variable Bio.Restriction.Restriction.BbvI.size=Bio.Restriction.Restriction.BbvI-class.html#size,Variable Bio.Restriction.Restriction.BbvII.size=Bio.Restriction.Restriction.BbvII-class.html#size,Variable Bio.Restriction.Restriction.BccI.size=Bio.Restriction.Restriction.BccI-class.html#size,Variable Bio.Restriction.Restriction.Bce83I.size=Bio.Restriction.Restriction.Bce83I-class.html#size,Variable Bio.Restriction.Restriction.BceAI.size=Bio.Restriction.Restriction.BceAI-class.html#size,Variable Bio.Restriction.Restriction.BcefI.size=Bio.Restriction.Restriction.BcefI-class.html#size,Variable Bio.Restriction.Restriction.BcgI.size=Bio.Restriction.Restriction.BcgI-class.html#size,Variable Bio.Restriction.Restriction.BciT130I.size=Bio.Restriction.Restriction.BciT130I-class.html#size,Variable Bio.Restriction.Restriction.BciVI.size=Bio.Restriction.Restriction.BciVI-class.html#size,Variable Bio.Restriction.Restriction.BclI.size=Bio.Restriction.Restriction.BclI-class.html#size,Variable Bio.Restriction.Restriction.BcnI.size=Bio.Restriction.Restriction.BcnI-class.html#size,Variable Bio.Restriction.Restriction.BcoDI.size=Bio.Restriction.Restriction.BcoDI-class.html#size,Variable Bio.Restriction.Restriction.BcuI.size=Bio.Restriction.Restriction.BcuI-class.html#size,Variable Bio.Restriction.Restriction.BdaI.size=Bio.Restriction.Restriction.BdaI-class.html#size,Variable Bio.Restriction.Restriction.BetI.size=Bio.Restriction.Restriction.BetI-class.html#size,Variable Bio.Restriction.Restriction.BfaI.size=Bio.Restriction.Restriction.BfaI-class.html#size,Variable Bio.Restriction.Restriction.BfiI.size=Bio.Restriction.Restriction.BfiI-class.html#size,Variable Bio.Restriction.Restriction.BfmI.size=Bio.Restriction.Restriction.BfmI-class.html#size,Variable Bio.Restriction.Restriction.BfoI.size=Bio.Restriction.Restriction.BfoI-class.html#size,Variable Bio.Restriction.Restriction.BfrI.size=Bio.Restriction.Restriction.BfrI-class.html#size,Variable Bio.Restriction.Restriction.BfuAI.size=Bio.Restriction.Restriction.BfuAI-class.html#size,Variable Bio.Restriction.Restriction.BfuCI.size=Bio.Restriction.Restriction.BfuCI-class.html#size,Variable Bio.Restriction.Restriction.BfuI.size=Bio.Restriction.Restriction.BfuI-class.html#size,Variable Bio.Restriction.Restriction.BglI.size=Bio.Restriction.Restriction.BglI-class.html#size,Variable Bio.Restriction.Restriction.BglII.size=Bio.Restriction.Restriction.BglII-class.html#size,Variable Bio.Restriction.Restriction.BinI.size=Bio.Restriction.Restriction.BinI-class.html#size,Variable Bio.Restriction.Restriction.BisI.size=Bio.Restriction.Restriction.BisI-class.html#size,Variable Bio.Restriction.Restriction.BlnI.size=Bio.Restriction.Restriction.BlnI-class.html#size,Variable Bio.Restriction.Restriction.BlpI.size=Bio.Restriction.Restriction.BlpI-class.html#size,Variable Bio.Restriction.Restriction.BlsI.size=Bio.Restriction.Restriction.BlsI-class.html#size,Variable Bio.Restriction.Restriction.BmcAI.size=Bio.Restriction.Restriction.BmcAI-class.html#size,Variable Bio.Restriction.Restriction.Bme1390I.size=Bio.Restriction.Restriction.Bme1390I-class.html#size,Variable Bio.Restriction.Restriction.Bme18I.size=Bio.Restriction.Restriction.Bme18I-class.html#size,Variable Bio.Restriction.Restriction.BmeDI.size=Bio.Restriction.Restriction.BmeDI-class.html#size,Variable Bio.Restriction.Restriction.BmeRI.size=Bio.Restriction.Restriction.BmeRI-class.html#size,Variable Bio.Restriction.Restriction.BmeT110I.size=Bio.Restriction.Restriction.BmeT110I-class.html#size,Variable Bio.Restriction.Restriction.BmgBI.size=Bio.Restriction.Restriction.BmgBI-class.html#size,Variable Bio.Restriction.Restriction.BmgI.size=Bio.Restriction.Restriction.BmgI-class.html#size,Variable Bio.Restriction.Restriction.BmgT120I.size=Bio.Restriction.Restriction.BmgT120I-class.html#size,Variable Bio.Restriction.Restriction.BmiI.size=Bio.Restriction.Restriction.BmiI-class.html#size,Variable Bio.Restriction.Restriction.BmrFI.size=Bio.Restriction.Restriction.BmrFI-class.html#size,Variable Bio.Restriction.Restriction.BmrI.size=Bio.Restriction.Restriction.BmrI-class.html#size,Variable Bio.Restriction.Restriction.BmsI.size=Bio.Restriction.Restriction.BmsI-class.html#size,Variable Bio.Restriction.Restriction.BmtI.size=Bio.Restriction.Restriction.BmtI-class.html#size,Variable Bio.Restriction.Restriction.BmuI.size=Bio.Restriction.Restriction.BmuI-class.html#size,Variable Bio.Restriction.Restriction.BoxI.size=Bio.Restriction.Restriction.BoxI-class.html#size,Variable Bio.Restriction.Restriction.BpiI.size=Bio.Restriction.Restriction.BpiI-class.html#size,Variable Bio.Restriction.Restriction.BplI.size=Bio.Restriction.Restriction.BplI-class.html#size,Variable Bio.Restriction.Restriction.BpmI.size=Bio.Restriction.Restriction.BpmI-class.html#size,Variable Bio.Restriction.Restriction.Bpu10I.size=Bio.Restriction.Restriction.Bpu10I-class.html#size,Variable Bio.Restriction.Restriction.Bpu1102I.size=Bio.Restriction.Restriction.Bpu1102I-class.html#size,Variable Bio.Restriction.Restriction.Bpu14I.size=Bio.Restriction.Restriction.Bpu14I-class.html#size,Variable Bio.Restriction.Restriction.BpuEI.size=Bio.Restriction.Restriction.BpuEI-class.html#size,Variable Bio.Restriction.Restriction.BpuMI.size=Bio.Restriction.Restriction.BpuMI-class.html#size,Variable Bio.Restriction.Restriction.BpvUI.size=Bio.Restriction.Restriction.BpvUI-class.html#size,Variable Bio.Restriction.Restriction.Bsa29I.size=Bio.Restriction.Restriction.Bsa29I-class.html#size,Variable Bio.Restriction.Restriction.BsaAI.size=Bio.Restriction.Restriction.BsaAI-class.html#size,Variable Bio.Restriction.Restriction.BsaBI.size=Bio.Restriction.Restriction.BsaBI-class.html#size,Variable Bio.Restriction.Restriction.BsaHI.size=Bio.Restriction.Restriction.BsaHI-class.html#size,Variable Bio.Restriction.Restriction.BsaI.size=Bio.Restriction.Restriction.BsaI-class.html#size,Variable Bio.Restriction.Restriction.BsaJI.size=Bio.Restriction.Restriction.BsaJI-class.html#size,Variable Bio.Restriction.Restriction.BsaWI.size=Bio.Restriction.Restriction.BsaWI-class.html#size,Variable Bio.Restriction.Restriction.BsaXI.size=Bio.Restriction.Restriction.BsaXI-class.html#size,Variable Bio.Restriction.Restriction.BsbI.size=Bio.Restriction.Restriction.BsbI-class.html#size,Variable Bio.Restriction.Restriction.Bsc4I.size=Bio.Restriction.Restriction.Bsc4I-class.html#size,Variable Bio.Restriction.Restriction.BscAI.size=Bio.Restriction.Restriction.BscAI-class.html#size,Variable Bio.Restriction.Restriction.BscGI.size=Bio.Restriction.Restriction.BscGI-class.html#size,Variable Bio.Restriction.Restriction.Bse118I.size=Bio.Restriction.Restriction.Bse118I-class.html#size,Variable Bio.Restriction.Restriction.Bse1I.size=Bio.Restriction.Restriction.Bse1I-class.html#size,Variable Bio.Restriction.Restriction.Bse21I.size=Bio.Restriction.Restriction.Bse21I-class.html#size,Variable Bio.Restriction.Restriction.Bse3DI.size=Bio.Restriction.Restriction.Bse3DI-class.html#size,Variable Bio.Restriction.Restriction.Bse8I.size=Bio.Restriction.Restriction.Bse8I-class.html#size,Variable Bio.Restriction.Restriction.BseAI.size=Bio.Restriction.Restriction.BseAI-class.html#size,Variable Bio.Restriction.Restriction.BseBI.size=Bio.Restriction.Restriction.BseBI-class.html#size,Variable Bio.Restriction.Restriction.BseCI.size=Bio.Restriction.Restriction.BseCI-class.html#size,Variable Bio.Restriction.Restriction.BseDI.size=Bio.Restriction.Restriction.BseDI-class.html#size,Variable Bio.Restriction.Restriction.BseGI.size=Bio.Restriction.Restriction.BseGI-class.html#size,Variable Bio.Restriction.Restriction.BseJI.size=Bio.Restriction.Restriction.BseJI-class.html#size,Variable Bio.Restriction.Restriction.BseLI.size=Bio.Restriction.Restriction.BseLI-class.html#size,Variable Bio.Restriction.Restriction.BseMI.size=Bio.Restriction.Restriction.BseMI-class.html#size,Variable Bio.Restriction.Restriction.BseMII.size=Bio.Restriction.Restriction.BseMII-class.html#size,Variable Bio.Restriction.Restriction.BseNI.size=Bio.Restriction.Restriction.BseNI-class.html#size,Variable Bio.Restriction.Restriction.BsePI.size=Bio.Restriction.Restriction.BsePI-class.html#size,Variable Bio.Restriction.Restriction.BseRI.size=Bio.Restriction.Restriction.BseRI-class.html#size,Variable Bio.Restriction.Restriction.BseSI.size=Bio.Restriction.Restriction.BseSI-class.html#size,Variable Bio.Restriction.Restriction.BseX3I.size=Bio.Restriction.Restriction.BseX3I-class.html#size,Variable Bio.Restriction.Restriction.BseXI.size=Bio.Restriction.Restriction.BseXI-class.html#size,Variable Bio.Restriction.Restriction.BseYI.size=Bio.Restriction.Restriction.BseYI-class.html#size,Variable Bio.Restriction.Restriction.BsgI.size=Bio.Restriction.Restriction.BsgI-class.html#size,Variable Bio.Restriction.Restriction.Bsh1236I.size=Bio.Restriction.Restriction.Bsh1236I-class.html#size,Variable Bio.Restriction.Restriction.Bsh1285I.size=Bio.Restriction.Restriction.Bsh1285I-class.html#size,Variable Bio.Restriction.Restriction.BshFI.size=Bio.Restriction.Restriction.BshFI-class.html#size,Variable Bio.Restriction.Restriction.BshNI.size=Bio.Restriction.Restriction.BshNI-class.html#size,Variable Bio.Restriction.Restriction.BshTI.size=Bio.Restriction.Restriction.BshTI-class.html#size,Variable Bio.Restriction.Restriction.BshVI.size=Bio.Restriction.Restriction.BshVI-class.html#size,Variable Bio.Restriction.Restriction.BsiEI.size=Bio.Restriction.Restriction.BsiEI-class.html#size,Variable Bio.Restriction.Restriction.BsiHKAI.size=Bio.Restriction.Restriction.BsiHKAI-class.html#size,Variable Bio.Restriction.Restriction.BsiHKCI.size=Bio.Restriction.Restriction.BsiHKCI-class.html#size,Variable Bio.Restriction.Restriction.BsiI.size=Bio.Restriction.Restriction.BsiI-class.html#size,Variable Bio.Restriction.Restriction.BsiSI.size=Bio.Restriction.Restriction.BsiSI-class.html#size,Variable Bio.Restriction.Restriction.BsiWI.size=Bio.Restriction.Restriction.BsiWI-class.html#size,Variable Bio.Restriction.Restriction.BsiYI.size=Bio.Restriction.Restriction.BsiYI-class.html#size,Variable Bio.Restriction.Restriction.BslFI.size=Bio.Restriction.Restriction.BslFI-class.html#size,Variable Bio.Restriction.Restriction.BslI.size=Bio.Restriction.Restriction.BslI-class.html#size,Variable Bio.Restriction.Restriction.BsmAI.size=Bio.Restriction.Restriction.BsmAI-class.html#size,Variable Bio.Restriction.Restriction.BsmBI.size=Bio.Restriction.Restriction.BsmBI-class.html#size,Variable Bio.Restriction.Restriction.BsmFI.size=Bio.Restriction.Restriction.BsmFI-class.html#size,Variable Bio.Restriction.Restriction.BsmI.size=Bio.Restriction.Restriction.BsmI-class.html#size,Variable Bio.Restriction.Restriction.BsnI.size=Bio.Restriction.Restriction.BsnI-class.html#size,Variable Bio.Restriction.Restriction.Bso31I.size=Bio.Restriction.Restriction.Bso31I-class.html#size,Variable Bio.Restriction.Restriction.BsoBI.size=Bio.Restriction.Restriction.BsoBI-class.html#size,Variable Bio.Restriction.Restriction.Bsp119I.size=Bio.Restriction.Restriction.Bsp119I-class.html#size,Variable Bio.Restriction.Restriction.Bsp120I.size=Bio.Restriction.Restriction.Bsp120I-class.html#size,Variable Bio.Restriction.Restriction.Bsp1286I.size=Bio.Restriction.Restriction.Bsp1286I-class.html#size,Variable Bio.Restriction.Restriction.Bsp13I.size=Bio.Restriction.Restriction.Bsp13I-class.html#size,Variable Bio.Restriction.Restriction.Bsp1407I.size=Bio.Restriction.Restriction.Bsp1407I-class.html#size,Variable Bio.Restriction.Restriction.Bsp143I.size=Bio.Restriction.Restriction.Bsp143I-class.html#size,Variable Bio.Restriction.Restriction.Bsp1720I.size=Bio.Restriction.Restriction.Bsp1720I-class.html#size,Variable Bio.Restriction.Restriction.Bsp19I.size=Bio.Restriction.Restriction.Bsp19I-class.html#size,Variable Bio.Restriction.Restriction.Bsp24I.size=Bio.Restriction.Restriction.Bsp24I-class.html#size,Variable Bio.Restriction.Restriction.Bsp68I.size=Bio.Restriction.Restriction.Bsp68I-class.html#size,Variable Bio.Restriction.Restriction.BspACI.size=Bio.Restriction.Restriction.BspACI-class.html#size,Variable Bio.Restriction.Restriction.BspCNI.size=Bio.Restriction.Restriction.BspCNI-class.html#size,Variable Bio.Restriction.Restriction.BspD6I.size=Bio.Restriction.Restriction.BspD6I-class.html#size,Variable Bio.Restriction.Restriction.BspDI.size=Bio.Restriction.Restriction.BspDI-class.html#size,Variable Bio.Restriction.Restriction.BspEI.size=Bio.Restriction.Restriction.BspEI-class.html#size,Variable Bio.Restriction.Restriction.BspFNI.size=Bio.Restriction.Restriction.BspFNI-class.html#size,Variable Bio.Restriction.Restriction.BspGI.size=Bio.Restriction.Restriction.BspGI-class.html#size,Variable Bio.Restriction.Restriction.BspHI.size=Bio.Restriction.Restriction.BspHI-class.html#size,Variable Bio.Restriction.Restriction.BspLI.size=Bio.Restriction.Restriction.BspLI-class.html#size,Variable Bio.Restriction.Restriction.BspLU11I.size=Bio.Restriction.Restriction.BspLU11I-class.html#size,Variable Bio.Restriction.Restriction.BspMI.size=Bio.Restriction.Restriction.BspMI-class.html#size,Variable Bio.Restriction.Restriction.BspMII.size=Bio.Restriction.Restriction.BspMII-class.html#size,Variable Bio.Restriction.Restriction.BspNCI.size=Bio.Restriction.Restriction.BspNCI-class.html#size,Variable Bio.Restriction.Restriction.BspOI.size=Bio.Restriction.Restriction.BspOI-class.html#size,Variable Bio.Restriction.Restriction.BspPI.size=Bio.Restriction.Restriction.BspPI-class.html#size,Variable Bio.Restriction.Restriction.BspQI.size=Bio.Restriction.Restriction.BspQI-class.html#size,Variable Bio.Restriction.Restriction.BspT104I.size=Bio.Restriction.Restriction.BspT104I-class.html#size,Variable Bio.Restriction.Restriction.BspT107I.size=Bio.Restriction.Restriction.BspT107I-class.html#size,Variable Bio.Restriction.Restriction.BspTI.size=Bio.Restriction.Restriction.BspTI-class.html#size,Variable Bio.Restriction.Restriction.BsrBI.size=Bio.Restriction.Restriction.BsrBI-class.html#size,Variable Bio.Restriction.Restriction.BsrDI.size=Bio.Restriction.Restriction.BsrDI-class.html#size,Variable Bio.Restriction.Restriction.BsrFI.size=Bio.Restriction.Restriction.BsrFI-class.html#size,Variable Bio.Restriction.Restriction.BsrGI.size=Bio.Restriction.Restriction.BsrGI-class.html#size,Variable Bio.Restriction.Restriction.BsrI.size=Bio.Restriction.Restriction.BsrI-class.html#size,Variable Bio.Restriction.Restriction.BsrSI.size=Bio.Restriction.Restriction.BsrSI-class.html#size,Variable Bio.Restriction.Restriction.BssAI.size=Bio.Restriction.Restriction.BssAI-class.html#size,Variable Bio.Restriction.Restriction.BssECI.size=Bio.Restriction.Restriction.BssECI-class.html#size,Variable Bio.Restriction.Restriction.BssHII.size=Bio.Restriction.Restriction.BssHII-class.html#size,Variable Bio.Restriction.Restriction.BssKI.size=Bio.Restriction.Restriction.BssKI-class.html#size,Variable Bio.Restriction.Restriction.BssMI.size=Bio.Restriction.Restriction.BssMI-class.html#size,Variable Bio.Restriction.Restriction.BssNAI.size=Bio.Restriction.Restriction.BssNAI-class.html#size,Variable Bio.Restriction.Restriction.BssNI.size=Bio.Restriction.Restriction.BssNI-class.html#size,Variable Bio.Restriction.Restriction.BssSI.size=Bio.Restriction.Restriction.BssSI-class.html#size,Variable Bio.Restriction.Restriction.BssT1I.size=Bio.Restriction.Restriction.BssT1I-class.html#size,Variable Bio.Restriction.Restriction.Bst1107I.size=Bio.Restriction.Restriction.Bst1107I-class.html#size,Variable Bio.Restriction.Restriction.Bst2BI.size=Bio.Restriction.Restriction.Bst2BI-class.html#size,Variable Bio.Restriction.Restriction.Bst2UI.size=Bio.Restriction.Restriction.Bst2UI-class.html#size,Variable Bio.Restriction.Restriction.Bst4CI.size=Bio.Restriction.Restriction.Bst4CI-class.html#size,Variable Bio.Restriction.Restriction.Bst6I.size=Bio.Restriction.Restriction.Bst6I-class.html#size,Variable Bio.Restriction.Restriction.BstACI.size=Bio.Restriction.Restriction.BstACI-class.html#size,Variable Bio.Restriction.Restriction.BstAFI.size=Bio.Restriction.Restriction.BstAFI-class.html#size,Variable Bio.Restriction.Restriction.BstAPI.size=Bio.Restriction.Restriction.BstAPI-class.html#size,Variable Bio.Restriction.Restriction.BstAUI.size=Bio.Restriction.Restriction.BstAUI-class.html#size,Variable Bio.Restriction.Restriction.BstBAI.size=Bio.Restriction.Restriction.BstBAI-class.html#size,Variable Bio.Restriction.Restriction.BstBI.size=Bio.Restriction.Restriction.BstBI-class.html#size,Variable Bio.Restriction.Restriction.BstC8I.size=Bio.Restriction.Restriction.BstC8I-class.html#size,Variable Bio.Restriction.Restriction.BstDEI.size=Bio.Restriction.Restriction.BstDEI-class.html#size,Variable Bio.Restriction.Restriction.BstDSI.size=Bio.Restriction.Restriction.BstDSI-class.html#size,Variable Bio.Restriction.Restriction.BstEII.size=Bio.Restriction.Restriction.BstEII-class.html#size,Variable Bio.Restriction.Restriction.BstENI.size=Bio.Restriction.Restriction.BstENI-class.html#size,Variable Bio.Restriction.Restriction.BstF5I.size=Bio.Restriction.Restriction.BstF5I-class.html#size,Variable Bio.Restriction.Restriction.BstFNI.size=Bio.Restriction.Restriction.BstFNI-class.html#size,Variable Bio.Restriction.Restriction.BstH2I.size=Bio.Restriction.Restriction.BstH2I-class.html#size,Variable Bio.Restriction.Restriction.BstHHI.size=Bio.Restriction.Restriction.BstHHI-class.html#size,Variable Bio.Restriction.Restriction.BstKTI.size=Bio.Restriction.Restriction.BstKTI-class.html#size,Variable Bio.Restriction.Restriction.BstMAI.size=Bio.Restriction.Restriction.BstMAI-class.html#size,Variable Bio.Restriction.Restriction.BstMBI.size=Bio.Restriction.Restriction.BstMBI-class.html#size,Variable Bio.Restriction.Restriction.BstMCI.size=Bio.Restriction.Restriction.BstMCI-class.html#size,Variable Bio.Restriction.Restriction.BstMWI.size=Bio.Restriction.Restriction.BstMWI-class.html#size,Variable Bio.Restriction.Restriction.BstNI.size=Bio.Restriction.Restriction.BstNI-class.html#size,Variable Bio.Restriction.Restriction.BstNSI.size=Bio.Restriction.Restriction.BstNSI-class.html#size,Variable Bio.Restriction.Restriction.BstOI.size=Bio.Restriction.Restriction.BstOI-class.html#size,Variable Bio.Restriction.Restriction.BstPAI.size=Bio.Restriction.Restriction.BstPAI-class.html#size,Variable Bio.Restriction.Restriction.BstPI.size=Bio.Restriction.Restriction.BstPI-class.html#size,Variable Bio.Restriction.Restriction.BstSCI.size=Bio.Restriction.Restriction.BstSCI-class.html#size,Variable Bio.Restriction.Restriction.BstSFI.size=Bio.Restriction.Restriction.BstSFI-class.html#size,Variable Bio.Restriction.Restriction.BstSLI.size=Bio.Restriction.Restriction.BstSLI-class.html#size,Variable Bio.Restriction.Restriction.BstSNI.size=Bio.Restriction.Restriction.BstSNI-class.html#size,Variable Bio.Restriction.Restriction.BstUI.size=Bio.Restriction.Restriction.BstUI-class.html#size,Variable Bio.Restriction.Restriction.BstV1I.size=Bio.Restriction.Restriction.BstV1I-class.html#size,Variable Bio.Restriction.Restriction.BstV2I.size=Bio.Restriction.Restriction.BstV2I-class.html#size,Variable Bio.Restriction.Restriction.BstX2I.size=Bio.Restriction.Restriction.BstX2I-class.html#size,Variable Bio.Restriction.Restriction.BstXI.size=Bio.Restriction.Restriction.BstXI-class.html#size,Variable Bio.Restriction.Restriction.BstYI.size=Bio.Restriction.Restriction.BstYI-class.html#size,Variable Bio.Restriction.Restriction.BstZ17I.size=Bio.Restriction.Restriction.BstZ17I-class.html#size,Variable Bio.Restriction.Restriction.BstZI.size=Bio.Restriction.Restriction.BstZI-class.html#size,Variable Bio.Restriction.Restriction.Bsu15I.size=Bio.Restriction.Restriction.Bsu15I-class.html#size,Variable Bio.Restriction.Restriction.Bsu36I.size=Bio.Restriction.Restriction.Bsu36I-class.html#size,Variable Bio.Restriction.Restriction.BsuI.size=Bio.Restriction.Restriction.BsuI-class.html#size,Variable Bio.Restriction.Restriction.BsuRI.size=Bio.Restriction.Restriction.BsuRI-class.html#size,Variable Bio.Restriction.Restriction.BtgI.size=Bio.Restriction.Restriction.BtgI-class.html#size,Variable Bio.Restriction.Restriction.BtgZI.size=Bio.Restriction.Restriction.BtgZI-class.html#size,Variable Bio.Restriction.Restriction.BthCI.size=Bio.Restriction.Restriction.BthCI-class.html#size,Variable Bio.Restriction.Restriction.BtrI.size=Bio.Restriction.Restriction.BtrI-class.html#size,Variable Bio.Restriction.Restriction.BtsCI.size=Bio.Restriction.Restriction.BtsCI-class.html#size,Variable Bio.Restriction.Restriction.BtsI.size=Bio.Restriction.Restriction.BtsI-class.html#size,Variable Bio.Restriction.Restriction.BtsIMutI.size=Bio.Restriction.Restriction.BtsIMutI-class.html#size,Variable Bio.Restriction.Restriction.BtuMI.size=Bio.Restriction.Restriction.BtuMI-class.html#size,Variable Bio.Restriction.Restriction.BveI.size=Bio.Restriction.Restriction.BveI-class.html#size,Variable Bio.Restriction.Restriction.Cac8I.size=Bio.Restriction.Restriction.Cac8I-class.html#size,Variable Bio.Restriction.Restriction.CaiI.size=Bio.Restriction.Restriction.CaiI-class.html#size,Variable Bio.Restriction.Restriction.CauII.size=Bio.Restriction.Restriction.CauII-class.html#size,Variable Bio.Restriction.Restriction.CchII.size=Bio.Restriction.Restriction.CchII-class.html#size,Variable Bio.Restriction.Restriction.CchIII.size=Bio.Restriction.Restriction.CchIII-class.html#size,Variable Bio.Restriction.Restriction.CciI.size=Bio.Restriction.Restriction.CciI-class.html#size,Variable Bio.Restriction.Restriction.CciNI.size=Bio.Restriction.Restriction.CciNI-class.html#size,Variable Bio.Restriction.Restriction.Cdi630V.size=Bio.Restriction.Restriction.Cdi630V-class.html#size,Variable Bio.Restriction.Restriction.CdiI.size=Bio.Restriction.Restriction.CdiI-class.html#size,Variable Bio.Restriction.Restriction.CdpI.size=Bio.Restriction.Restriction.CdpI-class.html#size,Variable Bio.Restriction.Restriction.CfoI.size=Bio.Restriction.Restriction.CfoI-class.html#size,Variable Bio.Restriction.Restriction.Cfr10I.size=Bio.Restriction.Restriction.Cfr10I-class.html#size,Variable Bio.Restriction.Restriction.Cfr13I.size=Bio.Restriction.Restriction.Cfr13I-class.html#size,Variable Bio.Restriction.Restriction.Cfr42I.size=Bio.Restriction.Restriction.Cfr42I-class.html#size,Variable Bio.Restriction.Restriction.Cfr9I.size=Bio.Restriction.Restriction.Cfr9I-class.html#size,Variable Bio.Restriction.Restriction.CfrI.size=Bio.Restriction.Restriction.CfrI-class.html#size,Variable Bio.Restriction.Restriction.Cgl13032I.size=Bio.Restriction.Restriction.Cgl13032I-class.html#size,Variable Bio.Restriction.Restriction.Cgl13032II.size=Bio.Restriction.Restriction.Cgl13032II-class.html#size,Variable Bio.Restriction.Restriction.ChaI.size=Bio.Restriction.Restriction.ChaI-class.html#size,Variable Bio.Restriction.Restriction.CjeFIII.size=Bio.Restriction.Restriction.CjeFIII-class.html#size,Variable Bio.Restriction.Restriction.CjeFV.size=Bio.Restriction.Restriction.CjeFV-class.html#size,Variable Bio.Restriction.Restriction.CjeI.size=Bio.Restriction.Restriction.CjeI-class.html#size,Variable Bio.Restriction.Restriction.CjeNII.size=Bio.Restriction.Restriction.CjeNII-class.html#size,Variable Bio.Restriction.Restriction.CjeNIII.size=Bio.Restriction.Restriction.CjeNIII-class.html#size,Variable Bio.Restriction.Restriction.CjeP659IV.size=Bio.Restriction.Restriction.CjeP659IV-class.html#size,Variable Bio.Restriction.Restriction.CjePI.size=Bio.Restriction.Restriction.CjePI-class.html#size,Variable Bio.Restriction.Restriction.CjuI.size=Bio.Restriction.Restriction.CjuI-class.html#size,Variable Bio.Restriction.Restriction.CjuII.size=Bio.Restriction.Restriction.CjuII-class.html#size,Variable Bio.Restriction.Restriction.ClaI.size=Bio.Restriction.Restriction.ClaI-class.html#size,Variable Bio.Restriction.Restriction.CpoI.size=Bio.Restriction.Restriction.CpoI-class.html#size,Variable Bio.Restriction.Restriction.CseI.size=Bio.Restriction.Restriction.CseI-class.html#size,Variable Bio.Restriction.Restriction.CsiI.size=Bio.Restriction.Restriction.CsiI-class.html#size,Variable Bio.Restriction.Restriction.Csp6I.size=Bio.Restriction.Restriction.Csp6I-class.html#size,Variable Bio.Restriction.Restriction.CspAI.size=Bio.Restriction.Restriction.CspAI-class.html#size,Variable Bio.Restriction.Restriction.CspCI.size=Bio.Restriction.Restriction.CspCI-class.html#size,Variable Bio.Restriction.Restriction.CspI.size=Bio.Restriction.Restriction.CspI-class.html#size,Variable Bio.Restriction.Restriction.CstMI.size=Bio.Restriction.Restriction.CstMI-class.html#size,Variable Bio.Restriction.Restriction.CviAII.size=Bio.Restriction.Restriction.CviAII-class.html#size,Variable Bio.Restriction.Restriction.CviJI.size=Bio.Restriction.Restriction.CviJI-class.html#size,Variable Bio.Restriction.Restriction.CviKI_1.size=Bio.Restriction.Restriction.CviKI_1-class.html#size,Variable Bio.Restriction.Restriction.CviQI.size=Bio.Restriction.Restriction.CviQI-class.html#size,Variable Bio.Restriction.Restriction.CviRI.size=Bio.Restriction.Restriction.CviRI-class.html#size,Variable Bio.Restriction.Restriction.DdeI.size=Bio.Restriction.Restriction.DdeI-class.html#size,Variable Bio.Restriction.Restriction.DinI.size=Bio.Restriction.Restriction.DinI-class.html#size,Variable Bio.Restriction.Restriction.DpnI.size=Bio.Restriction.Restriction.DpnI-class.html#size,Variable Bio.Restriction.Restriction.DpnII.size=Bio.Restriction.Restriction.DpnII-class.html#size,Variable Bio.Restriction.Restriction.DraI.size=Bio.Restriction.Restriction.DraI-class.html#size,Variable Bio.Restriction.Restriction.DraII.size=Bio.Restriction.Restriction.DraII-class.html#size,Variable Bio.Restriction.Restriction.DraIII.size=Bio.Restriction.Restriction.DraIII-class.html#size,Variable Bio.Restriction.Restriction.DraRI.size=Bio.Restriction.Restriction.DraRI-class.html#size,Variable Bio.Restriction.Restriction.DrdI.size=Bio.Restriction.Restriction.DrdI-class.html#size,Variable Bio.Restriction.Restriction.DrdII.size=Bio.Restriction.Restriction.DrdII-class.html#size,Variable Bio.Restriction.Restriction.DriI.size=Bio.Restriction.Restriction.DriI-class.html#size,Variable Bio.Restriction.Restriction.DsaI.size=Bio.Restriction.Restriction.DsaI-class.html#size,Variable Bio.Restriction.Restriction.DseDI.size=Bio.Restriction.Restriction.DseDI-class.html#size,Variable Bio.Restriction.Restriction.EaeI.size=Bio.Restriction.Restriction.EaeI-class.html#size,Variable Bio.Restriction.Restriction.EagI.size=Bio.Restriction.Restriction.EagI-class.html#size,Variable Bio.Restriction.Restriction.Eam1104I.size=Bio.Restriction.Restriction.Eam1104I-class.html#size,Variable Bio.Restriction.Restriction.Eam1105I.size=Bio.Restriction.Restriction.Eam1105I-class.html#size,Variable Bio.Restriction.Restriction.EarI.size=Bio.Restriction.Restriction.EarI-class.html#size,Variable Bio.Restriction.Restriction.EciI.size=Bio.Restriction.Restriction.EciI-class.html#size,Variable Bio.Restriction.Restriction.Ecl136II.size=Bio.Restriction.Restriction.Ecl136II-class.html#size,Variable Bio.Restriction.Restriction.EclXI.size=Bio.Restriction.Restriction.EclXI-class.html#size,Variable Bio.Restriction.Restriction.Eco105I.size=Bio.Restriction.Restriction.Eco105I-class.html#size,Variable Bio.Restriction.Restriction.Eco130I.size=Bio.Restriction.Restriction.Eco130I-class.html#size,Variable Bio.Restriction.Restriction.Eco147I.size=Bio.Restriction.Restriction.Eco147I-class.html#size,Variable Bio.Restriction.Restriction.Eco24I.size=Bio.Restriction.Restriction.Eco24I-class.html#size,Variable Bio.Restriction.Restriction.Eco31I.size=Bio.Restriction.Restriction.Eco31I-class.html#size,Variable Bio.Restriction.Restriction.Eco32I.size=Bio.Restriction.Restriction.Eco32I-class.html#size,Variable Bio.Restriction.Restriction.Eco47I.size=Bio.Restriction.Restriction.Eco47I-class.html#size,Variable Bio.Restriction.Restriction.Eco47III.size=Bio.Restriction.Restriction.Eco47III-class.html#size,Variable Bio.Restriction.Restriction.Eco52I.size=Bio.Restriction.Restriction.Eco52I-class.html#size,Variable Bio.Restriction.Restriction.Eco53kI.size=Bio.Restriction.Restriction.Eco53kI-class.html#size,Variable Bio.Restriction.Restriction.Eco57I.size=Bio.Restriction.Restriction.Eco57I-class.html#size,Variable Bio.Restriction.Restriction.Eco57MI.size=Bio.Restriction.Restriction.Eco57MI-class.html#size,Variable Bio.Restriction.Restriction.Eco72I.size=Bio.Restriction.Restriction.Eco72I-class.html#size,Variable Bio.Restriction.Restriction.Eco81I.size=Bio.Restriction.Restriction.Eco81I-class.html#size,Variable Bio.Restriction.Restriction.Eco88I.size=Bio.Restriction.Restriction.Eco88I-class.html#size,Variable Bio.Restriction.Restriction.Eco91I.size=Bio.Restriction.Restriction.Eco91I-class.html#size,Variable Bio.Restriction.Restriction.EcoHI.size=Bio.Restriction.Restriction.EcoHI-class.html#size,Variable Bio.Restriction.Restriction.EcoICRI.size=Bio.Restriction.Restriction.EcoICRI-class.html#size,Variable Bio.Restriction.Restriction.EcoNI.size=Bio.Restriction.Restriction.EcoNI-class.html#size,Variable Bio.Restriction.Restriction.EcoO109I.size=Bio.Restriction.Restriction.EcoO109I-class.html#size,Variable Bio.Restriction.Restriction.EcoO65I.size=Bio.Restriction.Restriction.EcoO65I-class.html#size,Variable Bio.Restriction.Restriction.EcoRI.size=Bio.Restriction.Restriction.EcoRI-class.html#size,Variable Bio.Restriction.Restriction.EcoRII.size=Bio.Restriction.Restriction.EcoRII-class.html#size,Variable Bio.Restriction.Restriction.EcoRV.size=Bio.Restriction.Restriction.EcoRV-class.html#size,Variable Bio.Restriction.Restriction.EcoT14I.size=Bio.Restriction.Restriction.EcoT14I-class.html#size,Variable Bio.Restriction.Restriction.EcoT22I.size=Bio.Restriction.Restriction.EcoT22I-class.html#size,Variable Bio.Restriction.Restriction.EcoT38I.size=Bio.Restriction.Restriction.EcoT38I-class.html#size,Variable Bio.Restriction.Restriction.EgeI.size=Bio.Restriction.Restriction.EgeI-class.html#size,Variable Bio.Restriction.Restriction.EheI.size=Bio.Restriction.Restriction.EheI-class.html#size,Variable Bio.Restriction.Restriction.ErhI.size=Bio.Restriction.Restriction.ErhI-class.html#size,Variable Bio.Restriction.Restriction.EsaBC3I.size=Bio.Restriction.Restriction.EsaBC3I-class.html#size,Variable Bio.Restriction.Restriction.EsaSSI.size=Bio.Restriction.Restriction.EsaSSI-class.html#size,Variable Bio.Restriction.Restriction.Esp3I.size=Bio.Restriction.Restriction.Esp3I-class.html#size,Variable Bio.Restriction.Restriction.EspI.size=Bio.Restriction.Restriction.EspI-class.html#size,Variable Bio.Restriction.Restriction.FaeI.size=Bio.Restriction.Restriction.FaeI-class.html#size,Variable Bio.Restriction.Restriction.FaiI.size=Bio.Restriction.Restriction.FaiI-class.html#size,Variable Bio.Restriction.Restriction.FalI.size=Bio.Restriction.Restriction.FalI-class.html#size,Variable Bio.Restriction.Restriction.FaqI.size=Bio.Restriction.Restriction.FaqI-class.html#size,Variable Bio.Restriction.Restriction.FatI.size=Bio.Restriction.Restriction.FatI-class.html#size,Variable Bio.Restriction.Restriction.FauI.size=Bio.Restriction.Restriction.FauI-class.html#size,Variable Bio.Restriction.Restriction.FauNDI.size=Bio.Restriction.Restriction.FauNDI-class.html#size,Variable Bio.Restriction.Restriction.FbaI.size=Bio.Restriction.Restriction.FbaI-class.html#size,Variable Bio.Restriction.Restriction.FblI.size=Bio.Restriction.Restriction.FblI-class.html#size,Variable Bio.Restriction.Restriction.FinI.size=Bio.Restriction.Restriction.FinI-class.html#size,Variable Bio.Restriction.Restriction.FmuI.size=Bio.Restriction.Restriction.FmuI-class.html#size,Variable Bio.Restriction.Restriction.Fnu4HI.size=Bio.Restriction.Restriction.Fnu4HI-class.html#size,Variable Bio.Restriction.Restriction.FnuDII.size=Bio.Restriction.Restriction.FnuDII-class.html#size,Variable Bio.Restriction.Restriction.FokI.size=Bio.Restriction.Restriction.FokI-class.html#size,Variable Bio.Restriction.Restriction.FriOI.size=Bio.Restriction.Restriction.FriOI-class.html#size,Variable Bio.Restriction.Restriction.FseI.size=Bio.Restriction.Restriction.FseI-class.html#size,Variable Bio.Restriction.Restriction.Fsp4HI.size=Bio.Restriction.Restriction.Fsp4HI-class.html#size,Variable Bio.Restriction.Restriction.FspAI.size=Bio.Restriction.Restriction.FspAI-class.html#size,Variable Bio.Restriction.Restriction.FspBI.size=Bio.Restriction.Restriction.FspBI-class.html#size,Variable Bio.Restriction.Restriction.FspEI.size=Bio.Restriction.Restriction.FspEI-class.html#size,Variable Bio.Restriction.Restriction.FspI.size=Bio.Restriction.Restriction.FspI-class.html#size,Variable Bio.Restriction.Restriction.GauT27I.size=Bio.Restriction.Restriction.GauT27I-class.html#size,Variable Bio.Restriction.Restriction.GdiII.size=Bio.Restriction.Restriction.GdiII-class.html#size,Variable Bio.Restriction.Restriction.GlaI.size=Bio.Restriction.Restriction.GlaI-class.html#size,Variable Bio.Restriction.Restriction.GluI.size=Bio.Restriction.Restriction.GluI-class.html#size,Variable Bio.Restriction.Restriction.GsaI.size=Bio.Restriction.Restriction.GsaI-class.html#size,Variable Bio.Restriction.Restriction.GsuI.size=Bio.Restriction.Restriction.GsuI-class.html#size,Variable Bio.Restriction.Restriction.HaeI.size=Bio.Restriction.Restriction.HaeI-class.html#size,Variable Bio.Restriction.Restriction.HaeII.size=Bio.Restriction.Restriction.HaeII-class.html#size,Variable Bio.Restriction.Restriction.HaeIII.size=Bio.Restriction.Restriction.HaeIII-class.html#size,Variable Bio.Restriction.Restriction.HapII.size=Bio.Restriction.Restriction.HapII-class.html#size,Variable Bio.Restriction.Restriction.HauII.size=Bio.Restriction.Restriction.HauII-class.html#size,Variable Bio.Restriction.Restriction.HgaI.size=Bio.Restriction.Restriction.HgaI-class.html#size,Variable Bio.Restriction.Restriction.HgiAI.size=Bio.Restriction.Restriction.HgiAI-class.html#size,Variable Bio.Restriction.Restriction.HgiCI.size=Bio.Restriction.Restriction.HgiCI-class.html#size,Variable Bio.Restriction.Restriction.HgiEII.size=Bio.Restriction.Restriction.HgiEII-class.html#size,Variable Bio.Restriction.Restriction.HgiJII.size=Bio.Restriction.Restriction.HgiJII-class.html#size,Variable Bio.Restriction.Restriction.HhaI.size=Bio.Restriction.Restriction.HhaI-class.html#size,Variable Bio.Restriction.Restriction.Hin1I.size=Bio.Restriction.Restriction.Hin1I-class.html#size,Variable Bio.Restriction.Restriction.Hin1II.size=Bio.Restriction.Restriction.Hin1II-class.html#size,Variable Bio.Restriction.Restriction.Hin4I.size=Bio.Restriction.Restriction.Hin4I-class.html#size,Variable Bio.Restriction.Restriction.Hin4II.size=Bio.Restriction.Restriction.Hin4II-class.html#size,Variable Bio.Restriction.Restriction.Hin6I.size=Bio.Restriction.Restriction.Hin6I-class.html#size,Variable Bio.Restriction.Restriction.HinP1I.size=Bio.Restriction.Restriction.HinP1I-class.html#size,Variable Bio.Restriction.Restriction.HincII.size=Bio.Restriction.Restriction.HincII-class.html#size,Variable Bio.Restriction.Restriction.HindII.size=Bio.Restriction.Restriction.HindII-class.html#size,Variable Bio.Restriction.Restriction.HindIII.size=Bio.Restriction.Restriction.HindIII-class.html#size,Variable Bio.Restriction.Restriction.HinfI.size=Bio.Restriction.Restriction.HinfI-class.html#size,Variable Bio.Restriction.Restriction.HpaI.size=Bio.Restriction.Restriction.HpaI-class.html#size,Variable Bio.Restriction.Restriction.HpaII.size=Bio.Restriction.Restriction.HpaII-class.html#size,Variable Bio.Restriction.Restriction.HphI.size=Bio.Restriction.Restriction.HphI-class.html#size,Variable Bio.Restriction.Restriction.Hpy166II.size=Bio.Restriction.Restriction.Hpy166II-class.html#size,Variable Bio.Restriction.Restriction.Hpy178III.size=Bio.Restriction.Restriction.Hpy178III-class.html#size,Variable Bio.Restriction.Restriction.Hpy188I.size=Bio.Restriction.Restriction.Hpy188I-class.html#size,Variable Bio.Restriction.Restriction.Hpy188III.size=Bio.Restriction.Restriction.Hpy188III-class.html#size,Variable Bio.Restriction.Restriction.Hpy8I.size=Bio.Restriction.Restriction.Hpy8I-class.html#size,Variable Bio.Restriction.Restriction.Hpy99I.size=Bio.Restriction.Restriction.Hpy99I-class.html#size,Variable Bio.Restriction.Restriction.Hpy99XIII.size=Bio.Restriction.Restriction.Hpy99XIII-class.html#size,Variable Bio.Restriction.Restriction.Hpy99XIV.size=Bio.Restriction.Restriction.Hpy99XIV-class.html#size,Variable Bio.Restriction.Restriction.HpyAV.size=Bio.Restriction.Restriction.HpyAV-class.html#size,Variable Bio.Restriction.Restriction.HpyCH4III.size=Bio.Restriction.Restriction.HpyCH4III-class.html#size,Variable Bio.Restriction.Restriction.HpyCH4IV.size=Bio.Restriction.Restriction.HpyCH4IV-class.html#size,Variable Bio.Restriction.Restriction.HpyCH4V.size=Bio.Restriction.Restriction.HpyCH4V-class.html#size,Variable Bio.Restriction.Restriction.HpyF10VI.size=Bio.Restriction.Restriction.HpyF10VI-class.html#size,Variable Bio.Restriction.Restriction.HpyF3I.size=Bio.Restriction.Restriction.HpyF3I-class.html#size,Variable Bio.Restriction.Restriction.HpySE526I.size=Bio.Restriction.Restriction.HpySE526I-class.html#size,Variable Bio.Restriction.Restriction.Hsp92I.size=Bio.Restriction.Restriction.Hsp92I-class.html#size,Variable Bio.Restriction.Restriction.Hsp92II.size=Bio.Restriction.Restriction.Hsp92II-class.html#size,Variable Bio.Restriction.Restriction.HspAI.size=Bio.Restriction.Restriction.HspAI-class.html#size,Variable Bio.Restriction.Restriction.Jma19592I.size=Bio.Restriction.Restriction.Jma19592I-class.html#size,Variable Bio.Restriction.Restriction.KasI.size=Bio.Restriction.Restriction.KasI-class.html#size,Variable Bio.Restriction.Restriction.KflI.size=Bio.Restriction.Restriction.KflI-class.html#size,Variable Bio.Restriction.Restriction.Kpn2I.size=Bio.Restriction.Restriction.Kpn2I-class.html#size,Variable Bio.Restriction.Restriction.KpnI.size=Bio.Restriction.Restriction.KpnI-class.html#size,Variable Bio.Restriction.Restriction.KroI.size=Bio.Restriction.Restriction.KroI-class.html#size,Variable Bio.Restriction.Restriction.Ksp22I.size=Bio.Restriction.Restriction.Ksp22I-class.html#size,Variable Bio.Restriction.Restriction.Ksp632I.size=Bio.Restriction.Restriction.Ksp632I-class.html#size,Variable Bio.Restriction.Restriction.KspAI.size=Bio.Restriction.Restriction.KspAI-class.html#size,Variable Bio.Restriction.Restriction.KspI.size=Bio.Restriction.Restriction.KspI-class.html#size,Variable Bio.Restriction.Restriction.Kzo9I.size=Bio.Restriction.Restriction.Kzo9I-class.html#size,Variable Bio.Restriction.Restriction.LguI.size=Bio.Restriction.Restriction.LguI-class.html#size,Variable Bio.Restriction.Restriction.LpnI.size=Bio.Restriction.Restriction.LpnI-class.html#size,Variable Bio.Restriction.Restriction.LpnPI.size=Bio.Restriction.Restriction.LpnPI-class.html#size,Variable Bio.Restriction.Restriction.Lsp1109I.size=Bio.Restriction.Restriction.Lsp1109I-class.html#size,Variable Bio.Restriction.Restriction.LweI.size=Bio.Restriction.Restriction.LweI-class.html#size,Variable Bio.Restriction.Restriction.MabI.size=Bio.Restriction.Restriction.MabI-class.html#size,Variable Bio.Restriction.Restriction.MaeI.size=Bio.Restriction.Restriction.MaeI-class.html#size,Variable Bio.Restriction.Restriction.MaeII.size=Bio.Restriction.Restriction.MaeII-class.html#size,Variable Bio.Restriction.Restriction.MaeIII.size=Bio.Restriction.Restriction.MaeIII-class.html#size,Variable Bio.Restriction.Restriction.MalI.size=Bio.Restriction.Restriction.MalI-class.html#size,Variable Bio.Restriction.Restriction.MaqI.size=Bio.Restriction.Restriction.MaqI-class.html#size,Variable Bio.Restriction.Restriction.MauBI.size=Bio.Restriction.Restriction.MauBI-class.html#size,Variable Bio.Restriction.Restriction.MbiI.size=Bio.Restriction.Restriction.MbiI-class.html#size,Variable Bio.Restriction.Restriction.MboI.size=Bio.Restriction.Restriction.MboI-class.html#size,Variable Bio.Restriction.Restriction.MboII.size=Bio.Restriction.Restriction.MboII-class.html#size,Variable Bio.Restriction.Restriction.McaTI.size=Bio.Restriction.Restriction.McaTI-class.html#size,Variable Bio.Restriction.Restriction.McrI.size=Bio.Restriction.Restriction.McrI-class.html#size,Variable Bio.Restriction.Restriction.MfeI.size=Bio.Restriction.Restriction.MfeI-class.html#size,Variable Bio.Restriction.Restriction.MflI.size=Bio.Restriction.Restriction.MflI-class.html#size,Variable Bio.Restriction.Restriction.MhlI.size=Bio.Restriction.Restriction.MhlI-class.html#size,Variable Bio.Restriction.Restriction.MjaIV.size=Bio.Restriction.Restriction.MjaIV-class.html#size,Variable Bio.Restriction.Restriction.MkaDII.size=Bio.Restriction.Restriction.MkaDII-class.html#size,Variable Bio.Restriction.Restriction.MlsI.size=Bio.Restriction.Restriction.MlsI-class.html#size,Variable Bio.Restriction.Restriction.MluCI.size=Bio.Restriction.Restriction.MluCI-class.html#size,Variable Bio.Restriction.Restriction.MluI.size=Bio.Restriction.Restriction.MluI-class.html#size,Variable Bio.Restriction.Restriction.MluNI.size=Bio.Restriction.Restriction.MluNI-class.html#size,Variable Bio.Restriction.Restriction.Mly113I.size=Bio.Restriction.Restriction.Mly113I-class.html#size,Variable Bio.Restriction.Restriction.MlyI.size=Bio.Restriction.Restriction.MlyI-class.html#size,Variable Bio.Restriction.Restriction.MmeI.size=Bio.Restriction.Restriction.MmeI-class.html#size,Variable Bio.Restriction.Restriction.MnlI.size=Bio.Restriction.Restriction.MnlI-class.html#size,Variable Bio.Restriction.Restriction.Mph1103I.size=Bio.Restriction.Restriction.Mph1103I-class.html#size,Variable Bio.Restriction.Restriction.MreI.size=Bio.Restriction.Restriction.MreI-class.html#size,Variable Bio.Restriction.Restriction.MroI.size=Bio.Restriction.Restriction.MroI-class.html#size,Variable Bio.Restriction.Restriction.MroNI.size=Bio.Restriction.Restriction.MroNI-class.html#size,Variable Bio.Restriction.Restriction.MroXI.size=Bio.Restriction.Restriction.MroXI-class.html#size,Variable Bio.Restriction.Restriction.MscI.size=Bio.Restriction.Restriction.MscI-class.html#size,Variable Bio.Restriction.Restriction.MseI.size=Bio.Restriction.Restriction.MseI-class.html#size,Variable Bio.Restriction.Restriction.MslI.size=Bio.Restriction.Restriction.MslI-class.html#size,Variable Bio.Restriction.Restriction.Msp20I.size=Bio.Restriction.Restriction.Msp20I-class.html#size,Variable Bio.Restriction.Restriction.MspA1I.size=Bio.Restriction.Restriction.MspA1I-class.html#size,Variable Bio.Restriction.Restriction.MspCI.size=Bio.Restriction.Restriction.MspCI-class.html#size,Variable Bio.Restriction.Restriction.MspI.size=Bio.Restriction.Restriction.MspI-class.html#size,Variable Bio.Restriction.Restriction.MspJI.size=Bio.Restriction.Restriction.MspJI-class.html#size,Variable Bio.Restriction.Restriction.MspR9I.size=Bio.Restriction.Restriction.MspR9I-class.html#size,Variable Bio.Restriction.Restriction.MssI.size=Bio.Restriction.Restriction.MssI-class.html#size,Variable Bio.Restriction.Restriction.MstI.size=Bio.Restriction.Restriction.MstI-class.html#size,Variable Bio.Restriction.Restriction.MunI.size=Bio.Restriction.Restriction.MunI-class.html#size,Variable Bio.Restriction.Restriction.Mva1269I.size=Bio.Restriction.Restriction.Mva1269I-class.html#size,Variable Bio.Restriction.Restriction.MvaI.size=Bio.Restriction.Restriction.MvaI-class.html#size,Variable Bio.Restriction.Restriction.MvnI.size=Bio.Restriction.Restriction.MvnI-class.html#size,Variable Bio.Restriction.Restriction.MvrI.size=Bio.Restriction.Restriction.MvrI-class.html#size,Variable Bio.Restriction.Restriction.MwoI.size=Bio.Restriction.Restriction.MwoI-class.html#size,Variable Bio.Restriction.Restriction.NaeI.size=Bio.Restriction.Restriction.NaeI-class.html#size,Variable Bio.Restriction.Restriction.NarI.size=Bio.Restriction.Restriction.NarI-class.html#size,Variable Bio.Restriction.Restriction.NciI.size=Bio.Restriction.Restriction.NciI-class.html#size,Variable Bio.Restriction.Restriction.NcoI.size=Bio.Restriction.Restriction.NcoI-class.html#size,Variable Bio.Restriction.Restriction.NdeI.size=Bio.Restriction.Restriction.NdeI-class.html#size,Variable Bio.Restriction.Restriction.NdeII.size=Bio.Restriction.Restriction.NdeII-class.html#size,Variable Bio.Restriction.Restriction.NgoAVIII.size=Bio.Restriction.Restriction.NgoAVIII-class.html#size,Variable Bio.Restriction.Restriction.NgoMIV.size=Bio.Restriction.Restriction.NgoMIV-class.html#size,Variable Bio.Restriction.Restriction.NhaXI.size=Bio.Restriction.Restriction.NhaXI-class.html#size,Variable Bio.Restriction.Restriction.NheI.size=Bio.Restriction.Restriction.NheI-class.html#size,Variable Bio.Restriction.Restriction.NlaCI.size=Bio.Restriction.Restriction.NlaCI-class.html#size,Variable Bio.Restriction.Restriction.NlaIII.size=Bio.Restriction.Restriction.NlaIII-class.html#size,Variable Bio.Restriction.Restriction.NlaIV.size=Bio.Restriction.Restriction.NlaIV-class.html#size,Variable Bio.Restriction.Restriction.Nli3877I.size=Bio.Restriction.Restriction.Nli3877I-class.html#size,Variable Bio.Restriction.Restriction.NmeAIII.size=Bio.Restriction.Restriction.NmeAIII-class.html#size,Variable Bio.Restriction.Restriction.NmeDI.size=Bio.Restriction.Restriction.NmeDI-class.html#size,Variable Bio.Restriction.Restriction.NmuCI.size=Bio.Restriction.Restriction.NmuCI-class.html#size,Variable Bio.Restriction.Restriction.NotI.size=Bio.Restriction.Restriction.NotI-class.html#size,Variable Bio.Restriction.Restriction.NruI.size=Bio.Restriction.Restriction.NruI-class.html#size,Variable Bio.Restriction.Restriction.NsbI.size=Bio.Restriction.Restriction.NsbI-class.html#size,Variable Bio.Restriction.Restriction.NsiI.size=Bio.Restriction.Restriction.NsiI-class.html#size,Variable Bio.Restriction.Restriction.NspBII.size=Bio.Restriction.Restriction.NspBII-class.html#size,Variable Bio.Restriction.Restriction.NspI.size=Bio.Restriction.Restriction.NspI-class.html#size,Variable Bio.Restriction.Restriction.NspV.size=Bio.Restriction.Restriction.NspV-class.html#size,Variable Bio.Restriction.Restriction.OliI.size=Bio.Restriction.Restriction.OliI-class.html#size,Variable Bio.Restriction.Restriction.PabI.size=Bio.Restriction.Restriction.PabI-class.html#size,Variable Bio.Restriction.Restriction.PacI.size=Bio.Restriction.Restriction.PacI-class.html#size,Variable Bio.Restriction.Restriction.PaeI.size=Bio.Restriction.Restriction.PaeI-class.html#size,Variable Bio.Restriction.Restriction.PaeR7I.size=Bio.Restriction.Restriction.PaeR7I-class.html#size,Variable Bio.Restriction.Restriction.PagI.size=Bio.Restriction.Restriction.PagI-class.html#size,Variable Bio.Restriction.Restriction.PalAI.size=Bio.Restriction.Restriction.PalAI-class.html#size,Variable Bio.Restriction.Restriction.PasI.size=Bio.Restriction.Restriction.PasI-class.html#size,Variable Bio.Restriction.Restriction.PauI.size=Bio.Restriction.Restriction.PauI-class.html#size,Variable Bio.Restriction.Restriction.PceI.size=Bio.Restriction.Restriction.PceI-class.html#size,Variable Bio.Restriction.Restriction.PciI.size=Bio.Restriction.Restriction.PciI-class.html#size,Variable Bio.Restriction.Restriction.PciSI.size=Bio.Restriction.Restriction.PciSI-class.html#size,Variable Bio.Restriction.Restriction.PcsI.size=Bio.Restriction.Restriction.PcsI-class.html#size,Variable Bio.Restriction.Restriction.PctI.size=Bio.Restriction.Restriction.PctI-class.html#size,Variable Bio.Restriction.Restriction.PdiI.size=Bio.Restriction.Restriction.PdiI-class.html#size,Variable Bio.Restriction.Restriction.PdmI.size=Bio.Restriction.Restriction.PdmI-class.html#size,Variable Bio.Restriction.Restriction.PenI.size=Bio.Restriction.Restriction.PenI-class.html#size,Variable Bio.Restriction.Restriction.PfeI.size=Bio.Restriction.Restriction.PfeI-class.html#size,Variable Bio.Restriction.Restriction.Pfl1108I.size=Bio.Restriction.Restriction.Pfl1108I-class.html#size,Variable Bio.Restriction.Restriction.Pfl23II.size=Bio.Restriction.Restriction.Pfl23II-class.html#size,Variable Bio.Restriction.Restriction.PflFI.size=Bio.Restriction.Restriction.PflFI-class.html#size,Variable Bio.Restriction.Restriction.PflMI.size=Bio.Restriction.Restriction.PflMI-class.html#size,Variable Bio.Restriction.Restriction.PfoI.size=Bio.Restriction.Restriction.PfoI-class.html#size,Variable Bio.Restriction.Restriction.PinAI.size=Bio.Restriction.Restriction.PinAI-class.html#size,Variable Bio.Restriction.Restriction.PlaDI.size=Bio.Restriction.Restriction.PlaDI-class.html#size,Variable Bio.Restriction.Restriction.Ple19I.size=Bio.Restriction.Restriction.Ple19I-class.html#size,Variable Bio.Restriction.Restriction.PleI.size=Bio.Restriction.Restriction.PleI-class.html#size,Variable Bio.Restriction.Restriction.PluTI.size=Bio.Restriction.Restriction.PluTI-class.html#size,Variable Bio.Restriction.Restriction.PmaCI.size=Bio.Restriction.Restriction.PmaCI-class.html#size,Variable Bio.Restriction.Restriction.PmeI.size=Bio.Restriction.Restriction.PmeI-class.html#size,Variable Bio.Restriction.Restriction.PmlI.size=Bio.Restriction.Restriction.PmlI-class.html#size,Variable Bio.Restriction.Restriction.PpiI.size=Bio.Restriction.Restriction.PpiI-class.html#size,Variable Bio.Restriction.Restriction.PpsI.size=Bio.Restriction.Restriction.PpsI-class.html#size,Variable Bio.Restriction.Restriction.Ppu10I.size=Bio.Restriction.Restriction.Ppu10I-class.html#size,Variable Bio.Restriction.Restriction.Ppu21I.size=Bio.Restriction.Restriction.Ppu21I-class.html#size,Variable Bio.Restriction.Restriction.PpuMI.size=Bio.Restriction.Restriction.PpuMI-class.html#size,Variable Bio.Restriction.Restriction.PscI.size=Bio.Restriction.Restriction.PscI-class.html#size,Variable Bio.Restriction.Restriction.PshAI.size=Bio.Restriction.Restriction.PshAI-class.html#size,Variable Bio.Restriction.Restriction.PshBI.size=Bio.Restriction.Restriction.PshBI-class.html#size,Variable Bio.Restriction.Restriction.PsiI.size=Bio.Restriction.Restriction.PsiI-class.html#size,Variable Bio.Restriction.Restriction.Psp03I.size=Bio.Restriction.Restriction.Psp03I-class.html#size,Variable Bio.Restriction.Restriction.Psp124BI.size=Bio.Restriction.Restriction.Psp124BI-class.html#size,Variable Bio.Restriction.Restriction.Psp1406I.size=Bio.Restriction.Restriction.Psp1406I-class.html#size,Variable Bio.Restriction.Restriction.Psp5II.size=Bio.Restriction.Restriction.Psp5II-class.html#size,Variable Bio.Restriction.Restriction.Psp6I.size=Bio.Restriction.Restriction.Psp6I-class.html#size,Variable Bio.Restriction.Restriction.PspCI.size=Bio.Restriction.Restriction.PspCI-class.html#size,Variable Bio.Restriction.Restriction.PspEI.size=Bio.Restriction.Restriction.PspEI-class.html#size,Variable Bio.Restriction.Restriction.PspGI.size=Bio.Restriction.Restriction.PspGI-class.html#size,Variable Bio.Restriction.Restriction.PspLI.size=Bio.Restriction.Restriction.PspLI-class.html#size,Variable Bio.Restriction.Restriction.PspN4I.size=Bio.Restriction.Restriction.PspN4I-class.html#size,Variable Bio.Restriction.Restriction.PspOMI.size=Bio.Restriction.Restriction.PspOMI-class.html#size,Variable Bio.Restriction.Restriction.PspOMII.size=Bio.Restriction.Restriction.PspOMII-class.html#size,Variable Bio.Restriction.Restriction.PspPI.size=Bio.Restriction.Restriction.PspPI-class.html#size,Variable Bio.Restriction.Restriction.PspPPI.size=Bio.Restriction.Restriction.PspPPI-class.html#size,Variable Bio.Restriction.Restriction.PspPRI.size=Bio.Restriction.Restriction.PspPRI-class.html#size,Variable Bio.Restriction.Restriction.PspXI.size=Bio.Restriction.Restriction.PspXI-class.html#size,Variable Bio.Restriction.Restriction.PsrI.size=Bio.Restriction.Restriction.PsrI-class.html#size,Variable Bio.Restriction.Restriction.PssI.size=Bio.Restriction.Restriction.PssI-class.html#size,Variable Bio.Restriction.Restriction.PstI.size=Bio.Restriction.Restriction.PstI-class.html#size,Variable Bio.Restriction.Restriction.PstNI.size=Bio.Restriction.Restriction.PstNI-class.html#size,Variable Bio.Restriction.Restriction.PsuI.size=Bio.Restriction.Restriction.PsuI-class.html#size,Variable Bio.Restriction.Restriction.PsyI.size=Bio.Restriction.Restriction.PsyI-class.html#size,Variable Bio.Restriction.Restriction.PteI.size=Bio.Restriction.Restriction.PteI-class.html#size,Variable Bio.Restriction.Restriction.PvuI.size=Bio.Restriction.Restriction.PvuI-class.html#size,Variable Bio.Restriction.Restriction.PvuII.size=Bio.Restriction.Restriction.PvuII-class.html#size,Variable Bio.Restriction.Restriction.R2_BceSIV.size=Bio.Restriction.Restriction.R2_BceSIV-class.html#size,Variable Bio.Restriction.Restriction.RceI.size=Bio.Restriction.Restriction.RceI-class.html#size,Variable Bio.Restriction.Restriction.RdeGBI.size=Bio.Restriction.Restriction.RdeGBI-class.html#size,Variable Bio.Restriction.Restriction.RdeGBII.size=Bio.Restriction.Restriction.RdeGBII-class.html#size,Variable Bio.Restriction.Restriction.RdeGBIII.size=Bio.Restriction.Restriction.RdeGBIII-class.html#size,Variable Bio.Restriction.Restriction.RflFIII.size=Bio.Restriction.Restriction.RflFIII-class.html#size,Variable Bio.Restriction.Restriction.RgaI.size=Bio.Restriction.Restriction.RgaI-class.html#size,Variable Bio.Restriction.Restriction.RigI.size=Bio.Restriction.Restriction.RigI-class.html#size,Variable Bio.Restriction.Restriction.RlaI.size=Bio.Restriction.Restriction.RlaI-class.html#size,Variable Bio.Restriction.Restriction.RleAI.size=Bio.Restriction.Restriction.RleAI-class.html#size,Variable Bio.Restriction.Restriction.RpaB5I.size=Bio.Restriction.Restriction.RpaB5I-class.html#size,Variable Bio.Restriction.Restriction.RpaBI.size=Bio.Restriction.Restriction.RpaBI-class.html#size,Variable Bio.Restriction.Restriction.RpaI.size=Bio.Restriction.Restriction.RpaI-class.html#size,Variable Bio.Restriction.Restriction.RpaTI.size=Bio.Restriction.Restriction.RpaTI-class.html#size,Variable Bio.Restriction.Restriction.RruI.size=Bio.Restriction.Restriction.RruI-class.html#size,Variable Bio.Restriction.Restriction.RsaI.size=Bio.Restriction.Restriction.RsaI-class.html#size,Variable Bio.Restriction.Restriction.RsaNI.size=Bio.Restriction.Restriction.RsaNI-class.html#size,Variable Bio.Restriction.Restriction.RseI.size=Bio.Restriction.Restriction.RseI-class.html#size,Variable Bio.Restriction.Restriction.Rsr2I.size=Bio.Restriction.Restriction.Rsr2I-class.html#size,Variable Bio.Restriction.Restriction.RsrII.size=Bio.Restriction.Restriction.RsrII-class.html#size,Variable Bio.Restriction.Restriction.SacI.size=Bio.Restriction.Restriction.SacI-class.html#size,Variable Bio.Restriction.Restriction.SacII.size=Bio.Restriction.Restriction.SacII-class.html#size,Variable Bio.Restriction.Restriction.SalI.size=Bio.Restriction.Restriction.SalI-class.html#size,Variable Bio.Restriction.Restriction.SanDI.size=Bio.Restriction.Restriction.SanDI-class.html#size,Variable Bio.Restriction.Restriction.SapI.size=Bio.Restriction.Restriction.SapI-class.html#size,Variable Bio.Restriction.Restriction.SaqAI.size=Bio.Restriction.Restriction.SaqAI-class.html#size,Variable Bio.Restriction.Restriction.SatI.size=Bio.Restriction.Restriction.SatI-class.html#size,Variable Bio.Restriction.Restriction.Sau3AI.size=Bio.Restriction.Restriction.Sau3AI-class.html#size,Variable Bio.Restriction.Restriction.Sau96I.size=Bio.Restriction.Restriction.Sau96I-class.html#size,Variable Bio.Restriction.Restriction.SauI.size=Bio.Restriction.Restriction.SauI-class.html#size,Variable Bio.Restriction.Restriction.SbfI.size=Bio.Restriction.Restriction.SbfI-class.html#size,Variable Bio.Restriction.Restriction.ScaI.size=Bio.Restriction.Restriction.ScaI-class.html#size,Variable Bio.Restriction.Restriction.SchI.size=Bio.Restriction.Restriction.SchI-class.html#size,Variable Bio.Restriction.Restriction.SciI.size=Bio.Restriction.Restriction.SciI-class.html#size,Variable Bio.Restriction.Restriction.ScrFI.size=Bio.Restriction.Restriction.ScrFI-class.html#size,Variable Bio.Restriction.Restriction.SdaI.size=Bio.Restriction.Restriction.SdaI-class.html#size,Variable Bio.Restriction.Restriction.SdeAI.size=Bio.Restriction.Restriction.SdeAI-class.html#size,Variable Bio.Restriction.Restriction.SdeOSI.size=Bio.Restriction.Restriction.SdeOSI-class.html#size,Variable Bio.Restriction.Restriction.SduI.size=Bio.Restriction.Restriction.SduI-class.html#size,Variable Bio.Restriction.Restriction.SecI.size=Bio.Restriction.Restriction.SecI-class.html#size,Variable Bio.Restriction.Restriction.SelI.size=Bio.Restriction.Restriction.SelI-class.html#size,Variable Bio.Restriction.Restriction.SetI.size=Bio.Restriction.Restriction.SetI-class.html#size,Variable Bio.Restriction.Restriction.SexAI.size=Bio.Restriction.Restriction.SexAI-class.html#size,Variable Bio.Restriction.Restriction.SfaAI.size=Bio.Restriction.Restriction.SfaAI-class.html#size,Variable Bio.Restriction.Restriction.SfaNI.size=Bio.Restriction.Restriction.SfaNI-class.html#size,Variable Bio.Restriction.Restriction.SfcI.size=Bio.Restriction.Restriction.SfcI-class.html#size,Variable Bio.Restriction.Restriction.SfeI.size=Bio.Restriction.Restriction.SfeI-class.html#size,Variable Bio.Restriction.Restriction.SfiI.size=Bio.Restriction.Restriction.SfiI-class.html#size,Variable Bio.Restriction.Restriction.SfoI.size=Bio.Restriction.Restriction.SfoI-class.html#size,Variable Bio.Restriction.Restriction.Sfr274I.size=Bio.Restriction.Restriction.Sfr274I-class.html#size,Variable Bio.Restriction.Restriction.Sfr303I.size=Bio.Restriction.Restriction.Sfr303I-class.html#size,Variable Bio.Restriction.Restriction.SfuI.size=Bio.Restriction.Restriction.SfuI-class.html#size,Variable Bio.Restriction.Restriction.SgeI.size=Bio.Restriction.Restriction.SgeI-class.html#size,Variable Bio.Restriction.Restriction.SgfI.size=Bio.Restriction.Restriction.SgfI-class.html#size,Variable Bio.Restriction.Restriction.SgrAI.size=Bio.Restriction.Restriction.SgrAI-class.html#size,Variable Bio.Restriction.Restriction.SgrBI.size=Bio.Restriction.Restriction.SgrBI-class.html#size,Variable Bio.Restriction.Restriction.SgrDI.size=Bio.Restriction.Restriction.SgrDI-class.html#size,Variable Bio.Restriction.Restriction.SgrTI.size=Bio.Restriction.Restriction.SgrTI-class.html#size,Variable Bio.Restriction.Restriction.SgsI.size=Bio.Restriction.Restriction.SgsI-class.html#size,Variable Bio.Restriction.Restriction.SimI.size=Bio.Restriction.Restriction.SimI-class.html#size,Variable Bio.Restriction.Restriction.SlaI.size=Bio.Restriction.Restriction.SlaI-class.html#size,Variable Bio.Restriction.Restriction.SmaI.size=Bio.Restriction.Restriction.SmaI-class.html#size,Variable Bio.Restriction.Restriction.SmiI.size=Bio.Restriction.Restriction.SmiI-class.html#size,Variable Bio.Restriction.Restriction.SmiMI.size=Bio.Restriction.Restriction.SmiMI-class.html#size,Variable Bio.Restriction.Restriction.SmlI.size=Bio.Restriction.Restriction.SmlI-class.html#size,Variable Bio.Restriction.Restriction.SmoI.size=Bio.Restriction.Restriction.SmoI-class.html#size,Variable Bio.Restriction.Restriction.SnaBI.size=Bio.Restriction.Restriction.SnaBI-class.html#size,Variable Bio.Restriction.Restriction.SnaI.size=Bio.Restriction.Restriction.SnaI-class.html#size,Variable Bio.Restriction.Restriction.Sno506I.size=Bio.Restriction.Restriction.Sno506I-class.html#size,Variable Bio.Restriction.Restriction.SpeI.size=Bio.Restriction.Restriction.SpeI-class.html#size,Variable Bio.Restriction.Restriction.SphI.size=Bio.Restriction.Restriction.SphI-class.html#size,Variable Bio.Restriction.Restriction.SplI.size=Bio.Restriction.Restriction.SplI-class.html#size,Variable Bio.Restriction.Restriction.SpoDI.size=Bio.Restriction.Restriction.SpoDI-class.html#size,Variable Bio.Restriction.Restriction.SrfI.size=Bio.Restriction.Restriction.SrfI-class.html#size,Variable Bio.Restriction.Restriction.Sse232I.size=Bio.Restriction.Restriction.Sse232I-class.html#size,Variable Bio.Restriction.Restriction.Sse8387I.size=Bio.Restriction.Restriction.Sse8387I-class.html#size,Variable Bio.Restriction.Restriction.Sse8647I.size=Bio.Restriction.Restriction.Sse8647I-class.html#size,Variable Bio.Restriction.Restriction.Sse9I.size=Bio.Restriction.Restriction.Sse9I-class.html#size,Variable Bio.Restriction.Restriction.SseBI.size=Bio.Restriction.Restriction.SseBI-class.html#size,Variable Bio.Restriction.Restriction.SsiI.size=Bio.Restriction.Restriction.SsiI-class.html#size,Variable Bio.Restriction.Restriction.SspD5I.size=Bio.Restriction.Restriction.SspD5I-class.html#size,Variable Bio.Restriction.Restriction.SspDI.size=Bio.Restriction.Restriction.SspDI-class.html#size,Variable Bio.Restriction.Restriction.SspI.size=Bio.Restriction.Restriction.SspI-class.html#size,Variable Bio.Restriction.Restriction.SstE37I.size=Bio.Restriction.Restriction.SstE37I-class.html#size,Variable Bio.Restriction.Restriction.SstI.size=Bio.Restriction.Restriction.SstI-class.html#size,Variable Bio.Restriction.Restriction.Sth132I.size=Bio.Restriction.Restriction.Sth132I-class.html#size,Variable Bio.Restriction.Restriction.Sth302II.size=Bio.Restriction.Restriction.Sth302II-class.html#size,Variable Bio.Restriction.Restriction.StrI.size=Bio.Restriction.Restriction.StrI-class.html#size,Variable Bio.Restriction.Restriction.StsI.size=Bio.Restriction.Restriction.StsI-class.html#size,Variable Bio.Restriction.Restriction.StuI.size=Bio.Restriction.Restriction.StuI-class.html#size,Variable Bio.Restriction.Restriction.StyD4I.size=Bio.Restriction.Restriction.StyD4I-class.html#size,Variable Bio.Restriction.Restriction.StyI.size=Bio.Restriction.Restriction.StyI-class.html#size,Variable Bio.Restriction.Restriction.SwaI.size=Bio.Restriction.Restriction.SwaI-class.html#size,Variable Bio.Restriction.Restriction.TaaI.size=Bio.Restriction.Restriction.TaaI-class.html#size,Variable Bio.Restriction.Restriction.TaiI.size=Bio.Restriction.Restriction.TaiI-class.html#size,Variable Bio.Restriction.Restriction.TaqI.size=Bio.Restriction.Restriction.TaqI-class.html#size,Variable Bio.Restriction.Restriction.TaqII.size=Bio.Restriction.Restriction.TaqII-class.html#size,Variable Bio.Restriction.Restriction.TasI.size=Bio.Restriction.Restriction.TasI-class.html#size,Variable Bio.Restriction.Restriction.TatI.size=Bio.Restriction.Restriction.TatI-class.html#size,Variable Bio.Restriction.Restriction.TauI.size=Bio.Restriction.Restriction.TauI-class.html#size,Variable Bio.Restriction.Restriction.TfiI.size=Bio.Restriction.Restriction.TfiI-class.html#size,Variable Bio.Restriction.Restriction.Tru1I.size=Bio.Restriction.Restriction.Tru1I-class.html#size,Variable Bio.Restriction.Restriction.Tru9I.size=Bio.Restriction.Restriction.Tru9I-class.html#size,Variable Bio.Restriction.Restriction.TscAI.size=Bio.Restriction.Restriction.TscAI-class.html#size,Variable Bio.Restriction.Restriction.TseFI.size=Bio.Restriction.Restriction.TseFI-class.html#size,Variable Bio.Restriction.Restriction.TseI.size=Bio.Restriction.Restriction.TseI-class.html#size,Variable Bio.Restriction.Restriction.TsoI.size=Bio.Restriction.Restriction.TsoI-class.html#size,Variable Bio.Restriction.Restriction.Tsp45I.size=Bio.Restriction.Restriction.Tsp45I-class.html#size,Variable Bio.Restriction.Restriction.Tsp4CI.size=Bio.Restriction.Restriction.Tsp4CI-class.html#size,Variable Bio.Restriction.Restriction.TspDTI.size=Bio.Restriction.Restriction.TspDTI-class.html#size,Variable Bio.Restriction.Restriction.TspEI.size=Bio.Restriction.Restriction.TspEI-class.html#size,Variable Bio.Restriction.Restriction.TspGWI.size=Bio.Restriction.Restriction.TspGWI-class.html#size,Variable Bio.Restriction.Restriction.TspMI.size=Bio.Restriction.Restriction.TspMI-class.html#size,Variable Bio.Restriction.Restriction.TspRI.size=Bio.Restriction.Restriction.TspRI-class.html#size,Variable Bio.Restriction.Restriction.TssI.size=Bio.Restriction.Restriction.TssI-class.html#size,Variable Bio.Restriction.Restriction.TstI.size=Bio.Restriction.Restriction.TstI-class.html#size,Variable Bio.Restriction.Restriction.TsuI.size=Bio.Restriction.Restriction.TsuI-class.html#size,Variable Bio.Restriction.Restriction.Tth111I.size=Bio.Restriction.Restriction.Tth111I-class.html#size,Variable Bio.Restriction.Restriction.Tth111II.size=Bio.Restriction.Restriction.Tth111II-class.html#size,Variable Bio.Restriction.Restriction.UbaF11I.size=Bio.Restriction.Restriction.UbaF11I-class.html#size,Variable Bio.Restriction.Restriction.UbaF12I.size=Bio.Restriction.Restriction.UbaF12I-class.html#size,Variable Bio.Restriction.Restriction.UbaF13I.size=Bio.Restriction.Restriction.UbaF13I-class.html#size,Variable Bio.Restriction.Restriction.UbaF14I.size=Bio.Restriction.Restriction.UbaF14I-class.html#size,Variable Bio.Restriction.Restriction.UbaF9I.size=Bio.Restriction.Restriction.UbaF9I-class.html#size,Variable Bio.Restriction.Restriction.UbaPI.size=Bio.Restriction.Restriction.UbaPI-class.html#size,Variable Bio.Restriction.Restriction.UcoMSI.size=Bio.Restriction.Restriction.UcoMSI-class.html#size,Variable Bio.Restriction.Restriction.UnbI.size=Bio.Restriction.Restriction.UnbI-class.html#size,Variable Bio.Restriction.Restriction.Van91I.size=Bio.Restriction.Restriction.Van91I-class.html#size,Variable Bio.Restriction.Restriction.Vha464I.size=Bio.Restriction.Restriction.Vha464I-class.html#size,Variable Bio.Restriction.Restriction.VneI.size=Bio.Restriction.Restriction.VneI-class.html#size,Variable Bio.Restriction.Restriction.VpaK11AI.size=Bio.Restriction.Restriction.VpaK11AI-class.html#size,Variable Bio.Restriction.Restriction.VpaK11BI.size=Bio.Restriction.Restriction.VpaK11BI-class.html#size,Variable Bio.Restriction.Restriction.VspI.size=Bio.Restriction.Restriction.VspI-class.html#size,Variable Bio.Restriction.Restriction.WviI.size=Bio.Restriction.Restriction.WviI-class.html#size,Variable Bio.Restriction.Restriction.XagI.size=Bio.Restriction.Restriction.XagI-class.html#size,Variable Bio.Restriction.Restriction.XapI.size=Bio.Restriction.Restriction.XapI-class.html#size,Variable Bio.Restriction.Restriction.XbaI.size=Bio.Restriction.Restriction.XbaI-class.html#size,Variable Bio.Restriction.Restriction.XceI.size=Bio.Restriction.Restriction.XceI-class.html#size,Variable Bio.Restriction.Restriction.XcmI.size=Bio.Restriction.Restriction.XcmI-class.html#size,Variable Bio.Restriction.Restriction.XhoI.size=Bio.Restriction.Restriction.XhoI-class.html#size,Variable Bio.Restriction.Restriction.XhoII.size=Bio.Restriction.Restriction.XhoII-class.html#size,Variable Bio.Restriction.Restriction.XmaI.size=Bio.Restriction.Restriction.XmaI-class.html#size,Variable Bio.Restriction.Restriction.XmaIII.size=Bio.Restriction.Restriction.XmaIII-class.html#size,Variable Bio.Restriction.Restriction.XmaJI.size=Bio.Restriction.Restriction.XmaJI-class.html#size,Variable Bio.Restriction.Restriction.XmiI.size=Bio.Restriction.Restriction.XmiI-class.html#size,Variable Bio.Restriction.Restriction.XmnI.size=Bio.Restriction.Restriction.XmnI-class.html#size,Variable Bio.Restriction.Restriction.XspI.size=Bio.Restriction.Restriction.XspI-class.html#size,Variable Bio.Restriction.Restriction.YkrI.size=Bio.Restriction.Restriction.YkrI-class.html#size,Variable Bio.Restriction.Restriction.ZraI.size=Bio.Restriction.Restriction.ZraI-class.html#size,Variable Bio.Restriction.Restriction.ZrmI.size=Bio.Restriction.Restriction.ZrmI-class.html#size,Variable Bio.Restriction.Restriction.Zsp2I.size=Bio.Restriction.Restriction.Zsp2I-class.html#size,Variable Bio.codonalign.codonalphabet.CodonAlphabet.size=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#size"><a title="Bio.Alphabet.Alphabet.size
Bio.Alphabet.Reduced.HPModel.size
Bio.Alphabet.Reduced.Murphy10.size
Bio.Alphabet.Reduced.Murphy15.size
Bio.Alphabet.Reduced.Murphy4.size
Bio.Alphabet.Reduced.Murphy8.size
Bio.Alphabet.Reduced.PC5.size
Bio.Alphabet.SingleLetterAlphabet.size
Bio.Alphabet.ThreeLetterProtein.size
Bio.GenBank._FeatureConsumer.size
Bio.GenBank._RecordConsumer.size
Bio.Restriction.Restriction.AanI.size
Bio.Restriction.Restriction.AarI.size
Bio.Restriction.Restriction.AasI.size
Bio.Restriction.Restriction.AatII.size
Bio.Restriction.Restriction.AbaSI.size
Bio.Restriction.Restriction.AbsI.size
Bio.Restriction.Restriction.Acc16I.size
Bio.Restriction.Restriction.Acc36I.size
Bio.Restriction.Restriction.Acc65I.size
Bio.Restriction.Restriction.AccB1I.size
Bio.Restriction.Restriction.AccB7I.size
Bio.Restriction.Restriction.AccBSI.size
Bio.Restriction.Restriction.AccI.size
Bio.Restriction.Restriction.AccII.size
Bio.Restriction.Restriction.AccIII.size
Bio.Restriction.Restriction.AceIII.size
Bio.Restriction.Restriction.AciI.size
Bio.Restriction.Restriction.AclI.size
Bio.Restriction.Restriction.AclWI.size
Bio.Restriction.Restriction.AcoI.size
Bio.Restriction.Restriction.AcsI.size
Bio.Restriction.Restriction.AcuI.size
Bio.Restriction.Restriction.AcvI.size
Bio.Restriction.Restriction.AcyI.size
Bio.Restriction.Restriction.AdeI.size
Bio.Restriction.Restriction.AfaI.size
Bio.Restriction.Restriction.AfeI.size
Bio.Restriction.Restriction.AfiI.size
Bio.Restriction.Restriction.AflII.size
Bio.Restriction.Restriction.AflIII.size
Bio.Restriction.Restriction.AgeI.size
Bio.Restriction.Restriction.AgsI.size
Bio.Restriction.Restriction.AhaIII.size
Bio.Restriction.Restriction.AhdI.size
Bio.Restriction.Restriction.AhlI.size
Bio.Restriction.Restriction.AjiI.size
Bio.Restriction.Restriction.AjnI.size
Bio.Restriction.Restriction.AjuI.size
Bio.Restriction.Restriction.AleI.size
Bio.Restriction.Restriction.AlfI.size
Bio.Restriction.Restriction.AloI.size
Bio.Restriction.Restriction.AluBI.size
Bio.Restriction.Restriction.AluI.size
Bio.Restriction.Restriction.Alw21I.size
Bio.Restriction.Restriction.Alw26I.size
Bio.Restriction.Restriction.Alw44I.size
Bio.Restriction.Restriction.AlwFI.size
Bio.Restriction.Restriction.AlwI.size
Bio.Restriction.Restriction.AlwNI.size
Bio.Restriction.Restriction.Ama87I.size
Bio.Restriction.Restriction.Aor13HI.size
Bio.Restriction.Restriction.Aor51HI.size
Bio.Restriction.Restriction.AoxI.size
Bio.Restriction.Restriction.ApaBI.size
Bio.Restriction.Restriction.ApaI.size
Bio.Restriction.Restriction.ApaLI.size
Bio.Restriction.Restriction.ApeKI.size
Bio.Restriction.Restriction.ApoI.size
Bio.Restriction.Restriction.ApyPI.size
Bio.Restriction.Restriction.AquII.size
Bio.Restriction.Restriction.AquIII.size
Bio.Restriction.Restriction.AquIV.size
Bio.Restriction.Restriction.ArsI.size
Bio.Restriction.Restriction.AscI.size
Bio.Restriction.Restriction.AseI.size
Bio.Restriction.Restriction.Asi256I.size
Bio.Restriction.Restriction.AsiGI.size
Bio.Restriction.Restriction.AsiSI.size
Bio.Restriction.Restriction.Asp700I.size
Bio.Restriction.Restriction.Asp718I.size
Bio.Restriction.Restriction.AspA2I.size
Bio.Restriction.Restriction.AspBHI.size
Bio.Restriction.Restriction.AspLEI.size
Bio.Restriction.Restriction.AspS9I.size
Bio.Restriction.Restriction.AssI.size
Bio.Restriction.Restriction.AsuC2I.size
Bio.Restriction.Restriction.AsuHPI.size
Bio.Restriction.Restriction.AsuI.size
Bio.Restriction.Restriction.AsuII.size
Bio.Restriction.Restriction.AsuNHI.size
Bio.Restriction.Restriction.AvaI.size
Bio.Restriction.Restriction.AvaII.size
Bio.Restriction.Restriction.AvaIII.size
Bio.Restriction.Restriction.AvrII.size
Bio.Restriction.Restriction.AxyI.size
Bio.Restriction.Restriction.BaeGI.size
Bio.Restriction.Restriction.BaeI.size
Bio.Restriction.Restriction.BalI.size
Bio.Restriction.Restriction.BamHI.size
Bio.Restriction.Restriction.BanI.size
Bio.Restriction.Restriction.BanII.size
Bio.Restriction.Restriction.BarI.size
Bio.Restriction.Restriction.BasI.size
Bio.Restriction.Restriction.BauI.size
Bio.Restriction.Restriction.Bbr7I.size
Bio.Restriction.Restriction.BbrPI.size
Bio.Restriction.Restriction.BbsI.size
Bio.Restriction.Restriction.Bbv12I.size
Bio.Restriction.Restriction.BbvCI.size
Bio.Restriction.Restriction.BbvI.size
Bio.Restriction.Restriction.BbvII.size
Bio.Restriction.Restriction.BccI.size
Bio.Restriction.Restriction.Bce83I.size
Bio.Restriction.Restriction.BceAI.size
Bio.Restriction.Restriction.BcefI.size
Bio.Restriction.Restriction.BcgI.size
Bio.Restriction.Restriction.BciT130I.size
Bio.Restriction.Restriction.BciVI.size
Bio.Restriction.Restriction.BclI.size
Bio.Restriction.Restriction.BcnI.size
Bio.Restriction.Restriction.BcoDI.size
Bio.Restriction.Restriction.BcuI.size
Bio.Restriction.Restriction.BdaI.size
Bio.Restriction.Restriction.BetI.size
Bio.Restriction.Restriction.BfaI.size
Bio.Restriction.Restriction.BfiI.size
Bio.Restriction.Restriction.BfmI.size
Bio.Restriction.Restriction.BfoI.size
Bio.Restriction.Restriction.BfrI.size
Bio.Restriction.Restriction.BfuAI.size
Bio.Restriction.Restriction.BfuCI.size
Bio.Restriction.Restriction.BfuI.size
Bio.Restriction.Restriction.BglI.size
Bio.Restriction.Restriction.BglII.size
Bio.Restriction.Restriction.BinI.size
Bio.Restriction.Restriction.BisI.size
Bio.Restriction.Restriction.BlnI.size
Bio.Restriction.Restriction.BlpI.size
Bio.Restriction.Restriction.BlsI.size
Bio.Restriction.Restriction.BmcAI.size
Bio.Restriction.Restriction.Bme1390I.size
Bio.Restriction.Restriction.Bme18I.size
Bio.Restriction.Restriction.BmeDI.size
Bio.Restriction.Restriction.BmeRI.size
Bio.Restriction.Restriction.BmeT110I.size
Bio.Restriction.Restriction.BmgBI.size
Bio.Restriction.Restriction.BmgI.size
Bio.Restriction.Restriction.BmgT120I.size
Bio.Restriction.Restriction.BmiI.size
Bio.Restriction.Restriction.BmrFI.size
Bio.Restriction.Restriction.BmrI.size
Bio.Restriction.Restriction.BmsI.size
Bio.Restriction.Restriction.BmtI.size
Bio.Restriction.Restriction.BmuI.size
Bio.Restriction.Restriction.BoxI.size
Bio.Restriction.Restriction.BpiI.size
Bio.Restriction.Restriction.BplI.size
Bio.Restriction.Restriction.BpmI.size
Bio.Restriction.Restriction.Bpu10I.size
Bio.Restriction.Restriction.Bpu1102I.size
Bio.Restriction.Restriction.Bpu14I.size
Bio.Restriction.Restriction.BpuEI.size
Bio.Restriction.Restriction.BpuMI.size
Bio.Restriction.Restriction.BpvUI.size
Bio.Restriction.Restriction.Bsa29I.size
Bio.Restriction.Restriction.BsaAI.size
Bio.Restriction.Restriction.BsaBI.size
Bio.Restriction.Restriction.BsaHI.size
Bio.Restriction.Restriction.BsaI.size
Bio.Restriction.Restriction.BsaJI.size
Bio.Restriction.Restriction.BsaWI.size
Bio.Restriction.Restriction.BsaXI.size
Bio.Restriction.Restriction.BsbI.size
Bio.Restriction.Restriction.Bsc4I.size
Bio.Restriction.Restriction.BscAI.size
Bio.Restriction.Restriction.BscGI.size
Bio.Restriction.Restriction.Bse118I.size
Bio.Restriction.Restriction.Bse1I.size
Bio.Restriction.Restriction.Bse21I.size
Bio.Restriction.Restriction.Bse3DI.size
Bio.Restriction.Restriction.Bse8I.size
Bio.Restriction.Restriction.BseAI.size
Bio.Restriction.Restriction.BseBI.size
Bio.Restriction.Restriction.BseCI.size
Bio.Restriction.Restriction.BseDI.size
Bio.Restriction.Restriction.BseGI.size
Bio.Restriction.Restriction.BseJI.size
Bio.Restriction.Restriction.BseLI.size
Bio.Restriction.Restriction.BseMI.size
Bio.Restriction.Restriction.BseMII.size
Bio.Restriction.Restriction.BseNI.size
Bio.Restriction.Restriction.BsePI.size
Bio.Restriction.Restriction.BseRI.size
Bio.Restriction.Restriction.BseSI.size
Bio.Restriction.Restriction.BseX3I.size
Bio.Restriction.Restriction.BseXI.size
Bio.Restriction.Restriction.BseYI.size
Bio.Restriction.Restriction.BsgI.size
Bio.Restriction.Restriction.Bsh1236I.size
Bio.Restriction.Restriction.Bsh1285I.size
Bio.Restriction.Restriction.BshFI.size
Bio.Restriction.Restriction.BshNI.size
Bio.Restriction.Restriction.BshTI.size
Bio.Restriction.Restriction.BshVI.size
Bio.Restriction.Restriction.BsiEI.size
Bio.Restriction.Restriction.BsiHKAI.size
Bio.Restriction.Restriction.BsiHKCI.size
Bio.Restriction.Restriction.BsiI.size
Bio.Restriction.Restriction.BsiSI.size
Bio.Restriction.Restriction.BsiWI.size
Bio.Restriction.Restriction.BsiYI.size
Bio.Restriction.Restriction.BslFI.size
Bio.Restriction.Restriction.BslI.size
Bio.Restriction.Restriction.BsmAI.size
Bio.Restriction.Restriction.BsmBI.size
Bio.Restriction.Restriction.BsmFI.size
Bio.Restriction.Restriction.BsmI.size
Bio.Restriction.Restriction.BsnI.size
Bio.Restriction.Restriction.Bso31I.size
Bio.Restriction.Restriction.BsoBI.size
Bio.Restriction.Restriction.Bsp119I.size
Bio.Restriction.Restriction.Bsp120I.size
Bio.Restriction.Restriction.Bsp1286I.size
Bio.Restriction.Restriction.Bsp13I.size
Bio.Restriction.Restriction.Bsp1407I.size
Bio.Restriction.Restriction.Bsp143I.size
Bio.Restriction.Restriction.Bsp1720I.size
Bio.Restriction.Restriction.Bsp19I.size
Bio.Restriction.Restriction.Bsp24I.size
Bio.Restriction.Restriction.Bsp68I.size
Bio.Restriction.Restriction.BspACI.size
Bio.Restriction.Restriction.BspCNI.size
Bio.Restriction.Restriction.BspD6I.size
Bio.Restriction.Restriction.BspDI.size
Bio.Restriction.Restriction.BspEI.size
Bio.Restriction.Restriction.BspFNI.size
Bio.Restriction.Restriction.BspGI.size
Bio.Restriction.Restriction.BspHI.size
Bio.Restriction.Restriction.BspLI.size
Bio.Restriction.Restriction.BspLU11I.size
Bio.Restriction.Restriction.BspMI.size
Bio.Restriction.Restriction.BspMII.size
Bio.Restriction.Restriction.BspNCI.size
Bio.Restriction.Restriction.BspOI.size
Bio.Restriction.Restriction.BspPI.size
Bio.Restriction.Restriction.BspQI.size
Bio.Restriction.Restriction.BspT104I.size
Bio.Restriction.Restriction.BspT107I.size
Bio.Restriction.Restriction.BspTI.size
Bio.Restriction.Restriction.BsrBI.size
Bio.Restriction.Restriction.BsrDI.size
Bio.Restriction.Restriction.BsrFI.size
Bio.Restriction.Restriction.BsrGI.size
Bio.Restriction.Restriction.BsrI.size
Bio.Restriction.Restriction.BsrSI.size
Bio.Restriction.Restriction.BssAI.size
Bio.Restriction.Restriction.BssECI.size
Bio.Restriction.Restriction.BssHII.size
Bio.Restriction.Restriction.BssKI.size
Bio.Restriction.Restriction.BssMI.size
Bio.Restriction.Restriction.BssNAI.size
Bio.Restriction.Restriction.BssNI.size
Bio.Restriction.Restriction.BssSI.size
Bio.Restriction.Restriction.BssT1I.size
Bio.Restriction.Restriction.Bst1107I.size
Bio.Restriction.Restriction.Bst2BI.size
Bio.Restriction.Restriction.Bst2UI.size
Bio.Restriction.Restriction.Bst4CI.size
Bio.Restriction.Restriction.Bst6I.size
Bio.Restriction.Restriction.BstACI.size
Bio.Restriction.Restriction.BstAFI.size
Bio.Restriction.Restriction.BstAPI.size
Bio.Restriction.Restriction.BstAUI.size
Bio.Restriction.Restriction.BstBAI.size
Bio.Restriction.Restriction.BstBI.size
Bio.Restriction.Restriction.BstC8I.size
Bio.Restriction.Restriction.BstDEI.size
Bio.Restriction.Restriction.BstDSI.size
Bio.Restriction.Restriction.BstEII.size
Bio.Restriction.Restriction.BstENI.size
Bio.Restriction.Restriction.BstF5I.size
Bio.Restriction.Restriction.BstFNI.size
Bio.Restriction.Restriction.BstH2I.size
Bio.Restriction.Restriction.BstHHI.size
Bio.Restriction.Restriction.BstKTI.size
Bio.Restriction.Restriction.BstMAI.size
Bio.Restriction.Restriction.BstMBI.size
Bio.Restriction.Restriction.BstMCI.size
Bio.Restriction.Restriction.BstMWI.size
Bio.Restriction.Restriction.BstNI.size
Bio.Restriction.Restriction.BstNSI.size
Bio.Restriction.Restriction.BstOI.size
Bio.Restriction.Restriction.BstPAI.size
Bio.Restriction.Restriction.BstPI.size
Bio.Restriction.Restriction.BstSCI.size
Bio.Restriction.Restriction.BstSFI.size
Bio.Restriction.Restriction.BstSLI.size
Bio.Restriction.Restriction.BstSNI.size
Bio.Restriction.Restriction.BstUI.size
Bio.Restriction.Restriction.BstV1I.size
Bio.Restriction.Restriction.BstV2I.size
Bio.Restriction.Restriction.BstX2I.size
Bio.Restriction.Restriction.BstXI.size
Bio.Restriction.Restriction.BstYI.size
Bio.Restriction.Restriction.BstZ17I.size
Bio.Restriction.Restriction.BstZI.size
Bio.Restriction.Restriction.Bsu15I.size
Bio.Restriction.Restriction.Bsu36I.size
Bio.Restriction.Restriction.BsuI.size
Bio.Restriction.Restriction.BsuRI.size
Bio.Restriction.Restriction.BtgI.size
Bio.Restriction.Restriction.BtgZI.size
Bio.Restriction.Restriction.BthCI.size
Bio.Restriction.Restriction.BtrI.size
Bio.Restriction.Restriction.BtsCI.size
Bio.Restriction.Restriction.BtsI.size
Bio.Restriction.Restriction.BtsIMutI.size
Bio.Restriction.Restriction.BtuMI.size
Bio.Restriction.Restriction.BveI.size
Bio.Restriction.Restriction.Cac8I.size
Bio.Restriction.Restriction.CaiI.size
Bio.Restriction.Restriction.CauII.size
Bio.Restriction.Restriction.CchII.size
Bio.Restriction.Restriction.CchIII.size
Bio.Restriction.Restriction.CciI.size
Bio.Restriction.Restriction.CciNI.size
Bio.Restriction.Restriction.Cdi630V.size
Bio.Restriction.Restriction.CdiI.size
Bio.Restriction.Restriction.CdpI.size
Bio.Restriction.Restriction.CfoI.size
Bio.Restriction.Restriction.Cfr10I.size
Bio.Restriction.Restriction.Cfr13I.size
Bio.Restriction.Restriction.Cfr42I.size
Bio.Restriction.Restriction.Cfr9I.size
Bio.Restriction.Restriction.CfrI.size
Bio.Restriction.Restriction.Cgl13032I.size
Bio.Restriction.Restriction.Cgl13032II.size
Bio.Restriction.Restriction.ChaI.size
Bio.Restriction.Restriction.CjeFIII.size
Bio.Restriction.Restriction.CjeFV.size
Bio.Restriction.Restriction.CjeI.size
Bio.Restriction.Restriction.CjeNII.size
Bio.Restriction.Restriction.CjeNIII.size
Bio.Restriction.Restriction.CjeP659IV.size
Bio.Restriction.Restriction.CjePI.size
Bio.Restriction.Restriction.CjuI.size
Bio.Restriction.Restriction.CjuII.size
Bio.Restriction.Restriction.ClaI.size
Bio.Restriction.Restriction.CpoI.size
Bio.Restriction.Restriction.CseI.size
Bio.Restriction.Restriction.CsiI.size
Bio.Restriction.Restriction.Csp6I.size
Bio.Restriction.Restriction.CspAI.size
Bio.Restriction.Restriction.CspCI.size
Bio.Restriction.Restriction.CspI.size
Bio.Restriction.Restriction.CstMI.size
Bio.Restriction.Restriction.CviAII.size
Bio.Restriction.Restriction.CviJI.size
Bio.Restriction.Restriction.CviKI_1.size
Bio.Restriction.Restriction.CviQI.size
Bio.Restriction.Restriction.CviRI.size
Bio.Restriction.Restriction.DdeI.size
Bio.Restriction.Restriction.DinI.size
Bio.Restriction.Restriction.DpnI.size
Bio.Restriction.Restriction.DpnII.size
Bio.Restriction.Restriction.DraI.size
Bio.Restriction.Restriction.DraII.size
Bio.Restriction.Restriction.DraIII.size
Bio.Restriction.Restriction.DraRI.size
Bio.Restriction.Restriction.DrdI.size
Bio.Restriction.Restriction.DrdII.size
Bio.Restriction.Restriction.DriI.size
Bio.Restriction.Restriction.DsaI.size
Bio.Restriction.Restriction.DseDI.size
Bio.Restriction.Restriction.EaeI.size
Bio.Restriction.Restriction.EagI.size
Bio.Restriction.Restriction.Eam1104I.size
Bio.Restriction.Restriction.Eam1105I.size
Bio.Restriction.Restriction.EarI.size
Bio.Restriction.Restriction.EciI.size
Bio.Restriction.Restriction.Ecl136II.size
Bio.Restriction.Restriction.EclXI.size
Bio.Restriction.Restriction.Eco105I.size
Bio.Restriction.Restriction.Eco130I.size
Bio.Restriction.Restriction.Eco147I.size
Bio.Restriction.Restriction.Eco24I.size
Bio.Restriction.Restriction.Eco31I.size
Bio.Restriction.Restriction.Eco32I.size
Bio.Restriction.Restriction.Eco47I.size
Bio.Restriction.Restriction.Eco47III.size
Bio.Restriction.Restriction.Eco52I.size
Bio.Restriction.Restriction.Eco53kI.size
Bio.Restriction.Restriction.Eco57I.size
Bio.Restriction.Restriction.Eco57MI.size
Bio.Restriction.Restriction.Eco72I.size
Bio.Restriction.Restriction.Eco81I.size
Bio.Restriction.Restriction.Eco88I.size
Bio.Restriction.Restriction.Eco91I.size
Bio.Restriction.Restriction.EcoHI.size
Bio.Restriction.Restriction.EcoICRI.size
Bio.Restriction.Restriction.EcoNI.size
Bio.Restriction.Restriction.EcoO109I.size
Bio.Restriction.Restriction.EcoO65I.size
Bio.Restriction.Restriction.EcoRI.size
Bio.Restriction.Restriction.EcoRII.size
Bio.Restriction.Restriction.EcoRV.size
Bio.Restriction.Restriction.EcoT14I.size
Bio.Restriction.Restriction.EcoT22I.size
Bio.Restriction.Restriction.EcoT38I.size
Bio.Restriction.Restriction.EgeI.size
Bio.Restriction.Restriction.EheI.size
Bio.Restriction.Restriction.ErhI.size
Bio.Restriction.Restriction.EsaBC3I.size
Bio.Restriction.Restriction.EsaSSI.size
Bio.Restriction.Restriction.Esp3I.size
Bio.Restriction.Restriction.EspI.size
Bio.Restriction.Restriction.FaeI.size
Bio.Restriction.Restriction.FaiI.size
Bio.Restriction.Restriction.FalI.size
Bio.Restriction.Restriction.FaqI.size
Bio.Restriction.Restriction.FatI.size
Bio.Restriction.Restriction.FauI.size
Bio.Restriction.Restriction.FauNDI.size
Bio.Restriction.Restriction.FbaI.size
Bio.Restriction.Restriction.FblI.size
Bio.Restriction.Restriction.FinI.size
Bio.Restriction.Restriction.FmuI.size
Bio.Restriction.Restriction.Fnu4HI.size
Bio.Restriction.Restriction.FnuDII.size
Bio.Restriction.Restriction.FokI.size
Bio.Restriction.Restriction.FriOI.size
Bio.Restriction.Restriction.FseI.size
Bio.Restriction.Restriction.Fsp4HI.size
Bio.Restriction.Restriction.FspAI.size
Bio.Restriction.Restriction.FspBI.size
Bio.Restriction.Restriction.FspEI.size
Bio.Restriction.Restriction.FspI.size
Bio.Restriction.Restriction.GauT27I.size
Bio.Restriction.Restriction.GdiII.size
Bio.Restriction.Restriction.GlaI.size
Bio.Restriction.Restriction.GluI.size
Bio.Restriction.Restriction.GsaI.size
Bio.Restriction.Restriction.GsuI.size
Bio.Restriction.Restriction.HaeI.size
Bio.Restriction.Restriction.HaeII.size
Bio.Restriction.Restriction.HaeIII.size
Bio.Restriction.Restriction.HapII.size
Bio.Restriction.Restriction.HauII.size
Bio.Restriction.Restriction.HgaI.size
Bio.Restriction.Restriction.HgiAI.size
Bio.Restriction.Restriction.HgiCI.size
Bio.Restriction.Restriction.HgiEII.size
Bio.Restriction.Restriction.HgiJII.size
Bio.Restriction.Restriction.HhaI.size
Bio.Restriction.Restriction.Hin1I.size
Bio.Restriction.Restriction.Hin1II.size
Bio.Restriction.Restriction.Hin4I.size
Bio.Restriction.Restriction.Hin4II.size
Bio.Restriction.Restriction.Hin6I.size
Bio.Restriction.Restriction.HinP1I.size
Bio.Restriction.Restriction.HincII.size
Bio.Restriction.Restriction.HindII.size
Bio.Restriction.Restriction.HindIII.size
Bio.Restriction.Restriction.HinfI.size
Bio.Restriction.Restriction.HpaI.size
Bio.Restriction.Restriction.HpaII.size
Bio.Restriction.Restriction.HphI.size
Bio.Restriction.Restriction.Hpy166II.size
Bio.Restriction.Restriction.Hpy178III.size
Bio.Restriction.Restriction.Hpy188I.size
Bio.Restriction.Restriction.Hpy188III.size
Bio.Restriction.Restriction.Hpy8I.size
Bio.Restriction.Restriction.Hpy99I.size
Bio.Restriction.Restriction.Hpy99XIII.size
Bio.Restriction.Restriction.Hpy99XIV.size
Bio.Restriction.Restriction.HpyAV.size
Bio.Restriction.Restriction.HpyCH4III.size
Bio.Restriction.Restriction.HpyCH4IV.size
Bio.Restriction.Restriction.HpyCH4V.size
Bio.Restriction.Restriction.HpyF10VI.size
Bio.Restriction.Restriction.HpyF3I.size
Bio.Restriction.Restriction.HpySE526I.size
Bio.Restriction.Restriction.Hsp92I.size
Bio.Restriction.Restriction.Hsp92II.size
Bio.Restriction.Restriction.HspAI.size
Bio.Restriction.Restriction.Jma19592I.size
Bio.Restriction.Restriction.KasI.size
Bio.Restriction.Restriction.KflI.size
Bio.Restriction.Restriction.Kpn2I.size
Bio.Restriction.Restriction.KpnI.size
Bio.Restriction.Restriction.KroI.size
Bio.Restriction.Restriction.Ksp22I.size
Bio.Restriction.Restriction.Ksp632I.size
Bio.Restriction.Restriction.KspAI.size
Bio.Restriction.Restriction.KspI.size
Bio.Restriction.Restriction.Kzo9I.size
Bio.Restriction.Restriction.LguI.size
Bio.Restriction.Restriction.LpnI.size
Bio.Restriction.Restriction.LpnPI.size
Bio.Restriction.Restriction.Lsp1109I.size
Bio.Restriction.Restriction.LweI.size
Bio.Restriction.Restriction.MabI.size
Bio.Restriction.Restriction.MaeI.size
Bio.Restriction.Restriction.MaeII.size
Bio.Restriction.Restriction.MaeIII.size
Bio.Restriction.Restriction.MalI.size
Bio.Restriction.Restriction.MaqI.size
Bio.Restriction.Restriction.MauBI.size
Bio.Restriction.Restriction.MbiI.size
Bio.Restriction.Restriction.MboI.size
Bio.Restriction.Restriction.MboII.size
Bio.Restriction.Restriction.McaTI.size
Bio.Restriction.Restriction.McrI.size
Bio.Restriction.Restriction.MfeI.size
Bio.Restriction.Restriction.MflI.size
Bio.Restriction.Restriction.MhlI.size
Bio.Restriction.Restriction.MjaIV.size
Bio.Restriction.Restriction.MkaDII.size
Bio.Restriction.Restriction.MlsI.size
Bio.Restriction.Restriction.MluCI.size
Bio.Restriction.Restriction.MluI.size
Bio.Restriction.Restriction.MluNI.size
Bio.Restriction.Restriction.Mly113I.size
Bio.Restriction.Restriction.MlyI.size
Bio.Restriction.Restriction.MmeI.size
Bio.Restriction.Restriction.MnlI.size
Bio.Restriction.Restriction.Mph1103I.size
Bio.Restriction.Restriction.MreI.size
Bio.Restriction.Restriction.MroI.size
Bio.Restriction.Restriction.MroNI.size
Bio.Restriction.Restriction.MroXI.size
Bio.Restriction.Restriction.MscI.size
Bio.Restriction.Restriction.MseI.size
Bio.Restriction.Restriction.MslI.size
Bio.Restriction.Restriction.Msp20I.size
Bio.Restriction.Restriction.MspA1I.size
Bio.Restriction.Restriction.MspCI.size
Bio.Restriction.Restriction.MspI.size
Bio.Restriction.Restriction.MspJI.size
Bio.Restriction.Restriction.MspR9I.size
Bio.Restriction.Restriction.MssI.size
Bio.Restriction.Restriction.MstI.size
Bio.Restriction.Restriction.MunI.size
Bio.Restriction.Restriction.Mva1269I.size
Bio.Restriction.Restriction.MvaI.size
Bio.Restriction.Restriction.MvnI.size
Bio.Restriction.Restriction.MvrI.size
Bio.Restriction.Restriction.MwoI.size
Bio.Restriction.Restriction.NaeI.size
Bio.Restriction.Restriction.NarI.size
Bio.Restriction.Restriction.NciI.size
Bio.Restriction.Restriction.NcoI.size
Bio.Restriction.Restriction.NdeI.size
Bio.Restriction.Restriction.NdeII.size
Bio.Restriction.Restriction.NgoAVIII.size
Bio.Restriction.Restriction.NgoMIV.size
Bio.Restriction.Restriction.NhaXI.size
Bio.Restriction.Restriction.NheI.size
Bio.Restriction.Restriction.NlaCI.size
Bio.Restriction.Restriction.NlaIII.size
Bio.Restriction.Restriction.NlaIV.size
Bio.Restriction.Restriction.Nli3877I.size
Bio.Restriction.Restriction.NmeAIII.size
Bio.Restriction.Restriction.NmeDI.size
Bio.Restriction.Restriction.NmuCI.size
Bio.Restriction.Restriction.NotI.size
Bio.Restriction.Restriction.NruI.size
Bio.Restriction.Restriction.NsbI.size
Bio.Restriction.Restriction.NsiI.size
Bio.Restriction.Restriction.NspBII.size
Bio.Restriction.Restriction.NspI.size
Bio.Restriction.Restriction.NspV.size
Bio.Restriction.Restriction.OliI.size
Bio.Restriction.Restriction.PabI.size
Bio.Restriction.Restriction.PacI.size
Bio.Restriction.Restriction.PaeI.size
Bio.Restriction.Restriction.PaeR7I.size
Bio.Restriction.Restriction.PagI.size
Bio.Restriction.Restriction.PalAI.size
Bio.Restriction.Restriction.PasI.size
Bio.Restriction.Restriction.PauI.size
Bio.Restriction.Restriction.PceI.size
Bio.Restriction.Restriction.PciI.size
Bio.Restriction.Restriction.PciSI.size
Bio.Restriction.Restriction.PcsI.size
Bio.Restriction.Restriction.PctI.size
Bio.Restriction.Restriction.PdiI.size
Bio.Restriction.Restriction.PdmI.size
Bio.Restriction.Restriction.PenI.size
Bio.Restriction.Restriction.PfeI.size
Bio.Restriction.Restriction.Pfl1108I.size
Bio.Restriction.Restriction.Pfl23II.size
Bio.Restriction.Restriction.PflFI.size
Bio.Restriction.Restriction.PflMI.size
Bio.Restriction.Restriction.PfoI.size
Bio.Restriction.Restriction.PinAI.size
Bio.Restriction.Restriction.PlaDI.size
Bio.Restriction.Restriction.Ple19I.size
Bio.Restriction.Restriction.PleI.size
Bio.Restriction.Restriction.PluTI.size
Bio.Restriction.Restriction.PmaCI.size
Bio.Restriction.Restriction.PmeI.size
Bio.Restriction.Restriction.PmlI.size
Bio.Restriction.Restriction.PpiI.size
Bio.Restriction.Restriction.PpsI.size
Bio.Restriction.Restriction.Ppu10I.size
Bio.Restriction.Restriction.Ppu21I.size
Bio.Restriction.Restriction.PpuMI.size
Bio.Restriction.Restriction.PscI.size
Bio.Restriction.Restriction.PshAI.size
Bio.Restriction.Restriction.PshBI.size
Bio.Restriction.Restriction.PsiI.size
Bio.Restriction.Restriction.Psp03I.size
Bio.Restriction.Restriction.Psp124BI.size
Bio.Restriction.Restriction.Psp1406I.size
Bio.Restriction.Restriction.Psp5II.size
Bio.Restriction.Restriction.Psp6I.size
Bio.Restriction.Restriction.PspCI.size
Bio.Restriction.Restriction.PspEI.size
Bio.Restriction.Restriction.PspGI.size
Bio.Restriction.Restriction.PspLI.size
Bio.Restriction.Restriction.PspN4I.size
Bio.Restriction.Restriction.PspOMI.size
Bio.Restriction.Restriction.PspOMII.size
Bio.Restriction.Restriction.PspPI.size
Bio.Restriction.Restriction.PspPPI.size
Bio.Restriction.Restriction.PspPRI.size
Bio.Restriction.Restriction.PspXI.size
Bio.Restriction.Restriction.PsrI.size
Bio.Restriction.Restriction.PssI.size
Bio.Restriction.Restriction.PstI.size
Bio.Restriction.Restriction.PstNI.size
Bio.Restriction.Restriction.PsuI.size
Bio.Restriction.Restriction.PsyI.size
Bio.Restriction.Restriction.PteI.size
Bio.Restriction.Restriction.PvuI.size
Bio.Restriction.Restriction.PvuII.size
Bio.Restriction.Restriction.R2_BceSIV.size
Bio.Restriction.Restriction.RceI.size
Bio.Restriction.Restriction.RdeGBI.size
Bio.Restriction.Restriction.RdeGBII.size
Bio.Restriction.Restriction.RdeGBIII.size
Bio.Restriction.Restriction.RflFIII.size
Bio.Restriction.Restriction.RgaI.size
Bio.Restriction.Restriction.RigI.size
Bio.Restriction.Restriction.RlaI.size
Bio.Restriction.Restriction.RleAI.size
Bio.Restriction.Restriction.RpaB5I.size
Bio.Restriction.Restriction.RpaBI.size
Bio.Restriction.Restriction.RpaI.size
Bio.Restriction.Restriction.RpaTI.size
Bio.Restriction.Restriction.RruI.size
Bio.Restriction.Restriction.RsaI.size
Bio.Restriction.Restriction.RsaNI.size
Bio.Restriction.Restriction.RseI.size
Bio.Restriction.Restriction.Rsr2I.size
Bio.Restriction.Restriction.RsrII.size
Bio.Restriction.Restriction.SacI.size
Bio.Restriction.Restriction.SacII.size
Bio.Restriction.Restriction.SalI.size
Bio.Restriction.Restriction.SanDI.size
Bio.Restriction.Restriction.SapI.size
Bio.Restriction.Restriction.SaqAI.size
Bio.Restriction.Restriction.SatI.size
Bio.Restriction.Restriction.Sau3AI.size
Bio.Restriction.Restriction.Sau96I.size
Bio.Restriction.Restriction.SauI.size
Bio.Restriction.Restriction.SbfI.size
Bio.Restriction.Restriction.ScaI.size
Bio.Restriction.Restriction.SchI.size
Bio.Restriction.Restriction.SciI.size
Bio.Restriction.Restriction.ScrFI.size
Bio.Restriction.Restriction.SdaI.size
Bio.Restriction.Restriction.SdeAI.size
Bio.Restriction.Restriction.SdeOSI.size
Bio.Restriction.Restriction.SduI.size
Bio.Restriction.Restriction.SecI.size
Bio.Restriction.Restriction.SelI.size
Bio.Restriction.Restriction.SetI.size
Bio.Restriction.Restriction.SexAI.size
Bio.Restriction.Restriction.SfaAI.size
Bio.Restriction.Restriction.SfaNI.size
Bio.Restriction.Restriction.SfcI.size
Bio.Restriction.Restriction.SfeI.size
Bio.Restriction.Restriction.SfiI.size
Bio.Restriction.Restriction.SfoI.size
Bio.Restriction.Restriction.Sfr274I.size
Bio.Restriction.Restriction.Sfr303I.size
Bio.Restriction.Restriction.SfuI.size
Bio.Restriction.Restriction.SgeI.size
Bio.Restriction.Restriction.SgfI.size
Bio.Restriction.Restriction.SgrAI.size
Bio.Restriction.Restriction.SgrBI.size
Bio.Restriction.Restriction.SgrDI.size
Bio.Restriction.Restriction.SgrTI.size
Bio.Restriction.Restriction.SgsI.size
Bio.Restriction.Restriction.SimI.size
Bio.Restriction.Restriction.SlaI.size
Bio.Restriction.Restriction.SmaI.size
Bio.Restriction.Restriction.SmiI.size
Bio.Restriction.Restriction.SmiMI.size
Bio.Restriction.Restriction.SmlI.size
Bio.Restriction.Restriction.SmoI.size
Bio.Restriction.Restriction.SnaBI.size
Bio.Restriction.Restriction.SnaI.size
Bio.Restriction.Restriction.Sno506I.size
Bio.Restriction.Restriction.SpeI.size
Bio.Restriction.Restriction.SphI.size
Bio.Restriction.Restriction.SplI.size
Bio.Restriction.Restriction.SpoDI.size
Bio.Restriction.Restriction.SrfI.size
Bio.Restriction.Restriction.Sse232I.size
Bio.Restriction.Restriction.Sse8387I.size
Bio.Restriction.Restriction.Sse8647I.size
Bio.Restriction.Restriction.Sse9I.size
Bio.Restriction.Restriction.SseBI.size
Bio.Restriction.Restriction.SsiI.size
Bio.Restriction.Restriction.SspD5I.size
Bio.Restriction.Restriction.SspDI.size
Bio.Restriction.Restriction.SspI.size
Bio.Restriction.Restriction.SstE37I.size
Bio.Restriction.Restriction.SstI.size
Bio.Restriction.Restriction.Sth132I.size
Bio.Restriction.Restriction.Sth302II.size
Bio.Restriction.Restriction.StrI.size
Bio.Restriction.Restriction.StsI.size
Bio.Restriction.Restriction.StuI.size
Bio.Restriction.Restriction.StyD4I.size
Bio.Restriction.Restriction.StyI.size
Bio.Restriction.Restriction.SwaI.size
Bio.Restriction.Restriction.TaaI.size
Bio.Restriction.Restriction.TaiI.size
Bio.Restriction.Restriction.TaqI.size
Bio.Restriction.Restriction.TaqII.size
Bio.Restriction.Restriction.TasI.size
Bio.Restriction.Restriction.TatI.size
Bio.Restriction.Restriction.TauI.size
Bio.Restriction.Restriction.TfiI.size
Bio.Restriction.Restriction.Tru1I.size
Bio.Restriction.Restriction.Tru9I.size
Bio.Restriction.Restriction.TscAI.size
Bio.Restriction.Restriction.TseFI.size
Bio.Restriction.Restriction.TseI.size
Bio.Restriction.Restriction.TsoI.size
Bio.Restriction.Restriction.Tsp45I.size
Bio.Restriction.Restriction.Tsp4CI.size
Bio.Restriction.Restriction.TspDTI.size
Bio.Restriction.Restriction.TspEI.size
Bio.Restriction.Restriction.TspGWI.size
Bio.Restriction.Restriction.TspMI.size
Bio.Restriction.Restriction.TspRI.size
Bio.Restriction.Restriction.TssI.size
Bio.Restriction.Restriction.TstI.size
Bio.Restriction.Restriction.TsuI.size
Bio.Restriction.Restriction.Tth111I.size
Bio.Restriction.Restriction.Tth111II.size
Bio.Restriction.Restriction.UbaF11I.size
Bio.Restriction.Restriction.UbaF12I.size
Bio.Restriction.Restriction.UbaF13I.size
Bio.Restriction.Restriction.UbaF14I.size
Bio.Restriction.Restriction.UbaF9I.size
Bio.Restriction.Restriction.UbaPI.size
Bio.Restriction.Restriction.UcoMSI.size
Bio.Restriction.Restriction.UnbI.size
Bio.Restriction.Restriction.Van91I.size
Bio.Restriction.Restriction.Vha464I.size
Bio.Restriction.Restriction.VneI.size
Bio.Restriction.Restriction.VpaK11AI.size
Bio.Restriction.Restriction.VpaK11BI.size
Bio.Restriction.Restriction.VspI.size
Bio.Restriction.Restriction.WviI.size
Bio.Restriction.Restriction.XagI.size
Bio.Restriction.Restriction.XapI.size
Bio.Restriction.Restriction.XbaI.size
Bio.Restriction.Restriction.XceI.size
Bio.Restriction.Restriction.XcmI.size
Bio.Restriction.Restriction.XhoI.size
Bio.Restriction.Restriction.XhoII.size
Bio.Restriction.Restriction.XmaI.size
Bio.Restriction.Restriction.XmaIII.size
Bio.Restriction.Restriction.XmaJI.size
Bio.Restriction.Restriction.XmiI.size
Bio.Restriction.Restriction.XmnI.size
Bio.Restriction.Restriction.XspI.size
Bio.Restriction.Restriction.YkrI.size
Bio.Restriction.Restriction.ZraI.size
Bio.Restriction.Restriction.ZrmI.size
Bio.Restriction.Restriction.Zsp2I.size
Bio.codonalign.codonalphabet.CodonAlphabet.size" class="py-name" href="#" onclick="return doclink('link-631', 'size', 'link-631');">size</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1282"></a><tt class="py-lineno">1282</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.residue_type"></a><div id="_RecordConsumer.residue_type-def"><a name="L1283"></a><tt class="py-lineno">1283</tt> <a class="py-toggle" href="#" id="_RecordConsumer.residue_type-toggle" onclick="return toggle('_RecordConsumer.residue_type');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#residue_type">residue_type</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.residue_type-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.residue_type-expanded"><a name="L1284"></a><tt class="py-lineno">1284</tt>  <tt class="py-line">        <tt class="py-comment"># Be lenient about parsing, but technically lowercase residue types are malformed.</tt> </tt>
<a name="L1285"></a><tt class="py-lineno">1285</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">'dna'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">content</tt> <tt class="py-keyword">or</tt> <tt class="py-string">'rna'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">content</tt><tt class="py-op">:</tt> </tt>
<a name="L1286"></a><tt class="py-lineno">1286</tt>  <tt class="py-line">            <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L1287"></a><tt class="py-lineno">1287</tt>  <tt class="py-line">            <tt class="py-keyword">from</tt> <tt id="link-632" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-632', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-633" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-633', 'BiopythonParserWarning', 'link-401');">BiopythonParserWarning</a></tt> </tt>
<a name="L1288"></a><tt class="py-lineno">1288</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid seq_type (%s): DNA/RNA should be uppercase."</tt> <tt class="py-op">%</tt> <tt class="py-name">content</tt><tt class="py-op">,</tt> </tt>
<a name="L1289"></a><tt class="py-lineno">1289</tt>  <tt class="py-line">                          <tt id="link-634" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-634', 'BiopythonParserWarning', 'link-401');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1290"></a><tt class="py-lineno">1290</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-635" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-635', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-636" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.residue_type()=Bio.GenBank._FeatureConsumer-class.html#residue_type,Method Bio.GenBank._RecordConsumer.residue_type()=Bio.GenBank._RecordConsumer-class.html#residue_type"><a title="Bio.GenBank._FeatureConsumer.residue_type
Bio.GenBank._RecordConsumer.residue_type" class="py-name" href="#" onclick="return doclink('link-636', 'residue_type', 'link-636');">residue_type</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1291"></a><tt class="py-lineno">1291</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.data_file_division"></a><div id="_RecordConsumer.data_file_division-def"><a name="L1292"></a><tt class="py-lineno">1292</tt> <a class="py-toggle" href="#" id="_RecordConsumer.data_file_division-toggle" onclick="return toggle('_RecordConsumer.data_file_division');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#data_file_division">data_file_division</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.data_file_division-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.data_file_division-expanded"><a name="L1293"></a><tt class="py-lineno">1293</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-637" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-637', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-638" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.data_file_division()=Bio.GenBank._FeatureConsumer-class.html#data_file_division,Method Bio.GenBank._RecordConsumer.data_file_division()=Bio.GenBank._RecordConsumer-class.html#data_file_division"><a title="Bio.GenBank._FeatureConsumer.data_file_division
Bio.GenBank._RecordConsumer.data_file_division" class="py-name" href="#" onclick="return doclink('link-638', 'data_file_division', 'link-638');">data_file_division</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1294"></a><tt class="py-lineno">1294</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.date"></a><div id="_RecordConsumer.date-def"><a name="L1295"></a><tt class="py-lineno">1295</tt> <a class="py-toggle" href="#" id="_RecordConsumer.date-toggle" onclick="return toggle('_RecordConsumer.date');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#date">date</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.date-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.date-expanded"><a name="L1296"></a><tt class="py-lineno">1296</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-639" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-639', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-640" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.date()=Bio.GenBank._FeatureConsumer-class.html#date,Method Bio.GenBank._RecordConsumer.date()=Bio.GenBank._RecordConsumer-class.html#date,Method Bio.Phylo.PhyloXMLIO.Parser.date()=Bio.Phylo.PhyloXMLIO.Parser-class.html#date,Method Bio.Phylo.PhyloXMLIO.Writer.date()=Bio.Phylo.PhyloXMLIO.Writer-class.html#date"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-640', 'date', 'link-640');">date</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1297"></a><tt class="py-lineno">1297</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.definition"></a><div id="_RecordConsumer.definition-def"><a name="L1298"></a><tt class="py-lineno">1298</tt> <a class="py-toggle" href="#" id="_RecordConsumer.definition-toggle" onclick="return toggle('_RecordConsumer.definition');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#definition">definition</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.definition-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.definition-expanded"><a name="L1299"></a><tt class="py-lineno">1299</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-641" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-641', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-642" class="py-name"><a title="Bio.GenBank._FeatureConsumer.definition
Bio.GenBank._RecordConsumer.definition" class="py-name" href="#" onclick="return doclink('link-642', 'definition', 'link-284');">definition</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1300"></a><tt class="py-lineno">1300</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.accession"></a><div id="_RecordConsumer.accession-def"><a name="L1301"></a><tt class="py-lineno">1301</tt> <a class="py-toggle" href="#" id="_RecordConsumer.accession-toggle" onclick="return toggle('_RecordConsumer.accession');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#accession">accession</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.accession-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.accession-expanded"><a name="L1302"></a><tt class="py-lineno">1302</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-643" class="py-name"><a title="Bio.FSSP.fssp_rec.align.acc" class="py-name" href="#" onclick="return doclink('link-643', 'acc', 'link-289');">acc</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-644" class="py-name"><a title="Bio.GenBank._BaseGenBankConsumer._split_accessions" class="py-name" href="#" onclick="return doclink('link-644', '_split_accessions', 'link-288');">_split_accessions</a></tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1303"></a><tt class="py-lineno">1303</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-645" class="py-name"><a title="Bio.FSSP.fssp_rec.align.acc" class="py-name" href="#" onclick="return doclink('link-645', 'acc', 'link-289');">acc</a></tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-646" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-646', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-647" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-647', 'accession', 'link-238');">accession</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1304"></a><tt class="py-lineno">1304</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-648" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-648', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-649" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-649', 'accession', 'link-238');">accession</a></tt><tt class="py-op">.</tt><tt id="link-650" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-650', 'append', 'link-213');">append</a></tt><tt class="py-op">(</tt><tt id="link-651" class="py-name"><a title="Bio.FSSP.fssp_rec.align.acc" class="py-name" href="#" onclick="return doclink('link-651', 'acc', 'link-289');">acc</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1305"></a><tt class="py-lineno">1305</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.nid"></a><div id="_RecordConsumer.nid-def"><a name="L1306"></a><tt class="py-lineno">1306</tt> <a class="py-toggle" href="#" id="_RecordConsumer.nid-toggle" onclick="return toggle('_RecordConsumer.nid');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#nid">nid</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.nid-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.nid-expanded"><a name="L1307"></a><tt class="py-lineno">1307</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-652" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-652', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-653" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.nid()=Bio.GenBank._FeatureConsumer-class.html#nid,Method Bio.GenBank._RecordConsumer.nid()=Bio.GenBank._RecordConsumer-class.html#nid"><a title="Bio.GenBank._FeatureConsumer.nid
Bio.GenBank._RecordConsumer.nid" class="py-name" href="#" onclick="return doclink('link-653', 'nid', 'link-653');">nid</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1308"></a><tt class="py-lineno">1308</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.pid"></a><div id="_RecordConsumer.pid-def"><a name="L1309"></a><tt class="py-lineno">1309</tt> <a class="py-toggle" href="#" id="_RecordConsumer.pid-toggle" onclick="return toggle('_RecordConsumer.pid');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#pid">pid</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.pid-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.pid-expanded"><a name="L1310"></a><tt class="py-lineno">1310</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-654" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-654', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-655" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.pid()=Bio.GenBank._FeatureConsumer-class.html#pid,Method Bio.GenBank._RecordConsumer.pid()=Bio.GenBank._RecordConsumer-class.html#pid"><a title="Bio.GenBank._FeatureConsumer.pid
Bio.GenBank._RecordConsumer.pid" class="py-name" href="#" onclick="return doclink('link-655', 'pid', 'link-655');">pid</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1311"></a><tt class="py-lineno">1311</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.version"></a><div id="_RecordConsumer.version-def"><a name="L1312"></a><tt class="py-lineno">1312</tt> <a class="py-toggle" href="#" id="_RecordConsumer.version-toggle" onclick="return toggle('_RecordConsumer.version');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#version">version</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.version-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.version-expanded"><a name="L1313"></a><tt class="py-lineno">1313</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-656" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-656', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-657" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-657', 'version', 'link-336');">version</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1314"></a><tt class="py-lineno">1314</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.db_source"></a><div id="_RecordConsumer.db_source-def"><a name="L1315"></a><tt class="py-lineno">1315</tt> <a class="py-toggle" href="#" id="_RecordConsumer.db_source-toggle" onclick="return toggle('_RecordConsumer.db_source');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#db_source">db_source</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.db_source-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.db_source-expanded"><a name="L1316"></a><tt class="py-lineno">1316</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-658" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-658', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-659" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.db_source()=Bio.GenBank._FeatureConsumer-class.html#db_source,Method Bio.GenBank._RecordConsumer.db_source()=Bio.GenBank._RecordConsumer-class.html#db_source"><a title="Bio.GenBank._FeatureConsumer.db_source
Bio.GenBank._RecordConsumer.db_source" class="py-name" href="#" onclick="return doclink('link-659', 'db_source', 'link-659');">db_source</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt><tt class="py-op">.</tt><tt id="link-660" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-660', 'rstrip', 'link-214');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1317"></a><tt class="py-lineno">1317</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.gi"></a><div id="_RecordConsumer.gi-def"><a name="L1318"></a><tt class="py-lineno">1318</tt> <a class="py-toggle" href="#" id="_RecordConsumer.gi-toggle" onclick="return toggle('_RecordConsumer.gi');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#gi">gi</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.gi-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.gi-expanded"><a name="L1319"></a><tt class="py-lineno">1319</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-661" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-661', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-662" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.gi()=Bio.GenBank._FeatureConsumer-class.html#gi,Method Bio.GenBank._RecordConsumer.gi()=Bio.GenBank._RecordConsumer-class.html#gi"><a title="Bio.GenBank._FeatureConsumer.gi
Bio.GenBank._RecordConsumer.gi" class="py-name" href="#" onclick="return doclink('link-662', 'gi', 'link-662');">gi</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1320"></a><tt class="py-lineno">1320</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.keywords"></a><div id="_RecordConsumer.keywords-def"><a name="L1321"></a><tt class="py-lineno">1321</tt> <a class="py-toggle" href="#" id="_RecordConsumer.keywords-toggle" onclick="return toggle('_RecordConsumer.keywords');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#keywords">keywords</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.keywords-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.keywords-expanded"><a name="L1322"></a><tt class="py-lineno">1322</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-663" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-663', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-664" class="py-name"><a title="Bio.GenBank._FeatureConsumer.keywords
Bio.GenBank._RecordConsumer.keywords" class="py-name" href="#" onclick="return doclink('link-664', 'keywords', 'link-230');">keywords</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-665" class="py-name"><a title="Bio.GenBank._BaseGenBankConsumer._split_keywords" class="py-name" href="#" onclick="return doclink('link-665', '_split_keywords', 'link-350');">_split_keywords</a></tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1323"></a><tt class="py-lineno">1323</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.project"></a><div id="_RecordConsumer.project-def"><a name="L1324"></a><tt class="py-lineno">1324</tt> <a class="py-toggle" href="#" id="_RecordConsumer.project-toggle" onclick="return toggle('_RecordConsumer.project');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#project">project</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.project-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.project-expanded"><a name="L1325"></a><tt class="py-lineno">1325</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-666" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-666', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">projects</tt><tt class="py-op">.</tt><tt id="link-667" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-667', 'extend', 'link-248');">extend</a></tt><tt class="py-op">(</tt><tt class="py-name">p</tt> <tt class="py-keyword">for</tt> <tt class="py-name">p</tt> <tt class="py-keyword">in</tt> <tt class="py-name">content</tt><tt class="py-op">.</tt><tt id="link-668" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-668', 'split', 'link-158');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt class="py-name">p</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1326"></a><tt class="py-lineno">1326</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.dblink"></a><div id="_RecordConsumer.dblink-def"><a name="L1327"></a><tt class="py-lineno">1327</tt> <a class="py-toggle" href="#" id="_RecordConsumer.dblink-toggle" onclick="return toggle('_RecordConsumer.dblink');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#dblink">dblink</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.dblink-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.dblink-expanded"><a name="L1328"></a><tt class="py-lineno">1328</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-669" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-669', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">dblinks</tt><tt class="py-op">.</tt><tt id="link-670" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-670', 'append', 'link-213');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1329"></a><tt class="py-lineno">1329</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.segment"></a><div id="_RecordConsumer.segment-def"><a name="L1330"></a><tt class="py-lineno">1330</tt> <a class="py-toggle" href="#" id="_RecordConsumer.segment-toggle" onclick="return toggle('_RecordConsumer.segment');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#segment">segment</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.segment-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.segment-expanded"><a name="L1331"></a><tt class="py-lineno">1331</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-671" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-671', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-672" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.segment()=Bio.GenBank._FeatureConsumer-class.html#segment,Method Bio.GenBank._RecordConsumer.segment()=Bio.GenBank._RecordConsumer-class.html#segment"><a title="Bio.GenBank._FeatureConsumer.segment
Bio.GenBank._RecordConsumer.segment" class="py-name" href="#" onclick="return doclink('link-672', 'segment', 'link-672');">segment</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1332"></a><tt class="py-lineno">1332</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.source"></a><div id="_RecordConsumer.source-def"><a name="L1333"></a><tt class="py-lineno">1333</tt> <a class="py-toggle" href="#" id="_RecordConsumer.source-toggle" onclick="return toggle('_RecordConsumer.source');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#source">source</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.source-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.source-expanded"><a name="L1334"></a><tt class="py-lineno">1334</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-673" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-673', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-674" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.source()=Bio.GenBank._FeatureConsumer-class.html#source,Method Bio.GenBank._RecordConsumer.source()=Bio.GenBank._RecordConsumer-class.html#source,Method Bio.Pathway.Network.source()=Bio.Pathway.Network-class.html#source"><a title="Bio.GenBank._FeatureConsumer.source
Bio.GenBank._RecordConsumer.source
Bio.Pathway.Network.source" class="py-name" href="#" onclick="return doclink('link-674', 'source', 'link-674');">source</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1335"></a><tt class="py-lineno">1335</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.organism"></a><div id="_RecordConsumer.organism-def"><a name="L1336"></a><tt class="py-lineno">1336</tt> <a class="py-toggle" href="#" id="_RecordConsumer.organism-toggle" onclick="return toggle('_RecordConsumer.organism');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#organism">organism</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.organism-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.organism-expanded"><a name="L1337"></a><tt class="py-lineno">1337</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-675" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-675', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-676" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.organism()=Bio.GenBank._FeatureConsumer-class.html#organism,Method Bio.GenBank._RecordConsumer.organism()=Bio.GenBank._RecordConsumer-class.html#organism"><a title="Bio.GenBank._FeatureConsumer.organism
Bio.GenBank._RecordConsumer.organism" class="py-name" href="#" onclick="return doclink('link-676', 'organism', 'link-676');">organism</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1338"></a><tt class="py-lineno">1338</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.taxonomy"></a><div id="_RecordConsumer.taxonomy-def"><a name="L1339"></a><tt class="py-lineno">1339</tt> <a class="py-toggle" href="#" id="_RecordConsumer.taxonomy-toggle" onclick="return toggle('_RecordConsumer.taxonomy');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#taxonomy">taxonomy</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.taxonomy-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.taxonomy-expanded"><a name="L1340"></a><tt class="py-lineno">1340</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-677" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-677', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-678" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.taxonomy()=Bio.GenBank._FeatureConsumer-class.html#taxonomy,Method Bio.GenBank._RecordConsumer.taxonomy()=Bio.GenBank._RecordConsumer-class.html#taxonomy,Variable Bio.Phylo.PhyloXML.Clade.taxonomy=Bio.Phylo.PhyloXML.Clade-class.html#taxonomy,Method Bio.Phylo.PhyloXMLIO.Writer.taxonomy()=Bio.Phylo.PhyloXMLIO.Writer-class.html#taxonomy"><a title="Bio.GenBank._FeatureConsumer.taxonomy
Bio.GenBank._RecordConsumer.taxonomy
Bio.Phylo.PhyloXML.Clade.taxonomy
Bio.Phylo.PhyloXMLIO.Writer.taxonomy" class="py-name" href="#" onclick="return doclink('link-678', 'taxonomy', 'link-678');">taxonomy</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-679" class="py-name"><a title="Bio.GenBank._BaseGenBankConsumer._split_taxonomy" class="py-name" href="#" onclick="return doclink('link-679', '_split_taxonomy', 'link-360');">_split_taxonomy</a></tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1341"></a><tt class="py-lineno">1341</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.reference_num"></a><div id="_RecordConsumer.reference_num-def"><a name="L1342"></a><tt class="py-lineno">1342</tt> <a class="py-toggle" href="#" id="_RecordConsumer.reference_num-toggle" onclick="return toggle('_RecordConsumer.reference_num');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#reference_num">reference_num</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.reference_num-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.reference_num-expanded"><a name="L1343"></a><tt class="py-lineno">1343</tt>  <tt class="py-line">        <tt class="py-docstring">"""Grab the reference number and signal the start of a new reference.</tt> </tt>
<a name="L1344"></a><tt class="py-lineno">1344</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1345"></a><tt class="py-lineno">1345</tt>  <tt class="py-line">        <tt class="py-comment"># check if we have a reference to add</tt> </tt>
<a name="L1346"></a><tt class="py-lineno">1346</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1347"></a><tt class="py-lineno">1347</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-680" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-680', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">references</tt><tt class="py-op">.</tt><tt id="link-681" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-681', 'append', 'link-213');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">)</tt> </tt>
<a name="L1348"></a><tt class="py-lineno">1348</tt>  <tt class="py-line"> </tt>
<a name="L1349"></a><tt class="py-lineno">1349</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt class="py-op">.</tt> <tt class="py-keyword">import</tt> <tt id="link-682" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-682', 'Record', 'link-618');">Record</a></tt> </tt>
<a name="L1350"></a><tt class="py-lineno">1350</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt> <tt class="py-op">=</tt> <tt id="link-683" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-683', 'Record', 'link-618');">Record</a></tt><tt class="py-op">.</tt><tt id="link-684" class="py-name"><a title="Bio.ExPASy.Prodoc.Reference
Bio.GenBank.Record.Reference
Bio.Phylo.PhyloXML.Reference
Bio.SeqFeature.Reference
Bio.SwissProt.Reference" class="py-name" href="#" onclick="return doclink('link-684', 'Reference', 'link-372');">Reference</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1351"></a><tt class="py-lineno">1351</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt id="link-685" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Pathway.number=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#number"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway.number" class="py-name" href="#" onclick="return doclink('link-685', 'number', 'link-685');">number</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1352"></a><tt class="py-lineno">1352</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.reference_bases"></a><div id="_RecordConsumer.reference_bases-def"><a name="L1353"></a><tt class="py-lineno">1353</tt> <a class="py-toggle" href="#" id="_RecordConsumer.reference_bases-toggle" onclick="return toggle('_RecordConsumer.reference_bases');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#reference_bases">reference_bases</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.reference_bases-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.reference_bases-expanded"><a name="L1354"></a><tt class="py-lineno">1354</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt class="py-name">bases</tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1355"></a><tt class="py-lineno">1355</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.authors"></a><div id="_RecordConsumer.authors-def"><a name="L1356"></a><tt class="py-lineno">1356</tt> <a class="py-toggle" href="#" id="_RecordConsumer.authors-toggle" onclick="return toggle('_RecordConsumer.authors');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#authors">authors</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.authors-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.authors-expanded"><a name="L1357"></a><tt class="py-lineno">1357</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt id="link-686" class="py-name"><a title="Bio.GenBank._FeatureConsumer.authors
Bio.GenBank._RecordConsumer.authors" class="py-name" href="#" onclick="return doclink('link-686', 'authors', 'link-394');">authors</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1358"></a><tt class="py-lineno">1358</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.consrtm"></a><div id="_RecordConsumer.consrtm-def"><a name="L1359"></a><tt class="py-lineno">1359</tt> <a class="py-toggle" href="#" id="_RecordConsumer.consrtm-toggle" onclick="return toggle('_RecordConsumer.consrtm');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#consrtm">consrtm</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.consrtm-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.consrtm-expanded"><a name="L1360"></a><tt class="py-lineno">1360</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt id="link-687" class="py-name"><a title="Bio.GenBank._FeatureConsumer.consrtm
Bio.GenBank._RecordConsumer.consrtm" class="py-name" href="#" onclick="return doclink('link-687', 'consrtm', 'link-397');">consrtm</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1361"></a><tt class="py-lineno">1361</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.title"></a><div id="_RecordConsumer.title-def"><a name="L1362"></a><tt class="py-lineno">1362</tt> <a class="py-toggle" href="#" id="_RecordConsumer.title-toggle" onclick="return toggle('_RecordConsumer.title');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#title">title</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.title-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.title-expanded"><a name="L1363"></a><tt class="py-lineno">1363</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1364"></a><tt class="py-lineno">1364</tt>  <tt class="py-line">            <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L1365"></a><tt class="py-lineno">1365</tt>  <tt class="py-line">            <tt class="py-keyword">from</tt> <tt id="link-688" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-688', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-689" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-689', 'BiopythonParserWarning', 'link-401');">BiopythonParserWarning</a></tt> </tt>
<a name="L1366"></a><tt class="py-lineno">1366</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"GenBank TITLE line without REFERENCE line."</tt><tt class="py-op">,</tt> </tt>
<a name="L1367"></a><tt class="py-lineno">1367</tt>  <tt class="py-line">                          <tt id="link-690" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-690', 'BiopythonParserWarning', 'link-401');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1368"></a><tt class="py-lineno">1368</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L1369"></a><tt class="py-lineno">1369</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt id="link-691" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-691', 'title', 'link-403');">title</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1370"></a><tt class="py-lineno">1370</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.journal"></a><div id="_RecordConsumer.journal-def"><a name="L1371"></a><tt class="py-lineno">1371</tt> <a class="py-toggle" href="#" id="_RecordConsumer.journal-toggle" onclick="return toggle('_RecordConsumer.journal');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#journal">journal</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.journal-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.journal-expanded"><a name="L1372"></a><tt class="py-lineno">1372</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt id="link-692" class="py-name"><a title="Bio.GenBank._FeatureConsumer.journal
Bio.GenBank._RecordConsumer.journal" class="py-name" href="#" onclick="return doclink('link-692', 'journal', 'link-406');">journal</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1373"></a><tt class="py-lineno">1373</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.medline_id"></a><div id="_RecordConsumer.medline_id-def"><a name="L1374"></a><tt class="py-lineno">1374</tt> <a class="py-toggle" href="#" id="_RecordConsumer.medline_id-toggle" onclick="return toggle('_RecordConsumer.medline_id');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#medline_id">medline_id</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.medline_id-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.medline_id-expanded"><a name="L1375"></a><tt class="py-lineno">1375</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt id="link-693" class="py-name"><a title="Bio.GenBank._FeatureConsumer.medline_id
Bio.GenBank._RecordConsumer.medline_id" class="py-name" href="#" onclick="return doclink('link-693', 'medline_id', 'link-409');">medline_id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1376"></a><tt class="py-lineno">1376</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.pubmed_id"></a><div id="_RecordConsumer.pubmed_id-def"><a name="L1377"></a><tt class="py-lineno">1377</tt> <a class="py-toggle" href="#" id="_RecordConsumer.pubmed_id-toggle" onclick="return toggle('_RecordConsumer.pubmed_id');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#pubmed_id">pubmed_id</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.pubmed_id-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.pubmed_id-expanded"><a name="L1378"></a><tt class="py-lineno">1378</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt id="link-694" class="py-name"><a title="Bio.GenBank._FeatureConsumer.pubmed_id
Bio.GenBank._RecordConsumer.pubmed_id" class="py-name" href="#" onclick="return doclink('link-694', 'pubmed_id', 'link-410');">pubmed_id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1379"></a><tt class="py-lineno">1379</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.remark"></a><div id="_RecordConsumer.remark-def"><a name="L1380"></a><tt class="py-lineno">1380</tt> <a class="py-toggle" href="#" id="_RecordConsumer.remark-toggle" onclick="return toggle('_RecordConsumer.remark');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#remark">remark</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.remark-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.remark-expanded"><a name="L1381"></a><tt class="py-lineno">1381</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">.</tt><tt id="link-695" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.remark()=Bio.GenBank._FeatureConsumer-class.html#remark,Method Bio.GenBank._RecordConsumer.remark()=Bio.GenBank._RecordConsumer-class.html#remark"><a title="Bio.GenBank._FeatureConsumer.remark
Bio.GenBank._RecordConsumer.remark" class="py-name" href="#" onclick="return doclink('link-695', 'remark', 'link-695');">remark</a></tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1382"></a><tt class="py-lineno">1382</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.comment"></a><div id="_RecordConsumer.comment-def"><a name="L1383"></a><tt class="py-lineno">1383</tt> <a class="py-toggle" href="#" id="_RecordConsumer.comment-toggle" onclick="return toggle('_RecordConsumer.comment');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#comment">comment</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.comment-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.comment-expanded"><a name="L1384"></a><tt class="py-lineno">1384</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-696" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-696', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-697" class="py-name"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-697', 'comment', 'link-411');">comment</a></tt> <tt class="py-op">+=</tt> <tt class="py-string">"\n"</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1385"></a><tt class="py-lineno">1385</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.primary_ref_line"></a><div id="_RecordConsumer.primary_ref_line-def"><a name="L1386"></a><tt class="py-lineno">1386</tt> <a class="py-toggle" href="#" id="_RecordConsumer.primary_ref_line-toggle" onclick="return toggle('_RecordConsumer.primary_ref_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#primary_ref_line">primary_ref_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.primary_ref_line-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.primary_ref_line-expanded"><a name="L1387"></a><tt class="py-lineno">1387</tt>  <tt class="py-line">        <tt class="py-docstring">"""Data for the PRIMARY line"""</tt> </tt>
<a name="L1388"></a><tt class="py-lineno">1388</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-698" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-698', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-699" class="py-name" targets="Method Bio.GenBank._RecordConsumer.primary()=Bio.GenBank._RecordConsumer-class.html#primary"><a title="Bio.GenBank._RecordConsumer.primary" class="py-name" href="#" onclick="return doclink('link-699', 'primary', 'link-699');">primary</a></tt><tt class="py-op">.</tt><tt id="link-700" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-700', 'append', 'link-213');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1389"></a><tt class="py-lineno">1389</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.primary"></a><div id="_RecordConsumer.primary-def"><a name="L1390"></a><tt class="py-lineno">1390</tt> <a class="py-toggle" href="#" id="_RecordConsumer.primary-toggle" onclick="return toggle('_RecordConsumer.primary');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#primary">primary</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.primary-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.primary-expanded"><a name="L1391"></a><tt class="py-lineno">1391</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L1392"></a><tt class="py-lineno">1392</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.features_line"></a><div id="_RecordConsumer.features_line-def"><a name="L1393"></a><tt class="py-lineno">1393</tt> <a class="py-toggle" href="#" id="_RecordConsumer.features_line-toggle" onclick="return toggle('_RecordConsumer.features_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#features_line">features_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.features_line-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.features_line-expanded"><a name="L1394"></a><tt class="py-lineno">1394</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get ready for the feature table when we reach the FEATURE line.</tt> </tt>
<a name="L1395"></a><tt class="py-lineno">1395</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1396"></a><tt class="py-lineno">1396</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-701" class="py-name"><a title="Bio.GenBank._FeatureConsumer.start_feature_table
Bio.GenBank._RecordConsumer.start_feature_table" class="py-name" href="#" onclick="return doclink('link-701', 'start_feature_table', 'link-418');">start_feature_table</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1397"></a><tt class="py-lineno">1397</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.start_feature_table"></a><div id="_RecordConsumer.start_feature_table-def"><a name="L1398"></a><tt class="py-lineno">1398</tt> <a class="py-toggle" href="#" id="_RecordConsumer.start_feature_table-toggle" onclick="return toggle('_RecordConsumer.start_feature_table');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#start_feature_table">start_feature_table</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.start_feature_table-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.start_feature_table-expanded"><a name="L1399"></a><tt class="py-lineno">1399</tt>  <tt class="py-line">        <tt class="py-docstring">"""Signal the start of the feature table.</tt> </tt>
<a name="L1400"></a><tt class="py-lineno">1400</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1401"></a><tt class="py-lineno">1401</tt>  <tt class="py-line">        <tt class="py-comment"># we need to add on the last reference</tt> </tt>
<a name="L1402"></a><tt class="py-lineno">1402</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1403"></a><tt class="py-lineno">1403</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-702" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-702', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">references</tt><tt class="py-op">.</tt><tt id="link-703" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-703', 'append', 'link-213');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_reference</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1404"></a><tt class="py-lineno">1404</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.feature_key"></a><div id="_RecordConsumer.feature_key-def"><a name="L1405"></a><tt class="py-lineno">1405</tt> <a class="py-toggle" href="#" id="_RecordConsumer.feature_key-toggle" onclick="return toggle('_RecordConsumer.feature_key');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#feature_key">feature_key</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.feature_key-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.feature_key-expanded"><a name="L1406"></a><tt class="py-lineno">1406</tt>  <tt class="py-line">        <tt class="py-docstring">"""Grab the key of the feature and signal the start of a new feature.</tt> </tt>
<a name="L1407"></a><tt class="py-lineno">1407</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1408"></a><tt class="py-lineno">1408</tt>  <tt class="py-line">        <tt class="py-comment"># first add on feature information if we've got any</tt> </tt>
<a name="L1409"></a><tt class="py-lineno">1409</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-704" class="py-name" targets="Method Bio.GenBank._RecordConsumer._add_feature()=Bio.GenBank._RecordConsumer-class.html#_add_feature"><a title="Bio.GenBank._RecordConsumer._add_feature" class="py-name" href="#" onclick="return doclink('link-704', '_add_feature', 'link-704');">_add_feature</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1410"></a><tt class="py-lineno">1410</tt>  <tt class="py-line"> </tt>
<a name="L1411"></a><tt class="py-lineno">1411</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt class="py-op">.</tt> <tt class="py-keyword">import</tt> <tt id="link-705" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-705', 'Record', 'link-618');">Record</a></tt> </tt>
<a name="L1412"></a><tt class="py-lineno">1412</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_feature</tt> <tt class="py-op">=</tt> <tt id="link-706" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-706', 'Record', 'link-618');">Record</a></tt><tt class="py-op">.</tt><tt id="link-707" class="py-name" targets="Class Bio.GenBank.Record.Feature=Bio.GenBank.Record.Feature-class.html,Class Bio.Graphics.GenomeDiagram._Feature.Feature=Bio.Graphics.GenomeDiagram._Feature.Feature-class.html"><a title="Bio.GenBank.Record.Feature
Bio.Graphics.GenomeDiagram._Feature.Feature" class="py-name" href="#" onclick="return doclink('link-707', 'Feature', 'link-707');">Feature</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1413"></a><tt class="py-lineno">1413</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_feature</tt><tt class="py-op">.</tt><tt class="py-name">key</tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1414"></a><tt class="py-lineno">1414</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer._add_feature"></a><div id="_RecordConsumer._add_feature-def"><a name="L1415"></a><tt class="py-lineno">1415</tt> <a class="py-toggle" href="#" id="_RecordConsumer._add_feature-toggle" onclick="return toggle('_RecordConsumer._add_feature');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#_add_feature">_add_feature</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer._add_feature-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer._add_feature-expanded"><a name="L1416"></a><tt class="py-lineno">1416</tt>  <tt class="py-line">        <tt class="py-docstring">"""Utility function to add a feature to the Record.</tt> </tt>
<a name="L1417"></a><tt class="py-lineno">1417</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1418"></a><tt class="py-lineno">1418</tt>  <tt class="py-line"><tt class="py-docstring">        This does all of the appropriate checking to make sure we haven't</tt> </tt>
<a name="L1419"></a><tt class="py-lineno">1419</tt>  <tt class="py-line"><tt class="py-docstring">        left any info behind, and that we are only adding info if it</tt> </tt>
<a name="L1420"></a><tt class="py-lineno">1420</tt>  <tt class="py-line"><tt class="py-docstring">        exists.</tt> </tt>
<a name="L1421"></a><tt class="py-lineno">1421</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1422"></a><tt class="py-lineno">1422</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_feature</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1423"></a><tt class="py-lineno">1423</tt>  <tt class="py-line">            <tt class="py-comment"># if we have a left over qualifier, add it to the qualifiers</tt> </tt>
<a name="L1424"></a><tt class="py-lineno">1424</tt>  <tt class="py-line">            <tt class="py-comment"># on the current feature</tt> </tt>
<a name="L1425"></a><tt class="py-lineno">1425</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_qualifier</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1426"></a><tt class="py-lineno">1426</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_feature</tt><tt class="py-op">.</tt><tt class="py-name">qualifiers</tt><tt class="py-op">.</tt><tt id="link-708" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-708', 'append', 'link-213');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_qualifier</tt><tt class="py-op">)</tt> </tt>
<a name="L1427"></a><tt class="py-lineno">1427</tt>  <tt class="py-line"> </tt>
<a name="L1428"></a><tt class="py-lineno">1428</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_qualifier</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L1429"></a><tt class="py-lineno">1429</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-709" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-709', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-710" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-710', 'features', 'link-426');">features</a></tt><tt class="py-op">.</tt><tt id="link-711" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-711', 'append', 'link-213');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_feature</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1430"></a><tt class="py-lineno">1430</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.location"></a><div id="_RecordConsumer.location-def"><a name="L1431"></a><tt class="py-lineno">1431</tt> <a class="py-toggle" href="#" id="_RecordConsumer.location-toggle" onclick="return toggle('_RecordConsumer.location');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#location">location</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.location-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.location-expanded"><a name="L1432"></a><tt class="py-lineno">1432</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_feature</tt><tt class="py-op">.</tt><tt id="link-712" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-712', 'location', 'link-381');">location</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-713" class="py-name"><a title="Bio.GenBank._BaseGenBankConsumer._clean_location" class="py-name" href="#" onclick="return doclink('link-713', '_clean_location', 'link-428');">_clean_location</a></tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1433"></a><tt class="py-lineno">1433</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.feature_qualifier"></a><div id="_RecordConsumer.feature_qualifier-def"><a name="L1434"></a><tt class="py-lineno">1434</tt> <a class="py-toggle" href="#" id="_RecordConsumer.feature_qualifier-toggle" onclick="return toggle('_RecordConsumer.feature_qualifier');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#feature_qualifier">feature_qualifier</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">key</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.feature_qualifier-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.feature_qualifier-expanded"><a name="L1435"></a><tt class="py-lineno">1435</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-714" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.feature_qualifier_name()=Bio.GenBank._FeatureConsumer-class.html#feature_qualifier_name,Method Bio.GenBank._RecordConsumer.feature_qualifier_name()=Bio.GenBank._RecordConsumer-class.html#feature_qualifier_name"><a title="Bio.GenBank._FeatureConsumer.feature_qualifier_name
Bio.GenBank._RecordConsumer.feature_qualifier_name" class="py-name" href="#" onclick="return doclink('link-714', 'feature_qualifier_name', 'link-714');">feature_qualifier_name</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1436"></a><tt class="py-lineno">1436</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-715" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-715', 'value', 'link-544');">value</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1437"></a><tt class="py-lineno">1437</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-716" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.feature_qualifier_description()=Bio.GenBank._FeatureConsumer-class.html#feature_qualifier_description,Method Bio.GenBank._RecordConsumer.feature_qualifier_description()=Bio.GenBank._RecordConsumer-class.html#feature_qualifier_description"><a title="Bio.GenBank._FeatureConsumer.feature_qualifier_description
Bio.GenBank._RecordConsumer.feature_qualifier_description" class="py-name" href="#" onclick="return doclink('link-716', 'feature_qualifier_description', 'link-716');">feature_qualifier_description</a></tt><tt class="py-op">(</tt><tt id="link-717" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-717', 'value', 'link-544');">value</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1438"></a><tt class="py-lineno">1438</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.feature_qualifier_name"></a><div id="_RecordConsumer.feature_qualifier_name-def"><a name="L1439"></a><tt class="py-lineno">1439</tt> <a class="py-toggle" href="#" id="_RecordConsumer.feature_qualifier_name-toggle" onclick="return toggle('_RecordConsumer.feature_qualifier_name');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#feature_qualifier_name">feature_qualifier_name</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content_list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.feature_qualifier_name-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.feature_qualifier_name-expanded"><a name="L1440"></a><tt class="py-lineno">1440</tt>  <tt class="py-line">        <tt class="py-docstring">"""Deal with qualifier names</tt> </tt>
<a name="L1441"></a><tt class="py-lineno">1441</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1442"></a><tt class="py-lineno">1442</tt>  <tt class="py-line"><tt class="py-docstring">        We receive a list of keys, since you can have valueless keys such as</tt> </tt>
<a name="L1443"></a><tt class="py-lineno">1443</tt>  <tt class="py-line"><tt class="py-docstring">        /pseudo which would be passed in with the next key (since no other</tt> </tt>
<a name="L1444"></a><tt class="py-lineno">1444</tt>  <tt class="py-line"><tt class="py-docstring">        tags separate them in the file)</tt> </tt>
<a name="L1445"></a><tt class="py-lineno">1445</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1446"></a><tt class="py-lineno">1446</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt class="py-op">.</tt> <tt class="py-keyword">import</tt> <tt id="link-718" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-718', 'Record', 'link-618');">Record</a></tt> </tt>
<a name="L1447"></a><tt class="py-lineno">1447</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">content</tt> <tt class="py-keyword">in</tt> <tt class="py-name">content_list</tt><tt class="py-op">:</tt> </tt>
<a name="L1448"></a><tt class="py-lineno">1448</tt>  <tt class="py-line">            <tt class="py-comment"># the record parser keeps the /s -- add them if we don't have 'em</tt> </tt>
<a name="L1449"></a><tt class="py-lineno">1449</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">content</tt><tt class="py-op">.</tt><tt id="link-719" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-719', 'startswith', 'link-150');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"/"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1450"></a><tt class="py-lineno">1450</tt>  <tt class="py-line">                <tt class="py-name">content</tt> <tt class="py-op">=</tt> <tt class="py-string">"/%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">content</tt> </tt>
<a name="L1451"></a><tt class="py-lineno">1451</tt>  <tt class="py-line">            <tt class="py-comment"># add on a qualifier if we've got one</tt> </tt>
<a name="L1452"></a><tt class="py-lineno">1452</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_qualifier</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1453"></a><tt class="py-lineno">1453</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_feature</tt><tt class="py-op">.</tt><tt class="py-name">qualifiers</tt><tt class="py-op">.</tt><tt id="link-720" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-720', 'append', 'link-213');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_qualifier</tt><tt class="py-op">)</tt> </tt>
<a name="L1454"></a><tt class="py-lineno">1454</tt>  <tt class="py-line"> </tt>
<a name="L1455"></a><tt class="py-lineno">1455</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_qualifier</tt> <tt class="py-op">=</tt> <tt id="link-721" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-721', 'Record', 'link-618');">Record</a></tt><tt class="py-op">.</tt><tt id="link-722" class="py-name" targets="Class Bio.GenBank.Record.Qualifier=Bio.GenBank.Record.Qualifier-class.html"><a title="Bio.GenBank.Record.Qualifier" class="py-name" href="#" onclick="return doclink('link-722', 'Qualifier', 'link-722');">Qualifier</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1456"></a><tt class="py-lineno">1456</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_qualifier</tt><tt class="py-op">.</tt><tt class="py-name">key</tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1457"></a><tt class="py-lineno">1457</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.feature_qualifier_description"></a><div id="_RecordConsumer.feature_qualifier_description-def"><a name="L1458"></a><tt class="py-lineno">1458</tt> <a class="py-toggle" href="#" id="_RecordConsumer.feature_qualifier_description-toggle" onclick="return toggle('_RecordConsumer.feature_qualifier_description');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#feature_qualifier_description">feature_qualifier_description</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.feature_qualifier_description-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.feature_qualifier_description-expanded"><a name="L1459"></a><tt class="py-lineno">1459</tt>  <tt class="py-line">        <tt class="py-comment"># if we have info then the qualifier key should have a ='s</tt> </tt>
<a name="L1460"></a><tt class="py-lineno">1460</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">'='</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_qualifier</tt><tt class="py-op">.</tt><tt class="py-name">key</tt><tt class="py-op">:</tt> </tt>
<a name="L1461"></a><tt class="py-lineno">1461</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_qualifier</tt><tt class="py-op">.</tt><tt class="py-name">key</tt> <tt class="py-op">=</tt> <tt class="py-string">"%s="</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_qualifier</tt><tt class="py-op">.</tt><tt class="py-name">key</tt> </tt>
<a name="L1462"></a><tt class="py-lineno">1462</tt>  <tt class="py-line">        <tt class="py-name">cur_content</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-723" class="py-name" targets="Method Bio.GenBank._BaseGenBankConsumer._remove_newlines()=Bio.GenBank._BaseGenBankConsumer-class.html#_remove_newlines"><a title="Bio.GenBank._BaseGenBankConsumer._remove_newlines" class="py-name" href="#" onclick="return doclink('link-723', '_remove_newlines', 'link-723');">_remove_newlines</a></tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">)</tt> </tt>
<a name="L1463"></a><tt class="py-lineno">1463</tt>  <tt class="py-line">        <tt class="py-comment"># remove all spaces from the value if it is a type where spaces</tt> </tt>
<a name="L1464"></a><tt class="py-lineno">1464</tt>  <tt class="py-line">        <tt class="py-comment"># are not important</tt> </tt>
<a name="L1465"></a><tt class="py-lineno">1465</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">remove_space_key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt id="link-724" class="py-name"><a title="Bio.GenBank._BaseGenBankConsumer.remove_space_keys" class="py-name" href="#" onclick="return doclink('link-724', 'remove_space_keys', 'link-228');">remove_space_keys</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1466"></a><tt class="py-lineno">1466</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">remove_space_key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_qualifier</tt><tt class="py-op">.</tt><tt class="py-name">key</tt><tt class="py-op">:</tt> </tt>
<a name="L1467"></a><tt class="py-lineno">1467</tt>  <tt class="py-line">                <tt class="py-name">cur_content</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-725" class="py-name" targets="Method Bio.GenBank._BaseGenBankConsumer._remove_spaces()=Bio.GenBank._BaseGenBankConsumer-class.html#_remove_spaces"><a title="Bio.GenBank._BaseGenBankConsumer._remove_spaces" class="py-name" href="#" onclick="return doclink('link-725', '_remove_spaces', 'link-725');">_remove_spaces</a></tt><tt class="py-op">(</tt><tt class="py-name">cur_content</tt><tt class="py-op">)</tt> </tt>
<a name="L1468"></a><tt class="py-lineno">1468</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_cur_qualifier</tt><tt class="py-op">.</tt><tt id="link-726" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-726', 'value', 'link-544');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-727" class="py-name" targets="Method Bio.GenBank._BaseGenBankConsumer._normalize_spaces()=Bio.GenBank._BaseGenBankConsumer-class.html#_normalize_spaces"><a title="Bio.GenBank._BaseGenBankConsumer._normalize_spaces" class="py-name" href="#" onclick="return doclink('link-727', '_normalize_spaces', 'link-727');">_normalize_spaces</a></tt><tt class="py-op">(</tt><tt class="py-name">cur_content</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1469"></a><tt class="py-lineno">1469</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.base_count"></a><div id="_RecordConsumer.base_count-def"><a name="L1470"></a><tt class="py-lineno">1470</tt> <a class="py-toggle" href="#" id="_RecordConsumer.base_count-toggle" onclick="return toggle('_RecordConsumer.base_count');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#base_count">base_count</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.base_count-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.base_count-expanded"><a name="L1471"></a><tt class="py-lineno">1471</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-728" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-728', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">base_counts</tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1472"></a><tt class="py-lineno">1472</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.origin_name"></a><div id="_RecordConsumer.origin_name-def"><a name="L1473"></a><tt class="py-lineno">1473</tt> <a class="py-toggle" href="#" id="_RecordConsumer.origin_name-toggle" onclick="return toggle('_RecordConsumer.origin_name');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#origin_name">origin_name</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.origin_name-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.origin_name-expanded"><a name="L1474"></a><tt class="py-lineno">1474</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-729" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-729', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">origin</tt> <tt class="py-op">=</tt> <tt class="py-name">content</tt> </tt>
</div><a name="L1475"></a><tt class="py-lineno">1475</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.contig_location"></a><div id="_RecordConsumer.contig_location-def"><a name="L1476"></a><tt class="py-lineno">1476</tt> <a class="py-toggle" href="#" id="_RecordConsumer.contig_location-toggle" onclick="return toggle('_RecordConsumer.contig_location');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#contig_location">contig_location</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.contig_location-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.contig_location-expanded"><a name="L1477"></a><tt class="py-lineno">1477</tt>  <tt class="py-line">        <tt class="py-docstring">"""Signal that we have contig information to add to the record.</tt> </tt>
<a name="L1478"></a><tt class="py-lineno">1478</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1479"></a><tt class="py-lineno">1479</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-730" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-730', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">contig</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-731" class="py-name"><a title="Bio.GenBank._BaseGenBankConsumer._clean_location" class="py-name" href="#" onclick="return doclink('link-731', '_clean_location', 'link-428');">_clean_location</a></tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1480"></a><tt class="py-lineno">1480</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.sequence"></a><div id="_RecordConsumer.sequence-def"><a name="L1481"></a><tt class="py-lineno">1481</tt> <a class="py-toggle" href="#" id="_RecordConsumer.sequence-toggle" onclick="return toggle('_RecordConsumer.sequence');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#sequence">sequence</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.sequence-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.sequence-expanded"><a name="L1482"></a><tt class="py-lineno">1482</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add sequence information to a list of sequence strings.</tt> </tt>
<a name="L1483"></a><tt class="py-lineno">1483</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1484"></a><tt class="py-lineno">1484</tt>  <tt class="py-line"><tt class="py-docstring">        This removes spaces in the data and uppercases the sequence, and</tt> </tt>
<a name="L1485"></a><tt class="py-lineno">1485</tt>  <tt class="py-line"><tt class="py-docstring">        then adds it to a list of sequences. Later on we'll join this</tt> </tt>
<a name="L1486"></a><tt class="py-lineno">1486</tt>  <tt class="py-line"><tt class="py-docstring">        list together to make the final sequence. This is faster than</tt> </tt>
<a name="L1487"></a><tt class="py-lineno">1487</tt>  <tt class="py-line"><tt class="py-docstring">        adding on the new string every time.</tt> </tt>
<a name="L1488"></a><tt class="py-lineno">1488</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1489"></a><tt class="py-lineno">1489</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-string">' '</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">content</tt> </tt>
<a name="L1490"></a><tt class="py-lineno">1490</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_data</tt><tt class="py-op">.</tt><tt id="link-732" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-732', 'append', 'link-213');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">content</tt><tt class="py-op">.</tt><tt id="link-733" class="py-name"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-733', 'upper', 'link-433');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1491"></a><tt class="py-lineno">1491</tt>  <tt class="py-line"> </tt>
<a name="_RecordConsumer.record_end"></a><div id="_RecordConsumer.record_end-def"><a name="L1492"></a><tt class="py-lineno">1492</tt> <a class="py-toggle" href="#" id="_RecordConsumer.record_end-toggle" onclick="return toggle('_RecordConsumer.record_end');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank._RecordConsumer-class.html#record_end">record_end</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">content</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_RecordConsumer.record_end-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_RecordConsumer.record_end-expanded"><a name="L1493"></a><tt class="py-lineno">1493</tt>  <tt class="py-line">        <tt class="py-docstring">"""Signal the end of the record and do any necessary clean-up.</tt> </tt>
<a name="L1494"></a><tt class="py-lineno">1494</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1495"></a><tt class="py-lineno">1495</tt>  <tt class="py-line">        <tt class="py-comment"># add together all of the sequence parts to create the</tt> </tt>
<a name="L1496"></a><tt class="py-lineno">1496</tt>  <tt class="py-line">        <tt class="py-comment"># final sequence string</tt> </tt>
<a name="L1497"></a><tt class="py-lineno">1497</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-734" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-734', 'data', 'link-223');">data</a></tt><tt class="py-op">.</tt><tt id="link-735" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-735', 'sequence', 'link-585');">sequence</a></tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_data</tt><tt class="py-op">)</tt> </tt>
<a name="L1498"></a><tt class="py-lineno">1498</tt>  <tt class="py-line">        <tt class="py-comment"># add on the last feature</tt> </tt>
<a name="L1499"></a><tt class="py-lineno">1499</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-736" class="py-name"><a title="Bio.GenBank._RecordConsumer._add_feature" class="py-name" href="#" onclick="return doclink('link-736', '_add_feature', 'link-704');">_add_feature</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1500"></a><tt class="py-lineno">1500</tt>  <tt class="py-line"> </tt>
<a name="L1501"></a><tt class="py-lineno">1501</tt>  <tt class="py-line"> </tt>
<a name="parse"></a><div id="parse-def"><a name="L1502"></a><tt class="py-lineno">1502</tt> <a class="py-toggle" href="#" id="parse-toggle" onclick="return toggle('parse');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank-module.html#parse">parse</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="parse-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="parse-expanded"><a name="L1503"></a><tt class="py-lineno">1503</tt>  <tt class="py-line">    <tt class="py-docstring">"""Iterate over GenBank formatted entries as Record objects.</tt> </tt>
<a name="L1504"></a><tt class="py-lineno">1504</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1505"></a><tt class="py-lineno">1505</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import GenBank</tt> </tt>
<a name="L1506"></a><tt class="py-lineno">1506</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; with open("GenBank/NC_000932.gb") as handle:</tt> </tt>
<a name="L1507"></a><tt class="py-lineno">1507</tt>  <tt class="py-line"><tt class="py-docstring">    ...     for record in GenBank.parse(handle):</tt> </tt>
<a name="L1508"></a><tt class="py-lineno">1508</tt>  <tt class="py-line"><tt class="py-docstring">    ...         print(record.accession)</tt> </tt>
<a name="L1509"></a><tt class="py-lineno">1509</tt>  <tt class="py-line"><tt class="py-docstring">    ['NC_000932']</tt> </tt>
<a name="L1510"></a><tt class="py-lineno">1510</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1511"></a><tt class="py-lineno">1511</tt>  <tt class="py-line"><tt class="py-docstring">    To get SeqRecord objects use Bio.SeqIO.parse(..., format="gb")</tt> </tt>
<a name="L1512"></a><tt class="py-lineno">1512</tt>  <tt class="py-line"><tt class="py-docstring">    instead.</tt> </tt>
<a name="L1513"></a><tt class="py-lineno">1513</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1514"></a><tt class="py-lineno">1514</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">iter</tt><tt class="py-op">(</tt><tt id="link-737" class="py-name" targets="Class Bio.Blast.NCBIStandalone.Iterator=Bio.Blast.NCBIStandalone.Iterator-class.html,Class Bio.GenBank.Iterator=Bio.GenBank.Iterator-class.html"><a title="Bio.Blast.NCBIStandalone.Iterator
Bio.GenBank.Iterator" class="py-name" href="#" onclick="return doclink('link-737', 'Iterator', 'link-737');">Iterator</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt id="link-738" class="py-name" targets="Class Bio.GenBank.RecordParser=Bio.GenBank.RecordParser-class.html"><a title="Bio.GenBank.RecordParser" class="py-name" href="#" onclick="return doclink('link-738', 'RecordParser', 'link-738');">RecordParser</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1515"></a><tt class="py-lineno">1515</tt>  <tt class="py-line"> </tt>
<a name="L1516"></a><tt class="py-lineno">1516</tt>  <tt class="py-line"> </tt>
<a name="read"></a><div id="read-def"><a name="L1517"></a><tt class="py-lineno">1517</tt> <a class="py-toggle" href="#" id="read-toggle" onclick="return toggle('read');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank-module.html#read">read</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="read-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="read-expanded"><a name="L1518"></a><tt class="py-lineno">1518</tt>  <tt class="py-line">    <tt class="py-docstring">"""Read a handle containing a single GenBank entry as a Record object.</tt> </tt>
<a name="L1519"></a><tt class="py-lineno">1519</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1520"></a><tt class="py-lineno">1520</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import GenBank</tt> </tt>
<a name="L1521"></a><tt class="py-lineno">1521</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; with open("GenBank/NC_000932.gb") as handle:</tt> </tt>
<a name="L1522"></a><tt class="py-lineno">1522</tt>  <tt class="py-line"><tt class="py-docstring">    ...     record = GenBank.read(handle)</tt> </tt>
<a name="L1523"></a><tt class="py-lineno">1523</tt>  <tt class="py-line"><tt class="py-docstring">    ...     print(record.accession)</tt> </tt>
<a name="L1524"></a><tt class="py-lineno">1524</tt>  <tt class="py-line"><tt class="py-docstring">    ['NC_000932']</tt> </tt>
<a name="L1525"></a><tt class="py-lineno">1525</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1526"></a><tt class="py-lineno">1526</tt>  <tt class="py-line"><tt class="py-docstring">    To get a SeqRecord object use Bio.SeqIO.read(..., format="gb")</tt> </tt>
<a name="L1527"></a><tt class="py-lineno">1527</tt>  <tt class="py-line"><tt class="py-docstring">    instead.</tt> </tt>
<a name="L1528"></a><tt class="py-lineno">1528</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1529"></a><tt class="py-lineno">1529</tt>  <tt class="py-line">    <tt class="py-name">iterator</tt> <tt class="py-op">=</tt> <tt id="link-739" class="py-name"><a title="Bio.AlignIO.parse
Bio.Blast.NCBIStandalone.BlastErrorParser.parse
Bio.Blast.NCBIStandalone.BlastParser.parse
Bio.Blast.NCBIStandalone.PSIBlastParser.parse
Bio.Blast.NCBIXML.parse
Bio.Compass.parse
Bio.Emboss.Primer3.parse
Bio.Entrez.Parser.DataHandler.parse
Bio.Entrez.parse
Bio.ExPASy.Enzyme.parse
Bio.ExPASy.Prodoc.parse
Bio.ExPASy.Prosite.parse
Bio.GenBank.FeatureParser.parse
Bio.GenBank.RecordParser.parse
Bio.GenBank.Scanner.InsdcScanner.parse
Bio.Geo.parse
Bio.KEGG.Compound.parse
Bio.KEGG.Enzyme.parse
Bio.KEGG.KGML.KGML_parser.KGMLParser.parse
Bio.KEGG.KGML.KGML_parser.parse
Bio.KEGG.Map.parse
Bio.Medline.parse
Bio.Motif.parse
Bio.ParserSupport.AbstractParser.parse
Bio.Phylo.CDAOIO.Parser.parse
Bio.Phylo.CDAOIO.parse
Bio.Phylo.NeXMLIO.Parser.parse
Bio.Phylo.NeXMLIO.parse
Bio.Phylo.NewickIO.Parser.parse
Bio.Phylo.NewickIO.parse
Bio.Phylo.NexusIO.parse
Bio.Phylo.PhyloXMLIO.Parser.parse
Bio.Phylo.PhyloXMLIO.parse
Bio.Phylo._io.parse
Bio.SCOP.Cla.parse
Bio.SCOP.Des.parse
Bio.SCOP.Dom.parse
Bio.SCOP.Hie.parse
Bio.SCOP.Raf.parse
Bio.SearchIO.parse
Bio.SeqIO.UniprotIO.Parser.parse
Bio.SeqIO.parse
Bio.Sequencing.Ace.parse
Bio.Sequencing.Phd.parse
Bio.SwissProt.KeyWList.parse
Bio.SwissProt.parse
Bio.UniGene.parse
Bio.Wise.psw.parse
Bio.motifs.parse" class="py-name" href="#" onclick="return doclink('link-739', 'parse', 'link-216');">parse</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L1530"></a><tt class="py-lineno">1530</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1531"></a><tt class="py-lineno">1531</tt>  <tt class="py-line">        <tt class="py-name">first</tt> <tt class="py-op">=</tt> <tt id="link-740" class="py-name" targets="Method Bio.AlignIO.Interfaces.AlignmentIterator.next()=Bio.AlignIO.Interfaces.AlignmentIterator-class.html#next,Method Bio.Blast.NCBIStandalone.Iterator.next()=Bio.Blast.NCBIStandalone.Iterator-class.html#next,Method Bio.Blast.ParseBlastTable.BlastTableReader.next()=Bio.Blast.ParseBlastTable.BlastTableReader-class.html#next,Method Bio.DocSQL.IterationCursor.next()=Bio.DocSQL.IterationCursor-class.html#next,Method Bio.File.UndoHandle.next()=Bio.File.UndoHandle-class.html#next,Method Bio.GenBank.Iterator.next()=Bio.GenBank.Iterator-class.html#next,Method Bio.Nexus.Nexus.CharBuffer.next()=Bio.Nexus.Nexus.CharBuffer-class.html#next,Method Bio.PopGen.GenePop.Controller._FileIterator.next()=Bio.PopGen.GenePop.Controller._FileIterator-class.html#next,Method Bio.SeqIO.Interfaces.SequenceIterator.next()=Bio.SeqIO.Interfaces.SequenceIterator-class.html#next,Method Bio.bgzf.BgzfReader.next()=Bio.bgzf.BgzfReader-class.html#next"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-740', 'next', 'link-740');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">iterator</tt><tt class="py-op">)</tt> </tt>
<a name="L1532"></a><tt class="py-lineno">1532</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L1533"></a><tt class="py-lineno">1533</tt>  <tt class="py-line">        <tt class="py-name">first</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L1534"></a><tt class="py-lineno">1534</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">first</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1535"></a><tt class="py-lineno">1535</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"No records found in handle"</tt><tt class="py-op">)</tt> </tt>
<a name="L1536"></a><tt class="py-lineno">1536</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1537"></a><tt class="py-lineno">1537</tt>  <tt class="py-line">        <tt class="py-name">second</tt> <tt class="py-op">=</tt> <tt id="link-741" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-741', 'next', 'link-740');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">iterator</tt><tt class="py-op">)</tt> </tt>
<a name="L1538"></a><tt class="py-lineno">1538</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L1539"></a><tt class="py-lineno">1539</tt>  <tt class="py-line">        <tt class="py-name">second</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L1540"></a><tt class="py-lineno">1540</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">second</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1541"></a><tt class="py-lineno">1541</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"More than one record found in handle"</tt><tt class="py-op">)</tt> </tt>
<a name="L1542"></a><tt class="py-lineno">1542</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">first</tt> </tt>
</div><a name="L1543"></a><tt class="py-lineno">1543</tt>  <tt class="py-line"> </tt>
<a name="L1544"></a><tt class="py-lineno">1544</tt>  <tt class="py-line"> </tt>
<a name="_test"></a><div id="_test-def"><a name="L1545"></a><tt class="py-lineno">1545</tt> <a class="py-toggle" href="#" id="_test-toggle" onclick="return toggle('_test');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank-module.html#_test">_test</a><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_test-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_test-expanded"><a name="L1546"></a><tt class="py-lineno">1546</tt>  <tt class="py-line">    <tt class="py-docstring">"""Run the Bio.GenBank module's doctests."""</tt> </tt>
<a name="L1547"></a><tt class="py-lineno">1547</tt>  <tt class="py-line">    <tt class="py-keyword">import</tt> <tt class="py-name">doctest</tt> </tt>
<a name="L1548"></a><tt class="py-lineno">1548</tt>  <tt class="py-line">    <tt class="py-keyword">import</tt> <tt class="py-name">os</tt> </tt>
<a name="L1549"></a><tt class="py-lineno">1549</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">isdir</tt><tt class="py-op">(</tt><tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-string">".."</tt><tt class="py-op">,</tt> <tt class="py-string">".."</tt><tt class="py-op">,</tt> <tt class="py-string">"Tests"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1550"></a><tt class="py-lineno">1550</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Running doctests..."</tt><tt class="py-op">)</tt> </tt>
<a name="L1551"></a><tt class="py-lineno">1551</tt>  <tt class="py-line">        <tt class="py-name">cur_dir</tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">abspath</tt><tt class="py-op">(</tt><tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">curdir</tt><tt class="py-op">)</tt> </tt>
<a name="L1552"></a><tt class="py-lineno">1552</tt>  <tt class="py-line">        <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">chdir</tt><tt class="py-op">(</tt><tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-string">".."</tt><tt class="py-op">,</tt> <tt class="py-string">".."</tt><tt class="py-op">,</tt> <tt class="py-string">"Tests"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1553"></a><tt class="py-lineno">1553</tt>  <tt class="py-line">        <tt class="py-name">doctest</tt><tt class="py-op">.</tt><tt class="py-name">testmod</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1554"></a><tt class="py-lineno">1554</tt>  <tt class="py-line">        <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">chdir</tt><tt class="py-op">(</tt><tt class="py-name">cur_dir</tt><tt class="py-op">)</tt> </tt>
<a name="L1555"></a><tt class="py-lineno">1555</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">cur_dir</tt> </tt>
<a name="L1556"></a><tt class="py-lineno">1556</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Done"</tt><tt class="py-op">)</tt> </tt>
<a name="L1557"></a><tt class="py-lineno">1557</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">isdir</tt><tt class="py-op">(</tt><tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-string">"Tests"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1558"></a><tt class="py-lineno">1558</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Running doctests..."</tt><tt class="py-op">)</tt> </tt>
<a name="L1559"></a><tt class="py-lineno">1559</tt>  <tt class="py-line">        <tt class="py-name">cur_dir</tt> <tt class="py-op">=</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">abspath</tt><tt class="py-op">(</tt><tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">curdir</tt><tt class="py-op">)</tt> </tt>
<a name="L1560"></a><tt class="py-lineno">1560</tt>  <tt class="py-line">        <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">chdir</tt><tt class="py-op">(</tt><tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-string">"Tests"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1561"></a><tt class="py-lineno">1561</tt>  <tt class="py-line">        <tt class="py-name">doctest</tt><tt class="py-op">.</tt><tt class="py-name">testmod</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1562"></a><tt class="py-lineno">1562</tt>  <tt class="py-line">        <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">chdir</tt><tt class="py-op">(</tt><tt class="py-name">cur_dir</tt><tt class="py-op">)</tt> </tt>
<a name="L1563"></a><tt class="py-lineno">1563</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">cur_dir</tt> </tt>
<a name="L1564"></a><tt class="py-lineno">1564</tt>  <tt class="py-line">        <tt class="py-keyword">print</tt><tt class="py-op">(</tt><tt class="py-string">"Done"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1565"></a><tt class="py-lineno">1565</tt>  <tt class="py-line"> </tt>
<a name="L1566"></a><tt class="py-lineno">1566</tt>  <tt class="py-line"><tt class="py-keyword">if</tt> <tt class="py-name">__name__</tt> <tt class="py-op">==</tt> <tt class="py-string">"__main__"</tt><tt class="py-op">:</tt> </tt>
<a name="L1567"></a><tt class="py-lineno">1567</tt>  <tt class="py-line">    <tt id="link-742" class="py-name" targets="Function Bio.Align.Applications._ClustalOmega._test()=Bio.Align.Applications._ClustalOmega-module.html#_test,Function Bio.Align.Applications._Clustalw._test()=Bio.Align.Applications._Clustalw-module.html#_test,Function Bio.Align.Applications._Dialign._test()=Bio.Align.Applications._Dialign-module.html#_test,Function Bio.Align.Applications._MSAProbs._test()=Bio.Align.Applications._MSAProbs-module.html#_test,Function Bio.Align.Applications._Muscle._test()=Bio.Align.Applications._Muscle-module.html#_test,Function Bio.Align.Applications._Prank._test()=Bio.Align.Applications._Prank-module.html#_test,Function Bio.Align.Applications._Probcons._test()=Bio.Align.Applications._Probcons-module.html#_test,Function Bio.Align.Applications._TCoffee._test()=Bio.Align.Applications._TCoffee-module.html#_test,Function Bio.Align.Generic._test()=Bio.Align.Generic-module.html#_test,Function Bio.Application._test()=Bio.Application-module.html#_test,Function Bio.Blast.Applications._test()=Bio.Blast.Applications-module.html#_test,Function Bio.DocSQL._test()=Bio.DocSQL-module.html#_test,Function Bio.Emboss.Applications._test()=Bio.Emboss.Applications-module.html#_test,Function Bio.Entrez._test()=Bio.Entrez-module.html#_test,Function Bio.KDTree.KDTree'._test()=Bio.KDTree.KDTree%27-module.html#_test,Function Bio.PDB.Selection._test()=Bio.PDB.Selection-module.html#_test,Function Bio.Seq._test()=Bio.Seq-module.html#_test,Function Bio.SeqUtils.MeltingTemp._test()=Bio.SeqUtils.MeltingTemp-module.html#_test,Function Bio.SeqUtils._test()=Bio.SeqUtils-module.html#_test,Function Bio.Sequencing.Applications._Novoalign._test()=Bio.Sequencing.Applications._Novoalign-module.html#_test,Function Bio.Sequencing.Applications._bwa._test()=Bio.Sequencing.Applications._bwa-module.html#_test,Function Bio.Sequencing.Applications._samtools._test()=Bio.Sequencing.Applications._samtools-module.html#_test,Function Bio.Statistics.lowess._test()=Bio.Statistics.lowess-module.html#_test,Function Bio.Wise._test()=Bio.Wise-module.html#_test,Function Bio.Wise.dnal._test()=Bio.Wise.dnal-module.html#_test,Function Bio.Wise.psw._test()=Bio.Wise.psw-module.html#_test,Function Bio.motifs.applications._alignace._test()=Bio.motifs.applications._alignace-module.html#_test,Function Bio.motifs.applications._xxmotif._test()=Bio.motifs.applications._xxmotif-module.html#_test,Function Bio.pairwise2._test()=Bio.pairwise2-module.html#_test"><a title="Bio.Align.Applications._ClustalOmega._test
Bio.Align.Applications._Clustalw._test
Bio.Align.Applications._Dialign._test
Bio.Align.Applications._MSAProbs._test
Bio.Align.Applications._Muscle._test
Bio.Align.Applications._Prank._test
Bio.Align.Applications._Probcons._test
Bio.Align.Applications._TCoffee._test
Bio.Align.Generic._test
Bio.Application._test
Bio.Blast.Applications._test
Bio.DocSQL._test
Bio.Emboss.Applications._test
Bio.Entrez._test
Bio.KDTree.KDTree'._test
Bio.PDB.Selection._test
Bio.Seq._test
Bio.SeqUtils.MeltingTemp._test
Bio.SeqUtils._test
Bio.Sequencing.Applications._Novoalign._test
Bio.Sequencing.Applications._bwa._test
Bio.Sequencing.Applications._samtools._test
Bio.Statistics.lowess._test
Bio.Wise._test
Bio.Wise.dnal._test
Bio.Wise.psw._test
Bio.motifs.applications._alignace._test
Bio.motifs.applications._xxmotif._test
Bio.pairwise2._test" class="py-name" href="#" onclick="return doclink('link-742', '_test', 'link-742');">_test</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1568"></a><tt class="py-lineno">1568</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Oct 21 18:04:55 2015
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
