<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.GenBank.Record</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.GenBank-module.html">Package&nbsp;GenBank</a> ::
        Module&nbsp;Record
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.GenBank.Record-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.GenBank.Record-module.html">Module Bio.GenBank.Record</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-docstring">"""Hold GenBank data in a straightforward format.</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-docstring">classes:</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-docstring">    - Record - All of the information in a GenBank record.</tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-docstring">    - Reference - hold reference data for a record.</tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-docstring">    - Feature - Hold the information in a Feature Table.</tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-docstring">    - Qualifier - Qualifiers on a Feature.</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-docstring">17-MAR-2009: added support for WGS and WGS_SCAFLD lines.  Ying Huang &amp; Iddo Friedberg</tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-comment"># local stuff</tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio.GenBank=Bio.GenBank-module.html"><a title="Bio.GenBank" class="py-name" href="#" onclick="return doclink('link-1', 'GenBank', 'link-1');">GenBank</a></tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-name">__docformat__</tt> <tt class="py-op">=</tt> <tt class="py-string">"restructuredtext en"</tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"> </tt>
<a name="_wrapped_genbank"></a><div id="_wrapped_genbank-def"><a name="L23"></a><tt class="py-lineno"> 23</tt> <a class="py-toggle" href="#" id="_wrapped_genbank-toggle" onclick="return toggle('_wrapped_genbank');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record-module.html#_wrapped_genbank">_wrapped_genbank</a><tt class="py-op">(</tt><tt class="py-param">information</tt><tt class="py-op">,</tt> <tt class="py-param">indent</tt><tt class="py-op">,</tt> <tt class="py-param">wrap_space</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-param">split_char</tt><tt class="py-op">=</tt><tt class="py-string">" "</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_wrapped_genbank-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_wrapped_genbank-expanded"><a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line">    <tt class="py-docstring">"""Write a line of GenBank info that can wrap over multiple lines.</tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"><tt class="py-docstring">    This takes a line of information which can potentially wrap over</tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"><tt class="py-docstring">    multiple lines, and breaks it up with carriage returns and</tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"><tt class="py-docstring">    indentation so it fits properly into a GenBank record.</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"><tt class="py-docstring">    Arguments:</tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"><tt class="py-docstring">        - information - The string holding the information we want</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-docstring">          wrapped in GenBank method.</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-docstring">        - indent - The indentation on the lines we are writing.</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"><tt class="py-docstring">        - wrap_space - Whether or not to wrap only on spaces in the</tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"><tt class="py-docstring">          information.</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"><tt class="py-docstring">        - split_char - A specific character to split the lines on. By default</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"><tt class="py-docstring">          spaces are used.</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line">    <tt class="py-name">info_length</tt> <tt class="py-op">=</tt> <tt id="link-2" class="py-name" targets="Class Bio.Affy.CelFile.Record=Bio.Affy.CelFile.Record-class.html,Module Bio.Blast.Record=Bio.Blast.Record-module.html,Class Bio.Cluster.Record=Bio.Cluster.Record-class.html,Class Bio.Compass.Record=Bio.Compass.Record-class.html,Class Bio.Emboss.Primer3.Record=Bio.Emboss.Primer3.Record-class.html,Class Bio.ExPASy.Enzyme.Record=Bio.ExPASy.Enzyme.Record-class.html,Class Bio.ExPASy.Prodoc.Record=Bio.ExPASy.Prodoc.Record-class.html,Class Bio.ExPASy.Prosite.Record=Bio.ExPASy.Prosite.Record-class.html,Class Bio.ExPASy.ScanProsite.Record=Bio.ExPASy.ScanProsite.Record-class.html,Module Bio.GenBank.Record=Bio.GenBank.Record-module.html,Class Bio.GenBank.Record.Record=Bio.GenBank.Record.Record-class.html,Module Bio.Geo.Record=Bio.Geo.Record-module.html,Class Bio.Geo.Record.Record=Bio.Geo.Record.Record-class.html,Class Bio.KEGG.Compound.Record=Bio.KEGG.Compound.Record-class.html,Class Bio.KEGG.Enzyme.Record=Bio.KEGG.Enzyme.Record-class.html,Class Bio.Medline.Record=Bio.Medline.Record-class.html,Class Bio.Motif.Parsers.AlignAce.Record=Bio.Motif.Parsers.AlignAce.Record-class.html,Class Bio.Motif.Parsers.MAST.Record=Bio.Motif.Parsers.MAST.Record-class.html,Class Bio.PopGen.FDist.Record=Bio.PopGen.FDist.Record-class.html,Class Bio.PopGen.GenePop.LargeFileParser.Record=Bio.PopGen.GenePop.LargeFileParser.Record-class.html,Class Bio.PopGen.GenePop.Record=Bio.PopGen.GenePop.Record-class.html,Class Bio.SCOP.Cla.Record=Bio.SCOP.Cla.Record-class.html,Class Bio.SCOP.Des.Record=Bio.SCOP.Des.Record-class.html,Class Bio.SCOP.Dom.Record=Bio.SCOP.Dom.Record-class.html,Class Bio.SCOP.Hie.Record=Bio.SCOP.Hie.Record-class.html,Class Bio.Sequencing.Phd.Record=Bio.Sequencing.Phd.Record-class.html,Class Bio.SwissProt.KeyWList.Record=Bio.SwissProt.KeyWList.Record-class.html,Class Bio.SwissProt.Record=Bio.SwissProt.Record-class.html,Class Bio.UniGene.Record=Bio.UniGene.Record-class.html,Class Bio.motifs.alignace.Record=Bio.motifs.alignace.Record-class.html,Class Bio.motifs.jaspar.Record=Bio.motifs.jaspar.Record-class.html,Class Bio.motifs.mast.Record=Bio.motifs.mast.Record-class.html,Class Bio.motifs.meme.Record=Bio.motifs.meme.Record-class.html,Class Bio.motifs.transfac.Record=Bio.motifs.transfac.Record-class.html"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-2', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-3" class="py-name" targets="Variable Bio.GenBank.Record.Record.GB_LINE_LENGTH=Bio.GenBank.Record.Record-class.html#GB_LINE_LENGTH"><a title="Bio.GenBank.Record.Record.GB_LINE_LENGTH" class="py-name" href="#" onclick="return doclink('link-3', 'GB_LINE_LENGTH', 'link-3');">GB_LINE_LENGTH</a></tt> <tt class="py-op">-</tt> <tt class="py-name">indent</tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line"> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">information</tt><tt class="py-op">:</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line">        <tt class="py-comment"># GenBank files use "." for missing data</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">".\n"</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line"> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">wrap_space</tt><tt class="py-op">:</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line">        <tt class="py-name">info_parts</tt> <tt class="py-op">=</tt> <tt class="py-name">information</tt><tt class="py-op">.</tt><tt id="link-4" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-4', 'split', 'link-4');">split</a></tt><tt class="py-op">(</tt><tt class="py-name">split_char</tt><tt class="py-op">)</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line">        <tt class="py-name">cur_pos</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line">        <tt class="py-name">info_parts</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">cur_pos</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">information</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line">            <tt class="py-name">info_parts</tt><tt class="py-op">.</tt><tt id="link-5" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-5', 'append', 'link-5');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">information</tt><tt class="py-op">[</tt><tt class="py-name">cur_pos</tt><tt class="py-op">:</tt> <tt class="py-name">cur_pos</tt> <tt class="py-op">+</tt> <tt class="py-name">info_length</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line">            <tt class="py-name">cur_pos</tt> <tt class="py-op">+=</tt> <tt class="py-name">info_length</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line">    <tt class="py-comment"># first get the information string split up by line</tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line">    <tt class="py-name">output_parts</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line">    <tt class="py-name">cur_part</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">info_part</tt> <tt class="py-keyword">in</tt> <tt class="py-name">info_parts</tt><tt class="py-op">:</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">cur_part</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt> <tt class="py-op">+</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">info_part</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-name">info_length</tt><tt class="py-op">:</tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">cur_part</tt><tt class="py-op">:</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">split_char</tt> <tt class="py-op">!=</tt> <tt class="py-string">" "</tt><tt class="py-op">:</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line">                    <tt class="py-name">cur_part</tt> <tt class="py-op">+=</tt> <tt class="py-name">split_char</tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line">                <tt class="py-name">output_parts</tt><tt class="py-op">.</tt><tt id="link-6" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-6', 'append', 'link-5');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">cur_part</tt><tt class="py-op">)</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line">            <tt class="py-name">cur_part</tt> <tt class="py-op">=</tt> <tt class="py-name">info_part</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">cur_part</tt> <tt class="py-op">==</tt> <tt class="py-string">""</tt><tt class="py-op">:</tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line">                <tt class="py-name">cur_part</tt> <tt class="py-op">=</tt> <tt class="py-name">info_part</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line">                <tt class="py-name">cur_part</tt> <tt class="py-op">+=</tt> <tt class="py-name">split_char</tt> <tt class="py-op">+</tt> <tt class="py-name">info_part</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line"> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line">    <tt class="py-comment"># add the last bit of information to the output</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">cur_part</tt><tt class="py-op">:</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line">        <tt class="py-name">output_parts</tt><tt class="py-op">.</tt><tt id="link-7" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-7', 'append', 'link-5');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">cur_part</tt><tt class="py-op">)</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line"> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line">    <tt class="py-comment"># now format the information string for return</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line">    <tt class="py-name">output_info</tt> <tt class="py-op">=</tt> <tt class="py-name">output_parts</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">output_part</tt> <tt class="py-keyword">in</tt> <tt class="py-name">output_parts</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line">        <tt class="py-name">output_info</tt> <tt class="py-op">+=</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt class="py-name">indent</tt> <tt class="py-op">+</tt> <tt class="py-name">output_part</tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line"> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">output_info</tt> </tt>
</div><a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line"> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line"> </tt>
<a name="_indent_genbank"></a><div id="_indent_genbank-def"><a name="L86"></a><tt class="py-lineno"> 86</tt> <a class="py-toggle" href="#" id="_indent_genbank-toggle" onclick="return toggle('_indent_genbank');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record-module.html#_indent_genbank">_indent_genbank</a><tt class="py-op">(</tt><tt class="py-param">information</tt><tt class="py-op">,</tt> <tt class="py-param">indent</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_indent_genbank-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_indent_genbank-expanded"><a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line">    <tt class="py-docstring">"""Write out information with the specified indent.</tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line"><tt class="py-docstring">    Unlike _wrapped_genbank, this function makes no attempt to wrap</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line"><tt class="py-docstring">    lines -- it assumes that the information already has newlines in the</tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line"><tt class="py-docstring">    appropriate places, and will add the specified indent to the start of</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line"><tt class="py-docstring">    each line.</tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line">    <tt class="py-comment"># split the info into lines based on line breaks</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line">    <tt class="py-name">info_parts</tt> <tt class="py-op">=</tt> <tt class="py-name">information</tt><tt class="py-op">.</tt><tt id="link-8" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-8', 'split', 'link-4');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">"\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line"> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line">    <tt class="py-comment"># the first line will have no indent</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line">    <tt class="py-name">output_info</tt> <tt class="py-op">=</tt> <tt class="py-name">info_parts</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">info_part</tt> <tt class="py-keyword">in</tt> <tt class="py-name">info_parts</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line">        <tt class="py-name">output_info</tt> <tt class="py-op">+=</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt class="py-name">indent</tt> <tt class="py-op">+</tt> <tt class="py-name">info_part</tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line"> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">output_info</tt> </tt>
</div><a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line"> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line"> </tt>
<a name="Record"></a><div id="Record-def"><a name="L105"></a><tt class="py-lineno">105</tt> <a class="py-toggle" href="#" id="Record-toggle" onclick="return toggle('Record');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.GenBank.Record.Record-class.html">Record</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Record-expanded"><a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line">    <tt class="py-docstring">"""Hold GenBank information in a format similar to the original record.</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line"><tt class="py-docstring">    The Record class is meant to make data easy to get to when you are</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line"><tt class="py-docstring">    just interested in looking at GenBank data.</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line"><tt class="py-docstring">    Attributes:</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line"><tt class="py-docstring">        - locus - The name specified after the LOCUS keyword in the GenBank</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line"><tt class="py-docstring">          record. This may be the accession number, or a clone id or something else.</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line"><tt class="py-docstring">        - size - The size of the record.</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line"><tt class="py-docstring">        - residue_type - The type of residues making up the sequence in this</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line"><tt class="py-docstring">          record. Normally something like RNA, DNA or PROTEIN, but may be as</tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line"><tt class="py-docstring">          esoteric as 'ss-RNA circular'.</tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line"><tt class="py-docstring">        - data_file_division - The division this record is stored under in</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line"><tt class="py-docstring">          GenBank (ie. PLN -&gt; plants; PRI -&gt; humans, primates; BCT -&gt; bacteria...)</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line"><tt class="py-docstring">        - date - The date of submission of the record, in a form like '28-JUL-1998'</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line"><tt class="py-docstring">        - accession - list of all accession numbers for the sequence.</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line"><tt class="py-docstring">        - nid - Nucleotide identifier number.</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line"><tt class="py-docstring">        - pid - Proteint identifier number</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line"><tt class="py-docstring">        - version - The accession number + version (ie. AB01234.2)</tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line"><tt class="py-docstring">        - db_source - Information about the database the record came from</tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line"><tt class="py-docstring">        - gi - The NCBI gi identifier for the record.</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line"><tt class="py-docstring">        - keywords - A list of keywords related to the record.</tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line"><tt class="py-docstring">        - segment - If the record is one of a series, this is info about which</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line"><tt class="py-docstring">          segment this record is (something like '1 of 6').</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line"><tt class="py-docstring">        - source - The source of material where the sequence came from.</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line"><tt class="py-docstring">        - organism - The genus and species of the organism (ie. 'Homo sapiens')</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line"><tt class="py-docstring">        - taxonomy - A listing of the taxonomic classification of the organism,</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line"><tt class="py-docstring">          starting general and getting more specific.</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line"><tt class="py-docstring">        - references - A list of Reference objects.</tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line"><tt class="py-docstring">        - comment - Text with any kind of comment about the record.</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line"><tt class="py-docstring">        - features - A listing of Features making up the feature table.</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line"><tt class="py-docstring">        - base_counts - A string with the counts of bases for the sequence.</tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line"><tt class="py-docstring">        - origin - A string specifying info about the origin of the sequence.</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line"><tt class="py-docstring">        - sequence - A string with the sequence itself.</tt> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line"><tt class="py-docstring">        - contig - A string of location information for a CONTIG in a RefSeq file</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line"><tt class="py-docstring">        - project - The genome sequencing project numbers</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line"><tt class="py-docstring">          (will be replaced by the dblink cross-references in 2009).</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line"><tt class="py-docstring">        - dblinks - The genome sequencing project number(s) and other links.</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line"><tt class="py-docstring">          (will replace the project information in 2009).</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line">    <tt class="py-comment"># constants for outputting GenBank information</tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line">    <tt id="link-9" class="py-name"><a title="Bio.GenBank.Record.Record.GB_LINE_LENGTH" class="py-name" href="#" onclick="return doclink('link-9', 'GB_LINE_LENGTH', 'link-3');">GB_LINE_LENGTH</a></tt> <tt class="py-op">=</tt> <tt class="py-number">79</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line">    <tt id="link-10" class="py-name" targets="Variable Bio.GenBank.Record.Record.GB_BASE_INDENT=Bio.GenBank.Record.Record-class.html#GB_BASE_INDENT"><a title="Bio.GenBank.Record.Record.GB_BASE_INDENT" class="py-name" href="#" onclick="return doclink('link-10', 'GB_BASE_INDENT', 'link-10');">GB_BASE_INDENT</a></tt> <tt class="py-op">=</tt> <tt class="py-number">12</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">    <tt id="link-11" class="py-name" targets="Variable Bio.GenBank.Record.Record.GB_FEATURE_INDENT=Bio.GenBank.Record.Record-class.html#GB_FEATURE_INDENT"><a title="Bio.GenBank.Record.Record.GB_FEATURE_INDENT" class="py-name" href="#" onclick="return doclink('link-11', 'GB_FEATURE_INDENT', 'link-11');">GB_FEATURE_INDENT</a></tt> <tt class="py-op">=</tt> <tt class="py-number">21</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">    <tt id="link-12" class="py-name" targets="Variable Bio.GenBank.Record.Record.GB_INTERNAL_INDENT=Bio.GenBank.Record.Record-class.html#GB_INTERNAL_INDENT"><a title="Bio.GenBank.Record.Record.GB_INTERNAL_INDENT" class="py-name" href="#" onclick="return doclink('link-12', 'GB_INTERNAL_INDENT', 'link-12');">GB_INTERNAL_INDENT</a></tt> <tt class="py-op">=</tt> <tt class="py-number">2</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line">    <tt id="link-13" class="py-name" targets="Variable Bio.GenBank.Record.Record.GB_OTHER_INTERNAL_INDENT=Bio.GenBank.Record.Record-class.html#GB_OTHER_INTERNAL_INDENT"><a title="Bio.GenBank.Record.Record.GB_OTHER_INTERNAL_INDENT" class="py-name" href="#" onclick="return doclink('link-13', 'GB_OTHER_INTERNAL_INDENT', 'link-13');">GB_OTHER_INTERNAL_INDENT</a></tt> <tt class="py-op">=</tt> <tt class="py-number">3</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">    <tt id="link-14" class="py-name" targets="Variable Bio.GenBank.Record.Record.GB_FEATURE_INTERNAL_INDENT=Bio.GenBank.Record.Record-class.html#GB_FEATURE_INTERNAL_INDENT"><a title="Bio.GenBank.Record.Record.GB_FEATURE_INTERNAL_INDENT" class="py-name" href="#" onclick="return doclink('link-14', 'GB_FEATURE_INTERNAL_INDENT', 'link-14');">GB_FEATURE_INTERNAL_INDENT</a></tt> <tt class="py-op">=</tt> <tt class="py-number">5</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line">    <tt id="link-15" class="py-name" targets="Variable Bio.GenBank.Record.Record.GB_SEQUENCE_INDENT=Bio.GenBank.Record.Record-class.html#GB_SEQUENCE_INDENT"><a title="Bio.GenBank.Record.Record.GB_SEQUENCE_INDENT" class="py-name" href="#" onclick="return doclink('link-15', 'GB_SEQUENCE_INDENT', 'link-15');">GB_SEQUENCE_INDENT</a></tt> <tt class="py-op">=</tt> <tt class="py-number">9</tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line"> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line">    <tt id="link-16" class="py-name" targets="Variable Bio.GenBank.Record.Record.BASE_FORMAT=Bio.GenBank.Record.Record-class.html#BASE_FORMAT"><a title="Bio.GenBank.Record.Record.BASE_FORMAT" class="py-name" href="#" onclick="return doclink('link-16', 'BASE_FORMAT', 'link-16');">BASE_FORMAT</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"%-"</tt> <tt class="py-op">+</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-17" class="py-name"><a title="Bio.GenBank.Record.Record.GB_BASE_INDENT" class="py-name" href="#" onclick="return doclink('link-17', 'GB_BASE_INDENT', 'link-10');">GB_BASE_INDENT</a></tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">"s"</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">    <tt id="link-18" class="py-name" targets="Variable Bio.GenBank.Record.Record.INTERNAL_FORMAT=Bio.GenBank.Record.Record-class.html#INTERNAL_FORMAT"><a title="Bio.GenBank.Record.Record.INTERNAL_FORMAT" class="py-name" href="#" onclick="return doclink('link-18', 'INTERNAL_FORMAT', 'link-18');">INTERNAL_FORMAT</a></tt> <tt class="py-op">=</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt id="link-19" class="py-name"><a title="Bio.GenBank.Record.Record.GB_INTERNAL_INDENT" class="py-name" href="#" onclick="return doclink('link-19', 'GB_INTERNAL_INDENT', 'link-12');">GB_INTERNAL_INDENT</a></tt> <tt class="py-op">+</tt> <tt class="py-string">"%-"</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">                      <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-20" class="py-name"><a title="Bio.GenBank.Record.Record.GB_BASE_INDENT" class="py-name" href="#" onclick="return doclink('link-20', 'GB_BASE_INDENT', 'link-10');">GB_BASE_INDENT</a></tt> <tt class="py-op">-</tt> <tt id="link-21" class="py-name"><a title="Bio.GenBank.Record.Record.GB_INTERNAL_INDENT" class="py-name" href="#" onclick="return doclink('link-21', 'GB_INTERNAL_INDENT', 'link-12');">GB_INTERNAL_INDENT</a></tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">"s"</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">    <tt id="link-22" class="py-name" targets="Variable Bio.GenBank.Record.Record.OTHER_INTERNAL_FORMAT=Bio.GenBank.Record.Record-class.html#OTHER_INTERNAL_FORMAT"><a title="Bio.GenBank.Record.Record.OTHER_INTERNAL_FORMAT" class="py-name" href="#" onclick="return doclink('link-22', 'OTHER_INTERNAL_FORMAT', 'link-22');">OTHER_INTERNAL_FORMAT</a></tt> <tt class="py-op">=</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt id="link-23" class="py-name"><a title="Bio.GenBank.Record.Record.GB_OTHER_INTERNAL_INDENT" class="py-name" href="#" onclick="return doclink('link-23', 'GB_OTHER_INTERNAL_INDENT', 'link-13');">GB_OTHER_INTERNAL_INDENT</a></tt> <tt class="py-op">+</tt> <tt class="py-string">"%-"</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">                            <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-24" class="py-name"><a title="Bio.GenBank.Record.Record.GB_BASE_INDENT" class="py-name" href="#" onclick="return doclink('link-24', 'GB_BASE_INDENT', 'link-10');">GB_BASE_INDENT</a></tt> <tt class="py-op">-</tt> <tt id="link-25" class="py-name"><a title="Bio.GenBank.Record.Record.GB_OTHER_INTERNAL_INDENT" class="py-name" href="#" onclick="return doclink('link-25', 'GB_OTHER_INTERNAL_INDENT', 'link-13');">GB_OTHER_INTERNAL_INDENT</a></tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">                            <tt class="py-string">"s"</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line"> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line">    <tt id="link-26" class="py-name" targets="Variable Bio.GenBank.Record.Record.BASE_FEATURE_FORMAT=Bio.GenBank.Record.Record-class.html#BASE_FEATURE_FORMAT"><a title="Bio.GenBank.Record.Record.BASE_FEATURE_FORMAT" class="py-name" href="#" onclick="return doclink('link-26', 'BASE_FEATURE_FORMAT', 'link-26');">BASE_FEATURE_FORMAT</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"%-"</tt> <tt class="py-op">+</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-27" class="py-name"><a title="Bio.GenBank.Record.Record.GB_FEATURE_INDENT" class="py-name" href="#" onclick="return doclink('link-27', 'GB_FEATURE_INDENT', 'link-11');">GB_FEATURE_INDENT</a></tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">"s"</tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">    <tt id="link-28" class="py-name" targets="Variable Bio.GenBank.Record.Record.INTERNAL_FEATURE_FORMAT=Bio.GenBank.Record.Record-class.html#INTERNAL_FEATURE_FORMAT"><a title="Bio.GenBank.Record.Record.INTERNAL_FEATURE_FORMAT" class="py-name" href="#" onclick="return doclink('link-28', 'INTERNAL_FEATURE_FORMAT', 'link-28');">INTERNAL_FEATURE_FORMAT</a></tt> <tt class="py-op">=</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt id="link-29" class="py-name"><a title="Bio.GenBank.Record.Record.GB_FEATURE_INTERNAL_INDENT" class="py-name" href="#" onclick="return doclink('link-29', 'GB_FEATURE_INTERNAL_INDENT', 'link-14');">GB_FEATURE_INTERNAL_INDENT</a></tt> <tt class="py-op">+</tt> <tt class="py-string">"%-"</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line">                              <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-30" class="py-name"><a title="Bio.GenBank.Record.Record.GB_FEATURE_INDENT" class="py-name" href="#" onclick="return doclink('link-30', 'GB_FEATURE_INDENT', 'link-11');">GB_FEATURE_INDENT</a></tt> <tt class="py-op">-</tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line">                                  <tt id="link-31" class="py-name"><a title="Bio.GenBank.Record.Record.GB_FEATURE_INTERNAL_INDENT" class="py-name" href="#" onclick="return doclink('link-31', 'GB_FEATURE_INTERNAL_INDENT', 'link-14');">GB_FEATURE_INTERNAL_INDENT</a></tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">"s"</tt> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">    <tt id="link-32" class="py-name" targets="Variable Bio.GenBank.Record.Record.SEQUENCE_FORMAT=Bio.GenBank.Record.Record-class.html#SEQUENCE_FORMAT"><a title="Bio.GenBank.Record.Record.SEQUENCE_FORMAT" class="py-name" href="#" onclick="return doclink('link-32', 'SEQUENCE_FORMAT', 'link-32');">SEQUENCE_FORMAT</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"%"</tt> <tt class="py-op">+</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-33" class="py-name"><a title="Bio.GenBank.Record.Record.GB_SEQUENCE_INDENT" class="py-name" href="#" onclick="return doclink('link-33', 'GB_SEQUENCE_INDENT', 'link-15');">GB_SEQUENCE_INDENT</a></tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">"s"</tt> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line"> </tt>
<a name="Record.__init__"></a><div id="Record.__init__-def"><a name="L169"></a><tt class="py-lineno">169</tt> <a class="py-toggle" href="#" id="Record.__init__-toggle" onclick="return toggle('Record.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Record-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record.__init__-expanded"><a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-34" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.locus()=Bio.GenBank._FeatureConsumer-class.html#locus,Method Bio.GenBank._RecordConsumer.locus()=Bio.GenBank._RecordConsumer-class.html#locus"><a title="Bio.GenBank._FeatureConsumer.locus
Bio.GenBank._RecordConsumer.locus" class="py-name" href="#" onclick="return doclink('link-34', 'locus', 'link-34');">locus</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-35" class="py-name" targets="Variable Bio.Alphabet.Alphabet.size=Bio.Alphabet.Alphabet-class.html#size,Variable Bio.Alphabet.Reduced.HPModel.size=Bio.Alphabet.Reduced.HPModel-class.html#size,Variable Bio.Alphabet.Reduced.Murphy10.size=Bio.Alphabet.Reduced.Murphy10-class.html#size,Variable Bio.Alphabet.Reduced.Murphy15.size=Bio.Alphabet.Reduced.Murphy15-class.html#size,Variable Bio.Alphabet.Reduced.Murphy4.size=Bio.Alphabet.Reduced.Murphy4-class.html#size,Variable Bio.Alphabet.Reduced.Murphy8.size=Bio.Alphabet.Reduced.Murphy8-class.html#size,Variable Bio.Alphabet.Reduced.PC5.size=Bio.Alphabet.Reduced.PC5-class.html#size,Variable Bio.Alphabet.SingleLetterAlphabet.size=Bio.Alphabet.SingleLetterAlphabet-class.html#size,Variable Bio.Alphabet.ThreeLetterProtein.size=Bio.Alphabet.ThreeLetterProtein-class.html#size,Method Bio.GenBank._FeatureConsumer.size()=Bio.GenBank._FeatureConsumer-class.html#size,Method Bio.GenBank._RecordConsumer.size()=Bio.GenBank._RecordConsumer-class.html#size,Variable Bio.Restriction.Restriction.AanI.size=Bio.Restriction.Restriction.AanI-class.html#size,Variable Bio.Restriction.Restriction.AarI.size=Bio.Restriction.Restriction.AarI-class.html#size,Variable Bio.Restriction.Restriction.AasI.size=Bio.Restriction.Restriction.AasI-class.html#size,Variable Bio.Restriction.Restriction.AatII.size=Bio.Restriction.Restriction.AatII-class.html#size,Variable Bio.Restriction.Restriction.AbaSI.size=Bio.Restriction.Restriction.AbaSI-class.html#size,Variable Bio.Restriction.Restriction.AbsI.size=Bio.Restriction.Restriction.AbsI-class.html#size,Variable Bio.Restriction.Restriction.Acc16I.size=Bio.Restriction.Restriction.Acc16I-class.html#size,Variable Bio.Restriction.Restriction.Acc36I.size=Bio.Restriction.Restriction.Acc36I-class.html#size,Variable Bio.Restriction.Restriction.Acc65I.size=Bio.Restriction.Restriction.Acc65I-class.html#size,Variable Bio.Restriction.Restriction.AccB1I.size=Bio.Restriction.Restriction.AccB1I-class.html#size,Variable Bio.Restriction.Restriction.AccB7I.size=Bio.Restriction.Restriction.AccB7I-class.html#size,Variable Bio.Restriction.Restriction.AccBSI.size=Bio.Restriction.Restriction.AccBSI-class.html#size,Variable Bio.Restriction.Restriction.AccI.size=Bio.Restriction.Restriction.AccI-class.html#size,Variable Bio.Restriction.Restriction.AccII.size=Bio.Restriction.Restriction.AccII-class.html#size,Variable Bio.Restriction.Restriction.AccIII.size=Bio.Restriction.Restriction.AccIII-class.html#size,Variable Bio.Restriction.Restriction.AceIII.size=Bio.Restriction.Restriction.AceIII-class.html#size,Variable Bio.Restriction.Restriction.AciI.size=Bio.Restriction.Restriction.AciI-class.html#size,Variable Bio.Restriction.Restriction.AclI.size=Bio.Restriction.Restriction.AclI-class.html#size,Variable Bio.Restriction.Restriction.AclWI.size=Bio.Restriction.Restriction.AclWI-class.html#size,Variable Bio.Restriction.Restriction.AcoI.size=Bio.Restriction.Restriction.AcoI-class.html#size,Variable Bio.Restriction.Restriction.AcsI.size=Bio.Restriction.Restriction.AcsI-class.html#size,Variable Bio.Restriction.Restriction.AcuI.size=Bio.Restriction.Restriction.AcuI-class.html#size,Variable Bio.Restriction.Restriction.AcvI.size=Bio.Restriction.Restriction.AcvI-class.html#size,Variable Bio.Restriction.Restriction.AcyI.size=Bio.Restriction.Restriction.AcyI-class.html#size,Variable Bio.Restriction.Restriction.AdeI.size=Bio.Restriction.Restriction.AdeI-class.html#size,Variable Bio.Restriction.Restriction.AfaI.size=Bio.Restriction.Restriction.AfaI-class.html#size,Variable Bio.Restriction.Restriction.AfeI.size=Bio.Restriction.Restriction.AfeI-class.html#size,Variable Bio.Restriction.Restriction.AfiI.size=Bio.Restriction.Restriction.AfiI-class.html#size,Variable Bio.Restriction.Restriction.AflII.size=Bio.Restriction.Restriction.AflII-class.html#size,Variable Bio.Restriction.Restriction.AflIII.size=Bio.Restriction.Restriction.AflIII-class.html#size,Variable Bio.Restriction.Restriction.AgeI.size=Bio.Restriction.Restriction.AgeI-class.html#size,Variable Bio.Restriction.Restriction.AgsI.size=Bio.Restriction.Restriction.AgsI-class.html#size,Variable Bio.Restriction.Restriction.AhaIII.size=Bio.Restriction.Restriction.AhaIII-class.html#size,Variable Bio.Restriction.Restriction.AhdI.size=Bio.Restriction.Restriction.AhdI-class.html#size,Variable Bio.Restriction.Restriction.AhlI.size=Bio.Restriction.Restriction.AhlI-class.html#size,Variable Bio.Restriction.Restriction.AjiI.size=Bio.Restriction.Restriction.AjiI-class.html#size,Variable Bio.Restriction.Restriction.AjnI.size=Bio.Restriction.Restriction.AjnI-class.html#size,Variable Bio.Restriction.Restriction.AjuI.size=Bio.Restriction.Restriction.AjuI-class.html#size,Variable Bio.Restriction.Restriction.AleI.size=Bio.Restriction.Restriction.AleI-class.html#size,Variable Bio.Restriction.Restriction.AlfI.size=Bio.Restriction.Restriction.AlfI-class.html#size,Variable Bio.Restriction.Restriction.AloI.size=Bio.Restriction.Restriction.AloI-class.html#size,Variable Bio.Restriction.Restriction.AluBI.size=Bio.Restriction.Restriction.AluBI-class.html#size,Variable Bio.Restriction.Restriction.AluI.size=Bio.Restriction.Restriction.AluI-class.html#size,Variable Bio.Restriction.Restriction.Alw21I.size=Bio.Restriction.Restriction.Alw21I-class.html#size,Variable Bio.Restriction.Restriction.Alw26I.size=Bio.Restriction.Restriction.Alw26I-class.html#size,Variable Bio.Restriction.Restriction.Alw44I.size=Bio.Restriction.Restriction.Alw44I-class.html#size,Variable Bio.Restriction.Restriction.AlwFI.size=Bio.Restriction.Restriction.AlwFI-class.html#size,Variable Bio.Restriction.Restriction.AlwI.size=Bio.Restriction.Restriction.AlwI-class.html#size,Variable Bio.Restriction.Restriction.AlwNI.size=Bio.Restriction.Restriction.AlwNI-class.html#size,Variable Bio.Restriction.Restriction.Ama87I.size=Bio.Restriction.Restriction.Ama87I-class.html#size,Variable Bio.Restriction.Restriction.Aor13HI.size=Bio.Restriction.Restriction.Aor13HI-class.html#size,Variable Bio.Restriction.Restriction.Aor51HI.size=Bio.Restriction.Restriction.Aor51HI-class.html#size,Variable Bio.Restriction.Restriction.AoxI.size=Bio.Restriction.Restriction.AoxI-class.html#size,Variable Bio.Restriction.Restriction.ApaBI.size=Bio.Restriction.Restriction.ApaBI-class.html#size,Variable Bio.Restriction.Restriction.ApaI.size=Bio.Restriction.Restriction.ApaI-class.html#size,Variable Bio.Restriction.Restriction.ApaLI.size=Bio.Restriction.Restriction.ApaLI-class.html#size,Variable Bio.Restriction.Restriction.ApeKI.size=Bio.Restriction.Restriction.ApeKI-class.html#size,Variable Bio.Restriction.Restriction.ApoI.size=Bio.Restriction.Restriction.ApoI-class.html#size,Variable Bio.Restriction.Restriction.ApyPI.size=Bio.Restriction.Restriction.ApyPI-class.html#size,Variable Bio.Restriction.Restriction.AquII.size=Bio.Restriction.Restriction.AquII-class.html#size,Variable Bio.Restriction.Restriction.AquIII.size=Bio.Restriction.Restriction.AquIII-class.html#size,Variable Bio.Restriction.Restriction.AquIV.size=Bio.Restriction.Restriction.AquIV-class.html#size,Variable Bio.Restriction.Restriction.ArsI.size=Bio.Restriction.Restriction.ArsI-class.html#size,Variable Bio.Restriction.Restriction.AscI.size=Bio.Restriction.Restriction.AscI-class.html#size,Variable Bio.Restriction.Restriction.AseI.size=Bio.Restriction.Restriction.AseI-class.html#size,Variable Bio.Restriction.Restriction.Asi256I.size=Bio.Restriction.Restriction.Asi256I-class.html#size,Variable Bio.Restriction.Restriction.AsiGI.size=Bio.Restriction.Restriction.AsiGI-class.html#size,Variable Bio.Restriction.Restriction.AsiSI.size=Bio.Restriction.Restriction.AsiSI-class.html#size,Variable Bio.Restriction.Restriction.Asp700I.size=Bio.Restriction.Restriction.Asp700I-class.html#size,Variable Bio.Restriction.Restriction.Asp718I.size=Bio.Restriction.Restriction.Asp718I-class.html#size,Variable Bio.Restriction.Restriction.AspA2I.size=Bio.Restriction.Restriction.AspA2I-class.html#size,Variable Bio.Restriction.Restriction.AspBHI.size=Bio.Restriction.Restriction.AspBHI-class.html#size,Variable Bio.Restriction.Restriction.AspLEI.size=Bio.Restriction.Restriction.AspLEI-class.html#size,Variable Bio.Restriction.Restriction.AspS9I.size=Bio.Restriction.Restriction.AspS9I-class.html#size,Variable Bio.Restriction.Restriction.AssI.size=Bio.Restriction.Restriction.AssI-class.html#size,Variable Bio.Restriction.Restriction.AsuC2I.size=Bio.Restriction.Restriction.AsuC2I-class.html#size,Variable Bio.Restriction.Restriction.AsuHPI.size=Bio.Restriction.Restriction.AsuHPI-class.html#size,Variable Bio.Restriction.Restriction.AsuI.size=Bio.Restriction.Restriction.AsuI-class.html#size,Variable Bio.Restriction.Restriction.AsuII.size=Bio.Restriction.Restriction.AsuII-class.html#size,Variable Bio.Restriction.Restriction.AsuNHI.size=Bio.Restriction.Restriction.AsuNHI-class.html#size,Variable Bio.Restriction.Restriction.AvaI.size=Bio.Restriction.Restriction.AvaI-class.html#size,Variable Bio.Restriction.Restriction.AvaII.size=Bio.Restriction.Restriction.AvaII-class.html#size,Variable Bio.Restriction.Restriction.AvaIII.size=Bio.Restriction.Restriction.AvaIII-class.html#size,Variable Bio.Restriction.Restriction.AvrII.size=Bio.Restriction.Restriction.AvrII-class.html#size,Variable Bio.Restriction.Restriction.AxyI.size=Bio.Restriction.Restriction.AxyI-class.html#size,Variable Bio.Restriction.Restriction.BaeGI.size=Bio.Restriction.Restriction.BaeGI-class.html#size,Variable Bio.Restriction.Restriction.BaeI.size=Bio.Restriction.Restriction.BaeI-class.html#size,Variable Bio.Restriction.Restriction.BalI.size=Bio.Restriction.Restriction.BalI-class.html#size,Variable Bio.Restriction.Restriction.BamHI.size=Bio.Restriction.Restriction.BamHI-class.html#size,Variable Bio.Restriction.Restriction.BanI.size=Bio.Restriction.Restriction.BanI-class.html#size,Variable Bio.Restriction.Restriction.BanII.size=Bio.Restriction.Restriction.BanII-class.html#size,Variable Bio.Restriction.Restriction.BarI.size=Bio.Restriction.Restriction.BarI-class.html#size,Variable Bio.Restriction.Restriction.BasI.size=Bio.Restriction.Restriction.BasI-class.html#size,Variable Bio.Restriction.Restriction.BauI.size=Bio.Restriction.Restriction.BauI-class.html#size,Variable Bio.Restriction.Restriction.Bbr7I.size=Bio.Restriction.Restriction.Bbr7I-class.html#size,Variable Bio.Restriction.Restriction.BbrPI.size=Bio.Restriction.Restriction.BbrPI-class.html#size,Variable Bio.Restriction.Restriction.BbsI.size=Bio.Restriction.Restriction.BbsI-class.html#size,Variable Bio.Restriction.Restriction.Bbv12I.size=Bio.Restriction.Restriction.Bbv12I-class.html#size,Variable Bio.Restriction.Restriction.BbvCI.size=Bio.Restriction.Restriction.BbvCI-class.html#size,Variable Bio.Restriction.Restriction.BbvI.size=Bio.Restriction.Restriction.BbvI-class.html#size,Variable Bio.Restriction.Restriction.BbvII.size=Bio.Restriction.Restriction.BbvII-class.html#size,Variable Bio.Restriction.Restriction.BccI.size=Bio.Restriction.Restriction.BccI-class.html#size,Variable Bio.Restriction.Restriction.Bce83I.size=Bio.Restriction.Restriction.Bce83I-class.html#size,Variable Bio.Restriction.Restriction.BceAI.size=Bio.Restriction.Restriction.BceAI-class.html#size,Variable Bio.Restriction.Restriction.BcefI.size=Bio.Restriction.Restriction.BcefI-class.html#size,Variable Bio.Restriction.Restriction.BcgI.size=Bio.Restriction.Restriction.BcgI-class.html#size,Variable Bio.Restriction.Restriction.BciT130I.size=Bio.Restriction.Restriction.BciT130I-class.html#size,Variable Bio.Restriction.Restriction.BciVI.size=Bio.Restriction.Restriction.BciVI-class.html#size,Variable Bio.Restriction.Restriction.BclI.size=Bio.Restriction.Restriction.BclI-class.html#size,Variable Bio.Restriction.Restriction.BcnI.size=Bio.Restriction.Restriction.BcnI-class.html#size,Variable Bio.Restriction.Restriction.BcoDI.size=Bio.Restriction.Restriction.BcoDI-class.html#size,Variable Bio.Restriction.Restriction.BcuI.size=Bio.Restriction.Restriction.BcuI-class.html#size,Variable Bio.Restriction.Restriction.BdaI.size=Bio.Restriction.Restriction.BdaI-class.html#size,Variable Bio.Restriction.Restriction.BetI.size=Bio.Restriction.Restriction.BetI-class.html#size,Variable Bio.Restriction.Restriction.BfaI.size=Bio.Restriction.Restriction.BfaI-class.html#size,Variable Bio.Restriction.Restriction.BfiI.size=Bio.Restriction.Restriction.BfiI-class.html#size,Variable Bio.Restriction.Restriction.BfmI.size=Bio.Restriction.Restriction.BfmI-class.html#size,Variable Bio.Restriction.Restriction.BfoI.size=Bio.Restriction.Restriction.BfoI-class.html#size,Variable Bio.Restriction.Restriction.BfrI.size=Bio.Restriction.Restriction.BfrI-class.html#size,Variable Bio.Restriction.Restriction.BfuAI.size=Bio.Restriction.Restriction.BfuAI-class.html#size,Variable Bio.Restriction.Restriction.BfuCI.size=Bio.Restriction.Restriction.BfuCI-class.html#size,Variable Bio.Restriction.Restriction.BfuI.size=Bio.Restriction.Restriction.BfuI-class.html#size,Variable Bio.Restriction.Restriction.BglI.size=Bio.Restriction.Restriction.BglI-class.html#size,Variable Bio.Restriction.Restriction.BglII.size=Bio.Restriction.Restriction.BglII-class.html#size,Variable Bio.Restriction.Restriction.BinI.size=Bio.Restriction.Restriction.BinI-class.html#size,Variable Bio.Restriction.Restriction.BisI.size=Bio.Restriction.Restriction.BisI-class.html#size,Variable Bio.Restriction.Restriction.BlnI.size=Bio.Restriction.Restriction.BlnI-class.html#size,Variable Bio.Restriction.Restriction.BlpI.size=Bio.Restriction.Restriction.BlpI-class.html#size,Variable Bio.Restriction.Restriction.BlsI.size=Bio.Restriction.Restriction.BlsI-class.html#size,Variable Bio.Restriction.Restriction.BmcAI.size=Bio.Restriction.Restriction.BmcAI-class.html#size,Variable Bio.Restriction.Restriction.Bme1390I.size=Bio.Restriction.Restriction.Bme1390I-class.html#size,Variable Bio.Restriction.Restriction.Bme18I.size=Bio.Restriction.Restriction.Bme18I-class.html#size,Variable Bio.Restriction.Restriction.BmeDI.size=Bio.Restriction.Restriction.BmeDI-class.html#size,Variable Bio.Restriction.Restriction.BmeRI.size=Bio.Restriction.Restriction.BmeRI-class.html#size,Variable Bio.Restriction.Restriction.BmeT110I.size=Bio.Restriction.Restriction.BmeT110I-class.html#size,Variable Bio.Restriction.Restriction.BmgBI.size=Bio.Restriction.Restriction.BmgBI-class.html#size,Variable Bio.Restriction.Restriction.BmgI.size=Bio.Restriction.Restriction.BmgI-class.html#size,Variable Bio.Restriction.Restriction.BmgT120I.size=Bio.Restriction.Restriction.BmgT120I-class.html#size,Variable Bio.Restriction.Restriction.BmiI.size=Bio.Restriction.Restriction.BmiI-class.html#size,Variable Bio.Restriction.Restriction.BmrFI.size=Bio.Restriction.Restriction.BmrFI-class.html#size,Variable Bio.Restriction.Restriction.BmrI.size=Bio.Restriction.Restriction.BmrI-class.html#size,Variable Bio.Restriction.Restriction.BmsI.size=Bio.Restriction.Restriction.BmsI-class.html#size,Variable Bio.Restriction.Restriction.BmtI.size=Bio.Restriction.Restriction.BmtI-class.html#size,Variable Bio.Restriction.Restriction.BmuI.size=Bio.Restriction.Restriction.BmuI-class.html#size,Variable Bio.Restriction.Restriction.BoxI.size=Bio.Restriction.Restriction.BoxI-class.html#size,Variable Bio.Restriction.Restriction.BpiI.size=Bio.Restriction.Restriction.BpiI-class.html#size,Variable Bio.Restriction.Restriction.BplI.size=Bio.Restriction.Restriction.BplI-class.html#size,Variable Bio.Restriction.Restriction.BpmI.size=Bio.Restriction.Restriction.BpmI-class.html#size,Variable Bio.Restriction.Restriction.Bpu10I.size=Bio.Restriction.Restriction.Bpu10I-class.html#size,Variable Bio.Restriction.Restriction.Bpu1102I.size=Bio.Restriction.Restriction.Bpu1102I-class.html#size,Variable Bio.Restriction.Restriction.Bpu14I.size=Bio.Restriction.Restriction.Bpu14I-class.html#size,Variable Bio.Restriction.Restriction.BpuEI.size=Bio.Restriction.Restriction.BpuEI-class.html#size,Variable Bio.Restriction.Restriction.BpuMI.size=Bio.Restriction.Restriction.BpuMI-class.html#size,Variable Bio.Restriction.Restriction.BpvUI.size=Bio.Restriction.Restriction.BpvUI-class.html#size,Variable Bio.Restriction.Restriction.Bsa29I.size=Bio.Restriction.Restriction.Bsa29I-class.html#size,Variable Bio.Restriction.Restriction.BsaAI.size=Bio.Restriction.Restriction.BsaAI-class.html#size,Variable Bio.Restriction.Restriction.BsaBI.size=Bio.Restriction.Restriction.BsaBI-class.html#size,Variable Bio.Restriction.Restriction.BsaHI.size=Bio.Restriction.Restriction.BsaHI-class.html#size,Variable Bio.Restriction.Restriction.BsaI.size=Bio.Restriction.Restriction.BsaI-class.html#size,Variable Bio.Restriction.Restriction.BsaJI.size=Bio.Restriction.Restriction.BsaJI-class.html#size,Variable Bio.Restriction.Restriction.BsaWI.size=Bio.Restriction.Restriction.BsaWI-class.html#size,Variable Bio.Restriction.Restriction.BsaXI.size=Bio.Restriction.Restriction.BsaXI-class.html#size,Variable Bio.Restriction.Restriction.BsbI.size=Bio.Restriction.Restriction.BsbI-class.html#size,Variable Bio.Restriction.Restriction.Bsc4I.size=Bio.Restriction.Restriction.Bsc4I-class.html#size,Variable Bio.Restriction.Restriction.BscAI.size=Bio.Restriction.Restriction.BscAI-class.html#size,Variable Bio.Restriction.Restriction.BscGI.size=Bio.Restriction.Restriction.BscGI-class.html#size,Variable Bio.Restriction.Restriction.Bse118I.size=Bio.Restriction.Restriction.Bse118I-class.html#size,Variable Bio.Restriction.Restriction.Bse1I.size=Bio.Restriction.Restriction.Bse1I-class.html#size,Variable Bio.Restriction.Restriction.Bse21I.size=Bio.Restriction.Restriction.Bse21I-class.html#size,Variable Bio.Restriction.Restriction.Bse3DI.size=Bio.Restriction.Restriction.Bse3DI-class.html#size,Variable Bio.Restriction.Restriction.Bse8I.size=Bio.Restriction.Restriction.Bse8I-class.html#size,Variable Bio.Restriction.Restriction.BseAI.size=Bio.Restriction.Restriction.BseAI-class.html#size,Variable Bio.Restriction.Restriction.BseBI.size=Bio.Restriction.Restriction.BseBI-class.html#size,Variable Bio.Restriction.Restriction.BseCI.size=Bio.Restriction.Restriction.BseCI-class.html#size,Variable Bio.Restriction.Restriction.BseDI.size=Bio.Restriction.Restriction.BseDI-class.html#size,Variable Bio.Restriction.Restriction.BseGI.size=Bio.Restriction.Restriction.BseGI-class.html#size,Variable Bio.Restriction.Restriction.BseJI.size=Bio.Restriction.Restriction.BseJI-class.html#size,Variable Bio.Restriction.Restriction.BseLI.size=Bio.Restriction.Restriction.BseLI-class.html#size,Variable Bio.Restriction.Restriction.BseMI.size=Bio.Restriction.Restriction.BseMI-class.html#size,Variable Bio.Restriction.Restriction.BseMII.size=Bio.Restriction.Restriction.BseMII-class.html#size,Variable Bio.Restriction.Restriction.BseNI.size=Bio.Restriction.Restriction.BseNI-class.html#size,Variable Bio.Restriction.Restriction.BsePI.size=Bio.Restriction.Restriction.BsePI-class.html#size,Variable Bio.Restriction.Restriction.BseRI.size=Bio.Restriction.Restriction.BseRI-class.html#size,Variable Bio.Restriction.Restriction.BseSI.size=Bio.Restriction.Restriction.BseSI-class.html#size,Variable Bio.Restriction.Restriction.BseX3I.size=Bio.Restriction.Restriction.BseX3I-class.html#size,Variable Bio.Restriction.Restriction.BseXI.size=Bio.Restriction.Restriction.BseXI-class.html#size,Variable Bio.Restriction.Restriction.BseYI.size=Bio.Restriction.Restriction.BseYI-class.html#size,Variable Bio.Restriction.Restriction.BsgI.size=Bio.Restriction.Restriction.BsgI-class.html#size,Variable Bio.Restriction.Restriction.Bsh1236I.size=Bio.Restriction.Restriction.Bsh1236I-class.html#size,Variable Bio.Restriction.Restriction.Bsh1285I.size=Bio.Restriction.Restriction.Bsh1285I-class.html#size,Variable Bio.Restriction.Restriction.BshFI.size=Bio.Restriction.Restriction.BshFI-class.html#size,Variable Bio.Restriction.Restriction.BshNI.size=Bio.Restriction.Restriction.BshNI-class.html#size,Variable Bio.Restriction.Restriction.BshTI.size=Bio.Restriction.Restriction.BshTI-class.html#size,Variable Bio.Restriction.Restriction.BshVI.size=Bio.Restriction.Restriction.BshVI-class.html#size,Variable Bio.Restriction.Restriction.BsiEI.size=Bio.Restriction.Restriction.BsiEI-class.html#size,Variable Bio.Restriction.Restriction.BsiHKAI.size=Bio.Restriction.Restriction.BsiHKAI-class.html#size,Variable Bio.Restriction.Restriction.BsiHKCI.size=Bio.Restriction.Restriction.BsiHKCI-class.html#size,Variable Bio.Restriction.Restriction.BsiI.size=Bio.Restriction.Restriction.BsiI-class.html#size,Variable Bio.Restriction.Restriction.BsiSI.size=Bio.Restriction.Restriction.BsiSI-class.html#size,Variable Bio.Restriction.Restriction.BsiWI.size=Bio.Restriction.Restriction.BsiWI-class.html#size,Variable Bio.Restriction.Restriction.BsiYI.size=Bio.Restriction.Restriction.BsiYI-class.html#size,Variable Bio.Restriction.Restriction.BslFI.size=Bio.Restriction.Restriction.BslFI-class.html#size,Variable Bio.Restriction.Restriction.BslI.size=Bio.Restriction.Restriction.BslI-class.html#size,Variable Bio.Restriction.Restriction.BsmAI.size=Bio.Restriction.Restriction.BsmAI-class.html#size,Variable Bio.Restriction.Restriction.BsmBI.size=Bio.Restriction.Restriction.BsmBI-class.html#size,Variable Bio.Restriction.Restriction.BsmFI.size=Bio.Restriction.Restriction.BsmFI-class.html#size,Variable Bio.Restriction.Restriction.BsmI.size=Bio.Restriction.Restriction.BsmI-class.html#size,Variable Bio.Restriction.Restriction.BsnI.size=Bio.Restriction.Restriction.BsnI-class.html#size,Variable Bio.Restriction.Restriction.Bso31I.size=Bio.Restriction.Restriction.Bso31I-class.html#size,Variable Bio.Restriction.Restriction.BsoBI.size=Bio.Restriction.Restriction.BsoBI-class.html#size,Variable Bio.Restriction.Restriction.Bsp119I.size=Bio.Restriction.Restriction.Bsp119I-class.html#size,Variable Bio.Restriction.Restriction.Bsp120I.size=Bio.Restriction.Restriction.Bsp120I-class.html#size,Variable Bio.Restriction.Restriction.Bsp1286I.size=Bio.Restriction.Restriction.Bsp1286I-class.html#size,Variable Bio.Restriction.Restriction.Bsp13I.size=Bio.Restriction.Restriction.Bsp13I-class.html#size,Variable Bio.Restriction.Restriction.Bsp1407I.size=Bio.Restriction.Restriction.Bsp1407I-class.html#size,Variable Bio.Restriction.Restriction.Bsp143I.size=Bio.Restriction.Restriction.Bsp143I-class.html#size,Variable Bio.Restriction.Restriction.Bsp1720I.size=Bio.Restriction.Restriction.Bsp1720I-class.html#size,Variable Bio.Restriction.Restriction.Bsp19I.size=Bio.Restriction.Restriction.Bsp19I-class.html#size,Variable Bio.Restriction.Restriction.Bsp24I.size=Bio.Restriction.Restriction.Bsp24I-class.html#size,Variable Bio.Restriction.Restriction.Bsp68I.size=Bio.Restriction.Restriction.Bsp68I-class.html#size,Variable Bio.Restriction.Restriction.BspACI.size=Bio.Restriction.Restriction.BspACI-class.html#size,Variable Bio.Restriction.Restriction.BspCNI.size=Bio.Restriction.Restriction.BspCNI-class.html#size,Variable Bio.Restriction.Restriction.BspD6I.size=Bio.Restriction.Restriction.BspD6I-class.html#size,Variable Bio.Restriction.Restriction.BspDI.size=Bio.Restriction.Restriction.BspDI-class.html#size,Variable Bio.Restriction.Restriction.BspEI.size=Bio.Restriction.Restriction.BspEI-class.html#size,Variable Bio.Restriction.Restriction.BspFNI.size=Bio.Restriction.Restriction.BspFNI-class.html#size,Variable Bio.Restriction.Restriction.BspGI.size=Bio.Restriction.Restriction.BspGI-class.html#size,Variable Bio.Restriction.Restriction.BspHI.size=Bio.Restriction.Restriction.BspHI-class.html#size,Variable Bio.Restriction.Restriction.BspLI.size=Bio.Restriction.Restriction.BspLI-class.html#size,Variable Bio.Restriction.Restriction.BspLU11I.size=Bio.Restriction.Restriction.BspLU11I-class.html#size,Variable Bio.Restriction.Restriction.BspMI.size=Bio.Restriction.Restriction.BspMI-class.html#size,Variable Bio.Restriction.Restriction.BspMII.size=Bio.Restriction.Restriction.BspMII-class.html#size,Variable Bio.Restriction.Restriction.BspNCI.size=Bio.Restriction.Restriction.BspNCI-class.html#size,Variable Bio.Restriction.Restriction.BspOI.size=Bio.Restriction.Restriction.BspOI-class.html#size,Variable Bio.Restriction.Restriction.BspPI.size=Bio.Restriction.Restriction.BspPI-class.html#size,Variable Bio.Restriction.Restriction.BspQI.size=Bio.Restriction.Restriction.BspQI-class.html#size,Variable Bio.Restriction.Restriction.BspT104I.size=Bio.Restriction.Restriction.BspT104I-class.html#size,Variable Bio.Restriction.Restriction.BspT107I.size=Bio.Restriction.Restriction.BspT107I-class.html#size,Variable Bio.Restriction.Restriction.BspTI.size=Bio.Restriction.Restriction.BspTI-class.html#size,Variable Bio.Restriction.Restriction.BsrBI.size=Bio.Restriction.Restriction.BsrBI-class.html#size,Variable Bio.Restriction.Restriction.BsrDI.size=Bio.Restriction.Restriction.BsrDI-class.html#size,Variable Bio.Restriction.Restriction.BsrFI.size=Bio.Restriction.Restriction.BsrFI-class.html#size,Variable Bio.Restriction.Restriction.BsrGI.size=Bio.Restriction.Restriction.BsrGI-class.html#size,Variable Bio.Restriction.Restriction.BsrI.size=Bio.Restriction.Restriction.BsrI-class.html#size,Variable Bio.Restriction.Restriction.BsrSI.size=Bio.Restriction.Restriction.BsrSI-class.html#size,Variable Bio.Restriction.Restriction.BssAI.size=Bio.Restriction.Restriction.BssAI-class.html#size,Variable Bio.Restriction.Restriction.BssECI.size=Bio.Restriction.Restriction.BssECI-class.html#size,Variable Bio.Restriction.Restriction.BssHII.size=Bio.Restriction.Restriction.BssHII-class.html#size,Variable Bio.Restriction.Restriction.BssKI.size=Bio.Restriction.Restriction.BssKI-class.html#size,Variable Bio.Restriction.Restriction.BssMI.size=Bio.Restriction.Restriction.BssMI-class.html#size,Variable Bio.Restriction.Restriction.BssNAI.size=Bio.Restriction.Restriction.BssNAI-class.html#size,Variable Bio.Restriction.Restriction.BssNI.size=Bio.Restriction.Restriction.BssNI-class.html#size,Variable Bio.Restriction.Restriction.BssSI.size=Bio.Restriction.Restriction.BssSI-class.html#size,Variable Bio.Restriction.Restriction.BssT1I.size=Bio.Restriction.Restriction.BssT1I-class.html#size,Variable Bio.Restriction.Restriction.Bst1107I.size=Bio.Restriction.Restriction.Bst1107I-class.html#size,Variable Bio.Restriction.Restriction.Bst2BI.size=Bio.Restriction.Restriction.Bst2BI-class.html#size,Variable Bio.Restriction.Restriction.Bst2UI.size=Bio.Restriction.Restriction.Bst2UI-class.html#size,Variable Bio.Restriction.Restriction.Bst4CI.size=Bio.Restriction.Restriction.Bst4CI-class.html#size,Variable Bio.Restriction.Restriction.Bst6I.size=Bio.Restriction.Restriction.Bst6I-class.html#size,Variable Bio.Restriction.Restriction.BstACI.size=Bio.Restriction.Restriction.BstACI-class.html#size,Variable Bio.Restriction.Restriction.BstAFI.size=Bio.Restriction.Restriction.BstAFI-class.html#size,Variable Bio.Restriction.Restriction.BstAPI.size=Bio.Restriction.Restriction.BstAPI-class.html#size,Variable Bio.Restriction.Restriction.BstAUI.size=Bio.Restriction.Restriction.BstAUI-class.html#size,Variable Bio.Restriction.Restriction.BstBAI.size=Bio.Restriction.Restriction.BstBAI-class.html#size,Variable Bio.Restriction.Restriction.BstBI.size=Bio.Restriction.Restriction.BstBI-class.html#size,Variable Bio.Restriction.Restriction.BstC8I.size=Bio.Restriction.Restriction.BstC8I-class.html#size,Variable Bio.Restriction.Restriction.BstDEI.size=Bio.Restriction.Restriction.BstDEI-class.html#size,Variable Bio.Restriction.Restriction.BstDSI.size=Bio.Restriction.Restriction.BstDSI-class.html#size,Variable Bio.Restriction.Restriction.BstEII.size=Bio.Restriction.Restriction.BstEII-class.html#size,Variable Bio.Restriction.Restriction.BstENI.size=Bio.Restriction.Restriction.BstENI-class.html#size,Variable Bio.Restriction.Restriction.BstF5I.size=Bio.Restriction.Restriction.BstF5I-class.html#size,Variable Bio.Restriction.Restriction.BstFNI.size=Bio.Restriction.Restriction.BstFNI-class.html#size,Variable Bio.Restriction.Restriction.BstH2I.size=Bio.Restriction.Restriction.BstH2I-class.html#size,Variable Bio.Restriction.Restriction.BstHHI.size=Bio.Restriction.Restriction.BstHHI-class.html#size,Variable Bio.Restriction.Restriction.BstKTI.size=Bio.Restriction.Restriction.BstKTI-class.html#size,Variable Bio.Restriction.Restriction.BstMAI.size=Bio.Restriction.Restriction.BstMAI-class.html#size,Variable Bio.Restriction.Restriction.BstMBI.size=Bio.Restriction.Restriction.BstMBI-class.html#size,Variable Bio.Restriction.Restriction.BstMCI.size=Bio.Restriction.Restriction.BstMCI-class.html#size,Variable Bio.Restriction.Restriction.BstMWI.size=Bio.Restriction.Restriction.BstMWI-class.html#size,Variable Bio.Restriction.Restriction.BstNI.size=Bio.Restriction.Restriction.BstNI-class.html#size,Variable Bio.Restriction.Restriction.BstNSI.size=Bio.Restriction.Restriction.BstNSI-class.html#size,Variable Bio.Restriction.Restriction.BstOI.size=Bio.Restriction.Restriction.BstOI-class.html#size,Variable Bio.Restriction.Restriction.BstPAI.size=Bio.Restriction.Restriction.BstPAI-class.html#size,Variable Bio.Restriction.Restriction.BstPI.size=Bio.Restriction.Restriction.BstPI-class.html#size,Variable Bio.Restriction.Restriction.BstSCI.size=Bio.Restriction.Restriction.BstSCI-class.html#size,Variable Bio.Restriction.Restriction.BstSFI.size=Bio.Restriction.Restriction.BstSFI-class.html#size,Variable Bio.Restriction.Restriction.BstSLI.size=Bio.Restriction.Restriction.BstSLI-class.html#size,Variable Bio.Restriction.Restriction.BstSNI.size=Bio.Restriction.Restriction.BstSNI-class.html#size,Variable Bio.Restriction.Restriction.BstUI.size=Bio.Restriction.Restriction.BstUI-class.html#size,Variable Bio.Restriction.Restriction.BstV1I.size=Bio.Restriction.Restriction.BstV1I-class.html#size,Variable Bio.Restriction.Restriction.BstV2I.size=Bio.Restriction.Restriction.BstV2I-class.html#size,Variable Bio.Restriction.Restriction.BstX2I.size=Bio.Restriction.Restriction.BstX2I-class.html#size,Variable Bio.Restriction.Restriction.BstXI.size=Bio.Restriction.Restriction.BstXI-class.html#size,Variable Bio.Restriction.Restriction.BstYI.size=Bio.Restriction.Restriction.BstYI-class.html#size,Variable Bio.Restriction.Restriction.BstZ17I.size=Bio.Restriction.Restriction.BstZ17I-class.html#size,Variable Bio.Restriction.Restriction.BstZI.size=Bio.Restriction.Restriction.BstZI-class.html#size,Variable Bio.Restriction.Restriction.Bsu15I.size=Bio.Restriction.Restriction.Bsu15I-class.html#size,Variable Bio.Restriction.Restriction.Bsu36I.size=Bio.Restriction.Restriction.Bsu36I-class.html#size,Variable Bio.Restriction.Restriction.BsuI.size=Bio.Restriction.Restriction.BsuI-class.html#size,Variable Bio.Restriction.Restriction.BsuRI.size=Bio.Restriction.Restriction.BsuRI-class.html#size,Variable Bio.Restriction.Restriction.BtgI.size=Bio.Restriction.Restriction.BtgI-class.html#size,Variable Bio.Restriction.Restriction.BtgZI.size=Bio.Restriction.Restriction.BtgZI-class.html#size,Variable Bio.Restriction.Restriction.BthCI.size=Bio.Restriction.Restriction.BthCI-class.html#size,Variable Bio.Restriction.Restriction.BtrI.size=Bio.Restriction.Restriction.BtrI-class.html#size,Variable Bio.Restriction.Restriction.BtsCI.size=Bio.Restriction.Restriction.BtsCI-class.html#size,Variable Bio.Restriction.Restriction.BtsI.size=Bio.Restriction.Restriction.BtsI-class.html#size,Variable Bio.Restriction.Restriction.BtsIMutI.size=Bio.Restriction.Restriction.BtsIMutI-class.html#size,Variable Bio.Restriction.Restriction.BtuMI.size=Bio.Restriction.Restriction.BtuMI-class.html#size,Variable Bio.Restriction.Restriction.BveI.size=Bio.Restriction.Restriction.BveI-class.html#size,Variable Bio.Restriction.Restriction.Cac8I.size=Bio.Restriction.Restriction.Cac8I-class.html#size,Variable Bio.Restriction.Restriction.CaiI.size=Bio.Restriction.Restriction.CaiI-class.html#size,Variable Bio.Restriction.Restriction.CauII.size=Bio.Restriction.Restriction.CauII-class.html#size,Variable Bio.Restriction.Restriction.CchII.size=Bio.Restriction.Restriction.CchII-class.html#size,Variable Bio.Restriction.Restriction.CchIII.size=Bio.Restriction.Restriction.CchIII-class.html#size,Variable Bio.Restriction.Restriction.CciI.size=Bio.Restriction.Restriction.CciI-class.html#size,Variable Bio.Restriction.Restriction.CciNI.size=Bio.Restriction.Restriction.CciNI-class.html#size,Variable Bio.Restriction.Restriction.Cdi630V.size=Bio.Restriction.Restriction.Cdi630V-class.html#size,Variable Bio.Restriction.Restriction.CdiI.size=Bio.Restriction.Restriction.CdiI-class.html#size,Variable Bio.Restriction.Restriction.CdpI.size=Bio.Restriction.Restriction.CdpI-class.html#size,Variable Bio.Restriction.Restriction.CfoI.size=Bio.Restriction.Restriction.CfoI-class.html#size,Variable Bio.Restriction.Restriction.Cfr10I.size=Bio.Restriction.Restriction.Cfr10I-class.html#size,Variable Bio.Restriction.Restriction.Cfr13I.size=Bio.Restriction.Restriction.Cfr13I-class.html#size,Variable Bio.Restriction.Restriction.Cfr42I.size=Bio.Restriction.Restriction.Cfr42I-class.html#size,Variable Bio.Restriction.Restriction.Cfr9I.size=Bio.Restriction.Restriction.Cfr9I-class.html#size,Variable Bio.Restriction.Restriction.CfrI.size=Bio.Restriction.Restriction.CfrI-class.html#size,Variable Bio.Restriction.Restriction.Cgl13032I.size=Bio.Restriction.Restriction.Cgl13032I-class.html#size,Variable Bio.Restriction.Restriction.Cgl13032II.size=Bio.Restriction.Restriction.Cgl13032II-class.html#size,Variable Bio.Restriction.Restriction.ChaI.size=Bio.Restriction.Restriction.ChaI-class.html#size,Variable Bio.Restriction.Restriction.CjeFIII.size=Bio.Restriction.Restriction.CjeFIII-class.html#size,Variable Bio.Restriction.Restriction.CjeFV.size=Bio.Restriction.Restriction.CjeFV-class.html#size,Variable Bio.Restriction.Restriction.CjeI.size=Bio.Restriction.Restriction.CjeI-class.html#size,Variable Bio.Restriction.Restriction.CjeNII.size=Bio.Restriction.Restriction.CjeNII-class.html#size,Variable Bio.Restriction.Restriction.CjeNIII.size=Bio.Restriction.Restriction.CjeNIII-class.html#size,Variable Bio.Restriction.Restriction.CjeP659IV.size=Bio.Restriction.Restriction.CjeP659IV-class.html#size,Variable Bio.Restriction.Restriction.CjePI.size=Bio.Restriction.Restriction.CjePI-class.html#size,Variable Bio.Restriction.Restriction.CjuI.size=Bio.Restriction.Restriction.CjuI-class.html#size,Variable Bio.Restriction.Restriction.CjuII.size=Bio.Restriction.Restriction.CjuII-class.html#size,Variable Bio.Restriction.Restriction.ClaI.size=Bio.Restriction.Restriction.ClaI-class.html#size,Variable Bio.Restriction.Restriction.CpoI.size=Bio.Restriction.Restriction.CpoI-class.html#size,Variable Bio.Restriction.Restriction.CseI.size=Bio.Restriction.Restriction.CseI-class.html#size,Variable Bio.Restriction.Restriction.CsiI.size=Bio.Restriction.Restriction.CsiI-class.html#size,Variable Bio.Restriction.Restriction.Csp6I.size=Bio.Restriction.Restriction.Csp6I-class.html#size,Variable Bio.Restriction.Restriction.CspAI.size=Bio.Restriction.Restriction.CspAI-class.html#size,Variable Bio.Restriction.Restriction.CspCI.size=Bio.Restriction.Restriction.CspCI-class.html#size,Variable Bio.Restriction.Restriction.CspI.size=Bio.Restriction.Restriction.CspI-class.html#size,Variable Bio.Restriction.Restriction.CstMI.size=Bio.Restriction.Restriction.CstMI-class.html#size,Variable Bio.Restriction.Restriction.CviAII.size=Bio.Restriction.Restriction.CviAII-class.html#size,Variable Bio.Restriction.Restriction.CviJI.size=Bio.Restriction.Restriction.CviJI-class.html#size,Variable Bio.Restriction.Restriction.CviKI_1.size=Bio.Restriction.Restriction.CviKI_1-class.html#size,Variable Bio.Restriction.Restriction.CviQI.size=Bio.Restriction.Restriction.CviQI-class.html#size,Variable Bio.Restriction.Restriction.CviRI.size=Bio.Restriction.Restriction.CviRI-class.html#size,Variable Bio.Restriction.Restriction.DdeI.size=Bio.Restriction.Restriction.DdeI-class.html#size,Variable Bio.Restriction.Restriction.DinI.size=Bio.Restriction.Restriction.DinI-class.html#size,Variable Bio.Restriction.Restriction.DpnI.size=Bio.Restriction.Restriction.DpnI-class.html#size,Variable Bio.Restriction.Restriction.DpnII.size=Bio.Restriction.Restriction.DpnII-class.html#size,Variable Bio.Restriction.Restriction.DraI.size=Bio.Restriction.Restriction.DraI-class.html#size,Variable Bio.Restriction.Restriction.DraII.size=Bio.Restriction.Restriction.DraII-class.html#size,Variable Bio.Restriction.Restriction.DraIII.size=Bio.Restriction.Restriction.DraIII-class.html#size,Variable Bio.Restriction.Restriction.DraRI.size=Bio.Restriction.Restriction.DraRI-class.html#size,Variable Bio.Restriction.Restriction.DrdI.size=Bio.Restriction.Restriction.DrdI-class.html#size,Variable Bio.Restriction.Restriction.DrdII.size=Bio.Restriction.Restriction.DrdII-class.html#size,Variable Bio.Restriction.Restriction.DriI.size=Bio.Restriction.Restriction.DriI-class.html#size,Variable Bio.Restriction.Restriction.DsaI.size=Bio.Restriction.Restriction.DsaI-class.html#size,Variable Bio.Restriction.Restriction.DseDI.size=Bio.Restriction.Restriction.DseDI-class.html#size,Variable Bio.Restriction.Restriction.EaeI.size=Bio.Restriction.Restriction.EaeI-class.html#size,Variable Bio.Restriction.Restriction.EagI.size=Bio.Restriction.Restriction.EagI-class.html#size,Variable Bio.Restriction.Restriction.Eam1104I.size=Bio.Restriction.Restriction.Eam1104I-class.html#size,Variable Bio.Restriction.Restriction.Eam1105I.size=Bio.Restriction.Restriction.Eam1105I-class.html#size,Variable Bio.Restriction.Restriction.EarI.size=Bio.Restriction.Restriction.EarI-class.html#size,Variable Bio.Restriction.Restriction.EciI.size=Bio.Restriction.Restriction.EciI-class.html#size,Variable Bio.Restriction.Restriction.Ecl136II.size=Bio.Restriction.Restriction.Ecl136II-class.html#size,Variable Bio.Restriction.Restriction.EclXI.size=Bio.Restriction.Restriction.EclXI-class.html#size,Variable Bio.Restriction.Restriction.Eco105I.size=Bio.Restriction.Restriction.Eco105I-class.html#size,Variable Bio.Restriction.Restriction.Eco130I.size=Bio.Restriction.Restriction.Eco130I-class.html#size,Variable Bio.Restriction.Restriction.Eco147I.size=Bio.Restriction.Restriction.Eco147I-class.html#size,Variable Bio.Restriction.Restriction.Eco24I.size=Bio.Restriction.Restriction.Eco24I-class.html#size,Variable Bio.Restriction.Restriction.Eco31I.size=Bio.Restriction.Restriction.Eco31I-class.html#size,Variable Bio.Restriction.Restriction.Eco32I.size=Bio.Restriction.Restriction.Eco32I-class.html#size,Variable Bio.Restriction.Restriction.Eco47I.size=Bio.Restriction.Restriction.Eco47I-class.html#size,Variable Bio.Restriction.Restriction.Eco47III.size=Bio.Restriction.Restriction.Eco47III-class.html#size,Variable Bio.Restriction.Restriction.Eco52I.size=Bio.Restriction.Restriction.Eco52I-class.html#size,Variable Bio.Restriction.Restriction.Eco53kI.size=Bio.Restriction.Restriction.Eco53kI-class.html#size,Variable Bio.Restriction.Restriction.Eco57I.size=Bio.Restriction.Restriction.Eco57I-class.html#size,Variable Bio.Restriction.Restriction.Eco57MI.size=Bio.Restriction.Restriction.Eco57MI-class.html#size,Variable Bio.Restriction.Restriction.Eco72I.size=Bio.Restriction.Restriction.Eco72I-class.html#size,Variable Bio.Restriction.Restriction.Eco81I.size=Bio.Restriction.Restriction.Eco81I-class.html#size,Variable Bio.Restriction.Restriction.Eco88I.size=Bio.Restriction.Restriction.Eco88I-class.html#size,Variable Bio.Restriction.Restriction.Eco91I.size=Bio.Restriction.Restriction.Eco91I-class.html#size,Variable Bio.Restriction.Restriction.EcoHI.size=Bio.Restriction.Restriction.EcoHI-class.html#size,Variable Bio.Restriction.Restriction.EcoICRI.size=Bio.Restriction.Restriction.EcoICRI-class.html#size,Variable Bio.Restriction.Restriction.EcoNI.size=Bio.Restriction.Restriction.EcoNI-class.html#size,Variable Bio.Restriction.Restriction.EcoO109I.size=Bio.Restriction.Restriction.EcoO109I-class.html#size,Variable Bio.Restriction.Restriction.EcoO65I.size=Bio.Restriction.Restriction.EcoO65I-class.html#size,Variable Bio.Restriction.Restriction.EcoRI.size=Bio.Restriction.Restriction.EcoRI-class.html#size,Variable Bio.Restriction.Restriction.EcoRII.size=Bio.Restriction.Restriction.EcoRII-class.html#size,Variable Bio.Restriction.Restriction.EcoRV.size=Bio.Restriction.Restriction.EcoRV-class.html#size,Variable Bio.Restriction.Restriction.EcoT14I.size=Bio.Restriction.Restriction.EcoT14I-class.html#size,Variable Bio.Restriction.Restriction.EcoT22I.size=Bio.Restriction.Restriction.EcoT22I-class.html#size,Variable Bio.Restriction.Restriction.EcoT38I.size=Bio.Restriction.Restriction.EcoT38I-class.html#size,Variable Bio.Restriction.Restriction.EgeI.size=Bio.Restriction.Restriction.EgeI-class.html#size,Variable Bio.Restriction.Restriction.EheI.size=Bio.Restriction.Restriction.EheI-class.html#size,Variable Bio.Restriction.Restriction.ErhI.size=Bio.Restriction.Restriction.ErhI-class.html#size,Variable Bio.Restriction.Restriction.EsaBC3I.size=Bio.Restriction.Restriction.EsaBC3I-class.html#size,Variable Bio.Restriction.Restriction.EsaSSI.size=Bio.Restriction.Restriction.EsaSSI-class.html#size,Variable Bio.Restriction.Restriction.Esp3I.size=Bio.Restriction.Restriction.Esp3I-class.html#size,Variable Bio.Restriction.Restriction.EspI.size=Bio.Restriction.Restriction.EspI-class.html#size,Variable Bio.Restriction.Restriction.FaeI.size=Bio.Restriction.Restriction.FaeI-class.html#size,Variable Bio.Restriction.Restriction.FaiI.size=Bio.Restriction.Restriction.FaiI-class.html#size,Variable Bio.Restriction.Restriction.FalI.size=Bio.Restriction.Restriction.FalI-class.html#size,Variable Bio.Restriction.Restriction.FaqI.size=Bio.Restriction.Restriction.FaqI-class.html#size,Variable Bio.Restriction.Restriction.FatI.size=Bio.Restriction.Restriction.FatI-class.html#size,Variable Bio.Restriction.Restriction.FauI.size=Bio.Restriction.Restriction.FauI-class.html#size,Variable Bio.Restriction.Restriction.FauNDI.size=Bio.Restriction.Restriction.FauNDI-class.html#size,Variable Bio.Restriction.Restriction.FbaI.size=Bio.Restriction.Restriction.FbaI-class.html#size,Variable Bio.Restriction.Restriction.FblI.size=Bio.Restriction.Restriction.FblI-class.html#size,Variable Bio.Restriction.Restriction.FinI.size=Bio.Restriction.Restriction.FinI-class.html#size,Variable Bio.Restriction.Restriction.FmuI.size=Bio.Restriction.Restriction.FmuI-class.html#size,Variable Bio.Restriction.Restriction.Fnu4HI.size=Bio.Restriction.Restriction.Fnu4HI-class.html#size,Variable Bio.Restriction.Restriction.FnuDII.size=Bio.Restriction.Restriction.FnuDII-class.html#size,Variable Bio.Restriction.Restriction.FokI.size=Bio.Restriction.Restriction.FokI-class.html#size,Variable Bio.Restriction.Restriction.FriOI.size=Bio.Restriction.Restriction.FriOI-class.html#size,Variable Bio.Restriction.Restriction.FseI.size=Bio.Restriction.Restriction.FseI-class.html#size,Variable Bio.Restriction.Restriction.Fsp4HI.size=Bio.Restriction.Restriction.Fsp4HI-class.html#size,Variable Bio.Restriction.Restriction.FspAI.size=Bio.Restriction.Restriction.FspAI-class.html#size,Variable Bio.Restriction.Restriction.FspBI.size=Bio.Restriction.Restriction.FspBI-class.html#size,Variable Bio.Restriction.Restriction.FspEI.size=Bio.Restriction.Restriction.FspEI-class.html#size,Variable Bio.Restriction.Restriction.FspI.size=Bio.Restriction.Restriction.FspI-class.html#size,Variable Bio.Restriction.Restriction.GauT27I.size=Bio.Restriction.Restriction.GauT27I-class.html#size,Variable Bio.Restriction.Restriction.GdiII.size=Bio.Restriction.Restriction.GdiII-class.html#size,Variable Bio.Restriction.Restriction.GlaI.size=Bio.Restriction.Restriction.GlaI-class.html#size,Variable Bio.Restriction.Restriction.GluI.size=Bio.Restriction.Restriction.GluI-class.html#size,Variable Bio.Restriction.Restriction.GsaI.size=Bio.Restriction.Restriction.GsaI-class.html#size,Variable Bio.Restriction.Restriction.GsuI.size=Bio.Restriction.Restriction.GsuI-class.html#size,Variable Bio.Restriction.Restriction.HaeI.size=Bio.Restriction.Restriction.HaeI-class.html#size,Variable Bio.Restriction.Restriction.HaeII.size=Bio.Restriction.Restriction.HaeII-class.html#size,Variable Bio.Restriction.Restriction.HaeIII.size=Bio.Restriction.Restriction.HaeIII-class.html#size,Variable Bio.Restriction.Restriction.HapII.size=Bio.Restriction.Restriction.HapII-class.html#size,Variable Bio.Restriction.Restriction.HauII.size=Bio.Restriction.Restriction.HauII-class.html#size,Variable Bio.Restriction.Restriction.HgaI.size=Bio.Restriction.Restriction.HgaI-class.html#size,Variable Bio.Restriction.Restriction.HgiAI.size=Bio.Restriction.Restriction.HgiAI-class.html#size,Variable Bio.Restriction.Restriction.HgiCI.size=Bio.Restriction.Restriction.HgiCI-class.html#size,Variable Bio.Restriction.Restriction.HgiEII.size=Bio.Restriction.Restriction.HgiEII-class.html#size,Variable Bio.Restriction.Restriction.HgiJII.size=Bio.Restriction.Restriction.HgiJII-class.html#size,Variable Bio.Restriction.Restriction.HhaI.size=Bio.Restriction.Restriction.HhaI-class.html#size,Variable Bio.Restriction.Restriction.Hin1I.size=Bio.Restriction.Restriction.Hin1I-class.html#size,Variable Bio.Restriction.Restriction.Hin1II.size=Bio.Restriction.Restriction.Hin1II-class.html#size,Variable Bio.Restriction.Restriction.Hin4I.size=Bio.Restriction.Restriction.Hin4I-class.html#size,Variable Bio.Restriction.Restriction.Hin4II.size=Bio.Restriction.Restriction.Hin4II-class.html#size,Variable Bio.Restriction.Restriction.Hin6I.size=Bio.Restriction.Restriction.Hin6I-class.html#size,Variable Bio.Restriction.Restriction.HinP1I.size=Bio.Restriction.Restriction.HinP1I-class.html#size,Variable Bio.Restriction.Restriction.HincII.size=Bio.Restriction.Restriction.HincII-class.html#size,Variable Bio.Restriction.Restriction.HindII.size=Bio.Restriction.Restriction.HindII-class.html#size,Variable Bio.Restriction.Restriction.HindIII.size=Bio.Restriction.Restriction.HindIII-class.html#size,Variable Bio.Restriction.Restriction.HinfI.size=Bio.Restriction.Restriction.HinfI-class.html#size,Variable Bio.Restriction.Restriction.HpaI.size=Bio.Restriction.Restriction.HpaI-class.html#size,Variable Bio.Restriction.Restriction.HpaII.size=Bio.Restriction.Restriction.HpaII-class.html#size,Variable Bio.Restriction.Restriction.HphI.size=Bio.Restriction.Restriction.HphI-class.html#size,Variable Bio.Restriction.Restriction.Hpy166II.size=Bio.Restriction.Restriction.Hpy166II-class.html#size,Variable Bio.Restriction.Restriction.Hpy178III.size=Bio.Restriction.Restriction.Hpy178III-class.html#size,Variable Bio.Restriction.Restriction.Hpy188I.size=Bio.Restriction.Restriction.Hpy188I-class.html#size,Variable Bio.Restriction.Restriction.Hpy188III.size=Bio.Restriction.Restriction.Hpy188III-class.html#size,Variable Bio.Restriction.Restriction.Hpy8I.size=Bio.Restriction.Restriction.Hpy8I-class.html#size,Variable Bio.Restriction.Restriction.Hpy99I.size=Bio.Restriction.Restriction.Hpy99I-class.html#size,Variable Bio.Restriction.Restriction.Hpy99XIII.size=Bio.Restriction.Restriction.Hpy99XIII-class.html#size,Variable Bio.Restriction.Restriction.Hpy99XIV.size=Bio.Restriction.Restriction.Hpy99XIV-class.html#size,Variable Bio.Restriction.Restriction.HpyAV.size=Bio.Restriction.Restriction.HpyAV-class.html#size,Variable Bio.Restriction.Restriction.HpyCH4III.size=Bio.Restriction.Restriction.HpyCH4III-class.html#size,Variable Bio.Restriction.Restriction.HpyCH4IV.size=Bio.Restriction.Restriction.HpyCH4IV-class.html#size,Variable Bio.Restriction.Restriction.HpyCH4V.size=Bio.Restriction.Restriction.HpyCH4V-class.html#size,Variable Bio.Restriction.Restriction.HpyF10VI.size=Bio.Restriction.Restriction.HpyF10VI-class.html#size,Variable Bio.Restriction.Restriction.HpyF3I.size=Bio.Restriction.Restriction.HpyF3I-class.html#size,Variable Bio.Restriction.Restriction.HpySE526I.size=Bio.Restriction.Restriction.HpySE526I-class.html#size,Variable Bio.Restriction.Restriction.Hsp92I.size=Bio.Restriction.Restriction.Hsp92I-class.html#size,Variable Bio.Restriction.Restriction.Hsp92II.size=Bio.Restriction.Restriction.Hsp92II-class.html#size,Variable Bio.Restriction.Restriction.HspAI.size=Bio.Restriction.Restriction.HspAI-class.html#size,Variable Bio.Restriction.Restriction.Jma19592I.size=Bio.Restriction.Restriction.Jma19592I-class.html#size,Variable Bio.Restriction.Restriction.KasI.size=Bio.Restriction.Restriction.KasI-class.html#size,Variable Bio.Restriction.Restriction.KflI.size=Bio.Restriction.Restriction.KflI-class.html#size,Variable Bio.Restriction.Restriction.Kpn2I.size=Bio.Restriction.Restriction.Kpn2I-class.html#size,Variable Bio.Restriction.Restriction.KpnI.size=Bio.Restriction.Restriction.KpnI-class.html#size,Variable Bio.Restriction.Restriction.KroI.size=Bio.Restriction.Restriction.KroI-class.html#size,Variable Bio.Restriction.Restriction.Ksp22I.size=Bio.Restriction.Restriction.Ksp22I-class.html#size,Variable Bio.Restriction.Restriction.Ksp632I.size=Bio.Restriction.Restriction.Ksp632I-class.html#size,Variable Bio.Restriction.Restriction.KspAI.size=Bio.Restriction.Restriction.KspAI-class.html#size,Variable Bio.Restriction.Restriction.KspI.size=Bio.Restriction.Restriction.KspI-class.html#size,Variable Bio.Restriction.Restriction.Kzo9I.size=Bio.Restriction.Restriction.Kzo9I-class.html#size,Variable Bio.Restriction.Restriction.LguI.size=Bio.Restriction.Restriction.LguI-class.html#size,Variable Bio.Restriction.Restriction.LpnI.size=Bio.Restriction.Restriction.LpnI-class.html#size,Variable Bio.Restriction.Restriction.LpnPI.size=Bio.Restriction.Restriction.LpnPI-class.html#size,Variable Bio.Restriction.Restriction.Lsp1109I.size=Bio.Restriction.Restriction.Lsp1109I-class.html#size,Variable Bio.Restriction.Restriction.LweI.size=Bio.Restriction.Restriction.LweI-class.html#size,Variable Bio.Restriction.Restriction.MabI.size=Bio.Restriction.Restriction.MabI-class.html#size,Variable Bio.Restriction.Restriction.MaeI.size=Bio.Restriction.Restriction.MaeI-class.html#size,Variable Bio.Restriction.Restriction.MaeII.size=Bio.Restriction.Restriction.MaeII-class.html#size,Variable Bio.Restriction.Restriction.MaeIII.size=Bio.Restriction.Restriction.MaeIII-class.html#size,Variable Bio.Restriction.Restriction.MalI.size=Bio.Restriction.Restriction.MalI-class.html#size,Variable Bio.Restriction.Restriction.MaqI.size=Bio.Restriction.Restriction.MaqI-class.html#size,Variable Bio.Restriction.Restriction.MauBI.size=Bio.Restriction.Restriction.MauBI-class.html#size,Variable Bio.Restriction.Restriction.MbiI.size=Bio.Restriction.Restriction.MbiI-class.html#size,Variable Bio.Restriction.Restriction.MboI.size=Bio.Restriction.Restriction.MboI-class.html#size,Variable Bio.Restriction.Restriction.MboII.size=Bio.Restriction.Restriction.MboII-class.html#size,Variable Bio.Restriction.Restriction.McaTI.size=Bio.Restriction.Restriction.McaTI-class.html#size,Variable Bio.Restriction.Restriction.McrI.size=Bio.Restriction.Restriction.McrI-class.html#size,Variable Bio.Restriction.Restriction.MfeI.size=Bio.Restriction.Restriction.MfeI-class.html#size,Variable Bio.Restriction.Restriction.MflI.size=Bio.Restriction.Restriction.MflI-class.html#size,Variable Bio.Restriction.Restriction.MhlI.size=Bio.Restriction.Restriction.MhlI-class.html#size,Variable Bio.Restriction.Restriction.MjaIV.size=Bio.Restriction.Restriction.MjaIV-class.html#size,Variable Bio.Restriction.Restriction.MkaDII.size=Bio.Restriction.Restriction.MkaDII-class.html#size,Variable Bio.Restriction.Restriction.MlsI.size=Bio.Restriction.Restriction.MlsI-class.html#size,Variable Bio.Restriction.Restriction.MluCI.size=Bio.Restriction.Restriction.MluCI-class.html#size,Variable Bio.Restriction.Restriction.MluI.size=Bio.Restriction.Restriction.MluI-class.html#size,Variable Bio.Restriction.Restriction.MluNI.size=Bio.Restriction.Restriction.MluNI-class.html#size,Variable Bio.Restriction.Restriction.Mly113I.size=Bio.Restriction.Restriction.Mly113I-class.html#size,Variable Bio.Restriction.Restriction.MlyI.size=Bio.Restriction.Restriction.MlyI-class.html#size,Variable Bio.Restriction.Restriction.MmeI.size=Bio.Restriction.Restriction.MmeI-class.html#size,Variable Bio.Restriction.Restriction.MnlI.size=Bio.Restriction.Restriction.MnlI-class.html#size,Variable Bio.Restriction.Restriction.Mph1103I.size=Bio.Restriction.Restriction.Mph1103I-class.html#size,Variable Bio.Restriction.Restriction.MreI.size=Bio.Restriction.Restriction.MreI-class.html#size,Variable Bio.Restriction.Restriction.MroI.size=Bio.Restriction.Restriction.MroI-class.html#size,Variable Bio.Restriction.Restriction.MroNI.size=Bio.Restriction.Restriction.MroNI-class.html#size,Variable Bio.Restriction.Restriction.MroXI.size=Bio.Restriction.Restriction.MroXI-class.html#size,Variable Bio.Restriction.Restriction.MscI.size=Bio.Restriction.Restriction.MscI-class.html#size,Variable Bio.Restriction.Restriction.MseI.size=Bio.Restriction.Restriction.MseI-class.html#size,Variable Bio.Restriction.Restriction.MslI.size=Bio.Restriction.Restriction.MslI-class.html#size,Variable Bio.Restriction.Restriction.Msp20I.size=Bio.Restriction.Restriction.Msp20I-class.html#size,Variable Bio.Restriction.Restriction.MspA1I.size=Bio.Restriction.Restriction.MspA1I-class.html#size,Variable Bio.Restriction.Restriction.MspCI.size=Bio.Restriction.Restriction.MspCI-class.html#size,Variable Bio.Restriction.Restriction.MspI.size=Bio.Restriction.Restriction.MspI-class.html#size,Variable Bio.Restriction.Restriction.MspJI.size=Bio.Restriction.Restriction.MspJI-class.html#size,Variable Bio.Restriction.Restriction.MspR9I.size=Bio.Restriction.Restriction.MspR9I-class.html#size,Variable Bio.Restriction.Restriction.MssI.size=Bio.Restriction.Restriction.MssI-class.html#size,Variable Bio.Restriction.Restriction.MstI.size=Bio.Restriction.Restriction.MstI-class.html#size,Variable Bio.Restriction.Restriction.MunI.size=Bio.Restriction.Restriction.MunI-class.html#size,Variable Bio.Restriction.Restriction.Mva1269I.size=Bio.Restriction.Restriction.Mva1269I-class.html#size,Variable Bio.Restriction.Restriction.MvaI.size=Bio.Restriction.Restriction.MvaI-class.html#size,Variable Bio.Restriction.Restriction.MvnI.size=Bio.Restriction.Restriction.MvnI-class.html#size,Variable Bio.Restriction.Restriction.MvrI.size=Bio.Restriction.Restriction.MvrI-class.html#size,Variable Bio.Restriction.Restriction.MwoI.size=Bio.Restriction.Restriction.MwoI-class.html#size,Variable Bio.Restriction.Restriction.NaeI.size=Bio.Restriction.Restriction.NaeI-class.html#size,Variable Bio.Restriction.Restriction.NarI.size=Bio.Restriction.Restriction.NarI-class.html#size,Variable Bio.Restriction.Restriction.NciI.size=Bio.Restriction.Restriction.NciI-class.html#size,Variable Bio.Restriction.Restriction.NcoI.size=Bio.Restriction.Restriction.NcoI-class.html#size,Variable Bio.Restriction.Restriction.NdeI.size=Bio.Restriction.Restriction.NdeI-class.html#size,Variable Bio.Restriction.Restriction.NdeII.size=Bio.Restriction.Restriction.NdeII-class.html#size,Variable Bio.Restriction.Restriction.NgoAVIII.size=Bio.Restriction.Restriction.NgoAVIII-class.html#size,Variable Bio.Restriction.Restriction.NgoMIV.size=Bio.Restriction.Restriction.NgoMIV-class.html#size,Variable Bio.Restriction.Restriction.NhaXI.size=Bio.Restriction.Restriction.NhaXI-class.html#size,Variable Bio.Restriction.Restriction.NheI.size=Bio.Restriction.Restriction.NheI-class.html#size,Variable Bio.Restriction.Restriction.NlaCI.size=Bio.Restriction.Restriction.NlaCI-class.html#size,Variable Bio.Restriction.Restriction.NlaIII.size=Bio.Restriction.Restriction.NlaIII-class.html#size,Variable Bio.Restriction.Restriction.NlaIV.size=Bio.Restriction.Restriction.NlaIV-class.html#size,Variable Bio.Restriction.Restriction.Nli3877I.size=Bio.Restriction.Restriction.Nli3877I-class.html#size,Variable Bio.Restriction.Restriction.NmeAIII.size=Bio.Restriction.Restriction.NmeAIII-class.html#size,Variable Bio.Restriction.Restriction.NmeDI.size=Bio.Restriction.Restriction.NmeDI-class.html#size,Variable Bio.Restriction.Restriction.NmuCI.size=Bio.Restriction.Restriction.NmuCI-class.html#size,Variable Bio.Restriction.Restriction.NotI.size=Bio.Restriction.Restriction.NotI-class.html#size,Variable Bio.Restriction.Restriction.NruI.size=Bio.Restriction.Restriction.NruI-class.html#size,Variable Bio.Restriction.Restriction.NsbI.size=Bio.Restriction.Restriction.NsbI-class.html#size,Variable Bio.Restriction.Restriction.NsiI.size=Bio.Restriction.Restriction.NsiI-class.html#size,Variable Bio.Restriction.Restriction.NspBII.size=Bio.Restriction.Restriction.NspBII-class.html#size,Variable Bio.Restriction.Restriction.NspI.size=Bio.Restriction.Restriction.NspI-class.html#size,Variable Bio.Restriction.Restriction.NspV.size=Bio.Restriction.Restriction.NspV-class.html#size,Variable Bio.Restriction.Restriction.OliI.size=Bio.Restriction.Restriction.OliI-class.html#size,Variable Bio.Restriction.Restriction.PabI.size=Bio.Restriction.Restriction.PabI-class.html#size,Variable Bio.Restriction.Restriction.PacI.size=Bio.Restriction.Restriction.PacI-class.html#size,Variable Bio.Restriction.Restriction.PaeI.size=Bio.Restriction.Restriction.PaeI-class.html#size,Variable Bio.Restriction.Restriction.PaeR7I.size=Bio.Restriction.Restriction.PaeR7I-class.html#size,Variable Bio.Restriction.Restriction.PagI.size=Bio.Restriction.Restriction.PagI-class.html#size,Variable Bio.Restriction.Restriction.PalAI.size=Bio.Restriction.Restriction.PalAI-class.html#size,Variable Bio.Restriction.Restriction.PasI.size=Bio.Restriction.Restriction.PasI-class.html#size,Variable Bio.Restriction.Restriction.PauI.size=Bio.Restriction.Restriction.PauI-class.html#size,Variable Bio.Restriction.Restriction.PceI.size=Bio.Restriction.Restriction.PceI-class.html#size,Variable Bio.Restriction.Restriction.PciI.size=Bio.Restriction.Restriction.PciI-class.html#size,Variable Bio.Restriction.Restriction.PciSI.size=Bio.Restriction.Restriction.PciSI-class.html#size,Variable Bio.Restriction.Restriction.PcsI.size=Bio.Restriction.Restriction.PcsI-class.html#size,Variable Bio.Restriction.Restriction.PctI.size=Bio.Restriction.Restriction.PctI-class.html#size,Variable Bio.Restriction.Restriction.PdiI.size=Bio.Restriction.Restriction.PdiI-class.html#size,Variable Bio.Restriction.Restriction.PdmI.size=Bio.Restriction.Restriction.PdmI-class.html#size,Variable Bio.Restriction.Restriction.PenI.size=Bio.Restriction.Restriction.PenI-class.html#size,Variable Bio.Restriction.Restriction.PfeI.size=Bio.Restriction.Restriction.PfeI-class.html#size,Variable Bio.Restriction.Restriction.Pfl1108I.size=Bio.Restriction.Restriction.Pfl1108I-class.html#size,Variable Bio.Restriction.Restriction.Pfl23II.size=Bio.Restriction.Restriction.Pfl23II-class.html#size,Variable Bio.Restriction.Restriction.PflFI.size=Bio.Restriction.Restriction.PflFI-class.html#size,Variable Bio.Restriction.Restriction.PflMI.size=Bio.Restriction.Restriction.PflMI-class.html#size,Variable Bio.Restriction.Restriction.PfoI.size=Bio.Restriction.Restriction.PfoI-class.html#size,Variable Bio.Restriction.Restriction.PinAI.size=Bio.Restriction.Restriction.PinAI-class.html#size,Variable Bio.Restriction.Restriction.PlaDI.size=Bio.Restriction.Restriction.PlaDI-class.html#size,Variable Bio.Restriction.Restriction.Ple19I.size=Bio.Restriction.Restriction.Ple19I-class.html#size,Variable Bio.Restriction.Restriction.PleI.size=Bio.Restriction.Restriction.PleI-class.html#size,Variable Bio.Restriction.Restriction.PluTI.size=Bio.Restriction.Restriction.PluTI-class.html#size,Variable Bio.Restriction.Restriction.PmaCI.size=Bio.Restriction.Restriction.PmaCI-class.html#size,Variable Bio.Restriction.Restriction.PmeI.size=Bio.Restriction.Restriction.PmeI-class.html#size,Variable Bio.Restriction.Restriction.PmlI.size=Bio.Restriction.Restriction.PmlI-class.html#size,Variable Bio.Restriction.Restriction.PpiI.size=Bio.Restriction.Restriction.PpiI-class.html#size,Variable Bio.Restriction.Restriction.PpsI.size=Bio.Restriction.Restriction.PpsI-class.html#size,Variable Bio.Restriction.Restriction.Ppu10I.size=Bio.Restriction.Restriction.Ppu10I-class.html#size,Variable Bio.Restriction.Restriction.Ppu21I.size=Bio.Restriction.Restriction.Ppu21I-class.html#size,Variable Bio.Restriction.Restriction.PpuMI.size=Bio.Restriction.Restriction.PpuMI-class.html#size,Variable Bio.Restriction.Restriction.PscI.size=Bio.Restriction.Restriction.PscI-class.html#size,Variable Bio.Restriction.Restriction.PshAI.size=Bio.Restriction.Restriction.PshAI-class.html#size,Variable Bio.Restriction.Restriction.PshBI.size=Bio.Restriction.Restriction.PshBI-class.html#size,Variable Bio.Restriction.Restriction.PsiI.size=Bio.Restriction.Restriction.PsiI-class.html#size,Variable Bio.Restriction.Restriction.Psp03I.size=Bio.Restriction.Restriction.Psp03I-class.html#size,Variable Bio.Restriction.Restriction.Psp124BI.size=Bio.Restriction.Restriction.Psp124BI-class.html#size,Variable Bio.Restriction.Restriction.Psp1406I.size=Bio.Restriction.Restriction.Psp1406I-class.html#size,Variable Bio.Restriction.Restriction.Psp5II.size=Bio.Restriction.Restriction.Psp5II-class.html#size,Variable Bio.Restriction.Restriction.Psp6I.size=Bio.Restriction.Restriction.Psp6I-class.html#size,Variable Bio.Restriction.Restriction.PspCI.size=Bio.Restriction.Restriction.PspCI-class.html#size,Variable Bio.Restriction.Restriction.PspEI.size=Bio.Restriction.Restriction.PspEI-class.html#size,Variable Bio.Restriction.Restriction.PspGI.size=Bio.Restriction.Restriction.PspGI-class.html#size,Variable Bio.Restriction.Restriction.PspLI.size=Bio.Restriction.Restriction.PspLI-class.html#size,Variable Bio.Restriction.Restriction.PspN4I.size=Bio.Restriction.Restriction.PspN4I-class.html#size,Variable Bio.Restriction.Restriction.PspOMI.size=Bio.Restriction.Restriction.PspOMI-class.html#size,Variable Bio.Restriction.Restriction.PspOMII.size=Bio.Restriction.Restriction.PspOMII-class.html#size,Variable Bio.Restriction.Restriction.PspPI.size=Bio.Restriction.Restriction.PspPI-class.html#size,Variable Bio.Restriction.Restriction.PspPPI.size=Bio.Restriction.Restriction.PspPPI-class.html#size,Variable Bio.Restriction.Restriction.PspPRI.size=Bio.Restriction.Restriction.PspPRI-class.html#size,Variable Bio.Restriction.Restriction.PspXI.size=Bio.Restriction.Restriction.PspXI-class.html#size,Variable Bio.Restriction.Restriction.PsrI.size=Bio.Restriction.Restriction.PsrI-class.html#size,Variable Bio.Restriction.Restriction.PssI.size=Bio.Restriction.Restriction.PssI-class.html#size,Variable Bio.Restriction.Restriction.PstI.size=Bio.Restriction.Restriction.PstI-class.html#size,Variable Bio.Restriction.Restriction.PstNI.size=Bio.Restriction.Restriction.PstNI-class.html#size,Variable Bio.Restriction.Restriction.PsuI.size=Bio.Restriction.Restriction.PsuI-class.html#size,Variable Bio.Restriction.Restriction.PsyI.size=Bio.Restriction.Restriction.PsyI-class.html#size,Variable Bio.Restriction.Restriction.PteI.size=Bio.Restriction.Restriction.PteI-class.html#size,Variable Bio.Restriction.Restriction.PvuI.size=Bio.Restriction.Restriction.PvuI-class.html#size,Variable Bio.Restriction.Restriction.PvuII.size=Bio.Restriction.Restriction.PvuII-class.html#size,Variable Bio.Restriction.Restriction.R2_BceSIV.size=Bio.Restriction.Restriction.R2_BceSIV-class.html#size,Variable Bio.Restriction.Restriction.RceI.size=Bio.Restriction.Restriction.RceI-class.html#size,Variable Bio.Restriction.Restriction.RdeGBI.size=Bio.Restriction.Restriction.RdeGBI-class.html#size,Variable Bio.Restriction.Restriction.RdeGBII.size=Bio.Restriction.Restriction.RdeGBII-class.html#size,Variable Bio.Restriction.Restriction.RdeGBIII.size=Bio.Restriction.Restriction.RdeGBIII-class.html#size,Variable Bio.Restriction.Restriction.RflFIII.size=Bio.Restriction.Restriction.RflFIII-class.html#size,Variable Bio.Restriction.Restriction.RgaI.size=Bio.Restriction.Restriction.RgaI-class.html#size,Variable Bio.Restriction.Restriction.RigI.size=Bio.Restriction.Restriction.RigI-class.html#size,Variable Bio.Restriction.Restriction.RlaI.size=Bio.Restriction.Restriction.RlaI-class.html#size,Variable Bio.Restriction.Restriction.RleAI.size=Bio.Restriction.Restriction.RleAI-class.html#size,Variable Bio.Restriction.Restriction.RpaB5I.size=Bio.Restriction.Restriction.RpaB5I-class.html#size,Variable Bio.Restriction.Restriction.RpaBI.size=Bio.Restriction.Restriction.RpaBI-class.html#size,Variable Bio.Restriction.Restriction.RpaI.size=Bio.Restriction.Restriction.RpaI-class.html#size,Variable Bio.Restriction.Restriction.RpaTI.size=Bio.Restriction.Restriction.RpaTI-class.html#size,Variable Bio.Restriction.Restriction.RruI.size=Bio.Restriction.Restriction.RruI-class.html#size,Variable Bio.Restriction.Restriction.RsaI.size=Bio.Restriction.Restriction.RsaI-class.html#size,Variable Bio.Restriction.Restriction.RsaNI.size=Bio.Restriction.Restriction.RsaNI-class.html#size,Variable Bio.Restriction.Restriction.RseI.size=Bio.Restriction.Restriction.RseI-class.html#size,Variable Bio.Restriction.Restriction.Rsr2I.size=Bio.Restriction.Restriction.Rsr2I-class.html#size,Variable Bio.Restriction.Restriction.RsrII.size=Bio.Restriction.Restriction.RsrII-class.html#size,Variable Bio.Restriction.Restriction.SacI.size=Bio.Restriction.Restriction.SacI-class.html#size,Variable Bio.Restriction.Restriction.SacII.size=Bio.Restriction.Restriction.SacII-class.html#size,Variable Bio.Restriction.Restriction.SalI.size=Bio.Restriction.Restriction.SalI-class.html#size,Variable Bio.Restriction.Restriction.SanDI.size=Bio.Restriction.Restriction.SanDI-class.html#size,Variable Bio.Restriction.Restriction.SapI.size=Bio.Restriction.Restriction.SapI-class.html#size,Variable Bio.Restriction.Restriction.SaqAI.size=Bio.Restriction.Restriction.SaqAI-class.html#size,Variable Bio.Restriction.Restriction.SatI.size=Bio.Restriction.Restriction.SatI-class.html#size,Variable Bio.Restriction.Restriction.Sau3AI.size=Bio.Restriction.Restriction.Sau3AI-class.html#size,Variable Bio.Restriction.Restriction.Sau96I.size=Bio.Restriction.Restriction.Sau96I-class.html#size,Variable Bio.Restriction.Restriction.SauI.size=Bio.Restriction.Restriction.SauI-class.html#size,Variable Bio.Restriction.Restriction.SbfI.size=Bio.Restriction.Restriction.SbfI-class.html#size,Variable Bio.Restriction.Restriction.ScaI.size=Bio.Restriction.Restriction.ScaI-class.html#size,Variable Bio.Restriction.Restriction.SchI.size=Bio.Restriction.Restriction.SchI-class.html#size,Variable Bio.Restriction.Restriction.SciI.size=Bio.Restriction.Restriction.SciI-class.html#size,Variable Bio.Restriction.Restriction.ScrFI.size=Bio.Restriction.Restriction.ScrFI-class.html#size,Variable Bio.Restriction.Restriction.SdaI.size=Bio.Restriction.Restriction.SdaI-class.html#size,Variable Bio.Restriction.Restriction.SdeAI.size=Bio.Restriction.Restriction.SdeAI-class.html#size,Variable Bio.Restriction.Restriction.SdeOSI.size=Bio.Restriction.Restriction.SdeOSI-class.html#size,Variable Bio.Restriction.Restriction.SduI.size=Bio.Restriction.Restriction.SduI-class.html#size,Variable Bio.Restriction.Restriction.SecI.size=Bio.Restriction.Restriction.SecI-class.html#size,Variable Bio.Restriction.Restriction.SelI.size=Bio.Restriction.Restriction.SelI-class.html#size,Variable Bio.Restriction.Restriction.SetI.size=Bio.Restriction.Restriction.SetI-class.html#size,Variable Bio.Restriction.Restriction.SexAI.size=Bio.Restriction.Restriction.SexAI-class.html#size,Variable Bio.Restriction.Restriction.SfaAI.size=Bio.Restriction.Restriction.SfaAI-class.html#size,Variable Bio.Restriction.Restriction.SfaNI.size=Bio.Restriction.Restriction.SfaNI-class.html#size,Variable Bio.Restriction.Restriction.SfcI.size=Bio.Restriction.Restriction.SfcI-class.html#size,Variable Bio.Restriction.Restriction.SfeI.size=Bio.Restriction.Restriction.SfeI-class.html#size,Variable Bio.Restriction.Restriction.SfiI.size=Bio.Restriction.Restriction.SfiI-class.html#size,Variable Bio.Restriction.Restriction.SfoI.size=Bio.Restriction.Restriction.SfoI-class.html#size,Variable Bio.Restriction.Restriction.Sfr274I.size=Bio.Restriction.Restriction.Sfr274I-class.html#size,Variable Bio.Restriction.Restriction.Sfr303I.size=Bio.Restriction.Restriction.Sfr303I-class.html#size,Variable Bio.Restriction.Restriction.SfuI.size=Bio.Restriction.Restriction.SfuI-class.html#size,Variable Bio.Restriction.Restriction.SgeI.size=Bio.Restriction.Restriction.SgeI-class.html#size,Variable Bio.Restriction.Restriction.SgfI.size=Bio.Restriction.Restriction.SgfI-class.html#size,Variable Bio.Restriction.Restriction.SgrAI.size=Bio.Restriction.Restriction.SgrAI-class.html#size,Variable Bio.Restriction.Restriction.SgrBI.size=Bio.Restriction.Restriction.SgrBI-class.html#size,Variable Bio.Restriction.Restriction.SgrDI.size=Bio.Restriction.Restriction.SgrDI-class.html#size,Variable Bio.Restriction.Restriction.SgrTI.size=Bio.Restriction.Restriction.SgrTI-class.html#size,Variable Bio.Restriction.Restriction.SgsI.size=Bio.Restriction.Restriction.SgsI-class.html#size,Variable Bio.Restriction.Restriction.SimI.size=Bio.Restriction.Restriction.SimI-class.html#size,Variable Bio.Restriction.Restriction.SlaI.size=Bio.Restriction.Restriction.SlaI-class.html#size,Variable Bio.Restriction.Restriction.SmaI.size=Bio.Restriction.Restriction.SmaI-class.html#size,Variable Bio.Restriction.Restriction.SmiI.size=Bio.Restriction.Restriction.SmiI-class.html#size,Variable Bio.Restriction.Restriction.SmiMI.size=Bio.Restriction.Restriction.SmiMI-class.html#size,Variable Bio.Restriction.Restriction.SmlI.size=Bio.Restriction.Restriction.SmlI-class.html#size,Variable Bio.Restriction.Restriction.SmoI.size=Bio.Restriction.Restriction.SmoI-class.html#size,Variable Bio.Restriction.Restriction.SnaBI.size=Bio.Restriction.Restriction.SnaBI-class.html#size,Variable Bio.Restriction.Restriction.SnaI.size=Bio.Restriction.Restriction.SnaI-class.html#size,Variable Bio.Restriction.Restriction.Sno506I.size=Bio.Restriction.Restriction.Sno506I-class.html#size,Variable Bio.Restriction.Restriction.SpeI.size=Bio.Restriction.Restriction.SpeI-class.html#size,Variable Bio.Restriction.Restriction.SphI.size=Bio.Restriction.Restriction.SphI-class.html#size,Variable Bio.Restriction.Restriction.SplI.size=Bio.Restriction.Restriction.SplI-class.html#size,Variable Bio.Restriction.Restriction.SpoDI.size=Bio.Restriction.Restriction.SpoDI-class.html#size,Variable Bio.Restriction.Restriction.SrfI.size=Bio.Restriction.Restriction.SrfI-class.html#size,Variable Bio.Restriction.Restriction.Sse232I.size=Bio.Restriction.Restriction.Sse232I-class.html#size,Variable Bio.Restriction.Restriction.Sse8387I.size=Bio.Restriction.Restriction.Sse8387I-class.html#size,Variable Bio.Restriction.Restriction.Sse8647I.size=Bio.Restriction.Restriction.Sse8647I-class.html#size,Variable Bio.Restriction.Restriction.Sse9I.size=Bio.Restriction.Restriction.Sse9I-class.html#size,Variable Bio.Restriction.Restriction.SseBI.size=Bio.Restriction.Restriction.SseBI-class.html#size,Variable Bio.Restriction.Restriction.SsiI.size=Bio.Restriction.Restriction.SsiI-class.html#size,Variable Bio.Restriction.Restriction.SspD5I.size=Bio.Restriction.Restriction.SspD5I-class.html#size,Variable Bio.Restriction.Restriction.SspDI.size=Bio.Restriction.Restriction.SspDI-class.html#size,Variable Bio.Restriction.Restriction.SspI.size=Bio.Restriction.Restriction.SspI-class.html#size,Variable Bio.Restriction.Restriction.SstE37I.size=Bio.Restriction.Restriction.SstE37I-class.html#size,Variable Bio.Restriction.Restriction.SstI.size=Bio.Restriction.Restriction.SstI-class.html#size,Variable Bio.Restriction.Restriction.Sth132I.size=Bio.Restriction.Restriction.Sth132I-class.html#size,Variable Bio.Restriction.Restriction.Sth302II.size=Bio.Restriction.Restriction.Sth302II-class.html#size,Variable Bio.Restriction.Restriction.StrI.size=Bio.Restriction.Restriction.StrI-class.html#size,Variable Bio.Restriction.Restriction.StsI.size=Bio.Restriction.Restriction.StsI-class.html#size,Variable Bio.Restriction.Restriction.StuI.size=Bio.Restriction.Restriction.StuI-class.html#size,Variable Bio.Restriction.Restriction.StyD4I.size=Bio.Restriction.Restriction.StyD4I-class.html#size,Variable Bio.Restriction.Restriction.StyI.size=Bio.Restriction.Restriction.StyI-class.html#size,Variable Bio.Restriction.Restriction.SwaI.size=Bio.Restriction.Restriction.SwaI-class.html#size,Variable Bio.Restriction.Restriction.TaaI.size=Bio.Restriction.Restriction.TaaI-class.html#size,Variable Bio.Restriction.Restriction.TaiI.size=Bio.Restriction.Restriction.TaiI-class.html#size,Variable Bio.Restriction.Restriction.TaqI.size=Bio.Restriction.Restriction.TaqI-class.html#size,Variable Bio.Restriction.Restriction.TaqII.size=Bio.Restriction.Restriction.TaqII-class.html#size,Variable Bio.Restriction.Restriction.TasI.size=Bio.Restriction.Restriction.TasI-class.html#size,Variable Bio.Restriction.Restriction.TatI.size=Bio.Restriction.Restriction.TatI-class.html#size,Variable Bio.Restriction.Restriction.TauI.size=Bio.Restriction.Restriction.TauI-class.html#size,Variable Bio.Restriction.Restriction.TfiI.size=Bio.Restriction.Restriction.TfiI-class.html#size,Variable Bio.Restriction.Restriction.Tru1I.size=Bio.Restriction.Restriction.Tru1I-class.html#size,Variable Bio.Restriction.Restriction.Tru9I.size=Bio.Restriction.Restriction.Tru9I-class.html#size,Variable Bio.Restriction.Restriction.TscAI.size=Bio.Restriction.Restriction.TscAI-class.html#size,Variable Bio.Restriction.Restriction.TseFI.size=Bio.Restriction.Restriction.TseFI-class.html#size,Variable Bio.Restriction.Restriction.TseI.size=Bio.Restriction.Restriction.TseI-class.html#size,Variable Bio.Restriction.Restriction.TsoI.size=Bio.Restriction.Restriction.TsoI-class.html#size,Variable Bio.Restriction.Restriction.Tsp45I.size=Bio.Restriction.Restriction.Tsp45I-class.html#size,Variable Bio.Restriction.Restriction.Tsp4CI.size=Bio.Restriction.Restriction.Tsp4CI-class.html#size,Variable Bio.Restriction.Restriction.TspDTI.size=Bio.Restriction.Restriction.TspDTI-class.html#size,Variable Bio.Restriction.Restriction.TspEI.size=Bio.Restriction.Restriction.TspEI-class.html#size,Variable Bio.Restriction.Restriction.TspGWI.size=Bio.Restriction.Restriction.TspGWI-class.html#size,Variable Bio.Restriction.Restriction.TspMI.size=Bio.Restriction.Restriction.TspMI-class.html#size,Variable Bio.Restriction.Restriction.TspRI.size=Bio.Restriction.Restriction.TspRI-class.html#size,Variable Bio.Restriction.Restriction.TssI.size=Bio.Restriction.Restriction.TssI-class.html#size,Variable Bio.Restriction.Restriction.TstI.size=Bio.Restriction.Restriction.TstI-class.html#size,Variable Bio.Restriction.Restriction.TsuI.size=Bio.Restriction.Restriction.TsuI-class.html#size,Variable Bio.Restriction.Restriction.Tth111I.size=Bio.Restriction.Restriction.Tth111I-class.html#size,Variable Bio.Restriction.Restriction.Tth111II.size=Bio.Restriction.Restriction.Tth111II-class.html#size,Variable Bio.Restriction.Restriction.UbaF11I.size=Bio.Restriction.Restriction.UbaF11I-class.html#size,Variable Bio.Restriction.Restriction.UbaF12I.size=Bio.Restriction.Restriction.UbaF12I-class.html#size,Variable Bio.Restriction.Restriction.UbaF13I.size=Bio.Restriction.Restriction.UbaF13I-class.html#size,Variable Bio.Restriction.Restriction.UbaF14I.size=Bio.Restriction.Restriction.UbaF14I-class.html#size,Variable Bio.Restriction.Restriction.UbaF9I.size=Bio.Restriction.Restriction.UbaF9I-class.html#size,Variable Bio.Restriction.Restriction.UbaPI.size=Bio.Restriction.Restriction.UbaPI-class.html#size,Variable Bio.Restriction.Restriction.UcoMSI.size=Bio.Restriction.Restriction.UcoMSI-class.html#size,Variable Bio.Restriction.Restriction.UnbI.size=Bio.Restriction.Restriction.UnbI-class.html#size,Variable Bio.Restriction.Restriction.Van91I.size=Bio.Restriction.Restriction.Van91I-class.html#size,Variable Bio.Restriction.Restriction.Vha464I.size=Bio.Restriction.Restriction.Vha464I-class.html#size,Variable Bio.Restriction.Restriction.VneI.size=Bio.Restriction.Restriction.VneI-class.html#size,Variable Bio.Restriction.Restriction.VpaK11AI.size=Bio.Restriction.Restriction.VpaK11AI-class.html#size,Variable Bio.Restriction.Restriction.VpaK11BI.size=Bio.Restriction.Restriction.VpaK11BI-class.html#size,Variable Bio.Restriction.Restriction.VspI.size=Bio.Restriction.Restriction.VspI-class.html#size,Variable Bio.Restriction.Restriction.WviI.size=Bio.Restriction.Restriction.WviI-class.html#size,Variable Bio.Restriction.Restriction.XagI.size=Bio.Restriction.Restriction.XagI-class.html#size,Variable Bio.Restriction.Restriction.XapI.size=Bio.Restriction.Restriction.XapI-class.html#size,Variable Bio.Restriction.Restriction.XbaI.size=Bio.Restriction.Restriction.XbaI-class.html#size,Variable Bio.Restriction.Restriction.XceI.size=Bio.Restriction.Restriction.XceI-class.html#size,Variable Bio.Restriction.Restriction.XcmI.size=Bio.Restriction.Restriction.XcmI-class.html#size,Variable Bio.Restriction.Restriction.XhoI.size=Bio.Restriction.Restriction.XhoI-class.html#size,Variable Bio.Restriction.Restriction.XhoII.size=Bio.Restriction.Restriction.XhoII-class.html#size,Variable Bio.Restriction.Restriction.XmaI.size=Bio.Restriction.Restriction.XmaI-class.html#size,Variable Bio.Restriction.Restriction.XmaIII.size=Bio.Restriction.Restriction.XmaIII-class.html#size,Variable Bio.Restriction.Restriction.XmaJI.size=Bio.Restriction.Restriction.XmaJI-class.html#size,Variable Bio.Restriction.Restriction.XmiI.size=Bio.Restriction.Restriction.XmiI-class.html#size,Variable Bio.Restriction.Restriction.XmnI.size=Bio.Restriction.Restriction.XmnI-class.html#size,Variable Bio.Restriction.Restriction.XspI.size=Bio.Restriction.Restriction.XspI-class.html#size,Variable Bio.Restriction.Restriction.YkrI.size=Bio.Restriction.Restriction.YkrI-class.html#size,Variable Bio.Restriction.Restriction.ZraI.size=Bio.Restriction.Restriction.ZraI-class.html#size,Variable Bio.Restriction.Restriction.ZrmI.size=Bio.Restriction.Restriction.ZrmI-class.html#size,Variable Bio.Restriction.Restriction.Zsp2I.size=Bio.Restriction.Restriction.Zsp2I-class.html#size,Variable Bio.codonalign.codonalphabet.CodonAlphabet.size=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#size"><a title="Bio.Alphabet.Alphabet.size
Bio.Alphabet.Reduced.HPModel.size
Bio.Alphabet.Reduced.Murphy10.size
Bio.Alphabet.Reduced.Murphy15.size
Bio.Alphabet.Reduced.Murphy4.size
Bio.Alphabet.Reduced.Murphy8.size
Bio.Alphabet.Reduced.PC5.size
Bio.Alphabet.SingleLetterAlphabet.size
Bio.Alphabet.ThreeLetterProtein.size
Bio.GenBank._FeatureConsumer.size
Bio.GenBank._RecordConsumer.size
Bio.Restriction.Restriction.AanI.size
Bio.Restriction.Restriction.AarI.size
Bio.Restriction.Restriction.AasI.size
Bio.Restriction.Restriction.AatII.size
Bio.Restriction.Restriction.AbaSI.size
Bio.Restriction.Restriction.AbsI.size
Bio.Restriction.Restriction.Acc16I.size
Bio.Restriction.Restriction.Acc36I.size
Bio.Restriction.Restriction.Acc65I.size
Bio.Restriction.Restriction.AccB1I.size
Bio.Restriction.Restriction.AccB7I.size
Bio.Restriction.Restriction.AccBSI.size
Bio.Restriction.Restriction.AccI.size
Bio.Restriction.Restriction.AccII.size
Bio.Restriction.Restriction.AccIII.size
Bio.Restriction.Restriction.AceIII.size
Bio.Restriction.Restriction.AciI.size
Bio.Restriction.Restriction.AclI.size
Bio.Restriction.Restriction.AclWI.size
Bio.Restriction.Restriction.AcoI.size
Bio.Restriction.Restriction.AcsI.size
Bio.Restriction.Restriction.AcuI.size
Bio.Restriction.Restriction.AcvI.size
Bio.Restriction.Restriction.AcyI.size
Bio.Restriction.Restriction.AdeI.size
Bio.Restriction.Restriction.AfaI.size
Bio.Restriction.Restriction.AfeI.size
Bio.Restriction.Restriction.AfiI.size
Bio.Restriction.Restriction.AflII.size
Bio.Restriction.Restriction.AflIII.size
Bio.Restriction.Restriction.AgeI.size
Bio.Restriction.Restriction.AgsI.size
Bio.Restriction.Restriction.AhaIII.size
Bio.Restriction.Restriction.AhdI.size
Bio.Restriction.Restriction.AhlI.size
Bio.Restriction.Restriction.AjiI.size
Bio.Restriction.Restriction.AjnI.size
Bio.Restriction.Restriction.AjuI.size
Bio.Restriction.Restriction.AleI.size
Bio.Restriction.Restriction.AlfI.size
Bio.Restriction.Restriction.AloI.size
Bio.Restriction.Restriction.AluBI.size
Bio.Restriction.Restriction.AluI.size
Bio.Restriction.Restriction.Alw21I.size
Bio.Restriction.Restriction.Alw26I.size
Bio.Restriction.Restriction.Alw44I.size
Bio.Restriction.Restriction.AlwFI.size
Bio.Restriction.Restriction.AlwI.size
Bio.Restriction.Restriction.AlwNI.size
Bio.Restriction.Restriction.Ama87I.size
Bio.Restriction.Restriction.Aor13HI.size
Bio.Restriction.Restriction.Aor51HI.size
Bio.Restriction.Restriction.AoxI.size
Bio.Restriction.Restriction.ApaBI.size
Bio.Restriction.Restriction.ApaI.size
Bio.Restriction.Restriction.ApaLI.size
Bio.Restriction.Restriction.ApeKI.size
Bio.Restriction.Restriction.ApoI.size
Bio.Restriction.Restriction.ApyPI.size
Bio.Restriction.Restriction.AquII.size
Bio.Restriction.Restriction.AquIII.size
Bio.Restriction.Restriction.AquIV.size
Bio.Restriction.Restriction.ArsI.size
Bio.Restriction.Restriction.AscI.size
Bio.Restriction.Restriction.AseI.size
Bio.Restriction.Restriction.Asi256I.size
Bio.Restriction.Restriction.AsiGI.size
Bio.Restriction.Restriction.AsiSI.size
Bio.Restriction.Restriction.Asp700I.size
Bio.Restriction.Restriction.Asp718I.size
Bio.Restriction.Restriction.AspA2I.size
Bio.Restriction.Restriction.AspBHI.size
Bio.Restriction.Restriction.AspLEI.size
Bio.Restriction.Restriction.AspS9I.size
Bio.Restriction.Restriction.AssI.size
Bio.Restriction.Restriction.AsuC2I.size
Bio.Restriction.Restriction.AsuHPI.size
Bio.Restriction.Restriction.AsuI.size
Bio.Restriction.Restriction.AsuII.size
Bio.Restriction.Restriction.AsuNHI.size
Bio.Restriction.Restriction.AvaI.size
Bio.Restriction.Restriction.AvaII.size
Bio.Restriction.Restriction.AvaIII.size
Bio.Restriction.Restriction.AvrII.size
Bio.Restriction.Restriction.AxyI.size
Bio.Restriction.Restriction.BaeGI.size
Bio.Restriction.Restriction.BaeI.size
Bio.Restriction.Restriction.BalI.size
Bio.Restriction.Restriction.BamHI.size
Bio.Restriction.Restriction.BanI.size
Bio.Restriction.Restriction.BanII.size
Bio.Restriction.Restriction.BarI.size
Bio.Restriction.Restriction.BasI.size
Bio.Restriction.Restriction.BauI.size
Bio.Restriction.Restriction.Bbr7I.size
Bio.Restriction.Restriction.BbrPI.size
Bio.Restriction.Restriction.BbsI.size
Bio.Restriction.Restriction.Bbv12I.size
Bio.Restriction.Restriction.BbvCI.size
Bio.Restriction.Restriction.BbvI.size
Bio.Restriction.Restriction.BbvII.size
Bio.Restriction.Restriction.BccI.size
Bio.Restriction.Restriction.Bce83I.size
Bio.Restriction.Restriction.BceAI.size
Bio.Restriction.Restriction.BcefI.size
Bio.Restriction.Restriction.BcgI.size
Bio.Restriction.Restriction.BciT130I.size
Bio.Restriction.Restriction.BciVI.size
Bio.Restriction.Restriction.BclI.size
Bio.Restriction.Restriction.BcnI.size
Bio.Restriction.Restriction.BcoDI.size
Bio.Restriction.Restriction.BcuI.size
Bio.Restriction.Restriction.BdaI.size
Bio.Restriction.Restriction.BetI.size
Bio.Restriction.Restriction.BfaI.size
Bio.Restriction.Restriction.BfiI.size
Bio.Restriction.Restriction.BfmI.size
Bio.Restriction.Restriction.BfoI.size
Bio.Restriction.Restriction.BfrI.size
Bio.Restriction.Restriction.BfuAI.size
Bio.Restriction.Restriction.BfuCI.size
Bio.Restriction.Restriction.BfuI.size
Bio.Restriction.Restriction.BglI.size
Bio.Restriction.Restriction.BglII.size
Bio.Restriction.Restriction.BinI.size
Bio.Restriction.Restriction.BisI.size
Bio.Restriction.Restriction.BlnI.size
Bio.Restriction.Restriction.BlpI.size
Bio.Restriction.Restriction.BlsI.size
Bio.Restriction.Restriction.BmcAI.size
Bio.Restriction.Restriction.Bme1390I.size
Bio.Restriction.Restriction.Bme18I.size
Bio.Restriction.Restriction.BmeDI.size
Bio.Restriction.Restriction.BmeRI.size
Bio.Restriction.Restriction.BmeT110I.size
Bio.Restriction.Restriction.BmgBI.size
Bio.Restriction.Restriction.BmgI.size
Bio.Restriction.Restriction.BmgT120I.size
Bio.Restriction.Restriction.BmiI.size
Bio.Restriction.Restriction.BmrFI.size
Bio.Restriction.Restriction.BmrI.size
Bio.Restriction.Restriction.BmsI.size
Bio.Restriction.Restriction.BmtI.size
Bio.Restriction.Restriction.BmuI.size
Bio.Restriction.Restriction.BoxI.size
Bio.Restriction.Restriction.BpiI.size
Bio.Restriction.Restriction.BplI.size
Bio.Restriction.Restriction.BpmI.size
Bio.Restriction.Restriction.Bpu10I.size
Bio.Restriction.Restriction.Bpu1102I.size
Bio.Restriction.Restriction.Bpu14I.size
Bio.Restriction.Restriction.BpuEI.size
Bio.Restriction.Restriction.BpuMI.size
Bio.Restriction.Restriction.BpvUI.size
Bio.Restriction.Restriction.Bsa29I.size
Bio.Restriction.Restriction.BsaAI.size
Bio.Restriction.Restriction.BsaBI.size
Bio.Restriction.Restriction.BsaHI.size
Bio.Restriction.Restriction.BsaI.size
Bio.Restriction.Restriction.BsaJI.size
Bio.Restriction.Restriction.BsaWI.size
Bio.Restriction.Restriction.BsaXI.size
Bio.Restriction.Restriction.BsbI.size
Bio.Restriction.Restriction.Bsc4I.size
Bio.Restriction.Restriction.BscAI.size
Bio.Restriction.Restriction.BscGI.size
Bio.Restriction.Restriction.Bse118I.size
Bio.Restriction.Restriction.Bse1I.size
Bio.Restriction.Restriction.Bse21I.size
Bio.Restriction.Restriction.Bse3DI.size
Bio.Restriction.Restriction.Bse8I.size
Bio.Restriction.Restriction.BseAI.size
Bio.Restriction.Restriction.BseBI.size
Bio.Restriction.Restriction.BseCI.size
Bio.Restriction.Restriction.BseDI.size
Bio.Restriction.Restriction.BseGI.size
Bio.Restriction.Restriction.BseJI.size
Bio.Restriction.Restriction.BseLI.size
Bio.Restriction.Restriction.BseMI.size
Bio.Restriction.Restriction.BseMII.size
Bio.Restriction.Restriction.BseNI.size
Bio.Restriction.Restriction.BsePI.size
Bio.Restriction.Restriction.BseRI.size
Bio.Restriction.Restriction.BseSI.size
Bio.Restriction.Restriction.BseX3I.size
Bio.Restriction.Restriction.BseXI.size
Bio.Restriction.Restriction.BseYI.size
Bio.Restriction.Restriction.BsgI.size
Bio.Restriction.Restriction.Bsh1236I.size
Bio.Restriction.Restriction.Bsh1285I.size
Bio.Restriction.Restriction.BshFI.size
Bio.Restriction.Restriction.BshNI.size
Bio.Restriction.Restriction.BshTI.size
Bio.Restriction.Restriction.BshVI.size
Bio.Restriction.Restriction.BsiEI.size
Bio.Restriction.Restriction.BsiHKAI.size
Bio.Restriction.Restriction.BsiHKCI.size
Bio.Restriction.Restriction.BsiI.size
Bio.Restriction.Restriction.BsiSI.size
Bio.Restriction.Restriction.BsiWI.size
Bio.Restriction.Restriction.BsiYI.size
Bio.Restriction.Restriction.BslFI.size
Bio.Restriction.Restriction.BslI.size
Bio.Restriction.Restriction.BsmAI.size
Bio.Restriction.Restriction.BsmBI.size
Bio.Restriction.Restriction.BsmFI.size
Bio.Restriction.Restriction.BsmI.size
Bio.Restriction.Restriction.BsnI.size
Bio.Restriction.Restriction.Bso31I.size
Bio.Restriction.Restriction.BsoBI.size
Bio.Restriction.Restriction.Bsp119I.size
Bio.Restriction.Restriction.Bsp120I.size
Bio.Restriction.Restriction.Bsp1286I.size
Bio.Restriction.Restriction.Bsp13I.size
Bio.Restriction.Restriction.Bsp1407I.size
Bio.Restriction.Restriction.Bsp143I.size
Bio.Restriction.Restriction.Bsp1720I.size
Bio.Restriction.Restriction.Bsp19I.size
Bio.Restriction.Restriction.Bsp24I.size
Bio.Restriction.Restriction.Bsp68I.size
Bio.Restriction.Restriction.BspACI.size
Bio.Restriction.Restriction.BspCNI.size
Bio.Restriction.Restriction.BspD6I.size
Bio.Restriction.Restriction.BspDI.size
Bio.Restriction.Restriction.BspEI.size
Bio.Restriction.Restriction.BspFNI.size
Bio.Restriction.Restriction.BspGI.size
Bio.Restriction.Restriction.BspHI.size
Bio.Restriction.Restriction.BspLI.size
Bio.Restriction.Restriction.BspLU11I.size
Bio.Restriction.Restriction.BspMI.size
Bio.Restriction.Restriction.BspMII.size
Bio.Restriction.Restriction.BspNCI.size
Bio.Restriction.Restriction.BspOI.size
Bio.Restriction.Restriction.BspPI.size
Bio.Restriction.Restriction.BspQI.size
Bio.Restriction.Restriction.BspT104I.size
Bio.Restriction.Restriction.BspT107I.size
Bio.Restriction.Restriction.BspTI.size
Bio.Restriction.Restriction.BsrBI.size
Bio.Restriction.Restriction.BsrDI.size
Bio.Restriction.Restriction.BsrFI.size
Bio.Restriction.Restriction.BsrGI.size
Bio.Restriction.Restriction.BsrI.size
Bio.Restriction.Restriction.BsrSI.size
Bio.Restriction.Restriction.BssAI.size
Bio.Restriction.Restriction.BssECI.size
Bio.Restriction.Restriction.BssHII.size
Bio.Restriction.Restriction.BssKI.size
Bio.Restriction.Restriction.BssMI.size
Bio.Restriction.Restriction.BssNAI.size
Bio.Restriction.Restriction.BssNI.size
Bio.Restriction.Restriction.BssSI.size
Bio.Restriction.Restriction.BssT1I.size
Bio.Restriction.Restriction.Bst1107I.size
Bio.Restriction.Restriction.Bst2BI.size
Bio.Restriction.Restriction.Bst2UI.size
Bio.Restriction.Restriction.Bst4CI.size
Bio.Restriction.Restriction.Bst6I.size
Bio.Restriction.Restriction.BstACI.size
Bio.Restriction.Restriction.BstAFI.size
Bio.Restriction.Restriction.BstAPI.size
Bio.Restriction.Restriction.BstAUI.size
Bio.Restriction.Restriction.BstBAI.size
Bio.Restriction.Restriction.BstBI.size
Bio.Restriction.Restriction.BstC8I.size
Bio.Restriction.Restriction.BstDEI.size
Bio.Restriction.Restriction.BstDSI.size
Bio.Restriction.Restriction.BstEII.size
Bio.Restriction.Restriction.BstENI.size
Bio.Restriction.Restriction.BstF5I.size
Bio.Restriction.Restriction.BstFNI.size
Bio.Restriction.Restriction.BstH2I.size
Bio.Restriction.Restriction.BstHHI.size
Bio.Restriction.Restriction.BstKTI.size
Bio.Restriction.Restriction.BstMAI.size
Bio.Restriction.Restriction.BstMBI.size
Bio.Restriction.Restriction.BstMCI.size
Bio.Restriction.Restriction.BstMWI.size
Bio.Restriction.Restriction.BstNI.size
Bio.Restriction.Restriction.BstNSI.size
Bio.Restriction.Restriction.BstOI.size
Bio.Restriction.Restriction.BstPAI.size
Bio.Restriction.Restriction.BstPI.size
Bio.Restriction.Restriction.BstSCI.size
Bio.Restriction.Restriction.BstSFI.size
Bio.Restriction.Restriction.BstSLI.size
Bio.Restriction.Restriction.BstSNI.size
Bio.Restriction.Restriction.BstUI.size
Bio.Restriction.Restriction.BstV1I.size
Bio.Restriction.Restriction.BstV2I.size
Bio.Restriction.Restriction.BstX2I.size
Bio.Restriction.Restriction.BstXI.size
Bio.Restriction.Restriction.BstYI.size
Bio.Restriction.Restriction.BstZ17I.size
Bio.Restriction.Restriction.BstZI.size
Bio.Restriction.Restriction.Bsu15I.size
Bio.Restriction.Restriction.Bsu36I.size
Bio.Restriction.Restriction.BsuI.size
Bio.Restriction.Restriction.BsuRI.size
Bio.Restriction.Restriction.BtgI.size
Bio.Restriction.Restriction.BtgZI.size
Bio.Restriction.Restriction.BthCI.size
Bio.Restriction.Restriction.BtrI.size
Bio.Restriction.Restriction.BtsCI.size
Bio.Restriction.Restriction.BtsI.size
Bio.Restriction.Restriction.BtsIMutI.size
Bio.Restriction.Restriction.BtuMI.size
Bio.Restriction.Restriction.BveI.size
Bio.Restriction.Restriction.Cac8I.size
Bio.Restriction.Restriction.CaiI.size
Bio.Restriction.Restriction.CauII.size
Bio.Restriction.Restriction.CchII.size
Bio.Restriction.Restriction.CchIII.size
Bio.Restriction.Restriction.CciI.size
Bio.Restriction.Restriction.CciNI.size
Bio.Restriction.Restriction.Cdi630V.size
Bio.Restriction.Restriction.CdiI.size
Bio.Restriction.Restriction.CdpI.size
Bio.Restriction.Restriction.CfoI.size
Bio.Restriction.Restriction.Cfr10I.size
Bio.Restriction.Restriction.Cfr13I.size
Bio.Restriction.Restriction.Cfr42I.size
Bio.Restriction.Restriction.Cfr9I.size
Bio.Restriction.Restriction.CfrI.size
Bio.Restriction.Restriction.Cgl13032I.size
Bio.Restriction.Restriction.Cgl13032II.size
Bio.Restriction.Restriction.ChaI.size
Bio.Restriction.Restriction.CjeFIII.size
Bio.Restriction.Restriction.CjeFV.size
Bio.Restriction.Restriction.CjeI.size
Bio.Restriction.Restriction.CjeNII.size
Bio.Restriction.Restriction.CjeNIII.size
Bio.Restriction.Restriction.CjeP659IV.size
Bio.Restriction.Restriction.CjePI.size
Bio.Restriction.Restriction.CjuI.size
Bio.Restriction.Restriction.CjuII.size
Bio.Restriction.Restriction.ClaI.size
Bio.Restriction.Restriction.CpoI.size
Bio.Restriction.Restriction.CseI.size
Bio.Restriction.Restriction.CsiI.size
Bio.Restriction.Restriction.Csp6I.size
Bio.Restriction.Restriction.CspAI.size
Bio.Restriction.Restriction.CspCI.size
Bio.Restriction.Restriction.CspI.size
Bio.Restriction.Restriction.CstMI.size
Bio.Restriction.Restriction.CviAII.size
Bio.Restriction.Restriction.CviJI.size
Bio.Restriction.Restriction.CviKI_1.size
Bio.Restriction.Restriction.CviQI.size
Bio.Restriction.Restriction.CviRI.size
Bio.Restriction.Restriction.DdeI.size
Bio.Restriction.Restriction.DinI.size
Bio.Restriction.Restriction.DpnI.size
Bio.Restriction.Restriction.DpnII.size
Bio.Restriction.Restriction.DraI.size
Bio.Restriction.Restriction.DraII.size
Bio.Restriction.Restriction.DraIII.size
Bio.Restriction.Restriction.DraRI.size
Bio.Restriction.Restriction.DrdI.size
Bio.Restriction.Restriction.DrdII.size
Bio.Restriction.Restriction.DriI.size
Bio.Restriction.Restriction.DsaI.size
Bio.Restriction.Restriction.DseDI.size
Bio.Restriction.Restriction.EaeI.size
Bio.Restriction.Restriction.EagI.size
Bio.Restriction.Restriction.Eam1104I.size
Bio.Restriction.Restriction.Eam1105I.size
Bio.Restriction.Restriction.EarI.size
Bio.Restriction.Restriction.EciI.size
Bio.Restriction.Restriction.Ecl136II.size
Bio.Restriction.Restriction.EclXI.size
Bio.Restriction.Restriction.Eco105I.size
Bio.Restriction.Restriction.Eco130I.size
Bio.Restriction.Restriction.Eco147I.size
Bio.Restriction.Restriction.Eco24I.size
Bio.Restriction.Restriction.Eco31I.size
Bio.Restriction.Restriction.Eco32I.size
Bio.Restriction.Restriction.Eco47I.size
Bio.Restriction.Restriction.Eco47III.size
Bio.Restriction.Restriction.Eco52I.size
Bio.Restriction.Restriction.Eco53kI.size
Bio.Restriction.Restriction.Eco57I.size
Bio.Restriction.Restriction.Eco57MI.size
Bio.Restriction.Restriction.Eco72I.size
Bio.Restriction.Restriction.Eco81I.size
Bio.Restriction.Restriction.Eco88I.size
Bio.Restriction.Restriction.Eco91I.size
Bio.Restriction.Restriction.EcoHI.size
Bio.Restriction.Restriction.EcoICRI.size
Bio.Restriction.Restriction.EcoNI.size
Bio.Restriction.Restriction.EcoO109I.size
Bio.Restriction.Restriction.EcoO65I.size
Bio.Restriction.Restriction.EcoRI.size
Bio.Restriction.Restriction.EcoRII.size
Bio.Restriction.Restriction.EcoRV.size
Bio.Restriction.Restriction.EcoT14I.size
Bio.Restriction.Restriction.EcoT22I.size
Bio.Restriction.Restriction.EcoT38I.size
Bio.Restriction.Restriction.EgeI.size
Bio.Restriction.Restriction.EheI.size
Bio.Restriction.Restriction.ErhI.size
Bio.Restriction.Restriction.EsaBC3I.size
Bio.Restriction.Restriction.EsaSSI.size
Bio.Restriction.Restriction.Esp3I.size
Bio.Restriction.Restriction.EspI.size
Bio.Restriction.Restriction.FaeI.size
Bio.Restriction.Restriction.FaiI.size
Bio.Restriction.Restriction.FalI.size
Bio.Restriction.Restriction.FaqI.size
Bio.Restriction.Restriction.FatI.size
Bio.Restriction.Restriction.FauI.size
Bio.Restriction.Restriction.FauNDI.size
Bio.Restriction.Restriction.FbaI.size
Bio.Restriction.Restriction.FblI.size
Bio.Restriction.Restriction.FinI.size
Bio.Restriction.Restriction.FmuI.size
Bio.Restriction.Restriction.Fnu4HI.size
Bio.Restriction.Restriction.FnuDII.size
Bio.Restriction.Restriction.FokI.size
Bio.Restriction.Restriction.FriOI.size
Bio.Restriction.Restriction.FseI.size
Bio.Restriction.Restriction.Fsp4HI.size
Bio.Restriction.Restriction.FspAI.size
Bio.Restriction.Restriction.FspBI.size
Bio.Restriction.Restriction.FspEI.size
Bio.Restriction.Restriction.FspI.size
Bio.Restriction.Restriction.GauT27I.size
Bio.Restriction.Restriction.GdiII.size
Bio.Restriction.Restriction.GlaI.size
Bio.Restriction.Restriction.GluI.size
Bio.Restriction.Restriction.GsaI.size
Bio.Restriction.Restriction.GsuI.size
Bio.Restriction.Restriction.HaeI.size
Bio.Restriction.Restriction.HaeII.size
Bio.Restriction.Restriction.HaeIII.size
Bio.Restriction.Restriction.HapII.size
Bio.Restriction.Restriction.HauII.size
Bio.Restriction.Restriction.HgaI.size
Bio.Restriction.Restriction.HgiAI.size
Bio.Restriction.Restriction.HgiCI.size
Bio.Restriction.Restriction.HgiEII.size
Bio.Restriction.Restriction.HgiJII.size
Bio.Restriction.Restriction.HhaI.size
Bio.Restriction.Restriction.Hin1I.size
Bio.Restriction.Restriction.Hin1II.size
Bio.Restriction.Restriction.Hin4I.size
Bio.Restriction.Restriction.Hin4II.size
Bio.Restriction.Restriction.Hin6I.size
Bio.Restriction.Restriction.HinP1I.size
Bio.Restriction.Restriction.HincII.size
Bio.Restriction.Restriction.HindII.size
Bio.Restriction.Restriction.HindIII.size
Bio.Restriction.Restriction.HinfI.size
Bio.Restriction.Restriction.HpaI.size
Bio.Restriction.Restriction.HpaII.size
Bio.Restriction.Restriction.HphI.size
Bio.Restriction.Restriction.Hpy166II.size
Bio.Restriction.Restriction.Hpy178III.size
Bio.Restriction.Restriction.Hpy188I.size
Bio.Restriction.Restriction.Hpy188III.size
Bio.Restriction.Restriction.Hpy8I.size
Bio.Restriction.Restriction.Hpy99I.size
Bio.Restriction.Restriction.Hpy99XIII.size
Bio.Restriction.Restriction.Hpy99XIV.size
Bio.Restriction.Restriction.HpyAV.size
Bio.Restriction.Restriction.HpyCH4III.size
Bio.Restriction.Restriction.HpyCH4IV.size
Bio.Restriction.Restriction.HpyCH4V.size
Bio.Restriction.Restriction.HpyF10VI.size
Bio.Restriction.Restriction.HpyF3I.size
Bio.Restriction.Restriction.HpySE526I.size
Bio.Restriction.Restriction.Hsp92I.size
Bio.Restriction.Restriction.Hsp92II.size
Bio.Restriction.Restriction.HspAI.size
Bio.Restriction.Restriction.Jma19592I.size
Bio.Restriction.Restriction.KasI.size
Bio.Restriction.Restriction.KflI.size
Bio.Restriction.Restriction.Kpn2I.size
Bio.Restriction.Restriction.KpnI.size
Bio.Restriction.Restriction.KroI.size
Bio.Restriction.Restriction.Ksp22I.size
Bio.Restriction.Restriction.Ksp632I.size
Bio.Restriction.Restriction.KspAI.size
Bio.Restriction.Restriction.KspI.size
Bio.Restriction.Restriction.Kzo9I.size
Bio.Restriction.Restriction.LguI.size
Bio.Restriction.Restriction.LpnI.size
Bio.Restriction.Restriction.LpnPI.size
Bio.Restriction.Restriction.Lsp1109I.size
Bio.Restriction.Restriction.LweI.size
Bio.Restriction.Restriction.MabI.size
Bio.Restriction.Restriction.MaeI.size
Bio.Restriction.Restriction.MaeII.size
Bio.Restriction.Restriction.MaeIII.size
Bio.Restriction.Restriction.MalI.size
Bio.Restriction.Restriction.MaqI.size
Bio.Restriction.Restriction.MauBI.size
Bio.Restriction.Restriction.MbiI.size
Bio.Restriction.Restriction.MboI.size
Bio.Restriction.Restriction.MboII.size
Bio.Restriction.Restriction.McaTI.size
Bio.Restriction.Restriction.McrI.size
Bio.Restriction.Restriction.MfeI.size
Bio.Restriction.Restriction.MflI.size
Bio.Restriction.Restriction.MhlI.size
Bio.Restriction.Restriction.MjaIV.size
Bio.Restriction.Restriction.MkaDII.size
Bio.Restriction.Restriction.MlsI.size
Bio.Restriction.Restriction.MluCI.size
Bio.Restriction.Restriction.MluI.size
Bio.Restriction.Restriction.MluNI.size
Bio.Restriction.Restriction.Mly113I.size
Bio.Restriction.Restriction.MlyI.size
Bio.Restriction.Restriction.MmeI.size
Bio.Restriction.Restriction.MnlI.size
Bio.Restriction.Restriction.Mph1103I.size
Bio.Restriction.Restriction.MreI.size
Bio.Restriction.Restriction.MroI.size
Bio.Restriction.Restriction.MroNI.size
Bio.Restriction.Restriction.MroXI.size
Bio.Restriction.Restriction.MscI.size
Bio.Restriction.Restriction.MseI.size
Bio.Restriction.Restriction.MslI.size
Bio.Restriction.Restriction.Msp20I.size
Bio.Restriction.Restriction.MspA1I.size
Bio.Restriction.Restriction.MspCI.size
Bio.Restriction.Restriction.MspI.size
Bio.Restriction.Restriction.MspJI.size
Bio.Restriction.Restriction.MspR9I.size
Bio.Restriction.Restriction.MssI.size
Bio.Restriction.Restriction.MstI.size
Bio.Restriction.Restriction.MunI.size
Bio.Restriction.Restriction.Mva1269I.size
Bio.Restriction.Restriction.MvaI.size
Bio.Restriction.Restriction.MvnI.size
Bio.Restriction.Restriction.MvrI.size
Bio.Restriction.Restriction.MwoI.size
Bio.Restriction.Restriction.NaeI.size
Bio.Restriction.Restriction.NarI.size
Bio.Restriction.Restriction.NciI.size
Bio.Restriction.Restriction.NcoI.size
Bio.Restriction.Restriction.NdeI.size
Bio.Restriction.Restriction.NdeII.size
Bio.Restriction.Restriction.NgoAVIII.size
Bio.Restriction.Restriction.NgoMIV.size
Bio.Restriction.Restriction.NhaXI.size
Bio.Restriction.Restriction.NheI.size
Bio.Restriction.Restriction.NlaCI.size
Bio.Restriction.Restriction.NlaIII.size
Bio.Restriction.Restriction.NlaIV.size
Bio.Restriction.Restriction.Nli3877I.size
Bio.Restriction.Restriction.NmeAIII.size
Bio.Restriction.Restriction.NmeDI.size
Bio.Restriction.Restriction.NmuCI.size
Bio.Restriction.Restriction.NotI.size
Bio.Restriction.Restriction.NruI.size
Bio.Restriction.Restriction.NsbI.size
Bio.Restriction.Restriction.NsiI.size
Bio.Restriction.Restriction.NspBII.size
Bio.Restriction.Restriction.NspI.size
Bio.Restriction.Restriction.NspV.size
Bio.Restriction.Restriction.OliI.size
Bio.Restriction.Restriction.PabI.size
Bio.Restriction.Restriction.PacI.size
Bio.Restriction.Restriction.PaeI.size
Bio.Restriction.Restriction.PaeR7I.size
Bio.Restriction.Restriction.PagI.size
Bio.Restriction.Restriction.PalAI.size
Bio.Restriction.Restriction.PasI.size
Bio.Restriction.Restriction.PauI.size
Bio.Restriction.Restriction.PceI.size
Bio.Restriction.Restriction.PciI.size
Bio.Restriction.Restriction.PciSI.size
Bio.Restriction.Restriction.PcsI.size
Bio.Restriction.Restriction.PctI.size
Bio.Restriction.Restriction.PdiI.size
Bio.Restriction.Restriction.PdmI.size
Bio.Restriction.Restriction.PenI.size
Bio.Restriction.Restriction.PfeI.size
Bio.Restriction.Restriction.Pfl1108I.size
Bio.Restriction.Restriction.Pfl23II.size
Bio.Restriction.Restriction.PflFI.size
Bio.Restriction.Restriction.PflMI.size
Bio.Restriction.Restriction.PfoI.size
Bio.Restriction.Restriction.PinAI.size
Bio.Restriction.Restriction.PlaDI.size
Bio.Restriction.Restriction.Ple19I.size
Bio.Restriction.Restriction.PleI.size
Bio.Restriction.Restriction.PluTI.size
Bio.Restriction.Restriction.PmaCI.size
Bio.Restriction.Restriction.PmeI.size
Bio.Restriction.Restriction.PmlI.size
Bio.Restriction.Restriction.PpiI.size
Bio.Restriction.Restriction.PpsI.size
Bio.Restriction.Restriction.Ppu10I.size
Bio.Restriction.Restriction.Ppu21I.size
Bio.Restriction.Restriction.PpuMI.size
Bio.Restriction.Restriction.PscI.size
Bio.Restriction.Restriction.PshAI.size
Bio.Restriction.Restriction.PshBI.size
Bio.Restriction.Restriction.PsiI.size
Bio.Restriction.Restriction.Psp03I.size
Bio.Restriction.Restriction.Psp124BI.size
Bio.Restriction.Restriction.Psp1406I.size
Bio.Restriction.Restriction.Psp5II.size
Bio.Restriction.Restriction.Psp6I.size
Bio.Restriction.Restriction.PspCI.size
Bio.Restriction.Restriction.PspEI.size
Bio.Restriction.Restriction.PspGI.size
Bio.Restriction.Restriction.PspLI.size
Bio.Restriction.Restriction.PspN4I.size
Bio.Restriction.Restriction.PspOMI.size
Bio.Restriction.Restriction.PspOMII.size
Bio.Restriction.Restriction.PspPI.size
Bio.Restriction.Restriction.PspPPI.size
Bio.Restriction.Restriction.PspPRI.size
Bio.Restriction.Restriction.PspXI.size
Bio.Restriction.Restriction.PsrI.size
Bio.Restriction.Restriction.PssI.size
Bio.Restriction.Restriction.PstI.size
Bio.Restriction.Restriction.PstNI.size
Bio.Restriction.Restriction.PsuI.size
Bio.Restriction.Restriction.PsyI.size
Bio.Restriction.Restriction.PteI.size
Bio.Restriction.Restriction.PvuI.size
Bio.Restriction.Restriction.PvuII.size
Bio.Restriction.Restriction.R2_BceSIV.size
Bio.Restriction.Restriction.RceI.size
Bio.Restriction.Restriction.RdeGBI.size
Bio.Restriction.Restriction.RdeGBII.size
Bio.Restriction.Restriction.RdeGBIII.size
Bio.Restriction.Restriction.RflFIII.size
Bio.Restriction.Restriction.RgaI.size
Bio.Restriction.Restriction.RigI.size
Bio.Restriction.Restriction.RlaI.size
Bio.Restriction.Restriction.RleAI.size
Bio.Restriction.Restriction.RpaB5I.size
Bio.Restriction.Restriction.RpaBI.size
Bio.Restriction.Restriction.RpaI.size
Bio.Restriction.Restriction.RpaTI.size
Bio.Restriction.Restriction.RruI.size
Bio.Restriction.Restriction.RsaI.size
Bio.Restriction.Restriction.RsaNI.size
Bio.Restriction.Restriction.RseI.size
Bio.Restriction.Restriction.Rsr2I.size
Bio.Restriction.Restriction.RsrII.size
Bio.Restriction.Restriction.SacI.size
Bio.Restriction.Restriction.SacII.size
Bio.Restriction.Restriction.SalI.size
Bio.Restriction.Restriction.SanDI.size
Bio.Restriction.Restriction.SapI.size
Bio.Restriction.Restriction.SaqAI.size
Bio.Restriction.Restriction.SatI.size
Bio.Restriction.Restriction.Sau3AI.size
Bio.Restriction.Restriction.Sau96I.size
Bio.Restriction.Restriction.SauI.size
Bio.Restriction.Restriction.SbfI.size
Bio.Restriction.Restriction.ScaI.size
Bio.Restriction.Restriction.SchI.size
Bio.Restriction.Restriction.SciI.size
Bio.Restriction.Restriction.ScrFI.size
Bio.Restriction.Restriction.SdaI.size
Bio.Restriction.Restriction.SdeAI.size
Bio.Restriction.Restriction.SdeOSI.size
Bio.Restriction.Restriction.SduI.size
Bio.Restriction.Restriction.SecI.size
Bio.Restriction.Restriction.SelI.size
Bio.Restriction.Restriction.SetI.size
Bio.Restriction.Restriction.SexAI.size
Bio.Restriction.Restriction.SfaAI.size
Bio.Restriction.Restriction.SfaNI.size
Bio.Restriction.Restriction.SfcI.size
Bio.Restriction.Restriction.SfeI.size
Bio.Restriction.Restriction.SfiI.size
Bio.Restriction.Restriction.SfoI.size
Bio.Restriction.Restriction.Sfr274I.size
Bio.Restriction.Restriction.Sfr303I.size
Bio.Restriction.Restriction.SfuI.size
Bio.Restriction.Restriction.SgeI.size
Bio.Restriction.Restriction.SgfI.size
Bio.Restriction.Restriction.SgrAI.size
Bio.Restriction.Restriction.SgrBI.size
Bio.Restriction.Restriction.SgrDI.size
Bio.Restriction.Restriction.SgrTI.size
Bio.Restriction.Restriction.SgsI.size
Bio.Restriction.Restriction.SimI.size
Bio.Restriction.Restriction.SlaI.size
Bio.Restriction.Restriction.SmaI.size
Bio.Restriction.Restriction.SmiI.size
Bio.Restriction.Restriction.SmiMI.size
Bio.Restriction.Restriction.SmlI.size
Bio.Restriction.Restriction.SmoI.size
Bio.Restriction.Restriction.SnaBI.size
Bio.Restriction.Restriction.SnaI.size
Bio.Restriction.Restriction.Sno506I.size
Bio.Restriction.Restriction.SpeI.size
Bio.Restriction.Restriction.SphI.size
Bio.Restriction.Restriction.SplI.size
Bio.Restriction.Restriction.SpoDI.size
Bio.Restriction.Restriction.SrfI.size
Bio.Restriction.Restriction.Sse232I.size
Bio.Restriction.Restriction.Sse8387I.size
Bio.Restriction.Restriction.Sse8647I.size
Bio.Restriction.Restriction.Sse9I.size
Bio.Restriction.Restriction.SseBI.size
Bio.Restriction.Restriction.SsiI.size
Bio.Restriction.Restriction.SspD5I.size
Bio.Restriction.Restriction.SspDI.size
Bio.Restriction.Restriction.SspI.size
Bio.Restriction.Restriction.SstE37I.size
Bio.Restriction.Restriction.SstI.size
Bio.Restriction.Restriction.Sth132I.size
Bio.Restriction.Restriction.Sth302II.size
Bio.Restriction.Restriction.StrI.size
Bio.Restriction.Restriction.StsI.size
Bio.Restriction.Restriction.StuI.size
Bio.Restriction.Restriction.StyD4I.size
Bio.Restriction.Restriction.StyI.size
Bio.Restriction.Restriction.SwaI.size
Bio.Restriction.Restriction.TaaI.size
Bio.Restriction.Restriction.TaiI.size
Bio.Restriction.Restriction.TaqI.size
Bio.Restriction.Restriction.TaqII.size
Bio.Restriction.Restriction.TasI.size
Bio.Restriction.Restriction.TatI.size
Bio.Restriction.Restriction.TauI.size
Bio.Restriction.Restriction.TfiI.size
Bio.Restriction.Restriction.Tru1I.size
Bio.Restriction.Restriction.Tru9I.size
Bio.Restriction.Restriction.TscAI.size
Bio.Restriction.Restriction.TseFI.size
Bio.Restriction.Restriction.TseI.size
Bio.Restriction.Restriction.TsoI.size
Bio.Restriction.Restriction.Tsp45I.size
Bio.Restriction.Restriction.Tsp4CI.size
Bio.Restriction.Restriction.TspDTI.size
Bio.Restriction.Restriction.TspEI.size
Bio.Restriction.Restriction.TspGWI.size
Bio.Restriction.Restriction.TspMI.size
Bio.Restriction.Restriction.TspRI.size
Bio.Restriction.Restriction.TssI.size
Bio.Restriction.Restriction.TstI.size
Bio.Restriction.Restriction.TsuI.size
Bio.Restriction.Restriction.Tth111I.size
Bio.Restriction.Restriction.Tth111II.size
Bio.Restriction.Restriction.UbaF11I.size
Bio.Restriction.Restriction.UbaF12I.size
Bio.Restriction.Restriction.UbaF13I.size
Bio.Restriction.Restriction.UbaF14I.size
Bio.Restriction.Restriction.UbaF9I.size
Bio.Restriction.Restriction.UbaPI.size
Bio.Restriction.Restriction.UcoMSI.size
Bio.Restriction.Restriction.UnbI.size
Bio.Restriction.Restriction.Van91I.size
Bio.Restriction.Restriction.Vha464I.size
Bio.Restriction.Restriction.VneI.size
Bio.Restriction.Restriction.VpaK11AI.size
Bio.Restriction.Restriction.VpaK11BI.size
Bio.Restriction.Restriction.VspI.size
Bio.Restriction.Restriction.WviI.size
Bio.Restriction.Restriction.XagI.size
Bio.Restriction.Restriction.XapI.size
Bio.Restriction.Restriction.XbaI.size
Bio.Restriction.Restriction.XceI.size
Bio.Restriction.Restriction.XcmI.size
Bio.Restriction.Restriction.XhoI.size
Bio.Restriction.Restriction.XhoII.size
Bio.Restriction.Restriction.XmaI.size
Bio.Restriction.Restriction.XmaIII.size
Bio.Restriction.Restriction.XmaJI.size
Bio.Restriction.Restriction.XmiI.size
Bio.Restriction.Restriction.XmnI.size
Bio.Restriction.Restriction.XspI.size
Bio.Restriction.Restriction.YkrI.size
Bio.Restriction.Restriction.ZraI.size
Bio.Restriction.Restriction.ZrmI.size
Bio.Restriction.Restriction.Zsp2I.size
Bio.codonalign.codonalphabet.CodonAlphabet.size" class="py-name" href="#" onclick="return doclink('link-35', 'size', 'link-35');">size</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-36" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.residue_type()=Bio.GenBank._FeatureConsumer-class.html#residue_type,Method Bio.GenBank._RecordConsumer.residue_type()=Bio.GenBank._RecordConsumer-class.html#residue_type"><a title="Bio.GenBank._FeatureConsumer.residue_type
Bio.GenBank._RecordConsumer.residue_type" class="py-name" href="#" onclick="return doclink('link-36', 'residue_type', 'link-36');">residue_type</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-37" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.data_file_division()=Bio.GenBank._FeatureConsumer-class.html#data_file_division,Method Bio.GenBank._RecordConsumer.data_file_division()=Bio.GenBank._RecordConsumer-class.html#data_file_division"><a title="Bio.GenBank._FeatureConsumer.data_file_division
Bio.GenBank._RecordConsumer.data_file_division" class="py-name" href="#" onclick="return doclink('link-37', 'data_file_division', 'link-37');">data_file_division</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-38" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.date()=Bio.GenBank._FeatureConsumer-class.html#date,Method Bio.GenBank._RecordConsumer.date()=Bio.GenBank._RecordConsumer-class.html#date,Method Bio.Phylo.PhyloXMLIO.Parser.date()=Bio.Phylo.PhyloXMLIO.Parser-class.html#date,Method Bio.Phylo.PhyloXMLIO.Writer.date()=Bio.Phylo.PhyloXMLIO.Writer-class.html#date"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-38', 'date', 'link-38');">date</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-39" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.definition()=Bio.GenBank._FeatureConsumer-class.html#definition,Method Bio.GenBank._RecordConsumer.definition()=Bio.GenBank._RecordConsumer-class.html#definition"><a title="Bio.GenBank._FeatureConsumer.definition
Bio.GenBank._RecordConsumer.definition" class="py-name" href="#" onclick="return doclink('link-39', 'definition', 'link-39');">definition</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-40" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.accession()=Bio.GenBank._FeatureConsumer-class.html#accession,Method Bio.GenBank._RecordConsumer.accession()=Bio.GenBank._RecordConsumer-class.html#accession,Method Bio.Phylo.PhyloXMLIO.Parser.accession()=Bio.Phylo.PhyloXMLIO.Parser-class.html#accession,Method Bio.Phylo.PhyloXMLIO.Writer.accession()=Bio.Phylo.PhyloXMLIO.Writer-class.html#accession"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-40', 'accession', 'link-40');">accession</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-41" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.nid()=Bio.GenBank._FeatureConsumer-class.html#nid,Method Bio.GenBank._RecordConsumer.nid()=Bio.GenBank._RecordConsumer-class.html#nid"><a title="Bio.GenBank._FeatureConsumer.nid
Bio.GenBank._RecordConsumer.nid" class="py-name" href="#" onclick="return doclink('link-41', 'nid', 'link-41');">nid</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-42" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.pid()=Bio.GenBank._FeatureConsumer-class.html#pid,Method Bio.GenBank._RecordConsumer.pid()=Bio.GenBank._RecordConsumer-class.html#pid"><a title="Bio.GenBank._FeatureConsumer.pid
Bio.GenBank._RecordConsumer.pid" class="py-name" href="#" onclick="return doclink('link-42', 'pid', 'link-42');">pid</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-43" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HeaderConsumer.version()=Bio.Blast.NCBIStandalone._HeaderConsumer-class.html#version,Function Bio.Cluster.cluster.version()=Bio.Cluster.cluster-module.html#version,Method Bio.GenBank._FeatureConsumer.version()=Bio.GenBank._FeatureConsumer-class.html#version,Method Bio.GenBank._RecordConsumer.version()=Bio.GenBank._RecordConsumer-class.html#version,Variable Bio.motifs.jaspar.Motif.version=Bio.motifs.jaspar.Motif-class.html#version"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-43', 'version', 'link-43');">version</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">projects</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dblinks</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-44" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.db_source()=Bio.GenBank._FeatureConsumer-class.html#db_source,Method Bio.GenBank._RecordConsumer.db_source()=Bio.GenBank._RecordConsumer-class.html#db_source"><a title="Bio.GenBank._FeatureConsumer.db_source
Bio.GenBank._RecordConsumer.db_source" class="py-name" href="#" onclick="return doclink('link-44', 'db_source', 'link-44');">db_source</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-45" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.gi()=Bio.GenBank._FeatureConsumer-class.html#gi,Method Bio.GenBank._RecordConsumer.gi()=Bio.GenBank._RecordConsumer-class.html#gi"><a title="Bio.GenBank._FeatureConsumer.gi
Bio.GenBank._RecordConsumer.gi" class="py-name" href="#" onclick="return doclink('link-45', 'gi', 'link-45');">gi</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-46" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.keywords()=Bio.GenBank._FeatureConsumer-class.html#keywords,Method Bio.GenBank._RecordConsumer.keywords()=Bio.GenBank._RecordConsumer-class.html#keywords"><a title="Bio.GenBank._FeatureConsumer.keywords
Bio.GenBank._RecordConsumer.keywords" class="py-name" href="#" onclick="return doclink('link-46', 'keywords', 'link-46');">keywords</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-47" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.segment()=Bio.GenBank._FeatureConsumer-class.html#segment,Method Bio.GenBank._RecordConsumer.segment()=Bio.GenBank._RecordConsumer-class.html#segment"><a title="Bio.GenBank._FeatureConsumer.segment
Bio.GenBank._RecordConsumer.segment" class="py-name" href="#" onclick="return doclink('link-47', 'segment', 'link-47');">segment</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-48" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.source()=Bio.GenBank._FeatureConsumer-class.html#source,Method Bio.GenBank._RecordConsumer.source()=Bio.GenBank._RecordConsumer-class.html#source,Method Bio.Pathway.Network.source()=Bio.Pathway.Network-class.html#source"><a title="Bio.GenBank._FeatureConsumer.source
Bio.GenBank._RecordConsumer.source
Bio.Pathway.Network.source" class="py-name" href="#" onclick="return doclink('link-48', 'source', 'link-48');">source</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-49" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.organism()=Bio.GenBank._FeatureConsumer-class.html#organism,Method Bio.GenBank._RecordConsumer.organism()=Bio.GenBank._RecordConsumer-class.html#organism"><a title="Bio.GenBank._FeatureConsumer.organism
Bio.GenBank._RecordConsumer.organism" class="py-name" href="#" onclick="return doclink('link-49', 'organism', 'link-49');">organism</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-50" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.taxonomy()=Bio.GenBank._FeatureConsumer-class.html#taxonomy,Method Bio.GenBank._RecordConsumer.taxonomy()=Bio.GenBank._RecordConsumer-class.html#taxonomy,Variable Bio.Phylo.PhyloXML.Clade.taxonomy=Bio.Phylo.PhyloXML.Clade-class.html#taxonomy,Method Bio.Phylo.PhyloXMLIO.Writer.taxonomy()=Bio.Phylo.PhyloXMLIO.Writer-class.html#taxonomy"><a title="Bio.GenBank._FeatureConsumer.taxonomy
Bio.GenBank._RecordConsumer.taxonomy
Bio.Phylo.PhyloXML.Clade.taxonomy
Bio.Phylo.PhyloXMLIO.Writer.taxonomy" class="py-name" href="#" onclick="return doclink('link-50', 'taxonomy', 'link-50');">taxonomy</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">references</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-51" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.comment()=Bio.GenBank._FeatureConsumer-class.html#comment,Method Bio.GenBank._RecordConsumer.comment()=Bio.GenBank._RecordConsumer-class.html#comment"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-51', 'comment', 'link-51');">comment</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-52" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.features=BioSQL.BioSeq.DBSeqRecord-class.html#features"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-52', 'features', 'link-52');">features</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">base_counts</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">origin</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-53" class="py-name" targets="Method Bio.FSSP.FSSPAlignDict.sequence()=Bio.FSSP.FSSPAlignDict-class.html#sequence,Method Bio.GenBank._FeatureConsumer.sequence()=Bio.GenBank._FeatureConsumer-class.html#sequence,Method Bio.GenBank._RecordConsumer.sequence()=Bio.GenBank._RecordConsumer-class.html#sequence,Method Bio.Phylo.PhyloXMLIO.Writer.sequence()=Bio.Phylo.PhyloXMLIO.Writer-class.html#sequence"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-53', 'sequence', 'link-53');">sequence</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">contig</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-54" class="py-name" targets="Method Bio.GenBank._RecordConsumer.primary()=Bio.GenBank._RecordConsumer-class.html#primary"><a title="Bio.GenBank._RecordConsumer.primary" class="py-name" href="#" onclick="return doclink('link-54', 'primary', 'link-54');">primary</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-55" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.wgs()=Bio.GenBank._FeatureConsumer-class.html#wgs,Method Bio.GenBank._RecordConsumer.wgs()=Bio.GenBank._RecordConsumer-class.html#wgs"><a title="Bio.GenBank._FeatureConsumer.wgs
Bio.GenBank._RecordConsumer.wgs" class="py-name" href="#" onclick="return doclink('link-55', 'wgs', 'link-55');">wgs</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">wgs_scafld</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div><a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line"> </tt>
<a name="Record.__str__"></a><div id="Record.__str__-def"><a name="L200"></a><tt class="py-lineno">200</tt> <a class="py-toggle" href="#" id="Record.__str__-toggle" onclick="return toggle('Record.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Record-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record.__str__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record.__str__-expanded"><a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line">        <tt class="py-docstring">"""Provide a GenBank formatted output option for a Record.</tt> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line"><tt class="py-docstring">        The objective of this is to provide an easy way to read in a GenBank</tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line"><tt class="py-docstring">        record, modify it somehow, and then output it in 'GenBank format.'</tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line"><tt class="py-docstring">        We are striving to make this work so that a parsed Record that is</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line"><tt class="py-docstring">        output using this function will look exactly like the original</tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line"><tt class="py-docstring">        record.</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line"><tt class="py-docstring">        Much of the output is based on format description info at:</tt> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line"><tt class="py-docstring">        ftp://ncbi.nlm.nih.gov/genbank/gbrel.txt</tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-56" class="py-name" targets="Method Bio.GenBank.Record.Record._locus_line()=Bio.GenBank.Record.Record-class.html#_locus_line"><a title="Bio.GenBank.Record.Record._locus_line" class="py-name" href="#" onclick="return doclink('link-56', '_locus_line', 'link-56');">_locus_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-57" class="py-name" targets="Method Bio.GenBank.Record.Record._definition_line()=Bio.GenBank.Record.Record-class.html#_definition_line"><a title="Bio.GenBank.Record.Record._definition_line" class="py-name" href="#" onclick="return doclink('link-57', '_definition_line', 'link-57');">_definition_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-58" class="py-name" targets="Method Bio.GenBank.Record.Record._accession_line()=Bio.GenBank.Record.Record-class.html#_accession_line"><a title="Bio.GenBank.Record.Record._accession_line" class="py-name" href="#" onclick="return doclink('link-58', '_accession_line', 'link-58');">_accession_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-59" class="py-name" targets="Method Bio.GenBank.Record.Record._version_line()=Bio.GenBank.Record.Record-class.html#_version_line"><a title="Bio.GenBank.Record.Record._version_line" class="py-name" href="#" onclick="return doclink('link-59', '_version_line', 'link-59');">_version_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-60" class="py-name" targets="Method Bio.GenBank.Record.Record._project_line()=Bio.GenBank.Record.Record-class.html#_project_line"><a title="Bio.GenBank.Record.Record._project_line" class="py-name" href="#" onclick="return doclink('link-60', '_project_line', 'link-60');">_project_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-61" class="py-name" targets="Method Bio.GenBank.Record.Record._dblink_line()=Bio.GenBank.Record.Record-class.html#_dblink_line"><a title="Bio.GenBank.Record.Record._dblink_line" class="py-name" href="#" onclick="return doclink('link-61', '_dblink_line', 'link-61');">_dblink_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-62" class="py-name" targets="Method Bio.GenBank.Record.Record._nid_line()=Bio.GenBank.Record.Record-class.html#_nid_line"><a title="Bio.GenBank.Record.Record._nid_line" class="py-name" href="#" onclick="return doclink('link-62', '_nid_line', 'link-62');">_nid_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-63" class="py-name" targets="Method Bio.GenBank.Record.Record._pid_line()=Bio.GenBank.Record.Record-class.html#_pid_line"><a title="Bio.GenBank.Record.Record._pid_line" class="py-name" href="#" onclick="return doclink('link-63', '_pid_line', 'link-63');">_pid_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-64" class="py-name" targets="Method Bio.GenBank.Record.Record._keywords_line()=Bio.GenBank.Record.Record-class.html#_keywords_line"><a title="Bio.GenBank.Record.Record._keywords_line" class="py-name" href="#" onclick="return doclink('link-64', '_keywords_line', 'link-64');">_keywords_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-65" class="py-name" targets="Method Bio.GenBank.Record.Record._db_source_line()=Bio.GenBank.Record.Record-class.html#_db_source_line"><a title="Bio.GenBank.Record.Record._db_source_line" class="py-name" href="#" onclick="return doclink('link-65', '_db_source_line', 'link-65');">_db_source_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-66" class="py-name" targets="Method Bio.GenBank.Record.Record._segment_line()=Bio.GenBank.Record.Record-class.html#_segment_line"><a title="Bio.GenBank.Record.Record._segment_line" class="py-name" href="#" onclick="return doclink('link-66', '_segment_line', 'link-66');">_segment_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-67" class="py-name" targets="Method Bio.GenBank.Record.Record._source_line()=Bio.GenBank.Record.Record-class.html#_source_line"><a title="Bio.GenBank.Record.Record._source_line" class="py-name" href="#" onclick="return doclink('link-67', '_source_line', 'link-67');">_source_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-68" class="py-name" targets="Method Bio.GenBank.Record.Record._organism_line()=Bio.GenBank.Record.Record-class.html#_organism_line"><a title="Bio.GenBank.Record.Record._organism_line" class="py-name" href="#" onclick="return doclink('link-68', '_organism_line', 'link-68');">_organism_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-69" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HeaderConsumer.reference()=Bio.Blast.NCBIStandalone._HeaderConsumer-class.html#reference,Method Bio.Phylo.PhyloXMLIO.Parser.reference()=Bio.Phylo.PhyloXMLIO.Parser-class.html#reference,Method Bio.Phylo.PhyloXMLIO.Writer.reference()=Bio.Phylo.PhyloXMLIO.Writer-class.html#reference"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-69', 'reference', 'link-69');">reference</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">references</tt><tt class="py-op">:</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-70" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-70', 'reference', 'link-69');">reference</a></tt><tt class="py-op">)</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-71" class="py-name" targets="Method Bio.GenBank.Record.Record._comment_line()=Bio.GenBank.Record.Record-class.html#_comment_line"><a title="Bio.GenBank.Record.Record._comment_line" class="py-name" href="#" onclick="return doclink('link-71', '_comment_line', 'link-71');">_comment_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-72" class="py-name" targets="Method Bio.GenBank.Record.Record._features_line()=Bio.GenBank.Record.Record-class.html#_features_line"><a title="Bio.GenBank.Record.Record._features_line" class="py-name" href="#" onclick="return doclink('link-72', '_features_line', 'link-72');">_features_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">feature</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-73" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-73', 'features', 'link-52');">features</a></tt><tt class="py-op">:</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">feature</tt><tt class="py-op">)</tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-74" class="py-name" targets="Method Bio.GenBank.Record.Record._base_count_line()=Bio.GenBank.Record.Record-class.html#_base_count_line"><a title="Bio.GenBank.Record.Record._base_count_line" class="py-name" href="#" onclick="return doclink('link-74', '_base_count_line', 'link-74');">_base_count_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-75" class="py-name" targets="Method Bio.GenBank.Record.Record._origin_line()=Bio.GenBank.Record.Record-class.html#_origin_line"><a title="Bio.GenBank.Record.Record._origin_line" class="py-name" href="#" onclick="return doclink('link-75', '_origin_line', 'link-75');">_origin_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-76" class="py-name" targets="Method Bio.GenBank.Record.Record._sequence_line()=Bio.GenBank.Record.Record-class.html#_sequence_line"><a title="Bio.GenBank.Record.Record._sequence_line" class="py-name" href="#" onclick="return doclink('link-76', '_sequence_line', 'link-76');">_sequence_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-77" class="py-name" targets="Method Bio.GenBank.Record.Record._wgs_line()=Bio.GenBank.Record.Record-class.html#_wgs_line"><a title="Bio.GenBank.Record.Record._wgs_line" class="py-name" href="#" onclick="return doclink('link-77', '_wgs_line', 'link-77');">_wgs_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-78" class="py-name" targets="Method Bio.GenBank.Record.Record._wgs_scafld_line()=Bio.GenBank.Record.Record-class.html#_wgs_scafld_line"><a title="Bio.GenBank.Record.Record._wgs_scafld_line" class="py-name" href="#" onclick="return doclink('link-78', '_wgs_scafld_line', 'link-78');">_wgs_scafld_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-79" class="py-name" targets="Method Bio.GenBank.Record.Record._contig_line()=Bio.GenBank.Record.Record-class.html#_contig_line"><a title="Bio.GenBank.Record.Record._contig_line" class="py-name" href="#" onclick="return doclink('link-79', '_contig_line', 'link-79');">_contig_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">"//"</tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line"> </tt>
<a name="Record._locus_line"></a><div id="Record._locus_line-def"><a name="L241"></a><tt class="py-lineno">241</tt> <a class="py-toggle" href="#" id="Record._locus_line-toggle" onclick="return toggle('Record._locus_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Record-class.html#_locus_line">_locus_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record._locus_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record._locus_line-expanded"><a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line">        <tt class="py-docstring">"""Provide the output string for the LOCUS line.</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-string">"LOCUS"</tt> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt class="py-number">7</tt>  <tt class="py-comment"># 6-12 spaces</tt> </tt>
<a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">"%-9s"</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-80" class="py-name"><a title="Bio.GenBank._FeatureConsumer.locus
Bio.GenBank._RecordConsumer.locus" class="py-name" href="#" onclick="return doclink('link-80', 'locus', 'link-34');">locus</a></tt> </tt>
<a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">" "</tt>  <tt class="py-comment"># 22 space</tt> </tt>
<a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">"%7s"</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-81" class="py-name"><a title="Bio.Alphabet.Alphabet.size
Bio.Alphabet.Reduced.HPModel.size
Bio.Alphabet.Reduced.Murphy10.size
Bio.Alphabet.Reduced.Murphy15.size
Bio.Alphabet.Reduced.Murphy4.size
Bio.Alphabet.Reduced.Murphy8.size
Bio.Alphabet.Reduced.PC5.size
Bio.Alphabet.SingleLetterAlphabet.size
Bio.Alphabet.ThreeLetterProtein.size
Bio.GenBank._FeatureConsumer.size
Bio.GenBank._RecordConsumer.size
Bio.Restriction.Restriction.AanI.size
Bio.Restriction.Restriction.AarI.size
Bio.Restriction.Restriction.AasI.size
Bio.Restriction.Restriction.AatII.size
Bio.Restriction.Restriction.AbaSI.size
Bio.Restriction.Restriction.AbsI.size
Bio.Restriction.Restriction.Acc16I.size
Bio.Restriction.Restriction.Acc36I.size
Bio.Restriction.Restriction.Acc65I.size
Bio.Restriction.Restriction.AccB1I.size
Bio.Restriction.Restriction.AccB7I.size
Bio.Restriction.Restriction.AccBSI.size
Bio.Restriction.Restriction.AccI.size
Bio.Restriction.Restriction.AccII.size
Bio.Restriction.Restriction.AccIII.size
Bio.Restriction.Restriction.AceIII.size
Bio.Restriction.Restriction.AciI.size
Bio.Restriction.Restriction.AclI.size
Bio.Restriction.Restriction.AclWI.size
Bio.Restriction.Restriction.AcoI.size
Bio.Restriction.Restriction.AcsI.size
Bio.Restriction.Restriction.AcuI.size
Bio.Restriction.Restriction.AcvI.size
Bio.Restriction.Restriction.AcyI.size
Bio.Restriction.Restriction.AdeI.size
Bio.Restriction.Restriction.AfaI.size
Bio.Restriction.Restriction.AfeI.size
Bio.Restriction.Restriction.AfiI.size
Bio.Restriction.Restriction.AflII.size
Bio.Restriction.Restriction.AflIII.size
Bio.Restriction.Restriction.AgeI.size
Bio.Restriction.Restriction.AgsI.size
Bio.Restriction.Restriction.AhaIII.size
Bio.Restriction.Restriction.AhdI.size
Bio.Restriction.Restriction.AhlI.size
Bio.Restriction.Restriction.AjiI.size
Bio.Restriction.Restriction.AjnI.size
Bio.Restriction.Restriction.AjuI.size
Bio.Restriction.Restriction.AleI.size
Bio.Restriction.Restriction.AlfI.size
Bio.Restriction.Restriction.AloI.size
Bio.Restriction.Restriction.AluBI.size
Bio.Restriction.Restriction.AluI.size
Bio.Restriction.Restriction.Alw21I.size
Bio.Restriction.Restriction.Alw26I.size
Bio.Restriction.Restriction.Alw44I.size
Bio.Restriction.Restriction.AlwFI.size
Bio.Restriction.Restriction.AlwI.size
Bio.Restriction.Restriction.AlwNI.size
Bio.Restriction.Restriction.Ama87I.size
Bio.Restriction.Restriction.Aor13HI.size
Bio.Restriction.Restriction.Aor51HI.size
Bio.Restriction.Restriction.AoxI.size
Bio.Restriction.Restriction.ApaBI.size
Bio.Restriction.Restriction.ApaI.size
Bio.Restriction.Restriction.ApaLI.size
Bio.Restriction.Restriction.ApeKI.size
Bio.Restriction.Restriction.ApoI.size
Bio.Restriction.Restriction.ApyPI.size
Bio.Restriction.Restriction.AquII.size
Bio.Restriction.Restriction.AquIII.size
Bio.Restriction.Restriction.AquIV.size
Bio.Restriction.Restriction.ArsI.size
Bio.Restriction.Restriction.AscI.size
Bio.Restriction.Restriction.AseI.size
Bio.Restriction.Restriction.Asi256I.size
Bio.Restriction.Restriction.AsiGI.size
Bio.Restriction.Restriction.AsiSI.size
Bio.Restriction.Restriction.Asp700I.size
Bio.Restriction.Restriction.Asp718I.size
Bio.Restriction.Restriction.AspA2I.size
Bio.Restriction.Restriction.AspBHI.size
Bio.Restriction.Restriction.AspLEI.size
Bio.Restriction.Restriction.AspS9I.size
Bio.Restriction.Restriction.AssI.size
Bio.Restriction.Restriction.AsuC2I.size
Bio.Restriction.Restriction.AsuHPI.size
Bio.Restriction.Restriction.AsuI.size
Bio.Restriction.Restriction.AsuII.size
Bio.Restriction.Restriction.AsuNHI.size
Bio.Restriction.Restriction.AvaI.size
Bio.Restriction.Restriction.AvaII.size
Bio.Restriction.Restriction.AvaIII.size
Bio.Restriction.Restriction.AvrII.size
Bio.Restriction.Restriction.AxyI.size
Bio.Restriction.Restriction.BaeGI.size
Bio.Restriction.Restriction.BaeI.size
Bio.Restriction.Restriction.BalI.size
Bio.Restriction.Restriction.BamHI.size
Bio.Restriction.Restriction.BanI.size
Bio.Restriction.Restriction.BanII.size
Bio.Restriction.Restriction.BarI.size
Bio.Restriction.Restriction.BasI.size
Bio.Restriction.Restriction.BauI.size
Bio.Restriction.Restriction.Bbr7I.size
Bio.Restriction.Restriction.BbrPI.size
Bio.Restriction.Restriction.BbsI.size
Bio.Restriction.Restriction.Bbv12I.size
Bio.Restriction.Restriction.BbvCI.size
Bio.Restriction.Restriction.BbvI.size
Bio.Restriction.Restriction.BbvII.size
Bio.Restriction.Restriction.BccI.size
Bio.Restriction.Restriction.Bce83I.size
Bio.Restriction.Restriction.BceAI.size
Bio.Restriction.Restriction.BcefI.size
Bio.Restriction.Restriction.BcgI.size
Bio.Restriction.Restriction.BciT130I.size
Bio.Restriction.Restriction.BciVI.size
Bio.Restriction.Restriction.BclI.size
Bio.Restriction.Restriction.BcnI.size
Bio.Restriction.Restriction.BcoDI.size
Bio.Restriction.Restriction.BcuI.size
Bio.Restriction.Restriction.BdaI.size
Bio.Restriction.Restriction.BetI.size
Bio.Restriction.Restriction.BfaI.size
Bio.Restriction.Restriction.BfiI.size
Bio.Restriction.Restriction.BfmI.size
Bio.Restriction.Restriction.BfoI.size
Bio.Restriction.Restriction.BfrI.size
Bio.Restriction.Restriction.BfuAI.size
Bio.Restriction.Restriction.BfuCI.size
Bio.Restriction.Restriction.BfuI.size
Bio.Restriction.Restriction.BglI.size
Bio.Restriction.Restriction.BglII.size
Bio.Restriction.Restriction.BinI.size
Bio.Restriction.Restriction.BisI.size
Bio.Restriction.Restriction.BlnI.size
Bio.Restriction.Restriction.BlpI.size
Bio.Restriction.Restriction.BlsI.size
Bio.Restriction.Restriction.BmcAI.size
Bio.Restriction.Restriction.Bme1390I.size
Bio.Restriction.Restriction.Bme18I.size
Bio.Restriction.Restriction.BmeDI.size
Bio.Restriction.Restriction.BmeRI.size
Bio.Restriction.Restriction.BmeT110I.size
Bio.Restriction.Restriction.BmgBI.size
Bio.Restriction.Restriction.BmgI.size
Bio.Restriction.Restriction.BmgT120I.size
Bio.Restriction.Restriction.BmiI.size
Bio.Restriction.Restriction.BmrFI.size
Bio.Restriction.Restriction.BmrI.size
Bio.Restriction.Restriction.BmsI.size
Bio.Restriction.Restriction.BmtI.size
Bio.Restriction.Restriction.BmuI.size
Bio.Restriction.Restriction.BoxI.size
Bio.Restriction.Restriction.BpiI.size
Bio.Restriction.Restriction.BplI.size
Bio.Restriction.Restriction.BpmI.size
Bio.Restriction.Restriction.Bpu10I.size
Bio.Restriction.Restriction.Bpu1102I.size
Bio.Restriction.Restriction.Bpu14I.size
Bio.Restriction.Restriction.BpuEI.size
Bio.Restriction.Restriction.BpuMI.size
Bio.Restriction.Restriction.BpvUI.size
Bio.Restriction.Restriction.Bsa29I.size
Bio.Restriction.Restriction.BsaAI.size
Bio.Restriction.Restriction.BsaBI.size
Bio.Restriction.Restriction.BsaHI.size
Bio.Restriction.Restriction.BsaI.size
Bio.Restriction.Restriction.BsaJI.size
Bio.Restriction.Restriction.BsaWI.size
Bio.Restriction.Restriction.BsaXI.size
Bio.Restriction.Restriction.BsbI.size
Bio.Restriction.Restriction.Bsc4I.size
Bio.Restriction.Restriction.BscAI.size
Bio.Restriction.Restriction.BscGI.size
Bio.Restriction.Restriction.Bse118I.size
Bio.Restriction.Restriction.Bse1I.size
Bio.Restriction.Restriction.Bse21I.size
Bio.Restriction.Restriction.Bse3DI.size
Bio.Restriction.Restriction.Bse8I.size
Bio.Restriction.Restriction.BseAI.size
Bio.Restriction.Restriction.BseBI.size
Bio.Restriction.Restriction.BseCI.size
Bio.Restriction.Restriction.BseDI.size
Bio.Restriction.Restriction.BseGI.size
Bio.Restriction.Restriction.BseJI.size
Bio.Restriction.Restriction.BseLI.size
Bio.Restriction.Restriction.BseMI.size
Bio.Restriction.Restriction.BseMII.size
Bio.Restriction.Restriction.BseNI.size
Bio.Restriction.Restriction.BsePI.size
Bio.Restriction.Restriction.BseRI.size
Bio.Restriction.Restriction.BseSI.size
Bio.Restriction.Restriction.BseX3I.size
Bio.Restriction.Restriction.BseXI.size
Bio.Restriction.Restriction.BseYI.size
Bio.Restriction.Restriction.BsgI.size
Bio.Restriction.Restriction.Bsh1236I.size
Bio.Restriction.Restriction.Bsh1285I.size
Bio.Restriction.Restriction.BshFI.size
Bio.Restriction.Restriction.BshNI.size
Bio.Restriction.Restriction.BshTI.size
Bio.Restriction.Restriction.BshVI.size
Bio.Restriction.Restriction.BsiEI.size
Bio.Restriction.Restriction.BsiHKAI.size
Bio.Restriction.Restriction.BsiHKCI.size
Bio.Restriction.Restriction.BsiI.size
Bio.Restriction.Restriction.BsiSI.size
Bio.Restriction.Restriction.BsiWI.size
Bio.Restriction.Restriction.BsiYI.size
Bio.Restriction.Restriction.BslFI.size
Bio.Restriction.Restriction.BslI.size
Bio.Restriction.Restriction.BsmAI.size
Bio.Restriction.Restriction.BsmBI.size
Bio.Restriction.Restriction.BsmFI.size
Bio.Restriction.Restriction.BsmI.size
Bio.Restriction.Restriction.BsnI.size
Bio.Restriction.Restriction.Bso31I.size
Bio.Restriction.Restriction.BsoBI.size
Bio.Restriction.Restriction.Bsp119I.size
Bio.Restriction.Restriction.Bsp120I.size
Bio.Restriction.Restriction.Bsp1286I.size
Bio.Restriction.Restriction.Bsp13I.size
Bio.Restriction.Restriction.Bsp1407I.size
Bio.Restriction.Restriction.Bsp143I.size
Bio.Restriction.Restriction.Bsp1720I.size
Bio.Restriction.Restriction.Bsp19I.size
Bio.Restriction.Restriction.Bsp24I.size
Bio.Restriction.Restriction.Bsp68I.size
Bio.Restriction.Restriction.BspACI.size
Bio.Restriction.Restriction.BspCNI.size
Bio.Restriction.Restriction.BspD6I.size
Bio.Restriction.Restriction.BspDI.size
Bio.Restriction.Restriction.BspEI.size
Bio.Restriction.Restriction.BspFNI.size
Bio.Restriction.Restriction.BspGI.size
Bio.Restriction.Restriction.BspHI.size
Bio.Restriction.Restriction.BspLI.size
Bio.Restriction.Restriction.BspLU11I.size
Bio.Restriction.Restriction.BspMI.size
Bio.Restriction.Restriction.BspMII.size
Bio.Restriction.Restriction.BspNCI.size
Bio.Restriction.Restriction.BspOI.size
Bio.Restriction.Restriction.BspPI.size
Bio.Restriction.Restriction.BspQI.size
Bio.Restriction.Restriction.BspT104I.size
Bio.Restriction.Restriction.BspT107I.size
Bio.Restriction.Restriction.BspTI.size
Bio.Restriction.Restriction.BsrBI.size
Bio.Restriction.Restriction.BsrDI.size
Bio.Restriction.Restriction.BsrFI.size
Bio.Restriction.Restriction.BsrGI.size
Bio.Restriction.Restriction.BsrI.size
Bio.Restriction.Restriction.BsrSI.size
Bio.Restriction.Restriction.BssAI.size
Bio.Restriction.Restriction.BssECI.size
Bio.Restriction.Restriction.BssHII.size
Bio.Restriction.Restriction.BssKI.size
Bio.Restriction.Restriction.BssMI.size
Bio.Restriction.Restriction.BssNAI.size
Bio.Restriction.Restriction.BssNI.size
Bio.Restriction.Restriction.BssSI.size
Bio.Restriction.Restriction.BssT1I.size
Bio.Restriction.Restriction.Bst1107I.size
Bio.Restriction.Restriction.Bst2BI.size
Bio.Restriction.Restriction.Bst2UI.size
Bio.Restriction.Restriction.Bst4CI.size
Bio.Restriction.Restriction.Bst6I.size
Bio.Restriction.Restriction.BstACI.size
Bio.Restriction.Restriction.BstAFI.size
Bio.Restriction.Restriction.BstAPI.size
Bio.Restriction.Restriction.BstAUI.size
Bio.Restriction.Restriction.BstBAI.size
Bio.Restriction.Restriction.BstBI.size
Bio.Restriction.Restriction.BstC8I.size
Bio.Restriction.Restriction.BstDEI.size
Bio.Restriction.Restriction.BstDSI.size
Bio.Restriction.Restriction.BstEII.size
Bio.Restriction.Restriction.BstENI.size
Bio.Restriction.Restriction.BstF5I.size
Bio.Restriction.Restriction.BstFNI.size
Bio.Restriction.Restriction.BstH2I.size
Bio.Restriction.Restriction.BstHHI.size
Bio.Restriction.Restriction.BstKTI.size
Bio.Restriction.Restriction.BstMAI.size
Bio.Restriction.Restriction.BstMBI.size
Bio.Restriction.Restriction.BstMCI.size
Bio.Restriction.Restriction.BstMWI.size
Bio.Restriction.Restriction.BstNI.size
Bio.Restriction.Restriction.BstNSI.size
Bio.Restriction.Restriction.BstOI.size
Bio.Restriction.Restriction.BstPAI.size
Bio.Restriction.Restriction.BstPI.size
Bio.Restriction.Restriction.BstSCI.size
Bio.Restriction.Restriction.BstSFI.size
Bio.Restriction.Restriction.BstSLI.size
Bio.Restriction.Restriction.BstSNI.size
Bio.Restriction.Restriction.BstUI.size
Bio.Restriction.Restriction.BstV1I.size
Bio.Restriction.Restriction.BstV2I.size
Bio.Restriction.Restriction.BstX2I.size
Bio.Restriction.Restriction.BstXI.size
Bio.Restriction.Restriction.BstYI.size
Bio.Restriction.Restriction.BstZ17I.size
Bio.Restriction.Restriction.BstZI.size
Bio.Restriction.Restriction.Bsu15I.size
Bio.Restriction.Restriction.Bsu36I.size
Bio.Restriction.Restriction.BsuI.size
Bio.Restriction.Restriction.BsuRI.size
Bio.Restriction.Restriction.BtgI.size
Bio.Restriction.Restriction.BtgZI.size
Bio.Restriction.Restriction.BthCI.size
Bio.Restriction.Restriction.BtrI.size
Bio.Restriction.Restriction.BtsCI.size
Bio.Restriction.Restriction.BtsI.size
Bio.Restriction.Restriction.BtsIMutI.size
Bio.Restriction.Restriction.BtuMI.size
Bio.Restriction.Restriction.BveI.size
Bio.Restriction.Restriction.Cac8I.size
Bio.Restriction.Restriction.CaiI.size
Bio.Restriction.Restriction.CauII.size
Bio.Restriction.Restriction.CchII.size
Bio.Restriction.Restriction.CchIII.size
Bio.Restriction.Restriction.CciI.size
Bio.Restriction.Restriction.CciNI.size
Bio.Restriction.Restriction.Cdi630V.size
Bio.Restriction.Restriction.CdiI.size
Bio.Restriction.Restriction.CdpI.size
Bio.Restriction.Restriction.CfoI.size
Bio.Restriction.Restriction.Cfr10I.size
Bio.Restriction.Restriction.Cfr13I.size
Bio.Restriction.Restriction.Cfr42I.size
Bio.Restriction.Restriction.Cfr9I.size
Bio.Restriction.Restriction.CfrI.size
Bio.Restriction.Restriction.Cgl13032I.size
Bio.Restriction.Restriction.Cgl13032II.size
Bio.Restriction.Restriction.ChaI.size
Bio.Restriction.Restriction.CjeFIII.size
Bio.Restriction.Restriction.CjeFV.size
Bio.Restriction.Restriction.CjeI.size
Bio.Restriction.Restriction.CjeNII.size
Bio.Restriction.Restriction.CjeNIII.size
Bio.Restriction.Restriction.CjeP659IV.size
Bio.Restriction.Restriction.CjePI.size
Bio.Restriction.Restriction.CjuI.size
Bio.Restriction.Restriction.CjuII.size
Bio.Restriction.Restriction.ClaI.size
Bio.Restriction.Restriction.CpoI.size
Bio.Restriction.Restriction.CseI.size
Bio.Restriction.Restriction.CsiI.size
Bio.Restriction.Restriction.Csp6I.size
Bio.Restriction.Restriction.CspAI.size
Bio.Restriction.Restriction.CspCI.size
Bio.Restriction.Restriction.CspI.size
Bio.Restriction.Restriction.CstMI.size
Bio.Restriction.Restriction.CviAII.size
Bio.Restriction.Restriction.CviJI.size
Bio.Restriction.Restriction.CviKI_1.size
Bio.Restriction.Restriction.CviQI.size
Bio.Restriction.Restriction.CviRI.size
Bio.Restriction.Restriction.DdeI.size
Bio.Restriction.Restriction.DinI.size
Bio.Restriction.Restriction.DpnI.size
Bio.Restriction.Restriction.DpnII.size
Bio.Restriction.Restriction.DraI.size
Bio.Restriction.Restriction.DraII.size
Bio.Restriction.Restriction.DraIII.size
Bio.Restriction.Restriction.DraRI.size
Bio.Restriction.Restriction.DrdI.size
Bio.Restriction.Restriction.DrdII.size
Bio.Restriction.Restriction.DriI.size
Bio.Restriction.Restriction.DsaI.size
Bio.Restriction.Restriction.DseDI.size
Bio.Restriction.Restriction.EaeI.size
Bio.Restriction.Restriction.EagI.size
Bio.Restriction.Restriction.Eam1104I.size
Bio.Restriction.Restriction.Eam1105I.size
Bio.Restriction.Restriction.EarI.size
Bio.Restriction.Restriction.EciI.size
Bio.Restriction.Restriction.Ecl136II.size
Bio.Restriction.Restriction.EclXI.size
Bio.Restriction.Restriction.Eco105I.size
Bio.Restriction.Restriction.Eco130I.size
Bio.Restriction.Restriction.Eco147I.size
Bio.Restriction.Restriction.Eco24I.size
Bio.Restriction.Restriction.Eco31I.size
Bio.Restriction.Restriction.Eco32I.size
Bio.Restriction.Restriction.Eco47I.size
Bio.Restriction.Restriction.Eco47III.size
Bio.Restriction.Restriction.Eco52I.size
Bio.Restriction.Restriction.Eco53kI.size
Bio.Restriction.Restriction.Eco57I.size
Bio.Restriction.Restriction.Eco57MI.size
Bio.Restriction.Restriction.Eco72I.size
Bio.Restriction.Restriction.Eco81I.size
Bio.Restriction.Restriction.Eco88I.size
Bio.Restriction.Restriction.Eco91I.size
Bio.Restriction.Restriction.EcoHI.size
Bio.Restriction.Restriction.EcoICRI.size
Bio.Restriction.Restriction.EcoNI.size
Bio.Restriction.Restriction.EcoO109I.size
Bio.Restriction.Restriction.EcoO65I.size
Bio.Restriction.Restriction.EcoRI.size
Bio.Restriction.Restriction.EcoRII.size
Bio.Restriction.Restriction.EcoRV.size
Bio.Restriction.Restriction.EcoT14I.size
Bio.Restriction.Restriction.EcoT22I.size
Bio.Restriction.Restriction.EcoT38I.size
Bio.Restriction.Restriction.EgeI.size
Bio.Restriction.Restriction.EheI.size
Bio.Restriction.Restriction.ErhI.size
Bio.Restriction.Restriction.EsaBC3I.size
Bio.Restriction.Restriction.EsaSSI.size
Bio.Restriction.Restriction.Esp3I.size
Bio.Restriction.Restriction.EspI.size
Bio.Restriction.Restriction.FaeI.size
Bio.Restriction.Restriction.FaiI.size
Bio.Restriction.Restriction.FalI.size
Bio.Restriction.Restriction.FaqI.size
Bio.Restriction.Restriction.FatI.size
Bio.Restriction.Restriction.FauI.size
Bio.Restriction.Restriction.FauNDI.size
Bio.Restriction.Restriction.FbaI.size
Bio.Restriction.Restriction.FblI.size
Bio.Restriction.Restriction.FinI.size
Bio.Restriction.Restriction.FmuI.size
Bio.Restriction.Restriction.Fnu4HI.size
Bio.Restriction.Restriction.FnuDII.size
Bio.Restriction.Restriction.FokI.size
Bio.Restriction.Restriction.FriOI.size
Bio.Restriction.Restriction.FseI.size
Bio.Restriction.Restriction.Fsp4HI.size
Bio.Restriction.Restriction.FspAI.size
Bio.Restriction.Restriction.FspBI.size
Bio.Restriction.Restriction.FspEI.size
Bio.Restriction.Restriction.FspI.size
Bio.Restriction.Restriction.GauT27I.size
Bio.Restriction.Restriction.GdiII.size
Bio.Restriction.Restriction.GlaI.size
Bio.Restriction.Restriction.GluI.size
Bio.Restriction.Restriction.GsaI.size
Bio.Restriction.Restriction.GsuI.size
Bio.Restriction.Restriction.HaeI.size
Bio.Restriction.Restriction.HaeII.size
Bio.Restriction.Restriction.HaeIII.size
Bio.Restriction.Restriction.HapII.size
Bio.Restriction.Restriction.HauII.size
Bio.Restriction.Restriction.HgaI.size
Bio.Restriction.Restriction.HgiAI.size
Bio.Restriction.Restriction.HgiCI.size
Bio.Restriction.Restriction.HgiEII.size
Bio.Restriction.Restriction.HgiJII.size
Bio.Restriction.Restriction.HhaI.size
Bio.Restriction.Restriction.Hin1I.size
Bio.Restriction.Restriction.Hin1II.size
Bio.Restriction.Restriction.Hin4I.size
Bio.Restriction.Restriction.Hin4II.size
Bio.Restriction.Restriction.Hin6I.size
Bio.Restriction.Restriction.HinP1I.size
Bio.Restriction.Restriction.HincII.size
Bio.Restriction.Restriction.HindII.size
Bio.Restriction.Restriction.HindIII.size
Bio.Restriction.Restriction.HinfI.size
Bio.Restriction.Restriction.HpaI.size
Bio.Restriction.Restriction.HpaII.size
Bio.Restriction.Restriction.HphI.size
Bio.Restriction.Restriction.Hpy166II.size
Bio.Restriction.Restriction.Hpy178III.size
Bio.Restriction.Restriction.Hpy188I.size
Bio.Restriction.Restriction.Hpy188III.size
Bio.Restriction.Restriction.Hpy8I.size
Bio.Restriction.Restriction.Hpy99I.size
Bio.Restriction.Restriction.Hpy99XIII.size
Bio.Restriction.Restriction.Hpy99XIV.size
Bio.Restriction.Restriction.HpyAV.size
Bio.Restriction.Restriction.HpyCH4III.size
Bio.Restriction.Restriction.HpyCH4IV.size
Bio.Restriction.Restriction.HpyCH4V.size
Bio.Restriction.Restriction.HpyF10VI.size
Bio.Restriction.Restriction.HpyF3I.size
Bio.Restriction.Restriction.HpySE526I.size
Bio.Restriction.Restriction.Hsp92I.size
Bio.Restriction.Restriction.Hsp92II.size
Bio.Restriction.Restriction.HspAI.size
Bio.Restriction.Restriction.Jma19592I.size
Bio.Restriction.Restriction.KasI.size
Bio.Restriction.Restriction.KflI.size
Bio.Restriction.Restriction.Kpn2I.size
Bio.Restriction.Restriction.KpnI.size
Bio.Restriction.Restriction.KroI.size
Bio.Restriction.Restriction.Ksp22I.size
Bio.Restriction.Restriction.Ksp632I.size
Bio.Restriction.Restriction.KspAI.size
Bio.Restriction.Restriction.KspI.size
Bio.Restriction.Restriction.Kzo9I.size
Bio.Restriction.Restriction.LguI.size
Bio.Restriction.Restriction.LpnI.size
Bio.Restriction.Restriction.LpnPI.size
Bio.Restriction.Restriction.Lsp1109I.size
Bio.Restriction.Restriction.LweI.size
Bio.Restriction.Restriction.MabI.size
Bio.Restriction.Restriction.MaeI.size
Bio.Restriction.Restriction.MaeII.size
Bio.Restriction.Restriction.MaeIII.size
Bio.Restriction.Restriction.MalI.size
Bio.Restriction.Restriction.MaqI.size
Bio.Restriction.Restriction.MauBI.size
Bio.Restriction.Restriction.MbiI.size
Bio.Restriction.Restriction.MboI.size
Bio.Restriction.Restriction.MboII.size
Bio.Restriction.Restriction.McaTI.size
Bio.Restriction.Restriction.McrI.size
Bio.Restriction.Restriction.MfeI.size
Bio.Restriction.Restriction.MflI.size
Bio.Restriction.Restriction.MhlI.size
Bio.Restriction.Restriction.MjaIV.size
Bio.Restriction.Restriction.MkaDII.size
Bio.Restriction.Restriction.MlsI.size
Bio.Restriction.Restriction.MluCI.size
Bio.Restriction.Restriction.MluI.size
Bio.Restriction.Restriction.MluNI.size
Bio.Restriction.Restriction.Mly113I.size
Bio.Restriction.Restriction.MlyI.size
Bio.Restriction.Restriction.MmeI.size
Bio.Restriction.Restriction.MnlI.size
Bio.Restriction.Restriction.Mph1103I.size
Bio.Restriction.Restriction.MreI.size
Bio.Restriction.Restriction.MroI.size
Bio.Restriction.Restriction.MroNI.size
Bio.Restriction.Restriction.MroXI.size
Bio.Restriction.Restriction.MscI.size
Bio.Restriction.Restriction.MseI.size
Bio.Restriction.Restriction.MslI.size
Bio.Restriction.Restriction.Msp20I.size
Bio.Restriction.Restriction.MspA1I.size
Bio.Restriction.Restriction.MspCI.size
Bio.Restriction.Restriction.MspI.size
Bio.Restriction.Restriction.MspJI.size
Bio.Restriction.Restriction.MspR9I.size
Bio.Restriction.Restriction.MssI.size
Bio.Restriction.Restriction.MstI.size
Bio.Restriction.Restriction.MunI.size
Bio.Restriction.Restriction.Mva1269I.size
Bio.Restriction.Restriction.MvaI.size
Bio.Restriction.Restriction.MvnI.size
Bio.Restriction.Restriction.MvrI.size
Bio.Restriction.Restriction.MwoI.size
Bio.Restriction.Restriction.NaeI.size
Bio.Restriction.Restriction.NarI.size
Bio.Restriction.Restriction.NciI.size
Bio.Restriction.Restriction.NcoI.size
Bio.Restriction.Restriction.NdeI.size
Bio.Restriction.Restriction.NdeII.size
Bio.Restriction.Restriction.NgoAVIII.size
Bio.Restriction.Restriction.NgoMIV.size
Bio.Restriction.Restriction.NhaXI.size
Bio.Restriction.Restriction.NheI.size
Bio.Restriction.Restriction.NlaCI.size
Bio.Restriction.Restriction.NlaIII.size
Bio.Restriction.Restriction.NlaIV.size
Bio.Restriction.Restriction.Nli3877I.size
Bio.Restriction.Restriction.NmeAIII.size
Bio.Restriction.Restriction.NmeDI.size
Bio.Restriction.Restriction.NmuCI.size
Bio.Restriction.Restriction.NotI.size
Bio.Restriction.Restriction.NruI.size
Bio.Restriction.Restriction.NsbI.size
Bio.Restriction.Restriction.NsiI.size
Bio.Restriction.Restriction.NspBII.size
Bio.Restriction.Restriction.NspI.size
Bio.Restriction.Restriction.NspV.size
Bio.Restriction.Restriction.OliI.size
Bio.Restriction.Restriction.PabI.size
Bio.Restriction.Restriction.PacI.size
Bio.Restriction.Restriction.PaeI.size
Bio.Restriction.Restriction.PaeR7I.size
Bio.Restriction.Restriction.PagI.size
Bio.Restriction.Restriction.PalAI.size
Bio.Restriction.Restriction.PasI.size
Bio.Restriction.Restriction.PauI.size
Bio.Restriction.Restriction.PceI.size
Bio.Restriction.Restriction.PciI.size
Bio.Restriction.Restriction.PciSI.size
Bio.Restriction.Restriction.PcsI.size
Bio.Restriction.Restriction.PctI.size
Bio.Restriction.Restriction.PdiI.size
Bio.Restriction.Restriction.PdmI.size
Bio.Restriction.Restriction.PenI.size
Bio.Restriction.Restriction.PfeI.size
Bio.Restriction.Restriction.Pfl1108I.size
Bio.Restriction.Restriction.Pfl23II.size
Bio.Restriction.Restriction.PflFI.size
Bio.Restriction.Restriction.PflMI.size
Bio.Restriction.Restriction.PfoI.size
Bio.Restriction.Restriction.PinAI.size
Bio.Restriction.Restriction.PlaDI.size
Bio.Restriction.Restriction.Ple19I.size
Bio.Restriction.Restriction.PleI.size
Bio.Restriction.Restriction.PluTI.size
Bio.Restriction.Restriction.PmaCI.size
Bio.Restriction.Restriction.PmeI.size
Bio.Restriction.Restriction.PmlI.size
Bio.Restriction.Restriction.PpiI.size
Bio.Restriction.Restriction.PpsI.size
Bio.Restriction.Restriction.Ppu10I.size
Bio.Restriction.Restriction.Ppu21I.size
Bio.Restriction.Restriction.PpuMI.size
Bio.Restriction.Restriction.PscI.size
Bio.Restriction.Restriction.PshAI.size
Bio.Restriction.Restriction.PshBI.size
Bio.Restriction.Restriction.PsiI.size
Bio.Restriction.Restriction.Psp03I.size
Bio.Restriction.Restriction.Psp124BI.size
Bio.Restriction.Restriction.Psp1406I.size
Bio.Restriction.Restriction.Psp5II.size
Bio.Restriction.Restriction.Psp6I.size
Bio.Restriction.Restriction.PspCI.size
Bio.Restriction.Restriction.PspEI.size
Bio.Restriction.Restriction.PspGI.size
Bio.Restriction.Restriction.PspLI.size
Bio.Restriction.Restriction.PspN4I.size
Bio.Restriction.Restriction.PspOMI.size
Bio.Restriction.Restriction.PspOMII.size
Bio.Restriction.Restriction.PspPI.size
Bio.Restriction.Restriction.PspPPI.size
Bio.Restriction.Restriction.PspPRI.size
Bio.Restriction.Restriction.PspXI.size
Bio.Restriction.Restriction.PsrI.size
Bio.Restriction.Restriction.PssI.size
Bio.Restriction.Restriction.PstI.size
Bio.Restriction.Restriction.PstNI.size
Bio.Restriction.Restriction.PsuI.size
Bio.Restriction.Restriction.PsyI.size
Bio.Restriction.Restriction.PteI.size
Bio.Restriction.Restriction.PvuI.size
Bio.Restriction.Restriction.PvuII.size
Bio.Restriction.Restriction.R2_BceSIV.size
Bio.Restriction.Restriction.RceI.size
Bio.Restriction.Restriction.RdeGBI.size
Bio.Restriction.Restriction.RdeGBII.size
Bio.Restriction.Restriction.RdeGBIII.size
Bio.Restriction.Restriction.RflFIII.size
Bio.Restriction.Restriction.RgaI.size
Bio.Restriction.Restriction.RigI.size
Bio.Restriction.Restriction.RlaI.size
Bio.Restriction.Restriction.RleAI.size
Bio.Restriction.Restriction.RpaB5I.size
Bio.Restriction.Restriction.RpaBI.size
Bio.Restriction.Restriction.RpaI.size
Bio.Restriction.Restriction.RpaTI.size
Bio.Restriction.Restriction.RruI.size
Bio.Restriction.Restriction.RsaI.size
Bio.Restriction.Restriction.RsaNI.size
Bio.Restriction.Restriction.RseI.size
Bio.Restriction.Restriction.Rsr2I.size
Bio.Restriction.Restriction.RsrII.size
Bio.Restriction.Restriction.SacI.size
Bio.Restriction.Restriction.SacII.size
Bio.Restriction.Restriction.SalI.size
Bio.Restriction.Restriction.SanDI.size
Bio.Restriction.Restriction.SapI.size
Bio.Restriction.Restriction.SaqAI.size
Bio.Restriction.Restriction.SatI.size
Bio.Restriction.Restriction.Sau3AI.size
Bio.Restriction.Restriction.Sau96I.size
Bio.Restriction.Restriction.SauI.size
Bio.Restriction.Restriction.SbfI.size
Bio.Restriction.Restriction.ScaI.size
Bio.Restriction.Restriction.SchI.size
Bio.Restriction.Restriction.SciI.size
Bio.Restriction.Restriction.ScrFI.size
Bio.Restriction.Restriction.SdaI.size
Bio.Restriction.Restriction.SdeAI.size
Bio.Restriction.Restriction.SdeOSI.size
Bio.Restriction.Restriction.SduI.size
Bio.Restriction.Restriction.SecI.size
Bio.Restriction.Restriction.SelI.size
Bio.Restriction.Restriction.SetI.size
Bio.Restriction.Restriction.SexAI.size
Bio.Restriction.Restriction.SfaAI.size
Bio.Restriction.Restriction.SfaNI.size
Bio.Restriction.Restriction.SfcI.size
Bio.Restriction.Restriction.SfeI.size
Bio.Restriction.Restriction.SfiI.size
Bio.Restriction.Restriction.SfoI.size
Bio.Restriction.Restriction.Sfr274I.size
Bio.Restriction.Restriction.Sfr303I.size
Bio.Restriction.Restriction.SfuI.size
Bio.Restriction.Restriction.SgeI.size
Bio.Restriction.Restriction.SgfI.size
Bio.Restriction.Restriction.SgrAI.size
Bio.Restriction.Restriction.SgrBI.size
Bio.Restriction.Restriction.SgrDI.size
Bio.Restriction.Restriction.SgrTI.size
Bio.Restriction.Restriction.SgsI.size
Bio.Restriction.Restriction.SimI.size
Bio.Restriction.Restriction.SlaI.size
Bio.Restriction.Restriction.SmaI.size
Bio.Restriction.Restriction.SmiI.size
Bio.Restriction.Restriction.SmiMI.size
Bio.Restriction.Restriction.SmlI.size
Bio.Restriction.Restriction.SmoI.size
Bio.Restriction.Restriction.SnaBI.size
Bio.Restriction.Restriction.SnaI.size
Bio.Restriction.Restriction.Sno506I.size
Bio.Restriction.Restriction.SpeI.size
Bio.Restriction.Restriction.SphI.size
Bio.Restriction.Restriction.SplI.size
Bio.Restriction.Restriction.SpoDI.size
Bio.Restriction.Restriction.SrfI.size
Bio.Restriction.Restriction.Sse232I.size
Bio.Restriction.Restriction.Sse8387I.size
Bio.Restriction.Restriction.Sse8647I.size
Bio.Restriction.Restriction.Sse9I.size
Bio.Restriction.Restriction.SseBI.size
Bio.Restriction.Restriction.SsiI.size
Bio.Restriction.Restriction.SspD5I.size
Bio.Restriction.Restriction.SspDI.size
Bio.Restriction.Restriction.SspI.size
Bio.Restriction.Restriction.SstE37I.size
Bio.Restriction.Restriction.SstI.size
Bio.Restriction.Restriction.Sth132I.size
Bio.Restriction.Restriction.Sth302II.size
Bio.Restriction.Restriction.StrI.size
Bio.Restriction.Restriction.StsI.size
Bio.Restriction.Restriction.StuI.size
Bio.Restriction.Restriction.StyD4I.size
Bio.Restriction.Restriction.StyI.size
Bio.Restriction.Restriction.SwaI.size
Bio.Restriction.Restriction.TaaI.size
Bio.Restriction.Restriction.TaiI.size
Bio.Restriction.Restriction.TaqI.size
Bio.Restriction.Restriction.TaqII.size
Bio.Restriction.Restriction.TasI.size
Bio.Restriction.Restriction.TatI.size
Bio.Restriction.Restriction.TauI.size
Bio.Restriction.Restriction.TfiI.size
Bio.Restriction.Restriction.Tru1I.size
Bio.Restriction.Restriction.Tru9I.size
Bio.Restriction.Restriction.TscAI.size
Bio.Restriction.Restriction.TseFI.size
Bio.Restriction.Restriction.TseI.size
Bio.Restriction.Restriction.TsoI.size
Bio.Restriction.Restriction.Tsp45I.size
Bio.Restriction.Restriction.Tsp4CI.size
Bio.Restriction.Restriction.TspDTI.size
Bio.Restriction.Restriction.TspEI.size
Bio.Restriction.Restriction.TspGWI.size
Bio.Restriction.Restriction.TspMI.size
Bio.Restriction.Restriction.TspRI.size
Bio.Restriction.Restriction.TssI.size
Bio.Restriction.Restriction.TstI.size
Bio.Restriction.Restriction.TsuI.size
Bio.Restriction.Restriction.Tth111I.size
Bio.Restriction.Restriction.Tth111II.size
Bio.Restriction.Restriction.UbaF11I.size
Bio.Restriction.Restriction.UbaF12I.size
Bio.Restriction.Restriction.UbaF13I.size
Bio.Restriction.Restriction.UbaF14I.size
Bio.Restriction.Restriction.UbaF9I.size
Bio.Restriction.Restriction.UbaPI.size
Bio.Restriction.Restriction.UcoMSI.size
Bio.Restriction.Restriction.UnbI.size
Bio.Restriction.Restriction.Van91I.size
Bio.Restriction.Restriction.Vha464I.size
Bio.Restriction.Restriction.VneI.size
Bio.Restriction.Restriction.VpaK11AI.size
Bio.Restriction.Restriction.VpaK11BI.size
Bio.Restriction.Restriction.VspI.size
Bio.Restriction.Restriction.WviI.size
Bio.Restriction.Restriction.XagI.size
Bio.Restriction.Restriction.XapI.size
Bio.Restriction.Restriction.XbaI.size
Bio.Restriction.Restriction.XceI.size
Bio.Restriction.Restriction.XcmI.size
Bio.Restriction.Restriction.XhoI.size
Bio.Restriction.Restriction.XhoII.size
Bio.Restriction.Restriction.XmaI.size
Bio.Restriction.Restriction.XmaIII.size
Bio.Restriction.Restriction.XmaJI.size
Bio.Restriction.Restriction.XmiI.size
Bio.Restriction.Restriction.XmnI.size
Bio.Restriction.Restriction.XspI.size
Bio.Restriction.Restriction.YkrI.size
Bio.Restriction.Restriction.ZraI.size
Bio.Restriction.Restriction.ZrmI.size
Bio.Restriction.Restriction.Zsp2I.size
Bio.codonalign.codonalphabet.CodonAlphabet.size" class="py-name" href="#" onclick="return doclink('link-81', 'size', 'link-35');">size</a></tt> </tt>
<a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"PROTEIN"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-82" class="py-name"><a title="Bio.GenBank._FeatureConsumer.residue_type
Bio.GenBank._RecordConsumer.residue_type" class="py-name" href="#" onclick="return doclink('link-82', 'residue_type', 'link-36');">residue_type</a></tt><tt class="py-op">:</tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">" aa"</tt> </tt>
<a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">" bp "</tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line"> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line">        <tt class="py-comment"># treat circular types differently, since they'll have long residue</tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line">        <tt class="py-comment"># types</tt> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"circular"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-83" class="py-name"><a title="Bio.GenBank._FeatureConsumer.residue_type
Bio.GenBank._RecordConsumer.residue_type" class="py-name" href="#" onclick="return doclink('link-83', 'residue_type', 'link-36');">residue_type</a></tt><tt class="py-op">:</tt> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">"%17s"</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-84" class="py-name"><a title="Bio.GenBank._FeatureConsumer.residue_type
Bio.GenBank._RecordConsumer.residue_type" class="py-name" href="#" onclick="return doclink('link-84', 'residue_type', 'link-36');">residue_type</a></tt> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line">        <tt class="py-comment"># second case: ss-DNA types of records</tt> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-string">"-"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-85" class="py-name"><a title="Bio.GenBank._FeatureConsumer.residue_type
Bio.GenBank._RecordConsumer.residue_type" class="py-name" href="#" onclick="return doclink('link-85', 'residue_type', 'link-36');">residue_type</a></tt><tt class="py-op">:</tt> </tt>
<a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">"%7s"</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-86" class="py-name"><a title="Bio.GenBank._FeatureConsumer.residue_type
Bio.GenBank._RecordConsumer.residue_type" class="py-name" href="#" onclick="return doclink('link-86', 'residue_type', 'link-36');">residue_type</a></tt> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt class="py-number">10</tt>  <tt class="py-comment"># spaces for circular</tt> </tt>
<a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt class="py-number">3</tt>  <tt class="py-comment"># spaces for stuff like ss-</tt> </tt>
<a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">"%-4s"</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-87" class="py-name"><a title="Bio.GenBank._FeatureConsumer.residue_type
Bio.GenBank._RecordConsumer.residue_type" class="py-name" href="#" onclick="return doclink('link-87', 'residue_type', 'link-36');">residue_type</a></tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt class="py-number">10</tt>  <tt class="py-comment"># spaces for circular</tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line"> </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt class="py-number">2</tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">"%3s"</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-88" class="py-name"><a title="Bio.GenBank._FeatureConsumer.data_file_division
Bio.GenBank._RecordConsumer.data_file_division" class="py-name" href="#" onclick="return doclink('link-88', 'data_file_division', 'link-37');">data_file_division</a></tt> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt class="py-number">7</tt>  <tt class="py-comment"># spaces for 56-63</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">"%11s"</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-89" class="py-name"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-89', 'date', 'link-38');">date</a></tt> </tt>
<a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">"\n"</tt> </tt>
<a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line"> </tt>
<a name="Record._definition_line"></a><div id="Record._definition_line-def"><a name="L274"></a><tt class="py-lineno">274</tt> <a class="py-toggle" href="#" id="Record._definition_line-toggle" onclick="return toggle('Record._definition_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Record-class.html#_definition_line">_definition_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record._definition_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record._definition_line-expanded"><a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line">        <tt class="py-docstring">"""Provide output for the DEFINITION line.</tt> </tt>
<a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt id="link-90" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-90', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-91" class="py-name"><a title="Bio.GenBank.Record.Record.BASE_FORMAT" class="py-name" href="#" onclick="return doclink('link-91', 'BASE_FORMAT', 'link-16');">BASE_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"DEFINITION"</tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-92" class="py-name" targets="Function Bio.GenBank.Record._wrapped_genbank()=Bio.GenBank.Record-module.html#_wrapped_genbank"><a title="Bio.GenBank.Record._wrapped_genbank" class="py-name" href="#" onclick="return doclink('link-92', '_wrapped_genbank', 'link-92');">_wrapped_genbank</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-93" class="py-name"><a title="Bio.GenBank._FeatureConsumer.definition
Bio.GenBank._RecordConsumer.definition" class="py-name" href="#" onclick="return doclink('link-93', 'definition', 'link-39');">definition</a></tt><tt class="py-op">,</tt> <tt id="link-94" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-94', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-95" class="py-name"><a title="Bio.GenBank.Record.Record.GB_BASE_INDENT" class="py-name" href="#" onclick="return doclink('link-95', 'GB_BASE_INDENT', 'link-10');">GB_BASE_INDENT</a></tt><tt class="py-op">)</tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line"> </tt>
<a name="Record._accession_line"></a><div id="Record._accession_line-def"><a name="L281"></a><tt class="py-lineno">281</tt> <a class="py-toggle" href="#" id="Record._accession_line-toggle" onclick="return toggle('Record._accession_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Record-class.html#_accession_line">_accession_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record._accession_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record._accession_line-expanded"><a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line">        <tt class="py-docstring">"""Output for the ACCESSION line.</tt> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-96" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-96', 'accession', 'link-40');">accession</a></tt><tt class="py-op">:</tt> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt id="link-97" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-97', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-98" class="py-name"><a title="Bio.GenBank.Record.Record.BASE_FORMAT" class="py-name" href="#" onclick="return doclink('link-98', 'BASE_FORMAT', 'link-16');">BASE_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"ACCESSION"</tt> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line"> </tt>
<a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line">            <tt class="py-name">acc_info</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt id="link-99" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-99', 'accession', 'link-40');">accession</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-100" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-100', 'accession', 'link-40');">accession</a></tt><tt class="py-op">:</tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line">                <tt class="py-name">acc_info</tt> <tt class="py-op">+=</tt> <tt class="py-string">"%s "</tt> <tt class="py-op">%</tt> <tt id="link-101" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-101', 'accession', 'link-40');">accession</a></tt> </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line">            <tt class="py-comment"># strip off an extra space at the end</tt> </tt>
<a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line">            <tt class="py-name">acc_info</tt> <tt class="py-op">=</tt> <tt class="py-name">acc_info</tt><tt class="py-op">.</tt><tt id="link-102" class="py-name" targets="Method Bio.Seq.Seq.rstrip()=Bio.Seq.Seq-class.html#rstrip"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-102', 'rstrip', 'link-102');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-103" class="py-name"><a title="Bio.GenBank.Record._wrapped_genbank" class="py-name" href="#" onclick="return doclink('link-103', '_wrapped_genbank', 'link-92');">_wrapped_genbank</a></tt><tt class="py-op">(</tt><tt class="py-name">acc_info</tt><tt class="py-op">,</tt> <tt id="link-104" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-104', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-105" class="py-name"><a title="Bio.GenBank.Record.Record.GB_BASE_INDENT" class="py-name" href="#" onclick="return doclink('link-105', 'GB_BASE_INDENT', 'link-10');">GB_BASE_INDENT</a></tt><tt class="py-op">)</tt> </tt>
<a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line"> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line"> </tt>
<a name="Record._version_line"></a><div id="Record._version_line-def"><a name="L298"></a><tt class="py-lineno">298</tt> <a class="py-toggle" href="#" id="Record._version_line-toggle" onclick="return toggle('Record._version_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Record-class.html#_version_line">_version_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record._version_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record._version_line-expanded"><a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line">        <tt class="py-docstring">"""Output for the VERSION line.</tt> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-106" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-106', 'version', 'link-43');">version</a></tt><tt class="py-op">:</tt> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt id="link-107" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-107', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-108" class="py-name"><a title="Bio.GenBank.Record.Record.BASE_FORMAT" class="py-name" href="#" onclick="return doclink('link-108', 'BASE_FORMAT', 'link-16');">BASE_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"VERSION"</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-109" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-109', 'version', 'link-43');">version</a></tt> </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">"  GI:"</tt> </tt>
<a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">"%s\n"</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-110" class="py-name"><a title="Bio.GenBank._FeatureConsumer.gi
Bio.GenBank._RecordConsumer.gi" class="py-name" href="#" onclick="return doclink('link-110', 'gi', 'link-45');">gi</a></tt> </tt>
<a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line"> </tt>
<a name="Record._project_line"></a><div id="Record._project_line-def"><a name="L310"></a><tt class="py-lineno">310</tt> <a class="py-toggle" href="#" id="Record._project_line-toggle" onclick="return toggle('Record._project_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Record-class.html#_project_line">_project_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record._project_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record._project_line-expanded"><a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">projects</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt id="link-111" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-111', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-112" class="py-name"><a title="Bio.GenBank.Record.Record.BASE_FORMAT" class="py-name" href="#" onclick="return doclink('link-112', 'BASE_FORMAT', 'link-16');">BASE_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"PROJECT"</tt> </tt>
<a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">"%s\n"</tt> <tt class="py-op">%</tt> <tt class="py-string">"  "</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">projects</tt><tt class="py-op">)</tt> </tt>
<a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line"> </tt>
<a name="Record._dblink_line"></a><div id="Record._dblink_line-def"><a name="L317"></a><tt class="py-lineno">317</tt> <a class="py-toggle" href="#" id="Record._dblink_line-toggle" onclick="return toggle('Record._dblink_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Record-class.html#_dblink_line">_dblink_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record._dblink_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record._dblink_line-expanded"><a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dblinks</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt id="link-113" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-113', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-114" class="py-name"><a title="Bio.GenBank.Record.Record.BASE_FORMAT" class="py-name" href="#" onclick="return doclink('link-114', 'BASE_FORMAT', 'link-16');">BASE_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"DBLINK"</tt> </tt>
<a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line">            <tt class="py-name">dblink_info</tt> <tt class="py-op">=</tt> <tt class="py-string">"\n"</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dblinks</tt><tt class="py-op">)</tt> </tt>
<a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-115" class="py-name"><a title="Bio.GenBank.Record._wrapped_genbank" class="py-name" href="#" onclick="return doclink('link-115', '_wrapped_genbank', 'link-92');">_wrapped_genbank</a></tt><tt class="py-op">(</tt><tt class="py-name">dblink_info</tt><tt class="py-op">,</tt> <tt id="link-116" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-116', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-117" class="py-name"><a title="Bio.GenBank.Record.Record.GB_BASE_INDENT" class="py-name" href="#" onclick="return doclink('link-117', 'GB_BASE_INDENT', 'link-10');">GB_BASE_INDENT</a></tt><tt class="py-op">)</tt> </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line"> </tt>
<a name="Record._nid_line"></a><div id="Record._nid_line-def"><a name="L325"></a><tt class="py-lineno">325</tt> <a class="py-toggle" href="#" id="Record._nid_line-toggle" onclick="return toggle('Record._nid_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Record-class.html#_nid_line">_nid_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record._nid_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record._nid_line-expanded"><a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line">        <tt class="py-docstring">"""Output for the NID line. Use of NID is obsolete in GenBank files.</tt> </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-118" class="py-name"><a title="Bio.GenBank._FeatureConsumer.nid
Bio.GenBank._RecordConsumer.nid" class="py-name" href="#" onclick="return doclink('link-118', 'nid', 'link-41');">nid</a></tt><tt class="py-op">:</tt> </tt>
<a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt id="link-119" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-119', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-120" class="py-name"><a title="Bio.GenBank.Record.Record.BASE_FORMAT" class="py-name" href="#" onclick="return doclink('link-120', 'BASE_FORMAT', 'link-16');">BASE_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"NID"</tt> </tt>
<a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">"%s\n"</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-121" class="py-name"><a title="Bio.GenBank._FeatureConsumer.nid
Bio.GenBank._RecordConsumer.nid" class="py-name" href="#" onclick="return doclink('link-121', 'nid', 'link-41');">nid</a></tt> </tt>
<a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L333"></a><tt class="py-lineno">333</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L334"></a><tt class="py-lineno">334</tt>  <tt class="py-line"> </tt>
<a name="Record._pid_line"></a><div id="Record._pid_line-def"><a name="L335"></a><tt class="py-lineno">335</tt> <a class="py-toggle" href="#" id="Record._pid_line-toggle" onclick="return toggle('Record._pid_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Record-class.html#_pid_line">_pid_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record._pid_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record._pid_line-expanded"><a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line">        <tt class="py-docstring">"""Output for PID line. Presumedly, PID usage is also obsolete.</tt> </tt>
<a name="L337"></a><tt class="py-lineno">337</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-122" class="py-name"><a title="Bio.GenBank._FeatureConsumer.pid
Bio.GenBank._RecordConsumer.pid" class="py-name" href="#" onclick="return doclink('link-122', 'pid', 'link-42');">pid</a></tt><tt class="py-op">:</tt> </tt>
<a name="L339"></a><tt class="py-lineno">339</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt id="link-123" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-123', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-124" class="py-name"><a title="Bio.GenBank.Record.Record.BASE_FORMAT" class="py-name" href="#" onclick="return doclink('link-124', 'BASE_FORMAT', 'link-16');">BASE_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"PID"</tt> </tt>
<a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">"%s\n"</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-125" class="py-name"><a title="Bio.GenBank._FeatureConsumer.pid
Bio.GenBank._RecordConsumer.pid" class="py-name" href="#" onclick="return doclink('link-125', 'pid', 'link-42');">pid</a></tt> </tt>
<a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L343"></a><tt class="py-lineno">343</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L344"></a><tt class="py-lineno">344</tt>  <tt class="py-line"> </tt>
<a name="Record._keywords_line"></a><div id="Record._keywords_line-def"><a name="L345"></a><tt class="py-lineno">345</tt> <a class="py-toggle" href="#" id="Record._keywords_line-toggle" onclick="return toggle('Record._keywords_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Record-class.html#_keywords_line">_keywords_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record._keywords_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record._keywords_line-expanded"><a name="L346"></a><tt class="py-lineno">346</tt>  <tt class="py-line">        <tt class="py-docstring">"""Output for the KEYWORDS line.</tt> </tt>
<a name="L347"></a><tt class="py-lineno">347</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L349"></a><tt class="py-lineno">349</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-126" class="py-name"><a title="Bio.GenBank._FeatureConsumer.keywords
Bio.GenBank._RecordConsumer.keywords" class="py-name" href="#" onclick="return doclink('link-126', 'keywords', 'link-46');">keywords</a></tt><tt class="py-op">)</tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-127" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-127', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-128" class="py-name"><a title="Bio.GenBank.Record.Record.BASE_FORMAT" class="py-name" href="#" onclick="return doclink('link-128', 'BASE_FORMAT', 'link-16');">BASE_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"KEYWORDS"</tt> </tt>
<a name="L351"></a><tt class="py-lineno">351</tt>  <tt class="py-line">            <tt class="py-name">keyword_info</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L352"></a><tt class="py-lineno">352</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">keyword</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-129" class="py-name"><a title="Bio.GenBank._FeatureConsumer.keywords
Bio.GenBank._RecordConsumer.keywords" class="py-name" href="#" onclick="return doclink('link-129', 'keywords', 'link-46');">keywords</a></tt><tt class="py-op">:</tt> </tt>
<a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line">                <tt class="py-name">keyword_info</tt> <tt class="py-op">+=</tt> <tt class="py-string">"%s; "</tt> <tt class="py-op">%</tt> <tt class="py-name">keyword</tt> </tt>
<a name="L354"></a><tt class="py-lineno">354</tt>  <tt class="py-line">            <tt class="py-comment"># replace the ; at the end with a period</tt> </tt>
<a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line">            <tt class="py-name">keyword_info</tt> <tt class="py-op">=</tt> <tt class="py-name">keyword_info</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line">            <tt class="py-name">keyword_info</tt> <tt class="py-op">+=</tt> <tt class="py-string">"."</tt> </tt>
<a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line"> </tt>
<a name="L358"></a><tt class="py-lineno">358</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-130" class="py-name"><a title="Bio.GenBank.Record._wrapped_genbank" class="py-name" href="#" onclick="return doclink('link-130', '_wrapped_genbank', 'link-92');">_wrapped_genbank</a></tt><tt class="py-op">(</tt><tt class="py-name">keyword_info</tt><tt class="py-op">,</tt> </tt>
<a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line">                                       <tt id="link-131" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-131', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-132" class="py-name"><a title="Bio.GenBank.Record.Record.GB_BASE_INDENT" class="py-name" href="#" onclick="return doclink('link-132', 'GB_BASE_INDENT', 'link-10');">GB_BASE_INDENT</a></tt><tt class="py-op">)</tt> </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line"> </tt>
<a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line"> </tt>
<a name="Record._db_source_line"></a><div id="Record._db_source_line-def"><a name="L363"></a><tt class="py-lineno">363</tt> <a class="py-toggle" href="#" id="Record._db_source_line-toggle" onclick="return toggle('Record._db_source_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Record-class.html#_db_source_line">_db_source_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record._db_source_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record._db_source_line-expanded"><a name="L364"></a><tt class="py-lineno">364</tt>  <tt class="py-line">        <tt class="py-docstring">"""Output for DBSOURCE line.</tt> </tt>
<a name="L365"></a><tt class="py-lineno">365</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L366"></a><tt class="py-lineno">366</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-133" class="py-name"><a title="Bio.GenBank._FeatureConsumer.db_source
Bio.GenBank._RecordConsumer.db_source" class="py-name" href="#" onclick="return doclink('link-133', 'db_source', 'link-44');">db_source</a></tt><tt class="py-op">:</tt> </tt>
<a name="L367"></a><tt class="py-lineno">367</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt id="link-134" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-134', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-135" class="py-name"><a title="Bio.GenBank.Record.Record.BASE_FORMAT" class="py-name" href="#" onclick="return doclink('link-135', 'BASE_FORMAT', 'link-16');">BASE_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"DBSOURCE"</tt> </tt>
<a name="L368"></a><tt class="py-lineno">368</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">"%s\n"</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-136" class="py-name"><a title="Bio.GenBank._FeatureConsumer.db_source
Bio.GenBank._RecordConsumer.db_source" class="py-name" href="#" onclick="return doclink('link-136', 'db_source', 'link-44');">db_source</a></tt> </tt>
<a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L371"></a><tt class="py-lineno">371</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L372"></a><tt class="py-lineno">372</tt>  <tt class="py-line"> </tt>
<a name="Record._segment_line"></a><div id="Record._segment_line-def"><a name="L373"></a><tt class="py-lineno">373</tt> <a class="py-toggle" href="#" id="Record._segment_line-toggle" onclick="return toggle('Record._segment_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Record-class.html#_segment_line">_segment_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record._segment_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record._segment_line-expanded"><a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line">        <tt class="py-docstring">"""Output for the SEGMENT line.</tt> </tt>
<a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L376"></a><tt class="py-lineno">376</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L377"></a><tt class="py-lineno">377</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-137" class="py-name"><a title="Bio.GenBank._FeatureConsumer.segment
Bio.GenBank._RecordConsumer.segment" class="py-name" href="#" onclick="return doclink('link-137', 'segment', 'link-47');">segment</a></tt><tt class="py-op">:</tt> </tt>
<a name="L378"></a><tt class="py-lineno">378</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-138" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-138', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-139" class="py-name"><a title="Bio.GenBank.Record.Record.BASE_FORMAT" class="py-name" href="#" onclick="return doclink('link-139', 'BASE_FORMAT', 'link-16');">BASE_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"SEGMENT"</tt> </tt>
<a name="L379"></a><tt class="py-lineno">379</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-140" class="py-name"><a title="Bio.GenBank.Record._wrapped_genbank" class="py-name" href="#" onclick="return doclink('link-140', '_wrapped_genbank', 'link-92');">_wrapped_genbank</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-141" class="py-name"><a title="Bio.GenBank._FeatureConsumer.segment
Bio.GenBank._RecordConsumer.segment" class="py-name" href="#" onclick="return doclink('link-141', 'segment', 'link-47');">segment</a></tt><tt class="py-op">,</tt> <tt id="link-142" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-142', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-143" class="py-name"><a title="Bio.GenBank.Record.Record.GB_BASE_INDENT" class="py-name" href="#" onclick="return doclink('link-143', 'GB_BASE_INDENT', 'link-10');">GB_BASE_INDENT</a></tt><tt class="py-op">)</tt> </tt>
<a name="L380"></a><tt class="py-lineno">380</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L381"></a><tt class="py-lineno">381</tt>  <tt class="py-line"> </tt>
<a name="Record._source_line"></a><div id="Record._source_line-def"><a name="L382"></a><tt class="py-lineno">382</tt> <a class="py-toggle" href="#" id="Record._source_line-toggle" onclick="return toggle('Record._source_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Record-class.html#_source_line">_source_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record._source_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record._source_line-expanded"><a name="L383"></a><tt class="py-lineno">383</tt>  <tt class="py-line">        <tt class="py-docstring">"""Output for SOURCE line on where the sample came from.</tt> </tt>
<a name="L384"></a><tt class="py-lineno">384</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L385"></a><tt class="py-lineno">385</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt id="link-144" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-144', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-145" class="py-name"><a title="Bio.GenBank.Record.Record.BASE_FORMAT" class="py-name" href="#" onclick="return doclink('link-145', 'BASE_FORMAT', 'link-16');">BASE_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"SOURCE"</tt> </tt>
<a name="L386"></a><tt class="py-lineno">386</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-146" class="py-name"><a title="Bio.GenBank.Record._wrapped_genbank" class="py-name" href="#" onclick="return doclink('link-146', '_wrapped_genbank', 'link-92');">_wrapped_genbank</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-147" class="py-name"><a title="Bio.GenBank._FeatureConsumer.source
Bio.GenBank._RecordConsumer.source
Bio.Pathway.Network.source" class="py-name" href="#" onclick="return doclink('link-147', 'source', 'link-48');">source</a></tt><tt class="py-op">,</tt> <tt id="link-148" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-148', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-149" class="py-name"><a title="Bio.GenBank.Record.Record.GB_BASE_INDENT" class="py-name" href="#" onclick="return doclink('link-149', 'GB_BASE_INDENT', 'link-10');">GB_BASE_INDENT</a></tt><tt class="py-op">)</tt> </tt>
<a name="L387"></a><tt class="py-lineno">387</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L388"></a><tt class="py-lineno">388</tt>  <tt class="py-line"> </tt>
<a name="Record._organism_line"></a><div id="Record._organism_line-def"><a name="L389"></a><tt class="py-lineno">389</tt> <a class="py-toggle" href="#" id="Record._organism_line-toggle" onclick="return toggle('Record._organism_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Record-class.html#_organism_line">_organism_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record._organism_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record._organism_line-expanded"><a name="L390"></a><tt class="py-lineno">390</tt>  <tt class="py-line">        <tt class="py-docstring">"""Output for ORGANISM line with taxonomy info.</tt> </tt>
<a name="L391"></a><tt class="py-lineno">391</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L392"></a><tt class="py-lineno">392</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt id="link-150" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-150', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-151" class="py-name"><a title="Bio.GenBank.Record.Record.INTERNAL_FORMAT" class="py-name" href="#" onclick="return doclink('link-151', 'INTERNAL_FORMAT', 'link-18');">INTERNAL_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"ORGANISM"</tt> </tt>
<a name="L393"></a><tt class="py-lineno">393</tt>  <tt class="py-line">        <tt class="py-comment"># Now that species names can be too long, this line can wrap (Bug 2591)</tt> </tt>
<a name="L394"></a><tt class="py-lineno">394</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-152" class="py-name"><a title="Bio.GenBank.Record._wrapped_genbank" class="py-name" href="#" onclick="return doclink('link-152', '_wrapped_genbank', 'link-92');">_wrapped_genbank</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-153" class="py-name"><a title="Bio.GenBank._FeatureConsumer.organism
Bio.GenBank._RecordConsumer.organism" class="py-name" href="#" onclick="return doclink('link-153', 'organism', 'link-49');">organism</a></tt><tt class="py-op">,</tt> <tt id="link-154" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-154', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-155" class="py-name"><a title="Bio.GenBank.Record.Record.GB_BASE_INDENT" class="py-name" href="#" onclick="return doclink('link-155', 'GB_BASE_INDENT', 'link-10');">GB_BASE_INDENT</a></tt><tt class="py-op">)</tt> </tt>
<a name="L395"></a><tt class="py-lineno">395</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt id="link-156" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-156', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-157" class="py-name"><a title="Bio.GenBank.Record.Record.GB_BASE_INDENT" class="py-name" href="#" onclick="return doclink('link-157', 'GB_BASE_INDENT', 'link-10');">GB_BASE_INDENT</a></tt> </tt>
<a name="L396"></a><tt class="py-lineno">396</tt>  <tt class="py-line">        <tt class="py-name">taxonomy_info</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L397"></a><tt class="py-lineno">397</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">tax</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-158" class="py-name"><a title="Bio.GenBank._FeatureConsumer.taxonomy
Bio.GenBank._RecordConsumer.taxonomy
Bio.Phylo.PhyloXML.Clade.taxonomy
Bio.Phylo.PhyloXMLIO.Writer.taxonomy" class="py-name" href="#" onclick="return doclink('link-158', 'taxonomy', 'link-50');">taxonomy</a></tt><tt class="py-op">:</tt> </tt>
<a name="L398"></a><tt class="py-lineno">398</tt>  <tt class="py-line">            <tt class="py-name">taxonomy_info</tt> <tt class="py-op">+=</tt> <tt class="py-string">"%s; "</tt> <tt class="py-op">%</tt> <tt class="py-name">tax</tt> </tt>
<a name="L399"></a><tt class="py-lineno">399</tt>  <tt class="py-line">        <tt class="py-comment"># replace the ; at the end with a period</tt> </tt>
<a name="L400"></a><tt class="py-lineno">400</tt>  <tt class="py-line">        <tt class="py-name">taxonomy_info</tt> <tt class="py-op">=</tt> <tt class="py-name">taxonomy_info</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> </tt>
<a name="L401"></a><tt class="py-lineno">401</tt>  <tt class="py-line">        <tt class="py-name">taxonomy_info</tt> <tt class="py-op">+=</tt> <tt class="py-string">"."</tt> </tt>
<a name="L402"></a><tt class="py-lineno">402</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-159" class="py-name"><a title="Bio.GenBank.Record._wrapped_genbank" class="py-name" href="#" onclick="return doclink('link-159', '_wrapped_genbank', 'link-92');">_wrapped_genbank</a></tt><tt class="py-op">(</tt><tt class="py-name">taxonomy_info</tt><tt class="py-op">,</tt> <tt id="link-160" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-160', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-161" class="py-name"><a title="Bio.GenBank.Record.Record.GB_BASE_INDENT" class="py-name" href="#" onclick="return doclink('link-161', 'GB_BASE_INDENT', 'link-10');">GB_BASE_INDENT</a></tt><tt class="py-op">)</tt> </tt>
<a name="L403"></a><tt class="py-lineno">403</tt>  <tt class="py-line"> </tt>
<a name="L404"></a><tt class="py-lineno">404</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L405"></a><tt class="py-lineno">405</tt>  <tt class="py-line"> </tt>
<a name="Record._comment_line"></a><div id="Record._comment_line-def"><a name="L406"></a><tt class="py-lineno">406</tt> <a class="py-toggle" href="#" id="Record._comment_line-toggle" onclick="return toggle('Record._comment_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Record-class.html#_comment_line">_comment_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record._comment_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record._comment_line-expanded"><a name="L407"></a><tt class="py-lineno">407</tt>  <tt class="py-line">        <tt class="py-docstring">"""Output for the COMMENT lines.</tt> </tt>
<a name="L408"></a><tt class="py-lineno">408</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L409"></a><tt class="py-lineno">409</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L410"></a><tt class="py-lineno">410</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-162" class="py-name"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-162', 'comment', 'link-51');">comment</a></tt><tt class="py-op">:</tt> </tt>
<a name="L411"></a><tt class="py-lineno">411</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-163" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-163', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-164" class="py-name"><a title="Bio.GenBank.Record.Record.BASE_FORMAT" class="py-name" href="#" onclick="return doclink('link-164', 'BASE_FORMAT', 'link-16');">BASE_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"COMMENT"</tt> </tt>
<a name="L412"></a><tt class="py-lineno">412</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-165" class="py-name" targets="Function Bio.GenBank.Record._indent_genbank()=Bio.GenBank.Record-module.html#_indent_genbank"><a title="Bio.GenBank.Record._indent_genbank" class="py-name" href="#" onclick="return doclink('link-165', '_indent_genbank', 'link-165');">_indent_genbank</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-166" class="py-name"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-166', 'comment', 'link-51');">comment</a></tt><tt class="py-op">,</tt> </tt>
<a name="L413"></a><tt class="py-lineno">413</tt>  <tt class="py-line">                                      <tt id="link-167" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-167', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-168" class="py-name"><a title="Bio.GenBank.Record.Record.GB_BASE_INDENT" class="py-name" href="#" onclick="return doclink('link-168', 'GB_BASE_INDENT', 'link-10');">GB_BASE_INDENT</a></tt><tt class="py-op">)</tt> </tt>
<a name="L414"></a><tt class="py-lineno">414</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L415"></a><tt class="py-lineno">415</tt>  <tt class="py-line"> </tt>
<a name="Record._features_line"></a><div id="Record._features_line-def"><a name="L416"></a><tt class="py-lineno">416</tt> <a class="py-toggle" href="#" id="Record._features_line-toggle" onclick="return toggle('Record._features_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Record-class.html#_features_line">_features_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record._features_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record._features_line-expanded"><a name="L417"></a><tt class="py-lineno">417</tt>  <tt class="py-line">        <tt class="py-docstring">"""Output for the FEATURES line.</tt> </tt>
<a name="L418"></a><tt class="py-lineno">418</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L419"></a><tt class="py-lineno">419</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L420"></a><tt class="py-lineno">420</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-169" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-169', 'features', 'link-52');">features</a></tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L421"></a><tt class="py-lineno">421</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-170" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-170', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-171" class="py-name"><a title="Bio.GenBank.Record.Record.BASE_FEATURE_FORMAT" class="py-name" href="#" onclick="return doclink('link-171', 'BASE_FEATURE_FORMAT', 'link-26');">BASE_FEATURE_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"FEATURES"</tt> </tt>
<a name="L422"></a><tt class="py-lineno">422</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">"Location/Qualifiers\n"</tt> </tt>
<a name="L423"></a><tt class="py-lineno">423</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L424"></a><tt class="py-lineno">424</tt>  <tt class="py-line"> </tt>
<a name="Record._base_count_line"></a><div id="Record._base_count_line-def"><a name="L425"></a><tt class="py-lineno">425</tt> <a class="py-toggle" href="#" id="Record._base_count_line-toggle" onclick="return toggle('Record._base_count_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Record-class.html#_base_count_line">_base_count_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record._base_count_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record._base_count_line-expanded"><a name="L426"></a><tt class="py-lineno">426</tt>  <tt class="py-line">        <tt class="py-docstring">"""Output for the BASE COUNT line with base information.</tt> </tt>
<a name="L427"></a><tt class="py-lineno">427</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L428"></a><tt class="py-lineno">428</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L429"></a><tt class="py-lineno">429</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">base_counts</tt><tt class="py-op">:</tt> </tt>
<a name="L430"></a><tt class="py-lineno">430</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-172" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-172', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-173" class="py-name"><a title="Bio.GenBank.Record.Record.BASE_FORMAT" class="py-name" href="#" onclick="return doclink('link-173', 'BASE_FORMAT', 'link-16');">BASE_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"BASE COUNT  "</tt> </tt>
<a name="L431"></a><tt class="py-lineno">431</tt>  <tt class="py-line">            <tt class="py-comment"># split up the base counts into their individual parts</tt> </tt>
<a name="L432"></a><tt class="py-lineno">432</tt>  <tt class="py-line">            <tt class="py-name">count_parts</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">base_counts</tt><tt class="py-op">.</tt><tt id="link-174" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-174', 'split', 'link-4');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">" "</tt><tt class="py-op">)</tt> </tt>
<a name="L433"></a><tt class="py-lineno">433</tt>  <tt class="py-line">            <tt class="py-keyword">while</tt> <tt class="py-string">''</tt> <tt class="py-keyword">in</tt> <tt class="py-name">count_parts</tt><tt class="py-op">:</tt> </tt>
<a name="L434"></a><tt class="py-lineno">434</tt>  <tt class="py-line">                <tt class="py-name">count_parts</tt><tt class="py-op">.</tt><tt id="link-175" class="py-name" targets="Method Bio.Crystal.Chain.remove()=Bio.Crystal.Chain-class.html#remove,Method Bio.Graphics.BasicChromosome._ChromosomeComponent.remove()=Bio.Graphics.BasicChromosome._ChromosomeComponent-class.html#remove,Method Bio.Restriction.Restriction.RestrictionBatch.remove()=Bio.Restriction.Restriction.RestrictionBatch-class.html#remove,Method Bio.Seq.MutableSeq.remove()=Bio.Seq.MutableSeq-class.html#remove,Method BioSQL.Loader.DatabaseRemover.remove()=BioSQL.Loader.DatabaseRemover-class.html#remove"><a title="Bio.Crystal.Chain.remove
Bio.Graphics.BasicChromosome._ChromosomeComponent.remove
Bio.Restriction.Restriction.RestrictionBatch.remove
Bio.Seq.MutableSeq.remove
BioSQL.Loader.DatabaseRemover.remove" class="py-name" href="#" onclick="return doclink('link-175', 'remove', 'link-175');">remove</a></tt><tt class="py-op">(</tt><tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L435"></a><tt class="py-lineno">435</tt>  <tt class="py-line">            <tt class="py-comment"># deal with the standard case, with a normal origin line</tt> </tt>
<a name="L436"></a><tt class="py-lineno">436</tt>  <tt class="py-line">            <tt class="py-comment"># like: 474 a    356 c    428 g    364 t</tt> </tt>
<a name="L437"></a><tt class="py-lineno">437</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">count_parts</tt><tt class="py-op">)</tt> <tt class="py-op">%</tt> <tt class="py-number">2</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L438"></a><tt class="py-lineno">438</tt>  <tt class="py-line">                <tt class="py-keyword">while</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">count_parts</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L439"></a><tt class="py-lineno">439</tt>  <tt class="py-line">                    <tt class="py-name">count_info</tt> <tt class="py-op">=</tt> <tt class="py-name">count_parts</tt><tt class="py-op">.</tt><tt id="link-176" class="py-name" targets="Method Bio.File._IndexedSeqFileDict.pop()=Bio.File._IndexedSeqFileDict-class.html#pop,Method Bio.SearchIO._model.hit.Hit.pop()=Bio.SearchIO._model.hit.Hit-class.html#pop,Method Bio.SearchIO._model.query.QueryResult.pop()=Bio.SearchIO._model.query.QueryResult-class.html#pop,Method Bio.Seq.MutableSeq.pop()=Bio.Seq.MutableSeq-class.html#pop,Method Bio._py3k._ordereddict.OrderedDict.pop()=Bio._py3k._ordereddict.OrderedDict-class.html#pop"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop
Bio._py3k._ordereddict.OrderedDict.pop" class="py-name" href="#" onclick="return doclink('link-176', 'pop', 'link-176');">pop</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L440"></a><tt class="py-lineno">440</tt>  <tt class="py-line">                    <tt class="py-name">count_type</tt> <tt class="py-op">=</tt> <tt class="py-name">count_parts</tt><tt class="py-op">.</tt><tt id="link-177" class="py-name"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop
Bio._py3k._ordereddict.OrderedDict.pop" class="py-name" href="#" onclick="return doclink('link-177', 'pop', 'link-176');">pop</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L441"></a><tt class="py-lineno">441</tt>  <tt class="py-line"> </tt>
<a name="L442"></a><tt class="py-lineno">442</tt>  <tt class="py-line">                    <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">"%7s %s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">count_info</tt><tt class="py-op">,</tt> <tt class="py-name">count_type</tt><tt class="py-op">)</tt> </tt>
<a name="L443"></a><tt class="py-lineno">443</tt>  <tt class="py-line">            <tt class="py-comment"># deal with ugly ORIGIN lines like:</tt> </tt>
<a name="L444"></a><tt class="py-lineno">444</tt>  <tt class="py-line">            <tt class="py-comment"># 1311257 a2224835 c2190093 g1309889 t</tt> </tt>
<a name="L445"></a><tt class="py-lineno">445</tt>  <tt class="py-line">            <tt class="py-comment"># by just outputting the raw information</tt> </tt>
<a name="L446"></a><tt class="py-lineno">446</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L447"></a><tt class="py-lineno">447</tt>  <tt class="py-line">                <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">base_counts</tt> </tt>
<a name="L448"></a><tt class="py-lineno">448</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">"\n"</tt> </tt>
<a name="L449"></a><tt class="py-lineno">449</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L450"></a><tt class="py-lineno">450</tt>  <tt class="py-line"> </tt>
<a name="Record._origin_line"></a><div id="Record._origin_line-def"><a name="L451"></a><tt class="py-lineno">451</tt> <a class="py-toggle" href="#" id="Record._origin_line-toggle" onclick="return toggle('Record._origin_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Record-class.html#_origin_line">_origin_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record._origin_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record._origin_line-expanded"><a name="L452"></a><tt class="py-lineno">452</tt>  <tt class="py-line">        <tt class="py-docstring">"""Output for the ORIGIN line</tt> </tt>
<a name="L453"></a><tt class="py-lineno">453</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L454"></a><tt class="py-lineno">454</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L455"></a><tt class="py-lineno">455</tt>  <tt class="py-line">        <tt class="py-comment"># only output the ORIGIN line if we have a sequence</tt> </tt>
<a name="L456"></a><tt class="py-lineno">456</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-178" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-178', 'sequence', 'link-53');">sequence</a></tt><tt class="py-op">:</tt> </tt>
<a name="L457"></a><tt class="py-lineno">457</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-179" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-179', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-180" class="py-name"><a title="Bio.GenBank.Record.Record.BASE_FORMAT" class="py-name" href="#" onclick="return doclink('link-180', 'BASE_FORMAT', 'link-16');">BASE_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"ORIGIN"</tt> </tt>
<a name="L458"></a><tt class="py-lineno">458</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">origin</tt><tt class="py-op">:</tt> </tt>
<a name="L459"></a><tt class="py-lineno">459</tt>  <tt class="py-line">                <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-181" class="py-name"><a title="Bio.GenBank.Record._wrapped_genbank" class="py-name" href="#" onclick="return doclink('link-181', '_wrapped_genbank', 'link-92');">_wrapped_genbank</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">origin</tt><tt class="py-op">,</tt> </tt>
<a name="L460"></a><tt class="py-lineno">460</tt>  <tt class="py-line">                                           <tt id="link-182" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-182', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-183" class="py-name"><a title="Bio.GenBank.Record.Record.GB_BASE_INDENT" class="py-name" href="#" onclick="return doclink('link-183', 'GB_BASE_INDENT', 'link-10');">GB_BASE_INDENT</a></tt><tt class="py-op">)</tt> </tt>
<a name="L461"></a><tt class="py-lineno">461</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L462"></a><tt class="py-lineno">462</tt>  <tt class="py-line">                <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">"\n"</tt> </tt>
<a name="L463"></a><tt class="py-lineno">463</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L464"></a><tt class="py-lineno">464</tt>  <tt class="py-line"> </tt>
<a name="Record._sequence_line"></a><div id="Record._sequence_line-def"><a name="L465"></a><tt class="py-lineno">465</tt> <a class="py-toggle" href="#" id="Record._sequence_line-toggle" onclick="return toggle('Record._sequence_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Record-class.html#_sequence_line">_sequence_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record._sequence_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record._sequence_line-expanded"><a name="L466"></a><tt class="py-lineno">466</tt>  <tt class="py-line">        <tt class="py-docstring">"""Output for all of the sequence.</tt> </tt>
<a name="L467"></a><tt class="py-lineno">467</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L468"></a><tt class="py-lineno">468</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L469"></a><tt class="py-lineno">469</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-184" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-184', 'sequence', 'link-53');">sequence</a></tt><tt class="py-op">:</tt> </tt>
<a name="L470"></a><tt class="py-lineno">470</tt>  <tt class="py-line">            <tt class="py-name">cur_seq_pos</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L471"></a><tt class="py-lineno">471</tt>  <tt class="py-line">            <tt class="py-keyword">while</tt> <tt class="py-name">cur_seq_pos</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-185" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-185', 'sequence', 'link-53');">sequence</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L472"></a><tt class="py-lineno">472</tt>  <tt class="py-line">                <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-186" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-186', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-187" class="py-name"><a title="Bio.GenBank.Record.Record.SEQUENCE_FORMAT" class="py-name" href="#" onclick="return doclink('link-187', 'SEQUENCE_FORMAT', 'link-32');">SEQUENCE_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">cur_seq_pos</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L473"></a><tt class="py-lineno">473</tt>  <tt class="py-line"> </tt>
<a name="L474"></a><tt class="py-lineno">474</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">section</tt> <tt class="py-keyword">in</tt> <tt id="link-188" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.range()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#range,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.range()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#range,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Track.Track.range()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#range"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-188', 'range', 'link-188');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">6</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L475"></a><tt class="py-lineno">475</tt>  <tt class="py-line">                    <tt class="py-name">start_pos</tt> <tt class="py-op">=</tt> <tt class="py-name">cur_seq_pos</tt> <tt class="py-op">+</tt> <tt class="py-name">section</tt> <tt class="py-op">*</tt> <tt class="py-number">10</tt> </tt>
<a name="L476"></a><tt class="py-lineno">476</tt>  <tt class="py-line">                    <tt class="py-name">end_pos</tt> <tt class="py-op">=</tt> <tt class="py-name">start_pos</tt> <tt class="py-op">+</tt> <tt class="py-number">10</tt> </tt>
<a name="L477"></a><tt class="py-lineno">477</tt>  <tt class="py-line">                    <tt class="py-name">seq_section</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-189" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-189', 'sequence', 'link-53');">sequence</a></tt><tt class="py-op">[</tt><tt class="py-name">start_pos</tt><tt class="py-op">:</tt><tt class="py-name">end_pos</tt><tt class="py-op">]</tt> </tt>
<a name="L478"></a><tt class="py-lineno">478</tt>  <tt class="py-line">                    <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">" %s"</tt> <tt class="py-op">%</tt> <tt class="py-name">seq_section</tt><tt class="py-op">.</tt><tt id="link-190" class="py-name" targets="Method Bio.Seq.Seq.lower()=Bio.Seq.Seq-class.html#lower,Method Bio.Seq.UnknownSeq.lower()=Bio.Seq.UnknownSeq-class.html#lower,Method Bio.SeqRecord.SeqRecord.lower()=Bio.SeqRecord.SeqRecord-class.html#lower"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-190', 'lower', 'link-190');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L479"></a><tt class="py-lineno">479</tt>  <tt class="py-line"> </tt>
<a name="L480"></a><tt class="py-lineno">480</tt>  <tt class="py-line">                    <tt class="py-comment"># stop looping if we are out of sequence</tt> </tt>
<a name="L481"></a><tt class="py-lineno">481</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">end_pos</tt> <tt class="py-op">&gt;</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-191" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.sequence
Bio.GenBank._FeatureConsumer.sequence
Bio.GenBank._RecordConsumer.sequence
Bio.Phylo.PhyloXMLIO.Writer.sequence" class="py-name" href="#" onclick="return doclink('link-191', 'sequence', 'link-53');">sequence</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L482"></a><tt class="py-lineno">482</tt>  <tt class="py-line">                        <tt class="py-keyword">break</tt> </tt>
<a name="L483"></a><tt class="py-lineno">483</tt>  <tt class="py-line"> </tt>
<a name="L484"></a><tt class="py-lineno">484</tt>  <tt class="py-line">                <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">"\n"</tt> </tt>
<a name="L485"></a><tt class="py-lineno">485</tt>  <tt class="py-line">                <tt class="py-name">cur_seq_pos</tt> <tt class="py-op">+=</tt> <tt class="py-number">60</tt> </tt>
<a name="L486"></a><tt class="py-lineno">486</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L487"></a><tt class="py-lineno">487</tt>  <tt class="py-line"> </tt>
<a name="Record._wgs_line"></a><div id="Record._wgs_line-def"><a name="L488"></a><tt class="py-lineno">488</tt> <a class="py-toggle" href="#" id="Record._wgs_line-toggle" onclick="return toggle('Record._wgs_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Record-class.html#_wgs_line">_wgs_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record._wgs_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record._wgs_line-expanded"><a name="L489"></a><tt class="py-lineno">489</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L490"></a><tt class="py-lineno">490</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-192" class="py-name"><a title="Bio.GenBank._FeatureConsumer.wgs
Bio.GenBank._RecordConsumer.wgs" class="py-name" href="#" onclick="return doclink('link-192', 'wgs', 'link-55');">wgs</a></tt><tt class="py-op">:</tt> </tt>
<a name="L491"></a><tt class="py-lineno">491</tt>  <tt class="py-line">                    <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-193" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-193', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-194" class="py-name"><a title="Bio.GenBank.Record.Record.BASE_FORMAT" class="py-name" href="#" onclick="return doclink('link-194', 'BASE_FORMAT', 'link-16');">BASE_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"WGS"</tt> </tt>
<a name="L492"></a><tt class="py-lineno">492</tt>  <tt class="py-line">                    <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-195" class="py-name"><a title="Bio.GenBank._FeatureConsumer.wgs
Bio.GenBank._RecordConsumer.wgs" class="py-name" href="#" onclick="return doclink('link-195', 'wgs', 'link-55');">wgs</a></tt> </tt>
<a name="L493"></a><tt class="py-lineno">493</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L494"></a><tt class="py-lineno">494</tt>  <tt class="py-line"> </tt>
<a name="Record._wgs_scafld_line"></a><div id="Record._wgs_scafld_line-def"><a name="L495"></a><tt class="py-lineno">495</tt> <a class="py-toggle" href="#" id="Record._wgs_scafld_line-toggle" onclick="return toggle('Record._wgs_scafld_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Record-class.html#_wgs_scafld_line">_wgs_scafld_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record._wgs_scafld_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record._wgs_scafld_line-expanded"><a name="L496"></a><tt class="py-lineno">496</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L497"></a><tt class="py-lineno">497</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">wgs_scafld</tt><tt class="py-op">:</tt> </tt>
<a name="L498"></a><tt class="py-lineno">498</tt>  <tt class="py-line">                    <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-196" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-196', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-197" class="py-name"><a title="Bio.GenBank.Record.Record.BASE_FORMAT" class="py-name" href="#" onclick="return doclink('link-197', 'BASE_FORMAT', 'link-16');">BASE_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"WGS_SCAFLD"</tt> </tt>
<a name="L499"></a><tt class="py-lineno">499</tt>  <tt class="py-line">                    <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">wgs_scafld</tt> </tt>
<a name="L500"></a><tt class="py-lineno">500</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L501"></a><tt class="py-lineno">501</tt>  <tt class="py-line"> </tt>
<a name="Record._contig_line"></a><div id="Record._contig_line-def"><a name="L502"></a><tt class="py-lineno">502</tt> <a class="py-toggle" href="#" id="Record._contig_line-toggle" onclick="return toggle('Record._contig_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Record-class.html#_contig_line">_contig_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Record._contig_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Record._contig_line-expanded"><a name="L503"></a><tt class="py-lineno">503</tt>  <tt class="py-line">        <tt class="py-docstring">"""Output for CONTIG location information from RefSeq.</tt> </tt>
<a name="L504"></a><tt class="py-lineno">504</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L505"></a><tt class="py-lineno">505</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L506"></a><tt class="py-lineno">506</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">contig</tt><tt class="py-op">:</tt> </tt>
<a name="L507"></a><tt class="py-lineno">507</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-198" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-198', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-199" class="py-name"><a title="Bio.GenBank.Record.Record.BASE_FORMAT" class="py-name" href="#" onclick="return doclink('link-199', 'BASE_FORMAT', 'link-16');">BASE_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"CONTIG"</tt> </tt>
<a name="L508"></a><tt class="py-lineno">508</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-200" class="py-name"><a title="Bio.GenBank.Record._wrapped_genbank" class="py-name" href="#" onclick="return doclink('link-200', '_wrapped_genbank', 'link-92');">_wrapped_genbank</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">contig</tt><tt class="py-op">,</tt> </tt>
<a name="L509"></a><tt class="py-lineno">509</tt>  <tt class="py-line">                                       <tt id="link-201" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-201', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-202" class="py-name"><a title="Bio.GenBank.Record.Record.GB_BASE_INDENT" class="py-name" href="#" onclick="return doclink('link-202', 'GB_BASE_INDENT', 'link-10');">GB_BASE_INDENT</a></tt><tt class="py-op">,</tt> <tt class="py-name">split_char</tt><tt class="py-op">=</tt><tt class="py-string">','</tt><tt class="py-op">)</tt> </tt>
<a name="L510"></a><tt class="py-lineno">510</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div></div><a name="L511"></a><tt class="py-lineno">511</tt>  <tt class="py-line"> </tt>
<a name="L512"></a><tt class="py-lineno">512</tt>  <tt class="py-line"> </tt>
<a name="Reference"></a><div id="Reference-def"><a name="L513"></a><tt class="py-lineno">513</tt> <a class="py-toggle" href="#" id="Reference-toggle" onclick="return toggle('Reference');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.GenBank.Record.Reference-class.html">Reference</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reference-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Reference-expanded"><a name="L514"></a><tt class="py-lineno">514</tt>  <tt class="py-line">    <tt class="py-docstring">"""Hold information from a GenBank reference.</tt> </tt>
<a name="L515"></a><tt class="py-lineno">515</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L516"></a><tt class="py-lineno">516</tt>  <tt class="py-line"><tt class="py-docstring">    Attributes:</tt> </tt>
<a name="L517"></a><tt class="py-lineno">517</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L518"></a><tt class="py-lineno">518</tt>  <tt class="py-line"><tt class="py-docstring">        - number - The number of the reference in the listing of references.</tt> </tt>
<a name="L519"></a><tt class="py-lineno">519</tt>  <tt class="py-line"><tt class="py-docstring">        - bases - The bases in the sequence the reference refers to.</tt> </tt>
<a name="L520"></a><tt class="py-lineno">520</tt>  <tt class="py-line"><tt class="py-docstring">        - authors - String with all of the authors.</tt> </tt>
<a name="L521"></a><tt class="py-lineno">521</tt>  <tt class="py-line"><tt class="py-docstring">        - consrtm - Consortium the authors belong to.</tt> </tt>
<a name="L522"></a><tt class="py-lineno">522</tt>  <tt class="py-line"><tt class="py-docstring">        - title - The title of the reference.</tt> </tt>
<a name="L523"></a><tt class="py-lineno">523</tt>  <tt class="py-line"><tt class="py-docstring">        - journal - Information about the journal where the reference appeared.</tt> </tt>
<a name="L524"></a><tt class="py-lineno">524</tt>  <tt class="py-line"><tt class="py-docstring">        - medline_id - The medline id for the reference.</tt> </tt>
<a name="L525"></a><tt class="py-lineno">525</tt>  <tt class="py-line"><tt class="py-docstring">        - pubmed_id - The pubmed_id for the reference.</tt> </tt>
<a name="L526"></a><tt class="py-lineno">526</tt>  <tt class="py-line"><tt class="py-docstring">        - remark - Free-form remarks about the reference.</tt> </tt>
<a name="L527"></a><tt class="py-lineno">527</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="Reference.__init__"></a><div id="Reference.__init__-def"><a name="L528"></a><tt class="py-lineno">528</tt> <a class="py-toggle" href="#" id="Reference.__init__-toggle" onclick="return toggle('Reference.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Reference-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reference.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Reference.__init__-expanded"><a name="L529"></a><tt class="py-lineno">529</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-203" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Pathway.number=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#number"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway.number" class="py-name" href="#" onclick="return doclink('link-203', 'number', 'link-203');">number</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L530"></a><tt class="py-lineno">530</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">bases</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L531"></a><tt class="py-lineno">531</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-204" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.authors()=Bio.GenBank._FeatureConsumer-class.html#authors,Method Bio.GenBank._RecordConsumer.authors()=Bio.GenBank._RecordConsumer-class.html#authors"><a title="Bio.GenBank._FeatureConsumer.authors
Bio.GenBank._RecordConsumer.authors" class="py-name" href="#" onclick="return doclink('link-204', 'authors', 'link-204');">authors</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L532"></a><tt class="py-lineno">532</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-205" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.consrtm()=Bio.GenBank._FeatureConsumer-class.html#consrtm,Method Bio.GenBank._RecordConsumer.consrtm()=Bio.GenBank._RecordConsumer-class.html#consrtm"><a title="Bio.GenBank._FeatureConsumer.consrtm
Bio.GenBank._RecordConsumer.consrtm" class="py-name" href="#" onclick="return doclink('link-205', 'consrtm', 'link-205');">consrtm</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L533"></a><tt class="py-lineno">533</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-206" class="py-name" targets="Method Bio.Blast.NCBIStandalone._AlignmentConsumer.title()=Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#title,Method Bio.GenBank._FeatureConsumer.title()=Bio.GenBank._FeatureConsumer-class.html#title,Method Bio.GenBank._RecordConsumer.title()=Bio.GenBank._RecordConsumer-class.html#title"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-206', 'title', 'link-206');">title</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L534"></a><tt class="py-lineno">534</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-207" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.journal()=Bio.GenBank._FeatureConsumer-class.html#journal,Method Bio.GenBank._RecordConsumer.journal()=Bio.GenBank._RecordConsumer-class.html#journal"><a title="Bio.GenBank._FeatureConsumer.journal
Bio.GenBank._RecordConsumer.journal" class="py-name" href="#" onclick="return doclink('link-207', 'journal', 'link-207');">journal</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L535"></a><tt class="py-lineno">535</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-208" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.medline_id()=Bio.GenBank._FeatureConsumer-class.html#medline_id,Method Bio.GenBank._RecordConsumer.medline_id()=Bio.GenBank._RecordConsumer-class.html#medline_id"><a title="Bio.GenBank._FeatureConsumer.medline_id
Bio.GenBank._RecordConsumer.medline_id" class="py-name" href="#" onclick="return doclink('link-208', 'medline_id', 'link-208');">medline_id</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L536"></a><tt class="py-lineno">536</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-209" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.pubmed_id()=Bio.GenBank._FeatureConsumer-class.html#pubmed_id,Method Bio.GenBank._RecordConsumer.pubmed_id()=Bio.GenBank._RecordConsumer-class.html#pubmed_id"><a title="Bio.GenBank._FeatureConsumer.pubmed_id
Bio.GenBank._RecordConsumer.pubmed_id" class="py-name" href="#" onclick="return doclink('link-209', 'pubmed_id', 'link-209');">pubmed_id</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L537"></a><tt class="py-lineno">537</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-210" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.remark()=Bio.GenBank._FeatureConsumer-class.html#remark,Method Bio.GenBank._RecordConsumer.remark()=Bio.GenBank._RecordConsumer-class.html#remark"><a title="Bio.GenBank._FeatureConsumer.remark
Bio.GenBank._RecordConsumer.remark" class="py-name" href="#" onclick="return doclink('link-210', 'remark', 'link-210');">remark</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
</div><a name="L538"></a><tt class="py-lineno">538</tt>  <tt class="py-line"> </tt>
<a name="Reference.__str__"></a><div id="Reference.__str__-def"><a name="L539"></a><tt class="py-lineno">539</tt> <a class="py-toggle" href="#" id="Reference.__str__-toggle" onclick="return toggle('Reference.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Reference-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reference.__str__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Reference.__str__-expanded"><a name="L540"></a><tt class="py-lineno">540</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-211" class="py-name" targets="Method Bio.GenBank.Record.Reference._reference_line()=Bio.GenBank.Record.Reference-class.html#_reference_line"><a title="Bio.GenBank.Record.Reference._reference_line" class="py-name" href="#" onclick="return doclink('link-211', '_reference_line', 'link-211');">_reference_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L541"></a><tt class="py-lineno">541</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-212" class="py-name" targets="Method Bio.GenBank.Record.Reference._authors_line()=Bio.GenBank.Record.Reference-class.html#_authors_line"><a title="Bio.GenBank.Record.Reference._authors_line" class="py-name" href="#" onclick="return doclink('link-212', '_authors_line', 'link-212');">_authors_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L542"></a><tt class="py-lineno">542</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-213" class="py-name" targets="Method Bio.GenBank.Record.Reference._consrtm_line()=Bio.GenBank.Record.Reference-class.html#_consrtm_line"><a title="Bio.GenBank.Record.Reference._consrtm_line" class="py-name" href="#" onclick="return doclink('link-213', '_consrtm_line', 'link-213');">_consrtm_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L543"></a><tt class="py-lineno">543</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-214" class="py-name" targets="Method Bio.GenBank.Record.Reference._title_line()=Bio.GenBank.Record.Reference-class.html#_title_line"><a title="Bio.GenBank.Record.Reference._title_line" class="py-name" href="#" onclick="return doclink('link-214', '_title_line', 'link-214');">_title_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L544"></a><tt class="py-lineno">544</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-215" class="py-name" targets="Method Bio.GenBank.Record.Reference._journal_line()=Bio.GenBank.Record.Reference-class.html#_journal_line"><a title="Bio.GenBank.Record.Reference._journal_line" class="py-name" href="#" onclick="return doclink('link-215', '_journal_line', 'link-215');">_journal_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L545"></a><tt class="py-lineno">545</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-216" class="py-name" targets="Method Bio.GenBank.Record.Reference._medline_line()=Bio.GenBank.Record.Reference-class.html#_medline_line"><a title="Bio.GenBank.Record.Reference._medline_line" class="py-name" href="#" onclick="return doclink('link-216', '_medline_line', 'link-216');">_medline_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L546"></a><tt class="py-lineno">546</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-217" class="py-name" targets="Method Bio.GenBank.Record.Reference._pubmed_line()=Bio.GenBank.Record.Reference-class.html#_pubmed_line"><a title="Bio.GenBank.Record.Reference._pubmed_line" class="py-name" href="#" onclick="return doclink('link-217', '_pubmed_line', 'link-217');">_pubmed_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L547"></a><tt class="py-lineno">547</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-218" class="py-name" targets="Method Bio.GenBank.Record.Reference._remark_line()=Bio.GenBank.Record.Reference-class.html#_remark_line"><a title="Bio.GenBank.Record.Reference._remark_line" class="py-name" href="#" onclick="return doclink('link-218', '_remark_line', 'link-218');">_remark_line</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L548"></a><tt class="py-lineno">548</tt>  <tt class="py-line"> </tt>
<a name="L549"></a><tt class="py-lineno">549</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L550"></a><tt class="py-lineno">550</tt>  <tt class="py-line"> </tt>
<a name="Reference._reference_line"></a><div id="Reference._reference_line-def"><a name="L551"></a><tt class="py-lineno">551</tt> <a class="py-toggle" href="#" id="Reference._reference_line-toggle" onclick="return toggle('Reference._reference_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Reference-class.html#_reference_line">_reference_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reference._reference_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Reference._reference_line-expanded"><a name="L552"></a><tt class="py-lineno">552</tt>  <tt class="py-line">        <tt class="py-docstring">"""Output for REFERENCE lines.</tt> </tt>
<a name="L553"></a><tt class="py-lineno">553</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L554"></a><tt class="py-lineno">554</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt id="link-219" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-219', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-220" class="py-name"><a title="Bio.GenBank.Record.Record.BASE_FORMAT" class="py-name" href="#" onclick="return doclink('link-220', 'BASE_FORMAT', 'link-16');">BASE_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"REFERENCE"</tt> </tt>
<a name="L555"></a><tt class="py-lineno">555</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-221" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway.number" class="py-name" href="#" onclick="return doclink('link-221', 'number', 'link-203');">number</a></tt><tt class="py-op">:</tt> </tt>
<a name="L556"></a><tt class="py-lineno">556</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">bases</tt><tt class="py-op">:</tt> </tt>
<a name="L557"></a><tt class="py-lineno">557</tt>  <tt class="py-line">                <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">"%-3s"</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-222" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway.number" class="py-name" href="#" onclick="return doclink('link-222', 'number', 'link-203');">number</a></tt> </tt>
<a name="L558"></a><tt class="py-lineno">558</tt>  <tt class="py-line">                <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">"%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">bases</tt> </tt>
<a name="L559"></a><tt class="py-lineno">559</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L560"></a><tt class="py-lineno">560</tt>  <tt class="py-line">                <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">"%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-223" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway.number" class="py-name" href="#" onclick="return doclink('link-223', 'number', 'link-203');">number</a></tt> </tt>
<a name="L561"></a><tt class="py-lineno">561</tt>  <tt class="py-line"> </tt>
<a name="L562"></a><tt class="py-lineno">562</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">"\n"</tt> </tt>
<a name="L563"></a><tt class="py-lineno">563</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L564"></a><tt class="py-lineno">564</tt>  <tt class="py-line"> </tt>
<a name="Reference._authors_line"></a><div id="Reference._authors_line-def"><a name="L565"></a><tt class="py-lineno">565</tt> <a class="py-toggle" href="#" id="Reference._authors_line-toggle" onclick="return toggle('Reference._authors_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Reference-class.html#_authors_line">_authors_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reference._authors_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Reference._authors_line-expanded"><a name="L566"></a><tt class="py-lineno">566</tt>  <tt class="py-line">        <tt class="py-docstring">"""Output for AUTHORS information.</tt> </tt>
<a name="L567"></a><tt class="py-lineno">567</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L568"></a><tt class="py-lineno">568</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L569"></a><tt class="py-lineno">569</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-224" class="py-name"><a title="Bio.GenBank._FeatureConsumer.authors
Bio.GenBank._RecordConsumer.authors" class="py-name" href="#" onclick="return doclink('link-224', 'authors', 'link-204');">authors</a></tt><tt class="py-op">:</tt> </tt>
<a name="L570"></a><tt class="py-lineno">570</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-225" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-225', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-226" class="py-name"><a title="Bio.GenBank.Record.Record.INTERNAL_FORMAT" class="py-name" href="#" onclick="return doclink('link-226', 'INTERNAL_FORMAT', 'link-18');">INTERNAL_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"AUTHORS"</tt> </tt>
<a name="L571"></a><tt class="py-lineno">571</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-227" class="py-name"><a title="Bio.GenBank.Record._wrapped_genbank" class="py-name" href="#" onclick="return doclink('link-227', '_wrapped_genbank', 'link-92');">_wrapped_genbank</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-228" class="py-name"><a title="Bio.GenBank._FeatureConsumer.authors
Bio.GenBank._RecordConsumer.authors" class="py-name" href="#" onclick="return doclink('link-228', 'authors', 'link-204');">authors</a></tt><tt class="py-op">,</tt> <tt id="link-229" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-229', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-230" class="py-name"><a title="Bio.GenBank.Record.Record.GB_BASE_INDENT" class="py-name" href="#" onclick="return doclink('link-230', 'GB_BASE_INDENT', 'link-10');">GB_BASE_INDENT</a></tt><tt class="py-op">)</tt> </tt>
<a name="L572"></a><tt class="py-lineno">572</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L573"></a><tt class="py-lineno">573</tt>  <tt class="py-line"> </tt>
<a name="Reference._consrtm_line"></a><div id="Reference._consrtm_line-def"><a name="L574"></a><tt class="py-lineno">574</tt> <a class="py-toggle" href="#" id="Reference._consrtm_line-toggle" onclick="return toggle('Reference._consrtm_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Reference-class.html#_consrtm_line">_consrtm_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reference._consrtm_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Reference._consrtm_line-expanded"><a name="L575"></a><tt class="py-lineno">575</tt>  <tt class="py-line">        <tt class="py-docstring">"""Output for CONSRTM information.</tt> </tt>
<a name="L576"></a><tt class="py-lineno">576</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L577"></a><tt class="py-lineno">577</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L578"></a><tt class="py-lineno">578</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-231" class="py-name"><a title="Bio.GenBank._FeatureConsumer.consrtm
Bio.GenBank._RecordConsumer.consrtm" class="py-name" href="#" onclick="return doclink('link-231', 'consrtm', 'link-205');">consrtm</a></tt><tt class="py-op">:</tt> </tt>
<a name="L579"></a><tt class="py-lineno">579</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-232" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-232', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-233" class="py-name"><a title="Bio.GenBank.Record.Record.INTERNAL_FORMAT" class="py-name" href="#" onclick="return doclink('link-233', 'INTERNAL_FORMAT', 'link-18');">INTERNAL_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"CONSRTM"</tt> </tt>
<a name="L580"></a><tt class="py-lineno">580</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-234" class="py-name"><a title="Bio.GenBank.Record._wrapped_genbank" class="py-name" href="#" onclick="return doclink('link-234', '_wrapped_genbank', 'link-92');">_wrapped_genbank</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-235" class="py-name"><a title="Bio.GenBank._FeatureConsumer.consrtm
Bio.GenBank._RecordConsumer.consrtm" class="py-name" href="#" onclick="return doclink('link-235', 'consrtm', 'link-205');">consrtm</a></tt><tt class="py-op">,</tt> <tt id="link-236" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-236', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-237" class="py-name"><a title="Bio.GenBank.Record.Record.GB_BASE_INDENT" class="py-name" href="#" onclick="return doclink('link-237', 'GB_BASE_INDENT', 'link-10');">GB_BASE_INDENT</a></tt><tt class="py-op">)</tt> </tt>
<a name="L581"></a><tt class="py-lineno">581</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L582"></a><tt class="py-lineno">582</tt>  <tt class="py-line"> </tt>
<a name="Reference._title_line"></a><div id="Reference._title_line-def"><a name="L583"></a><tt class="py-lineno">583</tt> <a class="py-toggle" href="#" id="Reference._title_line-toggle" onclick="return toggle('Reference._title_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Reference-class.html#_title_line">_title_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reference._title_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Reference._title_line-expanded"><a name="L584"></a><tt class="py-lineno">584</tt>  <tt class="py-line">        <tt class="py-docstring">"""Output for TITLE information.</tt> </tt>
<a name="L585"></a><tt class="py-lineno">585</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L586"></a><tt class="py-lineno">586</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L587"></a><tt class="py-lineno">587</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-238" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-238', 'title', 'link-206');">title</a></tt><tt class="py-op">:</tt> </tt>
<a name="L588"></a><tt class="py-lineno">588</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-239" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-239', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-240" class="py-name"><a title="Bio.GenBank.Record.Record.INTERNAL_FORMAT" class="py-name" href="#" onclick="return doclink('link-240', 'INTERNAL_FORMAT', 'link-18');">INTERNAL_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"TITLE"</tt> </tt>
<a name="L589"></a><tt class="py-lineno">589</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-241" class="py-name"><a title="Bio.GenBank.Record._wrapped_genbank" class="py-name" href="#" onclick="return doclink('link-241', '_wrapped_genbank', 'link-92');">_wrapped_genbank</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-242" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-242', 'title', 'link-206');">title</a></tt><tt class="py-op">,</tt> <tt id="link-243" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-243', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-244" class="py-name"><a title="Bio.GenBank.Record.Record.GB_BASE_INDENT" class="py-name" href="#" onclick="return doclink('link-244', 'GB_BASE_INDENT', 'link-10');">GB_BASE_INDENT</a></tt><tt class="py-op">)</tt> </tt>
<a name="L590"></a><tt class="py-lineno">590</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L591"></a><tt class="py-lineno">591</tt>  <tt class="py-line"> </tt>
<a name="Reference._journal_line"></a><div id="Reference._journal_line-def"><a name="L592"></a><tt class="py-lineno">592</tt> <a class="py-toggle" href="#" id="Reference._journal_line-toggle" onclick="return toggle('Reference._journal_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Reference-class.html#_journal_line">_journal_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reference._journal_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Reference._journal_line-expanded"><a name="L593"></a><tt class="py-lineno">593</tt>  <tt class="py-line">        <tt class="py-docstring">"""Output for JOURNAL information.</tt> </tt>
<a name="L594"></a><tt class="py-lineno">594</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L595"></a><tt class="py-lineno">595</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L596"></a><tt class="py-lineno">596</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-245" class="py-name"><a title="Bio.GenBank._FeatureConsumer.journal
Bio.GenBank._RecordConsumer.journal" class="py-name" href="#" onclick="return doclink('link-245', 'journal', 'link-207');">journal</a></tt><tt class="py-op">:</tt> </tt>
<a name="L597"></a><tt class="py-lineno">597</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-246" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-246', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-247" class="py-name"><a title="Bio.GenBank.Record.Record.INTERNAL_FORMAT" class="py-name" href="#" onclick="return doclink('link-247', 'INTERNAL_FORMAT', 'link-18');">INTERNAL_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"JOURNAL"</tt> </tt>
<a name="L598"></a><tt class="py-lineno">598</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-248" class="py-name"><a title="Bio.GenBank.Record._wrapped_genbank" class="py-name" href="#" onclick="return doclink('link-248', '_wrapped_genbank', 'link-92');">_wrapped_genbank</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-249" class="py-name"><a title="Bio.GenBank._FeatureConsumer.journal
Bio.GenBank._RecordConsumer.journal" class="py-name" href="#" onclick="return doclink('link-249', 'journal', 'link-207');">journal</a></tt><tt class="py-op">,</tt> <tt id="link-250" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-250', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-251" class="py-name"><a title="Bio.GenBank.Record.Record.GB_BASE_INDENT" class="py-name" href="#" onclick="return doclink('link-251', 'GB_BASE_INDENT', 'link-10');">GB_BASE_INDENT</a></tt><tt class="py-op">)</tt> </tt>
<a name="L599"></a><tt class="py-lineno">599</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L600"></a><tt class="py-lineno">600</tt>  <tt class="py-line"> </tt>
<a name="Reference._medline_line"></a><div id="Reference._medline_line-def"><a name="L601"></a><tt class="py-lineno">601</tt> <a class="py-toggle" href="#" id="Reference._medline_line-toggle" onclick="return toggle('Reference._medline_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Reference-class.html#_medline_line">_medline_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reference._medline_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Reference._medline_line-expanded"><a name="L602"></a><tt class="py-lineno">602</tt>  <tt class="py-line">        <tt class="py-docstring">"""Output for MEDLINE information.</tt> </tt>
<a name="L603"></a><tt class="py-lineno">603</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L604"></a><tt class="py-lineno">604</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L605"></a><tt class="py-lineno">605</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-252" class="py-name"><a title="Bio.GenBank._FeatureConsumer.medline_id
Bio.GenBank._RecordConsumer.medline_id" class="py-name" href="#" onclick="return doclink('link-252', 'medline_id', 'link-208');">medline_id</a></tt><tt class="py-op">:</tt> </tt>
<a name="L606"></a><tt class="py-lineno">606</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-253" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-253', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-254" class="py-name"><a title="Bio.GenBank.Record.Record.INTERNAL_FORMAT" class="py-name" href="#" onclick="return doclink('link-254', 'INTERNAL_FORMAT', 'link-18');">INTERNAL_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"MEDLINE"</tt> </tt>
<a name="L607"></a><tt class="py-lineno">607</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-255" class="py-name"><a title="Bio.GenBank._FeatureConsumer.medline_id
Bio.GenBank._RecordConsumer.medline_id" class="py-name" href="#" onclick="return doclink('link-255', 'medline_id', 'link-208');">medline_id</a></tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt> </tt>
<a name="L608"></a><tt class="py-lineno">608</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L609"></a><tt class="py-lineno">609</tt>  <tt class="py-line"> </tt>
<a name="Reference._pubmed_line"></a><div id="Reference._pubmed_line-def"><a name="L610"></a><tt class="py-lineno">610</tt> <a class="py-toggle" href="#" id="Reference._pubmed_line-toggle" onclick="return toggle('Reference._pubmed_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Reference-class.html#_pubmed_line">_pubmed_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reference._pubmed_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Reference._pubmed_line-expanded"><a name="L611"></a><tt class="py-lineno">611</tt>  <tt class="py-line">        <tt class="py-docstring">"""Output for PUBMED information.</tt> </tt>
<a name="L612"></a><tt class="py-lineno">612</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L613"></a><tt class="py-lineno">613</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L614"></a><tt class="py-lineno">614</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-256" class="py-name"><a title="Bio.GenBank._FeatureConsumer.pubmed_id
Bio.GenBank._RecordConsumer.pubmed_id" class="py-name" href="#" onclick="return doclink('link-256', 'pubmed_id', 'link-209');">pubmed_id</a></tt><tt class="py-op">:</tt> </tt>
<a name="L615"></a><tt class="py-lineno">615</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-257" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-257', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-258" class="py-name"><a title="Bio.GenBank.Record.Record.OTHER_INTERNAL_FORMAT" class="py-name" href="#" onclick="return doclink('link-258', 'OTHER_INTERNAL_FORMAT', 'link-22');">OTHER_INTERNAL_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"PUBMED"</tt> </tt>
<a name="L616"></a><tt class="py-lineno">616</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-259" class="py-name"><a title="Bio.GenBank._FeatureConsumer.pubmed_id
Bio.GenBank._RecordConsumer.pubmed_id" class="py-name" href="#" onclick="return doclink('link-259', 'pubmed_id', 'link-209');">pubmed_id</a></tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt> </tt>
<a name="L617"></a><tt class="py-lineno">617</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div><a name="L618"></a><tt class="py-lineno">618</tt>  <tt class="py-line"> </tt>
<a name="Reference._remark_line"></a><div id="Reference._remark_line-def"><a name="L619"></a><tt class="py-lineno">619</tt> <a class="py-toggle" href="#" id="Reference._remark_line-toggle" onclick="return toggle('Reference._remark_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Reference-class.html#_remark_line">_remark_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Reference._remark_line-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Reference._remark_line-expanded"><a name="L620"></a><tt class="py-lineno">620</tt>  <tt class="py-line">        <tt class="py-docstring">"""Output for REMARK information.</tt> </tt>
<a name="L621"></a><tt class="py-lineno">621</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L622"></a><tt class="py-lineno">622</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L623"></a><tt class="py-lineno">623</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-260" class="py-name"><a title="Bio.GenBank._FeatureConsumer.remark
Bio.GenBank._RecordConsumer.remark" class="py-name" href="#" onclick="return doclink('link-260', 'remark', 'link-210');">remark</a></tt><tt class="py-op">:</tt> </tt>
<a name="L624"></a><tt class="py-lineno">624</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-261" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-261', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-262" class="py-name"><a title="Bio.GenBank.Record.Record.INTERNAL_FORMAT" class="py-name" href="#" onclick="return doclink('link-262', 'INTERNAL_FORMAT', 'link-18');">INTERNAL_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-string">"REMARK"</tt> </tt>
<a name="L625"></a><tt class="py-lineno">625</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-263" class="py-name"><a title="Bio.GenBank.Record._wrapped_genbank" class="py-name" href="#" onclick="return doclink('link-263', '_wrapped_genbank', 'link-92');">_wrapped_genbank</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-264" class="py-name"><a title="Bio.GenBank._FeatureConsumer.remark
Bio.GenBank._RecordConsumer.remark" class="py-name" href="#" onclick="return doclink('link-264', 'remark', 'link-210');">remark</a></tt><tt class="py-op">,</tt> <tt id="link-265" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-265', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-266" class="py-name"><a title="Bio.GenBank.Record.Record.GB_BASE_INDENT" class="py-name" href="#" onclick="return doclink('link-266', 'GB_BASE_INDENT', 'link-10');">GB_BASE_INDENT</a></tt><tt class="py-op">)</tt> </tt>
<a name="L626"></a><tt class="py-lineno">626</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div></div><a name="L627"></a><tt class="py-lineno">627</tt>  <tt class="py-line"> </tt>
<a name="L628"></a><tt class="py-lineno">628</tt>  <tt class="py-line"> </tt>
<a name="Feature"></a><div id="Feature-def"><a name="L629"></a><tt class="py-lineno">629</tt> <a class="py-toggle" href="#" id="Feature-toggle" onclick="return toggle('Feature');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.GenBank.Record.Feature-class.html">Feature</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Feature-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Feature-expanded"><a name="L630"></a><tt class="py-lineno">630</tt>  <tt class="py-line">    <tt class="py-docstring">"""Hold information about a Feature in the Feature Table of GenBank record.</tt> </tt>
<a name="L631"></a><tt class="py-lineno">631</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L632"></a><tt class="py-lineno">632</tt>  <tt class="py-line"><tt class="py-docstring">    Attributes:</tt> </tt>
<a name="L633"></a><tt class="py-lineno">633</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L634"></a><tt class="py-lineno">634</tt>  <tt class="py-line"><tt class="py-docstring">        - key - The key name of the featue (ie. source)</tt> </tt>
<a name="L635"></a><tt class="py-lineno">635</tt>  <tt class="py-line"><tt class="py-docstring">        - location - The string specifying the location of the feature.</tt> </tt>
<a name="L636"></a><tt class="py-lineno">636</tt>  <tt class="py-line"><tt class="py-docstring">        - qualfiers - A listing Qualifier objects in the feature.</tt> </tt>
<a name="L637"></a><tt class="py-lineno">637</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="Feature.__init__"></a><div id="Feature.__init__-def"><a name="L638"></a><tt class="py-lineno">638</tt> <a class="py-toggle" href="#" id="Feature.__init__-toggle" onclick="return toggle('Feature.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Feature-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Feature.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Feature.__init__-expanded"><a name="L639"></a><tt class="py-lineno">639</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">key</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L640"></a><tt class="py-lineno">640</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-267" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.location()=Bio.GenBank._FeatureConsumer-class.html#location,Method Bio.GenBank._RecordConsumer.location()=Bio.GenBank._RecordConsumer-class.html#location,Method Bio.Phylo.PhyloXMLIO.Writer.location()=Bio.Phylo.PhyloXMLIO.Writer-class.html#location"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-267', 'location', 'link-267');">location</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L641"></a><tt class="py-lineno">641</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">qualifiers</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div><a name="L642"></a><tt class="py-lineno">642</tt>  <tt class="py-line"> </tt>
<a name="Feature.__str__"></a><div id="Feature.__str__-def"><a name="L643"></a><tt class="py-lineno">643</tt> <a class="py-toggle" href="#" id="Feature.__str__-toggle" onclick="return toggle('Feature.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Feature-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Feature.__str__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Feature.__str__-expanded"><a name="L644"></a><tt class="py-lineno">644</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">=</tt> <tt id="link-268" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-268', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-269" class="py-name"><a title="Bio.GenBank.Record.Record.INTERNAL_FEATURE_FORMAT" class="py-name" href="#" onclick="return doclink('link-269', 'INTERNAL_FEATURE_FORMAT', 'link-28');">INTERNAL_FEATURE_FORMAT</a></tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">key</tt> </tt>
<a name="L645"></a><tt class="py-lineno">645</tt>  <tt class="py-line">        <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-270" class="py-name"><a title="Bio.GenBank.Record._wrapped_genbank" class="py-name" href="#" onclick="return doclink('link-270', '_wrapped_genbank', 'link-92');">_wrapped_genbank</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-271" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-271', 'location', 'link-267');">location</a></tt><tt class="py-op">,</tt> <tt id="link-272" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-272', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-273" class="py-name"><a title="Bio.GenBank.Record.Record.GB_FEATURE_INDENT" class="py-name" href="#" onclick="return doclink('link-273', 'GB_FEATURE_INDENT', 'link-11');">GB_FEATURE_INDENT</a></tt><tt class="py-op">,</tt> </tt>
<a name="L646"></a><tt class="py-lineno">646</tt>  <tt class="py-line">                                   <tt class="py-name">split_char</tt><tt class="py-op">=</tt><tt class="py-string">','</tt><tt class="py-op">)</tt> </tt>
<a name="L647"></a><tt class="py-lineno">647</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">qualifier</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">qualifiers</tt><tt class="py-op">:</tt> </tt>
<a name="L648"></a><tt class="py-lineno">648</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt class="py-string">" "</tt> <tt class="py-op">*</tt> <tt id="link-274" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-274', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-275" class="py-name"><a title="Bio.GenBank.Record.Record.GB_FEATURE_INDENT" class="py-name" href="#" onclick="return doclink('link-275', 'GB_FEATURE_INDENT', 'link-11');">GB_FEATURE_INDENT</a></tt> </tt>
<a name="L649"></a><tt class="py-lineno">649</tt>  <tt class="py-line"> </tt>
<a name="L650"></a><tt class="py-lineno">650</tt>  <tt class="py-line">            <tt class="py-comment"># determine whether we can wrap on spaces</tt> </tt>
<a name="L651"></a><tt class="py-lineno">651</tt>  <tt class="py-line">            <tt class="py-name">space_wrap</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L652"></a><tt class="py-lineno">652</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">no_space_key</tt> <tt class="py-keyword">in</tt> \ </tt>
<a name="L653"></a><tt class="py-lineno">653</tt>  <tt class="py-line">                    <tt id="link-276" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-276', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-277" class="py-name"><a title="Bio.GenBank" class="py-name" href="#" onclick="return doclink('link-277', 'GenBank', 'link-1');">GenBank</a></tt><tt class="py-op">.</tt><tt id="link-278" class="py-name" targets="Class Bio.GenBank._BaseGenBankConsumer=Bio.GenBank._BaseGenBankConsumer-class.html"><a title="Bio.GenBank._BaseGenBankConsumer" class="py-name" href="#" onclick="return doclink('link-278', '_BaseGenBankConsumer', 'link-278');">_BaseGenBankConsumer</a></tt><tt class="py-op">.</tt><tt id="link-279" class="py-name" targets="Variable Bio.GenBank._BaseGenBankConsumer.remove_space_keys=Bio.GenBank._BaseGenBankConsumer-class.html#remove_space_keys"><a title="Bio.GenBank._BaseGenBankConsumer.remove_space_keys" class="py-name" href="#" onclick="return doclink('link-279', 'remove_space_keys', 'link-279');">remove_space_keys</a></tt><tt class="py-op">:</tt> </tt>
<a name="L654"></a><tt class="py-lineno">654</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">no_space_key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qualifier</tt><tt class="py-op">.</tt><tt class="py-name">key</tt><tt class="py-op">:</tt> </tt>
<a name="L655"></a><tt class="py-lineno">655</tt>  <tt class="py-line">                    <tt class="py-name">space_wrap</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L656"></a><tt class="py-lineno">656</tt>  <tt class="py-line"> </tt>
<a name="L657"></a><tt class="py-lineno">657</tt>  <tt class="py-line">            <tt class="py-name">output</tt> <tt class="py-op">+=</tt> <tt id="link-280" class="py-name"><a title="Bio.GenBank.Record._wrapped_genbank" class="py-name" href="#" onclick="return doclink('link-280', '_wrapped_genbank', 'link-92');">_wrapped_genbank</a></tt><tt class="py-op">(</tt><tt class="py-name">qualifier</tt><tt class="py-op">.</tt><tt class="py-name">key</tt> <tt class="py-op">+</tt> <tt class="py-name">qualifier</tt><tt class="py-op">.</tt><tt id="link-281" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.value()=Bio.Phylo.PhyloXMLIO.Writer-class.html#value"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-281', 'value', 'link-281');">value</a></tt><tt class="py-op">,</tt> </tt>
<a name="L658"></a><tt class="py-lineno">658</tt>  <tt class="py-line">                                       <tt id="link-282" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-282', 'Record', 'link-2');">Record</a></tt><tt class="py-op">.</tt><tt id="link-283" class="py-name"><a title="Bio.GenBank.Record.Record.GB_FEATURE_INDENT" class="py-name" href="#" onclick="return doclink('link-283', 'GB_FEATURE_INDENT', 'link-11');">GB_FEATURE_INDENT</a></tt><tt class="py-op">,</tt> <tt class="py-name">space_wrap</tt><tt class="py-op">)</tt> </tt>
<a name="L659"></a><tt class="py-lineno">659</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">output</tt> </tt>
</div></div><a name="L660"></a><tt class="py-lineno">660</tt>  <tt class="py-line"> </tt>
<a name="L661"></a><tt class="py-lineno">661</tt>  <tt class="py-line"> </tt>
<a name="Qualifier"></a><div id="Qualifier-def"><a name="L662"></a><tt class="py-lineno">662</tt> <a class="py-toggle" href="#" id="Qualifier-toggle" onclick="return toggle('Qualifier');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.GenBank.Record.Qualifier-class.html">Qualifier</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Qualifier-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Qualifier-expanded"><a name="L663"></a><tt class="py-lineno">663</tt>  <tt class="py-line">    <tt class="py-docstring">"""Hold information about a qualifier in a GenBank feature.</tt> </tt>
<a name="L664"></a><tt class="py-lineno">664</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L665"></a><tt class="py-lineno">665</tt>  <tt class="py-line"><tt class="py-docstring">    Attributes:</tt> </tt>
<a name="L666"></a><tt class="py-lineno">666</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L667"></a><tt class="py-lineno">667</tt>  <tt class="py-line"><tt class="py-docstring">        - key - The key name of the qualifier (ie. /organism=)</tt> </tt>
<a name="L668"></a><tt class="py-lineno">668</tt>  <tt class="py-line"><tt class="py-docstring">        - value - The value of the qualifier ("Dictyostelium discoideum").</tt> </tt>
<a name="L669"></a><tt class="py-lineno">669</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="Qualifier.__init__"></a><div id="Qualifier.__init__-def"><a name="L670"></a><tt class="py-lineno">670</tt> <a class="py-toggle" href="#" id="Qualifier.__init__-toggle" onclick="return toggle('Qualifier.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.GenBank.Record.Qualifier-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Qualifier.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Qualifier.__init__-expanded"><a name="L671"></a><tt class="py-lineno">671</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">key</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L672"></a><tt class="py-lineno">672</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-284" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-284', 'value', 'link-281');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
</div></div><a name="L673"></a><tt class="py-lineno">673</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Oct 21 18:04:54 2015
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
