<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.SeqUtils</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        Package&nbsp;SeqUtils
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.SeqUtils-module.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== PACKAGE DESCRIPTION ==================== -->
<h1 class="epydoc">Package SeqUtils</h1><p class="nomargin-top"><span class="codelink"><a href="Bio.SeqUtils-pysrc.html">source&nbsp;code</a></span></p>
Miscellaneous functions for dealing with sequences.

<!-- ==================== SUBMODULES ==================== -->
<a name="section-Submodules"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Submodules</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Submodules"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
  <tr><td class="summary">
  <ul class="nomargin">
    <li> <strong class="uidlink"><a href="Bio.SeqUtils.CheckSum-module.html">Bio.SeqUtils.CheckSum</a></strong>: <em class="summary">Functions to calculate assorted sequence checksums.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqUtils.CodonUsage-module.html">Bio.SeqUtils.CodonUsage</a></strong>: <em class="summary">Methods for codon usage calculations.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqUtils.CodonUsageIndices-module.html">Bio.SeqUtils.CodonUsageIndices</a></strong>: <em class="summary">Codon adaption indxes, including Sharp and Li (1987) E. coli index.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqUtils.IsoelectricPoint-module.html">Bio.SeqUtils.IsoelectricPoint</a></strong>: <em class="summary">Calculate isoelectric points of polypeptides using methods of Bjellqvist.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqUtils.MeltingTemp-module.html">Bio.SeqUtils.MeltingTemp</a></strong>: <em class="summary">Calculate the melting temperature of nucleotide sequences.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqUtils.ProtParam-module.html">Bio.SeqUtils.ProtParam</a></strong>: <em class="summary">Simple protein analysis.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqUtils.ProtParamData-module.html">Bio.SeqUtils.ProtParamData</a></strong>: <em class="summary">Indices to be used with ProtParam.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqUtils.lcc-module.html">Bio.SeqUtils.lcc</a></strong>: <em class="summary">Local Composition Complexity.</em>    </li>
  </ul></td></tr>
</table>

<br />
<!-- ==================== FUNCTIONS ==================== -->
<a name="section-Functions"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Functions</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Functions"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqUtils-module.html#GC" class="summary-sig-name">GC</a>(<span class="summary-sig-arg">seq</span>)</span><br />
      Calculate G+C content, return percentage (as float between 0 and 100).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqUtils-pysrc.html#GC">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqUtils-module.html#GC123" class="summary-sig-name">GC123</a>(<span class="summary-sig-arg">seq</span>)</span><br />
      Calculate G+C content: total, for first, second and third positions.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqUtils-pysrc.html#GC123">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqUtils-module.html#GC_skew" class="summary-sig-name">GC_skew</a>(<span class="summary-sig-arg">seq</span>,
        <span class="summary-sig-arg">window</span>=<span class="summary-sig-default">100</span>)</span><br />
      Calculate GC skew (G-C)/(G+C) for multiple windows along the sequence.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqUtils-pysrc.html#GC_skew">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="xGC_skew"></a><span class="summary-sig-name">xGC_skew</span>(<span class="summary-sig-arg">seq</span>,
        <span class="summary-sig-arg">window</span>=<span class="summary-sig-default">1000</span>,
        <span class="summary-sig-arg">zoom</span>=<span class="summary-sig-default">100</span>,
        <span class="summary-sig-arg">r</span>=<span class="summary-sig-default">300</span>,
        <span class="summary-sig-arg">px</span>=<span class="summary-sig-default">100</span>,
        <span class="summary-sig-arg">py</span>=<span class="summary-sig-default">100</span>)</span><br />
      Calculate and plot normal and accumulated GC skew (GRAPHICS !!!).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqUtils-pysrc.html#xGC_skew">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqUtils-module.html#nt_search" class="summary-sig-name">nt_search</a>(<span class="summary-sig-arg">seq</span>,
        <span class="summary-sig-arg">subseq</span>)</span><br />
      Search for a DNA subseq in sequence, return list of [subseq, positions].</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqUtils-pysrc.html#nt_search">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqUtils-module.html#seq3" class="summary-sig-name">seq3</a>(<span class="summary-sig-arg">seq</span>,
        <span class="summary-sig-arg">custom_map</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">undef_code</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">Xaa</code><code class="variable-quote">'</code></span>)</span><br />
      Convert protein sequence from one-letter to three-letter code.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqUtils-pysrc.html#seq3">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqUtils-module.html#seq1" class="summary-sig-name">seq1</a>(<span class="summary-sig-arg">seq</span>,
        <span class="summary-sig-arg">custom_map</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">undef_code</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">X</code><code class="variable-quote">'</code></span>)</span><br />
      Convert protein sequence from three-letter to one-letter code.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqUtils-pysrc.html#seq1">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqUtils-module.html#molecular_weight" class="summary-sig-name">molecular_weight</a>(<span class="summary-sig-arg">seq</span>,
        <span class="summary-sig-arg">seq_type</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">double_stranded</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">circular</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">monoisotopic</span>=<span class="summary-sig-default">False</span>)</span><br />
      Calculate the molecular mass of DNA, RNA or protein sequences as float.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqUtils-pysrc.html#molecular_weight">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SeqUtils-module.html#six_frame_translations" class="summary-sig-name">six_frame_translations</a>(<span class="summary-sig-arg">seq</span>,
        <span class="summary-sig-arg">genetic_code</span>=<span class="summary-sig-default">1</span>)</span><br />
      Return pretty string showing the 6 frame translations and GC content.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SeqUtils-pysrc.html#six_frame_translations">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
</table>
<!-- ==================== VARIABLES ==================== -->
<a name="section-Variables"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Variables</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Variables"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="__package__"></a><span class="summary-name">__package__</span> = <code title="'Bio.SeqUtils'"><code class="variable-quote">'</code><code class="variable-string">Bio.SeqUtils</code><code class="variable-quote">'</code></code>
    </td>
  </tr>
</table>
<!-- ==================== FUNCTION DETAILS ==================== -->
<a name="section-FunctionDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Function Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-FunctionDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="GC"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">GC</span>(<span class="sig-arg">seq</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqUtils-pysrc.html#GC">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculate G+C content, return percentage (as float between 0 and 100).</p>
<p>Copes mixed case sequences, and with the ambiguous nucleotide S (G or C)
when counting the G and C content.  The percentage is calculated against
the full length, e.g.:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqUtils <span class="py-keyword">import</span> GC
<span class="py-prompt">&gt;&gt;&gt; </span>GC(<span class="py-string">&quot;ACTGN&quot;</span>)
<span class="py-output">40.0</span></pre>
<p>Note that this will return zero for an empty sequence.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="GC123"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">GC123</span>(<span class="sig-arg">seq</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqUtils-pysrc.html#GC123">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculate G+C content: total, for first, second and third positions.</p>
<p>Returns a tuple of four floats (percentages between 0 and 100) for the
entire sequence, and the three codon positions.  e.g.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqUtils <span class="py-keyword">import</span> GC123
<span class="py-prompt">&gt;&gt;&gt; </span>GC123(<span class="py-string">&quot;ACTGTN&quot;</span>)
<span class="py-output">(40.0, 50.0, 50.0, 0.0)</span></pre>
<p>Copes with mixed case sequences, but does NOT deal with ambiguous
nucleotides.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="GC_skew"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">GC_skew</span>(<span class="sig-arg">seq</span>,
        <span class="sig-arg">window</span>=<span class="sig-default">100</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqUtils-pysrc.html#GC_skew">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculate GC skew (G-C)/(G+C) for multiple windows along the sequence.</p>
<p>Returns a list of ratios (floats), controlled by the length of the sequence
and the size of the window.</p>
<p>Returns 0 for windows without any G/C by handling zero division errors.</p>
<p>Does NOT look at any ambiguous nucleotides.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="nt_search"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">nt_search</span>(<span class="sig-arg">seq</span>,
        <span class="sig-arg">subseq</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqUtils-pysrc.html#nt_search">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Search for a DNA subseq in sequence, return list of [subseq, positions].</p>
<p>Use ambiguous values (like N = A or T or C or G, R = A or G etc.),
searches only on forward strand.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="seq3"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">seq3</span>(<span class="sig-arg">seq</span>,
        <span class="sig-arg">custom_map</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">undef_code</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">Xaa</code><code class="variable-quote">'</code></span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqUtils-pysrc.html#seq3">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Convert protein sequence from one-letter to three-letter code.</p>
<p>The single required input argument 'seq' should be a protein sequence using
single letter codes, either as a Python string or as a Seq or MutableSeq
object.</p>
<p>This function returns the amino acid sequence as a string using the three
letter amino acid codes. Output follows the IUPAC standard (including
ambiguous characters B for &quot;Asx&quot;, J for &quot;Xle&quot; and X for &quot;Xaa&quot;, and also U
for &quot;Sel&quot; and O for &quot;Pyl&quot;) plus &quot;Ter&quot; for a terminator given as an
asterisk. Any unknown character (including possible gap characters),
is changed into 'Xaa' by default.</p>
<p>e.g.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqUtils <span class="py-keyword">import</span> seq3
<span class="py-prompt">&gt;&gt;&gt; </span>seq3(<span class="py-string">&quot;MAIVMGRWKGAR*&quot;</span>)
<span class="py-output">'MetAlaIleValMetGlyArgTrpLysGlyAlaArgTer'</span></pre>
<p>You can set a custom translation of the codon termination code using the
dictionary &quot;custom_map&quot; argument (which defaults to {'*': 'Ter'}), e.g.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>seq3(<span class="py-string">&quot;MAIVMGRWKGAR*&quot;</span>, custom_map={<span class="py-string">&quot;*&quot;</span>: <span class="py-string">&quot;***&quot;</span>})
<span class="py-output">'MetAlaIleValMetGlyArgTrpLysGlyAlaArg***'</span></pre>
<p>You can also set a custom translation for non-amino acid characters, such
as '-', using the &quot;undef_code&quot; argument, e.g.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>seq3(<span class="py-string">&quot;MAIVMGRWKGA--R*&quot;</span>, undef_code=<span class="py-string">'---'</span>)
<span class="py-output">'MetAlaIleValMetGlyArgTrpLysGlyAla------ArgTer'</span></pre>
<p>If not given, &quot;undef_code&quot; defaults to &quot;Xaa&quot;, e.g.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>seq3(<span class="py-string">&quot;MAIVMGRWKGA--R*&quot;</span>)
<span class="py-output">'MetAlaIleValMetGlyArgTrpLysGlyAlaXaaXaaArgTer'</span></pre>
<p>This function was inspired by BioPerl's seq3.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="seq1"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">seq1</span>(<span class="sig-arg">seq</span>,
        <span class="sig-arg">custom_map</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">undef_code</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">X</code><code class="variable-quote">'</code></span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqUtils-pysrc.html#seq1">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Convert protein sequence from three-letter to one-letter code.</p>
<p>The single required input argument 'seq' should be a protein sequence
using three-letter codes, either as a Python string or as a Seq or
MutableSeq object.</p>
<p>This function returns the amino acid sequence as a string using the one
letter amino acid codes. Output follows the IUPAC standard (including
ambiguous characters &quot;B&quot; for &quot;Asx&quot;, &quot;J&quot; for &quot;Xle&quot;, &quot;X&quot; for &quot;Xaa&quot;, &quot;U&quot; for
&quot;Sel&quot;, and &quot;O&quot; for &quot;Pyl&quot;) plus &quot;*&quot; for a terminator given the &quot;Ter&quot; code.
Any unknown character (including possible gap characters), is changed
into '-' by default.</p>
<p>e.g.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqUtils <span class="py-keyword">import</span> seq1
<span class="py-prompt">&gt;&gt;&gt; </span>seq1(<span class="py-string">&quot;MetAlaIleValMetGlyArgTrpLysGlyAlaArgTer&quot;</span>)
<span class="py-output">'MAIVMGRWKGAR*'</span></pre>
<p>The input is case insensitive, e.g.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqUtils <span class="py-keyword">import</span> seq1
<span class="py-prompt">&gt;&gt;&gt; </span>seq1(<span class="py-string">&quot;METalaIlEValMetGLYArgtRplysGlyAlaARGTer&quot;</span>)
<span class="py-output">'MAIVMGRWKGAR*'</span></pre>
<p>You can set a custom translation of the codon termination code using the
dictionary &quot;custom_map&quot; argument (defaulting to {'Ter': '*'}), e.g.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>seq1(<span class="py-string">&quot;MetAlaIleValMetGlyArgTrpLysGlyAla***&quot;</span>, custom_map={<span class="py-string">&quot;***&quot;</span>: <span class="py-string">&quot;*&quot;</span>})
<span class="py-output">'MAIVMGRWKGA*'</span></pre>
<p>You can also set a custom translation for non-amino acid characters, such
as '-', using the &quot;undef_code&quot; argument, e.g.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>seq1(<span class="py-string">&quot;MetAlaIleValMetGlyArgTrpLysGlyAla------ArgTer&quot;</span>, undef_code=<span class="py-string">'?'</span>)
<span class="py-output">'MAIVMGRWKGA??R*'</span></pre>
<p>If not given, &quot;undef_code&quot; defaults to &quot;X&quot;, e.g.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>seq1(<span class="py-string">&quot;MetAlaIleValMetGlyArgTrpLysGlyAla------ArgTer&quot;</span>)
<span class="py-output">'MAIVMGRWKGAXXR*'</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="molecular_weight"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">molecular_weight</span>(<span class="sig-arg">seq</span>,
        <span class="sig-arg">seq_type</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">double_stranded</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">circular</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">monoisotopic</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqUtils-pysrc.html#molecular_weight">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Calculate the molecular mass of DNA, RNA or protein sequences as float.</p>
<p>Only unambiguous letters are allowed. Nucleotide sequences are assumed to
have a 5' phosphate.</p>
<dl class="rst-docutils">
<dt>Arguments:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>seq: String or Biopython sequence object.</li>
<li>seq_type: The default (None) is to take the alphabet from the seq
argument, or assume DNA if the seq argument is a string. Override this
with a string 'DNA', 'RNA', or 'protein'.</li>
<li>double_stranded: Calculate the mass for the double stranded molecule?</li>
<li>circular: Is the molecule circular (has no ends)?</li>
<li>monoisotopic: Use the monoisotopic mass tables?</li>
</ul>
</dd>
</dl>
<p>Note that for backwards compatibility, if the seq argument is a string,
or Seq object with a generic alphabet, and no seq_type is specified
(i.e. left as None), then DNA is assumed.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(<span class="py-string">&quot;%0.2f&quot;</span> % molecular_weight(<span class="py-string">&quot;AGC&quot;</span>))
<span class="py-output">949.61</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(<span class="py-string">&quot;%0.2f&quot;</span> % molecular_weight(Seq(<span class="py-string">&quot;AGC&quot;</span>)))
<span class="py-output">949.61</span></pre>
<p>However, it is better to be explicit - for example with strings:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(<span class="py-string">&quot;%0.2f&quot;</span> % molecular_weight(<span class="py-string">&quot;AGC&quot;</span>, <span class="py-string">&quot;DNA&quot;</span>))
<span class="py-output">949.61</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(<span class="py-string">&quot;%0.2f&quot;</span> % molecular_weight(<span class="py-string">&quot;AGC&quot;</span>, <span class="py-string">&quot;RNA&quot;</span>))
<span class="py-output">997.61</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(<span class="py-string">&quot;%0.2f&quot;</span> % molecular_weight(<span class="py-string">&quot;AGC&quot;</span>, <span class="py-string">&quot;protein&quot;</span>))
<span class="py-output">249.29</span></pre>
<p>Or, with the sequence alphabet:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> generic_dna, generic_rna, generic_protein
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(<span class="py-string">&quot;%0.2f&quot;</span> % molecular_weight(Seq(<span class="py-string">&quot;AGC&quot;</span>, generic_dna)))
<span class="py-output">949.61</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(<span class="py-string">&quot;%0.2f&quot;</span> % molecular_weight(Seq(<span class="py-string">&quot;AGC&quot;</span>, generic_rna)))
<span class="py-output">997.61</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(<span class="py-string">&quot;%0.2f&quot;</span> % molecular_weight(Seq(<span class="py-string">&quot;AGC&quot;</span>, generic_protein)))
<span class="py-output">249.29</span></pre>
<p>Also note that contradictory sequence alphabets and seq_type will also
give an exception:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> generic_dna
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(<span class="py-string">&quot;%0.2f&quot;</span> % molecular_weight(Seq(<span class="py-string">&quot;AGC&quot;</span>, generic_dna), <span class="py-string">&quot;RNA&quot;</span>))
<span class="py-except">Traceback (most recent call last):</span>
<span class="py-except">  ...</span>
<span class="py-except">ValueError: seq_type='RNA' contradicts DNA from seq alphabet</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="six_frame_translations"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">six_frame_translations</span>(<span class="sig-arg">seq</span>,
        <span class="sig-arg">genetic_code</span>=<span class="sig-default">1</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SeqUtils-pysrc.html#six_frame_translations">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Return pretty string showing the 6 frame translations and GC content.</p>
<p>Nice looking 6 frame translation with GC content - code from xbbtools
similar to DNA Striders six-frame translation</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqUtils <span class="py-keyword">import</span> six_frame_translations
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(six_frame_translations(<span class="py-string">&quot;AUGGCCAUUGUAAUGGGCCGCUGA&quot;</span>))
<span class="py-output">GC_Frame: a:5 t:0 g:8 c:5</span>
<span class="py-output">Sequence: auggccauug ... gggccgcuga, 24 nt, 54.17 %GC</span>
<span class="py-output">&lt;BLANKLINE&gt;</span>
<span class="py-output">&lt;BLANKLINE&gt;</span>
<span class="py-output">1/1</span>
<span class="py-output">  G  H  C  N  G  P  L</span>
<span class="py-output"> W  P  L  *  W  A  A</span>
<span class="py-output">M  A  I  V  M  G  R  *</span>
<span class="py-output">auggccauuguaaugggccgcuga   54 %</span>
<span class="py-output">uaccgguaacauuacccggcgacu</span>
<span class="py-output">A  M  T  I  P  R  Q</span>
<span class="py-output"> H  G  N  Y  H  A  A  S</span>
<span class="py-output">  P  W  Q  L  P  G  S</span>
<span class="py-output">&lt;BLANKLINE&gt;</span>
<span class="py-output">&lt;BLANKLINE&gt;</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Nov  6 13:32:24 2019
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
