<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.codonalign.codonalignment</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.codonalign-module.html">Package&nbsp;codonalign</a> ::
        Module&nbsp;codonalignment
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.codonalign.codonalignment-module.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== MODULE DESCRIPTION ==================== -->
<h1 class="epydoc">Module codonalignment</h1><p class="nomargin-top"><span class="codelink"><a href="Bio.codonalign.codonalignment-pysrc.html">source&nbsp;code</a></span></p>
<p>Code for dealing with Codon Alignment.</p>
<p>CodonAlignment class is interited from MultipleSeqAlignment class. This is
the core class to deal with codon alignment in biopython.</p>

<!-- ==================== CLASSES ==================== -->
<a name="section-Classes"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Classes</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Classes"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.codonalign.codonalignment.CodonAlignment-class.html" class="summary-name">CodonAlignment</a><br />
      Codon Alignment class that inherits from MultipleSeqAlignment.
    </td>
  </tr>
</table>
<!-- ==================== FUNCTIONS ==================== -->
<a name="section-Functions"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Functions</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Functions"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.codonalign.codonalignment-module.html#mktest" class="summary-sig-name">mktest</a>(<span class="summary-sig-arg">codon_alns</span>,
        <span class="summary-sig-arg">codon_table</span>=<span class="summary-sig-default">default_codon_table</span>,
        <span class="summary-sig-arg">alpha</span>=<span class="summary-sig-default">0.05</span>)</span><br />
      McDonald-Kreitman test for neutrality (PMID: 1904993) This method
counts changes rather than sites (<a class="rst-reference external" href="http://mkt.uab.es/mkt/help_mkt.asp" target="_top">http://mkt.uab.es/mkt/help_mkt.asp</a>).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.codonalign.codonalignment-pysrc.html#mktest">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.codonalign.codonalignment-module.html#_get_codon2codon_matrix" class="summary-sig-name" onclick="show_private();">_get_codon2codon_matrix</a>(<span class="summary-sig-arg">codon_table</span>=<span class="summary-sig-default">default_codon_table</span>)</span><br />
      Function to get codon codon subsitution matrix.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.codonalign.codonalignment-pysrc.html#_get_codon2codon_matrix">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.codonalign.codonalignment-module.html#_dijkstra" class="summary-sig-name" onclick="show_private();">_dijkstra</a>(<span class="summary-sig-arg">graph</span>,
        <span class="summary-sig-arg">start</span>,
        <span class="summary-sig-arg">end</span>)</span><br />
      Dijkstra's algorithm Python implementation.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.codonalign.codonalignment-pysrc.html#_dijkstra">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_count_replacement"></a><span class="summary-sig-name">_count_replacement</span>(<span class="summary-sig-arg">codon_set</span>,
        <span class="summary-sig-arg">G</span>)</span><br />
      Count replacement needed for a given codon_set (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.codonalign.codonalignment-pysrc.html#_count_replacement">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.codonalign.codonalignment-module.html#_prim" class="summary-sig-name" onclick="show_private();">_prim</a>(<span class="summary-sig-arg">G</span>)</span><br />
      Prim's algorithm to find minimum spanning tree.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.codonalign.codonalignment-pysrc.html#_prim">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_get_subgraph"></a><span class="summary-sig-name">_get_subgraph</span>(<span class="summary-sig-arg">codons</span>,
        <span class="summary-sig-arg">G</span>)</span><br />
      Get the subgraph that contains all codons in list (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.codonalign.codonalignment-pysrc.html#_get_subgraph">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.codonalign.codonalignment-module.html#_G_test" class="summary-sig-name" onclick="show_private();">_G_test</a>(<span class="summary-sig-arg">site_counts</span>)</span><br />
      G test for 2x2 contingency table (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.codonalign.codonalignment-pysrc.html#_G_test">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
</table>
<!-- ==================== VARIABLES ==================== -->
<a name="section-Variables"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Variables</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Variables"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="__package__"></a><span class="summary-name">__package__</span> = <code title="'Bio.codonalign'"><code class="variable-quote">'</code><code class="variable-string">Bio.codonalign</code><code class="variable-quote">'</code></code>
    </td>
  </tr>
</table>
<!-- ==================== FUNCTION DETAILS ==================== -->
<a name="section-FunctionDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Function Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-FunctionDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="mktest"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">mktest</span>(<span class="sig-arg">codon_alns</span>,
        <span class="sig-arg">codon_table</span>=<span class="sig-default">default_codon_table</span>,
        <span class="sig-arg">alpha</span>=<span class="sig-default">0.05</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.codonalign.codonalignment-pysrc.html#mktest">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>McDonald-Kreitman test for neutrality (PMID: 1904993) This method
counts changes rather than sites (<a class="rst-rst-reference external rst-reference external" href="http://mkt.uab.es/mkt/help_mkt.asp" target="_top">http://mkt.uab.es/mkt/help_mkt.asp</a>).
Arguments:</p>
<blockquote>
<ul class="rst-simple">
<li>codon_alns  - list of CodonAlignment to compare (each
CodonAlignment object corresponds to gene
sampled from a species)</li>
</ul>
</blockquote>
<p>Return the p-value of test result</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_get_codon2codon_matrix"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_get_codon2codon_matrix</span>(<span class="sig-arg">codon_table</span>=<span class="sig-default">default_codon_table</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.codonalign.codonalignment-pysrc.html#_get_codon2codon_matrix">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  Function to get codon codon subsitution matrix. Elements
in the matrix are number of synonymous and nonsynonymous
substitutions required for the substitution (PRIVATE).
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_dijkstra"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_dijkstra</span>(<span class="sig-arg">graph</span>,
        <span class="sig-arg">start</span>,
        <span class="sig-arg">end</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.codonalign.codonalignment-pysrc.html#_dijkstra">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Dijkstra's algorithm Python implementation.
Algorithm adapted from
<a class="rst-reference external" href="http://thomas.pelletier.im/2010/02/dijkstras-algorithm-python-implementation/" target="_top">http://thomas.pelletier.im/2010/02/dijkstras-algorithm-python-implementation/</a>.
However, an abvious bug in:</p>
<pre class="rst-literal-block">
if D[child_node] &gt;(&lt;) D[node] + child_value:
</pre>
<p>is fixed.
This function will return the distance between start and end.</p>
<p>Arguments:</p>
<blockquote>
<ul class="rst-simple">
<li>graph: Dictionnary of dictionnary (keys are vertices).</li>
<li>start: Start vertex.</li>
<li>end: End vertex.</li>
</ul>
</blockquote>
<dl class="rst-docutils">
<dt>Output:</dt>
<dd>List of vertices from the beggining to the end.</dd>
</dl>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_prim"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_prim</span>(<span class="sig-arg">G</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.codonalign.codonalignment-pysrc.html#_prim">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  Prim's algorithm to find minimum spanning tree. Code is adapted from
<a class="rst-reference external" href="http://programmingpraxis.com/2010/04/09/minimum-spanning-tree-prims-algorithm/" target="_top">http://programmingpraxis.com/2010/04/09/minimum-spanning-tree-prims-algorithm/</a>
(PRIVATE).
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_G_test"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_G_test</span>(<span class="sig-arg">site_counts</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.codonalign.codonalignment-pysrc.html#_G_test">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>G test for 2x2 contingency table (PRIVATE).
Argument:</p>
<blockquote>
<ul class="rst-simple">
<li>site_counts - [syn_fix, nonsyn_fix, syn_poly, nonsyn_poly]</li>
</ul>
</blockquote>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>round(_G_test([17, 7, 42, 2]), 7)
<span class="py-output">0.004924</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Oct 21 18:01:30 2015
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
