<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.Graphics.GenomeDiagram._Colors</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.Graphics-module.html">Package&nbsp;Graphics</a> ::
        <a href="Bio.Graphics.GenomeDiagram-module.html">Package&nbsp;GenomeDiagram</a> ::
        Module&nbsp;_Colors
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.Graphics.GenomeDiagram._Colors-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.Graphics.GenomeDiagram._Colors-module.html" onclick="show_private();">Module Bio.Graphics.GenomeDiagram._Colors</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2003-2008 by Leighton Pritchard.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># This file is part of the Biopython distribution and governed by your</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># choice of the "Biopython License Agreement" or the "BSD 3-Clause License".</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-comment"># Please see the LICENSE file that should have been included as part of this</tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-comment"># package.</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-comment"># Contact:       Leighton Pritchard, The James Hutton Institute,</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-comment">#                Invergowrie, Dundee, Scotland, DD2 5DA, UK</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-comment">#                Leighton.Pritchard@hutton.ac.uk</tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-comment">################################################################################</tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-docstring">"""Colors module.</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-docstring">Provides:</tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-docstring">- ColorTranslator - class to convert tuples of integers and floats into</tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-docstring">  colors.Color objects</tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-docstring">For drawing capabilities, this module uses reportlab to define colors:</tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-docstring">http://www.reportlab.com</tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-comment"># ReportLab imports</tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">__future__</tt> <tt class="py-keyword">import</tt> <tt class="py-name">print_function</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-1', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">basestring</tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">reportlab</tt><tt class="py-op">.</tt><tt class="py-name">lib</tt> <tt class="py-keyword">import</tt> <tt class="py-name">colors</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"> </tt>
<a name="ColorTranslator"></a><div id="ColorTranslator-def"><a name="L31"></a><tt class="py-lineno"> 31</tt> <a class="py-toggle" href="#" id="ColorTranslator-toggle" onclick="return toggle('ColorTranslator');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Colors.ColorTranslator-class.html">ColorTranslator</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ColorTranslator-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="ColorTranslator-expanded"><a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line">    <tt class="py-docstring">"""Class providing methods for translating representations of color into.</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-docstring">    Examples</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-docstring">    --------</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio.Graphics import GenomeDiagram</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; gdct=GenomeDiagram._Colors.ColorTranslator()</tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(gdct.float1_color((0.5, 0.5, 0.5)))</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line"><tt class="py-docstring">    Color(.5,.5,.5,1)</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(gdct.int255_color((1, 75, 240)))</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"><tt class="py-docstring">    Color(.003922,.294118,.941176,1)</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(gdct.artemis_color(7))</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line"><tt class="py-docstring">    Color(1,1,0,1)</tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(gdct.scheme_color(2))</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line"><tt class="py-docstring">    Color(1,0,0,1)</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; gdct.get_artemis_colorscheme()</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line"><tt class="py-docstring">    {0: (Color(1,1,1,1), 'pathogenicity, adaptation, chaperones'), 1: (Color(.39,.39,.39,1), 'energy metabolism'), 2: (Color(1,0,0,1), 'information transfer'), 3: (Color(0,1,0,1), 'surface'), 4: (Color(0,0,1,1), 'stable RNA'), 5: (Color(0,1,1,1), 'degradation of large molecules'), 6: (Color(1,0,1,1), 'degradation of small molecules'), 7: (Color(1,1,0,1), 'central/intermediary/miscellaneous metabolism'), 8: (Color(.6,.98,.6,1), 'unknown'), 9: (Color(.53,.81,.98,1), 'regulators'), 10: (Color(1,.65,0,1), 'conserved hypotheticals'), 11: (Color(.78,.59,.39,1), 'pseudogenes and partial genes'), 12: (Color(1,.78,.78,1), 'phage/IS elements'), 13: (Color(.7,.7,.7,1), 'some miscellaneous information'), 14: (Color(0,0,0,1), ''), 15: (Color(1,.25,.25,1), 'secondary metabolism'), 16: (Color(1,.5,.5,1), ''), 17: (Color(1,.75,.75,1), '')}</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(gdct.translate((0.5, 0.5, 0.5)))</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line"><tt class="py-docstring">    Color(.5,.5,.5,1)</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(gdct.translate((1, 75, 240)))</tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line"><tt class="py-docstring">    Color(.003922,.294118,.941176,1)</tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(gdct.translate(7))</tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line"><tt class="py-docstring">    Color(1,1,0,1)</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(gdct.translate(2))</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line"><tt class="py-docstring">    Color(1,0,0,1)</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line"> </tt>
<a name="ColorTranslator.__init__"></a><div id="ColorTranslator.__init__-def"><a name="L60"></a><tt class="py-lineno"> 60</tt> <a class="py-toggle" href="#" id="ColorTranslator.__init__-toggle" onclick="return toggle('ColorTranslator.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Colors.ColorTranslator-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">filename</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ColorTranslator.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ColorTranslator.__init__-expanded"><a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize.</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line"><tt class="py-docstring">        Argument filename is the location of a file containing</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line"><tt class="py-docstring">        colorscheme information.</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_artemis_colorscheme</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line">            <tt class="py-number">0</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-name">colors</tt><tt class="py-op">.</tt><tt class="py-name">Color</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">"pathogenicity, adaptation, chaperones"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line">            <tt class="py-number">1</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-name">colors</tt><tt class="py-op">.</tt><tt class="py-name">Color</tt><tt class="py-op">(</tt><tt class="py-number">0.39</tt><tt class="py-op">,</tt> <tt class="py-number">0.39</tt><tt class="py-op">,</tt> <tt class="py-number">0.39</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">"energy metabolism"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line">            <tt class="py-number">2</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-name">colors</tt><tt class="py-op">.</tt><tt class="py-name">Color</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">"information transfer"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line">            <tt class="py-number">3</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-name">colors</tt><tt class="py-op">.</tt><tt class="py-name">Color</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">"surface"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line">            <tt class="py-number">4</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-name">colors</tt><tt class="py-op">.</tt><tt class="py-name">Color</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">"stable RNA"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line">            <tt class="py-number">5</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-name">colors</tt><tt class="py-op">.</tt><tt class="py-name">Color</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">"degradation of large molecules"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line">            <tt class="py-number">6</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-name">colors</tt><tt class="py-op">.</tt><tt class="py-name">Color</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">"degradation of small molecules"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line">            <tt class="py-number">7</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-name">colors</tt><tt class="py-op">.</tt><tt class="py-name">Color</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">"central/intermediary/miscellaneous metabolism"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line">            <tt class="py-number">8</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-name">colors</tt><tt class="py-op">.</tt><tt class="py-name">Color</tt><tt class="py-op">(</tt><tt class="py-number">0.60</tt><tt class="py-op">,</tt> <tt class="py-number">0.98</tt><tt class="py-op">,</tt> <tt class="py-number">0.60</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">"unknown"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line">            <tt class="py-number">9</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-name">colors</tt><tt class="py-op">.</tt><tt class="py-name">Color</tt><tt class="py-op">(</tt><tt class="py-number">0.53</tt><tt class="py-op">,</tt> <tt class="py-number">0.81</tt><tt class="py-op">,</tt> <tt class="py-number">0.98</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">"regulators"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line">            <tt class="py-number">10</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-name">colors</tt><tt class="py-op">.</tt><tt class="py-name">Color</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">0.65</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">"conserved hypotheticals"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line">            <tt class="py-number">11</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-name">colors</tt><tt class="py-op">.</tt><tt class="py-name">Color</tt><tt class="py-op">(</tt><tt class="py-number">0.78</tt><tt class="py-op">,</tt> <tt class="py-number">0.59</tt><tt class="py-op">,</tt> <tt class="py-number">0.39</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">"pseudogenes and partial genes"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line">            <tt class="py-number">12</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-name">colors</tt><tt class="py-op">.</tt><tt class="py-name">Color</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">0.78</tt><tt class="py-op">,</tt> <tt class="py-number">0.78</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">"phage/IS elements"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line">            <tt class="py-number">13</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-name">colors</tt><tt class="py-op">.</tt><tt class="py-name">Color</tt><tt class="py-op">(</tt><tt class="py-number">0.70</tt><tt class="py-op">,</tt> <tt class="py-number">0.70</tt><tt class="py-op">,</tt> <tt class="py-number">0.70</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">"some miscellaneous information"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line">            <tt class="py-number">14</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-name">colors</tt><tt class="py-op">.</tt><tt class="py-name">Color</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line">            <tt class="py-number">15</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-name">colors</tt><tt class="py-op">.</tt><tt class="py-name">Color</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">0.25</tt><tt class="py-op">,</tt> <tt class="py-number">0.25</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">"secondary metabolism"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line">            <tt class="py-number">16</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-name">colors</tt><tt class="py-op">.</tt><tt class="py-name">Color</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">0.5</tt><tt class="py-op">,</tt> <tt class="py-number">0.5</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line">            <tt class="py-number">17</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-name">colors</tt><tt class="py-op">.</tt><tt class="py-name">Color</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">0.75</tt><tt class="py-op">,</tt> <tt class="py-number">0.75</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line">        <tt class="py-op">}</tt>      <tt class="py-comment"># Hardwired Artemis color scheme</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_colorscheme</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">filename</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-2" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.read_colorscheme()=Bio.Graphics.GenomeDiagram._Colors.ColorTranslator-class.html#read_colorscheme"><a title="Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.read_colorscheme" class="py-name" href="#" onclick="return doclink('link-2', 'read_colorscheme', 'link-2');">read_colorscheme</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">)</tt>  <tt class="py-comment"># Imported color scheme</tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_colorscheme</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_artemis_colorscheme</tt> </tt>
</div><a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line"> </tt>
<a name="ColorTranslator.translate"></a><div id="ColorTranslator.translate-def"><a name="L92"></a><tt class="py-lineno"> 92</tt> <a class="py-toggle" href="#" id="ColorTranslator.translate-toggle" onclick="return toggle('ColorTranslator.translate');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Colors.ColorTranslator-class.html#translate">translate</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">color</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">colour</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ColorTranslator.translate-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ColorTranslator.translate-expanded"><a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line">        <tt class="py-docstring">"""Translate a color into a ReportLab Color object.</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line"><tt class="py-docstring">        - color - Color defined as an int, a tuple of three ints 0-&gt;255</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line"><tt class="py-docstring">          or a tuple of three floats 0 -&gt; 1, or a string giving</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line"><tt class="py-docstring">          one of the named colors defined by ReportLab, or a</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line"><tt class="py-docstring">          ReportLab color object (returned as is).</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line"><tt class="py-docstring">        - colour - Backards compatible alias using UK spelling (which</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line"><tt class="py-docstring">          will over-ride any color argument).</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line"><tt class="py-docstring">        Returns a colors.Color object, determined semi-intelligently</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line"><tt class="py-docstring">        depending on the input values</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line">        <tt class="py-comment"># Let the UK spelling (colour) override the USA spelling (color)</tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">colour</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line">            <tt id="link-3" class="py-name" targets="Variable Bio.Phylo.BaseTree.Clade.color=Bio.Phylo.BaseTree.Clade-class.html#color,Method Bio.Phylo.PhyloXMLIO.Parser.color()=Bio.Phylo.PhyloXMLIO.Parser-class.html#color,Method Bio.Phylo.PhyloXMLIO.Writer.color()=Bio.Phylo.PhyloXMLIO.Writer-class.html#color"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-3', 'color', 'link-3');">color</a></tt> <tt class="py-op">=</tt> <tt class="py-name">colour</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line"> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-4" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-4', 'color', 'link-3');">color</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Passed color (or colour) must be a valid color type"</tt><tt class="py-op">)</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-5" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-5', 'color', 'link-3');">color</a></tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">            <tt id="link-6" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-6', 'color', 'link-3');">color</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-7" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.scheme_color()=Bio.Graphics.GenomeDiagram._Colors.ColorTranslator-class.html#scheme_color"><a title="Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.scheme_color" class="py-name" href="#" onclick="return doclink('link-7', 'scheme_color', 'link-7');">scheme_color</a></tt><tt class="py-op">(</tt><tt id="link-8" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-8', 'color', 'link-3');">color</a></tt><tt class="py-op">)</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-9" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-9', 'color', 'link-3');">color</a></tt><tt class="py-op">,</tt> <tt class="py-name">colors</tt><tt class="py-op">.</tt><tt class="py-name">Color</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-10" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-10', 'color', 'link-3');">color</a></tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-11" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-11', 'color', 'link-3');">color</a></tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line">            <tt class="py-comment"># Assume its a named reportlab color like "red".</tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line">            <tt id="link-12" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-12', 'color', 'link-3');">color</a></tt> <tt class="py-op">=</tt> <tt class="py-name">colors</tt><tt class="py-op">.</tt><tt class="py-name">toColor</tt><tt class="py-op">(</tt><tt id="link-13" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-13', 'color', 'link-3');">color</a></tt><tt class="py-op">)</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-14" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-14', 'color', 'link-3');">color</a></tt><tt class="py-op">,</tt> <tt class="py-name">tuple</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-15" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-15', 'color', 'link-3');">color</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">float</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">            <tt id="link-16" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-16', 'color', 'link-3');">color</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-17" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.float1_color()=Bio.Graphics.GenomeDiagram._Colors.ColorTranslator-class.html#float1_color"><a title="Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.float1_color" class="py-name" href="#" onclick="return doclink('link-17', 'float1_color', 'link-17');">float1_color</a></tt><tt class="py-op">(</tt><tt id="link-18" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-18', 'color', 'link-3');">color</a></tt><tt class="py-op">)</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-19" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-19', 'color', 'link-3');">color</a></tt><tt class="py-op">,</tt> <tt class="py-name">tuple</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-20" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-20', 'color', 'link-3');">color</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line">            <tt id="link-21" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-21', 'color', 'link-3');">color</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-22" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.int255_color()=Bio.Graphics.GenomeDiagram._Colors.ColorTranslator-class.html#int255_color"><a title="Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.int255_color" class="py-name" href="#" onclick="return doclink('link-22', 'int255_color', 'link-22');">int255_color</a></tt><tt class="py-op">(</tt><tt id="link-23" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-23', 'color', 'link-3');">color</a></tt><tt class="py-op">)</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-24" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-24', 'color', 'link-3');">color</a></tt> </tt>
</div><a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line"> </tt>
<a name="ColorTranslator.read_colorscheme"></a><div id="ColorTranslator.read_colorscheme-def"><a name="L126"></a><tt class="py-lineno">126</tt> <a class="py-toggle" href="#" id="ColorTranslator.read_colorscheme-toggle" onclick="return toggle('ColorTranslator.read_colorscheme');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Colors.ColorTranslator-class.html#read_colorscheme">read_colorscheme</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">filename</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ColorTranslator.read_colorscheme-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ColorTranslator.read_colorscheme-expanded"><a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line">        <tt class="py-docstring">r"""Load colour scheme from file.</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line"><tt class="py-docstring">        Reads information from a file containing color information and stores</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line"><tt class="py-docstring">        it internally.</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line"><tt class="py-docstring">        Argument filename is the location of a file defining colors in</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line"><tt class="py-docstring">        tab-separated format plaintext as::</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line"><tt class="py-docstring">            INT \t RED \t GREEN \t BLUE \t Comment</tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line"><tt class="py-docstring">        Where RED, GREEN and BLUE are intensities in the range 0 -&gt; 255, e.g.::</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line"><tt class="py-docstring">            2 \t 255 \t 0 \t 0 \t Red: Information transfer</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line">        <tt class="py-keyword">with</tt> <tt id="link-25" class="py-name" targets="Function Bio.bgzf.open()=Bio.bgzf-module.html#open"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-25', 'open', 'link-25');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">filename</tt><tt class="py-op">,</tt> <tt class="py-string">'r'</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-26" class="py-name" targets="Method Bio.File.UndoHandle.readlines()=Bio.File.UndoHandle-class.html#readlines"><a title="Bio.File.UndoHandle.readlines" class="py-name" href="#" onclick="return doclink('link-26', 'readlines', 'link-26');">readlines</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">lines</tt><tt class="py-op">:</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">lines</tt><tt class="py-op">:</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line">                <tt id="link-27" class="py-name" targets="Variable BioSQL.BioSeq.DBSeq.data=BioSQL.BioSeq.DBSeq-class.html#data"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-27', 'data', 'link-27');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-28" class="py-name" targets="Method Bio.Seq.Seq.strip()=Bio.Seq.Seq-class.html#strip"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-28', 'strip', 'link-28');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-29" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-29', 'split', 'link-29');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">'\t'</tt><tt class="py-op">)</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line">                <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line">                    <tt class="py-name">label</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-30" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-30', 'data', 'link-27');">data</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line">                    <tt id="link-31" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.red()=Bio.Phylo.PhyloXMLIO.Writer-class.html#red"><a title="Bio.Phylo.PhyloXMLIO.Writer.red" class="py-name" href="#" onclick="return doclink('link-31', 'red', 'link-31');">red</a></tt><tt class="py-op">,</tt> <tt id="link-32" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.green()=Bio.Phylo.PhyloXMLIO.Writer-class.html#green"><a title="Bio.Phylo.PhyloXMLIO.Writer.green" class="py-name" href="#" onclick="return doclink('link-32', 'green', 'link-32');">green</a></tt><tt class="py-op">,</tt> <tt id="link-33" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.blue()=Bio.Phylo.PhyloXMLIO.Writer-class.html#blue"><a title="Bio.Phylo.PhyloXMLIO.Writer.blue" class="py-name" href="#" onclick="return doclink('link-33', 'blue', 'link-33');">blue</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-34" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-34', 'data', 'link-27');">data</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-35" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-35', 'data', 'link-27');">data</a></tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-36" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-36', 'data', 'link-27');">data</a></tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-37" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-37', 'data', 'link-27');">data</a></tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">4</tt><tt class="py-op">:</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line">                        <tt id="link-38" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.comment()=Bio.GenBank._FeatureConsumer-class.html#comment,Method Bio.GenBank._RecordConsumer.comment()=Bio.GenBank._RecordConsumer-class.html#comment"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-38', 'comment', 'link-38');">comment</a></tt> <tt class="py-op">=</tt> <tt id="link-39" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-39', 'data', 'link-27');">data</a></tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">                        <tt id="link-40" class="py-name"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-40', 'comment', 'link-38');">comment</a></tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_colorscheme</tt><tt class="py-op">[</tt><tt class="py-name">label</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-41" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.int255_color" class="py-name" href="#" onclick="return doclink('link-41', 'int255_color', 'link-22');">int255_color</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt id="link-42" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.red" class="py-name" href="#" onclick="return doclink('link-42', 'red', 'link-31');">red</a></tt><tt class="py-op">,</tt> <tt id="link-43" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.green" class="py-name" href="#" onclick="return doclink('link-43', 'green', 'link-32');">green</a></tt><tt class="py-op">,</tt> <tt id="link-44" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.blue" class="py-name" href="#" onclick="return doclink('link-44', 'blue', 'link-33');">blue</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">                                                <tt id="link-45" class="py-name"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-45', 'comment', 'link-38');">comment</a></tt><tt class="py-op">)</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line">                <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Expected INT \t INT \t INT \t INT \t string input"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line"> </tt>
<a name="ColorTranslator.get_artemis_colorscheme"></a><div id="ColorTranslator.get_artemis_colorscheme-def"><a name="L157"></a><tt class="py-lineno">157</tt> <a class="py-toggle" href="#" id="ColorTranslator.get_artemis_colorscheme-toggle" onclick="return toggle('ColorTranslator.get_artemis_colorscheme');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Colors.ColorTranslator-class.html#get_artemis_colorscheme">get_artemis_colorscheme</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ColorTranslator.get_artemis_colorscheme-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ColorTranslator.get_artemis_colorscheme-expanded"><a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the Artemis color scheme as a dictionary."""</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_artemis_colorscheme</tt> </tt>
</div><a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line"> </tt>
<a name="ColorTranslator.artemis_color"></a><div id="ColorTranslator.artemis_color-def"><a name="L161"></a><tt class="py-lineno">161</tt> <a class="py-toggle" href="#" id="ColorTranslator.artemis_color-toggle" onclick="return toggle('ColorTranslator.artemis_color');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Colors.ColorTranslator-class.html#artemis_color">artemis_color</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ColorTranslator.artemis_color-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ColorTranslator.artemis_color-expanded"><a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">        <tt class="py-docstring">"""Artemis color (integer) to ReportLab Color object.</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line"><tt class="py-docstring">        - value: An int representing a functional class in the Artemis</tt> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line"><tt class="py-docstring">          color scheme (see www.sanger.ac.uk for a description),</tt> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line"><tt class="py-docstring">          or a string from a GenBank feature annotation for the</tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line"><tt class="py-docstring">          color which may be dot delimited (in which case the</tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line"><tt class="py-docstring">          first value is used).</tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line"><tt class="py-docstring">        Takes an int representing a functional class in the Artemis color</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line"><tt class="py-docstring">        scheme, and returns the appropriate colors.Color object</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">            <tt id="link-46" class="py-name" targets="Variable Bio.Entrez.Parser.Consumer.value=Bio.Entrez.Parser.Consumer-class.html#value,Variable Bio.Entrez.Parser.DictionaryConsumer.value=Bio.Entrez.Parser.DictionaryConsumer-class.html#value,Variable Bio.Entrez.Parser.ErrorConsumer.value=Bio.Entrez.Parser.ErrorConsumer-class.html#value,Variable Bio.Entrez.Parser.IntegerConsumer.value=Bio.Entrez.Parser.IntegerConsumer-class.html#value,Variable Bio.Entrez.Parser.ListConsumer.value=Bio.Entrez.Parser.ListConsumer-class.html#value,Variable Bio.Entrez.Parser.StringConsumer.value=Bio.Entrez.Parser.StringConsumer-class.html#value,Method Bio.Phylo.PhyloXMLIO.Writer.value()=Bio.Phylo.PhyloXMLIO.Writer-class.html#value"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-46', 'value', 'link-46');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-47" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-47', 'value', 'link-46');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-48" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-48', 'value', 'link-46');">value</a></tt><tt class="py-op">.</tt><tt id="link-49" class="py-name" targets="Method Bio.Crystal.Chain.count()=Bio.Crystal.Chain-class.html#count,Method Bio.Seq.MutableSeq.count()=Bio.Seq.MutableSeq-class.html#count,Method Bio.Seq.Seq.count()=Bio.Seq.Seq-class.html#count,Method Bio.Seq.UnknownSeq.count()=Bio.Seq.UnknownSeq-class.html#count,Method Bio.motifs.Instances.count()=Bio.motifs.Instances-class.html#count"><a title="Bio.Crystal.Chain.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-49', 'count', 'link-49');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">'.'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># dot-delimited</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line">                <tt id="link-50" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-50', 'value', 'link-46');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-51" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-51', 'value', 'link-46');">value</a></tt><tt class="py-op">.</tt><tt id="link-52" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-52', 'split', 'link-29');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">'.'</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt>  <tt class="py-comment"># Use only first integer</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-53" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-53', 'value', 'link-46');">value</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_artemis_colorscheme</tt><tt class="py-op">:</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_artemis_colorscheme</tt><tt class="py-op">[</tt><tt id="link-54" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-54', 'value', 'link-46');">value</a></tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Artemis color out of range: %d"</tt> <tt class="py-op">%</tt> <tt id="link-55" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-55', 'value', 'link-46');">value</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line"> </tt>
<a name="ColorTranslator.get_colorscheme"></a><div id="ColorTranslator.get_colorscheme-def"><a name="L187"></a><tt class="py-lineno">187</tt> <a class="py-toggle" href="#" id="ColorTranslator.get_colorscheme-toggle" onclick="return toggle('ColorTranslator.get_colorscheme');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Colors.ColorTranslator-class.html#get_colorscheme">get_colorscheme</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ColorTranslator.get_colorscheme-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ColorTranslator.get_colorscheme-expanded"><a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the user-defined color scheme as a dictionary."""</tt> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_colorscheme</tt> </tt>
</div><a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line"> </tt>
<a name="ColorTranslator.scheme_color"></a><div id="ColorTranslator.scheme_color-def"><a name="L191"></a><tt class="py-lineno">191</tt> <a class="py-toggle" href="#" id="ColorTranslator.scheme_color-toggle" onclick="return toggle('ColorTranslator.scheme_color');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Colors.ColorTranslator-class.html#scheme_color">scheme_color</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ColorTranslator.scheme_color-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ColorTranslator.scheme_color-expanded"><a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">        <tt class="py-docstring">"""Map a user-defined color integer to a ReportLab Color object.</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line"><tt class="py-docstring">        - value: An int representing a single color in the user-defined</tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line"><tt class="py-docstring">          color scheme</tt> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line"><tt class="py-docstring">        Takes an int representing a user-defined color and returns the</tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line"><tt class="py-docstring">        appropriate colors.Color object.</tt> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-56" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-56', 'value', 'link-46');">value</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_colorscheme</tt><tt class="py-op">:</tt> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_colorscheme</tt><tt class="py-op">[</tt><tt id="link-57" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-57', 'value', 'link-46');">value</a></tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Scheme color out of range: %d"</tt> <tt class="py-op">%</tt> <tt id="link-58" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-58', 'value', 'link-46');">value</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line"> </tt>
<a name="ColorTranslator.int255_color"></a><div id="ColorTranslator.int255_color-def"><a name="L205"></a><tt class="py-lineno">205</tt> <a class="py-toggle" href="#" id="ColorTranslator.int255_color-toggle" onclick="return toggle('ColorTranslator.int255_color');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Colors.ColorTranslator-class.html#int255_color">int255_color</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">values</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ColorTranslator.int255_color-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ColorTranslator.int255_color-expanded"><a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line">        <tt class="py-docstring">"""Map integer (red, green, blue) tuple to a ReportLab Color object.</tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line"><tt class="py-docstring">        - values: A tuple of (red, green, blue) intensities as</tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line"><tt class="py-docstring">          integers in the range 0-&gt;255</tt> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line"><tt class="py-docstring">        Takes a tuple of (red, green, blue) intensity values in the range</tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line"><tt class="py-docstring">        0 -&gt; 255 and returns an appropriate colors.Color object.</tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line">        <tt id="link-59" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.red" class="py-name" href="#" onclick="return doclink('link-59', 'red', 'link-31');">red</a></tt><tt class="py-op">,</tt> <tt id="link-60" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.green" class="py-name" href="#" onclick="return doclink('link-60', 'green', 'link-32');">green</a></tt><tt class="py-op">,</tt> <tt id="link-61" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.blue" class="py-name" href="#" onclick="return doclink('link-61', 'blue', 'link-33');">blue</a></tt> <tt class="py-op">=</tt> <tt id="link-62" class="py-name" targets="Method Bio.Crystal.Crystal.values()=Bio.Crystal.Crystal-class.html#values,Method Bio.File._IndexedSeqFileDict.values()=Bio.File._IndexedSeqFileDict-class.html#values,Method Bio.Phylo.PhyloXML.Events.values()=Bio.Phylo.PhyloXML.Events-class.html#values,Method BioSQL.BioSeqDatabase.BioSeqDatabase.values()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#values,Method BioSQL.BioSeqDatabase.DBServer.values()=BioSQL.BioSeqDatabase.DBServer-class.html#values"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-62', 'values', 'link-62');">values</a></tt> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line">        <tt class="py-name">factor</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> <tt class="py-op">/</tt> <tt class="py-number">255.</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line">        <tt id="link-63" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.red" class="py-name" href="#" onclick="return doclink('link-63', 'red', 'link-31');">red</a></tt><tt class="py-op">,</tt> <tt id="link-64" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.green" class="py-name" href="#" onclick="return doclink('link-64', 'green', 'link-32');">green</a></tt><tt class="py-op">,</tt> <tt id="link-65" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.blue" class="py-name" href="#" onclick="return doclink('link-65', 'blue', 'link-33');">blue</a></tt> <tt class="py-op">=</tt> <tt id="link-66" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.red" class="py-name" href="#" onclick="return doclink('link-66', 'red', 'link-31');">red</a></tt> <tt class="py-op">*</tt> <tt class="py-name">factor</tt><tt class="py-op">,</tt> <tt id="link-67" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.green" class="py-name" href="#" onclick="return doclink('link-67', 'green', 'link-32');">green</a></tt> <tt class="py-op">*</tt> <tt class="py-name">factor</tt><tt class="py-op">,</tt> <tt id="link-68" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.blue" class="py-name" href="#" onclick="return doclink('link-68', 'blue', 'link-33');">blue</a></tt> <tt class="py-op">*</tt> <tt class="py-name">factor</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">colors</tt><tt class="py-op">.</tt><tt class="py-name">Color</tt><tt class="py-op">(</tt><tt id="link-69" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.red" class="py-name" href="#" onclick="return doclink('link-69', 'red', 'link-31');">red</a></tt><tt class="py-op">,</tt> <tt id="link-70" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.green" class="py-name" href="#" onclick="return doclink('link-70', 'green', 'link-32');">green</a></tt><tt class="py-op">,</tt> <tt id="link-71" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.blue" class="py-name" href="#" onclick="return doclink('link-71', 'blue', 'link-33');">blue</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line"> </tt>
<a name="ColorTranslator.float1_color"></a><div id="ColorTranslator.float1_color-def"><a name="L219"></a><tt class="py-lineno">219</tt> <a class="py-toggle" href="#" id="ColorTranslator.float1_color-toggle" onclick="return toggle('ColorTranslator.float1_color');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._Colors.ColorTranslator-class.html#float1_color">float1_color</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">values</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ColorTranslator.float1_color-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="ColorTranslator.float1_color-expanded"><a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line">        <tt class="py-docstring">"""Map float (red, green, blue) tuple to a ReportLab Color object.</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line"><tt class="py-docstring">        - values: A tuple of (red, green, blue) intensities as floats</tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line"><tt class="py-docstring">          in the range 0 -&gt; 1</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line"><tt class="py-docstring">        Takes a tuple of (red, green, blue) intensity values in the range</tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line"><tt class="py-docstring">        0 -&gt; 1 and returns an appropriate colors.Color object.</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line">        <tt id="link-72" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.red" class="py-name" href="#" onclick="return doclink('link-72', 'red', 'link-31');">red</a></tt><tt class="py-op">,</tt> <tt id="link-73" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.green" class="py-name" href="#" onclick="return doclink('link-73', 'green', 'link-32');">green</a></tt><tt class="py-op">,</tt> <tt id="link-74" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.blue" class="py-name" href="#" onclick="return doclink('link-74', 'blue', 'link-33');">blue</a></tt> <tt class="py-op">=</tt> <tt id="link-75" class="py-name"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-75', 'values', 'link-62');">values</a></tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">colors</tt><tt class="py-op">.</tt><tt class="py-name">Color</tt><tt class="py-op">(</tt><tt id="link-76" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.red" class="py-name" href="#" onclick="return doclink('link-76', 'red', 'link-31');">red</a></tt><tt class="py-op">,</tt> <tt id="link-77" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.green" class="py-name" href="#" onclick="return doclink('link-77', 'green', 'link-32');">green</a></tt><tt class="py-op">,</tt> <tt id="link-78" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.blue" class="py-name" href="#" onclick="return doclink('link-78', 'blue', 'link-33');">blue</a></tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line"> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line"> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line"><tt class="py-keyword">if</tt> <tt class="py-name">__name__</tt> <tt class="py-op">==</tt> <tt class="py-string">'__main__'</tt><tt class="py-op">:</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-79" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-79', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-80" class="py-name" targets="Module Bio.Phylo._utils=Bio.Phylo._utils-module.html,Module Bio.SearchIO._utils=Bio.SearchIO._utils-module.html,Module Bio._utils=Bio._utils-module.html"><a title="Bio.Phylo._utils
Bio.SearchIO._utils
Bio._utils" class="py-name" href="#" onclick="return doclink('link-80', '_utils', 'link-80');">_utils</a></tt> <tt class="py-keyword">import</tt> <tt id="link-81" class="py-name" targets="Function Bio._utils.run_doctest()=Bio._utils-module.html#run_doctest"><a title="Bio._utils.run_doctest" class="py-name" href="#" onclick="return doclink('link-81', 'run_doctest', 'link-81');">run_doctest</a></tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line">    <tt id="link-82" class="py-name"><a title="Bio._utils.run_doctest" class="py-name" href="#" onclick="return doclink('link-82', 'run_doctest', 'link-81');">run_doctest</a></tt><tt class="py-op">(</tt><tt class="py-name">verbose</tt><tt class="py-op">=</tt><tt class="py-number">2</tt><tt class="py-op">)</tt> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Tue Dec 18 15:58:19 2018
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
