<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.TogoWS</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        Package&nbsp;TogoWS
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.TogoWS-module.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== PACKAGE DESCRIPTION ==================== -->
<h1 class="epydoc">Package TogoWS</h1><p class="nomargin-top"><span class="codelink"><a href="Bio.TogoWS-pysrc.html">source&nbsp;code</a></span></p>
<p>Provides code to access the TogoWS integrated websevices of DBCLS, Japan.</p>
<p>This module aims to make the TogoWS (from DBCLS, Japan) easier to use. See:
<a class="rst-reference external" href="http://togows.dbcls.jp/" target="_top">http://togows.dbcls.jp/</a></p>
<p>The TogoWS REST service provides simple access to a range of databases, acting
as a proxy to shield you from all the different provider APIs. This works using
simple URLs (which this module will construct for you). For more details, see
<a class="rst-reference external" href="http://togows.dbcls.jp/site/en/rest.html" target="_top">http://togows.dbcls.jp/site/en/rest.html</a></p>
<p>The functionality is somewhat similar to Biopython's Bio.Entrez module which
provides access to the NCBI's Entrez Utilities (E-Utils) which also covers a
wide range of databases.</p>
<p>Currently TogoWS does not provide any usage guidelines (unlike the NCBI whose
requirements are reasonably clear). To avoid risking overloading the service,
Biopython will only allow three calls per second.</p>
<p>The TogoWS SOAP service offers a more complex API for calling web services
(essentially calling remote functions) provided by DDBJ, KEGG and PDBj. For
example, this allows you to run a remote BLAST search at the DDBJ. This is
not yet covered by this module, however there are lots of Python examples
on the TogoWS website using the SOAPpy python library. See:
<a class="rst-reference external" href="http://togows.dbcls.jp/site/en/soap.html" target="_top">http://togows.dbcls.jp/site/en/soap.html</a>
<a class="rst-reference external" href="http://soapy.sourceforge.net/" target="_top">http://soapy.sourceforge.net/</a></p>

<!-- ==================== FUNCTIONS ==================== -->
<a name="section-Functions"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Functions</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Functions"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_get_fields"></a><span class="summary-sig-name">_get_fields</span>(<span class="summary-sig-arg">url</span>)</span><br />
      Query a TogoWS URL for a plain text list of values (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.TogoWS-pysrc.html#_get_fields">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_get_entry_dbs"></a><span class="summary-sig-name">_get_entry_dbs</span>()</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.TogoWS-pysrc.html#_get_entry_dbs">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_get_entry_fields"></a><span class="summary-sig-name">_get_entry_fields</span>(<span class="summary-sig-arg">db</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.TogoWS-pysrc.html#_get_entry_fields">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_get_entry_formats"></a><span class="summary-sig-name">_get_entry_formats</span>(<span class="summary-sig-arg">db</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.TogoWS-pysrc.html#_get_entry_formats">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_get_convert_formats"></a><span class="summary-sig-name">_get_convert_formats</span>()</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.TogoWS-pysrc.html#_get_convert_formats">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.TogoWS-module.html#entry" class="summary-sig-name">entry</a>(<span class="summary-sig-arg">db</span>,
        <span class="summary-sig-arg">id</span>,
        <span class="summary-sig-arg">format</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">field</span>=<span class="summary-sig-default">None</span>)</span><br />
      Call TogoWS 'entry' to fetch a record.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.TogoWS-pysrc.html#entry">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.TogoWS-module.html#search_count" class="summary-sig-name">search_count</a>(<span class="summary-sig-arg">db</span>,
        <span class="summary-sig-arg">query</span>)</span><br />
      Call TogoWS search count to see how many matches a search gives.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.TogoWS-pysrc.html#search_count">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.TogoWS-module.html#search_iter" class="summary-sig-name">search_iter</a>(<span class="summary-sig-arg">db</span>,
        <span class="summary-sig-arg">query</span>,
        <span class="summary-sig-arg">limit</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">batch</span>=<span class="summary-sig-default">100</span>)</span><br />
      Call TogoWS search iteratating over the results (generator function).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.TogoWS-pysrc.html#search_iter">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.TogoWS-module.html#search" class="summary-sig-name">search</a>(<span class="summary-sig-arg">db</span>,
        <span class="summary-sig-arg">query</span>,
        <span class="summary-sig-arg">offset</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">limit</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">format</span>=<span class="summary-sig-default">None</span>)</span><br />
      Call TogoWS search.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.TogoWS-pysrc.html#search">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.TogoWS-module.html#convert" class="summary-sig-name">convert</a>(<span class="summary-sig-arg">data</span>,
        <span class="summary-sig-arg">in_format</span>,
        <span class="summary-sig-arg">out_format</span>)</span><br />
      Call TogoWS for file format conversion.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.TogoWS-pysrc.html#convert">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.TogoWS-module.html#_open" class="summary-sig-name" onclick="show_private();">_open</a>(<span class="summary-sig-arg">url</span>,
        <span class="summary-sig-arg">post</span>=<span class="summary-sig-default">None</span>)</span><br />
      Build the URL and open a handle to it (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.TogoWS-pysrc.html#_open">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
</table>
<!-- ==================== VARIABLES ==================== -->
<a name="section-Variables"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Variables</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Variables"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="_BASE_URL"></a><span class="summary-name">_BASE_URL</span> = <code title="'http://togows.dbcls.jp'"><code class="variable-quote">'</code><code class="variable-string">http://togows.dbcls.jp</code><code class="variable-quote">'</code></code>
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="_search_db_names"></a><span class="summary-name">_search_db_names</span> = <code title="None">None</code><br />
      hash(x)
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="_entry_db_names"></a><span class="summary-name">_entry_db_names</span> = <code title="None">None</code><br />
      hash(x)
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="_entry_db_fields"></a><span class="summary-name">_entry_db_fields</span> = <code title="{}"><code class="variable-group">{</code><code class="variable-group">}</code></code>
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="_entry_db_formats"></a><span class="summary-name">_entry_db_formats</span> = <code title="{}"><code class="variable-group">{</code><code class="variable-group">}</code></code>
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="_convert_formats"></a><span class="summary-name">_convert_formats</span> = <code title="[]"><code class="variable-group">[</code><code class="variable-group">]</code></code>
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="__package__"></a><span class="summary-name">__package__</span> = <code title="'Bio.TogoWS'"><code class="variable-quote">'</code><code class="variable-string">Bio.TogoWS</code><code class="variable-quote">'</code></code>
    </td>
  </tr>
</table>
<!-- ==================== FUNCTION DETAILS ==================== -->
<a name="section-FunctionDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Function Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-FunctionDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="entry"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">entry</span>(<span class="sig-arg">db</span>,
        <span class="sig-arg">id</span>,
        <span class="sig-arg">format</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">field</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.TogoWS-pysrc.html#entry">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Call TogoWS 'entry' to fetch a record.</p>
<dl class="rst-docutils">
<dt>Arguments:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>db - database (string), see list below.</li>
<li>id - identier (string) or a list of identifiers (either as a list of
strings or a single string with comma separators).</li>
<li>format - return data file format (string), options depend on the database
e.g. &quot;xml&quot;, &quot;json&quot;, &quot;gff&quot;, &quot;fasta&quot;, &quot;ttl&quot; (RDF Turtle)</li>
<li>field - specific field from within the database record (string)
e.g. &quot;au&quot; or &quot;authors&quot; for pubmed.</li>
</ul>
</dd>
</dl>
<p>At the time of writing, this includes the following:</p>
<pre class="rst-literal-block">
KEGG: compound, drug, enzyme, genes, glycan, orthology, reaction,
      module, pathway
DDBj: ddbj, dad, pdb
NCBI: nuccore, nucest, nucgss, nucleotide, protein, gene, onim,
      homologue, snp, mesh, pubmed
EBI:  embl, uniprot, uniparc, uniref100, uniref90, uniref50
</pre>
<p>For the current list, please see <a class="rst-reference external" href="http://togows.dbcls.jp/entry/" target="_top">http://togows.dbcls.jp/entry/</a></p>
<p>This function is essentially equivalent to the NCBI Entrez service
EFetch, available in Biopython as Bio.Entrez.efetch(...), but that
does not offer field extraction.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="search_count"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">search_count</span>(<span class="sig-arg">db</span>,
        <span class="sig-arg">query</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.TogoWS-pysrc.html#search_count">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Call TogoWS search count to see how many matches a search gives.</p>
<dl class="rst-docutils">
<dt>Arguments:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>db - database (string), see <a class="rst-reference external" href="http://togows.dbcls.jp/search" target="_top">http://togows.dbcls.jp/search</a></li>
<li>query - search term (string)</li>
</ul>
</dd>
</dl>
<p>You could then use the count to download a large set of search results in
batches using the offset and limit options to Bio.TogoWS.search(). In
general however the Bio.TogoWS.search_iter() function is simpler to use.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="search_iter"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">search_iter</span>(<span class="sig-arg">db</span>,
        <span class="sig-arg">query</span>,
        <span class="sig-arg">limit</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">batch</span>=<span class="sig-default">100</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.TogoWS-pysrc.html#search_iter">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Call TogoWS search iteratating over the results (generator function).</p>
<dl class="rst-docutils">
<dt>Arguments:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>db - database (string), see <a class="rst-reference external" href="http://togows.dbcls.jp/search" target="_top">http://togows.dbcls.jp/search</a></li>
<li>query - search term (string)</li>
<li>limit - optional upper bound on number of search results</li>
<li>batch - number of search results to pull back each time talk to
TogoWS (currently limited to 100).</li>
</ul>
</dd>
</dl>
<p>You would use this function within a for loop, e.g.</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">for</span> id <span class="py-keyword">in</span> search_iter(<span class="py-string">&quot;pubmed&quot;</span>, <span class="py-string">&quot;lung+cancer+drug&quot;</span>, limit=10):
<span class="py-more">... </span>    <span class="py-keyword">print</span>(id) <span class="py-comment"># maybe fetch data with entry?</span></pre>
<p>Internally this first calls the Bio.TogoWS.search_count() and then
uses Bio.TogoWS.search() to get the results in batches.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="search"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">search</span>(<span class="sig-arg">db</span>,
        <span class="sig-arg">query</span>,
        <span class="sig-arg">offset</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">limit</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">format</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.TogoWS-pysrc.html#search">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Call TogoWS search.</p>
<p>This is a low level wrapper for the TogoWS search function, which
can return results in a several formats. In general, the search_iter
function is more suitable for end users.</p>
<dl class="rst-docutils">
<dt>Arguments:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>db - database (string), see <a class="rst-reference external" href="http://togows.dbcls.jp/search/" target="_top">http://togows.dbcls.jp/search/</a></li>
<li>query - search term (string)</li>
<li>offset, limit - optional integers specifying which result to start from
(1 based) and the number of results to return.</li>
<li>format - return data file format (string), e.g. &quot;json&quot;, &quot;ttl&quot; (RDF)
By default plain text is returned, one result per line.</li>
</ul>
</dd>
</dl>
<p>At the time of writing, TogoWS applies a default count limit of 100
search results, and this is an upper bound. To access more results,
use the offset argument or the search_iter(...) function.</p>
<p>TogoWS supports a long list of databases, including many from the NCBI
(e.g. &quot;ncbi-pubmed&quot; or &quot;pubmed&quot;, &quot;ncbi-genbank&quot; or &quot;genbank&quot;, and
&quot;ncbi-taxonomy&quot;), EBI (e.g. &quot;ebi-ebml&quot; or &quot;embl&quot;, &quot;ebi-uniprot&quot; or
&quot;uniprot, &quot;ebi-go&quot;), and KEGG (e.g. &quot;kegg-compound&quot; or &quot;compound&quot;).
For the current list, see <a class="rst-reference external" href="http://togows.dbcls.jp/search/" target="_top">http://togows.dbcls.jp/search/</a></p>
<p>The NCBI provide the Entrez Search service (ESearch) which is similar,
available in Biopython as the Bio.Entrez.esearch() function.</p>
<p>See also the function Bio.TogoWS.search_count() which returns the number
of matches found, and the Bio.TogoWS.search_iter() function which allows
you to iterate over the search results (taking care of batching for you).</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="convert"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">convert</span>(<span class="sig-arg">data</span>,
        <span class="sig-arg">in_format</span>,
        <span class="sig-arg">out_format</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.TogoWS-pysrc.html#convert">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Call TogoWS for file format conversion.</p>
<dl class="rst-docutils">
<dt>Arguments:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>data - string or handle containing input record(s)</li>
<li>in_format - string describing the input file format (e.g. &quot;genbank&quot;)</li>
<li>out_format - string describing the requested output format (e.g. &quot;fasta&quot;)</li>
</ul>
</dd>
</dl>
<p>For a list of supported conversions (e.g. &quot;genbank&quot; to &quot;fasta&quot;), see
<a class="rst-reference external" href="http://togows.dbcls.jp/convert/" target="_top">http://togows.dbcls.jp/convert/</a></p>
<p>Note that Biopython has built in support for conversion of sequence and
alignnent file formats (functions Bio.SeqIO.convert and Bio.AlignIO.convert)</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_open"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_open</span>(<span class="sig-arg">url</span>,
        <span class="sig-arg">post</span>=<span class="sig-default">None</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.TogoWS-pysrc.html#_open">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Build the URL and open a handle to it (PRIVATE).</p>
<p>Open a handle to TogoWS, will raise an IOError if it encounters an error.</p>
<p>In the absence of clear guidelines, this function enforces a limit of
&quot;up to three queries per second&quot; to avoid abusing the TogoWS servers.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Apr  4 10:48:06 2018
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
