<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.Phylo.PhyloXML.Sequence</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.Phylo-module.html">Package&nbsp;Phylo</a> ::
        <a href="Bio.Phylo.PhyloXML-module.html">Module&nbsp;PhyloXML</a> ::
        Class&nbsp;Sequence
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.Phylo.PhyloXML.Sequence-class.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== CLASS DESCRIPTION ==================== -->
<h1 class="epydoc">Class Sequence</h1><p class="nomargin-top"><span class="codelink"><a href="Bio.Phylo.PhyloXML-pysrc.html#Sequence">source&nbsp;code</a></span></p>
<pre class="base-tree">
          object --+        
                   |        
<a href="Bio.Phylo.BaseTree.TreeElement-class.html">BaseTree.TreeElement</a> --+    
                       |    
            <a href="Bio.Phylo.PhyloXML.PhyloElement-class.html">PhyloElement</a> --+
                           |
                          <strong class="uidshort">Sequence</strong>
</pre>

<hr />
<p>A molecular sequence (Protein, DNA, RNA) associated with a node.</p>
<p>One intended use for <tt class="rst-docutils literal">id_ref</tt> is to link a sequence to a taxonomy (via the
taxonomy's <tt class="rst-docutils literal">id_source</tt>) in case of multiple sequences and taxonomies per
node.</p>

<!-- ==================== INSTANCE METHODS ==================== -->
<a name="section-InstanceMethods"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Instance Methods</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-InstanceMethods"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Phylo.PhyloXML.Sequence-class.html#__init__" class="summary-sig-name">__init__</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">type</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">id_ref</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">id_source</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">symbol</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">accession</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">name</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">location</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">mol_seq</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">uri</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">domain_architecture</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">annotations</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">other</span>=<span class="summary-sig-default">None</span>)</span><br />
      Initialize value for a Sequence object.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Phylo.PhyloXML-pysrc.html#Sequence.__init__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Phylo.PhyloXML.Sequence-class.html#to_seqrecord" class="summary-sig-name">to_seqrecord</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Create a SeqRecord object from this Sequence instance.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Phylo.PhyloXML-pysrc.html#Sequence.to_seqrecord">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="get_alphabet"></a><span class="summary-sig-name">get_alphabet</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Get the alphabet for the sequence.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Phylo.PhyloXML-pysrc.html#Sequence.get_alphabet">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code><a href="Bio.Phylo.BaseTree.TreeElement-class.html">BaseTree.TreeElement</a></code></b>:
      <code><a href="Bio.Phylo.BaseTree.TreeElement-class.html#__repr__">__repr__</a></code>,
      <code><a href="Bio.Phylo.BaseTree.TreeElement-class.html#__str__">__str__</a></code>
      </p>
    <p class="indent-wrapped-lines"><b>Inherited from <code>object</code></b>:
      <code>__delattr__</code>,
      <code>__format__</code>,
      <code>__getattribute__</code>,
      <code>__hash__</code>,
      <code>__new__</code>,
      <code>__reduce__</code>,
      <code>__reduce_ex__</code>,
      <code>__setattr__</code>,
      <code>__sizeof__</code>,
      <code>__subclasshook__</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== CLASS METHODS ==================== -->
<a name="section-ClassMethods"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Class Methods</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-ClassMethods"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="from_seqrecord"></a><span class="summary-sig-name">from_seqrecord</span>(<span class="summary-sig-arg">cls</span>,
        <span class="summary-sig-arg">record</span>,
        <span class="summary-sig-arg">is_aligned</span>=<span class="summary-sig-default">None</span>)</span><br />
      Create a new PhyloXML Sequence from a SeqRecord object.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Phylo.PhyloXML-pysrc.html#Sequence.from_seqrecord">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
</table>
<!-- ==================== CLASS VARIABLES ==================== -->
<a name="section-ClassVariables"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Class Variables</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-ClassVariables"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.Phylo.PhyloXML.Sequence-class.html#alphabets" class="summary-name">alphabets</a> = <code title="{'dna': DNAAlphabet(),
 'protein': ProteinAlphabet(),
 'rna': RNAAlphabet()}"><code class="variable-group">{</code><code class="variable-quote">'</code><code class="variable-string">dna</code><code class="variable-quote">'</code><code class="variable-op">: </code>DNAAlphabet()<code class="variable-op">, </code><code class="variable-quote">'</code><code class="variable-string">protein</code><code class="variable-quote">'</code><code class="variable-op">: </code>ProteinAlphabet(<code class="variable-ellipsis">...</code></code>
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="re_symbol"></a><span class="summary-name">re_symbol</span> = <code title="re.compile(r'\S{1,10}')">re.compile(r'\S<code class="re-op">{1,10}</code>')</code>
    </td>
  </tr>
</table>
<!-- ==================== PROPERTIES ==================== -->
<a name="section-Properties"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Properties</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Properties"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code>object</code></b>:
      <code>__class__</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== METHOD DETAILS ==================== -->
<a name="section-MethodDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Method Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-MethodDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="__init__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__init__</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">type</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">id_ref</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">id_source</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">symbol</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">accession</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">name</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">location</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">mol_seq</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">uri</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">domain_architecture</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">annotations</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">other</span>=<span class="sig-default">None</span>)</span>
    <br /><em class="fname">(Constructor)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Phylo.PhyloXML-pysrc.html#Sequence.__init__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  Initialize value for a Sequence object.
  <dl class="fields">
    <dt>Parameters:</dt>
    <dd><ul class="nomargin-top">
        <li><strong class="pname"><code>type</code></strong> ({'dna', 'rna', 'protein'}) - type of molecule this sequence represents</li>
        <li><strong class="pname"><code>id_ref</code></strong> (string) - reference to another resource</li>
        <li><strong class="pname"><code>id_source</code></strong> (string) - source for the reference</li>
        <li><strong class="pname"><code>symbol</code></strong> (string) - short symbol of the sequence, e.g. 'ACTM' (max. 10 chars)</li>
        <li><strong class="pname"><code>accession</code></strong> (Accession) - accession code for this sequence.</li>
        <li><strong class="pname"><code>name</code></strong> (string) - full name of the sequence, e.g. 'muscle Actin'</li>
        <li><strong class="pname"><code>location</code></strong> - location of a sequence on a genome/chromosome.</li>
        <li><strong class="pname"><code>mol_seq</code></strong> (MolSeq) - the molecular sequence itself</li>
        <li><strong class="pname"><code>uri</code></strong> (Uri) - link</li>
        <li><strong class="pname"><code>annotations</code></strong> (list of Annotation objects) - annotations on this sequence</li>
        <li><strong class="pname"><code>domain_architecture</code></strong> (DomainArchitecture) - protein domains on this sequence</li>
        <li><strong class="pname"><code>other</code></strong> (list of Other objects) - non-phyloXML elements</li>
    </ul></dd>
    <dt>Overrides:
        object.__init__
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="to_seqrecord"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">to_seqrecord</span>(<span class="sig-arg">self</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Phylo.PhyloXML-pysrc.html#Sequence.to_seqrecord">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Create a SeqRecord object from this Sequence instance.</p>
<p>The seqrecord.annotations dictionary is packed like so:</p>
<pre class="rst-literal-block">
{ # Sequence attributes with no SeqRecord equivalent:
  'id_ref': self.id_ref,
  'id_source': self.id_source,
  'location': self.location,
  'uri': { 'value': self.uri.value,
                  'desc': self.uri.desc,
                  'type': self.uri.type },
  # Sequence.annotations attribute (list of Annotations)
  'annotations': [{'ref': ann.ref,
                   'source': ann.source,
                   'evidence': ann.evidence,
                   'type': ann.type,
                   'confidence': [ann.confidence.value,
                                  ann.confidence.type],
                   'properties': [{'value': prop.value,
                                    'ref': prop.ref,
                                    'applies_to': prop.applies_to,
                                    'datatype': prop.datatype,
                                    'unit': prop.unit,
                                    'id_ref': prop.id_ref}
                                   for prop in ann.properties],
                  } for ann in self.annotations],
}
</pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== CLASS VARIABLE DETAILS ==================== -->
<a name="section-ClassVariableDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Class Variable Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-ClassVariableDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="alphabets"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <h3 class="epydoc">alphabets</h3>
  
  <dl class="fields">
  </dl>
  <dl class="fields">
    <dt>Value:</dt>
      <dd><table><tr><td><pre class="variable">
<code class="variable-group">{</code><code class="variable-quote">'</code><code class="variable-string">dna</code><code class="variable-quote">'</code><code class="variable-op">: </code>DNAAlphabet()<code class="variable-op">,</code>
 <code class="variable-quote">'</code><code class="variable-string">protein</code><code class="variable-quote">'</code><code class="variable-op">: </code>ProteinAlphabet()<code class="variable-op">,</code>
 <code class="variable-quote">'</code><code class="variable-string">rna</code><code class="variable-quote">'</code><code class="variable-op">: </code>RNAAlphabet()<code class="variable-group">}</code>
</pre></td></tr></table>
</dd>
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Tue Jul 16 17:01:07 2019
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
