<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.Blast.NCBIXML.BlastParser</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.Blast-module.html">Package&nbsp;Blast</a> ::
        <a href="Bio.Blast.NCBIXML-module.html">Module&nbsp;NCBIXML</a> ::
        Class&nbsp;BlastParser
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.Blast.NCBIXML.BlastParser-class.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== CLASS DESCRIPTION ==================== -->
<h1 class="epydoc">Class BlastParser</h1><p class="nomargin-top"><span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser">source&nbsp;code</a></span></p>
<pre class="base-tree">
xml.sax.handler.ContentHandler --+    
                                 |    
                        <a href="Bio.Blast.NCBIXML._XMLparser-class.html" onclick="show_private();">_XMLparser</a> --+
                                     |
                                    <strong class="uidshort">BlastParser</strong>
</pre>

<hr />
<p>Parse XML BLAST data into a Record.Blast object.</p>
<p>Parses XML output from BLAST (direct use discouraged).
This (now) returns a list of Blast records.
Historically it returned a single Blast record.
You are expected to use this via the parse or read functions.</p>
<dl class="rst-docutils">
<dt>All XML 'action' methods are private methods and may be:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li><tt class="rst-docutils literal">_start_TAG</tt> called when the start tag is found</li>
<li><tt class="rst-docutils literal">_end_TAG</tt> called when the end tag is found</li>
</ul>
</dd>
</dl>

<!-- ==================== INSTANCE METHODS ==================== -->
<a name="section-InstanceMethods"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Instance Methods</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-InstanceMethods"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Blast.NCBIXML.BlastParser-class.html#__init__" class="summary-sig-name">__init__</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">debug</span>=<span class="summary-sig-default">0</span>)</span><br />
      Constructor.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser.__init__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="reset"></a><span class="summary-sig-name">reset</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Reset all the data allowing reuse of the BlastParser() object.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser.reset">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_start_Iteration"></a><span class="summary-sig-name">_start_Iteration</span>(<span class="summary-sig-arg">self</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._start_Iteration">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Iteration"></a><span class="summary-sig-name">_end_Iteration</span>(<span class="summary-sig-arg">self</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Iteration">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Blast.NCBIXML.BlastParser-class.html#_end_BlastOutput_program" class="summary-sig-name" onclick="show_private();">_end_BlastOutput_program</a>(<span class="summary-sig-arg">self</span>)</span><br />
      BLAST program, e.g., blastp, blastn, etc.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_BlastOutput_program">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Blast.NCBIXML.BlastParser-class.html#_end_BlastOutput_version" class="summary-sig-name" onclick="show_private();">_end_BlastOutput_version</a>(<span class="summary-sig-arg">self</span>)</span><br />
      version number and date of the BLAST engine.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_BlastOutput_version">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Blast.NCBIXML.BlastParser-class.html#_end_BlastOutput_reference" class="summary-sig-name" onclick="show_private();">_end_BlastOutput_reference</a>(<span class="summary-sig-arg">self</span>)</span><br />
      a reference to the article describing the algorithm (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_BlastOutput_reference">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Blast.NCBIXML.BlastParser-class.html#_end_BlastOutput_db" class="summary-sig-name" onclick="show_private();">_end_BlastOutput_db</a>(<span class="summary-sig-arg">self</span>)</span><br />
      the database(s) searched (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_BlastOutput_db">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Blast.NCBIXML.BlastParser-class.html#_end_BlastOutput_query_ID" class="summary-sig-name" onclick="show_private();">_end_BlastOutput_query_ID</a>(<span class="summary-sig-arg">self</span>)</span><br />
      the identifier of the query (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_BlastOutput_query_ID">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Blast.NCBIXML.BlastParser-class.html#_end_BlastOutput_query_def" class="summary-sig-name" onclick="show_private();">_end_BlastOutput_query_def</a>(<span class="summary-sig-arg">self</span>)</span><br />
      the definition line of the query (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_BlastOutput_query_def">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Blast.NCBIXML.BlastParser-class.html#_end_BlastOutput_query_len" class="summary-sig-name" onclick="show_private();">_end_BlastOutput_query_len</a>(<span class="summary-sig-arg">self</span>)</span><br />
      the length of the query (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_BlastOutput_query_len">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Iteration_query_ID"></a><span class="summary-sig-name">_end_Iteration_query_ID</span>(<span class="summary-sig-arg">self</span>)</span><br />
      the identifier of the query (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Iteration_query_ID">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Iteration_query_def"></a><span class="summary-sig-name">_end_Iteration_query_def</span>(<span class="summary-sig-arg">self</span>)</span><br />
      the definition line of the query (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Iteration_query_def">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Iteration_query_len"></a><span class="summary-sig-name">_end_Iteration_query_len</span>(<span class="summary-sig-arg">self</span>)</span><br />
      the length of the query (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Iteration_query_len">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_BlastOutput_hits"></a><span class="summary-sig-name">_end_BlastOutput_hits</span>(<span class="summary-sig-arg">self</span>)</span><br />
      hits to the database sequences, one for every sequence (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_BlastOutput_hits">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Parameters_matrix"></a><span class="summary-sig-name">_end_Parameters_matrix</span>(<span class="summary-sig-arg">self</span>)</span><br />
      matrix used (-M on legacy BLAST) (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Parameters_matrix">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Parameters_expect"></a><span class="summary-sig-name">_end_Parameters_expect</span>(<span class="summary-sig-arg">self</span>)</span><br />
      expect values cutoff (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Parameters_expect">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Parameters_sc_match"></a><span class="summary-sig-name">_end_Parameters_sc_match</span>(<span class="summary-sig-arg">self</span>)</span><br />
      match score for nucleotide-nucleotide comparaison (-r) (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Parameters_sc_match">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Parameters_sc_mismatch"></a><span class="summary-sig-name">_end_Parameters_sc_mismatch</span>(<span class="summary-sig-arg">self</span>)</span><br />
      mismatch penalty for nucleotide-nucleotide comparaison (-r) (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Parameters_sc_mismatch">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Parameters_gap_open"></a><span class="summary-sig-name">_end_Parameters_gap_open</span>(<span class="summary-sig-arg">self</span>)</span><br />
      gap existence cost (-G) (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Parameters_gap_open">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Parameters_gap_extend"></a><span class="summary-sig-name">_end_Parameters_gap_extend</span>(<span class="summary-sig-arg">self</span>)</span><br />
      gap extension cose (-E) (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Parameters_gap_extend">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Parameters_filter"></a><span class="summary-sig-name">_end_Parameters_filter</span>(<span class="summary-sig-arg">self</span>)</span><br />
      filtering options (-F) (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Parameters_filter">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_start_Hit"></a><span class="summary-sig-name">_start_Hit</span>(<span class="summary-sig-arg">self</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._start_Hit">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Hit"></a><span class="summary-sig-name">_end_Hit</span>(<span class="summary-sig-arg">self</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Hit">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Hit_id"></a><span class="summary-sig-name">_end_Hit_id</span>(<span class="summary-sig-arg">self</span>)</span><br />
      identifier of the database sequence (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Hit_id">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Hit_def"></a><span class="summary-sig-name">_end_Hit_def</span>(<span class="summary-sig-arg">self</span>)</span><br />
      definition line of the database sequence (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Hit_def">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Hit_accession"></a><span class="summary-sig-name">_end_Hit_accession</span>(<span class="summary-sig-arg">self</span>)</span><br />
      accession of the database sequence (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Hit_accession">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Hit_len"></a><span class="summary-sig-name">_end_Hit_len</span>(<span class="summary-sig-arg">self</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Hit_len">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_start_Hsp"></a><span class="summary-sig-name">_start_Hsp</span>(<span class="summary-sig-arg">self</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._start_Hsp">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Hsp_score"></a><span class="summary-sig-name">_end_Hsp_score</span>(<span class="summary-sig-arg">self</span>)</span><br />
      raw score of HSP (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Hsp_score">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Hsp_bit_score"></a><span class="summary-sig-name">_end_Hsp_bit_score</span>(<span class="summary-sig-arg">self</span>)</span><br />
      bit score of HSP (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Hsp_bit_score">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Hsp_evalue"></a><span class="summary-sig-name">_end_Hsp_evalue</span>(<span class="summary-sig-arg">self</span>)</span><br />
      expect value of the HSP (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Hsp_evalue">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Hsp_query_from"></a><span class="summary-sig-name">_end_Hsp_query_from</span>(<span class="summary-sig-arg">self</span>)</span><br />
      offset of query at the start of the alignment (one-offset) (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Hsp_query_from">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Hsp_query_to"></a><span class="summary-sig-name">_end_Hsp_query_to</span>(<span class="summary-sig-arg">self</span>)</span><br />
      offset of query at the end of the alignment (one-offset) (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Hsp_query_to">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Hsp_hit_from"></a><span class="summary-sig-name">_end_Hsp_hit_from</span>(<span class="summary-sig-arg">self</span>)</span><br />
      offset of the database at the start of the alignment (one-offset) (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Hsp_hit_from">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Hsp_hit_to"></a><span class="summary-sig-name">_end_Hsp_hit_to</span>(<span class="summary-sig-arg">self</span>)</span><br />
      offset of the database at the end of the alignment (one-offset) (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Hsp_hit_to">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Hsp_query_frame"></a><span class="summary-sig-name">_end_Hsp_query_frame</span>(<span class="summary-sig-arg">self</span>)</span><br />
      frame of the query if applicable (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Hsp_query_frame">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Hsp_hit_frame"></a><span class="summary-sig-name">_end_Hsp_hit_frame</span>(<span class="summary-sig-arg">self</span>)</span><br />
      frame of the database sequence if applicable (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Hsp_hit_frame">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Hsp_identity"></a><span class="summary-sig-name">_end_Hsp_identity</span>(<span class="summary-sig-arg">self</span>)</span><br />
      number of identities in the alignment (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Hsp_identity">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Hsp_positive"></a><span class="summary-sig-name">_end_Hsp_positive</span>(<span class="summary-sig-arg">self</span>)</span><br />
      number of positive (conservative) substitutions in the alignment (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Hsp_positive">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Hsp_gaps"></a><span class="summary-sig-name">_end_Hsp_gaps</span>(<span class="summary-sig-arg">self</span>)</span><br />
      number of gaps in the alignment (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Hsp_gaps">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Hsp_align_len"></a><span class="summary-sig-name">_end_Hsp_align_len</span>(<span class="summary-sig-arg">self</span>)</span><br />
      length of the alignment (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Hsp_align_len">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Hsp_qseq"></a><span class="summary-sig-name">_end_Hsp_qseq</span>(<span class="summary-sig-arg">self</span>)</span><br />
      alignment string for the query (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Hsp_qseq">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Hsp_hseq"></a><span class="summary-sig-name">_end_Hsp_hseq</span>(<span class="summary-sig-arg">self</span>)</span><br />
      alignment string for the database (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Hsp_hseq">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Hsp_midline"></a><span class="summary-sig-name">_end_Hsp_midline</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Formatting middle line as normally seen in BLAST report (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Hsp_midline">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Statistics_db_num"></a><span class="summary-sig-name">_end_Statistics_db_num</span>(<span class="summary-sig-arg">self</span>)</span><br />
      number of sequences in the database (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Statistics_db_num">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Statistics_db_len"></a><span class="summary-sig-name">_end_Statistics_db_len</span>(<span class="summary-sig-arg">self</span>)</span><br />
      number of letters in the database (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Statistics_db_len">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Statistics_hsp_len"></a><span class="summary-sig-name">_end_Statistics_hsp_len</span>(<span class="summary-sig-arg">self</span>)</span><br />
      the effective HSP length (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Statistics_hsp_len">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Statistics_eff_space"></a><span class="summary-sig-name">_end_Statistics_eff_space</span>(<span class="summary-sig-arg">self</span>)</span><br />
      the effective search space (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Statistics_eff_space">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Statistics_kappa"></a><span class="summary-sig-name">_end_Statistics_kappa</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Karlin-Altschul parameter K (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Statistics_kappa">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Statistics_lambda"></a><span class="summary-sig-name">_end_Statistics_lambda</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Karlin-Altschul parameter Lambda (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Statistics_lambda">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_end_Statistics_entropy"></a><span class="summary-sig-name">_end_Statistics_entropy</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Karlin-Altschul parameter H (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_Statistics_entropy">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code><a href="Bio.Blast.NCBIXML._XMLparser-class.html" onclick="show_private();">_XMLparser</a></code></b>:
      <code><a href="Bio.Blast.NCBIXML._XMLparser-class.html#characters">characters</a></code>,
      <code><a href="Bio.Blast.NCBIXML._XMLparser-class.html#endElement">endElement</a></code>,
      <code><a href="Bio.Blast.NCBIXML._XMLparser-class.html#startElement">startElement</a></code>
      </p>
    <div class="private">    <p class="indent-wrapped-lines"><b>Inherited from <code><a href="Bio.Blast.NCBIXML._XMLparser-class.html" onclick="show_private();">_XMLparser</a></code></b> (private):
      <code><a href="Bio.Blast.NCBIXML._XMLparser-class.html#_secure_name" onclick="show_private();">_secure_name</a></code>
      </p></div>
    <p class="indent-wrapped-lines"><b>Inherited from <code>xml.sax.handler.ContentHandler</code></b>:
      <code>endDocument</code>,
      <code>endElementNS</code>,
      <code>endPrefixMapping</code>,
      <code>ignorableWhitespace</code>,
      <code>processingInstruction</code>,
      <code>setDocumentLocator</code>,
      <code>skippedEntity</code>,
      <code>startDocument</code>,
      <code>startElementNS</code>,
      <code>startPrefixMapping</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== METHOD DETAILS ==================== -->
<a name="section-MethodDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Method Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-MethodDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="__init__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__init__</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">debug</span>=<span class="sig-default">0</span>)</span>
    <br /><em class="fname">(Constructor)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser.__init__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Constructor.</p>
<p>debug - integer, amount of debug information to print</p>
  <dl class="fields">
    <dt>Overrides:
        xml.sax.handler.ContentHandler.__init__
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="_end_BlastOutput_program"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_end_BlastOutput_program</span>(<span class="sig-arg">self</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_BlastOutput_program">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>BLAST program, e.g., blastp, blastn, etc.</p>
<p>Save this to put on each blast record object</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_end_BlastOutput_version"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_end_BlastOutput_version</span>(<span class="sig-arg">self</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_BlastOutput_version">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>version number and date of the BLAST engine.</p>
<p>e.g. &quot;BLASTX 2.2.12 [Aug-07-2005]&quot; but there can also be
variants like &quot;BLASTP 2.2.18+&quot; without the date.</p>
<p>Save this to put on each blast record object</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_end_BlastOutput_reference"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_end_BlastOutput_reference</span>(<span class="sig-arg">self</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_BlastOutput_reference">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>a reference to the article describing the algorithm (PRIVATE).</p>
<p>Save this to put on each blast record object</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_end_BlastOutput_db"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_end_BlastOutput_db</span>(<span class="sig-arg">self</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_BlastOutput_db">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>the database(s) searched (PRIVATE).</p>
<p>Save this to put on each blast record object</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_end_BlastOutput_query_ID"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_end_BlastOutput_query_ID</span>(<span class="sig-arg">self</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_BlastOutput_query_ID">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>the identifier of the query (PRIVATE).</p>
<p>Important in old pre 2.2.14 BLAST, for recent versions
&lt;Iteration_query-ID&gt; is enough</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_end_BlastOutput_query_def"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_end_BlastOutput_query_def</span>(<span class="sig-arg">self</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_BlastOutput_query_def">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>the definition line of the query (PRIVATE).</p>
<p>Important in old pre 2.2.14 BLAST, for recent versions
&lt;Iteration_query-def&gt; is enough</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_end_BlastOutput_query_len"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_end_BlastOutput_query_len</span>(<span class="sig-arg">self</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Blast.NCBIXML-pysrc.html#BlastParser._end_BlastOutput_query_len">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>the length of the query (PRIVATE).</p>
<p>Important in old pre 2.2.14 BLAST, for recent versions
&lt;Iteration_query-len&gt; is enough</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Oct 21 18:01:37 2015
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
