<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.Entrez</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        Package&nbsp;Entrez
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.Entrez-module.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== PACKAGE DESCRIPTION ==================== -->
<h1 class="epydoc">Package Entrez</h1><p class="nomargin-top"><span class="codelink"><a href="Bio.Entrez-pysrc.html">source&nbsp;code</a></span></p>
<p>Provides code to access NCBI over the WWW.</p>
<p>The main Entrez web page is available at:
<a class="rst-reference external" href="http://www.ncbi.nlm.nih.gov/Entrez/" target="_top">http://www.ncbi.nlm.nih.gov/Entrez/</a></p>
<p>Entrez Programming Utilities web page is available at:
<a class="rst-reference external" href="http://www.ncbi.nlm.nih.gov/books/NBK25501/" target="_top">http://www.ncbi.nlm.nih.gov/books/NBK25501/</a></p>
<p>This module provides a number of functions like <tt class="rst-docutils literal">efetch</tt> (short for
Entrez Fetch) which will return the data as a handle object. This is
a standard interface used in Python for reading data from a file, or
in this case a remote network connection, and provides methods like
<tt class="rst-docutils literal">.read()</tt> or offers iteration over the contents line by line. See
also &quot;What the heck is a handle?&quot; in the Biopython Tutorial and
Cookbook: <a class="rst-reference external" href="http://biopython.org/DIST/docs/tutorial/Tutorial.html" target="_top">http://biopython.org/DIST/docs/tutorial/Tutorial.html</a>
<a class="rst-reference external" href="http://biopython.org/DIST/docs/tutorial/Tutorial.pdf" target="_top">http://biopython.org/DIST/docs/tutorial/Tutorial.pdf</a></p>
<p>Unlike a handle to a file on disk from the <tt class="rst-docutils literal">open(filename)</tt> function,
which has a <tt class="rst-docutils literal">.name</tt> attribute giving the filename, the handles from
<tt class="rst-docutils literal">Bio.Entrez</tt> all have a <tt class="rst-docutils literal">.url</tt> attribute instead giving the URL
used to connect to the NCBI Entrez API.</p>
<p>All the functions that send requests to the NCBI Entrez API will
automatically respect the NCBI rate limit (of 3 requests per second
without an API key, or 10 requests per second with an API key) and
will automatically retry when encountering transient failures
(i.e. connection failures or HTTP 5XX codes). By default, Biopython
does a maximum of three tries before giving up, and sleeps for 15
seconds between tries. You can tweak these parameters by setting
<tt class="rst-docutils literal">Bio.Entrez.max_tries</tt> and <tt class="rst-docutils literal">Bio.Entrez.sleep_between_tries</tt>.</p>
<p>The Entrez module also provides an XML parser which takes a handle
as input.</p>
<p>Variables:</p>
<blockquote>
<ul>
<li><p class="rst-first">email        Set the Entrez email parameter (default is not set).</p>
</li>
<li><p class="rst-first">tool         Set the Entrez tool parameter (default is <tt class="rst-docutils literal">biopython</tt>).</p>
</li>
<li><dl class="rst-first rst-docutils">
<dt>api_key      Personal API key from NCBI. If not set, only 3 queries per</dt>
<dd><p class="rst-first rst-last">seconds are allowed. 10 queries per seconds otherwise with a
valid API key.</p>
</dd>
</dl>
</li>
</ul>
</blockquote>
<p>Functions:</p>
<blockquote>
<ul>
<li><p class="rst-first">efetch       Retrieves records in the requested format from a list of one or
more primary IDs or from the user's environment</p>
</li>
<li><p class="rst-first">epost        Posts a file containing a list of primary IDs for future use in
the user's environment to use with subsequent search strategies</p>
</li>
<li><p class="rst-first">esearch      Searches and retrieves primary IDs (for use in EFetch, ELink,
and ESummary) and term translations and optionally retains
results for future use in the user's environment.</p>
</li>
<li><p class="rst-first">elink        Checks for the existence of an external or Related Articles link
from a list of one or more primary IDs.  Retrieves primary IDs
and relevancy scores for links to Entrez databases or Related
Articles;  creates a hyperlink to the primary LinkOut provider
for a specific ID and database, or lists LinkOut URLs
and Attributes for multiple IDs.</p>
</li>
<li><p class="rst-first">einfo        Provides field index term counts, last update, and available
links for each database.</p>
</li>
<li><p class="rst-first">esummary     Retrieves document summaries from a list of primary IDs or from
the user's environment.</p>
</li>
<li><p class="rst-first">egquery      Provides Entrez database counts in XML for a single search
using Global Query.</p>
</li>
<li><p class="rst-first">espell       Retrieves spelling suggestions.</p>
</li>
<li><p class="rst-first">ecitmatch    Retrieves PubMed IDs (PMIDs) that correspond to a set of
input citation strings.</p>
</li>
<li><p class="rst-first">read         Parses the XML results returned by any of the above functions.
Typical usage is:</p>
<blockquote>
<blockquote>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio <span class="py-keyword">import</span> Entrez
<span class="py-prompt">&gt;&gt;&gt; </span>Entrez.email = <span class="py-string">&quot;Your.Name.Here@example.org&quot;</span>
<span class="py-prompt">&gt;&gt;&gt; </span>handle = Entrez.einfo() <span class="py-comment"># or esearch, efetch, ...</span>
<span class="py-prompt">&gt;&gt;&gt; </span>record = Entrez.read(handle)
<span class="py-prompt">&gt;&gt;&gt; </span>handle.close()</pre>
</blockquote>
<p>where record is now a Python dictionary or list.</p>
</blockquote>
</li>
<li><p class="rst-first">parse        Parses the XML results returned by those of the above functions
which can return multiple records - such as efetch, esummary
and elink. Typical usage is:</p>
<blockquote>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span>handle = Entrez.esummary(db=<span class="py-string">&quot;pubmed&quot;</span>, id=<span class="py-string">&quot;19304878,14630660&quot;</span>, retmode=<span class="py-string">&quot;xml&quot;</span>)
<span class="py-prompt">&gt;&gt;&gt; </span>records = Entrez.parse(handle)
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">for</span> record <span class="py-keyword">in</span> records:
<span class="py-more">... </span>    <span class="py-comment"># each record is a Python dictionary or list.</span>
<span class="py-more">... </span>    <span class="py-keyword">print</span>(record[<span class="py-string">'Title'</span>])
<span class="py-output">Biopython: freely available Python tools for computational molecular biology and bioinformatics.</span>
<span class="py-output">PDB file parser and structure class implemented in Python.</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>handle.close()</pre>
</blockquote>
<p>This function is appropriate only if the XML file contains
multiple records, and is particular useful for large files.</p>
</li>
<li><p class="rst-first">_open        Internally used function.</p>
</li>
</ul>
</blockquote>

<!-- ==================== SUBMODULES ==================== -->
<a name="section-Submodules"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Submodules</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Submodules"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
  <tr><td class="summary">
  <ul class="nomargin">
    <li> <strong class="uidlink"><a href="Bio.Entrez.Parser-module.html">Bio.Entrez.Parser</a></strong>: <em class="summary">Parser for XML results returned by NCBI's Entrez Utilities.</em>    </li>
  </ul></td></tr>
</table>

<br />
<!-- ==================== FUNCTIONS ==================== -->
<a name="section-Functions"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Functions</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Functions"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Entrez-module.html#epost" class="summary-sig-name">epost</a>(<span class="summary-sig-arg">db</span>,
        <span class="summary-sig-arg">**keywds</span>)</span><br />
      Post a file of identifiers for future use.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Entrez-pysrc.html#epost">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Entrez-module.html#efetch" class="summary-sig-name">efetch</a>(<span class="summary-sig-arg">db</span>,
        <span class="summary-sig-arg">**keywords</span>)</span><br />
      Fetch Entrez results which are returned as a handle.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Entrez-pysrc.html#efetch">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Entrez-module.html#esearch" class="summary-sig-name">esearch</a>(<span class="summary-sig-arg">db</span>,
        <span class="summary-sig-arg">term</span>,
        <span class="summary-sig-arg">**keywds</span>)</span><br />
      Run an Entrez search and return a handle to the results.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Entrez-pysrc.html#esearch">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Entrez-module.html#elink" class="summary-sig-name">elink</a>(<span class="summary-sig-arg">**keywds</span>)</span><br />
      Check for linked external articles and return a handle.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Entrez-pysrc.html#elink">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Entrez-module.html#einfo" class="summary-sig-name">einfo</a>(<span class="summary-sig-arg">**keywds</span>)</span><br />
      Return a summary of the Entrez databases as a results handle.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Entrez-pysrc.html#einfo">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Entrez-module.html#esummary" class="summary-sig-name">esummary</a>(<span class="summary-sig-arg">**keywds</span>)</span><br />
      Retrieve document summaries as a results handle.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Entrez-pysrc.html#esummary">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Entrez-module.html#egquery" class="summary-sig-name">egquery</a>(<span class="summary-sig-arg">**keywds</span>)</span><br />
      Provide Entrez database counts for a global search.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Entrez-pysrc.html#egquery">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Entrez-module.html#espell" class="summary-sig-name">espell</a>(<span class="summary-sig-arg">**keywds</span>)</span><br />
      Retrieve spelling suggestions as a results handle.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Entrez-pysrc.html#espell">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_update_ecitmatch_variables"></a><span class="summary-sig-name">_update_ecitmatch_variables</span>(<span class="summary-sig-arg">keywds</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Entrez-pysrc.html#_update_ecitmatch_variables">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Entrez-module.html#ecitmatch" class="summary-sig-name">ecitmatch</a>(<span class="summary-sig-arg">**keywds</span>)</span><br />
      Retrieve PMIDs for input citation strings, returned as a handle.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Entrez-pysrc.html#ecitmatch">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Entrez-module.html#read" class="summary-sig-name">read</a>(<span class="summary-sig-arg">handle</span>,
        <span class="summary-sig-arg">validate</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">escape</span>=<span class="summary-sig-default">False</span>)</span><br />
      Parse an XML file from the NCBI Entrez Utilities into python objects.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Entrez-pysrc.html#read">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Entrez-module.html#parse" class="summary-sig-name">parse</a>(<span class="summary-sig-arg">handle</span>,
        <span class="summary-sig-arg">validate</span>=<span class="summary-sig-default">True</span>,
        <span class="summary-sig-arg">escape</span>=<span class="summary-sig-default">False</span>)</span><br />
      Parse an XML file from the NCBI Entrez Utilities into python objects.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Entrez-pysrc.html#parse">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.Entrez-module.html#_open" class="summary-sig-name" onclick="show_private();">_open</a>(<span class="summary-sig-arg">cgi</span>,
        <span class="summary-sig-arg">params</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">post</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">ecitmatch</span>=<span class="summary-sig-default">False</span>)</span><br />
      Build the URL and open a handle to it (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Entrez-pysrc.html#_open">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_construct_params"></a><span class="summary-sig-name">_construct_params</span>(<span class="summary-sig-arg">params</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Entrez-pysrc.html#_construct_params">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_encode_options"></a><span class="summary-sig-name">_encode_options</span>(<span class="summary-sig-arg">ecitmatch</span>,
        <span class="summary-sig-arg">params</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Entrez-pysrc.html#_encode_options">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_construct_cgi"></a><span class="summary-sig-name">_construct_cgi</span>(<span class="summary-sig-arg">cgi</span>,
        <span class="summary-sig-arg">post</span>,
        <span class="summary-sig-arg">options</span>)</span></td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.Entrez-pysrc.html#_construct_cgi">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
</table>
<!-- ==================== VARIABLES ==================== -->
<a name="section-Variables"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Variables</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Variables"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="email"></a><span class="summary-name">email</span> = <code title="None">None</code><br />
      hash(x)
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="max_tries"></a><span class="summary-name">max_tries</span> = <code title="3">3</code><br />
      hash(x)
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="sleep_between_tries"></a><span class="summary-name">sleep_between_tries</span> = <code title="15">15</code>
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="tool"></a><span class="summary-name">tool</span> = <code title="'biopython'"><code class="variable-quote">'</code><code class="variable-string">biopython</code><code class="variable-quote">'</code></code>
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="api_key"></a><span class="summary-name">api_key</span> = <code title="None">None</code><br />
      hash(x)
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="__package__"></a><span class="summary-name">__package__</span> = <code title="'Bio.Entrez'"><code class="variable-quote">'</code><code class="variable-string">Bio.Entrez</code><code class="variable-quote">'</code></code>
    </td>
  </tr>
</table>
<!-- ==================== FUNCTION DETAILS ==================== -->
<a name="section-FunctionDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Function Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-FunctionDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="epost"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">epost</span>(<span class="sig-arg">db</span>,
        <span class="sig-arg">**keywds</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Entrez-pysrc.html#epost">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Post a file of identifiers for future use.</p>
<p>Posts a file containing a list of UIs for future use in the user's
environment to use with subsequent search strategies.</p>
<p>See the online documentation for an explanation of the parameters:
<a class="rst-reference external" href="http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.EPost" target="_top">http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.EPost</a></p>
<p>Return a handle to the results.</p>
<p>Raises an IOError exception if there's a network error.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="efetch"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">efetch</span>(<span class="sig-arg">db</span>,
        <span class="sig-arg">**keywords</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Entrez-pysrc.html#efetch">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Fetch Entrez results which are returned as a handle.</p>
<p>EFetch retrieves records in the requested format from a list of one or
more UIs or from user's environment.</p>
<p>See the online documentation for an explanation of the parameters:
<a class="rst-reference external" href="http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.EFetch" target="_top">http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.EFetch</a></p>
<p>Return a handle to the results.</p>
<p>Raises an IOError exception if there's a network error.</p>
<p>Short example:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio <span class="py-keyword">import</span> Entrez
<span class="py-prompt">&gt;&gt;&gt; </span>Entrez.email = <span class="py-string">&quot;Your.Name.Here@example.org&quot;</span>
<span class="py-prompt">&gt;&gt;&gt; </span>handle = Entrez.efetch(db=<span class="py-string">&quot;nucleotide&quot;</span>, id=<span class="py-string">&quot;AY851612&quot;</span>, rettype=<span class="py-string">&quot;gb&quot;</span>, retmode=<span class="py-string">&quot;text&quot;</span>)
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(handle.readline().strip())
<span class="py-output">LOCUS       AY851612                 892 bp    DNA     linear   PLN 10-APR-2007</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>handle.close()</pre>
<p>This will automatically use an HTTP POST rather than HTTP GET if there
are over 200 identifiers as recommended by the NCBI.</p>
<p><strong>Warning:</strong> The NCBI changed the default retmode in Feb 2012, so many
databases which previously returned text output now give XML.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="esearch"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">esearch</span>(<span class="sig-arg">db</span>,
        <span class="sig-arg">term</span>,
        <span class="sig-arg">**keywds</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Entrez-pysrc.html#esearch">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Run an Entrez search and return a handle to the results.</p>
<p>ESearch searches and retrieves primary IDs (for use in EFetch, ELink
and ESummary) and term translations, and optionally retains results
for future use in the user's environment.</p>
<p>See the online documentation for an explanation of the parameters:
<a class="rst-reference external" href="http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.ESearch" target="_top">http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.ESearch</a></p>
<p>Return a handle to the results which are always in XML format.</p>
<p>Raises an IOError exception if there's a network error.</p>
<p>Short example:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio <span class="py-keyword">import</span> Entrez
<span class="py-prompt">&gt;&gt;&gt; </span>Entrez.email = <span class="py-string">&quot;Your.Name.Here@example.org&quot;</span>
<span class="py-prompt">&gt;&gt;&gt; </span>handle = Entrez.esearch(db=<span class="py-string">&quot;nucleotide&quot;</span>, retmax=10, term=<span class="py-string">&quot;opuntia[ORGN] accD&quot;</span>, idtype=<span class="py-string">&quot;acc&quot;</span>)
<span class="py-prompt">&gt;&gt;&gt; </span>record = Entrez.read(handle)
<span class="py-prompt">&gt;&gt;&gt; </span>handle.close()
<span class="py-prompt">&gt;&gt;&gt; </span>int(record[<span class="py-string">&quot;Count&quot;</span>]) &gt;= 2
<span class="py-output">True</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-string">&quot;EF590893.1&quot;</span> <span class="py-keyword">in</span> record[<span class="py-string">&quot;IdList&quot;</span>]
<span class="py-output">True</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-string">&quot;EF590892.1&quot;</span> <span class="py-keyword">in</span> record[<span class="py-string">&quot;IdList&quot;</span>]
<span class="py-output">True</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="elink"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">elink</span>(<span class="sig-arg">**keywds</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Entrez-pysrc.html#elink">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Check for linked external articles and return a handle.</p>
<p>ELink checks for the existence of an external or Related Articles link
from a list of one or more primary IDs;  retrieves IDs and relevancy
scores for links to Entrez databases or Related Articles; creates a
hyperlink to the primary LinkOut provider for a specific ID and
database, or lists LinkOut URLs and attributes for multiple IDs.</p>
<p>See the online documentation for an explanation of the parameters:
<a class="rst-reference external" href="http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.ELink" target="_top">http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.ELink</a></p>
<p>Return a handle to the results, by default in XML format.</p>
<p>Raises an IOError exception if there's a network error.</p>
<p>This example finds articles related to the Biopython application
note's entry in the PubMed database:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio <span class="py-keyword">import</span> Entrez
<span class="py-prompt">&gt;&gt;&gt; </span>Entrez.email = <span class="py-string">&quot;Your.Name.Here@example.org&quot;</span>
<span class="py-prompt">&gt;&gt;&gt; </span>pmid = <span class="py-string">&quot;19304878&quot;</span>
<span class="py-prompt">&gt;&gt;&gt; </span>handle = Entrez.elink(dbfrom=<span class="py-string">&quot;pubmed&quot;</span>, id=pmid, linkname=<span class="py-string">&quot;pubmed_pubmed&quot;</span>)
<span class="py-prompt">&gt;&gt;&gt; </span>record = Entrez.read(handle)
<span class="py-prompt">&gt;&gt;&gt; </span>handle.close()
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(record[0][<span class="py-string">&quot;LinkSetDb&quot;</span>][0][<span class="py-string">&quot;LinkName&quot;</span>])
<span class="py-output">pubmed_pubmed</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>linked = [link[<span class="py-string">&quot;Id&quot;</span>] <span class="py-keyword">for</span> link <span class="py-keyword">in</span> record[0][<span class="py-string">&quot;LinkSetDb&quot;</span>][0][<span class="py-string">&quot;Link&quot;</span>]]
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-string">&quot;17121776&quot;</span> <span class="py-keyword">in</span> linked
<span class="py-output">True</span></pre>
<p>This is explained in much more detail in the Biopython Tutorial.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="einfo"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">einfo</span>(<span class="sig-arg">**keywds</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Entrez-pysrc.html#einfo">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Return a summary of the Entrez databases as a results handle.</p>
<p>EInfo provides field names, index term counts, last update, and
available links for each Entrez database.</p>
<p>See the online documentation for an explanation of the parameters:
<a class="rst-reference external" href="http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.EInfo" target="_top">http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.EInfo</a></p>
<p>Return a handle to the results, by default in XML format.</p>
<p>Raises an IOError exception if there's a network error.</p>
<p>Short example:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio <span class="py-keyword">import</span> Entrez
<span class="py-prompt">&gt;&gt;&gt; </span>Entrez.email = <span class="py-string">&quot;Your.Name.Here@example.org&quot;</span>
<span class="py-prompt">&gt;&gt;&gt; </span>record = Entrez.read(Entrez.einfo())
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-string">'pubmed'</span> <span class="py-keyword">in</span> record[<span class="py-string">'DbList'</span>]
<span class="py-output">True</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="esummary"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">esummary</span>(<span class="sig-arg">**keywds</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Entrez-pysrc.html#esummary">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Retrieve document summaries as a results handle.</p>
<p>ESummary retrieves document summaries from a list of primary IDs or
from the user's environment.</p>
<p>See the online documentation for an explanation of the parameters:
<a class="rst-reference external" href="http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.ESummary" target="_top">http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.ESummary</a></p>
<p>Return a handle to the results, by default in XML format.</p>
<p>Raises an IOError exception if there's a network error.</p>
<p>This example discovers more about entry 19923 in the structure
database:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio <span class="py-keyword">import</span> Entrez
<span class="py-prompt">&gt;&gt;&gt; </span>Entrez.email = <span class="py-string">&quot;Your.Name.Here@example.org&quot;</span>
<span class="py-prompt">&gt;&gt;&gt; </span>handle = Entrez.esummary(db=<span class="py-string">&quot;structure&quot;</span>, id=<span class="py-string">&quot;19923&quot;</span>)
<span class="py-prompt">&gt;&gt;&gt; </span>record = Entrez.read(handle)
<span class="py-prompt">&gt;&gt;&gt; </span>handle.close()
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(record[0][<span class="py-string">&quot;Id&quot;</span>])
<span class="py-output">19923</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(record[0][<span class="py-string">&quot;PdbDescr&quot;</span>])
<span class="py-output">Crystal Structure Of E. Coli Aconitase B</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="egquery"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">egquery</span>(<span class="sig-arg">**keywds</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Entrez-pysrc.html#egquery">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Provide Entrez database counts for a global search.</p>
<p>EGQuery provides Entrez database counts in XML for a single search
using Global Query.</p>
<p>See the online documentation for an explanation of the parameters:
<a class="rst-reference external" href="http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.EGQuery" target="_top">http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.EGQuery</a></p>
<p>Return a handle to the results in XML format.</p>
<p>Raises an IOError exception if there's a network error.</p>
<p>This quick example based on a longer version from the Biopython
Tutorial just checks there are over 60 matches for 'Biopython'
in PubMedCentral:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio <span class="py-keyword">import</span> Entrez
<span class="py-prompt">&gt;&gt;&gt; </span>Entrez.email = <span class="py-string">&quot;Your.Name.Here@example.org&quot;</span>
<span class="py-prompt">&gt;&gt;&gt; </span>handle = Entrez.egquery(term=<span class="py-string">&quot;biopython&quot;</span>)
<span class="py-prompt">&gt;&gt;&gt; </span>record = Entrez.read(handle)
<span class="py-prompt">&gt;&gt;&gt; </span>handle.close()
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">for</span> row <span class="py-keyword">in</span> record[<span class="py-string">&quot;eGQueryResult&quot;</span>]:
<span class="py-more">... </span>    <span class="py-keyword">if</span> <span class="py-string">&quot;pmc&quot;</span> <span class="py-keyword">in</span> row[<span class="py-string">&quot;DbName&quot;</span>]:
<span class="py-more">... </span>        <span class="py-keyword">print</span>(int(row[<span class="py-string">&quot;Count&quot;</span>]) &gt; 60)
<span class="py-output">True</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="espell"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">espell</span>(<span class="sig-arg">**keywds</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Entrez-pysrc.html#espell">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Retrieve spelling suggestions as a results handle.</p>
<p>ESpell retrieves spelling suggestions, if available.</p>
<p>See the online documentation for an explanation of the parameters:
<a class="rst-reference external" href="http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.ESpell" target="_top">http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.ESpell</a></p>
<p>Return a handle to the results, by default in XML format.</p>
<p>Raises an IOError exception if there's a network error.</p>
<p>Short example:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio <span class="py-keyword">import</span> Entrez
<span class="py-prompt">&gt;&gt;&gt; </span>Entrez.email = <span class="py-string">&quot;Your.Name.Here@example.org&quot;</span>
<span class="py-prompt">&gt;&gt;&gt; </span>record = Entrez.read(Entrez.espell(term=<span class="py-string">&quot;biopythooon&quot;</span>))
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(record[<span class="py-string">&quot;Query&quot;</span>])
<span class="py-output">biopythooon</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(record[<span class="py-string">&quot;CorrectedQuery&quot;</span>])
<span class="py-output">biopython</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="ecitmatch"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">ecitmatch</span>(<span class="sig-arg">**keywds</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Entrez-pysrc.html#ecitmatch">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Retrieve PMIDs for input citation strings, returned as a handle.</p>
<p>ECitMatch retrieves PubMed IDs (PMIDs) that correspond to a set of input
citation strings.</p>
<p>See the online documentation for an explanation of the parameters:
<a class="rst-reference external" href="http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.ECitMatch" target="_top">http://www.ncbi.nlm.nih.gov/books/NBK25499/#chapter4.ECitMatch</a></p>
<p>Return a handle to the results, by default in plain text</p>
<p>Raises an IOError exception if there's a network error.</p>
<p>Short example:</p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio <span class="py-keyword">import</span> Entrez
<span class="py-prompt">&gt;&gt;&gt; </span>Entrez.email = <span class="py-string">&quot;Your.Name.Here@example.org&quot;</span>
<span class="py-prompt">&gt;&gt;&gt; </span>citation_1 = {<span class="py-string">&quot;journal_title&quot;</span>: <span class="py-string">&quot;proc natl acad sci u s a&quot;</span>,
<span class="py-more">... </span>              <span class="py-string">&quot;year&quot;</span>: <span class="py-string">&quot;1991&quot;</span>, <span class="py-string">&quot;volume&quot;</span>: <span class="py-string">&quot;88&quot;</span>, <span class="py-string">&quot;first_page&quot;</span>: <span class="py-string">&quot;3248&quot;</span>,
<span class="py-more">... </span>              <span class="py-string">&quot;author_name&quot;</span>: <span class="py-string">&quot;mann bj&quot;</span>, <span class="py-string">&quot;key&quot;</span>: <span class="py-string">&quot;citation_1&quot;</span>}
<span class="py-prompt">&gt;&gt;&gt; </span>handle = Entrez.ecitmatch(db=<span class="py-string">&quot;pubmed&quot;</span>, bdata=[citation_1])
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(handle.read().strip().split(<span class="py-string">&quot;|&quot;</span>))
<span class="py-output">['proc natl acad sci u s a', '1991', '88', '3248', 'mann bj', 'citation_1', '2014248']</span>
<span class="py-output"></span><span class="py-prompt">&gt;&gt;&gt; </span>handle.close()</pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="read"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">read</span>(<span class="sig-arg">handle</span>,
        <span class="sig-arg">validate</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">escape</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Entrez-pysrc.html#read">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Parse an XML file from the NCBI Entrez Utilities into python objects.</p>
<p>This function parses an XML file created by NCBI's Entrez Utilities,
returning a multilevel data structure of Python lists and dictionaries.
Most XML files returned by NCBI's Entrez Utilities can be parsed by
this function, provided its DTD is available. Biopython includes the
DTDs for most commonly used Entrez Utilities.</p>
<p>If validate is True (default), the parser will validate the XML file
against the DTD, and raise an error if the XML file contains tags that
are not represented in the DTD. If validate is False, the parser will
simply skip such tags.</p>
<p>If escape is True, all characters that are not valid HTML are replaced
by HTML escape characters to guarantee that the returned strings are
valid HTML fragments. For example, a less-than sign (&lt;) is replaced by
&amp;lt;. If escape is False (default), the string is returned as is.</p>
<p>Whereas the data structure seems to consist of generic Python lists,
dictionaries, strings, and so on, each of these is actually a class
derived from the base type. This allows us to store the attributes
(if any) of each element in a dictionary my_element.attributes, and
the tag name in my_element.tag.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="parse"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">parse</span>(<span class="sig-arg">handle</span>,
        <span class="sig-arg">validate</span>=<span class="sig-default">True</span>,
        <span class="sig-arg">escape</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Entrez-pysrc.html#parse">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Parse an XML file from the NCBI Entrez Utilities into python objects.</p>
<p>This function parses an XML file created by NCBI's Entrez Utilities,
returning a multilevel data structure of Python lists and dictionaries.
This function is suitable for XML files that (in Python) can be represented
as a list of individual records. Whereas 'read' reads the complete file
and returns a single Python list, 'parse' is a generator function that
returns the records one by one. This function is therefore particularly
useful for parsing large files.</p>
<p>Most XML files returned by NCBI's Entrez Utilities can be parsed by
this function, provided its DTD is available. Biopython includes the
DTDs for most commonly used Entrez Utilities.</p>
<p>If validate is True (default), the parser will validate the XML file
against the DTD, and raise an error if the XML file contains tags that
are not represented in the DTD. If validate is False, the parser will
simply skip such tags.</p>
<p>If escape is True, all characters that are not valid HTML are replaced
by HTML escape characters to guarantee that the returned strings are
valid HTML fragments. For example, a less-than sign (&lt;) is replaced by
&amp;lt;. If escape is False (default), the string is returned as is.</p>
<p>Whereas the data structure seems to consist of generic Python lists,
dictionaries, strings, and so on, each of these is actually a class
derived from the base type. This allows us to store the attributes
(if any) of each element in a dictionary my_element.attributes, and
the tag name in my_element.tag.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_open"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_open</span>(<span class="sig-arg">cgi</span>,
        <span class="sig-arg">params</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">post</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">ecitmatch</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.Entrez-pysrc.html#_open">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Build the URL and open a handle to it (PRIVATE).</p>
<p>Open a handle to Entrez.  cgi is the URL for the cgi script to access.
params is a dictionary with the options to pass to it.  Does some
simple error checking, and will raise an IOError if it encounters one.</p>
<p>The argument post should be a boolean to explicitly control if an HTTP
POST should be used rather an HTTP GET based on the query length.
By default (post=None), POST is used if the URL encoded parameters would
be over 1000 characters long.</p>
<p>This function also enforces the &quot;up to three queries per second rule&quot;
to avoid abusing the NCBI servers.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Tue Jul 16 17:00:52 2019
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
