<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.SearchIO.ExonerateIO.exonerate_text</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.SearchIO-module.html">Package&nbsp;SearchIO</a> ::
        <a href="Bio.SearchIO.ExonerateIO-module.html">Package&nbsp;ExonerateIO</a> ::
        Module&nbsp;exonerate_text
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.SearchIO.ExonerateIO.exonerate_text-module.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== MODULE DESCRIPTION ==================== -->
<h1 class="epydoc">Module exonerate_text</h1><p class="nomargin-top"><span class="codelink"><a href="Bio.SearchIO.ExonerateIO.exonerate_text-pysrc.html">source&nbsp;code</a></span></p>
Bio.SearchIO parser for Exonerate plain text output format.

<!-- ==================== CLASSES ==================== -->
<a name="section-Classes"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Classes</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Classes"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextParser-class.html" class="summary-name">ExonerateTextParser</a><br />
      Parser for Exonerate plain text output.
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextIndexer-class.html" class="summary-name">ExonerateTextIndexer</a><br />
      Indexer class for Exonerate plain text.
    </td>
  </tr>
</table>
<!-- ==================== FUNCTIONS ==================== -->
<a name="section-Functions"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Functions</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Functions"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_flip_codons"></a><span class="summary-sig-name">_flip_codons</span>(<span class="summary-sig-arg">codon_seq</span>,
        <span class="summary-sig-arg">target_seq</span>)</span><br />
      Flips the codon characters from one seq to another (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SearchIO.ExonerateIO.exonerate_text-pysrc.html#_flip_codons">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_get_block_coords"></a><span class="summary-sig-name">_get_block_coords</span>(<span class="summary-sig-arg">parsed_seq</span>,
        <span class="summary-sig-arg">row_dict</span>,
        <span class="summary-sig-arg">has_ner</span>=<span class="summary-sig-default">False</span>)</span><br />
      Return a list of start, end coordinates for each given block in the sequence (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SearchIO.ExonerateIO.exonerate_text-pysrc.html#_get_block_coords">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_get_inter_coords" class="summary-sig-name" onclick="show_private();">_get_inter_coords</a>(<span class="summary-sig-arg">coords</span>,
        <span class="summary-sig-arg">strand</span>=<span class="summary-sig-default">1</span>)</span><br />
      Return list of pairs covering intervening ranges (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SearchIO.ExonerateIO.exonerate_text-pysrc.html#_get_inter_coords">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_stitch_rows"></a><span class="summary-sig-name">_stitch_rows</span>(<span class="summary-sig-arg">raw_rows</span>)</span><br />
      Stitches together the parsed alignment rows and returns them in a list (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SearchIO.ExonerateIO.exonerate_text-pysrc.html#_stitch_rows">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_get_row_dict"></a><span class="summary-sig-name">_get_row_dict</span>(<span class="summary-sig-arg">row_len</span>,
        <span class="summary-sig-arg">model</span>)</span><br />
      Return a dictionary of row indices for parsing alignment blocks (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SearchIO.ExonerateIO.exonerate_text-pysrc.html#_get_row_dict">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_get_blocks"></a><span class="summary-sig-name">_get_blocks</span>(<span class="summary-sig-arg">rows</span>,
        <span class="summary-sig-arg">coords</span>,
        <span class="summary-sig-arg">idx</span>)</span><br />
      Return a list of dictionaries of sequences split by the coordinates (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SearchIO.ExonerateIO.exonerate_text-pysrc.html#_get_blocks">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_get_scodon_moves"></a><span class="summary-sig-name">_get_scodon_moves</span>(<span class="summary-sig-arg">tmp_seq_blocks</span>)</span><br />
      Get a dictionary of split codon locations relative to each fragment end (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SearchIO.ExonerateIO.exonerate_text-pysrc.html#_get_scodon_moves">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_clean_blocks"></a><span class="summary-sig-name">_clean_blocks</span>(<span class="summary-sig-arg">tmp_seq_blocks</span>)</span><br />
      Remove curly braces (split codon markers) from the given sequences (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SearchIO.ExonerateIO.exonerate_text-pysrc.html#_clean_blocks">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_comp_intron_lens"></a><span class="summary-sig-name">_comp_intron_lens</span>(<span class="summary-sig-arg">seq_type</span>,
        <span class="summary-sig-arg">inter_blocks</span>,
        <span class="summary-sig-arg">raw_inter_lens</span>)</span><br />
      Return the length of introns between fragments (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SearchIO.ExonerateIO.exonerate_text-pysrc.html#_comp_intron_lens">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_comp_coords"></a><span class="summary-sig-name">_comp_coords</span>(<span class="summary-sig-arg">hsp</span>,
        <span class="summary-sig-arg">seq_type</span>,
        <span class="summary-sig-arg">inter_lens</span>)</span><br />
      Fill the block coordinates of the given hsp dictionary (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SearchIO.ExonerateIO.exonerate_text-pysrc.html#_comp_coords">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_comp_split_codons"></a><span class="summary-sig-name">_comp_split_codons</span>(<span class="summary-sig-arg">hsp</span>,
        <span class="summary-sig-arg">seq_type</span>,
        <span class="summary-sig-arg">scodon_moves</span>)</span><br />
      Compute positions of split codons, store in given HSP dictionary (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.SearchIO.ExonerateIO.exonerate_text-pysrc.html#_comp_split_codons">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
</table>
<!-- ==================== VARIABLES ==================== -->
<a name="section-Variables"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Variables</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Variables"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="_RE_ALN_ROW"></a><span class="summary-name">_RE_ALN_ROW</span> = <code title="re.compile(r'\s*\d+\s+: (.*) :\s+\d+')">re.compile(r'\s<code class="re-op">*</code>\d<code class="re-op">+</code>\s<code class="re-op">+</code>: <code class="re-group">(</code>.<code class="re-op">*</code><code class="re-group">)</code> :\s<code class="re-op">+</code>\d<code class="re-op">+</code>')</code>
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_RE_EXON" class="summary-name" onclick="show_private();">_RE_EXON</a> = <code title="re.compile(r'[atgc ]{2,}?(?:(?:[&lt;&gt;]+ \w+ Intron \d+ [&lt;&gt;]+)|(?:\.+))[at\
gc ]{2,}?')">re.compile(r'<code class="re-group">[</code>atgc <code class="re-group">]</code><code class="re-op">{2,}?</code><code class="re-group">(?:</code><code class="re-group">(?:</code><code class="re-group">[</code>&lt;&gt;<code class="re-group">]</code><code class="re-op">+</code> \w<code class="re-op">+</code> Intron \d<code class="re-op">+</code><code class="variable-ellipsis">...</code></code>
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.SearchIO.ExonerateIO.exonerate_text-module.html#_RE_EXON_LEN" class="summary-name" onclick="show_private();">_RE_EXON_LEN</a> = <code title="re.compile(r'(?:(\d+) bp // (\d+) bp)|(?:(\d+) bp)')">re.compile(r'<code class="re-group">(?:</code><code class="re-group">(</code>\d<code class="re-op">+</code><code class="re-group">)</code> bp // <code class="re-group">(</code>\d<code class="re-op">+</code><code class="re-group">)</code> bp<code class="re-group">)</code><code class="re-op">|</code><code class="re-group">(?:</code><code class="re-group">(</code>\d<code class="re-op">+</code><code class="re-group">)</code> <code class="variable-ellipsis">...</code></code>
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="_RE_NER"></a><span class="summary-name">_RE_NER</span> = <code title="re.compile(r'--&lt;\s+\d+\s+&gt;--')">re.compile(r'--&lt;\s<code class="re-op">+</code>\d<code class="re-op">+</code>\s<code class="re-op">+</code>&gt;--')</code>
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="_RE_NER_LEN"></a><span class="summary-name">_RE_NER_LEN</span> = <code title="re.compile(r'--&lt;\s+(\d+)\s+&gt;--')">re.compile(r'--&lt;\s<code class="re-op">+</code><code class="re-group">(</code>\d<code class="re-op">+</code><code class="re-group">)</code>\s<code class="re-op">+</code>&gt;--')</code>
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="_RE_SCODON_START"></a><span class="summary-name">_RE_SCODON_START</span> = <code title="re.compile(r'\{(\w{1,2})\}$')">re.compile(r'\{<code class="re-group">(</code>\w<code class="re-op">{1,2}</code><code class="re-group">)</code>\}$')</code>
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="_RE_SCODON_END"></a><span class="summary-name">_RE_SCODON_END</span> = <code title="re.compile(r'^\{(\w{1,2})\}')">re.compile(r'^\{<code class="re-group">(</code>\w<code class="re-op">{1,2}</code><code class="re-group">)</code>\}')</code>
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="__package__"></a><span class="summary-name">__package__</span> = <code title="'Bio.SearchIO.ExonerateIO'"><code class="variable-quote">'</code><code class="variable-string">Bio.SearchIO.ExonerateIO</code><code class="variable-quote">'</code></code>
    </td>
  </tr>
</table>
<!-- ==================== FUNCTION DETAILS ==================== -->
<a name="section-FunctionDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Function Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-FunctionDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="_get_inter_coords"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_get_inter_coords</span>(<span class="sig-arg">coords</span>,
        <span class="sig-arg">strand</span>=<span class="sig-default">1</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.SearchIO.ExonerateIO.exonerate_text-pysrc.html#_get_inter_coords">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Return list of pairs covering intervening ranges (PRIVATE).</p>
<p>From the given pairs of coordinates, returns a list of pairs
covering the intervening ranges.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== VARIABLES DETAILS ==================== -->
<a name="section-VariablesDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Variables Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-VariablesDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="_RE_EXON"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <h3 class="epydoc">_RE_EXON</h3>
  
  <dl class="fields">
  </dl>
  <dl class="fields">
    <dt>Value:</dt>
      <dd><table><tr><td><pre class="variable">
re.compile(r'<code class="re-group">[</code>atgc <code class="re-group">]</code><code class="re-op">{2,}?</code><code class="re-group">(?:</code><code class="re-group">(?:</code><code class="re-group">[</code>&lt;&gt;<code class="re-group">]</code><code class="re-op">+</code> \w<code class="re-op">+</code> Intron \d<code class="re-op">+</code> <code class="re-group">[</code>&lt;&gt;<code class="re-group">]</code><code class="re-op">+</code><code class="re-group">)</code><code class="re-op">|</code><code class="re-group">(?:</code>\.<code class="re-op">+</code><code class="re-group">)</code><code class="re-group">)</code><code class="re-group">[</code>at<span class="variable-linewrap"><img src="crarr.png" alt="\" /></span>
gc <code class="re-group">]</code><code class="re-op">{2,}?</code>')
</pre></td></tr></table>
</dd>
  </dl>
</td></tr></table>
</div>
<a name="_RE_EXON_LEN"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <h3 class="epydoc">_RE_EXON_LEN</h3>
  
  <dl class="fields">
  </dl>
  <dl class="fields">
    <dt>Value:</dt>
      <dd><table><tr><td><pre class="variable">
re.compile(r'<code class="re-group">(?:</code><code class="re-group">(</code>\d<code class="re-op">+</code><code class="re-group">)</code> bp // <code class="re-group">(</code>\d<code class="re-op">+</code><code class="re-group">)</code> bp<code class="re-group">)</code><code class="re-op">|</code><code class="re-group">(?:</code><code class="re-group">(</code>\d<code class="re-op">+</code><code class="re-group">)</code> bp<code class="re-group">)</code>')
</pre></td></tr></table>
</dd>
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Tue Dec 18 15:56:20 2018
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
