<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.PDB.mmcifio</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.PDB-module.html">Package&nbsp;PDB</a> ::
        Module&nbsp;mmcifio
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.PDB.mmcifio-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.PDB.mmcifio-module.html">Module Bio.PDB.mmcifio</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2017 Joe Greener. All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># This file is part of the Biopython distribution and governed by your</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># choice of the "Biopython License Agreement" or the "BSD 3-Clause License".</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-comment"># Please see the LICENSE file that should have been included as part of this</tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-comment"># package.</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-docstring">"""Write an mmCIF file.</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-docstring">See https://www.iucr.org/resources/cif/spec/version1.1/cifsyntax for syntax.</tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">re</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">collections</tt> <tt class="py-keyword">import</tt> <tt class="py-name">defaultdict</tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-1', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">basestring</tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-2" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-2', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-3" class="py-name" targets="Package Bio.PDB=Bio.PDB-module.html"><a title="Bio.PDB" class="py-name" href="#" onclick="return doclink('link-3', 'PDB', 'link-3');">PDB</a></tt><tt class="py-op">.</tt><tt id="link-4" class="py-name" targets="Module Bio.PDB.StructureBuilder=Bio.PDB.StructureBuilder-module.html,Class Bio.PDB.StructureBuilder.StructureBuilder=Bio.PDB.StructureBuilder.StructureBuilder-class.html"><a title="Bio.PDB.StructureBuilder
Bio.PDB.StructureBuilder.StructureBuilder" class="py-name" href="#" onclick="return doclink('link-4', 'StructureBuilder', 'link-4');">StructureBuilder</a></tt> <tt class="py-keyword">import</tt> <tt id="link-5" class="py-name"><a title="Bio.PDB.StructureBuilder
Bio.PDB.StructureBuilder.StructureBuilder" class="py-name" href="#" onclick="return doclink('link-5', 'StructureBuilder', 'link-4');">StructureBuilder</a></tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-6" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-6', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-7" class="py-name"><a title="Bio.PDB" class="py-name" href="#" onclick="return doclink('link-7', 'PDB', 'link-3');">PDB</a></tt><tt class="py-op">.</tt><tt id="link-8" class="py-name" targets="Class Bio.PDB.PDBIO'.PDBIO=Bio.PDB.PDBIO%27.PDBIO-class.html"><a title="Bio.PDB.PDBIO'.PDBIO" class="py-name" href="#" onclick="return doclink('link-8', 'PDBIO', 'link-8');">PDBIO</a></tt> <tt class="py-keyword">import</tt> <tt id="link-9" class="py-name" targets="Class Bio.PDB.PDBIO'.Select=Bio.PDB.PDBIO%27.Select-class.html"><a title="Bio.PDB.PDBIO'.Select" class="py-name" href="#" onclick="return doclink('link-9', 'Select', 'link-9');">Select</a></tt><tt class="py-op">,</tt> <tt id="link-10" class="py-name" targets="Class Bio.PDB.PDBIO.StructureIO=Bio.PDB.PDBIO.StructureIO-class.html"><a title="Bio.PDB.PDBIO.StructureIO" class="py-name" href="#" onclick="return doclink('link-10', 'StructureIO', 'link-10');">StructureIO</a></tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-comment"># If certain entries should have a certain order of keys, that is specified here</tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt id="link-11" class="py-name" targets="Variable Bio.PDB.mmcifio.mmcif_order=Bio.PDB.mmcifio-module.html#mmcif_order"><a title="Bio.PDB.mmcifio.mmcif_order" class="py-name" href="#" onclick="return doclink('link-11', 'mmcif_order', 'link-11');">mmcif_order</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line">    <tt class="py-string">"_atom_site"</tt><tt class="py-op">:</tt> <tt class="py-op">[</tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line">        <tt class="py-string">"group_PDB"</tt><tt class="py-op">,</tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line">        <tt class="py-string">"id"</tt><tt class="py-op">,</tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line">        <tt class="py-string">"type_symbol"</tt><tt class="py-op">,</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line">        <tt class="py-string">"label_atom_id"</tt><tt class="py-op">,</tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line">        <tt class="py-string">"label_alt_id"</tt><tt class="py-op">,</tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line">        <tt class="py-string">"label_comp_id"</tt><tt class="py-op">,</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line">        <tt class="py-string">"label_asym_id"</tt><tt class="py-op">,</tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line">        <tt class="py-string">"label_entity_id"</tt><tt class="py-op">,</tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line">        <tt class="py-string">"label_seq_id"</tt><tt class="py-op">,</tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line">        <tt class="py-string">"pdbx_PDB_ins_code"</tt><tt class="py-op">,</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line">        <tt class="py-string">"Cartn_x"</tt><tt class="py-op">,</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line">        <tt class="py-string">"Cartn_y"</tt><tt class="py-op">,</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line">        <tt class="py-string">"Cartn_z"</tt><tt class="py-op">,</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line">        <tt class="py-string">"occupancy"</tt><tt class="py-op">,</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line">        <tt class="py-string">"B_iso_or_equiv"</tt><tt class="py-op">,</tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line">        <tt class="py-string">"pdbx_formal_charge"</tt><tt class="py-op">,</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line">        <tt class="py-string">"auth_seq_id"</tt><tt class="py-op">,</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line">        <tt class="py-string">"auth_comp_id"</tt><tt class="py-op">,</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line">        <tt class="py-string">"auth_asym_id"</tt><tt class="py-op">,</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line">        <tt class="py-string">"auth_atom_id"</tt><tt class="py-op">,</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line">        <tt class="py-string">"pdbx_PDB_model_num"</tt><tt class="py-op">,</tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line">    <tt class="py-op">]</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line"><tt class="py-op">}</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line"> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line"> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line"><tt id="link-12" class="py-name" targets="Variable Bio.PDB.PDBIO'._select=Bio.PDB.PDBIO%27-module.html#_select,Variable Bio.PDB.mmcifio._select=Bio.PDB.mmcifio-module.html#_select,Variable Bio.PDB.mmtf.mmtfio._select=Bio.PDB.mmtf.mmtfio-module.html#_select"><a title="Bio.PDB.PDBIO'._select
Bio.PDB.mmcifio._select
Bio.PDB.mmtf.mmtfio._select" class="py-name" href="#" onclick="return doclink('link-12', '_select', 'link-12');">_select</a></tt> <tt class="py-op">=</tt> <tt id="link-13" class="py-name"><a title="Bio.PDB.PDBIO'.Select" class="py-name" href="#" onclick="return doclink('link-13', 'Select', 'link-9');">Select</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line"> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line"> </tt>
<a name="MMCIFIO"></a><div id="MMCIFIO-def"><a name="L51"></a><tt class="py-lineno"> 51</tt> <a class="py-toggle" href="#" id="MMCIFIO-toggle" onclick="return toggle('MMCIFIO');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.PDB.mmcifio.MMCIFIO-class.html">MMCIFIO</a><tt class="py-op">(</tt><tt class="py-base-class">StructureIO</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MMCIFIO-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="MMCIFIO-expanded"><a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line">    <tt class="py-docstring">"""Write a Structure object or a mmCIF dictionary as a mmCIF file.</tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line"><tt class="py-docstring">    Examples</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line"><tt class="py-docstring">    --------</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.PDB import MMCIFParser</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio.PDB.mmcifio import MMCIFIO</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; parser = MMCIFParser()</tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; structure = parser.get_structure("1a8o", "PDB/1A8O.cif")</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; io=MMCIFIO()</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; io.set_structure(structure)</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; io.save("bio-pdb-mmcifio-out.cif")</tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; import os</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; os.remove("bio-pdb-mmcifio-out.cif")  # tidy up</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line"> </tt>
<a name="MMCIFIO.__init__"></a><div id="MMCIFIO.__init__-def"><a name="L69"></a><tt class="py-lineno"> 69</tt> <a class="py-toggle" href="#" id="MMCIFIO.__init__-toggle" onclick="return toggle('MMCIFIO.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.PDB.mmcifio.MMCIFIO-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MMCIFIO.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MMCIFIO.__init__-expanded"><a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialise."""</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line"> </tt>
<a name="MMCIFIO.set_dict"></a><div id="MMCIFIO.set_dict-def"><a name="L73"></a><tt class="py-lineno"> 73</tt> <a class="py-toggle" href="#" id="MMCIFIO.set_dict-toggle" onclick="return toggle('MMCIFIO.set_dict');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.PDB.mmcifio.MMCIFIO-class.html#set_dict">set_dict</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">dic</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MMCIFIO.set_dict-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MMCIFIO.set_dict-expanded"><a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line">        <tt class="py-docstring">"""Set the mmCIF dictionary to be written out."""</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dic</tt> <tt class="py-op">=</tt> <tt class="py-name">dic</tt> </tt>
</div><a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line"> </tt>
<a name="MMCIFIO.save"></a><div id="MMCIFIO.save-def"><a name="L77"></a><tt class="py-lineno"> 77</tt> <a class="py-toggle" href="#" id="MMCIFIO.save-toggle" onclick="return toggle('MMCIFIO.save');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.PDB.mmcifio.MMCIFIO-class.html#save">save</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">filepath</tt><tt class="py-op">,</tt> <tt class="py-param">select</tt><tt class="py-op">=</tt><tt id="link-14" class="py-name"><a title="Bio.PDB.PDBIO'._select
Bio.PDB.mmcifio._select
Bio.PDB.mmtf.mmtfio._select" class="py-name" href="#" onclick="return doclink('link-14', '_select', 'link-12');">_select</a></tt><tt class="py-op">,</tt> <tt class="py-param">preserve_atom_numbering</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MMCIFIO.save-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MMCIFIO.save-expanded"><a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line">        <tt class="py-docstring">"""Save the structure to a file.</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line"><tt class="py-docstring">        :param filepath: output file</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line"><tt class="py-docstring">        :type filepath: string or filehandle</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line"><tt class="py-docstring">        :param select: selects which entities will be written.</tt> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line"><tt class="py-docstring">        :type select: object</tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line"><tt class="py-docstring">        Typically select is a subclass of L{Select}, it should</tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line"><tt class="py-docstring">        have the following methods:</tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line"><tt class="py-docstring">         - accept_model(model)</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line"><tt class="py-docstring">         - accept_chain(chain)</tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line"><tt class="py-docstring">         - accept_residue(residue)</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line"><tt class="py-docstring">         - accept_atom(atom)</tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line"><tt class="py-docstring">        These methods should return 1 if the entity is to be</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line"><tt class="py-docstring">        written out, 0 otherwise.</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line">        <tt class="py-comment"># Similar to the PDBIO save method, we check if the filepath is a</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line">        <tt class="py-comment"># string for a filepath or an open file handle</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">filepath</tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line">            <tt class="py-name">fp</tt> <tt class="py-op">=</tt> <tt id="link-15" class="py-name" targets="Function Bio.bgzf.open()=Bio.bgzf-module.html#open"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-15', 'open', 'link-15');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">filepath</tt><tt class="py-op">,</tt> <tt class="py-string">"w"</tt><tt class="py-op">)</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line">            <tt class="py-name">close_file</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line">            <tt class="py-name">fp</tt> <tt class="py-op">=</tt> <tt class="py-name">filepath</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line">            <tt class="py-name">close_file</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">        <tt class="py-comment"># Decide whether to save a Structure object or an mmCIF dictionary</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-string">"structure"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-16" class="py-name" targets="Method Bio.PDB.mmcifio.MMCIFIO._save_structure()=Bio.PDB.mmcifio.MMCIFIO-class.html#_save_structure,Method Bio.PDB.mmtf.mmtfio.MMTFIO._save_structure()=Bio.PDB.mmtf.mmtfio.MMTFIO-class.html#_save_structure"><a title="Bio.PDB.mmcifio.MMCIFIO._save_structure
Bio.PDB.mmtf.mmtfio.MMTFIO._save_structure" class="py-name" href="#" onclick="return doclink('link-16', '_save_structure', 'link-16');">_save_structure</a></tt><tt class="py-op">(</tt><tt class="py-name">fp</tt><tt class="py-op">,</tt> <tt class="py-name">select</tt><tt class="py-op">,</tt> <tt class="py-name">preserve_atom_numbering</tt><tt class="py-op">)</tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-string">"dic"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-17" class="py-name" targets="Method Bio.PDB.mmcifio.MMCIFIO._save_dict()=Bio.PDB.mmcifio.MMCIFIO-class.html#_save_dict"><a title="Bio.PDB.mmcifio.MMCIFIO._save_dict" class="py-name" href="#" onclick="return doclink('link-17', '_save_dict', 'link-17');">_save_dict</a></tt><tt class="py-op">(</tt><tt class="py-name">fp</tt><tt class="py-op">)</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">                <tt class="py-string">"Use set_structure or set_dict to set a structure or dictionary to write out"</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">            <tt class="py-op">)</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">close_file</tt><tt class="py-op">:</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">            <tt class="py-name">fp</tt><tt class="py-op">.</tt><tt id="link-18" class="py-name" targets="Method Bio.File._IndexedSeqFileDict.close()=Bio.File._IndexedSeqFileDict-class.html#close,Method Bio.File._SQLiteManySeqFilesDict.close()=Bio.File._SQLiteManySeqFilesDict-class.html#close,Method Bio.SeqIO.SffIO._AddTellHandle.close()=Bio.SeqIO.SffIO._AddTellHandle-class.html#close,Method Bio.bgzf.BgzfReader.close()=Bio.bgzf.BgzfReader-class.html#close,Method Bio.bgzf.BgzfWriter.close()=Bio.bgzf.BgzfWriter-class.html#close,Method BioSQL.BioSeqDatabase.Adaptor.close()=BioSQL.BioSeqDatabase.Adaptor-class.html#close,Method BioSQL.BioSeqDatabase.DBServer.close()=BioSQL.BioSeqDatabase.DBServer-class.html#close"><a title="Bio.File._IndexedSeqFileDict.close
Bio.File._SQLiteManySeqFilesDict.close
Bio.SeqIO.SffIO._AddTellHandle.close
Bio.bgzf.BgzfReader.close
Bio.bgzf.BgzfWriter.close
BioSQL.BioSeqDatabase.Adaptor.close
BioSQL.BioSeqDatabase.DBServer.close" class="py-name" href="#" onclick="return doclink('link-18', 'close', 'link-18');">close</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line"> </tt>
<a name="MMCIFIO._save_dict"></a><div id="MMCIFIO._save_dict-def"><a name="L117"></a><tt class="py-lineno">117</tt> <a class="py-toggle" href="#" id="MMCIFIO._save_dict-toggle" onclick="return toggle('MMCIFIO._save_dict');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.PDB.mmcifio.MMCIFIO-class.html#_save_dict">_save_dict</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">out_file</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MMCIFIO._save_dict-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MMCIFIO._save_dict-expanded"><a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line">        <tt class="py-comment"># Form dictionary where key is first part of mmCIF key and value is list</tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line">        <tt class="py-comment"># of corresponding second parts</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">        <tt class="py-name">key_lists</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dic</tt><tt class="py-op">:</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">key</tt> <tt class="py-op">==</tt> <tt class="py-string">"data_"</tt><tt class="py-op">:</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line">                <tt class="py-name">data_val</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dic</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line">                <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-19" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-19', 'split', 'link-19');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">r"\."</tt><tt class="py-op">,</tt> <tt class="py-name">key</tt><tt class="py-op">)</tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">s</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt class="py-name">key_lists</tt><tt class="py-op">:</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line">                        <tt class="py-name">key_lists</tt><tt class="py-op">[</tt><tt class="py-name">s</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-20" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-20', 'append', 'link-20');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line">                        <tt class="py-name">key_lists</tt><tt class="py-op">[</tt><tt class="py-name">s</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">s</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid key in mmCIF dictionary: "</tt> <tt class="py-op">+</tt> <tt class="py-name">key</tt><tt class="py-op">)</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line"> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line">        <tt class="py-comment"># Re-order lists if an order has been specified</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line">        <tt class="py-comment"># Not all elements from the specified order are necessarily present</tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">key_list</tt> <tt class="py-keyword">in</tt> <tt class="py-name">key_lists</tt><tt class="py-op">.</tt><tt id="link-21" class="py-name" targets="Method Bio.Align.substitution_matrices.Array.items()=Bio.Align.substitution_matrices.Array-class.html#items,Method Bio.Crystal.Crystal.items()=Bio.Crystal.Crystal-class.html#items,Method Bio.File._IndexedSeqFileDict.items()=Bio.File._IndexedSeqFileDict-class.html#items,Method Bio.Phylo.PhyloXML.Events.items()=Bio.Phylo.PhyloXML.Events-class.html#items,Variable Bio.SearchIO._model.query.QueryResult.items=Bio.SearchIO._model.query.QueryResult-class.html#items,Method BioSQL.BioSeqDatabase.BioSeqDatabase.items()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#items,Method BioSQL.BioSeqDatabase.DBServer.items()=BioSQL.BioSeqDatabase.DBServer-class.html#items"><a title="Bio.Align.substitution_matrices.Array.items
Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-21', 'items', 'link-21');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt id="link-22" class="py-name"><a title="Bio.PDB.mmcifio.mmcif_order" class="py-name" href="#" onclick="return doclink('link-22', 'mmcif_order', 'link-11');">mmcif_order</a></tt><tt class="py-op">:</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line">                <tt class="py-name">inds</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt id="link-23" class="py-name" targets="Variable Bio.PDB.Polypeptide.i=Bio.PDB.Polypeptide-module.html#i,Variable Bio.SeqIO.SffIO.i=Bio.SeqIO.SffIO-module.html#i"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-23', 'i', 'link-23');">i</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">key_list</tt><tt class="py-op">:</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line">                    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line">                        <tt class="py-name">inds</tt><tt class="py-op">.</tt><tt id="link-24" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-24', 'append', 'link-20');">append</a></tt><tt class="py-op">(</tt><tt id="link-25" class="py-name"><a title="Bio.PDB.mmcifio.mmcif_order" class="py-name" href="#" onclick="return doclink('link-25', 'mmcif_order', 'link-11');">mmcif_order</a></tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-26" class="py-name" targets="Method Bio.Crystal.Chain.index()=Bio.Crystal.Chain-class.html#index,Method Bio.SCOP.Raf.SeqMap.index()=Bio.SCOP.Raf.SeqMap-class.html#index,Method Bio.SearchIO._model.hit.Hit.index()=Bio.SearchIO._model.hit.Hit-class.html#index,Method Bio.SearchIO._model.query.QueryResult.index()=Bio.SearchIO._model.query.QueryResult-class.html#index,Function Bio.SearchIO.index()=Bio.SearchIO-module.html#index,Method Bio.Seq.MutableSeq.index()=Bio.Seq.MutableSeq-class.html#index,Method Bio.Seq.Seq.index()=Bio.Seq.Seq-class.html#index,Function Bio.SeqIO.index()=Bio.SeqIO-module.html#index"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.Seq.Seq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-26', 'index', 'link-26');">index</a></tt><tt class="py-op">(</tt><tt id="link-27" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-27', 'i', 'link-23');">i</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line">                    <tt class="py-comment"># Unrecognised key - add at end</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line">                    <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line">                        <tt class="py-name">inds</tt><tt class="py-op">.</tt><tt id="link-28" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-28', 'append', 'link-20');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-29" class="py-name"><a title="Bio.PDB.mmcifio.mmcif_order" class="py-name" href="#" onclick="return doclink('link-29', 'mmcif_order', 'link-11');">mmcif_order</a></tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line">                <tt class="py-name">key_lists</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">k</tt> <tt class="py-keyword">for</tt> <tt class="py-name">_</tt><tt class="py-op">,</tt> <tt class="py-name">k</tt> <tt class="py-keyword">in</tt> <tt class="py-name">sorted</tt><tt class="py-op">(</tt><tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">inds</tt><tt class="py-op">,</tt> <tt class="py-name">key_list</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line"> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line">        <tt class="py-comment"># Write out top data_ line</tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">data_val</tt><tt class="py-op">:</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line">            <tt class="py-name">out_file</tt><tt class="py-op">.</tt><tt id="link-30" class="py-name" targets="Function Bio.AlignIO.write()=Bio.AlignIO-module.html#write,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.write()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#write,Method Bio.Phylo.CDAOIO.Writer.write()=Bio.Phylo.CDAOIO.Writer-class.html#write,Function Bio.Phylo.CDAOIO.write()=Bio.Phylo.CDAOIO-module.html#write,Method Bio.Phylo.NeXMLIO.Writer.write()=Bio.Phylo.NeXMLIO.Writer-class.html#write,Function Bio.Phylo.NeXMLIO.write()=Bio.Phylo.NeXMLIO-module.html#write,Method Bio.Phylo.NewickIO.Writer.write()=Bio.Phylo.NewickIO.Writer-class.html#write,Function Bio.Phylo.NewickIO.write()=Bio.Phylo.NewickIO-module.html#write,Function Bio.Phylo.NexusIO.write()=Bio.Phylo.NexusIO-module.html#write,Method Bio.Phylo.PhyloXMLIO.Writer.write()=Bio.Phylo.PhyloXMLIO.Writer-class.html#write,Function Bio.Phylo.PhyloXMLIO.write()=Bio.Phylo.PhyloXMLIO-module.html#write,Function Bio.Phylo._io.write()=Bio.Phylo._io-module.html#write,Function Bio.SearchIO.write()=Bio.SearchIO-module.html#write,Function Bio.SeqIO.write()=Bio.SeqIO-module.html#write,Method Bio.bgzf.BgzfWriter.write()=Bio.bgzf.BgzfWriter-class.html#write,Function Bio.motifs.clusterbuster.write()=Bio.motifs.clusterbuster-module.html#write,Function Bio.motifs.jaspar.write()=Bio.motifs.jaspar-module.html#write,Function Bio.motifs.pfm.write()=Bio.motifs.pfm-module.html#write,Function Bio.motifs.transfac.write()=Bio.motifs.transfac-module.html#write,Function Bio.motifs.write()=Bio.motifs-module.html#write,Method Bio.phenotype.phen_micro.JsonWriter.write()=Bio.phenotype.phen_micro.JsonWriter-class.html#write,Function Bio.phenotype.write()=Bio.phenotype-module.html#write"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.clusterbuster.write
Bio.motifs.jaspar.write
Bio.motifs.pfm.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-30', 'write', 'link-30');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"data_"</tt> <tt class="py-op">+</tt> <tt class="py-name">data_val</tt> <tt class="py-op">+</tt> <tt class="py-string">"\n#\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line"> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">key_list</tt> <tt class="py-keyword">in</tt> <tt class="py-name">key_lists</tt><tt class="py-op">.</tt><tt id="link-31" class="py-name"><a title="Bio.Align.substitution_matrices.Array.items
Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-31', 'items', 'link-21');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line">            <tt class="py-comment"># Pick a sample mmCIF value, which can be a list or a single value</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">            <tt class="py-name">sample_val</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dic</tt><tt class="py-op">[</tt><tt class="py-name">key</tt> <tt class="py-op">+</tt> <tt class="py-string">"."</tt> <tt class="py-op">+</tt> <tt class="py-name">key_list</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line">            <tt class="py-name">n_vals</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">sample_val</tt><tt class="py-op">)</tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">            <tt class="py-comment"># Check the mmCIF dictionary has consistent list sizes</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt id="link-32" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-32', 'i', 'link-23');">i</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">key_list</tt><tt class="py-op">:</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">                <tt class="py-name">val</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dic</tt><tt class="py-op">[</tt><tt class="py-name">key</tt> <tt class="py-op">+</tt> <tt class="py-string">"."</tt> <tt class="py-op">+</tt> <tt id="link-33" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-33', 'i', 'link-23');">i</a></tt><tt class="py-op">]</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-op">(</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">                    <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">sample_val</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">                    <tt class="py-keyword">and</tt> <tt class="py-op">(</tt><tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">val</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">val</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">n_vals</tt><tt class="py-op">)</tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">                <tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-op">(</tt><tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">sample_val</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">val</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line">                        <tt class="py-string">"Inconsistent list sizes in mmCIF dictionary: "</tt> <tt class="py-op">+</tt> <tt class="py-name">key</tt> <tt class="py-op">+</tt> <tt class="py-string">"."</tt> <tt class="py-op">+</tt> <tt id="link-34" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-34', 'i', 'link-23');">i</a></tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">                    <tt class="py-op">)</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line">            <tt class="py-comment"># If the value is a single value, write as key-value pairs</tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">sample_val</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-op">(</tt> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">                <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">sample_val</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">sample_val</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">            <tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line">                <tt class="py-name">m</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line">                <tt class="py-comment"># Find the maximum key length</tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt id="link-35" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-35', 'i', 'link-23');">i</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">key_list</tt><tt class="py-op">:</tt> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-36" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-36', 'i', 'link-23');">i</a></tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-name">m</tt><tt class="py-op">:</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line">                        <tt class="py-name">m</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-37" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-37', 'i', 'link-23');">i</a></tt><tt class="py-op">)</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt id="link-38" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-38', 'i', 'link-23');">i</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">key_list</tt><tt class="py-op">:</tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line">                    <tt class="py-comment"># If the value is a single item list, just take the value</tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">sample_val</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line">                        <tt class="py-name">value_no_list</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dic</tt><tt class="py-op">[</tt><tt class="py-name">key</tt> <tt class="py-op">+</tt> <tt class="py-string">"."</tt> <tt class="py-op">+</tt> <tt id="link-39" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-39', 'i', 'link-23');">i</a></tt><tt class="py-op">]</tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line">                        <tt class="py-name">value_no_list</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dic</tt><tt class="py-op">[</tt><tt class="py-name">key</tt> <tt class="py-op">+</tt> <tt class="py-string">"."</tt> <tt class="py-op">+</tt> <tt id="link-40" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-40', 'i', 'link-23');">i</a></tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line">                    <tt class="py-name">out_file</tt><tt class="py-op">.</tt><tt id="link-41" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.clusterbuster.write
Bio.motifs.jaspar.write
Bio.motifs.pfm.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-41', 'write', 'link-30');">write</a></tt><tt class="py-op">(</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">                        <tt class="py-string">"{k: &lt;{width}}"</tt><tt class="py-op">.</tt><tt id="link-42" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.format()=Bio.Align.MultipleSeqAlignment-class.html#format,Method Bio.Align.PairwiseAlignment.format()=Bio.Align.PairwiseAlignment-class.html#format,Method Bio.Phylo.BaseTree.Tree.format()=Bio.Phylo.BaseTree.Tree-class.html#format,Method Bio.Restriction.Restriction.RestrictionBatch.format()=Bio.Restriction.Restriction.RestrictionBatch-class.html#format,Method Bio.SeqRecord.SeqRecord.format()=Bio.SeqRecord.SeqRecord-class.html#format,Method Bio.SubsMat.SeqMat.format()=Bio.SubsMat.SeqMat-class.html#format,Method Bio.motifs.Motif.format()=Bio.motifs.Motif-class.html#format"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-42', 'format', 'link-42');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">k</tt><tt class="py-op">=</tt><tt class="py-name">key</tt> <tt class="py-op">+</tt> <tt class="py-string">"."</tt> <tt class="py-op">+</tt> <tt id="link-43" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-43', 'i', 'link-23');">i</a></tt><tt class="py-op">,</tt> <tt id="link-44" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.width=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#width,Method Bio.Phylo.PhyloXMLIO.Writer.width()=Bio.Phylo.PhyloXMLIO.Writer-class.html#width"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.width
Bio.Phylo.PhyloXMLIO.Writer.width" class="py-name" href="#" onclick="return doclink('link-44', 'width', 'link-44');">width</a></tt><tt class="py-op">=</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-name">m</tt> <tt class="py-op">+</tt> <tt class="py-number">4</tt><tt class="py-op">)</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">                        <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-45" class="py-name" targets="Method Bio.PDB.mmcifio.MMCIFIO._format_mmcif_col()=Bio.PDB.mmcifio.MMCIFIO-class.html#_format_mmcif_col"><a title="Bio.PDB.mmcifio.MMCIFIO._format_mmcif_col" class="py-name" href="#" onclick="return doclink('link-45', '_format_mmcif_col', 'link-45');">_format_mmcif_col</a></tt><tt class="py-op">(</tt><tt class="py-name">value_no_list</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">value_no_list</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">                        <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">                    <tt class="py-op">)</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line">            <tt class="py-comment"># If the value is more than one value, write as keys then a value table</tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">sample_val</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">                <tt class="py-name">out_file</tt><tt class="py-op">.</tt><tt id="link-46" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.clusterbuster.write
Bio.motifs.jaspar.write
Bio.motifs.pfm.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-46', 'write', 'link-30');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"loop_\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line">                <tt class="py-name">col_widths</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line">                <tt class="py-comment"># Write keys and find max widths for each set of values</tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt id="link-47" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-47', 'i', 'link-23');">i</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">key_list</tt><tt class="py-op">:</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">                    <tt class="py-name">out_file</tt><tt class="py-op">.</tt><tt id="link-48" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.clusterbuster.write
Bio.motifs.jaspar.write
Bio.motifs.pfm.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-48', 'write', 'link-30');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">key</tt> <tt class="py-op">+</tt> <tt class="py-string">"."</tt> <tt class="py-op">+</tt> <tt id="link-49" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-49', 'i', 'link-23');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">                    <tt class="py-name">col_widths</tt><tt class="py-op">[</tt><tt id="link-50" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-50', 'i', 'link-23');">i</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-name">val</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dic</tt><tt class="py-op">[</tt><tt class="py-name">key</tt> <tt class="py-op">+</tt> <tt class="py-string">"."</tt> <tt class="py-op">+</tt> <tt id="link-51" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-51', 'i', 'link-23');">i</a></tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">                        <tt class="py-name">len_val</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">val</tt><tt class="py-op">)</tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line">                        <tt class="py-comment"># If the value requires quoting it will add 2 characters</tt> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-52" class="py-name" targets="Method Bio.PDB.mmcifio.MMCIFIO._requires_quote()=Bio.PDB.mmcifio.MMCIFIO-class.html#_requires_quote"><a title="Bio.PDB.mmcifio.MMCIFIO._requires_quote" class="py-name" href="#" onclick="return doclink('link-52', '_requires_quote', 'link-52');">_requires_quote</a></tt><tt class="py-op">(</tt><tt class="py-name">val</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-53" class="py-name" targets="Method Bio.PDB.mmcifio.MMCIFIO._requires_newline()=Bio.PDB.mmcifio.MMCIFIO-class.html#_requires_newline"><a title="Bio.PDB.mmcifio.MMCIFIO._requires_newline" class="py-name" href="#" onclick="return doclink('link-53', '_requires_newline', 'link-53');">_requires_newline</a></tt><tt class="py-op">(</tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line">                            <tt class="py-name">val</tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line">                        <tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line">                            <tt class="py-name">len_val</tt> <tt class="py-op">+=</tt> <tt class="py-number">2</tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">len_val</tt> <tt class="py-op">&gt;</tt> <tt class="py-name">col_widths</tt><tt class="py-op">[</tt><tt id="link-54" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-54', 'i', 'link-23');">i</a></tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line">                            <tt class="py-name">col_widths</tt><tt class="py-op">[</tt><tt id="link-55" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-55', 'i', 'link-23');">i</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">len_val</tt> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line">                <tt class="py-comment"># Technically the max of the sum of the column widths is 2048</tt> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line"> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line">                <tt class="py-comment"># Write the values as rows</tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt id="link-56" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-56', 'i', 'link-23');">i</a></tt> <tt class="py-keyword">in</tt> <tt id="link-57" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.range()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#range,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.range()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#range,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Track.Track.range()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#range"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-57', 'range', 'link-57');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">n_vals</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-name">col</tt> <tt class="py-keyword">in</tt> <tt class="py-name">key_list</tt><tt class="py-op">:</tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line">                        <tt class="py-name">out_file</tt><tt class="py-op">.</tt><tt id="link-58" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.clusterbuster.write
Bio.motifs.jaspar.write
Bio.motifs.pfm.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-58', 'write', 'link-30');">write</a></tt><tt class="py-op">(</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line">                            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-59" class="py-name"><a title="Bio.PDB.mmcifio.MMCIFIO._format_mmcif_col" class="py-name" href="#" onclick="return doclink('link-59', '_format_mmcif_col', 'link-45');">_format_mmcif_col</a></tt><tt class="py-op">(</tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line">                                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dic</tt><tt class="py-op">[</tt><tt class="py-name">key</tt> <tt class="py-op">+</tt> <tt class="py-string">"."</tt> <tt class="py-op">+</tt> <tt class="py-name">col</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt id="link-60" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-60', 'i', 'link-23');">i</a></tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">col_widths</tt><tt class="py-op">[</tt><tt class="py-name">col</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line">                            <tt class="py-op">)</tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line">                        <tt class="py-op">)</tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line">                    <tt class="py-name">out_file</tt><tt class="py-op">.</tt><tt id="link-61" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.clusterbuster.write
Bio.motifs.jaspar.write
Bio.motifs.pfm.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-61', 'write', 'link-30');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line">                    <tt class="py-string">"Invalid type in mmCIF dictionary: "</tt> <tt class="py-op">+</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-62" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.type()=Bio.Phylo.PhyloXMLIO.Writer-class.html#type"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-62', 'type', 'link-62');">type</a></tt><tt class="py-op">(</tt><tt class="py-name">sample_val</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line">                <tt class="py-op">)</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line">            <tt class="py-name">out_file</tt><tt class="py-op">.</tt><tt id="link-63" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.clusterbuster.write
Bio.motifs.jaspar.write
Bio.motifs.pfm.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-63', 'write', 'link-30');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"#\n"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line"> </tt>
<a name="MMCIFIO._format_mmcif_col"></a><div id="MMCIFIO._format_mmcif_col-def"><a name="L219"></a><tt class="py-lineno">219</tt> <a class="py-toggle" href="#" id="MMCIFIO._format_mmcif_col-toggle" onclick="return toggle('MMCIFIO._format_mmcif_col');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.PDB.mmcifio.MMCIFIO-class.html#_format_mmcif_col">_format_mmcif_col</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">val</tt><tt class="py-op">,</tt> <tt class="py-param">col_width</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MMCIFIO._format_mmcif_col-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MMCIFIO._format_mmcif_col-expanded"><a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line">        <tt class="py-comment"># Format a mmCIF data value by enclosing with quotes or semicolon lines</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line">        <tt class="py-comment"># where appropriate. See</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line">        <tt class="py-comment"># https://www.iucr.org/resources/cif/spec/version1.1/cifsyntax for</tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line">        <tt class="py-comment"># syntax.</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line"> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line">        <tt class="py-comment"># If there is a newline or quotes cannot be contained, use semicolon</tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line">        <tt class="py-comment"># and newline construct</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-64" class="py-name"><a title="Bio.PDB.mmcifio.MMCIFIO._requires_newline" class="py-name" href="#" onclick="return doclink('link-64', '_requires_newline', 'link-53');">_requires_newline</a></tt><tt class="py-op">(</tt><tt class="py-name">val</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-string">"\n;"</tt> <tt class="py-op">+</tt> <tt class="py-name">val</tt> <tt class="py-op">+</tt> <tt class="py-string">"\n;\n"</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-65" class="py-name"><a title="Bio.PDB.mmcifio.MMCIFIO._requires_quote" class="py-name" href="#" onclick="return doclink('link-65', '_requires_quote', 'link-52');">_requires_quote</a></tt><tt class="py-op">(</tt><tt class="py-name">val</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line">            <tt class="py-comment"># Choose quote character</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-string">"' "</tt> <tt class="py-keyword">in</tt> <tt class="py-name">val</tt><tt class="py-op">:</tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-string">"{v: &lt;{width}}"</tt><tt class="py-op">.</tt><tt id="link-66" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-66', 'format', 'link-42');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">v</tt><tt class="py-op">=</tt><tt class="py-string">'"'</tt> <tt class="py-op">+</tt> <tt class="py-name">val</tt> <tt class="py-op">+</tt> <tt class="py-string">'"'</tt><tt class="py-op">,</tt> <tt id="link-67" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.width
Bio.Phylo.PhyloXMLIO.Writer.width" class="py-name" href="#" onclick="return doclink('link-67', 'width', 'link-44');">width</a></tt><tt class="py-op">=</tt><tt class="py-name">col_width</tt><tt class="py-op">)</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-string">"{v: &lt;{width}}"</tt><tt class="py-op">.</tt><tt id="link-68" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-68', 'format', 'link-42');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">v</tt><tt class="py-op">=</tt><tt class="py-string">"'"</tt> <tt class="py-op">+</tt> <tt class="py-name">val</tt> <tt class="py-op">+</tt> <tt class="py-string">"'"</tt><tt class="py-op">,</tt> <tt id="link-69" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.width
Bio.Phylo.PhyloXMLIO.Writer.width" class="py-name" href="#" onclick="return doclink('link-69', 'width', 'link-44');">width</a></tt><tt class="py-op">=</tt><tt class="py-name">col_width</tt><tt class="py-op">)</tt> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line">        <tt class="py-comment"># Safe to not quote</tt> </tt>
<a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line">        <tt class="py-comment"># Numbers must not be quoted</tt> </tt>
<a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-string">"{v: &lt;{width}}"</tt><tt class="py-op">.</tt><tt id="link-70" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-70', 'format', 'link-42');">format</a></tt><tt class="py-op">(</tt><tt class="py-name">v</tt><tt class="py-op">=</tt><tt class="py-name">val</tt><tt class="py-op">,</tt> <tt id="link-71" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.width
Bio.Phylo.PhyloXMLIO.Writer.width" class="py-name" href="#" onclick="return doclink('link-71', 'width', 'link-44');">width</a></tt><tt class="py-op">=</tt><tt class="py-name">col_width</tt><tt class="py-op">)</tt> </tt>
</div><a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line"> </tt>
<a name="MMCIFIO._requires_newline"></a><div id="MMCIFIO._requires_newline-def"><a name="L240"></a><tt class="py-lineno">240</tt> <a class="py-toggle" href="#" id="MMCIFIO._requires_newline-toggle" onclick="return toggle('MMCIFIO._requires_newline');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.PDB.mmcifio.MMCIFIO-class.html#_requires_newline">_requires_newline</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">val</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MMCIFIO._requires_newline-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MMCIFIO._requires_newline-expanded"><a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line">        <tt class="py-comment"># Technically the space can be a tab too</tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"\n"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">val</tt> <tt class="py-keyword">or</tt> <tt class="py-op">(</tt><tt class="py-string">"' "</tt> <tt class="py-keyword">in</tt> <tt class="py-name">val</tt> <tt class="py-keyword">and</tt> <tt class="py-string">'" '</tt> <tt class="py-keyword">in</tt> <tt class="py-name">val</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">True</tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">False</tt> </tt>
</div><a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line"> </tt>
<a name="MMCIFIO._requires_quote"></a><div id="MMCIFIO._requires_quote-def"><a name="L247"></a><tt class="py-lineno">247</tt> <a class="py-toggle" href="#" id="MMCIFIO._requires_quote-toggle" onclick="return toggle('MMCIFIO._requires_quote');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.PDB.mmcifio.MMCIFIO-class.html#_requires_quote">_requires_quote</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">val</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MMCIFIO._requires_quote-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MMCIFIO._requires_quote-expanded"><a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line">        <tt class="py-comment"># Technically the words should be case-insensitive</tt> </tt>
<a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-op">(</tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line">            <tt class="py-string">" "</tt> <tt class="py-keyword">in</tt> <tt class="py-name">val</tt> </tt>
<a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line">            <tt class="py-keyword">or</tt> <tt class="py-string">"'"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">val</tt> </tt>
<a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line">            <tt class="py-keyword">or</tt> <tt class="py-string">'"'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">val</tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line">            <tt class="py-keyword">or</tt> <tt class="py-name">val</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">"_"</tt><tt class="py-op">,</tt> <tt class="py-string">"#"</tt><tt class="py-op">,</tt> <tt class="py-string">"$"</tt><tt class="py-op">,</tt> <tt class="py-string">"["</tt><tt class="py-op">,</tt> <tt class="py-string">"]"</tt><tt class="py-op">,</tt> <tt class="py-string">";"</tt><tt class="py-op">]</tt> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line">            <tt class="py-keyword">or</tt> <tt class="py-name">val</tt><tt class="py-op">.</tt><tt id="link-72" class="py-name" targets="Method Bio.Seq.Seq.startswith()=Bio.Seq.Seq-class.html#startswith"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-72', 'startswith', 'link-72');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"data_"</tt><tt class="py-op">)</tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line">            <tt class="py-keyword">or</tt> <tt class="py-name">val</tt><tt class="py-op">.</tt><tt id="link-73" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-73', 'startswith', 'link-72');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"save_"</tt><tt class="py-op">)</tt> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line">            <tt class="py-keyword">or</tt> <tt class="py-name">val</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">"loop_"</tt><tt class="py-op">,</tt> <tt class="py-string">"stop_"</tt><tt class="py-op">,</tt> <tt class="py-string">"global_"</tt><tt class="py-op">]</tt> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line">        <tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">True</tt> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">False</tt> </tt>
</div><a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line"> </tt>
<a name="MMCIFIO._get_label_asym_id"></a><div id="MMCIFIO._get_label_asym_id-def"><a name="L262"></a><tt class="py-lineno">262</tt> <a class="py-toggle" href="#" id="MMCIFIO._get_label_asym_id-toggle" onclick="return toggle('MMCIFIO._get_label_asym_id');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.PDB.mmcifio.MMCIFIO-class.html#_get_label_asym_id">_get_label_asym_id</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">entity_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MMCIFIO._get_label_asym_id-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MMCIFIO._get_label_asym_id-expanded"><a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line">        <tt class="py-comment"># Convert a positive integer into a chain ID</tt> </tt>
<a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line">        <tt class="py-comment"># Goes A to Z, then AA to ZA, AB to ZB etc</tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line">        <tt class="py-comment"># This is in line with existing mmCIF files</tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line">        <tt class="py-name">div</tt> <tt class="py-op">=</tt> <tt class="py-name">entity_id</tt> </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line">        <tt class="py-name">out</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">div</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line">            <tt class="py-name">mod</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">div</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> <tt class="py-op">%</tt> <tt class="py-number">26</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line">            <tt class="py-name">out</tt> <tt class="py-op">+=</tt> <tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-number">65</tt> <tt class="py-op">+</tt> <tt class="py-name">mod</tt><tt class="py-op">)</tt> </tt>
<a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line">            <tt class="py-name">div</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">div</tt> <tt class="py-op">-</tt> <tt class="py-name">mod</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-number">26</tt><tt class="py-op">)</tt> </tt>
<a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">out</tt> </tt>
</div><a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line"> </tt>
<a name="MMCIFIO._save_structure"></a><div id="MMCIFIO._save_structure-def"><a name="L274"></a><tt class="py-lineno">274</tt> <a class="py-toggle" href="#" id="MMCIFIO._save_structure-toggle" onclick="return toggle('MMCIFIO._save_structure');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.PDB.mmcifio.MMCIFIO-class.html#_save_structure">_save_structure</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">out_file</tt><tt class="py-op">,</tt> <tt class="py-param">select</tt><tt class="py-op">,</tt> <tt class="py-param">preserve_atom_numbering</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MMCIFIO._save_structure-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MMCIFIO._save_structure-expanded"><a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line">        <tt class="py-name">atom_dict</tt> <tt class="py-op">=</tt> <tt class="py-name">defaultdict</tt><tt class="py-op">(</tt><tt class="py-name">list</tt><tt class="py-op">)</tt> </tt>
<a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line"> </tt>
<a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">model</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-74" class="py-name" targets="Variable Bio.SubsMat.MatrixInfo.structure=Bio.SubsMat.MatrixInfo-module.html#structure"><a title="Bio.SubsMat.MatrixInfo.structure" class="py-name" href="#" onclick="return doclink('link-74', 'structure', 'link-74');">structure</a></tt><tt class="py-op">.</tt><tt id="link-75" class="py-name" targets="Method Bio.PDB.Entity.Entity.get_list()=Bio.PDB.Entity.Entity-class.html#get_list"><a title="Bio.PDB.Entity.Entity.get_list" class="py-name" href="#" onclick="return doclink('link-75', 'get_list', 'link-75');">get_list</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">select</tt><tt class="py-op">.</tt><tt id="link-76" class="py-name" targets="Method Bio.PDB.Dice.ChainSelector.accept_model()=Bio.PDB.Dice.ChainSelector-class.html#accept_model,Method Bio.PDB.PDBIO'.Select.accept_model()=Bio.PDB.PDBIO%27.Select-class.html#accept_model"><a title="Bio.PDB.Dice.ChainSelector.accept_model
Bio.PDB.PDBIO'.Select.accept_model" class="py-name" href="#" onclick="return doclink('link-76', 'accept_model', 'link-76');">accept_model</a></tt><tt class="py-op">(</tt><tt class="py-name">model</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line">                <tt class="py-keyword">continue</tt> </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line">            <tt class="py-comment"># mmCIF files with a single model have it specified as model 1</tt> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">model</tt><tt class="py-op">.</tt><tt class="py-name">serial_num</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line">                <tt class="py-name">model_n</tt> <tt class="py-op">=</tt> <tt class="py-string">"1"</tt> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line">                <tt class="py-name">model_n</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">model</tt><tt class="py-op">.</tt><tt class="py-name">serial_num</tt><tt class="py-op">)</tt> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line">            <tt class="py-comment"># This is used to write label_entity_id and label_asym_id and</tt> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line">            <tt class="py-comment"># increments from 1, changing with each molecule</tt> </tt>
<a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line">            <tt class="py-name">entity_id</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">preserve_atom_numbering</tt><tt class="py-op">:</tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line">                <tt class="py-name">atom_number</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">chain</tt> <tt class="py-keyword">in</tt> <tt class="py-name">model</tt><tt class="py-op">.</tt><tt id="link-77" class="py-name"><a title="Bio.PDB.Entity.Entity.get_list" class="py-name" href="#" onclick="return doclink('link-77', 'get_list', 'link-75');">get_list</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">select</tt><tt class="py-op">.</tt><tt id="link-78" class="py-name" targets="Method Bio.PDB.Dice.ChainSelector.accept_chain()=Bio.PDB.Dice.ChainSelector-class.html#accept_chain,Method Bio.PDB.PDBIO'.Select.accept_chain()=Bio.PDB.PDBIO%27.Select-class.html#accept_chain"><a title="Bio.PDB.Dice.ChainSelector.accept_chain
Bio.PDB.PDBIO'.Select.accept_chain" class="py-name" href="#" onclick="return doclink('link-78', 'accept_chain', 'link-78');">accept_chain</a></tt><tt class="py-op">(</tt><tt class="py-name">chain</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line">                    <tt class="py-keyword">continue</tt> </tt>
<a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line">                <tt id="link-79" class="py-name" targets="Variable Bio.FSSP.fssp_rec.align.chain_id=Bio.FSSP.fssp_rec.align-class.html#chain_id"><a title="Bio.FSSP.fssp_rec.align.chain_id" class="py-name" href="#" onclick="return doclink('link-79', 'chain_id', 'link-79');">chain_id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">chain</tt><tt class="py-op">.</tt><tt id="link-80" class="py-name" targets="Method Bio.Nexus.Nodes.Node.get_id()=Bio.Nexus.Nodes.Node-class.html#get_id,Method Bio.PDB.Atom.Atom.get_id()=Bio.PDB.Atom.Atom-class.html#get_id,Method Bio.PDB.Entity.DisorderedEntityWrapper.get_id()=Bio.PDB.Entity.DisorderedEntityWrapper-class.html#get_id,Method Bio.PDB.Entity.Entity.get_id()=Bio.PDB.Entity.Entity-class.html#get_id,Method Bio.PDB.FragmentMapper'.Fragment.get_id()=Bio.PDB.FragmentMapper%27.Fragment-class.html#get_id"><a title="Bio.Nexus.Nodes.Node.get_id
Bio.PDB.Atom.Atom.get_id
Bio.PDB.Entity.DisorderedEntityWrapper.get_id
Bio.PDB.Entity.Entity.get_id
Bio.PDB.FragmentMapper'.Fragment.get_id" class="py-name" href="#" onclick="return doclink('link-80', 'get_id', 'link-80');">get_id</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-81" class="py-name"><a title="Bio.FSSP.fssp_rec.align.chain_id" class="py-name" href="#" onclick="return doclink('link-81', 'chain_id', 'link-79');">chain_id</a></tt> <tt class="py-op">==</tt> <tt class="py-string">" "</tt><tt class="py-op">:</tt> </tt>
<a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line">                    <tt id="link-82" class="py-name"><a title="Bio.FSSP.fssp_rec.align.chain_id" class="py-name" href="#" onclick="return doclink('link-82', 'chain_id', 'link-79');">chain_id</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"."</tt> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line">                <tt class="py-comment"># This is used to write label_seq_id and increments from 1,</tt> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line">                <tt class="py-comment"># remaining blank for hetero residues</tt> </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line">                <tt class="py-name">residue_number</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line">                <tt class="py-name">prev_residue_type</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line">                <tt class="py-name">prev_resname</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">residue</tt> <tt class="py-keyword">in</tt> <tt class="py-name">chain</tt><tt class="py-op">.</tt><tt id="link-83" class="py-name" targets="Method Bio.PDB.Chain.Chain.get_unpacked_list()=Bio.PDB.Chain.Chain-class.html#get_unpacked_list,Method Bio.PDB.Residue.Residue.get_unpacked_list()=Bio.PDB.Residue.Residue-class.html#get_unpacked_list"><a title="Bio.PDB.Chain.Chain.get_unpacked_list
Bio.PDB.Residue.Residue.get_unpacked_list" class="py-name" href="#" onclick="return doclink('link-83', 'get_unpacked_list', 'link-83');">get_unpacked_list</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">select</tt><tt class="py-op">.</tt><tt id="link-84" class="py-name" targets="Method Bio.PDB.Dice.ChainSelector.accept_residue()=Bio.PDB.Dice.ChainSelector-class.html#accept_residue,Method Bio.PDB.PDBIO'.Select.accept_residue()=Bio.PDB.PDBIO%27.Select-class.html#accept_residue"><a title="Bio.PDB.Dice.ChainSelector.accept_residue
Bio.PDB.PDBIO'.Select.accept_residue" class="py-name" href="#" onclick="return doclink('link-84', 'accept_residue', 'link-84');">accept_residue</a></tt><tt class="py-op">(</tt><tt class="py-name">residue</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line">                        <tt class="py-keyword">continue</tt> </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line">                    <tt class="py-name">hetfield</tt><tt class="py-op">,</tt> <tt class="py-name">resseq</tt><tt class="py-op">,</tt> <tt class="py-name">icode</tt> <tt class="py-op">=</tt> <tt class="py-name">residue</tt><tt class="py-op">.</tt><tt id="link-85" class="py-name"><a title="Bio.Nexus.Nodes.Node.get_id
Bio.PDB.Atom.Atom.get_id
Bio.PDB.Entity.DisorderedEntityWrapper.get_id
Bio.PDB.Entity.Entity.get_id
Bio.PDB.FragmentMapper'.Fragment.get_id" class="py-name" href="#" onclick="return doclink('link-85', 'get_id', 'link-80');">get_id</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">hetfield</tt> <tt class="py-op">==</tt> <tt class="py-string">" "</tt><tt class="py-op">:</tt> </tt>
<a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line">                        <tt id="link-86" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.residue_type()=Bio.GenBank._FeatureConsumer-class.html#residue_type,Method Bio.GenBank._RecordConsumer.residue_type()=Bio.GenBank._RecordConsumer-class.html#residue_type"><a title="Bio.GenBank._FeatureConsumer.residue_type
Bio.GenBank._RecordConsumer.residue_type" class="py-name" href="#" onclick="return doclink('link-86', 'residue_type', 'link-86');">residue_type</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"ATOM"</tt> </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line">                        <tt class="py-name">label_seq_id</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">residue_number</tt><tt class="py-op">)</tt> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line">                        <tt class="py-name">residue_number</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line">                        <tt id="link-87" class="py-name"><a title="Bio.GenBank._FeatureConsumer.residue_type
Bio.GenBank._RecordConsumer.residue_type" class="py-name" href="#" onclick="return doclink('link-87', 'residue_type', 'link-86');">residue_type</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"HETATM"</tt> </tt>
<a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line">                        <tt class="py-name">label_seq_id</tt> <tt class="py-op">=</tt> <tt class="py-string">"."</tt> </tt>
<a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line">                    <tt class="py-name">resseq</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">resseq</tt><tt class="py-op">)</tt> </tt>
<a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">icode</tt> <tt class="py-op">==</tt> <tt class="py-string">" "</tt><tt class="py-op">:</tt> </tt>
<a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line">                        <tt class="py-name">icode</tt> <tt class="py-op">=</tt> <tt class="py-string">"?"</tt> </tt>
<a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line">                    <tt class="py-name">resname</tt> <tt class="py-op">=</tt> <tt class="py-name">residue</tt><tt class="py-op">.</tt><tt id="link-88" class="py-name" targets="Method Bio.PDB.Residue.Residue.get_resname()=Bio.PDB.Residue.Residue-class.html#get_resname"><a title="Bio.PDB.Residue.Residue.get_resname" class="py-name" href="#" onclick="return doclink('link-88', 'get_resname', 'link-88');">get_resname</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line">                    <tt class="py-comment"># Check if the molecule changes within the chain</tt> </tt>
<a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line">                    <tt class="py-comment"># This will always increment for the first residue in a</tt> </tt>
<a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line">                    <tt class="py-comment"># chain due to the starting values above</tt> </tt>
<a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt id="link-89" class="py-name"><a title="Bio.GenBank._FeatureConsumer.residue_type
Bio.GenBank._RecordConsumer.residue_type" class="py-name" href="#" onclick="return doclink('link-89', 'residue_type', 'link-86');">residue_type</a></tt> <tt class="py-op">!=</tt> <tt class="py-name">prev_residue_type</tt> <tt class="py-keyword">or</tt> <tt class="py-op">(</tt> </tt>
<a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line">                        <tt id="link-90" class="py-name"><a title="Bio.GenBank._FeatureConsumer.residue_type
Bio.GenBank._RecordConsumer.residue_type" class="py-name" href="#" onclick="return doclink('link-90', 'residue_type', 'link-86');">residue_type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"HETATM"</tt> <tt class="py-keyword">and</tt> <tt class="py-name">resname</tt> <tt class="py-op">!=</tt> <tt class="py-name">prev_resname</tt> </tt>
<a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line">                    <tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line">                        <tt class="py-name">entity_id</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line">                    <tt class="py-name">prev_residue_type</tt> <tt class="py-op">=</tt> <tt id="link-91" class="py-name"><a title="Bio.GenBank._FeatureConsumer.residue_type
Bio.GenBank._RecordConsumer.residue_type" class="py-name" href="#" onclick="return doclink('link-91', 'residue_type', 'link-86');">residue_type</a></tt> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line">                    <tt class="py-name">prev_resname</tt> <tt class="py-op">=</tt> <tt class="py-name">resname</tt> </tt>
<a name="L325"></a><tt class="py-lineno">325</tt>  <tt class="py-line">                    <tt class="py-name">label_asym_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-92" class="py-name" targets="Method Bio.PDB.mmcifio.MMCIFIO._get_label_asym_id()=Bio.PDB.mmcifio.MMCIFIO-class.html#_get_label_asym_id"><a title="Bio.PDB.mmcifio.MMCIFIO._get_label_asym_id" class="py-name" href="#" onclick="return doclink('link-92', '_get_label_asym_id', 'link-92');">_get_label_asym_id</a></tt><tt class="py-op">(</tt><tt class="py-name">entity_id</tt><tt class="py-op">)</tt> </tt>
<a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-name">atom</tt> <tt class="py-keyword">in</tt> <tt class="py-name">residue</tt><tt class="py-op">.</tt><tt id="link-93" class="py-name"><a title="Bio.PDB.Chain.Chain.get_unpacked_list
Bio.PDB.Residue.Residue.get_unpacked_list" class="py-name" href="#" onclick="return doclink('link-93', 'get_unpacked_list', 'link-83');">get_unpacked_list</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">select</tt><tt class="py-op">.</tt><tt id="link-94" class="py-name" targets="Method Bio.PDB.Dice.ChainSelector.accept_atom()=Bio.PDB.Dice.ChainSelector-class.html#accept_atom,Method Bio.PDB.PDBIO'.Select.accept_atom()=Bio.PDB.PDBIO%27.Select-class.html#accept_atom"><a title="Bio.PDB.Dice.ChainSelector.accept_atom
Bio.PDB.PDBIO'.Select.accept_atom" class="py-name" href="#" onclick="return doclink('link-94', 'accept_atom', 'link-94');">accept_atom</a></tt><tt class="py-op">(</tt><tt class="py-name">atom</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line">                            <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-string">"_atom_site.group_PDB"</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-95" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-95', 'append', 'link-20');">append</a></tt><tt class="py-op">(</tt><tt id="link-96" class="py-name"><a title="Bio.GenBank._FeatureConsumer.residue_type
Bio.GenBank._RecordConsumer.residue_type" class="py-name" href="#" onclick="return doclink('link-96', 'residue_type', 'link-86');">residue_type</a></tt><tt class="py-op">)</tt> </tt>
<a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line">                            <tt class="py-keyword">if</tt> <tt class="py-name">preserve_atom_numbering</tt><tt class="py-op">:</tt> </tt>
<a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line">                                <tt class="py-name">atom_number</tt> <tt class="py-op">=</tt> <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-97" class="py-name" targets="Method Bio.PDB.Atom.Atom.get_serial_number()=Bio.PDB.Atom.Atom-class.html#get_serial_number"><a title="Bio.PDB.Atom.Atom.get_serial_number" class="py-name" href="#" onclick="return doclink('link-97', 'get_serial_number', 'link-97');">get_serial_number</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line">                            <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-string">"_atom_site.id"</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-98" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-98', 'append', 'link-20');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">atom_number</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line">                            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">preserve_atom_numbering</tt><tt class="py-op">:</tt> </tt>
<a name="L333"></a><tt class="py-lineno">333</tt>  <tt class="py-line">                                <tt class="py-name">atom_number</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L334"></a><tt class="py-lineno">334</tt>  <tt class="py-line">                            <tt id="link-99" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.element=Bio.KEGG.KGML.KGML_pathway.Component-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Entry.element=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Graphics.element=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.element=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.element=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Relation.element=Bio.KEGG.KGML.KGML_pathway.Relation-class.html#element,Variable Bio.Phylo._cdao_owl.element=Bio.Phylo._cdao_owl-module.html#element"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-99', 'element', 'link-99');">element</a></tt> <tt class="py-op">=</tt> <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-100" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-100', 'element', 'link-99');">element</a></tt><tt class="py-op">.</tt><tt id="link-101" class="py-name" targets="Method Bio.Seq.Seq.strip()=Bio.Seq.Seq-class.html#strip"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-101', 'strip', 'link-101');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L335"></a><tt class="py-lineno">335</tt>  <tt class="py-line">                            <tt class="py-keyword">if</tt> <tt id="link-102" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-102', 'element', 'link-99');">element</a></tt> <tt class="py-op">==</tt> <tt class="py-string">""</tt><tt class="py-op">:</tt> </tt>
<a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line">                                <tt id="link-103" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-103', 'element', 'link-99');">element</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"?"</tt> </tt>
<a name="L337"></a><tt class="py-lineno">337</tt>  <tt class="py-line">                            <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-string">"_atom_site.type_symbol"</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-104" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-104', 'append', 'link-20');">append</a></tt><tt class="py-op">(</tt><tt id="link-105" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-105', 'element', 'link-99');">element</a></tt><tt class="py-op">)</tt> </tt>
<a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line">                            <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-string">"_atom_site.label_atom_id"</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-106" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-106', 'append', 'link-20');">append</a></tt><tt class="py-op">(</tt> </tt>
<a name="L339"></a><tt class="py-lineno">339</tt>  <tt class="py-line">                                <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-107" class="py-name" targets="Method Bio.PDB.Atom.Atom.get_name()=Bio.PDB.Atom.Atom-class.html#get_name"><a title="Bio.PDB.Atom.Atom.get_name" class="py-name" href="#" onclick="return doclink('link-107', 'get_name', 'link-107');">get_name</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-108" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-108', 'strip', 'link-101');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line">                            <tt class="py-op">)</tt> </tt>
<a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line">                            <tt class="py-name">altloc</tt> <tt class="py-op">=</tt> <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-109" class="py-name" targets="Method Bio.PDB.Atom.Atom.get_altloc()=Bio.PDB.Atom.Atom-class.html#get_altloc"><a title="Bio.PDB.Atom.Atom.get_altloc" class="py-name" href="#" onclick="return doclink('link-109', 'get_altloc', 'link-109');">get_altloc</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line">                            <tt class="py-keyword">if</tt> <tt class="py-name">altloc</tt> <tt class="py-op">==</tt> <tt class="py-string">" "</tt><tt class="py-op">:</tt> </tt>
<a name="L343"></a><tt class="py-lineno">343</tt>  <tt class="py-line">                                <tt class="py-name">altloc</tt> <tt class="py-op">=</tt> <tt class="py-string">"."</tt> </tt>
<a name="L344"></a><tt class="py-lineno">344</tt>  <tt class="py-line">                            <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-string">"_atom_site.label_alt_id"</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-110" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-110', 'append', 'link-20');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">altloc</tt><tt class="py-op">)</tt> </tt>
<a name="L345"></a><tt class="py-lineno">345</tt>  <tt class="py-line">                            <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-string">"_atom_site.label_comp_id"</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-111" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-111', 'append', 'link-20');">append</a></tt><tt class="py-op">(</tt> </tt>
<a name="L346"></a><tt class="py-lineno">346</tt>  <tt class="py-line">                                <tt class="py-name">resname</tt><tt class="py-op">.</tt><tt id="link-112" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-112', 'strip', 'link-101');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L347"></a><tt class="py-lineno">347</tt>  <tt class="py-line">                            <tt class="py-op">)</tt> </tt>
<a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line">                            <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-string">"_atom_site.label_asym_id"</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-113" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-113', 'append', 'link-20');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">label_asym_id</tt><tt class="py-op">)</tt> </tt>
<a name="L349"></a><tt class="py-lineno">349</tt>  <tt class="py-line">                            <tt class="py-comment"># The entity ID should be the same for similar chains</tt> </tt>
<a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line">                            <tt class="py-comment"># However this is non-trivial to calculate so we write "?"</tt> </tt>
<a name="L351"></a><tt class="py-lineno">351</tt>  <tt class="py-line">                            <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-string">"_atom_site.label_entity_id"</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-114" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-114', 'append', 'link-20');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"?"</tt><tt class="py-op">)</tt> </tt>
<a name="L352"></a><tt class="py-lineno">352</tt>  <tt class="py-line">                            <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-string">"_atom_site.label_seq_id"</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-115" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-115', 'append', 'link-20');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">label_seq_id</tt><tt class="py-op">)</tt> </tt>
<a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line">                            <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-string">"_atom_site.pdbx_PDB_ins_code"</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-116" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-116', 'append', 'link-20');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">icode</tt><tt class="py-op">)</tt> </tt>
<a name="L354"></a><tt class="py-lineno">354</tt>  <tt class="py-line">                            <tt class="py-name">coord</tt> <tt class="py-op">=</tt> <tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-117" class="py-name" targets="Method Bio.PDB.Atom.Atom.get_coord()=Bio.PDB.Atom.Atom-class.html#get_coord"><a title="Bio.PDB.Atom.Atom.get_coord" class="py-name" href="#" onclick="return doclink('link-117', 'get_coord', 'link-117');">get_coord</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line">                            <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-string">"_atom_site.Cartn_x"</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-118" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-118', 'append', 'link-20');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"%.3f"</tt> <tt class="py-op">%</tt> <tt class="py-name">coord</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line">                            <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-string">"_atom_site.Cartn_y"</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-119" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-119', 'append', 'link-20');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"%.3f"</tt> <tt class="py-op">%</tt> <tt class="py-name">coord</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line">                            <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-string">"_atom_site.Cartn_z"</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-120" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-120', 'append', 'link-20');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"%.3f"</tt> <tt class="py-op">%</tt> <tt class="py-name">coord</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L358"></a><tt class="py-lineno">358</tt>  <tt class="py-line">                            <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-string">"_atom_site.occupancy"</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-121" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-121', 'append', 'link-20');">append</a></tt><tt class="py-op">(</tt> </tt>
<a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line">                                <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-122" class="py-name" targets="Method Bio.PDB.Atom.Atom.get_occupancy()=Bio.PDB.Atom.Atom-class.html#get_occupancy"><a title="Bio.PDB.Atom.Atom.get_occupancy" class="py-name" href="#" onclick="return doclink('link-122', 'get_occupancy', 'link-122');">get_occupancy</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line">                            <tt class="py-op">)</tt> </tt>
<a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line">                            <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-string">"_atom_site.B_iso_or_equiv"</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-123" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-123', 'append', 'link-20');">append</a></tt><tt class="py-op">(</tt> </tt>
<a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line">                                <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">atom</tt><tt class="py-op">.</tt><tt id="link-124" class="py-name" targets="Method Bio.PDB.Atom.Atom.get_bfactor()=Bio.PDB.Atom.Atom-class.html#get_bfactor"><a title="Bio.PDB.Atom.Atom.get_bfactor" class="py-name" href="#" onclick="return doclink('link-124', 'get_bfactor', 'link-124');">get_bfactor</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L363"></a><tt class="py-lineno">363</tt>  <tt class="py-line">                            <tt class="py-op">)</tt> </tt>
<a name="L364"></a><tt class="py-lineno">364</tt>  <tt class="py-line">                            <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-string">"_atom_site.auth_seq_id"</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-125" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-125', 'append', 'link-20');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">resseq</tt><tt class="py-op">)</tt> </tt>
<a name="L365"></a><tt class="py-lineno">365</tt>  <tt class="py-line">                            <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-string">"_atom_site.auth_asym_id"</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-126" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-126', 'append', 'link-20');">append</a></tt><tt class="py-op">(</tt><tt id="link-127" class="py-name"><a title="Bio.FSSP.fssp_rec.align.chain_id" class="py-name" href="#" onclick="return doclink('link-127', 'chain_id', 'link-79');">chain_id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L366"></a><tt class="py-lineno">366</tt>  <tt class="py-line">                            <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-string">"_atom_site.pdbx_PDB_model_num"</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-128" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-128', 'append', 'link-20');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">model_n</tt><tt class="py-op">)</tt> </tt>
<a name="L367"></a><tt class="py-lineno">367</tt>  <tt class="py-line"> </tt>
<a name="L368"></a><tt class="py-lineno">368</tt>  <tt class="py-line">        <tt class="py-comment"># Data block name is the structure ID with special characters removed</tt> </tt>
<a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line">        <tt class="py-name">structure_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-129" class="py-name"><a title="Bio.SubsMat.MatrixInfo.structure" class="py-name" href="#" onclick="return doclink('link-129', 'structure', 'link-74');">structure</a></tt><tt class="py-op">.</tt><tt id="link-130" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.id=Bio.KEGG.KGML.KGML_pathway.Component-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Entry.id=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.id=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#id,Variable Bio.PDB.Entity.Entity.id=Bio.PDB.Entity.Entity-class.html#id,Method Bio.Phylo.PhyloXMLIO.Parser.id()=Bio.Phylo.PhyloXMLIO.Parser-class.html#id,Method Bio.Phylo.PhyloXMLIO.Writer.id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#id,Variable Bio.SearchIO._model.hit.Hit.id=Bio.SearchIO._model.hit.Hit-class.html#id,Variable Bio.SearchIO._model.query.QueryResult.id=Bio.SearchIO._model.query.QueryResult-class.html#id"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-130', 'id', 'link-130');">id</a></tt> </tt>
<a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">c</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">"#"</tt><tt class="py-op">,</tt> <tt class="py-string">"$"</tt><tt class="py-op">,</tt> <tt class="py-string">"'"</tt><tt class="py-op">,</tt> <tt class="py-string">'"'</tt><tt class="py-op">,</tt> <tt class="py-string">"["</tt><tt class="py-op">,</tt> <tt class="py-string">"]"</tt><tt class="py-op">,</tt> <tt class="py-string">" "</tt><tt class="py-op">,</tt> <tt class="py-string">"\t"</tt><tt class="py-op">,</tt> <tt class="py-string">"\n"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L371"></a><tt class="py-lineno">371</tt>  <tt class="py-line">            <tt class="py-name">structure_id</tt> <tt class="py-op">=</tt> <tt class="py-name">structure_id</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">)</tt> </tt>
<a name="L372"></a><tt class="py-lineno">372</tt>  <tt class="py-line">        <tt class="py-name">atom_dict</tt><tt class="py-op">[</tt><tt class="py-string">"data_"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">structure_id</tt> </tt>
<a name="L373"></a><tt class="py-lineno">373</tt>  <tt class="py-line"> </tt>
<a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line">        <tt class="py-comment"># Set the dictionary and write out using the generic dictionary method</tt> </tt>
<a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dic</tt> <tt class="py-op">=</tt> <tt class="py-name">atom_dict</tt> </tt>
<a name="L376"></a><tt class="py-lineno">376</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-131" class="py-name"><a title="Bio.PDB.mmcifio.MMCIFIO._save_dict" class="py-name" href="#" onclick="return doclink('link-131', '_save_dict', 'link-17');">_save_dict</a></tt><tt class="py-op">(</tt><tt class="py-name">out_file</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L377"></a><tt class="py-lineno">377</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Nov  6 13:34:03 2019
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
