<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.SearchIO.HmmerIO.hmmer3_text</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.SearchIO-module.html">Package&nbsp;SearchIO</a> ::
        <a href="Bio.SearchIO.HmmerIO-module.html">Package&nbsp;HmmerIO</a> ::
        Module&nbsp;hmmer3_text
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.SearchIO.HmmerIO.hmmer3_text-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.SearchIO.HmmerIO.hmmer3_text-module.html">Module Bio.SearchIO.HmmerIO.hmmer3_text</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2012 by Wibowo Arindrarto.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-docstring">"""Bio.SearchIO parser for HMMER plain text output format."""</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">re</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-1', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt id="link-2" class="py-name" targets="Function Bio._py3k._as_bytes()=Bio._py3k-module.html#_as_bytes"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-2', '_as_bytes', 'link-2');">_as_bytes</a></tt><tt class="py-op">,</tt> <tt id="link-3" class="py-name" targets="Function Bio._py3k._bytes_to_string()=Bio._py3k-module.html#_bytes_to_string"><a title="Bio._py3k._bytes_to_string" class="py-name" href="#" onclick="return doclink('link-3', '_bytes_to_string', 'link-3');">_bytes_to_string</a></tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-4" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-4', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-5" class="py-name" targets="Module Bio.Phylo._utils=Bio.Phylo._utils-module.html,Module Bio.SearchIO._utils=Bio.SearchIO._utils-module.html,Module Bio._utils=Bio._utils-module.html"><a title="Bio.Phylo._utils
Bio.SearchIO._utils
Bio._utils" class="py-name" href="#" onclick="return doclink('link-5', '_utils', 'link-5');">_utils</a></tt> <tt class="py-keyword">import</tt> <tt id="link-6" class="py-name" targets="Function Bio._utils.read_forward()=Bio._utils-module.html#read_forward"><a title="Bio._utils.read_forward" class="py-name" href="#" onclick="return doclink('link-6', 'read_forward', 'link-6');">read_forward</a></tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-7" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-7', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-8" class="py-name" targets="Package Bio.Alphabet=Bio.Alphabet-module.html,Class Bio.Alphabet.Alphabet=Bio.Alphabet.Alphabet-class.html"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-8', 'Alphabet', 'link-8');">Alphabet</a></tt> <tt class="py-keyword">import</tt> <tt id="link-9" class="py-name" targets="Variable Bio.Alphabet.generic_protein=Bio.Alphabet-module.html#generic_protein"><a title="Bio.Alphabet.generic_protein" class="py-name" href="#" onclick="return doclink('link-9', 'generic_protein', 'link-9');">generic_protein</a></tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-10" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-10', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-11" class="py-name" targets="Package Bio.SearchIO=Bio.SearchIO-module.html"><a title="Bio.SearchIO" class="py-name" href="#" onclick="return doclink('link-11', 'SearchIO', 'link-11');">SearchIO</a></tt><tt class="py-op">.</tt><tt id="link-12" class="py-name" targets="Package Bio.SearchIO._model=Bio.SearchIO._model-module.html"><a title="Bio.SearchIO._model" class="py-name" href="#" onclick="return doclink('link-12', '_model', 'link-12');">_model</a></tt> <tt class="py-keyword">import</tt> <tt id="link-13" class="py-name" targets="Class Bio.SearchIO._model.query.QueryResult=Bio.SearchIO._model.query.QueryResult-class.html"><a title="Bio.SearchIO._model.query.QueryResult" class="py-name" href="#" onclick="return doclink('link-13', 'QueryResult', 'link-13');">QueryResult</a></tt><tt class="py-op">,</tt> <tt id="link-14" class="py-name" targets="Class Bio.SearchIO._model.hit.Hit=Bio.SearchIO._model.hit.Hit-class.html"><a title="Bio.SearchIO._model.hit.Hit" class="py-name" href="#" onclick="return doclink('link-14', 'Hit', 'link-14');">Hit</a></tt><tt class="py-op">,</tt> <tt id="link-15" class="py-name" targets="Class Bio.Blast.Record.HSP=Bio.Blast.Record.HSP-class.html,Class Bio.SearchIO._model.hsp.HSP=Bio.SearchIO._model.hsp.HSP-class.html"><a title="Bio.Blast.Record.HSP
Bio.SearchIO._model.hsp.HSP" class="py-name" href="#" onclick="return doclink('link-15', 'HSP', 'link-15');">HSP</a></tt><tt class="py-op">,</tt> <tt id="link-16" class="py-name" targets="Class Bio.SearchIO._model.hsp.HSPFragment=Bio.SearchIO._model.hsp.HSPFragment-class.html"><a title="Bio.SearchIO._model.hsp.HSPFragment" class="py-name" href="#" onclick="return doclink('link-16', 'HSPFragment', 'link-16');">HSPFragment</a></tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt><tt id="link-17" class="py-name" targets="Module Bio.SearchIO.ExonerateIO._base=Bio.SearchIO.ExonerateIO._base-module.html,Module Bio.SearchIO.HmmerIO._base=Bio.SearchIO.HmmerIO._base-module.html,Module Bio.SearchIO._model._base=Bio.SearchIO._model._base-module.html"><a title="Bio.SearchIO.ExonerateIO._base
Bio.SearchIO.HmmerIO._base
Bio.SearchIO._model._base" class="py-name" href="#" onclick="return doclink('link-17', '_base', 'link-17');">_base</a></tt> <tt class="py-keyword">import</tt> <tt id="link-18" class="py-name" targets="Class Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer=Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer-class.html"><a title="Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer" class="py-name" href="#" onclick="return doclink('link-18', '_BaseHmmerTextIndexer', 'link-18');">_BaseHmmerTextIndexer</a></tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-name">__all__</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">'Hmmer3TextParser'</tt><tt class="py-op">,</tt> <tt class="py-string">'Hmmer3TextIndexer'</tt><tt class="py-op">]</tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-name">__docformat__</tt> <tt class="py-op">=</tt> <tt class="py-string">"restructuredtext en"</tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-comment"># precompile regex patterns for faster processing</tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-comment"># regex for program name capture</tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt id="link-19" class="py-name" targets="Variable Bio.SearchIO.HmmerIO.hmmer3_text._RE_PROGRAM=Bio.SearchIO.HmmerIO.hmmer3_text-module.html#_RE_PROGRAM"><a title="Bio.SearchIO.HmmerIO.hmmer3_text._RE_PROGRAM" class="py-name" href="#" onclick="return doclink('link-19', '_RE_PROGRAM', 'link-19');">_RE_PROGRAM</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'^# (\w*hmm\w+) :: .*$'</tt><tt class="py-op">)</tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-comment"># regex for version string capture</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"><tt id="link-20" class="py-name" targets="Variable Bio.SearchIO.BlastIO.blast_xml._RE_VERSION=Bio.SearchIO.BlastIO.blast_xml-module.html#_RE_VERSION,Variable Bio.SearchIO.HmmerIO.hmmer3_text._RE_VERSION=Bio.SearchIO.HmmerIO.hmmer3_text-module.html#_RE_VERSION"><a title="Bio.SearchIO.BlastIO.blast_xml._RE_VERSION
Bio.SearchIO.HmmerIO.hmmer3_text._RE_VERSION" class="py-name" href="#" onclick="return doclink('link-20', '_RE_VERSION', 'link-20');">_RE_VERSION</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'# \w+ ([\w+\.]+) .*; http.*$'</tt><tt class="py-op">)</tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"><tt class="py-comment"># regex for option string capture</tt> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"><tt id="link-21" class="py-name" targets="Variable Bio.SearchIO.HmmerIO.hmmer3_text._RE_OPT=Bio.SearchIO.HmmerIO.hmmer3_text-module.html#_RE_OPT"><a title="Bio.SearchIO.HmmerIO.hmmer3_text._RE_OPT" class="py-name" href="#" onclick="return doclink('link-21', '_RE_OPT', 'link-21');">_RE_OPT</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'^# (.+):\s+(.+)$'</tt><tt class="py-op">)</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"><tt class="py-comment"># regex for parsing query id and length, for parsing</tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"><tt id="link-22" class="py-name" targets="Variable Bio.SearchIO.HmmerIO.hmmer3_text._QRE_ID_LEN_PTN=Bio.SearchIO.HmmerIO.hmmer3_text-module.html#_QRE_ID_LEN_PTN"><a title="Bio.SearchIO.HmmerIO.hmmer3_text._QRE_ID_LEN_PTN" class="py-name" href="#" onclick="return doclink('link-22', '_QRE_ID_LEN_PTN', 'link-22');">_QRE_ID_LEN_PTN</a></tt> <tt class="py-op">=</tt> <tt class="py-string">r'^Query:\s*(.*)\s+\[\w=(\d+)\]'</tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"><tt id="link-23" class="py-name" targets="Variable Bio.SearchIO.HmmerIO.hmmer3_text._QRE_ID_LEN=Bio.SearchIO.HmmerIO.hmmer3_text-module.html#_QRE_ID_LEN"><a title="Bio.SearchIO.HmmerIO.hmmer3_text._QRE_ID_LEN" class="py-name" href="#" onclick="return doclink('link-23', '_QRE_ID_LEN', 'link-23');">_QRE_ID_LEN</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt id="link-24" class="py-name"><a title="Bio.SearchIO.HmmerIO.hmmer3_text._QRE_ID_LEN_PTN" class="py-name" href="#" onclick="return doclink('link-24', '_QRE_ID_LEN_PTN', 'link-22');">_QRE_ID_LEN_PTN</a></tt><tt class="py-op">)</tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"><tt class="py-comment"># regex for hsp validation</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt id="link-25" class="py-name" targets="Variable Bio.SearchIO.HmmerIO.hmmer3_text._HRE_VALIDATE=Bio.SearchIO.HmmerIO.hmmer3_text-module.html#_HRE_VALIDATE"><a title="Bio.SearchIO.HmmerIO.hmmer3_text._HRE_VALIDATE" class="py-name" href="#" onclick="return doclink('link-25', '_HRE_VALIDATE', 'link-25');">_HRE_VALIDATE</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'score:\s(-?\d+\.?\d+)\sbits.*value:\s(.*)'</tt><tt class="py-op">)</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-comment"># regexes for parsing hsp alignment blocks</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt id="link-26" class="py-name" targets="Variable Bio.SearchIO.HmmerIO.hmmer3_text._HRE_ANNOT_LINE=Bio.SearchIO.HmmerIO.hmmer3_text-module.html#_HRE_ANNOT_LINE"><a title="Bio.SearchIO.HmmerIO.hmmer3_text._HRE_ANNOT_LINE" class="py-name" href="#" onclick="return doclink('link-26', '_HRE_ANNOT_LINE', 'link-26');">_HRE_ANNOT_LINE</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'^(\s+)(.+)\s(\w+)'</tt><tt class="py-op">)</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"><tt id="link-27" class="py-name" targets="Variable Bio.SearchIO.HmmerIO.hmmer3_text._HRE_ID_LINE=Bio.SearchIO.HmmerIO.hmmer3_text-module.html#_HRE_ID_LINE"><a title="Bio.SearchIO.HmmerIO.hmmer3_text._HRE_ID_LINE" class="py-name" href="#" onclick="return doclink('link-27', '_HRE_ID_LINE', 'link-27');">_HRE_ID_LINE</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'^(\s+\S+\s+[0-9-]+ )(.+?)(\s+[0-9-]+)'</tt><tt class="py-op">)</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"> </tt>
<a name="Hmmer3TextParser"></a><div id="Hmmer3TextParser-def"><a name="L39"></a><tt class="py-lineno"> 39</tt> <a class="py-toggle" href="#" id="Hmmer3TextParser-toggle" onclick="return toggle('Hmmer3TextParser');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html">Hmmer3TextParser</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Hmmer3TextParser-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Hmmer3TextParser-expanded"><a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line">    <tt class="py-docstring">"""Parser for the HMMER 3.0 text output."""</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line"> </tt>
<a name="Hmmer3TextParser.__init__"></a><div id="Hmmer3TextParser.__init__-def"><a name="L43"></a><tt class="py-lineno"> 43</tt> <a class="py-toggle" href="#" id="Hmmer3TextParser.__init__-toggle" onclick="return toggle('Hmmer3TextParser.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Hmmer3TextParser.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Hmmer3TextParser.__init__-expanded"><a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-28" class="py-name"><a title="Bio._utils.read_forward" class="py-name" href="#" onclick="return doclink('link-28', 'read_forward', 'link-6');">read_forward</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_meta</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-29" class="py-name" targets="Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser._parse_preamble()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser-class.html#_parse_preamble,Method Bio.SearchIO.FastaIO.FastaM10Parser._parse_preamble()=Bio.SearchIO.FastaIO.FastaM10Parser-class.html#_parse_preamble,Method Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser._parse_preamble()=Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html#_parse_preamble"><a title="Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser._parse_preamble
Bio.SearchIO.FastaIO.FastaM10Parser._parse_preamble
Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser._parse_preamble" class="py-name" href="#" onclick="return doclink('link-29', '_parse_preamble', 'link-29');">_parse_preamble</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line"> </tt>
<a name="Hmmer3TextParser.__iter__"></a><div id="Hmmer3TextParser.__iter__-def"><a name="L48"></a><tt class="py-lineno"> 48</tt> <a class="py-toggle" href="#" id="Hmmer3TextParser.__iter__-toggle" onclick="return toggle('Hmmer3TextParser.__iter__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html#__iter__">__iter__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Hmmer3TextParser.__iter__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Hmmer3TextParser.__iter__-expanded"><a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">qresult</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-30" class="py-name" targets="Method Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._parse_qresult()=Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#_parse_qresult,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser._parse_qresult()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser-class.html#_parse_qresult,Method Bio.SearchIO.BlatIO.BlatPslParser._parse_qresult()=Bio.SearchIO.BlatIO.BlatPslParser-class.html#_parse_qresult,Method Bio.SearchIO.ExonerateIO._base._BaseExonerateParser._parse_qresult()=Bio.SearchIO.ExonerateIO._base._BaseExonerateParser-class.html#_parse_qresult,Method Bio.SearchIO.FastaIO.FastaM10Parser._parse_qresult()=Bio.SearchIO.FastaIO.FastaM10Parser-class.html#_parse_qresult,Method Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabParser._parse_qresult()=Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabParser-class.html#_parse_qresult,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser._parse_qresult()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser-class.html#_parse_qresult,Method Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser._parse_qresult()=Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html#_parse_qresult"><a title="Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._parse_qresult
Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser._parse_qresult
Bio.SearchIO.BlatIO.BlatPslParser._parse_qresult
Bio.SearchIO.ExonerateIO._base._BaseExonerateParser._parse_qresult
Bio.SearchIO.FastaIO.FastaM10Parser._parse_qresult
Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabParser._parse_qresult
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser._parse_qresult
Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser._parse_qresult" class="py-name" href="#" onclick="return doclink('link-30', '_parse_qresult', 'link-30');">_parse_qresult</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line">            <tt class="py-keyword">yield</tt> <tt class="py-name">qresult</tt> </tt>
</div><a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line"> </tt>
<a name="Hmmer3TextParser._read_until"></a><div id="Hmmer3TextParser._read_until-def"><a name="L52"></a><tt class="py-lineno"> 52</tt> <a class="py-toggle" href="#" id="Hmmer3TextParser._read_until-toggle" onclick="return toggle('Hmmer3TextParser._read_until');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html#_read_until">_read_until</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">bool_func</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Hmmer3TextParser._read_until-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Hmmer3TextParser._read_until-expanded"><a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line">        <tt class="py-docstring">"""Reads the file handle until the given function returns True."""</tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-keyword">or</tt> <tt class="py-name">bool_func</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-31" class="py-name"><a title="Bio._utils.read_forward" class="py-name" href="#" onclick="return doclink('link-31', 'read_forward', 'link-6');">read_forward</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
</div><a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line"> </tt>
<a name="Hmmer3TextParser._parse_preamble"></a><div id="Hmmer3TextParser._parse_preamble-def"><a name="L60"></a><tt class="py-lineno"> 60</tt> <a class="py-toggle" href="#" id="Hmmer3TextParser._parse_preamble-toggle" onclick="return toggle('Hmmer3TextParser._parse_preamble');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html#_parse_preamble">_parse_preamble</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Hmmer3TextParser._parse_preamble-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Hmmer3TextParser._parse_preamble-expanded"><a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line">        <tt class="py-docstring">"""Parses HMMER preamble (lines beginning with '#')."""</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line">        <tt class="py-name">meta</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line">        <tt class="py-comment"># bool flag for storing state ~ whether we are parsing the option</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line">        <tt class="py-comment"># lines or not</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line">        <tt class="py-name">has_opts</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line">            <tt class="py-comment"># no pound sign means we've left the preamble</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-32" class="py-name" targets="Method Bio.Seq.Seq.startswith()=Bio.Seq.Seq-class.html#startswith"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-32', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'#'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line">            <tt class="py-comment"># dashes could either mean we are entering or leaving the options</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line">            <tt class="py-comment"># section ~ so it's a switch for the has_opts flag</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-string">'- - -'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">has_opts</tt><tt class="py-op">:</tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line">                    <tt class="py-comment"># if flag is false, that means we're entering opts</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line">                    <tt class="py-comment"># so switch the flag accordingly</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line">                    <tt class="py-name">has_opts</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line">                    <tt class="py-comment"># if flag is true, that means we've reached the end of opts</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line">                    <tt class="py-comment"># so we can break out of the function</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-keyword">not</tt> <tt class="py-name">has_opts</tt><tt class="py-op">:</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line">                <tt class="py-comment"># try parsing program</tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line">                <tt class="py-name">regx</tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-33" class="py-name" targets="Method Bio.KDTree.KDTree'.KDTree.search()=Bio.KDTree.KDTree%27.KDTree-class.html#search,Method Bio.PDB.NeighborSearch'.NeighborSearch.search()=Bio.PDB.NeighborSearch%27.NeighborSearch-class.html#search,Method Bio.Phylo.TreeConstruction.NNITreeSearcher.search()=Bio.Phylo.TreeConstruction.NNITreeSearcher-class.html#search,Method Bio.Phylo.TreeConstruction.TreeSearcher.search()=Bio.Phylo.TreeConstruction.TreeSearcher-class.html#search,Class Method Bio.Restriction.Restriction.AbstractCut.search()=Bio.Restriction.Restriction.AbstractCut-class.html#search,Method Bio.Restriction.Restriction.RestrictionBatch.search()=Bio.Restriction.Restriction.RestrictionBatch-class.html#search,Function Bio.SCOP.search()=Bio.SCOP-module.html#search,Function Bio.TogoWS.search()=Bio.TogoWS-module.html#search,Method Bio.motifs.Instances.search()=Bio.motifs.Instances-class.html#search,Method Bio.motifs.matrix.PositionSpecificScoringMatrix.search()=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#search"><a title="Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-33', 'search', 'link-33');">search</a></tt><tt class="py-op">(</tt><tt id="link-34" class="py-name"><a title="Bio.SearchIO.HmmerIO.hmmer3_text._RE_PROGRAM" class="py-name" href="#" onclick="return doclink('link-34', '_RE_PROGRAM', 'link-19');">_RE_PROGRAM</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">regx</tt><tt class="py-op">:</tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line">                    <tt class="py-name">meta</tt><tt class="py-op">[</tt><tt class="py-string">'program'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">regx</tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line">                <tt class="py-comment"># try parsing version</tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line">                <tt class="py-name">regx</tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-35" class="py-name"><a title="Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-35', 'search', 'link-33');">search</a></tt><tt class="py-op">(</tt><tt id="link-36" class="py-name"><a title="Bio.SearchIO.BlastIO.blast_xml._RE_VERSION
Bio.SearchIO.HmmerIO.hmmer3_text._RE_VERSION" class="py-name" href="#" onclick="return doclink('link-36', '_RE_VERSION', 'link-20');">_RE_VERSION</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">regx</tt><tt class="py-op">:</tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line">                    <tt class="py-name">meta</tt><tt class="py-op">[</tt><tt class="py-string">'version'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">regx</tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">has_opts</tt><tt class="py-op">:</tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line">                <tt class="py-name">regx</tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-37" class="py-name"><a title="Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-37', 'search', 'link-33');">search</a></tt><tt class="py-op">(</tt><tt id="link-38" class="py-name"><a title="Bio.SearchIO.HmmerIO.hmmer3_text._RE_OPT" class="py-name" href="#" onclick="return doclink('link-38', '_RE_OPT', 'link-21');">_RE_OPT</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line">                <tt class="py-comment"># if target in regx.group(1), then we store the key as target</tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-string">'target'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">regx</tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line">                    <tt class="py-name">meta</tt><tt class="py-op">[</tt><tt class="py-string">'target'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">regx</tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">2</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-39" class="py-name" targets="Method Bio.Seq.Seq.strip()=Bio.Seq.Seq-class.html#strip"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-39', 'strip', 'link-39');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line">                    <tt class="py-name">meta</tt><tt class="py-op">[</tt><tt class="py-name">regx</tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">regx</tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">2</tt><tt class="py-op">)</tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line"> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-40" class="py-name"><a title="Bio._utils.read_forward" class="py-name" href="#" onclick="return doclink('link-40', 'read_forward', 'link-6');">read_forward</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line"> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">meta</tt> </tt>
</div><a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line"> </tt>
<a name="Hmmer3TextParser._parse_qresult"></a><div id="Hmmer3TextParser._parse_qresult-def"><a name="L102"></a><tt class="py-lineno">102</tt> <a class="py-toggle" href="#" id="Hmmer3TextParser._parse_qresult-toggle" onclick="return toggle('Hmmer3TextParser._parse_qresult');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html#_parse_qresult">_parse_qresult</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Hmmer3TextParser._parse_qresult-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Hmmer3TextParser._parse_qresult-expanded"><a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line">        <tt class="py-docstring">"""Parses a HMMER3 query block."""</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line"> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-41" class="py-name" targets="Method Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser._read_until()=Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html#_read_until"><a title="Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser._read_until" class="py-name" href="#" onclick="return doclink('link-41', '_read_until', 'link-41');">_read_until</a></tt><tt class="py-op">(</tt><tt class="py-keyword">lambda</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-42" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-42', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Query:'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line"> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line"> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line">            <tt class="py-comment"># get query id and length</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line">            <tt class="py-name">regx</tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-43" class="py-name"><a title="Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-43', 'search', 'link-33');">search</a></tt><tt class="py-op">(</tt><tt id="link-44" class="py-name"><a title="Bio.SearchIO.HmmerIO.hmmer3_text._QRE_ID_LEN" class="py-name" href="#" onclick="return doclink('link-44', '_QRE_ID_LEN', 'link-23');">_QRE_ID_LEN</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line">            <tt class="py-name">qid</tt> <tt class="py-op">=</tt> <tt class="py-name">regx</tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-45" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-45', 'strip', 'link-39');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">            <tt class="py-comment"># store qresult attributes</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">            <tt class="py-name">qresult_attrs</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">                <tt class="py-string">'seq_len'</tt><tt class="py-op">:</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">regx</tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">2</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">                <tt class="py-string">'program'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_meta</tt><tt class="py-op">.</tt><tt id="link-46" class="py-name" targets="Method Bio.Crystal.Crystal.get()=Bio.Crystal.Crystal-class.html#get,Method Bio.Data.CodonTable.AmbiguousForwardTable.get()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#get,Method Bio.File._IndexedSeqFileDict.get()=Bio.File._IndexedSeqFileDict-class.html#get,Method Bio.File._IndexedSeqFileProxy.get()=Bio.File._IndexedSeqFileProxy-class.html#get,Method Bio.File._SQLiteManySeqFilesDict.get()=Bio.File._SQLiteManySeqFilesDict-class.html#get,Method Bio.Restriction.Restriction.RestrictionBatch.get()=Bio.Restriction.Restriction.RestrictionBatch-class.html#get,Method Bio.SearchIO._index.SearchIndexer.get()=Bio.SearchIO._index.SearchIndexer-class.html#get,Method Bio.SeqIO._index.SeqFileRandomAccess.get()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#get,Method Bio.SeqIO._index.SffRandomAccess.get()=Bio.SeqIO._index.SffRandomAccess-class.html#get,Method Bio.SeqIO._index.SffTrimedRandomAccess.get()=Bio.SeqIO._index.SffTrimedRandomAccess-class.html#get,Method Bio.SeqIO._index.UniprotRandomAccess.get()=Bio.SeqIO._index.UniprotRandomAccess-class.html#get"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-46', 'get', 'link-46');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'program'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">                <tt class="py-string">'version'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_meta</tt><tt class="py-op">.</tt><tt id="link-47" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-47', 'get', 'link-46');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'version'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line">                <tt class="py-string">'target'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_meta</tt><tt class="py-op">.</tt><tt id="link-48" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-48', 'get', 'link-46');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'target'</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line">            <tt class="py-op">}</tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line"> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">            <tt class="py-comment"># get description and accession, if they exist</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">            <tt class="py-name">qdesc</tt> <tt class="py-op">=</tt> <tt class="py-string">'&lt;unknown description&gt;'</tt>  <tt class="py-comment"># placeholder</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line">            <tt class="py-keyword">while</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-49" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-49', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Scores for '</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-50" class="py-name"><a title="Bio._utils.read_forward" class="py-name" href="#" onclick="return doclink('link-50', 'read_forward', 'link-6');">read_forward</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line"> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-51" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-51', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Accession:'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line">                    <tt id="link-52" class="py-name" targets="Variable Bio.FSSP.fssp_rec.align.acc=Bio.FSSP.fssp_rec.align-class.html#acc"><a title="Bio.FSSP.fssp_rec.align.acc" class="py-name" href="#" onclick="return doclink('link-52', 'acc', 'link-52');">acc</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-53" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-53', 'strip', 'link-39');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-54" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-54', 'split', 'link-54');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">' '</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line">                    <tt class="py-name">qresult_attrs</tt><tt class="py-op">[</tt><tt class="py-string">'accession'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-55" class="py-name"><a title="Bio.FSSP.fssp_rec.align.acc" class="py-name" href="#" onclick="return doclink('link-55', 'acc', 'link-52');">acc</a></tt><tt class="py-op">.</tt><tt id="link-56" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-56', 'strip', 'link-39');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-57" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-57', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Description:'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line">                    <tt class="py-name">qdesc</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-58" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-58', 'strip', 'link-39');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-59" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-59', 'split', 'link-54');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">' '</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-60" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-60', 'strip', 'link-39');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line">                    <tt class="py-name">qresult_attrs</tt><tt class="py-op">[</tt><tt class="py-string">'description'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">qdesc</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line"> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line">            <tt class="py-comment"># parse the query hits</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line">            <tt class="py-keyword">while</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-keyword">and</tt> <tt class="py-string">'//'</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line">                <tt class="py-name">hit_list</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-61" class="py-name" targets="Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser._parse_hit()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser-class.html#_parse_hit,Method Bio.SearchIO.FastaIO.FastaM10Parser._parse_hit()=Bio.SearchIO.FastaIO.FastaM10Parser-class.html#_parse_hit,Method Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser._parse_hit()=Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html#_parse_hit"><a title="Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser._parse_hit
Bio.SearchIO.FastaIO.FastaM10Parser._parse_hit
Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser._parse_hit" class="py-name" href="#" onclick="return doclink('link-61', '_parse_hit', 'link-61');">_parse_hit</a></tt><tt class="py-op">(</tt><tt class="py-name">qid</tt><tt class="py-op">,</tt> <tt class="py-name">qdesc</tt><tt class="py-op">)</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line">                <tt class="py-comment"># read through the statistics summary</tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line">                <tt class="py-comment"># TODO: parse and store this information?</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-62" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-62', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Internal pipeline'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line">                    <tt class="py-keyword">while</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-keyword">and</tt> <tt class="py-string">'//'</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line">                        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-63" class="py-name"><a title="Bio._utils.read_forward" class="py-name" href="#" onclick="return doclink('link-63', 'read_forward', 'link-6');">read_forward</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line"> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line">            <tt class="py-comment"># create qresult, set its attributes and yield</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line">            <tt class="py-comment"># not initializing hit_list directly to handle empty hits</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line">            <tt class="py-comment"># (i.e. need to set its query description manually)</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line">            <tt class="py-name">qresult</tt> <tt class="py-op">=</tt> <tt id="link-64" class="py-name"><a title="Bio.SearchIO._model.query.QueryResult" class="py-name" href="#" onclick="return doclink('link-64', 'QueryResult', 'link-13');">QueryResult</a></tt><tt class="py-op">(</tt><tt id="link-65" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.id=Bio.KEGG.KGML.KGML_pathway.Component-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Entry.id=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.id=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#id,Method Bio.Phylo.PhyloXMLIO.Parser.id()=Bio.Phylo.PhyloXMLIO.Parser-class.html#id,Method Bio.Phylo.PhyloXMLIO.Writer.id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#id,Variable Bio.SearchIO._model.hit.Hit.id=Bio.SearchIO._model.hit.Hit-class.html#id,Variable Bio.SearchIO._model.query.QueryResult.id=Bio.SearchIO._model.query.QueryResult-class.html#id"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-65', 'id', 'link-65');">id</a></tt><tt class="py-op">=</tt><tt class="py-name">qid</tt><tt class="py-op">,</tt> <tt id="link-66" class="py-name" targets="Variable Bio.SearchIO._model.query.QueryResult.hits=Bio.SearchIO._model.query.QueryResult-class.html#hits"><a title="Bio.SearchIO._model.query.QueryResult.hits" class="py-name" href="#" onclick="return doclink('link-66', 'hits', 'link-66');">hits</a></tt><tt class="py-op">=</tt><tt class="py-name">hit_list</tt><tt class="py-op">)</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">attr</tt><tt class="py-op">,</tt> <tt id="link-67" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.value()=Bio.Phylo.PhyloXMLIO.Writer-class.html#value"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-67', 'value', 'link-67');">value</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">qresult_attrs</tt><tt class="py-op">.</tt><tt id="link-68" class="py-name" targets="Method Bio.Crystal.Crystal.items()=Bio.Crystal.Crystal-class.html#items,Method Bio.File._IndexedSeqFileDict.items()=Bio.File._IndexedSeqFileDict-class.html#items,Method Bio.Phylo.PhyloXML.Events.items()=Bio.Phylo.PhyloXML.Events-class.html#items,Variable Bio.SearchIO._model.query.QueryResult.items=Bio.SearchIO._model.query.QueryResult-class.html#items,Method Bio._py3k._ordereddict.OrderedDict.items()=Bio._py3k._ordereddict.OrderedDict-class.html#items,Method BioSQL.BioSeqDatabase.BioSeqDatabase.items()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#items,Method BioSQL.BioSeqDatabase.DBServer.items()=BioSQL.BioSeqDatabase.DBServer-class.html#items"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
Bio._py3k._ordereddict.OrderedDict.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-68', 'items', 'link-68');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line">                <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt class="py-name">qresult</tt><tt class="py-op">,</tt> <tt class="py-name">attr</tt><tt class="py-op">,</tt> <tt id="link-69" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-69', 'value', 'link-67');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line">            <tt class="py-keyword">yield</tt> <tt class="py-name">qresult</tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-70" class="py-name"><a title="Bio._utils.read_forward" class="py-name" href="#" onclick="return doclink('link-70', 'read_forward', 'link-6');">read_forward</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line"> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">            <tt class="py-comment"># Skip line beginning with '# Alignment of', which are output</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">            <tt class="py-comment"># when running phmmer with the '-A' flag.</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-71" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-71', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'# Alignment of'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-72" class="py-name" targets="Method Bio.File.UndoHandle.readline()=Bio.File.UndoHandle-class.html#readline,Method Bio.bgzf.BgzfReader.readline()=Bio.bgzf.BgzfReader-class.html#readline"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-72', 'readline', 'link-72');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line"> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">            <tt class="py-comment"># HMMER &gt;= 3.1 outputs '[ok]' at the end of all results file,</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line">            <tt class="py-comment"># which means we can break the main loop when we see the line</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-string">'[ok]'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
</div><a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line"> </tt>
<a name="Hmmer3TextParser._parse_hit"></a><div id="Hmmer3TextParser._parse_hit-def"><a name="L160"></a><tt class="py-lineno">160</tt> <a class="py-toggle" href="#" id="Hmmer3TextParser._parse_hit-toggle" onclick="return toggle('Hmmer3TextParser._parse_hit');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html#_parse_hit">_parse_hit</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">qid</tt><tt class="py-op">,</tt> <tt class="py-param">qdesc</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Hmmer3TextParser._parse_hit-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Hmmer3TextParser._parse_hit-expanded"><a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">        <tt class="py-docstring">"""Parses a HMMER3 hit block, beginning with the hit table."""</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">        <tt class="py-comment"># get to the end of the hit table delimiter and read one more line</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-73" class="py-name"><a title="Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser._read_until" class="py-name" href="#" onclick="return doclink('link-73', '_read_until', 'link-41');">_read_until</a></tt><tt class="py-op">(</tt><tt class="py-keyword">lambda</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">                <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-74" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-74', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'    ------- ------ -----'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-75" class="py-name"><a title="Bio._utils.read_forward" class="py-name" href="#" onclick="return doclink('link-75', 'read_forward', 'link-6');">read_forward</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line"> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">        <tt class="py-comment"># assume every hit is in inclusion threshold until the inclusion</tt> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">        <tt class="py-comment"># threshold line is encountered</tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line">        <tt class="py-name">is_included</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line"> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line">        <tt class="py-comment"># parse the hit table</tt> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line">        <tt class="py-name">hit_attr_list</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-76" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-76', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'  ------ inclusion'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line">                <tt class="py-name">is_included</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-77" class="py-name"><a title="Bio._utils.read_forward" class="py-name" href="#" onclick="return doclink('link-77', 'read_forward', 'link-6');">read_forward</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line">            <tt class="py-comment"># if there are no hits, then there are no hsps</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line">            <tt class="py-comment"># so we forward-read until 'Internal pipeline..'</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-78" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-78', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'   [No hits detected that satisfy '</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">                    <tt class="py-string">'reporting'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">                <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-79" class="py-name"><a title="Bio._utils.read_forward" class="py-name" href="#" onclick="return doclink('link-79', 'read_forward', 'link-6');">read_forward</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-80" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-80', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Internal pipeline'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line">                        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">hit_attr_list</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">                        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-81" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-81', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Domain annotation for each '</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line">                <tt class="py-name">hit_list</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-82" class="py-name" targets="Method Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser._create_hits()=Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html#_create_hits"><a title="Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser._create_hits" class="py-name" href="#" onclick="return doclink('link-82', '_create_hits', 'link-82');">_create_hits</a></tt><tt class="py-op">(</tt><tt class="py-name">hit_attr_list</tt><tt class="py-op">,</tt> <tt class="py-name">qid</tt><tt class="py-op">,</tt> <tt class="py-name">qdesc</tt><tt class="py-op">)</tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">hit_list</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">            <tt class="py-comment"># entering hit results row</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">            <tt class="py-comment"># parse the columns into a list</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line">            <tt class="py-name">row</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-83" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.x=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#x"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-83', 'x', 'link-83');">x</a></tt> <tt class="py-keyword">for</tt> <tt id="link-84" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-84', 'x', 'link-83');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-85" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-85', 'strip', 'link-39');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-86" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-86', 'split', 'link-54');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">' '</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt id="link-87" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-87', 'x', 'link-83');">x</a></tt><tt class="py-op">]</tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">            <tt class="py-comment"># join the description words if it's &gt;1 word</tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">10</tt><tt class="py-op">:</tt> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">                <tt class="py-name">row</tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">' '</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line">            <tt class="py-comment"># if there's no description, set it to an empty string</tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">10</tt><tt class="py-op">:</tt> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line">                <tt class="py-name">row</tt><tt class="py-op">.</tt><tt id="link-88" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-88', 'append', 'link-88');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line">                <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">10</tt> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line">            <tt class="py-comment"># create the hit object</tt> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line">            <tt class="py-name">hit_attrs</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line">                <tt class="py-string">'id'</tt><tt class="py-op">:</tt> <tt class="py-name">row</tt><tt class="py-op">[</tt><tt class="py-number">8</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line">                <tt class="py-string">'query_id'</tt><tt class="py-op">:</tt> <tt class="py-name">qid</tt><tt class="py-op">,</tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line">                <tt class="py-string">'evalue'</tt><tt class="py-op">:</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line">                <tt class="py-string">'bitscore'</tt><tt class="py-op">:</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line">                <tt class="py-string">'bias'</tt><tt class="py-op">:</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line">                <tt class="py-comment"># row[3:6] is not parsed, since the info is available</tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line">                <tt class="py-comment"># at the HSP level</tt> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line">                <tt class="py-string">'domain_exp_num'</tt><tt class="py-op">:</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line">                <tt class="py-string">'domain_obs_num'</tt><tt class="py-op">:</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">row</tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line">                <tt class="py-string">'description'</tt><tt class="py-op">:</tt> <tt class="py-name">row</tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line">                <tt class="py-string">'is_included'</tt><tt class="py-op">:</tt> <tt class="py-name">is_included</tt><tt class="py-op">,</tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line">            <tt class="py-op">}</tt> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line">            <tt class="py-name">hit_attr_list</tt><tt class="py-op">.</tt><tt id="link-89" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-89', 'append', 'link-88');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">hit_attrs</tt><tt class="py-op">)</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line"> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-90" class="py-name"><a title="Bio._utils.read_forward" class="py-name" href="#" onclick="return doclink('link-90', 'read_forward', 'link-6');">read_forward</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
</div><a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line"> </tt>
<a name="Hmmer3TextParser._create_hits"></a><div id="Hmmer3TextParser._create_hits-def"><a name="L219"></a><tt class="py-lineno">219</tt> <a class="py-toggle" href="#" id="Hmmer3TextParser._create_hits-toggle" onclick="return toggle('Hmmer3TextParser._create_hits');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html#_create_hits">_create_hits</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">hit_attrs</tt><tt class="py-op">,</tt> <tt class="py-param">qid</tt><tt class="py-op">,</tt> <tt class="py-param">qdesc</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Hmmer3TextParser._create_hits-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Hmmer3TextParser._create_hits-expanded"><a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line">        <tt class="py-docstring">"""Parses a HMMER3 hsp block, beginning with the hsp table."""</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line">        <tt class="py-comment"># read through until the beginning of the hsp block</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-91" class="py-name"><a title="Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser._read_until" class="py-name" href="#" onclick="return doclink('link-91', '_read_until', 'link-41');">_read_until</a></tt><tt class="py-op">(</tt><tt class="py-keyword">lambda</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-92" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-92', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Internal pipeline'</tt><tt class="py-op">)</tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line">                <tt class="py-keyword">or</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-93" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-93', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'&gt;&gt;'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line"> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line">        <tt class="py-comment"># start parsing the hsp block</tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line">        <tt class="py-name">hit_list</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-94" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-94', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Internal pipeline'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line">                <tt class="py-comment"># by this time we should've emptied the hit attr list</tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line">                <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">hit_attrs</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">hit_list</tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-95" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-95', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'&gt;&gt;'</tt><tt class="py-op">)</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">            <tt class="py-name">hid</tt><tt class="py-op">,</tt> <tt class="py-name">hdesc</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-string">'&gt;&gt; '</tt><tt class="py-op">)</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-96" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-96', 'split', 'link-54');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">'  '</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line">            <tt class="py-name">hdesc</tt> <tt class="py-op">=</tt> <tt class="py-name">hdesc</tt><tt class="py-op">.</tt><tt id="link-97" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-97', 'strip', 'link-39');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line"> </tt>
<a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line">            <tt class="py-comment"># read through the hsp table header and move one more line</tt> </tt>
<a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-98" class="py-name"><a title="Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser._read_until" class="py-name" href="#" onclick="return doclink('link-98', '_read_until', 'link-41');">_read_until</a></tt><tt class="py-op">(</tt><tt class="py-keyword">lambda</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line">                    <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-99" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-99', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">' ---   ------ ----- --------'</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line">                    <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-100" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-100', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'   [No individual domains'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-101" class="py-name"><a title="Bio._utils.read_forward" class="py-name" href="#" onclick="return doclink('link-101', 'read_forward', 'link-6');">read_forward</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line"> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line">            <tt class="py-comment"># parse the hsp table for the current hit</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line">            <tt class="py-name">hsp_list</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line">            <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line">                <tt class="py-comment"># break out of hsp parsing if there are no hits, it's the last hsp</tt> </tt>
<a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line">                <tt class="py-comment"># or it's the start of a new hit</tt> </tt>
<a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-102" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-102', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'   [No targets detected that satisfy'</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> \ </tt>
<a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line">                   <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-103" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-103', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'   [No individual domains'</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> \ </tt>
<a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line">                   <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-104" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-104', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Internal pipeline statistics summary:'</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> \ </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line">                   <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-105" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-105', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'  Alignments for each domain:'</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> \ </tt>
<a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line">                   <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-106" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-106', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'&gt;&gt;'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line"> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line">                    <tt class="py-name">hit_attr</tt> <tt class="py-op">=</tt> <tt class="py-name">hit_attrs</tt><tt class="py-op">.</tt><tt id="link-107" class="py-name" targets="Method Bio.File._IndexedSeqFileDict.pop()=Bio.File._IndexedSeqFileDict-class.html#pop,Method Bio.SearchIO._model.hit.Hit.pop()=Bio.SearchIO._model.hit.Hit-class.html#pop,Method Bio.SearchIO._model.query.QueryResult.pop()=Bio.SearchIO._model.query.QueryResult-class.html#pop,Method Bio.Seq.MutableSeq.pop()=Bio.Seq.MutableSeq-class.html#pop,Method Bio._py3k._ordereddict.OrderedDict.pop()=Bio._py3k._ordereddict.OrderedDict-class.html#pop"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop
Bio._py3k._ordereddict.OrderedDict.pop" class="py-name" href="#" onclick="return doclink('link-107', 'pop', 'link-107');">pop</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line">                    <tt id="link-108" class="py-name" targets="Module Bio.SearchIO._model.hit=Bio.SearchIO._model.hit-module.html,Variable Bio.SearchIO._model.hsp.HSP.hit=Bio.SearchIO._model.hsp.HSP-class.html#hit,Variable Bio.SearchIO._model.hsp.HSPFragment.hit=Bio.SearchIO._model.hsp.HSPFragment-class.html#hit"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-108', 'hit', 'link-108');">hit</a></tt> <tt class="py-op">=</tt> <tt id="link-109" class="py-name"><a title="Bio.SearchIO._model.hit.Hit" class="py-name" href="#" onclick="return doclink('link-109', 'Hit', 'link-14');">Hit</a></tt><tt class="py-op">(</tt><tt class="py-name">hsp_list</tt><tt class="py-op">)</tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-name">attr</tt><tt class="py-op">,</tt> <tt id="link-110" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-110', 'value', 'link-67');">value</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">hit_attr</tt><tt class="py-op">.</tt><tt id="link-111" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
Bio._py3k._ordereddict.OrderedDict.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-111', 'items', 'link-68');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">attr</tt> <tt class="py-op">==</tt> <tt class="py-string">"description"</tt><tt class="py-op">:</tt> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line">                            <tt class="py-name">cur_val</tt> <tt class="py-op">=</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt id="link-112" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-112', 'hit', 'link-108');">hit</a></tt><tt class="py-op">,</tt> <tt class="py-name">attr</tt><tt class="py-op">)</tt> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line">                            <tt class="py-keyword">if</tt> <tt class="py-name">cur_val</tt> <tt class="py-keyword">and</tt> <tt id="link-113" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-113', 'value', 'link-67');">value</a></tt> <tt class="py-keyword">and</tt> <tt class="py-name">cur_val</tt><tt class="py-op">.</tt><tt id="link-114" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-114', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt id="link-115" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-115', 'value', 'link-67');">value</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line">                                <tt class="py-keyword">continue</tt> </tt>
<a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line">                        <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt id="link-116" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-116', 'hit', 'link-108');">hit</a></tt><tt class="py-op">,</tt> <tt class="py-name">attr</tt><tt class="py-op">,</tt> <tt id="link-117" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-117', 'value', 'link-67');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-118" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-118', 'hit', 'link-108');">hit</a></tt><tt class="py-op">:</tt> </tt>
<a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line">                        <tt id="link-119" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-119', 'hit', 'link-108');">hit</a></tt><tt class="py-op">.</tt><tt id="link-120" class="py-name" targets="Variable Bio.SearchIO._model.hit.Hit.query_description=Bio.SearchIO._model.hit.Hit-class.html#query_description,Variable Bio.SearchIO._model.hsp.HSP.query_description=Bio.SearchIO._model.hsp.HSP-class.html#query_description,Variable Bio.SearchIO._model.hsp.HSPFragment.query_description=Bio.SearchIO._model.hsp.HSPFragment-class.html#query_description"><a title="Bio.SearchIO._model.hit.Hit.query_description
Bio.SearchIO._model.hsp.HSP.query_description
Bio.SearchIO._model.hsp.HSPFragment.query_description" class="py-name" href="#" onclick="return doclink('link-120', 'query_description', 'link-120');">query_description</a></tt> <tt class="py-op">=</tt> <tt class="py-name">qdesc</tt> </tt>
<a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line">                    <tt class="py-name">hit_list</tt><tt class="py-op">.</tt><tt id="link-121" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-121', 'append', 'link-88');">append</a></tt><tt class="py-op">(</tt><tt id="link-122" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-122', 'hit', 'link-108');">hit</a></tt><tt class="py-op">)</tt> </tt>
<a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line"> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line">                <tt class="py-name">parsed</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-123" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-123', 'x', 'link-83');">x</a></tt> <tt class="py-keyword">for</tt> <tt id="link-124" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-124', 'x', 'link-83');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-125" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-125', 'strip', 'link-39');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-126" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-126', 'split', 'link-54');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">' '</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt id="link-127" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-127', 'x', 'link-83');">x</a></tt><tt class="py-op">]</tt> </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line">                <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">parsed</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">16</tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line">                <tt class="py-comment"># parsed column order:</tt> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line">                <tt class="py-comment"># index, is_included, bitscore, bias, evalue_cond, evalue</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line">                <tt class="py-comment"># hmmfrom, hmmto, query_ends, hit_ends, alifrom, alito,</tt> </tt>
<a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line">                <tt class="py-comment"># envfrom, envto, acc_avg</tt> </tt>
<a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line">                <tt class="py-name">frag</tt> <tt class="py-op">=</tt> <tt id="link-128" class="py-name"><a title="Bio.SearchIO._model.hsp.HSPFragment" class="py-name" href="#" onclick="return doclink('link-128', 'HSPFragment', 'link-16');">HSPFragment</a></tt><tt class="py-op">(</tt><tt class="py-name">hid</tt><tt class="py-op">,</tt> <tt class="py-name">qid</tt><tt class="py-op">)</tt> </tt>
<a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line">                <tt class="py-comment"># set query and hit descriptions if they are defined / nonempty string</tt> </tt>
<a name="L274"></a><tt class="py-lineno">274</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">qdesc</tt><tt class="py-op">:</tt> </tt>
<a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line">                    <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-129" class="py-name"><a title="Bio.SearchIO._model.hit.Hit.query_description
Bio.SearchIO._model.hsp.HSP.query_description
Bio.SearchIO._model.hsp.HSPFragment.query_description" class="py-name" href="#" onclick="return doclink('link-129', 'query_description', 'link-120');">query_description</a></tt> <tt class="py-op">=</tt> <tt class="py-name">qdesc</tt> </tt>
<a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">hdesc</tt><tt class="py-op">:</tt> </tt>
<a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line">                    <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-130" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.hit_description=Bio.SearchIO._model.hsp.HSP-class.html#hit_description,Variable Bio.SearchIO._model.hsp.HSPFragment.hit_description=Bio.SearchIO._model.hsp.HSPFragment-class.html#hit_description"><a title="Bio.SearchIO._model.hsp.HSP.hit_description
Bio.SearchIO._model.hsp.HSPFragment.hit_description" class="py-name" href="#" onclick="return doclink('link-130', 'hit_description', 'link-130');">hit_description</a></tt> <tt class="py-op">=</tt> <tt class="py-name">hdesc</tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line">                <tt class="py-comment"># HMMER3 alphabets are always protein alphabets</tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line">                <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-131" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.alphabet=Bio.SearchIO._model.hsp.HSP-class.html#alphabet,Variable Bio.SearchIO._model.hsp.HSPFragment.alphabet=Bio.SearchIO._model.hsp.HSPFragment-class.html#alphabet"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-131', 'alphabet', 'link-131');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-132" class="py-name"><a title="Bio.Alphabet.generic_protein" class="py-name" href="#" onclick="return doclink('link-132', 'generic_protein', 'link-9');">generic_protein</a></tt> </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line">                <tt class="py-comment"># depending on whether the program is hmmsearch, hmmscan, or phmmer</tt> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line">                <tt class="py-comment"># {hmm,ali}{from,to} can either be hit_{from,to} or query_{from,to}</tt> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line">                <tt class="py-comment"># for hmmscan, hit is the hmm profile, query is the sequence</tt> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_meta</tt><tt class="py-op">.</tt><tt id="link-133" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-133', 'get', 'link-46');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'program'</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-string">'hmmscan'</tt><tt class="py-op">:</tt> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line">                    <tt class="py-comment"># adjust 'from' and 'to' coordinates to 0-based ones</tt> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line">                    <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-134" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.hit_start=Bio.SearchIO._model.hsp.HSP-class.html#hit_start,Variable Bio.SearchIO._model.hsp.HSPFragment.hit_start=Bio.SearchIO._model.hsp.HSPFragment-class.html#hit_start"><a title="Bio.SearchIO._model.hsp.HSP.hit_start
Bio.SearchIO._model.hsp.HSPFragment.hit_start" class="py-name" href="#" onclick="return doclink('link-134', 'hit_start', 'link-134');">hit_start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">parsed</tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line">                    <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-135" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.hit_end=Bio.SearchIO._model.hsp.HSP-class.html#hit_end,Variable Bio.SearchIO._model.hsp.HSPFragment.hit_end=Bio.SearchIO._model.hsp.HSPFragment-class.html#hit_end"><a title="Bio.SearchIO._model.hsp.HSP.hit_end
Bio.SearchIO._model.hsp.HSPFragment.hit_end" class="py-name" href="#" onclick="return doclink('link-135', 'hit_end', 'link-135');">hit_end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">parsed</tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line">                    <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-136" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.query_start=Bio.SearchIO._model.hsp.HSP-class.html#query_start,Variable Bio.SearchIO._model.hsp.HSPFragment.query_start=Bio.SearchIO._model.hsp.HSPFragment-class.html#query_start"><a title="Bio.SearchIO._model.hsp.HSP.query_start
Bio.SearchIO._model.hsp.HSPFragment.query_start" class="py-name" href="#" onclick="return doclink('link-136', 'query_start', 'link-136');">query_start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">parsed</tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">                    <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-137" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.query_end=Bio.SearchIO._model.hsp.HSP-class.html#query_end,Variable Bio.SearchIO._model.hsp.HSPFragment.query_end=Bio.SearchIO._model.hsp.HSPFragment-class.html#query_end"><a title="Bio.SearchIO._model.hsp.HSP.query_end
Bio.SearchIO._model.hsp.HSPFragment.query_end" class="py-name" href="#" onclick="return doclink('link-137', 'query_end', 'link-137');">query_end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">parsed</tt><tt class="py-op">[</tt><tt class="py-number">10</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_meta</tt><tt class="py-op">.</tt><tt id="link-138" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-138', 'get', 'link-46');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'program'</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">'hmmsearch'</tt><tt class="py-op">,</tt> <tt class="py-string">'phmmer'</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line">                    <tt class="py-comment"># adjust 'from' and 'to' coordinates to 0-based ones</tt> </tt>
<a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line">                    <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-139" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.hit_start
Bio.SearchIO._model.hsp.HSPFragment.hit_start" class="py-name" href="#" onclick="return doclink('link-139', 'hit_start', 'link-134');">hit_start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">parsed</tt><tt class="py-op">[</tt><tt class="py-number">9</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt> </tt>
<a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line">                    <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-140" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.hit_end
Bio.SearchIO._model.hsp.HSPFragment.hit_end" class="py-name" href="#" onclick="return doclink('link-140', 'hit_end', 'link-135');">hit_end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">parsed</tt><tt class="py-op">[</tt><tt class="py-number">10</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line">                    <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-141" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_start
Bio.SearchIO._model.hsp.HSPFragment.query_start" class="py-name" href="#" onclick="return doclink('link-141', 'query_start', 'link-136');">query_start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">parsed</tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line">                    <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-142" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_end
Bio.SearchIO._model.hsp.HSPFragment.query_end" class="py-name" href="#" onclick="return doclink('link-142', 'query_end', 'link-137');">query_end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">parsed</tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line">                <tt class="py-comment"># strand is always 0, since HMMER now only handles protein</tt> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line">                <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-143" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.hit_strand=Bio.SearchIO._model.hsp.HSP-class.html#hit_strand,Variable Bio.SearchIO._model.hsp.HSPFragment.hit_strand=Bio.SearchIO._model.hsp.HSPFragment-class.html#hit_strand"><a title="Bio.SearchIO._model.hsp.HSP.hit_strand
Bio.SearchIO._model.hsp.HSPFragment.hit_strand" class="py-name" href="#" onclick="return doclink('link-143', 'hit_strand', 'link-143');">hit_strand</a></tt> <tt class="py-op">=</tt> <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-144" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.query_strand=Bio.SearchIO._model.hsp.HSP-class.html#query_strand,Variable Bio.SearchIO._model.hsp.HSPFragment.query_strand=Bio.SearchIO._model.hsp.HSPFragment-class.html#query_strand"><a title="Bio.SearchIO._model.hsp.HSP.query_strand
Bio.SearchIO._model.hsp.HSPFragment.query_strand" class="py-name" href="#" onclick="return doclink('link-144', 'query_strand', 'link-144');">query_strand</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line"> </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line">                <tt id="link-145" class="py-name" targets="Module Bio.SearchIO._model.hsp=Bio.SearchIO._model.hsp-module.html"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-145', 'hsp', 'link-145');">hsp</a></tt> <tt class="py-op">=</tt> <tt id="link-146" class="py-name"><a title="Bio.Blast.Record.HSP
Bio.SearchIO._model.hsp.HSP" class="py-name" href="#" onclick="return doclink('link-146', 'HSP', 'link-15');">HSP</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">frag</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line">                <tt id="link-147" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-147', 'hsp', 'link-145');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">domain_index</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">parsed</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line">                <tt id="link-148" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-148', 'hsp', 'link-145');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">is_included</tt> <tt class="py-op">=</tt> <tt class="py-name">parsed</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'!'</tt> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line">                <tt id="link-149" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-149', 'hsp', 'link-145');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">bitscore</tt> <tt class="py-op">=</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">parsed</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line">                <tt id="link-150" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-150', 'hsp', 'link-145');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">bias</tt> <tt class="py-op">=</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">parsed</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line">                <tt id="link-151" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-151', 'hsp', 'link-145');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">evalue_cond</tt> <tt class="py-op">=</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">parsed</tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line">                <tt id="link-152" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-152', 'hsp', 'link-145');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">evalue</tt> <tt class="py-op">=</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">parsed</tt><tt class="py-op">[</tt><tt class="py-number">5</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_meta</tt><tt class="py-op">.</tt><tt id="link-153" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-153', 'get', 'link-46');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'program'</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-string">'hmmscan'</tt><tt class="py-op">:</tt> </tt>
<a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line">                    <tt class="py-comment"># adjust 'from' and 'to' coordinates to 0-based ones</tt> </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line">                    <tt id="link-154" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-154', 'hsp', 'link-145');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">hit_endtype</tt> <tt class="py-op">=</tt> <tt class="py-name">parsed</tt><tt class="py-op">[</tt><tt class="py-number">8</tt><tt class="py-op">]</tt> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line">                    <tt id="link-155" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-155', 'hsp', 'link-145');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">query_endtype</tt> <tt class="py-op">=</tt> <tt class="py-name">parsed</tt><tt class="py-op">[</tt><tt class="py-number">11</tt><tt class="py-op">]</tt> </tt>
<a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_meta</tt><tt class="py-op">.</tt><tt id="link-156" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-156', 'get', 'link-46');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'program'</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">'hmmsearch'</tt><tt class="py-op">,</tt> <tt class="py-string">'phmmer'</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line">                    <tt class="py-comment"># adjust 'from' and 'to' coordinates to 0-based ones</tt> </tt>
<a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line">                    <tt id="link-157" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-157', 'hsp', 'link-145');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">hit_endtype</tt> <tt class="py-op">=</tt> <tt class="py-name">parsed</tt><tt class="py-op">[</tt><tt class="py-number">11</tt><tt class="py-op">]</tt> </tt>
<a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line">                    <tt id="link-158" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-158', 'hsp', 'link-145');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">query_endtype</tt> <tt class="py-op">=</tt> <tt class="py-name">parsed</tt><tt class="py-op">[</tt><tt class="py-number">8</tt><tt class="py-op">]</tt> </tt>
<a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line">                <tt class="py-comment"># adjust 'from' and 'to' coordinates to 0-based ones</tt> </tt>
<a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line">                <tt id="link-159" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-159', 'hsp', 'link-145');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">env_start</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">parsed</tt><tt class="py-op">[</tt><tt class="py-number">12</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt> </tt>
<a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line">                <tt id="link-160" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-160', 'hsp', 'link-145');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">env_end</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">parsed</tt><tt class="py-op">[</tt><tt class="py-number">13</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line">                <tt id="link-161" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-161', 'hsp', 'link-145');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">env_endtype</tt> <tt class="py-op">=</tt> <tt class="py-name">parsed</tt><tt class="py-op">[</tt><tt class="py-number">14</tt><tt class="py-op">]</tt> </tt>
<a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line">                <tt id="link-162" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-162', 'hsp', 'link-145');">hsp</a></tt><tt class="py-op">.</tt><tt class="py-name">acc_avg</tt> <tt class="py-op">=</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">parsed</tt><tt class="py-op">[</tt><tt class="py-number">15</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line"> </tt>
<a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line">                <tt class="py-name">hsp_list</tt><tt class="py-op">.</tt><tt id="link-163" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-163', 'append', 'link-88');">append</a></tt><tt class="py-op">(</tt><tt id="link-164" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-164', 'hsp', 'link-145');">hsp</a></tt><tt class="py-op">)</tt> </tt>
<a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-165" class="py-name"><a title="Bio._utils.read_forward" class="py-name" href="#" onclick="return doclink('link-165', 'read_forward', 'link-6');">read_forward</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line"> </tt>
<a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line">            <tt class="py-comment"># parse the hsp alignments</tt> </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-166" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-166', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'  Alignments for each domain:'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-167" class="py-name" targets="Method Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser._parse_aln_block()=Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html#_parse_aln_block"><a title="Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser._parse_aln_block" class="py-name" href="#" onclick="return doclink('link-167', '_parse_aln_block', 'link-167');">_parse_aln_block</a></tt><tt class="py-op">(</tt><tt class="py-name">hid</tt><tt class="py-op">,</tt> <tt id="link-168" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-168', 'hit', 'link-108');">hit</a></tt><tt class="py-op">.</tt><tt id="link-169" class="py-name" targets="Variable Bio.SearchIO._model.hit.Hit.hsps=Bio.SearchIO._model.hit.Hit-class.html#hsps,Variable Bio.SearchIO._model.query.QueryResult.hsps=Bio.SearchIO._model.query.QueryResult-class.html#hsps"><a title="Bio.SearchIO._model.hit.Hit.hsps
Bio.SearchIO._model.query.QueryResult.hsps" class="py-name" href="#" onclick="return doclink('link-169', 'hsps', 'link-169');">hsps</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L325"></a><tt class="py-lineno">325</tt>  <tt class="py-line"> </tt>
<a name="Hmmer3TextParser._parse_aln_block"></a><div id="Hmmer3TextParser._parse_aln_block-def"><a name="L326"></a><tt class="py-lineno">326</tt> <a class="py-toggle" href="#" id="Hmmer3TextParser._parse_aln_block-toggle" onclick="return toggle('Hmmer3TextParser._parse_aln_block');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html#_parse_aln_block">_parse_aln_block</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">hid</tt><tt class="py-op">,</tt> <tt class="py-param">hsp_list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Hmmer3TextParser._parse_aln_block-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Hmmer3TextParser._parse_aln_block-expanded"><a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line">        <tt class="py-docstring">"""Parses a HMMER3 HSP alignment block."""</tt> </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-170" class="py-name"><a title="Bio._utils.read_forward" class="py-name" href="#" onclick="return doclink('link-170', 'read_forward', 'link-6');">read_forward</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line">        <tt class="py-name">dom_counter</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-171" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-171', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'&gt;&gt;'</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> \ </tt>
<a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-172" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-172', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Internal pipeline'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L333"></a><tt class="py-lineno">333</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">hsp_list</tt> </tt>
<a name="L334"></a><tt class="py-lineno">334</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-173" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-173', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'  == domain %i'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">dom_counter</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L335"></a><tt class="py-lineno">335</tt>  <tt class="py-line">            <tt class="py-comment"># alias hsp to local var</tt> </tt>
<a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line">            <tt class="py-comment"># but note that we're still changing the attrs of the actual</tt> </tt>
<a name="L337"></a><tt class="py-lineno">337</tt>  <tt class="py-line">            <tt class="py-comment"># hsp inside the qresult as we're not creating a copy</tt> </tt>
<a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line">            <tt class="py-name">frag</tt> <tt class="py-op">=</tt> <tt class="py-name">hsp_list</tt><tt class="py-op">[</tt><tt class="py-name">dom_counter</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L339"></a><tt class="py-lineno">339</tt>  <tt class="py-line">            <tt class="py-comment"># XXX: should we validate again here? regex is expensive..</tt> </tt>
<a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line">            <tt class="py-comment"># regx = re.search(_HRE_VALIDATE, self.line)</tt> </tt>
<a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line">            <tt class="py-comment"># assert hsp.bitscore == float(regx.group(1))</tt> </tt>
<a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line">            <tt class="py-comment"># assert hsp.evalue_cond == float(regx.group(2))</tt> </tt>
<a name="L343"></a><tt class="py-lineno">343</tt>  <tt class="py-line">            <tt class="py-name">hmmseq</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L344"></a><tt class="py-lineno">344</tt>  <tt class="py-line">            <tt class="py-name">aliseq</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L345"></a><tt class="py-lineno">345</tt>  <tt class="py-line">            <tt class="py-name">annot</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L346"></a><tt class="py-lineno">346</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-174" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-174', 'readline', 'link-72');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L347"></a><tt class="py-lineno">347</tt>  <tt class="py-line"> </tt>
<a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line">            <tt class="py-comment"># parse all the alignment blocks in the hsp</tt> </tt>
<a name="L349"></a><tt class="py-lineno">349</tt>  <tt class="py-line">            <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line"> </tt>
<a name="L351"></a><tt class="py-lineno">351</tt>  <tt class="py-line">                <tt class="py-name">regx</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L352"></a><tt class="py-lineno">352</tt>  <tt class="py-line"> </tt>
<a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line">                <tt class="py-comment"># check for hit or query line</tt> </tt>
<a name="L354"></a><tt class="py-lineno">354</tt>  <tt class="py-line">                <tt class="py-comment"># we don't check for the hit or query id specifically</tt> </tt>
<a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line">                <tt class="py-comment"># to anticipate special cases where query id == hit id</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line">                <tt class="py-name">regx</tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-175" class="py-name"><a title="Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-175', 'search', 'link-33');">search</a></tt><tt class="py-op">(</tt><tt id="link-176" class="py-name"><a title="Bio.SearchIO.HmmerIO.hmmer3_text._HRE_ID_LINE" class="py-name" href="#" onclick="return doclink('link-176', '_HRE_ID_LINE', 'link-27');">_HRE_ID_LINE</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">regx</tt><tt class="py-op">:</tt> </tt>
<a name="L358"></a><tt class="py-lineno">358</tt>  <tt class="py-line">                    <tt class="py-comment"># the first hit/query self.line we encounter is the hmmseq</tt> </tt>
<a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">hmmseq</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">aliseq</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line">                        <tt class="py-name">hmmseq</tt> <tt class="py-op">+=</tt> <tt class="py-name">regx</tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">2</tt><tt class="py-op">)</tt> </tt>
<a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line">                    <tt class="py-comment"># and for subsequent self.lines, len(hmmseq) is either</tt> </tt>
<a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line">                    <tt class="py-comment"># &gt; or == len(aliseq)</tt> </tt>
<a name="L363"></a><tt class="py-lineno">363</tt>  <tt class="py-line">                    <tt class="py-keyword">elif</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">hmmseq</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">aliseq</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L364"></a><tt class="py-lineno">364</tt>  <tt class="py-line">                        <tt class="py-name">aliseq</tt> <tt class="py-op">+=</tt> <tt class="py-name">regx</tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">2</tt><tt class="py-op">)</tt> </tt>
<a name="L365"></a><tt class="py-lineno">365</tt>  <tt class="py-line">                    <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">hmmseq</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">aliseq</tt><tt class="py-op">)</tt> </tt>
<a name="L366"></a><tt class="py-lineno">366</tt>  <tt class="py-line">                <tt class="py-comment"># check for start of new domain</tt> </tt>
<a name="L367"></a><tt class="py-lineno">367</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-177" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-177', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'  == domain'</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> \ </tt>
<a name="L368"></a><tt class="py-lineno">368</tt>  <tt class="py-line">                        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-178" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-178', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'&gt;&gt;'</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> \ </tt>
<a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line">                        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-179" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-179', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Internal pipeline'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line">                    <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-180" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.aln_annotation=Bio.SearchIO._model.hsp.HSP-class.html#aln_annotation"><a title="Bio.SearchIO._model.hsp.HSP.aln_annotation" class="py-name" href="#" onclick="return doclink('link-180', 'aln_annotation', 'link-180');">aln_annotation</a></tt> <tt class="py-op">=</tt> <tt class="py-name">annot</tt> </tt>
<a name="L371"></a><tt class="py-lineno">371</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_meta</tt><tt class="py-op">.</tt><tt id="link-181" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-181', 'get', 'link-46');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'program'</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-string">'hmmscan'</tt><tt class="py-op">:</tt> </tt>
<a name="L372"></a><tt class="py-lineno">372</tt>  <tt class="py-line">                        <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-182" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-182', 'hit', 'link-108');">hit</a></tt> <tt class="py-op">=</tt> <tt class="py-name">hmmseq</tt> </tt>
<a name="L373"></a><tt class="py-lineno">373</tt>  <tt class="py-line">                        <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-183" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HSPConsumer.query()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#query,Variable Bio.SearchIO._model.hsp.HSP.query=Bio.SearchIO._model.hsp.HSP-class.html#query,Variable Bio.SearchIO._model.hsp.HSPFragment.query=Bio.SearchIO._model.hsp.HSPFragment-class.html#query,Module Bio.SearchIO._model.query=Bio.SearchIO._model.query-module.html"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.query
Bio.SearchIO._model.hsp.HSP.query
Bio.SearchIO._model.hsp.HSPFragment.query
Bio.SearchIO._model.query" class="py-name" href="#" onclick="return doclink('link-183', 'query', 'link-183');">query</a></tt> <tt class="py-op">=</tt> <tt class="py-name">aliseq</tt> </tt>
<a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line">                    <tt class="py-keyword">elif</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_meta</tt><tt class="py-op">.</tt><tt id="link-184" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-184', 'get', 'link-46');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'program'</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">'hmmsearch'</tt><tt class="py-op">,</tt> <tt class="py-string">'phmmer'</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line">                        <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-185" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-185', 'hit', 'link-108');">hit</a></tt> <tt class="py-op">=</tt> <tt class="py-name">aliseq</tt> </tt>
<a name="L376"></a><tt class="py-lineno">376</tt>  <tt class="py-line">                        <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-186" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.query
Bio.SearchIO._model.hsp.HSP.query
Bio.SearchIO._model.hsp.HSPFragment.query
Bio.SearchIO._model.query" class="py-name" href="#" onclick="return doclink('link-186', 'query', 'link-183');">query</a></tt> <tt class="py-op">=</tt> <tt class="py-name">hmmseq</tt> </tt>
<a name="L377"></a><tt class="py-lineno">377</tt>  <tt class="py-line">                    <tt class="py-name">dom_counter</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L378"></a><tt class="py-lineno">378</tt>  <tt class="py-line">                    <tt class="py-name">hmmseq</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L379"></a><tt class="py-lineno">379</tt>  <tt class="py-line">                    <tt class="py-name">aliseq</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L380"></a><tt class="py-lineno">380</tt>  <tt class="py-line">                    <tt class="py-name">annot</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L381"></a><tt class="py-lineno">381</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L382"></a><tt class="py-lineno">382</tt>  <tt class="py-line">                <tt class="py-comment"># otherwise check if it's an annotation line and parse it</tt> </tt>
<a name="L383"></a><tt class="py-lineno">383</tt>  <tt class="py-line">                <tt class="py-comment"># len(hmmseq) is only != len(aliseq) when the cursor is parsing</tt> </tt>
<a name="L384"></a><tt class="py-lineno">384</tt>  <tt class="py-line">                <tt class="py-comment"># the similarity character. Since we're not parsing that, we</tt> </tt>
<a name="L385"></a><tt class="py-lineno">385</tt>  <tt class="py-line">                <tt class="py-comment"># check for when the condition is False (i.e. when it's ==)</tt> </tt>
<a name="L386"></a><tt class="py-lineno">386</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">hmmseq</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">aliseq</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L387"></a><tt class="py-lineno">387</tt>  <tt class="py-line">                    <tt class="py-name">regx</tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-187" class="py-name"><a title="Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-187', 'search', 'link-33');">search</a></tt><tt class="py-op">(</tt><tt id="link-188" class="py-name"><a title="Bio.SearchIO.HmmerIO.hmmer3_text._HRE_ANNOT_LINE" class="py-name" href="#" onclick="return doclink('link-188', '_HRE_ANNOT_LINE', 'link-26');">_HRE_ANNOT_LINE</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L388"></a><tt class="py-lineno">388</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">regx</tt><tt class="py-op">:</tt> </tt>
<a name="L389"></a><tt class="py-lineno">389</tt>  <tt class="py-line">                        <tt class="py-name">annot_name</tt> <tt class="py-op">=</tt> <tt class="py-name">regx</tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">3</tt><tt class="py-op">)</tt> </tt>
<a name="L390"></a><tt class="py-lineno">390</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">annot_name</tt> <tt class="py-keyword">in</tt> <tt class="py-name">annot</tt><tt class="py-op">:</tt> </tt>
<a name="L391"></a><tt class="py-lineno">391</tt>  <tt class="py-line">                            <tt class="py-name">annot</tt><tt class="py-op">[</tt><tt class="py-name">annot_name</tt><tt class="py-op">]</tt> <tt class="py-op">+=</tt> <tt class="py-name">regx</tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">2</tt><tt class="py-op">)</tt> </tt>
<a name="L392"></a><tt class="py-lineno">392</tt>  <tt class="py-line">                        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L393"></a><tt class="py-lineno">393</tt>  <tt class="py-line">                            <tt class="py-name">annot</tt><tt class="py-op">[</tt><tt class="py-name">annot_name</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">regx</tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">2</tt><tt class="py-op">)</tt> </tt>
<a name="L394"></a><tt class="py-lineno">394</tt>  <tt class="py-line"> </tt>
<a name="L395"></a><tt class="py-lineno">395</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-189" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-189', 'readline', 'link-72');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L396"></a><tt class="py-lineno">396</tt>  <tt class="py-line"> </tt>
<a name="L397"></a><tt class="py-lineno">397</tt>  <tt class="py-line"> </tt>
<a name="Hmmer3TextIndexer"></a><div id="Hmmer3TextIndexer-def"><a name="L398"></a><tt class="py-lineno">398</tt> <a class="py-toggle" href="#" id="Hmmer3TextIndexer-toggle" onclick="return toggle('Hmmer3TextIndexer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextIndexer-class.html">Hmmer3TextIndexer</a><tt class="py-op">(</tt><tt class="py-base-class">_BaseHmmerTextIndexer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Hmmer3TextIndexer-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Hmmer3TextIndexer-expanded"><a name="L399"></a><tt class="py-lineno">399</tt>  <tt class="py-line"> </tt>
<a name="L400"></a><tt class="py-lineno">400</tt>  <tt class="py-line">    <tt class="py-docstring">"""Indexer class for HMMER plain text output."""</tt> </tt>
<a name="L401"></a><tt class="py-lineno">401</tt>  <tt class="py-line"> </tt>
<a name="L402"></a><tt class="py-lineno">402</tt>  <tt class="py-line">    <tt id="link-190" class="py-name" targets="Variable Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer._parser=Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer-class.html#_parser"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer._parser" class="py-name" href="#" onclick="return doclink('link-190', '_parser', 'link-190');">_parser</a></tt> <tt class="py-op">=</tt> <tt id="link-191" class="py-name" targets="Class Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser=Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html"><a title="Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser" class="py-name" href="#" onclick="return doclink('link-191', 'Hmmer3TextParser', 'link-191');">Hmmer3TextParser</a></tt> </tt>
<a name="L403"></a><tt class="py-lineno">403</tt>  <tt class="py-line">    <tt id="link-192" class="py-name" targets="Variable Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextIndexer.qresult_start=Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextIndexer-class.html#qresult_start,Variable Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextIndexer.qresult_start=Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextIndexer-class.html#qresult_start"><a title="Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextIndexer.qresult_start
Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextIndexer.qresult_start" class="py-name" href="#" onclick="return doclink('link-192', 'qresult_start', 'link-192');">qresult_start</a></tt> <tt class="py-op">=</tt> <tt id="link-193" class="py-name"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-193', '_as_bytes', 'link-2');">_as_bytes</a></tt><tt class="py-op">(</tt><tt class="py-string">'Query: '</tt><tt class="py-op">)</tt> </tt>
<a name="L404"></a><tt class="py-lineno">404</tt>  <tt class="py-line">    <tt id="link-194" class="py-name" targets="Variable Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextIndexer.qresult_end=Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextIndexer-class.html#qresult_end,Variable Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextIndexer.qresult_end=Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextIndexer-class.html#qresult_end"><a title="Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextIndexer.qresult_end
Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextIndexer.qresult_end" class="py-name" href="#" onclick="return doclink('link-194', 'qresult_end', 'link-194');">qresult_end</a></tt> <tt class="py-op">=</tt> <tt id="link-195" class="py-name"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-195', '_as_bytes', 'link-2');">_as_bytes</a></tt><tt class="py-op">(</tt><tt class="py-string">'//'</tt><tt class="py-op">)</tt> </tt>
<a name="L405"></a><tt class="py-lineno">405</tt>  <tt class="py-line"> </tt>
<a name="Hmmer3TextIndexer.__iter__"></a><div id="Hmmer3TextIndexer.__iter__-def"><a name="L406"></a><tt class="py-lineno">406</tt> <a class="py-toggle" href="#" id="Hmmer3TextIndexer.__iter__-toggle" onclick="return toggle('Hmmer3TextIndexer.__iter__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextIndexer-class.html#__iter__">__iter__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Hmmer3TextIndexer.__iter__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Hmmer3TextIndexer.__iter__-expanded"><a name="L407"></a><tt class="py-lineno">407</tt>  <tt class="py-line">        <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_handle</tt> </tt>
<a name="L408"></a><tt class="py-lineno">408</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-196" class="py-name" targets="Method Bio.File.UndoHandle.seek()=Bio.File.UndoHandle-class.html#seek,Method Bio.SeqIO.SffIO._AddTellHandle.seek()=Bio.SeqIO.SffIO._AddTellHandle-class.html#seek,Method Bio.bgzf.BgzfReader.seek()=Bio.bgzf.BgzfReader-class.html#seek"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-196', 'seek', 'link-196');">seek</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L409"></a><tt class="py-lineno">409</tt>  <tt class="py-line">        <tt class="py-name">start_offset</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-197" class="py-name" targets="Method Bio.File.UndoHandle.tell()=Bio.File.UndoHandle-class.html#tell,Method Bio.SeqIO.SffIO._AddTellHandle.tell()=Bio.SeqIO.SffIO._AddTellHandle-class.html#tell,Method Bio.bgzf.BgzfReader.tell()=Bio.bgzf.BgzfReader-class.html#tell,Method Bio.bgzf.BgzfWriter.tell()=Bio.bgzf.BgzfWriter-class.html#tell"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-197', 'tell', 'link-197');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L410"></a><tt class="py-lineno">410</tt>  <tt class="py-line">        <tt class="py-name">regex_id</tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt id="link-198" class="py-name"><a title="Bio._py3k._as_bytes" class="py-name" href="#" onclick="return doclink('link-198', '_as_bytes', 'link-2');">_as_bytes</a></tt><tt class="py-op">(</tt><tt id="link-199" class="py-name"><a title="Bio.SearchIO.HmmerIO.hmmer3_text._QRE_ID_LEN_PTN" class="py-name" href="#" onclick="return doclink('link-199', '_QRE_ID_LEN_PTN', 'link-22');">_QRE_ID_LEN_PTN</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L411"></a><tt class="py-lineno">411</tt>  <tt class="py-line"> </tt>
<a name="L412"></a><tt class="py-lineno">412</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L413"></a><tt class="py-lineno">413</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-200" class="py-name"><a title="Bio._utils.read_forward" class="py-name" href="#" onclick="return doclink('link-200', 'read_forward', 'link-6');">read_forward</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L414"></a><tt class="py-lineno">414</tt>  <tt class="py-line">            <tt class="py-name">end_offset</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-201" class="py-name"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-201', 'tell', 'link-197');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L415"></a><tt class="py-lineno">415</tt>  <tt class="py-line"> </tt>
<a name="L416"></a><tt class="py-lineno">416</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-202" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-202', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-203" class="py-name"><a title="Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextIndexer.qresult_start
Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextIndexer.qresult_start" class="py-name" href="#" onclick="return doclink('link-203', 'qresult_start', 'link-192');">qresult_start</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L417"></a><tt class="py-lineno">417</tt>  <tt class="py-line">                <tt class="py-name">regx</tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-204" class="py-name"><a title="Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-204', 'search', 'link-33');">search</a></tt><tt class="py-op">(</tt><tt class="py-name">regex_id</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L418"></a><tt class="py-lineno">418</tt>  <tt class="py-line">                <tt class="py-name">qresult_key</tt> <tt class="py-op">=</tt> <tt class="py-name">regx</tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-205" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-205', 'strip', 'link-39');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L419"></a><tt class="py-lineno">419</tt>  <tt class="py-line">                <tt class="py-comment"># qresult start offset is the offset of this line</tt> </tt>
<a name="L420"></a><tt class="py-lineno">420</tt>  <tt class="py-line">                <tt class="py-comment"># (starts with the start mark)</tt> </tt>
<a name="L421"></a><tt class="py-lineno">421</tt>  <tt class="py-line">                <tt class="py-name">start_offset</tt> <tt class="py-op">=</tt> <tt class="py-name">end_offset</tt> <tt class="py-op">-</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L422"></a><tt class="py-lineno">422</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-206" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-206', 'startswith', 'link-32');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-207" class="py-name"><a title="Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextIndexer.qresult_end
Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextIndexer.qresult_end" class="py-name" href="#" onclick="return doclink('link-207', 'qresult_end', 'link-194');">qresult_end</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L423"></a><tt class="py-lineno">423</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt id="link-208" class="py-name"><a title="Bio._py3k._bytes_to_string" class="py-name" href="#" onclick="return doclink('link-208', '_bytes_to_string', 'link-3');">_bytes_to_string</a></tt><tt class="py-op">(</tt><tt class="py-name">qresult_key</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">start_offset</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt> </tt>
<a name="L424"></a><tt class="py-lineno">424</tt>  <tt class="py-line">                <tt class="py-name">start_offset</tt> <tt class="py-op">=</tt> <tt class="py-name">end_offset</tt> </tt>
<a name="L425"></a><tt class="py-lineno">425</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L426"></a><tt class="py-lineno">426</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
</div></div><a name="L427"></a><tt class="py-lineno">427</tt>  <tt class="py-line"> </tt>
<a name="L428"></a><tt class="py-lineno">428</tt>  <tt class="py-line"><tt class="py-comment"># if not used as a module, run the doctest</tt> </tt>
<a name="L429"></a><tt class="py-lineno">429</tt>  <tt class="py-line"><tt class="py-keyword">if</tt> <tt class="py-name">__name__</tt> <tt class="py-op">==</tt> <tt class="py-string">"__main__"</tt><tt class="py-op">:</tt> </tt>
<a name="L430"></a><tt class="py-lineno">430</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-209" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-209', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-210" class="py-name"><a title="Bio.Phylo._utils
Bio.SearchIO._utils
Bio._utils" class="py-name" href="#" onclick="return doclink('link-210', '_utils', 'link-5');">_utils</a></tt> <tt class="py-keyword">import</tt> <tt id="link-211" class="py-name" targets="Function Bio._utils.run_doctest()=Bio._utils-module.html#run_doctest"><a title="Bio._utils.run_doctest" class="py-name" href="#" onclick="return doclink('link-211', 'run_doctest', 'link-211');">run_doctest</a></tt> </tt>
<a name="L431"></a><tt class="py-lineno">431</tt>  <tt class="py-line">    <tt id="link-212" class="py-name"><a title="Bio._utils.run_doctest" class="py-name" href="#" onclick="return doclink('link-212', 'run_doctest', 'link-211');">run_doctest</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L432"></a><tt class="py-lineno">432</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Oct 21 18:04:31 2015
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
