<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.Phylo.CDAOIO</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.Phylo-module.html">Package&nbsp;Phylo</a> ::
        Module&nbsp;CDAOIO
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.Phylo.CDAOIO-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.Phylo.CDAOIO-module.html">Module Bio.Phylo.CDAOIO</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright (C) 2013 by Ben Morris (ben@bendmorris.com)</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># Based on Bio.Nexus, copyright 2005-2008 by Frank Kauff &amp; Cymon J. Cox</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># and Bio.Phylo.Newick, copyright 2009 by Eric Talevich.</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># All rights reserved.</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-comment"># license. Please see the LICENSE file that should have been included</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-docstring">"""I/O function wrappers for the RDF/CDAO file format.</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-docstring">This is an RDF format that conforms to the Comparative Data Analysis Ontology (CDAO).</tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-docstring">See: http://evolutionaryontology.org/cdao</tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-docstring">This module requires the librdf Python bindings (http://www.librdf.org)</tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-docstring">The CDAOIO.Parser, in addition to parsing text files, can also parse directly</tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-docstring">from a triple store that implements the Redland storage interface; similarly,</tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"><tt class="py-docstring">the CDAOIO.Writer can store triples in a triple store instead of serializing</tt> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-docstring">them to a file.</tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">os</tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-1', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">StringIO</tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-2" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-2', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-3" class="py-name" targets="Class Bio.MissingPythonDependencyError=Bio.MissingPythonDependencyError-class.html"><a title="Bio.MissingPythonDependencyError" class="py-name" href="#" onclick="return doclink('link-3', 'MissingPythonDependencyError', 'link-3');">MissingPythonDependencyError</a></tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-4" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-4', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-5" class="py-name" targets="Package Bio.Phylo=Bio.Phylo-module.html"><a title="Bio.Phylo" class="py-name" href="#" onclick="return doclink('link-5', 'Phylo', 'link-5');">Phylo</a></tt> <tt class="py-keyword">import</tt> <tt id="link-6" class="py-name" targets="Module Bio.Phylo.CDAO=Bio.Phylo.CDAO-module.html"><a title="Bio.Phylo.CDAO" class="py-name" href="#" onclick="return doclink('link-6', 'CDAO', 'link-6');">CDAO</a></tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt><tt id="link-7" class="py-name" targets="Module Bio.Phylo._cdao_owl=Bio.Phylo._cdao_owl-module.html"><a title="Bio.Phylo._cdao_owl" class="py-name" href="#" onclick="return doclink('link-7', '_cdao_owl', 'link-7');">_cdao_owl</a></tt> <tt class="py-keyword">import</tt> <tt id="link-8" class="py-name" targets="Variable Bio.Phylo._cdao_owl.cdao_namespaces=Bio.Phylo._cdao_owl-module.html#cdao_namespaces"><a title="Bio.Phylo._cdao_owl.cdao_namespaces" class="py-name" href="#" onclick="return doclink('link-8', 'cdao_namespaces', 'link-8');">cdao_namespaces</a></tt><tt class="py-op">,</tt> <tt id="link-9" class="py-name" targets="Function Bio.Phylo._cdao_owl.resolve_uri()=Bio.Phylo._cdao_owl-module.html#resolve_uri"><a title="Bio.Phylo._cdao_owl.resolve_uri" class="py-name" href="#" onclick="return doclink('link-9', 'resolve_uri', 'link-9');">resolve_uri</a></tt> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"><tt class="py-comment"># import of cdao_elements from ._cdao_owl removed in Biopython 1.74</tt> </tt>
<a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line"> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line">    <tt class="py-keyword">import</tt> <tt class="py-name">rdflib</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line">    <tt id="link-10" class="py-name" targets="Variable Bio.Phylo.CDAOIO.rdfver=Bio.Phylo.CDAOIO-module.html#rdfver"><a title="Bio.Phylo.CDAOIO.rdfver" class="py-name" href="#" onclick="return doclink('link-10', 'rdfver', 'link-10');">rdfver</a></tt> <tt class="py-op">=</tt> <tt class="py-name">rdflib</tt><tt class="py-op">.</tt><tt class="py-name">__version__</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt id="link-11" class="py-name"><a title="Bio.Phylo.CDAOIO.rdfver" class="py-name" href="#" onclick="return doclink('link-11', 'rdfver', 'link-10');">rdfver</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">"1"</tt><tt class="py-op">,</tt> <tt class="py-string">"2"</tt><tt class="py-op">]</tt> <tt class="py-keyword">or</tt> <tt class="py-op">(</tt><tt id="link-12" class="py-name"><a title="Bio.Phylo.CDAOIO.rdfver" class="py-name" href="#" onclick="return doclink('link-12', 'rdfver', 'link-10');">rdfver</a></tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">"3.0.0"</tt><tt class="py-op">,</tt> <tt class="py-string">"3.1.0"</tt><tt class="py-op">,</tt> <tt class="py-string">"3.2.0"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt id="link-13" class="py-name"><a title="Bio.MissingPythonDependencyError" class="py-name" href="#" onclick="return doclink('link-13', 'MissingPythonDependencyError', 'link-3');">MissingPythonDependencyError</a></tt><tt class="py-op">(</tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line">            <tt class="py-string">"Support for CDAO tree format requires RDFlib v3.2.1 or later."</tt><tt class="py-op">)</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line"><tt class="py-keyword">except</tt> <tt class="py-name">ImportError</tt><tt class="py-op">:</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line">    <tt class="py-keyword">raise</tt> <tt id="link-14" class="py-name"><a title="Bio.MissingPythonDependencyError" class="py-name" href="#" onclick="return doclink('link-14', 'MissingPythonDependencyError', 'link-3');">MissingPythonDependencyError</a></tt><tt class="py-op">(</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line">        <tt class="py-string">"Support for CDAO tree format requires RDFlib."</tt><tt class="py-op">)</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line"> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line"><tt id="link-15" class="py-name" targets="Variable Bio.Phylo.CDAOIO.RDF_NAMESPACES=Bio.Phylo.CDAOIO-module.html#RDF_NAMESPACES"><a title="Bio.Phylo.CDAOIO.RDF_NAMESPACES" class="py-name" href="#" onclick="return doclink('link-15', 'RDF_NAMESPACES', 'link-15');">RDF_NAMESPACES</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line">    <tt class="py-string">"owl"</tt><tt class="py-op">:</tt> <tt class="py-string">"http://www.w3.org/2002/07/owl#"</tt><tt class="py-op">,</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line">    <tt class="py-string">"rdf"</tt><tt class="py-op">:</tt> <tt class="py-string">"http://www.w3.org/1999/02/22-rdf-syntax-ns#"</tt><tt class="py-op">,</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line">    <tt class="py-string">"rdfs"</tt><tt class="py-op">:</tt> <tt class="py-string">"http://www.w3.org/2000/01/rdf-schema#"</tt><tt class="py-op">,</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line"><tt class="py-op">}</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line"><tt id="link-16" class="py-name"><a title="Bio.Phylo.CDAOIO.RDF_NAMESPACES" class="py-name" href="#" onclick="return doclink('link-16', 'RDF_NAMESPACES', 'link-15');">RDF_NAMESPACES</a></tt><tt class="py-op">.</tt><tt id="link-17" class="py-name" targets="Method Bio.Align.substitution_matrices.Array.update()=Bio.Align.substitution_matrices.Array-class.html#update,Method Bio.File._IndexedSeqFileDict.update()=Bio.File._IndexedSeqFileDict-class.html#update,Method Bio.Index._InMemoryIndex.update()=Bio.Index._InMemoryIndex-class.html#update,Method Bio.SeqRecord._RestrictedDict.update()=Bio.SeqRecord._RestrictedDict-class.html#update"><a title="Bio.Align.substitution_matrices.Array.update
Bio.File._IndexedSeqFileDict.update
Bio.Index._InMemoryIndex.update
Bio.SeqRecord._RestrictedDict.update" class="py-name" href="#" onclick="return doclink('link-17', 'update', 'link-17');">update</a></tt><tt class="py-op">(</tt><tt id="link-18" class="py-name"><a title="Bio.Phylo._cdao_owl.cdao_namespaces" class="py-name" href="#" onclick="return doclink('link-18', 'cdao_namespaces', 'link-8');">cdao_namespaces</a></tt><tt class="py-op">)</tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line"><tt class="py-comment"># pad node ids with zeroes until they're at least this length</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line"><tt id="link-19" class="py-name" targets="Variable Bio.Phylo.CDAOIO.ZEROES=Bio.Phylo.CDAOIO-module.html#ZEROES"><a title="Bio.Phylo.CDAOIO.ZEROES" class="py-name" href="#" onclick="return doclink('link-19', 'ZEROES', 'link-19');">ZEROES</a></tt> <tt class="py-op">=</tt> <tt class="py-number">8</tt> </tt>
<a name="qUri"></a><div id="qUri-def"><a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line"> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line"> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt> <a class="py-toggle" href="#" id="qUri-toggle" onclick="return toggle('qUri');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.CDAOIO-module.html#qUri">qUri</a><tt class="py-op">(</tt><tt class="py-param">x</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="qUri-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="qUri-expanded"><a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line">    <tt class="py-docstring">"""Resolve URI for librdf."""</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-20" class="py-name"><a title="Bio.Phylo._cdao_owl.resolve_uri" class="py-name" href="#" onclick="return doclink('link-20', 'resolve_uri', 'link-9');">resolve_uri</a></tt><tt class="py-op">(</tt><tt id="link-21" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.x=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#x,Variable Bio.Statistics.lowess.x=Bio.Statistics.lowess-module.html#x"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-21', 'x', 'link-21');">x</a></tt><tt class="py-op">,</tt> <tt class="py-name">namespaces</tt><tt class="py-op">=</tt><tt id="link-22" class="py-name"><a title="Bio.Phylo.CDAOIO.RDF_NAMESPACES" class="py-name" href="#" onclick="return doclink('link-22', 'RDF_NAMESPACES', 'link-15');">RDF_NAMESPACES</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line"> </tt>
<a name="format_label"></a><div id="format_label-def"><a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt> <a class="py-toggle" href="#" id="format_label-toggle" onclick="return toggle('format_label');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.CDAOIO-module.html#format_label">format_label</a><tt class="py-op">(</tt><tt class="py-param">x</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="format_label-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="format_label-expanded"><a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line">    <tt class="py-docstring">"""Format label for librdf."""</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-23" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-23', 'x', 'link-21');">x</a></tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"_"</tt><tt class="py-op">,</tt> <tt class="py-string">" "</tt><tt class="py-op">)</tt> </tt>
</div><a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line"> </tt>
<a name="parse"></a><div id="parse-def"><a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line"> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line"><tt class="py-comment"># ---------------------------------------------------------</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line"><tt class="py-comment"># Public API</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line"> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt> <a class="py-toggle" href="#" id="parse-toggle" onclick="return toggle('parse');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.CDAOIO-module.html#parse">parse</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="parse-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="parse-expanded"><a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line">    <tt class="py-docstring">"""Iterate over the trees in a CDAO file handle.</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line"><tt class="py-docstring">    :returns: generator of Bio.Phylo.CDAO.Tree objects.</tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-24" class="py-name" targets="Module Bio.Entrez.Parser=Bio.Entrez.Parser-module.html,Class Bio.ExPASy.ScanProsite.Parser=Bio.ExPASy.ScanProsite.Parser-class.html,Class Bio.Phylo.CDAOIO.Parser=Bio.Phylo.CDAOIO.Parser-class.html,Class Bio.Phylo.NeXMLIO.Parser=Bio.Phylo.NeXMLIO.Parser-class.html,Class Bio.Phylo.NewickIO.Parser=Bio.Phylo.NewickIO.Parser-class.html,Class Bio.Phylo.PhyloXMLIO.Parser=Bio.Phylo.PhyloXMLIO.Parser-class.html,Class Bio.SeqIO.UniprotIO.Parser=Bio.SeqIO.UniprotIO.Parser-class.html"><a title="Bio.Entrez.Parser
Bio.ExPASy.ScanProsite.Parser
Bio.Phylo.CDAOIO.Parser
Bio.Phylo.NeXMLIO.Parser
Bio.Phylo.NewickIO.Parser
Bio.Phylo.PhyloXMLIO.Parser
Bio.SeqIO.UniprotIO.Parser" class="py-name" href="#" onclick="return doclink('link-24', 'Parser', 'link-24');">Parser</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-25" class="py-name" targets="Function Bio.AlignIO.parse()=Bio.AlignIO-module.html#parse,Function Bio.Blast.NCBIXML.parse()=Bio.Blast.NCBIXML-module.html#parse,Function Bio.Compass.parse()=Bio.Compass-module.html#parse,Function Bio.Emboss.Primer3.parse()=Bio.Emboss.Primer3-module.html#parse,Method Bio.Entrez.Parser.DataHandler.parse()=Bio.Entrez.Parser.DataHandler-class.html#parse,Function Bio.Entrez.parse()=Bio.Entrez-module.html#parse,Function Bio.ExPASy.Enzyme.parse()=Bio.ExPASy.Enzyme-module.html#parse,Function Bio.ExPASy.Prodoc.parse()=Bio.ExPASy.Prodoc-module.html#parse,Function Bio.ExPASy.Prosite.parse()=Bio.ExPASy.Prosite-module.html#parse,Function Bio.ExPASy.cellosaurus.parse()=Bio.ExPASy.cellosaurus-module.html#parse,Method Bio.GenBank.FeatureParser.parse()=Bio.GenBank.FeatureParser-class.html#parse,Method Bio.GenBank.RecordParser.parse()=Bio.GenBank.RecordParser-class.html#parse,Method Bio.GenBank.Scanner.InsdcScanner.parse()=Bio.GenBank.Scanner.InsdcScanner-class.html#parse,Function Bio.Geo.parse()=Bio.Geo-module.html#parse,Function Bio.KEGG.Compound.parse()=Bio.KEGG.Compound-module.html#parse,Function Bio.KEGG.Enzyme.parse()=Bio.KEGG.Enzyme-module.html#parse,Function Bio.KEGG.Gene.parse()=Bio.KEGG.Gene-module.html#parse,Method Bio.KEGG.KGML.KGML_parser.KGMLParser.parse()=Bio.KEGG.KGML.KGML_parser.KGMLParser-class.html#parse,Function Bio.KEGG.KGML.KGML_parser.parse()=Bio.KEGG.KGML.KGML_parser-module.html#parse,Function Bio.KEGG.Map.parse()=Bio.KEGG.Map-module.html#parse,Function Bio.Medline.parse()=Bio.Medline-module.html#parse,Method Bio.Phylo.CDAOIO.Parser.parse()=Bio.Phylo.CDAOIO.Parser-class.html#parse,Function Bio.Phylo.CDAOIO.parse()=Bio.Phylo.CDAOIO-module.html#parse,Method Bio.Phylo.NeXMLIO.Parser.parse()=Bio.Phylo.NeXMLIO.Parser-class.html#parse,Function Bio.Phylo.NeXMLIO.parse()=Bio.Phylo.NeXMLIO-module.html#parse,Method Bio.Phylo.NewickIO.Parser.parse()=Bio.Phylo.NewickIO.Parser-class.html#parse,Function Bio.Phylo.NewickIO.parse()=Bio.Phylo.NewickIO-module.html#parse,Function Bio.Phylo.NexusIO.parse()=Bio.Phylo.NexusIO-module.html#parse,Method Bio.Phylo.PhyloXMLIO.Parser.parse()=Bio.Phylo.PhyloXMLIO.Parser-class.html#parse,Function Bio.Phylo.PhyloXMLIO.parse()=Bio.Phylo.PhyloXMLIO-module.html#parse,Function Bio.Phylo._io.parse()=Bio.Phylo._io-module.html#parse,Function Bio.SCOP.Cla.parse()=Bio.SCOP.Cla-module.html#parse,Function Bio.SCOP.Des.parse()=Bio.SCOP.Des-module.html#parse,Function Bio.SCOP.Dom.parse()=Bio.SCOP.Dom-module.html#parse,Function Bio.SCOP.Hie.parse()=Bio.SCOP.Hie-module.html#parse,Function Bio.SCOP.Raf.parse()=Bio.SCOP.Raf-module.html#parse,Method Bio.SearchIO._legacy.NCBIStandalone.BlastErrorParser.parse()=Bio.SearchIO._legacy.NCBIStandalone.BlastErrorParser-class.html#parse,Method Bio.SearchIO._legacy.NCBIStandalone.BlastParser.parse()=Bio.SearchIO._legacy.NCBIStandalone.BlastParser-class.html#parse,Method Bio.SearchIO._legacy.NCBIStandalone.PSIBlastParser.parse()=Bio.SearchIO._legacy.NCBIStandalone.PSIBlastParser-class.html#parse,Method Bio.SearchIO._legacy.ParserSupport.AbstractParser.parse()=Bio.SearchIO._legacy.ParserSupport.AbstractParser-class.html#parse,Function Bio.SearchIO.parse()=Bio.SearchIO-module.html#parse,Method Bio.SeqIO.UniprotIO.Parser.parse()=Bio.SeqIO.UniprotIO.Parser-class.html#parse,Function Bio.SeqIO.parse()=Bio.SeqIO-module.html#parse,Function Bio.Sequencing.Ace.parse()=Bio.Sequencing.Ace-module.html#parse,Function Bio.Sequencing.Phd.parse()=Bio.Sequencing.Phd-module.html#parse,Function Bio.SwissProt.KeyWList.parse()=Bio.SwissProt.KeyWList-module.html#parse,Function Bio.SwissProt.parse()=Bio.SwissProt-module.html#parse,Function Bio.UniGene.parse()=Bio.UniGene-module.html#parse,Function Bio.Wise.psw.parse()=Bio.Wise.psw-module.html#parse,Function Bio.motifs.parse()=Bio.motifs-module.html#parse,Function Bio.phenotype.parse()=Bio.phenotype-module.html#parse"><a title="Bio.AlignIO.parse
Bio.Blast.NCBIXML.parse
Bio.Compass.parse
Bio.Emboss.Primer3.parse
Bio.Entrez.Parser.DataHandler.parse
Bio.Entrez.parse
Bio.ExPASy.Enzyme.parse
Bio.ExPASy.Prodoc.parse
Bio.ExPASy.Prosite.parse
Bio.ExPASy.cellosaurus.parse
Bio.GenBank.FeatureParser.parse
Bio.GenBank.RecordParser.parse
Bio.GenBank.Scanner.InsdcScanner.parse
Bio.Geo.parse
Bio.KEGG.Compound.parse
Bio.KEGG.Enzyme.parse
Bio.KEGG.Gene.parse
Bio.KEGG.KGML.KGML_parser.KGMLParser.parse
Bio.KEGG.KGML.KGML_parser.parse
Bio.KEGG.Map.parse
Bio.Medline.parse
Bio.Phylo.CDAOIO.Parser.parse
Bio.Phylo.CDAOIO.parse
Bio.Phylo.NeXMLIO.Parser.parse
Bio.Phylo.NeXMLIO.parse
Bio.Phylo.NewickIO.Parser.parse
Bio.Phylo.NewickIO.parse
Bio.Phylo.NexusIO.parse
Bio.Phylo.PhyloXMLIO.Parser.parse
Bio.Phylo.PhyloXMLIO.parse
Bio.Phylo._io.parse
Bio.SCOP.Cla.parse
Bio.SCOP.Des.parse
Bio.SCOP.Dom.parse
Bio.SCOP.Hie.parse
Bio.SCOP.Raf.parse
Bio.SearchIO._legacy.NCBIStandalone.BlastErrorParser.parse
Bio.SearchIO._legacy.NCBIStandalone.BlastParser.parse
Bio.SearchIO._legacy.NCBIStandalone.PSIBlastParser.parse
Bio.SearchIO._legacy.ParserSupport.AbstractParser.parse
Bio.SearchIO.parse
Bio.SeqIO.UniprotIO.Parser.parse
Bio.SeqIO.parse
Bio.Sequencing.Ace.parse
Bio.Sequencing.Phd.parse
Bio.SwissProt.KeyWList.parse
Bio.SwissProt.parse
Bio.UniGene.parse
Bio.Wise.psw.parse
Bio.motifs.parse
Bio.phenotype.parse" class="py-name" href="#" onclick="return doclink('link-25', 'parse', 'link-25');">parse</a></tt><tt class="py-op">(</tt><tt class="py-op">**</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> </tt>
</div><a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line"> </tt>
<a name="write"></a><div id="write-def"><a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line"> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt> <a class="py-toggle" href="#" id="write-toggle" onclick="return toggle('write');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.CDAOIO-module.html#write">write</a><tt class="py-op">(</tt><tt class="py-param">trees</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">plain</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="write-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="write-expanded"><a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line">    <tt class="py-docstring">"""Write a trees in CDAO format to the given file handle.</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line"><tt class="py-docstring">    :returns: number of trees written.</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-26" class="py-name" targets="Class Bio.Phylo.CDAOIO.Writer=Bio.Phylo.CDAOIO.Writer-class.html,Class Bio.Phylo.NeXMLIO.Writer=Bio.Phylo.NeXMLIO.Writer-class.html,Class Bio.Phylo.NewickIO.Writer=Bio.Phylo.NewickIO.Writer-class.html,Class Bio.Phylo.PhyloXMLIO.Writer=Bio.Phylo.PhyloXMLIO.Writer-class.html"><a title="Bio.Phylo.CDAOIO.Writer
Bio.Phylo.NeXMLIO.Writer
Bio.Phylo.NewickIO.Writer
Bio.Phylo.PhyloXMLIO.Writer" class="py-name" href="#" onclick="return doclink('link-26', 'Writer', 'link-26');">Writer</a></tt><tt class="py-op">(</tt><tt class="py-name">trees</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-27" class="py-name" targets="Function Bio.AlignIO.write()=Bio.AlignIO-module.html#write,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.write()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#write,Method Bio.Phylo.CDAOIO.Writer.write()=Bio.Phylo.CDAOIO.Writer-class.html#write,Function Bio.Phylo.CDAOIO.write()=Bio.Phylo.CDAOIO-module.html#write,Method Bio.Phylo.NeXMLIO.Writer.write()=Bio.Phylo.NeXMLIO.Writer-class.html#write,Function Bio.Phylo.NeXMLIO.write()=Bio.Phylo.NeXMLIO-module.html#write,Method Bio.Phylo.NewickIO.Writer.write()=Bio.Phylo.NewickIO.Writer-class.html#write,Function Bio.Phylo.NewickIO.write()=Bio.Phylo.NewickIO-module.html#write,Function Bio.Phylo.NexusIO.write()=Bio.Phylo.NexusIO-module.html#write,Method Bio.Phylo.PhyloXMLIO.Writer.write()=Bio.Phylo.PhyloXMLIO.Writer-class.html#write,Function Bio.Phylo.PhyloXMLIO.write()=Bio.Phylo.PhyloXMLIO-module.html#write,Function Bio.Phylo._io.write()=Bio.Phylo._io-module.html#write,Function Bio.SearchIO.write()=Bio.SearchIO-module.html#write,Function Bio.SeqIO.write()=Bio.SeqIO-module.html#write,Method Bio.bgzf.BgzfWriter.write()=Bio.bgzf.BgzfWriter-class.html#write,Function Bio.motifs.clusterbuster.write()=Bio.motifs.clusterbuster-module.html#write,Function Bio.motifs.jaspar.write()=Bio.motifs.jaspar-module.html#write,Function Bio.motifs.pfm.write()=Bio.motifs.pfm-module.html#write,Function Bio.motifs.transfac.write()=Bio.motifs.transfac-module.html#write,Function Bio.motifs.write()=Bio.motifs-module.html#write,Method Bio.phenotype.phen_micro.JsonWriter.write()=Bio.phenotype.phen_micro.JsonWriter-class.html#write,Function Bio.phenotype.write()=Bio.phenotype-module.html#write"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.clusterbuster.write
Bio.motifs.jaspar.write
Bio.motifs.pfm.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-27', 'write', 'link-27');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-name">plain</tt><tt class="py-op">=</tt><tt class="py-name">plain</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> </tt>
</div><a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line"> </tt>
<a name="Parser"></a><div id="Parser-def"><a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line"> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line"><tt class="py-comment"># ---------------------------------------------------------</tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line"><tt class="py-comment"># Input</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line"> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt> <a class="py-toggle" href="#" id="Parser-toggle" onclick="return toggle('Parser');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.CDAOIO.Parser-class.html">Parser</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Parser-expanded"><a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line">    <tt class="py-docstring">"""Parse a CDAO tree given a file handle."""</tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line"> </tt>
<a name="Parser.__init__"></a><div id="Parser.__init__-def"><a name="L90"></a><tt class="py-lineno"> 90</tt> <a class="py-toggle" href="#" id="Parser.__init__-toggle" onclick="return toggle('Parser.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.CDAOIO.Parser-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.__init__-expanded"><a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize CDAO tree parser."""</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">graph</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">node_info</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-28" class="py-name" targets="Method Bio.Pathway.Rep.Graph.Graph.children()=Bio.Pathway.Rep.Graph.Graph-class.html#children,Method Bio.Pathway.Rep.MultiGraph.MultiGraph.children()=Bio.Pathway.Rep.MultiGraph.MultiGraph-class.html#children"><a title="Bio.Pathway.Rep.Graph.Graph.children
Bio.Pathway.Rep.MultiGraph.MultiGraph.children" class="py-name" href="#" onclick="return doclink('link-28', 'children', 'link-28');">children</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">rooted</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
</div><a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line"> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">classmethod</tt> </tt>
<a name="Parser.from_string"></a><div id="Parser.from_string-def"><a name="L99"></a><tt class="py-lineno"> 99</tt> <a class="py-toggle" href="#" id="Parser.from_string-toggle" onclick="return toggle('Parser.from_string');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.CDAOIO.Parser-class.html#from_string">from_string</a><tt class="py-op">(</tt><tt class="py-param">cls</tt><tt class="py-op">,</tt> <tt class="py-param">treetext</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.from_string-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.from_string-expanded"><a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line">        <tt class="py-docstring">"""Instantiate the class from the given string."""</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line">        <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">StringIO</tt><tt class="py-op">(</tt><tt class="py-name">treetext</tt><tt class="py-op">)</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">cls</tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
</div><a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line"> </tt>
<a name="Parser.parse"></a><div id="Parser.parse-def"><a name="L104"></a><tt class="py-lineno">104</tt> <a class="py-toggle" href="#" id="Parser.parse-toggle" onclick="return toggle('Parser.parse');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.CDAOIO.Parser-class.html#parse">parse</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.parse-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.parse-expanded"><a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">        <tt class="py-docstring">"""Parse the text stream this object was initialized with."""</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-29" class="py-name" targets="Method Bio.Phylo.CDAOIO.Parser.parse_handle_to_graph()=Bio.Phylo.CDAOIO.Parser-class.html#parse_handle_to_graph"><a title="Bio.Phylo.CDAOIO.Parser.parse_handle_to_graph" class="py-name" href="#" onclick="return doclink('link-29', 'parse_handle_to_graph', 'link-29');">parse_handle_to_graph</a></tt><tt class="py-op">(</tt><tt class="py-op">**</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-30" class="py-name" targets="Method Bio.Phylo.CDAOIO.Parser.parse_graph()=Bio.Phylo.CDAOIO.Parser-class.html#parse_graph"><a title="Bio.Phylo.CDAOIO.Parser.parse_graph" class="py-name" href="#" onclick="return doclink('link-30', 'parse_graph', 'link-30');">parse_graph</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line"> </tt>
<a name="Parser.parse_handle_to_graph"></a><div id="Parser.parse_handle_to_graph-def"><a name="L109"></a><tt class="py-lineno">109</tt> <a class="py-toggle" href="#" id="Parser.parse_handle_to_graph-toggle" onclick="return toggle('Parser.parse_handle_to_graph');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.CDAOIO.Parser-class.html#parse_handle_to_graph">parse_handle_to_graph</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">rooted</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line">                              <tt class="py-param">parse_format</tt><tt class="py-op">=</tt><tt class="py-string">"turtle"</tt><tt class="py-op">,</tt> <tt class="py-param">context</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.parse_handle_to_graph-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.parse_handle_to_graph-expanded"><a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line">        <tt class="py-docstring">"""Parse self.handle into RDF model self.model."""</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">graph</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">graph</tt> <tt class="py-op">=</tt> <tt class="py-name">rdflib</tt><tt class="py-op">.</tt><tt id="link-31" class="py-name" targets="Module Bio.Pathway.Rep.Graph=Bio.Pathway.Rep.Graph-module.html,Class Bio.Pathway.Rep.Graph.Graph=Bio.Pathway.Rep.Graph.Graph-class.html"><a title="Bio.Pathway.Rep.Graph
Bio.Pathway.Rep.Graph.Graph" class="py-name" href="#" onclick="return doclink('link-31', 'Graph', 'link-31');">Graph</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">        <tt class="py-name">graph</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">graph</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line"> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt> <tt class="py-keyword">in</tt> <tt id="link-32" class="py-name"><a title="Bio.Phylo.CDAOIO.RDF_NAMESPACES" class="py-name" href="#" onclick="return doclink('link-32', 'RDF_NAMESPACES', 'link-15');">RDF_NAMESPACES</a></tt><tt class="py-op">.</tt><tt id="link-33" class="py-name" targets="Method Bio.Align.substitution_matrices.Array.items()=Bio.Align.substitution_matrices.Array-class.html#items,Method Bio.Crystal.Crystal.items()=Bio.Crystal.Crystal-class.html#items,Method Bio.File._IndexedSeqFileDict.items()=Bio.File._IndexedSeqFileDict-class.html#items,Method Bio.Phylo.PhyloXML.Events.items()=Bio.Phylo.PhyloXML.Events-class.html#items,Variable Bio.SearchIO._model.query.QueryResult.items=Bio.SearchIO._model.query.QueryResult-class.html#items,Method BioSQL.BioSeqDatabase.BioSeqDatabase.items()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#items,Method BioSQL.BioSeqDatabase.DBServer.items()=BioSQL.BioSeqDatabase.DBServer-class.html#items"><a title="Bio.Align.substitution_matrices.Array.items
Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-33', 'items', 'link-33');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line">            <tt class="py-name">graph</tt><tt class="py-op">.</tt><tt class="py-name">bind</tt><tt class="py-op">(</tt><tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt><tt class="py-op">)</tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line"> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">rooted</tt> <tt class="py-op">=</tt> <tt class="py-name">rooted</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line"> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"base_uri"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">kwargs</tt><tt class="py-op">:</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line">            <tt class="py-name">base_uri</tt> <tt class="py-op">=</tt> <tt class="py-name">kwargs</tt><tt class="py-op">[</tt><tt class="py-string">"base_uri"</tt><tt class="py-op">]</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line">            <tt class="py-comment"># Windows style slashes cannot be used in an RDF URI</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line">            <tt class="py-name">base_uri</tt> <tt class="py-op">=</tt> <tt class="py-string">"file://"</tt> <tt class="py-op">+</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">abspath</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-34" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.name=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.name=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.name=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#name,Method Bio.Phylo.PhyloXMLIO.Writer.name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#name,Variable Bio.codonalign.codonalphabet.CodonAlphabet.name=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-34', 'name', 'link-34');">name</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"\\"</tt><tt class="py-op">,</tt> <tt class="py-string">"/"</tt><tt class="py-op">)</tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line"> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line">        <tt class="py-name">graph</tt><tt class="py-op">.</tt><tt id="link-35" class="py-name"><a title="Bio.AlignIO.parse
Bio.Blast.NCBIXML.parse
Bio.Compass.parse
Bio.Emboss.Primer3.parse
Bio.Entrez.Parser.DataHandler.parse
Bio.Entrez.parse
Bio.ExPASy.Enzyme.parse
Bio.ExPASy.Prodoc.parse
Bio.ExPASy.Prosite.parse
Bio.ExPASy.cellosaurus.parse
Bio.GenBank.FeatureParser.parse
Bio.GenBank.RecordParser.parse
Bio.GenBank.Scanner.InsdcScanner.parse
Bio.Geo.parse
Bio.KEGG.Compound.parse
Bio.KEGG.Enzyme.parse
Bio.KEGG.Gene.parse
Bio.KEGG.KGML.KGML_parser.KGMLParser.parse
Bio.KEGG.KGML.KGML_parser.parse
Bio.KEGG.Map.parse
Bio.Medline.parse
Bio.Phylo.CDAOIO.Parser.parse
Bio.Phylo.CDAOIO.parse
Bio.Phylo.NeXMLIO.Parser.parse
Bio.Phylo.NeXMLIO.parse
Bio.Phylo.NewickIO.Parser.parse
Bio.Phylo.NewickIO.parse
Bio.Phylo.NexusIO.parse
Bio.Phylo.PhyloXMLIO.Parser.parse
Bio.Phylo.PhyloXMLIO.parse
Bio.Phylo._io.parse
Bio.SCOP.Cla.parse
Bio.SCOP.Des.parse
Bio.SCOP.Dom.parse
Bio.SCOP.Hie.parse
Bio.SCOP.Raf.parse
Bio.SearchIO._legacy.NCBIStandalone.BlastErrorParser.parse
Bio.SearchIO._legacy.NCBIStandalone.BlastParser.parse
Bio.SearchIO._legacy.NCBIStandalone.PSIBlastParser.parse
Bio.SearchIO._legacy.ParserSupport.AbstractParser.parse
Bio.SearchIO.parse
Bio.SeqIO.UniprotIO.Parser.parse
Bio.SeqIO.parse
Bio.Sequencing.Ace.parse
Bio.Sequencing.Phd.parse
Bio.SwissProt.KeyWList.parse
Bio.SwissProt.parse
Bio.UniGene.parse
Bio.Wise.psw.parse
Bio.motifs.parse
Bio.phenotype.parse" class="py-name" href="#" onclick="return doclink('link-35', 'parse', 'link-25');">parse</a></tt><tt class="py-op">(</tt><tt class="py-name">file</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-name">publicID</tt><tt class="py-op">=</tt><tt class="py-name">base_uri</tt><tt class="py-op">,</tt> <tt id="link-36" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.format()=Bio.Align.MultipleSeqAlignment-class.html#format,Method Bio.Align.PairwiseAlignment.format()=Bio.Align.PairwiseAlignment-class.html#format,Method Bio.Phylo.BaseTree.Tree.format()=Bio.Phylo.BaseTree.Tree-class.html#format,Method Bio.Restriction.Restriction.RestrictionBatch.format()=Bio.Restriction.Restriction.RestrictionBatch-class.html#format,Method Bio.SeqRecord.SeqRecord.format()=Bio.SeqRecord.SeqRecord-class.html#format,Method Bio.SubsMat.SeqMat.format()=Bio.SubsMat.SeqMat-class.html#format,Method Bio.motifs.Motif.format()=Bio.motifs.Motif-class.html#format"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Align.PairwiseAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.SubsMat.SeqMat.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-36', 'format', 'link-36');">format</a></tt><tt class="py-op">=</tt><tt class="py-name">parse_format</tt><tt class="py-op">)</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line"> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-37" class="py-name"><a title="Bio.Phylo.CDAOIO.Parser.parse_graph" class="py-name" href="#" onclick="return doclink('link-37', 'parse_graph', 'link-30');">parse_graph</a></tt><tt class="py-op">(</tt><tt class="py-name">graph</tt><tt class="py-op">,</tt> <tt class="py-name">context</tt><tt class="py-op">=</tt><tt class="py-name">context</tt><tt class="py-op">)</tt> </tt>
</div><a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line"> </tt>
<a name="Parser.parse_graph"></a><div id="Parser.parse_graph-def"><a name="L131"></a><tt class="py-lineno">131</tt> <a class="py-toggle" href="#" id="Parser.parse_graph-toggle" onclick="return toggle('Parser.parse_graph');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.CDAOIO.Parser-class.html#parse_graph">parse_graph</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">graph</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">context</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.parse_graph-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.parse_graph-expanded"><a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line">        <tt class="py-docstring">"""Iterate over RDF model yielding CDAO.Tree instances."""</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">graph</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line">            <tt class="py-name">graph</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">graph</tt> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line"> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line">        <tt class="py-comment"># look up branch lengths/TUs for all nodes</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-38" class="py-name" targets="Method Bio.Phylo.CDAOIO.Parser.get_node_info()=Bio.Phylo.CDAOIO.Parser-class.html#get_node_info"><a title="Bio.Phylo.CDAOIO.Parser.get_node_info" class="py-name" href="#" onclick="return doclink('link-38', 'get_node_info', 'link-38');">get_node_info</a></tt><tt class="py-op">(</tt><tt class="py-name">graph</tt><tt class="py-op">,</tt> <tt class="py-name">context</tt><tt class="py-op">=</tt><tt class="py-name">context</tt><tt class="py-op">)</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line"> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">root_node</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tree_roots</tt><tt class="py-op">:</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line">            <tt id="link-39" class="py-name" targets="Variable Bio.Phylo.BaseTree.Tree.clade=Bio.Phylo.BaseTree.Tree-class.html#clade,Method Bio.Phylo.PhyloXMLIO.Writer.clade()=Bio.Phylo.PhyloXMLIO.Writer-class.html#clade"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-39', 'clade', 'link-39');">clade</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-40" class="py-name" targets="Method Bio.Phylo.CDAOIO.Parser.parse_children()=Bio.Phylo.CDAOIO.Parser-class.html#parse_children"><a title="Bio.Phylo.CDAOIO.Parser.parse_children" class="py-name" href="#" onclick="return doclink('link-40', 'parse_children', 'link-40');">parse_children</a></tt><tt class="py-op">(</tt><tt class="py-name">root_node</tt><tt class="py-op">)</tt> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line"> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line">            <tt class="py-keyword">yield</tt> <tt id="link-41" class="py-name"><a title="Bio.Phylo.CDAO" class="py-name" href="#" onclick="return doclink('link-41', 'CDAO', 'link-6');">CDAO</a></tt><tt class="py-op">.</tt><tt id="link-42" class="py-name" targets="Class Bio.Cluster.Tree=Bio.Cluster.Tree-class.html,Class Bio.Nexus.Trees.Tree=Bio.Nexus.Trees.Tree-class.html,Class Bio.Phylo.BaseTree.Tree=Bio.Phylo.BaseTree.Tree-class.html,Class Bio.Phylo.CDAO.Tree=Bio.Phylo.CDAO.Tree-class.html,Class Bio.Phylo.NeXML.Tree=Bio.Phylo.NeXML.Tree-class.html,Class Bio.Phylo.Newick.Tree=Bio.Phylo.Newick.Tree-class.html"><a title="Bio.Cluster.Tree
Bio.Nexus.Trees.Tree
Bio.Phylo.BaseTree.Tree
Bio.Phylo.CDAO.Tree
Bio.Phylo.NeXML.Tree
Bio.Phylo.Newick.Tree" class="py-name" href="#" onclick="return doclink('link-42', 'Tree', 'link-42');">Tree</a></tt><tt class="py-op">(</tt><tt id="link-43" class="py-name" targets="Variable Bio.Phylo.BaseTree.Clade.root=Bio.Phylo.BaseTree.Clade-class.html#root,Variable Bio.Phylo._cdao_owl.root=Bio.Phylo._cdao_owl-module.html#root"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-43', 'root', 'link-43');">root</a></tt><tt class="py-op">=</tt><tt id="link-44" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-44', 'clade', 'link-39');">clade</a></tt><tt class="py-op">,</tt> <tt class="py-name">rooted</tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">rooted</tt><tt class="py-op">)</tt> </tt>
</div><a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line"> </tt>
<a name="Parser.new_clade"></a><div id="Parser.new_clade-def"><a name="L144"></a><tt class="py-lineno">144</tt> <a class="py-toggle" href="#" id="Parser.new_clade-toggle" onclick="return toggle('Parser.new_clade');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.CDAOIO.Parser-class.html#new_clade">new_clade</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">node</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.new_clade-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.new_clade-expanded"><a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a CDAO.Clade object for a given named node."""</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line">        <tt class="py-name">result</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">node_info</tt><tt class="py-op">[</tt><tt id="link-45" class="py-name" targets="Method Bio.Nexus.Trees.Tree.node()=Bio.Nexus.Trees.Tree-class.html#node"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-45', 'node', 'link-45');">node</a></tt><tt class="py-op">]</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line"> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line">        <tt class="py-name">kwargs</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"branch_length"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">result</tt><tt class="py-op">:</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">            <tt class="py-name">kwargs</tt><tt class="py-op">[</tt><tt class="py-string">"branch_length"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">result</tt><tt class="py-op">[</tt><tt class="py-string">"branch_length"</tt><tt class="py-op">]</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"label"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">result</tt><tt class="py-op">:</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line">            <tt class="py-name">kwargs</tt><tt class="py-op">[</tt><tt class="py-string">"name"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">result</tt><tt class="py-op">[</tt><tt class="py-string">"label"</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"_"</tt><tt class="py-op">,</tt> <tt class="py-string">" "</tt><tt class="py-op">)</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"confidence"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">result</tt><tt class="py-op">:</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line">            <tt class="py-name">kwargs</tt><tt class="py-op">[</tt><tt class="py-string">"confidence"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">result</tt><tt class="py-op">[</tt><tt class="py-string">"confidence"</tt><tt class="py-op">]</tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line"> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line">        <tt id="link-46" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-46', 'clade', 'link-39');">clade</a></tt> <tt class="py-op">=</tt> <tt id="link-47" class="py-name"><a title="Bio.Phylo.CDAO" class="py-name" href="#" onclick="return doclink('link-47', 'CDAO', 'link-6');">CDAO</a></tt><tt class="py-op">.</tt><tt id="link-48" class="py-name" targets="Class Bio.Phylo.BaseTree.Clade=Bio.Phylo.BaseTree.Clade-class.html,Class Bio.Phylo.CDAO.Clade=Bio.Phylo.CDAO.Clade-class.html,Class Bio.Phylo.NeXML.Clade=Bio.Phylo.NeXML.Clade-class.html,Class Bio.Phylo.Newick.Clade=Bio.Phylo.Newick.Clade-class.html,Class Bio.Phylo.PhyloXML.Clade=Bio.Phylo.PhyloXML.Clade-class.html"><a title="Bio.Phylo.BaseTree.Clade
Bio.Phylo.CDAO.Clade
Bio.Phylo.NeXML.Clade
Bio.Phylo.Newick.Clade
Bio.Phylo.PhyloXML.Clade" class="py-name" href="#" onclick="return doclink('link-48', 'Clade', 'link-48');">Clade</a></tt><tt class="py-op">(</tt><tt class="py-op">**</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line"> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-49" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-49', 'clade', 'link-39');">clade</a></tt> </tt>
</div><a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line"> </tt>
<a name="Parser.get_node_info"></a><div id="Parser.get_node_info-def"><a name="L160"></a><tt class="py-lineno">160</tt> <a class="py-toggle" href="#" id="Parser.get_node_info-toggle" onclick="return toggle('Parser.get_node_info');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.CDAOIO.Parser-class.html#get_node_info">get_node_info</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">graph</tt><tt class="py-op">,</tt> <tt class="py-param">context</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.get_node_info-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.get_node_info-expanded"><a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a dictionary containing information about all nodes in the tree."""</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">node_info</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">obj_info</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-50" class="py-name"><a title="Bio.Pathway.Rep.Graph.Graph.children
Bio.Pathway.Rep.MultiGraph.MultiGraph.children" class="py-name" href="#" onclick="return doclink('link-50', 'children', 'link-28');">children</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-51" class="py-name" targets="Method Bio.Pathway.Rep.Graph.Graph.nodes()=Bio.Pathway.Rep.Graph.Graph-class.html#nodes,Method Bio.Pathway.Rep.MultiGraph.MultiGraph.nodes()=Bio.Pathway.Rep.MultiGraph.MultiGraph-class.html#nodes"><a title="Bio.Pathway.Rep.Graph.Graph.nodes
Bio.Pathway.Rep.MultiGraph.MultiGraph.nodes" class="py-name" href="#" onclick="return doclink('link-51', 'nodes', 'link-51');">nodes</a></tt> <tt class="py-op">=</tt> <tt id="link-52" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.set()=Bio.Nexus.Nexus.StepMatrix-class.html#set,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.set()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#set,Method Bio.SVDSuperimposer.SVDSuperimposer.set()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#set"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-52', 'set', 'link-52');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tree_roots</tt> <tt class="py-op">=</tt> <tt id="link-53" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-53', 'set', 'link-52');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line"> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">        <tt class="py-name">assignments</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line">            <tt id="link-54" class="py-name" targets="Function Bio.Phylo.CDAOIO.qUri()=Bio.Phylo.CDAOIO-module.html#qUri,Function Bio.Phylo.NeXMLIO.qUri()=Bio.Phylo.NeXMLIO-module.html#qUri"><a title="Bio.Phylo.CDAOIO.qUri
Bio.Phylo.NeXMLIO.qUri" class="py-name" href="#" onclick="return doclink('link-54', 'qUri', 'link-54');">qUri</a></tt><tt class="py-op">(</tt><tt class="py-string">"cdao:has_Parent"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt class="py-string">"parent"</tt><tt class="py-op">,</tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line">            <tt id="link-55" class="py-name"><a title="Bio.Phylo.CDAOIO.qUri
Bio.Phylo.NeXMLIO.qUri" class="py-name" href="#" onclick="return doclink('link-55', 'qUri', 'link-54');">qUri</a></tt><tt class="py-op">(</tt><tt class="py-string">"cdao:belongs_to_Edge_as_Child"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt class="py-string">"edge"</tt><tt class="py-op">,</tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line">            <tt id="link-56" class="py-name"><a title="Bio.Phylo.CDAOIO.qUri
Bio.Phylo.NeXMLIO.qUri" class="py-name" href="#" onclick="return doclink('link-56', 'qUri', 'link-54');">qUri</a></tt><tt class="py-op">(</tt><tt class="py-string">"cdao:has_Annotation"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt class="py-string">"annotation"</tt><tt class="py-op">,</tt> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line">            <tt id="link-57" class="py-name"><a title="Bio.Phylo.CDAOIO.qUri
Bio.Phylo.NeXMLIO.qUri" class="py-name" href="#" onclick="return doclink('link-57', 'qUri', 'link-54');">qUri</a></tt><tt class="py-op">(</tt><tt class="py-string">"cdao:has_Value"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt class="py-string">"value"</tt><tt class="py-op">,</tt> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line">            <tt id="link-58" class="py-name"><a title="Bio.Phylo.CDAOIO.qUri
Bio.Phylo.NeXMLIO.qUri" class="py-name" href="#" onclick="return doclink('link-58', 'qUri', 'link-54');">qUri</a></tt><tt class="py-op">(</tt><tt class="py-string">"cdao:represents_TU"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt class="py-string">"tu"</tt><tt class="py-op">,</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line">            <tt id="link-59" class="py-name"><a title="Bio.Phylo.CDAOIO.qUri
Bio.Phylo.NeXMLIO.qUri" class="py-name" href="#" onclick="return doclink('link-59', 'qUri', 'link-54');">qUri</a></tt><tt class="py-op">(</tt><tt class="py-string">"rdfs:label"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt class="py-string">"label"</tt><tt class="py-op">,</tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line">            <tt id="link-60" class="py-name"><a title="Bio.Phylo.CDAOIO.qUri
Bio.Phylo.NeXMLIO.qUri" class="py-name" href="#" onclick="return doclink('link-60', 'qUri', 'link-54');">qUri</a></tt><tt class="py-op">(</tt><tt class="py-string">"cdao:has_Support_Value"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> <tt class="py-string">"confidence"</tt><tt class="py-op">,</tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">        <tt class="py-op">}</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line"> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">s</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt><tt class="py-op">,</tt> <tt class="py-name">o</tt> <tt class="py-keyword">in</tt> <tt class="py-name">graph</tt><tt class="py-op">:</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line">            <tt class="py-comment"># process each RDF triple in the graph sequentially</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line"> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">            <tt class="py-name">s</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt><tt class="py-op">,</tt> <tt class="py-name">o</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">v</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">o</tt><tt class="py-op">)</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line"> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">s</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">obj_info</tt><tt class="py-op">:</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">obj_info</tt><tt class="py-op">[</tt><tt class="py-name">s</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line">            <tt class="py-name">this</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">obj_info</tt><tt class="py-op">[</tt><tt class="py-name">s</tt><tt class="py-op">]</tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line"> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line">                <tt class="py-comment"># if the predicate is one we care about, store information for</tt> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line">                <tt class="py-comment"># later</tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line">                <tt class="py-name">this</tt><tt class="py-op">[</tt><tt class="py-name">assignments</tt><tt class="py-op">[</tt><tt class="py-name">v</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">o</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">                <tt class="py-keyword">pass</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line"> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">v</tt> <tt class="py-op">==</tt> <tt id="link-61" class="py-name"><a title="Bio.Phylo.CDAOIO.qUri
Bio.Phylo.NeXMLIO.qUri" class="py-name" href="#" onclick="return doclink('link-61', 'qUri', 'link-54');">qUri</a></tt><tt class="py-op">(</tt><tt class="py-string">"rdf:type"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">o</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt id="link-62" class="py-name"><a title="Bio.Phylo.CDAOIO.qUri
Bio.Phylo.NeXMLIO.qUri" class="py-name" href="#" onclick="return doclink('link-62', 'qUri', 'link-54');">qUri</a></tt><tt class="py-op">(</tt><tt class="py-string">"cdao:AncestralNode"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-63" class="py-name"><a title="Bio.Phylo.CDAOIO.qUri
Bio.Phylo.NeXMLIO.qUri" class="py-name" href="#" onclick="return doclink('link-63', 'qUri', 'link-54');">qUri</a></tt><tt class="py-op">(</tt><tt class="py-string">"cdao:TerminalNode"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">                    <tt class="py-comment"># this is a tree node; store it in set of all nodes</tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-64" class="py-name"><a title="Bio.Pathway.Rep.Graph.Graph.nodes
Bio.Pathway.Rep.MultiGraph.MultiGraph.nodes" class="py-name" href="#" onclick="return doclink('link-64', 'nodes', 'link-51');">nodes</a></tt><tt class="py-op">.</tt><tt id="link-65" class="py-name" targets="Method Bio.Graphics.BasicChromosome._ChromosomeComponent.add()=Bio.Graphics.BasicChromosome._ChromosomeComponent-class.html#add,Method Bio.Nexus.Nexus.StepMatrix.add()=Bio.Nexus.Nexus.StepMatrix-class.html#add,Method Bio.Nexus.Nodes.Chain.add()=Bio.Nexus.Nodes.Chain-class.html#add,Method Bio.PDB.Entity.Entity.add()=Bio.PDB.Entity.Entity-class.html#add,Method Bio.PDB.Residue.DisorderedResidue.add()=Bio.PDB.Residue.DisorderedResidue-class.html#add,Method Bio.PDB.Residue.Residue.add()=Bio.PDB.Residue.Residue-class.html#add,Method Bio.Restriction.Restriction.RestrictionBatch.add()=Bio.Restriction.Restriction.RestrictionBatch-class.html#add"><a title="Bio.Graphics.BasicChromosome._ChromosomeComponent.add
Bio.Nexus.Nexus.StepMatrix.add
Bio.Nexus.Nodes.Chain.add
Bio.PDB.Entity.Entity.add
Bio.PDB.Residue.DisorderedResidue.add
Bio.PDB.Residue.Residue.add
Bio.Restriction.Restriction.RestrictionBatch.add" class="py-name" href="#" onclick="return doclink('link-65', 'add', 'link-65');">add</a></tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">)</tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">v</tt> <tt class="py-op">==</tt> <tt id="link-66" class="py-name"><a title="Bio.Phylo.CDAOIO.qUri
Bio.Phylo.NeXMLIO.qUri" class="py-name" href="#" onclick="return doclink('link-66', 'qUri', 'link-54');">qUri</a></tt><tt class="py-op">(</tt><tt class="py-string">"cdao:has_Root"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line">                <tt class="py-comment"># this is a tree; store its root in set of all tree roots</tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tree_roots</tt><tt class="py-op">.</tt><tt id="link-67" class="py-name"><a title="Bio.Graphics.BasicChromosome._ChromosomeComponent.add
Bio.Nexus.Nexus.StepMatrix.add
Bio.Nexus.Nodes.Chain.add
Bio.PDB.Entity.Entity.add
Bio.PDB.Residue.DisorderedResidue.add
Bio.PDB.Residue.Residue.add
Bio.Restriction.Restriction.RestrictionBatch.add" class="py-name" href="#" onclick="return doclink('link-67', 'add', 'link-65');">add</a></tt><tt class="py-op">(</tt><tt class="py-name">o</tt><tt class="py-op">)</tt> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line"> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-68" class="py-name"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-68', 'node', 'link-45');">node</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-69" class="py-name"><a title="Bio.Pathway.Rep.Graph.Graph.nodes
Bio.Pathway.Rep.MultiGraph.MultiGraph.nodes" class="py-name" href="#" onclick="return doclink('link-69', 'nodes', 'link-51');">nodes</a></tt><tt class="py-op">:</tt> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line">            <tt class="py-comment"># for each node, look up all information needed to create a</tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line">            <tt class="py-comment"># CDAO.Clade</tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">node_info</tt><tt class="py-op">[</tt><tt id="link-70" class="py-name"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-70', 'node', 'link-45');">node</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line">            <tt class="py-name">node_info</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">node_info</tt><tt class="py-op">[</tt><tt id="link-71" class="py-name"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-71', 'node', 'link-45');">node</a></tt><tt class="py-op">]</tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line"> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line">            <tt class="py-name">obj</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">obj_info</tt><tt class="py-op">[</tt><tt id="link-72" class="py-name"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-72', 'node', 'link-45');">node</a></tt><tt class="py-op">]</tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-string">"edge"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">obj</tt><tt class="py-op">:</tt> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line">                <tt class="py-comment"># if this object points to an edge, we need a branch length from</tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line">                <tt class="py-comment"># the annotation on that edge</tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line">                <tt class="py-name">edge</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">obj_info</tt><tt class="py-op">[</tt><tt class="py-name">obj</tt><tt class="py-op">[</tt><tt class="py-string">"edge"</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-string">"annotation"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">edge</tt><tt class="py-op">:</tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line">                    <tt id="link-73" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.annotation()=Bio.Phylo.PhyloXMLIO.Parser-class.html#annotation,Method Bio.Phylo.PhyloXMLIO.Writer.annotation()=Bio.Phylo.PhyloXMLIO.Writer-class.html#annotation"><a title="Bio.Phylo.PhyloXMLIO.Parser.annotation
Bio.Phylo.PhyloXMLIO.Writer.annotation" class="py-name" href="#" onclick="return doclink('link-73', 'annotation', 'link-73');">annotation</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">obj_info</tt><tt class="py-op">[</tt><tt class="py-name">edge</tt><tt class="py-op">[</tt><tt class="py-string">"annotation"</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-string">"value"</tt> <tt class="py-keyword">in</tt> <tt id="link-74" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.annotation
Bio.Phylo.PhyloXMLIO.Writer.annotation" class="py-name" href="#" onclick="return doclink('link-74', 'annotation', 'link-73');">annotation</a></tt><tt class="py-op">:</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line">                        <tt class="py-name">node_info</tt><tt class="py-op">[</tt><tt class="py-string">"branch_length"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt id="link-75" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.annotation
Bio.Phylo.PhyloXMLIO.Writer.annotation" class="py-name" href="#" onclick="return doclink('link-75', 'annotation', 'link-73');">annotation</a></tt><tt class="py-op">[</tt><tt class="py-string">"value"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line"> </tt>
<a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-string">"tu"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">obj</tt><tt class="py-op">:</tt> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line">                <tt class="py-comment"># if this object points to a TU, we need the label of that TU</tt> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line">                <tt class="py-name">tu</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">obj_info</tt><tt class="py-op">[</tt><tt class="py-name">obj</tt><tt class="py-op">[</tt><tt class="py-string">"tu"</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-string">"label"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">tu</tt><tt class="py-op">:</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line">                    <tt class="py-name">node_info</tt><tt class="py-op">[</tt><tt class="py-string">"label"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">tu</tt><tt class="py-op">[</tt><tt class="py-string">"label"</tt><tt class="py-op">]</tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line"> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-string">"parent"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">obj</tt><tt class="py-op">:</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line">                <tt class="py-comment"># store this node as a child of its parent, if it has one,</tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line">                <tt class="py-comment"># so that the tree can be traversed from parent to children</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line">                <tt class="py-name">parent</tt> <tt class="py-op">=</tt> <tt class="py-name">obj</tt><tt class="py-op">[</tt><tt class="py-string">"parent"</tt><tt class="py-op">]</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">parent</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-76" class="py-name"><a title="Bio.Pathway.Rep.Graph.Graph.children
Bio.Pathway.Rep.MultiGraph.MultiGraph.children" class="py-name" href="#" onclick="return doclink('link-76', 'children', 'link-28');">children</a></tt><tt class="py-op">:</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-77" class="py-name"><a title="Bio.Pathway.Rep.Graph.Graph.children
Bio.Pathway.Rep.MultiGraph.MultiGraph.children" class="py-name" href="#" onclick="return doclink('link-77', 'children', 'link-28');">children</a></tt><tt class="py-op">[</tt><tt class="py-name">parent</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-78" class="py-name"><a title="Bio.Pathway.Rep.Graph.Graph.children
Bio.Pathway.Rep.MultiGraph.MultiGraph.children" class="py-name" href="#" onclick="return doclink('link-78', 'children', 'link-28');">children</a></tt><tt class="py-op">[</tt><tt class="py-name">parent</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-79" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-79', 'append', 'link-79');">append</a></tt><tt class="py-op">(</tt><tt id="link-80" class="py-name"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-80', 'node', 'link-45');">node</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line"> </tt>
<a name="Parser.parse_children"></a><div id="Parser.parse_children-def"><a name="L232"></a><tt class="py-lineno">232</tt> <a class="py-toggle" href="#" id="Parser.parse_children-toggle" onclick="return toggle('Parser.parse_children');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.CDAOIO.Parser-class.html#parse_children">parse_children</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">node</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Parser.parse_children-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Parser.parse_children-expanded"><a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">        <tt class="py-docstring">"""Traverse the tree to create a nested clade structure.</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line"><tt class="py-docstring">        Return a CDAO.Clade, and calls itself recursively for each child,</tt> </tt>
<a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line"><tt class="py-docstring">        traversing the entire tree and creating a nested structure of CDAO.Clade</tt> </tt>
<a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line"><tt class="py-docstring">        objects.</tt> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line">        <tt id="link-81" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-81', 'clade', 'link-39');">clade</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-82" class="py-name" targets="Method Bio.Phylo.CDAOIO.Parser.new_clade()=Bio.Phylo.CDAOIO.Parser-class.html#new_clade,Method Bio.Phylo.NewickIO.Parser.new_clade()=Bio.Phylo.NewickIO.Parser-class.html#new_clade"><a title="Bio.Phylo.CDAOIO.Parser.new_clade
Bio.Phylo.NewickIO.Parser.new_clade" class="py-name" href="#" onclick="return doclink('link-82', 'new_clade', 'link-82');">new_clade</a></tt><tt class="py-op">(</tt><tt id="link-83" class="py-name"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-83', 'node', 'link-45');">node</a></tt><tt class="py-op">)</tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line"> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line">        <tt id="link-84" class="py-name"><a title="Bio.Pathway.Rep.Graph.Graph.children
Bio.Pathway.Rep.MultiGraph.MultiGraph.children" class="py-name" href="#" onclick="return doclink('link-84', 'children', 'link-28');">children</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-85" class="py-name"><a title="Bio.Pathway.Rep.Graph.Graph.children
Bio.Pathway.Rep.MultiGraph.MultiGraph.children" class="py-name" href="#" onclick="return doclink('link-85', 'children', 'link-28');">children</a></tt><tt class="py-op">[</tt><tt id="link-86" class="py-name"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-86', 'node', 'link-45');">node</a></tt><tt class="py-op">]</tt> <tt class="py-keyword">if</tt> <tt id="link-87" class="py-name"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-87', 'node', 'link-45');">node</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-88" class="py-name"><a title="Bio.Pathway.Rep.Graph.Graph.children
Bio.Pathway.Rep.MultiGraph.MultiGraph.children" class="py-name" href="#" onclick="return doclink('link-88', 'children', 'link-28');">children</a></tt> <tt class="py-keyword">else</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line">        <tt id="link-89" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-89', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt class="py-name">clades</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-90" class="py-name"><a title="Bio.Phylo.CDAOIO.Parser.parse_children" class="py-name" href="#" onclick="return doclink('link-90', 'parse_children', 'link-40');">parse_children</a></tt><tt class="py-op">(</tt><tt class="py-name">child_node</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">child_node</tt> <tt class="py-keyword">in</tt> <tt id="link-91" class="py-name"><a title="Bio.Pathway.Rep.Graph.Graph.children
Bio.Pathway.Rep.MultiGraph.MultiGraph.children" class="py-name" href="#" onclick="return doclink('link-91', 'children', 'link-28');">children</a></tt><tt class="py-op">]</tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line"> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-92" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-92', 'clade', 'link-39');">clade</a></tt> </tt>
</div></div><a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line"> </tt>
<a name="Writer"></a><div id="Writer-def"><a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line"> </tt>
<a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line"><tt class="py-comment"># ---------------------------------------------------------</tt> </tt>
<a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line"><tt class="py-comment"># Output</tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line"> </tt>
<a name="L251"></a><tt class="py-lineno">251</tt> <a class="py-toggle" href="#" id="Writer-toggle" onclick="return toggle('Writer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.CDAOIO.Writer-class.html">Writer</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Writer-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Writer-expanded"><a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line">    <tt class="py-docstring">"""Based on the writer in Bio.Nexus.Trees (str, to_string)."""</tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line"> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line">    <tt id="link-93" class="py-name" targets="Variable Bio.Phylo.CDAOIO.Writer.prefixes=Bio.Phylo.CDAOIO.Writer-class.html#prefixes"><a title="Bio.Phylo.CDAOIO.Writer.prefixes" class="py-name" href="#" onclick="return doclink('link-93', 'prefixes', 'link-93');">prefixes</a></tt> <tt class="py-op">=</tt> <tt id="link-94" class="py-name"><a title="Bio.Phylo.CDAOIO.RDF_NAMESPACES" class="py-name" href="#" onclick="return doclink('link-94', 'RDF_NAMESPACES', 'link-15');">RDF_NAMESPACES</a></tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line"> </tt>
<a name="Writer.__init__"></a><div id="Writer.__init__-def"><a name="L256"></a><tt class="py-lineno">256</tt> <a class="py-toggle" href="#" id="Writer.__init__-toggle" onclick="return toggle('Writer.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.CDAOIO.Writer-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">trees</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Writer.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Writer.__init__-expanded"><a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize parameters for writing a CDAO tree."""</tt> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">trees</tt> <tt class="py-op">=</tt> <tt class="py-name">trees</tt> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line"> </tt>
<a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">node_counter</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">edge_counter</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tu_counter</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tree_counter</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
</div><a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line"> </tt>
<a name="Writer.write"></a><div id="Writer.write-def"><a name="L265"></a><tt class="py-lineno">265</tt> <a class="py-toggle" href="#" id="Writer.write-toggle" onclick="return toggle('Writer.write');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.CDAOIO.Writer-class.html#write">write</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">tree_uri</tt><tt class="py-op">=</tt><tt class="py-string">""</tt><tt class="py-op">,</tt> <tt class="py-param">record_complete_ancestry</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line">              <tt class="py-param">rooted</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Writer.write-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Writer.write-expanded"><a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line">        <tt class="py-docstring">"""Write this instance's trees to a file handle."""</tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">rooted</tt> <tt class="py-op">=</tt> <tt class="py-name">rooted</tt> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">record_complete_ancestry</tt> <tt class="py-op">=</tt> <tt class="py-name">record_complete_ancestry</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line"> </tt>
<a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">tree_uri</tt> <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt class="py-name">tree_uri</tt><tt class="py-op">.</tt><tt id="link-95" class="py-name" targets="Method Bio.Seq.Seq.endswith()=Bio.Seq.Seq-class.html#endswith"><a title="Bio.Seq.Seq.endswith" class="py-name" href="#" onclick="return doclink('link-95', 'endswith', 'link-95');">endswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"/"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line">            <tt class="py-name">tree_uri</tt> <tt class="py-op">+=</tt> <tt class="py-string">"/"</tt> </tt>
<a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line"> </tt>
<a name="L274"></a><tt class="py-lineno">274</tt>  <tt class="py-line">        <tt class="py-name">trees</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">trees</tt> </tt>
<a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line"> </tt>
<a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">tree_uri</tt><tt class="py-op">:</tt> </tt>
<a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-96" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.clusterbuster.write
Bio.motifs.jaspar.write
Bio.motifs.pfm.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-96', 'write', 'link-27');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"@base &lt;%s&gt;\n"</tt> <tt class="py-op">%</tt> <tt class="py-name">tree_uri</tt><tt class="py-op">)</tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-97" class="py-name"><a title="Bio.Phylo.CDAOIO.Writer.prefixes" class="py-name" href="#" onclick="return doclink('link-97', 'prefixes', 'link-93');">prefixes</a></tt><tt class="py-op">.</tt><tt id="link-98" class="py-name"><a title="Bio.Align.substitution_matrices.Array.items
Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-98', 'items', 'link-33');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-99" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.clusterbuster.write
Bio.motifs.jaspar.write
Bio.motifs.pfm.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-99', 'write', 'link-27');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"@prefix %s: &lt;%s&gt; .\n"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line"> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-100" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.clusterbuster.write
Bio.motifs.jaspar.write
Bio.motifs.pfm.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-100', 'write', 'link-27');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"&lt;%s&gt; a owl:Ontology .\n"</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-101" class="py-name"><a title="Bio.Phylo.CDAOIO.Writer.prefixes" class="py-name" href="#" onclick="return doclink('link-101', 'prefixes', 'link-93');">prefixes</a></tt><tt class="py-op">[</tt><tt class="py-string">"cdao"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line"> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">tree</tt> <tt class="py-keyword">in</tt> <tt class="py-name">trees</tt><tt class="py-op">:</tt> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tree_counter</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tree_uri</tt> <tt class="py-op">=</tt> <tt class="py-string">"tree%s"</tt> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line"> </tt>
<a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line">            <tt class="py-name">first_clade</tt> <tt class="py-op">=</tt> <tt class="py-name">tree</tt><tt class="py-op">.</tt><tt id="link-102" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-102', 'clade', 'link-39');">clade</a></tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">            <tt class="py-name">statements</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-103" class="py-name" targets="Method Bio.Phylo.CDAOIO.Writer.process_clade()=Bio.Phylo.CDAOIO.Writer-class.html#process_clade,Method Bio.Phylo.NewickIO.Parser.process_clade()=Bio.Phylo.NewickIO.Parser-class.html#process_clade"><a title="Bio.Phylo.CDAOIO.Writer.process_clade
Bio.Phylo.NewickIO.Parser.process_clade" class="py-name" href="#" onclick="return doclink('link-103', 'process_clade', 'link-103');">process_clade</a></tt><tt class="py-op">(</tt><tt class="py-name">first_clade</tt><tt class="py-op">,</tt> <tt id="link-104" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-104', 'root', 'link-43');">root</a></tt><tt class="py-op">=</tt><tt class="py-name">tree</tt><tt class="py-op">)</tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">stmt</tt> <tt class="py-keyword">in</tt> <tt class="py-name">statements</tt><tt class="py-op">:</tt> </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-105" class="py-name" targets="Method Bio.Phylo.CDAOIO.Writer.add_stmt_to_handle()=Bio.Phylo.CDAOIO.Writer-class.html#add_stmt_to_handle"><a title="Bio.Phylo.CDAOIO.Writer.add_stmt_to_handle" class="py-name" href="#" onclick="return doclink('link-105', 'add_stmt_to_handle', 'link-105');">add_stmt_to_handle</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-name">stmt</tt><tt class="py-op">)</tt> </tt>
</div><a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line"> </tt>
<a name="Writer.add_stmt_to_handle"></a><div id="Writer.add_stmt_to_handle-def"><a name="L292"></a><tt class="py-lineno">292</tt> <a class="py-toggle" href="#" id="Writer.add_stmt_to_handle-toggle" onclick="return toggle('Writer.add_stmt_to_handle');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.CDAOIO.Writer-class.html#add_stmt_to_handle">add_stmt_to_handle</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">stmt</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Writer.add_stmt_to_handle-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Writer.add_stmt_to_handle-expanded"><a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add URI prefix to handle."""</tt> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line">        <tt class="py-comment"># apply URI prefixes</tt> </tt>
<a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line">        <tt class="py-name">stmt_strings</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">n</tt><tt class="py-op">,</tt> <tt class="py-name">part</tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt class="py-name">stmt</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">part</tt><tt class="py-op">,</tt> <tt class="py-name">rdflib</tt><tt class="py-op">.</tt><tt class="py-name">URIRef</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line">                <tt class="py-name">node_uri</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">part</tt><tt class="py-op">)</tt> </tt>
<a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line">                <tt class="py-name">changed</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt id="link-106" class="py-name" targets="Variable Bio.Phylo.NeXMLIO.prefix=Bio.Phylo.NeXMLIO-module.html#prefix,Variable Bio.Phylo.PhyloXMLIO.prefix=Bio.Phylo.PhyloXMLIO-module.html#prefix"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-106', 'prefix', 'link-106');">prefix</a></tt><tt class="py-op">,</tt> <tt id="link-107" class="py-name" targets="Variable Bio.Phylo.NeXMLIO.uri=Bio.Phylo.NeXMLIO-module.html#uri,Method Bio.Phylo.PhyloXMLIO.Parser.uri()=Bio.Phylo.PhyloXMLIO.Parser-class.html#uri,Method Bio.Phylo.PhyloXMLIO.Writer.uri()=Bio.Phylo.PhyloXMLIO.Writer-class.html#uri,Variable Bio.Phylo.PhyloXMLIO.uri=Bio.Phylo.PhyloXMLIO-module.html#uri"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-107', 'uri', 'link-107');">uri</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-108" class="py-name"><a title="Bio.Phylo.CDAOIO.Writer.prefixes" class="py-name" href="#" onclick="return doclink('link-108', 'prefixes', 'link-93');">prefixes</a></tt><tt class="py-op">.</tt><tt id="link-109" class="py-name"><a title="Bio.Align.substitution_matrices.Array.items
Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-109', 'items', 'link-33');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">node_uri</tt><tt class="py-op">.</tt><tt id="link-110" class="py-name" targets="Method Bio.Seq.Seq.startswith()=Bio.Seq.Seq-class.html#startswith"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-110', 'startswith', 'link-110');">startswith</a></tt><tt class="py-op">(</tt><tt id="link-111" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-111', 'uri', 'link-107');">uri</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line">                        <tt class="py-name">node_uri</tt> <tt class="py-op">=</tt> <tt class="py-name">node_uri</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt id="link-112" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-112', 'uri', 'link-107');">uri</a></tt><tt class="py-op">,</tt> <tt class="py-string">"%s:"</tt> <tt class="py-op">%</tt> <tt id="link-113" class="py-name"><a title="Bio.Phylo.NeXMLIO.prefix
Bio.Phylo.PhyloXMLIO.prefix" class="py-name" href="#" onclick="return doclink('link-113', 'prefix', 'link-106');">prefix</a></tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">node_uri</tt> <tt class="py-op">==</tt> <tt class="py-string">"rdf:type"</tt><tt class="py-op">:</tt> </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line">                            <tt class="py-name">node_uri</tt> <tt class="py-op">=</tt> <tt class="py-string">"a"</tt> </tt>
<a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line">                        <tt class="py-name">changed</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">changed</tt> <tt class="py-keyword">or</tt> <tt class="py-string">":"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">node_uri</tt><tt class="py-op">:</tt> </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line">                    <tt class="py-name">stmt_strings</tt><tt class="py-op">.</tt><tt id="link-114" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-114', 'append', 'link-79');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">node_uri</tt><tt class="py-op">)</tt> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line">                    <tt class="py-name">stmt_strings</tt><tt class="py-op">.</tt><tt id="link-115" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-115', 'append', 'link-79');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"&lt;%s&gt;"</tt> <tt class="py-op">%</tt> <tt class="py-name">node_uri</tt><tt class="py-op">)</tt> </tt>
<a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line"> </tt>
<a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">part</tt><tt class="py-op">,</tt> <tt class="py-name">rdflib</tt><tt class="py-op">.</tt><tt class="py-name">Literal</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line">                <tt class="py-name">stmt_strings</tt><tt class="py-op">.</tt><tt id="link-116" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-116', 'append', 'link-79');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">part</tt><tt class="py-op">.</tt><tt id="link-117" class="py-name" targets="Variable Bio.PDB.Polypeptide.n3=Bio.PDB.Polypeptide-module.html#n3"><a title="Bio.PDB.Polypeptide.n3" class="py-name" href="#" onclick="return doclink('link-117', 'n3', 'link-117');">n3</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line"> </tt>
<a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line">                <tt class="py-name">stmt_strings</tt><tt class="py-op">.</tt><tt id="link-118" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-118', 'append', 'link-79');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">part</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line"> </tt>
<a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-119" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.clusterbuster.write
Bio.motifs.jaspar.write
Bio.motifs.pfm.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-119', 'write', 'link-27');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"%s .\n"</tt> <tt class="py-op">%</tt> <tt class="py-string">" "</tt><tt class="py-op">.</tt><tt id="link-120" class="py-name" targets="Method Bio.Seq.MutableSeq.join()=Bio.Seq.MutableSeq-class.html#join,Method Bio.Seq.Seq.join()=Bio.Seq.Seq-class.html#join,Method Bio.Seq.UnknownSeq.join()=Bio.Seq.UnknownSeq-class.html#join"><a title="Bio.Seq.MutableSeq.join
Bio.Seq.Seq.join
Bio.Seq.UnknownSeq.join" class="py-name" href="#" onclick="return doclink('link-120', 'join', 'link-120');">join</a></tt><tt class="py-op">(</tt><tt class="py-name">stmt_strings</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line"> </tt>
<a name="Writer.process_clade"></a><div id="Writer.process_clade-def"><a name="L319"></a><tt class="py-lineno">319</tt> <a class="py-toggle" href="#" id="Writer.process_clade-toggle" onclick="return toggle('Writer.process_clade');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.CDAOIO.Writer-class.html#process_clade">process_clade</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">clade</tt><tt class="py-op">,</tt> <tt class="py-param">parent</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">root</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Writer.process_clade-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Writer.process_clade-expanded"><a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line">        <tt class="py-docstring">"""Recursively generate triples describing a tree of clades."""</tt> </tt>
<a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">node_counter</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line">        <tt id="link-121" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-121', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt id="link-122" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-122', 'uri', 'link-107');">uri</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"node%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">node_counter</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">zfill</tt><tt class="py-op">(</tt><tt id="link-123" class="py-name"><a title="Bio.Phylo.CDAOIO.ZEROES" class="py-name" href="#" onclick="return doclink('link-123', 'ZEROES', 'link-19');">ZEROES</a></tt><tt class="py-op">)</tt> </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">parent</tt><tt class="py-op">:</tt> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line">            <tt id="link-124" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-124', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt class="py-name">ancestors</tt> <tt class="py-op">=</tt> <tt class="py-name">parent</tt><tt class="py-op">.</tt><tt class="py-name">ancestors</tt> <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt class="py-name">parent</tt><tt class="py-op">.</tt><tt id="link-125" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-125', 'uri', 'link-107');">uri</a></tt><tt class="py-op">]</tt> </tt>
<a name="L325"></a><tt class="py-lineno">325</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line">            <tt id="link-126" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-126', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt class="py-name">ancestors</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line"> </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line">        <tt class="py-keyword">def</tt> <tt class="py-def-name">nUri</tt><tt class="py-op">(</tt><tt class="py-param">s</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line">            <tt class="py-comment"># nUri = lambda s: rdflib.URIRef(s)</tt> </tt>
<a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">rdflib</tt><tt class="py-op">.</tt><tt class="py-name">URIRef</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">)</tt> </tt>
</div><a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line"> </tt>
<a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line">        <tt class="py-keyword">def</tt> <tt class="py-def-name">pUri</tt><tt class="py-op">(</tt><tt class="py-param">s</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L333"></a><tt class="py-lineno">333</tt>  <tt class="py-line">            <tt class="py-comment"># pUri = lambda s: rdflib.URIRef(qUri(s))</tt> </tt>
<a name="L334"></a><tt class="py-lineno">334</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">rdflib</tt><tt class="py-op">.</tt><tt class="py-name">URIRef</tt><tt class="py-op">(</tt><tt id="link-127" class="py-name"><a title="Bio.Phylo.CDAOIO.qUri
Bio.Phylo.NeXMLIO.qUri" class="py-name" href="#" onclick="return doclink('link-127', 'qUri', 'link-54');">qUri</a></tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L335"></a><tt class="py-lineno">335</tt>  <tt class="py-line"> </tt>
<a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line">        <tt class="py-name">tree_id</tt> <tt class="py-op">=</tt> <tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt class="py-string">""</tt><tt class="py-op">)</tt> </tt>
<a name="L337"></a><tt class="py-lineno">337</tt>  <tt class="py-line"> </tt>
<a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line">        <tt class="py-name">statements</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L339"></a><tt class="py-lineno">339</tt>  <tt class="py-line"> </tt>
<a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-128" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-128', 'root', 'link-43');">root</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">False</tt><tt class="py-op">:</tt> </tt>
<a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line">            <tt class="py-comment"># create a cdao:RootedTree with reference to the tree root</tt> </tt>
<a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line">            <tt class="py-name">tree_type</tt> <tt class="py-op">=</tt> <tt class="py-name">pUri</tt><tt class="py-op">(</tt><tt class="py-string">"cdao:RootedTree"</tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">rooted</tt> <tt class="py-keyword">else</tt> <tt class="py-name">pUri</tt><tt class="py-op">(</tt> </tt>
<a name="L343"></a><tt class="py-lineno">343</tt>  <tt class="py-line">                <tt class="py-string">"cdao:UnrootedTree"</tt><tt class="py-op">)</tt> </tt>
<a name="L344"></a><tt class="py-lineno">344</tt>  <tt class="py-line"> </tt>
<a name="L345"></a><tt class="py-lineno">345</tt>  <tt class="py-line">            <tt class="py-name">statements</tt> <tt class="py-op">+=</tt> <tt class="py-op">[</tt> </tt>
<a name="L346"></a><tt class="py-lineno">346</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-name">tree_id</tt><tt class="py-op">,</tt> <tt class="py-name">pUri</tt><tt class="py-op">(</tt><tt class="py-string">"rdf:type"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">tree_type</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L347"></a><tt class="py-lineno">347</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-name">tree_id</tt><tt class="py-op">,</tt> <tt class="py-name">pUri</tt><tt class="py-op">(</tt><tt class="py-string">"cdao:has_Root"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt id="link-129" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-129', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt id="link-130" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-130', 'uri', 'link-107');">uri</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line">            <tt class="py-op">]</tt> </tt>
<a name="L349"></a><tt class="py-lineno">349</tt>  <tt class="py-line"> </tt>
<a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L351"></a><tt class="py-lineno">351</tt>  <tt class="py-line">                <tt class="py-name">tree_attributes</tt> <tt class="py-op">=</tt> <tt id="link-131" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-131', 'root', 'link-43');">root</a></tt><tt class="py-op">.</tt><tt class="py-name">attributes</tt> </tt>
<a name="L352"></a><tt class="py-lineno">352</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line">                <tt class="py-name">tree_attributes</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L354"></a><tt class="py-lineno">354</tt>  <tt class="py-line"> </tt>
<a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">predicate</tt><tt class="py-op">,</tt> <tt class="py-name">obj</tt> <tt class="py-keyword">in</tt> <tt class="py-name">tree_attributes</tt><tt class="py-op">:</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line">                <tt class="py-name">statements</tt><tt class="py-op">.</tt><tt id="link-132" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-132', 'append', 'link-79');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">tree_id</tt><tt class="py-op">,</tt> <tt class="py-name">predicate</tt><tt class="py-op">,</tt> <tt class="py-name">obj</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line"> </tt>
<a name="L358"></a><tt class="py-lineno">358</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-133" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-133', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt id="link-134" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-134', 'name', 'link-34');">name</a></tt><tt class="py-op">:</tt> </tt>
<a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line">            <tt class="py-comment"># create TU</tt> </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tu_counter</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line">            <tt class="py-name">tu_uri</tt> <tt class="py-op">=</tt> <tt class="py-string">"tu%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tu_counter</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">zfill</tt><tt class="py-op">(</tt><tt id="link-135" class="py-name"><a title="Bio.Phylo.CDAOIO.ZEROES" class="py-name" href="#" onclick="return doclink('link-135', 'ZEROES', 'link-19');">ZEROES</a></tt><tt class="py-op">)</tt> </tt>
<a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line"> </tt>
<a name="L363"></a><tt class="py-lineno">363</tt>  <tt class="py-line">            <tt class="py-name">statements</tt> <tt class="py-op">+=</tt> <tt class="py-op">[</tt> </tt>
<a name="L364"></a><tt class="py-lineno">364</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt class="py-name">tu_uri</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">pUri</tt><tt class="py-op">(</tt><tt class="py-string">"rdf:type"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">pUri</tt><tt class="py-op">(</tt><tt class="py-string">"cdao:TU"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L365"></a><tt class="py-lineno">365</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt id="link-136" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-136', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt id="link-137" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-137', 'uri', 'link-107');">uri</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">pUri</tt><tt class="py-op">(</tt> </tt>
<a name="L366"></a><tt class="py-lineno">366</tt>  <tt class="py-line">                    <tt class="py-string">"cdao:represents_TU"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt class="py-name">tu_uri</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L367"></a><tt class="py-lineno">367</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt class="py-name">tu_uri</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">pUri</tt><tt class="py-op">(</tt><tt class="py-string">"rdfs:label"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L368"></a><tt class="py-lineno">368</tt>  <tt class="py-line">                 <tt class="py-name">rdflib</tt><tt class="py-op">.</tt><tt class="py-name">Literal</tt><tt class="py-op">(</tt><tt id="link-138" class="py-name" targets="Function Bio.Phylo.CDAOIO.format_label()=Bio.Phylo.CDAOIO-module.html#format_label"><a title="Bio.Phylo.CDAOIO.format_label" class="py-name" href="#" onclick="return doclink('link-138', 'format_label', 'link-138');">format_label</a></tt><tt class="py-op">(</tt><tt id="link-139" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-139', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt id="link-140" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-140', 'name', 'link-34');">name</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line">            <tt class="py-op">]</tt> </tt>
<a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line"> </tt>
<a name="L371"></a><tt class="py-lineno">371</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L372"></a><tt class="py-lineno">372</tt>  <tt class="py-line">                <tt class="py-name">tu_attributes</tt> <tt class="py-op">=</tt> <tt id="link-141" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-141', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt class="py-name">tu_attributes</tt> </tt>
<a name="L373"></a><tt class="py-lineno">373</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line">                <tt class="py-name">tu_attributes</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line"> </tt>
<a name="L376"></a><tt class="py-lineno">376</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">predicate</tt><tt class="py-op">,</tt> <tt class="py-name">obj</tt> <tt class="py-keyword">in</tt> <tt class="py-name">tu_attributes</tt><tt class="py-op">:</tt> </tt>
<a name="L377"></a><tt class="py-lineno">377</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt class="py-op">(</tt><tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt class="py-name">tu_uri</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">predicate</tt><tt class="py-op">,</tt> <tt class="py-name">obj</tt><tt class="py-op">)</tt> </tt>
<a name="L378"></a><tt class="py-lineno">378</tt>  <tt class="py-line"> </tt>
<a name="L379"></a><tt class="py-lineno">379</tt>  <tt class="py-line">        <tt class="py-comment"># create this node</tt> </tt>
<a name="L380"></a><tt class="py-lineno">380</tt>  <tt class="py-line">        <tt id="link-142" class="py-name" targets="Variable Bio.Phylo._cdao_owl.node_type=Bio.Phylo._cdao_owl-module.html#node_type"><a title="Bio.Phylo._cdao_owl.node_type" class="py-name" href="#" onclick="return doclink('link-142', 'node_type', 'link-142');">node_type</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"cdao:TerminalNode"</tt> <tt class="py-keyword">if</tt> <tt id="link-143" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-143', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt id="link-144" class="py-name" targets="Method Bio.Nexus.Trees.Tree.is_terminal()=Bio.Nexus.Trees.Tree-class.html#is_terminal,Method Bio.Phylo.BaseTree.Clade.is_terminal()=Bio.Phylo.BaseTree.Clade-class.html#is_terminal,Method Bio.Phylo.BaseTree.Tree.is_terminal()=Bio.Phylo.BaseTree.Tree-class.html#is_terminal"><a title="Bio.Nexus.Trees.Tree.is_terminal
Bio.Phylo.BaseTree.Clade.is_terminal
Bio.Phylo.BaseTree.Tree.is_terminal" class="py-name" href="#" onclick="return doclink('link-144', 'is_terminal', 'link-144');">is_terminal</a></tt><tt class="py-op">(</tt> </tt>
<a name="L381"></a><tt class="py-lineno">381</tt>  <tt class="py-line">        <tt class="py-op">)</tt> <tt class="py-keyword">else</tt> <tt class="py-string">"cdao:AncestralNode"</tt> </tt>
<a name="L382"></a><tt class="py-lineno">382</tt>  <tt class="py-line">        <tt class="py-name">statements</tt> <tt class="py-op">+=</tt> <tt class="py-op">[</tt> </tt>
<a name="L383"></a><tt class="py-lineno">383</tt>  <tt class="py-line">            <tt class="py-op">(</tt><tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt id="link-145" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-145', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt id="link-146" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-146', 'uri', 'link-107');">uri</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">pUri</tt><tt class="py-op">(</tt><tt class="py-string">"rdf:type"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">pUri</tt><tt class="py-op">(</tt><tt id="link-147" class="py-name"><a title="Bio.Phylo._cdao_owl.node_type" class="py-name" href="#" onclick="return doclink('link-147', 'node_type', 'link-142');">node_type</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L384"></a><tt class="py-lineno">384</tt>  <tt class="py-line">            <tt class="py-op">(</tt><tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt id="link-148" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-148', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt id="link-149" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-149', 'uri', 'link-107');">uri</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">pUri</tt><tt class="py-op">(</tt> </tt>
<a name="L385"></a><tt class="py-lineno">385</tt>  <tt class="py-line">                <tt class="py-string">"cdao:belongs_to_Tree"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">tree_id</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L386"></a><tt class="py-lineno">386</tt>  <tt class="py-line">        <tt class="py-op">]</tt> </tt>
<a name="L387"></a><tt class="py-lineno">387</tt>  <tt class="py-line"> </tt>
<a name="L388"></a><tt class="py-lineno">388</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">parent</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L389"></a><tt class="py-lineno">389</tt>  <tt class="py-line">            <tt class="py-comment"># create edge from the parent node to this node</tt> </tt>
<a name="L390"></a><tt class="py-lineno">390</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">edge_counter</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L391"></a><tt class="py-lineno">391</tt>  <tt class="py-line">            <tt class="py-name">edge_uri</tt> <tt class="py-op">=</tt> <tt class="py-string">"edge%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">edge_counter</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">zfill</tt><tt class="py-op">(</tt><tt id="link-150" class="py-name"><a title="Bio.Phylo.CDAOIO.ZEROES" class="py-name" href="#" onclick="return doclink('link-150', 'ZEROES', 'link-19');">ZEROES</a></tt><tt class="py-op">)</tt> </tt>
<a name="L392"></a><tt class="py-lineno">392</tt>  <tt class="py-line"> </tt>
<a name="L393"></a><tt class="py-lineno">393</tt>  <tt class="py-line">            <tt class="py-name">statements</tt> <tt class="py-op">+=</tt> <tt class="py-op">[</tt> </tt>
<a name="L394"></a><tt class="py-lineno">394</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt class="py-name">edge_uri</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">pUri</tt><tt class="py-op">(</tt><tt class="py-string">"rdf:type"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">pUri</tt><tt class="py-op">(</tt><tt class="py-string">"cdao:DirectedEdge"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L395"></a><tt class="py-lineno">395</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt class="py-name">edge_uri</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">pUri</tt><tt class="py-op">(</tt> </tt>
<a name="L396"></a><tt class="py-lineno">396</tt>  <tt class="py-line">                    <tt class="py-string">"cdao:belongs_to_Tree"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">tree_id</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L397"></a><tt class="py-lineno">397</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt class="py-name">edge_uri</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">pUri</tt><tt class="py-op">(</tt><tt class="py-string">"cdao:has_Parent_Node"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L398"></a><tt class="py-lineno">398</tt>  <tt class="py-line">                 <tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt class="py-name">parent</tt><tt class="py-op">.</tt><tt id="link-151" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-151', 'uri', 'link-107');">uri</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L399"></a><tt class="py-lineno">399</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt class="py-name">edge_uri</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">pUri</tt><tt class="py-op">(</tt><tt class="py-string">"cdao:has_Child_Node"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L400"></a><tt class="py-lineno">400</tt>  <tt class="py-line">                 <tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt id="link-152" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-152', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt id="link-153" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-153', 'uri', 'link-107');">uri</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L401"></a><tt class="py-lineno">401</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt id="link-154" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-154', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt id="link-155" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-155', 'uri', 'link-107');">uri</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">pUri</tt><tt class="py-op">(</tt> </tt>
<a name="L402"></a><tt class="py-lineno">402</tt>  <tt class="py-line">                    <tt class="py-string">"cdao:belongs_to_Edge_as_Child"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt class="py-name">edge_uri</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L403"></a><tt class="py-lineno">403</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt id="link-156" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-156', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt id="link-157" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-157', 'uri', 'link-107');">uri</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">pUri</tt><tt class="py-op">(</tt><tt class="py-string">"cdao:has_Parent"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L404"></a><tt class="py-lineno">404</tt>  <tt class="py-line">                 <tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt class="py-name">parent</tt><tt class="py-op">.</tt><tt id="link-158" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-158', 'uri', 'link-107');">uri</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L405"></a><tt class="py-lineno">405</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt class="py-name">parent</tt><tt class="py-op">.</tt><tt id="link-159" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-159', 'uri', 'link-107');">uri</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">pUri</tt><tt class="py-op">(</tt> </tt>
<a name="L406"></a><tt class="py-lineno">406</tt>  <tt class="py-line">                    <tt class="py-string">"cdao:belongs_to_Edge_as_Parent"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt class="py-name">edge_uri</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L407"></a><tt class="py-lineno">407</tt>  <tt class="py-line">            <tt class="py-op">]</tt> </tt>
<a name="L408"></a><tt class="py-lineno">408</tt>  <tt class="py-line"> </tt>
<a name="L409"></a><tt class="py-lineno">409</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-160" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-160', 'clade', 'link-39');">clade</a></tt><tt class="py-op">,</tt> <tt class="py-string">"confidence"</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt id="link-161" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-161', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt id="link-162" class="py-name" targets="Variable Bio.Phylo.PhyloXML.Clade.confidence=Bio.Phylo.PhyloXML.Clade-class.html#confidence,Variable Bio.Phylo.PhyloXML.Phylogeny.confidence=Bio.Phylo.PhyloXML.Phylogeny-class.html#confidence,Method Bio.Phylo.PhyloXMLIO.Parser.confidence()=Bio.Phylo.PhyloXMLIO.Parser-class.html#confidence,Method Bio.Phylo.PhyloXMLIO.Writer.confidence()=Bio.Phylo.PhyloXMLIO.Writer-class.html#confidence"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-162', 'confidence', 'link-162');">confidence</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L410"></a><tt class="py-lineno">410</tt>  <tt class="py-line">                <tt id="link-163" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-163', 'confidence', 'link-162');">confidence</a></tt> <tt class="py-op">=</tt> <tt class="py-name">rdflib</tt><tt class="py-op">.</tt><tt class="py-name">Literal</tt><tt class="py-op">(</tt> </tt>
<a name="L411"></a><tt class="py-lineno">411</tt>  <tt class="py-line">                    <tt id="link-164" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-164', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt id="link-165" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-165', 'confidence', 'link-162');">confidence</a></tt><tt class="py-op">,</tt> <tt class="py-name">datatype</tt><tt class="py-op">=</tt><tt class="py-string">"http://www.w3.org/2001/XMLSchema#decimal"</tt><tt class="py-op">)</tt> </tt>
<a name="L412"></a><tt class="py-lineno">412</tt>  <tt class="py-line"> </tt>
<a name="L413"></a><tt class="py-lineno">413</tt>  <tt class="py-line">                <tt class="py-name">statements</tt> <tt class="py-op">+=</tt> <tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt id="link-166" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-166', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt id="link-167" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-167', 'uri', 'link-107');">uri</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L414"></a><tt class="py-lineno">414</tt>  <tt class="py-line">                                <tt class="py-name">pUri</tt><tt class="py-op">(</tt><tt class="py-string">"cdao:has_Support_Value"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-168" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-168', 'confidence', 'link-162');">confidence</a></tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L415"></a><tt class="py-lineno">415</tt>  <tt class="py-line"> </tt>
<a name="L416"></a><tt class="py-lineno">416</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">record_complete_ancestry</tt> <tt class="py-keyword">and</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-169" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-169', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt class="py-name">ancestors</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L417"></a><tt class="py-lineno">417</tt>  <tt class="py-line">                <tt class="py-name">statements</tt> <tt class="py-op">+=</tt> <tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt id="link-170" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-170', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt id="link-171" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-171', 'uri', 'link-107');">uri</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">pUri</tt><tt class="py-op">(</tt><tt class="py-string">"cdao:has_Ancestor"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt class="py-name">ancestor</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L418"></a><tt class="py-lineno">418</tt>  <tt class="py-line">                               <tt class="py-keyword">for</tt> <tt class="py-name">ancestor</tt> <tt class="py-keyword">in</tt> <tt id="link-172" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-172', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt class="py-name">ancestors</tt><tt class="py-op">]</tt> </tt>
<a name="L419"></a><tt class="py-lineno">419</tt>  <tt class="py-line"> </tt>
<a name="L420"></a><tt class="py-lineno">420</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-173" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-173', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt id="link-174" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.branch_length()=Bio.Phylo.PhyloXMLIO.Writer-class.html#branch_length"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-174', 'branch_length', 'link-174');">branch_length</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L421"></a><tt class="py-lineno">421</tt>  <tt class="py-line">                <tt class="py-comment"># add branch length</tt> </tt>
<a name="L422"></a><tt class="py-lineno">422</tt>  <tt class="py-line">                <tt class="py-name">edge_ann_uri</tt> <tt class="py-op">=</tt> <tt class="py-string">"edge_annotation%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt> </tt>
<a name="L423"></a><tt class="py-lineno">423</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">edge_counter</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">zfill</tt><tt class="py-op">(</tt><tt id="link-175" class="py-name"><a title="Bio.Phylo.CDAOIO.ZEROES" class="py-name" href="#" onclick="return doclink('link-175', 'ZEROES', 'link-19');">ZEROES</a></tt><tt class="py-op">)</tt> </tt>
<a name="L424"></a><tt class="py-lineno">424</tt>  <tt class="py-line"> </tt>
<a name="L425"></a><tt class="py-lineno">425</tt>  <tt class="py-line">                <tt id="link-176" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-176', 'branch_length', 'link-174');">branch_length</a></tt> <tt class="py-op">=</tt> <tt class="py-name">rdflib</tt><tt class="py-op">.</tt><tt class="py-name">Literal</tt><tt class="py-op">(</tt><tt id="link-177" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-177', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt id="link-178" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-178', 'branch_length', 'link-174');">branch_length</a></tt><tt class="py-op">,</tt> <tt class="py-name">datatype</tt><tt class="py-op">=</tt><tt class="py-name">rdflib</tt><tt class="py-op">.</tt><tt class="py-name">URIRef</tt><tt class="py-op">(</tt> </tt>
<a name="L426"></a><tt class="py-lineno">426</tt>  <tt class="py-line">                    <tt class="py-string">"http://www.w3.org/2001/XMLSchema#decimal"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L427"></a><tt class="py-lineno">427</tt>  <tt class="py-line">                <tt class="py-name">statements</tt> <tt class="py-op">+=</tt> <tt class="py-op">[</tt> </tt>
<a name="L428"></a><tt class="py-lineno">428</tt>  <tt class="py-line">                    <tt class="py-op">(</tt><tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt class="py-name">edge_ann_uri</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">pUri</tt><tt class="py-op">(</tt><tt class="py-string">"rdf:type"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L429"></a><tt class="py-lineno">429</tt>  <tt class="py-line">                     <tt class="py-name">pUri</tt><tt class="py-op">(</tt><tt class="py-string">"cdao:EdgeLength"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L430"></a><tt class="py-lineno">430</tt>  <tt class="py-line">                    <tt class="py-op">(</tt><tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt class="py-name">edge_uri</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">pUri</tt><tt class="py-op">(</tt><tt class="py-string">"cdao:has_Annotation"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L431"></a><tt class="py-lineno">431</tt>  <tt class="py-line">                     <tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt class="py-name">edge_ann_uri</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L432"></a><tt class="py-lineno">432</tt>  <tt class="py-line">                    <tt class="py-op">(</tt><tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt class="py-name">edge_ann_uri</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L433"></a><tt class="py-lineno">433</tt>  <tt class="py-line">                     <tt class="py-name">pUri</tt><tt class="py-op">(</tt><tt class="py-string">"cdao:has_Value"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-179" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-179', 'branch_length', 'link-174');">branch_length</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L434"></a><tt class="py-lineno">434</tt>  <tt class="py-line">                <tt class="py-op">]</tt> </tt>
<a name="L435"></a><tt class="py-lineno">435</tt>  <tt class="py-line"> </tt>
<a name="L436"></a><tt class="py-lineno">436</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L437"></a><tt class="py-lineno">437</tt>  <tt class="py-line">                <tt class="py-name">edge_attributes</tt> <tt class="py-op">=</tt> <tt id="link-180" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-180', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt class="py-name">edge_attributes</tt> </tt>
<a name="L438"></a><tt class="py-lineno">438</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L439"></a><tt class="py-lineno">439</tt>  <tt class="py-line">                <tt class="py-name">edge_attributes</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L440"></a><tt class="py-lineno">440</tt>  <tt class="py-line"> </tt>
<a name="L441"></a><tt class="py-lineno">441</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">predicate</tt><tt class="py-op">,</tt> <tt class="py-name">obj</tt> <tt class="py-keyword">in</tt> <tt class="py-name">edge_attributes</tt><tt class="py-op">:</tt> </tt>
<a name="L442"></a><tt class="py-lineno">442</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt class="py-op">(</tt><tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt class="py-name">edge_uri</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">predicate</tt><tt class="py-op">,</tt> <tt class="py-name">obj</tt><tt class="py-op">)</tt> </tt>
<a name="L443"></a><tt class="py-lineno">443</tt>  <tt class="py-line"> </tt>
<a name="L444"></a><tt class="py-lineno">444</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">stmt</tt> <tt class="py-keyword">in</tt> <tt class="py-name">statements</tt><tt class="py-op">:</tt> </tt>
<a name="L445"></a><tt class="py-lineno">445</tt>  <tt class="py-line">            <tt class="py-keyword">yield</tt> <tt class="py-name">stmt</tt> </tt>
<a name="L446"></a><tt class="py-lineno">446</tt>  <tt class="py-line"> </tt>
<a name="L447"></a><tt class="py-lineno">447</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L448"></a><tt class="py-lineno">448</tt>  <tt class="py-line">            <tt class="py-name">clade_attributes</tt> <tt class="py-op">=</tt> <tt id="link-181" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-181', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt class="py-name">attributes</tt> </tt>
<a name="L449"></a><tt class="py-lineno">449</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L450"></a><tt class="py-lineno">450</tt>  <tt class="py-line">            <tt class="py-name">clade_attributes</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L451"></a><tt class="py-lineno">451</tt>  <tt class="py-line"> </tt>
<a name="L452"></a><tt class="py-lineno">452</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">predicate</tt><tt class="py-op">,</tt> <tt class="py-name">obj</tt> <tt class="py-keyword">in</tt> <tt class="py-name">clade_attributes</tt><tt class="py-op">:</tt> </tt>
<a name="L453"></a><tt class="py-lineno">453</tt>  <tt class="py-line">            <tt class="py-keyword">yield</tt> <tt class="py-op">(</tt><tt class="py-name">nUri</tt><tt class="py-op">(</tt><tt id="link-182" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-182', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt id="link-183" class="py-name"><a title="Bio.Phylo.NeXMLIO.uri
Bio.Phylo.PhyloXMLIO.Parser.uri
Bio.Phylo.PhyloXMLIO.Writer.uri
Bio.Phylo.PhyloXMLIO.uri" class="py-name" href="#" onclick="return doclink('link-183', 'uri', 'link-107');">uri</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">predicate</tt><tt class="py-op">,</tt> <tt class="py-name">obj</tt><tt class="py-op">)</tt> </tt>
<a name="L454"></a><tt class="py-lineno">454</tt>  <tt class="py-line"> </tt>
<a name="L455"></a><tt class="py-lineno">455</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-184" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-184', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt id="link-185" class="py-name"><a title="Bio.Nexus.Trees.Tree.is_terminal
Bio.Phylo.BaseTree.Clade.is_terminal
Bio.Phylo.BaseTree.Tree.is_terminal" class="py-name" href="#" onclick="return doclink('link-185', 'is_terminal', 'link-144');">is_terminal</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L456"></a><tt class="py-lineno">456</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt id="link-186" class="py-name"><a title="Bio.Phylo.CDAOIO.Parser.new_clade
Bio.Phylo.NewickIO.Parser.new_clade" class="py-name" href="#" onclick="return doclink('link-186', 'new_clade', 'link-82');">new_clade</a></tt> <tt class="py-keyword">in</tt> <tt id="link-187" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-187', 'clade', 'link-39');">clade</a></tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">:</tt> </tt>
<a name="L457"></a><tt class="py-lineno">457</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">stmt</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-188" class="py-name"><a title="Bio.Phylo.CDAOIO.Writer.process_clade
Bio.Phylo.NewickIO.Parser.process_clade" class="py-name" href="#" onclick="return doclink('link-188', 'process_clade', 'link-103');">process_clade</a></tt><tt class="py-op">(</tt><tt id="link-189" class="py-name"><a title="Bio.Phylo.CDAOIO.Parser.new_clade
Bio.Phylo.NewickIO.Parser.new_clade" class="py-name" href="#" onclick="return doclink('link-189', 'new_clade', 'link-82');">new_clade</a></tt><tt class="py-op">,</tt> <tt class="py-name">parent</tt><tt class="py-op">=</tt><tt id="link-190" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-190', 'clade', 'link-39');">clade</a></tt><tt class="py-op">,</tt> <tt id="link-191" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-191', 'root', 'link-43');">root</a></tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L458"></a><tt class="py-lineno">458</tt>  <tt class="py-line">                    <tt class="py-keyword">yield</tt> <tt class="py-name">stmt</tt> </tt>
</div></div><a name="L459"></a><tt class="py-lineno">459</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Nov  6 13:34:09 2019
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
