<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.motifs.jaspar.db.JASPAR5</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.motifs-module.html">Package&nbsp;motifs</a> ::
        <a href="Bio.motifs.jaspar-module.html">Package&nbsp;jaspar</a> ::
        <a href="Bio.motifs.jaspar.db-module.html">Module&nbsp;db</a> ::
        Class&nbsp;JASPAR5
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.motifs.jaspar.db.JASPAR5-class.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== CLASS DESCRIPTION ==================== -->
<h1 class="epydoc">Class JASPAR5</h1><p class="nomargin-top"><span class="codelink"><a href="Bio.motifs.jaspar.db-pysrc.html#JASPAR5">source&nbsp;code</a></span></p>
<pre class="base-tree">
object --+
         |
        <strong class="uidshort">JASPAR5</strong>
</pre>

<hr />
<p>Class representing a JASPAR5 database.</p>
<p>Class representing a JASPAR5 DB. The methods within are loosely based
on the perl TFBS::DB::JASPAR5 module.</p>
<p>Note: We will only implement reading of JASPAR motifs from the DB.
Unlike the perl module, we will not attempt to implement any methods to
store JASPAR motifs or create a new DB at this time.</p>

<!-- ==================== INSTANCE METHODS ==================== -->
<a name="section-InstanceMethods"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Instance Methods</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-InstanceMethods"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.motifs.jaspar.db.JASPAR5-class.html#__init__" class="summary-sig-name">__init__</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">host</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">name</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">user</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">password</span>=<span class="summary-sig-default">None</span>)</span><br />
      Construct a JASPAR5 instance and connect to specified DB.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.motifs.jaspar.db-pysrc.html#JASPAR5.__init__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.motifs.jaspar.db.JASPAR5-class.html#__str__" class="summary-sig-name">__str__</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Return a string represention of the JASPAR5 DB connection.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.motifs.jaspar.db-pysrc.html#JASPAR5.__str__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.motifs.jaspar.db.JASPAR5-class.html#fetch_motif_by_id" class="summary-sig-name">fetch_motif_by_id</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">id</span>)</span><br />
      Fetch a single JASPAR motif from the DB by it's JASPAR matrix ID.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.motifs.jaspar.db-pysrc.html#JASPAR5.fetch_motif_by_id">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.motifs.jaspar.db.JASPAR5-class.html#fetch_motifs_by_name" class="summary-sig-name">fetch_motifs_by_name</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">name</span>)</span><br />
      Fetch a list of JASPAR motifs from a JASPAR DB by the given TF name(s).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.motifs.jaspar.db-pysrc.html#JASPAR5.fetch_motifs_by_name">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.motifs.jaspar.db.JASPAR5-class.html#fetch_motifs" class="summary-sig-name">fetch_motifs</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">collection</span>=<span class="summary-sig-default">JASPAR_DFLT_COLLECTION</span>,
        <span class="summary-sig-arg">tf_name</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">tf_class</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">tf_family</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">matrix_id</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">tax_group</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">species</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">pazar_id</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">data_type</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">medline</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">min_ic</span>=<span class="summary-sig-default">0</span>,
        <span class="summary-sig-arg">min_length</span>=<span class="summary-sig-default">0</span>,
        <span class="summary-sig-arg">min_sites</span>=<span class="summary-sig-default">0</span>,
        <span class="summary-sig-arg">all</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">all_versions</span>=<span class="summary-sig-default">False</span>)</span><br />
      Fetch jaspar.Record (list) of motifs using selection criteria.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.motifs.jaspar.db-pysrc.html#JASPAR5.fetch_motifs">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_fetch_latest_version"></a><span class="summary-sig-name">_fetch_latest_version</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">base_id</span>)</span><br />
      Get the latest version number for the given base_id (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.motifs.jaspar.db-pysrc.html#JASPAR5._fetch_latest_version">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.motifs.jaspar.db.JASPAR5-class.html#_fetch_internal_id" class="summary-sig-name" onclick="show_private();">_fetch_internal_id</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">base_id</span>,
        <span class="summary-sig-arg">version</span>)</span><br />
      Fetch the internal id for a base id + version (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.motifs.jaspar.db-pysrc.html#JASPAR5._fetch_internal_id">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_fetch_motif_by_internal_id"></a><span class="summary-sig-name">_fetch_motif_by_internal_id</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">int_id</span>)</span><br />
      Fetch basic motif information (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.motifs.jaspar.db-pysrc.html#JASPAR5._fetch_motif_by_internal_id">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.motifs.jaspar.db.JASPAR5-class.html#_fetch_counts_matrix" class="summary-sig-name" onclick="show_private();">_fetch_counts_matrix</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">int_id</span>)</span><br />
      Fetch the counts matrix from the JASPAR DB by the internal ID (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.motifs.jaspar.db-pysrc.html#JASPAR5._fetch_counts_matrix">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.motifs.jaspar.db.JASPAR5-class.html#_fetch_internal_id_list" class="summary-sig-name" onclick="show_private();">_fetch_internal_id_list</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">collection</span>=<span class="summary-sig-default">JASPAR_DFLT_COLLECTION</span>,
        <span class="summary-sig-arg">tf_name</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">tf_class</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">tf_family</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">matrix_id</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">tax_group</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">species</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">pazar_id</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">data_type</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">medline</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">all</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">all_versions</span>=<span class="summary-sig-default">False</span>)</span><br />
      Fetch list of internal JASPAR motif IDs.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.motifs.jaspar.db-pysrc.html#JASPAR5._fetch_internal_id_list">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.motifs.jaspar.db.JASPAR5-class.html#_is_latest_version" class="summary-sig-name" onclick="show_private();">_is_latest_version</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">int_id</span>)</span><br />
      Check if the internal ID represents the latest JASPAR matrix (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.motifs.jaspar.db-pysrc.html#JASPAR5._is_latest_version">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code>object</code></b>:
      <code>__delattr__</code>,
      <code>__format__</code>,
      <code>__getattribute__</code>,
      <code>__hash__</code>,
      <code>__new__</code>,
      <code>__reduce__</code>,
      <code>__reduce_ex__</code>,
      <code>__repr__</code>,
      <code>__setattr__</code>,
      <code>__sizeof__</code>,
      <code>__subclasshook__</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== PROPERTIES ==================== -->
<a name="section-Properties"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Properties</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Properties"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code>object</code></b>:
      <code>__class__</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== METHOD DETAILS ==================== -->
<a name="section-MethodDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Method Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-MethodDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="__init__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__init__</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">host</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">name</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">user</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">password</span>=<span class="sig-default">None</span>)</span>
    <br /><em class="fname">(Constructor)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.motifs.jaspar.db-pysrc.html#JASPAR5.__init__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Construct a JASPAR5 instance and connect to specified DB.</p>
<p>Arguments:</p>
<ul class="rst-simple">
<li>host - host name of the the JASPAR DB server</li>
<li>name - name of the JASPAR database</li>
<li>user - user name to connect to the JASPAR DB</li>
<li>password - JASPAR DB password</li>
</ul>
  <dl class="fields">
    <dt>Overrides:
        object.__init__
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="__str__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__str__</span>(<span class="sig-arg">self</span>)</span>
    <br /><em class="fname">(Informal representation operator)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.motifs.jaspar.db-pysrc.html#JASPAR5.__str__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  Return a string represention of the JASPAR5 DB connection.
  <dl class="fields">
    <dt>Overrides:
        object.__str__
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="fetch_motif_by_id"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">fetch_motif_by_id</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">id</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.motifs.jaspar.db-pysrc.html#JASPAR5.fetch_motif_by_id">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Fetch a single JASPAR motif from the DB by it's JASPAR matrix ID.</p>
<p>Example id 'MA0001.1'.</p>
<p>Arguments:</p>
<blockquote>
<ul class="rst-simple">
<li>id - JASPAR matrix ID. This may be a fully specified ID including
the version number (e.g. MA0049.2) or just the base ID (e.g.
MA0049). If only a base ID is provided, the latest version is
returned.</li>
</ul>
</blockquote>
<dl class="rst-docutils">
<dt>Returns:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>A Bio.motifs.jaspar.Motif object</li>
</ul>
</dd>
</dl>
<p><strong>NOTE:</strong> The perl TFBS module allows you to specify the type of matrix
to return (PFM, PWM, ICM) but matrices are always stored in JASPAR as
PFMs so this does not really belong here. Once a PFM is fetched the
pwm() and pssm() methods can be called to return the normalized and
log-odds matrices.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="fetch_motifs_by_name"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">fetch_motifs_by_name</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">name</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.motifs.jaspar.db-pysrc.html#JASPAR5.fetch_motifs_by_name">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Fetch a list of JASPAR motifs from a JASPAR DB by the given TF name(s).</p>
<p>Arguments:
name - a single name or list of names
Returns:
A list of Bio.motifs.jaspar.Motif objects</p>
<p>Notes:
Names are not guaranteed to be unique. There may be more than one
motif with the same name. Therefore even if name specifies a single
name, a list of motifs is returned. This just calls
self.fetch_motifs(collection = None, tf_name = name).</p>
<p>This behaviour is different from the TFBS perl module's
get_Matrix_by_name() method which always returns a single matrix,
issuing a warning message and returning the first matrix retrieved
in the case where multiple matrices have the same name.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="fetch_motifs"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">fetch_motifs</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">collection</span>=<span class="sig-default">JASPAR_DFLT_COLLECTION</span>,
        <span class="sig-arg">tf_name</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">tf_class</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">tf_family</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">matrix_id</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">tax_group</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">species</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">pazar_id</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">data_type</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">medline</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">min_ic</span>=<span class="sig-default">0</span>,
        <span class="sig-arg">min_length</span>=<span class="sig-default">0</span>,
        <span class="sig-arg">min_sites</span>=<span class="sig-default">0</span>,
        <span class="sig-arg">all</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">all_versions</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.motifs.jaspar.db-pysrc.html#JASPAR5.fetch_motifs">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Fetch jaspar.Record (list) of motifs using selection criteria.</p>
<p>Arguments:</p>
<pre class="rst-literal-block">
Except where obvious, all selection criteria arguments may be
specified as a single value or a list of values. Motifs must
meet ALL the specified selection criteria to be returned with
the precedent exceptions noted below.

all         - Takes precedent of all other selection criteria.
              Every motif is returned. If 'all_versions' is also
              specified, all versions of every motif are returned,
              otherwise just the latest version of every motif is
              returned.
matrix_id   - Takes precedence over all other selection criteria
              except 'all'.  Only motifs with the given JASPAR
              matrix ID(s) are returned. A matrix ID may be
              specified as just a base ID or full JASPAR IDs
              including version number. If only a base ID is
              provided for specific motif(s), then just the latest
              version of those motif(s) are returned unless
              'all_versions' is also specified.
collection  - Only motifs from the specified JASPAR collection(s)
              are returned. NOTE - if not specified, the collection
              defaults to CORE for all other selection criteria
              except 'all' and 'matrix_id'. To apply the other
              selection criteria across all JASPAR collections,
              explicitly set collection=None.
tf_name     - Only motifs with the given name(s) are returned.
tf_class    - Only motifs of the given TF class(es) are returned.
tf_family   - Only motifs from the given TF families are returned.
tax_group   - Only motifs belonging to the given taxonomic
              supergroups are returned (e.g. 'vertebrates',
              'insects', 'nematodes' etc.)
species     - Only motifs derived from the given species are
              returned.  Species are specified as taxonomy IDs.
data_type   - Only motifs generated with the given data type (e.g.
              ('ChIP-seq', 'PBM', 'SELEX' etc.) are returned.
              NOTE - must match exactly as stored in the database.
pazar_id    - Only motifs with the given PAZAR TF ID are returned.
medline     - Only motifs with the given medline (PubmMed IDs) are
              returned.
min_ic      - Only motifs whose profile matrices have at least this
              information content (specificty) are returned.
min_length  - Only motifs whose profiles are of at least this
              length are returned.
min_sites   - Only motifs compiled from at least these many binding
              sites are returned.
all_versions- Unless specified, just the latest version of motifs
              determined by the other selection criteria are
              returned. Otherwise all versions of the selected
              motifs are returned.
</pre>
<dl class="rst-docutils">
<dt>Returns:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>A Bio.motifs.jaspar.Record (list) of motifs.</li>
</ul>
</dd>
</dl>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_fetch_internal_id"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_fetch_internal_id</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">base_id</span>,
        <span class="sig-arg">version</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.motifs.jaspar.db-pysrc.html#JASPAR5._fetch_internal_id">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Fetch the internal id for a base id + version (PRIVATE).</p>
<p>Also checks if this combo exists or not.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_fetch_counts_matrix"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_fetch_counts_matrix</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">int_id</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.motifs.jaspar.db-pysrc.html#JASPAR5._fetch_counts_matrix">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Fetch the counts matrix from the JASPAR DB by the internal ID (PRIVATE).</p>
<p>Returns a Bio.motifs.matrix.GenericPositionMatrix</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_fetch_internal_id_list"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_fetch_internal_id_list</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">collection</span>=<span class="sig-default">JASPAR_DFLT_COLLECTION</span>,
        <span class="sig-arg">tf_name</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">tf_class</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">tf_family</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">matrix_id</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">tax_group</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">species</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">pazar_id</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">data_type</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">medline</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">all</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">all_versions</span>=<span class="sig-default">False</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.motifs.jaspar.db-pysrc.html#JASPAR5._fetch_internal_id_list">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Fetch list of internal JASPAR motif IDs.</p>
<p>Fetch a list of internal JASPAR motif IDs based on various passed
parameters which may then be used to fetch the rest of the motif data.</p>
<dl class="rst-docutils">
<dt>Caller:</dt>
<dd>fetch_motifs()</dd>
<dt>Arguments:</dt>
<dd>See arguments sections of fetch_motifs()</dd>
<dt>Returns:</dt>
<dd>A list of internal JASPAR motif IDs which match the given
selection criteria arguments.</dd>
</dl>
<p>Build an SQL query based on the selection arguments provided.</p>
<dl class="rst-docutils">
<dt>1: First add table joins and sub-clauses for criteria corresponding to</dt>
<dd>named fields from the MATRIX and MATRIX_SPECIES tables such as
collection, matrix ID, name, species etc.</dd>
<dt>2: Then add joins/sub-clauses for tag/value parameters from the</dt>
<dd>MATRIX_ANNOTATION table.</dd>
</dl>
<p>For the surviving matrices, the responsibility to do matrix-based
feature filtering such as ic, number of sites etc, fall on the
calling fetch_motifs() method.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_is_latest_version"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_is_latest_version</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">int_id</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.motifs.jaspar.db-pysrc.html#JASPAR5._is_latest_version">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Check if the internal ID represents the latest JASPAR matrix (PRIVATE).</p>
<p>Does this internal ID represent the latest version of the JASPAR
matrix (collapse on base ids)</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Tue Jul 16 17:05:00 2019
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
