<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.codonalign</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        Package&nbsp;codonalign
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.codonalign-module.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== PACKAGE DESCRIPTION ==================== -->
<h1 class="epydoc">Package codonalign</h1><p class="nomargin-top"><span class="codelink"><a href="Bio.codonalign-pysrc.html">source&nbsp;code</a></span></p>
Code for dealing with Codon Alignments.

<!-- ==================== SUBMODULES ==================== -->
<a name="section-Submodules"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Submodules</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Submodules"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
  <tr><td class="summary">
  <ul class="nomargin">
    <li> <strong class="uidlink"><a href="Bio.codonalign.chisq-module.html">Bio.codonalign.chisq</a></strong>: <em class="summary">Python implementation of chisqprob, to avoid SciPy dependency.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.codonalign.codonalignment-module.html">Bio.codonalign.codonalignment</a></strong>: <em class="summary">Code for dealing with Codon Alignment.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.codonalign.codonalphabet-module.html">Bio.codonalign.codonalphabet</a></strong>: <em class="summary">Code for Codon Alphabet.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.codonalign.codonseq-module.html">Bio.codonalign.codonseq</a></strong>: <em class="summary">Code for dealing with coding sequence.</em>    </li>
  </ul></td></tr>
</table>

<br />
<!-- ==================== FUNCTIONS ==================== -->
<a name="section-Functions"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Functions</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Functions"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.codonalign-module.html#build" class="summary-sig-name">build</a>(<span class="summary-sig-arg">pro_align</span>,
        <span class="summary-sig-arg">nucl_seqs</span>,
        <span class="summary-sig-arg">corr_dict</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">gap_char</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">-</code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">unknown</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">X</code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">codon_table</span>=<span class="summary-sig-default">NCBICodonTable(id=1, names=['Standard', 'SGC0'], ...)</span>,
        <span class="summary-sig-arg">alphabet</span>=<span class="summary-sig-default">None</span>,
        <span class="summary-sig-arg">complete_protein</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">anchor_len</span>=<span class="summary-sig-default">10</span>,
        <span class="summary-sig-arg">max_score</span>=<span class="summary-sig-default">10</span>)</span><br />
      Build a codon alignment from protein alignment and corresponding nucleotides.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.codonalign-pysrc.html#build">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_codons2re"></a><span class="summary-sig-name">_codons2re</span>(<span class="summary-sig-arg">codons</span>)</span><br />
      Generate regular expression based on a given list of codons (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.codonalign-pysrc.html#_codons2re">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_get_aa_regex"></a><span class="summary-sig-name">_get_aa_regex</span>(<span class="summary-sig-arg">codon_table</span>,
        <span class="summary-sig-arg">stop</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">*</code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">unknown</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">X</code><code class="variable-quote">'</code></span>)</span><br />
      Set up the regular expression of a given CodonTable (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.codonalign-pysrc.html#_get_aa_regex">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.codonalign-module.html#_check_corr" class="summary-sig-name" onclick="show_private();">_check_corr</a>(<span class="summary-sig-arg">pro</span>,
        <span class="summary-sig-arg">nucl</span>,
        <span class="summary-sig-arg">gap_char</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">-</code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">codon_table</span>=<span class="summary-sig-default">NCBICodonTable(id=1, names=['Standard', 'SGC0'], ...)</span>,
        <span class="summary-sig-arg">complete_protein</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">anchor_len</span>=<span class="summary-sig-default">10</span>)</span><br />
      Check if the nucleotide can be translated into the protein (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.codonalign-pysrc.html#_check_corr">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.codonalign-module.html#_get_shift_anchor_re" class="summary-sig-name" onclick="show_private();">_get_shift_anchor_re</a>(<span class="summary-sig-arg">sh_anc</span>,
        <span class="summary-sig-arg">sh_nuc</span>,
        <span class="summary-sig-arg">shift_val</span>,
        <span class="summary-sig-arg">aa2re</span>,
        <span class="summary-sig-arg">anchor_len</span>,
        <span class="summary-sig-arg">shift_id_pos</span>)</span><br />
      Find a regular expression matching a potentially shifted anchor (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.codonalign-pysrc.html#_get_shift_anchor_re">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_merge_aa2re"></a><span class="summary-sig-name">_merge_aa2re</span>(<span class="summary-sig-arg">aa1</span>,
        <span class="summary-sig-arg">aa2</span>,
        <span class="summary-sig-arg">shift_val</span>,
        <span class="summary-sig-arg">aa2re</span>,
        <span class="summary-sig-arg">reid</span>)</span><br />
      Merge two amino acids based on detected frame shift value (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.codonalign-pysrc.html#_merge_aa2re">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.codonalign-module.html#_get_codon_rec" class="summary-sig-name" onclick="show_private();">_get_codon_rec</a>(<span class="summary-sig-arg">pro</span>,
        <span class="summary-sig-arg">nucl</span>,
        <span class="summary-sig-arg">span_mode</span>,
        <span class="summary-sig-arg">alphabet</span>,
        <span class="summary-sig-arg">gap_char</span>=<span class="summary-sig-default"><code class="variable-quote">'</code><code class="variable-string">-</code><code class="variable-quote">'</code></span>,
        <span class="summary-sig-arg">codon_table</span>=<span class="summary-sig-default">NCBICodonTable(id=1, names=['Standard', 'SGC0'], ...)</span>,
        <span class="summary-sig-arg">complete_protein</span>=<span class="summary-sig-default">False</span>,
        <span class="summary-sig-arg">max_score</span>=<span class="summary-sig-default">10</span>)</span><br />
      Generate codon alignment based on regular re match (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.codonalign-pysrc.html#_get_codon_rec">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="Bio.codonalign-module.html#_align_shift_recs" class="summary-sig-name" onclick="show_private();">_align_shift_recs</a>(<span class="summary-sig-arg">recs</span>)</span><br />
      Build alignment according to the frameshift detected by _check_corr (PRIVATE).</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="Bio.codonalign-pysrc.html#_align_shift_recs">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
</table>
<!-- ==================== VARIABLES ==================== -->
<a name="section-Variables"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Variables</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Variables"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a name="__package__"></a><span class="summary-name">__package__</span> = <code title="'Bio.codonalign'"><code class="variable-quote">'</code><code class="variable-string">Bio.codonalign</code><code class="variable-quote">'</code></code>
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
        <a href="Bio.codonalign-module.html#__warningregistry__" class="summary-name">__warningregistry__</a> = <code title="{('Bio.codonalign is an experimental module which may undergo signific\
ant changes prior to its future official release.',
  &lt;class 'Bio.BiopythonExperimentalWarning'&gt;,
  27): True}"><code class="variable-group">{</code><code class="variable-group">(</code><code class="variable-quote">'</code><code class="variable-string">Bio.codonalign is an experimental mod</code><code class="variable-ellipsis">...</code></code>
    </td>
  </tr>
</table>
<!-- ==================== FUNCTION DETAILS ==================== -->
<a name="section-FunctionDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Function Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-FunctionDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="build"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">build</span>(<span class="sig-arg">pro_align</span>,
        <span class="sig-arg">nucl_seqs</span>,
        <span class="sig-arg">corr_dict</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">gap_char</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">-</code><code class="variable-quote">'</code></span>,
        <span class="sig-arg">unknown</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">X</code><code class="variable-quote">'</code></span>,
        <span class="sig-arg">codon_table</span>=<span class="sig-default">NCBICodonTable(id=1, names=['Standard', 'SGC0'], ...)</span>,
        <span class="sig-arg">alphabet</span>=<span class="sig-default">None</span>,
        <span class="sig-arg">complete_protein</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">anchor_len</span>=<span class="sig-default">10</span>,
        <span class="sig-arg">max_score</span>=<span class="sig-default">10</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.codonalign-pysrc.html#build">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Build a codon alignment from protein alignment and corresponding nucleotides.</p>
<dl class="rst-docutils">
<dt>Arguments:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>pro_align  - a protein MultipleSeqAlignment object</li>
<li>nucl_seqs - an object returned by SeqIO.parse or SeqIO.index
or a collection of SeqRecord.</li>
<li>alphabet   - alphabet for the returned codon alignment</li>
<li>corr_dict  - a dict that maps protein id to nucleotide id</li>
<li>complete_protein - whether the sequence begins with a start
codon</li>
</ul>
</dd>
</dl>
<p>Return a CodonAlignment object.</p>
<p>The example below answers this Biostars question: <a class="rst-reference external" href="https://www.biostars.org/p/89741/" target="_top">https://www.biostars.org/p/89741/</a></p>
<pre class="py-doctest">
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Alphabet <span class="py-keyword">import</span> generic_dna, generic_protein
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Seq <span class="py-keyword">import</span> Seq
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.SeqRecord <span class="py-keyword">import</span> SeqRecord
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.Align <span class="py-keyword">import</span> MultipleSeqAlignment
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">from</span> Bio.codonalign <span class="py-keyword">import</span> build
<span class="py-prompt">&gt;&gt;&gt; </span>seq1 = SeqRecord(Seq(<span class="py-string">'ATGTCTCGT'</span>, alphabet=generic_dna), id=<span class="py-string">'pro1'</span>)
<span class="py-prompt">&gt;&gt;&gt; </span>seq2 = SeqRecord(Seq(<span class="py-string">'ATGCGT'</span>, alphabet=generic_dna), id=<span class="py-string">'pro2'</span>)
<span class="py-prompt">&gt;&gt;&gt; </span>pro1 = SeqRecord(Seq(<span class="py-string">'MSR'</span>, alphabet=generic_protein), id=<span class="py-string">'pro1'</span>)
<span class="py-prompt">&gt;&gt;&gt; </span>pro2 = SeqRecord(Seq(<span class="py-string">'M-R'</span>, alphabet=generic_protein), id=<span class="py-string">'pro2'</span>)
<span class="py-prompt">&gt;&gt;&gt; </span>aln = MultipleSeqAlignment([pro1, pro2])
<span class="py-prompt">&gt;&gt;&gt; </span>codon_aln = build(aln, [seq1, seq2])
<span class="py-prompt">&gt;&gt;&gt; </span><span class="py-keyword">print</span>(codon_aln)
<span class="py-output">CodonAlphabet(Standard) CodonAlignment with 2 rows and 9 columns (3 codons)</span>
<span class="py-output">ATGTCTCGT pro1</span>
<span class="py-output">ATG---CGT pro2</span></pre>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_check_corr"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_check_corr</span>(<span class="sig-arg">pro</span>,
        <span class="sig-arg">nucl</span>,
        <span class="sig-arg">gap_char</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">-</code><code class="variable-quote">'</code></span>,
        <span class="sig-arg">codon_table</span>=<span class="sig-default">NCBICodonTable(id=1, names=['Standard', 'SGC0'], ...)</span>,
        <span class="sig-arg">complete_protein</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">anchor_len</span>=<span class="sig-default">10</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.codonalign-pysrc.html#_check_corr">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Check if the nucleotide can be translated into the protein (PRIVATE).</p>
<p>Expects two SeqRecord objects.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_get_shift_anchor_re"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_get_shift_anchor_re</span>(<span class="sig-arg">sh_anc</span>,
        <span class="sig-arg">sh_nuc</span>,
        <span class="sig-arg">shift_val</span>,
        <span class="sig-arg">aa2re</span>,
        <span class="sig-arg">anchor_len</span>,
        <span class="sig-arg">shift_id_pos</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.codonalign-pysrc.html#_get_shift_anchor_re">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Find a regular expression matching a potentially shifted anchor (PRIVATE).</p>
<dl class="rst-docutils">
<dt>Arguments:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>sh_anc    - shifted anchor sequence</li>
<li>sh_nuc    - potentially corresponding nucleotide sequence
of sh_anc</li>
<li>shift_val - 1 or 2 indicates forward frame shift, whereas
3*anchor_len-1 or 3*anchor_len-2 indicates
backward shift</li>
<li>aa2re     - aa to codon re dict</li>
<li>anchor_len - length of the anchor</li>
<li>shift_id_pos - specify current shift name we are at</li>
</ul>
</dd>
</dl>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_get_codon_rec"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_get_codon_rec</span>(<span class="sig-arg">pro</span>,
        <span class="sig-arg">nucl</span>,
        <span class="sig-arg">span_mode</span>,
        <span class="sig-arg">alphabet</span>,
        <span class="sig-arg">gap_char</span>=<span class="sig-default"><code class="variable-quote">'</code><code class="variable-string">-</code><code class="variable-quote">'</code></span>,
        <span class="sig-arg">codon_table</span>=<span class="sig-default">NCBICodonTable(id=1, names=['Standard', 'SGC0'], ...)</span>,
        <span class="sig-arg">complete_protein</span>=<span class="sig-default">False</span>,
        <span class="sig-arg">max_score</span>=<span class="sig-default">10</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.codonalign-pysrc.html#_get_codon_rec">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Generate codon alignment based on regular re match (PRIVATE).</p>
<p>span_mode is a tuple returned by _check_corr. The first element
is the span of a re search, and the second element is the mode
for the match.</p>
<dl class="rst-docutils">
<dt>mode</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>0: direct match</li>
<li>1: mismatch (no indels)</li>
<li>2: frameshift</li>
</ul>
</dd>
</dl>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="_align_shift_recs"></a>
<div class="private">
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">_align_shift_recs</span>(<span class="sig-arg">recs</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="Bio.codonalign-pysrc.html#_align_shift_recs">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Build alignment according to the frameshift detected by _check_corr (PRIVATE).</p>
<dl class="rst-docutils">
<dt>Argument:</dt>
<dd><ul class="rst-first rst-last rst-simple">
<li>recs - a list of SeqRecords containing a CodonSeq dictated
by a rf_table (with frameshift in some of them).</li>
</ul>
</dd>
</dl>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== VARIABLES DETAILS ==================== -->
<a name="section-VariablesDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Variables Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-VariablesDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="__warningregistry__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <h3 class="epydoc">__warningregistry__</h3>
  
  <dl class="fields">
  </dl>
  <dl class="fields">
    <dt>Value:</dt>
      <dd><table><tr><td><pre class="variable">
<code class="variable-group">{</code><code class="variable-group">(</code><code class="variable-quote">'</code><code class="variable-string">Bio.codonalign is an experimental module which may undergo signific</code><span class="variable-linewrap"><img src="crarr.png" alt="\" /></span>
<code class="variable-string">ant changes prior to its future official release.</code><code class="variable-quote">'</code><code class="variable-op">,</code>
  &lt;class 'Bio.BiopythonExperimentalWarning'&gt;<code class="variable-op">,</code>
  27<code class="variable-group">)</code><code class="variable-op">: </code>True<code class="variable-group">}</code>
</pre></td></tr></table>
</dd>
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Nov  6 13:32:24 2019
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
