<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.Blast.NCBIStandalone</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.Blast-module.html">Package&nbsp;Blast</a> ::
        Module&nbsp;NCBIStandalone
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.Blast.NCBIStandalone-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.Blast.NCBIStandalone-module.html">Module Bio.Blast.NCBIStandalone</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">   1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 1999-2000 by Jeffrey Chang.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">   2</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L3"></a><tt class="py-lineno">   3</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L4"></a><tt class="py-lineno">   4</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L5"></a><tt class="py-lineno">   5</tt>  <tt class="py-line"><tt class="py-comment"># Patches by Mike Poidinger to support multiple databases.</tt> </tt>
<a name="L6"></a><tt class="py-lineno">   6</tt>  <tt class="py-line"><tt class="py-comment"># Updated by Peter Cock in 2007 to do a better job on BLAST 2.2.15</tt> </tt>
<a name="L7"></a><tt class="py-lineno">   7</tt>  <tt class="py-line"> </tt>
<a name="L8"></a><tt class="py-lineno">   8</tt>  <tt class="py-line"><tt class="py-docstring">"""Code for calling standalone BLAST and parsing plain text output (DEPRECATED).</tt> </tt>
<a name="L9"></a><tt class="py-lineno">   9</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L10"></a><tt class="py-lineno">  10</tt>  <tt class="py-line"><tt class="py-docstring">Rather than parsing the human readable plain text BLAST output (which seems to</tt> </tt>
<a name="L11"></a><tt class="py-lineno">  11</tt>  <tt class="py-line"><tt class="py-docstring">change with every update to BLAST), we and the NBCI recommend you parse the</tt> </tt>
<a name="L12"></a><tt class="py-lineno">  12</tt>  <tt class="py-line"><tt class="py-docstring">XML output instead. The plain text parser in this module still works at the</tt> </tt>
<a name="L13"></a><tt class="py-lineno">  13</tt>  <tt class="py-line"><tt class="py-docstring">time of writing, but is considered obsolete and updating it to cope with the</tt> </tt>
<a name="L14"></a><tt class="py-lineno">  14</tt>  <tt class="py-line"><tt class="py-docstring">latest versions of BLAST is not a priority for us.</tt> </tt>
<a name="L15"></a><tt class="py-lineno">  15</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L16"></a><tt class="py-lineno">  16</tt>  <tt class="py-line"><tt class="py-docstring">This module also provides code to work with the "legacy" standalone version of</tt> </tt>
<a name="L17"></a><tt class="py-lineno">  17</tt>  <tt class="py-line"><tt class="py-docstring">NCBI BLAST, tools blastall, rpsblast and blastpgp via three helper functions of</tt> </tt>
<a name="L18"></a><tt class="py-lineno">  18</tt>  <tt class="py-line"><tt class="py-docstring">the same name. These functions are very limited for dealing with the output as</tt> </tt>
<a name="L19"></a><tt class="py-lineno">  19</tt>  <tt class="py-line"><tt class="py-docstring">files rather than handles, for which the wrappers in Bio.Blast.Applications are</tt> </tt>
<a name="L20"></a><tt class="py-lineno">  20</tt>  <tt class="py-line"><tt class="py-docstring">preferred. Furthermore, the NCBI themselves regard these command line tools as</tt> </tt>
<a name="L21"></a><tt class="py-lineno">  21</tt>  <tt class="py-line"><tt class="py-docstring">"legacy", and encourage using the new BLAST+ tools instead. Biopython has</tt> </tt>
<a name="L22"></a><tt class="py-lineno">  22</tt>  <tt class="py-line"><tt class="py-docstring">wrappers for these under Bio.Blast.Applications (see the tutorial).</tt> </tt>
<a name="L23"></a><tt class="py-lineno">  23</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L24"></a><tt class="py-lineno">  24</tt>  <tt class="py-line"> </tt>
<a name="L25"></a><tt class="py-lineno">  25</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">__future__</tt> <tt class="py-keyword">import</tt> <tt class="py-name">print_function</tt> </tt>
<a name="L26"></a><tt class="py-lineno">  26</tt>  <tt class="py-line"> </tt>
<a name="L27"></a><tt class="py-lineno">  27</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-1" class="py-name" targets="Class Bio.BiopythonDeprecationWarning=Bio.BiopythonDeprecationWarning-class.html"><a title="Bio.BiopythonDeprecationWarning" class="py-name" href="#" onclick="return doclink('link-1', 'BiopythonDeprecationWarning', 'link-1');">BiopythonDeprecationWarning</a></tt> </tt>
<a name="L28"></a><tt class="py-lineno">  28</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L29"></a><tt class="py-lineno">  29</tt>  <tt class="py-line"><tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"This module has been deprecated. Consider Bio.SearchIO for "</tt> </tt>
<a name="L30"></a><tt class="py-lineno">  30</tt>  <tt class="py-line">              <tt class="py-string">"parsing BLAST output instead."</tt><tt class="py-op">,</tt> <tt id="link-2" class="py-name"><a title="Bio.BiopythonDeprecationWarning" class="py-name" href="#" onclick="return doclink('link-2', 'BiopythonDeprecationWarning', 'link-1');">BiopythonDeprecationWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L31"></a><tt class="py-lineno">  31</tt>  <tt class="py-line"> </tt>
<a name="L32"></a><tt class="py-lineno">  32</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">os</tt> </tt>
<a name="L33"></a><tt class="py-lineno">  33</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">re</tt> </tt>
<a name="L34"></a><tt class="py-lineno">  34</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-3" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-3', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-4" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-4', '_py3k', 'link-4');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">StringIO</tt> </tt>
<a name="L35"></a><tt class="py-lineno">  35</tt>  <tt class="py-line"> </tt>
<a name="L36"></a><tt class="py-lineno">  36</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-5" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-5', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-6" class="py-name" targets="Module Bio.File=Bio.File-module.html"><a title="Bio.File" class="py-name" href="#" onclick="return doclink('link-6', 'File', 'link-6');">File</a></tt> </tt>
<a name="L37"></a><tt class="py-lineno">  37</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-7" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-7', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-8" class="py-name" targets="Module Bio.ParserSupport=Bio.ParserSupport-module.html"><a title="Bio.ParserSupport" class="py-name" href="#" onclick="return doclink('link-8', 'ParserSupport', 'link-8');">ParserSupport</a></tt> <tt class="py-keyword">import</tt> <tt class="py-op">*</tt> </tt>
<a name="L38"></a><tt class="py-lineno">  38</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-9" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-9', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-10" class="py-name" targets="Package Bio.Blast=Bio.Blast-module.html,Class Bio.Blast.Record.Blast=Bio.Blast.Record.Blast-class.html"><a title="Bio.Blast
Bio.Blast.Record.Blast" class="py-name" href="#" onclick="return doclink('link-10', 'Blast', 'link-10');">Blast</a></tt> <tt class="py-keyword">import</tt> <tt id="link-11" class="py-name" targets="Class Bio.Affy.CelFile.Record=Bio.Affy.CelFile.Record-class.html,Module Bio.Blast.Record=Bio.Blast.Record-module.html,Class Bio.Cluster.Record=Bio.Cluster.Record-class.html,Class Bio.Compass.Record=Bio.Compass.Record-class.html,Class Bio.Emboss.Primer3.Record=Bio.Emboss.Primer3.Record-class.html,Class Bio.ExPASy.Enzyme.Record=Bio.ExPASy.Enzyme.Record-class.html,Class Bio.ExPASy.Prodoc.Record=Bio.ExPASy.Prodoc.Record-class.html,Class Bio.ExPASy.Prosite.Record=Bio.ExPASy.Prosite.Record-class.html,Class Bio.ExPASy.ScanProsite.Record=Bio.ExPASy.ScanProsite.Record-class.html,Module Bio.GenBank.Record=Bio.GenBank.Record-module.html,Class Bio.GenBank.Record.Record=Bio.GenBank.Record.Record-class.html,Module Bio.Geo.Record=Bio.Geo.Record-module.html,Class Bio.Geo.Record.Record=Bio.Geo.Record.Record-class.html,Class Bio.KEGG.Compound.Record=Bio.KEGG.Compound.Record-class.html,Class Bio.KEGG.Enzyme.Record=Bio.KEGG.Enzyme.Record-class.html,Class Bio.Medline.Record=Bio.Medline.Record-class.html,Class Bio.Motif.Parsers.AlignAce.Record=Bio.Motif.Parsers.AlignAce.Record-class.html,Class Bio.Motif.Parsers.MAST.Record=Bio.Motif.Parsers.MAST.Record-class.html,Class Bio.PopGen.FDist.Record=Bio.PopGen.FDist.Record-class.html,Class Bio.PopGen.GenePop.LargeFileParser.Record=Bio.PopGen.GenePop.LargeFileParser.Record-class.html,Class Bio.PopGen.GenePop.Record=Bio.PopGen.GenePop.Record-class.html,Class Bio.SCOP.Cla.Record=Bio.SCOP.Cla.Record-class.html,Class Bio.SCOP.Des.Record=Bio.SCOP.Des.Record-class.html,Class Bio.SCOP.Dom.Record=Bio.SCOP.Dom.Record-class.html,Class Bio.SCOP.Hie.Record=Bio.SCOP.Hie.Record-class.html,Class Bio.Sequencing.Phd.Record=Bio.Sequencing.Phd.Record-class.html,Class Bio.SwissProt.KeyWList.Record=Bio.SwissProt.KeyWList.Record-class.html,Class Bio.SwissProt.Record=Bio.SwissProt.Record-class.html,Class Bio.UniGene.Record=Bio.UniGene.Record-class.html,Class Bio.motifs.alignace.Record=Bio.motifs.alignace.Record-class.html,Class Bio.motifs.jaspar.Record=Bio.motifs.jaspar.Record-class.html,Class Bio.motifs.mast.Record=Bio.motifs.mast.Record-class.html,Class Bio.motifs.meme.Record=Bio.motifs.meme.Record-class.html,Class Bio.motifs.transfac.Record=Bio.motifs.transfac.Record-class.html"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-11', 'Record', 'link-11');">Record</a></tt> </tt>
<a name="L39"></a><tt class="py-lineno">  39</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-12" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-12', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-13" class="py-name" targets="Package Bio.Application=Bio.Application-module.html"><a title="Bio.Application" class="py-name" href="#" onclick="return doclink('link-13', 'Application', 'link-13');">Application</a></tt> <tt class="py-keyword">import</tt> <tt id="link-14" class="py-name" targets="Function Bio.Application._escape_filename()=Bio.Application-module.html#_escape_filename"><a title="Bio.Application._escape_filename" class="py-name" href="#" onclick="return doclink('link-14', '_escape_filename', 'link-14');">_escape_filename</a></tt> </tt>
<a name="L40"></a><tt class="py-lineno">  40</tt>  <tt class="py-line"> </tt>
<a name="L41"></a><tt class="py-lineno">  41</tt>  <tt class="py-line"><tt class="py-name">__docformat__</tt> <tt class="py-op">=</tt> <tt class="py-string">"restructuredtext en"</tt> </tt>
<a name="L42"></a><tt class="py-lineno">  42</tt>  <tt class="py-line"> </tt>
<a name="L43"></a><tt class="py-lineno">  43</tt>  <tt class="py-line"><tt id="link-15" class="py-name" targets="Variable Bio.Blast.NCBIStandalone._score_e_re=Bio.Blast.NCBIStandalone-module.html#_score_e_re"><a title="Bio.Blast.NCBIStandalone._score_e_re" class="py-name" href="#" onclick="return doclink('link-15', '_score_e_re', 'link-15');">_score_e_re</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'Score +E'</tt><tt class="py-op">)</tt> </tt>
<a name="L44"></a><tt class="py-lineno">  44</tt>  <tt class="py-line"> </tt>
<a name="L45"></a><tt class="py-lineno">  45</tt>  <tt class="py-line"> </tt>
<a name="LowQualityBlastError"></a><div id="LowQualityBlastError-def"><a name="L46"></a><tt class="py-lineno">  46</tt> <a class="py-toggle" href="#" id="LowQualityBlastError-toggle" onclick="return toggle('LowQualityBlastError');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone.LowQualityBlastError-class.html">LowQualityBlastError</a><tt class="py-op">(</tt><tt class="py-base-class">Exception</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="LowQualityBlastError-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="LowQualityBlastError-expanded"><a name="L47"></a><tt class="py-lineno">  47</tt>  <tt class="py-line">    <tt class="py-docstring">"""Error caused by running a low quality sequence through BLAST.</tt> </tt>
<a name="L48"></a><tt class="py-lineno">  48</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L49"></a><tt class="py-lineno">  49</tt>  <tt class="py-line"><tt class="py-docstring">    When low quality sequences (like GenBank entries containing only</tt> </tt>
<a name="L50"></a><tt class="py-lineno">  50</tt>  <tt class="py-line"><tt class="py-docstring">    stretches of a single nucleotide) are BLASTed, they will result in</tt> </tt>
<a name="L51"></a><tt class="py-lineno">  51</tt>  <tt class="py-line"><tt class="py-docstring">    BLAST generating an error and not being able to perform the BLAST.</tt> </tt>
<a name="L52"></a><tt class="py-lineno">  52</tt>  <tt class="py-line"><tt class="py-docstring">    search. This error should be raised for the BLAST reports produced</tt> </tt>
<a name="L53"></a><tt class="py-lineno">  53</tt>  <tt class="py-line"><tt class="py-docstring">    in this case.</tt> </tt>
<a name="L54"></a><tt class="py-lineno">  54</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L55"></a><tt class="py-lineno">  55</tt>  <tt class="py-line">    <tt class="py-keyword">pass</tt> </tt>
</div><a name="L56"></a><tt class="py-lineno">  56</tt>  <tt class="py-line"> </tt>
<a name="L57"></a><tt class="py-lineno">  57</tt>  <tt class="py-line"> </tt>
<a name="ShortQueryBlastError"></a><div id="ShortQueryBlastError-def"><a name="L58"></a><tt class="py-lineno">  58</tt> <a class="py-toggle" href="#" id="ShortQueryBlastError-toggle" onclick="return toggle('ShortQueryBlastError');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone.ShortQueryBlastError-class.html">ShortQueryBlastError</a><tt class="py-op">(</tt><tt class="py-base-class">Exception</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="ShortQueryBlastError-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="ShortQueryBlastError-expanded"><a name="L59"></a><tt class="py-lineno">  59</tt>  <tt class="py-line">    <tt class="py-docstring">"""Error caused by running a short query sequence through BLAST.</tt> </tt>
<a name="L60"></a><tt class="py-lineno">  60</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L61"></a><tt class="py-lineno">  61</tt>  <tt class="py-line"><tt class="py-docstring">    If the query sequence is too short, BLAST outputs warnings and errors::</tt> </tt>
<a name="L62"></a><tt class="py-lineno">  62</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L63"></a><tt class="py-lineno">  63</tt>  <tt class="py-line"><tt class="py-docstring">        Searching[blastall] WARNING:  [000.000]  AT1G08320: SetUpBlastSearch failed.</tt> </tt>
<a name="L64"></a><tt class="py-lineno">  64</tt>  <tt class="py-line"><tt class="py-docstring">        [blastall] ERROR:  [000.000]  AT1G08320: Blast:</tt> </tt>
<a name="L65"></a><tt class="py-lineno">  65</tt>  <tt class="py-line"><tt class="py-docstring">        [blastall] ERROR:  [000.000]  AT1G08320: Blast: Query must be at least wordsize</tt> </tt>
<a name="L66"></a><tt class="py-lineno">  66</tt>  <tt class="py-line"><tt class="py-docstring">        done</tt> </tt>
<a name="L67"></a><tt class="py-lineno">  67</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L68"></a><tt class="py-lineno">  68</tt>  <tt class="py-line"><tt class="py-docstring">    This exception is raised when that condition is detected.</tt> </tt>
<a name="L69"></a><tt class="py-lineno">  69</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L70"></a><tt class="py-lineno">  70</tt>  <tt class="py-line">    <tt class="py-keyword">pass</tt> </tt>
</div><a name="L71"></a><tt class="py-lineno">  71</tt>  <tt class="py-line"> </tt>
<a name="L72"></a><tt class="py-lineno">  72</tt>  <tt class="py-line"> </tt>
<a name="_Scanner"></a><div id="_Scanner-def"><a name="L73"></a><tt class="py-lineno">  73</tt> <a class="py-toggle" href="#" id="_Scanner-toggle" onclick="return toggle('_Scanner');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._Scanner-class.html">_Scanner</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_Scanner-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_Scanner-expanded"><a name="L74"></a><tt class="py-lineno">  74</tt>  <tt class="py-line">    <tt class="py-docstring">"""Scan BLAST output from blastall or blastpgp.</tt> </tt>
<a name="L75"></a><tt class="py-lineno">  75</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L76"></a><tt class="py-lineno">  76</tt>  <tt class="py-line"><tt class="py-docstring">    Tested with blastall and blastpgp v2.0.10, v2.0.11</tt> </tt>
<a name="L77"></a><tt class="py-lineno">  77</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L78"></a><tt class="py-lineno">  78</tt>  <tt class="py-line"><tt class="py-docstring">    Methods:</tt> </tt>
<a name="L79"></a><tt class="py-lineno">  79</tt>  <tt class="py-line"><tt class="py-docstring">     - feed     Feed data into the scanner.</tt> </tt>
<a name="L80"></a><tt class="py-lineno">  80</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="_Scanner.feed"></a><div id="_Scanner.feed-def"><a name="L81"></a><tt class="py-lineno">  81</tt> <a class="py-toggle" href="#" id="_Scanner.feed-toggle" onclick="return toggle('_Scanner.feed');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._Scanner-class.html#feed">feed</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_Scanner.feed-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_Scanner.feed-expanded"><a name="L82"></a><tt class="py-lineno">  82</tt>  <tt class="py-line">        <tt class="py-docstring">"""S.feed(handle, consumer)</tt> </tt>
<a name="L83"></a><tt class="py-lineno">  83</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L84"></a><tt class="py-lineno">  84</tt>  <tt class="py-line"><tt class="py-docstring">        Feed in a BLAST report for scanning.  handle is a file-like</tt> </tt>
<a name="L85"></a><tt class="py-lineno">  85</tt>  <tt class="py-line"><tt class="py-docstring">        object that contains the BLAST report.  consumer is a Consumer</tt> </tt>
<a name="L86"></a><tt class="py-lineno">  86</tt>  <tt class="py-line"><tt class="py-docstring">        object that will receive events as the report is scanned.</tt> </tt>
<a name="L87"></a><tt class="py-lineno">  87</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L88"></a><tt class="py-lineno">  88</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt id="link-16" class="py-name"><a title="Bio.File" class="py-name" href="#" onclick="return doclink('link-16', 'File', 'link-6');">File</a></tt><tt class="py-op">.</tt><tt id="link-17" class="py-name" targets="Class Bio.File.UndoHandle=Bio.File.UndoHandle-class.html"><a title="Bio.File.UndoHandle" class="py-name" href="#" onclick="return doclink('link-17', 'UndoHandle', 'link-17');">UndoHandle</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L89"></a><tt class="py-lineno">  89</tt>  <tt class="py-line">            <tt class="py-name">uhandle</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt> </tt>
<a name="L90"></a><tt class="py-lineno">  90</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L91"></a><tt class="py-lineno">  91</tt>  <tt class="py-line">            <tt class="py-name">uhandle</tt> <tt class="py-op">=</tt> <tt id="link-18" class="py-name"><a title="Bio.File" class="py-name" href="#" onclick="return doclink('link-18', 'File', 'link-6');">File</a></tt><tt class="py-op">.</tt><tt id="link-19" class="py-name"><a title="Bio.File.UndoHandle" class="py-name" href="#" onclick="return doclink('link-19', 'UndoHandle', 'link-17');">UndoHandle</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L92"></a><tt class="py-lineno">  92</tt>  <tt class="py-line"> </tt>
<a name="L93"></a><tt class="py-lineno">  93</tt>  <tt class="py-line">        <tt class="py-comment"># Try to fast-forward to the beginning of the blast report.</tt> </tt>
<a name="L94"></a><tt class="py-lineno">  94</tt>  <tt class="py-line">        <tt id="link-20" class="py-name" targets="Function Bio.ParserSupport.read_and_call_until()=Bio.ParserSupport-module.html#read_and_call_until"><a title="Bio.ParserSupport.read_and_call_until" class="py-name" href="#" onclick="return doclink('link-20', 'read_and_call_until', 'link-20');">read_and_call_until</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-21" class="py-name" targets="Method Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent()=Bio.Blast.NCBIStandalone._BlastErrorConsumer-class.html#noevent"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-21', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt id="link-22" class="py-name" targets="Method Bio.Alphabet.Alphabet.contains()=Bio.Alphabet.Alphabet-class.html#contains,Method Bio.Alphabet.AlphabetEncoder.contains()=Bio.Alphabet.AlphabetEncoder-class.html#contains,Method Bio.Alphabet.Gapped.contains()=Bio.Alphabet.Gapped-class.html#contains,Method Bio.Alphabet.HasStopCodon.contains()=Bio.Alphabet.HasStopCodon-class.html#contains,Method Bio.Phylo.Consensus._BitString.contains()=Bio.Phylo.Consensus._BitString-class.html#contains"><a title="Bio.Alphabet.Alphabet.contains
Bio.Alphabet.AlphabetEncoder.contains
Bio.Alphabet.Gapped.contains
Bio.Alphabet.HasStopCodon.contains
Bio.Phylo.Consensus._BitString.contains" class="py-name" href="#" onclick="return doclink('link-22', 'contains', 'link-22');">contains</a></tt><tt class="py-op">=</tt><tt class="py-string">'BLAST'</tt><tt class="py-op">)</tt> </tt>
<a name="L95"></a><tt class="py-lineno">  95</tt>  <tt class="py-line">        <tt class="py-comment"># Now scan the BLAST report.</tt> </tt>
<a name="L96"></a><tt class="py-lineno">  96</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-23" class="py-name" targets="Method Bio.Blast.NCBIStandalone._Scanner._scan_header()=Bio.Blast.NCBIStandalone._Scanner-class.html#_scan_header"><a title="Bio.Blast.NCBIStandalone._Scanner._scan_header" class="py-name" href="#" onclick="return doclink('link-23', '_scan_header', 'link-23');">_scan_header</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">)</tt> </tt>
<a name="L97"></a><tt class="py-lineno">  97</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-24" class="py-name" targets="Method Bio.Blast.NCBIStandalone._Scanner._scan_rounds()=Bio.Blast.NCBIStandalone._Scanner-class.html#_scan_rounds"><a title="Bio.Blast.NCBIStandalone._Scanner._scan_rounds" class="py-name" href="#" onclick="return doclink('link-24', '_scan_rounds', 'link-24');">_scan_rounds</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">)</tt> </tt>
<a name="L98"></a><tt class="py-lineno">  98</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-25" class="py-name" targets="Method Bio.Blast.NCBIStandalone._Scanner._scan_database_report()=Bio.Blast.NCBIStandalone._Scanner-class.html#_scan_database_report"><a title="Bio.Blast.NCBIStandalone._Scanner._scan_database_report" class="py-name" href="#" onclick="return doclink('link-25', '_scan_database_report', 'link-25');">_scan_database_report</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">)</tt> </tt>
<a name="L99"></a><tt class="py-lineno">  99</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-26" class="py-name" targets="Method Bio.Blast.NCBIStandalone._Scanner._scan_parameters()=Bio.Blast.NCBIStandalone._Scanner-class.html#_scan_parameters"><a title="Bio.Blast.NCBIStandalone._Scanner._scan_parameters" class="py-name" href="#" onclick="return doclink('link-26', '_scan_parameters', 'link-26');">_scan_parameters</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">)</tt> </tt>
</div><a name="L100"></a><tt class="py-lineno"> 100</tt>  <tt class="py-line"> </tt>
<a name="_Scanner._scan_header"></a><div id="_Scanner._scan_header-def"><a name="L101"></a><tt class="py-lineno"> 101</tt> <a class="py-toggle" href="#" id="_Scanner._scan_header-toggle" onclick="return toggle('_Scanner._scan_header');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._Scanner-class.html#_scan_header">_scan_header</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">uhandle</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_Scanner._scan_header-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_Scanner._scan_header-expanded"><a name="L102"></a><tt class="py-lineno"> 102</tt>  <tt class="py-line">        <tt class="py-comment"># BLASTP 2.0.10 [Aug-26-1999]</tt> </tt>
<a name="L103"></a><tt class="py-lineno"> 103</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L104"></a><tt class="py-lineno"> 104</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L105"></a><tt class="py-lineno"> 105</tt>  <tt class="py-line">        <tt class="py-comment"># Reference: Altschul, Stephen F., Thomas L. Madden, Alejandro A. Schaf</tt> </tt>
<a name="L106"></a><tt class="py-lineno"> 106</tt>  <tt class="py-line">        <tt class="py-comment"># Jinghui Zhang, Zheng Zhang, Webb Miller, and David J. Lipman (1997),</tt> </tt>
<a name="L107"></a><tt class="py-lineno"> 107</tt>  <tt class="py-line">        <tt class="py-comment"># "Gapped BLAST and PSI-BLAST: a new generation of protein database sea</tt> </tt>
<a name="L108"></a><tt class="py-lineno"> 108</tt>  <tt class="py-line">        <tt class="py-comment"># programs",  Nucleic Acids Res. 25:3389-3402.</tt> </tt>
<a name="L109"></a><tt class="py-lineno"> 109</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L110"></a><tt class="py-lineno"> 110</tt>  <tt class="py-line">        <tt class="py-comment"># Query= test</tt> </tt>
<a name="L111"></a><tt class="py-lineno"> 111</tt>  <tt class="py-line">        <tt class="py-comment">#          (140 letters)</tt> </tt>
<a name="L112"></a><tt class="py-lineno"> 112</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L113"></a><tt class="py-lineno"> 113</tt>  <tt class="py-line">        <tt class="py-comment"># Database: sdqib40-1.35.seg.fa</tt> </tt>
<a name="L114"></a><tt class="py-lineno"> 114</tt>  <tt class="py-line">        <tt class="py-comment">#            1323 sequences; 223,339 total letters</tt> </tt>
<a name="L115"></a><tt class="py-lineno"> 115</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L116"></a><tt class="py-lineno"> 116</tt>  <tt class="py-line">        <tt class="py-comment"># ========================================================</tt> </tt>
<a name="L117"></a><tt class="py-lineno"> 117</tt>  <tt class="py-line">        <tt class="py-comment"># This next example is from the online version of Blast,</tt> </tt>
<a name="L118"></a><tt class="py-lineno"> 118</tt>  <tt class="py-line">        <tt class="py-comment"># note there are TWO references, an RID line, and also</tt> </tt>
<a name="L119"></a><tt class="py-lineno"> 119</tt>  <tt class="py-line">        <tt class="py-comment"># the database is BEFORE the query line.</tt> </tt>
<a name="L120"></a><tt class="py-lineno"> 120</tt>  <tt class="py-line">        <tt class="py-comment"># Note there possibleuse of non-ASCII in the author names.</tt> </tt>
<a name="L121"></a><tt class="py-lineno"> 121</tt>  <tt class="py-line">        <tt class="py-comment"># ========================================================</tt> </tt>
<a name="L122"></a><tt class="py-lineno"> 122</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L123"></a><tt class="py-lineno"> 123</tt>  <tt class="py-line">        <tt class="py-comment"># BLASTP 2.2.15 [Oct-15-2006]</tt> </tt>
<a name="L124"></a><tt class="py-lineno"> 124</tt>  <tt class="py-line">        <tt class="py-comment"># Reference: Altschul, Stephen F., Thomas L. Madden, Alejandro A. Sch??ffer,</tt> </tt>
<a name="L125"></a><tt class="py-lineno"> 125</tt>  <tt class="py-line">        <tt class="py-comment"># Jinghui Zhang, Zheng Zhang, Webb Miller, and David J. Lipman</tt> </tt>
<a name="L126"></a><tt class="py-lineno"> 126</tt>  <tt class="py-line">        <tt class="py-comment"># (1997), "Gapped BLAST and PSI-BLAST: a new generation of</tt> </tt>
<a name="L127"></a><tt class="py-lineno"> 127</tt>  <tt class="py-line">        <tt class="py-comment"># protein database search programs", Nucleic Acids Res. 25:3389-3402.</tt> </tt>
<a name="L128"></a><tt class="py-lineno"> 128</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L129"></a><tt class="py-lineno"> 129</tt>  <tt class="py-line">        <tt class="py-comment"># Reference: Sch??ffer, Alejandro A., L. Aravind, Thomas L. Madden, Sergei</tt> </tt>
<a name="L130"></a><tt class="py-lineno"> 130</tt>  <tt class="py-line">        <tt class="py-comment"># Shavirin, John L. Spouge, Yuri I. Wolf, Eugene V. Koonin, and</tt> </tt>
<a name="L131"></a><tt class="py-lineno"> 131</tt>  <tt class="py-line">        <tt class="py-comment"># Stephen F. Altschul (2001), "Improving the accuracy of PSI-BLAST</tt> </tt>
<a name="L132"></a><tt class="py-lineno"> 132</tt>  <tt class="py-line">        <tt class="py-comment"># protein database searches with composition-based statistics</tt> </tt>
<a name="L133"></a><tt class="py-lineno"> 133</tt>  <tt class="py-line">        <tt class="py-comment"># and other refinements", Nucleic Acids Res. 29:2994-3005.</tt> </tt>
<a name="L134"></a><tt class="py-lineno"> 134</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L135"></a><tt class="py-lineno"> 135</tt>  <tt class="py-line">        <tt class="py-comment"># RID: 1166022616-19998-65316425856.BLASTQ1</tt> </tt>
<a name="L136"></a><tt class="py-lineno"> 136</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L137"></a><tt class="py-lineno"> 137</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L138"></a><tt class="py-lineno"> 138</tt>  <tt class="py-line">        <tt class="py-comment"># Database: All non-redundant GenBank CDS</tt> </tt>
<a name="L139"></a><tt class="py-lineno"> 139</tt>  <tt class="py-line">        <tt class="py-comment"># translations+PDB+SwissProt+PIR+PRF excluding environmental samples</tt> </tt>
<a name="L140"></a><tt class="py-lineno"> 140</tt>  <tt class="py-line">        <tt class="py-comment">#            4,254,166 sequences; 1,462,033,012 total letters</tt> </tt>
<a name="L141"></a><tt class="py-lineno"> 141</tt>  <tt class="py-line">        <tt class="py-comment"># Query=  gi:16127998</tt> </tt>
<a name="L142"></a><tt class="py-lineno"> 142</tt>  <tt class="py-line">        <tt class="py-comment"># Length=428</tt> </tt>
<a name="L143"></a><tt class="py-lineno"> 143</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L144"></a><tt class="py-lineno"> 144</tt>  <tt class="py-line"> </tt>
<a name="L145"></a><tt class="py-lineno"> 145</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-27" class="py-name" targets="Method Bio.Blast.NCBIStandalone._BlastConsumer.start_header()=Bio.Blast.NCBIStandalone._BlastConsumer-class.html#start_header,Method Bio.Blast.NCBIStandalone._HeaderConsumer.start_header()=Bio.Blast.NCBIStandalone._HeaderConsumer-class.html#start_header,Method Bio.Blast.NCBIStandalone._PSIBlastConsumer.start_header()=Bio.Blast.NCBIStandalone._PSIBlastConsumer-class.html#start_header"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.start_header
Bio.Blast.NCBIStandalone._HeaderConsumer.start_header
Bio.Blast.NCBIStandalone._PSIBlastConsumer.start_header" class="py-name" href="#" onclick="return doclink('link-27', 'start_header', 'link-27');">start_header</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L146"></a><tt class="py-lineno"> 146</tt>  <tt class="py-line"> </tt>
<a name="L147"></a><tt class="py-lineno"> 147</tt>  <tt class="py-line">        <tt id="link-28" class="py-name" targets="Function Bio.ParserSupport.read_and_call()=Bio.ParserSupport-module.html#read_and_call"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-28', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-29" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HeaderConsumer.version()=Bio.Blast.NCBIStandalone._HeaderConsumer-class.html#version,Function Bio.Cluster.cluster.version()=Bio.Cluster.cluster-module.html#version,Method Bio.GenBank._FeatureConsumer.version()=Bio.GenBank._FeatureConsumer-class.html#version,Method Bio.GenBank._RecordConsumer.version()=Bio.GenBank._RecordConsumer-class.html#version,Variable Bio.motifs.jaspar.Motif.version=Bio.motifs.jaspar.Motif-class.html#version"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-29', 'version', 'link-29');">version</a></tt><tt class="py-op">,</tt> <tt id="link-30" class="py-name"><a title="Bio.Alphabet.Alphabet.contains
Bio.Alphabet.AlphabetEncoder.contains
Bio.Alphabet.Gapped.contains
Bio.Alphabet.HasStopCodon.contains
Bio.Phylo.Consensus._BitString.contains" class="py-name" href="#" onclick="return doclink('link-30', 'contains', 'link-22');">contains</a></tt><tt class="py-op">=</tt><tt class="py-string">'BLAST'</tt><tt class="py-op">)</tt> </tt>
<a name="L148"></a><tt class="py-lineno"> 148</tt>  <tt class="py-line">        <tt id="link-31" class="py-name" targets="Function Bio.ParserSupport.read_and_call_while()=Bio.ParserSupport-module.html#read_and_call_while"><a title="Bio.ParserSupport.read_and_call_while" class="py-name" href="#" onclick="return doclink('link-31', 'read_and_call_while', 'link-31');">read_and_call_while</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-32" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-32', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L149"></a><tt class="py-lineno"> 149</tt>  <tt class="py-line"> </tt>
<a name="L150"></a><tt class="py-lineno"> 150</tt>  <tt class="py-line">        <tt class="py-comment"># There might be a &lt;pre&gt; line, for qblast output.</tt> </tt>
<a name="L151"></a><tt class="py-lineno"> 151</tt>  <tt class="py-line">        <tt id="link-33" class="py-name" targets="Function Bio.ParserSupport.attempt_read_and_call()=Bio.ParserSupport-module.html#attempt_read_and_call"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-33', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-34" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-34', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt id="link-35" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.start=Bio.SeqFeature.CompoundLocation-class.html#start,Variable Bio.SeqFeature.FeatureLocation.start=Bio.SeqFeature.FeatureLocation-class.html#start"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-35', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">"&lt;pre&gt;"</tt><tt class="py-op">)</tt> </tt>
<a name="L152"></a><tt class="py-lineno"> 152</tt>  <tt class="py-line"> </tt>
<a name="L153"></a><tt class="py-lineno"> 153</tt>  <tt class="py-line">        <tt class="py-comment"># Read the reference(s)</tt> </tt>
<a name="L154"></a><tt class="py-lineno"> 154</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt id="link-36" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-36', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> </tt>
<a name="L155"></a><tt class="py-lineno"> 155</tt>  <tt class="py-line">                                <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-37" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HeaderConsumer.reference()=Bio.Blast.NCBIStandalone._HeaderConsumer-class.html#reference,Method Bio.Phylo.PhyloXMLIO.Parser.reference()=Bio.Phylo.PhyloXMLIO.Parser-class.html#reference,Method Bio.Phylo.PhyloXMLIO.Writer.reference()=Bio.Phylo.PhyloXMLIO.Writer-class.html#reference"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-37', 'reference', 'link-37');">reference</a></tt><tt class="py-op">,</tt> <tt id="link-38" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-38', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'Reference'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L156"></a><tt class="py-lineno"> 156</tt>  <tt class="py-line">            <tt class="py-comment"># References are normally multiline terminated by a blank line</tt> </tt>
<a name="L157"></a><tt class="py-lineno"> 157</tt>  <tt class="py-line">            <tt class="py-comment"># (or, based on the old code, the RID line)</tt> </tt>
<a name="L158"></a><tt class="py-lineno"> 158</tt>  <tt class="py-line">            <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L159"></a><tt class="py-lineno"> 159</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">uhandle</tt><tt class="py-op">.</tt><tt id="link-39" class="py-name" targets="Method Bio.File.UndoHandle.readline()=Bio.File.UndoHandle-class.html#readline,Method Bio.bgzf.BgzfReader.readline()=Bio.bgzf.BgzfReader-class.html#readline"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-39', 'readline', 'link-39');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L160"></a><tt class="py-lineno"> 160</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-40" class="py-name" targets="Function Bio.ParserSupport.is_blank_line()=Bio.ParserSupport-module.html#is_blank_line"><a title="Bio.ParserSupport.is_blank_line" class="py-name" href="#" onclick="return doclink('link-40', 'is_blank_line', 'link-40');">is_blank_line</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L161"></a><tt class="py-lineno"> 161</tt>  <tt class="py-line">                    <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-41" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-41', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L162"></a><tt class="py-lineno"> 162</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L163"></a><tt class="py-lineno"> 163</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-42" class="py-name" targets="Method Bio.Seq.Seq.startswith()=Bio.Seq.Seq-class.html#startswith"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-42', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"RID"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L164"></a><tt class="py-lineno"> 164</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L165"></a><tt class="py-lineno"> 165</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L166"></a><tt class="py-lineno"> 166</tt>  <tt class="py-line">                    <tt class="py-comment"># More of the reference</tt> </tt>
<a name="L167"></a><tt class="py-lineno"> 167</tt>  <tt class="py-line">                    <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-43" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-43', 'reference', 'link-37');">reference</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L168"></a><tt class="py-lineno"> 168</tt>  <tt class="py-line"> </tt>
<a name="L169"></a><tt class="py-lineno"> 169</tt>  <tt class="py-line">        <tt class="py-comment"># Deal with the optional RID: ...</tt> </tt>
<a name="L170"></a><tt class="py-lineno"> 170</tt>  <tt class="py-line">        <tt id="link-44" class="py-name"><a title="Bio.ParserSupport.read_and_call_while" class="py-name" href="#" onclick="return doclink('link-44', 'read_and_call_while', 'link-31');">read_and_call_while</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-45" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-45', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L171"></a><tt class="py-lineno"> 171</tt>  <tt class="py-line">        <tt id="link-46" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-46', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-47" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-47', 'reference', 'link-37');">reference</a></tt><tt class="py-op">,</tt> <tt id="link-48" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-48', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">"RID:"</tt><tt class="py-op">)</tt> </tt>
<a name="L172"></a><tt class="py-lineno"> 172</tt>  <tt class="py-line">        <tt id="link-49" class="py-name"><a title="Bio.ParserSupport.read_and_call_while" class="py-name" href="#" onclick="return doclink('link-49', 'read_and_call_while', 'link-31');">read_and_call_while</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-50" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-50', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L173"></a><tt class="py-lineno"> 173</tt>  <tt class="py-line"> </tt>
<a name="L174"></a><tt class="py-lineno"> 174</tt>  <tt class="py-line">        <tt class="py-comment"># blastpgp may have a reference for compositional score matrix</tt> </tt>
<a name="L175"></a><tt class="py-lineno"> 175</tt>  <tt class="py-line">        <tt class="py-comment"># adjustment (see Bug 2502):</tt> </tt>
<a name="L176"></a><tt class="py-lineno"> 176</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-51" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-51', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt> </tt>
<a name="L177"></a><tt class="py-lineno"> 177</tt>  <tt class="py-line">           <tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-52" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-52', 'reference', 'link-37');">reference</a></tt><tt class="py-op">,</tt> <tt id="link-53" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-53', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">"Reference"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L178"></a><tt class="py-lineno"> 178</tt>  <tt class="py-line">            <tt id="link-54" class="py-name"><a title="Bio.ParserSupport.read_and_call_until" class="py-name" href="#" onclick="return doclink('link-54', 'read_and_call_until', 'link-20');">read_and_call_until</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-55" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-55', 'reference', 'link-37');">reference</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L179"></a><tt class="py-lineno"> 179</tt>  <tt class="py-line">            <tt id="link-56" class="py-name"><a title="Bio.ParserSupport.read_and_call_while" class="py-name" href="#" onclick="return doclink('link-56', 'read_and_call_while', 'link-31');">read_and_call_while</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-57" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-57', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L180"></a><tt class="py-lineno"> 180</tt>  <tt class="py-line"> </tt>
<a name="L181"></a><tt class="py-lineno"> 181</tt>  <tt class="py-line">        <tt class="py-comment"># blastpgp has a Reference for composition-based statistics.</tt> </tt>
<a name="L182"></a><tt class="py-lineno"> 182</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-58" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-58', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt> </tt>
<a name="L183"></a><tt class="py-lineno"> 183</tt>  <tt class="py-line">           <tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-59" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-59', 'reference', 'link-37');">reference</a></tt><tt class="py-op">,</tt> <tt id="link-60" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-60', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">"Reference"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L184"></a><tt class="py-lineno"> 184</tt>  <tt class="py-line">            <tt id="link-61" class="py-name"><a title="Bio.ParserSupport.read_and_call_until" class="py-name" href="#" onclick="return doclink('link-61', 'read_and_call_until', 'link-20');">read_and_call_until</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-62" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-62', 'reference', 'link-37');">reference</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L185"></a><tt class="py-lineno"> 185</tt>  <tt class="py-line">            <tt id="link-63" class="py-name"><a title="Bio.ParserSupport.read_and_call_while" class="py-name" href="#" onclick="return doclink('link-63', 'read_and_call_while', 'link-31');">read_and_call_while</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-64" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-64', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L186"></a><tt class="py-lineno"> 186</tt>  <tt class="py-line"> </tt>
<a name="L187"></a><tt class="py-lineno"> 187</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">uhandle</tt><tt class="py-op">.</tt><tt id="link-65" class="py-name" targets="Method Bio.File.UndoHandle.peekline()=Bio.File.UndoHandle-class.html#peekline"><a title="Bio.File.UndoHandle.peekline" class="py-name" href="#" onclick="return doclink('link-65', 'peekline', 'link-65');">peekline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L188"></a><tt class="py-lineno"> 188</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-66" class="py-name" targets="Method Bio.Seq.Seq.strip()=Bio.Seq.Seq-class.html#strip"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-66', 'strip', 'link-66');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-string">""</tt> </tt>
<a name="L189"></a><tt class="py-lineno"> 189</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-67" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-67', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"RID:"</tt><tt class="py-op">)</tt> </tt>
<a name="L190"></a><tt class="py-lineno"> 190</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-68" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-68', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"Query="</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L191"></a><tt class="py-lineno"> 191</tt>  <tt class="py-line">            <tt class="py-comment"># This is an old style query then database...</tt> </tt>
<a name="L192"></a><tt class="py-lineno"> 192</tt>  <tt class="py-line"> </tt>
<a name="L193"></a><tt class="py-lineno"> 193</tt>  <tt class="py-line">            <tt class="py-comment"># Read the Query lines and the following blank line.</tt> </tt>
<a name="L194"></a><tt class="py-lineno"> 194</tt>  <tt class="py-line">            <tt id="link-69" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-69', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-70" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HeaderConsumer.query_info()=Bio.Blast.NCBIStandalone._HeaderConsumer-class.html#query_info"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.query_info" class="py-name" href="#" onclick="return doclink('link-70', 'query_info', 'link-70');">query_info</a></tt><tt class="py-op">,</tt> <tt id="link-71" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-71', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'Query='</tt><tt class="py-op">)</tt> </tt>
<a name="L195"></a><tt class="py-lineno"> 195</tt>  <tt class="py-line">            <tt id="link-72" class="py-name"><a title="Bio.ParserSupport.read_and_call_until" class="py-name" href="#" onclick="return doclink('link-72', 'read_and_call_until', 'link-20');">read_and_call_until</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-73" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.query_info" class="py-name" href="#" onclick="return doclink('link-73', 'query_info', 'link-70');">query_info</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L196"></a><tt class="py-lineno"> 196</tt>  <tt class="py-line">            <tt id="link-74" class="py-name"><a title="Bio.ParserSupport.read_and_call_while" class="py-name" href="#" onclick="return doclink('link-74', 'read_and_call_while', 'link-31');">read_and_call_while</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-75" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-75', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L197"></a><tt class="py-lineno"> 197</tt>  <tt class="py-line"> </tt>
<a name="L198"></a><tt class="py-lineno"> 198</tt>  <tt class="py-line">            <tt class="py-comment"># Read the database lines and the following blank line.</tt> </tt>
<a name="L199"></a><tt class="py-lineno"> 199</tt>  <tt class="py-line">            <tt id="link-76" class="py-name"><a title="Bio.ParserSupport.read_and_call_until" class="py-name" href="#" onclick="return doclink('link-76', 'read_and_call_until', 'link-20');">read_and_call_until</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-77" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HeaderConsumer.database_info()=Bio.Blast.NCBIStandalone._HeaderConsumer-class.html#database_info"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.database_info" class="py-name" href="#" onclick="return doclink('link-77', 'database_info', 'link-77');">database_info</a></tt><tt class="py-op">,</tt> <tt id="link-78" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.end=Bio.SeqFeature.CompoundLocation-class.html#end,Variable Bio.SeqFeature.FeatureLocation.end=Bio.SeqFeature.FeatureLocation-class.html#end"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-78', 'end', 'link-78');">end</a></tt><tt class="py-op">=</tt><tt class="py-string">'total letters'</tt><tt class="py-op">)</tt> </tt>
<a name="L200"></a><tt class="py-lineno"> 200</tt>  <tt class="py-line">            <tt id="link-79" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-79', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-80" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.database_info" class="py-name" href="#" onclick="return doclink('link-80', 'database_info', 'link-77');">database_info</a></tt><tt class="py-op">,</tt> <tt id="link-81" class="py-name"><a title="Bio.Alphabet.Alphabet.contains
Bio.Alphabet.AlphabetEncoder.contains
Bio.Alphabet.Gapped.contains
Bio.Alphabet.HasStopCodon.contains
Bio.Phylo.Consensus._BitString.contains" class="py-name" href="#" onclick="return doclink('link-81', 'contains', 'link-22');">contains</a></tt><tt class="py-op">=</tt><tt class="py-string">'sequences'</tt><tt class="py-op">)</tt> </tt>
<a name="L201"></a><tt class="py-lineno"> 201</tt>  <tt class="py-line">            <tt id="link-82" class="py-name"><a title="Bio.ParserSupport.read_and_call_while" class="py-name" href="#" onclick="return doclink('link-82', 'read_and_call_while', 'link-31');">read_and_call_while</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-83" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-83', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L202"></a><tt class="py-lineno"> 202</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-84" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-84', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"Database:"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L203"></a><tt class="py-lineno"> 203</tt>  <tt class="py-line">            <tt class="py-comment"># This is a new style database then query...</tt> </tt>
<a name="L204"></a><tt class="py-lineno"> 204</tt>  <tt class="py-line">            <tt id="link-85" class="py-name"><a title="Bio.ParserSupport.read_and_call_until" class="py-name" href="#" onclick="return doclink('link-85', 'read_and_call_until', 'link-20');">read_and_call_until</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-86" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.database_info" class="py-name" href="#" onclick="return doclink('link-86', 'database_info', 'link-77');">database_info</a></tt><tt class="py-op">,</tt> <tt id="link-87" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-87', 'end', 'link-78');">end</a></tt><tt class="py-op">=</tt><tt class="py-string">'total letters'</tt><tt class="py-op">)</tt> </tt>
<a name="L205"></a><tt class="py-lineno"> 205</tt>  <tt class="py-line">            <tt id="link-88" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-88', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-89" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.database_info" class="py-name" href="#" onclick="return doclink('link-89', 'database_info', 'link-77');">database_info</a></tt><tt class="py-op">,</tt> <tt id="link-90" class="py-name"><a title="Bio.Alphabet.Alphabet.contains
Bio.Alphabet.AlphabetEncoder.contains
Bio.Alphabet.Gapped.contains
Bio.Alphabet.HasStopCodon.contains
Bio.Phylo.Consensus._BitString.contains" class="py-name" href="#" onclick="return doclink('link-90', 'contains', 'link-22');">contains</a></tt><tt class="py-op">=</tt><tt class="py-string">'sequences'</tt><tt class="py-op">)</tt> </tt>
<a name="L206"></a><tt class="py-lineno"> 206</tt>  <tt class="py-line">            <tt id="link-91" class="py-name"><a title="Bio.ParserSupport.read_and_call_while" class="py-name" href="#" onclick="return doclink('link-91', 'read_and_call_while', 'link-31');">read_and_call_while</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-92" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-92', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L207"></a><tt class="py-lineno"> 207</tt>  <tt class="py-line"> </tt>
<a name="L208"></a><tt class="py-lineno"> 208</tt>  <tt class="py-line">            <tt class="py-comment"># Read the Query lines and the following blank line.</tt> </tt>
<a name="L209"></a><tt class="py-lineno"> 209</tt>  <tt class="py-line">            <tt class="py-comment"># Or, on BLAST 2.2.22+ there is no blank link - need to spot</tt> </tt>
<a name="L210"></a><tt class="py-lineno"> 210</tt>  <tt class="py-line">            <tt class="py-comment"># the "... Score     E" line instead.</tt> </tt>
<a name="L211"></a><tt class="py-lineno"> 211</tt>  <tt class="py-line">            <tt id="link-93" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-93', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-94" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.query_info" class="py-name" href="#" onclick="return doclink('link-94', 'query_info', 'link-70');">query_info</a></tt><tt class="py-op">,</tt> <tt id="link-95" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-95', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'Query='</tt><tt class="py-op">)</tt> </tt>
<a name="L212"></a><tt class="py-lineno"> 212</tt>  <tt class="py-line">            <tt class="py-comment"># BLAST 2.2.25+ has a blank line before Length=</tt> </tt>
<a name="L213"></a><tt class="py-lineno"> 213</tt>  <tt class="py-line">            <tt id="link-96" class="py-name"><a title="Bio.ParserSupport.read_and_call_until" class="py-name" href="#" onclick="return doclink('link-96', 'read_and_call_until', 'link-20');">read_and_call_until</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-97" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.query_info" class="py-name" href="#" onclick="return doclink('link-97', 'query_info', 'link-70');">query_info</a></tt><tt class="py-op">,</tt> <tt id="link-98" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-98', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'Length='</tt><tt class="py-op">)</tt> </tt>
<a name="L214"></a><tt class="py-lineno"> 214</tt>  <tt class="py-line">            <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L215"></a><tt class="py-lineno"> 215</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">uhandle</tt><tt class="py-op">.</tt><tt id="link-99" class="py-name"><a title="Bio.File.UndoHandle.peekline" class="py-name" href="#" onclick="return doclink('link-99', 'peekline', 'link-65');">peekline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L216"></a><tt class="py-lineno"> 216</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-100" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-100', 'strip', 'link-66');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt id="link-101" class="py-name"><a title="Bio.Blast.NCBIStandalone._score_e_re" class="py-name" href="#" onclick="return doclink('link-101', '_score_e_re', 'link-15');">_score_e_re</a></tt><tt class="py-op">.</tt><tt id="link-102" class="py-name" targets="Method Bio.KDTree.KDTree'.KDTree.search()=Bio.KDTree.KDTree%27.KDTree-class.html#search,Method Bio.PDB.NeighborSearch'.NeighborSearch.search()=Bio.PDB.NeighborSearch%27.NeighborSearch-class.html#search,Method Bio.Phylo.TreeConstruction.NNITreeSearcher.search()=Bio.Phylo.TreeConstruction.NNITreeSearcher-class.html#search,Method Bio.Phylo.TreeConstruction.TreeSearcher.search()=Bio.Phylo.TreeConstruction.TreeSearcher-class.html#search,Class Method Bio.Restriction.Restriction.AbstractCut.search()=Bio.Restriction.Restriction.AbstractCut-class.html#search,Method Bio.Restriction.Restriction.RestrictionBatch.search()=Bio.Restriction.Restriction.RestrictionBatch-class.html#search,Function Bio.SCOP.search()=Bio.SCOP-module.html#search,Function Bio.TogoWS.search()=Bio.TogoWS-module.html#search,Method Bio.motifs.Instances.search()=Bio.motifs.Instances-class.html#search,Method Bio.motifs.matrix.PositionSpecificScoringMatrix.search()=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#search"><a title="Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-102', 'search', 'link-102');">search</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L217"></a><tt class="py-lineno"> 217</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L218"></a><tt class="py-lineno"> 218</tt>  <tt class="py-line">                <tt class="py-comment"># It is more of the query (and its length)</tt> </tt>
<a name="L219"></a><tt class="py-lineno"> 219</tt>  <tt class="py-line">                <tt id="link-103" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-103', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-104" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.query_info" class="py-name" href="#" onclick="return doclink('link-104', 'query_info', 'link-70');">query_info</a></tt><tt class="py-op">)</tt> </tt>
<a name="L220"></a><tt class="py-lineno"> 220</tt>  <tt class="py-line">            <tt id="link-105" class="py-name"><a title="Bio.ParserSupport.read_and_call_while" class="py-name" href="#" onclick="return doclink('link-105', 'read_and_call_while', 'link-31');">read_and_call_while</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-106" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-106', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L221"></a><tt class="py-lineno"> 221</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L222"></a><tt class="py-lineno"> 222</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid header?"</tt><tt class="py-op">)</tt> </tt>
<a name="L223"></a><tt class="py-lineno"> 223</tt>  <tt class="py-line"> </tt>
<a name="L224"></a><tt class="py-lineno"> 224</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-107" class="py-name" targets="Method Bio.Blast.NCBIStandalone._BlastConsumer.end_header()=Bio.Blast.NCBIStandalone._BlastConsumer-class.html#end_header,Method Bio.Blast.NCBIStandalone._HeaderConsumer.end_header()=Bio.Blast.NCBIStandalone._HeaderConsumer-class.html#end_header,Method Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_header()=Bio.Blast.NCBIStandalone._PSIBlastConsumer-class.html#end_header"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.end_header
Bio.Blast.NCBIStandalone._HeaderConsumer.end_header
Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_header" class="py-name" href="#" onclick="return doclink('link-107', 'end_header', 'link-107');">end_header</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L225"></a><tt class="py-lineno"> 225</tt>  <tt class="py-line"> </tt>
<a name="_Scanner._scan_rounds"></a><div id="_Scanner._scan_rounds-def"><a name="L226"></a><tt class="py-lineno"> 226</tt> <a class="py-toggle" href="#" id="_Scanner._scan_rounds-toggle" onclick="return toggle('_Scanner._scan_rounds');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._Scanner-class.html#_scan_rounds">_scan_rounds</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">uhandle</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_Scanner._scan_rounds-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_Scanner._scan_rounds-expanded"><a name="L227"></a><tt class="py-lineno"> 227</tt>  <tt class="py-line">        <tt class="py-comment"># Scan a bunch of rounds.</tt> </tt>
<a name="L228"></a><tt class="py-lineno"> 228</tt>  <tt class="py-line">        <tt class="py-comment"># Each round begins with either a "Searching......" line</tt> </tt>
<a name="L229"></a><tt class="py-lineno"> 229</tt>  <tt class="py-line">        <tt class="py-comment"># or a 'Score     E' line followed by descriptions and alignments.</tt> </tt>
<a name="L230"></a><tt class="py-lineno"> 230</tt>  <tt class="py-line">        <tt class="py-comment"># The email server doesn't give the "Searching....." line.</tt> </tt>
<a name="L231"></a><tt class="py-lineno"> 231</tt>  <tt class="py-line">        <tt class="py-comment"># If there is no 'Searching.....' line then you'll first see a</tt> </tt>
<a name="L232"></a><tt class="py-lineno"> 232</tt>  <tt class="py-line">        <tt class="py-comment"># 'Results from round' line</tt> </tt>
<a name="L233"></a><tt class="py-lineno"> 233</tt>  <tt class="py-line"> </tt>
<a name="L234"></a><tt class="py-lineno"> 234</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-108" class="py-name" targets="Method Bio.Blast.NCBIStandalone._Scanner._eof()=Bio.Blast.NCBIStandalone._Scanner-class.html#_eof"><a title="Bio.Blast.NCBIStandalone._Scanner._eof" class="py-name" href="#" onclick="return doclink('link-108', '_eof', 'link-108');">_eof</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L235"></a><tt class="py-lineno"> 235</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-109" class="py-name" targets="Function Bio.ParserSupport.safe_peekline()=Bio.ParserSupport-module.html#safe_peekline"><a title="Bio.ParserSupport.safe_peekline" class="py-name" href="#" onclick="return doclink('link-109', 'safe_peekline', 'link-109');">safe_peekline</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">)</tt> </tt>
<a name="L236"></a><tt class="py-lineno"> 236</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-110" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-110', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Searching'</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> \ </tt>
<a name="L237"></a><tt class="py-lineno"> 237</tt>  <tt class="py-line">               <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-111" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-111', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Results from round'</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> \ </tt>
<a name="L238"></a><tt class="py-lineno"> 238</tt>  <tt class="py-line">               <tt id="link-112" class="py-name"><a title="Bio.Blast.NCBIStandalone._score_e_re" class="py-name" href="#" onclick="return doclink('link-112', '_score_e_re', 'link-15');">_score_e_re</a></tt><tt class="py-op">.</tt><tt id="link-113" class="py-name"><a title="Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-113', 'search', 'link-102');">search</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> \ </tt>
<a name="L239"></a><tt class="py-lineno"> 239</tt>  <tt class="py-line">               <tt class="py-string">'No hits found'</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L240"></a><tt class="py-lineno"> 240</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L241"></a><tt class="py-lineno"> 241</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-114" class="py-name" targets="Method Bio.Blast.NCBIStandalone._Scanner._scan_descriptions()=Bio.Blast.NCBIStandalone._Scanner-class.html#_scan_descriptions"><a title="Bio.Blast.NCBIStandalone._Scanner._scan_descriptions" class="py-name" href="#" onclick="return doclink('link-114', '_scan_descriptions', 'link-114');">_scan_descriptions</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">)</tt> </tt>
<a name="L242"></a><tt class="py-lineno"> 242</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-115" class="py-name" targets="Method Bio.Blast.NCBIStandalone._Scanner._scan_alignments()=Bio.Blast.NCBIStandalone._Scanner-class.html#_scan_alignments"><a title="Bio.Blast.NCBIStandalone._Scanner._scan_alignments" class="py-name" href="#" onclick="return doclink('link-115', '_scan_alignments', 'link-115');">_scan_alignments</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">)</tt> </tt>
</div><a name="L243"></a><tt class="py-lineno"> 243</tt>  <tt class="py-line"> </tt>
<a name="_Scanner._scan_descriptions"></a><div id="_Scanner._scan_descriptions-def"><a name="L244"></a><tt class="py-lineno"> 244</tt> <a class="py-toggle" href="#" id="_Scanner._scan_descriptions-toggle" onclick="return toggle('_Scanner._scan_descriptions');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._Scanner-class.html#_scan_descriptions">_scan_descriptions</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">uhandle</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_Scanner._scan_descriptions-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_Scanner._scan_descriptions-expanded"><a name="L245"></a><tt class="py-lineno"> 245</tt>  <tt class="py-line">        <tt class="py-comment"># Searching..................................................done</tt> </tt>
<a name="L246"></a><tt class="py-lineno"> 246</tt>  <tt class="py-line">        <tt class="py-comment"># Results from round 2</tt> </tt>
<a name="L247"></a><tt class="py-lineno"> 247</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L248"></a><tt class="py-lineno"> 248</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L249"></a><tt class="py-lineno"> 249</tt>  <tt class="py-line">        <tt class="py-comment">#                                                                    Sc</tt> </tt>
<a name="L250"></a><tt class="py-lineno"> 250</tt>  <tt class="py-line">        <tt class="py-comment"># Sequences producing significant alignments:                        (b</tt> </tt>
<a name="L251"></a><tt class="py-lineno"> 251</tt>  <tt class="py-line">        <tt class="py-comment"># Sequences used in model and found again:</tt> </tt>
<a name="L252"></a><tt class="py-lineno"> 252</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L253"></a><tt class="py-lineno"> 253</tt>  <tt class="py-line">        <tt class="py-comment"># d1tde_2 3.4.1.4.4 (119-244) Thioredoxin reductase [Escherichia ...</tt> </tt>
<a name="L254"></a><tt class="py-lineno"> 254</tt>  <tt class="py-line">        <tt class="py-comment"># d1tcob_ 1.31.1.5.16 Calcineurin regulatory subunit (B-chain) [B...</tt> </tt>
<a name="L255"></a><tt class="py-lineno"> 255</tt>  <tt class="py-line">        <tt class="py-comment"># d1symb_ 1.31.1.2.2 Calcyclin (S100) [RAT (RATTUS NORVEGICUS)]</tt> </tt>
<a name="L256"></a><tt class="py-lineno"> 256</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L257"></a><tt class="py-lineno"> 257</tt>  <tt class="py-line">        <tt class="py-comment"># Sequences not found previously or not previously below threshold:</tt> </tt>
<a name="L258"></a><tt class="py-lineno"> 258</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L259"></a><tt class="py-lineno"> 259</tt>  <tt class="py-line">        <tt class="py-comment"># d1osa__ 1.31.1.5.11 Calmodulin [Paramecium tetraurelia]</tt> </tt>
<a name="L260"></a><tt class="py-lineno"> 260</tt>  <tt class="py-line">        <tt class="py-comment"># d1aoza3 2.5.1.3.3 (339-552) Ascorbate oxidase [zucchini (Cucurb...</tt> </tt>
<a name="L261"></a><tt class="py-lineno"> 261</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L262"></a><tt class="py-lineno"> 262</tt>  <tt class="py-line"> </tt>
<a name="L263"></a><tt class="py-lineno"> 263</tt>  <tt class="py-line">        <tt class="py-comment"># If PSI-BLAST, may also have:</tt> </tt>
<a name="L264"></a><tt class="py-lineno"> 264</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L265"></a><tt class="py-lineno"> 265</tt>  <tt class="py-line">        <tt class="py-comment"># CONVERGED!</tt> </tt>
<a name="L266"></a><tt class="py-lineno"> 266</tt>  <tt class="py-line"> </tt>
<a name="L267"></a><tt class="py-lineno"> 267</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-116" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DescriptionConsumer.start_descriptions()=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#start_descriptions,Method Bio.Blast.NCBIStandalone._PSIBlastConsumer.start_descriptions()=Bio.Blast.NCBIStandalone._PSIBlastConsumer-class.html#start_descriptions"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.start_descriptions
Bio.Blast.NCBIStandalone._PSIBlastConsumer.start_descriptions" class="py-name" href="#" onclick="return doclink('link-116', 'start_descriptions', 'link-116');">start_descriptions</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L268"></a><tt class="py-lineno"> 268</tt>  <tt class="py-line"> </tt>
<a name="L269"></a><tt class="py-lineno"> 269</tt>  <tt class="py-line">        <tt class="py-comment"># Read 'Searching'</tt> </tt>
<a name="L270"></a><tt class="py-lineno"> 270</tt>  <tt class="py-line">        <tt class="py-comment"># This line seems to be missing in BLASTN 2.1.2 (others?)</tt> </tt>
<a name="L271"></a><tt class="py-lineno"> 271</tt>  <tt class="py-line">        <tt id="link-117" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-117', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-118" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-118', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt id="link-119" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-119', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'Searching'</tt><tt class="py-op">)</tt> </tt>
<a name="L272"></a><tt class="py-lineno"> 272</tt>  <tt class="py-line"> </tt>
<a name="L273"></a><tt class="py-lineno"> 273</tt>  <tt class="py-line">        <tt class="py-comment"># blastpgp 2.0.10 from NCBI 9/19/99 for Solaris sometimes crashes here.</tt> </tt>
<a name="L274"></a><tt class="py-lineno"> 274</tt>  <tt class="py-line">        <tt class="py-comment"># If this happens, the handle will yield no more information.</tt> </tt>
<a name="L275"></a><tt class="py-lineno"> 275</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">uhandle</tt><tt class="py-op">.</tt><tt id="link-120" class="py-name"><a title="Bio.File.UndoHandle.peekline" class="py-name" href="#" onclick="return doclink('link-120', 'peekline', 'link-65');">peekline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L276"></a><tt class="py-lineno"> 276</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unexpected end of blast report.  "</tt> <tt class="py-op">+</tt> </tt>
<a name="L277"></a><tt class="py-lineno"> 277</tt>  <tt class="py-line">                  <tt class="py-string">"Looks suspiciously like a PSI-BLAST crash."</tt><tt class="py-op">)</tt> </tt>
<a name="L278"></a><tt class="py-lineno"> 278</tt>  <tt class="py-line"> </tt>
<a name="L279"></a><tt class="py-lineno"> 279</tt>  <tt class="py-line">        <tt class="py-comment"># BLASTN 2.2.3 sometimes spews a bunch of warnings and errors here:</tt> </tt>
<a name="L280"></a><tt class="py-lineno"> 280</tt>  <tt class="py-line">        <tt class="py-comment"># Searching[blastall] WARNING:  [000.000]  AT1G08320: SetUpBlastSearch</tt> </tt>
<a name="L281"></a><tt class="py-lineno"> 281</tt>  <tt class="py-line">        <tt class="py-comment"># [blastall] ERROR:  [000.000]  AT1G08320: Blast:</tt> </tt>
<a name="L282"></a><tt class="py-lineno"> 282</tt>  <tt class="py-line">        <tt class="py-comment"># [blastall] ERROR:  [000.000]  AT1G08320: Blast: Query must be at leas</tt> </tt>
<a name="L283"></a><tt class="py-lineno"> 283</tt>  <tt class="py-line">        <tt class="py-comment"># done</tt> </tt>
<a name="L284"></a><tt class="py-lineno"> 284</tt>  <tt class="py-line">        <tt class="py-comment"># Reported by David Weisman.</tt> </tt>
<a name="L285"></a><tt class="py-lineno"> 285</tt>  <tt class="py-line">        <tt class="py-comment"># Check for these error lines and ignore them for now.  Let</tt> </tt>
<a name="L286"></a><tt class="py-lineno"> 286</tt>  <tt class="py-line">        <tt class="py-comment"># the BlastErrorParser deal with them.</tt> </tt>
<a name="L287"></a><tt class="py-lineno"> 287</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">uhandle</tt><tt class="py-op">.</tt><tt id="link-121" class="py-name"><a title="Bio.File.UndoHandle.peekline" class="py-name" href="#" onclick="return doclink('link-121', 'peekline', 'link-65');">peekline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L288"></a><tt class="py-lineno"> 288</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"ERROR:"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt> <tt class="py-keyword">or</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-122" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-122', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"done"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L289"></a><tt class="py-lineno"> 289</tt>  <tt class="py-line">            <tt id="link-123" class="py-name"><a title="Bio.ParserSupport.read_and_call_while" class="py-name" href="#" onclick="return doclink('link-123', 'read_and_call_while', 'link-31');">read_and_call_while</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-124" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-124', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt id="link-125" class="py-name"><a title="Bio.Alphabet.Alphabet.contains
Bio.Alphabet.AlphabetEncoder.contains
Bio.Alphabet.Gapped.contains
Bio.Alphabet.HasStopCodon.contains
Bio.Phylo.Consensus._BitString.contains" class="py-name" href="#" onclick="return doclink('link-125', 'contains', 'link-22');">contains</a></tt><tt class="py-op">=</tt><tt class="py-string">"ERROR:"</tt><tt class="py-op">)</tt> </tt>
<a name="L290"></a><tt class="py-lineno"> 290</tt>  <tt class="py-line">            <tt id="link-126" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-126', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-127" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-127', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt id="link-128" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-128', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">"done"</tt><tt class="py-op">)</tt> </tt>
<a name="L291"></a><tt class="py-lineno"> 291</tt>  <tt class="py-line"> </tt>
<a name="L292"></a><tt class="py-lineno"> 292</tt>  <tt class="py-line">        <tt class="py-comment"># Check to see if this is PSI-BLAST.</tt> </tt>
<a name="L293"></a><tt class="py-lineno"> 293</tt>  <tt class="py-line">        <tt class="py-comment"># If it is, the 'Searching' line will be followed by:</tt> </tt>
<a name="L294"></a><tt class="py-lineno"> 294</tt>  <tt class="py-line">        <tt class="py-comment"># (version 2.0.10)</tt> </tt>
<a name="L295"></a><tt class="py-lineno"> 295</tt>  <tt class="py-line">        <tt class="py-comment">#     Searching.............................</tt> </tt>
<a name="L296"></a><tt class="py-lineno"> 296</tt>  <tt class="py-line">        <tt class="py-comment">#     Results from round 2</tt> </tt>
<a name="L297"></a><tt class="py-lineno"> 297</tt>  <tt class="py-line">        <tt class="py-comment"># or (version 2.0.11)</tt> </tt>
<a name="L298"></a><tt class="py-lineno"> 298</tt>  <tt class="py-line">        <tt class="py-comment">#     Searching.............................</tt> </tt>
<a name="L299"></a><tt class="py-lineno"> 299</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L300"></a><tt class="py-lineno"> 300</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L301"></a><tt class="py-lineno"> 301</tt>  <tt class="py-line">        <tt class="py-comment">#     Results from round 2</tt> </tt>
<a name="L302"></a><tt class="py-lineno"> 302</tt>  <tt class="py-line"> </tt>
<a name="L303"></a><tt class="py-lineno"> 303</tt>  <tt class="py-line">        <tt class="py-comment"># Skip a bunch of blank lines.</tt> </tt>
<a name="L304"></a><tt class="py-lineno"> 304</tt>  <tt class="py-line">        <tt id="link-129" class="py-name"><a title="Bio.ParserSupport.read_and_call_while" class="py-name" href="#" onclick="return doclink('link-129', 'read_and_call_while', 'link-31');">read_and_call_while</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-130" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-130', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L305"></a><tt class="py-lineno"> 305</tt>  <tt class="py-line">        <tt class="py-comment"># Check for the results line if it's there.</tt> </tt>
<a name="L306"></a><tt class="py-lineno"> 306</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-131" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-131', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-132" class="py-name" targets="Method Bio.Blast.NCBIStandalone._BlastConsumer.round()=Bio.Blast.NCBIStandalone._BlastConsumer-class.html#round,Method Bio.Blast.NCBIStandalone._DescriptionConsumer.round()=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#round"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.round
Bio.Blast.NCBIStandalone._DescriptionConsumer.round" class="py-name" href="#" onclick="return doclink('link-132', 'round', 'link-132');">round</a></tt><tt class="py-op">,</tt> <tt id="link-133" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-133', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'Results'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L307"></a><tt class="py-lineno"> 307</tt>  <tt class="py-line">            <tt id="link-134" class="py-name"><a title="Bio.ParserSupport.read_and_call_while" class="py-name" href="#" onclick="return doclink('link-134', 'read_and_call_while', 'link-31');">read_and_call_while</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-135" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-135', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L308"></a><tt class="py-lineno"> 308</tt>  <tt class="py-line"> </tt>
<a name="L309"></a><tt class="py-lineno"> 309</tt>  <tt class="py-line">        <tt class="py-comment"># Three things can happen here:</tt> </tt>
<a name="L310"></a><tt class="py-lineno"> 310</tt>  <tt class="py-line">        <tt class="py-comment"># 1.  line contains 'Score     E'</tt> </tt>
<a name="L311"></a><tt class="py-lineno"> 311</tt>  <tt class="py-line">        <tt class="py-comment"># 2.  line contains "No hits found"</tt> </tt>
<a name="L312"></a><tt class="py-lineno"> 312</tt>  <tt class="py-line">        <tt class="py-comment"># 3.  no descriptions</tt> </tt>
<a name="L313"></a><tt class="py-lineno"> 313</tt>  <tt class="py-line">        <tt class="py-comment"># The first one begins a bunch of descriptions.  The last two</tt> </tt>
<a name="L314"></a><tt class="py-lineno"> 314</tt>  <tt class="py-line">        <tt class="py-comment"># indicates that no descriptions follow, and we should go straight</tt> </tt>
<a name="L315"></a><tt class="py-lineno"> 315</tt>  <tt class="py-line">        <tt class="py-comment"># to the alignments.</tt> </tt>
<a name="L316"></a><tt class="py-lineno"> 316</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-136" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-136', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt> </tt>
<a name="L317"></a><tt class="py-lineno"> 317</tt>  <tt class="py-line">           <tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-137" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DescriptionConsumer.description_header()=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#description_header"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description_header" class="py-name" href="#" onclick="return doclink('link-137', 'description_header', 'link-137');">description_header</a></tt><tt class="py-op">,</tt> </tt>
<a name="L318"></a><tt class="py-lineno"> 318</tt>  <tt class="py-line">           <tt class="py-name">has_re</tt><tt class="py-op">=</tt><tt id="link-138" class="py-name"><a title="Bio.Blast.NCBIStandalone._score_e_re" class="py-name" href="#" onclick="return doclink('link-138', '_score_e_re', 'link-15');">_score_e_re</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L319"></a><tt class="py-lineno"> 319</tt>  <tt class="py-line">            <tt class="py-comment"># Either case 2 or 3.  Look for "No hits found".</tt> </tt>
<a name="L320"></a><tt class="py-lineno"> 320</tt>  <tt class="py-line">            <tt id="link-139" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-139', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-140" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DescriptionConsumer.no_hits()=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#no_hits"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.no_hits" class="py-name" href="#" onclick="return doclink('link-140', 'no_hits', 'link-140');">no_hits</a></tt><tt class="py-op">,</tt> </tt>
<a name="L321"></a><tt class="py-lineno"> 321</tt>  <tt class="py-line">                                  <tt id="link-141" class="py-name"><a title="Bio.Alphabet.Alphabet.contains
Bio.Alphabet.AlphabetEncoder.contains
Bio.Alphabet.Gapped.contains
Bio.Alphabet.HasStopCodon.contains
Bio.Phylo.Consensus._BitString.contains" class="py-name" href="#" onclick="return doclink('link-141', 'contains', 'link-22');">contains</a></tt><tt class="py-op">=</tt><tt class="py-string">'No hits found'</tt><tt class="py-op">)</tt> </tt>
<a name="L322"></a><tt class="py-lineno"> 322</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L323"></a><tt class="py-lineno"> 323</tt>  <tt class="py-line">                <tt id="link-142" class="py-name"><a title="Bio.ParserSupport.read_and_call_while" class="py-name" href="#" onclick="return doclink('link-142', 'read_and_call_while', 'link-31');">read_and_call_while</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-143" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-143', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L324"></a><tt class="py-lineno"> 324</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt> <tt class="py-keyword">as</tt> <tt class="py-name">err</tt><tt class="py-op">:</tt> </tt>
<a name="L325"></a><tt class="py-lineno"> 325</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">err</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-string">"Unexpected end of stream."</tt><tt class="py-op">:</tt> </tt>
<a name="L326"></a><tt class="py-lineno"> 326</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt class="py-name">err</tt> </tt>
<a name="L327"></a><tt class="py-lineno"> 327</tt>  <tt class="py-line"> </tt>
<a name="L328"></a><tt class="py-lineno"> 328</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-144" class="py-name" targets="Method Bio.Blast.NCBIStandalone._BlastConsumer.end_descriptions()=Bio.Blast.NCBIStandalone._BlastConsumer-class.html#end_descriptions,Method Bio.Blast.NCBIStandalone._DescriptionConsumer.end_descriptions()=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#end_descriptions,Method Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_descriptions()=Bio.Blast.NCBIStandalone._PSIBlastConsumer-class.html#end_descriptions"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.end_descriptions
Bio.Blast.NCBIStandalone._DescriptionConsumer.end_descriptions
Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_descriptions" class="py-name" href="#" onclick="return doclink('link-144', 'end_descriptions', 'link-144');">end_descriptions</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L329"></a><tt class="py-lineno"> 329</tt>  <tt class="py-line">            <tt class="py-comment"># Stop processing.</tt> </tt>
<a name="L330"></a><tt class="py-lineno"> 330</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L331"></a><tt class="py-lineno"> 331</tt>  <tt class="py-line"> </tt>
<a name="L332"></a><tt class="py-lineno"> 332</tt>  <tt class="py-line">        <tt class="py-comment"># Read the score header lines</tt> </tt>
<a name="L333"></a><tt class="py-lineno"> 333</tt>  <tt class="py-line">        <tt id="link-145" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-145', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-146" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description_header" class="py-name" href="#" onclick="return doclink('link-146', 'description_header', 'link-137');">description_header</a></tt><tt class="py-op">,</tt> </tt>
<a name="L334"></a><tt class="py-lineno"> 334</tt>  <tt class="py-line">                      <tt id="link-147" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-147', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'Sequences producing'</tt><tt class="py-op">)</tt> </tt>
<a name="L335"></a><tt class="py-lineno"> 335</tt>  <tt class="py-line"> </tt>
<a name="L336"></a><tt class="py-lineno"> 336</tt>  <tt class="py-line">        <tt class="py-comment"># If PSI-BLAST, read the 'Sequences used in model' line.</tt> </tt>
<a name="L337"></a><tt class="py-lineno"> 337</tt>  <tt class="py-line">        <tt id="link-148" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-148', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-149" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DescriptionConsumer.model_sequences()=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#model_sequences"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.model_sequences" class="py-name" href="#" onclick="return doclink('link-149', 'model_sequences', 'link-149');">model_sequences</a></tt><tt class="py-op">,</tt> </tt>
<a name="L338"></a><tt class="py-lineno"> 338</tt>  <tt class="py-line">                              <tt id="link-150" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-150', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'Sequences used in model'</tt><tt class="py-op">)</tt> </tt>
<a name="L339"></a><tt class="py-lineno"> 339</tt>  <tt class="py-line">        <tt id="link-151" class="py-name"><a title="Bio.ParserSupport.read_and_call_while" class="py-name" href="#" onclick="return doclink('link-151', 'read_and_call_while', 'link-31');">read_and_call_while</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-152" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-152', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L340"></a><tt class="py-lineno"> 340</tt>  <tt class="py-line"> </tt>
<a name="L341"></a><tt class="py-lineno"> 341</tt>  <tt class="py-line">        <tt class="py-comment"># In BLAT, rather than a "No hits found" line, we just</tt> </tt>
<a name="L342"></a><tt class="py-lineno"> 342</tt>  <tt class="py-line">        <tt class="py-comment"># get no descriptions (and no alignments). This can be</tt> </tt>
<a name="L343"></a><tt class="py-lineno"> 343</tt>  <tt class="py-line">        <tt class="py-comment"># spotted because the next line is the database block:</tt> </tt>
<a name="L344"></a><tt class="py-lineno"> 344</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-153" class="py-name"><a title="Bio.ParserSupport.safe_peekline" class="py-name" href="#" onclick="return doclink('link-153', 'safe_peekline', 'link-109');">safe_peekline</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-154" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-154', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"  Database:"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L345"></a><tt class="py-lineno"> 345</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-155" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.end_descriptions
Bio.Blast.NCBIStandalone._DescriptionConsumer.end_descriptions
Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_descriptions" class="py-name" href="#" onclick="return doclink('link-155', 'end_descriptions', 'link-144');">end_descriptions</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L346"></a><tt class="py-lineno"> 346</tt>  <tt class="py-line">            <tt class="py-comment"># Stop processing.</tt> </tt>
<a name="L347"></a><tt class="py-lineno"> 347</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L348"></a><tt class="py-lineno"> 348</tt>  <tt class="py-line"> </tt>
<a name="L349"></a><tt class="py-lineno"> 349</tt>  <tt class="py-line">        <tt class="py-comment"># Read the descriptions and the following blank lines, making</tt> </tt>
<a name="L350"></a><tt class="py-lineno"> 350</tt>  <tt class="py-line">        <tt class="py-comment"># sure that there are descriptions.</tt> </tt>
<a name="L351"></a><tt class="py-lineno"> 351</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">uhandle</tt><tt class="py-op">.</tt><tt id="link-156" class="py-name"><a title="Bio.File.UndoHandle.peekline" class="py-name" href="#" onclick="return doclink('link-156', 'peekline', 'link-65');">peekline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-157" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-157', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Sequences not found'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L352"></a><tt class="py-lineno"> 352</tt>  <tt class="py-line">            <tt id="link-158" class="py-name"><a title="Bio.ParserSupport.read_and_call_until" class="py-name" href="#" onclick="return doclink('link-158', 'read_and_call_until', 'link-20');">read_and_call_until</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-159" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DescriptionConsumer.description()=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#description,Method Bio.Phylo.PhyloXMLIO.Writer.description()=Bio.Phylo.PhyloXMLIO.Writer-class.html#description,Variable Bio.SearchIO._model.hit.Hit.description=Bio.SearchIO._model.hit.Hit-class.html#description,Variable Bio.SearchIO._model.query.QueryResult.description=Bio.SearchIO._model.query.QueryResult-class.html#description"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-159', 'description', 'link-159');">description</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L353"></a><tt class="py-lineno"> 353</tt>  <tt class="py-line">            <tt id="link-160" class="py-name"><a title="Bio.ParserSupport.read_and_call_while" class="py-name" href="#" onclick="return doclink('link-160', 'read_and_call_while', 'link-31');">read_and_call_while</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-161" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-161', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L354"></a><tt class="py-lineno"> 354</tt>  <tt class="py-line"> </tt>
<a name="L355"></a><tt class="py-lineno"> 355</tt>  <tt class="py-line">        <tt class="py-comment"># If PSI-BLAST, read the 'Sequences not found' line followed</tt> </tt>
<a name="L356"></a><tt class="py-lineno"> 356</tt>  <tt class="py-line">        <tt class="py-comment"># by more descriptions.  However, I need to watch out for the</tt> </tt>
<a name="L357"></a><tt class="py-lineno"> 357</tt>  <tt class="py-line">        <tt class="py-comment"># case where there were no sequences not found previously, in</tt> </tt>
<a name="L358"></a><tt class="py-lineno"> 358</tt>  <tt class="py-line">        <tt class="py-comment"># which case there will be no more descriptions.</tt> </tt>
<a name="L359"></a><tt class="py-lineno"> 359</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-162" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-162', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-163" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DescriptionConsumer.nonmodel_sequences()=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#nonmodel_sequences"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.nonmodel_sequences" class="py-name" href="#" onclick="return doclink('link-163', 'nonmodel_sequences', 'link-163');">nonmodel_sequences</a></tt><tt class="py-op">,</tt> </tt>
<a name="L360"></a><tt class="py-lineno"> 360</tt>  <tt class="py-line">                                 <tt id="link-164" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-164', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'Sequences not found'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L361"></a><tt class="py-lineno"> 361</tt>  <tt class="py-line">            <tt class="py-comment"># Read the descriptions and the following blank lines.</tt> </tt>
<a name="L362"></a><tt class="py-lineno"> 362</tt>  <tt class="py-line">            <tt id="link-165" class="py-name"><a title="Bio.ParserSupport.read_and_call_while" class="py-name" href="#" onclick="return doclink('link-165', 'read_and_call_while', 'link-31');">read_and_call_while</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-166" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-166', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L363"></a><tt class="py-lineno"> 363</tt>  <tt class="py-line">            <tt class="py-name">l</tt> <tt class="py-op">=</tt> <tt id="link-167" class="py-name"><a title="Bio.ParserSupport.safe_peekline" class="py-name" href="#" onclick="return doclink('link-167', 'safe_peekline', 'link-109');">safe_peekline</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">)</tt> </tt>
<a name="L364"></a><tt class="py-lineno"> 364</tt>  <tt class="py-line">            <tt class="py-comment"># Brad -- added check for QUERY. On some PSI-BLAST outputs</tt> </tt>
<a name="L365"></a><tt class="py-lineno"> 365</tt>  <tt class="py-line">            <tt class="py-comment"># there will be a 'Sequences not found' line followed by no</tt> </tt>
<a name="L366"></a><tt class="py-lineno"> 366</tt>  <tt class="py-line">            <tt class="py-comment"># descriptions. Check for this case since the first thing you'll</tt> </tt>
<a name="L367"></a><tt class="py-lineno"> 367</tt>  <tt class="py-line">            <tt class="py-comment"># get is a blank line and then 'QUERY'</tt> </tt>
<a name="L368"></a><tt class="py-lineno"> 368</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">l</tt><tt class="py-op">.</tt><tt id="link-168" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-168', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'CONVERGED'</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-name">l</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-string">'&gt;'</tt> \ </tt>
<a name="L369"></a><tt class="py-lineno"> 369</tt>  <tt class="py-line">                    <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt class="py-name">l</tt><tt class="py-op">.</tt><tt id="link-169" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-169', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'QUERY'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L370"></a><tt class="py-lineno"> 370</tt>  <tt class="py-line">                <tt id="link-170" class="py-name"><a title="Bio.ParserSupport.read_and_call_until" class="py-name" href="#" onclick="return doclink('link-170', 'read_and_call_until', 'link-20');">read_and_call_until</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-171" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-171', 'description', 'link-159');">description</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L371"></a><tt class="py-lineno"> 371</tt>  <tt class="py-line">                <tt id="link-172" class="py-name"><a title="Bio.ParserSupport.read_and_call_while" class="py-name" href="#" onclick="return doclink('link-172', 'read_and_call_while', 'link-31');">read_and_call_while</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-173" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-173', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L372"></a><tt class="py-lineno"> 372</tt>  <tt class="py-line"> </tt>
<a name="L373"></a><tt class="py-lineno"> 373</tt>  <tt class="py-line">        <tt id="link-174" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-174', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-175" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DescriptionConsumer.converged()=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#converged"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.converged" class="py-name" href="#" onclick="return doclink('link-175', 'converged', 'link-175');">converged</a></tt><tt class="py-op">,</tt> <tt id="link-176" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-176', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'CONVERGED'</tt><tt class="py-op">)</tt> </tt>
<a name="L374"></a><tt class="py-lineno"> 374</tt>  <tt class="py-line">        <tt id="link-177" class="py-name"><a title="Bio.ParserSupport.read_and_call_while" class="py-name" href="#" onclick="return doclink('link-177', 'read_and_call_while', 'link-31');">read_and_call_while</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-178" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-178', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L375"></a><tt class="py-lineno"> 375</tt>  <tt class="py-line"> </tt>
<a name="L376"></a><tt class="py-lineno"> 376</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-179" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.end_descriptions
Bio.Blast.NCBIStandalone._DescriptionConsumer.end_descriptions
Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_descriptions" class="py-name" href="#" onclick="return doclink('link-179', 'end_descriptions', 'link-144');">end_descriptions</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L377"></a><tt class="py-lineno"> 377</tt>  <tt class="py-line"> </tt>
<a name="_Scanner._scan_alignments"></a><div id="_Scanner._scan_alignments-def"><a name="L378"></a><tt class="py-lineno"> 378</tt> <a class="py-toggle" href="#" id="_Scanner._scan_alignments-toggle" onclick="return toggle('_Scanner._scan_alignments');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._Scanner-class.html#_scan_alignments">_scan_alignments</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">uhandle</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_Scanner._scan_alignments-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_Scanner._scan_alignments-expanded"><a name="L379"></a><tt class="py-lineno"> 379</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-180" class="py-name"><a title="Bio.Blast.NCBIStandalone._Scanner._eof" class="py-name" href="#" onclick="return doclink('link-180', '_eof', 'link-108');">_eof</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L380"></a><tt class="py-lineno"> 380</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L381"></a><tt class="py-lineno"> 381</tt>  <tt class="py-line"> </tt>
<a name="L382"></a><tt class="py-lineno"> 382</tt>  <tt class="py-line">        <tt class="py-comment"># qblast inserts a helpful line here.</tt> </tt>
<a name="L383"></a><tt class="py-lineno"> 383</tt>  <tt class="py-line">        <tt id="link-181" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-181', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-182" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-182', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt id="link-183" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-183', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">"ALIGNMENTS"</tt><tt class="py-op">)</tt> </tt>
<a name="L384"></a><tt class="py-lineno"> 384</tt>  <tt class="py-line"> </tt>
<a name="L385"></a><tt class="py-lineno"> 385</tt>  <tt class="py-line">        <tt class="py-comment"># First, check to see if I'm at the database report.</tt> </tt>
<a name="L386"></a><tt class="py-lineno"> 386</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-184" class="py-name"><a title="Bio.ParserSupport.safe_peekline" class="py-name" href="#" onclick="return doclink('link-184', 'safe_peekline', 'link-109');">safe_peekline</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">)</tt> </tt>
<a name="L387"></a><tt class="py-lineno"> 387</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L388"></a><tt class="py-lineno"> 388</tt>  <tt class="py-line">            <tt class="py-comment"># EOF</tt> </tt>
<a name="L389"></a><tt class="py-lineno"> 389</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L390"></a><tt class="py-lineno"> 390</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-185" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-185', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'  Database'</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-186" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-186', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"Lambda"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L391"></a><tt class="py-lineno"> 391</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L392"></a><tt class="py-lineno"> 392</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'&gt;'</tt><tt class="py-op">:</tt> </tt>
<a name="L393"></a><tt class="py-lineno"> 393</tt>  <tt class="py-line">            <tt class="py-comment"># XXX make a better check here between pairwise and masterslave</tt> </tt>
<a name="L394"></a><tt class="py-lineno"> 394</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-187" class="py-name" targets="Method Bio.Blast.NCBIStandalone._Scanner._scan_pairwise_alignments()=Bio.Blast.NCBIStandalone._Scanner-class.html#_scan_pairwise_alignments"><a title="Bio.Blast.NCBIStandalone._Scanner._scan_pairwise_alignments" class="py-name" href="#" onclick="return doclink('link-187', '_scan_pairwise_alignments', 'link-187');">_scan_pairwise_alignments</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">)</tt> </tt>
<a name="L395"></a><tt class="py-lineno"> 395</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-188" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-188', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Effective'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L396"></a><tt class="py-lineno"> 396</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L397"></a><tt class="py-lineno"> 397</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L398"></a><tt class="py-lineno"> 398</tt>  <tt class="py-line">            <tt class="py-comment"># XXX put in a check to make sure I'm in a masterslave alignment</tt> </tt>
<a name="L399"></a><tt class="py-lineno"> 399</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-189" class="py-name" targets="Method Bio.Blast.NCBIStandalone._Scanner._scan_masterslave_alignment()=Bio.Blast.NCBIStandalone._Scanner-class.html#_scan_masterslave_alignment"><a title="Bio.Blast.NCBIStandalone._Scanner._scan_masterslave_alignment" class="py-name" href="#" onclick="return doclink('link-189', '_scan_masterslave_alignment', 'link-189');">_scan_masterslave_alignment</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">)</tt> </tt>
</div><a name="L400"></a><tt class="py-lineno"> 400</tt>  <tt class="py-line"> </tt>
<a name="_Scanner._scan_pairwise_alignments"></a><div id="_Scanner._scan_pairwise_alignments-def"><a name="L401"></a><tt class="py-lineno"> 401</tt> <a class="py-toggle" href="#" id="_Scanner._scan_pairwise_alignments-toggle" onclick="return toggle('_Scanner._scan_pairwise_alignments');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._Scanner-class.html#_scan_pairwise_alignments">_scan_pairwise_alignments</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">uhandle</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_Scanner._scan_pairwise_alignments-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_Scanner._scan_pairwise_alignments-expanded"><a name="L402"></a><tt class="py-lineno"> 402</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-190" class="py-name"><a title="Bio.Blast.NCBIStandalone._Scanner._eof" class="py-name" href="#" onclick="return doclink('link-190', '_eof', 'link-108');">_eof</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L403"></a><tt class="py-lineno"> 403</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-191" class="py-name"><a title="Bio.ParserSupport.safe_peekline" class="py-name" href="#" onclick="return doclink('link-191', 'safe_peekline', 'link-109');">safe_peekline</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">)</tt> </tt>
<a name="L404"></a><tt class="py-lineno"> 404</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-string">'&gt;'</tt><tt class="py-op">:</tt> </tt>
<a name="L405"></a><tt class="py-lineno"> 405</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L406"></a><tt class="py-lineno"> 406</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-192" class="py-name" targets="Method Bio.Blast.NCBIStandalone._Scanner._scan_one_pairwise_alignment()=Bio.Blast.NCBIStandalone._Scanner-class.html#_scan_one_pairwise_alignment"><a title="Bio.Blast.NCBIStandalone._Scanner._scan_one_pairwise_alignment" class="py-name" href="#" onclick="return doclink('link-192', '_scan_one_pairwise_alignment', 'link-192');">_scan_one_pairwise_alignment</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">)</tt> </tt>
</div><a name="L407"></a><tt class="py-lineno"> 407</tt>  <tt class="py-line"> </tt>
<a name="_Scanner._scan_one_pairwise_alignment"></a><div id="_Scanner._scan_one_pairwise_alignment-def"><a name="L408"></a><tt class="py-lineno"> 408</tt> <a class="py-toggle" href="#" id="_Scanner._scan_one_pairwise_alignment-toggle" onclick="return toggle('_Scanner._scan_one_pairwise_alignment');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._Scanner-class.html#_scan_one_pairwise_alignment">_scan_one_pairwise_alignment</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">uhandle</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_Scanner._scan_one_pairwise_alignment-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_Scanner._scan_one_pairwise_alignment-expanded"><a name="L409"></a><tt class="py-lineno"> 409</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-193" class="py-name"><a title="Bio.Blast.NCBIStandalone._Scanner._eof" class="py-name" href="#" onclick="return doclink('link-193', '_eof', 'link-108');">_eof</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L410"></a><tt class="py-lineno"> 410</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L411"></a><tt class="py-lineno"> 411</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-194" class="py-name" targets="Method Bio.Blast.NCBIStandalone._AlignmentConsumer.start_alignment()=Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#start_alignment"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.start_alignment" class="py-name" href="#" onclick="return doclink('link-194', 'start_alignment', 'link-194');">start_alignment</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L412"></a><tt class="py-lineno"> 412</tt>  <tt class="py-line"> </tt>
<a name="L413"></a><tt class="py-lineno"> 413</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-195" class="py-name" targets="Method Bio.Blast.NCBIStandalone._Scanner._scan_alignment_header()=Bio.Blast.NCBIStandalone._Scanner-class.html#_scan_alignment_header"><a title="Bio.Blast.NCBIStandalone._Scanner._scan_alignment_header" class="py-name" href="#" onclick="return doclink('link-195', '_scan_alignment_header', 'link-195');">_scan_alignment_header</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">)</tt> </tt>
<a name="L414"></a><tt class="py-lineno"> 414</tt>  <tt class="py-line"> </tt>
<a name="L415"></a><tt class="py-lineno"> 415</tt>  <tt class="py-line">        <tt class="py-comment"># Scan a bunch of score/alignment pairs.</tt> </tt>
<a name="L416"></a><tt class="py-lineno"> 416</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L417"></a><tt class="py-lineno"> 417</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-196" class="py-name"><a title="Bio.Blast.NCBIStandalone._Scanner._eof" class="py-name" href="#" onclick="return doclink('link-196', '_eof', 'link-108');">_eof</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L418"></a><tt class="py-lineno"> 418</tt>  <tt class="py-line">                <tt class="py-comment"># Shouldn't have issued that _scan_alignment_header event...</tt> </tt>
<a name="L419"></a><tt class="py-lineno"> 419</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L420"></a><tt class="py-lineno"> 420</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-197" class="py-name"><a title="Bio.ParserSupport.safe_peekline" class="py-name" href="#" onclick="return doclink('link-197', 'safe_peekline', 'link-109');">safe_peekline</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">)</tt> </tt>
<a name="L421"></a><tt class="py-lineno"> 421</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-198" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-198', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">' Score'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L422"></a><tt class="py-lineno"> 422</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L423"></a><tt class="py-lineno"> 423</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-199" class="py-name" targets="Method Bio.Blast.NCBIStandalone._Scanner._scan_hsp()=Bio.Blast.NCBIStandalone._Scanner-class.html#_scan_hsp"><a title="Bio.Blast.NCBIStandalone._Scanner._scan_hsp" class="py-name" href="#" onclick="return doclink('link-199', '_scan_hsp', 'link-199');">_scan_hsp</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">)</tt> </tt>
<a name="L424"></a><tt class="py-lineno"> 424</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-200" class="py-name" targets="Method Bio.Blast.NCBIStandalone._AlignmentConsumer.end_alignment()=Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#end_alignment,Method Bio.Blast.NCBIStandalone._BlastConsumer.end_alignment()=Bio.Blast.NCBIStandalone._BlastConsumer-class.html#end_alignment,Method Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_alignment()=Bio.Blast.NCBIStandalone._PSIBlastConsumer-class.html#end_alignment"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.end_alignment
Bio.Blast.NCBIStandalone._BlastConsumer.end_alignment
Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_alignment" class="py-name" href="#" onclick="return doclink('link-200', 'end_alignment', 'link-200');">end_alignment</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L425"></a><tt class="py-lineno"> 425</tt>  <tt class="py-line"> </tt>
<a name="_Scanner._scan_alignment_header"></a><div id="_Scanner._scan_alignment_header-def"><a name="L426"></a><tt class="py-lineno"> 426</tt> <a class="py-toggle" href="#" id="_Scanner._scan_alignment_header-toggle" onclick="return toggle('_Scanner._scan_alignment_header');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._Scanner-class.html#_scan_alignment_header">_scan_alignment_header</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">uhandle</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_Scanner._scan_alignment_header-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_Scanner._scan_alignment_header-expanded"><a name="L427"></a><tt class="py-lineno"> 427</tt>  <tt class="py-line">        <tt class="py-comment"># &gt;d1rip__ 2.24.7.1.1 Ribosomal S17 protein [Bacillus</tt> </tt>
<a name="L428"></a><tt class="py-lineno"> 428</tt>  <tt class="py-line">        <tt class="py-comment">#           stearothermophilus]</tt> </tt>
<a name="L429"></a><tt class="py-lineno"> 429</tt>  <tt class="py-line">        <tt class="py-comment">#           Length = 81</tt> </tt>
<a name="L430"></a><tt class="py-lineno"> 430</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L431"></a><tt class="py-lineno"> 431</tt>  <tt class="py-line">        <tt class="py-comment"># Or, more recently with different white space:</tt> </tt>
<a name="L432"></a><tt class="py-lineno"> 432</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L433"></a><tt class="py-lineno"> 433</tt>  <tt class="py-line">        <tt class="py-comment"># &gt;gi|15799684|ref|NP_285696.1| threonine synthase ...</tt> </tt>
<a name="L434"></a><tt class="py-lineno"> 434</tt>  <tt class="py-line">        <tt class="py-comment">#  gi|15829258|ref|NP_308031.1| threonine synthase</tt> </tt>
<a name="L435"></a><tt class="py-lineno"> 435</tt>  <tt class="py-line">        <tt class="py-comment">#  ...</tt> </tt>
<a name="L436"></a><tt class="py-lineno"> 436</tt>  <tt class="py-line">        <tt class="py-comment"># Length=428</tt> </tt>
<a name="L437"></a><tt class="py-lineno"> 437</tt>  <tt class="py-line">        <tt id="link-201" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-201', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-202" class="py-name" targets="Method Bio.Blast.NCBIStandalone._AlignmentConsumer.title()=Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#title,Method Bio.GenBank._FeatureConsumer.title()=Bio.GenBank._FeatureConsumer-class.html#title,Method Bio.GenBank._RecordConsumer.title()=Bio.GenBank._RecordConsumer-class.html#title"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-202', 'title', 'link-202');">title</a></tt><tt class="py-op">,</tt> <tt id="link-203" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-203', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'&gt;'</tt><tt class="py-op">)</tt> </tt>
<a name="L438"></a><tt class="py-lineno"> 438</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L439"></a><tt class="py-lineno"> 439</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-204" class="py-name" targets="Function Bio.ParserSupport.safe_readline()=Bio.ParserSupport-module.html#safe_readline"><a title="Bio.ParserSupport.safe_readline" class="py-name" href="#" onclick="return doclink('link-204', 'safe_readline', 'link-204');">safe_readline</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">)</tt> </tt>
<a name="L440"></a><tt class="py-lineno"> 440</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-205" class="py-name" targets="Method Bio.Seq.Seq.lstrip()=Bio.Seq.Seq-class.html#lstrip"><a title="Bio.Seq.Seq.lstrip" class="py-name" href="#" onclick="return doclink('link-205', 'lstrip', 'link-205');">lstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-206" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-206', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Length ='</tt><tt class="py-op">)</tt> \ </tt>
<a name="L441"></a><tt class="py-lineno"> 441</tt>  <tt class="py-line">            <tt class="py-keyword">or</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-207" class="py-name"><a title="Bio.Seq.Seq.lstrip" class="py-name" href="#" onclick="return doclink('link-207', 'lstrip', 'link-205');">lstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-208" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-208', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Length='</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L442"></a><tt class="py-lineno"> 442</tt>  <tt class="py-line">                <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-209" class="py-name" targets="Method Bio.Blast.NCBIStandalone._AlignmentConsumer.length()=Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#length"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-209', 'length', 'link-209');">length</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L443"></a><tt class="py-lineno"> 443</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L444"></a><tt class="py-lineno"> 444</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-210" class="py-name"><a title="Bio.ParserSupport.is_blank_line" class="py-name" href="#" onclick="return doclink('link-210', 'is_blank_line', 'link-40');">is_blank_line</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L445"></a><tt class="py-lineno"> 445</tt>  <tt class="py-line">                <tt class="py-comment"># Check to make sure I haven't missed the Length line</tt> </tt>
<a name="L446"></a><tt class="py-lineno"> 446</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"I missed the Length in an alignment header"</tt><tt class="py-op">)</tt> </tt>
<a name="L447"></a><tt class="py-lineno"> 447</tt>  <tt class="py-line">            <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-211" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-211', 'title', 'link-202');">title</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L448"></a><tt class="py-lineno"> 448</tt>  <tt class="py-line"> </tt>
<a name="L449"></a><tt class="py-lineno"> 449</tt>  <tt class="py-line">        <tt class="py-comment"># Older versions of BLAST will have a line with some spaces.</tt> </tt>
<a name="L450"></a><tt class="py-lineno"> 450</tt>  <tt class="py-line">        <tt class="py-comment"># Version 2.0.14 (maybe 2.0.13?) and above print a true blank line.</tt> </tt>
<a name="L451"></a><tt class="py-lineno"> 451</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-212" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-212', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-213" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-213', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> </tt>
<a name="L452"></a><tt class="py-lineno"> 452</tt>  <tt class="py-line">                                     <tt id="link-214" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-214', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'          '</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L453"></a><tt class="py-lineno"> 453</tt>  <tt class="py-line">            <tt id="link-215" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-215', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-216" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-216', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
</div><a name="L454"></a><tt class="py-lineno"> 454</tt>  <tt class="py-line"> </tt>
<a name="_Scanner._scan_hsp"></a><div id="_Scanner._scan_hsp-def"><a name="L455"></a><tt class="py-lineno"> 455</tt> <a class="py-toggle" href="#" id="_Scanner._scan_hsp-toggle" onclick="return toggle('_Scanner._scan_hsp');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._Scanner-class.html#_scan_hsp">_scan_hsp</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">uhandle</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_Scanner._scan_hsp-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_Scanner._scan_hsp-expanded"><a name="L456"></a><tt class="py-lineno"> 456</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-217" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HSPConsumer.start_hsp()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#start_hsp"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.start_hsp" class="py-name" href="#" onclick="return doclink('link-217', 'start_hsp', 'link-217');">start_hsp</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L457"></a><tt class="py-lineno"> 457</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-218" class="py-name" targets="Method Bio.Blast.NCBIStandalone._Scanner._scan_hsp_header()=Bio.Blast.NCBIStandalone._Scanner-class.html#_scan_hsp_header"><a title="Bio.Blast.NCBIStandalone._Scanner._scan_hsp_header" class="py-name" href="#" onclick="return doclink('link-218', '_scan_hsp_header', 'link-218');">_scan_hsp_header</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">)</tt> </tt>
<a name="L458"></a><tt class="py-lineno"> 458</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-219" class="py-name" targets="Method Bio.Blast.NCBIStandalone._Scanner._scan_hsp_alignment()=Bio.Blast.NCBIStandalone._Scanner-class.html#_scan_hsp_alignment"><a title="Bio.Blast.NCBIStandalone._Scanner._scan_hsp_alignment" class="py-name" href="#" onclick="return doclink('link-219', '_scan_hsp_alignment', 'link-219');">_scan_hsp_alignment</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">)</tt> </tt>
<a name="L459"></a><tt class="py-lineno"> 459</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-220" class="py-name" targets="Method Bio.Blast.NCBIStandalone._BlastConsumer.end_hsp()=Bio.Blast.NCBIStandalone._BlastConsumer-class.html#end_hsp,Method Bio.Blast.NCBIStandalone._HSPConsumer.end_hsp()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#end_hsp,Method Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_hsp()=Bio.Blast.NCBIStandalone._PSIBlastConsumer-class.html#end_hsp"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.end_hsp
Bio.Blast.NCBIStandalone._HSPConsumer.end_hsp
Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_hsp" class="py-name" href="#" onclick="return doclink('link-220', 'end_hsp', 'link-220');">end_hsp</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L460"></a><tt class="py-lineno"> 460</tt>  <tt class="py-line"> </tt>
<a name="_Scanner._scan_hsp_header"></a><div id="_Scanner._scan_hsp_header-def"><a name="L461"></a><tt class="py-lineno"> 461</tt> <a class="py-toggle" href="#" id="_Scanner._scan_hsp_header-toggle" onclick="return toggle('_Scanner._scan_hsp_header');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._Scanner-class.html#_scan_hsp_header">_scan_hsp_header</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">uhandle</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_Scanner._scan_hsp_header-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_Scanner._scan_hsp_header-expanded"><a name="L462"></a><tt class="py-lineno"> 462</tt>  <tt class="py-line">        <tt class="py-comment">#  Score = 22.7 bits (47), Expect = 2.5</tt> </tt>
<a name="L463"></a><tt class="py-lineno"> 463</tt>  <tt class="py-line">        <tt class="py-comment">#  Identities = 10/36 (27%), Positives = 18/36 (49%)</tt> </tt>
<a name="L464"></a><tt class="py-lineno"> 464</tt>  <tt class="py-line">        <tt class="py-comment">#  Strand = Plus / Plus</tt> </tt>
<a name="L465"></a><tt class="py-lineno"> 465</tt>  <tt class="py-line">        <tt class="py-comment">#  Frame = +3</tt> </tt>
<a name="L466"></a><tt class="py-lineno"> 466</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L467"></a><tt class="py-lineno"> 467</tt>  <tt class="py-line"> </tt>
<a name="L468"></a><tt class="py-lineno"> 468</tt>  <tt class="py-line">        <tt id="link-221" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-221', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-222" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HSPConsumer.score()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#score"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-222', 'score', 'link-222');">score</a></tt><tt class="py-op">,</tt> <tt id="link-223" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-223', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">' Score'</tt><tt class="py-op">)</tt> </tt>
<a name="L469"></a><tt class="py-lineno"> 469</tt>  <tt class="py-line">        <tt id="link-224" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-224', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-225" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HSPConsumer.identities()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#identities"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.identities" class="py-name" href="#" onclick="return doclink('link-225', 'identities', 'link-225');">identities</a></tt><tt class="py-op">,</tt> <tt id="link-226" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-226', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">' Identities'</tt><tt class="py-op">)</tt> </tt>
<a name="L470"></a><tt class="py-lineno"> 470</tt>  <tt class="py-line">        <tt class="py-comment"># BLASTN</tt> </tt>
<a name="L471"></a><tt class="py-lineno"> 471</tt>  <tt class="py-line">        <tt id="link-227" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-227', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-228" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HSPConsumer.strand()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#strand,Variable Bio.SeqFeature.CompoundLocation.strand=Bio.SeqFeature.CompoundLocation-class.html#strand,Variable Bio.SeqFeature.FeatureLocation.strand=Bio.SeqFeature.FeatureLocation-class.html#strand,Variable Bio.SeqFeature.SeqFeature.strand=Bio.SeqFeature.SeqFeature-class.html#strand"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-228', 'strand', 'link-228');">strand</a></tt><tt class="py-op">,</tt> <tt id="link-229" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-229', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">' Strand'</tt><tt class="py-op">)</tt> </tt>
<a name="L472"></a><tt class="py-lineno"> 472</tt>  <tt class="py-line">        <tt class="py-comment"># BLASTX, TBLASTN, TBLASTX</tt> </tt>
<a name="L473"></a><tt class="py-lineno"> 473</tt>  <tt class="py-line">        <tt id="link-230" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-230', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-231" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HSPConsumer.frame()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#frame"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.frame" class="py-name" href="#" onclick="return doclink('link-231', 'frame', 'link-231');">frame</a></tt><tt class="py-op">,</tt> <tt id="link-232" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-232', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">' Frame'</tt><tt class="py-op">)</tt> </tt>
<a name="L474"></a><tt class="py-lineno"> 474</tt>  <tt class="py-line">        <tt id="link-233" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-233', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-234" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-234', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
</div><a name="L475"></a><tt class="py-lineno"> 475</tt>  <tt class="py-line"> </tt>
<a name="_Scanner._scan_hsp_alignment"></a><div id="_Scanner._scan_hsp_alignment-def"><a name="L476"></a><tt class="py-lineno"> 476</tt> <a class="py-toggle" href="#" id="_Scanner._scan_hsp_alignment-toggle" onclick="return toggle('_Scanner._scan_hsp_alignment');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._Scanner-class.html#_scan_hsp_alignment">_scan_hsp_alignment</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">uhandle</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_Scanner._scan_hsp_alignment-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_Scanner._scan_hsp_alignment-expanded"><a name="L477"></a><tt class="py-lineno"> 477</tt>  <tt class="py-line">        <tt class="py-comment"># Query: 11 GRGVSACA-------TCDGFFYRNQKVAVIGGGNTAVEEALYLSNIASEVHLIHRRDGF</tt> </tt>
<a name="L478"></a><tt class="py-lineno"> 478</tt>  <tt class="py-line">        <tt class="py-comment">#           GRGVS+         TC    Y  + + V GGG+ + EE   L     +   I R+</tt> </tt>
<a name="L479"></a><tt class="py-lineno"> 479</tt>  <tt class="py-line">        <tt class="py-comment"># Sbjct: 12 GRGVSSVVRRCIHKPTCKE--YAVKIIDVTGGGSFSAEEVQELREATLKEVDILRKVSG</tt> </tt>
<a name="L480"></a><tt class="py-lineno"> 480</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L481"></a><tt class="py-lineno"> 481</tt>  <tt class="py-line">        <tt class="py-comment"># Query: 64 AEKILIKR 71</tt> </tt>
<a name="L482"></a><tt class="py-lineno"> 482</tt>  <tt class="py-line">        <tt class="py-comment">#              I +K</tt> </tt>
<a name="L483"></a><tt class="py-lineno"> 483</tt>  <tt class="py-line">        <tt class="py-comment"># Sbjct: 70 PNIIQLKD 77</tt> </tt>
<a name="L484"></a><tt class="py-lineno"> 484</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L485"></a><tt class="py-lineno"> 485</tt>  <tt class="py-line"> </tt>
<a name="L486"></a><tt class="py-lineno"> 486</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L487"></a><tt class="py-lineno"> 487</tt>  <tt class="py-line">            <tt class="py-comment"># Blastn adds an extra line filled with spaces before Query</tt> </tt>
<a name="L488"></a><tt class="py-lineno"> 488</tt>  <tt class="py-line">            <tt id="link-235" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-235', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-236" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-236', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt id="link-237" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-237', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'     '</tt><tt class="py-op">)</tt> </tt>
<a name="L489"></a><tt class="py-lineno"> 489</tt>  <tt class="py-line">            <tt id="link-238" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-238', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-239" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HSPConsumer.query()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#query,Variable Bio.SearchIO._model.hsp.HSP.query=Bio.SearchIO._model.hsp.HSP-class.html#query,Variable Bio.SearchIO._model.hsp.HSPFragment.query=Bio.SearchIO._model.hsp.HSPFragment-class.html#query,Module Bio.SearchIO._model.query=Bio.SearchIO._model.query-module.html"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.query
Bio.SearchIO._model.hsp.HSP.query
Bio.SearchIO._model.hsp.HSPFragment.query
Bio.SearchIO._model.query" class="py-name" href="#" onclick="return doclink('link-239', 'query', 'link-239');">query</a></tt><tt class="py-op">,</tt> <tt id="link-240" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-240', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'Query'</tt><tt class="py-op">)</tt> </tt>
<a name="L490"></a><tt class="py-lineno"> 490</tt>  <tt class="py-line">            <tt id="link-241" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-241', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-242" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HSPConsumer.align()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#align,Class Bio.FSSP.fssp_rec.align=Bio.FSSP.fssp_rec.align-class.html,Function Bio.Wise.align()=Bio.Wise-module.html#align,Function Bio.Wise.dnal.align()=Bio.Wise.dnal-module.html#align,Function Bio.Wise.psw.align()=Bio.Wise.psw-module.html#align,Variable Bio.pairwise2.align=Bio.pairwise2-module.html#align"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.align
Bio.FSSP.fssp_rec.align
Bio.Wise.align
Bio.Wise.dnal.align
Bio.Wise.psw.align
Bio.pairwise2.align" class="py-name" href="#" onclick="return doclink('link-242', 'align', 'link-242');">align</a></tt><tt class="py-op">,</tt> <tt id="link-243" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-243', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'     '</tt><tt class="py-op">)</tt> </tt>
<a name="L491"></a><tt class="py-lineno"> 491</tt>  <tt class="py-line">            <tt id="link-244" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-244', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-245" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HSPConsumer.sbjct()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#sbjct"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.sbjct" class="py-name" href="#" onclick="return doclink('link-245', 'sbjct', 'link-245');">sbjct</a></tt><tt class="py-op">,</tt> <tt id="link-246" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-246', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'Sbjct'</tt><tt class="py-op">)</tt> </tt>
<a name="L492"></a><tt class="py-lineno"> 492</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L493"></a><tt class="py-lineno"> 493</tt>  <tt class="py-line">                <tt id="link-247" class="py-name"><a title="Bio.ParserSupport.read_and_call_while" class="py-name" href="#" onclick="return doclink('link-247', 'read_and_call_while', 'link-31');">read_and_call_while</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-248" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-248', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L494"></a><tt class="py-lineno"> 494</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt> <tt class="py-keyword">as</tt> <tt class="py-name">err</tt><tt class="py-op">:</tt> </tt>
<a name="L495"></a><tt class="py-lineno"> 495</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">err</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-string">"Unexpected end of stream."</tt><tt class="py-op">:</tt> </tt>
<a name="L496"></a><tt class="py-lineno"> 496</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt class="py-name">err</tt> </tt>
<a name="L497"></a><tt class="py-lineno"> 497</tt>  <tt class="py-line">                <tt class="py-comment"># End of File (well, it looks like it with recent versions</tt> </tt>
<a name="L498"></a><tt class="py-lineno"> 498</tt>  <tt class="py-line">                <tt class="py-comment"># of BLAST for multiple queries after the Iterator class</tt> </tt>
<a name="L499"></a><tt class="py-lineno"> 499</tt>  <tt class="py-line">                <tt class="py-comment"># has broken up the whole file into chunks).</tt> </tt>
<a name="L500"></a><tt class="py-lineno"> 500</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L501"></a><tt class="py-lineno"> 501</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-249" class="py-name"><a title="Bio.ParserSupport.safe_peekline" class="py-name" href="#" onclick="return doclink('link-249', 'safe_peekline', 'link-109');">safe_peekline</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">)</tt> </tt>
<a name="L502"></a><tt class="py-lineno"> 502</tt>  <tt class="py-line">            <tt class="py-comment"># Alignment continues if I see a 'Query' or the spaces for Blastn.</tt> </tt>
<a name="L503"></a><tt class="py-lineno"> 503</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-250" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-250', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Query'</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-251" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-251', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'     '</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L504"></a><tt class="py-lineno"> 504</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
</div><a name="L505"></a><tt class="py-lineno"> 505</tt>  <tt class="py-line"> </tt>
<a name="_Scanner._scan_masterslave_alignment"></a><div id="_Scanner._scan_masterslave_alignment-def"><a name="L506"></a><tt class="py-lineno"> 506</tt> <a class="py-toggle" href="#" id="_Scanner._scan_masterslave_alignment-toggle" onclick="return toggle('_Scanner._scan_masterslave_alignment');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._Scanner-class.html#_scan_masterslave_alignment">_scan_masterslave_alignment</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">uhandle</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_Scanner._scan_masterslave_alignment-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_Scanner._scan_masterslave_alignment-expanded"><a name="L507"></a><tt class="py-lineno"> 507</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-252" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.start_alignment" class="py-name" href="#" onclick="return doclink('link-252', 'start_alignment', 'link-194');">start_alignment</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L508"></a><tt class="py-lineno"> 508</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L509"></a><tt class="py-lineno"> 509</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-253" class="py-name"><a title="Bio.ParserSupport.safe_readline" class="py-name" href="#" onclick="return doclink('link-253', 'safe_readline', 'link-204');">safe_readline</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">)</tt> </tt>
<a name="L510"></a><tt class="py-lineno"> 510</tt>  <tt class="py-line">            <tt class="py-comment"># Check to see whether I'm finished reading the alignment.</tt> </tt>
<a name="L511"></a><tt class="py-lineno"> 511</tt>  <tt class="py-line">            <tt class="py-comment"># This is indicated by 1) database section, 2) next psi-blast</tt> </tt>
<a name="L512"></a><tt class="py-lineno"> 512</tt>  <tt class="py-line">            <tt class="py-comment"># round, which can also be a 'Results from round' if no</tt> </tt>
<a name="L513"></a><tt class="py-lineno"> 513</tt>  <tt class="py-line">            <tt class="py-comment"># searching line is present</tt> </tt>
<a name="L514"></a><tt class="py-lineno"> 514</tt>  <tt class="py-line">            <tt class="py-comment"># patch by chapmanb</tt> </tt>
<a name="L515"></a><tt class="py-lineno"> 515</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-254" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-254', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Searching'</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> \ </tt>
<a name="L516"></a><tt class="py-lineno"> 516</tt>  <tt class="py-line">                    <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-255" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-255', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Results from round'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L517"></a><tt class="py-lineno"> 517</tt>  <tt class="py-line">                <tt class="py-name">uhandle</tt><tt class="py-op">.</tt><tt id="link-256" class="py-name" targets="Method Bio.File.UndoHandle.saveline()=Bio.File.UndoHandle-class.html#saveline"><a title="Bio.File.UndoHandle.saveline" class="py-name" href="#" onclick="return doclink('link-256', 'saveline', 'link-256');">saveline</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L518"></a><tt class="py-lineno"> 518</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L519"></a><tt class="py-lineno"> 519</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-257" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-257', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'  Database'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L520"></a><tt class="py-lineno"> 520</tt>  <tt class="py-line">                <tt class="py-name">uhandle</tt><tt class="py-op">.</tt><tt id="link-258" class="py-name"><a title="Bio.File.UndoHandle.saveline" class="py-name" href="#" onclick="return doclink('link-258', 'saveline', 'link-256');">saveline</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L521"></a><tt class="py-lineno"> 521</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L522"></a><tt class="py-lineno"> 522</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-259" class="py-name"><a title="Bio.ParserSupport.is_blank_line" class="py-name" href="#" onclick="return doclink('link-259', 'is_blank_line', 'link-40');">is_blank_line</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L523"></a><tt class="py-lineno"> 523</tt>  <tt class="py-line">                <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-260" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-260', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L524"></a><tt class="py-lineno"> 524</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L525"></a><tt class="py-lineno"> 525</tt>  <tt class="py-line">                <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-261" class="py-name" targets="Method Bio.Blast.NCBIStandalone._AlignmentConsumer.multalign()=Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#multalign"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.multalign" class="py-name" href="#" onclick="return doclink('link-261', 'multalign', 'link-261');">multalign</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L526"></a><tt class="py-lineno"> 526</tt>  <tt class="py-line">        <tt id="link-262" class="py-name"><a title="Bio.ParserSupport.read_and_call_while" class="py-name" href="#" onclick="return doclink('link-262', 'read_and_call_while', 'link-31');">read_and_call_while</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-263" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-263', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L527"></a><tt class="py-lineno"> 527</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-264" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.end_alignment
Bio.Blast.NCBIStandalone._BlastConsumer.end_alignment
Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_alignment" class="py-name" href="#" onclick="return doclink('link-264', 'end_alignment', 'link-200');">end_alignment</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L528"></a><tt class="py-lineno"> 528</tt>  <tt class="py-line"> </tt>
<a name="_Scanner._eof"></a><div id="_Scanner._eof-def"><a name="L529"></a><tt class="py-lineno"> 529</tt> <a class="py-toggle" href="#" id="_Scanner._eof-toggle" onclick="return toggle('_Scanner._eof');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._Scanner-class.html#_eof">_eof</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">uhandle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_Scanner._eof-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_Scanner._eof-expanded"><a name="L530"></a><tt class="py-lineno"> 530</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L531"></a><tt class="py-lineno"> 531</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-265" class="py-name"><a title="Bio.ParserSupport.safe_peekline" class="py-name" href="#" onclick="return doclink('link-265', 'safe_peekline', 'link-109');">safe_peekline</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">)</tt> </tt>
<a name="L532"></a><tt class="py-lineno"> 532</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt> <tt class="py-keyword">as</tt> <tt class="py-name">err</tt><tt class="py-op">:</tt> </tt>
<a name="L533"></a><tt class="py-lineno"> 533</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">err</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-string">"Unexpected end of stream."</tt><tt class="py-op">:</tt> </tt>
<a name="L534"></a><tt class="py-lineno"> 534</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">err</tt> </tt>
<a name="L535"></a><tt class="py-lineno"> 535</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L536"></a><tt class="py-lineno"> 536</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt> </tt>
</div><a name="L537"></a><tt class="py-lineno"> 537</tt>  <tt class="py-line"> </tt>
<a name="_Scanner._scan_database_report"></a><div id="_Scanner._scan_database_report-def"><a name="L538"></a><tt class="py-lineno"> 538</tt> <a class="py-toggle" href="#" id="_Scanner._scan_database_report-toggle" onclick="return toggle('_Scanner._scan_database_report');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._Scanner-class.html#_scan_database_report">_scan_database_report</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">uhandle</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_Scanner._scan_database_report-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_Scanner._scan_database_report-expanded"><a name="L539"></a><tt class="py-lineno"> 539</tt>  <tt class="py-line">        <tt class="py-comment">#   Database: sdqib40-1.35.seg.fa</tt> </tt>
<a name="L540"></a><tt class="py-lineno"> 540</tt>  <tt class="py-line">        <tt class="py-comment">#     Posted date:  Nov 1, 1999  4:25 PM</tt> </tt>
<a name="L541"></a><tt class="py-lineno"> 541</tt>  <tt class="py-line">        <tt class="py-comment">#   Number of letters in database: 223,339</tt> </tt>
<a name="L542"></a><tt class="py-lineno"> 542</tt>  <tt class="py-line">        <tt class="py-comment">#   Number of sequences in database:  1323</tt> </tt>
<a name="L543"></a><tt class="py-lineno"> 543</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L544"></a><tt class="py-lineno"> 544</tt>  <tt class="py-line">        <tt class="py-comment"># Lambda     K      H</tt> </tt>
<a name="L545"></a><tt class="py-lineno"> 545</tt>  <tt class="py-line">        <tt class="py-comment">#    0.322    0.133    0.369</tt> </tt>
<a name="L546"></a><tt class="py-lineno"> 546</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L547"></a><tt class="py-lineno"> 547</tt>  <tt class="py-line">        <tt class="py-comment"># Gapped</tt> </tt>
<a name="L548"></a><tt class="py-lineno"> 548</tt>  <tt class="py-line">        <tt class="py-comment"># Lambda     K      H</tt> </tt>
<a name="L549"></a><tt class="py-lineno"> 549</tt>  <tt class="py-line">        <tt class="py-comment">#    0.270   0.0470    0.230</tt> </tt>
<a name="L550"></a><tt class="py-lineno"> 550</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L551"></a><tt class="py-lineno"> 551</tt>  <tt class="py-line">        <tt class="py-comment">##########################################</tt> </tt>
<a name="L552"></a><tt class="py-lineno"> 552</tt>  <tt class="py-line">        <tt class="py-comment"># Or, more recently Blast 2.2.15 gives less blank lines</tt> </tt>
<a name="L553"></a><tt class="py-lineno"> 553</tt>  <tt class="py-line">        <tt class="py-comment">##########################################</tt> </tt>
<a name="L554"></a><tt class="py-lineno"> 554</tt>  <tt class="py-line">        <tt class="py-comment">#   Database: All non-redundant GenBank CDS translations+PDB+SwissProt+PIR+PRF excluding</tt> </tt>
<a name="L555"></a><tt class="py-lineno"> 555</tt>  <tt class="py-line">        <tt class="py-comment"># environmental samples</tt> </tt>
<a name="L556"></a><tt class="py-lineno"> 556</tt>  <tt class="py-line">        <tt class="py-comment">#     Posted date:  Dec 12, 2006  5:51 PM</tt> </tt>
<a name="L557"></a><tt class="py-lineno"> 557</tt>  <tt class="py-line">        <tt class="py-comment">#   Number of letters in database: 667,088,753</tt> </tt>
<a name="L558"></a><tt class="py-lineno"> 558</tt>  <tt class="py-line">        <tt class="py-comment">#   Number of sequences in database:  2,094,974</tt> </tt>
<a name="L559"></a><tt class="py-lineno"> 559</tt>  <tt class="py-line">        <tt class="py-comment"># Lambda     K      H</tt> </tt>
<a name="L560"></a><tt class="py-lineno"> 560</tt>  <tt class="py-line">        <tt class="py-comment">#    0.319    0.136    0.395</tt> </tt>
<a name="L561"></a><tt class="py-lineno"> 561</tt>  <tt class="py-line">        <tt class="py-comment"># Gapped</tt> </tt>
<a name="L562"></a><tt class="py-lineno"> 562</tt>  <tt class="py-line">        <tt class="py-comment"># Lambda     K      H</tt> </tt>
<a name="L563"></a><tt class="py-lineno"> 563</tt>  <tt class="py-line">        <tt class="py-comment">#    0.267   0.0410    0.140</tt> </tt>
<a name="L564"></a><tt class="py-lineno"> 564</tt>  <tt class="py-line"> </tt>
<a name="L565"></a><tt class="py-lineno"> 565</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-266" class="py-name"><a title="Bio.Blast.NCBIStandalone._Scanner._eof" class="py-name" href="#" onclick="return doclink('link-266', '_eof', 'link-108');">_eof</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L566"></a><tt class="py-lineno"> 566</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L567"></a><tt class="py-lineno"> 567</tt>  <tt class="py-line"> </tt>
<a name="L568"></a><tt class="py-lineno"> 568</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-267" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DatabaseReportConsumer.start_database_report()=Bio.Blast.NCBIStandalone._DatabaseReportConsumer-class.html#start_database_report"><a title="Bio.Blast.NCBIStandalone._DatabaseReportConsumer.start_database_report" class="py-name" href="#" onclick="return doclink('link-267', 'start_database_report', 'link-267');">start_database_report</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L569"></a><tt class="py-lineno"> 569</tt>  <tt class="py-line"> </tt>
<a name="L570"></a><tt class="py-lineno"> 570</tt>  <tt class="py-line">        <tt class="py-comment"># Subset of the database(s) listed below</tt> </tt>
<a name="L571"></a><tt class="py-lineno"> 571</tt>  <tt class="py-line">        <tt class="py-comment">#    Number of letters searched: 562,618,960</tt> </tt>
<a name="L572"></a><tt class="py-lineno"> 572</tt>  <tt class="py-line">        <tt class="py-comment">#    Number of sequences searched:  228,924</tt> </tt>
<a name="L573"></a><tt class="py-lineno"> 573</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-268" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-268', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-269" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-269', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt id="link-270" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-270', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">"  Subset"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L574"></a><tt class="py-lineno"> 574</tt>  <tt class="py-line">            <tt id="link-271" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-271', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-272" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-272', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt id="link-273" class="py-name"><a title="Bio.Alphabet.Alphabet.contains
Bio.Alphabet.AlphabetEncoder.contains
Bio.Alphabet.Gapped.contains
Bio.Alphabet.HasStopCodon.contains
Bio.Phylo.Consensus._BitString.contains" class="py-name" href="#" onclick="return doclink('link-273', 'contains', 'link-22');">contains</a></tt><tt class="py-op">=</tt><tt class="py-string">"letters"</tt><tt class="py-op">)</tt> </tt>
<a name="L575"></a><tt class="py-lineno"> 575</tt>  <tt class="py-line">            <tt id="link-274" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-274', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-275" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-275', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt id="link-276" class="py-name"><a title="Bio.Alphabet.Alphabet.contains
Bio.Alphabet.AlphabetEncoder.contains
Bio.Alphabet.Gapped.contains
Bio.Alphabet.HasStopCodon.contains
Bio.Phylo.Consensus._BitString.contains" class="py-name" href="#" onclick="return doclink('link-276', 'contains', 'link-22');">contains</a></tt><tt class="py-op">=</tt><tt class="py-string">"sequences"</tt><tt class="py-op">)</tt> </tt>
<a name="L576"></a><tt class="py-lineno"> 576</tt>  <tt class="py-line">            <tt id="link-277" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-277', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-278" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-278', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt id="link-279" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-279', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">"  "</tt><tt class="py-op">)</tt> </tt>
<a name="L577"></a><tt class="py-lineno"> 577</tt>  <tt class="py-line"> </tt>
<a name="L578"></a><tt class="py-lineno"> 578</tt>  <tt class="py-line">        <tt class="py-comment"># Sameet Mehta reported seeing output from BLASTN 2.2.9 that</tt> </tt>
<a name="L579"></a><tt class="py-lineno"> 579</tt>  <tt class="py-line">        <tt class="py-comment"># was missing the "Database" stanza completely.</tt> </tt>
<a name="L580"></a><tt class="py-lineno"> 580</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt id="link-280" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-280', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-281" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DatabaseReportConsumer.database()=Bio.Blast.NCBIStandalone._DatabaseReportConsumer-class.html#database"><a title="Bio.Blast.NCBIStandalone._DatabaseReportConsumer.database" class="py-name" href="#" onclick="return doclink('link-281', 'database', 'link-281');">database</a></tt><tt class="py-op">,</tt> </tt>
<a name="L581"></a><tt class="py-lineno"> 581</tt>  <tt class="py-line">                <tt id="link-282" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-282', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'  Database'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L582"></a><tt class="py-lineno"> 582</tt>  <tt class="py-line">            <tt class="py-comment"># BLAT output ends abruptly here, without any of the other</tt> </tt>
<a name="L583"></a><tt class="py-lineno"> 583</tt>  <tt class="py-line">            <tt class="py-comment"># information.  Check to see if this is the case.  If so,</tt> </tt>
<a name="L584"></a><tt class="py-lineno"> 584</tt>  <tt class="py-line">            <tt class="py-comment"># then end the database report here gracefully.</tt> </tt>
<a name="L585"></a><tt class="py-lineno"> 585</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">uhandle</tt><tt class="py-op">.</tt><tt id="link-283" class="py-name"><a title="Bio.File.UndoHandle.peekline" class="py-name" href="#" onclick="return doclink('link-283', 'peekline', 'link-65');">peekline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-284" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-284', 'strip', 'link-66');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> \ </tt>
<a name="L586"></a><tt class="py-lineno"> 586</tt>  <tt class="py-line">            <tt class="py-keyword">or</tt> <tt class="py-name">uhandle</tt><tt class="py-op">.</tt><tt id="link-285" class="py-name"><a title="Bio.File.UndoHandle.peekline" class="py-name" href="#" onclick="return doclink('link-285', 'peekline', 'link-65');">peekline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-286" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-286', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"BLAST"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L587"></a><tt class="py-lineno"> 587</tt>  <tt class="py-line">                <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-287" class="py-name" targets="Method Bio.Blast.NCBIStandalone._BlastConsumer.end_database_report()=Bio.Blast.NCBIStandalone._BlastConsumer-class.html#end_database_report,Method Bio.Blast.NCBIStandalone._DatabaseReportConsumer.end_database_report()=Bio.Blast.NCBIStandalone._DatabaseReportConsumer-class.html#end_database_report,Method Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_database_report()=Bio.Blast.NCBIStandalone._PSIBlastConsumer-class.html#end_database_report"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.end_database_report
Bio.Blast.NCBIStandalone._DatabaseReportConsumer.end_database_report
Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_database_report" class="py-name" href="#" onclick="return doclink('link-287', 'end_database_report', 'link-287');">end_database_report</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L588"></a><tt class="py-lineno"> 588</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> </tt>
<a name="L589"></a><tt class="py-lineno"> 589</tt>  <tt class="py-line"> </tt>
<a name="L590"></a><tt class="py-lineno"> 590</tt>  <tt class="py-line">            <tt class="py-comment"># Database can span multiple lines.</tt> </tt>
<a name="L591"></a><tt class="py-lineno"> 591</tt>  <tt class="py-line">            <tt id="link-288" class="py-name"><a title="Bio.ParserSupport.read_and_call_until" class="py-name" href="#" onclick="return doclink('link-288', 'read_and_call_until', 'link-20');">read_and_call_until</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-289" class="py-name"><a title="Bio.Blast.NCBIStandalone._DatabaseReportConsumer.database" class="py-name" href="#" onclick="return doclink('link-289', 'database', 'link-281');">database</a></tt><tt class="py-op">,</tt> <tt id="link-290" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-290', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'    Posted'</tt><tt class="py-op">)</tt> </tt>
<a name="L592"></a><tt class="py-lineno"> 592</tt>  <tt class="py-line">            <tt id="link-291" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-291', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-292" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DatabaseReportConsumer.posted_date()=Bio.Blast.NCBIStandalone._DatabaseReportConsumer-class.html#posted_date"><a title="Bio.Blast.NCBIStandalone._DatabaseReportConsumer.posted_date" class="py-name" href="#" onclick="return doclink('link-292', 'posted_date', 'link-292');">posted_date</a></tt><tt class="py-op">,</tt> <tt id="link-293" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-293', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'    Posted'</tt><tt class="py-op">)</tt> </tt>
<a name="L593"></a><tt class="py-lineno"> 593</tt>  <tt class="py-line">            <tt id="link-294" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-294', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-295" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DatabaseReportConsumer.num_letters_in_database()=Bio.Blast.NCBIStandalone._DatabaseReportConsumer-class.html#num_letters_in_database"><a title="Bio.Blast.NCBIStandalone._DatabaseReportConsumer.num_letters_in_database" class="py-name" href="#" onclick="return doclink('link-295', 'num_letters_in_database', 'link-295');">num_letters_in_database</a></tt><tt class="py-op">,</tt> </tt>
<a name="L594"></a><tt class="py-lineno"> 594</tt>  <tt class="py-line">                       <tt id="link-296" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-296', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'  Number of letters'</tt><tt class="py-op">)</tt> </tt>
<a name="L595"></a><tt class="py-lineno"> 595</tt>  <tt class="py-line">            <tt id="link-297" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-297', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-298" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DatabaseReportConsumer.num_sequences_in_database()=Bio.Blast.NCBIStandalone._DatabaseReportConsumer-class.html#num_sequences_in_database"><a title="Bio.Blast.NCBIStandalone._DatabaseReportConsumer.num_sequences_in_database" class="py-name" href="#" onclick="return doclink('link-298', 'num_sequences_in_database', 'link-298');">num_sequences_in_database</a></tt><tt class="py-op">,</tt> </tt>
<a name="L596"></a><tt class="py-lineno"> 596</tt>  <tt class="py-line">                       <tt id="link-299" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-299', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'  Number of sequences'</tt><tt class="py-op">)</tt> </tt>
<a name="L597"></a><tt class="py-lineno"> 597</tt>  <tt class="py-line">            <tt class="py-comment"># There may not be a line starting with spaces...</tt> </tt>
<a name="L598"></a><tt class="py-lineno"> 598</tt>  <tt class="py-line">            <tt id="link-300" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-300', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-301" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-301', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt id="link-302" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-302', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'  '</tt><tt class="py-op">)</tt> </tt>
<a name="L599"></a><tt class="py-lineno"> 599</tt>  <tt class="py-line"> </tt>
<a name="L600"></a><tt class="py-lineno"> 600</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt id="link-303" class="py-name"><a title="Bio.ParserSupport.safe_readline" class="py-name" href="#" onclick="return doclink('link-303', 'safe_readline', 'link-204');">safe_readline</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">)</tt> </tt>
<a name="L601"></a><tt class="py-lineno"> 601</tt>  <tt class="py-line">            <tt class="py-name">uhandle</tt><tt class="py-op">.</tt><tt id="link-304" class="py-name"><a title="Bio.File.UndoHandle.saveline" class="py-name" href="#" onclick="return doclink('link-304', 'saveline', 'link-256');">saveline</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L602"></a><tt class="py-lineno"> 602</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-string">'Lambda'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L603"></a><tt class="py-lineno"> 603</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L604"></a><tt class="py-lineno"> 604</tt>  <tt class="py-line"> </tt>
<a name="L605"></a><tt class="py-lineno"> 605</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L606"></a><tt class="py-lineno"> 606</tt>  <tt class="py-line">            <tt id="link-305" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-305', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-306" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-306', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt id="link-307" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-307', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'Lambda'</tt><tt class="py-op">)</tt> </tt>
<a name="L607"></a><tt class="py-lineno"> 607</tt>  <tt class="py-line">            <tt id="link-308" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-308', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-309" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DatabaseReportConsumer.ka_params()=Bio.Blast.NCBIStandalone._DatabaseReportConsumer-class.html#ka_params"><a title="Bio.Blast.NCBIStandalone._DatabaseReportConsumer.ka_params" class="py-name" href="#" onclick="return doclink('link-309', 'ka_params', 'link-309');">ka_params</a></tt><tt class="py-op">)</tt> </tt>
<a name="L608"></a><tt class="py-lineno"> 608</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt><tt class="py-op">:</tt> </tt>
<a name="L609"></a><tt class="py-lineno"> 609</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
<a name="L610"></a><tt class="py-lineno"> 610</tt>  <tt class="py-line"> </tt>
<a name="L611"></a><tt class="py-lineno"> 611</tt>  <tt class="py-line">        <tt class="py-comment"># This blank line is optional:</tt> </tt>
<a name="L612"></a><tt class="py-lineno"> 612</tt>  <tt class="py-line">        <tt id="link-310" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-310', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-311" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-311', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L613"></a><tt class="py-lineno"> 613</tt>  <tt class="py-line"> </tt>
<a name="L614"></a><tt class="py-lineno"> 614</tt>  <tt class="py-line">        <tt class="py-comment"># not BLASTP</tt> </tt>
<a name="L615"></a><tt class="py-lineno"> 615</tt>  <tt class="py-line">        <tt id="link-312" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-312', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-313" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DatabaseReportConsumer.gapped()=Bio.Blast.NCBIStandalone._DatabaseReportConsumer-class.html#gapped"><a title="Bio.Blast.NCBIStandalone._DatabaseReportConsumer.gapped" class="py-name" href="#" onclick="return doclink('link-313', 'gapped', 'link-313');">gapped</a></tt><tt class="py-op">,</tt> <tt id="link-314" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-314', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'Gapped'</tt><tt class="py-op">)</tt> </tt>
<a name="L616"></a><tt class="py-lineno"> 616</tt>  <tt class="py-line">        <tt class="py-comment"># not TBLASTX</tt> </tt>
<a name="L617"></a><tt class="py-lineno"> 617</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-315" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-315', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-316" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-316', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt id="link-317" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-317', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'Lambda'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L618"></a><tt class="py-lineno"> 618</tt>  <tt class="py-line">            <tt id="link-318" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-318', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-319" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DatabaseReportConsumer.ka_params_gap()=Bio.Blast.NCBIStandalone._DatabaseReportConsumer-class.html#ka_params_gap"><a title="Bio.Blast.NCBIStandalone._DatabaseReportConsumer.ka_params_gap" class="py-name" href="#" onclick="return doclink('link-319', 'ka_params_gap', 'link-319');">ka_params_gap</a></tt><tt class="py-op">)</tt> </tt>
<a name="L619"></a><tt class="py-lineno"> 619</tt>  <tt class="py-line"> </tt>
<a name="L620"></a><tt class="py-lineno"> 620</tt>  <tt class="py-line">        <tt class="py-comment"># Blast 2.2.4 can sometimes skip the whole parameter section.</tt> </tt>
<a name="L621"></a><tt class="py-lineno"> 621</tt>  <tt class="py-line">        <tt class="py-comment"># Thus, I need to be careful not to read past the end of the</tt> </tt>
<a name="L622"></a><tt class="py-lineno"> 622</tt>  <tt class="py-line">        <tt class="py-comment"># file.</tt> </tt>
<a name="L623"></a><tt class="py-lineno"> 623</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L624"></a><tt class="py-lineno"> 624</tt>  <tt class="py-line">            <tt id="link-320" class="py-name"><a title="Bio.ParserSupport.read_and_call_while" class="py-name" href="#" onclick="return doclink('link-320', 'read_and_call_while', 'link-31');">read_and_call_while</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-321" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-321', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> <tt class="py-name">blank</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L625"></a><tt class="py-lineno"> 625</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt> <tt class="py-keyword">as</tt> <tt id="link-322" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.x=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#x"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-322', 'x', 'link-322');">x</a></tt><tt class="py-op">:</tt> </tt>
<a name="L626"></a><tt class="py-lineno"> 626</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-323" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-323', 'x', 'link-322');">x</a></tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-string">"Unexpected end of stream."</tt><tt class="py-op">:</tt> </tt>
<a name="L627"></a><tt class="py-lineno"> 627</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> </tt>
<a name="L628"></a><tt class="py-lineno"> 628</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-324" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.end_database_report
Bio.Blast.NCBIStandalone._DatabaseReportConsumer.end_database_report
Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_database_report" class="py-name" href="#" onclick="return doclink('link-324', 'end_database_report', 'link-287');">end_database_report</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L629"></a><tt class="py-lineno"> 629</tt>  <tt class="py-line"> </tt>
<a name="_Scanner._scan_parameters"></a><div id="_Scanner._scan_parameters-def"><a name="L630"></a><tt class="py-lineno"> 630</tt> <a class="py-toggle" href="#" id="_Scanner._scan_parameters-toggle" onclick="return toggle('_Scanner._scan_parameters');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._Scanner-class.html#_scan_parameters">_scan_parameters</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">uhandle</tt><tt class="py-op">,</tt> <tt class="py-param">consumer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_Scanner._scan_parameters-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_Scanner._scan_parameters-expanded"><a name="L631"></a><tt class="py-lineno"> 631</tt>  <tt class="py-line">        <tt class="py-comment"># Matrix: BLOSUM62</tt> </tt>
<a name="L632"></a><tt class="py-lineno"> 632</tt>  <tt class="py-line">        <tt class="py-comment"># Gap Penalties: Existence: 11, Extension: 1</tt> </tt>
<a name="L633"></a><tt class="py-lineno"> 633</tt>  <tt class="py-line">        <tt class="py-comment"># Number of Hits to DB: 50604</tt> </tt>
<a name="L634"></a><tt class="py-lineno"> 634</tt>  <tt class="py-line">        <tt class="py-comment"># Number of Sequences: 1323</tt> </tt>
<a name="L635"></a><tt class="py-lineno"> 635</tt>  <tt class="py-line">        <tt class="py-comment"># Number of extensions: 1526</tt> </tt>
<a name="L636"></a><tt class="py-lineno"> 636</tt>  <tt class="py-line">        <tt class="py-comment"># Number of successful extensions: 6</tt> </tt>
<a name="L637"></a><tt class="py-lineno"> 637</tt>  <tt class="py-line">        <tt class="py-comment"># Number of sequences better than 10.0: 5</tt> </tt>
<a name="L638"></a><tt class="py-lineno"> 638</tt>  <tt class="py-line">        <tt class="py-comment"># Number of HSP's better than 10.0 without gapping: 5</tt> </tt>
<a name="L639"></a><tt class="py-lineno"> 639</tt>  <tt class="py-line">        <tt class="py-comment"># Number of HSP's successfully gapped in prelim test: 0</tt> </tt>
<a name="L640"></a><tt class="py-lineno"> 640</tt>  <tt class="py-line">        <tt class="py-comment"># Number of HSP's that attempted gapping in prelim test: 1</tt> </tt>
<a name="L641"></a><tt class="py-lineno"> 641</tt>  <tt class="py-line">        <tt class="py-comment"># Number of HSP's gapped (non-prelim): 5</tt> </tt>
<a name="L642"></a><tt class="py-lineno"> 642</tt>  <tt class="py-line">        <tt class="py-comment"># length of query: 140</tt> </tt>
<a name="L643"></a><tt class="py-lineno"> 643</tt>  <tt class="py-line">        <tt class="py-comment"># length of database: 223,339</tt> </tt>
<a name="L644"></a><tt class="py-lineno"> 644</tt>  <tt class="py-line">        <tt class="py-comment"># effective HSP length: 39</tt> </tt>
<a name="L645"></a><tt class="py-lineno"> 645</tt>  <tt class="py-line">        <tt class="py-comment"># effective length of query: 101</tt> </tt>
<a name="L646"></a><tt class="py-lineno"> 646</tt>  <tt class="py-line">        <tt class="py-comment"># effective length of database: 171,742</tt> </tt>
<a name="L647"></a><tt class="py-lineno"> 647</tt>  <tt class="py-line">        <tt class="py-comment"># effective search space: 17345942</tt> </tt>
<a name="L648"></a><tt class="py-lineno"> 648</tt>  <tt class="py-line">        <tt class="py-comment"># effective search space used: 17345942</tt> </tt>
<a name="L649"></a><tt class="py-lineno"> 649</tt>  <tt class="py-line">        <tt class="py-comment"># T: 11</tt> </tt>
<a name="L650"></a><tt class="py-lineno"> 650</tt>  <tt class="py-line">        <tt class="py-comment"># A: 40</tt> </tt>
<a name="L651"></a><tt class="py-lineno"> 651</tt>  <tt class="py-line">        <tt class="py-comment"># X1: 16 ( 7.4 bits)</tt> </tt>
<a name="L652"></a><tt class="py-lineno"> 652</tt>  <tt class="py-line">        <tt class="py-comment"># X2: 38 (14.8 bits)</tt> </tt>
<a name="L653"></a><tt class="py-lineno"> 653</tt>  <tt class="py-line">        <tt class="py-comment"># X3: 64 (24.9 bits)</tt> </tt>
<a name="L654"></a><tt class="py-lineno"> 654</tt>  <tt class="py-line">        <tt class="py-comment"># S1: 41 (21.9 bits)</tt> </tt>
<a name="L655"></a><tt class="py-lineno"> 655</tt>  <tt class="py-line">        <tt class="py-comment"># S2: 42 (20.8 bits)</tt> </tt>
<a name="L656"></a><tt class="py-lineno"> 656</tt>  <tt class="py-line">        <tt class="py-comment">##########################################</tt> </tt>
<a name="L657"></a><tt class="py-lineno"> 657</tt>  <tt class="py-line">        <tt class="py-comment"># Or, more recently Blast(x) 2.2.15 gives</tt> </tt>
<a name="L658"></a><tt class="py-lineno"> 658</tt>  <tt class="py-line">        <tt class="py-comment">##########################################</tt> </tt>
<a name="L659"></a><tt class="py-lineno"> 659</tt>  <tt class="py-line">        <tt class="py-comment"># Matrix: BLOSUM62</tt> </tt>
<a name="L660"></a><tt class="py-lineno"> 660</tt>  <tt class="py-line">        <tt class="py-comment"># Gap Penalties: Existence: 11, Extension: 1</tt> </tt>
<a name="L661"></a><tt class="py-lineno"> 661</tt>  <tt class="py-line">        <tt class="py-comment"># Number of Sequences: 4535438</tt> </tt>
<a name="L662"></a><tt class="py-lineno"> 662</tt>  <tt class="py-line">        <tt class="py-comment"># Number of Hits to DB: 2,588,844,100</tt> </tt>
<a name="L663"></a><tt class="py-lineno"> 663</tt>  <tt class="py-line">        <tt class="py-comment"># Number of extensions: 60427286</tt> </tt>
<a name="L664"></a><tt class="py-lineno"> 664</tt>  <tt class="py-line">        <tt class="py-comment"># Number of successful extensions: 126433</tt> </tt>
<a name="L665"></a><tt class="py-lineno"> 665</tt>  <tt class="py-line">        <tt class="py-comment"># Number of sequences better than  2.0: 30</tt> </tt>
<a name="L666"></a><tt class="py-lineno"> 666</tt>  <tt class="py-line">        <tt class="py-comment"># Number of HSP's gapped: 126387</tt> </tt>
<a name="L667"></a><tt class="py-lineno"> 667</tt>  <tt class="py-line">        <tt class="py-comment"># Number of HSP's successfully gapped: 35</tt> </tt>
<a name="L668"></a><tt class="py-lineno"> 668</tt>  <tt class="py-line">        <tt class="py-comment"># Length of query: 291</tt> </tt>
<a name="L669"></a><tt class="py-lineno"> 669</tt>  <tt class="py-line">        <tt class="py-comment"># Length of database: 1,573,298,872</tt> </tt>
<a name="L670"></a><tt class="py-lineno"> 670</tt>  <tt class="py-line">        <tt class="py-comment"># Length adjustment: 130</tt> </tt>
<a name="L671"></a><tt class="py-lineno"> 671</tt>  <tt class="py-line">        <tt class="py-comment"># Effective length of query: 161</tt> </tt>
<a name="L672"></a><tt class="py-lineno"> 672</tt>  <tt class="py-line">        <tt class="py-comment"># Effective length of database: 983,691,932</tt> </tt>
<a name="L673"></a><tt class="py-lineno"> 673</tt>  <tt class="py-line">        <tt class="py-comment"># Effective search space: 158374401052</tt> </tt>
<a name="L674"></a><tt class="py-lineno"> 674</tt>  <tt class="py-line">        <tt class="py-comment"># Effective search space used: 158374401052</tt> </tt>
<a name="L675"></a><tt class="py-lineno"> 675</tt>  <tt class="py-line">        <tt class="py-comment"># Neighboring words threshold: 12</tt> </tt>
<a name="L676"></a><tt class="py-lineno"> 676</tt>  <tt class="py-line">        <tt class="py-comment"># Window for multiple hits: 40</tt> </tt>
<a name="L677"></a><tt class="py-lineno"> 677</tt>  <tt class="py-line">        <tt class="py-comment"># X1: 16 ( 7.3 bits)</tt> </tt>
<a name="L678"></a><tt class="py-lineno"> 678</tt>  <tt class="py-line">        <tt class="py-comment"># X2: 38 (14.6 bits)</tt> </tt>
<a name="L679"></a><tt class="py-lineno"> 679</tt>  <tt class="py-line">        <tt class="py-comment"># X3: 64 (24.7 bits)</tt> </tt>
<a name="L680"></a><tt class="py-lineno"> 680</tt>  <tt class="py-line">        <tt class="py-comment"># S1: 41 (21.7 bits)</tt> </tt>
<a name="L681"></a><tt class="py-lineno"> 681</tt>  <tt class="py-line">        <tt class="py-comment"># S2: 32 (16.9 bits)</tt> </tt>
<a name="L682"></a><tt class="py-lineno"> 682</tt>  <tt class="py-line"> </tt>
<a name="L683"></a><tt class="py-lineno"> 683</tt>  <tt class="py-line">        <tt class="py-comment"># Blast 2.2.4 can sometimes skip the whole parameter section.</tt> </tt>
<a name="L684"></a><tt class="py-lineno"> 684</tt>  <tt class="py-line">        <tt class="py-comment"># BLAT also skips the whole parameter section.</tt> </tt>
<a name="L685"></a><tt class="py-lineno"> 685</tt>  <tt class="py-line">        <tt class="py-comment"># Thus, check to make sure that the parameter section really</tt> </tt>
<a name="L686"></a><tt class="py-lineno"> 686</tt>  <tt class="py-line">        <tt class="py-comment"># exists.</tt> </tt>
<a name="L687"></a><tt class="py-lineno"> 687</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">uhandle</tt><tt class="py-op">.</tt><tt id="link-325" class="py-name"><a title="Bio.File.UndoHandle.peekline" class="py-name" href="#" onclick="return doclink('link-325', 'peekline', 'link-65');">peekline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-326" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-326', 'strip', 'link-66');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L688"></a><tt class="py-lineno"> 688</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L689"></a><tt class="py-lineno"> 689</tt>  <tt class="py-line"> </tt>
<a name="L690"></a><tt class="py-lineno"> 690</tt>  <tt class="py-line">        <tt class="py-comment"># BLASTN 2.2.9 looks like it reverses the "Number of Hits" and</tt> </tt>
<a name="L691"></a><tt class="py-lineno"> 691</tt>  <tt class="py-line">        <tt class="py-comment"># "Number of Sequences" lines.</tt> </tt>
<a name="L692"></a><tt class="py-lineno"> 692</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-327" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.start_parameters()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#start_parameters"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.start_parameters" class="py-name" href="#" onclick="return doclink('link-327', 'start_parameters', 'link-327');">start_parameters</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L693"></a><tt class="py-lineno"> 693</tt>  <tt class="py-line"> </tt>
<a name="L694"></a><tt class="py-lineno"> 694</tt>  <tt class="py-line">        <tt class="py-comment"># Matrix line may be missing in BLASTN 2.2.9</tt> </tt>
<a name="L695"></a><tt class="py-lineno"> 695</tt>  <tt class="py-line">        <tt id="link-328" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-328', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-329" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.matrix()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#matrix,Module Bio.motifs.matrix=Bio.motifs.matrix-module.html"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.matrix
Bio.motifs.matrix" class="py-name" href="#" onclick="return doclink('link-329', 'matrix', 'link-329');">matrix</a></tt><tt class="py-op">,</tt> <tt id="link-330" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-330', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'Matrix'</tt><tt class="py-op">)</tt> </tt>
<a name="L696"></a><tt class="py-lineno"> 696</tt>  <tt class="py-line">        <tt class="py-comment"># not TBLASTX</tt> </tt>
<a name="L697"></a><tt class="py-lineno"> 697</tt>  <tt class="py-line">        <tt id="link-331" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-331', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-332" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.gap_penalties()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#gap_penalties"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.gap_penalties" class="py-name" href="#" onclick="return doclink('link-332', 'gap_penalties', 'link-332');">gap_penalties</a></tt><tt class="py-op">,</tt> <tt id="link-333" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-333', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'Gap'</tt><tt class="py-op">)</tt> </tt>
<a name="L698"></a><tt class="py-lineno"> 698</tt>  <tt class="py-line"> </tt>
<a name="L699"></a><tt class="py-lineno"> 699</tt>  <tt class="py-line">        <tt id="link-334" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-334', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-335" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.num_sequences()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#num_sequences"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.num_sequences" class="py-name" href="#" onclick="return doclink('link-335', 'num_sequences', 'link-335');">num_sequences</a></tt><tt class="py-op">,</tt> </tt>
<a name="L700"></a><tt class="py-lineno"> 700</tt>  <tt class="py-line">                              <tt id="link-336" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-336', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'Number of Sequences'</tt><tt class="py-op">)</tt> </tt>
<a name="L701"></a><tt class="py-lineno"> 701</tt>  <tt class="py-line">        <tt id="link-337" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-337', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-338" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.num_hits()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#num_hits"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.num_hits" class="py-name" href="#" onclick="return doclink('link-338', 'num_hits', 'link-338');">num_hits</a></tt><tt class="py-op">,</tt> </tt>
<a name="L702"></a><tt class="py-lineno"> 702</tt>  <tt class="py-line">                      <tt id="link-339" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-339', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'Number of Hits'</tt><tt class="py-op">)</tt> </tt>
<a name="L703"></a><tt class="py-lineno"> 703</tt>  <tt class="py-line">        <tt id="link-340" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-340', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-341" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.num_sequences" class="py-name" href="#" onclick="return doclink('link-341', 'num_sequences', 'link-335');">num_sequences</a></tt><tt class="py-op">,</tt> </tt>
<a name="L704"></a><tt class="py-lineno"> 704</tt>  <tt class="py-line">                              <tt id="link-342" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-342', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'Number of Sequences'</tt><tt class="py-op">)</tt> </tt>
<a name="L705"></a><tt class="py-lineno"> 705</tt>  <tt class="py-line">        <tt id="link-343" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-343', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-344" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.num_extends()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#num_extends"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.num_extends" class="py-name" href="#" onclick="return doclink('link-344', 'num_extends', 'link-344');">num_extends</a></tt><tt class="py-op">,</tt> </tt>
<a name="L706"></a><tt class="py-lineno"> 706</tt>  <tt class="py-line">                      <tt id="link-345" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-345', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'Number of extensions'</tt><tt class="py-op">)</tt> </tt>
<a name="L707"></a><tt class="py-lineno"> 707</tt>  <tt class="py-line">        <tt id="link-346" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-346', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-347" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.num_good_extends()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#num_good_extends"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.num_good_extends" class="py-name" href="#" onclick="return doclink('link-347', 'num_good_extends', 'link-347');">num_good_extends</a></tt><tt class="py-op">,</tt> </tt>
<a name="L708"></a><tt class="py-lineno"> 708</tt>  <tt class="py-line">                      <tt id="link-348" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-348', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'Number of successful'</tt><tt class="py-op">)</tt> </tt>
<a name="L709"></a><tt class="py-lineno"> 709</tt>  <tt class="py-line"> </tt>
<a name="L710"></a><tt class="py-lineno"> 710</tt>  <tt class="py-line">        <tt id="link-349" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-349', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-350" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.num_seqs_better_e()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#num_seqs_better_e"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.num_seqs_better_e" class="py-name" href="#" onclick="return doclink('link-350', 'num_seqs_better_e', 'link-350');">num_seqs_better_e</a></tt><tt class="py-op">,</tt> </tt>
<a name="L711"></a><tt class="py-lineno"> 711</tt>  <tt class="py-line">                      <tt id="link-351" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-351', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'Number of sequences'</tt><tt class="py-op">)</tt> </tt>
<a name="L712"></a><tt class="py-lineno"> 712</tt>  <tt class="py-line"> </tt>
<a name="L713"></a><tt class="py-lineno"> 713</tt>  <tt class="py-line">        <tt class="py-comment"># not BLASTN, TBLASTX</tt> </tt>
<a name="L714"></a><tt class="py-lineno"> 714</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-352" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-352', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-353" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.hsps_no_gap()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#hsps_no_gap"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.hsps_no_gap" class="py-name" href="#" onclick="return doclink('link-353', 'hsps_no_gap', 'link-353');">hsps_no_gap</a></tt><tt class="py-op">,</tt> </tt>
<a name="L715"></a><tt class="py-lineno"> 715</tt>  <tt class="py-line">                                 <tt id="link-354" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-354', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">"Number of HSP's better"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L716"></a><tt class="py-lineno"> 716</tt>  <tt class="py-line">            <tt class="py-comment"># BLASTN 2.2.9</tt> </tt>
<a name="L717"></a><tt class="py-lineno"> 717</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-355" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-355', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-356" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-356', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> </tt>
<a name="L718"></a><tt class="py-lineno"> 718</tt>  <tt class="py-line">                                     <tt id="link-357" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-357', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">"Number of HSP's gapped:"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L719"></a><tt class="py-lineno"> 719</tt>  <tt class="py-line">                <tt id="link-358" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-358', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-359" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-359', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> </tt>
<a name="L720"></a><tt class="py-lineno"> 720</tt>  <tt class="py-line">                              <tt id="link-360" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-360', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">"Number of HSP's successfully"</tt><tt class="py-op">)</tt> </tt>
<a name="L721"></a><tt class="py-lineno"> 721</tt>  <tt class="py-line">                <tt class="py-comment"># This is omitted in 2.2.15</tt> </tt>
<a name="L722"></a><tt class="py-lineno"> 722</tt>  <tt class="py-line">                <tt id="link-361" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-361', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-362" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-362', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> </tt>
<a name="L723"></a><tt class="py-lineno"> 723</tt>  <tt class="py-line">                              <tt id="link-363" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-363', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">"Number of extra gapped extensions"</tt><tt class="py-op">)</tt> </tt>
<a name="L724"></a><tt class="py-lineno"> 724</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L725"></a><tt class="py-lineno"> 725</tt>  <tt class="py-line">                <tt id="link-364" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-364', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-365" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.hsps_prelim_gapped()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#hsps_prelim_gapped"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.hsps_prelim_gapped" class="py-name" href="#" onclick="return doclink('link-365', 'hsps_prelim_gapped', 'link-365');">hsps_prelim_gapped</a></tt><tt class="py-op">,</tt> </tt>
<a name="L726"></a><tt class="py-lineno"> 726</tt>  <tt class="py-line">                              <tt id="link-366" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-366', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">"Number of HSP's successfully"</tt><tt class="py-op">)</tt> </tt>
<a name="L727"></a><tt class="py-lineno"> 727</tt>  <tt class="py-line">                <tt id="link-367" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-367', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt class="py-name">hsps_prelim_gap_attempted</tt><tt class="py-op">,</tt> </tt>
<a name="L728"></a><tt class="py-lineno"> 728</tt>  <tt class="py-line">                              <tt id="link-368" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-368', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">"Number of HSP's that"</tt><tt class="py-op">)</tt> </tt>
<a name="L729"></a><tt class="py-lineno"> 729</tt>  <tt class="py-line">                <tt id="link-369" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-369', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-370" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.hsps_gapped()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#hsps_gapped"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.hsps_gapped" class="py-name" href="#" onclick="return doclink('link-370', 'hsps_gapped', 'link-370');">hsps_gapped</a></tt><tt class="py-op">,</tt> </tt>
<a name="L730"></a><tt class="py-lineno"> 730</tt>  <tt class="py-line">                              <tt id="link-371" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-371', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">"Number of HSP's gapped"</tt><tt class="py-op">)</tt> </tt>
<a name="L731"></a><tt class="py-lineno"> 731</tt>  <tt class="py-line">        <tt class="py-comment"># e.g. BLASTX 2.2.15 where the "better" line is missing</tt> </tt>
<a name="L732"></a><tt class="py-lineno"> 732</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-372" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-372', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-373" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-373', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> </tt>
<a name="L733"></a><tt class="py-lineno"> 733</tt>  <tt class="py-line">                                     <tt id="link-374" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-374', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">"Number of HSP's gapped"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L734"></a><tt class="py-lineno"> 734</tt>  <tt class="py-line">            <tt id="link-375" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-375', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-376" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-376', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> </tt>
<a name="L735"></a><tt class="py-lineno"> 735</tt>  <tt class="py-line">                          <tt id="link-377" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-377', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">"Number of HSP's successfully"</tt><tt class="py-op">)</tt> </tt>
<a name="L736"></a><tt class="py-lineno"> 736</tt>  <tt class="py-line"> </tt>
<a name="L737"></a><tt class="py-lineno"> 737</tt>  <tt class="py-line">        <tt class="py-comment"># not in blastx 2.2.1</tt> </tt>
<a name="L738"></a><tt class="py-lineno"> 738</tt>  <tt class="py-line">        <tt id="link-378" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-378', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-379" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.query_length()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#query_length"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.query_length" class="py-name" href="#" onclick="return doclink('link-379', 'query_length', 'link-379');">query_length</a></tt><tt class="py-op">,</tt> </tt>
<a name="L739"></a><tt class="py-lineno"> 739</tt>  <tt class="py-line">                              <tt class="py-name">has_re</tt><tt class="py-op">=</tt><tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r"[Ll]ength of query"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L740"></a><tt class="py-lineno"> 740</tt>  <tt class="py-line">        <tt class="py-comment"># Not in BLASTX 2.2.22+</tt> </tt>
<a name="L741"></a><tt class="py-lineno"> 741</tt>  <tt class="py-line">        <tt id="link-380" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-380', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-381" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.database_length()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#database_length"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.database_length" class="py-name" href="#" onclick="return doclink('link-381', 'database_length', 'link-381');">database_length</a></tt><tt class="py-op">,</tt> </tt>
<a name="L742"></a><tt class="py-lineno"> 742</tt>  <tt class="py-line">                          <tt class="py-name">has_re</tt><tt class="py-op">=</tt><tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r"[Ll]ength of \s*[Dd]atabase"</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L743"></a><tt class="py-lineno"> 743</tt>  <tt class="py-line"> </tt>
<a name="L744"></a><tt class="py-lineno"> 744</tt>  <tt class="py-line">        <tt class="py-comment"># BLASTN 2.2.9</tt> </tt>
<a name="L745"></a><tt class="py-lineno"> 745</tt>  <tt class="py-line">        <tt id="link-382" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-382', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-383" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer.noevent" class="py-name" href="#" onclick="return doclink('link-383', 'noevent', 'link-21');">noevent</a></tt><tt class="py-op">,</tt> </tt>
<a name="L746"></a><tt class="py-lineno"> 746</tt>  <tt class="py-line">                              <tt id="link-384" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-384', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">"Length adjustment"</tt><tt class="py-op">)</tt> </tt>
<a name="L747"></a><tt class="py-lineno"> 747</tt>  <tt class="py-line">        <tt id="link-385" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-385', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-386" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.effective_hsp_length()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#effective_hsp_length"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.effective_hsp_length" class="py-name" href="#" onclick="return doclink('link-386', 'effective_hsp_length', 'link-386');">effective_hsp_length</a></tt><tt class="py-op">,</tt> </tt>
<a name="L748"></a><tt class="py-lineno"> 748</tt>  <tt class="py-line">                              <tt id="link-387" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-387', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'effective HSP'</tt><tt class="py-op">)</tt> </tt>
<a name="L749"></a><tt class="py-lineno"> 749</tt>  <tt class="py-line">        <tt class="py-comment"># Not in blastx 2.2.1</tt> </tt>
<a name="L750"></a><tt class="py-lineno"> 750</tt>  <tt class="py-line">        <tt id="link-388" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-388', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt> </tt>
<a name="L751"></a><tt class="py-lineno"> 751</tt>  <tt class="py-line">            <tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-389" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.effective_query_length()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#effective_query_length"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.effective_query_length" class="py-name" href="#" onclick="return doclink('link-389', 'effective_query_length', 'link-389');">effective_query_length</a></tt><tt class="py-op">,</tt> </tt>
<a name="L752"></a><tt class="py-lineno"> 752</tt>  <tt class="py-line">            <tt class="py-name">has_re</tt><tt class="py-op">=</tt><tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'[Ee]ffective length of query'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L753"></a><tt class="py-lineno"> 753</tt>  <tt class="py-line"> </tt>
<a name="L754"></a><tt class="py-lineno"> 754</tt>  <tt class="py-line">        <tt class="py-comment"># This is not in BLASTP 2.2.15</tt> </tt>
<a name="L755"></a><tt class="py-lineno"> 755</tt>  <tt class="py-line">        <tt id="link-390" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-390', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt> </tt>
<a name="L756"></a><tt class="py-lineno"> 756</tt>  <tt class="py-line">            <tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-391" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.effective_database_length()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#effective_database_length"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.effective_database_length" class="py-name" href="#" onclick="return doclink('link-391', 'effective_database_length', 'link-391');">effective_database_length</a></tt><tt class="py-op">,</tt> </tt>
<a name="L757"></a><tt class="py-lineno"> 757</tt>  <tt class="py-line">            <tt class="py-name">has_re</tt><tt class="py-op">=</tt><tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'[Ee]ffective length of \s*[Dd]atabase'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L758"></a><tt class="py-lineno"> 758</tt>  <tt class="py-line">        <tt class="py-comment"># Not in blastx 2.2.1, added a ':' to distinguish between</tt> </tt>
<a name="L759"></a><tt class="py-lineno"> 759</tt>  <tt class="py-line">        <tt class="py-comment"># this and the 'effective search space used' line</tt> </tt>
<a name="L760"></a><tt class="py-lineno"> 760</tt>  <tt class="py-line">        <tt id="link-392" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-392', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt> </tt>
<a name="L761"></a><tt class="py-lineno"> 761</tt>  <tt class="py-line">            <tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-393" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.effective_search_space()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#effective_search_space"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.effective_search_space" class="py-name" href="#" onclick="return doclink('link-393', 'effective_search_space', 'link-393');">effective_search_space</a></tt><tt class="py-op">,</tt> </tt>
<a name="L762"></a><tt class="py-lineno"> 762</tt>  <tt class="py-line">            <tt class="py-name">has_re</tt><tt class="py-op">=</tt><tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'[Ee]ffective search space:'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L763"></a><tt class="py-lineno"> 763</tt>  <tt class="py-line">        <tt class="py-comment"># Does not appear in BLASTP 2.0.5</tt> </tt>
<a name="L764"></a><tt class="py-lineno"> 764</tt>  <tt class="py-line">        <tt id="link-394" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-394', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt> </tt>
<a name="L765"></a><tt class="py-lineno"> 765</tt>  <tt class="py-line">            <tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-395" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.effective_search_space_used()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#effective_search_space_used"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.effective_search_space_used" class="py-name" href="#" onclick="return doclink('link-395', 'effective_search_space_used', 'link-395');">effective_search_space_used</a></tt><tt class="py-op">,</tt> </tt>
<a name="L766"></a><tt class="py-lineno"> 766</tt>  <tt class="py-line">            <tt class="py-name">has_re</tt><tt class="py-op">=</tt><tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'[Ee]ffective search space used'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L767"></a><tt class="py-lineno"> 767</tt>  <tt class="py-line"> </tt>
<a name="L768"></a><tt class="py-lineno"> 768</tt>  <tt class="py-line">        <tt class="py-comment"># BLASTX, TBLASTN, TBLASTX</tt> </tt>
<a name="L769"></a><tt class="py-lineno"> 769</tt>  <tt class="py-line">        <tt id="link-396" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-396', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-397" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.frameshift()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#frameshift"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.frameshift" class="py-name" href="#" onclick="return doclink('link-397', 'frameshift', 'link-397');">frameshift</a></tt><tt class="py-op">,</tt> <tt id="link-398" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-398', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'frameshift'</tt><tt class="py-op">)</tt> </tt>
<a name="L770"></a><tt class="py-lineno"> 770</tt>  <tt class="py-line"> </tt>
<a name="L771"></a><tt class="py-lineno"> 771</tt>  <tt class="py-line">        <tt class="py-comment"># not in BLASTN 2.2.9</tt> </tt>
<a name="L772"></a><tt class="py-lineno"> 772</tt>  <tt class="py-line">        <tt id="link-399" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-399', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-400" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.threshold()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#threshold"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.threshold" class="py-name" href="#" onclick="return doclink('link-400', 'threshold', 'link-400');">threshold</a></tt><tt class="py-op">,</tt> <tt id="link-401" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-401', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'T'</tt><tt class="py-op">)</tt> </tt>
<a name="L773"></a><tt class="py-lineno"> 773</tt>  <tt class="py-line">        <tt class="py-comment"># In BLASTX 2.2.15 replaced by: "Neighboring words threshold: 12"</tt> </tt>
<a name="L774"></a><tt class="py-lineno"> 774</tt>  <tt class="py-line">        <tt id="link-402" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-402', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-403" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.threshold" class="py-name" href="#" onclick="return doclink('link-403', 'threshold', 'link-400');">threshold</a></tt><tt class="py-op">,</tt> <tt id="link-404" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-404', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'Neighboring words threshold'</tt><tt class="py-op">)</tt> </tt>
<a name="L775"></a><tt class="py-lineno"> 775</tt>  <tt class="py-line"> </tt>
<a name="L776"></a><tt class="py-lineno"> 776</tt>  <tt class="py-line">        <tt class="py-comment"># not in BLASTX 2.2.15</tt> </tt>
<a name="L777"></a><tt class="py-lineno"> 777</tt>  <tt class="py-line">        <tt id="link-405" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-405', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-406" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.window_size()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#window_size"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.window_size" class="py-name" href="#" onclick="return doclink('link-406', 'window_size', 'link-406');">window_size</a></tt><tt class="py-op">,</tt> <tt id="link-407" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-407', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'A'</tt><tt class="py-op">)</tt> </tt>
<a name="L778"></a><tt class="py-lineno"> 778</tt>  <tt class="py-line">        <tt class="py-comment"># get this instead: "Window for multiple hits: 40"</tt> </tt>
<a name="L779"></a><tt class="py-lineno"> 779</tt>  <tt class="py-line">        <tt id="link-408" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-408', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-409" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.window_size" class="py-name" href="#" onclick="return doclink('link-409', 'window_size', 'link-406');">window_size</a></tt><tt class="py-op">,</tt> <tt id="link-410" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-410', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'Window for multiple hits'</tt><tt class="py-op">)</tt> </tt>
<a name="L780"></a><tt class="py-lineno"> 780</tt>  <tt class="py-line"> </tt>
<a name="L781"></a><tt class="py-lineno"> 781</tt>  <tt class="py-line">        <tt class="py-comment"># not in BLASTX 2.2.22+</tt> </tt>
<a name="L782"></a><tt class="py-lineno"> 782</tt>  <tt class="py-line">        <tt id="link-411" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-411', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-412" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.dropoff_1st_pass()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#dropoff_1st_pass"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.dropoff_1st_pass" class="py-name" href="#" onclick="return doclink('link-412', 'dropoff_1st_pass', 'link-412');">dropoff_1st_pass</a></tt><tt class="py-op">,</tt> <tt id="link-413" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-413', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'X1'</tt><tt class="py-op">)</tt> </tt>
<a name="L783"></a><tt class="py-lineno"> 783</tt>  <tt class="py-line">        <tt class="py-comment"># not TBLASTN</tt> </tt>
<a name="L784"></a><tt class="py-lineno"> 784</tt>  <tt class="py-line">        <tt id="link-414" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-414', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-415" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.gap_x_dropoff()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#gap_x_dropoff"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.gap_x_dropoff" class="py-name" href="#" onclick="return doclink('link-415', 'gap_x_dropoff', 'link-415');">gap_x_dropoff</a></tt><tt class="py-op">,</tt> <tt id="link-416" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-416', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'X2'</tt><tt class="py-op">)</tt> </tt>
<a name="L785"></a><tt class="py-lineno"> 785</tt>  <tt class="py-line"> </tt>
<a name="L786"></a><tt class="py-lineno"> 786</tt>  <tt class="py-line">        <tt class="py-comment"># not BLASTN, TBLASTX</tt> </tt>
<a name="L787"></a><tt class="py-lineno"> 787</tt>  <tt class="py-line">        <tt id="link-417" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-417', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-418" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.gap_x_dropoff_final()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#gap_x_dropoff_final"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.gap_x_dropoff_final" class="py-name" href="#" onclick="return doclink('link-418', 'gap_x_dropoff_final', 'link-418');">gap_x_dropoff_final</a></tt><tt class="py-op">,</tt> </tt>
<a name="L788"></a><tt class="py-lineno"> 788</tt>  <tt class="py-line">                              <tt id="link-419" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-419', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'X3'</tt><tt class="py-op">)</tt> </tt>
<a name="L789"></a><tt class="py-lineno"> 789</tt>  <tt class="py-line"> </tt>
<a name="L790"></a><tt class="py-lineno"> 790</tt>  <tt class="py-line">        <tt class="py-comment"># not TBLASTN</tt> </tt>
<a name="L791"></a><tt class="py-lineno"> 791</tt>  <tt class="py-line">        <tt id="link-420" class="py-name"><a title="Bio.ParserSupport.attempt_read_and_call" class="py-name" href="#" onclick="return doclink('link-420', 'attempt_read_and_call', 'link-33');">attempt_read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-421" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.gap_trigger()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#gap_trigger"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.gap_trigger" class="py-name" href="#" onclick="return doclink('link-421', 'gap_trigger', 'link-421');">gap_trigger</a></tt><tt class="py-op">,</tt> <tt id="link-422" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-422', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'S1'</tt><tt class="py-op">)</tt> </tt>
<a name="L792"></a><tt class="py-lineno"> 792</tt>  <tt class="py-line">        <tt class="py-comment"># not in blastx 2.2.1</tt> </tt>
<a name="L793"></a><tt class="py-lineno"> 793</tt>  <tt class="py-line">        <tt class="py-comment"># first we make sure we have additional lines to work with, if</tt> </tt>
<a name="L794"></a><tt class="py-lineno"> 794</tt>  <tt class="py-line">        <tt class="py-comment"># not then the file is done and we don't have a final S2</tt> </tt>
<a name="L795"></a><tt class="py-lineno"> 795</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-423" class="py-name"><a title="Bio.ParserSupport.is_blank_line" class="py-name" href="#" onclick="return doclink('link-423', 'is_blank_line', 'link-40');">is_blank_line</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">.</tt><tt id="link-424" class="py-name"><a title="Bio.File.UndoHandle.peekline" class="py-name" href="#" onclick="return doclink('link-424', 'peekline', 'link-65');">peekline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">allow_spaces</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L796"></a><tt class="py-lineno"> 796</tt>  <tt class="py-line">            <tt id="link-425" class="py-name"><a title="Bio.ParserSupport.read_and_call" class="py-name" href="#" onclick="return doclink('link-425', 'read_and_call', 'link-28');">read_and_call</a></tt><tt class="py-op">(</tt><tt class="py-name">uhandle</tt><tt class="py-op">,</tt> <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-426" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.blast_cutoff()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#blast_cutoff"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.blast_cutoff" class="py-name" href="#" onclick="return doclink('link-426', 'blast_cutoff', 'link-426');">blast_cutoff</a></tt><tt class="py-op">,</tt> <tt id="link-427" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-427', 'start', 'link-35');">start</a></tt><tt class="py-op">=</tt><tt class="py-string">'S2'</tt><tt class="py-op">)</tt> </tt>
<a name="L797"></a><tt class="py-lineno"> 797</tt>  <tt class="py-line"> </tt>
<a name="L798"></a><tt class="py-lineno"> 798</tt>  <tt class="py-line">        <tt class="py-name">consumer</tt><tt class="py-op">.</tt><tt id="link-428" class="py-name" targets="Method Bio.Blast.NCBIStandalone._BlastConsumer.end_parameters()=Bio.Blast.NCBIStandalone._BlastConsumer-class.html#end_parameters,Method Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_parameters()=Bio.Blast.NCBIStandalone._PSIBlastConsumer-class.html#end_parameters,Method Bio.Blast.NCBIStandalone._ParametersConsumer.end_parameters()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#end_parameters"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.end_parameters
Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_parameters
Bio.Blast.NCBIStandalone._ParametersConsumer.end_parameters" class="py-name" href="#" onclick="return doclink('link-428', 'end_parameters', 'link-428');">end_parameters</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L799"></a><tt class="py-lineno"> 799</tt>  <tt class="py-line"> </tt>
<a name="L800"></a><tt class="py-lineno"> 800</tt>  <tt class="py-line"> </tt>
<a name="BlastParser"></a><div id="BlastParser-def"><a name="L801"></a><tt class="py-lineno"> 801</tt> <a class="py-toggle" href="#" id="BlastParser-toggle" onclick="return toggle('BlastParser');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone.BlastParser-class.html">BlastParser</a><tt class="py-op">(</tt><tt class="py-base-class">AbstractParser</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastParser-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="BlastParser-expanded"><a name="L802"></a><tt class="py-lineno"> 802</tt>  <tt class="py-line">    <tt class="py-docstring">"""Parses BLAST data into a Record.Blast object.</tt> </tt>
<a name="L803"></a><tt class="py-lineno"> 803</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L804"></a><tt class="py-lineno"> 804</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="BlastParser.__init__"></a><div id="BlastParser.__init__-def"><a name="L805"></a><tt class="py-lineno"> 805</tt> <a class="py-toggle" href="#" id="BlastParser.__init__-toggle" onclick="return toggle('BlastParser.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone.BlastParser-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastParser.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="BlastParser.__init__-expanded"><a name="L806"></a><tt class="py-lineno"> 806</tt>  <tt class="py-line">        <tt class="py-docstring">"""__init__(self)"""</tt> </tt>
<a name="L807"></a><tt class="py-lineno"> 807</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_scanner</tt> <tt class="py-op">=</tt> <tt id="link-429" class="py-name" targets="Class Bio.Blast.NCBIStandalone._Scanner=Bio.Blast.NCBIStandalone._Scanner-class.html"><a title="Bio.Blast.NCBIStandalone._Scanner" class="py-name" href="#" onclick="return doclink('link-429', '_Scanner', 'link-429');">_Scanner</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L808"></a><tt class="py-lineno"> 808</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_consumer</tt> <tt class="py-op">=</tt> <tt id="link-430" class="py-name" targets="Class Bio.Blast.NCBIStandalone._BlastConsumer=Bio.Blast.NCBIStandalone._BlastConsumer-class.html"><a title="Bio.Blast.NCBIStandalone._BlastConsumer" class="py-name" href="#" onclick="return doclink('link-430', '_BlastConsumer', 'link-430');">_BlastConsumer</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L809"></a><tt class="py-lineno"> 809</tt>  <tt class="py-line"> </tt>
<a name="BlastParser.parse"></a><div id="BlastParser.parse-def"><a name="L810"></a><tt class="py-lineno"> 810</tt> <a class="py-toggle" href="#" id="BlastParser.parse-toggle" onclick="return toggle('BlastParser.parse');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone.BlastParser-class.html#parse">parse</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastParser.parse-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="BlastParser.parse-expanded"><a name="L811"></a><tt class="py-lineno"> 811</tt>  <tt class="py-line">        <tt class="py-docstring">"""parse(self, handle)"""</tt> </tt>
<a name="L812"></a><tt class="py-lineno"> 812</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_scanner</tt><tt class="py-op">.</tt><tt id="link-431" class="py-name" targets="Method Bio.Blast.NCBIStandalone._Scanner.feed()=Bio.Blast.NCBIStandalone._Scanner-class.html#feed,Method Bio.ExPASy.ScanProsite.Parser.feed()=Bio.ExPASy.ScanProsite.Parser-class.html#feed,Method Bio.GenBank.Scanner.InsdcScanner.feed()=Bio.GenBank.Scanner.InsdcScanner-class.html#feed"><a title="Bio.Blast.NCBIStandalone._Scanner.feed
Bio.ExPASy.ScanProsite.Parser.feed
Bio.GenBank.Scanner.InsdcScanner.feed" class="py-name" href="#" onclick="return doclink('link-431', 'feed', 'link-431');">feed</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_consumer</tt><tt class="py-op">)</tt> </tt>
<a name="L813"></a><tt class="py-lineno"> 813</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_consumer</tt><tt class="py-op">.</tt><tt id="link-432" class="py-name" targets="Variable BioSQL.BioSeq.DBSeq.data=BioSQL.BioSeq.DBSeq-class.html#data"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-432', 'data', 'link-432');">data</a></tt> </tt>
</div></div><a name="L814"></a><tt class="py-lineno"> 814</tt>  <tt class="py-line"> </tt>
<a name="L815"></a><tt class="py-lineno"> 815</tt>  <tt class="py-line"> </tt>
<a name="PSIBlastParser"></a><div id="PSIBlastParser-def"><a name="L816"></a><tt class="py-lineno"> 816</tt> <a class="py-toggle" href="#" id="PSIBlastParser-toggle" onclick="return toggle('PSIBlastParser');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone.PSIBlastParser-class.html">PSIBlastParser</a><tt class="py-op">(</tt><tt class="py-base-class">AbstractParser</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PSIBlastParser-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="PSIBlastParser-expanded"><a name="L817"></a><tt class="py-lineno"> 817</tt>  <tt class="py-line">    <tt class="py-docstring">"""Parses BLAST data into a Record.PSIBlast object.</tt> </tt>
<a name="L818"></a><tt class="py-lineno"> 818</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L819"></a><tt class="py-lineno"> 819</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="PSIBlastParser.__init__"></a><div id="PSIBlastParser.__init__-def"><a name="L820"></a><tt class="py-lineno"> 820</tt> <a class="py-toggle" href="#" id="PSIBlastParser.__init__-toggle" onclick="return toggle('PSIBlastParser.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone.PSIBlastParser-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PSIBlastParser.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="PSIBlastParser.__init__-expanded"><a name="L821"></a><tt class="py-lineno"> 821</tt>  <tt class="py-line">        <tt class="py-docstring">"""__init__(self)"""</tt> </tt>
<a name="L822"></a><tt class="py-lineno"> 822</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_scanner</tt> <tt class="py-op">=</tt> <tt id="link-433" class="py-name"><a title="Bio.Blast.NCBIStandalone._Scanner" class="py-name" href="#" onclick="return doclink('link-433', '_Scanner', 'link-429');">_Scanner</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L823"></a><tt class="py-lineno"> 823</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_consumer</tt> <tt class="py-op">=</tt> <tt id="link-434" class="py-name" targets="Class Bio.Blast.NCBIStandalone._PSIBlastConsumer=Bio.Blast.NCBIStandalone._PSIBlastConsumer-class.html"><a title="Bio.Blast.NCBIStandalone._PSIBlastConsumer" class="py-name" href="#" onclick="return doclink('link-434', '_PSIBlastConsumer', 'link-434');">_PSIBlastConsumer</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L824"></a><tt class="py-lineno"> 824</tt>  <tt class="py-line"> </tt>
<a name="PSIBlastParser.parse"></a><div id="PSIBlastParser.parse-def"><a name="L825"></a><tt class="py-lineno"> 825</tt> <a class="py-toggle" href="#" id="PSIBlastParser.parse-toggle" onclick="return toggle('PSIBlastParser.parse');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone.PSIBlastParser-class.html#parse">parse</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PSIBlastParser.parse-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="PSIBlastParser.parse-expanded"><a name="L826"></a><tt class="py-lineno"> 826</tt>  <tt class="py-line">        <tt class="py-docstring">"""parse(self, handle)"""</tt> </tt>
<a name="L827"></a><tt class="py-lineno"> 827</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_scanner</tt><tt class="py-op">.</tt><tt id="link-435" class="py-name"><a title="Bio.Blast.NCBIStandalone._Scanner.feed
Bio.ExPASy.ScanProsite.Parser.feed
Bio.GenBank.Scanner.InsdcScanner.feed" class="py-name" href="#" onclick="return doclink('link-435', 'feed', 'link-431');">feed</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_consumer</tt><tt class="py-op">)</tt> </tt>
<a name="L828"></a><tt class="py-lineno"> 828</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_consumer</tt><tt class="py-op">.</tt><tt id="link-436" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-436', 'data', 'link-432');">data</a></tt> </tt>
</div></div><a name="L829"></a><tt class="py-lineno"> 829</tt>  <tt class="py-line"> </tt>
<a name="L830"></a><tt class="py-lineno"> 830</tt>  <tt class="py-line"> </tt>
<a name="_HeaderConsumer"></a><div id="_HeaderConsumer-def"><a name="L831"></a><tt class="py-lineno"> 831</tt> <a class="py-toggle" href="#" id="_HeaderConsumer-toggle" onclick="return toggle('_HeaderConsumer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._HeaderConsumer-class.html">_HeaderConsumer</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_HeaderConsumer-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_HeaderConsumer-expanded"><a name="_HeaderConsumer.start_header"></a><div id="_HeaderConsumer.start_header-def"><a name="L832"></a><tt class="py-lineno"> 832</tt> <a class="py-toggle" href="#" id="_HeaderConsumer.start_header-toggle" onclick="return toggle('_HeaderConsumer.start_header');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._HeaderConsumer-class.html#start_header">start_header</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_HeaderConsumer.start_header-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_HeaderConsumer.start_header-expanded"><a name="L833"></a><tt class="py-lineno"> 833</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-437" class="py-name" targets="Variable Bio.AlignIO.ClustalIO.ClustalIterator._header=Bio.AlignIO.ClustalIO.ClustalIterator-class.html#_header,Variable Bio.AlignIO.EmbossIO.EmbossIterator._header=Bio.AlignIO.EmbossIO.EmbossIterator-class.html#_header,Variable Bio.AlignIO.PhylipIO.PhylipIterator._header=Bio.AlignIO.PhylipIO.PhylipIterator-class.html#_header,Variable Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header=Bio.AlignIO.PhylipIO.SequentialPhylipIterator-class.html#_header,Variable Bio.AlignIO.StockholmIO.StockholmIterator._header=Bio.AlignIO.StockholmIO.StockholmIterator-class.html#_header"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-437', '_header', 'link-437');">_header</a></tt> <tt class="py-op">=</tt> <tt id="link-438" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-438', 'Record', 'link-11');">Record</a></tt><tt class="py-op">.</tt><tt id="link-439" class="py-name" targets="Class Bio.Blast.Record.Header=Bio.Blast.Record.Header-class.html"><a title="Bio.Blast.Record.Header" class="py-name" href="#" onclick="return doclink('link-439', 'Header', 'link-439');">Header</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L834"></a><tt class="py-lineno"> 834</tt>  <tt class="py-line"> </tt>
<a name="_HeaderConsumer.version"></a><div id="_HeaderConsumer.version-def"><a name="L835"></a><tt class="py-lineno"> 835</tt> <a class="py-toggle" href="#" id="_HeaderConsumer.version-toggle" onclick="return toggle('_HeaderConsumer.version');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._HeaderConsumer-class.html#version">version</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_HeaderConsumer.version-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_HeaderConsumer.version-expanded"><a name="L836"></a><tt class="py-lineno"> 836</tt>  <tt class="py-line">        <tt class="py-name">c</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-440" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-440', 'split', 'link-440');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L837"></a><tt class="py-lineno"> 837</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-441" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-441', '_header', 'link-437');">_header</a></tt><tt class="py-op">.</tt><tt class="py-name">application</tt> <tt class="py-op">=</tt> <tt class="py-name">c</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L838"></a><tt class="py-lineno"> 838</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-442" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-442', '_header', 'link-437');">_header</a></tt><tt class="py-op">.</tt><tt id="link-443" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-443', 'version', 'link-29');">version</a></tt> <tt class="py-op">=</tt> <tt class="py-name">c</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L839"></a><tt class="py-lineno"> 839</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">c</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> </tt>
<a name="L840"></a><tt class="py-lineno"> 840</tt>  <tt class="py-line">            <tt class="py-comment"># The date is missing in the new C++ output from blastx 2.2.22+</tt> </tt>
<a name="L841"></a><tt class="py-lineno"> 841</tt>  <tt class="py-line">            <tt class="py-comment"># Just get "BLASTX 2.2.22+\n" and that's all.</tt> </tt>
<a name="L842"></a><tt class="py-lineno"> 842</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-444" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-444', '_header', 'link-437');">_header</a></tt><tt class="py-op">.</tt><tt id="link-445" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.date()=Bio.GenBank._FeatureConsumer-class.html#date,Method Bio.GenBank._RecordConsumer.date()=Bio.GenBank._RecordConsumer-class.html#date,Method Bio.Phylo.PhyloXMLIO.Parser.date()=Bio.Phylo.PhyloXMLIO.Parser-class.html#date,Method Bio.Phylo.PhyloXMLIO.Writer.date()=Bio.Phylo.PhyloXMLIO.Writer-class.html#date"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-445', 'date', 'link-445');">date</a></tt> <tt class="py-op">=</tt> <tt class="py-name">c</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
</div><a name="L843"></a><tt class="py-lineno"> 843</tt>  <tt class="py-line"> </tt>
<a name="_HeaderConsumer.reference"></a><div id="_HeaderConsumer.reference-def"><a name="L844"></a><tt class="py-lineno"> 844</tt> <a class="py-toggle" href="#" id="_HeaderConsumer.reference-toggle" onclick="return toggle('_HeaderConsumer.reference');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._HeaderConsumer-class.html#reference">reference</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_HeaderConsumer.reference-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_HeaderConsumer.reference-expanded"><a name="L845"></a><tt class="py-lineno"> 845</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-446" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-446', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Reference: '</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L846"></a><tt class="py-lineno"> 846</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-447" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-447', '_header', 'link-437');">_header</a></tt><tt class="py-op">.</tt><tt id="link-448" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-448', 'reference', 'link-37');">reference</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">11</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L847"></a><tt class="py-lineno"> 847</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L848"></a><tt class="py-lineno"> 848</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-449" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-449', '_header', 'link-437');">_header</a></tt><tt class="py-op">.</tt><tt id="link-450" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-450', 'reference', 'link-37');">reference</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-451" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-451', '_header', 'link-437');">_header</a></tt><tt class="py-op">.</tt><tt id="link-452" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-452', 'reference', 'link-37');">reference</a></tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt> </tt>
</div><a name="L849"></a><tt class="py-lineno"> 849</tt>  <tt class="py-line"> </tt>
<a name="_HeaderConsumer.query_info"></a><div id="_HeaderConsumer.query_info-def"><a name="L850"></a><tt class="py-lineno"> 850</tt> <a class="py-toggle" href="#" id="_HeaderConsumer.query_info-toggle" onclick="return toggle('_HeaderConsumer.query_info');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._HeaderConsumer-class.html#query_info">query_info</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_HeaderConsumer.query_info-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_HeaderConsumer.query_info-expanded"><a name="L851"></a><tt class="py-lineno"> 851</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-453" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-453', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Query= '</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L852"></a><tt class="py-lineno"> 852</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-454" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-454', '_header', 'link-437');">_header</a></tt><tt class="py-op">.</tt><tt id="link-455" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.query
Bio.SearchIO._model.hsp.HSP.query
Bio.SearchIO._model.hsp.HSPFragment.query
Bio.SearchIO._model.query" class="py-name" href="#" onclick="return doclink('link-455', 'query', 'link-239');">query</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-456" class="py-name"><a title="Bio.Seq.Seq.lstrip" class="py-name" href="#" onclick="return doclink('link-456', 'lstrip', 'link-205');">lstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L853"></a><tt class="py-lineno"> 853</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-457" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-457', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Length='</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L854"></a><tt class="py-lineno"> 854</tt>  <tt class="py-line">            <tt class="py-comment"># New style way to give the query length in BLAST 2.2.22+ (the C++ code)</tt> </tt>
<a name="L855"></a><tt class="py-lineno"> 855</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-458" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-458', '_header', 'link-437');">_header</a></tt><tt class="py-op">.</tt><tt class="py-name">query_letters</tt> <tt class="py-op">=</tt> <tt id="link-459" class="py-name" targets="Function Bio.Blast.NCBIStandalone._safe_int()=Bio.Blast.NCBIStandalone-module.html#_safe_int"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-459', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">7</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-460" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-460', 'strip', 'link-66');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L856"></a><tt class="py-lineno"> 856</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-461" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-461', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'       '</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># continuation of query_info</tt> </tt>
<a name="L857"></a><tt class="py-lineno"> 857</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-462" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-462', '_header', 'link-437');">_header</a></tt><tt class="py-op">.</tt><tt id="link-463" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.query
Bio.SearchIO._model.hsp.HSP.query
Bio.SearchIO._model.hsp.HSPFragment.query
Bio.SearchIO._model.query" class="py-name" href="#" onclick="return doclink('link-463', 'query', 'link-239');">query</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"%s%s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-464" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-464', '_header', 'link-437');">_header</a></tt><tt class="py-op">.</tt><tt id="link-465" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.query
Bio.SearchIO._model.hsp.HSP.query
Bio.SearchIO._model.hsp.HSPFragment.query
Bio.SearchIO._model.query" class="py-name" href="#" onclick="return doclink('link-465', 'query', 'link-239');">query</a></tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L858"></a><tt class="py-lineno"> 858</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L859"></a><tt class="py-lineno"> 859</tt>  <tt class="py-line">            <tt class="py-comment"># Hope it is the old style way to give the query length:</tt> </tt>
<a name="L860"></a><tt class="py-lineno"> 860</tt>  <tt class="py-line">            <tt id="link-466" class="py-name" targets="Variable Bio.Alphabet.Alphabet.letters=Bio.Alphabet.Alphabet-class.html#letters,Variable Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters=Bio.Alphabet.IUPAC.ExtendedIUPACDNA-class.html#letters,Variable Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters=Bio.Alphabet.IUPAC.ExtendedIUPACProtein-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters=Bio.Alphabet.IUPAC.IUPACAmbiguousDNA-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters=Bio.Alphabet.IUPAC.IUPACAmbiguousRNA-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACProtein.letters=Bio.Alphabet.IUPAC.IUPACProtein-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters=Bio.Alphabet.IUPAC.IUPACUnambiguousDNA-class.html#letters,Variable Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters=Bio.Alphabet.IUPAC.IUPACUnambiguousRNA-class.html#letters,Variable Bio.Alphabet.Reduced.HPModel.letters=Bio.Alphabet.Reduced.HPModel-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy10.letters=Bio.Alphabet.Reduced.Murphy10-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy15.letters=Bio.Alphabet.Reduced.Murphy15-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy4.letters=Bio.Alphabet.Reduced.Murphy4-class.html#letters,Variable Bio.Alphabet.Reduced.Murphy8.letters=Bio.Alphabet.Reduced.Murphy8-class.html#letters,Variable Bio.Alphabet.Reduced.PC5.letters=Bio.Alphabet.Reduced.PC5-class.html#letters,Variable Bio.Alphabet.SecondaryStructure.letters=Bio.Alphabet.SecondaryStructure-class.html#letters,Variable Bio.Alphabet.SingleLetterAlphabet.letters=Bio.Alphabet.SingleLetterAlphabet-class.html#letters,Variable Bio.Alphabet.ThreeLetterProtein.letters=Bio.Alphabet.ThreeLetterProtein-class.html#letters,Variable Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet.letters=Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet-class.html#letters,Variable Bio.codonalign.codonalphabet.CodonAlphabet.letters=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#letters"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-466', 'letters', 'link-466');">letters</a></tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-467" class="py-name" targets="Function Bio.Blast.NCBIStandalone._re_search()=Bio.Blast.NCBIStandalone-module.html#_re_search"><a title="Bio.Blast.NCBIStandalone._re_search" class="py-name" href="#" onclick="return doclink('link-467', '_re_search', 'link-467');">_re_search</a></tt><tt class="py-op">(</tt> </tt>
<a name="L861"></a><tt class="py-lineno"> 861</tt>  <tt class="py-line">                <tt class="py-string">r"([0-9,]+) letters"</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">,</tt> </tt>
<a name="L862"></a><tt class="py-lineno"> 862</tt>  <tt class="py-line">                <tt class="py-string">"I could not find the number of letters in line\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L863"></a><tt class="py-lineno"> 863</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-468" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-468', '_header', 'link-437');">_header</a></tt><tt class="py-op">.</tt><tt class="py-name">query_letters</tt> <tt class="py-op">=</tt> <tt id="link-469" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-469', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-470" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-470', 'letters', 'link-466');">letters</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L864"></a><tt class="py-lineno"> 864</tt>  <tt class="py-line"> </tt>
<a name="_HeaderConsumer.database_info"></a><div id="_HeaderConsumer.database_info-def"><a name="L865"></a><tt class="py-lineno"> 865</tt> <a class="py-toggle" href="#" id="_HeaderConsumer.database_info-toggle" onclick="return toggle('_HeaderConsumer.database_info');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._HeaderConsumer-class.html#database_info">database_info</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_HeaderConsumer.database_info-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_HeaderConsumer.database_info-expanded"><a name="L866"></a><tt class="py-lineno"> 866</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-471" class="py-name" targets="Method Bio.Seq.Seq.rstrip()=Bio.Seq.Seq-class.html#rstrip"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-471', 'rstrip', 'link-471');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L867"></a><tt class="py-lineno"> 867</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-472" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-472', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Database: '</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L868"></a><tt class="py-lineno"> 868</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-473" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-473', '_header', 'link-437');">_header</a></tt><tt class="py-op">.</tt><tt id="link-474" class="py-name"><a title="Bio.Blast.NCBIStandalone._DatabaseReportConsumer.database" class="py-name" href="#" onclick="return doclink('link-474', 'database', 'link-281');">database</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">10</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L869"></a><tt class="py-lineno"> 869</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-475" class="py-name" targets="Method Bio.Seq.Seq.endswith()=Bio.Seq.Seq-class.html#endswith"><a title="Bio.Seq.Seq.endswith" class="py-name" href="#" onclick="return doclink('link-475', 'endswith', 'link-475');">endswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'total letters'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L870"></a><tt class="py-lineno"> 870</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-476" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-476', '_header', 'link-437');">_header</a></tt><tt class="py-op">.</tt><tt id="link-477" class="py-name"><a title="Bio.Blast.NCBIStandalone._DatabaseReportConsumer.database" class="py-name" href="#" onclick="return doclink('link-477', 'database', 'link-281');">database</a></tt><tt class="py-op">:</tt> </tt>
<a name="L871"></a><tt class="py-lineno"> 871</tt>  <tt class="py-line">                <tt class="py-comment"># Need to include a space when merging multi line datase descr</tt> </tt>
<a name="L872"></a><tt class="py-lineno"> 872</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-478" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-478', '_header', 'link-437');">_header</a></tt><tt class="py-op">.</tt><tt id="link-479" class="py-name"><a title="Bio.Blast.NCBIStandalone._DatabaseReportConsumer.database" class="py-name" href="#" onclick="return doclink('link-479', 'database', 'link-281');">database</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-480" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-480', '_header', 'link-437');">_header</a></tt><tt class="py-op">.</tt><tt id="link-481" class="py-name"><a title="Bio.Blast.NCBIStandalone._DatabaseReportConsumer.database" class="py-name" href="#" onclick="return doclink('link-481', 'database', 'link-281');">database</a></tt> <tt class="py-op">+</tt> <tt class="py-string">" "</tt> <tt class="py-op">+</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-482" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-482', 'strip', 'link-66');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L873"></a><tt class="py-lineno"> 873</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L874"></a><tt class="py-lineno"> 874</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-483" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-483', '_header', 'link-437');">_header</a></tt><tt class="py-op">.</tt><tt id="link-484" class="py-name"><a title="Bio.Blast.NCBIStandalone._DatabaseReportConsumer.database" class="py-name" href="#" onclick="return doclink('link-484', 'database', 'link-281');">database</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-485" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-485', 'strip', 'link-66');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L875"></a><tt class="py-lineno"> 875</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L876"></a><tt class="py-lineno"> 876</tt>  <tt class="py-line">            <tt class="py-name">sequences</tt><tt class="py-op">,</tt> <tt id="link-486" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-486', 'letters', 'link-466');">letters</a></tt> <tt class="py-op">=</tt> <tt id="link-487" class="py-name"><a title="Bio.Blast.NCBIStandalone._re_search" class="py-name" href="#" onclick="return doclink('link-487', '_re_search', 'link-467');">_re_search</a></tt><tt class="py-op">(</tt> </tt>
<a name="L877"></a><tt class="py-lineno"> 877</tt>  <tt class="py-line">                <tt class="py-string">r"([0-9,]+) sequences; ([0-9,-]+) total letters"</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">,</tt> </tt>
<a name="L878"></a><tt class="py-lineno"> 878</tt>  <tt class="py-line">                <tt class="py-string">"I could not find the sequences and letters in line\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L879"></a><tt class="py-lineno"> 879</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-488" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-488', '_header', 'link-437');">_header</a></tt><tt class="py-op">.</tt><tt class="py-name">database_sequences</tt> <tt class="py-op">=</tt> <tt id="link-489" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-489', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt class="py-name">sequences</tt><tt class="py-op">)</tt> </tt>
<a name="L880"></a><tt class="py-lineno"> 880</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-490" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-490', '_header', 'link-437');">_header</a></tt><tt class="py-op">.</tt><tt class="py-name">database_letters</tt> <tt class="py-op">=</tt> <tt id="link-491" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-491', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-492" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-492', 'letters', 'link-466');">letters</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L881"></a><tt class="py-lineno"> 881</tt>  <tt class="py-line"> </tt>
<a name="_HeaderConsumer.end_header"></a><div id="_HeaderConsumer.end_header-def"><a name="L882"></a><tt class="py-lineno"> 882</tt> <a class="py-toggle" href="#" id="_HeaderConsumer.end_header-toggle" onclick="return toggle('_HeaderConsumer.end_header');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._HeaderConsumer-class.html#end_header">end_header</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_HeaderConsumer.end_header-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_HeaderConsumer.end_header-expanded"><a name="L883"></a><tt class="py-lineno"> 883</tt>  <tt class="py-line">        <tt class="py-comment"># Get rid of the trailing newlines</tt> </tt>
<a name="L884"></a><tt class="py-lineno"> 884</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-493" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-493', '_header', 'link-437');">_header</a></tt><tt class="py-op">.</tt><tt id="link-494" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-494', 'reference', 'link-37');">reference</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-495" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-495', '_header', 'link-437');">_header</a></tt><tt class="py-op">.</tt><tt id="link-496" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-496', 'reference', 'link-37');">reference</a></tt><tt class="py-op">.</tt><tt id="link-497" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-497', 'rstrip', 'link-471');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L885"></a><tt class="py-lineno"> 885</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-498" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-498', '_header', 'link-437');">_header</a></tt><tt class="py-op">.</tt><tt id="link-499" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.query
Bio.SearchIO._model.hsp.HSP.query
Bio.SearchIO._model.hsp.HSPFragment.query
Bio.SearchIO._model.query" class="py-name" href="#" onclick="return doclink('link-499', 'query', 'link-239');">query</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-500" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-500', '_header', 'link-437');">_header</a></tt><tt class="py-op">.</tt><tt id="link-501" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.query
Bio.SearchIO._model.hsp.HSP.query
Bio.SearchIO._model.hsp.HSPFragment.query
Bio.SearchIO._model.query" class="py-name" href="#" onclick="return doclink('link-501', 'query', 'link-239');">query</a></tt><tt class="py-op">.</tt><tt id="link-502" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-502', 'rstrip', 'link-471');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L886"></a><tt class="py-lineno"> 886</tt>  <tt class="py-line"> </tt>
<a name="L887"></a><tt class="py-lineno"> 887</tt>  <tt class="py-line"> </tt>
<a name="_DescriptionConsumer"></a><div id="_DescriptionConsumer-def"><a name="L888"></a><tt class="py-lineno"> 888</tt> <a class="py-toggle" href="#" id="_DescriptionConsumer-toggle" onclick="return toggle('_DescriptionConsumer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html">_DescriptionConsumer</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_DescriptionConsumer-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_DescriptionConsumer-expanded"><a name="_DescriptionConsumer.start_descriptions"></a><div id="_DescriptionConsumer.start_descriptions-def"><a name="L889"></a><tt class="py-lineno"> 889</tt> <a class="py-toggle" href="#" id="_DescriptionConsumer.start_descriptions-toggle" onclick="return toggle('_DescriptionConsumer.start_descriptions');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#start_descriptions">start_descriptions</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_DescriptionConsumer.start_descriptions-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_DescriptionConsumer.start_descriptions-expanded"><a name="L890"></a><tt class="py-lineno"> 890</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_descriptions</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L891"></a><tt class="py-lineno"> 891</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_model_sequences</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L892"></a><tt class="py-lineno"> 892</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_nonmodel_sequences</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L893"></a><tt class="py-lineno"> 893</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_converged</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L894"></a><tt class="py-lineno"> 894</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_type</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L895"></a><tt class="py-lineno"> 895</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_roundnum</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L896"></a><tt class="py-lineno"> 896</tt>  <tt class="py-line"> </tt>
<a name="L897"></a><tt class="py-lineno"> 897</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__has_n</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt>   <tt class="py-comment"># Does the description line contain an N value?</tt> </tt>
</div><a name="L898"></a><tt class="py-lineno"> 898</tt>  <tt class="py-line"> </tt>
<a name="_DescriptionConsumer.description_header"></a><div id="_DescriptionConsumer.description_header-def"><a name="L899"></a><tt class="py-lineno"> 899</tt> <a class="py-toggle" href="#" id="_DescriptionConsumer.description_header-toggle" onclick="return toggle('_DescriptionConsumer.description_header');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#description_header">description_header</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_DescriptionConsumer.description_header-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_DescriptionConsumer.description_header-expanded"><a name="L900"></a><tt class="py-lineno"> 900</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-503" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-503', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Sequences producing'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L901"></a><tt class="py-lineno"> 901</tt>  <tt class="py-line">            <tt class="py-name">cols</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-504" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-504', 'split', 'link-440');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L902"></a><tt class="py-lineno"> 902</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">'N'</tt><tt class="py-op">:</tt> </tt>
<a name="L903"></a><tt class="py-lineno"> 903</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__has_n</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
</div><a name="L904"></a><tt class="py-lineno"> 904</tt>  <tt class="py-line"> </tt>
<a name="_DescriptionConsumer.description"></a><div id="_DescriptionConsumer.description-def"><a name="L905"></a><tt class="py-lineno"> 905</tt> <a class="py-toggle" href="#" id="_DescriptionConsumer.description-toggle" onclick="return toggle('_DescriptionConsumer.description');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#description">description</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_DescriptionConsumer.description-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_DescriptionConsumer.description-expanded"><a name="L906"></a><tt class="py-lineno"> 906</tt>  <tt class="py-line">        <tt class="py-name">dh</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-505" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DescriptionConsumer._parse()=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#_parse,Method Bio.Nexus.Trees.Tree._parse()=Bio.Nexus.Trees.Tree-class.html#_parse,Method Bio.PDB.PDBParser'.PDBParser._parse()=Bio.PDB.PDBParser%27.PDBParser-class.html#_parse,Method Bio.SCOP.Residues.Residues._parse()=Bio.SCOP.Residues.Residues-class.html#_parse,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer._parse()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer-class.html#_parse,Method Bio.SearchIO._index.SearchIndexer._parse()=Bio.SearchIO._index.SearchIndexer-class.html#_parse"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer._parse
Bio.Nexus.Trees.Tree._parse
Bio.PDB.PDBParser'.PDBParser._parse
Bio.SCOP.Residues.Residues._parse
Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer._parse
Bio.SearchIO._index.SearchIndexer._parse" class="py-name" href="#" onclick="return doclink('link-505', '_parse', 'link-505');">_parse</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L907"></a><tt class="py-lineno"> 907</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_type</tt> <tt class="py-op">==</tt> <tt class="py-string">'model'</tt><tt class="py-op">:</tt> </tt>
<a name="L908"></a><tt class="py-lineno"> 908</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_model_sequences</tt><tt class="py-op">.</tt><tt id="link-506" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-506', 'append', 'link-506');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">dh</tt><tt class="py-op">)</tt> </tt>
<a name="L909"></a><tt class="py-lineno"> 909</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_type</tt> <tt class="py-op">==</tt> <tt class="py-string">'nonmodel'</tt><tt class="py-op">:</tt> </tt>
<a name="L910"></a><tt class="py-lineno"> 910</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_nonmodel_sequences</tt><tt class="py-op">.</tt><tt id="link-507" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-507', 'append', 'link-506');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">dh</tt><tt class="py-op">)</tt> </tt>
<a name="L911"></a><tt class="py-lineno"> 911</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L912"></a><tt class="py-lineno"> 912</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_descriptions</tt><tt class="py-op">.</tt><tt id="link-508" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-508', 'append', 'link-506');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">dh</tt><tt class="py-op">)</tt> </tt>
</div><a name="L913"></a><tt class="py-lineno"> 913</tt>  <tt class="py-line"> </tt>
<a name="_DescriptionConsumer.model_sequences"></a><div id="_DescriptionConsumer.model_sequences-def"><a name="L914"></a><tt class="py-lineno"> 914</tt> <a class="py-toggle" href="#" id="_DescriptionConsumer.model_sequences-toggle" onclick="return toggle('_DescriptionConsumer.model_sequences');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#model_sequences">model_sequences</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_DescriptionConsumer.model_sequences-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_DescriptionConsumer.model_sequences-expanded"><a name="L915"></a><tt class="py-lineno"> 915</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_type</tt> <tt class="py-op">=</tt> <tt class="py-string">'model'</tt> </tt>
</div><a name="L916"></a><tt class="py-lineno"> 916</tt>  <tt class="py-line"> </tt>
<a name="_DescriptionConsumer.nonmodel_sequences"></a><div id="_DescriptionConsumer.nonmodel_sequences-def"><a name="L917"></a><tt class="py-lineno"> 917</tt> <a class="py-toggle" href="#" id="_DescriptionConsumer.nonmodel_sequences-toggle" onclick="return toggle('_DescriptionConsumer.nonmodel_sequences');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#nonmodel_sequences">nonmodel_sequences</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_DescriptionConsumer.nonmodel_sequences-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_DescriptionConsumer.nonmodel_sequences-expanded"><a name="L918"></a><tt class="py-lineno"> 918</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_type</tt> <tt class="py-op">=</tt> <tt class="py-string">'nonmodel'</tt> </tt>
</div><a name="L919"></a><tt class="py-lineno"> 919</tt>  <tt class="py-line"> </tt>
<a name="_DescriptionConsumer.converged"></a><div id="_DescriptionConsumer.converged-def"><a name="L920"></a><tt class="py-lineno"> 920</tt> <a class="py-toggle" href="#" id="_DescriptionConsumer.converged-toggle" onclick="return toggle('_DescriptionConsumer.converged');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#converged">converged</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_DescriptionConsumer.converged-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_DescriptionConsumer.converged-expanded"><a name="L921"></a><tt class="py-lineno"> 921</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_converged</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
</div><a name="L922"></a><tt class="py-lineno"> 922</tt>  <tt class="py-line"> </tt>
<a name="_DescriptionConsumer.no_hits"></a><div id="_DescriptionConsumer.no_hits-def"><a name="L923"></a><tt class="py-lineno"> 923</tt> <a class="py-toggle" href="#" id="_DescriptionConsumer.no_hits-toggle" onclick="return toggle('_DescriptionConsumer.no_hits');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#no_hits">no_hits</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_DescriptionConsumer.no_hits-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_DescriptionConsumer.no_hits-expanded"><a name="L924"></a><tt class="py-lineno"> 924</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L925"></a><tt class="py-lineno"> 925</tt>  <tt class="py-line"> </tt>
<a name="_DescriptionConsumer.round"></a><div id="_DescriptionConsumer.round-def"><a name="L926"></a><tt class="py-lineno"> 926</tt> <a class="py-toggle" href="#" id="_DescriptionConsumer.round-toggle" onclick="return toggle('_DescriptionConsumer.round');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#round">round</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_DescriptionConsumer.round-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_DescriptionConsumer.round-expanded"><a name="L927"></a><tt class="py-lineno"> 927</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-509" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-509', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Results from round'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L928"></a><tt class="py-lineno"> 928</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"I didn't understand the round line\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L929"></a><tt class="py-lineno"> 929</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_roundnum</tt> <tt class="py-op">=</tt> <tt id="link-510" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-510', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">18</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-511" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-511', 'strip', 'link-66');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L930"></a><tt class="py-lineno"> 930</tt>  <tt class="py-line"> </tt>
<a name="_DescriptionConsumer.end_descriptions"></a><div id="_DescriptionConsumer.end_descriptions-def"><a name="L931"></a><tt class="py-lineno"> 931</tt> <a class="py-toggle" href="#" id="_DescriptionConsumer.end_descriptions-toggle" onclick="return toggle('_DescriptionConsumer.end_descriptions');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#end_descriptions">end_descriptions</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_DescriptionConsumer.end_descriptions-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_DescriptionConsumer.end_descriptions-expanded"><a name="L932"></a><tt class="py-lineno"> 932</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div><a name="L933"></a><tt class="py-lineno"> 933</tt>  <tt class="py-line"> </tt>
<a name="_DescriptionConsumer._parse"></a><div id="_DescriptionConsumer._parse-def"><a name="L934"></a><tt class="py-lineno"> 934</tt> <a class="py-toggle" href="#" id="_DescriptionConsumer._parse-toggle" onclick="return toggle('_DescriptionConsumer._parse');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#_parse">_parse</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">description_line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_DescriptionConsumer._parse-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_DescriptionConsumer._parse-expanded"><a name="L935"></a><tt class="py-lineno"> 935</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">description_line</tt>  <tt class="py-comment"># for convenience</tt> </tt>
<a name="L936"></a><tt class="py-lineno"> 936</tt>  <tt class="py-line">        <tt class="py-name">dh</tt> <tt class="py-op">=</tt> <tt id="link-512" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-512', 'Record', 'link-11');">Record</a></tt><tt class="py-op">.</tt><tt id="link-513" class="py-name" targets="Class Bio.Blast.Record.Description=Bio.Blast.Record.Description-class.html"><a title="Bio.Blast.Record.Description" class="py-name" href="#" onclick="return doclink('link-513', 'Description', 'link-513');">Description</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L937"></a><tt class="py-lineno"> 937</tt>  <tt class="py-line"> </tt>
<a name="L938"></a><tt class="py-lineno"> 938</tt>  <tt class="py-line">        <tt class="py-comment"># I need to separate the score and p-value from the title.</tt> </tt>
<a name="L939"></a><tt class="py-lineno"> 939</tt>  <tt class="py-line">        <tt class="py-comment"># sp|P21297|FLBT_CAUCR FLBT PROTEIN     [snip]         284  7e-77</tt> </tt>
<a name="L940"></a><tt class="py-lineno"> 940</tt>  <tt class="py-line">        <tt class="py-comment"># sp|P21297|FLBT_CAUCR FLBT PROTEIN     [snip]         284  7e-77  1</tt> </tt>
<a name="L941"></a><tt class="py-lineno"> 941</tt>  <tt class="py-line">        <tt class="py-comment"># special cases to handle:</tt> </tt>
<a name="L942"></a><tt class="py-lineno"> 942</tt>  <tt class="py-line">        <tt class="py-comment">#   - title must be preserved exactly (including whitespaces)</tt> </tt>
<a name="L943"></a><tt class="py-lineno"> 943</tt>  <tt class="py-line">        <tt class="py-comment">#   - score could be equal to e-value (not likely, but what if??)</tt> </tt>
<a name="L944"></a><tt class="py-lineno"> 944</tt>  <tt class="py-line">        <tt class="py-comment">#   - sometimes there's an "N" score of '1'.</tt> </tt>
<a name="L945"></a><tt class="py-lineno"> 945</tt>  <tt class="py-line">        <tt class="py-name">cols</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-514" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-514', 'split', 'link-440');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L946"></a><tt class="py-lineno"> 946</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">3</tt><tt class="py-op">:</tt> </tt>
<a name="L947"></a><tt class="py-lineno"> 947</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt> </tt>
<a name="L948"></a><tt class="py-lineno"> 948</tt>  <tt class="py-line">                  <tt class="py-string">"Line does not appear to contain description:\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L949"></a><tt class="py-lineno"> 949</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__has_n</tt><tt class="py-op">:</tt> </tt>
<a name="L950"></a><tt class="py-lineno"> 950</tt>  <tt class="py-line">            <tt id="link-515" class="py-name" targets="Variable Bio.PDB.Polypeptide.i=Bio.PDB.Polypeptide-module.html#i,Variable Bio.SeqIO.SffIO.i=Bio.SeqIO.SffIO-module.html#i"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-515', 'i', 'link-515');">i</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-516" class="py-name" targets="Method Bio.Seq.Seq.rfind()=Bio.Seq.Seq-class.html#rfind"><a title="Bio.Seq.Seq.rfind" class="py-name" href="#" onclick="return doclink('link-516', 'rfind', 'link-516');">rfind</a></tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt>        <tt class="py-comment"># find start of N</tt> </tt>
<a name="L951"></a><tt class="py-lineno"> 951</tt>  <tt class="py-line">            <tt id="link-517" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-517', 'i', 'link-515');">i</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-518" class="py-name"><a title="Bio.Seq.Seq.rfind" class="py-name" href="#" onclick="return doclink('link-518', 'rfind', 'link-516');">rfind</a></tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt id="link-519" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-519', 'i', 'link-515');">i</a></tt><tt class="py-op">)</tt>  <tt class="py-comment"># find start of p-value</tt> </tt>
<a name="L952"></a><tt class="py-lineno"> 952</tt>  <tt class="py-line">            <tt id="link-520" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-520', 'i', 'link-515');">i</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-521" class="py-name"><a title="Bio.Seq.Seq.rfind" class="py-name" href="#" onclick="return doclink('link-521', 'rfind', 'link-516');">rfind</a></tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt id="link-522" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-522', 'i', 'link-515');">i</a></tt><tt class="py-op">)</tt>  <tt class="py-comment"># find start of score</tt> </tt>
<a name="L953"></a><tt class="py-lineno"> 953</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L954"></a><tt class="py-lineno"> 954</tt>  <tt class="py-line">            <tt id="link-523" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-523', 'i', 'link-515');">i</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-524" class="py-name"><a title="Bio.Seq.Seq.rfind" class="py-name" href="#" onclick="return doclink('link-524', 'rfind', 'link-516');">rfind</a></tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt>        <tt class="py-comment"># find start of p-value</tt> </tt>
<a name="L955"></a><tt class="py-lineno"> 955</tt>  <tt class="py-line">            <tt id="link-525" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-525', 'i', 'link-515');">i</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-526" class="py-name"><a title="Bio.Seq.Seq.rfind" class="py-name" href="#" onclick="return doclink('link-526', 'rfind', 'link-516');">rfind</a></tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt id="link-527" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-527', 'i', 'link-515');">i</a></tt><tt class="py-op">)</tt>  <tt class="py-comment"># find start of score</tt> </tt>
<a name="L956"></a><tt class="py-lineno"> 956</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__has_n</tt><tt class="py-op">:</tt> </tt>
<a name="L957"></a><tt class="py-lineno"> 957</tt>  <tt class="py-line">            <tt class="py-name">dh</tt><tt class="py-op">.</tt><tt id="link-528" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-528', 'title', 'link-202');">title</a></tt><tt class="py-op">,</tt> <tt class="py-name">dh</tt><tt class="py-op">.</tt><tt id="link-529" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-529', 'score', 'link-222');">score</a></tt><tt class="py-op">,</tt> <tt class="py-name">dh</tt><tt class="py-op">.</tt><tt class="py-name">e</tt><tt class="py-op">,</tt> <tt class="py-name">dh</tt><tt class="py-op">.</tt><tt class="py-name">num_alignments</tt> <tt class="py-op">=</tt> \ </tt>
<a name="L958"></a><tt class="py-lineno"> 958</tt>  <tt class="py-line">                      <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt id="link-530" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-530', 'i', 'link-515');">i</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-531" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-531', 'rstrip', 'link-471');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L959"></a><tt class="py-lineno"> 959</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L960"></a><tt class="py-lineno"> 960</tt>  <tt class="py-line">            <tt class="py-name">dh</tt><tt class="py-op">.</tt><tt id="link-532" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-532', 'title', 'link-202');">title</a></tt><tt class="py-op">,</tt> <tt class="py-name">dh</tt><tt class="py-op">.</tt><tt id="link-533" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-533', 'score', 'link-222');">score</a></tt><tt class="py-op">,</tt> <tt class="py-name">dh</tt><tt class="py-op">.</tt><tt class="py-name">e</tt><tt class="py-op">,</tt> <tt class="py-name">dh</tt><tt class="py-op">.</tt><tt class="py-name">num_alignments</tt> <tt class="py-op">=</tt> \ </tt>
<a name="L961"></a><tt class="py-lineno"> 961</tt>  <tt class="py-line">                      <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt id="link-534" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-534', 'i', 'link-515');">i</a></tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-535" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-535', 'rstrip', 'link-471');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt> </tt>
<a name="L962"></a><tt class="py-lineno"> 962</tt>  <tt class="py-line">        <tt class="py-name">dh</tt><tt class="py-op">.</tt><tt class="py-name">num_alignments</tt> <tt class="py-op">=</tt> <tt id="link-536" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-536', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt class="py-name">dh</tt><tt class="py-op">.</tt><tt class="py-name">num_alignments</tt><tt class="py-op">)</tt> </tt>
<a name="L963"></a><tt class="py-lineno"> 963</tt>  <tt class="py-line">        <tt class="py-name">dh</tt><tt class="py-op">.</tt><tt id="link-537" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-537', 'score', 'link-222');">score</a></tt> <tt class="py-op">=</tt> <tt id="link-538" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-538', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt class="py-name">dh</tt><tt class="py-op">.</tt><tt id="link-539" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-539', 'score', 'link-222');">score</a></tt><tt class="py-op">)</tt> </tt>
<a name="L964"></a><tt class="py-lineno"> 964</tt>  <tt class="py-line">        <tt class="py-name">dh</tt><tt class="py-op">.</tt><tt class="py-name">e</tt> <tt class="py-op">=</tt> <tt id="link-540" class="py-name" targets="Function Bio.Blast.NCBIStandalone._safe_float()=Bio.Blast.NCBIStandalone-module.html#_safe_float"><a title="Bio.Blast.NCBIStandalone._safe_float" class="py-name" href="#" onclick="return doclink('link-540', '_safe_float', 'link-540');">_safe_float</a></tt><tt class="py-op">(</tt><tt class="py-name">dh</tt><tt class="py-op">.</tt><tt class="py-name">e</tt><tt class="py-op">)</tt> </tt>
<a name="L965"></a><tt class="py-lineno"> 965</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">dh</tt> </tt>
</div></div><a name="L966"></a><tt class="py-lineno"> 966</tt>  <tt class="py-line"> </tt>
<a name="L967"></a><tt class="py-lineno"> 967</tt>  <tt class="py-line"> </tt>
<a name="_AlignmentConsumer"></a><div id="_AlignmentConsumer-def"><a name="L968"></a><tt class="py-lineno"> 968</tt> <a class="py-toggle" href="#" id="_AlignmentConsumer-toggle" onclick="return toggle('_AlignmentConsumer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html">_AlignmentConsumer</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_AlignmentConsumer-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_AlignmentConsumer-expanded"><a name="L969"></a><tt class="py-lineno"> 969</tt>  <tt class="py-line">    <tt class="py-comment"># This is a little bit tricky.  An alignment can either be a</tt> </tt>
<a name="L970"></a><tt class="py-lineno"> 970</tt>  <tt class="py-line">    <tt class="py-comment"># pairwise alignment or a multiple alignment.  Since it's difficult</tt> </tt>
<a name="L971"></a><tt class="py-lineno"> 971</tt>  <tt class="py-line">    <tt class="py-comment"># to know a-priori which one the blast record will contain, I'm going</tt> </tt>
<a name="L972"></a><tt class="py-lineno"> 972</tt>  <tt class="py-line">    <tt class="py-comment"># to make one class that can parse both of them.</tt> </tt>
<a name="_AlignmentConsumer.start_alignment"></a><div id="_AlignmentConsumer.start_alignment-def"><a name="L973"></a><tt class="py-lineno"> 973</tt> <a class="py-toggle" href="#" id="_AlignmentConsumer.start_alignment-toggle" onclick="return toggle('_AlignmentConsumer.start_alignment');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#start_alignment">start_alignment</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_AlignmentConsumer.start_alignment-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_AlignmentConsumer.start_alignment-expanded"><a name="L974"></a><tt class="py-lineno"> 974</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_alignment</tt> <tt class="py-op">=</tt> <tt id="link-541" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-541', 'Record', 'link-11');">Record</a></tt><tt class="py-op">.</tt><tt id="link-542" class="py-name" targets="Class Bio.Align.Generic.Alignment=Bio.Align.Generic.Alignment-class.html,Class Bio.Blast.Record.Alignment=Bio.Blast.Record.Alignment-class.html,Class Bio.Wise.psw.Alignment=Bio.Wise.psw.Alignment-class.html"><a title="Bio.Align.Generic.Alignment
Bio.Blast.Record.Alignment
Bio.Wise.psw.Alignment" class="py-name" href="#" onclick="return doclink('link-542', 'Alignment', 'link-542');">Alignment</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L975"></a><tt class="py-lineno"> 975</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_multiple_alignment</tt> <tt class="py-op">=</tt> <tt id="link-543" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-543', 'Record', 'link-11');">Record</a></tt><tt class="py-op">.</tt><tt id="link-544" class="py-name" targets="Class Bio.Blast.Record.MultipleAlignment=Bio.Blast.Record.MultipleAlignment-class.html"><a title="Bio.Blast.Record.MultipleAlignment" class="py-name" href="#" onclick="return doclink('link-544', 'MultipleAlignment', 'link-544');">MultipleAlignment</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L976"></a><tt class="py-lineno"> 976</tt>  <tt class="py-line"> </tt>
<a name="_AlignmentConsumer.title"></a><div id="_AlignmentConsumer.title-def"><a name="L977"></a><tt class="py-lineno"> 977</tt> <a class="py-toggle" href="#" id="_AlignmentConsumer.title-toggle" onclick="return toggle('_AlignmentConsumer.title');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#title">title</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_AlignmentConsumer.title-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_AlignmentConsumer.title-expanded"><a name="L978"></a><tt class="py-lineno"> 978</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_alignment</tt><tt class="py-op">.</tt><tt id="link-545" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-545', 'title', 'link-202');">title</a></tt><tt class="py-op">:</tt> </tt>
<a name="L979"></a><tt class="py-lineno"> 979</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_alignment</tt><tt class="py-op">.</tt><tt id="link-546" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-546', 'title', 'link-202');">title</a></tt> <tt class="py-op">+=</tt> <tt class="py-string">" "</tt> </tt>
<a name="L980"></a><tt class="py-lineno"> 980</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_alignment</tt><tt class="py-op">.</tt><tt id="link-547" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-547', 'title', 'link-202');">title</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-548" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-548', 'strip', 'link-66');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L981"></a><tt class="py-lineno"> 981</tt>  <tt class="py-line"> </tt>
<a name="_AlignmentConsumer.length"></a><div id="_AlignmentConsumer.length-def"><a name="L982"></a><tt class="py-lineno"> 982</tt> <a class="py-toggle" href="#" id="_AlignmentConsumer.length-toggle" onclick="return toggle('_AlignmentConsumer.length');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#length">length</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_AlignmentConsumer.length-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_AlignmentConsumer.length-expanded"><a name="L983"></a><tt class="py-lineno"> 983</tt>  <tt class="py-line">        <tt class="py-comment"># e.g. "Length = 81" or more recently, "Length=428"</tt> </tt>
<a name="L984"></a><tt class="py-lineno"> 984</tt>  <tt class="py-line">        <tt id="link-549" class="py-name" targets="Variable Bio.SeqFeature.FeatureLocation.parts=Bio.SeqFeature.FeatureLocation-class.html#parts"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-549', 'parts', 'link-549');">parts</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">" "</tt><tt class="py-op">,</tt> <tt class="py-string">""</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-550" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-550', 'split', 'link-440');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">"="</tt><tt class="py-op">)</tt> </tt>
<a name="L985"></a><tt class="py-lineno"> 985</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-551" class="py-name"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-551', 'parts', 'link-549');">parts</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">2</tt><tt class="py-op">,</tt> <tt class="py-string">"Unrecognised format length line"</tt> </tt>
<a name="L986"></a><tt class="py-lineno"> 986</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_alignment</tt><tt class="py-op">.</tt><tt id="link-552" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-552', 'length', 'link-209');">length</a></tt> <tt class="py-op">=</tt> <tt id="link-553" class="py-name"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-553', 'parts', 'link-549');">parts</a></tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L987"></a><tt class="py-lineno"> 987</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_alignment</tt><tt class="py-op">.</tt><tt id="link-554" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-554', 'length', 'link-209');">length</a></tt> <tt class="py-op">=</tt> <tt id="link-555" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-555', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_alignment</tt><tt class="py-op">.</tt><tt id="link-556" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-556', 'length', 'link-209');">length</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L988"></a><tt class="py-lineno"> 988</tt>  <tt class="py-line"> </tt>
<a name="_AlignmentConsumer.multalign"></a><div id="_AlignmentConsumer.multalign-def"><a name="L989"></a><tt class="py-lineno"> 989</tt> <a class="py-toggle" href="#" id="_AlignmentConsumer.multalign-toggle" onclick="return toggle('_AlignmentConsumer.multalign');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#multalign">multalign</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_AlignmentConsumer.multalign-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_AlignmentConsumer.multalign-expanded"><a name="L990"></a><tt class="py-lineno"> 990</tt>  <tt class="py-line">        <tt class="py-comment"># Standalone version uses 'QUERY', while WWW version uses blast_tmp.</tt> </tt>
<a name="L991"></a><tt class="py-lineno"> 991</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-557" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-557', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'QUERY'</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-558" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-558', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'blast_tmp'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L992"></a><tt class="py-lineno"> 992</tt>  <tt class="py-line">            <tt class="py-comment"># If this is the first line of the multiple alignment,</tt> </tt>
<a name="L993"></a><tt class="py-lineno"> 993</tt>  <tt class="py-line">            <tt class="py-comment"># then I need to figure out how the line is formatted.</tt> </tt>
<a name="L994"></a><tt class="py-lineno"> 994</tt>  <tt class="py-line"> </tt>
<a name="L995"></a><tt class="py-lineno"> 995</tt>  <tt class="py-line">            <tt class="py-comment"># Format of line is:</tt> </tt>
<a name="L996"></a><tt class="py-lineno"> 996</tt>  <tt class="py-line">            <tt class="py-comment"># QUERY 1   acttg...gccagaggtggtttattcagtctccataagagaggggacaaacg 60</tt> </tt>
<a name="L997"></a><tt class="py-lineno"> 997</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L998"></a><tt class="py-lineno"> 998</tt>  <tt class="py-line">                <tt id="link-559" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.name=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.name=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.name=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#name,Method Bio.Phylo.PhyloXMLIO.Writer.name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#name,Variable Bio.codonalign.codonalphabet.CodonAlphabet.name=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-559', 'name', 'link-559');">name</a></tt><tt class="py-op">,</tt> <tt id="link-560" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-560', 'start', 'link-35');">start</a></tt><tt class="py-op">,</tt> <tt id="link-561" class="py-name" targets="Variable Bio.SeqRecord.SeqRecord.seq=Bio.SeqRecord.SeqRecord-class.html#seq,Variable BioSQL.BioSeq.DBSeqRecord.seq=BioSQL.BioSeq.DBSeqRecord-class.html#seq"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-561', 'seq', 'link-561');">seq</a></tt><tt class="py-op">,</tt> <tt id="link-562" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-562', 'end', 'link-78');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-563" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-563', 'split', 'link-440');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L999"></a><tt class="py-lineno"> 999</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L1000"></a><tt class="py-lineno">1000</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"I do not understand the line\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1001"></a><tt class="py-lineno">1001</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_start_index</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-564" class="py-name" targets="Method Bio.Crystal.Chain.index()=Bio.Crystal.Chain-class.html#index,Method Bio.SCOP.Raf.SeqMap.index()=Bio.SCOP.Raf.SeqMap-class.html#index,Method Bio.SearchIO._model.hit.Hit.index()=Bio.SearchIO._model.hit.Hit-class.html#index,Method Bio.SearchIO._model.query.QueryResult.index()=Bio.SearchIO._model.query.QueryResult-class.html#index,Function Bio.SearchIO.index()=Bio.SearchIO-module.html#index,Method Bio.Seq.MutableSeq.index()=Bio.Seq.MutableSeq-class.html#index,Function Bio.SeqIO.index()=Bio.SeqIO-module.html#index"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-564', 'index', 'link-564');">index</a></tt><tt class="py-op">(</tt><tt id="link-565" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-565', 'start', 'link-35');">start</a></tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-566" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-566', 'name', 'link-559');">name</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1002"></a><tt class="py-lineno">1002</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_index</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-567" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-567', 'index', 'link-564');">index</a></tt><tt class="py-op">(</tt><tt id="link-568" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-568', 'seq', 'link-561');">seq</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1003"></a><tt class="py-lineno">1003</tt>  <tt class="py-line">                                         <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_start_index</tt> <tt class="py-op">+</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-569" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-569', 'start', 'link-35');">start</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1004"></a><tt class="py-lineno">1004</tt>  <tt class="py-line">            <tt class="py-comment"># subtract 1 for the space</tt> </tt>
<a name="L1005"></a><tt class="py-lineno">1005</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_name_length</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_start_index</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt> </tt>
<a name="L1006"></a><tt class="py-lineno">1006</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_start_length</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_index</tt> <tt class="py-op">-</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_start_index</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt> </tt>
<a name="L1007"></a><tt class="py-lineno">1007</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_length</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-570" class="py-name"><a title="Bio.Seq.Seq.rfind" class="py-name" href="#" onclick="return doclink('link-570', 'rfind', 'link-516');">rfind</a></tt><tt class="py-op">(</tt><tt id="link-571" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-571', 'end', 'link-78');">end</a></tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_index</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt> </tt>
<a name="L1008"></a><tt class="py-lineno">1008</tt>  <tt class="py-line"> </tt>
<a name="L1009"></a><tt class="py-lineno">1009</tt>  <tt class="py-line">            <tt class="py-comment"># self._seq_index = line.index(seq)</tt> </tt>
<a name="L1010"></a><tt class="py-lineno">1010</tt>  <tt class="py-line">            <tt class="py-comment"># # subtract 1 for the space</tt> </tt>
<a name="L1011"></a><tt class="py-lineno">1011</tt>  <tt class="py-line">            <tt class="py-comment"># self._seq_length = line.rfind(end) - self._seq_index - 1</tt> </tt>
<a name="L1012"></a><tt class="py-lineno">1012</tt>  <tt class="py-line">            <tt class="py-comment"># self._start_index = line.index(start)</tt> </tt>
<a name="L1013"></a><tt class="py-lineno">1013</tt>  <tt class="py-line">            <tt class="py-comment"># self._start_length = self._seq_index - self._start_index - 1</tt> </tt>
<a name="L1014"></a><tt class="py-lineno">1014</tt>  <tt class="py-line">            <tt class="py-comment"># self._name_length = self._start_index</tt> </tt>
<a name="L1015"></a><tt class="py-lineno">1015</tt>  <tt class="py-line"> </tt>
<a name="L1016"></a><tt class="py-lineno">1016</tt>  <tt class="py-line">        <tt class="py-comment"># Extract the information from the line</tt> </tt>
<a name="L1017"></a><tt class="py-lineno">1017</tt>  <tt class="py-line">        <tt id="link-572" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-572', 'name', 'link-559');">name</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_name_length</tt><tt class="py-op">]</tt> </tt>
<a name="L1018"></a><tt class="py-lineno">1018</tt>  <tt class="py-line">        <tt id="link-573" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-573', 'name', 'link-559');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-574" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-574', 'name', 'link-559');">name</a></tt><tt class="py-op">.</tt><tt id="link-575" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-575', 'rstrip', 'link-471');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1019"></a><tt class="py-lineno">1019</tt>  <tt class="py-line">        <tt id="link-576" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-576', 'start', 'link-35');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_start_index</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_start_index</tt> <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_start_length</tt><tt class="py-op">]</tt> </tt>
<a name="L1020"></a><tt class="py-lineno">1020</tt>  <tt class="py-line">        <tt id="link-577" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-577', 'start', 'link-35');">start</a></tt> <tt class="py-op">=</tt> <tt id="link-578" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-578', 'start', 'link-35');">start</a></tt><tt class="py-op">.</tt><tt id="link-579" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-579', 'rstrip', 'link-471');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1021"></a><tt class="py-lineno">1021</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-580" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-580', 'start', 'link-35');">start</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1022"></a><tt class="py-lineno">1022</tt>  <tt class="py-line">            <tt id="link-581" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-581', 'start', 'link-35');">start</a></tt> <tt class="py-op">=</tt> <tt id="link-582" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-582', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-583" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-583', 'start', 'link-35');">start</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1023"></a><tt class="py-lineno">1023</tt>  <tt class="py-line">        <tt id="link-584" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-584', 'end', 'link-78');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_index</tt> <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_length</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-585" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-585', 'rstrip', 'link-471');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1024"></a><tt class="py-lineno">1024</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-586" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-586', 'end', 'link-78');">end</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1025"></a><tt class="py-lineno">1025</tt>  <tt class="py-line">            <tt id="link-587" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-587', 'end', 'link-78');">end</a></tt> <tt class="py-op">=</tt> <tt id="link-588" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-588', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-589" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-589', 'end', 'link-78');">end</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1026"></a><tt class="py-lineno">1026</tt>  <tt class="py-line">        <tt id="link-590" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-590', 'seq', 'link-561');">seq</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_index</tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_index</tt> <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_length</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-591" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-591', 'rstrip', 'link-471');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1027"></a><tt class="py-lineno">1027</tt>  <tt class="py-line">        <tt class="py-comment"># right pad the sequence with spaces if necessary</tt> </tt>
<a name="L1028"></a><tt class="py-lineno">1028</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-592" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-592', 'seq', 'link-561');">seq</a></tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_length</tt><tt class="py-op">:</tt> </tt>
<a name="L1029"></a><tt class="py-lineno">1029</tt>  <tt class="py-line">            <tt id="link-593" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-593', 'seq', 'link-561');">seq</a></tt> <tt class="py-op">+=</tt> <tt class="py-string">' '</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_length</tt> <tt class="py-op">-</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-594" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-594', 'seq', 'link-561');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1030"></a><tt class="py-lineno">1030</tt>  <tt class="py-line"> </tt>
<a name="L1031"></a><tt class="py-lineno">1031</tt>  <tt class="py-line">        <tt class="py-comment"># I need to make sure the sequence is aligned correctly with the query.</tt> </tt>
<a name="L1032"></a><tt class="py-lineno">1032</tt>  <tt class="py-line">        <tt class="py-comment"># First, I will find the length of the query.  Then, if necessary,</tt> </tt>
<a name="L1033"></a><tt class="py-lineno">1033</tt>  <tt class="py-line">        <tt class="py-comment"># I will pad my current sequence with spaces so that they will line</tt> </tt>
<a name="L1034"></a><tt class="py-lineno">1034</tt>  <tt class="py-line">        <tt class="py-comment"># up correctly.</tt> </tt>
<a name="L1035"></a><tt class="py-lineno">1035</tt>  <tt class="py-line"> </tt>
<a name="L1036"></a><tt class="py-lineno">1036</tt>  <tt class="py-line">        <tt class="py-comment"># Two possible things can happen:</tt> </tt>
<a name="L1037"></a><tt class="py-lineno">1037</tt>  <tt class="py-line">        <tt class="py-comment"># QUERY</tt> </tt>
<a name="L1038"></a><tt class="py-lineno">1038</tt>  <tt class="py-line">        <tt class="py-comment"># 504</tt> </tt>
<a name="L1039"></a><tt class="py-lineno">1039</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L1040"></a><tt class="py-lineno">1040</tt>  <tt class="py-line">        <tt class="py-comment"># QUERY</tt> </tt>
<a name="L1041"></a><tt class="py-lineno">1041</tt>  <tt class="py-line">        <tt class="py-comment"># 403</tt> </tt>
<a name="L1042"></a><tt class="py-lineno">1042</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L1043"></a><tt class="py-lineno">1043</tt>  <tt class="py-line">        <tt class="py-comment"># Sequence 504 will need padding at the end.  Since I won't know</tt> </tt>
<a name="L1044"></a><tt class="py-lineno">1044</tt>  <tt class="py-line">        <tt class="py-comment"># this until the end of the alignment, this will be handled in</tt> </tt>
<a name="L1045"></a><tt class="py-lineno">1045</tt>  <tt class="py-line">        <tt class="py-comment"># end_alignment.</tt> </tt>
<a name="L1046"></a><tt class="py-lineno">1046</tt>  <tt class="py-line">        <tt class="py-comment"># Sequence 403 will need padding before being added to the alignment.</tt> </tt>
<a name="L1047"></a><tt class="py-lineno">1047</tt>  <tt class="py-line"> </tt>
<a name="L1048"></a><tt class="py-lineno">1048</tt>  <tt class="py-line">        <tt id="link-595" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.align
Bio.FSSP.fssp_rec.align
Bio.Wise.align
Bio.Wise.dnal.align
Bio.Wise.psw.align
Bio.pairwise2.align" class="py-name" href="#" onclick="return doclink('link-595', 'align', 'link-242');">align</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_multiple_alignment</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt>  <tt class="py-comment"># for convenience</tt> </tt>
<a name="L1049"></a><tt class="py-lineno">1049</tt>  <tt class="py-line">        <tt id="link-596" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.align
Bio.FSSP.fssp_rec.align
Bio.Wise.align
Bio.Wise.dnal.align
Bio.Wise.psw.align
Bio.pairwise2.align" class="py-name" href="#" onclick="return doclink('link-596', 'align', 'link-242');">align</a></tt><tt class="py-op">.</tt><tt id="link-597" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-597', 'append', 'link-506');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt id="link-598" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-598', 'name', 'link-559');">name</a></tt><tt class="py-op">,</tt> <tt id="link-599" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-599', 'start', 'link-35');">start</a></tt><tt class="py-op">,</tt> <tt id="link-600" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-600', 'seq', 'link-561');">seq</a></tt><tt class="py-op">,</tt> <tt id="link-601" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-601', 'end', 'link-78');">end</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1050"></a><tt class="py-lineno">1050</tt>  <tt class="py-line"> </tt>
<a name="L1051"></a><tt class="py-lineno">1051</tt>  <tt class="py-line">        <tt class="py-comment"># This is old code that tried to line up all the sequences</tt> </tt>
<a name="L1052"></a><tt class="py-lineno">1052</tt>  <tt class="py-line">        <tt class="py-comment"># in a multiple alignment by using the sequence title's as</tt> </tt>
<a name="L1053"></a><tt class="py-lineno">1053</tt>  <tt class="py-line">        <tt class="py-comment"># identifiers.  The problem with this is that BLAST assigns</tt> </tt>
<a name="L1054"></a><tt class="py-lineno">1054</tt>  <tt class="py-line">        <tt class="py-comment"># different HSP's from the same sequence the same id.  Thus,</tt> </tt>
<a name="L1055"></a><tt class="py-lineno">1055</tt>  <tt class="py-line">        <tt class="py-comment"># in one alignment block, there may be multiple sequences with</tt> </tt>
<a name="L1056"></a><tt class="py-lineno">1056</tt>  <tt class="py-line">        <tt class="py-comment"># the same id.  I'm not sure how to handle this, so I'm not</tt> </tt>
<a name="L1057"></a><tt class="py-lineno">1057</tt>  <tt class="py-line">        <tt class="py-comment"># going to.</tt> </tt>
<a name="L1058"></a><tt class="py-lineno">1058</tt>  <tt class="py-line"> </tt>
<a name="L1059"></a><tt class="py-lineno">1059</tt>  <tt class="py-line">        <tt class="py-comment"># # If the sequence is the query, then just add it.</tt> </tt>
<a name="L1060"></a><tt class="py-lineno">1060</tt>  <tt class="py-line">        <tt class="py-comment"># if name == 'QUERY':</tt> </tt>
<a name="L1061"></a><tt class="py-lineno">1061</tt>  <tt class="py-line">        <tt class="py-comment">#     if len(align) == 0:</tt> </tt>
<a name="L1062"></a><tt class="py-lineno">1062</tt>  <tt class="py-line">        <tt class="py-comment">#         align.append((name, start, seq))</tt> </tt>
<a name="L1063"></a><tt class="py-lineno">1063</tt>  <tt class="py-line">        <tt class="py-comment">#     else:</tt> </tt>
<a name="L1064"></a><tt class="py-lineno">1064</tt>  <tt class="py-line">        <tt class="py-comment">#         aname, astart, aseq = align[0]</tt> </tt>
<a name="L1065"></a><tt class="py-lineno">1065</tt>  <tt class="py-line">        <tt class="py-comment">#         if name != aname:</tt> </tt>
<a name="L1066"></a><tt class="py-lineno">1066</tt>  <tt class="py-line">        <tt class="py-comment">#             raise ValueError, "Query is not the first sequence"</tt> </tt>
<a name="L1067"></a><tt class="py-lineno">1067</tt>  <tt class="py-line">        <tt class="py-comment">#         aseq = aseq + seq</tt> </tt>
<a name="L1068"></a><tt class="py-lineno">1068</tt>  <tt class="py-line">        <tt class="py-comment">#         align[0] = aname, astart, aseq</tt> </tt>
<a name="L1069"></a><tt class="py-lineno">1069</tt>  <tt class="py-line">        <tt class="py-comment"># else:</tt> </tt>
<a name="L1070"></a><tt class="py-lineno">1070</tt>  <tt class="py-line">        <tt class="py-comment">#     if len(align) == 0:</tt> </tt>
<a name="L1071"></a><tt class="py-lineno">1071</tt>  <tt class="py-line">        <tt class="py-comment">#         raise ValueError, "I could not find the query sequence"</tt> </tt>
<a name="L1072"></a><tt class="py-lineno">1072</tt>  <tt class="py-line">        <tt class="py-comment">#     qname, qstart, qseq = align[0]</tt> </tt>
<a name="L1073"></a><tt class="py-lineno">1073</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L1074"></a><tt class="py-lineno">1074</tt>  <tt class="py-line">        <tt class="py-comment">#     # Now find my sequence in the multiple alignment.</tt> </tt>
<a name="L1075"></a><tt class="py-lineno">1075</tt>  <tt class="py-line">        <tt class="py-comment">#     for i in range(1, len(align)):</tt> </tt>
<a name="L1076"></a><tt class="py-lineno">1076</tt>  <tt class="py-line">        <tt class="py-comment">#         aname, astart, aseq = align[i]</tt> </tt>
<a name="L1077"></a><tt class="py-lineno">1077</tt>  <tt class="py-line">        <tt class="py-comment">#         if name == aname:</tt> </tt>
<a name="L1078"></a><tt class="py-lineno">1078</tt>  <tt class="py-line">        <tt class="py-comment">#             index = i</tt> </tt>
<a name="L1079"></a><tt class="py-lineno">1079</tt>  <tt class="py-line">        <tt class="py-comment">#             break</tt> </tt>
<a name="L1080"></a><tt class="py-lineno">1080</tt>  <tt class="py-line">        <tt class="py-comment">#     else:</tt> </tt>
<a name="L1081"></a><tt class="py-lineno">1081</tt>  <tt class="py-line">        <tt class="py-comment">#         # If I couldn't find it, then add a new one.</tt> </tt>
<a name="L1082"></a><tt class="py-lineno">1082</tt>  <tt class="py-line">        <tt class="py-comment">#         align.append((None, None, None))</tt> </tt>
<a name="L1083"></a><tt class="py-lineno">1083</tt>  <tt class="py-line">        <tt class="py-comment">#         index = len(align)-1</tt> </tt>
<a name="L1084"></a><tt class="py-lineno">1084</tt>  <tt class="py-line">        <tt class="py-comment">#         # Make sure to left-pad it.</tt> </tt>
<a name="L1085"></a><tt class="py-lineno">1085</tt>  <tt class="py-line">        <tt class="py-comment">#         aname, astart, aseq = name, start, ' '*(len(qseq)-len(seq))</tt> </tt>
<a name="L1086"></a><tt class="py-lineno">1086</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L1087"></a><tt class="py-lineno">1087</tt>  <tt class="py-line">        <tt class="py-comment">#     if len(qseq) != len(aseq) + len(seq):</tt> </tt>
<a name="L1088"></a><tt class="py-lineno">1088</tt>  <tt class="py-line">        <tt class="py-comment">#         # If my sequences are shorter than the query sequence,</tt> </tt>
<a name="L1089"></a><tt class="py-lineno">1089</tt>  <tt class="py-line">        <tt class="py-comment">#         # then I will need to pad some spaces to make them line up.</tt> </tt>
<a name="L1090"></a><tt class="py-lineno">1090</tt>  <tt class="py-line">        <tt class="py-comment">#         # Since I've already right padded seq, that means aseq</tt> </tt>
<a name="L1091"></a><tt class="py-lineno">1091</tt>  <tt class="py-line">        <tt class="py-comment">#         # must be too short.</tt> </tt>
<a name="L1092"></a><tt class="py-lineno">1092</tt>  <tt class="py-line">        <tt class="py-comment">#         aseq = aseq + ' '*(len(qseq)-len(aseq)-len(seq))</tt> </tt>
<a name="L1093"></a><tt class="py-lineno">1093</tt>  <tt class="py-line">        <tt class="py-comment">#     aseq = aseq + seq</tt> </tt>
<a name="L1094"></a><tt class="py-lineno">1094</tt>  <tt class="py-line">        <tt class="py-comment">#     if astart is None:</tt> </tt>
<a name="L1095"></a><tt class="py-lineno">1095</tt>  <tt class="py-line">        <tt class="py-comment">#         astart = start</tt> </tt>
<a name="L1096"></a><tt class="py-lineno">1096</tt>  <tt class="py-line">        <tt class="py-comment">#     align[index] = aname, astart, aseq</tt> </tt>
<a name="L1097"></a><tt class="py-lineno">1097</tt>  <tt class="py-line"> </tt>
<a name="_AlignmentConsumer.end_alignment"></a><div id="_AlignmentConsumer.end_alignment-def"><a name="L1098"></a><tt class="py-lineno">1098</tt> <a class="py-toggle" href="#" id="_AlignmentConsumer.end_alignment-toggle" onclick="return toggle('_AlignmentConsumer.end_alignment');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#end_alignment">end_alignment</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_AlignmentConsumer.end_alignment-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_AlignmentConsumer.end_alignment-expanded"><a name="L1099"></a><tt class="py-lineno">1099</tt>  <tt class="py-line">        <tt class="py-comment"># Remove trailing newlines</tt> </tt>
<a name="L1100"></a><tt class="py-lineno">1100</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_alignment</tt><tt class="py-op">:</tt> </tt>
<a name="L1101"></a><tt class="py-lineno">1101</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_alignment</tt><tt class="py-op">.</tt><tt id="link-602" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-602', 'title', 'link-202');">title</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_alignment</tt><tt class="py-op">.</tt><tt id="link-603" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-603', 'title', 'link-202');">title</a></tt><tt class="py-op">.</tt><tt id="link-604" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-604', 'rstrip', 'link-471');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1102"></a><tt class="py-lineno">1102</tt>  <tt class="py-line"> </tt>
<a name="L1103"></a><tt class="py-lineno">1103</tt>  <tt class="py-line">        <tt class="py-comment"># This code is also obsolete.  See note above.</tt> </tt>
<a name="L1104"></a><tt class="py-lineno">1104</tt>  <tt class="py-line">        <tt class="py-comment"># If there's a multiple alignment, I will need to make sure</tt> </tt>
<a name="L1105"></a><tt class="py-lineno">1105</tt>  <tt class="py-line">        <tt class="py-comment"># all the sequences are aligned.  That is, I may need to</tt> </tt>
<a name="L1106"></a><tt class="py-lineno">1106</tt>  <tt class="py-line">        <tt class="py-comment"># right-pad the sequences.</tt> </tt>
<a name="L1107"></a><tt class="py-lineno">1107</tt>  <tt class="py-line">        <tt class="py-comment"># if self._multiple_alignment is not None:</tt> </tt>
<a name="L1108"></a><tt class="py-lineno">1108</tt>  <tt class="py-line">        <tt class="py-comment">#     align = self._multiple_alignment.alignment</tt> </tt>
<a name="L1109"></a><tt class="py-lineno">1109</tt>  <tt class="py-line">        <tt class="py-comment">#     seqlen = None</tt> </tt>
<a name="L1110"></a><tt class="py-lineno">1110</tt>  <tt class="py-line">        <tt class="py-comment">#     for i in range(len(align)):</tt> </tt>
<a name="L1111"></a><tt class="py-lineno">1111</tt>  <tt class="py-line">        <tt class="py-comment">#         name, start, seq = align[i]</tt> </tt>
<a name="L1112"></a><tt class="py-lineno">1112</tt>  <tt class="py-line">        <tt class="py-comment">#         if seqlen is None:</tt> </tt>
<a name="L1113"></a><tt class="py-lineno">1113</tt>  <tt class="py-line">        <tt class="py-comment">#             seqlen = len(seq)</tt> </tt>
<a name="L1114"></a><tt class="py-lineno">1114</tt>  <tt class="py-line">        <tt class="py-comment">#         else:</tt> </tt>
<a name="L1115"></a><tt class="py-lineno">1115</tt>  <tt class="py-line">        <tt class="py-comment">#             if len(seq) &lt; seqlen:</tt> </tt>
<a name="L1116"></a><tt class="py-lineno">1116</tt>  <tt class="py-line">        <tt class="py-comment">#                 seq = seq + ' '*(seqlen - len(seq))</tt> </tt>
<a name="L1117"></a><tt class="py-lineno">1117</tt>  <tt class="py-line">        <tt class="py-comment">#                 align[i] = name, start, seq</tt> </tt>
<a name="L1118"></a><tt class="py-lineno">1118</tt>  <tt class="py-line">        <tt class="py-comment">#             elif len(seq) &gt; seqlen:</tt> </tt>
<a name="L1119"></a><tt class="py-lineno">1119</tt>  <tt class="py-line">        <tt class="py-comment">#                 raise ValueError, \</tt> </tt>
<a name="L1120"></a><tt class="py-lineno">1120</tt>  <tt class="py-line">        <tt class="py-comment">#                       "Sequence %s is longer than the query" % name</tt> </tt>
<a name="L1121"></a><tt class="py-lineno">1121</tt>  <tt class="py-line"> </tt>
<a name="L1122"></a><tt class="py-lineno">1122</tt>  <tt class="py-line">        <tt class="py-comment"># Clean up some variables, if they exist.</tt> </tt>
<a name="L1123"></a><tt class="py-lineno">1123</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1124"></a><tt class="py-lineno">1124</tt>  <tt class="py-line">            <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_index</tt> </tt>
<a name="L1125"></a><tt class="py-lineno">1125</tt>  <tt class="py-line">            <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq_length</tt> </tt>
<a name="L1126"></a><tt class="py-lineno">1126</tt>  <tt class="py-line">            <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_start_index</tt> </tt>
<a name="L1127"></a><tt class="py-lineno">1127</tt>  <tt class="py-line">            <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_start_length</tt> </tt>
<a name="L1128"></a><tt class="py-lineno">1128</tt>  <tt class="py-line">            <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_name_length</tt> </tt>
<a name="L1129"></a><tt class="py-lineno">1129</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L1130"></a><tt class="py-lineno">1130</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
</div></div><a name="L1131"></a><tt class="py-lineno">1131</tt>  <tt class="py-line"> </tt>
<a name="L1132"></a><tt class="py-lineno">1132</tt>  <tt class="py-line"> </tt>
<a name="_HSPConsumer"></a><div id="_HSPConsumer-def"><a name="L1133"></a><tt class="py-lineno">1133</tt> <a class="py-toggle" href="#" id="_HSPConsumer-toggle" onclick="return toggle('_HSPConsumer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._HSPConsumer-class.html">_HSPConsumer</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_HSPConsumer-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_HSPConsumer-expanded"><a name="_HSPConsumer.start_hsp"></a><div id="_HSPConsumer.start_hsp-def"><a name="L1134"></a><tt class="py-lineno">1134</tt> <a class="py-toggle" href="#" id="_HSPConsumer.start_hsp-toggle" onclick="return toggle('_HSPConsumer.start_hsp');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._HSPConsumer-class.html#start_hsp">start_hsp</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_HSPConsumer.start_hsp-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_HSPConsumer.start_hsp-expanded"><a name="L1135"></a><tt class="py-lineno">1135</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt> <tt class="py-op">=</tt> <tt id="link-605" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-605', 'Record', 'link-11');">Record</a></tt><tt class="py-op">.</tt><tt id="link-606" class="py-name" targets="Class Bio.Blast.Record.HSP=Bio.Blast.Record.HSP-class.html,Class Bio.SearchIO._model.hsp.HSP=Bio.SearchIO._model.hsp.HSP-class.html"><a title="Bio.Blast.Record.HSP
Bio.SearchIO._model.hsp.HSP" class="py-name" href="#" onclick="return doclink('link-606', 'HSP', 'link-606');">HSP</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1136"></a><tt class="py-lineno">1136</tt>  <tt class="py-line"> </tt>
<a name="_HSPConsumer.score"></a><div id="_HSPConsumer.score-def"><a name="L1137"></a><tt class="py-lineno">1137</tt> <a class="py-toggle" href="#" id="_HSPConsumer.score-toggle" onclick="return toggle('_HSPConsumer.score');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._HSPConsumer-class.html#score">score</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_HSPConsumer.score-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_HSPConsumer.score-expanded"><a name="L1138"></a><tt class="py-lineno">1138</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt class="py-name">bits</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt id="link-607" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-607', 'score', 'link-222');">score</a></tt> <tt class="py-op">=</tt> <tt id="link-608" class="py-name"><a title="Bio.Blast.NCBIStandalone._re_search" class="py-name" href="#" onclick="return doclink('link-608', '_re_search', 'link-467');">_re_search</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1139"></a><tt class="py-lineno">1139</tt>  <tt class="py-line">            <tt class="py-string">r"Score =\s*([0-9.e+]+) bits \(([0-9]+)\)"</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">,</tt> </tt>
<a name="L1140"></a><tt class="py-lineno">1140</tt>  <tt class="py-line">            <tt class="py-string">"I could not find the score in line\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1141"></a><tt class="py-lineno">1141</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt id="link-609" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-609', 'score', 'link-222');">score</a></tt> <tt class="py-op">=</tt> <tt id="link-610" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_float" class="py-name" href="#" onclick="return doclink('link-610', '_safe_float', 'link-540');">_safe_float</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt id="link-611" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-611', 'score', 'link-222');">score</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1142"></a><tt class="py-lineno">1142</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt class="py-name">bits</tt> <tt class="py-op">=</tt> <tt id="link-612" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_float" class="py-name" href="#" onclick="return doclink('link-612', '_safe_float', 'link-540');">_safe_float</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt class="py-name">bits</tt><tt class="py-op">)</tt> </tt>
<a name="L1143"></a><tt class="py-lineno">1143</tt>  <tt class="py-line"> </tt>
<a name="L1144"></a><tt class="py-lineno">1144</tt>  <tt class="py-line">        <tt id="link-613" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-613', 'x', 'link-322');">x</a></tt><tt class="py-op">,</tt> <tt id="link-614" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.y=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#y"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-614', 'y', 'link-614');">y</a></tt> <tt class="py-op">=</tt> <tt id="link-615" class="py-name"><a title="Bio.Blast.NCBIStandalone._re_search" class="py-name" href="#" onclick="return doclink('link-615', '_re_search', 'link-467');">_re_search</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1145"></a><tt class="py-lineno">1145</tt>  <tt class="py-line">            <tt class="py-string">r"Expect\(?(\d*)\)? = +([0-9.e\-|\+]+)"</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">,</tt> </tt>
<a name="L1146"></a><tt class="py-lineno">1146</tt>  <tt class="py-line">            <tt class="py-string">"I could not find the expect in line\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1147"></a><tt class="py-lineno">1147</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-616" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-616', 'x', 'link-322');">x</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1148"></a><tt class="py-lineno">1148</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt class="py-name">num_alignments</tt> <tt class="py-op">=</tt> <tt id="link-617" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-617', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-618" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-618', 'x', 'link-322');">x</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1149"></a><tt class="py-lineno">1149</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1150"></a><tt class="py-lineno">1150</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt class="py-name">num_alignments</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L1151"></a><tt class="py-lineno">1151</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt class="py-name">expect</tt> <tt class="py-op">=</tt> <tt id="link-619" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_float" class="py-name" href="#" onclick="return doclink('link-619', '_safe_float', 'link-540');">_safe_float</a></tt><tt class="py-op">(</tt><tt id="link-620" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-620', 'y', 'link-614');">y</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1152"></a><tt class="py-lineno">1152</tt>  <tt class="py-line"> </tt>
<a name="_HSPConsumer.identities"></a><div id="_HSPConsumer.identities-def"><a name="L1153"></a><tt class="py-lineno">1153</tt> <a class="py-toggle" href="#" id="_HSPConsumer.identities-toggle" onclick="return toggle('_HSPConsumer.identities');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._HSPConsumer-class.html#identities">identities</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_HSPConsumer.identities-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_HSPConsumer.identities-expanded"><a name="L1154"></a><tt class="py-lineno">1154</tt>  <tt class="py-line">        <tt id="link-621" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-621', 'x', 'link-322');">x</a></tt><tt class="py-op">,</tt> <tt id="link-622" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-622', 'y', 'link-614');">y</a></tt> <tt class="py-op">=</tt> <tt id="link-623" class="py-name"><a title="Bio.Blast.NCBIStandalone._re_search" class="py-name" href="#" onclick="return doclink('link-623', '_re_search', 'link-467');">_re_search</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1155"></a><tt class="py-lineno">1155</tt>  <tt class="py-line">            <tt class="py-string">r"Identities = (\d+)\/(\d+)"</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">,</tt> </tt>
<a name="L1156"></a><tt class="py-lineno">1156</tt>  <tt class="py-line">            <tt class="py-string">"I could not find the identities in line\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1157"></a><tt class="py-lineno">1157</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt id="link-624" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.identities" class="py-name" href="#" onclick="return doclink('link-624', 'identities', 'link-225');">identities</a></tt> <tt class="py-op">=</tt> <tt id="link-625" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-625', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-626" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-626', 'x', 'link-322');">x</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-627" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-627', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-628" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-628', 'y', 'link-614');">y</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1158"></a><tt class="py-lineno">1158</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt class="py-name">align_length</tt> <tt class="py-op">=</tt> <tt id="link-629" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-629', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-630" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-630', 'y', 'link-614');">y</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1159"></a><tt class="py-lineno">1159</tt>  <tt class="py-line"> </tt>
<a name="L1160"></a><tt class="py-lineno">1160</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">'Positives'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L1161"></a><tt class="py-lineno">1161</tt>  <tt class="py-line">            <tt id="link-631" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-631', 'x', 'link-322');">x</a></tt><tt class="py-op">,</tt> <tt id="link-632" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-632', 'y', 'link-614');">y</a></tt> <tt class="py-op">=</tt> <tt id="link-633" class="py-name"><a title="Bio.Blast.NCBIStandalone._re_search" class="py-name" href="#" onclick="return doclink('link-633', '_re_search', 'link-467');">_re_search</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1162"></a><tt class="py-lineno">1162</tt>  <tt class="py-line">                <tt class="py-string">r"Positives = (\d+)\/(\d+)"</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">,</tt> </tt>
<a name="L1163"></a><tt class="py-lineno">1163</tt>  <tt class="py-line">                <tt class="py-string">"I could not find the positives in line\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1164"></a><tt class="py-lineno">1164</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt class="py-name">positives</tt> <tt class="py-op">=</tt> <tt id="link-634" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-634', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-635" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-635', 'x', 'link-322');">x</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-636" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-636', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-637" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-637', 'y', 'link-614');">y</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1165"></a><tt class="py-lineno">1165</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt class="py-name">align_length</tt> <tt class="py-op">==</tt> <tt id="link-638" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-638', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-639" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-639', 'y', 'link-614');">y</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1166"></a><tt class="py-lineno">1166</tt>  <tt class="py-line"> </tt>
<a name="L1167"></a><tt class="py-lineno">1167</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">'Gaps'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L1168"></a><tt class="py-lineno">1168</tt>  <tt class="py-line">            <tt id="link-640" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-640', 'x', 'link-322');">x</a></tt><tt class="py-op">,</tt> <tt id="link-641" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-641', 'y', 'link-614');">y</a></tt> <tt class="py-op">=</tt> <tt id="link-642" class="py-name"><a title="Bio.Blast.NCBIStandalone._re_search" class="py-name" href="#" onclick="return doclink('link-642', '_re_search', 'link-467');">_re_search</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1169"></a><tt class="py-lineno">1169</tt>  <tt class="py-line">                <tt class="py-string">r"Gaps = (\d+)\/(\d+)"</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">,</tt> </tt>
<a name="L1170"></a><tt class="py-lineno">1170</tt>  <tt class="py-line">                <tt class="py-string">"I could not find the gaps in line\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1171"></a><tt class="py-lineno">1171</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt class="py-name">gaps</tt> <tt class="py-op">=</tt> <tt id="link-643" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-643', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-644" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-644', 'x', 'link-322');">x</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-645" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-645', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-646" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-646', 'y', 'link-614');">y</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1172"></a><tt class="py-lineno">1172</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt class="py-name">align_length</tt> <tt class="py-op">==</tt> <tt id="link-647" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-647', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-648" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-648', 'y', 'link-614');">y</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1173"></a><tt class="py-lineno">1173</tt>  <tt class="py-line"> </tt>
<a name="_HSPConsumer.strand"></a><div id="_HSPConsumer.strand-def"><a name="L1174"></a><tt class="py-lineno">1174</tt> <a class="py-toggle" href="#" id="_HSPConsumer.strand-toggle" onclick="return toggle('_HSPConsumer.strand');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._HSPConsumer-class.html#strand">strand</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_HSPConsumer.strand-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_HSPConsumer.strand-expanded"><a name="L1175"></a><tt class="py-lineno">1175</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt id="link-649" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-649', 'strand', 'link-228');">strand</a></tt> <tt class="py-op">=</tt> <tt id="link-650" class="py-name"><a title="Bio.Blast.NCBIStandalone._re_search" class="py-name" href="#" onclick="return doclink('link-650', '_re_search', 'link-467');">_re_search</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1176"></a><tt class="py-lineno">1176</tt>  <tt class="py-line">            <tt class="py-string">r"Strand\s?=\s?(\w+)\s?/\s?(\w+)"</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">,</tt> </tt>
<a name="L1177"></a><tt class="py-lineno">1177</tt>  <tt class="py-line">            <tt class="py-string">"I could not find the strand in line\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1178"></a><tt class="py-lineno">1178</tt>  <tt class="py-line"> </tt>
<a name="_HSPConsumer.frame"></a><div id="_HSPConsumer.frame-def"><a name="L1179"></a><tt class="py-lineno">1179</tt> <a class="py-toggle" href="#" id="_HSPConsumer.frame-toggle" onclick="return toggle('_HSPConsumer.frame');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._HSPConsumer-class.html#frame">frame</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_HSPConsumer.frame-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_HSPConsumer.frame-expanded"><a name="L1180"></a><tt class="py-lineno">1180</tt>  <tt class="py-line">        <tt class="py-comment"># Frame can be in formats:</tt> </tt>
<a name="L1181"></a><tt class="py-lineno">1181</tt>  <tt class="py-line">        <tt class="py-comment"># Frame = +1</tt> </tt>
<a name="L1182"></a><tt class="py-lineno">1182</tt>  <tt class="py-line">        <tt class="py-comment"># Frame = +2 / +2</tt> </tt>
<a name="L1183"></a><tt class="py-lineno">1183</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">'/'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L1184"></a><tt class="py-lineno">1184</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt id="link-651" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.frame" class="py-name" href="#" onclick="return doclink('link-651', 'frame', 'link-231');">frame</a></tt> <tt class="py-op">=</tt> <tt id="link-652" class="py-name"><a title="Bio.Blast.NCBIStandalone._re_search" class="py-name" href="#" onclick="return doclink('link-652', '_re_search', 'link-467');">_re_search</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1185"></a><tt class="py-lineno">1185</tt>  <tt class="py-line">                <tt class="py-string">r"Frame\s?=\s?([-+][123])\s?/\s?([-+][123])"</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">,</tt> </tt>
<a name="L1186"></a><tt class="py-lineno">1186</tt>  <tt class="py-line">                <tt class="py-string">"I could not find the frame in line\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1187"></a><tt class="py-lineno">1187</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1188"></a><tt class="py-lineno">1188</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt id="link-653" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.frame" class="py-name" href="#" onclick="return doclink('link-653', 'frame', 'link-231');">frame</a></tt> <tt class="py-op">=</tt> <tt id="link-654" class="py-name"><a title="Bio.Blast.NCBIStandalone._re_search" class="py-name" href="#" onclick="return doclink('link-654', '_re_search', 'link-467');">_re_search</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1189"></a><tt class="py-lineno">1189</tt>  <tt class="py-line">                <tt class="py-string">r"Frame = ([-+][123])"</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">,</tt> </tt>
<a name="L1190"></a><tt class="py-lineno">1190</tt>  <tt class="py-line">                <tt class="py-string">"I could not find the frame in line\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1191"></a><tt class="py-lineno">1191</tt>  <tt class="py-line"> </tt>
<a name="L1192"></a><tt class="py-lineno">1192</tt>  <tt class="py-line">    <tt class="py-comment"># Match a space, if one is available.  Masahir Ishikawa found a</tt> </tt>
<a name="L1193"></a><tt class="py-lineno">1193</tt>  <tt class="py-line">    <tt class="py-comment"># case where there's no space between the start and the sequence:</tt> </tt>
<a name="L1194"></a><tt class="py-lineno">1194</tt>  <tt class="py-line">    <tt class="py-comment"># Query: 100tt 101</tt> </tt>
<a name="L1195"></a><tt class="py-lineno">1195</tt>  <tt class="py-line">    <tt class="py-comment"># line below modified by Yair Benita, Sep 2004</tt> </tt>
<a name="L1196"></a><tt class="py-lineno">1196</tt>  <tt class="py-line">    <tt class="py-comment"># Note that the colon is not always present. 2006</tt> </tt>
<a name="L1197"></a><tt class="py-lineno">1197</tt>  <tt class="py-line">    <tt id="link-655" class="py-name" targets="Variable Bio.Blast.NCBIStandalone._HSPConsumer._query_re=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#_query_re"><a title="Bio.Blast.NCBIStandalone._HSPConsumer._query_re" class="py-name" href="#" onclick="return doclink('link-655', '_query_re', 'link-655');">_query_re</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r"Query(:?) \s*(\d+)\s*(.+) (\d+)"</tt><tt class="py-op">)</tt> </tt>
<a name="L1198"></a><tt class="py-lineno">1198</tt>  <tt class="py-line"> </tt>
<a name="_HSPConsumer.query"></a><div id="_HSPConsumer.query-def"><a name="L1199"></a><tt class="py-lineno">1199</tt> <a class="py-toggle" href="#" id="_HSPConsumer.query-toggle" onclick="return toggle('_HSPConsumer.query');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._HSPConsumer-class.html#query">query</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_HSPConsumer.query-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_HSPConsumer.query-expanded"><a name="L1200"></a><tt class="py-lineno">1200</tt>  <tt class="py-line">        <tt class="py-name">m</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-656" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer._query_re" class="py-name" href="#" onclick="return doclink('link-656', '_query_re', 'link-655');">_query_re</a></tt><tt class="py-op">.</tt><tt id="link-657" class="py-name"><a title="Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-657', 'search', 'link-102');">search</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1201"></a><tt class="py-lineno">1201</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">m</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1202"></a><tt class="py-lineno">1202</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-658" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-658', 'strip', 'link-66');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-string">"Query        ------------------------------------------------------------"</tt><tt class="py-op">:</tt> </tt>
<a name="L1203"></a><tt class="py-lineno">1203</tt>  <tt class="py-line">                <tt class="py-comment"># Special case - long gap relative to the subject,</tt> </tt>
<a name="L1204"></a><tt class="py-lineno">1204</tt>  <tt class="py-line">                <tt class="py-comment"># note there is no start/end present, cannot update those</tt> </tt>
<a name="L1205"></a><tt class="py-lineno">1205</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt id="link-659" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.query
Bio.SearchIO._model.hsp.HSP.query
Bio.SearchIO._model.hsp.HSPFragment.query
Bio.SearchIO._model.query" class="py-name" href="#" onclick="return doclink('link-659', 'query', 'link-239');">query</a></tt> <tt class="py-op">+=</tt> <tt class="py-string">"-"</tt> <tt class="py-op">*</tt> <tt class="py-number">60</tt> </tt>
<a name="L1206"></a><tt class="py-lineno">1206</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_query_len</tt> <tt class="py-op">=</tt> <tt class="py-number">60</tt>  <tt class="py-comment"># number of dashes</tt> </tt>
<a name="L1207"></a><tt class="py-lineno">1207</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_query_start_index</tt> <tt class="py-op">=</tt> <tt class="py-number">13</tt>  <tt class="py-comment"># offset of first dash</tt> </tt>
<a name="L1208"></a><tt class="py-lineno">1208</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> </tt>
<a name="L1209"></a><tt class="py-lineno">1209</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"I could not find the query in line\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1210"></a><tt class="py-lineno">1210</tt>  <tt class="py-line"> </tt>
<a name="L1211"></a><tt class="py-lineno">1211</tt>  <tt class="py-line">        <tt class="py-comment"># line below modified by Yair Benita, Sep 2004.</tt> </tt>
<a name="L1212"></a><tt class="py-lineno">1212</tt>  <tt class="py-line">        <tt class="py-comment"># added the end attribute for the query</tt> </tt>
<a name="L1213"></a><tt class="py-lineno">1213</tt>  <tt class="py-line">        <tt class="py-name">colon</tt><tt class="py-op">,</tt> <tt id="link-660" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-660', 'start', 'link-35');">start</a></tt><tt class="py-op">,</tt> <tt id="link-661" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-661', 'seq', 'link-561');">seq</a></tt><tt class="py-op">,</tt> <tt id="link-662" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-662', 'end', 'link-78');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">m</tt><tt class="py-op">.</tt><tt class="py-name">groups</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1214"></a><tt class="py-lineno">1214</tt>  <tt class="py-line">        <tt id="link-663" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-663', 'seq', 'link-561');">seq</a></tt> <tt class="py-op">=</tt> <tt id="link-664" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-664', 'seq', 'link-561');">seq</a></tt><tt class="py-op">.</tt><tt id="link-665" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-665', 'strip', 'link-66');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1215"></a><tt class="py-lineno">1215</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt id="link-666" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.query
Bio.SearchIO._model.hsp.HSP.query
Bio.SearchIO._model.hsp.HSPFragment.query
Bio.SearchIO._model.query" class="py-name" href="#" onclick="return doclink('link-666', 'query', 'link-239');">query</a></tt> <tt class="py-op">+=</tt> <tt id="link-667" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-667', 'seq', 'link-561');">seq</a></tt> </tt>
<a name="L1216"></a><tt class="py-lineno">1216</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt id="link-668" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.query_start=Bio.SearchIO._model.hsp.HSP-class.html#query_start,Variable Bio.SearchIO._model.hsp.HSPFragment.query_start=Bio.SearchIO._model.hsp.HSPFragment-class.html#query_start"><a title="Bio.SearchIO._model.hsp.HSP.query_start
Bio.SearchIO._model.hsp.HSPFragment.query_start" class="py-name" href="#" onclick="return doclink('link-668', 'query_start', 'link-668');">query_start</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1217"></a><tt class="py-lineno">1217</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt id="link-669" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_start
Bio.SearchIO._model.hsp.HSPFragment.query_start" class="py-name" href="#" onclick="return doclink('link-669', 'query_start', 'link-668');">query_start</a></tt> <tt class="py-op">=</tt> <tt id="link-670" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-670', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-671" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-671', 'start', 'link-35');">start</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1218"></a><tt class="py-lineno">1218</tt>  <tt class="py-line"> </tt>
<a name="L1219"></a><tt class="py-lineno">1219</tt>  <tt class="py-line">        <tt class="py-comment"># line below added by Yair Benita, Sep 2004.</tt> </tt>
<a name="L1220"></a><tt class="py-lineno">1220</tt>  <tt class="py-line">        <tt class="py-comment"># added the end attribute for the query</tt> </tt>
<a name="L1221"></a><tt class="py-lineno">1221</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt id="link-672" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.query_end=Bio.SearchIO._model.hsp.HSP-class.html#query_end,Variable Bio.SearchIO._model.hsp.HSPFragment.query_end=Bio.SearchIO._model.hsp.HSPFragment-class.html#query_end"><a title="Bio.SearchIO._model.hsp.HSP.query_end
Bio.SearchIO._model.hsp.HSPFragment.query_end" class="py-name" href="#" onclick="return doclink('link-672', 'query_end', 'link-672');">query_end</a></tt> <tt class="py-op">=</tt> <tt id="link-673" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-673', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-674" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-674', 'end', 'link-78');">end</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1222"></a><tt class="py-lineno">1222</tt>  <tt class="py-line"> </tt>
<a name="L1223"></a><tt class="py-lineno">1223</tt>  <tt class="py-line">        <tt class="py-comment"># Get index for sequence start (regular expression element 3)</tt> </tt>
<a name="L1224"></a><tt class="py-lineno">1224</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_query_start_index</tt> <tt class="py-op">=</tt> <tt class="py-name">m</tt><tt class="py-op">.</tt><tt id="link-675" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-675', 'start', 'link-35');">start</a></tt><tt class="py-op">(</tt><tt class="py-number">3</tt><tt class="py-op">)</tt> </tt>
<a name="L1225"></a><tt class="py-lineno">1225</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_query_len</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-676" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-676', 'seq', 'link-561');">seq</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1226"></a><tt class="py-lineno">1226</tt>  <tt class="py-line"> </tt>
<a name="_HSPConsumer.align"></a><div id="_HSPConsumer.align-def"><a name="L1227"></a><tt class="py-lineno">1227</tt> <a class="py-toggle" href="#" id="_HSPConsumer.align-toggle" onclick="return toggle('_HSPConsumer.align');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._HSPConsumer-class.html#align">align</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_HSPConsumer.align-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_HSPConsumer.align-expanded"><a name="L1228"></a><tt class="py-lineno">1228</tt>  <tt class="py-line">        <tt id="link-677" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-677', 'seq', 'link-561');">seq</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_query_start_index</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-678" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-678', 'rstrip', 'link-471');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1229"></a><tt class="py-lineno">1229</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-679" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-679', 'seq', 'link-561');">seq</a></tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_query_len</tt><tt class="py-op">:</tt> </tt>
<a name="L1230"></a><tt class="py-lineno">1230</tt>  <tt class="py-line">            <tt class="py-comment"># Make sure the alignment is the same length as the query</tt> </tt>
<a name="L1231"></a><tt class="py-lineno">1231</tt>  <tt class="py-line">            <tt id="link-680" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-680', 'seq', 'link-561');">seq</a></tt> <tt class="py-op">+=</tt> <tt class="py-string">' '</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_query_len</tt> <tt class="py-op">-</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-681" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-681', 'seq', 'link-561');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1232"></a><tt class="py-lineno">1232</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-682" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-682', 'seq', 'link-561');">seq</a></tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_query_len</tt><tt class="py-op">:</tt> </tt>
<a name="L1233"></a><tt class="py-lineno">1233</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Match is longer than the query in line\n%s"</tt> </tt>
<a name="L1234"></a><tt class="py-lineno">1234</tt>  <tt class="py-line">                             <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1235"></a><tt class="py-lineno">1235</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt id="link-683" class="py-name" targets="Function Bio.triefind.match()=Bio.triefind-module.html#match"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-683', 'match', 'link-683');">match</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt id="link-684" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-684', 'match', 'link-683');">match</a></tt> <tt class="py-op">+</tt> <tt id="link-685" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-685', 'seq', 'link-561');">seq</a></tt> </tt>
</div><a name="L1236"></a><tt class="py-lineno">1236</tt>  <tt class="py-line"> </tt>
<a name="L1237"></a><tt class="py-lineno">1237</tt>  <tt class="py-line">    <tt class="py-comment"># To match how we do the query, cache the regular expression.</tt> </tt>
<a name="L1238"></a><tt class="py-lineno">1238</tt>  <tt class="py-line">    <tt class="py-comment"># Note that the colon is not always present.</tt> </tt>
<a name="L1239"></a><tt class="py-lineno">1239</tt>  <tt class="py-line">    <tt id="link-686" class="py-name" targets="Variable Bio.Blast.NCBIStandalone._HSPConsumer._sbjct_re=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#_sbjct_re"><a title="Bio.Blast.NCBIStandalone._HSPConsumer._sbjct_re" class="py-name" href="#" onclick="return doclink('link-686', '_sbjct_re', 'link-686');">_sbjct_re</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r"Sbjct(:?) \s*(\d+)\s*(.+) (\d+)"</tt><tt class="py-op">)</tt> </tt>
<a name="L1240"></a><tt class="py-lineno">1240</tt>  <tt class="py-line"> </tt>
<a name="_HSPConsumer.sbjct"></a><div id="_HSPConsumer.sbjct-def"><a name="L1241"></a><tt class="py-lineno">1241</tt> <a class="py-toggle" href="#" id="_HSPConsumer.sbjct-toggle" onclick="return toggle('_HSPConsumer.sbjct');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._HSPConsumer-class.html#sbjct">sbjct</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_HSPConsumer.sbjct-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_HSPConsumer.sbjct-expanded"><a name="L1242"></a><tt class="py-lineno">1242</tt>  <tt class="py-line">        <tt class="py-name">m</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-687" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer._sbjct_re" class="py-name" href="#" onclick="return doclink('link-687', '_sbjct_re', 'link-686');">_sbjct_re</a></tt><tt class="py-op">.</tt><tt id="link-688" class="py-name"><a title="Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-688', 'search', 'link-102');">search</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1243"></a><tt class="py-lineno">1243</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">m</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1244"></a><tt class="py-lineno">1244</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"I could not find the sbjct in line\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1245"></a><tt class="py-lineno">1245</tt>  <tt class="py-line">        <tt class="py-name">colon</tt><tt class="py-op">,</tt> <tt id="link-689" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-689', 'start', 'link-35');">start</a></tt><tt class="py-op">,</tt> <tt id="link-690" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-690', 'seq', 'link-561');">seq</a></tt><tt class="py-op">,</tt> <tt id="link-691" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-691', 'end', 'link-78');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">m</tt><tt class="py-op">.</tt><tt class="py-name">groups</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1246"></a><tt class="py-lineno">1246</tt>  <tt class="py-line">        <tt class="py-comment"># mikep 26/9/00</tt> </tt>
<a name="L1247"></a><tt class="py-lineno">1247</tt>  <tt class="py-line">        <tt class="py-comment"># On occasion, there is a blast hit with no subject match</tt> </tt>
<a name="L1248"></a><tt class="py-lineno">1248</tt>  <tt class="py-line">        <tt class="py-comment"># so far, it only occurs with 1-line short "matches"</tt> </tt>
<a name="L1249"></a><tt class="py-lineno">1249</tt>  <tt class="py-line">        <tt class="py-comment"># I have decided to let these pass as they appear</tt> </tt>
<a name="L1250"></a><tt class="py-lineno">1250</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-692" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-692', 'seq', 'link-561');">seq</a></tt><tt class="py-op">.</tt><tt id="link-693" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-693', 'strip', 'link-66');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1251"></a><tt class="py-lineno">1251</tt>  <tt class="py-line">            <tt id="link-694" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-694', 'seq', 'link-561');">seq</a></tt> <tt class="py-op">=</tt> <tt class="py-string">' '</tt> <tt class="py-op">*</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_query_len</tt> </tt>
<a name="L1252"></a><tt class="py-lineno">1252</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1253"></a><tt class="py-lineno">1253</tt>  <tt class="py-line">            <tt id="link-695" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-695', 'seq', 'link-561');">seq</a></tt> <tt class="py-op">=</tt> <tt id="link-696" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-696', 'seq', 'link-561');">seq</a></tt><tt class="py-op">.</tt><tt id="link-697" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-697', 'strip', 'link-66');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1254"></a><tt class="py-lineno">1254</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt id="link-698" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.sbjct" class="py-name" href="#" onclick="return doclink('link-698', 'sbjct', 'link-245');">sbjct</a></tt> <tt class="py-op">+=</tt> <tt id="link-699" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-699', 'seq', 'link-561');">seq</a></tt> </tt>
<a name="L1255"></a><tt class="py-lineno">1255</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt class="py-name">sbjct_start</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1256"></a><tt class="py-lineno">1256</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt class="py-name">sbjct_start</tt> <tt class="py-op">=</tt> <tt id="link-700" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-700', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-701" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-701', 'start', 'link-35');">start</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1257"></a><tt class="py-lineno">1257</tt>  <tt class="py-line"> </tt>
<a name="L1258"></a><tt class="py-lineno">1258</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt class="py-name">sbjct_end</tt> <tt class="py-op">=</tt> <tt id="link-702" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-702', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-703" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-703', 'end', 'link-78');">end</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1259"></a><tt class="py-lineno">1259</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-704" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-704', 'seq', 'link-561');">seq</a></tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_query_len</tt><tt class="py-op">:</tt> </tt>
<a name="L1260"></a><tt class="py-lineno">1260</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt> </tt>
<a name="L1261"></a><tt class="py-lineno">1261</tt>  <tt class="py-line">                  <tt class="py-string">"QUERY and SBJCT sequence lengths don't match (%i %r vs %i) in line\n%s"</tt> </tt>
<a name="L1262"></a><tt class="py-lineno">1262</tt>  <tt class="py-line">                  <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_query_len</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">.</tt><tt id="link-705" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.query
Bio.SearchIO._model.hsp.HSP.query
Bio.SearchIO._model.hsp.HSPFragment.query
Bio.SearchIO._model.query" class="py-name" href="#" onclick="return doclink('link-705', 'query', 'link-239');">query</a></tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-706" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-706', 'seq', 'link-561');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1263"></a><tt class="py-lineno">1263</tt>  <tt class="py-line"> </tt>
<a name="L1264"></a><tt class="py-lineno">1264</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_query_start_index</tt>   <tt class="py-comment"># clean up unused variables</tt> </tt>
<a name="L1265"></a><tt class="py-lineno">1265</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_query_len</tt> </tt>
</div><a name="L1266"></a><tt class="py-lineno">1266</tt>  <tt class="py-line"> </tt>
<a name="_HSPConsumer.end_hsp"></a><div id="_HSPConsumer.end_hsp-def"><a name="L1267"></a><tt class="py-lineno">1267</tt> <a class="py-toggle" href="#" id="_HSPConsumer.end_hsp-toggle" onclick="return toggle('_HSPConsumer.end_hsp');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._HSPConsumer-class.html#end_hsp">end_hsp</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_HSPConsumer.end_hsp-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_HSPConsumer.end_hsp-expanded"><a name="L1268"></a><tt class="py-lineno">1268</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div></div><a name="L1269"></a><tt class="py-lineno">1269</tt>  <tt class="py-line"> </tt>
<a name="L1270"></a><tt class="py-lineno">1270</tt>  <tt class="py-line"> </tt>
<a name="_DatabaseReportConsumer"></a><div id="_DatabaseReportConsumer-def"><a name="L1271"></a><tt class="py-lineno">1271</tt> <a class="py-toggle" href="#" id="_DatabaseReportConsumer-toggle" onclick="return toggle('_DatabaseReportConsumer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._DatabaseReportConsumer-class.html">_DatabaseReportConsumer</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_DatabaseReportConsumer-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_DatabaseReportConsumer-expanded"><a name="L1272"></a><tt class="py-lineno">1272</tt>  <tt class="py-line"> </tt>
<a name="_DatabaseReportConsumer.start_database_report"></a><div id="_DatabaseReportConsumer.start_database_report-def"><a name="L1273"></a><tt class="py-lineno">1273</tt> <a class="py-toggle" href="#" id="_DatabaseReportConsumer.start_database_report-toggle" onclick="return toggle('_DatabaseReportConsumer.start_database_report');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._DatabaseReportConsumer-class.html#start_database_report">start_database_report</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_DatabaseReportConsumer.start_database_report-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_DatabaseReportConsumer.start_database_report-expanded"><a name="L1274"></a><tt class="py-lineno">1274</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_dr</tt> <tt class="py-op">=</tt> <tt id="link-707" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-707', 'Record', 'link-11');">Record</a></tt><tt class="py-op">.</tt><tt id="link-708" class="py-name" targets="Class Bio.Blast.Record.DatabaseReport=Bio.Blast.Record.DatabaseReport-class.html"><a title="Bio.Blast.Record.DatabaseReport" class="py-name" href="#" onclick="return doclink('link-708', 'DatabaseReport', 'link-708');">DatabaseReport</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1275"></a><tt class="py-lineno">1275</tt>  <tt class="py-line"> </tt>
<a name="_DatabaseReportConsumer.database"></a><div id="_DatabaseReportConsumer.database-def"><a name="L1276"></a><tt class="py-lineno">1276</tt> <a class="py-toggle" href="#" id="_DatabaseReportConsumer.database-toggle" onclick="return toggle('_DatabaseReportConsumer.database');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._DatabaseReportConsumer-class.html#database">database</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_DatabaseReportConsumer.database-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_DatabaseReportConsumer.database-expanded"><a name="L1277"></a><tt class="py-lineno">1277</tt>  <tt class="py-line">        <tt class="py-name">m</tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-709" class="py-name"><a title="Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-709', 'search', 'link-102');">search</a></tt><tt class="py-op">(</tt><tt class="py-string">r"Database: (.+)$"</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1278"></a><tt class="py-lineno">1278</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">m</tt><tt class="py-op">:</tt> </tt>
<a name="L1279"></a><tt class="py-lineno">1279</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_dr</tt><tt class="py-op">.</tt><tt class="py-name">database_name</tt><tt class="py-op">.</tt><tt id="link-710" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-710', 'append', 'link-506');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">m</tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1280"></a><tt class="py-lineno">1280</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_dr</tt><tt class="py-op">.</tt><tt class="py-name">database_name</tt><tt class="py-op">:</tt> </tt>
<a name="L1281"></a><tt class="py-lineno">1281</tt>  <tt class="py-line">            <tt class="py-comment"># This must be a continuation of the previous name.</tt> </tt>
<a name="L1282"></a><tt class="py-lineno">1282</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_dr</tt><tt class="py-op">.</tt><tt class="py-name">database_name</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">"%s%s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_dr</tt><tt class="py-op">.</tt><tt class="py-name">database_name</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L1283"></a><tt class="py-lineno">1283</tt>  <tt class="py-line">                                                   <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-711" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-711', 'strip', 'link-66');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1284"></a><tt class="py-lineno">1284</tt>  <tt class="py-line"> </tt>
<a name="_DatabaseReportConsumer.posted_date"></a><div id="_DatabaseReportConsumer.posted_date-def"><a name="L1285"></a><tt class="py-lineno">1285</tt> <a class="py-toggle" href="#" id="_DatabaseReportConsumer.posted_date-toggle" onclick="return toggle('_DatabaseReportConsumer.posted_date');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._DatabaseReportConsumer-class.html#posted_date">posted_date</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_DatabaseReportConsumer.posted_date-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_DatabaseReportConsumer.posted_date-expanded"><a name="L1286"></a><tt class="py-lineno">1286</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_dr</tt><tt class="py-op">.</tt><tt id="link-712" class="py-name"><a title="Bio.Blast.NCBIStandalone._DatabaseReportConsumer.posted_date" class="py-name" href="#" onclick="return doclink('link-712', 'posted_date', 'link-292');">posted_date</a></tt><tt class="py-op">.</tt><tt id="link-713" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-713', 'append', 'link-506');">append</a></tt><tt class="py-op">(</tt><tt id="link-714" class="py-name"><a title="Bio.Blast.NCBIStandalone._re_search" class="py-name" href="#" onclick="return doclink('link-714', '_re_search', 'link-467');">_re_search</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1287"></a><tt class="py-lineno">1287</tt>  <tt class="py-line">            <tt class="py-string">r"Posted date:\s*(.+)$"</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">,</tt> </tt>
<a name="L1288"></a><tt class="py-lineno">1288</tt>  <tt class="py-line">            <tt class="py-string">"I could not find the posted date in line\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1289"></a><tt class="py-lineno">1289</tt>  <tt class="py-line"> </tt>
<a name="_DatabaseReportConsumer.num_letters_in_database"></a><div id="_DatabaseReportConsumer.num_letters_in_database-def"><a name="L1290"></a><tt class="py-lineno">1290</tt> <a class="py-toggle" href="#" id="_DatabaseReportConsumer.num_letters_in_database-toggle" onclick="return toggle('_DatabaseReportConsumer.num_letters_in_database');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._DatabaseReportConsumer-class.html#num_letters_in_database">num_letters_in_database</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_DatabaseReportConsumer.num_letters_in_database-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_DatabaseReportConsumer.num_letters_in_database-expanded"><a name="L1291"></a><tt class="py-lineno">1291</tt>  <tt class="py-line">        <tt id="link-715" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-715', 'letters', 'link-466');">letters</a></tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-716" class="py-name" targets="Function Bio.Blast.NCBIStandalone._get_cols()=Bio.Blast.NCBIStandalone-module.html#_get_cols"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-716', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1292"></a><tt class="py-lineno">1292</tt>  <tt class="py-line">            <tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">6</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">2</tt><tt class="py-op">:</tt> <tt class="py-string">"letters"</tt><tt class="py-op">,</tt> <tt class="py-number">4</tt><tt class="py-op">:</tt> <tt class="py-string">"database:"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1293"></a><tt class="py-lineno">1293</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_dr</tt><tt class="py-op">.</tt><tt id="link-717" class="py-name"><a title="Bio.Blast.NCBIStandalone._DatabaseReportConsumer.num_letters_in_database" class="py-name" href="#" onclick="return doclink('link-717', 'num_letters_in_database', 'link-295');">num_letters_in_database</a></tt><tt class="py-op">.</tt><tt id="link-718" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-718', 'append', 'link-506');">append</a></tt><tt class="py-op">(</tt><tt id="link-719" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-719', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-720" class="py-name"><a title="Bio.Alphabet.Alphabet.letters
Bio.Alphabet.IUPAC.ExtendedIUPACDNA.letters
Bio.Alphabet.IUPAC.ExtendedIUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACAmbiguousRNA.letters
Bio.Alphabet.IUPAC.IUPACProtein.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousDNA.letters
Bio.Alphabet.IUPAC.IUPACUnambiguousRNA.letters
Bio.Alphabet.Reduced.HPModel.letters
Bio.Alphabet.Reduced.Murphy10.letters
Bio.Alphabet.Reduced.Murphy15.letters
Bio.Alphabet.Reduced.Murphy4.letters
Bio.Alphabet.Reduced.Murphy8.letters
Bio.Alphabet.Reduced.PC5.letters
Bio.Alphabet.SecondaryStructure.letters
Bio.Alphabet.SingleLetterAlphabet.letters
Bio.Alphabet.ThreeLetterProtein.letters
Bio.NeuralNetwork.Gene.Schema.SchemaDNAAlphabet.letters
Bio.codonalign.codonalphabet.CodonAlphabet.letters" class="py-name" href="#" onclick="return doclink('link-720', 'letters', 'link-466');">letters</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1294"></a><tt class="py-lineno">1294</tt>  <tt class="py-line"> </tt>
<a name="_DatabaseReportConsumer.num_sequences_in_database"></a><div id="_DatabaseReportConsumer.num_sequences_in_database-def"><a name="L1295"></a><tt class="py-lineno">1295</tt> <a class="py-toggle" href="#" id="_DatabaseReportConsumer.num_sequences_in_database-toggle" onclick="return toggle('_DatabaseReportConsumer.num_sequences_in_database');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._DatabaseReportConsumer-class.html#num_sequences_in_database">num_sequences_in_database</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_DatabaseReportConsumer.num_sequences_in_database-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_DatabaseReportConsumer.num_sequences_in_database-expanded"><a name="L1296"></a><tt class="py-lineno">1296</tt>  <tt class="py-line">        <tt class="py-name">sequences</tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-721" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-721', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1297"></a><tt class="py-lineno">1297</tt>  <tt class="py-line">            <tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">6</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">2</tt><tt class="py-op">:</tt> <tt class="py-string">"sequences"</tt><tt class="py-op">,</tt> <tt class="py-number">4</tt><tt class="py-op">:</tt> <tt class="py-string">"database:"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1298"></a><tt class="py-lineno">1298</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_dr</tt><tt class="py-op">.</tt><tt id="link-722" class="py-name"><a title="Bio.Blast.NCBIStandalone._DatabaseReportConsumer.num_sequences_in_database" class="py-name" href="#" onclick="return doclink('link-722', 'num_sequences_in_database', 'link-298');">num_sequences_in_database</a></tt><tt class="py-op">.</tt><tt id="link-723" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-723', 'append', 'link-506');">append</a></tt><tt class="py-op">(</tt><tt id="link-724" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-724', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt class="py-name">sequences</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1299"></a><tt class="py-lineno">1299</tt>  <tt class="py-line"> </tt>
<a name="_DatabaseReportConsumer.ka_params"></a><div id="_DatabaseReportConsumer.ka_params-def"><a name="L1300"></a><tt class="py-lineno">1300</tt> <a class="py-toggle" href="#" id="_DatabaseReportConsumer.ka_params-toggle" onclick="return toggle('_DatabaseReportConsumer.ka_params');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._DatabaseReportConsumer-class.html#ka_params">ka_params</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_DatabaseReportConsumer.ka_params-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_DatabaseReportConsumer.ka_params-expanded"><a name="L1301"></a><tt class="py-lineno">1301</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_dr</tt><tt class="py-op">.</tt><tt id="link-725" class="py-name"><a title="Bio.Blast.NCBIStandalone._DatabaseReportConsumer.ka_params" class="py-name" href="#" onclick="return doclink('link-725', 'ka_params', 'link-309');">ka_params</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-726" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_float" class="py-name" href="#" onclick="return doclink('link-726', '_safe_float', 'link-540');">_safe_float</a></tt><tt class="py-op">(</tt><tt id="link-727" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-727', 'x', 'link-322');">x</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-728" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-728', 'x', 'link-322');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-729" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-729', 'split', 'link-440');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
</div><a name="L1302"></a><tt class="py-lineno">1302</tt>  <tt class="py-line"> </tt>
<a name="_DatabaseReportConsumer.gapped"></a><div id="_DatabaseReportConsumer.gapped-def"><a name="L1303"></a><tt class="py-lineno">1303</tt> <a class="py-toggle" href="#" id="_DatabaseReportConsumer.gapped-toggle" onclick="return toggle('_DatabaseReportConsumer.gapped');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._DatabaseReportConsumer-class.html#gapped">gapped</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_DatabaseReportConsumer.gapped-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_DatabaseReportConsumer.gapped-expanded"><a name="L1304"></a><tt class="py-lineno">1304</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_dr</tt><tt class="py-op">.</tt><tt id="link-730" class="py-name"><a title="Bio.Blast.NCBIStandalone._DatabaseReportConsumer.gapped" class="py-name" href="#" onclick="return doclink('link-730', 'gapped', 'link-313');">gapped</a></tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
</div><a name="L1305"></a><tt class="py-lineno">1305</tt>  <tt class="py-line"> </tt>
<a name="_DatabaseReportConsumer.ka_params_gap"></a><div id="_DatabaseReportConsumer.ka_params_gap-def"><a name="L1306"></a><tt class="py-lineno">1306</tt> <a class="py-toggle" href="#" id="_DatabaseReportConsumer.ka_params_gap-toggle" onclick="return toggle('_DatabaseReportConsumer.ka_params_gap');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._DatabaseReportConsumer-class.html#ka_params_gap">ka_params_gap</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_DatabaseReportConsumer.ka_params_gap-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_DatabaseReportConsumer.ka_params_gap-expanded"><a name="L1307"></a><tt class="py-lineno">1307</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_dr</tt><tt class="py-op">.</tt><tt id="link-731" class="py-name"><a title="Bio.Blast.NCBIStandalone._DatabaseReportConsumer.ka_params_gap" class="py-name" href="#" onclick="return doclink('link-731', 'ka_params_gap', 'link-319');">ka_params_gap</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-732" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_float" class="py-name" href="#" onclick="return doclink('link-732', '_safe_float', 'link-540');">_safe_float</a></tt><tt class="py-op">(</tt><tt id="link-733" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-733', 'x', 'link-322');">x</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-734" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-734', 'x', 'link-322');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-735" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-735', 'split', 'link-440');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
</div><a name="L1308"></a><tt class="py-lineno">1308</tt>  <tt class="py-line"> </tt>
<a name="_DatabaseReportConsumer.end_database_report"></a><div id="_DatabaseReportConsumer.end_database_report-def"><a name="L1309"></a><tt class="py-lineno">1309</tt> <a class="py-toggle" href="#" id="_DatabaseReportConsumer.end_database_report-toggle" onclick="return toggle('_DatabaseReportConsumer.end_database_report');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._DatabaseReportConsumer-class.html#end_database_report">end_database_report</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_DatabaseReportConsumer.end_database_report-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_DatabaseReportConsumer.end_database_report-expanded"><a name="L1310"></a><tt class="py-lineno">1310</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div></div><a name="L1311"></a><tt class="py-lineno">1311</tt>  <tt class="py-line"> </tt>
<a name="L1312"></a><tt class="py-lineno">1312</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer"></a><div id="_ParametersConsumer-def"><a name="L1313"></a><tt class="py-lineno">1313</tt> <a class="py-toggle" href="#" id="_ParametersConsumer-toggle" onclick="return toggle('_ParametersConsumer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html">_ParametersConsumer</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_ParametersConsumer-expanded"><a name="_ParametersConsumer.start_parameters"></a><div id="_ParametersConsumer.start_parameters-def"><a name="L1314"></a><tt class="py-lineno">1314</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.start_parameters-toggle" onclick="return toggle('_ParametersConsumer.start_parameters');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#start_parameters">start_parameters</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.start_parameters-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.start_parameters-expanded"><a name="L1315"></a><tt class="py-lineno">1315</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt> <tt class="py-op">=</tt> <tt id="link-736" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-736', 'Record', 'link-11');">Record</a></tt><tt class="py-op">.</tt><tt id="link-737" class="py-name" targets="Class Bio.Blast.Record.Parameters=Bio.Blast.Record.Parameters-class.html"><a title="Bio.Blast.Record.Parameters" class="py-name" href="#" onclick="return doclink('link-737', 'Parameters', 'link-737');">Parameters</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1316"></a><tt class="py-lineno">1316</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.matrix"></a><div id="_ParametersConsumer.matrix-def"><a name="L1317"></a><tt class="py-lineno">1317</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.matrix-toggle" onclick="return toggle('_ParametersConsumer.matrix');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#matrix">matrix</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.matrix-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.matrix-expanded"><a name="L1318"></a><tt class="py-lineno">1318</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-738" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.matrix
Bio.motifs.matrix" class="py-name" href="#" onclick="return doclink('link-738', 'matrix', 'link-329');">matrix</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">8</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-739" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-739', 'rstrip', 'link-471');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1319"></a><tt class="py-lineno">1319</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.gap_penalties"></a><div id="_ParametersConsumer.gap_penalties-def"><a name="L1320"></a><tt class="py-lineno">1320</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.gap_penalties-toggle" onclick="return toggle('_ParametersConsumer.gap_penalties');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#gap_penalties">gap_penalties</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.gap_penalties-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.gap_penalties-expanded"><a name="L1321"></a><tt class="py-lineno">1321</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-740" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.gap_penalties" class="py-name" href="#" onclick="return doclink('link-740', 'gap_penalties', 'link-332');">gap_penalties</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-741" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_float" class="py-name" href="#" onclick="return doclink('link-741', '_safe_float', 'link-540');">_safe_float</a></tt><tt class="py-op">(</tt><tt id="link-742" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-742', 'x', 'link-322');">x</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-743" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-743', 'x', 'link-322');">x</a></tt> <tt class="py-keyword">in</tt> <tt id="link-744" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-744', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1322"></a><tt class="py-lineno">1322</tt>  <tt class="py-line">            <tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-number">3</tt><tt class="py-op">,</tt> <tt class="py-number">5</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">6</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">2</tt><tt class="py-op">:</tt> <tt class="py-string">"Existence:"</tt><tt class="py-op">,</tt> <tt class="py-number">4</tt><tt class="py-op">:</tt> <tt class="py-string">"Extension:"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
</div><a name="L1323"></a><tt class="py-lineno">1323</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.num_hits"></a><div id="_ParametersConsumer.num_hits-def"><a name="L1324"></a><tt class="py-lineno">1324</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.num_hits-toggle" onclick="return toggle('_ParametersConsumer.num_hits');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#num_hits">num_hits</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.num_hits-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.num_hits-expanded"><a name="L1325"></a><tt class="py-lineno">1325</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">'1st pass'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L1326"></a><tt class="py-lineno">1326</tt>  <tt class="py-line">            <tt id="link-745" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-745', 'x', 'link-322');">x</a></tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-746" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-746', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">4</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">11</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">2</tt><tt class="py-op">:</tt> <tt class="py-string">"Hits"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1327"></a><tt class="py-lineno">1327</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-747" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.num_hits" class="py-name" href="#" onclick="return doclink('link-747', 'num_hits', 'link-338');">num_hits</a></tt> <tt class="py-op">=</tt> <tt id="link-748" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-748', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-749" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-749', 'x', 'link-322');">x</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1328"></a><tt class="py-lineno">1328</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1329"></a><tt class="py-lineno">1329</tt>  <tt class="py-line">            <tt id="link-750" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-750', 'x', 'link-322');">x</a></tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-751" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-751', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">6</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">2</tt><tt class="py-op">:</tt> <tt class="py-string">"Hits"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1330"></a><tt class="py-lineno">1330</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-752" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.num_hits" class="py-name" href="#" onclick="return doclink('link-752', 'num_hits', 'link-338');">num_hits</a></tt> <tt class="py-op">=</tt> <tt id="link-753" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-753', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-754" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-754', 'x', 'link-322');">x</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1331"></a><tt class="py-lineno">1331</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.num_sequences"></a><div id="_ParametersConsumer.num_sequences-def"><a name="L1332"></a><tt class="py-lineno">1332</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.num_sequences-toggle" onclick="return toggle('_ParametersConsumer.num_sequences');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#num_sequences">num_sequences</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.num_sequences-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.num_sequences-expanded"><a name="L1333"></a><tt class="py-lineno">1333</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">'1st pass'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L1334"></a><tt class="py-lineno">1334</tt>  <tt class="py-line">            <tt id="link-755" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-755', 'x', 'link-322');">x</a></tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-756" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-756', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">4</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">9</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">2</tt><tt class="py-op">:</tt> <tt class="py-string">"Sequences:"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1335"></a><tt class="py-lineno">1335</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-757" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.num_sequences" class="py-name" href="#" onclick="return doclink('link-757', 'num_sequences', 'link-335');">num_sequences</a></tt> <tt class="py-op">=</tt> <tt id="link-758" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-758', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-759" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-759', 'x', 'link-322');">x</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1336"></a><tt class="py-lineno">1336</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1337"></a><tt class="py-lineno">1337</tt>  <tt class="py-line">            <tt id="link-760" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-760', 'x', 'link-322');">x</a></tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-761" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-761', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">4</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">2</tt><tt class="py-op">:</tt> <tt class="py-string">"Sequences:"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1338"></a><tt class="py-lineno">1338</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-762" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.num_sequences" class="py-name" href="#" onclick="return doclink('link-762', 'num_sequences', 'link-335');">num_sequences</a></tt> <tt class="py-op">=</tt> <tt id="link-763" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-763', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-764" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-764', 'x', 'link-322');">x</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1339"></a><tt class="py-lineno">1339</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.num_extends"></a><div id="_ParametersConsumer.num_extends-def"><a name="L1340"></a><tt class="py-lineno">1340</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.num_extends-toggle" onclick="return toggle('_ParametersConsumer.num_extends');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#num_extends">num_extends</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.num_extends-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.num_extends-expanded"><a name="L1341"></a><tt class="py-lineno">1341</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">'1st pass'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L1342"></a><tt class="py-lineno">1342</tt>  <tt class="py-line">            <tt id="link-765" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-765', 'x', 'link-322');">x</a></tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-766" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-766', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">4</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">9</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">2</tt><tt class="py-op">:</tt> <tt class="py-string">"extensions:"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1343"></a><tt class="py-lineno">1343</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-767" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.num_extends" class="py-name" href="#" onclick="return doclink('link-767', 'num_extends', 'link-344');">num_extends</a></tt> <tt class="py-op">=</tt> <tt id="link-768" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-768', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-769" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-769', 'x', 'link-322');">x</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1344"></a><tt class="py-lineno">1344</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1345"></a><tt class="py-lineno">1345</tt>  <tt class="py-line">            <tt id="link-770" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-770', 'x', 'link-322');">x</a></tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-771" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-771', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">4</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">2</tt><tt class="py-op">:</tt> <tt class="py-string">"extensions:"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1346"></a><tt class="py-lineno">1346</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-772" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.num_extends" class="py-name" href="#" onclick="return doclink('link-772', 'num_extends', 'link-344');">num_extends</a></tt> <tt class="py-op">=</tt> <tt id="link-773" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-773', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-774" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-774', 'x', 'link-322');">x</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1347"></a><tt class="py-lineno">1347</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.num_good_extends"></a><div id="_ParametersConsumer.num_good_extends-def"><a name="L1348"></a><tt class="py-lineno">1348</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.num_good_extends-toggle" onclick="return toggle('_ParametersConsumer.num_good_extends');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#num_good_extends">num_good_extends</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.num_good_extends-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.num_good_extends-expanded"><a name="L1349"></a><tt class="py-lineno">1349</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">'1st pass'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L1350"></a><tt class="py-lineno">1350</tt>  <tt class="py-line">            <tt id="link-775" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-775', 'x', 'link-322');">x</a></tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-776" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-776', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">4</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">10</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">3</tt><tt class="py-op">:</tt> <tt class="py-string">"extensions:"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1351"></a><tt class="py-lineno">1351</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-777" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.num_good_extends" class="py-name" href="#" onclick="return doclink('link-777', 'num_good_extends', 'link-347');">num_good_extends</a></tt> <tt class="py-op">=</tt> <tt id="link-778" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-778', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-779" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-779', 'x', 'link-322');">x</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1352"></a><tt class="py-lineno">1352</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1353"></a><tt class="py-lineno">1353</tt>  <tt class="py-line">            <tt id="link-780" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-780', 'x', 'link-322');">x</a></tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-781" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-781', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">5</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">3</tt><tt class="py-op">:</tt> <tt class="py-string">"extensions:"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1354"></a><tt class="py-lineno">1354</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-782" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.num_good_extends" class="py-name" href="#" onclick="return doclink('link-782', 'num_good_extends', 'link-347');">num_good_extends</a></tt> <tt class="py-op">=</tt> <tt id="link-783" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-783', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-784" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-784', 'x', 'link-322');">x</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1355"></a><tt class="py-lineno">1355</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.num_seqs_better_e"></a><div id="_ParametersConsumer.num_seqs_better_e-def"><a name="L1356"></a><tt class="py-lineno">1356</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.num_seqs_better_e-toggle" onclick="return toggle('_ParametersConsumer.num_seqs_better_e');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#num_seqs_better_e">num_seqs_better_e</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.num_seqs_better_e-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.num_seqs_better_e-expanded"><a name="L1357"></a><tt class="py-lineno">1357</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-785" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.num_seqs_better_e" class="py-name" href="#" onclick="return doclink('link-785', 'num_seqs_better_e', 'link-350');">num_seqs_better_e</a></tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-786" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-786', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1358"></a><tt class="py-lineno">1358</tt>  <tt class="py-line">            <tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">7</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">2</tt><tt class="py-op">:</tt> <tt class="py-string">"sequences"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1359"></a><tt class="py-lineno">1359</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-787" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.num_seqs_better_e" class="py-name" href="#" onclick="return doclink('link-787', 'num_seqs_better_e', 'link-350');">num_seqs_better_e</a></tt> <tt class="py-op">=</tt> <tt id="link-788" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-788', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1360"></a><tt class="py-lineno">1360</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-789" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.num_seqs_better_e" class="py-name" href="#" onclick="return doclink('link-789', 'num_seqs_better_e', 'link-350');">num_seqs_better_e</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1361"></a><tt class="py-lineno">1361</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.hsps_no_gap"></a><div id="_ParametersConsumer.hsps_no_gap-def"><a name="L1362"></a><tt class="py-lineno">1362</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.hsps_no_gap-toggle" onclick="return toggle('_ParametersConsumer.hsps_no_gap');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#hsps_no_gap">hsps_no_gap</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.hsps_no_gap-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.hsps_no_gap-expanded"><a name="L1363"></a><tt class="py-lineno">1363</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-790" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.hsps_no_gap" class="py-name" href="#" onclick="return doclink('link-790', 'hsps_no_gap', 'link-353');">hsps_no_gap</a></tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-791" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-791', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1364"></a><tt class="py-lineno">1364</tt>  <tt class="py-line">            <tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">9</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">3</tt><tt class="py-op">:</tt> <tt class="py-string">"better"</tt><tt class="py-op">,</tt> <tt class="py-number">7</tt><tt class="py-op">:</tt> <tt class="py-string">"gapping:"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1365"></a><tt class="py-lineno">1365</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-792" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.hsps_no_gap" class="py-name" href="#" onclick="return doclink('link-792', 'hsps_no_gap', 'link-353');">hsps_no_gap</a></tt> <tt class="py-op">=</tt> <tt id="link-793" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-793', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-794" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.hsps_no_gap" class="py-name" href="#" onclick="return doclink('link-794', 'hsps_no_gap', 'link-353');">hsps_no_gap</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1366"></a><tt class="py-lineno">1366</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.hsps_prelim_gapped"></a><div id="_ParametersConsumer.hsps_prelim_gapped-def"><a name="L1367"></a><tt class="py-lineno">1367</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.hsps_prelim_gapped-toggle" onclick="return toggle('_ParametersConsumer.hsps_prelim_gapped');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#hsps_prelim_gapped">hsps_prelim_gapped</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.hsps_prelim_gapped-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.hsps_prelim_gapped-expanded"><a name="L1368"></a><tt class="py-lineno">1368</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-795" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.hsps_prelim_gapped" class="py-name" href="#" onclick="return doclink('link-795', 'hsps_prelim_gapped', 'link-365');">hsps_prelim_gapped</a></tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-796" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-796', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1369"></a><tt class="py-lineno">1369</tt>  <tt class="py-line">            <tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">9</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">4</tt><tt class="py-op">:</tt> <tt class="py-string">"gapped"</tt><tt class="py-op">,</tt> <tt class="py-number">6</tt><tt class="py-op">:</tt> <tt class="py-string">"prelim"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1370"></a><tt class="py-lineno">1370</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-797" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.hsps_prelim_gapped" class="py-name" href="#" onclick="return doclink('link-797', 'hsps_prelim_gapped', 'link-365');">hsps_prelim_gapped</a></tt> <tt class="py-op">=</tt> <tt id="link-798" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-798', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1371"></a><tt class="py-lineno">1371</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-799" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.hsps_prelim_gapped" class="py-name" href="#" onclick="return doclink('link-799', 'hsps_prelim_gapped', 'link-365');">hsps_prelim_gapped</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1372"></a><tt class="py-lineno">1372</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.hsps_prelim_gapped_attempted"></a><div id="_ParametersConsumer.hsps_prelim_gapped_attempted-def"><a name="L1373"></a><tt class="py-lineno">1373</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.hsps_prelim_gapped_attempted-toggle" onclick="return toggle('_ParametersConsumer.hsps_prelim_gapped_attempted');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#hsps_prelim_gapped_attempted">hsps_prelim_gapped_attempted</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.hsps_prelim_gapped_attempted-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.hsps_prelim_gapped_attempted-expanded"><a name="L1374"></a><tt class="py-lineno">1374</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-800" class="py-name" targets="Method Bio.Blast.NCBIStandalone._ParametersConsumer.hsps_prelim_gapped_attempted()=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#hsps_prelim_gapped_attempted"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.hsps_prelim_gapped_attempted" class="py-name" href="#" onclick="return doclink('link-800', 'hsps_prelim_gapped_attempted', 'link-800');">hsps_prelim_gapped_attempted</a></tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-801" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-801', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1375"></a><tt class="py-lineno">1375</tt>  <tt class="py-line">            <tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">10</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">4</tt><tt class="py-op">:</tt> <tt class="py-string">"attempted"</tt><tt class="py-op">,</tt> <tt class="py-number">7</tt><tt class="py-op">:</tt> <tt class="py-string">"prelim"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1376"></a><tt class="py-lineno">1376</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-802" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.hsps_prelim_gapped_attempted" class="py-name" href="#" onclick="return doclink('link-802', 'hsps_prelim_gapped_attempted', 'link-800');">hsps_prelim_gapped_attempted</a></tt> <tt class="py-op">=</tt> <tt id="link-803" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-803', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1377"></a><tt class="py-lineno">1377</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-804" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.hsps_prelim_gapped_attempted" class="py-name" href="#" onclick="return doclink('link-804', 'hsps_prelim_gapped_attempted', 'link-800');">hsps_prelim_gapped_attempted</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1378"></a><tt class="py-lineno">1378</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.hsps_gapped"></a><div id="_ParametersConsumer.hsps_gapped-def"><a name="L1379"></a><tt class="py-lineno">1379</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.hsps_gapped-toggle" onclick="return toggle('_ParametersConsumer.hsps_gapped');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#hsps_gapped">hsps_gapped</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.hsps_gapped-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.hsps_gapped-expanded"><a name="L1380"></a><tt class="py-lineno">1380</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-805" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.hsps_gapped" class="py-name" href="#" onclick="return doclink('link-805', 'hsps_gapped', 'link-370');">hsps_gapped</a></tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-806" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-806', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1381"></a><tt class="py-lineno">1381</tt>  <tt class="py-line">            <tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">6</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">3</tt><tt class="py-op">:</tt> <tt class="py-string">"gapped"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1382"></a><tt class="py-lineno">1382</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-807" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.hsps_gapped" class="py-name" href="#" onclick="return doclink('link-807', 'hsps_gapped', 'link-370');">hsps_gapped</a></tt> <tt class="py-op">=</tt> <tt id="link-808" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-808', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-809" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.hsps_gapped" class="py-name" href="#" onclick="return doclink('link-809', 'hsps_gapped', 'link-370');">hsps_gapped</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1383"></a><tt class="py-lineno">1383</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.query_length"></a><div id="_ParametersConsumer.query_length-def"><a name="L1384"></a><tt class="py-lineno">1384</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.query_length-toggle" onclick="return toggle('_ParametersConsumer.query_length');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#query_length">query_length</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.query_length-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.query_length-expanded"><a name="L1385"></a><tt class="py-lineno">1385</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-810" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.query_length" class="py-name" href="#" onclick="return doclink('link-810', 'query_length', 'link-379');">query_length</a></tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-811" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-811', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1386"></a><tt class="py-lineno">1386</tt>  <tt class="py-line">            <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-812" class="py-name" targets="Method Bio.Seq.Seq.lower()=Bio.Seq.Seq-class.html#lower,Method Bio.Seq.UnknownSeq.lower()=Bio.Seq.UnknownSeq-class.html#lower,Method Bio.SeqRecord.SeqRecord.lower()=Bio.SeqRecord.SeqRecord-class.html#lower"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-812', 'lower', 'link-812');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">4</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">0</tt><tt class="py-op">:</tt> <tt class="py-string">"length"</tt><tt class="py-op">,</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> <tt class="py-string">"query:"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1387"></a><tt class="py-lineno">1387</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-813" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.query_length" class="py-name" href="#" onclick="return doclink('link-813', 'query_length', 'link-379');">query_length</a></tt> <tt class="py-op">=</tt> <tt id="link-814" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-814', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-815" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.query_length" class="py-name" href="#" onclick="return doclink('link-815', 'query_length', 'link-379');">query_length</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1388"></a><tt class="py-lineno">1388</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.database_length"></a><div id="_ParametersConsumer.database_length-def"><a name="L1389"></a><tt class="py-lineno">1389</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.database_length-toggle" onclick="return toggle('_ParametersConsumer.database_length');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#database_length">database_length</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.database_length-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.database_length-expanded"><a name="L1390"></a><tt class="py-lineno">1390</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-816" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.database_length" class="py-name" href="#" onclick="return doclink('link-816', 'database_length', 'link-381');">database_length</a></tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-817" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-817', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1391"></a><tt class="py-lineno">1391</tt>  <tt class="py-line">            <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-818" class="py-name"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-818', 'lower', 'link-812');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">4</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">0</tt><tt class="py-op">:</tt> <tt class="py-string">"length"</tt><tt class="py-op">,</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> <tt class="py-string">"database:"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1392"></a><tt class="py-lineno">1392</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-819" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.database_length" class="py-name" href="#" onclick="return doclink('link-819', 'database_length', 'link-381');">database_length</a></tt> <tt class="py-op">=</tt> <tt id="link-820" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-820', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-821" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.database_length" class="py-name" href="#" onclick="return doclink('link-821', 'database_length', 'link-381');">database_length</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1393"></a><tt class="py-lineno">1393</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.effective_hsp_length"></a><div id="_ParametersConsumer.effective_hsp_length-def"><a name="L1394"></a><tt class="py-lineno">1394</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.effective_hsp_length-toggle" onclick="return toggle('_ParametersConsumer.effective_hsp_length');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#effective_hsp_length">effective_hsp_length</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.effective_hsp_length-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.effective_hsp_length-expanded"><a name="L1395"></a><tt class="py-lineno">1395</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-822" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.effective_hsp_length" class="py-name" href="#" onclick="return doclink('link-822', 'effective_hsp_length', 'link-386');">effective_hsp_length</a></tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-823" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-823', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1396"></a><tt class="py-lineno">1396</tt>  <tt class="py-line">            <tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">4</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> <tt class="py-string">"HSP"</tt><tt class="py-op">,</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> <tt class="py-string">"length:"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1397"></a><tt class="py-lineno">1397</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-824" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.effective_hsp_length" class="py-name" href="#" onclick="return doclink('link-824', 'effective_hsp_length', 'link-386');">effective_hsp_length</a></tt> <tt class="py-op">=</tt> <tt id="link-825" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-825', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1398"></a><tt class="py-lineno">1398</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-826" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.effective_hsp_length" class="py-name" href="#" onclick="return doclink('link-826', 'effective_hsp_length', 'link-386');">effective_hsp_length</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1399"></a><tt class="py-lineno">1399</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.effective_query_length"></a><div id="_ParametersConsumer.effective_query_length-def"><a name="L1400"></a><tt class="py-lineno">1400</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.effective_query_length-toggle" onclick="return toggle('_ParametersConsumer.effective_query_length');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#effective_query_length">effective_query_length</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.effective_query_length-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.effective_query_length-expanded"><a name="L1401"></a><tt class="py-lineno">1401</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-827" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.effective_query_length" class="py-name" href="#" onclick="return doclink('link-827', 'effective_query_length', 'link-389');">effective_query_length</a></tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-828" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-828', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1402"></a><tt class="py-lineno">1402</tt>  <tt class="py-line">            <tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">5</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> <tt class="py-string">"length"</tt><tt class="py-op">,</tt> <tt class="py-number">3</tt><tt class="py-op">:</tt> <tt class="py-string">"query:"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1403"></a><tt class="py-lineno">1403</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-829" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.effective_query_length" class="py-name" href="#" onclick="return doclink('link-829', 'effective_query_length', 'link-389');">effective_query_length</a></tt> <tt class="py-op">=</tt> <tt id="link-830" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-830', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1404"></a><tt class="py-lineno">1404</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-831" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.effective_query_length" class="py-name" href="#" onclick="return doclink('link-831', 'effective_query_length', 'link-389');">effective_query_length</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1405"></a><tt class="py-lineno">1405</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.effective_database_length"></a><div id="_ParametersConsumer.effective_database_length-def"><a name="L1406"></a><tt class="py-lineno">1406</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.effective_database_length-toggle" onclick="return toggle('_ParametersConsumer.effective_database_length');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#effective_database_length">effective_database_length</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.effective_database_length-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.effective_database_length-expanded"><a name="L1407"></a><tt class="py-lineno">1407</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-832" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.effective_database_length" class="py-name" href="#" onclick="return doclink('link-832', 'effective_database_length', 'link-391');">effective_database_length</a></tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-833" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-833', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1408"></a><tt class="py-lineno">1408</tt>  <tt class="py-line">            <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-834" class="py-name"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-834', 'lower', 'link-812');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">5</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> <tt class="py-string">"length"</tt><tt class="py-op">,</tt> <tt class="py-number">3</tt><tt class="py-op">:</tt> <tt class="py-string">"database:"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1409"></a><tt class="py-lineno">1409</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-835" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.effective_database_length" class="py-name" href="#" onclick="return doclink('link-835', 'effective_database_length', 'link-391');">effective_database_length</a></tt> <tt class="py-op">=</tt> <tt id="link-836" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-836', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1410"></a><tt class="py-lineno">1410</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-837" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.effective_database_length" class="py-name" href="#" onclick="return doclink('link-837', 'effective_database_length', 'link-391');">effective_database_length</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1411"></a><tt class="py-lineno">1411</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.effective_search_space"></a><div id="_ParametersConsumer.effective_search_space-def"><a name="L1412"></a><tt class="py-lineno">1412</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.effective_search_space-toggle" onclick="return toggle('_ParametersConsumer.effective_search_space');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#effective_search_space">effective_search_space</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.effective_search_space-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.effective_search_space-expanded"><a name="L1413"></a><tt class="py-lineno">1413</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-838" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.effective_search_space" class="py-name" href="#" onclick="return doclink('link-838', 'effective_search_space', 'link-393');">effective_search_space</a></tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-839" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-839', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1414"></a><tt class="py-lineno">1414</tt>  <tt class="py-line">            <tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">4</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> <tt class="py-string">"search"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1415"></a><tt class="py-lineno">1415</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-840" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.effective_search_space" class="py-name" href="#" onclick="return doclink('link-840', 'effective_search_space', 'link-393');">effective_search_space</a></tt> <tt class="py-op">=</tt> <tt id="link-841" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-841', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1416"></a><tt class="py-lineno">1416</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-842" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.effective_search_space" class="py-name" href="#" onclick="return doclink('link-842', 'effective_search_space', 'link-393');">effective_search_space</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1417"></a><tt class="py-lineno">1417</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.effective_search_space_used"></a><div id="_ParametersConsumer.effective_search_space_used-def"><a name="L1418"></a><tt class="py-lineno">1418</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.effective_search_space_used-toggle" onclick="return toggle('_ParametersConsumer.effective_search_space_used');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#effective_search_space_used">effective_search_space_used</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.effective_search_space_used-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.effective_search_space_used-expanded"><a name="L1419"></a><tt class="py-lineno">1419</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-843" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.effective_search_space_used" class="py-name" href="#" onclick="return doclink('link-843', 'effective_search_space_used', 'link-395');">effective_search_space_used</a></tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-844" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-844', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1420"></a><tt class="py-lineno">1420</tt>  <tt class="py-line">            <tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">5</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> <tt class="py-string">"search"</tt><tt class="py-op">,</tt> <tt class="py-number">3</tt><tt class="py-op">:</tt> <tt class="py-string">"used:"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1421"></a><tt class="py-lineno">1421</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-845" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.effective_search_space_used" class="py-name" href="#" onclick="return doclink('link-845', 'effective_search_space_used', 'link-395');">effective_search_space_used</a></tt> <tt class="py-op">=</tt> <tt id="link-846" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-846', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1422"></a><tt class="py-lineno">1422</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-847" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.effective_search_space_used" class="py-name" href="#" onclick="return doclink('link-847', 'effective_search_space_used', 'link-395');">effective_search_space_used</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1423"></a><tt class="py-lineno">1423</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.frameshift"></a><div id="_ParametersConsumer.frameshift-def"><a name="L1424"></a><tt class="py-lineno">1424</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.frameshift-toggle" onclick="return toggle('_ParametersConsumer.frameshift');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#frameshift">frameshift</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.frameshift-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.frameshift-expanded"><a name="L1425"></a><tt class="py-lineno">1425</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-848" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.frameshift" class="py-name" href="#" onclick="return doclink('link-848', 'frameshift', 'link-397');">frameshift</a></tt> <tt class="py-op">=</tt> <tt id="link-849" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-849', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1426"></a><tt class="py-lineno">1426</tt>  <tt class="py-line">           <tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-number">4</tt><tt class="py-op">,</tt> <tt class="py-number">5</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">6</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">0</tt><tt class="py-op">:</tt> <tt class="py-string">"frameshift"</tt><tt class="py-op">,</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> <tt class="py-string">"decay"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1427"></a><tt class="py-lineno">1427</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.threshold"></a><div id="_ParametersConsumer.threshold-def"><a name="L1428"></a><tt class="py-lineno">1428</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.threshold-toggle" onclick="return toggle('_ParametersConsumer.threshold');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#threshold">threshold</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.threshold-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.threshold-expanded"><a name="L1429"></a><tt class="py-lineno">1429</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"T:"</tt><tt class="py-op">:</tt> </tt>
<a name="L1430"></a><tt class="py-lineno">1430</tt>  <tt class="py-line">            <tt class="py-comment"># Assume its an old stlye line like "T: 123"</tt> </tt>
<a name="L1431"></a><tt class="py-lineno">1431</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-850" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.threshold" class="py-name" href="#" onclick="return doclink('link-850', 'threshold', 'link-400');">threshold</a></tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-851" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-851', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1432"></a><tt class="py-lineno">1432</tt>  <tt class="py-line">                <tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">2</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">0</tt><tt class="py-op">:</tt> <tt class="py-string">"T:"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1433"></a><tt class="py-lineno">1433</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">28</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"Neighboring words threshold:"</tt><tt class="py-op">:</tt> </tt>
<a name="L1434"></a><tt class="py-lineno">1434</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-852" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.threshold" class="py-name" href="#" onclick="return doclink('link-852', 'threshold', 'link-400');">threshold</a></tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-853" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-853', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1435"></a><tt class="py-lineno">1435</tt>  <tt class="py-line">                <tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-number">3</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">4</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">0</tt><tt class="py-op">:</tt> <tt class="py-string">"Neighboring"</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> <tt class="py-string">"words"</tt><tt class="py-op">,</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> <tt class="py-string">"threshold:"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1436"></a><tt class="py-lineno">1436</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1437"></a><tt class="py-lineno">1437</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unrecognised threshold line:\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1438"></a><tt class="py-lineno">1438</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-854" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.threshold" class="py-name" href="#" onclick="return doclink('link-854', 'threshold', 'link-400');">threshold</a></tt> <tt class="py-op">=</tt> <tt id="link-855" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-855', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-856" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.threshold" class="py-name" href="#" onclick="return doclink('link-856', 'threshold', 'link-400');">threshold</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1439"></a><tt class="py-lineno">1439</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.window_size"></a><div id="_ParametersConsumer.window_size-def"><a name="L1440"></a><tt class="py-lineno">1440</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.window_size-toggle" onclick="return toggle('_ParametersConsumer.window_size');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#window_size">window_size</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.window_size-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.window_size-expanded"><a name="L1441"></a><tt class="py-lineno">1441</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"A:"</tt><tt class="py-op">:</tt> </tt>
<a name="L1442"></a><tt class="py-lineno">1442</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-857" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.window_size" class="py-name" href="#" onclick="return doclink('link-857', 'window_size', 'link-406');">window_size</a></tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-858" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-858', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1443"></a><tt class="py-lineno">1443</tt>  <tt class="py-line">                <tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">2</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">0</tt><tt class="py-op">:</tt> <tt class="py-string">"A:"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1444"></a><tt class="py-lineno">1444</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">25</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"Window for multiple hits:"</tt><tt class="py-op">:</tt> </tt>
<a name="L1445"></a><tt class="py-lineno">1445</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-859" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.window_size" class="py-name" href="#" onclick="return doclink('link-859', 'window_size', 'link-406');">window_size</a></tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-860" class="py-name"><a title="Bio.Blast.NCBIStandalone._get_cols" class="py-name" href="#" onclick="return doclink('link-860', '_get_cols', 'link-716');">_get_cols</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1446"></a><tt class="py-lineno">1446</tt>  <tt class="py-line">                <tt class="py-name">line</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-number">4</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">ncols</tt><tt class="py-op">=</tt><tt class="py-number">5</tt><tt class="py-op">,</tt> <tt class="py-name">expected</tt><tt class="py-op">=</tt><tt class="py-op">{</tt><tt class="py-number">0</tt><tt class="py-op">:</tt> <tt class="py-string">"Window"</tt><tt class="py-op">,</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> <tt class="py-string">"multiple"</tt><tt class="py-op">,</tt> <tt class="py-number">3</tt><tt class="py-op">:</tt> <tt class="py-string">"hits:"</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L1447"></a><tt class="py-lineno">1447</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1448"></a><tt class="py-lineno">1448</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unrecognised window size line:\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1449"></a><tt class="py-lineno">1449</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-861" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.window_size" class="py-name" href="#" onclick="return doclink('link-861', 'window_size', 'link-406');">window_size</a></tt> <tt class="py-op">=</tt> <tt id="link-862" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-862', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-863" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.window_size" class="py-name" href="#" onclick="return doclink('link-863', 'window_size', 'link-406');">window_size</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1450"></a><tt class="py-lineno">1450</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.dropoff_1st_pass"></a><div id="_ParametersConsumer.dropoff_1st_pass-def"><a name="L1451"></a><tt class="py-lineno">1451</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.dropoff_1st_pass-toggle" onclick="return toggle('_ParametersConsumer.dropoff_1st_pass');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#dropoff_1st_pass">dropoff_1st_pass</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.dropoff_1st_pass-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.dropoff_1st_pass-expanded"><a name="L1452"></a><tt class="py-lineno">1452</tt>  <tt class="py-line">        <tt id="link-864" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-864', 'score', 'link-222');">score</a></tt><tt class="py-op">,</tt> <tt class="py-name">bits</tt> <tt class="py-op">=</tt> <tt id="link-865" class="py-name"><a title="Bio.Blast.NCBIStandalone._re_search" class="py-name" href="#" onclick="return doclink('link-865', '_re_search', 'link-467');">_re_search</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1453"></a><tt class="py-lineno">1453</tt>  <tt class="py-line">            <tt class="py-string">r"X1: (\d+) \(\s*([0-9,.]+) bits\)"</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">,</tt> </tt>
<a name="L1454"></a><tt class="py-lineno">1454</tt>  <tt class="py-line">            <tt class="py-string">"I could not find the dropoff in line\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1455"></a><tt class="py-lineno">1455</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-866" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.dropoff_1st_pass" class="py-name" href="#" onclick="return doclink('link-866', 'dropoff_1st_pass', 'link-412');">dropoff_1st_pass</a></tt> <tt class="py-op">=</tt> <tt id="link-867" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-867', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-868" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-868', 'score', 'link-222');">score</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-869" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_float" class="py-name" href="#" onclick="return doclink('link-869', '_safe_float', 'link-540');">_safe_float</a></tt><tt class="py-op">(</tt><tt class="py-name">bits</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1456"></a><tt class="py-lineno">1456</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.gap_x_dropoff"></a><div id="_ParametersConsumer.gap_x_dropoff-def"><a name="L1457"></a><tt class="py-lineno">1457</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.gap_x_dropoff-toggle" onclick="return toggle('_ParametersConsumer.gap_x_dropoff');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#gap_x_dropoff">gap_x_dropoff</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.gap_x_dropoff-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.gap_x_dropoff-expanded"><a name="L1458"></a><tt class="py-lineno">1458</tt>  <tt class="py-line">        <tt id="link-870" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-870', 'score', 'link-222');">score</a></tt><tt class="py-op">,</tt> <tt class="py-name">bits</tt> <tt class="py-op">=</tt> <tt id="link-871" class="py-name"><a title="Bio.Blast.NCBIStandalone._re_search" class="py-name" href="#" onclick="return doclink('link-871', '_re_search', 'link-467');">_re_search</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1459"></a><tt class="py-lineno">1459</tt>  <tt class="py-line">            <tt class="py-string">r"X2: (\d+) \(\s*([0-9,.]+) bits\)"</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">,</tt> </tt>
<a name="L1460"></a><tt class="py-lineno">1460</tt>  <tt class="py-line">            <tt class="py-string">"I could not find the gap dropoff in line\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1461"></a><tt class="py-lineno">1461</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-872" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.gap_x_dropoff" class="py-name" href="#" onclick="return doclink('link-872', 'gap_x_dropoff', 'link-415');">gap_x_dropoff</a></tt> <tt class="py-op">=</tt> <tt id="link-873" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-873', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-874" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-874', 'score', 'link-222');">score</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-875" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_float" class="py-name" href="#" onclick="return doclink('link-875', '_safe_float', 'link-540');">_safe_float</a></tt><tt class="py-op">(</tt><tt class="py-name">bits</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1462"></a><tt class="py-lineno">1462</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.gap_x_dropoff_final"></a><div id="_ParametersConsumer.gap_x_dropoff_final-def"><a name="L1463"></a><tt class="py-lineno">1463</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.gap_x_dropoff_final-toggle" onclick="return toggle('_ParametersConsumer.gap_x_dropoff_final');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#gap_x_dropoff_final">gap_x_dropoff_final</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.gap_x_dropoff_final-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.gap_x_dropoff_final-expanded"><a name="L1464"></a><tt class="py-lineno">1464</tt>  <tt class="py-line">        <tt id="link-876" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-876', 'score', 'link-222');">score</a></tt><tt class="py-op">,</tt> <tt class="py-name">bits</tt> <tt class="py-op">=</tt> <tt id="link-877" class="py-name"><a title="Bio.Blast.NCBIStandalone._re_search" class="py-name" href="#" onclick="return doclink('link-877', '_re_search', 'link-467');">_re_search</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1465"></a><tt class="py-lineno">1465</tt>  <tt class="py-line">            <tt class="py-string">r"X3: (\d+) \(\s*([0-9,.]+) bits\)"</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">,</tt> </tt>
<a name="L1466"></a><tt class="py-lineno">1466</tt>  <tt class="py-line">            <tt class="py-string">"I could not find the gap dropoff final in line\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1467"></a><tt class="py-lineno">1467</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-878" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.gap_x_dropoff_final" class="py-name" href="#" onclick="return doclink('link-878', 'gap_x_dropoff_final', 'link-418');">gap_x_dropoff_final</a></tt> <tt class="py-op">=</tt> <tt id="link-879" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-879', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-880" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-880', 'score', 'link-222');">score</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-881" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_float" class="py-name" href="#" onclick="return doclink('link-881', '_safe_float', 'link-540');">_safe_float</a></tt><tt class="py-op">(</tt><tt class="py-name">bits</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1468"></a><tt class="py-lineno">1468</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.gap_trigger"></a><div id="_ParametersConsumer.gap_trigger-def"><a name="L1469"></a><tt class="py-lineno">1469</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.gap_trigger-toggle" onclick="return toggle('_ParametersConsumer.gap_trigger');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#gap_trigger">gap_trigger</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.gap_trigger-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.gap_trigger-expanded"><a name="L1470"></a><tt class="py-lineno">1470</tt>  <tt class="py-line">        <tt id="link-882" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-882', 'score', 'link-222');">score</a></tt><tt class="py-op">,</tt> <tt class="py-name">bits</tt> <tt class="py-op">=</tt> <tt id="link-883" class="py-name"><a title="Bio.Blast.NCBIStandalone._re_search" class="py-name" href="#" onclick="return doclink('link-883', '_re_search', 'link-467');">_re_search</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1471"></a><tt class="py-lineno">1471</tt>  <tt class="py-line">            <tt class="py-string">r"S1: (\d+) \(\s*([0-9,.]+) bits\)"</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">,</tt> </tt>
<a name="L1472"></a><tt class="py-lineno">1472</tt>  <tt class="py-line">            <tt class="py-string">"I could not find the gap trigger in line\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1473"></a><tt class="py-lineno">1473</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-884" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.gap_trigger" class="py-name" href="#" onclick="return doclink('link-884', 'gap_trigger', 'link-421');">gap_trigger</a></tt> <tt class="py-op">=</tt> <tt id="link-885" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-885', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-886" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-886', 'score', 'link-222');">score</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-887" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_float" class="py-name" href="#" onclick="return doclink('link-887', '_safe_float', 'link-540');">_safe_float</a></tt><tt class="py-op">(</tt><tt class="py-name">bits</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1474"></a><tt class="py-lineno">1474</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.blast_cutoff"></a><div id="_ParametersConsumer.blast_cutoff-def"><a name="L1475"></a><tt class="py-lineno">1475</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.blast_cutoff-toggle" onclick="return toggle('_ParametersConsumer.blast_cutoff');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#blast_cutoff">blast_cutoff</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.blast_cutoff-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.blast_cutoff-expanded"><a name="L1476"></a><tt class="py-lineno">1476</tt>  <tt class="py-line">        <tt id="link-888" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-888', 'score', 'link-222');">score</a></tt><tt class="py-op">,</tt> <tt class="py-name">bits</tt> <tt class="py-op">=</tt> <tt id="link-889" class="py-name"><a title="Bio.Blast.NCBIStandalone._re_search" class="py-name" href="#" onclick="return doclink('link-889', '_re_search', 'link-467');">_re_search</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1477"></a><tt class="py-lineno">1477</tt>  <tt class="py-line">            <tt class="py-string">r"S2: (\d+) \(\s*([0-9,.]+) bits\)"</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">,</tt> </tt>
<a name="L1478"></a><tt class="py-lineno">1478</tt>  <tt class="py-line">            <tt class="py-string">"I could not find the blast cutoff in line\n%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1479"></a><tt class="py-lineno">1479</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt id="link-890" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer.blast_cutoff" class="py-name" href="#" onclick="return doclink('link-890', 'blast_cutoff', 'link-426');">blast_cutoff</a></tt> <tt class="py-op">=</tt> <tt id="link-891" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_int" class="py-name" href="#" onclick="return doclink('link-891', '_safe_int', 'link-459');">_safe_int</a></tt><tt class="py-op">(</tt><tt id="link-892" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.score" class="py-name" href="#" onclick="return doclink('link-892', 'score', 'link-222');">score</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-893" class="py-name"><a title="Bio.Blast.NCBIStandalone._safe_float" class="py-name" href="#" onclick="return doclink('link-893', '_safe_float', 'link-540');">_safe_float</a></tt><tt class="py-op">(</tt><tt class="py-name">bits</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1480"></a><tt class="py-lineno">1480</tt>  <tt class="py-line"> </tt>
<a name="_ParametersConsumer.end_parameters"></a><div id="_ParametersConsumer.end_parameters-def"><a name="L1481"></a><tt class="py-lineno">1481</tt> <a class="py-toggle" href="#" id="_ParametersConsumer.end_parameters-toggle" onclick="return toggle('_ParametersConsumer.end_parameters');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._ParametersConsumer-class.html#end_parameters">end_parameters</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_ParametersConsumer.end_parameters-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_ParametersConsumer.end_parameters-expanded"><a name="L1482"></a><tt class="py-lineno">1482</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
</div></div><a name="L1483"></a><tt class="py-lineno">1483</tt>  <tt class="py-line"> </tt>
<a name="L1484"></a><tt class="py-lineno">1484</tt>  <tt class="py-line"> </tt>
<a name="_BlastConsumer"></a><div id="_BlastConsumer-def"><a name="L1485"></a><tt class="py-lineno">1485</tt> <a class="py-toggle" href="#" id="_BlastConsumer-toggle" onclick="return toggle('_BlastConsumer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._BlastConsumer-class.html">_BlastConsumer</a><tt class="py-op">(</tt><tt class="py-base-class">AbstractConsumer</tt><tt class="py-op">,</tt> </tt>
<a name="L1486"></a><tt class="py-lineno">1486</tt>  <tt class="py-line">                     <tt class="py-base-class">_HeaderConsumer</tt><tt class="py-op">,</tt> </tt>
<a name="L1487"></a><tt class="py-lineno">1487</tt>  <tt class="py-line">                     <tt class="py-base-class">_DescriptionConsumer</tt><tt class="py-op">,</tt> </tt>
<a name="L1488"></a><tt class="py-lineno">1488</tt>  <tt class="py-line">                     <tt class="py-base-class">_AlignmentConsumer</tt><tt class="py-op">,</tt> </tt>
<a name="L1489"></a><tt class="py-lineno">1489</tt>  <tt class="py-line">                     <tt class="py-base-class">_HSPConsumer</tt><tt class="py-op">,</tt> </tt>
<a name="L1490"></a><tt class="py-lineno">1490</tt>  <tt class="py-line">                     <tt class="py-base-class">_DatabaseReportConsumer</tt><tt class="py-op">,</tt> </tt>
<a name="L1491"></a><tt class="py-lineno">1491</tt>  <tt class="py-line">                     <tt class="py-base-class">_ParametersConsumer</tt> </tt>
<a name="L1492"></a><tt class="py-lineno">1492</tt>  <tt class="py-line">                     <tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BlastConsumer-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_BlastConsumer-expanded"><a name="L1493"></a><tt class="py-lineno">1493</tt>  <tt class="py-line">    <tt class="py-comment"># This Consumer is inherits from many other consumer classes that handle</tt> </tt>
<a name="L1494"></a><tt class="py-lineno">1494</tt>  <tt class="py-line">    <tt class="py-comment"># the actual dirty work.  An alternate way to do it is to create objects</tt> </tt>
<a name="L1495"></a><tt class="py-lineno">1495</tt>  <tt class="py-line">    <tt class="py-comment"># of those classes and then delegate the parsing tasks to them in a</tt> </tt>
<a name="L1496"></a><tt class="py-lineno">1496</tt>  <tt class="py-line">    <tt class="py-comment"># decorator-type pattern.  The disadvantage of that is that the method</tt> </tt>
<a name="L1497"></a><tt class="py-lineno">1497</tt>  <tt class="py-line">    <tt class="py-comment"># names will need to be resolved in this classes.  However, using</tt> </tt>
<a name="L1498"></a><tt class="py-lineno">1498</tt>  <tt class="py-line">    <tt class="py-comment"># a decorator will retain more control in this class (which may or</tt> </tt>
<a name="L1499"></a><tt class="py-lineno">1499</tt>  <tt class="py-line">    <tt class="py-comment"># may not be a bad thing).  In addition, having each sub-consumer as</tt> </tt>
<a name="L1500"></a><tt class="py-lineno">1500</tt>  <tt class="py-line">    <tt class="py-comment"># its own object prevents this object's dictionary from being cluttered</tt> </tt>
<a name="L1501"></a><tt class="py-lineno">1501</tt>  <tt class="py-line">    <tt class="py-comment"># with members and reduces the chance of member collisions.</tt> </tt>
<a name="_BlastConsumer.__init__"></a><div id="_BlastConsumer.__init__-def"><a name="L1502"></a><tt class="py-lineno">1502</tt> <a class="py-toggle" href="#" id="_BlastConsumer.__init__-toggle" onclick="return toggle('_BlastConsumer.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._BlastConsumer-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BlastConsumer.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_BlastConsumer.__init__-expanded"><a name="L1503"></a><tt class="py-lineno">1503</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-894" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-894', 'data', 'link-432');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L1504"></a><tt class="py-lineno">1504</tt>  <tt class="py-line"> </tt>
<a name="_BlastConsumer.round"></a><div id="_BlastConsumer.round-def"><a name="L1505"></a><tt class="py-lineno">1505</tt> <a class="py-toggle" href="#" id="_BlastConsumer.round-toggle" onclick="return toggle('_BlastConsumer.round');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._BlastConsumer-class.html#round">round</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BlastConsumer.round-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_BlastConsumer.round-expanded"><a name="L1506"></a><tt class="py-lineno">1506</tt>  <tt class="py-line">        <tt class="py-comment"># Make sure nobody's trying to pass me PSI-BLAST data!</tt> </tt>
<a name="L1507"></a><tt class="py-lineno">1507</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"This consumer doesn't handle PSI-BLAST data"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1508"></a><tt class="py-lineno">1508</tt>  <tt class="py-line"> </tt>
<a name="_BlastConsumer.start_header"></a><div id="_BlastConsumer.start_header-def"><a name="L1509"></a><tt class="py-lineno">1509</tt> <a class="py-toggle" href="#" id="_BlastConsumer.start_header-toggle" onclick="return toggle('_BlastConsumer.start_header');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._BlastConsumer-class.html#start_header">start_header</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BlastConsumer.start_header-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_BlastConsumer.start_header-expanded"><a name="L1510"></a><tt class="py-lineno">1510</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-895" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-895', 'data', 'link-432');">data</a></tt> <tt class="py-op">=</tt> <tt id="link-896" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-896', 'Record', 'link-11');">Record</a></tt><tt class="py-op">.</tt><tt id="link-897" class="py-name"><a title="Bio.Blast
Bio.Blast.Record.Blast" class="py-name" href="#" onclick="return doclink('link-897', 'Blast', 'link-10');">Blast</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1511"></a><tt class="py-lineno">1511</tt>  <tt class="py-line">        <tt id="link-898" class="py-name" targets="Class Bio.Blast.NCBIStandalone._HeaderConsumer=Bio.Blast.NCBIStandalone._HeaderConsumer-class.html"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer" class="py-name" href="#" onclick="return doclink('link-898', '_HeaderConsumer', 'link-898');">_HeaderConsumer</a></tt><tt class="py-op">.</tt><tt id="link-899" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.start_header
Bio.Blast.NCBIStandalone._HeaderConsumer.start_header
Bio.Blast.NCBIStandalone._PSIBlastConsumer.start_header" class="py-name" href="#" onclick="return doclink('link-899', 'start_header', 'link-27');">start_header</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1512"></a><tt class="py-lineno">1512</tt>  <tt class="py-line"> </tt>
<a name="_BlastConsumer.end_header"></a><div id="_BlastConsumer.end_header-def"><a name="L1513"></a><tt class="py-lineno">1513</tt> <a class="py-toggle" href="#" id="_BlastConsumer.end_header-toggle" onclick="return toggle('_BlastConsumer.end_header');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._BlastConsumer-class.html#end_header">end_header</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BlastConsumer.end_header-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_BlastConsumer.end_header-expanded"><a name="L1514"></a><tt class="py-lineno">1514</tt>  <tt class="py-line">        <tt id="link-900" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer" class="py-name" href="#" onclick="return doclink('link-900', '_HeaderConsumer', 'link-898');">_HeaderConsumer</a></tt><tt class="py-op">.</tt><tt id="link-901" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.end_header
Bio.Blast.NCBIStandalone._HeaderConsumer.end_header
Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_header" class="py-name" href="#" onclick="return doclink('link-901', 'end_header', 'link-107');">end_header</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
<a name="L1515"></a><tt class="py-lineno">1515</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-902" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-902', 'data', 'link-432');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">__dict__</tt><tt class="py-op">.</tt><tt id="link-903" class="py-name" targets="Method Bio.File._IndexedSeqFileDict.update()=Bio.File._IndexedSeqFileDict-class.html#update,Method Bio.Index._InMemoryIndex.update()=Bio.Index._InMemoryIndex-class.html#update,Method Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.update()=Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer-class.html#update,Method Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.update()=Bio.NeuralNetwork.BackPropagation.Layer.InputLayer-class.html#update,Method Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.update()=Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer-class.html#update,Method Bio.SeqRecord._RestrictedDict.update()=Bio.SeqRecord._RestrictedDict-class.html#update,Method Bio._py3k._ordereddict.OrderedDict.update()=Bio._py3k._ordereddict.OrderedDict-class.html#update"><a title="Bio.File._IndexedSeqFileDict.update
Bio.Index._InMemoryIndex.update
Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.update
Bio.SeqRecord._RestrictedDict.update
Bio._py3k._ordereddict.OrderedDict.update" class="py-name" href="#" onclick="return doclink('link-903', 'update', 'link-903');">update</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-904" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-904', '_header', 'link-437');">_header</a></tt><tt class="py-op">.</tt><tt class="py-name">__dict__</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1516"></a><tt class="py-lineno">1516</tt>  <tt class="py-line"> </tt>
<a name="_BlastConsumer.end_descriptions"></a><div id="_BlastConsumer.end_descriptions-def"><a name="L1517"></a><tt class="py-lineno">1517</tt> <a class="py-toggle" href="#" id="_BlastConsumer.end_descriptions-toggle" onclick="return toggle('_BlastConsumer.end_descriptions');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._BlastConsumer-class.html#end_descriptions">end_descriptions</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BlastConsumer.end_descriptions-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_BlastConsumer.end_descriptions-expanded"><a name="L1518"></a><tt class="py-lineno">1518</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-905" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-905', 'data', 'link-432');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">descriptions</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_descriptions</tt> </tt>
</div><a name="L1519"></a><tt class="py-lineno">1519</tt>  <tt class="py-line"> </tt>
<a name="_BlastConsumer.end_alignment"></a><div id="_BlastConsumer.end_alignment-def"><a name="L1520"></a><tt class="py-lineno">1520</tt> <a class="py-toggle" href="#" id="_BlastConsumer.end_alignment-toggle" onclick="return toggle('_BlastConsumer.end_alignment');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._BlastConsumer-class.html#end_alignment">end_alignment</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BlastConsumer.end_alignment-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_BlastConsumer.end_alignment-expanded"><a name="L1521"></a><tt class="py-lineno">1521</tt>  <tt class="py-line">        <tt id="link-906" class="py-name" targets="Class Bio.Blast.NCBIStandalone._AlignmentConsumer=Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer" class="py-name" href="#" onclick="return doclink('link-906', '_AlignmentConsumer', 'link-906');">_AlignmentConsumer</a></tt><tt class="py-op">.</tt><tt id="link-907" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.end_alignment
Bio.Blast.NCBIStandalone._BlastConsumer.end_alignment
Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_alignment" class="py-name" href="#" onclick="return doclink('link-907', 'end_alignment', 'link-200');">end_alignment</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
<a name="L1522"></a><tt class="py-lineno">1522</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_alignment</tt><tt class="py-op">.</tt><tt id="link-908" class="py-name" targets="Variable Bio.SearchIO._model.hit.Hit.hsps=Bio.SearchIO._model.hit.Hit-class.html#hsps,Variable Bio.SearchIO._model.query.QueryResult.hsps=Bio.SearchIO._model.query.QueryResult-class.html#hsps"><a title="Bio.SearchIO._model.hit.Hit.hsps
Bio.SearchIO._model.query.QueryResult.hsps" class="py-name" href="#" onclick="return doclink('link-908', 'hsps', 'link-908');">hsps</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1523"></a><tt class="py-lineno">1523</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-909" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-909', 'data', 'link-432');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">alignments</tt><tt class="py-op">.</tt><tt id="link-910" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-910', 'append', 'link-506');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_alignment</tt><tt class="py-op">)</tt> </tt>
<a name="L1524"></a><tt class="py-lineno">1524</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_multiple_alignment</tt><tt class="py-op">.</tt><tt class="py-name">alignment</tt><tt class="py-op">:</tt> </tt>
<a name="L1525"></a><tt class="py-lineno">1525</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-911" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-911', 'data', 'link-432');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">multiple_alignment</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_multiple_alignment</tt> </tt>
</div><a name="L1526"></a><tt class="py-lineno">1526</tt>  <tt class="py-line"> </tt>
<a name="_BlastConsumer.end_hsp"></a><div id="_BlastConsumer.end_hsp-def"><a name="L1527"></a><tt class="py-lineno">1527</tt> <a class="py-toggle" href="#" id="_BlastConsumer.end_hsp-toggle" onclick="return toggle('_BlastConsumer.end_hsp');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._BlastConsumer-class.html#end_hsp">end_hsp</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BlastConsumer.end_hsp-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_BlastConsumer.end_hsp-expanded"><a name="L1528"></a><tt class="py-lineno">1528</tt>  <tt class="py-line">        <tt id="link-912" class="py-name" targets="Class Bio.Blast.NCBIStandalone._HSPConsumer=Bio.Blast.NCBIStandalone._HSPConsumer-class.html"><a title="Bio.Blast.NCBIStandalone._HSPConsumer" class="py-name" href="#" onclick="return doclink('link-912', '_HSPConsumer', 'link-912');">_HSPConsumer</a></tt><tt class="py-op">.</tt><tt id="link-913" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.end_hsp
Bio.Blast.NCBIStandalone._HSPConsumer.end_hsp
Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_hsp" class="py-name" href="#" onclick="return doclink('link-913', 'end_hsp', 'link-220');">end_hsp</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
<a name="L1529"></a><tt class="py-lineno">1529</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1530"></a><tt class="py-lineno">1530</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_alignment</tt><tt class="py-op">.</tt><tt id="link-914" class="py-name"><a title="Bio.SearchIO._model.hit.Hit.hsps
Bio.SearchIO._model.query.QueryResult.hsps" class="py-name" href="#" onclick="return doclink('link-914', 'hsps', 'link-908');">hsps</a></tt><tt class="py-op">.</tt><tt id="link-915" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-915', 'append', 'link-506');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">)</tt> </tt>
<a name="L1531"></a><tt class="py-lineno">1531</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L1532"></a><tt class="py-lineno">1532</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Found an HSP before an alignment"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1533"></a><tt class="py-lineno">1533</tt>  <tt class="py-line"> </tt>
<a name="_BlastConsumer.end_database_report"></a><div id="_BlastConsumer.end_database_report-def"><a name="L1534"></a><tt class="py-lineno">1534</tt> <a class="py-toggle" href="#" id="_BlastConsumer.end_database_report-toggle" onclick="return toggle('_BlastConsumer.end_database_report');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._BlastConsumer-class.html#end_database_report">end_database_report</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BlastConsumer.end_database_report-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_BlastConsumer.end_database_report-expanded"><a name="L1535"></a><tt class="py-lineno">1535</tt>  <tt class="py-line">        <tt id="link-916" class="py-name" targets="Class Bio.Blast.NCBIStandalone._DatabaseReportConsumer=Bio.Blast.NCBIStandalone._DatabaseReportConsumer-class.html"><a title="Bio.Blast.NCBIStandalone._DatabaseReportConsumer" class="py-name" href="#" onclick="return doclink('link-916', '_DatabaseReportConsumer', 'link-916');">_DatabaseReportConsumer</a></tt><tt class="py-op">.</tt><tt id="link-917" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.end_database_report
Bio.Blast.NCBIStandalone._DatabaseReportConsumer.end_database_report
Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_database_report" class="py-name" href="#" onclick="return doclink('link-917', 'end_database_report', 'link-287');">end_database_report</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
<a name="L1536"></a><tt class="py-lineno">1536</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-918" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-918', 'data', 'link-432');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">__dict__</tt><tt class="py-op">.</tt><tt id="link-919" class="py-name"><a title="Bio.File._IndexedSeqFileDict.update
Bio.Index._InMemoryIndex.update
Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.update
Bio.SeqRecord._RestrictedDict.update
Bio._py3k._ordereddict.OrderedDict.update" class="py-name" href="#" onclick="return doclink('link-919', 'update', 'link-903');">update</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_dr</tt><tt class="py-op">.</tt><tt class="py-name">__dict__</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1537"></a><tt class="py-lineno">1537</tt>  <tt class="py-line"> </tt>
<a name="_BlastConsumer.end_parameters"></a><div id="_BlastConsumer.end_parameters-def"><a name="L1538"></a><tt class="py-lineno">1538</tt> <a class="py-toggle" href="#" id="_BlastConsumer.end_parameters-toggle" onclick="return toggle('_BlastConsumer.end_parameters');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._BlastConsumer-class.html#end_parameters">end_parameters</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BlastConsumer.end_parameters-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_BlastConsumer.end_parameters-expanded"><a name="L1539"></a><tt class="py-lineno">1539</tt>  <tt class="py-line">        <tt id="link-920" class="py-name" targets="Class Bio.Blast.NCBIStandalone._ParametersConsumer=Bio.Blast.NCBIStandalone._ParametersConsumer-class.html"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer" class="py-name" href="#" onclick="return doclink('link-920', '_ParametersConsumer', 'link-920');">_ParametersConsumer</a></tt><tt class="py-op">.</tt><tt id="link-921" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.end_parameters
Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_parameters
Bio.Blast.NCBIStandalone._ParametersConsumer.end_parameters" class="py-name" href="#" onclick="return doclink('link-921', 'end_parameters', 'link-428');">end_parameters</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
<a name="L1540"></a><tt class="py-lineno">1540</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-922" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-922', 'data', 'link-432');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">__dict__</tt><tt class="py-op">.</tt><tt id="link-923" class="py-name"><a title="Bio.File._IndexedSeqFileDict.update
Bio.Index._InMemoryIndex.update
Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.update
Bio.SeqRecord._RestrictedDict.update
Bio._py3k._ordereddict.OrderedDict.update" class="py-name" href="#" onclick="return doclink('link-923', 'update', 'link-903');">update</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt class="py-name">__dict__</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1541"></a><tt class="py-lineno">1541</tt>  <tt class="py-line"> </tt>
<a name="L1542"></a><tt class="py-lineno">1542</tt>  <tt class="py-line"> </tt>
<a name="_PSIBlastConsumer"></a><div id="_PSIBlastConsumer-def"><a name="L1543"></a><tt class="py-lineno">1543</tt> <a class="py-toggle" href="#" id="_PSIBlastConsumer-toggle" onclick="return toggle('_PSIBlastConsumer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._PSIBlastConsumer-class.html">_PSIBlastConsumer</a><tt class="py-op">(</tt><tt class="py-base-class">AbstractConsumer</tt><tt class="py-op">,</tt> </tt>
<a name="L1544"></a><tt class="py-lineno">1544</tt>  <tt class="py-line">                        <tt class="py-base-class">_HeaderConsumer</tt><tt class="py-op">,</tt> </tt>
<a name="L1545"></a><tt class="py-lineno">1545</tt>  <tt class="py-line">                        <tt class="py-base-class">_DescriptionConsumer</tt><tt class="py-op">,</tt> </tt>
<a name="L1546"></a><tt class="py-lineno">1546</tt>  <tt class="py-line">                        <tt class="py-base-class">_AlignmentConsumer</tt><tt class="py-op">,</tt> </tt>
<a name="L1547"></a><tt class="py-lineno">1547</tt>  <tt class="py-line">                        <tt class="py-base-class">_HSPConsumer</tt><tt class="py-op">,</tt> </tt>
<a name="L1548"></a><tt class="py-lineno">1548</tt>  <tt class="py-line">                        <tt class="py-base-class">_DatabaseReportConsumer</tt><tt class="py-op">,</tt> </tt>
<a name="L1549"></a><tt class="py-lineno">1549</tt>  <tt class="py-line">                        <tt class="py-base-class">_ParametersConsumer</tt> </tt>
<a name="L1550"></a><tt class="py-lineno">1550</tt>  <tt class="py-line">                        <tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_PSIBlastConsumer-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_PSIBlastConsumer-expanded"><a name="_PSIBlastConsumer.__init__"></a><div id="_PSIBlastConsumer.__init__-def"><a name="L1551"></a><tt class="py-lineno">1551</tt> <a class="py-toggle" href="#" id="_PSIBlastConsumer.__init__-toggle" onclick="return toggle('_PSIBlastConsumer.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._PSIBlastConsumer-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_PSIBlastConsumer.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_PSIBlastConsumer.__init__-expanded"><a name="L1552"></a><tt class="py-lineno">1552</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-924" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-924', 'data', 'link-432');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L1553"></a><tt class="py-lineno">1553</tt>  <tt class="py-line"> </tt>
<a name="_PSIBlastConsumer.start_header"></a><div id="_PSIBlastConsumer.start_header-def"><a name="L1554"></a><tt class="py-lineno">1554</tt> <a class="py-toggle" href="#" id="_PSIBlastConsumer.start_header-toggle" onclick="return toggle('_PSIBlastConsumer.start_header');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._PSIBlastConsumer-class.html#start_header">start_header</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_PSIBlastConsumer.start_header-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_PSIBlastConsumer.start_header-expanded"><a name="L1555"></a><tt class="py-lineno">1555</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-925" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-925', 'data', 'link-432');">data</a></tt> <tt class="py-op">=</tt> <tt id="link-926" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-926', 'Record', 'link-11');">Record</a></tt><tt class="py-op">.</tt><tt id="link-927" class="py-name" targets="Class Bio.Blast.Record.PSIBlast=Bio.Blast.Record.PSIBlast-class.html"><a title="Bio.Blast.Record.PSIBlast" class="py-name" href="#" onclick="return doclink('link-927', 'PSIBlast', 'link-927');">PSIBlast</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1556"></a><tt class="py-lineno">1556</tt>  <tt class="py-line">        <tt id="link-928" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer" class="py-name" href="#" onclick="return doclink('link-928', '_HeaderConsumer', 'link-898');">_HeaderConsumer</a></tt><tt class="py-op">.</tt><tt id="link-929" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.start_header
Bio.Blast.NCBIStandalone._HeaderConsumer.start_header
Bio.Blast.NCBIStandalone._PSIBlastConsumer.start_header" class="py-name" href="#" onclick="return doclink('link-929', 'start_header', 'link-27');">start_header</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1557"></a><tt class="py-lineno">1557</tt>  <tt class="py-line"> </tt>
<a name="_PSIBlastConsumer.end_header"></a><div id="_PSIBlastConsumer.end_header-def"><a name="L1558"></a><tt class="py-lineno">1558</tt> <a class="py-toggle" href="#" id="_PSIBlastConsumer.end_header-toggle" onclick="return toggle('_PSIBlastConsumer.end_header');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._PSIBlastConsumer-class.html#end_header">end_header</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_PSIBlastConsumer.end_header-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_PSIBlastConsumer.end_header-expanded"><a name="L1559"></a><tt class="py-lineno">1559</tt>  <tt class="py-line">        <tt id="link-930" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer" class="py-name" href="#" onclick="return doclink('link-930', '_HeaderConsumer', 'link-898');">_HeaderConsumer</a></tt><tt class="py-op">.</tt><tt id="link-931" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.end_header
Bio.Blast.NCBIStandalone._HeaderConsumer.end_header
Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_header" class="py-name" href="#" onclick="return doclink('link-931', 'end_header', 'link-107');">end_header</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
<a name="L1560"></a><tt class="py-lineno">1560</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-932" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-932', 'data', 'link-432');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">__dict__</tt><tt class="py-op">.</tt><tt id="link-933" class="py-name"><a title="Bio.File._IndexedSeqFileDict.update
Bio.Index._InMemoryIndex.update
Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.update
Bio.SeqRecord._RestrictedDict.update
Bio._py3k._ordereddict.OrderedDict.update" class="py-name" href="#" onclick="return doclink('link-933', 'update', 'link-903');">update</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-934" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-934', '_header', 'link-437');">_header</a></tt><tt class="py-op">.</tt><tt class="py-name">__dict__</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1561"></a><tt class="py-lineno">1561</tt>  <tt class="py-line"> </tt>
<a name="_PSIBlastConsumer.start_descriptions"></a><div id="_PSIBlastConsumer.start_descriptions-def"><a name="L1562"></a><tt class="py-lineno">1562</tt> <a class="py-toggle" href="#" id="_PSIBlastConsumer.start_descriptions-toggle" onclick="return toggle('_PSIBlastConsumer.start_descriptions');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._PSIBlastConsumer-class.html#start_descriptions">start_descriptions</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_PSIBlastConsumer.start_descriptions-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_PSIBlastConsumer.start_descriptions-expanded"><a name="L1563"></a><tt class="py-lineno">1563</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_round</tt> <tt class="py-op">=</tt> <tt id="link-935" class="py-name"><a title="Bio.Affy.CelFile.Record
Bio.Blast.Record
Bio.Cluster.Record
Bio.Compass.Record
Bio.Emboss.Primer3.Record
Bio.ExPASy.Enzyme.Record
Bio.ExPASy.Prodoc.Record
Bio.ExPASy.Prosite.Record
Bio.ExPASy.ScanProsite.Record
Bio.GenBank.Record
Bio.GenBank.Record.Record
Bio.Geo.Record
Bio.Geo.Record.Record
Bio.KEGG.Compound.Record
Bio.KEGG.Enzyme.Record
Bio.Medline.Record
Bio.Motif.Parsers.AlignAce.Record
Bio.Motif.Parsers.MAST.Record
Bio.PopGen.FDist.Record
Bio.PopGen.GenePop.LargeFileParser.Record
Bio.PopGen.GenePop.Record
Bio.SCOP.Cla.Record
Bio.SCOP.Des.Record
Bio.SCOP.Dom.Record
Bio.SCOP.Hie.Record
Bio.Sequencing.Phd.Record
Bio.SwissProt.KeyWList.Record
Bio.SwissProt.Record
Bio.UniGene.Record
Bio.motifs.alignace.Record
Bio.motifs.jaspar.Record
Bio.motifs.mast.Record
Bio.motifs.meme.Record
Bio.motifs.transfac.Record" class="py-name" href="#" onclick="return doclink('link-935', 'Record', 'link-11');">Record</a></tt><tt class="py-op">.</tt><tt id="link-936" class="py-name" targets="Class Bio.Blast.Record.Round=Bio.Blast.Record.Round-class.html"><a title="Bio.Blast.Record.Round" class="py-name" href="#" onclick="return doclink('link-936', 'Round', 'link-936');">Round</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1564"></a><tt class="py-lineno">1564</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-937" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-937', 'data', 'link-432');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">rounds</tt><tt class="py-op">.</tt><tt id="link-938" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-938', 'append', 'link-506');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_round</tt><tt class="py-op">)</tt> </tt>
<a name="L1565"></a><tt class="py-lineno">1565</tt>  <tt class="py-line">        <tt id="link-939" class="py-name" targets="Class Bio.Blast.NCBIStandalone._DescriptionConsumer=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer" class="py-name" href="#" onclick="return doclink('link-939', '_DescriptionConsumer', 'link-939');">_DescriptionConsumer</a></tt><tt class="py-op">.</tt><tt id="link-940" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.start_descriptions
Bio.Blast.NCBIStandalone._PSIBlastConsumer.start_descriptions" class="py-name" href="#" onclick="return doclink('link-940', 'start_descriptions', 'link-116');">start_descriptions</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1566"></a><tt class="py-lineno">1566</tt>  <tt class="py-line"> </tt>
<a name="_PSIBlastConsumer.end_descriptions"></a><div id="_PSIBlastConsumer.end_descriptions-def"><a name="L1567"></a><tt class="py-lineno">1567</tt> <a class="py-toggle" href="#" id="_PSIBlastConsumer.end_descriptions-toggle" onclick="return toggle('_PSIBlastConsumer.end_descriptions');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._PSIBlastConsumer-class.html#end_descriptions">end_descriptions</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_PSIBlastConsumer.end_descriptions-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_PSIBlastConsumer.end_descriptions-expanded"><a name="L1568"></a><tt class="py-lineno">1568</tt>  <tt class="py-line">        <tt id="link-941" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer" class="py-name" href="#" onclick="return doclink('link-941', '_DescriptionConsumer', 'link-939');">_DescriptionConsumer</a></tt><tt class="py-op">.</tt><tt id="link-942" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.end_descriptions
Bio.Blast.NCBIStandalone._DescriptionConsumer.end_descriptions
Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_descriptions" class="py-name" href="#" onclick="return doclink('link-942', 'end_descriptions', 'link-144');">end_descriptions</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
<a name="L1569"></a><tt class="py-lineno">1569</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_round</tt><tt class="py-op">.</tt><tt id="link-943" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Pathway.number=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#number"><a title="Bio.KEGG.KGML.KGML_pathway.Pathway.number" class="py-name" href="#" onclick="return doclink('link-943', 'number', 'link-943');">number</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_roundnum</tt> </tt>
<a name="L1570"></a><tt class="py-lineno">1570</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_descriptions</tt><tt class="py-op">:</tt> </tt>
<a name="L1571"></a><tt class="py-lineno">1571</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_round</tt><tt class="py-op">.</tt><tt class="py-name">new_seqs</tt><tt class="py-op">.</tt><tt id="link-944" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.extend()=Bio.Align.MultipleSeqAlignment-class.html#extend,Method Bio.SCOP.Raf.SeqMap.extend()=Bio.SCOP.Raf.SeqMap-class.html#extend,Method Bio.Seq.MutableSeq.extend()=Bio.Seq.MutableSeq-class.html#extend"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-944', 'extend', 'link-944');">extend</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_descriptions</tt><tt class="py-op">)</tt> </tt>
<a name="L1572"></a><tt class="py-lineno">1572</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_round</tt><tt class="py-op">.</tt><tt class="py-name">reused_seqs</tt><tt class="py-op">.</tt><tt id="link-945" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-945', 'extend', 'link-944');">extend</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_model_sequences</tt><tt class="py-op">)</tt> </tt>
<a name="L1573"></a><tt class="py-lineno">1573</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_round</tt><tt class="py-op">.</tt><tt class="py-name">new_seqs</tt><tt class="py-op">.</tt><tt id="link-946" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-946', 'extend', 'link-944');">extend</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_nonmodel_sequences</tt><tt class="py-op">)</tt> </tt>
<a name="L1574"></a><tt class="py-lineno">1574</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_converged</tt><tt class="py-op">:</tt> </tt>
<a name="L1575"></a><tt class="py-lineno">1575</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-947" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-947', 'data', 'link-432');">data</a></tt><tt class="py-op">.</tt><tt id="link-948" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.converged" class="py-name" href="#" onclick="return doclink('link-948', 'converged', 'link-175');">converged</a></tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
</div><a name="L1576"></a><tt class="py-lineno">1576</tt>  <tt class="py-line"> </tt>
<a name="_PSIBlastConsumer.end_alignment"></a><div id="_PSIBlastConsumer.end_alignment-def"><a name="L1577"></a><tt class="py-lineno">1577</tt> <a class="py-toggle" href="#" id="_PSIBlastConsumer.end_alignment-toggle" onclick="return toggle('_PSIBlastConsumer.end_alignment');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._PSIBlastConsumer-class.html#end_alignment">end_alignment</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_PSIBlastConsumer.end_alignment-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_PSIBlastConsumer.end_alignment-expanded"><a name="L1578"></a><tt class="py-lineno">1578</tt>  <tt class="py-line">        <tt id="link-949" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer" class="py-name" href="#" onclick="return doclink('link-949', '_AlignmentConsumer', 'link-906');">_AlignmentConsumer</a></tt><tt class="py-op">.</tt><tt id="link-950" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.end_alignment
Bio.Blast.NCBIStandalone._BlastConsumer.end_alignment
Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_alignment" class="py-name" href="#" onclick="return doclink('link-950', 'end_alignment', 'link-200');">end_alignment</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
<a name="L1579"></a><tt class="py-lineno">1579</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_alignment</tt><tt class="py-op">.</tt><tt id="link-951" class="py-name"><a title="Bio.SearchIO._model.hit.Hit.hsps
Bio.SearchIO._model.query.QueryResult.hsps" class="py-name" href="#" onclick="return doclink('link-951', 'hsps', 'link-908');">hsps</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1580"></a><tt class="py-lineno">1580</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_round</tt><tt class="py-op">.</tt><tt class="py-name">alignments</tt><tt class="py-op">.</tt><tt id="link-952" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-952', 'append', 'link-506');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_alignment</tt><tt class="py-op">)</tt> </tt>
<a name="L1581"></a><tt class="py-lineno">1581</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_multiple_alignment</tt><tt class="py-op">:</tt> </tt>
<a name="L1582"></a><tt class="py-lineno">1582</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_round</tt><tt class="py-op">.</tt><tt class="py-name">multiple_alignment</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_multiple_alignment</tt> </tt>
</div><a name="L1583"></a><tt class="py-lineno">1583</tt>  <tt class="py-line"> </tt>
<a name="_PSIBlastConsumer.end_hsp"></a><div id="_PSIBlastConsumer.end_hsp-def"><a name="L1584"></a><tt class="py-lineno">1584</tt> <a class="py-toggle" href="#" id="_PSIBlastConsumer.end_hsp-toggle" onclick="return toggle('_PSIBlastConsumer.end_hsp');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._PSIBlastConsumer-class.html#end_hsp">end_hsp</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_PSIBlastConsumer.end_hsp-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_PSIBlastConsumer.end_hsp-expanded"><a name="L1585"></a><tt class="py-lineno">1585</tt>  <tt class="py-line">        <tt id="link-953" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer" class="py-name" href="#" onclick="return doclink('link-953', '_HSPConsumer', 'link-912');">_HSPConsumer</a></tt><tt class="py-op">.</tt><tt id="link-954" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.end_hsp
Bio.Blast.NCBIStandalone._HSPConsumer.end_hsp
Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_hsp" class="py-name" href="#" onclick="return doclink('link-954', 'end_hsp', 'link-220');">end_hsp</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
<a name="L1586"></a><tt class="py-lineno">1586</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1587"></a><tt class="py-lineno">1587</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_alignment</tt><tt class="py-op">.</tt><tt id="link-955" class="py-name"><a title="Bio.SearchIO._model.hit.Hit.hsps
Bio.SearchIO._model.query.QueryResult.hsps" class="py-name" href="#" onclick="return doclink('link-955', 'hsps', 'link-908');">hsps</a></tt><tt class="py-op">.</tt><tt id="link-956" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-956', 'append', 'link-506');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_hsp</tt><tt class="py-op">)</tt> </tt>
<a name="L1588"></a><tt class="py-lineno">1588</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L1589"></a><tt class="py-lineno">1589</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Found an HSP before an alignment"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1590"></a><tt class="py-lineno">1590</tt>  <tt class="py-line"> </tt>
<a name="_PSIBlastConsumer.end_database_report"></a><div id="_PSIBlastConsumer.end_database_report-def"><a name="L1591"></a><tt class="py-lineno">1591</tt> <a class="py-toggle" href="#" id="_PSIBlastConsumer.end_database_report-toggle" onclick="return toggle('_PSIBlastConsumer.end_database_report');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._PSIBlastConsumer-class.html#end_database_report">end_database_report</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_PSIBlastConsumer.end_database_report-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_PSIBlastConsumer.end_database_report-expanded"><a name="L1592"></a><tt class="py-lineno">1592</tt>  <tt class="py-line">        <tt id="link-957" class="py-name"><a title="Bio.Blast.NCBIStandalone._DatabaseReportConsumer" class="py-name" href="#" onclick="return doclink('link-957', '_DatabaseReportConsumer', 'link-916');">_DatabaseReportConsumer</a></tt><tt class="py-op">.</tt><tt id="link-958" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.end_database_report
Bio.Blast.NCBIStandalone._DatabaseReportConsumer.end_database_report
Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_database_report" class="py-name" href="#" onclick="return doclink('link-958', 'end_database_report', 'link-287');">end_database_report</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
<a name="L1593"></a><tt class="py-lineno">1593</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-959" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-959', 'data', 'link-432');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">__dict__</tt><tt class="py-op">.</tt><tt id="link-960" class="py-name"><a title="Bio.File._IndexedSeqFileDict.update
Bio.Index._InMemoryIndex.update
Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.update
Bio.SeqRecord._RestrictedDict.update
Bio._py3k._ordereddict.OrderedDict.update" class="py-name" href="#" onclick="return doclink('link-960', 'update', 'link-903');">update</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_dr</tt><tt class="py-op">.</tt><tt class="py-name">__dict__</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1594"></a><tt class="py-lineno">1594</tt>  <tt class="py-line"> </tt>
<a name="_PSIBlastConsumer.end_parameters"></a><div id="_PSIBlastConsumer.end_parameters-def"><a name="L1595"></a><tt class="py-lineno">1595</tt> <a class="py-toggle" href="#" id="_PSIBlastConsumer.end_parameters-toggle" onclick="return toggle('_PSIBlastConsumer.end_parameters');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._PSIBlastConsumer-class.html#end_parameters">end_parameters</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_PSIBlastConsumer.end_parameters-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_PSIBlastConsumer.end_parameters-expanded"><a name="L1596"></a><tt class="py-lineno">1596</tt>  <tt class="py-line">        <tt id="link-961" class="py-name"><a title="Bio.Blast.NCBIStandalone._ParametersConsumer" class="py-name" href="#" onclick="return doclink('link-961', '_ParametersConsumer', 'link-920');">_ParametersConsumer</a></tt><tt class="py-op">.</tt><tt id="link-962" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.end_parameters
Bio.Blast.NCBIStandalone._PSIBlastConsumer.end_parameters
Bio.Blast.NCBIStandalone._ParametersConsumer.end_parameters" class="py-name" href="#" onclick="return doclink('link-962', 'end_parameters', 'link-428');">end_parameters</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
<a name="L1597"></a><tt class="py-lineno">1597</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-963" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-963', 'data', 'link-432');">data</a></tt><tt class="py-op">.</tt><tt class="py-name">__dict__</tt><tt class="py-op">.</tt><tt id="link-964" class="py-name"><a title="Bio.File._IndexedSeqFileDict.update
Bio.Index._InMemoryIndex.update
Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.update
Bio.SeqRecord._RestrictedDict.update
Bio._py3k._ordereddict.OrderedDict.update" class="py-name" href="#" onclick="return doclink('link-964', 'update', 'link-903');">update</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_params</tt><tt class="py-op">.</tt><tt class="py-name">__dict__</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1598"></a><tt class="py-lineno">1598</tt>  <tt class="py-line"> </tt>
<a name="L1599"></a><tt class="py-lineno">1599</tt>  <tt class="py-line"> </tt>
<a name="Iterator"></a><div id="Iterator-def"><a name="L1600"></a><tt class="py-lineno">1600</tt> <a class="py-toggle" href="#" id="Iterator-toggle" onclick="return toggle('Iterator');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone.Iterator-class.html">Iterator</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Iterator-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Iterator-expanded"><a name="L1601"></a><tt class="py-lineno">1601</tt>  <tt class="py-line">    <tt class="py-docstring">"""Iterates over a file of multiple BLAST results.</tt> </tt>
<a name="L1602"></a><tt class="py-lineno">1602</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1603"></a><tt class="py-lineno">1603</tt>  <tt class="py-line"><tt class="py-docstring">    Methods:</tt> </tt>
<a name="L1604"></a><tt class="py-lineno">1604</tt>  <tt class="py-line"><tt class="py-docstring">    next   Return the next record from the stream, or None.</tt> </tt>
<a name="L1605"></a><tt class="py-lineno">1605</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1606"></a><tt class="py-lineno">1606</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="Iterator.__init__"></a><div id="Iterator.__init__-def"><a name="L1607"></a><tt class="py-lineno">1607</tt> <a class="py-toggle" href="#" id="Iterator.__init__-toggle" onclick="return toggle('Iterator.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone.Iterator-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">parser</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Iterator.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Iterator.__init__-expanded"><a name="L1608"></a><tt class="py-lineno">1608</tt>  <tt class="py-line">        <tt class="py-docstring">"""__init__(self, handle, parser=None)</tt> </tt>
<a name="L1609"></a><tt class="py-lineno">1609</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1610"></a><tt class="py-lineno">1610</tt>  <tt class="py-line"><tt class="py-docstring">        Create a new iterator.  handle is a file-like object.  parser</tt> </tt>
<a name="L1611"></a><tt class="py-lineno">1611</tt>  <tt class="py-line"><tt class="py-docstring">        is an optional Parser object to change the results into another form.</tt> </tt>
<a name="L1612"></a><tt class="py-lineno">1612</tt>  <tt class="py-line"><tt class="py-docstring">        If set to None, then the raw contents of the file will be returned.</tt> </tt>
<a name="L1613"></a><tt class="py-lineno">1613</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1614"></a><tt class="py-lineno">1614</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1615"></a><tt class="py-lineno">1615</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1616"></a><tt class="py-lineno">1616</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-965" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-965', 'readline', 'link-39');">readline</a></tt> </tt>
<a name="L1617"></a><tt class="py-lineno">1617</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt> </tt>
<a name="L1618"></a><tt class="py-lineno">1618</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt> </tt>
<a name="L1619"></a><tt class="py-lineno">1619</tt>  <tt class="py-line">                <tt class="py-string">"I expected a file handle or file-like object, got %s"</tt> </tt>
<a name="L1620"></a><tt class="py-lineno">1620</tt>  <tt class="py-line">                <tt class="py-op">%</tt> <tt id="link-966" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.type()=Bio.Phylo.PhyloXMLIO.Writer-class.html#type"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-966', 'type', 'link-966');">type</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1621"></a><tt class="py-lineno">1621</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_uhandle</tt> <tt class="py-op">=</tt> <tt id="link-967" class="py-name"><a title="Bio.File" class="py-name" href="#" onclick="return doclink('link-967', 'File', 'link-6');">File</a></tt><tt class="py-op">.</tt><tt id="link-968" class="py-name"><a title="Bio.File.UndoHandle" class="py-name" href="#" onclick="return doclink('link-968', 'UndoHandle', 'link-17');">UndoHandle</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L1622"></a><tt class="py-lineno">1622</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-969" class="py-name" targets="Variable Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer._parser=Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer-class.html#_parser"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer._parser" class="py-name" href="#" onclick="return doclink('link-969', '_parser', 'link-969');">_parser</a></tt> <tt class="py-op">=</tt> <tt class="py-name">parser</tt> </tt>
<a name="L1623"></a><tt class="py-lineno">1623</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-970" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-970', '_header', 'link-437');">_header</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div><a name="L1624"></a><tt class="py-lineno">1624</tt>  <tt class="py-line"> </tt>
<a name="Iterator.__next__"></a><div id="Iterator.__next__-def"><a name="L1625"></a><tt class="py-lineno">1625</tt> <a class="py-toggle" href="#" id="Iterator.__next__-toggle" onclick="return toggle('Iterator.__next__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone.Iterator-class.html#__next__">__next__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Iterator.__next__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Iterator.__next__-expanded"><a name="L1626"></a><tt class="py-lineno">1626</tt>  <tt class="py-line">        <tt class="py-docstring">"""next(self) -&gt; object</tt> </tt>
<a name="L1627"></a><tt class="py-lineno">1627</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1628"></a><tt class="py-lineno">1628</tt>  <tt class="py-line"><tt class="py-docstring">        Return the next Blast record from the file.  If no more records,</tt> </tt>
<a name="L1629"></a><tt class="py-lineno">1629</tt>  <tt class="py-line"><tt class="py-docstring">        return None.</tt> </tt>
<a name="L1630"></a><tt class="py-lineno">1630</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1631"></a><tt class="py-lineno">1631</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1632"></a><tt class="py-lineno">1632</tt>  <tt class="py-line">        <tt class="py-name">lines</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L1633"></a><tt class="py-lineno">1633</tt>  <tt class="py-line">        <tt id="link-971" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.query
Bio.SearchIO._model.hsp.HSP.query
Bio.SearchIO._model.hsp.HSPFragment.query
Bio.SearchIO._model.query" class="py-name" href="#" onclick="return doclink('link-971', 'query', 'link-239');">query</a></tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L1634"></a><tt class="py-lineno">1634</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L1635"></a><tt class="py-lineno">1635</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_uhandle</tt><tt class="py-op">.</tt><tt id="link-972" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-972', 'readline', 'link-39');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1636"></a><tt class="py-lineno">1636</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L1637"></a><tt class="py-lineno">1637</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L1638"></a><tt class="py-lineno">1638</tt>  <tt class="py-line">            <tt class="py-comment"># If I've reached the next one, then put the line back and stop.</tt> </tt>
<a name="L1639"></a><tt class="py-lineno">1639</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">lines</tt> <tt class="py-keyword">and</tt> <tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-973" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-973', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'BLAST'</tt><tt class="py-op">)</tt> </tt>
<a name="L1640"></a><tt class="py-lineno">1640</tt>  <tt class="py-line">                          <tt class="py-keyword">or</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-974" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-974', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'BLAST'</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L1641"></a><tt class="py-lineno">1641</tt>  <tt class="py-line">                          <tt class="py-keyword">or</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-975" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-975', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'&lt;?xml '</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1642"></a><tt class="py-lineno">1642</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_uhandle</tt><tt class="py-op">.</tt><tt id="link-976" class="py-name"><a title="Bio.File.UndoHandle.saveline" class="py-name" href="#" onclick="return doclink('link-976', 'saveline', 'link-256');">saveline</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1643"></a><tt class="py-lineno">1643</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L1644"></a><tt class="py-lineno">1644</tt>  <tt class="py-line">            <tt class="py-comment"># New style files omit the BLAST line to mark a new query:</tt> </tt>
<a name="L1645"></a><tt class="py-lineno">1645</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-977" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-977', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"Query="</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1646"></a><tt class="py-lineno">1646</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-978" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.query
Bio.SearchIO._model.hsp.HSP.query
Bio.SearchIO._model.hsp.HSPFragment.query
Bio.SearchIO._model.query" class="py-name" href="#" onclick="return doclink('link-978', 'query', 'link-239');">query</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1647"></a><tt class="py-lineno">1647</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-979" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-979', '_header', 'link-437');">_header</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1648"></a><tt class="py-lineno">1648</tt>  <tt class="py-line">                        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-980" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-980', '_header', 'link-437');">_header</a></tt> <tt class="py-op">=</tt> <tt class="py-name">lines</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L1649"></a><tt class="py-lineno">1649</tt>  <tt class="py-line">                    <tt id="link-981" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.query
Bio.SearchIO._model.hsp.HSP.query
Bio.SearchIO._model.hsp.HSPFragment.query
Bio.SearchIO._model.query" class="py-name" href="#" onclick="return doclink('link-981', 'query', 'link-239');">query</a></tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L1650"></a><tt class="py-lineno">1650</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1651"></a><tt class="py-lineno">1651</tt>  <tt class="py-line">                    <tt class="py-comment"># Start of another record</tt> </tt>
<a name="L1652"></a><tt class="py-lineno">1652</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_uhandle</tt><tt class="py-op">.</tt><tt id="link-982" class="py-name"><a title="Bio.File.UndoHandle.saveline" class="py-name" href="#" onclick="return doclink('link-982', 'saveline', 'link-256');">saveline</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1653"></a><tt class="py-lineno">1653</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L1654"></a><tt class="py-lineno">1654</tt>  <tt class="py-line">            <tt class="py-name">lines</tt><tt class="py-op">.</tt><tt id="link-983" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-983', 'append', 'link-506');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1655"></a><tt class="py-lineno">1655</tt>  <tt class="py-line"> </tt>
<a name="L1656"></a><tt class="py-lineno">1656</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-984" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.query
Bio.SearchIO._model.hsp.HSP.query
Bio.SearchIO._model.hsp.HSPFragment.query
Bio.SearchIO._model.query" class="py-name" href="#" onclick="return doclink('link-984', 'query', 'link-239');">query</a></tt> <tt class="py-keyword">and</tt> <tt class="py-string">"BLAST"</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">lines</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L1657"></a><tt class="py-lineno">1657</tt>  <tt class="py-line">            <tt class="py-comment"># Cheat and re-insert the header</tt> </tt>
<a name="L1658"></a><tt class="py-lineno">1658</tt>  <tt class="py-line">            <tt class="py-comment"># print "-"*50</tt> </tt>
<a name="L1659"></a><tt class="py-lineno">1659</tt>  <tt class="py-line">            <tt class="py-comment"># print "".join(self._header)</tt> </tt>
<a name="L1660"></a><tt class="py-lineno">1660</tt>  <tt class="py-line">            <tt class="py-comment"># print "-"*50</tt> </tt>
<a name="L1661"></a><tt class="py-lineno">1661</tt>  <tt class="py-line">            <tt class="py-comment"># print "".join(lines)</tt> </tt>
<a name="L1662"></a><tt class="py-lineno">1662</tt>  <tt class="py-line">            <tt class="py-comment"># print "-"*50</tt> </tt>
<a name="L1663"></a><tt class="py-lineno">1663</tt>  <tt class="py-line">            <tt class="py-name">lines</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-985" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator._header
Bio.AlignIO.EmbossIO.EmbossIterator._header
Bio.AlignIO.PhylipIO.PhylipIterator._header
Bio.AlignIO.PhylipIO.SequentialPhylipIterator._header
Bio.AlignIO.StockholmIO.StockholmIterator._header" class="py-name" href="#" onclick="return doclink('link-985', '_header', 'link-437');">_header</a></tt> <tt class="py-op">+</tt> <tt class="py-name">lines</tt> </tt>
<a name="L1664"></a><tt class="py-lineno">1664</tt>  <tt class="py-line"> </tt>
<a name="L1665"></a><tt class="py-lineno">1665</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">lines</tt><tt class="py-op">:</tt> </tt>
<a name="L1666"></a><tt class="py-lineno">1666</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
<a name="L1667"></a><tt class="py-lineno">1667</tt>  <tt class="py-line"> </tt>
<a name="L1668"></a><tt class="py-lineno">1668</tt>  <tt class="py-line">        <tt id="link-986" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-986', 'data', 'link-432');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">lines</tt><tt class="py-op">)</tt> </tt>
<a name="L1669"></a><tt class="py-lineno">1669</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-987" class="py-name"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer._parser" class="py-name" href="#" onclick="return doclink('link-987', '_parser', 'link-969');">_parser</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1670"></a><tt class="py-lineno">1670</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-988" class="py-name"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer._parser" class="py-name" href="#" onclick="return doclink('link-988', '_parser', 'link-969');">_parser</a></tt><tt class="py-op">.</tt><tt id="link-989" class="py-name" targets="Function Bio.AlignIO.parse()=Bio.AlignIO-module.html#parse,Method Bio.Blast.NCBIStandalone.BlastErrorParser.parse()=Bio.Blast.NCBIStandalone.BlastErrorParser-class.html#parse,Method Bio.Blast.NCBIStandalone.BlastParser.parse()=Bio.Blast.NCBIStandalone.BlastParser-class.html#parse,Method Bio.Blast.NCBIStandalone.PSIBlastParser.parse()=Bio.Blast.NCBIStandalone.PSIBlastParser-class.html#parse,Function Bio.Blast.NCBIXML.parse()=Bio.Blast.NCBIXML-module.html#parse,Function Bio.Compass.parse()=Bio.Compass-module.html#parse,Function Bio.Emboss.Primer3.parse()=Bio.Emboss.Primer3-module.html#parse,Method Bio.Entrez.Parser.DataHandler.parse()=Bio.Entrez.Parser.DataHandler-class.html#parse,Function Bio.Entrez.parse()=Bio.Entrez-module.html#parse,Function Bio.ExPASy.Enzyme.parse()=Bio.ExPASy.Enzyme-module.html#parse,Function Bio.ExPASy.Prodoc.parse()=Bio.ExPASy.Prodoc-module.html#parse,Function Bio.ExPASy.Prosite.parse()=Bio.ExPASy.Prosite-module.html#parse,Method Bio.GenBank.FeatureParser.parse()=Bio.GenBank.FeatureParser-class.html#parse,Method Bio.GenBank.RecordParser.parse()=Bio.GenBank.RecordParser-class.html#parse,Method Bio.GenBank.Scanner.InsdcScanner.parse()=Bio.GenBank.Scanner.InsdcScanner-class.html#parse,Function Bio.Geo.parse()=Bio.Geo-module.html#parse,Function Bio.KEGG.Compound.parse()=Bio.KEGG.Compound-module.html#parse,Function Bio.KEGG.Enzyme.parse()=Bio.KEGG.Enzyme-module.html#parse,Method Bio.KEGG.KGML.KGML_parser.KGMLParser.parse()=Bio.KEGG.KGML.KGML_parser.KGMLParser-class.html#parse,Function Bio.KEGG.KGML.KGML_parser.parse()=Bio.KEGG.KGML.KGML_parser-module.html#parse,Function Bio.KEGG.Map.parse()=Bio.KEGG.Map-module.html#parse,Function Bio.Medline.parse()=Bio.Medline-module.html#parse,Function Bio.Motif.parse()=Bio.Motif-module.html#parse,Method Bio.ParserSupport.AbstractParser.parse()=Bio.ParserSupport.AbstractParser-class.html#parse,Method Bio.Phylo.CDAOIO.Parser.parse()=Bio.Phylo.CDAOIO.Parser-class.html#parse,Function Bio.Phylo.CDAOIO.parse()=Bio.Phylo.CDAOIO-module.html#parse,Method Bio.Phylo.NeXMLIO.Parser.parse()=Bio.Phylo.NeXMLIO.Parser-class.html#parse,Function Bio.Phylo.NeXMLIO.parse()=Bio.Phylo.NeXMLIO-module.html#parse,Method Bio.Phylo.NewickIO.Parser.parse()=Bio.Phylo.NewickIO.Parser-class.html#parse,Function Bio.Phylo.NewickIO.parse()=Bio.Phylo.NewickIO-module.html#parse,Function Bio.Phylo.NexusIO.parse()=Bio.Phylo.NexusIO-module.html#parse,Method Bio.Phylo.PhyloXMLIO.Parser.parse()=Bio.Phylo.PhyloXMLIO.Parser-class.html#parse,Function Bio.Phylo.PhyloXMLIO.parse()=Bio.Phylo.PhyloXMLIO-module.html#parse,Function Bio.Phylo._io.parse()=Bio.Phylo._io-module.html#parse,Function Bio.SCOP.Cla.parse()=Bio.SCOP.Cla-module.html#parse,Function Bio.SCOP.Des.parse()=Bio.SCOP.Des-module.html#parse,Function Bio.SCOP.Dom.parse()=Bio.SCOP.Dom-module.html#parse,Function Bio.SCOP.Hie.parse()=Bio.SCOP.Hie-module.html#parse,Function Bio.SCOP.Raf.parse()=Bio.SCOP.Raf-module.html#parse,Function Bio.SearchIO.parse()=Bio.SearchIO-module.html#parse,Method Bio.SeqIO.UniprotIO.Parser.parse()=Bio.SeqIO.UniprotIO.Parser-class.html#parse,Function Bio.SeqIO.parse()=Bio.SeqIO-module.html#parse,Function Bio.Sequencing.Ace.parse()=Bio.Sequencing.Ace-module.html#parse,Function Bio.Sequencing.Phd.parse()=Bio.Sequencing.Phd-module.html#parse,Function Bio.SwissProt.KeyWList.parse()=Bio.SwissProt.KeyWList-module.html#parse,Function Bio.SwissProt.parse()=Bio.SwissProt-module.html#parse,Function Bio.UniGene.parse()=Bio.UniGene-module.html#parse,Function Bio.Wise.psw.parse()=Bio.Wise.psw-module.html#parse,Function Bio.motifs.parse()=Bio.motifs-module.html#parse"><a title="Bio.AlignIO.parse
Bio.Blast.NCBIStandalone.BlastErrorParser.parse
Bio.Blast.NCBIStandalone.BlastParser.parse
Bio.Blast.NCBIStandalone.PSIBlastParser.parse
Bio.Blast.NCBIXML.parse
Bio.Compass.parse
Bio.Emboss.Primer3.parse
Bio.Entrez.Parser.DataHandler.parse
Bio.Entrez.parse
Bio.ExPASy.Enzyme.parse
Bio.ExPASy.Prodoc.parse
Bio.ExPASy.Prosite.parse
Bio.GenBank.FeatureParser.parse
Bio.GenBank.RecordParser.parse
Bio.GenBank.Scanner.InsdcScanner.parse
Bio.Geo.parse
Bio.KEGG.Compound.parse
Bio.KEGG.Enzyme.parse
Bio.KEGG.KGML.KGML_parser.KGMLParser.parse
Bio.KEGG.KGML.KGML_parser.parse
Bio.KEGG.Map.parse
Bio.Medline.parse
Bio.Motif.parse
Bio.ParserSupport.AbstractParser.parse
Bio.Phylo.CDAOIO.Parser.parse
Bio.Phylo.CDAOIO.parse
Bio.Phylo.NeXMLIO.Parser.parse
Bio.Phylo.NeXMLIO.parse
Bio.Phylo.NewickIO.Parser.parse
Bio.Phylo.NewickIO.parse
Bio.Phylo.NexusIO.parse
Bio.Phylo.PhyloXMLIO.Parser.parse
Bio.Phylo.PhyloXMLIO.parse
Bio.Phylo._io.parse
Bio.SCOP.Cla.parse
Bio.SCOP.Des.parse
Bio.SCOP.Dom.parse
Bio.SCOP.Hie.parse
Bio.SCOP.Raf.parse
Bio.SearchIO.parse
Bio.SeqIO.UniprotIO.Parser.parse
Bio.SeqIO.parse
Bio.Sequencing.Ace.parse
Bio.Sequencing.Phd.parse
Bio.SwissProt.KeyWList.parse
Bio.SwissProt.parse
Bio.UniGene.parse
Bio.Wise.psw.parse
Bio.motifs.parse" class="py-name" href="#" onclick="return doclink('link-989', 'parse', 'link-989');">parse</a></tt><tt class="py-op">(</tt><tt class="py-name">StringIO</tt><tt class="py-op">(</tt><tt id="link-990" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-990', 'data', 'link-432');">data</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1671"></a><tt class="py-lineno">1671</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-991" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-991', 'data', 'link-432');">data</a></tt> </tt>
</div><a name="L1672"></a><tt class="py-lineno">1672</tt>  <tt class="py-line"> </tt>
<a name="L1673"></a><tt class="py-lineno">1673</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">version_info</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">3</tt><tt class="py-op">:</tt> </tt>
<a name="Iterator.next"></a><div id="Iterator.next-def"><a name="L1674"></a><tt class="py-lineno">1674</tt> <a class="py-toggle" href="#" id="Iterator.next-toggle" onclick="return toggle('Iterator.next');">-</a><tt class="py-line">        <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone.Iterator-class.html#next">next</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Iterator.next-collapsed" style="display:none;" pad="++++" indent="++++++++++++++++"></div><div id="Iterator.next-expanded"><a name="L1675"></a><tt class="py-lineno">1675</tt>  <tt class="py-line">            <tt class="py-docstring">"""Python 2 style alias for Python 3 style __next__ method."""</tt> </tt>
<a name="L1676"></a><tt class="py-lineno">1676</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-992" class="py-name" targets="Method Bio.AlignIO.ClustalIO.ClustalIterator.__next__()=Bio.AlignIO.ClustalIO.ClustalIterator-class.html#__next__,Method Bio.AlignIO.EmbossIO.EmbossIterator.__next__()=Bio.AlignIO.EmbossIO.EmbossIterator-class.html#__next__,Method Bio.AlignIO.Interfaces.AlignmentIterator.__next__()=Bio.AlignIO.Interfaces.AlignmentIterator-class.html#__next__,Method Bio.AlignIO.PhylipIO.PhylipIterator.__next__()=Bio.AlignIO.PhylipIO.PhylipIterator-class.html#__next__,Method Bio.AlignIO.PhylipIO.SequentialPhylipIterator.__next__()=Bio.AlignIO.PhylipIO.SequentialPhylipIterator-class.html#__next__,Method Bio.AlignIO.StockholmIO.StockholmIterator.__next__()=Bio.AlignIO.StockholmIO.StockholmIterator-class.html#__next__,Method Bio.Blast.NCBIStandalone.Iterator.__next__()=Bio.Blast.NCBIStandalone.Iterator-class.html#__next__,Method Bio.Blast.ParseBlastTable.BlastTableReader.__next__()=Bio.Blast.ParseBlastTable.BlastTableReader-class.html#__next__,Method Bio.DocSQL.IterationCursor.__next__()=Bio.DocSQL.IterationCursor-class.html#__next__,Method Bio.File.UndoHandle.__next__()=Bio.File.UndoHandle-class.html#__next__,Method Bio.GenBank.Iterator.__next__()=Bio.GenBank.Iterator-class.html#__next__,Method Bio.Nexus.Nexus.CharBuffer.__next__()=Bio.Nexus.Nexus.CharBuffer-class.html#__next__,Method Bio.PopGen.GenePop.Controller._FileIterator.__next__()=Bio.PopGen.GenePop.Controller._FileIterator-class.html#__next__,Method Bio.SeqIO.Interfaces.SequenceIterator.__next__()=Bio.SeqIO.Interfaces.SequenceIterator-class.html#__next__,Method Bio.bgzf.BgzfReader.__next__()=Bio.bgzf.BgzfReader-class.html#__next__"><a title="Bio.AlignIO.ClustalIO.ClustalIterator.__next__
Bio.AlignIO.EmbossIO.EmbossIterator.__next__
Bio.AlignIO.Interfaces.AlignmentIterator.__next__
Bio.AlignIO.PhylipIO.PhylipIterator.__next__
Bio.AlignIO.PhylipIO.SequentialPhylipIterator.__next__
Bio.AlignIO.StockholmIO.StockholmIterator.__next__
Bio.Blast.NCBIStandalone.Iterator.__next__
Bio.Blast.ParseBlastTable.BlastTableReader.__next__
Bio.DocSQL.IterationCursor.__next__
Bio.File.UndoHandle.__next__
Bio.GenBank.Iterator.__next__
Bio.Nexus.Nexus.CharBuffer.__next__
Bio.PopGen.GenePop.Controller._FileIterator.__next__
Bio.SeqIO.Interfaces.SequenceIterator.__next__
Bio.bgzf.BgzfReader.__next__" class="py-name" href="#" onclick="return doclink('link-992', '__next__', 'link-992');">__next__</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1677"></a><tt class="py-lineno">1677</tt>  <tt class="py-line"> </tt>
<a name="Iterator.__iter__"></a><div id="Iterator.__iter__-def"><a name="L1678"></a><tt class="py-lineno">1678</tt> <a class="py-toggle" href="#" id="Iterator.__iter__-toggle" onclick="return toggle('Iterator.__iter__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone.Iterator-class.html#__iter__">__iter__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Iterator.__iter__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Iterator.__iter__-expanded"><a name="L1679"></a><tt class="py-lineno">1679</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">iter</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-993" class="py-name"><a title="Bio.AlignIO.ClustalIO.ClustalIterator.__next__
Bio.AlignIO.EmbossIO.EmbossIterator.__next__
Bio.AlignIO.Interfaces.AlignmentIterator.__next__
Bio.AlignIO.PhylipIO.PhylipIterator.__next__
Bio.AlignIO.PhylipIO.SequentialPhylipIterator.__next__
Bio.AlignIO.StockholmIO.StockholmIterator.__next__
Bio.Blast.NCBIStandalone.Iterator.__next__
Bio.Blast.ParseBlastTable.BlastTableReader.__next__
Bio.DocSQL.IterationCursor.__next__
Bio.File.UndoHandle.__next__
Bio.GenBank.Iterator.__next__
Bio.Nexus.Nexus.CharBuffer.__next__
Bio.PopGen.GenePop.Controller._FileIterator.__next__
Bio.SeqIO.Interfaces.SequenceIterator.__next__
Bio.bgzf.BgzfReader.__next__" class="py-name" href="#" onclick="return doclink('link-993', '__next__', 'link-992');">__next__</a></tt><tt class="py-op">,</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1680"></a><tt class="py-lineno">1680</tt>  <tt class="py-line"> </tt>
<a name="L1681"></a><tt class="py-lineno">1681</tt>  <tt class="py-line"> </tt>
<a name="_re_search"></a><div id="_re_search-def"><a name="L1682"></a><tt class="py-lineno">1682</tt> <a class="py-toggle" href="#" id="_re_search-toggle" onclick="return toggle('_re_search');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone-module.html#_re_search">_re_search</a><tt class="py-op">(</tt><tt class="py-param">regex</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">,</tt> <tt class="py-param">error_msg</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_re_search-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_re_search-expanded"><a name="L1683"></a><tt class="py-lineno">1683</tt>  <tt class="py-line">    <tt class="py-name">m</tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-994" class="py-name"><a title="Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-994', 'search', 'link-102');">search</a></tt><tt class="py-op">(</tt><tt class="py-name">regex</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L1684"></a><tt class="py-lineno">1684</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">m</tt><tt class="py-op">:</tt> </tt>
<a name="L1685"></a><tt class="py-lineno">1685</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-name">error_msg</tt><tt class="py-op">)</tt> </tt>
<a name="L1686"></a><tt class="py-lineno">1686</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">m</tt><tt class="py-op">.</tt><tt class="py-name">groups</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1687"></a><tt class="py-lineno">1687</tt>  <tt class="py-line"> </tt>
<a name="L1688"></a><tt class="py-lineno">1688</tt>  <tt class="py-line"> </tt>
<a name="_get_cols"></a><div id="_get_cols-def"><a name="L1689"></a><tt class="py-lineno">1689</tt> <a class="py-toggle" href="#" id="_get_cols-toggle" onclick="return toggle('_get_cols');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone-module.html#_get_cols">_get_cols</a><tt class="py-op">(</tt><tt class="py-param">line</tt><tt class="py-op">,</tt> <tt class="py-param">cols_to_get</tt><tt class="py-op">,</tt> <tt class="py-param">ncols</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">expected</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_get_cols-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_get_cols-expanded"><a name="L1690"></a><tt class="py-lineno">1690</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">expected</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1691"></a><tt class="py-lineno">1691</tt>  <tt class="py-line">        <tt class="py-name">expected</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L1692"></a><tt class="py-lineno">1692</tt>  <tt class="py-line">    <tt class="py-name">cols</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-995" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-995', 'split', 'link-440');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1693"></a><tt class="py-lineno">1693</tt>  <tt class="py-line"> </tt>
<a name="L1694"></a><tt class="py-lineno">1694</tt>  <tt class="py-line">    <tt class="py-comment"># Check to make sure number of columns is correct</tt> </tt>
<a name="L1695"></a><tt class="py-lineno">1695</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">ncols</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">ncols</tt><tt class="py-op">:</tt> </tt>
<a name="L1696"></a><tt class="py-lineno">1696</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"I expected %d columns (got %d) in line\n%s"</tt> </tt>
<a name="L1697"></a><tt class="py-lineno">1697</tt>  <tt class="py-line">                         <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">ncols</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1698"></a><tt class="py-lineno">1698</tt>  <tt class="py-line"> </tt>
<a name="L1699"></a><tt class="py-lineno">1699</tt>  <tt class="py-line">    <tt class="py-comment"># Check to make sure columns contain the correct data</tt> </tt>
<a name="L1700"></a><tt class="py-lineno">1700</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">k</tt> <tt class="py-keyword">in</tt> <tt class="py-name">expected</tt><tt class="py-op">:</tt> </tt>
<a name="L1701"></a><tt class="py-lineno">1701</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-name">k</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-name">expected</tt><tt class="py-op">[</tt><tt class="py-name">k</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L1702"></a><tt class="py-lineno">1702</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"I expected '%s' in column %d in line\n%s"</tt> </tt>
<a name="L1703"></a><tt class="py-lineno">1703</tt>  <tt class="py-line">                             <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">expected</tt><tt class="py-op">[</tt><tt class="py-name">k</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">line</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1704"></a><tt class="py-lineno">1704</tt>  <tt class="py-line"> </tt>
<a name="L1705"></a><tt class="py-lineno">1705</tt>  <tt class="py-line">    <tt class="py-comment"># Construct the answer tuple</tt> </tt>
<a name="L1706"></a><tt class="py-lineno">1706</tt>  <tt class="py-line">    <tt id="link-996" class="py-name" targets="Variable Bio.Restriction.Restriction.AanI.results=Bio.Restriction.Restriction.AanI-class.html#results,Variable Bio.Restriction.Restriction.AarI.results=Bio.Restriction.Restriction.AarI-class.html#results,Variable Bio.Restriction.Restriction.AasI.results=Bio.Restriction.Restriction.AasI-class.html#results,Variable Bio.Restriction.Restriction.AatII.results=Bio.Restriction.Restriction.AatII-class.html#results,Variable Bio.Restriction.Restriction.AbaSI.results=Bio.Restriction.Restriction.AbaSI-class.html#results,Variable Bio.Restriction.Restriction.AbsI.results=Bio.Restriction.Restriction.AbsI-class.html#results,Variable Bio.Restriction.Restriction.Acc16I.results=Bio.Restriction.Restriction.Acc16I-class.html#results,Variable Bio.Restriction.Restriction.Acc36I.results=Bio.Restriction.Restriction.Acc36I-class.html#results,Variable Bio.Restriction.Restriction.Acc65I.results=Bio.Restriction.Restriction.Acc65I-class.html#results,Variable Bio.Restriction.Restriction.AccB1I.results=Bio.Restriction.Restriction.AccB1I-class.html#results,Variable Bio.Restriction.Restriction.AccB7I.results=Bio.Restriction.Restriction.AccB7I-class.html#results,Variable Bio.Restriction.Restriction.AccBSI.results=Bio.Restriction.Restriction.AccBSI-class.html#results,Variable Bio.Restriction.Restriction.AccI.results=Bio.Restriction.Restriction.AccI-class.html#results,Variable Bio.Restriction.Restriction.AccII.results=Bio.Restriction.Restriction.AccII-class.html#results,Variable Bio.Restriction.Restriction.AccIII.results=Bio.Restriction.Restriction.AccIII-class.html#results,Variable Bio.Restriction.Restriction.AceIII.results=Bio.Restriction.Restriction.AceIII-class.html#results,Variable Bio.Restriction.Restriction.AciI.results=Bio.Restriction.Restriction.AciI-class.html#results,Variable Bio.Restriction.Restriction.AclI.results=Bio.Restriction.Restriction.AclI-class.html#results,Variable Bio.Restriction.Restriction.AclWI.results=Bio.Restriction.Restriction.AclWI-class.html#results,Variable Bio.Restriction.Restriction.AcoI.results=Bio.Restriction.Restriction.AcoI-class.html#results,Variable Bio.Restriction.Restriction.AcsI.results=Bio.Restriction.Restriction.AcsI-class.html#results,Variable Bio.Restriction.Restriction.AcuI.results=Bio.Restriction.Restriction.AcuI-class.html#results,Variable Bio.Restriction.Restriction.AcvI.results=Bio.Restriction.Restriction.AcvI-class.html#results,Variable Bio.Restriction.Restriction.AcyI.results=Bio.Restriction.Restriction.AcyI-class.html#results,Variable Bio.Restriction.Restriction.AdeI.results=Bio.Restriction.Restriction.AdeI-class.html#results,Variable Bio.Restriction.Restriction.AfaI.results=Bio.Restriction.Restriction.AfaI-class.html#results,Variable Bio.Restriction.Restriction.AfeI.results=Bio.Restriction.Restriction.AfeI-class.html#results,Variable Bio.Restriction.Restriction.AfiI.results=Bio.Restriction.Restriction.AfiI-class.html#results,Variable Bio.Restriction.Restriction.AflII.results=Bio.Restriction.Restriction.AflII-class.html#results,Variable Bio.Restriction.Restriction.AflIII.results=Bio.Restriction.Restriction.AflIII-class.html#results,Variable Bio.Restriction.Restriction.AgeI.results=Bio.Restriction.Restriction.AgeI-class.html#results,Variable Bio.Restriction.Restriction.AgsI.results=Bio.Restriction.Restriction.AgsI-class.html#results,Variable Bio.Restriction.Restriction.AhaIII.results=Bio.Restriction.Restriction.AhaIII-class.html#results,Variable Bio.Restriction.Restriction.AhdI.results=Bio.Restriction.Restriction.AhdI-class.html#results,Variable Bio.Restriction.Restriction.AhlI.results=Bio.Restriction.Restriction.AhlI-class.html#results,Variable Bio.Restriction.Restriction.AjiI.results=Bio.Restriction.Restriction.AjiI-class.html#results,Variable Bio.Restriction.Restriction.AjnI.results=Bio.Restriction.Restriction.AjnI-class.html#results,Variable Bio.Restriction.Restriction.AjuI.results=Bio.Restriction.Restriction.AjuI-class.html#results,Variable Bio.Restriction.Restriction.AleI.results=Bio.Restriction.Restriction.AleI-class.html#results,Variable Bio.Restriction.Restriction.AlfI.results=Bio.Restriction.Restriction.AlfI-class.html#results,Variable Bio.Restriction.Restriction.AloI.results=Bio.Restriction.Restriction.AloI-class.html#results,Variable Bio.Restriction.Restriction.AluBI.results=Bio.Restriction.Restriction.AluBI-class.html#results,Variable Bio.Restriction.Restriction.AluI.results=Bio.Restriction.Restriction.AluI-class.html#results,Variable Bio.Restriction.Restriction.Alw21I.results=Bio.Restriction.Restriction.Alw21I-class.html#results,Variable Bio.Restriction.Restriction.Alw26I.results=Bio.Restriction.Restriction.Alw26I-class.html#results,Variable Bio.Restriction.Restriction.Alw44I.results=Bio.Restriction.Restriction.Alw44I-class.html#results,Variable Bio.Restriction.Restriction.AlwFI.results=Bio.Restriction.Restriction.AlwFI-class.html#results,Variable Bio.Restriction.Restriction.AlwI.results=Bio.Restriction.Restriction.AlwI-class.html#results,Variable Bio.Restriction.Restriction.AlwNI.results=Bio.Restriction.Restriction.AlwNI-class.html#results,Variable Bio.Restriction.Restriction.Ama87I.results=Bio.Restriction.Restriction.Ama87I-class.html#results,Variable Bio.Restriction.Restriction.Aor13HI.results=Bio.Restriction.Restriction.Aor13HI-class.html#results,Variable Bio.Restriction.Restriction.Aor51HI.results=Bio.Restriction.Restriction.Aor51HI-class.html#results,Variable Bio.Restriction.Restriction.AoxI.results=Bio.Restriction.Restriction.AoxI-class.html#results,Variable Bio.Restriction.Restriction.ApaBI.results=Bio.Restriction.Restriction.ApaBI-class.html#results,Variable Bio.Restriction.Restriction.ApaI.results=Bio.Restriction.Restriction.ApaI-class.html#results,Variable Bio.Restriction.Restriction.ApaLI.results=Bio.Restriction.Restriction.ApaLI-class.html#results,Variable Bio.Restriction.Restriction.ApeKI.results=Bio.Restriction.Restriction.ApeKI-class.html#results,Variable Bio.Restriction.Restriction.ApoI.results=Bio.Restriction.Restriction.ApoI-class.html#results,Variable Bio.Restriction.Restriction.ApyPI.results=Bio.Restriction.Restriction.ApyPI-class.html#results,Variable Bio.Restriction.Restriction.AquII.results=Bio.Restriction.Restriction.AquII-class.html#results,Variable Bio.Restriction.Restriction.AquIII.results=Bio.Restriction.Restriction.AquIII-class.html#results,Variable Bio.Restriction.Restriction.AquIV.results=Bio.Restriction.Restriction.AquIV-class.html#results,Variable Bio.Restriction.Restriction.ArsI.results=Bio.Restriction.Restriction.ArsI-class.html#results,Variable Bio.Restriction.Restriction.AscI.results=Bio.Restriction.Restriction.AscI-class.html#results,Variable Bio.Restriction.Restriction.AseI.results=Bio.Restriction.Restriction.AseI-class.html#results,Variable Bio.Restriction.Restriction.Asi256I.results=Bio.Restriction.Restriction.Asi256I-class.html#results,Variable Bio.Restriction.Restriction.AsiGI.results=Bio.Restriction.Restriction.AsiGI-class.html#results,Variable Bio.Restriction.Restriction.AsiSI.results=Bio.Restriction.Restriction.AsiSI-class.html#results,Variable Bio.Restriction.Restriction.Asp700I.results=Bio.Restriction.Restriction.Asp700I-class.html#results,Variable Bio.Restriction.Restriction.Asp718I.results=Bio.Restriction.Restriction.Asp718I-class.html#results,Variable Bio.Restriction.Restriction.AspA2I.results=Bio.Restriction.Restriction.AspA2I-class.html#results,Variable Bio.Restriction.Restriction.AspBHI.results=Bio.Restriction.Restriction.AspBHI-class.html#results,Variable Bio.Restriction.Restriction.AspLEI.results=Bio.Restriction.Restriction.AspLEI-class.html#results,Variable Bio.Restriction.Restriction.AspS9I.results=Bio.Restriction.Restriction.AspS9I-class.html#results,Variable Bio.Restriction.Restriction.AssI.results=Bio.Restriction.Restriction.AssI-class.html#results,Variable Bio.Restriction.Restriction.AsuC2I.results=Bio.Restriction.Restriction.AsuC2I-class.html#results,Variable Bio.Restriction.Restriction.AsuHPI.results=Bio.Restriction.Restriction.AsuHPI-class.html#results,Variable Bio.Restriction.Restriction.AsuI.results=Bio.Restriction.Restriction.AsuI-class.html#results,Variable Bio.Restriction.Restriction.AsuII.results=Bio.Restriction.Restriction.AsuII-class.html#results,Variable Bio.Restriction.Restriction.AsuNHI.results=Bio.Restriction.Restriction.AsuNHI-class.html#results,Variable Bio.Restriction.Restriction.AvaI.results=Bio.Restriction.Restriction.AvaI-class.html#results,Variable Bio.Restriction.Restriction.AvaII.results=Bio.Restriction.Restriction.AvaII-class.html#results,Variable Bio.Restriction.Restriction.AvaIII.results=Bio.Restriction.Restriction.AvaIII-class.html#results,Variable Bio.Restriction.Restriction.AvrII.results=Bio.Restriction.Restriction.AvrII-class.html#results,Variable Bio.Restriction.Restriction.AxyI.results=Bio.Restriction.Restriction.AxyI-class.html#results,Variable Bio.Restriction.Restriction.BaeGI.results=Bio.Restriction.Restriction.BaeGI-class.html#results,Variable Bio.Restriction.Restriction.BaeI.results=Bio.Restriction.Restriction.BaeI-class.html#results,Variable Bio.Restriction.Restriction.BalI.results=Bio.Restriction.Restriction.BalI-class.html#results,Variable Bio.Restriction.Restriction.BamHI.results=Bio.Restriction.Restriction.BamHI-class.html#results,Variable Bio.Restriction.Restriction.BanI.results=Bio.Restriction.Restriction.BanI-class.html#results,Variable Bio.Restriction.Restriction.BanII.results=Bio.Restriction.Restriction.BanII-class.html#results,Variable Bio.Restriction.Restriction.BarI.results=Bio.Restriction.Restriction.BarI-class.html#results,Variable Bio.Restriction.Restriction.BasI.results=Bio.Restriction.Restriction.BasI-class.html#results,Variable Bio.Restriction.Restriction.BauI.results=Bio.Restriction.Restriction.BauI-class.html#results,Variable Bio.Restriction.Restriction.Bbr7I.results=Bio.Restriction.Restriction.Bbr7I-class.html#results,Variable Bio.Restriction.Restriction.BbrPI.results=Bio.Restriction.Restriction.BbrPI-class.html#results,Variable Bio.Restriction.Restriction.BbsI.results=Bio.Restriction.Restriction.BbsI-class.html#results,Variable Bio.Restriction.Restriction.Bbv12I.results=Bio.Restriction.Restriction.Bbv12I-class.html#results,Variable Bio.Restriction.Restriction.BbvCI.results=Bio.Restriction.Restriction.BbvCI-class.html#results,Variable Bio.Restriction.Restriction.BbvI.results=Bio.Restriction.Restriction.BbvI-class.html#results,Variable Bio.Restriction.Restriction.BbvII.results=Bio.Restriction.Restriction.BbvII-class.html#results,Variable Bio.Restriction.Restriction.BccI.results=Bio.Restriction.Restriction.BccI-class.html#results,Variable Bio.Restriction.Restriction.Bce83I.results=Bio.Restriction.Restriction.Bce83I-class.html#results,Variable Bio.Restriction.Restriction.BceAI.results=Bio.Restriction.Restriction.BceAI-class.html#results,Variable Bio.Restriction.Restriction.BcefI.results=Bio.Restriction.Restriction.BcefI-class.html#results,Variable Bio.Restriction.Restriction.BcgI.results=Bio.Restriction.Restriction.BcgI-class.html#results,Variable Bio.Restriction.Restriction.BciT130I.results=Bio.Restriction.Restriction.BciT130I-class.html#results,Variable Bio.Restriction.Restriction.BciVI.results=Bio.Restriction.Restriction.BciVI-class.html#results,Variable Bio.Restriction.Restriction.BclI.results=Bio.Restriction.Restriction.BclI-class.html#results,Variable Bio.Restriction.Restriction.BcnI.results=Bio.Restriction.Restriction.BcnI-class.html#results,Variable Bio.Restriction.Restriction.BcoDI.results=Bio.Restriction.Restriction.BcoDI-class.html#results,Variable Bio.Restriction.Restriction.BcuI.results=Bio.Restriction.Restriction.BcuI-class.html#results,Variable Bio.Restriction.Restriction.BdaI.results=Bio.Restriction.Restriction.BdaI-class.html#results,Variable Bio.Restriction.Restriction.BetI.results=Bio.Restriction.Restriction.BetI-class.html#results,Variable Bio.Restriction.Restriction.BfaI.results=Bio.Restriction.Restriction.BfaI-class.html#results,Variable Bio.Restriction.Restriction.BfiI.results=Bio.Restriction.Restriction.BfiI-class.html#results,Variable Bio.Restriction.Restriction.BfmI.results=Bio.Restriction.Restriction.BfmI-class.html#results,Variable Bio.Restriction.Restriction.BfoI.results=Bio.Restriction.Restriction.BfoI-class.html#results,Variable Bio.Restriction.Restriction.BfrI.results=Bio.Restriction.Restriction.BfrI-class.html#results,Variable Bio.Restriction.Restriction.BfuAI.results=Bio.Restriction.Restriction.BfuAI-class.html#results,Variable Bio.Restriction.Restriction.BfuCI.results=Bio.Restriction.Restriction.BfuCI-class.html#results,Variable Bio.Restriction.Restriction.BfuI.results=Bio.Restriction.Restriction.BfuI-class.html#results,Variable Bio.Restriction.Restriction.BglI.results=Bio.Restriction.Restriction.BglI-class.html#results,Variable Bio.Restriction.Restriction.BglII.results=Bio.Restriction.Restriction.BglII-class.html#results,Variable Bio.Restriction.Restriction.BinI.results=Bio.Restriction.Restriction.BinI-class.html#results,Variable Bio.Restriction.Restriction.BisI.results=Bio.Restriction.Restriction.BisI-class.html#results,Variable Bio.Restriction.Restriction.BlnI.results=Bio.Restriction.Restriction.BlnI-class.html#results,Variable Bio.Restriction.Restriction.BlpI.results=Bio.Restriction.Restriction.BlpI-class.html#results,Variable Bio.Restriction.Restriction.BlsI.results=Bio.Restriction.Restriction.BlsI-class.html#results,Variable Bio.Restriction.Restriction.BmcAI.results=Bio.Restriction.Restriction.BmcAI-class.html#results,Variable Bio.Restriction.Restriction.Bme1390I.results=Bio.Restriction.Restriction.Bme1390I-class.html#results,Variable Bio.Restriction.Restriction.Bme18I.results=Bio.Restriction.Restriction.Bme18I-class.html#results,Variable Bio.Restriction.Restriction.BmeDI.results=Bio.Restriction.Restriction.BmeDI-class.html#results,Variable Bio.Restriction.Restriction.BmeRI.results=Bio.Restriction.Restriction.BmeRI-class.html#results,Variable Bio.Restriction.Restriction.BmeT110I.results=Bio.Restriction.Restriction.BmeT110I-class.html#results,Variable Bio.Restriction.Restriction.BmgBI.results=Bio.Restriction.Restriction.BmgBI-class.html#results,Variable Bio.Restriction.Restriction.BmgI.results=Bio.Restriction.Restriction.BmgI-class.html#results,Variable Bio.Restriction.Restriction.BmgT120I.results=Bio.Restriction.Restriction.BmgT120I-class.html#results,Variable Bio.Restriction.Restriction.BmiI.results=Bio.Restriction.Restriction.BmiI-class.html#results,Variable Bio.Restriction.Restriction.BmrFI.results=Bio.Restriction.Restriction.BmrFI-class.html#results,Variable Bio.Restriction.Restriction.BmrI.results=Bio.Restriction.Restriction.BmrI-class.html#results,Variable Bio.Restriction.Restriction.BmsI.results=Bio.Restriction.Restriction.BmsI-class.html#results,Variable Bio.Restriction.Restriction.BmtI.results=Bio.Restriction.Restriction.BmtI-class.html#results,Variable Bio.Restriction.Restriction.BmuI.results=Bio.Restriction.Restriction.BmuI-class.html#results,Variable Bio.Restriction.Restriction.BoxI.results=Bio.Restriction.Restriction.BoxI-class.html#results,Variable Bio.Restriction.Restriction.BpiI.results=Bio.Restriction.Restriction.BpiI-class.html#results,Variable Bio.Restriction.Restriction.BplI.results=Bio.Restriction.Restriction.BplI-class.html#results,Variable Bio.Restriction.Restriction.BpmI.results=Bio.Restriction.Restriction.BpmI-class.html#results,Variable Bio.Restriction.Restriction.Bpu10I.results=Bio.Restriction.Restriction.Bpu10I-class.html#results,Variable Bio.Restriction.Restriction.Bpu1102I.results=Bio.Restriction.Restriction.Bpu1102I-class.html#results,Variable Bio.Restriction.Restriction.Bpu14I.results=Bio.Restriction.Restriction.Bpu14I-class.html#results,Variable Bio.Restriction.Restriction.BpuEI.results=Bio.Restriction.Restriction.BpuEI-class.html#results,Variable Bio.Restriction.Restriction.BpuMI.results=Bio.Restriction.Restriction.BpuMI-class.html#results,Variable Bio.Restriction.Restriction.BpvUI.results=Bio.Restriction.Restriction.BpvUI-class.html#results,Variable Bio.Restriction.Restriction.Bsa29I.results=Bio.Restriction.Restriction.Bsa29I-class.html#results,Variable Bio.Restriction.Restriction.BsaAI.results=Bio.Restriction.Restriction.BsaAI-class.html#results,Variable Bio.Restriction.Restriction.BsaBI.results=Bio.Restriction.Restriction.BsaBI-class.html#results,Variable Bio.Restriction.Restriction.BsaHI.results=Bio.Restriction.Restriction.BsaHI-class.html#results,Variable Bio.Restriction.Restriction.BsaI.results=Bio.Restriction.Restriction.BsaI-class.html#results,Variable Bio.Restriction.Restriction.BsaJI.results=Bio.Restriction.Restriction.BsaJI-class.html#results,Variable Bio.Restriction.Restriction.BsaWI.results=Bio.Restriction.Restriction.BsaWI-class.html#results,Variable Bio.Restriction.Restriction.BsaXI.results=Bio.Restriction.Restriction.BsaXI-class.html#results,Variable Bio.Restriction.Restriction.BsbI.results=Bio.Restriction.Restriction.BsbI-class.html#results,Variable Bio.Restriction.Restriction.Bsc4I.results=Bio.Restriction.Restriction.Bsc4I-class.html#results,Variable Bio.Restriction.Restriction.BscAI.results=Bio.Restriction.Restriction.BscAI-class.html#results,Variable Bio.Restriction.Restriction.BscGI.results=Bio.Restriction.Restriction.BscGI-class.html#results,Variable Bio.Restriction.Restriction.Bse118I.results=Bio.Restriction.Restriction.Bse118I-class.html#results,Variable Bio.Restriction.Restriction.Bse1I.results=Bio.Restriction.Restriction.Bse1I-class.html#results,Variable Bio.Restriction.Restriction.Bse21I.results=Bio.Restriction.Restriction.Bse21I-class.html#results,Variable Bio.Restriction.Restriction.Bse3DI.results=Bio.Restriction.Restriction.Bse3DI-class.html#results,Variable Bio.Restriction.Restriction.Bse8I.results=Bio.Restriction.Restriction.Bse8I-class.html#results,Variable Bio.Restriction.Restriction.BseAI.results=Bio.Restriction.Restriction.BseAI-class.html#results,Variable Bio.Restriction.Restriction.BseBI.results=Bio.Restriction.Restriction.BseBI-class.html#results,Variable Bio.Restriction.Restriction.BseCI.results=Bio.Restriction.Restriction.BseCI-class.html#results,Variable Bio.Restriction.Restriction.BseDI.results=Bio.Restriction.Restriction.BseDI-class.html#results,Variable Bio.Restriction.Restriction.BseGI.results=Bio.Restriction.Restriction.BseGI-class.html#results,Variable Bio.Restriction.Restriction.BseJI.results=Bio.Restriction.Restriction.BseJI-class.html#results,Variable Bio.Restriction.Restriction.BseLI.results=Bio.Restriction.Restriction.BseLI-class.html#results,Variable Bio.Restriction.Restriction.BseMI.results=Bio.Restriction.Restriction.BseMI-class.html#results,Variable Bio.Restriction.Restriction.BseMII.results=Bio.Restriction.Restriction.BseMII-class.html#results,Variable Bio.Restriction.Restriction.BseNI.results=Bio.Restriction.Restriction.BseNI-class.html#results,Variable Bio.Restriction.Restriction.BsePI.results=Bio.Restriction.Restriction.BsePI-class.html#results,Variable Bio.Restriction.Restriction.BseRI.results=Bio.Restriction.Restriction.BseRI-class.html#results,Variable Bio.Restriction.Restriction.BseSI.results=Bio.Restriction.Restriction.BseSI-class.html#results,Variable Bio.Restriction.Restriction.BseX3I.results=Bio.Restriction.Restriction.BseX3I-class.html#results,Variable Bio.Restriction.Restriction.BseXI.results=Bio.Restriction.Restriction.BseXI-class.html#results,Variable Bio.Restriction.Restriction.BseYI.results=Bio.Restriction.Restriction.BseYI-class.html#results,Variable Bio.Restriction.Restriction.BsgI.results=Bio.Restriction.Restriction.BsgI-class.html#results,Variable Bio.Restriction.Restriction.Bsh1236I.results=Bio.Restriction.Restriction.Bsh1236I-class.html#results,Variable Bio.Restriction.Restriction.Bsh1285I.results=Bio.Restriction.Restriction.Bsh1285I-class.html#results,Variable Bio.Restriction.Restriction.BshFI.results=Bio.Restriction.Restriction.BshFI-class.html#results,Variable Bio.Restriction.Restriction.BshNI.results=Bio.Restriction.Restriction.BshNI-class.html#results,Variable Bio.Restriction.Restriction.BshTI.results=Bio.Restriction.Restriction.BshTI-class.html#results,Variable Bio.Restriction.Restriction.BshVI.results=Bio.Restriction.Restriction.BshVI-class.html#results,Variable Bio.Restriction.Restriction.BsiEI.results=Bio.Restriction.Restriction.BsiEI-class.html#results,Variable Bio.Restriction.Restriction.BsiHKAI.results=Bio.Restriction.Restriction.BsiHKAI-class.html#results,Variable Bio.Restriction.Restriction.BsiHKCI.results=Bio.Restriction.Restriction.BsiHKCI-class.html#results,Variable Bio.Restriction.Restriction.BsiI.results=Bio.Restriction.Restriction.BsiI-class.html#results,Variable Bio.Restriction.Restriction.BsiSI.results=Bio.Restriction.Restriction.BsiSI-class.html#results,Variable Bio.Restriction.Restriction.BsiWI.results=Bio.Restriction.Restriction.BsiWI-class.html#results,Variable Bio.Restriction.Restriction.BsiYI.results=Bio.Restriction.Restriction.BsiYI-class.html#results,Variable Bio.Restriction.Restriction.BslFI.results=Bio.Restriction.Restriction.BslFI-class.html#results,Variable Bio.Restriction.Restriction.BslI.results=Bio.Restriction.Restriction.BslI-class.html#results,Variable Bio.Restriction.Restriction.BsmAI.results=Bio.Restriction.Restriction.BsmAI-class.html#results,Variable Bio.Restriction.Restriction.BsmBI.results=Bio.Restriction.Restriction.BsmBI-class.html#results,Variable Bio.Restriction.Restriction.BsmFI.results=Bio.Restriction.Restriction.BsmFI-class.html#results,Variable Bio.Restriction.Restriction.BsmI.results=Bio.Restriction.Restriction.BsmI-class.html#results,Variable Bio.Restriction.Restriction.BsnI.results=Bio.Restriction.Restriction.BsnI-class.html#results,Variable Bio.Restriction.Restriction.Bso31I.results=Bio.Restriction.Restriction.Bso31I-class.html#results,Variable Bio.Restriction.Restriction.BsoBI.results=Bio.Restriction.Restriction.BsoBI-class.html#results,Variable Bio.Restriction.Restriction.Bsp119I.results=Bio.Restriction.Restriction.Bsp119I-class.html#results,Variable Bio.Restriction.Restriction.Bsp120I.results=Bio.Restriction.Restriction.Bsp120I-class.html#results,Variable Bio.Restriction.Restriction.Bsp1286I.results=Bio.Restriction.Restriction.Bsp1286I-class.html#results,Variable Bio.Restriction.Restriction.Bsp13I.results=Bio.Restriction.Restriction.Bsp13I-class.html#results,Variable Bio.Restriction.Restriction.Bsp1407I.results=Bio.Restriction.Restriction.Bsp1407I-class.html#results,Variable Bio.Restriction.Restriction.Bsp143I.results=Bio.Restriction.Restriction.Bsp143I-class.html#results,Variable Bio.Restriction.Restriction.Bsp1720I.results=Bio.Restriction.Restriction.Bsp1720I-class.html#results,Variable Bio.Restriction.Restriction.Bsp19I.results=Bio.Restriction.Restriction.Bsp19I-class.html#results,Variable Bio.Restriction.Restriction.Bsp24I.results=Bio.Restriction.Restriction.Bsp24I-class.html#results,Variable Bio.Restriction.Restriction.Bsp68I.results=Bio.Restriction.Restriction.Bsp68I-class.html#results,Variable Bio.Restriction.Restriction.BspACI.results=Bio.Restriction.Restriction.BspACI-class.html#results,Variable Bio.Restriction.Restriction.BspCNI.results=Bio.Restriction.Restriction.BspCNI-class.html#results,Variable Bio.Restriction.Restriction.BspD6I.results=Bio.Restriction.Restriction.BspD6I-class.html#results,Variable Bio.Restriction.Restriction.BspDI.results=Bio.Restriction.Restriction.BspDI-class.html#results,Variable Bio.Restriction.Restriction.BspEI.results=Bio.Restriction.Restriction.BspEI-class.html#results,Variable Bio.Restriction.Restriction.BspFNI.results=Bio.Restriction.Restriction.BspFNI-class.html#results,Variable Bio.Restriction.Restriction.BspGI.results=Bio.Restriction.Restriction.BspGI-class.html#results,Variable Bio.Restriction.Restriction.BspHI.results=Bio.Restriction.Restriction.BspHI-class.html#results,Variable Bio.Restriction.Restriction.BspLI.results=Bio.Restriction.Restriction.BspLI-class.html#results,Variable Bio.Restriction.Restriction.BspLU11I.results=Bio.Restriction.Restriction.BspLU11I-class.html#results,Variable Bio.Restriction.Restriction.BspMI.results=Bio.Restriction.Restriction.BspMI-class.html#results,Variable Bio.Restriction.Restriction.BspMII.results=Bio.Restriction.Restriction.BspMII-class.html#results,Variable Bio.Restriction.Restriction.BspNCI.results=Bio.Restriction.Restriction.BspNCI-class.html#results,Variable Bio.Restriction.Restriction.BspOI.results=Bio.Restriction.Restriction.BspOI-class.html#results,Variable Bio.Restriction.Restriction.BspPI.results=Bio.Restriction.Restriction.BspPI-class.html#results,Variable Bio.Restriction.Restriction.BspQI.results=Bio.Restriction.Restriction.BspQI-class.html#results,Variable Bio.Restriction.Restriction.BspT104I.results=Bio.Restriction.Restriction.BspT104I-class.html#results,Variable Bio.Restriction.Restriction.BspT107I.results=Bio.Restriction.Restriction.BspT107I-class.html#results,Variable Bio.Restriction.Restriction.BspTI.results=Bio.Restriction.Restriction.BspTI-class.html#results,Variable Bio.Restriction.Restriction.BsrBI.results=Bio.Restriction.Restriction.BsrBI-class.html#results,Variable Bio.Restriction.Restriction.BsrDI.results=Bio.Restriction.Restriction.BsrDI-class.html#results,Variable Bio.Restriction.Restriction.BsrFI.results=Bio.Restriction.Restriction.BsrFI-class.html#results,Variable Bio.Restriction.Restriction.BsrGI.results=Bio.Restriction.Restriction.BsrGI-class.html#results,Variable Bio.Restriction.Restriction.BsrI.results=Bio.Restriction.Restriction.BsrI-class.html#results,Variable Bio.Restriction.Restriction.BsrSI.results=Bio.Restriction.Restriction.BsrSI-class.html#results,Variable Bio.Restriction.Restriction.BssAI.results=Bio.Restriction.Restriction.BssAI-class.html#results,Variable Bio.Restriction.Restriction.BssECI.results=Bio.Restriction.Restriction.BssECI-class.html#results,Variable Bio.Restriction.Restriction.BssHII.results=Bio.Restriction.Restriction.BssHII-class.html#results,Variable Bio.Restriction.Restriction.BssKI.results=Bio.Restriction.Restriction.BssKI-class.html#results,Variable Bio.Restriction.Restriction.BssMI.results=Bio.Restriction.Restriction.BssMI-class.html#results,Variable Bio.Restriction.Restriction.BssNAI.results=Bio.Restriction.Restriction.BssNAI-class.html#results,Variable Bio.Restriction.Restriction.BssNI.results=Bio.Restriction.Restriction.BssNI-class.html#results,Variable Bio.Restriction.Restriction.BssSI.results=Bio.Restriction.Restriction.BssSI-class.html#results,Variable Bio.Restriction.Restriction.BssT1I.results=Bio.Restriction.Restriction.BssT1I-class.html#results,Variable Bio.Restriction.Restriction.Bst1107I.results=Bio.Restriction.Restriction.Bst1107I-class.html#results,Variable Bio.Restriction.Restriction.Bst2BI.results=Bio.Restriction.Restriction.Bst2BI-class.html#results,Variable Bio.Restriction.Restriction.Bst2UI.results=Bio.Restriction.Restriction.Bst2UI-class.html#results,Variable Bio.Restriction.Restriction.Bst4CI.results=Bio.Restriction.Restriction.Bst4CI-class.html#results,Variable Bio.Restriction.Restriction.Bst6I.results=Bio.Restriction.Restriction.Bst6I-class.html#results,Variable Bio.Restriction.Restriction.BstACI.results=Bio.Restriction.Restriction.BstACI-class.html#results,Variable Bio.Restriction.Restriction.BstAFI.results=Bio.Restriction.Restriction.BstAFI-class.html#results,Variable Bio.Restriction.Restriction.BstAPI.results=Bio.Restriction.Restriction.BstAPI-class.html#results,Variable Bio.Restriction.Restriction.BstAUI.results=Bio.Restriction.Restriction.BstAUI-class.html#results,Variable Bio.Restriction.Restriction.BstBAI.results=Bio.Restriction.Restriction.BstBAI-class.html#results,Variable Bio.Restriction.Restriction.BstBI.results=Bio.Restriction.Restriction.BstBI-class.html#results,Variable Bio.Restriction.Restriction.BstC8I.results=Bio.Restriction.Restriction.BstC8I-class.html#results,Variable Bio.Restriction.Restriction.BstDEI.results=Bio.Restriction.Restriction.BstDEI-class.html#results,Variable Bio.Restriction.Restriction.BstDSI.results=Bio.Restriction.Restriction.BstDSI-class.html#results,Variable Bio.Restriction.Restriction.BstEII.results=Bio.Restriction.Restriction.BstEII-class.html#results,Variable Bio.Restriction.Restriction.BstENI.results=Bio.Restriction.Restriction.BstENI-class.html#results,Variable Bio.Restriction.Restriction.BstF5I.results=Bio.Restriction.Restriction.BstF5I-class.html#results,Variable Bio.Restriction.Restriction.BstFNI.results=Bio.Restriction.Restriction.BstFNI-class.html#results,Variable Bio.Restriction.Restriction.BstH2I.results=Bio.Restriction.Restriction.BstH2I-class.html#results,Variable Bio.Restriction.Restriction.BstHHI.results=Bio.Restriction.Restriction.BstHHI-class.html#results,Variable Bio.Restriction.Restriction.BstKTI.results=Bio.Restriction.Restriction.BstKTI-class.html#results,Variable Bio.Restriction.Restriction.BstMAI.results=Bio.Restriction.Restriction.BstMAI-class.html#results,Variable Bio.Restriction.Restriction.BstMBI.results=Bio.Restriction.Restriction.BstMBI-class.html#results,Variable Bio.Restriction.Restriction.BstMCI.results=Bio.Restriction.Restriction.BstMCI-class.html#results,Variable Bio.Restriction.Restriction.BstMWI.results=Bio.Restriction.Restriction.BstMWI-class.html#results,Variable Bio.Restriction.Restriction.BstNI.results=Bio.Restriction.Restriction.BstNI-class.html#results,Variable Bio.Restriction.Restriction.BstNSI.results=Bio.Restriction.Restriction.BstNSI-class.html#results,Variable Bio.Restriction.Restriction.BstOI.results=Bio.Restriction.Restriction.BstOI-class.html#results,Variable Bio.Restriction.Restriction.BstPAI.results=Bio.Restriction.Restriction.BstPAI-class.html#results,Variable Bio.Restriction.Restriction.BstPI.results=Bio.Restriction.Restriction.BstPI-class.html#results,Variable Bio.Restriction.Restriction.BstSCI.results=Bio.Restriction.Restriction.BstSCI-class.html#results,Variable Bio.Restriction.Restriction.BstSFI.results=Bio.Restriction.Restriction.BstSFI-class.html#results,Variable Bio.Restriction.Restriction.BstSLI.results=Bio.Restriction.Restriction.BstSLI-class.html#results,Variable Bio.Restriction.Restriction.BstSNI.results=Bio.Restriction.Restriction.BstSNI-class.html#results,Variable Bio.Restriction.Restriction.BstUI.results=Bio.Restriction.Restriction.BstUI-class.html#results,Variable Bio.Restriction.Restriction.BstV1I.results=Bio.Restriction.Restriction.BstV1I-class.html#results,Variable Bio.Restriction.Restriction.BstV2I.results=Bio.Restriction.Restriction.BstV2I-class.html#results,Variable Bio.Restriction.Restriction.BstX2I.results=Bio.Restriction.Restriction.BstX2I-class.html#results,Variable Bio.Restriction.Restriction.BstXI.results=Bio.Restriction.Restriction.BstXI-class.html#results,Variable Bio.Restriction.Restriction.BstYI.results=Bio.Restriction.Restriction.BstYI-class.html#results,Variable Bio.Restriction.Restriction.BstZ17I.results=Bio.Restriction.Restriction.BstZ17I-class.html#results,Variable Bio.Restriction.Restriction.BstZI.results=Bio.Restriction.Restriction.BstZI-class.html#results,Variable Bio.Restriction.Restriction.Bsu15I.results=Bio.Restriction.Restriction.Bsu15I-class.html#results,Variable Bio.Restriction.Restriction.Bsu36I.results=Bio.Restriction.Restriction.Bsu36I-class.html#results,Variable Bio.Restriction.Restriction.BsuI.results=Bio.Restriction.Restriction.BsuI-class.html#results,Variable Bio.Restriction.Restriction.BsuRI.results=Bio.Restriction.Restriction.BsuRI-class.html#results,Variable Bio.Restriction.Restriction.BtgI.results=Bio.Restriction.Restriction.BtgI-class.html#results,Variable Bio.Restriction.Restriction.BtgZI.results=Bio.Restriction.Restriction.BtgZI-class.html#results,Variable Bio.Restriction.Restriction.BthCI.results=Bio.Restriction.Restriction.BthCI-class.html#results,Variable Bio.Restriction.Restriction.BtrI.results=Bio.Restriction.Restriction.BtrI-class.html#results,Variable Bio.Restriction.Restriction.BtsCI.results=Bio.Restriction.Restriction.BtsCI-class.html#results,Variable Bio.Restriction.Restriction.BtsI.results=Bio.Restriction.Restriction.BtsI-class.html#results,Variable Bio.Restriction.Restriction.BtsIMutI.results=Bio.Restriction.Restriction.BtsIMutI-class.html#results,Variable Bio.Restriction.Restriction.BtuMI.results=Bio.Restriction.Restriction.BtuMI-class.html#results,Variable Bio.Restriction.Restriction.BveI.results=Bio.Restriction.Restriction.BveI-class.html#results,Variable Bio.Restriction.Restriction.Cac8I.results=Bio.Restriction.Restriction.Cac8I-class.html#results,Variable Bio.Restriction.Restriction.CaiI.results=Bio.Restriction.Restriction.CaiI-class.html#results,Variable Bio.Restriction.Restriction.CauII.results=Bio.Restriction.Restriction.CauII-class.html#results,Variable Bio.Restriction.Restriction.CchII.results=Bio.Restriction.Restriction.CchII-class.html#results,Variable Bio.Restriction.Restriction.CchIII.results=Bio.Restriction.Restriction.CchIII-class.html#results,Variable Bio.Restriction.Restriction.CciI.results=Bio.Restriction.Restriction.CciI-class.html#results,Variable Bio.Restriction.Restriction.CciNI.results=Bio.Restriction.Restriction.CciNI-class.html#results,Variable Bio.Restriction.Restriction.Cdi630V.results=Bio.Restriction.Restriction.Cdi630V-class.html#results,Variable Bio.Restriction.Restriction.CdiI.results=Bio.Restriction.Restriction.CdiI-class.html#results,Variable Bio.Restriction.Restriction.CdpI.results=Bio.Restriction.Restriction.CdpI-class.html#results,Variable Bio.Restriction.Restriction.CfoI.results=Bio.Restriction.Restriction.CfoI-class.html#results,Variable Bio.Restriction.Restriction.Cfr10I.results=Bio.Restriction.Restriction.Cfr10I-class.html#results,Variable Bio.Restriction.Restriction.Cfr13I.results=Bio.Restriction.Restriction.Cfr13I-class.html#results,Variable Bio.Restriction.Restriction.Cfr42I.results=Bio.Restriction.Restriction.Cfr42I-class.html#results,Variable Bio.Restriction.Restriction.Cfr9I.results=Bio.Restriction.Restriction.Cfr9I-class.html#results,Variable Bio.Restriction.Restriction.CfrI.results=Bio.Restriction.Restriction.CfrI-class.html#results,Variable Bio.Restriction.Restriction.Cgl13032I.results=Bio.Restriction.Restriction.Cgl13032I-class.html#results,Variable Bio.Restriction.Restriction.Cgl13032II.results=Bio.Restriction.Restriction.Cgl13032II-class.html#results,Variable Bio.Restriction.Restriction.ChaI.results=Bio.Restriction.Restriction.ChaI-class.html#results,Variable Bio.Restriction.Restriction.CjeFIII.results=Bio.Restriction.Restriction.CjeFIII-class.html#results,Variable Bio.Restriction.Restriction.CjeFV.results=Bio.Restriction.Restriction.CjeFV-class.html#results,Variable Bio.Restriction.Restriction.CjeI.results=Bio.Restriction.Restriction.CjeI-class.html#results,Variable Bio.Restriction.Restriction.CjeNII.results=Bio.Restriction.Restriction.CjeNII-class.html#results,Variable Bio.Restriction.Restriction.CjeNIII.results=Bio.Restriction.Restriction.CjeNIII-class.html#results,Variable Bio.Restriction.Restriction.CjeP659IV.results=Bio.Restriction.Restriction.CjeP659IV-class.html#results,Variable Bio.Restriction.Restriction.CjePI.results=Bio.Restriction.Restriction.CjePI-class.html#results,Variable Bio.Restriction.Restriction.CjuI.results=Bio.Restriction.Restriction.CjuI-class.html#results,Variable Bio.Restriction.Restriction.CjuII.results=Bio.Restriction.Restriction.CjuII-class.html#results,Variable Bio.Restriction.Restriction.ClaI.results=Bio.Restriction.Restriction.ClaI-class.html#results,Variable Bio.Restriction.Restriction.CpoI.results=Bio.Restriction.Restriction.CpoI-class.html#results,Variable Bio.Restriction.Restriction.CseI.results=Bio.Restriction.Restriction.CseI-class.html#results,Variable Bio.Restriction.Restriction.CsiI.results=Bio.Restriction.Restriction.CsiI-class.html#results,Variable Bio.Restriction.Restriction.Csp6I.results=Bio.Restriction.Restriction.Csp6I-class.html#results,Variable Bio.Restriction.Restriction.CspAI.results=Bio.Restriction.Restriction.CspAI-class.html#results,Variable Bio.Restriction.Restriction.CspCI.results=Bio.Restriction.Restriction.CspCI-class.html#results,Variable Bio.Restriction.Restriction.CspI.results=Bio.Restriction.Restriction.CspI-class.html#results,Variable Bio.Restriction.Restriction.CstMI.results=Bio.Restriction.Restriction.CstMI-class.html#results,Variable Bio.Restriction.Restriction.CviAII.results=Bio.Restriction.Restriction.CviAII-class.html#results,Variable Bio.Restriction.Restriction.CviJI.results=Bio.Restriction.Restriction.CviJI-class.html#results,Variable Bio.Restriction.Restriction.CviKI_1.results=Bio.Restriction.Restriction.CviKI_1-class.html#results,Variable Bio.Restriction.Restriction.CviQI.results=Bio.Restriction.Restriction.CviQI-class.html#results,Variable Bio.Restriction.Restriction.CviRI.results=Bio.Restriction.Restriction.CviRI-class.html#results,Variable Bio.Restriction.Restriction.DdeI.results=Bio.Restriction.Restriction.DdeI-class.html#results,Variable Bio.Restriction.Restriction.DinI.results=Bio.Restriction.Restriction.DinI-class.html#results,Variable Bio.Restriction.Restriction.DpnI.results=Bio.Restriction.Restriction.DpnI-class.html#results,Variable Bio.Restriction.Restriction.DpnII.results=Bio.Restriction.Restriction.DpnII-class.html#results,Variable Bio.Restriction.Restriction.DraI.results=Bio.Restriction.Restriction.DraI-class.html#results,Variable Bio.Restriction.Restriction.DraII.results=Bio.Restriction.Restriction.DraII-class.html#results,Variable Bio.Restriction.Restriction.DraIII.results=Bio.Restriction.Restriction.DraIII-class.html#results,Variable Bio.Restriction.Restriction.DraRI.results=Bio.Restriction.Restriction.DraRI-class.html#results,Variable Bio.Restriction.Restriction.DrdI.results=Bio.Restriction.Restriction.DrdI-class.html#results,Variable Bio.Restriction.Restriction.DrdII.results=Bio.Restriction.Restriction.DrdII-class.html#results,Variable Bio.Restriction.Restriction.DriI.results=Bio.Restriction.Restriction.DriI-class.html#results,Variable Bio.Restriction.Restriction.DsaI.results=Bio.Restriction.Restriction.DsaI-class.html#results,Variable Bio.Restriction.Restriction.DseDI.results=Bio.Restriction.Restriction.DseDI-class.html#results,Variable Bio.Restriction.Restriction.EaeI.results=Bio.Restriction.Restriction.EaeI-class.html#results,Variable Bio.Restriction.Restriction.EagI.results=Bio.Restriction.Restriction.EagI-class.html#results,Variable Bio.Restriction.Restriction.Eam1104I.results=Bio.Restriction.Restriction.Eam1104I-class.html#results,Variable Bio.Restriction.Restriction.Eam1105I.results=Bio.Restriction.Restriction.Eam1105I-class.html#results,Variable Bio.Restriction.Restriction.EarI.results=Bio.Restriction.Restriction.EarI-class.html#results,Variable Bio.Restriction.Restriction.EciI.results=Bio.Restriction.Restriction.EciI-class.html#results,Variable Bio.Restriction.Restriction.Ecl136II.results=Bio.Restriction.Restriction.Ecl136II-class.html#results,Variable Bio.Restriction.Restriction.EclXI.results=Bio.Restriction.Restriction.EclXI-class.html#results,Variable Bio.Restriction.Restriction.Eco105I.results=Bio.Restriction.Restriction.Eco105I-class.html#results,Variable Bio.Restriction.Restriction.Eco130I.results=Bio.Restriction.Restriction.Eco130I-class.html#results,Variable Bio.Restriction.Restriction.Eco147I.results=Bio.Restriction.Restriction.Eco147I-class.html#results,Variable Bio.Restriction.Restriction.Eco24I.results=Bio.Restriction.Restriction.Eco24I-class.html#results,Variable Bio.Restriction.Restriction.Eco31I.results=Bio.Restriction.Restriction.Eco31I-class.html#results,Variable Bio.Restriction.Restriction.Eco32I.results=Bio.Restriction.Restriction.Eco32I-class.html#results,Variable Bio.Restriction.Restriction.Eco47I.results=Bio.Restriction.Restriction.Eco47I-class.html#results,Variable Bio.Restriction.Restriction.Eco47III.results=Bio.Restriction.Restriction.Eco47III-class.html#results,Variable Bio.Restriction.Restriction.Eco52I.results=Bio.Restriction.Restriction.Eco52I-class.html#results,Variable Bio.Restriction.Restriction.Eco53kI.results=Bio.Restriction.Restriction.Eco53kI-class.html#results,Variable Bio.Restriction.Restriction.Eco57I.results=Bio.Restriction.Restriction.Eco57I-class.html#results,Variable Bio.Restriction.Restriction.Eco57MI.results=Bio.Restriction.Restriction.Eco57MI-class.html#results,Variable Bio.Restriction.Restriction.Eco72I.results=Bio.Restriction.Restriction.Eco72I-class.html#results,Variable Bio.Restriction.Restriction.Eco81I.results=Bio.Restriction.Restriction.Eco81I-class.html#results,Variable Bio.Restriction.Restriction.Eco88I.results=Bio.Restriction.Restriction.Eco88I-class.html#results,Variable Bio.Restriction.Restriction.Eco91I.results=Bio.Restriction.Restriction.Eco91I-class.html#results,Variable Bio.Restriction.Restriction.EcoHI.results=Bio.Restriction.Restriction.EcoHI-class.html#results,Variable Bio.Restriction.Restriction.EcoICRI.results=Bio.Restriction.Restriction.EcoICRI-class.html#results,Variable Bio.Restriction.Restriction.EcoNI.results=Bio.Restriction.Restriction.EcoNI-class.html#results,Variable Bio.Restriction.Restriction.EcoO109I.results=Bio.Restriction.Restriction.EcoO109I-class.html#results,Variable Bio.Restriction.Restriction.EcoO65I.results=Bio.Restriction.Restriction.EcoO65I-class.html#results,Variable Bio.Restriction.Restriction.EcoRI.results=Bio.Restriction.Restriction.EcoRI-class.html#results,Variable Bio.Restriction.Restriction.EcoRII.results=Bio.Restriction.Restriction.EcoRII-class.html#results,Variable Bio.Restriction.Restriction.EcoRV.results=Bio.Restriction.Restriction.EcoRV-class.html#results,Variable Bio.Restriction.Restriction.EcoT14I.results=Bio.Restriction.Restriction.EcoT14I-class.html#results,Variable Bio.Restriction.Restriction.EcoT22I.results=Bio.Restriction.Restriction.EcoT22I-class.html#results,Variable Bio.Restriction.Restriction.EcoT38I.results=Bio.Restriction.Restriction.EcoT38I-class.html#results,Variable Bio.Restriction.Restriction.EgeI.results=Bio.Restriction.Restriction.EgeI-class.html#results,Variable Bio.Restriction.Restriction.EheI.results=Bio.Restriction.Restriction.EheI-class.html#results,Variable Bio.Restriction.Restriction.ErhI.results=Bio.Restriction.Restriction.ErhI-class.html#results,Variable Bio.Restriction.Restriction.EsaBC3I.results=Bio.Restriction.Restriction.EsaBC3I-class.html#results,Variable Bio.Restriction.Restriction.EsaSSI.results=Bio.Restriction.Restriction.EsaSSI-class.html#results,Variable Bio.Restriction.Restriction.Esp3I.results=Bio.Restriction.Restriction.Esp3I-class.html#results,Variable Bio.Restriction.Restriction.EspI.results=Bio.Restriction.Restriction.EspI-class.html#results,Variable Bio.Restriction.Restriction.FaeI.results=Bio.Restriction.Restriction.FaeI-class.html#results,Variable Bio.Restriction.Restriction.FaiI.results=Bio.Restriction.Restriction.FaiI-class.html#results,Variable Bio.Restriction.Restriction.FalI.results=Bio.Restriction.Restriction.FalI-class.html#results,Variable Bio.Restriction.Restriction.FaqI.results=Bio.Restriction.Restriction.FaqI-class.html#results,Variable Bio.Restriction.Restriction.FatI.results=Bio.Restriction.Restriction.FatI-class.html#results,Variable Bio.Restriction.Restriction.FauI.results=Bio.Restriction.Restriction.FauI-class.html#results,Variable Bio.Restriction.Restriction.FauNDI.results=Bio.Restriction.Restriction.FauNDI-class.html#results,Variable Bio.Restriction.Restriction.FbaI.results=Bio.Restriction.Restriction.FbaI-class.html#results,Variable Bio.Restriction.Restriction.FblI.results=Bio.Restriction.Restriction.FblI-class.html#results,Variable Bio.Restriction.Restriction.FinI.results=Bio.Restriction.Restriction.FinI-class.html#results,Variable Bio.Restriction.Restriction.FmuI.results=Bio.Restriction.Restriction.FmuI-class.html#results,Variable Bio.Restriction.Restriction.Fnu4HI.results=Bio.Restriction.Restriction.Fnu4HI-class.html#results,Variable Bio.Restriction.Restriction.FnuDII.results=Bio.Restriction.Restriction.FnuDII-class.html#results,Variable Bio.Restriction.Restriction.FokI.results=Bio.Restriction.Restriction.FokI-class.html#results,Variable Bio.Restriction.Restriction.FriOI.results=Bio.Restriction.Restriction.FriOI-class.html#results,Variable Bio.Restriction.Restriction.FseI.results=Bio.Restriction.Restriction.FseI-class.html#results,Variable Bio.Restriction.Restriction.Fsp4HI.results=Bio.Restriction.Restriction.Fsp4HI-class.html#results,Variable Bio.Restriction.Restriction.FspAI.results=Bio.Restriction.Restriction.FspAI-class.html#results,Variable Bio.Restriction.Restriction.FspBI.results=Bio.Restriction.Restriction.FspBI-class.html#results,Variable Bio.Restriction.Restriction.FspEI.results=Bio.Restriction.Restriction.FspEI-class.html#results,Variable Bio.Restriction.Restriction.FspI.results=Bio.Restriction.Restriction.FspI-class.html#results,Variable Bio.Restriction.Restriction.GauT27I.results=Bio.Restriction.Restriction.GauT27I-class.html#results,Variable Bio.Restriction.Restriction.GdiII.results=Bio.Restriction.Restriction.GdiII-class.html#results,Variable Bio.Restriction.Restriction.GlaI.results=Bio.Restriction.Restriction.GlaI-class.html#results,Variable Bio.Restriction.Restriction.GluI.results=Bio.Restriction.Restriction.GluI-class.html#results,Variable Bio.Restriction.Restriction.GsaI.results=Bio.Restriction.Restriction.GsaI-class.html#results,Variable Bio.Restriction.Restriction.GsuI.results=Bio.Restriction.Restriction.GsuI-class.html#results,Variable Bio.Restriction.Restriction.HaeI.results=Bio.Restriction.Restriction.HaeI-class.html#results,Variable Bio.Restriction.Restriction.HaeII.results=Bio.Restriction.Restriction.HaeII-class.html#results,Variable Bio.Restriction.Restriction.HaeIII.results=Bio.Restriction.Restriction.HaeIII-class.html#results,Variable Bio.Restriction.Restriction.HapII.results=Bio.Restriction.Restriction.HapII-class.html#results,Variable Bio.Restriction.Restriction.HauII.results=Bio.Restriction.Restriction.HauII-class.html#results,Variable Bio.Restriction.Restriction.HgaI.results=Bio.Restriction.Restriction.HgaI-class.html#results,Variable Bio.Restriction.Restriction.HgiAI.results=Bio.Restriction.Restriction.HgiAI-class.html#results,Variable Bio.Restriction.Restriction.HgiCI.results=Bio.Restriction.Restriction.HgiCI-class.html#results,Variable Bio.Restriction.Restriction.HgiEII.results=Bio.Restriction.Restriction.HgiEII-class.html#results,Variable Bio.Restriction.Restriction.HgiJII.results=Bio.Restriction.Restriction.HgiJII-class.html#results,Variable Bio.Restriction.Restriction.HhaI.results=Bio.Restriction.Restriction.HhaI-class.html#results,Variable Bio.Restriction.Restriction.Hin1I.results=Bio.Restriction.Restriction.Hin1I-class.html#results,Variable Bio.Restriction.Restriction.Hin1II.results=Bio.Restriction.Restriction.Hin1II-class.html#results,Variable Bio.Restriction.Restriction.Hin4I.results=Bio.Restriction.Restriction.Hin4I-class.html#results,Variable Bio.Restriction.Restriction.Hin4II.results=Bio.Restriction.Restriction.Hin4II-class.html#results,Variable Bio.Restriction.Restriction.Hin6I.results=Bio.Restriction.Restriction.Hin6I-class.html#results,Variable Bio.Restriction.Restriction.HinP1I.results=Bio.Restriction.Restriction.HinP1I-class.html#results,Variable Bio.Restriction.Restriction.HincII.results=Bio.Restriction.Restriction.HincII-class.html#results,Variable Bio.Restriction.Restriction.HindII.results=Bio.Restriction.Restriction.HindII-class.html#results,Variable Bio.Restriction.Restriction.HindIII.results=Bio.Restriction.Restriction.HindIII-class.html#results,Variable Bio.Restriction.Restriction.HinfI.results=Bio.Restriction.Restriction.HinfI-class.html#results,Variable Bio.Restriction.Restriction.HpaI.results=Bio.Restriction.Restriction.HpaI-class.html#results,Variable Bio.Restriction.Restriction.HpaII.results=Bio.Restriction.Restriction.HpaII-class.html#results,Variable Bio.Restriction.Restriction.HphI.results=Bio.Restriction.Restriction.HphI-class.html#results,Variable Bio.Restriction.Restriction.Hpy166II.results=Bio.Restriction.Restriction.Hpy166II-class.html#results,Variable Bio.Restriction.Restriction.Hpy178III.results=Bio.Restriction.Restriction.Hpy178III-class.html#results,Variable Bio.Restriction.Restriction.Hpy188I.results=Bio.Restriction.Restriction.Hpy188I-class.html#results,Variable Bio.Restriction.Restriction.Hpy188III.results=Bio.Restriction.Restriction.Hpy188III-class.html#results,Variable Bio.Restriction.Restriction.Hpy8I.results=Bio.Restriction.Restriction.Hpy8I-class.html#results,Variable Bio.Restriction.Restriction.Hpy99I.results=Bio.Restriction.Restriction.Hpy99I-class.html#results,Variable Bio.Restriction.Restriction.Hpy99XIII.results=Bio.Restriction.Restriction.Hpy99XIII-class.html#results,Variable Bio.Restriction.Restriction.Hpy99XIV.results=Bio.Restriction.Restriction.Hpy99XIV-class.html#results,Variable Bio.Restriction.Restriction.HpyAV.results=Bio.Restriction.Restriction.HpyAV-class.html#results,Variable Bio.Restriction.Restriction.HpyCH4III.results=Bio.Restriction.Restriction.HpyCH4III-class.html#results,Variable Bio.Restriction.Restriction.HpyCH4IV.results=Bio.Restriction.Restriction.HpyCH4IV-class.html#results,Variable Bio.Restriction.Restriction.HpyCH4V.results=Bio.Restriction.Restriction.HpyCH4V-class.html#results,Variable Bio.Restriction.Restriction.HpyF10VI.results=Bio.Restriction.Restriction.HpyF10VI-class.html#results,Variable Bio.Restriction.Restriction.HpyF3I.results=Bio.Restriction.Restriction.HpyF3I-class.html#results,Variable Bio.Restriction.Restriction.HpySE526I.results=Bio.Restriction.Restriction.HpySE526I-class.html#results,Variable Bio.Restriction.Restriction.Hsp92I.results=Bio.Restriction.Restriction.Hsp92I-class.html#results,Variable Bio.Restriction.Restriction.Hsp92II.results=Bio.Restriction.Restriction.Hsp92II-class.html#results,Variable Bio.Restriction.Restriction.HspAI.results=Bio.Restriction.Restriction.HspAI-class.html#results,Variable Bio.Restriction.Restriction.Jma19592I.results=Bio.Restriction.Restriction.Jma19592I-class.html#results,Variable Bio.Restriction.Restriction.KasI.results=Bio.Restriction.Restriction.KasI-class.html#results,Variable Bio.Restriction.Restriction.KflI.results=Bio.Restriction.Restriction.KflI-class.html#results,Variable Bio.Restriction.Restriction.Kpn2I.results=Bio.Restriction.Restriction.Kpn2I-class.html#results,Variable Bio.Restriction.Restriction.KpnI.results=Bio.Restriction.Restriction.KpnI-class.html#results,Variable Bio.Restriction.Restriction.KroI.results=Bio.Restriction.Restriction.KroI-class.html#results,Variable Bio.Restriction.Restriction.Ksp22I.results=Bio.Restriction.Restriction.Ksp22I-class.html#results,Variable Bio.Restriction.Restriction.Ksp632I.results=Bio.Restriction.Restriction.Ksp632I-class.html#results,Variable Bio.Restriction.Restriction.KspAI.results=Bio.Restriction.Restriction.KspAI-class.html#results,Variable Bio.Restriction.Restriction.KspI.results=Bio.Restriction.Restriction.KspI-class.html#results,Variable Bio.Restriction.Restriction.Kzo9I.results=Bio.Restriction.Restriction.Kzo9I-class.html#results,Variable Bio.Restriction.Restriction.LguI.results=Bio.Restriction.Restriction.LguI-class.html#results,Variable Bio.Restriction.Restriction.LpnI.results=Bio.Restriction.Restriction.LpnI-class.html#results,Variable Bio.Restriction.Restriction.LpnPI.results=Bio.Restriction.Restriction.LpnPI-class.html#results,Variable Bio.Restriction.Restriction.Lsp1109I.results=Bio.Restriction.Restriction.Lsp1109I-class.html#results,Variable Bio.Restriction.Restriction.LweI.results=Bio.Restriction.Restriction.LweI-class.html#results,Variable Bio.Restriction.Restriction.MabI.results=Bio.Restriction.Restriction.MabI-class.html#results,Variable Bio.Restriction.Restriction.MaeI.results=Bio.Restriction.Restriction.MaeI-class.html#results,Variable Bio.Restriction.Restriction.MaeII.results=Bio.Restriction.Restriction.MaeII-class.html#results,Variable Bio.Restriction.Restriction.MaeIII.results=Bio.Restriction.Restriction.MaeIII-class.html#results,Variable Bio.Restriction.Restriction.MalI.results=Bio.Restriction.Restriction.MalI-class.html#results,Variable Bio.Restriction.Restriction.MaqI.results=Bio.Restriction.Restriction.MaqI-class.html#results,Variable Bio.Restriction.Restriction.MauBI.results=Bio.Restriction.Restriction.MauBI-class.html#results,Variable Bio.Restriction.Restriction.MbiI.results=Bio.Restriction.Restriction.MbiI-class.html#results,Variable Bio.Restriction.Restriction.MboI.results=Bio.Restriction.Restriction.MboI-class.html#results,Variable Bio.Restriction.Restriction.MboII.results=Bio.Restriction.Restriction.MboII-class.html#results,Variable Bio.Restriction.Restriction.McaTI.results=Bio.Restriction.Restriction.McaTI-class.html#results,Variable Bio.Restriction.Restriction.McrI.results=Bio.Restriction.Restriction.McrI-class.html#results,Variable Bio.Restriction.Restriction.MfeI.results=Bio.Restriction.Restriction.MfeI-class.html#results,Variable Bio.Restriction.Restriction.MflI.results=Bio.Restriction.Restriction.MflI-class.html#results,Variable Bio.Restriction.Restriction.MhlI.results=Bio.Restriction.Restriction.MhlI-class.html#results,Variable Bio.Restriction.Restriction.MjaIV.results=Bio.Restriction.Restriction.MjaIV-class.html#results,Variable Bio.Restriction.Restriction.MkaDII.results=Bio.Restriction.Restriction.MkaDII-class.html#results,Variable Bio.Restriction.Restriction.MlsI.results=Bio.Restriction.Restriction.MlsI-class.html#results,Variable Bio.Restriction.Restriction.MluCI.results=Bio.Restriction.Restriction.MluCI-class.html#results,Variable Bio.Restriction.Restriction.MluI.results=Bio.Restriction.Restriction.MluI-class.html#results,Variable Bio.Restriction.Restriction.MluNI.results=Bio.Restriction.Restriction.MluNI-class.html#results,Variable Bio.Restriction.Restriction.Mly113I.results=Bio.Restriction.Restriction.Mly113I-class.html#results,Variable Bio.Restriction.Restriction.MlyI.results=Bio.Restriction.Restriction.MlyI-class.html#results,Variable Bio.Restriction.Restriction.MmeI.results=Bio.Restriction.Restriction.MmeI-class.html#results,Variable Bio.Restriction.Restriction.MnlI.results=Bio.Restriction.Restriction.MnlI-class.html#results,Variable Bio.Restriction.Restriction.Mph1103I.results=Bio.Restriction.Restriction.Mph1103I-class.html#results,Variable Bio.Restriction.Restriction.MreI.results=Bio.Restriction.Restriction.MreI-class.html#results,Variable Bio.Restriction.Restriction.MroI.results=Bio.Restriction.Restriction.MroI-class.html#results,Variable Bio.Restriction.Restriction.MroNI.results=Bio.Restriction.Restriction.MroNI-class.html#results,Variable Bio.Restriction.Restriction.MroXI.results=Bio.Restriction.Restriction.MroXI-class.html#results,Variable Bio.Restriction.Restriction.MscI.results=Bio.Restriction.Restriction.MscI-class.html#results,Variable Bio.Restriction.Restriction.MseI.results=Bio.Restriction.Restriction.MseI-class.html#results,Variable Bio.Restriction.Restriction.MslI.results=Bio.Restriction.Restriction.MslI-class.html#results,Variable Bio.Restriction.Restriction.Msp20I.results=Bio.Restriction.Restriction.Msp20I-class.html#results,Variable Bio.Restriction.Restriction.MspA1I.results=Bio.Restriction.Restriction.MspA1I-class.html#results,Variable Bio.Restriction.Restriction.MspCI.results=Bio.Restriction.Restriction.MspCI-class.html#results,Variable Bio.Restriction.Restriction.MspI.results=Bio.Restriction.Restriction.MspI-class.html#results,Variable Bio.Restriction.Restriction.MspJI.results=Bio.Restriction.Restriction.MspJI-class.html#results,Variable Bio.Restriction.Restriction.MspR9I.results=Bio.Restriction.Restriction.MspR9I-class.html#results,Variable Bio.Restriction.Restriction.MssI.results=Bio.Restriction.Restriction.MssI-class.html#results,Variable Bio.Restriction.Restriction.MstI.results=Bio.Restriction.Restriction.MstI-class.html#results,Variable Bio.Restriction.Restriction.MunI.results=Bio.Restriction.Restriction.MunI-class.html#results,Variable Bio.Restriction.Restriction.Mva1269I.results=Bio.Restriction.Restriction.Mva1269I-class.html#results,Variable Bio.Restriction.Restriction.MvaI.results=Bio.Restriction.Restriction.MvaI-class.html#results,Variable Bio.Restriction.Restriction.MvnI.results=Bio.Restriction.Restriction.MvnI-class.html#results,Variable Bio.Restriction.Restriction.MvrI.results=Bio.Restriction.Restriction.MvrI-class.html#results,Variable Bio.Restriction.Restriction.MwoI.results=Bio.Restriction.Restriction.MwoI-class.html#results,Variable Bio.Restriction.Restriction.NaeI.results=Bio.Restriction.Restriction.NaeI-class.html#results,Variable Bio.Restriction.Restriction.NarI.results=Bio.Restriction.Restriction.NarI-class.html#results,Variable Bio.Restriction.Restriction.NciI.results=Bio.Restriction.Restriction.NciI-class.html#results,Variable Bio.Restriction.Restriction.NcoI.results=Bio.Restriction.Restriction.NcoI-class.html#results,Variable Bio.Restriction.Restriction.NdeI.results=Bio.Restriction.Restriction.NdeI-class.html#results,Variable Bio.Restriction.Restriction.NdeII.results=Bio.Restriction.Restriction.NdeII-class.html#results,Variable Bio.Restriction.Restriction.NgoAVIII.results=Bio.Restriction.Restriction.NgoAVIII-class.html#results,Variable Bio.Restriction.Restriction.NgoMIV.results=Bio.Restriction.Restriction.NgoMIV-class.html#results,Variable Bio.Restriction.Restriction.NhaXI.results=Bio.Restriction.Restriction.NhaXI-class.html#results,Variable Bio.Restriction.Restriction.NheI.results=Bio.Restriction.Restriction.NheI-class.html#results,Variable Bio.Restriction.Restriction.NlaCI.results=Bio.Restriction.Restriction.NlaCI-class.html#results,Variable Bio.Restriction.Restriction.NlaIII.results=Bio.Restriction.Restriction.NlaIII-class.html#results,Variable Bio.Restriction.Restriction.NlaIV.results=Bio.Restriction.Restriction.NlaIV-class.html#results,Variable Bio.Restriction.Restriction.Nli3877I.results=Bio.Restriction.Restriction.Nli3877I-class.html#results,Variable Bio.Restriction.Restriction.NmeAIII.results=Bio.Restriction.Restriction.NmeAIII-class.html#results,Variable Bio.Restriction.Restriction.NmeDI.results=Bio.Restriction.Restriction.NmeDI-class.html#results,Variable Bio.Restriction.Restriction.NmuCI.results=Bio.Restriction.Restriction.NmuCI-class.html#results,Variable Bio.Restriction.Restriction.NotI.results=Bio.Restriction.Restriction.NotI-class.html#results,Variable Bio.Restriction.Restriction.NruI.results=Bio.Restriction.Restriction.NruI-class.html#results,Variable Bio.Restriction.Restriction.NsbI.results=Bio.Restriction.Restriction.NsbI-class.html#results,Variable Bio.Restriction.Restriction.NsiI.results=Bio.Restriction.Restriction.NsiI-class.html#results,Variable Bio.Restriction.Restriction.NspBII.results=Bio.Restriction.Restriction.NspBII-class.html#results,Variable Bio.Restriction.Restriction.NspI.results=Bio.Restriction.Restriction.NspI-class.html#results,Variable Bio.Restriction.Restriction.NspV.results=Bio.Restriction.Restriction.NspV-class.html#results,Variable Bio.Restriction.Restriction.OliI.results=Bio.Restriction.Restriction.OliI-class.html#results,Variable Bio.Restriction.Restriction.PabI.results=Bio.Restriction.Restriction.PabI-class.html#results,Variable Bio.Restriction.Restriction.PacI.results=Bio.Restriction.Restriction.PacI-class.html#results,Variable Bio.Restriction.Restriction.PaeI.results=Bio.Restriction.Restriction.PaeI-class.html#results,Variable Bio.Restriction.Restriction.PaeR7I.results=Bio.Restriction.Restriction.PaeR7I-class.html#results,Variable Bio.Restriction.Restriction.PagI.results=Bio.Restriction.Restriction.PagI-class.html#results,Variable Bio.Restriction.Restriction.PalAI.results=Bio.Restriction.Restriction.PalAI-class.html#results,Variable Bio.Restriction.Restriction.PasI.results=Bio.Restriction.Restriction.PasI-class.html#results,Variable Bio.Restriction.Restriction.PauI.results=Bio.Restriction.Restriction.PauI-class.html#results,Variable Bio.Restriction.Restriction.PceI.results=Bio.Restriction.Restriction.PceI-class.html#results,Variable Bio.Restriction.Restriction.PciI.results=Bio.Restriction.Restriction.PciI-class.html#results,Variable Bio.Restriction.Restriction.PciSI.results=Bio.Restriction.Restriction.PciSI-class.html#results,Variable Bio.Restriction.Restriction.PcsI.results=Bio.Restriction.Restriction.PcsI-class.html#results,Variable Bio.Restriction.Restriction.PctI.results=Bio.Restriction.Restriction.PctI-class.html#results,Variable Bio.Restriction.Restriction.PdiI.results=Bio.Restriction.Restriction.PdiI-class.html#results,Variable Bio.Restriction.Restriction.PdmI.results=Bio.Restriction.Restriction.PdmI-class.html#results,Variable Bio.Restriction.Restriction.PenI.results=Bio.Restriction.Restriction.PenI-class.html#results,Variable Bio.Restriction.Restriction.PfeI.results=Bio.Restriction.Restriction.PfeI-class.html#results,Variable Bio.Restriction.Restriction.Pfl1108I.results=Bio.Restriction.Restriction.Pfl1108I-class.html#results,Variable Bio.Restriction.Restriction.Pfl23II.results=Bio.Restriction.Restriction.Pfl23II-class.html#results,Variable Bio.Restriction.Restriction.PflFI.results=Bio.Restriction.Restriction.PflFI-class.html#results,Variable Bio.Restriction.Restriction.PflMI.results=Bio.Restriction.Restriction.PflMI-class.html#results,Variable Bio.Restriction.Restriction.PfoI.results=Bio.Restriction.Restriction.PfoI-class.html#results,Variable Bio.Restriction.Restriction.PinAI.results=Bio.Restriction.Restriction.PinAI-class.html#results,Variable Bio.Restriction.Restriction.PlaDI.results=Bio.Restriction.Restriction.PlaDI-class.html#results,Variable Bio.Restriction.Restriction.Ple19I.results=Bio.Restriction.Restriction.Ple19I-class.html#results,Variable Bio.Restriction.Restriction.PleI.results=Bio.Restriction.Restriction.PleI-class.html#results,Variable Bio.Restriction.Restriction.PluTI.results=Bio.Restriction.Restriction.PluTI-class.html#results,Variable Bio.Restriction.Restriction.PmaCI.results=Bio.Restriction.Restriction.PmaCI-class.html#results,Variable Bio.Restriction.Restriction.PmeI.results=Bio.Restriction.Restriction.PmeI-class.html#results,Variable Bio.Restriction.Restriction.PmlI.results=Bio.Restriction.Restriction.PmlI-class.html#results,Variable Bio.Restriction.Restriction.PpiI.results=Bio.Restriction.Restriction.PpiI-class.html#results,Variable Bio.Restriction.Restriction.PpsI.results=Bio.Restriction.Restriction.PpsI-class.html#results,Variable Bio.Restriction.Restriction.Ppu10I.results=Bio.Restriction.Restriction.Ppu10I-class.html#results,Variable Bio.Restriction.Restriction.Ppu21I.results=Bio.Restriction.Restriction.Ppu21I-class.html#results,Variable Bio.Restriction.Restriction.PpuMI.results=Bio.Restriction.Restriction.PpuMI-class.html#results,Variable Bio.Restriction.Restriction.PscI.results=Bio.Restriction.Restriction.PscI-class.html#results,Variable Bio.Restriction.Restriction.PshAI.results=Bio.Restriction.Restriction.PshAI-class.html#results,Variable Bio.Restriction.Restriction.PshBI.results=Bio.Restriction.Restriction.PshBI-class.html#results,Variable Bio.Restriction.Restriction.PsiI.results=Bio.Restriction.Restriction.PsiI-class.html#results,Variable Bio.Restriction.Restriction.Psp03I.results=Bio.Restriction.Restriction.Psp03I-class.html#results,Variable Bio.Restriction.Restriction.Psp124BI.results=Bio.Restriction.Restriction.Psp124BI-class.html#results,Variable Bio.Restriction.Restriction.Psp1406I.results=Bio.Restriction.Restriction.Psp1406I-class.html#results,Variable Bio.Restriction.Restriction.Psp5II.results=Bio.Restriction.Restriction.Psp5II-class.html#results,Variable Bio.Restriction.Restriction.Psp6I.results=Bio.Restriction.Restriction.Psp6I-class.html#results,Variable Bio.Restriction.Restriction.PspCI.results=Bio.Restriction.Restriction.PspCI-class.html#results,Variable Bio.Restriction.Restriction.PspEI.results=Bio.Restriction.Restriction.PspEI-class.html#results,Variable Bio.Restriction.Restriction.PspGI.results=Bio.Restriction.Restriction.PspGI-class.html#results,Variable Bio.Restriction.Restriction.PspLI.results=Bio.Restriction.Restriction.PspLI-class.html#results,Variable Bio.Restriction.Restriction.PspN4I.results=Bio.Restriction.Restriction.PspN4I-class.html#results,Variable Bio.Restriction.Restriction.PspOMI.results=Bio.Restriction.Restriction.PspOMI-class.html#results,Variable Bio.Restriction.Restriction.PspOMII.results=Bio.Restriction.Restriction.PspOMII-class.html#results,Variable Bio.Restriction.Restriction.PspPI.results=Bio.Restriction.Restriction.PspPI-class.html#results,Variable Bio.Restriction.Restriction.PspPPI.results=Bio.Restriction.Restriction.PspPPI-class.html#results,Variable Bio.Restriction.Restriction.PspPRI.results=Bio.Restriction.Restriction.PspPRI-class.html#results,Variable Bio.Restriction.Restriction.PspXI.results=Bio.Restriction.Restriction.PspXI-class.html#results,Variable Bio.Restriction.Restriction.PsrI.results=Bio.Restriction.Restriction.PsrI-class.html#results,Variable Bio.Restriction.Restriction.PssI.results=Bio.Restriction.Restriction.PssI-class.html#results,Variable Bio.Restriction.Restriction.PstI.results=Bio.Restriction.Restriction.PstI-class.html#results,Variable Bio.Restriction.Restriction.PstNI.results=Bio.Restriction.Restriction.PstNI-class.html#results,Variable Bio.Restriction.Restriction.PsuI.results=Bio.Restriction.Restriction.PsuI-class.html#results,Variable Bio.Restriction.Restriction.PsyI.results=Bio.Restriction.Restriction.PsyI-class.html#results,Variable Bio.Restriction.Restriction.PteI.results=Bio.Restriction.Restriction.PteI-class.html#results,Variable Bio.Restriction.Restriction.PvuI.results=Bio.Restriction.Restriction.PvuI-class.html#results,Variable Bio.Restriction.Restriction.PvuII.results=Bio.Restriction.Restriction.PvuII-class.html#results,Variable Bio.Restriction.Restriction.R2_BceSIV.results=Bio.Restriction.Restriction.R2_BceSIV-class.html#results,Variable Bio.Restriction.Restriction.RceI.results=Bio.Restriction.Restriction.RceI-class.html#results,Variable Bio.Restriction.Restriction.RdeGBI.results=Bio.Restriction.Restriction.RdeGBI-class.html#results,Variable Bio.Restriction.Restriction.RdeGBII.results=Bio.Restriction.Restriction.RdeGBII-class.html#results,Variable Bio.Restriction.Restriction.RdeGBIII.results=Bio.Restriction.Restriction.RdeGBIII-class.html#results,Variable Bio.Restriction.Restriction.RflFIII.results=Bio.Restriction.Restriction.RflFIII-class.html#results,Variable Bio.Restriction.Restriction.RgaI.results=Bio.Restriction.Restriction.RgaI-class.html#results,Variable Bio.Restriction.Restriction.RigI.results=Bio.Restriction.Restriction.RigI-class.html#results,Variable Bio.Restriction.Restriction.RlaI.results=Bio.Restriction.Restriction.RlaI-class.html#results,Variable Bio.Restriction.Restriction.RleAI.results=Bio.Restriction.Restriction.RleAI-class.html#results,Variable Bio.Restriction.Restriction.RpaB5I.results=Bio.Restriction.Restriction.RpaB5I-class.html#results,Variable Bio.Restriction.Restriction.RpaBI.results=Bio.Restriction.Restriction.RpaBI-class.html#results,Variable Bio.Restriction.Restriction.RpaI.results=Bio.Restriction.Restriction.RpaI-class.html#results,Variable Bio.Restriction.Restriction.RpaTI.results=Bio.Restriction.Restriction.RpaTI-class.html#results,Variable Bio.Restriction.Restriction.RruI.results=Bio.Restriction.Restriction.RruI-class.html#results,Variable Bio.Restriction.Restriction.RsaI.results=Bio.Restriction.Restriction.RsaI-class.html#results,Variable Bio.Restriction.Restriction.RsaNI.results=Bio.Restriction.Restriction.RsaNI-class.html#results,Variable Bio.Restriction.Restriction.RseI.results=Bio.Restriction.Restriction.RseI-class.html#results,Variable Bio.Restriction.Restriction.Rsr2I.results=Bio.Restriction.Restriction.Rsr2I-class.html#results,Variable Bio.Restriction.Restriction.RsrII.results=Bio.Restriction.Restriction.RsrII-class.html#results,Variable Bio.Restriction.Restriction.SacI.results=Bio.Restriction.Restriction.SacI-class.html#results,Variable Bio.Restriction.Restriction.SacII.results=Bio.Restriction.Restriction.SacII-class.html#results,Variable Bio.Restriction.Restriction.SalI.results=Bio.Restriction.Restriction.SalI-class.html#results,Variable Bio.Restriction.Restriction.SanDI.results=Bio.Restriction.Restriction.SanDI-class.html#results,Variable Bio.Restriction.Restriction.SapI.results=Bio.Restriction.Restriction.SapI-class.html#results,Variable Bio.Restriction.Restriction.SaqAI.results=Bio.Restriction.Restriction.SaqAI-class.html#results,Variable Bio.Restriction.Restriction.SatI.results=Bio.Restriction.Restriction.SatI-class.html#results,Variable Bio.Restriction.Restriction.Sau3AI.results=Bio.Restriction.Restriction.Sau3AI-class.html#results,Variable Bio.Restriction.Restriction.Sau96I.results=Bio.Restriction.Restriction.Sau96I-class.html#results,Variable Bio.Restriction.Restriction.SauI.results=Bio.Restriction.Restriction.SauI-class.html#results,Variable Bio.Restriction.Restriction.SbfI.results=Bio.Restriction.Restriction.SbfI-class.html#results,Variable Bio.Restriction.Restriction.ScaI.results=Bio.Restriction.Restriction.ScaI-class.html#results,Variable Bio.Restriction.Restriction.SchI.results=Bio.Restriction.Restriction.SchI-class.html#results,Variable Bio.Restriction.Restriction.SciI.results=Bio.Restriction.Restriction.SciI-class.html#results,Variable Bio.Restriction.Restriction.ScrFI.results=Bio.Restriction.Restriction.ScrFI-class.html#results,Variable Bio.Restriction.Restriction.SdaI.results=Bio.Restriction.Restriction.SdaI-class.html#results,Variable Bio.Restriction.Restriction.SdeAI.results=Bio.Restriction.Restriction.SdeAI-class.html#results,Variable Bio.Restriction.Restriction.SdeOSI.results=Bio.Restriction.Restriction.SdeOSI-class.html#results,Variable Bio.Restriction.Restriction.SduI.results=Bio.Restriction.Restriction.SduI-class.html#results,Variable Bio.Restriction.Restriction.SecI.results=Bio.Restriction.Restriction.SecI-class.html#results,Variable Bio.Restriction.Restriction.SelI.results=Bio.Restriction.Restriction.SelI-class.html#results,Variable Bio.Restriction.Restriction.SetI.results=Bio.Restriction.Restriction.SetI-class.html#results,Variable Bio.Restriction.Restriction.SexAI.results=Bio.Restriction.Restriction.SexAI-class.html#results,Variable Bio.Restriction.Restriction.SfaAI.results=Bio.Restriction.Restriction.SfaAI-class.html#results,Variable Bio.Restriction.Restriction.SfaNI.results=Bio.Restriction.Restriction.SfaNI-class.html#results,Variable Bio.Restriction.Restriction.SfcI.results=Bio.Restriction.Restriction.SfcI-class.html#results,Variable Bio.Restriction.Restriction.SfeI.results=Bio.Restriction.Restriction.SfeI-class.html#results,Variable Bio.Restriction.Restriction.SfiI.results=Bio.Restriction.Restriction.SfiI-class.html#results,Variable Bio.Restriction.Restriction.SfoI.results=Bio.Restriction.Restriction.SfoI-class.html#results,Variable Bio.Restriction.Restriction.Sfr274I.results=Bio.Restriction.Restriction.Sfr274I-class.html#results,Variable Bio.Restriction.Restriction.Sfr303I.results=Bio.Restriction.Restriction.Sfr303I-class.html#results,Variable Bio.Restriction.Restriction.SfuI.results=Bio.Restriction.Restriction.SfuI-class.html#results,Variable Bio.Restriction.Restriction.SgeI.results=Bio.Restriction.Restriction.SgeI-class.html#results,Variable Bio.Restriction.Restriction.SgfI.results=Bio.Restriction.Restriction.SgfI-class.html#results,Variable Bio.Restriction.Restriction.SgrAI.results=Bio.Restriction.Restriction.SgrAI-class.html#results,Variable Bio.Restriction.Restriction.SgrBI.results=Bio.Restriction.Restriction.SgrBI-class.html#results,Variable Bio.Restriction.Restriction.SgrDI.results=Bio.Restriction.Restriction.SgrDI-class.html#results,Variable Bio.Restriction.Restriction.SgrTI.results=Bio.Restriction.Restriction.SgrTI-class.html#results,Variable Bio.Restriction.Restriction.SgsI.results=Bio.Restriction.Restriction.SgsI-class.html#results,Variable Bio.Restriction.Restriction.SimI.results=Bio.Restriction.Restriction.SimI-class.html#results,Variable Bio.Restriction.Restriction.SlaI.results=Bio.Restriction.Restriction.SlaI-class.html#results,Variable Bio.Restriction.Restriction.SmaI.results=Bio.Restriction.Restriction.SmaI-class.html#results,Variable Bio.Restriction.Restriction.SmiI.results=Bio.Restriction.Restriction.SmiI-class.html#results,Variable Bio.Restriction.Restriction.SmiMI.results=Bio.Restriction.Restriction.SmiMI-class.html#results,Variable Bio.Restriction.Restriction.SmlI.results=Bio.Restriction.Restriction.SmlI-class.html#results,Variable Bio.Restriction.Restriction.SmoI.results=Bio.Restriction.Restriction.SmoI-class.html#results,Variable Bio.Restriction.Restriction.SnaBI.results=Bio.Restriction.Restriction.SnaBI-class.html#results,Variable Bio.Restriction.Restriction.SnaI.results=Bio.Restriction.Restriction.SnaI-class.html#results,Variable Bio.Restriction.Restriction.Sno506I.results=Bio.Restriction.Restriction.Sno506I-class.html#results,Variable Bio.Restriction.Restriction.SpeI.results=Bio.Restriction.Restriction.SpeI-class.html#results,Variable Bio.Restriction.Restriction.SphI.results=Bio.Restriction.Restriction.SphI-class.html#results,Variable Bio.Restriction.Restriction.SplI.results=Bio.Restriction.Restriction.SplI-class.html#results,Variable Bio.Restriction.Restriction.SpoDI.results=Bio.Restriction.Restriction.SpoDI-class.html#results,Variable Bio.Restriction.Restriction.SrfI.results=Bio.Restriction.Restriction.SrfI-class.html#results,Variable Bio.Restriction.Restriction.Sse232I.results=Bio.Restriction.Restriction.Sse232I-class.html#results,Variable Bio.Restriction.Restriction.Sse8387I.results=Bio.Restriction.Restriction.Sse8387I-class.html#results,Variable Bio.Restriction.Restriction.Sse8647I.results=Bio.Restriction.Restriction.Sse8647I-class.html#results,Variable Bio.Restriction.Restriction.Sse9I.results=Bio.Restriction.Restriction.Sse9I-class.html#results,Variable Bio.Restriction.Restriction.SseBI.results=Bio.Restriction.Restriction.SseBI-class.html#results,Variable Bio.Restriction.Restriction.SsiI.results=Bio.Restriction.Restriction.SsiI-class.html#results,Variable Bio.Restriction.Restriction.SspD5I.results=Bio.Restriction.Restriction.SspD5I-class.html#results,Variable Bio.Restriction.Restriction.SspDI.results=Bio.Restriction.Restriction.SspDI-class.html#results,Variable Bio.Restriction.Restriction.SspI.results=Bio.Restriction.Restriction.SspI-class.html#results,Variable Bio.Restriction.Restriction.SstE37I.results=Bio.Restriction.Restriction.SstE37I-class.html#results,Variable Bio.Restriction.Restriction.SstI.results=Bio.Restriction.Restriction.SstI-class.html#results,Variable Bio.Restriction.Restriction.Sth132I.results=Bio.Restriction.Restriction.Sth132I-class.html#results,Variable Bio.Restriction.Restriction.Sth302II.results=Bio.Restriction.Restriction.Sth302II-class.html#results,Variable Bio.Restriction.Restriction.StrI.results=Bio.Restriction.Restriction.StrI-class.html#results,Variable Bio.Restriction.Restriction.StsI.results=Bio.Restriction.Restriction.StsI-class.html#results,Variable Bio.Restriction.Restriction.StuI.results=Bio.Restriction.Restriction.StuI-class.html#results,Variable Bio.Restriction.Restriction.StyD4I.results=Bio.Restriction.Restriction.StyD4I-class.html#results,Variable Bio.Restriction.Restriction.StyI.results=Bio.Restriction.Restriction.StyI-class.html#results,Variable Bio.Restriction.Restriction.SwaI.results=Bio.Restriction.Restriction.SwaI-class.html#results,Variable Bio.Restriction.Restriction.TaaI.results=Bio.Restriction.Restriction.TaaI-class.html#results,Variable Bio.Restriction.Restriction.TaiI.results=Bio.Restriction.Restriction.TaiI-class.html#results,Variable Bio.Restriction.Restriction.TaqI.results=Bio.Restriction.Restriction.TaqI-class.html#results,Variable Bio.Restriction.Restriction.TaqII.results=Bio.Restriction.Restriction.TaqII-class.html#results,Variable Bio.Restriction.Restriction.TasI.results=Bio.Restriction.Restriction.TasI-class.html#results,Variable Bio.Restriction.Restriction.TatI.results=Bio.Restriction.Restriction.TatI-class.html#results,Variable Bio.Restriction.Restriction.TauI.results=Bio.Restriction.Restriction.TauI-class.html#results,Variable Bio.Restriction.Restriction.TfiI.results=Bio.Restriction.Restriction.TfiI-class.html#results,Variable Bio.Restriction.Restriction.Tru1I.results=Bio.Restriction.Restriction.Tru1I-class.html#results,Variable Bio.Restriction.Restriction.Tru9I.results=Bio.Restriction.Restriction.Tru9I-class.html#results,Variable Bio.Restriction.Restriction.TscAI.results=Bio.Restriction.Restriction.TscAI-class.html#results,Variable Bio.Restriction.Restriction.TseFI.results=Bio.Restriction.Restriction.TseFI-class.html#results,Variable Bio.Restriction.Restriction.TseI.results=Bio.Restriction.Restriction.TseI-class.html#results,Variable Bio.Restriction.Restriction.TsoI.results=Bio.Restriction.Restriction.TsoI-class.html#results,Variable Bio.Restriction.Restriction.Tsp45I.results=Bio.Restriction.Restriction.Tsp45I-class.html#results,Variable Bio.Restriction.Restriction.Tsp4CI.results=Bio.Restriction.Restriction.Tsp4CI-class.html#results,Variable Bio.Restriction.Restriction.TspDTI.results=Bio.Restriction.Restriction.TspDTI-class.html#results,Variable Bio.Restriction.Restriction.TspEI.results=Bio.Restriction.Restriction.TspEI-class.html#results,Variable Bio.Restriction.Restriction.TspGWI.results=Bio.Restriction.Restriction.TspGWI-class.html#results,Variable Bio.Restriction.Restriction.TspMI.results=Bio.Restriction.Restriction.TspMI-class.html#results,Variable Bio.Restriction.Restriction.TspRI.results=Bio.Restriction.Restriction.TspRI-class.html#results,Variable Bio.Restriction.Restriction.TssI.results=Bio.Restriction.Restriction.TssI-class.html#results,Variable Bio.Restriction.Restriction.TstI.results=Bio.Restriction.Restriction.TstI-class.html#results,Variable Bio.Restriction.Restriction.TsuI.results=Bio.Restriction.Restriction.TsuI-class.html#results,Variable Bio.Restriction.Restriction.Tth111I.results=Bio.Restriction.Restriction.Tth111I-class.html#results,Variable Bio.Restriction.Restriction.Tth111II.results=Bio.Restriction.Restriction.Tth111II-class.html#results,Variable Bio.Restriction.Restriction.UbaF11I.results=Bio.Restriction.Restriction.UbaF11I-class.html#results,Variable Bio.Restriction.Restriction.UbaF12I.results=Bio.Restriction.Restriction.UbaF12I-class.html#results,Variable Bio.Restriction.Restriction.UbaF13I.results=Bio.Restriction.Restriction.UbaF13I-class.html#results,Variable Bio.Restriction.Restriction.UbaF14I.results=Bio.Restriction.Restriction.UbaF14I-class.html#results,Variable Bio.Restriction.Restriction.UbaF9I.results=Bio.Restriction.Restriction.UbaF9I-class.html#results,Variable Bio.Restriction.Restriction.UbaPI.results=Bio.Restriction.Restriction.UbaPI-class.html#results,Variable Bio.Restriction.Restriction.UcoMSI.results=Bio.Restriction.Restriction.UcoMSI-class.html#results,Variable Bio.Restriction.Restriction.UnbI.results=Bio.Restriction.Restriction.UnbI-class.html#results,Variable Bio.Restriction.Restriction.Van91I.results=Bio.Restriction.Restriction.Van91I-class.html#results,Variable Bio.Restriction.Restriction.Vha464I.results=Bio.Restriction.Restriction.Vha464I-class.html#results,Variable Bio.Restriction.Restriction.VneI.results=Bio.Restriction.Restriction.VneI-class.html#results,Variable Bio.Restriction.Restriction.VpaK11AI.results=Bio.Restriction.Restriction.VpaK11AI-class.html#results,Variable Bio.Restriction.Restriction.VpaK11BI.results=Bio.Restriction.Restriction.VpaK11BI-class.html#results,Variable Bio.Restriction.Restriction.VspI.results=Bio.Restriction.Restriction.VspI-class.html#results,Variable Bio.Restriction.Restriction.WviI.results=Bio.Restriction.Restriction.WviI-class.html#results,Variable Bio.Restriction.Restriction.XagI.results=Bio.Restriction.Restriction.XagI-class.html#results,Variable Bio.Restriction.Restriction.XapI.results=Bio.Restriction.Restriction.XapI-class.html#results,Variable Bio.Restriction.Restriction.XbaI.results=Bio.Restriction.Restriction.XbaI-class.html#results,Variable Bio.Restriction.Restriction.XceI.results=Bio.Restriction.Restriction.XceI-class.html#results,Variable Bio.Restriction.Restriction.XcmI.results=Bio.Restriction.Restriction.XcmI-class.html#results,Variable Bio.Restriction.Restriction.XhoI.results=Bio.Restriction.Restriction.XhoI-class.html#results,Variable Bio.Restriction.Restriction.XhoII.results=Bio.Restriction.Restriction.XhoII-class.html#results,Variable Bio.Restriction.Restriction.XmaI.results=Bio.Restriction.Restriction.XmaI-class.html#results,Variable Bio.Restriction.Restriction.XmaIII.results=Bio.Restriction.Restriction.XmaIII-class.html#results,Variable Bio.Restriction.Restriction.XmaJI.results=Bio.Restriction.Restriction.XmaJI-class.html#results,Variable Bio.Restriction.Restriction.XmiI.results=Bio.Restriction.Restriction.XmiI-class.html#results,Variable Bio.Restriction.Restriction.XmnI.results=Bio.Restriction.Restriction.XmnI-class.html#results,Variable Bio.Restriction.Restriction.XspI.results=Bio.Restriction.Restriction.XspI-class.html#results,Variable Bio.Restriction.Restriction.YkrI.results=Bio.Restriction.Restriction.YkrI-class.html#results,Variable Bio.Restriction.Restriction.ZraI.results=Bio.Restriction.Restriction.ZraI-class.html#results,Variable Bio.Restriction.Restriction.ZrmI.results=Bio.Restriction.Restriction.ZrmI-class.html#results,Variable Bio.Restriction.Restriction.Zsp2I.results=Bio.Restriction.Restriction.Zsp2I-class.html#results"><a title="Bio.Restriction.Restriction.AanI.results
Bio.Restriction.Restriction.AarI.results
Bio.Restriction.Restriction.AasI.results
Bio.Restriction.Restriction.AatII.results
Bio.Restriction.Restriction.AbaSI.results
Bio.Restriction.Restriction.AbsI.results
Bio.Restriction.Restriction.Acc16I.results
Bio.Restriction.Restriction.Acc36I.results
Bio.Restriction.Restriction.Acc65I.results
Bio.Restriction.Restriction.AccB1I.results
Bio.Restriction.Restriction.AccB7I.results
Bio.Restriction.Restriction.AccBSI.results
Bio.Restriction.Restriction.AccI.results
Bio.Restriction.Restriction.AccII.results
Bio.Restriction.Restriction.AccIII.results
Bio.Restriction.Restriction.AceIII.results
Bio.Restriction.Restriction.AciI.results
Bio.Restriction.Restriction.AclI.results
Bio.Restriction.Restriction.AclWI.results
Bio.Restriction.Restriction.AcoI.results
Bio.Restriction.Restriction.AcsI.results
Bio.Restriction.Restriction.AcuI.results
Bio.Restriction.Restriction.AcvI.results
Bio.Restriction.Restriction.AcyI.results
Bio.Restriction.Restriction.AdeI.results
Bio.Restriction.Restriction.AfaI.results
Bio.Restriction.Restriction.AfeI.results
Bio.Restriction.Restriction.AfiI.results
Bio.Restriction.Restriction.AflII.results
Bio.Restriction.Restriction.AflIII.results
Bio.Restriction.Restriction.AgeI.results
Bio.Restriction.Restriction.AgsI.results
Bio.Restriction.Restriction.AhaIII.results
Bio.Restriction.Restriction.AhdI.results
Bio.Restriction.Restriction.AhlI.results
Bio.Restriction.Restriction.AjiI.results
Bio.Restriction.Restriction.AjnI.results
Bio.Restriction.Restriction.AjuI.results
Bio.Restriction.Restriction.AleI.results
Bio.Restriction.Restriction.AlfI.results
Bio.Restriction.Restriction.AloI.results
Bio.Restriction.Restriction.AluBI.results
Bio.Restriction.Restriction.AluI.results
Bio.Restriction.Restriction.Alw21I.results
Bio.Restriction.Restriction.Alw26I.results
Bio.Restriction.Restriction.Alw44I.results
Bio.Restriction.Restriction.AlwFI.results
Bio.Restriction.Restriction.AlwI.results
Bio.Restriction.Restriction.AlwNI.results
Bio.Restriction.Restriction.Ama87I.results
Bio.Restriction.Restriction.Aor13HI.results
Bio.Restriction.Restriction.Aor51HI.results
Bio.Restriction.Restriction.AoxI.results
Bio.Restriction.Restriction.ApaBI.results
Bio.Restriction.Restriction.ApaI.results
Bio.Restriction.Restriction.ApaLI.results
Bio.Restriction.Restriction.ApeKI.results
Bio.Restriction.Restriction.ApoI.results
Bio.Restriction.Restriction.ApyPI.results
Bio.Restriction.Restriction.AquII.results
Bio.Restriction.Restriction.AquIII.results
Bio.Restriction.Restriction.AquIV.results
Bio.Restriction.Restriction.ArsI.results
Bio.Restriction.Restriction.AscI.results
Bio.Restriction.Restriction.AseI.results
Bio.Restriction.Restriction.Asi256I.results
Bio.Restriction.Restriction.AsiGI.results
Bio.Restriction.Restriction.AsiSI.results
Bio.Restriction.Restriction.Asp700I.results
Bio.Restriction.Restriction.Asp718I.results
Bio.Restriction.Restriction.AspA2I.results
Bio.Restriction.Restriction.AspBHI.results
Bio.Restriction.Restriction.AspLEI.results
Bio.Restriction.Restriction.AspS9I.results
Bio.Restriction.Restriction.AssI.results
Bio.Restriction.Restriction.AsuC2I.results
Bio.Restriction.Restriction.AsuHPI.results
Bio.Restriction.Restriction.AsuI.results
Bio.Restriction.Restriction.AsuII.results
Bio.Restriction.Restriction.AsuNHI.results
Bio.Restriction.Restriction.AvaI.results
Bio.Restriction.Restriction.AvaII.results
Bio.Restriction.Restriction.AvaIII.results
Bio.Restriction.Restriction.AvrII.results
Bio.Restriction.Restriction.AxyI.results
Bio.Restriction.Restriction.BaeGI.results
Bio.Restriction.Restriction.BaeI.results
Bio.Restriction.Restriction.BalI.results
Bio.Restriction.Restriction.BamHI.results
Bio.Restriction.Restriction.BanI.results
Bio.Restriction.Restriction.BanII.results
Bio.Restriction.Restriction.BarI.results
Bio.Restriction.Restriction.BasI.results
Bio.Restriction.Restriction.BauI.results
Bio.Restriction.Restriction.Bbr7I.results
Bio.Restriction.Restriction.BbrPI.results
Bio.Restriction.Restriction.BbsI.results
Bio.Restriction.Restriction.Bbv12I.results
Bio.Restriction.Restriction.BbvCI.results
Bio.Restriction.Restriction.BbvI.results
Bio.Restriction.Restriction.BbvII.results
Bio.Restriction.Restriction.BccI.results
Bio.Restriction.Restriction.Bce83I.results
Bio.Restriction.Restriction.BceAI.results
Bio.Restriction.Restriction.BcefI.results
Bio.Restriction.Restriction.BcgI.results
Bio.Restriction.Restriction.BciT130I.results
Bio.Restriction.Restriction.BciVI.results
Bio.Restriction.Restriction.BclI.results
Bio.Restriction.Restriction.BcnI.results
Bio.Restriction.Restriction.BcoDI.results
Bio.Restriction.Restriction.BcuI.results
Bio.Restriction.Restriction.BdaI.results
Bio.Restriction.Restriction.BetI.results
Bio.Restriction.Restriction.BfaI.results
Bio.Restriction.Restriction.BfiI.results
Bio.Restriction.Restriction.BfmI.results
Bio.Restriction.Restriction.BfoI.results
Bio.Restriction.Restriction.BfrI.results
Bio.Restriction.Restriction.BfuAI.results
Bio.Restriction.Restriction.BfuCI.results
Bio.Restriction.Restriction.BfuI.results
Bio.Restriction.Restriction.BglI.results
Bio.Restriction.Restriction.BglII.results
Bio.Restriction.Restriction.BinI.results
Bio.Restriction.Restriction.BisI.results
Bio.Restriction.Restriction.BlnI.results
Bio.Restriction.Restriction.BlpI.results
Bio.Restriction.Restriction.BlsI.results
Bio.Restriction.Restriction.BmcAI.results
Bio.Restriction.Restriction.Bme1390I.results
Bio.Restriction.Restriction.Bme18I.results
Bio.Restriction.Restriction.BmeDI.results
Bio.Restriction.Restriction.BmeRI.results
Bio.Restriction.Restriction.BmeT110I.results
Bio.Restriction.Restriction.BmgBI.results
Bio.Restriction.Restriction.BmgI.results
Bio.Restriction.Restriction.BmgT120I.results
Bio.Restriction.Restriction.BmiI.results
Bio.Restriction.Restriction.BmrFI.results
Bio.Restriction.Restriction.BmrI.results
Bio.Restriction.Restriction.BmsI.results
Bio.Restriction.Restriction.BmtI.results
Bio.Restriction.Restriction.BmuI.results
Bio.Restriction.Restriction.BoxI.results
Bio.Restriction.Restriction.BpiI.results
Bio.Restriction.Restriction.BplI.results
Bio.Restriction.Restriction.BpmI.results
Bio.Restriction.Restriction.Bpu10I.results
Bio.Restriction.Restriction.Bpu1102I.results
Bio.Restriction.Restriction.Bpu14I.results
Bio.Restriction.Restriction.BpuEI.results
Bio.Restriction.Restriction.BpuMI.results
Bio.Restriction.Restriction.BpvUI.results
Bio.Restriction.Restriction.Bsa29I.results
Bio.Restriction.Restriction.BsaAI.results
Bio.Restriction.Restriction.BsaBI.results
Bio.Restriction.Restriction.BsaHI.results
Bio.Restriction.Restriction.BsaI.results
Bio.Restriction.Restriction.BsaJI.results
Bio.Restriction.Restriction.BsaWI.results
Bio.Restriction.Restriction.BsaXI.results
Bio.Restriction.Restriction.BsbI.results
Bio.Restriction.Restriction.Bsc4I.results
Bio.Restriction.Restriction.BscAI.results
Bio.Restriction.Restriction.BscGI.results
Bio.Restriction.Restriction.Bse118I.results
Bio.Restriction.Restriction.Bse1I.results
Bio.Restriction.Restriction.Bse21I.results
Bio.Restriction.Restriction.Bse3DI.results
Bio.Restriction.Restriction.Bse8I.results
Bio.Restriction.Restriction.BseAI.results
Bio.Restriction.Restriction.BseBI.results
Bio.Restriction.Restriction.BseCI.results
Bio.Restriction.Restriction.BseDI.results
Bio.Restriction.Restriction.BseGI.results
Bio.Restriction.Restriction.BseJI.results
Bio.Restriction.Restriction.BseLI.results
Bio.Restriction.Restriction.BseMI.results
Bio.Restriction.Restriction.BseMII.results
Bio.Restriction.Restriction.BseNI.results
Bio.Restriction.Restriction.BsePI.results
Bio.Restriction.Restriction.BseRI.results
Bio.Restriction.Restriction.BseSI.results
Bio.Restriction.Restriction.BseX3I.results
Bio.Restriction.Restriction.BseXI.results
Bio.Restriction.Restriction.BseYI.results
Bio.Restriction.Restriction.BsgI.results
Bio.Restriction.Restriction.Bsh1236I.results
Bio.Restriction.Restriction.Bsh1285I.results
Bio.Restriction.Restriction.BshFI.results
Bio.Restriction.Restriction.BshNI.results
Bio.Restriction.Restriction.BshTI.results
Bio.Restriction.Restriction.BshVI.results
Bio.Restriction.Restriction.BsiEI.results
Bio.Restriction.Restriction.BsiHKAI.results
Bio.Restriction.Restriction.BsiHKCI.results
Bio.Restriction.Restriction.BsiI.results
Bio.Restriction.Restriction.BsiSI.results
Bio.Restriction.Restriction.BsiWI.results
Bio.Restriction.Restriction.BsiYI.results
Bio.Restriction.Restriction.BslFI.results
Bio.Restriction.Restriction.BslI.results
Bio.Restriction.Restriction.BsmAI.results
Bio.Restriction.Restriction.BsmBI.results
Bio.Restriction.Restriction.BsmFI.results
Bio.Restriction.Restriction.BsmI.results
Bio.Restriction.Restriction.BsnI.results
Bio.Restriction.Restriction.Bso31I.results
Bio.Restriction.Restriction.BsoBI.results
Bio.Restriction.Restriction.Bsp119I.results
Bio.Restriction.Restriction.Bsp120I.results
Bio.Restriction.Restriction.Bsp1286I.results
Bio.Restriction.Restriction.Bsp13I.results
Bio.Restriction.Restriction.Bsp1407I.results
Bio.Restriction.Restriction.Bsp143I.results
Bio.Restriction.Restriction.Bsp1720I.results
Bio.Restriction.Restriction.Bsp19I.results
Bio.Restriction.Restriction.Bsp24I.results
Bio.Restriction.Restriction.Bsp68I.results
Bio.Restriction.Restriction.BspACI.results
Bio.Restriction.Restriction.BspCNI.results
Bio.Restriction.Restriction.BspD6I.results
Bio.Restriction.Restriction.BspDI.results
Bio.Restriction.Restriction.BspEI.results
Bio.Restriction.Restriction.BspFNI.results
Bio.Restriction.Restriction.BspGI.results
Bio.Restriction.Restriction.BspHI.results
Bio.Restriction.Restriction.BspLI.results
Bio.Restriction.Restriction.BspLU11I.results
Bio.Restriction.Restriction.BspMI.results
Bio.Restriction.Restriction.BspMII.results
Bio.Restriction.Restriction.BspNCI.results
Bio.Restriction.Restriction.BspOI.results
Bio.Restriction.Restriction.BspPI.results
Bio.Restriction.Restriction.BspQI.results
Bio.Restriction.Restriction.BspT104I.results
Bio.Restriction.Restriction.BspT107I.results
Bio.Restriction.Restriction.BspTI.results
Bio.Restriction.Restriction.BsrBI.results
Bio.Restriction.Restriction.BsrDI.results
Bio.Restriction.Restriction.BsrFI.results
Bio.Restriction.Restriction.BsrGI.results
Bio.Restriction.Restriction.BsrI.results
Bio.Restriction.Restriction.BsrSI.results
Bio.Restriction.Restriction.BssAI.results
Bio.Restriction.Restriction.BssECI.results
Bio.Restriction.Restriction.BssHII.results
Bio.Restriction.Restriction.BssKI.results
Bio.Restriction.Restriction.BssMI.results
Bio.Restriction.Restriction.BssNAI.results
Bio.Restriction.Restriction.BssNI.results
Bio.Restriction.Restriction.BssSI.results
Bio.Restriction.Restriction.BssT1I.results
Bio.Restriction.Restriction.Bst1107I.results
Bio.Restriction.Restriction.Bst2BI.results
Bio.Restriction.Restriction.Bst2UI.results
Bio.Restriction.Restriction.Bst4CI.results
Bio.Restriction.Restriction.Bst6I.results
Bio.Restriction.Restriction.BstACI.results
Bio.Restriction.Restriction.BstAFI.results
Bio.Restriction.Restriction.BstAPI.results
Bio.Restriction.Restriction.BstAUI.results
Bio.Restriction.Restriction.BstBAI.results
Bio.Restriction.Restriction.BstBI.results
Bio.Restriction.Restriction.BstC8I.results
Bio.Restriction.Restriction.BstDEI.results
Bio.Restriction.Restriction.BstDSI.results
Bio.Restriction.Restriction.BstEII.results
Bio.Restriction.Restriction.BstENI.results
Bio.Restriction.Restriction.BstF5I.results
Bio.Restriction.Restriction.BstFNI.results
Bio.Restriction.Restriction.BstH2I.results
Bio.Restriction.Restriction.BstHHI.results
Bio.Restriction.Restriction.BstKTI.results
Bio.Restriction.Restriction.BstMAI.results
Bio.Restriction.Restriction.BstMBI.results
Bio.Restriction.Restriction.BstMCI.results
Bio.Restriction.Restriction.BstMWI.results
Bio.Restriction.Restriction.BstNI.results
Bio.Restriction.Restriction.BstNSI.results
Bio.Restriction.Restriction.BstOI.results
Bio.Restriction.Restriction.BstPAI.results
Bio.Restriction.Restriction.BstPI.results
Bio.Restriction.Restriction.BstSCI.results
Bio.Restriction.Restriction.BstSFI.results
Bio.Restriction.Restriction.BstSLI.results
Bio.Restriction.Restriction.BstSNI.results
Bio.Restriction.Restriction.BstUI.results
Bio.Restriction.Restriction.BstV1I.results
Bio.Restriction.Restriction.BstV2I.results
Bio.Restriction.Restriction.BstX2I.results
Bio.Restriction.Restriction.BstXI.results
Bio.Restriction.Restriction.BstYI.results
Bio.Restriction.Restriction.BstZ17I.results
Bio.Restriction.Restriction.BstZI.results
Bio.Restriction.Restriction.Bsu15I.results
Bio.Restriction.Restriction.Bsu36I.results
Bio.Restriction.Restriction.BsuI.results
Bio.Restriction.Restriction.BsuRI.results
Bio.Restriction.Restriction.BtgI.results
Bio.Restriction.Restriction.BtgZI.results
Bio.Restriction.Restriction.BthCI.results
Bio.Restriction.Restriction.BtrI.results
Bio.Restriction.Restriction.BtsCI.results
Bio.Restriction.Restriction.BtsI.results
Bio.Restriction.Restriction.BtsIMutI.results
Bio.Restriction.Restriction.BtuMI.results
Bio.Restriction.Restriction.BveI.results
Bio.Restriction.Restriction.Cac8I.results
Bio.Restriction.Restriction.CaiI.results
Bio.Restriction.Restriction.CauII.results
Bio.Restriction.Restriction.CchII.results
Bio.Restriction.Restriction.CchIII.results
Bio.Restriction.Restriction.CciI.results
Bio.Restriction.Restriction.CciNI.results
Bio.Restriction.Restriction.Cdi630V.results
Bio.Restriction.Restriction.CdiI.results
Bio.Restriction.Restriction.CdpI.results
Bio.Restriction.Restriction.CfoI.results
Bio.Restriction.Restriction.Cfr10I.results
Bio.Restriction.Restriction.Cfr13I.results
Bio.Restriction.Restriction.Cfr42I.results
Bio.Restriction.Restriction.Cfr9I.results
Bio.Restriction.Restriction.CfrI.results
Bio.Restriction.Restriction.Cgl13032I.results
Bio.Restriction.Restriction.Cgl13032II.results
Bio.Restriction.Restriction.ChaI.results
Bio.Restriction.Restriction.CjeFIII.results
Bio.Restriction.Restriction.CjeFV.results
Bio.Restriction.Restriction.CjeI.results
Bio.Restriction.Restriction.CjeNII.results
Bio.Restriction.Restriction.CjeNIII.results
Bio.Restriction.Restriction.CjeP659IV.results
Bio.Restriction.Restriction.CjePI.results
Bio.Restriction.Restriction.CjuI.results
Bio.Restriction.Restriction.CjuII.results
Bio.Restriction.Restriction.ClaI.results
Bio.Restriction.Restriction.CpoI.results
Bio.Restriction.Restriction.CseI.results
Bio.Restriction.Restriction.CsiI.results
Bio.Restriction.Restriction.Csp6I.results
Bio.Restriction.Restriction.CspAI.results
Bio.Restriction.Restriction.CspCI.results
Bio.Restriction.Restriction.CspI.results
Bio.Restriction.Restriction.CstMI.results
Bio.Restriction.Restriction.CviAII.results
Bio.Restriction.Restriction.CviJI.results
Bio.Restriction.Restriction.CviKI_1.results
Bio.Restriction.Restriction.CviQI.results
Bio.Restriction.Restriction.CviRI.results
Bio.Restriction.Restriction.DdeI.results
Bio.Restriction.Restriction.DinI.results
Bio.Restriction.Restriction.DpnI.results
Bio.Restriction.Restriction.DpnII.results
Bio.Restriction.Restriction.DraI.results
Bio.Restriction.Restriction.DraII.results
Bio.Restriction.Restriction.DraIII.results
Bio.Restriction.Restriction.DraRI.results
Bio.Restriction.Restriction.DrdI.results
Bio.Restriction.Restriction.DrdII.results
Bio.Restriction.Restriction.DriI.results
Bio.Restriction.Restriction.DsaI.results
Bio.Restriction.Restriction.DseDI.results
Bio.Restriction.Restriction.EaeI.results
Bio.Restriction.Restriction.EagI.results
Bio.Restriction.Restriction.Eam1104I.results
Bio.Restriction.Restriction.Eam1105I.results
Bio.Restriction.Restriction.EarI.results
Bio.Restriction.Restriction.EciI.results
Bio.Restriction.Restriction.Ecl136II.results
Bio.Restriction.Restriction.EclXI.results
Bio.Restriction.Restriction.Eco105I.results
Bio.Restriction.Restriction.Eco130I.results
Bio.Restriction.Restriction.Eco147I.results
Bio.Restriction.Restriction.Eco24I.results
Bio.Restriction.Restriction.Eco31I.results
Bio.Restriction.Restriction.Eco32I.results
Bio.Restriction.Restriction.Eco47I.results
Bio.Restriction.Restriction.Eco47III.results
Bio.Restriction.Restriction.Eco52I.results
Bio.Restriction.Restriction.Eco53kI.results
Bio.Restriction.Restriction.Eco57I.results
Bio.Restriction.Restriction.Eco57MI.results
Bio.Restriction.Restriction.Eco72I.results
Bio.Restriction.Restriction.Eco81I.results
Bio.Restriction.Restriction.Eco88I.results
Bio.Restriction.Restriction.Eco91I.results
Bio.Restriction.Restriction.EcoHI.results
Bio.Restriction.Restriction.EcoICRI.results
Bio.Restriction.Restriction.EcoNI.results
Bio.Restriction.Restriction.EcoO109I.results
Bio.Restriction.Restriction.EcoO65I.results
Bio.Restriction.Restriction.EcoRI.results
Bio.Restriction.Restriction.EcoRII.results
Bio.Restriction.Restriction.EcoRV.results
Bio.Restriction.Restriction.EcoT14I.results
Bio.Restriction.Restriction.EcoT22I.results
Bio.Restriction.Restriction.EcoT38I.results
Bio.Restriction.Restriction.EgeI.results
Bio.Restriction.Restriction.EheI.results
Bio.Restriction.Restriction.ErhI.results
Bio.Restriction.Restriction.EsaBC3I.results
Bio.Restriction.Restriction.EsaSSI.results
Bio.Restriction.Restriction.Esp3I.results
Bio.Restriction.Restriction.EspI.results
Bio.Restriction.Restriction.FaeI.results
Bio.Restriction.Restriction.FaiI.results
Bio.Restriction.Restriction.FalI.results
Bio.Restriction.Restriction.FaqI.results
Bio.Restriction.Restriction.FatI.results
Bio.Restriction.Restriction.FauI.results
Bio.Restriction.Restriction.FauNDI.results
Bio.Restriction.Restriction.FbaI.results
Bio.Restriction.Restriction.FblI.results
Bio.Restriction.Restriction.FinI.results
Bio.Restriction.Restriction.FmuI.results
Bio.Restriction.Restriction.Fnu4HI.results
Bio.Restriction.Restriction.FnuDII.results
Bio.Restriction.Restriction.FokI.results
Bio.Restriction.Restriction.FriOI.results
Bio.Restriction.Restriction.FseI.results
Bio.Restriction.Restriction.Fsp4HI.results
Bio.Restriction.Restriction.FspAI.results
Bio.Restriction.Restriction.FspBI.results
Bio.Restriction.Restriction.FspEI.results
Bio.Restriction.Restriction.FspI.results
Bio.Restriction.Restriction.GauT27I.results
Bio.Restriction.Restriction.GdiII.results
Bio.Restriction.Restriction.GlaI.results
Bio.Restriction.Restriction.GluI.results
Bio.Restriction.Restriction.GsaI.results
Bio.Restriction.Restriction.GsuI.results
Bio.Restriction.Restriction.HaeI.results
Bio.Restriction.Restriction.HaeII.results
Bio.Restriction.Restriction.HaeIII.results
Bio.Restriction.Restriction.HapII.results
Bio.Restriction.Restriction.HauII.results
Bio.Restriction.Restriction.HgaI.results
Bio.Restriction.Restriction.HgiAI.results
Bio.Restriction.Restriction.HgiCI.results
Bio.Restriction.Restriction.HgiEII.results
Bio.Restriction.Restriction.HgiJII.results
Bio.Restriction.Restriction.HhaI.results
Bio.Restriction.Restriction.Hin1I.results
Bio.Restriction.Restriction.Hin1II.results
Bio.Restriction.Restriction.Hin4I.results
Bio.Restriction.Restriction.Hin4II.results
Bio.Restriction.Restriction.Hin6I.results
Bio.Restriction.Restriction.HinP1I.results
Bio.Restriction.Restriction.HincII.results
Bio.Restriction.Restriction.HindII.results
Bio.Restriction.Restriction.HindIII.results
Bio.Restriction.Restriction.HinfI.results
Bio.Restriction.Restriction.HpaI.results
Bio.Restriction.Restriction.HpaII.results
Bio.Restriction.Restriction.HphI.results
Bio.Restriction.Restriction.Hpy166II.results
Bio.Restriction.Restriction.Hpy178III.results
Bio.Restriction.Restriction.Hpy188I.results
Bio.Restriction.Restriction.Hpy188III.results
Bio.Restriction.Restriction.Hpy8I.results
Bio.Restriction.Restriction.Hpy99I.results
Bio.Restriction.Restriction.Hpy99XIII.results
Bio.Restriction.Restriction.Hpy99XIV.results
Bio.Restriction.Restriction.HpyAV.results
Bio.Restriction.Restriction.HpyCH4III.results
Bio.Restriction.Restriction.HpyCH4IV.results
Bio.Restriction.Restriction.HpyCH4V.results
Bio.Restriction.Restriction.HpyF10VI.results
Bio.Restriction.Restriction.HpyF3I.results
Bio.Restriction.Restriction.HpySE526I.results
Bio.Restriction.Restriction.Hsp92I.results
Bio.Restriction.Restriction.Hsp92II.results
Bio.Restriction.Restriction.HspAI.results
Bio.Restriction.Restriction.Jma19592I.results
Bio.Restriction.Restriction.KasI.results
Bio.Restriction.Restriction.KflI.results
Bio.Restriction.Restriction.Kpn2I.results
Bio.Restriction.Restriction.KpnI.results
Bio.Restriction.Restriction.KroI.results
Bio.Restriction.Restriction.Ksp22I.results
Bio.Restriction.Restriction.Ksp632I.results
Bio.Restriction.Restriction.KspAI.results
Bio.Restriction.Restriction.KspI.results
Bio.Restriction.Restriction.Kzo9I.results
Bio.Restriction.Restriction.LguI.results
Bio.Restriction.Restriction.LpnI.results
Bio.Restriction.Restriction.LpnPI.results
Bio.Restriction.Restriction.Lsp1109I.results
Bio.Restriction.Restriction.LweI.results
Bio.Restriction.Restriction.MabI.results
Bio.Restriction.Restriction.MaeI.results
Bio.Restriction.Restriction.MaeII.results
Bio.Restriction.Restriction.MaeIII.results
Bio.Restriction.Restriction.MalI.results
Bio.Restriction.Restriction.MaqI.results
Bio.Restriction.Restriction.MauBI.results
Bio.Restriction.Restriction.MbiI.results
Bio.Restriction.Restriction.MboI.results
Bio.Restriction.Restriction.MboII.results
Bio.Restriction.Restriction.McaTI.results
Bio.Restriction.Restriction.McrI.results
Bio.Restriction.Restriction.MfeI.results
Bio.Restriction.Restriction.MflI.results
Bio.Restriction.Restriction.MhlI.results
Bio.Restriction.Restriction.MjaIV.results
Bio.Restriction.Restriction.MkaDII.results
Bio.Restriction.Restriction.MlsI.results
Bio.Restriction.Restriction.MluCI.results
Bio.Restriction.Restriction.MluI.results
Bio.Restriction.Restriction.MluNI.results
Bio.Restriction.Restriction.Mly113I.results
Bio.Restriction.Restriction.MlyI.results
Bio.Restriction.Restriction.MmeI.results
Bio.Restriction.Restriction.MnlI.results
Bio.Restriction.Restriction.Mph1103I.results
Bio.Restriction.Restriction.MreI.results
Bio.Restriction.Restriction.MroI.results
Bio.Restriction.Restriction.MroNI.results
Bio.Restriction.Restriction.MroXI.results
Bio.Restriction.Restriction.MscI.results
Bio.Restriction.Restriction.MseI.results
Bio.Restriction.Restriction.MslI.results
Bio.Restriction.Restriction.Msp20I.results
Bio.Restriction.Restriction.MspA1I.results
Bio.Restriction.Restriction.MspCI.results
Bio.Restriction.Restriction.MspI.results
Bio.Restriction.Restriction.MspJI.results
Bio.Restriction.Restriction.MspR9I.results
Bio.Restriction.Restriction.MssI.results
Bio.Restriction.Restriction.MstI.results
Bio.Restriction.Restriction.MunI.results
Bio.Restriction.Restriction.Mva1269I.results
Bio.Restriction.Restriction.MvaI.results
Bio.Restriction.Restriction.MvnI.results
Bio.Restriction.Restriction.MvrI.results
Bio.Restriction.Restriction.MwoI.results
Bio.Restriction.Restriction.NaeI.results
Bio.Restriction.Restriction.NarI.results
Bio.Restriction.Restriction.NciI.results
Bio.Restriction.Restriction.NcoI.results
Bio.Restriction.Restriction.NdeI.results
Bio.Restriction.Restriction.NdeII.results
Bio.Restriction.Restriction.NgoAVIII.results
Bio.Restriction.Restriction.NgoMIV.results
Bio.Restriction.Restriction.NhaXI.results
Bio.Restriction.Restriction.NheI.results
Bio.Restriction.Restriction.NlaCI.results
Bio.Restriction.Restriction.NlaIII.results
Bio.Restriction.Restriction.NlaIV.results
Bio.Restriction.Restriction.Nli3877I.results
Bio.Restriction.Restriction.NmeAIII.results
Bio.Restriction.Restriction.NmeDI.results
Bio.Restriction.Restriction.NmuCI.results
Bio.Restriction.Restriction.NotI.results
Bio.Restriction.Restriction.NruI.results
Bio.Restriction.Restriction.NsbI.results
Bio.Restriction.Restriction.NsiI.results
Bio.Restriction.Restriction.NspBII.results
Bio.Restriction.Restriction.NspI.results
Bio.Restriction.Restriction.NspV.results
Bio.Restriction.Restriction.OliI.results
Bio.Restriction.Restriction.PabI.results
Bio.Restriction.Restriction.PacI.results
Bio.Restriction.Restriction.PaeI.results
Bio.Restriction.Restriction.PaeR7I.results
Bio.Restriction.Restriction.PagI.results
Bio.Restriction.Restriction.PalAI.results
Bio.Restriction.Restriction.PasI.results
Bio.Restriction.Restriction.PauI.results
Bio.Restriction.Restriction.PceI.results
Bio.Restriction.Restriction.PciI.results
Bio.Restriction.Restriction.PciSI.results
Bio.Restriction.Restriction.PcsI.results
Bio.Restriction.Restriction.PctI.results
Bio.Restriction.Restriction.PdiI.results
Bio.Restriction.Restriction.PdmI.results
Bio.Restriction.Restriction.PenI.results
Bio.Restriction.Restriction.PfeI.results
Bio.Restriction.Restriction.Pfl1108I.results
Bio.Restriction.Restriction.Pfl23II.results
Bio.Restriction.Restriction.PflFI.results
Bio.Restriction.Restriction.PflMI.results
Bio.Restriction.Restriction.PfoI.results
Bio.Restriction.Restriction.PinAI.results
Bio.Restriction.Restriction.PlaDI.results
Bio.Restriction.Restriction.Ple19I.results
Bio.Restriction.Restriction.PleI.results
Bio.Restriction.Restriction.PluTI.results
Bio.Restriction.Restriction.PmaCI.results
Bio.Restriction.Restriction.PmeI.results
Bio.Restriction.Restriction.PmlI.results
Bio.Restriction.Restriction.PpiI.results
Bio.Restriction.Restriction.PpsI.results
Bio.Restriction.Restriction.Ppu10I.results
Bio.Restriction.Restriction.Ppu21I.results
Bio.Restriction.Restriction.PpuMI.results
Bio.Restriction.Restriction.PscI.results
Bio.Restriction.Restriction.PshAI.results
Bio.Restriction.Restriction.PshBI.results
Bio.Restriction.Restriction.PsiI.results
Bio.Restriction.Restriction.Psp03I.results
Bio.Restriction.Restriction.Psp124BI.results
Bio.Restriction.Restriction.Psp1406I.results
Bio.Restriction.Restriction.Psp5II.results
Bio.Restriction.Restriction.Psp6I.results
Bio.Restriction.Restriction.PspCI.results
Bio.Restriction.Restriction.PspEI.results
Bio.Restriction.Restriction.PspGI.results
Bio.Restriction.Restriction.PspLI.results
Bio.Restriction.Restriction.PspN4I.results
Bio.Restriction.Restriction.PspOMI.results
Bio.Restriction.Restriction.PspOMII.results
Bio.Restriction.Restriction.PspPI.results
Bio.Restriction.Restriction.PspPPI.results
Bio.Restriction.Restriction.PspPRI.results
Bio.Restriction.Restriction.PspXI.results
Bio.Restriction.Restriction.PsrI.results
Bio.Restriction.Restriction.PssI.results
Bio.Restriction.Restriction.PstI.results
Bio.Restriction.Restriction.PstNI.results
Bio.Restriction.Restriction.PsuI.results
Bio.Restriction.Restriction.PsyI.results
Bio.Restriction.Restriction.PteI.results
Bio.Restriction.Restriction.PvuI.results
Bio.Restriction.Restriction.PvuII.results
Bio.Restriction.Restriction.R2_BceSIV.results
Bio.Restriction.Restriction.RceI.results
Bio.Restriction.Restriction.RdeGBI.results
Bio.Restriction.Restriction.RdeGBII.results
Bio.Restriction.Restriction.RdeGBIII.results
Bio.Restriction.Restriction.RflFIII.results
Bio.Restriction.Restriction.RgaI.results
Bio.Restriction.Restriction.RigI.results
Bio.Restriction.Restriction.RlaI.results
Bio.Restriction.Restriction.RleAI.results
Bio.Restriction.Restriction.RpaB5I.results
Bio.Restriction.Restriction.RpaBI.results
Bio.Restriction.Restriction.RpaI.results
Bio.Restriction.Restriction.RpaTI.results
Bio.Restriction.Restriction.RruI.results
Bio.Restriction.Restriction.RsaI.results
Bio.Restriction.Restriction.RsaNI.results
Bio.Restriction.Restriction.RseI.results
Bio.Restriction.Restriction.Rsr2I.results
Bio.Restriction.Restriction.RsrII.results
Bio.Restriction.Restriction.SacI.results
Bio.Restriction.Restriction.SacII.results
Bio.Restriction.Restriction.SalI.results
Bio.Restriction.Restriction.SanDI.results
Bio.Restriction.Restriction.SapI.results
Bio.Restriction.Restriction.SaqAI.results
Bio.Restriction.Restriction.SatI.results
Bio.Restriction.Restriction.Sau3AI.results
Bio.Restriction.Restriction.Sau96I.results
Bio.Restriction.Restriction.SauI.results
Bio.Restriction.Restriction.SbfI.results
Bio.Restriction.Restriction.ScaI.results
Bio.Restriction.Restriction.SchI.results
Bio.Restriction.Restriction.SciI.results
Bio.Restriction.Restriction.ScrFI.results
Bio.Restriction.Restriction.SdaI.results
Bio.Restriction.Restriction.SdeAI.results
Bio.Restriction.Restriction.SdeOSI.results
Bio.Restriction.Restriction.SduI.results
Bio.Restriction.Restriction.SecI.results
Bio.Restriction.Restriction.SelI.results
Bio.Restriction.Restriction.SetI.results
Bio.Restriction.Restriction.SexAI.results
Bio.Restriction.Restriction.SfaAI.results
Bio.Restriction.Restriction.SfaNI.results
Bio.Restriction.Restriction.SfcI.results
Bio.Restriction.Restriction.SfeI.results
Bio.Restriction.Restriction.SfiI.results
Bio.Restriction.Restriction.SfoI.results
Bio.Restriction.Restriction.Sfr274I.results
Bio.Restriction.Restriction.Sfr303I.results
Bio.Restriction.Restriction.SfuI.results
Bio.Restriction.Restriction.SgeI.results
Bio.Restriction.Restriction.SgfI.results
Bio.Restriction.Restriction.SgrAI.results
Bio.Restriction.Restriction.SgrBI.results
Bio.Restriction.Restriction.SgrDI.results
Bio.Restriction.Restriction.SgrTI.results
Bio.Restriction.Restriction.SgsI.results
Bio.Restriction.Restriction.SimI.results
Bio.Restriction.Restriction.SlaI.results
Bio.Restriction.Restriction.SmaI.results
Bio.Restriction.Restriction.SmiI.results
Bio.Restriction.Restriction.SmiMI.results
Bio.Restriction.Restriction.SmlI.results
Bio.Restriction.Restriction.SmoI.results
Bio.Restriction.Restriction.SnaBI.results
Bio.Restriction.Restriction.SnaI.results
Bio.Restriction.Restriction.Sno506I.results
Bio.Restriction.Restriction.SpeI.results
Bio.Restriction.Restriction.SphI.results
Bio.Restriction.Restriction.SplI.results
Bio.Restriction.Restriction.SpoDI.results
Bio.Restriction.Restriction.SrfI.results
Bio.Restriction.Restriction.Sse232I.results
Bio.Restriction.Restriction.Sse8387I.results
Bio.Restriction.Restriction.Sse8647I.results
Bio.Restriction.Restriction.Sse9I.results
Bio.Restriction.Restriction.SseBI.results
Bio.Restriction.Restriction.SsiI.results
Bio.Restriction.Restriction.SspD5I.results
Bio.Restriction.Restriction.SspDI.results
Bio.Restriction.Restriction.SspI.results
Bio.Restriction.Restriction.SstE37I.results
Bio.Restriction.Restriction.SstI.results
Bio.Restriction.Restriction.Sth132I.results
Bio.Restriction.Restriction.Sth302II.results
Bio.Restriction.Restriction.StrI.results
Bio.Restriction.Restriction.StsI.results
Bio.Restriction.Restriction.StuI.results
Bio.Restriction.Restriction.StyD4I.results
Bio.Restriction.Restriction.StyI.results
Bio.Restriction.Restriction.SwaI.results
Bio.Restriction.Restriction.TaaI.results
Bio.Restriction.Restriction.TaiI.results
Bio.Restriction.Restriction.TaqI.results
Bio.Restriction.Restriction.TaqII.results
Bio.Restriction.Restriction.TasI.results
Bio.Restriction.Restriction.TatI.results
Bio.Restriction.Restriction.TauI.results
Bio.Restriction.Restriction.TfiI.results
Bio.Restriction.Restriction.Tru1I.results
Bio.Restriction.Restriction.Tru9I.results
Bio.Restriction.Restriction.TscAI.results
Bio.Restriction.Restriction.TseFI.results
Bio.Restriction.Restriction.TseI.results
Bio.Restriction.Restriction.TsoI.results
Bio.Restriction.Restriction.Tsp45I.results
Bio.Restriction.Restriction.Tsp4CI.results
Bio.Restriction.Restriction.TspDTI.results
Bio.Restriction.Restriction.TspEI.results
Bio.Restriction.Restriction.TspGWI.results
Bio.Restriction.Restriction.TspMI.results
Bio.Restriction.Restriction.TspRI.results
Bio.Restriction.Restriction.TssI.results
Bio.Restriction.Restriction.TstI.results
Bio.Restriction.Restriction.TsuI.results
Bio.Restriction.Restriction.Tth111I.results
Bio.Restriction.Restriction.Tth111II.results
Bio.Restriction.Restriction.UbaF11I.results
Bio.Restriction.Restriction.UbaF12I.results
Bio.Restriction.Restriction.UbaF13I.results
Bio.Restriction.Restriction.UbaF14I.results
Bio.Restriction.Restriction.UbaF9I.results
Bio.Restriction.Restriction.UbaPI.results
Bio.Restriction.Restriction.UcoMSI.results
Bio.Restriction.Restriction.UnbI.results
Bio.Restriction.Restriction.Van91I.results
Bio.Restriction.Restriction.Vha464I.results
Bio.Restriction.Restriction.VneI.results
Bio.Restriction.Restriction.VpaK11AI.results
Bio.Restriction.Restriction.VpaK11BI.results
Bio.Restriction.Restriction.VspI.results
Bio.Restriction.Restriction.WviI.results
Bio.Restriction.Restriction.XagI.results
Bio.Restriction.Restriction.XapI.results
Bio.Restriction.Restriction.XbaI.results
Bio.Restriction.Restriction.XceI.results
Bio.Restriction.Restriction.XcmI.results
Bio.Restriction.Restriction.XhoI.results
Bio.Restriction.Restriction.XhoII.results
Bio.Restriction.Restriction.XmaI.results
Bio.Restriction.Restriction.XmaIII.results
Bio.Restriction.Restriction.XmaJI.results
Bio.Restriction.Restriction.XmiI.results
Bio.Restriction.Restriction.XmnI.results
Bio.Restriction.Restriction.XspI.results
Bio.Restriction.Restriction.YkrI.results
Bio.Restriction.Restriction.ZraI.results
Bio.Restriction.Restriction.ZrmI.results
Bio.Restriction.Restriction.Zsp2I.results" class="py-name" href="#" onclick="return doclink('link-996', 'results', 'link-996');">results</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L1707"></a><tt class="py-lineno">1707</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">c</tt> <tt class="py-keyword">in</tt> <tt class="py-name">cols_to_get</tt><tt class="py-op">:</tt> </tt>
<a name="L1708"></a><tt class="py-lineno">1708</tt>  <tt class="py-line">        <tt id="link-997" class="py-name"><a title="Bio.Restriction.Restriction.AanI.results
Bio.Restriction.Restriction.AarI.results
Bio.Restriction.Restriction.AasI.results
Bio.Restriction.Restriction.AatII.results
Bio.Restriction.Restriction.AbaSI.results
Bio.Restriction.Restriction.AbsI.results
Bio.Restriction.Restriction.Acc16I.results
Bio.Restriction.Restriction.Acc36I.results
Bio.Restriction.Restriction.Acc65I.results
Bio.Restriction.Restriction.AccB1I.results
Bio.Restriction.Restriction.AccB7I.results
Bio.Restriction.Restriction.AccBSI.results
Bio.Restriction.Restriction.AccI.results
Bio.Restriction.Restriction.AccII.results
Bio.Restriction.Restriction.AccIII.results
Bio.Restriction.Restriction.AceIII.results
Bio.Restriction.Restriction.AciI.results
Bio.Restriction.Restriction.AclI.results
Bio.Restriction.Restriction.AclWI.results
Bio.Restriction.Restriction.AcoI.results
Bio.Restriction.Restriction.AcsI.results
Bio.Restriction.Restriction.AcuI.results
Bio.Restriction.Restriction.AcvI.results
Bio.Restriction.Restriction.AcyI.results
Bio.Restriction.Restriction.AdeI.results
Bio.Restriction.Restriction.AfaI.results
Bio.Restriction.Restriction.AfeI.results
Bio.Restriction.Restriction.AfiI.results
Bio.Restriction.Restriction.AflII.results
Bio.Restriction.Restriction.AflIII.results
Bio.Restriction.Restriction.AgeI.results
Bio.Restriction.Restriction.AgsI.results
Bio.Restriction.Restriction.AhaIII.results
Bio.Restriction.Restriction.AhdI.results
Bio.Restriction.Restriction.AhlI.results
Bio.Restriction.Restriction.AjiI.results
Bio.Restriction.Restriction.AjnI.results
Bio.Restriction.Restriction.AjuI.results
Bio.Restriction.Restriction.AleI.results
Bio.Restriction.Restriction.AlfI.results
Bio.Restriction.Restriction.AloI.results
Bio.Restriction.Restriction.AluBI.results
Bio.Restriction.Restriction.AluI.results
Bio.Restriction.Restriction.Alw21I.results
Bio.Restriction.Restriction.Alw26I.results
Bio.Restriction.Restriction.Alw44I.results
Bio.Restriction.Restriction.AlwFI.results
Bio.Restriction.Restriction.AlwI.results
Bio.Restriction.Restriction.AlwNI.results
Bio.Restriction.Restriction.Ama87I.results
Bio.Restriction.Restriction.Aor13HI.results
Bio.Restriction.Restriction.Aor51HI.results
Bio.Restriction.Restriction.AoxI.results
Bio.Restriction.Restriction.ApaBI.results
Bio.Restriction.Restriction.ApaI.results
Bio.Restriction.Restriction.ApaLI.results
Bio.Restriction.Restriction.ApeKI.results
Bio.Restriction.Restriction.ApoI.results
Bio.Restriction.Restriction.ApyPI.results
Bio.Restriction.Restriction.AquII.results
Bio.Restriction.Restriction.AquIII.results
Bio.Restriction.Restriction.AquIV.results
Bio.Restriction.Restriction.ArsI.results
Bio.Restriction.Restriction.AscI.results
Bio.Restriction.Restriction.AseI.results
Bio.Restriction.Restriction.Asi256I.results
Bio.Restriction.Restriction.AsiGI.results
Bio.Restriction.Restriction.AsiSI.results
Bio.Restriction.Restriction.Asp700I.results
Bio.Restriction.Restriction.Asp718I.results
Bio.Restriction.Restriction.AspA2I.results
Bio.Restriction.Restriction.AspBHI.results
Bio.Restriction.Restriction.AspLEI.results
Bio.Restriction.Restriction.AspS9I.results
Bio.Restriction.Restriction.AssI.results
Bio.Restriction.Restriction.AsuC2I.results
Bio.Restriction.Restriction.AsuHPI.results
Bio.Restriction.Restriction.AsuI.results
Bio.Restriction.Restriction.AsuII.results
Bio.Restriction.Restriction.AsuNHI.results
Bio.Restriction.Restriction.AvaI.results
Bio.Restriction.Restriction.AvaII.results
Bio.Restriction.Restriction.AvaIII.results
Bio.Restriction.Restriction.AvrII.results
Bio.Restriction.Restriction.AxyI.results
Bio.Restriction.Restriction.BaeGI.results
Bio.Restriction.Restriction.BaeI.results
Bio.Restriction.Restriction.BalI.results
Bio.Restriction.Restriction.BamHI.results
Bio.Restriction.Restriction.BanI.results
Bio.Restriction.Restriction.BanII.results
Bio.Restriction.Restriction.BarI.results
Bio.Restriction.Restriction.BasI.results
Bio.Restriction.Restriction.BauI.results
Bio.Restriction.Restriction.Bbr7I.results
Bio.Restriction.Restriction.BbrPI.results
Bio.Restriction.Restriction.BbsI.results
Bio.Restriction.Restriction.Bbv12I.results
Bio.Restriction.Restriction.BbvCI.results
Bio.Restriction.Restriction.BbvI.results
Bio.Restriction.Restriction.BbvII.results
Bio.Restriction.Restriction.BccI.results
Bio.Restriction.Restriction.Bce83I.results
Bio.Restriction.Restriction.BceAI.results
Bio.Restriction.Restriction.BcefI.results
Bio.Restriction.Restriction.BcgI.results
Bio.Restriction.Restriction.BciT130I.results
Bio.Restriction.Restriction.BciVI.results
Bio.Restriction.Restriction.BclI.results
Bio.Restriction.Restriction.BcnI.results
Bio.Restriction.Restriction.BcoDI.results
Bio.Restriction.Restriction.BcuI.results
Bio.Restriction.Restriction.BdaI.results
Bio.Restriction.Restriction.BetI.results
Bio.Restriction.Restriction.BfaI.results
Bio.Restriction.Restriction.BfiI.results
Bio.Restriction.Restriction.BfmI.results
Bio.Restriction.Restriction.BfoI.results
Bio.Restriction.Restriction.BfrI.results
Bio.Restriction.Restriction.BfuAI.results
Bio.Restriction.Restriction.BfuCI.results
Bio.Restriction.Restriction.BfuI.results
Bio.Restriction.Restriction.BglI.results
Bio.Restriction.Restriction.BglII.results
Bio.Restriction.Restriction.BinI.results
Bio.Restriction.Restriction.BisI.results
Bio.Restriction.Restriction.BlnI.results
Bio.Restriction.Restriction.BlpI.results
Bio.Restriction.Restriction.BlsI.results
Bio.Restriction.Restriction.BmcAI.results
Bio.Restriction.Restriction.Bme1390I.results
Bio.Restriction.Restriction.Bme18I.results
Bio.Restriction.Restriction.BmeDI.results
Bio.Restriction.Restriction.BmeRI.results
Bio.Restriction.Restriction.BmeT110I.results
Bio.Restriction.Restriction.BmgBI.results
Bio.Restriction.Restriction.BmgI.results
Bio.Restriction.Restriction.BmgT120I.results
Bio.Restriction.Restriction.BmiI.results
Bio.Restriction.Restriction.BmrFI.results
Bio.Restriction.Restriction.BmrI.results
Bio.Restriction.Restriction.BmsI.results
Bio.Restriction.Restriction.BmtI.results
Bio.Restriction.Restriction.BmuI.results
Bio.Restriction.Restriction.BoxI.results
Bio.Restriction.Restriction.BpiI.results
Bio.Restriction.Restriction.BplI.results
Bio.Restriction.Restriction.BpmI.results
Bio.Restriction.Restriction.Bpu10I.results
Bio.Restriction.Restriction.Bpu1102I.results
Bio.Restriction.Restriction.Bpu14I.results
Bio.Restriction.Restriction.BpuEI.results
Bio.Restriction.Restriction.BpuMI.results
Bio.Restriction.Restriction.BpvUI.results
Bio.Restriction.Restriction.Bsa29I.results
Bio.Restriction.Restriction.BsaAI.results
Bio.Restriction.Restriction.BsaBI.results
Bio.Restriction.Restriction.BsaHI.results
Bio.Restriction.Restriction.BsaI.results
Bio.Restriction.Restriction.BsaJI.results
Bio.Restriction.Restriction.BsaWI.results
Bio.Restriction.Restriction.BsaXI.results
Bio.Restriction.Restriction.BsbI.results
Bio.Restriction.Restriction.Bsc4I.results
Bio.Restriction.Restriction.BscAI.results
Bio.Restriction.Restriction.BscGI.results
Bio.Restriction.Restriction.Bse118I.results
Bio.Restriction.Restriction.Bse1I.results
Bio.Restriction.Restriction.Bse21I.results
Bio.Restriction.Restriction.Bse3DI.results
Bio.Restriction.Restriction.Bse8I.results
Bio.Restriction.Restriction.BseAI.results
Bio.Restriction.Restriction.BseBI.results
Bio.Restriction.Restriction.BseCI.results
Bio.Restriction.Restriction.BseDI.results
Bio.Restriction.Restriction.BseGI.results
Bio.Restriction.Restriction.BseJI.results
Bio.Restriction.Restriction.BseLI.results
Bio.Restriction.Restriction.BseMI.results
Bio.Restriction.Restriction.BseMII.results
Bio.Restriction.Restriction.BseNI.results
Bio.Restriction.Restriction.BsePI.results
Bio.Restriction.Restriction.BseRI.results
Bio.Restriction.Restriction.BseSI.results
Bio.Restriction.Restriction.BseX3I.results
Bio.Restriction.Restriction.BseXI.results
Bio.Restriction.Restriction.BseYI.results
Bio.Restriction.Restriction.BsgI.results
Bio.Restriction.Restriction.Bsh1236I.results
Bio.Restriction.Restriction.Bsh1285I.results
Bio.Restriction.Restriction.BshFI.results
Bio.Restriction.Restriction.BshNI.results
Bio.Restriction.Restriction.BshTI.results
Bio.Restriction.Restriction.BshVI.results
Bio.Restriction.Restriction.BsiEI.results
Bio.Restriction.Restriction.BsiHKAI.results
Bio.Restriction.Restriction.BsiHKCI.results
Bio.Restriction.Restriction.BsiI.results
Bio.Restriction.Restriction.BsiSI.results
Bio.Restriction.Restriction.BsiWI.results
Bio.Restriction.Restriction.BsiYI.results
Bio.Restriction.Restriction.BslFI.results
Bio.Restriction.Restriction.BslI.results
Bio.Restriction.Restriction.BsmAI.results
Bio.Restriction.Restriction.BsmBI.results
Bio.Restriction.Restriction.BsmFI.results
Bio.Restriction.Restriction.BsmI.results
Bio.Restriction.Restriction.BsnI.results
Bio.Restriction.Restriction.Bso31I.results
Bio.Restriction.Restriction.BsoBI.results
Bio.Restriction.Restriction.Bsp119I.results
Bio.Restriction.Restriction.Bsp120I.results
Bio.Restriction.Restriction.Bsp1286I.results
Bio.Restriction.Restriction.Bsp13I.results
Bio.Restriction.Restriction.Bsp1407I.results
Bio.Restriction.Restriction.Bsp143I.results
Bio.Restriction.Restriction.Bsp1720I.results
Bio.Restriction.Restriction.Bsp19I.results
Bio.Restriction.Restriction.Bsp24I.results
Bio.Restriction.Restriction.Bsp68I.results
Bio.Restriction.Restriction.BspACI.results
Bio.Restriction.Restriction.BspCNI.results
Bio.Restriction.Restriction.BspD6I.results
Bio.Restriction.Restriction.BspDI.results
Bio.Restriction.Restriction.BspEI.results
Bio.Restriction.Restriction.BspFNI.results
Bio.Restriction.Restriction.BspGI.results
Bio.Restriction.Restriction.BspHI.results
Bio.Restriction.Restriction.BspLI.results
Bio.Restriction.Restriction.BspLU11I.results
Bio.Restriction.Restriction.BspMI.results
Bio.Restriction.Restriction.BspMII.results
Bio.Restriction.Restriction.BspNCI.results
Bio.Restriction.Restriction.BspOI.results
Bio.Restriction.Restriction.BspPI.results
Bio.Restriction.Restriction.BspQI.results
Bio.Restriction.Restriction.BspT104I.results
Bio.Restriction.Restriction.BspT107I.results
Bio.Restriction.Restriction.BspTI.results
Bio.Restriction.Restriction.BsrBI.results
Bio.Restriction.Restriction.BsrDI.results
Bio.Restriction.Restriction.BsrFI.results
Bio.Restriction.Restriction.BsrGI.results
Bio.Restriction.Restriction.BsrI.results
Bio.Restriction.Restriction.BsrSI.results
Bio.Restriction.Restriction.BssAI.results
Bio.Restriction.Restriction.BssECI.results
Bio.Restriction.Restriction.BssHII.results
Bio.Restriction.Restriction.BssKI.results
Bio.Restriction.Restriction.BssMI.results
Bio.Restriction.Restriction.BssNAI.results
Bio.Restriction.Restriction.BssNI.results
Bio.Restriction.Restriction.BssSI.results
Bio.Restriction.Restriction.BssT1I.results
Bio.Restriction.Restriction.Bst1107I.results
Bio.Restriction.Restriction.Bst2BI.results
Bio.Restriction.Restriction.Bst2UI.results
Bio.Restriction.Restriction.Bst4CI.results
Bio.Restriction.Restriction.Bst6I.results
Bio.Restriction.Restriction.BstACI.results
Bio.Restriction.Restriction.BstAFI.results
Bio.Restriction.Restriction.BstAPI.results
Bio.Restriction.Restriction.BstAUI.results
Bio.Restriction.Restriction.BstBAI.results
Bio.Restriction.Restriction.BstBI.results
Bio.Restriction.Restriction.BstC8I.results
Bio.Restriction.Restriction.BstDEI.results
Bio.Restriction.Restriction.BstDSI.results
Bio.Restriction.Restriction.BstEII.results
Bio.Restriction.Restriction.BstENI.results
Bio.Restriction.Restriction.BstF5I.results
Bio.Restriction.Restriction.BstFNI.results
Bio.Restriction.Restriction.BstH2I.results
Bio.Restriction.Restriction.BstHHI.results
Bio.Restriction.Restriction.BstKTI.results
Bio.Restriction.Restriction.BstMAI.results
Bio.Restriction.Restriction.BstMBI.results
Bio.Restriction.Restriction.BstMCI.results
Bio.Restriction.Restriction.BstMWI.results
Bio.Restriction.Restriction.BstNI.results
Bio.Restriction.Restriction.BstNSI.results
Bio.Restriction.Restriction.BstOI.results
Bio.Restriction.Restriction.BstPAI.results
Bio.Restriction.Restriction.BstPI.results
Bio.Restriction.Restriction.BstSCI.results
Bio.Restriction.Restriction.BstSFI.results
Bio.Restriction.Restriction.BstSLI.results
Bio.Restriction.Restriction.BstSNI.results
Bio.Restriction.Restriction.BstUI.results
Bio.Restriction.Restriction.BstV1I.results
Bio.Restriction.Restriction.BstV2I.results
Bio.Restriction.Restriction.BstX2I.results
Bio.Restriction.Restriction.BstXI.results
Bio.Restriction.Restriction.BstYI.results
Bio.Restriction.Restriction.BstZ17I.results
Bio.Restriction.Restriction.BstZI.results
Bio.Restriction.Restriction.Bsu15I.results
Bio.Restriction.Restriction.Bsu36I.results
Bio.Restriction.Restriction.BsuI.results
Bio.Restriction.Restriction.BsuRI.results
Bio.Restriction.Restriction.BtgI.results
Bio.Restriction.Restriction.BtgZI.results
Bio.Restriction.Restriction.BthCI.results
Bio.Restriction.Restriction.BtrI.results
Bio.Restriction.Restriction.BtsCI.results
Bio.Restriction.Restriction.BtsI.results
Bio.Restriction.Restriction.BtsIMutI.results
Bio.Restriction.Restriction.BtuMI.results
Bio.Restriction.Restriction.BveI.results
Bio.Restriction.Restriction.Cac8I.results
Bio.Restriction.Restriction.CaiI.results
Bio.Restriction.Restriction.CauII.results
Bio.Restriction.Restriction.CchII.results
Bio.Restriction.Restriction.CchIII.results
Bio.Restriction.Restriction.CciI.results
Bio.Restriction.Restriction.CciNI.results
Bio.Restriction.Restriction.Cdi630V.results
Bio.Restriction.Restriction.CdiI.results
Bio.Restriction.Restriction.CdpI.results
Bio.Restriction.Restriction.CfoI.results
Bio.Restriction.Restriction.Cfr10I.results
Bio.Restriction.Restriction.Cfr13I.results
Bio.Restriction.Restriction.Cfr42I.results
Bio.Restriction.Restriction.Cfr9I.results
Bio.Restriction.Restriction.CfrI.results
Bio.Restriction.Restriction.Cgl13032I.results
Bio.Restriction.Restriction.Cgl13032II.results
Bio.Restriction.Restriction.ChaI.results
Bio.Restriction.Restriction.CjeFIII.results
Bio.Restriction.Restriction.CjeFV.results
Bio.Restriction.Restriction.CjeI.results
Bio.Restriction.Restriction.CjeNII.results
Bio.Restriction.Restriction.CjeNIII.results
Bio.Restriction.Restriction.CjeP659IV.results
Bio.Restriction.Restriction.CjePI.results
Bio.Restriction.Restriction.CjuI.results
Bio.Restriction.Restriction.CjuII.results
Bio.Restriction.Restriction.ClaI.results
Bio.Restriction.Restriction.CpoI.results
Bio.Restriction.Restriction.CseI.results
Bio.Restriction.Restriction.CsiI.results
Bio.Restriction.Restriction.Csp6I.results
Bio.Restriction.Restriction.CspAI.results
Bio.Restriction.Restriction.CspCI.results
Bio.Restriction.Restriction.CspI.results
Bio.Restriction.Restriction.CstMI.results
Bio.Restriction.Restriction.CviAII.results
Bio.Restriction.Restriction.CviJI.results
Bio.Restriction.Restriction.CviKI_1.results
Bio.Restriction.Restriction.CviQI.results
Bio.Restriction.Restriction.CviRI.results
Bio.Restriction.Restriction.DdeI.results
Bio.Restriction.Restriction.DinI.results
Bio.Restriction.Restriction.DpnI.results
Bio.Restriction.Restriction.DpnII.results
Bio.Restriction.Restriction.DraI.results
Bio.Restriction.Restriction.DraII.results
Bio.Restriction.Restriction.DraIII.results
Bio.Restriction.Restriction.DraRI.results
Bio.Restriction.Restriction.DrdI.results
Bio.Restriction.Restriction.DrdII.results
Bio.Restriction.Restriction.DriI.results
Bio.Restriction.Restriction.DsaI.results
Bio.Restriction.Restriction.DseDI.results
Bio.Restriction.Restriction.EaeI.results
Bio.Restriction.Restriction.EagI.results
Bio.Restriction.Restriction.Eam1104I.results
Bio.Restriction.Restriction.Eam1105I.results
Bio.Restriction.Restriction.EarI.results
Bio.Restriction.Restriction.EciI.results
Bio.Restriction.Restriction.Ecl136II.results
Bio.Restriction.Restriction.EclXI.results
Bio.Restriction.Restriction.Eco105I.results
Bio.Restriction.Restriction.Eco130I.results
Bio.Restriction.Restriction.Eco147I.results
Bio.Restriction.Restriction.Eco24I.results
Bio.Restriction.Restriction.Eco31I.results
Bio.Restriction.Restriction.Eco32I.results
Bio.Restriction.Restriction.Eco47I.results
Bio.Restriction.Restriction.Eco47III.results
Bio.Restriction.Restriction.Eco52I.results
Bio.Restriction.Restriction.Eco53kI.results
Bio.Restriction.Restriction.Eco57I.results
Bio.Restriction.Restriction.Eco57MI.results
Bio.Restriction.Restriction.Eco72I.results
Bio.Restriction.Restriction.Eco81I.results
Bio.Restriction.Restriction.Eco88I.results
Bio.Restriction.Restriction.Eco91I.results
Bio.Restriction.Restriction.EcoHI.results
Bio.Restriction.Restriction.EcoICRI.results
Bio.Restriction.Restriction.EcoNI.results
Bio.Restriction.Restriction.EcoO109I.results
Bio.Restriction.Restriction.EcoO65I.results
Bio.Restriction.Restriction.EcoRI.results
Bio.Restriction.Restriction.EcoRII.results
Bio.Restriction.Restriction.EcoRV.results
Bio.Restriction.Restriction.EcoT14I.results
Bio.Restriction.Restriction.EcoT22I.results
Bio.Restriction.Restriction.EcoT38I.results
Bio.Restriction.Restriction.EgeI.results
Bio.Restriction.Restriction.EheI.results
Bio.Restriction.Restriction.ErhI.results
Bio.Restriction.Restriction.EsaBC3I.results
Bio.Restriction.Restriction.EsaSSI.results
Bio.Restriction.Restriction.Esp3I.results
Bio.Restriction.Restriction.EspI.results
Bio.Restriction.Restriction.FaeI.results
Bio.Restriction.Restriction.FaiI.results
Bio.Restriction.Restriction.FalI.results
Bio.Restriction.Restriction.FaqI.results
Bio.Restriction.Restriction.FatI.results
Bio.Restriction.Restriction.FauI.results
Bio.Restriction.Restriction.FauNDI.results
Bio.Restriction.Restriction.FbaI.results
Bio.Restriction.Restriction.FblI.results
Bio.Restriction.Restriction.FinI.results
Bio.Restriction.Restriction.FmuI.results
Bio.Restriction.Restriction.Fnu4HI.results
Bio.Restriction.Restriction.FnuDII.results
Bio.Restriction.Restriction.FokI.results
Bio.Restriction.Restriction.FriOI.results
Bio.Restriction.Restriction.FseI.results
Bio.Restriction.Restriction.Fsp4HI.results
Bio.Restriction.Restriction.FspAI.results
Bio.Restriction.Restriction.FspBI.results
Bio.Restriction.Restriction.FspEI.results
Bio.Restriction.Restriction.FspI.results
Bio.Restriction.Restriction.GauT27I.results
Bio.Restriction.Restriction.GdiII.results
Bio.Restriction.Restriction.GlaI.results
Bio.Restriction.Restriction.GluI.results
Bio.Restriction.Restriction.GsaI.results
Bio.Restriction.Restriction.GsuI.results
Bio.Restriction.Restriction.HaeI.results
Bio.Restriction.Restriction.HaeII.results
Bio.Restriction.Restriction.HaeIII.results
Bio.Restriction.Restriction.HapII.results
Bio.Restriction.Restriction.HauII.results
Bio.Restriction.Restriction.HgaI.results
Bio.Restriction.Restriction.HgiAI.results
Bio.Restriction.Restriction.HgiCI.results
Bio.Restriction.Restriction.HgiEII.results
Bio.Restriction.Restriction.HgiJII.results
Bio.Restriction.Restriction.HhaI.results
Bio.Restriction.Restriction.Hin1I.results
Bio.Restriction.Restriction.Hin1II.results
Bio.Restriction.Restriction.Hin4I.results
Bio.Restriction.Restriction.Hin4II.results
Bio.Restriction.Restriction.Hin6I.results
Bio.Restriction.Restriction.HinP1I.results
Bio.Restriction.Restriction.HincII.results
Bio.Restriction.Restriction.HindII.results
Bio.Restriction.Restriction.HindIII.results
Bio.Restriction.Restriction.HinfI.results
Bio.Restriction.Restriction.HpaI.results
Bio.Restriction.Restriction.HpaII.results
Bio.Restriction.Restriction.HphI.results
Bio.Restriction.Restriction.Hpy166II.results
Bio.Restriction.Restriction.Hpy178III.results
Bio.Restriction.Restriction.Hpy188I.results
Bio.Restriction.Restriction.Hpy188III.results
Bio.Restriction.Restriction.Hpy8I.results
Bio.Restriction.Restriction.Hpy99I.results
Bio.Restriction.Restriction.Hpy99XIII.results
Bio.Restriction.Restriction.Hpy99XIV.results
Bio.Restriction.Restriction.HpyAV.results
Bio.Restriction.Restriction.HpyCH4III.results
Bio.Restriction.Restriction.HpyCH4IV.results
Bio.Restriction.Restriction.HpyCH4V.results
Bio.Restriction.Restriction.HpyF10VI.results
Bio.Restriction.Restriction.HpyF3I.results
Bio.Restriction.Restriction.HpySE526I.results
Bio.Restriction.Restriction.Hsp92I.results
Bio.Restriction.Restriction.Hsp92II.results
Bio.Restriction.Restriction.HspAI.results
Bio.Restriction.Restriction.Jma19592I.results
Bio.Restriction.Restriction.KasI.results
Bio.Restriction.Restriction.KflI.results
Bio.Restriction.Restriction.Kpn2I.results
Bio.Restriction.Restriction.KpnI.results
Bio.Restriction.Restriction.KroI.results
Bio.Restriction.Restriction.Ksp22I.results
Bio.Restriction.Restriction.Ksp632I.results
Bio.Restriction.Restriction.KspAI.results
Bio.Restriction.Restriction.KspI.results
Bio.Restriction.Restriction.Kzo9I.results
Bio.Restriction.Restriction.LguI.results
Bio.Restriction.Restriction.LpnI.results
Bio.Restriction.Restriction.LpnPI.results
Bio.Restriction.Restriction.Lsp1109I.results
Bio.Restriction.Restriction.LweI.results
Bio.Restriction.Restriction.MabI.results
Bio.Restriction.Restriction.MaeI.results
Bio.Restriction.Restriction.MaeII.results
Bio.Restriction.Restriction.MaeIII.results
Bio.Restriction.Restriction.MalI.results
Bio.Restriction.Restriction.MaqI.results
Bio.Restriction.Restriction.MauBI.results
Bio.Restriction.Restriction.MbiI.results
Bio.Restriction.Restriction.MboI.results
Bio.Restriction.Restriction.MboII.results
Bio.Restriction.Restriction.McaTI.results
Bio.Restriction.Restriction.McrI.results
Bio.Restriction.Restriction.MfeI.results
Bio.Restriction.Restriction.MflI.results
Bio.Restriction.Restriction.MhlI.results
Bio.Restriction.Restriction.MjaIV.results
Bio.Restriction.Restriction.MkaDII.results
Bio.Restriction.Restriction.MlsI.results
Bio.Restriction.Restriction.MluCI.results
Bio.Restriction.Restriction.MluI.results
Bio.Restriction.Restriction.MluNI.results
Bio.Restriction.Restriction.Mly113I.results
Bio.Restriction.Restriction.MlyI.results
Bio.Restriction.Restriction.MmeI.results
Bio.Restriction.Restriction.MnlI.results
Bio.Restriction.Restriction.Mph1103I.results
Bio.Restriction.Restriction.MreI.results
Bio.Restriction.Restriction.MroI.results
Bio.Restriction.Restriction.MroNI.results
Bio.Restriction.Restriction.MroXI.results
Bio.Restriction.Restriction.MscI.results
Bio.Restriction.Restriction.MseI.results
Bio.Restriction.Restriction.MslI.results
Bio.Restriction.Restriction.Msp20I.results
Bio.Restriction.Restriction.MspA1I.results
Bio.Restriction.Restriction.MspCI.results
Bio.Restriction.Restriction.MspI.results
Bio.Restriction.Restriction.MspJI.results
Bio.Restriction.Restriction.MspR9I.results
Bio.Restriction.Restriction.MssI.results
Bio.Restriction.Restriction.MstI.results
Bio.Restriction.Restriction.MunI.results
Bio.Restriction.Restriction.Mva1269I.results
Bio.Restriction.Restriction.MvaI.results
Bio.Restriction.Restriction.MvnI.results
Bio.Restriction.Restriction.MvrI.results
Bio.Restriction.Restriction.MwoI.results
Bio.Restriction.Restriction.NaeI.results
Bio.Restriction.Restriction.NarI.results
Bio.Restriction.Restriction.NciI.results
Bio.Restriction.Restriction.NcoI.results
Bio.Restriction.Restriction.NdeI.results
Bio.Restriction.Restriction.NdeII.results
Bio.Restriction.Restriction.NgoAVIII.results
Bio.Restriction.Restriction.NgoMIV.results
Bio.Restriction.Restriction.NhaXI.results
Bio.Restriction.Restriction.NheI.results
Bio.Restriction.Restriction.NlaCI.results
Bio.Restriction.Restriction.NlaIII.results
Bio.Restriction.Restriction.NlaIV.results
Bio.Restriction.Restriction.Nli3877I.results
Bio.Restriction.Restriction.NmeAIII.results
Bio.Restriction.Restriction.NmeDI.results
Bio.Restriction.Restriction.NmuCI.results
Bio.Restriction.Restriction.NotI.results
Bio.Restriction.Restriction.NruI.results
Bio.Restriction.Restriction.NsbI.results
Bio.Restriction.Restriction.NsiI.results
Bio.Restriction.Restriction.NspBII.results
Bio.Restriction.Restriction.NspI.results
Bio.Restriction.Restriction.NspV.results
Bio.Restriction.Restriction.OliI.results
Bio.Restriction.Restriction.PabI.results
Bio.Restriction.Restriction.PacI.results
Bio.Restriction.Restriction.PaeI.results
Bio.Restriction.Restriction.PaeR7I.results
Bio.Restriction.Restriction.PagI.results
Bio.Restriction.Restriction.PalAI.results
Bio.Restriction.Restriction.PasI.results
Bio.Restriction.Restriction.PauI.results
Bio.Restriction.Restriction.PceI.results
Bio.Restriction.Restriction.PciI.results
Bio.Restriction.Restriction.PciSI.results
Bio.Restriction.Restriction.PcsI.results
Bio.Restriction.Restriction.PctI.results
Bio.Restriction.Restriction.PdiI.results
Bio.Restriction.Restriction.PdmI.results
Bio.Restriction.Restriction.PenI.results
Bio.Restriction.Restriction.PfeI.results
Bio.Restriction.Restriction.Pfl1108I.results
Bio.Restriction.Restriction.Pfl23II.results
Bio.Restriction.Restriction.PflFI.results
Bio.Restriction.Restriction.PflMI.results
Bio.Restriction.Restriction.PfoI.results
Bio.Restriction.Restriction.PinAI.results
Bio.Restriction.Restriction.PlaDI.results
Bio.Restriction.Restriction.Ple19I.results
Bio.Restriction.Restriction.PleI.results
Bio.Restriction.Restriction.PluTI.results
Bio.Restriction.Restriction.PmaCI.results
Bio.Restriction.Restriction.PmeI.results
Bio.Restriction.Restriction.PmlI.results
Bio.Restriction.Restriction.PpiI.results
Bio.Restriction.Restriction.PpsI.results
Bio.Restriction.Restriction.Ppu10I.results
Bio.Restriction.Restriction.Ppu21I.results
Bio.Restriction.Restriction.PpuMI.results
Bio.Restriction.Restriction.PscI.results
Bio.Restriction.Restriction.PshAI.results
Bio.Restriction.Restriction.PshBI.results
Bio.Restriction.Restriction.PsiI.results
Bio.Restriction.Restriction.Psp03I.results
Bio.Restriction.Restriction.Psp124BI.results
Bio.Restriction.Restriction.Psp1406I.results
Bio.Restriction.Restriction.Psp5II.results
Bio.Restriction.Restriction.Psp6I.results
Bio.Restriction.Restriction.PspCI.results
Bio.Restriction.Restriction.PspEI.results
Bio.Restriction.Restriction.PspGI.results
Bio.Restriction.Restriction.PspLI.results
Bio.Restriction.Restriction.PspN4I.results
Bio.Restriction.Restriction.PspOMI.results
Bio.Restriction.Restriction.PspOMII.results
Bio.Restriction.Restriction.PspPI.results
Bio.Restriction.Restriction.PspPPI.results
Bio.Restriction.Restriction.PspPRI.results
Bio.Restriction.Restriction.PspXI.results
Bio.Restriction.Restriction.PsrI.results
Bio.Restriction.Restriction.PssI.results
Bio.Restriction.Restriction.PstI.results
Bio.Restriction.Restriction.PstNI.results
Bio.Restriction.Restriction.PsuI.results
Bio.Restriction.Restriction.PsyI.results
Bio.Restriction.Restriction.PteI.results
Bio.Restriction.Restriction.PvuI.results
Bio.Restriction.Restriction.PvuII.results
Bio.Restriction.Restriction.R2_BceSIV.results
Bio.Restriction.Restriction.RceI.results
Bio.Restriction.Restriction.RdeGBI.results
Bio.Restriction.Restriction.RdeGBII.results
Bio.Restriction.Restriction.RdeGBIII.results
Bio.Restriction.Restriction.RflFIII.results
Bio.Restriction.Restriction.RgaI.results
Bio.Restriction.Restriction.RigI.results
Bio.Restriction.Restriction.RlaI.results
Bio.Restriction.Restriction.RleAI.results
Bio.Restriction.Restriction.RpaB5I.results
Bio.Restriction.Restriction.RpaBI.results
Bio.Restriction.Restriction.RpaI.results
Bio.Restriction.Restriction.RpaTI.results
Bio.Restriction.Restriction.RruI.results
Bio.Restriction.Restriction.RsaI.results
Bio.Restriction.Restriction.RsaNI.results
Bio.Restriction.Restriction.RseI.results
Bio.Restriction.Restriction.Rsr2I.results
Bio.Restriction.Restriction.RsrII.results
Bio.Restriction.Restriction.SacI.results
Bio.Restriction.Restriction.SacII.results
Bio.Restriction.Restriction.SalI.results
Bio.Restriction.Restriction.SanDI.results
Bio.Restriction.Restriction.SapI.results
Bio.Restriction.Restriction.SaqAI.results
Bio.Restriction.Restriction.SatI.results
Bio.Restriction.Restriction.Sau3AI.results
Bio.Restriction.Restriction.Sau96I.results
Bio.Restriction.Restriction.SauI.results
Bio.Restriction.Restriction.SbfI.results
Bio.Restriction.Restriction.ScaI.results
Bio.Restriction.Restriction.SchI.results
Bio.Restriction.Restriction.SciI.results
Bio.Restriction.Restriction.ScrFI.results
Bio.Restriction.Restriction.SdaI.results
Bio.Restriction.Restriction.SdeAI.results
Bio.Restriction.Restriction.SdeOSI.results
Bio.Restriction.Restriction.SduI.results
Bio.Restriction.Restriction.SecI.results
Bio.Restriction.Restriction.SelI.results
Bio.Restriction.Restriction.SetI.results
Bio.Restriction.Restriction.SexAI.results
Bio.Restriction.Restriction.SfaAI.results
Bio.Restriction.Restriction.SfaNI.results
Bio.Restriction.Restriction.SfcI.results
Bio.Restriction.Restriction.SfeI.results
Bio.Restriction.Restriction.SfiI.results
Bio.Restriction.Restriction.SfoI.results
Bio.Restriction.Restriction.Sfr274I.results
Bio.Restriction.Restriction.Sfr303I.results
Bio.Restriction.Restriction.SfuI.results
Bio.Restriction.Restriction.SgeI.results
Bio.Restriction.Restriction.SgfI.results
Bio.Restriction.Restriction.SgrAI.results
Bio.Restriction.Restriction.SgrBI.results
Bio.Restriction.Restriction.SgrDI.results
Bio.Restriction.Restriction.SgrTI.results
Bio.Restriction.Restriction.SgsI.results
Bio.Restriction.Restriction.SimI.results
Bio.Restriction.Restriction.SlaI.results
Bio.Restriction.Restriction.SmaI.results
Bio.Restriction.Restriction.SmiI.results
Bio.Restriction.Restriction.SmiMI.results
Bio.Restriction.Restriction.SmlI.results
Bio.Restriction.Restriction.SmoI.results
Bio.Restriction.Restriction.SnaBI.results
Bio.Restriction.Restriction.SnaI.results
Bio.Restriction.Restriction.Sno506I.results
Bio.Restriction.Restriction.SpeI.results
Bio.Restriction.Restriction.SphI.results
Bio.Restriction.Restriction.SplI.results
Bio.Restriction.Restriction.SpoDI.results
Bio.Restriction.Restriction.SrfI.results
Bio.Restriction.Restriction.Sse232I.results
Bio.Restriction.Restriction.Sse8387I.results
Bio.Restriction.Restriction.Sse8647I.results
Bio.Restriction.Restriction.Sse9I.results
Bio.Restriction.Restriction.SseBI.results
Bio.Restriction.Restriction.SsiI.results
Bio.Restriction.Restriction.SspD5I.results
Bio.Restriction.Restriction.SspDI.results
Bio.Restriction.Restriction.SspI.results
Bio.Restriction.Restriction.SstE37I.results
Bio.Restriction.Restriction.SstI.results
Bio.Restriction.Restriction.Sth132I.results
Bio.Restriction.Restriction.Sth302II.results
Bio.Restriction.Restriction.StrI.results
Bio.Restriction.Restriction.StsI.results
Bio.Restriction.Restriction.StuI.results
Bio.Restriction.Restriction.StyD4I.results
Bio.Restriction.Restriction.StyI.results
Bio.Restriction.Restriction.SwaI.results
Bio.Restriction.Restriction.TaaI.results
Bio.Restriction.Restriction.TaiI.results
Bio.Restriction.Restriction.TaqI.results
Bio.Restriction.Restriction.TaqII.results
Bio.Restriction.Restriction.TasI.results
Bio.Restriction.Restriction.TatI.results
Bio.Restriction.Restriction.TauI.results
Bio.Restriction.Restriction.TfiI.results
Bio.Restriction.Restriction.Tru1I.results
Bio.Restriction.Restriction.Tru9I.results
Bio.Restriction.Restriction.TscAI.results
Bio.Restriction.Restriction.TseFI.results
Bio.Restriction.Restriction.TseI.results
Bio.Restriction.Restriction.TsoI.results
Bio.Restriction.Restriction.Tsp45I.results
Bio.Restriction.Restriction.Tsp4CI.results
Bio.Restriction.Restriction.TspDTI.results
Bio.Restriction.Restriction.TspEI.results
Bio.Restriction.Restriction.TspGWI.results
Bio.Restriction.Restriction.TspMI.results
Bio.Restriction.Restriction.TspRI.results
Bio.Restriction.Restriction.TssI.results
Bio.Restriction.Restriction.TstI.results
Bio.Restriction.Restriction.TsuI.results
Bio.Restriction.Restriction.Tth111I.results
Bio.Restriction.Restriction.Tth111II.results
Bio.Restriction.Restriction.UbaF11I.results
Bio.Restriction.Restriction.UbaF12I.results
Bio.Restriction.Restriction.UbaF13I.results
Bio.Restriction.Restriction.UbaF14I.results
Bio.Restriction.Restriction.UbaF9I.results
Bio.Restriction.Restriction.UbaPI.results
Bio.Restriction.Restriction.UcoMSI.results
Bio.Restriction.Restriction.UnbI.results
Bio.Restriction.Restriction.Van91I.results
Bio.Restriction.Restriction.Vha464I.results
Bio.Restriction.Restriction.VneI.results
Bio.Restriction.Restriction.VpaK11AI.results
Bio.Restriction.Restriction.VpaK11BI.results
Bio.Restriction.Restriction.VspI.results
Bio.Restriction.Restriction.WviI.results
Bio.Restriction.Restriction.XagI.results
Bio.Restriction.Restriction.XapI.results
Bio.Restriction.Restriction.XbaI.results
Bio.Restriction.Restriction.XceI.results
Bio.Restriction.Restriction.XcmI.results
Bio.Restriction.Restriction.XhoI.results
Bio.Restriction.Restriction.XhoII.results
Bio.Restriction.Restriction.XmaI.results
Bio.Restriction.Restriction.XmaIII.results
Bio.Restriction.Restriction.XmaJI.results
Bio.Restriction.Restriction.XmiI.results
Bio.Restriction.Restriction.XmnI.results
Bio.Restriction.Restriction.XspI.results
Bio.Restriction.Restriction.YkrI.results
Bio.Restriction.Restriction.ZraI.results
Bio.Restriction.Restriction.ZrmI.results
Bio.Restriction.Restriction.Zsp2I.results" class="py-name" href="#" onclick="return doclink('link-997', 'results', 'link-996');">results</a></tt><tt class="py-op">.</tt><tt id="link-998" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-998', 'append', 'link-506');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">cols</tt><tt class="py-op">[</tt><tt class="py-name">c</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L1709"></a><tt class="py-lineno">1709</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">tuple</tt><tt class="py-op">(</tt><tt id="link-999" class="py-name"><a title="Bio.Restriction.Restriction.AanI.results
Bio.Restriction.Restriction.AarI.results
Bio.Restriction.Restriction.AasI.results
Bio.Restriction.Restriction.AatII.results
Bio.Restriction.Restriction.AbaSI.results
Bio.Restriction.Restriction.AbsI.results
Bio.Restriction.Restriction.Acc16I.results
Bio.Restriction.Restriction.Acc36I.results
Bio.Restriction.Restriction.Acc65I.results
Bio.Restriction.Restriction.AccB1I.results
Bio.Restriction.Restriction.AccB7I.results
Bio.Restriction.Restriction.AccBSI.results
Bio.Restriction.Restriction.AccI.results
Bio.Restriction.Restriction.AccII.results
Bio.Restriction.Restriction.AccIII.results
Bio.Restriction.Restriction.AceIII.results
Bio.Restriction.Restriction.AciI.results
Bio.Restriction.Restriction.AclI.results
Bio.Restriction.Restriction.AclWI.results
Bio.Restriction.Restriction.AcoI.results
Bio.Restriction.Restriction.AcsI.results
Bio.Restriction.Restriction.AcuI.results
Bio.Restriction.Restriction.AcvI.results
Bio.Restriction.Restriction.AcyI.results
Bio.Restriction.Restriction.AdeI.results
Bio.Restriction.Restriction.AfaI.results
Bio.Restriction.Restriction.AfeI.results
Bio.Restriction.Restriction.AfiI.results
Bio.Restriction.Restriction.AflII.results
Bio.Restriction.Restriction.AflIII.results
Bio.Restriction.Restriction.AgeI.results
Bio.Restriction.Restriction.AgsI.results
Bio.Restriction.Restriction.AhaIII.results
Bio.Restriction.Restriction.AhdI.results
Bio.Restriction.Restriction.AhlI.results
Bio.Restriction.Restriction.AjiI.results
Bio.Restriction.Restriction.AjnI.results
Bio.Restriction.Restriction.AjuI.results
Bio.Restriction.Restriction.AleI.results
Bio.Restriction.Restriction.AlfI.results
Bio.Restriction.Restriction.AloI.results
Bio.Restriction.Restriction.AluBI.results
Bio.Restriction.Restriction.AluI.results
Bio.Restriction.Restriction.Alw21I.results
Bio.Restriction.Restriction.Alw26I.results
Bio.Restriction.Restriction.Alw44I.results
Bio.Restriction.Restriction.AlwFI.results
Bio.Restriction.Restriction.AlwI.results
Bio.Restriction.Restriction.AlwNI.results
Bio.Restriction.Restriction.Ama87I.results
Bio.Restriction.Restriction.Aor13HI.results
Bio.Restriction.Restriction.Aor51HI.results
Bio.Restriction.Restriction.AoxI.results
Bio.Restriction.Restriction.ApaBI.results
Bio.Restriction.Restriction.ApaI.results
Bio.Restriction.Restriction.ApaLI.results
Bio.Restriction.Restriction.ApeKI.results
Bio.Restriction.Restriction.ApoI.results
Bio.Restriction.Restriction.ApyPI.results
Bio.Restriction.Restriction.AquII.results
Bio.Restriction.Restriction.AquIII.results
Bio.Restriction.Restriction.AquIV.results
Bio.Restriction.Restriction.ArsI.results
Bio.Restriction.Restriction.AscI.results
Bio.Restriction.Restriction.AseI.results
Bio.Restriction.Restriction.Asi256I.results
Bio.Restriction.Restriction.AsiGI.results
Bio.Restriction.Restriction.AsiSI.results
Bio.Restriction.Restriction.Asp700I.results
Bio.Restriction.Restriction.Asp718I.results
Bio.Restriction.Restriction.AspA2I.results
Bio.Restriction.Restriction.AspBHI.results
Bio.Restriction.Restriction.AspLEI.results
Bio.Restriction.Restriction.AspS9I.results
Bio.Restriction.Restriction.AssI.results
Bio.Restriction.Restriction.AsuC2I.results
Bio.Restriction.Restriction.AsuHPI.results
Bio.Restriction.Restriction.AsuI.results
Bio.Restriction.Restriction.AsuII.results
Bio.Restriction.Restriction.AsuNHI.results
Bio.Restriction.Restriction.AvaI.results
Bio.Restriction.Restriction.AvaII.results
Bio.Restriction.Restriction.AvaIII.results
Bio.Restriction.Restriction.AvrII.results
Bio.Restriction.Restriction.AxyI.results
Bio.Restriction.Restriction.BaeGI.results
Bio.Restriction.Restriction.BaeI.results
Bio.Restriction.Restriction.BalI.results
Bio.Restriction.Restriction.BamHI.results
Bio.Restriction.Restriction.BanI.results
Bio.Restriction.Restriction.BanII.results
Bio.Restriction.Restriction.BarI.results
Bio.Restriction.Restriction.BasI.results
Bio.Restriction.Restriction.BauI.results
Bio.Restriction.Restriction.Bbr7I.results
Bio.Restriction.Restriction.BbrPI.results
Bio.Restriction.Restriction.BbsI.results
Bio.Restriction.Restriction.Bbv12I.results
Bio.Restriction.Restriction.BbvCI.results
Bio.Restriction.Restriction.BbvI.results
Bio.Restriction.Restriction.BbvII.results
Bio.Restriction.Restriction.BccI.results
Bio.Restriction.Restriction.Bce83I.results
Bio.Restriction.Restriction.BceAI.results
Bio.Restriction.Restriction.BcefI.results
Bio.Restriction.Restriction.BcgI.results
Bio.Restriction.Restriction.BciT130I.results
Bio.Restriction.Restriction.BciVI.results
Bio.Restriction.Restriction.BclI.results
Bio.Restriction.Restriction.BcnI.results
Bio.Restriction.Restriction.BcoDI.results
Bio.Restriction.Restriction.BcuI.results
Bio.Restriction.Restriction.BdaI.results
Bio.Restriction.Restriction.BetI.results
Bio.Restriction.Restriction.BfaI.results
Bio.Restriction.Restriction.BfiI.results
Bio.Restriction.Restriction.BfmI.results
Bio.Restriction.Restriction.BfoI.results
Bio.Restriction.Restriction.BfrI.results
Bio.Restriction.Restriction.BfuAI.results
Bio.Restriction.Restriction.BfuCI.results
Bio.Restriction.Restriction.BfuI.results
Bio.Restriction.Restriction.BglI.results
Bio.Restriction.Restriction.BglII.results
Bio.Restriction.Restriction.BinI.results
Bio.Restriction.Restriction.BisI.results
Bio.Restriction.Restriction.BlnI.results
Bio.Restriction.Restriction.BlpI.results
Bio.Restriction.Restriction.BlsI.results
Bio.Restriction.Restriction.BmcAI.results
Bio.Restriction.Restriction.Bme1390I.results
Bio.Restriction.Restriction.Bme18I.results
Bio.Restriction.Restriction.BmeDI.results
Bio.Restriction.Restriction.BmeRI.results
Bio.Restriction.Restriction.BmeT110I.results
Bio.Restriction.Restriction.BmgBI.results
Bio.Restriction.Restriction.BmgI.results
Bio.Restriction.Restriction.BmgT120I.results
Bio.Restriction.Restriction.BmiI.results
Bio.Restriction.Restriction.BmrFI.results
Bio.Restriction.Restriction.BmrI.results
Bio.Restriction.Restriction.BmsI.results
Bio.Restriction.Restriction.BmtI.results
Bio.Restriction.Restriction.BmuI.results
Bio.Restriction.Restriction.BoxI.results
Bio.Restriction.Restriction.BpiI.results
Bio.Restriction.Restriction.BplI.results
Bio.Restriction.Restriction.BpmI.results
Bio.Restriction.Restriction.Bpu10I.results
Bio.Restriction.Restriction.Bpu1102I.results
Bio.Restriction.Restriction.Bpu14I.results
Bio.Restriction.Restriction.BpuEI.results
Bio.Restriction.Restriction.BpuMI.results
Bio.Restriction.Restriction.BpvUI.results
Bio.Restriction.Restriction.Bsa29I.results
Bio.Restriction.Restriction.BsaAI.results
Bio.Restriction.Restriction.BsaBI.results
Bio.Restriction.Restriction.BsaHI.results
Bio.Restriction.Restriction.BsaI.results
Bio.Restriction.Restriction.BsaJI.results
Bio.Restriction.Restriction.BsaWI.results
Bio.Restriction.Restriction.BsaXI.results
Bio.Restriction.Restriction.BsbI.results
Bio.Restriction.Restriction.Bsc4I.results
Bio.Restriction.Restriction.BscAI.results
Bio.Restriction.Restriction.BscGI.results
Bio.Restriction.Restriction.Bse118I.results
Bio.Restriction.Restriction.Bse1I.results
Bio.Restriction.Restriction.Bse21I.results
Bio.Restriction.Restriction.Bse3DI.results
Bio.Restriction.Restriction.Bse8I.results
Bio.Restriction.Restriction.BseAI.results
Bio.Restriction.Restriction.BseBI.results
Bio.Restriction.Restriction.BseCI.results
Bio.Restriction.Restriction.BseDI.results
Bio.Restriction.Restriction.BseGI.results
Bio.Restriction.Restriction.BseJI.results
Bio.Restriction.Restriction.BseLI.results
Bio.Restriction.Restriction.BseMI.results
Bio.Restriction.Restriction.BseMII.results
Bio.Restriction.Restriction.BseNI.results
Bio.Restriction.Restriction.BsePI.results
Bio.Restriction.Restriction.BseRI.results
Bio.Restriction.Restriction.BseSI.results
Bio.Restriction.Restriction.BseX3I.results
Bio.Restriction.Restriction.BseXI.results
Bio.Restriction.Restriction.BseYI.results
Bio.Restriction.Restriction.BsgI.results
Bio.Restriction.Restriction.Bsh1236I.results
Bio.Restriction.Restriction.Bsh1285I.results
Bio.Restriction.Restriction.BshFI.results
Bio.Restriction.Restriction.BshNI.results
Bio.Restriction.Restriction.BshTI.results
Bio.Restriction.Restriction.BshVI.results
Bio.Restriction.Restriction.BsiEI.results
Bio.Restriction.Restriction.BsiHKAI.results
Bio.Restriction.Restriction.BsiHKCI.results
Bio.Restriction.Restriction.BsiI.results
Bio.Restriction.Restriction.BsiSI.results
Bio.Restriction.Restriction.BsiWI.results
Bio.Restriction.Restriction.BsiYI.results
Bio.Restriction.Restriction.BslFI.results
Bio.Restriction.Restriction.BslI.results
Bio.Restriction.Restriction.BsmAI.results
Bio.Restriction.Restriction.BsmBI.results
Bio.Restriction.Restriction.BsmFI.results
Bio.Restriction.Restriction.BsmI.results
Bio.Restriction.Restriction.BsnI.results
Bio.Restriction.Restriction.Bso31I.results
Bio.Restriction.Restriction.BsoBI.results
Bio.Restriction.Restriction.Bsp119I.results
Bio.Restriction.Restriction.Bsp120I.results
Bio.Restriction.Restriction.Bsp1286I.results
Bio.Restriction.Restriction.Bsp13I.results
Bio.Restriction.Restriction.Bsp1407I.results
Bio.Restriction.Restriction.Bsp143I.results
Bio.Restriction.Restriction.Bsp1720I.results
Bio.Restriction.Restriction.Bsp19I.results
Bio.Restriction.Restriction.Bsp24I.results
Bio.Restriction.Restriction.Bsp68I.results
Bio.Restriction.Restriction.BspACI.results
Bio.Restriction.Restriction.BspCNI.results
Bio.Restriction.Restriction.BspD6I.results
Bio.Restriction.Restriction.BspDI.results
Bio.Restriction.Restriction.BspEI.results
Bio.Restriction.Restriction.BspFNI.results
Bio.Restriction.Restriction.BspGI.results
Bio.Restriction.Restriction.BspHI.results
Bio.Restriction.Restriction.BspLI.results
Bio.Restriction.Restriction.BspLU11I.results
Bio.Restriction.Restriction.BspMI.results
Bio.Restriction.Restriction.BspMII.results
Bio.Restriction.Restriction.BspNCI.results
Bio.Restriction.Restriction.BspOI.results
Bio.Restriction.Restriction.BspPI.results
Bio.Restriction.Restriction.BspQI.results
Bio.Restriction.Restriction.BspT104I.results
Bio.Restriction.Restriction.BspT107I.results
Bio.Restriction.Restriction.BspTI.results
Bio.Restriction.Restriction.BsrBI.results
Bio.Restriction.Restriction.BsrDI.results
Bio.Restriction.Restriction.BsrFI.results
Bio.Restriction.Restriction.BsrGI.results
Bio.Restriction.Restriction.BsrI.results
Bio.Restriction.Restriction.BsrSI.results
Bio.Restriction.Restriction.BssAI.results
Bio.Restriction.Restriction.BssECI.results
Bio.Restriction.Restriction.BssHII.results
Bio.Restriction.Restriction.BssKI.results
Bio.Restriction.Restriction.BssMI.results
Bio.Restriction.Restriction.BssNAI.results
Bio.Restriction.Restriction.BssNI.results
Bio.Restriction.Restriction.BssSI.results
Bio.Restriction.Restriction.BssT1I.results
Bio.Restriction.Restriction.Bst1107I.results
Bio.Restriction.Restriction.Bst2BI.results
Bio.Restriction.Restriction.Bst2UI.results
Bio.Restriction.Restriction.Bst4CI.results
Bio.Restriction.Restriction.Bst6I.results
Bio.Restriction.Restriction.BstACI.results
Bio.Restriction.Restriction.BstAFI.results
Bio.Restriction.Restriction.BstAPI.results
Bio.Restriction.Restriction.BstAUI.results
Bio.Restriction.Restriction.BstBAI.results
Bio.Restriction.Restriction.BstBI.results
Bio.Restriction.Restriction.BstC8I.results
Bio.Restriction.Restriction.BstDEI.results
Bio.Restriction.Restriction.BstDSI.results
Bio.Restriction.Restriction.BstEII.results
Bio.Restriction.Restriction.BstENI.results
Bio.Restriction.Restriction.BstF5I.results
Bio.Restriction.Restriction.BstFNI.results
Bio.Restriction.Restriction.BstH2I.results
Bio.Restriction.Restriction.BstHHI.results
Bio.Restriction.Restriction.BstKTI.results
Bio.Restriction.Restriction.BstMAI.results
Bio.Restriction.Restriction.BstMBI.results
Bio.Restriction.Restriction.BstMCI.results
Bio.Restriction.Restriction.BstMWI.results
Bio.Restriction.Restriction.BstNI.results
Bio.Restriction.Restriction.BstNSI.results
Bio.Restriction.Restriction.BstOI.results
Bio.Restriction.Restriction.BstPAI.results
Bio.Restriction.Restriction.BstPI.results
Bio.Restriction.Restriction.BstSCI.results
Bio.Restriction.Restriction.BstSFI.results
Bio.Restriction.Restriction.BstSLI.results
Bio.Restriction.Restriction.BstSNI.results
Bio.Restriction.Restriction.BstUI.results
Bio.Restriction.Restriction.BstV1I.results
Bio.Restriction.Restriction.BstV2I.results
Bio.Restriction.Restriction.BstX2I.results
Bio.Restriction.Restriction.BstXI.results
Bio.Restriction.Restriction.BstYI.results
Bio.Restriction.Restriction.BstZ17I.results
Bio.Restriction.Restriction.BstZI.results
Bio.Restriction.Restriction.Bsu15I.results
Bio.Restriction.Restriction.Bsu36I.results
Bio.Restriction.Restriction.BsuI.results
Bio.Restriction.Restriction.BsuRI.results
Bio.Restriction.Restriction.BtgI.results
Bio.Restriction.Restriction.BtgZI.results
Bio.Restriction.Restriction.BthCI.results
Bio.Restriction.Restriction.BtrI.results
Bio.Restriction.Restriction.BtsCI.results
Bio.Restriction.Restriction.BtsI.results
Bio.Restriction.Restriction.BtsIMutI.results
Bio.Restriction.Restriction.BtuMI.results
Bio.Restriction.Restriction.BveI.results
Bio.Restriction.Restriction.Cac8I.results
Bio.Restriction.Restriction.CaiI.results
Bio.Restriction.Restriction.CauII.results
Bio.Restriction.Restriction.CchII.results
Bio.Restriction.Restriction.CchIII.results
Bio.Restriction.Restriction.CciI.results
Bio.Restriction.Restriction.CciNI.results
Bio.Restriction.Restriction.Cdi630V.results
Bio.Restriction.Restriction.CdiI.results
Bio.Restriction.Restriction.CdpI.results
Bio.Restriction.Restriction.CfoI.results
Bio.Restriction.Restriction.Cfr10I.results
Bio.Restriction.Restriction.Cfr13I.results
Bio.Restriction.Restriction.Cfr42I.results
Bio.Restriction.Restriction.Cfr9I.results
Bio.Restriction.Restriction.CfrI.results
Bio.Restriction.Restriction.Cgl13032I.results
Bio.Restriction.Restriction.Cgl13032II.results
Bio.Restriction.Restriction.ChaI.results
Bio.Restriction.Restriction.CjeFIII.results
Bio.Restriction.Restriction.CjeFV.results
Bio.Restriction.Restriction.CjeI.results
Bio.Restriction.Restriction.CjeNII.results
Bio.Restriction.Restriction.CjeNIII.results
Bio.Restriction.Restriction.CjeP659IV.results
Bio.Restriction.Restriction.CjePI.results
Bio.Restriction.Restriction.CjuI.results
Bio.Restriction.Restriction.CjuII.results
Bio.Restriction.Restriction.ClaI.results
Bio.Restriction.Restriction.CpoI.results
Bio.Restriction.Restriction.CseI.results
Bio.Restriction.Restriction.CsiI.results
Bio.Restriction.Restriction.Csp6I.results
Bio.Restriction.Restriction.CspAI.results
Bio.Restriction.Restriction.CspCI.results
Bio.Restriction.Restriction.CspI.results
Bio.Restriction.Restriction.CstMI.results
Bio.Restriction.Restriction.CviAII.results
Bio.Restriction.Restriction.CviJI.results
Bio.Restriction.Restriction.CviKI_1.results
Bio.Restriction.Restriction.CviQI.results
Bio.Restriction.Restriction.CviRI.results
Bio.Restriction.Restriction.DdeI.results
Bio.Restriction.Restriction.DinI.results
Bio.Restriction.Restriction.DpnI.results
Bio.Restriction.Restriction.DpnII.results
Bio.Restriction.Restriction.DraI.results
Bio.Restriction.Restriction.DraII.results
Bio.Restriction.Restriction.DraIII.results
Bio.Restriction.Restriction.DraRI.results
Bio.Restriction.Restriction.DrdI.results
Bio.Restriction.Restriction.DrdII.results
Bio.Restriction.Restriction.DriI.results
Bio.Restriction.Restriction.DsaI.results
Bio.Restriction.Restriction.DseDI.results
Bio.Restriction.Restriction.EaeI.results
Bio.Restriction.Restriction.EagI.results
Bio.Restriction.Restriction.Eam1104I.results
Bio.Restriction.Restriction.Eam1105I.results
Bio.Restriction.Restriction.EarI.results
Bio.Restriction.Restriction.EciI.results
Bio.Restriction.Restriction.Ecl136II.results
Bio.Restriction.Restriction.EclXI.results
Bio.Restriction.Restriction.Eco105I.results
Bio.Restriction.Restriction.Eco130I.results
Bio.Restriction.Restriction.Eco147I.results
Bio.Restriction.Restriction.Eco24I.results
Bio.Restriction.Restriction.Eco31I.results
Bio.Restriction.Restriction.Eco32I.results
Bio.Restriction.Restriction.Eco47I.results
Bio.Restriction.Restriction.Eco47III.results
Bio.Restriction.Restriction.Eco52I.results
Bio.Restriction.Restriction.Eco53kI.results
Bio.Restriction.Restriction.Eco57I.results
Bio.Restriction.Restriction.Eco57MI.results
Bio.Restriction.Restriction.Eco72I.results
Bio.Restriction.Restriction.Eco81I.results
Bio.Restriction.Restriction.Eco88I.results
Bio.Restriction.Restriction.Eco91I.results
Bio.Restriction.Restriction.EcoHI.results
Bio.Restriction.Restriction.EcoICRI.results
Bio.Restriction.Restriction.EcoNI.results
Bio.Restriction.Restriction.EcoO109I.results
Bio.Restriction.Restriction.EcoO65I.results
Bio.Restriction.Restriction.EcoRI.results
Bio.Restriction.Restriction.EcoRII.results
Bio.Restriction.Restriction.EcoRV.results
Bio.Restriction.Restriction.EcoT14I.results
Bio.Restriction.Restriction.EcoT22I.results
Bio.Restriction.Restriction.EcoT38I.results
Bio.Restriction.Restriction.EgeI.results
Bio.Restriction.Restriction.EheI.results
Bio.Restriction.Restriction.ErhI.results
Bio.Restriction.Restriction.EsaBC3I.results
Bio.Restriction.Restriction.EsaSSI.results
Bio.Restriction.Restriction.Esp3I.results
Bio.Restriction.Restriction.EspI.results
Bio.Restriction.Restriction.FaeI.results
Bio.Restriction.Restriction.FaiI.results
Bio.Restriction.Restriction.FalI.results
Bio.Restriction.Restriction.FaqI.results
Bio.Restriction.Restriction.FatI.results
Bio.Restriction.Restriction.FauI.results
Bio.Restriction.Restriction.FauNDI.results
Bio.Restriction.Restriction.FbaI.results
Bio.Restriction.Restriction.FblI.results
Bio.Restriction.Restriction.FinI.results
Bio.Restriction.Restriction.FmuI.results
Bio.Restriction.Restriction.Fnu4HI.results
Bio.Restriction.Restriction.FnuDII.results
Bio.Restriction.Restriction.FokI.results
Bio.Restriction.Restriction.FriOI.results
Bio.Restriction.Restriction.FseI.results
Bio.Restriction.Restriction.Fsp4HI.results
Bio.Restriction.Restriction.FspAI.results
Bio.Restriction.Restriction.FspBI.results
Bio.Restriction.Restriction.FspEI.results
Bio.Restriction.Restriction.FspI.results
Bio.Restriction.Restriction.GauT27I.results
Bio.Restriction.Restriction.GdiII.results
Bio.Restriction.Restriction.GlaI.results
Bio.Restriction.Restriction.GluI.results
Bio.Restriction.Restriction.GsaI.results
Bio.Restriction.Restriction.GsuI.results
Bio.Restriction.Restriction.HaeI.results
Bio.Restriction.Restriction.HaeII.results
Bio.Restriction.Restriction.HaeIII.results
Bio.Restriction.Restriction.HapII.results
Bio.Restriction.Restriction.HauII.results
Bio.Restriction.Restriction.HgaI.results
Bio.Restriction.Restriction.HgiAI.results
Bio.Restriction.Restriction.HgiCI.results
Bio.Restriction.Restriction.HgiEII.results
Bio.Restriction.Restriction.HgiJII.results
Bio.Restriction.Restriction.HhaI.results
Bio.Restriction.Restriction.Hin1I.results
Bio.Restriction.Restriction.Hin1II.results
Bio.Restriction.Restriction.Hin4I.results
Bio.Restriction.Restriction.Hin4II.results
Bio.Restriction.Restriction.Hin6I.results
Bio.Restriction.Restriction.HinP1I.results
Bio.Restriction.Restriction.HincII.results
Bio.Restriction.Restriction.HindII.results
Bio.Restriction.Restriction.HindIII.results
Bio.Restriction.Restriction.HinfI.results
Bio.Restriction.Restriction.HpaI.results
Bio.Restriction.Restriction.HpaII.results
Bio.Restriction.Restriction.HphI.results
Bio.Restriction.Restriction.Hpy166II.results
Bio.Restriction.Restriction.Hpy178III.results
Bio.Restriction.Restriction.Hpy188I.results
Bio.Restriction.Restriction.Hpy188III.results
Bio.Restriction.Restriction.Hpy8I.results
Bio.Restriction.Restriction.Hpy99I.results
Bio.Restriction.Restriction.Hpy99XIII.results
Bio.Restriction.Restriction.Hpy99XIV.results
Bio.Restriction.Restriction.HpyAV.results
Bio.Restriction.Restriction.HpyCH4III.results
Bio.Restriction.Restriction.HpyCH4IV.results
Bio.Restriction.Restriction.HpyCH4V.results
Bio.Restriction.Restriction.HpyF10VI.results
Bio.Restriction.Restriction.HpyF3I.results
Bio.Restriction.Restriction.HpySE526I.results
Bio.Restriction.Restriction.Hsp92I.results
Bio.Restriction.Restriction.Hsp92II.results
Bio.Restriction.Restriction.HspAI.results
Bio.Restriction.Restriction.Jma19592I.results
Bio.Restriction.Restriction.KasI.results
Bio.Restriction.Restriction.KflI.results
Bio.Restriction.Restriction.Kpn2I.results
Bio.Restriction.Restriction.KpnI.results
Bio.Restriction.Restriction.KroI.results
Bio.Restriction.Restriction.Ksp22I.results
Bio.Restriction.Restriction.Ksp632I.results
Bio.Restriction.Restriction.KspAI.results
Bio.Restriction.Restriction.KspI.results
Bio.Restriction.Restriction.Kzo9I.results
Bio.Restriction.Restriction.LguI.results
Bio.Restriction.Restriction.LpnI.results
Bio.Restriction.Restriction.LpnPI.results
Bio.Restriction.Restriction.Lsp1109I.results
Bio.Restriction.Restriction.LweI.results
Bio.Restriction.Restriction.MabI.results
Bio.Restriction.Restriction.MaeI.results
Bio.Restriction.Restriction.MaeII.results
Bio.Restriction.Restriction.MaeIII.results
Bio.Restriction.Restriction.MalI.results
Bio.Restriction.Restriction.MaqI.results
Bio.Restriction.Restriction.MauBI.results
Bio.Restriction.Restriction.MbiI.results
Bio.Restriction.Restriction.MboI.results
Bio.Restriction.Restriction.MboII.results
Bio.Restriction.Restriction.McaTI.results
Bio.Restriction.Restriction.McrI.results
Bio.Restriction.Restriction.MfeI.results
Bio.Restriction.Restriction.MflI.results
Bio.Restriction.Restriction.MhlI.results
Bio.Restriction.Restriction.MjaIV.results
Bio.Restriction.Restriction.MkaDII.results
Bio.Restriction.Restriction.MlsI.results
Bio.Restriction.Restriction.MluCI.results
Bio.Restriction.Restriction.MluI.results
Bio.Restriction.Restriction.MluNI.results
Bio.Restriction.Restriction.Mly113I.results
Bio.Restriction.Restriction.MlyI.results
Bio.Restriction.Restriction.MmeI.results
Bio.Restriction.Restriction.MnlI.results
Bio.Restriction.Restriction.Mph1103I.results
Bio.Restriction.Restriction.MreI.results
Bio.Restriction.Restriction.MroI.results
Bio.Restriction.Restriction.MroNI.results
Bio.Restriction.Restriction.MroXI.results
Bio.Restriction.Restriction.MscI.results
Bio.Restriction.Restriction.MseI.results
Bio.Restriction.Restriction.MslI.results
Bio.Restriction.Restriction.Msp20I.results
Bio.Restriction.Restriction.MspA1I.results
Bio.Restriction.Restriction.MspCI.results
Bio.Restriction.Restriction.MspI.results
Bio.Restriction.Restriction.MspJI.results
Bio.Restriction.Restriction.MspR9I.results
Bio.Restriction.Restriction.MssI.results
Bio.Restriction.Restriction.MstI.results
Bio.Restriction.Restriction.MunI.results
Bio.Restriction.Restriction.Mva1269I.results
Bio.Restriction.Restriction.MvaI.results
Bio.Restriction.Restriction.MvnI.results
Bio.Restriction.Restriction.MvrI.results
Bio.Restriction.Restriction.MwoI.results
Bio.Restriction.Restriction.NaeI.results
Bio.Restriction.Restriction.NarI.results
Bio.Restriction.Restriction.NciI.results
Bio.Restriction.Restriction.NcoI.results
Bio.Restriction.Restriction.NdeI.results
Bio.Restriction.Restriction.NdeII.results
Bio.Restriction.Restriction.NgoAVIII.results
Bio.Restriction.Restriction.NgoMIV.results
Bio.Restriction.Restriction.NhaXI.results
Bio.Restriction.Restriction.NheI.results
Bio.Restriction.Restriction.NlaCI.results
Bio.Restriction.Restriction.NlaIII.results
Bio.Restriction.Restriction.NlaIV.results
Bio.Restriction.Restriction.Nli3877I.results
Bio.Restriction.Restriction.NmeAIII.results
Bio.Restriction.Restriction.NmeDI.results
Bio.Restriction.Restriction.NmuCI.results
Bio.Restriction.Restriction.NotI.results
Bio.Restriction.Restriction.NruI.results
Bio.Restriction.Restriction.NsbI.results
Bio.Restriction.Restriction.NsiI.results
Bio.Restriction.Restriction.NspBII.results
Bio.Restriction.Restriction.NspI.results
Bio.Restriction.Restriction.NspV.results
Bio.Restriction.Restriction.OliI.results
Bio.Restriction.Restriction.PabI.results
Bio.Restriction.Restriction.PacI.results
Bio.Restriction.Restriction.PaeI.results
Bio.Restriction.Restriction.PaeR7I.results
Bio.Restriction.Restriction.PagI.results
Bio.Restriction.Restriction.PalAI.results
Bio.Restriction.Restriction.PasI.results
Bio.Restriction.Restriction.PauI.results
Bio.Restriction.Restriction.PceI.results
Bio.Restriction.Restriction.PciI.results
Bio.Restriction.Restriction.PciSI.results
Bio.Restriction.Restriction.PcsI.results
Bio.Restriction.Restriction.PctI.results
Bio.Restriction.Restriction.PdiI.results
Bio.Restriction.Restriction.PdmI.results
Bio.Restriction.Restriction.PenI.results
Bio.Restriction.Restriction.PfeI.results
Bio.Restriction.Restriction.Pfl1108I.results
Bio.Restriction.Restriction.Pfl23II.results
Bio.Restriction.Restriction.PflFI.results
Bio.Restriction.Restriction.PflMI.results
Bio.Restriction.Restriction.PfoI.results
Bio.Restriction.Restriction.PinAI.results
Bio.Restriction.Restriction.PlaDI.results
Bio.Restriction.Restriction.Ple19I.results
Bio.Restriction.Restriction.PleI.results
Bio.Restriction.Restriction.PluTI.results
Bio.Restriction.Restriction.PmaCI.results
Bio.Restriction.Restriction.PmeI.results
Bio.Restriction.Restriction.PmlI.results
Bio.Restriction.Restriction.PpiI.results
Bio.Restriction.Restriction.PpsI.results
Bio.Restriction.Restriction.Ppu10I.results
Bio.Restriction.Restriction.Ppu21I.results
Bio.Restriction.Restriction.PpuMI.results
Bio.Restriction.Restriction.PscI.results
Bio.Restriction.Restriction.PshAI.results
Bio.Restriction.Restriction.PshBI.results
Bio.Restriction.Restriction.PsiI.results
Bio.Restriction.Restriction.Psp03I.results
Bio.Restriction.Restriction.Psp124BI.results
Bio.Restriction.Restriction.Psp1406I.results
Bio.Restriction.Restriction.Psp5II.results
Bio.Restriction.Restriction.Psp6I.results
Bio.Restriction.Restriction.PspCI.results
Bio.Restriction.Restriction.PspEI.results
Bio.Restriction.Restriction.PspGI.results
Bio.Restriction.Restriction.PspLI.results
Bio.Restriction.Restriction.PspN4I.results
Bio.Restriction.Restriction.PspOMI.results
Bio.Restriction.Restriction.PspOMII.results
Bio.Restriction.Restriction.PspPI.results
Bio.Restriction.Restriction.PspPPI.results
Bio.Restriction.Restriction.PspPRI.results
Bio.Restriction.Restriction.PspXI.results
Bio.Restriction.Restriction.PsrI.results
Bio.Restriction.Restriction.PssI.results
Bio.Restriction.Restriction.PstI.results
Bio.Restriction.Restriction.PstNI.results
Bio.Restriction.Restriction.PsuI.results
Bio.Restriction.Restriction.PsyI.results
Bio.Restriction.Restriction.PteI.results
Bio.Restriction.Restriction.PvuI.results
Bio.Restriction.Restriction.PvuII.results
Bio.Restriction.Restriction.R2_BceSIV.results
Bio.Restriction.Restriction.RceI.results
Bio.Restriction.Restriction.RdeGBI.results
Bio.Restriction.Restriction.RdeGBII.results
Bio.Restriction.Restriction.RdeGBIII.results
Bio.Restriction.Restriction.RflFIII.results
Bio.Restriction.Restriction.RgaI.results
Bio.Restriction.Restriction.RigI.results
Bio.Restriction.Restriction.RlaI.results
Bio.Restriction.Restriction.RleAI.results
Bio.Restriction.Restriction.RpaB5I.results
Bio.Restriction.Restriction.RpaBI.results
Bio.Restriction.Restriction.RpaI.results
Bio.Restriction.Restriction.RpaTI.results
Bio.Restriction.Restriction.RruI.results
Bio.Restriction.Restriction.RsaI.results
Bio.Restriction.Restriction.RsaNI.results
Bio.Restriction.Restriction.RseI.results
Bio.Restriction.Restriction.Rsr2I.results
Bio.Restriction.Restriction.RsrII.results
Bio.Restriction.Restriction.SacI.results
Bio.Restriction.Restriction.SacII.results
Bio.Restriction.Restriction.SalI.results
Bio.Restriction.Restriction.SanDI.results
Bio.Restriction.Restriction.SapI.results
Bio.Restriction.Restriction.SaqAI.results
Bio.Restriction.Restriction.SatI.results
Bio.Restriction.Restriction.Sau3AI.results
Bio.Restriction.Restriction.Sau96I.results
Bio.Restriction.Restriction.SauI.results
Bio.Restriction.Restriction.SbfI.results
Bio.Restriction.Restriction.ScaI.results
Bio.Restriction.Restriction.SchI.results
Bio.Restriction.Restriction.SciI.results
Bio.Restriction.Restriction.ScrFI.results
Bio.Restriction.Restriction.SdaI.results
Bio.Restriction.Restriction.SdeAI.results
Bio.Restriction.Restriction.SdeOSI.results
Bio.Restriction.Restriction.SduI.results
Bio.Restriction.Restriction.SecI.results
Bio.Restriction.Restriction.SelI.results
Bio.Restriction.Restriction.SetI.results
Bio.Restriction.Restriction.SexAI.results
Bio.Restriction.Restriction.SfaAI.results
Bio.Restriction.Restriction.SfaNI.results
Bio.Restriction.Restriction.SfcI.results
Bio.Restriction.Restriction.SfeI.results
Bio.Restriction.Restriction.SfiI.results
Bio.Restriction.Restriction.SfoI.results
Bio.Restriction.Restriction.Sfr274I.results
Bio.Restriction.Restriction.Sfr303I.results
Bio.Restriction.Restriction.SfuI.results
Bio.Restriction.Restriction.SgeI.results
Bio.Restriction.Restriction.SgfI.results
Bio.Restriction.Restriction.SgrAI.results
Bio.Restriction.Restriction.SgrBI.results
Bio.Restriction.Restriction.SgrDI.results
Bio.Restriction.Restriction.SgrTI.results
Bio.Restriction.Restriction.SgsI.results
Bio.Restriction.Restriction.SimI.results
Bio.Restriction.Restriction.SlaI.results
Bio.Restriction.Restriction.SmaI.results
Bio.Restriction.Restriction.SmiI.results
Bio.Restriction.Restriction.SmiMI.results
Bio.Restriction.Restriction.SmlI.results
Bio.Restriction.Restriction.SmoI.results
Bio.Restriction.Restriction.SnaBI.results
Bio.Restriction.Restriction.SnaI.results
Bio.Restriction.Restriction.Sno506I.results
Bio.Restriction.Restriction.SpeI.results
Bio.Restriction.Restriction.SphI.results
Bio.Restriction.Restriction.SplI.results
Bio.Restriction.Restriction.SpoDI.results
Bio.Restriction.Restriction.SrfI.results
Bio.Restriction.Restriction.Sse232I.results
Bio.Restriction.Restriction.Sse8387I.results
Bio.Restriction.Restriction.Sse8647I.results
Bio.Restriction.Restriction.Sse9I.results
Bio.Restriction.Restriction.SseBI.results
Bio.Restriction.Restriction.SsiI.results
Bio.Restriction.Restriction.SspD5I.results
Bio.Restriction.Restriction.SspDI.results
Bio.Restriction.Restriction.SspI.results
Bio.Restriction.Restriction.SstE37I.results
Bio.Restriction.Restriction.SstI.results
Bio.Restriction.Restriction.Sth132I.results
Bio.Restriction.Restriction.Sth302II.results
Bio.Restriction.Restriction.StrI.results
Bio.Restriction.Restriction.StsI.results
Bio.Restriction.Restriction.StuI.results
Bio.Restriction.Restriction.StyD4I.results
Bio.Restriction.Restriction.StyI.results
Bio.Restriction.Restriction.SwaI.results
Bio.Restriction.Restriction.TaaI.results
Bio.Restriction.Restriction.TaiI.results
Bio.Restriction.Restriction.TaqI.results
Bio.Restriction.Restriction.TaqII.results
Bio.Restriction.Restriction.TasI.results
Bio.Restriction.Restriction.TatI.results
Bio.Restriction.Restriction.TauI.results
Bio.Restriction.Restriction.TfiI.results
Bio.Restriction.Restriction.Tru1I.results
Bio.Restriction.Restriction.Tru9I.results
Bio.Restriction.Restriction.TscAI.results
Bio.Restriction.Restriction.TseFI.results
Bio.Restriction.Restriction.TseI.results
Bio.Restriction.Restriction.TsoI.results
Bio.Restriction.Restriction.Tsp45I.results
Bio.Restriction.Restriction.Tsp4CI.results
Bio.Restriction.Restriction.TspDTI.results
Bio.Restriction.Restriction.TspEI.results
Bio.Restriction.Restriction.TspGWI.results
Bio.Restriction.Restriction.TspMI.results
Bio.Restriction.Restriction.TspRI.results
Bio.Restriction.Restriction.TssI.results
Bio.Restriction.Restriction.TstI.results
Bio.Restriction.Restriction.TsuI.results
Bio.Restriction.Restriction.Tth111I.results
Bio.Restriction.Restriction.Tth111II.results
Bio.Restriction.Restriction.UbaF11I.results
Bio.Restriction.Restriction.UbaF12I.results
Bio.Restriction.Restriction.UbaF13I.results
Bio.Restriction.Restriction.UbaF14I.results
Bio.Restriction.Restriction.UbaF9I.results
Bio.Restriction.Restriction.UbaPI.results
Bio.Restriction.Restriction.UcoMSI.results
Bio.Restriction.Restriction.UnbI.results
Bio.Restriction.Restriction.Van91I.results
Bio.Restriction.Restriction.Vha464I.results
Bio.Restriction.Restriction.VneI.results
Bio.Restriction.Restriction.VpaK11AI.results
Bio.Restriction.Restriction.VpaK11BI.results
Bio.Restriction.Restriction.VspI.results
Bio.Restriction.Restriction.WviI.results
Bio.Restriction.Restriction.XagI.results
Bio.Restriction.Restriction.XapI.results
Bio.Restriction.Restriction.XbaI.results
Bio.Restriction.Restriction.XceI.results
Bio.Restriction.Restriction.XcmI.results
Bio.Restriction.Restriction.XhoI.results
Bio.Restriction.Restriction.XhoII.results
Bio.Restriction.Restriction.XmaI.results
Bio.Restriction.Restriction.XmaIII.results
Bio.Restriction.Restriction.XmaJI.results
Bio.Restriction.Restriction.XmiI.results
Bio.Restriction.Restriction.XmnI.results
Bio.Restriction.Restriction.XspI.results
Bio.Restriction.Restriction.YkrI.results
Bio.Restriction.Restriction.ZraI.results
Bio.Restriction.Restriction.ZrmI.results
Bio.Restriction.Restriction.Zsp2I.results" class="py-name" href="#" onclick="return doclink('link-999', 'results', 'link-996');">results</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1710"></a><tt class="py-lineno">1710</tt>  <tt class="py-line"> </tt>
<a name="L1711"></a><tt class="py-lineno">1711</tt>  <tt class="py-line"> </tt>
<a name="_safe_int"></a><div id="_safe_int-def"><a name="L1712"></a><tt class="py-lineno">1712</tt> <a class="py-toggle" href="#" id="_safe_int-toggle" onclick="return toggle('_safe_int');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone-module.html#_safe_int">_safe_int</a><tt class="py-op">(</tt><tt class="py-param">str</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_safe_int-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_safe_int-expanded"><a name="L1713"></a><tt class="py-lineno">1713</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1714"></a><tt class="py-lineno">1714</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">)</tt> </tt>
<a name="L1715"></a><tt class="py-lineno">1715</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L1716"></a><tt class="py-lineno">1716</tt>  <tt class="py-line">        <tt class="py-comment"># Something went wrong.  Try to clean up the string.</tt> </tt>
<a name="L1717"></a><tt class="py-lineno">1717</tt>  <tt class="py-line">        <tt class="py-comment"># Remove all commas from the string</tt> </tt>
<a name="L1718"></a><tt class="py-lineno">1718</tt>  <tt class="py-line">        <tt class="py-name">str</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">','</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L1719"></a><tt class="py-lineno">1719</tt>  <tt class="py-line">    <tt class="py-comment"># try again after removing commas.</tt> </tt>
<a name="L1720"></a><tt class="py-lineno">1720</tt>  <tt class="py-line">    <tt class="py-comment"># Note int() will return a long rather than overflow</tt> </tt>
<a name="L1721"></a><tt class="py-lineno">1721</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1722"></a><tt class="py-lineno">1722</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">)</tt> </tt>
<a name="L1723"></a><tt class="py-lineno">1723</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L1724"></a><tt class="py-lineno">1724</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
<a name="L1725"></a><tt class="py-lineno">1725</tt>  <tt class="py-line">    <tt class="py-comment"># Call float to handle things like "54.3", note could lose precision, e.g.</tt> </tt>
<a name="L1726"></a><tt class="py-lineno">1726</tt>  <tt class="py-line">    <tt class="py-comment"># &gt;&gt;&gt; int("5399354557888517312")</tt> </tt>
<a name="L1727"></a><tt class="py-lineno">1727</tt>  <tt class="py-line">    <tt class="py-comment"># 5399354557888517312</tt> </tt>
<a name="L1728"></a><tt class="py-lineno">1728</tt>  <tt class="py-line">    <tt class="py-comment"># &gt;&gt;&gt; int(float("5399354557888517312"))</tt> </tt>
<a name="L1729"></a><tt class="py-lineno">1729</tt>  <tt class="py-line">    <tt class="py-comment"># 5399354557888517120</tt> </tt>
<a name="L1730"></a><tt class="py-lineno">1730</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1731"></a><tt class="py-lineno">1731</tt>  <tt class="py-line"> </tt>
<a name="L1732"></a><tt class="py-lineno">1732</tt>  <tt class="py-line"> </tt>
<a name="_safe_float"></a><div id="_safe_float-def"><a name="L1733"></a><tt class="py-lineno">1733</tt> <a class="py-toggle" href="#" id="_safe_float-toggle" onclick="return toggle('_safe_float');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone-module.html#_safe_float">_safe_float</a><tt class="py-op">(</tt><tt class="py-param">str</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_safe_float-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_safe_float-expanded"><a name="L1734"></a><tt class="py-lineno">1734</tt>  <tt class="py-line">    <tt class="py-comment"># Thomas Rosleff Soerensen (rosleff@mpiz-koeln.mpg.de) noted that</tt> </tt>
<a name="L1735"></a><tt class="py-lineno">1735</tt>  <tt class="py-line">    <tt class="py-comment"># float('e-172') does not produce an error on his platform.  Thus,</tt> </tt>
<a name="L1736"></a><tt class="py-lineno">1736</tt>  <tt class="py-line">    <tt class="py-comment"># we need to check the string for this condition.</tt> </tt>
<a name="L1737"></a><tt class="py-lineno">1737</tt>  <tt class="py-line"> </tt>
<a name="L1738"></a><tt class="py-lineno">1738</tt>  <tt class="py-line">    <tt class="py-comment"># Sometimes BLAST leaves of the '1' in front of an exponent.</tt> </tt>
<a name="L1739"></a><tt class="py-lineno">1739</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">str</tt> <tt class="py-keyword">and</tt> <tt class="py-name">str</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">'E'</tt><tt class="py-op">,</tt> <tt class="py-string">'e'</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L1740"></a><tt class="py-lineno">1740</tt>  <tt class="py-line">        <tt class="py-name">str</tt> <tt class="py-op">=</tt> <tt class="py-string">'1'</tt> <tt class="py-op">+</tt> <tt class="py-name">str</tt> </tt>
<a name="L1741"></a><tt class="py-lineno">1741</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1742"></a><tt class="py-lineno">1742</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">)</tt> </tt>
<a name="L1743"></a><tt class="py-lineno">1743</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L1744"></a><tt class="py-lineno">1744</tt>  <tt class="py-line">        <tt class="py-comment"># Remove all commas from the string</tt> </tt>
<a name="L1745"></a><tt class="py-lineno">1745</tt>  <tt class="py-line">        <tt class="py-name">str</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">','</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L1746"></a><tt class="py-lineno">1746</tt>  <tt class="py-line">    <tt class="py-comment"># try again.</tt> </tt>
<a name="L1747"></a><tt class="py-lineno">1747</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1748"></a><tt class="py-lineno">1748</tt>  <tt class="py-line"> </tt>
<a name="L1749"></a><tt class="py-lineno">1749</tt>  <tt class="py-line"> </tt>
<a name="_BlastErrorConsumer"></a><div id="_BlastErrorConsumer-def"><a name="L1750"></a><tt class="py-lineno">1750</tt> <a class="py-toggle" href="#" id="_BlastErrorConsumer-toggle" onclick="return toggle('_BlastErrorConsumer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._BlastErrorConsumer-class.html">_BlastErrorConsumer</a><tt class="py-op">(</tt><tt class="py-base-class">_BlastConsumer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BlastErrorConsumer-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_BlastErrorConsumer-expanded"><a name="_BlastErrorConsumer.__init__"></a><div id="_BlastErrorConsumer.__init__-def"><a name="L1751"></a><tt class="py-lineno">1751</tt> <a class="py-toggle" href="#" id="_BlastErrorConsumer.__init__-toggle" onclick="return toggle('_BlastErrorConsumer.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._BlastErrorConsumer-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BlastErrorConsumer.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_BlastErrorConsumer.__init__-expanded"><a name="L1752"></a><tt class="py-lineno">1752</tt>  <tt class="py-line">        <tt id="link-1000" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer" class="py-name" href="#" onclick="return doclink('link-1000', '_BlastConsumer', 'link-430');">_BlastConsumer</a></tt><tt class="py-op">.</tt><tt id="link-1001" class="py-name" targets="Method Bio.Affy.CelFile.Record.__init__()=Bio.Affy.CelFile.Record-class.html#__init__,Method Bio.Align.AlignInfo.PSSM.__init__()=Bio.Align.AlignInfo.PSSM-class.html#__init__,Method Bio.Align.AlignInfo.SummaryInfo.__init__()=Bio.Align.AlignInfo.SummaryInfo-class.html#__init__,Method Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline.__init__()=Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline-class.html#__init__,Method Bio.Align.Applications._Clustalw.ClustalwCommandline.__init__()=Bio.Align.Applications._Clustalw.ClustalwCommandline-class.html#__init__,Method Bio.Align.Applications._Dialign.DialignCommandline.__init__()=Bio.Align.Applications._Dialign.DialignCommandline-class.html#__init__,Method Bio.Align.Applications._MSAProbs.MSAProbsCommandline.__init__()=Bio.Align.Applications._MSAProbs.MSAProbsCommandline-class.html#__init__,Method Bio.Align.Applications._Mafft.MafftCommandline.__init__()=Bio.Align.Applications._Mafft.MafftCommandline-class.html#__init__,Method Bio.Align.Applications._Muscle.MuscleCommandline.__init__()=Bio.Align.Applications._Muscle.MuscleCommandline-class.html#__init__,Method Bio.Align.Applications._Prank.PrankCommandline.__init__()=Bio.Align.Applications._Prank.PrankCommandline-class.html#__init__,Method Bio.Align.Applications._Probcons.ProbconsCommandline.__init__()=Bio.Align.Applications._Probcons.ProbconsCommandline-class.html#__init__,Method Bio.Align.Applications._TCoffee.TCoffeeCommandline.__init__()=Bio.Align.Applications._TCoffee.TCoffeeCommandline-class.html#__init__,Method Bio.Align.Generic.Alignment.__init__()=Bio.Align.Generic.Alignment-class.html#__init__,Method Bio.Align.MultipleSeqAlignment.__init__()=Bio.Align.MultipleSeqAlignment-class.html#__init__,Method Bio.AlignIO.Interfaces.AlignmentIterator.__init__()=Bio.AlignIO.Interfaces.AlignmentIterator-class.html#__init__,Method Bio.AlignIO.Interfaces.AlignmentWriter.__init__()=Bio.AlignIO.Interfaces.AlignmentWriter-class.html#__init__,Method Bio.AlignIO.Interfaces.SequentialAlignmentWriter.__init__()=Bio.AlignIO.Interfaces.SequentialAlignmentWriter-class.html#__init__,Method Bio.Alphabet.AlphabetEncoder.__init__()=Bio.Alphabet.AlphabetEncoder-class.html#__init__,Method Bio.Alphabet.Gapped.__init__()=Bio.Alphabet.Gapped-class.html#__init__,Method Bio.Alphabet.HasStopCodon.__init__()=Bio.Alphabet.HasStopCodon-class.html#__init__,Method Bio.Application.AbstractCommandline.__init__()=Bio.Application.AbstractCommandline-class.html#__init__,Method Bio.Application.ApplicationError.__init__()=Bio.Application.ApplicationError-class.html#__init__,Method Bio.Application._AbstractParameter.__init__()=Bio.Application._AbstractParameter-class.html#__init__,Method Bio.Application._Argument.__init__()=Bio.Application._Argument-class.html#__init__,Method Bio.Application._Option.__init__()=Bio.Application._Option-class.html#__init__,Method Bio.Application._StaticArgument.__init__()=Bio.Application._StaticArgument-class.html#__init__,Method Bio.Application._Switch.__init__()=Bio.Application._Switch-class.html#__init__,Method Bio.Blast.Applications.NcbiblastformatterCommandline.__init__()=Bio.Blast.Applications.NcbiblastformatterCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastnCommandline.__init__()=Bio.Blast.Applications.NcbiblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastpCommandline.__init__()=Bio.Blast.Applications.NcbiblastpCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastxCommandline.__init__()=Bio.Blast.Applications.NcbiblastxCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbideltablastCommandline.__init__()=Bio.Blast.Applications.NcbideltablastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbipsiblastCommandline.__init__()=Bio.Blast.Applications.NcbipsiblastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbirpsblastCommandline.__init__()=Bio.Blast.Applications.NcbirpsblastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbirpstblastnCommandline.__init__()=Bio.Blast.Applications.NcbirpstblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbitblastnCommandline.__init__()=Bio.Blast.Applications.NcbitblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbitblastxCommandline.__init__()=Bio.Blast.Applications.NcbitblastxCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbibaseblastCommandline.__init__()=Bio.Blast.Applications._NcbibaseblastCommandline-class.html#__init__,Method Bio.Blast.Applications._Ncbiblast2SeqCommandline.__init__()=Bio.Blast.Applications._Ncbiblast2SeqCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbiblastCommandline.__init__()=Bio.Blast.Applications._NcbiblastCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbiblastMain2SeqCommandline.__init__()=Bio.Blast.Applications._NcbiblastMain2SeqCommandline-class.html#__init__,Method Bio.Blast.NCBIStandalone.BlastErrorParser.__init__()=Bio.Blast.NCBIStandalone.BlastErrorParser-class.html#__init__,Method Bio.Blast.NCBIStandalone.BlastParser.__init__()=Bio.Blast.NCBIStandalone.BlastParser-class.html#__init__,Method Bio.Blast.NCBIStandalone.Iterator.__init__()=Bio.Blast.NCBIStandalone.Iterator-class.html#__init__,Method Bio.Blast.NCBIStandalone.PSIBlastParser.__init__()=Bio.Blast.NCBIStandalone.PSIBlastParser-class.html#__init__,Method Bio.Blast.NCBIStandalone._BlastConsumer.__init__()=Bio.Blast.NCBIStandalone._BlastConsumer-class.html#__init__,Method Bio.Blast.NCBIStandalone._BlastErrorConsumer.__init__()=Bio.Blast.NCBIStandalone._BlastErrorConsumer-class.html#__init__,Method Bio.Blast.NCBIStandalone._PSIBlastConsumer.__init__()=Bio.Blast.NCBIStandalone._PSIBlastConsumer-class.html#__init__,Method Bio.Blast.NCBIXML.BlastParser.__init__()=Bio.Blast.NCBIXML.BlastParser-class.html#__init__,Method Bio.Blast.NCBIXML._XMLparser.__init__()=Bio.Blast.NCBIXML._XMLparser-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableEntry.__init__()=Bio.Blast.ParseBlastTable.BlastTableEntry-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableReader.__init__()=Bio.Blast.ParseBlastTable.BlastTableReader-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableRec.__init__()=Bio.Blast.ParseBlastTable.BlastTableRec-class.html#__init__,Method Bio.Blast.Record.Alignment.__init__()=Bio.Blast.Record.Alignment-class.html#__init__,Method Bio.Blast.Record.Blast.__init__()=Bio.Blast.Record.Blast-class.html#__init__,Method Bio.Blast.Record.DatabaseReport.__init__()=Bio.Blast.Record.DatabaseReport-class.html#__init__,Method Bio.Blast.Record.Description.__init__()=Bio.Blast.Record.Description-class.html#__init__,Method Bio.Blast.Record.HSP.__init__()=Bio.Blast.Record.HSP-class.html#__init__,Method Bio.Blast.Record.Header.__init__()=Bio.Blast.Record.Header-class.html#__init__,Method Bio.Blast.Record.MultipleAlignment.__init__()=Bio.Blast.Record.MultipleAlignment-class.html#__init__,Method Bio.Blast.Record.PSIBlast.__init__()=Bio.Blast.Record.PSIBlast-class.html#__init__,Method Bio.Blast.Record.Parameters.__init__()=Bio.Blast.Record.Parameters-class.html#__init__,Method Bio.Blast.Record.Round.__init__()=Bio.Blast.Record.Round-class.html#__init__,Method Bio.CAPS.CAPSMap.__init__()=Bio.CAPS.CAPSMap-class.html#__init__,Method Bio.CAPS.DifferentialCutsite.__init__()=Bio.CAPS.DifferentialCutsite-class.html#__init__,Method Bio.Cluster.Record.__init__()=Bio.Cluster.Record-class.html#__init__,Method Bio.Compass.Record.__init__()=Bio.Compass.Record-class.html#__init__,Method Bio.Crystal.Chain.__init__()=Bio.Crystal.Chain-class.html#__init__,Method Bio.Crystal.Crystal.__init__()=Bio.Crystal.Crystal-class.html#__init__,Method Bio.Crystal.Hetero.__init__()=Bio.Crystal.Hetero-class.html#__init__,Method Bio.Data.CodonTable.AmbiguousCodonTable.__init__()=Bio.Data.CodonTable.AmbiguousCodonTable-class.html#__init__,Method Bio.Data.CodonTable.AmbiguousForwardTable.__init__()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#__init__,Method Bio.Data.CodonTable.CodonTable.__init__()=Bio.Data.CodonTable.CodonTable-class.html#__init__,Method Bio.Data.CodonTable.NCBICodonTable.__init__()=Bio.Data.CodonTable.NCBICodonTable-class.html#__init__,Method Bio.DocSQL.Create.__init__()=Bio.DocSQL.Create-class.html#__init__,Method Bio.DocSQL.Insert.__init__()=Bio.DocSQL.Insert-class.html#__init__,Method Bio.DocSQL.IterationCursor.__init__()=Bio.DocSQL.IterationCursor-class.html#__init__,Method Bio.DocSQL.Query.__init__()=Bio.DocSQL.Query-class.html#__init__,Method Bio.DocSQL.QueryAll.__init__()=Bio.DocSQL.QueryAll-class.html#__init__,Method Bio.DocSQL.QueryGeneric.__init__()=Bio.DocSQL.QueryGeneric-class.html#__init__,Method Bio.DocSQL.QueryRow.__init__()=Bio.DocSQL.QueryRow-class.html#__init__,Method Bio.DocSQL.QuerySingle.__init__()=Bio.DocSQL.QuerySingle-class.html#__init__,Method Bio.Emboss.Applications.DiffseqCommandline.__init__()=Bio.Emboss.Applications.DiffseqCommandline-class.html#__init__,Method Bio.Emboss.Applications.EInvertedCommandline.__init__()=Bio.Emboss.Applications.EInvertedCommandline-class.html#__init__,Method Bio.Emboss.Applications.ETandemCommandline.__init__()=Bio.Emboss.Applications.ETandemCommandline-class.html#__init__,Method Bio.Emboss.Applications.Est2GenomeCommandline.__init__()=Bio.Emboss.Applications.Est2GenomeCommandline-class.html#__init__,Method Bio.Emboss.Applications.FConsenseCommandline.__init__()=Bio.Emboss.Applications.FConsenseCommandline-class.html#__init__,Method Bio.Emboss.Applications.FDNADistCommandline.__init__()=Bio.Emboss.Applications.FDNADistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FDNAParsCommandline.__init__()=Bio.Emboss.Applications.FDNAParsCommandline-class.html#__init__,Method Bio.Emboss.Applications.FNeighborCommandline.__init__()=Bio.Emboss.Applications.FNeighborCommandline-class.html#__init__,Method Bio.Emboss.Applications.FProtDistCommandline.__init__()=Bio.Emboss.Applications.FProtDistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FProtParsCommandline.__init__()=Bio.Emboss.Applications.FProtParsCommandline-class.html#__init__,Method Bio.Emboss.Applications.FSeqBootCommandline.__init__()=Bio.Emboss.Applications.FSeqBootCommandline-class.html#__init__,Method Bio.Emboss.Applications.FTreeDistCommandline.__init__()=Bio.Emboss.Applications.FTreeDistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FuzznucCommandline.__init__()=Bio.Emboss.Applications.FuzznucCommandline-class.html#__init__,Method Bio.Emboss.Applications.IepCommandline.__init__()=Bio.Emboss.Applications.IepCommandline-class.html#__init__,Method Bio.Emboss.Applications.NeedleCommandline.__init__()=Bio.Emboss.Applications.NeedleCommandline-class.html#__init__,Method Bio.Emboss.Applications.NeedleallCommandline.__init__()=Bio.Emboss.Applications.NeedleallCommandline-class.html#__init__,Method Bio.Emboss.Applications.PalindromeCommandline.__init__()=Bio.Emboss.Applications.PalindromeCommandline-class.html#__init__,Method Bio.Emboss.Applications.Primer3Commandline.__init__()=Bio.Emboss.Applications.Primer3Commandline-class.html#__init__,Method Bio.Emboss.Applications.PrimerSearchCommandline.__init__()=Bio.Emboss.Applications.PrimerSearchCommandline-class.html#__init__,Method Bio.Emboss.Applications.SeqmatchallCommandline.__init__()=Bio.Emboss.Applications.SeqmatchallCommandline-class.html#__init__,Method Bio.Emboss.Applications.SeqretCommandline.__init__()=Bio.Emboss.Applications.SeqretCommandline-class.html#__init__,Method Bio.Emboss.Applications.StretcherCommandline.__init__()=Bio.Emboss.Applications.StretcherCommandline-class.html#__init__,Method Bio.Emboss.Applications.TranalignCommandline.__init__()=Bio.Emboss.Applications.TranalignCommandline-class.html#__init__,Method Bio.Emboss.Applications.WaterCommandline.__init__()=Bio.Emboss.Applications.WaterCommandline-class.html#__init__,Method Bio.Emboss.Applications._EmbossCommandLine.__init__()=Bio.Emboss.Applications._EmbossCommandLine-class.html#__init__,Method Bio.Emboss.Applications._EmbossMinimalCommandLine.__init__()=Bio.Emboss.Applications._EmbossMinimalCommandLine-class.html#__init__,Method Bio.Emboss.Primer3.Primers.__init__()=Bio.Emboss.Primer3.Primers-class.html#__init__,Method Bio.Emboss.Primer3.Record.__init__()=Bio.Emboss.Primer3.Record-class.html#__init__,Method Bio.Emboss.PrimerSearch.Amplifier.__init__()=Bio.Emboss.PrimerSearch.Amplifier-class.html#__init__,Method Bio.Emboss.PrimerSearch.InputRecord.__init__()=Bio.Emboss.PrimerSearch.InputRecord-class.html#__init__,Method Bio.Emboss.PrimerSearch.OutputRecord.__init__()=Bio.Emboss.PrimerSearch.OutputRecord-class.html#__init__,Method Bio.Entrez.Parser.CorruptedXMLError.__init__()=Bio.Entrez.Parser.CorruptedXMLError-class.html#__init__,Method Bio.Entrez.Parser.DataHandler.__init__()=Bio.Entrez.Parser.DataHandler-class.html#__init__,Method Bio.Entrez.Parser.NotXMLError.__init__()=Bio.Entrez.Parser.NotXMLError-class.html#__init__,Method Bio.Entrez.Parser.StructureElement.__init__()=Bio.Entrez.Parser.StructureElement-class.html#__init__,Method Bio.Entrez.Parser.ValidationError.__init__()=Bio.Entrez.Parser.ValidationError-class.html#__init__,Method Bio.ExPASy.Enzyme.Record.__init__()=Bio.ExPASy.Enzyme.Record-class.html#__init__,Method Bio.ExPASy.Prodoc.Record.__init__()=Bio.ExPASy.Prodoc.Record-class.html#__init__,Method Bio.ExPASy.Prodoc.Reference.__init__()=Bio.ExPASy.Prodoc.Reference-class.html#__init__,Method Bio.ExPASy.Prosite.Record.__init__()=Bio.ExPASy.Prosite.Record-class.html#__init__,Method Bio.ExPASy.ScanProsite.ContentHandler.__init__()=Bio.ExPASy.ScanProsite.ContentHandler-class.html#__init__,Method Bio.ExPASy.ScanProsite.Parser.__init__()=Bio.ExPASy.ScanProsite.Parser-class.html#__init__,Method Bio.ExPASy.ScanProsite.Record.__init__()=Bio.ExPASy.ScanProsite.Record-class.html#__init__,Method Bio.FSSP.FSSPAlignDict.__init__()=Bio.FSSP.FSSPAlignDict-class.html#__init__,Method Bio.FSSP.FSSPAlignRec.__init__()=Bio.FSSP.FSSPAlignRec-class.html#__init__,Method Bio.FSSP.FSSPHeader.__init__()=Bio.FSSP.FSSPHeader-class.html#__init__,Method Bio.FSSP.FSSPSumRec.__init__()=Bio.FSSP.FSSPSumRec-class.html#__init__,Method Bio.FSSP.FSSPTools.FSSPMultAlign.__init__()=Bio.FSSP.FSSPTools.FSSPMultAlign-class.html#__init__,Method Bio.FSSP.PosAlign.__init__()=Bio.FSSP.PosAlign-class.html#__init__,Method Bio.FSSP.fssp_rec.fff_rec.__init__()=Bio.FSSP.fssp_rec.fff_rec-class.html#__init__,Method Bio.File.UndoHandle.__init__()=Bio.File.UndoHandle-class.html#__init__,Method Bio.File._IndexedSeqFileDict.__init__()=Bio.File._IndexedSeqFileDict-class.html#__init__,Method Bio.File._SQLiteManySeqFilesDict.__init__()=Bio.File._SQLiteManySeqFilesDict-class.html#__init__,Method Bio.GA.Crossover.General.SafeFitnessCrossover.__init__()=Bio.GA.Crossover.General.SafeFitnessCrossover-class.html#__init__,Method Bio.GA.Crossover.GeneralPoint.GeneralPointCrossover.__init__()=Bio.GA.Crossover.GeneralPoint.GeneralPointCrossover-class.html#__init__,Method Bio.GA.Crossover.GeneralPoint.InterleaveCrossover.__init__()=Bio.GA.Crossover.GeneralPoint.InterleaveCrossover-class.html#__init__,Method Bio.GA.Crossover.Point.SinglePointCrossover.__init__()=Bio.GA.Crossover.Point.SinglePointCrossover-class.html#__init__,Method Bio.GA.Crossover.TwoPoint.TwoPointCrossover.__init__()=Bio.GA.Crossover.TwoPoint.TwoPointCrossover-class.html#__init__,Method Bio.GA.Crossover.Uniform.UniformCrossover.__init__()=Bio.GA.Crossover.Uniform.UniformCrossover-class.html#__init__,Method Bio.GA.Evolver.GenerationEvolver.__init__()=Bio.GA.Evolver.GenerationEvolver-class.html#__init__,Method Bio.GA.Evolver.SteadyStateEvolver.__init__()=Bio.GA.Evolver.SteadyStateEvolver-class.html#__init__,Method Bio.GA.Mutation.General.SafeFitnessMutation.__init__()=Bio.GA.Mutation.General.SafeFitnessMutation-class.html#__init__,Method Bio.GA.Mutation.Simple.ConversionMutation.__init__()=Bio.GA.Mutation.Simple.ConversionMutation-class.html#__init__,Method Bio.GA.Mutation.Simple.SinglePositionMutation.__init__()=Bio.GA.Mutation.Simple.SinglePositionMutation-class.html#__init__,Method Bio.GA.Organism.Organism.__init__()=Bio.GA.Organism.Organism-class.html#__init__,Method Bio.GA.Repair.Stabilizing.AmbiguousRepair.__init__()=Bio.GA.Repair.Stabilizing.AmbiguousRepair-class.html#__init__,Method Bio.GA.Selection.Abstract.AbstractSelection.__init__()=Bio.GA.Selection.Abstract.AbstractSelection-class.html#__init__,Method Bio.GA.Selection.Diversity.DiversitySelection.__init__()=Bio.GA.Selection.Diversity.DiversitySelection-class.html#__init__,Method Bio.GA.Selection.RouletteWheel.RouletteWheelSelection.__init__()=Bio.GA.Selection.RouletteWheel.RouletteWheelSelection-class.html#__init__,Method Bio.GA.Selection.Tournament.TournamentSelection.__init__()=Bio.GA.Selection.Tournament.TournamentSelection-class.html#__init__,Method Bio.GenBank.FeatureParser.__init__()=Bio.GenBank.FeatureParser-class.html#__init__,Method Bio.GenBank.Iterator.__init__()=Bio.GenBank.Iterator-class.html#__init__,Method Bio.GenBank.Record.Feature.__init__()=Bio.GenBank.Record.Feature-class.html#__init__,Method Bio.GenBank.Record.Qualifier.__init__()=Bio.GenBank.Record.Qualifier-class.html#__init__,Method Bio.GenBank.Record.Record.__init__()=Bio.GenBank.Record.Record-class.html#__init__,Method Bio.GenBank.Record.Reference.__init__()=Bio.GenBank.Record.Reference-class.html#__init__,Method Bio.GenBank.RecordParser.__init__()=Bio.GenBank.RecordParser-class.html#__init__,Method Bio.GenBank.Scanner.InsdcScanner.__init__()=Bio.GenBank.Scanner.InsdcScanner-class.html#__init__,Method Bio.GenBank._BaseGenBankConsumer.__init__()=Bio.GenBank._BaseGenBankConsumer-class.html#__init__,Method Bio.GenBank._FeatureConsumer.__init__()=Bio.GenBank._FeatureConsumer-class.html#__init__,Method Bio.GenBank._RecordConsumer.__init__()=Bio.GenBank._RecordConsumer-class.html#__init__,Method Bio.GenBank.utils.FeatureValueCleaner.__init__()=Bio.GenBank.utils.FeatureValueCleaner-class.html#__init__,Method Bio.Geo.Record.Record.__init__()=Bio.Geo.Record.Record-class.html#__init__,Method Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment.__init__()=Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome.Chromosome.__init__()=Bio.Graphics.BasicChromosome.Chromosome-class.html#__init__,Method Bio.Graphics.BasicChromosome.ChromosomeSegment.__init__()=Bio.Graphics.BasicChromosome.ChromosomeSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome.Organism.__init__()=Bio.Graphics.BasicChromosome.Organism-class.html#__init__,Method Bio.Graphics.BasicChromosome.TelomereSegment.__init__()=Bio.Graphics.BasicChromosome.TelomereSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome._ChromosomeComponent.__init__()=Bio.Graphics.BasicChromosome._ChromosomeComponent-class.html#__init__,Method Bio.Graphics.ColorSpiral.ColorSpiral.__init__()=Bio.Graphics.ColorSpiral.ColorSpiral-class.html#__init__,Method Bio.Graphics.Comparative.ComparativeScatterPlot.__init__()=Bio.Graphics.Comparative.ComparativeScatterPlot-class.html#__init__,Method Bio.Graphics.DisplayRepresentation.ChromosomeCounts.__init__()=Bio.Graphics.DisplayRepresentation.ChromosomeCounts-class.html#__init__,Method Bio.Graphics.Distribution.BarChartDistribution.__init__()=Bio.Graphics.Distribution.BarChartDistribution-class.html#__init__,Method Bio.Graphics.Distribution.DistributionPage.__init__()=Bio.Graphics.Distribution.DistributionPage-class.html#__init__,Method Bio.Graphics.Distribution.LineDistribution.__init__()=Bio.Graphics.Distribution.LineDistribution-class.html#__init__,Method Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.__init__()=Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.__init__()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.__init__()=Bio.Graphics.GenomeDiagram._Colors.ColorTranslator-class.html#__init__,Method Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.__init__()=Bio.Graphics.GenomeDiagram._CrossLink.CrossLink-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.__init__()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Feature.Feature.__init__()=Bio.Graphics.GenomeDiagram._Feature.Feature-class.html#__init__,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.__init__()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.__init__()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#__init__,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.__init__()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#__init__,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.__init__()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Track.Track.__init__()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#__init__,Method Bio.Graphics.KGML_vis.KGMLCanvas.__init__()=Bio.Graphics.KGML_vis.KGMLCanvas-class.html#__init__,Method Bio.HMM.DynamicProgramming.AbstractDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.AbstractDPAlgorithms-class.html#__init__,Method Bio.HMM.DynamicProgramming.LogDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.LogDPAlgorithms-class.html#__init__,Method Bio.HMM.DynamicProgramming.ScaledDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.ScaledDPAlgorithms-class.html#__init__,Method Bio.HMM.MarkovModel.HiddenMarkovModel.__init__()=Bio.HMM.MarkovModel.HiddenMarkovModel-class.html#__init__,Method Bio.HMM.MarkovModel.MarkovModelBuilder.__init__()=Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#__init__,Method Bio.HMM.Trainer.AbstractTrainer.__init__()=Bio.HMM.Trainer.AbstractTrainer-class.html#__init__,Method Bio.HMM.Trainer.BaumWelchTrainer.__init__()=Bio.HMM.Trainer.BaumWelchTrainer-class.html#__init__,Method Bio.HMM.Trainer.KnownStateTrainer.__init__()=Bio.HMM.Trainer.KnownStateTrainer-class.html#__init__,Method Bio.HMM.Trainer.TrainingSequence.__init__()=Bio.HMM.Trainer.TrainingSequence-class.html#__init__,Method Bio.Index._InMemoryIndex.__init__()=Bio.Index._InMemoryIndex-class.html#__init__,Method Bio.Index._ShelveIndex.__init__()=Bio.Index._ShelveIndex-class.html#__init__,Method Bio.KDTree.KDTree'.KDTree.__init__()=Bio.KDTree.KDTree%27.KDTree-class.html#__init__,Method Bio.KEGG.Compound.Record.__init__()=Bio.KEGG.Compound.Record-class.html#__init__,Method Bio.KEGG.Enzyme.Record.__init__()=Bio.KEGG.Enzyme.Record-class.html#__init__,Method Bio.KEGG.KGML.KGML_parser.KGMLParser.__init__()=Bio.KEGG.KGML.KGML_parser.KGMLParser-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Component.__init__()=Bio.KEGG.KGML.KGML_pathway.Component-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Entry.__init__()=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Graphics.__init__()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Pathway.__init__()=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Reaction.__init__()=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Relation.__init__()=Bio.KEGG.KGML.KGML_pathway.Relation-class.html#__init__,Method Bio.LogisticRegression.LogisticRegression.__init__()=Bio.LogisticRegression.LogisticRegression-class.html#__init__,Method Bio.MarkovModel.MarkovModel.__init__()=Bio.MarkovModel.MarkovModel-class.html#__init__,Method Bio.MaxEntropy.MaxEntropy.__init__()=Bio.MaxEntropy.MaxEntropy-class.html#__init__,Method Bio.Motif.Parsers.AlignAce.Record.__init__()=Bio.Motif.Parsers.AlignAce.Record-class.html#__init__,Method Bio.Motif.Parsers.MAST.Record.__init__()=Bio.Motif.Parsers.MAST.Record-class.html#__init__,Method Bio.Motif.Parsers.MEME.MEMEInstance.__init__()=Bio.Motif.Parsers.MEME.MEMEInstance-class.html#__init__,Method Bio.Motif.Parsers.MEME.MEMEMotif.__init__()=Bio.Motif.Parsers.MEME.MEMEMotif-class.html#__init__,Method Bio.Motif.Parsers.MEME.MEMERecord.__init__()=Bio.Motif.Parsers.MEME.MEMERecord-class.html#__init__,Method Bio.Motif.Thresholds.ScoreDistribution.__init__()=Bio.Motif.Thresholds.ScoreDistribution-class.html#__init__,Method Bio.Motif._Motif.Motif.__init__()=Bio.Motif._Motif.Motif-class.html#__init__,Method Bio.NMR.xpktools.Peaklist.__init__()=Bio.NMR.xpktools.Peaklist-class.html#__init__,Method Bio.NMR.xpktools.XpkEntry.__init__()=Bio.NMR.xpktools.XpkEntry-class.html#__init__,Method Bio.NaiveBayes.NaiveBayes.__init__()=Bio.NaiveBayes.NaiveBayes-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.AbstractLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.AbstractLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.InputLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Network.BasicNetwork.__init__()=Bio.NeuralNetwork.BackPropagation.Network.BasicNetwork-class.html#__init__,Method Bio.NeuralNetwork.Gene.Motif.MotifCoder.__init__()=Bio.NeuralNetwork.Gene.Motif.MotifCoder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Motif.MotifFinder.__init__()=Bio.NeuralNetwork.Gene.Motif.MotifFinder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Pattern.PatternIO.__init__()=Bio.NeuralNetwork.Gene.Pattern.PatternIO-class.html#__init__,Method Bio.NeuralNetwork.Gene.Pattern.PatternRepository.__init__()=Bio.NeuralNetwork.Gene.Pattern.PatternRepository-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness.__init__()=Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder.__init__()=Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness.__init__()=Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator.__init__()=Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.Schema.__init__()=Bio.NeuralNetwork.Gene.Schema.Schema-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SchemaCoder.__init__()=Bio.NeuralNetwork.Gene.Schema.SchemaCoder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SchemaFactory.__init__()=Bio.NeuralNetwork.Gene.Schema.SchemaFactory-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SchemaFinder.__init__()=Bio.NeuralNetwork.Gene.Schema.SchemaFinder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SimpleFinisher.__init__()=Bio.NeuralNetwork.Gene.Schema.SimpleFinisher-class.html#__init__,Method Bio.NeuralNetwork.Gene.Signature.SignatureCoder.__init__()=Bio.NeuralNetwork.Gene.Signature.SignatureCoder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Signature.SignatureFinder.__init__()=Bio.NeuralNetwork.Gene.Signature.SignatureFinder-class.html#__init__,Method Bio.NeuralNetwork.StopTraining.ValidationIncreaseStop.__init__()=Bio.NeuralNetwork.StopTraining.ValidationIncreaseStop-class.html#__init__,Method Bio.NeuralNetwork.Training.ExampleManager.__init__()=Bio.NeuralNetwork.Training.ExampleManager-class.html#__init__,Method Bio.NeuralNetwork.Training.TrainingExample.__init__()=Bio.NeuralNetwork.Training.TrainingExample-class.html#__init__,Method Bio.Nexus.Nexus.Block.__init__()=Bio.Nexus.Nexus.Block-class.html#__init__,Method Bio.Nexus.Nexus.CharBuffer.__init__()=Bio.Nexus.Nexus.CharBuffer-class.html#__init__,Method Bio.Nexus.Nexus.Commandline.__init__()=Bio.Nexus.Nexus.Commandline-class.html#__init__,Method Bio.Nexus.Nexus.Nexus.__init__()=Bio.Nexus.Nexus.Nexus-class.html#__init__,Method Bio.Nexus.Nexus.StepMatrix.__init__()=Bio.Nexus.Nexus.StepMatrix-class.html#__init__,Method Bio.Nexus.Nodes.Chain.__init__()=Bio.Nexus.Nodes.Chain-class.html#__init__,Method Bio.Nexus.Nodes.Node.__init__()=Bio.Nexus.Nodes.Node-class.html#__init__,Method Bio.Nexus.Trees.NodeData.__init__()=Bio.Nexus.Trees.NodeData-class.html#__init__,Method Bio.Nexus.Trees.Tree.__init__()=Bio.Nexus.Trees.Tree-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractPropertyMap-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap-class.html#__init__,Method Bio.PDB.Atom.Atom.__init__()=Bio.PDB.Atom.Atom-class.html#__init__,Method Bio.PDB.Atom.DisorderedAtom.__init__()=Bio.PDB.Atom.DisorderedAtom-class.html#__init__,Method Bio.PDB.Chain.Chain.__init__()=Bio.PDB.Chain.Chain-class.html#__init__,Method Bio.PDB.DSSP'.DSSP.__init__()=Bio.PDB.DSSP%27.DSSP-class.html#__init__,Method Bio.PDB.Dice.ChainSelector.__init__()=Bio.PDB.Dice.ChainSelector-class.html#__init__,Method Bio.PDB.Entity.DisorderedEntityWrapper.__init__()=Bio.PDB.Entity.DisorderedEntityWrapper-class.html#__init__,Method Bio.PDB.Entity.Entity.__init__()=Bio.PDB.Entity.Entity-class.html#__init__,Method Bio.PDB.FragmentMapper'.Fragment.__init__()=Bio.PDB.FragmentMapper%27.Fragment-class.html#__init__,Method Bio.PDB.FragmentMapper'.FragmentMapper.__init__()=Bio.PDB.FragmentMapper%27.FragmentMapper-class.html#__init__,Method Bio.PDB.HSExposure.ExposureCN.__init__()=Bio.PDB.HSExposure.ExposureCN-class.html#__init__,Method Bio.PDB.HSExposure.HSExposureCA.__init__()=Bio.PDB.HSExposure.HSExposureCA-class.html#__init__,Method Bio.PDB.HSExposure.HSExposureCB.__init__()=Bio.PDB.HSExposure.HSExposureCB-class.html#__init__,Method Bio.PDB.HSExposure._AbstractHSExposure.__init__()=Bio.PDB.HSExposure._AbstractHSExposure-class.html#__init__,Method Bio.PDB.MMCIF2Dict.MMCIF2Dict.__init__()=Bio.PDB.MMCIF2Dict.MMCIF2Dict-class.html#__init__,Method Bio.PDB.MMCIFParser'.MMCIFParser.__init__()=Bio.PDB.MMCIFParser%27.MMCIFParser-class.html#__init__,Method Bio.PDB.Model.Model.__init__()=Bio.PDB.Model.Model-class.html#__init__,Method Bio.PDB.NACCESS.NACCESS.__init__()=Bio.PDB.NACCESS.NACCESS-class.html#__init__,Method Bio.PDB.NACCESS.NACCESS_atomic.__init__()=Bio.PDB.NACCESS.NACCESS_atomic-class.html#__init__,Method Bio.PDB.NeighborSearch'.NeighborSearch.__init__()=Bio.PDB.NeighborSearch%27.NeighborSearch-class.html#__init__,Method Bio.PDB.PDBIO'.PDBIO.__init__()=Bio.PDB.PDBIO%27.PDBIO-class.html#__init__,Method Bio.PDB.PDBList'.PDBList.__init__()=Bio.PDB.PDBList%27.PDBList-class.html#__init__,Method Bio.PDB.PDBParser'.PDBParser.__init__()=Bio.PDB.PDBParser%27.PDBParser-class.html#__init__,Method Bio.PDB.PSEA.PSEA.__init__()=Bio.PDB.PSEA.PSEA-class.html#__init__,Method Bio.PDB.Polypeptide.CaPPBuilder.__init__()=Bio.PDB.Polypeptide.CaPPBuilder-class.html#__init__,Method Bio.PDB.Polypeptide.PPBuilder.__init__()=Bio.PDB.Polypeptide.PPBuilder-class.html#__init__,Method Bio.PDB.Polypeptide._PPBuilder.__init__()=Bio.PDB.Polypeptide._PPBuilder-class.html#__init__,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.__init__()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#__init__,Method Bio.PDB.Residue.DisorderedResidue.__init__()=Bio.PDB.Residue.DisorderedResidue-class.html#__init__,Method Bio.PDB.Residue.Residue.__init__()=Bio.PDB.Residue.Residue-class.html#__init__,Method Bio.PDB.ResidueDepth'.ResidueDepth.__init__()=Bio.PDB.ResidueDepth%27.ResidueDepth-class.html#__init__,Method Bio.PDB.Structure.Structure.__init__()=Bio.PDB.Structure.Structure-class.html#__init__,Method Bio.PDB.StructureAlignment'.StructureAlignment.__init__()=Bio.PDB.StructureAlignment%27.StructureAlignment-class.html#__init__,Method Bio.PDB.StructureBuilder.StructureBuilder.__init__()=Bio.PDB.StructureBuilder.StructureBuilder-class.html#__init__,Method Bio.PDB.Superimposer'.Superimposer.__init__()=Bio.PDB.Superimposer%27.Superimposer-class.html#__init__,Method Bio.PDB.Vector'.Vector.__init__()=Bio.PDB.Vector%27.Vector-class.html#__init__,Method Bio.ParserSupport.EventGenerator.__init__()=Bio.ParserSupport.EventGenerator-class.html#__init__,Method Bio.ParserSupport.TaggingConsumer.__init__()=Bio.ParserSupport.TaggingConsumer-class.html#__init__,Method Bio.Pathway.Network.__init__()=Bio.Pathway.Network-class.html#__init__,Method Bio.Pathway.Reaction.__init__()=Bio.Pathway.Reaction-class.html#__init__,Method Bio.Pathway.Rep.Graph.Graph.__init__()=Bio.Pathway.Rep.Graph.Graph-class.html#__init__,Method Bio.Pathway.Rep.MultiGraph.MultiGraph.__init__()=Bio.Pathway.Rep.MultiGraph.MultiGraph-class.html#__init__,Method Bio.Pathway.System.__init__()=Bio.Pathway.System-class.html#__init__,Method Bio.Phylo.Applications._Fasttree.FastTreeCommandline.__init__()=Bio.Phylo.Applications._Fasttree.FastTreeCommandline-class.html#__init__,Method Bio.Phylo.Applications._Phyml.PhymlCommandline.__init__()=Bio.Phylo.Applications._Phyml.PhymlCommandline-class.html#__init__,Method Bio.Phylo.Applications._Raxml.RaxmlCommandline.__init__()=Bio.Phylo.Applications._Raxml.RaxmlCommandline-class.html#__init__,Method Bio.Phylo.BaseTree.BranchColor.__init__()=Bio.Phylo.BaseTree.BranchColor-class.html#__init__,Method Bio.Phylo.BaseTree.Clade.__init__()=Bio.Phylo.BaseTree.Clade-class.html#__init__,Method Bio.Phylo.BaseTree.Tree.__init__()=Bio.Phylo.BaseTree.Tree-class.html#__init__,Method Bio.Phylo.CDAO.Clade.__init__()=Bio.Phylo.CDAO.Clade-class.html#__init__,Method Bio.Phylo.CDAO.Tree.__init__()=Bio.Phylo.CDAO.Tree-class.html#__init__,Method Bio.Phylo.CDAOIO.Parser.__init__()=Bio.Phylo.CDAOIO.Parser-class.html#__init__,Method Bio.Phylo.CDAOIO.Writer.__init__()=Bio.Phylo.CDAOIO.Writer-class.html#__init__,Method Bio.Phylo.NeXML.Clade.__init__()=Bio.Phylo.NeXML.Clade-class.html#__init__,Method Bio.Phylo.NeXML.Tree.__init__()=Bio.Phylo.NeXML.Tree-class.html#__init__,Method Bio.Phylo.NeXMLIO.Parser.__init__()=Bio.Phylo.NeXMLIO.Parser-class.html#__init__,Method Bio.Phylo.NeXMLIO.Writer.__init__()=Bio.Phylo.NeXMLIO.Writer-class.html#__init__,Method Bio.Phylo.Newick.Clade.__init__()=Bio.Phylo.Newick.Clade-class.html#__init__,Method Bio.Phylo.Newick.Tree.__init__()=Bio.Phylo.Newick.Tree-class.html#__init__,Method Bio.Phylo.NewickIO.Parser.__init__()=Bio.Phylo.NewickIO.Parser-class.html#__init__,Method Bio.Phylo.NewickIO.Writer.__init__()=Bio.Phylo.NewickIO.Writer-class.html#__init__,Method Bio.Phylo.PAML._paml.Paml.__init__()=Bio.Phylo.PAML._paml.Paml-class.html#__init__,Method Bio.Phylo.PAML.baseml.Baseml.__init__()=Bio.Phylo.PAML.baseml.Baseml-class.html#__init__,Method Bio.Phylo.PAML.codeml.Codeml.__init__()=Bio.Phylo.PAML.codeml.Codeml-class.html#__init__,Method Bio.Phylo.PAML.yn00.Yn00.__init__()=Bio.Phylo.PAML.yn00.Yn00-class.html#__init__,Method Bio.Phylo.PhyloXML.Accession.__init__()=Bio.Phylo.PhyloXML.Accession-class.html#__init__,Method Bio.Phylo.PhyloXML.Annotation.__init__()=Bio.Phylo.PhyloXML.Annotation-class.html#__init__,Method Bio.Phylo.PhyloXML.BinaryCharacters.__init__()=Bio.Phylo.PhyloXML.BinaryCharacters-class.html#__init__,Method Bio.Phylo.PhyloXML.BranchColor.__init__()=Bio.Phylo.PhyloXML.BranchColor-class.html#__init__,Method Bio.Phylo.PhyloXML.Clade.__init__()=Bio.Phylo.PhyloXML.Clade-class.html#__init__,Method Bio.Phylo.PhyloXML.CladeRelation.__init__()=Bio.Phylo.PhyloXML.CladeRelation-class.html#__init__,Method Bio.Phylo.PhyloXML.Confidence.__init__()=Bio.Phylo.PhyloXML.Confidence-class.html#__init__,Method Bio.Phylo.PhyloXML.Date.__init__()=Bio.Phylo.PhyloXML.Date-class.html#__init__,Method Bio.Phylo.PhyloXML.Distribution.__init__()=Bio.Phylo.PhyloXML.Distribution-class.html#__init__,Method Bio.Phylo.PhyloXML.DomainArchitecture.__init__()=Bio.Phylo.PhyloXML.DomainArchitecture-class.html#__init__,Method Bio.Phylo.PhyloXML.Events.__init__()=Bio.Phylo.PhyloXML.Events-class.html#__init__,Method Bio.Phylo.PhyloXML.Id.__init__()=Bio.Phylo.PhyloXML.Id-class.html#__init__,Method Bio.Phylo.PhyloXML.MolSeq.__init__()=Bio.Phylo.PhyloXML.MolSeq-class.html#__init__,Method Bio.Phylo.PhyloXML.Other.__init__()=Bio.Phylo.PhyloXML.Other-class.html#__init__,Method Bio.Phylo.PhyloXML.Phylogeny.__init__()=Bio.Phylo.PhyloXML.Phylogeny-class.html#__init__,Method Bio.Phylo.PhyloXML.Phyloxml.__init__()=Bio.Phylo.PhyloXML.Phyloxml-class.html#__init__,Method Bio.Phylo.PhyloXML.Point.__init__()=Bio.Phylo.PhyloXML.Point-class.html#__init__,Method Bio.Phylo.PhyloXML.Polygon.__init__()=Bio.Phylo.PhyloXML.Polygon-class.html#__init__,Method Bio.Phylo.PhyloXML.Property.__init__()=Bio.Phylo.PhyloXML.Property-class.html#__init__,Method Bio.Phylo.PhyloXML.ProteinDomain.__init__()=Bio.Phylo.PhyloXML.ProteinDomain-class.html#__init__,Method Bio.Phylo.PhyloXML.Reference.__init__()=Bio.Phylo.PhyloXML.Reference-class.html#__init__,Method Bio.Phylo.PhyloXML.Sequence.__init__()=Bio.Phylo.PhyloXML.Sequence-class.html#__init__,Method Bio.Phylo.PhyloXML.SequenceRelation.__init__()=Bio.Phylo.PhyloXML.SequenceRelation-class.html#__init__,Method Bio.Phylo.PhyloXML.Taxonomy.__init__()=Bio.Phylo.PhyloXML.Taxonomy-class.html#__init__,Method Bio.Phylo.PhyloXML.Uri.__init__()=Bio.Phylo.PhyloXML.Uri-class.html#__init__,Method Bio.Phylo.PhyloXMLIO.Parser.__init__()=Bio.Phylo.PhyloXMLIO.Parser-class.html#__init__,Method Bio.Phylo.PhyloXMLIO.Writer.__init__()=Bio.Phylo.PhyloXMLIO.Writer-class.html#__init__,Method Bio.Phylo.TreeConstruction.DistanceCalculator.__init__()=Bio.Phylo.TreeConstruction.DistanceCalculator-class.html#__init__,Method Bio.Phylo.TreeConstruction.DistanceTreeConstructor.__init__()=Bio.Phylo.TreeConstruction.DistanceTreeConstructor-class.html#__init__,Method Bio.Phylo.TreeConstruction.NNITreeSearcher.__init__()=Bio.Phylo.TreeConstruction.NNITreeSearcher-class.html#__init__,Method Bio.Phylo.TreeConstruction.ParsimonyScorer.__init__()=Bio.Phylo.TreeConstruction.ParsimonyScorer-class.html#__init__,Method Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor.__init__()=Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor-class.html#__init__,Method Bio.Phylo.TreeConstruction._DistanceMatrix.__init__()=Bio.Phylo.TreeConstruction._DistanceMatrix-class.html#__init__,Method Bio.Phylo.TreeConstruction._Matrix.__init__()=Bio.Phylo.TreeConstruction._Matrix-class.html#__init__,Method Bio.PopGen.Async.Async.__init__()=Bio.PopGen.Async.Async-class.html#__init__,Method Bio.PopGen.Async.DirectoryRetriever.__init__()=Bio.PopGen.Async.DirectoryRetriever-class.html#__init__,Method Bio.PopGen.Async.FileRetriever.__init__()=Bio.PopGen.Async.FileRetriever-class.html#__init__,Method Bio.PopGen.Async.Local.Local.__init__()=Bio.PopGen.Async.Local.Local-class.html#__init__,Method Bio.PopGen.FDist.Async.FDistAsync.__init__()=Bio.PopGen.FDist.Async.FDistAsync-class.html#__init__,Method Bio.PopGen.FDist.Async.SplitFDist.__init__()=Bio.PopGen.FDist.Async.SplitFDist-class.html#__init__,Method Bio.PopGen.FDist.Controller.FDistController.__init__()=Bio.PopGen.FDist.Controller.FDistController-class.html#__init__,Method Bio.PopGen.FDist.Record.__init__()=Bio.PopGen.FDist.Record-class.html#__init__,Method Bio.PopGen.GenePop.Controller.GenePopController.__init__()=Bio.PopGen.GenePop.Controller.GenePopController-class.html#__init__,Method Bio.PopGen.GenePop.Controller._FileIterator.__init__()=Bio.PopGen.GenePop.Controller._FileIterator-class.html#__init__,Method Bio.PopGen.GenePop.Controller._GenePopCommandline.__init__()=Bio.PopGen.GenePop.Controller._GenePopCommandline-class.html#__init__,Method Bio.PopGen.GenePop.EasyController.EasyController.__init__()=Bio.PopGen.GenePop.EasyController.EasyController-class.html#__init__,Method Bio.PopGen.GenePop.FileParser.FileRecord.__init__()=Bio.PopGen.GenePop.FileParser.FileRecord-class.html#__init__,Method Bio.PopGen.GenePop.LargeFileParser.Record.__init__()=Bio.PopGen.GenePop.LargeFileParser.Record-class.html#__init__,Method Bio.PopGen.GenePop.Record.__init__()=Bio.PopGen.GenePop.Record-class.html#__init__,Method Bio.PopGen.SimCoal.Async.SimCoalCache.__init__()=Bio.PopGen.SimCoal.Async.SimCoalCache-class.html#__init__,Method Bio.PopGen.SimCoal.Cache.SimCoalCache.__init__()=Bio.PopGen.SimCoal.Cache.SimCoalCache-class.html#__init__,Method Bio.PopGen.SimCoal.Controller.FastSimCoalController.__init__()=Bio.PopGen.SimCoal.Controller.FastSimCoalController-class.html#__init__,Method Bio.PopGen.SimCoal.Controller.SimCoalController.__init__()=Bio.PopGen.SimCoal.Controller.SimCoalController-class.html#__init__,Method Bio.PopGen.SimCoal.Controller._FastSimCoalCommandLine.__init__()=Bio.PopGen.SimCoal.Controller._FastSimCoalCommandLine-class.html#__init__,Method Bio.Restriction.PrintFormat.PrintFormat.__init__()=Bio.Restriction.PrintFormat.PrintFormat-class.html#__init__,Method Bio.Restriction.Restriction.Analysis.__init__()=Bio.Restriction.Restriction.Analysis-class.html#__init__,Method Bio.Restriction.Restriction.FormattedSeq.__init__()=Bio.Restriction.Restriction.FormattedSeq-class.html#__init__,Method Bio.Restriction.Restriction.RestrictionBatch.__init__()=Bio.Restriction.Restriction.RestrictionBatch-class.html#__init__,Method Bio.Restriction.Restriction.RestrictionType.__init__()=Bio.Restriction.Restriction.RestrictionType-class.html#__init__,Method Bio.SCOP.Astral.__init__()=Bio.SCOP.Astral-class.html#__init__,Method Bio.SCOP.Cla.Index.__init__()=Bio.SCOP.Cla.Index-class.html#__init__,Method Bio.SCOP.Cla.Record.__init__()=Bio.SCOP.Cla.Record-class.html#__init__,Method Bio.SCOP.Des.Record.__init__()=Bio.SCOP.Des.Record-class.html#__init__,Method Bio.SCOP.Dom.Record.__init__()=Bio.SCOP.Dom.Record-class.html#__init__,Method Bio.SCOP.Domain.__init__()=Bio.SCOP.Domain-class.html#__init__,Method Bio.SCOP.Hie.Record.__init__()=Bio.SCOP.Hie.Record-class.html#__init__,Method Bio.SCOP.Node.__init__()=Bio.SCOP.Node-class.html#__init__,Method Bio.SCOP.Raf.Res.__init__()=Bio.SCOP.Raf.Res-class.html#__init__,Method Bio.SCOP.Raf.SeqMap.__init__()=Bio.SCOP.Raf.SeqMap-class.html#__init__,Method Bio.SCOP.Raf.SeqMapIndex.__init__()=Bio.SCOP.Raf.SeqMapIndex-class.html#__init__,Method Bio.SCOP.Residues.Residues.__init__()=Bio.SCOP.Residues.Residues-class.html#__init__,Method Bio.SCOP.Scop.__init__()=Bio.SCOP.Scop-class.html#__init__,Method Bio.SVDSuperimposer.SVDSuperimposer.__init__()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabParser.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_text.BlastTextParser.__init__()=Bio.SearchIO.BlastIO.blast_text.BlastTextParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator.__init__()=Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslIndexer.__init__()=Bio.SearchIO.BlatIO.BlatPslIndexer-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslParser.__init__()=Bio.SearchIO.BlatIO.BlatPslParser-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslWriter.__init__()=Bio.SearchIO.BlatIO.BlatPslWriter-class.html#__init__,Method Bio.SearchIO.ExonerateIO._base._BaseExonerateParser.__init__()=Bio.SearchIO.ExonerateIO._base._BaseExonerateParser-class.html#__init__,Method Bio.SearchIO.FastaIO.FastaM10Indexer.__init__()=Bio.SearchIO.FastaIO.FastaM10Indexer-class.html#__init__,Method Bio.SearchIO.FastaIO.FastaM10Parser.__init__()=Bio.SearchIO.FastaIO.FastaM10Parser-class.html#__init__,Method Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer.__init__()=Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser.__init__()=Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.__init__()=Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser.__init__()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.__init__()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser.__init__()=Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html#__init__,Method Bio.SearchIO._index.SearchIndexer.__init__()=Bio.SearchIO._index.SearchIndexer-class.html#__init__,Method Bio.SearchIO._model.hit.Hit.__init__()=Bio.SearchIO._model.hit.Hit-class.html#__init__,Method Bio.SearchIO._model.hsp.HSP.__init__()=Bio.SearchIO._model.hsp.HSP-class.html#__init__,Method Bio.SearchIO._model.hsp.HSPFragment.__init__()=Bio.SearchIO._model.hsp.HSPFragment-class.html#__init__,Method Bio.SearchIO._model.query.QueryResult.__init__()=Bio.SearchIO._model.query.QueryResult-class.html#__init__,Method Bio.Seq.MutableSeq.__init__()=Bio.Seq.MutableSeq-class.html#__init__,Method Bio.Seq.Seq.__init__()=Bio.Seq.Seq-class.html#__init__,Method Bio.Seq.UnknownSeq.__init__()=Bio.Seq.UnknownSeq-class.html#__init__,Method Bio.SeqFeature.CompoundLocation.__init__()=Bio.SeqFeature.CompoundLocation-class.html#__init__,Method Bio.SeqFeature.FeatureLocation.__init__()=Bio.SeqFeature.FeatureLocation-class.html#__init__,Method Bio.SeqFeature.PositionGap.__init__()=Bio.SeqFeature.PositionGap-class.html#__init__,Method Bio.SeqFeature.Reference.__init__()=Bio.SeqFeature.Reference-class.html#__init__,Method Bio.SeqFeature.SeqFeature.__init__()=Bio.SeqFeature.SeqFeature-class.html#__init__,Method Bio.SeqIO.FastaIO.FastaWriter.__init__()=Bio.SeqIO.FastaIO.FastaWriter-class.html#__init__,Method Bio.SeqIO.Interfaces.SequenceIterator.__init__()=Bio.SeqIO.Interfaces.SequenceIterator-class.html#__init__,Method Bio.SeqIO.Interfaces.SequenceWriter.__init__()=Bio.SeqIO.Interfaces.SequenceWriter-class.html#__init__,Method Bio.SeqIO.Interfaces.SequentialSequenceWriter.__init__()=Bio.SeqIO.Interfaces.SequentialSequenceWriter-class.html#__init__,Method Bio.SeqIO.PhdIO.PhdWriter.__init__()=Bio.SeqIO.PhdIO.PhdWriter-class.html#__init__,Method Bio.SeqIO.QualityIO.QualPhredWriter.__init__()=Bio.SeqIO.QualityIO.QualPhredWriter-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.SeqXmlIterator.__init__()=Bio.SeqIO.SeqXmlIO.SeqXmlIterator-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.SeqXmlWriter.__init__()=Bio.SeqIO.SeqXmlIO.SeqXmlWriter-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.XMLRecordIterator.__init__()=Bio.SeqIO.SeqXmlIO.XMLRecordIterator-class.html#__init__,Method Bio.SeqIO.SffIO.SffWriter.__init__()=Bio.SeqIO.SffIO.SffWriter-class.html#__init__,Method Bio.SeqIO.SffIO._AddTellHandle.__init__()=Bio.SeqIO.SffIO._AddTellHandle-class.html#__init__,Method Bio.SeqIO.UniprotIO.Parser.__init__()=Bio.SeqIO.UniprotIO.Parser-class.html#__init__,Method Bio.SeqIO._index.IntelliGeneticsRandomAccess.__init__()=Bio.SeqIO._index.IntelliGeneticsRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SeqFileRandomAccess.__init__()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SequentialSeqFileRandomAccess.__init__()=Bio.SeqIO._index.SequentialSeqFileRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SffRandomAccess.__init__()=Bio.SeqIO._index.SffRandomAccess-class.html#__init__,Method Bio.SeqRecord.SeqRecord.__init__()=Bio.SeqRecord.SeqRecord-class.html#__init__,Method Bio.SeqRecord._RestrictedDict.__init__()=Bio.SeqRecord._RestrictedDict-class.html#__init__,Method Bio.SeqUtils.CodonUsage.CodonAdaptationIndex.__init__()=Bio.SeqUtils.CodonUsage.CodonAdaptationIndex-class.html#__init__,Method Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.__init__()=Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint-class.html#__init__,Method Bio.SeqUtils.ProtParam.ProteinAnalysis.__init__()=Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#__init__,Method Bio.Sequencing.Ace.ACEFileRecord.__init__()=Bio.Sequencing.Ace.ACEFileRecord-class.html#__init__,Method Bio.Sequencing.Ace.Contig.__init__()=Bio.Sequencing.Ace.Contig-class.html#__init__,Method Bio.Sequencing.Ace.Reads.__init__()=Bio.Sequencing.Ace.Reads-class.html#__init__,Method Bio.Sequencing.Ace.af.__init__()=Bio.Sequencing.Ace.af-class.html#__init__,Method Bio.Sequencing.Ace.bs.__init__()=Bio.Sequencing.Ace.bs-class.html#__init__,Method Bio.Sequencing.Ace.ct.__init__()=Bio.Sequencing.Ace.ct-class.html#__init__,Method Bio.Sequencing.Ace.ds.__init__()=Bio.Sequencing.Ace.ds-class.html#__init__,Method Bio.Sequencing.Ace.qa.__init__()=Bio.Sequencing.Ace.qa-class.html#__init__,Method Bio.Sequencing.Ace.rd.__init__()=Bio.Sequencing.Ace.rd-class.html#__init__,Method Bio.Sequencing.Ace.rt.__init__()=Bio.Sequencing.Ace.rt-class.html#__init__,Method Bio.Sequencing.Ace.wa.__init__()=Bio.Sequencing.Ace.wa-class.html#__init__,Method Bio.Sequencing.Ace.wr.__init__()=Bio.Sequencing.Ace.wr-class.html#__init__,Method Bio.Sequencing.Applications._Novoalign.NovoalignCommandline.__init__()=Bio.Sequencing.Applications._Novoalign.NovoalignCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaAlignCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaAlignCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaBwaswCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaBwaswCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaIndexCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaIndexCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaSampeCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaSampeCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaSamseCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaSamseCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsSortCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsSortCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline-class.html#__init__,Method Bio.Sequencing.Phd.Record.__init__()=Bio.Sequencing.Phd.Record-class.html#__init__,Method Bio.SubsMat.FreqTable.FreqTable.__init__()=Bio.SubsMat.FreqTable.FreqTable-class.html#__init__,Method Bio.SubsMat.SeqMat.__init__()=Bio.SubsMat.SeqMat-class.html#__init__,Method Bio.SwissProt.KeyWList.Record.__init__()=Bio.SwissProt.KeyWList.Record-class.html#__init__,Method Bio.SwissProt.Record.__init__()=Bio.SwissProt.Record-class.html#__init__,Method Bio.SwissProt.Reference.__init__()=Bio.SwissProt.Reference-class.html#__init__,Method Bio.UniGene.ProtsimLine.__init__()=Bio.UniGene.ProtsimLine-class.html#__init__,Method Bio.UniGene.Record.__init__()=Bio.UniGene.Record-class.html#__init__,Method Bio.UniGene.STSLine.__init__()=Bio.UniGene.STSLine-class.html#__init__,Method Bio.UniGene.SequenceLine.__init__()=Bio.UniGene.SequenceLine-class.html#__init__,Method Bio.Wise.dnal.Statistics.__init__()=Bio.Wise.dnal.Statistics-class.html#__init__,Method Bio.Wise.psw.AlignmentColumn.__init__()=Bio.Wise.psw.AlignmentColumn-class.html#__init__,Method Bio.Wise.psw.ColumnUnit.__init__()=Bio.Wise.psw.ColumnUnit-class.html#__init__,Method Bio._py3k._ordereddict.OrderedDict.__init__()=Bio._py3k._ordereddict.OrderedDict-class.html#__init__,Method Bio.bgzf.BgzfReader.__init__()=Bio.bgzf.BgzfReader-class.html#__init__,Method Bio.bgzf.BgzfWriter.__init__()=Bio.bgzf.BgzfWriter-class.html#__init__,Method Bio.codonalign.codonalignment.CodonAlignment.__init__()=Bio.codonalign.codonalignment.CodonAlignment-class.html#__init__,Method Bio.codonalign.codonseq.CodonSeq.__init__()=Bio.codonalign.codonseq.CodonSeq-class.html#__init__,Method Bio.kNN.kNN.__init__()=Bio.kNN.kNN-class.html#__init__,Method Bio.motifs.Instances.__init__()=Bio.motifs.Instances-class.html#__init__,Method Bio.motifs.Motif.__init__()=Bio.motifs.Motif-class.html#__init__,Method Bio.motifs.alignace.Record.__init__()=Bio.motifs.alignace.Record-class.html#__init__,Method Bio.motifs.applications._alignace.AlignAceCommandline.__init__()=Bio.motifs.applications._alignace.AlignAceCommandline-class.html#__init__,Method Bio.motifs.applications._alignace.CompareAceCommandline.__init__()=Bio.motifs.applications._alignace.CompareAceCommandline-class.html#__init__,Method Bio.motifs.applications._xxmotif.XXmotifCommandline.__init__()=Bio.motifs.applications._xxmotif.XXmotifCommandline-class.html#__init__,Method Bio.motifs.jaspar.Motif.__init__()=Bio.motifs.jaspar.Motif-class.html#__init__,Method Bio.motifs.jaspar.Record.__init__()=Bio.motifs.jaspar.Record-class.html#__init__,Method Bio.motifs.jaspar.db.JASPAR5.__init__()=Bio.motifs.jaspar.db.JASPAR5-class.html#__init__,Method Bio.motifs.mast.Record.__init__()=Bio.motifs.mast.Record-class.html#__init__,Method Bio.motifs.matrix.GenericPositionMatrix.__init__()=Bio.motifs.matrix.GenericPositionMatrix-class.html#__init__,Method Bio.motifs.matrix.PositionWeightMatrix.__init__()=Bio.motifs.matrix.PositionWeightMatrix-class.html#__init__,Method Bio.motifs.meme.Instance.__init__()=Bio.motifs.meme.Instance-class.html#__init__,Method Bio.motifs.meme.Motif.__init__()=Bio.motifs.meme.Motif-class.html#__init__,Method Bio.motifs.meme.Record.__init__()=Bio.motifs.meme.Record-class.html#__init__,Method Bio.motifs.thresholds.ScoreDistribution.__init__()=Bio.motifs.thresholds.ScoreDistribution-class.html#__init__,Method Bio.motifs.transfac.Record.__init__()=Bio.motifs.transfac.Record-class.html#__init__,Method Bio.pairwise2.affine_penalty.__init__()=Bio.pairwise2.affine_penalty-class.html#__init__,Method Bio.pairwise2.dictionary_match.__init__()=Bio.pairwise2.dictionary_match-class.html#__init__,Method Bio.pairwise2.identity_match.__init__()=Bio.pairwise2.identity_match-class.html#__init__,Method BioSQL.BioSeq.DBSeq.__init__()=BioSQL.BioSeq.DBSeq-class.html#__init__,Method BioSQL.BioSeq.DBSeqRecord.__init__()=BioSQL.BioSeq.DBSeqRecord-class.html#__init__,Method BioSQL.BioSeqDatabase.Adaptor.__init__()=BioSQL.BioSeqDatabase.Adaptor-class.html#__init__,Method BioSQL.BioSeqDatabase.BioSeqDatabase.__init__()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#__init__,Method BioSQL.BioSeqDatabase.DBServer.__init__()=BioSQL.BioSeqDatabase.DBServer-class.html#__init__,Method BioSQL.BioSeqDatabase._CursorWrapper.__init__()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#__init__,Method BioSQL.DBUtils.Generic_dbutils.__init__()=BioSQL.DBUtils.Generic_dbutils-class.html#__init__,Method BioSQL.Loader.DatabaseLoader.__init__()=BioSQL.Loader.DatabaseLoader-class.html#__init__,Method BioSQL.Loader.DatabaseRemover.__init__()=BioSQL.Loader.DatabaseRemover-class.html#__init__"><a title="Bio.Affy.CelFile.Record.__init__
Bio.Align.AlignInfo.PSSM.__init__
Bio.Align.AlignInfo.SummaryInfo.__init__
Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline.__init__
Bio.Align.Applications._Clustalw.ClustalwCommandline.__init__
Bio.Align.Applications._Dialign.DialignCommandline.__init__
Bio.Align.Applications._MSAProbs.MSAProbsCommandline.__init__
Bio.Align.Applications._Mafft.MafftCommandline.__init__
Bio.Align.Applications._Muscle.MuscleCommandline.__init__
Bio.Align.Applications._Prank.PrankCommandline.__init__
Bio.Align.Applications._Probcons.ProbconsCommandline.__init__
Bio.Align.Applications._TCoffee.TCoffeeCommandline.__init__
Bio.Align.Generic.Alignment.__init__
Bio.Align.MultipleSeqAlignment.__init__
Bio.AlignIO.Interfaces.AlignmentIterator.__init__
Bio.AlignIO.Interfaces.AlignmentWriter.__init__
Bio.AlignIO.Interfaces.SequentialAlignmentWriter.__init__
Bio.Alphabet.AlphabetEncoder.__init__
Bio.Alphabet.Gapped.__init__
Bio.Alphabet.HasStopCodon.__init__
Bio.Application.AbstractCommandline.__init__
Bio.Application.ApplicationError.__init__
Bio.Application._AbstractParameter.__init__
Bio.Application._Argument.__init__
Bio.Application._Option.__init__
Bio.Application._StaticArgument.__init__
Bio.Application._Switch.__init__
Bio.Blast.Applications.NcbiblastformatterCommandline.__init__
Bio.Blast.Applications.NcbiblastnCommandline.__init__
Bio.Blast.Applications.NcbiblastpCommandline.__init__
Bio.Blast.Applications.NcbiblastxCommandline.__init__
Bio.Blast.Applications.NcbideltablastCommandline.__init__
Bio.Blast.Applications.NcbipsiblastCommandline.__init__
Bio.Blast.Applications.NcbirpsblastCommandline.__init__
Bio.Blast.Applications.NcbirpstblastnCommandline.__init__
Bio.Blast.Applications.NcbitblastnCommandline.__init__
Bio.Blast.Applications.NcbitblastxCommandline.__init__
Bio.Blast.Applications._NcbibaseblastCommandline.__init__
Bio.Blast.Applications._Ncbiblast2SeqCommandline.__init__
Bio.Blast.Applications._NcbiblastCommandline.__init__
Bio.Blast.Applications._NcbiblastMain2SeqCommandline.__init__
Bio.Blast.NCBIStandalone.BlastErrorParser.__init__
Bio.Blast.NCBIStandalone.BlastParser.__init__
Bio.Blast.NCBIStandalone.Iterator.__init__
Bio.Blast.NCBIStandalone.PSIBlastParser.__init__
Bio.Blast.NCBIStandalone._BlastConsumer.__init__
Bio.Blast.NCBIStandalone._BlastErrorConsumer.__init__
Bio.Blast.NCBIStandalone._PSIBlastConsumer.__init__
Bio.Blast.NCBIXML.BlastParser.__init__
Bio.Blast.NCBIXML._XMLparser.__init__
Bio.Blast.ParseBlastTable.BlastTableEntry.__init__
Bio.Blast.ParseBlastTable.BlastTableReader.__init__
Bio.Blast.ParseBlastTable.BlastTableRec.__init__
Bio.Blast.Record.Alignment.__init__
Bio.Blast.Record.Blast.__init__
Bio.Blast.Record.DatabaseReport.__init__
Bio.Blast.Record.Description.__init__
Bio.Blast.Record.HSP.__init__
Bio.Blast.Record.Header.__init__
Bio.Blast.Record.MultipleAlignment.__init__
Bio.Blast.Record.PSIBlast.__init__
Bio.Blast.Record.Parameters.__init__
Bio.Blast.Record.Round.__init__
Bio.CAPS.CAPSMap.__init__
Bio.CAPS.DifferentialCutsite.__init__
Bio.Cluster.Record.__init__
Bio.Compass.Record.__init__
Bio.Crystal.Chain.__init__
Bio.Crystal.Crystal.__init__
Bio.Crystal.Hetero.__init__
Bio.Data.CodonTable.AmbiguousCodonTable.__init__
Bio.Data.CodonTable.AmbiguousForwardTable.__init__
Bio.Data.CodonTable.CodonTable.__init__
Bio.Data.CodonTable.NCBICodonTable.__init__
Bio.DocSQL.Create.__init__
Bio.DocSQL.Insert.__init__
Bio.DocSQL.IterationCursor.__init__
Bio.DocSQL.Query.__init__
Bio.DocSQL.QueryAll.__init__
Bio.DocSQL.QueryGeneric.__init__
Bio.DocSQL.QueryRow.__init__
Bio.DocSQL.QuerySingle.__init__
Bio.Emboss.Applications.DiffseqCommandline.__init__
Bio.Emboss.Applications.EInvertedCommandline.__init__
Bio.Emboss.Applications.ETandemCommandline.__init__
Bio.Emboss.Applications.Est2GenomeCommandline.__init__
Bio.Emboss.Applications.FConsenseCommandline.__init__
Bio.Emboss.Applications.FDNADistCommandline.__init__
Bio.Emboss.Applications.FDNAParsCommandline.__init__
Bio.Emboss.Applications.FNeighborCommandline.__init__
Bio.Emboss.Applications.FProtDistCommandline.__init__
Bio.Emboss.Applications.FProtParsCommandline.__init__
Bio.Emboss.Applications.FSeqBootCommandline.__init__
Bio.Emboss.Applications.FTreeDistCommandline.__init__
Bio.Emboss.Applications.FuzznucCommandline.__init__
Bio.Emboss.Applications.IepCommandline.__init__
Bio.Emboss.Applications.NeedleCommandline.__init__
Bio.Emboss.Applications.NeedleallCommandline.__init__
Bio.Emboss.Applications.PalindromeCommandline.__init__
Bio.Emboss.Applications.Primer3Commandline.__init__
Bio.Emboss.Applications.PrimerSearchCommandline.__init__
Bio.Emboss.Applications.SeqmatchallCommandline.__init__
Bio.Emboss.Applications.SeqretCommandline.__init__
Bio.Emboss.Applications.StretcherCommandline.__init__
Bio.Emboss.Applications.TranalignCommandline.__init__
Bio.Emboss.Applications.WaterCommandline.__init__
Bio.Emboss.Applications._EmbossCommandLine.__init__
Bio.Emboss.Applications._EmbossMinimalCommandLine.__init__
Bio.Emboss.Primer3.Primers.__init__
Bio.Emboss.Primer3.Record.__init__
Bio.Emboss.PrimerSearch.Amplifier.__init__
Bio.Emboss.PrimerSearch.InputRecord.__init__
Bio.Emboss.PrimerSearch.OutputRecord.__init__
Bio.Entrez.Parser.CorruptedXMLError.__init__
Bio.Entrez.Parser.DataHandler.__init__
Bio.Entrez.Parser.NotXMLError.__init__
Bio.Entrez.Parser.StructureElement.__init__
Bio.Entrez.Parser.ValidationError.__init__
Bio.ExPASy.Enzyme.Record.__init__
Bio.ExPASy.Prodoc.Record.__init__
Bio.ExPASy.Prodoc.Reference.__init__
Bio.ExPASy.Prosite.Record.__init__
Bio.ExPASy.ScanProsite.ContentHandler.__init__
Bio.ExPASy.ScanProsite.Parser.__init__
Bio.ExPASy.ScanProsite.Record.__init__
Bio.FSSP.FSSPAlignDict.__init__
Bio.FSSP.FSSPAlignRec.__init__
Bio.FSSP.FSSPHeader.__init__
Bio.FSSP.FSSPSumRec.__init__
Bio.FSSP.FSSPTools.FSSPMultAlign.__init__
Bio.FSSP.PosAlign.__init__
Bio.FSSP.fssp_rec.fff_rec.__init__
Bio.File.UndoHandle.__init__
Bio.File._IndexedSeqFileDict.__init__
Bio.File._SQLiteManySeqFilesDict.__init__
Bio.GA.Crossover.General.SafeFitnessCrossover.__init__
Bio.GA.Crossover.GeneralPoint.GeneralPointCrossover.__init__
Bio.GA.Crossover.GeneralPoint.InterleaveCrossover.__init__
Bio.GA.Crossover.Point.SinglePointCrossover.__init__
Bio.GA.Crossover.TwoPoint.TwoPointCrossover.__init__
Bio.GA.Crossover.Uniform.UniformCrossover.__init__
Bio.GA.Evolver.GenerationEvolver.__init__
Bio.GA.Evolver.SteadyStateEvolver.__init__
Bio.GA.Mutation.General.SafeFitnessMutation.__init__
Bio.GA.Mutation.Simple.ConversionMutation.__init__
Bio.GA.Mutation.Simple.SinglePositionMutation.__init__
Bio.GA.Organism.Organism.__init__
Bio.GA.Repair.Stabilizing.AmbiguousRepair.__init__
Bio.GA.Selection.Abstract.AbstractSelection.__init__
Bio.GA.Selection.Diversity.DiversitySelection.__init__
Bio.GA.Selection.RouletteWheel.RouletteWheelSelection.__init__
Bio.GA.Selection.Tournament.TournamentSelection.__init__
Bio.GenBank.FeatureParser.__init__
Bio.GenBank.Iterator.__init__
Bio.GenBank.Record.Feature.__init__
Bio.GenBank.Record.Qualifier.__init__
Bio.GenBank.Record.Record.__init__
Bio.GenBank.Record.Reference.__init__
Bio.GenBank.RecordParser.__init__
Bio.GenBank.Scanner.InsdcScanner.__init__
Bio.GenBank._BaseGenBankConsumer.__init__
Bio.GenBank._FeatureConsumer.__init__
Bio.GenBank._RecordConsumer.__init__
Bio.GenBank.utils.FeatureValueCleaner.__init__
Bio.Geo.Record.Record.__init__
Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment.__init__
Bio.Graphics.BasicChromosome.Chromosome.__init__
Bio.Graphics.BasicChromosome.ChromosomeSegment.__init__
Bio.Graphics.BasicChromosome.Organism.__init__
Bio.Graphics.BasicChromosome.TelomereSegment.__init__
Bio.Graphics.BasicChromosome._ChromosomeComponent.__init__
Bio.Graphics.ColorSpiral.ColorSpiral.__init__
Bio.Graphics.Comparative.ComparativeScatterPlot.__init__
Bio.Graphics.DisplayRepresentation.ChromosomeCounts.__init__
Bio.Graphics.Distribution.BarChartDistribution.__init__
Bio.Graphics.Distribution.DistributionPage.__init__
Bio.Graphics.Distribution.LineDistribution.__init__
Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.__init__
Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.__init__
Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.__init__
Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.__init__
Bio.Graphics.GenomeDiagram._Diagram.Diagram.__init__
Bio.Graphics.GenomeDiagram._Feature.Feature.__init__
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.__init__
Bio.Graphics.GenomeDiagram._Graph.GraphData.__init__
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.__init__
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.__init__
Bio.Graphics.GenomeDiagram._Track.Track.__init__
Bio.Graphics.KGML_vis.KGMLCanvas.__init__
Bio.HMM.DynamicProgramming.AbstractDPAlgorithms.__init__
Bio.HMM.DynamicProgramming.LogDPAlgorithms.__init__
Bio.HMM.DynamicProgramming.ScaledDPAlgorithms.__init__
Bio.HMM.MarkovModel.HiddenMarkovModel.__init__
Bio.HMM.MarkovModel.MarkovModelBuilder.__init__
Bio.HMM.Trainer.AbstractTrainer.__init__
Bio.HMM.Trainer.BaumWelchTrainer.__init__
Bio.HMM.Trainer.KnownStateTrainer.__init__
Bio.HMM.Trainer.TrainingSequence.__init__
Bio.Index._InMemoryIndex.__init__
Bio.Index._ShelveIndex.__init__
Bio.KDTree.KDTree'.KDTree.__init__
Bio.KEGG.Compound.Record.__init__
Bio.KEGG.Enzyme.Record.__init__
Bio.KEGG.KGML.KGML_parser.KGMLParser.__init__
Bio.KEGG.KGML.KGML_pathway.Component.__init__
Bio.KEGG.KGML.KGML_pathway.Entry.__init__
Bio.KEGG.KGML.KGML_pathway.Graphics.__init__
Bio.KEGG.KGML.KGML_pathway.Pathway.__init__
Bio.KEGG.KGML.KGML_pathway.Reaction.__init__
Bio.KEGG.KGML.KGML_pathway.Relation.__init__
Bio.LogisticRegression.LogisticRegression.__init__
Bio.MarkovModel.MarkovModel.__init__
Bio.MaxEntropy.MaxEntropy.__init__
Bio.Motif.Parsers.AlignAce.Record.__init__
Bio.Motif.Parsers.MAST.Record.__init__
Bio.Motif.Parsers.MEME.MEMEInstance.__init__
Bio.Motif.Parsers.MEME.MEMEMotif.__init__
Bio.Motif.Parsers.MEME.MEMERecord.__init__
Bio.Motif.Thresholds.ScoreDistribution.__init__
Bio.Motif._Motif.Motif.__init__
Bio.NMR.xpktools.Peaklist.__init__
Bio.NMR.xpktools.XpkEntry.__init__
Bio.NaiveBayes.NaiveBayes.__init__
Bio.NeuralNetwork.BackPropagation.Layer.AbstractLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.__init__
Bio.NeuralNetwork.BackPropagation.Network.BasicNetwork.__init__
Bio.NeuralNetwork.Gene.Motif.MotifCoder.__init__
Bio.NeuralNetwork.Gene.Motif.MotifFinder.__init__
Bio.NeuralNetwork.Gene.Pattern.PatternIO.__init__
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.__init__
Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness.__init__
Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder.__init__
Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness.__init__
Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator.__init__
Bio.NeuralNetwork.Gene.Schema.Schema.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaCoder.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaFactory.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaFinder.__init__
Bio.NeuralNetwork.Gene.Schema.SimpleFinisher.__init__
Bio.NeuralNetwork.Gene.Signature.SignatureCoder.__init__
Bio.NeuralNetwork.Gene.Signature.SignatureFinder.__init__
Bio.NeuralNetwork.StopTraining.ValidationIncreaseStop.__init__
Bio.NeuralNetwork.Training.ExampleManager.__init__
Bio.NeuralNetwork.Training.TrainingExample.__init__
Bio.Nexus.Nexus.Block.__init__
Bio.Nexus.Nexus.CharBuffer.__init__
Bio.Nexus.Nexus.Commandline.__init__
Bio.Nexus.Nexus.Nexus.__init__
Bio.Nexus.Nexus.StepMatrix.__init__
Bio.Nexus.Nodes.Chain.__init__
Bio.Nexus.Nodes.Node.__init__
Bio.Nexus.Trees.NodeData.__init__
Bio.Nexus.Trees.Tree.__init__
Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap.__init__
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__init__
Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap.__init__
Bio.PDB.Atom.Atom.__init__
Bio.PDB.Atom.DisorderedAtom.__init__
Bio.PDB.Chain.Chain.__init__
Bio.PDB.DSSP'.DSSP.__init__
Bio.PDB.Dice.ChainSelector.__init__
Bio.PDB.Entity.DisorderedEntityWrapper.__init__
Bio.PDB.Entity.Entity.__init__
Bio.PDB.FragmentMapper'.Fragment.__init__
Bio.PDB.FragmentMapper'.FragmentMapper.__init__
Bio.PDB.HSExposure.ExposureCN.__init__
Bio.PDB.HSExposure.HSExposureCA.__init__
Bio.PDB.HSExposure.HSExposureCB.__init__
Bio.PDB.HSExposure._AbstractHSExposure.__init__
Bio.PDB.MMCIF2Dict.MMCIF2Dict.__init__
Bio.PDB.MMCIFParser'.MMCIFParser.__init__
Bio.PDB.Model.Model.__init__
Bio.PDB.NACCESS.NACCESS.__init__
Bio.PDB.NACCESS.NACCESS_atomic.__init__
Bio.PDB.NeighborSearch'.NeighborSearch.__init__
Bio.PDB.PDBIO'.PDBIO.__init__
Bio.PDB.PDBList'.PDBList.__init__
Bio.PDB.PDBParser'.PDBParser.__init__
Bio.PDB.PSEA.PSEA.__init__
Bio.PDB.Polypeptide.CaPPBuilder.__init__
Bio.PDB.Polypeptide.PPBuilder.__init__
Bio.PDB.Polypeptide._PPBuilder.__init__
Bio.PDB.QCPSuperimposer.QCPSuperimposer.__init__
Bio.PDB.Residue.DisorderedResidue.__init__
Bio.PDB.Residue.Residue.__init__
Bio.PDB.ResidueDepth'.ResidueDepth.__init__
Bio.PDB.Structure.Structure.__init__
Bio.PDB.StructureAlignment'.StructureAlignment.__init__
Bio.PDB.StructureBuilder.StructureBuilder.__init__
Bio.PDB.Superimposer'.Superimposer.__init__
Bio.PDB.Vector'.Vector.__init__
Bio.ParserSupport.EventGenerator.__init__
Bio.ParserSupport.TaggingConsumer.__init__
Bio.Pathway.Network.__init__
Bio.Pathway.Reaction.__init__
Bio.Pathway.Rep.Graph.Graph.__init__
Bio.Pathway.Rep.MultiGraph.MultiGraph.__init__
Bio.Pathway.System.__init__
Bio.Phylo.Applications._Fasttree.FastTreeCommandline.__init__
Bio.Phylo.Applications._Phyml.PhymlCommandline.__init__
Bio.Phylo.Applications._Raxml.RaxmlCommandline.__init__
Bio.Phylo.BaseTree.BranchColor.__init__
Bio.Phylo.BaseTree.Clade.__init__
Bio.Phylo.BaseTree.Tree.__init__
Bio.Phylo.CDAO.Clade.__init__
Bio.Phylo.CDAO.Tree.__init__
Bio.Phylo.CDAOIO.Parser.__init__
Bio.Phylo.CDAOIO.Writer.__init__
Bio.Phylo.NeXML.Clade.__init__
Bio.Phylo.NeXML.Tree.__init__
Bio.Phylo.NeXMLIO.Parser.__init__
Bio.Phylo.NeXMLIO.Writer.__init__
Bio.Phylo.Newick.Clade.__init__
Bio.Phylo.Newick.Tree.__init__
Bio.Phylo.NewickIO.Parser.__init__
Bio.Phylo.NewickIO.Writer.__init__
Bio.Phylo.PAML._paml.Paml.__init__
Bio.Phylo.PAML.baseml.Baseml.__init__
Bio.Phylo.PAML.codeml.Codeml.__init__
Bio.Phylo.PAML.yn00.Yn00.__init__
Bio.Phylo.PhyloXML.Accession.__init__
Bio.Phylo.PhyloXML.Annotation.__init__
Bio.Phylo.PhyloXML.BinaryCharacters.__init__
Bio.Phylo.PhyloXML.BranchColor.__init__
Bio.Phylo.PhyloXML.Clade.__init__
Bio.Phylo.PhyloXML.CladeRelation.__init__
Bio.Phylo.PhyloXML.Confidence.__init__
Bio.Phylo.PhyloXML.Date.__init__
Bio.Phylo.PhyloXML.Distribution.__init__
Bio.Phylo.PhyloXML.DomainArchitecture.__init__
Bio.Phylo.PhyloXML.Events.__init__
Bio.Phylo.PhyloXML.Id.__init__
Bio.Phylo.PhyloXML.MolSeq.__init__
Bio.Phylo.PhyloXML.Other.__init__
Bio.Phylo.PhyloXML.Phylogeny.__init__
Bio.Phylo.PhyloXML.Phyloxml.__init__
Bio.Phylo.PhyloXML.Point.__init__
Bio.Phylo.PhyloXML.Polygon.__init__
Bio.Phylo.PhyloXML.Property.__init__
Bio.Phylo.PhyloXML.ProteinDomain.__init__
Bio.Phylo.PhyloXML.Reference.__init__
Bio.Phylo.PhyloXML.Sequence.__init__
Bio.Phylo.PhyloXML.SequenceRelation.__init__
Bio.Phylo.PhyloXML.Taxonomy.__init__
Bio.Phylo.PhyloXML.Uri.__init__
Bio.Phylo.PhyloXMLIO.Parser.__init__
Bio.Phylo.PhyloXMLIO.Writer.__init__
Bio.Phylo.TreeConstruction.DistanceCalculator.__init__
Bio.Phylo.TreeConstruction.DistanceTreeConstructor.__init__
Bio.Phylo.TreeConstruction.NNITreeSearcher.__init__
Bio.Phylo.TreeConstruction.ParsimonyScorer.__init__
Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor.__init__
Bio.Phylo.TreeConstruction._DistanceMatrix.__init__
Bio.Phylo.TreeConstruction._Matrix.__init__
Bio.PopGen.Async.Async.__init__
Bio.PopGen.Async.DirectoryRetriever.__init__
Bio.PopGen.Async.FileRetriever.__init__
Bio.PopGen.Async.Local.Local.__init__
Bio.PopGen.FDist.Async.FDistAsync.__init__
Bio.PopGen.FDist.Async.SplitFDist.__init__
Bio.PopGen.FDist.Controller.FDistController.__init__
Bio.PopGen.FDist.Record.__init__
Bio.PopGen.GenePop.Controller.GenePopController.__init__
Bio.PopGen.GenePop.Controller._FileIterator.__init__
Bio.PopGen.GenePop.Controller._GenePopCommandline.__init__
Bio.PopGen.GenePop.EasyController.EasyController.__init__
Bio.PopGen.GenePop.FileParser.FileRecord.__init__
Bio.PopGen.GenePop.LargeFileParser.Record.__init__
Bio.PopGen.GenePop.Record.__init__
Bio.PopGen.SimCoal.Async.SimCoalCache.__init__
Bio.PopGen.SimCoal.Cache.SimCoalCache.__init__
Bio.PopGen.SimCoal.Controller.FastSimCoalController.__init__
Bio.PopGen.SimCoal.Controller.SimCoalController.__init__
Bio.PopGen.SimCoal.Controller._FastSimCoalCommandLine.__init__
Bio.Restriction.PrintFormat.PrintFormat.__init__
Bio.Restriction.Restriction.Analysis.__init__
Bio.Restriction.Restriction.FormattedSeq.__init__
Bio.Restriction.Restriction.RestrictionBatch.__init__
Bio.Restriction.Restriction.RestrictionType.__init__
Bio.SCOP.Astral.__init__
Bio.SCOP.Cla.Index.__init__
Bio.SCOP.Cla.Record.__init__
Bio.SCOP.Des.Record.__init__
Bio.SCOP.Dom.Record.__init__
Bio.SCOP.Domain.__init__
Bio.SCOP.Hie.Record.__init__
Bio.SCOP.Node.__init__
Bio.SCOP.Raf.Res.__init__
Bio.SCOP.Raf.SeqMap.__init__
Bio.SCOP.Raf.SeqMapIndex.__init__
Bio.SCOP.Residues.Residues.__init__
Bio.SCOP.Scop.__init__
Bio.SVDSuperimposer.SVDSuperimposer.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabParser.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.__init__
Bio.SearchIO.BlastIO.blast_text.BlastTextParser.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.__init__
Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator.__init__
Bio.SearchIO.BlatIO.BlatPslIndexer.__init__
Bio.SearchIO.BlatIO.BlatPslParser.__init__
Bio.SearchIO.BlatIO.BlatPslWriter.__init__
Bio.SearchIO.ExonerateIO._base._BaseExonerateParser.__init__
Bio.SearchIO.FastaIO.FastaM10Indexer.__init__
Bio.SearchIO.FastaIO.FastaM10Parser.__init__
Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer.__init__
Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser.__init__
Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.__init__
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser.__init__
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.__init__
Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser.__init__
Bio.SearchIO._index.SearchIndexer.__init__
Bio.SearchIO._model.hit.Hit.__init__
Bio.SearchIO._model.hsp.HSP.__init__
Bio.SearchIO._model.hsp.HSPFragment.__init__
Bio.SearchIO._model.query.QueryResult.__init__
Bio.Seq.MutableSeq.__init__
Bio.Seq.Seq.__init__
Bio.Seq.UnknownSeq.__init__
Bio.SeqFeature.CompoundLocation.__init__
Bio.SeqFeature.FeatureLocation.__init__
Bio.SeqFeature.PositionGap.__init__
Bio.SeqFeature.Reference.__init__
Bio.SeqFeature.SeqFeature.__init__
Bio.SeqIO.FastaIO.FastaWriter.__init__
Bio.SeqIO.Interfaces.SequenceIterator.__init__
Bio.SeqIO.Interfaces.SequenceWriter.__init__
Bio.SeqIO.Interfaces.SequentialSequenceWriter.__init__
Bio.SeqIO.PhdIO.PhdWriter.__init__
Bio.SeqIO.QualityIO.QualPhredWriter.__init__
Bio.SeqIO.SeqXmlIO.SeqXmlIterator.__init__
Bio.SeqIO.SeqXmlIO.SeqXmlWriter.__init__
Bio.SeqIO.SeqXmlIO.XMLRecordIterator.__init__
Bio.SeqIO.SffIO.SffWriter.__init__
Bio.SeqIO.SffIO._AddTellHandle.__init__
Bio.SeqIO.UniprotIO.Parser.__init__
Bio.SeqIO._index.IntelliGeneticsRandomAccess.__init__
Bio.SeqIO._index.SeqFileRandomAccess.__init__
Bio.SeqIO._index.SequentialSeqFileRandomAccess.__init__
Bio.SeqIO._index.SffRandomAccess.__init__
Bio.SeqRecord.SeqRecord.__init__
Bio.SeqRecord._RestrictedDict.__init__
Bio.SeqUtils.CodonUsage.CodonAdaptationIndex.__init__
Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.__init__
Bio.SeqUtils.ProtParam.ProteinAnalysis.__init__
Bio.Sequencing.Ace.ACEFileRecord.__init__
Bio.Sequencing.Ace.Contig.__init__
Bio.Sequencing.Ace.Reads.__init__
Bio.Sequencing.Ace.af.__init__
Bio.Sequencing.Ace.bs.__init__
Bio.Sequencing.Ace.ct.__init__
Bio.Sequencing.Ace.ds.__init__
Bio.Sequencing.Ace.qa.__init__
Bio.Sequencing.Ace.rd.__init__
Bio.Sequencing.Ace.rt.__init__
Bio.Sequencing.Ace.wa.__init__
Bio.Sequencing.Ace.wr.__init__
Bio.Sequencing.Applications._Novoalign.NovoalignCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaAlignCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaBwaswCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaIndexCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaSampeCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaSamseCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsSortCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline.__init__
Bio.Sequencing.Phd.Record.__init__
Bio.SubsMat.FreqTable.FreqTable.__init__
Bio.SubsMat.SeqMat.__init__
Bio.SwissProt.KeyWList.Record.__init__
Bio.SwissProt.Record.__init__
Bio.SwissProt.Reference.__init__
Bio.UniGene.ProtsimLine.__init__
Bio.UniGene.Record.__init__
Bio.UniGene.STSLine.__init__
Bio.UniGene.SequenceLine.__init__
Bio.Wise.dnal.Statistics.__init__
Bio.Wise.psw.AlignmentColumn.__init__
Bio.Wise.psw.ColumnUnit.__init__
Bio._py3k._ordereddict.OrderedDict.__init__
Bio.bgzf.BgzfReader.__init__
Bio.bgzf.BgzfWriter.__init__
Bio.codonalign.codonalignment.CodonAlignment.__init__
Bio.codonalign.codonseq.CodonSeq.__init__
Bio.kNN.kNN.__init__
Bio.motifs.Instances.__init__
Bio.motifs.Motif.__init__
Bio.motifs.alignace.Record.__init__
Bio.motifs.applications._alignace.AlignAceCommandline.__init__
Bio.motifs.applications._alignace.CompareAceCommandline.__init__
Bio.motifs.applications._xxmotif.XXmotifCommandline.__init__
Bio.motifs.jaspar.Motif.__init__
Bio.motifs.jaspar.Record.__init__
Bio.motifs.jaspar.db.JASPAR5.__init__
Bio.motifs.mast.Record.__init__
Bio.motifs.matrix.GenericPositionMatrix.__init__
Bio.motifs.matrix.PositionWeightMatrix.__init__
Bio.motifs.meme.Instance.__init__
Bio.motifs.meme.Motif.__init__
Bio.motifs.meme.Record.__init__
Bio.motifs.thresholds.ScoreDistribution.__init__
Bio.motifs.transfac.Record.__init__
Bio.pairwise2.affine_penalty.__init__
Bio.pairwise2.dictionary_match.__init__
Bio.pairwise2.identity_match.__init__
BioSQL.BioSeq.DBSeq.__init__
BioSQL.BioSeq.DBSeqRecord.__init__
BioSQL.BioSeqDatabase.Adaptor.__init__
BioSQL.BioSeqDatabase.BioSeqDatabase.__init__
BioSQL.BioSeqDatabase.DBServer.__init__
BioSQL.BioSeqDatabase._CursorWrapper.__init__
BioSQL.DBUtils.Generic_dbutils.__init__
BioSQL.Loader.DatabaseLoader.__init__
BioSQL.Loader.DatabaseRemover.__init__" class="py-name" href="#" onclick="return doclink('link-1001', '__init__', 'link-1001');">__init__</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1753"></a><tt class="py-lineno">1753</tt>  <tt class="py-line"> </tt>
<a name="_BlastErrorConsumer.noevent"></a><div id="_BlastErrorConsumer.noevent-def"><a name="L1754"></a><tt class="py-lineno">1754</tt> <a class="py-toggle" href="#" id="_BlastErrorConsumer.noevent-toggle" onclick="return toggle('_BlastErrorConsumer.noevent');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone._BlastErrorConsumer-class.html#noevent">noevent</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BlastErrorConsumer.noevent-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="_BlastErrorConsumer.noevent-expanded"><a name="L1755"></a><tt class="py-lineno">1755</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">'Query must be at least wordsize'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L1756"></a><tt class="py-lineno">1756</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt id="link-1002" class="py-name" targets="Class Bio.Blast.NCBIStandalone.ShortQueryBlastError=Bio.Blast.NCBIStandalone.ShortQueryBlastError-class.html"><a title="Bio.Blast.NCBIStandalone.ShortQueryBlastError" class="py-name" href="#" onclick="return doclink('link-1002', 'ShortQueryBlastError', 'link-1002');">ShortQueryBlastError</a></tt><tt class="py-op">(</tt><tt class="py-string">"Query must be at least wordsize"</tt><tt class="py-op">)</tt> </tt>
<a name="L1757"></a><tt class="py-lineno">1757</tt>  <tt class="py-line">        <tt class="py-comment"># Now pass the line back up to the superclass.</tt> </tt>
<a name="L1758"></a><tt class="py-lineno">1758</tt>  <tt class="py-line">        <tt class="py-name">method</tt> <tt class="py-op">=</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt id="link-1003" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer" class="py-name" href="#" onclick="return doclink('link-1003', '_BlastConsumer', 'link-430');">_BlastConsumer</a></tt><tt class="py-op">,</tt> <tt class="py-string">'noevent'</tt><tt class="py-op">,</tt> </tt>
<a name="L1759"></a><tt class="py-lineno">1759</tt>  <tt class="py-line">                         <tt id="link-1004" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer" class="py-name" href="#" onclick="return doclink('link-1004', '_BlastConsumer', 'link-430');">_BlastConsumer</a></tt><tt class="py-op">.</tt><tt id="link-1005" class="py-name" targets="Method Bio.Alphabet.AlphabetEncoder.__getattr__()=Bio.Alphabet.AlphabetEncoder-class.html#__getattr__,Method Bio.Data.CodonTable.AmbiguousCodonTable.__getattr__()=Bio.Data.CodonTable.AmbiguousCodonTable-class.html#__getattr__,Method Bio.DocSQL.QueryRow.__getattr__()=Bio.DocSQL.QueryRow-class.html#__getattr__,Method Bio.File.UndoHandle.__getattr__()=Bio.File.UndoHandle-class.html#__getattr__,Method Bio.GenBank._BaseGenBankConsumer.__getattr__()=Bio.GenBank._BaseGenBankConsumer-class.html#__getattr__,Method Bio.Graphics.GenomeDiagram._Feature.Feature.__getattr__()=Bio.Graphics.GenomeDiagram._Feature.Feature-class.html#__getattr__,Method Bio.PDB.Entity.DisorderedEntityWrapper.__getattr__()=Bio.PDB.Entity.DisorderedEntityWrapper-class.html#__getattr__,Method Bio.ParserSupport.AbstractConsumer.__getattr__()=Bio.ParserSupport.AbstractConsumer-class.html#__getattr__,Method Bio.ParserSupport.TaggingConsumer.__getattr__()=Bio.ParserSupport.TaggingConsumer-class.html#__getattr__"><a title="Bio.Alphabet.AlphabetEncoder.__getattr__
Bio.Data.CodonTable.AmbiguousCodonTable.__getattr__
Bio.DocSQL.QueryRow.__getattr__
Bio.File.UndoHandle.__getattr__
Bio.GenBank._BaseGenBankConsumer.__getattr__
Bio.Graphics.GenomeDiagram._Feature.Feature.__getattr__
Bio.PDB.Entity.DisorderedEntityWrapper.__getattr__
Bio.ParserSupport.AbstractConsumer.__getattr__
Bio.ParserSupport.TaggingConsumer.__getattr__" class="py-name" href="#" onclick="return doclink('link-1005', '__getattr__', 'link-1005');">__getattr__</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-string">'noevent'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1760"></a><tt class="py-lineno">1760</tt>  <tt class="py-line">        <tt class="py-name">method</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1761"></a><tt class="py-lineno">1761</tt>  <tt class="py-line"> </tt>
<a name="L1762"></a><tt class="py-lineno">1762</tt>  <tt class="py-line"> </tt>
<a name="BlastErrorParser"></a><div id="BlastErrorParser-def"><a name="L1763"></a><tt class="py-lineno">1763</tt> <a class="py-toggle" href="#" id="BlastErrorParser-toggle" onclick="return toggle('BlastErrorParser');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone.BlastErrorParser-class.html">BlastErrorParser</a><tt class="py-op">(</tt><tt class="py-base-class">AbstractParser</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastErrorParser-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="BlastErrorParser-expanded"><a name="L1764"></a><tt class="py-lineno">1764</tt>  <tt class="py-line">    <tt class="py-docstring">"""Attempt to catch and diagnose BLAST errors while parsing.</tt> </tt>
<a name="L1765"></a><tt class="py-lineno">1765</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1766"></a><tt class="py-lineno">1766</tt>  <tt class="py-line"><tt class="py-docstring">    This utilizes the BlastParser module but adds an additional layer</tt> </tt>
<a name="L1767"></a><tt class="py-lineno">1767</tt>  <tt class="py-line"><tt class="py-docstring">    of complexity on top of it by attempting to diagnose ValueErrors</tt> </tt>
<a name="L1768"></a><tt class="py-lineno">1768</tt>  <tt class="py-line"><tt class="py-docstring">    that may actually indicate problems during BLAST parsing.</tt> </tt>
<a name="L1769"></a><tt class="py-lineno">1769</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1770"></a><tt class="py-lineno">1770</tt>  <tt class="py-line"><tt class="py-docstring">    Current BLAST problems this detects are:</tt> </tt>
<a name="L1771"></a><tt class="py-lineno">1771</tt>  <tt class="py-line"><tt class="py-docstring">    o LowQualityBlastError - When BLASTing really low quality sequences</tt> </tt>
<a name="L1772"></a><tt class="py-lineno">1772</tt>  <tt class="py-line"><tt class="py-docstring">    (ie. some GenBank entries which are just short streches of a single</tt> </tt>
<a name="L1773"></a><tt class="py-lineno">1773</tt>  <tt class="py-line"><tt class="py-docstring">    nucleotide), BLAST will report an error with the sequence and be</tt> </tt>
<a name="L1774"></a><tt class="py-lineno">1774</tt>  <tt class="py-line"><tt class="py-docstring">    unable to search with this. This will lead to a badly formatted</tt> </tt>
<a name="L1775"></a><tt class="py-lineno">1775</tt>  <tt class="py-line"><tt class="py-docstring">    BLAST report that the parsers choke on. The parser will convert the</tt> </tt>
<a name="L1776"></a><tt class="py-lineno">1776</tt>  <tt class="py-line"><tt class="py-docstring">    ValueError to a LowQualityBlastError and attempt to provide useful</tt> </tt>
<a name="L1777"></a><tt class="py-lineno">1777</tt>  <tt class="py-line"><tt class="py-docstring">    information.</tt> </tt>
<a name="L1778"></a><tt class="py-lineno">1778</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="BlastErrorParser.__init__"></a><div id="BlastErrorParser.__init__-def"><a name="L1779"></a><tt class="py-lineno">1779</tt> <a class="py-toggle" href="#" id="BlastErrorParser.__init__-toggle" onclick="return toggle('BlastErrorParser.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone.BlastErrorParser-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">bad_report_handle</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastErrorParser.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="BlastErrorParser.__init__-expanded"><a name="L1780"></a><tt class="py-lineno">1780</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize a parser that tries to catch BlastErrors.</tt> </tt>
<a name="L1781"></a><tt class="py-lineno">1781</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1782"></a><tt class="py-lineno">1782</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L1783"></a><tt class="py-lineno">1783</tt>  <tt class="py-line"><tt class="py-docstring">        o bad_report_handle - An optional argument specifying a handle</tt> </tt>
<a name="L1784"></a><tt class="py-lineno">1784</tt>  <tt class="py-line"><tt class="py-docstring">        where bad reports should be sent. This would allow you to save</tt> </tt>
<a name="L1785"></a><tt class="py-lineno">1785</tt>  <tt class="py-line"><tt class="py-docstring">        all of the bad reports to a file, for instance. If no handle</tt> </tt>
<a name="L1786"></a><tt class="py-lineno">1786</tt>  <tt class="py-line"><tt class="py-docstring">        is specified, the bad reports will not be saved.</tt> </tt>
<a name="L1787"></a><tt class="py-lineno">1787</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1788"></a><tt class="py-lineno">1788</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_bad_report_handle</tt> <tt class="py-op">=</tt> <tt class="py-name">bad_report_handle</tt> </tt>
<a name="L1789"></a><tt class="py-lineno">1789</tt>  <tt class="py-line"> </tt>
<a name="L1790"></a><tt class="py-lineno">1790</tt>  <tt class="py-line">        <tt class="py-comment"># self._b_parser = BlastParser()</tt> </tt>
<a name="L1791"></a><tt class="py-lineno">1791</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_scanner</tt> <tt class="py-op">=</tt> <tt id="link-1006" class="py-name"><a title="Bio.Blast.NCBIStandalone._Scanner" class="py-name" href="#" onclick="return doclink('link-1006', '_Scanner', 'link-429');">_Scanner</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1792"></a><tt class="py-lineno">1792</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_consumer</tt> <tt class="py-op">=</tt> <tt id="link-1007" class="py-name" targets="Class Bio.Blast.NCBIStandalone._BlastErrorConsumer=Bio.Blast.NCBIStandalone._BlastErrorConsumer-class.html"><a title="Bio.Blast.NCBIStandalone._BlastErrorConsumer" class="py-name" href="#" onclick="return doclink('link-1007', '_BlastErrorConsumer', 'link-1007');">_BlastErrorConsumer</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1793"></a><tt class="py-lineno">1793</tt>  <tt class="py-line"> </tt>
<a name="BlastErrorParser.parse"></a><div id="BlastErrorParser.parse-def"><a name="L1794"></a><tt class="py-lineno">1794</tt> <a class="py-toggle" href="#" id="BlastErrorParser.parse-toggle" onclick="return toggle('BlastErrorParser.parse');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone.BlastErrorParser-class.html#parse">parse</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastErrorParser.parse-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="BlastErrorParser.parse-expanded"><a name="L1795"></a><tt class="py-lineno">1795</tt>  <tt class="py-line">        <tt class="py-docstring">"""Parse a handle, attempting to diagnose errors.</tt> </tt>
<a name="L1796"></a><tt class="py-lineno">1796</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1797"></a><tt class="py-lineno">1797</tt>  <tt class="py-line">        <tt id="link-1008" class="py-name"><a title="Bio.Restriction.Restriction.AanI.results
Bio.Restriction.Restriction.AarI.results
Bio.Restriction.Restriction.AasI.results
Bio.Restriction.Restriction.AatII.results
Bio.Restriction.Restriction.AbaSI.results
Bio.Restriction.Restriction.AbsI.results
Bio.Restriction.Restriction.Acc16I.results
Bio.Restriction.Restriction.Acc36I.results
Bio.Restriction.Restriction.Acc65I.results
Bio.Restriction.Restriction.AccB1I.results
Bio.Restriction.Restriction.AccB7I.results
Bio.Restriction.Restriction.AccBSI.results
Bio.Restriction.Restriction.AccI.results
Bio.Restriction.Restriction.AccII.results
Bio.Restriction.Restriction.AccIII.results
Bio.Restriction.Restriction.AceIII.results
Bio.Restriction.Restriction.AciI.results
Bio.Restriction.Restriction.AclI.results
Bio.Restriction.Restriction.AclWI.results
Bio.Restriction.Restriction.AcoI.results
Bio.Restriction.Restriction.AcsI.results
Bio.Restriction.Restriction.AcuI.results
Bio.Restriction.Restriction.AcvI.results
Bio.Restriction.Restriction.AcyI.results
Bio.Restriction.Restriction.AdeI.results
Bio.Restriction.Restriction.AfaI.results
Bio.Restriction.Restriction.AfeI.results
Bio.Restriction.Restriction.AfiI.results
Bio.Restriction.Restriction.AflII.results
Bio.Restriction.Restriction.AflIII.results
Bio.Restriction.Restriction.AgeI.results
Bio.Restriction.Restriction.AgsI.results
Bio.Restriction.Restriction.AhaIII.results
Bio.Restriction.Restriction.AhdI.results
Bio.Restriction.Restriction.AhlI.results
Bio.Restriction.Restriction.AjiI.results
Bio.Restriction.Restriction.AjnI.results
Bio.Restriction.Restriction.AjuI.results
Bio.Restriction.Restriction.AleI.results
Bio.Restriction.Restriction.AlfI.results
Bio.Restriction.Restriction.AloI.results
Bio.Restriction.Restriction.AluBI.results
Bio.Restriction.Restriction.AluI.results
Bio.Restriction.Restriction.Alw21I.results
Bio.Restriction.Restriction.Alw26I.results
Bio.Restriction.Restriction.Alw44I.results
Bio.Restriction.Restriction.AlwFI.results
Bio.Restriction.Restriction.AlwI.results
Bio.Restriction.Restriction.AlwNI.results
Bio.Restriction.Restriction.Ama87I.results
Bio.Restriction.Restriction.Aor13HI.results
Bio.Restriction.Restriction.Aor51HI.results
Bio.Restriction.Restriction.AoxI.results
Bio.Restriction.Restriction.ApaBI.results
Bio.Restriction.Restriction.ApaI.results
Bio.Restriction.Restriction.ApaLI.results
Bio.Restriction.Restriction.ApeKI.results
Bio.Restriction.Restriction.ApoI.results
Bio.Restriction.Restriction.ApyPI.results
Bio.Restriction.Restriction.AquII.results
Bio.Restriction.Restriction.AquIII.results
Bio.Restriction.Restriction.AquIV.results
Bio.Restriction.Restriction.ArsI.results
Bio.Restriction.Restriction.AscI.results
Bio.Restriction.Restriction.AseI.results
Bio.Restriction.Restriction.Asi256I.results
Bio.Restriction.Restriction.AsiGI.results
Bio.Restriction.Restriction.AsiSI.results
Bio.Restriction.Restriction.Asp700I.results
Bio.Restriction.Restriction.Asp718I.results
Bio.Restriction.Restriction.AspA2I.results
Bio.Restriction.Restriction.AspBHI.results
Bio.Restriction.Restriction.AspLEI.results
Bio.Restriction.Restriction.AspS9I.results
Bio.Restriction.Restriction.AssI.results
Bio.Restriction.Restriction.AsuC2I.results
Bio.Restriction.Restriction.AsuHPI.results
Bio.Restriction.Restriction.AsuI.results
Bio.Restriction.Restriction.AsuII.results
Bio.Restriction.Restriction.AsuNHI.results
Bio.Restriction.Restriction.AvaI.results
Bio.Restriction.Restriction.AvaII.results
Bio.Restriction.Restriction.AvaIII.results
Bio.Restriction.Restriction.AvrII.results
Bio.Restriction.Restriction.AxyI.results
Bio.Restriction.Restriction.BaeGI.results
Bio.Restriction.Restriction.BaeI.results
Bio.Restriction.Restriction.BalI.results
Bio.Restriction.Restriction.BamHI.results
Bio.Restriction.Restriction.BanI.results
Bio.Restriction.Restriction.BanII.results
Bio.Restriction.Restriction.BarI.results
Bio.Restriction.Restriction.BasI.results
Bio.Restriction.Restriction.BauI.results
Bio.Restriction.Restriction.Bbr7I.results
Bio.Restriction.Restriction.BbrPI.results
Bio.Restriction.Restriction.BbsI.results
Bio.Restriction.Restriction.Bbv12I.results
Bio.Restriction.Restriction.BbvCI.results
Bio.Restriction.Restriction.BbvI.results
Bio.Restriction.Restriction.BbvII.results
Bio.Restriction.Restriction.BccI.results
Bio.Restriction.Restriction.Bce83I.results
Bio.Restriction.Restriction.BceAI.results
Bio.Restriction.Restriction.BcefI.results
Bio.Restriction.Restriction.BcgI.results
Bio.Restriction.Restriction.BciT130I.results
Bio.Restriction.Restriction.BciVI.results
Bio.Restriction.Restriction.BclI.results
Bio.Restriction.Restriction.BcnI.results
Bio.Restriction.Restriction.BcoDI.results
Bio.Restriction.Restriction.BcuI.results
Bio.Restriction.Restriction.BdaI.results
Bio.Restriction.Restriction.BetI.results
Bio.Restriction.Restriction.BfaI.results
Bio.Restriction.Restriction.BfiI.results
Bio.Restriction.Restriction.BfmI.results
Bio.Restriction.Restriction.BfoI.results
Bio.Restriction.Restriction.BfrI.results
Bio.Restriction.Restriction.BfuAI.results
Bio.Restriction.Restriction.BfuCI.results
Bio.Restriction.Restriction.BfuI.results
Bio.Restriction.Restriction.BglI.results
Bio.Restriction.Restriction.BglII.results
Bio.Restriction.Restriction.BinI.results
Bio.Restriction.Restriction.BisI.results
Bio.Restriction.Restriction.BlnI.results
Bio.Restriction.Restriction.BlpI.results
Bio.Restriction.Restriction.BlsI.results
Bio.Restriction.Restriction.BmcAI.results
Bio.Restriction.Restriction.Bme1390I.results
Bio.Restriction.Restriction.Bme18I.results
Bio.Restriction.Restriction.BmeDI.results
Bio.Restriction.Restriction.BmeRI.results
Bio.Restriction.Restriction.BmeT110I.results
Bio.Restriction.Restriction.BmgBI.results
Bio.Restriction.Restriction.BmgI.results
Bio.Restriction.Restriction.BmgT120I.results
Bio.Restriction.Restriction.BmiI.results
Bio.Restriction.Restriction.BmrFI.results
Bio.Restriction.Restriction.BmrI.results
Bio.Restriction.Restriction.BmsI.results
Bio.Restriction.Restriction.BmtI.results
Bio.Restriction.Restriction.BmuI.results
Bio.Restriction.Restriction.BoxI.results
Bio.Restriction.Restriction.BpiI.results
Bio.Restriction.Restriction.BplI.results
Bio.Restriction.Restriction.BpmI.results
Bio.Restriction.Restriction.Bpu10I.results
Bio.Restriction.Restriction.Bpu1102I.results
Bio.Restriction.Restriction.Bpu14I.results
Bio.Restriction.Restriction.BpuEI.results
Bio.Restriction.Restriction.BpuMI.results
Bio.Restriction.Restriction.BpvUI.results
Bio.Restriction.Restriction.Bsa29I.results
Bio.Restriction.Restriction.BsaAI.results
Bio.Restriction.Restriction.BsaBI.results
Bio.Restriction.Restriction.BsaHI.results
Bio.Restriction.Restriction.BsaI.results
Bio.Restriction.Restriction.BsaJI.results
Bio.Restriction.Restriction.BsaWI.results
Bio.Restriction.Restriction.BsaXI.results
Bio.Restriction.Restriction.BsbI.results
Bio.Restriction.Restriction.Bsc4I.results
Bio.Restriction.Restriction.BscAI.results
Bio.Restriction.Restriction.BscGI.results
Bio.Restriction.Restriction.Bse118I.results
Bio.Restriction.Restriction.Bse1I.results
Bio.Restriction.Restriction.Bse21I.results
Bio.Restriction.Restriction.Bse3DI.results
Bio.Restriction.Restriction.Bse8I.results
Bio.Restriction.Restriction.BseAI.results
Bio.Restriction.Restriction.BseBI.results
Bio.Restriction.Restriction.BseCI.results
Bio.Restriction.Restriction.BseDI.results
Bio.Restriction.Restriction.BseGI.results
Bio.Restriction.Restriction.BseJI.results
Bio.Restriction.Restriction.BseLI.results
Bio.Restriction.Restriction.BseMI.results
Bio.Restriction.Restriction.BseMII.results
Bio.Restriction.Restriction.BseNI.results
Bio.Restriction.Restriction.BsePI.results
Bio.Restriction.Restriction.BseRI.results
Bio.Restriction.Restriction.BseSI.results
Bio.Restriction.Restriction.BseX3I.results
Bio.Restriction.Restriction.BseXI.results
Bio.Restriction.Restriction.BseYI.results
Bio.Restriction.Restriction.BsgI.results
Bio.Restriction.Restriction.Bsh1236I.results
Bio.Restriction.Restriction.Bsh1285I.results
Bio.Restriction.Restriction.BshFI.results
Bio.Restriction.Restriction.BshNI.results
Bio.Restriction.Restriction.BshTI.results
Bio.Restriction.Restriction.BshVI.results
Bio.Restriction.Restriction.BsiEI.results
Bio.Restriction.Restriction.BsiHKAI.results
Bio.Restriction.Restriction.BsiHKCI.results
Bio.Restriction.Restriction.BsiI.results
Bio.Restriction.Restriction.BsiSI.results
Bio.Restriction.Restriction.BsiWI.results
Bio.Restriction.Restriction.BsiYI.results
Bio.Restriction.Restriction.BslFI.results
Bio.Restriction.Restriction.BslI.results
Bio.Restriction.Restriction.BsmAI.results
Bio.Restriction.Restriction.BsmBI.results
Bio.Restriction.Restriction.BsmFI.results
Bio.Restriction.Restriction.BsmI.results
Bio.Restriction.Restriction.BsnI.results
Bio.Restriction.Restriction.Bso31I.results
Bio.Restriction.Restriction.BsoBI.results
Bio.Restriction.Restriction.Bsp119I.results
Bio.Restriction.Restriction.Bsp120I.results
Bio.Restriction.Restriction.Bsp1286I.results
Bio.Restriction.Restriction.Bsp13I.results
Bio.Restriction.Restriction.Bsp1407I.results
Bio.Restriction.Restriction.Bsp143I.results
Bio.Restriction.Restriction.Bsp1720I.results
Bio.Restriction.Restriction.Bsp19I.results
Bio.Restriction.Restriction.Bsp24I.results
Bio.Restriction.Restriction.Bsp68I.results
Bio.Restriction.Restriction.BspACI.results
Bio.Restriction.Restriction.BspCNI.results
Bio.Restriction.Restriction.BspD6I.results
Bio.Restriction.Restriction.BspDI.results
Bio.Restriction.Restriction.BspEI.results
Bio.Restriction.Restriction.BspFNI.results
Bio.Restriction.Restriction.BspGI.results
Bio.Restriction.Restriction.BspHI.results
Bio.Restriction.Restriction.BspLI.results
Bio.Restriction.Restriction.BspLU11I.results
Bio.Restriction.Restriction.BspMI.results
Bio.Restriction.Restriction.BspMII.results
Bio.Restriction.Restriction.BspNCI.results
Bio.Restriction.Restriction.BspOI.results
Bio.Restriction.Restriction.BspPI.results
Bio.Restriction.Restriction.BspQI.results
Bio.Restriction.Restriction.BspT104I.results
Bio.Restriction.Restriction.BspT107I.results
Bio.Restriction.Restriction.BspTI.results
Bio.Restriction.Restriction.BsrBI.results
Bio.Restriction.Restriction.BsrDI.results
Bio.Restriction.Restriction.BsrFI.results
Bio.Restriction.Restriction.BsrGI.results
Bio.Restriction.Restriction.BsrI.results
Bio.Restriction.Restriction.BsrSI.results
Bio.Restriction.Restriction.BssAI.results
Bio.Restriction.Restriction.BssECI.results
Bio.Restriction.Restriction.BssHII.results
Bio.Restriction.Restriction.BssKI.results
Bio.Restriction.Restriction.BssMI.results
Bio.Restriction.Restriction.BssNAI.results
Bio.Restriction.Restriction.BssNI.results
Bio.Restriction.Restriction.BssSI.results
Bio.Restriction.Restriction.BssT1I.results
Bio.Restriction.Restriction.Bst1107I.results
Bio.Restriction.Restriction.Bst2BI.results
Bio.Restriction.Restriction.Bst2UI.results
Bio.Restriction.Restriction.Bst4CI.results
Bio.Restriction.Restriction.Bst6I.results
Bio.Restriction.Restriction.BstACI.results
Bio.Restriction.Restriction.BstAFI.results
Bio.Restriction.Restriction.BstAPI.results
Bio.Restriction.Restriction.BstAUI.results
Bio.Restriction.Restriction.BstBAI.results
Bio.Restriction.Restriction.BstBI.results
Bio.Restriction.Restriction.BstC8I.results
Bio.Restriction.Restriction.BstDEI.results
Bio.Restriction.Restriction.BstDSI.results
Bio.Restriction.Restriction.BstEII.results
Bio.Restriction.Restriction.BstENI.results
Bio.Restriction.Restriction.BstF5I.results
Bio.Restriction.Restriction.BstFNI.results
Bio.Restriction.Restriction.BstH2I.results
Bio.Restriction.Restriction.BstHHI.results
Bio.Restriction.Restriction.BstKTI.results
Bio.Restriction.Restriction.BstMAI.results
Bio.Restriction.Restriction.BstMBI.results
Bio.Restriction.Restriction.BstMCI.results
Bio.Restriction.Restriction.BstMWI.results
Bio.Restriction.Restriction.BstNI.results
Bio.Restriction.Restriction.BstNSI.results
Bio.Restriction.Restriction.BstOI.results
Bio.Restriction.Restriction.BstPAI.results
Bio.Restriction.Restriction.BstPI.results
Bio.Restriction.Restriction.BstSCI.results
Bio.Restriction.Restriction.BstSFI.results
Bio.Restriction.Restriction.BstSLI.results
Bio.Restriction.Restriction.BstSNI.results
Bio.Restriction.Restriction.BstUI.results
Bio.Restriction.Restriction.BstV1I.results
Bio.Restriction.Restriction.BstV2I.results
Bio.Restriction.Restriction.BstX2I.results
Bio.Restriction.Restriction.BstXI.results
Bio.Restriction.Restriction.BstYI.results
Bio.Restriction.Restriction.BstZ17I.results
Bio.Restriction.Restriction.BstZI.results
Bio.Restriction.Restriction.Bsu15I.results
Bio.Restriction.Restriction.Bsu36I.results
Bio.Restriction.Restriction.BsuI.results
Bio.Restriction.Restriction.BsuRI.results
Bio.Restriction.Restriction.BtgI.results
Bio.Restriction.Restriction.BtgZI.results
Bio.Restriction.Restriction.BthCI.results
Bio.Restriction.Restriction.BtrI.results
Bio.Restriction.Restriction.BtsCI.results
Bio.Restriction.Restriction.BtsI.results
Bio.Restriction.Restriction.BtsIMutI.results
Bio.Restriction.Restriction.BtuMI.results
Bio.Restriction.Restriction.BveI.results
Bio.Restriction.Restriction.Cac8I.results
Bio.Restriction.Restriction.CaiI.results
Bio.Restriction.Restriction.CauII.results
Bio.Restriction.Restriction.CchII.results
Bio.Restriction.Restriction.CchIII.results
Bio.Restriction.Restriction.CciI.results
Bio.Restriction.Restriction.CciNI.results
Bio.Restriction.Restriction.Cdi630V.results
Bio.Restriction.Restriction.CdiI.results
Bio.Restriction.Restriction.CdpI.results
Bio.Restriction.Restriction.CfoI.results
Bio.Restriction.Restriction.Cfr10I.results
Bio.Restriction.Restriction.Cfr13I.results
Bio.Restriction.Restriction.Cfr42I.results
Bio.Restriction.Restriction.Cfr9I.results
Bio.Restriction.Restriction.CfrI.results
Bio.Restriction.Restriction.Cgl13032I.results
Bio.Restriction.Restriction.Cgl13032II.results
Bio.Restriction.Restriction.ChaI.results
Bio.Restriction.Restriction.CjeFIII.results
Bio.Restriction.Restriction.CjeFV.results
Bio.Restriction.Restriction.CjeI.results
Bio.Restriction.Restriction.CjeNII.results
Bio.Restriction.Restriction.CjeNIII.results
Bio.Restriction.Restriction.CjeP659IV.results
Bio.Restriction.Restriction.CjePI.results
Bio.Restriction.Restriction.CjuI.results
Bio.Restriction.Restriction.CjuII.results
Bio.Restriction.Restriction.ClaI.results
Bio.Restriction.Restriction.CpoI.results
Bio.Restriction.Restriction.CseI.results
Bio.Restriction.Restriction.CsiI.results
Bio.Restriction.Restriction.Csp6I.results
Bio.Restriction.Restriction.CspAI.results
Bio.Restriction.Restriction.CspCI.results
Bio.Restriction.Restriction.CspI.results
Bio.Restriction.Restriction.CstMI.results
Bio.Restriction.Restriction.CviAII.results
Bio.Restriction.Restriction.CviJI.results
Bio.Restriction.Restriction.CviKI_1.results
Bio.Restriction.Restriction.CviQI.results
Bio.Restriction.Restriction.CviRI.results
Bio.Restriction.Restriction.DdeI.results
Bio.Restriction.Restriction.DinI.results
Bio.Restriction.Restriction.DpnI.results
Bio.Restriction.Restriction.DpnII.results
Bio.Restriction.Restriction.DraI.results
Bio.Restriction.Restriction.DraII.results
Bio.Restriction.Restriction.DraIII.results
Bio.Restriction.Restriction.DraRI.results
Bio.Restriction.Restriction.DrdI.results
Bio.Restriction.Restriction.DrdII.results
Bio.Restriction.Restriction.DriI.results
Bio.Restriction.Restriction.DsaI.results
Bio.Restriction.Restriction.DseDI.results
Bio.Restriction.Restriction.EaeI.results
Bio.Restriction.Restriction.EagI.results
Bio.Restriction.Restriction.Eam1104I.results
Bio.Restriction.Restriction.Eam1105I.results
Bio.Restriction.Restriction.EarI.results
Bio.Restriction.Restriction.EciI.results
Bio.Restriction.Restriction.Ecl136II.results
Bio.Restriction.Restriction.EclXI.results
Bio.Restriction.Restriction.Eco105I.results
Bio.Restriction.Restriction.Eco130I.results
Bio.Restriction.Restriction.Eco147I.results
Bio.Restriction.Restriction.Eco24I.results
Bio.Restriction.Restriction.Eco31I.results
Bio.Restriction.Restriction.Eco32I.results
Bio.Restriction.Restriction.Eco47I.results
Bio.Restriction.Restriction.Eco47III.results
Bio.Restriction.Restriction.Eco52I.results
Bio.Restriction.Restriction.Eco53kI.results
Bio.Restriction.Restriction.Eco57I.results
Bio.Restriction.Restriction.Eco57MI.results
Bio.Restriction.Restriction.Eco72I.results
Bio.Restriction.Restriction.Eco81I.results
Bio.Restriction.Restriction.Eco88I.results
Bio.Restriction.Restriction.Eco91I.results
Bio.Restriction.Restriction.EcoHI.results
Bio.Restriction.Restriction.EcoICRI.results
Bio.Restriction.Restriction.EcoNI.results
Bio.Restriction.Restriction.EcoO109I.results
Bio.Restriction.Restriction.EcoO65I.results
Bio.Restriction.Restriction.EcoRI.results
Bio.Restriction.Restriction.EcoRII.results
Bio.Restriction.Restriction.EcoRV.results
Bio.Restriction.Restriction.EcoT14I.results
Bio.Restriction.Restriction.EcoT22I.results
Bio.Restriction.Restriction.EcoT38I.results
Bio.Restriction.Restriction.EgeI.results
Bio.Restriction.Restriction.EheI.results
Bio.Restriction.Restriction.ErhI.results
Bio.Restriction.Restriction.EsaBC3I.results
Bio.Restriction.Restriction.EsaSSI.results
Bio.Restriction.Restriction.Esp3I.results
Bio.Restriction.Restriction.EspI.results
Bio.Restriction.Restriction.FaeI.results
Bio.Restriction.Restriction.FaiI.results
Bio.Restriction.Restriction.FalI.results
Bio.Restriction.Restriction.FaqI.results
Bio.Restriction.Restriction.FatI.results
Bio.Restriction.Restriction.FauI.results
Bio.Restriction.Restriction.FauNDI.results
Bio.Restriction.Restriction.FbaI.results
Bio.Restriction.Restriction.FblI.results
Bio.Restriction.Restriction.FinI.results
Bio.Restriction.Restriction.FmuI.results
Bio.Restriction.Restriction.Fnu4HI.results
Bio.Restriction.Restriction.FnuDII.results
Bio.Restriction.Restriction.FokI.results
Bio.Restriction.Restriction.FriOI.results
Bio.Restriction.Restriction.FseI.results
Bio.Restriction.Restriction.Fsp4HI.results
Bio.Restriction.Restriction.FspAI.results
Bio.Restriction.Restriction.FspBI.results
Bio.Restriction.Restriction.FspEI.results
Bio.Restriction.Restriction.FspI.results
Bio.Restriction.Restriction.GauT27I.results
Bio.Restriction.Restriction.GdiII.results
Bio.Restriction.Restriction.GlaI.results
Bio.Restriction.Restriction.GluI.results
Bio.Restriction.Restriction.GsaI.results
Bio.Restriction.Restriction.GsuI.results
Bio.Restriction.Restriction.HaeI.results
Bio.Restriction.Restriction.HaeII.results
Bio.Restriction.Restriction.HaeIII.results
Bio.Restriction.Restriction.HapII.results
Bio.Restriction.Restriction.HauII.results
Bio.Restriction.Restriction.HgaI.results
Bio.Restriction.Restriction.HgiAI.results
Bio.Restriction.Restriction.HgiCI.results
Bio.Restriction.Restriction.HgiEII.results
Bio.Restriction.Restriction.HgiJII.results
Bio.Restriction.Restriction.HhaI.results
Bio.Restriction.Restriction.Hin1I.results
Bio.Restriction.Restriction.Hin1II.results
Bio.Restriction.Restriction.Hin4I.results
Bio.Restriction.Restriction.Hin4II.results
Bio.Restriction.Restriction.Hin6I.results
Bio.Restriction.Restriction.HinP1I.results
Bio.Restriction.Restriction.HincII.results
Bio.Restriction.Restriction.HindII.results
Bio.Restriction.Restriction.HindIII.results
Bio.Restriction.Restriction.HinfI.results
Bio.Restriction.Restriction.HpaI.results
Bio.Restriction.Restriction.HpaII.results
Bio.Restriction.Restriction.HphI.results
Bio.Restriction.Restriction.Hpy166II.results
Bio.Restriction.Restriction.Hpy178III.results
Bio.Restriction.Restriction.Hpy188I.results
Bio.Restriction.Restriction.Hpy188III.results
Bio.Restriction.Restriction.Hpy8I.results
Bio.Restriction.Restriction.Hpy99I.results
Bio.Restriction.Restriction.Hpy99XIII.results
Bio.Restriction.Restriction.Hpy99XIV.results
Bio.Restriction.Restriction.HpyAV.results
Bio.Restriction.Restriction.HpyCH4III.results
Bio.Restriction.Restriction.HpyCH4IV.results
Bio.Restriction.Restriction.HpyCH4V.results
Bio.Restriction.Restriction.HpyF10VI.results
Bio.Restriction.Restriction.HpyF3I.results
Bio.Restriction.Restriction.HpySE526I.results
Bio.Restriction.Restriction.Hsp92I.results
Bio.Restriction.Restriction.Hsp92II.results
Bio.Restriction.Restriction.HspAI.results
Bio.Restriction.Restriction.Jma19592I.results
Bio.Restriction.Restriction.KasI.results
Bio.Restriction.Restriction.KflI.results
Bio.Restriction.Restriction.Kpn2I.results
Bio.Restriction.Restriction.KpnI.results
Bio.Restriction.Restriction.KroI.results
Bio.Restriction.Restriction.Ksp22I.results
Bio.Restriction.Restriction.Ksp632I.results
Bio.Restriction.Restriction.KspAI.results
Bio.Restriction.Restriction.KspI.results
Bio.Restriction.Restriction.Kzo9I.results
Bio.Restriction.Restriction.LguI.results
Bio.Restriction.Restriction.LpnI.results
Bio.Restriction.Restriction.LpnPI.results
Bio.Restriction.Restriction.Lsp1109I.results
Bio.Restriction.Restriction.LweI.results
Bio.Restriction.Restriction.MabI.results
Bio.Restriction.Restriction.MaeI.results
Bio.Restriction.Restriction.MaeII.results
Bio.Restriction.Restriction.MaeIII.results
Bio.Restriction.Restriction.MalI.results
Bio.Restriction.Restriction.MaqI.results
Bio.Restriction.Restriction.MauBI.results
Bio.Restriction.Restriction.MbiI.results
Bio.Restriction.Restriction.MboI.results
Bio.Restriction.Restriction.MboII.results
Bio.Restriction.Restriction.McaTI.results
Bio.Restriction.Restriction.McrI.results
Bio.Restriction.Restriction.MfeI.results
Bio.Restriction.Restriction.MflI.results
Bio.Restriction.Restriction.MhlI.results
Bio.Restriction.Restriction.MjaIV.results
Bio.Restriction.Restriction.MkaDII.results
Bio.Restriction.Restriction.MlsI.results
Bio.Restriction.Restriction.MluCI.results
Bio.Restriction.Restriction.MluI.results
Bio.Restriction.Restriction.MluNI.results
Bio.Restriction.Restriction.Mly113I.results
Bio.Restriction.Restriction.MlyI.results
Bio.Restriction.Restriction.MmeI.results
Bio.Restriction.Restriction.MnlI.results
Bio.Restriction.Restriction.Mph1103I.results
Bio.Restriction.Restriction.MreI.results
Bio.Restriction.Restriction.MroI.results
Bio.Restriction.Restriction.MroNI.results
Bio.Restriction.Restriction.MroXI.results
Bio.Restriction.Restriction.MscI.results
Bio.Restriction.Restriction.MseI.results
Bio.Restriction.Restriction.MslI.results
Bio.Restriction.Restriction.Msp20I.results
Bio.Restriction.Restriction.MspA1I.results
Bio.Restriction.Restriction.MspCI.results
Bio.Restriction.Restriction.MspI.results
Bio.Restriction.Restriction.MspJI.results
Bio.Restriction.Restriction.MspR9I.results
Bio.Restriction.Restriction.MssI.results
Bio.Restriction.Restriction.MstI.results
Bio.Restriction.Restriction.MunI.results
Bio.Restriction.Restriction.Mva1269I.results
Bio.Restriction.Restriction.MvaI.results
Bio.Restriction.Restriction.MvnI.results
Bio.Restriction.Restriction.MvrI.results
Bio.Restriction.Restriction.MwoI.results
Bio.Restriction.Restriction.NaeI.results
Bio.Restriction.Restriction.NarI.results
Bio.Restriction.Restriction.NciI.results
Bio.Restriction.Restriction.NcoI.results
Bio.Restriction.Restriction.NdeI.results
Bio.Restriction.Restriction.NdeII.results
Bio.Restriction.Restriction.NgoAVIII.results
Bio.Restriction.Restriction.NgoMIV.results
Bio.Restriction.Restriction.NhaXI.results
Bio.Restriction.Restriction.NheI.results
Bio.Restriction.Restriction.NlaCI.results
Bio.Restriction.Restriction.NlaIII.results
Bio.Restriction.Restriction.NlaIV.results
Bio.Restriction.Restriction.Nli3877I.results
Bio.Restriction.Restriction.NmeAIII.results
Bio.Restriction.Restriction.NmeDI.results
Bio.Restriction.Restriction.NmuCI.results
Bio.Restriction.Restriction.NotI.results
Bio.Restriction.Restriction.NruI.results
Bio.Restriction.Restriction.NsbI.results
Bio.Restriction.Restriction.NsiI.results
Bio.Restriction.Restriction.NspBII.results
Bio.Restriction.Restriction.NspI.results
Bio.Restriction.Restriction.NspV.results
Bio.Restriction.Restriction.OliI.results
Bio.Restriction.Restriction.PabI.results
Bio.Restriction.Restriction.PacI.results
Bio.Restriction.Restriction.PaeI.results
Bio.Restriction.Restriction.PaeR7I.results
Bio.Restriction.Restriction.PagI.results
Bio.Restriction.Restriction.PalAI.results
Bio.Restriction.Restriction.PasI.results
Bio.Restriction.Restriction.PauI.results
Bio.Restriction.Restriction.PceI.results
Bio.Restriction.Restriction.PciI.results
Bio.Restriction.Restriction.PciSI.results
Bio.Restriction.Restriction.PcsI.results
Bio.Restriction.Restriction.PctI.results
Bio.Restriction.Restriction.PdiI.results
Bio.Restriction.Restriction.PdmI.results
Bio.Restriction.Restriction.PenI.results
Bio.Restriction.Restriction.PfeI.results
Bio.Restriction.Restriction.Pfl1108I.results
Bio.Restriction.Restriction.Pfl23II.results
Bio.Restriction.Restriction.PflFI.results
Bio.Restriction.Restriction.PflMI.results
Bio.Restriction.Restriction.PfoI.results
Bio.Restriction.Restriction.PinAI.results
Bio.Restriction.Restriction.PlaDI.results
Bio.Restriction.Restriction.Ple19I.results
Bio.Restriction.Restriction.PleI.results
Bio.Restriction.Restriction.PluTI.results
Bio.Restriction.Restriction.PmaCI.results
Bio.Restriction.Restriction.PmeI.results
Bio.Restriction.Restriction.PmlI.results
Bio.Restriction.Restriction.PpiI.results
Bio.Restriction.Restriction.PpsI.results
Bio.Restriction.Restriction.Ppu10I.results
Bio.Restriction.Restriction.Ppu21I.results
Bio.Restriction.Restriction.PpuMI.results
Bio.Restriction.Restriction.PscI.results
Bio.Restriction.Restriction.PshAI.results
Bio.Restriction.Restriction.PshBI.results
Bio.Restriction.Restriction.PsiI.results
Bio.Restriction.Restriction.Psp03I.results
Bio.Restriction.Restriction.Psp124BI.results
Bio.Restriction.Restriction.Psp1406I.results
Bio.Restriction.Restriction.Psp5II.results
Bio.Restriction.Restriction.Psp6I.results
Bio.Restriction.Restriction.PspCI.results
Bio.Restriction.Restriction.PspEI.results
Bio.Restriction.Restriction.PspGI.results
Bio.Restriction.Restriction.PspLI.results
Bio.Restriction.Restriction.PspN4I.results
Bio.Restriction.Restriction.PspOMI.results
Bio.Restriction.Restriction.PspOMII.results
Bio.Restriction.Restriction.PspPI.results
Bio.Restriction.Restriction.PspPPI.results
Bio.Restriction.Restriction.PspPRI.results
Bio.Restriction.Restriction.PspXI.results
Bio.Restriction.Restriction.PsrI.results
Bio.Restriction.Restriction.PssI.results
Bio.Restriction.Restriction.PstI.results
Bio.Restriction.Restriction.PstNI.results
Bio.Restriction.Restriction.PsuI.results
Bio.Restriction.Restriction.PsyI.results
Bio.Restriction.Restriction.PteI.results
Bio.Restriction.Restriction.PvuI.results
Bio.Restriction.Restriction.PvuII.results
Bio.Restriction.Restriction.R2_BceSIV.results
Bio.Restriction.Restriction.RceI.results
Bio.Restriction.Restriction.RdeGBI.results
Bio.Restriction.Restriction.RdeGBII.results
Bio.Restriction.Restriction.RdeGBIII.results
Bio.Restriction.Restriction.RflFIII.results
Bio.Restriction.Restriction.RgaI.results
Bio.Restriction.Restriction.RigI.results
Bio.Restriction.Restriction.RlaI.results
Bio.Restriction.Restriction.RleAI.results
Bio.Restriction.Restriction.RpaB5I.results
Bio.Restriction.Restriction.RpaBI.results
Bio.Restriction.Restriction.RpaI.results
Bio.Restriction.Restriction.RpaTI.results
Bio.Restriction.Restriction.RruI.results
Bio.Restriction.Restriction.RsaI.results
Bio.Restriction.Restriction.RsaNI.results
Bio.Restriction.Restriction.RseI.results
Bio.Restriction.Restriction.Rsr2I.results
Bio.Restriction.Restriction.RsrII.results
Bio.Restriction.Restriction.SacI.results
Bio.Restriction.Restriction.SacII.results
Bio.Restriction.Restriction.SalI.results
Bio.Restriction.Restriction.SanDI.results
Bio.Restriction.Restriction.SapI.results
Bio.Restriction.Restriction.SaqAI.results
Bio.Restriction.Restriction.SatI.results
Bio.Restriction.Restriction.Sau3AI.results
Bio.Restriction.Restriction.Sau96I.results
Bio.Restriction.Restriction.SauI.results
Bio.Restriction.Restriction.SbfI.results
Bio.Restriction.Restriction.ScaI.results
Bio.Restriction.Restriction.SchI.results
Bio.Restriction.Restriction.SciI.results
Bio.Restriction.Restriction.ScrFI.results
Bio.Restriction.Restriction.SdaI.results
Bio.Restriction.Restriction.SdeAI.results
Bio.Restriction.Restriction.SdeOSI.results
Bio.Restriction.Restriction.SduI.results
Bio.Restriction.Restriction.SecI.results
Bio.Restriction.Restriction.SelI.results
Bio.Restriction.Restriction.SetI.results
Bio.Restriction.Restriction.SexAI.results
Bio.Restriction.Restriction.SfaAI.results
Bio.Restriction.Restriction.SfaNI.results
Bio.Restriction.Restriction.SfcI.results
Bio.Restriction.Restriction.SfeI.results
Bio.Restriction.Restriction.SfiI.results
Bio.Restriction.Restriction.SfoI.results
Bio.Restriction.Restriction.Sfr274I.results
Bio.Restriction.Restriction.Sfr303I.results
Bio.Restriction.Restriction.SfuI.results
Bio.Restriction.Restriction.SgeI.results
Bio.Restriction.Restriction.SgfI.results
Bio.Restriction.Restriction.SgrAI.results
Bio.Restriction.Restriction.SgrBI.results
Bio.Restriction.Restriction.SgrDI.results
Bio.Restriction.Restriction.SgrTI.results
Bio.Restriction.Restriction.SgsI.results
Bio.Restriction.Restriction.SimI.results
Bio.Restriction.Restriction.SlaI.results
Bio.Restriction.Restriction.SmaI.results
Bio.Restriction.Restriction.SmiI.results
Bio.Restriction.Restriction.SmiMI.results
Bio.Restriction.Restriction.SmlI.results
Bio.Restriction.Restriction.SmoI.results
Bio.Restriction.Restriction.SnaBI.results
Bio.Restriction.Restriction.SnaI.results
Bio.Restriction.Restriction.Sno506I.results
Bio.Restriction.Restriction.SpeI.results
Bio.Restriction.Restriction.SphI.results
Bio.Restriction.Restriction.SplI.results
Bio.Restriction.Restriction.SpoDI.results
Bio.Restriction.Restriction.SrfI.results
Bio.Restriction.Restriction.Sse232I.results
Bio.Restriction.Restriction.Sse8387I.results
Bio.Restriction.Restriction.Sse8647I.results
Bio.Restriction.Restriction.Sse9I.results
Bio.Restriction.Restriction.SseBI.results
Bio.Restriction.Restriction.SsiI.results
Bio.Restriction.Restriction.SspD5I.results
Bio.Restriction.Restriction.SspDI.results
Bio.Restriction.Restriction.SspI.results
Bio.Restriction.Restriction.SstE37I.results
Bio.Restriction.Restriction.SstI.results
Bio.Restriction.Restriction.Sth132I.results
Bio.Restriction.Restriction.Sth302II.results
Bio.Restriction.Restriction.StrI.results
Bio.Restriction.Restriction.StsI.results
Bio.Restriction.Restriction.StuI.results
Bio.Restriction.Restriction.StyD4I.results
Bio.Restriction.Restriction.StyI.results
Bio.Restriction.Restriction.SwaI.results
Bio.Restriction.Restriction.TaaI.results
Bio.Restriction.Restriction.TaiI.results
Bio.Restriction.Restriction.TaqI.results
Bio.Restriction.Restriction.TaqII.results
Bio.Restriction.Restriction.TasI.results
Bio.Restriction.Restriction.TatI.results
Bio.Restriction.Restriction.TauI.results
Bio.Restriction.Restriction.TfiI.results
Bio.Restriction.Restriction.Tru1I.results
Bio.Restriction.Restriction.Tru9I.results
Bio.Restriction.Restriction.TscAI.results
Bio.Restriction.Restriction.TseFI.results
Bio.Restriction.Restriction.TseI.results
Bio.Restriction.Restriction.TsoI.results
Bio.Restriction.Restriction.Tsp45I.results
Bio.Restriction.Restriction.Tsp4CI.results
Bio.Restriction.Restriction.TspDTI.results
Bio.Restriction.Restriction.TspEI.results
Bio.Restriction.Restriction.TspGWI.results
Bio.Restriction.Restriction.TspMI.results
Bio.Restriction.Restriction.TspRI.results
Bio.Restriction.Restriction.TssI.results
Bio.Restriction.Restriction.TstI.results
Bio.Restriction.Restriction.TsuI.results
Bio.Restriction.Restriction.Tth111I.results
Bio.Restriction.Restriction.Tth111II.results
Bio.Restriction.Restriction.UbaF11I.results
Bio.Restriction.Restriction.UbaF12I.results
Bio.Restriction.Restriction.UbaF13I.results
Bio.Restriction.Restriction.UbaF14I.results
Bio.Restriction.Restriction.UbaF9I.results
Bio.Restriction.Restriction.UbaPI.results
Bio.Restriction.Restriction.UcoMSI.results
Bio.Restriction.Restriction.UnbI.results
Bio.Restriction.Restriction.Van91I.results
Bio.Restriction.Restriction.Vha464I.results
Bio.Restriction.Restriction.VneI.results
Bio.Restriction.Restriction.VpaK11AI.results
Bio.Restriction.Restriction.VpaK11BI.results
Bio.Restriction.Restriction.VspI.results
Bio.Restriction.Restriction.WviI.results
Bio.Restriction.Restriction.XagI.results
Bio.Restriction.Restriction.XapI.results
Bio.Restriction.Restriction.XbaI.results
Bio.Restriction.Restriction.XceI.results
Bio.Restriction.Restriction.XcmI.results
Bio.Restriction.Restriction.XhoI.results
Bio.Restriction.Restriction.XhoII.results
Bio.Restriction.Restriction.XmaI.results
Bio.Restriction.Restriction.XmaIII.results
Bio.Restriction.Restriction.XmaJI.results
Bio.Restriction.Restriction.XmiI.results
Bio.Restriction.Restriction.XmnI.results
Bio.Restriction.Restriction.XspI.results
Bio.Restriction.Restriction.YkrI.results
Bio.Restriction.Restriction.ZraI.results
Bio.Restriction.Restriction.ZrmI.results
Bio.Restriction.Restriction.Zsp2I.results" class="py-name" href="#" onclick="return doclink('link-1008', 'results', 'link-996');">results</a></tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-1009" class="py-name" targets="Function Bio.Affy.CelFile.read()=Bio.Affy.CelFile-module.html#read,Function Bio.AlignIO.read()=Bio.AlignIO-module.html#read,Function Bio.Blast.NCBIXML.read()=Bio.Blast.NCBIXML-module.html#read,Function Bio.Cluster.read()=Bio.Cluster-module.html#read,Function Bio.Compass.read()=Bio.Compass-module.html#read,Function Bio.Emboss.Primer3.read()=Bio.Emboss.Primer3-module.html#read,Function Bio.Emboss.PrimerSearch.read()=Bio.Emboss.PrimerSearch-module.html#read,Method Bio.Entrez.Parser.DataHandler.read()=Bio.Entrez.Parser.DataHandler-class.html#read,Function Bio.Entrez.read()=Bio.Entrez-module.html#read,Function Bio.ExPASy.Enzyme.read()=Bio.ExPASy.Enzyme-module.html#read,Function Bio.ExPASy.Prodoc.read()=Bio.ExPASy.Prodoc-module.html#read,Function Bio.ExPASy.Prosite.read()=Bio.ExPASy.Prosite-module.html#read,Function Bio.ExPASy.ScanProsite.read()=Bio.ExPASy.ScanProsite-module.html#read,Method Bio.File.UndoHandle.read()=Bio.File.UndoHandle-class.html#read,Function Bio.KEGG.KGML.KGML_parser.read()=Bio.KEGG.KGML.KGML_parser-module.html#read,Function Bio.Medline.read()=Bio.Medline-module.html#read,Function Bio.Motif.Parsers.AlignAce.read()=Bio.Motif.Parsers.AlignAce-module.html#read,Function Bio.Motif.Parsers.MAST.read()=Bio.Motif.Parsers.MAST-module.html#read,Function Bio.Motif.Parsers.MEME.read()=Bio.Motif.Parsers.MEME-module.html#read,Function Bio.Motif.read()=Bio.Motif-module.html#read,Method Bio.NeuralNetwork.Gene.Pattern.PatternIO.read()=Bio.NeuralNetwork.Gene.Pattern.PatternIO-class.html#read,Method Bio.Nexus.Nexus.Nexus.read()=Bio.Nexus.Nexus.Nexus-class.html#read,Function Bio.Phylo.PAML.baseml.read()=Bio.Phylo.PAML.baseml-module.html#read,Function Bio.Phylo.PAML.codeml.read()=Bio.Phylo.PAML.codeml-module.html#read,Function Bio.Phylo.PAML.yn00.read()=Bio.Phylo.PAML.yn00-module.html#read,Method Bio.Phylo.PhyloXMLIO.Parser.read()=Bio.Phylo.PhyloXMLIO.Parser-class.html#read,Function Bio.Phylo.PhyloXMLIO.read()=Bio.Phylo.PhyloXMLIO-module.html#read,Function Bio.Phylo._io.read()=Bio.Phylo._io-module.html#read,Function Bio.PopGen.GenePop.FileParser.read()=Bio.PopGen.GenePop.FileParser-module.html#read,Function Bio.PopGen.GenePop.LargeFileParser.read()=Bio.PopGen.GenePop.LargeFileParser-module.html#read,Function Bio.PopGen.GenePop.read()=Bio.PopGen.GenePop-module.html#read,Function Bio.SearchIO.read()=Bio.SearchIO-module.html#read,Method Bio.SeqIO.SffIO._AddTellHandle.read()=Bio.SeqIO.SffIO._AddTellHandle-class.html#read,Function Bio.SeqIO.read()=Bio.SeqIO-module.html#read,Function Bio.Sequencing.Ace.read()=Bio.Sequencing.Ace-module.html#read,Function Bio.Sequencing.Phd.read()=Bio.Sequencing.Phd-module.html#read,Function Bio.SwissProt.read()=Bio.SwissProt-module.html#read,Function Bio.UniGene.read()=Bio.UniGene-module.html#read,Method Bio.bgzf.BgzfReader.read()=Bio.bgzf.BgzfReader-class.html#read,Function Bio.motifs.alignace.read()=Bio.motifs.alignace-module.html#read,Function Bio.motifs.jaspar.read()=Bio.motifs.jaspar-module.html#read,Function Bio.motifs.mast.read()=Bio.motifs.mast-module.html#read,Function Bio.motifs.meme.read()=Bio.motifs.meme-module.html#read,Function Bio.motifs.read()=Bio.motifs-module.html#read,Function Bio.motifs.transfac.read()=Bio.motifs.transfac-module.html#read"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-1009', 'read', 'link-1009');">read</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1798"></a><tt class="py-lineno">1798</tt>  <tt class="py-line"> </tt>
<a name="L1799"></a><tt class="py-lineno">1799</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1800"></a><tt class="py-lineno">1800</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_scanner</tt><tt class="py-op">.</tt><tt id="link-1010" class="py-name"><a title="Bio.Blast.NCBIStandalone._Scanner.feed
Bio.ExPASy.ScanProsite.Parser.feed
Bio.GenBank.Scanner.InsdcScanner.feed" class="py-name" href="#" onclick="return doclink('link-1010', 'feed', 'link-431');">feed</a></tt><tt class="py-op">(</tt><tt class="py-name">StringIO</tt><tt class="py-op">(</tt><tt id="link-1011" class="py-name"><a title="Bio.Restriction.Restriction.AanI.results
Bio.Restriction.Restriction.AarI.results
Bio.Restriction.Restriction.AasI.results
Bio.Restriction.Restriction.AatII.results
Bio.Restriction.Restriction.AbaSI.results
Bio.Restriction.Restriction.AbsI.results
Bio.Restriction.Restriction.Acc16I.results
Bio.Restriction.Restriction.Acc36I.results
Bio.Restriction.Restriction.Acc65I.results
Bio.Restriction.Restriction.AccB1I.results
Bio.Restriction.Restriction.AccB7I.results
Bio.Restriction.Restriction.AccBSI.results
Bio.Restriction.Restriction.AccI.results
Bio.Restriction.Restriction.AccII.results
Bio.Restriction.Restriction.AccIII.results
Bio.Restriction.Restriction.AceIII.results
Bio.Restriction.Restriction.AciI.results
Bio.Restriction.Restriction.AclI.results
Bio.Restriction.Restriction.AclWI.results
Bio.Restriction.Restriction.AcoI.results
Bio.Restriction.Restriction.AcsI.results
Bio.Restriction.Restriction.AcuI.results
Bio.Restriction.Restriction.AcvI.results
Bio.Restriction.Restriction.AcyI.results
Bio.Restriction.Restriction.AdeI.results
Bio.Restriction.Restriction.AfaI.results
Bio.Restriction.Restriction.AfeI.results
Bio.Restriction.Restriction.AfiI.results
Bio.Restriction.Restriction.AflII.results
Bio.Restriction.Restriction.AflIII.results
Bio.Restriction.Restriction.AgeI.results
Bio.Restriction.Restriction.AgsI.results
Bio.Restriction.Restriction.AhaIII.results
Bio.Restriction.Restriction.AhdI.results
Bio.Restriction.Restriction.AhlI.results
Bio.Restriction.Restriction.AjiI.results
Bio.Restriction.Restriction.AjnI.results
Bio.Restriction.Restriction.AjuI.results
Bio.Restriction.Restriction.AleI.results
Bio.Restriction.Restriction.AlfI.results
Bio.Restriction.Restriction.AloI.results
Bio.Restriction.Restriction.AluBI.results
Bio.Restriction.Restriction.AluI.results
Bio.Restriction.Restriction.Alw21I.results
Bio.Restriction.Restriction.Alw26I.results
Bio.Restriction.Restriction.Alw44I.results
Bio.Restriction.Restriction.AlwFI.results
Bio.Restriction.Restriction.AlwI.results
Bio.Restriction.Restriction.AlwNI.results
Bio.Restriction.Restriction.Ama87I.results
Bio.Restriction.Restriction.Aor13HI.results
Bio.Restriction.Restriction.Aor51HI.results
Bio.Restriction.Restriction.AoxI.results
Bio.Restriction.Restriction.ApaBI.results
Bio.Restriction.Restriction.ApaI.results
Bio.Restriction.Restriction.ApaLI.results
Bio.Restriction.Restriction.ApeKI.results
Bio.Restriction.Restriction.ApoI.results
Bio.Restriction.Restriction.ApyPI.results
Bio.Restriction.Restriction.AquII.results
Bio.Restriction.Restriction.AquIII.results
Bio.Restriction.Restriction.AquIV.results
Bio.Restriction.Restriction.ArsI.results
Bio.Restriction.Restriction.AscI.results
Bio.Restriction.Restriction.AseI.results
Bio.Restriction.Restriction.Asi256I.results
Bio.Restriction.Restriction.AsiGI.results
Bio.Restriction.Restriction.AsiSI.results
Bio.Restriction.Restriction.Asp700I.results
Bio.Restriction.Restriction.Asp718I.results
Bio.Restriction.Restriction.AspA2I.results
Bio.Restriction.Restriction.AspBHI.results
Bio.Restriction.Restriction.AspLEI.results
Bio.Restriction.Restriction.AspS9I.results
Bio.Restriction.Restriction.AssI.results
Bio.Restriction.Restriction.AsuC2I.results
Bio.Restriction.Restriction.AsuHPI.results
Bio.Restriction.Restriction.AsuI.results
Bio.Restriction.Restriction.AsuII.results
Bio.Restriction.Restriction.AsuNHI.results
Bio.Restriction.Restriction.AvaI.results
Bio.Restriction.Restriction.AvaII.results
Bio.Restriction.Restriction.AvaIII.results
Bio.Restriction.Restriction.AvrII.results
Bio.Restriction.Restriction.AxyI.results
Bio.Restriction.Restriction.BaeGI.results
Bio.Restriction.Restriction.BaeI.results
Bio.Restriction.Restriction.BalI.results
Bio.Restriction.Restriction.BamHI.results
Bio.Restriction.Restriction.BanI.results
Bio.Restriction.Restriction.BanII.results
Bio.Restriction.Restriction.BarI.results
Bio.Restriction.Restriction.BasI.results
Bio.Restriction.Restriction.BauI.results
Bio.Restriction.Restriction.Bbr7I.results
Bio.Restriction.Restriction.BbrPI.results
Bio.Restriction.Restriction.BbsI.results
Bio.Restriction.Restriction.Bbv12I.results
Bio.Restriction.Restriction.BbvCI.results
Bio.Restriction.Restriction.BbvI.results
Bio.Restriction.Restriction.BbvII.results
Bio.Restriction.Restriction.BccI.results
Bio.Restriction.Restriction.Bce83I.results
Bio.Restriction.Restriction.BceAI.results
Bio.Restriction.Restriction.BcefI.results
Bio.Restriction.Restriction.BcgI.results
Bio.Restriction.Restriction.BciT130I.results
Bio.Restriction.Restriction.BciVI.results
Bio.Restriction.Restriction.BclI.results
Bio.Restriction.Restriction.BcnI.results
Bio.Restriction.Restriction.BcoDI.results
Bio.Restriction.Restriction.BcuI.results
Bio.Restriction.Restriction.BdaI.results
Bio.Restriction.Restriction.BetI.results
Bio.Restriction.Restriction.BfaI.results
Bio.Restriction.Restriction.BfiI.results
Bio.Restriction.Restriction.BfmI.results
Bio.Restriction.Restriction.BfoI.results
Bio.Restriction.Restriction.BfrI.results
Bio.Restriction.Restriction.BfuAI.results
Bio.Restriction.Restriction.BfuCI.results
Bio.Restriction.Restriction.BfuI.results
Bio.Restriction.Restriction.BglI.results
Bio.Restriction.Restriction.BglII.results
Bio.Restriction.Restriction.BinI.results
Bio.Restriction.Restriction.BisI.results
Bio.Restriction.Restriction.BlnI.results
Bio.Restriction.Restriction.BlpI.results
Bio.Restriction.Restriction.BlsI.results
Bio.Restriction.Restriction.BmcAI.results
Bio.Restriction.Restriction.Bme1390I.results
Bio.Restriction.Restriction.Bme18I.results
Bio.Restriction.Restriction.BmeDI.results
Bio.Restriction.Restriction.BmeRI.results
Bio.Restriction.Restriction.BmeT110I.results
Bio.Restriction.Restriction.BmgBI.results
Bio.Restriction.Restriction.BmgI.results
Bio.Restriction.Restriction.BmgT120I.results
Bio.Restriction.Restriction.BmiI.results
Bio.Restriction.Restriction.BmrFI.results
Bio.Restriction.Restriction.BmrI.results
Bio.Restriction.Restriction.BmsI.results
Bio.Restriction.Restriction.BmtI.results
Bio.Restriction.Restriction.BmuI.results
Bio.Restriction.Restriction.BoxI.results
Bio.Restriction.Restriction.BpiI.results
Bio.Restriction.Restriction.BplI.results
Bio.Restriction.Restriction.BpmI.results
Bio.Restriction.Restriction.Bpu10I.results
Bio.Restriction.Restriction.Bpu1102I.results
Bio.Restriction.Restriction.Bpu14I.results
Bio.Restriction.Restriction.BpuEI.results
Bio.Restriction.Restriction.BpuMI.results
Bio.Restriction.Restriction.BpvUI.results
Bio.Restriction.Restriction.Bsa29I.results
Bio.Restriction.Restriction.BsaAI.results
Bio.Restriction.Restriction.BsaBI.results
Bio.Restriction.Restriction.BsaHI.results
Bio.Restriction.Restriction.BsaI.results
Bio.Restriction.Restriction.BsaJI.results
Bio.Restriction.Restriction.BsaWI.results
Bio.Restriction.Restriction.BsaXI.results
Bio.Restriction.Restriction.BsbI.results
Bio.Restriction.Restriction.Bsc4I.results
Bio.Restriction.Restriction.BscAI.results
Bio.Restriction.Restriction.BscGI.results
Bio.Restriction.Restriction.Bse118I.results
Bio.Restriction.Restriction.Bse1I.results
Bio.Restriction.Restriction.Bse21I.results
Bio.Restriction.Restriction.Bse3DI.results
Bio.Restriction.Restriction.Bse8I.results
Bio.Restriction.Restriction.BseAI.results
Bio.Restriction.Restriction.BseBI.results
Bio.Restriction.Restriction.BseCI.results
Bio.Restriction.Restriction.BseDI.results
Bio.Restriction.Restriction.BseGI.results
Bio.Restriction.Restriction.BseJI.results
Bio.Restriction.Restriction.BseLI.results
Bio.Restriction.Restriction.BseMI.results
Bio.Restriction.Restriction.BseMII.results
Bio.Restriction.Restriction.BseNI.results
Bio.Restriction.Restriction.BsePI.results
Bio.Restriction.Restriction.BseRI.results
Bio.Restriction.Restriction.BseSI.results
Bio.Restriction.Restriction.BseX3I.results
Bio.Restriction.Restriction.BseXI.results
Bio.Restriction.Restriction.BseYI.results
Bio.Restriction.Restriction.BsgI.results
Bio.Restriction.Restriction.Bsh1236I.results
Bio.Restriction.Restriction.Bsh1285I.results
Bio.Restriction.Restriction.BshFI.results
Bio.Restriction.Restriction.BshNI.results
Bio.Restriction.Restriction.BshTI.results
Bio.Restriction.Restriction.BshVI.results
Bio.Restriction.Restriction.BsiEI.results
Bio.Restriction.Restriction.BsiHKAI.results
Bio.Restriction.Restriction.BsiHKCI.results
Bio.Restriction.Restriction.BsiI.results
Bio.Restriction.Restriction.BsiSI.results
Bio.Restriction.Restriction.BsiWI.results
Bio.Restriction.Restriction.BsiYI.results
Bio.Restriction.Restriction.BslFI.results
Bio.Restriction.Restriction.BslI.results
Bio.Restriction.Restriction.BsmAI.results
Bio.Restriction.Restriction.BsmBI.results
Bio.Restriction.Restriction.BsmFI.results
Bio.Restriction.Restriction.BsmI.results
Bio.Restriction.Restriction.BsnI.results
Bio.Restriction.Restriction.Bso31I.results
Bio.Restriction.Restriction.BsoBI.results
Bio.Restriction.Restriction.Bsp119I.results
Bio.Restriction.Restriction.Bsp120I.results
Bio.Restriction.Restriction.Bsp1286I.results
Bio.Restriction.Restriction.Bsp13I.results
Bio.Restriction.Restriction.Bsp1407I.results
Bio.Restriction.Restriction.Bsp143I.results
Bio.Restriction.Restriction.Bsp1720I.results
Bio.Restriction.Restriction.Bsp19I.results
Bio.Restriction.Restriction.Bsp24I.results
Bio.Restriction.Restriction.Bsp68I.results
Bio.Restriction.Restriction.BspACI.results
Bio.Restriction.Restriction.BspCNI.results
Bio.Restriction.Restriction.BspD6I.results
Bio.Restriction.Restriction.BspDI.results
Bio.Restriction.Restriction.BspEI.results
Bio.Restriction.Restriction.BspFNI.results
Bio.Restriction.Restriction.BspGI.results
Bio.Restriction.Restriction.BspHI.results
Bio.Restriction.Restriction.BspLI.results
Bio.Restriction.Restriction.BspLU11I.results
Bio.Restriction.Restriction.BspMI.results
Bio.Restriction.Restriction.BspMII.results
Bio.Restriction.Restriction.BspNCI.results
Bio.Restriction.Restriction.BspOI.results
Bio.Restriction.Restriction.BspPI.results
Bio.Restriction.Restriction.BspQI.results
Bio.Restriction.Restriction.BspT104I.results
Bio.Restriction.Restriction.BspT107I.results
Bio.Restriction.Restriction.BspTI.results
Bio.Restriction.Restriction.BsrBI.results
Bio.Restriction.Restriction.BsrDI.results
Bio.Restriction.Restriction.BsrFI.results
Bio.Restriction.Restriction.BsrGI.results
Bio.Restriction.Restriction.BsrI.results
Bio.Restriction.Restriction.BsrSI.results
Bio.Restriction.Restriction.BssAI.results
Bio.Restriction.Restriction.BssECI.results
Bio.Restriction.Restriction.BssHII.results
Bio.Restriction.Restriction.BssKI.results
Bio.Restriction.Restriction.BssMI.results
Bio.Restriction.Restriction.BssNAI.results
Bio.Restriction.Restriction.BssNI.results
Bio.Restriction.Restriction.BssSI.results
Bio.Restriction.Restriction.BssT1I.results
Bio.Restriction.Restriction.Bst1107I.results
Bio.Restriction.Restriction.Bst2BI.results
Bio.Restriction.Restriction.Bst2UI.results
Bio.Restriction.Restriction.Bst4CI.results
Bio.Restriction.Restriction.Bst6I.results
Bio.Restriction.Restriction.BstACI.results
Bio.Restriction.Restriction.BstAFI.results
Bio.Restriction.Restriction.BstAPI.results
Bio.Restriction.Restriction.BstAUI.results
Bio.Restriction.Restriction.BstBAI.results
Bio.Restriction.Restriction.BstBI.results
Bio.Restriction.Restriction.BstC8I.results
Bio.Restriction.Restriction.BstDEI.results
Bio.Restriction.Restriction.BstDSI.results
Bio.Restriction.Restriction.BstEII.results
Bio.Restriction.Restriction.BstENI.results
Bio.Restriction.Restriction.BstF5I.results
Bio.Restriction.Restriction.BstFNI.results
Bio.Restriction.Restriction.BstH2I.results
Bio.Restriction.Restriction.BstHHI.results
Bio.Restriction.Restriction.BstKTI.results
Bio.Restriction.Restriction.BstMAI.results
Bio.Restriction.Restriction.BstMBI.results
Bio.Restriction.Restriction.BstMCI.results
Bio.Restriction.Restriction.BstMWI.results
Bio.Restriction.Restriction.BstNI.results
Bio.Restriction.Restriction.BstNSI.results
Bio.Restriction.Restriction.BstOI.results
Bio.Restriction.Restriction.BstPAI.results
Bio.Restriction.Restriction.BstPI.results
Bio.Restriction.Restriction.BstSCI.results
Bio.Restriction.Restriction.BstSFI.results
Bio.Restriction.Restriction.BstSLI.results
Bio.Restriction.Restriction.BstSNI.results
Bio.Restriction.Restriction.BstUI.results
Bio.Restriction.Restriction.BstV1I.results
Bio.Restriction.Restriction.BstV2I.results
Bio.Restriction.Restriction.BstX2I.results
Bio.Restriction.Restriction.BstXI.results
Bio.Restriction.Restriction.BstYI.results
Bio.Restriction.Restriction.BstZ17I.results
Bio.Restriction.Restriction.BstZI.results
Bio.Restriction.Restriction.Bsu15I.results
Bio.Restriction.Restriction.Bsu36I.results
Bio.Restriction.Restriction.BsuI.results
Bio.Restriction.Restriction.BsuRI.results
Bio.Restriction.Restriction.BtgI.results
Bio.Restriction.Restriction.BtgZI.results
Bio.Restriction.Restriction.BthCI.results
Bio.Restriction.Restriction.BtrI.results
Bio.Restriction.Restriction.BtsCI.results
Bio.Restriction.Restriction.BtsI.results
Bio.Restriction.Restriction.BtsIMutI.results
Bio.Restriction.Restriction.BtuMI.results
Bio.Restriction.Restriction.BveI.results
Bio.Restriction.Restriction.Cac8I.results
Bio.Restriction.Restriction.CaiI.results
Bio.Restriction.Restriction.CauII.results
Bio.Restriction.Restriction.CchII.results
Bio.Restriction.Restriction.CchIII.results
Bio.Restriction.Restriction.CciI.results
Bio.Restriction.Restriction.CciNI.results
Bio.Restriction.Restriction.Cdi630V.results
Bio.Restriction.Restriction.CdiI.results
Bio.Restriction.Restriction.CdpI.results
Bio.Restriction.Restriction.CfoI.results
Bio.Restriction.Restriction.Cfr10I.results
Bio.Restriction.Restriction.Cfr13I.results
Bio.Restriction.Restriction.Cfr42I.results
Bio.Restriction.Restriction.Cfr9I.results
Bio.Restriction.Restriction.CfrI.results
Bio.Restriction.Restriction.Cgl13032I.results
Bio.Restriction.Restriction.Cgl13032II.results
Bio.Restriction.Restriction.ChaI.results
Bio.Restriction.Restriction.CjeFIII.results
Bio.Restriction.Restriction.CjeFV.results
Bio.Restriction.Restriction.CjeI.results
Bio.Restriction.Restriction.CjeNII.results
Bio.Restriction.Restriction.CjeNIII.results
Bio.Restriction.Restriction.CjeP659IV.results
Bio.Restriction.Restriction.CjePI.results
Bio.Restriction.Restriction.CjuI.results
Bio.Restriction.Restriction.CjuII.results
Bio.Restriction.Restriction.ClaI.results
Bio.Restriction.Restriction.CpoI.results
Bio.Restriction.Restriction.CseI.results
Bio.Restriction.Restriction.CsiI.results
Bio.Restriction.Restriction.Csp6I.results
Bio.Restriction.Restriction.CspAI.results
Bio.Restriction.Restriction.CspCI.results
Bio.Restriction.Restriction.CspI.results
Bio.Restriction.Restriction.CstMI.results
Bio.Restriction.Restriction.CviAII.results
Bio.Restriction.Restriction.CviJI.results
Bio.Restriction.Restriction.CviKI_1.results
Bio.Restriction.Restriction.CviQI.results
Bio.Restriction.Restriction.CviRI.results
Bio.Restriction.Restriction.DdeI.results
Bio.Restriction.Restriction.DinI.results
Bio.Restriction.Restriction.DpnI.results
Bio.Restriction.Restriction.DpnII.results
Bio.Restriction.Restriction.DraI.results
Bio.Restriction.Restriction.DraII.results
Bio.Restriction.Restriction.DraIII.results
Bio.Restriction.Restriction.DraRI.results
Bio.Restriction.Restriction.DrdI.results
Bio.Restriction.Restriction.DrdII.results
Bio.Restriction.Restriction.DriI.results
Bio.Restriction.Restriction.DsaI.results
Bio.Restriction.Restriction.DseDI.results
Bio.Restriction.Restriction.EaeI.results
Bio.Restriction.Restriction.EagI.results
Bio.Restriction.Restriction.Eam1104I.results
Bio.Restriction.Restriction.Eam1105I.results
Bio.Restriction.Restriction.EarI.results
Bio.Restriction.Restriction.EciI.results
Bio.Restriction.Restriction.Ecl136II.results
Bio.Restriction.Restriction.EclXI.results
Bio.Restriction.Restriction.Eco105I.results
Bio.Restriction.Restriction.Eco130I.results
Bio.Restriction.Restriction.Eco147I.results
Bio.Restriction.Restriction.Eco24I.results
Bio.Restriction.Restriction.Eco31I.results
Bio.Restriction.Restriction.Eco32I.results
Bio.Restriction.Restriction.Eco47I.results
Bio.Restriction.Restriction.Eco47III.results
Bio.Restriction.Restriction.Eco52I.results
Bio.Restriction.Restriction.Eco53kI.results
Bio.Restriction.Restriction.Eco57I.results
Bio.Restriction.Restriction.Eco57MI.results
Bio.Restriction.Restriction.Eco72I.results
Bio.Restriction.Restriction.Eco81I.results
Bio.Restriction.Restriction.Eco88I.results
Bio.Restriction.Restriction.Eco91I.results
Bio.Restriction.Restriction.EcoHI.results
Bio.Restriction.Restriction.EcoICRI.results
Bio.Restriction.Restriction.EcoNI.results
Bio.Restriction.Restriction.EcoO109I.results
Bio.Restriction.Restriction.EcoO65I.results
Bio.Restriction.Restriction.EcoRI.results
Bio.Restriction.Restriction.EcoRII.results
Bio.Restriction.Restriction.EcoRV.results
Bio.Restriction.Restriction.EcoT14I.results
Bio.Restriction.Restriction.EcoT22I.results
Bio.Restriction.Restriction.EcoT38I.results
Bio.Restriction.Restriction.EgeI.results
Bio.Restriction.Restriction.EheI.results
Bio.Restriction.Restriction.ErhI.results
Bio.Restriction.Restriction.EsaBC3I.results
Bio.Restriction.Restriction.EsaSSI.results
Bio.Restriction.Restriction.Esp3I.results
Bio.Restriction.Restriction.EspI.results
Bio.Restriction.Restriction.FaeI.results
Bio.Restriction.Restriction.FaiI.results
Bio.Restriction.Restriction.FalI.results
Bio.Restriction.Restriction.FaqI.results
Bio.Restriction.Restriction.FatI.results
Bio.Restriction.Restriction.FauI.results
Bio.Restriction.Restriction.FauNDI.results
Bio.Restriction.Restriction.FbaI.results
Bio.Restriction.Restriction.FblI.results
Bio.Restriction.Restriction.FinI.results
Bio.Restriction.Restriction.FmuI.results
Bio.Restriction.Restriction.Fnu4HI.results
Bio.Restriction.Restriction.FnuDII.results
Bio.Restriction.Restriction.FokI.results
Bio.Restriction.Restriction.FriOI.results
Bio.Restriction.Restriction.FseI.results
Bio.Restriction.Restriction.Fsp4HI.results
Bio.Restriction.Restriction.FspAI.results
Bio.Restriction.Restriction.FspBI.results
Bio.Restriction.Restriction.FspEI.results
Bio.Restriction.Restriction.FspI.results
Bio.Restriction.Restriction.GauT27I.results
Bio.Restriction.Restriction.GdiII.results
Bio.Restriction.Restriction.GlaI.results
Bio.Restriction.Restriction.GluI.results
Bio.Restriction.Restriction.GsaI.results
Bio.Restriction.Restriction.GsuI.results
Bio.Restriction.Restriction.HaeI.results
Bio.Restriction.Restriction.HaeII.results
Bio.Restriction.Restriction.HaeIII.results
Bio.Restriction.Restriction.HapII.results
Bio.Restriction.Restriction.HauII.results
Bio.Restriction.Restriction.HgaI.results
Bio.Restriction.Restriction.HgiAI.results
Bio.Restriction.Restriction.HgiCI.results
Bio.Restriction.Restriction.HgiEII.results
Bio.Restriction.Restriction.HgiJII.results
Bio.Restriction.Restriction.HhaI.results
Bio.Restriction.Restriction.Hin1I.results
Bio.Restriction.Restriction.Hin1II.results
Bio.Restriction.Restriction.Hin4I.results
Bio.Restriction.Restriction.Hin4II.results
Bio.Restriction.Restriction.Hin6I.results
Bio.Restriction.Restriction.HinP1I.results
Bio.Restriction.Restriction.HincII.results
Bio.Restriction.Restriction.HindII.results
Bio.Restriction.Restriction.HindIII.results
Bio.Restriction.Restriction.HinfI.results
Bio.Restriction.Restriction.HpaI.results
Bio.Restriction.Restriction.HpaII.results
Bio.Restriction.Restriction.HphI.results
Bio.Restriction.Restriction.Hpy166II.results
Bio.Restriction.Restriction.Hpy178III.results
Bio.Restriction.Restriction.Hpy188I.results
Bio.Restriction.Restriction.Hpy188III.results
Bio.Restriction.Restriction.Hpy8I.results
Bio.Restriction.Restriction.Hpy99I.results
Bio.Restriction.Restriction.Hpy99XIII.results
Bio.Restriction.Restriction.Hpy99XIV.results
Bio.Restriction.Restriction.HpyAV.results
Bio.Restriction.Restriction.HpyCH4III.results
Bio.Restriction.Restriction.HpyCH4IV.results
Bio.Restriction.Restriction.HpyCH4V.results
Bio.Restriction.Restriction.HpyF10VI.results
Bio.Restriction.Restriction.HpyF3I.results
Bio.Restriction.Restriction.HpySE526I.results
Bio.Restriction.Restriction.Hsp92I.results
Bio.Restriction.Restriction.Hsp92II.results
Bio.Restriction.Restriction.HspAI.results
Bio.Restriction.Restriction.Jma19592I.results
Bio.Restriction.Restriction.KasI.results
Bio.Restriction.Restriction.KflI.results
Bio.Restriction.Restriction.Kpn2I.results
Bio.Restriction.Restriction.KpnI.results
Bio.Restriction.Restriction.KroI.results
Bio.Restriction.Restriction.Ksp22I.results
Bio.Restriction.Restriction.Ksp632I.results
Bio.Restriction.Restriction.KspAI.results
Bio.Restriction.Restriction.KspI.results
Bio.Restriction.Restriction.Kzo9I.results
Bio.Restriction.Restriction.LguI.results
Bio.Restriction.Restriction.LpnI.results
Bio.Restriction.Restriction.LpnPI.results
Bio.Restriction.Restriction.Lsp1109I.results
Bio.Restriction.Restriction.LweI.results
Bio.Restriction.Restriction.MabI.results
Bio.Restriction.Restriction.MaeI.results
Bio.Restriction.Restriction.MaeII.results
Bio.Restriction.Restriction.MaeIII.results
Bio.Restriction.Restriction.MalI.results
Bio.Restriction.Restriction.MaqI.results
Bio.Restriction.Restriction.MauBI.results
Bio.Restriction.Restriction.MbiI.results
Bio.Restriction.Restriction.MboI.results
Bio.Restriction.Restriction.MboII.results
Bio.Restriction.Restriction.McaTI.results
Bio.Restriction.Restriction.McrI.results
Bio.Restriction.Restriction.MfeI.results
Bio.Restriction.Restriction.MflI.results
Bio.Restriction.Restriction.MhlI.results
Bio.Restriction.Restriction.MjaIV.results
Bio.Restriction.Restriction.MkaDII.results
Bio.Restriction.Restriction.MlsI.results
Bio.Restriction.Restriction.MluCI.results
Bio.Restriction.Restriction.MluI.results
Bio.Restriction.Restriction.MluNI.results
Bio.Restriction.Restriction.Mly113I.results
Bio.Restriction.Restriction.MlyI.results
Bio.Restriction.Restriction.MmeI.results
Bio.Restriction.Restriction.MnlI.results
Bio.Restriction.Restriction.Mph1103I.results
Bio.Restriction.Restriction.MreI.results
Bio.Restriction.Restriction.MroI.results
Bio.Restriction.Restriction.MroNI.results
Bio.Restriction.Restriction.MroXI.results
Bio.Restriction.Restriction.MscI.results
Bio.Restriction.Restriction.MseI.results
Bio.Restriction.Restriction.MslI.results
Bio.Restriction.Restriction.Msp20I.results
Bio.Restriction.Restriction.MspA1I.results
Bio.Restriction.Restriction.MspCI.results
Bio.Restriction.Restriction.MspI.results
Bio.Restriction.Restriction.MspJI.results
Bio.Restriction.Restriction.MspR9I.results
Bio.Restriction.Restriction.MssI.results
Bio.Restriction.Restriction.MstI.results
Bio.Restriction.Restriction.MunI.results
Bio.Restriction.Restriction.Mva1269I.results
Bio.Restriction.Restriction.MvaI.results
Bio.Restriction.Restriction.MvnI.results
Bio.Restriction.Restriction.MvrI.results
Bio.Restriction.Restriction.MwoI.results
Bio.Restriction.Restriction.NaeI.results
Bio.Restriction.Restriction.NarI.results
Bio.Restriction.Restriction.NciI.results
Bio.Restriction.Restriction.NcoI.results
Bio.Restriction.Restriction.NdeI.results
Bio.Restriction.Restriction.NdeII.results
Bio.Restriction.Restriction.NgoAVIII.results
Bio.Restriction.Restriction.NgoMIV.results
Bio.Restriction.Restriction.NhaXI.results
Bio.Restriction.Restriction.NheI.results
Bio.Restriction.Restriction.NlaCI.results
Bio.Restriction.Restriction.NlaIII.results
Bio.Restriction.Restriction.NlaIV.results
Bio.Restriction.Restriction.Nli3877I.results
Bio.Restriction.Restriction.NmeAIII.results
Bio.Restriction.Restriction.NmeDI.results
Bio.Restriction.Restriction.NmuCI.results
Bio.Restriction.Restriction.NotI.results
Bio.Restriction.Restriction.NruI.results
Bio.Restriction.Restriction.NsbI.results
Bio.Restriction.Restriction.NsiI.results
Bio.Restriction.Restriction.NspBII.results
Bio.Restriction.Restriction.NspI.results
Bio.Restriction.Restriction.NspV.results
Bio.Restriction.Restriction.OliI.results
Bio.Restriction.Restriction.PabI.results
Bio.Restriction.Restriction.PacI.results
Bio.Restriction.Restriction.PaeI.results
Bio.Restriction.Restriction.PaeR7I.results
Bio.Restriction.Restriction.PagI.results
Bio.Restriction.Restriction.PalAI.results
Bio.Restriction.Restriction.PasI.results
Bio.Restriction.Restriction.PauI.results
Bio.Restriction.Restriction.PceI.results
Bio.Restriction.Restriction.PciI.results
Bio.Restriction.Restriction.PciSI.results
Bio.Restriction.Restriction.PcsI.results
Bio.Restriction.Restriction.PctI.results
Bio.Restriction.Restriction.PdiI.results
Bio.Restriction.Restriction.PdmI.results
Bio.Restriction.Restriction.PenI.results
Bio.Restriction.Restriction.PfeI.results
Bio.Restriction.Restriction.Pfl1108I.results
Bio.Restriction.Restriction.Pfl23II.results
Bio.Restriction.Restriction.PflFI.results
Bio.Restriction.Restriction.PflMI.results
Bio.Restriction.Restriction.PfoI.results
Bio.Restriction.Restriction.PinAI.results
Bio.Restriction.Restriction.PlaDI.results
Bio.Restriction.Restriction.Ple19I.results
Bio.Restriction.Restriction.PleI.results
Bio.Restriction.Restriction.PluTI.results
Bio.Restriction.Restriction.PmaCI.results
Bio.Restriction.Restriction.PmeI.results
Bio.Restriction.Restriction.PmlI.results
Bio.Restriction.Restriction.PpiI.results
Bio.Restriction.Restriction.PpsI.results
Bio.Restriction.Restriction.Ppu10I.results
Bio.Restriction.Restriction.Ppu21I.results
Bio.Restriction.Restriction.PpuMI.results
Bio.Restriction.Restriction.PscI.results
Bio.Restriction.Restriction.PshAI.results
Bio.Restriction.Restriction.PshBI.results
Bio.Restriction.Restriction.PsiI.results
Bio.Restriction.Restriction.Psp03I.results
Bio.Restriction.Restriction.Psp124BI.results
Bio.Restriction.Restriction.Psp1406I.results
Bio.Restriction.Restriction.Psp5II.results
Bio.Restriction.Restriction.Psp6I.results
Bio.Restriction.Restriction.PspCI.results
Bio.Restriction.Restriction.PspEI.results
Bio.Restriction.Restriction.PspGI.results
Bio.Restriction.Restriction.PspLI.results
Bio.Restriction.Restriction.PspN4I.results
Bio.Restriction.Restriction.PspOMI.results
Bio.Restriction.Restriction.PspOMII.results
Bio.Restriction.Restriction.PspPI.results
Bio.Restriction.Restriction.PspPPI.results
Bio.Restriction.Restriction.PspPRI.results
Bio.Restriction.Restriction.PspXI.results
Bio.Restriction.Restriction.PsrI.results
Bio.Restriction.Restriction.PssI.results
Bio.Restriction.Restriction.PstI.results
Bio.Restriction.Restriction.PstNI.results
Bio.Restriction.Restriction.PsuI.results
Bio.Restriction.Restriction.PsyI.results
Bio.Restriction.Restriction.PteI.results
Bio.Restriction.Restriction.PvuI.results
Bio.Restriction.Restriction.PvuII.results
Bio.Restriction.Restriction.R2_BceSIV.results
Bio.Restriction.Restriction.RceI.results
Bio.Restriction.Restriction.RdeGBI.results
Bio.Restriction.Restriction.RdeGBII.results
Bio.Restriction.Restriction.RdeGBIII.results
Bio.Restriction.Restriction.RflFIII.results
Bio.Restriction.Restriction.RgaI.results
Bio.Restriction.Restriction.RigI.results
Bio.Restriction.Restriction.RlaI.results
Bio.Restriction.Restriction.RleAI.results
Bio.Restriction.Restriction.RpaB5I.results
Bio.Restriction.Restriction.RpaBI.results
Bio.Restriction.Restriction.RpaI.results
Bio.Restriction.Restriction.RpaTI.results
Bio.Restriction.Restriction.RruI.results
Bio.Restriction.Restriction.RsaI.results
Bio.Restriction.Restriction.RsaNI.results
Bio.Restriction.Restriction.RseI.results
Bio.Restriction.Restriction.Rsr2I.results
Bio.Restriction.Restriction.RsrII.results
Bio.Restriction.Restriction.SacI.results
Bio.Restriction.Restriction.SacII.results
Bio.Restriction.Restriction.SalI.results
Bio.Restriction.Restriction.SanDI.results
Bio.Restriction.Restriction.SapI.results
Bio.Restriction.Restriction.SaqAI.results
Bio.Restriction.Restriction.SatI.results
Bio.Restriction.Restriction.Sau3AI.results
Bio.Restriction.Restriction.Sau96I.results
Bio.Restriction.Restriction.SauI.results
Bio.Restriction.Restriction.SbfI.results
Bio.Restriction.Restriction.ScaI.results
Bio.Restriction.Restriction.SchI.results
Bio.Restriction.Restriction.SciI.results
Bio.Restriction.Restriction.ScrFI.results
Bio.Restriction.Restriction.SdaI.results
Bio.Restriction.Restriction.SdeAI.results
Bio.Restriction.Restriction.SdeOSI.results
Bio.Restriction.Restriction.SduI.results
Bio.Restriction.Restriction.SecI.results
Bio.Restriction.Restriction.SelI.results
Bio.Restriction.Restriction.SetI.results
Bio.Restriction.Restriction.SexAI.results
Bio.Restriction.Restriction.SfaAI.results
Bio.Restriction.Restriction.SfaNI.results
Bio.Restriction.Restriction.SfcI.results
Bio.Restriction.Restriction.SfeI.results
Bio.Restriction.Restriction.SfiI.results
Bio.Restriction.Restriction.SfoI.results
Bio.Restriction.Restriction.Sfr274I.results
Bio.Restriction.Restriction.Sfr303I.results
Bio.Restriction.Restriction.SfuI.results
Bio.Restriction.Restriction.SgeI.results
Bio.Restriction.Restriction.SgfI.results
Bio.Restriction.Restriction.SgrAI.results
Bio.Restriction.Restriction.SgrBI.results
Bio.Restriction.Restriction.SgrDI.results
Bio.Restriction.Restriction.SgrTI.results
Bio.Restriction.Restriction.SgsI.results
Bio.Restriction.Restriction.SimI.results
Bio.Restriction.Restriction.SlaI.results
Bio.Restriction.Restriction.SmaI.results
Bio.Restriction.Restriction.SmiI.results
Bio.Restriction.Restriction.SmiMI.results
Bio.Restriction.Restriction.SmlI.results
Bio.Restriction.Restriction.SmoI.results
Bio.Restriction.Restriction.SnaBI.results
Bio.Restriction.Restriction.SnaI.results
Bio.Restriction.Restriction.Sno506I.results
Bio.Restriction.Restriction.SpeI.results
Bio.Restriction.Restriction.SphI.results
Bio.Restriction.Restriction.SplI.results
Bio.Restriction.Restriction.SpoDI.results
Bio.Restriction.Restriction.SrfI.results
Bio.Restriction.Restriction.Sse232I.results
Bio.Restriction.Restriction.Sse8387I.results
Bio.Restriction.Restriction.Sse8647I.results
Bio.Restriction.Restriction.Sse9I.results
Bio.Restriction.Restriction.SseBI.results
Bio.Restriction.Restriction.SsiI.results
Bio.Restriction.Restriction.SspD5I.results
Bio.Restriction.Restriction.SspDI.results
Bio.Restriction.Restriction.SspI.results
Bio.Restriction.Restriction.SstE37I.results
Bio.Restriction.Restriction.SstI.results
Bio.Restriction.Restriction.Sth132I.results
Bio.Restriction.Restriction.Sth302II.results
Bio.Restriction.Restriction.StrI.results
Bio.Restriction.Restriction.StsI.results
Bio.Restriction.Restriction.StuI.results
Bio.Restriction.Restriction.StyD4I.results
Bio.Restriction.Restriction.StyI.results
Bio.Restriction.Restriction.SwaI.results
Bio.Restriction.Restriction.TaaI.results
Bio.Restriction.Restriction.TaiI.results
Bio.Restriction.Restriction.TaqI.results
Bio.Restriction.Restriction.TaqII.results
Bio.Restriction.Restriction.TasI.results
Bio.Restriction.Restriction.TatI.results
Bio.Restriction.Restriction.TauI.results
Bio.Restriction.Restriction.TfiI.results
Bio.Restriction.Restriction.Tru1I.results
Bio.Restriction.Restriction.Tru9I.results
Bio.Restriction.Restriction.TscAI.results
Bio.Restriction.Restriction.TseFI.results
Bio.Restriction.Restriction.TseI.results
Bio.Restriction.Restriction.TsoI.results
Bio.Restriction.Restriction.Tsp45I.results
Bio.Restriction.Restriction.Tsp4CI.results
Bio.Restriction.Restriction.TspDTI.results
Bio.Restriction.Restriction.TspEI.results
Bio.Restriction.Restriction.TspGWI.results
Bio.Restriction.Restriction.TspMI.results
Bio.Restriction.Restriction.TspRI.results
Bio.Restriction.Restriction.TssI.results
Bio.Restriction.Restriction.TstI.results
Bio.Restriction.Restriction.TsuI.results
Bio.Restriction.Restriction.Tth111I.results
Bio.Restriction.Restriction.Tth111II.results
Bio.Restriction.Restriction.UbaF11I.results
Bio.Restriction.Restriction.UbaF12I.results
Bio.Restriction.Restriction.UbaF13I.results
Bio.Restriction.Restriction.UbaF14I.results
Bio.Restriction.Restriction.UbaF9I.results
Bio.Restriction.Restriction.UbaPI.results
Bio.Restriction.Restriction.UcoMSI.results
Bio.Restriction.Restriction.UnbI.results
Bio.Restriction.Restriction.Van91I.results
Bio.Restriction.Restriction.Vha464I.results
Bio.Restriction.Restriction.VneI.results
Bio.Restriction.Restriction.VpaK11AI.results
Bio.Restriction.Restriction.VpaK11BI.results
Bio.Restriction.Restriction.VspI.results
Bio.Restriction.Restriction.WviI.results
Bio.Restriction.Restriction.XagI.results
Bio.Restriction.Restriction.XapI.results
Bio.Restriction.Restriction.XbaI.results
Bio.Restriction.Restriction.XceI.results
Bio.Restriction.Restriction.XcmI.results
Bio.Restriction.Restriction.XhoI.results
Bio.Restriction.Restriction.XhoII.results
Bio.Restriction.Restriction.XmaI.results
Bio.Restriction.Restriction.XmaIII.results
Bio.Restriction.Restriction.XmaJI.results
Bio.Restriction.Restriction.XmiI.results
Bio.Restriction.Restriction.XmnI.results
Bio.Restriction.Restriction.XspI.results
Bio.Restriction.Restriction.YkrI.results
Bio.Restriction.Restriction.ZraI.results
Bio.Restriction.Restriction.ZrmI.results
Bio.Restriction.Restriction.Zsp2I.results" class="py-name" href="#" onclick="return doclink('link-1011', 'results', 'link-996');">results</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_consumer</tt><tt class="py-op">)</tt> </tt>
<a name="L1801"></a><tt class="py-lineno">1801</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L1802"></a><tt class="py-lineno">1802</tt>  <tt class="py-line">            <tt class="py-comment"># if we have a bad_report_file, save the info to it first</tt> </tt>
<a name="L1803"></a><tt class="py-lineno">1803</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_bad_report_handle</tt><tt class="py-op">:</tt> </tt>
<a name="L1804"></a><tt class="py-lineno">1804</tt>  <tt class="py-line">                <tt class="py-comment"># send the info to the error handle</tt> </tt>
<a name="L1805"></a><tt class="py-lineno">1805</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_bad_report_handle</tt><tt class="py-op">.</tt><tt id="link-1012" class="py-name" targets="Function Bio.AlignIO.write()=Bio.AlignIO-module.html#write,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.write()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#write,Method Bio.NeuralNetwork.Gene.Pattern.PatternIO.write()=Bio.NeuralNetwork.Gene.Pattern.PatternIO-class.html#write,Method Bio.Phylo.CDAOIO.Writer.write()=Bio.Phylo.CDAOIO.Writer-class.html#write,Function Bio.Phylo.CDAOIO.write()=Bio.Phylo.CDAOIO-module.html#write,Method Bio.Phylo.NeXMLIO.Writer.write()=Bio.Phylo.NeXMLIO.Writer-class.html#write,Function Bio.Phylo.NeXMLIO.write()=Bio.Phylo.NeXMLIO-module.html#write,Method Bio.Phylo.NewickIO.Writer.write()=Bio.Phylo.NewickIO.Writer-class.html#write,Function Bio.Phylo.NewickIO.write()=Bio.Phylo.NewickIO-module.html#write,Function Bio.Phylo.NexusIO.write()=Bio.Phylo.NexusIO-module.html#write,Method Bio.Phylo.PhyloXMLIO.Writer.write()=Bio.Phylo.PhyloXMLIO.Writer-class.html#write,Function Bio.Phylo.PhyloXMLIO.write()=Bio.Phylo.PhyloXMLIO-module.html#write,Function Bio.Phylo._io.write()=Bio.Phylo._io-module.html#write,Function Bio.SearchIO.write()=Bio.SearchIO-module.html#write,Function Bio.SeqIO.write()=Bio.SeqIO-module.html#write,Method Bio.bgzf.BgzfWriter.write()=Bio.bgzf.BgzfWriter-class.html#write,Function Bio.motifs.jaspar.write()=Bio.motifs.jaspar-module.html#write,Function Bio.motifs.transfac.write()=Bio.motifs.transfac-module.html#write,Function Bio.motifs.write()=Bio.motifs-module.html#write"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-1012', 'write', 'link-1012');">write</a></tt><tt class="py-op">(</tt><tt id="link-1013" class="py-name"><a title="Bio.Restriction.Restriction.AanI.results
Bio.Restriction.Restriction.AarI.results
Bio.Restriction.Restriction.AasI.results
Bio.Restriction.Restriction.AatII.results
Bio.Restriction.Restriction.AbaSI.results
Bio.Restriction.Restriction.AbsI.results
Bio.Restriction.Restriction.Acc16I.results
Bio.Restriction.Restriction.Acc36I.results
Bio.Restriction.Restriction.Acc65I.results
Bio.Restriction.Restriction.AccB1I.results
Bio.Restriction.Restriction.AccB7I.results
Bio.Restriction.Restriction.AccBSI.results
Bio.Restriction.Restriction.AccI.results
Bio.Restriction.Restriction.AccII.results
Bio.Restriction.Restriction.AccIII.results
Bio.Restriction.Restriction.AceIII.results
Bio.Restriction.Restriction.AciI.results
Bio.Restriction.Restriction.AclI.results
Bio.Restriction.Restriction.AclWI.results
Bio.Restriction.Restriction.AcoI.results
Bio.Restriction.Restriction.AcsI.results
Bio.Restriction.Restriction.AcuI.results
Bio.Restriction.Restriction.AcvI.results
Bio.Restriction.Restriction.AcyI.results
Bio.Restriction.Restriction.AdeI.results
Bio.Restriction.Restriction.AfaI.results
Bio.Restriction.Restriction.AfeI.results
Bio.Restriction.Restriction.AfiI.results
Bio.Restriction.Restriction.AflII.results
Bio.Restriction.Restriction.AflIII.results
Bio.Restriction.Restriction.AgeI.results
Bio.Restriction.Restriction.AgsI.results
Bio.Restriction.Restriction.AhaIII.results
Bio.Restriction.Restriction.AhdI.results
Bio.Restriction.Restriction.AhlI.results
Bio.Restriction.Restriction.AjiI.results
Bio.Restriction.Restriction.AjnI.results
Bio.Restriction.Restriction.AjuI.results
Bio.Restriction.Restriction.AleI.results
Bio.Restriction.Restriction.AlfI.results
Bio.Restriction.Restriction.AloI.results
Bio.Restriction.Restriction.AluBI.results
Bio.Restriction.Restriction.AluI.results
Bio.Restriction.Restriction.Alw21I.results
Bio.Restriction.Restriction.Alw26I.results
Bio.Restriction.Restriction.Alw44I.results
Bio.Restriction.Restriction.AlwFI.results
Bio.Restriction.Restriction.AlwI.results
Bio.Restriction.Restriction.AlwNI.results
Bio.Restriction.Restriction.Ama87I.results
Bio.Restriction.Restriction.Aor13HI.results
Bio.Restriction.Restriction.Aor51HI.results
Bio.Restriction.Restriction.AoxI.results
Bio.Restriction.Restriction.ApaBI.results
Bio.Restriction.Restriction.ApaI.results
Bio.Restriction.Restriction.ApaLI.results
Bio.Restriction.Restriction.ApeKI.results
Bio.Restriction.Restriction.ApoI.results
Bio.Restriction.Restriction.ApyPI.results
Bio.Restriction.Restriction.AquII.results
Bio.Restriction.Restriction.AquIII.results
Bio.Restriction.Restriction.AquIV.results
Bio.Restriction.Restriction.ArsI.results
Bio.Restriction.Restriction.AscI.results
Bio.Restriction.Restriction.AseI.results
Bio.Restriction.Restriction.Asi256I.results
Bio.Restriction.Restriction.AsiGI.results
Bio.Restriction.Restriction.AsiSI.results
Bio.Restriction.Restriction.Asp700I.results
Bio.Restriction.Restriction.Asp718I.results
Bio.Restriction.Restriction.AspA2I.results
Bio.Restriction.Restriction.AspBHI.results
Bio.Restriction.Restriction.AspLEI.results
Bio.Restriction.Restriction.AspS9I.results
Bio.Restriction.Restriction.AssI.results
Bio.Restriction.Restriction.AsuC2I.results
Bio.Restriction.Restriction.AsuHPI.results
Bio.Restriction.Restriction.AsuI.results
Bio.Restriction.Restriction.AsuII.results
Bio.Restriction.Restriction.AsuNHI.results
Bio.Restriction.Restriction.AvaI.results
Bio.Restriction.Restriction.AvaII.results
Bio.Restriction.Restriction.AvaIII.results
Bio.Restriction.Restriction.AvrII.results
Bio.Restriction.Restriction.AxyI.results
Bio.Restriction.Restriction.BaeGI.results
Bio.Restriction.Restriction.BaeI.results
Bio.Restriction.Restriction.BalI.results
Bio.Restriction.Restriction.BamHI.results
Bio.Restriction.Restriction.BanI.results
Bio.Restriction.Restriction.BanII.results
Bio.Restriction.Restriction.BarI.results
Bio.Restriction.Restriction.BasI.results
Bio.Restriction.Restriction.BauI.results
Bio.Restriction.Restriction.Bbr7I.results
Bio.Restriction.Restriction.BbrPI.results
Bio.Restriction.Restriction.BbsI.results
Bio.Restriction.Restriction.Bbv12I.results
Bio.Restriction.Restriction.BbvCI.results
Bio.Restriction.Restriction.BbvI.results
Bio.Restriction.Restriction.BbvII.results
Bio.Restriction.Restriction.BccI.results
Bio.Restriction.Restriction.Bce83I.results
Bio.Restriction.Restriction.BceAI.results
Bio.Restriction.Restriction.BcefI.results
Bio.Restriction.Restriction.BcgI.results
Bio.Restriction.Restriction.BciT130I.results
Bio.Restriction.Restriction.BciVI.results
Bio.Restriction.Restriction.BclI.results
Bio.Restriction.Restriction.BcnI.results
Bio.Restriction.Restriction.BcoDI.results
Bio.Restriction.Restriction.BcuI.results
Bio.Restriction.Restriction.BdaI.results
Bio.Restriction.Restriction.BetI.results
Bio.Restriction.Restriction.BfaI.results
Bio.Restriction.Restriction.BfiI.results
Bio.Restriction.Restriction.BfmI.results
Bio.Restriction.Restriction.BfoI.results
Bio.Restriction.Restriction.BfrI.results
Bio.Restriction.Restriction.BfuAI.results
Bio.Restriction.Restriction.BfuCI.results
Bio.Restriction.Restriction.BfuI.results
Bio.Restriction.Restriction.BglI.results
Bio.Restriction.Restriction.BglII.results
Bio.Restriction.Restriction.BinI.results
Bio.Restriction.Restriction.BisI.results
Bio.Restriction.Restriction.BlnI.results
Bio.Restriction.Restriction.BlpI.results
Bio.Restriction.Restriction.BlsI.results
Bio.Restriction.Restriction.BmcAI.results
Bio.Restriction.Restriction.Bme1390I.results
Bio.Restriction.Restriction.Bme18I.results
Bio.Restriction.Restriction.BmeDI.results
Bio.Restriction.Restriction.BmeRI.results
Bio.Restriction.Restriction.BmeT110I.results
Bio.Restriction.Restriction.BmgBI.results
Bio.Restriction.Restriction.BmgI.results
Bio.Restriction.Restriction.BmgT120I.results
Bio.Restriction.Restriction.BmiI.results
Bio.Restriction.Restriction.BmrFI.results
Bio.Restriction.Restriction.BmrI.results
Bio.Restriction.Restriction.BmsI.results
Bio.Restriction.Restriction.BmtI.results
Bio.Restriction.Restriction.BmuI.results
Bio.Restriction.Restriction.BoxI.results
Bio.Restriction.Restriction.BpiI.results
Bio.Restriction.Restriction.BplI.results
Bio.Restriction.Restriction.BpmI.results
Bio.Restriction.Restriction.Bpu10I.results
Bio.Restriction.Restriction.Bpu1102I.results
Bio.Restriction.Restriction.Bpu14I.results
Bio.Restriction.Restriction.BpuEI.results
Bio.Restriction.Restriction.BpuMI.results
Bio.Restriction.Restriction.BpvUI.results
Bio.Restriction.Restriction.Bsa29I.results
Bio.Restriction.Restriction.BsaAI.results
Bio.Restriction.Restriction.BsaBI.results
Bio.Restriction.Restriction.BsaHI.results
Bio.Restriction.Restriction.BsaI.results
Bio.Restriction.Restriction.BsaJI.results
Bio.Restriction.Restriction.BsaWI.results
Bio.Restriction.Restriction.BsaXI.results
Bio.Restriction.Restriction.BsbI.results
Bio.Restriction.Restriction.Bsc4I.results
Bio.Restriction.Restriction.BscAI.results
Bio.Restriction.Restriction.BscGI.results
Bio.Restriction.Restriction.Bse118I.results
Bio.Restriction.Restriction.Bse1I.results
Bio.Restriction.Restriction.Bse21I.results
Bio.Restriction.Restriction.Bse3DI.results
Bio.Restriction.Restriction.Bse8I.results
Bio.Restriction.Restriction.BseAI.results
Bio.Restriction.Restriction.BseBI.results
Bio.Restriction.Restriction.BseCI.results
Bio.Restriction.Restriction.BseDI.results
Bio.Restriction.Restriction.BseGI.results
Bio.Restriction.Restriction.BseJI.results
Bio.Restriction.Restriction.BseLI.results
Bio.Restriction.Restriction.BseMI.results
Bio.Restriction.Restriction.BseMII.results
Bio.Restriction.Restriction.BseNI.results
Bio.Restriction.Restriction.BsePI.results
Bio.Restriction.Restriction.BseRI.results
Bio.Restriction.Restriction.BseSI.results
Bio.Restriction.Restriction.BseX3I.results
Bio.Restriction.Restriction.BseXI.results
Bio.Restriction.Restriction.BseYI.results
Bio.Restriction.Restriction.BsgI.results
Bio.Restriction.Restriction.Bsh1236I.results
Bio.Restriction.Restriction.Bsh1285I.results
Bio.Restriction.Restriction.BshFI.results
Bio.Restriction.Restriction.BshNI.results
Bio.Restriction.Restriction.BshTI.results
Bio.Restriction.Restriction.BshVI.results
Bio.Restriction.Restriction.BsiEI.results
Bio.Restriction.Restriction.BsiHKAI.results
Bio.Restriction.Restriction.BsiHKCI.results
Bio.Restriction.Restriction.BsiI.results
Bio.Restriction.Restriction.BsiSI.results
Bio.Restriction.Restriction.BsiWI.results
Bio.Restriction.Restriction.BsiYI.results
Bio.Restriction.Restriction.BslFI.results
Bio.Restriction.Restriction.BslI.results
Bio.Restriction.Restriction.BsmAI.results
Bio.Restriction.Restriction.BsmBI.results
Bio.Restriction.Restriction.BsmFI.results
Bio.Restriction.Restriction.BsmI.results
Bio.Restriction.Restriction.BsnI.results
Bio.Restriction.Restriction.Bso31I.results
Bio.Restriction.Restriction.BsoBI.results
Bio.Restriction.Restriction.Bsp119I.results
Bio.Restriction.Restriction.Bsp120I.results
Bio.Restriction.Restriction.Bsp1286I.results
Bio.Restriction.Restriction.Bsp13I.results
Bio.Restriction.Restriction.Bsp1407I.results
Bio.Restriction.Restriction.Bsp143I.results
Bio.Restriction.Restriction.Bsp1720I.results
Bio.Restriction.Restriction.Bsp19I.results
Bio.Restriction.Restriction.Bsp24I.results
Bio.Restriction.Restriction.Bsp68I.results
Bio.Restriction.Restriction.BspACI.results
Bio.Restriction.Restriction.BspCNI.results
Bio.Restriction.Restriction.BspD6I.results
Bio.Restriction.Restriction.BspDI.results
Bio.Restriction.Restriction.BspEI.results
Bio.Restriction.Restriction.BspFNI.results
Bio.Restriction.Restriction.BspGI.results
Bio.Restriction.Restriction.BspHI.results
Bio.Restriction.Restriction.BspLI.results
Bio.Restriction.Restriction.BspLU11I.results
Bio.Restriction.Restriction.BspMI.results
Bio.Restriction.Restriction.BspMII.results
Bio.Restriction.Restriction.BspNCI.results
Bio.Restriction.Restriction.BspOI.results
Bio.Restriction.Restriction.BspPI.results
Bio.Restriction.Restriction.BspQI.results
Bio.Restriction.Restriction.BspT104I.results
Bio.Restriction.Restriction.BspT107I.results
Bio.Restriction.Restriction.BspTI.results
Bio.Restriction.Restriction.BsrBI.results
Bio.Restriction.Restriction.BsrDI.results
Bio.Restriction.Restriction.BsrFI.results
Bio.Restriction.Restriction.BsrGI.results
Bio.Restriction.Restriction.BsrI.results
Bio.Restriction.Restriction.BsrSI.results
Bio.Restriction.Restriction.BssAI.results
Bio.Restriction.Restriction.BssECI.results
Bio.Restriction.Restriction.BssHII.results
Bio.Restriction.Restriction.BssKI.results
Bio.Restriction.Restriction.BssMI.results
Bio.Restriction.Restriction.BssNAI.results
Bio.Restriction.Restriction.BssNI.results
Bio.Restriction.Restriction.BssSI.results
Bio.Restriction.Restriction.BssT1I.results
Bio.Restriction.Restriction.Bst1107I.results
Bio.Restriction.Restriction.Bst2BI.results
Bio.Restriction.Restriction.Bst2UI.results
Bio.Restriction.Restriction.Bst4CI.results
Bio.Restriction.Restriction.Bst6I.results
Bio.Restriction.Restriction.BstACI.results
Bio.Restriction.Restriction.BstAFI.results
Bio.Restriction.Restriction.BstAPI.results
Bio.Restriction.Restriction.BstAUI.results
Bio.Restriction.Restriction.BstBAI.results
Bio.Restriction.Restriction.BstBI.results
Bio.Restriction.Restriction.BstC8I.results
Bio.Restriction.Restriction.BstDEI.results
Bio.Restriction.Restriction.BstDSI.results
Bio.Restriction.Restriction.BstEII.results
Bio.Restriction.Restriction.BstENI.results
Bio.Restriction.Restriction.BstF5I.results
Bio.Restriction.Restriction.BstFNI.results
Bio.Restriction.Restriction.BstH2I.results
Bio.Restriction.Restriction.BstHHI.results
Bio.Restriction.Restriction.BstKTI.results
Bio.Restriction.Restriction.BstMAI.results
Bio.Restriction.Restriction.BstMBI.results
Bio.Restriction.Restriction.BstMCI.results
Bio.Restriction.Restriction.BstMWI.results
Bio.Restriction.Restriction.BstNI.results
Bio.Restriction.Restriction.BstNSI.results
Bio.Restriction.Restriction.BstOI.results
Bio.Restriction.Restriction.BstPAI.results
Bio.Restriction.Restriction.BstPI.results
Bio.Restriction.Restriction.BstSCI.results
Bio.Restriction.Restriction.BstSFI.results
Bio.Restriction.Restriction.BstSLI.results
Bio.Restriction.Restriction.BstSNI.results
Bio.Restriction.Restriction.BstUI.results
Bio.Restriction.Restriction.BstV1I.results
Bio.Restriction.Restriction.BstV2I.results
Bio.Restriction.Restriction.BstX2I.results
Bio.Restriction.Restriction.BstXI.results
Bio.Restriction.Restriction.BstYI.results
Bio.Restriction.Restriction.BstZ17I.results
Bio.Restriction.Restriction.BstZI.results
Bio.Restriction.Restriction.Bsu15I.results
Bio.Restriction.Restriction.Bsu36I.results
Bio.Restriction.Restriction.BsuI.results
Bio.Restriction.Restriction.BsuRI.results
Bio.Restriction.Restriction.BtgI.results
Bio.Restriction.Restriction.BtgZI.results
Bio.Restriction.Restriction.BthCI.results
Bio.Restriction.Restriction.BtrI.results
Bio.Restriction.Restriction.BtsCI.results
Bio.Restriction.Restriction.BtsI.results
Bio.Restriction.Restriction.BtsIMutI.results
Bio.Restriction.Restriction.BtuMI.results
Bio.Restriction.Restriction.BveI.results
Bio.Restriction.Restriction.Cac8I.results
Bio.Restriction.Restriction.CaiI.results
Bio.Restriction.Restriction.CauII.results
Bio.Restriction.Restriction.CchII.results
Bio.Restriction.Restriction.CchIII.results
Bio.Restriction.Restriction.CciI.results
Bio.Restriction.Restriction.CciNI.results
Bio.Restriction.Restriction.Cdi630V.results
Bio.Restriction.Restriction.CdiI.results
Bio.Restriction.Restriction.CdpI.results
Bio.Restriction.Restriction.CfoI.results
Bio.Restriction.Restriction.Cfr10I.results
Bio.Restriction.Restriction.Cfr13I.results
Bio.Restriction.Restriction.Cfr42I.results
Bio.Restriction.Restriction.Cfr9I.results
Bio.Restriction.Restriction.CfrI.results
Bio.Restriction.Restriction.Cgl13032I.results
Bio.Restriction.Restriction.Cgl13032II.results
Bio.Restriction.Restriction.ChaI.results
Bio.Restriction.Restriction.CjeFIII.results
Bio.Restriction.Restriction.CjeFV.results
Bio.Restriction.Restriction.CjeI.results
Bio.Restriction.Restriction.CjeNII.results
Bio.Restriction.Restriction.CjeNIII.results
Bio.Restriction.Restriction.CjeP659IV.results
Bio.Restriction.Restriction.CjePI.results
Bio.Restriction.Restriction.CjuI.results
Bio.Restriction.Restriction.CjuII.results
Bio.Restriction.Restriction.ClaI.results
Bio.Restriction.Restriction.CpoI.results
Bio.Restriction.Restriction.CseI.results
Bio.Restriction.Restriction.CsiI.results
Bio.Restriction.Restriction.Csp6I.results
Bio.Restriction.Restriction.CspAI.results
Bio.Restriction.Restriction.CspCI.results
Bio.Restriction.Restriction.CspI.results
Bio.Restriction.Restriction.CstMI.results
Bio.Restriction.Restriction.CviAII.results
Bio.Restriction.Restriction.CviJI.results
Bio.Restriction.Restriction.CviKI_1.results
Bio.Restriction.Restriction.CviQI.results
Bio.Restriction.Restriction.CviRI.results
Bio.Restriction.Restriction.DdeI.results
Bio.Restriction.Restriction.DinI.results
Bio.Restriction.Restriction.DpnI.results
Bio.Restriction.Restriction.DpnII.results
Bio.Restriction.Restriction.DraI.results
Bio.Restriction.Restriction.DraII.results
Bio.Restriction.Restriction.DraIII.results
Bio.Restriction.Restriction.DraRI.results
Bio.Restriction.Restriction.DrdI.results
Bio.Restriction.Restriction.DrdII.results
Bio.Restriction.Restriction.DriI.results
Bio.Restriction.Restriction.DsaI.results
Bio.Restriction.Restriction.DseDI.results
Bio.Restriction.Restriction.EaeI.results
Bio.Restriction.Restriction.EagI.results
Bio.Restriction.Restriction.Eam1104I.results
Bio.Restriction.Restriction.Eam1105I.results
Bio.Restriction.Restriction.EarI.results
Bio.Restriction.Restriction.EciI.results
Bio.Restriction.Restriction.Ecl136II.results
Bio.Restriction.Restriction.EclXI.results
Bio.Restriction.Restriction.Eco105I.results
Bio.Restriction.Restriction.Eco130I.results
Bio.Restriction.Restriction.Eco147I.results
Bio.Restriction.Restriction.Eco24I.results
Bio.Restriction.Restriction.Eco31I.results
Bio.Restriction.Restriction.Eco32I.results
Bio.Restriction.Restriction.Eco47I.results
Bio.Restriction.Restriction.Eco47III.results
Bio.Restriction.Restriction.Eco52I.results
Bio.Restriction.Restriction.Eco53kI.results
Bio.Restriction.Restriction.Eco57I.results
Bio.Restriction.Restriction.Eco57MI.results
Bio.Restriction.Restriction.Eco72I.results
Bio.Restriction.Restriction.Eco81I.results
Bio.Restriction.Restriction.Eco88I.results
Bio.Restriction.Restriction.Eco91I.results
Bio.Restriction.Restriction.EcoHI.results
Bio.Restriction.Restriction.EcoICRI.results
Bio.Restriction.Restriction.EcoNI.results
Bio.Restriction.Restriction.EcoO109I.results
Bio.Restriction.Restriction.EcoO65I.results
Bio.Restriction.Restriction.EcoRI.results
Bio.Restriction.Restriction.EcoRII.results
Bio.Restriction.Restriction.EcoRV.results
Bio.Restriction.Restriction.EcoT14I.results
Bio.Restriction.Restriction.EcoT22I.results
Bio.Restriction.Restriction.EcoT38I.results
Bio.Restriction.Restriction.EgeI.results
Bio.Restriction.Restriction.EheI.results
Bio.Restriction.Restriction.ErhI.results
Bio.Restriction.Restriction.EsaBC3I.results
Bio.Restriction.Restriction.EsaSSI.results
Bio.Restriction.Restriction.Esp3I.results
Bio.Restriction.Restriction.EspI.results
Bio.Restriction.Restriction.FaeI.results
Bio.Restriction.Restriction.FaiI.results
Bio.Restriction.Restriction.FalI.results
Bio.Restriction.Restriction.FaqI.results
Bio.Restriction.Restriction.FatI.results
Bio.Restriction.Restriction.FauI.results
Bio.Restriction.Restriction.FauNDI.results
Bio.Restriction.Restriction.FbaI.results
Bio.Restriction.Restriction.FblI.results
Bio.Restriction.Restriction.FinI.results
Bio.Restriction.Restriction.FmuI.results
Bio.Restriction.Restriction.Fnu4HI.results
Bio.Restriction.Restriction.FnuDII.results
Bio.Restriction.Restriction.FokI.results
Bio.Restriction.Restriction.FriOI.results
Bio.Restriction.Restriction.FseI.results
Bio.Restriction.Restriction.Fsp4HI.results
Bio.Restriction.Restriction.FspAI.results
Bio.Restriction.Restriction.FspBI.results
Bio.Restriction.Restriction.FspEI.results
Bio.Restriction.Restriction.FspI.results
Bio.Restriction.Restriction.GauT27I.results
Bio.Restriction.Restriction.GdiII.results
Bio.Restriction.Restriction.GlaI.results
Bio.Restriction.Restriction.GluI.results
Bio.Restriction.Restriction.GsaI.results
Bio.Restriction.Restriction.GsuI.results
Bio.Restriction.Restriction.HaeI.results
Bio.Restriction.Restriction.HaeII.results
Bio.Restriction.Restriction.HaeIII.results
Bio.Restriction.Restriction.HapII.results
Bio.Restriction.Restriction.HauII.results
Bio.Restriction.Restriction.HgaI.results
Bio.Restriction.Restriction.HgiAI.results
Bio.Restriction.Restriction.HgiCI.results
Bio.Restriction.Restriction.HgiEII.results
Bio.Restriction.Restriction.HgiJII.results
Bio.Restriction.Restriction.HhaI.results
Bio.Restriction.Restriction.Hin1I.results
Bio.Restriction.Restriction.Hin1II.results
Bio.Restriction.Restriction.Hin4I.results
Bio.Restriction.Restriction.Hin4II.results
Bio.Restriction.Restriction.Hin6I.results
Bio.Restriction.Restriction.HinP1I.results
Bio.Restriction.Restriction.HincII.results
Bio.Restriction.Restriction.HindII.results
Bio.Restriction.Restriction.HindIII.results
Bio.Restriction.Restriction.HinfI.results
Bio.Restriction.Restriction.HpaI.results
Bio.Restriction.Restriction.HpaII.results
Bio.Restriction.Restriction.HphI.results
Bio.Restriction.Restriction.Hpy166II.results
Bio.Restriction.Restriction.Hpy178III.results
Bio.Restriction.Restriction.Hpy188I.results
Bio.Restriction.Restriction.Hpy188III.results
Bio.Restriction.Restriction.Hpy8I.results
Bio.Restriction.Restriction.Hpy99I.results
Bio.Restriction.Restriction.Hpy99XIII.results
Bio.Restriction.Restriction.Hpy99XIV.results
Bio.Restriction.Restriction.HpyAV.results
Bio.Restriction.Restriction.HpyCH4III.results
Bio.Restriction.Restriction.HpyCH4IV.results
Bio.Restriction.Restriction.HpyCH4V.results
Bio.Restriction.Restriction.HpyF10VI.results
Bio.Restriction.Restriction.HpyF3I.results
Bio.Restriction.Restriction.HpySE526I.results
Bio.Restriction.Restriction.Hsp92I.results
Bio.Restriction.Restriction.Hsp92II.results
Bio.Restriction.Restriction.HspAI.results
Bio.Restriction.Restriction.Jma19592I.results
Bio.Restriction.Restriction.KasI.results
Bio.Restriction.Restriction.KflI.results
Bio.Restriction.Restriction.Kpn2I.results
Bio.Restriction.Restriction.KpnI.results
Bio.Restriction.Restriction.KroI.results
Bio.Restriction.Restriction.Ksp22I.results
Bio.Restriction.Restriction.Ksp632I.results
Bio.Restriction.Restriction.KspAI.results
Bio.Restriction.Restriction.KspI.results
Bio.Restriction.Restriction.Kzo9I.results
Bio.Restriction.Restriction.LguI.results
Bio.Restriction.Restriction.LpnI.results
Bio.Restriction.Restriction.LpnPI.results
Bio.Restriction.Restriction.Lsp1109I.results
Bio.Restriction.Restriction.LweI.results
Bio.Restriction.Restriction.MabI.results
Bio.Restriction.Restriction.MaeI.results
Bio.Restriction.Restriction.MaeII.results
Bio.Restriction.Restriction.MaeIII.results
Bio.Restriction.Restriction.MalI.results
Bio.Restriction.Restriction.MaqI.results
Bio.Restriction.Restriction.MauBI.results
Bio.Restriction.Restriction.MbiI.results
Bio.Restriction.Restriction.MboI.results
Bio.Restriction.Restriction.MboII.results
Bio.Restriction.Restriction.McaTI.results
Bio.Restriction.Restriction.McrI.results
Bio.Restriction.Restriction.MfeI.results
Bio.Restriction.Restriction.MflI.results
Bio.Restriction.Restriction.MhlI.results
Bio.Restriction.Restriction.MjaIV.results
Bio.Restriction.Restriction.MkaDII.results
Bio.Restriction.Restriction.MlsI.results
Bio.Restriction.Restriction.MluCI.results
Bio.Restriction.Restriction.MluI.results
Bio.Restriction.Restriction.MluNI.results
Bio.Restriction.Restriction.Mly113I.results
Bio.Restriction.Restriction.MlyI.results
Bio.Restriction.Restriction.MmeI.results
Bio.Restriction.Restriction.MnlI.results
Bio.Restriction.Restriction.Mph1103I.results
Bio.Restriction.Restriction.MreI.results
Bio.Restriction.Restriction.MroI.results
Bio.Restriction.Restriction.MroNI.results
Bio.Restriction.Restriction.MroXI.results
Bio.Restriction.Restriction.MscI.results
Bio.Restriction.Restriction.MseI.results
Bio.Restriction.Restriction.MslI.results
Bio.Restriction.Restriction.Msp20I.results
Bio.Restriction.Restriction.MspA1I.results
Bio.Restriction.Restriction.MspCI.results
Bio.Restriction.Restriction.MspI.results
Bio.Restriction.Restriction.MspJI.results
Bio.Restriction.Restriction.MspR9I.results
Bio.Restriction.Restriction.MssI.results
Bio.Restriction.Restriction.MstI.results
Bio.Restriction.Restriction.MunI.results
Bio.Restriction.Restriction.Mva1269I.results
Bio.Restriction.Restriction.MvaI.results
Bio.Restriction.Restriction.MvnI.results
Bio.Restriction.Restriction.MvrI.results
Bio.Restriction.Restriction.MwoI.results
Bio.Restriction.Restriction.NaeI.results
Bio.Restriction.Restriction.NarI.results
Bio.Restriction.Restriction.NciI.results
Bio.Restriction.Restriction.NcoI.results
Bio.Restriction.Restriction.NdeI.results
Bio.Restriction.Restriction.NdeII.results
Bio.Restriction.Restriction.NgoAVIII.results
Bio.Restriction.Restriction.NgoMIV.results
Bio.Restriction.Restriction.NhaXI.results
Bio.Restriction.Restriction.NheI.results
Bio.Restriction.Restriction.NlaCI.results
Bio.Restriction.Restriction.NlaIII.results
Bio.Restriction.Restriction.NlaIV.results
Bio.Restriction.Restriction.Nli3877I.results
Bio.Restriction.Restriction.NmeAIII.results
Bio.Restriction.Restriction.NmeDI.results
Bio.Restriction.Restriction.NmuCI.results
Bio.Restriction.Restriction.NotI.results
Bio.Restriction.Restriction.NruI.results
Bio.Restriction.Restriction.NsbI.results
Bio.Restriction.Restriction.NsiI.results
Bio.Restriction.Restriction.NspBII.results
Bio.Restriction.Restriction.NspI.results
Bio.Restriction.Restriction.NspV.results
Bio.Restriction.Restriction.OliI.results
Bio.Restriction.Restriction.PabI.results
Bio.Restriction.Restriction.PacI.results
Bio.Restriction.Restriction.PaeI.results
Bio.Restriction.Restriction.PaeR7I.results
Bio.Restriction.Restriction.PagI.results
Bio.Restriction.Restriction.PalAI.results
Bio.Restriction.Restriction.PasI.results
Bio.Restriction.Restriction.PauI.results
Bio.Restriction.Restriction.PceI.results
Bio.Restriction.Restriction.PciI.results
Bio.Restriction.Restriction.PciSI.results
Bio.Restriction.Restriction.PcsI.results
Bio.Restriction.Restriction.PctI.results
Bio.Restriction.Restriction.PdiI.results
Bio.Restriction.Restriction.PdmI.results
Bio.Restriction.Restriction.PenI.results
Bio.Restriction.Restriction.PfeI.results
Bio.Restriction.Restriction.Pfl1108I.results
Bio.Restriction.Restriction.Pfl23II.results
Bio.Restriction.Restriction.PflFI.results
Bio.Restriction.Restriction.PflMI.results
Bio.Restriction.Restriction.PfoI.results
Bio.Restriction.Restriction.PinAI.results
Bio.Restriction.Restriction.PlaDI.results
Bio.Restriction.Restriction.Ple19I.results
Bio.Restriction.Restriction.PleI.results
Bio.Restriction.Restriction.PluTI.results
Bio.Restriction.Restriction.PmaCI.results
Bio.Restriction.Restriction.PmeI.results
Bio.Restriction.Restriction.PmlI.results
Bio.Restriction.Restriction.PpiI.results
Bio.Restriction.Restriction.PpsI.results
Bio.Restriction.Restriction.Ppu10I.results
Bio.Restriction.Restriction.Ppu21I.results
Bio.Restriction.Restriction.PpuMI.results
Bio.Restriction.Restriction.PscI.results
Bio.Restriction.Restriction.PshAI.results
Bio.Restriction.Restriction.PshBI.results
Bio.Restriction.Restriction.PsiI.results
Bio.Restriction.Restriction.Psp03I.results
Bio.Restriction.Restriction.Psp124BI.results
Bio.Restriction.Restriction.Psp1406I.results
Bio.Restriction.Restriction.Psp5II.results
Bio.Restriction.Restriction.Psp6I.results
Bio.Restriction.Restriction.PspCI.results
Bio.Restriction.Restriction.PspEI.results
Bio.Restriction.Restriction.PspGI.results
Bio.Restriction.Restriction.PspLI.results
Bio.Restriction.Restriction.PspN4I.results
Bio.Restriction.Restriction.PspOMI.results
Bio.Restriction.Restriction.PspOMII.results
Bio.Restriction.Restriction.PspPI.results
Bio.Restriction.Restriction.PspPPI.results
Bio.Restriction.Restriction.PspPRI.results
Bio.Restriction.Restriction.PspXI.results
Bio.Restriction.Restriction.PsrI.results
Bio.Restriction.Restriction.PssI.results
Bio.Restriction.Restriction.PstI.results
Bio.Restriction.Restriction.PstNI.results
Bio.Restriction.Restriction.PsuI.results
Bio.Restriction.Restriction.PsyI.results
Bio.Restriction.Restriction.PteI.results
Bio.Restriction.Restriction.PvuI.results
Bio.Restriction.Restriction.PvuII.results
Bio.Restriction.Restriction.R2_BceSIV.results
Bio.Restriction.Restriction.RceI.results
Bio.Restriction.Restriction.RdeGBI.results
Bio.Restriction.Restriction.RdeGBII.results
Bio.Restriction.Restriction.RdeGBIII.results
Bio.Restriction.Restriction.RflFIII.results
Bio.Restriction.Restriction.RgaI.results
Bio.Restriction.Restriction.RigI.results
Bio.Restriction.Restriction.RlaI.results
Bio.Restriction.Restriction.RleAI.results
Bio.Restriction.Restriction.RpaB5I.results
Bio.Restriction.Restriction.RpaBI.results
Bio.Restriction.Restriction.RpaI.results
Bio.Restriction.Restriction.RpaTI.results
Bio.Restriction.Restriction.RruI.results
Bio.Restriction.Restriction.RsaI.results
Bio.Restriction.Restriction.RsaNI.results
Bio.Restriction.Restriction.RseI.results
Bio.Restriction.Restriction.Rsr2I.results
Bio.Restriction.Restriction.RsrII.results
Bio.Restriction.Restriction.SacI.results
Bio.Restriction.Restriction.SacII.results
Bio.Restriction.Restriction.SalI.results
Bio.Restriction.Restriction.SanDI.results
Bio.Restriction.Restriction.SapI.results
Bio.Restriction.Restriction.SaqAI.results
Bio.Restriction.Restriction.SatI.results
Bio.Restriction.Restriction.Sau3AI.results
Bio.Restriction.Restriction.Sau96I.results
Bio.Restriction.Restriction.SauI.results
Bio.Restriction.Restriction.SbfI.results
Bio.Restriction.Restriction.ScaI.results
Bio.Restriction.Restriction.SchI.results
Bio.Restriction.Restriction.SciI.results
Bio.Restriction.Restriction.ScrFI.results
Bio.Restriction.Restriction.SdaI.results
Bio.Restriction.Restriction.SdeAI.results
Bio.Restriction.Restriction.SdeOSI.results
Bio.Restriction.Restriction.SduI.results
Bio.Restriction.Restriction.SecI.results
Bio.Restriction.Restriction.SelI.results
Bio.Restriction.Restriction.SetI.results
Bio.Restriction.Restriction.SexAI.results
Bio.Restriction.Restriction.SfaAI.results
Bio.Restriction.Restriction.SfaNI.results
Bio.Restriction.Restriction.SfcI.results
Bio.Restriction.Restriction.SfeI.results
Bio.Restriction.Restriction.SfiI.results
Bio.Restriction.Restriction.SfoI.results
Bio.Restriction.Restriction.Sfr274I.results
Bio.Restriction.Restriction.Sfr303I.results
Bio.Restriction.Restriction.SfuI.results
Bio.Restriction.Restriction.SgeI.results
Bio.Restriction.Restriction.SgfI.results
Bio.Restriction.Restriction.SgrAI.results
Bio.Restriction.Restriction.SgrBI.results
Bio.Restriction.Restriction.SgrDI.results
Bio.Restriction.Restriction.SgrTI.results
Bio.Restriction.Restriction.SgsI.results
Bio.Restriction.Restriction.SimI.results
Bio.Restriction.Restriction.SlaI.results
Bio.Restriction.Restriction.SmaI.results
Bio.Restriction.Restriction.SmiI.results
Bio.Restriction.Restriction.SmiMI.results
Bio.Restriction.Restriction.SmlI.results
Bio.Restriction.Restriction.SmoI.results
Bio.Restriction.Restriction.SnaBI.results
Bio.Restriction.Restriction.SnaI.results
Bio.Restriction.Restriction.Sno506I.results
Bio.Restriction.Restriction.SpeI.results
Bio.Restriction.Restriction.SphI.results
Bio.Restriction.Restriction.SplI.results
Bio.Restriction.Restriction.SpoDI.results
Bio.Restriction.Restriction.SrfI.results
Bio.Restriction.Restriction.Sse232I.results
Bio.Restriction.Restriction.Sse8387I.results
Bio.Restriction.Restriction.Sse8647I.results
Bio.Restriction.Restriction.Sse9I.results
Bio.Restriction.Restriction.SseBI.results
Bio.Restriction.Restriction.SsiI.results
Bio.Restriction.Restriction.SspD5I.results
Bio.Restriction.Restriction.SspDI.results
Bio.Restriction.Restriction.SspI.results
Bio.Restriction.Restriction.SstE37I.results
Bio.Restriction.Restriction.SstI.results
Bio.Restriction.Restriction.Sth132I.results
Bio.Restriction.Restriction.Sth302II.results
Bio.Restriction.Restriction.StrI.results
Bio.Restriction.Restriction.StsI.results
Bio.Restriction.Restriction.StuI.results
Bio.Restriction.Restriction.StyD4I.results
Bio.Restriction.Restriction.StyI.results
Bio.Restriction.Restriction.SwaI.results
Bio.Restriction.Restriction.TaaI.results
Bio.Restriction.Restriction.TaiI.results
Bio.Restriction.Restriction.TaqI.results
Bio.Restriction.Restriction.TaqII.results
Bio.Restriction.Restriction.TasI.results
Bio.Restriction.Restriction.TatI.results
Bio.Restriction.Restriction.TauI.results
Bio.Restriction.Restriction.TfiI.results
Bio.Restriction.Restriction.Tru1I.results
Bio.Restriction.Restriction.Tru9I.results
Bio.Restriction.Restriction.TscAI.results
Bio.Restriction.Restriction.TseFI.results
Bio.Restriction.Restriction.TseI.results
Bio.Restriction.Restriction.TsoI.results
Bio.Restriction.Restriction.Tsp45I.results
Bio.Restriction.Restriction.Tsp4CI.results
Bio.Restriction.Restriction.TspDTI.results
Bio.Restriction.Restriction.TspEI.results
Bio.Restriction.Restriction.TspGWI.results
Bio.Restriction.Restriction.TspMI.results
Bio.Restriction.Restriction.TspRI.results
Bio.Restriction.Restriction.TssI.results
Bio.Restriction.Restriction.TstI.results
Bio.Restriction.Restriction.TsuI.results
Bio.Restriction.Restriction.Tth111I.results
Bio.Restriction.Restriction.Tth111II.results
Bio.Restriction.Restriction.UbaF11I.results
Bio.Restriction.Restriction.UbaF12I.results
Bio.Restriction.Restriction.UbaF13I.results
Bio.Restriction.Restriction.UbaF14I.results
Bio.Restriction.Restriction.UbaF9I.results
Bio.Restriction.Restriction.UbaPI.results
Bio.Restriction.Restriction.UcoMSI.results
Bio.Restriction.Restriction.UnbI.results
Bio.Restriction.Restriction.Van91I.results
Bio.Restriction.Restriction.Vha464I.results
Bio.Restriction.Restriction.VneI.results
Bio.Restriction.Restriction.VpaK11AI.results
Bio.Restriction.Restriction.VpaK11BI.results
Bio.Restriction.Restriction.VspI.results
Bio.Restriction.Restriction.WviI.results
Bio.Restriction.Restriction.XagI.results
Bio.Restriction.Restriction.XapI.results
Bio.Restriction.Restriction.XbaI.results
Bio.Restriction.Restriction.XceI.results
Bio.Restriction.Restriction.XcmI.results
Bio.Restriction.Restriction.XhoI.results
Bio.Restriction.Restriction.XhoII.results
Bio.Restriction.Restriction.XmaI.results
Bio.Restriction.Restriction.XmaIII.results
Bio.Restriction.Restriction.XmaJI.results
Bio.Restriction.Restriction.XmiI.results
Bio.Restriction.Restriction.XmnI.results
Bio.Restriction.Restriction.XspI.results
Bio.Restriction.Restriction.YkrI.results
Bio.Restriction.Restriction.ZraI.results
Bio.Restriction.Restriction.ZrmI.results
Bio.Restriction.Restriction.Zsp2I.results" class="py-name" href="#" onclick="return doclink('link-1013', 'results', 'link-996');">results</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1806"></a><tt class="py-lineno">1806</tt>  <tt class="py-line"> </tt>
<a name="L1807"></a><tt class="py-lineno">1807</tt>  <tt class="py-line">            <tt class="py-comment"># now we want to try and diagnose the error</tt> </tt>
<a name="L1808"></a><tt class="py-lineno">1808</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-1014" class="py-name" targets="Method Bio.Blast.NCBIStandalone.BlastErrorParser._diagnose_error()=Bio.Blast.NCBIStandalone.BlastErrorParser-class.html#_diagnose_error"><a title="Bio.Blast.NCBIStandalone.BlastErrorParser._diagnose_error" class="py-name" href="#" onclick="return doclink('link-1014', '_diagnose_error', 'link-1014');">_diagnose_error</a></tt><tt class="py-op">(</tt> </tt>
<a name="L1809"></a><tt class="py-lineno">1809</tt>  <tt class="py-line">                <tt class="py-name">StringIO</tt><tt class="py-op">(</tt><tt id="link-1015" class="py-name"><a title="Bio.Restriction.Restriction.AanI.results
Bio.Restriction.Restriction.AarI.results
Bio.Restriction.Restriction.AasI.results
Bio.Restriction.Restriction.AatII.results
Bio.Restriction.Restriction.AbaSI.results
Bio.Restriction.Restriction.AbsI.results
Bio.Restriction.Restriction.Acc16I.results
Bio.Restriction.Restriction.Acc36I.results
Bio.Restriction.Restriction.Acc65I.results
Bio.Restriction.Restriction.AccB1I.results
Bio.Restriction.Restriction.AccB7I.results
Bio.Restriction.Restriction.AccBSI.results
Bio.Restriction.Restriction.AccI.results
Bio.Restriction.Restriction.AccII.results
Bio.Restriction.Restriction.AccIII.results
Bio.Restriction.Restriction.AceIII.results
Bio.Restriction.Restriction.AciI.results
Bio.Restriction.Restriction.AclI.results
Bio.Restriction.Restriction.AclWI.results
Bio.Restriction.Restriction.AcoI.results
Bio.Restriction.Restriction.AcsI.results
Bio.Restriction.Restriction.AcuI.results
Bio.Restriction.Restriction.AcvI.results
Bio.Restriction.Restriction.AcyI.results
Bio.Restriction.Restriction.AdeI.results
Bio.Restriction.Restriction.AfaI.results
Bio.Restriction.Restriction.AfeI.results
Bio.Restriction.Restriction.AfiI.results
Bio.Restriction.Restriction.AflII.results
Bio.Restriction.Restriction.AflIII.results
Bio.Restriction.Restriction.AgeI.results
Bio.Restriction.Restriction.AgsI.results
Bio.Restriction.Restriction.AhaIII.results
Bio.Restriction.Restriction.AhdI.results
Bio.Restriction.Restriction.AhlI.results
Bio.Restriction.Restriction.AjiI.results
Bio.Restriction.Restriction.AjnI.results
Bio.Restriction.Restriction.AjuI.results
Bio.Restriction.Restriction.AleI.results
Bio.Restriction.Restriction.AlfI.results
Bio.Restriction.Restriction.AloI.results
Bio.Restriction.Restriction.AluBI.results
Bio.Restriction.Restriction.AluI.results
Bio.Restriction.Restriction.Alw21I.results
Bio.Restriction.Restriction.Alw26I.results
Bio.Restriction.Restriction.Alw44I.results
Bio.Restriction.Restriction.AlwFI.results
Bio.Restriction.Restriction.AlwI.results
Bio.Restriction.Restriction.AlwNI.results
Bio.Restriction.Restriction.Ama87I.results
Bio.Restriction.Restriction.Aor13HI.results
Bio.Restriction.Restriction.Aor51HI.results
Bio.Restriction.Restriction.AoxI.results
Bio.Restriction.Restriction.ApaBI.results
Bio.Restriction.Restriction.ApaI.results
Bio.Restriction.Restriction.ApaLI.results
Bio.Restriction.Restriction.ApeKI.results
Bio.Restriction.Restriction.ApoI.results
Bio.Restriction.Restriction.ApyPI.results
Bio.Restriction.Restriction.AquII.results
Bio.Restriction.Restriction.AquIII.results
Bio.Restriction.Restriction.AquIV.results
Bio.Restriction.Restriction.ArsI.results
Bio.Restriction.Restriction.AscI.results
Bio.Restriction.Restriction.AseI.results
Bio.Restriction.Restriction.Asi256I.results
Bio.Restriction.Restriction.AsiGI.results
Bio.Restriction.Restriction.AsiSI.results
Bio.Restriction.Restriction.Asp700I.results
Bio.Restriction.Restriction.Asp718I.results
Bio.Restriction.Restriction.AspA2I.results
Bio.Restriction.Restriction.AspBHI.results
Bio.Restriction.Restriction.AspLEI.results
Bio.Restriction.Restriction.AspS9I.results
Bio.Restriction.Restriction.AssI.results
Bio.Restriction.Restriction.AsuC2I.results
Bio.Restriction.Restriction.AsuHPI.results
Bio.Restriction.Restriction.AsuI.results
Bio.Restriction.Restriction.AsuII.results
Bio.Restriction.Restriction.AsuNHI.results
Bio.Restriction.Restriction.AvaI.results
Bio.Restriction.Restriction.AvaII.results
Bio.Restriction.Restriction.AvaIII.results
Bio.Restriction.Restriction.AvrII.results
Bio.Restriction.Restriction.AxyI.results
Bio.Restriction.Restriction.BaeGI.results
Bio.Restriction.Restriction.BaeI.results
Bio.Restriction.Restriction.BalI.results
Bio.Restriction.Restriction.BamHI.results
Bio.Restriction.Restriction.BanI.results
Bio.Restriction.Restriction.BanII.results
Bio.Restriction.Restriction.BarI.results
Bio.Restriction.Restriction.BasI.results
Bio.Restriction.Restriction.BauI.results
Bio.Restriction.Restriction.Bbr7I.results
Bio.Restriction.Restriction.BbrPI.results
Bio.Restriction.Restriction.BbsI.results
Bio.Restriction.Restriction.Bbv12I.results
Bio.Restriction.Restriction.BbvCI.results
Bio.Restriction.Restriction.BbvI.results
Bio.Restriction.Restriction.BbvII.results
Bio.Restriction.Restriction.BccI.results
Bio.Restriction.Restriction.Bce83I.results
Bio.Restriction.Restriction.BceAI.results
Bio.Restriction.Restriction.BcefI.results
Bio.Restriction.Restriction.BcgI.results
Bio.Restriction.Restriction.BciT130I.results
Bio.Restriction.Restriction.BciVI.results
Bio.Restriction.Restriction.BclI.results
Bio.Restriction.Restriction.BcnI.results
Bio.Restriction.Restriction.BcoDI.results
Bio.Restriction.Restriction.BcuI.results
Bio.Restriction.Restriction.BdaI.results
Bio.Restriction.Restriction.BetI.results
Bio.Restriction.Restriction.BfaI.results
Bio.Restriction.Restriction.BfiI.results
Bio.Restriction.Restriction.BfmI.results
Bio.Restriction.Restriction.BfoI.results
Bio.Restriction.Restriction.BfrI.results
Bio.Restriction.Restriction.BfuAI.results
Bio.Restriction.Restriction.BfuCI.results
Bio.Restriction.Restriction.BfuI.results
Bio.Restriction.Restriction.BglI.results
Bio.Restriction.Restriction.BglII.results
Bio.Restriction.Restriction.BinI.results
Bio.Restriction.Restriction.BisI.results
Bio.Restriction.Restriction.BlnI.results
Bio.Restriction.Restriction.BlpI.results
Bio.Restriction.Restriction.BlsI.results
Bio.Restriction.Restriction.BmcAI.results
Bio.Restriction.Restriction.Bme1390I.results
Bio.Restriction.Restriction.Bme18I.results
Bio.Restriction.Restriction.BmeDI.results
Bio.Restriction.Restriction.BmeRI.results
Bio.Restriction.Restriction.BmeT110I.results
Bio.Restriction.Restriction.BmgBI.results
Bio.Restriction.Restriction.BmgI.results
Bio.Restriction.Restriction.BmgT120I.results
Bio.Restriction.Restriction.BmiI.results
Bio.Restriction.Restriction.BmrFI.results
Bio.Restriction.Restriction.BmrI.results
Bio.Restriction.Restriction.BmsI.results
Bio.Restriction.Restriction.BmtI.results
Bio.Restriction.Restriction.BmuI.results
Bio.Restriction.Restriction.BoxI.results
Bio.Restriction.Restriction.BpiI.results
Bio.Restriction.Restriction.BplI.results
Bio.Restriction.Restriction.BpmI.results
Bio.Restriction.Restriction.Bpu10I.results
Bio.Restriction.Restriction.Bpu1102I.results
Bio.Restriction.Restriction.Bpu14I.results
Bio.Restriction.Restriction.BpuEI.results
Bio.Restriction.Restriction.BpuMI.results
Bio.Restriction.Restriction.BpvUI.results
Bio.Restriction.Restriction.Bsa29I.results
Bio.Restriction.Restriction.BsaAI.results
Bio.Restriction.Restriction.BsaBI.results
Bio.Restriction.Restriction.BsaHI.results
Bio.Restriction.Restriction.BsaI.results
Bio.Restriction.Restriction.BsaJI.results
Bio.Restriction.Restriction.BsaWI.results
Bio.Restriction.Restriction.BsaXI.results
Bio.Restriction.Restriction.BsbI.results
Bio.Restriction.Restriction.Bsc4I.results
Bio.Restriction.Restriction.BscAI.results
Bio.Restriction.Restriction.BscGI.results
Bio.Restriction.Restriction.Bse118I.results
Bio.Restriction.Restriction.Bse1I.results
Bio.Restriction.Restriction.Bse21I.results
Bio.Restriction.Restriction.Bse3DI.results
Bio.Restriction.Restriction.Bse8I.results
Bio.Restriction.Restriction.BseAI.results
Bio.Restriction.Restriction.BseBI.results
Bio.Restriction.Restriction.BseCI.results
Bio.Restriction.Restriction.BseDI.results
Bio.Restriction.Restriction.BseGI.results
Bio.Restriction.Restriction.BseJI.results
Bio.Restriction.Restriction.BseLI.results
Bio.Restriction.Restriction.BseMI.results
Bio.Restriction.Restriction.BseMII.results
Bio.Restriction.Restriction.BseNI.results
Bio.Restriction.Restriction.BsePI.results
Bio.Restriction.Restriction.BseRI.results
Bio.Restriction.Restriction.BseSI.results
Bio.Restriction.Restriction.BseX3I.results
Bio.Restriction.Restriction.BseXI.results
Bio.Restriction.Restriction.BseYI.results
Bio.Restriction.Restriction.BsgI.results
Bio.Restriction.Restriction.Bsh1236I.results
Bio.Restriction.Restriction.Bsh1285I.results
Bio.Restriction.Restriction.BshFI.results
Bio.Restriction.Restriction.BshNI.results
Bio.Restriction.Restriction.BshTI.results
Bio.Restriction.Restriction.BshVI.results
Bio.Restriction.Restriction.BsiEI.results
Bio.Restriction.Restriction.BsiHKAI.results
Bio.Restriction.Restriction.BsiHKCI.results
Bio.Restriction.Restriction.BsiI.results
Bio.Restriction.Restriction.BsiSI.results
Bio.Restriction.Restriction.BsiWI.results
Bio.Restriction.Restriction.BsiYI.results
Bio.Restriction.Restriction.BslFI.results
Bio.Restriction.Restriction.BslI.results
Bio.Restriction.Restriction.BsmAI.results
Bio.Restriction.Restriction.BsmBI.results
Bio.Restriction.Restriction.BsmFI.results
Bio.Restriction.Restriction.BsmI.results
Bio.Restriction.Restriction.BsnI.results
Bio.Restriction.Restriction.Bso31I.results
Bio.Restriction.Restriction.BsoBI.results
Bio.Restriction.Restriction.Bsp119I.results
Bio.Restriction.Restriction.Bsp120I.results
Bio.Restriction.Restriction.Bsp1286I.results
Bio.Restriction.Restriction.Bsp13I.results
Bio.Restriction.Restriction.Bsp1407I.results
Bio.Restriction.Restriction.Bsp143I.results
Bio.Restriction.Restriction.Bsp1720I.results
Bio.Restriction.Restriction.Bsp19I.results
Bio.Restriction.Restriction.Bsp24I.results
Bio.Restriction.Restriction.Bsp68I.results
Bio.Restriction.Restriction.BspACI.results
Bio.Restriction.Restriction.BspCNI.results
Bio.Restriction.Restriction.BspD6I.results
Bio.Restriction.Restriction.BspDI.results
Bio.Restriction.Restriction.BspEI.results
Bio.Restriction.Restriction.BspFNI.results
Bio.Restriction.Restriction.BspGI.results
Bio.Restriction.Restriction.BspHI.results
Bio.Restriction.Restriction.BspLI.results
Bio.Restriction.Restriction.BspLU11I.results
Bio.Restriction.Restriction.BspMI.results
Bio.Restriction.Restriction.BspMII.results
Bio.Restriction.Restriction.BspNCI.results
Bio.Restriction.Restriction.BspOI.results
Bio.Restriction.Restriction.BspPI.results
Bio.Restriction.Restriction.BspQI.results
Bio.Restriction.Restriction.BspT104I.results
Bio.Restriction.Restriction.BspT107I.results
Bio.Restriction.Restriction.BspTI.results
Bio.Restriction.Restriction.BsrBI.results
Bio.Restriction.Restriction.BsrDI.results
Bio.Restriction.Restriction.BsrFI.results
Bio.Restriction.Restriction.BsrGI.results
Bio.Restriction.Restriction.BsrI.results
Bio.Restriction.Restriction.BsrSI.results
Bio.Restriction.Restriction.BssAI.results
Bio.Restriction.Restriction.BssECI.results
Bio.Restriction.Restriction.BssHII.results
Bio.Restriction.Restriction.BssKI.results
Bio.Restriction.Restriction.BssMI.results
Bio.Restriction.Restriction.BssNAI.results
Bio.Restriction.Restriction.BssNI.results
Bio.Restriction.Restriction.BssSI.results
Bio.Restriction.Restriction.BssT1I.results
Bio.Restriction.Restriction.Bst1107I.results
Bio.Restriction.Restriction.Bst2BI.results
Bio.Restriction.Restriction.Bst2UI.results
Bio.Restriction.Restriction.Bst4CI.results
Bio.Restriction.Restriction.Bst6I.results
Bio.Restriction.Restriction.BstACI.results
Bio.Restriction.Restriction.BstAFI.results
Bio.Restriction.Restriction.BstAPI.results
Bio.Restriction.Restriction.BstAUI.results
Bio.Restriction.Restriction.BstBAI.results
Bio.Restriction.Restriction.BstBI.results
Bio.Restriction.Restriction.BstC8I.results
Bio.Restriction.Restriction.BstDEI.results
Bio.Restriction.Restriction.BstDSI.results
Bio.Restriction.Restriction.BstEII.results
Bio.Restriction.Restriction.BstENI.results
Bio.Restriction.Restriction.BstF5I.results
Bio.Restriction.Restriction.BstFNI.results
Bio.Restriction.Restriction.BstH2I.results
Bio.Restriction.Restriction.BstHHI.results
Bio.Restriction.Restriction.BstKTI.results
Bio.Restriction.Restriction.BstMAI.results
Bio.Restriction.Restriction.BstMBI.results
Bio.Restriction.Restriction.BstMCI.results
Bio.Restriction.Restriction.BstMWI.results
Bio.Restriction.Restriction.BstNI.results
Bio.Restriction.Restriction.BstNSI.results
Bio.Restriction.Restriction.BstOI.results
Bio.Restriction.Restriction.BstPAI.results
Bio.Restriction.Restriction.BstPI.results
Bio.Restriction.Restriction.BstSCI.results
Bio.Restriction.Restriction.BstSFI.results
Bio.Restriction.Restriction.BstSLI.results
Bio.Restriction.Restriction.BstSNI.results
Bio.Restriction.Restriction.BstUI.results
Bio.Restriction.Restriction.BstV1I.results
Bio.Restriction.Restriction.BstV2I.results
Bio.Restriction.Restriction.BstX2I.results
Bio.Restriction.Restriction.BstXI.results
Bio.Restriction.Restriction.BstYI.results
Bio.Restriction.Restriction.BstZ17I.results
Bio.Restriction.Restriction.BstZI.results
Bio.Restriction.Restriction.Bsu15I.results
Bio.Restriction.Restriction.Bsu36I.results
Bio.Restriction.Restriction.BsuI.results
Bio.Restriction.Restriction.BsuRI.results
Bio.Restriction.Restriction.BtgI.results
Bio.Restriction.Restriction.BtgZI.results
Bio.Restriction.Restriction.BthCI.results
Bio.Restriction.Restriction.BtrI.results
Bio.Restriction.Restriction.BtsCI.results
Bio.Restriction.Restriction.BtsI.results
Bio.Restriction.Restriction.BtsIMutI.results
Bio.Restriction.Restriction.BtuMI.results
Bio.Restriction.Restriction.BveI.results
Bio.Restriction.Restriction.Cac8I.results
Bio.Restriction.Restriction.CaiI.results
Bio.Restriction.Restriction.CauII.results
Bio.Restriction.Restriction.CchII.results
Bio.Restriction.Restriction.CchIII.results
Bio.Restriction.Restriction.CciI.results
Bio.Restriction.Restriction.CciNI.results
Bio.Restriction.Restriction.Cdi630V.results
Bio.Restriction.Restriction.CdiI.results
Bio.Restriction.Restriction.CdpI.results
Bio.Restriction.Restriction.CfoI.results
Bio.Restriction.Restriction.Cfr10I.results
Bio.Restriction.Restriction.Cfr13I.results
Bio.Restriction.Restriction.Cfr42I.results
Bio.Restriction.Restriction.Cfr9I.results
Bio.Restriction.Restriction.CfrI.results
Bio.Restriction.Restriction.Cgl13032I.results
Bio.Restriction.Restriction.Cgl13032II.results
Bio.Restriction.Restriction.ChaI.results
Bio.Restriction.Restriction.CjeFIII.results
Bio.Restriction.Restriction.CjeFV.results
Bio.Restriction.Restriction.CjeI.results
Bio.Restriction.Restriction.CjeNII.results
Bio.Restriction.Restriction.CjeNIII.results
Bio.Restriction.Restriction.CjeP659IV.results
Bio.Restriction.Restriction.CjePI.results
Bio.Restriction.Restriction.CjuI.results
Bio.Restriction.Restriction.CjuII.results
Bio.Restriction.Restriction.ClaI.results
Bio.Restriction.Restriction.CpoI.results
Bio.Restriction.Restriction.CseI.results
Bio.Restriction.Restriction.CsiI.results
Bio.Restriction.Restriction.Csp6I.results
Bio.Restriction.Restriction.CspAI.results
Bio.Restriction.Restriction.CspCI.results
Bio.Restriction.Restriction.CspI.results
Bio.Restriction.Restriction.CstMI.results
Bio.Restriction.Restriction.CviAII.results
Bio.Restriction.Restriction.CviJI.results
Bio.Restriction.Restriction.CviKI_1.results
Bio.Restriction.Restriction.CviQI.results
Bio.Restriction.Restriction.CviRI.results
Bio.Restriction.Restriction.DdeI.results
Bio.Restriction.Restriction.DinI.results
Bio.Restriction.Restriction.DpnI.results
Bio.Restriction.Restriction.DpnII.results
Bio.Restriction.Restriction.DraI.results
Bio.Restriction.Restriction.DraII.results
Bio.Restriction.Restriction.DraIII.results
Bio.Restriction.Restriction.DraRI.results
Bio.Restriction.Restriction.DrdI.results
Bio.Restriction.Restriction.DrdII.results
Bio.Restriction.Restriction.DriI.results
Bio.Restriction.Restriction.DsaI.results
Bio.Restriction.Restriction.DseDI.results
Bio.Restriction.Restriction.EaeI.results
Bio.Restriction.Restriction.EagI.results
Bio.Restriction.Restriction.Eam1104I.results
Bio.Restriction.Restriction.Eam1105I.results
Bio.Restriction.Restriction.EarI.results
Bio.Restriction.Restriction.EciI.results
Bio.Restriction.Restriction.Ecl136II.results
Bio.Restriction.Restriction.EclXI.results
Bio.Restriction.Restriction.Eco105I.results
Bio.Restriction.Restriction.Eco130I.results
Bio.Restriction.Restriction.Eco147I.results
Bio.Restriction.Restriction.Eco24I.results
Bio.Restriction.Restriction.Eco31I.results
Bio.Restriction.Restriction.Eco32I.results
Bio.Restriction.Restriction.Eco47I.results
Bio.Restriction.Restriction.Eco47III.results
Bio.Restriction.Restriction.Eco52I.results
Bio.Restriction.Restriction.Eco53kI.results
Bio.Restriction.Restriction.Eco57I.results
Bio.Restriction.Restriction.Eco57MI.results
Bio.Restriction.Restriction.Eco72I.results
Bio.Restriction.Restriction.Eco81I.results
Bio.Restriction.Restriction.Eco88I.results
Bio.Restriction.Restriction.Eco91I.results
Bio.Restriction.Restriction.EcoHI.results
Bio.Restriction.Restriction.EcoICRI.results
Bio.Restriction.Restriction.EcoNI.results
Bio.Restriction.Restriction.EcoO109I.results
Bio.Restriction.Restriction.EcoO65I.results
Bio.Restriction.Restriction.EcoRI.results
Bio.Restriction.Restriction.EcoRII.results
Bio.Restriction.Restriction.EcoRV.results
Bio.Restriction.Restriction.EcoT14I.results
Bio.Restriction.Restriction.EcoT22I.results
Bio.Restriction.Restriction.EcoT38I.results
Bio.Restriction.Restriction.EgeI.results
Bio.Restriction.Restriction.EheI.results
Bio.Restriction.Restriction.ErhI.results
Bio.Restriction.Restriction.EsaBC3I.results
Bio.Restriction.Restriction.EsaSSI.results
Bio.Restriction.Restriction.Esp3I.results
Bio.Restriction.Restriction.EspI.results
Bio.Restriction.Restriction.FaeI.results
Bio.Restriction.Restriction.FaiI.results
Bio.Restriction.Restriction.FalI.results
Bio.Restriction.Restriction.FaqI.results
Bio.Restriction.Restriction.FatI.results
Bio.Restriction.Restriction.FauI.results
Bio.Restriction.Restriction.FauNDI.results
Bio.Restriction.Restriction.FbaI.results
Bio.Restriction.Restriction.FblI.results
Bio.Restriction.Restriction.FinI.results
Bio.Restriction.Restriction.FmuI.results
Bio.Restriction.Restriction.Fnu4HI.results
Bio.Restriction.Restriction.FnuDII.results
Bio.Restriction.Restriction.FokI.results
Bio.Restriction.Restriction.FriOI.results
Bio.Restriction.Restriction.FseI.results
Bio.Restriction.Restriction.Fsp4HI.results
Bio.Restriction.Restriction.FspAI.results
Bio.Restriction.Restriction.FspBI.results
Bio.Restriction.Restriction.FspEI.results
Bio.Restriction.Restriction.FspI.results
Bio.Restriction.Restriction.GauT27I.results
Bio.Restriction.Restriction.GdiII.results
Bio.Restriction.Restriction.GlaI.results
Bio.Restriction.Restriction.GluI.results
Bio.Restriction.Restriction.GsaI.results
Bio.Restriction.Restriction.GsuI.results
Bio.Restriction.Restriction.HaeI.results
Bio.Restriction.Restriction.HaeII.results
Bio.Restriction.Restriction.HaeIII.results
Bio.Restriction.Restriction.HapII.results
Bio.Restriction.Restriction.HauII.results
Bio.Restriction.Restriction.HgaI.results
Bio.Restriction.Restriction.HgiAI.results
Bio.Restriction.Restriction.HgiCI.results
Bio.Restriction.Restriction.HgiEII.results
Bio.Restriction.Restriction.HgiJII.results
Bio.Restriction.Restriction.HhaI.results
Bio.Restriction.Restriction.Hin1I.results
Bio.Restriction.Restriction.Hin1II.results
Bio.Restriction.Restriction.Hin4I.results
Bio.Restriction.Restriction.Hin4II.results
Bio.Restriction.Restriction.Hin6I.results
Bio.Restriction.Restriction.HinP1I.results
Bio.Restriction.Restriction.HincII.results
Bio.Restriction.Restriction.HindII.results
Bio.Restriction.Restriction.HindIII.results
Bio.Restriction.Restriction.HinfI.results
Bio.Restriction.Restriction.HpaI.results
Bio.Restriction.Restriction.HpaII.results
Bio.Restriction.Restriction.HphI.results
Bio.Restriction.Restriction.Hpy166II.results
Bio.Restriction.Restriction.Hpy178III.results
Bio.Restriction.Restriction.Hpy188I.results
Bio.Restriction.Restriction.Hpy188III.results
Bio.Restriction.Restriction.Hpy8I.results
Bio.Restriction.Restriction.Hpy99I.results
Bio.Restriction.Restriction.Hpy99XIII.results
Bio.Restriction.Restriction.Hpy99XIV.results
Bio.Restriction.Restriction.HpyAV.results
Bio.Restriction.Restriction.HpyCH4III.results
Bio.Restriction.Restriction.HpyCH4IV.results
Bio.Restriction.Restriction.HpyCH4V.results
Bio.Restriction.Restriction.HpyF10VI.results
Bio.Restriction.Restriction.HpyF3I.results
Bio.Restriction.Restriction.HpySE526I.results
Bio.Restriction.Restriction.Hsp92I.results
Bio.Restriction.Restriction.Hsp92II.results
Bio.Restriction.Restriction.HspAI.results
Bio.Restriction.Restriction.Jma19592I.results
Bio.Restriction.Restriction.KasI.results
Bio.Restriction.Restriction.KflI.results
Bio.Restriction.Restriction.Kpn2I.results
Bio.Restriction.Restriction.KpnI.results
Bio.Restriction.Restriction.KroI.results
Bio.Restriction.Restriction.Ksp22I.results
Bio.Restriction.Restriction.Ksp632I.results
Bio.Restriction.Restriction.KspAI.results
Bio.Restriction.Restriction.KspI.results
Bio.Restriction.Restriction.Kzo9I.results
Bio.Restriction.Restriction.LguI.results
Bio.Restriction.Restriction.LpnI.results
Bio.Restriction.Restriction.LpnPI.results
Bio.Restriction.Restriction.Lsp1109I.results
Bio.Restriction.Restriction.LweI.results
Bio.Restriction.Restriction.MabI.results
Bio.Restriction.Restriction.MaeI.results
Bio.Restriction.Restriction.MaeII.results
Bio.Restriction.Restriction.MaeIII.results
Bio.Restriction.Restriction.MalI.results
Bio.Restriction.Restriction.MaqI.results
Bio.Restriction.Restriction.MauBI.results
Bio.Restriction.Restriction.MbiI.results
Bio.Restriction.Restriction.MboI.results
Bio.Restriction.Restriction.MboII.results
Bio.Restriction.Restriction.McaTI.results
Bio.Restriction.Restriction.McrI.results
Bio.Restriction.Restriction.MfeI.results
Bio.Restriction.Restriction.MflI.results
Bio.Restriction.Restriction.MhlI.results
Bio.Restriction.Restriction.MjaIV.results
Bio.Restriction.Restriction.MkaDII.results
Bio.Restriction.Restriction.MlsI.results
Bio.Restriction.Restriction.MluCI.results
Bio.Restriction.Restriction.MluI.results
Bio.Restriction.Restriction.MluNI.results
Bio.Restriction.Restriction.Mly113I.results
Bio.Restriction.Restriction.MlyI.results
Bio.Restriction.Restriction.MmeI.results
Bio.Restriction.Restriction.MnlI.results
Bio.Restriction.Restriction.Mph1103I.results
Bio.Restriction.Restriction.MreI.results
Bio.Restriction.Restriction.MroI.results
Bio.Restriction.Restriction.MroNI.results
Bio.Restriction.Restriction.MroXI.results
Bio.Restriction.Restriction.MscI.results
Bio.Restriction.Restriction.MseI.results
Bio.Restriction.Restriction.MslI.results
Bio.Restriction.Restriction.Msp20I.results
Bio.Restriction.Restriction.MspA1I.results
Bio.Restriction.Restriction.MspCI.results
Bio.Restriction.Restriction.MspI.results
Bio.Restriction.Restriction.MspJI.results
Bio.Restriction.Restriction.MspR9I.results
Bio.Restriction.Restriction.MssI.results
Bio.Restriction.Restriction.MstI.results
Bio.Restriction.Restriction.MunI.results
Bio.Restriction.Restriction.Mva1269I.results
Bio.Restriction.Restriction.MvaI.results
Bio.Restriction.Restriction.MvnI.results
Bio.Restriction.Restriction.MvrI.results
Bio.Restriction.Restriction.MwoI.results
Bio.Restriction.Restriction.NaeI.results
Bio.Restriction.Restriction.NarI.results
Bio.Restriction.Restriction.NciI.results
Bio.Restriction.Restriction.NcoI.results
Bio.Restriction.Restriction.NdeI.results
Bio.Restriction.Restriction.NdeII.results
Bio.Restriction.Restriction.NgoAVIII.results
Bio.Restriction.Restriction.NgoMIV.results
Bio.Restriction.Restriction.NhaXI.results
Bio.Restriction.Restriction.NheI.results
Bio.Restriction.Restriction.NlaCI.results
Bio.Restriction.Restriction.NlaIII.results
Bio.Restriction.Restriction.NlaIV.results
Bio.Restriction.Restriction.Nli3877I.results
Bio.Restriction.Restriction.NmeAIII.results
Bio.Restriction.Restriction.NmeDI.results
Bio.Restriction.Restriction.NmuCI.results
Bio.Restriction.Restriction.NotI.results
Bio.Restriction.Restriction.NruI.results
Bio.Restriction.Restriction.NsbI.results
Bio.Restriction.Restriction.NsiI.results
Bio.Restriction.Restriction.NspBII.results
Bio.Restriction.Restriction.NspI.results
Bio.Restriction.Restriction.NspV.results
Bio.Restriction.Restriction.OliI.results
Bio.Restriction.Restriction.PabI.results
Bio.Restriction.Restriction.PacI.results
Bio.Restriction.Restriction.PaeI.results
Bio.Restriction.Restriction.PaeR7I.results
Bio.Restriction.Restriction.PagI.results
Bio.Restriction.Restriction.PalAI.results
Bio.Restriction.Restriction.PasI.results
Bio.Restriction.Restriction.PauI.results
Bio.Restriction.Restriction.PceI.results
Bio.Restriction.Restriction.PciI.results
Bio.Restriction.Restriction.PciSI.results
Bio.Restriction.Restriction.PcsI.results
Bio.Restriction.Restriction.PctI.results
Bio.Restriction.Restriction.PdiI.results
Bio.Restriction.Restriction.PdmI.results
Bio.Restriction.Restriction.PenI.results
Bio.Restriction.Restriction.PfeI.results
Bio.Restriction.Restriction.Pfl1108I.results
Bio.Restriction.Restriction.Pfl23II.results
Bio.Restriction.Restriction.PflFI.results
Bio.Restriction.Restriction.PflMI.results
Bio.Restriction.Restriction.PfoI.results
Bio.Restriction.Restriction.PinAI.results
Bio.Restriction.Restriction.PlaDI.results
Bio.Restriction.Restriction.Ple19I.results
Bio.Restriction.Restriction.PleI.results
Bio.Restriction.Restriction.PluTI.results
Bio.Restriction.Restriction.PmaCI.results
Bio.Restriction.Restriction.PmeI.results
Bio.Restriction.Restriction.PmlI.results
Bio.Restriction.Restriction.PpiI.results
Bio.Restriction.Restriction.PpsI.results
Bio.Restriction.Restriction.Ppu10I.results
Bio.Restriction.Restriction.Ppu21I.results
Bio.Restriction.Restriction.PpuMI.results
Bio.Restriction.Restriction.PscI.results
Bio.Restriction.Restriction.PshAI.results
Bio.Restriction.Restriction.PshBI.results
Bio.Restriction.Restriction.PsiI.results
Bio.Restriction.Restriction.Psp03I.results
Bio.Restriction.Restriction.Psp124BI.results
Bio.Restriction.Restriction.Psp1406I.results
Bio.Restriction.Restriction.Psp5II.results
Bio.Restriction.Restriction.Psp6I.results
Bio.Restriction.Restriction.PspCI.results
Bio.Restriction.Restriction.PspEI.results
Bio.Restriction.Restriction.PspGI.results
Bio.Restriction.Restriction.PspLI.results
Bio.Restriction.Restriction.PspN4I.results
Bio.Restriction.Restriction.PspOMI.results
Bio.Restriction.Restriction.PspOMII.results
Bio.Restriction.Restriction.PspPI.results
Bio.Restriction.Restriction.PspPPI.results
Bio.Restriction.Restriction.PspPRI.results
Bio.Restriction.Restriction.PspXI.results
Bio.Restriction.Restriction.PsrI.results
Bio.Restriction.Restriction.PssI.results
Bio.Restriction.Restriction.PstI.results
Bio.Restriction.Restriction.PstNI.results
Bio.Restriction.Restriction.PsuI.results
Bio.Restriction.Restriction.PsyI.results
Bio.Restriction.Restriction.PteI.results
Bio.Restriction.Restriction.PvuI.results
Bio.Restriction.Restriction.PvuII.results
Bio.Restriction.Restriction.R2_BceSIV.results
Bio.Restriction.Restriction.RceI.results
Bio.Restriction.Restriction.RdeGBI.results
Bio.Restriction.Restriction.RdeGBII.results
Bio.Restriction.Restriction.RdeGBIII.results
Bio.Restriction.Restriction.RflFIII.results
Bio.Restriction.Restriction.RgaI.results
Bio.Restriction.Restriction.RigI.results
Bio.Restriction.Restriction.RlaI.results
Bio.Restriction.Restriction.RleAI.results
Bio.Restriction.Restriction.RpaB5I.results
Bio.Restriction.Restriction.RpaBI.results
Bio.Restriction.Restriction.RpaI.results
Bio.Restriction.Restriction.RpaTI.results
Bio.Restriction.Restriction.RruI.results
Bio.Restriction.Restriction.RsaI.results
Bio.Restriction.Restriction.RsaNI.results
Bio.Restriction.Restriction.RseI.results
Bio.Restriction.Restriction.Rsr2I.results
Bio.Restriction.Restriction.RsrII.results
Bio.Restriction.Restriction.SacI.results
Bio.Restriction.Restriction.SacII.results
Bio.Restriction.Restriction.SalI.results
Bio.Restriction.Restriction.SanDI.results
Bio.Restriction.Restriction.SapI.results
Bio.Restriction.Restriction.SaqAI.results
Bio.Restriction.Restriction.SatI.results
Bio.Restriction.Restriction.Sau3AI.results
Bio.Restriction.Restriction.Sau96I.results
Bio.Restriction.Restriction.SauI.results
Bio.Restriction.Restriction.SbfI.results
Bio.Restriction.Restriction.ScaI.results
Bio.Restriction.Restriction.SchI.results
Bio.Restriction.Restriction.SciI.results
Bio.Restriction.Restriction.ScrFI.results
Bio.Restriction.Restriction.SdaI.results
Bio.Restriction.Restriction.SdeAI.results
Bio.Restriction.Restriction.SdeOSI.results
Bio.Restriction.Restriction.SduI.results
Bio.Restriction.Restriction.SecI.results
Bio.Restriction.Restriction.SelI.results
Bio.Restriction.Restriction.SetI.results
Bio.Restriction.Restriction.SexAI.results
Bio.Restriction.Restriction.SfaAI.results
Bio.Restriction.Restriction.SfaNI.results
Bio.Restriction.Restriction.SfcI.results
Bio.Restriction.Restriction.SfeI.results
Bio.Restriction.Restriction.SfiI.results
Bio.Restriction.Restriction.SfoI.results
Bio.Restriction.Restriction.Sfr274I.results
Bio.Restriction.Restriction.Sfr303I.results
Bio.Restriction.Restriction.SfuI.results
Bio.Restriction.Restriction.SgeI.results
Bio.Restriction.Restriction.SgfI.results
Bio.Restriction.Restriction.SgrAI.results
Bio.Restriction.Restriction.SgrBI.results
Bio.Restriction.Restriction.SgrDI.results
Bio.Restriction.Restriction.SgrTI.results
Bio.Restriction.Restriction.SgsI.results
Bio.Restriction.Restriction.SimI.results
Bio.Restriction.Restriction.SlaI.results
Bio.Restriction.Restriction.SmaI.results
Bio.Restriction.Restriction.SmiI.results
Bio.Restriction.Restriction.SmiMI.results
Bio.Restriction.Restriction.SmlI.results
Bio.Restriction.Restriction.SmoI.results
Bio.Restriction.Restriction.SnaBI.results
Bio.Restriction.Restriction.SnaI.results
Bio.Restriction.Restriction.Sno506I.results
Bio.Restriction.Restriction.SpeI.results
Bio.Restriction.Restriction.SphI.results
Bio.Restriction.Restriction.SplI.results
Bio.Restriction.Restriction.SpoDI.results
Bio.Restriction.Restriction.SrfI.results
Bio.Restriction.Restriction.Sse232I.results
Bio.Restriction.Restriction.Sse8387I.results
Bio.Restriction.Restriction.Sse8647I.results
Bio.Restriction.Restriction.Sse9I.results
Bio.Restriction.Restriction.SseBI.results
Bio.Restriction.Restriction.SsiI.results
Bio.Restriction.Restriction.SspD5I.results
Bio.Restriction.Restriction.SspDI.results
Bio.Restriction.Restriction.SspI.results
Bio.Restriction.Restriction.SstE37I.results
Bio.Restriction.Restriction.SstI.results
Bio.Restriction.Restriction.Sth132I.results
Bio.Restriction.Restriction.Sth302II.results
Bio.Restriction.Restriction.StrI.results
Bio.Restriction.Restriction.StsI.results
Bio.Restriction.Restriction.StuI.results
Bio.Restriction.Restriction.StyD4I.results
Bio.Restriction.Restriction.StyI.results
Bio.Restriction.Restriction.SwaI.results
Bio.Restriction.Restriction.TaaI.results
Bio.Restriction.Restriction.TaiI.results
Bio.Restriction.Restriction.TaqI.results
Bio.Restriction.Restriction.TaqII.results
Bio.Restriction.Restriction.TasI.results
Bio.Restriction.Restriction.TatI.results
Bio.Restriction.Restriction.TauI.results
Bio.Restriction.Restriction.TfiI.results
Bio.Restriction.Restriction.Tru1I.results
Bio.Restriction.Restriction.Tru9I.results
Bio.Restriction.Restriction.TscAI.results
Bio.Restriction.Restriction.TseFI.results
Bio.Restriction.Restriction.TseI.results
Bio.Restriction.Restriction.TsoI.results
Bio.Restriction.Restriction.Tsp45I.results
Bio.Restriction.Restriction.Tsp4CI.results
Bio.Restriction.Restriction.TspDTI.results
Bio.Restriction.Restriction.TspEI.results
Bio.Restriction.Restriction.TspGWI.results
Bio.Restriction.Restriction.TspMI.results
Bio.Restriction.Restriction.TspRI.results
Bio.Restriction.Restriction.TssI.results
Bio.Restriction.Restriction.TstI.results
Bio.Restriction.Restriction.TsuI.results
Bio.Restriction.Restriction.Tth111I.results
Bio.Restriction.Restriction.Tth111II.results
Bio.Restriction.Restriction.UbaF11I.results
Bio.Restriction.Restriction.UbaF12I.results
Bio.Restriction.Restriction.UbaF13I.results
Bio.Restriction.Restriction.UbaF14I.results
Bio.Restriction.Restriction.UbaF9I.results
Bio.Restriction.Restriction.UbaPI.results
Bio.Restriction.Restriction.UcoMSI.results
Bio.Restriction.Restriction.UnbI.results
Bio.Restriction.Restriction.Van91I.results
Bio.Restriction.Restriction.Vha464I.results
Bio.Restriction.Restriction.VneI.results
Bio.Restriction.Restriction.VpaK11AI.results
Bio.Restriction.Restriction.VpaK11BI.results
Bio.Restriction.Restriction.VspI.results
Bio.Restriction.Restriction.WviI.results
Bio.Restriction.Restriction.XagI.results
Bio.Restriction.Restriction.XapI.results
Bio.Restriction.Restriction.XbaI.results
Bio.Restriction.Restriction.XceI.results
Bio.Restriction.Restriction.XcmI.results
Bio.Restriction.Restriction.XhoI.results
Bio.Restriction.Restriction.XhoII.results
Bio.Restriction.Restriction.XmaI.results
Bio.Restriction.Restriction.XmaIII.results
Bio.Restriction.Restriction.XmaJI.results
Bio.Restriction.Restriction.XmiI.results
Bio.Restriction.Restriction.XmnI.results
Bio.Restriction.Restriction.XspI.results
Bio.Restriction.Restriction.YkrI.results
Bio.Restriction.Restriction.ZraI.results
Bio.Restriction.Restriction.ZrmI.results
Bio.Restriction.Restriction.Zsp2I.results" class="py-name" href="#" onclick="return doclink('link-1015', 'results', 'link-996');">results</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_consumer</tt><tt class="py-op">.</tt><tt id="link-1016" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-1016', 'data', 'link-432');">data</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1810"></a><tt class="py-lineno">1810</tt>  <tt class="py-line"> </tt>
<a name="L1811"></a><tt class="py-lineno">1811</tt>  <tt class="py-line">            <tt class="py-comment"># if we got here we can't figure out the problem</tt> </tt>
<a name="L1812"></a><tt class="py-lineno">1812</tt>  <tt class="py-line">            <tt class="py-comment"># so we should pass along the syntax error we got</tt> </tt>
<a name="L1813"></a><tt class="py-lineno">1813</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> </tt>
<a name="L1814"></a><tt class="py-lineno">1814</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_consumer</tt><tt class="py-op">.</tt><tt id="link-1017" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-1017', 'data', 'link-432');">data</a></tt> </tt>
</div><a name="L1815"></a><tt class="py-lineno">1815</tt>  <tt class="py-line"> </tt>
<a name="BlastErrorParser._diagnose_error"></a><div id="BlastErrorParser._diagnose_error-def"><a name="L1816"></a><tt class="py-lineno">1816</tt> <a class="py-toggle" href="#" id="BlastErrorParser._diagnose_error-toggle" onclick="return toggle('BlastErrorParser._diagnose_error');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Blast.NCBIStandalone.BlastErrorParser-class.html#_diagnose_error">_diagnose_error</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">data_record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BlastErrorParser._diagnose_error-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="BlastErrorParser._diagnose_error-expanded"><a name="L1817"></a><tt class="py-lineno">1817</tt>  <tt class="py-line">        <tt class="py-docstring">"""Attempt to diagnose an error in the passed handle.</tt> </tt>
<a name="L1818"></a><tt class="py-lineno">1818</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1819"></a><tt class="py-lineno">1819</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L1820"></a><tt class="py-lineno">1820</tt>  <tt class="py-line"><tt class="py-docstring">        o handle - The handle potentially containing the error</tt> </tt>
<a name="L1821"></a><tt class="py-lineno">1821</tt>  <tt class="py-line"><tt class="py-docstring">        o data_record - The data record partially created by the consumer.</tt> </tt>
<a name="L1822"></a><tt class="py-lineno">1822</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1823"></a><tt class="py-lineno">1823</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-1018" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-1018', 'readline', 'link-39');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1824"></a><tt class="py-lineno">1824</tt>  <tt class="py-line"> </tt>
<a name="L1825"></a><tt class="py-lineno">1825</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L1826"></a><tt class="py-lineno">1826</tt>  <tt class="py-line">            <tt class="py-comment"># 'Searchingdone' instead of 'Searching......done' seems</tt> </tt>
<a name="L1827"></a><tt class="py-lineno">1827</tt>  <tt class="py-line">            <tt class="py-comment"># to indicate a failure to perform the BLAST due to</tt> </tt>
<a name="L1828"></a><tt class="py-lineno">1828</tt>  <tt class="py-line">            <tt class="py-comment"># low quality sequence</tt> </tt>
<a name="L1829"></a><tt class="py-lineno">1829</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-1019" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-1019', 'startswith', 'link-42');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Searchingdone'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1830"></a><tt class="py-lineno">1830</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt id="link-1020" class="py-name" targets="Class Bio.Blast.NCBIStandalone.LowQualityBlastError=Bio.Blast.NCBIStandalone.LowQualityBlastError-class.html"><a title="Bio.Blast.NCBIStandalone.LowQualityBlastError" class="py-name" href="#" onclick="return doclink('link-1020', 'LowQualityBlastError', 'link-1020');">LowQualityBlastError</a></tt><tt class="py-op">(</tt><tt class="py-string">"Blast failure occurred on query: "</tt><tt class="py-op">,</tt> </tt>
<a name="L1831"></a><tt class="py-lineno">1831</tt>  <tt class="py-line">                                           <tt class="py-name">data_record</tt><tt class="py-op">.</tt><tt id="link-1021" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.query
Bio.SearchIO._model.hsp.HSP.query
Bio.SearchIO._model.hsp.HSPFragment.query
Bio.SearchIO._model.query" class="py-name" href="#" onclick="return doclink('link-1021', 'query', 'link-239');">query</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1832"></a><tt class="py-lineno">1832</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-1022" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-1022', 'readline', 'link-39');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1833"></a><tt class="py-lineno">1833</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Oct 21 18:05:04 2015
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
