<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.Phylo.BaseTree</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.Phylo-module.html">Package&nbsp;Phylo</a> ::
        Module&nbsp;BaseTree
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.Phylo.BaseTree-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.Phylo.BaseTree-module.html">Module Bio.Phylo.BaseTree</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">   1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright (C) 2009 by Eric Talevich (eric.talevich@gmail.com)</tt> </tt>
<a name="L2"></a><tt class="py-lineno">   2</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L3"></a><tt class="py-lineno">   3</tt>  <tt class="py-line"><tt class="py-comment"># license. Please see the LICENSE file that should have been included</tt> </tt>
<a name="L4"></a><tt class="py-lineno">   4</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L5"></a><tt class="py-lineno">   5</tt>  <tt class="py-line"> </tt>
<a name="L6"></a><tt class="py-lineno">   6</tt>  <tt class="py-line"><tt class="py-docstring">"""Base classes for Bio.Phylo objects.</tt> </tt>
<a name="L7"></a><tt class="py-lineno">   7</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L8"></a><tt class="py-lineno">   8</tt>  <tt class="py-line"><tt class="py-docstring">All object representations for phylogenetic trees should derive from these base</tt> </tt>
<a name="L9"></a><tt class="py-lineno">   9</tt>  <tt class="py-line"><tt class="py-docstring">classes in order to use the common methods defined on them.</tt> </tt>
<a name="L10"></a><tt class="py-lineno">  10</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L11"></a><tt class="py-lineno">  11</tt>  <tt class="py-line"> </tt>
<a name="L12"></a><tt class="py-lineno">  12</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-1', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">basestring</tt><tt class="py-op">,</tt> <tt id="link-2" class="py-name" targets="Function Bio.FSSP.FSSPTools.filter()=Bio.FSSP.FSSPTools-module.html#filter,Method Bio.SearchIO._model.hit.Hit.filter()=Bio.SearchIO._model.hit.Hit-class.html#filter"><a title="Bio.FSSP.FSSPTools.filter
Bio.SearchIO._model.hit.Hit.filter" class="py-name" href="#" onclick="return doclink('link-2', 'filter', 'link-2');">filter</a></tt><tt class="py-op">,</tt> <tt class="py-name">unicode</tt><tt class="py-op">,</tt> <tt class="py-name">zip</tt> </tt>
<a name="L13"></a><tt class="py-lineno">  13</tt>  <tt class="py-line"> </tt>
<a name="L14"></a><tt class="py-lineno">  14</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">collections</tt> </tt>
<a name="L15"></a><tt class="py-lineno">  15</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt id="link-3" class="py-name" targets="Method Bio.Crystal.Crystal.copy()=Bio.Crystal.Crystal-class.html#copy,Method Bio.File._IndexedSeqFileDict.copy()=Bio.File._IndexedSeqFileDict-class.html#copy,Method Bio.PDB.Atom.Atom.copy()=Bio.PDB.Atom.Atom-class.html#copy,Method Bio.PDB.Entity.Entity.copy()=Bio.PDB.Entity.Entity-class.html#copy,Method Bio.PDB.vectors.Vector.copy()=Bio.PDB.vectors.Vector-class.html#copy"><a title="Bio.Crystal.Crystal.copy
Bio.File._IndexedSeqFileDict.copy
Bio.PDB.Atom.Atom.copy
Bio.PDB.Entity.Entity.copy
Bio.PDB.vectors.Vector.copy" class="py-name" href="#" onclick="return doclink('link-3', 'copy', 'link-3');">copy</a></tt> </tt>
<a name="L16"></a><tt class="py-lineno">  16</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">itertools</tt> </tt>
<a name="L17"></a><tt class="py-lineno">  17</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">random</tt> </tt>
<a name="L18"></a><tt class="py-lineno">  18</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">re</tt> </tt>
<a name="L19"></a><tt class="py-lineno">  19</tt>  <tt class="py-line"> </tt>
<a name="L20"></a><tt class="py-lineno">  20</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-4" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-4', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-5" class="py-name" targets="Module Bio.Phylo._utils=Bio.Phylo._utils-module.html,Module Bio.SearchIO._utils=Bio.SearchIO._utils-module.html,Module Bio._utils=Bio._utils-module.html"><a title="Bio.Phylo._utils
Bio.SearchIO._utils
Bio._utils" class="py-name" href="#" onclick="return doclink('link-5', '_utils', 'link-5');">_utils</a></tt> </tt>
<a name="L21"></a><tt class="py-lineno">  21</tt>  <tt class="py-line"> </tt>
<a name="L22"></a><tt class="py-lineno">  22</tt>  <tt class="py-line"><tt class="py-comment"># NB: On Python 2, repr() and str() are specified to return byte strings, not</tt> </tt>
<a name="L23"></a><tt class="py-lineno">  23</tt>  <tt class="py-line"><tt class="py-comment"># unicode. On Python 3, it's the opposite. Horrible.</tt> </tt>
<a name="L24"></a><tt class="py-lineno">  24</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">sys</tt> </tt>
<a name="L25"></a><tt class="py-lineno">  25</tt>  <tt class="py-line"><tt class="py-keyword">if</tt> <tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">version_info</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">3</tt><tt class="py-op">:</tt> </tt>
<a name="as_string"></a><div id="as_string-def"><a name="L26"></a><tt class="py-lineno">  26</tt> <a class="py-toggle" href="#" id="as_string-toggle" onclick="return toggle('as_string');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree-module.html#as_string">as_string</a><tt class="py-op">(</tt><tt class="py-param">s</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="as_string-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="as_string-expanded"><a name="L27"></a><tt class="py-lineno">  27</tt>  <tt class="py-line">        <tt class="py-docstring">"""Encode string to UTF-8."""</tt> </tt>
<a name="L28"></a><tt class="py-lineno">  28</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">,</tt> <tt class="py-name">unicode</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L29"></a><tt class="py-lineno">  29</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">s</tt><tt class="py-op">.</tt><tt class="py-name">encode</tt><tt class="py-op">(</tt><tt class="py-string">'utf-8'</tt><tt class="py-op">)</tt> </tt>
<a name="L30"></a><tt class="py-lineno">  30</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">)</tt> </tt>
</div><a name="L31"></a><tt class="py-lineno">  31</tt>  <tt class="py-line"><tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L32"></a><tt class="py-lineno">  32</tt>  <tt class="py-line">    <tt id="link-6" class="py-name" targets="Function Bio.Phylo.BaseTree.as_string()=Bio.Phylo.BaseTree-module.html#as_string,Method Bio.Restriction.Restriction.RestrictionBatch.as_string()=Bio.Restriction.Restriction.RestrictionBatch-class.html#as_string"><a title="Bio.Phylo.BaseTree.as_string
Bio.Restriction.Restriction.RestrictionBatch.as_string" class="py-name" href="#" onclick="return doclink('link-6', 'as_string', 'link-6');">as_string</a></tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt> </tt>
<a name="_level_traverse"></a><div id="_level_traverse-def"><a name="L33"></a><tt class="py-lineno">  33</tt>  <tt class="py-line"> </tt>
<a name="L34"></a><tt class="py-lineno">  34</tt>  <tt class="py-line"> </tt>
<a name="L35"></a><tt class="py-lineno">  35</tt>  <tt class="py-line"><tt class="py-comment"># General tree-traversal algorithms</tt> </tt>
<a name="L36"></a><tt class="py-lineno">  36</tt>  <tt class="py-line"> </tt>
<a name="L37"></a><tt class="py-lineno">  37</tt> <a class="py-toggle" href="#" id="_level_traverse-toggle" onclick="return toggle('_level_traverse');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree-module.html#_level_traverse">_level_traverse</a><tt class="py-op">(</tt><tt class="py-param">root</tt><tt class="py-op">,</tt> <tt class="py-param">get_children</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_level_traverse-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_level_traverse-expanded"><a name="L38"></a><tt class="py-lineno">  38</tt>  <tt class="py-line">    <tt class="py-docstring">"""Traverse a tree in breadth-first (level) order (PRIVATE)."""</tt> </tt>
<a name="L39"></a><tt class="py-lineno">  39</tt>  <tt class="py-line">    <tt class="py-name">Q</tt> <tt class="py-op">=</tt> <tt class="py-name">collections</tt><tt class="py-op">.</tt><tt class="py-name">deque</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt id="link-7" class="py-name" targets="Variable Bio.Phylo.BaseTree.Clade.root=Bio.Phylo.BaseTree.Clade-class.html#root,Variable Bio.Phylo._cdao_owl.root=Bio.Phylo._cdao_owl-module.html#root"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-7', 'root', 'link-7');">root</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L40"></a><tt class="py-lineno">  40</tt>  <tt class="py-line">    <tt class="py-keyword">while</tt> <tt class="py-name">Q</tt><tt class="py-op">:</tt> </tt>
<a name="L41"></a><tt class="py-lineno">  41</tt>  <tt class="py-line">        <tt class="py-name">v</tt> <tt class="py-op">=</tt> <tt class="py-name">Q</tt><tt class="py-op">.</tt><tt class="py-name">popleft</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L42"></a><tt class="py-lineno">  42</tt>  <tt class="py-line">        <tt class="py-keyword">yield</tt> <tt class="py-name">v</tt> </tt>
<a name="L43"></a><tt class="py-lineno">  43</tt>  <tt class="py-line">        <tt class="py-name">Q</tt><tt class="py-op">.</tt><tt id="link-8" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.extend()=Bio.Align.MultipleSeqAlignment-class.html#extend,Method Bio.SCOP.Raf.SeqMap.extend()=Bio.SCOP.Raf.SeqMap-class.html#extend,Method Bio.Seq.MutableSeq.extend()=Bio.Seq.MutableSeq-class.html#extend"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-8', 'extend', 'link-8');">extend</a></tt><tt class="py-op">(</tt><tt class="py-name">get_children</tt><tt class="py-op">(</tt><tt class="py-name">v</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L44"></a><tt class="py-lineno">  44</tt>  <tt class="py-line"> </tt>
<a name="_preorder_traverse"></a><div id="_preorder_traverse-def"><a name="L45"></a><tt class="py-lineno">  45</tt>  <tt class="py-line"> </tt>
<a name="L46"></a><tt class="py-lineno">  46</tt> <a class="py-toggle" href="#" id="_preorder_traverse-toggle" onclick="return toggle('_preorder_traverse');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree-module.html#_preorder_traverse">_preorder_traverse</a><tt class="py-op">(</tt><tt class="py-param">root</tt><tt class="py-op">,</tt> <tt class="py-param">get_children</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_preorder_traverse-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_preorder_traverse-expanded"><a name="L47"></a><tt class="py-lineno">  47</tt>  <tt class="py-line">    <tt class="py-docstring">"""Traverse a tree in depth-first pre-order (parent before children) (PRIVATE)."""</tt> </tt>
<a name="L48"></a><tt class="py-lineno">  48</tt>  <tt class="py-line">    <tt class="py-keyword">def</tt> <tt class="py-def-name">dfs</tt><tt class="py-op">(</tt><tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L49"></a><tt class="py-lineno">  49</tt>  <tt class="py-line">        <tt class="py-keyword">yield</tt> <tt class="py-name">elem</tt> </tt>
<a name="L50"></a><tt class="py-lineno">  50</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">v</tt> <tt class="py-keyword">in</tt> <tt class="py-name">get_children</tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L51"></a><tt class="py-lineno">  51</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">u</tt> <tt class="py-keyword">in</tt> <tt class="py-name">dfs</tt><tt class="py-op">(</tt><tt class="py-name">v</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L52"></a><tt class="py-lineno">  52</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt class="py-name">u</tt> </tt>
</div><a name="L53"></a><tt class="py-lineno">  53</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">elem</tt> <tt class="py-keyword">in</tt> <tt class="py-name">dfs</tt><tt class="py-op">(</tt><tt id="link-9" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-9', 'root', 'link-7');">root</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L54"></a><tt class="py-lineno">  54</tt>  <tt class="py-line">        <tt class="py-keyword">yield</tt> <tt class="py-name">elem</tt> </tt>
</div><a name="L55"></a><tt class="py-lineno">  55</tt>  <tt class="py-line"> </tt>
<a name="_postorder_traverse"></a><div id="_postorder_traverse-def"><a name="L56"></a><tt class="py-lineno">  56</tt>  <tt class="py-line"> </tt>
<a name="L57"></a><tt class="py-lineno">  57</tt> <a class="py-toggle" href="#" id="_postorder_traverse-toggle" onclick="return toggle('_postorder_traverse');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree-module.html#_postorder_traverse">_postorder_traverse</a><tt class="py-op">(</tt><tt class="py-param">root</tt><tt class="py-op">,</tt> <tt class="py-param">get_children</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_postorder_traverse-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_postorder_traverse-expanded"><a name="L58"></a><tt class="py-lineno">  58</tt>  <tt class="py-line">    <tt class="py-docstring">"""Traverse a tree in depth-first post-order (children before parent) (PRIVATE)."""</tt> </tt>
<a name="L59"></a><tt class="py-lineno">  59</tt>  <tt class="py-line">    <tt class="py-keyword">def</tt> <tt class="py-def-name">dfs</tt><tt class="py-op">(</tt><tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L60"></a><tt class="py-lineno">  60</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">v</tt> <tt class="py-keyword">in</tt> <tt class="py-name">get_children</tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L61"></a><tt class="py-lineno">  61</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">u</tt> <tt class="py-keyword">in</tt> <tt class="py-name">dfs</tt><tt class="py-op">(</tt><tt class="py-name">v</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L62"></a><tt class="py-lineno">  62</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt class="py-name">u</tt> </tt>
<a name="L63"></a><tt class="py-lineno">  63</tt>  <tt class="py-line">        <tt class="py-keyword">yield</tt> <tt class="py-name">elem</tt> </tt>
</div><a name="L64"></a><tt class="py-lineno">  64</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">elem</tt> <tt class="py-keyword">in</tt> <tt class="py-name">dfs</tt><tt class="py-op">(</tt><tt id="link-10" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-10', 'root', 'link-7');">root</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L65"></a><tt class="py-lineno">  65</tt>  <tt class="py-line">        <tt class="py-keyword">yield</tt> <tt class="py-name">elem</tt> </tt>
</div><a name="L66"></a><tt class="py-lineno">  66</tt>  <tt class="py-line"> </tt>
<a name="_sorted_attrs"></a><div id="_sorted_attrs-def"><a name="L67"></a><tt class="py-lineno">  67</tt>  <tt class="py-line"> </tt>
<a name="L68"></a><tt class="py-lineno">  68</tt> <a class="py-toggle" href="#" id="_sorted_attrs-toggle" onclick="return toggle('_sorted_attrs');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree-module.html#_sorted_attrs">_sorted_attrs</a><tt class="py-op">(</tt><tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_sorted_attrs-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_sorted_attrs-expanded"><a name="L69"></a><tt class="py-lineno">  69</tt>  <tt class="py-line">    <tt class="py-docstring">"""Get a flat list of elem's attributes, sorted for consistency (PRIVATE)."""</tt> </tt>
<a name="L70"></a><tt class="py-lineno">  70</tt>  <tt class="py-line">    <tt class="py-name">singles</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L71"></a><tt class="py-lineno">  71</tt>  <tt class="py-line">    <tt class="py-name">lists</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L72"></a><tt class="py-lineno">  72</tt>  <tt class="py-line">    <tt class="py-comment"># Sort attributes for consistent results</tt> </tt>
<a name="L73"></a><tt class="py-lineno">  73</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">attrname</tt><tt class="py-op">,</tt> <tt class="py-name">child</tt> <tt class="py-keyword">in</tt> <tt class="py-name">sorted</tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">__dict__</tt><tt class="py-op">.</tt><tt id="link-11" class="py-name" targets="Method Bio.Crystal.Crystal.items()=Bio.Crystal.Crystal-class.html#items,Method Bio.File._IndexedSeqFileDict.items()=Bio.File._IndexedSeqFileDict-class.html#items,Method Bio.Phylo.PhyloXML.Events.items()=Bio.Phylo.PhyloXML.Events-class.html#items,Variable Bio.SearchIO._model.query.QueryResult.items=Bio.SearchIO._model.query.QueryResult-class.html#items,Method BioSQL.BioSeqDatabase.BioSeqDatabase.items()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#items,Method BioSQL.BioSeqDatabase.DBServer.items()=BioSQL.BioSeqDatabase.DBServer-class.html#items"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-11', 'items', 'link-11');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L74"></a><tt class="py-lineno">  74</tt>  <tt class="py-line">                                  <tt class="py-name">key</tt><tt class="py-op">=</tt><tt class="py-keyword">lambda</tt> <tt class="py-name">kv</tt><tt class="py-op">:</tt> <tt class="py-name">kv</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L75"></a><tt class="py-lineno">  75</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">child</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L76"></a><tt class="py-lineno">  76</tt>  <tt class="py-line">            <tt class="py-keyword">continue</tt> </tt>
<a name="L77"></a><tt class="py-lineno">  77</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">child</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L78"></a><tt class="py-lineno">  78</tt>  <tt class="py-line">            <tt class="py-name">lists</tt><tt class="py-op">.</tt><tt id="link-12" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-12', 'extend', 'link-8');">extend</a></tt><tt class="py-op">(</tt><tt class="py-name">child</tt><tt class="py-op">)</tt> </tt>
<a name="L79"></a><tt class="py-lineno">  79</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L80"></a><tt class="py-lineno">  80</tt>  <tt class="py-line">            <tt class="py-name">singles</tt><tt class="py-op">.</tt><tt id="link-13" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-13', 'append', 'link-13');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">child</tt><tt class="py-op">)</tt> </tt>
<a name="L81"></a><tt class="py-lineno">  81</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt id="link-14" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.x=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#x,Variable Bio.Statistics.lowess.x=Bio.Statistics.lowess-module.html#x"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-14', 'x', 'link-14');">x</a></tt> <tt class="py-keyword">for</tt> <tt id="link-15" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-15', 'x', 'link-14');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">singles</tt> <tt class="py-op">+</tt> <tt class="py-name">lists</tt> </tt>
<a name="L82"></a><tt class="py-lineno">  82</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-16" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-16', 'x', 'link-14');">x</a></tt><tt class="py-op">,</tt> <tt id="link-17" class="py-name" targets="Class Bio.Phylo.BaseTree.TreeElement=Bio.Phylo.BaseTree.TreeElement-class.html"><a title="Bio.Phylo.BaseTree.TreeElement" class="py-name" href="#" onclick="return doclink('link-17', 'TreeElement', 'link-17');">TreeElement</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L83"></a><tt class="py-lineno">  83</tt>  <tt class="py-line"> </tt>
<a name="_identity_matcher"></a><div id="_identity_matcher-def"><a name="L84"></a><tt class="py-lineno">  84</tt>  <tt class="py-line"> </tt>
<a name="L85"></a><tt class="py-lineno">  85</tt>  <tt class="py-line"><tt class="py-comment"># Factory functions to generalize searching for clades/nodes</tt> </tt>
<a name="L86"></a><tt class="py-lineno">  86</tt>  <tt class="py-line"> </tt>
<a name="L87"></a><tt class="py-lineno">  87</tt> <a class="py-toggle" href="#" id="_identity_matcher-toggle" onclick="return toggle('_identity_matcher');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree-module.html#_identity_matcher">_identity_matcher</a><tt class="py-op">(</tt><tt class="py-param">target</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_identity_matcher-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_identity_matcher-expanded"><a name="L88"></a><tt class="py-lineno">  88</tt>  <tt class="py-line">    <tt class="py-docstring">"""Match a node to the target object by identity (PRIVATE)."""</tt> </tt>
<a name="L89"></a><tt class="py-lineno">  89</tt>  <tt class="py-line">    <tt class="py-keyword">def</tt> <tt class="py-def-name">match</tt><tt class="py-op">(</tt><tt class="py-param">node</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L90"></a><tt class="py-lineno">  90</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt id="link-18" class="py-name" targets="Method Bio.Nexus.Trees.Tree.node()=Bio.Nexus.Trees.Tree-class.html#node"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-18', 'node', 'link-18');">node</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">target</tt><tt class="py-op">)</tt> </tt>
</div><a name="L91"></a><tt class="py-lineno">  91</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-19" class="py-name" targets="Function Bio.triefind.match()=Bio.triefind-module.html#match"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-19', 'match', 'link-19');">match</a></tt> </tt>
</div><a name="L92"></a><tt class="py-lineno">  92</tt>  <tt class="py-line"> </tt>
<a name="_class_matcher"></a><div id="_class_matcher-def"><a name="L93"></a><tt class="py-lineno">  93</tt>  <tt class="py-line"> </tt>
<a name="L94"></a><tt class="py-lineno">  94</tt> <a class="py-toggle" href="#" id="_class_matcher-toggle" onclick="return toggle('_class_matcher');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree-module.html#_class_matcher">_class_matcher</a><tt class="py-op">(</tt><tt class="py-param">target_cls</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_class_matcher-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_class_matcher-expanded"><a name="L95"></a><tt class="py-lineno">  95</tt>  <tt class="py-line">    <tt class="py-docstring">"""Match a node if it's an instance of the given class (PRIVATE)."""</tt> </tt>
<a name="L96"></a><tt class="py-lineno">  96</tt>  <tt class="py-line">    <tt class="py-keyword">def</tt> <tt class="py-def-name">match</tt><tt class="py-op">(</tt><tt class="py-param">node</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L97"></a><tt class="py-lineno">  97</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-20" class="py-name"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-20', 'node', 'link-18');">node</a></tt><tt class="py-op">,</tt> <tt class="py-name">target_cls</tt><tt class="py-op">)</tt> </tt>
</div><a name="L98"></a><tt class="py-lineno">  98</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-21" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-21', 'match', 'link-19');">match</a></tt> </tt>
</div><a name="L99"></a><tt class="py-lineno">  99</tt>  <tt class="py-line"> </tt>
<a name="_string_matcher"></a><div id="_string_matcher-def"><a name="L100"></a><tt class="py-lineno"> 100</tt>  <tt class="py-line"> </tt>
<a name="L101"></a><tt class="py-lineno"> 101</tt> <a class="py-toggle" href="#" id="_string_matcher-toggle" onclick="return toggle('_string_matcher');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree-module.html#_string_matcher">_string_matcher</a><tt class="py-op">(</tt><tt class="py-param">target</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_string_matcher-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_string_matcher-expanded"><a name="L102"></a><tt class="py-lineno"> 102</tt>  <tt class="py-line">    <tt class="py-keyword">def</tt> <tt class="py-def-name">match</tt><tt class="py-op">(</tt><tt class="py-param">node</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L103"></a><tt class="py-lineno"> 103</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-22" class="py-name"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-22', 'node', 'link-18');">node</a></tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt id="link-23" class="py-name" targets="Class Bio.Phylo.BaseTree.Clade=Bio.Phylo.BaseTree.Clade-class.html,Class Bio.Phylo.CDAO.Clade=Bio.Phylo.CDAO.Clade-class.html,Class Bio.Phylo.NeXML.Clade=Bio.Phylo.NeXML.Clade-class.html,Class Bio.Phylo.Newick.Clade=Bio.Phylo.Newick.Clade-class.html,Class Bio.Phylo.PhyloXML.Clade=Bio.Phylo.PhyloXML.Clade-class.html"><a title="Bio.Phylo.BaseTree.Clade
Bio.Phylo.CDAO.Clade
Bio.Phylo.NeXML.Clade
Bio.Phylo.Newick.Clade
Bio.Phylo.PhyloXML.Clade" class="py-name" href="#" onclick="return doclink('link-23', 'Clade', 'link-23');">Clade</a></tt><tt class="py-op">,</tt> <tt id="link-24" class="py-name" targets="Class Bio.Cluster.Tree=Bio.Cluster.Tree-class.html,Class Bio.Nexus.Trees.Tree=Bio.Nexus.Trees.Tree-class.html,Class Bio.Phylo.BaseTree.Tree=Bio.Phylo.BaseTree.Tree-class.html,Class Bio.Phylo.CDAO.Tree=Bio.Phylo.CDAO.Tree-class.html,Class Bio.Phylo.NeXML.Tree=Bio.Phylo.NeXML.Tree-class.html,Class Bio.Phylo.Newick.Tree=Bio.Phylo.Newick.Tree-class.html"><a title="Bio.Cluster.Tree
Bio.Nexus.Trees.Tree
Bio.Phylo.BaseTree.Tree
Bio.Phylo.CDAO.Tree
Bio.Phylo.NeXML.Tree
Bio.Phylo.Newick.Tree" class="py-name" href="#" onclick="return doclink('link-24', 'Tree', 'link-24');">Tree</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L104"></a><tt class="py-lineno"> 104</tt>  <tt class="py-line">            <tt class="py-comment"># Avoid triggering specialized or recursive magic methods</tt> </tt>
<a name="L105"></a><tt class="py-lineno"> 105</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-25" class="py-name"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-25', 'node', 'link-18');">node</a></tt><tt class="py-op">.</tt><tt id="link-26" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.name=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.name=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.name=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#name,Method Bio.Phylo.PhyloXMLIO.Writer.name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#name,Variable Bio.codonalign.codonalphabet.CodonAlphabet.name=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-26', 'name', 'link-26');">name</a></tt> <tt class="py-op">==</tt> <tt class="py-name">target</tt> </tt>
<a name="L106"></a><tt class="py-lineno"> 106</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-27" class="py-name"><a title="Bio.Phylo.BaseTree.as_string
Bio.Restriction.Restriction.RestrictionBatch.as_string" class="py-name" href="#" onclick="return doclink('link-27', 'as_string', 'link-6');">as_string</a></tt><tt class="py-op">(</tt><tt id="link-28" class="py-name"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-28', 'node', 'link-18');">node</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">target</tt> </tt>
</div><a name="L107"></a><tt class="py-lineno"> 107</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-29" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-29', 'match', 'link-19');">match</a></tt> </tt>
</div><a name="L108"></a><tt class="py-lineno"> 108</tt>  <tt class="py-line"> </tt>
<a name="_attribute_matcher"></a><div id="_attribute_matcher-def"><a name="L109"></a><tt class="py-lineno"> 109</tt>  <tt class="py-line"> </tt>
<a name="L110"></a><tt class="py-lineno"> 110</tt> <a class="py-toggle" href="#" id="_attribute_matcher-toggle" onclick="return toggle('_attribute_matcher');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree-module.html#_attribute_matcher">_attribute_matcher</a><tt class="py-op">(</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_attribute_matcher-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_attribute_matcher-expanded"><a name="L111"></a><tt class="py-lineno"> 111</tt>  <tt class="py-line">    <tt class="py-docstring">"""Match a node by specified attribute values (PRIVATE).</tt> </tt>
<a name="L112"></a><tt class="py-lineno"> 112</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L113"></a><tt class="py-lineno"> 113</tt>  <tt class="py-line"><tt class="py-docstring">    ``terminal`` is a special case: True restricts the search to external (leaf)</tt> </tt>
<a name="L114"></a><tt class="py-lineno"> 114</tt>  <tt class="py-line"><tt class="py-docstring">    nodes, False restricts to internal nodes, and None allows all tree elements</tt> </tt>
<a name="L115"></a><tt class="py-lineno"> 115</tt>  <tt class="py-line"><tt class="py-docstring">    to be searched, including phyloXML annotations.</tt> </tt>
<a name="L116"></a><tt class="py-lineno"> 116</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L117"></a><tt class="py-lineno"> 117</tt>  <tt class="py-line"><tt class="py-docstring">    Otherwise, for a tree element to match the specification (i.e. for the</tt> </tt>
<a name="L118"></a><tt class="py-lineno"> 118</tt>  <tt class="py-line"><tt class="py-docstring">    function produced by `_attribute_matcher` to return True when given a tree</tt> </tt>
<a name="L119"></a><tt class="py-lineno"> 119</tt>  <tt class="py-line"><tt class="py-docstring">    element), it must have each of the attributes specified by the keys and</tt> </tt>
<a name="L120"></a><tt class="py-lineno"> 120</tt>  <tt class="py-line"><tt class="py-docstring">    match each of the corresponding values -- think 'and', not 'or', for</tt> </tt>
<a name="L121"></a><tt class="py-lineno"> 121</tt>  <tt class="py-line"><tt class="py-docstring">    multiple keys.</tt> </tt>
<a name="L122"></a><tt class="py-lineno"> 122</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L123"></a><tt class="py-lineno"> 123</tt>  <tt class="py-line">    <tt class="py-keyword">def</tt> <tt class="py-def-name">match</tt><tt class="py-op">(</tt><tt class="py-param">node</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L124"></a><tt class="py-lineno"> 124</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">'terminal'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">kwargs</tt><tt class="py-op">:</tt> </tt>
<a name="L125"></a><tt class="py-lineno"> 125</tt>  <tt class="py-line">            <tt class="py-comment"># Special case: restrict to internal/external/any nodes</tt> </tt>
<a name="L126"></a><tt class="py-lineno"> 126</tt>  <tt class="py-line">            <tt class="py-name">kwa_copy</tt> <tt class="py-op">=</tt> <tt class="py-name">kwargs</tt><tt class="py-op">.</tt><tt id="link-30" class="py-name"><a title="Bio.Crystal.Crystal.copy
Bio.File._IndexedSeqFileDict.copy
Bio.PDB.Atom.Atom.copy
Bio.PDB.Entity.Entity.copy
Bio.PDB.vectors.Vector.copy" class="py-name" href="#" onclick="return doclink('link-30', 'copy', 'link-3');">copy</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L127"></a><tt class="py-lineno"> 127</tt>  <tt class="py-line">            <tt class="py-name">pattern</tt> <tt class="py-op">=</tt> <tt class="py-name">kwa_copy</tt><tt class="py-op">.</tt><tt id="link-31" class="py-name" targets="Method Bio.File._IndexedSeqFileDict.pop()=Bio.File._IndexedSeqFileDict-class.html#pop,Method Bio.SearchIO._model.hit.Hit.pop()=Bio.SearchIO._model.hit.Hit-class.html#pop,Method Bio.SearchIO._model.query.QueryResult.pop()=Bio.SearchIO._model.query.QueryResult-class.html#pop,Method Bio.Seq.MutableSeq.pop()=Bio.Seq.MutableSeq-class.html#pop"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop" class="py-name" href="#" onclick="return doclink('link-31', 'pop', 'link-31');">pop</a></tt><tt class="py-op">(</tt><tt class="py-string">'terminal'</tt><tt class="py-op">)</tt> </tt>
<a name="L128"></a><tt class="py-lineno"> 128</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-op">(</tt><tt class="py-name">pattern</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> </tt>
<a name="L129"></a><tt class="py-lineno"> 129</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-keyword">not</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-32" class="py-name"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-32', 'node', 'link-18');">node</a></tt><tt class="py-op">,</tt> <tt class="py-string">'is_terminal'</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> </tt>
<a name="L130"></a><tt class="py-lineno"> 130</tt>  <tt class="py-line">                    <tt id="link-33" class="py-name"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-33', 'node', 'link-18');">node</a></tt><tt class="py-op">.</tt><tt id="link-34" class="py-name" targets="Method Bio.Nexus.Trees.Tree.is_terminal()=Bio.Nexus.Trees.Tree-class.html#is_terminal,Method Bio.Phylo.BaseTree.Clade.is_terminal()=Bio.Phylo.BaseTree.Clade-class.html#is_terminal,Method Bio.Phylo.BaseTree.Tree.is_terminal()=Bio.Phylo.BaseTree.Tree-class.html#is_terminal"><a title="Bio.Nexus.Trees.Tree.is_terminal
Bio.Phylo.BaseTree.Clade.is_terminal
Bio.Phylo.BaseTree.Tree.is_terminal" class="py-name" href="#" onclick="return doclink('link-34', 'is_terminal', 'link-34');">is_terminal</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">pattern</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L131"></a><tt class="py-lineno"> 131</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">False</tt> </tt>
<a name="L132"></a><tt class="py-lineno"> 132</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L133"></a><tt class="py-lineno"> 133</tt>  <tt class="py-line">            <tt class="py-name">kwa_copy</tt> <tt class="py-op">=</tt> <tt class="py-name">kwargs</tt> </tt>
<a name="L134"></a><tt class="py-lineno"> 134</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">pattern</tt> <tt class="py-keyword">in</tt> <tt class="py-name">kwa_copy</tt><tt class="py-op">.</tt><tt id="link-35" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-35', 'items', 'link-11');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L135"></a><tt class="py-lineno"> 135</tt>  <tt class="py-line">            <tt class="py-comment"># Nodes must match all other specified attributes</tt> </tt>
<a name="L136"></a><tt class="py-lineno"> 136</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt id="link-36" class="py-name"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-36', 'node', 'link-18');">node</a></tt><tt class="py-op">,</tt> <tt class="py-name">key</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L137"></a><tt class="py-lineno"> 137</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">False</tt> </tt>
<a name="L138"></a><tt class="py-lineno"> 138</tt>  <tt class="py-line">            <tt class="py-name">target</tt> <tt class="py-op">=</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt id="link-37" class="py-name"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-37', 'node', 'link-18');">node</a></tt><tt class="py-op">,</tt> <tt class="py-name">key</tt><tt class="py-op">)</tt> </tt>
<a name="L139"></a><tt class="py-lineno"> 139</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">pattern</tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L140"></a><tt class="py-lineno"> 140</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">target</tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> </tt>
<a name="L141"></a><tt class="py-lineno"> 141</tt>  <tt class="py-line">                        <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-38" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-38', 'match', 'link-19');">match</a></tt><tt class="py-op">(</tt><tt class="py-name">pattern</tt> <tt class="py-op">+</tt> <tt class="py-string">'$'</tt><tt class="py-op">,</tt> <tt class="py-name">target</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L142"></a><tt class="py-lineno"> 142</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">pattern</tt><tt class="py-op">,</tt> <tt class="py-name">bool</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L143"></a><tt class="py-lineno"> 143</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-name">pattern</tt> <tt class="py-op">==</tt> <tt class="py-name">bool</tt><tt class="py-op">(</tt><tt class="py-name">target</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L144"></a><tt class="py-lineno"> 144</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">pattern</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L145"></a><tt class="py-lineno"> 145</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-name">pattern</tt> <tt class="py-op">==</tt> <tt class="py-name">target</tt><tt class="py-op">)</tt> </tt>
<a name="L146"></a><tt class="py-lineno"> 146</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">pattern</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L147"></a><tt class="py-lineno"> 147</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-name">target</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
<a name="L148"></a><tt class="py-lineno"> 148</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">'invalid query type: %s'</tt> <tt class="py-op">%</tt> <tt id="link-39" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.type()=Bio.Phylo.PhyloXMLIO.Writer-class.html#type"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-39', 'type', 'link-39');">type</a></tt><tt class="py-op">(</tt><tt class="py-name">pattern</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L149"></a><tt class="py-lineno"> 149</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">True</tt> </tt>
</div><a name="L150"></a><tt class="py-lineno"> 150</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-40" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-40', 'match', 'link-19');">match</a></tt> </tt>
</div><a name="L151"></a><tt class="py-lineno"> 151</tt>  <tt class="py-line"> </tt>
<a name="_function_matcher"></a><div id="_function_matcher-def"><a name="L152"></a><tt class="py-lineno"> 152</tt>  <tt class="py-line"> </tt>
<a name="L153"></a><tt class="py-lineno"> 153</tt> <a class="py-toggle" href="#" id="_function_matcher-toggle" onclick="return toggle('_function_matcher');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree-module.html#_function_matcher">_function_matcher</a><tt class="py-op">(</tt><tt class="py-param">matcher_func</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_function_matcher-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_function_matcher-expanded"><a name="L154"></a><tt class="py-lineno"> 154</tt>  <tt class="py-line">    <tt class="py-docstring">"""Safer attribute lookup -- returns False instead of raising an error (PRIVATE)."""</tt> </tt>
<a name="L155"></a><tt class="py-lineno"> 155</tt>  <tt class="py-line">    <tt class="py-keyword">def</tt> <tt class="py-def-name">match</tt><tt class="py-op">(</tt><tt class="py-param">node</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L156"></a><tt class="py-lineno"> 156</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L157"></a><tt class="py-lineno"> 157</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">matcher_func</tt><tt class="py-op">(</tt><tt id="link-41" class="py-name"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-41', 'node', 'link-18');">node</a></tt><tt class="py-op">)</tt> </tt>
<a name="L158"></a><tt class="py-lineno"> 158</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-op">(</tt><tt class="py-name">LookupError</tt><tt class="py-op">,</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">,</tt> <tt class="py-name">ValueError</tt><tt class="py-op">,</tt> <tt class="py-name">TypeError</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L159"></a><tt class="py-lineno"> 159</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">False</tt> </tt>
</div><a name="L160"></a><tt class="py-lineno"> 160</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-42" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-42', 'match', 'link-19');">match</a></tt> </tt>
</div><a name="L161"></a><tt class="py-lineno"> 161</tt>  <tt class="py-line"> </tt>
<a name="_object_matcher"></a><div id="_object_matcher-def"><a name="L162"></a><tt class="py-lineno"> 162</tt>  <tt class="py-line"> </tt>
<a name="L163"></a><tt class="py-lineno"> 163</tt> <a class="py-toggle" href="#" id="_object_matcher-toggle" onclick="return toggle('_object_matcher');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree-module.html#_object_matcher">_object_matcher</a><tt class="py-op">(</tt><tt class="py-param">obj</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_object_matcher-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_object_matcher-expanded"><a name="L164"></a><tt class="py-lineno"> 164</tt>  <tt class="py-line">    <tt class="py-docstring">"""Retrieve a matcher function by passing an arbitrary object (PRIVATE).</tt> </tt>
<a name="L165"></a><tt class="py-lineno"> 165</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L166"></a><tt class="py-lineno"> 166</tt>  <tt class="py-line"><tt class="py-docstring">    i.e. passing a `TreeElement` such as a `Clade` or `Tree` instance returns an</tt> </tt>
<a name="L167"></a><tt class="py-lineno"> 167</tt>  <tt class="py-line"><tt class="py-docstring">    identity matcher, passing a type such as the `PhyloXML.Taxonomy` class</tt> </tt>
<a name="L168"></a><tt class="py-lineno"> 168</tt>  <tt class="py-line"><tt class="py-docstring">    returns a class matcher, and passing a dictionary returns an attribute</tt> </tt>
<a name="L169"></a><tt class="py-lineno"> 169</tt>  <tt class="py-line"><tt class="py-docstring">    matcher.</tt> </tt>
<a name="L170"></a><tt class="py-lineno"> 170</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L171"></a><tt class="py-lineno"> 171</tt>  <tt class="py-line"><tt class="py-docstring">    The resulting 'match' function returns True when given an object matching</tt> </tt>
<a name="L172"></a><tt class="py-lineno"> 172</tt>  <tt class="py-line"><tt class="py-docstring">    the specification (identity, type or attribute values), otherwise False.</tt> </tt>
<a name="L173"></a><tt class="py-lineno"> 173</tt>  <tt class="py-line"><tt class="py-docstring">    This is useful for writing functions that search the tree, and probably</tt> </tt>
<a name="L174"></a><tt class="py-lineno"> 174</tt>  <tt class="py-line"><tt class="py-docstring">    shouldn't be used directly by the end user.</tt> </tt>
<a name="L175"></a><tt class="py-lineno"> 175</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L176"></a><tt class="py-lineno"> 176</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">,</tt> <tt id="link-43" class="py-name"><a title="Bio.Phylo.BaseTree.TreeElement" class="py-name" href="#" onclick="return doclink('link-43', 'TreeElement', 'link-17');">TreeElement</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L177"></a><tt class="py-lineno"> 177</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-44" class="py-name" targets="Function Bio.Phylo.BaseTree._identity_matcher()=Bio.Phylo.BaseTree-module.html#_identity_matcher"><a title="Bio.Phylo.BaseTree._identity_matcher" class="py-name" href="#" onclick="return doclink('link-44', '_identity_matcher', 'link-44');">_identity_matcher</a></tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">)</tt> </tt>
<a name="L178"></a><tt class="py-lineno"> 178</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">,</tt> <tt id="link-45" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-45', 'type', 'link-39');">type</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L179"></a><tt class="py-lineno"> 179</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-46" class="py-name" targets="Function Bio.Phylo.BaseTree._class_matcher()=Bio.Phylo.BaseTree-module.html#_class_matcher"><a title="Bio.Phylo.BaseTree._class_matcher" class="py-name" href="#" onclick="return doclink('link-46', '_class_matcher', 'link-46');">_class_matcher</a></tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">)</tt> </tt>
<a name="L180"></a><tt class="py-lineno"> 180</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L181"></a><tt class="py-lineno"> 181</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-47" class="py-name" targets="Function Bio.Phylo.BaseTree._string_matcher()=Bio.Phylo.BaseTree-module.html#_string_matcher"><a title="Bio.Phylo.BaseTree._string_matcher" class="py-name" href="#" onclick="return doclink('link-47', '_string_matcher', 'link-47');">_string_matcher</a></tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">)</tt> </tt>
<a name="L182"></a><tt class="py-lineno"> 182</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">,</tt> <tt class="py-name">dict</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L183"></a><tt class="py-lineno"> 183</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-48" class="py-name" targets="Function Bio.Phylo.BaseTree._attribute_matcher()=Bio.Phylo.BaseTree-module.html#_attribute_matcher"><a title="Bio.Phylo.BaseTree._attribute_matcher" class="py-name" href="#" onclick="return doclink('link-48', '_attribute_matcher', 'link-48');">_attribute_matcher</a></tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">)</tt> </tt>
<a name="L184"></a><tt class="py-lineno"> 184</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">callable</tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L185"></a><tt class="py-lineno"> 185</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-49" class="py-name" targets="Function Bio.Phylo.BaseTree._function_matcher()=Bio.Phylo.BaseTree-module.html#_function_matcher"><a title="Bio.Phylo.BaseTree._function_matcher" class="py-name" href="#" onclick="return doclink('link-49', '_function_matcher', 'link-49');">_function_matcher</a></tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">)</tt> </tt>
<a name="L186"></a><tt class="py-lineno"> 186</tt>  <tt class="py-line">    <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"%s (type %s) is not a valid type for comparison."</tt> </tt>
<a name="L187"></a><tt class="py-lineno"> 187</tt>  <tt class="py-line">                     <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">,</tt> <tt id="link-50" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-50', 'type', 'link-39');">type</a></tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L188"></a><tt class="py-lineno"> 188</tt>  <tt class="py-line"> </tt>
<a name="_combine_matchers"></a><div id="_combine_matchers-def"><a name="L189"></a><tt class="py-lineno"> 189</tt>  <tt class="py-line"> </tt>
<a name="L190"></a><tt class="py-lineno"> 190</tt> <a class="py-toggle" href="#" id="_combine_matchers-toggle" onclick="return toggle('_combine_matchers');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree-module.html#_combine_matchers">_combine_matchers</a><tt class="py-op">(</tt><tt class="py-param">target</tt><tt class="py-op">,</tt> <tt class="py-param">kwargs</tt><tt class="py-op">,</tt> <tt class="py-param">require_spec</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_combine_matchers-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_combine_matchers-expanded"><a name="L191"></a><tt class="py-lineno"> 191</tt>  <tt class="py-line">    <tt class="py-docstring">"""Merge target specifications with keyword arguments (PRIVATE).</tt> </tt>
<a name="L192"></a><tt class="py-lineno"> 192</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L193"></a><tt class="py-lineno"> 193</tt>  <tt class="py-line"><tt class="py-docstring">    Dispatch the components to the various matcher functions, then merge into a</tt> </tt>
<a name="L194"></a><tt class="py-lineno"> 194</tt>  <tt class="py-line"><tt class="py-docstring">    single boolean function.</tt> </tt>
<a name="L195"></a><tt class="py-lineno"> 195</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L196"></a><tt class="py-lineno"> 196</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">target</tt><tt class="py-op">:</tt> </tt>
<a name="L197"></a><tt class="py-lineno"> 197</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">kwargs</tt><tt class="py-op">:</tt> </tt>
<a name="L198"></a><tt class="py-lineno"> 198</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">require_spec</tt><tt class="py-op">:</tt> </tt>
<a name="L199"></a><tt class="py-lineno"> 199</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"you must specify a target object or keyword "</tt> </tt>
<a name="L200"></a><tt class="py-lineno"> 200</tt>  <tt class="py-line">                                 <tt class="py-string">"arguments."</tt><tt class="py-op">)</tt> </tt>
<a name="L201"></a><tt class="py-lineno"> 201</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-keyword">lambda</tt> <tt id="link-51" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-51', 'x', 'link-14');">x</a></tt><tt class="py-op">:</tt> <tt class="py-name">True</tt> </tt>
<a name="L202"></a><tt class="py-lineno"> 202</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-52" class="py-name"><a title="Bio.Phylo.BaseTree._attribute_matcher" class="py-name" href="#" onclick="return doclink('link-52', '_attribute_matcher', 'link-48');">_attribute_matcher</a></tt><tt class="py-op">(</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> </tt>
<a name="L203"></a><tt class="py-lineno"> 203</tt>  <tt class="py-line">    <tt class="py-name">match_obj</tt> <tt class="py-op">=</tt> <tt id="link-53" class="py-name" targets="Function Bio.Phylo.BaseTree._object_matcher()=Bio.Phylo.BaseTree-module.html#_object_matcher"><a title="Bio.Phylo.BaseTree._object_matcher" class="py-name" href="#" onclick="return doclink('link-53', '_object_matcher', 'link-53');">_object_matcher</a></tt><tt class="py-op">(</tt><tt class="py-name">target</tt><tt class="py-op">)</tt> </tt>
<a name="L204"></a><tt class="py-lineno"> 204</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">kwargs</tt><tt class="py-op">:</tt> </tt>
<a name="L205"></a><tt class="py-lineno"> 205</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">match_obj</tt> </tt>
<a name="L206"></a><tt class="py-lineno"> 206</tt>  <tt class="py-line">    <tt class="py-name">match_kwargs</tt> <tt class="py-op">=</tt> <tt id="link-54" class="py-name"><a title="Bio.Phylo.BaseTree._attribute_matcher" class="py-name" href="#" onclick="return doclink('link-54', '_attribute_matcher', 'link-48');">_attribute_matcher</a></tt><tt class="py-op">(</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> </tt>
<a name="L207"></a><tt class="py-lineno"> 207</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-keyword">lambda</tt> <tt id="link-55" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-55', 'x', 'link-14');">x</a></tt><tt class="py-op">:</tt> <tt class="py-name">match_obj</tt><tt class="py-op">(</tt><tt id="link-56" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-56', 'x', 'link-14');">x</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-name">match_kwargs</tt><tt class="py-op">(</tt><tt id="link-57" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-57', 'x', 'link-14');">x</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L208"></a><tt class="py-lineno"> 208</tt>  <tt class="py-line"> </tt>
<a name="_combine_args"></a><div id="_combine_args-def"><a name="L209"></a><tt class="py-lineno"> 209</tt>  <tt class="py-line"> </tt>
<a name="L210"></a><tt class="py-lineno"> 210</tt> <a class="py-toggle" href="#" id="_combine_args-toggle" onclick="return toggle('_combine_args');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree-module.html#_combine_args">_combine_args</a><tt class="py-op">(</tt><tt class="py-param">first</tt><tt class="py-op">,</tt> <tt class="py-op">*</tt><tt class="py-param">rest</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_combine_args-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_combine_args-expanded"><a name="L211"></a><tt class="py-lineno"> 211</tt>  <tt class="py-line">    <tt class="py-docstring">"""Convert ``[targets]`` or ``*targets`` arguments to a single iterable (PRIVATE).</tt> </tt>
<a name="L212"></a><tt class="py-lineno"> 212</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L213"></a><tt class="py-lineno"> 213</tt>  <tt class="py-line"><tt class="py-docstring">    This helps other functions work like the built-in functions `max` and</tt> </tt>
<a name="L214"></a><tt class="py-lineno"> 214</tt>  <tt class="py-line"><tt class="py-docstring">    `min`.</tt> </tt>
<a name="L215"></a><tt class="py-lineno"> 215</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L216"></a><tt class="py-lineno"> 216</tt>  <tt class="py-line">    <tt class="py-comment"># Background: is_monophyletic takes a single list or iterable (like the</tt> </tt>
<a name="L217"></a><tt class="py-lineno"> 217</tt>  <tt class="py-line">    <tt class="py-comment"># same method in Bio.Nexus.Trees); root_with_outgroup and common_ancestor</tt> </tt>
<a name="L218"></a><tt class="py-lineno"> 218</tt>  <tt class="py-line">    <tt class="py-comment"># take separate arguments. This mismatch was in the initial release and I</tt> </tt>
<a name="L219"></a><tt class="py-lineno"> 219</tt>  <tt class="py-line">    <tt class="py-comment"># didn't notice the inconsistency until after Biopython 1.55. I can think</tt> </tt>
<a name="L220"></a><tt class="py-lineno"> 220</tt>  <tt class="py-line">    <tt class="py-comment"># of cases where either style is more convenient, so let's support both</tt> </tt>
<a name="L221"></a><tt class="py-lineno"> 221</tt>  <tt class="py-line">    <tt class="py-comment"># (for backward compatibility and consistency between methods).</tt> </tt>
<a name="L222"></a><tt class="py-lineno"> 222</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">first</tt><tt class="py-op">,</tt> <tt class="py-string">'__iter__'</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt class="py-op">(</tt><tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">first</tt><tt class="py-op">,</tt> <tt id="link-58" class="py-name"><a title="Bio.Phylo.BaseTree.TreeElement" class="py-name" href="#" onclick="return doclink('link-58', 'TreeElement', 'link-17');">TreeElement</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> </tt>
<a name="L223"></a><tt class="py-lineno"> 223</tt>  <tt class="py-line">                                           <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">first</tt><tt class="py-op">,</tt> <tt id="link-59" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-59', 'type', 'link-39');">type</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> </tt>
<a name="L224"></a><tt class="py-lineno"> 224</tt>  <tt class="py-line">                                           <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">first</tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> </tt>
<a name="L225"></a><tt class="py-lineno"> 225</tt>  <tt class="py-line">                                           <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">first</tt><tt class="py-op">,</tt> <tt class="py-name">dict</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L226"></a><tt class="py-lineno"> 226</tt>  <tt class="py-line">        <tt class="py-comment"># `terminals` is an iterable of targets</tt> </tt>
<a name="L227"></a><tt class="py-lineno"> 227</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-60" class="py-name" targets="Method Bio.Nexus.Nexus.CharBuffer.rest()=Bio.Nexus.Nexus.CharBuffer-class.html#rest"><a title="Bio.Nexus.Nexus.CharBuffer.rest" class="py-name" href="#" onclick="return doclink('link-60', 'rest', 'link-60');">rest</a></tt><tt class="py-op">:</tt> </tt>
<a name="L228"></a><tt class="py-lineno"> 228</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Arguments must be either a single list of "</tt> </tt>
<a name="L229"></a><tt class="py-lineno"> 229</tt>  <tt class="py-line">                             <tt class="py-string">"targets, or separately specified targets "</tt> </tt>
<a name="L230"></a><tt class="py-lineno"> 230</tt>  <tt class="py-line">                             <tt class="py-string">"(e.g. foo(t1, t2, t3)), but not both."</tt><tt class="py-op">)</tt> </tt>
<a name="L231"></a><tt class="py-lineno"> 231</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">first</tt> </tt>
<a name="L232"></a><tt class="py-lineno"> 232</tt>  <tt class="py-line">    <tt class="py-comment"># `terminals` is a single target -- wrap in a container</tt> </tt>
<a name="L233"></a><tt class="py-lineno"> 233</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">itertools</tt><tt class="py-op">.</tt><tt class="py-name">chain</tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">first</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt id="link-61" class="py-name"><a title="Bio.Nexus.Nexus.CharBuffer.rest" class="py-name" href="#" onclick="return doclink('link-61', 'rest', 'link-60');">rest</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L234"></a><tt class="py-lineno"> 234</tt>  <tt class="py-line"> </tt>
<a name="TreeElement"></a><div id="TreeElement-def"><a name="L235"></a><tt class="py-lineno"> 235</tt>  <tt class="py-line"> </tt>
<a name="L236"></a><tt class="py-lineno"> 236</tt>  <tt class="py-line"><tt class="py-comment"># Class definitions</tt> </tt>
<a name="L237"></a><tt class="py-lineno"> 237</tt>  <tt class="py-line"> </tt>
<a name="L238"></a><tt class="py-lineno"> 238</tt> <a class="py-toggle" href="#" id="TreeElement-toggle" onclick="return toggle('TreeElement');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.TreeElement-class.html">TreeElement</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="TreeElement-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="TreeElement-expanded"><a name="L239"></a><tt class="py-lineno"> 239</tt>  <tt class="py-line">    <tt class="py-docstring">"""Base class for all Bio.Phylo classes."""</tt> </tt>
<a name="L240"></a><tt class="py-lineno"> 240</tt>  <tt class="py-line"> </tt>
<a name="TreeElement.__repr__"></a><div id="TreeElement.__repr__-def"><a name="L241"></a><tt class="py-lineno"> 241</tt> <a class="py-toggle" href="#" id="TreeElement.__repr__-toggle" onclick="return toggle('TreeElement.__repr__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.TreeElement-class.html#__repr__">__repr__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="TreeElement.__repr__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="TreeElement.__repr__-expanded"><a name="L242"></a><tt class="py-lineno"> 242</tt>  <tt class="py-line">        <tt class="py-docstring">"""Show this object's constructor with its primitive arguments."""</tt> </tt>
<a name="L243"></a><tt class="py-lineno"> 243</tt>  <tt class="py-line">        <tt class="py-keyword">def</tt> <tt class="py-def-name">pair_as_kwarg_string</tt><tt class="py-op">(</tt><tt class="py-param">key</tt><tt class="py-op">,</tt> <tt class="py-param">val</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L244"></a><tt class="py-lineno"> 244</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">val</tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L245"></a><tt class="py-lineno"> 245</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-string">"%s='%s'"</tt> </tt>
<a name="L246"></a><tt class="py-lineno"> 246</tt>  <tt class="py-line">                        <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">,</tt> <tt id="link-62" class="py-name"><a title="Bio.Phylo._utils
Bio.SearchIO._utils
Bio._utils" class="py-name" href="#" onclick="return doclink('link-62', '_utils', 'link-5');">_utils</a></tt><tt class="py-op">.</tt><tt id="link-63" class="py-name" targets="Function Bio._utils.trim_str()=Bio._utils-module.html#trim_str"><a title="Bio._utils.trim_str" class="py-name" href="#" onclick="return doclink('link-63', 'trim_str', 'link-63');">trim_str</a></tt><tt class="py-op">(</tt><tt id="link-64" class="py-name"><a title="Bio.Phylo.BaseTree.as_string
Bio.Restriction.Restriction.RestrictionBatch.as_string" class="py-name" href="#" onclick="return doclink('link-64', 'as_string', 'link-6');">as_string</a></tt><tt class="py-op">(</tt><tt class="py-name">val</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">60</tt><tt class="py-op">,</tt> <tt class="py-string">'...'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L247"></a><tt class="py-lineno"> 247</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-string">"%s=%s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">val</tt><tt class="py-op">)</tt> </tt>
</div><a name="L248"></a><tt class="py-lineno"> 248</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-string">'%s(%s)'</tt> </tt>
<a name="L249"></a><tt class="py-lineno"> 249</tt>  <tt class="py-line">                <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">,</tt> </tt>
<a name="L250"></a><tt class="py-lineno"> 250</tt>  <tt class="py-line">                   <tt class="py-string">', '</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">pair_as_kwarg_string</tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">val</tt><tt class="py-op">)</tt> </tt>
<a name="L251"></a><tt class="py-lineno"> 251</tt>  <tt class="py-line">                             <tt class="py-keyword">for</tt> <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">val</tt> <tt class="py-keyword">in</tt> <tt class="py-name">sorted</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__dict__</tt><tt class="py-op">.</tt><tt id="link-65" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-65', 'items', 'link-11');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L252"></a><tt class="py-lineno"> 252</tt>  <tt class="py-line">                             <tt class="py-keyword">if</tt> <tt class="py-name">val</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> </tt>
<a name="L253"></a><tt class="py-lineno"> 253</tt>  <tt class="py-line">                             <tt id="link-66" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-66', 'type', 'link-39');">type</a></tt><tt class="py-op">(</tt><tt class="py-name">val</tt><tt class="py-op">)</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">,</tt> <tt class="py-name">float</tt><tt class="py-op">,</tt> <tt class="py-name">bool</tt><tt class="py-op">,</tt> <tt class="py-name">unicode</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L254"></a><tt class="py-lineno"> 254</tt>  <tt class="py-line"> </tt>
<a name="L255"></a><tt class="py-lineno"> 255</tt>  <tt class="py-line">    <tt id="link-67" class="py-name" targets="Method Bio.Align.AlignInfo.PSSM.__str__()=Bio.Align.AlignInfo.PSSM-class.html#__str__,Method Bio.Align.MultipleSeqAlignment.__str__()=Bio.Align.MultipleSeqAlignment-class.html#__str__,Method Bio.Align.PairwiseAlignment.__str__()=Bio.Align.PairwiseAlignment-class.html#__str__,Method Bio.Application.AbstractCommandline.__str__()=Bio.Application.AbstractCommandline-class.html#__str__,Method Bio.Application.ApplicationError.__str__()=Bio.Application.ApplicationError-class.html#__str__,Method Bio.Application._AbstractParameter.__str__()=Bio.Application._AbstractParameter-class.html#__str__,Method Bio.Application._Argument.__str__()=Bio.Application._Argument-class.html#__str__,Method Bio.Application._ArgumentList.__str__()=Bio.Application._ArgumentList-class.html#__str__,Method Bio.Application._Option.__str__()=Bio.Application._Option-class.html#__str__,Method Bio.Application._StaticArgument.__str__()=Bio.Application._StaticArgument-class.html#__str__,Method Bio.Application._Switch.__str__()=Bio.Application._Switch-class.html#__str__,Method Bio.Blast.Record.Alignment.__str__()=Bio.Blast.Record.Alignment-class.html#__str__,Method Bio.Blast.Record.Description.__str__()=Bio.Blast.Record.Description-class.html#__str__,Method Bio.Blast.Record.HSP.__str__()=Bio.Blast.Record.HSP-class.html#__str__,Method Bio.Crystal.Chain.__str__()=Bio.Crystal.Chain-class.html#__str__,Method Bio.Crystal.Crystal.__str__()=Bio.Crystal.Crystal-class.html#__str__,Method Bio.Crystal.Hetero.__str__()=Bio.Crystal.Hetero-class.html#__str__,Method Bio.Data.CodonTable.CodonTable.__str__()=Bio.Data.CodonTable.CodonTable-class.html#__str__,Method Bio.Emboss.PrimerSearch.InputRecord.__str__()=Bio.Emboss.PrimerSearch.InputRecord-class.html#__str__,Method Bio.Entrez.Parser.CorruptedXMLError.__str__()=Bio.Entrez.Parser.CorruptedXMLError-class.html#__str__,Method Bio.Entrez.Parser.NotXMLError.__str__()=Bio.Entrez.Parser.NotXMLError-class.html#__str__,Method Bio.Entrez.Parser.ValidationError.__str__()=Bio.Entrez.Parser.ValidationError-class.html#__str__,Method Bio.ExPASy.Enzyme.Record.__str__()=Bio.ExPASy.Enzyme.Record-class.html#__str__,Method Bio.ExPASy.cellosaurus.Record.__str__()=Bio.ExPASy.cellosaurus.Record-class.html#__str__,Method Bio.File._IndexedSeqFileDict.__str__()=Bio.File._IndexedSeqFileDict-class.html#__str__,Method Bio.GenBank.Record.Feature.__str__()=Bio.GenBank.Record.Feature-class.html#__str__,Method Bio.GenBank.Record.Qualifier.__str__()=Bio.GenBank.Record.Qualifier-class.html#__str__,Method Bio.GenBank.Record.Record.__str__()=Bio.GenBank.Record.Record-class.html#__str__,Method Bio.GenBank.Record.Reference.__str__()=Bio.GenBank.Record.Reference-class.html#__str__,Method Bio.Geo.Record.Record.__str__()=Bio.Geo.Record.Record-class.html#__str__,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.__str__()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#__str__,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.__str__()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#__str__,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.__str__()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#__str__,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.__str__()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#__str__,Method Bio.Graphics.GenomeDiagram._Track.Track.__str__()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#__str__,Method Bio.KEGG.Compound.Record.__str__()=Bio.KEGG.Compound.Record-class.html#__str__,Method Bio.KEGG.Enzyme.Record.__str__()=Bio.KEGG.Enzyme.Record-class.html#__str__,Method Bio.KEGG.Gene.Record.__str__()=Bio.KEGG.Gene.Record-class.html#__str__,Method Bio.KEGG.KGML.KGML_pathway.Entry.__str__()=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#__str__,Method Bio.KEGG.KGML.KGML_pathway.Pathway.__str__()=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#__str__,Method Bio.KEGG.KGML.KGML_pathway.Reaction.__str__()=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#__str__,Method Bio.KEGG.KGML.KGML_pathway.Relation.__str__()=Bio.KEGG.KGML.KGML_pathway.Relation-class.html#__str__,Method Bio.MarkovModel.MarkovModel.__str__()=Bio.MarkovModel.MarkovModel-class.html#__str__,Method Bio.Nexus.StandardData.StandardData.__str__()=Bio.Nexus.StandardData.StandardData-class.html#__str__,Method Bio.Nexus.Trees.Tree.__str__()=Bio.Nexus.Trees.Tree-class.html#__str__,Method Bio.Pathway.Interaction.__str__()=Bio.Pathway.Interaction-class.html#__str__,Method Bio.Pathway.Network.__str__()=Bio.Pathway.Network-class.html#__str__,Method Bio.Pathway.Reaction.__str__()=Bio.Pathway.Reaction-class.html#__str__,Method Bio.Pathway.Rep.Graph.Graph.__str__()=Bio.Pathway.Rep.Graph.Graph-class.html#__str__,Method Bio.Pathway.Rep.MultiGraph.MultiGraph.__str__()=Bio.Pathway.Rep.MultiGraph.MultiGraph-class.html#__str__,Method Bio.Pathway.System.__str__()=Bio.Pathway.System-class.html#__str__,Method Bio.Phylo.BaseTree.BranchColor.__str__()=Bio.Phylo.BaseTree.BranchColor-class.html#__str__,Method Bio.Phylo.BaseTree.Clade.__str__()=Bio.Phylo.BaseTree.Clade-class.html#__str__,Method Bio.Phylo.BaseTree.Tree.__str__()=Bio.Phylo.BaseTree.Tree-class.html#__str__,Method Bio.Phylo.PhyloXML.Accession.__str__()=Bio.Phylo.PhyloXML.Accession-class.html#__str__,Method Bio.Phylo.PhyloXML.Date.__str__()=Bio.Phylo.PhyloXML.Date-class.html#__str__,Method Bio.Phylo.PhyloXML.Id.__str__()=Bio.Phylo.PhyloXML.Id-class.html#__str__,Method Bio.Phylo.PhyloXML.MolSeq.__str__()=Bio.Phylo.PhyloXML.MolSeq-class.html#__str__,Method Bio.Phylo.PhyloXML.Phyloxml.__str__()=Bio.Phylo.PhyloXML.Phyloxml-class.html#__str__,Method Bio.Phylo.PhyloXML.Polygon.__str__()=Bio.Phylo.PhyloXML.Polygon-class.html#__str__,Method Bio.Phylo.PhyloXML.Taxonomy.__str__()=Bio.Phylo.PhyloXML.Taxonomy-class.html#__str__,Method Bio.Phylo.PhyloXML.Uri.__str__()=Bio.Phylo.PhyloXML.Uri-class.html#__str__,Method Bio.Phylo.TreeConstruction._Matrix.__str__()=Bio.Phylo.TreeConstruction._Matrix-class.html#__str__,Method Bio.PopGen.GenePop.FileParser.FileRecord.__str__()=Bio.PopGen.GenePop.FileParser.FileRecord-class.html#__str__,Method Bio.PopGen.GenePop.Record.__str__()=Bio.PopGen.GenePop.Record-class.html#__str__,Method Bio.Restriction.Restriction.RestrictionBatch.__str__()=Bio.Restriction.Restriction.RestrictionBatch-class.html#__str__,Method Bio.Restriction.Restriction.RestrictionType.__str__()=Bio.Restriction.Restriction.RestrictionType-class.html#__str__,Method Bio.SCOP.Cla.Record.__str__()=Bio.SCOP.Cla.Record-class.html#__str__,Method Bio.SCOP.Des.Record.__str__()=Bio.SCOP.Des.Record-class.html#__str__,Method Bio.SCOP.Dom.Record.__str__()=Bio.SCOP.Dom.Record-class.html#__str__,Method Bio.SCOP.Domain.__str__()=Bio.SCOP.Domain-class.html#__str__,Method Bio.SCOP.Hie.Record.__str__()=Bio.SCOP.Hie.Record-class.html#__str__,Method Bio.SCOP.Node.__str__()=Bio.SCOP.Node-class.html#__str__,Method Bio.SCOP.Residues.Residues.__str__()=Bio.SCOP.Residues.Residues-class.html#__str__,Method Bio.SearchIO._model.hit.Hit.__str__()=Bio.SearchIO._model.hit.Hit-class.html#__str__,Method Bio.SearchIO._model.hsp.HSP.__str__()=Bio.SearchIO._model.hsp.HSP-class.html#__str__,Method Bio.SearchIO._model.hsp.HSPFragment.__str__()=Bio.SearchIO._model.hsp.HSPFragment-class.html#__str__,Method Bio.SearchIO._model.query.QueryResult.__str__()=Bio.SearchIO._model.query.QueryResult-class.html#__str__,Method Bio.Seq.MutableSeq.__str__()=Bio.Seq.MutableSeq-class.html#__str__,Method Bio.Seq.Seq.__str__()=Bio.Seq.Seq-class.html#__str__,Method Bio.Seq.UnknownSeq.__str__()=Bio.Seq.UnknownSeq-class.html#__str__,Method Bio.SeqFeature.AfterPosition.__str__()=Bio.SeqFeature.AfterPosition-class.html#__str__,Method Bio.SeqFeature.BeforePosition.__str__()=Bio.SeqFeature.BeforePosition-class.html#__str__,Method Bio.SeqFeature.BetweenPosition.__str__()=Bio.SeqFeature.BetweenPosition-class.html#__str__,Method Bio.SeqFeature.CompoundLocation.__str__()=Bio.SeqFeature.CompoundLocation-class.html#__str__,Method Bio.SeqFeature.FeatureLocation.__str__()=Bio.SeqFeature.FeatureLocation-class.html#__str__,Method Bio.SeqFeature.OneOfPosition.__str__()=Bio.SeqFeature.OneOfPosition-class.html#__str__,Method Bio.SeqFeature.PositionGap.__str__()=Bio.SeqFeature.PositionGap-class.html#__str__,Method Bio.SeqFeature.Reference.__str__()=Bio.SeqFeature.Reference-class.html#__str__,Method Bio.SeqFeature.SeqFeature.__str__()=Bio.SeqFeature.SeqFeature-class.html#__str__,Method Bio.SeqFeature.WithinPosition.__str__()=Bio.SeqFeature.WithinPosition-class.html#__str__,Method Bio.SeqRecord.SeqRecord.__str__()=Bio.SeqRecord.SeqRecord-class.html#__str__,Method Bio.SubsMat.SeqMat.__str__()=Bio.SubsMat.SeqMat-class.html#__str__,Method Bio.Wise.dnal.Statistics.__str__()=Bio.Wise.dnal.Statistics-class.html#__str__,Method Bio.Wise.psw.ColumnUnit.__str__()=Bio.Wise.psw.ColumnUnit-class.html#__str__,Method Bio.codonalign.codonalignment.CodonAlignment.__str__()=Bio.codonalign.codonalignment.CodonAlignment-class.html#__str__,Method Bio.motifs.Instances.__str__()=Bio.motifs.Instances-class.html#__str__,Method Bio.motifs.Motif.__str__()=Bio.motifs.Motif-class.html#__str__,Method Bio.motifs.jaspar.Motif.__str__()=Bio.motifs.jaspar.Motif-class.html#__str__,Method Bio.motifs.jaspar.Record.__str__()=Bio.motifs.jaspar.Record-class.html#__str__,Method Bio.motifs.jaspar.db.JASPAR5.__str__()=Bio.motifs.jaspar.db.JASPAR5-class.html#__str__,Method Bio.motifs.matrix.GenericPositionMatrix.__str__()=Bio.motifs.matrix.GenericPositionMatrix-class.html#__str__,Method Bio.motifs.transfac.Record.__str__()=Bio.motifs.transfac.Record-class.html#__str__,Method Bio.phenotype.phen_micro.PlateRecord.__str__()=Bio.phenotype.phen_micro.PlateRecord-class.html#__str__,Method Bio.phenotype.phen_micro.WellRecord.__str__()=Bio.phenotype.phen_micro.WellRecord-class.html#__str__,Method BioSQL.BioSeq.DBSeq.__str__()=BioSQL.BioSeq.DBSeq-class.html#__str__"><a title="Bio.Align.AlignInfo.PSSM.__str__
Bio.Align.MultipleSeqAlignment.__str__
Bio.Align.PairwiseAlignment.__str__
Bio.Application.AbstractCommandline.__str__
Bio.Application.ApplicationError.__str__
Bio.Application._AbstractParameter.__str__
Bio.Application._Argument.__str__
Bio.Application._ArgumentList.__str__
Bio.Application._Option.__str__
Bio.Application._StaticArgument.__str__
Bio.Application._Switch.__str__
Bio.Blast.Record.Alignment.__str__
Bio.Blast.Record.Description.__str__
Bio.Blast.Record.HSP.__str__
Bio.Crystal.Chain.__str__
Bio.Crystal.Crystal.__str__
Bio.Crystal.Hetero.__str__
Bio.Data.CodonTable.CodonTable.__str__
Bio.Emboss.PrimerSearch.InputRecord.__str__
Bio.Entrez.Parser.CorruptedXMLError.__str__
Bio.Entrez.Parser.NotXMLError.__str__
Bio.Entrez.Parser.ValidationError.__str__
Bio.ExPASy.Enzyme.Record.__str__
Bio.ExPASy.cellosaurus.Record.__str__
Bio.File._IndexedSeqFileDict.__str__
Bio.GenBank.Record.Feature.__str__
Bio.GenBank.Record.Qualifier.__str__
Bio.GenBank.Record.Record.__str__
Bio.GenBank.Record.Reference.__str__
Bio.Geo.Record.Record.__str__
Bio.Graphics.GenomeDiagram._Diagram.Diagram.__str__
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.__str__
Bio.Graphics.GenomeDiagram._Graph.GraphData.__str__
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.__str__
Bio.Graphics.GenomeDiagram._Track.Track.__str__
Bio.KEGG.Compound.Record.__str__
Bio.KEGG.Enzyme.Record.__str__
Bio.KEGG.Gene.Record.__str__
Bio.KEGG.KGML.KGML_pathway.Entry.__str__
Bio.KEGG.KGML.KGML_pathway.Pathway.__str__
Bio.KEGG.KGML.KGML_pathway.Reaction.__str__
Bio.KEGG.KGML.KGML_pathway.Relation.__str__
Bio.MarkovModel.MarkovModel.__str__
Bio.Nexus.StandardData.StandardData.__str__
Bio.Nexus.Trees.Tree.__str__
Bio.Pathway.Interaction.__str__
Bio.Pathway.Network.__str__
Bio.Pathway.Reaction.__str__
Bio.Pathway.Rep.Graph.Graph.__str__
Bio.Pathway.Rep.MultiGraph.MultiGraph.__str__
Bio.Pathway.System.__str__
Bio.Phylo.BaseTree.BranchColor.__str__
Bio.Phylo.BaseTree.Clade.__str__
Bio.Phylo.BaseTree.Tree.__str__
Bio.Phylo.PhyloXML.Accession.__str__
Bio.Phylo.PhyloXML.Date.__str__
Bio.Phylo.PhyloXML.Id.__str__
Bio.Phylo.PhyloXML.MolSeq.__str__
Bio.Phylo.PhyloXML.Phyloxml.__str__
Bio.Phylo.PhyloXML.Polygon.__str__
Bio.Phylo.PhyloXML.Taxonomy.__str__
Bio.Phylo.PhyloXML.Uri.__str__
Bio.Phylo.TreeConstruction._Matrix.__str__
Bio.PopGen.GenePop.FileParser.FileRecord.__str__
Bio.PopGen.GenePop.Record.__str__
Bio.Restriction.Restriction.RestrictionBatch.__str__
Bio.Restriction.Restriction.RestrictionType.__str__
Bio.SCOP.Cla.Record.__str__
Bio.SCOP.Des.Record.__str__
Bio.SCOP.Dom.Record.__str__
Bio.SCOP.Domain.__str__
Bio.SCOP.Hie.Record.__str__
Bio.SCOP.Node.__str__
Bio.SCOP.Residues.Residues.__str__
Bio.SearchIO._model.hit.Hit.__str__
Bio.SearchIO._model.hsp.HSP.__str__
Bio.SearchIO._model.hsp.HSPFragment.__str__
Bio.SearchIO._model.query.QueryResult.__str__
Bio.Seq.MutableSeq.__str__
Bio.Seq.Seq.__str__
Bio.Seq.UnknownSeq.__str__
Bio.SeqFeature.AfterPosition.__str__
Bio.SeqFeature.BeforePosition.__str__
Bio.SeqFeature.BetweenPosition.__str__
Bio.SeqFeature.CompoundLocation.__str__
Bio.SeqFeature.FeatureLocation.__str__
Bio.SeqFeature.OneOfPosition.__str__
Bio.SeqFeature.PositionGap.__str__
Bio.SeqFeature.Reference.__str__
Bio.SeqFeature.SeqFeature.__str__
Bio.SeqFeature.WithinPosition.__str__
Bio.SeqRecord.SeqRecord.__str__
Bio.SubsMat.SeqMat.__str__
Bio.Wise.dnal.Statistics.__str__
Bio.Wise.psw.ColumnUnit.__str__
Bio.codonalign.codonalignment.CodonAlignment.__str__
Bio.motifs.Instances.__str__
Bio.motifs.Motif.__str__
Bio.motifs.jaspar.Motif.__str__
Bio.motifs.jaspar.Record.__str__
Bio.motifs.jaspar.db.JASPAR5.__str__
Bio.motifs.matrix.GenericPositionMatrix.__str__
Bio.motifs.transfac.Record.__str__
Bio.phenotype.phen_micro.PlateRecord.__str__
Bio.phenotype.phen_micro.WellRecord.__str__
BioSQL.BioSeq.DBSeq.__str__" class="py-name" href="#" onclick="return doclink('link-67', '__str__', 'link-67');">__str__</a></tt> <tt class="py-op">=</tt> <tt id="link-68" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.__repr__()=Bio.Align.MultipleSeqAlignment-class.html#__repr__,Method Bio.AlignIO.MafIO.MafIndex.__repr__()=Bio.AlignIO.MafIO.MafIndex-class.html#__repr__,Method Bio.Alphabet.Alphabet.__repr__()=Bio.Alphabet.Alphabet-class.html#__repr__,Method Bio.Alphabet.AlphabetEncoder.__repr__()=Bio.Alphabet.AlphabetEncoder-class.html#__repr__,Method Bio.Application.AbstractCommandline.__repr__()=Bio.Application.AbstractCommandline-class.html#__repr__,Method Bio.Application.ApplicationError.__repr__()=Bio.Application.ApplicationError-class.html#__repr__,Method Bio.Crystal.Crystal.__repr__()=Bio.Crystal.Crystal-class.html#__repr__,Method Bio.Crystal.Hetero.__repr__()=Bio.Crystal.Hetero-class.html#__repr__,Method Bio.Entrez.Parser.DictionaryElement.__repr__()=Bio.Entrez.Parser.DictionaryElement-class.html#__repr__,Method Bio.Entrez.Parser.IntegerElement.__repr__()=Bio.Entrez.Parser.IntegerElement-class.html#__repr__,Method Bio.Entrez.Parser.ListElement.__repr__()=Bio.Entrez.Parser.ListElement-class.html#__repr__,Method Bio.Entrez.Parser.StringElement.__repr__()=Bio.Entrez.Parser.StringElement-class.html#__repr__,Method Bio.Entrez.Parser.UnicodeElement.__repr__()=Bio.Entrez.Parser.UnicodeElement-class.html#__repr__,Method Bio.ExPASy.Enzyme.Record.__repr__()=Bio.ExPASy.Enzyme.Record-class.html#__repr__,Method Bio.ExPASy.cellosaurus.Record.__repr__()=Bio.ExPASy.cellosaurus.Record-class.html#__repr__,Method Bio.FSSP.FSSPSumRec.__repr__()=Bio.FSSP.FSSPSumRec-class.html#__repr__,Method Bio.FSSP.PosAlign.__repr__()=Bio.FSSP.PosAlign-class.html#__repr__,Method Bio.FSSP.fssp_rec.fff_rec.__repr__()=Bio.FSSP.fssp_rec.fff_rec-class.html#__repr__,Method Bio.File._IndexedSeqFileDict.__repr__()=Bio.File._IndexedSeqFileDict-class.html#__repr__,Method Bio.File._SQLiteManySeqFilesDict.__repr__()=Bio.File._SQLiteManySeqFilesDict-class.html#__repr__,Method Bio.GenBank.Record.Feature.__repr__()=Bio.GenBank.Record.Feature-class.html#__repr__,Method Bio.GenBank.Record.Qualifier.__repr__()=Bio.GenBank.Record.Qualifier-class.html#__repr__,Method Bio.PDB.Atom.Atom.__repr__()=Bio.PDB.Atom.Atom-class.html#__repr__,Method Bio.PDB.Atom.DisorderedAtom.__repr__()=Bio.PDB.Atom.DisorderedAtom-class.html#__repr__,Method Bio.PDB.Chain.Chain.__repr__()=Bio.PDB.Chain.Chain-class.html#__repr__,Method Bio.PDB.FragmentMapper'.Fragment.__repr__()=Bio.PDB.FragmentMapper%27.Fragment-class.html#__repr__,Method Bio.PDB.Model.Model.__repr__()=Bio.PDB.Model.Model-class.html#__repr__,Method Bio.PDB.PDBIO'.Select.__repr__()=Bio.PDB.PDBIO%27.Select-class.html#__repr__,Method Bio.PDB.Polypeptide.Polypeptide.__repr__()=Bio.PDB.Polypeptide.Polypeptide-class.html#__repr__,Method Bio.PDB.Residue.DisorderedResidue.__repr__()=Bio.PDB.Residue.DisorderedResidue-class.html#__repr__,Method Bio.PDB.Residue.Residue.__repr__()=Bio.PDB.Residue.Residue-class.html#__repr__,Method Bio.PDB.Structure.Structure.__repr__()=Bio.PDB.Structure.Structure-class.html#__repr__,Method Bio.PDB.vectors.Vector.__repr__()=Bio.PDB.vectors.Vector-class.html#__repr__,Method Bio.Pathway.Interaction.__repr__()=Bio.Pathway.Interaction-class.html#__repr__,Method Bio.Pathway.Network.__repr__()=Bio.Pathway.Network-class.html#__repr__,Method Bio.Pathway.Reaction.__repr__()=Bio.Pathway.Reaction-class.html#__repr__,Method Bio.Pathway.Rep.Graph.Graph.__repr__()=Bio.Pathway.Rep.Graph.Graph-class.html#__repr__,Method Bio.Pathway.Rep.MultiGraph.MultiGraph.__repr__()=Bio.Pathway.Rep.MultiGraph.MultiGraph-class.html#__repr__,Method Bio.Pathway.System.__repr__()=Bio.Pathway.System-class.html#__repr__,Method Bio.Phylo.BaseTree.BranchColor.__repr__()=Bio.Phylo.BaseTree.BranchColor-class.html#__repr__,Method Bio.Phylo.BaseTree.TreeElement.__repr__()=Bio.Phylo.BaseTree.TreeElement-class.html#__repr__,Method Bio.Phylo.Consensus._BitString.__repr__()=Bio.Phylo.Consensus._BitString-class.html#__repr__,Method Bio.Phylo.TreeConstruction._Matrix.__repr__()=Bio.Phylo.TreeConstruction._Matrix-class.html#__repr__,Method Bio.Restriction.Restriction.Analysis.__repr__()=Bio.Restriction.Restriction.Analysis-class.html#__repr__,Method Bio.Restriction.Restriction.FormattedSeq.__repr__()=Bio.Restriction.Restriction.FormattedSeq-class.html#__repr__,Method Bio.Restriction.Restriction.RestrictionBatch.__repr__()=Bio.Restriction.Restriction.RestrictionBatch-class.html#__repr__,Method Bio.Restriction.Restriction.RestrictionType.__repr__()=Bio.Restriction.Restriction.RestrictionType-class.html#__repr__,Method Bio.SearchIO._model.hit.Hit.__repr__()=Bio.SearchIO._model.hit.Hit-class.html#__repr__,Method Bio.SearchIO._model.hsp.HSP.__repr__()=Bio.SearchIO._model.hsp.HSP-class.html#__repr__,Method Bio.SearchIO._model.hsp.HSPFragment.__repr__()=Bio.SearchIO._model.hsp.HSPFragment-class.html#__repr__,Method Bio.SearchIO._model.query.QueryResult.__repr__()=Bio.SearchIO._model.query.QueryResult-class.html#__repr__,Method Bio.Seq.MutableSeq.__repr__()=Bio.Seq.MutableSeq-class.html#__repr__,Method Bio.Seq.Seq.__repr__()=Bio.Seq.Seq-class.html#__repr__,Method Bio.Seq.UnknownSeq.__repr__()=Bio.Seq.UnknownSeq-class.html#__repr__,Method Bio.SeqFeature.AbstractPosition.__repr__()=Bio.SeqFeature.AbstractPosition-class.html#__repr__,Method Bio.SeqFeature.AfterPosition.__repr__()=Bio.SeqFeature.AfterPosition-class.html#__repr__,Method Bio.SeqFeature.BeforePosition.__repr__()=Bio.SeqFeature.BeforePosition-class.html#__repr__,Method Bio.SeqFeature.BetweenPosition.__repr__()=Bio.SeqFeature.BetweenPosition-class.html#__repr__,Method Bio.SeqFeature.CompoundLocation.__repr__()=Bio.SeqFeature.CompoundLocation-class.html#__repr__,Method Bio.SeqFeature.ExactPosition.__repr__()=Bio.SeqFeature.ExactPosition-class.html#__repr__,Method Bio.SeqFeature.FeatureLocation.__repr__()=Bio.SeqFeature.FeatureLocation-class.html#__repr__,Method Bio.SeqFeature.OneOfPosition.__repr__()=Bio.SeqFeature.OneOfPosition-class.html#__repr__,Method Bio.SeqFeature.PositionGap.__repr__()=Bio.SeqFeature.PositionGap-class.html#__repr__,Method Bio.SeqFeature.Reference.__repr__()=Bio.SeqFeature.Reference-class.html#__repr__,Method Bio.SeqFeature.SeqFeature.__repr__()=Bio.SeqFeature.SeqFeature-class.html#__repr__,Method Bio.SeqFeature.UnknownPosition.__repr__()=Bio.SeqFeature.UnknownPosition-class.html#__repr__,Method Bio.SeqFeature.WithinPosition.__repr__()=Bio.SeqFeature.WithinPosition-class.html#__repr__,Method Bio.SeqRecord.SeqRecord.__repr__()=Bio.SeqRecord.SeqRecord-class.html#__repr__,Method Bio.UniGene.ProtsimLine.__repr__()=Bio.UniGene.ProtsimLine-class.html#__repr__,Method Bio.UniGene.Record.__repr__()=Bio.UniGene.Record-class.html#__repr__,Method Bio.UniGene.STSLine.__repr__()=Bio.UniGene.STSLine-class.html#__repr__,Method Bio.UniGene.SequenceLine.__repr__()=Bio.UniGene.SequenceLine-class.html#__repr__,Method Bio.Wise.psw.AlignmentColumn.__repr__()=Bio.Wise.psw.AlignmentColumn-class.html#__repr__,Method Bio.codonalign.codonalphabet.CodonAlphabet.__repr__()=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#__repr__,Method Bio.phenotype.phen_micro.PlateRecord.__repr__()=Bio.phenotype.phen_micro.PlateRecord-class.html#__repr__,Method Bio.phenotype.phen_micro.WellRecord.__repr__()=Bio.phenotype.phen_micro.WellRecord-class.html#__repr__,Method BioSQL.BioSeqDatabase.BioSeqDatabase.__repr__()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#__repr__,Method BioSQL.BioSeqDatabase.DBServer.__repr__()=BioSQL.BioSeqDatabase.DBServer-class.html#__repr__"><a title="Bio.Align.MultipleSeqAlignment.__repr__
Bio.AlignIO.MafIO.MafIndex.__repr__
Bio.Alphabet.Alphabet.__repr__
Bio.Alphabet.AlphabetEncoder.__repr__
Bio.Application.AbstractCommandline.__repr__
Bio.Application.ApplicationError.__repr__
Bio.Crystal.Crystal.__repr__
Bio.Crystal.Hetero.__repr__
Bio.Entrez.Parser.DictionaryElement.__repr__
Bio.Entrez.Parser.IntegerElement.__repr__
Bio.Entrez.Parser.ListElement.__repr__
Bio.Entrez.Parser.StringElement.__repr__
Bio.Entrez.Parser.UnicodeElement.__repr__
Bio.ExPASy.Enzyme.Record.__repr__
Bio.ExPASy.cellosaurus.Record.__repr__
Bio.FSSP.FSSPSumRec.__repr__
Bio.FSSP.PosAlign.__repr__
Bio.FSSP.fssp_rec.fff_rec.__repr__
Bio.File._IndexedSeqFileDict.__repr__
Bio.File._SQLiteManySeqFilesDict.__repr__
Bio.GenBank.Record.Feature.__repr__
Bio.GenBank.Record.Qualifier.__repr__
Bio.PDB.Atom.Atom.__repr__
Bio.PDB.Atom.DisorderedAtom.__repr__
Bio.PDB.Chain.Chain.__repr__
Bio.PDB.FragmentMapper'.Fragment.__repr__
Bio.PDB.Model.Model.__repr__
Bio.PDB.PDBIO'.Select.__repr__
Bio.PDB.Polypeptide.Polypeptide.__repr__
Bio.PDB.Residue.DisorderedResidue.__repr__
Bio.PDB.Residue.Residue.__repr__
Bio.PDB.Structure.Structure.__repr__
Bio.PDB.vectors.Vector.__repr__
Bio.Pathway.Interaction.__repr__
Bio.Pathway.Network.__repr__
Bio.Pathway.Reaction.__repr__
Bio.Pathway.Rep.Graph.Graph.__repr__
Bio.Pathway.Rep.MultiGraph.MultiGraph.__repr__
Bio.Pathway.System.__repr__
Bio.Phylo.BaseTree.BranchColor.__repr__
Bio.Phylo.BaseTree.TreeElement.__repr__
Bio.Phylo.Consensus._BitString.__repr__
Bio.Phylo.TreeConstruction._Matrix.__repr__
Bio.Restriction.Restriction.Analysis.__repr__
Bio.Restriction.Restriction.FormattedSeq.__repr__
Bio.Restriction.Restriction.RestrictionBatch.__repr__
Bio.Restriction.Restriction.RestrictionType.__repr__
Bio.SearchIO._model.hit.Hit.__repr__
Bio.SearchIO._model.hsp.HSP.__repr__
Bio.SearchIO._model.hsp.HSPFragment.__repr__
Bio.SearchIO._model.query.QueryResult.__repr__
Bio.Seq.MutableSeq.__repr__
Bio.Seq.Seq.__repr__
Bio.Seq.UnknownSeq.__repr__
Bio.SeqFeature.AbstractPosition.__repr__
Bio.SeqFeature.AfterPosition.__repr__
Bio.SeqFeature.BeforePosition.__repr__
Bio.SeqFeature.BetweenPosition.__repr__
Bio.SeqFeature.CompoundLocation.__repr__
Bio.SeqFeature.ExactPosition.__repr__
Bio.SeqFeature.FeatureLocation.__repr__
Bio.SeqFeature.OneOfPosition.__repr__
Bio.SeqFeature.PositionGap.__repr__
Bio.SeqFeature.Reference.__repr__
Bio.SeqFeature.SeqFeature.__repr__
Bio.SeqFeature.UnknownPosition.__repr__
Bio.SeqFeature.WithinPosition.__repr__
Bio.SeqRecord.SeqRecord.__repr__
Bio.UniGene.ProtsimLine.__repr__
Bio.UniGene.Record.__repr__
Bio.UniGene.STSLine.__repr__
Bio.UniGene.SequenceLine.__repr__
Bio.Wise.psw.AlignmentColumn.__repr__
Bio.codonalign.codonalphabet.CodonAlphabet.__repr__
Bio.phenotype.phen_micro.PlateRecord.__repr__
Bio.phenotype.phen_micro.WellRecord.__repr__
BioSQL.BioSeqDatabase.BioSeqDatabase.__repr__
BioSQL.BioSeqDatabase.DBServer.__repr__" class="py-name" href="#" onclick="return doclink('link-68', '__repr__', 'link-68');">__repr__</a></tt> </tt>
</div><a name="L256"></a><tt class="py-lineno"> 256</tt>  <tt class="py-line"> </tt>
<a name="TreeMixin"></a><div id="TreeMixin-def"><a name="L257"></a><tt class="py-lineno"> 257</tt>  <tt class="py-line"> </tt>
<a name="L258"></a><tt class="py-lineno"> 258</tt> <a class="py-toggle" href="#" id="TreeMixin-toggle" onclick="return toggle('TreeMixin');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.TreeMixin-class.html">TreeMixin</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="TreeMixin-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="TreeMixin-expanded"><a name="L259"></a><tt class="py-lineno"> 259</tt>  <tt class="py-line">    <tt class="py-docstring">"""Methods for Tree- and Clade-based classes.</tt> </tt>
<a name="L260"></a><tt class="py-lineno"> 260</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L261"></a><tt class="py-lineno"> 261</tt>  <tt class="py-line"><tt class="py-docstring">    This lets `Tree` and `Clade` support the same traversal and searching</tt> </tt>
<a name="L262"></a><tt class="py-lineno"> 262</tt>  <tt class="py-line"><tt class="py-docstring">    operations without requiring Clade to inherit from Tree, so Clade isn't</tt> </tt>
<a name="L263"></a><tt class="py-lineno"> 263</tt>  <tt class="py-line"><tt class="py-docstring">    required to have all of Tree's attributes -- just ``root`` (a Clade</tt> </tt>
<a name="L264"></a><tt class="py-lineno"> 264</tt>  <tt class="py-line"><tt class="py-docstring">    instance) and ``is_terminal``.</tt> </tt>
<a name="L265"></a><tt class="py-lineno"> 265</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L266"></a><tt class="py-lineno"> 266</tt>  <tt class="py-line"> </tt>
<a name="L267"></a><tt class="py-lineno"> 267</tt>  <tt class="py-line">    <tt class="py-comment"># Traversal methods</tt> </tt>
<a name="L268"></a><tt class="py-lineno"> 268</tt>  <tt class="py-line"> </tt>
<a name="TreeMixin._filter_search"></a><div id="TreeMixin._filter_search-def"><a name="L269"></a><tt class="py-lineno"> 269</tt> <a class="py-toggle" href="#" id="TreeMixin._filter_search-toggle" onclick="return toggle('TreeMixin._filter_search');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.TreeMixin-class.html#_filter_search">_filter_search</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">filter_func</tt><tt class="py-op">,</tt> <tt class="py-param">order</tt><tt class="py-op">,</tt> <tt class="py-param">follow_attrs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="TreeMixin._filter_search-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="TreeMixin._filter_search-expanded"><a name="L270"></a><tt class="py-lineno"> 270</tt>  <tt class="py-line">        <tt class="py-docstring">"""Perform a BFS or DFS traversal through all elements in this tree (PRIVATE).</tt> </tt>
<a name="L271"></a><tt class="py-lineno"> 271</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L272"></a><tt class="py-lineno"> 272</tt>  <tt class="py-line"><tt class="py-docstring">        :returns: generator of all elements for which `filter_func` is True.</tt> </tt>
<a name="L273"></a><tt class="py-lineno"> 273</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L274"></a><tt class="py-lineno"> 274</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L275"></a><tt class="py-lineno"> 275</tt>  <tt class="py-line">        <tt class="py-name">order_opts</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'preorder'</tt><tt class="py-op">:</tt> <tt id="link-69" class="py-name" targets="Function Bio.Phylo.BaseTree._preorder_traverse()=Bio.Phylo.BaseTree-module.html#_preorder_traverse"><a title="Bio.Phylo.BaseTree._preorder_traverse" class="py-name" href="#" onclick="return doclink('link-69', '_preorder_traverse', 'link-69');">_preorder_traverse</a></tt><tt class="py-op">,</tt> </tt>
<a name="L276"></a><tt class="py-lineno"> 276</tt>  <tt class="py-line">                      <tt class="py-string">'postorder'</tt><tt class="py-op">:</tt> <tt id="link-70" class="py-name" targets="Function Bio.Phylo.BaseTree._postorder_traverse()=Bio.Phylo.BaseTree-module.html#_postorder_traverse"><a title="Bio.Phylo.BaseTree._postorder_traverse" class="py-name" href="#" onclick="return doclink('link-70', '_postorder_traverse', 'link-70');">_postorder_traverse</a></tt><tt class="py-op">,</tt> </tt>
<a name="L277"></a><tt class="py-lineno"> 277</tt>  <tt class="py-line">                      <tt class="py-string">'level'</tt><tt class="py-op">:</tt> <tt id="link-71" class="py-name" targets="Function Bio.Phylo.BaseTree._level_traverse()=Bio.Phylo.BaseTree-module.html#_level_traverse"><a title="Bio.Phylo.BaseTree._level_traverse" class="py-name" href="#" onclick="return doclink('link-71', '_level_traverse', 'link-71');">_level_traverse</a></tt><tt class="py-op">}</tt> </tt>
<a name="L278"></a><tt class="py-lineno"> 278</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L279"></a><tt class="py-lineno"> 279</tt>  <tt class="py-line">            <tt class="py-name">order_func</tt> <tt class="py-op">=</tt> <tt class="py-name">order_opts</tt><tt class="py-op">[</tt><tt class="py-name">order</tt><tt class="py-op">]</tt> </tt>
<a name="L280"></a><tt class="py-lineno"> 280</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L281"></a><tt class="py-lineno"> 281</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid order '%s'; must be one of: %s"</tt> </tt>
<a name="L282"></a><tt class="py-lineno"> 282</tt>  <tt class="py-line">                             <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">order</tt><tt class="py-op">,</tt> <tt class="py-name">tuple</tt><tt class="py-op">(</tt><tt class="py-name">order_opts</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L283"></a><tt class="py-lineno"> 283</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">follow_attrs</tt><tt class="py-op">:</tt> </tt>
<a name="L284"></a><tt class="py-lineno"> 284</tt>  <tt class="py-line">            <tt class="py-name">get_children</tt> <tt class="py-op">=</tt> <tt id="link-72" class="py-name" targets="Function Bio.Phylo.BaseTree._sorted_attrs()=Bio.Phylo.BaseTree-module.html#_sorted_attrs"><a title="Bio.Phylo.BaseTree._sorted_attrs" class="py-name" href="#" onclick="return doclink('link-72', '_sorted_attrs', 'link-72');">_sorted_attrs</a></tt> </tt>
<a name="L285"></a><tt class="py-lineno"> 285</tt>  <tt class="py-line">            <tt id="link-73" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-73', 'root', 'link-7');">root</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt> </tt>
<a name="L286"></a><tt class="py-lineno"> 286</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L287"></a><tt class="py-lineno"> 287</tt>  <tt class="py-line">            <tt class="py-name">get_children</tt> <tt class="py-op">=</tt> <tt class="py-keyword">lambda</tt> <tt class="py-name">elem</tt><tt class="py-op">:</tt> <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt> </tt>
<a name="L288"></a><tt class="py-lineno"> 288</tt>  <tt class="py-line">            <tt id="link-74" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-74', 'root', 'link-7');">root</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-75" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-75', 'root', 'link-7');">root</a></tt> </tt>
<a name="L289"></a><tt class="py-lineno"> 289</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-76" class="py-name"><a title="Bio.FSSP.FSSPTools.filter
Bio.SearchIO._model.hit.Hit.filter" class="py-name" href="#" onclick="return doclink('link-76', 'filter', 'link-2');">filter</a></tt><tt class="py-op">(</tt><tt class="py-name">filter_func</tt><tt class="py-op">,</tt> <tt class="py-name">order_func</tt><tt class="py-op">(</tt><tt id="link-77" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-77', 'root', 'link-7');">root</a></tt><tt class="py-op">,</tt> <tt class="py-name">get_children</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L290"></a><tt class="py-lineno"> 290</tt>  <tt class="py-line"> </tt>
<a name="TreeMixin.find_any"></a><div id="TreeMixin.find_any-def"><a name="L291"></a><tt class="py-lineno"> 291</tt> <a class="py-toggle" href="#" id="TreeMixin.find_any-toggle" onclick="return toggle('TreeMixin.find_any');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.TreeMixin-class.html#find_any">find_any</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-op">*</tt><tt class="py-param">args</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="TreeMixin.find_any-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="TreeMixin.find_any-expanded"><a name="L292"></a><tt class="py-lineno"> 292</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the first element found by find_elements(), or None.</tt> </tt>
<a name="L293"></a><tt class="py-lineno"> 293</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L294"></a><tt class="py-lineno"> 294</tt>  <tt class="py-line"><tt class="py-docstring">        This is also useful for checking whether any matching element exists in</tt> </tt>
<a name="L295"></a><tt class="py-lineno"> 295</tt>  <tt class="py-line"><tt class="py-docstring">        the tree, and can be used in a conditional expression.</tt> </tt>
<a name="L296"></a><tt class="py-lineno"> 296</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L297"></a><tt class="py-lineno"> 297</tt>  <tt class="py-line">        <tt id="link-78" class="py-name" targets="Variable Bio.SearchIO._model.query.QueryResult.hits=Bio.SearchIO._model.query.QueryResult-class.html#hits"><a title="Bio.SearchIO._model.query.QueryResult.hits" class="py-name" href="#" onclick="return doclink('link-78', 'hits', 'link-78');">hits</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-79" class="py-name" targets="Method Bio.Phylo.BaseTree.TreeMixin.find_elements()=Bio.Phylo.BaseTree.TreeMixin-class.html#find_elements"><a title="Bio.Phylo.BaseTree.TreeMixin.find_elements" class="py-name" href="#" onclick="return doclink('link-79', 'find_elements', 'link-79');">find_elements</a></tt><tt class="py-op">(</tt><tt class="py-op">*</tt><tt class="py-name">args</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> </tt>
<a name="L298"></a><tt class="py-lineno"> 298</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L299"></a><tt class="py-lineno"> 299</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-80" class="py-name" targets="Method Bio.AlignIO.Interfaces.AlignmentIterator.next()=Bio.AlignIO.Interfaces.AlignmentIterator-class.html#next,Method Bio.Blast.ParseBlastTable.BlastTableReader.next()=Bio.Blast.ParseBlastTable.BlastTableReader-class.html#next,Method Bio.File.UndoHandle.next()=Bio.File.UndoHandle-class.html#next,Method Bio.GenBank.Iterator.next()=Bio.GenBank.Iterator-class.html#next,Method Bio.Nexus.Nexus.CharBuffer.next()=Bio.Nexus.Nexus.CharBuffer-class.html#next,Method Bio.Nexus.StandardData.StandardData.next()=Bio.Nexus.StandardData.StandardData-class.html#next,Method Bio.PopGen.GenePop.Controller._FileIterator.next()=Bio.PopGen.GenePop.Controller._FileIterator-class.html#next,Method Bio.SearchIO._legacy.NCBIStandalone.Iterator.next()=Bio.SearchIO._legacy.NCBIStandalone.Iterator-class.html#next,Method Bio.SeqIO.Interfaces.SequenceIterator.next()=Bio.SeqIO.Interfaces.SequenceIterator-class.html#next,Method Bio.bgzf.BgzfReader.next()=Bio.bgzf.BgzfReader-class.html#next"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.Nexus.StandardData.StandardData.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SearchIO._legacy.NCBIStandalone.Iterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-80', 'next', 'link-80');">next</a></tt><tt class="py-op">(</tt><tt id="link-81" class="py-name"><a title="Bio.SearchIO._model.query.QueryResult.hits" class="py-name" href="#" onclick="return doclink('link-81', 'hits', 'link-78');">hits</a></tt><tt class="py-op">)</tt> </tt>
<a name="L300"></a><tt class="py-lineno"> 300</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L301"></a><tt class="py-lineno"> 301</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L302"></a><tt class="py-lineno"> 302</tt>  <tt class="py-line"> </tt>
<a name="TreeMixin.find_elements"></a><div id="TreeMixin.find_elements-def"><a name="L303"></a><tt class="py-lineno"> 303</tt> <a class="py-toggle" href="#" id="TreeMixin.find_elements-toggle" onclick="return toggle('TreeMixin.find_elements');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.TreeMixin-class.html#find_elements">find_elements</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">target</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">terminal</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">order</tt><tt class="py-op">=</tt><tt class="py-string">'preorder'</tt><tt class="py-op">,</tt> </tt>
<a name="L304"></a><tt class="py-lineno"> 304</tt>  <tt class="py-line">                      <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="TreeMixin.find_elements-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="TreeMixin.find_elements-expanded"><a name="L305"></a><tt class="py-lineno"> 305</tt>  <tt class="py-line">        <tt class="py-docstring">"""Find all tree elements matching the given attributes.</tt> </tt>
<a name="L306"></a><tt class="py-lineno"> 306</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L307"></a><tt class="py-lineno"> 307</tt>  <tt class="py-line"><tt class="py-docstring">        The arbitrary keyword arguments indicate the attribute name of the</tt> </tt>
<a name="L308"></a><tt class="py-lineno"> 308</tt>  <tt class="py-line"><tt class="py-docstring">        sub-element and the value to match: string, integer or boolean. Strings</tt> </tt>
<a name="L309"></a><tt class="py-lineno"> 309</tt>  <tt class="py-line"><tt class="py-docstring">        are evaluated as regular expression matches; integers are compared</tt> </tt>
<a name="L310"></a><tt class="py-lineno"> 310</tt>  <tt class="py-line"><tt class="py-docstring">        directly for equality, and booleans evaluate the attribute's truth value</tt> </tt>
<a name="L311"></a><tt class="py-lineno"> 311</tt>  <tt class="py-line"><tt class="py-docstring">        (True or False) before comparing. To handle nonzero floats, search with</tt> </tt>
<a name="L312"></a><tt class="py-lineno"> 312</tt>  <tt class="py-line"><tt class="py-docstring">        a boolean argument, then filter the result manually.</tt> </tt>
<a name="L313"></a><tt class="py-lineno"> 313</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L314"></a><tt class="py-lineno"> 314</tt>  <tt class="py-line"><tt class="py-docstring">        If no keyword arguments are given, then just the class type is used for</tt> </tt>
<a name="L315"></a><tt class="py-lineno"> 315</tt>  <tt class="py-line"><tt class="py-docstring">        matching.</tt> </tt>
<a name="L316"></a><tt class="py-lineno"> 316</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L317"></a><tt class="py-lineno"> 317</tt>  <tt class="py-line"><tt class="py-docstring">        The result is an iterable through all matching objects, by depth-first</tt> </tt>
<a name="L318"></a><tt class="py-lineno"> 318</tt>  <tt class="py-line"><tt class="py-docstring">        search. (Not necessarily the same order as the elements appear in the</tt> </tt>
<a name="L319"></a><tt class="py-lineno"> 319</tt>  <tt class="py-line"><tt class="py-docstring">        source file!)</tt> </tt>
<a name="L320"></a><tt class="py-lineno"> 320</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L321"></a><tt class="py-lineno"> 321</tt>  <tt class="py-line"><tt class="py-docstring">        :Parameters:</tt> </tt>
<a name="L322"></a><tt class="py-lineno"> 322</tt>  <tt class="py-line"><tt class="py-docstring">            target : TreeElement instance, type, dict, or callable</tt> </tt>
<a name="L323"></a><tt class="py-lineno"> 323</tt>  <tt class="py-line"><tt class="py-docstring">                Specifies the characteristics to search for. (The default,</tt> </tt>
<a name="L324"></a><tt class="py-lineno"> 324</tt>  <tt class="py-line"><tt class="py-docstring">                TreeElement, matches any standard Bio.Phylo type.)</tt> </tt>
<a name="L325"></a><tt class="py-lineno"> 325</tt>  <tt class="py-line"><tt class="py-docstring">            terminal : bool</tt> </tt>
<a name="L326"></a><tt class="py-lineno"> 326</tt>  <tt class="py-line"><tt class="py-docstring">                A boolean value to select for or against terminal nodes (a.k.a.</tt> </tt>
<a name="L327"></a><tt class="py-lineno"> 327</tt>  <tt class="py-line"><tt class="py-docstring">                leaf nodes). True searches for only terminal nodes, False</tt> </tt>
<a name="L328"></a><tt class="py-lineno"> 328</tt>  <tt class="py-line"><tt class="py-docstring">                excludes terminal nodes, and the default, None, searches both</tt> </tt>
<a name="L329"></a><tt class="py-lineno"> 329</tt>  <tt class="py-line"><tt class="py-docstring">                terminal and non-terminal nodes, as well as any tree elements</tt> </tt>
<a name="L330"></a><tt class="py-lineno"> 330</tt>  <tt class="py-line"><tt class="py-docstring">                lacking the ``is_terminal`` method.</tt> </tt>
<a name="L331"></a><tt class="py-lineno"> 331</tt>  <tt class="py-line"><tt class="py-docstring">            order : {'preorder', 'postorder', 'level'}</tt> </tt>
<a name="L332"></a><tt class="py-lineno"> 332</tt>  <tt class="py-line"><tt class="py-docstring">                Tree traversal order: 'preorder' (default) is depth-first</tt> </tt>
<a name="L333"></a><tt class="py-lineno"> 333</tt>  <tt class="py-line"><tt class="py-docstring">                search, 'postorder' is DFS with child nodes preceding parents,</tt> </tt>
<a name="L334"></a><tt class="py-lineno"> 334</tt>  <tt class="py-line"><tt class="py-docstring">                and 'level' is breadth-first search.</tt> </tt>
<a name="L335"></a><tt class="py-lineno"> 335</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L336"></a><tt class="py-lineno"> 336</tt>  <tt class="py-line"><tt class="py-docstring">        Examples</tt> </tt>
<a name="L337"></a><tt class="py-lineno"> 337</tt>  <tt class="py-line"><tt class="py-docstring">        --------</tt> </tt>
<a name="L338"></a><tt class="py-lineno"> 338</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from Bio import Phylo</tt> </tt>
<a name="L339"></a><tt class="py-lineno"> 339</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; phx = Phylo.PhyloXMLIO.read('PhyloXML/phyloxml_examples.xml')</tt> </tt>
<a name="L340"></a><tt class="py-lineno"> 340</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; matches = phx.phylogenies[5].find_elements(code='OCTVU')</tt> </tt>
<a name="L341"></a><tt class="py-lineno"> 341</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; next(matches)</tt> </tt>
<a name="L342"></a><tt class="py-lineno"> 342</tt>  <tt class="py-line"><tt class="py-docstring">        Taxonomy(code='OCTVU', scientific_name='Octopus vulgaris')</tt> </tt>
<a name="L343"></a><tt class="py-lineno"> 343</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L344"></a><tt class="py-lineno"> 344</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L345"></a><tt class="py-lineno"> 345</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">terminal</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L346"></a><tt class="py-lineno"> 346</tt>  <tt class="py-line">            <tt class="py-name">kwargs</tt><tt class="py-op">[</tt><tt class="py-string">'terminal'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">terminal</tt> </tt>
<a name="L347"></a><tt class="py-lineno"> 347</tt>  <tt class="py-line">        <tt class="py-name">is_matching_elem</tt> <tt class="py-op">=</tt> <tt id="link-82" class="py-name" targets="Function Bio.Phylo.BaseTree._combine_matchers()=Bio.Phylo.BaseTree-module.html#_combine_matchers"><a title="Bio.Phylo.BaseTree._combine_matchers" class="py-name" href="#" onclick="return doclink('link-82', '_combine_matchers', 'link-82');">_combine_matchers</a></tt><tt class="py-op">(</tt><tt class="py-name">target</tt><tt class="py-op">,</tt> <tt class="py-name">kwargs</tt><tt class="py-op">,</tt> <tt class="py-name">False</tt><tt class="py-op">)</tt> </tt>
<a name="L348"></a><tt class="py-lineno"> 348</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-83" class="py-name" targets="Method Bio.Phylo.BaseTree.TreeMixin._filter_search()=Bio.Phylo.BaseTree.TreeMixin-class.html#_filter_search"><a title="Bio.Phylo.BaseTree.TreeMixin._filter_search" class="py-name" href="#" onclick="return doclink('link-83', '_filter_search', 'link-83');">_filter_search</a></tt><tt class="py-op">(</tt><tt class="py-name">is_matching_elem</tt><tt class="py-op">,</tt> <tt class="py-name">order</tt><tt class="py-op">,</tt> <tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
</div><a name="L349"></a><tt class="py-lineno"> 349</tt>  <tt class="py-line"> </tt>
<a name="TreeMixin.find_clades"></a><div id="TreeMixin.find_clades-def"><a name="L350"></a><tt class="py-lineno"> 350</tt> <a class="py-toggle" href="#" id="TreeMixin.find_clades-toggle" onclick="return toggle('TreeMixin.find_clades');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.TreeMixin-class.html#find_clades">find_clades</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">target</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">terminal</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">order</tt><tt class="py-op">=</tt><tt class="py-string">'preorder'</tt><tt class="py-op">,</tt> </tt>
<a name="L351"></a><tt class="py-lineno"> 351</tt>  <tt class="py-line">                    <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="TreeMixin.find_clades-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="TreeMixin.find_clades-expanded"><a name="L352"></a><tt class="py-lineno"> 352</tt>  <tt class="py-line">        <tt class="py-docstring">"""Find each clade containing a matching element.</tt> </tt>
<a name="L353"></a><tt class="py-lineno"> 353</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L354"></a><tt class="py-lineno"> 354</tt>  <tt class="py-line"><tt class="py-docstring">        That is, find each element as with find_elements(), but return the</tt> </tt>
<a name="L355"></a><tt class="py-lineno"> 355</tt>  <tt class="py-line"><tt class="py-docstring">        corresponding clade object. (This is usually what you want.)</tt> </tt>
<a name="L356"></a><tt class="py-lineno"> 356</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L357"></a><tt class="py-lineno"> 357</tt>  <tt class="py-line"><tt class="py-docstring">        :returns: an iterable through all matching objects, searching</tt> </tt>
<a name="L358"></a><tt class="py-lineno"> 358</tt>  <tt class="py-line"><tt class="py-docstring">            depth-first (preorder) by default.</tt> </tt>
<a name="L359"></a><tt class="py-lineno"> 359</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L360"></a><tt class="py-lineno"> 360</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L361"></a><tt class="py-lineno"> 361</tt>  <tt class="py-line">        <tt class="py-keyword">def</tt> <tt class="py-def-name">match_attrs</tt><tt class="py-op">(</tt><tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L362"></a><tt class="py-lineno"> 362</tt>  <tt class="py-line">            <tt class="py-name">orig_clades</tt> <tt class="py-op">=</tt> <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">__dict__</tt><tt class="py-op">.</tt><tt id="link-84" class="py-name"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop" class="py-name" href="#" onclick="return doclink('link-84', 'pop', 'link-31');">pop</a></tt><tt class="py-op">(</tt><tt class="py-string">'clades'</tt><tt class="py-op">)</tt> </tt>
<a name="L363"></a><tt class="py-lineno"> 363</tt>  <tt class="py-line">            <tt class="py-name">found</tt> <tt class="py-op">=</tt> <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-85" class="py-name" targets="Method Bio.Phylo.BaseTree.TreeMixin.find_any()=Bio.Phylo.BaseTree.TreeMixin-class.html#find_any"><a title="Bio.Phylo.BaseTree.TreeMixin.find_any" class="py-name" href="#" onclick="return doclink('link-85', 'find_any', 'link-85');">find_any</a></tt><tt class="py-op">(</tt><tt class="py-name">target</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> </tt>
<a name="L364"></a><tt class="py-lineno"> 364</tt>  <tt class="py-line">            <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt> <tt class="py-op">=</tt> <tt class="py-name">orig_clades</tt> </tt>
<a name="L365"></a><tt class="py-lineno"> 365</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-name">found</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
</div><a name="L366"></a><tt class="py-lineno"> 366</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">terminal</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L367"></a><tt class="py-lineno"> 367</tt>  <tt class="py-line">            <tt class="py-name">is_matching_elem</tt> <tt class="py-op">=</tt> <tt class="py-name">match_attrs</tt> </tt>
<a name="L368"></a><tt class="py-lineno"> 368</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L369"></a><tt class="py-lineno"> 369</tt>  <tt class="py-line">            <tt class="py-keyword">def</tt> <tt class="py-def-name">is_matching_elem</tt><tt class="py-op">(</tt><tt class="py-param">elem</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L370"></a><tt class="py-lineno"> 370</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">.</tt><tt id="link-86" class="py-name"><a title="Bio.Nexus.Trees.Tree.is_terminal
Bio.Phylo.BaseTree.Clade.is_terminal
Bio.Phylo.BaseTree.Tree.is_terminal" class="py-name" href="#" onclick="return doclink('link-86', 'is_terminal', 'link-34');">is_terminal</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">terminal</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> </tt>
<a name="L371"></a><tt class="py-lineno"> 371</tt>  <tt class="py-line">                        <tt class="py-name">match_attrs</tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L372"></a><tt class="py-lineno"> 372</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-87" class="py-name"><a title="Bio.Phylo.BaseTree.TreeMixin._filter_search" class="py-name" href="#" onclick="return doclink('link-87', '_filter_search', 'link-83');">_filter_search</a></tt><tt class="py-op">(</tt><tt class="py-name">is_matching_elem</tt><tt class="py-op">,</tt> <tt class="py-name">order</tt><tt class="py-op">,</tt> <tt class="py-name">False</tt><tt class="py-op">)</tt> </tt>
</div><a name="L373"></a><tt class="py-lineno"> 373</tt>  <tt class="py-line"> </tt>
<a name="TreeMixin.get_path"></a><div id="TreeMixin.get_path-def"><a name="L374"></a><tt class="py-lineno"> 374</tt> <a class="py-toggle" href="#" id="TreeMixin.get_path-toggle" onclick="return toggle('TreeMixin.get_path');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.TreeMixin-class.html#get_path">get_path</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">target</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="TreeMixin.get_path-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="TreeMixin.get_path-expanded"><a name="L375"></a><tt class="py-lineno"> 375</tt>  <tt class="py-line">        <tt class="py-docstring">"""List the clades directly between this root and the given target.</tt> </tt>
<a name="L376"></a><tt class="py-lineno"> 376</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L377"></a><tt class="py-lineno"> 377</tt>  <tt class="py-line"><tt class="py-docstring">        :returns: list of all clade objects along this path, ending with the</tt> </tt>
<a name="L378"></a><tt class="py-lineno"> 378</tt>  <tt class="py-line"><tt class="py-docstring">            given target, but excluding the root clade.</tt> </tt>
<a name="L379"></a><tt class="py-lineno"> 379</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L380"></a><tt class="py-lineno"> 380</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L381"></a><tt class="py-lineno"> 381</tt>  <tt class="py-line">        <tt class="py-comment"># Only one path will work -- ignore weights and visits</tt> </tt>
<a name="L382"></a><tt class="py-lineno"> 382</tt>  <tt class="py-line">        <tt class="py-name">path</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L383"></a><tt class="py-lineno"> 383</tt>  <tt class="py-line">        <tt id="link-88" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-88', 'match', 'link-19');">match</a></tt> <tt class="py-op">=</tt> <tt id="link-89" class="py-name"><a title="Bio.Phylo.BaseTree._combine_matchers" class="py-name" href="#" onclick="return doclink('link-89', '_combine_matchers', 'link-82');">_combine_matchers</a></tt><tt class="py-op">(</tt><tt class="py-name">target</tt><tt class="py-op">,</tt> <tt class="py-name">kwargs</tt><tt class="py-op">,</tt> <tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L384"></a><tt class="py-lineno"> 384</tt>  <tt class="py-line"> </tt>
<a name="L385"></a><tt class="py-lineno"> 385</tt>  <tt class="py-line">        <tt class="py-keyword">def</tt> <tt class="py-def-name">check_in_path</tt><tt class="py-op">(</tt><tt class="py-param">v</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L386"></a><tt class="py-lineno"> 386</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-90" class="py-name"><a title="Bio.triefind.match" class="py-name" href="#" onclick="return doclink('link-90', 'match', 'link-19');">match</a></tt><tt class="py-op">(</tt><tt class="py-name">v</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L387"></a><tt class="py-lineno"> 387</tt>  <tt class="py-line">                <tt class="py-name">path</tt><tt class="py-op">.</tt><tt id="link-91" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-91', 'append', 'link-13');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">v</tt><tt class="py-op">)</tt> </tt>
<a name="L388"></a><tt class="py-lineno"> 388</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">True</tt> </tt>
<a name="L389"></a><tt class="py-lineno"> 389</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">v</tt><tt class="py-op">.</tt><tt id="link-92" class="py-name"><a title="Bio.Nexus.Trees.Tree.is_terminal
Bio.Phylo.BaseTree.Clade.is_terminal
Bio.Phylo.BaseTree.Tree.is_terminal" class="py-name" href="#" onclick="return doclink('link-92', 'is_terminal', 'link-34');">is_terminal</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L390"></a><tt class="py-lineno"> 390</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">False</tt> </tt>
<a name="L391"></a><tt class="py-lineno"> 391</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">child</tt> <tt class="py-keyword">in</tt> <tt class="py-name">v</tt><tt class="py-op">:</tt> </tt>
<a name="L392"></a><tt class="py-lineno"> 392</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">check_in_path</tt><tt class="py-op">(</tt><tt class="py-name">child</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L393"></a><tt class="py-lineno"> 393</tt>  <tt class="py-line">                    <tt class="py-name">path</tt><tt class="py-op">.</tt><tt id="link-93" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-93', 'append', 'link-13');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">v</tt><tt class="py-op">)</tt> </tt>
<a name="L394"></a><tt class="py-lineno"> 394</tt>  <tt class="py-line">                    <tt class="py-keyword">return</tt> <tt class="py-name">True</tt> </tt>
<a name="L395"></a><tt class="py-lineno"> 395</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">False</tt> </tt>
</div><a name="L396"></a><tt class="py-lineno"> 396</tt>  <tt class="py-line"> </tt>
<a name="L397"></a><tt class="py-lineno"> 397</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">check_in_path</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-94" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-94', 'root', 'link-7');">root</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L398"></a><tt class="py-lineno"> 398</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
<a name="L399"></a><tt class="py-lineno"> 399</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">path</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">2</tt><tt class="py-op">:</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
</div><a name="L400"></a><tt class="py-lineno"> 400</tt>  <tt class="py-line"> </tt>
<a name="TreeMixin.get_nonterminals"></a><div id="TreeMixin.get_nonterminals-def"><a name="L401"></a><tt class="py-lineno"> 401</tt> <a class="py-toggle" href="#" id="TreeMixin.get_nonterminals-toggle" onclick="return toggle('TreeMixin.get_nonterminals');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.TreeMixin-class.html#get_nonterminals">get_nonterminals</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">order</tt><tt class="py-op">=</tt><tt class="py-string">'preorder'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="TreeMixin.get_nonterminals-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="TreeMixin.get_nonterminals-expanded"><a name="L402"></a><tt class="py-lineno"> 402</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get a list of all of this tree's nonterminal (internal) nodes."""</tt> </tt>
<a name="L403"></a><tt class="py-lineno"> 403</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-95" class="py-name" targets="Method Bio.Phylo.BaseTree.TreeMixin.find_clades()=Bio.Phylo.BaseTree.TreeMixin-class.html#find_clades"><a title="Bio.Phylo.BaseTree.TreeMixin.find_clades" class="py-name" href="#" onclick="return doclink('link-95', 'find_clades', 'link-95');">find_clades</a></tt><tt class="py-op">(</tt><tt class="py-name">terminal</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt> <tt class="py-name">order</tt><tt class="py-op">=</tt><tt class="py-name">order</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L404"></a><tt class="py-lineno"> 404</tt>  <tt class="py-line"> </tt>
<a name="TreeMixin.get_terminals"></a><div id="TreeMixin.get_terminals-def"><a name="L405"></a><tt class="py-lineno"> 405</tt> <a class="py-toggle" href="#" id="TreeMixin.get_terminals-toggle" onclick="return toggle('TreeMixin.get_terminals');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.TreeMixin-class.html#get_terminals">get_terminals</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">order</tt><tt class="py-op">=</tt><tt class="py-string">'preorder'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="TreeMixin.get_terminals-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="TreeMixin.get_terminals-expanded"><a name="L406"></a><tt class="py-lineno"> 406</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get a list of all of this tree's terminal (leaf) nodes."""</tt> </tt>
<a name="L407"></a><tt class="py-lineno"> 407</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-96" class="py-name"><a title="Bio.Phylo.BaseTree.TreeMixin.find_clades" class="py-name" href="#" onclick="return doclink('link-96', 'find_clades', 'link-95');">find_clades</a></tt><tt class="py-op">(</tt><tt class="py-name">terminal</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">,</tt> <tt class="py-name">order</tt><tt class="py-op">=</tt><tt class="py-name">order</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L408"></a><tt class="py-lineno"> 408</tt>  <tt class="py-line"> </tt>
<a name="TreeMixin.trace"></a><div id="TreeMixin.trace-def"><a name="L409"></a><tt class="py-lineno"> 409</tt> <a class="py-toggle" href="#" id="TreeMixin.trace-toggle" onclick="return toggle('TreeMixin.trace');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.TreeMixin-class.html#trace">trace</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">,</tt> <tt class="py-param">finish</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="TreeMixin.trace-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="TreeMixin.trace-expanded"><a name="L410"></a><tt class="py-lineno"> 410</tt>  <tt class="py-line">        <tt class="py-docstring">"""List of all clade object between two targets in this tree.</tt> </tt>
<a name="L411"></a><tt class="py-lineno"> 411</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L412"></a><tt class="py-lineno"> 412</tt>  <tt class="py-line"><tt class="py-docstring">        Excluding `start`, including `finish`.</tt> </tt>
<a name="L413"></a><tt class="py-lineno"> 413</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L414"></a><tt class="py-lineno"> 414</tt>  <tt class="py-line">        <tt class="py-name">mrca</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-97" class="py-name" targets="Method Bio.Nexus.Trees.Tree.common_ancestor()=Bio.Nexus.Trees.Tree-class.html#common_ancestor,Method Bio.Phylo.BaseTree.TreeMixin.common_ancestor()=Bio.Phylo.BaseTree.TreeMixin-class.html#common_ancestor"><a title="Bio.Nexus.Trees.Tree.common_ancestor
Bio.Phylo.BaseTree.TreeMixin.common_ancestor" class="py-name" href="#" onclick="return doclink('link-97', 'common_ancestor', 'link-97');">common_ancestor</a></tt><tt class="py-op">(</tt><tt id="link-98" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.start=Bio.SeqFeature.CompoundLocation-class.html#start,Variable Bio.SeqFeature.FeatureLocation.start=Bio.SeqFeature.FeatureLocation-class.html#start"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-98', 'start', 'link-98');">start</a></tt><tt class="py-op">,</tt> <tt class="py-name">finish</tt><tt class="py-op">)</tt> </tt>
<a name="L415"></a><tt class="py-lineno"> 415</tt>  <tt class="py-line">        <tt class="py-name">fromstart</tt> <tt class="py-op">=</tt> <tt class="py-name">mrca</tt><tt class="py-op">.</tt><tt id="link-99" class="py-name" targets="Method Bio.Phylo.BaseTree.TreeMixin.get_path()=Bio.Phylo.BaseTree.TreeMixin-class.html#get_path"><a title="Bio.Phylo.BaseTree.TreeMixin.get_path" class="py-name" href="#" onclick="return doclink('link-99', 'get_path', 'link-99');">get_path</a></tt><tt class="py-op">(</tt><tt id="link-100" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-100', 'start', 'link-98');">start</a></tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">2</tt><tt class="py-op">:</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L416"></a><tt class="py-lineno"> 416</tt>  <tt class="py-line">        <tt class="py-name">to</tt> <tt class="py-op">=</tt> <tt class="py-name">mrca</tt><tt class="py-op">.</tt><tt id="link-101" class="py-name"><a title="Bio.Phylo.BaseTree.TreeMixin.get_path" class="py-name" href="#" onclick="return doclink('link-101', 'get_path', 'link-99');">get_path</a></tt><tt class="py-op">(</tt><tt class="py-name">finish</tt><tt class="py-op">)</tt> </tt>
<a name="L417"></a><tt class="py-lineno"> 417</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">fromstart</tt> <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt class="py-name">mrca</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-name">to</tt> </tt>
</div><a name="L418"></a><tt class="py-lineno"> 418</tt>  <tt class="py-line"> </tt>
<a name="L419"></a><tt class="py-lineno"> 419</tt>  <tt class="py-line">    <tt class="py-comment"># Information methods</tt> </tt>
<a name="L420"></a><tt class="py-lineno"> 420</tt>  <tt class="py-line"> </tt>
<a name="TreeMixin.common_ancestor"></a><div id="TreeMixin.common_ancestor-def"><a name="L421"></a><tt class="py-lineno"> 421</tt> <a class="py-toggle" href="#" id="TreeMixin.common_ancestor-toggle" onclick="return toggle('TreeMixin.common_ancestor');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.TreeMixin-class.html#common_ancestor">common_ancestor</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">targets</tt><tt class="py-op">,</tt> <tt class="py-op">*</tt><tt class="py-param">more_targets</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="TreeMixin.common_ancestor-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="TreeMixin.common_ancestor-expanded"><a name="L422"></a><tt class="py-lineno"> 422</tt>  <tt class="py-line">        <tt class="py-docstring">"""Most recent common ancestor (clade) of all the given targets.</tt> </tt>
<a name="L423"></a><tt class="py-lineno"> 423</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L424"></a><tt class="py-lineno"> 424</tt>  <tt class="py-line"><tt class="py-docstring">        Edge cases:</tt> </tt>
<a name="L425"></a><tt class="py-lineno"> 425</tt>  <tt class="py-line"><tt class="py-docstring">         - If no target is given, returns self.root</tt> </tt>
<a name="L426"></a><tt class="py-lineno"> 426</tt>  <tt class="py-line"><tt class="py-docstring">         - If 1 target is given, returns the target</tt> </tt>
<a name="L427"></a><tt class="py-lineno"> 427</tt>  <tt class="py-line"><tt class="py-docstring">         - If any target is not found in this tree, raises a ValueError</tt> </tt>
<a name="L428"></a><tt class="py-lineno"> 428</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L429"></a><tt class="py-lineno"> 429</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L430"></a><tt class="py-lineno"> 430</tt>  <tt class="py-line">        <tt class="py-name">paths</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-102" class="py-name"><a title="Bio.Phylo.BaseTree.TreeMixin.get_path" class="py-name" href="#" onclick="return doclink('link-102', 'get_path', 'link-99');">get_path</a></tt><tt class="py-op">(</tt><tt class="py-name">t</tt><tt class="py-op">)</tt> </tt>
<a name="L431"></a><tt class="py-lineno"> 431</tt>  <tt class="py-line">                 <tt class="py-keyword">for</tt> <tt class="py-name">t</tt> <tt class="py-keyword">in</tt> <tt id="link-103" class="py-name" targets="Function Bio.Phylo.BaseTree._combine_args()=Bio.Phylo.BaseTree-module.html#_combine_args"><a title="Bio.Phylo.BaseTree._combine_args" class="py-name" href="#" onclick="return doclink('link-103', '_combine_args', 'link-103');">_combine_args</a></tt><tt class="py-op">(</tt><tt class="py-name">targets</tt><tt class="py-op">,</tt> <tt class="py-op">*</tt><tt class="py-name">more_targets</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L432"></a><tt class="py-lineno"> 432</tt>  <tt class="py-line">        <tt class="py-comment"># Validation -- otherwise izip throws a spooky error below</tt> </tt>
<a name="L433"></a><tt class="py-lineno"> 433</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">p</tt><tt class="py-op">,</tt> <tt class="py-name">t</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">paths</tt><tt class="py-op">,</tt> <tt class="py-name">targets</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L434"></a><tt class="py-lineno"> 434</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">p</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L435"></a><tt class="py-lineno"> 435</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"target %s is not in this tree"</tt> <tt class="py-op">%</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">t</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L436"></a><tt class="py-lineno"> 436</tt>  <tt class="py-line">        <tt class="py-name">mrca</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-104" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-104', 'root', 'link-7');">root</a></tt> </tt>
<a name="L437"></a><tt class="py-lineno"> 437</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">level</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-op">*</tt><tt class="py-name">paths</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L438"></a><tt class="py-lineno"> 438</tt>  <tt class="py-line">            <tt id="link-105" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.ref=Bio.SeqFeature.CompoundLocation-class.html#ref,Variable Bio.SeqFeature.SeqFeature.ref=Bio.SeqFeature.SeqFeature-class.html#ref"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-105', 'ref', 'link-105');">ref</a></tt> <tt class="py-op">=</tt> <tt class="py-name">level</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L439"></a><tt class="py-lineno"> 439</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt id="link-106" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.other()=Bio.Phylo.PhyloXMLIO.Parser-class.html#other,Method Bio.Phylo.PhyloXMLIO.Writer.other()=Bio.Phylo.PhyloXMLIO.Writer-class.html#other"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-106', 'other', 'link-106');">other</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">level</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L440"></a><tt class="py-lineno"> 440</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-107" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-107', 'ref', 'link-105');">ref</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt id="link-108" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-108', 'other', 'link-106');">other</a></tt><tt class="py-op">:</tt> </tt>
<a name="L441"></a><tt class="py-lineno"> 441</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L442"></a><tt class="py-lineno"> 442</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L443"></a><tt class="py-lineno"> 443</tt>  <tt class="py-line">                <tt class="py-name">mrca</tt> <tt class="py-op">=</tt> <tt id="link-109" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-109', 'ref', 'link-105');">ref</a></tt> </tt>
<a name="L444"></a><tt class="py-lineno"> 444</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-110" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-110', 'ref', 'link-105');">ref</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">mrca</tt><tt class="py-op">:</tt> </tt>
<a name="L445"></a><tt class="py-lineno"> 445</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L446"></a><tt class="py-lineno"> 446</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">mrca</tt> </tt>
</div><a name="L447"></a><tt class="py-lineno"> 447</tt>  <tt class="py-line"> </tt>
<a name="TreeMixin.count_terminals"></a><div id="TreeMixin.count_terminals-def"><a name="L448"></a><tt class="py-lineno"> 448</tt> <a class="py-toggle" href="#" id="TreeMixin.count_terminals-toggle" onclick="return toggle('TreeMixin.count_terminals');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.TreeMixin-class.html#count_terminals">count_terminals</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="TreeMixin.count_terminals-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="TreeMixin.count_terminals-expanded"><a name="L449"></a><tt class="py-lineno"> 449</tt>  <tt class="py-line">        <tt class="py-docstring">"""Count the number of terminal (leaf) nodes within this tree."""</tt> </tt>
<a name="L450"></a><tt class="py-lineno"> 450</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-111" class="py-name"><a title="Bio.Phylo._utils
Bio.SearchIO._utils
Bio._utils" class="py-name" href="#" onclick="return doclink('link-111', '_utils', 'link-5');">_utils</a></tt><tt class="py-op">.</tt><tt id="link-112" class="py-name" targets="Function Bio._utils.iterlen()=Bio._utils-module.html#iterlen"><a title="Bio._utils.iterlen" class="py-name" href="#" onclick="return doclink('link-112', 'iterlen', 'link-112');">iterlen</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-113" class="py-name"><a title="Bio.Phylo.BaseTree.TreeMixin.find_clades" class="py-name" href="#" onclick="return doclink('link-113', 'find_clades', 'link-95');">find_clades</a></tt><tt class="py-op">(</tt><tt class="py-name">terminal</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L451"></a><tt class="py-lineno"> 451</tt>  <tt class="py-line"> </tt>
<a name="TreeMixin.depths"></a><div id="TreeMixin.depths-def"><a name="L452"></a><tt class="py-lineno"> 452</tt> <a class="py-toggle" href="#" id="TreeMixin.depths-toggle" onclick="return toggle('TreeMixin.depths');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.TreeMixin-class.html#depths">depths</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">unit_branch_lengths</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># noqa: D402</tt> </tt>
</div><a name="L453"></a><tt class="py-lineno"> 453</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a mapping of tree clades to depths (by branch length).</tt> </tt>
<a name="L454"></a><tt class="py-lineno"> 454</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L455"></a><tt class="py-lineno"> 455</tt>  <tt class="py-line"><tt class="py-docstring">        :Parameters:</tt> </tt>
<a name="L456"></a><tt class="py-lineno"> 456</tt>  <tt class="py-line"><tt class="py-docstring">            unit_branch_lengths : bool</tt> </tt>
<a name="L457"></a><tt class="py-lineno"> 457</tt>  <tt class="py-line"><tt class="py-docstring">                If True, count only the number of branches (levels in the tree).</tt> </tt>
<a name="L458"></a><tt class="py-lineno"> 458</tt>  <tt class="py-line"><tt class="py-docstring">                By default the distance is the cumulative branch length leading</tt> </tt>
<a name="L459"></a><tt class="py-lineno"> 459</tt>  <tt class="py-line"><tt class="py-docstring">                to the clade.</tt> </tt>
<a name="L460"></a><tt class="py-lineno"> 460</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L461"></a><tt class="py-lineno"> 461</tt>  <tt class="py-line"><tt class="py-docstring">        :returns: dict of {clade: depth}, where keys are all of the Clade</tt> </tt>
<a name="L462"></a><tt class="py-lineno"> 462</tt>  <tt class="py-line"><tt class="py-docstring">            instances in the tree, and values are the distance from the root to</tt> </tt>
<a name="L463"></a><tt class="py-lineno"> 463</tt>  <tt class="py-line"><tt class="py-docstring">            each clade (including terminals).</tt> </tt>
<a name="L464"></a><tt class="py-lineno"> 464</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L465"></a><tt class="py-lineno"> 465</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L466"></a><tt class="py-lineno"> 466</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">unit_branch_lengths</tt><tt class="py-op">:</tt> </tt>
<a name="L467"></a><tt class="py-lineno"> 467</tt>  <tt class="py-line">            <tt class="py-name">depth_of</tt> <tt class="py-op">=</tt> <tt class="py-keyword">lambda</tt> <tt class="py-name">c</tt><tt class="py-op">:</tt> <tt class="py-number">1</tt> </tt>
<a name="L468"></a><tt class="py-lineno"> 468</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L469"></a><tt class="py-lineno"> 469</tt>  <tt class="py-line">            <tt class="py-name">depth_of</tt> <tt class="py-op">=</tt> <tt class="py-keyword">lambda</tt> <tt class="py-name">c</tt><tt class="py-op">:</tt> <tt class="py-name">c</tt><tt class="py-op">.</tt><tt id="link-114" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.branch_length()=Bio.Phylo.PhyloXMLIO.Writer-class.html#branch_length"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-114', 'branch_length', 'link-114');">branch_length</a></tt> <tt class="py-keyword">or</tt> <tt class="py-number">0</tt> </tt>
<a name="L470"></a><tt class="py-lineno"> 470</tt>  <tt class="py-line">        <tt id="link-115" class="py-name" targets="Method Bio.Phylo.BaseTree.TreeMixin.depths()=Bio.Phylo.BaseTree.TreeMixin-class.html#depths"><a title="Bio.Phylo.BaseTree.TreeMixin.depths" class="py-name" href="#" onclick="return doclink('link-115', 'depths', 'link-115');">depths</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L471"></a><tt class="py-lineno"> 471</tt>  <tt class="py-line"> </tt>
<a name="L472"></a><tt class="py-lineno"> 472</tt>  <tt class="py-line">        <tt class="py-keyword">def</tt> <tt class="py-def-name">update_depths</tt><tt class="py-op">(</tt><tt class="py-param">node</tt><tt class="py-op">,</tt> <tt class="py-param">curr_depth</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L473"></a><tt class="py-lineno"> 473</tt>  <tt class="py-line">            <tt id="link-116" class="py-name"><a title="Bio.Phylo.BaseTree.TreeMixin.depths" class="py-name" href="#" onclick="return doclink('link-116', 'depths', 'link-115');">depths</a></tt><tt class="py-op">[</tt><tt id="link-117" class="py-name"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-117', 'node', 'link-18');">node</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">curr_depth</tt> </tt>
<a name="L474"></a><tt class="py-lineno"> 474</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">child</tt> <tt class="py-keyword">in</tt> <tt id="link-118" class="py-name"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-118', 'node', 'link-18');">node</a></tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">:</tt> </tt>
<a name="L475"></a><tt class="py-lineno"> 475</tt>  <tt class="py-line">                <tt class="py-name">new_depth</tt> <tt class="py-op">=</tt> <tt class="py-name">curr_depth</tt> <tt class="py-op">+</tt> <tt class="py-name">depth_of</tt><tt class="py-op">(</tt><tt class="py-name">child</tt><tt class="py-op">)</tt> </tt>
<a name="L476"></a><tt class="py-lineno"> 476</tt>  <tt class="py-line">                <tt class="py-name">update_depths</tt><tt class="py-op">(</tt><tt class="py-name">child</tt><tt class="py-op">,</tt> <tt class="py-name">new_depth</tt><tt class="py-op">)</tt> </tt>
</div><a name="L477"></a><tt class="py-lineno"> 477</tt>  <tt class="py-line"> </tt>
<a name="L478"></a><tt class="py-lineno"> 478</tt>  <tt class="py-line">        <tt class="py-name">update_depths</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-119" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-119', 'root', 'link-7');">root</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-120" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-120', 'root', 'link-7');">root</a></tt><tt class="py-op">.</tt><tt id="link-121" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-121', 'branch_length', 'link-114');">branch_length</a></tt> <tt class="py-keyword">or</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L479"></a><tt class="py-lineno"> 479</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-122" class="py-name"><a title="Bio.Phylo.BaseTree.TreeMixin.depths" class="py-name" href="#" onclick="return doclink('link-122', 'depths', 'link-115');">depths</a></tt> </tt>
</div><a name="L480"></a><tt class="py-lineno"> 480</tt>  <tt class="py-line"> </tt>
<a name="TreeMixin.distance"></a><div id="TreeMixin.distance-def"><a name="L481"></a><tt class="py-lineno"> 481</tt> <a class="py-toggle" href="#" id="TreeMixin.distance-toggle" onclick="return toggle('TreeMixin.distance');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.TreeMixin-class.html#distance">distance</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">target1</tt><tt class="py-op">,</tt> <tt class="py-param">target2</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="TreeMixin.distance-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="TreeMixin.distance-expanded"><a name="L482"></a><tt class="py-lineno"> 482</tt>  <tt class="py-line">        <tt class="py-docstring">"""Calculate the sum of the branch lengths between two targets.</tt> </tt>
<a name="L483"></a><tt class="py-lineno"> 483</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L484"></a><tt class="py-lineno"> 484</tt>  <tt class="py-line"><tt class="py-docstring">        If only one target is specified, the other is the root of this tree.</tt> </tt>
<a name="L485"></a><tt class="py-lineno"> 485</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L486"></a><tt class="py-lineno"> 486</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">target2</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L487"></a><tt class="py-lineno"> 487</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-123" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.sum()=Bio.Nexus.Nexus.StepMatrix-class.html#sum,Method Bio.SubsMat.SeqMat.sum()=Bio.SubsMat.SeqMat-class.html#sum"><a title="Bio.Nexus.Nexus.StepMatrix.sum
Bio.SubsMat.SeqMat.sum" class="py-name" href="#" onclick="return doclink('link-123', 'sum', 'link-123');">sum</a></tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">.</tt><tt id="link-124" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-124', 'branch_length', 'link-114');">branch_length</a></tt> <tt class="py-keyword">for</tt> <tt class="py-name">n</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-125" class="py-name"><a title="Bio.Phylo.BaseTree.TreeMixin.get_path" class="py-name" href="#" onclick="return doclink('link-125', 'get_path', 'link-99');">get_path</a></tt><tt class="py-op">(</tt><tt class="py-name">target1</tt><tt class="py-op">)</tt> </tt>
<a name="L488"></a><tt class="py-lineno"> 488</tt>  <tt class="py-line">                       <tt class="py-keyword">if</tt> <tt class="py-name">n</tt><tt class="py-op">.</tt><tt id="link-126" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-126', 'branch_length', 'link-114');">branch_length</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
<a name="L489"></a><tt class="py-lineno"> 489</tt>  <tt class="py-line">        <tt class="py-name">mrca</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-127" class="py-name"><a title="Bio.Nexus.Trees.Tree.common_ancestor
Bio.Phylo.BaseTree.TreeMixin.common_ancestor" class="py-name" href="#" onclick="return doclink('link-127', 'common_ancestor', 'link-97');">common_ancestor</a></tt><tt class="py-op">(</tt><tt class="py-name">target1</tt><tt class="py-op">,</tt> <tt class="py-name">target2</tt><tt class="py-op">)</tt> </tt>
<a name="L490"></a><tt class="py-lineno"> 490</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">mrca</tt><tt class="py-op">.</tt><tt id="link-128" class="py-name" targets="Method Bio.Nexus.Trees.Tree.distance()=Bio.Nexus.Trees.Tree-class.html#distance,Method Bio.Phylo.BaseTree.TreeMixin.distance()=Bio.Phylo.BaseTree.TreeMixin-class.html#distance"><a title="Bio.Nexus.Trees.Tree.distance
Bio.Phylo.BaseTree.TreeMixin.distance" class="py-name" href="#" onclick="return doclink('link-128', 'distance', 'link-128');">distance</a></tt><tt class="py-op">(</tt><tt class="py-name">target1</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-name">mrca</tt><tt class="py-op">.</tt><tt id="link-129" class="py-name"><a title="Bio.Nexus.Trees.Tree.distance
Bio.Phylo.BaseTree.TreeMixin.distance" class="py-name" href="#" onclick="return doclink('link-129', 'distance', 'link-128');">distance</a></tt><tt class="py-op">(</tt><tt class="py-name">target2</tt><tt class="py-op">)</tt> </tt>
</div><a name="L491"></a><tt class="py-lineno"> 491</tt>  <tt class="py-line"> </tt>
<a name="TreeMixin.is_bifurcating"></a><div id="TreeMixin.is_bifurcating-def"><a name="L492"></a><tt class="py-lineno"> 492</tt> <a class="py-toggle" href="#" id="TreeMixin.is_bifurcating-toggle" onclick="return toggle('TreeMixin.is_bifurcating');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.TreeMixin-class.html#is_bifurcating">is_bifurcating</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="TreeMixin.is_bifurcating-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="TreeMixin.is_bifurcating-expanded"><a name="L493"></a><tt class="py-lineno"> 493</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return True if tree downstream of node is strictly bifurcating.</tt> </tt>
<a name="L494"></a><tt class="py-lineno"> 494</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L495"></a><tt class="py-lineno"> 495</tt>  <tt class="py-line"><tt class="py-docstring">        I.e., all nodes have either 2 or 0 children (internal or external,</tt> </tt>
<a name="L496"></a><tt class="py-lineno"> 496</tt>  <tt class="py-line"><tt class="py-docstring">        respectively). The root may have 3 descendents and still be considered</tt> </tt>
<a name="L497"></a><tt class="py-lineno"> 497</tt>  <tt class="py-line"><tt class="py-docstring">        part of a bifurcating tree, because it has no ancestor.</tt> </tt>
<a name="L498"></a><tt class="py-lineno"> 498</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L499"></a><tt class="py-lineno"> 499</tt>  <tt class="py-line">        <tt class="py-comment"># Root can be trifurcating</tt> </tt>
<a name="L500"></a><tt class="py-lineno"> 500</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt id="link-130" class="py-name"><a title="Bio.Cluster.Tree
Bio.Nexus.Trees.Tree
Bio.Phylo.BaseTree.Tree
Bio.Phylo.CDAO.Tree
Bio.Phylo.NeXML.Tree
Bio.Phylo.Newick.Tree" class="py-name" href="#" onclick="return doclink('link-130', 'Tree', 'link-24');">Tree</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-131" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-131', 'root', 'link-7');">root</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">3</tt><tt class="py-op">:</tt> </tt>
<a name="L501"></a><tt class="py-lineno"> 501</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-132" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-132', 'root', 'link-7');">root</a></tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-133" class="py-name" targets="Method Bio.Nexus.Trees.Tree.is_bifurcating()=Bio.Nexus.Trees.Tree-class.html#is_bifurcating,Method Bio.Phylo.BaseTree.TreeMixin.is_bifurcating()=Bio.Phylo.BaseTree.TreeMixin-class.html#is_bifurcating"><a title="Bio.Nexus.Trees.Tree.is_bifurcating
Bio.Phylo.BaseTree.TreeMixin.is_bifurcating" class="py-name" href="#" onclick="return doclink('link-133', 'is_bifurcating', 'link-133');">is_bifurcating</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> </tt>
<a name="L502"></a><tt class="py-lineno"> 502</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-134" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-134', 'root', 'link-7');">root</a></tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-135" class="py-name"><a title="Bio.Nexus.Trees.Tree.is_bifurcating
Bio.Phylo.BaseTree.TreeMixin.is_bifurcating" class="py-name" href="#" onclick="return doclink('link-135', 'is_bifurcating', 'link-133');">is_bifurcating</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> </tt>
<a name="L503"></a><tt class="py-lineno"> 503</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-136" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-136', 'root', 'link-7');">root</a></tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-137" class="py-name"><a title="Bio.Nexus.Trees.Tree.is_bifurcating
Bio.Phylo.BaseTree.TreeMixin.is_bifurcating" class="py-name" href="#" onclick="return doclink('link-137', 'is_bifurcating', 'link-133');">is_bifurcating</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L504"></a><tt class="py-lineno"> 504</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-138" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-138', 'root', 'link-7');">root</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> </tt>
<a name="L505"></a><tt class="py-lineno"> 505</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-139" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-139', 'root', 'link-7');">root</a></tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-140" class="py-name"><a title="Bio.Nexus.Trees.Tree.is_bifurcating
Bio.Phylo.BaseTree.TreeMixin.is_bifurcating" class="py-name" href="#" onclick="return doclink('link-140', 'is_bifurcating', 'link-133');">is_bifurcating</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> </tt>
<a name="L506"></a><tt class="py-lineno"> 506</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-141" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-141', 'root', 'link-7');">root</a></tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-142" class="py-name"><a title="Bio.Nexus.Trees.Tree.is_bifurcating
Bio.Phylo.BaseTree.TreeMixin.is_bifurcating" class="py-name" href="#" onclick="return doclink('link-142', 'is_bifurcating', 'link-133');">is_bifurcating</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L507"></a><tt class="py-lineno"> 507</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-143" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-143', 'root', 'link-7');">root</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L508"></a><tt class="py-lineno"> 508</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">True</tt> </tt>
<a name="L509"></a><tt class="py-lineno"> 509</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">False</tt> </tt>
</div><a name="L510"></a><tt class="py-lineno"> 510</tt>  <tt class="py-line"> </tt>
<a name="TreeMixin.is_monophyletic"></a><div id="TreeMixin.is_monophyletic-def"><a name="L511"></a><tt class="py-lineno"> 511</tt> <a class="py-toggle" href="#" id="TreeMixin.is_monophyletic-toggle" onclick="return toggle('TreeMixin.is_monophyletic');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.TreeMixin-class.html#is_monophyletic">is_monophyletic</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">terminals</tt><tt class="py-op">,</tt> <tt class="py-op">*</tt><tt class="py-param">more_terminals</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="TreeMixin.is_monophyletic-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="TreeMixin.is_monophyletic-expanded"><a name="L512"></a><tt class="py-lineno"> 512</tt>  <tt class="py-line">        <tt class="py-docstring">"""MRCA of terminals if they comprise a complete subclade, or False.</tt> </tt>
<a name="L513"></a><tt class="py-lineno"> 513</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L514"></a><tt class="py-lineno"> 514</tt>  <tt class="py-line"><tt class="py-docstring">        I.e., there exists a clade such that its terminals are the same set as</tt> </tt>
<a name="L515"></a><tt class="py-lineno"> 515</tt>  <tt class="py-line"><tt class="py-docstring">        the given targets.</tt> </tt>
<a name="L516"></a><tt class="py-lineno"> 516</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L517"></a><tt class="py-lineno"> 517</tt>  <tt class="py-line"><tt class="py-docstring">        The given targets must be terminals of the tree.</tt> </tt>
<a name="L518"></a><tt class="py-lineno"> 518</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L519"></a><tt class="py-lineno"> 519</tt>  <tt class="py-line"><tt class="py-docstring">        To match both `Bio.Nexus.Trees` and the other multi-target methods in</tt> </tt>
<a name="L520"></a><tt class="py-lineno"> 520</tt>  <tt class="py-line"><tt class="py-docstring">        Bio.Phylo, arguments to this method can be specified either of two ways:</tt> </tt>
<a name="L521"></a><tt class="py-lineno"> 521</tt>  <tt class="py-line"><tt class="py-docstring">        (i) as a single list of targets, or (ii) separately specified targets,</tt> </tt>
<a name="L522"></a><tt class="py-lineno"> 522</tt>  <tt class="py-line"><tt class="py-docstring">        e.g. is_monophyletic(t1, t2, t3) -- but not both.</tt> </tt>
<a name="L523"></a><tt class="py-lineno"> 523</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L524"></a><tt class="py-lineno"> 524</tt>  <tt class="py-line"><tt class="py-docstring">        For convenience, this method returns the common ancestor (MCRA) of the</tt> </tt>
<a name="L525"></a><tt class="py-lineno"> 525</tt>  <tt class="py-line"><tt class="py-docstring">        targets if they are monophyletic (instead of the value True), and False</tt> </tt>
<a name="L526"></a><tt class="py-lineno"> 526</tt>  <tt class="py-line"><tt class="py-docstring">        otherwise.</tt> </tt>
<a name="L527"></a><tt class="py-lineno"> 527</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L528"></a><tt class="py-lineno"> 528</tt>  <tt class="py-line"><tt class="py-docstring">        :returns: common ancestor if terminals are monophyletic, otherwise False.</tt> </tt>
<a name="L529"></a><tt class="py-lineno"> 529</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L530"></a><tt class="py-lineno"> 530</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L531"></a><tt class="py-lineno"> 531</tt>  <tt class="py-line">        <tt class="py-name">target_set</tt> <tt class="py-op">=</tt> <tt id="link-144" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.set()=Bio.Nexus.Nexus.StepMatrix-class.html#set,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.set()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#set,Method Bio.SVDSuperimposer.SVDSuperimposer.set()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#set"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-144', 'set', 'link-144');">set</a></tt><tt class="py-op">(</tt><tt id="link-145" class="py-name"><a title="Bio.Phylo.BaseTree._combine_args" class="py-name" href="#" onclick="return doclink('link-145', '_combine_args', 'link-103');">_combine_args</a></tt><tt class="py-op">(</tt><tt class="py-name">terminals</tt><tt class="py-op">,</tt> <tt class="py-op">*</tt><tt class="py-name">more_terminals</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L532"></a><tt class="py-lineno"> 532</tt>  <tt class="py-line">        <tt class="py-name">current</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-146" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-146', 'root', 'link-7');">root</a></tt> </tt>
<a name="L533"></a><tt class="py-lineno"> 533</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L534"></a><tt class="py-lineno"> 534</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-147" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-147', 'set', 'link-144');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">current</tt><tt class="py-op">.</tt><tt id="link-148" class="py-name" targets="Method Bio.Nexus.Trees.Tree.get_terminals()=Bio.Nexus.Trees.Tree-class.html#get_terminals,Method Bio.Phylo.BaseTree.TreeMixin.get_terminals()=Bio.Phylo.BaseTree.TreeMixin-class.html#get_terminals"><a title="Bio.Nexus.Trees.Tree.get_terminals
Bio.Phylo.BaseTree.TreeMixin.get_terminals" class="py-name" href="#" onclick="return doclink('link-148', 'get_terminals', 'link-148');">get_terminals</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-name">target_set</tt><tt class="py-op">:</tt> </tt>
<a name="L535"></a><tt class="py-lineno"> 535</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">current</tt> </tt>
<a name="L536"></a><tt class="py-lineno"> 536</tt>  <tt class="py-line">            <tt class="py-comment"># Try a narrower subclade</tt> </tt>
<a name="L537"></a><tt class="py-lineno"> 537</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">subclade</tt> <tt class="py-keyword">in</tt> <tt class="py-name">current</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">:</tt> </tt>
<a name="L538"></a><tt class="py-lineno"> 538</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-149" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-149', 'set', 'link-144');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">subclade</tt><tt class="py-op">.</tt><tt id="link-150" class="py-name"><a title="Bio.Nexus.Trees.Tree.get_terminals
Bio.Phylo.BaseTree.TreeMixin.get_terminals" class="py-name" href="#" onclick="return doclink('link-150', 'get_terminals', 'link-148');">get_terminals</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">issuperset</tt><tt class="py-op">(</tt><tt class="py-name">target_set</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L539"></a><tt class="py-lineno"> 539</tt>  <tt class="py-line">                    <tt class="py-name">current</tt> <tt class="py-op">=</tt> <tt class="py-name">subclade</tt> </tt>
<a name="L540"></a><tt class="py-lineno"> 540</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L541"></a><tt class="py-lineno"> 541</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L542"></a><tt class="py-lineno"> 542</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">False</tt> </tt>
</div><a name="L543"></a><tt class="py-lineno"> 543</tt>  <tt class="py-line"> </tt>
<a name="TreeMixin.is_parent_of"></a><div id="TreeMixin.is_parent_of-def"><a name="L544"></a><tt class="py-lineno"> 544</tt> <a class="py-toggle" href="#" id="TreeMixin.is_parent_of-toggle" onclick="return toggle('TreeMixin.is_parent_of');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.TreeMixin-class.html#is_parent_of">is_parent_of</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">target</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="TreeMixin.is_parent_of-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="TreeMixin.is_parent_of-expanded"><a name="L545"></a><tt class="py-lineno"> 545</tt>  <tt class="py-line">        <tt class="py-docstring">"""Check if target is a descendent of this tree.</tt> </tt>
<a name="L546"></a><tt class="py-lineno"> 546</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L547"></a><tt class="py-lineno"> 547</tt>  <tt class="py-line"><tt class="py-docstring">        Not required to be a direct descendent.</tt> </tt>
<a name="L548"></a><tt class="py-lineno"> 548</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L549"></a><tt class="py-lineno"> 549</tt>  <tt class="py-line"><tt class="py-docstring">        To check only direct descendents of a clade, simply use list membership</tt> </tt>
<a name="L550"></a><tt class="py-lineno"> 550</tt>  <tt class="py-line"><tt class="py-docstring">        testing: ``if subclade in clade: ...``</tt> </tt>
<a name="L551"></a><tt class="py-lineno"> 551</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L552"></a><tt class="py-lineno"> 552</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-151" class="py-name"><a title="Bio.Phylo.BaseTree.TreeMixin.get_path" class="py-name" href="#" onclick="return doclink('link-151', 'get_path', 'link-99');">get_path</a></tt><tt class="py-op">(</tt><tt class="py-name">target</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> </tt>
</div><a name="L553"></a><tt class="py-lineno"> 553</tt>  <tt class="py-line"> </tt>
<a name="TreeMixin.is_preterminal"></a><div id="TreeMixin.is_preterminal-def"><a name="L554"></a><tt class="py-lineno"> 554</tt> <a class="py-toggle" href="#" id="TreeMixin.is_preterminal-toggle" onclick="return toggle('TreeMixin.is_preterminal');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.TreeMixin-class.html#is_preterminal">is_preterminal</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="TreeMixin.is_preterminal-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="TreeMixin.is_preterminal-expanded"><a name="L555"></a><tt class="py-lineno"> 555</tt>  <tt class="py-line">        <tt class="py-docstring">"""Check if all direct descendents are terminal."""</tt> </tt>
<a name="L556"></a><tt class="py-lineno"> 556</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-152" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-152', 'root', 'link-7');">root</a></tt><tt class="py-op">.</tt><tt id="link-153" class="py-name"><a title="Bio.Nexus.Trees.Tree.is_terminal
Bio.Phylo.BaseTree.Clade.is_terminal
Bio.Phylo.BaseTree.Tree.is_terminal" class="py-name" href="#" onclick="return doclink('link-153', 'is_terminal', 'link-34');">is_terminal</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L557"></a><tt class="py-lineno"> 557</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">False</tt> </tt>
<a name="L558"></a><tt class="py-lineno"> 558</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-154" class="py-name" targets="Variable Bio.Phylo.BaseTree.Tree.clade=Bio.Phylo.BaseTree.Tree-class.html#clade,Method Bio.Phylo.PhyloXMLIO.Writer.clade()=Bio.Phylo.PhyloXMLIO.Writer-class.html#clade"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-154', 'clade', 'link-154');">clade</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-155" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-155', 'root', 'link-7');">root</a></tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">:</tt> </tt>
<a name="L559"></a><tt class="py-lineno"> 559</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-156" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-156', 'clade', 'link-154');">clade</a></tt><tt class="py-op">.</tt><tt id="link-157" class="py-name"><a title="Bio.Nexus.Trees.Tree.is_terminal
Bio.Phylo.BaseTree.Clade.is_terminal
Bio.Phylo.BaseTree.Tree.is_terminal" class="py-name" href="#" onclick="return doclink('link-157', 'is_terminal', 'link-34');">is_terminal</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L560"></a><tt class="py-lineno"> 560</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">False</tt> </tt>
<a name="L561"></a><tt class="py-lineno"> 561</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">True</tt> </tt>
</div><a name="L562"></a><tt class="py-lineno"> 562</tt>  <tt class="py-line"> </tt>
<a name="TreeMixin.total_branch_length"></a><div id="TreeMixin.total_branch_length-def"><a name="L563"></a><tt class="py-lineno"> 563</tt> <a class="py-toggle" href="#" id="TreeMixin.total_branch_length-toggle" onclick="return toggle('TreeMixin.total_branch_length');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.TreeMixin-class.html#total_branch_length">total_branch_length</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="TreeMixin.total_branch_length-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="TreeMixin.total_branch_length-expanded"><a name="L564"></a><tt class="py-lineno"> 564</tt>  <tt class="py-line">        <tt class="py-docstring">"""Calculate the sum of all the branch lengths in this tree."""</tt> </tt>
<a name="L565"></a><tt class="py-lineno"> 565</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-158" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.sum
Bio.SubsMat.SeqMat.sum" class="py-name" href="#" onclick="return doclink('link-158', 'sum', 'link-123');">sum</a></tt><tt class="py-op">(</tt><tt id="link-159" class="py-name"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-159', 'node', 'link-18');">node</a></tt><tt class="py-op">.</tt><tt id="link-160" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-160', 'branch_length', 'link-114');">branch_length</a></tt> </tt>
<a name="L566"></a><tt class="py-lineno"> 566</tt>  <tt class="py-line">                   <tt class="py-keyword">for</tt> <tt id="link-161" class="py-name"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-161', 'node', 'link-18');">node</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-162" class="py-name"><a title="Bio.Phylo.BaseTree.TreeMixin.find_clades" class="py-name" href="#" onclick="return doclink('link-162', 'find_clades', 'link-95');">find_clades</a></tt><tt class="py-op">(</tt><tt id="link-163" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-163', 'branch_length', 'link-114');">branch_length</a></tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L567"></a><tt class="py-lineno"> 567</tt>  <tt class="py-line"> </tt>
<a name="L568"></a><tt class="py-lineno"> 568</tt>  <tt class="py-line">    <tt class="py-comment"># Tree manipulation methods</tt> </tt>
<a name="L569"></a><tt class="py-lineno"> 569</tt>  <tt class="py-line"> </tt>
<a name="TreeMixin.collapse"></a><div id="TreeMixin.collapse-def"><a name="L570"></a><tt class="py-lineno"> 570</tt> <a class="py-toggle" href="#" id="TreeMixin.collapse-toggle" onclick="return toggle('TreeMixin.collapse');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.TreeMixin-class.html#collapse">collapse</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">target</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="TreeMixin.collapse-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="TreeMixin.collapse-expanded"><a name="L571"></a><tt class="py-lineno"> 571</tt>  <tt class="py-line">        <tt class="py-docstring">"""Delete target from the tree, relinking its children to its parent.</tt> </tt>
<a name="L572"></a><tt class="py-lineno"> 572</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L573"></a><tt class="py-lineno"> 573</tt>  <tt class="py-line"><tt class="py-docstring">        :returns: the parent clade.</tt> </tt>
<a name="L574"></a><tt class="py-lineno"> 574</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L575"></a><tt class="py-lineno"> 575</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L576"></a><tt class="py-lineno"> 576</tt>  <tt class="py-line">        <tt class="py-name">path</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-164" class="py-name"><a title="Bio.Phylo.BaseTree.TreeMixin.get_path" class="py-name" href="#" onclick="return doclink('link-164', 'get_path', 'link-99');">get_path</a></tt><tt class="py-op">(</tt><tt class="py-name">target</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> </tt>
<a name="L577"></a><tt class="py-lineno"> 577</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">path</tt><tt class="py-op">:</tt> </tt>
<a name="L578"></a><tt class="py-lineno"> 578</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"couldn't collapse %s in this tree"</tt> </tt>
<a name="L579"></a><tt class="py-lineno"> 579</tt>  <tt class="py-line">                             <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">target</tt> <tt class="py-keyword">or</tt> <tt class="py-name">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L580"></a><tt class="py-lineno"> 580</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">path</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L581"></a><tt class="py-lineno"> 581</tt>  <tt class="py-line">            <tt class="py-name">parent</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-165" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-165', 'root', 'link-7');">root</a></tt> </tt>
<a name="L582"></a><tt class="py-lineno"> 582</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L583"></a><tt class="py-lineno"> 583</tt>  <tt class="py-line">            <tt class="py-name">parent</tt> <tt class="py-op">=</tt> <tt class="py-name">path</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> </tt>
<a name="L584"></a><tt class="py-lineno"> 584</tt>  <tt class="py-line">        <tt class="py-name">popped</tt> <tt class="py-op">=</tt> <tt class="py-name">parent</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">.</tt><tt id="link-166" class="py-name"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop" class="py-name" href="#" onclick="return doclink('link-166', 'pop', 'link-31');">pop</a></tt><tt class="py-op">(</tt><tt class="py-name">parent</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">.</tt><tt id="link-167" class="py-name" targets="Method Bio.Crystal.Chain.index()=Bio.Crystal.Chain-class.html#index,Method Bio.SCOP.Raf.SeqMap.index()=Bio.SCOP.Raf.SeqMap-class.html#index,Method Bio.SearchIO._model.hit.Hit.index()=Bio.SearchIO._model.hit.Hit-class.html#index,Method Bio.SearchIO._model.query.QueryResult.index()=Bio.SearchIO._model.query.QueryResult-class.html#index,Function Bio.SearchIO.index()=Bio.SearchIO-module.html#index,Method Bio.Seq.MutableSeq.index()=Bio.Seq.MutableSeq-class.html#index,Function Bio.SeqIO.index()=Bio.SeqIO-module.html#index"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-167', 'index', 'link-167');">index</a></tt><tt class="py-op">(</tt><tt class="py-name">path</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L585"></a><tt class="py-lineno"> 585</tt>  <tt class="py-line">        <tt class="py-name">extra_length</tt> <tt class="py-op">=</tt> <tt class="py-name">popped</tt><tt class="py-op">.</tt><tt id="link-168" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-168', 'branch_length', 'link-114');">branch_length</a></tt> <tt class="py-keyword">or</tt> <tt class="py-number">0</tt> </tt>
<a name="L586"></a><tt class="py-lineno"> 586</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">child</tt> <tt class="py-keyword">in</tt> <tt class="py-name">popped</tt><tt class="py-op">:</tt> </tt>
<a name="L587"></a><tt class="py-lineno"> 587</tt>  <tt class="py-line">            <tt class="py-name">child</tt><tt class="py-op">.</tt><tt id="link-169" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-169', 'branch_length', 'link-114');">branch_length</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">extra_length</tt> </tt>
<a name="L588"></a><tt class="py-lineno"> 588</tt>  <tt class="py-line">        <tt class="py-name">parent</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">.</tt><tt id="link-170" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-170', 'extend', 'link-8');">extend</a></tt><tt class="py-op">(</tt><tt class="py-name">popped</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">)</tt> </tt>
<a name="L589"></a><tt class="py-lineno"> 589</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">parent</tt> </tt>
</div><a name="L590"></a><tt class="py-lineno"> 590</tt>  <tt class="py-line"> </tt>
<a name="TreeMixin.collapse_all"></a><div id="TreeMixin.collapse_all-def"><a name="L591"></a><tt class="py-lineno"> 591</tt> <a class="py-toggle" href="#" id="TreeMixin.collapse_all-toggle" onclick="return toggle('TreeMixin.collapse_all');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.TreeMixin-class.html#collapse_all">collapse_all</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">target</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="TreeMixin.collapse_all-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="TreeMixin.collapse_all-expanded"><a name="L592"></a><tt class="py-lineno"> 592</tt>  <tt class="py-line">        <tt class="py-docstring">"""Collapse all the descendents of this tree, leaving only terminals.</tt> </tt>
<a name="L593"></a><tt class="py-lineno"> 593</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L594"></a><tt class="py-lineno"> 594</tt>  <tt class="py-line"><tt class="py-docstring">        Total branch lengths are preserved, i.e. the distance to each terminal</tt> </tt>
<a name="L595"></a><tt class="py-lineno"> 595</tt>  <tt class="py-line"><tt class="py-docstring">        stays the same.</tt> </tt>
<a name="L596"></a><tt class="py-lineno"> 596</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L597"></a><tt class="py-lineno"> 597</tt>  <tt class="py-line"><tt class="py-docstring">        For example, this will safely collapse nodes with poor bootstrap</tt> </tt>
<a name="L598"></a><tt class="py-lineno"> 598</tt>  <tt class="py-line"><tt class="py-docstring">        support:</tt> </tt>
<a name="L599"></a><tt class="py-lineno"> 599</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L600"></a><tt class="py-lineno"> 600</tt>  <tt class="py-line"><tt class="py-docstring">            &gt;&gt;&gt; from Bio import Phylo</tt> </tt>
<a name="L601"></a><tt class="py-lineno"> 601</tt>  <tt class="py-line"><tt class="py-docstring">            &gt;&gt;&gt; tree = Phylo.read('PhyloXML/apaf.xml', 'phyloxml')</tt> </tt>
<a name="L602"></a><tt class="py-lineno"> 602</tt>  <tt class="py-line"><tt class="py-docstring">            &gt;&gt;&gt; print("Total branch length %0.2f" % tree.total_branch_length())</tt> </tt>
<a name="L603"></a><tt class="py-lineno"> 603</tt>  <tt class="py-line"><tt class="py-docstring">            Total branch length 20.44</tt> </tt>
<a name="L604"></a><tt class="py-lineno"> 604</tt>  <tt class="py-line"><tt class="py-docstring">            &gt;&gt;&gt; tree.collapse_all(lambda c: c.confidence is not None and c.confidence &lt; 70)</tt> </tt>
<a name="L605"></a><tt class="py-lineno"> 605</tt>  <tt class="py-line"><tt class="py-docstring">            &gt;&gt;&gt; print("Total branch length %0.2f" % tree.total_branch_length())</tt> </tt>
<a name="L606"></a><tt class="py-lineno"> 606</tt>  <tt class="py-line"><tt class="py-docstring">            Total branch length 21.37</tt> </tt>
<a name="L607"></a><tt class="py-lineno"> 607</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L608"></a><tt class="py-lineno"> 608</tt>  <tt class="py-line"><tt class="py-docstring">        This implementation avoids strange side-effects by using level-order</tt> </tt>
<a name="L609"></a><tt class="py-lineno"> 609</tt>  <tt class="py-line"><tt class="py-docstring">        traversal and testing all clade properties (versus the target</tt> </tt>
<a name="L610"></a><tt class="py-lineno"> 610</tt>  <tt class="py-line"><tt class="py-docstring">        specification) up front. In particular, if a clade meets the target</tt> </tt>
<a name="L611"></a><tt class="py-lineno"> 611</tt>  <tt class="py-line"><tt class="py-docstring">        specification in the original tree, it will be collapsed.  For example,</tt> </tt>
<a name="L612"></a><tt class="py-lineno"> 612</tt>  <tt class="py-line"><tt class="py-docstring">        if the condition is:</tt> </tt>
<a name="L613"></a><tt class="py-lineno"> 613</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L614"></a><tt class="py-lineno"> 614</tt>  <tt class="py-line"><tt class="py-docstring">            &gt;&gt;&gt; from Bio import Phylo</tt> </tt>
<a name="L615"></a><tt class="py-lineno"> 615</tt>  <tt class="py-line"><tt class="py-docstring">            &gt;&gt;&gt; tree = Phylo.read('PhyloXML/apaf.xml', 'phyloxml')</tt> </tt>
<a name="L616"></a><tt class="py-lineno"> 616</tt>  <tt class="py-line"><tt class="py-docstring">            &gt;&gt;&gt; print("Total branch length %0.2f" % tree.total_branch_length())</tt> </tt>
<a name="L617"></a><tt class="py-lineno"> 617</tt>  <tt class="py-line"><tt class="py-docstring">            Total branch length 20.44</tt> </tt>
<a name="L618"></a><tt class="py-lineno"> 618</tt>  <tt class="py-line"><tt class="py-docstring">            &gt;&gt;&gt; tree.collapse_all(lambda c: c.branch_length &lt; 0.1)</tt> </tt>
<a name="L619"></a><tt class="py-lineno"> 619</tt>  <tt class="py-line"><tt class="py-docstring">            &gt;&gt;&gt; print("Total branch length %0.2f" % tree.total_branch_length())</tt> </tt>
<a name="L620"></a><tt class="py-lineno"> 620</tt>  <tt class="py-line"><tt class="py-docstring">            Total branch length 21.13</tt> </tt>
<a name="L621"></a><tt class="py-lineno"> 621</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L622"></a><tt class="py-lineno"> 622</tt>  <tt class="py-line"><tt class="py-docstring">        Collapsing a clade's parent node adds the parent's branch length to the</tt> </tt>
<a name="L623"></a><tt class="py-lineno"> 623</tt>  <tt class="py-line"><tt class="py-docstring">        child, so during the execution of collapse_all, a clade's branch_length</tt> </tt>
<a name="L624"></a><tt class="py-lineno"> 624</tt>  <tt class="py-line"><tt class="py-docstring">        may increase. In this implementation, clades are collapsed according to</tt> </tt>
<a name="L625"></a><tt class="py-lineno"> 625</tt>  <tt class="py-line"><tt class="py-docstring">        their properties in the original tree, not the properties when tree</tt> </tt>
<a name="L626"></a><tt class="py-lineno"> 626</tt>  <tt class="py-line"><tt class="py-docstring">        traversal reaches the clade. (It's easier to debug.) If you want the</tt> </tt>
<a name="L627"></a><tt class="py-lineno"> 627</tt>  <tt class="py-line"><tt class="py-docstring">        other behavior (incremental testing), modifying the source code of this</tt> </tt>
<a name="L628"></a><tt class="py-lineno"> 628</tt>  <tt class="py-line"><tt class="py-docstring">        function is straightforward.</tt> </tt>
<a name="L629"></a><tt class="py-lineno"> 629</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L630"></a><tt class="py-lineno"> 630</tt>  <tt class="py-line">        <tt class="py-comment"># Read the iterable into a list to protect against in-place changes</tt> </tt>
<a name="L631"></a><tt class="py-lineno"> 631</tt>  <tt class="py-line">        <tt id="link-171" class="py-name" targets="Function Bio.Phylo.NeXMLIO.matches()=Bio.Phylo.NeXMLIO-module.html#matches"><a title="Bio.Phylo.NeXMLIO.matches" class="py-name" href="#" onclick="return doclink('link-171', 'matches', 'link-171');">matches</a></tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-172" class="py-name"><a title="Bio.Phylo.BaseTree.TreeMixin.find_clades" class="py-name" href="#" onclick="return doclink('link-172', 'find_clades', 'link-95');">find_clades</a></tt><tt class="py-op">(</tt><tt class="py-name">target</tt><tt class="py-op">,</tt> <tt class="py-name">False</tt><tt class="py-op">,</tt> <tt class="py-string">'level'</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L632"></a><tt class="py-lineno"> 632</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-173" class="py-name"><a title="Bio.Phylo.NeXMLIO.matches" class="py-name" href="#" onclick="return doclink('link-173', 'matches', 'link-171');">matches</a></tt><tt class="py-op">:</tt> </tt>
<a name="L633"></a><tt class="py-lineno"> 633</tt>  <tt class="py-line">            <tt class="py-comment"># No matching nodes to collapse</tt> </tt>
<a name="L634"></a><tt class="py-lineno"> 634</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L635"></a><tt class="py-lineno"> 635</tt>  <tt class="py-line">        <tt class="py-comment"># Skip the root node -- it can't be collapsed</tt> </tt>
<a name="L636"></a><tt class="py-lineno"> 636</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-174" class="py-name"><a title="Bio.Phylo.NeXMLIO.matches" class="py-name" href="#" onclick="return doclink('link-174', 'matches', 'link-171');">matches</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-175" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-175', 'root', 'link-7');">root</a></tt><tt class="py-op">:</tt> </tt>
<a name="L637"></a><tt class="py-lineno"> 637</tt>  <tt class="py-line">            <tt id="link-176" class="py-name"><a title="Bio.Phylo.NeXMLIO.matches" class="py-name" href="#" onclick="return doclink('link-176', 'matches', 'link-171');">matches</a></tt><tt class="py-op">.</tt><tt id="link-177" class="py-name"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop" class="py-name" href="#" onclick="return doclink('link-177', 'pop', 'link-31');">pop</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L638"></a><tt class="py-lineno"> 638</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-178" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-178', 'clade', 'link-154');">clade</a></tt> <tt class="py-keyword">in</tt> <tt id="link-179" class="py-name"><a title="Bio.Phylo.NeXMLIO.matches" class="py-name" href="#" onclick="return doclink('link-179', 'matches', 'link-171');">matches</a></tt><tt class="py-op">:</tt> </tt>
<a name="L639"></a><tt class="py-lineno"> 639</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-180" class="py-name" targets="Method Bio.Nexus.Nodes.Chain.collapse()=Bio.Nexus.Nodes.Chain-class.html#collapse,Method Bio.Phylo.BaseTree.TreeMixin.collapse()=Bio.Phylo.BaseTree.TreeMixin-class.html#collapse"><a title="Bio.Nexus.Nodes.Chain.collapse
Bio.Phylo.BaseTree.TreeMixin.collapse" class="py-name" href="#" onclick="return doclink('link-180', 'collapse', 'link-180');">collapse</a></tt><tt class="py-op">(</tt><tt id="link-181" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-181', 'clade', 'link-154');">clade</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L640"></a><tt class="py-lineno"> 640</tt>  <tt class="py-line"> </tt>
<a name="TreeMixin.ladderize"></a><div id="TreeMixin.ladderize-def"><a name="L641"></a><tt class="py-lineno"> 641</tt> <a class="py-toggle" href="#" id="TreeMixin.ladderize-toggle" onclick="return toggle('TreeMixin.ladderize');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.TreeMixin-class.html#ladderize">ladderize</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">reverse</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="TreeMixin.ladderize-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="TreeMixin.ladderize-expanded"><a name="L642"></a><tt class="py-lineno"> 642</tt>  <tt class="py-line">        <tt class="py-docstring">"""Sort clades in-place according to the number of terminal nodes.</tt> </tt>
<a name="L643"></a><tt class="py-lineno"> 643</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L644"></a><tt class="py-lineno"> 644</tt>  <tt class="py-line"><tt class="py-docstring">        Deepest clades are last by default. Use ``reverse=True`` to sort clades</tt> </tt>
<a name="L645"></a><tt class="py-lineno"> 645</tt>  <tt class="py-line"><tt class="py-docstring">        deepest-to-shallowest.</tt> </tt>
<a name="L646"></a><tt class="py-lineno"> 646</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L647"></a><tt class="py-lineno"> 647</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-182" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-182', 'root', 'link-7');">root</a></tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">.</tt><tt id="link-183" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.sort()=Bio.Align.MultipleSeqAlignment-class.html#sort,Method Bio.Cluster.Tree.sort()=Bio.Cluster.Tree-class.html#sort,Method Bio.PDB.Residue.DisorderedResidue.sort()=Bio.PDB.Residue.DisorderedResidue-class.html#sort,Method Bio.PDB.Residue.Residue.sort()=Bio.PDB.Residue.Residue-class.html#sort,Method Bio.SearchIO._model.hit.Hit.sort()=Bio.SearchIO._model.hit.Hit-class.html#sort,Method Bio.SearchIO._model.query.QueryResult.sort()=Bio.SearchIO._model.query.QueryResult-class.html#sort,Method Bio.Sequencing.Ace.ACEFileRecord.sort()=Bio.Sequencing.Ace.ACEFileRecord-class.html#sort"><a title="Bio.Align.MultipleSeqAlignment.sort
Bio.Cluster.Tree.sort
Bio.PDB.Residue.DisorderedResidue.sort
Bio.PDB.Residue.Residue.sort
Bio.SearchIO._model.hit.Hit.sort
Bio.SearchIO._model.query.QueryResult.sort
Bio.Sequencing.Ace.ACEFileRecord.sort" class="py-name" href="#" onclick="return doclink('link-183', 'sort', 'link-183');">sort</a></tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">=</tt><tt class="py-keyword">lambda</tt> <tt class="py-name">c</tt><tt class="py-op">:</tt> <tt class="py-name">c</tt><tt class="py-op">.</tt><tt id="link-184" class="py-name" targets="Method Bio.Nexus.Trees.Tree.count_terminals()=Bio.Nexus.Trees.Tree-class.html#count_terminals,Method Bio.Phylo.BaseTree.TreeMixin.count_terminals()=Bio.Phylo.BaseTree.TreeMixin-class.html#count_terminals"><a title="Bio.Nexus.Trees.Tree.count_terminals
Bio.Phylo.BaseTree.TreeMixin.count_terminals" class="py-name" href="#" onclick="return doclink('link-184', 'count_terminals', 'link-184');">count_terminals</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L648"></a><tt class="py-lineno"> 648</tt>  <tt class="py-line">                              <tt id="link-185" class="py-name" targets="Method Bio.Pathway.Reaction.reverse()=Bio.Pathway.Reaction-class.html#reverse,Method Bio.Seq.MutableSeq.reverse()=Bio.Seq.MutableSeq-class.html#reverse"><a title="Bio.Pathway.Reaction.reverse
Bio.Seq.MutableSeq.reverse" class="py-name" href="#" onclick="return doclink('link-185', 'reverse', 'link-185');">reverse</a></tt><tt class="py-op">=</tt><tt id="link-186" class="py-name"><a title="Bio.Pathway.Reaction.reverse
Bio.Seq.MutableSeq.reverse" class="py-name" href="#" onclick="return doclink('link-186', 'reverse', 'link-185');">reverse</a></tt><tt class="py-op">)</tt> </tt>
<a name="L649"></a><tt class="py-lineno"> 649</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">subclade</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-187" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-187', 'root', 'link-7');">root</a></tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">:</tt> </tt>
<a name="L650"></a><tt class="py-lineno"> 650</tt>  <tt class="py-line">            <tt class="py-name">subclade</tt><tt class="py-op">.</tt><tt id="link-188" class="py-name" targets="Method Bio.Phylo.BaseTree.TreeMixin.ladderize()=Bio.Phylo.BaseTree.TreeMixin-class.html#ladderize"><a title="Bio.Phylo.BaseTree.TreeMixin.ladderize" class="py-name" href="#" onclick="return doclink('link-188', 'ladderize', 'link-188');">ladderize</a></tt><tt class="py-op">(</tt><tt id="link-189" class="py-name"><a title="Bio.Pathway.Reaction.reverse
Bio.Seq.MutableSeq.reverse" class="py-name" href="#" onclick="return doclink('link-189', 'reverse', 'link-185');">reverse</a></tt><tt class="py-op">=</tt><tt id="link-190" class="py-name"><a title="Bio.Pathway.Reaction.reverse
Bio.Seq.MutableSeq.reverse" class="py-name" href="#" onclick="return doclink('link-190', 'reverse', 'link-185');">reverse</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L651"></a><tt class="py-lineno"> 651</tt>  <tt class="py-line"> </tt>
<a name="TreeMixin.prune"></a><div id="TreeMixin.prune-def"><a name="L652"></a><tt class="py-lineno"> 652</tt> <a class="py-toggle" href="#" id="TreeMixin.prune-toggle" onclick="return toggle('TreeMixin.prune');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.TreeMixin-class.html#prune">prune</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">target</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="TreeMixin.prune-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="TreeMixin.prune-expanded"><a name="L653"></a><tt class="py-lineno"> 653</tt>  <tt class="py-line">        <tt class="py-docstring">"""Prunes a terminal clade from the tree.</tt> </tt>
<a name="L654"></a><tt class="py-lineno"> 654</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L655"></a><tt class="py-lineno"> 655</tt>  <tt class="py-line"><tt class="py-docstring">        If taxon is from a bifurcation, the connecting node will be collapsed</tt> </tt>
<a name="L656"></a><tt class="py-lineno"> 656</tt>  <tt class="py-line"><tt class="py-docstring">        and its branch length added to remaining terminal node. This might be no</tt> </tt>
<a name="L657"></a><tt class="py-lineno"> 657</tt>  <tt class="py-line"><tt class="py-docstring">        longer be a meaningful value.</tt> </tt>
<a name="L658"></a><tt class="py-lineno"> 658</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L659"></a><tt class="py-lineno"> 659</tt>  <tt class="py-line"><tt class="py-docstring">        :returns: parent clade of the pruned target</tt> </tt>
<a name="L660"></a><tt class="py-lineno"> 660</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L661"></a><tt class="py-lineno"> 661</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L662"></a><tt class="py-lineno"> 662</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">'terminal'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">kwargs</tt> <tt class="py-keyword">and</tt> <tt class="py-name">kwargs</tt><tt class="py-op">[</tt><tt class="py-string">'terminal'</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L663"></a><tt class="py-lineno"> 663</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"target must be terminal"</tt><tt class="py-op">)</tt> </tt>
<a name="L664"></a><tt class="py-lineno"> 664</tt>  <tt class="py-line">        <tt class="py-name">path</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-191" class="py-name"><a title="Bio.Phylo.BaseTree.TreeMixin.get_path" class="py-name" href="#" onclick="return doclink('link-191', 'get_path', 'link-99');">get_path</a></tt><tt class="py-op">(</tt><tt class="py-name">target</tt><tt class="py-op">,</tt> <tt class="py-name">terminal</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> </tt>
<a name="L665"></a><tt class="py-lineno"> 665</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">path</tt><tt class="py-op">:</tt> </tt>
<a name="L666"></a><tt class="py-lineno"> 666</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"can't find a matching target below this root"</tt><tt class="py-op">)</tt> </tt>
<a name="L667"></a><tt class="py-lineno"> 667</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">path</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L668"></a><tt class="py-lineno"> 668</tt>  <tt class="py-line">            <tt class="py-name">parent</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-192" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-192', 'root', 'link-7');">root</a></tt> </tt>
<a name="L669"></a><tt class="py-lineno"> 669</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L670"></a><tt class="py-lineno"> 670</tt>  <tt class="py-line">            <tt class="py-name">parent</tt> <tt class="py-op">=</tt> <tt class="py-name">path</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> </tt>
<a name="L671"></a><tt class="py-lineno"> 671</tt>  <tt class="py-line">        <tt class="py-name">parent</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">.</tt><tt id="link-193" class="py-name" targets="Method Bio.Crystal.Chain.remove()=Bio.Crystal.Chain-class.html#remove,Method Bio.Graphics.BasicChromosome._ChromosomeComponent.remove()=Bio.Graphics.BasicChromosome._ChromosomeComponent-class.html#remove,Method Bio.Restriction.Restriction.RestrictionBatch.remove()=Bio.Restriction.Restriction.RestrictionBatch-class.html#remove,Method Bio.Seq.MutableSeq.remove()=Bio.Seq.MutableSeq-class.html#remove,Method BioSQL.Loader.DatabaseRemover.remove()=BioSQL.Loader.DatabaseRemover-class.html#remove"><a title="Bio.Crystal.Chain.remove
Bio.Graphics.BasicChromosome._ChromosomeComponent.remove
Bio.Restriction.Restriction.RestrictionBatch.remove
Bio.Seq.MutableSeq.remove
BioSQL.Loader.DatabaseRemover.remove" class="py-name" href="#" onclick="return doclink('link-193', 'remove', 'link-193');">remove</a></tt><tt class="py-op">(</tt><tt class="py-name">path</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L672"></a><tt class="py-lineno"> 672</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">parent</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L673"></a><tt class="py-lineno"> 673</tt>  <tt class="py-line">            <tt class="py-comment"># We deleted a branch from a bifurcation</tt> </tt>
<a name="L674"></a><tt class="py-lineno"> 674</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">parent</tt> <tt class="py-op">==</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-194" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-194', 'root', 'link-7');">root</a></tt><tt class="py-op">:</tt> </tt>
<a name="L675"></a><tt class="py-lineno"> 675</tt>  <tt class="py-line">                <tt class="py-comment"># If we're at the root, move the root upwards</tt> </tt>
<a name="L676"></a><tt class="py-lineno"> 676</tt>  <tt class="py-line">                <tt class="py-comment"># NB: This loses the length of the original branch</tt> </tt>
<a name="L677"></a><tt class="py-lineno"> 677</tt>  <tt class="py-line">                <tt class="py-name">newroot</tt> <tt class="py-op">=</tt> <tt class="py-name">parent</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L678"></a><tt class="py-lineno"> 678</tt>  <tt class="py-line">                <tt class="py-name">newroot</tt><tt class="py-op">.</tt><tt id="link-195" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-195', 'branch_length', 'link-114');">branch_length</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L679"></a><tt class="py-lineno"> 679</tt>  <tt class="py-line">                <tt class="py-name">parent</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-196" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-196', 'root', 'link-7');">root</a></tt> <tt class="py-op">=</tt> <tt class="py-name">newroot</tt> </tt>
<a name="L680"></a><tt class="py-lineno"> 680</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L681"></a><tt class="py-lineno"> 681</tt>  <tt class="py-line">                <tt class="py-comment"># If we're not at the root, collapse this parent</tt> </tt>
<a name="L682"></a><tt class="py-lineno"> 682</tt>  <tt class="py-line">                <tt class="py-name">child</tt> <tt class="py-op">=</tt> <tt class="py-name">parent</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L683"></a><tt class="py-lineno"> 683</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">child</tt><tt class="py-op">.</tt><tt id="link-197" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-197', 'branch_length', 'link-114');">branch_length</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L684"></a><tt class="py-lineno"> 684</tt>  <tt class="py-line">                    <tt class="py-name">child</tt><tt class="py-op">.</tt><tt id="link-198" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-198', 'branch_length', 'link-114');">branch_length</a></tt> <tt class="py-op">+=</tt> <tt class="py-op">(</tt><tt class="py-name">parent</tt><tt class="py-op">.</tt><tt id="link-199" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-199', 'branch_length', 'link-114');">branch_length</a></tt> <tt class="py-keyword">or</tt> <tt class="py-number">0.0</tt><tt class="py-op">)</tt> </tt>
<a name="L685"></a><tt class="py-lineno"> 685</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">path</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">3</tt><tt class="py-op">:</tt> </tt>
<a name="L686"></a><tt class="py-lineno"> 686</tt>  <tt class="py-line">                    <tt class="py-name">grandparent</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-200" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-200', 'root', 'link-7');">root</a></tt> </tt>
<a name="L687"></a><tt class="py-lineno"> 687</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L688"></a><tt class="py-lineno"> 688</tt>  <tt class="py-line">                    <tt class="py-name">grandparent</tt> <tt class="py-op">=</tt> <tt class="py-name">path</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">3</tt><tt class="py-op">]</tt> </tt>
<a name="L689"></a><tt class="py-lineno"> 689</tt>  <tt class="py-line">                <tt class="py-comment"># Replace parent with child at the same place in grandparent</tt> </tt>
<a name="L690"></a><tt class="py-lineno"> 690</tt>  <tt class="py-line">                <tt id="link-201" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-201', 'index', 'link-167');">index</a></tt> <tt class="py-op">=</tt> <tt class="py-name">grandparent</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">.</tt><tt id="link-202" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-202', 'index', 'link-167');">index</a></tt><tt class="py-op">(</tt><tt class="py-name">parent</tt><tt class="py-op">)</tt> </tt>
<a name="L691"></a><tt class="py-lineno"> 691</tt>  <tt class="py-line">                <tt class="py-name">grandparent</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">.</tt><tt id="link-203" class="py-name"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop" class="py-name" href="#" onclick="return doclink('link-203', 'pop', 'link-31');">pop</a></tt><tt class="py-op">(</tt><tt id="link-204" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-204', 'index', 'link-167');">index</a></tt><tt class="py-op">)</tt> </tt>
<a name="L692"></a><tt class="py-lineno"> 692</tt>  <tt class="py-line">                <tt class="py-name">grandparent</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">.</tt><tt id="link-205" class="py-name" targets="Method Bio.Crystal.Chain.insert()=Bio.Crystal.Chain-class.html#insert,Method Bio.PDB.Entity.Entity.insert()=Bio.PDB.Entity.Entity-class.html#insert,Method Bio.Phylo.TreeConstruction._Matrix.insert()=Bio.Phylo.TreeConstruction._Matrix-class.html#insert,Method Bio.Seq.MutableSeq.insert()=Bio.Seq.MutableSeq-class.html#insert"><a title="Bio.Crystal.Chain.insert
Bio.PDB.Entity.Entity.insert
Bio.Phylo.TreeConstruction._Matrix.insert
Bio.Seq.MutableSeq.insert" class="py-name" href="#" onclick="return doclink('link-205', 'insert', 'link-205');">insert</a></tt><tt class="py-op">(</tt><tt id="link-206" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-206', 'index', 'link-167');">index</a></tt><tt class="py-op">,</tt> <tt class="py-name">child</tt><tt class="py-op">)</tt> </tt>
<a name="L693"></a><tt class="py-lineno"> 693</tt>  <tt class="py-line">                <tt class="py-name">parent</tt> <tt class="py-op">=</tt> <tt class="py-name">grandparent</tt> </tt>
<a name="L694"></a><tt class="py-lineno"> 694</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">parent</tt> </tt>
</div><a name="L695"></a><tt class="py-lineno"> 695</tt>  <tt class="py-line"> </tt>
<a name="TreeMixin.split"></a><div id="TreeMixin.split-def"><a name="L696"></a><tt class="py-lineno"> 696</tt> <a class="py-toggle" href="#" id="TreeMixin.split-toggle" onclick="return toggle('TreeMixin.split');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.TreeMixin-class.html#split">split</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">n</tt><tt class="py-op">=</tt><tt class="py-number">2</tt><tt class="py-op">,</tt> <tt class="py-param">branch_length</tt><tt class="py-op">=</tt><tt class="py-number">1.0</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="TreeMixin.split-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="TreeMixin.split-expanded"><a name="L697"></a><tt class="py-lineno"> 697</tt>  <tt class="py-line">        <tt class="py-docstring">"""Generate n (default 2) new descendants.</tt> </tt>
<a name="L698"></a><tt class="py-lineno"> 698</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L699"></a><tt class="py-lineno"> 699</tt>  <tt class="py-line"><tt class="py-docstring">        In a species tree, this is a speciation event.</tt> </tt>
<a name="L700"></a><tt class="py-lineno"> 700</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L701"></a><tt class="py-lineno"> 701</tt>  <tt class="py-line"><tt class="py-docstring">        New clades have the given branch_length and the same name as this</tt> </tt>
<a name="L702"></a><tt class="py-lineno"> 702</tt>  <tt class="py-line"><tt class="py-docstring">        clade's root plus an integer suffix (counting from 0). For example,</tt> </tt>
<a name="L703"></a><tt class="py-lineno"> 703</tt>  <tt class="py-line"><tt class="py-docstring">        splitting a clade named "A" produces sub-clades named "A0" and "A1".</tt> </tt>
<a name="L704"></a><tt class="py-lineno"> 704</tt>  <tt class="py-line"><tt class="py-docstring">        If the clade has no name, the prefix "n" is used for child nodes, e.g.</tt> </tt>
<a name="L705"></a><tt class="py-lineno"> 705</tt>  <tt class="py-line"><tt class="py-docstring">        "n0" and "n1".</tt> </tt>
<a name="L706"></a><tt class="py-lineno"> 706</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L707"></a><tt class="py-lineno"> 707</tt>  <tt class="py-line">        <tt class="py-name">clade_cls</tt> <tt class="py-op">=</tt> <tt id="link-207" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-207', 'type', 'link-39');">type</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-208" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-208', 'root', 'link-7');">root</a></tt><tt class="py-op">)</tt> </tt>
<a name="L708"></a><tt class="py-lineno"> 708</tt>  <tt class="py-line">        <tt class="py-name">base_name</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-209" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-209', 'root', 'link-7');">root</a></tt><tt class="py-op">.</tt><tt id="link-210" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-210', 'name', 'link-26');">name</a></tt> <tt class="py-keyword">or</tt> <tt class="py-string">'n'</tt> </tt>
<a name="L709"></a><tt class="py-lineno"> 709</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-211" class="py-name" targets="Variable Bio.PDB.Polypeptide.i=Bio.PDB.Polypeptide-module.html#i,Variable Bio.SeqIO.SffIO.i=Bio.SeqIO.SffIO-module.html#i"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-211', 'i', 'link-211');">i</a></tt> <tt class="py-keyword">in</tt> <tt id="link-212" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.range()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#range,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.range()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#range,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Track.Track.range()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#range"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-212', 'range', 'link-212');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">n</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L710"></a><tt class="py-lineno"> 710</tt>  <tt class="py-line">            <tt id="link-213" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-213', 'clade', 'link-154');">clade</a></tt> <tt class="py-op">=</tt> <tt class="py-name">clade_cls</tt><tt class="py-op">(</tt><tt id="link-214" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-214', 'name', 'link-26');">name</a></tt><tt class="py-op">=</tt><tt class="py-name">base_name</tt> <tt class="py-op">+</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-215" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-215', 'i', 'link-211');">i</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L711"></a><tt class="py-lineno"> 711</tt>  <tt class="py-line">                              <tt id="link-216" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-216', 'branch_length', 'link-114');">branch_length</a></tt><tt class="py-op">=</tt><tt id="link-217" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-217', 'branch_length', 'link-114');">branch_length</a></tt><tt class="py-op">)</tt> </tt>
<a name="L712"></a><tt class="py-lineno"> 712</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-218" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-218', 'root', 'link-7');">root</a></tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">.</tt><tt id="link-219" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-219', 'append', 'link-13');">append</a></tt><tt class="py-op">(</tt><tt id="link-220" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-220', 'clade', 'link-154');">clade</a></tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L713"></a><tt class="py-lineno"> 713</tt>  <tt class="py-line"> </tt>
<a name="Tree"></a><div id="Tree-def"><a name="L714"></a><tt class="py-lineno"> 714</tt>  <tt class="py-line"> </tt>
<a name="L715"></a><tt class="py-lineno"> 715</tt> <a class="py-toggle" href="#" id="Tree-toggle" onclick="return toggle('Tree');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.Tree-class.html">Tree</a><tt class="py-op">(</tt><tt class="py-base-class">TreeElement</tt><tt class="py-op">,</tt> <tt class="py-base-class">TreeMixin</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Tree-expanded"><a name="L716"></a><tt class="py-lineno"> 716</tt>  <tt class="py-line">    <tt class="py-docstring">"""A phylogenetic tree, containing global info for the phylogeny.</tt> </tt>
<a name="L717"></a><tt class="py-lineno"> 717</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L718"></a><tt class="py-lineno"> 718</tt>  <tt class="py-line"><tt class="py-docstring">    The structure and node-specific data is accessible through the 'root'</tt> </tt>
<a name="L719"></a><tt class="py-lineno"> 719</tt>  <tt class="py-line"><tt class="py-docstring">    clade attached to the Tree instance.</tt> </tt>
<a name="L720"></a><tt class="py-lineno"> 720</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L721"></a><tt class="py-lineno"> 721</tt>  <tt class="py-line"><tt class="py-docstring">    :Parameters:</tt> </tt>
<a name="L722"></a><tt class="py-lineno"> 722</tt>  <tt class="py-line"><tt class="py-docstring">        root : Clade</tt> </tt>
<a name="L723"></a><tt class="py-lineno"> 723</tt>  <tt class="py-line"><tt class="py-docstring">            The starting node of the tree. If the tree is rooted, this will</tt> </tt>
<a name="L724"></a><tt class="py-lineno"> 724</tt>  <tt class="py-line"><tt class="py-docstring">            usually be the root node.</tt> </tt>
<a name="L725"></a><tt class="py-lineno"> 725</tt>  <tt class="py-line"><tt class="py-docstring">        rooted : bool</tt> </tt>
<a name="L726"></a><tt class="py-lineno"> 726</tt>  <tt class="py-line"><tt class="py-docstring">            Whether or not the tree is rooted. By default, a tree is assumed to</tt> </tt>
<a name="L727"></a><tt class="py-lineno"> 727</tt>  <tt class="py-line"><tt class="py-docstring">            be rooted.</tt> </tt>
<a name="L728"></a><tt class="py-lineno"> 728</tt>  <tt class="py-line"><tt class="py-docstring">        id : str</tt> </tt>
<a name="L729"></a><tt class="py-lineno"> 729</tt>  <tt class="py-line"><tt class="py-docstring">            The identifier of the tree, if there is one.</tt> </tt>
<a name="L730"></a><tt class="py-lineno"> 730</tt>  <tt class="py-line"><tt class="py-docstring">        name : str</tt> </tt>
<a name="L731"></a><tt class="py-lineno"> 731</tt>  <tt class="py-line"><tt class="py-docstring">            The name of the tree, in essence a label.</tt> </tt>
<a name="L732"></a><tt class="py-lineno"> 732</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L733"></a><tt class="py-lineno"> 733</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L734"></a><tt class="py-lineno"> 734</tt>  <tt class="py-line"> </tt>
<a name="Tree.__init__"></a><div id="Tree.__init__-def"><a name="L735"></a><tt class="py-lineno"> 735</tt> <a class="py-toggle" href="#" id="Tree.__init__-toggle" onclick="return toggle('Tree.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.Tree-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">root</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">rooted</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">,</tt> <tt class="py-param">id</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Tree.__init__-expanded"><a name="L736"></a><tt class="py-lineno"> 736</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize parameter for phylogenetic tree."""</tt> </tt>
<a name="L737"></a><tt class="py-lineno"> 737</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-221" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-221', 'root', 'link-7');">root</a></tt> <tt class="py-op">=</tt> <tt id="link-222" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-222', 'root', 'link-7');">root</a></tt> <tt class="py-keyword">or</tt> <tt id="link-223" class="py-name"><a title="Bio.Phylo.BaseTree.Clade
Bio.Phylo.CDAO.Clade
Bio.Phylo.NeXML.Clade
Bio.Phylo.Newick.Clade
Bio.Phylo.PhyloXML.Clade" class="py-name" href="#" onclick="return doclink('link-223', 'Clade', 'link-23');">Clade</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L738"></a><tt class="py-lineno"> 738</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">rooted</tt> <tt class="py-op">=</tt> <tt class="py-name">rooted</tt> </tt>
<a name="L739"></a><tt class="py-lineno"> 739</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-224" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.id=Bio.KEGG.KGML.KGML_pathway.Component-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Entry.id=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.id=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#id,Variable Bio.PDB.Entity.Entity.id=Bio.PDB.Entity.Entity-class.html#id,Method Bio.Phylo.PhyloXMLIO.Parser.id()=Bio.Phylo.PhyloXMLIO.Parser-class.html#id,Method Bio.Phylo.PhyloXMLIO.Writer.id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#id,Variable Bio.SearchIO._model.hit.Hit.id=Bio.SearchIO._model.hit.Hit-class.html#id,Variable Bio.SearchIO._model.query.QueryResult.id=Bio.SearchIO._model.query.QueryResult-class.html#id"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-224', 'id', 'link-224');">id</a></tt> <tt class="py-op">=</tt> <tt id="link-225" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-225', 'id', 'link-224');">id</a></tt> </tt>
<a name="L740"></a><tt class="py-lineno"> 740</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-226" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-226', 'name', 'link-26');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-227" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-227', 'name', 'link-26');">name</a></tt> </tt>
</div><a name="L741"></a><tt class="py-lineno"> 741</tt>  <tt class="py-line"> </tt>
<a name="L742"></a><tt class="py-lineno"> 742</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">classmethod</tt> </tt>
<a name="Tree.from_clade"></a><div id="Tree.from_clade-def"><a name="L743"></a><tt class="py-lineno"> 743</tt> <a class="py-toggle" href="#" id="Tree.from_clade-toggle" onclick="return toggle('Tree.from_clade');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.Tree-class.html#from_clade">from_clade</a><tt class="py-op">(</tt><tt class="py-param">cls</tt><tt class="py-op">,</tt> <tt class="py-param">clade</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.from_clade-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Tree.from_clade-expanded"><a name="L744"></a><tt class="py-lineno"> 744</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a new Tree object given a clade.</tt> </tt>
<a name="L745"></a><tt class="py-lineno"> 745</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L746"></a><tt class="py-lineno"> 746</tt>  <tt class="py-line"><tt class="py-docstring">        Keyword arguments are the usual `Tree` constructor parameters.</tt> </tt>
<a name="L747"></a><tt class="py-lineno"> 747</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L748"></a><tt class="py-lineno"> 748</tt>  <tt class="py-line">        <tt id="link-228" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-228', 'root', 'link-7');">root</a></tt> <tt class="py-op">=</tt> <tt id="link-229" class="py-name"><a title="Bio.Crystal.Crystal.copy
Bio.File._IndexedSeqFileDict.copy
Bio.PDB.Atom.Atom.copy
Bio.PDB.Entity.Entity.copy
Bio.PDB.vectors.Vector.copy" class="py-name" href="#" onclick="return doclink('link-229', 'copy', 'link-3');">copy</a></tt><tt class="py-op">.</tt><tt class="py-name">deepcopy</tt><tt class="py-op">(</tt><tt id="link-230" class="py-name"><a title="Bio.Phylo.BaseTree.Tree.clade
Bio.Phylo.PhyloXMLIO.Writer.clade" class="py-name" href="#" onclick="return doclink('link-230', 'clade', 'link-154');">clade</a></tt><tt class="py-op">)</tt> </tt>
<a name="L749"></a><tt class="py-lineno"> 749</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">cls</tt><tt class="py-op">(</tt><tt id="link-231" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-231', 'root', 'link-7');">root</a></tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> </tt>
</div><a name="L750"></a><tt class="py-lineno"> 750</tt>  <tt class="py-line"> </tt>
<a name="L751"></a><tt class="py-lineno"> 751</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">classmethod</tt> </tt>
<a name="Tree.randomized"></a><div id="Tree.randomized-def"><a name="L752"></a><tt class="py-lineno"> 752</tt> <a class="py-toggle" href="#" id="Tree.randomized-toggle" onclick="return toggle('Tree.randomized');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.Tree-class.html#randomized">randomized</a><tt class="py-op">(</tt><tt class="py-param">cls</tt><tt class="py-op">,</tt> <tt class="py-param">taxa</tt><tt class="py-op">,</tt> <tt class="py-param">branch_length</tt><tt class="py-op">=</tt><tt class="py-number">1.0</tt><tt class="py-op">,</tt> <tt class="py-param">branch_stdev</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.randomized-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Tree.randomized-expanded"><a name="L753"></a><tt class="py-lineno"> 753</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a randomized bifurcating tree given a list of taxa.</tt> </tt>
<a name="L754"></a><tt class="py-lineno"> 754</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L755"></a><tt class="py-lineno"> 755</tt>  <tt class="py-line"><tt class="py-docstring">        :param taxa: Either an integer specifying the number of taxa to create</tt> </tt>
<a name="L756"></a><tt class="py-lineno"> 756</tt>  <tt class="py-line"><tt class="py-docstring">            (automatically named taxon#), or an iterable of taxon names, as</tt> </tt>
<a name="L757"></a><tt class="py-lineno"> 757</tt>  <tt class="py-line"><tt class="py-docstring">            strings.</tt> </tt>
<a name="L758"></a><tt class="py-lineno"> 758</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L759"></a><tt class="py-lineno"> 759</tt>  <tt class="py-line"><tt class="py-docstring">        :returns: a tree of the same type as this class.</tt> </tt>
<a name="L760"></a><tt class="py-lineno"> 760</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L761"></a><tt class="py-lineno"> 761</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L762"></a><tt class="py-lineno"> 762</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">taxa</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L763"></a><tt class="py-lineno"> 763</tt>  <tt class="py-line">            <tt class="py-name">taxa</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-string">'taxon%s'</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-232" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-232', 'i', 'link-211');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-233" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-233', 'i', 'link-211');">i</a></tt> <tt class="py-keyword">in</tt> <tt id="link-234" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-234', 'range', 'link-212');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">taxa</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L764"></a><tt class="py-lineno"> 764</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">taxa</tt><tt class="py-op">,</tt> <tt class="py-string">'__iter__'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L765"></a><tt class="py-lineno"> 765</tt>  <tt class="py-line">            <tt class="py-name">taxa</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">taxa</tt><tt class="py-op">)</tt> </tt>
<a name="L766"></a><tt class="py-lineno"> 766</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L767"></a><tt class="py-lineno"> 767</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"taxa argument must be integer (# taxa) or "</tt> </tt>
<a name="L768"></a><tt class="py-lineno"> 768</tt>  <tt class="py-line">                            <tt class="py-string">"iterable of taxon names."</tt><tt class="py-op">)</tt> </tt>
<a name="L769"></a><tt class="py-lineno"> 769</tt>  <tt class="py-line">        <tt class="py-name">rtree</tt> <tt class="py-op">=</tt> <tt class="py-name">cls</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L770"></a><tt class="py-lineno"> 770</tt>  <tt class="py-line">        <tt class="py-name">terminals</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">rtree</tt><tt class="py-op">.</tt><tt id="link-235" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-235', 'root', 'link-7');">root</a></tt><tt class="py-op">]</tt> </tt>
<a name="L771"></a><tt class="py-lineno"> 771</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">terminals</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">taxa</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L772"></a><tt class="py-lineno"> 772</tt>  <tt class="py-line">            <tt class="py-name">newsplit</tt> <tt class="py-op">=</tt> <tt class="py-name">random</tt><tt class="py-op">.</tt><tt class="py-name">choice</tt><tt class="py-op">(</tt><tt class="py-name">terminals</tt><tt class="py-op">)</tt> </tt>
<a name="L773"></a><tt class="py-lineno"> 773</tt>  <tt class="py-line">            <tt class="py-name">newsplit</tt><tt class="py-op">.</tt><tt id="link-236" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-236', 'split', 'link-236');">split</a></tt><tt class="py-op">(</tt><tt id="link-237" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-237', 'branch_length', 'link-114');">branch_length</a></tt><tt class="py-op">=</tt><tt id="link-238" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-238', 'branch_length', 'link-114');">branch_length</a></tt><tt class="py-op">)</tt> </tt>
<a name="L774"></a><tt class="py-lineno"> 774</tt>  <tt class="py-line">            <tt class="py-name">newterms</tt> <tt class="py-op">=</tt> <tt class="py-name">newsplit</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt> </tt>
<a name="L775"></a><tt class="py-lineno"> 775</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">branch_stdev</tt><tt class="py-op">:</tt> </tt>
<a name="L776"></a><tt class="py-lineno"> 776</tt>  <tt class="py-line">                <tt class="py-comment"># Add some noise to the branch lengths</tt> </tt>
<a name="L777"></a><tt class="py-lineno"> 777</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">nt</tt> <tt class="py-keyword">in</tt> <tt class="py-name">newterms</tt><tt class="py-op">:</tt> </tt>
<a name="L778"></a><tt class="py-lineno"> 778</tt>  <tt class="py-line">                    <tt class="py-name">nt</tt><tt class="py-op">.</tt><tt id="link-239" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-239', 'branch_length', 'link-114');">branch_length</a></tt> <tt class="py-op">=</tt> <tt id="link-240" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.max=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#max"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-240', 'max', 'link-240');">max</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> </tt>
<a name="L779"></a><tt class="py-lineno"> 779</tt>  <tt class="py-line">                                           <tt class="py-name">random</tt><tt class="py-op">.</tt><tt class="py-name">gauss</tt><tt class="py-op">(</tt><tt id="link-241" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-241', 'branch_length', 'link-114');">branch_length</a></tt><tt class="py-op">,</tt> <tt class="py-name">branch_stdev</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L780"></a><tt class="py-lineno"> 780</tt>  <tt class="py-line">            <tt class="py-name">terminals</tt><tt class="py-op">.</tt><tt id="link-242" class="py-name"><a title="Bio.Crystal.Chain.remove
Bio.Graphics.BasicChromosome._ChromosomeComponent.remove
Bio.Restriction.Restriction.RestrictionBatch.remove
Bio.Seq.MutableSeq.remove
BioSQL.Loader.DatabaseRemover.remove" class="py-name" href="#" onclick="return doclink('link-242', 'remove', 'link-193');">remove</a></tt><tt class="py-op">(</tt><tt class="py-name">newsplit</tt><tt class="py-op">)</tt> </tt>
<a name="L781"></a><tt class="py-lineno"> 781</tt>  <tt class="py-line">            <tt class="py-name">terminals</tt><tt class="py-op">.</tt><tt id="link-243" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-243', 'extend', 'link-8');">extend</a></tt><tt class="py-op">(</tt><tt class="py-name">newterms</tt><tt class="py-op">)</tt> </tt>
<a name="L782"></a><tt class="py-lineno"> 782</tt>  <tt class="py-line">        <tt class="py-comment"># Distribute taxon labels randomly</tt> </tt>
<a name="L783"></a><tt class="py-lineno"> 783</tt>  <tt class="py-line">        <tt class="py-name">random</tt><tt class="py-op">.</tt><tt class="py-name">shuffle</tt><tt class="py-op">(</tt><tt class="py-name">taxa</tt><tt class="py-op">)</tt> </tt>
<a name="L784"></a><tt class="py-lineno"> 784</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-244" class="py-name"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-244', 'node', 'link-18');">node</a></tt><tt class="py-op">,</tt> <tt id="link-245" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-245', 'name', 'link-26');">name</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">terminals</tt><tt class="py-op">,</tt> <tt class="py-name">taxa</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L785"></a><tt class="py-lineno"> 785</tt>  <tt class="py-line">            <tt id="link-246" class="py-name"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-246', 'node', 'link-18');">node</a></tt><tt class="py-op">.</tt><tt id="link-247" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-247', 'name', 'link-26');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-248" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-248', 'name', 'link-26');">name</a></tt> </tt>
<a name="L786"></a><tt class="py-lineno"> 786</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">rtree</tt> </tt>
</div><a name="L787"></a><tt class="py-lineno"> 787</tt>  <tt class="py-line"> </tt>
<a name="L788"></a><tt class="py-lineno"> 788</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="Tree.clade"></a><div id="Tree.clade-def"><a name="L789"></a><tt class="py-lineno"> 789</tt> <a class="py-toggle" href="#" id="Tree.clade-toggle" onclick="return toggle('Tree.clade');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.Tree-class.html#clade">clade</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.clade-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Tree.clade-expanded"><a name="L790"></a><tt class="py-lineno"> 790</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return first clade in this tree (not itself)."""</tt> </tt>
<a name="L791"></a><tt class="py-lineno"> 791</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-249" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-249', 'root', 'link-7');">root</a></tt> </tt>
</div><a name="L792"></a><tt class="py-lineno"> 792</tt>  <tt class="py-line"> </tt>
<a name="Tree.as_phyloxml"></a><div id="Tree.as_phyloxml-def"><a name="L793"></a><tt class="py-lineno"> 793</tt> <a class="py-toggle" href="#" id="Tree.as_phyloxml-toggle" onclick="return toggle('Tree.as_phyloxml');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.Tree-class.html#as_phyloxml">as_phyloxml</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.as_phyloxml-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Tree.as_phyloxml-expanded"><a name="L794"></a><tt class="py-lineno"> 794</tt>  <tt class="py-line">        <tt class="py-docstring">"""Convert this tree to a PhyloXML-compatible Phylogeny.</tt> </tt>
<a name="L795"></a><tt class="py-lineno"> 795</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L796"></a><tt class="py-lineno"> 796</tt>  <tt class="py-line"><tt class="py-docstring">        This lets you use the additional annotation types PhyloXML defines, and</tt> </tt>
<a name="L797"></a><tt class="py-lineno"> 797</tt>  <tt class="py-line"><tt class="py-docstring">        save this information when you write this tree as 'phyloxml'.</tt> </tt>
<a name="L798"></a><tt class="py-lineno"> 798</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L799"></a><tt class="py-lineno"> 799</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt id="link-250" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-250', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-251" class="py-name" targets="Package Bio.Phylo=Bio.Phylo-module.html"><a title="Bio.Phylo" class="py-name" href="#" onclick="return doclink('link-251', 'Phylo', 'link-251');">Phylo</a></tt><tt class="py-op">.</tt><tt id="link-252" class="py-name" targets="Module Bio.Phylo.PhyloXML=Bio.Phylo.PhyloXML-module.html"><a title="Bio.Phylo.PhyloXML" class="py-name" href="#" onclick="return doclink('link-252', 'PhyloXML', 'link-252');">PhyloXML</a></tt> <tt class="py-keyword">import</tt> <tt id="link-253" class="py-name" targets="Class Bio.Phylo.PhyloXML.Phylogeny=Bio.Phylo.PhyloXML.Phylogeny-class.html"><a title="Bio.Phylo.PhyloXML.Phylogeny" class="py-name" href="#" onclick="return doclink('link-253', 'Phylogeny', 'link-253');">Phylogeny</a></tt> </tt>
<a name="L800"></a><tt class="py-lineno"> 800</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-254" class="py-name"><a title="Bio.Phylo.PhyloXML.Phylogeny" class="py-name" href="#" onclick="return doclink('link-254', 'Phylogeny', 'link-253');">Phylogeny</a></tt><tt class="py-op">.</tt><tt id="link-255" class="py-name" targets="Class Method Bio.Phylo.PhyloXML.Phylogeny.from_tree()=Bio.Phylo.PhyloXML.Phylogeny-class.html#from_tree"><a title="Bio.Phylo.PhyloXML.Phylogeny.from_tree" class="py-name" href="#" onclick="return doclink('link-255', 'from_tree', 'link-255');">from_tree</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> </tt>
</div><a name="L801"></a><tt class="py-lineno"> 801</tt>  <tt class="py-line"> </tt>
<a name="L802"></a><tt class="py-lineno"> 802</tt>  <tt class="py-line">    <tt class="py-comment"># XXX Py3 Compatibility: In Python 3.0+, **kwargs can be replaced with the</tt> </tt>
<a name="L803"></a><tt class="py-lineno"> 803</tt>  <tt class="py-line">    <tt class="py-comment"># named keyword argument outgroup_branch_length=None</tt> </tt>
<a name="Tree.root_with_outgroup"></a><div id="Tree.root_with_outgroup-def"><a name="L804"></a><tt class="py-lineno"> 804</tt> <a class="py-toggle" href="#" id="Tree.root_with_outgroup-toggle" onclick="return toggle('Tree.root_with_outgroup');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.Tree-class.html#root_with_outgroup">root_with_outgroup</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">outgroup_targets</tt><tt class="py-op">,</tt> <tt class="py-op">*</tt><tt class="py-param">more_targets</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.root_with_outgroup-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Tree.root_with_outgroup-expanded"><a name="L805"></a><tt class="py-lineno"> 805</tt>  <tt class="py-line">        <tt class="py-docstring">"""Reroot this tree with the outgroup clade containing outgroup_targets.</tt> </tt>
<a name="L806"></a><tt class="py-lineno"> 806</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L807"></a><tt class="py-lineno"> 807</tt>  <tt class="py-line"><tt class="py-docstring">        Operates in-place.</tt> </tt>
<a name="L808"></a><tt class="py-lineno"> 808</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L809"></a><tt class="py-lineno"> 809</tt>  <tt class="py-line"><tt class="py-docstring">        Edge cases:</tt> </tt>
<a name="L810"></a><tt class="py-lineno"> 810</tt>  <tt class="py-line"><tt class="py-docstring">         - If ``outgroup == self.root``, no change</tt> </tt>
<a name="L811"></a><tt class="py-lineno"> 811</tt>  <tt class="py-line"><tt class="py-docstring">         - If outgroup is terminal, create new bifurcating root node with a</tt> </tt>
<a name="L812"></a><tt class="py-lineno"> 812</tt>  <tt class="py-line"><tt class="py-docstring">           0-length branch to the outgroup</tt> </tt>
<a name="L813"></a><tt class="py-lineno"> 813</tt>  <tt class="py-line"><tt class="py-docstring">         - If outgroup is internal, use the given outgroup node as the new</tt> </tt>
<a name="L814"></a><tt class="py-lineno"> 814</tt>  <tt class="py-line"><tt class="py-docstring">           trifurcating root, keeping branches the same</tt> </tt>
<a name="L815"></a><tt class="py-lineno"> 815</tt>  <tt class="py-line"><tt class="py-docstring">         - If the original root was bifurcating, drop it from the tree,</tt> </tt>
<a name="L816"></a><tt class="py-lineno"> 816</tt>  <tt class="py-line"><tt class="py-docstring">           preserving total branch lengths</tt> </tt>
<a name="L817"></a><tt class="py-lineno"> 817</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L818"></a><tt class="py-lineno"> 818</tt>  <tt class="py-line"><tt class="py-docstring">        :param outgroup_branch_length: length of the branch leading to the</tt> </tt>
<a name="L819"></a><tt class="py-lineno"> 819</tt>  <tt class="py-line"><tt class="py-docstring">            outgroup after rerooting. If not specified (None), then:</tt> </tt>
<a name="L820"></a><tt class="py-lineno"> 820</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L821"></a><tt class="py-lineno"> 821</tt>  <tt class="py-line"><tt class="py-docstring">            - If the outgroup is an internal node (not a single terminal taxon),</tt> </tt>
<a name="L822"></a><tt class="py-lineno"> 822</tt>  <tt class="py-line"><tt class="py-docstring">              then use that node as the new root.</tt> </tt>
<a name="L823"></a><tt class="py-lineno"> 823</tt>  <tt class="py-line"><tt class="py-docstring">            - Otherwise, create a new root node as the parent of the outgroup.</tt> </tt>
<a name="L824"></a><tt class="py-lineno"> 824</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L825"></a><tt class="py-lineno"> 825</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L826"></a><tt class="py-lineno"> 826</tt>  <tt class="py-line">        <tt class="py-comment"># This raises a ValueError if any target is not in this tree</tt> </tt>
<a name="L827"></a><tt class="py-lineno"> 827</tt>  <tt class="py-line">        <tt class="py-comment"># Otherwise, common_ancestor guarantees outgroup is in this tree</tt> </tt>
<a name="L828"></a><tt class="py-lineno"> 828</tt>  <tt class="py-line">        <tt class="py-name">outgroup</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-256" class="py-name"><a title="Bio.Nexus.Trees.Tree.common_ancestor
Bio.Phylo.BaseTree.TreeMixin.common_ancestor" class="py-name" href="#" onclick="return doclink('link-256', 'common_ancestor', 'link-97');">common_ancestor</a></tt><tt class="py-op">(</tt><tt class="py-name">outgroup_targets</tt><tt class="py-op">,</tt> <tt class="py-op">*</tt><tt class="py-name">more_targets</tt><tt class="py-op">)</tt> </tt>
<a name="L829"></a><tt class="py-lineno"> 829</tt>  <tt class="py-line">        <tt class="py-name">outgroup_path</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-257" class="py-name"><a title="Bio.Phylo.BaseTree.TreeMixin.get_path" class="py-name" href="#" onclick="return doclink('link-257', 'get_path', 'link-99');">get_path</a></tt><tt class="py-op">(</tt><tt class="py-name">outgroup</tt><tt class="py-op">)</tt> </tt>
<a name="L830"></a><tt class="py-lineno"> 830</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">outgroup_path</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L831"></a><tt class="py-lineno"> 831</tt>  <tt class="py-line">            <tt class="py-comment"># Outgroup is the current root -- no change</tt> </tt>
<a name="L832"></a><tt class="py-lineno"> 832</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L833"></a><tt class="py-lineno"> 833</tt>  <tt class="py-line"> </tt>
<a name="L834"></a><tt class="py-lineno"> 834</tt>  <tt class="py-line">        <tt class="py-name">prev_blen</tt> <tt class="py-op">=</tt> <tt class="py-name">outgroup</tt><tt class="py-op">.</tt><tt id="link-258" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-258', 'branch_length', 'link-114');">branch_length</a></tt> <tt class="py-keyword">or</tt> <tt class="py-number">0.0</tt> </tt>
<a name="L835"></a><tt class="py-lineno"> 835</tt>  <tt class="py-line">        <tt class="py-comment"># Hideous kludge because Py2.x doesn't allow keyword args after *args</tt> </tt>
<a name="L836"></a><tt class="py-lineno"> 836</tt>  <tt class="py-line">        <tt class="py-name">outgroup_branch_length</tt> <tt class="py-op">=</tt> <tt class="py-name">kwargs</tt><tt class="py-op">.</tt><tt id="link-259" class="py-name" targets="Method Bio.Crystal.Crystal.get()=Bio.Crystal.Crystal-class.html#get,Method Bio.Data.CodonTable.AmbiguousForwardTable.get()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#get,Method Bio.File._IndexedSeqFileDict.get()=Bio.File._IndexedSeqFileDict-class.html#get,Method Bio.File._IndexedSeqFileProxy.get()=Bio.File._IndexedSeqFileProxy-class.html#get,Method Bio.File._SQLiteManySeqFilesDict.get()=Bio.File._SQLiteManySeqFilesDict-class.html#get,Method Bio.Restriction.Restriction.RestrictionBatch.get()=Bio.Restriction.Restriction.RestrictionBatch-class.html#get,Method Bio.SearchIO._index.SearchIndexer.get()=Bio.SearchIO._index.SearchIndexer-class.html#get,Method Bio.SeqIO._index.SeqFileRandomAccess.get()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#get,Method Bio.SeqIO._index.SffRandomAccess.get()=Bio.SeqIO._index.SffRandomAccess-class.html#get,Method Bio.SeqIO._index.SffTrimedRandomAccess.get()=Bio.SeqIO._index.SffTrimedRandomAccess-class.html#get,Method Bio.SeqIO._index.UniprotRandomAccess.get()=Bio.SeqIO._index.UniprotRandomAccess-class.html#get"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-259', 'get', 'link-259');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'outgroup_branch_length'</tt><tt class="py-op">)</tt> </tt>
<a name="L837"></a><tt class="py-lineno"> 837</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">outgroup_branch_length</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L838"></a><tt class="py-lineno"> 838</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-op">(</tt><tt class="py-number">0</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">outgroup_branch_length</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">prev_blen</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L839"></a><tt class="py-lineno"> 839</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"outgroup_branch_length must be between 0 "</tt> </tt>
<a name="L840"></a><tt class="py-lineno"> 840</tt>  <tt class="py-line">                                 <tt class="py-string">"and the original length of the branch "</tt> </tt>
<a name="L841"></a><tt class="py-lineno"> 841</tt>  <tt class="py-line">                                 <tt class="py-string">"leading to the outgroup."</tt><tt class="py-op">)</tt> </tt>
<a name="L842"></a><tt class="py-lineno"> 842</tt>  <tt class="py-line"> </tt>
<a name="L843"></a><tt class="py-lineno"> 843</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">outgroup</tt><tt class="py-op">.</tt><tt id="link-260" class="py-name"><a title="Bio.Nexus.Trees.Tree.is_terminal
Bio.Phylo.BaseTree.Clade.is_terminal
Bio.Phylo.BaseTree.Tree.is_terminal" class="py-name" href="#" onclick="return doclink('link-260', 'is_terminal', 'link-34');">is_terminal</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">outgroup_branch_length</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L844"></a><tt class="py-lineno"> 844</tt>  <tt class="py-line">            <tt class="py-comment"># Create a new root with a 0-length branch to the outgroup</tt> </tt>
<a name="L845"></a><tt class="py-lineno"> 845</tt>  <tt class="py-line">            <tt class="py-name">outgroup</tt><tt class="py-op">.</tt><tt id="link-261" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-261', 'branch_length', 'link-114');">branch_length</a></tt> <tt class="py-op">=</tt> <tt class="py-name">outgroup_branch_length</tt> <tt class="py-keyword">or</tt> <tt class="py-number">0.0</tt> </tt>
<a name="L846"></a><tt class="py-lineno"> 846</tt>  <tt class="py-line">            <tt class="py-name">new_root</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-262" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-262', 'root', 'link-7');">root</a></tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt> </tt>
<a name="L847"></a><tt class="py-lineno"> 847</tt>  <tt class="py-line">                <tt id="link-263" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-263', 'branch_length', 'link-114');">branch_length</a></tt><tt class="py-op">=</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-264" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-264', 'root', 'link-7');">root</a></tt><tt class="py-op">.</tt><tt id="link-265" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-265', 'branch_length', 'link-114');">branch_length</a></tt><tt class="py-op">,</tt> <tt class="py-name">clades</tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-name">outgroup</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L848"></a><tt class="py-lineno"> 848</tt>  <tt class="py-line">            <tt class="py-comment"># The first branch reversal (see the upcoming loop) is modified</tt> </tt>
<a name="L849"></a><tt class="py-lineno"> 849</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">outgroup_path</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L850"></a><tt class="py-lineno"> 850</tt>  <tt class="py-line">                <tt class="py-comment"># No nodes between the original root and outgroup to rearrange.</tt> </tt>
<a name="L851"></a><tt class="py-lineno"> 851</tt>  <tt class="py-line">                <tt class="py-comment"># Most of the code below will be skipped, but we still need</tt> </tt>
<a name="L852"></a><tt class="py-lineno"> 852</tt>  <tt class="py-line">                <tt class="py-comment"># 'new_parent' pointing at the new root.</tt> </tt>
<a name="L853"></a><tt class="py-lineno"> 853</tt>  <tt class="py-line">                <tt class="py-name">new_parent</tt> <tt class="py-op">=</tt> <tt class="py-name">new_root</tt> </tt>
<a name="L854"></a><tt class="py-lineno"> 854</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L855"></a><tt class="py-lineno"> 855</tt>  <tt class="py-line">                <tt class="py-name">parent</tt> <tt class="py-op">=</tt> <tt class="py-name">outgroup_path</tt><tt class="py-op">.</tt><tt id="link-266" class="py-name"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop" class="py-name" href="#" onclick="return doclink('link-266', 'pop', 'link-31');">pop</a></tt><tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">2</tt><tt class="py-op">)</tt> </tt>
<a name="L856"></a><tt class="py-lineno"> 856</tt>  <tt class="py-line">                <tt class="py-comment"># First iteration of reversing the path to the outgroup</tt> </tt>
<a name="L857"></a><tt class="py-lineno"> 857</tt>  <tt class="py-line">                <tt class="py-name">parent</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">.</tt><tt id="link-267" class="py-name"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop" class="py-name" href="#" onclick="return doclink('link-267', 'pop', 'link-31');">pop</a></tt><tt class="py-op">(</tt><tt class="py-name">parent</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">.</tt><tt id="link-268" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-268', 'index', 'link-167');">index</a></tt><tt class="py-op">(</tt><tt class="py-name">outgroup</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L858"></a><tt class="py-lineno"> 858</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-name">prev_blen</tt><tt class="py-op">,</tt> <tt class="py-name">parent</tt><tt class="py-op">.</tt><tt id="link-269" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-269', 'branch_length', 'link-114');">branch_length</a></tt><tt class="py-op">)</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">parent</tt><tt class="py-op">.</tt><tt id="link-270" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-270', 'branch_length', 'link-114');">branch_length</a></tt><tt class="py-op">,</tt> </tt>
<a name="L859"></a><tt class="py-lineno"> 859</tt>  <tt class="py-line">                                                     <tt class="py-name">prev_blen</tt> <tt class="py-op">-</tt> <tt class="py-name">outgroup</tt><tt class="py-op">.</tt><tt id="link-271" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-271', 'branch_length', 'link-114');">branch_length</a></tt><tt class="py-op">)</tt> </tt>
<a name="L860"></a><tt class="py-lineno"> 860</tt>  <tt class="py-line">                <tt class="py-name">new_root</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">.</tt><tt id="link-272" class="py-name"><a title="Bio.Crystal.Chain.insert
Bio.PDB.Entity.Entity.insert
Bio.Phylo.TreeConstruction._Matrix.insert
Bio.Seq.MutableSeq.insert" class="py-name" href="#" onclick="return doclink('link-272', 'insert', 'link-205');">insert</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">parent</tt><tt class="py-op">)</tt> </tt>
<a name="L861"></a><tt class="py-lineno"> 861</tt>  <tt class="py-line">                <tt class="py-name">new_parent</tt> <tt class="py-op">=</tt> <tt class="py-name">parent</tt> </tt>
<a name="L862"></a><tt class="py-lineno"> 862</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L863"></a><tt class="py-lineno"> 863</tt>  <tt class="py-line">            <tt class="py-comment"># Use the given outgroup node as the new (trifurcating) root</tt> </tt>
<a name="L864"></a><tt class="py-lineno"> 864</tt>  <tt class="py-line">            <tt class="py-name">new_root</tt> <tt class="py-op">=</tt> <tt class="py-name">outgroup</tt> </tt>
<a name="L865"></a><tt class="py-lineno"> 865</tt>  <tt class="py-line">            <tt class="py-name">new_root</tt><tt class="py-op">.</tt><tt id="link-273" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-273', 'branch_length', 'link-114');">branch_length</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-274" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-274', 'root', 'link-7');">root</a></tt><tt class="py-op">.</tt><tt id="link-275" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-275', 'branch_length', 'link-114');">branch_length</a></tt> </tt>
<a name="L866"></a><tt class="py-lineno"> 866</tt>  <tt class="py-line">            <tt class="py-name">new_parent</tt> <tt class="py-op">=</tt> <tt class="py-name">new_root</tt> </tt>
<a name="L867"></a><tt class="py-lineno"> 867</tt>  <tt class="py-line"> </tt>
<a name="L868"></a><tt class="py-lineno"> 868</tt>  <tt class="py-line">        <tt class="py-comment"># Tracing the outgroup lineage backwards, reattach the subclades under a</tt> </tt>
<a name="L869"></a><tt class="py-lineno"> 869</tt>  <tt class="py-line">        <tt class="py-comment"># new root clade. Reverse the branches directly above the outgroup in</tt> </tt>
<a name="L870"></a><tt class="py-lineno"> 870</tt>  <tt class="py-line">        <tt class="py-comment"># the tree, but keep the descendants of those clades as they are.</tt> </tt>
<a name="L871"></a><tt class="py-lineno"> 871</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">parent</tt> <tt class="py-keyword">in</tt> <tt class="py-name">outgroup_path</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">2</tt><tt class="py-op">:</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L872"></a><tt class="py-lineno"> 872</tt>  <tt class="py-line">            <tt class="py-name">parent</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">.</tt><tt id="link-276" class="py-name"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop" class="py-name" href="#" onclick="return doclink('link-276', 'pop', 'link-31');">pop</a></tt><tt class="py-op">(</tt><tt class="py-name">parent</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">.</tt><tt id="link-277" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-277', 'index', 'link-167');">index</a></tt><tt class="py-op">(</tt><tt class="py-name">new_parent</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L873"></a><tt class="py-lineno"> 873</tt>  <tt class="py-line">            <tt class="py-name">prev_blen</tt><tt class="py-op">,</tt> <tt class="py-name">parent</tt><tt class="py-op">.</tt><tt id="link-278" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-278', 'branch_length', 'link-114');">branch_length</a></tt> <tt class="py-op">=</tt> <tt class="py-name">parent</tt><tt class="py-op">.</tt><tt id="link-279" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-279', 'branch_length', 'link-114');">branch_length</a></tt><tt class="py-op">,</tt> <tt class="py-name">prev_blen</tt> </tt>
<a name="L874"></a><tt class="py-lineno"> 874</tt>  <tt class="py-line">            <tt class="py-name">new_parent</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">.</tt><tt id="link-280" class="py-name"><a title="Bio.Crystal.Chain.insert
Bio.PDB.Entity.Entity.insert
Bio.Phylo.TreeConstruction._Matrix.insert
Bio.Seq.MutableSeq.insert" class="py-name" href="#" onclick="return doclink('link-280', 'insert', 'link-205');">insert</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">parent</tt><tt class="py-op">)</tt> </tt>
<a name="L875"></a><tt class="py-lineno"> 875</tt>  <tt class="py-line">            <tt class="py-name">new_parent</tt> <tt class="py-op">=</tt> <tt class="py-name">parent</tt> </tt>
<a name="L876"></a><tt class="py-lineno"> 876</tt>  <tt class="py-line"> </tt>
<a name="L877"></a><tt class="py-lineno"> 877</tt>  <tt class="py-line">        <tt class="py-comment"># Finally, handle the original root according to number of descendents</tt> </tt>
<a name="L878"></a><tt class="py-lineno"> 878</tt>  <tt class="py-line">        <tt class="py-name">old_root</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-281" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-281', 'root', 'link-7');">root</a></tt> </tt>
<a name="L879"></a><tt class="py-lineno"> 879</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">outgroup</tt> <tt class="py-keyword">in</tt> <tt class="py-name">old_root</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">:</tt> </tt>
<a name="L880"></a><tt class="py-lineno"> 880</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">outgroup_path</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt> </tt>
<a name="L881"></a><tt class="py-lineno"> 881</tt>  <tt class="py-line">            <tt class="py-name">old_root</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">.</tt><tt id="link-282" class="py-name"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop" class="py-name" href="#" onclick="return doclink('link-282', 'pop', 'link-31');">pop</a></tt><tt class="py-op">(</tt><tt class="py-name">old_root</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">.</tt><tt id="link-283" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-283', 'index', 'link-167');">index</a></tt><tt class="py-op">(</tt><tt class="py-name">outgroup</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L882"></a><tt class="py-lineno"> 882</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L883"></a><tt class="py-lineno"> 883</tt>  <tt class="py-line">            <tt class="py-name">old_root</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">.</tt><tt id="link-284" class="py-name"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop" class="py-name" href="#" onclick="return doclink('link-284', 'pop', 'link-31');">pop</a></tt><tt class="py-op">(</tt><tt class="py-name">old_root</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">.</tt><tt id="link-285" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-285', 'index', 'link-167');">index</a></tt><tt class="py-op">(</tt><tt class="py-name">new_parent</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L884"></a><tt class="py-lineno"> 884</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">old_root</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L885"></a><tt class="py-lineno"> 885</tt>  <tt class="py-line">            <tt class="py-comment"># Delete the old bifurcating root &amp; add branch lengths</tt> </tt>
<a name="L886"></a><tt class="py-lineno"> 886</tt>  <tt class="py-line">            <tt class="py-name">ingroup</tt> <tt class="py-op">=</tt> <tt class="py-name">old_root</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L887"></a><tt class="py-lineno"> 887</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">ingroup</tt><tt class="py-op">.</tt><tt id="link-286" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-286', 'branch_length', 'link-114');">branch_length</a></tt><tt class="py-op">:</tt> </tt>
<a name="L888"></a><tt class="py-lineno"> 888</tt>  <tt class="py-line">                <tt class="py-name">ingroup</tt><tt class="py-op">.</tt><tt id="link-287" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-287', 'branch_length', 'link-114');">branch_length</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">prev_blen</tt> </tt>
<a name="L889"></a><tt class="py-lineno"> 889</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L890"></a><tt class="py-lineno"> 890</tt>  <tt class="py-line">                <tt class="py-name">ingroup</tt><tt class="py-op">.</tt><tt id="link-288" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-288', 'branch_length', 'link-114');">branch_length</a></tt> <tt class="py-op">=</tt> <tt class="py-name">prev_blen</tt> </tt>
<a name="L891"></a><tt class="py-lineno"> 891</tt>  <tt class="py-line">            <tt class="py-name">new_parent</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">.</tt><tt id="link-289" class="py-name"><a title="Bio.Crystal.Chain.insert
Bio.PDB.Entity.Entity.insert
Bio.Phylo.TreeConstruction._Matrix.insert
Bio.Seq.MutableSeq.insert" class="py-name" href="#" onclick="return doclink('link-289', 'insert', 'link-205');">insert</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">ingroup</tt><tt class="py-op">)</tt> </tt>
<a name="L892"></a><tt class="py-lineno"> 892</tt>  <tt class="py-line">            <tt class="py-comment"># ENH: If annotations are attached to old_root, do... something.</tt> </tt>
<a name="L893"></a><tt class="py-lineno"> 893</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L894"></a><tt class="py-lineno"> 894</tt>  <tt class="py-line">            <tt class="py-comment"># Keep the old trifurcating/polytomous root as an internal node</tt> </tt>
<a name="L895"></a><tt class="py-lineno"> 895</tt>  <tt class="py-line">            <tt class="py-name">old_root</tt><tt class="py-op">.</tt><tt id="link-290" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-290', 'branch_length', 'link-114');">branch_length</a></tt> <tt class="py-op">=</tt> <tt class="py-name">prev_blen</tt> </tt>
<a name="L896"></a><tt class="py-lineno"> 896</tt>  <tt class="py-line">            <tt class="py-name">new_parent</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">.</tt><tt id="link-291" class="py-name"><a title="Bio.Crystal.Chain.insert
Bio.PDB.Entity.Entity.insert
Bio.Phylo.TreeConstruction._Matrix.insert
Bio.Seq.MutableSeq.insert" class="py-name" href="#" onclick="return doclink('link-291', 'insert', 'link-205');">insert</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">old_root</tt><tt class="py-op">)</tt> </tt>
<a name="L897"></a><tt class="py-lineno"> 897</tt>  <tt class="py-line"> </tt>
<a name="L898"></a><tt class="py-lineno"> 898</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-292" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-292', 'root', 'link-7');">root</a></tt> <tt class="py-op">=</tt> <tt class="py-name">new_root</tt> </tt>
<a name="L899"></a><tt class="py-lineno"> 899</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">rooted</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L900"></a><tt class="py-lineno"> 900</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> </tt>
</div><a name="L901"></a><tt class="py-lineno"> 901</tt>  <tt class="py-line"> </tt>
<a name="Tree.root_at_midpoint"></a><div id="Tree.root_at_midpoint-def"><a name="L902"></a><tt class="py-lineno"> 902</tt> <a class="py-toggle" href="#" id="Tree.root_at_midpoint-toggle" onclick="return toggle('Tree.root_at_midpoint');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.Tree-class.html#root_at_midpoint">root_at_midpoint</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.root_at_midpoint-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Tree.root_at_midpoint-expanded"><a name="L903"></a><tt class="py-lineno"> 903</tt>  <tt class="py-line">        <tt class="py-docstring">"""Root the tree at the midpoint of the two most distant taxa.</tt> </tt>
<a name="L904"></a><tt class="py-lineno"> 904</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L905"></a><tt class="py-lineno"> 905</tt>  <tt class="py-line"><tt class="py-docstring">        This operates in-place, leaving a bifurcating root. The topology of the</tt> </tt>
<a name="L906"></a><tt class="py-lineno"> 906</tt>  <tt class="py-line"><tt class="py-docstring">        tree is otherwise retained, though no guarantees are made about the</tt> </tt>
<a name="L907"></a><tt class="py-lineno"> 907</tt>  <tt class="py-line"><tt class="py-docstring">        stability of clade/node/taxon ordering.</tt> </tt>
<a name="L908"></a><tt class="py-lineno"> 908</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L909"></a><tt class="py-lineno"> 909</tt>  <tt class="py-line">        <tt class="py-comment"># Identify the largest pairwise distance</tt> </tt>
<a name="L910"></a><tt class="py-lineno"> 910</tt>  <tt class="py-line">        <tt class="py-name">max_distance</tt> <tt class="py-op">=</tt> <tt class="py-number">0.0</tt> </tt>
<a name="L911"></a><tt class="py-lineno"> 911</tt>  <tt class="py-line">        <tt class="py-name">tips</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-293" class="py-name"><a title="Bio.Nexus.Trees.Tree.get_terminals
Bio.Phylo.BaseTree.TreeMixin.get_terminals" class="py-name" href="#" onclick="return doclink('link-293', 'get_terminals', 'link-148');">get_terminals</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L912"></a><tt class="py-lineno"> 912</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">tip</tt> <tt class="py-keyword">in</tt> <tt class="py-name">tips</tt><tt class="py-op">:</tt> </tt>
<a name="L913"></a><tt class="py-lineno"> 913</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-294" class="py-name" targets="Method Bio.Nexus.Trees.Tree.root_with_outgroup()=Bio.Nexus.Trees.Tree-class.html#root_with_outgroup,Method Bio.Phylo.BaseTree.Tree.root_with_outgroup()=Bio.Phylo.BaseTree.Tree-class.html#root_with_outgroup"><a title="Bio.Nexus.Trees.Tree.root_with_outgroup
Bio.Phylo.BaseTree.Tree.root_with_outgroup" class="py-name" href="#" onclick="return doclink('link-294', 'root_with_outgroup', 'link-294');">root_with_outgroup</a></tt><tt class="py-op">(</tt><tt class="py-name">tip</tt><tt class="py-op">)</tt> </tt>
<a name="L914"></a><tt class="py-lineno"> 914</tt>  <tt class="py-line">            <tt class="py-name">new_max</tt> <tt class="py-op">=</tt> <tt id="link-295" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-295', 'max', 'link-240');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-296" class="py-name"><a title="Bio.Phylo.BaseTree.TreeMixin.depths" class="py-name" href="#" onclick="return doclink('link-296', 'depths', 'link-115');">depths</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-297" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-297', 'items', 'link-11');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">key</tt><tt class="py-op">=</tt><tt class="py-keyword">lambda</tt> <tt class="py-name">nd</tt><tt class="py-op">:</tt> <tt class="py-name">nd</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L915"></a><tt class="py-lineno"> 915</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">new_max</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">&gt;</tt> <tt class="py-name">max_distance</tt><tt class="py-op">:</tt> </tt>
<a name="L916"></a><tt class="py-lineno"> 916</tt>  <tt class="py-line">                <tt class="py-name">tip1</tt> <tt class="py-op">=</tt> <tt class="py-name">tip</tt> </tt>
<a name="L917"></a><tt class="py-lineno"> 917</tt>  <tt class="py-line">                <tt class="py-name">tip2</tt> <tt class="py-op">=</tt> <tt class="py-name">new_max</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L918"></a><tt class="py-lineno"> 918</tt>  <tt class="py-line">                <tt class="py-name">max_distance</tt> <tt class="py-op">=</tt> <tt class="py-name">new_max</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L919"></a><tt class="py-lineno"> 919</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-298" class="py-name"><a title="Bio.Nexus.Trees.Tree.root_with_outgroup
Bio.Phylo.BaseTree.Tree.root_with_outgroup" class="py-name" href="#" onclick="return doclink('link-298', 'root_with_outgroup', 'link-294');">root_with_outgroup</a></tt><tt class="py-op">(</tt><tt class="py-name">tip1</tt><tt class="py-op">)</tt> </tt>
<a name="L920"></a><tt class="py-lineno"> 920</tt>  <tt class="py-line">        <tt class="py-comment"># Depth to go from the ingroup tip toward the outgroup tip</tt> </tt>
<a name="L921"></a><tt class="py-lineno"> 921</tt>  <tt class="py-line">        <tt class="py-name">root_remainder</tt> <tt class="py-op">=</tt> <tt class="py-number">0.5</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-name">max_distance</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-299" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-299', 'root', 'link-7');">root</a></tt><tt class="py-op">.</tt><tt id="link-300" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-300', 'branch_length', 'link-114');">branch_length</a></tt> <tt class="py-keyword">or</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L922"></a><tt class="py-lineno"> 922</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">root_remainder</tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">0</tt> </tt>
<a name="L923"></a><tt class="py-lineno"> 923</tt>  <tt class="py-line">        <tt class="py-comment"># Identify the midpoint and reroot there.</tt> </tt>
<a name="L924"></a><tt class="py-lineno"> 924</tt>  <tt class="py-line">        <tt class="py-comment"># Trace the path to the outgroup tip until all of the root depth has</tt> </tt>
<a name="L925"></a><tt class="py-lineno"> 925</tt>  <tt class="py-line">        <tt class="py-comment"># been traveled/accounted for.</tt> </tt>
<a name="L926"></a><tt class="py-lineno"> 926</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-301" class="py-name"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-301', 'node', 'link-18');">node</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-302" class="py-name"><a title="Bio.Phylo.BaseTree.TreeMixin.get_path" class="py-name" href="#" onclick="return doclink('link-302', 'get_path', 'link-99');">get_path</a></tt><tt class="py-op">(</tt><tt class="py-name">tip2</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L927"></a><tt class="py-lineno"> 927</tt>  <tt class="py-line">            <tt class="py-name">root_remainder</tt> <tt class="py-op">-=</tt> <tt id="link-303" class="py-name"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-303', 'node', 'link-18');">node</a></tt><tt class="py-op">.</tt><tt id="link-304" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-304', 'branch_length', 'link-114');">branch_length</a></tt> </tt>
<a name="L928"></a><tt class="py-lineno"> 928</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">root_remainder</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L929"></a><tt class="py-lineno"> 929</tt>  <tt class="py-line">                <tt class="py-name">outgroup_node</tt> <tt class="py-op">=</tt> <tt id="link-305" class="py-name"><a title="Bio.Nexus.Trees.Tree.node" class="py-name" href="#" onclick="return doclink('link-305', 'node', 'link-18');">node</a></tt> </tt>
<a name="L930"></a><tt class="py-lineno"> 930</tt>  <tt class="py-line">                <tt class="py-name">outgroup_branch_length</tt> <tt class="py-op">=</tt> <tt class="py-op">-</tt><tt class="py-name">root_remainder</tt> </tt>
<a name="L931"></a><tt class="py-lineno"> 931</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L932"></a><tt class="py-lineno"> 932</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L933"></a><tt class="py-lineno"> 933</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Somehow, failed to find the midpoint!"</tt><tt class="py-op">)</tt> </tt>
<a name="L934"></a><tt class="py-lineno"> 934</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-306" class="py-name"><a title="Bio.Nexus.Trees.Tree.root_with_outgroup
Bio.Phylo.BaseTree.Tree.root_with_outgroup" class="py-name" href="#" onclick="return doclink('link-306', 'root_with_outgroup', 'link-294');">root_with_outgroup</a></tt><tt class="py-op">(</tt><tt class="py-name">outgroup_node</tt><tt class="py-op">,</tt> </tt>
<a name="L935"></a><tt class="py-lineno"> 935</tt>  <tt class="py-line">                                <tt class="py-name">outgroup_branch_length</tt><tt class="py-op">=</tt><tt class="py-name">outgroup_branch_length</tt><tt class="py-op">)</tt> </tt>
</div><a name="L936"></a><tt class="py-lineno"> 936</tt>  <tt class="py-line"> </tt>
<a name="L937"></a><tt class="py-lineno"> 937</tt>  <tt class="py-line">    <tt class="py-comment"># Method assumed by TreeMixin</tt> </tt>
<a name="L938"></a><tt class="py-lineno"> 938</tt>  <tt class="py-line"> </tt>
<a name="Tree.is_terminal"></a><div id="Tree.is_terminal-def"><a name="L939"></a><tt class="py-lineno"> 939</tt> <a class="py-toggle" href="#" id="Tree.is_terminal-toggle" onclick="return toggle('Tree.is_terminal');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.Tree-class.html#is_terminal">is_terminal</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.is_terminal-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Tree.is_terminal-expanded"><a name="L940"></a><tt class="py-lineno"> 940</tt>  <tt class="py-line">        <tt class="py-docstring">"""Check if the root of this tree is terminal."""</tt> </tt>
<a name="L941"></a><tt class="py-lineno"> 941</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-307" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.root
Bio.Phylo._cdao_owl.root" class="py-name" href="#" onclick="return doclink('link-307', 'root', 'link-7');">root</a></tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">)</tt> </tt>
</div><a name="L942"></a><tt class="py-lineno"> 942</tt>  <tt class="py-line"> </tt>
<a name="L943"></a><tt class="py-lineno"> 943</tt>  <tt class="py-line">    <tt class="py-comment"># Convention from SeqRecord and Alignment classes</tt> </tt>
<a name="L944"></a><tt class="py-lineno"> 944</tt>  <tt class="py-line"> </tt>
<a name="Tree.__format__"></a><div id="Tree.__format__-def"><a name="L945"></a><tt class="py-lineno"> 945</tt> <a class="py-toggle" href="#" id="Tree.__format__-toggle" onclick="return toggle('Tree.__format__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.Tree-class.html#__format__">__format__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">format_spec</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.__format__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Tree.__format__-expanded"><a name="L946"></a><tt class="py-lineno"> 946</tt>  <tt class="py-line">        <tt class="py-docstring">"""Serialize the tree as a string in the specified file format.</tt> </tt>
<a name="L947"></a><tt class="py-lineno"> 947</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L948"></a><tt class="py-lineno"> 948</tt>  <tt class="py-line"><tt class="py-docstring">        This method supports the ``format`` built-in function added in Python</tt> </tt>
<a name="L949"></a><tt class="py-lineno"> 949</tt>  <tt class="py-line"><tt class="py-docstring">        2.6/3.0.</tt> </tt>
<a name="L950"></a><tt class="py-lineno"> 950</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L951"></a><tt class="py-lineno"> 951</tt>  <tt class="py-line"><tt class="py-docstring">        :param format_spec: a lower-case string supported by `Bio.Phylo.write`</tt> </tt>
<a name="L952"></a><tt class="py-lineno"> 952</tt>  <tt class="py-line"><tt class="py-docstring">            as an output file format.</tt> </tt>
<a name="L953"></a><tt class="py-lineno"> 953</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L954"></a><tt class="py-lineno"> 954</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L955"></a><tt class="py-lineno"> 955</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">format_spec</tt><tt class="py-op">:</tt> </tt>
<a name="L956"></a><tt class="py-lineno"> 956</tt>  <tt class="py-line">            <tt class="py-keyword">from</tt> <tt id="link-308" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-308', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-309" class="py-name"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-309', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">StringIO</tt> </tt>
<a name="L957"></a><tt class="py-lineno"> 957</tt>  <tt class="py-line">            <tt class="py-keyword">from</tt> <tt id="link-310" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-310', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-311" class="py-name"><a title="Bio.Phylo" class="py-name" href="#" onclick="return doclink('link-311', 'Phylo', 'link-251');">Phylo</a></tt> <tt class="py-keyword">import</tt> <tt id="link-312" class="py-name" targets="Module Bio.Phylo._io=Bio.Phylo._io-module.html"><a title="Bio.Phylo._io" class="py-name" href="#" onclick="return doclink('link-312', '_io', 'link-312');">_io</a></tt> </tt>
<a name="L958"></a><tt class="py-lineno"> 958</tt>  <tt class="py-line">            <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">StringIO</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L959"></a><tt class="py-lineno"> 959</tt>  <tt class="py-line">            <tt id="link-313" class="py-name"><a title="Bio.Phylo._io" class="py-name" href="#" onclick="return doclink('link-313', '_io', 'link-312');">_io</a></tt><tt class="py-op">.</tt><tt id="link-314" class="py-name" targets="Function Bio.AlignIO.write()=Bio.AlignIO-module.html#write,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.write()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#write,Method Bio.Phylo.CDAOIO.Writer.write()=Bio.Phylo.CDAOIO.Writer-class.html#write,Function Bio.Phylo.CDAOIO.write()=Bio.Phylo.CDAOIO-module.html#write,Method Bio.Phylo.NeXMLIO.Writer.write()=Bio.Phylo.NeXMLIO.Writer-class.html#write,Function Bio.Phylo.NeXMLIO.write()=Bio.Phylo.NeXMLIO-module.html#write,Method Bio.Phylo.NewickIO.Writer.write()=Bio.Phylo.NewickIO.Writer-class.html#write,Function Bio.Phylo.NewickIO.write()=Bio.Phylo.NewickIO-module.html#write,Function Bio.Phylo.NexusIO.write()=Bio.Phylo.NexusIO-module.html#write,Method Bio.Phylo.PhyloXMLIO.Writer.write()=Bio.Phylo.PhyloXMLIO.Writer-class.html#write,Function Bio.Phylo.PhyloXMLIO.write()=Bio.Phylo.PhyloXMLIO-module.html#write,Function Bio.Phylo._io.write()=Bio.Phylo._io-module.html#write,Function Bio.SearchIO.write()=Bio.SearchIO-module.html#write,Function Bio.SeqIO.write()=Bio.SeqIO-module.html#write,Method Bio.bgzf.BgzfWriter.write()=Bio.bgzf.BgzfWriter-class.html#write,Function Bio.motifs.jaspar.write()=Bio.motifs.jaspar-module.html#write,Function Bio.motifs.transfac.write()=Bio.motifs.transfac-module.html#write,Function Bio.motifs.write()=Bio.motifs-module.html#write,Method Bio.phenotype.phen_micro.JsonWriter.write()=Bio.phenotype.phen_micro.JsonWriter-class.html#write,Function Bio.phenotype.write()=Bio.phenotype-module.html#write"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write
Bio.phenotype.phen_micro.JsonWriter.write
Bio.phenotype.write" class="py-name" href="#" onclick="return doclink('link-314', 'write', 'link-314');">write</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">handle</tt><tt class="py-op">,</tt> <tt class="py-name">format_spec</tt><tt class="py-op">)</tt> </tt>
<a name="L960"></a><tt class="py-lineno"> 960</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt class="py-name">getvalue</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L961"></a><tt class="py-lineno"> 961</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L962"></a><tt class="py-lineno"> 962</tt>  <tt class="py-line">            <tt class="py-comment"># Follow python convention and default to using __str__</tt> </tt>
<a name="L963"></a><tt class="py-lineno"> 963</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
</div><a name="L964"></a><tt class="py-lineno"> 964</tt>  <tt class="py-line"> </tt>
<a name="Tree.format"></a><div id="Tree.format-def"><a name="L965"></a><tt class="py-lineno"> 965</tt> <a class="py-toggle" href="#" id="Tree.format-toggle" onclick="return toggle('Tree.format');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.Tree-class.html#format">format</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">format</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.format-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Tree.format-expanded"><a name="L966"></a><tt class="py-lineno"> 966</tt>  <tt class="py-line">        <tt class="py-docstring">"""Serialize the tree as a string in the specified file format.</tt> </tt>
<a name="L967"></a><tt class="py-lineno"> 967</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L968"></a><tt class="py-lineno"> 968</tt>  <tt class="py-line"><tt class="py-docstring">        This duplicates the __format__ magic method for pre-2.6 Pythons.</tt> </tt>
<a name="L969"></a><tt class="py-lineno"> 969</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L970"></a><tt class="py-lineno"> 970</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-315" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.__format__()=Bio.Align.MultipleSeqAlignment-class.html#__format__,Method Bio.Align.PairwiseAlignment.__format__()=Bio.Align.PairwiseAlignment-class.html#__format__,Method Bio.Phylo.BaseTree.Tree.__format__()=Bio.Phylo.BaseTree.Tree-class.html#__format__,Method Bio.SeqRecord.SeqRecord.__format__()=Bio.SeqRecord.SeqRecord-class.html#__format__"><a title="Bio.Align.MultipleSeqAlignment.__format__
Bio.Align.PairwiseAlignment.__format__
Bio.Phylo.BaseTree.Tree.__format__
Bio.SeqRecord.SeqRecord.__format__" class="py-name" href="#" onclick="return doclink('link-315', '__format__', 'link-315');">__format__</a></tt><tt class="py-op">(</tt><tt id="link-316" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.format()=Bio.Align.MultipleSeqAlignment-class.html#format,Method Bio.Phylo.BaseTree.Tree.format()=Bio.Phylo.BaseTree.Tree-class.html#format,Method Bio.Restriction.Restriction.RestrictionBatch.format()=Bio.Restriction.Restriction.RestrictionBatch-class.html#format,Method Bio.SeqRecord.SeqRecord.format()=Bio.SeqRecord.SeqRecord-class.html#format,Method Bio.motifs.Motif.format()=Bio.motifs.Motif-class.html#format"><a title="Bio.Align.MultipleSeqAlignment.format
Bio.Phylo.BaseTree.Tree.format
Bio.Restriction.Restriction.RestrictionBatch.format
Bio.SeqRecord.SeqRecord.format
Bio.motifs.Motif.format" class="py-name" href="#" onclick="return doclink('link-316', 'format', 'link-316');">format</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L971"></a><tt class="py-lineno"> 971</tt>  <tt class="py-line"> </tt>
<a name="L972"></a><tt class="py-lineno"> 972</tt>  <tt class="py-line">    <tt class="py-comment"># Pretty-printer for the entire tree hierarchy</tt> </tt>
<a name="L973"></a><tt class="py-lineno"> 973</tt>  <tt class="py-line"> </tt>
<a name="Tree.__str__"></a><div id="Tree.__str__-def"><a name="L974"></a><tt class="py-lineno"> 974</tt> <a class="py-toggle" href="#" id="Tree.__str__-toggle" onclick="return toggle('Tree.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.Tree-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Tree.__str__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Tree.__str__-expanded"><a name="L975"></a><tt class="py-lineno"> 975</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a string representation of the entire tree.</tt> </tt>
<a name="L976"></a><tt class="py-lineno"> 976</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L977"></a><tt class="py-lineno"> 977</tt>  <tt class="py-line"><tt class="py-docstring">        Serialize each sub-clade recursively using ``repr`` to create a summary</tt> </tt>
<a name="L978"></a><tt class="py-lineno"> 978</tt>  <tt class="py-line"><tt class="py-docstring">        of the object structure.</tt> </tt>
<a name="L979"></a><tt class="py-lineno"> 979</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L980"></a><tt class="py-lineno"> 980</tt>  <tt class="py-line">        <tt class="py-name">TAB</tt> <tt class="py-op">=</tt> <tt class="py-string">'    '</tt> </tt>
<a name="L981"></a><tt class="py-lineno"> 981</tt>  <tt class="py-line">        <tt class="py-name">textlines</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L982"></a><tt class="py-lineno"> 982</tt>  <tt class="py-line"> </tt>
<a name="L983"></a><tt class="py-lineno"> 983</tt>  <tt class="py-line">        <tt class="py-keyword">def</tt> <tt class="py-def-name">print_tree</tt><tt class="py-op">(</tt><tt class="py-param">obj</tt><tt class="py-op">,</tt> <tt class="py-param">indent</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L984"></a><tt class="py-lineno"> 984</tt>  <tt class="py-line">            <tt class="py-docstring">"""Recursively serialize sub-elements.</tt> </tt>
<a name="L985"></a><tt class="py-lineno"> 985</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L986"></a><tt class="py-lineno"> 986</tt>  <tt class="py-line"><tt class="py-docstring">            This closes over textlines and modifies it in-place.</tt> </tt>
<a name="L987"></a><tt class="py-lineno"> 987</tt>  <tt class="py-line"><tt class="py-docstring">            """</tt> </tt>
<a name="L988"></a><tt class="py-lineno"> 988</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt id="link-317" class="py-name"><a title="Bio.Cluster.Tree
Bio.Nexus.Trees.Tree
Bio.Phylo.BaseTree.Tree
Bio.Phylo.CDAO.Tree
Bio.Phylo.NeXML.Tree
Bio.Phylo.Newick.Tree" class="py-name" href="#" onclick="return doclink('link-317', 'Tree', 'link-24');">Tree</a></tt><tt class="py-op">,</tt> <tt id="link-318" class="py-name"><a title="Bio.Phylo.BaseTree.Clade
Bio.Phylo.CDAO.Clade
Bio.Phylo.NeXML.Clade
Bio.Phylo.Newick.Clade
Bio.Phylo.PhyloXML.Clade" class="py-name" href="#" onclick="return doclink('link-318', 'Clade', 'link-23');">Clade</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L989"></a><tt class="py-lineno"> 989</tt>  <tt class="py-line">                <tt class="py-comment"># Avoid infinite recursion or special formatting from str()</tt> </tt>
<a name="L990"></a><tt class="py-lineno"> 990</tt>  <tt class="py-line">                <tt class="py-name">objstr</tt> <tt class="py-op">=</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">)</tt> </tt>
<a name="L991"></a><tt class="py-lineno"> 991</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L992"></a><tt class="py-lineno"> 992</tt>  <tt class="py-line">                <tt class="py-name">objstr</tt> <tt class="py-op">=</tt> <tt id="link-319" class="py-name"><a title="Bio.Phylo.BaseTree.as_string
Bio.Restriction.Restriction.RestrictionBatch.as_string" class="py-name" href="#" onclick="return doclink('link-319', 'as_string', 'link-6');">as_string</a></tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">)</tt> </tt>
<a name="L993"></a><tt class="py-lineno"> 993</tt>  <tt class="py-line">            <tt class="py-name">textlines</tt><tt class="py-op">.</tt><tt id="link-320" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-320', 'append', 'link-13');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">TAB</tt> <tt class="py-op">*</tt> <tt class="py-name">indent</tt> <tt class="py-op">+</tt> <tt class="py-name">objstr</tt><tt class="py-op">)</tt> </tt>
<a name="L994"></a><tt class="py-lineno"> 994</tt>  <tt class="py-line">            <tt class="py-name">indent</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L995"></a><tt class="py-lineno"> 995</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">attr</tt> <tt class="py-keyword">in</tt> <tt class="py-name">obj</tt><tt class="py-op">.</tt><tt class="py-name">__dict__</tt><tt class="py-op">:</tt> </tt>
<a name="L996"></a><tt class="py-lineno"> 996</tt>  <tt class="py-line">                <tt class="py-name">child</tt> <tt class="py-op">=</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">obj</tt><tt class="py-op">,</tt> <tt class="py-name">attr</tt><tt class="py-op">)</tt> </tt>
<a name="L997"></a><tt class="py-lineno"> 997</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">child</tt><tt class="py-op">,</tt> <tt id="link-321" class="py-name"><a title="Bio.Phylo.BaseTree.TreeElement" class="py-name" href="#" onclick="return doclink('link-321', 'TreeElement', 'link-17');">TreeElement</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L998"></a><tt class="py-lineno"> 998</tt>  <tt class="py-line">                    <tt class="py-name">print_tree</tt><tt class="py-op">(</tt><tt class="py-name">child</tt><tt class="py-op">,</tt> <tt class="py-name">indent</tt><tt class="py-op">)</tt> </tt>
<a name="L999"></a><tt class="py-lineno"> 999</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">child</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1000"></a><tt class="py-lineno">1000</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-name">elem</tt> <tt class="py-keyword">in</tt> <tt class="py-name">child</tt><tt class="py-op">:</tt> </tt>
<a name="L1001"></a><tt class="py-lineno">1001</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt id="link-322" class="py-name"><a title="Bio.Phylo.BaseTree.TreeElement" class="py-name" href="#" onclick="return doclink('link-322', 'TreeElement', 'link-17');">TreeElement</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1002"></a><tt class="py-lineno">1002</tt>  <tt class="py-line">                            <tt class="py-name">print_tree</tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">,</tt> <tt class="py-name">indent</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1003"></a><tt class="py-lineno">1003</tt>  <tt class="py-line"> </tt>
<a name="L1004"></a><tt class="py-lineno">1004</tt>  <tt class="py-line">        <tt class="py-name">print_tree</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L1005"></a><tt class="py-lineno">1005</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">'\n'</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">textlines</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1006"></a><tt class="py-lineno">1006</tt>  <tt class="py-line"> </tt>
<a name="Clade"></a><div id="Clade-def"><a name="L1007"></a><tt class="py-lineno">1007</tt>  <tt class="py-line"> </tt>
<a name="L1008"></a><tt class="py-lineno">1008</tt> <a class="py-toggle" href="#" id="Clade-toggle" onclick="return toggle('Clade');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.Clade-class.html">Clade</a><tt class="py-op">(</tt><tt class="py-base-class">TreeElement</tt><tt class="py-op">,</tt> <tt class="py-base-class">TreeMixin</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Clade-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="Clade-expanded"><a name="L1009"></a><tt class="py-lineno">1009</tt>  <tt class="py-line">    <tt class="py-docstring">"""A recursively defined sub-tree.</tt> </tt>
<a name="L1010"></a><tt class="py-lineno">1010</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1011"></a><tt class="py-lineno">1011</tt>  <tt class="py-line"><tt class="py-docstring">    :Parameters:</tt> </tt>
<a name="L1012"></a><tt class="py-lineno">1012</tt>  <tt class="py-line"><tt class="py-docstring">        branch_length : str</tt> </tt>
<a name="L1013"></a><tt class="py-lineno">1013</tt>  <tt class="py-line"><tt class="py-docstring">            The length of the branch leading to the root node of this clade.</tt> </tt>
<a name="L1014"></a><tt class="py-lineno">1014</tt>  <tt class="py-line"><tt class="py-docstring">        name : str</tt> </tt>
<a name="L1015"></a><tt class="py-lineno">1015</tt>  <tt class="py-line"><tt class="py-docstring">            The clade's name (a label).</tt> </tt>
<a name="L1016"></a><tt class="py-lineno">1016</tt>  <tt class="py-line"><tt class="py-docstring">        clades : list</tt> </tt>
<a name="L1017"></a><tt class="py-lineno">1017</tt>  <tt class="py-line"><tt class="py-docstring">            Sub-trees rooted directly under this tree's root.</tt> </tt>
<a name="L1018"></a><tt class="py-lineno">1018</tt>  <tt class="py-line"><tt class="py-docstring">        confidence : number</tt> </tt>
<a name="L1019"></a><tt class="py-lineno">1019</tt>  <tt class="py-line"><tt class="py-docstring">            Support.</tt> </tt>
<a name="L1020"></a><tt class="py-lineno">1020</tt>  <tt class="py-line"><tt class="py-docstring">        color : BranchColor</tt> </tt>
<a name="L1021"></a><tt class="py-lineno">1021</tt>  <tt class="py-line"><tt class="py-docstring">            The display color of the branch and descendents.</tt> </tt>
<a name="L1022"></a><tt class="py-lineno">1022</tt>  <tt class="py-line"><tt class="py-docstring">        width : number</tt> </tt>
<a name="L1023"></a><tt class="py-lineno">1023</tt>  <tt class="py-line"><tt class="py-docstring">            The display width of the branch and descendents.</tt> </tt>
<a name="L1024"></a><tt class="py-lineno">1024</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1025"></a><tt class="py-lineno">1025</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1026"></a><tt class="py-lineno">1026</tt>  <tt class="py-line"> </tt>
<a name="Clade.__init__"></a><div id="Clade.__init__-def"><a name="L1027"></a><tt class="py-lineno">1027</tt> <a class="py-toggle" href="#" id="Clade.__init__-toggle" onclick="return toggle('Clade.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.Clade-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">branch_length</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">clades</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L1028"></a><tt class="py-lineno">1028</tt>  <tt class="py-line">                 <tt class="py-param">confidence</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">color</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">width</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Clade.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Clade.__init__-expanded"><a name="L1029"></a><tt class="py-lineno">1029</tt>  <tt class="py-line">        <tt class="py-docstring">"""Define parameters for the Clade tree."""</tt> </tt>
<a name="L1030"></a><tt class="py-lineno">1030</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-323" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-323', 'branch_length', 'link-114');">branch_length</a></tt> <tt class="py-op">=</tt> <tt id="link-324" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.branch_length" class="py-name" href="#" onclick="return doclink('link-324', 'branch_length', 'link-114');">branch_length</a></tt> </tt>
<a name="L1031"></a><tt class="py-lineno">1031</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-325" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-325', 'name', 'link-26');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-326" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-326', 'name', 'link-26');">name</a></tt> </tt>
<a name="L1032"></a><tt class="py-lineno">1032</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt> <tt class="py-op">=</tt> <tt class="py-name">clades</tt> <tt class="py-keyword">or</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L1033"></a><tt class="py-lineno">1033</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-327" class="py-name" targets="Variable Bio.Phylo.PhyloXML.Clade.confidence=Bio.Phylo.PhyloXML.Clade-class.html#confidence,Variable Bio.Phylo.PhyloXML.Phylogeny.confidence=Bio.Phylo.PhyloXML.Phylogeny-class.html#confidence,Method Bio.Phylo.PhyloXMLIO.Parser.confidence()=Bio.Phylo.PhyloXMLIO.Parser-class.html#confidence,Method Bio.Phylo.PhyloXMLIO.Writer.confidence()=Bio.Phylo.PhyloXMLIO.Writer-class.html#confidence"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-327', 'confidence', 'link-327');">confidence</a></tt> <tt class="py-op">=</tt> <tt id="link-328" class="py-name"><a title="Bio.Phylo.PhyloXML.Clade.confidence
Bio.Phylo.PhyloXML.Phylogeny.confidence
Bio.Phylo.PhyloXMLIO.Parser.confidence
Bio.Phylo.PhyloXMLIO.Writer.confidence" class="py-name" href="#" onclick="return doclink('link-328', 'confidence', 'link-327');">confidence</a></tt> </tt>
<a name="L1034"></a><tt class="py-lineno">1034</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-329" class="py-name" targets="Variable Bio.Phylo.BaseTree.Clade.color=Bio.Phylo.BaseTree.Clade-class.html#color,Method Bio.Phylo.PhyloXMLIO.Parser.color()=Bio.Phylo.PhyloXMLIO.Parser-class.html#color,Method Bio.Phylo.PhyloXMLIO.Writer.color()=Bio.Phylo.PhyloXMLIO.Writer-class.html#color"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-329', 'color', 'link-329');">color</a></tt> <tt class="py-op">=</tt> <tt id="link-330" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-330', 'color', 'link-329');">color</a></tt> </tt>
<a name="L1035"></a><tt class="py-lineno">1035</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-331" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.width=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#width,Method Bio.Phylo.PhyloXMLIO.Writer.width()=Bio.Phylo.PhyloXMLIO.Writer-class.html#width"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.width
Bio.Phylo.PhyloXMLIO.Writer.width" class="py-name" href="#" onclick="return doclink('link-331', 'width', 'link-331');">width</a></tt> <tt class="py-op">=</tt> <tt id="link-332" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.width
Bio.Phylo.PhyloXMLIO.Writer.width" class="py-name" href="#" onclick="return doclink('link-332', 'width', 'link-331');">width</a></tt> </tt>
</div><a name="L1036"></a><tt class="py-lineno">1036</tt>  <tt class="py-line"> </tt>
<a name="L1037"></a><tt class="py-lineno">1037</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">property</tt> </tt>
<a name="Clade.root"></a><div id="Clade.root-def"><a name="L1038"></a><tt class="py-lineno">1038</tt> <a class="py-toggle" href="#" id="Clade.root-toggle" onclick="return toggle('Clade.root');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.Clade-class.html#root">root</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Clade.root-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Clade.root-expanded"><a name="L1039"></a><tt class="py-lineno">1039</tt>  <tt class="py-line">        <tt class="py-docstring">"""Allow TreeMixin methods to traverse clades properly."""</tt> </tt>
<a name="L1040"></a><tt class="py-lineno">1040</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt> </tt>
</div><a name="L1041"></a><tt class="py-lineno">1041</tt>  <tt class="py-line"> </tt>
<a name="Clade.is_terminal"></a><div id="Clade.is_terminal-def"><a name="L1042"></a><tt class="py-lineno">1042</tt> <a class="py-toggle" href="#" id="Clade.is_terminal-toggle" onclick="return toggle('Clade.is_terminal');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.Clade-class.html#is_terminal">is_terminal</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Clade.is_terminal-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Clade.is_terminal-expanded"><a name="L1043"></a><tt class="py-lineno">1043</tt>  <tt class="py-line">        <tt class="py-docstring">"""Check if this is a terminal (leaf) node."""</tt> </tt>
<a name="L1044"></a><tt class="py-lineno">1044</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1045"></a><tt class="py-lineno">1045</tt>  <tt class="py-line"> </tt>
<a name="L1046"></a><tt class="py-lineno">1046</tt>  <tt class="py-line">    <tt class="py-comment"># Sequence-type behavior methods</tt> </tt>
<a name="L1047"></a><tt class="py-lineno">1047</tt>  <tt class="py-line"> </tt>
<a name="Clade.__getitem__"></a><div id="Clade.__getitem__-def"><a name="L1048"></a><tt class="py-lineno">1048</tt> <a class="py-toggle" href="#" id="Clade.__getitem__-toggle" onclick="return toggle('Clade.__getitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.Clade-class.html#__getitem__">__getitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">index</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Clade.__getitem__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Clade.__getitem__-expanded"><a name="L1049"></a><tt class="py-lineno">1049</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get clades by index (integer or slice)."""</tt> </tt>
<a name="L1050"></a><tt class="py-lineno">1050</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-333" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-333', 'index', 'link-167');">index</a></tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">int</tt><tt class="py-op">,</tt> <tt class="py-name">slice</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1051"></a><tt class="py-lineno">1051</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">[</tt><tt id="link-334" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-334', 'index', 'link-167');">index</a></tt><tt class="py-op">]</tt> </tt>
<a name="L1052"></a><tt class="py-lineno">1052</tt>  <tt class="py-line">        <tt id="link-335" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-335', 'ref', 'link-105');">ref</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt> </tt>
<a name="L1053"></a><tt class="py-lineno">1053</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">idx</tt> <tt class="py-keyword">in</tt> <tt id="link-336" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-336', 'index', 'link-167');">index</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1054"></a><tt class="py-lineno">1054</tt>  <tt class="py-line">            <tt id="link-337" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-337', 'ref', 'link-105');">ref</a></tt> <tt class="py-op">=</tt> <tt id="link-338" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-338', 'ref', 'link-105');">ref</a></tt><tt class="py-op">[</tt><tt class="py-name">idx</tt><tt class="py-op">]</tt> </tt>
<a name="L1055"></a><tt class="py-lineno">1055</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-339" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-339', 'ref', 'link-105');">ref</a></tt> </tt>
</div><a name="L1056"></a><tt class="py-lineno">1056</tt>  <tt class="py-line"> </tt>
<a name="Clade.__iter__"></a><div id="Clade.__iter__-def"><a name="L1057"></a><tt class="py-lineno">1057</tt> <a class="py-toggle" href="#" id="Clade.__iter__-toggle" onclick="return toggle('Clade.__iter__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.Clade-class.html#__iter__">__iter__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Clade.__iter__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Clade.__iter__-expanded"><a name="L1058"></a><tt class="py-lineno">1058</tt>  <tt class="py-line">        <tt class="py-docstring">"""Iterate through this tree's direct descendent clades (sub-trees)."""</tt> </tt>
<a name="L1059"></a><tt class="py-lineno">1059</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">iter</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1060"></a><tt class="py-lineno">1060</tt>  <tt class="py-line"> </tt>
<a name="Clade.__len__"></a><div id="Clade.__len__-def"><a name="L1061"></a><tt class="py-lineno">1061</tt> <a class="py-toggle" href="#" id="Clade.__len__-toggle" onclick="return toggle('Clade.__len__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.Clade-class.html#__len__">__len__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Clade.__len__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Clade.__len__-expanded"><a name="L1062"></a><tt class="py-lineno">1062</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the number of clades directy under the root."""</tt> </tt>
<a name="L1063"></a><tt class="py-lineno">1063</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">clades</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1064"></a><tt class="py-lineno">1064</tt>  <tt class="py-line"> </tt>
<a name="L1065"></a><tt class="py-lineno">1065</tt>  <tt class="py-line">    <tt class="py-comment"># Python 3:</tt> </tt>
<a name="Clade.__bool__"></a><div id="Clade.__bool__-def"><a name="L1066"></a><tt class="py-lineno">1066</tt> <a class="py-toggle" href="#" id="Clade.__bool__-toggle" onclick="return toggle('Clade.__bool__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.Clade-class.html#__bool__">__bool__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Clade.__bool__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Clade.__bool__-expanded"><a name="L1067"></a><tt class="py-lineno">1067</tt>  <tt class="py-line">        <tt class="py-docstring">"""Boolean value of an instance of this class (True).</tt> </tt>
<a name="L1068"></a><tt class="py-lineno">1068</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1069"></a><tt class="py-lineno">1069</tt>  <tt class="py-line"><tt class="py-docstring">        NB: If this method is not defined, but ``__len__``  is, then the object</tt> </tt>
<a name="L1070"></a><tt class="py-lineno">1070</tt>  <tt class="py-line"><tt class="py-docstring">        is considered true if the result of ``__len__()`` is nonzero. We want</tt> </tt>
<a name="L1071"></a><tt class="py-lineno">1071</tt>  <tt class="py-line"><tt class="py-docstring">        Clade instances to always be considered True.</tt> </tt>
<a name="L1072"></a><tt class="py-lineno">1072</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1073"></a><tt class="py-lineno">1073</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">True</tt> </tt>
</div><a name="L1074"></a><tt class="py-lineno">1074</tt>  <tt class="py-line">    <tt class="py-comment"># Python 2:</tt> </tt>
<a name="L1075"></a><tt class="py-lineno">1075</tt>  <tt class="py-line">    <tt id="link-340" class="py-name" targets="Method Bio.SeqFeature.CompoundLocation.__nonzero__()=Bio.SeqFeature.CompoundLocation-class.html#__nonzero__,Method Bio.SeqFeature.FeatureLocation.__nonzero__()=Bio.SeqFeature.FeatureLocation-class.html#__nonzero__"><a title="Bio.SeqFeature.CompoundLocation.__nonzero__
Bio.SeqFeature.FeatureLocation.__nonzero__" class="py-name" href="#" onclick="return doclink('link-340', '__nonzero__', 'link-340');">__nonzero__</a></tt> <tt class="py-op">=</tt> <tt id="link-341" class="py-name" targets="Method Bio.Phylo.BaseTree.Clade.__bool__()=Bio.Phylo.BaseTree.Clade-class.html#__bool__,Method Bio.SearchIO._model.hit.Hit.__bool__()=Bio.SearchIO._model.hit.Hit-class.html#__bool__,Method Bio.SearchIO._model.hsp.HSP.__bool__()=Bio.SearchIO._model.hsp.HSP-class.html#__bool__,Method Bio.SearchIO._model.query.QueryResult.__bool__()=Bio.SearchIO._model.query.QueryResult-class.html#__bool__,Method Bio.SeqFeature.SeqFeature.__bool__()=Bio.SeqFeature.SeqFeature-class.html#__bool__,Method Bio.SeqRecord.SeqRecord.__bool__()=Bio.SeqRecord.SeqRecord-class.html#__bool__"><a title="Bio.Phylo.BaseTree.Clade.__bool__
Bio.SearchIO._model.hit.Hit.__bool__
Bio.SearchIO._model.hsp.HSP.__bool__
Bio.SearchIO._model.query.QueryResult.__bool__
Bio.SeqFeature.SeqFeature.__bool__
Bio.SeqRecord.SeqRecord.__bool__" class="py-name" href="#" onclick="return doclink('link-341', '__bool__', 'link-341');">__bool__</a></tt> </tt>
<a name="L1076"></a><tt class="py-lineno">1076</tt>  <tt class="py-line"> </tt>
<a name="Clade.__str__"></a><div id="Clade.__str__-def"><a name="L1077"></a><tt class="py-lineno">1077</tt> <a class="py-toggle" href="#" id="Clade.__str__-toggle" onclick="return toggle('Clade.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.Clade-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Clade.__str__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Clade.__str__-expanded"><a name="L1078"></a><tt class="py-lineno">1078</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return name of the class instance."""</tt> </tt>
<a name="L1079"></a><tt class="py-lineno">1079</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-342" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-342', 'name', 'link-26');">name</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1080"></a><tt class="py-lineno">1080</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-343" class="py-name"><a title="Bio.Phylo._utils
Bio.SearchIO._utils
Bio._utils" class="py-name" href="#" onclick="return doclink('link-343', '_utils', 'link-5');">_utils</a></tt><tt class="py-op">.</tt><tt id="link-344" class="py-name"><a title="Bio._utils.trim_str" class="py-name" href="#" onclick="return doclink('link-344', 'trim_str', 'link-63');">trim_str</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-345" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-345', 'name', 'link-26');">name</a></tt><tt class="py-op">,</tt> <tt class="py-number">40</tt><tt class="py-op">,</tt> <tt class="py-string">'...'</tt><tt class="py-op">)</tt> </tt>
<a name="L1081"></a><tt class="py-lineno">1081</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt> </tt>
</div><a name="L1082"></a><tt class="py-lineno">1082</tt>  <tt class="py-line"> </tt>
<a name="L1083"></a><tt class="py-lineno">1083</tt>  <tt class="py-line">    <tt class="py-comment"># Syntax sugar for setting the branch color</tt> </tt>
<a name="Clade._get_color"></a><div id="Clade._get_color-def"><a name="L1084"></a><tt class="py-lineno">1084</tt> <a class="py-toggle" href="#" id="Clade._get_color-toggle" onclick="return toggle('Clade._get_color');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.Clade-class.html#_get_color">_get_color</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Clade._get_color-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Clade._get_color-expanded"><a name="L1085"></a><tt class="py-lineno">1085</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_color</tt> </tt>
</div><a name="L1086"></a><tt class="py-lineno">1086</tt>  <tt class="py-line"> </tt>
<a name="Clade._set_color"></a><div id="Clade._set_color-def"><a name="L1087"></a><tt class="py-lineno">1087</tt> <a class="py-toggle" href="#" id="Clade._set_color-toggle" onclick="return toggle('Clade._set_color');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.Clade-class.html#_set_color">_set_color</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">arg</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Clade._set_color-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="Clade._set_color-expanded"><a name="L1088"></a><tt class="py-lineno">1088</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">arg</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt> <tt class="py-keyword">or</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">arg</tt><tt class="py-op">,</tt> <tt id="link-346" class="py-name" targets="Class Bio.Phylo.BaseTree.BranchColor=Bio.Phylo.BaseTree.BranchColor-class.html,Class Bio.Phylo.PhyloXML.BranchColor=Bio.Phylo.PhyloXML.BranchColor-class.html"><a title="Bio.Phylo.BaseTree.BranchColor
Bio.Phylo.PhyloXML.BranchColor" class="py-name" href="#" onclick="return doclink('link-346', 'BranchColor', 'link-346');">BranchColor</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1089"></a><tt class="py-lineno">1089</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_color</tt> <tt class="py-op">=</tt> <tt class="py-name">arg</tt> </tt>
<a name="L1090"></a><tt class="py-lineno">1090</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">arg</tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1091"></a><tt class="py-lineno">1091</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">arg</tt> <tt class="py-keyword">in</tt> <tt id="link-347" class="py-name"><a title="Bio.Phylo.BaseTree.BranchColor
Bio.Phylo.PhyloXML.BranchColor" class="py-name" href="#" onclick="return doclink('link-347', 'BranchColor', 'link-346');">BranchColor</a></tt><tt class="py-op">.</tt><tt id="link-348" class="py-name" targets="Variable Bio.Phylo.BaseTree.BranchColor.color_names=Bio.Phylo.BaseTree.BranchColor-class.html#color_names"><a title="Bio.Phylo.BaseTree.BranchColor.color_names" class="py-name" href="#" onclick="return doclink('link-348', 'color_names', 'link-348');">color_names</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1092"></a><tt class="py-lineno">1092</tt>  <tt class="py-line">                <tt class="py-comment"># Known color name</tt> </tt>
<a name="L1093"></a><tt class="py-lineno">1093</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_color</tt> <tt class="py-op">=</tt> <tt id="link-349" class="py-name"><a title="Bio.Phylo.BaseTree.BranchColor
Bio.Phylo.PhyloXML.BranchColor" class="py-name" href="#" onclick="return doclink('link-349', 'BranchColor', 'link-346');">BranchColor</a></tt><tt class="py-op">.</tt><tt id="link-350" class="py-name" targets="Class Method Bio.Phylo.BaseTree.BranchColor.from_name()=Bio.Phylo.BaseTree.BranchColor-class.html#from_name"><a title="Bio.Phylo.BaseTree.BranchColor.from_name" class="py-name" href="#" onclick="return doclink('link-350', 'from_name', 'link-350');">from_name</a></tt><tt class="py-op">(</tt><tt class="py-name">arg</tt><tt class="py-op">)</tt> </tt>
<a name="L1094"></a><tt class="py-lineno">1094</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">arg</tt><tt class="py-op">.</tt><tt id="link-351" class="py-name" targets="Method Bio.Seq.Seq.startswith()=Bio.Seq.Seq-class.html#startswith"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-351', 'startswith', 'link-351');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'#'</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">arg</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">7</tt><tt class="py-op">:</tt> </tt>
<a name="L1095"></a><tt class="py-lineno">1095</tt>  <tt class="py-line">                <tt class="py-comment"># HTML-style hex string</tt> </tt>
<a name="L1096"></a><tt class="py-lineno">1096</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_color</tt> <tt class="py-op">=</tt> <tt id="link-352" class="py-name"><a title="Bio.Phylo.BaseTree.BranchColor
Bio.Phylo.PhyloXML.BranchColor" class="py-name" href="#" onclick="return doclink('link-352', 'BranchColor', 'link-346');">BranchColor</a></tt><tt class="py-op">.</tt><tt id="link-353" class="py-name" targets="Class Method Bio.Phylo.BaseTree.BranchColor.from_hex()=Bio.Phylo.BaseTree.BranchColor-class.html#from_hex"><a title="Bio.Phylo.BaseTree.BranchColor.from_hex" class="py-name" href="#" onclick="return doclink('link-353', 'from_hex', 'link-353');">from_hex</a></tt><tt class="py-op">(</tt><tt class="py-name">arg</tt><tt class="py-op">)</tt> </tt>
<a name="L1097"></a><tt class="py-lineno">1097</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1098"></a><tt class="py-lineno">1098</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"invalid color string %s"</tt> <tt class="py-op">%</tt> <tt class="py-name">arg</tt><tt class="py-op">)</tt> </tt>
<a name="L1099"></a><tt class="py-lineno">1099</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">arg</tt><tt class="py-op">,</tt> <tt class="py-string">'__iter__'</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">arg</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">3</tt><tt class="py-op">:</tt> </tt>
<a name="L1100"></a><tt class="py-lineno">1100</tt>  <tt class="py-line">            <tt class="py-comment"># RGB triplet</tt> </tt>
<a name="L1101"></a><tt class="py-lineno">1101</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_color</tt> <tt class="py-op">=</tt> <tt id="link-354" class="py-name"><a title="Bio.Phylo.BaseTree.BranchColor
Bio.Phylo.PhyloXML.BranchColor" class="py-name" href="#" onclick="return doclink('link-354', 'BranchColor', 'link-346');">BranchColor</a></tt><tt class="py-op">(</tt><tt class="py-op">*</tt><tt class="py-name">arg</tt><tt class="py-op">)</tt> </tt>
<a name="L1102"></a><tt class="py-lineno">1102</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1103"></a><tt class="py-lineno">1103</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"invalid color value %s"</tt> <tt class="py-op">%</tt> <tt class="py-name">arg</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1104"></a><tt class="py-lineno">1104</tt>  <tt class="py-line"> </tt>
<a name="L1105"></a><tt class="py-lineno">1105</tt>  <tt class="py-line">    <tt id="link-355" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-355', 'color', 'link-329');">color</a></tt> <tt class="py-op">=</tt> <tt id="link-356" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.property()=Bio.Phylo.PhyloXMLIO.Parser-class.html#property,Method Bio.Phylo.PhyloXMLIO.Writer.property()=Bio.Phylo.PhyloXMLIO.Writer-class.html#property"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-356', 'property', 'link-356');">property</a></tt><tt class="py-op">(</tt><tt id="link-357" class="py-name" targets="Method Bio.Phylo.BaseTree.Clade._get_color()=Bio.Phylo.BaseTree.Clade-class.html#_get_color"><a title="Bio.Phylo.BaseTree.Clade._get_color" class="py-name" href="#" onclick="return doclink('link-357', '_get_color', 'link-357');">_get_color</a></tt><tt class="py-op">,</tt> <tt id="link-358" class="py-name" targets="Method Bio.Phylo.BaseTree.Clade._set_color()=Bio.Phylo.BaseTree.Clade-class.html#_set_color"><a title="Bio.Phylo.BaseTree.Clade._set_color" class="py-name" href="#" onclick="return doclink('link-358', '_set_color', 'link-358');">_set_color</a></tt><tt class="py-op">,</tt> <tt class="py-name">doc</tt><tt class="py-op">=</tt><tt class="py-string">"Branch color."</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1106"></a><tt class="py-lineno">1106</tt>  <tt class="py-line"> </tt>
<a name="BranchColor"></a><div id="BranchColor-def"><a name="L1107"></a><tt class="py-lineno">1107</tt>  <tt class="py-line"> </tt>
<a name="L1108"></a><tt class="py-lineno">1108</tt> <a class="py-toggle" href="#" id="BranchColor-toggle" onclick="return toggle('BranchColor');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.BranchColor-class.html">BranchColor</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BranchColor-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="BranchColor-expanded"><a name="L1109"></a><tt class="py-lineno">1109</tt>  <tt class="py-line">    <tt class="py-docstring">"""Indicates the color of a clade when rendered graphically.</tt> </tt>
<a name="L1110"></a><tt class="py-lineno">1110</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1111"></a><tt class="py-lineno">1111</tt>  <tt class="py-line"><tt class="py-docstring">    The color should be interpreted by client code (e.g. visualization</tt> </tt>
<a name="L1112"></a><tt class="py-lineno">1112</tt>  <tt class="py-line"><tt class="py-docstring">    programs) as applying to the whole clade, unless overwritten by the</tt> </tt>
<a name="L1113"></a><tt class="py-lineno">1113</tt>  <tt class="py-line"><tt class="py-docstring">    color(s) of sub-clades.</tt> </tt>
<a name="L1114"></a><tt class="py-lineno">1114</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1115"></a><tt class="py-lineno">1115</tt>  <tt class="py-line"><tt class="py-docstring">    Color values must be integers from 0 to 255.</tt> </tt>
<a name="L1116"></a><tt class="py-lineno">1116</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1117"></a><tt class="py-lineno">1117</tt>  <tt class="py-line"> </tt>
<a name="L1118"></a><tt class="py-lineno">1118</tt>  <tt class="py-line">    <tt id="link-359" class="py-name"><a title="Bio.Phylo.BaseTree.BranchColor.color_names" class="py-name" href="#" onclick="return doclink('link-359', 'color_names', 'link-348');">color_names</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L1119"></a><tt class="py-lineno">1119</tt>  <tt class="py-line">        <tt class="py-string">'red'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">255</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1120"></a><tt class="py-lineno">1120</tt>  <tt class="py-line">        <tt class="py-string">'r'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">255</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1121"></a><tt class="py-lineno">1121</tt>  <tt class="py-line">        <tt class="py-string">'yellow'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">255</tt><tt class="py-op">,</tt> <tt class="py-number">255</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1122"></a><tt class="py-lineno">1122</tt>  <tt class="py-line">        <tt class="py-string">'y'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">255</tt><tt class="py-op">,</tt> <tt class="py-number">255</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1123"></a><tt class="py-lineno">1123</tt>  <tt class="py-line">        <tt class="py-string">'green'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">128</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1124"></a><tt class="py-lineno">1124</tt>  <tt class="py-line">        <tt class="py-string">'g'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">128</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1125"></a><tt class="py-lineno">1125</tt>  <tt class="py-line">        <tt class="py-string">'cyan'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">255</tt><tt class="py-op">,</tt> <tt class="py-number">255</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1126"></a><tt class="py-lineno">1126</tt>  <tt class="py-line">        <tt class="py-string">'c'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">255</tt><tt class="py-op">,</tt> <tt class="py-number">255</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1127"></a><tt class="py-lineno">1127</tt>  <tt class="py-line">        <tt class="py-string">'blue'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">255</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1128"></a><tt class="py-lineno">1128</tt>  <tt class="py-line">        <tt class="py-string">'b'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">255</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1129"></a><tt class="py-lineno">1129</tt>  <tt class="py-line">        <tt class="py-string">'magenta'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">255</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">255</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1130"></a><tt class="py-lineno">1130</tt>  <tt class="py-line">        <tt class="py-string">'m'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">255</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">255</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1131"></a><tt class="py-lineno">1131</tt>  <tt class="py-line">        <tt class="py-string">'black'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1132"></a><tt class="py-lineno">1132</tt>  <tt class="py-line">        <tt class="py-string">'k'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1133"></a><tt class="py-lineno">1133</tt>  <tt class="py-line">        <tt class="py-string">'white'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">255</tt><tt class="py-op">,</tt> <tt class="py-number">255</tt><tt class="py-op">,</tt> <tt class="py-number">255</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1134"></a><tt class="py-lineno">1134</tt>  <tt class="py-line">        <tt class="py-string">'w'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">255</tt><tt class="py-op">,</tt> <tt class="py-number">255</tt><tt class="py-op">,</tt> <tt class="py-number">255</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1135"></a><tt class="py-lineno">1135</tt>  <tt class="py-line">        <tt class="py-comment"># Names standardized in HTML/CSS spec</tt> </tt>
<a name="L1136"></a><tt class="py-lineno">1136</tt>  <tt class="py-line">        <tt class="py-comment"># http://w3schools.com/html/html_colornames.asp</tt> </tt>
<a name="L1137"></a><tt class="py-lineno">1137</tt>  <tt class="py-line">        <tt class="py-string">'maroon'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">128</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1138"></a><tt class="py-lineno">1138</tt>  <tt class="py-line">        <tt class="py-string">'olive'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">128</tt><tt class="py-op">,</tt> <tt class="py-number">128</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1139"></a><tt class="py-lineno">1139</tt>  <tt class="py-line">        <tt class="py-string">'lime'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">255</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1140"></a><tt class="py-lineno">1140</tt>  <tt class="py-line">        <tt class="py-string">'aqua'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">255</tt><tt class="py-op">,</tt> <tt class="py-number">255</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1141"></a><tt class="py-lineno">1141</tt>  <tt class="py-line">        <tt class="py-string">'teal'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">128</tt><tt class="py-op">,</tt> <tt class="py-number">128</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1142"></a><tt class="py-lineno">1142</tt>  <tt class="py-line">        <tt class="py-string">'navy'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">128</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1143"></a><tt class="py-lineno">1143</tt>  <tt class="py-line">        <tt class="py-string">'fuchsia'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">255</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">255</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1144"></a><tt class="py-lineno">1144</tt>  <tt class="py-line">        <tt class="py-string">'purple'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">128</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">128</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1145"></a><tt class="py-lineno">1145</tt>  <tt class="py-line">        <tt class="py-string">'silver'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">192</tt><tt class="py-op">,</tt> <tt class="py-number">192</tt><tt class="py-op">,</tt> <tt class="py-number">192</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1146"></a><tt class="py-lineno">1146</tt>  <tt class="py-line">        <tt class="py-string">'gray'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">128</tt><tt class="py-op">,</tt> <tt class="py-number">128</tt><tt class="py-op">,</tt> <tt class="py-number">128</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1147"></a><tt class="py-lineno">1147</tt>  <tt class="py-line">        <tt class="py-comment"># More definitions from matplotlib/gcolor2</tt> </tt>
<a name="L1148"></a><tt class="py-lineno">1148</tt>  <tt class="py-line">        <tt class="py-string">'grey'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">128</tt><tt class="py-op">,</tt> <tt class="py-number">128</tt><tt class="py-op">,</tt> <tt class="py-number">128</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1149"></a><tt class="py-lineno">1149</tt>  <tt class="py-line">        <tt class="py-string">'pink'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">255</tt><tt class="py-op">,</tt> <tt class="py-number">192</tt><tt class="py-op">,</tt> <tt class="py-number">203</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1150"></a><tt class="py-lineno">1150</tt>  <tt class="py-line">        <tt class="py-string">'salmon'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">250</tt><tt class="py-op">,</tt> <tt class="py-number">128</tt><tt class="py-op">,</tt> <tt class="py-number">114</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1151"></a><tt class="py-lineno">1151</tt>  <tt class="py-line">        <tt class="py-string">'orange'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">255</tt><tt class="py-op">,</tt> <tt class="py-number">165</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1152"></a><tt class="py-lineno">1152</tt>  <tt class="py-line">        <tt class="py-string">'gold'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">255</tt><tt class="py-op">,</tt> <tt class="py-number">215</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1153"></a><tt class="py-lineno">1153</tt>  <tt class="py-line">        <tt class="py-string">'tan'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">210</tt><tt class="py-op">,</tt> <tt class="py-number">180</tt><tt class="py-op">,</tt> <tt class="py-number">140</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1154"></a><tt class="py-lineno">1154</tt>  <tt class="py-line">        <tt class="py-string">'brown'</tt><tt class="py-op">:</tt> <tt class="py-op">(</tt><tt class="py-number">165</tt><tt class="py-op">,</tt> <tt class="py-number">42</tt><tt class="py-op">,</tt> <tt class="py-number">42</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1155"></a><tt class="py-lineno">1155</tt>  <tt class="py-line">        <tt class="py-op">}</tt> </tt>
<a name="L1156"></a><tt class="py-lineno">1156</tt>  <tt class="py-line"> </tt>
<a name="BranchColor.__init__"></a><div id="BranchColor.__init__-def"><a name="L1157"></a><tt class="py-lineno">1157</tt> <a class="py-toggle" href="#" id="BranchColor.__init__-toggle" onclick="return toggle('BranchColor.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.BranchColor-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">red</tt><tt class="py-op">,</tt> <tt class="py-param">green</tt><tt class="py-op">,</tt> <tt class="py-param">blue</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BranchColor.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="BranchColor.__init__-expanded"><a name="L1158"></a><tt class="py-lineno">1158</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize BranchColor for a tree."""</tt> </tt>
<a name="L1159"></a><tt class="py-lineno">1159</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-360" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-360', 'color', 'link-329');">color</a></tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt id="link-361" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.red()=Bio.Phylo.PhyloXMLIO.Writer-class.html#red"><a title="Bio.Phylo.PhyloXMLIO.Writer.red" class="py-name" href="#" onclick="return doclink('link-361', 'red', 'link-361');">red</a></tt><tt class="py-op">,</tt> <tt id="link-362" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.green()=Bio.Phylo.PhyloXMLIO.Writer-class.html#green"><a title="Bio.Phylo.PhyloXMLIO.Writer.green" class="py-name" href="#" onclick="return doclink('link-362', 'green', 'link-362');">green</a></tt><tt class="py-op">,</tt> <tt id="link-363" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.blue()=Bio.Phylo.PhyloXMLIO.Writer-class.html#blue"><a title="Bio.Phylo.PhyloXMLIO.Writer.blue" class="py-name" href="#" onclick="return doclink('link-363', 'blue', 'link-363');">blue</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1160"></a><tt class="py-lineno">1160</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt class="py-op">(</tt><tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-364" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-364', 'color', 'link-329');">color</a></tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> </tt>
<a name="L1161"></a><tt class="py-lineno">1161</tt>  <tt class="py-line">                    <tt class="py-number">0</tt> <tt class="py-op">&lt;=</tt> <tt id="link-365" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-365', 'color', 'link-329');">color</a></tt> <tt class="py-op">&lt;=</tt> <tt class="py-number">255</tt> </tt>
<a name="L1162"></a><tt class="py-lineno">1162</tt>  <tt class="py-line">                    <tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">"Color values must be integers between 0 and 255."</tt> </tt>
<a name="L1163"></a><tt class="py-lineno">1163</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-366" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.red" class="py-name" href="#" onclick="return doclink('link-366', 'red', 'link-361');">red</a></tt> <tt class="py-op">=</tt> <tt id="link-367" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.red" class="py-name" href="#" onclick="return doclink('link-367', 'red', 'link-361');">red</a></tt> </tt>
<a name="L1164"></a><tt class="py-lineno">1164</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-368" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.green" class="py-name" href="#" onclick="return doclink('link-368', 'green', 'link-362');">green</a></tt> <tt class="py-op">=</tt> <tt id="link-369" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.green" class="py-name" href="#" onclick="return doclink('link-369', 'green', 'link-362');">green</a></tt> </tt>
<a name="L1165"></a><tt class="py-lineno">1165</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-370" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.blue" class="py-name" href="#" onclick="return doclink('link-370', 'blue', 'link-363');">blue</a></tt> <tt class="py-op">=</tt> <tt id="link-371" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.blue" class="py-name" href="#" onclick="return doclink('link-371', 'blue', 'link-363');">blue</a></tt> </tt>
</div><a name="L1166"></a><tt class="py-lineno">1166</tt>  <tt class="py-line"> </tt>
<a name="L1167"></a><tt class="py-lineno">1167</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">classmethod</tt> </tt>
<a name="BranchColor.from_hex"></a><div id="BranchColor.from_hex-def"><a name="L1168"></a><tt class="py-lineno">1168</tt> <a class="py-toggle" href="#" id="BranchColor.from_hex-toggle" onclick="return toggle('BranchColor.from_hex');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.BranchColor-class.html#from_hex">from_hex</a><tt class="py-op">(</tt><tt class="py-param">cls</tt><tt class="py-op">,</tt> <tt class="py-param">hexstr</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BranchColor.from_hex-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="BranchColor.from_hex-expanded"><a name="L1169"></a><tt class="py-lineno">1169</tt>  <tt class="py-line">        <tt class="py-docstring">"""Construct a BranchColor object from a hexadecimal string.</tt> </tt>
<a name="L1170"></a><tt class="py-lineno">1170</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1171"></a><tt class="py-lineno">1171</tt>  <tt class="py-line"><tt class="py-docstring">        The string format is the same style used in HTML and CSS, such as</tt> </tt>
<a name="L1172"></a><tt class="py-lineno">1172</tt>  <tt class="py-line"><tt class="py-docstring">        '#FF8000' for an RGB value of (255, 128, 0).</tt> </tt>
<a name="L1173"></a><tt class="py-lineno">1173</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1174"></a><tt class="py-lineno">1174</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-op">(</tt><tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">hexstr</tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> </tt>
<a name="L1175"></a><tt class="py-lineno">1175</tt>  <tt class="py-line">                <tt class="py-name">hexstr</tt><tt class="py-op">.</tt><tt id="link-372" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-372', 'startswith', 'link-351');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'#'</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> </tt>
<a name="L1176"></a><tt class="py-lineno">1176</tt>  <tt class="py-line">                <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">hexstr</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">7</tt> </tt>
<a name="L1177"></a><tt class="py-lineno">1177</tt>  <tt class="py-line">                <tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-string">"need a 24-bit hexadecimal string, e.g. #000000"</tt> </tt>
<a name="L1178"></a><tt class="py-lineno">1178</tt>  <tt class="py-line"> </tt>
<a name="L1179"></a><tt class="py-lineno">1179</tt>  <tt class="py-line">        <tt class="py-name">RGB</tt> <tt class="py-op">=</tt> <tt class="py-name">hexstr</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-number">3</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">hexstr</tt><tt class="py-op">[</tt><tt class="py-number">3</tt><tt class="py-op">:</tt><tt class="py-number">5</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">hexstr</tt><tt class="py-op">[</tt><tt class="py-number">5</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L1180"></a><tt class="py-lineno">1180</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">cls</tt><tt class="py-op">(</tt><tt class="py-op">*</tt><tt class="py-op">[</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-string">'0x'</tt> <tt class="py-op">+</tt> <tt class="py-name">cc</tt><tt class="py-op">,</tt> <tt class="py-name">base</tt><tt class="py-op">=</tt><tt class="py-number">16</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">cc</tt> <tt class="py-keyword">in</tt> <tt class="py-name">RGB</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1181"></a><tt class="py-lineno">1181</tt>  <tt class="py-line"> </tt>
<a name="L1182"></a><tt class="py-lineno">1182</tt>  <tt class="py-line">    <tt class="py-decorator">@</tt><tt class="py-decorator">classmethod</tt> </tt>
<a name="BranchColor.from_name"></a><div id="BranchColor.from_name-def"><a name="L1183"></a><tt class="py-lineno">1183</tt> <a class="py-toggle" href="#" id="BranchColor.from_name-toggle" onclick="return toggle('BranchColor.from_name');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.BranchColor-class.html#from_name">from_name</a><tt class="py-op">(</tt><tt class="py-param">cls</tt><tt class="py-op">,</tt> <tt class="py-param">colorname</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BranchColor.from_name-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="BranchColor.from_name-expanded"><a name="L1184"></a><tt class="py-lineno">1184</tt>  <tt class="py-line">        <tt class="py-docstring">"""Construct a BranchColor object by the color's name."""</tt> </tt>
<a name="L1185"></a><tt class="py-lineno">1185</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">cls</tt><tt class="py-op">(</tt><tt class="py-op">*</tt><tt class="py-name">cls</tt><tt class="py-op">.</tt><tt id="link-373" class="py-name"><a title="Bio.Phylo.BaseTree.BranchColor.color_names" class="py-name" href="#" onclick="return doclink('link-373', 'color_names', 'link-348');">color_names</a></tt><tt class="py-op">[</tt><tt class="py-name">colorname</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1186"></a><tt class="py-lineno">1186</tt>  <tt class="py-line"> </tt>
<a name="BranchColor.to_hex"></a><div id="BranchColor.to_hex-def"><a name="L1187"></a><tt class="py-lineno">1187</tt> <a class="py-toggle" href="#" id="BranchColor.to_hex-toggle" onclick="return toggle('BranchColor.to_hex');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.BranchColor-class.html#to_hex">to_hex</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BranchColor.to_hex-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="BranchColor.to_hex-expanded"><a name="L1188"></a><tt class="py-lineno">1188</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a 24-bit hexadecimal RGB representation of this color.</tt> </tt>
<a name="L1189"></a><tt class="py-lineno">1189</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1190"></a><tt class="py-lineno">1190</tt>  <tt class="py-line"><tt class="py-docstring">        The returned string is suitable for use in HTML/CSS, as a color</tt> </tt>
<a name="L1191"></a><tt class="py-lineno">1191</tt>  <tt class="py-line"><tt class="py-docstring">        parameter in matplotlib, and perhaps other situations.</tt> </tt>
<a name="L1192"></a><tt class="py-lineno">1192</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1193"></a><tt class="py-lineno">1193</tt>  <tt class="py-line"><tt class="py-docstring">        Examples</tt> </tt>
<a name="L1194"></a><tt class="py-lineno">1194</tt>  <tt class="py-line"><tt class="py-docstring">        --------</tt> </tt>
<a name="L1195"></a><tt class="py-lineno">1195</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; bc = BranchColor(12, 200, 100)</tt> </tt>
<a name="L1196"></a><tt class="py-lineno">1196</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; bc.to_hex()</tt> </tt>
<a name="L1197"></a><tt class="py-lineno">1197</tt>  <tt class="py-line"><tt class="py-docstring">        '#0cc864'</tt> </tt>
<a name="L1198"></a><tt class="py-lineno">1198</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1199"></a><tt class="py-lineno">1199</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1200"></a><tt class="py-lineno">1200</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">"#%02x%02x%02x"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-374" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.red" class="py-name" href="#" onclick="return doclink('link-374', 'red', 'link-361');">red</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-375" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.green" class="py-name" href="#" onclick="return doclink('link-375', 'green', 'link-362');">green</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-376" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.blue" class="py-name" href="#" onclick="return doclink('link-376', 'blue', 'link-363');">blue</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1201"></a><tt class="py-lineno">1201</tt>  <tt class="py-line"> </tt>
<a name="BranchColor.to_rgb"></a><div id="BranchColor.to_rgb-def"><a name="L1202"></a><tt class="py-lineno">1202</tt> <a class="py-toggle" href="#" id="BranchColor.to_rgb-toggle" onclick="return toggle('BranchColor.to_rgb');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.BranchColor-class.html#to_rgb">to_rgb</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BranchColor.to_rgb-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="BranchColor.to_rgb-expanded"><a name="L1203"></a><tt class="py-lineno">1203</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a tuple of RGB values (0 to 255) representing this color.</tt> </tt>
<a name="L1204"></a><tt class="py-lineno">1204</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1205"></a><tt class="py-lineno">1205</tt>  <tt class="py-line"><tt class="py-docstring">        Examples</tt> </tt>
<a name="L1206"></a><tt class="py-lineno">1206</tt>  <tt class="py-line"><tt class="py-docstring">        --------</tt> </tt>
<a name="L1207"></a><tt class="py-lineno">1207</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; bc = BranchColor(255, 165, 0)</tt> </tt>
<a name="L1208"></a><tt class="py-lineno">1208</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; bc.to_rgb()</tt> </tt>
<a name="L1209"></a><tt class="py-lineno">1209</tt>  <tt class="py-line"><tt class="py-docstring">        (255, 165, 0)</tt> </tt>
<a name="L1210"></a><tt class="py-lineno">1210</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1211"></a><tt class="py-lineno">1211</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1212"></a><tt class="py-lineno">1212</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-377" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.red" class="py-name" href="#" onclick="return doclink('link-377', 'red', 'link-361');">red</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-378" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.green" class="py-name" href="#" onclick="return doclink('link-378', 'green', 'link-362');">green</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-379" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.blue" class="py-name" href="#" onclick="return doclink('link-379', 'blue', 'link-363');">blue</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1213"></a><tt class="py-lineno">1213</tt>  <tt class="py-line"> </tt>
<a name="BranchColor.__repr__"></a><div id="BranchColor.__repr__-def"><a name="L1214"></a><tt class="py-lineno">1214</tt> <a class="py-toggle" href="#" id="BranchColor.__repr__-toggle" onclick="return toggle('BranchColor.__repr__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.BranchColor-class.html#__repr__">__repr__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BranchColor.__repr__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="BranchColor.__repr__-expanded"><a name="L1215"></a><tt class="py-lineno">1215</tt>  <tt class="py-line">        <tt class="py-docstring">"""Preserve the standard RGB order when representing this object."""</tt> </tt>
<a name="L1216"></a><tt class="py-lineno">1216</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-string">'%s(red=%d, green=%d, blue=%d)'</tt> </tt>
<a name="L1217"></a><tt class="py-lineno">1217</tt>  <tt class="py-line">                <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-380" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.red" class="py-name" href="#" onclick="return doclink('link-380', 'red', 'link-361');">red</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-381" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.green" class="py-name" href="#" onclick="return doclink('link-381', 'green', 'link-362');">green</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-382" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.blue" class="py-name" href="#" onclick="return doclink('link-382', 'blue', 'link-363');">blue</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1218"></a><tt class="py-lineno">1218</tt>  <tt class="py-line"> </tt>
<a name="BranchColor.__str__"></a><div id="BranchColor.__str__-def"><a name="L1219"></a><tt class="py-lineno">1219</tt> <a class="py-toggle" href="#" id="BranchColor.__str__-toggle" onclick="return toggle('BranchColor.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Phylo.BaseTree.BranchColor-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BranchColor.__str__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="BranchColor.__str__-expanded"><a name="L1220"></a><tt class="py-lineno">1220</tt>  <tt class="py-line">        <tt class="py-docstring">"""Show the color's RGB values."""</tt> </tt>
<a name="L1221"></a><tt class="py-lineno">1221</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">"(%d, %d, %d)"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-383" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.red" class="py-name" href="#" onclick="return doclink('link-383', 'red', 'link-361');">red</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-384" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.green" class="py-name" href="#" onclick="return doclink('link-384', 'green', 'link-362');">green</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-385" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.blue" class="py-name" href="#" onclick="return doclink('link-385', 'blue', 'link-363');">blue</a></tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1222"></a><tt class="py-lineno">1222</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Tue Dec 18 15:58:15 2018
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
