<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.SearchIO.ExonerateIO._base</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.SearchIO-module.html">Package&nbsp;SearchIO</a> ::
        <a href="Bio.SearchIO.ExonerateIO-module.html">Package&nbsp;ExonerateIO</a> ::
        Module&nbsp;_base
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.SearchIO.ExonerateIO._base-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.SearchIO.ExonerateIO._base-module.html" onclick="show_private();">Module Bio.SearchIO.ExonerateIO._base</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2012 by Wibowo Arindrarto.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-docstring">"""Bio.SearchIO abstract base parser for Exonerate standard output format."""</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">re</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">functools</tt> <tt class="py-keyword">import</tt> <tt class="py-name">reduce</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio.SearchIO=Bio.SearchIO-module.html"><a title="Bio.SearchIO" class="py-name" href="#" onclick="return doclink('link-1', 'SearchIO', 'link-1');">SearchIO</a></tt><tt class="py-op">.</tt><tt id="link-2" class="py-name" targets="Module Bio.SearchIO._index=Bio.SearchIO._index-module.html,Module Bio.SeqIO._index=Bio.SeqIO._index-module.html"><a title="Bio.SearchIO._index
Bio.SeqIO._index" class="py-name" href="#" onclick="return doclink('link-2', '_index', 'link-2');">_index</a></tt> <tt class="py-keyword">import</tt> <tt id="link-3" class="py-name" targets="Class Bio.SearchIO._index.SearchIndexer=Bio.SearchIO._index.SearchIndexer-class.html"><a title="Bio.SearchIO._index.SearchIndexer" class="py-name" href="#" onclick="return doclink('link-3', 'SearchIndexer', 'link-3');">SearchIndexer</a></tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-4" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-4', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-5" class="py-name"><a title="Bio.SearchIO" class="py-name" href="#" onclick="return doclink('link-5', 'SearchIO', 'link-1');">SearchIO</a></tt><tt class="py-op">.</tt><tt id="link-6" class="py-name" targets="Package Bio.SearchIO._model=Bio.SearchIO._model-module.html"><a title="Bio.SearchIO._model" class="py-name" href="#" onclick="return doclink('link-6', '_model', 'link-6');">_model</a></tt> <tt class="py-keyword">import</tt> <tt id="link-7" class="py-name" targets="Class Bio.SearchIO._model.query.QueryResult=Bio.SearchIO._model.query.QueryResult-class.html"><a title="Bio.SearchIO._model.query.QueryResult" class="py-name" href="#" onclick="return doclink('link-7', 'QueryResult', 'link-7');">QueryResult</a></tt><tt class="py-op">,</tt> <tt id="link-8" class="py-name" targets="Class Bio.SearchIO._model.hit.Hit=Bio.SearchIO._model.hit.Hit-class.html"><a title="Bio.SearchIO._model.hit.Hit" class="py-name" href="#" onclick="return doclink('link-8', 'Hit', 'link-8');">Hit</a></tt><tt class="py-op">,</tt> <tt id="link-9" class="py-name" targets="Class Bio.Blast.Record.HSP=Bio.Blast.Record.HSP-class.html,Class Bio.SearchIO._model.hsp.HSP=Bio.SearchIO._model.hsp.HSP-class.html"><a title="Bio.Blast.Record.HSP
Bio.SearchIO._model.hsp.HSP" class="py-name" href="#" onclick="return doclink('link-9', 'HSP', 'link-9');">HSP</a></tt><tt class="py-op">,</tt> <tt id="link-10" class="py-name" targets="Class Bio.SearchIO._model.hsp.HSPFragment=Bio.SearchIO._model.hsp.HSPFragment-class.html"><a title="Bio.SearchIO._model.hsp.HSPFragment" class="py-name" href="#" onclick="return doclink('link-10', 'HSPFragment', 'link-10');">HSPFragment</a></tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-11" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-11', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-12" class="py-name" targets="Package Bio.SeqUtils=Bio.SeqUtils-module.html"><a title="Bio.SeqUtils" class="py-name" href="#" onclick="return doclink('link-12', 'SeqUtils', 'link-12');">SeqUtils</a></tt> <tt class="py-keyword">import</tt> <tt id="link-13" class="py-name" targets="Function Bio.SeqUtils.seq1()=Bio.SeqUtils-module.html#seq1"><a title="Bio.SeqUtils.seq1" class="py-name" href="#" onclick="return doclink('link-13', 'seq1', 'link-13');">seq1</a></tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-comment"># strand char-value mapping</tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt id="link-14" class="py-name" targets="Variable Bio.SearchIO.ExonerateIO._base._STRAND_MAP=Bio.SearchIO.ExonerateIO._base-module.html#_STRAND_MAP"><a title="Bio.SearchIO.ExonerateIO._base._STRAND_MAP" class="py-name" href="#" onclick="return doclink('link-14', '_STRAND_MAP', 'link-14');">_STRAND_MAP</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'+'</tt><tt class="py-op">:</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-string">'-'</tt><tt class="py-op">:</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-string">'.'</tt><tt class="py-op">:</tt> <tt class="py-number">0</tt><tt class="py-op">}</tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt id="link-15" class="py-name" targets="Variable Bio.SearchIO.ExonerateIO._base._RE_SHIFTS=Bio.SearchIO.ExonerateIO._base-module.html#_RE_SHIFTS"><a title="Bio.SearchIO.ExonerateIO._base._RE_SHIFTS" class="py-name" href="#" onclick="return doclink('link-15', '_RE_SHIFTS', 'link-15');">_RE_SHIFTS</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'(#+)'</tt><tt class="py-op">)</tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-comment"># regex for checking whether a vulgar line has protein/translated components</tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt id="link-16" class="py-name" targets="Variable Bio.SearchIO.ExonerateIO._base._RE_TRANS=Bio.SearchIO.ExonerateIO._base-module.html#_RE_TRANS"><a title="Bio.SearchIO.ExonerateIO._base._RE_TRANS" class="py-name" href="#" onclick="return doclink('link-16', '_RE_TRANS', 'link-16');">_RE_TRANS</a></tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt class="py-name">compile</tt><tt class="py-op">(</tt><tt class="py-string">r'[53ISCF]'</tt><tt class="py-op">)</tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"> </tt>
<a name="_set_frame"></a><div id="_set_frame-def"><a name="L24"></a><tt class="py-lineno"> 24</tt> <a class="py-toggle" href="#" id="_set_frame-toggle" onclick="return toggle('_set_frame');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO._base-module.html#_set_frame">_set_frame</a><tt class="py-op">(</tt><tt class="py-param">frag</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_set_frame-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_set_frame-expanded"><a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line">    <tt class="py-docstring">"""Set the HSPFragment frames (PRIVATE)."""</tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line">    <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-17" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.hit_frame=Bio.SearchIO._model.hsp.HSP-class.html#hit_frame,Variable Bio.SearchIO._model.hsp.HSPFragment.hit_frame=Bio.SearchIO._model.hsp.HSPFragment-class.html#hit_frame"><a title="Bio.SearchIO._model.hsp.HSP.hit_frame
Bio.SearchIO._model.hsp.HSPFragment.hit_frame" class="py-name" href="#" onclick="return doclink('link-17', 'hit_frame', 'link-17');">hit_frame</a></tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-18" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.hit_start=Bio.SearchIO._model.hsp.HSP-class.html#hit_start,Variable Bio.SearchIO._model.hsp.HSPFragment.hit_start=Bio.SearchIO._model.hsp.HSPFragment-class.html#hit_start"><a title="Bio.SearchIO._model.hsp.HSP.hit_start
Bio.SearchIO._model.hsp.HSPFragment.hit_start" class="py-name" href="#" onclick="return doclink('link-18', 'hit_start', 'link-18');">hit_start</a></tt> <tt class="py-op">%</tt> <tt class="py-number">3</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-19" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.hit_strand=Bio.SearchIO._model.hsp.HSP-class.html#hit_strand,Variable Bio.SearchIO._model.hsp.HSPFragment.hit_strand=Bio.SearchIO._model.hsp.HSPFragment-class.html#hit_strand"><a title="Bio.SearchIO._model.hsp.HSP.hit_strand
Bio.SearchIO._model.hsp.HSPFragment.hit_strand" class="py-name" href="#" onclick="return doclink('link-19', 'hit_strand', 'link-19');">hit_strand</a></tt> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line">    <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-20" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.query_frame=Bio.SearchIO._model.hsp.HSP-class.html#query_frame,Variable Bio.SearchIO._model.hsp.HSPFragment.query_frame=Bio.SearchIO._model.hsp.HSPFragment-class.html#query_frame"><a title="Bio.SearchIO._model.hsp.HSP.query_frame
Bio.SearchIO._model.hsp.HSPFragment.query_frame" class="py-name" href="#" onclick="return doclink('link-20', 'query_frame', 'link-20');">query_frame</a></tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-21" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.query_start=Bio.SearchIO._model.hsp.HSP-class.html#query_start,Variable Bio.SearchIO._model.hsp.HSPFragment.query_start=Bio.SearchIO._model.hsp.HSPFragment-class.html#query_start"><a title="Bio.SearchIO._model.hsp.HSP.query_start
Bio.SearchIO._model.hsp.HSPFragment.query_start" class="py-name" href="#" onclick="return doclink('link-21', 'query_start', 'link-21');">query_start</a></tt> <tt class="py-op">%</tt> <tt class="py-number">3</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-22" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.query_strand=Bio.SearchIO._model.hsp.HSP-class.html#query_strand,Variable Bio.SearchIO._model.hsp.HSPFragment.query_strand=Bio.SearchIO._model.hsp.HSPFragment-class.html#query_strand"><a title="Bio.SearchIO._model.hsp.HSP.query_strand
Bio.SearchIO._model.hsp.HSPFragment.query_strand" class="py-name" href="#" onclick="return doclink('link-22', 'query_strand', 'link-22');">query_strand</a></tt> </tt>
</div><a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"> </tt>
<a name="_make_triplets"></a><div id="_make_triplets-def"><a name="L30"></a><tt class="py-lineno"> 30</tt> <a class="py-toggle" href="#" id="_make_triplets-toggle" onclick="return toggle('_make_triplets');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO._base-module.html#_make_triplets">_make_triplets</a><tt class="py-op">(</tt><tt class="py-param">seq</tt><tt class="py-op">,</tt> <tt class="py-param">phase</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_make_triplets-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_make_triplets-expanded"><a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line">    <tt class="py-docstring">"""Select a valid amino acid sequence given a 3-letter code input (PRIVATE).</tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-docstring">    This function takes a single three-letter amino acid sequence and the phase</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-docstring">    of the sequence to return the longest intact amino acid sequence possible.</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-docstring">    Parts of the input sequence before and after the selected sequence are also</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"><tt class="py-docstring">    returned.</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"><tt class="py-docstring">    This is an internal private function and is meant for parsing Exonerate's</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line"><tt class="py-docstring">    three-letter amino acid output.</tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio.SearchIO.ExonerateIO._base import _make_triplets</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _make_triplets('GlyThrSerAlaPro')</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line"><tt class="py-docstring">    ('', ['Gly', 'Thr', 'Ser', 'Ala', 'Pro'], '')</tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _make_triplets('yThSerAla', phase=1)</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line"><tt class="py-docstring">    ('y', ['Thr', 'Ser', 'Ala'], '')</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _make_triplets('yThSerAlaPr', phase=1)</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line"><tt class="py-docstring">    ('y', ['Thr', 'Ser', 'Ala'], 'Pr')</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line">    <tt class="py-name">pre</tt> <tt class="py-op">=</tt> <tt id="link-23" class="py-name" targets="Variable Bio.SeqRecord.SeqRecord.seq=Bio.SeqRecord.SeqRecord-class.html#seq,Variable BioSQL.BioSeq.DBSeqRecord.seq=BioSQL.BioSeq.DBSeqRecord-class.html#seq"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-23', 'seq', 'link-23');">seq</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-name">phase</tt><tt class="py-op">]</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line">    <tt class="py-name">np_seq</tt> <tt class="py-op">=</tt> <tt id="link-24" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-24', 'seq', 'link-23');">seq</a></tt><tt class="py-op">[</tt><tt class="py-name">phase</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line">    <tt class="py-name">non_triplets</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">np_seq</tt><tt class="py-op">)</tt> <tt class="py-op">%</tt> <tt class="py-number">3</tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line">    <tt class="py-name">post</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">non_triplets</tt> <tt class="py-keyword">else</tt> <tt class="py-name">np_seq</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt> <tt class="py-op">*</tt> <tt class="py-name">non_triplets</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line">    <tt class="py-name">intacts</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">np_seq</tt><tt class="py-op">[</tt><tt class="py-number">3</tt> <tt class="py-op">*</tt> <tt id="link-25" class="py-name" targets="Variable Bio.PDB.Polypeptide.i=Bio.PDB.Polypeptide-module.html#i,Variable Bio.SeqIO.SffIO.i=Bio.SeqIO.SffIO-module.html#i"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-25', 'i', 'link-25');">i</a></tt><tt class="py-op">:</tt><tt class="py-number">3</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt id="link-26" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-26', 'i', 'link-25');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line">               <tt class="py-keyword">for</tt> <tt id="link-27" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-27', 'i', 'link-25');">i</a></tt> <tt class="py-keyword">in</tt> <tt id="link-28" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.range()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#range,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.range()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#range,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Track.Track.range()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#range"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-28', 'range', 'link-28');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">np_seq</tt><tt class="py-op">)</tt> <tt class="py-op">//</tt> <tt class="py-number">3</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">pre</tt><tt class="py-op">,</tt> <tt class="py-name">intacts</tt><tt class="py-op">,</tt> <tt class="py-name">post</tt> </tt>
</div><a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line"> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line"> </tt>
<a name="_get_fragments_coord"></a><div id="_get_fragments_coord-def"><a name="L59"></a><tt class="py-lineno"> 59</tt> <a class="py-toggle" href="#" id="_get_fragments_coord-toggle" onclick="return toggle('_get_fragments_coord');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO._base-module.html#_get_fragments_coord">_get_fragments_coord</a><tt class="py-op">(</tt><tt class="py-param">frags</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_get_fragments_coord-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_get_fragments_coord-expanded"><a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line">    <tt class="py-docstring">"""Return the letter coordinate of the given list of fragments (PRIVATE).</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line"><tt class="py-docstring">    This function takes a list of three-letter amino acid sequences and</tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line"><tt class="py-docstring">    returns a list of coordinates for each fragment had all the input</tt> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line"><tt class="py-docstring">    sequences been flattened.</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line"><tt class="py-docstring">    This is an internal private function and is meant for parsing Exonerate's</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line"><tt class="py-docstring">    three-letter amino acid output.</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio.SearchIO.ExonerateIO._base import _get_fragments_coord</tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _get_fragments_coord(['Thr', 'Ser', 'Ala'])</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line"><tt class="py-docstring">    [0, 3, 6]</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _get_fragments_coord(['Thr', 'SerAlaPro', 'GlyLeu'])</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line"><tt class="py-docstring">    [0, 3, 12, ]</tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _get_fragments_coord(['Thr', 'SerAlaPro', 'GlyLeu', 'Cys'])</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line"><tt class="py-docstring">    [0, 3, 12, 18]</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">frags</tt><tt class="py-op">:</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line">    <tt class="py-comment"># first fragment always starts from position 0</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line">    <tt class="py-name">init</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">reduce</tt><tt class="py-op">(</tt><tt class="py-keyword">lambda</tt> <tt id="link-29" class="py-name" targets="Variable Bio.FSSP.fssp_rec.align.acc=Bio.FSSP.fssp_rec.align-class.html#acc"><a title="Bio.FSSP.fssp_rec.align.acc" class="py-name" href="#" onclick="return doclink('link-29', 'acc', 'link-29');">acc</a></tt><tt class="py-op">,</tt> <tt class="py-name">frag</tt><tt class="py-op">:</tt> <tt id="link-30" class="py-name"><a title="Bio.FSSP.fssp_rec.align.acc" class="py-name" href="#" onclick="return doclink('link-30', 'acc', 'link-29');">acc</a></tt> <tt class="py-op">+</tt> <tt class="py-op">[</tt><tt id="link-31" class="py-name"><a title="Bio.FSSP.fssp_rec.align.acc" class="py-name" href="#" onclick="return doclink('link-31', 'acc', 'link-29');">acc</a></tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line">                  <tt class="py-name">frags</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">init</tt><tt class="py-op">)</tt> </tt>
</div><a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line"> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line"> </tt>
<a name="_get_fragments_phase"></a><div id="_get_fragments_phase-def"><a name="L86"></a><tt class="py-lineno"> 86</tt> <a class="py-toggle" href="#" id="_get_fragments_phase-toggle" onclick="return toggle('_get_fragments_phase');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO._base-module.html#_get_fragments_phase">_get_fragments_phase</a><tt class="py-op">(</tt><tt class="py-param">frags</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_get_fragments_phase-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_get_fragments_phase-expanded"><a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line">    <tt class="py-docstring">"""Return the phases of the given list of 3-letter amino acid fragments (PRIVATE).</tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line"><tt class="py-docstring">    This is an internal private function and is meant for parsing Exonerate's</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line"><tt class="py-docstring">    three-letter amino acid output.</tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio.SearchIO.ExonerateIO._base import _get_fragments_phase</tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _get_fragments_phase(['Thr', 'Ser', 'Ala'])</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line"><tt class="py-docstring">    [0, 0, 0]</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _get_fragments_phase(['ThrSe', 'rAla'])</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line"><tt class="py-docstring">    [0, 1]</tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _get_fragments_phase(['ThrSe', 'rAlaLeu', 'ProCys'])</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line"><tt class="py-docstring">    [0, 1, 0]</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _get_fragments_phase(['ThrSe', 'rAlaLeuP', 'roCys'])</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line"><tt class="py-docstring">    [0, 1, 2]</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _get_fragments_phase(['ThrSe', 'rAlaLeuPr', 'oCys'])</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line"><tt class="py-docstring">    [0, 1, 1]</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-number">3</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt id="link-32" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.x=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#x,Variable Bio.Statistics.lowess.x=Bio.Statistics.lowess-module.html#x"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-32', 'x', 'link-32');">x</a></tt> <tt class="py-op">%</tt> <tt class="py-number">3</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">%</tt> <tt class="py-number">3</tt> <tt class="py-keyword">for</tt> <tt id="link-33" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-33', 'x', 'link-32');">x</a></tt> <tt class="py-keyword">in</tt> <tt id="link-34" class="py-name" targets="Function Bio.SearchIO.ExonerateIO._base._get_fragments_coord()=Bio.SearchIO.ExonerateIO._base-module.html#_get_fragments_coord"><a title="Bio.SearchIO.ExonerateIO._base._get_fragments_coord" class="py-name" href="#" onclick="return doclink('link-34', '_get_fragments_coord', 'link-34');">_get_fragments_coord</a></tt><tt class="py-op">(</tt><tt class="py-name">frags</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
</div><a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line"> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line"> </tt>
<a name="_adjust_aa_seq"></a><div id="_adjust_aa_seq-def"><a name="L108"></a><tt class="py-lineno">108</tt> <a class="py-toggle" href="#" id="_adjust_aa_seq-toggle" onclick="return toggle('_adjust_aa_seq');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO._base-module.html#_adjust_aa_seq">_adjust_aa_seq</a><tt class="py-op">(</tt><tt class="py-param">fraglist</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_adjust_aa_seq-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_adjust_aa_seq-expanded"><a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line">    <tt class="py-docstring">"""Transform 3-letter AA codes of input fragments to one-letter codes (PRIVATE).</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line"><tt class="py-docstring">    Argument fraglist should be a list of HSPFragments objects.</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">    <tt class="py-name">custom_map</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'***'</tt><tt class="py-op">:</tt> <tt class="py-string">'*'</tt><tt class="py-op">,</tt> <tt class="py-string">'&lt;-&gt;'</tt><tt class="py-op">:</tt> <tt class="py-string">'-'</tt><tt class="py-op">}</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">    <tt class="py-name">hsp_hstart</tt> <tt class="py-op">=</tt> <tt class="py-name">fraglist</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-35" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.hit_start
Bio.SearchIO._model.hsp.HSPFragment.hit_start" class="py-name" href="#" onclick="return doclink('link-35', 'hit_start', 'link-18');">hit_start</a></tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">    <tt class="py-name">hsp_qstart</tt> <tt class="py-op">=</tt> <tt class="py-name">fraglist</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-36" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_start
Bio.SearchIO._model.hsp.HSPFragment.query_start" class="py-name" href="#" onclick="return doclink('link-36', 'query_start', 'link-21');">query_start</a></tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">    <tt class="py-name">frag_phases</tt> <tt class="py-op">=</tt> <tt id="link-37" class="py-name" targets="Function Bio.SearchIO.ExonerateIO._base._get_fragments_phase()=Bio.SearchIO.ExonerateIO._base-module.html#_get_fragments_phase"><a title="Bio.SearchIO.ExonerateIO._base._get_fragments_phase" class="py-name" href="#" onclick="return doclink('link-37', '_get_fragments_phase', 'link-37');">_get_fragments_phase</a></tt><tt class="py-op">(</tt><tt class="py-name">fraglist</tt><tt class="py-op">)</tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">frag</tt><tt class="py-op">,</tt> <tt class="py-name">phase</tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">fraglist</tt><tt class="py-op">,</tt> <tt class="py-name">frag_phases</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-38" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_strand
Bio.SearchIO._model.hsp.HSPFragment.query_strand" class="py-name" href="#" onclick="return doclink('link-38', 'query_strand', 'link-22');">query_strand</a></tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt> <tt class="py-keyword">or</tt> <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-39" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.hit_strand
Bio.SearchIO._model.hsp.HSPFragment.hit_strand" class="py-name" href="#" onclick="return doclink('link-39', 'hit_strand', 'link-19');">hit_strand</a></tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line">        <tt class="py-comment"># hit step may be -1 as we're aligning to DNA</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">        <tt class="py-name">hstep</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> <tt class="py-keyword">if</tt> <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-40" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.hit_strand
Bio.SearchIO._model.hsp.HSPFragment.hit_strand" class="py-name" href="#" onclick="return doclink('link-40', 'hit_strand', 'link-19');">hit_strand</a></tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">0</tt> <tt class="py-keyword">else</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line"> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line">        <tt class="py-comment"># set fragment phase</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line">        <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt class="py-name">phase</tt> <tt class="py-op">=</tt> <tt class="py-name">phase</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line"> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line">        <tt class="py-comment"># fragment should have a length that is a multiple of 3</tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line">        <tt class="py-comment"># assert len(frag) % 3 == 0</tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line">        <tt class="py-name">qseq</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-41" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HSPConsumer.query()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#query,Variable Bio.SearchIO._model.hsp.HSP.query=Bio.SearchIO._model.hsp.HSP-class.html#query,Variable Bio.SearchIO._model.hsp.HSPFragment.query=Bio.SearchIO._model.hsp.HSPFragment-class.html#query,Module Bio.SearchIO._model.query=Bio.SearchIO._model.query-module.html"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.query
Bio.SearchIO._model.hsp.HSP.query
Bio.SearchIO._model.hsp.HSPFragment.query
Bio.SearchIO._model.query" class="py-name" href="#" onclick="return doclink('link-41', 'query', 'link-41');">query</a></tt><tt class="py-op">.</tt><tt id="link-42" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-42', 'seq', 'link-23');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line">        <tt class="py-name">q_triplets_pre</tt><tt class="py-op">,</tt> <tt class="py-name">q_triplets</tt><tt class="py-op">,</tt> <tt class="py-name">q_triplets_post</tt> <tt class="py-op">=</tt> \ </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line">            <tt id="link-43" class="py-name" targets="Function Bio.SearchIO.ExonerateIO._base._make_triplets()=Bio.SearchIO.ExonerateIO._base-module.html#_make_triplets"><a title="Bio.SearchIO.ExonerateIO._base._make_triplets" class="py-name" href="#" onclick="return doclink('link-43', '_make_triplets', 'link-43');">_make_triplets</a></tt><tt class="py-op">(</tt><tt class="py-name">qseq</tt><tt class="py-op">,</tt> <tt class="py-name">phase</tt><tt class="py-op">)</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line"> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line">        <tt class="py-name">hseq</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-44" class="py-name" targets="Module Bio.SearchIO._model.hit=Bio.SearchIO._model.hit-module.html,Variable Bio.SearchIO._model.hsp.HSP.hit=Bio.SearchIO._model.hsp.HSP-class.html#hit,Variable Bio.SearchIO._model.hsp.HSPFragment.hit=Bio.SearchIO._model.hsp.HSPFragment-class.html#hit"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-44', 'hit', 'link-44');">hit</a></tt><tt class="py-op">.</tt><tt id="link-45" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-45', 'seq', 'link-23');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line">        <tt class="py-name">h_triplets_pre</tt><tt class="py-op">,</tt> <tt class="py-name">h_triplets</tt><tt class="py-op">,</tt> <tt class="py-name">h_triplets_post</tt> <tt class="py-op">=</tt> \ </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line">            <tt id="link-46" class="py-name"><a title="Bio.SearchIO.ExonerateIO._base._make_triplets" class="py-name" href="#" onclick="return doclink('link-46', '_make_triplets', 'link-43');">_make_triplets</a></tt><tt class="py-op">(</tt><tt class="py-name">hseq</tt><tt class="py-op">,</tt> <tt class="py-name">phase</tt><tt class="py-op">)</tt> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line"> </tt>
<a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line">        <tt class="py-comment"># get one letter codes</tt> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line">        <tt class="py-comment"># and replace gap codon markers and termination characters</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line">        <tt class="py-name">hseq1_pre</tt> <tt class="py-op">=</tt> <tt class="py-string">"X"</tt> <tt class="py-keyword">if</tt> <tt class="py-name">h_triplets_pre</tt> <tt class="py-keyword">else</tt> <tt class="py-string">""</tt> </tt>
<a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line">        <tt class="py-name">hseq1_post</tt> <tt class="py-op">=</tt> <tt class="py-string">"X"</tt> <tt class="py-keyword">if</tt> <tt class="py-name">h_triplets_post</tt> <tt class="py-keyword">else</tt> <tt class="py-string">""</tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line">        <tt class="py-name">hseq1</tt> <tt class="py-op">=</tt> <tt id="link-47" class="py-name"><a title="Bio.SeqUtils.seq1" class="py-name" href="#" onclick="return doclink('link-47', 'seq1', 'link-13');">seq1</a></tt><tt class="py-op">(</tt><tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">h_triplets</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">custom_map</tt><tt class="py-op">=</tt><tt class="py-name">custom_map</tt><tt class="py-op">)</tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line">        <tt class="py-name">hstart</tt> <tt class="py-op">=</tt> <tt class="py-name">hsp_hstart</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">hseq1_pre</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt class="py-name">hstep</tt><tt class="py-op">)</tt> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line">        <tt class="py-name">hend</tt> <tt class="py-op">=</tt> <tt class="py-name">hstart</tt> <tt class="py-op">+</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">hseq1</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'-'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt class="py-name">hstep</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line"> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line">        <tt class="py-name">qseq1_pre</tt> <tt class="py-op">=</tt> <tt class="py-string">"X"</tt> <tt class="py-keyword">if</tt> <tt class="py-name">q_triplets_pre</tt> <tt class="py-keyword">else</tt> <tt class="py-string">""</tt> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line">        <tt class="py-name">qseq1_post</tt> <tt class="py-op">=</tt> <tt class="py-string">"X"</tt> <tt class="py-keyword">if</tt> <tt class="py-name">q_triplets_post</tt> <tt class="py-keyword">else</tt> <tt class="py-string">""</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line">        <tt class="py-name">qseq1</tt> <tt class="py-op">=</tt> <tt id="link-48" class="py-name"><a title="Bio.SeqUtils.seq1" class="py-name" href="#" onclick="return doclink('link-48', 'seq1', 'link-13');">seq1</a></tt><tt class="py-op">(</tt><tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">q_triplets</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">custom_map</tt><tt class="py-op">=</tt><tt class="py-name">custom_map</tt><tt class="py-op">)</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line">        <tt class="py-name">qstart</tt> <tt class="py-op">=</tt> <tt class="py-name">hsp_qstart</tt> <tt class="py-op">+</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">qseq1_pre</tt><tt class="py-op">)</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line">        <tt class="py-name">qend</tt> <tt class="py-op">=</tt> <tt class="py-name">qstart</tt> <tt class="py-op">+</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">qseq1</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'-'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line"> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line">        <tt class="py-comment"># replace the old frag sequences with the new ones</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">        <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-49" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-49', 'hit', 'link-44');">hit</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">        <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-50" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.query
Bio.SearchIO._model.hsp.HSP.query
Bio.SearchIO._model.hsp.HSPFragment.query
Bio.SearchIO._model.query" class="py-name" href="#" onclick="return doclink('link-50', 'query', 'link-41');">query</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line">        <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-51" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-51', 'hit', 'link-44');">hit</a></tt> <tt class="py-op">=</tt> <tt class="py-name">hseq1_pre</tt> <tt class="py-op">+</tt> <tt class="py-name">hseq1</tt> <tt class="py-op">+</tt> <tt class="py-name">hseq1_post</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">        <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-52" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.query
Bio.SearchIO._model.hsp.HSP.query
Bio.SearchIO._model.hsp.HSPFragment.query
Bio.SearchIO._model.query" class="py-name" href="#" onclick="return doclink('link-52', 'query', 'link-41');">query</a></tt> <tt class="py-op">=</tt> <tt class="py-name">qseq1_pre</tt> <tt class="py-op">+</tt> <tt class="py-name">qseq1</tt> <tt class="py-op">+</tt> <tt class="py-name">qseq1_post</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line"> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">        <tt class="py-comment"># set coordinates for the protein sequence</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-53" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_strand
Bio.SearchIO._model.hsp.HSPFragment.query_strand" class="py-name" href="#" onclick="return doclink('link-53', 'query_strand', 'link-22');">query_strand</a></tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">            <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-54" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_start
Bio.SearchIO._model.hsp.HSPFragment.query_start" class="py-name" href="#" onclick="return doclink('link-54', 'query_start', 'link-21');">query_start</a></tt><tt class="py-op">,</tt> <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-55" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.query_end=Bio.SearchIO._model.hsp.HSP-class.html#query_end,Variable Bio.SearchIO._model.hsp.HSPFragment.query_end=Bio.SearchIO._model.hsp.HSPFragment-class.html#query_end"><a title="Bio.SearchIO._model.hsp.HSP.query_end
Bio.SearchIO._model.hsp.HSPFragment.query_end" class="py-name" href="#" onclick="return doclink('link-55', 'query_end', 'link-55');">query_end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">qstart</tt><tt class="py-op">,</tt> <tt class="py-name">qend</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-56" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.hit_strand
Bio.SearchIO._model.hsp.HSPFragment.hit_strand" class="py-name" href="#" onclick="return doclink('link-56', 'hit_strand', 'link-19');">hit_strand</a></tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">            <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-57" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.hit_start
Bio.SearchIO._model.hsp.HSPFragment.hit_start" class="py-name" href="#" onclick="return doclink('link-57', 'hit_start', 'link-18');">hit_start</a></tt><tt class="py-op">,</tt> <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-58" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.hit_end=Bio.SearchIO._model.hsp.HSP-class.html#hit_end,Variable Bio.SearchIO._model.hsp.HSPFragment.hit_end=Bio.SearchIO._model.hsp.HSPFragment-class.html#hit_end"><a title="Bio.SearchIO._model.hsp.HSP.hit_end
Bio.SearchIO._model.hsp.HSPFragment.hit_end" class="py-name" href="#" onclick="return doclink('link-58', 'hit_end', 'link-58');">hit_end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">hstart</tt><tt class="py-op">,</tt> <tt class="py-name">hend</tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line"> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">        <tt class="py-comment"># update alignment annotation</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">        <tt class="py-comment"># by turning them into list of triplets</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">annot</tt><tt class="py-op">,</tt> <tt class="py-name">annotseq</tt> <tt class="py-keyword">in</tt> <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-59" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.aln_annotation=Bio.SearchIO._model.hsp.HSP-class.html#aln_annotation"><a title="Bio.SearchIO._model.hsp.HSP.aln_annotation" class="py-name" href="#" onclick="return doclink('link-59', 'aln_annotation', 'link-59');">aln_annotation</a></tt><tt class="py-op">.</tt><tt id="link-60" class="py-name" targets="Method Bio.Crystal.Crystal.items()=Bio.Crystal.Crystal-class.html#items,Method Bio.File._IndexedSeqFileDict.items()=Bio.File._IndexedSeqFileDict-class.html#items,Method Bio.Phylo.PhyloXML.Events.items()=Bio.Phylo.PhyloXML.Events-class.html#items,Variable Bio.SearchIO._model.query.QueryResult.items=Bio.SearchIO._model.query.QueryResult-class.html#items,Method BioSQL.BioSeqDatabase.BioSeqDatabase.items()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#items,Method BioSQL.BioSeqDatabase.DBServer.items()=BioSQL.BioSeqDatabase.DBServer-class.html#items"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-60', 'items', 'link-60');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">            <tt class="py-name">pre</tt><tt class="py-op">,</tt> <tt class="py-name">intact</tt><tt class="py-op">,</tt> <tt class="py-name">post</tt> <tt class="py-op">=</tt> <tt id="link-61" class="py-name"><a title="Bio.SearchIO.ExonerateIO._base._make_triplets" class="py-name" href="#" onclick="return doclink('link-61', '_make_triplets', 'link-43');">_make_triplets</a></tt><tt class="py-op">(</tt><tt class="py-name">annotseq</tt><tt class="py-op">,</tt> <tt class="py-name">phase</tt><tt class="py-op">)</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line">            <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-62" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.aln_annotation" class="py-name" href="#" onclick="return doclink('link-62', 'aln_annotation', 'link-59');">aln_annotation</a></tt><tt class="py-op">[</tt><tt class="py-name">annot</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> \ </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line">                <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-63" class="py-name" targets="Function Bio.FSSP.FSSPTools.filter()=Bio.FSSP.FSSPTools-module.html#filter,Method Bio.SearchIO._model.hit.Hit.filter()=Bio.SearchIO._model.hit.Hit-class.html#filter"><a title="Bio.FSSP.FSSPTools.filter
Bio.SearchIO._model.hit.Hit.filter" class="py-name" href="#" onclick="return doclink('link-63', 'filter', 'link-63');">filter</a></tt><tt class="py-op">(</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-name">pre</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-name">intact</tt> <tt class="py-op">+</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-64" class="py-name"><a title="Bio.FSSP.FSSPTools.filter
Bio.SearchIO._model.hit.Hit.filter" class="py-name" href="#" onclick="return doclink('link-64', 'filter', 'link-63');">filter</a></tt><tt class="py-op">(</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-name">post</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line"> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">        <tt class="py-comment"># update values for next iteration</tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line">        <tt class="py-name">hsp_hstart</tt><tt class="py-op">,</tt> <tt class="py-name">hsp_qstart</tt> <tt class="py-op">=</tt> <tt class="py-name">hend</tt><tt class="py-op">,</tt> <tt class="py-name">qend</tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line"> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">fraglist</tt> </tt>
</div><a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line"> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line"> </tt>
<a name="_split_fragment"></a><div id="_split_fragment-def"><a name="L174"></a><tt class="py-lineno">174</tt> <a class="py-toggle" href="#" id="_split_fragment-toggle" onclick="return toggle('_split_fragment');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO._base-module.html#_split_fragment">_split_fragment</a><tt class="py-op">(</tt><tt class="py-param">frag</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_split_fragment-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_split_fragment-expanded"><a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line">    <tt class="py-docstring">"""Split one HSPFragment containing frame-shifted alignment into two (PRIVATE)."""</tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">    <tt class="py-comment"># given an HSPFragment object with frameshift(s), this method splits it</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line">    <tt class="py-comment"># into fragments without frameshifts by sequentially chopping it off</tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">    <tt class="py-comment"># starting from the beginning</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line">    <tt class="py-name">simil</tt> <tt class="py-op">=</tt> <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-65" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.aln_annotation" class="py-name" href="#" onclick="return doclink('link-65', 'aln_annotation', 'link-59');">aln_annotation</a></tt><tt class="py-op">[</tt><tt class="py-string">'similarity'</tt><tt class="py-op">]</tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line">    <tt class="py-comment"># we should have at least 1 frame shift for splitting</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">simil</tt><tt class="py-op">.</tt><tt id="link-66" class="py-name" targets="Method Bio.Crystal.Chain.count()=Bio.Crystal.Chain-class.html#count,Method Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count()=Bio.NeuralNetwork.Gene.Pattern.PatternRepository-class.html#count,Method Bio.Seq.MutableSeq.count()=Bio.Seq.MutableSeq-class.html#count,Method Bio.Seq.Seq.count()=Bio.Seq.Seq-class.html#count,Method Bio.Seq.UnknownSeq.count()=Bio.Seq.UnknownSeq-class.html#count,Method Bio.motifs.Instances.count()=Bio.motifs.Instances-class.html#count"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-66', 'count', 'link-66');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">'#'</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line"> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">    <tt class="py-name">split_frags</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">    <tt class="py-name">qstep</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> <tt class="py-keyword">if</tt> <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-67" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_strand
Bio.SearchIO._model.hsp.HSPFragment.query_strand" class="py-name" href="#" onclick="return doclink('link-67', 'query_strand', 'link-22');">query_strand</a></tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">0</tt> <tt class="py-keyword">else</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line">    <tt class="py-name">hstep</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> <tt class="py-keyword">if</tt> <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-68" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.hit_strand
Bio.SearchIO._model.hsp.HSPFragment.hit_strand" class="py-name" href="#" onclick="return doclink('link-68', 'hit_strand', 'link-19');">hit_strand</a></tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">0</tt> <tt class="py-keyword">else</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line">    <tt class="py-name">qpos</tt> <tt class="py-op">=</tt> <tt id="link-69" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.min=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#min"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-69', 'min', 'link-69');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-70" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.query_range=Bio.SearchIO._model.hsp.HSP-class.html#query_range,Variable Bio.SearchIO._model.hsp.HSPFragment.query_range=Bio.SearchIO._model.hsp.HSPFragment-class.html#query_range"><a title="Bio.SearchIO._model.hsp.HSP.query_range
Bio.SearchIO._model.hsp.HSPFragment.query_range" class="py-name" href="#" onclick="return doclink('link-70', 'query_range', 'link-70');">query_range</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt class="py-name">qstep</tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">0</tt> <tt class="py-keyword">else</tt> <tt id="link-71" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.max=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#max"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-71', 'max', 'link-71');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-72" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_range
Bio.SearchIO._model.hsp.HSPFragment.query_range" class="py-name" href="#" onclick="return doclink('link-72', 'query_range', 'link-70');">query_range</a></tt><tt class="py-op">)</tt> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">    <tt class="py-name">hpos</tt> <tt class="py-op">=</tt> <tt id="link-73" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-73', 'min', 'link-69');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-74" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.hit_range=Bio.SearchIO._model.hsp.HSP-class.html#hit_range,Variable Bio.SearchIO._model.hsp.HSPFragment.hit_range=Bio.SearchIO._model.hsp.HSPFragment-class.html#hit_range"><a title="Bio.SearchIO._model.hsp.HSP.hit_range
Bio.SearchIO._model.hsp.HSPFragment.hit_range" class="py-name" href="#" onclick="return doclink('link-74', 'hit_range', 'link-74');">hit_range</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">if</tt> <tt class="py-name">qstep</tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">0</tt> <tt class="py-keyword">else</tt> <tt id="link-75" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-75', 'max', 'link-71');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-76" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.hit_range
Bio.SearchIO._model.hsp.HSPFragment.hit_range" class="py-name" href="#" onclick="return doclink('link-76', 'hit_range', 'link-74');">hit_range</a></tt><tt class="py-op">)</tt> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line">    <tt class="py-name">abs_pos</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line">    <tt class="py-comment"># split according to hit, then query</tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line">    <tt class="py-keyword">while</tt> <tt class="py-name">simil</tt><tt class="py-op">:</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line"> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line">            <tt class="py-name">shifts</tt> <tt class="py-op">=</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-77" class="py-name" targets="Method Bio.AlignIO.MafIO.MafIndex.search()=Bio.AlignIO.MafIO.MafIndex-class.html#search,Method Bio.KDTree.KDTree'.KDTree.search()=Bio.KDTree.KDTree%27.KDTree-class.html#search,Method Bio.PDB.NeighborSearch'.NeighborSearch.search()=Bio.PDB.NeighborSearch%27.NeighborSearch-class.html#search,Method Bio.Phylo.TreeConstruction.NNITreeSearcher.search()=Bio.Phylo.TreeConstruction.NNITreeSearcher-class.html#search,Method Bio.Phylo.TreeConstruction.TreeSearcher.search()=Bio.Phylo.TreeConstruction.TreeSearcher-class.html#search,Class Method Bio.Restriction.Restriction.AbstractCut.search()=Bio.Restriction.Restriction.AbstractCut-class.html#search,Method Bio.Restriction.Restriction.RestrictionBatch.search()=Bio.Restriction.Restriction.RestrictionBatch-class.html#search,Function Bio.SCOP.search()=Bio.SCOP-module.html#search,Function Bio.TogoWS.search()=Bio.TogoWS-module.html#search,Method Bio.motifs.Instances.search()=Bio.motifs.Instances-class.html#search,Method Bio.motifs.matrix.PositionSpecificScoringMatrix.search()=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#search"><a title="Bio.AlignIO.MafIO.MafIndex.search
Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-77', 'search', 'link-77');">search</a></tt><tt class="py-op">(</tt><tt id="link-78" class="py-name"><a title="Bio.SearchIO.ExonerateIO._base._RE_SHIFTS" class="py-name" href="#" onclick="return doclink('link-78', '_RE_SHIFTS', 'link-15');">_RE_SHIFTS</a></tt><tt class="py-op">,</tt> <tt class="py-name">simil</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">group</tt><tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">            <tt class="py-name">s_start</tt> <tt class="py-op">=</tt> <tt class="py-name">simil</tt><tt class="py-op">.</tt><tt id="link-79" class="py-name" targets="Method Bio.NeuralNetwork.Gene.Motif.MotifFinder.find()=Bio.NeuralNetwork.Gene.Motif.MotifFinder-class.html#find,Method Bio.NeuralNetwork.Gene.Schema.SchemaFinder.find()=Bio.NeuralNetwork.Gene.Schema.SchemaFinder-class.html#find,Method Bio.NeuralNetwork.Gene.Signature.SignatureFinder.find()=Bio.NeuralNetwork.Gene.Signature.SignatureFinder-class.html#find,Method Bio.Seq.Seq.find()=Bio.Seq.Seq-class.html#find,Function Bio.triefind.find()=Bio.triefind-module.html#find"><a title="Bio.NeuralNetwork.Gene.Motif.MotifFinder.find
Bio.NeuralNetwork.Gene.Schema.SchemaFinder.find
Bio.NeuralNetwork.Gene.Signature.SignatureFinder.find
Bio.Seq.Seq.find
Bio.triefind.find" class="py-name" href="#" onclick="return doclink('link-79', 'find', 'link-79');">find</a></tt><tt class="py-op">(</tt><tt class="py-name">shifts</tt><tt class="py-op">)</tt> </tt>
<a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line">            <tt class="py-name">s_stop</tt> <tt class="py-op">=</tt> <tt class="py-name">s_start</tt> <tt class="py-op">+</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">shifts</tt><tt class="py-op">)</tt> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">            <tt id="link-80" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-80', 'split', 'link-80');">split</a></tt> <tt class="py-op">=</tt> <tt class="py-name">frag</tt><tt class="py-op">[</tt><tt class="py-name">abs_pos</tt><tt class="py-op">:</tt><tt class="py-name">abs_pos</tt> <tt class="py-op">+</tt> <tt class="py-name">s_start</tt><tt class="py-op">]</tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">AttributeError</tt><tt class="py-op">:</tt>  <tt class="py-comment"># no '#' in simil, i.e. last frag</tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line">            <tt class="py-name">shifts</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line">            <tt class="py-name">s_start</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line">            <tt class="py-name">s_stop</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">simil</tt><tt class="py-op">)</tt> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line">            <tt id="link-81" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-81', 'split', 'link-80');">split</a></tt> <tt class="py-op">=</tt> <tt class="py-name">frag</tt><tt class="py-op">[</tt><tt class="py-name">abs_pos</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line"> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line">        <tt class="py-comment"># coordinates for the split strand</tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line">        <tt class="py-name">qstart</tt><tt class="py-op">,</tt> <tt class="py-name">hstart</tt> <tt class="py-op">=</tt> <tt class="py-name">qpos</tt><tt class="py-op">,</tt> <tt class="py-name">hpos</tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line">        <tt class="py-name">qpos</tt> <tt class="py-op">+=</tt> <tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-82" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-82', 'split', 'link-80');">split</a></tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt id="link-83" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.sum()=Bio.Nexus.Nexus.StepMatrix-class.html#sum,Method Bio.SubsMat.SeqMat.sum()=Bio.SubsMat.SeqMat-class.html#sum"><a title="Bio.Nexus.Nexus.StepMatrix.sum
Bio.SubsMat.SeqMat.sum" class="py-name" href="#" onclick="return doclink('link-83', 'sum', 'link-83');">sum</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-84" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-84', 'split', 'link-80');">split</a></tt><tt class="py-op">.</tt><tt id="link-85" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.query
Bio.SearchIO._model.hsp.HSP.query
Bio.SearchIO._model.hsp.HSPFragment.query
Bio.SearchIO._model.query" class="py-name" href="#" onclick="return doclink('link-85', 'query', 'link-41');">query</a></tt><tt class="py-op">.</tt><tt id="link-86" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-86', 'seq', 'link-23');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-87" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-87', 'count', 'link-66');">count</a></tt><tt class="py-op">(</tt><tt id="link-88" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-88', 'x', 'link-32');">x</a></tt><tt class="py-op">)</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt id="link-89" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-89', 'x', 'link-32');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'-'</tt><tt class="py-op">,</tt> <tt class="py-string">'&lt;'</tt><tt class="py-op">,</tt> <tt class="py-string">'&gt;'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt class="py-name">qstep</tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line">        <tt class="py-name">hpos</tt> <tt class="py-op">+=</tt> <tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-90" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-90', 'split', 'link-80');">split</a></tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt id="link-91" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.sum
Bio.SubsMat.SeqMat.sum" class="py-name" href="#" onclick="return doclink('link-91', 'sum', 'link-83');">sum</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-92" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-92', 'split', 'link-80');">split</a></tt><tt class="py-op">.</tt><tt id="link-93" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-93', 'hit', 'link-44');">hit</a></tt><tt class="py-op">.</tt><tt id="link-94" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-94', 'seq', 'link-23');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-95" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-95', 'count', 'link-66');">count</a></tt><tt class="py-op">(</tt><tt id="link-96" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-96', 'x', 'link-32');">x</a></tt><tt class="py-op">)</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt id="link-97" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-97', 'x', 'link-32');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'-'</tt><tt class="py-op">,</tt> <tt class="py-string">'&lt;'</tt><tt class="py-op">,</tt> <tt class="py-string">'&gt;'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt class="py-name">hstep</tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line"> </tt>
<a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line">        <tt id="link-98" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-98', 'split', 'link-80');">split</a></tt><tt class="py-op">.</tt><tt id="link-99" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.hit_start
Bio.SearchIO._model.hsp.HSPFragment.hit_start" class="py-name" href="#" onclick="return doclink('link-99', 'hit_start', 'link-18');">hit_start</a></tt> <tt class="py-op">=</tt> <tt id="link-100" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-100', 'min', 'link-69');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">hstart</tt><tt class="py-op">,</tt> <tt class="py-name">hpos</tt><tt class="py-op">)</tt> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line">        <tt id="link-101" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-101', 'split', 'link-80');">split</a></tt><tt class="py-op">.</tt><tt id="link-102" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_start
Bio.SearchIO._model.hsp.HSPFragment.query_start" class="py-name" href="#" onclick="return doclink('link-102', 'query_start', 'link-21');">query_start</a></tt> <tt class="py-op">=</tt> <tt id="link-103" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-103', 'min', 'link-69');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">qstart</tt><tt class="py-op">,</tt> <tt class="py-name">qpos</tt><tt class="py-op">)</tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line">        <tt id="link-104" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-104', 'split', 'link-80');">split</a></tt><tt class="py-op">.</tt><tt id="link-105" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.hit_end
Bio.SearchIO._model.hsp.HSPFragment.hit_end" class="py-name" href="#" onclick="return doclink('link-105', 'hit_end', 'link-58');">hit_end</a></tt> <tt class="py-op">=</tt> <tt id="link-106" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-106', 'max', 'link-71');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">hstart</tt><tt class="py-op">,</tt> <tt class="py-name">hpos</tt><tt class="py-op">)</tt> </tt>
<a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line">        <tt id="link-107" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-107', 'split', 'link-80');">split</a></tt><tt class="py-op">.</tt><tt id="link-108" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_end
Bio.SearchIO._model.hsp.HSPFragment.query_end" class="py-name" href="#" onclick="return doclink('link-108', 'query_end', 'link-55');">query_end</a></tt> <tt class="py-op">=</tt> <tt id="link-109" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-109', 'max', 'link-71');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">qstart</tt><tt class="py-op">,</tt> <tt class="py-name">qpos</tt><tt class="py-op">)</tt> </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line"> </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line">        <tt class="py-comment"># account for frameshift length</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line">        <tt class="py-name">abs_slice</tt> <tt class="py-op">=</tt> <tt class="py-name">slice</tt><tt class="py-op">(</tt><tt class="py-name">abs_pos</tt> <tt class="py-op">+</tt> <tt class="py-name">s_start</tt><tt class="py-op">,</tt> <tt class="py-name">abs_pos</tt> <tt class="py-op">+</tt> <tt class="py-name">s_stop</tt><tt class="py-op">)</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-110" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.aln_annotation" class="py-name" href="#" onclick="return doclink('link-110', 'aln_annotation', 'link-59');">aln_annotation</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> </tt>
<a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line">            <tt class="py-name">seqs</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">[</tt><tt class="py-name">abs_slice</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-111" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.query
Bio.SearchIO._model.hsp.HSP.query
Bio.SearchIO._model.hsp.HSPFragment.query
Bio.SearchIO._model.query" class="py-name" href="#" onclick="return doclink('link-111', 'query', 'link-41');">query</a></tt><tt class="py-op">.</tt><tt id="link-112" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-112', 'seq', 'link-23');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line">                    <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">[</tt><tt class="py-name">abs_slice</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-113" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-113', 'hit', 'link-44');">hit</a></tt><tt class="py-op">.</tt><tt id="link-114" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-114', 'seq', 'link-23');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-115" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.aln_annotation" class="py-name" href="#" onclick="return doclink('link-115', 'aln_annotation', 'link-59');">aln_annotation</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">3</tt><tt class="py-op">:</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line">            <tt class="py-name">seqs</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">[</tt><tt class="py-name">abs_slice</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-116" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.aln_annotation" class="py-name" href="#" onclick="return doclink('link-116', 'aln_annotation', 'link-59');">aln_annotation</a></tt><tt class="py-op">[</tt><tt class="py-string">'query_annotation'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line">                    <tt class="py-name">frag</tt><tt class="py-op">[</tt><tt class="py-name">abs_slice</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-117" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.aln_annotation" class="py-name" href="#" onclick="return doclink('link-117', 'aln_annotation', 'link-59');">aln_annotation</a></tt><tt class="py-op">[</tt><tt class="py-string">'hit_annotation'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt><tt class="py-op">)</tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">'#'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">seqs</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line">            <tt class="py-name">qpos</tt> <tt class="py-op">+=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">shifts</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt class="py-name">qstep</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-string">'#'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">seqs</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line">            <tt class="py-name">hpos</tt> <tt class="py-op">+=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">shifts</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt class="py-name">hstep</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line"> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line">        <tt class="py-comment"># set frame</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line">        <tt id="link-118" class="py-name" targets="Function Bio.SearchIO.ExonerateIO._base._set_frame()=Bio.SearchIO.ExonerateIO._base-module.html#_set_frame"><a title="Bio.SearchIO.ExonerateIO._base._set_frame" class="py-name" href="#" onclick="return doclink('link-118', '_set_frame', 'link-118');">_set_frame</a></tt><tt class="py-op">(</tt><tt id="link-119" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-119', 'split', 'link-80');">split</a></tt><tt class="py-op">)</tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line">        <tt class="py-name">split_frags</tt><tt class="py-op">.</tt><tt id="link-120" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-120', 'append', 'link-120');">append</a></tt><tt class="py-op">(</tt><tt id="link-121" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-121', 'split', 'link-80');">split</a></tt><tt class="py-op">)</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line">        <tt class="py-comment"># set similarity string and absolute position for the next loop</tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line">        <tt class="py-name">simil</tt> <tt class="py-op">=</tt> <tt class="py-name">simil</tt><tt class="py-op">[</tt><tt class="py-name">s_stop</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">        <tt class="py-name">abs_pos</tt> <tt class="py-op">+=</tt> <tt class="py-name">s_stop</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line"> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">split_frags</tt> </tt>
</div><a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line"> </tt>
<a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line"> </tt>
<a name="_create_hsp"></a><div id="_create_hsp-def"><a name="L238"></a><tt class="py-lineno">238</tt> <a class="py-toggle" href="#" id="_create_hsp-toggle" onclick="return toggle('_create_hsp');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO._base-module.html#_create_hsp">_create_hsp</a><tt class="py-op">(</tt><tt class="py-param">hid</tt><tt class="py-op">,</tt> <tt class="py-param">qid</tt><tt class="py-op">,</tt> <tt class="py-param">hspd</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_create_hsp-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_create_hsp-expanded"><a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line">    <tt class="py-docstring">"""Return a list of HSP objects from the given parsed HSP values (PRIVATE)."""</tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line">    <tt class="py-name">frags</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line">    <tt class="py-comment"># we are iterating over query_ranges, but hit_ranges works just as well</tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">idx</tt><tt class="py-op">,</tt> <tt class="py-name">qcoords</tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt class="py-name">hspd</tt><tt class="py-op">[</tt><tt class="py-string">'query_ranges'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line">        <tt class="py-comment"># get sequences, create object</tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line">        <tt class="py-name">hseqlist</tt> <tt class="py-op">=</tt> <tt class="py-name">hspd</tt><tt class="py-op">.</tt><tt id="link-122" class="py-name" targets="Method Bio.Crystal.Crystal.get()=Bio.Crystal.Crystal-class.html#get,Method Bio.Data.CodonTable.AmbiguousForwardTable.get()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#get,Method Bio.File._IndexedSeqFileDict.get()=Bio.File._IndexedSeqFileDict-class.html#get,Method Bio.File._IndexedSeqFileProxy.get()=Bio.File._IndexedSeqFileProxy-class.html#get,Method Bio.File._SQLiteManySeqFilesDict.get()=Bio.File._SQLiteManySeqFilesDict-class.html#get,Method Bio.Restriction.Restriction.RestrictionBatch.get()=Bio.Restriction.Restriction.RestrictionBatch-class.html#get,Method Bio.SearchIO._index.SearchIndexer.get()=Bio.SearchIO._index.SearchIndexer-class.html#get,Method Bio.SeqIO._index.SeqFileRandomAccess.get()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#get,Method Bio.SeqIO._index.SffRandomAccess.get()=Bio.SeqIO._index.SffRandomAccess-class.html#get,Method Bio.SeqIO._index.SffTrimedRandomAccess.get()=Bio.SeqIO._index.SffTrimedRandomAccess-class.html#get,Method Bio.SeqIO._index.UniprotRandomAccess.get()=Bio.SeqIO._index.UniprotRandomAccess-class.html#get"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-122', 'get', 'link-122');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'hit'</tt><tt class="py-op">)</tt> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line">        <tt class="py-name">hseq</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> <tt class="py-keyword">if</tt> <tt class="py-name">hseqlist</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt> <tt class="py-keyword">else</tt> <tt class="py-name">hseqlist</tt><tt class="py-op">[</tt><tt class="py-name">idx</tt><tt class="py-op">]</tt> </tt>
<a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line">        <tt class="py-name">qseqlist</tt> <tt class="py-op">=</tt> <tt class="py-name">hspd</tt><tt class="py-op">.</tt><tt id="link-123" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-123', 'get', 'link-122');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'query'</tt><tt class="py-op">)</tt> </tt>
<a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line">        <tt class="py-name">qseq</tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> <tt class="py-keyword">if</tt> <tt class="py-name">qseqlist</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt> <tt class="py-keyword">else</tt> <tt class="py-name">qseqlist</tt><tt class="py-op">[</tt><tt class="py-name">idx</tt><tt class="py-op">]</tt> </tt>
<a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line">        <tt class="py-name">frag</tt> <tt class="py-op">=</tt> <tt id="link-124" class="py-name"><a title="Bio.SearchIO._model.hsp.HSPFragment" class="py-name" href="#" onclick="return doclink('link-124', 'HSPFragment', 'link-10');">HSPFragment</a></tt><tt class="py-op">(</tt><tt class="py-name">hid</tt><tt class="py-op">,</tt> <tt class="py-name">qid</tt><tt class="py-op">,</tt> <tt id="link-125" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-125', 'hit', 'link-44');">hit</a></tt><tt class="py-op">=</tt><tt class="py-name">hseq</tt><tt class="py-op">,</tt> <tt id="link-126" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.query
Bio.SearchIO._model.hsp.HSP.query
Bio.SearchIO._model.hsp.HSPFragment.query
Bio.SearchIO._model.query" class="py-name" href="#" onclick="return doclink('link-126', 'query', 'link-41');">query</a></tt><tt class="py-op">=</tt><tt class="py-name">qseq</tt><tt class="py-op">)</tt> </tt>
<a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line">        <tt class="py-comment"># coordinates</tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line">        <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-127" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_start
Bio.SearchIO._model.hsp.HSPFragment.query_start" class="py-name" href="#" onclick="return doclink('link-127', 'query_start', 'link-21');">query_start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">qcoords</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line">        <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-128" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_end
Bio.SearchIO._model.hsp.HSPFragment.query_end" class="py-name" href="#" onclick="return doclink('link-128', 'query_end', 'link-55');">query_end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">qcoords</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line">        <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-129" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.hit_start
Bio.SearchIO._model.hsp.HSPFragment.hit_start" class="py-name" href="#" onclick="return doclink('link-129', 'hit_start', 'link-18');">hit_start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">hspd</tt><tt class="py-op">[</tt><tt class="py-string">'hit_ranges'</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">idx</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line">        <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-130" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.hit_end
Bio.SearchIO._model.hsp.HSPFragment.hit_end" class="py-name" href="#" onclick="return doclink('link-130', 'hit_end', 'link-58');">hit_end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">hspd</tt><tt class="py-op">[</tt><tt class="py-string">'hit_ranges'</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-name">idx</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line">        <tt class="py-comment"># alignment annotation</tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line">            <tt class="py-name">aln_annot</tt> <tt class="py-op">=</tt> <tt class="py-name">hspd</tt><tt class="py-op">.</tt><tt id="link-131" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-131', 'get', 'link-122');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'aln_annotation'</tt><tt class="py-op">,</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt id="link-132" class="py-name" targets="Variable Bio.Entrez.Parser.Consumer.value=Bio.Entrez.Parser.Consumer-class.html#value,Variable Bio.Entrez.Parser.DictionaryConsumer.value=Bio.Entrez.Parser.DictionaryConsumer-class.html#value,Variable Bio.Entrez.Parser.ErrorConsumer.value=Bio.Entrez.Parser.ErrorConsumer-class.html#value,Variable Bio.Entrez.Parser.IntegerConsumer.value=Bio.Entrez.Parser.IntegerConsumer-class.html#value,Variable Bio.Entrez.Parser.ListConsumer.value=Bio.Entrez.Parser.ListConsumer-class.html#value,Variable Bio.Entrez.Parser.StringConsumer.value=Bio.Entrez.Parser.StringConsumer-class.html#value,Method Bio.Phylo.PhyloXMLIO.Writer.value()=Bio.Phylo.PhyloXMLIO.Writer-class.html#value"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-132', 'value', 'link-132');">value</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">aln_annot</tt><tt class="py-op">.</tt><tt id="link-133" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-133', 'items', 'link-60');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line">                <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-134" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.aln_annotation" class="py-name" href="#" onclick="return doclink('link-134', 'aln_annotation', 'link-59');">aln_annotation</a></tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-135" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-135', 'value', 'link-132');">value</a></tt><tt class="py-op">[</tt><tt class="py-name">idx</tt><tt class="py-op">]</tt> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">IndexError</tt><tt class="py-op">:</tt> </tt>
<a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line">        <tt class="py-comment"># strands</tt> </tt>
<a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line">        <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-136" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_strand
Bio.SearchIO._model.hsp.HSPFragment.query_strand" class="py-name" href="#" onclick="return doclink('link-136', 'query_strand', 'link-22');">query_strand</a></tt> <tt class="py-op">=</tt> <tt class="py-name">hspd</tt><tt class="py-op">[</tt><tt class="py-string">'query_strand'</tt><tt class="py-op">]</tt> </tt>
<a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line">        <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-137" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.hit_strand
Bio.SearchIO._model.hsp.HSPFragment.hit_strand" class="py-name" href="#" onclick="return doclink('link-137', 'hit_strand', 'link-19');">hit_strand</a></tt> <tt class="py-op">=</tt> <tt class="py-name">hspd</tt><tt class="py-op">[</tt><tt class="py-string">'hit_strand'</tt><tt class="py-op">]</tt> </tt>
<a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line">        <tt class="py-comment"># and append the hsp object to the list</tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-138" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.aln_annotation" class="py-name" href="#" onclick="return doclink('link-138', 'aln_annotation', 'link-59');">aln_annotation</a></tt><tt class="py-op">.</tt><tt id="link-139" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-139', 'get', 'link-122');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'similarity'</tt><tt class="py-op">)</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-string">'#'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-140" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.aln_annotation" class="py-name" href="#" onclick="return doclink('link-140', 'aln_annotation', 'link-59');">aln_annotation</a></tt><tt class="py-op">[</tt><tt class="py-string">'similarity'</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line">                <tt class="py-name">frags</tt><tt class="py-op">.</tt><tt id="link-141" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.extend()=Bio.Align.MultipleSeqAlignment-class.html#extend,Method Bio.SCOP.Raf.SeqMap.extend()=Bio.SCOP.Raf.SeqMap-class.html#extend,Method Bio.Seq.MutableSeq.extend()=Bio.Seq.MutableSeq-class.html#extend"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-141', 'extend', 'link-141');">extend</a></tt><tt class="py-op">(</tt><tt id="link-142" class="py-name" targets="Function Bio.SearchIO.ExonerateIO._base._split_fragment()=Bio.SearchIO.ExonerateIO._base-module.html#_split_fragment"><a title="Bio.SearchIO.ExonerateIO._base._split_fragment" class="py-name" href="#" onclick="return doclink('link-142', '_split_fragment', 'link-142');">_split_fragment</a></tt><tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line">                <tt class="py-keyword">continue</tt> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line">        <tt class="py-comment"># try to set frame if there are translation in the alignment</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-143" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.aln_annotation" class="py-name" href="#" onclick="return doclink('link-143', 'aln_annotation', 'link-59');">aln_annotation</a></tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1</tt> <tt class="py-keyword">or</tt> \ </tt>
<a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line">           <tt class="py-name">frag</tt><tt class="py-op">.</tt><tt id="link-144" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.query_strand
Bio.SearchIO._model.hsp.HSPFragment.query_strand" class="py-name" href="#" onclick="return doclink('link-144', 'query_strand', 'link-22');">query_strand</a></tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt> <tt class="py-keyword">or</tt> \ </tt>
<a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line">           <tt class="py-op">(</tt><tt class="py-string">'vulgar_comp'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">hspd</tt> <tt class="py-keyword">and</tt> <tt class="py-name">re</tt><tt class="py-op">.</tt><tt id="link-145" class="py-name"><a title="Bio.AlignIO.MafIO.MafIndex.search
Bio.KDTree.KDTree'.KDTree.search
Bio.PDB.NeighborSearch'.NeighborSearch.search
Bio.Phylo.TreeConstruction.NNITreeSearcher.search
Bio.Phylo.TreeConstruction.TreeSearcher.search
Bio.Restriction.Restriction.AbstractCut.search
Bio.Restriction.Restriction.RestrictionBatch.search
Bio.SCOP.search
Bio.TogoWS.search
Bio.motifs.Instances.search
Bio.motifs.matrix.PositionSpecificScoringMatrix.search" class="py-name" href="#" onclick="return doclink('link-145', 'search', 'link-77');">search</a></tt><tt class="py-op">(</tt><tt id="link-146" class="py-name"><a title="Bio.SearchIO.ExonerateIO._base._RE_TRANS" class="py-name" href="#" onclick="return doclink('link-146', '_RE_TRANS', 'link-16');">_RE_TRANS</a></tt><tt class="py-op">,</tt> <tt class="py-name">hspd</tt><tt class="py-op">[</tt><tt class="py-string">'vulgar_comp'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line">            <tt id="link-147" class="py-name"><a title="Bio.SearchIO.ExonerateIO._base._set_frame" class="py-name" href="#" onclick="return doclink('link-147', '_set_frame', 'link-118');">_set_frame</a></tt><tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">)</tt> </tt>
<a name="L274"></a><tt class="py-lineno">274</tt>  <tt class="py-line"> </tt>
<a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line">        <tt class="py-name">frags</tt><tt class="py-op">.</tt><tt id="link-148" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-148', 'append', 'link-120');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">frag</tt><tt class="py-op">)</tt> </tt>
<a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line"> </tt>
<a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line">    <tt class="py-comment"># if the query is protein, we need to change the hit and query sequences</tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line">    <tt class="py-comment"># from three-letter amino acid codes to one letter, and adjust their</tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line">    <tt class="py-comment"># coordinates accordingly</tt> </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">frags</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-149" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.aln_annotation" class="py-name" href="#" onclick="return doclink('link-149', 'aln_annotation', 'link-59');">aln_annotation</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt>  <tt class="py-comment"># 2 annotations == protein query</tt> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line">        <tt class="py-name">frags</tt> <tt class="py-op">=</tt> <tt id="link-150" class="py-name" targets="Function Bio.SearchIO.ExonerateIO._base._adjust_aa_seq()=Bio.SearchIO.ExonerateIO._base-module.html#_adjust_aa_seq"><a title="Bio.SearchIO.ExonerateIO._base._adjust_aa_seq" class="py-name" href="#" onclick="return doclink('link-150', '_adjust_aa_seq', 'link-150');">_adjust_aa_seq</a></tt><tt class="py-op">(</tt><tt class="py-name">frags</tt><tt class="py-op">)</tt> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line"> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line">    <tt id="link-151" class="py-name" targets="Module Bio.SearchIO._model.hsp=Bio.SearchIO._model.hsp-module.html"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-151', 'hsp', 'link-151');">hsp</a></tt> <tt class="py-op">=</tt> <tt id="link-152" class="py-name"><a title="Bio.Blast.Record.HSP
Bio.SearchIO._model.hsp.HSP" class="py-name" href="#" onclick="return doclink('link-152', 'HSP', 'link-9');">HSP</a></tt><tt class="py-op">(</tt><tt class="py-name">frags</tt><tt class="py-op">)</tt> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line">    <tt class="py-comment"># set hsp-specific attributes</tt> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">attr</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'score'</tt><tt class="py-op">,</tt> <tt class="py-string">'hit_split_codons'</tt><tt class="py-op">,</tt> <tt class="py-string">'query_split_codons'</tt><tt class="py-op">,</tt> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line">            <tt class="py-string">'model'</tt><tt class="py-op">,</tt> <tt class="py-string">'vulgar_comp'</tt><tt class="py-op">,</tt> <tt class="py-string">'cigar_comp'</tt><tt class="py-op">,</tt> <tt class="py-string">'alphabet'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">attr</tt> <tt class="py-keyword">in</tt> <tt class="py-name">hspd</tt><tt class="py-op">:</tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">            <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt id="link-153" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-153', 'hsp', 'link-151');">hsp</a></tt><tt class="py-op">,</tt> <tt class="py-name">attr</tt><tt class="py-op">,</tt> <tt class="py-name">hspd</tt><tt class="py-op">[</tt><tt class="py-name">attr</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line"> </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-154" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-154', 'hsp', 'link-151');">hsp</a></tt> </tt>
</div><a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line"> </tt>
<a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line"> </tt>
<a name="_parse_hit_or_query_line"></a><div id="_parse_hit_or_query_line-def"><a name="L293"></a><tt class="py-lineno">293</tt> <a class="py-toggle" href="#" id="_parse_hit_or_query_line-toggle" onclick="return toggle('_parse_hit_or_query_line');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO._base-module.html#_parse_hit_or_query_line">_parse_hit_or_query_line</a><tt class="py-op">(</tt><tt class="py-param">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_parse_hit_or_query_line-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_parse_hit_or_query_line-expanded"><a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line">    <tt class="py-docstring">"""Parse the 'Query:' line of exonerate alignment outputs (PRIVATE)."""</tt> </tt>
<a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line">        <tt class="py-name">mark</tt><tt class="py-op">,</tt> <tt id="link-155" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.id=Bio.KEGG.KGML.KGML_pathway.Component-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Entry.id=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.id=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#id,Variable Bio.PDB.Entity.Entity.id=Bio.PDB.Entity.Entity-class.html#id,Method Bio.Phylo.PhyloXMLIO.Parser.id()=Bio.Phylo.PhyloXMLIO.Parser-class.html#id,Method Bio.Phylo.PhyloXMLIO.Writer.id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#id,Variable Bio.SearchIO._model.hit.Hit.id=Bio.SearchIO._model.hit.Hit-class.html#id,Variable Bio.SearchIO._model.query.QueryResult.id=Bio.SearchIO._model.query.QueryResult-class.html#id"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-155', 'id', 'link-155');">id</a></tt><tt class="py-op">,</tt> <tt id="link-156" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.desc()=Bio.Phylo.PhyloXMLIO.Writer-class.html#desc"><a title="Bio.Phylo.PhyloXMLIO.Writer.desc" class="py-name" href="#" onclick="return doclink('link-156', 'desc', 'link-156');">desc</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-157" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-157', 'split', 'link-80');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">' '</tt><tt class="py-op">,</tt> <tt class="py-number">2</tt><tt class="py-op">)</tt> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt>  <tt class="py-comment"># no desc</tt> </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line">        <tt class="py-name">mark</tt><tt class="py-op">,</tt> <tt id="link-158" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-158', 'id', 'link-155');">id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-159" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-159', 'split', 'link-80');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">' '</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line">        <tt id="link-160" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.desc" class="py-name" href="#" onclick="return doclink('link-160', 'desc', 'link-156');">desc</a></tt> <tt class="py-op">=</tt> <tt class="py-string">''</tt> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line"> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-161" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-161', 'id', 'link-155');">id</a></tt><tt class="py-op">,</tt> <tt id="link-162" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.desc" class="py-name" href="#" onclick="return doclink('link-162', 'desc', 'link-156');">desc</a></tt> </tt>
</div><a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line"> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line"> </tt>
<a name="_BaseExonerateParser"></a><div id="_BaseExonerateParser-def"><a name="L304"></a><tt class="py-lineno">304</tt> <a class="py-toggle" href="#" id="_BaseExonerateParser-toggle" onclick="return toggle('_BaseExonerateParser');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO._base._BaseExonerateParser-class.html">_BaseExonerateParser</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BaseExonerateParser-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_BaseExonerateParser-expanded"><a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line">    <tt class="py-docstring">"""Abstract iterator for exonerate format."""</tt> </tt>
<a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line"> </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line">    <tt id="link-163" class="py-name" targets="Variable Bio.SearchIO.ExonerateIO._base._BaseExonerateParser._ALN_MARK=Bio.SearchIO.ExonerateIO._base._BaseExonerateParser-class.html#_ALN_MARK,Variable Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarParser._ALN_MARK=Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarParser-class.html#_ALN_MARK,Variable Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextParser._ALN_MARK=Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextParser-class.html#_ALN_MARK,Variable Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarParser._ALN_MARK=Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarParser-class.html#_ALN_MARK"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateParser._ALN_MARK
Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarParser._ALN_MARK
Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextParser._ALN_MARK
Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarParser._ALN_MARK" class="py-name" href="#" onclick="return doclink('link-163', '_ALN_MARK', 'link-163');">_ALN_MARK</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line"> </tt>
<a name="_BaseExonerateParser.__init__"></a><div id="_BaseExonerateParser.__init__-def"><a name="L309"></a><tt class="py-lineno">309</tt> <a class="py-toggle" href="#" id="_BaseExonerateParser.__init__-toggle" onclick="return toggle('_BaseExonerateParser.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO._base._BaseExonerateParser-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BaseExonerateParser.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="_BaseExonerateParser.__init__-expanded"><a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt> </tt>
<a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">has_c4_alignment</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
</div><a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line"> </tt>
<a name="_BaseExonerateParser.__iter__"></a><div id="_BaseExonerateParser.__iter__-def"><a name="L313"></a><tt class="py-lineno">313</tt> <a class="py-toggle" href="#" id="_BaseExonerateParser.__iter__-toggle" onclick="return toggle('_BaseExonerateParser.__iter__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO._base._BaseExonerateParser-class.html#__iter__">__iter__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BaseExonerateParser.__iter__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="_BaseExonerateParser.__iter__-expanded"><a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line">        <tt class="py-comment"># read line until the first alignment block or cigar/vulgar lines</tt> </tt>
<a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-164" class="py-name" targets="Method Bio.File.UndoHandle.readline()=Bio.File.UndoHandle-class.html#readline,Method Bio.bgzf.BgzfReader.readline()=Bio.bgzf.BgzfReader-class.html#readline"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-164', 'readline', 'link-164');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line">            <tt class="py-comment"># flag for human-readable alignment block</tt> </tt>
<a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-165" class="py-name" targets="Method Bio.Seq.Seq.startswith()=Bio.Seq.Seq-class.html#startswith"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-165', 'startswith', 'link-165');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'C4 Alignment:'</tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> \ </tt>
<a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">has_c4_alignment</tt><tt class="py-op">:</tt> </tt>
<a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">has_c4_alignment</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-166" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-166', 'startswith', 'link-165');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'C4 Alignment:'</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> \ </tt>
<a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-167" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-167', 'startswith', 'link-165');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'vulgar:'</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> \ </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-168" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-168', 'startswith', 'link-165');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'cigar:'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L325"></a><tt class="py-lineno">325</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-keyword">or</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-169" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-169', 'startswith', 'link-165');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'-- completed '</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line"> </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">qresult</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-170" class="py-name" targets="Method Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._parse_qresult()=Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#_parse_qresult,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser._parse_qresult()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser-class.html#_parse_qresult,Method Bio.SearchIO.BlatIO.BlatPslParser._parse_qresult()=Bio.SearchIO.BlatIO.BlatPslParser-class.html#_parse_qresult,Method Bio.SearchIO.ExonerateIO._base._BaseExonerateParser._parse_qresult()=Bio.SearchIO.ExonerateIO._base._BaseExonerateParser-class.html#_parse_qresult,Method Bio.SearchIO.FastaIO.FastaM10Parser._parse_qresult()=Bio.SearchIO.FastaIO.FastaM10Parser-class.html#_parse_qresult,Method Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabParser._parse_qresult()=Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabParser-class.html#_parse_qresult,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser._parse_qresult()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser-class.html#_parse_qresult,Method Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser._parse_qresult()=Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html#_parse_qresult,Method Bio.SearchIO.InterproscanIO.interproscan_xml.InterproscanXmlParser._parse_qresult()=Bio.SearchIO.InterproscanIO.interproscan_xml.InterproscanXmlParser-class.html#_parse_qresult"><a title="Bio.SearchIO.BlastIO.blast_tab.BlastTabParser._parse_qresult
Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser._parse_qresult
Bio.SearchIO.BlatIO.BlatPslParser._parse_qresult
Bio.SearchIO.ExonerateIO._base._BaseExonerateParser._parse_qresult
Bio.SearchIO.FastaIO.FastaM10Parser._parse_qresult
Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabParser._parse_qresult
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser._parse_qresult
Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser._parse_qresult
Bio.SearchIO.InterproscanIO.interproscan_xml.InterproscanXmlParser._parse_qresult" class="py-name" href="#" onclick="return doclink('link-170', '_parse_qresult', 'link-170');">_parse_qresult</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line">            <tt class="py-name">qresult</tt><tt class="py-op">.</tt><tt class="py-name">program</tt> <tt class="py-op">=</tt> <tt class="py-string">'exonerate'</tt> </tt>
<a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line">            <tt class="py-comment"># HACK: so that all descriptions are set</tt> </tt>
<a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line">            <tt class="py-name">qresult</tt><tt class="py-op">.</tt><tt id="link-171" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DescriptionConsumer.description()=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#description,Method Bio.Phylo.PhyloXMLIO.Writer.description()=Bio.Phylo.PhyloXMLIO.Writer-class.html#description,Variable Bio.SearchIO._model.hit.Hit.description=Bio.SearchIO._model.hit.Hit-class.html#description,Variable Bio.SearchIO._model.query.QueryResult.description=Bio.SearchIO._model.query.QueryResult-class.html#description"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-171', 'description', 'link-171');">description</a></tt> <tt class="py-op">=</tt> <tt class="py-name">qresult</tt><tt class="py-op">.</tt><tt id="link-172" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-172', 'description', 'link-171');">description</a></tt> </tt>
<a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt id="link-173" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-173', 'hit', 'link-44');">hit</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">qresult</tt><tt class="py-op">:</tt> </tt>
<a name="L333"></a><tt class="py-lineno">333</tt>  <tt class="py-line">                <tt id="link-174" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-174', 'hit', 'link-44');">hit</a></tt><tt class="py-op">.</tt><tt id="link-175" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-175', 'description', 'link-171');">description</a></tt> <tt class="py-op">=</tt> <tt id="link-176" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-176', 'hit', 'link-44');">hit</a></tt><tt class="py-op">.</tt><tt id="link-177" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-177', 'description', 'link-171');">description</a></tt> </tt>
<a name="L334"></a><tt class="py-lineno">334</tt>  <tt class="py-line">            <tt class="py-keyword">yield</tt> <tt class="py-name">qresult</tt> </tt>
</div><a name="L335"></a><tt class="py-lineno">335</tt>  <tt class="py-line"> </tt>
<a name="_BaseExonerateParser.read_until"></a><div id="_BaseExonerateParser.read_until-def"><a name="L336"></a><tt class="py-lineno">336</tt> <a class="py-toggle" href="#" id="_BaseExonerateParser.read_until-toggle" onclick="return toggle('_BaseExonerateParser.read_until');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO._base._BaseExonerateParser-class.html#read_until">read_until</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">bool_func</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BaseExonerateParser.read_until-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="_BaseExonerateParser.read_until-expanded"><a name="L337"></a><tt class="py-lineno">337</tt>  <tt class="py-line">        <tt class="py-docstring">"""Read the file handle until the given bool function returns True."""</tt> </tt>
<a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L339"></a><tt class="py-lineno">339</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-keyword">or</tt> <tt class="py-name">bool_func</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> </tt>
<a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-178" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-178', 'readline', 'link-164');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L343"></a><tt class="py-lineno">343</tt>  <tt class="py-line"> </tt>
<a name="_BaseExonerateParser.parse_alignment_block"></a><div id="_BaseExonerateParser.parse_alignment_block-def"><a name="L344"></a><tt class="py-lineno">344</tt> <a class="py-toggle" href="#" id="_BaseExonerateParser.parse_alignment_block-toggle" onclick="return toggle('_BaseExonerateParser.parse_alignment_block');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO._base._BaseExonerateParser-class.html#parse_alignment_block">parse_alignment_block</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">header</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BaseExonerateParser.parse_alignment_block-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="_BaseExonerateParser.parse_alignment_block-expanded"><a name="L345"></a><tt class="py-lineno">345</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">NotImplementedError</tt><tt class="py-op">(</tt><tt class="py-string">"Subclass must implement this"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L346"></a><tt class="py-lineno">346</tt>  <tt class="py-line"> </tt>
<a name="_BaseExonerateParser._parse_alignment_header"></a><div id="_BaseExonerateParser._parse_alignment_header-def"><a name="L347"></a><tt class="py-lineno">347</tt> <a class="py-toggle" href="#" id="_BaseExonerateParser._parse_alignment_header-toggle" onclick="return toggle('_BaseExonerateParser._parse_alignment_header');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO._base._BaseExonerateParser-class.html#_parse_alignment_header">_parse_alignment_header</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BaseExonerateParser._parse_alignment_header-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="_BaseExonerateParser._parse_alignment_header-expanded"><a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line">        <tt class="py-comment"># read all header lines and store them</tt> </tt>
<a name="L349"></a><tt class="py-lineno">349</tt>  <tt class="py-line">        <tt class="py-name">aln_header</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line">        <tt class="py-comment"># header is everything before the first empty line</tt> </tt>
<a name="L351"></a><tt class="py-lineno">351</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-179" class="py-name" targets="Method Bio.Seq.Seq.strip()=Bio.Seq.Seq-class.html#strip"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-179', 'strip', 'link-179');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L352"></a><tt class="py-lineno">352</tt>  <tt class="py-line">            <tt class="py-name">aln_header</tt><tt class="py-op">.</tt><tt id="link-180" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-180', 'append', 'link-120');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-181" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-181', 'strip', 'link-179');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-182" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-182', 'readline', 'link-164');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L354"></a><tt class="py-lineno">354</tt>  <tt class="py-line">        <tt class="py-comment"># then parse them</tt> </tt>
<a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line">        <tt class="py-name">qresult</tt><tt class="py-op">,</tt> <tt id="link-183" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-183', 'hit', 'link-44');">hit</a></tt><tt class="py-op">,</tt> <tt id="link-184" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-184', 'hsp', 'link-151');">hsp</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt><tt class="py-op">,</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt><tt class="py-op">,</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">aln_header</tt><tt class="py-op">:</tt> </tt>
<a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line">            <tt class="py-comment"># query line</tt> </tt>
<a name="L358"></a><tt class="py-lineno">358</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-185" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-185', 'startswith', 'link-165');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Query:'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line">                <tt class="py-name">qresult</tt><tt class="py-op">[</tt><tt class="py-string">'id'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">qresult</tt><tt class="py-op">[</tt><tt class="py-string">'description'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> \ </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line">                        <tt id="link-186" class="py-name" targets="Function Bio.SearchIO.ExonerateIO._base._parse_hit_or_query_line()=Bio.SearchIO.ExonerateIO._base-module.html#_parse_hit_or_query_line"><a title="Bio.SearchIO.ExonerateIO._base._parse_hit_or_query_line" class="py-name" href="#" onclick="return doclink('link-186', '_parse_hit_or_query_line', 'link-186');">_parse_hit_or_query_line</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line">            <tt class="py-comment"># target line</tt> </tt>
<a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-187" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-187', 'startswith', 'link-165');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Target:'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L363"></a><tt class="py-lineno">363</tt>  <tt class="py-line">                <tt id="link-188" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-188', 'hit', 'link-44');">hit</a></tt><tt class="py-op">[</tt><tt class="py-string">'id'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt id="link-189" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-189', 'hit', 'link-44');">hit</a></tt><tt class="py-op">[</tt><tt class="py-string">'description'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> \ </tt>
<a name="L364"></a><tt class="py-lineno">364</tt>  <tt class="py-line">                        <tt id="link-190" class="py-name"><a title="Bio.SearchIO.ExonerateIO._base._parse_hit_or_query_line" class="py-name" href="#" onclick="return doclink('link-190', '_parse_hit_or_query_line', 'link-186');">_parse_hit_or_query_line</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> </tt>
<a name="L365"></a><tt class="py-lineno">365</tt>  <tt class="py-line">            <tt class="py-comment"># model line</tt> </tt>
<a name="L366"></a><tt class="py-lineno">366</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-191" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-191', 'startswith', 'link-165');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Model:'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L367"></a><tt class="py-lineno">367</tt>  <tt class="py-line">                <tt class="py-name">qresult</tt><tt class="py-op">[</tt><tt class="py-string">'model'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-192" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-192', 'split', 'link-80');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">' '</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L368"></a><tt class="py-lineno">368</tt>  <tt class="py-line">            <tt class="py-comment"># score line</tt> </tt>
<a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-193" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-193', 'startswith', 'link-165');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Raw score:'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line">                <tt id="link-194" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-194', 'hsp', 'link-151');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'score'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-195" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-195', 'split', 'link-80');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">' '</tt><tt class="py-op">,</tt> <tt class="py-number">2</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> </tt>
<a name="L371"></a><tt class="py-lineno">371</tt>  <tt class="py-line">            <tt class="py-comment"># query range line</tt> </tt>
<a name="L372"></a><tt class="py-lineno">372</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-196" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-196', 'startswith', 'link-165');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Query range:'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L373"></a><tt class="py-lineno">373</tt>  <tt class="py-line">                <tt class="py-comment"># line is always 'Query range: \d+ -&gt; \d+', so we can pluck</tt> </tt>
<a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line">                <tt class="py-comment"># the numbers directly</tt> </tt>
<a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line">                <tt id="link-197" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-197', 'hsp', 'link-151');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'query_start'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt id="link-198" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-198', 'hsp', 'link-151');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'query_end'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-199" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-199', 'split', 'link-80');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">' '</tt><tt class="py-op">,</tt> <tt class="py-number">4</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">:</tt><tt class="py-number">5</tt><tt class="py-op">:</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> </tt>
<a name="L376"></a><tt class="py-lineno">376</tt>  <tt class="py-line">            <tt class="py-comment"># hit range line</tt> </tt>
<a name="L377"></a><tt class="py-lineno">377</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-200" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-200', 'startswith', 'link-165');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Target range:'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L378"></a><tt class="py-lineno">378</tt>  <tt class="py-line">                <tt class="py-comment"># same logic with query range</tt> </tt>
<a name="L379"></a><tt class="py-lineno">379</tt>  <tt class="py-line">                <tt id="link-201" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-201', 'hsp', 'link-151');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'hit_start'</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt id="link-202" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-202', 'hsp', 'link-151');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'hit_end'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-203" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-203', 'split', 'link-80');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">' '</tt><tt class="py-op">,</tt> <tt class="py-number">4</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">:</tt><tt class="py-number">5</tt><tt class="py-op">:</tt><tt class="py-number">2</tt><tt class="py-op">]</tt> </tt>
<a name="L380"></a><tt class="py-lineno">380</tt>  <tt class="py-line"> </tt>
<a name="L381"></a><tt class="py-lineno">381</tt>  <tt class="py-line">        <tt class="py-comment"># determine strand</tt> </tt>
<a name="L382"></a><tt class="py-lineno">382</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">qresult</tt><tt class="py-op">[</tt><tt class="py-string">'description'</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-204" class="py-name" targets="Method Bio.Seq.Seq.endswith()=Bio.Seq.Seq-class.html#endswith"><a title="Bio.Seq.Seq.endswith" class="py-name" href="#" onclick="return doclink('link-204', 'endswith', 'link-204');">endswith</a></tt><tt class="py-op">(</tt><tt class="py-string">':[revcomp]'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L383"></a><tt class="py-lineno">383</tt>  <tt class="py-line">            <tt id="link-205" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-205', 'hsp', 'link-151');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'query_strand'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">'-'</tt> </tt>
<a name="L384"></a><tt class="py-lineno">384</tt>  <tt class="py-line">            <tt class="py-name">qresult</tt><tt class="py-op">[</tt><tt class="py-string">'description'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">qresult</tt><tt class="py-op">[</tt><tt class="py-string">'description'</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">':[revcomp]'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L385"></a><tt class="py-lineno">385</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-string">'protein2'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qresult</tt><tt class="py-op">[</tt><tt class="py-string">'model'</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L386"></a><tt class="py-lineno">386</tt>  <tt class="py-line">            <tt id="link-206" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-206', 'hsp', 'link-151');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'query_strand'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">'.'</tt> </tt>
<a name="L387"></a><tt class="py-lineno">387</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L388"></a><tt class="py-lineno">388</tt>  <tt class="py-line">            <tt id="link-207" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-207', 'hsp', 'link-151');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'query_strand'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">'+'</tt> </tt>
<a name="L389"></a><tt class="py-lineno">389</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-208" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-208', 'hit', 'link-44');">hit</a></tt><tt class="py-op">[</tt><tt class="py-string">'description'</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-209" class="py-name"><a title="Bio.Seq.Seq.endswith" class="py-name" href="#" onclick="return doclink('link-209', 'endswith', 'link-204');">endswith</a></tt><tt class="py-op">(</tt><tt class="py-string">':[revcomp]'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L390"></a><tt class="py-lineno">390</tt>  <tt class="py-line">            <tt id="link-210" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-210', 'hsp', 'link-151');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'hit_strand'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">'-'</tt> </tt>
<a name="L391"></a><tt class="py-lineno">391</tt>  <tt class="py-line">            <tt id="link-211" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-211', 'hit', 'link-44');">hit</a></tt><tt class="py-op">[</tt><tt class="py-string">'description'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-212" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-212', 'hit', 'link-44');">hit</a></tt><tt class="py-op">[</tt><tt class="py-string">'description'</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">':[revcomp]'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt> </tt>
<a name="L392"></a><tt class="py-lineno">392</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-string">'2protein'</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qresult</tt><tt class="py-op">[</tt><tt class="py-string">'model'</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L393"></a><tt class="py-lineno">393</tt>  <tt class="py-line">            <tt id="link-213" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-213', 'hsp', 'link-151');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'hit_strand'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">'.'</tt> </tt>
<a name="L394"></a><tt class="py-lineno">394</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L395"></a><tt class="py-lineno">395</tt>  <tt class="py-line">            <tt id="link-214" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-214', 'hsp', 'link-151');">hsp</a></tt><tt class="py-op">[</tt><tt class="py-string">'hit_strand'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">'+'</tt> </tt>
<a name="L396"></a><tt class="py-lineno">396</tt>  <tt class="py-line"> </tt>
<a name="L397"></a><tt class="py-lineno">397</tt>  <tt class="py-line">        <tt class="py-comment"># NOTE: we haven't processed the coordinates types</tt> </tt>
<a name="L398"></a><tt class="py-lineno">398</tt>  <tt class="py-line">        <tt class="py-comment"># and the strands are not yet Biopython's standard (1 / -1 / 0)</tt> </tt>
<a name="L399"></a><tt class="py-lineno">399</tt>  <tt class="py-line">        <tt class="py-comment"># since it's easier if we do the conversion later</tt> </tt>
<a name="L400"></a><tt class="py-lineno">400</tt>  <tt class="py-line"> </tt>
<a name="L401"></a><tt class="py-lineno">401</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">{</tt><tt class="py-string">'qresult'</tt><tt class="py-op">:</tt> <tt class="py-name">qresult</tt><tt class="py-op">,</tt> <tt class="py-string">'hit'</tt><tt class="py-op">:</tt> <tt id="link-215" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-215', 'hit', 'link-44');">hit</a></tt><tt class="py-op">,</tt> <tt class="py-string">'hsp'</tt><tt class="py-op">:</tt> <tt id="link-216" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-216', 'hsp', 'link-151');">hsp</a></tt><tt class="py-op">}</tt> </tt>
</div><a name="L402"></a><tt class="py-lineno">402</tt>  <tt class="py-line"> </tt>
<a name="_BaseExonerateParser._parse_qresult"></a><div id="_BaseExonerateParser._parse_qresult-def"><a name="L403"></a><tt class="py-lineno">403</tt> <a class="py-toggle" href="#" id="_BaseExonerateParser._parse_qresult-toggle" onclick="return toggle('_BaseExonerateParser._parse_qresult');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO._base._BaseExonerateParser-class.html#_parse_qresult">_parse_qresult</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BaseExonerateParser._parse_qresult-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="_BaseExonerateParser._parse_qresult-expanded"><a name="L404"></a><tt class="py-lineno">404</tt>  <tt class="py-line">        <tt class="py-comment"># state values</tt> </tt>
<a name="L405"></a><tt class="py-lineno">405</tt>  <tt class="py-line">        <tt class="py-name">state_EOF</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L406"></a><tt class="py-lineno">406</tt>  <tt class="py-line">        <tt class="py-name">state_QRES_NEW</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L407"></a><tt class="py-lineno">407</tt>  <tt class="py-line">        <tt class="py-name">state_QRES_SAME</tt> <tt class="py-op">=</tt> <tt class="py-number">3</tt> </tt>
<a name="L408"></a><tt class="py-lineno">408</tt>  <tt class="py-line">        <tt class="py-name">state_HIT_NEW</tt> <tt class="py-op">=</tt> <tt class="py-number">2</tt> </tt>
<a name="L409"></a><tt class="py-lineno">409</tt>  <tt class="py-line">        <tt class="py-name">state_HIT_SAME</tt> <tt class="py-op">=</tt> <tt class="py-number">4</tt> </tt>
<a name="L410"></a><tt class="py-lineno">410</tt>  <tt class="py-line">        <tt class="py-comment"># initial dummies</tt> </tt>
<a name="L411"></a><tt class="py-lineno">411</tt>  <tt class="py-line">        <tt class="py-name">qres_state</tt><tt class="py-op">,</tt> <tt class="py-name">hit_state</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt> </tt>
<a name="L412"></a><tt class="py-lineno">412</tt>  <tt class="py-line">        <tt class="py-name">file_state</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L413"></a><tt class="py-lineno">413</tt>  <tt class="py-line">        <tt class="py-name">cur_qid</tt><tt class="py-op">,</tt> <tt class="py-name">cur_hid</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt> </tt>
<a name="L414"></a><tt class="py-lineno">414</tt>  <tt class="py-line">        <tt class="py-name">prev_qid</tt><tt class="py-op">,</tt> <tt class="py-name">prev_hid</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt> </tt>
<a name="L415"></a><tt class="py-lineno">415</tt>  <tt class="py-line">        <tt class="py-name">cur</tt><tt class="py-op">,</tt> <tt class="py-name">prev</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt> </tt>
<a name="L416"></a><tt class="py-lineno">416</tt>  <tt class="py-line">        <tt class="py-name">hit_list</tt><tt class="py-op">,</tt> <tt class="py-name">hsp_list</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L417"></a><tt class="py-lineno">417</tt>  <tt class="py-line">        <tt class="py-comment"># if the file has c4 alignments, use that as the alignment mark</tt> </tt>
<a name="L418"></a><tt class="py-lineno">418</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">has_c4_alignment</tt><tt class="py-op">:</tt> </tt>
<a name="L419"></a><tt class="py-lineno">419</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-217" class="py-name"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateParser._ALN_MARK
Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarParser._ALN_MARK
Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextParser._ALN_MARK
Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarParser._ALN_MARK" class="py-name" href="#" onclick="return doclink('link-217', '_ALN_MARK', 'link-163');">_ALN_MARK</a></tt> <tt class="py-op">=</tt> <tt class="py-string">'C4 Alignment:'</tt> </tt>
<a name="L420"></a><tt class="py-lineno">420</tt>  <tt class="py-line"> </tt>
<a name="L421"></a><tt class="py-lineno">421</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L422"></a><tt class="py-lineno">422</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-218" class="py-name" targets="Method Bio.SearchIO.ExonerateIO._base._BaseExonerateParser.read_until()=Bio.SearchIO.ExonerateIO._base._BaseExonerateParser-class.html#read_until"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateParser.read_until" class="py-name" href="#" onclick="return doclink('link-218', 'read_until', 'link-218');">read_until</a></tt><tt class="py-op">(</tt><tt class="py-keyword">lambda</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-219" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-219', 'startswith', 'link-165');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-220" class="py-name"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateParser._ALN_MARK
Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarParser._ALN_MARK
Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextParser._ALN_MARK
Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarParser._ALN_MARK" class="py-name" href="#" onclick="return doclink('link-220', '_ALN_MARK', 'link-163');">_ALN_MARK</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L423"></a><tt class="py-lineno">423</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">cur</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L424"></a><tt class="py-lineno">424</tt>  <tt class="py-line">                <tt class="py-name">prev</tt> <tt class="py-op">=</tt> <tt class="py-name">cur</tt> </tt>
<a name="L425"></a><tt class="py-lineno">425</tt>  <tt class="py-line">                <tt class="py-name">prev_qid</tt> <tt class="py-op">=</tt> <tt class="py-name">cur_qid</tt> </tt>
<a name="L426"></a><tt class="py-lineno">426</tt>  <tt class="py-line">                <tt class="py-name">prev_hid</tt> <tt class="py-op">=</tt> <tt class="py-name">cur_hid</tt> </tt>
<a name="L427"></a><tt class="py-lineno">427</tt>  <tt class="py-line">            <tt class="py-comment"># only parse the result row if it's not EOF</tt> </tt>
<a name="L428"></a><tt class="py-lineno">428</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L429"></a><tt class="py-lineno">429</tt>  <tt class="py-line">                <tt class="py-keyword">assert</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-221" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-221', 'startswith', 'link-165');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-222" class="py-name"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateParser._ALN_MARK
Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarParser._ALN_MARK
Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextParser._ALN_MARK
Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarParser._ALN_MARK" class="py-name" href="#" onclick="return doclink('link-222', '_ALN_MARK', 'link-163');">_ALN_MARK</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> </tt>
<a name="L430"></a><tt class="py-lineno">430</tt>  <tt class="py-line">                <tt class="py-comment"># create temp dicts for storing parsed values</tt> </tt>
<a name="L431"></a><tt class="py-lineno">431</tt>  <tt class="py-line">                <tt id="link-223" class="py-name" targets="Variable Bio.Wise.dnal.Statistics.header=Bio.Wise.dnal.Statistics-class.html#header"><a title="Bio.Wise.dnal.Statistics.header" class="py-name" href="#" onclick="return doclink('link-223', 'header', 'link-223');">header</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'qresult'</tt><tt class="py-op">:</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt><tt class="py-op">,</tt> <tt class="py-string">'hit'</tt><tt class="py-op">:</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt><tt class="py-op">,</tt> <tt class="py-string">'hsp'</tt><tt class="py-op">:</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt><tt class="py-op">}</tt> </tt>
<a name="L432"></a><tt class="py-lineno">432</tt>  <tt class="py-line">                <tt class="py-comment"># if the file has c4 alignments, try to parse the header</tt> </tt>
<a name="L433"></a><tt class="py-lineno">433</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">has_c4_alignment</tt><tt class="py-op">:</tt> </tt>
<a name="L434"></a><tt class="py-lineno">434</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-224" class="py-name"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateParser.read_until" class="py-name" href="#" onclick="return doclink('link-224', 'read_until', 'link-218');">read_until</a></tt><tt class="py-op">(</tt><tt class="py-keyword">lambda</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L435"></a><tt class="py-lineno">435</tt>  <tt class="py-line">                            <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-225" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-225', 'strip', 'link-179');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-226" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-226', 'startswith', 'link-165');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'Query:'</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L436"></a><tt class="py-lineno">436</tt>  <tt class="py-line">                    <tt id="link-227" class="py-name"><a title="Bio.Wise.dnal.Statistics.header" class="py-name" href="#" onclick="return doclink('link-227', 'header', 'link-223');">header</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-228" class="py-name" targets="Method Bio.SearchIO.ExonerateIO._base._BaseExonerateParser._parse_alignment_header()=Bio.SearchIO.ExonerateIO._base._BaseExonerateParser-class.html#_parse_alignment_header"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateParser._parse_alignment_header" class="py-name" href="#" onclick="return doclink('link-228', '_parse_alignment_header', 'link-228');">_parse_alignment_header</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L437"></a><tt class="py-lineno">437</tt>  <tt class="py-line">                <tt class="py-comment"># parse the block contents</tt> </tt>
<a name="L438"></a><tt class="py-lineno">438</tt>  <tt class="py-line">                <tt class="py-name">cur</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-229" class="py-name" targets="Method Bio.SearchIO.ExonerateIO._base._BaseExonerateParser.parse_alignment_block()=Bio.SearchIO.ExonerateIO._base._BaseExonerateParser-class.html#parse_alignment_block,Method Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarParser.parse_alignment_block()=Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarParser-class.html#parse_alignment_block,Method Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextParser.parse_alignment_block()=Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextParser-class.html#parse_alignment_block,Method Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarParser.parse_alignment_block()=Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarParser-class.html#parse_alignment_block"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateParser.parse_alignment_block
Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarParser.parse_alignment_block
Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextParser.parse_alignment_block
Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarParser.parse_alignment_block" class="py-name" href="#" onclick="return doclink('link-229', 'parse_alignment_block', 'link-229');">parse_alignment_block</a></tt><tt class="py-op">(</tt><tt id="link-230" class="py-name"><a title="Bio.Wise.dnal.Statistics.header" class="py-name" href="#" onclick="return doclink('link-230', 'header', 'link-223');">header</a></tt><tt class="py-op">)</tt> </tt>
<a name="L439"></a><tt class="py-lineno">439</tt>  <tt class="py-line">                <tt class="py-name">cur_qid</tt> <tt class="py-op">=</tt> <tt class="py-name">cur</tt><tt class="py-op">[</tt><tt class="py-string">'qresult'</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-string">'id'</tt><tt class="py-op">]</tt> </tt>
<a name="L440"></a><tt class="py-lineno">440</tt>  <tt class="py-line">                <tt class="py-name">cur_hid</tt> <tt class="py-op">=</tt> <tt class="py-name">cur</tt><tt class="py-op">[</tt><tt class="py-string">'hit'</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-string">'id'</tt><tt class="py-op">]</tt> </tt>
<a name="L441"></a><tt class="py-lineno">441</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-keyword">or</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-231" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-231', 'startswith', 'link-165');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">'-- completed '</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L442"></a><tt class="py-lineno">442</tt>  <tt class="py-line">                <tt class="py-name">file_state</tt> <tt class="py-op">=</tt> <tt class="py-name">state_EOF</tt> </tt>
<a name="L443"></a><tt class="py-lineno">443</tt>  <tt class="py-line">                <tt class="py-name">cur_qid</tt><tt class="py-op">,</tt> <tt class="py-name">cur_hid</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt> </tt>
<a name="L444"></a><tt class="py-lineno">444</tt>  <tt class="py-line"> </tt>
<a name="L445"></a><tt class="py-lineno">445</tt>  <tt class="py-line">            <tt class="py-comment"># get the state of hit and qresult</tt> </tt>
<a name="L446"></a><tt class="py-lineno">446</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">prev_qid</tt> <tt class="py-op">!=</tt> <tt class="py-name">cur_qid</tt><tt class="py-op">:</tt> </tt>
<a name="L447"></a><tt class="py-lineno">447</tt>  <tt class="py-line">                <tt class="py-name">qres_state</tt> <tt class="py-op">=</tt> <tt class="py-name">state_QRES_NEW</tt> </tt>
<a name="L448"></a><tt class="py-lineno">448</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L449"></a><tt class="py-lineno">449</tt>  <tt class="py-line">                <tt class="py-name">qres_state</tt> <tt class="py-op">=</tt> <tt class="py-name">state_QRES_SAME</tt> </tt>
<a name="L450"></a><tt class="py-lineno">450</tt>  <tt class="py-line">            <tt class="py-comment"># new hits are hits with different ids or hits in a new query</tt> </tt>
<a name="L451"></a><tt class="py-lineno">451</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">prev_hid</tt> <tt class="py-op">!=</tt> <tt class="py-name">cur_hid</tt> <tt class="py-keyword">or</tt> <tt class="py-name">qres_state</tt> <tt class="py-op">==</tt> <tt class="py-name">state_QRES_NEW</tt><tt class="py-op">:</tt> </tt>
<a name="L452"></a><tt class="py-lineno">452</tt>  <tt class="py-line">                <tt class="py-name">hit_state</tt> <tt class="py-op">=</tt> <tt class="py-name">state_HIT_NEW</tt> </tt>
<a name="L453"></a><tt class="py-lineno">453</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L454"></a><tt class="py-lineno">454</tt>  <tt class="py-line">                <tt class="py-name">hit_state</tt> <tt class="py-op">=</tt> <tt class="py-name">state_HIT_SAME</tt> </tt>
<a name="L455"></a><tt class="py-lineno">455</tt>  <tt class="py-line"> </tt>
<a name="L456"></a><tt class="py-lineno">456</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">prev</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L457"></a><tt class="py-lineno">457</tt>  <tt class="py-line">                <tt id="link-232" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-232', 'hsp', 'link-151');">hsp</a></tt> <tt class="py-op">=</tt> <tt id="link-233" class="py-name" targets="Function Bio.SearchIO.BlatIO._create_hsp()=Bio.SearchIO.BlatIO-module.html#_create_hsp,Function Bio.SearchIO.ExonerateIO._base._create_hsp()=Bio.SearchIO.ExonerateIO._base-module.html#_create_hsp"><a title="Bio.SearchIO.BlatIO._create_hsp
Bio.SearchIO.ExonerateIO._base._create_hsp" class="py-name" href="#" onclick="return doclink('link-233', '_create_hsp', 'link-233');">_create_hsp</a></tt><tt class="py-op">(</tt><tt class="py-name">prev_hid</tt><tt class="py-op">,</tt> <tt class="py-name">prev_qid</tt><tt class="py-op">,</tt> <tt class="py-name">prev</tt><tt class="py-op">[</tt><tt class="py-string">'hsp'</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L458"></a><tt class="py-lineno">458</tt>  <tt class="py-line">                <tt class="py-name">hsp_list</tt><tt class="py-op">.</tt><tt id="link-234" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-234', 'append', 'link-120');">append</a></tt><tt class="py-op">(</tt><tt id="link-235" class="py-name"><a title="Bio.SearchIO._model.hsp" class="py-name" href="#" onclick="return doclink('link-235', 'hsp', 'link-151');">hsp</a></tt><tt class="py-op">)</tt> </tt>
<a name="L459"></a><tt class="py-lineno">459</tt>  <tt class="py-line"> </tt>
<a name="L460"></a><tt class="py-lineno">460</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">hit_state</tt> <tt class="py-op">==</tt> <tt class="py-name">state_HIT_NEW</tt><tt class="py-op">:</tt> </tt>
<a name="L461"></a><tt class="py-lineno">461</tt>  <tt class="py-line">                    <tt id="link-236" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-236', 'hit', 'link-44');">hit</a></tt> <tt class="py-op">=</tt> <tt id="link-237" class="py-name"><a title="Bio.SearchIO._model.hit.Hit" class="py-name" href="#" onclick="return doclink('link-237', 'Hit', 'link-8');">Hit</a></tt><tt class="py-op">(</tt><tt class="py-name">hsp_list</tt><tt class="py-op">)</tt> </tt>
<a name="L462"></a><tt class="py-lineno">462</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-name">attr</tt><tt class="py-op">,</tt> <tt id="link-238" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-238', 'value', 'link-132');">value</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">prev</tt><tt class="py-op">[</tt><tt class="py-string">'hit'</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-239" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-239', 'items', 'link-60');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L463"></a><tt class="py-lineno">463</tt>  <tt class="py-line">                        <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt id="link-240" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-240', 'hit', 'link-44');">hit</a></tt><tt class="py-op">,</tt> <tt class="py-name">attr</tt><tt class="py-op">,</tt> <tt id="link-241" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-241', 'value', 'link-132');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L464"></a><tt class="py-lineno">464</tt>  <tt class="py-line">                    <tt class="py-name">hit_list</tt><tt class="py-op">.</tt><tt id="link-242" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-242', 'append', 'link-120');">append</a></tt><tt class="py-op">(</tt><tt id="link-243" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-243', 'hit', 'link-44');">hit</a></tt><tt class="py-op">)</tt> </tt>
<a name="L465"></a><tt class="py-lineno">465</tt>  <tt class="py-line">                    <tt class="py-name">hsp_list</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L466"></a><tt class="py-lineno">466</tt>  <tt class="py-line"> </tt>
<a name="L467"></a><tt class="py-lineno">467</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">qres_state</tt> <tt class="py-op">==</tt> <tt class="py-name">state_QRES_NEW</tt> <tt class="py-keyword">or</tt> <tt class="py-name">file_state</tt> <tt class="py-op">==</tt> <tt class="py-name">state_EOF</tt><tt class="py-op">:</tt> </tt>
<a name="L468"></a><tt class="py-lineno">468</tt>  <tt class="py-line">                    <tt class="py-name">qresult</tt> <tt class="py-op">=</tt> <tt id="link-244" class="py-name"><a title="Bio.SearchIO._model.query.QueryResult" class="py-name" href="#" onclick="return doclink('link-244', 'QueryResult', 'link-7');">QueryResult</a></tt><tt class="py-op">(</tt><tt id="link-245" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-245', 'id', 'link-155');">id</a></tt><tt class="py-op">=</tt><tt class="py-name">prev_qid</tt><tt class="py-op">)</tt> </tt>
<a name="L469"></a><tt class="py-lineno">469</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt id="link-246" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-246', 'hit', 'link-44');">hit</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">hit_list</tt><tt class="py-op">:</tt> </tt>
<a name="L470"></a><tt class="py-lineno">470</tt>  <tt class="py-line">                        <tt class="py-comment"># not using append since Exonerate may separate the</tt> </tt>
<a name="L471"></a><tt class="py-lineno">471</tt>  <tt class="py-line">                        <tt class="py-comment"># same hit if it has different strands</tt> </tt>
<a name="L472"></a><tt class="py-lineno">472</tt>  <tt class="py-line">                        <tt class="py-name">qresult</tt><tt class="py-op">.</tt><tt id="link-247" class="py-name" targets="Method Bio.SearchIO._model.query.QueryResult.absorb()=Bio.SearchIO._model.query.QueryResult-class.html#absorb"><a title="Bio.SearchIO._model.query.QueryResult.absorb" class="py-name" href="#" onclick="return doclink('link-247', 'absorb', 'link-247');">absorb</a></tt><tt class="py-op">(</tt><tt id="link-248" class="py-name"><a title="Bio.SearchIO._model.hit
Bio.SearchIO._model.hsp.HSP.hit
Bio.SearchIO._model.hsp.HSPFragment.hit" class="py-name" href="#" onclick="return doclink('link-248', 'hit', 'link-44');">hit</a></tt><tt class="py-op">)</tt> </tt>
<a name="L473"></a><tt class="py-lineno">473</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-name">attr</tt><tt class="py-op">,</tt> <tt id="link-249" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-249', 'value', 'link-132');">value</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">prev</tt><tt class="py-op">[</tt><tt class="py-string">'qresult'</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-250" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-250', 'items', 'link-60');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L474"></a><tt class="py-lineno">474</tt>  <tt class="py-line">                        <tt class="py-name">setattr</tt><tt class="py-op">(</tt><tt class="py-name">qresult</tt><tt class="py-op">,</tt> <tt class="py-name">attr</tt><tt class="py-op">,</tt> <tt id="link-251" class="py-name"><a title="Bio.Entrez.Parser.Consumer.value
Bio.Entrez.Parser.DictionaryConsumer.value
Bio.Entrez.Parser.ErrorConsumer.value
Bio.Entrez.Parser.IntegerConsumer.value
Bio.Entrez.Parser.ListConsumer.value
Bio.Entrez.Parser.StringConsumer.value
Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-251', 'value', 'link-132');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L475"></a><tt class="py-lineno">475</tt>  <tt class="py-line">                    <tt class="py-keyword">yield</tt> <tt class="py-name">qresult</tt> </tt>
<a name="L476"></a><tt class="py-lineno">476</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">file_state</tt> <tt class="py-op">==</tt> <tt class="py-name">state_EOF</tt><tt class="py-op">:</tt> </tt>
<a name="L477"></a><tt class="py-lineno">477</tt>  <tt class="py-line">                        <tt class="py-keyword">break</tt> </tt>
<a name="L478"></a><tt class="py-lineno">478</tt>  <tt class="py-line">                    <tt class="py-name">hit_list</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L479"></a><tt class="py-lineno">479</tt>  <tt class="py-line"> </tt>
<a name="L480"></a><tt class="py-lineno">480</tt>  <tt class="py-line">            <tt class="py-comment"># only readline() here if we're not parsing C4 alignments</tt> </tt>
<a name="L481"></a><tt class="py-lineno">481</tt>  <tt class="py-line">            <tt class="py-comment"># C4 alignments readline() is handled by its parse_alignment_block</tt> </tt>
<a name="L482"></a><tt class="py-lineno">482</tt>  <tt class="py-line">            <tt class="py-comment"># function</tt> </tt>
<a name="L483"></a><tt class="py-lineno">483</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">has_c4_alignment</tt><tt class="py-op">:</tt> </tt>
<a name="L484"></a><tt class="py-lineno">484</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-252" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-252', 'readline', 'link-164');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L485"></a><tt class="py-lineno">485</tt>  <tt class="py-line"> </tt>
<a name="L486"></a><tt class="py-lineno">486</tt>  <tt class="py-line"> </tt>
<a name="_BaseExonerateIndexer"></a><div id="_BaseExonerateIndexer-def"><a name="L487"></a><tt class="py-lineno">487</tt> <a class="py-toggle" href="#" id="_BaseExonerateIndexer-toggle" onclick="return toggle('_BaseExonerateIndexer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer-class.html">_BaseExonerateIndexer</a><tt class="py-op">(</tt><tt class="py-base-class">SearchIndexer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BaseExonerateIndexer-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_BaseExonerateIndexer-expanded"><a name="L488"></a><tt class="py-lineno">488</tt>  <tt class="py-line">    <tt class="py-docstring">"""Indexer class for Exonerate plain text."""</tt> </tt>
<a name="L489"></a><tt class="py-lineno">489</tt>  <tt class="py-line"> </tt>
<a name="L490"></a><tt class="py-lineno">490</tt>  <tt class="py-line">    <tt id="link-253" class="py-name" targets="Variable Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer._parser=Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer-class.html#_parser"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer._parser" class="py-name" href="#" onclick="return doclink('link-253', '_parser', 'link-253');">_parser</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt>  <tt class="py-comment"># should be defined by subclass</tt> </tt>
<a name="L491"></a><tt class="py-lineno">491</tt>  <tt class="py-line">    <tt id="link-254" class="py-name" targets="Variable Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer._query_mark=Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer-class.html#_query_mark,Variable Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarIndexer._query_mark=Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarIndexer-class.html#_query_mark,Variable Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextIndexer._query_mark=Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextIndexer-class.html#_query_mark,Variable Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarIndexer._query_mark=Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarIndexer-class.html#_query_mark"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer._query_mark
Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarIndexer._query_mark
Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextIndexer._query_mark
Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarIndexer._query_mark" class="py-name" href="#" onclick="return doclink('link-254', '_query_mark', 'link-254');">_query_mark</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt>  <tt class="py-comment"># this one too</tt> </tt>
<a name="L492"></a><tt class="py-lineno">492</tt>  <tt class="py-line"> </tt>
<a name="_BaseExonerateIndexer.get_qresult_id"></a><div id="_BaseExonerateIndexer.get_qresult_id-def"><a name="L493"></a><tt class="py-lineno">493</tt> <a class="py-toggle" href="#" id="_BaseExonerateIndexer.get_qresult_id-toggle" onclick="return toggle('_BaseExonerateIndexer.get_qresult_id');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer-class.html#get_qresult_id">get_qresult_id</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">pos</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BaseExonerateIndexer.get_qresult_id-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="_BaseExonerateIndexer.get_qresult_id-expanded"><a name="L494"></a><tt class="py-lineno">494</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">NotImplementedError</tt><tt class="py-op">(</tt><tt class="py-string">"Should be defined by subclass"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L495"></a><tt class="py-lineno">495</tt>  <tt class="py-line"> </tt>
<a name="_BaseExonerateIndexer.__iter__"></a><div id="_BaseExonerateIndexer.__iter__-def"><a name="L496"></a><tt class="py-lineno">496</tt> <a class="py-toggle" href="#" id="_BaseExonerateIndexer.__iter__-toggle" onclick="return toggle('_BaseExonerateIndexer.__iter__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer-class.html#__iter__">__iter__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_BaseExonerateIndexer.__iter__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="_BaseExonerateIndexer.__iter__-expanded"><a name="L497"></a><tt class="py-lineno">497</tt>  <tt class="py-line">        <tt class="py-docstring">"""Iterate over the file handle; yields key, start offset, and length."""</tt> </tt>
<a name="L498"></a><tt class="py-lineno">498</tt>  <tt class="py-line">        <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_handle</tt> </tt>
<a name="L499"></a><tt class="py-lineno">499</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-255" class="py-name" targets="Method Bio.File.UndoHandle.seek()=Bio.File.UndoHandle-class.html#seek,Method Bio.SeqIO.SffIO._AddTellHandle.seek()=Bio.SeqIO.SffIO._AddTellHandle-class.html#seek,Method Bio.bgzf.BgzfReader.seek()=Bio.bgzf.BgzfReader-class.html#seek"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-255', 'seek', 'link-255');">seek</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L500"></a><tt class="py-lineno">500</tt>  <tt class="py-line">        <tt class="py-name">qresult_key</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L501"></a><tt class="py-lineno">501</tt>  <tt class="py-line"> </tt>
<a name="L502"></a><tt class="py-lineno">502</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L503"></a><tt class="py-lineno">503</tt>  <tt class="py-line">            <tt class="py-name">start_offset</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-256" class="py-name" targets="Method Bio.File.UndoHandle.tell()=Bio.File.UndoHandle-class.html#tell,Method Bio.SeqIO.SffIO._AddTellHandle.tell()=Bio.SeqIO.SffIO._AddTellHandle-class.html#tell,Method Bio.bgzf.BgzfReader.tell()=Bio.bgzf.BgzfReader-class.html#tell,Method Bio.bgzf.BgzfWriter.tell()=Bio.bgzf.BgzfWriter-class.html#tell"><a title="Bio.File.UndoHandle.tell
Bio.SeqIO.SffIO._AddTellHandle.tell
Bio.bgzf.BgzfReader.tell
Bio.bgzf.BgzfWriter.tell" class="py-name" href="#" onclick="return doclink('link-256', 'tell', 'link-256');">tell</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L504"></a><tt class="py-lineno">504</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-257" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-257', 'readline', 'link-164');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L505"></a><tt class="py-lineno">505</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-258" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-258', 'startswith', 'link-165');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-259" class="py-name"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer._query_mark
Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarIndexer._query_mark
Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextIndexer._query_mark
Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarIndexer._query_mark" class="py-name" href="#" onclick="return doclink('link-259', '_query_mark', 'link-254');">_query_mark</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L506"></a><tt class="py-lineno">506</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">qresult_key</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L507"></a><tt class="py-lineno">507</tt>  <tt class="py-line">                    <tt class="py-name">qresult_key</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-260" class="py-name" targets="Method Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer.get_qresult_id()=Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer-class.html#get_qresult_id,Method Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarIndexer.get_qresult_id()=Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarIndexer-class.html#get_qresult_id,Method Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextIndexer.get_qresult_id()=Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextIndexer-class.html#get_qresult_id,Method Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarIndexer.get_qresult_id()=Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarIndexer-class.html#get_qresult_id"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer.get_qresult_id
Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarIndexer.get_qresult_id
Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextIndexer.get_qresult_id
Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarIndexer.get_qresult_id" class="py-name" href="#" onclick="return doclink('link-260', 'get_qresult_id', 'link-260');">get_qresult_id</a></tt><tt class="py-op">(</tt><tt class="py-name">start_offset</tt><tt class="py-op">)</tt> </tt>
<a name="L508"></a><tt class="py-lineno">508</tt>  <tt class="py-line">                    <tt class="py-name">qresult_offset</tt> <tt class="py-op">=</tt> <tt class="py-name">start_offset</tt> </tt>
<a name="L509"></a><tt class="py-lineno">509</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L510"></a><tt class="py-lineno">510</tt>  <tt class="py-line">                    <tt class="py-name">curr_key</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-261" class="py-name"><a title="Bio.SearchIO.ExonerateIO._base._BaseExonerateIndexer.get_qresult_id
Bio.SearchIO.ExonerateIO.exonerate_cigar.ExonerateCigarIndexer.get_qresult_id
Bio.SearchIO.ExonerateIO.exonerate_text.ExonerateTextIndexer.get_qresult_id
Bio.SearchIO.ExonerateIO.exonerate_vulgar.ExonerateVulgarIndexer.get_qresult_id" class="py-name" href="#" onclick="return doclink('link-261', 'get_qresult_id', 'link-260');">get_qresult_id</a></tt><tt class="py-op">(</tt><tt class="py-name">start_offset</tt><tt class="py-op">)</tt> </tt>
<a name="L511"></a><tt class="py-lineno">511</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">curr_key</tt> <tt class="py-op">!=</tt> <tt class="py-name">qresult_key</tt><tt class="py-op">:</tt> </tt>
<a name="L512"></a><tt class="py-lineno">512</tt>  <tt class="py-line">                        <tt class="py-keyword">yield</tt> <tt class="py-name">qresult_key</tt><tt class="py-op">,</tt> <tt class="py-name">qresult_offset</tt><tt class="py-op">,</tt> \ </tt>
<a name="L513"></a><tt class="py-lineno">513</tt>  <tt class="py-line">                                <tt class="py-name">start_offset</tt> <tt class="py-op">-</tt> <tt class="py-name">qresult_offset</tt> </tt>
<a name="L514"></a><tt class="py-lineno">514</tt>  <tt class="py-line">                        <tt class="py-name">qresult_key</tt> <tt class="py-op">=</tt> <tt class="py-name">curr_key</tt> </tt>
<a name="L515"></a><tt class="py-lineno">515</tt>  <tt class="py-line">                        <tt class="py-name">qresult_offset</tt> <tt class="py-op">=</tt> <tt class="py-name">start_offset</tt> </tt>
<a name="L516"></a><tt class="py-lineno">516</tt>  <tt class="py-line">                        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-262" class="py-name"><a title="Bio.File.UndoHandle.seek
Bio.SeqIO.SffIO._AddTellHandle.seek
Bio.bgzf.BgzfReader.seek" class="py-name" href="#" onclick="return doclink('link-262', 'seek', 'link-255');">seek</a></tt><tt class="py-op">(</tt><tt class="py-name">qresult_offset</tt><tt class="py-op">)</tt> </tt>
<a name="L517"></a><tt class="py-lineno">517</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L518"></a><tt class="py-lineno">518</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt class="py-name">qresult_key</tt><tt class="py-op">,</tt> <tt class="py-name">qresult_offset</tt><tt class="py-op">,</tt> \ </tt>
<a name="L519"></a><tt class="py-lineno">519</tt>  <tt class="py-line">                        <tt class="py-name">start_offset</tt> <tt class="py-op">-</tt> <tt class="py-name">qresult_offset</tt> </tt>
<a name="L520"></a><tt class="py-lineno">520</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
</div></div><a name="L521"></a><tt class="py-lineno">521</tt>  <tt class="py-line"> </tt>
<a name="L522"></a><tt class="py-lineno">522</tt>  <tt class="py-line"> </tt>
<a name="L523"></a><tt class="py-lineno">523</tt>  <tt class="py-line"><tt class="py-comment"># if not used as a module, run the doctest</tt> </tt>
<a name="L524"></a><tt class="py-lineno">524</tt>  <tt class="py-line"><tt class="py-keyword">if</tt> <tt class="py-name">__name__</tt> <tt class="py-op">==</tt> <tt class="py-string">"__main__"</tt><tt class="py-op">:</tt> </tt>
<a name="L525"></a><tt class="py-lineno">525</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-263" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-263', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-264" class="py-name" targets="Module Bio.Phylo._utils=Bio.Phylo._utils-module.html,Module Bio.SearchIO._utils=Bio.SearchIO._utils-module.html,Module Bio._utils=Bio._utils-module.html"><a title="Bio.Phylo._utils
Bio.SearchIO._utils
Bio._utils" class="py-name" href="#" onclick="return doclink('link-264', '_utils', 'link-264');">_utils</a></tt> <tt class="py-keyword">import</tt> <tt id="link-265" class="py-name" targets="Function Bio._utils.run_doctest()=Bio._utils-module.html#run_doctest"><a title="Bio._utils.run_doctest" class="py-name" href="#" onclick="return doclink('link-265', 'run_doctest', 'link-265');">run_doctest</a></tt> </tt>
<a name="L526"></a><tt class="py-lineno">526</tt>  <tt class="py-line">    <tt id="link-266" class="py-name"><a title="Bio._utils.run_doctest" class="py-name" href="#" onclick="return doclink('link-266', 'run_doctest', 'link-265');">run_doctest</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L527"></a><tt class="py-lineno">527</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Fri Jun 22 16:36:14 2018
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
