<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.Graphics.GenomeDiagram._CircularDrawer</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.Graphics-module.html">Package&nbsp;Graphics</a> ::
        <a href="Bio.Graphics.GenomeDiagram-module.html">Package&nbsp;GenomeDiagram</a> ::
        Module&nbsp;_CircularDrawer
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.Graphics.GenomeDiagram._CircularDrawer-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.Graphics.GenomeDiagram._CircularDrawer-module.html" onclick="show_private();">Module Bio.Graphics.GenomeDiagram._CircularDrawer</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">   1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2003-2008 by Leighton Pritchard.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">   2</tt>  <tt class="py-line"><tt class="py-comment"># Revisions copyright 2008-2017 by Peter Cock.</tt> </tt>
<a name="L3"></a><tt class="py-lineno">   3</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L4"></a><tt class="py-lineno">   4</tt>  <tt class="py-line"><tt class="py-comment"># This file is part of the Biopython distribution and governed by your</tt> </tt>
<a name="L5"></a><tt class="py-lineno">   5</tt>  <tt class="py-line"><tt class="py-comment"># choice of the "Biopython License Agreement" or the "BSD 3-Clause License".</tt> </tt>
<a name="L6"></a><tt class="py-lineno">   6</tt>  <tt class="py-line"><tt class="py-comment"># Please see the LICENSE file that should have been included as part of this</tt> </tt>
<a name="L7"></a><tt class="py-lineno">   7</tt>  <tt class="py-line"><tt class="py-comment"># package.</tt> </tt>
<a name="L8"></a><tt class="py-lineno">   8</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L9"></a><tt class="py-lineno">   9</tt>  <tt class="py-line"><tt class="py-comment"># Contact:       Leighton Pritchard, The James Hutton Institute,</tt> </tt>
<a name="L10"></a><tt class="py-lineno">  10</tt>  <tt class="py-line"><tt class="py-comment">#                Invergowrie, Dundee, Scotland, DD2 5DA, UK</tt> </tt>
<a name="L11"></a><tt class="py-lineno">  11</tt>  <tt class="py-line"><tt class="py-comment">#                Leighton.Pritchard@hutton.ac.uk</tt> </tt>
<a name="L12"></a><tt class="py-lineno">  12</tt>  <tt class="py-line"><tt class="py-comment">################################################################################</tt> </tt>
<a name="L13"></a><tt class="py-lineno">  13</tt>  <tt class="py-line"> </tt>
<a name="L14"></a><tt class="py-lineno">  14</tt>  <tt class="py-line"><tt class="py-docstring">"""CircularDrawer module for GenomeDiagram."""</tt> </tt>
<a name="L15"></a><tt class="py-lineno">  15</tt>  <tt class="py-line"> </tt>
<a name="L16"></a><tt class="py-lineno">  16</tt>  <tt class="py-line"><tt class="py-comment"># ReportLab imports</tt> </tt>
<a name="L17"></a><tt class="py-lineno">  17</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">__future__</tt> <tt class="py-keyword">import</tt> <tt class="py-name">print_function</tt> </tt>
<a name="L18"></a><tt class="py-lineno">  18</tt>  <tt class="py-line"> </tt>
<a name="L19"></a><tt class="py-lineno">  19</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">reportlab</tt><tt class="py-op">.</tt><tt class="py-name">graphics</tt><tt class="py-op">.</tt><tt class="py-name">shapes</tt> <tt class="py-keyword">import</tt> <tt class="py-name">Drawing</tt><tt class="py-op">,</tt> <tt class="py-name">String</tt><tt class="py-op">,</tt> <tt class="py-name">Group</tt><tt class="py-op">,</tt> <tt class="py-name">Line</tt><tt class="py-op">,</tt> <tt class="py-name">Circle</tt><tt class="py-op">,</tt> <tt id="link-0" class="py-name" targets="Class Bio.Phylo.PhyloXML.Polygon=Bio.Phylo.PhyloXML.Polygon-class.html"><a title="Bio.Phylo.PhyloXML.Polygon" class="py-name" href="#" onclick="return doclink('link-0', 'Polygon', 'link-0');">Polygon</a></tt> </tt>
<a name="L20"></a><tt class="py-lineno">  20</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">reportlab</tt><tt class="py-op">.</tt><tt class="py-name">lib</tt> <tt class="py-keyword">import</tt> <tt class="py-name">colors</tt> </tt>
<a name="L21"></a><tt class="py-lineno">  21</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">reportlab</tt><tt class="py-op">.</tt><tt class="py-name">graphics</tt><tt class="py-op">.</tt><tt class="py-name">shapes</tt> <tt class="py-keyword">import</tt> <tt class="py-name">ArcPath</tt> </tt>
<a name="L22"></a><tt class="py-lineno">  22</tt>  <tt class="py-line"> </tt>
<a name="L23"></a><tt class="py-lineno">  23</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-1" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-1', 'Bio', 'link-1');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-2" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-2', '_py3k', 'link-2');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt id="link-3" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.range()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#range,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.range()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#range,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Track.Track.range()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#range"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-3', 'range', 'link-3');">range</a></tt> </tt>
<a name="L24"></a><tt class="py-lineno">  24</tt>  <tt class="py-line"> </tt>
<a name="L25"></a><tt class="py-lineno">  25</tt>  <tt class="py-line"><tt class="py-comment"># GenomeDiagram imports</tt> </tt>
<a name="L26"></a><tt class="py-lineno">  26</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt><tt id="link-4" class="py-name" targets="Module Bio.Graphics.GenomeDiagram._AbstractDrawer=Bio.Graphics.GenomeDiagram._AbstractDrawer-module.html"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer" class="py-name" href="#" onclick="return doclink('link-4', '_AbstractDrawer', 'link-4');">_AbstractDrawer</a></tt> <tt class="py-keyword">import</tt> <tt id="link-5" class="py-name" targets="Class Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer=Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer-class.html"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer" class="py-name" href="#" onclick="return doclink('link-5', 'AbstractDrawer', 'link-5');">AbstractDrawer</a></tt><tt class="py-op">,</tt> <tt id="link-6" class="py-name" targets="Function Bio.Graphics.GenomeDiagram._AbstractDrawer.draw_polygon()=Bio.Graphics.GenomeDiagram._AbstractDrawer-module.html#draw_polygon"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer.draw_polygon" class="py-name" href="#" onclick="return doclink('link-6', 'draw_polygon', 'link-6');">draw_polygon</a></tt><tt class="py-op">,</tt> <tt id="link-7" class="py-name" targets="Function Bio.Graphics.GenomeDiagram._AbstractDrawer.intermediate_points()=Bio.Graphics.GenomeDiagram._AbstractDrawer-module.html#intermediate_points"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer.intermediate_points" class="py-name" href="#" onclick="return doclink('link-7', 'intermediate_points', 'link-7');">intermediate_points</a></tt> </tt>
<a name="L27"></a><tt class="py-lineno">  27</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt><tt id="link-8" class="py-name"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer" class="py-name" href="#" onclick="return doclink('link-8', '_AbstractDrawer', 'link-4');">_AbstractDrawer</a></tt> <tt class="py-keyword">import</tt> <tt id="link-9" class="py-name" targets="Function Bio.Graphics.GenomeDiagram._AbstractDrawer._stroke_and_fill_colors()=Bio.Graphics.GenomeDiagram._AbstractDrawer-module.html#_stroke_and_fill_colors"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer._stroke_and_fill_colors" class="py-name" href="#" onclick="return doclink('link-9', '_stroke_and_fill_colors', 'link-9');">_stroke_and_fill_colors</a></tt> </tt>
<a name="L28"></a><tt class="py-lineno">  28</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt><tt id="link-10" class="py-name" targets="Module Bio.Graphics.GenomeDiagram._FeatureSet=Bio.Graphics.GenomeDiagram._FeatureSet-module.html"><a title="Bio.Graphics.GenomeDiagram._FeatureSet" class="py-name" href="#" onclick="return doclink('link-10', '_FeatureSet', 'link-10');">_FeatureSet</a></tt> <tt class="py-keyword">import</tt> <tt id="link-11" class="py-name" targets="Class Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html"><a title="Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet" class="py-name" href="#" onclick="return doclink('link-11', 'FeatureSet', 'link-11');">FeatureSet</a></tt> </tt>
<a name="L29"></a><tt class="py-lineno">  29</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt><tt id="link-12" class="py-name" targets="Module Bio.Graphics.GenomeDiagram._GraphSet=Bio.Graphics.GenomeDiagram._GraphSet-module.html"><a title="Bio.Graphics.GenomeDiagram._GraphSet" class="py-name" href="#" onclick="return doclink('link-12', '_GraphSet', 'link-12');">_GraphSet</a></tt> <tt class="py-keyword">import</tt> <tt id="link-13" class="py-name" targets="Class Bio.Graphics.GenomeDiagram._GraphSet.GraphSet=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html"><a title="Bio.Graphics.GenomeDiagram._GraphSet.GraphSet" class="py-name" href="#" onclick="return doclink('link-13', 'GraphSet', 'link-13');">GraphSet</a></tt> </tt>
<a name="L30"></a><tt class="py-lineno">  30</tt>  <tt class="py-line"> </tt>
<a name="L31"></a><tt class="py-lineno">  31</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">math</tt> <tt class="py-keyword">import</tt> <tt id="link-14" class="py-name" targets="Method Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi()=Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint-class.html#pi"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-14', 'pi', 'link-14');">pi</a></tt><tt class="py-op">,</tt> <tt class="py-name">cos</tt><tt class="py-op">,</tt> <tt class="py-name">sin</tt> </tt>
<a name="L32"></a><tt class="py-lineno">  32</tt>  <tt class="py-line"> </tt>
<a name="L33"></a><tt class="py-lineno">  33</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer"></a><div id="CircularDrawer-def"><a name="L34"></a><tt class="py-lineno">  34</tt> <a class="py-toggle" href="#" id="CircularDrawer-toggle" onclick="return toggle('CircularDrawer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html">CircularDrawer</a><tt class="py-op">(</tt><tt class="py-base-class">AbstractDrawer</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="CircularDrawer-expanded"><a name="L35"></a><tt class="py-lineno">  35</tt>  <tt class="py-line">    <tt class="py-docstring">"""Object for drawing circular diagrams.</tt> </tt>
<a name="L36"></a><tt class="py-lineno">  36</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L37"></a><tt class="py-lineno">  37</tt>  <tt class="py-line"><tt class="py-docstring">    Attributes:</tt> </tt>
<a name="L38"></a><tt class="py-lineno">  38</tt>  <tt class="py-line"><tt class="py-docstring">     - tracklines    Boolean for whether to draw lines dilineating tracks</tt> </tt>
<a name="L39"></a><tt class="py-lineno">  39</tt>  <tt class="py-line"><tt class="py-docstring">     - pagesize      Tuple describing the size of the page in pixels</tt> </tt>
<a name="L40"></a><tt class="py-lineno">  40</tt>  <tt class="py-line"><tt class="py-docstring">     - x0            Float X co-ord for leftmost point of drawable area</tt> </tt>
<a name="L41"></a><tt class="py-lineno">  41</tt>  <tt class="py-line"><tt class="py-docstring">     - xlim          Float X co-ord for rightmost point of drawable area</tt> </tt>
<a name="L42"></a><tt class="py-lineno">  42</tt>  <tt class="py-line"><tt class="py-docstring">     - y0            Float Y co-ord for lowest point of drawable area</tt> </tt>
<a name="L43"></a><tt class="py-lineno">  43</tt>  <tt class="py-line"><tt class="py-docstring">     - ylim          Float Y co-ord for topmost point of drawable area</tt> </tt>
<a name="L44"></a><tt class="py-lineno">  44</tt>  <tt class="py-line"><tt class="py-docstring">     - pagewidth     Float pixel width of drawable area</tt> </tt>
<a name="L45"></a><tt class="py-lineno">  45</tt>  <tt class="py-line"><tt class="py-docstring">     - pageheight    Float pixel height of drawable area</tt> </tt>
<a name="L46"></a><tt class="py-lineno">  46</tt>  <tt class="py-line"><tt class="py-docstring">     - xcenter       Float X co-ord of center of drawable area</tt> </tt>
<a name="L47"></a><tt class="py-lineno">  47</tt>  <tt class="py-line"><tt class="py-docstring">     - ycenter       Float Y co-ord of center of drawable area</tt> </tt>
<a name="L48"></a><tt class="py-lineno">  48</tt>  <tt class="py-line"><tt class="py-docstring">     - start         Int, base to start drawing from</tt> </tt>
<a name="L49"></a><tt class="py-lineno">  49</tt>  <tt class="py-line"><tt class="py-docstring">     - end           Int, base to stop drawing at</tt> </tt>
<a name="L50"></a><tt class="py-lineno">  50</tt>  <tt class="py-line"><tt class="py-docstring">     - length        Size of sequence to be drawn</tt> </tt>
<a name="L51"></a><tt class="py-lineno">  51</tt>  <tt class="py-line"><tt class="py-docstring">     - track_size    Float (0-&gt;1) the proportion of the track height to draw in</tt> </tt>
<a name="L52"></a><tt class="py-lineno">  52</tt>  <tt class="py-line"><tt class="py-docstring">     - drawing       Drawing canvas</tt> </tt>
<a name="L53"></a><tt class="py-lineno">  53</tt>  <tt class="py-line"><tt class="py-docstring">     - drawn_tracks  List of ints denoting which tracks are to be drawn</tt> </tt>
<a name="L54"></a><tt class="py-lineno">  54</tt>  <tt class="py-line"><tt class="py-docstring">     - current_track_level   Int denoting which track is currently being drawn</tt> </tt>
<a name="L55"></a><tt class="py-lineno">  55</tt>  <tt class="py-line"><tt class="py-docstring">     - track_offsets     Dictionary of number of pixels that each track top,</tt> </tt>
<a name="L56"></a><tt class="py-lineno">  56</tt>  <tt class="py-line"><tt class="py-docstring">       center and bottom is offset from the base of a fragment, keyed by track</tt> </tt>
<a name="L57"></a><tt class="py-lineno">  57</tt>  <tt class="py-line"><tt class="py-docstring">     - sweep     Float (0-&gt;1) the proportion of the circle circumference to</tt> </tt>
<a name="L58"></a><tt class="py-lineno">  58</tt>  <tt class="py-line"><tt class="py-docstring">       use for the diagram</tt> </tt>
<a name="L59"></a><tt class="py-lineno">  59</tt>  <tt class="py-line"><tt class="py-docstring">     - cross_track_links List of tuples each with four entries (track A,</tt> </tt>
<a name="L60"></a><tt class="py-lineno">  60</tt>  <tt class="py-line"><tt class="py-docstring">       feature A, track B, feature B) to be linked.</tt> </tt>
<a name="L61"></a><tt class="py-lineno">  61</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L62"></a><tt class="py-lineno">  62</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L63"></a><tt class="py-lineno">  63</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer.__init__"></a><div id="CircularDrawer.__init__-def"><a name="L64"></a><tt class="py-lineno">  64</tt> <a class="py-toggle" href="#" id="CircularDrawer.__init__-toggle" onclick="return toggle('CircularDrawer.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">parent</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">pagesize</tt><tt class="py-op">=</tt><tt class="py-string">'A3'</tt><tt class="py-op">,</tt> <tt class="py-param">orientation</tt><tt class="py-op">=</tt><tt class="py-string">'landscape'</tt><tt class="py-op">,</tt> </tt>
<a name="L65"></a><tt class="py-lineno">  65</tt>  <tt class="py-line">                 <tt class="py-param">x</tt><tt class="py-op">=</tt><tt class="py-number">0.05</tt><tt class="py-op">,</tt> <tt class="py-param">y</tt><tt class="py-op">=</tt><tt class="py-number">0.05</tt><tt class="py-op">,</tt> <tt class="py-param">xl</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">xr</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">yt</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">yb</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L66"></a><tt class="py-lineno">  66</tt>  <tt class="py-line">                 <tt class="py-param">start</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">tracklines</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-param">track_size</tt><tt class="py-op">=</tt><tt class="py-number">0.75</tt><tt class="py-op">,</tt> </tt>
<a name="L67"></a><tt class="py-lineno">  67</tt>  <tt class="py-line">                 <tt class="py-param">circular</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-param">circle_core</tt><tt class="py-op">=</tt><tt class="py-number">0.0</tt><tt class="py-op">,</tt> <tt class="py-param">cross_track_links</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CircularDrawer.__init__-expanded"><a name="L68"></a><tt class="py-lineno">  68</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create CircularDrawer object.</tt> </tt>
<a name="L69"></a><tt class="py-lineno">  69</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L70"></a><tt class="py-lineno">  70</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L71"></a><tt class="py-lineno">  71</tt>  <tt class="py-line"><tt class="py-docstring">         - parent    Diagram object containing the data that the drawer</tt> </tt>
<a name="L72"></a><tt class="py-lineno">  72</tt>  <tt class="py-line"><tt class="py-docstring">           draws</tt> </tt>
<a name="L73"></a><tt class="py-lineno">  73</tt>  <tt class="py-line"><tt class="py-docstring">         - pagesize  String describing the ISO size of the image, or a tuple</tt> </tt>
<a name="L74"></a><tt class="py-lineno">  74</tt>  <tt class="py-line"><tt class="py-docstring">           of pixels</tt> </tt>
<a name="L75"></a><tt class="py-lineno">  75</tt>  <tt class="py-line"><tt class="py-docstring">         - orientation   String describing the required orientation of the</tt> </tt>
<a name="L76"></a><tt class="py-lineno">  76</tt>  <tt class="py-line"><tt class="py-docstring">           final drawing ('landscape' or 'portrait')</tt> </tt>
<a name="L77"></a><tt class="py-lineno">  77</tt>  <tt class="py-line"><tt class="py-docstring">         - x         Float (0-&gt;1) describing the relative size of the X</tt> </tt>
<a name="L78"></a><tt class="py-lineno">  78</tt>  <tt class="py-line"><tt class="py-docstring">           margins to the page</tt> </tt>
<a name="L79"></a><tt class="py-lineno">  79</tt>  <tt class="py-line"><tt class="py-docstring">         - y         Float (0-&gt;1) describing the relative size of the Y</tt> </tt>
<a name="L80"></a><tt class="py-lineno">  80</tt>  <tt class="py-line"><tt class="py-docstring">           margins to the page</tt> </tt>
<a name="L81"></a><tt class="py-lineno">  81</tt>  <tt class="py-line"><tt class="py-docstring">         - xl        Float (0-&gt;1) describing the relative size of the left X</tt> </tt>
<a name="L82"></a><tt class="py-lineno">  82</tt>  <tt class="py-line"><tt class="py-docstring">           margin to the page (overrides x)</tt> </tt>
<a name="L83"></a><tt class="py-lineno">  83</tt>  <tt class="py-line"><tt class="py-docstring">         - xl        Float (0-&gt;1) describing the relative size of the left X</tt> </tt>
<a name="L84"></a><tt class="py-lineno">  84</tt>  <tt class="py-line"><tt class="py-docstring">           margin to the page (overrides x)</tt> </tt>
<a name="L85"></a><tt class="py-lineno">  85</tt>  <tt class="py-line"><tt class="py-docstring">         - xr        Float (0-&gt;1) describing the relative size of the right X</tt> </tt>
<a name="L86"></a><tt class="py-lineno">  86</tt>  <tt class="py-line"><tt class="py-docstring">           margin to the page (overrides x)</tt> </tt>
<a name="L87"></a><tt class="py-lineno">  87</tt>  <tt class="py-line"><tt class="py-docstring">         - yt        Float (0-&gt;1) describing the relative size of the top Y</tt> </tt>
<a name="L88"></a><tt class="py-lineno">  88</tt>  <tt class="py-line"><tt class="py-docstring">           margin to the page (overrides y)</tt> </tt>
<a name="L89"></a><tt class="py-lineno">  89</tt>  <tt class="py-line"><tt class="py-docstring">         - yb        Float (0-&gt;1) describing the relative size of the lower Y</tt> </tt>
<a name="L90"></a><tt class="py-lineno">  90</tt>  <tt class="py-line"><tt class="py-docstring">           margin to the page (overrides y)</tt> </tt>
<a name="L91"></a><tt class="py-lineno">  91</tt>  <tt class="py-line"><tt class="py-docstring">         - start     Int, the position to begin drawing the diagram at</tt> </tt>
<a name="L92"></a><tt class="py-lineno">  92</tt>  <tt class="py-line"><tt class="py-docstring">         - end       Int, the position to stop drawing the diagram at</tt> </tt>
<a name="L93"></a><tt class="py-lineno">  93</tt>  <tt class="py-line"><tt class="py-docstring">         - tracklines    Boolean flag to show (or not) lines delineating tracks</tt> </tt>
<a name="L94"></a><tt class="py-lineno">  94</tt>  <tt class="py-line"><tt class="py-docstring">           on the diagram</tt> </tt>
<a name="L95"></a><tt class="py-lineno">  95</tt>  <tt class="py-line"><tt class="py-docstring">         - track_size    The proportion of the available track height that</tt> </tt>
<a name="L96"></a><tt class="py-lineno">  96</tt>  <tt class="py-line"><tt class="py-docstring">           should be taken up in drawing</tt> </tt>
<a name="L97"></a><tt class="py-lineno">  97</tt>  <tt class="py-line"><tt class="py-docstring">         - circular      Boolean flaw to show whether the passed sequence is</tt> </tt>
<a name="L98"></a><tt class="py-lineno">  98</tt>  <tt class="py-line"><tt class="py-docstring">           circular or not</tt> </tt>
<a name="L99"></a><tt class="py-lineno">  99</tt>  <tt class="py-line"><tt class="py-docstring">         - circle_core   The proportion of the available radius to leave</tt> </tt>
<a name="L100"></a><tt class="py-lineno"> 100</tt>  <tt class="py-line"><tt class="py-docstring">           empty at the center of a circular diagram (0 to 1).</tt> </tt>
<a name="L101"></a><tt class="py-lineno"> 101</tt>  <tt class="py-line"><tt class="py-docstring">         - cross_track_links List of tuples each with four entries (track A,</tt> </tt>
<a name="L102"></a><tt class="py-lineno"> 102</tt>  <tt class="py-line"><tt class="py-docstring">           feature A, track B, feature B) to be linked.</tt> </tt>
<a name="L103"></a><tt class="py-lineno"> 103</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L104"></a><tt class="py-lineno"> 104</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L105"></a><tt class="py-lineno"> 105</tt>  <tt class="py-line">        <tt class="py-comment"># Use the superclass' instantiation method</tt> </tt>
<a name="L106"></a><tt class="py-lineno"> 106</tt>  <tt class="py-line">        <tt id="link-15" class="py-name"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer" class="py-name" href="#" onclick="return doclink('link-15', 'AbstractDrawer', 'link-5');">AbstractDrawer</a></tt><tt class="py-op">.</tt><tt id="link-16" class="py-name" targets="Method Bio.Affy.CelFile.ParserError.__init__()=Bio.Affy.CelFile.ParserError-class.html#__init__,Method Bio.Affy.CelFile.Record.__init__()=Bio.Affy.CelFile.Record-class.html#__init__,Method Bio.Align.AlignInfo.PSSM.__init__()=Bio.Align.AlignInfo.PSSM-class.html#__init__,Method Bio.Align.AlignInfo.SummaryInfo.__init__()=Bio.Align.AlignInfo.SummaryInfo-class.html#__init__,Method Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline.__init__()=Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline-class.html#__init__,Method Bio.Align.Applications._Clustalw.ClustalwCommandline.__init__()=Bio.Align.Applications._Clustalw.ClustalwCommandline-class.html#__init__,Method Bio.Align.Applications._Dialign.DialignCommandline.__init__()=Bio.Align.Applications._Dialign.DialignCommandline-class.html#__init__,Method Bio.Align.Applications._MSAProbs.MSAProbsCommandline.__init__()=Bio.Align.Applications._MSAProbs.MSAProbsCommandline-class.html#__init__,Method Bio.Align.Applications._Mafft.MafftCommandline.__init__()=Bio.Align.Applications._Mafft.MafftCommandline-class.html#__init__,Method Bio.Align.Applications._Muscle.MuscleCommandline.__init__()=Bio.Align.Applications._Muscle.MuscleCommandline-class.html#__init__,Method Bio.Align.Applications._Prank.PrankCommandline.__init__()=Bio.Align.Applications._Prank.PrankCommandline-class.html#__init__,Method Bio.Align.Applications._Probcons.ProbconsCommandline.__init__()=Bio.Align.Applications._Probcons.ProbconsCommandline-class.html#__init__,Method Bio.Align.Applications._TCoffee.TCoffeeCommandline.__init__()=Bio.Align.Applications._TCoffee.TCoffeeCommandline-class.html#__init__,Method Bio.Align.MultipleSeqAlignment.__init__()=Bio.Align.MultipleSeqAlignment-class.html#__init__,Method Bio.Align.PairwiseAlignment.__init__()=Bio.Align.PairwiseAlignment-class.html#__init__,Method Bio.Align.PairwiseAlignments.__init__()=Bio.Align.PairwiseAlignments-class.html#__init__,Method Bio.AlignIO.Interfaces.AlignmentIterator.__init__()=Bio.AlignIO.Interfaces.AlignmentIterator-class.html#__init__,Method Bio.AlignIO.Interfaces.AlignmentWriter.__init__()=Bio.AlignIO.Interfaces.AlignmentWriter-class.html#__init__,Method Bio.AlignIO.Interfaces.SequentialAlignmentWriter.__init__()=Bio.AlignIO.Interfaces.SequentialAlignmentWriter-class.html#__init__,Method Bio.AlignIO.MafIO.MafIndex.__init__()=Bio.AlignIO.MafIO.MafIndex-class.html#__init__,Method Bio.AlignIO.MauveIO.MauveWriter.__init__()=Bio.AlignIO.MauveIO.MauveWriter-class.html#__init__,Method Bio.Alphabet.AlphabetEncoder.__init__()=Bio.Alphabet.AlphabetEncoder-class.html#__init__,Method Bio.Alphabet.Gapped.__init__()=Bio.Alphabet.Gapped-class.html#__init__,Method Bio.Alphabet.HasStopCodon.__init__()=Bio.Alphabet.HasStopCodon-class.html#__init__,Method Bio.Application.AbstractCommandline.__init__()=Bio.Application.AbstractCommandline-class.html#__init__,Method Bio.Application.ApplicationError.__init__()=Bio.Application.ApplicationError-class.html#__init__,Method Bio.Application._AbstractParameter.__init__()=Bio.Application._AbstractParameter-class.html#__init__,Method Bio.Application._Argument.__init__()=Bio.Application._Argument-class.html#__init__,Method Bio.Application._Option.__init__()=Bio.Application._Option-class.html#__init__,Method Bio.Application._StaticArgument.__init__()=Bio.Application._StaticArgument-class.html#__init__,Method Bio.Application._Switch.__init__()=Bio.Application._Switch-class.html#__init__,Method Bio.Blast.Applications.NcbiblastformatterCommandline.__init__()=Bio.Blast.Applications.NcbiblastformatterCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastnCommandline.__init__()=Bio.Blast.Applications.NcbiblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastpCommandline.__init__()=Bio.Blast.Applications.NcbiblastpCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastxCommandline.__init__()=Bio.Blast.Applications.NcbiblastxCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbideltablastCommandline.__init__()=Bio.Blast.Applications.NcbideltablastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbimakeblastdbCommandline.__init__()=Bio.Blast.Applications.NcbimakeblastdbCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbipsiblastCommandline.__init__()=Bio.Blast.Applications.NcbipsiblastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbirpsblastCommandline.__init__()=Bio.Blast.Applications.NcbirpsblastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbirpstblastnCommandline.__init__()=Bio.Blast.Applications.NcbirpstblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbitblastnCommandline.__init__()=Bio.Blast.Applications.NcbitblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbitblastxCommandline.__init__()=Bio.Blast.Applications.NcbitblastxCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbibaseblastCommandline.__init__()=Bio.Blast.Applications._NcbibaseblastCommandline-class.html#__init__,Method Bio.Blast.Applications._Ncbiblast2SeqCommandline.__init__()=Bio.Blast.Applications._Ncbiblast2SeqCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbiblastCommandline.__init__()=Bio.Blast.Applications._NcbiblastCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbiblastMain2SeqCommandline.__init__()=Bio.Blast.Applications._NcbiblastMain2SeqCommandline-class.html#__init__,Method Bio.Blast.NCBIXML.BlastParser.__init__()=Bio.Blast.NCBIXML.BlastParser-class.html#__init__,Method Bio.Blast.NCBIXML._XMLparser.__init__()=Bio.Blast.NCBIXML._XMLparser-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableEntry.__init__()=Bio.Blast.ParseBlastTable.BlastTableEntry-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableReader.__init__()=Bio.Blast.ParseBlastTable.BlastTableReader-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableRec.__init__()=Bio.Blast.ParseBlastTable.BlastTableRec-class.html#__init__,Method Bio.Blast.Record.Alignment.__init__()=Bio.Blast.Record.Alignment-class.html#__init__,Method Bio.Blast.Record.Blast.__init__()=Bio.Blast.Record.Blast-class.html#__init__,Method Bio.Blast.Record.DatabaseReport.__init__()=Bio.Blast.Record.DatabaseReport-class.html#__init__,Method Bio.Blast.Record.Description.__init__()=Bio.Blast.Record.Description-class.html#__init__,Method Bio.Blast.Record.DescriptionExt.__init__()=Bio.Blast.Record.DescriptionExt-class.html#__init__,Method Bio.Blast.Record.DescriptionExtItem.__init__()=Bio.Blast.Record.DescriptionExtItem-class.html#__init__,Method Bio.Blast.Record.HSP.__init__()=Bio.Blast.Record.HSP-class.html#__init__,Method Bio.Blast.Record.Header.__init__()=Bio.Blast.Record.Header-class.html#__init__,Method Bio.Blast.Record.MultipleAlignment.__init__()=Bio.Blast.Record.MultipleAlignment-class.html#__init__,Method Bio.Blast.Record.PSIBlast.__init__()=Bio.Blast.Record.PSIBlast-class.html#__init__,Method Bio.Blast.Record.Parameters.__init__()=Bio.Blast.Record.Parameters-class.html#__init__,Method Bio.Blast.Record.Round.__init__()=Bio.Blast.Record.Round-class.html#__init__,Method Bio.CAPS.CAPSMap.__init__()=Bio.CAPS.CAPSMap-class.html#__init__,Method Bio.CAPS.DifferentialCutsite.__init__()=Bio.CAPS.DifferentialCutsite-class.html#__init__,Method Bio.Cluster.Record.__init__()=Bio.Cluster.Record-class.html#__init__,Method Bio.Compass.Record.__init__()=Bio.Compass.Record-class.html#__init__,Method Bio.Crystal.Chain.__init__()=Bio.Crystal.Chain-class.html#__init__,Method Bio.Crystal.Crystal.__init__()=Bio.Crystal.Crystal-class.html#__init__,Method Bio.Crystal.Hetero.__init__()=Bio.Crystal.Hetero-class.html#__init__,Method Bio.Data.CodonTable.AmbiguousCodonTable.__init__()=Bio.Data.CodonTable.AmbiguousCodonTable-class.html#__init__,Method Bio.Data.CodonTable.AmbiguousForwardTable.__init__()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#__init__,Method Bio.Data.CodonTable.CodonTable.__init__()=Bio.Data.CodonTable.CodonTable-class.html#__init__,Method Bio.Data.CodonTable.NCBICodonTable.__init__()=Bio.Data.CodonTable.NCBICodonTable-class.html#__init__,Method Bio.Emboss.Applications.DiffseqCommandline.__init__()=Bio.Emboss.Applications.DiffseqCommandline-class.html#__init__,Method Bio.Emboss.Applications.EInvertedCommandline.__init__()=Bio.Emboss.Applications.EInvertedCommandline-class.html#__init__,Method Bio.Emboss.Applications.ETandemCommandline.__init__()=Bio.Emboss.Applications.ETandemCommandline-class.html#__init__,Method Bio.Emboss.Applications.Est2GenomeCommandline.__init__()=Bio.Emboss.Applications.Est2GenomeCommandline-class.html#__init__,Method Bio.Emboss.Applications.FConsenseCommandline.__init__()=Bio.Emboss.Applications.FConsenseCommandline-class.html#__init__,Method Bio.Emboss.Applications.FDNADistCommandline.__init__()=Bio.Emboss.Applications.FDNADistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FDNAParsCommandline.__init__()=Bio.Emboss.Applications.FDNAParsCommandline-class.html#__init__,Method Bio.Emboss.Applications.FNeighborCommandline.__init__()=Bio.Emboss.Applications.FNeighborCommandline-class.html#__init__,Method Bio.Emboss.Applications.FProtDistCommandline.__init__()=Bio.Emboss.Applications.FProtDistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FProtParsCommandline.__init__()=Bio.Emboss.Applications.FProtParsCommandline-class.html#__init__,Method Bio.Emboss.Applications.FSeqBootCommandline.__init__()=Bio.Emboss.Applications.FSeqBootCommandline-class.html#__init__,Method Bio.Emboss.Applications.FTreeDistCommandline.__init__()=Bio.Emboss.Applications.FTreeDistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FuzznucCommandline.__init__()=Bio.Emboss.Applications.FuzznucCommandline-class.html#__init__,Method Bio.Emboss.Applications.FuzzproCommandline.__init__()=Bio.Emboss.Applications.FuzzproCommandline-class.html#__init__,Method Bio.Emboss.Applications.IepCommandline.__init__()=Bio.Emboss.Applications.IepCommandline-class.html#__init__,Method Bio.Emboss.Applications.NeedleCommandline.__init__()=Bio.Emboss.Applications.NeedleCommandline-class.html#__init__,Method Bio.Emboss.Applications.NeedleallCommandline.__init__()=Bio.Emboss.Applications.NeedleallCommandline-class.html#__init__,Method Bio.Emboss.Applications.PalindromeCommandline.__init__()=Bio.Emboss.Applications.PalindromeCommandline-class.html#__init__,Method Bio.Emboss.Applications.Primer3Commandline.__init__()=Bio.Emboss.Applications.Primer3Commandline-class.html#__init__,Method Bio.Emboss.Applications.PrimerSearchCommandline.__init__()=Bio.Emboss.Applications.PrimerSearchCommandline-class.html#__init__,Method Bio.Emboss.Applications.SeqmatchallCommandline.__init__()=Bio.Emboss.Applications.SeqmatchallCommandline-class.html#__init__,Method Bio.Emboss.Applications.SeqretCommandline.__init__()=Bio.Emboss.Applications.SeqretCommandline-class.html#__init__,Method Bio.Emboss.Applications.StretcherCommandline.__init__()=Bio.Emboss.Applications.StretcherCommandline-class.html#__init__,Method Bio.Emboss.Applications.TranalignCommandline.__init__()=Bio.Emboss.Applications.TranalignCommandline-class.html#__init__,Method Bio.Emboss.Applications.WaterCommandline.__init__()=Bio.Emboss.Applications.WaterCommandline-class.html#__init__,Method Bio.Emboss.Applications._EmbossCommandLine.__init__()=Bio.Emboss.Applications._EmbossCommandLine-class.html#__init__,Method Bio.Emboss.Applications._EmbossMinimalCommandLine.__init__()=Bio.Emboss.Applications._EmbossMinimalCommandLine-class.html#__init__,Method Bio.Emboss.Primer3.Primers.__init__()=Bio.Emboss.Primer3.Primers-class.html#__init__,Method Bio.Emboss.Primer3.Record.__init__()=Bio.Emboss.Primer3.Record-class.html#__init__,Method Bio.Emboss.PrimerSearch.Amplifier.__init__()=Bio.Emboss.PrimerSearch.Amplifier-class.html#__init__,Method Bio.Emboss.PrimerSearch.InputRecord.__init__()=Bio.Emboss.PrimerSearch.InputRecord-class.html#__init__,Method Bio.Emboss.PrimerSearch.OutputRecord.__init__()=Bio.Emboss.PrimerSearch.OutputRecord-class.html#__init__,Method Bio.Entrez.Parser.CorruptedXMLError.__init__()=Bio.Entrez.Parser.CorruptedXMLError-class.html#__init__,Method Bio.Entrez.Parser.DataHandler.__init__()=Bio.Entrez.Parser.DataHandler-class.html#__init__,Method Bio.Entrez.Parser.DictionaryElement.__init__()=Bio.Entrez.Parser.DictionaryElement-class.html#__init__,Method Bio.Entrez.Parser.ListElement.__init__()=Bio.Entrez.Parser.ListElement-class.html#__init__,Method Bio.Entrez.Parser.NoneElement.__init__()=Bio.Entrez.Parser.NoneElement-class.html#__init__,Method Bio.Entrez.Parser.NotXMLError.__init__()=Bio.Entrez.Parser.NotXMLError-class.html#__init__,Method Bio.Entrez.Parser.ValidationError.__init__()=Bio.Entrez.Parser.ValidationError-class.html#__init__,Method Bio.ExPASy.Enzyme.Record.__init__()=Bio.ExPASy.Enzyme.Record-class.html#__init__,Method Bio.ExPASy.Prodoc.Record.__init__()=Bio.ExPASy.Prodoc.Record-class.html#__init__,Method Bio.ExPASy.Prodoc.Reference.__init__()=Bio.ExPASy.Prodoc.Reference-class.html#__init__,Method Bio.ExPASy.Prosite.Record.__init__()=Bio.ExPASy.Prosite.Record-class.html#__init__,Method Bio.ExPASy.ScanProsite.ContentHandler.__init__()=Bio.ExPASy.ScanProsite.ContentHandler-class.html#__init__,Method Bio.ExPASy.ScanProsite.Parser.__init__()=Bio.ExPASy.ScanProsite.Parser-class.html#__init__,Method Bio.ExPASy.ScanProsite.Record.__init__()=Bio.ExPASy.ScanProsite.Record-class.html#__init__,Method Bio.ExPASy.cellosaurus.Record.__init__()=Bio.ExPASy.cellosaurus.Record-class.html#__init__,Method Bio.FSSP.FSSPAlignDict.__init__()=Bio.FSSP.FSSPAlignDict-class.html#__init__,Method Bio.FSSP.FSSPAlignRec.__init__()=Bio.FSSP.FSSPAlignRec-class.html#__init__,Method Bio.FSSP.FSSPHeader.__init__()=Bio.FSSP.FSSPHeader-class.html#__init__,Method Bio.FSSP.FSSPSumRec.__init__()=Bio.FSSP.FSSPSumRec-class.html#__init__,Method Bio.FSSP.FSSPTools.FSSPMultAlign.__init__()=Bio.FSSP.FSSPTools.FSSPMultAlign-class.html#__init__,Method Bio.FSSP.PosAlign.__init__()=Bio.FSSP.PosAlign-class.html#__init__,Method Bio.FSSP.fssp_rec.fff_rec.__init__()=Bio.FSSP.fssp_rec.fff_rec-class.html#__init__,Method Bio.File.UndoHandle.__init__()=Bio.File.UndoHandle-class.html#__init__,Method Bio.File._IndexedSeqFileDict.__init__()=Bio.File._IndexedSeqFileDict-class.html#__init__,Method Bio.File._SQLiteManySeqFilesDict.__init__()=Bio.File._SQLiteManySeqFilesDict-class.html#__init__,Method Bio.GenBank.FeatureParser.__init__()=Bio.GenBank.FeatureParser-class.html#__init__,Method Bio.GenBank.Iterator.__init__()=Bio.GenBank.Iterator-class.html#__init__,Method Bio.GenBank.Record.Feature.__init__()=Bio.GenBank.Record.Feature-class.html#__init__,Method Bio.GenBank.Record.Qualifier.__init__()=Bio.GenBank.Record.Qualifier-class.html#__init__,Method Bio.GenBank.Record.Record.__init__()=Bio.GenBank.Record.Record-class.html#__init__,Method Bio.GenBank.Record.Reference.__init__()=Bio.GenBank.Record.Reference-class.html#__init__,Method Bio.GenBank.RecordParser.__init__()=Bio.GenBank.RecordParser-class.html#__init__,Method Bio.GenBank.Scanner.InsdcScanner.__init__()=Bio.GenBank.Scanner.InsdcScanner-class.html#__init__,Method Bio.GenBank._BaseGenBankConsumer.__init__()=Bio.GenBank._BaseGenBankConsumer-class.html#__init__,Method Bio.GenBank._FeatureConsumer.__init__()=Bio.GenBank._FeatureConsumer-class.html#__init__,Method Bio.GenBank._RecordConsumer.__init__()=Bio.GenBank._RecordConsumer-class.html#__init__,Method Bio.GenBank.utils.FeatureValueCleaner.__init__()=Bio.GenBank.utils.FeatureValueCleaner-class.html#__init__,Method Bio.Geo.Record.Record.__init__()=Bio.Geo.Record.Record-class.html#__init__,Method Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment.__init__()=Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome.Chromosome.__init__()=Bio.Graphics.BasicChromosome.Chromosome-class.html#__init__,Method Bio.Graphics.BasicChromosome.ChromosomeSegment.__init__()=Bio.Graphics.BasicChromosome.ChromosomeSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome.Organism.__init__()=Bio.Graphics.BasicChromosome.Organism-class.html#__init__,Method Bio.Graphics.BasicChromosome.TelomereSegment.__init__()=Bio.Graphics.BasicChromosome.TelomereSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome._ChromosomeComponent.__init__()=Bio.Graphics.BasicChromosome._ChromosomeComponent-class.html#__init__,Method Bio.Graphics.ColorSpiral.ColorSpiral.__init__()=Bio.Graphics.ColorSpiral.ColorSpiral-class.html#__init__,Method Bio.Graphics.Comparative.ComparativeScatterPlot.__init__()=Bio.Graphics.Comparative.ComparativeScatterPlot-class.html#__init__,Method Bio.Graphics.DisplayRepresentation.ChromosomeCounts.__init__()=Bio.Graphics.DisplayRepresentation.ChromosomeCounts-class.html#__init__,Method Bio.Graphics.Distribution.BarChartDistribution.__init__()=Bio.Graphics.Distribution.BarChartDistribution-class.html#__init__,Method Bio.Graphics.Distribution.DistributionPage.__init__()=Bio.Graphics.Distribution.DistributionPage-class.html#__init__,Method Bio.Graphics.Distribution.LineDistribution.__init__()=Bio.Graphics.Distribution.LineDistribution-class.html#__init__,Method Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.__init__()=Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.__init__()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.__init__()=Bio.Graphics.GenomeDiagram._Colors.ColorTranslator-class.html#__init__,Method Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.__init__()=Bio.Graphics.GenomeDiagram._CrossLink.CrossLink-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.__init__()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Feature.Feature.__init__()=Bio.Graphics.GenomeDiagram._Feature.Feature-class.html#__init__,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.__init__()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.__init__()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#__init__,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.__init__()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#__init__,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.__init__()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Track.Track.__init__()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#__init__,Method Bio.Graphics.KGML_vis.KGMLCanvas.__init__()=Bio.Graphics.KGML_vis.KGMLCanvas-class.html#__init__,Method Bio.HMM.DynamicProgramming.AbstractDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.AbstractDPAlgorithms-class.html#__init__,Method Bio.HMM.DynamicProgramming.LogDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.LogDPAlgorithms-class.html#__init__,Method Bio.HMM.DynamicProgramming.ScaledDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.ScaledDPAlgorithms-class.html#__init__,Method Bio.HMM.MarkovModel.HiddenMarkovModel.__init__()=Bio.HMM.MarkovModel.HiddenMarkovModel-class.html#__init__,Method Bio.HMM.MarkovModel.MarkovModelBuilder.__init__()=Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#__init__,Method Bio.HMM.Trainer.AbstractTrainer.__init__()=Bio.HMM.Trainer.AbstractTrainer-class.html#__init__,Method Bio.HMM.Trainer.BaumWelchTrainer.__init__()=Bio.HMM.Trainer.BaumWelchTrainer-class.html#__init__,Method Bio.HMM.Trainer.KnownStateTrainer.__init__()=Bio.HMM.Trainer.KnownStateTrainer-class.html#__init__,Method Bio.HMM.Trainer.TrainingSequence.__init__()=Bio.HMM.Trainer.TrainingSequence-class.html#__init__,Method Bio.Index._InMemoryIndex.__init__()=Bio.Index._InMemoryIndex-class.html#__init__,Method Bio.Index._ShelveIndex.__init__()=Bio.Index._ShelveIndex-class.html#__init__,Method Bio.KDTree.KDTree'.KDTree.__init__()=Bio.KDTree.KDTree%27.KDTree-class.html#__init__,Method Bio.KEGG.Compound.Record.__init__()=Bio.KEGG.Compound.Record-class.html#__init__,Method Bio.KEGG.Enzyme.Record.__init__()=Bio.KEGG.Enzyme.Record-class.html#__init__,Method Bio.KEGG.Gene.Record.__init__()=Bio.KEGG.Gene.Record-class.html#__init__,Method Bio.KEGG.KGML.KGML_parser.KGMLParser.__init__()=Bio.KEGG.KGML.KGML_parser.KGMLParser-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Component.__init__()=Bio.KEGG.KGML.KGML_pathway.Component-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Entry.__init__()=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Graphics.__init__()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Pathway.__init__()=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Reaction.__init__()=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Relation.__init__()=Bio.KEGG.KGML.KGML_pathway.Relation-class.html#__init__,Method Bio.LogisticRegression.LogisticRegression.__init__()=Bio.LogisticRegression.LogisticRegression-class.html#__init__,Method Bio.MarkovModel.MarkovModel.__init__()=Bio.MarkovModel.MarkovModel-class.html#__init__,Method Bio.MaxEntropy.MaxEntropy.__init__()=Bio.MaxEntropy.MaxEntropy-class.html#__init__,Method Bio.NMR.xpktools.Peaklist.__init__()=Bio.NMR.xpktools.Peaklist-class.html#__init__,Method Bio.NMR.xpktools.XpkEntry.__init__()=Bio.NMR.xpktools.XpkEntry-class.html#__init__,Method Bio.NaiveBayes.NaiveBayes.__init__()=Bio.NaiveBayes.NaiveBayes-class.html#__init__,Method Bio.Nexus.Nexus.Block.__init__()=Bio.Nexus.Nexus.Block-class.html#__init__,Method Bio.Nexus.Nexus.CharBuffer.__init__()=Bio.Nexus.Nexus.CharBuffer-class.html#__init__,Method Bio.Nexus.Nexus.Commandline.__init__()=Bio.Nexus.Nexus.Commandline-class.html#__init__,Method Bio.Nexus.Nexus.Nexus.__init__()=Bio.Nexus.Nexus.Nexus-class.html#__init__,Method Bio.Nexus.Nexus.StepMatrix.__init__()=Bio.Nexus.Nexus.StepMatrix-class.html#__init__,Method Bio.Nexus.Nodes.Chain.__init__()=Bio.Nexus.Nodes.Chain-class.html#__init__,Method Bio.Nexus.Nodes.Node.__init__()=Bio.Nexus.Nodes.Node-class.html#__init__,Method Bio.Nexus.StandardData.StandardData.__init__()=Bio.Nexus.StandardData.StandardData-class.html#__init__,Method Bio.Nexus.Trees.NodeData.__init__()=Bio.Nexus.Trees.NodeData-class.html#__init__,Method Bio.Nexus.Trees.Tree.__init__()=Bio.Nexus.Trees.Tree-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractPropertyMap-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap-class.html#__init__,Method Bio.PDB.Atom.Atom.__init__()=Bio.PDB.Atom.Atom-class.html#__init__,Method Bio.PDB.Atom.DisorderedAtom.__init__()=Bio.PDB.Atom.DisorderedAtom-class.html#__init__,Method Bio.PDB.Chain.Chain.__init__()=Bio.PDB.Chain.Chain-class.html#__init__,Method Bio.PDB.DSSP'.DSSP.__init__()=Bio.PDB.DSSP%27.DSSP-class.html#__init__,Method Bio.PDB.Dice.ChainSelector.__init__()=Bio.PDB.Dice.ChainSelector-class.html#__init__,Method Bio.PDB.Entity.DisorderedEntityWrapper.__init__()=Bio.PDB.Entity.DisorderedEntityWrapper-class.html#__init__,Method Bio.PDB.Entity.Entity.__init__()=Bio.PDB.Entity.Entity-class.html#__init__,Method Bio.PDB.FragmentMapper'.Fragment.__init__()=Bio.PDB.FragmentMapper%27.Fragment-class.html#__init__,Method Bio.PDB.FragmentMapper'.FragmentMapper.__init__()=Bio.PDB.FragmentMapper%27.FragmentMapper-class.html#__init__,Method Bio.PDB.HSExposure.ExposureCN.__init__()=Bio.PDB.HSExposure.ExposureCN-class.html#__init__,Method Bio.PDB.HSExposure.HSExposureCA.__init__()=Bio.PDB.HSExposure.HSExposureCA-class.html#__init__,Method Bio.PDB.HSExposure.HSExposureCB.__init__()=Bio.PDB.HSExposure.HSExposureCB-class.html#__init__,Method Bio.PDB.HSExposure._AbstractHSExposure.__init__()=Bio.PDB.HSExposure._AbstractHSExposure-class.html#__init__,Method Bio.PDB.MMCIF2Dict.MMCIF2Dict.__init__()=Bio.PDB.MMCIF2Dict.MMCIF2Dict-class.html#__init__,Method Bio.PDB.MMCIFParser'.FastMMCIFParser.__init__()=Bio.PDB.MMCIFParser%27.FastMMCIFParser-class.html#__init__,Method Bio.PDB.MMCIFParser'.MMCIFParser.__init__()=Bio.PDB.MMCIFParser%27.MMCIFParser-class.html#__init__,Method Bio.PDB.Model.Model.__init__()=Bio.PDB.Model.Model-class.html#__init__,Method Bio.PDB.NACCESS.NACCESS.__init__()=Bio.PDB.NACCESS.NACCESS-class.html#__init__,Method Bio.PDB.NACCESS.NACCESS_atomic.__init__()=Bio.PDB.NACCESS.NACCESS_atomic-class.html#__init__,Method Bio.PDB.NeighborSearch'.NeighborSearch.__init__()=Bio.PDB.NeighborSearch%27.NeighborSearch-class.html#__init__,Method Bio.PDB.PDBIO'.PDBIO.__init__()=Bio.PDB.PDBIO%27.PDBIO-class.html#__init__,Method Bio.PDB.PDBList'.PDBList.__init__()=Bio.PDB.PDBList%27.PDBList-class.html#__init__,Method Bio.PDB.PDBParser'.PDBParser.__init__()=Bio.PDB.PDBParser%27.PDBParser-class.html#__init__,Method Bio.PDB.PSEA.PSEA.__init__()=Bio.PDB.PSEA.PSEA-class.html#__init__,Method Bio.PDB.Polypeptide.CaPPBuilder.__init__()=Bio.PDB.Polypeptide.CaPPBuilder-class.html#__init__,Method Bio.PDB.Polypeptide.PPBuilder.__init__()=Bio.PDB.Polypeptide.PPBuilder-class.html#__init__,Method Bio.PDB.Polypeptide._PPBuilder.__init__()=Bio.PDB.Polypeptide._PPBuilder-class.html#__init__,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.__init__()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#__init__,Method Bio.PDB.Residue.DisorderedResidue.__init__()=Bio.PDB.Residue.DisorderedResidue-class.html#__init__,Method Bio.PDB.Residue.Residue.__init__()=Bio.PDB.Residue.Residue-class.html#__init__,Method Bio.PDB.ResidueDepth'.ResidueDepth.__init__()=Bio.PDB.ResidueDepth%27.ResidueDepth-class.html#__init__,Method Bio.PDB.Structure.Structure.__init__()=Bio.PDB.Structure.Structure-class.html#__init__,Method Bio.PDB.StructureAlignment'.StructureAlignment.__init__()=Bio.PDB.StructureAlignment%27.StructureAlignment-class.html#__init__,Method Bio.PDB.StructureBuilder.StructureBuilder.__init__()=Bio.PDB.StructureBuilder.StructureBuilder-class.html#__init__,Method Bio.PDB.Superimposer'.Superimposer.__init__()=Bio.PDB.Superimposer%27.Superimposer-class.html#__init__,Method Bio.PDB.mmcifio.MMCIFIO.__init__()=Bio.PDB.mmcifio.MMCIFIO-class.html#__init__,Method Bio.PDB.mmtf.DefaultParser.StructureDecoder.__init__()=Bio.PDB.mmtf.DefaultParser.StructureDecoder-class.html#__init__,Method Bio.PDB.vectors.Vector.__init__()=Bio.PDB.vectors.Vector-class.html#__init__,Method Bio.Pathway.Network.__init__()=Bio.Pathway.Network-class.html#__init__,Method Bio.Pathway.Reaction.__init__()=Bio.Pathway.Reaction-class.html#__init__,Method Bio.Pathway.Rep.Graph.Graph.__init__()=Bio.Pathway.Rep.Graph.Graph-class.html#__init__,Method Bio.Pathway.Rep.MultiGraph.MultiGraph.__init__()=Bio.Pathway.Rep.MultiGraph.MultiGraph-class.html#__init__,Method Bio.Pathway.System.__init__()=Bio.Pathway.System-class.html#__init__,Method Bio.Phylo.Applications._Fasttree.FastTreeCommandline.__init__()=Bio.Phylo.Applications._Fasttree.FastTreeCommandline-class.html#__init__,Method Bio.Phylo.Applications._Phyml.PhymlCommandline.__init__()=Bio.Phylo.Applications._Phyml.PhymlCommandline-class.html#__init__,Method Bio.Phylo.Applications._Raxml.RaxmlCommandline.__init__()=Bio.Phylo.Applications._Raxml.RaxmlCommandline-class.html#__init__,Method Bio.Phylo.BaseTree.BranchColor.__init__()=Bio.Phylo.BaseTree.BranchColor-class.html#__init__,Method Bio.Phylo.BaseTree.Clade.__init__()=Bio.Phylo.BaseTree.Clade-class.html#__init__,Method Bio.Phylo.BaseTree.Tree.__init__()=Bio.Phylo.BaseTree.Tree-class.html#__init__,Method Bio.Phylo.CDAO.Clade.__init__()=Bio.Phylo.CDAO.Clade-class.html#__init__,Method Bio.Phylo.CDAO.Tree.__init__()=Bio.Phylo.CDAO.Tree-class.html#__init__,Method Bio.Phylo.CDAOIO.Parser.__init__()=Bio.Phylo.CDAOIO.Parser-class.html#__init__,Method Bio.Phylo.CDAOIO.Writer.__init__()=Bio.Phylo.CDAOIO.Writer-class.html#__init__,Method Bio.Phylo.NeXML.Clade.__init__()=Bio.Phylo.NeXML.Clade-class.html#__init__,Method Bio.Phylo.NeXML.Tree.__init__()=Bio.Phylo.NeXML.Tree-class.html#__init__,Method Bio.Phylo.NeXMLIO.Parser.__init__()=Bio.Phylo.NeXMLIO.Parser-class.html#__init__,Method Bio.Phylo.NeXMLIO.Writer.__init__()=Bio.Phylo.NeXMLIO.Writer-class.html#__init__,Method Bio.Phylo.Newick.Clade.__init__()=Bio.Phylo.Newick.Clade-class.html#__init__,Method Bio.Phylo.Newick.Tree.__init__()=Bio.Phylo.Newick.Tree-class.html#__init__,Method Bio.Phylo.NewickIO.Parser.__init__()=Bio.Phylo.NewickIO.Parser-class.html#__init__,Method Bio.Phylo.NewickIO.Writer.__init__()=Bio.Phylo.NewickIO.Writer-class.html#__init__,Method Bio.Phylo.PAML._paml.Paml.__init__()=Bio.Phylo.PAML._paml.Paml-class.html#__init__,Method Bio.Phylo.PAML.baseml.Baseml.__init__()=Bio.Phylo.PAML.baseml.Baseml-class.html#__init__,Method Bio.Phylo.PAML.codeml.Codeml.__init__()=Bio.Phylo.PAML.codeml.Codeml-class.html#__init__,Method Bio.Phylo.PAML.yn00.Yn00.__init__()=Bio.Phylo.PAML.yn00.Yn00-class.html#__init__,Method Bio.Phylo.PhyloXML.Accession.__init__()=Bio.Phylo.PhyloXML.Accession-class.html#__init__,Method Bio.Phylo.PhyloXML.Annotation.__init__()=Bio.Phylo.PhyloXML.Annotation-class.html#__init__,Method Bio.Phylo.PhyloXML.BinaryCharacters.__init__()=Bio.Phylo.PhyloXML.BinaryCharacters-class.html#__init__,Method Bio.Phylo.PhyloXML.BranchColor.__init__()=Bio.Phylo.PhyloXML.BranchColor-class.html#__init__,Method Bio.Phylo.PhyloXML.Clade.__init__()=Bio.Phylo.PhyloXML.Clade-class.html#__init__,Method Bio.Phylo.PhyloXML.CladeRelation.__init__()=Bio.Phylo.PhyloXML.CladeRelation-class.html#__init__,Method Bio.Phylo.PhyloXML.Confidence.__init__()=Bio.Phylo.PhyloXML.Confidence-class.html#__init__,Method Bio.Phylo.PhyloXML.Date.__init__()=Bio.Phylo.PhyloXML.Date-class.html#__init__,Method Bio.Phylo.PhyloXML.Distribution.__init__()=Bio.Phylo.PhyloXML.Distribution-class.html#__init__,Method Bio.Phylo.PhyloXML.DomainArchitecture.__init__()=Bio.Phylo.PhyloXML.DomainArchitecture-class.html#__init__,Method Bio.Phylo.PhyloXML.Events.__init__()=Bio.Phylo.PhyloXML.Events-class.html#__init__,Method Bio.Phylo.PhyloXML.Id.__init__()=Bio.Phylo.PhyloXML.Id-class.html#__init__,Method Bio.Phylo.PhyloXML.MolSeq.__init__()=Bio.Phylo.PhyloXML.MolSeq-class.html#__init__,Method Bio.Phylo.PhyloXML.Other.__init__()=Bio.Phylo.PhyloXML.Other-class.html#__init__,Method Bio.Phylo.PhyloXML.Phylogeny.__init__()=Bio.Phylo.PhyloXML.Phylogeny-class.html#__init__,Method Bio.Phylo.PhyloXML.Phyloxml.__init__()=Bio.Phylo.PhyloXML.Phyloxml-class.html#__init__,Method Bio.Phylo.PhyloXML.Point.__init__()=Bio.Phylo.PhyloXML.Point-class.html#__init__,Method Bio.Phylo.PhyloXML.Polygon.__init__()=Bio.Phylo.PhyloXML.Polygon-class.html#__init__,Method Bio.Phylo.PhyloXML.Property.__init__()=Bio.Phylo.PhyloXML.Property-class.html#__init__,Method Bio.Phylo.PhyloXML.ProteinDomain.__init__()=Bio.Phylo.PhyloXML.ProteinDomain-class.html#__init__,Method Bio.Phylo.PhyloXML.Reference.__init__()=Bio.Phylo.PhyloXML.Reference-class.html#__init__,Method Bio.Phylo.PhyloXML.Sequence.__init__()=Bio.Phylo.PhyloXML.Sequence-class.html#__init__,Method Bio.Phylo.PhyloXML.SequenceRelation.__init__()=Bio.Phylo.PhyloXML.SequenceRelation-class.html#__init__,Method Bio.Phylo.PhyloXML.Taxonomy.__init__()=Bio.Phylo.PhyloXML.Taxonomy-class.html#__init__,Method Bio.Phylo.PhyloXML.Uri.__init__()=Bio.Phylo.PhyloXML.Uri-class.html#__init__,Method Bio.Phylo.PhyloXMLIO.Parser.__init__()=Bio.Phylo.PhyloXMLIO.Parser-class.html#__init__,Method Bio.Phylo.PhyloXMLIO.Writer.__init__()=Bio.Phylo.PhyloXMLIO.Writer-class.html#__init__,Method Bio.Phylo.TreeConstruction.DistanceCalculator.__init__()=Bio.Phylo.TreeConstruction.DistanceCalculator-class.html#__init__,Method Bio.Phylo.TreeConstruction.DistanceMatrix.__init__()=Bio.Phylo.TreeConstruction.DistanceMatrix-class.html#__init__,Method Bio.Phylo.TreeConstruction.DistanceTreeConstructor.__init__()=Bio.Phylo.TreeConstruction.DistanceTreeConstructor-class.html#__init__,Method Bio.Phylo.TreeConstruction.NNITreeSearcher.__init__()=Bio.Phylo.TreeConstruction.NNITreeSearcher-class.html#__init__,Method Bio.Phylo.TreeConstruction.ParsimonyScorer.__init__()=Bio.Phylo.TreeConstruction.ParsimonyScorer-class.html#__init__,Method Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor.__init__()=Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor-class.html#__init__,Method Bio.Phylo.TreeConstruction._Matrix.__init__()=Bio.Phylo.TreeConstruction._Matrix-class.html#__init__,Method Bio.PopGen.GenePop.Controller.GenePopController.__init__()=Bio.PopGen.GenePop.Controller.GenePopController-class.html#__init__,Method Bio.PopGen.GenePop.Controller._FileIterator.__init__()=Bio.PopGen.GenePop.Controller._FileIterator-class.html#__init__,Method Bio.PopGen.GenePop.Controller._GenePopCommandline.__init__()=Bio.PopGen.GenePop.Controller._GenePopCommandline-class.html#__init__,Method Bio.PopGen.GenePop.EasyController.EasyController.__init__()=Bio.PopGen.GenePop.EasyController.EasyController-class.html#__init__,Method Bio.PopGen.GenePop.FileParser.FileRecord.__init__()=Bio.PopGen.GenePop.FileParser.FileRecord-class.html#__init__,Method Bio.PopGen.GenePop.LargeFileParser.Record.__init__()=Bio.PopGen.GenePop.LargeFileParser.Record-class.html#__init__,Method Bio.PopGen.GenePop.Record.__init__()=Bio.PopGen.GenePop.Record-class.html#__init__,Method Bio.Restriction.PrintFormat.PrintFormat.__init__()=Bio.Restriction.PrintFormat.PrintFormat-class.html#__init__,Method Bio.Restriction.Restriction.Analysis.__init__()=Bio.Restriction.Restriction.Analysis-class.html#__init__,Method Bio.Restriction.Restriction.FormattedSeq.__init__()=Bio.Restriction.Restriction.FormattedSeq-class.html#__init__,Method Bio.Restriction.Restriction.RestrictionBatch.__init__()=Bio.Restriction.Restriction.RestrictionBatch-class.html#__init__,Method Bio.Restriction.Restriction.RestrictionType.__init__()=Bio.Restriction.Restriction.RestrictionType-class.html#__init__,Method Bio.SCOP.Astral.__init__()=Bio.SCOP.Astral-class.html#__init__,Method Bio.SCOP.Cla.Index.__init__()=Bio.SCOP.Cla.Index-class.html#__init__,Method Bio.SCOP.Cla.Record.__init__()=Bio.SCOP.Cla.Record-class.html#__init__,Method Bio.SCOP.Des.Record.__init__()=Bio.SCOP.Des.Record-class.html#__init__,Method Bio.SCOP.Dom.Record.__init__()=Bio.SCOP.Dom.Record-class.html#__init__,Method Bio.SCOP.Domain.__init__()=Bio.SCOP.Domain-class.html#__init__,Method Bio.SCOP.Hie.Record.__init__()=Bio.SCOP.Hie.Record-class.html#__init__,Method Bio.SCOP.Node.__init__()=Bio.SCOP.Node-class.html#__init__,Method Bio.SCOP.Raf.Res.__init__()=Bio.SCOP.Raf.Res-class.html#__init__,Method Bio.SCOP.Raf.SeqMap.__init__()=Bio.SCOP.Raf.SeqMap-class.html#__init__,Method Bio.SCOP.Raf.SeqMapIndex.__init__()=Bio.SCOP.Raf.SeqMapIndex-class.html#__init__,Method Bio.SCOP.Residues.Residues.__init__()=Bio.SCOP.Residues.Residues-class.html#__init__,Method Bio.SCOP.Scop.__init__()=Bio.SCOP.Scop-class.html#__init__,Method Bio.SVDSuperimposer.SVDSuperimposer.__init__()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabParser.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_text.BlastTextParser.__init__()=Bio.SearchIO.BlastIO.blast_text.BlastTextParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator.__init__()=Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslIndexer.__init__()=Bio.SearchIO.BlatIO.BlatPslIndexer-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslParser.__init__()=Bio.SearchIO.BlatIO.BlatPslParser-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslWriter.__init__()=Bio.SearchIO.BlatIO.BlatPslWriter-class.html#__init__,Method Bio.SearchIO.ExonerateIO._base._BaseExonerateParser.__init__()=Bio.SearchIO.ExonerateIO._base._BaseExonerateParser-class.html#__init__,Method Bio.SearchIO.FastaIO.FastaM10Indexer.__init__()=Bio.SearchIO.FastaIO.FastaM10Indexer-class.html#__init__,Method Bio.SearchIO.FastaIO.FastaM10Parser.__init__()=Bio.SearchIO.FastaIO.FastaM10Parser-class.html#__init__,Method Bio.SearchIO.HHsuiteIO.hhsuite2_text.Hhsuite2TextParser.__init__()=Bio.SearchIO.HHsuiteIO.hhsuite2_text.Hhsuite2TextParser-class.html#__init__,Method Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer.__init__()=Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser.__init__()=Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.__init__()=Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser.__init__()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.__init__()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser.__init__()=Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html#__init__,Method Bio.SearchIO.InterproscanIO.interproscan_xml.InterproscanXmlParser.__init__()=Bio.SearchIO.InterproscanIO.interproscan_xml.InterproscanXmlParser-class.html#__init__,Method Bio.SearchIO._index.SearchIndexer.__init__()=Bio.SearchIO._index.SearchIndexer-class.html#__init__,Method Bio.SearchIO._legacy.NCBIStandalone.BlastErrorParser.__init__()=Bio.SearchIO._legacy.NCBIStandalone.BlastErrorParser-class.html#__init__,Method Bio.SearchIO._legacy.NCBIStandalone.BlastParser.__init__()=Bio.SearchIO._legacy.NCBIStandalone.BlastParser-class.html#__init__,Method Bio.SearchIO._legacy.NCBIStandalone.Iterator.__init__()=Bio.SearchIO._legacy.NCBIStandalone.Iterator-class.html#__init__,Method Bio.SearchIO._legacy.NCBIStandalone.PSIBlastParser.__init__()=Bio.SearchIO._legacy.NCBIStandalone.PSIBlastParser-class.html#__init__,Method Bio.SearchIO._legacy.NCBIStandalone._BlastConsumer.__init__()=Bio.SearchIO._legacy.NCBIStandalone._BlastConsumer-class.html#__init__,Method Bio.SearchIO._legacy.NCBIStandalone._BlastErrorConsumer.__init__()=Bio.SearchIO._legacy.NCBIStandalone._BlastErrorConsumer-class.html#__init__,Method Bio.SearchIO._legacy.NCBIStandalone._PSIBlastConsumer.__init__()=Bio.SearchIO._legacy.NCBIStandalone._PSIBlastConsumer-class.html#__init__,Method Bio.SearchIO._legacy.NCBIStandalone._Scanner.__init__()=Bio.SearchIO._legacy.NCBIStandalone._Scanner-class.html#__init__,Method Bio.SearchIO._legacy.ParserSupport.TaggingConsumer.__init__()=Bio.SearchIO._legacy.ParserSupport.TaggingConsumer-class.html#__init__,Method Bio.SearchIO._model.hit.Hit.__init__()=Bio.SearchIO._model.hit.Hit-class.html#__init__,Method Bio.SearchIO._model.hsp.HSP.__init__()=Bio.SearchIO._model.hsp.HSP-class.html#__init__,Method Bio.SearchIO._model.hsp.HSPFragment.__init__()=Bio.SearchIO._model.hsp.HSPFragment-class.html#__init__,Method Bio.SearchIO._model.query.QueryResult.__init__()=Bio.SearchIO._model.query.QueryResult-class.html#__init__,Method Bio.Seq.MutableSeq.__init__()=Bio.Seq.MutableSeq-class.html#__init__,Method Bio.Seq.Seq.__init__()=Bio.Seq.Seq-class.html#__init__,Method Bio.Seq.UnknownSeq.__init__()=Bio.Seq.UnknownSeq-class.html#__init__,Method Bio.SeqFeature.CompoundLocation.__init__()=Bio.SeqFeature.CompoundLocation-class.html#__init__,Method Bio.SeqFeature.FeatureLocation.__init__()=Bio.SeqFeature.FeatureLocation-class.html#__init__,Method Bio.SeqFeature.PositionGap.__init__()=Bio.SeqFeature.PositionGap-class.html#__init__,Method Bio.SeqFeature.Reference.__init__()=Bio.SeqFeature.Reference-class.html#__init__,Method Bio.SeqFeature.SeqFeature.__init__()=Bio.SeqFeature.SeqFeature-class.html#__init__,Method Bio.SeqIO.FastaIO.FastaTwoLineWriter.__init__()=Bio.SeqIO.FastaIO.FastaTwoLineWriter-class.html#__init__,Method Bio.SeqIO.FastaIO.FastaWriter.__init__()=Bio.SeqIO.FastaIO.FastaWriter-class.html#__init__,Method Bio.SeqIO.Interfaces.SequenceIterator.__init__()=Bio.SeqIO.Interfaces.SequenceIterator-class.html#__init__,Method Bio.SeqIO.Interfaces.SequenceWriter.__init__()=Bio.SeqIO.Interfaces.SequenceWriter-class.html#__init__,Method Bio.SeqIO.Interfaces.SequentialSequenceWriter.__init__()=Bio.SeqIO.Interfaces.SequentialSequenceWriter-class.html#__init__,Method Bio.SeqIO.NibIO.NibWriter.__init__()=Bio.SeqIO.NibIO.NibWriter-class.html#__init__,Method Bio.SeqIO.PhdIO.PhdWriter.__init__()=Bio.SeqIO.PhdIO.PhdWriter-class.html#__init__,Method Bio.SeqIO.PirIO.PirWriter.__init__()=Bio.SeqIO.PirIO.PirWriter-class.html#__init__,Method Bio.SeqIO.QualityIO.QualPhredWriter.__init__()=Bio.SeqIO.QualityIO.QualPhredWriter-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.SeqXmlIterator.__init__()=Bio.SeqIO.SeqXmlIO.SeqXmlIterator-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.SeqXmlWriter.__init__()=Bio.SeqIO.SeqXmlIO.SeqXmlWriter-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.XMLRecordIterator.__init__()=Bio.SeqIO.SeqXmlIO.XMLRecordIterator-class.html#__init__,Method Bio.SeqIO.SffIO.SffWriter.__init__()=Bio.SeqIO.SffIO.SffWriter-class.html#__init__,Method Bio.SeqIO.SffIO._AddTellHandle.__init__()=Bio.SeqIO.SffIO._AddTellHandle-class.html#__init__,Method Bio.SeqIO.UniprotIO.Parser.__init__()=Bio.SeqIO.UniprotIO.Parser-class.html#__init__,Method Bio.SeqIO._index.IntelliGeneticsRandomAccess.__init__()=Bio.SeqIO._index.IntelliGeneticsRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SeqFileRandomAccess.__init__()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SequentialSeqFileRandomAccess.__init__()=Bio.SeqIO._index.SequentialSeqFileRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SffRandomAccess.__init__()=Bio.SeqIO._index.SffRandomAccess-class.html#__init__,Method Bio.SeqRecord.SeqRecord.__init__()=Bio.SeqRecord.SeqRecord-class.html#__init__,Method Bio.SeqRecord._RestrictedDict.__init__()=Bio.SeqRecord._RestrictedDict-class.html#__init__,Method Bio.SeqUtils.CodonUsage.CodonAdaptationIndex.__init__()=Bio.SeqUtils.CodonUsage.CodonAdaptationIndex-class.html#__init__,Method Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.__init__()=Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint-class.html#__init__,Method Bio.SeqUtils.ProtParam.ProteinAnalysis.__init__()=Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#__init__,Method Bio.Sequencing.Ace.ACEFileRecord.__init__()=Bio.Sequencing.Ace.ACEFileRecord-class.html#__init__,Method Bio.Sequencing.Ace.Contig.__init__()=Bio.Sequencing.Ace.Contig-class.html#__init__,Method Bio.Sequencing.Ace.Reads.__init__()=Bio.Sequencing.Ace.Reads-class.html#__init__,Method Bio.Sequencing.Ace.af.__init__()=Bio.Sequencing.Ace.af-class.html#__init__,Method Bio.Sequencing.Ace.bs.__init__()=Bio.Sequencing.Ace.bs-class.html#__init__,Method Bio.Sequencing.Ace.ct.__init__()=Bio.Sequencing.Ace.ct-class.html#__init__,Method Bio.Sequencing.Ace.ds.__init__()=Bio.Sequencing.Ace.ds-class.html#__init__,Method Bio.Sequencing.Ace.qa.__init__()=Bio.Sequencing.Ace.qa-class.html#__init__,Method Bio.Sequencing.Ace.rd.__init__()=Bio.Sequencing.Ace.rd-class.html#__init__,Method Bio.Sequencing.Ace.rt.__init__()=Bio.Sequencing.Ace.rt-class.html#__init__,Method Bio.Sequencing.Ace.wa.__init__()=Bio.Sequencing.Ace.wa-class.html#__init__,Method Bio.Sequencing.Ace.wr.__init__()=Bio.Sequencing.Ace.wr-class.html#__init__,Method Bio.Sequencing.Applications._Novoalign.NovoalignCommandline.__init__()=Bio.Sequencing.Applications._Novoalign.NovoalignCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaAlignCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaAlignCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaBwaswCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaBwaswCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaIndexCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaIndexCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaMemCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaMemCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaSampeCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaSampeCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaSamseCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaSamseCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsVersion0xSortCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsVersion0xSortCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsVersion1xSortCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsVersion1xSortCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline-class.html#__init__,Method Bio.Sequencing.Phd.Record.__init__()=Bio.Sequencing.Phd.Record-class.html#__init__,Method Bio.SubsMat.FreqTable.FreqTable.__init__()=Bio.SubsMat.FreqTable.FreqTable-class.html#__init__,Method Bio.SubsMat.SeqMat.__init__()=Bio.SubsMat.SeqMat-class.html#__init__,Method Bio.SwissProt.KeyWList.Record.__init__()=Bio.SwissProt.KeyWList.Record-class.html#__init__,Method Bio.SwissProt.Record.__init__()=Bio.SwissProt.Record-class.html#__init__,Method Bio.SwissProt.Reference.__init__()=Bio.SwissProt.Reference-class.html#__init__,Method Bio.UniGene.ProtsimLine.__init__()=Bio.UniGene.ProtsimLine-class.html#__init__,Method Bio.UniGene.Record.__init__()=Bio.UniGene.Record-class.html#__init__,Method Bio.UniGene.STSLine.__init__()=Bio.UniGene.STSLine-class.html#__init__,Method Bio.UniGene.SequenceLine.__init__()=Bio.UniGene.SequenceLine-class.html#__init__,Method Bio.Wise.dnal.Statistics.__init__()=Bio.Wise.dnal.Statistics-class.html#__init__,Method Bio.Wise.psw.AlignmentColumn.__init__()=Bio.Wise.psw.AlignmentColumn-class.html#__init__,Method Bio.Wise.psw.ColumnUnit.__init__()=Bio.Wise.psw.ColumnUnit-class.html#__init__,Method Bio.bgzf.BgzfReader.__init__()=Bio.bgzf.BgzfReader-class.html#__init__,Method Bio.bgzf.BgzfWriter.__init__()=Bio.bgzf.BgzfWriter-class.html#__init__,Method Bio.codonalign.codonalignment.CodonAlignment.__init__()=Bio.codonalign.codonalignment.CodonAlignment-class.html#__init__,Method Bio.codonalign.codonseq.CodonSeq.__init__()=Bio.codonalign.codonseq.CodonSeq-class.html#__init__,Method Bio.kNN.kNN.__init__()=Bio.kNN.kNN-class.html#__init__,Method Bio.motifs.Instances.__init__()=Bio.motifs.Instances-class.html#__init__,Method Bio.motifs.Motif.__init__()=Bio.motifs.Motif-class.html#__init__,Method Bio.motifs.alignace.Record.__init__()=Bio.motifs.alignace.Record-class.html#__init__,Method Bio.motifs.applications._xxmotif.XXmotifCommandline.__init__()=Bio.motifs.applications._xxmotif.XXmotifCommandline-class.html#__init__,Method Bio.motifs.jaspar.Motif.__init__()=Bio.motifs.jaspar.Motif-class.html#__init__,Method Bio.motifs.jaspar.Record.__init__()=Bio.motifs.jaspar.Record-class.html#__init__,Method Bio.motifs.jaspar.db.JASPAR5.__init__()=Bio.motifs.jaspar.db.JASPAR5-class.html#__init__,Method Bio.motifs.mast.Record.__init__()=Bio.motifs.mast.Record-class.html#__init__,Method Bio.motifs.matrix.GenericPositionMatrix.__init__()=Bio.motifs.matrix.GenericPositionMatrix-class.html#__init__,Method Bio.motifs.matrix.PositionWeightMatrix.__init__()=Bio.motifs.matrix.PositionWeightMatrix-class.html#__init__,Method Bio.motifs.meme.Instance.__init__()=Bio.motifs.meme.Instance-class.html#__init__,Method Bio.motifs.meme.Motif.__init__()=Bio.motifs.meme.Motif-class.html#__init__,Method Bio.motifs.meme.Record.__init__()=Bio.motifs.meme.Record-class.html#__init__,Method Bio.motifs.minimal.Record.__init__()=Bio.motifs.minimal.Record-class.html#__init__,Method Bio.motifs.thresholds.ScoreDistribution.__init__()=Bio.motifs.thresholds.ScoreDistribution-class.html#__init__,Method Bio.motifs.transfac.Record.__init__()=Bio.motifs.transfac.Record-class.html#__init__,Method Bio.motifs.xms.XMSScanner.__init__()=Bio.motifs.xms.XMSScanner-class.html#__init__,Method Bio.pairwise2.affine_penalty.__init__()=Bio.pairwise2.affine_penalty-class.html#__init__,Method Bio.pairwise2.dictionary_match.__init__()=Bio.pairwise2.dictionary_match-class.html#__init__,Method Bio.pairwise2.identity_match.__init__()=Bio.pairwise2.identity_match-class.html#__init__,Method Bio.phenotype.phen_micro.JsonWriter.__init__()=Bio.phenotype.phen_micro.JsonWriter-class.html#__init__,Method Bio.phenotype.phen_micro.PlateRecord.__init__()=Bio.phenotype.phen_micro.PlateRecord-class.html#__init__,Method Bio.phenotype.phen_micro.WellRecord.__init__()=Bio.phenotype.phen_micro.WellRecord-class.html#__init__,Method BioSQL.BioSeq.DBSeq.__init__()=BioSQL.BioSeq.DBSeq-class.html#__init__,Method BioSQL.BioSeq.DBSeqRecord.__init__()=BioSQL.BioSeq.DBSeqRecord-class.html#__init__,Method BioSQL.BioSeqDatabase.Adaptor.__init__()=BioSQL.BioSeqDatabase.Adaptor-class.html#__init__,Method BioSQL.BioSeqDatabase.BioSeqDatabase.__init__()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#__init__,Method BioSQL.BioSeqDatabase.DBServer.__init__()=BioSQL.BioSeqDatabase.DBServer-class.html#__init__,Method BioSQL.BioSeqDatabase._CursorWrapper.__init__()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#__init__,Method BioSQL.DBUtils.Generic_dbutils.__init__()=BioSQL.DBUtils.Generic_dbutils-class.html#__init__,Method BioSQL.Loader.DatabaseLoader.__init__()=BioSQL.Loader.DatabaseLoader-class.html#__init__,Method BioSQL.Loader.DatabaseRemover.__init__()=BioSQL.Loader.DatabaseRemover-class.html#__init__"><a title="Bio.Affy.CelFile.ParserError.__init__
Bio.Affy.CelFile.Record.__init__
Bio.Align.AlignInfo.PSSM.__init__
Bio.Align.AlignInfo.SummaryInfo.__init__
Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline.__init__
Bio.Align.Applications._Clustalw.ClustalwCommandline.__init__
Bio.Align.Applications._Dialign.DialignCommandline.__init__
Bio.Align.Applications._MSAProbs.MSAProbsCommandline.__init__
Bio.Align.Applications._Mafft.MafftCommandline.__init__
Bio.Align.Applications._Muscle.MuscleCommandline.__init__
Bio.Align.Applications._Prank.PrankCommandline.__init__
Bio.Align.Applications._Probcons.ProbconsCommandline.__init__
Bio.Align.Applications._TCoffee.TCoffeeCommandline.__init__
Bio.Align.MultipleSeqAlignment.__init__
Bio.Align.PairwiseAlignment.__init__
Bio.Align.PairwiseAlignments.__init__
Bio.AlignIO.Interfaces.AlignmentIterator.__init__
Bio.AlignIO.Interfaces.AlignmentWriter.__init__
Bio.AlignIO.Interfaces.SequentialAlignmentWriter.__init__
Bio.AlignIO.MafIO.MafIndex.__init__
Bio.AlignIO.MauveIO.MauveWriter.__init__
Bio.Alphabet.AlphabetEncoder.__init__
Bio.Alphabet.Gapped.__init__
Bio.Alphabet.HasStopCodon.__init__
Bio.Application.AbstractCommandline.__init__
Bio.Application.ApplicationError.__init__
Bio.Application._AbstractParameter.__init__
Bio.Application._Argument.__init__
Bio.Application._Option.__init__
Bio.Application._StaticArgument.__init__
Bio.Application._Switch.__init__
Bio.Blast.Applications.NcbiblastformatterCommandline.__init__
Bio.Blast.Applications.NcbiblastnCommandline.__init__
Bio.Blast.Applications.NcbiblastpCommandline.__init__
Bio.Blast.Applications.NcbiblastxCommandline.__init__
Bio.Blast.Applications.NcbideltablastCommandline.__init__
Bio.Blast.Applications.NcbimakeblastdbCommandline.__init__
Bio.Blast.Applications.NcbipsiblastCommandline.__init__
Bio.Blast.Applications.NcbirpsblastCommandline.__init__
Bio.Blast.Applications.NcbirpstblastnCommandline.__init__
Bio.Blast.Applications.NcbitblastnCommandline.__init__
Bio.Blast.Applications.NcbitblastxCommandline.__init__
Bio.Blast.Applications._NcbibaseblastCommandline.__init__
Bio.Blast.Applications._Ncbiblast2SeqCommandline.__init__
Bio.Blast.Applications._NcbiblastCommandline.__init__
Bio.Blast.Applications._NcbiblastMain2SeqCommandline.__init__
Bio.Blast.NCBIXML.BlastParser.__init__
Bio.Blast.NCBIXML._XMLparser.__init__
Bio.Blast.ParseBlastTable.BlastTableEntry.__init__
Bio.Blast.ParseBlastTable.BlastTableReader.__init__
Bio.Blast.ParseBlastTable.BlastTableRec.__init__
Bio.Blast.Record.Alignment.__init__
Bio.Blast.Record.Blast.__init__
Bio.Blast.Record.DatabaseReport.__init__
Bio.Blast.Record.Description.__init__
Bio.Blast.Record.DescriptionExt.__init__
Bio.Blast.Record.DescriptionExtItem.__init__
Bio.Blast.Record.HSP.__init__
Bio.Blast.Record.Header.__init__
Bio.Blast.Record.MultipleAlignment.__init__
Bio.Blast.Record.PSIBlast.__init__
Bio.Blast.Record.Parameters.__init__
Bio.Blast.Record.Round.__init__
Bio.CAPS.CAPSMap.__init__
Bio.CAPS.DifferentialCutsite.__init__
Bio.Cluster.Record.__init__
Bio.Compass.Record.__init__
Bio.Crystal.Chain.__init__
Bio.Crystal.Crystal.__init__
Bio.Crystal.Hetero.__init__
Bio.Data.CodonTable.AmbiguousCodonTable.__init__
Bio.Data.CodonTable.AmbiguousForwardTable.__init__
Bio.Data.CodonTable.CodonTable.__init__
Bio.Data.CodonTable.NCBICodonTable.__init__
Bio.Emboss.Applications.DiffseqCommandline.__init__
Bio.Emboss.Applications.EInvertedCommandline.__init__
Bio.Emboss.Applications.ETandemCommandline.__init__
Bio.Emboss.Applications.Est2GenomeCommandline.__init__
Bio.Emboss.Applications.FConsenseCommandline.__init__
Bio.Emboss.Applications.FDNADistCommandline.__init__
Bio.Emboss.Applications.FDNAParsCommandline.__init__
Bio.Emboss.Applications.FNeighborCommandline.__init__
Bio.Emboss.Applications.FProtDistCommandline.__init__
Bio.Emboss.Applications.FProtParsCommandline.__init__
Bio.Emboss.Applications.FSeqBootCommandline.__init__
Bio.Emboss.Applications.FTreeDistCommandline.__init__
Bio.Emboss.Applications.FuzznucCommandline.__init__
Bio.Emboss.Applications.FuzzproCommandline.__init__
Bio.Emboss.Applications.IepCommandline.__init__
Bio.Emboss.Applications.NeedleCommandline.__init__
Bio.Emboss.Applications.NeedleallCommandline.__init__
Bio.Emboss.Applications.PalindromeCommandline.__init__
Bio.Emboss.Applications.Primer3Commandline.__init__
Bio.Emboss.Applications.PrimerSearchCommandline.__init__
Bio.Emboss.Applications.SeqmatchallCommandline.__init__
Bio.Emboss.Applications.SeqretCommandline.__init__
Bio.Emboss.Applications.StretcherCommandline.__init__
Bio.Emboss.Applications.TranalignCommandline.__init__
Bio.Emboss.Applications.WaterCommandline.__init__
Bio.Emboss.Applications._EmbossCommandLine.__init__
Bio.Emboss.Applications._EmbossMinimalCommandLine.__init__
Bio.Emboss.Primer3.Primers.__init__
Bio.Emboss.Primer3.Record.__init__
Bio.Emboss.PrimerSearch.Amplifier.__init__
Bio.Emboss.PrimerSearch.InputRecord.__init__
Bio.Emboss.PrimerSearch.OutputRecord.__init__
Bio.Entrez.Parser.CorruptedXMLError.__init__
Bio.Entrez.Parser.DataHandler.__init__
Bio.Entrez.Parser.DictionaryElement.__init__
Bio.Entrez.Parser.ListElement.__init__
Bio.Entrez.Parser.NoneElement.__init__
Bio.Entrez.Parser.NotXMLError.__init__
Bio.Entrez.Parser.ValidationError.__init__
Bio.ExPASy.Enzyme.Record.__init__
Bio.ExPASy.Prodoc.Record.__init__
Bio.ExPASy.Prodoc.Reference.__init__
Bio.ExPASy.Prosite.Record.__init__
Bio.ExPASy.ScanProsite.ContentHandler.__init__
Bio.ExPASy.ScanProsite.Parser.__init__
Bio.ExPASy.ScanProsite.Record.__init__
Bio.ExPASy.cellosaurus.Record.__init__
Bio.FSSP.FSSPAlignDict.__init__
Bio.FSSP.FSSPAlignRec.__init__
Bio.FSSP.FSSPHeader.__init__
Bio.FSSP.FSSPSumRec.__init__
Bio.FSSP.FSSPTools.FSSPMultAlign.__init__
Bio.FSSP.PosAlign.__init__
Bio.FSSP.fssp_rec.fff_rec.__init__
Bio.File.UndoHandle.__init__
Bio.File._IndexedSeqFileDict.__init__
Bio.File._SQLiteManySeqFilesDict.__init__
Bio.GenBank.FeatureParser.__init__
Bio.GenBank.Iterator.__init__
Bio.GenBank.Record.Feature.__init__
Bio.GenBank.Record.Qualifier.__init__
Bio.GenBank.Record.Record.__init__
Bio.GenBank.Record.Reference.__init__
Bio.GenBank.RecordParser.__init__
Bio.GenBank.Scanner.InsdcScanner.__init__
Bio.GenBank._BaseGenBankConsumer.__init__
Bio.GenBank._FeatureConsumer.__init__
Bio.GenBank._RecordConsumer.__init__
Bio.GenBank.utils.FeatureValueCleaner.__init__
Bio.Geo.Record.Record.__init__
Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment.__init__
Bio.Graphics.BasicChromosome.Chromosome.__init__
Bio.Graphics.BasicChromosome.ChromosomeSegment.__init__
Bio.Graphics.BasicChromosome.Organism.__init__
Bio.Graphics.BasicChromosome.TelomereSegment.__init__
Bio.Graphics.BasicChromosome._ChromosomeComponent.__init__
Bio.Graphics.ColorSpiral.ColorSpiral.__init__
Bio.Graphics.Comparative.ComparativeScatterPlot.__init__
Bio.Graphics.DisplayRepresentation.ChromosomeCounts.__init__
Bio.Graphics.Distribution.BarChartDistribution.__init__
Bio.Graphics.Distribution.DistributionPage.__init__
Bio.Graphics.Distribution.LineDistribution.__init__
Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.__init__
Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.__init__
Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.__init__
Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.__init__
Bio.Graphics.GenomeDiagram._Diagram.Diagram.__init__
Bio.Graphics.GenomeDiagram._Feature.Feature.__init__
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.__init__
Bio.Graphics.GenomeDiagram._Graph.GraphData.__init__
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.__init__
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.__init__
Bio.Graphics.GenomeDiagram._Track.Track.__init__
Bio.Graphics.KGML_vis.KGMLCanvas.__init__
Bio.HMM.DynamicProgramming.AbstractDPAlgorithms.__init__
Bio.HMM.DynamicProgramming.LogDPAlgorithms.__init__
Bio.HMM.DynamicProgramming.ScaledDPAlgorithms.__init__
Bio.HMM.MarkovModel.HiddenMarkovModel.__init__
Bio.HMM.MarkovModel.MarkovModelBuilder.__init__
Bio.HMM.Trainer.AbstractTrainer.__init__
Bio.HMM.Trainer.BaumWelchTrainer.__init__
Bio.HMM.Trainer.KnownStateTrainer.__init__
Bio.HMM.Trainer.TrainingSequence.__init__
Bio.Index._InMemoryIndex.__init__
Bio.Index._ShelveIndex.__init__
Bio.KDTree.KDTree'.KDTree.__init__
Bio.KEGG.Compound.Record.__init__
Bio.KEGG.Enzyme.Record.__init__
Bio.KEGG.Gene.Record.__init__
Bio.KEGG.KGML.KGML_parser.KGMLParser.__init__
Bio.KEGG.KGML.KGML_pathway.Component.__init__
Bio.KEGG.KGML.KGML_pathway.Entry.__init__
Bio.KEGG.KGML.KGML_pathway.Graphics.__init__
Bio.KEGG.KGML.KGML_pathway.Pathway.__init__
Bio.KEGG.KGML.KGML_pathway.Reaction.__init__
Bio.KEGG.KGML.KGML_pathway.Relation.__init__
Bio.LogisticRegression.LogisticRegression.__init__
Bio.MarkovModel.MarkovModel.__init__
Bio.MaxEntropy.MaxEntropy.__init__
Bio.NMR.xpktools.Peaklist.__init__
Bio.NMR.xpktools.XpkEntry.__init__
Bio.NaiveBayes.NaiveBayes.__init__
Bio.Nexus.Nexus.Block.__init__
Bio.Nexus.Nexus.CharBuffer.__init__
Bio.Nexus.Nexus.Commandline.__init__
Bio.Nexus.Nexus.Nexus.__init__
Bio.Nexus.Nexus.StepMatrix.__init__
Bio.Nexus.Nodes.Chain.__init__
Bio.Nexus.Nodes.Node.__init__
Bio.Nexus.StandardData.StandardData.__init__
Bio.Nexus.Trees.NodeData.__init__
Bio.Nexus.Trees.Tree.__init__
Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap.__init__
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__init__
Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap.__init__
Bio.PDB.Atom.Atom.__init__
Bio.PDB.Atom.DisorderedAtom.__init__
Bio.PDB.Chain.Chain.__init__
Bio.PDB.DSSP'.DSSP.__init__
Bio.PDB.Dice.ChainSelector.__init__
Bio.PDB.Entity.DisorderedEntityWrapper.__init__
Bio.PDB.Entity.Entity.__init__
Bio.PDB.FragmentMapper'.Fragment.__init__
Bio.PDB.FragmentMapper'.FragmentMapper.__init__
Bio.PDB.HSExposure.ExposureCN.__init__
Bio.PDB.HSExposure.HSExposureCA.__init__
Bio.PDB.HSExposure.HSExposureCB.__init__
Bio.PDB.HSExposure._AbstractHSExposure.__init__
Bio.PDB.MMCIF2Dict.MMCIF2Dict.__init__
Bio.PDB.MMCIFParser'.FastMMCIFParser.__init__
Bio.PDB.MMCIFParser'.MMCIFParser.__init__
Bio.PDB.Model.Model.__init__
Bio.PDB.NACCESS.NACCESS.__init__
Bio.PDB.NACCESS.NACCESS_atomic.__init__
Bio.PDB.NeighborSearch'.NeighborSearch.__init__
Bio.PDB.PDBIO'.PDBIO.__init__
Bio.PDB.PDBList'.PDBList.__init__
Bio.PDB.PDBParser'.PDBParser.__init__
Bio.PDB.PSEA.PSEA.__init__
Bio.PDB.Polypeptide.CaPPBuilder.__init__
Bio.PDB.Polypeptide.PPBuilder.__init__
Bio.PDB.Polypeptide._PPBuilder.__init__
Bio.PDB.QCPSuperimposer.QCPSuperimposer.__init__
Bio.PDB.Residue.DisorderedResidue.__init__
Bio.PDB.Residue.Residue.__init__
Bio.PDB.ResidueDepth'.ResidueDepth.__init__
Bio.PDB.Structure.Structure.__init__
Bio.PDB.StructureAlignment'.StructureAlignment.__init__
Bio.PDB.StructureBuilder.StructureBuilder.__init__
Bio.PDB.Superimposer'.Superimposer.__init__
Bio.PDB.mmcifio.MMCIFIO.__init__
Bio.PDB.mmtf.DefaultParser.StructureDecoder.__init__
Bio.PDB.vectors.Vector.__init__
Bio.Pathway.Network.__init__
Bio.Pathway.Reaction.__init__
Bio.Pathway.Rep.Graph.Graph.__init__
Bio.Pathway.Rep.MultiGraph.MultiGraph.__init__
Bio.Pathway.System.__init__
Bio.Phylo.Applications._Fasttree.FastTreeCommandline.__init__
Bio.Phylo.Applications._Phyml.PhymlCommandline.__init__
Bio.Phylo.Applications._Raxml.RaxmlCommandline.__init__
Bio.Phylo.BaseTree.BranchColor.__init__
Bio.Phylo.BaseTree.Clade.__init__
Bio.Phylo.BaseTree.Tree.__init__
Bio.Phylo.CDAO.Clade.__init__
Bio.Phylo.CDAO.Tree.__init__
Bio.Phylo.CDAOIO.Parser.__init__
Bio.Phylo.CDAOIO.Writer.__init__
Bio.Phylo.NeXML.Clade.__init__
Bio.Phylo.NeXML.Tree.__init__
Bio.Phylo.NeXMLIO.Parser.__init__
Bio.Phylo.NeXMLIO.Writer.__init__
Bio.Phylo.Newick.Clade.__init__
Bio.Phylo.Newick.Tree.__init__
Bio.Phylo.NewickIO.Parser.__init__
Bio.Phylo.NewickIO.Writer.__init__
Bio.Phylo.PAML._paml.Paml.__init__
Bio.Phylo.PAML.baseml.Baseml.__init__
Bio.Phylo.PAML.codeml.Codeml.__init__
Bio.Phylo.PAML.yn00.Yn00.__init__
Bio.Phylo.PhyloXML.Accession.__init__
Bio.Phylo.PhyloXML.Annotation.__init__
Bio.Phylo.PhyloXML.BinaryCharacters.__init__
Bio.Phylo.PhyloXML.BranchColor.__init__
Bio.Phylo.PhyloXML.Clade.__init__
Bio.Phylo.PhyloXML.CladeRelation.__init__
Bio.Phylo.PhyloXML.Confidence.__init__
Bio.Phylo.PhyloXML.Date.__init__
Bio.Phylo.PhyloXML.Distribution.__init__
Bio.Phylo.PhyloXML.DomainArchitecture.__init__
Bio.Phylo.PhyloXML.Events.__init__
Bio.Phylo.PhyloXML.Id.__init__
Bio.Phylo.PhyloXML.MolSeq.__init__
Bio.Phylo.PhyloXML.Other.__init__
Bio.Phylo.PhyloXML.Phylogeny.__init__
Bio.Phylo.PhyloXML.Phyloxml.__init__
Bio.Phylo.PhyloXML.Point.__init__
Bio.Phylo.PhyloXML.Polygon.__init__
Bio.Phylo.PhyloXML.Property.__init__
Bio.Phylo.PhyloXML.ProteinDomain.__init__
Bio.Phylo.PhyloXML.Reference.__init__
Bio.Phylo.PhyloXML.Sequence.__init__
Bio.Phylo.PhyloXML.SequenceRelation.__init__
Bio.Phylo.PhyloXML.Taxonomy.__init__
Bio.Phylo.PhyloXML.Uri.__init__
Bio.Phylo.PhyloXMLIO.Parser.__init__
Bio.Phylo.PhyloXMLIO.Writer.__init__
Bio.Phylo.TreeConstruction.DistanceCalculator.__init__
Bio.Phylo.TreeConstruction.DistanceMatrix.__init__
Bio.Phylo.TreeConstruction.DistanceTreeConstructor.__init__
Bio.Phylo.TreeConstruction.NNITreeSearcher.__init__
Bio.Phylo.TreeConstruction.ParsimonyScorer.__init__
Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor.__init__
Bio.Phylo.TreeConstruction._Matrix.__init__
Bio.PopGen.GenePop.Controller.GenePopController.__init__
Bio.PopGen.GenePop.Controller._FileIterator.__init__
Bio.PopGen.GenePop.Controller._GenePopCommandline.__init__
Bio.PopGen.GenePop.EasyController.EasyController.__init__
Bio.PopGen.GenePop.FileParser.FileRecord.__init__
Bio.PopGen.GenePop.LargeFileParser.Record.__init__
Bio.PopGen.GenePop.Record.__init__
Bio.Restriction.PrintFormat.PrintFormat.__init__
Bio.Restriction.Restriction.Analysis.__init__
Bio.Restriction.Restriction.FormattedSeq.__init__
Bio.Restriction.Restriction.RestrictionBatch.__init__
Bio.Restriction.Restriction.RestrictionType.__init__
Bio.SCOP.Astral.__init__
Bio.SCOP.Cla.Index.__init__
Bio.SCOP.Cla.Record.__init__
Bio.SCOP.Des.Record.__init__
Bio.SCOP.Dom.Record.__init__
Bio.SCOP.Domain.__init__
Bio.SCOP.Hie.Record.__init__
Bio.SCOP.Node.__init__
Bio.SCOP.Raf.Res.__init__
Bio.SCOP.Raf.SeqMap.__init__
Bio.SCOP.Raf.SeqMapIndex.__init__
Bio.SCOP.Residues.Residues.__init__
Bio.SCOP.Scop.__init__
Bio.SVDSuperimposer.SVDSuperimposer.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabParser.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.__init__
Bio.SearchIO.BlastIO.blast_text.BlastTextParser.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.__init__
Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator.__init__
Bio.SearchIO.BlatIO.BlatPslIndexer.__init__
Bio.SearchIO.BlatIO.BlatPslParser.__init__
Bio.SearchIO.BlatIO.BlatPslWriter.__init__
Bio.SearchIO.ExonerateIO._base._BaseExonerateParser.__init__
Bio.SearchIO.FastaIO.FastaM10Indexer.__init__
Bio.SearchIO.FastaIO.FastaM10Parser.__init__
Bio.SearchIO.HHsuiteIO.hhsuite2_text.Hhsuite2TextParser.__init__
Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer.__init__
Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser.__init__
Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.__init__
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser.__init__
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.__init__
Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser.__init__
Bio.SearchIO.InterproscanIO.interproscan_xml.InterproscanXmlParser.__init__
Bio.SearchIO._index.SearchIndexer.__init__
Bio.SearchIO._legacy.NCBIStandalone.BlastErrorParser.__init__
Bio.SearchIO._legacy.NCBIStandalone.BlastParser.__init__
Bio.SearchIO._legacy.NCBIStandalone.Iterator.__init__
Bio.SearchIO._legacy.NCBIStandalone.PSIBlastParser.__init__
Bio.SearchIO._legacy.NCBIStandalone._BlastConsumer.__init__
Bio.SearchIO._legacy.NCBIStandalone._BlastErrorConsumer.__init__
Bio.SearchIO._legacy.NCBIStandalone._PSIBlastConsumer.__init__
Bio.SearchIO._legacy.NCBIStandalone._Scanner.__init__
Bio.SearchIO._legacy.ParserSupport.TaggingConsumer.__init__
Bio.SearchIO._model.hit.Hit.__init__
Bio.SearchIO._model.hsp.HSP.__init__
Bio.SearchIO._model.hsp.HSPFragment.__init__
Bio.SearchIO._model.query.QueryResult.__init__
Bio.Seq.MutableSeq.__init__
Bio.Seq.Seq.__init__
Bio.Seq.UnknownSeq.__init__
Bio.SeqFeature.CompoundLocation.__init__
Bio.SeqFeature.FeatureLocation.__init__
Bio.SeqFeature.PositionGap.__init__
Bio.SeqFeature.Reference.__init__
Bio.SeqFeature.SeqFeature.__init__
Bio.SeqIO.FastaIO.FastaTwoLineWriter.__init__
Bio.SeqIO.FastaIO.FastaWriter.__init__
Bio.SeqIO.Interfaces.SequenceIterator.__init__
Bio.SeqIO.Interfaces.SequenceWriter.__init__
Bio.SeqIO.Interfaces.SequentialSequenceWriter.__init__
Bio.SeqIO.NibIO.NibWriter.__init__
Bio.SeqIO.PhdIO.PhdWriter.__init__
Bio.SeqIO.PirIO.PirWriter.__init__
Bio.SeqIO.QualityIO.QualPhredWriter.__init__
Bio.SeqIO.SeqXmlIO.SeqXmlIterator.__init__
Bio.SeqIO.SeqXmlIO.SeqXmlWriter.__init__
Bio.SeqIO.SeqXmlIO.XMLRecordIterator.__init__
Bio.SeqIO.SffIO.SffWriter.__init__
Bio.SeqIO.SffIO._AddTellHandle.__init__
Bio.SeqIO.UniprotIO.Parser.__init__
Bio.SeqIO._index.IntelliGeneticsRandomAccess.__init__
Bio.SeqIO._index.SeqFileRandomAccess.__init__
Bio.SeqIO._index.SequentialSeqFileRandomAccess.__init__
Bio.SeqIO._index.SffRandomAccess.__init__
Bio.SeqRecord.SeqRecord.__init__
Bio.SeqRecord._RestrictedDict.__init__
Bio.SeqUtils.CodonUsage.CodonAdaptationIndex.__init__
Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.__init__
Bio.SeqUtils.ProtParam.ProteinAnalysis.__init__
Bio.Sequencing.Ace.ACEFileRecord.__init__
Bio.Sequencing.Ace.Contig.__init__
Bio.Sequencing.Ace.Reads.__init__
Bio.Sequencing.Ace.af.__init__
Bio.Sequencing.Ace.bs.__init__
Bio.Sequencing.Ace.ct.__init__
Bio.Sequencing.Ace.ds.__init__
Bio.Sequencing.Ace.qa.__init__
Bio.Sequencing.Ace.rd.__init__
Bio.Sequencing.Ace.rt.__init__
Bio.Sequencing.Ace.wa.__init__
Bio.Sequencing.Ace.wr.__init__
Bio.Sequencing.Applications._Novoalign.NovoalignCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaAlignCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaBwaswCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaIndexCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaMemCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaSampeCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaSamseCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsVersion0xSortCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsVersion1xSortCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline.__init__
Bio.Sequencing.Phd.Record.__init__
Bio.SubsMat.FreqTable.FreqTable.__init__
Bio.SubsMat.SeqMat.__init__
Bio.SwissProt.KeyWList.Record.__init__
Bio.SwissProt.Record.__init__
Bio.SwissProt.Reference.__init__
Bio.UniGene.ProtsimLine.__init__
Bio.UniGene.Record.__init__
Bio.UniGene.STSLine.__init__
Bio.UniGene.SequenceLine.__init__
Bio.Wise.dnal.Statistics.__init__
Bio.Wise.psw.AlignmentColumn.__init__
Bio.Wise.psw.ColumnUnit.__init__
Bio.bgzf.BgzfReader.__init__
Bio.bgzf.BgzfWriter.__init__
Bio.codonalign.codonalignment.CodonAlignment.__init__
Bio.codonalign.codonseq.CodonSeq.__init__
Bio.kNN.kNN.__init__
Bio.motifs.Instances.__init__
Bio.motifs.Motif.__init__
Bio.motifs.alignace.Record.__init__
Bio.motifs.applications._xxmotif.XXmotifCommandline.__init__
Bio.motifs.jaspar.Motif.__init__
Bio.motifs.jaspar.Record.__init__
Bio.motifs.jaspar.db.JASPAR5.__init__
Bio.motifs.mast.Record.__init__
Bio.motifs.matrix.GenericPositionMatrix.__init__
Bio.motifs.matrix.PositionWeightMatrix.__init__
Bio.motifs.meme.Instance.__init__
Bio.motifs.meme.Motif.__init__
Bio.motifs.meme.Record.__init__
Bio.motifs.minimal.Record.__init__
Bio.motifs.thresholds.ScoreDistribution.__init__
Bio.motifs.transfac.Record.__init__
Bio.motifs.xms.XMSScanner.__init__
Bio.pairwise2.affine_penalty.__init__
Bio.pairwise2.dictionary_match.__init__
Bio.pairwise2.identity_match.__init__
Bio.phenotype.phen_micro.JsonWriter.__init__
Bio.phenotype.phen_micro.PlateRecord.__init__
Bio.phenotype.phen_micro.WellRecord.__init__
BioSQL.BioSeq.DBSeq.__init__
BioSQL.BioSeq.DBSeqRecord.__init__
BioSQL.BioSeqDatabase.Adaptor.__init__
BioSQL.BioSeqDatabase.BioSeqDatabase.__init__
BioSQL.BioSeqDatabase.DBServer.__init__
BioSQL.BioSeqDatabase._CursorWrapper.__init__
BioSQL.DBUtils.Generic_dbutils.__init__
BioSQL.Loader.DatabaseLoader.__init__
BioSQL.Loader.DatabaseRemover.__init__" class="py-name" href="#" onclick="return doclink('link-16', '__init__', 'link-16');">__init__</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-name">parent</tt><tt class="py-op">,</tt> <tt class="py-name">pagesize</tt><tt class="py-op">,</tt> <tt class="py-name">orientation</tt><tt class="py-op">,</tt> </tt>
<a name="L107"></a><tt class="py-lineno"> 107</tt>  <tt class="py-line">                                <tt id="link-17" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.x=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#x,Variable Bio.Statistics.lowess.x=Bio.Statistics.lowess-module.html#x"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-17', 'x', 'link-17');">x</a></tt><tt class="py-op">,</tt> <tt id="link-18" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.y=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#y"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-18', 'y', 'link-18');">y</a></tt><tt class="py-op">,</tt> <tt class="py-name">xl</tt><tt class="py-op">,</tt> <tt class="py-name">xr</tt><tt class="py-op">,</tt> <tt class="py-name">yt</tt><tt class="py-op">,</tt> <tt class="py-name">yb</tt><tt class="py-op">,</tt> <tt id="link-19" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.start=Bio.SeqFeature.CompoundLocation-class.html#start,Variable Bio.SeqFeature.FeatureLocation.start=Bio.SeqFeature.FeatureLocation-class.html#start"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-19', 'start', 'link-19');">start</a></tt><tt class="py-op">,</tt> <tt id="link-20" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.end=Bio.SeqFeature.CompoundLocation-class.html#end,Variable Bio.SeqFeature.FeatureLocation.end=Bio.SeqFeature.FeatureLocation-class.html#end"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-20', 'end', 'link-20');">end</a></tt><tt class="py-op">,</tt> </tt>
<a name="L108"></a><tt class="py-lineno"> 108</tt>  <tt class="py-line">                                <tt class="py-name">tracklines</tt><tt class="py-op">,</tt> <tt class="py-name">cross_track_links</tt><tt class="py-op">)</tt> </tt>
<a name="L109"></a><tt class="py-lineno"> 109</tt>  <tt class="py-line"> </tt>
<a name="L110"></a><tt class="py-lineno"> 110</tt>  <tt class="py-line">        <tt class="py-comment"># Useful measurements on the page</tt> </tt>
<a name="L111"></a><tt class="py-lineno"> 111</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">track_size</tt> <tt class="py-op">=</tt> <tt class="py-name">track_size</tt> </tt>
<a name="L112"></a><tt class="py-lineno"> 112</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">circle_core</tt> <tt class="py-op">=</tt> <tt class="py-name">circle_core</tt> </tt>
<a name="L113"></a><tt class="py-lineno"> 113</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">circular</tt><tt class="py-op">:</tt>   <tt class="py-comment"># Determine the proportion of the circumference</tt> </tt>
<a name="L114"></a><tt class="py-lineno"> 114</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">sweep</tt> <tt class="py-op">=</tt> <tt class="py-number">0.9</tt>    <tt class="py-comment"># around which information will be drawn</tt> </tt>
<a name="L115"></a><tt class="py-lineno"> 115</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L116"></a><tt class="py-lineno"> 116</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">sweep</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
</div><a name="L117"></a><tt class="py-lineno"> 117</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer.set_track_heights"></a><div id="CircularDrawer.set_track_heights-def"><a name="L118"></a><tt class="py-lineno"> 118</tt> <a class="py-toggle" href="#" id="CircularDrawer.set_track_heights-toggle" onclick="return toggle('CircularDrawer.set_track_heights');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#set_track_heights">set_track_heights</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer.set_track_heights-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CircularDrawer.set_track_heights-expanded"><a name="L119"></a><tt class="py-lineno"> 119</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize track heights.</tt> </tt>
<a name="L120"></a><tt class="py-lineno"> 120</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L121"></a><tt class="py-lineno"> 121</tt>  <tt class="py-line"><tt class="py-docstring">        Since tracks may not be of identical heights, the bottom and top</tt> </tt>
<a name="L122"></a><tt class="py-lineno"> 122</tt>  <tt class="py-line"><tt class="py-docstring">        radius for each track is stored in a dictionary - self.track_radii,</tt> </tt>
<a name="L123"></a><tt class="py-lineno"> 123</tt>  <tt class="py-line"><tt class="py-docstring">        keyed by track number</tt> </tt>
<a name="L124"></a><tt class="py-lineno"> 124</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L125"></a><tt class="py-lineno"> 125</tt>  <tt class="py-line">        <tt class="py-name">bot_track</tt> <tt class="py-op">=</tt> <tt id="link-21" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.min=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#min"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-21', 'min', 'link-21');">min</a></tt><tt class="py-op">(</tt><tt id="link-22" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-22', 'min', 'link-21');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">drawn_tracks</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L126"></a><tt class="py-lineno"> 126</tt>  <tt class="py-line">        <tt class="py-name">top_track</tt> <tt class="py-op">=</tt> <tt id="link-23" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.max=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#max"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-23', 'max', 'link-23');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">drawn_tracks</tt><tt class="py-op">)</tt>  <tt class="py-comment"># The 'highest' track to draw</tt> </tt>
<a name="L127"></a><tt class="py-lineno"> 127</tt>  <tt class="py-line"> </tt>
<a name="L128"></a><tt class="py-lineno"> 128</tt>  <tt class="py-line">        <tt class="py-name">trackunit_sum</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt>  <tt class="py-comment"># Total number of 'units' taken up by all tracks</tt> </tt>
<a name="L129"></a><tt class="py-lineno"> 129</tt>  <tt class="py-line">        <tt class="py-name">trackunits</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt>  <tt class="py-comment"># Start and &amp; units for each track keyed by track number</tt> </tt>
<a name="L130"></a><tt class="py-lineno"> 130</tt>  <tt class="py-line">        <tt class="py-name">heightholder</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt>  <tt class="py-comment"># placeholder variable</tt> </tt>
<a name="L131"></a><tt class="py-lineno"> 131</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">track</tt> <tt class="py-keyword">in</tt> <tt id="link-24" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-24', 'range', 'link-3');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">bot_track</tt><tt class="py-op">,</tt> <tt class="py-name">top_track</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># track numbers to 'draw'</tt> </tt>
<a name="L132"></a><tt class="py-lineno"> 132</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L133"></a><tt class="py-lineno"> 133</tt>  <tt class="py-line">                <tt class="py-name">trackheight</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_parent</tt><tt class="py-op">[</tt><tt class="py-name">track</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-25" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.height=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#height"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.height" class="py-name" href="#" onclick="return doclink('link-25', 'height', 'link-25');">height</a></tt>  <tt class="py-comment"># Get track height</tt> </tt>
<a name="L134"></a><tt class="py-lineno"> 134</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">Exception</tt><tt class="py-op">:</tt>  <tt class="py-comment"># TODO: ValueError? IndexError?</tt> </tt>
<a name="L135"></a><tt class="py-lineno"> 135</tt>  <tt class="py-line">                <tt class="py-name">trackheight</tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> </tt>
<a name="L136"></a><tt class="py-lineno"> 136</tt>  <tt class="py-line">            <tt class="py-name">trackunit_sum</tt> <tt class="py-op">+=</tt> <tt class="py-name">trackheight</tt>  <tt class="py-comment"># increment total track unit height</tt> </tt>
<a name="L137"></a><tt class="py-lineno"> 137</tt>  <tt class="py-line">            <tt class="py-name">trackunits</tt><tt class="py-op">[</tt><tt class="py-name">track</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">heightholder</tt><tt class="py-op">,</tt> <tt class="py-name">heightholder</tt> <tt class="py-op">+</tt> <tt class="py-name">trackheight</tt><tt class="py-op">)</tt> </tt>
<a name="L138"></a><tt class="py-lineno"> 138</tt>  <tt class="py-line">            <tt class="py-name">heightholder</tt> <tt class="py-op">+=</tt> <tt class="py-name">trackheight</tt>  <tt class="py-comment"># move to next height</tt> </tt>
<a name="L139"></a><tt class="py-lineno"> 139</tt>  <tt class="py-line"> </tt>
<a name="L140"></a><tt class="py-lineno"> 140</tt>  <tt class="py-line">        <tt class="py-name">max_radius</tt> <tt class="py-op">=</tt> <tt class="py-number">0.5</tt> <tt class="py-op">*</tt> <tt id="link-26" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-26', 'min', 'link-21');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">pagewidth</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">pageheight</tt><tt class="py-op">)</tt> </tt>
<a name="L141"></a><tt class="py-lineno"> 141</tt>  <tt class="py-line">        <tt class="py-name">trackunit_height</tt> <tt class="py-op">=</tt> <tt class="py-name">max_radius</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-number">1</tt> <tt class="py-op">-</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">circle_core</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-name">trackunit_sum</tt> </tt>
<a name="L142"></a><tt class="py-lineno"> 142</tt>  <tt class="py-line">        <tt class="py-name">track_core</tt> <tt class="py-op">=</tt> <tt class="py-name">max_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">circle_core</tt> </tt>
<a name="L143"></a><tt class="py-lineno"> 143</tt>  <tt class="py-line"> </tt>
<a name="L144"></a><tt class="py-lineno"> 144</tt>  <tt class="py-line">        <tt class="py-comment"># Calculate top and bottom radii for each track</tt> </tt>
<a name="L145"></a><tt class="py-lineno"> 145</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">track_radii</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt>  <tt class="py-comment"># The inner, outer and center radii for each track</tt> </tt>
<a name="L146"></a><tt class="py-lineno"> 146</tt>  <tt class="py-line">        <tt class="py-name">track_crop</tt> <tt class="py-op">=</tt> <tt class="py-name">trackunit_height</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-number">1</tt> <tt class="py-op">-</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">track_size</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-number">2.</tt>  <tt class="py-comment"># 'step back' in pixels</tt> </tt>
<a name="L147"></a><tt class="py-lineno"> 147</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">track</tt> <tt class="py-keyword">in</tt> <tt class="py-name">trackunits</tt><tt class="py-op">:</tt> </tt>
<a name="L148"></a><tt class="py-lineno"> 148</tt>  <tt class="py-line">            <tt class="py-name">top</tt> <tt class="py-op">=</tt> <tt class="py-name">trackunits</tt><tt class="py-op">[</tt><tt class="py-name">track</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">*</tt> <tt class="py-name">trackunit_height</tt> <tt class="py-op">-</tt> <tt class="py-name">track_crop</tt> <tt class="py-op">+</tt> <tt class="py-name">track_core</tt> </tt>
<a name="L149"></a><tt class="py-lineno"> 149</tt>  <tt class="py-line">            <tt class="py-name">btm</tt> <tt class="py-op">=</tt> <tt class="py-name">trackunits</tt><tt class="py-op">[</tt><tt class="py-name">track</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">*</tt> <tt class="py-name">trackunit_height</tt> <tt class="py-op">+</tt> <tt class="py-name">track_crop</tt> <tt class="py-op">+</tt> <tt class="py-name">track_core</tt> </tt>
<a name="L150"></a><tt class="py-lineno"> 150</tt>  <tt class="py-line">            <tt class="py-name">ctr</tt> <tt class="py-op">=</tt> <tt class="py-name">btm</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt class="py-name">top</tt> <tt class="py-op">-</tt> <tt class="py-name">btm</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-number">2.</tt> </tt>
<a name="L151"></a><tt class="py-lineno"> 151</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">track_radii</tt><tt class="py-op">[</tt><tt class="py-name">track</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">btm</tt><tt class="py-op">,</tt> <tt class="py-name">ctr</tt><tt class="py-op">,</tt> <tt class="py-name">top</tt><tt class="py-op">)</tt> </tt>
</div><a name="L152"></a><tt class="py-lineno"> 152</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer.draw"></a><div id="CircularDrawer.draw-def"><a name="L153"></a><tt class="py-lineno"> 153</tt> <a class="py-toggle" href="#" id="CircularDrawer.draw-toggle" onclick="return toggle('CircularDrawer.draw');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw">draw</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer.draw-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CircularDrawer.draw-expanded"><a name="L154"></a><tt class="py-lineno"> 154</tt>  <tt class="py-line">        <tt class="py-docstring">"""Draw a circular diagram of the stored data."""</tt> </tt>
<a name="L155"></a><tt class="py-lineno"> 155</tt>  <tt class="py-line">        <tt class="py-comment"># Instantiate the drawing canvas</tt> </tt>
<a name="L156"></a><tt class="py-lineno"> 156</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">drawing</tt> <tt class="py-op">=</tt> <tt class="py-name">Drawing</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">pagesize</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">pagesize</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L157"></a><tt class="py-lineno"> 157</tt>  <tt class="py-line"> </tt>
<a name="L158"></a><tt class="py-lineno"> 158</tt>  <tt class="py-line">        <tt class="py-name">feature_elements</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>           <tt class="py-comment"># holds feature elements</tt> </tt>
<a name="L159"></a><tt class="py-lineno"> 159</tt>  <tt class="py-line">        <tt class="py-name">feature_labels</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>             <tt class="py-comment"># holds feature labels</tt> </tt>
<a name="L160"></a><tt class="py-lineno"> 160</tt>  <tt class="py-line">        <tt class="py-name">greytrack_bgs</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>              <tt class="py-comment"># holds track background</tt> </tt>
<a name="L161"></a><tt class="py-lineno"> 161</tt>  <tt class="py-line">        <tt class="py-name">greytrack_labels</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>           <tt class="py-comment"># holds track foreground labels</tt> </tt>
<a name="L162"></a><tt class="py-lineno"> 162</tt>  <tt class="py-line">        <tt class="py-name">scale_axes</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>                 <tt class="py-comment"># holds scale axes</tt> </tt>
<a name="L163"></a><tt class="py-lineno"> 163</tt>  <tt class="py-line">        <tt class="py-name">scale_labels</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>               <tt class="py-comment"># holds scale axis labels</tt> </tt>
<a name="L164"></a><tt class="py-lineno"> 164</tt>  <tt class="py-line"> </tt>
<a name="L165"></a><tt class="py-lineno"> 165</tt>  <tt class="py-line">        <tt class="py-comment"># Get tracks to be drawn and set track sizes</tt> </tt>
<a name="L166"></a><tt class="py-lineno"> 166</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">drawn_tracks</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_parent</tt><tt class="py-op">.</tt><tt id="link-27" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.get_drawn_levels()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#get_drawn_levels"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.get_drawn_levels" class="py-name" href="#" onclick="return doclink('link-27', 'get_drawn_levels', 'link-27');">get_drawn_levels</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L167"></a><tt class="py-lineno"> 167</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-28" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.set_track_heights()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#set_track_heights,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.set_track_heights()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#set_track_heights"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.set_track_heights
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.set_track_heights" class="py-name" href="#" onclick="return doclink('link-28', 'set_track_heights', 'link-28');">set_track_heights</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L168"></a><tt class="py-lineno"> 168</tt>  <tt class="py-line"> </tt>
<a name="L169"></a><tt class="py-lineno"> 169</tt>  <tt class="py-line">        <tt class="py-comment"># Go through each track in the parent (if it is to be drawn) one by</tt> </tt>
<a name="L170"></a><tt class="py-lineno"> 170</tt>  <tt class="py-line">        <tt class="py-comment"># one and collate the data as drawing elements</tt> </tt>
<a name="L171"></a><tt class="py-lineno"> 171</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">track_level</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_parent</tt><tt class="py-op">.</tt><tt id="link-29" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.get_drawn_levels" class="py-name" href="#" onclick="return doclink('link-29', 'get_drawn_levels', 'link-27');">get_drawn_levels</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L172"></a><tt class="py-lineno"> 172</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">current_track_level</tt> <tt class="py-op">=</tt> <tt class="py-name">track_level</tt> </tt>
<a name="L173"></a><tt class="py-lineno"> 173</tt>  <tt class="py-line">            <tt class="py-name">track</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_parent</tt><tt class="py-op">[</tt><tt class="py-name">track_level</tt><tt class="py-op">]</tt> </tt>
<a name="L174"></a><tt class="py-lineno"> 174</tt>  <tt class="py-line">            <tt class="py-name">gbgs</tt><tt class="py-op">,</tt> <tt class="py-name">glabels</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-30" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw_greytrack()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw_greytrack,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw_greytrack()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#draw_greytrack"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw_greytrack
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw_greytrack" class="py-name" href="#" onclick="return doclink('link-30', 'draw_greytrack', 'link-30');">draw_greytrack</a></tt><tt class="py-op">(</tt><tt class="py-name">track</tt><tt class="py-op">)</tt>    <tt class="py-comment"># Greytracks</tt> </tt>
<a name="L175"></a><tt class="py-lineno"> 175</tt>  <tt class="py-line">            <tt class="py-name">greytrack_bgs</tt><tt class="py-op">.</tt><tt id="link-31" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-31', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">gbgs</tt><tt class="py-op">)</tt> </tt>
<a name="L176"></a><tt class="py-lineno"> 176</tt>  <tt class="py-line">            <tt class="py-name">greytrack_labels</tt><tt class="py-op">.</tt><tt id="link-32" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-32', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">glabels</tt><tt class="py-op">)</tt> </tt>
<a name="L177"></a><tt class="py-lineno"> 177</tt>  <tt class="py-line">            <tt id="link-33" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.features=BioSQL.BioSeq.DBSeqRecord-class.html#features"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-33', 'features', 'link-33');">features</a></tt><tt class="py-op">,</tt> <tt class="py-name">flabels</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-34" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw_track()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw_track,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw_track()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#draw_track"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw_track
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw_track" class="py-name" href="#" onclick="return doclink('link-34', 'draw_track', 'link-34');">draw_track</a></tt><tt class="py-op">(</tt><tt class="py-name">track</tt><tt class="py-op">)</tt>   <tt class="py-comment"># Features and graphs</tt> </tt>
<a name="L178"></a><tt class="py-lineno"> 178</tt>  <tt class="py-line">            <tt class="py-name">feature_elements</tt><tt class="py-op">.</tt><tt id="link-35" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-35', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt id="link-36" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-36', 'features', 'link-33');">features</a></tt><tt class="py-op">)</tt> </tt>
<a name="L179"></a><tt class="py-lineno"> 179</tt>  <tt class="py-line">            <tt class="py-name">feature_labels</tt><tt class="py-op">.</tt><tt id="link-37" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-37', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">flabels</tt><tt class="py-op">)</tt> </tt>
<a name="L180"></a><tt class="py-lineno"> 180</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">scale</tt><tt class="py-op">:</tt> </tt>
<a name="L181"></a><tt class="py-lineno"> 181</tt>  <tt class="py-line">                <tt class="py-name">axes</tt><tt class="py-op">,</tt> <tt class="py-name">slabels</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-38" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw_scale()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw_scale,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw_scale()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#draw_scale"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw_scale
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw_scale" class="py-name" href="#" onclick="return doclink('link-38', 'draw_scale', 'link-38');">draw_scale</a></tt><tt class="py-op">(</tt><tt class="py-name">track</tt><tt class="py-op">)</tt>       <tt class="py-comment"># Scale axes</tt> </tt>
<a name="L182"></a><tt class="py-lineno"> 182</tt>  <tt class="py-line">                <tt class="py-name">scale_axes</tt><tt class="py-op">.</tt><tt id="link-39" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-39', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">axes</tt><tt class="py-op">)</tt> </tt>
<a name="L183"></a><tt class="py-lineno"> 183</tt>  <tt class="py-line">                <tt class="py-name">scale_labels</tt><tt class="py-op">.</tt><tt id="link-40" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-40', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">slabels</tt><tt class="py-op">)</tt> </tt>
<a name="L184"></a><tt class="py-lineno"> 184</tt>  <tt class="py-line"> </tt>
<a name="L185"></a><tt class="py-lineno"> 185</tt>  <tt class="py-line">        <tt class="py-name">feature_cross_links</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L186"></a><tt class="py-lineno"> 186</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">cross_link_obj</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">cross_track_links</tt><tt class="py-op">:</tt> </tt>
<a name="L187"></a><tt class="py-lineno"> 187</tt>  <tt class="py-line">            <tt class="py-name">cross_link_elements</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-41" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw_cross_link()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw_cross_link,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw_cross_link()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#draw_cross_link"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw_cross_link
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw_cross_link" class="py-name" href="#" onclick="return doclink('link-41', 'draw_cross_link', 'link-41');">draw_cross_link</a></tt><tt class="py-op">(</tt><tt class="py-name">cross_link_obj</tt><tt class="py-op">)</tt> </tt>
<a name="L188"></a><tt class="py-lineno"> 188</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">cross_link_elements</tt><tt class="py-op">:</tt> </tt>
<a name="L189"></a><tt class="py-lineno"> 189</tt>  <tt class="py-line">                <tt class="py-name">feature_cross_links</tt><tt class="py-op">.</tt><tt id="link-42" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-42', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">cross_link_elements</tt><tt class="py-op">)</tt> </tt>
<a name="L190"></a><tt class="py-lineno"> 190</tt>  <tt class="py-line"> </tt>
<a name="L191"></a><tt class="py-lineno"> 191</tt>  <tt class="py-line">        <tt class="py-comment"># Groups listed in order of addition to page (from back to front)</tt> </tt>
<a name="L192"></a><tt class="py-lineno"> 192</tt>  <tt class="py-line">        <tt class="py-comment"># Draw track backgrounds</tt> </tt>
<a name="L193"></a><tt class="py-lineno"> 193</tt>  <tt class="py-line">        <tt class="py-comment"># Draw feature cross track links</tt> </tt>
<a name="L194"></a><tt class="py-lineno"> 194</tt>  <tt class="py-line">        <tt class="py-comment"># Draw features and graphs</tt> </tt>
<a name="L195"></a><tt class="py-lineno"> 195</tt>  <tt class="py-line">        <tt class="py-comment"># Draw scale axes</tt> </tt>
<a name="L196"></a><tt class="py-lineno"> 196</tt>  <tt class="py-line">        <tt class="py-comment"># Draw scale labels</tt> </tt>
<a name="L197"></a><tt class="py-lineno"> 197</tt>  <tt class="py-line">        <tt class="py-comment"># Draw feature labels</tt> </tt>
<a name="L198"></a><tt class="py-lineno"> 198</tt>  <tt class="py-line">        <tt class="py-comment"># Draw track labels</tt> </tt>
<a name="L199"></a><tt class="py-lineno"> 199</tt>  <tt class="py-line">        <tt class="py-name">element_groups</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">greytrack_bgs</tt><tt class="py-op">,</tt> <tt class="py-name">feature_cross_links</tt><tt class="py-op">,</tt> </tt>
<a name="L200"></a><tt class="py-lineno"> 200</tt>  <tt class="py-line">                          <tt class="py-name">feature_elements</tt><tt class="py-op">,</tt> </tt>
<a name="L201"></a><tt class="py-lineno"> 201</tt>  <tt class="py-line">                          <tt class="py-name">scale_axes</tt><tt class="py-op">,</tt> <tt class="py-name">scale_labels</tt><tt class="py-op">,</tt> </tt>
<a name="L202"></a><tt class="py-lineno"> 202</tt>  <tt class="py-line">                          <tt class="py-name">feature_labels</tt><tt class="py-op">,</tt> <tt class="py-name">greytrack_labels</tt> </tt>
<a name="L203"></a><tt class="py-lineno"> 203</tt>  <tt class="py-line">                          <tt class="py-op">]</tt> </tt>
<a name="L204"></a><tt class="py-lineno"> 204</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">element_group</tt> <tt class="py-keyword">in</tt> <tt class="py-name">element_groups</tt><tt class="py-op">:</tt> </tt>
<a name="L205"></a><tt class="py-lineno"> 205</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">element_list</tt> <tt class="py-keyword">in</tt> <tt class="py-name">element_group</tt><tt class="py-op">:</tt> </tt>
<a name="L206"></a><tt class="py-lineno"> 206</tt>  <tt class="py-line">                <tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">drawing</tt><tt class="py-op">.</tt><tt id="link-43" class="py-name" targets="Method Bio.Graphics.BasicChromosome._ChromosomeComponent.add()=Bio.Graphics.BasicChromosome._ChromosomeComponent-class.html#add,Method Bio.Nexus.Nexus.StepMatrix.add()=Bio.Nexus.Nexus.StepMatrix-class.html#add,Method Bio.Nexus.Nodes.Chain.add()=Bio.Nexus.Nodes.Chain-class.html#add,Method Bio.PDB.Entity.Entity.add()=Bio.PDB.Entity.Entity-class.html#add,Method Bio.PDB.Residue.DisorderedResidue.add()=Bio.PDB.Residue.DisorderedResidue-class.html#add,Method Bio.PDB.Residue.Residue.add()=Bio.PDB.Residue.Residue-class.html#add,Method Bio.Restriction.Restriction.RestrictionBatch.add()=Bio.Restriction.Restriction.RestrictionBatch-class.html#add"><a title="Bio.Graphics.BasicChromosome._ChromosomeComponent.add
Bio.Nexus.Nexus.StepMatrix.add
Bio.Nexus.Nodes.Chain.add
Bio.PDB.Entity.Entity.add
Bio.PDB.Residue.DisorderedResidue.add
Bio.PDB.Residue.Residue.add
Bio.Restriction.Restriction.RestrictionBatch.add" class="py-name" href="#" onclick="return doclink('link-43', 'add', 'link-43');">add</a></tt><tt class="py-op">(</tt><tt id="link-44" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.element=Bio.KEGG.KGML.KGML_pathway.Component-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Entry.element=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Graphics.element=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.element=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.element=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#element,Variable Bio.KEGG.KGML.KGML_pathway.Relation.element=Bio.KEGG.KGML.KGML_pathway.Relation-class.html#element,Variable Bio.Phylo._cdao_owl.element=Bio.Phylo._cdao_owl-module.html#element"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-44', 'element', 'link-44');">element</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-45" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.element
Bio.KEGG.KGML.KGML_pathway.Entry.element
Bio.KEGG.KGML.KGML_pathway.Graphics.element
Bio.KEGG.KGML.KGML_pathway.Pathway.element
Bio.KEGG.KGML.KGML_pathway.Reaction.element
Bio.KEGG.KGML.KGML_pathway.Relation.element
Bio.Phylo._cdao_owl.element" class="py-name" href="#" onclick="return doclink('link-45', 'element', 'link-44');">element</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">element_list</tt><tt class="py-op">]</tt> </tt>
<a name="L207"></a><tt class="py-lineno"> 207</tt>  <tt class="py-line"> </tt>
<a name="L208"></a><tt class="py-lineno"> 208</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">tracklines</tt><tt class="py-op">:</tt> </tt>
<a name="L209"></a><tt class="py-lineno"> 209</tt>  <tt class="py-line">            <tt class="py-comment"># Draw test tracks over top of diagram</tt> </tt>
<a name="L210"></a><tt class="py-lineno"> 210</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-46" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw_test_tracks()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw_test_tracks,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw_test_tracks()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#draw_test_tracks"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw_test_tracks
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw_test_tracks" class="py-name" href="#" onclick="return doclink('link-46', 'draw_test_tracks', 'link-46');">draw_test_tracks</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L211"></a><tt class="py-lineno"> 211</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer.draw_track"></a><div id="CircularDrawer.draw_track-def"><a name="L212"></a><tt class="py-lineno"> 212</tt> <a class="py-toggle" href="#" id="CircularDrawer.draw_track-toggle" onclick="return toggle('CircularDrawer.draw_track');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw_track">draw_track</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">track</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer.draw_track-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CircularDrawer.draw_track-expanded"><a name="L213"></a><tt class="py-lineno"> 213</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return list of track elements and list of track labels."""</tt> </tt>
<a name="L214"></a><tt class="py-lineno"> 214</tt>  <tt class="py-line">        <tt class="py-name">track_elements</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>  <tt class="py-comment"># Holds elements for features and graphs</tt> </tt>
<a name="L215"></a><tt class="py-lineno"> 215</tt>  <tt class="py-line">        <tt class="py-name">track_labels</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>   <tt class="py-comment"># Holds labels for features and graphs</tt> </tt>
<a name="L216"></a><tt class="py-lineno"> 216</tt>  <tt class="py-line"> </tt>
<a name="L217"></a><tt class="py-lineno"> 217</tt>  <tt class="py-line">        <tt class="py-comment"># Distribution dictionary for dealing with different set types</tt> </tt>
<a name="L218"></a><tt class="py-lineno"> 218</tt>  <tt class="py-line">        <tt class="py-name">set_methods</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt id="link-47" class="py-name"><a title="Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet" class="py-name" href="#" onclick="return doclink('link-47', 'FeatureSet', 'link-11');">FeatureSet</a></tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-48" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw_feature_set()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw_feature_set,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw_feature_set()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#draw_feature_set"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw_feature_set
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw_feature_set" class="py-name" href="#" onclick="return doclink('link-48', 'draw_feature_set', 'link-48');">draw_feature_set</a></tt><tt class="py-op">,</tt> </tt>
<a name="L219"></a><tt class="py-lineno"> 219</tt>  <tt class="py-line">                       <tt id="link-49" class="py-name"><a title="Bio.Graphics.GenomeDiagram._GraphSet.GraphSet" class="py-name" href="#" onclick="return doclink('link-49', 'GraphSet', 'link-13');">GraphSet</a></tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-50" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw_graph_set()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw_graph_set,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw_graph_set()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#draw_graph_set"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw_graph_set
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw_graph_set" class="py-name" href="#" onclick="return doclink('link-50', 'draw_graph_set', 'link-50');">draw_graph_set</a></tt> </tt>
<a name="L220"></a><tt class="py-lineno"> 220</tt>  <tt class="py-line">                       <tt class="py-op">}</tt> </tt>
<a name="L221"></a><tt class="py-lineno"> 221</tt>  <tt class="py-line"> </tt>
<a name="L222"></a><tt class="py-lineno"> 222</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-51" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.set()=Bio.Nexus.Nexus.StepMatrix-class.html#set,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.set()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#set,Method Bio.SVDSuperimposer.SVDSuperimposer.set()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#set"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-51', 'set', 'link-51');">set</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">track</tt><tt class="py-op">.</tt><tt id="link-52" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Track.Track.get_sets()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#get_sets"><a title="Bio.Graphics.GenomeDiagram._Track.Track.get_sets" class="py-name" href="#" onclick="return doclink('link-52', 'get_sets', 'link-52');">get_sets</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># Draw the feature or graph sets</tt> </tt>
<a name="L223"></a><tt class="py-lineno"> 223</tt>  <tt class="py-line">            <tt id="link-53" class="py-name" targets="Method Bio.Restriction.Restriction.RestrictionBatch.elements()=Bio.Restriction.Restriction.RestrictionBatch-class.html#elements"><a title="Bio.Restriction.Restriction.RestrictionBatch.elements" class="py-name" href="#" onclick="return doclink('link-53', 'elements', 'link-53');">elements</a></tt><tt class="py-op">,</tt> <tt id="link-54" class="py-name" targets="Method Bio.Pathway.Rep.Graph.Graph.labels()=Bio.Pathway.Rep.Graph.Graph-class.html#labels,Method Bio.Pathway.Rep.MultiGraph.MultiGraph.labels()=Bio.Pathway.Rep.MultiGraph.MultiGraph-class.html#labels"><a title="Bio.Pathway.Rep.Graph.Graph.labels
Bio.Pathway.Rep.MultiGraph.MultiGraph.labels" class="py-name" href="#" onclick="return doclink('link-54', 'labels', 'link-54');">labels</a></tt> <tt class="py-op">=</tt> <tt class="py-name">set_methods</tt><tt class="py-op">[</tt><tt id="link-55" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-55', 'set', 'link-51');">set</a></tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">]</tt><tt class="py-op">(</tt><tt id="link-56" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-56', 'set', 'link-51');">set</a></tt><tt class="py-op">)</tt> </tt>
<a name="L224"></a><tt class="py-lineno"> 224</tt>  <tt class="py-line">            <tt class="py-name">track_elements</tt> <tt class="py-op">+=</tt> <tt id="link-57" class="py-name"><a title="Bio.Restriction.Restriction.RestrictionBatch.elements" class="py-name" href="#" onclick="return doclink('link-57', 'elements', 'link-53');">elements</a></tt> </tt>
<a name="L225"></a><tt class="py-lineno"> 225</tt>  <tt class="py-line">            <tt class="py-name">track_labels</tt> <tt class="py-op">+=</tt> <tt id="link-58" class="py-name"><a title="Bio.Pathway.Rep.Graph.Graph.labels
Bio.Pathway.Rep.MultiGraph.MultiGraph.labels" class="py-name" href="#" onclick="return doclink('link-58', 'labels', 'link-54');">labels</a></tt> </tt>
<a name="L226"></a><tt class="py-lineno"> 226</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">track_elements</tt><tt class="py-op">,</tt> <tt class="py-name">track_labels</tt> </tt>
</div><a name="L227"></a><tt class="py-lineno"> 227</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer.draw_feature_set"></a><div id="CircularDrawer.draw_feature_set-def"><a name="L228"></a><tt class="py-lineno"> 228</tt> <a class="py-toggle" href="#" id="CircularDrawer.draw_feature_set-toggle" onclick="return toggle('CircularDrawer.draw_feature_set');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw_feature_set">draw_feature_set</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">set</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer.draw_feature_set-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CircularDrawer.draw_feature_set-expanded"><a name="L229"></a><tt class="py-lineno"> 229</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return list of feature elements and list of labels for them."""</tt> </tt>
<a name="L230"></a><tt class="py-lineno"> 230</tt>  <tt class="py-line">        <tt class="py-comment"># print 'draw feature set'</tt> </tt>
<a name="L231"></a><tt class="py-lineno"> 231</tt>  <tt class="py-line">        <tt class="py-name">feature_elements</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>  <tt class="py-comment"># Holds diagram elements belonging to the features</tt> </tt>
<a name="L232"></a><tt class="py-lineno"> 232</tt>  <tt class="py-line">        <tt class="py-name">label_elements</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>   <tt class="py-comment"># Holds diagram elements belonging to feature labels</tt> </tt>
<a name="L233"></a><tt class="py-lineno"> 233</tt>  <tt class="py-line"> </tt>
<a name="L234"></a><tt class="py-lineno"> 234</tt>  <tt class="py-line">        <tt class="py-comment"># Collect all the elements for the feature set</tt> </tt>
<a name="L235"></a><tt class="py-lineno"> 235</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">feature</tt> <tt class="py-keyword">in</tt> <tt id="link-59" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-59', 'set', 'link-51');">set</a></tt><tt class="py-op">.</tt><tt id="link-60" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.get_features()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#get_features"><a title="Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.get_features" class="py-name" href="#" onclick="return doclink('link-60', 'get_features', 'link-60');">get_features</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L236"></a><tt class="py-lineno"> 236</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-61" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.is_in_bounds()=Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer-class.html#is_in_bounds"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.is_in_bounds" class="py-name" href="#" onclick="return doclink('link-61', 'is_in_bounds', 'link-61');">is_in_bounds</a></tt><tt class="py-op">(</tt><tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-62" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-62', 'start', 'link-19');">start</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-63" class="py-name"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.is_in_bounds" class="py-name" href="#" onclick="return doclink('link-63', 'is_in_bounds', 'link-61');">is_in_bounds</a></tt><tt class="py-op">(</tt><tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-64" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-64', 'end', 'link-20');">end</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L237"></a><tt class="py-lineno"> 237</tt>  <tt class="py-line">                <tt id="link-65" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-65', 'features', 'link-33');">features</a></tt><tt class="py-op">,</tt> <tt id="link-66" class="py-name"><a title="Bio.Pathway.Rep.Graph.Graph.labels
Bio.Pathway.Rep.MultiGraph.MultiGraph.labels" class="py-name" href="#" onclick="return doclink('link-66', 'labels', 'link-54');">labels</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-67" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw_feature()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw_feature,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw_feature()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#draw_feature"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw_feature
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw_feature" class="py-name" href="#" onclick="return doclink('link-67', 'draw_feature', 'link-67');">draw_feature</a></tt><tt class="py-op">(</tt><tt class="py-name">feature</tt><tt class="py-op">)</tt> </tt>
<a name="L238"></a><tt class="py-lineno"> 238</tt>  <tt class="py-line">                <tt class="py-name">feature_elements</tt> <tt class="py-op">+=</tt> <tt id="link-68" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-68', 'features', 'link-33');">features</a></tt> </tt>
<a name="L239"></a><tt class="py-lineno"> 239</tt>  <tt class="py-line">                <tt class="py-name">label_elements</tt> <tt class="py-op">+=</tt> <tt id="link-69" class="py-name"><a title="Bio.Pathway.Rep.Graph.Graph.labels
Bio.Pathway.Rep.MultiGraph.MultiGraph.labels" class="py-name" href="#" onclick="return doclink('link-69', 'labels', 'link-54');">labels</a></tt> </tt>
<a name="L240"></a><tt class="py-lineno"> 240</tt>  <tt class="py-line"> </tt>
<a name="L241"></a><tt class="py-lineno"> 241</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">feature_elements</tt><tt class="py-op">,</tt> <tt class="py-name">label_elements</tt> </tt>
</div><a name="L242"></a><tt class="py-lineno"> 242</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer.draw_feature"></a><div id="CircularDrawer.draw_feature-def"><a name="L243"></a><tt class="py-lineno"> 243</tt> <a class="py-toggle" href="#" id="CircularDrawer.draw_feature-toggle" onclick="return toggle('CircularDrawer.draw_feature');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw_feature">draw_feature</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">feature</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer.draw_feature-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CircularDrawer.draw_feature-expanded"><a name="L244"></a><tt class="py-lineno"> 244</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return list of feature elements and list of labels for them."""</tt> </tt>
<a name="L245"></a><tt class="py-lineno"> 245</tt>  <tt class="py-line">        <tt class="py-name">feature_elements</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>  <tt class="py-comment"># Holds drawable elements for a single feature</tt> </tt>
<a name="L246"></a><tt class="py-lineno"> 246</tt>  <tt class="py-line">        <tt class="py-name">label_elements</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>   <tt class="py-comment"># Holds labels for a single feature</tt> </tt>
<a name="L247"></a><tt class="py-lineno"> 247</tt>  <tt class="py-line"> </tt>
<a name="L248"></a><tt class="py-lineno"> 248</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt class="py-name">hide</tt><tt class="py-op">:</tt>    <tt class="py-comment"># Don't show feature: return early</tt> </tt>
<a name="L249"></a><tt class="py-lineno"> 249</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">feature_elements</tt><tt class="py-op">,</tt> <tt class="py-name">label_elements</tt> </tt>
<a name="L250"></a><tt class="py-lineno"> 250</tt>  <tt class="py-line"> </tt>
<a name="L251"></a><tt class="py-lineno"> 251</tt>  <tt class="py-line">        <tt id="link-70" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-70', 'start', 'link-19');">start</a></tt><tt class="py-op">,</tt> <tt id="link-71" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-71', 'end', 'link-20');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-72" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer._current_track_start_end()=Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer-class.html#_current_track_start_end"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer._current_track_start_end" class="py-name" href="#" onclick="return doclink('link-72', '_current_track_start_end', 'link-72');">_current_track_start_end</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L252"></a><tt class="py-lineno"> 252</tt>  <tt class="py-line">        <tt class="py-comment"># A single feature may be split into subfeatures, so loop over them</tt> </tt>
<a name="L253"></a><tt class="py-lineno"> 253</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">locstart</tt><tt class="py-op">,</tt> <tt class="py-name">locend</tt> <tt class="py-keyword">in</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt class="py-name">locations</tt><tt class="py-op">:</tt> </tt>
<a name="L254"></a><tt class="py-lineno"> 254</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">locend</tt> <tt class="py-op">&lt;</tt> <tt id="link-73" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-73', 'start', 'link-19');">start</a></tt><tt class="py-op">:</tt> </tt>
<a name="L255"></a><tt class="py-lineno"> 255</tt>  <tt class="py-line">                <tt class="py-keyword">continue</tt> </tt>
<a name="L256"></a><tt class="py-lineno"> 256</tt>  <tt class="py-line">            <tt class="py-name">locstart</tt> <tt class="py-op">=</tt> <tt id="link-74" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-74', 'max', 'link-23');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">locstart</tt><tt class="py-op">,</tt> <tt id="link-75" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-75', 'start', 'link-19');">start</a></tt><tt class="py-op">)</tt> </tt>
<a name="L257"></a><tt class="py-lineno"> 257</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-76" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-76', 'end', 'link-20');">end</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-name">locstart</tt><tt class="py-op">:</tt> </tt>
<a name="L258"></a><tt class="py-lineno"> 258</tt>  <tt class="py-line">                <tt class="py-keyword">continue</tt> </tt>
<a name="L259"></a><tt class="py-lineno"> 259</tt>  <tt class="py-line">            <tt class="py-name">locend</tt> <tt class="py-op">=</tt> <tt id="link-77" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-77', 'min', 'link-21');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">locend</tt><tt class="py-op">,</tt> <tt id="link-78" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-78', 'end', 'link-20');">end</a></tt><tt class="py-op">)</tt> </tt>
<a name="L260"></a><tt class="py-lineno"> 260</tt>  <tt class="py-line">            <tt class="py-comment"># Get sigil for the feature/ each subfeature</tt> </tt>
<a name="L261"></a><tt class="py-lineno"> 261</tt>  <tt class="py-line">            <tt class="py-name">feature_sigil</tt><tt class="py-op">,</tt> <tt class="py-name">label</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-79" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.get_feature_sigil()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#get_feature_sigil,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.get_feature_sigil()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#get_feature_sigil"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.get_feature_sigil
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.get_feature_sigil" class="py-name" href="#" onclick="return doclink('link-79', 'get_feature_sigil', 'link-79');">get_feature_sigil</a></tt><tt class="py-op">(</tt><tt class="py-name">feature</tt><tt class="py-op">,</tt> <tt class="py-name">locstart</tt><tt class="py-op">,</tt> <tt class="py-name">locend</tt><tt class="py-op">)</tt> </tt>
<a name="L262"></a><tt class="py-lineno"> 262</tt>  <tt class="py-line">            <tt class="py-name">feature_elements</tt><tt class="py-op">.</tt><tt id="link-80" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-80', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">feature_sigil</tt><tt class="py-op">)</tt> </tt>
<a name="L263"></a><tt class="py-lineno"> 263</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">label</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt>   <tt class="py-comment"># If there's a label</tt> </tt>
<a name="L264"></a><tt class="py-lineno"> 264</tt>  <tt class="py-line">                <tt class="py-name">label_elements</tt><tt class="py-op">.</tt><tt id="link-81" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-81', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">label</tt><tt class="py-op">)</tt> </tt>
<a name="L265"></a><tt class="py-lineno"> 265</tt>  <tt class="py-line"> </tt>
<a name="L266"></a><tt class="py-lineno"> 266</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">feature_elements</tt><tt class="py-op">,</tt> <tt class="py-name">label_elements</tt> </tt>
</div><a name="L267"></a><tt class="py-lineno"> 267</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer.get_feature_sigil"></a><div id="CircularDrawer.get_feature_sigil-def"><a name="L268"></a><tt class="py-lineno"> 268</tt> <a class="py-toggle" href="#" id="CircularDrawer.get_feature_sigil-toggle" onclick="return toggle('CircularDrawer.get_feature_sigil');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#get_feature_sigil">get_feature_sigil</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">feature</tt><tt class="py-op">,</tt> <tt class="py-param">locstart</tt><tt class="py-op">,</tt> <tt class="py-param">locend</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer.get_feature_sigil-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CircularDrawer.get_feature_sigil-expanded"><a name="L269"></a><tt class="py-lineno"> 269</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return graphics for feature, and any required label for it.</tt> </tt>
<a name="L270"></a><tt class="py-lineno"> 270</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L271"></a><tt class="py-lineno"> 271</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L272"></a><tt class="py-lineno"> 272</tt>  <tt class="py-line"><tt class="py-docstring">         - feature       Feature object</tt> </tt>
<a name="L273"></a><tt class="py-lineno"> 273</tt>  <tt class="py-line"><tt class="py-docstring">         - locstart      The start position of the feature</tt> </tt>
<a name="L274"></a><tt class="py-lineno"> 274</tt>  <tt class="py-line"><tt class="py-docstring">         - locend        The end position of the feature</tt> </tt>
<a name="L275"></a><tt class="py-lineno"> 275</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L276"></a><tt class="py-lineno"> 276</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L277"></a><tt class="py-lineno"> 277</tt>  <tt class="py-line">        <tt class="py-comment"># Establish the co-ordinates for the sigil</tt> </tt>
<a name="L278"></a><tt class="py-lineno"> 278</tt>  <tt class="py-line">        <tt class="py-name">btm</tt><tt class="py-op">,</tt> <tt class="py-name">ctr</tt><tt class="py-op">,</tt> <tt class="py-name">top</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">track_radii</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">current_track_level</tt><tt class="py-op">]</tt> </tt>
<a name="L279"></a><tt class="py-lineno"> 279</tt>  <tt class="py-line"> </tt>
<a name="L280"></a><tt class="py-lineno"> 280</tt>  <tt class="py-line">        <tt class="py-name">startangle</tt><tt class="py-op">,</tt> <tt class="py-name">startcos</tt><tt class="py-op">,</tt> <tt class="py-name">startsin</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-82" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.canvas_angle()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#canvas_angle"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.canvas_angle" class="py-name" href="#" onclick="return doclink('link-82', 'canvas_angle', 'link-82');">canvas_angle</a></tt><tt class="py-op">(</tt><tt class="py-name">locstart</tt><tt class="py-op">)</tt> </tt>
<a name="L281"></a><tt class="py-lineno"> 281</tt>  <tt class="py-line">        <tt class="py-name">endangle</tt><tt class="py-op">,</tt> <tt class="py-name">endcos</tt><tt class="py-op">,</tt> <tt class="py-name">endsin</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-83" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.canvas_angle" class="py-name" href="#" onclick="return doclink('link-83', 'canvas_angle', 'link-82');">canvas_angle</a></tt><tt class="py-op">(</tt><tt class="py-name">locend</tt><tt class="py-op">)</tt> </tt>
<a name="L282"></a><tt class="py-lineno"> 282</tt>  <tt class="py-line">        <tt class="py-name">midangle</tt><tt class="py-op">,</tt> <tt class="py-name">midcos</tt><tt class="py-op">,</tt> <tt class="py-name">midsin</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-84" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.canvas_angle" class="py-name" href="#" onclick="return doclink('link-84', 'canvas_angle', 'link-82');">canvas_angle</a></tt><tt class="py-op">(</tt><tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">locend</tt> <tt class="py-op">+</tt> <tt class="py-name">locstart</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-number">2</tt><tt class="py-op">)</tt> </tt>
<a name="L283"></a><tt class="py-lineno"> 283</tt>  <tt class="py-line"> </tt>
<a name="L284"></a><tt class="py-lineno"> 284</tt>  <tt class="py-line">        <tt class="py-comment"># Distribution dictionary for various ways of drawing the feature</tt> </tt>
<a name="L285"></a><tt class="py-lineno"> 285</tt>  <tt class="py-line">        <tt class="py-comment"># Each method takes the inner and outer radii, the start and end angle</tt> </tt>
<a name="L286"></a><tt class="py-lineno"> 286</tt>  <tt class="py-line">        <tt class="py-comment"># subtended at the diagram center, and the color as arguments</tt> </tt>
<a name="L287"></a><tt class="py-lineno"> 287</tt>  <tt class="py-line">        <tt class="py-name">draw_methods</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'BOX'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-85" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_sigil_box()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#_draw_sigil_box,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer._draw_sigil_box()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#_draw_sigil_box"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_sigil_box
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer._draw_sigil_box" class="py-name" href="#" onclick="return doclink('link-85', '_draw_sigil_box', 'link-85');">_draw_sigil_box</a></tt><tt class="py-op">,</tt> </tt>
<a name="L288"></a><tt class="py-lineno"> 288</tt>  <tt class="py-line">                        <tt class="py-string">'OCTO'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-86" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_sigil_cut_corner_box()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#_draw_sigil_cut_corner_box"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_sigil_cut_corner_box" class="py-name" href="#" onclick="return doclink('link-86', '_draw_sigil_cut_corner_box', 'link-86');">_draw_sigil_cut_corner_box</a></tt><tt class="py-op">,</tt> </tt>
<a name="L289"></a><tt class="py-lineno"> 289</tt>  <tt class="py-line">                        <tt class="py-string">'JAGGY'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-87" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_sigil_jaggy()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#_draw_sigil_jaggy,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer._draw_sigil_jaggy()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#_draw_sigil_jaggy"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_sigil_jaggy
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer._draw_sigil_jaggy" class="py-name" href="#" onclick="return doclink('link-87', '_draw_sigil_jaggy', 'link-87');">_draw_sigil_jaggy</a></tt><tt class="py-op">,</tt> </tt>
<a name="L290"></a><tt class="py-lineno"> 290</tt>  <tt class="py-line">                        <tt class="py-string">'ARROW'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-88" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_sigil_arrow()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#_draw_sigil_arrow,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer._draw_sigil_arrow()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#_draw_sigil_arrow"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_sigil_arrow
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer._draw_sigil_arrow" class="py-name" href="#" onclick="return doclink('link-88', '_draw_sigil_arrow', 'link-88');">_draw_sigil_arrow</a></tt><tt class="py-op">,</tt> </tt>
<a name="L291"></a><tt class="py-lineno"> 291</tt>  <tt class="py-line">                        <tt class="py-string">'BIGARROW'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-89" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_sigil_big_arrow()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#_draw_sigil_big_arrow,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer._draw_sigil_big_arrow()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#_draw_sigil_big_arrow"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_sigil_big_arrow
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer._draw_sigil_big_arrow" class="py-name" href="#" onclick="return doclink('link-89', '_draw_sigil_big_arrow', 'link-89');">_draw_sigil_big_arrow</a></tt><tt class="py-op">,</tt> </tt>
<a name="L292"></a><tt class="py-lineno"> 292</tt>  <tt class="py-line">                        <tt class="py-op">}</tt> </tt>
<a name="L293"></a><tt class="py-lineno"> 293</tt>  <tt class="py-line"> </tt>
<a name="L294"></a><tt class="py-lineno"> 294</tt>  <tt class="py-line">        <tt class="py-comment"># Get sigil for the feature, location dependent on the feature strand</tt> </tt>
<a name="L295"></a><tt class="py-lineno"> 295</tt>  <tt class="py-line">        <tt class="py-name">method</tt> <tt class="py-op">=</tt> <tt class="py-name">draw_methods</tt><tt class="py-op">[</tt><tt class="py-name">feature</tt><tt class="py-op">.</tt><tt class="py-name">sigil</tt><tt class="py-op">]</tt> </tt>
<a name="L296"></a><tt class="py-lineno"> 296</tt>  <tt class="py-line">        <tt class="py-name">kwargs</tt><tt class="py-op">[</tt><tt class="py-string">'head_length_ratio'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt class="py-name">arrowhead_length</tt> </tt>
<a name="L297"></a><tt class="py-lineno"> 297</tt>  <tt class="py-line">        <tt class="py-name">kwargs</tt><tt class="py-op">[</tt><tt class="py-string">'shaft_height_ratio'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt class="py-name">arrowshaft_height</tt> </tt>
<a name="L298"></a><tt class="py-lineno"> 298</tt>  <tt class="py-line"> </tt>
<a name="L299"></a><tt class="py-lineno"> 299</tt>  <tt class="py-line">        <tt class="py-comment"># Support for clickable links... needs ReportLab 2.4 or later</tt> </tt>
<a name="L300"></a><tt class="py-lineno"> 300</tt>  <tt class="py-line">        <tt class="py-comment"># which added support for links in SVG output.</tt> </tt>
<a name="L301"></a><tt class="py-lineno"> 301</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">feature</tt><tt class="py-op">,</tt> <tt class="py-string">"url"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L302"></a><tt class="py-lineno"> 302</tt>  <tt class="py-line">            <tt class="py-name">kwargs</tt><tt class="py-op">[</tt><tt class="py-string">"hrefURL"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt class="py-name">url</tt> </tt>
<a name="L303"></a><tt class="py-lineno"> 303</tt>  <tt class="py-line">            <tt class="py-name">kwargs</tt><tt class="py-op">[</tt><tt class="py-string">"hrefTitle"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-90" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.name=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.name=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.name=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#name,Method Bio.Phylo.PhyloXMLIO.Writer.name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#name,Variable Bio.codonalign.codonalphabet.CodonAlphabet.name=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-90', 'name', 'link-90');">name</a></tt> </tt>
<a name="L304"></a><tt class="py-lineno"> 304</tt>  <tt class="py-line"> </tt>
<a name="L305"></a><tt class="py-lineno"> 305</tt>  <tt class="py-line">        <tt class="py-name">sigil</tt> <tt class="py-op">=</tt> <tt class="py-name">method</tt><tt class="py-op">(</tt><tt class="py-name">btm</tt><tt class="py-op">,</tt> <tt class="py-name">ctr</tt><tt class="py-op">,</tt> <tt class="py-name">top</tt><tt class="py-op">,</tt> <tt class="py-name">startangle</tt><tt class="py-op">,</tt> <tt class="py-name">endangle</tt><tt class="py-op">,</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-91" class="py-name" targets="Method Bio.SearchIO._legacy.NCBIStandalone._HSPConsumer.strand()=Bio.SearchIO._legacy.NCBIStandalone._HSPConsumer-class.html#strand,Variable Bio.SeqFeature.CompoundLocation.strand=Bio.SeqFeature.CompoundLocation-class.html#strand,Variable Bio.SeqFeature.FeatureLocation.strand=Bio.SeqFeature.FeatureLocation-class.html#strand,Variable Bio.SeqFeature.SeqFeature.strand=Bio.SeqFeature.SeqFeature-class.html#strand"><a title="Bio.SearchIO._legacy.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-91', 'strand', 'link-91');">strand</a></tt><tt class="py-op">,</tt> </tt>
<a name="L306"></a><tt class="py-lineno"> 306</tt>  <tt class="py-line">                       <tt id="link-92" class="py-name" targets="Variable Bio.Phylo.BaseTree.Clade.color=Bio.Phylo.BaseTree.Clade-class.html#color,Method Bio.Phylo.PhyloXMLIO.Parser.color()=Bio.Phylo.PhyloXMLIO.Parser-class.html#color,Method Bio.Phylo.PhyloXMLIO.Writer.color()=Bio.Phylo.PhyloXMLIO.Writer-class.html#color"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-92', 'color', 'link-92');">color</a></tt><tt class="py-op">=</tt><tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-93" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-93', 'color', 'link-92');">color</a></tt><tt class="py-op">,</tt> <tt class="py-name">border</tt><tt class="py-op">=</tt><tt class="py-name">feature</tt><tt class="py-op">.</tt><tt class="py-name">border</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> </tt>
<a name="L307"></a><tt class="py-lineno"> 307</tt>  <tt class="py-line"> </tt>
<a name="L308"></a><tt class="py-lineno"> 308</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt class="py-name">label</tt><tt class="py-op">:</tt>   <tt class="py-comment"># Feature needs a label</tt> </tt>
<a name="L309"></a><tt class="py-lineno"> 309</tt>  <tt class="py-line">            <tt class="py-comment"># The spaces are a hack to force a little space between the label</tt> </tt>
<a name="L310"></a><tt class="py-lineno"> 310</tt>  <tt class="py-line">            <tt class="py-comment"># and the edge of the feature</tt> </tt>
<a name="L311"></a><tt class="py-lineno"> 311</tt>  <tt class="py-line">            <tt class="py-name">label</tt> <tt class="py-op">=</tt> <tt class="py-name">String</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-string">" %s "</tt> <tt class="py-op">%</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-94" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-94', 'name', 'link-90');">name</a></tt><tt class="py-op">.</tt><tt id="link-95" class="py-name" targets="Method Bio.Seq.Seq.strip()=Bio.Seq.Seq-class.html#strip"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-95', 'strip', 'link-95');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L312"></a><tt class="py-lineno"> 312</tt>  <tt class="py-line">                           <tt class="py-name">fontName</tt><tt class="py-op">=</tt><tt class="py-name">feature</tt><tt class="py-op">.</tt><tt class="py-name">label_font</tt><tt class="py-op">,</tt> </tt>
<a name="L313"></a><tt class="py-lineno"> 313</tt>  <tt class="py-line">                           <tt class="py-name">fontSize</tt><tt class="py-op">=</tt><tt class="py-name">feature</tt><tt class="py-op">.</tt><tt class="py-name">label_size</tt><tt class="py-op">,</tt> </tt>
<a name="L314"></a><tt class="py-lineno"> 314</tt>  <tt class="py-line">                           <tt class="py-name">fillColor</tt><tt class="py-op">=</tt><tt class="py-name">feature</tt><tt class="py-op">.</tt><tt class="py-name">label_color</tt><tt class="py-op">)</tt> </tt>
<a name="L315"></a><tt class="py-lineno"> 315</tt>  <tt class="py-line">            <tt class="py-name">labelgroup</tt> <tt class="py-op">=</tt> <tt class="py-name">Group</tt><tt class="py-op">(</tt><tt class="py-name">label</tt><tt class="py-op">)</tt> </tt>
<a name="L316"></a><tt class="py-lineno"> 316</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt class="py-name">label_strand</tt><tt class="py-op">:</tt> </tt>
<a name="L317"></a><tt class="py-lineno"> 317</tt>  <tt class="py-line">                <tt id="link-96" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-96', 'strand', 'link-91');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt class="py-name">label_strand</tt> </tt>
<a name="L318"></a><tt class="py-lineno"> 318</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L319"></a><tt class="py-lineno"> 319</tt>  <tt class="py-line">                <tt id="link-97" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-97', 'strand', 'link-91');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-98" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-98', 'strand', 'link-91');">strand</a></tt> </tt>
<a name="L320"></a><tt class="py-lineno"> 320</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt class="py-name">label_position</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'start'</tt><tt class="py-op">,</tt> <tt class="py-string">"5'"</tt><tt class="py-op">,</tt> <tt class="py-string">'left'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L321"></a><tt class="py-lineno"> 321</tt>  <tt class="py-line">                <tt class="py-comment"># Position the label at the feature's start</tt> </tt>
<a name="L322"></a><tt class="py-lineno"> 322</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-99" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-99', 'strand', 'link-91');">strand</a></tt> <tt class="py-op">!=</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L323"></a><tt class="py-lineno"> 323</tt>  <tt class="py-line">                    <tt class="py-name">label_angle</tt> <tt class="py-op">=</tt> <tt class="py-name">startangle</tt> <tt class="py-op">+</tt> <tt class="py-number">0.5</tt> <tt class="py-op">*</tt> <tt id="link-100" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-100', 'pi', 'link-14');">pi</a></tt>  <tt class="py-comment"># Make text radial</tt> </tt>
<a name="L324"></a><tt class="py-lineno"> 324</tt>  <tt class="py-line">                    <tt class="py-name">sinval</tt><tt class="py-op">,</tt> <tt class="py-name">cosval</tt> <tt class="py-op">=</tt> <tt class="py-name">startsin</tt><tt class="py-op">,</tt> <tt class="py-name">startcos</tt> </tt>
<a name="L325"></a><tt class="py-lineno"> 325</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L326"></a><tt class="py-lineno"> 326</tt>  <tt class="py-line">                    <tt class="py-name">label_angle</tt> <tt class="py-op">=</tt> <tt class="py-name">endangle</tt> <tt class="py-op">+</tt> <tt class="py-number">0.5</tt> <tt class="py-op">*</tt> <tt id="link-101" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-101', 'pi', 'link-14');">pi</a></tt>  <tt class="py-comment"># Make text radial</tt> </tt>
<a name="L327"></a><tt class="py-lineno"> 327</tt>  <tt class="py-line">                    <tt class="py-name">sinval</tt><tt class="py-op">,</tt> <tt class="py-name">cosval</tt> <tt class="py-op">=</tt> <tt class="py-name">endsin</tt><tt class="py-op">,</tt> <tt class="py-name">endcos</tt> </tt>
<a name="L328"></a><tt class="py-lineno"> 328</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt class="py-name">label_position</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'middle'</tt><tt class="py-op">,</tt> <tt class="py-string">'center'</tt><tt class="py-op">,</tt> <tt class="py-string">'centre'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L329"></a><tt class="py-lineno"> 329</tt>  <tt class="py-line">                <tt class="py-comment"># Position the label at the feature's midpoint</tt> </tt>
<a name="L330"></a><tt class="py-lineno"> 330</tt>  <tt class="py-line">                <tt class="py-name">label_angle</tt> <tt class="py-op">=</tt> <tt class="py-name">midangle</tt> <tt class="py-op">+</tt> <tt class="py-number">0.5</tt> <tt class="py-op">*</tt> <tt id="link-102" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-102', 'pi', 'link-14');">pi</a></tt>  <tt class="py-comment"># Make text radial</tt> </tt>
<a name="L331"></a><tt class="py-lineno"> 331</tt>  <tt class="py-line">                <tt class="py-name">sinval</tt><tt class="py-op">,</tt> <tt class="py-name">cosval</tt> <tt class="py-op">=</tt> <tt class="py-name">midsin</tt><tt class="py-op">,</tt> <tt class="py-name">midcos</tt> </tt>
<a name="L332"></a><tt class="py-lineno"> 332</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt class="py-name">label_position</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-string">'end'</tt><tt class="py-op">,</tt> <tt class="py-string">"3'"</tt><tt class="py-op">,</tt> <tt class="py-string">'right'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L333"></a><tt class="py-lineno"> 333</tt>  <tt class="py-line">                <tt class="py-comment"># Position the label at the feature's end</tt> </tt>
<a name="L334"></a><tt class="py-lineno"> 334</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-103" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-103', 'strand', 'link-91');">strand</a></tt> <tt class="py-op">!=</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L335"></a><tt class="py-lineno"> 335</tt>  <tt class="py-line">                    <tt class="py-name">label_angle</tt> <tt class="py-op">=</tt> <tt class="py-name">endangle</tt> <tt class="py-op">+</tt> <tt class="py-number">0.5</tt> <tt class="py-op">*</tt> <tt id="link-104" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-104', 'pi', 'link-14');">pi</a></tt>  <tt class="py-comment"># Make text radial</tt> </tt>
<a name="L336"></a><tt class="py-lineno"> 336</tt>  <tt class="py-line">                    <tt class="py-name">sinval</tt><tt class="py-op">,</tt> <tt class="py-name">cosval</tt> <tt class="py-op">=</tt> <tt class="py-name">endsin</tt><tt class="py-op">,</tt> <tt class="py-name">endcos</tt> </tt>
<a name="L337"></a><tt class="py-lineno"> 337</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L338"></a><tt class="py-lineno"> 338</tt>  <tt class="py-line">                    <tt class="py-name">label_angle</tt> <tt class="py-op">=</tt> <tt class="py-name">startangle</tt> <tt class="py-op">+</tt> <tt class="py-number">0.5</tt> <tt class="py-op">*</tt> <tt id="link-105" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-105', 'pi', 'link-14');">pi</a></tt>  <tt class="py-comment"># Make text radial</tt> </tt>
<a name="L339"></a><tt class="py-lineno"> 339</tt>  <tt class="py-line">                    <tt class="py-name">sinval</tt><tt class="py-op">,</tt> <tt class="py-name">cosval</tt> <tt class="py-op">=</tt> <tt class="py-name">startsin</tt><tt class="py-op">,</tt> <tt class="py-name">startcos</tt> </tt>
<a name="L340"></a><tt class="py-lineno"> 340</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">startangle</tt> <tt class="py-op">&lt;</tt> <tt id="link-106" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-106', 'pi', 'link-14');">pi</a></tt><tt class="py-op">:</tt> </tt>
<a name="L341"></a><tt class="py-lineno"> 341</tt>  <tt class="py-line">                <tt class="py-comment"># Default to placing the label the bottom of the feature</tt> </tt>
<a name="L342"></a><tt class="py-lineno"> 342</tt>  <tt class="py-line">                <tt class="py-comment"># as drawn on the page, meaning feature end on left half</tt> </tt>
<a name="L343"></a><tt class="py-lineno"> 343</tt>  <tt class="py-line">                <tt class="py-name">label_angle</tt> <tt class="py-op">=</tt> <tt class="py-name">endangle</tt> <tt class="py-op">+</tt> <tt class="py-number">0.5</tt> <tt class="py-op">*</tt> <tt id="link-107" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-107', 'pi', 'link-14');">pi</a></tt>  <tt class="py-comment"># Make text radial</tt> </tt>
<a name="L344"></a><tt class="py-lineno"> 344</tt>  <tt class="py-line">                <tt class="py-name">sinval</tt><tt class="py-op">,</tt> <tt class="py-name">cosval</tt> <tt class="py-op">=</tt> <tt class="py-name">endsin</tt><tt class="py-op">,</tt> <tt class="py-name">endcos</tt> </tt>
<a name="L345"></a><tt class="py-lineno"> 345</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L346"></a><tt class="py-lineno"> 346</tt>  <tt class="py-line">                <tt class="py-comment"># Default to placing the label on the bottom of the feature,</tt> </tt>
<a name="L347"></a><tt class="py-lineno"> 347</tt>  <tt class="py-line">                <tt class="py-comment"># which means the feature end when on right hand half</tt> </tt>
<a name="L348"></a><tt class="py-lineno"> 348</tt>  <tt class="py-line">                <tt class="py-name">label_angle</tt> <tt class="py-op">=</tt> <tt class="py-name">startangle</tt> <tt class="py-op">+</tt> <tt class="py-number">0.5</tt> <tt class="py-op">*</tt> <tt id="link-108" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-108', 'pi', 'link-14');">pi</a></tt>  <tt class="py-comment"># Make text radial</tt> </tt>
<a name="L349"></a><tt class="py-lineno"> 349</tt>  <tt class="py-line">                <tt class="py-name">sinval</tt><tt class="py-op">,</tt> <tt class="py-name">cosval</tt> <tt class="py-op">=</tt> <tt class="py-name">startsin</tt><tt class="py-op">,</tt> <tt class="py-name">startcos</tt> </tt>
<a name="L350"></a><tt class="py-lineno"> 350</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-109" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-109', 'strand', 'link-91');">strand</a></tt> <tt class="py-op">!=</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L351"></a><tt class="py-lineno"> 351</tt>  <tt class="py-line">                <tt class="py-comment"># Feature label on top</tt> </tt>
<a name="L352"></a><tt class="py-lineno"> 352</tt>  <tt class="py-line">                <tt class="py-name">radius</tt> <tt class="py-op">=</tt> <tt class="py-name">top</tt> </tt>
<a name="L353"></a><tt class="py-lineno"> 353</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">startangle</tt> <tt class="py-op">&lt;</tt> <tt id="link-110" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-110', 'pi', 'link-14');">pi</a></tt><tt class="py-op">:</tt>  <tt class="py-comment"># Turn text round</tt> </tt>
<a name="L354"></a><tt class="py-lineno"> 354</tt>  <tt class="py-line">                    <tt class="py-name">label_angle</tt> <tt class="py-op">-=</tt> <tt id="link-111" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-111', 'pi', 'link-14');">pi</a></tt> </tt>
<a name="L355"></a><tt class="py-lineno"> 355</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L356"></a><tt class="py-lineno"> 356</tt>  <tt class="py-line">                    <tt class="py-name">labelgroup</tt><tt class="py-op">.</tt><tt class="py-name">contents</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">textAnchor</tt> <tt class="py-op">=</tt> <tt class="py-string">'end'</tt> </tt>
<a name="L357"></a><tt class="py-lineno"> 357</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L358"></a><tt class="py-lineno"> 358</tt>  <tt class="py-line">                <tt class="py-comment"># Feature label on bottom</tt> </tt>
<a name="L359"></a><tt class="py-lineno"> 359</tt>  <tt class="py-line">                <tt class="py-name">radius</tt> <tt class="py-op">=</tt> <tt class="py-name">btm</tt> </tt>
<a name="L360"></a><tt class="py-lineno"> 360</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">startangle</tt> <tt class="py-op">&lt;</tt> <tt id="link-112" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-112', 'pi', 'link-14');">pi</a></tt><tt class="py-op">:</tt>  <tt class="py-comment"># Turn text round and anchor end</tt> </tt>
<a name="L361"></a><tt class="py-lineno"> 361</tt>  <tt class="py-line">                    <tt class="py-name">label_angle</tt> <tt class="py-op">-=</tt> <tt id="link-113" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-113', 'pi', 'link-14');">pi</a></tt> </tt>
<a name="L362"></a><tt class="py-lineno"> 362</tt>  <tt class="py-line">                    <tt class="py-name">labelgroup</tt><tt class="py-op">.</tt><tt class="py-name">contents</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt class="py-name">textAnchor</tt> <tt class="py-op">=</tt> <tt class="py-string">'end'</tt> </tt>
<a name="L363"></a><tt class="py-lineno"> 363</tt>  <tt class="py-line">            <tt class="py-name">x_pos</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt> <tt class="py-op">+</tt> <tt class="py-name">radius</tt> <tt class="py-op">*</tt> <tt class="py-name">sinval</tt> </tt>
<a name="L364"></a><tt class="py-lineno"> 364</tt>  <tt class="py-line">            <tt class="py-name">y_pos</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt> <tt class="py-op">+</tt> <tt class="py-name">radius</tt> <tt class="py-op">*</tt> <tt class="py-name">cosval</tt> </tt>
<a name="L365"></a><tt class="py-lineno"> 365</tt>  <tt class="py-line">            <tt class="py-name">coslabel</tt> <tt class="py-op">=</tt> <tt class="py-name">cos</tt><tt class="py-op">(</tt><tt class="py-name">label_angle</tt><tt class="py-op">)</tt> </tt>
<a name="L366"></a><tt class="py-lineno"> 366</tt>  <tt class="py-line">            <tt class="py-name">sinlabel</tt> <tt class="py-op">=</tt> <tt class="py-name">sin</tt><tt class="py-op">(</tt><tt class="py-name">label_angle</tt><tt class="py-op">)</tt> </tt>
<a name="L367"></a><tt class="py-lineno"> 367</tt>  <tt class="py-line">            <tt class="py-name">labelgroup</tt><tt class="py-op">.</tt><tt id="link-114" class="py-name" targets="Method Bio.PDB.Atom.Atom.transform()=Bio.PDB.Atom.Atom-class.html#transform,Method Bio.PDB.Entity.Entity.transform()=Bio.PDB.Entity.Entity-class.html#transform"><a title="Bio.PDB.Atom.Atom.transform
Bio.PDB.Entity.Entity.transform" class="py-name" href="#" onclick="return doclink('link-114', 'transform', 'link-114');">transform</a></tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">coslabel</tt><tt class="py-op">,</tt> <tt class="py-op">-</tt><tt class="py-name">sinlabel</tt><tt class="py-op">,</tt> <tt class="py-name">sinlabel</tt><tt class="py-op">,</tt> <tt class="py-name">coslabel</tt><tt class="py-op">,</tt> </tt>
<a name="L368"></a><tt class="py-lineno"> 368</tt>  <tt class="py-line">                                    <tt class="py-name">x_pos</tt><tt class="py-op">,</tt> <tt class="py-name">y_pos</tt><tt class="py-op">)</tt> </tt>
<a name="L369"></a><tt class="py-lineno"> 369</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L370"></a><tt class="py-lineno"> 370</tt>  <tt class="py-line">            <tt class="py-comment"># No label required</tt> </tt>
<a name="L371"></a><tt class="py-lineno"> 371</tt>  <tt class="py-line">            <tt class="py-name">labelgroup</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L372"></a><tt class="py-lineno"> 372</tt>  <tt class="py-line">        <tt class="py-comment"># if locstart &gt; locend:</tt> </tt>
<a name="L373"></a><tt class="py-lineno"> 373</tt>  <tt class="py-line">        <tt class="py-comment">#    print locstart, locend, feature.strand, sigil, feature.name</tt> </tt>
<a name="L374"></a><tt class="py-lineno"> 374</tt>  <tt class="py-line">        <tt class="py-comment"># print locstart, locend, feature.name</tt> </tt>
<a name="L375"></a><tt class="py-lineno"> 375</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">sigil</tt><tt class="py-op">,</tt> <tt class="py-name">labelgroup</tt> </tt>
</div><a name="L376"></a><tt class="py-lineno"> 376</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer.draw_cross_link"></a><div id="CircularDrawer.draw_cross_link-def"><a name="L377"></a><tt class="py-lineno"> 377</tt> <a class="py-toggle" href="#" id="CircularDrawer.draw_cross_link-toggle" onclick="return toggle('CircularDrawer.draw_cross_link');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw_cross_link">draw_cross_link</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">cross_link</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer.draw_cross_link-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CircularDrawer.draw_cross_link-expanded"><a name="L378"></a><tt class="py-lineno"> 378</tt>  <tt class="py-line">        <tt class="py-docstring">"""Draw a cross-link between features."""</tt> </tt>
<a name="L379"></a><tt class="py-lineno"> 379</tt>  <tt class="py-line">        <tt id="link-115" class="py-name" targets="Variable Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.startA=Bio.Graphics.GenomeDiagram._CrossLink.CrossLink-class.html#startA"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.startA" class="py-name" href="#" onclick="return doclink('link-115', 'startA', 'link-115');">startA</a></tt> <tt class="py-op">=</tt> <tt class="py-name">cross_link</tt><tt class="py-op">.</tt><tt id="link-116" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.startA" class="py-name" href="#" onclick="return doclink('link-116', 'startA', 'link-115');">startA</a></tt> </tt>
<a name="L380"></a><tt class="py-lineno"> 380</tt>  <tt class="py-line">        <tt id="link-117" class="py-name" targets="Variable Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.startB=Bio.Graphics.GenomeDiagram._CrossLink.CrossLink-class.html#startB"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.startB" class="py-name" href="#" onclick="return doclink('link-117', 'startB', 'link-117');">startB</a></tt> <tt class="py-op">=</tt> <tt class="py-name">cross_link</tt><tt class="py-op">.</tt><tt id="link-118" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.startB" class="py-name" href="#" onclick="return doclink('link-118', 'startB', 'link-117');">startB</a></tt> </tt>
<a name="L381"></a><tt class="py-lineno"> 381</tt>  <tt class="py-line">        <tt id="link-119" class="py-name" targets="Variable Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.endA=Bio.Graphics.GenomeDiagram._CrossLink.CrossLink-class.html#endA"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.endA" class="py-name" href="#" onclick="return doclink('link-119', 'endA', 'link-119');">endA</a></tt> <tt class="py-op">=</tt> <tt class="py-name">cross_link</tt><tt class="py-op">.</tt><tt id="link-120" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.endA" class="py-name" href="#" onclick="return doclink('link-120', 'endA', 'link-119');">endA</a></tt> </tt>
<a name="L382"></a><tt class="py-lineno"> 382</tt>  <tt class="py-line">        <tt id="link-121" class="py-name" targets="Variable Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.endB=Bio.Graphics.GenomeDiagram._CrossLink.CrossLink-class.html#endB"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.endB" class="py-name" href="#" onclick="return doclink('link-121', 'endB', 'link-121');">endB</a></tt> <tt class="py-op">=</tt> <tt class="py-name">cross_link</tt><tt class="py-op">.</tt><tt id="link-122" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.endB" class="py-name" href="#" onclick="return doclink('link-122', 'endB', 'link-121');">endB</a></tt> </tt>
<a name="L383"></a><tt class="py-lineno"> 383</tt>  <tt class="py-line"> </tt>
<a name="L384"></a><tt class="py-lineno"> 384</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-123" class="py-name"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.is_in_bounds" class="py-name" href="#" onclick="return doclink('link-123', 'is_in_bounds', 'link-61');">is_in_bounds</a></tt><tt class="py-op">(</tt><tt id="link-124" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.startA" class="py-name" href="#" onclick="return doclink('link-124', 'startA', 'link-115');">startA</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-125" class="py-name"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.is_in_bounds" class="py-name" href="#" onclick="return doclink('link-125', 'is_in_bounds', 'link-61');">is_in_bounds</a></tt><tt class="py-op">(</tt><tt id="link-126" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.endA" class="py-name" href="#" onclick="return doclink('link-126', 'endA', 'link-119');">endA</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L385"></a><tt class="py-lineno"> 385</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
<a name="L386"></a><tt class="py-lineno"> 386</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-127" class="py-name"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.is_in_bounds" class="py-name" href="#" onclick="return doclink('link-127', 'is_in_bounds', 'link-61');">is_in_bounds</a></tt><tt class="py-op">(</tt><tt id="link-128" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.startB" class="py-name" href="#" onclick="return doclink('link-128', 'startB', 'link-117');">startB</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-129" class="py-name"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.is_in_bounds" class="py-name" href="#" onclick="return doclink('link-129', 'is_in_bounds', 'link-61');">is_in_bounds</a></tt><tt class="py-op">(</tt><tt id="link-130" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.endB" class="py-name" href="#" onclick="return doclink('link-130', 'endB', 'link-121');">endB</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L387"></a><tt class="py-lineno"> 387</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
<a name="L388"></a><tt class="py-lineno"> 388</tt>  <tt class="py-line"> </tt>
<a name="L389"></a><tt class="py-lineno"> 389</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-131" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.startA" class="py-name" href="#" onclick="return doclink('link-131', 'startA', 'link-115');">startA</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-132" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-132', 'start', 'link-19');">start</a></tt><tt class="py-op">:</tt> </tt>
<a name="L390"></a><tt class="py-lineno"> 390</tt>  <tt class="py-line">            <tt id="link-133" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.startA" class="py-name" href="#" onclick="return doclink('link-133', 'startA', 'link-115');">startA</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-134" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-134', 'start', 'link-19');">start</a></tt> </tt>
<a name="L391"></a><tt class="py-lineno"> 391</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-135" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.startB" class="py-name" href="#" onclick="return doclink('link-135', 'startB', 'link-117');">startB</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-136" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-136', 'start', 'link-19');">start</a></tt><tt class="py-op">:</tt> </tt>
<a name="L392"></a><tt class="py-lineno"> 392</tt>  <tt class="py-line">            <tt id="link-137" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.startB" class="py-name" href="#" onclick="return doclink('link-137', 'startB', 'link-117');">startB</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-138" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-138', 'start', 'link-19');">start</a></tt> </tt>
<a name="L393"></a><tt class="py-lineno"> 393</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-139" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-139', 'end', 'link-20');">end</a></tt> <tt class="py-op">&lt;</tt> <tt id="link-140" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.endA" class="py-name" href="#" onclick="return doclink('link-140', 'endA', 'link-119');">endA</a></tt><tt class="py-op">:</tt> </tt>
<a name="L394"></a><tt class="py-lineno"> 394</tt>  <tt class="py-line">            <tt id="link-141" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.endA" class="py-name" href="#" onclick="return doclink('link-141', 'endA', 'link-119');">endA</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-142" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-142', 'end', 'link-20');">end</a></tt> </tt>
<a name="L395"></a><tt class="py-lineno"> 395</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-143" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-143', 'end', 'link-20');">end</a></tt> <tt class="py-op">&lt;</tt> <tt id="link-144" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.endB" class="py-name" href="#" onclick="return doclink('link-144', 'endB', 'link-121');">endB</a></tt><tt class="py-op">:</tt> </tt>
<a name="L396"></a><tt class="py-lineno"> 396</tt>  <tt class="py-line">            <tt id="link-145" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.endB" class="py-name" href="#" onclick="return doclink('link-145', 'endB', 'link-121');">endB</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-146" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-146', 'end', 'link-20');">end</a></tt> </tt>
<a name="L397"></a><tt class="py-lineno"> 397</tt>  <tt class="py-line"> </tt>
<a name="L398"></a><tt class="py-lineno"> 398</tt>  <tt class="py-line">        <tt class="py-name">trackobjA</tt> <tt class="py-op">=</tt> <tt class="py-name">cross_link</tt><tt class="py-op">.</tt><tt id="link-147" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._CrossLink.CrossLink._trackA()=Bio.Graphics.GenomeDiagram._CrossLink.CrossLink-class.html#_trackA"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink._trackA" class="py-name" href="#" onclick="return doclink('link-147', '_trackA', 'link-147');">_trackA</a></tt><tt class="py-op">(</tt><tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_parent</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt><tt class="py-op">.</tt><tt id="link-148" class="py-name" targets="Method Bio.Crystal.Crystal.values()=Bio.Crystal.Crystal-class.html#values,Method Bio.File._IndexedSeqFileDict.values()=Bio.File._IndexedSeqFileDict-class.html#values,Method Bio.Phylo.PhyloXML.Events.values()=Bio.Phylo.PhyloXML.Events-class.html#values,Method BioSQL.BioSeqDatabase.BioSeqDatabase.values()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#values,Method BioSQL.BioSeqDatabase.DBServer.values()=BioSQL.BioSeqDatabase.DBServer-class.html#values"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-148', 'values', 'link-148');">values</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L399"></a><tt class="py-lineno"> 399</tt>  <tt class="py-line">        <tt class="py-name">trackobjB</tt> <tt class="py-op">=</tt> <tt class="py-name">cross_link</tt><tt class="py-op">.</tt><tt id="link-149" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._CrossLink.CrossLink._trackB()=Bio.Graphics.GenomeDiagram._CrossLink.CrossLink-class.html#_trackB"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink._trackB" class="py-name" href="#" onclick="return doclink('link-149', '_trackB', 'link-149');">_trackB</a></tt><tt class="py-op">(</tt><tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_parent</tt><tt class="py-op">.</tt><tt class="py-name">tracks</tt><tt class="py-op">.</tt><tt id="link-150" class="py-name"><a title="Bio.Crystal.Crystal.values
Bio.File._IndexedSeqFileDict.values
Bio.Phylo.PhyloXML.Events.values
BioSQL.BioSeqDatabase.BioSeqDatabase.values
BioSQL.BioSeqDatabase.DBServer.values" class="py-name" href="#" onclick="return doclink('link-150', 'values', 'link-148');">values</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L400"></a><tt class="py-lineno"> 400</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">trackobjA</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> </tt>
<a name="L401"></a><tt class="py-lineno"> 401</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">trackobjB</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> </tt>
<a name="L402"></a><tt class="py-lineno"> 402</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">trackobjA</tt> <tt class="py-op">==</tt> <tt class="py-name">trackobjB</tt><tt class="py-op">:</tt> </tt>
<a name="L403"></a><tt class="py-lineno"> 403</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">NotImplementedError</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L404"></a><tt class="py-lineno"> 404</tt>  <tt class="py-line"> </tt>
<a name="L405"></a><tt class="py-lineno"> 405</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">trackobjA</tt><tt class="py-op">.</tt><tt id="link-151" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-151', 'start', 'link-19');">start</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L406"></a><tt class="py-lineno"> 406</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-152" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.endA" class="py-name" href="#" onclick="return doclink('link-152', 'endA', 'link-119');">endA</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-name">trackobjA</tt><tt class="py-op">.</tt><tt id="link-153" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-153', 'start', 'link-19');">start</a></tt><tt class="py-op">:</tt> </tt>
<a name="L407"></a><tt class="py-lineno"> 407</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> </tt>
<a name="L408"></a><tt class="py-lineno"> 408</tt>  <tt class="py-line">            <tt id="link-154" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.startA" class="py-name" href="#" onclick="return doclink('link-154', 'startA', 'link-115');">startA</a></tt> <tt class="py-op">=</tt> <tt id="link-155" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-155', 'max', 'link-23');">max</a></tt><tt class="py-op">(</tt><tt id="link-156" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.startA" class="py-name" href="#" onclick="return doclink('link-156', 'startA', 'link-115');">startA</a></tt><tt class="py-op">,</tt> <tt class="py-name">trackobjA</tt><tt class="py-op">.</tt><tt id="link-157" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-157', 'start', 'link-19');">start</a></tt><tt class="py-op">)</tt> </tt>
<a name="L409"></a><tt class="py-lineno"> 409</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">trackobjA</tt><tt class="py-op">.</tt><tt id="link-158" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-158', 'end', 'link-20');">end</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L410"></a><tt class="py-lineno"> 410</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">trackobjA</tt><tt class="py-op">.</tt><tt id="link-159" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-159', 'end', 'link-20');">end</a></tt> <tt class="py-op">&lt;</tt> <tt id="link-160" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.startA" class="py-name" href="#" onclick="return doclink('link-160', 'startA', 'link-115');">startA</a></tt><tt class="py-op">:</tt> </tt>
<a name="L411"></a><tt class="py-lineno"> 411</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> </tt>
<a name="L412"></a><tt class="py-lineno"> 412</tt>  <tt class="py-line">            <tt id="link-161" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.endA" class="py-name" href="#" onclick="return doclink('link-161', 'endA', 'link-119');">endA</a></tt> <tt class="py-op">=</tt> <tt id="link-162" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-162', 'min', 'link-21');">min</a></tt><tt class="py-op">(</tt><tt id="link-163" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.endA" class="py-name" href="#" onclick="return doclink('link-163', 'endA', 'link-119');">endA</a></tt><tt class="py-op">,</tt> <tt class="py-name">trackobjA</tt><tt class="py-op">.</tt><tt id="link-164" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-164', 'end', 'link-20');">end</a></tt><tt class="py-op">)</tt> </tt>
<a name="L413"></a><tt class="py-lineno"> 413</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">trackobjB</tt><tt class="py-op">.</tt><tt id="link-165" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-165', 'start', 'link-19');">start</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L414"></a><tt class="py-lineno"> 414</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-166" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.endB" class="py-name" href="#" onclick="return doclink('link-166', 'endB', 'link-121');">endB</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-name">trackobjB</tt><tt class="py-op">.</tt><tt id="link-167" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-167', 'start', 'link-19');">start</a></tt><tt class="py-op">:</tt> </tt>
<a name="L415"></a><tt class="py-lineno"> 415</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> </tt>
<a name="L416"></a><tt class="py-lineno"> 416</tt>  <tt class="py-line">            <tt id="link-168" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.startB" class="py-name" href="#" onclick="return doclink('link-168', 'startB', 'link-117');">startB</a></tt> <tt class="py-op">=</tt> <tt id="link-169" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-169', 'max', 'link-23');">max</a></tt><tt class="py-op">(</tt><tt id="link-170" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.startB" class="py-name" href="#" onclick="return doclink('link-170', 'startB', 'link-117');">startB</a></tt><tt class="py-op">,</tt> <tt class="py-name">trackobjB</tt><tt class="py-op">.</tt><tt id="link-171" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-171', 'start', 'link-19');">start</a></tt><tt class="py-op">)</tt> </tt>
<a name="L417"></a><tt class="py-lineno"> 417</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">trackobjB</tt><tt class="py-op">.</tt><tt id="link-172" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-172', 'end', 'link-20');">end</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L418"></a><tt class="py-lineno"> 418</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">trackobjB</tt><tt class="py-op">.</tt><tt id="link-173" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-173', 'end', 'link-20');">end</a></tt> <tt class="py-op">&lt;</tt> <tt id="link-174" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.startB" class="py-name" href="#" onclick="return doclink('link-174', 'startB', 'link-117');">startB</a></tt><tt class="py-op">:</tt> </tt>
<a name="L419"></a><tt class="py-lineno"> 419</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> </tt>
<a name="L420"></a><tt class="py-lineno"> 420</tt>  <tt class="py-line">            <tt id="link-175" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.endB" class="py-name" href="#" onclick="return doclink('link-175', 'endB', 'link-121');">endB</a></tt> <tt class="py-op">=</tt> <tt id="link-176" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-176', 'min', 'link-21');">min</a></tt><tt class="py-op">(</tt><tt id="link-177" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.endB" class="py-name" href="#" onclick="return doclink('link-177', 'endB', 'link-121');">endB</a></tt><tt class="py-op">,</tt> <tt class="py-name">trackobjB</tt><tt class="py-op">.</tt><tt id="link-178" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-178', 'end', 'link-20');">end</a></tt><tt class="py-op">)</tt> </tt>
<a name="L421"></a><tt class="py-lineno"> 421</tt>  <tt class="py-line"> </tt>
<a name="L422"></a><tt class="py-lineno"> 422</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">track_level</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_parent</tt><tt class="py-op">.</tt><tt id="link-179" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.get_drawn_levels" class="py-name" href="#" onclick="return doclink('link-179', 'get_drawn_levels', 'link-27');">get_drawn_levels</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L423"></a><tt class="py-lineno"> 423</tt>  <tt class="py-line">            <tt class="py-name">track</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_parent</tt><tt class="py-op">[</tt><tt class="py-name">track_level</tt><tt class="py-op">]</tt> </tt>
<a name="L424"></a><tt class="py-lineno"> 424</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">track</tt> <tt class="py-op">==</tt> <tt class="py-name">trackobjA</tt><tt class="py-op">:</tt> </tt>
<a name="L425"></a><tt class="py-lineno"> 425</tt>  <tt class="py-line">                <tt class="py-name">trackA</tt> <tt class="py-op">=</tt> <tt class="py-name">track_level</tt> </tt>
<a name="L426"></a><tt class="py-lineno"> 426</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">track</tt> <tt class="py-op">==</tt> <tt class="py-name">trackobjB</tt><tt class="py-op">:</tt> </tt>
<a name="L427"></a><tt class="py-lineno"> 427</tt>  <tt class="py-line">                <tt class="py-name">trackB</tt> <tt class="py-op">=</tt> <tt class="py-name">track_level</tt> </tt>
<a name="L428"></a><tt class="py-lineno"> 428</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">trackA</tt> <tt class="py-op">==</tt> <tt class="py-name">trackB</tt><tt class="py-op">:</tt> </tt>
<a name="L429"></a><tt class="py-lineno"> 429</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">NotImplementedError</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L430"></a><tt class="py-lineno"> 430</tt>  <tt class="py-line"> </tt>
<a name="L431"></a><tt class="py-lineno"> 431</tt>  <tt class="py-line">        <tt class="py-name">startangleA</tt><tt class="py-op">,</tt> <tt class="py-name">startcosA</tt><tt class="py-op">,</tt> <tt class="py-name">startsinA</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-180" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.canvas_angle" class="py-name" href="#" onclick="return doclink('link-180', 'canvas_angle', 'link-82');">canvas_angle</a></tt><tt class="py-op">(</tt><tt id="link-181" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.startA" class="py-name" href="#" onclick="return doclink('link-181', 'startA', 'link-115');">startA</a></tt><tt class="py-op">)</tt> </tt>
<a name="L432"></a><tt class="py-lineno"> 432</tt>  <tt class="py-line">        <tt class="py-name">startangleB</tt><tt class="py-op">,</tt> <tt class="py-name">startcosB</tt><tt class="py-op">,</tt> <tt class="py-name">startsinB</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-182" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.canvas_angle" class="py-name" href="#" onclick="return doclink('link-182', 'canvas_angle', 'link-82');">canvas_angle</a></tt><tt class="py-op">(</tt><tt id="link-183" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.startB" class="py-name" href="#" onclick="return doclink('link-183', 'startB', 'link-117');">startB</a></tt><tt class="py-op">)</tt> </tt>
<a name="L433"></a><tt class="py-lineno"> 433</tt>  <tt class="py-line">        <tt class="py-name">endangleA</tt><tt class="py-op">,</tt> <tt class="py-name">endcosA</tt><tt class="py-op">,</tt> <tt class="py-name">endsinA</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-184" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.canvas_angle" class="py-name" href="#" onclick="return doclink('link-184', 'canvas_angle', 'link-82');">canvas_angle</a></tt><tt class="py-op">(</tt><tt id="link-185" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.endA" class="py-name" href="#" onclick="return doclink('link-185', 'endA', 'link-119');">endA</a></tt><tt class="py-op">)</tt> </tt>
<a name="L434"></a><tt class="py-lineno"> 434</tt>  <tt class="py-line">        <tt class="py-name">endangleB</tt><tt class="py-op">,</tt> <tt class="py-name">endcosB</tt><tt class="py-op">,</tt> <tt class="py-name">endsinB</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-186" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.canvas_angle" class="py-name" href="#" onclick="return doclink('link-186', 'canvas_angle', 'link-82');">canvas_angle</a></tt><tt class="py-op">(</tt><tt id="link-187" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.endB" class="py-name" href="#" onclick="return doclink('link-187', 'endB', 'link-121');">endB</a></tt><tt class="py-op">)</tt> </tt>
<a name="L435"></a><tt class="py-lineno"> 435</tt>  <tt class="py-line"> </tt>
<a name="L436"></a><tt class="py-lineno"> 436</tt>  <tt class="py-line">        <tt class="py-name">btmA</tt><tt class="py-op">,</tt> <tt class="py-name">ctrA</tt><tt class="py-op">,</tt> <tt class="py-name">topA</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">track_radii</tt><tt class="py-op">[</tt><tt class="py-name">trackA</tt><tt class="py-op">]</tt> </tt>
<a name="L437"></a><tt class="py-lineno"> 437</tt>  <tt class="py-line">        <tt class="py-name">btmB</tt><tt class="py-op">,</tt> <tt class="py-name">ctrB</tt><tt class="py-op">,</tt> <tt class="py-name">topB</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">track_radii</tt><tt class="py-op">[</tt><tt class="py-name">trackB</tt><tt class="py-op">]</tt> </tt>
<a name="L438"></a><tt class="py-lineno"> 438</tt>  <tt class="py-line"> </tt>
<a name="L439"></a><tt class="py-lineno"> 439</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">ctrA</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">ctrB</tt><tt class="py-op">:</tt> </tt>
<a name="L440"></a><tt class="py-lineno"> 440</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-188" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_arc_poly()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#_draw_arc_poly"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_arc_poly" class="py-name" href="#" onclick="return doclink('link-188', '_draw_arc_poly', 'link-188');">_draw_arc_poly</a></tt><tt class="py-op">(</tt><tt class="py-name">topA</tt><tt class="py-op">,</tt> <tt class="py-name">btmB</tt><tt class="py-op">,</tt> </tt>
<a name="L441"></a><tt class="py-lineno"> 441</tt>  <tt class="py-line">                                        <tt class="py-name">startangleA</tt><tt class="py-op">,</tt> <tt class="py-name">endangleA</tt><tt class="py-op">,</tt> </tt>
<a name="L442"></a><tt class="py-lineno"> 442</tt>  <tt class="py-line">                                        <tt class="py-name">startangleB</tt><tt class="py-op">,</tt> <tt class="py-name">endangleB</tt><tt class="py-op">,</tt> </tt>
<a name="L443"></a><tt class="py-lineno"> 443</tt>  <tt class="py-line">                                        <tt class="py-name">cross_link</tt><tt class="py-op">.</tt><tt id="link-189" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-189', 'color', 'link-92');">color</a></tt><tt class="py-op">,</tt> <tt class="py-name">cross_link</tt><tt class="py-op">.</tt><tt class="py-name">border</tt><tt class="py-op">,</tt> </tt>
<a name="L444"></a><tt class="py-lineno"> 444</tt>  <tt class="py-line">                                        <tt class="py-name">cross_link</tt><tt class="py-op">.</tt><tt class="py-name">flip</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L445"></a><tt class="py-lineno"> 445</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L446"></a><tt class="py-lineno"> 446</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-190" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_arc_poly" class="py-name" href="#" onclick="return doclink('link-190', '_draw_arc_poly', 'link-188');">_draw_arc_poly</a></tt><tt class="py-op">(</tt><tt class="py-name">btmA</tt><tt class="py-op">,</tt> <tt class="py-name">topB</tt><tt class="py-op">,</tt> </tt>
<a name="L447"></a><tt class="py-lineno"> 447</tt>  <tt class="py-line">                                        <tt class="py-name">startangleA</tt><tt class="py-op">,</tt> <tt class="py-name">endangleA</tt><tt class="py-op">,</tt> </tt>
<a name="L448"></a><tt class="py-lineno"> 448</tt>  <tt class="py-line">                                        <tt class="py-name">startangleB</tt><tt class="py-op">,</tt> <tt class="py-name">endangleB</tt><tt class="py-op">,</tt> </tt>
<a name="L449"></a><tt class="py-lineno"> 449</tt>  <tt class="py-line">                                        <tt class="py-name">cross_link</tt><tt class="py-op">.</tt><tt id="link-191" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-191', 'color', 'link-92');">color</a></tt><tt class="py-op">,</tt> <tt class="py-name">cross_link</tt><tt class="py-op">.</tt><tt class="py-name">border</tt><tt class="py-op">,</tt> </tt>
<a name="L450"></a><tt class="py-lineno"> 450</tt>  <tt class="py-line">                                        <tt class="py-name">cross_link</tt><tt class="py-op">.</tt><tt class="py-name">flip</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
</div><a name="L451"></a><tt class="py-lineno"> 451</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer.draw_graph_set"></a><div id="CircularDrawer.draw_graph_set-def"><a name="L452"></a><tt class="py-lineno"> 452</tt> <a class="py-toggle" href="#" id="CircularDrawer.draw_graph_set-toggle" onclick="return toggle('CircularDrawer.draw_graph_set');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw_graph_set">draw_graph_set</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">set</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer.draw_graph_set-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CircularDrawer.draw_graph_set-expanded"><a name="L453"></a><tt class="py-lineno"> 453</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return list of graph elements and list of their labels.</tt> </tt>
<a name="L454"></a><tt class="py-lineno"> 454</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L455"></a><tt class="py-lineno"> 455</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L456"></a><tt class="py-lineno"> 456</tt>  <tt class="py-line"><tt class="py-docstring">         - set       GraphSet object</tt> </tt>
<a name="L457"></a><tt class="py-lineno"> 457</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L458"></a><tt class="py-lineno"> 458</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L459"></a><tt class="py-lineno"> 459</tt>  <tt class="py-line">        <tt class="py-comment"># print 'draw graph set'</tt> </tt>
<a name="L460"></a><tt class="py-lineno"> 460</tt>  <tt class="py-line">        <tt id="link-192" class="py-name"><a title="Bio.Restriction.Restriction.RestrictionBatch.elements" class="py-name" href="#" onclick="return doclink('link-192', 'elements', 'link-53');">elements</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>  <tt class="py-comment"># Holds graph elements</tt> </tt>
<a name="L461"></a><tt class="py-lineno"> 461</tt>  <tt class="py-line"> </tt>
<a name="L462"></a><tt class="py-lineno"> 462</tt>  <tt class="py-line">        <tt class="py-comment"># Distribution dictionary for how to draw the graph</tt> </tt>
<a name="L463"></a><tt class="py-lineno"> 463</tt>  <tt class="py-line">        <tt class="py-name">style_methods</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-string">'line'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-193" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw_line_graph()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw_line_graph,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw_line_graph()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#draw_line_graph"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw_line_graph
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw_line_graph" class="py-name" href="#" onclick="return doclink('link-193', 'draw_line_graph', 'link-193');">draw_line_graph</a></tt><tt class="py-op">,</tt> </tt>
<a name="L464"></a><tt class="py-lineno"> 464</tt>  <tt class="py-line">                         <tt class="py-string">'heat'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-194" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw_heat_graph()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw_heat_graph,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw_heat_graph()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#draw_heat_graph"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw_heat_graph
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw_heat_graph" class="py-name" href="#" onclick="return doclink('link-194', 'draw_heat_graph', 'link-194');">draw_heat_graph</a></tt><tt class="py-op">,</tt> </tt>
<a name="L465"></a><tt class="py-lineno"> 465</tt>  <tt class="py-line">                         <tt class="py-string">'bar'</tt><tt class="py-op">:</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-195" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw_bar_graph()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw_bar_graph,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw_bar_graph()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#draw_bar_graph"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw_bar_graph
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw_bar_graph" class="py-name" href="#" onclick="return doclink('link-195', 'draw_bar_graph', 'link-195');">draw_bar_graph</a></tt> </tt>
<a name="L466"></a><tt class="py-lineno"> 466</tt>  <tt class="py-line">                         <tt class="py-op">}</tt> </tt>
<a name="L467"></a><tt class="py-lineno"> 467</tt>  <tt class="py-line"> </tt>
<a name="L468"></a><tt class="py-lineno"> 468</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">graph</tt> <tt class="py-keyword">in</tt> <tt id="link-196" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-196', 'set', 'link-51');">set</a></tt><tt class="py-op">.</tt><tt id="link-197" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.get_graphs()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#get_graphs"><a title="Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.get_graphs" class="py-name" href="#" onclick="return doclink('link-197', 'get_graphs', 'link-197');">get_graphs</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L469"></a><tt class="py-lineno"> 469</tt>  <tt class="py-line">            <tt id="link-198" class="py-name"><a title="Bio.Restriction.Restriction.RestrictionBatch.elements" class="py-name" href="#" onclick="return doclink('link-198', 'elements', 'link-53');">elements</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">style_methods</tt><tt class="py-op">[</tt><tt class="py-name">graph</tt><tt class="py-op">.</tt><tt class="py-name">style</tt><tt class="py-op">]</tt><tt class="py-op">(</tt><tt class="py-name">graph</tt><tt class="py-op">)</tt> </tt>
<a name="L470"></a><tt class="py-lineno"> 470</tt>  <tt class="py-line"> </tt>
<a name="L471"></a><tt class="py-lineno"> 471</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-199" class="py-name"><a title="Bio.Restriction.Restriction.RestrictionBatch.elements" class="py-name" href="#" onclick="return doclink('link-199', 'elements', 'link-53');">elements</a></tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
</div><a name="L472"></a><tt class="py-lineno"> 472</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer.draw_line_graph"></a><div id="CircularDrawer.draw_line_graph-def"><a name="L473"></a><tt class="py-lineno"> 473</tt> <a class="py-toggle" href="#" id="CircularDrawer.draw_line_graph-toggle" onclick="return toggle('CircularDrawer.draw_line_graph');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw_line_graph">draw_line_graph</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">graph</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer.draw_line_graph-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CircularDrawer.draw_line_graph-expanded"><a name="L474"></a><tt class="py-lineno"> 474</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return line graph as list of drawable elements.</tt> </tt>
<a name="L475"></a><tt class="py-lineno"> 475</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L476"></a><tt class="py-lineno"> 476</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L477"></a><tt class="py-lineno"> 477</tt>  <tt class="py-line"><tt class="py-docstring">         - graph     GraphData object</tt> </tt>
<a name="L478"></a><tt class="py-lineno"> 478</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L479"></a><tt class="py-lineno"> 479</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L480"></a><tt class="py-lineno"> 480</tt>  <tt class="py-line">        <tt class="py-name">line_elements</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>  <tt class="py-comment"># holds drawable elements</tt> </tt>
<a name="L481"></a><tt class="py-lineno"> 481</tt>  <tt class="py-line"> </tt>
<a name="L482"></a><tt class="py-lineno"> 482</tt>  <tt class="py-line">        <tt class="py-comment"># Get graph data</tt> </tt>
<a name="L483"></a><tt class="py-lineno"> 483</tt>  <tt class="py-line">        <tt id="link-200" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.data_quartiles()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#data_quartiles"><a title="Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.data_quartiles" class="py-name" href="#" onclick="return doclink('link-200', 'data_quartiles', 'link-200');">data_quartiles</a></tt> <tt class="py-op">=</tt> <tt class="py-name">graph</tt><tt class="py-op">.</tt><tt id="link-201" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Graph.GraphData.quartiles()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#quartiles"><a title="Bio.Graphics.GenomeDiagram._Graph.GraphData.quartiles" class="py-name" href="#" onclick="return doclink('link-201', 'quartiles', 'link-201');">quartiles</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L484"></a><tt class="py-lineno"> 484</tt>  <tt class="py-line">        <tt class="py-name">minval</tt><tt class="py-op">,</tt> <tt class="py-name">maxval</tt> <tt class="py-op">=</tt> <tt id="link-202" class="py-name"><a title="Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.data_quartiles" class="py-name" href="#" onclick="return doclink('link-202', 'data_quartiles', 'link-200');">data_quartiles</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt id="link-203" class="py-name"><a title="Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.data_quartiles" class="py-name" href="#" onclick="return doclink('link-203', 'data_quartiles', 'link-200');">data_quartiles</a></tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt> </tt>
<a name="L485"></a><tt class="py-lineno"> 485</tt>  <tt class="py-line">        <tt class="py-name">btm</tt><tt class="py-op">,</tt> <tt class="py-name">ctr</tt><tt class="py-op">,</tt> <tt class="py-name">top</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">track_radii</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">current_track_level</tt><tt class="py-op">]</tt> </tt>
<a name="L486"></a><tt class="py-lineno"> 486</tt>  <tt class="py-line">        <tt class="py-name">trackheight</tt> <tt class="py-op">=</tt> <tt class="py-number">0.5</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-name">top</tt> <tt class="py-op">-</tt> <tt class="py-name">btm</tt><tt class="py-op">)</tt> </tt>
<a name="L487"></a><tt class="py-lineno"> 487</tt>  <tt class="py-line">        <tt class="py-name">datarange</tt> <tt class="py-op">=</tt> <tt class="py-name">maxval</tt> <tt class="py-op">-</tt> <tt class="py-name">minval</tt> </tt>
<a name="L488"></a><tt class="py-lineno"> 488</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">datarange</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L489"></a><tt class="py-lineno"> 489</tt>  <tt class="py-line">            <tt class="py-name">datarange</tt> <tt class="py-op">=</tt> <tt class="py-name">trackheight</tt> </tt>
<a name="L490"></a><tt class="py-lineno"> 490</tt>  <tt class="py-line"> </tt>
<a name="L491"></a><tt class="py-lineno"> 491</tt>  <tt class="py-line">        <tt id="link-204" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-204', 'start', 'link-19');">start</a></tt><tt class="py-op">,</tt> <tt id="link-205" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-205', 'end', 'link-20');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-206" class="py-name"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer._current_track_start_end" class="py-name" href="#" onclick="return doclink('link-206', '_current_track_start_end', 'link-72');">_current_track_start_end</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L492"></a><tt class="py-lineno"> 492</tt>  <tt class="py-line">        <tt id="link-207" class="py-name" targets="Variable BioSQL.BioSeq.DBSeq.data=BioSQL.BioSeq.DBSeq-class.html#data"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-207', 'data', 'link-207');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-name">graph</tt><tt class="py-op">[</tt><tt id="link-208" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-208', 'start', 'link-19');">start</a></tt><tt class="py-op">:</tt><tt id="link-209" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-209', 'end', 'link-20');">end</a></tt><tt class="py-op">]</tt> </tt>
<a name="L493"></a><tt class="py-lineno"> 493</tt>  <tt class="py-line"> </tt>
<a name="L494"></a><tt class="py-lineno"> 494</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-210" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-210', 'data', 'link-207');">data</a></tt><tt class="py-op">:</tt> </tt>
<a name="L495"></a><tt class="py-lineno"> 495</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L496"></a><tt class="py-lineno"> 496</tt>  <tt class="py-line"> </tt>
<a name="L497"></a><tt class="py-lineno"> 497</tt>  <tt class="py-line">        <tt class="py-comment"># midval is the value at which the x-axis is plotted, and is the</tt> </tt>
<a name="L498"></a><tt class="py-lineno"> 498</tt>  <tt class="py-line">        <tt class="py-comment"># central ring in the track</tt> </tt>
<a name="L499"></a><tt class="py-lineno"> 499</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">graph</tt><tt class="py-op">.</tt><tt class="py-name">center</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L500"></a><tt class="py-lineno"> 500</tt>  <tt class="py-line">            <tt class="py-name">midval</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">maxval</tt> <tt class="py-op">+</tt> <tt class="py-name">minval</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-number">2.</tt> </tt>
<a name="L501"></a><tt class="py-lineno"> 501</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L502"></a><tt class="py-lineno"> 502</tt>  <tt class="py-line">            <tt class="py-name">midval</tt> <tt class="py-op">=</tt> <tt class="py-name">graph</tt><tt class="py-op">.</tt><tt class="py-name">center</tt> </tt>
<a name="L503"></a><tt class="py-lineno"> 503</tt>  <tt class="py-line">        <tt class="py-comment"># Whichever is the greatest difference: max-midval or min-midval, is</tt> </tt>
<a name="L504"></a><tt class="py-lineno"> 504</tt>  <tt class="py-line">        <tt class="py-comment"># taken to specify the number of pixel units resolved along the</tt> </tt>
<a name="L505"></a><tt class="py-lineno"> 505</tt>  <tt class="py-line">        <tt class="py-comment"># y-axis</tt> </tt>
<a name="L506"></a><tt class="py-lineno"> 506</tt>  <tt class="py-line">        <tt class="py-name">resolution</tt> <tt class="py-op">=</tt> <tt id="link-211" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-211', 'max', 'link-23');">max</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">midval</tt> <tt class="py-op">-</tt> <tt class="py-name">minval</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">maxval</tt> <tt class="py-op">-</tt> <tt class="py-name">midval</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L507"></a><tt class="py-lineno"> 507</tt>  <tt class="py-line"> </tt>
<a name="L508"></a><tt class="py-lineno"> 508</tt>  <tt class="py-line">        <tt class="py-comment"># Start from first data point</tt> </tt>
<a name="L509"></a><tt class="py-lineno"> 509</tt>  <tt class="py-line">        <tt class="py-name">pos</tt><tt class="py-op">,</tt> <tt class="py-name">val</tt> <tt class="py-op">=</tt> <tt id="link-212" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-212', 'data', 'link-207');">data</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L510"></a><tt class="py-lineno"> 510</tt>  <tt class="py-line">        <tt class="py-name">lastangle</tt><tt class="py-op">,</tt> <tt class="py-name">lastcos</tt><tt class="py-op">,</tt> <tt class="py-name">lastsin</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-213" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.canvas_angle" class="py-name" href="#" onclick="return doclink('link-213', 'canvas_angle', 'link-82');">canvas_angle</a></tt><tt class="py-op">(</tt><tt class="py-name">pos</tt><tt class="py-op">)</tt> </tt>
<a name="L511"></a><tt class="py-lineno"> 511</tt>  <tt class="py-line">        <tt class="py-comment"># We calculate the track height</tt> </tt>
<a name="L512"></a><tt class="py-lineno"> 512</tt>  <tt class="py-line">        <tt class="py-name">posheight</tt> <tt class="py-op">=</tt> <tt class="py-name">trackheight</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-name">val</tt> <tt class="py-op">-</tt> <tt class="py-name">midval</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-name">resolution</tt> <tt class="py-op">+</tt> <tt class="py-name">ctr</tt> </tt>
<a name="L513"></a><tt class="py-lineno"> 513</tt>  <tt class="py-line">        <tt class="py-name">lastx</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt> <tt class="py-op">+</tt> <tt class="py-name">posheight</tt> <tt class="py-op">*</tt> <tt class="py-name">lastsin</tt>  <tt class="py-comment"># start xy coords</tt> </tt>
<a name="L514"></a><tt class="py-lineno"> 514</tt>  <tt class="py-line">        <tt class="py-name">lasty</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt> <tt class="py-op">+</tt> <tt class="py-name">posheight</tt> <tt class="py-op">*</tt> <tt class="py-name">lastcos</tt> </tt>
<a name="L515"></a><tt class="py-lineno"> 515</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">pos</tt><tt class="py-op">,</tt> <tt class="py-name">val</tt> <tt class="py-keyword">in</tt> <tt id="link-214" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-214', 'data', 'link-207');">data</a></tt><tt class="py-op">:</tt> </tt>
<a name="L516"></a><tt class="py-lineno"> 516</tt>  <tt class="py-line">            <tt class="py-name">posangle</tt><tt class="py-op">,</tt> <tt class="py-name">poscos</tt><tt class="py-op">,</tt> <tt class="py-name">possin</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-215" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.canvas_angle" class="py-name" href="#" onclick="return doclink('link-215', 'canvas_angle', 'link-82');">canvas_angle</a></tt><tt class="py-op">(</tt><tt class="py-name">pos</tt><tt class="py-op">)</tt> </tt>
<a name="L517"></a><tt class="py-lineno"> 517</tt>  <tt class="py-line">            <tt class="py-name">posheight</tt> <tt class="py-op">=</tt> <tt class="py-name">trackheight</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-name">val</tt> <tt class="py-op">-</tt> <tt class="py-name">midval</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-name">resolution</tt> <tt class="py-op">+</tt> <tt class="py-name">ctr</tt> </tt>
<a name="L518"></a><tt class="py-lineno"> 518</tt>  <tt class="py-line">            <tt id="link-216" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-216', 'x', 'link-17');">x</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt> <tt class="py-op">+</tt> <tt class="py-name">posheight</tt> <tt class="py-op">*</tt> <tt class="py-name">possin</tt>   <tt class="py-comment"># next xy coords</tt> </tt>
<a name="L519"></a><tt class="py-lineno"> 519</tt>  <tt class="py-line">            <tt id="link-217" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-217', 'y', 'link-18');">y</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt> <tt class="py-op">+</tt> <tt class="py-name">posheight</tt> <tt class="py-op">*</tt> <tt class="py-name">poscos</tt> </tt>
<a name="L520"></a><tt class="py-lineno"> 520</tt>  <tt class="py-line">            <tt class="py-name">line_elements</tt><tt class="py-op">.</tt><tt id="link-218" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-218', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">Line</tt><tt class="py-op">(</tt><tt class="py-name">lastx</tt><tt class="py-op">,</tt> <tt class="py-name">lasty</tt><tt class="py-op">,</tt> <tt id="link-219" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-219', 'x', 'link-17');">x</a></tt><tt class="py-op">,</tt> <tt id="link-220" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-220', 'y', 'link-18');">y</a></tt><tt class="py-op">,</tt> </tt>
<a name="L521"></a><tt class="py-lineno"> 521</tt>  <tt class="py-line">                                      <tt class="py-name">strokeColor</tt><tt class="py-op">=</tt><tt class="py-name">graph</tt><tt class="py-op">.</tt><tt class="py-name">poscolor</tt><tt class="py-op">,</tt> </tt>
<a name="L522"></a><tt class="py-lineno"> 522</tt>  <tt class="py-line">                                      <tt class="py-name">strokeWidth</tt><tt class="py-op">=</tt><tt class="py-name">graph</tt><tt class="py-op">.</tt><tt class="py-name">linewidth</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L523"></a><tt class="py-lineno"> 523</tt>  <tt class="py-line">            <tt class="py-name">lastx</tt><tt class="py-op">,</tt> <tt class="py-name">lasty</tt><tt class="py-op">,</tt> <tt class="py-op">=</tt> <tt id="link-221" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-221', 'x', 'link-17');">x</a></tt><tt class="py-op">,</tt> <tt id="link-222" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-222', 'y', 'link-18');">y</a></tt> </tt>
<a name="L524"></a><tt class="py-lineno"> 524</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">line_elements</tt> </tt>
</div><a name="L525"></a><tt class="py-lineno"> 525</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer.draw_bar_graph"></a><div id="CircularDrawer.draw_bar_graph-def"><a name="L526"></a><tt class="py-lineno"> 526</tt> <a class="py-toggle" href="#" id="CircularDrawer.draw_bar_graph-toggle" onclick="return toggle('CircularDrawer.draw_bar_graph');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw_bar_graph">draw_bar_graph</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">graph</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer.draw_bar_graph-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CircularDrawer.draw_bar_graph-expanded"><a name="L527"></a><tt class="py-lineno"> 527</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return list of drawable elements for a bar graph.</tt> </tt>
<a name="L528"></a><tt class="py-lineno"> 528</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L529"></a><tt class="py-lineno"> 529</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L530"></a><tt class="py-lineno"> 530</tt>  <tt class="py-line"><tt class="py-docstring">         - graph     Graph object</tt> </tt>
<a name="L531"></a><tt class="py-lineno"> 531</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L532"></a><tt class="py-lineno"> 532</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L533"></a><tt class="py-lineno"> 533</tt>  <tt class="py-line">        <tt class="py-comment"># At each point contained in the graph data, we draw a vertical bar</tt> </tt>
<a name="L534"></a><tt class="py-lineno"> 534</tt>  <tt class="py-line">        <tt class="py-comment"># from the track center to the height of the datapoint value (positive</tt> </tt>
<a name="L535"></a><tt class="py-lineno"> 535</tt>  <tt class="py-line">        <tt class="py-comment"># values go up in one color, negative go down in the alternative</tt> </tt>
<a name="L536"></a><tt class="py-lineno"> 536</tt>  <tt class="py-line">        <tt class="py-comment"># color).</tt> </tt>
<a name="L537"></a><tt class="py-lineno"> 537</tt>  <tt class="py-line">        <tt class="py-name">bar_elements</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L538"></a><tt class="py-lineno"> 538</tt>  <tt class="py-line"> </tt>
<a name="L539"></a><tt class="py-lineno"> 539</tt>  <tt class="py-line">        <tt class="py-comment"># Set the number of pixels per unit for the data</tt> </tt>
<a name="L540"></a><tt class="py-lineno"> 540</tt>  <tt class="py-line">        <tt id="link-223" class="py-name"><a title="Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.data_quartiles" class="py-name" href="#" onclick="return doclink('link-223', 'data_quartiles', 'link-200');">data_quartiles</a></tt> <tt class="py-op">=</tt> <tt class="py-name">graph</tt><tt class="py-op">.</tt><tt id="link-224" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Graph.GraphData.quartiles" class="py-name" href="#" onclick="return doclink('link-224', 'quartiles', 'link-201');">quartiles</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L541"></a><tt class="py-lineno"> 541</tt>  <tt class="py-line">        <tt class="py-name">minval</tt><tt class="py-op">,</tt> <tt class="py-name">maxval</tt> <tt class="py-op">=</tt> <tt id="link-225" class="py-name"><a title="Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.data_quartiles" class="py-name" href="#" onclick="return doclink('link-225', 'data_quartiles', 'link-200');">data_quartiles</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt id="link-226" class="py-name"><a title="Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.data_quartiles" class="py-name" href="#" onclick="return doclink('link-226', 'data_quartiles', 'link-200');">data_quartiles</a></tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt> </tt>
<a name="L542"></a><tt class="py-lineno"> 542</tt>  <tt class="py-line">        <tt class="py-name">btm</tt><tt class="py-op">,</tt> <tt class="py-name">ctr</tt><tt class="py-op">,</tt> <tt class="py-name">top</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">track_radii</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">current_track_level</tt><tt class="py-op">]</tt> </tt>
<a name="L543"></a><tt class="py-lineno"> 543</tt>  <tt class="py-line">        <tt class="py-name">trackheight</tt> <tt class="py-op">=</tt> <tt class="py-number">0.5</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-name">top</tt> <tt class="py-op">-</tt> <tt class="py-name">btm</tt><tt class="py-op">)</tt> </tt>
<a name="L544"></a><tt class="py-lineno"> 544</tt>  <tt class="py-line">        <tt class="py-name">datarange</tt> <tt class="py-op">=</tt> <tt class="py-name">maxval</tt> <tt class="py-op">-</tt> <tt class="py-name">minval</tt> </tt>
<a name="L545"></a><tt class="py-lineno"> 545</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">datarange</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L546"></a><tt class="py-lineno"> 546</tt>  <tt class="py-line">            <tt class="py-name">datarange</tt> <tt class="py-op">=</tt> <tt class="py-name">trackheight</tt> </tt>
<a name="L547"></a><tt class="py-lineno"> 547</tt>  <tt class="py-line">        <tt id="link-227" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-227', 'data', 'link-207');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-name">graph</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-228" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-228', 'start', 'link-19');">start</a></tt><tt class="py-op">:</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-229" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-229', 'end', 'link-20');">end</a></tt><tt class="py-op">]</tt> </tt>
<a name="L548"></a><tt class="py-lineno"> 548</tt>  <tt class="py-line">        <tt class="py-comment"># midval is the value at which the x-axis is plotted, and is the</tt> </tt>
<a name="L549"></a><tt class="py-lineno"> 549</tt>  <tt class="py-line">        <tt class="py-comment"># central ring in the track</tt> </tt>
<a name="L550"></a><tt class="py-lineno"> 550</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">graph</tt><tt class="py-op">.</tt><tt class="py-name">center</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L551"></a><tt class="py-lineno"> 551</tt>  <tt class="py-line">            <tt class="py-name">midval</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">maxval</tt> <tt class="py-op">+</tt> <tt class="py-name">minval</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-number">2.</tt> </tt>
<a name="L552"></a><tt class="py-lineno"> 552</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L553"></a><tt class="py-lineno"> 553</tt>  <tt class="py-line">            <tt class="py-name">midval</tt> <tt class="py-op">=</tt> <tt class="py-name">graph</tt><tt class="py-op">.</tt><tt class="py-name">center</tt> </tt>
<a name="L554"></a><tt class="py-lineno"> 554</tt>  <tt class="py-line"> </tt>
<a name="L555"></a><tt class="py-lineno"> 555</tt>  <tt class="py-line">        <tt class="py-comment"># Convert data into 'binned' blocks, covering half the distance to the</tt> </tt>
<a name="L556"></a><tt class="py-lineno"> 556</tt>  <tt class="py-line">        <tt class="py-comment"># next data point on either side, accounting for the ends of fragments</tt> </tt>
<a name="L557"></a><tt class="py-lineno"> 557</tt>  <tt class="py-line">        <tt class="py-comment"># and tracks</tt> </tt>
<a name="L558"></a><tt class="py-lineno"> 558</tt>  <tt class="py-line">        <tt id="link-230" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-230', 'start', 'link-19');">start</a></tt><tt class="py-op">,</tt> <tt id="link-231" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-231', 'end', 'link-20');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-232" class="py-name"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer._current_track_start_end" class="py-name" href="#" onclick="return doclink('link-232', '_current_track_start_end', 'link-72');">_current_track_start_end</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L559"></a><tt class="py-lineno"> 559</tt>  <tt class="py-line">        <tt id="link-233" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-233', 'data', 'link-207');">data</a></tt> <tt class="py-op">=</tt> <tt id="link-234" class="py-name"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer.intermediate_points" class="py-name" href="#" onclick="return doclink('link-234', 'intermediate_points', 'link-7');">intermediate_points</a></tt><tt class="py-op">(</tt><tt id="link-235" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-235', 'start', 'link-19');">start</a></tt><tt class="py-op">,</tt> <tt id="link-236" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-236', 'end', 'link-20');">end</a></tt><tt class="py-op">,</tt> <tt class="py-name">graph</tt><tt class="py-op">[</tt><tt id="link-237" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-237', 'start', 'link-19');">start</a></tt><tt class="py-op">:</tt><tt id="link-238" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-238', 'end', 'link-20');">end</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L560"></a><tt class="py-lineno"> 560</tt>  <tt class="py-line"> </tt>
<a name="L561"></a><tt class="py-lineno"> 561</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-239" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-239', 'data', 'link-207');">data</a></tt><tt class="py-op">:</tt> </tt>
<a name="L562"></a><tt class="py-lineno"> 562</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L563"></a><tt class="py-lineno"> 563</tt>  <tt class="py-line"> </tt>
<a name="L564"></a><tt class="py-lineno"> 564</tt>  <tt class="py-line">        <tt class="py-comment"># Whichever is the greatest difference: max-midval or min-midval, is</tt> </tt>
<a name="L565"></a><tt class="py-lineno"> 565</tt>  <tt class="py-line">        <tt class="py-comment"># taken to specify the number of pixel units resolved along the</tt> </tt>
<a name="L566"></a><tt class="py-lineno"> 566</tt>  <tt class="py-line">        <tt class="py-comment"># y-axis</tt> </tt>
<a name="L567"></a><tt class="py-lineno"> 567</tt>  <tt class="py-line">        <tt class="py-name">resolution</tt> <tt class="py-op">=</tt> <tt id="link-240" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-240', 'max', 'link-23');">max</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">midval</tt> <tt class="py-op">-</tt> <tt class="py-name">minval</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">maxval</tt> <tt class="py-op">-</tt> <tt class="py-name">midval</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L568"></a><tt class="py-lineno"> 568</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">resolution</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L569"></a><tt class="py-lineno"> 569</tt>  <tt class="py-line">            <tt class="py-name">resolution</tt> <tt class="py-op">=</tt> <tt class="py-name">trackheight</tt> </tt>
<a name="L570"></a><tt class="py-lineno"> 570</tt>  <tt class="py-line"> </tt>
<a name="L571"></a><tt class="py-lineno"> 571</tt>  <tt class="py-line">        <tt class="py-comment"># Create elements for the bar graph based on newdata</tt> </tt>
<a name="L572"></a><tt class="py-lineno"> 572</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">pos0</tt><tt class="py-op">,</tt> <tt class="py-name">pos1</tt><tt class="py-op">,</tt> <tt class="py-name">val</tt> <tt class="py-keyword">in</tt> <tt id="link-241" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-241', 'data', 'link-207');">data</a></tt><tt class="py-op">:</tt> </tt>
<a name="L573"></a><tt class="py-lineno"> 573</tt>  <tt class="py-line">            <tt class="py-name">pos0angle</tt><tt class="py-op">,</tt> <tt class="py-name">pos0cos</tt><tt class="py-op">,</tt> <tt class="py-name">pos0sin</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-242" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.canvas_angle" class="py-name" href="#" onclick="return doclink('link-242', 'canvas_angle', 'link-82');">canvas_angle</a></tt><tt class="py-op">(</tt><tt class="py-name">pos0</tt><tt class="py-op">)</tt> </tt>
<a name="L574"></a><tt class="py-lineno"> 574</tt>  <tt class="py-line">            <tt class="py-name">pos1angle</tt><tt class="py-op">,</tt> <tt class="py-name">pos1cos</tt><tt class="py-op">,</tt> <tt class="py-name">pos1sin</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-243" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.canvas_angle" class="py-name" href="#" onclick="return doclink('link-243', 'canvas_angle', 'link-82');">canvas_angle</a></tt><tt class="py-op">(</tt><tt class="py-name">pos1</tt><tt class="py-op">)</tt> </tt>
<a name="L575"></a><tt class="py-lineno"> 575</tt>  <tt class="py-line"> </tt>
<a name="L576"></a><tt class="py-lineno"> 576</tt>  <tt class="py-line">            <tt class="py-name">barval</tt> <tt class="py-op">=</tt> <tt class="py-name">trackheight</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-name">val</tt> <tt class="py-op">-</tt> <tt class="py-name">midval</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-name">resolution</tt> </tt>
<a name="L577"></a><tt class="py-lineno"> 577</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">barval</tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L578"></a><tt class="py-lineno"> 578</tt>  <tt class="py-line">                <tt class="py-name">barcolor</tt> <tt class="py-op">=</tt> <tt class="py-name">graph</tt><tt class="py-op">.</tt><tt class="py-name">poscolor</tt> </tt>
<a name="L579"></a><tt class="py-lineno"> 579</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L580"></a><tt class="py-lineno"> 580</tt>  <tt class="py-line">                <tt class="py-name">barcolor</tt> <tt class="py-op">=</tt> <tt class="py-name">graph</tt><tt class="py-op">.</tt><tt class="py-name">negcolor</tt> </tt>
<a name="L581"></a><tt class="py-lineno"> 581</tt>  <tt class="py-line"> </tt>
<a name="L582"></a><tt class="py-lineno"> 582</tt>  <tt class="py-line">            <tt class="py-comment"># Draw bar</tt> </tt>
<a name="L583"></a><tt class="py-lineno"> 583</tt>  <tt class="py-line">            <tt class="py-name">bar_elements</tt><tt class="py-op">.</tt><tt id="link-244" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-244', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-245" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_arc()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#_draw_arc"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_arc" class="py-name" href="#" onclick="return doclink('link-245', '_draw_arc', 'link-245');">_draw_arc</a></tt><tt class="py-op">(</tt><tt class="py-name">ctr</tt><tt class="py-op">,</tt> <tt class="py-name">ctr</tt> <tt class="py-op">+</tt> <tt class="py-name">barval</tt><tt class="py-op">,</tt> <tt class="py-name">pos0angle</tt><tt class="py-op">,</tt> </tt>
<a name="L584"></a><tt class="py-lineno"> 584</tt>  <tt class="py-line">                                               <tt class="py-name">pos1angle</tt><tt class="py-op">,</tt> <tt class="py-name">barcolor</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L585"></a><tt class="py-lineno"> 585</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">bar_elements</tt> </tt>
</div><a name="L586"></a><tt class="py-lineno"> 586</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer.draw_heat_graph"></a><div id="CircularDrawer.draw_heat_graph-def"><a name="L587"></a><tt class="py-lineno"> 587</tt> <a class="py-toggle" href="#" id="CircularDrawer.draw_heat_graph-toggle" onclick="return toggle('CircularDrawer.draw_heat_graph');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw_heat_graph">draw_heat_graph</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">graph</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer.draw_heat_graph-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CircularDrawer.draw_heat_graph-expanded"><a name="L588"></a><tt class="py-lineno"> 588</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return list of drawable elements for the heat graph.</tt> </tt>
<a name="L589"></a><tt class="py-lineno"> 589</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L590"></a><tt class="py-lineno"> 590</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L591"></a><tt class="py-lineno"> 591</tt>  <tt class="py-line"><tt class="py-docstring">         - graph     Graph object</tt> </tt>
<a name="L592"></a><tt class="py-lineno"> 592</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L593"></a><tt class="py-lineno"> 593</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L594"></a><tt class="py-lineno"> 594</tt>  <tt class="py-line">        <tt class="py-comment"># At each point contained in the graph data, we draw a box that is the</tt> </tt>
<a name="L595"></a><tt class="py-lineno"> 595</tt>  <tt class="py-line">        <tt class="py-comment"># full height of the track, extending from the midpoint between the</tt> </tt>
<a name="L596"></a><tt class="py-lineno"> 596</tt>  <tt class="py-line">        <tt class="py-comment"># previous and current data points to the midpoint between the current</tt> </tt>
<a name="L597"></a><tt class="py-lineno"> 597</tt>  <tt class="py-line">        <tt class="py-comment"># and next data points</tt> </tt>
<a name="L598"></a><tt class="py-lineno"> 598</tt>  <tt class="py-line">        <tt class="py-name">heat_elements</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>  <tt class="py-comment"># holds drawable elements</tt> </tt>
<a name="L599"></a><tt class="py-lineno"> 599</tt>  <tt class="py-line"> </tt>
<a name="L600"></a><tt class="py-lineno"> 600</tt>  <tt class="py-line">        <tt class="py-comment"># Get graph data</tt> </tt>
<a name="L601"></a><tt class="py-lineno"> 601</tt>  <tt class="py-line">        <tt id="link-246" class="py-name"><a title="Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.data_quartiles" class="py-name" href="#" onclick="return doclink('link-246', 'data_quartiles', 'link-200');">data_quartiles</a></tt> <tt class="py-op">=</tt> <tt class="py-name">graph</tt><tt class="py-op">.</tt><tt id="link-247" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Graph.GraphData.quartiles" class="py-name" href="#" onclick="return doclink('link-247', 'quartiles', 'link-201');">quartiles</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L602"></a><tt class="py-lineno"> 602</tt>  <tt class="py-line">        <tt class="py-name">minval</tt><tt class="py-op">,</tt> <tt class="py-name">maxval</tt> <tt class="py-op">=</tt> <tt id="link-248" class="py-name"><a title="Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.data_quartiles" class="py-name" href="#" onclick="return doclink('link-248', 'data_quartiles', 'link-200');">data_quartiles</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt id="link-249" class="py-name"><a title="Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.data_quartiles" class="py-name" href="#" onclick="return doclink('link-249', 'data_quartiles', 'link-200');">data_quartiles</a></tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt> </tt>
<a name="L603"></a><tt class="py-lineno"> 603</tt>  <tt class="py-line">        <tt class="py-name">midval</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">maxval</tt> <tt class="py-op">+</tt> <tt class="py-name">minval</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-number">2.</tt>  <tt class="py-comment"># mid is the value at the X-axis</tt> </tt>
<a name="L604"></a><tt class="py-lineno"> 604</tt>  <tt class="py-line">        <tt class="py-name">btm</tt><tt class="py-op">,</tt> <tt class="py-name">ctr</tt><tt class="py-op">,</tt> <tt class="py-name">top</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">track_radii</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">current_track_level</tt><tt class="py-op">]</tt> </tt>
<a name="L605"></a><tt class="py-lineno"> 605</tt>  <tt class="py-line">        <tt class="py-name">trackheight</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">top</tt> <tt class="py-op">-</tt> <tt class="py-name">btm</tt><tt class="py-op">)</tt> </tt>
<a name="L606"></a><tt class="py-lineno"> 606</tt>  <tt class="py-line"> </tt>
<a name="L607"></a><tt class="py-lineno"> 607</tt>  <tt class="py-line">        <tt id="link-250" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-250', 'start', 'link-19');">start</a></tt><tt class="py-op">,</tt> <tt id="link-251" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-251', 'end', 'link-20');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-252" class="py-name"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer._current_track_start_end" class="py-name" href="#" onclick="return doclink('link-252', '_current_track_start_end', 'link-72');">_current_track_start_end</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L608"></a><tt class="py-lineno"> 608</tt>  <tt class="py-line">        <tt id="link-253" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-253', 'data', 'link-207');">data</a></tt> <tt class="py-op">=</tt> <tt id="link-254" class="py-name"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer.intermediate_points" class="py-name" href="#" onclick="return doclink('link-254', 'intermediate_points', 'link-7');">intermediate_points</a></tt><tt class="py-op">(</tt><tt id="link-255" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-255', 'start', 'link-19');">start</a></tt><tt class="py-op">,</tt> <tt id="link-256" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-256', 'end', 'link-20');">end</a></tt><tt class="py-op">,</tt> <tt class="py-name">graph</tt><tt class="py-op">[</tt><tt id="link-257" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-257', 'start', 'link-19');">start</a></tt><tt class="py-op">:</tt><tt id="link-258" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-258', 'end', 'link-20');">end</a></tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L609"></a><tt class="py-lineno"> 609</tt>  <tt class="py-line"> </tt>
<a name="L610"></a><tt class="py-lineno"> 610</tt>  <tt class="py-line">        <tt class="py-comment"># Create elements on the graph, indicating a large positive value by</tt> </tt>
<a name="L611"></a><tt class="py-lineno"> 611</tt>  <tt class="py-line">        <tt class="py-comment"># the graph's poscolor, and a large negative value by the graph's</tt> </tt>
<a name="L612"></a><tt class="py-lineno"> 612</tt>  <tt class="py-line">        <tt class="py-comment"># negcolor attributes</tt> </tt>
<a name="L613"></a><tt class="py-lineno"> 613</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">pos0</tt><tt class="py-op">,</tt> <tt class="py-name">pos1</tt><tt class="py-op">,</tt> <tt class="py-name">val</tt> <tt class="py-keyword">in</tt> <tt id="link-259" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-259', 'data', 'link-207');">data</a></tt><tt class="py-op">:</tt> </tt>
<a name="L614"></a><tt class="py-lineno"> 614</tt>  <tt class="py-line">            <tt class="py-name">pos0angle</tt><tt class="py-op">,</tt> <tt class="py-name">pos0cos</tt><tt class="py-op">,</tt> <tt class="py-name">pos0sin</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-260" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.canvas_angle" class="py-name" href="#" onclick="return doclink('link-260', 'canvas_angle', 'link-82');">canvas_angle</a></tt><tt class="py-op">(</tt><tt class="py-name">pos0</tt><tt class="py-op">)</tt> </tt>
<a name="L615"></a><tt class="py-lineno"> 615</tt>  <tt class="py-line">            <tt class="py-name">pos1angle</tt><tt class="py-op">,</tt> <tt class="py-name">pos1cos</tt><tt class="py-op">,</tt> <tt class="py-name">pos1sin</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-261" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.canvas_angle" class="py-name" href="#" onclick="return doclink('link-261', 'canvas_angle', 'link-82');">canvas_angle</a></tt><tt class="py-op">(</tt><tt class="py-name">pos1</tt><tt class="py-op">)</tt> </tt>
<a name="L616"></a><tt class="py-lineno"> 616</tt>  <tt class="py-line"> </tt>
<a name="L617"></a><tt class="py-lineno"> 617</tt>  <tt class="py-line">            <tt class="py-comment"># Calculate the heat color, based on the differential between</tt> </tt>
<a name="L618"></a><tt class="py-lineno"> 618</tt>  <tt class="py-line">            <tt class="py-comment"># the value and the median value</tt> </tt>
<a name="L619"></a><tt class="py-lineno"> 619</tt>  <tt class="py-line">            <tt class="py-name">heat</tt> <tt class="py-op">=</tt> <tt class="py-name">colors</tt><tt class="py-op">.</tt><tt class="py-name">linearlyInterpolatedColor</tt><tt class="py-op">(</tt><tt class="py-name">graph</tt><tt class="py-op">.</tt><tt class="py-name">poscolor</tt><tt class="py-op">,</tt> </tt>
<a name="L620"></a><tt class="py-lineno"> 620</tt>  <tt class="py-line">                                                    <tt class="py-name">graph</tt><tt class="py-op">.</tt><tt class="py-name">negcolor</tt><tt class="py-op">,</tt> </tt>
<a name="L621"></a><tt class="py-lineno"> 621</tt>  <tt class="py-line">                                                    <tt class="py-name">maxval</tt><tt class="py-op">,</tt> <tt class="py-name">minval</tt><tt class="py-op">,</tt> <tt class="py-name">val</tt><tt class="py-op">)</tt> </tt>
<a name="L622"></a><tt class="py-lineno"> 622</tt>  <tt class="py-line"> </tt>
<a name="L623"></a><tt class="py-lineno"> 623</tt>  <tt class="py-line">            <tt class="py-comment"># Draw heat box</tt> </tt>
<a name="L624"></a><tt class="py-lineno"> 624</tt>  <tt class="py-line">            <tt class="py-name">heat_elements</tt><tt class="py-op">.</tt><tt id="link-262" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-262', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-263" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_arc" class="py-name" href="#" onclick="return doclink('link-263', '_draw_arc', 'link-245');">_draw_arc</a></tt><tt class="py-op">(</tt><tt class="py-name">btm</tt><tt class="py-op">,</tt> <tt class="py-name">top</tt><tt class="py-op">,</tt> <tt class="py-name">pos0angle</tt><tt class="py-op">,</tt> <tt class="py-name">pos1angle</tt><tt class="py-op">,</tt> </tt>
<a name="L625"></a><tt class="py-lineno"> 625</tt>  <tt class="py-line">                                                <tt class="py-name">heat</tt><tt class="py-op">,</tt> <tt class="py-name">border</tt><tt class="py-op">=</tt><tt class="py-name">heat</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L626"></a><tt class="py-lineno"> 626</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">heat_elements</tt> </tt>
</div><a name="L627"></a><tt class="py-lineno"> 627</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer.draw_scale"></a><div id="CircularDrawer.draw_scale-def"><a name="L628"></a><tt class="py-lineno"> 628</tt> <a class="py-toggle" href="#" id="CircularDrawer.draw_scale-toggle" onclick="return toggle('CircularDrawer.draw_scale');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw_scale">draw_scale</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">track</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer.draw_scale-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CircularDrawer.draw_scale-expanded"><a name="L629"></a><tt class="py-lineno"> 629</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return list of elements in the scale and list of their labels.</tt> </tt>
<a name="L630"></a><tt class="py-lineno"> 630</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L631"></a><tt class="py-lineno"> 631</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L632"></a><tt class="py-lineno"> 632</tt>  <tt class="py-line"><tt class="py-docstring">         - track     Track object</tt> </tt>
<a name="L633"></a><tt class="py-lineno"> 633</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L634"></a><tt class="py-lineno"> 634</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L635"></a><tt class="py-lineno"> 635</tt>  <tt class="py-line">        <tt class="py-name">scale_elements</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>  <tt class="py-comment"># holds axes and ticks</tt> </tt>
<a name="L636"></a><tt class="py-lineno"> 636</tt>  <tt class="py-line">        <tt class="py-name">scale_labels</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>  <tt class="py-comment"># holds labels</tt> </tt>
<a name="L637"></a><tt class="py-lineno"> 637</tt>  <tt class="py-line"> </tt>
<a name="L638"></a><tt class="py-lineno"> 638</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">scale</tt><tt class="py-op">:</tt> </tt>
<a name="L639"></a><tt class="py-lineno"> 639</tt>  <tt class="py-line">            <tt class="py-comment"># no scale required, exit early</tt> </tt>
<a name="L640"></a><tt class="py-lineno"> 640</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L641"></a><tt class="py-lineno"> 641</tt>  <tt class="py-line"> </tt>
<a name="L642"></a><tt class="py-lineno"> 642</tt>  <tt class="py-line">        <tt class="py-comment"># Get track locations</tt> </tt>
<a name="L643"></a><tt class="py-lineno"> 643</tt>  <tt class="py-line">        <tt class="py-name">btm</tt><tt class="py-op">,</tt> <tt class="py-name">ctr</tt><tt class="py-op">,</tt> <tt class="py-name">top</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">track_radii</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">current_track_level</tt><tt class="py-op">]</tt> </tt>
<a name="L644"></a><tt class="py-lineno"> 644</tt>  <tt class="py-line">        <tt class="py-name">trackheight</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">top</tt> <tt class="py-op">-</tt> <tt class="py-name">ctr</tt><tt class="py-op">)</tt> </tt>
<a name="L645"></a><tt class="py-lineno"> 645</tt>  <tt class="py-line"> </tt>
<a name="L646"></a><tt class="py-lineno"> 646</tt>  <tt class="py-line">        <tt class="py-comment"># X-axis</tt> </tt>
<a name="L647"></a><tt class="py-lineno"> 647</tt>  <tt class="py-line">        <tt id="link-264" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-264', 'start', 'link-19');">start</a></tt><tt class="py-op">,</tt> <tt id="link-265" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-265', 'end', 'link-20');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-266" class="py-name"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer._current_track_start_end" class="py-name" href="#" onclick="return doclink('link-266', '_current_track_start_end', 'link-72');">_current_track_start_end</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L648"></a><tt class="py-lineno"> 648</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">track</tt><tt class="py-op">.</tt><tt id="link-267" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-267', 'start', 'link-19');">start</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-keyword">or</tt> <tt class="py-name">track</tt><tt class="py-op">.</tt><tt id="link-268" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-268', 'end', 'link-20');">end</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L649"></a><tt class="py-lineno"> 649</tt>  <tt class="py-line">            <tt class="py-comment"># Draw an arc, leaving out the wedge</tt> </tt>
<a name="L650"></a><tt class="py-lineno"> 650</tt>  <tt class="py-line">            <tt class="py-name">p</tt> <tt class="py-op">=</tt> <tt class="py-name">ArcPath</tt><tt class="py-op">(</tt><tt class="py-name">strokeColor</tt><tt class="py-op">=</tt><tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">scale_color</tt><tt class="py-op">,</tt> <tt class="py-name">fillColor</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
<a name="L651"></a><tt class="py-lineno"> 651</tt>  <tt class="py-line">            <tt class="py-name">startangle</tt><tt class="py-op">,</tt> <tt class="py-name">startcos</tt><tt class="py-op">,</tt> <tt class="py-name">startsin</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-269" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.canvas_angle" class="py-name" href="#" onclick="return doclink('link-269', 'canvas_angle', 'link-82');">canvas_angle</a></tt><tt class="py-op">(</tt><tt id="link-270" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-270', 'start', 'link-19');">start</a></tt><tt class="py-op">)</tt> </tt>
<a name="L652"></a><tt class="py-lineno"> 652</tt>  <tt class="py-line">            <tt class="py-name">endangle</tt><tt class="py-op">,</tt> <tt class="py-name">endcos</tt><tt class="py-op">,</tt> <tt class="py-name">endsin</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-271" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.canvas_angle" class="py-name" href="#" onclick="return doclink('link-271', 'canvas_angle', 'link-82');">canvas_angle</a></tt><tt class="py-op">(</tt><tt id="link-272" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-272', 'end', 'link-20');">end</a></tt><tt class="py-op">)</tt> </tt>
<a name="L653"></a><tt class="py-lineno"> 653</tt>  <tt class="py-line">            <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">addArc</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt><tt class="py-op">,</tt> <tt class="py-name">ctr</tt><tt class="py-op">,</tt> </tt>
<a name="L654"></a><tt class="py-lineno"> 654</tt>  <tt class="py-line">                     <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">endangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-273" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-273', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L655"></a><tt class="py-lineno"> 655</tt>  <tt class="py-line">                     <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">startangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-274" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-274', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L656"></a><tt class="py-lineno"> 656</tt>  <tt class="py-line">            <tt class="py-name">scale_elements</tt><tt class="py-op">.</tt><tt id="link-275" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-275', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">p</tt><tt class="py-op">)</tt> </tt>
<a name="L657"></a><tt class="py-lineno"> 657</tt>  <tt class="py-line">            <tt class="py-keyword">del</tt> <tt class="py-name">p</tt> </tt>
<a name="L658"></a><tt class="py-lineno"> 658</tt>  <tt class="py-line">            <tt class="py-comment"># Y-axis start marker</tt> </tt>
<a name="L659"></a><tt class="py-lineno"> 659</tt>  <tt class="py-line">            <tt class="py-name">x0</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt> <tt class="py-op">+</tt> <tt class="py-name">btm</tt> <tt class="py-op">*</tt> <tt class="py-name">startsin</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt> <tt class="py-op">+</tt> <tt class="py-name">btm</tt> <tt class="py-op">*</tt> <tt class="py-name">startcos</tt> </tt>
<a name="L660"></a><tt class="py-lineno"> 660</tt>  <tt class="py-line">            <tt class="py-name">x1</tt><tt class="py-op">,</tt> <tt class="py-name">y1</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt> <tt class="py-op">+</tt> <tt class="py-name">top</tt> <tt class="py-op">*</tt> <tt class="py-name">startsin</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt> <tt class="py-op">+</tt> <tt class="py-name">top</tt> <tt class="py-op">*</tt> <tt class="py-name">startcos</tt> </tt>
<a name="L661"></a><tt class="py-lineno"> 661</tt>  <tt class="py-line">            <tt class="py-name">scale_elements</tt><tt class="py-op">.</tt><tt id="link-276" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-276', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">Line</tt><tt class="py-op">(</tt><tt class="py-name">x0</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt><tt class="py-op">,</tt> <tt class="py-name">x1</tt><tt class="py-op">,</tt> <tt class="py-name">y1</tt><tt class="py-op">,</tt> <tt class="py-name">strokeColor</tt><tt class="py-op">=</tt><tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">scale_color</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L662"></a><tt class="py-lineno"> 662</tt>  <tt class="py-line">            <tt class="py-comment"># Y-axis end marker</tt> </tt>
<a name="L663"></a><tt class="py-lineno"> 663</tt>  <tt class="py-line">            <tt class="py-name">x0</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt> <tt class="py-op">+</tt> <tt class="py-name">btm</tt> <tt class="py-op">*</tt> <tt class="py-name">endsin</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt> <tt class="py-op">+</tt> <tt class="py-name">btm</tt> <tt class="py-op">*</tt> <tt class="py-name">endcos</tt> </tt>
<a name="L664"></a><tt class="py-lineno"> 664</tt>  <tt class="py-line">            <tt class="py-name">x1</tt><tt class="py-op">,</tt> <tt class="py-name">y1</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt> <tt class="py-op">+</tt> <tt class="py-name">top</tt> <tt class="py-op">*</tt> <tt class="py-name">endsin</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt> <tt class="py-op">+</tt> <tt class="py-name">top</tt> <tt class="py-op">*</tt> <tt class="py-name">endcos</tt> </tt>
<a name="L665"></a><tt class="py-lineno"> 665</tt>  <tt class="py-line">            <tt class="py-name">scale_elements</tt><tt class="py-op">.</tt><tt id="link-277" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-277', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">Line</tt><tt class="py-op">(</tt><tt class="py-name">x0</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt><tt class="py-op">,</tt> <tt class="py-name">x1</tt><tt class="py-op">,</tt> <tt class="py-name">y1</tt><tt class="py-op">,</tt> <tt class="py-name">strokeColor</tt><tt class="py-op">=</tt><tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">scale_color</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L666"></a><tt class="py-lineno"> 666</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">sweep</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L667"></a><tt class="py-lineno"> 667</tt>  <tt class="py-line">            <tt class="py-comment"># Draw an arc, leaving out the wedge</tt> </tt>
<a name="L668"></a><tt class="py-lineno"> 668</tt>  <tt class="py-line">            <tt class="py-name">p</tt> <tt class="py-op">=</tt> <tt class="py-name">ArcPath</tt><tt class="py-op">(</tt><tt class="py-name">strokeColor</tt><tt class="py-op">=</tt><tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">scale_color</tt><tt class="py-op">,</tt> <tt class="py-name">fillColor</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
<a name="L669"></a><tt class="py-lineno"> 669</tt>  <tt class="py-line">            <tt class="py-comment"># Note reportlab counts angles anti-clockwise from the horizontal</tt> </tt>
<a name="L670"></a><tt class="py-lineno"> 670</tt>  <tt class="py-line">            <tt class="py-comment"># (as in mathematics, e.g. complex numbers and polar coordinates)</tt> </tt>
<a name="L671"></a><tt class="py-lineno"> 671</tt>  <tt class="py-line">            <tt class="py-comment"># in degrees.</tt> </tt>
<a name="L672"></a><tt class="py-lineno"> 672</tt>  <tt class="py-line">            <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">addArc</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt><tt class="py-op">,</tt> <tt class="py-name">ctr</tt><tt class="py-op">,</tt> </tt>
<a name="L673"></a><tt class="py-lineno"> 673</tt>  <tt class="py-line">                     <tt class="py-name">startangledegrees</tt><tt class="py-op">=</tt><tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-number">360</tt> <tt class="py-op">*</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">sweep</tt><tt class="py-op">,</tt> </tt>
<a name="L674"></a><tt class="py-lineno"> 674</tt>  <tt class="py-line">                     <tt class="py-name">endangledegrees</tt><tt class="py-op">=</tt><tt class="py-number">90</tt><tt class="py-op">)</tt> </tt>
<a name="L675"></a><tt class="py-lineno"> 675</tt>  <tt class="py-line">            <tt class="py-name">scale_elements</tt><tt class="py-op">.</tt><tt id="link-278" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-278', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">p</tt><tt class="py-op">)</tt> </tt>
<a name="L676"></a><tt class="py-lineno"> 676</tt>  <tt class="py-line">            <tt class="py-keyword">del</tt> <tt class="py-name">p</tt> </tt>
<a name="L677"></a><tt class="py-lineno"> 677</tt>  <tt class="py-line">            <tt class="py-comment"># Y-axis start marker</tt> </tt>
<a name="L678"></a><tt class="py-lineno"> 678</tt>  <tt class="py-line">            <tt class="py-name">x0</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt> <tt class="py-op">+</tt> <tt class="py-name">btm</tt> </tt>
<a name="L679"></a><tt class="py-lineno"> 679</tt>  <tt class="py-line">            <tt class="py-name">x1</tt><tt class="py-op">,</tt> <tt class="py-name">y1</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt> <tt class="py-op">+</tt> <tt class="py-name">top</tt> </tt>
<a name="L680"></a><tt class="py-lineno"> 680</tt>  <tt class="py-line">            <tt class="py-name">scale_elements</tt><tt class="py-op">.</tt><tt id="link-279" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-279', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">Line</tt><tt class="py-op">(</tt><tt class="py-name">x0</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt><tt class="py-op">,</tt> <tt class="py-name">x1</tt><tt class="py-op">,</tt> <tt class="py-name">y1</tt><tt class="py-op">,</tt> <tt class="py-name">strokeColor</tt><tt class="py-op">=</tt><tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">scale_color</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L681"></a><tt class="py-lineno"> 681</tt>  <tt class="py-line">            <tt class="py-comment"># Y-axis end marker</tt> </tt>
<a name="L682"></a><tt class="py-lineno"> 682</tt>  <tt class="py-line">            <tt class="py-name">alpha</tt> <tt class="py-op">=</tt> <tt class="py-number">2</tt> <tt class="py-op">*</tt> <tt id="link-280" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-280', 'pi', 'link-14');">pi</a></tt> <tt class="py-op">*</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">sweep</tt> </tt>
<a name="L683"></a><tt class="py-lineno"> 683</tt>  <tt class="py-line">            <tt class="py-name">x0</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt> <tt class="py-op">+</tt> <tt class="py-name">btm</tt> <tt class="py-op">*</tt> <tt class="py-name">sin</tt><tt class="py-op">(</tt><tt class="py-name">alpha</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt> <tt class="py-op">+</tt> <tt class="py-name">btm</tt> <tt class="py-op">*</tt> <tt class="py-name">cos</tt><tt class="py-op">(</tt><tt class="py-name">alpha</tt><tt class="py-op">)</tt> </tt>
<a name="L684"></a><tt class="py-lineno"> 684</tt>  <tt class="py-line">            <tt class="py-name">x1</tt><tt class="py-op">,</tt> <tt class="py-name">y1</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt> <tt class="py-op">+</tt> <tt class="py-name">top</tt> <tt class="py-op">*</tt> <tt class="py-name">sin</tt><tt class="py-op">(</tt><tt class="py-name">alpha</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt> <tt class="py-op">+</tt> <tt class="py-name">top</tt> <tt class="py-op">*</tt> <tt class="py-name">cos</tt><tt class="py-op">(</tt><tt class="py-name">alpha</tt><tt class="py-op">)</tt> </tt>
<a name="L685"></a><tt class="py-lineno"> 685</tt>  <tt class="py-line">            <tt class="py-name">scale_elements</tt><tt class="py-op">.</tt><tt id="link-281" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-281', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">Line</tt><tt class="py-op">(</tt><tt class="py-name">x0</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt><tt class="py-op">,</tt> <tt class="py-name">x1</tt><tt class="py-op">,</tt> <tt class="py-name">y1</tt><tt class="py-op">,</tt> <tt class="py-name">strokeColor</tt><tt class="py-op">=</tt><tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">scale_color</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L686"></a><tt class="py-lineno"> 686</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L687"></a><tt class="py-lineno"> 687</tt>  <tt class="py-line">            <tt class="py-comment"># Draw a full circle</tt> </tt>
<a name="L688"></a><tt class="py-lineno"> 688</tt>  <tt class="py-line">            <tt class="py-name">scale_elements</tt><tt class="py-op">.</tt><tt id="link-282" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-282', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">Circle</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt><tt class="py-op">,</tt> <tt class="py-name">ctr</tt><tt class="py-op">,</tt> </tt>
<a name="L689"></a><tt class="py-lineno"> 689</tt>  <tt class="py-line">                                         <tt class="py-name">strokeColor</tt><tt class="py-op">=</tt><tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">scale_color</tt><tt class="py-op">,</tt> </tt>
<a name="L690"></a><tt class="py-lineno"> 690</tt>  <tt class="py-line">                                         <tt class="py-name">fillColor</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L691"></a><tt class="py-lineno"> 691</tt>  <tt class="py-line"> </tt>
<a name="L692"></a><tt class="py-lineno"> 692</tt>  <tt class="py-line">        <tt id="link-283" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-283', 'start', 'link-19');">start</a></tt><tt class="py-op">,</tt> <tt id="link-284" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-284', 'end', 'link-20');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-285" class="py-name"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer._current_track_start_end" class="py-name" href="#" onclick="return doclink('link-285', '_current_track_start_end', 'link-72');">_current_track_start_end</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L693"></a><tt class="py-lineno"> 693</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">scale_ticks</tt><tt class="py-op">:</tt>  <tt class="py-comment"># Ticks are required on the scale</tt> </tt>
<a name="L694"></a><tt class="py-lineno"> 694</tt>  <tt class="py-line">            <tt class="py-comment"># Draw large ticks</tt> </tt>
<a name="L695"></a><tt class="py-lineno"> 695</tt>  <tt class="py-line">            <tt class="py-comment"># I want the ticks to be consistently positioned relative to</tt> </tt>
<a name="L696"></a><tt class="py-lineno"> 696</tt>  <tt class="py-line">            <tt class="py-comment"># the start of the sequence (position 0), not relative to the</tt> </tt>
<a name="L697"></a><tt class="py-lineno"> 697</tt>  <tt class="py-line">            <tt class="py-comment"># current viewpoint (self.start and self.end)</tt> </tt>
<a name="L698"></a><tt class="py-lineno"> 698</tt>  <tt class="py-line"> </tt>
<a name="L699"></a><tt class="py-lineno"> 699</tt>  <tt class="py-line">            <tt class="py-name">ticklen</tt> <tt class="py-op">=</tt> <tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">scale_largeticks</tt> <tt class="py-op">*</tt> <tt class="py-name">trackheight</tt> </tt>
<a name="L700"></a><tt class="py-lineno"> 700</tt>  <tt class="py-line">            <tt class="py-name">tickiterval</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">scale_largetick_interval</tt><tt class="py-op">)</tt> </tt>
<a name="L701"></a><tt class="py-lineno"> 701</tt>  <tt class="py-line">            <tt class="py-comment"># Note that we could just start the list of ticks using</tt> </tt>
<a name="L702"></a><tt class="py-lineno"> 702</tt>  <tt class="py-line">            <tt class="py-comment"># range(0,self.end,tickinterval) and the filter out the</tt> </tt>
<a name="L703"></a><tt class="py-lineno"> 703</tt>  <tt class="py-line">            <tt class="py-comment"># ones before self.start - but this seems wasteful.</tt> </tt>
<a name="L704"></a><tt class="py-lineno"> 704</tt>  <tt class="py-line">            <tt class="py-comment"># Using tickiterval * (self.start/tickiterval) is a shortcut.</tt> </tt>
<a name="L705"></a><tt class="py-lineno"> 705</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">tickpos</tt> <tt class="py-keyword">in</tt> <tt id="link-286" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-286', 'range', 'link-3');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">tickiterval</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-287" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-287', 'start', 'link-19');">start</a></tt> <tt class="py-op">//</tt> <tt class="py-name">tickiterval</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L706"></a><tt class="py-lineno"> 706</tt>  <tt class="py-line">                                 <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-288" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-288', 'end', 'link-20');">end</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">tickiterval</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L707"></a><tt class="py-lineno"> 707</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">tickpos</tt> <tt class="py-op">&lt;=</tt> <tt id="link-289" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-289', 'start', 'link-19');">start</a></tt> <tt class="py-keyword">or</tt> <tt id="link-290" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-290', 'end', 'link-20');">end</a></tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">tickpos</tt><tt class="py-op">:</tt> </tt>
<a name="L708"></a><tt class="py-lineno"> 708</tt>  <tt class="py-line">                    <tt class="py-keyword">continue</tt> </tt>
<a name="L709"></a><tt class="py-lineno"> 709</tt>  <tt class="py-line">                <tt class="py-name">tick</tt><tt class="py-op">,</tt> <tt class="py-name">label</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-291" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw_tick()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw_tick,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw_tick()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#draw_tick"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw_tick
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw_tick" class="py-name" href="#" onclick="return doclink('link-291', 'draw_tick', 'link-291');">draw_tick</a></tt><tt class="py-op">(</tt><tt class="py-name">tickpos</tt><tt class="py-op">,</tt> <tt class="py-name">ctr</tt><tt class="py-op">,</tt> <tt class="py-name">ticklen</tt><tt class="py-op">,</tt> </tt>
<a name="L710"></a><tt class="py-lineno"> 710</tt>  <tt class="py-line">                                             <tt class="py-name">track</tt><tt class="py-op">,</tt> </tt>
<a name="L711"></a><tt class="py-lineno"> 711</tt>  <tt class="py-line">                                             <tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">scale_largetick_labels</tt><tt class="py-op">)</tt> </tt>
<a name="L712"></a><tt class="py-lineno"> 712</tt>  <tt class="py-line">                <tt class="py-name">scale_elements</tt><tt class="py-op">.</tt><tt id="link-292" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-292', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">tick</tt><tt class="py-op">)</tt> </tt>
<a name="L713"></a><tt class="py-lineno"> 713</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">label</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt>  <tt class="py-comment"># If there's a label, add it</tt> </tt>
<a name="L714"></a><tt class="py-lineno"> 714</tt>  <tt class="py-line">                    <tt class="py-name">scale_labels</tt><tt class="py-op">.</tt><tt id="link-293" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-293', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">label</tt><tt class="py-op">)</tt> </tt>
<a name="L715"></a><tt class="py-lineno"> 715</tt>  <tt class="py-line">            <tt class="py-comment"># Draw small ticks</tt> </tt>
<a name="L716"></a><tt class="py-lineno"> 716</tt>  <tt class="py-line">            <tt class="py-name">ticklen</tt> <tt class="py-op">=</tt> <tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">scale_smallticks</tt> <tt class="py-op">*</tt> <tt class="py-name">trackheight</tt> </tt>
<a name="L717"></a><tt class="py-lineno"> 717</tt>  <tt class="py-line">            <tt class="py-name">tickiterval</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">scale_smalltick_interval</tt><tt class="py-op">)</tt> </tt>
<a name="L718"></a><tt class="py-lineno"> 718</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">tickpos</tt> <tt class="py-keyword">in</tt> <tt id="link-294" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-294', 'range', 'link-3');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">tickiterval</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-295" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-295', 'start', 'link-19');">start</a></tt> <tt class="py-op">//</tt> <tt class="py-name">tickiterval</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L719"></a><tt class="py-lineno"> 719</tt>  <tt class="py-line">                                 <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-296" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-296', 'end', 'link-20');">end</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">tickiterval</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L720"></a><tt class="py-lineno"> 720</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">tickpos</tt> <tt class="py-op">&lt;=</tt> <tt id="link-297" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-297', 'start', 'link-19');">start</a></tt> <tt class="py-keyword">or</tt> <tt id="link-298" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-298', 'end', 'link-20');">end</a></tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">tickpos</tt><tt class="py-op">:</tt> </tt>
<a name="L721"></a><tt class="py-lineno"> 721</tt>  <tt class="py-line">                    <tt class="py-keyword">continue</tt> </tt>
<a name="L722"></a><tt class="py-lineno"> 722</tt>  <tt class="py-line">                <tt class="py-name">tick</tt><tt class="py-op">,</tt> <tt class="py-name">label</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-299" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.draw_tick
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.draw_tick" class="py-name" href="#" onclick="return doclink('link-299', 'draw_tick', 'link-291');">draw_tick</a></tt><tt class="py-op">(</tt><tt class="py-name">tickpos</tt><tt class="py-op">,</tt> <tt class="py-name">ctr</tt><tt class="py-op">,</tt> <tt class="py-name">ticklen</tt><tt class="py-op">,</tt> </tt>
<a name="L723"></a><tt class="py-lineno"> 723</tt>  <tt class="py-line">                                             <tt class="py-name">track</tt><tt class="py-op">,</tt> </tt>
<a name="L724"></a><tt class="py-lineno"> 724</tt>  <tt class="py-line">                                             <tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">scale_smalltick_labels</tt><tt class="py-op">)</tt> </tt>
<a name="L725"></a><tt class="py-lineno"> 725</tt>  <tt class="py-line">                <tt class="py-name">scale_elements</tt><tt class="py-op">.</tt><tt id="link-300" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-300', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">tick</tt><tt class="py-op">)</tt> </tt>
<a name="L726"></a><tt class="py-lineno"> 726</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">label</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt>  <tt class="py-comment"># If there's a label, add it</tt> </tt>
<a name="L727"></a><tt class="py-lineno"> 727</tt>  <tt class="py-line">                    <tt class="py-name">scale_labels</tt><tt class="py-op">.</tt><tt id="link-301" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-301', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">label</tt><tt class="py-op">)</tt> </tt>
<a name="L728"></a><tt class="py-lineno"> 728</tt>  <tt class="py-line"> </tt>
<a name="L729"></a><tt class="py-lineno"> 729</tt>  <tt class="py-line">        <tt class="py-comment"># Check to see if the track contains a graph - if it does, get the</tt> </tt>
<a name="L730"></a><tt class="py-lineno"> 730</tt>  <tt class="py-line">        <tt class="py-comment"># minimum and maximum values, and put them on the scale Y-axis</tt> </tt>
<a name="L731"></a><tt class="py-lineno"> 731</tt>  <tt class="py-line">        <tt class="py-comment"># at 60 degree intervals, ordering the labels by graph_id</tt> </tt>
<a name="L732"></a><tt class="py-lineno"> 732</tt>  <tt class="py-line">        <tt class="py-name">startangle</tt><tt class="py-op">,</tt> <tt class="py-name">startcos</tt><tt class="py-op">,</tt> <tt class="py-name">startsin</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-302" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.canvas_angle" class="py-name" href="#" onclick="return doclink('link-302', 'canvas_angle', 'link-82');">canvas_angle</a></tt><tt class="py-op">(</tt><tt id="link-303" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-303', 'start', 'link-19');">start</a></tt><tt class="py-op">)</tt> </tt>
<a name="L733"></a><tt class="py-lineno"> 733</tt>  <tt class="py-line">        <tt class="py-name">endangle</tt><tt class="py-op">,</tt> <tt class="py-name">endcos</tt><tt class="py-op">,</tt> <tt class="py-name">endsin</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-304" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.canvas_angle" class="py-name" href="#" onclick="return doclink('link-304', 'canvas_angle', 'link-82');">canvas_angle</a></tt><tt class="py-op">(</tt><tt id="link-305" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-305', 'end', 'link-20');">end</a></tt><tt class="py-op">)</tt> </tt>
<a name="L734"></a><tt class="py-lineno"> 734</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">axis_labels</tt><tt class="py-op">:</tt> </tt>
<a name="L735"></a><tt class="py-lineno"> 735</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt id="link-306" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-306', 'set', 'link-51');">set</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">track</tt><tt class="py-op">.</tt><tt id="link-307" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Track.Track.get_sets" class="py-name" href="#" onclick="return doclink('link-307', 'get_sets', 'link-52');">get_sets</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L736"></a><tt class="py-lineno"> 736</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt id="link-308" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-308', 'set', 'link-51');">set</a></tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt> <tt class="py-keyword">is</tt> <tt id="link-309" class="py-name"><a title="Bio.Graphics.GenomeDiagram._GraphSet.GraphSet" class="py-name" href="#" onclick="return doclink('link-309', 'GraphSet', 'link-13');">GraphSet</a></tt><tt class="py-op">:</tt> </tt>
<a name="L737"></a><tt class="py-lineno"> 737</tt>  <tt class="py-line">                    <tt class="py-comment"># Y-axis</tt> </tt>
<a name="L738"></a><tt class="py-lineno"> 738</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-name">n</tt> <tt class="py-keyword">in</tt> <tt id="link-310" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-310', 'range', 'link-3');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">7</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L739"></a><tt class="py-lineno"> 739</tt>  <tt class="py-line">                        <tt id="link-311" class="py-name" targets="Method Bio.PDB.vectors.Vector.angle()=Bio.PDB.vectors.Vector-class.html#angle"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-311', 'angle', 'link-311');">angle</a></tt> <tt class="py-op">=</tt> <tt class="py-name">n</tt> <tt class="py-op">*</tt> <tt class="py-number">1.0471975511965976</tt> </tt>
<a name="L740"></a><tt class="py-lineno"> 740</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt id="link-312" class="py-name"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-312', 'angle', 'link-311');">angle</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-name">startangle</tt> <tt class="py-keyword">or</tt> <tt class="py-name">endangle</tt> <tt class="py-op">&lt;</tt> <tt id="link-313" class="py-name"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-313', 'angle', 'link-311');">angle</a></tt><tt class="py-op">:</tt> </tt>
<a name="L741"></a><tt class="py-lineno"> 741</tt>  <tt class="py-line">                            <tt class="py-keyword">continue</tt> </tt>
<a name="L742"></a><tt class="py-lineno"> 742</tt>  <tt class="py-line">                        <tt class="py-name">ticksin</tt><tt class="py-op">,</tt> <tt class="py-name">tickcos</tt> <tt class="py-op">=</tt> <tt class="py-name">sin</tt><tt class="py-op">(</tt><tt id="link-314" class="py-name"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-314', 'angle', 'link-311');">angle</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">cos</tt><tt class="py-op">(</tt><tt id="link-315" class="py-name"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-315', 'angle', 'link-311');">angle</a></tt><tt class="py-op">)</tt> </tt>
<a name="L743"></a><tt class="py-lineno"> 743</tt>  <tt class="py-line">                        <tt class="py-name">x0</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt> <tt class="py-op">+</tt> <tt class="py-name">btm</tt> <tt class="py-op">*</tt> <tt class="py-name">ticksin</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt> <tt class="py-op">+</tt> <tt class="py-name">btm</tt> <tt class="py-op">*</tt> <tt class="py-name">tickcos</tt> </tt>
<a name="L744"></a><tt class="py-lineno"> 744</tt>  <tt class="py-line">                        <tt class="py-name">x1</tt><tt class="py-op">,</tt> <tt class="py-name">y1</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt> <tt class="py-op">+</tt> <tt class="py-name">top</tt> <tt class="py-op">*</tt> <tt class="py-name">ticksin</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt> <tt class="py-op">+</tt> <tt class="py-name">top</tt> <tt class="py-op">*</tt> <tt class="py-name">tickcos</tt> </tt>
<a name="L745"></a><tt class="py-lineno"> 745</tt>  <tt class="py-line">                        <tt class="py-name">scale_elements</tt><tt class="py-op">.</tt><tt id="link-316" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-316', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">Line</tt><tt class="py-op">(</tt><tt class="py-name">x0</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt><tt class="py-op">,</tt> <tt class="py-name">x1</tt><tt class="py-op">,</tt> <tt class="py-name">y1</tt><tt class="py-op">,</tt> </tt>
<a name="L746"></a><tt class="py-lineno"> 746</tt>  <tt class="py-line">                                                   <tt class="py-name">strokeColor</tt><tt class="py-op">=</tt><tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">scale_color</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L747"></a><tt class="py-lineno"> 747</tt>  <tt class="py-line"> </tt>
<a name="L748"></a><tt class="py-lineno"> 748</tt>  <tt class="py-line">                        <tt class="py-name">graph_label_min</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L749"></a><tt class="py-lineno"> 749</tt>  <tt class="py-line">                        <tt class="py-name">graph_label_max</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L750"></a><tt class="py-lineno"> 750</tt>  <tt class="py-line">                        <tt class="py-name">graph_label_mid</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L751"></a><tt class="py-lineno"> 751</tt>  <tt class="py-line">                        <tt class="py-keyword">for</tt> <tt class="py-name">graph</tt> <tt class="py-keyword">in</tt> <tt id="link-317" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-317', 'set', 'link-51');">set</a></tt><tt class="py-op">.</tt><tt id="link-318" class="py-name"><a title="Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.get_graphs" class="py-name" href="#" onclick="return doclink('link-318', 'get_graphs', 'link-197');">get_graphs</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L752"></a><tt class="py-lineno"> 752</tt>  <tt class="py-line">                            <tt id="link-319" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Graph.GraphData.quartiles" class="py-name" href="#" onclick="return doclink('link-319', 'quartiles', 'link-201');">quartiles</a></tt> <tt class="py-op">=</tt> <tt class="py-name">graph</tt><tt class="py-op">.</tt><tt id="link-320" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Graph.GraphData.quartiles" class="py-name" href="#" onclick="return doclink('link-320', 'quartiles', 'link-201');">quartiles</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L753"></a><tt class="py-lineno"> 753</tt>  <tt class="py-line">                            <tt class="py-name">minval</tt><tt class="py-op">,</tt> <tt class="py-name">maxval</tt> <tt class="py-op">=</tt> <tt id="link-321" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Graph.GraphData.quartiles" class="py-name" href="#" onclick="return doclink('link-321', 'quartiles', 'link-201');">quartiles</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt id="link-322" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Graph.GraphData.quartiles" class="py-name" href="#" onclick="return doclink('link-322', 'quartiles', 'link-201');">quartiles</a></tt><tt class="py-op">[</tt><tt class="py-number">4</tt><tt class="py-op">]</tt> </tt>
<a name="L754"></a><tt class="py-lineno"> 754</tt>  <tt class="py-line">                            <tt class="py-keyword">if</tt> <tt class="py-name">graph</tt><tt class="py-op">.</tt><tt class="py-name">center</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L755"></a><tt class="py-lineno"> 755</tt>  <tt class="py-line">                                <tt class="py-name">midval</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">maxval</tt> <tt class="py-op">+</tt> <tt class="py-name">minval</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-number">2.</tt> </tt>
<a name="L756"></a><tt class="py-lineno"> 756</tt>  <tt class="py-line">                                <tt class="py-name">graph_label_min</tt><tt class="py-op">.</tt><tt id="link-323" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-323', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"%.3f"</tt> <tt class="py-op">%</tt> <tt class="py-name">minval</tt><tt class="py-op">)</tt> </tt>
<a name="L757"></a><tt class="py-lineno"> 757</tt>  <tt class="py-line">                                <tt class="py-name">graph_label_max</tt><tt class="py-op">.</tt><tt id="link-324" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-324', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"%.3f"</tt> <tt class="py-op">%</tt> <tt class="py-name">maxval</tt><tt class="py-op">)</tt> </tt>
<a name="L758"></a><tt class="py-lineno"> 758</tt>  <tt class="py-line">                                <tt class="py-name">graph_label_mid</tt><tt class="py-op">.</tt><tt id="link-325" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-325', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"%.3f"</tt> <tt class="py-op">%</tt> <tt class="py-name">midval</tt><tt class="py-op">)</tt> </tt>
<a name="L759"></a><tt class="py-lineno"> 759</tt>  <tt class="py-line">                            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L760"></a><tt class="py-lineno"> 760</tt>  <tt class="py-line">                                <tt class="py-name">diff</tt> <tt class="py-op">=</tt> <tt id="link-326" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-326', 'max', 'link-23');">max</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">graph</tt><tt class="py-op">.</tt><tt class="py-name">center</tt> <tt class="py-op">-</tt> <tt class="py-name">minval</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L761"></a><tt class="py-lineno"> 761</tt>  <tt class="py-line">                                           <tt class="py-op">(</tt><tt class="py-name">maxval</tt> <tt class="py-op">-</tt> <tt class="py-name">graph</tt><tt class="py-op">.</tt><tt class="py-name">center</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L762"></a><tt class="py-lineno"> 762</tt>  <tt class="py-line">                                <tt class="py-name">minval</tt> <tt class="py-op">=</tt> <tt class="py-name">graph</tt><tt class="py-op">.</tt><tt class="py-name">center</tt> <tt class="py-op">-</tt> <tt class="py-name">diff</tt> </tt>
<a name="L763"></a><tt class="py-lineno"> 763</tt>  <tt class="py-line">                                <tt class="py-name">maxval</tt> <tt class="py-op">=</tt> <tt class="py-name">graph</tt><tt class="py-op">.</tt><tt class="py-name">center</tt> <tt class="py-op">+</tt> <tt class="py-name">diff</tt> </tt>
<a name="L764"></a><tt class="py-lineno"> 764</tt>  <tt class="py-line">                                <tt class="py-name">midval</tt> <tt class="py-op">=</tt> <tt class="py-name">graph</tt><tt class="py-op">.</tt><tt class="py-name">center</tt> </tt>
<a name="L765"></a><tt class="py-lineno"> 765</tt>  <tt class="py-line">                                <tt class="py-name">graph_label_mid</tt><tt class="py-op">.</tt><tt id="link-327" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-327', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"%.3f"</tt> <tt class="py-op">%</tt> <tt class="py-name">midval</tt><tt class="py-op">)</tt> </tt>
<a name="L766"></a><tt class="py-lineno"> 766</tt>  <tt class="py-line">                                <tt class="py-name">graph_label_min</tt><tt class="py-op">.</tt><tt id="link-328" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-328', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"%.3f"</tt> <tt class="py-op">%</tt> <tt class="py-name">minval</tt><tt class="py-op">)</tt> </tt>
<a name="L767"></a><tt class="py-lineno"> 767</tt>  <tt class="py-line">                                <tt class="py-name">graph_label_max</tt><tt class="py-op">.</tt><tt id="link-329" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-329', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"%.3f"</tt> <tt class="py-op">%</tt> <tt class="py-name">maxval</tt><tt class="py-op">)</tt> </tt>
<a name="L768"></a><tt class="py-lineno"> 768</tt>  <tt class="py-line">                        <tt class="py-name">xmid</tt><tt class="py-op">,</tt> <tt class="py-name">ymid</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">x1</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-number">2.</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">y1</tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-number">2.</tt> </tt>
<a name="L769"></a><tt class="py-lineno"> 769</tt>  <tt class="py-line">                        <tt class="py-keyword">for</tt> <tt class="py-name">limit</tt><tt class="py-op">,</tt> <tt id="link-330" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-330', 'x', 'link-17');">x</a></tt><tt class="py-op">,</tt> <tt id="link-331" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-331', 'y', 'link-18');">y</a></tt><tt class="py-op">,</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">graph_label_min</tt><tt class="py-op">,</tt> <tt class="py-name">x0</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L770"></a><tt class="py-lineno"> 770</tt>  <tt class="py-line">                                             <tt class="py-op">(</tt><tt class="py-name">graph_label_max</tt><tt class="py-op">,</tt> <tt class="py-name">x1</tt><tt class="py-op">,</tt> <tt class="py-name">y1</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L771"></a><tt class="py-lineno"> 771</tt>  <tt class="py-line">                                             <tt class="py-op">(</tt><tt class="py-name">graph_label_mid</tt><tt class="py-op">,</tt> <tt class="py-name">xmid</tt><tt class="py-op">,</tt> <tt class="py-name">ymid</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L772"></a><tt class="py-lineno"> 772</tt>  <tt class="py-line">                            <tt class="py-name">label</tt> <tt class="py-op">=</tt> <tt class="py-name">String</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-string">";"</tt><tt class="py-op">.</tt><tt id="link-332" class="py-name" targets="Method Bio.Seq.MutableSeq.join()=Bio.Seq.MutableSeq-class.html#join,Method Bio.Seq.Seq.join()=Bio.Seq.Seq-class.html#join,Method Bio.Seq.UnknownSeq.join()=Bio.Seq.UnknownSeq-class.html#join"><a title="Bio.Seq.MutableSeq.join
Bio.Seq.Seq.join
Bio.Seq.UnknownSeq.join" class="py-name" href="#" onclick="return doclink('link-332', 'join', 'link-332');">join</a></tt><tt class="py-op">(</tt><tt class="py-name">limit</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L773"></a><tt class="py-lineno"> 773</tt>  <tt class="py-line">                                           <tt class="py-name">fontName</tt><tt class="py-op">=</tt><tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">scale_font</tt><tt class="py-op">,</tt> </tt>
<a name="L774"></a><tt class="py-lineno"> 774</tt>  <tt class="py-line">                                           <tt class="py-name">fontSize</tt><tt class="py-op">=</tt><tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">scale_fontsize</tt><tt class="py-op">,</tt> </tt>
<a name="L775"></a><tt class="py-lineno"> 775</tt>  <tt class="py-line">                                           <tt class="py-name">fillColor</tt><tt class="py-op">=</tt><tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">scale_color</tt><tt class="py-op">)</tt> </tt>
<a name="L776"></a><tt class="py-lineno"> 776</tt>  <tt class="py-line">                            <tt class="py-name">label</tt><tt class="py-op">.</tt><tt class="py-name">textAnchor</tt> <tt class="py-op">=</tt> <tt class="py-string">'middle'</tt> </tt>
<a name="L777"></a><tt class="py-lineno"> 777</tt>  <tt class="py-line">                            <tt class="py-name">labelgroup</tt> <tt class="py-op">=</tt> <tt class="py-name">Group</tt><tt class="py-op">(</tt><tt class="py-name">label</tt><tt class="py-op">)</tt> </tt>
<a name="L778"></a><tt class="py-lineno"> 778</tt>  <tt class="py-line">                            <tt class="py-name">labelgroup</tt><tt class="py-op">.</tt><tt id="link-333" class="py-name"><a title="Bio.PDB.Atom.Atom.transform
Bio.PDB.Entity.Entity.transform" class="py-name" href="#" onclick="return doclink('link-333', 'transform', 'link-114');">transform</a></tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">tickcos</tt><tt class="py-op">,</tt> <tt class="py-op">-</tt><tt class="py-name">ticksin</tt><tt class="py-op">,</tt> </tt>
<a name="L779"></a><tt class="py-lineno"> 779</tt>  <tt class="py-line">                                                    <tt class="py-name">ticksin</tt><tt class="py-op">,</tt> <tt class="py-name">tickcos</tt><tt class="py-op">,</tt> </tt>
<a name="L780"></a><tt class="py-lineno"> 780</tt>  <tt class="py-line">                                                    <tt id="link-334" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-334', 'x', 'link-17');">x</a></tt><tt class="py-op">,</tt> <tt id="link-335" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-335', 'y', 'link-18');">y</a></tt><tt class="py-op">)</tt> </tt>
<a name="L781"></a><tt class="py-lineno"> 781</tt>  <tt class="py-line">                            <tt class="py-name">scale_labels</tt><tt class="py-op">.</tt><tt id="link-336" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-336', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">labelgroup</tt><tt class="py-op">)</tt> </tt>
<a name="L782"></a><tt class="py-lineno"> 782</tt>  <tt class="py-line"> </tt>
<a name="L783"></a><tt class="py-lineno"> 783</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">scale_elements</tt><tt class="py-op">,</tt> <tt class="py-name">scale_labels</tt> </tt>
</div><a name="L784"></a><tt class="py-lineno"> 784</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer.draw_tick"></a><div id="CircularDrawer.draw_tick-def"><a name="L785"></a><tt class="py-lineno"> 785</tt> <a class="py-toggle" href="#" id="CircularDrawer.draw_tick-toggle" onclick="return toggle('CircularDrawer.draw_tick');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw_tick">draw_tick</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">tickpos</tt><tt class="py-op">,</tt> <tt class="py-param">ctr</tt><tt class="py-op">,</tt> <tt class="py-param">ticklen</tt><tt class="py-op">,</tt> <tt class="py-param">track</tt><tt class="py-op">,</tt> <tt class="py-param">draw_label</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer.draw_tick-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CircularDrawer.draw_tick-expanded"><a name="L786"></a><tt class="py-lineno"> 786</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return drawing element for a tick on the scale.</tt> </tt>
<a name="L787"></a><tt class="py-lineno"> 787</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L788"></a><tt class="py-lineno"> 788</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L789"></a><tt class="py-lineno"> 789</tt>  <tt class="py-line"><tt class="py-docstring">         - tickpos   Int, position of the tick on the sequence</tt> </tt>
<a name="L790"></a><tt class="py-lineno"> 790</tt>  <tt class="py-line"><tt class="py-docstring">         - ctr       Float, Y co-ord of the center of the track</tt> </tt>
<a name="L791"></a><tt class="py-lineno"> 791</tt>  <tt class="py-line"><tt class="py-docstring">         - ticklen   How long to draw the tick</tt> </tt>
<a name="L792"></a><tt class="py-lineno"> 792</tt>  <tt class="py-line"><tt class="py-docstring">         - track     Track, the track the tick is drawn on</tt> </tt>
<a name="L793"></a><tt class="py-lineno"> 793</tt>  <tt class="py-line"><tt class="py-docstring">         - draw_label    Boolean, write the tick label?</tt> </tt>
<a name="L794"></a><tt class="py-lineno"> 794</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L795"></a><tt class="py-lineno"> 795</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L796"></a><tt class="py-lineno"> 796</tt>  <tt class="py-line">        <tt class="py-comment"># Calculate tick co-ordinates</tt> </tt>
<a name="L797"></a><tt class="py-lineno"> 797</tt>  <tt class="py-line">        <tt class="py-name">tickangle</tt><tt class="py-op">,</tt> <tt class="py-name">tickcos</tt><tt class="py-op">,</tt> <tt class="py-name">ticksin</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-337" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.canvas_angle" class="py-name" href="#" onclick="return doclink('link-337', 'canvas_angle', 'link-82');">canvas_angle</a></tt><tt class="py-op">(</tt><tt class="py-name">tickpos</tt><tt class="py-op">)</tt> </tt>
<a name="L798"></a><tt class="py-lineno"> 798</tt>  <tt class="py-line">        <tt class="py-name">x0</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt> <tt class="py-op">+</tt> <tt class="py-name">ctr</tt> <tt class="py-op">*</tt> <tt class="py-name">ticksin</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt> <tt class="py-op">+</tt> <tt class="py-name">ctr</tt> <tt class="py-op">*</tt> <tt class="py-name">tickcos</tt> </tt>
<a name="L799"></a><tt class="py-lineno"> 799</tt>  <tt class="py-line">        <tt class="py-name">x1</tt><tt class="py-op">,</tt> <tt class="py-name">y1</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt class="py-name">ctr</tt> <tt class="py-op">+</tt> <tt class="py-name">ticklen</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt class="py-name">ticksin</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt class="py-name">ctr</tt> <tt class="py-op">+</tt> <tt class="py-name">ticklen</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt class="py-name">tickcos</tt> </tt>
<a name="L800"></a><tt class="py-lineno"> 800</tt>  <tt class="py-line">        <tt class="py-comment"># Calculate height of text label so it can be offset on lower half</tt> </tt>
<a name="L801"></a><tt class="py-lineno"> 801</tt>  <tt class="py-line">        <tt class="py-comment"># of diagram</tt> </tt>
<a name="L802"></a><tt class="py-lineno"> 802</tt>  <tt class="py-line">        <tt class="py-comment"># LP: not used, as not all fonts have ascent_descent data in reportlab.pdfbase._fontdata</tt> </tt>
<a name="L803"></a><tt class="py-lineno"> 803</tt>  <tt class="py-line">        <tt class="py-comment"># label_offset = _fontdata.ascent_descent[track.scale_font][0]*\</tt> </tt>
<a name="L804"></a><tt class="py-lineno"> 804</tt>  <tt class="py-line">        <tt class="py-comment">#               track.scale_fontsize/1000.</tt> </tt>
<a name="L805"></a><tt class="py-lineno"> 805</tt>  <tt class="py-line">        <tt class="py-name">tick</tt> <tt class="py-op">=</tt> <tt class="py-name">Line</tt><tt class="py-op">(</tt><tt class="py-name">x0</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt><tt class="py-op">,</tt> <tt class="py-name">x1</tt><tt class="py-op">,</tt> <tt class="py-name">y1</tt><tt class="py-op">,</tt> <tt class="py-name">strokeColor</tt><tt class="py-op">=</tt><tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">scale_color</tt><tt class="py-op">)</tt> </tt>
<a name="L806"></a><tt class="py-lineno"> 806</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">draw_label</tt><tt class="py-op">:</tt> </tt>
<a name="L807"></a><tt class="py-lineno"> 807</tt>  <tt class="py-line">            <tt class="py-comment"># Put tick position on as label</tt> </tt>
<a name="L808"></a><tt class="py-lineno"> 808</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">scale_format</tt> <tt class="py-op">==</tt> <tt class="py-string">'SInt'</tt><tt class="py-op">:</tt> </tt>
<a name="L809"></a><tt class="py-lineno"> 809</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">tickpos</tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">1000000</tt><tt class="py-op">:</tt> </tt>
<a name="L810"></a><tt class="py-lineno"> 810</tt>  <tt class="py-line">                    <tt class="py-name">tickstring</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">tickpos</tt> <tt class="py-op">//</tt> <tt class="py-number">1000000</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">" Mbp"</tt> </tt>
<a name="L811"></a><tt class="py-lineno"> 811</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">tickpos</tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">1000</tt><tt class="py-op">:</tt> </tt>
<a name="L812"></a><tt class="py-lineno"> 812</tt>  <tt class="py-line">                    <tt class="py-name">tickstring</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">tickpos</tt> <tt class="py-op">//</tt> <tt class="py-number">1000</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">" Kbp"</tt> </tt>
<a name="L813"></a><tt class="py-lineno"> 813</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L814"></a><tt class="py-lineno"> 814</tt>  <tt class="py-line">                    <tt class="py-name">tickstring</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">tickpos</tt><tt class="py-op">)</tt> </tt>
<a name="L815"></a><tt class="py-lineno"> 815</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L816"></a><tt class="py-lineno"> 816</tt>  <tt class="py-line">                <tt class="py-name">tickstring</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">tickpos</tt><tt class="py-op">)</tt> </tt>
<a name="L817"></a><tt class="py-lineno"> 817</tt>  <tt class="py-line">            <tt class="py-name">label</tt> <tt class="py-op">=</tt> <tt class="py-name">String</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">tickstring</tt><tt class="py-op">,</tt>  <tt class="py-comment"># Make label string</tt> </tt>
<a name="L818"></a><tt class="py-lineno"> 818</tt>  <tt class="py-line">                           <tt class="py-name">fontName</tt><tt class="py-op">=</tt><tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">scale_font</tt><tt class="py-op">,</tt> </tt>
<a name="L819"></a><tt class="py-lineno"> 819</tt>  <tt class="py-line">                           <tt class="py-name">fontSize</tt><tt class="py-op">=</tt><tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">scale_fontsize</tt><tt class="py-op">,</tt> </tt>
<a name="L820"></a><tt class="py-lineno"> 820</tt>  <tt class="py-line">                           <tt class="py-name">fillColor</tt><tt class="py-op">=</tt><tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">scale_color</tt><tt class="py-op">)</tt> </tt>
<a name="L821"></a><tt class="py-lineno"> 821</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">tickangle</tt> <tt class="py-op">&gt;</tt> <tt id="link-338" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-338', 'pi', 'link-14');">pi</a></tt><tt class="py-op">:</tt> </tt>
<a name="L822"></a><tt class="py-lineno"> 822</tt>  <tt class="py-line">                <tt class="py-name">label</tt><tt class="py-op">.</tt><tt class="py-name">textAnchor</tt> <tt class="py-op">=</tt> <tt class="py-string">'end'</tt> </tt>
<a name="L823"></a><tt class="py-lineno"> 823</tt>  <tt class="py-line">            <tt class="py-comment"># LP: This label_offset depends on ascent_descent data, which is not available for all</tt> </tt>
<a name="L824"></a><tt class="py-lineno"> 824</tt>  <tt class="py-line">            <tt class="py-comment"># fonts, so has been deprecated.</tt> </tt>
<a name="L825"></a><tt class="py-lineno"> 825</tt>  <tt class="py-line">            <tt class="py-comment"># if 0.5*pi &lt; tickangle &lt; 1.5*pi:</tt> </tt>
<a name="L826"></a><tt class="py-lineno"> 826</tt>  <tt class="py-line">            <tt class="py-comment">#    y1 -= label_offset</tt> </tt>
<a name="L827"></a><tt class="py-lineno"> 827</tt>  <tt class="py-line">            <tt class="py-name">labelgroup</tt> <tt class="py-op">=</tt> <tt class="py-name">Group</tt><tt class="py-op">(</tt><tt class="py-name">label</tt><tt class="py-op">)</tt> </tt>
<a name="L828"></a><tt class="py-lineno"> 828</tt>  <tt class="py-line">            <tt class="py-name">labelgroup</tt><tt class="py-op">.</tt><tt id="link-339" class="py-name"><a title="Bio.PDB.Atom.Atom.transform
Bio.PDB.Entity.Entity.transform" class="py-name" href="#" onclick="return doclink('link-339', 'transform', 'link-114');">transform</a></tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">x1</tt><tt class="py-op">,</tt> <tt class="py-name">y1</tt><tt class="py-op">)</tt> </tt>
<a name="L829"></a><tt class="py-lineno"> 829</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L830"></a><tt class="py-lineno"> 830</tt>  <tt class="py-line">            <tt class="py-name">labelgroup</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L831"></a><tt class="py-lineno"> 831</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">tick</tt><tt class="py-op">,</tt> <tt class="py-name">labelgroup</tt> </tt>
</div><a name="L832"></a><tt class="py-lineno"> 832</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer.draw_test_tracks"></a><div id="CircularDrawer.draw_test_tracks-def"><a name="L833"></a><tt class="py-lineno"> 833</tt> <a class="py-toggle" href="#" id="CircularDrawer.draw_test_tracks-toggle" onclick="return toggle('CircularDrawer.draw_test_tracks');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw_test_tracks">draw_test_tracks</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer.draw_test_tracks-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CircularDrawer.draw_test_tracks-expanded"><a name="L834"></a><tt class="py-lineno"> 834</tt>  <tt class="py-line">        <tt class="py-docstring">"""Draw blue test tracks with grene line down their center."""</tt> </tt>
<a name="L835"></a><tt class="py-lineno"> 835</tt>  <tt class="py-line">        <tt class="py-comment"># Add lines only for drawn tracks</tt> </tt>
<a name="L836"></a><tt class="py-lineno"> 836</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">track</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">drawn_tracks</tt><tt class="py-op">:</tt> </tt>
<a name="L837"></a><tt class="py-lineno"> 837</tt>  <tt class="py-line">            <tt class="py-name">btm</tt><tt class="py-op">,</tt> <tt class="py-name">ctr</tt><tt class="py-op">,</tt> <tt class="py-name">top</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">track_radii</tt><tt class="py-op">[</tt><tt class="py-name">track</tt><tt class="py-op">]</tt> </tt>
<a name="L838"></a><tt class="py-lineno"> 838</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">drawing</tt><tt class="py-op">.</tt><tt id="link-340" class="py-name"><a title="Bio.Graphics.BasicChromosome._ChromosomeComponent.add
Bio.Nexus.Nexus.StepMatrix.add
Bio.Nexus.Nodes.Chain.add
Bio.PDB.Entity.Entity.add
Bio.PDB.Residue.DisorderedResidue.add
Bio.PDB.Residue.Residue.add
Bio.Restriction.Restriction.RestrictionBatch.add" class="py-name" href="#" onclick="return doclink('link-340', 'add', 'link-43');">add</a></tt><tt class="py-op">(</tt><tt class="py-name">Circle</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt><tt class="py-op">,</tt> <tt class="py-name">top</tt><tt class="py-op">,</tt> </tt>
<a name="L839"></a><tt class="py-lineno"> 839</tt>  <tt class="py-line">                                    <tt class="py-name">strokeColor</tt><tt class="py-op">=</tt><tt class="py-name">colors</tt><tt class="py-op">.</tt><tt id="link-341" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.blue()=Bio.Phylo.PhyloXMLIO.Writer-class.html#blue"><a title="Bio.Phylo.PhyloXMLIO.Writer.blue" class="py-name" href="#" onclick="return doclink('link-341', 'blue', 'link-341');">blue</a></tt><tt class="py-op">,</tt> </tt>
<a name="L840"></a><tt class="py-lineno"> 840</tt>  <tt class="py-line">                                    <tt class="py-name">fillColor</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">)</tt>  <tt class="py-comment"># top line</tt> </tt>
<a name="L841"></a><tt class="py-lineno"> 841</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">drawing</tt><tt class="py-op">.</tt><tt id="link-342" class="py-name"><a title="Bio.Graphics.BasicChromosome._ChromosomeComponent.add
Bio.Nexus.Nexus.StepMatrix.add
Bio.Nexus.Nodes.Chain.add
Bio.PDB.Entity.Entity.add
Bio.PDB.Residue.DisorderedResidue.add
Bio.PDB.Residue.Residue.add
Bio.Restriction.Restriction.RestrictionBatch.add" class="py-name" href="#" onclick="return doclink('link-342', 'add', 'link-43');">add</a></tt><tt class="py-op">(</tt><tt class="py-name">Circle</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt><tt class="py-op">,</tt> <tt class="py-name">ctr</tt><tt class="py-op">,</tt> </tt>
<a name="L842"></a><tt class="py-lineno"> 842</tt>  <tt class="py-line">                                    <tt class="py-name">strokeColor</tt><tt class="py-op">=</tt><tt class="py-name">colors</tt><tt class="py-op">.</tt><tt id="link-343" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.green()=Bio.Phylo.PhyloXMLIO.Writer-class.html#green"><a title="Bio.Phylo.PhyloXMLIO.Writer.green" class="py-name" href="#" onclick="return doclink('link-343', 'green', 'link-343');">green</a></tt><tt class="py-op">,</tt> </tt>
<a name="L843"></a><tt class="py-lineno"> 843</tt>  <tt class="py-line">                                    <tt class="py-name">fillColor</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">)</tt>  <tt class="py-comment"># middle line</tt> </tt>
<a name="L844"></a><tt class="py-lineno"> 844</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">drawing</tt><tt class="py-op">.</tt><tt id="link-344" class="py-name"><a title="Bio.Graphics.BasicChromosome._ChromosomeComponent.add
Bio.Nexus.Nexus.StepMatrix.add
Bio.Nexus.Nodes.Chain.add
Bio.PDB.Entity.Entity.add
Bio.PDB.Residue.DisorderedResidue.add
Bio.PDB.Residue.Residue.add
Bio.Restriction.Restriction.RestrictionBatch.add" class="py-name" href="#" onclick="return doclink('link-344', 'add', 'link-43');">add</a></tt><tt class="py-op">(</tt><tt class="py-name">Circle</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt><tt class="py-op">,</tt> <tt class="py-name">btm</tt><tt class="py-op">,</tt> </tt>
<a name="L845"></a><tt class="py-lineno"> 845</tt>  <tt class="py-line">                                    <tt class="py-name">strokeColor</tt><tt class="py-op">=</tt><tt class="py-name">colors</tt><tt class="py-op">.</tt><tt id="link-345" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.blue" class="py-name" href="#" onclick="return doclink('link-345', 'blue', 'link-341');">blue</a></tt><tt class="py-op">,</tt> </tt>
<a name="L846"></a><tt class="py-lineno"> 846</tt>  <tt class="py-line">                                    <tt class="py-name">fillColor</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">)</tt>  <tt class="py-comment"># bottom line</tt> </tt>
</div><a name="L847"></a><tt class="py-lineno"> 847</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer.draw_greytrack"></a><div id="CircularDrawer.draw_greytrack-def"><a name="L848"></a><tt class="py-lineno"> 848</tt> <a class="py-toggle" href="#" id="CircularDrawer.draw_greytrack-toggle" onclick="return toggle('CircularDrawer.draw_greytrack');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#draw_greytrack">draw_greytrack</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">track</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer.draw_greytrack-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CircularDrawer.draw_greytrack-expanded"><a name="L849"></a><tt class="py-lineno"> 849</tt>  <tt class="py-line">        <tt class="py-docstring">"""Drawing element for grey background to passed Track object."""</tt> </tt>
<a name="L850"></a><tt class="py-lineno"> 850</tt>  <tt class="py-line">        <tt class="py-name">greytrack_bgs</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>  <tt class="py-comment"># Holds track backgrounds</tt> </tt>
<a name="L851"></a><tt class="py-lineno"> 851</tt>  <tt class="py-line">        <tt class="py-name">greytrack_labels</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt>  <tt class="py-comment"># Holds track foreground labels</tt> </tt>
<a name="L852"></a><tt class="py-lineno"> 852</tt>  <tt class="py-line"> </tt>
<a name="L853"></a><tt class="py-lineno"> 853</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">greytrack</tt><tt class="py-op">:</tt>  <tt class="py-comment"># No greytrack required, return early</tt> </tt>
<a name="L854"></a><tt class="py-lineno"> 854</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L855"></a><tt class="py-lineno"> 855</tt>  <tt class="py-line"> </tt>
<a name="L856"></a><tt class="py-lineno"> 856</tt>  <tt class="py-line">        <tt class="py-comment"># Get track location</tt> </tt>
<a name="L857"></a><tt class="py-lineno"> 857</tt>  <tt class="py-line">        <tt class="py-name">btm</tt><tt class="py-op">,</tt> <tt class="py-name">ctr</tt><tt class="py-op">,</tt> <tt class="py-name">top</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">track_radii</tt><tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">current_track_level</tt><tt class="py-op">]</tt> </tt>
<a name="L858"></a><tt class="py-lineno"> 858</tt>  <tt class="py-line"> </tt>
<a name="L859"></a><tt class="py-lineno"> 859</tt>  <tt class="py-line">        <tt id="link-346" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-346', 'start', 'link-19');">start</a></tt><tt class="py-op">,</tt> <tt id="link-347" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-347', 'end', 'link-20');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-348" class="py-name"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer._current_track_start_end" class="py-name" href="#" onclick="return doclink('link-348', '_current_track_start_end', 'link-72');">_current_track_start_end</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L860"></a><tt class="py-lineno"> 860</tt>  <tt class="py-line">        <tt class="py-name">startangle</tt><tt class="py-op">,</tt> <tt class="py-name">startcos</tt><tt class="py-op">,</tt> <tt class="py-name">startsin</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-349" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.canvas_angle" class="py-name" href="#" onclick="return doclink('link-349', 'canvas_angle', 'link-82');">canvas_angle</a></tt><tt class="py-op">(</tt><tt id="link-350" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-350', 'start', 'link-19');">start</a></tt><tt class="py-op">)</tt> </tt>
<a name="L861"></a><tt class="py-lineno"> 861</tt>  <tt class="py-line">        <tt class="py-name">endangle</tt><tt class="py-op">,</tt> <tt class="py-name">endcos</tt><tt class="py-op">,</tt> <tt class="py-name">endsin</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-351" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.canvas_angle" class="py-name" href="#" onclick="return doclink('link-351', 'canvas_angle', 'link-82');">canvas_angle</a></tt><tt class="py-op">(</tt><tt id="link-352" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-352', 'end', 'link-20');">end</a></tt><tt class="py-op">)</tt> </tt>
<a name="L862"></a><tt class="py-lineno"> 862</tt>  <tt class="py-line"> </tt>
<a name="L863"></a><tt class="py-lineno"> 863</tt>  <tt class="py-line">        <tt class="py-comment"># Make background</tt> </tt>
<a name="L864"></a><tt class="py-lineno"> 864</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">track</tt><tt class="py-op">.</tt><tt id="link-353" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-353', 'start', 'link-19');">start</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt> <tt class="py-keyword">or</tt> <tt class="py-name">track</tt><tt class="py-op">.</tt><tt id="link-354" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-354', 'end', 'link-20');">end</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L865"></a><tt class="py-lineno"> 865</tt>  <tt class="py-line">            <tt class="py-comment"># Draw an arc, leaving out the wedge</tt> </tt>
<a name="L866"></a><tt class="py-lineno"> 866</tt>  <tt class="py-line">            <tt class="py-name">p</tt> <tt class="py-op">=</tt> <tt class="py-name">ArcPath</tt><tt class="py-op">(</tt><tt class="py-name">strokeColor</tt><tt class="py-op">=</tt><tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">scale_color</tt><tt class="py-op">,</tt> <tt class="py-name">fillColor</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
<a name="L867"></a><tt class="py-lineno"> 867</tt>  <tt class="py-line">            <tt class="py-name">greytrack_bgs</tt><tt class="py-op">.</tt><tt id="link-355" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-355', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-356" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_arc" class="py-name" href="#" onclick="return doclink('link-356', '_draw_arc', 'link-245');">_draw_arc</a></tt><tt class="py-op">(</tt><tt class="py-name">btm</tt><tt class="py-op">,</tt> <tt class="py-name">top</tt><tt class="py-op">,</tt> <tt class="py-name">startangle</tt><tt class="py-op">,</tt> <tt class="py-name">endangle</tt><tt class="py-op">,</tt> </tt>
<a name="L868"></a><tt class="py-lineno"> 868</tt>  <tt class="py-line">                                                <tt class="py-name">colors</tt><tt class="py-op">.</tt><tt class="py-name">Color</tt><tt class="py-op">(</tt><tt class="py-number">0.96</tt><tt class="py-op">,</tt> <tt class="py-number">0.96</tt><tt class="py-op">,</tt> <tt class="py-number">0.96</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L869"></a><tt class="py-lineno"> 869</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">sweep</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L870"></a><tt class="py-lineno"> 870</tt>  <tt class="py-line">            <tt class="py-comment"># Make a partial circle, a large arc box</tt> </tt>
<a name="L871"></a><tt class="py-lineno"> 871</tt>  <tt class="py-line">            <tt class="py-comment"># This method assumes the correct center for us.</tt> </tt>
<a name="L872"></a><tt class="py-lineno"> 872</tt>  <tt class="py-line">            <tt class="py-name">greytrack_bgs</tt><tt class="py-op">.</tt><tt id="link-357" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-357', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-358" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_arc" class="py-name" href="#" onclick="return doclink('link-358', '_draw_arc', 'link-245');">_draw_arc</a></tt><tt class="py-op">(</tt><tt class="py-name">btm</tt><tt class="py-op">,</tt> <tt class="py-name">top</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">2</tt> <tt class="py-op">*</tt> <tt id="link-359" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-359', 'pi', 'link-14');">pi</a></tt> <tt class="py-op">*</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">sweep</tt><tt class="py-op">,</tt> </tt>
<a name="L873"></a><tt class="py-lineno"> 873</tt>  <tt class="py-line">                                                <tt class="py-name">colors</tt><tt class="py-op">.</tt><tt class="py-name">Color</tt><tt class="py-op">(</tt><tt class="py-number">0.96</tt><tt class="py-op">,</tt> <tt class="py-number">0.96</tt><tt class="py-op">,</tt> <tt class="py-number">0.96</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L874"></a><tt class="py-lineno"> 874</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L875"></a><tt class="py-lineno"> 875</tt>  <tt class="py-line">            <tt class="py-comment"># Make a full circle (using a VERY thick linewidth)</tt> </tt>
<a name="L876"></a><tt class="py-lineno"> 876</tt>  <tt class="py-line">            <tt class="py-name">greytrack_bgs</tt><tt class="py-op">.</tt><tt id="link-360" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-360', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">Circle</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt><tt class="py-op">,</tt> <tt class="py-name">ctr</tt><tt class="py-op">,</tt> </tt>
<a name="L877"></a><tt class="py-lineno"> 877</tt>  <tt class="py-line">                                        <tt class="py-name">strokeColor</tt><tt class="py-op">=</tt><tt class="py-name">colors</tt><tt class="py-op">.</tt><tt class="py-name">Color</tt><tt class="py-op">(</tt><tt class="py-number">0.96</tt><tt class="py-op">,</tt> <tt class="py-number">0.96</tt><tt class="py-op">,</tt> <tt class="py-number">0.96</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L878"></a><tt class="py-lineno"> 878</tt>  <tt class="py-line">                                        <tt class="py-name">fillColor</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-name">strokeWidth</tt><tt class="py-op">=</tt><tt class="py-name">top</tt> <tt class="py-op">-</tt> <tt class="py-name">btm</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L879"></a><tt class="py-lineno"> 879</tt>  <tt class="py-line"> </tt>
<a name="L880"></a><tt class="py-lineno"> 880</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">greytrack_labels</tt><tt class="py-op">:</tt> </tt>
<a name="L881"></a><tt class="py-lineno"> 881</tt>  <tt class="py-line">            <tt class="py-comment"># Labels are required for this track</tt> </tt>
<a name="L882"></a><tt class="py-lineno"> 882</tt>  <tt class="py-line">            <tt class="py-name">labelstep</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-361" class="py-name" targets="Method Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.length()=Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer-class.html#length"><a title="Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-361', 'length', 'link-361');">length</a></tt> <tt class="py-op">//</tt> <tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">greytrack_labels</tt>  <tt class="py-comment"># label interval</tt> </tt>
<a name="L883"></a><tt class="py-lineno"> 883</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">pos</tt> <tt class="py-keyword">in</tt> <tt id="link-362" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-362', 'range', 'link-3');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-363" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-363', 'start', 'link-19');">start</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-364" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-364', 'end', 'link-20');">end</a></tt><tt class="py-op">,</tt> <tt class="py-name">labelstep</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L884"></a><tt class="py-lineno"> 884</tt>  <tt class="py-line">                <tt class="py-name">label</tt> <tt class="py-op">=</tt> <tt class="py-name">String</tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">track</tt><tt class="py-op">.</tt><tt id="link-365" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-365', 'name', 'link-90');">name</a></tt><tt class="py-op">,</tt>            <tt class="py-comment"># Add a new label at</tt> </tt>
<a name="L885"></a><tt class="py-lineno"> 885</tt>  <tt class="py-line">                               <tt class="py-name">fontName</tt><tt class="py-op">=</tt><tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">greytrack_font</tt><tt class="py-op">,</tt>   <tt class="py-comment"># each interval</tt> </tt>
<a name="L886"></a><tt class="py-lineno"> 886</tt>  <tt class="py-line">                               <tt class="py-name">fontSize</tt><tt class="py-op">=</tt><tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">greytrack_fontsize</tt><tt class="py-op">,</tt> </tt>
<a name="L887"></a><tt class="py-lineno"> 887</tt>  <tt class="py-line">                               <tt class="py-name">fillColor</tt><tt class="py-op">=</tt><tt class="py-name">track</tt><tt class="py-op">.</tt><tt class="py-name">greytrack_fontcolor</tt><tt class="py-op">)</tt> </tt>
<a name="L888"></a><tt class="py-lineno"> 888</tt>  <tt class="py-line">                <tt class="py-name">theta</tt><tt class="py-op">,</tt> <tt class="py-name">costheta</tt><tt class="py-op">,</tt> <tt class="py-name">sintheta</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-366" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.canvas_angle" class="py-name" href="#" onclick="return doclink('link-366', 'canvas_angle', 'link-82');">canvas_angle</a></tt><tt class="py-op">(</tt><tt class="py-name">pos</tt><tt class="py-op">)</tt> </tt>
<a name="L889"></a><tt class="py-lineno"> 889</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">theta</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">startangle</tt> <tt class="py-keyword">or</tt> <tt class="py-name">endangle</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">theta</tt><tt class="py-op">:</tt> </tt>
<a name="L890"></a><tt class="py-lineno"> 890</tt>  <tt class="py-line">                    <tt class="py-keyword">continue</tt> </tt>
<a name="L891"></a><tt class="py-lineno"> 891</tt>  <tt class="py-line">                <tt id="link-367" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-367', 'x', 'link-17');">x</a></tt><tt class="py-op">,</tt> <tt id="link-368" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-368', 'y', 'link-18');">y</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt> <tt class="py-op">+</tt> <tt class="py-name">btm</tt> <tt class="py-op">*</tt> <tt class="py-name">sintheta</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt> <tt class="py-op">+</tt> <tt class="py-name">btm</tt> <tt class="py-op">*</tt> <tt class="py-name">costheta</tt>  <tt class="py-comment"># start text halfway up marker</tt> </tt>
<a name="L892"></a><tt class="py-lineno"> 892</tt>  <tt class="py-line">                <tt class="py-name">labelgroup</tt> <tt class="py-op">=</tt> <tt class="py-name">Group</tt><tt class="py-op">(</tt><tt class="py-name">label</tt><tt class="py-op">)</tt> </tt>
<a name="L893"></a><tt class="py-lineno"> 893</tt>  <tt class="py-line">                <tt class="py-name">labelangle</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">sweep</tt> <tt class="py-op">*</tt> <tt class="py-number">2</tt> <tt class="py-op">*</tt> <tt id="link-369" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-369', 'pi', 'link-14');">pi</a></tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-name">pos</tt> <tt class="py-op">-</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-370" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-370', 'start', 'link-19');">start</a></tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-371" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-371', 'length', 'link-361');">length</a></tt> <tt class="py-op">-</tt> <tt id="link-372" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-372', 'pi', 'link-14');">pi</a></tt> <tt class="py-op">/</tt> <tt class="py-number">2</tt> </tt>
<a name="L894"></a><tt class="py-lineno"> 894</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">theta</tt> <tt class="py-op">&gt;</tt> <tt id="link-373" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-373', 'pi', 'link-14');">pi</a></tt><tt class="py-op">:</tt> </tt>
<a name="L895"></a><tt class="py-lineno"> 895</tt>  <tt class="py-line">                    <tt class="py-name">label</tt><tt class="py-op">.</tt><tt class="py-name">textAnchor</tt> <tt class="py-op">=</tt> <tt class="py-string">'end'</tt>    <tt class="py-comment"># Anchor end of text to inner radius</tt> </tt>
<a name="L896"></a><tt class="py-lineno"> 896</tt>  <tt class="py-line">                    <tt class="py-name">labelangle</tt> <tt class="py-op">+=</tt> <tt id="link-374" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-374', 'pi', 'link-14');">pi</a></tt>            <tt class="py-comment"># and reorient it</tt> </tt>
<a name="L897"></a><tt class="py-lineno"> 897</tt>  <tt class="py-line">                <tt class="py-name">cosA</tt><tt class="py-op">,</tt> <tt class="py-name">sinA</tt> <tt class="py-op">=</tt> <tt class="py-name">cos</tt><tt class="py-op">(</tt><tt class="py-name">labelangle</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">sin</tt><tt class="py-op">(</tt><tt class="py-name">labelangle</tt><tt class="py-op">)</tt> </tt>
<a name="L898"></a><tt class="py-lineno"> 898</tt>  <tt class="py-line">                <tt class="py-name">labelgroup</tt><tt class="py-op">.</tt><tt id="link-375" class="py-name"><a title="Bio.PDB.Atom.Atom.transform
Bio.PDB.Entity.Entity.transform" class="py-name" href="#" onclick="return doclink('link-375', 'transform', 'link-114');">transform</a></tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">cosA</tt><tt class="py-op">,</tt> <tt class="py-op">-</tt><tt class="py-name">sinA</tt><tt class="py-op">,</tt> <tt class="py-name">sinA</tt><tt class="py-op">,</tt> </tt>
<a name="L899"></a><tt class="py-lineno"> 899</tt>  <tt class="py-line">                                        <tt class="py-name">cosA</tt><tt class="py-op">,</tt> <tt id="link-376" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-376', 'x', 'link-17');">x</a></tt><tt class="py-op">,</tt> <tt id="link-377" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-377', 'y', 'link-18');">y</a></tt><tt class="py-op">)</tt> </tt>
<a name="L900"></a><tt class="py-lineno"> 900</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-378" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-378', 'length', 'link-361');">length</a></tt> <tt class="py-op">-</tt> <tt id="link-379" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-379', 'x', 'link-17');">x</a></tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">labelstep</tt><tt class="py-op">:</tt>  <tt class="py-comment"># Don't overrun the circle</tt> </tt>
<a name="L901"></a><tt class="py-lineno"> 901</tt>  <tt class="py-line">                    <tt class="py-name">greytrack_labels</tt><tt class="py-op">.</tt><tt id="link-380" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-380', 'append', 'link-31');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">labelgroup</tt><tt class="py-op">)</tt> </tt>
<a name="L902"></a><tt class="py-lineno"> 902</tt>  <tt class="py-line"> </tt>
<a name="L903"></a><tt class="py-lineno"> 903</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">greytrack_bgs</tt><tt class="py-op">,</tt> <tt class="py-name">greytrack_labels</tt> </tt>
</div><a name="L904"></a><tt class="py-lineno"> 904</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer.canvas_angle"></a><div id="CircularDrawer.canvas_angle-def"><a name="L905"></a><tt class="py-lineno"> 905</tt> <a class="py-toggle" href="#" id="CircularDrawer.canvas_angle-toggle" onclick="return toggle('CircularDrawer.canvas_angle');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#canvas_angle">canvas_angle</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">base</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer.canvas_angle-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CircularDrawer.canvas_angle-expanded"><a name="L906"></a><tt class="py-lineno"> 906</tt>  <tt class="py-line">        <tt class="py-docstring">"""Given base-pair position, return (angle, cosine, sin) (PRIVATE)."""</tt> </tt>
<a name="L907"></a><tt class="py-lineno"> 907</tt>  <tt class="py-line">        <tt id="link-381" class="py-name"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-381', 'angle', 'link-311');">angle</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">sweep</tt> <tt class="py-op">*</tt> <tt class="py-number">2</tt> <tt class="py-op">*</tt> <tt id="link-382" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-382', 'pi', 'link-14');">pi</a></tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-name">base</tt> <tt class="py-op">-</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-383" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-383', 'start', 'link-19');">start</a></tt><tt class="py-op">)</tt> <tt class="py-op">/</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-384" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-384', 'length', 'link-361');">length</a></tt> </tt>
<a name="L908"></a><tt class="py-lineno"> 908</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt id="link-385" class="py-name"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-385', 'angle', 'link-311');">angle</a></tt><tt class="py-op">,</tt> <tt class="py-name">cos</tt><tt class="py-op">(</tt><tt id="link-386" class="py-name"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-386', 'angle', 'link-311');">angle</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">sin</tt><tt class="py-op">(</tt><tt id="link-387" class="py-name"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-387', 'angle', 'link-311');">angle</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L909"></a><tt class="py-lineno"> 909</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer._draw_sigil_box"></a><div id="CircularDrawer._draw_sigil_box-def"><a name="L910"></a><tt class="py-lineno"> 910</tt> <a class="py-toggle" href="#" id="CircularDrawer._draw_sigil_box-toggle" onclick="return toggle('CircularDrawer._draw_sigil_box');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#_draw_sigil_box">_draw_sigil_box</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">bottom</tt><tt class="py-op">,</tt> <tt class="py-param">center</tt><tt class="py-op">,</tt> <tt class="py-param">top</tt><tt class="py-op">,</tt> </tt>
<a name="L911"></a><tt class="py-lineno"> 911</tt>  <tt class="py-line">                        <tt class="py-param">startangle</tt><tt class="py-op">,</tt> <tt class="py-param">endangle</tt><tt class="py-op">,</tt> <tt class="py-param">strand</tt><tt class="py-op">,</tt> </tt>
<a name="L912"></a><tt class="py-lineno"> 912</tt>  <tt class="py-line">                        <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer._draw_sigil_box-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CircularDrawer._draw_sigil_box-expanded"><a name="L913"></a><tt class="py-lineno"> 913</tt>  <tt class="py-line">        <tt class="py-docstring">"""Draw BOX sigil (PRIVATE)."""</tt> </tt>
<a name="L914"></a><tt class="py-lineno"> 914</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-388" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-388', 'strand', 'link-91');">strand</a></tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L915"></a><tt class="py-lineno"> 915</tt>  <tt class="py-line">            <tt class="py-name">inner_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">center</tt> </tt>
<a name="L916"></a><tt class="py-lineno"> 916</tt>  <tt class="py-line">            <tt class="py-name">outer_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">top</tt> </tt>
<a name="L917"></a><tt class="py-lineno"> 917</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-389" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-389', 'strand', 'link-91');">strand</a></tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L918"></a><tt class="py-lineno"> 918</tt>  <tt class="py-line">            <tt class="py-name">inner_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">bottom</tt> </tt>
<a name="L919"></a><tt class="py-lineno"> 919</tt>  <tt class="py-line">            <tt class="py-name">outer_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">center</tt> </tt>
<a name="L920"></a><tt class="py-lineno"> 920</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L921"></a><tt class="py-lineno"> 921</tt>  <tt class="py-line">            <tt class="py-name">inner_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">bottom</tt> </tt>
<a name="L922"></a><tt class="py-lineno"> 922</tt>  <tt class="py-line">            <tt class="py-name">outer_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">top</tt> </tt>
<a name="L923"></a><tt class="py-lineno"> 923</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-390" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_arc" class="py-name" href="#" onclick="return doclink('link-390', '_draw_arc', 'link-245');">_draw_arc</a></tt><tt class="py-op">(</tt><tt class="py-name">inner_radius</tt><tt class="py-op">,</tt> <tt class="py-name">outer_radius</tt><tt class="py-op">,</tt> <tt class="py-name">startangle</tt><tt class="py-op">,</tt> <tt class="py-name">endangle</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> </tt>
</div><a name="L924"></a><tt class="py-lineno"> 924</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer._draw_arc"></a><div id="CircularDrawer._draw_arc-def"><a name="L925"></a><tt class="py-lineno"> 925</tt> <a class="py-toggle" href="#" id="CircularDrawer._draw_arc-toggle" onclick="return toggle('CircularDrawer._draw_arc');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#_draw_arc">_draw_arc</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">inner_radius</tt><tt class="py-op">,</tt> <tt class="py-param">outer_radius</tt><tt class="py-op">,</tt> <tt class="py-param">startangle</tt><tt class="py-op">,</tt> <tt class="py-param">endangle</tt><tt class="py-op">,</tt> </tt>
<a name="L926"></a><tt class="py-lineno"> 926</tt>  <tt class="py-line">                  <tt class="py-param">color</tt><tt class="py-op">,</tt> <tt class="py-param">border</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">colour</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer._draw_arc-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CircularDrawer._draw_arc-expanded"><a name="L927"></a><tt class="py-lineno"> 927</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return closed path describing an arc box (PRIVATE).</tt> </tt>
<a name="L928"></a><tt class="py-lineno"> 928</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L929"></a><tt class="py-lineno"> 929</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L930"></a><tt class="py-lineno"> 930</tt>  <tt class="py-line"><tt class="py-docstring">         - inner_radius  Float distance of inside of arc from drawing center</tt> </tt>
<a name="L931"></a><tt class="py-lineno"> 931</tt>  <tt class="py-line"><tt class="py-docstring">         - outer_radius  Float distance of outside of arc from drawing center</tt> </tt>
<a name="L932"></a><tt class="py-lineno"> 932</tt>  <tt class="py-line"><tt class="py-docstring">         - startangle    Float angle subtended by start of arc at drawing center</tt> </tt>
<a name="L933"></a><tt class="py-lineno"> 933</tt>  <tt class="py-line"><tt class="py-docstring">           (in radians)</tt> </tt>
<a name="L934"></a><tt class="py-lineno"> 934</tt>  <tt class="py-line"><tt class="py-docstring">         - endangle      Float angle subtended by end of arc at drawing center</tt> </tt>
<a name="L935"></a><tt class="py-lineno"> 935</tt>  <tt class="py-line"><tt class="py-docstring">           (in radians)</tt> </tt>
<a name="L936"></a><tt class="py-lineno"> 936</tt>  <tt class="py-line"><tt class="py-docstring">         - color        colors.Color object for arc (overridden by backwards</tt> </tt>
<a name="L937"></a><tt class="py-lineno"> 937</tt>  <tt class="py-line"><tt class="py-docstring">           compatible argument with UK spelling, colour).</tt> </tt>
<a name="L938"></a><tt class="py-lineno"> 938</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L939"></a><tt class="py-lineno"> 939</tt>  <tt class="py-line"><tt class="py-docstring">        Returns a closed path object describing an arced box corresponding to</tt> </tt>
<a name="L940"></a><tt class="py-lineno"> 940</tt>  <tt class="py-line"><tt class="py-docstring">        the passed values.  For very small angles, a simple four sided</tt> </tt>
<a name="L941"></a><tt class="py-lineno"> 941</tt>  <tt class="py-line"><tt class="py-docstring">        polygon is used.</tt> </tt>
<a name="L942"></a><tt class="py-lineno"> 942</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L943"></a><tt class="py-lineno"> 943</tt>  <tt class="py-line">        <tt class="py-comment"># Let the UK spelling (colour) override the USA spelling (color)</tt> </tt>
<a name="L944"></a><tt class="py-lineno"> 944</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">colour</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L945"></a><tt class="py-lineno"> 945</tt>  <tt class="py-line">            <tt id="link-391" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-391', 'color', 'link-92');">color</a></tt> <tt class="py-op">=</tt> <tt class="py-name">colour</tt> </tt>
<a name="L946"></a><tt class="py-lineno"> 946</tt>  <tt class="py-line"> </tt>
<a name="L947"></a><tt class="py-lineno"> 947</tt>  <tt class="py-line">        <tt class="py-name">strokecolor</tt><tt class="py-op">,</tt> <tt id="link-392" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-392', 'color', 'link-92');">color</a></tt> <tt class="py-op">=</tt> <tt id="link-393" class="py-name"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer._stroke_and_fill_colors" class="py-name" href="#" onclick="return doclink('link-393', '_stroke_and_fill_colors', 'link-9');">_stroke_and_fill_colors</a></tt><tt class="py-op">(</tt><tt id="link-394" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-394', 'color', 'link-92');">color</a></tt><tt class="py-op">,</tt> <tt class="py-name">border</tt><tt class="py-op">)</tt> </tt>
<a name="L948"></a><tt class="py-lineno"> 948</tt>  <tt class="py-line"> </tt>
<a name="L949"></a><tt class="py-lineno"> 949</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-395" class="py-name" targets="Method Bio.FSSP.FSSPAlignDict.abs()=Bio.FSSP.FSSPAlignDict-class.html#abs"><a title="Bio.FSSP.FSSPAlignDict.abs" class="py-name" href="#" onclick="return doclink('link-395', 'abs', 'link-395');">abs</a></tt><tt class="py-op">(</tt><tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">endangle</tt> <tt class="py-op">-</tt> <tt class="py-name">startangle</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">.01</tt><tt class="py-op">:</tt> </tt>
<a name="L950"></a><tt class="py-lineno"> 950</tt>  <tt class="py-line">            <tt class="py-comment"># Wide arc, must use full curves</tt> </tt>
<a name="L951"></a><tt class="py-lineno"> 951</tt>  <tt class="py-line">            <tt class="py-name">p</tt> <tt class="py-op">=</tt> <tt class="py-name">ArcPath</tt><tt class="py-op">(</tt><tt class="py-name">strokeColor</tt><tt class="py-op">=</tt><tt class="py-name">strokecolor</tt><tt class="py-op">,</tt> </tt>
<a name="L952"></a><tt class="py-lineno"> 952</tt>  <tt class="py-line">                        <tt class="py-name">fillColor</tt><tt class="py-op">=</tt><tt id="link-396" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-396', 'color', 'link-92');">color</a></tt><tt class="py-op">,</tt> </tt>
<a name="L953"></a><tt class="py-lineno"> 953</tt>  <tt class="py-line">                        <tt class="py-name">strokewidth</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L954"></a><tt class="py-lineno"> 954</tt>  <tt class="py-line">            <tt class="py-comment"># Note reportlab counts angles anti-clockwise from the horizontal</tt> </tt>
<a name="L955"></a><tt class="py-lineno"> 955</tt>  <tt class="py-line">            <tt class="py-comment"># (as in mathematics, e.g. complex numbers and polar coordinates)</tt> </tt>
<a name="L956"></a><tt class="py-lineno"> 956</tt>  <tt class="py-line">            <tt class="py-comment"># but we use clockwise from the vertical.  Also reportlab uses</tt> </tt>
<a name="L957"></a><tt class="py-lineno"> 957</tt>  <tt class="py-line">            <tt class="py-comment"># degrees, but we use radians.</tt> </tt>
<a name="L958"></a><tt class="py-lineno"> 958</tt>  <tt class="py-line">            <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">addArc</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt><tt class="py-op">,</tt> <tt class="py-name">inner_radius</tt><tt class="py-op">,</tt> </tt>
<a name="L959"></a><tt class="py-lineno"> 959</tt>  <tt class="py-line">                     <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">endangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-397" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-397', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">startangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-398" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-398', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L960"></a><tt class="py-lineno"> 960</tt>  <tt class="py-line">                     <tt class="py-name">moveTo</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L961"></a><tt class="py-lineno"> 961</tt>  <tt class="py-line">            <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">addArc</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt><tt class="py-op">,</tt> <tt class="py-name">outer_radius</tt><tt class="py-op">,</tt> </tt>
<a name="L962"></a><tt class="py-lineno"> 962</tt>  <tt class="py-line">                     <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">endangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-399" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-399', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">startangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-400" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-400', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L963"></a><tt class="py-lineno"> 963</tt>  <tt class="py-line">                     <tt id="link-401" class="py-name" targets="Method Bio.Pathway.Reaction.reverse()=Bio.Pathway.Reaction-class.html#reverse,Method Bio.Seq.MutableSeq.reverse()=Bio.Seq.MutableSeq-class.html#reverse"><a title="Bio.Pathway.Reaction.reverse
Bio.Seq.MutableSeq.reverse" class="py-name" href="#" onclick="return doclink('link-401', 'reverse', 'link-401');">reverse</a></tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L964"></a><tt class="py-lineno"> 964</tt>  <tt class="py-line">            <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">closePath</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L965"></a><tt class="py-lineno"> 965</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">p</tt> </tt>
<a name="L966"></a><tt class="py-lineno"> 966</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L967"></a><tt class="py-lineno"> 967</tt>  <tt class="py-line">            <tt class="py-comment"># Cheat and just use a four sided polygon.</tt> </tt>
<a name="L968"></a><tt class="py-lineno"> 968</tt>  <tt class="py-line">            <tt class="py-comment"># Calculate trig values for angle and coordinates</tt> </tt>
<a name="L969"></a><tt class="py-lineno"> 969</tt>  <tt class="py-line">            <tt class="py-name">startcos</tt><tt class="py-op">,</tt> <tt class="py-name">startsin</tt> <tt class="py-op">=</tt> <tt class="py-name">cos</tt><tt class="py-op">(</tt><tt class="py-name">startangle</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">sin</tt><tt class="py-op">(</tt><tt class="py-name">startangle</tt><tt class="py-op">)</tt> </tt>
<a name="L970"></a><tt class="py-lineno"> 970</tt>  <tt class="py-line">            <tt class="py-name">endcos</tt><tt class="py-op">,</tt> <tt class="py-name">endsin</tt> <tt class="py-op">=</tt> <tt class="py-name">cos</tt><tt class="py-op">(</tt><tt class="py-name">endangle</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">sin</tt><tt class="py-op">(</tt><tt class="py-name">endangle</tt><tt class="py-op">)</tt> </tt>
<a name="L971"></a><tt class="py-lineno"> 971</tt>  <tt class="py-line">            <tt class="py-name">x0</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt>  <tt class="py-comment"># origin of the circle</tt> </tt>
<a name="L972"></a><tt class="py-lineno"> 972</tt>  <tt class="py-line">            <tt class="py-name">x1</tt><tt class="py-op">,</tt> <tt class="py-name">y1</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">inner_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">startsin</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">inner_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">startcos</tt><tt class="py-op">)</tt> </tt>
<a name="L973"></a><tt class="py-lineno"> 973</tt>  <tt class="py-line">            <tt class="py-name">x2</tt><tt class="py-op">,</tt> <tt class="py-name">y2</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">inner_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">endsin</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">inner_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">endcos</tt><tt class="py-op">)</tt> </tt>
<a name="L974"></a><tt class="py-lineno"> 974</tt>  <tt class="py-line">            <tt class="py-name">x3</tt><tt class="py-op">,</tt> <tt class="py-name">y3</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">outer_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">endsin</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">outer_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">endcos</tt><tt class="py-op">)</tt> </tt>
<a name="L975"></a><tt class="py-lineno"> 975</tt>  <tt class="py-line">            <tt class="py-name">x4</tt><tt class="py-op">,</tt> <tt class="py-name">y4</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">outer_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">startsin</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">outer_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">startcos</tt><tt class="py-op">)</tt> </tt>
<a name="L976"></a><tt class="py-lineno"> 976</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-402" class="py-name"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer.draw_polygon" class="py-name" href="#" onclick="return doclink('link-402', 'draw_polygon', 'link-6');">draw_polygon</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">x1</tt><tt class="py-op">,</tt> <tt class="py-name">y1</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">x2</tt><tt class="py-op">,</tt> <tt class="py-name">y2</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">x3</tt><tt class="py-op">,</tt> <tt class="py-name">y3</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">x4</tt><tt class="py-op">,</tt> <tt class="py-name">y4</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt id="link-403" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-403', 'color', 'link-92');">color</a></tt><tt class="py-op">,</tt> <tt class="py-name">border</tt><tt class="py-op">)</tt> </tt>
</div><a name="L977"></a><tt class="py-lineno"> 977</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer._draw_arc_line"></a><div id="CircularDrawer._draw_arc_line-def"><a name="L978"></a><tt class="py-lineno"> 978</tt> <a class="py-toggle" href="#" id="CircularDrawer._draw_arc_line-toggle" onclick="return toggle('CircularDrawer._draw_arc_line');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#_draw_arc_line">_draw_arc_line</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">path</tt><tt class="py-op">,</tt> <tt class="py-param">start_radius</tt><tt class="py-op">,</tt> <tt class="py-param">end_radius</tt><tt class="py-op">,</tt> <tt class="py-param">start_angle</tt><tt class="py-op">,</tt> <tt class="py-param">end_angle</tt><tt class="py-op">,</tt> </tt>
<a name="L979"></a><tt class="py-lineno"> 979</tt>  <tt class="py-line">                       <tt class="py-param">move</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer._draw_arc_line-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CircularDrawer._draw_arc_line-expanded"><a name="L980"></a><tt class="py-lineno"> 980</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add a list of points to a path object (PRIVATE).</tt> </tt>
<a name="L981"></a><tt class="py-lineno"> 981</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L982"></a><tt class="py-lineno"> 982</tt>  <tt class="py-line"><tt class="py-docstring">        Assumes angles given are in degrees!</tt> </tt>
<a name="L983"></a><tt class="py-lineno"> 983</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L984"></a><tt class="py-lineno"> 984</tt>  <tt class="py-line"><tt class="py-docstring">        Represents what would be a straight line on a linear diagram.</tt> </tt>
<a name="L985"></a><tt class="py-lineno"> 985</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L986"></a><tt class="py-lineno"> 986</tt>  <tt class="py-line">        <tt class="py-name">x0</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt>  <tt class="py-comment"># origin of the circle</tt> </tt>
<a name="L987"></a><tt class="py-lineno"> 987</tt>  <tt class="py-line">        <tt class="py-name">radius_diff</tt> <tt class="py-op">=</tt> <tt class="py-name">end_radius</tt> <tt class="py-op">-</tt> <tt class="py-name">start_radius</tt> </tt>
<a name="L988"></a><tt class="py-lineno"> 988</tt>  <tt class="py-line">        <tt class="py-name">angle_diff</tt> <tt class="py-op">=</tt> <tt class="py-name">end_angle</tt> <tt class="py-op">-</tt> <tt class="py-name">start_angle</tt> </tt>
<a name="L989"></a><tt class="py-lineno"> 989</tt>  <tt class="py-line">        <tt class="py-name">dx</tt> <tt class="py-op">=</tt> <tt class="py-number">0.01</tt>  <tt class="py-comment"># heuristic</tt> </tt>
<a name="L990"></a><tt class="py-lineno"> 990</tt>  <tt class="py-line">        <tt id="link-404" class="py-name" targets="Variable Bio.Graphics.ColorSpiral.ColorSpiral.a=Bio.Graphics.ColorSpiral.ColorSpiral-class.html#a"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-404', 'a', 'link-404');">a</a></tt> <tt class="py-op">=</tt> <tt class="py-name">start_angle</tt> <tt class="py-op">*</tt> <tt id="link-405" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-405', 'pi', 'link-14');">pi</a></tt> <tt class="py-op">/</tt> <tt class="py-number">180</tt> </tt>
<a name="L991"></a><tt class="py-lineno"> 991</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">move</tt><tt class="py-op">:</tt> </tt>
<a name="L992"></a><tt class="py-lineno"> 992</tt>  <tt class="py-line">            <tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">moveTo</tt><tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">start_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">cos</tt><tt class="py-op">(</tt><tt id="link-406" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-406', 'a', 'link-404');">a</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">start_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">sin</tt><tt class="py-op">(</tt><tt id="link-407" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-407', 'a', 'link-404');">a</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L993"></a><tt class="py-lineno"> 993</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L994"></a><tt class="py-lineno"> 994</tt>  <tt class="py-line">            <tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">lineTo</tt><tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">start_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">cos</tt><tt class="py-op">(</tt><tt id="link-408" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-408', 'a', 'link-404');">a</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">start_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">sin</tt><tt class="py-op">(</tt><tt id="link-409" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-409', 'a', 'link-404');">a</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L995"></a><tt class="py-lineno"> 995</tt>  <tt class="py-line">        <tt id="link-410" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-410', 'x', 'link-17');">x</a></tt> <tt class="py-op">=</tt> <tt class="py-name">dx</tt> </tt>
<a name="L996"></a><tt class="py-lineno"> 996</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-number">0.01</tt> <tt class="py-op">&lt;=</tt> <tt id="link-411" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.abs" class="py-name" href="#" onclick="return doclink('link-411', 'abs', 'link-395');">abs</a></tt><tt class="py-op">(</tt><tt class="py-name">dx</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L997"></a><tt class="py-lineno"> 997</tt>  <tt class="py-line">            <tt class="py-keyword">while</tt> <tt id="link-412" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-412', 'x', 'link-17');">x</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L998"></a><tt class="py-lineno"> 998</tt>  <tt class="py-line">                <tt class="py-name">r</tt> <tt class="py-op">=</tt> <tt class="py-name">start_radius</tt> <tt class="py-op">+</tt> <tt id="link-413" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-413', 'x', 'link-17');">x</a></tt> <tt class="py-op">*</tt> <tt class="py-name">radius_diff</tt> </tt>
<a name="L999"></a><tt class="py-lineno"> 999</tt>  <tt class="py-line">                <tt id="link-414" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-414', 'a', 'link-404');">a</a></tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">start_angle</tt> <tt class="py-op">+</tt> <tt id="link-415" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-415', 'x', 'link-17');">x</a></tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-name">angle_diff</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt id="link-416" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-416', 'pi', 'link-14');">pi</a></tt> <tt class="py-op">/</tt> <tt class="py-number">180</tt>  <tt class="py-comment"># to radians for sin/cos</tt> </tt>
<a name="L1000"></a><tt class="py-lineno">1000</tt>  <tt class="py-line">                <tt class="py-comment"># print x0+r*cos(a), y0+r*sin(a)</tt> </tt>
<a name="L1001"></a><tt class="py-lineno">1001</tt>  <tt class="py-line">                <tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">lineTo</tt><tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">r</tt> <tt class="py-op">*</tt> <tt class="py-name">cos</tt><tt class="py-op">(</tt><tt id="link-417" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-417', 'a', 'link-404');">a</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">r</tt> <tt class="py-op">*</tt> <tt class="py-name">sin</tt><tt class="py-op">(</tt><tt id="link-418" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-418', 'a', 'link-404');">a</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1002"></a><tt class="py-lineno">1002</tt>  <tt class="py-line">                <tt id="link-419" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x
Bio.Statistics.lowess.x" class="py-name" href="#" onclick="return doclink('link-419', 'x', 'link-17');">x</a></tt> <tt class="py-op">+=</tt> <tt class="py-name">dx</tt> </tt>
<a name="L1003"></a><tt class="py-lineno">1003</tt>  <tt class="py-line">        <tt id="link-420" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-420', 'a', 'link-404');">a</a></tt> <tt class="py-op">=</tt> <tt class="py-name">end_angle</tt> <tt class="py-op">*</tt> <tt id="link-421" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-421', 'pi', 'link-14');">pi</a></tt> <tt class="py-op">/</tt> <tt class="py-number">180</tt> </tt>
<a name="L1004"></a><tt class="py-lineno">1004</tt>  <tt class="py-line">        <tt class="py-name">path</tt><tt class="py-op">.</tt><tt class="py-name">lineTo</tt><tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">end_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">cos</tt><tt class="py-op">(</tt><tt id="link-422" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-422', 'a', 'link-404');">a</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">end_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">sin</tt><tt class="py-op">(</tt><tt id="link-423" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-423', 'a', 'link-404');">a</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1005"></a><tt class="py-lineno">1005</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer._draw_arc_poly"></a><div id="CircularDrawer._draw_arc_poly-def"><a name="L1006"></a><tt class="py-lineno">1006</tt> <a class="py-toggle" href="#" id="CircularDrawer._draw_arc_poly-toggle" onclick="return toggle('CircularDrawer._draw_arc_poly');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#_draw_arc_poly">_draw_arc_poly</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">inner_radius</tt><tt class="py-op">,</tt> <tt class="py-param">outer_radius</tt><tt class="py-op">,</tt> </tt>
<a name="L1007"></a><tt class="py-lineno">1007</tt>  <tt class="py-line">                       <tt class="py-param">inner_startangle</tt><tt class="py-op">,</tt> <tt class="py-param">inner_endangle</tt><tt class="py-op">,</tt> </tt>
<a name="L1008"></a><tt class="py-lineno">1008</tt>  <tt class="py-line">                       <tt class="py-param">outer_startangle</tt><tt class="py-op">,</tt> <tt class="py-param">outer_endangle</tt><tt class="py-op">,</tt> </tt>
<a name="L1009"></a><tt class="py-lineno">1009</tt>  <tt class="py-line">                       <tt class="py-param">color</tt><tt class="py-op">,</tt> <tt class="py-param">border</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">flip</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">,</tt> </tt>
<a name="L1010"></a><tt class="py-lineno">1010</tt>  <tt class="py-line">                       <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer._draw_arc_poly-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CircularDrawer._draw_arc_poly-expanded"><a name="L1011"></a><tt class="py-lineno">1011</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return polygon path describing an arc."""</tt> </tt>
<a name="L1012"></a><tt class="py-lineno">1012</tt>  <tt class="py-line">        <tt class="py-name">strokecolor</tt><tt class="py-op">,</tt> <tt id="link-424" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-424', 'color', 'link-92');">color</a></tt> <tt class="py-op">=</tt> <tt id="link-425" class="py-name"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer._stroke_and_fill_colors" class="py-name" href="#" onclick="return doclink('link-425', '_stroke_and_fill_colors', 'link-9');">_stroke_and_fill_colors</a></tt><tt class="py-op">(</tt><tt id="link-426" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-426', 'color', 'link-92');">color</a></tt><tt class="py-op">,</tt> <tt class="py-name">border</tt><tt class="py-op">)</tt> </tt>
<a name="L1013"></a><tt class="py-lineno">1013</tt>  <tt class="py-line"> </tt>
<a name="L1014"></a><tt class="py-lineno">1014</tt>  <tt class="py-line">        <tt class="py-name">x0</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt>      <tt class="py-comment"># origin of the circle</tt> </tt>
<a name="L1015"></a><tt class="py-lineno">1015</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-427" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.abs" class="py-name" href="#" onclick="return doclink('link-427', 'abs', 'link-395');">abs</a></tt><tt class="py-op">(</tt><tt class="py-name">inner_endangle</tt> <tt class="py-op">-</tt> <tt class="py-name">outer_startangle</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0.01</tt> \ </tt>
<a name="L1016"></a><tt class="py-lineno">1016</tt>  <tt class="py-line">        <tt class="py-keyword">or</tt> <tt id="link-428" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.abs" class="py-name" href="#" onclick="return doclink('link-428', 'abs', 'link-395');">abs</a></tt><tt class="py-op">(</tt><tt class="py-name">outer_endangle</tt> <tt class="py-op">-</tt> <tt class="py-name">inner_startangle</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0.01</tt> \ </tt>
<a name="L1017"></a><tt class="py-lineno">1017</tt>  <tt class="py-line">        <tt class="py-keyword">or</tt> <tt id="link-429" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.abs" class="py-name" href="#" onclick="return doclink('link-429', 'abs', 'link-395');">abs</a></tt><tt class="py-op">(</tt><tt class="py-name">inner_startangle</tt> <tt class="py-op">-</tt> <tt class="py-name">outer_startangle</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0.01</tt> \ </tt>
<a name="L1018"></a><tt class="py-lineno">1018</tt>  <tt class="py-line">        <tt class="py-keyword">or</tt> <tt id="link-430" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.abs" class="py-name" href="#" onclick="return doclink('link-430', 'abs', 'link-395');">abs</a></tt><tt class="py-op">(</tt><tt class="py-name">outer_startangle</tt> <tt class="py-op">-</tt> <tt class="py-name">outer_startangle</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0.01</tt><tt class="py-op">:</tt> </tt>
<a name="L1019"></a><tt class="py-lineno">1019</tt>  <tt class="py-line">            <tt class="py-comment"># Wide arc, must use full curves</tt> </tt>
<a name="L1020"></a><tt class="py-lineno">1020</tt>  <tt class="py-line">            <tt class="py-name">p</tt> <tt class="py-op">=</tt> <tt class="py-name">ArcPath</tt><tt class="py-op">(</tt><tt class="py-name">strokeColor</tt><tt class="py-op">=</tt><tt class="py-name">strokecolor</tt><tt class="py-op">,</tt> </tt>
<a name="L1021"></a><tt class="py-lineno">1021</tt>  <tt class="py-line">                        <tt class="py-name">fillColor</tt><tt class="py-op">=</tt><tt id="link-431" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-431', 'color', 'link-92');">color</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1022"></a><tt class="py-lineno">1022</tt>  <tt class="py-line">                        <tt class="py-comment"># default is mitre/miter which can stick out too much:</tt> </tt>
<a name="L1023"></a><tt class="py-lineno">1023</tt>  <tt class="py-line">                        <tt class="py-name">strokeLineJoin</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">,</tt>  <tt class="py-comment"># 1=round</tt> </tt>
<a name="L1024"></a><tt class="py-lineno">1024</tt>  <tt class="py-line">                        <tt class="py-name">strokewidth</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L1025"></a><tt class="py-lineno">1025</tt>  <tt class="py-line">            <tt class="py-comment"># Note reportlab counts angles anti-clockwise from the horizontal</tt> </tt>
<a name="L1026"></a><tt class="py-lineno">1026</tt>  <tt class="py-line">            <tt class="py-comment"># (as in mathematics, e.g. complex numbers and polar coordinates)</tt> </tt>
<a name="L1027"></a><tt class="py-lineno">1027</tt>  <tt class="py-line">            <tt class="py-comment"># but we use clockwise from the vertical.  Also reportlab uses</tt> </tt>
<a name="L1028"></a><tt class="py-lineno">1028</tt>  <tt class="py-line">            <tt class="py-comment"># degrees, but we use radians.</tt> </tt>
<a name="L1029"></a><tt class="py-lineno">1029</tt>  <tt class="py-line">            <tt class="py-name">i_start</tt> <tt class="py-op">=</tt> <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">inner_startangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-432" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-432', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1030"></a><tt class="py-lineno">1030</tt>  <tt class="py-line">            <tt class="py-name">i_end</tt> <tt class="py-op">=</tt> <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">inner_endangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-433" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-433', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1031"></a><tt class="py-lineno">1031</tt>  <tt class="py-line">            <tt class="py-name">o_start</tt> <tt class="py-op">=</tt> <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">outer_startangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-434" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-434', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1032"></a><tt class="py-lineno">1032</tt>  <tt class="py-line">            <tt class="py-name">o_end</tt> <tt class="py-op">=</tt> <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">outer_endangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-435" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-435', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1033"></a><tt class="py-lineno">1033</tt>  <tt class="py-line">            <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">addArc</tt><tt class="py-op">(</tt><tt class="py-name">x0</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt><tt class="py-op">,</tt> <tt class="py-name">inner_radius</tt><tt class="py-op">,</tt> <tt class="py-name">i_end</tt><tt class="py-op">,</tt> <tt class="py-name">i_start</tt><tt class="py-op">,</tt> </tt>
<a name="L1034"></a><tt class="py-lineno">1034</tt>  <tt class="py-line">                     <tt class="py-name">moveTo</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">,</tt> <tt id="link-436" class="py-name"><a title="Bio.Pathway.Reaction.reverse
Bio.Seq.MutableSeq.reverse" class="py-name" href="#" onclick="return doclink('link-436', 'reverse', 'link-401');">reverse</a></tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L1035"></a><tt class="py-lineno">1035</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">flip</tt><tt class="py-op">:</tt> </tt>
<a name="L1036"></a><tt class="py-lineno">1036</tt>  <tt class="py-line">                <tt class="py-comment"># Flipped, join end to start,</tt> </tt>
<a name="L1037"></a><tt class="py-lineno">1037</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-437" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_arc_line()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#_draw_arc_line"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_arc_line" class="py-name" href="#" onclick="return doclink('link-437', '_draw_arc_line', 'link-437');">_draw_arc_line</a></tt><tt class="py-op">(</tt><tt class="py-name">p</tt><tt class="py-op">,</tt> <tt class="py-name">inner_radius</tt><tt class="py-op">,</tt> <tt class="py-name">outer_radius</tt><tt class="py-op">,</tt> <tt class="py-name">i_end</tt><tt class="py-op">,</tt> <tt class="py-name">o_start</tt><tt class="py-op">)</tt> </tt>
<a name="L1038"></a><tt class="py-lineno">1038</tt>  <tt class="py-line">                <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">addArc</tt><tt class="py-op">(</tt><tt class="py-name">x0</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt><tt class="py-op">,</tt> <tt class="py-name">outer_radius</tt><tt class="py-op">,</tt> <tt class="py-name">o_end</tt><tt class="py-op">,</tt> <tt class="py-name">o_start</tt><tt class="py-op">,</tt> <tt id="link-438" class="py-name"><a title="Bio.Pathway.Reaction.reverse
Bio.Seq.MutableSeq.reverse" class="py-name" href="#" onclick="return doclink('link-438', 'reverse', 'link-401');">reverse</a></tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L1039"></a><tt class="py-lineno">1039</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-439" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_arc_line" class="py-name" href="#" onclick="return doclink('link-439', '_draw_arc_line', 'link-437');">_draw_arc_line</a></tt><tt class="py-op">(</tt><tt class="py-name">p</tt><tt class="py-op">,</tt> <tt class="py-name">outer_radius</tt><tt class="py-op">,</tt> <tt class="py-name">inner_radius</tt><tt class="py-op">,</tt> <tt class="py-name">o_end</tt><tt class="py-op">,</tt> <tt class="py-name">i_start</tt><tt class="py-op">)</tt> </tt>
<a name="L1040"></a><tt class="py-lineno">1040</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1041"></a><tt class="py-lineno">1041</tt>  <tt class="py-line">                <tt class="py-comment"># Not flipped, join start to start, end to end</tt> </tt>
<a name="L1042"></a><tt class="py-lineno">1042</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-440" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_arc_line" class="py-name" href="#" onclick="return doclink('link-440', '_draw_arc_line', 'link-437');">_draw_arc_line</a></tt><tt class="py-op">(</tt><tt class="py-name">p</tt><tt class="py-op">,</tt> <tt class="py-name">inner_radius</tt><tt class="py-op">,</tt> <tt class="py-name">outer_radius</tt><tt class="py-op">,</tt> <tt class="py-name">i_end</tt><tt class="py-op">,</tt> <tt class="py-name">o_end</tt><tt class="py-op">)</tt> </tt>
<a name="L1043"></a><tt class="py-lineno">1043</tt>  <tt class="py-line">                <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">addArc</tt><tt class="py-op">(</tt><tt class="py-name">x0</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt><tt class="py-op">,</tt> <tt class="py-name">outer_radius</tt><tt class="py-op">,</tt> <tt class="py-name">o_end</tt><tt class="py-op">,</tt> <tt class="py-name">o_start</tt><tt class="py-op">,</tt> </tt>
<a name="L1044"></a><tt class="py-lineno">1044</tt>  <tt class="py-line">                         <tt id="link-441" class="py-name"><a title="Bio.Pathway.Reaction.reverse
Bio.Seq.MutableSeq.reverse" class="py-name" href="#" onclick="return doclink('link-441', 'reverse', 'link-401');">reverse</a></tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt> </tt>
<a name="L1045"></a><tt class="py-lineno">1045</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-442" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_arc_line" class="py-name" href="#" onclick="return doclink('link-442', '_draw_arc_line', 'link-437');">_draw_arc_line</a></tt><tt class="py-op">(</tt><tt class="py-name">p</tt><tt class="py-op">,</tt> <tt class="py-name">outer_radius</tt><tt class="py-op">,</tt> <tt class="py-name">inner_radius</tt><tt class="py-op">,</tt> <tt class="py-name">o_start</tt><tt class="py-op">,</tt> <tt class="py-name">i_start</tt><tt class="py-op">)</tt> </tt>
<a name="L1046"></a><tt class="py-lineno">1046</tt>  <tt class="py-line">            <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">closePath</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1047"></a><tt class="py-lineno">1047</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">p</tt> </tt>
<a name="L1048"></a><tt class="py-lineno">1048</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1049"></a><tt class="py-lineno">1049</tt>  <tt class="py-line">            <tt class="py-comment"># Cheat and just use a four sided polygon.</tt> </tt>
<a name="L1050"></a><tt class="py-lineno">1050</tt>  <tt class="py-line">            <tt class="py-comment"># Calculate trig values for angle and coordinates</tt> </tt>
<a name="L1051"></a><tt class="py-lineno">1051</tt>  <tt class="py-line">            <tt class="py-name">inner_startcos</tt><tt class="py-op">,</tt> <tt class="py-name">inner_startsin</tt> <tt class="py-op">=</tt> <tt class="py-name">cos</tt><tt class="py-op">(</tt><tt class="py-name">inner_startangle</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">sin</tt><tt class="py-op">(</tt><tt class="py-name">inner_startangle</tt><tt class="py-op">)</tt> </tt>
<a name="L1052"></a><tt class="py-lineno">1052</tt>  <tt class="py-line">            <tt class="py-name">inner_endcos</tt><tt class="py-op">,</tt> <tt class="py-name">inner_endsin</tt> <tt class="py-op">=</tt> <tt class="py-name">cos</tt><tt class="py-op">(</tt><tt class="py-name">inner_endangle</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">sin</tt><tt class="py-op">(</tt><tt class="py-name">inner_endangle</tt><tt class="py-op">)</tt> </tt>
<a name="L1053"></a><tt class="py-lineno">1053</tt>  <tt class="py-line">            <tt class="py-name">outer_startcos</tt><tt class="py-op">,</tt> <tt class="py-name">outer_startsin</tt> <tt class="py-op">=</tt> <tt class="py-name">cos</tt><tt class="py-op">(</tt><tt class="py-name">outer_startangle</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">sin</tt><tt class="py-op">(</tt><tt class="py-name">outer_startangle</tt><tt class="py-op">)</tt> </tt>
<a name="L1054"></a><tt class="py-lineno">1054</tt>  <tt class="py-line">            <tt class="py-name">outer_endcos</tt><tt class="py-op">,</tt> <tt class="py-name">outer_endsin</tt> <tt class="py-op">=</tt> <tt class="py-name">cos</tt><tt class="py-op">(</tt><tt class="py-name">outer_endangle</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">sin</tt><tt class="py-op">(</tt><tt class="py-name">outer_endangle</tt><tt class="py-op">)</tt> </tt>
<a name="L1055"></a><tt class="py-lineno">1055</tt>  <tt class="py-line">            <tt class="py-name">x1</tt><tt class="py-op">,</tt> <tt class="py-name">y1</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">inner_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">inner_startsin</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">inner_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">inner_startcos</tt><tt class="py-op">)</tt> </tt>
<a name="L1056"></a><tt class="py-lineno">1056</tt>  <tt class="py-line">            <tt class="py-name">x2</tt><tt class="py-op">,</tt> <tt class="py-name">y2</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">inner_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">inner_endsin</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">inner_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">inner_endcos</tt><tt class="py-op">)</tt> </tt>
<a name="L1057"></a><tt class="py-lineno">1057</tt>  <tt class="py-line">            <tt class="py-name">x3</tt><tt class="py-op">,</tt> <tt class="py-name">y3</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">outer_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">outer_endsin</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">outer_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">outer_endcos</tt><tt class="py-op">)</tt> </tt>
<a name="L1058"></a><tt class="py-lineno">1058</tt>  <tt class="py-line">            <tt class="py-name">x4</tt><tt class="py-op">,</tt> <tt class="py-name">y4</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">outer_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">outer_startsin</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">outer_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">outer_startcos</tt><tt class="py-op">)</tt> </tt>
<a name="L1059"></a><tt class="py-lineno">1059</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-443" class="py-name"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer.draw_polygon" class="py-name" href="#" onclick="return doclink('link-443', 'draw_polygon', 'link-6');">draw_polygon</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">x1</tt><tt class="py-op">,</tt> <tt class="py-name">y1</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">x2</tt><tt class="py-op">,</tt> <tt class="py-name">y2</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">x3</tt><tt class="py-op">,</tt> <tt class="py-name">y3</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">x4</tt><tt class="py-op">,</tt> <tt class="py-name">y4</tt><tt class="py-op">)</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt id="link-444" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-444', 'color', 'link-92');">color</a></tt><tt class="py-op">,</tt> <tt class="py-name">border</tt><tt class="py-op">,</tt> </tt>
<a name="L1060"></a><tt class="py-lineno">1060</tt>  <tt class="py-line">                                <tt class="py-comment"># default is mitre/miter which can stick out too much:</tt> </tt>
<a name="L1061"></a><tt class="py-lineno">1061</tt>  <tt class="py-line">                                <tt class="py-name">strokeLineJoin</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">,</tt>  <tt class="py-comment"># 1=round</tt> </tt>
<a name="L1062"></a><tt class="py-lineno">1062</tt>  <tt class="py-line">                                <tt class="py-op">)</tt> </tt>
</div><a name="L1063"></a><tt class="py-lineno">1063</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer._draw_sigil_cut_corner_box"></a><div id="CircularDrawer._draw_sigil_cut_corner_box-def"><a name="L1064"></a><tt class="py-lineno">1064</tt> <a class="py-toggle" href="#" id="CircularDrawer._draw_sigil_cut_corner_box-toggle" onclick="return toggle('CircularDrawer._draw_sigil_cut_corner_box');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#_draw_sigil_cut_corner_box">_draw_sigil_cut_corner_box</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">bottom</tt><tt class="py-op">,</tt> <tt class="py-param">center</tt><tt class="py-op">,</tt> <tt class="py-param">top</tt><tt class="py-op">,</tt> </tt>
<a name="L1065"></a><tt class="py-lineno">1065</tt>  <tt class="py-line">                                   <tt class="py-param">startangle</tt><tt class="py-op">,</tt> <tt class="py-param">endangle</tt><tt class="py-op">,</tt> <tt class="py-param">strand</tt><tt class="py-op">,</tt> </tt>
<a name="L1066"></a><tt class="py-lineno">1066</tt>  <tt class="py-line">                                   <tt class="py-param">color</tt><tt class="py-op">,</tt> <tt class="py-param">border</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">corner</tt><tt class="py-op">=</tt><tt class="py-number">0.5</tt><tt class="py-op">,</tt> </tt>
<a name="L1067"></a><tt class="py-lineno">1067</tt>  <tt class="py-line">                                   <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer._draw_sigil_cut_corner_box-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CircularDrawer._draw_sigil_cut_corner_box-expanded"><a name="L1068"></a><tt class="py-lineno">1068</tt>  <tt class="py-line">        <tt class="py-docstring">"""Draw OCTO sigil, box with corners cut off (PRIVATE)."""</tt> </tt>
<a name="L1069"></a><tt class="py-lineno">1069</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-445" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-445', 'strand', 'link-91');">strand</a></tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L1070"></a><tt class="py-lineno">1070</tt>  <tt class="py-line">            <tt class="py-name">inner_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">center</tt> </tt>
<a name="L1071"></a><tt class="py-lineno">1071</tt>  <tt class="py-line">            <tt class="py-name">outer_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">top</tt> </tt>
<a name="L1072"></a><tt class="py-lineno">1072</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-446" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-446', 'strand', 'link-91');">strand</a></tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L1073"></a><tt class="py-lineno">1073</tt>  <tt class="py-line">            <tt class="py-name">inner_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">bottom</tt> </tt>
<a name="L1074"></a><tt class="py-lineno">1074</tt>  <tt class="py-line">            <tt class="py-name">outer_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">center</tt> </tt>
<a name="L1075"></a><tt class="py-lineno">1075</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1076"></a><tt class="py-lineno">1076</tt>  <tt class="py-line">            <tt class="py-name">inner_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">bottom</tt> </tt>
<a name="L1077"></a><tt class="py-lineno">1077</tt>  <tt class="py-line">            <tt class="py-name">outer_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">top</tt> </tt>
<a name="L1078"></a><tt class="py-lineno">1078</tt>  <tt class="py-line"> </tt>
<a name="L1079"></a><tt class="py-lineno">1079</tt>  <tt class="py-line">        <tt class="py-name">strokecolor</tt><tt class="py-op">,</tt> <tt id="link-447" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-447', 'color', 'link-92');">color</a></tt> <tt class="py-op">=</tt> <tt id="link-448" class="py-name"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer._stroke_and_fill_colors" class="py-name" href="#" onclick="return doclink('link-448', '_stroke_and_fill_colors', 'link-9');">_stroke_and_fill_colors</a></tt><tt class="py-op">(</tt><tt id="link-449" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-449', 'color', 'link-92');">color</a></tt><tt class="py-op">,</tt> <tt class="py-name">border</tt><tt class="py-op">)</tt> </tt>
<a name="L1080"></a><tt class="py-lineno">1080</tt>  <tt class="py-line"> </tt>
<a name="L1081"></a><tt class="py-lineno">1081</tt>  <tt class="py-line">        <tt class="py-name">startangle</tt><tt class="py-op">,</tt> <tt class="py-name">endangle</tt> <tt class="py-op">=</tt> <tt id="link-450" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-450', 'min', 'link-21');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">startangle</tt><tt class="py-op">,</tt> <tt class="py-name">endangle</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-451" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-451', 'max', 'link-23');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">startangle</tt><tt class="py-op">,</tt> <tt class="py-name">endangle</tt><tt class="py-op">)</tt> </tt>
<a name="L1082"></a><tt class="py-lineno">1082</tt>  <tt class="py-line">        <tt id="link-452" class="py-name"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-452', 'angle', 'link-311');">angle</a></tt> <tt class="py-op">=</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">endangle</tt> <tt class="py-op">-</tt> <tt class="py-name">startangle</tt><tt class="py-op">)</tt> </tt>
<a name="L1083"></a><tt class="py-lineno">1083</tt>  <tt class="py-line"> </tt>
<a name="L1084"></a><tt class="py-lineno">1084</tt>  <tt class="py-line">        <tt class="py-name">middle_radius</tt> <tt class="py-op">=</tt> <tt class="py-number">0.5</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-name">inner_radius</tt> <tt class="py-op">+</tt> <tt class="py-name">outer_radius</tt><tt class="py-op">)</tt> </tt>
<a name="L1085"></a><tt class="py-lineno">1085</tt>  <tt class="py-line">        <tt class="py-name">boxheight</tt> <tt class="py-op">=</tt> <tt class="py-name">outer_radius</tt> <tt class="py-op">-</tt> <tt class="py-name">inner_radius</tt> </tt>
<a name="L1086"></a><tt class="py-lineno">1086</tt>  <tt class="py-line"> </tt>
<a name="L1087"></a><tt class="py-lineno">1087</tt>  <tt class="py-line">        <tt class="py-name">corner_len</tt> <tt class="py-op">=</tt> <tt id="link-453" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-453', 'min', 'link-21');">min</a></tt><tt class="py-op">(</tt><tt class="py-number">0.5</tt> <tt class="py-op">*</tt> <tt class="py-name">boxheight</tt><tt class="py-op">,</tt> <tt class="py-number">0.5</tt> <tt class="py-op">*</tt> <tt class="py-name">boxheight</tt> <tt class="py-op">*</tt> <tt class="py-name">corner</tt><tt class="py-op">)</tt> </tt>
<a name="L1088"></a><tt class="py-lineno">1088</tt>  <tt class="py-line">        <tt class="py-name">shaft_inner_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">inner_radius</tt> <tt class="py-op">+</tt> <tt class="py-name">corner_len</tt> </tt>
<a name="L1089"></a><tt class="py-lineno">1089</tt>  <tt class="py-line">        <tt class="py-name">shaft_outer_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">outer_radius</tt> <tt class="py-op">-</tt> <tt class="py-name">corner_len</tt> </tt>
<a name="L1090"></a><tt class="py-lineno">1090</tt>  <tt class="py-line"> </tt>
<a name="L1091"></a><tt class="py-lineno">1091</tt>  <tt class="py-line">        <tt class="py-name">cornerangle_delta</tt> <tt class="py-op">=</tt> <tt id="link-454" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-454', 'max', 'link-23');">max</a></tt><tt class="py-op">(</tt><tt class="py-number">0.0</tt><tt class="py-op">,</tt> <tt id="link-455" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-455', 'min', 'link-21');">min</a></tt><tt class="py-op">(</tt><tt id="link-456" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.abs" class="py-name" href="#" onclick="return doclink('link-456', 'abs', 'link-395');">abs</a></tt><tt class="py-op">(</tt><tt class="py-name">boxheight</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt class="py-number">0.5</tt> <tt class="py-op">*</tt> <tt class="py-name">corner</tt> <tt class="py-op">/</tt> <tt class="py-name">middle_radius</tt><tt class="py-op">,</tt> <tt id="link-457" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.abs" class="py-name" href="#" onclick="return doclink('link-457', 'abs', 'link-395');">abs</a></tt><tt class="py-op">(</tt><tt id="link-458" class="py-name"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-458', 'angle', 'link-311');">angle</a></tt> <tt class="py-op">*</tt> <tt class="py-number">0.5</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1092"></a><tt class="py-lineno">1092</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-459" class="py-name"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-459', 'angle', 'link-311');">angle</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1093"></a><tt class="py-lineno">1093</tt>  <tt class="py-line">            <tt class="py-name">cornerangle_delta</tt> <tt class="py-op">*=</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt>  <tt class="py-comment"># reverse it</tt> </tt>
<a name="L1094"></a><tt class="py-lineno">1094</tt>  <tt class="py-line"> </tt>
<a name="L1095"></a><tt class="py-lineno">1095</tt>  <tt class="py-line">        <tt class="py-comment"># Calculate trig values for angle and coordinates</tt> </tt>
<a name="L1096"></a><tt class="py-lineno">1096</tt>  <tt class="py-line">        <tt class="py-name">startcos</tt><tt class="py-op">,</tt> <tt class="py-name">startsin</tt> <tt class="py-op">=</tt> <tt class="py-name">cos</tt><tt class="py-op">(</tt><tt class="py-name">startangle</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">sin</tt><tt class="py-op">(</tt><tt class="py-name">startangle</tt><tt class="py-op">)</tt> </tt>
<a name="L1097"></a><tt class="py-lineno">1097</tt>  <tt class="py-line">        <tt class="py-name">endcos</tt><tt class="py-op">,</tt> <tt class="py-name">endsin</tt> <tt class="py-op">=</tt> <tt class="py-name">cos</tt><tt class="py-op">(</tt><tt class="py-name">endangle</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">sin</tt><tt class="py-op">(</tt><tt class="py-name">endangle</tt><tt class="py-op">)</tt> </tt>
<a name="L1098"></a><tt class="py-lineno">1098</tt>  <tt class="py-line">        <tt class="py-name">x0</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt>      <tt class="py-comment"># origin of the circle</tt> </tt>
<a name="L1099"></a><tt class="py-lineno">1099</tt>  <tt class="py-line">        <tt class="py-name">p</tt> <tt class="py-op">=</tt> <tt class="py-name">ArcPath</tt><tt class="py-op">(</tt><tt class="py-name">strokeColor</tt><tt class="py-op">=</tt><tt class="py-name">strokecolor</tt><tt class="py-op">,</tt> </tt>
<a name="L1100"></a><tt class="py-lineno">1100</tt>  <tt class="py-line">                    <tt class="py-name">fillColor</tt><tt class="py-op">=</tt><tt id="link-460" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-460', 'color', 'link-92');">color</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1101"></a><tt class="py-lineno">1101</tt>  <tt class="py-line">                    <tt class="py-name">strokeLineJoin</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">,</tt>  <tt class="py-comment"># 1=round</tt> </tt>
<a name="L1102"></a><tt class="py-lineno">1102</tt>  <tt class="py-line">                    <tt class="py-name">strokewidth</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> </tt>
<a name="L1103"></a><tt class="py-lineno">1103</tt>  <tt class="py-line">                    <tt class="py-op">**</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> </tt>
<a name="L1104"></a><tt class="py-lineno">1104</tt>  <tt class="py-line">        <tt class="py-comment"># Inner curved edge</tt> </tt>
<a name="L1105"></a><tt class="py-lineno">1105</tt>  <tt class="py-line">        <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">addArc</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt><tt class="py-op">,</tt> <tt class="py-name">inner_radius</tt><tt class="py-op">,</tt> </tt>
<a name="L1106"></a><tt class="py-lineno">1106</tt>  <tt class="py-line">                 <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">endangle</tt> <tt class="py-op">-</tt> <tt class="py-name">cornerangle_delta</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-461" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-461', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1107"></a><tt class="py-lineno">1107</tt>  <tt class="py-line">                 <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">startangle</tt> <tt class="py-op">+</tt> <tt class="py-name">cornerangle_delta</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-462" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-462', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1108"></a><tt class="py-lineno">1108</tt>  <tt class="py-line">                 <tt class="py-name">moveTo</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L1109"></a><tt class="py-lineno">1109</tt>  <tt class="py-line">        <tt class="py-comment"># Corner edge - straight lines assumes small angle!</tt> </tt>
<a name="L1110"></a><tt class="py-lineno">1110</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - Use self._draw_arc_line(p, ...) here if we expose corner setting</tt> </tt>
<a name="L1111"></a><tt class="py-lineno">1111</tt>  <tt class="py-line">        <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">lineTo</tt><tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">shaft_inner_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">startsin</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">shaft_inner_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">startcos</tt><tt class="py-op">)</tt> </tt>
<a name="L1112"></a><tt class="py-lineno">1112</tt>  <tt class="py-line">        <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">lineTo</tt><tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">shaft_outer_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">startsin</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">shaft_outer_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">startcos</tt><tt class="py-op">)</tt> </tt>
<a name="L1113"></a><tt class="py-lineno">1113</tt>  <tt class="py-line">        <tt class="py-comment"># Outer curved edge</tt> </tt>
<a name="L1114"></a><tt class="py-lineno">1114</tt>  <tt class="py-line">        <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">addArc</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt><tt class="py-op">,</tt> <tt class="py-name">outer_radius</tt><tt class="py-op">,</tt> </tt>
<a name="L1115"></a><tt class="py-lineno">1115</tt>  <tt class="py-line">                 <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">endangle</tt> <tt class="py-op">-</tt> <tt class="py-name">cornerangle_delta</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-463" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-463', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1116"></a><tt class="py-lineno">1116</tt>  <tt class="py-line">                 <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">startangle</tt> <tt class="py-op">+</tt> <tt class="py-name">cornerangle_delta</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-464" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-464', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1117"></a><tt class="py-lineno">1117</tt>  <tt class="py-line">                 <tt id="link-465" class="py-name"><a title="Bio.Pathway.Reaction.reverse
Bio.Seq.MutableSeq.reverse" class="py-name" href="#" onclick="return doclink('link-465', 'reverse', 'link-401');">reverse</a></tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L1118"></a><tt class="py-lineno">1118</tt>  <tt class="py-line">        <tt class="py-comment"># Corner edges</tt> </tt>
<a name="L1119"></a><tt class="py-lineno">1119</tt>  <tt class="py-line">        <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">lineTo</tt><tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">shaft_outer_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">endsin</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">shaft_outer_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">endcos</tt><tt class="py-op">)</tt> </tt>
<a name="L1120"></a><tt class="py-lineno">1120</tt>  <tt class="py-line">        <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">lineTo</tt><tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">shaft_inner_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">endsin</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">shaft_inner_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">endcos</tt><tt class="py-op">)</tt> </tt>
<a name="L1121"></a><tt class="py-lineno">1121</tt>  <tt class="py-line">        <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">closePath</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1122"></a><tt class="py-lineno">1122</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">p</tt> </tt>
</div><a name="L1123"></a><tt class="py-lineno">1123</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer._draw_sigil_arrow"></a><div id="CircularDrawer._draw_sigil_arrow-def"><a name="L1124"></a><tt class="py-lineno">1124</tt> <a class="py-toggle" href="#" id="CircularDrawer._draw_sigil_arrow-toggle" onclick="return toggle('CircularDrawer._draw_sigil_arrow');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#_draw_sigil_arrow">_draw_sigil_arrow</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">bottom</tt><tt class="py-op">,</tt> <tt class="py-param">center</tt><tt class="py-op">,</tt> <tt class="py-param">top</tt><tt class="py-op">,</tt> </tt>
<a name="L1125"></a><tt class="py-lineno">1125</tt>  <tt class="py-line">                          <tt class="py-param">startangle</tt><tt class="py-op">,</tt> <tt class="py-param">endangle</tt><tt class="py-op">,</tt> <tt class="py-param">strand</tt><tt class="py-op">,</tt> </tt>
<a name="L1126"></a><tt class="py-lineno">1126</tt>  <tt class="py-line">                          <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer._draw_sigil_arrow-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CircularDrawer._draw_sigil_arrow-expanded"><a name="L1127"></a><tt class="py-lineno">1127</tt>  <tt class="py-line">        <tt class="py-docstring">"""Draw ARROW sigil (PRIVATE)."""</tt> </tt>
<a name="L1128"></a><tt class="py-lineno">1128</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-466" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-466', 'strand', 'link-91');">strand</a></tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L1129"></a><tt class="py-lineno">1129</tt>  <tt class="py-line">            <tt class="py-name">inner_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">center</tt> </tt>
<a name="L1130"></a><tt class="py-lineno">1130</tt>  <tt class="py-line">            <tt class="py-name">outer_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">top</tt> </tt>
<a name="L1131"></a><tt class="py-lineno">1131</tt>  <tt class="py-line">            <tt class="py-name">orientation</tt> <tt class="py-op">=</tt> <tt class="py-string">"right"</tt> </tt>
<a name="L1132"></a><tt class="py-lineno">1132</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-467" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-467', 'strand', 'link-91');">strand</a></tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L1133"></a><tt class="py-lineno">1133</tt>  <tt class="py-line">            <tt class="py-name">inner_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">bottom</tt> </tt>
<a name="L1134"></a><tt class="py-lineno">1134</tt>  <tt class="py-line">            <tt class="py-name">outer_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">center</tt> </tt>
<a name="L1135"></a><tt class="py-lineno">1135</tt>  <tt class="py-line">            <tt class="py-name">orientation</tt> <tt class="py-op">=</tt> <tt class="py-string">"left"</tt> </tt>
<a name="L1136"></a><tt class="py-lineno">1136</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1137"></a><tt class="py-lineno">1137</tt>  <tt class="py-line">            <tt class="py-name">inner_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">bottom</tt> </tt>
<a name="L1138"></a><tt class="py-lineno">1138</tt>  <tt class="py-line">            <tt class="py-name">outer_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">top</tt> </tt>
<a name="L1139"></a><tt class="py-lineno">1139</tt>  <tt class="py-line">            <tt class="py-name">orientation</tt> <tt class="py-op">=</tt> <tt class="py-string">"right"</tt>  <tt class="py-comment"># backwards compatibility</tt> </tt>
<a name="L1140"></a><tt class="py-lineno">1140</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-468" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_arc_arrow()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#_draw_arc_arrow"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_arc_arrow" class="py-name" href="#" onclick="return doclink('link-468', '_draw_arc_arrow', 'link-468');">_draw_arc_arrow</a></tt><tt class="py-op">(</tt><tt class="py-name">inner_radius</tt><tt class="py-op">,</tt> <tt class="py-name">outer_radius</tt><tt class="py-op">,</tt> <tt class="py-name">startangle</tt><tt class="py-op">,</tt> <tt class="py-name">endangle</tt><tt class="py-op">,</tt> </tt>
<a name="L1141"></a><tt class="py-lineno">1141</tt>  <tt class="py-line">                                    <tt class="py-name">orientation</tt><tt class="py-op">=</tt><tt class="py-name">orientation</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1142"></a><tt class="py-lineno">1142</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer._draw_sigil_big_arrow"></a><div id="CircularDrawer._draw_sigil_big_arrow-def"><a name="L1143"></a><tt class="py-lineno">1143</tt> <a class="py-toggle" href="#" id="CircularDrawer._draw_sigil_big_arrow-toggle" onclick="return toggle('CircularDrawer._draw_sigil_big_arrow');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#_draw_sigil_big_arrow">_draw_sigil_big_arrow</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">bottom</tt><tt class="py-op">,</tt> <tt class="py-param">center</tt><tt class="py-op">,</tt> <tt class="py-param">top</tt><tt class="py-op">,</tt> </tt>
<a name="L1144"></a><tt class="py-lineno">1144</tt>  <tt class="py-line">                              <tt class="py-param">startangle</tt><tt class="py-op">,</tt> <tt class="py-param">endangle</tt><tt class="py-op">,</tt> <tt class="py-param">strand</tt><tt class="py-op">,</tt> </tt>
<a name="L1145"></a><tt class="py-lineno">1145</tt>  <tt class="py-line">                              <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer._draw_sigil_big_arrow-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CircularDrawer._draw_sigil_big_arrow-expanded"><a name="L1146"></a><tt class="py-lineno">1146</tt>  <tt class="py-line">        <tt class="py-docstring">"""Draw BIGARROW sigil, like ARROW but straddles the axis (PRIVATE)."""</tt> </tt>
<a name="L1147"></a><tt class="py-lineno">1147</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-469" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-469', 'strand', 'link-91');">strand</a></tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L1148"></a><tt class="py-lineno">1148</tt>  <tt class="py-line">            <tt class="py-name">orientation</tt> <tt class="py-op">=</tt> <tt class="py-string">"left"</tt> </tt>
<a name="L1149"></a><tt class="py-lineno">1149</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1150"></a><tt class="py-lineno">1150</tt>  <tt class="py-line">            <tt class="py-name">orientation</tt> <tt class="py-op">=</tt> <tt class="py-string">"right"</tt> </tt>
<a name="L1151"></a><tt class="py-lineno">1151</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-470" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_arc_arrow" class="py-name" href="#" onclick="return doclink('link-470', '_draw_arc_arrow', 'link-468');">_draw_arc_arrow</a></tt><tt class="py-op">(</tt><tt class="py-name">bottom</tt><tt class="py-op">,</tt> <tt class="py-name">top</tt><tt class="py-op">,</tt> <tt class="py-name">startangle</tt><tt class="py-op">,</tt> <tt class="py-name">endangle</tt><tt class="py-op">,</tt> </tt>
<a name="L1152"></a><tt class="py-lineno">1152</tt>  <tt class="py-line">                                    <tt class="py-name">orientation</tt><tt class="py-op">=</tt><tt class="py-name">orientation</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1153"></a><tt class="py-lineno">1153</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer._draw_arc_arrow"></a><div id="CircularDrawer._draw_arc_arrow-def"><a name="L1154"></a><tt class="py-lineno">1154</tt> <a class="py-toggle" href="#" id="CircularDrawer._draw_arc_arrow-toggle" onclick="return toggle('CircularDrawer._draw_arc_arrow');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#_draw_arc_arrow">_draw_arc_arrow</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">inner_radius</tt><tt class="py-op">,</tt> <tt class="py-param">outer_radius</tt><tt class="py-op">,</tt> <tt class="py-param">startangle</tt><tt class="py-op">,</tt> <tt class="py-param">endangle</tt><tt class="py-op">,</tt> </tt>
<a name="L1155"></a><tt class="py-lineno">1155</tt>  <tt class="py-line">                        <tt class="py-param">color</tt><tt class="py-op">,</tt> <tt class="py-param">border</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L1156"></a><tt class="py-lineno">1156</tt>  <tt class="py-line">                        <tt class="py-param">shaft_height_ratio</tt><tt class="py-op">=</tt><tt class="py-number">0.4</tt><tt class="py-op">,</tt> <tt class="py-param">head_length_ratio</tt><tt class="py-op">=</tt><tt class="py-number">0.5</tt><tt class="py-op">,</tt> </tt>
<a name="L1157"></a><tt class="py-lineno">1157</tt>  <tt class="py-line">                        <tt class="py-param">orientation</tt><tt class="py-op">=</tt><tt class="py-string">'right'</tt><tt class="py-op">,</tt> <tt class="py-param">colour</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer._draw_arc_arrow-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CircularDrawer._draw_arc_arrow-expanded"><a name="L1158"></a><tt class="py-lineno">1158</tt>  <tt class="py-line">        <tt class="py-docstring">"""Draw an arrow along an arc (PRIVATE)."""</tt> </tt>
<a name="L1159"></a><tt class="py-lineno">1159</tt>  <tt class="py-line">        <tt class="py-comment"># Let the UK spelling (colour) override the USA spelling (color)</tt> </tt>
<a name="L1160"></a><tt class="py-lineno">1160</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">colour</tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1161"></a><tt class="py-lineno">1161</tt>  <tt class="py-line">            <tt id="link-471" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-471', 'color', 'link-92');">color</a></tt> <tt class="py-op">=</tt> <tt class="py-name">colour</tt> </tt>
<a name="L1162"></a><tt class="py-lineno">1162</tt>  <tt class="py-line"> </tt>
<a name="L1163"></a><tt class="py-lineno">1163</tt>  <tt class="py-line">        <tt class="py-name">strokecolor</tt><tt class="py-op">,</tt> <tt id="link-472" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-472', 'color', 'link-92');">color</a></tt> <tt class="py-op">=</tt> <tt id="link-473" class="py-name"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer._stroke_and_fill_colors" class="py-name" href="#" onclick="return doclink('link-473', '_stroke_and_fill_colors', 'link-9');">_stroke_and_fill_colors</a></tt><tt class="py-op">(</tt><tt id="link-474" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-474', 'color', 'link-92');">color</a></tt><tt class="py-op">,</tt> <tt class="py-name">border</tt><tt class="py-op">)</tt> </tt>
<a name="L1164"></a><tt class="py-lineno">1164</tt>  <tt class="py-line"> </tt>
<a name="L1165"></a><tt class="py-lineno">1165</tt>  <tt class="py-line">        <tt class="py-comment"># if orientation == 'right':</tt> </tt>
<a name="L1166"></a><tt class="py-lineno">1166</tt>  <tt class="py-line">        <tt class="py-comment">#    startangle, endangle = min(startangle, endangle), max(startangle, endangle)</tt> </tt>
<a name="L1167"></a><tt class="py-lineno">1167</tt>  <tt class="py-line">        <tt class="py-comment"># elif orientation == 'left':</tt> </tt>
<a name="L1168"></a><tt class="py-lineno">1168</tt>  <tt class="py-line">        <tt class="py-comment">#    startangle, endangle = max(startangle, endangle), min(startangle, endangle)</tt> </tt>
<a name="L1169"></a><tt class="py-lineno">1169</tt>  <tt class="py-line">        <tt class="py-comment"># else:</tt> </tt>
<a name="L1170"></a><tt class="py-lineno">1170</tt>  <tt class="py-line">        <tt class="py-name">startangle</tt><tt class="py-op">,</tt> <tt class="py-name">endangle</tt> <tt class="py-op">=</tt> <tt id="link-475" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-475', 'min', 'link-21');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">startangle</tt><tt class="py-op">,</tt> <tt class="py-name">endangle</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-476" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-476', 'max', 'link-23');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">startangle</tt><tt class="py-op">,</tt> <tt class="py-name">endangle</tt><tt class="py-op">)</tt> </tt>
<a name="L1171"></a><tt class="py-lineno">1171</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">orientation</tt> <tt class="py-op">!=</tt> <tt class="py-string">"left"</tt> <tt class="py-keyword">and</tt> <tt class="py-name">orientation</tt> <tt class="py-op">!=</tt> <tt class="py-string">"right"</tt><tt class="py-op">:</tt> </tt>
<a name="L1172"></a><tt class="py-lineno">1172</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid orientation %s, should be 'left' or 'right'"</tt> </tt>
<a name="L1173"></a><tt class="py-lineno">1173</tt>  <tt class="py-line">                             <tt class="py-op">%</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">orientation</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1174"></a><tt class="py-lineno">1174</tt>  <tt class="py-line"> </tt>
<a name="L1175"></a><tt class="py-lineno">1175</tt>  <tt class="py-line">        <tt id="link-477" class="py-name"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-477', 'angle', 'link-311');">angle</a></tt> <tt class="py-op">=</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">endangle</tt> <tt class="py-op">-</tt> <tt class="py-name">startangle</tt><tt class="py-op">)</tt>    <tt class="py-comment"># angle subtended by arc</tt> </tt>
<a name="L1176"></a><tt class="py-lineno">1176</tt>  <tt class="py-line">        <tt class="py-name">middle_radius</tt> <tt class="py-op">=</tt> <tt class="py-number">0.5</tt> <tt class="py-op">*</tt> <tt class="py-op">(</tt><tt class="py-name">inner_radius</tt> <tt class="py-op">+</tt> <tt class="py-name">outer_radius</tt><tt class="py-op">)</tt> </tt>
<a name="L1177"></a><tt class="py-lineno">1177</tt>  <tt class="py-line">        <tt class="py-name">boxheight</tt> <tt class="py-op">=</tt> <tt class="py-name">outer_radius</tt> <tt class="py-op">-</tt> <tt class="py-name">inner_radius</tt> </tt>
<a name="L1178"></a><tt class="py-lineno">1178</tt>  <tt class="py-line">        <tt class="py-name">shaft_height</tt> <tt class="py-op">=</tt> <tt class="py-name">boxheight</tt> <tt class="py-op">*</tt> <tt class="py-name">shaft_height_ratio</tt> </tt>
<a name="L1179"></a><tt class="py-lineno">1179</tt>  <tt class="py-line">        <tt class="py-name">shaft_inner_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">middle_radius</tt> <tt class="py-op">-</tt> <tt class="py-number">0.5</tt> <tt class="py-op">*</tt> <tt class="py-name">shaft_height</tt> </tt>
<a name="L1180"></a><tt class="py-lineno">1180</tt>  <tt class="py-line">        <tt class="py-name">shaft_outer_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">middle_radius</tt> <tt class="py-op">+</tt> <tt class="py-number">0.5</tt> <tt class="py-op">*</tt> <tt class="py-name">shaft_height</tt> </tt>
<a name="L1181"></a><tt class="py-lineno">1181</tt>  <tt class="py-line">        <tt class="py-name">headangle_delta</tt> <tt class="py-op">=</tt> <tt id="link-478" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-478', 'max', 'link-23');">max</a></tt><tt class="py-op">(</tt><tt class="py-number">0.0</tt><tt class="py-op">,</tt> <tt id="link-479" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-479', 'min', 'link-21');">min</a></tt><tt class="py-op">(</tt><tt id="link-480" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.abs" class="py-name" href="#" onclick="return doclink('link-480', 'abs', 'link-395');">abs</a></tt><tt class="py-op">(</tt><tt class="py-name">boxheight</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt class="py-name">head_length_ratio</tt> <tt class="py-op">/</tt> <tt class="py-name">middle_radius</tt><tt class="py-op">,</tt> <tt id="link-481" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.abs" class="py-name" href="#" onclick="return doclink('link-481', 'abs', 'link-395');">abs</a></tt><tt class="py-op">(</tt><tt id="link-482" class="py-name"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-482', 'angle', 'link-311');">angle</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1182"></a><tt class="py-lineno">1182</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-483" class="py-name"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-483', 'angle', 'link-311');">angle</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1183"></a><tt class="py-lineno">1183</tt>  <tt class="py-line">            <tt class="py-name">headangle_delta</tt> <tt class="py-op">*=</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt>  <tt class="py-comment"># reverse it</tt> </tt>
<a name="L1184"></a><tt class="py-lineno">1184</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">orientation</tt> <tt class="py-op">==</tt> <tt class="py-string">"right"</tt><tt class="py-op">:</tt> </tt>
<a name="L1185"></a><tt class="py-lineno">1185</tt>  <tt class="py-line">            <tt class="py-name">headangle</tt> <tt class="py-op">=</tt> <tt class="py-name">endangle</tt> <tt class="py-op">-</tt> <tt class="py-name">headangle_delta</tt> </tt>
<a name="L1186"></a><tt class="py-lineno">1186</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1187"></a><tt class="py-lineno">1187</tt>  <tt class="py-line">            <tt class="py-name">headangle</tt> <tt class="py-op">=</tt> <tt class="py-name">startangle</tt> <tt class="py-op">+</tt> <tt class="py-name">headangle_delta</tt> </tt>
<a name="L1188"></a><tt class="py-lineno">1188</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">startangle</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">endangle</tt><tt class="py-op">:</tt> </tt>
<a name="L1189"></a><tt class="py-lineno">1189</tt>  <tt class="py-line">            <tt class="py-name">headangle</tt> <tt class="py-op">=</tt> <tt id="link-484" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-484', 'max', 'link-23');">max</a></tt><tt class="py-op">(</tt><tt id="link-485" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-485', 'min', 'link-21');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">headangle</tt><tt class="py-op">,</tt> <tt class="py-name">endangle</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">startangle</tt><tt class="py-op">)</tt> </tt>
<a name="L1190"></a><tt class="py-lineno">1190</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1191"></a><tt class="py-lineno">1191</tt>  <tt class="py-line">            <tt class="py-name">headangle</tt> <tt class="py-op">=</tt> <tt id="link-486" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-486', 'max', 'link-23');">max</a></tt><tt class="py-op">(</tt><tt id="link-487" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-487', 'min', 'link-21');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">headangle</tt><tt class="py-op">,</tt> <tt class="py-name">startangle</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">endangle</tt><tt class="py-op">)</tt> </tt>
<a name="L1192"></a><tt class="py-lineno">1192</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-op">(</tt><tt class="py-name">startangle</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">headangle</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">endangle</tt> </tt>
<a name="L1193"></a><tt class="py-lineno">1193</tt>  <tt class="py-line">                <tt class="py-keyword">or</tt> <tt class="py-name">endangle</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">headangle</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">startangle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1194"></a><tt class="py-lineno">1194</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">RuntimeError</tt><tt class="py-op">(</tt><tt class="py-string">"Problem drawing arrow, invalid positions. "</tt> </tt>
<a name="L1195"></a><tt class="py-lineno">1195</tt>  <tt class="py-line">                               <tt class="py-string">"Start angle: %s, Head angle: %s, "</tt> </tt>
<a name="L1196"></a><tt class="py-lineno">1196</tt>  <tt class="py-line">                               <tt class="py-string">"End angle: %s, Angle: %s"</tt> </tt>
<a name="L1197"></a><tt class="py-lineno">1197</tt>  <tt class="py-line">                               <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">startangle</tt><tt class="py-op">,</tt> <tt class="py-name">headangle</tt><tt class="py-op">,</tt> <tt class="py-name">endangle</tt><tt class="py-op">,</tt> <tt id="link-488" class="py-name"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-488', 'angle', 'link-311');">angle</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1198"></a><tt class="py-lineno">1198</tt>  <tt class="py-line"> </tt>
<a name="L1199"></a><tt class="py-lineno">1199</tt>  <tt class="py-line">        <tt class="py-comment"># Calculate trig values for angle and coordinates</tt> </tt>
<a name="L1200"></a><tt class="py-lineno">1200</tt>  <tt class="py-line">        <tt class="py-name">startcos</tt><tt class="py-op">,</tt> <tt class="py-name">startsin</tt> <tt class="py-op">=</tt> <tt class="py-name">cos</tt><tt class="py-op">(</tt><tt class="py-name">startangle</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">sin</tt><tt class="py-op">(</tt><tt class="py-name">startangle</tt><tt class="py-op">)</tt> </tt>
<a name="L1201"></a><tt class="py-lineno">1201</tt>  <tt class="py-line">        <tt class="py-name">headcos</tt><tt class="py-op">,</tt> <tt class="py-name">headsin</tt> <tt class="py-op">=</tt> <tt class="py-name">cos</tt><tt class="py-op">(</tt><tt class="py-name">headangle</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">sin</tt><tt class="py-op">(</tt><tt class="py-name">headangle</tt><tt class="py-op">)</tt> </tt>
<a name="L1202"></a><tt class="py-lineno">1202</tt>  <tt class="py-line">        <tt class="py-name">endcos</tt><tt class="py-op">,</tt> <tt class="py-name">endsin</tt> <tt class="py-op">=</tt> <tt class="py-name">cos</tt><tt class="py-op">(</tt><tt class="py-name">endangle</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">sin</tt><tt class="py-op">(</tt><tt class="py-name">endangle</tt><tt class="py-op">)</tt> </tt>
<a name="L1203"></a><tt class="py-lineno">1203</tt>  <tt class="py-line">        <tt class="py-name">x0</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt>      <tt class="py-comment"># origin of the circle</tt> </tt>
<a name="L1204"></a><tt class="py-lineno">1204</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-number">0.5</tt> <tt class="py-op">&gt;=</tt> <tt id="link-489" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.abs" class="py-name" href="#" onclick="return doclink('link-489', 'abs', 'link-395');">abs</a></tt><tt class="py-op">(</tt><tt id="link-490" class="py-name"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-490', 'angle', 'link-311');">angle</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">and</tt> <tt id="link-491" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.abs" class="py-name" href="#" onclick="return doclink('link-491', 'abs', 'link-395');">abs</a></tt><tt class="py-op">(</tt><tt class="py-name">headangle_delta</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;=</tt> <tt id="link-492" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.abs" class="py-name" href="#" onclick="return doclink('link-492', 'abs', 'link-395');">abs</a></tt><tt class="py-op">(</tt><tt id="link-493" class="py-name"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-493', 'angle', 'link-311');">angle</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1205"></a><tt class="py-lineno">1205</tt>  <tt class="py-line">            <tt class="py-comment"># If the angle is small, and the arrow is all head,</tt> </tt>
<a name="L1206"></a><tt class="py-lineno">1206</tt>  <tt class="py-line">            <tt class="py-comment"># cheat and just use a triangle.</tt> </tt>
<a name="L1207"></a><tt class="py-lineno">1207</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">orientation</tt> <tt class="py-op">==</tt> <tt class="py-string">"right"</tt><tt class="py-op">:</tt> </tt>
<a name="L1208"></a><tt class="py-lineno">1208</tt>  <tt class="py-line">                <tt class="py-name">x1</tt><tt class="py-op">,</tt> <tt class="py-name">y1</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">inner_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">startsin</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">inner_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">startcos</tt><tt class="py-op">)</tt> </tt>
<a name="L1209"></a><tt class="py-lineno">1209</tt>  <tt class="py-line">                <tt class="py-name">x2</tt><tt class="py-op">,</tt> <tt class="py-name">y2</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">outer_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">startsin</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">outer_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">startcos</tt><tt class="py-op">)</tt> </tt>
<a name="L1210"></a><tt class="py-lineno">1210</tt>  <tt class="py-line">                <tt class="py-name">x3</tt><tt class="py-op">,</tt> <tt class="py-name">y3</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">middle_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">endsin</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">middle_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">endcos</tt><tt class="py-op">)</tt> </tt>
<a name="L1211"></a><tt class="py-lineno">1211</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1212"></a><tt class="py-lineno">1212</tt>  <tt class="py-line">                <tt class="py-name">x1</tt><tt class="py-op">,</tt> <tt class="py-name">y1</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">inner_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">endsin</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">inner_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">endcos</tt><tt class="py-op">)</tt> </tt>
<a name="L1213"></a><tt class="py-lineno">1213</tt>  <tt class="py-line">                <tt class="py-name">x2</tt><tt class="py-op">,</tt> <tt class="py-name">y2</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">outer_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">endsin</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">outer_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">endcos</tt><tt class="py-op">)</tt> </tt>
<a name="L1214"></a><tt class="py-lineno">1214</tt>  <tt class="py-line">                <tt class="py-name">x3</tt><tt class="py-op">,</tt> <tt class="py-name">y3</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">middle_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">startsin</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">middle_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">startcos</tt><tt class="py-op">)</tt> </tt>
<a name="L1215"></a><tt class="py-lineno">1215</tt>  <tt class="py-line">            <tt class="py-comment"># return draw_polygon([(x1,y1),(x2,y2),(x3,y3)], color, border,</tt> </tt>
<a name="L1216"></a><tt class="py-lineno">1216</tt>  <tt class="py-line">            <tt class="py-comment">#                    stroke_line_join=1)</tt> </tt>
<a name="L1217"></a><tt class="py-lineno">1217</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-494" class="py-name"><a title="Bio.Phylo.PhyloXML.Polygon" class="py-name" href="#" onclick="return doclink('link-494', 'Polygon', 'link-0');">Polygon</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">x1</tt><tt class="py-op">,</tt> <tt class="py-name">y1</tt><tt class="py-op">,</tt> <tt class="py-name">x2</tt><tt class="py-op">,</tt> <tt class="py-name">y2</tt><tt class="py-op">,</tt> <tt class="py-name">x3</tt><tt class="py-op">,</tt> <tt class="py-name">y3</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L1218"></a><tt class="py-lineno">1218</tt>  <tt class="py-line">                           <tt class="py-name">strokeColor</tt><tt class="py-op">=</tt><tt class="py-name">border</tt> <tt class="py-keyword">or</tt> <tt id="link-495" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-495', 'color', 'link-92');">color</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1219"></a><tt class="py-lineno">1219</tt>  <tt class="py-line">                           <tt class="py-name">fillColor</tt><tt class="py-op">=</tt><tt id="link-496" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-496', 'color', 'link-92');">color</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1220"></a><tt class="py-lineno">1220</tt>  <tt class="py-line">                           <tt class="py-name">strokeLineJoin</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">,</tt>  <tt class="py-comment"># 1=round, not mitre!</tt> </tt>
<a name="L1221"></a><tt class="py-lineno">1221</tt>  <tt class="py-line">                           <tt class="py-name">strokewidth</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L1222"></a><tt class="py-lineno">1222</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">orientation</tt> <tt class="py-op">==</tt> <tt class="py-string">"right"</tt><tt class="py-op">:</tt> </tt>
<a name="L1223"></a><tt class="py-lineno">1223</tt>  <tt class="py-line">            <tt class="py-name">p</tt> <tt class="py-op">=</tt> <tt class="py-name">ArcPath</tt><tt class="py-op">(</tt><tt class="py-name">strokeColor</tt><tt class="py-op">=</tt><tt class="py-name">strokecolor</tt><tt class="py-op">,</tt> </tt>
<a name="L1224"></a><tt class="py-lineno">1224</tt>  <tt class="py-line">                        <tt class="py-name">fillColor</tt><tt class="py-op">=</tt><tt id="link-497" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-497', 'color', 'link-92');">color</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1225"></a><tt class="py-lineno">1225</tt>  <tt class="py-line">                        <tt class="py-comment"># default is mitre/miter which can stick out too much:</tt> </tt>
<a name="L1226"></a><tt class="py-lineno">1226</tt>  <tt class="py-line">                        <tt class="py-name">strokeLineJoin</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">,</tt>  <tt class="py-comment"># 1=round</tt> </tt>
<a name="L1227"></a><tt class="py-lineno">1227</tt>  <tt class="py-line">                        <tt class="py-name">strokewidth</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> </tt>
<a name="L1228"></a><tt class="py-lineno">1228</tt>  <tt class="py-line">                        <tt class="py-op">**</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> </tt>
<a name="L1229"></a><tt class="py-lineno">1229</tt>  <tt class="py-line">            <tt class="py-comment"># Note reportlab counts angles anti-clockwise from the horizontal</tt> </tt>
<a name="L1230"></a><tt class="py-lineno">1230</tt>  <tt class="py-line">            <tt class="py-comment"># (as in mathematics, e.g. complex numbers and polar coordinates)</tt> </tt>
<a name="L1231"></a><tt class="py-lineno">1231</tt>  <tt class="py-line">            <tt class="py-comment"># but we use clockwise from the vertical.  Also reportlab uses</tt> </tt>
<a name="L1232"></a><tt class="py-lineno">1232</tt>  <tt class="py-line">            <tt class="py-comment"># degrees, but we use radians.</tt> </tt>
<a name="L1233"></a><tt class="py-lineno">1233</tt>  <tt class="py-line">            <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">addArc</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt><tt class="py-op">,</tt> <tt class="py-name">shaft_inner_radius</tt><tt class="py-op">,</tt> </tt>
<a name="L1234"></a><tt class="py-lineno">1234</tt>  <tt class="py-line">                     <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">headangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-498" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-498', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">startangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-499" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-499', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1235"></a><tt class="py-lineno">1235</tt>  <tt class="py-line">                     <tt class="py-name">moveTo</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L1236"></a><tt class="py-lineno">1236</tt>  <tt class="py-line">            <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">addArc</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt><tt class="py-op">,</tt> <tt class="py-name">shaft_outer_radius</tt><tt class="py-op">,</tt> </tt>
<a name="L1237"></a><tt class="py-lineno">1237</tt>  <tt class="py-line">                     <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">headangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-500" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-500', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">startangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-501" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-501', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1238"></a><tt class="py-lineno">1238</tt>  <tt class="py-line">                     <tt id="link-502" class="py-name"><a title="Bio.Pathway.Reaction.reverse
Bio.Seq.MutableSeq.reverse" class="py-name" href="#" onclick="return doclink('link-502', 'reverse', 'link-401');">reverse</a></tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L1239"></a><tt class="py-lineno">1239</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-503" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.abs" class="py-name" href="#" onclick="return doclink('link-503', 'abs', 'link-395');">abs</a></tt><tt class="py-op">(</tt><tt id="link-504" class="py-name"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-504', 'angle', 'link-311');">angle</a></tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0.5</tt><tt class="py-op">:</tt> </tt>
<a name="L1240"></a><tt class="py-lineno">1240</tt>  <tt class="py-line">                <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">lineTo</tt><tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">outer_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">headsin</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">outer_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">headcos</tt><tt class="py-op">)</tt> </tt>
<a name="L1241"></a><tt class="py-lineno">1241</tt>  <tt class="py-line">                <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">lineTo</tt><tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">middle_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">endsin</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">middle_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">endcos</tt><tt class="py-op">)</tt> </tt>
<a name="L1242"></a><tt class="py-lineno">1242</tt>  <tt class="py-line">                <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">lineTo</tt><tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">inner_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">headsin</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">inner_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">headcos</tt><tt class="py-op">)</tt> </tt>
<a name="L1243"></a><tt class="py-lineno">1243</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1244"></a><tt class="py-lineno">1244</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-505" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_arc_line" class="py-name" href="#" onclick="return doclink('link-505', '_draw_arc_line', 'link-437');">_draw_arc_line</a></tt><tt class="py-op">(</tt><tt class="py-name">p</tt><tt class="py-op">,</tt> <tt class="py-name">outer_radius</tt><tt class="py-op">,</tt> <tt class="py-name">middle_radius</tt><tt class="py-op">,</tt> </tt>
<a name="L1245"></a><tt class="py-lineno">1245</tt>  <tt class="py-line">                                    <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">headangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-506" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-506', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1246"></a><tt class="py-lineno">1246</tt>  <tt class="py-line">                                    <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">endangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-507" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-507', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1247"></a><tt class="py-lineno">1247</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-508" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_arc_line" class="py-name" href="#" onclick="return doclink('link-508', '_draw_arc_line', 'link-437');">_draw_arc_line</a></tt><tt class="py-op">(</tt><tt class="py-name">p</tt><tt class="py-op">,</tt> <tt class="py-name">middle_radius</tt><tt class="py-op">,</tt> <tt class="py-name">inner_radius</tt><tt class="py-op">,</tt> </tt>
<a name="L1248"></a><tt class="py-lineno">1248</tt>  <tt class="py-line">                                    <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">endangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-509" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-509', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1249"></a><tt class="py-lineno">1249</tt>  <tt class="py-line">                                    <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">headangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-510" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-510', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1250"></a><tt class="py-lineno">1250</tt>  <tt class="py-line">            <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">closePath</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1251"></a><tt class="py-lineno">1251</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">p</tt> </tt>
<a name="L1252"></a><tt class="py-lineno">1252</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1253"></a><tt class="py-lineno">1253</tt>  <tt class="py-line">            <tt class="py-name">p</tt> <tt class="py-op">=</tt> <tt class="py-name">ArcPath</tt><tt class="py-op">(</tt><tt class="py-name">strokeColor</tt><tt class="py-op">=</tt><tt class="py-name">strokecolor</tt><tt class="py-op">,</tt> </tt>
<a name="L1254"></a><tt class="py-lineno">1254</tt>  <tt class="py-line">                        <tt class="py-name">fillColor</tt><tt class="py-op">=</tt><tt id="link-511" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-511', 'color', 'link-92');">color</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1255"></a><tt class="py-lineno">1255</tt>  <tt class="py-line">                        <tt class="py-comment"># default is mitre/miter which can stick out too much:</tt> </tt>
<a name="L1256"></a><tt class="py-lineno">1256</tt>  <tt class="py-line">                        <tt class="py-name">strokeLineJoin</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">,</tt>  <tt class="py-comment"># 1=round</tt> </tt>
<a name="L1257"></a><tt class="py-lineno">1257</tt>  <tt class="py-line">                        <tt class="py-name">strokewidth</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> </tt>
<a name="L1258"></a><tt class="py-lineno">1258</tt>  <tt class="py-line">                        <tt class="py-op">**</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> </tt>
<a name="L1259"></a><tt class="py-lineno">1259</tt>  <tt class="py-line">            <tt class="py-comment"># Note reportlab counts angles anti-clockwise from the horizontal</tt> </tt>
<a name="L1260"></a><tt class="py-lineno">1260</tt>  <tt class="py-line">            <tt class="py-comment"># (as in mathematics, e.g. complex numbers and polar coordinates)</tt> </tt>
<a name="L1261"></a><tt class="py-lineno">1261</tt>  <tt class="py-line">            <tt class="py-comment"># but we use clockwise from the vertical.  Also reportlab uses</tt> </tt>
<a name="L1262"></a><tt class="py-lineno">1262</tt>  <tt class="py-line">            <tt class="py-comment"># degrees, but we use radians.</tt> </tt>
<a name="L1263"></a><tt class="py-lineno">1263</tt>  <tt class="py-line">            <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">addArc</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt><tt class="py-op">,</tt> <tt class="py-name">shaft_inner_radius</tt><tt class="py-op">,</tt> </tt>
<a name="L1264"></a><tt class="py-lineno">1264</tt>  <tt class="py-line">                     <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">endangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-512" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-512', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">headangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-513" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-513', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1265"></a><tt class="py-lineno">1265</tt>  <tt class="py-line">                     <tt class="py-name">moveTo</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">,</tt> <tt id="link-514" class="py-name"><a title="Bio.Pathway.Reaction.reverse
Bio.Seq.MutableSeq.reverse" class="py-name" href="#" onclick="return doclink('link-514', 'reverse', 'link-401');">reverse</a></tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L1266"></a><tt class="py-lineno">1266</tt>  <tt class="py-line">            <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">addArc</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt><tt class="py-op">,</tt> <tt class="py-name">shaft_outer_radius</tt><tt class="py-op">,</tt> </tt>
<a name="L1267"></a><tt class="py-lineno">1267</tt>  <tt class="py-line">                     <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">endangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-515" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-515', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">headangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-516" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-516', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1268"></a><tt class="py-lineno">1268</tt>  <tt class="py-line">                     <tt id="link-517" class="py-name"><a title="Bio.Pathway.Reaction.reverse
Bio.Seq.MutableSeq.reverse" class="py-name" href="#" onclick="return doclink('link-517', 'reverse', 'link-401');">reverse</a></tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt> </tt>
<a name="L1269"></a><tt class="py-lineno">1269</tt>  <tt class="py-line">            <tt class="py-comment"># Note - two staight lines is only a good approximation for small</tt> </tt>
<a name="L1270"></a><tt class="py-lineno">1270</tt>  <tt class="py-line">            <tt class="py-comment"># head angle, in general will need to curved lines here:</tt> </tt>
<a name="L1271"></a><tt class="py-lineno">1271</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-518" class="py-name"><a title="Bio.FSSP.FSSPAlignDict.abs" class="py-name" href="#" onclick="return doclink('link-518', 'abs', 'link-395');">abs</a></tt><tt class="py-op">(</tt><tt id="link-519" class="py-name"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-519', 'angle', 'link-311');">angle</a></tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0.5</tt><tt class="py-op">:</tt> </tt>
<a name="L1272"></a><tt class="py-lineno">1272</tt>  <tt class="py-line">                <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">lineTo</tt><tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">outer_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">headsin</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">outer_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">headcos</tt><tt class="py-op">)</tt> </tt>
<a name="L1273"></a><tt class="py-lineno">1273</tt>  <tt class="py-line">                <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">lineTo</tt><tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">middle_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">startsin</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">middle_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">startcos</tt><tt class="py-op">)</tt> </tt>
<a name="L1274"></a><tt class="py-lineno">1274</tt>  <tt class="py-line">                <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">lineTo</tt><tt class="py-op">(</tt><tt class="py-name">x0</tt> <tt class="py-op">+</tt> <tt class="py-name">inner_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">headsin</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">+</tt> <tt class="py-name">inner_radius</tt> <tt class="py-op">*</tt> <tt class="py-name">headcos</tt><tt class="py-op">)</tt> </tt>
<a name="L1275"></a><tt class="py-lineno">1275</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1276"></a><tt class="py-lineno">1276</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-520" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_arc_line" class="py-name" href="#" onclick="return doclink('link-520', '_draw_arc_line', 'link-437');">_draw_arc_line</a></tt><tt class="py-op">(</tt><tt class="py-name">p</tt><tt class="py-op">,</tt> <tt class="py-name">outer_radius</tt><tt class="py-op">,</tt> <tt class="py-name">middle_radius</tt><tt class="py-op">,</tt> </tt>
<a name="L1277"></a><tt class="py-lineno">1277</tt>  <tt class="py-line">                                    <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">headangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-521" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-521', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1278"></a><tt class="py-lineno">1278</tt>  <tt class="py-line">                                    <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">startangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-522" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-522', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1279"></a><tt class="py-lineno">1279</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-523" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_arc_line" class="py-name" href="#" onclick="return doclink('link-523', '_draw_arc_line', 'link-437');">_draw_arc_line</a></tt><tt class="py-op">(</tt><tt class="py-name">p</tt><tt class="py-op">,</tt> <tt class="py-name">middle_radius</tt><tt class="py-op">,</tt> <tt class="py-name">inner_radius</tt><tt class="py-op">,</tt> </tt>
<a name="L1280"></a><tt class="py-lineno">1280</tt>  <tt class="py-line">                                    <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">startangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-524" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-524', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1281"></a><tt class="py-lineno">1281</tt>  <tt class="py-line">                                    <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">headangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-525" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-525', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1282"></a><tt class="py-lineno">1282</tt>  <tt class="py-line">            <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">closePath</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1283"></a><tt class="py-lineno">1283</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">p</tt> </tt>
</div><a name="L1284"></a><tt class="py-lineno">1284</tt>  <tt class="py-line"> </tt>
<a name="CircularDrawer._draw_sigil_jaggy"></a><div id="CircularDrawer._draw_sigil_jaggy-def"><a name="L1285"></a><tt class="py-lineno">1285</tt> <a class="py-toggle" href="#" id="CircularDrawer._draw_sigil_jaggy-toggle" onclick="return toggle('CircularDrawer._draw_sigil_jaggy');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#_draw_sigil_jaggy">_draw_sigil_jaggy</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">bottom</tt><tt class="py-op">,</tt> <tt class="py-param">center</tt><tt class="py-op">,</tt> <tt class="py-param">top</tt><tt class="py-op">,</tt> </tt>
<a name="L1286"></a><tt class="py-lineno">1286</tt>  <tt class="py-line">                          <tt class="py-param">startangle</tt><tt class="py-op">,</tt> <tt class="py-param">endangle</tt><tt class="py-op">,</tt> <tt class="py-param">strand</tt><tt class="py-op">,</tt> </tt>
<a name="L1287"></a><tt class="py-lineno">1287</tt>  <tt class="py-line">                          <tt class="py-param">color</tt><tt class="py-op">,</tt> <tt class="py-param">border</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L1288"></a><tt class="py-lineno">1288</tt>  <tt class="py-line">                          <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="CircularDrawer._draw_sigil_jaggy-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="CircularDrawer._draw_sigil_jaggy-expanded"><a name="L1289"></a><tt class="py-lineno">1289</tt>  <tt class="py-line">        <tt class="py-docstring">"""Draw JAGGY sigil (PRIVATE).</tt> </tt>
<a name="L1290"></a><tt class="py-lineno">1290</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1291"></a><tt class="py-lineno">1291</tt>  <tt class="py-line"><tt class="py-docstring">        Although we may in future expose the head/tail jaggy lengths, for now</tt> </tt>
<a name="L1292"></a><tt class="py-lineno">1292</tt>  <tt class="py-line"><tt class="py-docstring">        both the left and right edges are drawn jagged.</tt> </tt>
<a name="L1293"></a><tt class="py-lineno">1293</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1294"></a><tt class="py-lineno">1294</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-526" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-526', 'strand', 'link-91');">strand</a></tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L1295"></a><tt class="py-lineno">1295</tt>  <tt class="py-line">            <tt class="py-name">inner_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">center</tt> </tt>
<a name="L1296"></a><tt class="py-lineno">1296</tt>  <tt class="py-line">            <tt class="py-name">outer_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">top</tt> </tt>
<a name="L1297"></a><tt class="py-lineno">1297</tt>  <tt class="py-line">            <tt class="py-name">teeth</tt> <tt class="py-op">=</tt> <tt class="py-number">2</tt> </tt>
<a name="L1298"></a><tt class="py-lineno">1298</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-527" class="py-name"><a title="Bio.SearchIO._legacy.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-527', 'strand', 'link-91');">strand</a></tt> <tt class="py-op">==</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L1299"></a><tt class="py-lineno">1299</tt>  <tt class="py-line">            <tt class="py-name">inner_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">bottom</tt> </tt>
<a name="L1300"></a><tt class="py-lineno">1300</tt>  <tt class="py-line">            <tt class="py-name">outer_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">center</tt> </tt>
<a name="L1301"></a><tt class="py-lineno">1301</tt>  <tt class="py-line">            <tt class="py-name">teeth</tt> <tt class="py-op">=</tt> <tt class="py-number">2</tt> </tt>
<a name="L1302"></a><tt class="py-lineno">1302</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1303"></a><tt class="py-lineno">1303</tt>  <tt class="py-line">            <tt class="py-name">inner_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">bottom</tt> </tt>
<a name="L1304"></a><tt class="py-lineno">1304</tt>  <tt class="py-line">            <tt class="py-name">outer_radius</tt> <tt class="py-op">=</tt> <tt class="py-name">top</tt> </tt>
<a name="L1305"></a><tt class="py-lineno">1305</tt>  <tt class="py-line">            <tt class="py-name">teeth</tt> <tt class="py-op">=</tt> <tt class="py-number">4</tt> </tt>
<a name="L1306"></a><tt class="py-lineno">1306</tt>  <tt class="py-line"> </tt>
<a name="L1307"></a><tt class="py-lineno">1307</tt>  <tt class="py-line">        <tt class="py-comment"># TODO, expose these settings?</tt> </tt>
<a name="L1308"></a><tt class="py-lineno">1308</tt>  <tt class="py-line">        <tt class="py-name">tail_length_ratio</tt> <tt class="py-op">=</tt> <tt class="py-number">1.0</tt> </tt>
<a name="L1309"></a><tt class="py-lineno">1309</tt>  <tt class="py-line">        <tt class="py-name">head_length_ratio</tt> <tt class="py-op">=</tt> <tt class="py-number">1.0</tt> </tt>
<a name="L1310"></a><tt class="py-lineno">1310</tt>  <tt class="py-line"> </tt>
<a name="L1311"></a><tt class="py-lineno">1311</tt>  <tt class="py-line">        <tt class="py-name">strokecolor</tt><tt class="py-op">,</tt> <tt id="link-528" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-528', 'color', 'link-92');">color</a></tt> <tt class="py-op">=</tt> <tt id="link-529" class="py-name"><a title="Bio.Graphics.GenomeDiagram._AbstractDrawer._stroke_and_fill_colors" class="py-name" href="#" onclick="return doclink('link-529', '_stroke_and_fill_colors', 'link-9');">_stroke_and_fill_colors</a></tt><tt class="py-op">(</tt><tt id="link-530" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-530', 'color', 'link-92');">color</a></tt><tt class="py-op">,</tt> <tt class="py-name">border</tt><tt class="py-op">)</tt> </tt>
<a name="L1312"></a><tt class="py-lineno">1312</tt>  <tt class="py-line"> </tt>
<a name="L1313"></a><tt class="py-lineno">1313</tt>  <tt class="py-line">        <tt class="py-name">startangle</tt><tt class="py-op">,</tt> <tt class="py-name">endangle</tt> <tt class="py-op">=</tt> <tt id="link-531" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-531', 'min', 'link-21');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">startangle</tt><tt class="py-op">,</tt> <tt class="py-name">endangle</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-532" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-532', 'max', 'link-23');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">startangle</tt><tt class="py-op">,</tt> <tt class="py-name">endangle</tt><tt class="py-op">)</tt> </tt>
<a name="L1314"></a><tt class="py-lineno">1314</tt>  <tt class="py-line">        <tt id="link-533" class="py-name"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-533', 'angle', 'link-311');">angle</a></tt> <tt class="py-op">=</tt> <tt class="py-name">float</tt><tt class="py-op">(</tt><tt class="py-name">endangle</tt> <tt class="py-op">-</tt> <tt class="py-name">startangle</tt><tt class="py-op">)</tt>    <tt class="py-comment"># angle subtended by arc</tt> </tt>
<a name="L1315"></a><tt class="py-lineno">1315</tt>  <tt class="py-line">        <tt id="link-534" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.height" class="py-name" href="#" onclick="return doclink('link-534', 'height', 'link-25');">height</a></tt> <tt class="py-op">=</tt> <tt class="py-name">outer_radius</tt> <tt class="py-op">-</tt> <tt class="py-name">inner_radius</tt> </tt>
<a name="L1316"></a><tt class="py-lineno">1316</tt>  <tt class="py-line"> </tt>
<a name="L1317"></a><tt class="py-lineno">1317</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">startangle</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">endangle</tt> <tt class="py-keyword">and</tt> <tt id="link-535" class="py-name"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-535', 'angle', 'link-311');">angle</a></tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">0</tt> </tt>
<a name="L1318"></a><tt class="py-lineno">1318</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">head_length_ratio</tt> <tt class="py-keyword">and</tt> <tt class="py-name">tail_length_ratio</tt><tt class="py-op">:</tt> </tt>
<a name="L1319"></a><tt class="py-lineno">1319</tt>  <tt class="py-line">            <tt class="py-name">headangle</tt> <tt class="py-op">=</tt> <tt id="link-536" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-536', 'max', 'link-23');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">endangle</tt> <tt class="py-op">-</tt> <tt id="link-537" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-537', 'min', 'link-21');">min</a></tt><tt class="py-op">(</tt><tt id="link-538" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.height" class="py-name" href="#" onclick="return doclink('link-538', 'height', 'link-25');">height</a></tt> <tt class="py-op">*</tt> <tt class="py-name">head_length_ratio</tt> <tt class="py-op">/</tt> <tt class="py-op">(</tt><tt class="py-name">center</tt> <tt class="py-op">*</tt> <tt class="py-name">teeth</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-539" class="py-name"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-539', 'angle', 'link-311');">angle</a></tt> <tt class="py-op">*</tt> <tt class="py-number">0.5</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">startangle</tt><tt class="py-op">)</tt> </tt>
<a name="L1320"></a><tt class="py-lineno">1320</tt>  <tt class="py-line">            <tt class="py-name">tailangle</tt> <tt class="py-op">=</tt> <tt id="link-540" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-540', 'min', 'link-21');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">startangle</tt> <tt class="py-op">+</tt> <tt id="link-541" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-541', 'min', 'link-21');">min</a></tt><tt class="py-op">(</tt><tt id="link-542" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.height" class="py-name" href="#" onclick="return doclink('link-542', 'height', 'link-25');">height</a></tt> <tt class="py-op">*</tt> <tt class="py-name">tail_length_ratio</tt> <tt class="py-op">/</tt> <tt class="py-op">(</tt><tt class="py-name">center</tt> <tt class="py-op">*</tt> <tt class="py-name">teeth</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-543" class="py-name"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-543', 'angle', 'link-311');">angle</a></tt> <tt class="py-op">*</tt> <tt class="py-number">0.5</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">endangle</tt><tt class="py-op">)</tt> </tt>
<a name="L1321"></a><tt class="py-lineno">1321</tt>  <tt class="py-line">            <tt class="py-comment"># With very small features, can due to floating point calculations</tt> </tt>
<a name="L1322"></a><tt class="py-lineno">1322</tt>  <tt class="py-line">            <tt class="py-comment"># violate the assertion below that start &lt;= tail &lt;= head &lt;= end</tt> </tt>
<a name="L1323"></a><tt class="py-lineno">1323</tt>  <tt class="py-line">            <tt class="py-name">tailangle</tt> <tt class="py-op">=</tt> <tt id="link-544" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-544', 'min', 'link-21');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">tailangle</tt><tt class="py-op">,</tt> <tt class="py-name">headangle</tt><tt class="py-op">)</tt> </tt>
<a name="L1324"></a><tt class="py-lineno">1324</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">head_length_ratio</tt><tt class="py-op">:</tt> </tt>
<a name="L1325"></a><tt class="py-lineno">1325</tt>  <tt class="py-line">            <tt class="py-name">headangle</tt> <tt class="py-op">=</tt> <tt id="link-545" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-545', 'max', 'link-23');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">endangle</tt> <tt class="py-op">-</tt> <tt id="link-546" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-546', 'min', 'link-21');">min</a></tt><tt class="py-op">(</tt><tt id="link-547" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.height" class="py-name" href="#" onclick="return doclink('link-547', 'height', 'link-25');">height</a></tt> <tt class="py-op">*</tt> <tt class="py-name">head_length_ratio</tt> <tt class="py-op">/</tt> <tt class="py-op">(</tt><tt class="py-name">center</tt> <tt class="py-op">*</tt> <tt class="py-name">teeth</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-548" class="py-name"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-548', 'angle', 'link-311');">angle</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">startangle</tt><tt class="py-op">)</tt> </tt>
<a name="L1326"></a><tt class="py-lineno">1326</tt>  <tt class="py-line">            <tt class="py-name">tailangle</tt> <tt class="py-op">=</tt> <tt class="py-name">startangle</tt> </tt>
<a name="L1327"></a><tt class="py-lineno">1327</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1328"></a><tt class="py-lineno">1328</tt>  <tt class="py-line">            <tt class="py-name">headangle</tt> <tt class="py-op">=</tt> <tt class="py-name">endangle</tt> </tt>
<a name="L1329"></a><tt class="py-lineno">1329</tt>  <tt class="py-line">            <tt class="py-name">tailangle</tt> <tt class="py-op">=</tt> <tt id="link-549" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-549', 'min', 'link-21');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">startangle</tt> <tt class="py-op">+</tt> <tt id="link-550" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-550', 'min', 'link-21');">min</a></tt><tt class="py-op">(</tt><tt id="link-551" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.height" class="py-name" href="#" onclick="return doclink('link-551', 'height', 'link-25');">height</a></tt> <tt class="py-op">*</tt> <tt class="py-name">tail_length_ratio</tt> <tt class="py-op">/</tt> <tt class="py-op">(</tt><tt class="py-name">center</tt> <tt class="py-op">*</tt> <tt class="py-name">teeth</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-552" class="py-name"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-552', 'angle', 'link-311');">angle</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">endangle</tt><tt class="py-op">)</tt> </tt>
<a name="L1330"></a><tt class="py-lineno">1330</tt>  <tt class="py-line"> </tt>
<a name="L1331"></a><tt class="py-lineno">1331</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">startangle</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">tailangle</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">headangle</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">endangle</tt><tt class="py-op">:</tt> </tt>
<a name="L1332"></a><tt class="py-lineno">1332</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">RuntimeError</tt><tt class="py-op">(</tt><tt class="py-string">"Problem drawing jaggy sigil, invalid "</tt> </tt>
<a name="L1333"></a><tt class="py-lineno">1333</tt>  <tt class="py-line">                               <tt class="py-string">"positions. Start angle: %s, "</tt> </tt>
<a name="L1334"></a><tt class="py-lineno">1334</tt>  <tt class="py-line">                               <tt class="py-string">"Tail angle: %s, Head angle: %s, End angle %s, "</tt> </tt>
<a name="L1335"></a><tt class="py-lineno">1335</tt>  <tt class="py-line">                               <tt class="py-string">"Angle: %s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">startangle</tt><tt class="py-op">,</tt> <tt class="py-name">tailangle</tt><tt class="py-op">,</tt> <tt class="py-name">headangle</tt><tt class="py-op">,</tt> </tt>
<a name="L1336"></a><tt class="py-lineno">1336</tt>  <tt class="py-line">                                              <tt class="py-name">endangle</tt><tt class="py-op">,</tt> <tt id="link-553" class="py-name"><a title="Bio.PDB.vectors.Vector.angle" class="py-name" href="#" onclick="return doclink('link-553', 'angle', 'link-311');">angle</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1337"></a><tt class="py-lineno">1337</tt>  <tt class="py-line"> </tt>
<a name="L1338"></a><tt class="py-lineno">1338</tt>  <tt class="py-line">        <tt class="py-comment"># Calculate trig values for angle and coordinates</tt> </tt>
<a name="L1339"></a><tt class="py-lineno">1339</tt>  <tt class="py-line">        <tt class="py-name">startcos</tt><tt class="py-op">,</tt> <tt class="py-name">startsin</tt> <tt class="py-op">=</tt> <tt class="py-name">cos</tt><tt class="py-op">(</tt><tt class="py-name">startangle</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">sin</tt><tt class="py-op">(</tt><tt class="py-name">startangle</tt><tt class="py-op">)</tt> </tt>
<a name="L1340"></a><tt class="py-lineno">1340</tt>  <tt class="py-line">        <tt class="py-name">headcos</tt><tt class="py-op">,</tt> <tt class="py-name">headsin</tt> <tt class="py-op">=</tt> <tt class="py-name">cos</tt><tt class="py-op">(</tt><tt class="py-name">headangle</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">sin</tt><tt class="py-op">(</tt><tt class="py-name">headangle</tt><tt class="py-op">)</tt> </tt>
<a name="L1341"></a><tt class="py-lineno">1341</tt>  <tt class="py-line">        <tt class="py-name">endcos</tt><tt class="py-op">,</tt> <tt class="py-name">endsin</tt> <tt class="py-op">=</tt> <tt class="py-name">cos</tt><tt class="py-op">(</tt><tt class="py-name">endangle</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">sin</tt><tt class="py-op">(</tt><tt class="py-name">endangle</tt><tt class="py-op">)</tt> </tt>
<a name="L1342"></a><tt class="py-lineno">1342</tt>  <tt class="py-line">        <tt class="py-name">x0</tt><tt class="py-op">,</tt> <tt class="py-name">y0</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt>      <tt class="py-comment"># origin of the circle</tt> </tt>
<a name="L1343"></a><tt class="py-lineno">1343</tt>  <tt class="py-line"> </tt>
<a name="L1344"></a><tt class="py-lineno">1344</tt>  <tt class="py-line">        <tt class="py-name">p</tt> <tt class="py-op">=</tt> <tt class="py-name">ArcPath</tt><tt class="py-op">(</tt><tt class="py-name">strokeColor</tt><tt class="py-op">=</tt><tt class="py-name">strokecolor</tt><tt class="py-op">,</tt> </tt>
<a name="L1345"></a><tt class="py-lineno">1345</tt>  <tt class="py-line">                    <tt class="py-name">fillColor</tt><tt class="py-op">=</tt><tt id="link-554" class="py-name"><a title="Bio.Phylo.BaseTree.Clade.color
Bio.Phylo.PhyloXMLIO.Parser.color
Bio.Phylo.PhyloXMLIO.Writer.color" class="py-name" href="#" onclick="return doclink('link-554', 'color', 'link-92');">color</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1346"></a><tt class="py-lineno">1346</tt>  <tt class="py-line">                    <tt class="py-comment"># default is mitre/miter which can stick out too much:</tt> </tt>
<a name="L1347"></a><tt class="py-lineno">1347</tt>  <tt class="py-line">                    <tt class="py-name">strokeLineJoin</tt><tt class="py-op">=</tt><tt class="py-number">1</tt><tt class="py-op">,</tt>  <tt class="py-comment"># 1=round</tt> </tt>
<a name="L1348"></a><tt class="py-lineno">1348</tt>  <tt class="py-line">                    <tt class="py-name">strokewidth</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> </tt>
<a name="L1349"></a><tt class="py-lineno">1349</tt>  <tt class="py-line">                    <tt class="py-op">**</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> </tt>
<a name="L1350"></a><tt class="py-lineno">1350</tt>  <tt class="py-line">        <tt class="py-comment"># Note reportlab counts angles anti-clockwise from the horizontal</tt> </tt>
<a name="L1351"></a><tt class="py-lineno">1351</tt>  <tt class="py-line">        <tt class="py-comment"># (as in mathematics, e.g. complex numbers and polar coordinates)</tt> </tt>
<a name="L1352"></a><tt class="py-lineno">1352</tt>  <tt class="py-line">        <tt class="py-comment"># but we use clockwise from the vertical.  Also reportlab uses</tt> </tt>
<a name="L1353"></a><tt class="py-lineno">1353</tt>  <tt class="py-line">        <tt class="py-comment"># degrees, but we use radians.</tt> </tt>
<a name="L1354"></a><tt class="py-lineno">1354</tt>  <tt class="py-line">        <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">addArc</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt><tt class="py-op">,</tt> <tt class="py-name">inner_radius</tt><tt class="py-op">,</tt> </tt>
<a name="L1355"></a><tt class="py-lineno">1355</tt>  <tt class="py-line">                 <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">headangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-555" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-555', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">tailangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-556" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-556', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1356"></a><tt class="py-lineno">1356</tt>  <tt class="py-line">                 <tt class="py-name">moveTo</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L1357"></a><tt class="py-lineno">1357</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-557" class="py-name" targets="Variable Bio.PDB.Polypeptide.i=Bio.PDB.Polypeptide-module.html#i,Variable Bio.SeqIO.SffIO.i=Bio.SeqIO.SffIO-module.html#i"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-557', 'i', 'link-557');">i</a></tt> <tt class="py-keyword">in</tt> <tt id="link-558" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-558', 'range', 'link-3');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">teeth</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1358"></a><tt class="py-lineno">1358</tt>  <tt class="py-line">            <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">addArc</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt><tt class="py-op">,</tt> <tt class="py-name">inner_radius</tt> <tt class="py-op">+</tt> <tt id="link-559" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-559', 'i', 'link-557');">i</a></tt> <tt class="py-op">*</tt> <tt id="link-560" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.height" class="py-name" href="#" onclick="return doclink('link-560', 'height', 'link-25');">height</a></tt> <tt class="py-op">/</tt> <tt class="py-name">teeth</tt><tt class="py-op">,</tt> </tt>
<a name="L1359"></a><tt class="py-lineno">1359</tt>  <tt class="py-line">                     <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">tailangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-561" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-561', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">startangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-562" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-562', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1360"></a><tt class="py-lineno">1360</tt>  <tt class="py-line">            <tt class="py-comment"># Curved line needed when drawing long jaggies</tt> </tt>
<a name="L1361"></a><tt class="py-lineno">1361</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-563" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_arc_line" class="py-name" href="#" onclick="return doclink('link-563', '_draw_arc_line', 'link-437');">_draw_arc_line</a></tt><tt class="py-op">(</tt><tt class="py-name">p</tt><tt class="py-op">,</tt> </tt>
<a name="L1362"></a><tt class="py-lineno">1362</tt>  <tt class="py-line">                                <tt class="py-name">inner_radius</tt> <tt class="py-op">+</tt> <tt id="link-564" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-564', 'i', 'link-557');">i</a></tt> <tt class="py-op">*</tt> <tt id="link-565" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.height" class="py-name" href="#" onclick="return doclink('link-565', 'height', 'link-25');">height</a></tt> <tt class="py-op">/</tt> <tt class="py-name">teeth</tt><tt class="py-op">,</tt> </tt>
<a name="L1363"></a><tt class="py-lineno">1363</tt>  <tt class="py-line">                                <tt class="py-name">inner_radius</tt> <tt class="py-op">+</tt> <tt class="py-op">(</tt><tt id="link-566" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-566', 'i', 'link-557');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt id="link-567" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.height" class="py-name" href="#" onclick="return doclink('link-567', 'height', 'link-25');">height</a></tt> <tt class="py-op">/</tt> <tt class="py-name">teeth</tt><tt class="py-op">,</tt> </tt>
<a name="L1364"></a><tt class="py-lineno">1364</tt>  <tt class="py-line">                                <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">startangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-568" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-568', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1365"></a><tt class="py-lineno">1365</tt>  <tt class="py-line">                                <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">tailangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-569" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-569', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1366"></a><tt class="py-lineno">1366</tt>  <tt class="py-line">        <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">addArc</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt><tt class="py-op">,</tt> <tt class="py-name">outer_radius</tt><tt class="py-op">,</tt> </tt>
<a name="L1367"></a><tt class="py-lineno">1367</tt>  <tt class="py-line">                 <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">headangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-570" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-570', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">tailangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-571" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-571', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1368"></a><tt class="py-lineno">1368</tt>  <tt class="py-line">                 <tt id="link-572" class="py-name"><a title="Bio.Pathway.Reaction.reverse
Bio.Seq.MutableSeq.reverse" class="py-name" href="#" onclick="return doclink('link-572', 'reverse', 'link-401');">reverse</a></tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L1369"></a><tt class="py-lineno">1369</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-573" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-573', 'i', 'link-557');">i</a></tt> <tt class="py-keyword">in</tt> <tt id="link-574" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-574', 'range', 'link-3');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">teeth</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1370"></a><tt class="py-lineno">1370</tt>  <tt class="py-line">            <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">addArc</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">xcenter</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">ycenter</tt><tt class="py-op">,</tt> </tt>
<a name="L1371"></a><tt class="py-lineno">1371</tt>  <tt class="py-line">                     <tt class="py-name">outer_radius</tt> <tt class="py-op">-</tt> <tt id="link-575" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-575', 'i', 'link-557');">i</a></tt> <tt class="py-op">*</tt> <tt id="link-576" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.height" class="py-name" href="#" onclick="return doclink('link-576', 'height', 'link-25');">height</a></tt> <tt class="py-op">/</tt> <tt class="py-name">teeth</tt><tt class="py-op">,</tt> </tt>
<a name="L1372"></a><tt class="py-lineno">1372</tt>  <tt class="py-line">                     <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">endangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-577" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-577', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">headangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-578" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-578', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1373"></a><tt class="py-lineno">1373</tt>  <tt class="py-line">                     <tt id="link-579" class="py-name"><a title="Bio.Pathway.Reaction.reverse
Bio.Seq.MutableSeq.reverse" class="py-name" href="#" onclick="return doclink('link-579', 'reverse', 'link-401');">reverse</a></tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt> </tt>
<a name="L1374"></a><tt class="py-lineno">1374</tt>  <tt class="py-line">            <tt class="py-comment"># Curved line needed when drawing long jaggies</tt> </tt>
<a name="L1375"></a><tt class="py-lineno">1375</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-580" class="py-name"><a title="Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer._draw_arc_line" class="py-name" href="#" onclick="return doclink('link-580', '_draw_arc_line', 'link-437');">_draw_arc_line</a></tt><tt class="py-op">(</tt><tt class="py-name">p</tt><tt class="py-op">,</tt> </tt>
<a name="L1376"></a><tt class="py-lineno">1376</tt>  <tt class="py-line">                                <tt class="py-name">outer_radius</tt> <tt class="py-op">-</tt> <tt id="link-581" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-581', 'i', 'link-557');">i</a></tt> <tt class="py-op">*</tt> <tt id="link-582" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.height" class="py-name" href="#" onclick="return doclink('link-582', 'height', 'link-25');">height</a></tt> <tt class="py-op">/</tt> <tt class="py-name">teeth</tt><tt class="py-op">,</tt> </tt>
<a name="L1377"></a><tt class="py-lineno">1377</tt>  <tt class="py-line">                                <tt class="py-name">outer_radius</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt id="link-583" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-583', 'i', 'link-557');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> <tt class="py-op">*</tt> <tt id="link-584" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.height" class="py-name" href="#" onclick="return doclink('link-584', 'height', 'link-25');">height</a></tt> <tt class="py-op">/</tt> <tt class="py-name">teeth</tt><tt class="py-op">,</tt> </tt>
<a name="L1378"></a><tt class="py-lineno">1378</tt>  <tt class="py-line">                                <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">endangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-585" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-585', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1379"></a><tt class="py-lineno">1379</tt>  <tt class="py-line">                                <tt class="py-number">90</tt> <tt class="py-op">-</tt> <tt class="py-op">(</tt><tt class="py-name">headangle</tt> <tt class="py-op">*</tt> <tt class="py-number">180</tt> <tt class="py-op">/</tt> <tt id="link-586" class="py-name"><a title="Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.pi" class="py-name" href="#" onclick="return doclink('link-586', 'pi', 'link-14');">pi</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1380"></a><tt class="py-lineno">1380</tt>  <tt class="py-line">        <tt class="py-name">p</tt><tt class="py-op">.</tt><tt class="py-name">closePath</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1381"></a><tt class="py-lineno">1381</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">p</tt> </tt>
</div></div><a name="L1382"></a><tt class="py-lineno">1382</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Tue Jul 16 17:06:07 2019
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
