<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Bio.SeqIO.QualityIO</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="Bio-module.html">Package&nbsp;Bio</a> ::
        <a href="Bio.SeqIO-module.html">Package&nbsp;SeqIO</a> ::
        Module&nbsp;QualityIO
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="Bio.SeqIO.QualityIO-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="Bio.SeqIO.QualityIO-module.html">Module Bio.SeqIO.QualityIO</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">   1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2009-2010 by Peter Cock.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">   2</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L3"></a><tt class="py-lineno">   3</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L4"></a><tt class="py-lineno">   4</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L5"></a><tt class="py-lineno">   5</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L6"></a><tt class="py-lineno">   6</tt>  <tt class="py-line"><tt class="py-comment"># This module is for reading and writing FASTQ and QUAL format files as</tt> </tt>
<a name="L7"></a><tt class="py-lineno">   7</tt>  <tt class="py-line"><tt class="py-comment"># SeqRecord objects, and is expected to be used via the Bio.SeqIO API.</tt> </tt>
<a name="L8"></a><tt class="py-lineno">   8</tt>  <tt class="py-line"> </tt>
<a name="L9"></a><tt class="py-lineno">   9</tt>  <tt class="py-line"><tt class="py-docstring">"""Bio.SeqIO support for the FASTQ and QUAL file formats.</tt> </tt>
<a name="L10"></a><tt class="py-lineno">  10</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L11"></a><tt class="py-lineno">  11</tt>  <tt class="py-line"><tt class="py-docstring">Note that you are expected to use this code via the Bio.SeqIO interface, as</tt> </tt>
<a name="L12"></a><tt class="py-lineno">  12</tt>  <tt class="py-line"><tt class="py-docstring">shown below.</tt> </tt>
<a name="L13"></a><tt class="py-lineno">  13</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L14"></a><tt class="py-lineno">  14</tt>  <tt class="py-line"><tt class="py-docstring">The FASTQ file format is used frequently at the Wellcome Trust Sanger Institute</tt> </tt>
<a name="L15"></a><tt class="py-lineno">  15</tt>  <tt class="py-line"><tt class="py-docstring">to bundle a FASTA sequence and its PHRED quality data (integers between 0 and</tt> </tt>
<a name="L16"></a><tt class="py-lineno">  16</tt>  <tt class="py-line"><tt class="py-docstring">90).  Rather than using a single FASTQ file, often paired FASTA and QUAL files</tt> </tt>
<a name="L17"></a><tt class="py-lineno">  17</tt>  <tt class="py-line"><tt class="py-docstring">are used containing the sequence and the quality information separately.</tt> </tt>
<a name="L18"></a><tt class="py-lineno">  18</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L19"></a><tt class="py-lineno">  19</tt>  <tt class="py-line"><tt class="py-docstring">The PHRED software reads DNA sequencing trace files, calls bases, and</tt> </tt>
<a name="L20"></a><tt class="py-lineno">  20</tt>  <tt class="py-line"><tt class="py-docstring">assigns a non-negative quality value to each called base using a logged</tt> </tt>
<a name="L21"></a><tt class="py-lineno">  21</tt>  <tt class="py-line"><tt class="py-docstring">transformation of the error probability, Q = -10 log10( Pe ), for example::</tt> </tt>
<a name="L22"></a><tt class="py-lineno">  22</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L23"></a><tt class="py-lineno">  23</tt>  <tt class="py-line"><tt class="py-docstring">    Pe = 1.0,         Q =  0</tt> </tt>
<a name="L24"></a><tt class="py-lineno">  24</tt>  <tt class="py-line"><tt class="py-docstring">    Pe = 0.1,         Q = 10</tt> </tt>
<a name="L25"></a><tt class="py-lineno">  25</tt>  <tt class="py-line"><tt class="py-docstring">    Pe = 0.01,        Q = 20</tt> </tt>
<a name="L26"></a><tt class="py-lineno">  26</tt>  <tt class="py-line"><tt class="py-docstring">    ...</tt> </tt>
<a name="L27"></a><tt class="py-lineno">  27</tt>  <tt class="py-line"><tt class="py-docstring">    Pe = 0.00000001,  Q = 80</tt> </tt>
<a name="L28"></a><tt class="py-lineno">  28</tt>  <tt class="py-line"><tt class="py-docstring">    Pe = 0.000000001, Q = 90</tt> </tt>
<a name="L29"></a><tt class="py-lineno">  29</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L30"></a><tt class="py-lineno">  30</tt>  <tt class="py-line"><tt class="py-docstring">In typical raw sequence reads, the PHRED quality valuea will be from 0 to 40.</tt> </tt>
<a name="L31"></a><tt class="py-lineno">  31</tt>  <tt class="py-line"><tt class="py-docstring">In the QUAL format these quality values are held as space separated text in</tt> </tt>
<a name="L32"></a><tt class="py-lineno">  32</tt>  <tt class="py-line"><tt class="py-docstring">a FASTA like file format.  In the FASTQ format, each quality values is encoded</tt> </tt>
<a name="L33"></a><tt class="py-lineno">  33</tt>  <tt class="py-line"><tt class="py-docstring">with a single ASCI character using chr(Q+33), meaning zero maps to the</tt> </tt>
<a name="L34"></a><tt class="py-lineno">  34</tt>  <tt class="py-line"><tt class="py-docstring">character "!" and for example 80 maps to "q".  For the Sanger FASTQ standard</tt> </tt>
<a name="L35"></a><tt class="py-lineno">  35</tt>  <tt class="py-line"><tt class="py-docstring">the allowed range of PHRED scores is 0 to 93 inclusive. The sequences and</tt> </tt>
<a name="L36"></a><tt class="py-lineno">  36</tt>  <tt class="py-line"><tt class="py-docstring">quality are then stored in pairs in a FASTA like format.</tt> </tt>
<a name="L37"></a><tt class="py-lineno">  37</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L38"></a><tt class="py-lineno">  38</tt>  <tt class="py-line"><tt class="py-docstring">Unfortunately there is no official document describing the FASTQ file format,</tt> </tt>
<a name="L39"></a><tt class="py-lineno">  39</tt>  <tt class="py-line"><tt class="py-docstring">and worse, several related but different variants exist. For more details,</tt> </tt>
<a name="L40"></a><tt class="py-lineno">  40</tt>  <tt class="py-line"><tt class="py-docstring">please read this open access publication::</tt> </tt>
<a name="L41"></a><tt class="py-lineno">  41</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L42"></a><tt class="py-lineno">  42</tt>  <tt class="py-line"><tt class="py-docstring">    The Sanger FASTQ file format for sequences with quality scores, and the</tt> </tt>
<a name="L43"></a><tt class="py-lineno">  43</tt>  <tt class="py-line"><tt class="py-docstring">    Solexa/Illumina FASTQ variants.</tt> </tt>
<a name="L44"></a><tt class="py-lineno">  44</tt>  <tt class="py-line"><tt class="py-docstring">    P.J.A.Cock (Biopython), C.J.Fields (BioPerl), N.Goto (BioRuby),</tt> </tt>
<a name="L45"></a><tt class="py-lineno">  45</tt>  <tt class="py-line"><tt class="py-docstring">    M.L.Heuer (BioJava) and P.M. Rice (EMBOSS).</tt> </tt>
<a name="L46"></a><tt class="py-lineno">  46</tt>  <tt class="py-line"><tt class="py-docstring">    Nucleic Acids Research 2010 38(6):1767-1771</tt> </tt>
<a name="L47"></a><tt class="py-lineno">  47</tt>  <tt class="py-line"><tt class="py-docstring">    http://dx.doi.org/10.1093/nar/gkp1137</tt> </tt>
<a name="L48"></a><tt class="py-lineno">  48</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L49"></a><tt class="py-lineno">  49</tt>  <tt class="py-line"><tt class="py-docstring">The good news is that Roche 454 sequencers can output files in the QUAL format,</tt> </tt>
<a name="L50"></a><tt class="py-lineno">  50</tt>  <tt class="py-line"><tt class="py-docstring">and sensibly they use PHREP style scores like Sanger.  Converting a pair of</tt> </tt>
<a name="L51"></a><tt class="py-lineno">  51</tt>  <tt class="py-line"><tt class="py-docstring">FASTA and QUAL files into a Sanger style FASTQ file is easy. To extract QUAL</tt> </tt>
<a name="L52"></a><tt class="py-lineno">  52</tt>  <tt class="py-line"><tt class="py-docstring">files from a Roche 454 SFF binary file, use the Roche off instrument command</tt> </tt>
<a name="L53"></a><tt class="py-lineno">  53</tt>  <tt class="py-line"><tt class="py-docstring">line tool "sffinfo" with the -q or -qual argument.  You can extract a matching</tt> </tt>
<a name="L54"></a><tt class="py-lineno">  54</tt>  <tt class="py-line"><tt class="py-docstring">FASTA file using the -s or -seq argument instead.</tt> </tt>
<a name="L55"></a><tt class="py-lineno">  55</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L56"></a><tt class="py-lineno">  56</tt>  <tt class="py-line"><tt class="py-docstring">The bad news is that Solexa/Illumina did things differently - they have their</tt> </tt>
<a name="L57"></a><tt class="py-lineno">  57</tt>  <tt class="py-line"><tt class="py-docstring">own scoring system AND their own incompatible versions of the FASTQ format.</tt> </tt>
<a name="L58"></a><tt class="py-lineno">  58</tt>  <tt class="py-line"><tt class="py-docstring">Solexa/Illumina quality scores use Q = - 10 log10 ( Pe / (1-Pe) ), which can</tt> </tt>
<a name="L59"></a><tt class="py-lineno">  59</tt>  <tt class="py-line"><tt class="py-docstring">be negative.  PHRED scores and Solexa scores are NOT interchangeable (but a</tt> </tt>
<a name="L60"></a><tt class="py-lineno">  60</tt>  <tt class="py-line"><tt class="py-docstring">reasonable mapping can be achieved between them, and they are approximately</tt> </tt>
<a name="L61"></a><tt class="py-lineno">  61</tt>  <tt class="py-line"><tt class="py-docstring">equal for higher quality reads).</tt> </tt>
<a name="L62"></a><tt class="py-lineno">  62</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L63"></a><tt class="py-lineno">  63</tt>  <tt class="py-line"><tt class="py-docstring">Confusingly early Solexa pipelines produced a FASTQ like file but using their</tt> </tt>
<a name="L64"></a><tt class="py-lineno">  64</tt>  <tt class="py-line"><tt class="py-docstring">own score mapping and an ASCII offset of 64. To make things worse, for the</tt> </tt>
<a name="L65"></a><tt class="py-lineno">  65</tt>  <tt class="py-line"><tt class="py-docstring">Solexa/Illumina pipeline 1.3 onwards, they introduced a third variant of the</tt> </tt>
<a name="L66"></a><tt class="py-lineno">  66</tt>  <tt class="py-line"><tt class="py-docstring">FASTQ file format, this time using PHRED scores (which is more consistent) but</tt> </tt>
<a name="L67"></a><tt class="py-lineno">  67</tt>  <tt class="py-line"><tt class="py-docstring">with an ASCII offset of 64.</tt> </tt>
<a name="L68"></a><tt class="py-lineno">  68</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L69"></a><tt class="py-lineno">  69</tt>  <tt class="py-line"><tt class="py-docstring">i.e. There are at least THREE different and INCOMPATIBLE variants of the FASTQ</tt> </tt>
<a name="L70"></a><tt class="py-lineno">  70</tt>  <tt class="py-line"><tt class="py-docstring">file format: The original Sanger PHRED standard, and two from Solexa/Illumina.</tt> </tt>
<a name="L71"></a><tt class="py-lineno">  71</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L72"></a><tt class="py-lineno">  72</tt>  <tt class="py-line"><tt class="py-docstring">The good news is that as of CASAVA version 1.8, Illumina sequencers will</tt> </tt>
<a name="L73"></a><tt class="py-lineno">  73</tt>  <tt class="py-line"><tt class="py-docstring">produce FASTQ files using the standard Sanger encoding.</tt> </tt>
<a name="L74"></a><tt class="py-lineno">  74</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L75"></a><tt class="py-lineno">  75</tt>  <tt class="py-line"><tt class="py-docstring">You are expected to use this module via the Bio.SeqIO functions, with the</tt> </tt>
<a name="L76"></a><tt class="py-lineno">  76</tt>  <tt class="py-line"><tt class="py-docstring">following format names:</tt> </tt>
<a name="L77"></a><tt class="py-lineno">  77</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L78"></a><tt class="py-lineno">  78</tt>  <tt class="py-line"><tt class="py-docstring">    - "qual" means simple quality files using PHRED scores (e.g. from Roche 454)</tt> </tt>
<a name="L79"></a><tt class="py-lineno">  79</tt>  <tt class="py-line"><tt class="py-docstring">    - "fastq" means Sanger style FASTQ files using PHRED scores and an ASCII</tt> </tt>
<a name="L80"></a><tt class="py-lineno">  80</tt>  <tt class="py-line"><tt class="py-docstring">      offset of 33 (e.g. from the NCBI Short Read Archive and Illumina 1.8+).</tt> </tt>
<a name="L81"></a><tt class="py-lineno">  81</tt>  <tt class="py-line"><tt class="py-docstring">      These can potentially hold PHRED scores from 0 to 93.</tt> </tt>
<a name="L82"></a><tt class="py-lineno">  82</tt>  <tt class="py-line"><tt class="py-docstring">    - "fastq-sanger" is an alias for "fastq".</tt> </tt>
<a name="L83"></a><tt class="py-lineno">  83</tt>  <tt class="py-line"><tt class="py-docstring">    - "fastq-solexa" means old Solexa (and also very early Illumina) style FASTQ</tt> </tt>
<a name="L84"></a><tt class="py-lineno">  84</tt>  <tt class="py-line"><tt class="py-docstring">      files, using Solexa scores with an ASCII offset 64. These can hold Solexa</tt> </tt>
<a name="L85"></a><tt class="py-lineno">  85</tt>  <tt class="py-line"><tt class="py-docstring">      scores from -5 to 62.</tt> </tt>
<a name="L86"></a><tt class="py-lineno">  86</tt>  <tt class="py-line"><tt class="py-docstring">    - "fastq-illumina" means newer Illumina 1.3 to 1.7 style FASTQ files, using</tt> </tt>
<a name="L87"></a><tt class="py-lineno">  87</tt>  <tt class="py-line"><tt class="py-docstring">      PHRED scores but with an ASCII offset 64, allowing PHRED scores from 0</tt> </tt>
<a name="L88"></a><tt class="py-lineno">  88</tt>  <tt class="py-line"><tt class="py-docstring">      to 62.</tt> </tt>
<a name="L89"></a><tt class="py-lineno">  89</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L90"></a><tt class="py-lineno">  90</tt>  <tt class="py-line"><tt class="py-docstring">We could potentially add support for "qual-solexa" meaning QUAL files which</tt> </tt>
<a name="L91"></a><tt class="py-lineno">  91</tt>  <tt class="py-line"><tt class="py-docstring">contain Solexa scores, but thus far there isn't any reason to use such files.</tt> </tt>
<a name="L92"></a><tt class="py-lineno">  92</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L93"></a><tt class="py-lineno">  93</tt>  <tt class="py-line"><tt class="py-docstring">For example, consider the following short FASTQ file::</tt> </tt>
<a name="L94"></a><tt class="py-lineno">  94</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L95"></a><tt class="py-lineno">  95</tt>  <tt class="py-line"><tt class="py-docstring">    @EAS54_6_R1_2_1_413_324</tt> </tt>
<a name="L96"></a><tt class="py-lineno">  96</tt>  <tt class="py-line"><tt class="py-docstring">    CCCTTCTTGTCTTCAGCGTTTCTCC</tt> </tt>
<a name="L97"></a><tt class="py-lineno">  97</tt>  <tt class="py-line"><tt class="py-docstring">    +</tt> </tt>
<a name="L98"></a><tt class="py-lineno">  98</tt>  <tt class="py-line"><tt class="py-docstring">    ;;3;;;;;;;;;;;;7;;;;;;;88</tt> </tt>
<a name="L99"></a><tt class="py-lineno">  99</tt>  <tt class="py-line"><tt class="py-docstring">    @EAS54_6_R1_2_1_540_792</tt> </tt>
<a name="L100"></a><tt class="py-lineno"> 100</tt>  <tt class="py-line"><tt class="py-docstring">    TTGGCAGGCCAAGGCCGATGGATCA</tt> </tt>
<a name="L101"></a><tt class="py-lineno"> 101</tt>  <tt class="py-line"><tt class="py-docstring">    +</tt> </tt>
<a name="L102"></a><tt class="py-lineno"> 102</tt>  <tt class="py-line"><tt class="py-docstring">    ;;;;;;;;;;;7;;;;;-;;;3;83</tt> </tt>
<a name="L103"></a><tt class="py-lineno"> 103</tt>  <tt class="py-line"><tt class="py-docstring">    @EAS54_6_R1_2_1_443_348</tt> </tt>
<a name="L104"></a><tt class="py-lineno"> 104</tt>  <tt class="py-line"><tt class="py-docstring">    GTTGCTTCTGGCGTGGGTGGGGGGG</tt> </tt>
<a name="L105"></a><tt class="py-lineno"> 105</tt>  <tt class="py-line"><tt class="py-docstring">    +</tt> </tt>
<a name="L106"></a><tt class="py-lineno"> 106</tt>  <tt class="py-line"><tt class="py-docstring">    ;;;;;;;;;;;9;7;;.7;393333</tt> </tt>
<a name="L107"></a><tt class="py-lineno"> 107</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L108"></a><tt class="py-lineno"> 108</tt>  <tt class="py-line"><tt class="py-docstring">This contains three reads of length 25.  From the read length these were</tt> </tt>
<a name="L109"></a><tt class="py-lineno"> 109</tt>  <tt class="py-line"><tt class="py-docstring">probably originally from an early Solexa/Illumina sequencer but this file</tt> </tt>
<a name="L110"></a><tt class="py-lineno"> 110</tt>  <tt class="py-line"><tt class="py-docstring">follows the Sanger FASTQ convention (PHRED style qualities with an ASCII</tt> </tt>
<a name="L111"></a><tt class="py-lineno"> 111</tt>  <tt class="py-line"><tt class="py-docstring">offet of 33).  This means we can parse this file using Bio.SeqIO using</tt> </tt>
<a name="L112"></a><tt class="py-lineno"> 112</tt>  <tt class="py-line"><tt class="py-docstring">"fastq" as the format name:</tt> </tt>
<a name="L113"></a><tt class="py-lineno"> 113</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L114"></a><tt class="py-lineno"> 114</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L115"></a><tt class="py-lineno"> 115</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; for record in SeqIO.parse("Quality/example.fastq", "fastq"):</tt> </tt>
<a name="L116"></a><tt class="py-lineno"> 116</tt>  <tt class="py-line"><tt class="py-docstring">...     print("%s %s" % (record.id, record.seq))</tt> </tt>
<a name="L117"></a><tt class="py-lineno"> 117</tt>  <tt class="py-line"><tt class="py-docstring">EAS54_6_R1_2_1_413_324 CCCTTCTTGTCTTCAGCGTTTCTCC</tt> </tt>
<a name="L118"></a><tt class="py-lineno"> 118</tt>  <tt class="py-line"><tt class="py-docstring">EAS54_6_R1_2_1_540_792 TTGGCAGGCCAAGGCCGATGGATCA</tt> </tt>
<a name="L119"></a><tt class="py-lineno"> 119</tt>  <tt class="py-line"><tt class="py-docstring">EAS54_6_R1_2_1_443_348 GTTGCTTCTGGCGTGGGTGGGGGGG</tt> </tt>
<a name="L120"></a><tt class="py-lineno"> 120</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L121"></a><tt class="py-lineno"> 121</tt>  <tt class="py-line"><tt class="py-docstring">The qualities are held as a list of integers in each record's annotation:</tt> </tt>
<a name="L122"></a><tt class="py-lineno"> 122</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L123"></a><tt class="py-lineno"> 123</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; print(record)</tt> </tt>
<a name="L124"></a><tt class="py-lineno"> 124</tt>  <tt class="py-line"><tt class="py-docstring">ID: EAS54_6_R1_2_1_443_348</tt> </tt>
<a name="L125"></a><tt class="py-lineno"> 125</tt>  <tt class="py-line"><tt class="py-docstring">Name: EAS54_6_R1_2_1_443_348</tt> </tt>
<a name="L126"></a><tt class="py-lineno"> 126</tt>  <tt class="py-line"><tt class="py-docstring">Description: EAS54_6_R1_2_1_443_348</tt> </tt>
<a name="L127"></a><tt class="py-lineno"> 127</tt>  <tt class="py-line"><tt class="py-docstring">Number of features: 0</tt> </tt>
<a name="L128"></a><tt class="py-lineno"> 128</tt>  <tt class="py-line"><tt class="py-docstring">Per letter annotation for: phred_quality</tt> </tt>
<a name="L129"></a><tt class="py-lineno"> 129</tt>  <tt class="py-line"><tt class="py-docstring">Seq('GTTGCTTCTGGCGTGGGTGGGGGGG', SingleLetterAlphabet())</tt> </tt>
<a name="L130"></a><tt class="py-lineno"> 130</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; print(record.letter_annotations["phred_quality"])</tt> </tt>
<a name="L131"></a><tt class="py-lineno"> 131</tt>  <tt class="py-line"><tt class="py-docstring">[26, 26, 26, 26, 26, 26, 26, 26, 26, 26, 26, 24, 26, 22, 26, 26, 13, 22, 26, 18, 24, 18, 18, 18, 18]</tt> </tt>
<a name="L132"></a><tt class="py-lineno"> 132</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L133"></a><tt class="py-lineno"> 133</tt>  <tt class="py-line"><tt class="py-docstring">You can use the SeqRecord format method to show this in the QUAL format:</tt> </tt>
<a name="L134"></a><tt class="py-lineno"> 134</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L135"></a><tt class="py-lineno"> 135</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; print(record.format("qual"))</tt> </tt>
<a name="L136"></a><tt class="py-lineno"> 136</tt>  <tt class="py-line"><tt class="py-docstring">&gt;EAS54_6_R1_2_1_443_348</tt> </tt>
<a name="L137"></a><tt class="py-lineno"> 137</tt>  <tt class="py-line"><tt class="py-docstring">26 26 26 26 26 26 26 26 26 26 26 24 26 22 26 26 13 22 26 18</tt> </tt>
<a name="L138"></a><tt class="py-lineno"> 138</tt>  <tt class="py-line"><tt class="py-docstring">24 18 18 18 18</tt> </tt>
<a name="L139"></a><tt class="py-lineno"> 139</tt>  <tt class="py-line"><tt class="py-docstring">&lt;BLANKLINE&gt;</tt> </tt>
<a name="L140"></a><tt class="py-lineno"> 140</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L141"></a><tt class="py-lineno"> 141</tt>  <tt class="py-line"><tt class="py-docstring">Or go back to the FASTQ format, use "fastq" (or "fastq-sanger"):</tt> </tt>
<a name="L142"></a><tt class="py-lineno"> 142</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L143"></a><tt class="py-lineno"> 143</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; print(record.format("fastq"))</tt> </tt>
<a name="L144"></a><tt class="py-lineno"> 144</tt>  <tt class="py-line"><tt class="py-docstring">@EAS54_6_R1_2_1_443_348</tt> </tt>
<a name="L145"></a><tt class="py-lineno"> 145</tt>  <tt class="py-line"><tt class="py-docstring">GTTGCTTCTGGCGTGGGTGGGGGGG</tt> </tt>
<a name="L146"></a><tt class="py-lineno"> 146</tt>  <tt class="py-line"><tt class="py-docstring">+</tt> </tt>
<a name="L147"></a><tt class="py-lineno"> 147</tt>  <tt class="py-line"><tt class="py-docstring">;;;;;;;;;;;9;7;;.7;393333</tt> </tt>
<a name="L148"></a><tt class="py-lineno"> 148</tt>  <tt class="py-line"><tt class="py-docstring">&lt;BLANKLINE&gt;</tt> </tt>
<a name="L149"></a><tt class="py-lineno"> 149</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L150"></a><tt class="py-lineno"> 150</tt>  <tt class="py-line"><tt class="py-docstring">Or, using the Illumina 1.3+ FASTQ encoding (PHRED values with an ASCII offset</tt> </tt>
<a name="L151"></a><tt class="py-lineno"> 151</tt>  <tt class="py-line"><tt class="py-docstring">of 64):</tt> </tt>
<a name="L152"></a><tt class="py-lineno"> 152</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L153"></a><tt class="py-lineno"> 153</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; print(record.format("fastq-illumina"))</tt> </tt>
<a name="L154"></a><tt class="py-lineno"> 154</tt>  <tt class="py-line"><tt class="py-docstring">@EAS54_6_R1_2_1_443_348</tt> </tt>
<a name="L155"></a><tt class="py-lineno"> 155</tt>  <tt class="py-line"><tt class="py-docstring">GTTGCTTCTGGCGTGGGTGGGGGGG</tt> </tt>
<a name="L156"></a><tt class="py-lineno"> 156</tt>  <tt class="py-line"><tt class="py-docstring">+</tt> </tt>
<a name="L157"></a><tt class="py-lineno"> 157</tt>  <tt class="py-line"><tt class="py-docstring">ZZZZZZZZZZZXZVZZMVZRXRRRR</tt> </tt>
<a name="L158"></a><tt class="py-lineno"> 158</tt>  <tt class="py-line"><tt class="py-docstring">&lt;BLANKLINE&gt;</tt> </tt>
<a name="L159"></a><tt class="py-lineno"> 159</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L160"></a><tt class="py-lineno"> 160</tt>  <tt class="py-line"><tt class="py-docstring">You can also get Biopython to convert the scores and show a Solexa style</tt> </tt>
<a name="L161"></a><tt class="py-lineno"> 161</tt>  <tt class="py-line"><tt class="py-docstring">FASTQ file:</tt> </tt>
<a name="L162"></a><tt class="py-lineno"> 162</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L163"></a><tt class="py-lineno"> 163</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; print(record.format("fastq-solexa"))</tt> </tt>
<a name="L164"></a><tt class="py-lineno"> 164</tt>  <tt class="py-line"><tt class="py-docstring">@EAS54_6_R1_2_1_443_348</tt> </tt>
<a name="L165"></a><tt class="py-lineno"> 165</tt>  <tt class="py-line"><tt class="py-docstring">GTTGCTTCTGGCGTGGGTGGGGGGG</tt> </tt>
<a name="L166"></a><tt class="py-lineno"> 166</tt>  <tt class="py-line"><tt class="py-docstring">+</tt> </tt>
<a name="L167"></a><tt class="py-lineno"> 167</tt>  <tt class="py-line"><tt class="py-docstring">ZZZZZZZZZZZXZVZZMVZRXRRRR</tt> </tt>
<a name="L168"></a><tt class="py-lineno"> 168</tt>  <tt class="py-line"><tt class="py-docstring">&lt;BLANKLINE&gt;</tt> </tt>
<a name="L169"></a><tt class="py-lineno"> 169</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L170"></a><tt class="py-lineno"> 170</tt>  <tt class="py-line"><tt class="py-docstring">Notice that this is actually the same output as above using "fastq-illumina"</tt> </tt>
<a name="L171"></a><tt class="py-lineno"> 171</tt>  <tt class="py-line"><tt class="py-docstring">as the format! The reason for this is all these scores are high enough that</tt> </tt>
<a name="L172"></a><tt class="py-lineno"> 172</tt>  <tt class="py-line"><tt class="py-docstring">the PHRED and Solexa scores are almost equal. The differences become apparent</tt> </tt>
<a name="L173"></a><tt class="py-lineno"> 173</tt>  <tt class="py-line"><tt class="py-docstring">for poor quality reads. See the functions solexa_quality_from_phred and</tt> </tt>
<a name="L174"></a><tt class="py-lineno"> 174</tt>  <tt class="py-line"><tt class="py-docstring">phred_quality_from_solexa for more details.</tt> </tt>
<a name="L175"></a><tt class="py-lineno"> 175</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L176"></a><tt class="py-lineno"> 176</tt>  <tt class="py-line"><tt class="py-docstring">If you wanted to trim your sequences (perhaps to remove low quality regions,</tt> </tt>
<a name="L177"></a><tt class="py-lineno"> 177</tt>  <tt class="py-line"><tt class="py-docstring">or to remove a primer sequence), try slicing the SeqRecord objects.  e.g.</tt> </tt>
<a name="L178"></a><tt class="py-lineno"> 178</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L179"></a><tt class="py-lineno"> 179</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; sub_rec = record[5:15]</tt> </tt>
<a name="L180"></a><tt class="py-lineno"> 180</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; print(sub_rec)</tt> </tt>
<a name="L181"></a><tt class="py-lineno"> 181</tt>  <tt class="py-line"><tt class="py-docstring">ID: EAS54_6_R1_2_1_443_348</tt> </tt>
<a name="L182"></a><tt class="py-lineno"> 182</tt>  <tt class="py-line"><tt class="py-docstring">Name: EAS54_6_R1_2_1_443_348</tt> </tt>
<a name="L183"></a><tt class="py-lineno"> 183</tt>  <tt class="py-line"><tt class="py-docstring">Description: EAS54_6_R1_2_1_443_348</tt> </tt>
<a name="L184"></a><tt class="py-lineno"> 184</tt>  <tt class="py-line"><tt class="py-docstring">Number of features: 0</tt> </tt>
<a name="L185"></a><tt class="py-lineno"> 185</tt>  <tt class="py-line"><tt class="py-docstring">Per letter annotation for: phred_quality</tt> </tt>
<a name="L186"></a><tt class="py-lineno"> 186</tt>  <tt class="py-line"><tt class="py-docstring">Seq('TTCTGGCGTG', SingleLetterAlphabet())</tt> </tt>
<a name="L187"></a><tt class="py-lineno"> 187</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; print(sub_rec.letter_annotations["phred_quality"])</tt> </tt>
<a name="L188"></a><tt class="py-lineno"> 188</tt>  <tt class="py-line"><tt class="py-docstring">[26, 26, 26, 26, 26, 26, 24, 26, 22, 26]</tt> </tt>
<a name="L189"></a><tt class="py-lineno"> 189</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; print(sub_rec.format("fastq"))</tt> </tt>
<a name="L190"></a><tt class="py-lineno"> 190</tt>  <tt class="py-line"><tt class="py-docstring">@EAS54_6_R1_2_1_443_348</tt> </tt>
<a name="L191"></a><tt class="py-lineno"> 191</tt>  <tt class="py-line"><tt class="py-docstring">TTCTGGCGTG</tt> </tt>
<a name="L192"></a><tt class="py-lineno"> 192</tt>  <tt class="py-line"><tt class="py-docstring">+</tt> </tt>
<a name="L193"></a><tt class="py-lineno"> 193</tt>  <tt class="py-line"><tt class="py-docstring">;;;;;;9;7;</tt> </tt>
<a name="L194"></a><tt class="py-lineno"> 194</tt>  <tt class="py-line"><tt class="py-docstring">&lt;BLANKLINE&gt;</tt> </tt>
<a name="L195"></a><tt class="py-lineno"> 195</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L196"></a><tt class="py-lineno"> 196</tt>  <tt class="py-line"><tt class="py-docstring">If you wanted to, you could read in this FASTQ file, and save it as a QUAL file:</tt> </tt>
<a name="L197"></a><tt class="py-lineno"> 197</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L198"></a><tt class="py-lineno"> 198</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L199"></a><tt class="py-lineno"> 199</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; record_iterator = SeqIO.parse("Quality/example.fastq", "fastq")</tt> </tt>
<a name="L200"></a><tt class="py-lineno"> 200</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; with open("Quality/temp.qual", "w") as out_handle:</tt> </tt>
<a name="L201"></a><tt class="py-lineno"> 201</tt>  <tt class="py-line"><tt class="py-docstring">...     SeqIO.write(record_iterator, out_handle, "qual")</tt> </tt>
<a name="L202"></a><tt class="py-lineno"> 202</tt>  <tt class="py-line"><tt class="py-docstring">3</tt> </tt>
<a name="L203"></a><tt class="py-lineno"> 203</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L204"></a><tt class="py-lineno"> 204</tt>  <tt class="py-line"><tt class="py-docstring">You can of course read in a QUAL file, such as the one we just created:</tt> </tt>
<a name="L205"></a><tt class="py-lineno"> 205</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L206"></a><tt class="py-lineno"> 206</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L207"></a><tt class="py-lineno"> 207</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; for record in SeqIO.parse("Quality/temp.qual", "qual"):</tt> </tt>
<a name="L208"></a><tt class="py-lineno"> 208</tt>  <tt class="py-line"><tt class="py-docstring">...     print("%s %s" % (record.id, record.seq))</tt> </tt>
<a name="L209"></a><tt class="py-lineno"> 209</tt>  <tt class="py-line"><tt class="py-docstring">EAS54_6_R1_2_1_413_324 ?????????????????????????</tt> </tt>
<a name="L210"></a><tt class="py-lineno"> 210</tt>  <tt class="py-line"><tt class="py-docstring">EAS54_6_R1_2_1_540_792 ?????????????????????????</tt> </tt>
<a name="L211"></a><tt class="py-lineno"> 211</tt>  <tt class="py-line"><tt class="py-docstring">EAS54_6_R1_2_1_443_348 ?????????????????????????</tt> </tt>
<a name="L212"></a><tt class="py-lineno"> 212</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L213"></a><tt class="py-lineno"> 213</tt>  <tt class="py-line"><tt class="py-docstring">Notice that QUAL files don't have a proper sequence present!  But the quality</tt> </tt>
<a name="L214"></a><tt class="py-lineno"> 214</tt>  <tt class="py-line"><tt class="py-docstring">information is there:</tt> </tt>
<a name="L215"></a><tt class="py-lineno"> 215</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L216"></a><tt class="py-lineno"> 216</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; print(record)</tt> </tt>
<a name="L217"></a><tt class="py-lineno"> 217</tt>  <tt class="py-line"><tt class="py-docstring">ID: EAS54_6_R1_2_1_443_348</tt> </tt>
<a name="L218"></a><tt class="py-lineno"> 218</tt>  <tt class="py-line"><tt class="py-docstring">Name: EAS54_6_R1_2_1_443_348</tt> </tt>
<a name="L219"></a><tt class="py-lineno"> 219</tt>  <tt class="py-line"><tt class="py-docstring">Description: EAS54_6_R1_2_1_443_348</tt> </tt>
<a name="L220"></a><tt class="py-lineno"> 220</tt>  <tt class="py-line"><tt class="py-docstring">Number of features: 0</tt> </tt>
<a name="L221"></a><tt class="py-lineno"> 221</tt>  <tt class="py-line"><tt class="py-docstring">Per letter annotation for: phred_quality</tt> </tt>
<a name="L222"></a><tt class="py-lineno"> 222</tt>  <tt class="py-line"><tt class="py-docstring">UnknownSeq(25, alphabet = SingleLetterAlphabet(), character = '?')</tt> </tt>
<a name="L223"></a><tt class="py-lineno"> 223</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; print(record.letter_annotations["phred_quality"])</tt> </tt>
<a name="L224"></a><tt class="py-lineno"> 224</tt>  <tt class="py-line"><tt class="py-docstring">[26, 26, 26, 26, 26, 26, 26, 26, 26, 26, 26, 24, 26, 22, 26, 26, 13, 22, 26, 18, 24, 18, 18, 18, 18]</tt> </tt>
<a name="L225"></a><tt class="py-lineno"> 225</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L226"></a><tt class="py-lineno"> 226</tt>  <tt class="py-line"><tt class="py-docstring">Just to keep things tidy, if you are following this example yourself, you can</tt> </tt>
<a name="L227"></a><tt class="py-lineno"> 227</tt>  <tt class="py-line"><tt class="py-docstring">delete this temporary file now:</tt> </tt>
<a name="L228"></a><tt class="py-lineno"> 228</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L229"></a><tt class="py-lineno"> 229</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; import os</tt> </tt>
<a name="L230"></a><tt class="py-lineno"> 230</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; os.remove("Quality/temp.qual")</tt> </tt>
<a name="L231"></a><tt class="py-lineno"> 231</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L232"></a><tt class="py-lineno"> 232</tt>  <tt class="py-line"><tt class="py-docstring">Sometimes you won't have a FASTQ file, but rather just a pair of FASTA and QUAL</tt> </tt>
<a name="L233"></a><tt class="py-lineno"> 233</tt>  <tt class="py-line"><tt class="py-docstring">files.  Because the Bio.SeqIO system is designed for reading single files, you</tt> </tt>
<a name="L234"></a><tt class="py-lineno"> 234</tt>  <tt class="py-line"><tt class="py-docstring">would have to read the two in separately and then combine the data.  However,</tt> </tt>
<a name="L235"></a><tt class="py-lineno"> 235</tt>  <tt class="py-line"><tt class="py-docstring">since this is such a common thing to want to do, there is a helper iterator</tt> </tt>
<a name="L236"></a><tt class="py-lineno"> 236</tt>  <tt class="py-line"><tt class="py-docstring">defined in this module that does this for you - PairedFastaQualIterator.</tt> </tt>
<a name="L237"></a><tt class="py-lineno"> 237</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L238"></a><tt class="py-lineno"> 238</tt>  <tt class="py-line"><tt class="py-docstring">Alternatively, if you have enough RAM to hold all the records in memory at once,</tt> </tt>
<a name="L239"></a><tt class="py-lineno"> 239</tt>  <tt class="py-line"><tt class="py-docstring">then a simple dictionary approach would work:</tt> </tt>
<a name="L240"></a><tt class="py-lineno"> 240</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L241"></a><tt class="py-lineno"> 241</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L242"></a><tt class="py-lineno"> 242</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; reads = SeqIO.to_dict(SeqIO.parse("Quality/example.fasta", "fasta"))</tt> </tt>
<a name="L243"></a><tt class="py-lineno"> 243</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; for rec in SeqIO.parse("Quality/example.qual", "qual"):</tt> </tt>
<a name="L244"></a><tt class="py-lineno"> 244</tt>  <tt class="py-line"><tt class="py-docstring">...     reads[rec.id].letter_annotations["phred_quality"]=rec.letter_annotations["phred_quality"]</tt> </tt>
<a name="L245"></a><tt class="py-lineno"> 245</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L246"></a><tt class="py-lineno"> 246</tt>  <tt class="py-line"><tt class="py-docstring">You can then access any record by its key, and get both the sequence and the</tt> </tt>
<a name="L247"></a><tt class="py-lineno"> 247</tt>  <tt class="py-line"><tt class="py-docstring">quality scores.</tt> </tt>
<a name="L248"></a><tt class="py-lineno"> 248</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L249"></a><tt class="py-lineno"> 249</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; print(reads["EAS54_6_R1_2_1_540_792"].format("fastq"))</tt> </tt>
<a name="L250"></a><tt class="py-lineno"> 250</tt>  <tt class="py-line"><tt class="py-docstring">@EAS54_6_R1_2_1_540_792</tt> </tt>
<a name="L251"></a><tt class="py-lineno"> 251</tt>  <tt class="py-line"><tt class="py-docstring">TTGGCAGGCCAAGGCCGATGGATCA</tt> </tt>
<a name="L252"></a><tt class="py-lineno"> 252</tt>  <tt class="py-line"><tt class="py-docstring">+</tt> </tt>
<a name="L253"></a><tt class="py-lineno"> 253</tt>  <tt class="py-line"><tt class="py-docstring">;;;;;;;;;;;7;;;;;-;;;3;83</tt> </tt>
<a name="L254"></a><tt class="py-lineno"> 254</tt>  <tt class="py-line"><tt class="py-docstring">&lt;BLANKLINE&gt;</tt> </tt>
<a name="L255"></a><tt class="py-lineno"> 255</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L256"></a><tt class="py-lineno"> 256</tt>  <tt class="py-line"><tt class="py-docstring">It is important that you explicitly tell Bio.SeqIO which FASTQ variant you are</tt> </tt>
<a name="L257"></a><tt class="py-lineno"> 257</tt>  <tt class="py-line"><tt class="py-docstring">using ("fastq" or "fastq-sanger" for the Sanger standard using PHRED values,</tt> </tt>
<a name="L258"></a><tt class="py-lineno"> 258</tt>  <tt class="py-line"><tt class="py-docstring">"fastq-solexa" for the original Solexa/Illumina variant, or "fastq-illumina"</tt> </tt>
<a name="L259"></a><tt class="py-lineno"> 259</tt>  <tt class="py-line"><tt class="py-docstring">for the more recent variant), as this cannot be detected reliably</tt> </tt>
<a name="L260"></a><tt class="py-lineno"> 260</tt>  <tt class="py-line"><tt class="py-docstring">automatically.</tt> </tt>
<a name="L261"></a><tt class="py-lineno"> 261</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L262"></a><tt class="py-lineno"> 262</tt>  <tt class="py-line"><tt class="py-docstring">To illustrate this problem, let's consider an artifical example:</tt> </tt>
<a name="L263"></a><tt class="py-lineno"> 263</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L264"></a><tt class="py-lineno"> 264</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L265"></a><tt class="py-lineno"> 265</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; from Bio.Alphabet import generic_dna</tt> </tt>
<a name="L266"></a><tt class="py-lineno"> 266</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; from Bio.SeqRecord import SeqRecord</tt> </tt>
<a name="L267"></a><tt class="py-lineno"> 267</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; test = SeqRecord(Seq("NACGTACGTA", generic_dna), id="Test",</tt> </tt>
<a name="L268"></a><tt class="py-lineno"> 268</tt>  <tt class="py-line"><tt class="py-docstring">... description="Made up!")</tt> </tt>
<a name="L269"></a><tt class="py-lineno"> 269</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; print(test.format("fasta"))</tt> </tt>
<a name="L270"></a><tt class="py-lineno"> 270</tt>  <tt class="py-line"><tt class="py-docstring">&gt;Test Made up!</tt> </tt>
<a name="L271"></a><tt class="py-lineno"> 271</tt>  <tt class="py-line"><tt class="py-docstring">NACGTACGTA</tt> </tt>
<a name="L272"></a><tt class="py-lineno"> 272</tt>  <tt class="py-line"><tt class="py-docstring">&lt;BLANKLINE&gt;</tt> </tt>
<a name="L273"></a><tt class="py-lineno"> 273</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; print(test.format("fastq"))</tt> </tt>
<a name="L274"></a><tt class="py-lineno"> 274</tt>  <tt class="py-line"><tt class="py-docstring">Traceback (most recent call last):</tt> </tt>
<a name="L275"></a><tt class="py-lineno"> 275</tt>  <tt class="py-line"><tt class="py-docstring"> ...</tt> </tt>
<a name="L276"></a><tt class="py-lineno"> 276</tt>  <tt class="py-line"><tt class="py-docstring">ValueError: No suitable quality scores found in letter_annotations of SeqRecord (id=Test).</tt> </tt>
<a name="L277"></a><tt class="py-lineno"> 277</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L278"></a><tt class="py-lineno"> 278</tt>  <tt class="py-line"><tt class="py-docstring">We created a sample SeqRecord, and can show it in FASTA format - but for QUAL</tt> </tt>
<a name="L279"></a><tt class="py-lineno"> 279</tt>  <tt class="py-line"><tt class="py-docstring">or FASTQ format we need to provide some quality scores. These are held as a</tt> </tt>
<a name="L280"></a><tt class="py-lineno"> 280</tt>  <tt class="py-line"><tt class="py-docstring">list of integers (one for each base) in the letter_annotations dictionary:</tt> </tt>
<a name="L281"></a><tt class="py-lineno"> 281</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L282"></a><tt class="py-lineno"> 282</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; test.letter_annotations["phred_quality"] = [0, 1, 2, 3, 4, 5, 10, 20, 30, 40]</tt> </tt>
<a name="L283"></a><tt class="py-lineno"> 283</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; print(test.format("qual"))</tt> </tt>
<a name="L284"></a><tt class="py-lineno"> 284</tt>  <tt class="py-line"><tt class="py-docstring">&gt;Test Made up!</tt> </tt>
<a name="L285"></a><tt class="py-lineno"> 285</tt>  <tt class="py-line"><tt class="py-docstring">0 1 2 3 4 5 10 20 30 40</tt> </tt>
<a name="L286"></a><tt class="py-lineno"> 286</tt>  <tt class="py-line"><tt class="py-docstring">&lt;BLANKLINE&gt;</tt> </tt>
<a name="L287"></a><tt class="py-lineno"> 287</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; print(test.format("fastq"))</tt> </tt>
<a name="L288"></a><tt class="py-lineno"> 288</tt>  <tt class="py-line"><tt class="py-docstring">@Test Made up!</tt> </tt>
<a name="L289"></a><tt class="py-lineno"> 289</tt>  <tt class="py-line"><tt class="py-docstring">NACGTACGTA</tt> </tt>
<a name="L290"></a><tt class="py-lineno"> 290</tt>  <tt class="py-line"><tt class="py-docstring">+</tt> </tt>
<a name="L291"></a><tt class="py-lineno"> 291</tt>  <tt class="py-line"><tt class="py-docstring">!"#$%&amp;+5?I</tt> </tt>
<a name="L292"></a><tt class="py-lineno"> 292</tt>  <tt class="py-line"><tt class="py-docstring">&lt;BLANKLINE&gt;</tt> </tt>
<a name="L293"></a><tt class="py-lineno"> 293</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L294"></a><tt class="py-lineno"> 294</tt>  <tt class="py-line"><tt class="py-docstring">We can check this FASTQ encoding - the first PHRED quality was zero, and this</tt> </tt>
<a name="L295"></a><tt class="py-lineno"> 295</tt>  <tt class="py-line"><tt class="py-docstring">mapped to a exclamation mark, while the final score was 40 and this mapped to</tt> </tt>
<a name="L296"></a><tt class="py-lineno"> 296</tt>  <tt class="py-line"><tt class="py-docstring">the letter "I":</tt> </tt>
<a name="L297"></a><tt class="py-lineno"> 297</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L298"></a><tt class="py-lineno"> 298</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; ord('!') - 33</tt> </tt>
<a name="L299"></a><tt class="py-lineno"> 299</tt>  <tt class="py-line"><tt class="py-docstring">0</tt> </tt>
<a name="L300"></a><tt class="py-lineno"> 300</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; ord('I') - 33</tt> </tt>
<a name="L301"></a><tt class="py-lineno"> 301</tt>  <tt class="py-line"><tt class="py-docstring">40</tt> </tt>
<a name="L302"></a><tt class="py-lineno"> 302</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; [ord(letter)-33 for letter in '!"#$%&amp;+5?I']</tt> </tt>
<a name="L303"></a><tt class="py-lineno"> 303</tt>  <tt class="py-line"><tt class="py-docstring">[0, 1, 2, 3, 4, 5, 10, 20, 30, 40]</tt> </tt>
<a name="L304"></a><tt class="py-lineno"> 304</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L305"></a><tt class="py-lineno"> 305</tt>  <tt class="py-line"><tt class="py-docstring">Similarly, we could produce an Illumina 1.3 to 1.7 style FASTQ file using PHRED</tt> </tt>
<a name="L306"></a><tt class="py-lineno"> 306</tt>  <tt class="py-line"><tt class="py-docstring">scores with an offset of 64:</tt> </tt>
<a name="L307"></a><tt class="py-lineno"> 307</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L308"></a><tt class="py-lineno"> 308</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; print(test.format("fastq-illumina"))</tt> </tt>
<a name="L309"></a><tt class="py-lineno"> 309</tt>  <tt class="py-line"><tt class="py-docstring">@Test Made up!</tt> </tt>
<a name="L310"></a><tt class="py-lineno"> 310</tt>  <tt class="py-line"><tt class="py-docstring">NACGTACGTA</tt> </tt>
<a name="L311"></a><tt class="py-lineno"> 311</tt>  <tt class="py-line"><tt class="py-docstring">+</tt> </tt>
<a name="L312"></a><tt class="py-lineno"> 312</tt>  <tt class="py-line"><tt class="py-docstring">@ABCDEJT^h</tt> </tt>
<a name="L313"></a><tt class="py-lineno"> 313</tt>  <tt class="py-line"><tt class="py-docstring">&lt;BLANKLINE&gt;</tt> </tt>
<a name="L314"></a><tt class="py-lineno"> 314</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L315"></a><tt class="py-lineno"> 315</tt>  <tt class="py-line"><tt class="py-docstring">And we can check this too - the first PHRED score was zero, and this mapped to</tt> </tt>
<a name="L316"></a><tt class="py-lineno"> 316</tt>  <tt class="py-line"><tt class="py-docstring">"@", while the final score was 40 and this mapped to "h":</tt> </tt>
<a name="L317"></a><tt class="py-lineno"> 317</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L318"></a><tt class="py-lineno"> 318</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; ord("@") - 64</tt> </tt>
<a name="L319"></a><tt class="py-lineno"> 319</tt>  <tt class="py-line"><tt class="py-docstring">0</tt> </tt>
<a name="L320"></a><tt class="py-lineno"> 320</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; ord("h") - 64</tt> </tt>
<a name="L321"></a><tt class="py-lineno"> 321</tt>  <tt class="py-line"><tt class="py-docstring">40</tt> </tt>
<a name="L322"></a><tt class="py-lineno"> 322</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; [ord(letter)-64 for letter in "@ABCDEJT^h"]</tt> </tt>
<a name="L323"></a><tt class="py-lineno"> 323</tt>  <tt class="py-line"><tt class="py-docstring">[0, 1, 2, 3, 4, 5, 10, 20, 30, 40]</tt> </tt>
<a name="L324"></a><tt class="py-lineno"> 324</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L325"></a><tt class="py-lineno"> 325</tt>  <tt class="py-line"><tt class="py-docstring">Notice how different the standard Sanger FASTQ and the Illumina 1.3 to 1.7 style</tt> </tt>
<a name="L326"></a><tt class="py-lineno"> 326</tt>  <tt class="py-line"><tt class="py-docstring">FASTQ files look for the same data! Then we have the older Solexa/Illumina</tt> </tt>
<a name="L327"></a><tt class="py-lineno"> 327</tt>  <tt class="py-line"><tt class="py-docstring">format to consider which encodes Solexa scores instead of PHRED scores.</tt> </tt>
<a name="L328"></a><tt class="py-lineno"> 328</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L329"></a><tt class="py-lineno"> 329</tt>  <tt class="py-line"><tt class="py-docstring">First let's see what Biopython says if we convert the PHRED scores into Solexa</tt> </tt>
<a name="L330"></a><tt class="py-lineno"> 330</tt>  <tt class="py-line"><tt class="py-docstring">scores (rounding to one decimal place):</tt> </tt>
<a name="L331"></a><tt class="py-lineno"> 331</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L332"></a><tt class="py-lineno"> 332</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; for q in [0, 1, 2, 3, 4, 5, 10, 20, 30, 40]:</tt> </tt>
<a name="L333"></a><tt class="py-lineno"> 333</tt>  <tt class="py-line"><tt class="py-docstring">...     print("PHRED %i maps to Solexa %0.1f" % (q, solexa_quality_from_phred(q)))</tt> </tt>
<a name="L334"></a><tt class="py-lineno"> 334</tt>  <tt class="py-line"><tt class="py-docstring">PHRED 0 maps to Solexa -5.0</tt> </tt>
<a name="L335"></a><tt class="py-lineno"> 335</tt>  <tt class="py-line"><tt class="py-docstring">PHRED 1 maps to Solexa -5.0</tt> </tt>
<a name="L336"></a><tt class="py-lineno"> 336</tt>  <tt class="py-line"><tt class="py-docstring">PHRED 2 maps to Solexa -2.3</tt> </tt>
<a name="L337"></a><tt class="py-lineno"> 337</tt>  <tt class="py-line"><tt class="py-docstring">PHRED 3 maps to Solexa -0.0</tt> </tt>
<a name="L338"></a><tt class="py-lineno"> 338</tt>  <tt class="py-line"><tt class="py-docstring">PHRED 4 maps to Solexa 1.8</tt> </tt>
<a name="L339"></a><tt class="py-lineno"> 339</tt>  <tt class="py-line"><tt class="py-docstring">PHRED 5 maps to Solexa 3.3</tt> </tt>
<a name="L340"></a><tt class="py-lineno"> 340</tt>  <tt class="py-line"><tt class="py-docstring">PHRED 10 maps to Solexa 9.5</tt> </tt>
<a name="L341"></a><tt class="py-lineno"> 341</tt>  <tt class="py-line"><tt class="py-docstring">PHRED 20 maps to Solexa 20.0</tt> </tt>
<a name="L342"></a><tt class="py-lineno"> 342</tt>  <tt class="py-line"><tt class="py-docstring">PHRED 30 maps to Solexa 30.0</tt> </tt>
<a name="L343"></a><tt class="py-lineno"> 343</tt>  <tt class="py-line"><tt class="py-docstring">PHRED 40 maps to Solexa 40.0</tt> </tt>
<a name="L344"></a><tt class="py-lineno"> 344</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L345"></a><tt class="py-lineno"> 345</tt>  <tt class="py-line"><tt class="py-docstring">Now here is the record using the old Solexa style FASTQ file:</tt> </tt>
<a name="L346"></a><tt class="py-lineno"> 346</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L347"></a><tt class="py-lineno"> 347</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; print(test.format("fastq-solexa"))</tt> </tt>
<a name="L348"></a><tt class="py-lineno"> 348</tt>  <tt class="py-line"><tt class="py-docstring">@Test Made up!</tt> </tt>
<a name="L349"></a><tt class="py-lineno"> 349</tt>  <tt class="py-line"><tt class="py-docstring">NACGTACGTA</tt> </tt>
<a name="L350"></a><tt class="py-lineno"> 350</tt>  <tt class="py-line"><tt class="py-docstring">+</tt> </tt>
<a name="L351"></a><tt class="py-lineno"> 351</tt>  <tt class="py-line"><tt class="py-docstring">;;&gt;@BCJT^h</tt> </tt>
<a name="L352"></a><tt class="py-lineno"> 352</tt>  <tt class="py-line"><tt class="py-docstring">&lt;BLANKLINE&gt;</tt> </tt>
<a name="L353"></a><tt class="py-lineno"> 353</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L354"></a><tt class="py-lineno"> 354</tt>  <tt class="py-line"><tt class="py-docstring">Again, this is using an ASCII offset of 64, so we can check the Solexa scores:</tt> </tt>
<a name="L355"></a><tt class="py-lineno"> 355</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L356"></a><tt class="py-lineno"> 356</tt>  <tt class="py-line"><tt class="py-docstring">&gt;&gt;&gt; [ord(letter)-64 for letter in ";;&gt;@BCJT^h"]</tt> </tt>
<a name="L357"></a><tt class="py-lineno"> 357</tt>  <tt class="py-line"><tt class="py-docstring">[-5, -5, -2, 0, 2, 3, 10, 20, 30, 40]</tt> </tt>
<a name="L358"></a><tt class="py-lineno"> 358</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L359"></a><tt class="py-lineno"> 359</tt>  <tt class="py-line"><tt class="py-docstring">This explains why the last few letters of this FASTQ output matched that using</tt> </tt>
<a name="L360"></a><tt class="py-lineno"> 360</tt>  <tt class="py-line"><tt class="py-docstring">the Illumina 1.3 to 1.7 format - high quality PHRED scores and Solexa scores</tt> </tt>
<a name="L361"></a><tt class="py-lineno"> 361</tt>  <tt class="py-line"><tt class="py-docstring">are approximately equal.</tt> </tt>
<a name="L362"></a><tt class="py-lineno"> 362</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L363"></a><tt class="py-lineno"> 363</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L364"></a><tt class="py-lineno"> 364</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">__future__</tt> <tt class="py-keyword">import</tt> <tt class="py-name">print_function</tt> </tt>
<a name="L365"></a><tt class="py-lineno"> 365</tt>  <tt class="py-line"> </tt>
<a name="L366"></a><tt class="py-lineno"> 366</tt>  <tt class="py-line"><tt class="py-name">__docformat__</tt> <tt class="py-op">=</tt> <tt class="py-string">"restructuredtext en"</tt>  <tt class="py-comment"># Don't just use plain text in epydoc API pages!</tt> </tt>
<a name="L367"></a><tt class="py-lineno"> 367</tt>  <tt class="py-line"> </tt>
<a name="L368"></a><tt class="py-lineno"> 368</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio.Alphabet=Bio.Alphabet-module.html,Class Bio.Alphabet.Alphabet=Bio.Alphabet.Alphabet-class.html"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-1', 'Alphabet', 'link-1');">Alphabet</a></tt> <tt class="py-keyword">import</tt> <tt id="link-2" class="py-name" targets="Variable Bio.Alphabet.single_letter_alphabet=Bio.Alphabet-module.html#single_letter_alphabet"><a title="Bio.Alphabet.single_letter_alphabet" class="py-name" href="#" onclick="return doclink('link-2', 'single_letter_alphabet', 'link-2');">single_letter_alphabet</a></tt> </tt>
<a name="L369"></a><tt class="py-lineno"> 369</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-3" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-3', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-4" class="py-name" targets="Module Bio.Seq=Bio.Seq-module.html,Class Bio.Seq.Seq=Bio.Seq.Seq-class.html"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-4', 'Seq', 'link-4');">Seq</a></tt> <tt class="py-keyword">import</tt> <tt id="link-5" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-5', 'Seq', 'link-4');">Seq</a></tt><tt class="py-op">,</tt> <tt id="link-6" class="py-name" targets="Class Bio.Seq.UnknownSeq=Bio.Seq.UnknownSeq-class.html"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-6', 'UnknownSeq', 'link-6');">UnknownSeq</a></tt> </tt>
<a name="L370"></a><tt class="py-lineno"> 370</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-7" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-7', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-8" class="py-name" targets="Module Bio.SeqRecord=Bio.SeqRecord-module.html,Class Bio.SeqRecord.SeqRecord=Bio.SeqRecord.SeqRecord-class.html"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-8', 'SeqRecord', 'link-8');">SeqRecord</a></tt> <tt class="py-keyword">import</tt> <tt id="link-9" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-9', 'SeqRecord', 'link-8');">SeqRecord</a></tt> </tt>
<a name="L371"></a><tt class="py-lineno"> 371</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-10" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-10', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-11" class="py-name" targets="Package Bio.SeqIO=Bio.SeqIO-module.html"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-11', 'SeqIO', 'link-11');">SeqIO</a></tt><tt class="py-op">.</tt><tt id="link-12" class="py-name" targets="Module Bio.AlignIO.Interfaces=Bio.AlignIO.Interfaces-module.html,Module Bio.SeqIO.Interfaces=Bio.SeqIO.Interfaces-module.html"><a title="Bio.AlignIO.Interfaces
Bio.SeqIO.Interfaces" class="py-name" href="#" onclick="return doclink('link-12', 'Interfaces', 'link-12');">Interfaces</a></tt> <tt class="py-keyword">import</tt> <tt id="link-13" class="py-name" targets="Class Bio.SeqIO.Interfaces.SequentialSequenceWriter=Bio.SeqIO.Interfaces.SequentialSequenceWriter-class.html"><a title="Bio.SeqIO.Interfaces.SequentialSequenceWriter" class="py-name" href="#" onclick="return doclink('link-13', 'SequentialSequenceWriter', 'link-13');">SequentialSequenceWriter</a></tt> </tt>
<a name="L372"></a><tt class="py-lineno"> 372</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">math</tt> <tt class="py-keyword">import</tt> <tt class="py-name">log</tt> </tt>
<a name="L373"></a><tt class="py-lineno"> 373</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L374"></a><tt class="py-lineno"> 374</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-14" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-14', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-15" class="py-name" targets="Class Bio.BiopythonWarning=Bio.BiopythonWarning-class.html"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-15', 'BiopythonWarning', 'link-15');">BiopythonWarning</a></tt><tt class="py-op">,</tt> <tt id="link-16" class="py-name" targets="Class Bio.BiopythonParserWarning=Bio.BiopythonParserWarning-class.html"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-16', 'BiopythonParserWarning', 'link-16');">BiopythonParserWarning</a></tt> </tt>
<a name="L375"></a><tt class="py-lineno"> 375</tt>  <tt class="py-line"> </tt>
<a name="L376"></a><tt class="py-lineno"> 376</tt>  <tt class="py-line"> </tt>
<a name="L377"></a><tt class="py-lineno"> 377</tt>  <tt class="py-line"><tt class="py-comment"># define score offsets. See discussion for differences between Sanger and</tt> </tt>
<a name="L378"></a><tt class="py-lineno"> 378</tt>  <tt class="py-line"><tt class="py-comment"># Solexa offsets.</tt> </tt>
<a name="L379"></a><tt class="py-lineno"> 379</tt>  <tt class="py-line"><tt id="link-17" class="py-name" targets="Variable Bio.SeqIO.QualityIO.SANGER_SCORE_OFFSET=Bio.SeqIO.QualityIO-module.html#SANGER_SCORE_OFFSET"><a title="Bio.SeqIO.QualityIO.SANGER_SCORE_OFFSET" class="py-name" href="#" onclick="return doclink('link-17', 'SANGER_SCORE_OFFSET', 'link-17');">SANGER_SCORE_OFFSET</a></tt> <tt class="py-op">=</tt> <tt class="py-number">33</tt> </tt>
<a name="L380"></a><tt class="py-lineno"> 380</tt>  <tt class="py-line"><tt id="link-18" class="py-name" targets="Variable Bio.SeqIO.QualityIO.SOLEXA_SCORE_OFFSET=Bio.SeqIO.QualityIO-module.html#SOLEXA_SCORE_OFFSET"><a title="Bio.SeqIO.QualityIO.SOLEXA_SCORE_OFFSET" class="py-name" href="#" onclick="return doclink('link-18', 'SOLEXA_SCORE_OFFSET', 'link-18');">SOLEXA_SCORE_OFFSET</a></tt> <tt class="py-op">=</tt> <tt class="py-number">64</tt> </tt>
<a name="L381"></a><tt class="py-lineno"> 381</tt>  <tt class="py-line"> </tt>
<a name="L382"></a><tt class="py-lineno"> 382</tt>  <tt class="py-line"> </tt>
<a name="solexa_quality_from_phred"></a><div id="solexa_quality_from_phred-def"><a name="L383"></a><tt class="py-lineno"> 383</tt> <a class="py-toggle" href="#" id="solexa_quality_from_phred-toggle" onclick="return toggle('solexa_quality_from_phred');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.QualityIO-module.html#solexa_quality_from_phred">solexa_quality_from_phred</a><tt class="py-op">(</tt><tt class="py-param">phred_quality</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="solexa_quality_from_phred-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="solexa_quality_from_phred-expanded"><a name="L384"></a><tt class="py-lineno"> 384</tt>  <tt class="py-line">    <tt class="py-docstring">"""Covert a PHRED quality (range 0 to about 90) to a Solexa quality.</tt> </tt>
<a name="L385"></a><tt class="py-lineno"> 385</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L386"></a><tt class="py-lineno"> 386</tt>  <tt class="py-line"><tt class="py-docstring">    PHRED and Solexa quality scores are both log transformations of a</tt> </tt>
<a name="L387"></a><tt class="py-lineno"> 387</tt>  <tt class="py-line"><tt class="py-docstring">    probality of error (high score = low probability of error). This function</tt> </tt>
<a name="L388"></a><tt class="py-lineno"> 388</tt>  <tt class="py-line"><tt class="py-docstring">    takes a PHRED score, transforms it back to a probability of error, and</tt> </tt>
<a name="L389"></a><tt class="py-lineno"> 389</tt>  <tt class="py-line"><tt class="py-docstring">    then re-expresses it as a Solexa score. This assumes the error estimates</tt> </tt>
<a name="L390"></a><tt class="py-lineno"> 390</tt>  <tt class="py-line"><tt class="py-docstring">    are equivalent.</tt> </tt>
<a name="L391"></a><tt class="py-lineno"> 391</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L392"></a><tt class="py-lineno"> 392</tt>  <tt class="py-line"><tt class="py-docstring">    How does this work exactly? Well the PHRED quality is minus ten times the</tt> </tt>
<a name="L393"></a><tt class="py-lineno"> 393</tt>  <tt class="py-line"><tt class="py-docstring">    base ten logarithm of the probability of error::</tt> </tt>
<a name="L394"></a><tt class="py-lineno"> 394</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L395"></a><tt class="py-lineno"> 395</tt>  <tt class="py-line"><tt class="py-docstring">        phred_quality = -10*log(error,10)</tt> </tt>
<a name="L396"></a><tt class="py-lineno"> 396</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L397"></a><tt class="py-lineno"> 397</tt>  <tt class="py-line"><tt class="py-docstring">    Therefore, turning this round::</tt> </tt>
<a name="L398"></a><tt class="py-lineno"> 398</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L399"></a><tt class="py-lineno"> 399</tt>  <tt class="py-line"><tt class="py-docstring">        error = 10 ** (- phred_quality / 10)</tt> </tt>
<a name="L400"></a><tt class="py-lineno"> 400</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L401"></a><tt class="py-lineno"> 401</tt>  <tt class="py-line"><tt class="py-docstring">    Now, Solexa qualities use a different log transformation::</tt> </tt>
<a name="L402"></a><tt class="py-lineno"> 402</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L403"></a><tt class="py-lineno"> 403</tt>  <tt class="py-line"><tt class="py-docstring">        solexa_quality = -10*log(error/(1-error),10)</tt> </tt>
<a name="L404"></a><tt class="py-lineno"> 404</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L405"></a><tt class="py-lineno"> 405</tt>  <tt class="py-line"><tt class="py-docstring">    After substitution and a little manipulation we get::</tt> </tt>
<a name="L406"></a><tt class="py-lineno"> 406</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L407"></a><tt class="py-lineno"> 407</tt>  <tt class="py-line"><tt class="py-docstring">         solexa_quality = 10*log(10**(phred_quality/10.0) - 1, 10)</tt> </tt>
<a name="L408"></a><tt class="py-lineno"> 408</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L409"></a><tt class="py-lineno"> 409</tt>  <tt class="py-line"><tt class="py-docstring">    However, real Solexa files use a minimum quality of -5. This does have a</tt> </tt>
<a name="L410"></a><tt class="py-lineno"> 410</tt>  <tt class="py-line"><tt class="py-docstring">    good reason - a random base call would be correct 25% of the time,</tt> </tt>
<a name="L411"></a><tt class="py-lineno"> 411</tt>  <tt class="py-line"><tt class="py-docstring">    and thus have a probability of error of 0.75, which gives 1.25 as the PHRED</tt> </tt>
<a name="L412"></a><tt class="py-lineno"> 412</tt>  <tt class="py-line"><tt class="py-docstring">    quality, or -4.77 as the Solexa quality. Thus (after rounding), a random</tt> </tt>
<a name="L413"></a><tt class="py-lineno"> 413</tt>  <tt class="py-line"><tt class="py-docstring">    nucleotide read would have a PHRED quality of 1, or a Solexa quality of -5.</tt> </tt>
<a name="L414"></a><tt class="py-lineno"> 414</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L415"></a><tt class="py-lineno"> 415</tt>  <tt class="py-line"><tt class="py-docstring">    Taken literally, this logarithic formula would map a PHRED quality of zero</tt> </tt>
<a name="L416"></a><tt class="py-lineno"> 416</tt>  <tt class="py-line"><tt class="py-docstring">    to a Solexa quality of minus infinity. Of course, taken literally, a PHRED</tt> </tt>
<a name="L417"></a><tt class="py-lineno"> 417</tt>  <tt class="py-line"><tt class="py-docstring">    score of zero means a probability of error of one (i.e. the base call is</tt> </tt>
<a name="L418"></a><tt class="py-lineno"> 418</tt>  <tt class="py-line"><tt class="py-docstring">    definitely wrong), which is worse than random! In practice, a PHRED quality</tt> </tt>
<a name="L419"></a><tt class="py-lineno"> 419</tt>  <tt class="py-line"><tt class="py-docstring">    of zero usually means a default value, or perhaps random - and therefore</tt> </tt>
<a name="L420"></a><tt class="py-lineno"> 420</tt>  <tt class="py-line"><tt class="py-docstring">    mapping it to the minimum Solexa score of -5 is reasonable.</tt> </tt>
<a name="L421"></a><tt class="py-lineno"> 421</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L422"></a><tt class="py-lineno"> 422</tt>  <tt class="py-line"><tt class="py-docstring">    In conclusion, we follow EMBOSS, and take this logarithmic formula but also</tt> </tt>
<a name="L423"></a><tt class="py-lineno"> 423</tt>  <tt class="py-line"><tt class="py-docstring">    apply a minimum value of -5.0 for the Solexa quality, and also map a PHRED</tt> </tt>
<a name="L424"></a><tt class="py-lineno"> 424</tt>  <tt class="py-line"><tt class="py-docstring">    quality of zero to -5.0 as well.</tt> </tt>
<a name="L425"></a><tt class="py-lineno"> 425</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L426"></a><tt class="py-lineno"> 426</tt>  <tt class="py-line"><tt class="py-docstring">    Note this function will return a floating point number, it is up to you to</tt> </tt>
<a name="L427"></a><tt class="py-lineno"> 427</tt>  <tt class="py-line"><tt class="py-docstring">    round this to the nearest integer if appropriate.  e.g.</tt> </tt>
<a name="L428"></a><tt class="py-lineno"> 428</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L429"></a><tt class="py-lineno"> 429</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%0.2f" % round(solexa_quality_from_phred(80), 2))</tt> </tt>
<a name="L430"></a><tt class="py-lineno"> 430</tt>  <tt class="py-line"><tt class="py-docstring">    80.00</tt> </tt>
<a name="L431"></a><tt class="py-lineno"> 431</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%0.2f" % round(solexa_quality_from_phred(50), 2))</tt> </tt>
<a name="L432"></a><tt class="py-lineno"> 432</tt>  <tt class="py-line"><tt class="py-docstring">    50.00</tt> </tt>
<a name="L433"></a><tt class="py-lineno"> 433</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%0.2f" % round(solexa_quality_from_phred(20), 2))</tt> </tt>
<a name="L434"></a><tt class="py-lineno"> 434</tt>  <tt class="py-line"><tt class="py-docstring">    19.96</tt> </tt>
<a name="L435"></a><tt class="py-lineno"> 435</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%0.2f" % round(solexa_quality_from_phred(10), 2))</tt> </tt>
<a name="L436"></a><tt class="py-lineno"> 436</tt>  <tt class="py-line"><tt class="py-docstring">    9.54</tt> </tt>
<a name="L437"></a><tt class="py-lineno"> 437</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%0.2f" % round(solexa_quality_from_phred(5), 2))</tt> </tt>
<a name="L438"></a><tt class="py-lineno"> 438</tt>  <tt class="py-line"><tt class="py-docstring">    3.35</tt> </tt>
<a name="L439"></a><tt class="py-lineno"> 439</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%0.2f" % round(solexa_quality_from_phred(4), 2))</tt> </tt>
<a name="L440"></a><tt class="py-lineno"> 440</tt>  <tt class="py-line"><tt class="py-docstring">    1.80</tt> </tt>
<a name="L441"></a><tt class="py-lineno"> 441</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%0.2f" % round(solexa_quality_from_phred(3), 2))</tt> </tt>
<a name="L442"></a><tt class="py-lineno"> 442</tt>  <tt class="py-line"><tt class="py-docstring">    -0.02</tt> </tt>
<a name="L443"></a><tt class="py-lineno"> 443</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%0.2f" % round(solexa_quality_from_phred(2), 2))</tt> </tt>
<a name="L444"></a><tt class="py-lineno"> 444</tt>  <tt class="py-line"><tt class="py-docstring">    -2.33</tt> </tt>
<a name="L445"></a><tt class="py-lineno"> 445</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%0.2f" % round(solexa_quality_from_phred(1), 2))</tt> </tt>
<a name="L446"></a><tt class="py-lineno"> 446</tt>  <tt class="py-line"><tt class="py-docstring">    -5.00</tt> </tt>
<a name="L447"></a><tt class="py-lineno"> 447</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%0.2f" % round(solexa_quality_from_phred(0), 2))</tt> </tt>
<a name="L448"></a><tt class="py-lineno"> 448</tt>  <tt class="py-line"><tt class="py-docstring">    -5.00</tt> </tt>
<a name="L449"></a><tt class="py-lineno"> 449</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L450"></a><tt class="py-lineno"> 450</tt>  <tt class="py-line"><tt class="py-docstring">    Notice that for high quality reads PHRED and Solexa scores are numerically</tt> </tt>
<a name="L451"></a><tt class="py-lineno"> 451</tt>  <tt class="py-line"><tt class="py-docstring">    equal. The differences are important for poor quality reads, where PHRED</tt> </tt>
<a name="L452"></a><tt class="py-lineno"> 452</tt>  <tt class="py-line"><tt class="py-docstring">    has a minimum of zero but Solexa scores can be negative.</tt> </tt>
<a name="L453"></a><tt class="py-lineno"> 453</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L454"></a><tt class="py-lineno"> 454</tt>  <tt class="py-line"><tt class="py-docstring">    Finally, as a special case where None is used for a "missing value", None</tt> </tt>
<a name="L455"></a><tt class="py-lineno"> 455</tt>  <tt class="py-line"><tt class="py-docstring">    is returned:</tt> </tt>
<a name="L456"></a><tt class="py-lineno"> 456</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L457"></a><tt class="py-lineno"> 457</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(solexa_quality_from_phred(None))</tt> </tt>
<a name="L458"></a><tt class="py-lineno"> 458</tt>  <tt class="py-line"><tt class="py-docstring">    None</tt> </tt>
<a name="L459"></a><tt class="py-lineno"> 459</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L460"></a><tt class="py-lineno"> 460</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">phred_quality</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L461"></a><tt class="py-lineno"> 461</tt>  <tt class="py-line">        <tt class="py-comment"># Assume None is used as some kind of NULL or NA value; return None</tt> </tt>
<a name="L462"></a><tt class="py-lineno"> 462</tt>  <tt class="py-line">        <tt class="py-comment"># e.g. Bio.SeqIO gives Ace contig gaps a quality of None.</tt> </tt>
<a name="L463"></a><tt class="py-lineno"> 463</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
<a name="L464"></a><tt class="py-lineno"> 464</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">phred_quality</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L465"></a><tt class="py-lineno"> 465</tt>  <tt class="py-line">        <tt class="py-comment"># Solexa uses a minimum value of -5, which after rounding matches a</tt> </tt>
<a name="L466"></a><tt class="py-lineno"> 466</tt>  <tt class="py-line">        <tt class="py-comment"># random nucleotide base call.</tt> </tt>
<a name="L467"></a><tt class="py-lineno"> 467</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-19" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.max=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#max"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-19', 'max', 'link-19');">max</a></tt><tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">5.0</tt><tt class="py-op">,</tt> <tt class="py-number">10</tt> <tt class="py-op">*</tt> <tt class="py-name">log</tt><tt class="py-op">(</tt><tt class="py-number">10</tt> <tt class="py-op">**</tt> <tt class="py-op">(</tt><tt class="py-name">phred_quality</tt> <tt class="py-op">/</tt> <tt class="py-number">10.0</tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">10</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L468"></a><tt class="py-lineno"> 468</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">phred_quality</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L469"></a><tt class="py-lineno"> 469</tt>  <tt class="py-line">        <tt class="py-comment"># Special case, map to -5 as discussed in the docstring</tt> </tt>
<a name="L470"></a><tt class="py-lineno"> 470</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">-</tt><tt class="py-number">5.0</tt> </tt>
<a name="L471"></a><tt class="py-lineno"> 471</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L472"></a><tt class="py-lineno"> 472</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"PHRED qualities must be positive (or zero), not %s"</tt> </tt>
<a name="L473"></a><tt class="py-lineno"> 473</tt>  <tt class="py-line">                         <tt class="py-op">%</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">phred_quality</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L474"></a><tt class="py-lineno"> 474</tt>  <tt class="py-line"> </tt>
<a name="L475"></a><tt class="py-lineno"> 475</tt>  <tt class="py-line"> </tt>
<a name="phred_quality_from_solexa"></a><div id="phred_quality_from_solexa-def"><a name="L476"></a><tt class="py-lineno"> 476</tt> <a class="py-toggle" href="#" id="phred_quality_from_solexa-toggle" onclick="return toggle('phred_quality_from_solexa');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.QualityIO-module.html#phred_quality_from_solexa">phred_quality_from_solexa</a><tt class="py-op">(</tt><tt class="py-param">solexa_quality</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="phred_quality_from_solexa-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="phred_quality_from_solexa-expanded"><a name="L477"></a><tt class="py-lineno"> 477</tt>  <tt class="py-line">    <tt class="py-docstring">"""Convert a Solexa quality (which can be negative) to a PHRED quality.</tt> </tt>
<a name="L478"></a><tt class="py-lineno"> 478</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L479"></a><tt class="py-lineno"> 479</tt>  <tt class="py-line"><tt class="py-docstring">    PHRED and Solexa quality scores are both log transformations of a</tt> </tt>
<a name="L480"></a><tt class="py-lineno"> 480</tt>  <tt class="py-line"><tt class="py-docstring">    probality of error (high score = low probability of error). This function</tt> </tt>
<a name="L481"></a><tt class="py-lineno"> 481</tt>  <tt class="py-line"><tt class="py-docstring">    takes a Solexa score, transforms it back to a probability of error, and</tt> </tt>
<a name="L482"></a><tt class="py-lineno"> 482</tt>  <tt class="py-line"><tt class="py-docstring">    then re-expresses it as a PHRED score. This assumes the error estimates</tt> </tt>
<a name="L483"></a><tt class="py-lineno"> 483</tt>  <tt class="py-line"><tt class="py-docstring">    are equivalent.</tt> </tt>
<a name="L484"></a><tt class="py-lineno"> 484</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L485"></a><tt class="py-lineno"> 485</tt>  <tt class="py-line"><tt class="py-docstring">    The underlying formulas are given in the documentation for the sister</tt> </tt>
<a name="L486"></a><tt class="py-lineno"> 486</tt>  <tt class="py-line"><tt class="py-docstring">    function solexa_quality_from_phred, in this case the operation is::</tt> </tt>
<a name="L487"></a><tt class="py-lineno"> 487</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L488"></a><tt class="py-lineno"> 488</tt>  <tt class="py-line"><tt class="py-docstring">        phred_quality = 10*log(10**(solexa_quality/10.0) + 1, 10)</tt> </tt>
<a name="L489"></a><tt class="py-lineno"> 489</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L490"></a><tt class="py-lineno"> 490</tt>  <tt class="py-line"><tt class="py-docstring">    This will return a floating point number, it is up to you to round this to</tt> </tt>
<a name="L491"></a><tt class="py-lineno"> 491</tt>  <tt class="py-line"><tt class="py-docstring">    the nearest integer if appropriate.  e.g.</tt> </tt>
<a name="L492"></a><tt class="py-lineno"> 492</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L493"></a><tt class="py-lineno"> 493</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%0.2f" % round(phred_quality_from_solexa(80), 2))</tt> </tt>
<a name="L494"></a><tt class="py-lineno"> 494</tt>  <tt class="py-line"><tt class="py-docstring">    80.00</tt> </tt>
<a name="L495"></a><tt class="py-lineno"> 495</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%0.2f" % round(phred_quality_from_solexa(20), 2))</tt> </tt>
<a name="L496"></a><tt class="py-lineno"> 496</tt>  <tt class="py-line"><tt class="py-docstring">    20.04</tt> </tt>
<a name="L497"></a><tt class="py-lineno"> 497</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%0.2f" % round(phred_quality_from_solexa(10), 2))</tt> </tt>
<a name="L498"></a><tt class="py-lineno"> 498</tt>  <tt class="py-line"><tt class="py-docstring">    10.41</tt> </tt>
<a name="L499"></a><tt class="py-lineno"> 499</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%0.2f" % round(phred_quality_from_solexa(0), 2))</tt> </tt>
<a name="L500"></a><tt class="py-lineno"> 500</tt>  <tt class="py-line"><tt class="py-docstring">    3.01</tt> </tt>
<a name="L501"></a><tt class="py-lineno"> 501</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%0.2f" % round(phred_quality_from_solexa(-5), 2))</tt> </tt>
<a name="L502"></a><tt class="py-lineno"> 502</tt>  <tt class="py-line"><tt class="py-docstring">    1.19</tt> </tt>
<a name="L503"></a><tt class="py-lineno"> 503</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L504"></a><tt class="py-lineno"> 504</tt>  <tt class="py-line"><tt class="py-docstring">    Note that a solexa_quality less then -5 is not expected, will trigger a</tt> </tt>
<a name="L505"></a><tt class="py-lineno"> 505</tt>  <tt class="py-line"><tt class="py-docstring">    warning, but will still be converted as per the logarithmic mapping</tt> </tt>
<a name="L506"></a><tt class="py-lineno"> 506</tt>  <tt class="py-line"><tt class="py-docstring">    (giving a number between 0 and 1.19 back).</tt> </tt>
<a name="L507"></a><tt class="py-lineno"> 507</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L508"></a><tt class="py-lineno"> 508</tt>  <tt class="py-line"><tt class="py-docstring">    As a special case where None is used for a "missing value", None is</tt> </tt>
<a name="L509"></a><tt class="py-lineno"> 509</tt>  <tt class="py-line"><tt class="py-docstring">    returned:</tt> </tt>
<a name="L510"></a><tt class="py-lineno"> 510</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L511"></a><tt class="py-lineno"> 511</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(phred_quality_from_solexa(None))</tt> </tt>
<a name="L512"></a><tt class="py-lineno"> 512</tt>  <tt class="py-line"><tt class="py-docstring">    None</tt> </tt>
<a name="L513"></a><tt class="py-lineno"> 513</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L514"></a><tt class="py-lineno"> 514</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">solexa_quality</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L515"></a><tt class="py-lineno"> 515</tt>  <tt class="py-line">        <tt class="py-comment"># Assume None is used as some kind of NULL or NA value; return None</tt> </tt>
<a name="L516"></a><tt class="py-lineno"> 516</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
<a name="L517"></a><tt class="py-lineno"> 517</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">solexa_quality</tt> <tt class="py-op">&lt;</tt> <tt class="py-op">-</tt><tt class="py-number">5</tt><tt class="py-op">:</tt> </tt>
<a name="L518"></a><tt class="py-lineno"> 518</tt>  <tt class="py-line">        <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Solexa quality less than -5 passed, %s"</tt> </tt>
<a name="L519"></a><tt class="py-lineno"> 519</tt>  <tt class="py-line">                      <tt class="py-op">%</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">solexa_quality</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-20" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-20', 'BiopythonWarning', 'link-15');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L520"></a><tt class="py-lineno"> 520</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-number">10</tt> <tt class="py-op">*</tt> <tt class="py-name">log</tt><tt class="py-op">(</tt><tt class="py-number">10</tt> <tt class="py-op">**</tt> <tt class="py-op">(</tt><tt class="py-name">solexa_quality</tt> <tt class="py-op">/</tt> <tt class="py-number">10.0</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-number">10</tt><tt class="py-op">)</tt> </tt>
</div><a name="L521"></a><tt class="py-lineno"> 521</tt>  <tt class="py-line"> </tt>
<a name="L522"></a><tt class="py-lineno"> 522</tt>  <tt class="py-line"> </tt>
<a name="_get_phred_quality"></a><div id="_get_phred_quality-def"><a name="L523"></a><tt class="py-lineno"> 523</tt> <a class="py-toggle" href="#" id="_get_phred_quality-toggle" onclick="return toggle('_get_phred_quality');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.QualityIO-module.html#_get_phred_quality">_get_phred_quality</a><tt class="py-op">(</tt><tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_get_phred_quality-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_get_phred_quality-expanded"><a name="L524"></a><tt class="py-lineno"> 524</tt>  <tt class="py-line">    <tt class="py-docstring">"""Extract PHRED qualities from a SeqRecord's letter_annotations (PRIVATE).</tt> </tt>
<a name="L525"></a><tt class="py-lineno"> 525</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L526"></a><tt class="py-lineno"> 526</tt>  <tt class="py-line"><tt class="py-docstring">    If there are no PHRED qualities, but there are Solexa qualities, those are</tt> </tt>
<a name="L527"></a><tt class="py-lineno"> 527</tt>  <tt class="py-line"><tt class="py-docstring">    used instead after conversion.</tt> </tt>
<a name="L528"></a><tt class="py-lineno"> 528</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L529"></a><tt class="py-lineno"> 529</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L530"></a><tt class="py-lineno"> 530</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-21" class="py-name" targets="Variable Bio.SeqRecord.SeqRecord.letter_annotations=Bio.SeqRecord.SeqRecord-class.html#letter_annotations"><a title="Bio.SeqRecord.SeqRecord.letter_annotations" class="py-name" href="#" onclick="return doclink('link-21', 'letter_annotations', 'link-21');">letter_annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"phred_quality"</tt><tt class="py-op">]</tt> </tt>
<a name="L531"></a><tt class="py-lineno"> 531</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L532"></a><tt class="py-lineno"> 532</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
<a name="L533"></a><tt class="py-lineno"> 533</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L534"></a><tt class="py-lineno"> 534</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt id="link-22" class="py-name" targets="Function Bio.SeqIO.QualityIO.phred_quality_from_solexa()=Bio.SeqIO.QualityIO-module.html#phred_quality_from_solexa"><a title="Bio.SeqIO.QualityIO.phred_quality_from_solexa" class="py-name" href="#" onclick="return doclink('link-22', 'phred_quality_from_solexa', 'link-22');">phred_quality_from_solexa</a></tt><tt class="py-op">(</tt><tt class="py-name">q</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> </tt>
<a name="L535"></a><tt class="py-lineno"> 535</tt>  <tt class="py-line">                <tt class="py-name">q</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-23" class="py-name"><a title="Bio.SeqRecord.SeqRecord.letter_annotations" class="py-name" href="#" onclick="return doclink('link-23', 'letter_annotations', 'link-21');">letter_annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"solexa_quality"</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L536"></a><tt class="py-lineno"> 536</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L537"></a><tt class="py-lineno"> 537</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"No suitable quality scores found in "</tt> </tt>
<a name="L538"></a><tt class="py-lineno"> 538</tt>  <tt class="py-line">                         <tt class="py-string">"letter_annotations of SeqRecord (id=%s)."</tt> </tt>
<a name="L539"></a><tt class="py-lineno"> 539</tt>  <tt class="py-line">                         <tt class="py-op">%</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-24" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.id=Bio.KEGG.KGML.KGML_pathway.Component-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Entry.id=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.id=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#id,Method Bio.Phylo.PhyloXMLIO.Parser.id()=Bio.Phylo.PhyloXMLIO.Parser-class.html#id,Method Bio.Phylo.PhyloXMLIO.Writer.id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#id,Variable Bio.SearchIO._model.hit.Hit.id=Bio.SearchIO._model.hit.Hit-class.html#id,Variable Bio.SearchIO._model.query.QueryResult.id=Bio.SearchIO._model.query.QueryResult-class.html#id"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-24', 'id', 'link-24');">id</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L540"></a><tt class="py-lineno"> 540</tt>  <tt class="py-line"> </tt>
<a name="L541"></a><tt class="py-lineno"> 541</tt>  <tt class="py-line"><tt class="py-comment"># Only map 0 to 93, we need to give a warning on truncating at 93</tt> </tt>
<a name="L542"></a><tt class="py-lineno"> 542</tt>  <tt class="py-line"><tt id="link-25" class="py-name" targets="Variable Bio.SeqIO.QualityIO._phred_to_sanger_quality_str=Bio.SeqIO.QualityIO-module.html#_phred_to_sanger_quality_str"><a title="Bio.SeqIO.QualityIO._phred_to_sanger_quality_str" class="py-name" href="#" onclick="return doclink('link-25', '_phred_to_sanger_quality_str', 'link-25');">_phred_to_sanger_quality_str</a></tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">qp</tt><tt class="py-op">,</tt> <tt class="py-name">chr</tt><tt class="py-op">(</tt><tt id="link-26" class="py-name" targets="Variable Bio.motifs.matrix.PositionSpecificScoringMatrix.min=Bio.motifs.matrix.PositionSpecificScoringMatrix-class.html#min"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-26', 'min', 'link-26');">min</a></tt><tt class="py-op">(</tt><tt class="py-number">126</tt><tt class="py-op">,</tt> <tt class="py-name">qp</tt> <tt class="py-op">+</tt> <tt id="link-27" class="py-name"><a title="Bio.SeqIO.QualityIO.SANGER_SCORE_OFFSET" class="py-name" href="#" onclick="return doclink('link-27', 'SANGER_SCORE_OFFSET', 'link-17');">SANGER_SCORE_OFFSET</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L543"></a><tt class="py-lineno"> 543</tt>  <tt class="py-line">                                    <tt class="py-keyword">for</tt> <tt class="py-name">qp</tt> <tt class="py-keyword">in</tt> <tt id="link-28" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.range()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#range,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.range()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#range,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Track.Track.range()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#range"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-28', 'range', 'link-28');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">93</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L544"></a><tt class="py-lineno"> 544</tt>  <tt class="py-line"><tt class="py-comment"># Only map -5 to 93, we need to give a warning on truncating at 93</tt> </tt>
<a name="L545"></a><tt class="py-lineno"> 545</tt>  <tt class="py-line"><tt id="link-29" class="py-name" targets="Variable Bio.SeqIO.QualityIO._solexa_to_sanger_quality_str=Bio.SeqIO.QualityIO-module.html#_solexa_to_sanger_quality_str"><a title="Bio.SeqIO.QualityIO._solexa_to_sanger_quality_str" class="py-name" href="#" onclick="return doclink('link-29', '_solexa_to_sanger_quality_str', 'link-29');">_solexa_to_sanger_quality_str</a></tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt> </tt>
<a name="L546"></a><tt class="py-lineno"> 546</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-name">qs</tt><tt class="py-op">,</tt> <tt class="py-name">chr</tt><tt class="py-op">(</tt><tt id="link-30" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-30', 'min', 'link-26');">min</a></tt><tt class="py-op">(</tt><tt class="py-number">126</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-31" class="py-name" targets="Method Bio.Blast.NCBIStandalone._BlastConsumer.round()=Bio.Blast.NCBIStandalone._BlastConsumer-class.html#round,Method Bio.Blast.NCBIStandalone._DescriptionConsumer.round()=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#round"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.round
Bio.Blast.NCBIStandalone._DescriptionConsumer.round" class="py-name" href="#" onclick="return doclink('link-31', 'round', 'link-31');">round</a></tt><tt class="py-op">(</tt><tt id="link-32" class="py-name"><a title="Bio.SeqIO.QualityIO.phred_quality_from_solexa" class="py-name" href="#" onclick="return doclink('link-32', 'phred_quality_from_solexa', 'link-22');">phred_quality_from_solexa</a></tt><tt class="py-op">(</tt><tt class="py-name">qs</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> </tt>
<a name="L547"></a><tt class="py-lineno"> 547</tt>  <tt class="py-line">     <tt id="link-33" class="py-name"><a title="Bio.SeqIO.QualityIO.SANGER_SCORE_OFFSET" class="py-name" href="#" onclick="return doclink('link-33', 'SANGER_SCORE_OFFSET', 'link-17');">SANGER_SCORE_OFFSET</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L548"></a><tt class="py-lineno"> 548</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">qs</tt> <tt class="py-keyword">in</tt> <tt id="link-34" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-34', 'range', 'link-28');">range</a></tt><tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">5</tt><tt class="py-op">,</tt> <tt class="py-number">93</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L549"></a><tt class="py-lineno"> 549</tt>  <tt class="py-line"> </tt>
<a name="L550"></a><tt class="py-lineno"> 550</tt>  <tt class="py-line"> </tt>
<a name="_get_sanger_quality_str"></a><div id="_get_sanger_quality_str-def"><a name="L551"></a><tt class="py-lineno"> 551</tt> <a class="py-toggle" href="#" id="_get_sanger_quality_str-toggle" onclick="return toggle('_get_sanger_quality_str');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.QualityIO-module.html#_get_sanger_quality_str">_get_sanger_quality_str</a><tt class="py-op">(</tt><tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_get_sanger_quality_str-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_get_sanger_quality_str-expanded"><a name="L552"></a><tt class="py-lineno"> 552</tt>  <tt class="py-line">    <tt class="py-docstring">"""Returns a Sanger FASTQ encoded quality string (PRIVATE).</tt> </tt>
<a name="L553"></a><tt class="py-lineno"> 553</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L554"></a><tt class="py-lineno"> 554</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio.Seq import Seq</tt> </tt>
<a name="L555"></a><tt class="py-lineno"> 555</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio.SeqRecord import SeqRecord</tt> </tt>
<a name="L556"></a><tt class="py-lineno"> 556</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; r = SeqRecord(Seq("ACGTAN"), id="Test",</tt> </tt>
<a name="L557"></a><tt class="py-lineno"> 557</tt>  <tt class="py-line"><tt class="py-docstring">    ...               letter_annotations = {"phred_quality":[50, 40, 30, 20, 10, 0]})</tt> </tt>
<a name="L558"></a><tt class="py-lineno"> 558</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _get_sanger_quality_str(r)</tt> </tt>
<a name="L559"></a><tt class="py-lineno"> 559</tt>  <tt class="py-line"><tt class="py-docstring">    'SI?5+!'</tt> </tt>
<a name="L560"></a><tt class="py-lineno"> 560</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L561"></a><tt class="py-lineno"> 561</tt>  <tt class="py-line"><tt class="py-docstring">    If as in the above example (or indeed a SeqRecord parser with Bio.SeqIO),</tt> </tt>
<a name="L562"></a><tt class="py-lineno"> 562</tt>  <tt class="py-line"><tt class="py-docstring">    the PHRED qualities are integers, this function is able to use a very fast</tt> </tt>
<a name="L563"></a><tt class="py-lineno"> 563</tt>  <tt class="py-line"><tt class="py-docstring">    pre-cached mapping. However, if they are floats which differ slightly, then</tt> </tt>
<a name="L564"></a><tt class="py-lineno"> 564</tt>  <tt class="py-line"><tt class="py-docstring">    it has to do the appropriate rounding - which is slower:</tt> </tt>
<a name="L565"></a><tt class="py-lineno"> 565</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L566"></a><tt class="py-lineno"> 566</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; r2 = SeqRecord(Seq("ACGTAN"), id="Test2",</tt> </tt>
<a name="L567"></a><tt class="py-lineno"> 567</tt>  <tt class="py-line"><tt class="py-docstring">    ...      letter_annotations = {"phred_quality":[50.0, 40.05, 29.99, 20, 9.55, 0.01]})</tt> </tt>
<a name="L568"></a><tt class="py-lineno"> 568</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _get_sanger_quality_str(r2)</tt> </tt>
<a name="L569"></a><tt class="py-lineno"> 569</tt>  <tt class="py-line"><tt class="py-docstring">    'SI?5+!'</tt> </tt>
<a name="L570"></a><tt class="py-lineno"> 570</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L571"></a><tt class="py-lineno"> 571</tt>  <tt class="py-line"><tt class="py-docstring">    If your scores include a None value, this raises an exception:</tt> </tt>
<a name="L572"></a><tt class="py-lineno"> 572</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L573"></a><tt class="py-lineno"> 573</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; r3 = SeqRecord(Seq("ACGTAN"), id="Test3",</tt> </tt>
<a name="L574"></a><tt class="py-lineno"> 574</tt>  <tt class="py-line"><tt class="py-docstring">    ...               letter_annotations = {"phred_quality":[50, 40, 30, 20, 10, None]})</tt> </tt>
<a name="L575"></a><tt class="py-lineno"> 575</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _get_sanger_quality_str(r3)</tt> </tt>
<a name="L576"></a><tt class="py-lineno"> 576</tt>  <tt class="py-line"><tt class="py-docstring">    Traceback (most recent call last):</tt> </tt>
<a name="L577"></a><tt class="py-lineno"> 577</tt>  <tt class="py-line"><tt class="py-docstring">       ...</tt> </tt>
<a name="L578"></a><tt class="py-lineno"> 578</tt>  <tt class="py-line"><tt class="py-docstring">    TypeError: A quality value of None was found</tt> </tt>
<a name="L579"></a><tt class="py-lineno"> 579</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L580"></a><tt class="py-lineno"> 580</tt>  <tt class="py-line"><tt class="py-docstring">    If (strangely) your record has both PHRED and Solexa scores, then the PHRED</tt> </tt>
<a name="L581"></a><tt class="py-lineno"> 581</tt>  <tt class="py-line"><tt class="py-docstring">    scores are used in preference:</tt> </tt>
<a name="L582"></a><tt class="py-lineno"> 582</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L583"></a><tt class="py-lineno"> 583</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; r4 = SeqRecord(Seq("ACGTAN"), id="Test4",</tt> </tt>
<a name="L584"></a><tt class="py-lineno"> 584</tt>  <tt class="py-line"><tt class="py-docstring">    ...               letter_annotations = {"phred_quality":[50, 40, 30, 20, 10, 0],</tt> </tt>
<a name="L585"></a><tt class="py-lineno"> 585</tt>  <tt class="py-line"><tt class="py-docstring">    ...                                     "solexa_quality":[-5, -4, 0, None, 0, 40]})</tt> </tt>
<a name="L586"></a><tt class="py-lineno"> 586</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _get_sanger_quality_str(r4)</tt> </tt>
<a name="L587"></a><tt class="py-lineno"> 587</tt>  <tt class="py-line"><tt class="py-docstring">    'SI?5+!'</tt> </tt>
<a name="L588"></a><tt class="py-lineno"> 588</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L589"></a><tt class="py-lineno"> 589</tt>  <tt class="py-line"><tt class="py-docstring">    If there are no PHRED scores, but there are Solexa scores, these are used</tt> </tt>
<a name="L590"></a><tt class="py-lineno"> 590</tt>  <tt class="py-line"><tt class="py-docstring">    instead (after the appropriate conversion):</tt> </tt>
<a name="L591"></a><tt class="py-lineno"> 591</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L592"></a><tt class="py-lineno"> 592</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; r5 = SeqRecord(Seq("ACGTAN"), id="Test5",</tt> </tt>
<a name="L593"></a><tt class="py-lineno"> 593</tt>  <tt class="py-line"><tt class="py-docstring">    ...      letter_annotations = {"solexa_quality":[40, 30, 20, 10, 0, -5]})</tt> </tt>
<a name="L594"></a><tt class="py-lineno"> 594</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _get_sanger_quality_str(r5)</tt> </tt>
<a name="L595"></a><tt class="py-lineno"> 595</tt>  <tt class="py-line"><tt class="py-docstring">    'I?5+$"'</tt> </tt>
<a name="L596"></a><tt class="py-lineno"> 596</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L597"></a><tt class="py-lineno"> 597</tt>  <tt class="py-line"><tt class="py-docstring">    Again, integer Solexa scores can be looked up in a pre-cached mapping making</tt> </tt>
<a name="L598"></a><tt class="py-lineno"> 598</tt>  <tt class="py-line"><tt class="py-docstring">    this very fast. You can still use approximate floating point scores:</tt> </tt>
<a name="L599"></a><tt class="py-lineno"> 599</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L600"></a><tt class="py-lineno"> 600</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; r6 = SeqRecord(Seq("ACGTAN"), id="Test6",</tt> </tt>
<a name="L601"></a><tt class="py-lineno"> 601</tt>  <tt class="py-line"><tt class="py-docstring">    ...      letter_annotations = {"solexa_quality":[40.1, 29.7, 20.01, 10, 0.0, -4.9]})</tt> </tt>
<a name="L602"></a><tt class="py-lineno"> 602</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; _get_sanger_quality_str(r6)</tt> </tt>
<a name="L603"></a><tt class="py-lineno"> 603</tt>  <tt class="py-line"><tt class="py-docstring">    'I?5+$"'</tt> </tt>
<a name="L604"></a><tt class="py-lineno"> 604</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L605"></a><tt class="py-lineno"> 605</tt>  <tt class="py-line"><tt class="py-docstring">    Notice that due to the limited range of printable ASCII characters, a</tt> </tt>
<a name="L606"></a><tt class="py-lineno"> 606</tt>  <tt class="py-line"><tt class="py-docstring">    PHRED quality of 93 is the maximum that can be held in an Illumina FASTQ</tt> </tt>
<a name="L607"></a><tt class="py-lineno"> 607</tt>  <tt class="py-line"><tt class="py-docstring">    file (using ASCII 126, the tilde). This function will issue a warning</tt> </tt>
<a name="L608"></a><tt class="py-lineno"> 608</tt>  <tt class="py-line"><tt class="py-docstring">    in this situation.</tt> </tt>
<a name="L609"></a><tt class="py-lineno"> 609</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L610"></a><tt class="py-lineno"> 610</tt>  <tt class="py-line">    <tt class="py-comment"># TODO - This functions works and is fast, but it is also ugly</tt> </tt>
<a name="L611"></a><tt class="py-lineno"> 611</tt>  <tt class="py-line">    <tt class="py-comment"># and there is considerable repetition of code for the other</tt> </tt>
<a name="L612"></a><tt class="py-lineno"> 612</tt>  <tt class="py-line">    <tt class="py-comment"># two FASTQ variants.</tt> </tt>
<a name="L613"></a><tt class="py-lineno"> 613</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L614"></a><tt class="py-lineno"> 614</tt>  <tt class="py-line">        <tt class="py-comment"># These take priority (in case both Solexa and PHRED scores found)</tt> </tt>
<a name="L615"></a><tt class="py-lineno"> 615</tt>  <tt class="py-line">        <tt class="py-name">qualities</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-35" class="py-name"><a title="Bio.SeqRecord.SeqRecord.letter_annotations" class="py-name" href="#" onclick="return doclink('link-35', 'letter_annotations', 'link-21');">letter_annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"phred_quality"</tt><tt class="py-op">]</tt> </tt>
<a name="L616"></a><tt class="py-lineno"> 616</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L617"></a><tt class="py-lineno"> 617</tt>  <tt class="py-line">        <tt class="py-comment"># Fall back on solexa scores...</tt> </tt>
<a name="L618"></a><tt class="py-lineno"> 618</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
<a name="L619"></a><tt class="py-lineno"> 619</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L620"></a><tt class="py-lineno"> 620</tt>  <tt class="py-line">        <tt class="py-comment"># Try and use the precomputed mapping:</tt> </tt>
<a name="L621"></a><tt class="py-lineno"> 621</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L622"></a><tt class="py-lineno"> 622</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt id="link-36" class="py-name"><a title="Bio.SeqIO.QualityIO._phred_to_sanger_quality_str" class="py-name" href="#" onclick="return doclink('link-36', '_phred_to_sanger_quality_str', 'link-25');">_phred_to_sanger_quality_str</a></tt><tt class="py-op">[</tt><tt class="py-name">qp</tt><tt class="py-op">]</tt> </tt>
<a name="L623"></a><tt class="py-lineno"> 623</tt>  <tt class="py-line">                           <tt class="py-keyword">for</tt> <tt class="py-name">qp</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qualities</tt><tt class="py-op">)</tt> </tt>
<a name="L624"></a><tt class="py-lineno"> 624</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L625"></a><tt class="py-lineno"> 625</tt>  <tt class="py-line">            <tt class="py-comment"># Could be a float, or a None in the list, or a high value.</tt> </tt>
<a name="L626"></a><tt class="py-lineno"> 626</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
<a name="L627"></a><tt class="py-lineno"> 627</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">None</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qualities</tt><tt class="py-op">:</tt> </tt>
<a name="L628"></a><tt class="py-lineno"> 628</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"A quality value of None was found"</tt><tt class="py-op">)</tt> </tt>
<a name="L629"></a><tt class="py-lineno"> 629</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-37" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-37', 'max', 'link-19');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">qualities</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">93.5</tt><tt class="py-op">:</tt> </tt>
<a name="L630"></a><tt class="py-lineno"> 630</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Data loss - max PHRED quality 93 in Sanger FASTQ"</tt><tt class="py-op">,</tt> </tt>
<a name="L631"></a><tt class="py-lineno"> 631</tt>  <tt class="py-line">                          <tt id="link-38" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-38', 'BiopythonWarning', 'link-15');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L632"></a><tt class="py-lineno"> 632</tt>  <tt class="py-line">        <tt class="py-comment"># This will apply the truncation at 93, giving max ASCII 126</tt> </tt>
<a name="L633"></a><tt class="py-lineno"> 633</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt id="link-39" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-39', 'min', 'link-26');">min</a></tt><tt class="py-op">(</tt><tt class="py-number">126</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-40" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.round
Bio.Blast.NCBIStandalone._DescriptionConsumer.round" class="py-name" href="#" onclick="return doclink('link-40', 'round', 'link-31');">round</a></tt><tt class="py-op">(</tt><tt class="py-name">qp</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt id="link-41" class="py-name"><a title="Bio.SeqIO.QualityIO.SANGER_SCORE_OFFSET" class="py-name" href="#" onclick="return doclink('link-41', 'SANGER_SCORE_OFFSET', 'link-17');">SANGER_SCORE_OFFSET</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L634"></a><tt class="py-lineno"> 634</tt>  <tt class="py-line">                       <tt class="py-keyword">for</tt> <tt class="py-name">qp</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qualities</tt><tt class="py-op">)</tt> </tt>
<a name="L635"></a><tt class="py-lineno"> 635</tt>  <tt class="py-line">    <tt class="py-comment"># Fall back on the Solexa scores...</tt> </tt>
<a name="L636"></a><tt class="py-lineno"> 636</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L637"></a><tt class="py-lineno"> 637</tt>  <tt class="py-line">        <tt class="py-name">qualities</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-42" class="py-name"><a title="Bio.SeqRecord.SeqRecord.letter_annotations" class="py-name" href="#" onclick="return doclink('link-42', 'letter_annotations', 'link-21');">letter_annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"solexa_quality"</tt><tt class="py-op">]</tt> </tt>
<a name="L638"></a><tt class="py-lineno"> 638</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L639"></a><tt class="py-lineno"> 639</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"No suitable quality scores found in "</tt> </tt>
<a name="L640"></a><tt class="py-lineno"> 640</tt>  <tt class="py-line">                         <tt class="py-string">"letter_annotations of SeqRecord (id=%s)."</tt> </tt>
<a name="L641"></a><tt class="py-lineno"> 641</tt>  <tt class="py-line">                         <tt class="py-op">%</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-43" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-43', 'id', 'link-24');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L642"></a><tt class="py-lineno"> 642</tt>  <tt class="py-line">    <tt class="py-comment"># Try and use the precomputed mapping:</tt> </tt>
<a name="L643"></a><tt class="py-lineno"> 643</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L644"></a><tt class="py-lineno"> 644</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt id="link-44" class="py-name"><a title="Bio.SeqIO.QualityIO._solexa_to_sanger_quality_str" class="py-name" href="#" onclick="return doclink('link-44', '_solexa_to_sanger_quality_str', 'link-29');">_solexa_to_sanger_quality_str</a></tt><tt class="py-op">[</tt><tt class="py-name">qs</tt><tt class="py-op">]</tt> </tt>
<a name="L645"></a><tt class="py-lineno"> 645</tt>  <tt class="py-line">                       <tt class="py-keyword">for</tt> <tt class="py-name">qs</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qualities</tt><tt class="py-op">)</tt> </tt>
<a name="L646"></a><tt class="py-lineno"> 646</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L647"></a><tt class="py-lineno"> 647</tt>  <tt class="py-line">        <tt class="py-comment"># Either no PHRED scores, or something odd like a float or None</tt> </tt>
<a name="L648"></a><tt class="py-lineno"> 648</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
<a name="L649"></a><tt class="py-lineno"> 649</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">None</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qualities</tt><tt class="py-op">:</tt> </tt>
<a name="L650"></a><tt class="py-lineno"> 650</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"A quality value of None was found"</tt><tt class="py-op">)</tt> </tt>
<a name="L651"></a><tt class="py-lineno"> 651</tt>  <tt class="py-line">    <tt class="py-comment"># Must do this the slow way, first converting the PHRED scores into</tt> </tt>
<a name="L652"></a><tt class="py-lineno"> 652</tt>  <tt class="py-line">    <tt class="py-comment"># Solexa scores:</tt> </tt>
<a name="L653"></a><tt class="py-lineno"> 653</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt id="link-45" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-45', 'max', 'link-19');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">qualities</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">93.5</tt><tt class="py-op">:</tt> </tt>
<a name="L654"></a><tt class="py-lineno"> 654</tt>  <tt class="py-line">        <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Data loss - max PHRED quality 93 in Sanger FASTQ"</tt><tt class="py-op">,</tt> </tt>
<a name="L655"></a><tt class="py-lineno"> 655</tt>  <tt class="py-line">                      <tt id="link-46" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-46', 'BiopythonWarning', 'link-15');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L656"></a><tt class="py-lineno"> 656</tt>  <tt class="py-line">    <tt class="py-comment"># This will apply the truncation at 93, giving max ASCII 126</tt> </tt>
<a name="L657"></a><tt class="py-lineno"> 657</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt id="link-47" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-47', 'min', 'link-26');">min</a></tt><tt class="py-op">(</tt><tt class="py-number">126</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-48" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.round
Bio.Blast.NCBIStandalone._DescriptionConsumer.round" class="py-name" href="#" onclick="return doclink('link-48', 'round', 'link-31');">round</a></tt><tt class="py-op">(</tt><tt id="link-49" class="py-name"><a title="Bio.SeqIO.QualityIO.phred_quality_from_solexa" class="py-name" href="#" onclick="return doclink('link-49', 'phred_quality_from_solexa', 'link-22');">phred_quality_from_solexa</a></tt><tt class="py-op">(</tt><tt class="py-name">qs</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt id="link-50" class="py-name"><a title="Bio.SeqIO.QualityIO.SANGER_SCORE_OFFSET" class="py-name" href="#" onclick="return doclink('link-50', 'SANGER_SCORE_OFFSET', 'link-17');">SANGER_SCORE_OFFSET</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L658"></a><tt class="py-lineno"> 658</tt>  <tt class="py-line">                   <tt class="py-keyword">for</tt> <tt class="py-name">qs</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qualities</tt><tt class="py-op">)</tt> </tt>
</div><a name="L659"></a><tt class="py-lineno"> 659</tt>  <tt class="py-line"> </tt>
<a name="L660"></a><tt class="py-lineno"> 660</tt>  <tt class="py-line"><tt class="py-comment"># Only map 0 to 62, we need to give a warning on truncating at 62</tt> </tt>
<a name="L661"></a><tt class="py-lineno"> 661</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-number">62</tt> <tt class="py-op">+</tt> <tt id="link-51" class="py-name"><a title="Bio.SeqIO.QualityIO.SOLEXA_SCORE_OFFSET" class="py-name" href="#" onclick="return doclink('link-51', 'SOLEXA_SCORE_OFFSET', 'link-18');">SOLEXA_SCORE_OFFSET</a></tt> <tt class="py-op">==</tt> <tt class="py-number">126</tt> </tt>
<a name="L662"></a><tt class="py-lineno"> 662</tt>  <tt class="py-line"><tt id="link-52" class="py-name" targets="Variable Bio.SeqIO.QualityIO._phred_to_illumina_quality_str=Bio.SeqIO.QualityIO-module.html#_phred_to_illumina_quality_str"><a title="Bio.SeqIO.QualityIO._phred_to_illumina_quality_str" class="py-name" href="#" onclick="return doclink('link-52', '_phred_to_illumina_quality_str', 'link-52');">_phred_to_illumina_quality_str</a></tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">qp</tt><tt class="py-op">,</tt> <tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-name">qp</tt> <tt class="py-op">+</tt> <tt id="link-53" class="py-name"><a title="Bio.SeqIO.QualityIO.SOLEXA_SCORE_OFFSET" class="py-name" href="#" onclick="return doclink('link-53', 'SOLEXA_SCORE_OFFSET', 'link-18');">SOLEXA_SCORE_OFFSET</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L663"></a><tt class="py-lineno"> 663</tt>  <tt class="py-line">                                      <tt class="py-keyword">for</tt> <tt class="py-name">qp</tt> <tt class="py-keyword">in</tt> <tt id="link-54" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-54', 'range', 'link-28');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">62</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L664"></a><tt class="py-lineno"> 664</tt>  <tt class="py-line"><tt class="py-comment"># Only map -5 to 62, we need to give a warning on truncating at 62</tt> </tt>
<a name="L665"></a><tt class="py-lineno"> 665</tt>  <tt class="py-line"><tt id="link-55" class="py-name" targets="Variable Bio.SeqIO.QualityIO._solexa_to_illumina_quality_str=Bio.SeqIO.QualityIO-module.html#_solexa_to_illumina_quality_str"><a title="Bio.SeqIO.QualityIO._solexa_to_illumina_quality_str" class="py-name" href="#" onclick="return doclink('link-55', '_solexa_to_illumina_quality_str', 'link-55');">_solexa_to_illumina_quality_str</a></tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt> </tt>
<a name="L666"></a><tt class="py-lineno"> 666</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-name">qs</tt><tt class="py-op">,</tt> <tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-56" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.round
Bio.Blast.NCBIStandalone._DescriptionConsumer.round" class="py-name" href="#" onclick="return doclink('link-56', 'round', 'link-31');">round</a></tt><tt class="py-op">(</tt><tt id="link-57" class="py-name"><a title="Bio.SeqIO.QualityIO.phred_quality_from_solexa" class="py-name" href="#" onclick="return doclink('link-57', 'phred_quality_from_solexa', 'link-22');">phred_quality_from_solexa</a></tt><tt class="py-op">(</tt><tt class="py-name">qs</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt id="link-58" class="py-name"><a title="Bio.SeqIO.QualityIO.SOLEXA_SCORE_OFFSET" class="py-name" href="#" onclick="return doclink('link-58', 'SOLEXA_SCORE_OFFSET', 'link-18');">SOLEXA_SCORE_OFFSET</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L667"></a><tt class="py-lineno"> 667</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">qs</tt> <tt class="py-keyword">in</tt> <tt id="link-59" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-59', 'range', 'link-28');">range</a></tt><tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">5</tt><tt class="py-op">,</tt> <tt class="py-number">62</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L668"></a><tt class="py-lineno"> 668</tt>  <tt class="py-line"> </tt>
<a name="L669"></a><tt class="py-lineno"> 669</tt>  <tt class="py-line"> </tt>
<a name="_get_illumina_quality_str"></a><div id="_get_illumina_quality_str-def"><a name="L670"></a><tt class="py-lineno"> 670</tt> <a class="py-toggle" href="#" id="_get_illumina_quality_str-toggle" onclick="return toggle('_get_illumina_quality_str');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.QualityIO-module.html#_get_illumina_quality_str">_get_illumina_quality_str</a><tt class="py-op">(</tt><tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_get_illumina_quality_str-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_get_illumina_quality_str-expanded"><a name="L671"></a><tt class="py-lineno"> 671</tt>  <tt class="py-line">    <tt class="py-docstring">"""Returns an Illumina 1.3 to 1.7 FASTQ encoded quality string (PRIVATE).</tt> </tt>
<a name="L672"></a><tt class="py-lineno"> 672</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L673"></a><tt class="py-lineno"> 673</tt>  <tt class="py-line"><tt class="py-docstring">    Notice that due to the limited range of printable ASCII characters, a</tt> </tt>
<a name="L674"></a><tt class="py-lineno"> 674</tt>  <tt class="py-line"><tt class="py-docstring">    PHRED quality of 62 is the maximum that can be held in an Illumina FASTQ</tt> </tt>
<a name="L675"></a><tt class="py-lineno"> 675</tt>  <tt class="py-line"><tt class="py-docstring">    file (using ASCII 126, the tilde). This function will issue a warning</tt> </tt>
<a name="L676"></a><tt class="py-lineno"> 676</tt>  <tt class="py-line"><tt class="py-docstring">    in this situation.</tt> </tt>
<a name="L677"></a><tt class="py-lineno"> 677</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L678"></a><tt class="py-lineno"> 678</tt>  <tt class="py-line">    <tt class="py-comment"># TODO - This functions works and is fast, but it is also ugly</tt> </tt>
<a name="L679"></a><tt class="py-lineno"> 679</tt>  <tt class="py-line">    <tt class="py-comment"># and there is considerable repetition of code for the other</tt> </tt>
<a name="L680"></a><tt class="py-lineno"> 680</tt>  <tt class="py-line">    <tt class="py-comment"># two FASTQ variants.</tt> </tt>
<a name="L681"></a><tt class="py-lineno"> 681</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L682"></a><tt class="py-lineno"> 682</tt>  <tt class="py-line">        <tt class="py-comment"># These take priority (in case both Solexa and PHRED scores found)</tt> </tt>
<a name="L683"></a><tt class="py-lineno"> 683</tt>  <tt class="py-line">        <tt class="py-name">qualities</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-60" class="py-name"><a title="Bio.SeqRecord.SeqRecord.letter_annotations" class="py-name" href="#" onclick="return doclink('link-60', 'letter_annotations', 'link-21');">letter_annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"phred_quality"</tt><tt class="py-op">]</tt> </tt>
<a name="L684"></a><tt class="py-lineno"> 684</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L685"></a><tt class="py-lineno"> 685</tt>  <tt class="py-line">        <tt class="py-comment"># Fall back on solexa scores...</tt> </tt>
<a name="L686"></a><tt class="py-lineno"> 686</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
<a name="L687"></a><tt class="py-lineno"> 687</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L688"></a><tt class="py-lineno"> 688</tt>  <tt class="py-line">        <tt class="py-comment"># Try and use the precomputed mapping:</tt> </tt>
<a name="L689"></a><tt class="py-lineno"> 689</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L690"></a><tt class="py-lineno"> 690</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt id="link-61" class="py-name"><a title="Bio.SeqIO.QualityIO._phred_to_illumina_quality_str" class="py-name" href="#" onclick="return doclink('link-61', '_phred_to_illumina_quality_str', 'link-52');">_phred_to_illumina_quality_str</a></tt><tt class="py-op">[</tt><tt class="py-name">qp</tt><tt class="py-op">]</tt> </tt>
<a name="L691"></a><tt class="py-lineno"> 691</tt>  <tt class="py-line">                           <tt class="py-keyword">for</tt> <tt class="py-name">qp</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qualities</tt><tt class="py-op">)</tt> </tt>
<a name="L692"></a><tt class="py-lineno"> 692</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L693"></a><tt class="py-lineno"> 693</tt>  <tt class="py-line">            <tt class="py-comment"># Could be a float, or a None in the list, or a high value.</tt> </tt>
<a name="L694"></a><tt class="py-lineno"> 694</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
<a name="L695"></a><tt class="py-lineno"> 695</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">None</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qualities</tt><tt class="py-op">:</tt> </tt>
<a name="L696"></a><tt class="py-lineno"> 696</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"A quality value of None was found"</tt><tt class="py-op">)</tt> </tt>
<a name="L697"></a><tt class="py-lineno"> 697</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-62" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-62', 'max', 'link-19');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">qualities</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">62.5</tt><tt class="py-op">:</tt> </tt>
<a name="L698"></a><tt class="py-lineno"> 698</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Data loss - max PHRED quality 62 in Illumina FASTQ"</tt><tt class="py-op">,</tt> </tt>
<a name="L699"></a><tt class="py-lineno"> 699</tt>  <tt class="py-line">                          <tt id="link-63" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-63', 'BiopythonWarning', 'link-15');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L700"></a><tt class="py-lineno"> 700</tt>  <tt class="py-line">        <tt class="py-comment"># This will apply the truncation at 62, giving max ASCII 126</tt> </tt>
<a name="L701"></a><tt class="py-lineno"> 701</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt id="link-64" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-64', 'min', 'link-26');">min</a></tt><tt class="py-op">(</tt><tt class="py-number">126</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-65" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.round
Bio.Blast.NCBIStandalone._DescriptionConsumer.round" class="py-name" href="#" onclick="return doclink('link-65', 'round', 'link-31');">round</a></tt><tt class="py-op">(</tt><tt class="py-name">qp</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt id="link-66" class="py-name"><a title="Bio.SeqIO.QualityIO.SOLEXA_SCORE_OFFSET" class="py-name" href="#" onclick="return doclink('link-66', 'SOLEXA_SCORE_OFFSET', 'link-18');">SOLEXA_SCORE_OFFSET</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L702"></a><tt class="py-lineno"> 702</tt>  <tt class="py-line">                       <tt class="py-keyword">for</tt> <tt class="py-name">qp</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qualities</tt><tt class="py-op">)</tt> </tt>
<a name="L703"></a><tt class="py-lineno"> 703</tt>  <tt class="py-line">    <tt class="py-comment"># Fall back on the Solexa scores...</tt> </tt>
<a name="L704"></a><tt class="py-lineno"> 704</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L705"></a><tt class="py-lineno"> 705</tt>  <tt class="py-line">        <tt class="py-name">qualities</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-67" class="py-name"><a title="Bio.SeqRecord.SeqRecord.letter_annotations" class="py-name" href="#" onclick="return doclink('link-67', 'letter_annotations', 'link-21');">letter_annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"solexa_quality"</tt><tt class="py-op">]</tt> </tt>
<a name="L706"></a><tt class="py-lineno"> 706</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L707"></a><tt class="py-lineno"> 707</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"No suitable quality scores found in "</tt> </tt>
<a name="L708"></a><tt class="py-lineno"> 708</tt>  <tt class="py-line">                         <tt class="py-string">"letter_annotations of SeqRecord (id=%s)."</tt> </tt>
<a name="L709"></a><tt class="py-lineno"> 709</tt>  <tt class="py-line">                         <tt class="py-op">%</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-68" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-68', 'id', 'link-24');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L710"></a><tt class="py-lineno"> 710</tt>  <tt class="py-line">    <tt class="py-comment"># Try and use the precomputed mapping:</tt> </tt>
<a name="L711"></a><tt class="py-lineno"> 711</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L712"></a><tt class="py-lineno"> 712</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt id="link-69" class="py-name"><a title="Bio.SeqIO.QualityIO._solexa_to_illumina_quality_str" class="py-name" href="#" onclick="return doclink('link-69', '_solexa_to_illumina_quality_str', 'link-55');">_solexa_to_illumina_quality_str</a></tt><tt class="py-op">[</tt><tt class="py-name">qs</tt><tt class="py-op">]</tt> </tt>
<a name="L713"></a><tt class="py-lineno"> 713</tt>  <tt class="py-line">                       <tt class="py-keyword">for</tt> <tt class="py-name">qs</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qualities</tt><tt class="py-op">)</tt> </tt>
<a name="L714"></a><tt class="py-lineno"> 714</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L715"></a><tt class="py-lineno"> 715</tt>  <tt class="py-line">        <tt class="py-comment"># Either no PHRED scores, or something odd like a float or None</tt> </tt>
<a name="L716"></a><tt class="py-lineno"> 716</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
<a name="L717"></a><tt class="py-lineno"> 717</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">None</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qualities</tt><tt class="py-op">:</tt> </tt>
<a name="L718"></a><tt class="py-lineno"> 718</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"A quality value of None was found"</tt><tt class="py-op">)</tt> </tt>
<a name="L719"></a><tt class="py-lineno"> 719</tt>  <tt class="py-line">    <tt class="py-comment"># Must do this the slow way, first converting the PHRED scores into</tt> </tt>
<a name="L720"></a><tt class="py-lineno"> 720</tt>  <tt class="py-line">    <tt class="py-comment"># Solexa scores:</tt> </tt>
<a name="L721"></a><tt class="py-lineno"> 721</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt id="link-70" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-70', 'max', 'link-19');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">qualities</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">62.5</tt><tt class="py-op">:</tt> </tt>
<a name="L722"></a><tt class="py-lineno"> 722</tt>  <tt class="py-line">        <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Data loss - max PHRED quality 62 in Illumina FASTQ"</tt><tt class="py-op">,</tt> </tt>
<a name="L723"></a><tt class="py-lineno"> 723</tt>  <tt class="py-line">                      <tt id="link-71" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-71', 'BiopythonWarning', 'link-15');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L724"></a><tt class="py-lineno"> 724</tt>  <tt class="py-line">    <tt class="py-comment"># This will apply the truncation at 62, giving max ASCII 126</tt> </tt>
<a name="L725"></a><tt class="py-lineno"> 725</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt id="link-72" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-72', 'min', 'link-26');">min</a></tt><tt class="py-op">(</tt><tt class="py-number">126</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-73" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.round
Bio.Blast.NCBIStandalone._DescriptionConsumer.round" class="py-name" href="#" onclick="return doclink('link-73', 'round', 'link-31');">round</a></tt><tt class="py-op">(</tt><tt id="link-74" class="py-name"><a title="Bio.SeqIO.QualityIO.phred_quality_from_solexa" class="py-name" href="#" onclick="return doclink('link-74', 'phred_quality_from_solexa', 'link-22');">phred_quality_from_solexa</a></tt><tt class="py-op">(</tt><tt class="py-name">qs</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt id="link-75" class="py-name"><a title="Bio.SeqIO.QualityIO.SOLEXA_SCORE_OFFSET" class="py-name" href="#" onclick="return doclink('link-75', 'SOLEXA_SCORE_OFFSET', 'link-18');">SOLEXA_SCORE_OFFSET</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L726"></a><tt class="py-lineno"> 726</tt>  <tt class="py-line">                   <tt class="py-keyword">for</tt> <tt class="py-name">qs</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qualities</tt><tt class="py-op">)</tt> </tt>
</div><a name="L727"></a><tt class="py-lineno"> 727</tt>  <tt class="py-line"> </tt>
<a name="L728"></a><tt class="py-lineno"> 728</tt>  <tt class="py-line"><tt class="py-comment"># Only map 0 to 62, we need to give a warning on truncating at 62</tt> </tt>
<a name="L729"></a><tt class="py-lineno"> 729</tt>  <tt class="py-line"><tt class="py-keyword">assert</tt> <tt class="py-number">62</tt> <tt class="py-op">+</tt> <tt id="link-76" class="py-name"><a title="Bio.SeqIO.QualityIO.SOLEXA_SCORE_OFFSET" class="py-name" href="#" onclick="return doclink('link-76', 'SOLEXA_SCORE_OFFSET', 'link-18');">SOLEXA_SCORE_OFFSET</a></tt> <tt class="py-op">==</tt> <tt class="py-number">126</tt> </tt>
<a name="L730"></a><tt class="py-lineno"> 730</tt>  <tt class="py-line"><tt id="link-77" class="py-name" targets="Variable Bio.SeqIO.QualityIO._solexa_to_solexa_quality_str=Bio.SeqIO.QualityIO-module.html#_solexa_to_solexa_quality_str"><a title="Bio.SeqIO.QualityIO._solexa_to_solexa_quality_str" class="py-name" href="#" onclick="return doclink('link-77', '_solexa_to_solexa_quality_str', 'link-77');">_solexa_to_solexa_quality_str</a></tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">qs</tt><tt class="py-op">,</tt> <tt class="py-name">chr</tt><tt class="py-op">(</tt><tt id="link-78" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-78', 'min', 'link-26');">min</a></tt><tt class="py-op">(</tt><tt class="py-number">126</tt><tt class="py-op">,</tt> <tt class="py-name">qs</tt> <tt class="py-op">+</tt> <tt id="link-79" class="py-name"><a title="Bio.SeqIO.QualityIO.SOLEXA_SCORE_OFFSET" class="py-name" href="#" onclick="return doclink('link-79', 'SOLEXA_SCORE_OFFSET', 'link-18');">SOLEXA_SCORE_OFFSET</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L731"></a><tt class="py-lineno"> 731</tt>  <tt class="py-line">                                     <tt class="py-keyword">for</tt> <tt class="py-name">qs</tt> <tt class="py-keyword">in</tt> <tt id="link-80" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-80', 'range', 'link-28');">range</a></tt><tt class="py-op">(</tt><tt class="py-op">-</tt><tt class="py-number">5</tt><tt class="py-op">,</tt> <tt class="py-number">62</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L732"></a><tt class="py-lineno"> 732</tt>  <tt class="py-line"><tt class="py-comment"># Only map -5 to 62, we need to give a warning on truncating at 62</tt> </tt>
<a name="L733"></a><tt class="py-lineno"> 733</tt>  <tt class="py-line"><tt id="link-81" class="py-name" targets="Variable Bio.SeqIO.QualityIO._phred_to_solexa_quality_str=Bio.SeqIO.QualityIO-module.html#_phred_to_solexa_quality_str"><a title="Bio.SeqIO.QualityIO._phred_to_solexa_quality_str" class="py-name" href="#" onclick="return doclink('link-81', '_phred_to_solexa_quality_str', 'link-81');">_phred_to_solexa_quality_str</a></tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt> </tt>
<a name="L734"></a><tt class="py-lineno"> 734</tt>  <tt class="py-line">    <tt class="py-op">(</tt><tt class="py-name">qp</tt><tt class="py-op">,</tt> <tt class="py-name">chr</tt><tt class="py-op">(</tt><tt id="link-82" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-82', 'min', 'link-26');">min</a></tt><tt class="py-op">(</tt><tt class="py-number">126</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-83" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.round
Bio.Blast.NCBIStandalone._DescriptionConsumer.round" class="py-name" href="#" onclick="return doclink('link-83', 'round', 'link-31');">round</a></tt><tt class="py-op">(</tt><tt id="link-84" class="py-name" targets="Function Bio.SeqIO.QualityIO.solexa_quality_from_phred()=Bio.SeqIO.QualityIO-module.html#solexa_quality_from_phred"><a title="Bio.SeqIO.QualityIO.solexa_quality_from_phred" class="py-name" href="#" onclick="return doclink('link-84', 'solexa_quality_from_phred', 'link-84');">solexa_quality_from_phred</a></tt><tt class="py-op">(</tt><tt class="py-name">qp</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> </tt>
<a name="L735"></a><tt class="py-lineno"> 735</tt>  <tt class="py-line">     <tt id="link-85" class="py-name"><a title="Bio.SeqIO.QualityIO.SOLEXA_SCORE_OFFSET" class="py-name" href="#" onclick="return doclink('link-85', 'SOLEXA_SCORE_OFFSET', 'link-18');">SOLEXA_SCORE_OFFSET</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L736"></a><tt class="py-lineno"> 736</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">qp</tt> <tt class="py-keyword">in</tt> <tt id="link-86" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-86', 'range', 'link-28');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">62</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L737"></a><tt class="py-lineno"> 737</tt>  <tt class="py-line"> </tt>
<a name="L738"></a><tt class="py-lineno"> 738</tt>  <tt class="py-line"> </tt>
<a name="_get_solexa_quality_str"></a><div id="_get_solexa_quality_str-def"><a name="L739"></a><tt class="py-lineno"> 739</tt> <a class="py-toggle" href="#" id="_get_solexa_quality_str-toggle" onclick="return toggle('_get_solexa_quality_str');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.QualityIO-module.html#_get_solexa_quality_str">_get_solexa_quality_str</a><tt class="py-op">(</tt><tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_get_solexa_quality_str-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="_get_solexa_quality_str-expanded"><a name="L740"></a><tt class="py-lineno"> 740</tt>  <tt class="py-line">    <tt class="py-docstring">"""Returns a Solexa FASTQ encoded quality string (PRIVATE).</tt> </tt>
<a name="L741"></a><tt class="py-lineno"> 741</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L742"></a><tt class="py-lineno"> 742</tt>  <tt class="py-line"><tt class="py-docstring">    Notice that due to the limited range of printable ASCII characters, a</tt> </tt>
<a name="L743"></a><tt class="py-lineno"> 743</tt>  <tt class="py-line"><tt class="py-docstring">    Solexa quality of 62 is the maximum that can be held in a Solexa FASTQ</tt> </tt>
<a name="L744"></a><tt class="py-lineno"> 744</tt>  <tt class="py-line"><tt class="py-docstring">    file (using ASCII 126, the tilde). This function will issue a warning</tt> </tt>
<a name="L745"></a><tt class="py-lineno"> 745</tt>  <tt class="py-line"><tt class="py-docstring">    in this situation.</tt> </tt>
<a name="L746"></a><tt class="py-lineno"> 746</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L747"></a><tt class="py-lineno"> 747</tt>  <tt class="py-line">    <tt class="py-comment"># TODO - This functions works and is fast, but it is also ugly</tt> </tt>
<a name="L748"></a><tt class="py-lineno"> 748</tt>  <tt class="py-line">    <tt class="py-comment"># and there is considerable repetition of code for the other</tt> </tt>
<a name="L749"></a><tt class="py-lineno"> 749</tt>  <tt class="py-line">    <tt class="py-comment"># two FASTQ variants.</tt> </tt>
<a name="L750"></a><tt class="py-lineno"> 750</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L751"></a><tt class="py-lineno"> 751</tt>  <tt class="py-line">        <tt class="py-comment"># These take priority (in case both Solexa and PHRED scores found)</tt> </tt>
<a name="L752"></a><tt class="py-lineno"> 752</tt>  <tt class="py-line">        <tt class="py-name">qualities</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-87" class="py-name"><a title="Bio.SeqRecord.SeqRecord.letter_annotations" class="py-name" href="#" onclick="return doclink('link-87', 'letter_annotations', 'link-21');">letter_annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"solexa_quality"</tt><tt class="py-op">]</tt> </tt>
<a name="L753"></a><tt class="py-lineno"> 753</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L754"></a><tt class="py-lineno"> 754</tt>  <tt class="py-line">        <tt class="py-comment"># Fall back on PHRED scores...</tt> </tt>
<a name="L755"></a><tt class="py-lineno"> 755</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
<a name="L756"></a><tt class="py-lineno"> 756</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L757"></a><tt class="py-lineno"> 757</tt>  <tt class="py-line">        <tt class="py-comment"># Try and use the precomputed mapping:</tt> </tt>
<a name="L758"></a><tt class="py-lineno"> 758</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L759"></a><tt class="py-lineno"> 759</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt id="link-88" class="py-name"><a title="Bio.SeqIO.QualityIO._solexa_to_solexa_quality_str" class="py-name" href="#" onclick="return doclink('link-88', '_solexa_to_solexa_quality_str', 'link-77');">_solexa_to_solexa_quality_str</a></tt><tt class="py-op">[</tt><tt class="py-name">qs</tt><tt class="py-op">]</tt> </tt>
<a name="L760"></a><tt class="py-lineno"> 760</tt>  <tt class="py-line">                           <tt class="py-keyword">for</tt> <tt class="py-name">qs</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qualities</tt><tt class="py-op">)</tt> </tt>
<a name="L761"></a><tt class="py-lineno"> 761</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L762"></a><tt class="py-lineno"> 762</tt>  <tt class="py-line">            <tt class="py-comment"># Could be a float, or a None in the list, or a high value.</tt> </tt>
<a name="L763"></a><tt class="py-lineno"> 763</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
<a name="L764"></a><tt class="py-lineno"> 764</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">None</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qualities</tt><tt class="py-op">:</tt> </tt>
<a name="L765"></a><tt class="py-lineno"> 765</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"A quality value of None was found"</tt><tt class="py-op">)</tt> </tt>
<a name="L766"></a><tt class="py-lineno"> 766</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-89" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-89', 'max', 'link-19');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">qualities</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">62.5</tt><tt class="py-op">:</tt> </tt>
<a name="L767"></a><tt class="py-lineno"> 767</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Data loss - max Solexa quality 62 in Solexa FASTQ"</tt><tt class="py-op">,</tt> </tt>
<a name="L768"></a><tt class="py-lineno"> 768</tt>  <tt class="py-line">                          <tt id="link-90" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-90', 'BiopythonWarning', 'link-15');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L769"></a><tt class="py-lineno"> 769</tt>  <tt class="py-line">        <tt class="py-comment"># This will apply the truncation at 62, giving max ASCII 126</tt> </tt>
<a name="L770"></a><tt class="py-lineno"> 770</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt id="link-91" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-91', 'min', 'link-26');">min</a></tt><tt class="py-op">(</tt><tt class="py-number">126</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-92" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.round
Bio.Blast.NCBIStandalone._DescriptionConsumer.round" class="py-name" href="#" onclick="return doclink('link-92', 'round', 'link-31');">round</a></tt><tt class="py-op">(</tt><tt class="py-name">qs</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt id="link-93" class="py-name"><a title="Bio.SeqIO.QualityIO.SOLEXA_SCORE_OFFSET" class="py-name" href="#" onclick="return doclink('link-93', 'SOLEXA_SCORE_OFFSET', 'link-18');">SOLEXA_SCORE_OFFSET</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L771"></a><tt class="py-lineno"> 771</tt>  <tt class="py-line">                       <tt class="py-keyword">for</tt> <tt class="py-name">qs</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qualities</tt><tt class="py-op">)</tt> </tt>
<a name="L772"></a><tt class="py-lineno"> 772</tt>  <tt class="py-line">    <tt class="py-comment"># Fall back on the PHRED scores...</tt> </tt>
<a name="L773"></a><tt class="py-lineno"> 773</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L774"></a><tt class="py-lineno"> 774</tt>  <tt class="py-line">        <tt class="py-name">qualities</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-94" class="py-name"><a title="Bio.SeqRecord.SeqRecord.letter_annotations" class="py-name" href="#" onclick="return doclink('link-94', 'letter_annotations', 'link-21');">letter_annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"phred_quality"</tt><tt class="py-op">]</tt> </tt>
<a name="L775"></a><tt class="py-lineno"> 775</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L776"></a><tt class="py-lineno"> 776</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"No suitable quality scores found in "</tt> </tt>
<a name="L777"></a><tt class="py-lineno"> 777</tt>  <tt class="py-line">                         <tt class="py-string">"letter_annotations of SeqRecord (id=%s)."</tt> </tt>
<a name="L778"></a><tt class="py-lineno"> 778</tt>  <tt class="py-line">                         <tt class="py-op">%</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-95" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-95', 'id', 'link-24');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L779"></a><tt class="py-lineno"> 779</tt>  <tt class="py-line">    <tt class="py-comment"># Try and use the precomputed mapping:</tt> </tt>
<a name="L780"></a><tt class="py-lineno"> 780</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L781"></a><tt class="py-lineno"> 781</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt id="link-96" class="py-name"><a title="Bio.SeqIO.QualityIO._phred_to_solexa_quality_str" class="py-name" href="#" onclick="return doclink('link-96', '_phred_to_solexa_quality_str', 'link-81');">_phred_to_solexa_quality_str</a></tt><tt class="py-op">[</tt><tt class="py-name">qp</tt><tt class="py-op">]</tt> </tt>
<a name="L782"></a><tt class="py-lineno"> 782</tt>  <tt class="py-line">                       <tt class="py-keyword">for</tt> <tt class="py-name">qp</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qualities</tt><tt class="py-op">)</tt> </tt>
<a name="L783"></a><tt class="py-lineno"> 783</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L784"></a><tt class="py-lineno"> 784</tt>  <tt class="py-line">        <tt class="py-comment"># Either no PHRED scores, or something odd like a float or None</tt> </tt>
<a name="L785"></a><tt class="py-lineno"> 785</tt>  <tt class="py-line">        <tt class="py-comment"># or too big to be in the cache</tt> </tt>
<a name="L786"></a><tt class="py-lineno"> 786</tt>  <tt class="py-line">        <tt class="py-keyword">pass</tt> </tt>
<a name="L787"></a><tt class="py-lineno"> 787</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">None</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qualities</tt><tt class="py-op">:</tt> </tt>
<a name="L788"></a><tt class="py-lineno"> 788</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"A quality value of None was found"</tt><tt class="py-op">)</tt> </tt>
<a name="L789"></a><tt class="py-lineno"> 789</tt>  <tt class="py-line">    <tt class="py-comment"># Must do this the slow way, first converting the PHRED scores into</tt> </tt>
<a name="L790"></a><tt class="py-lineno"> 790</tt>  <tt class="py-line">    <tt class="py-comment"># Solexa scores:</tt> </tt>
<a name="L791"></a><tt class="py-lineno"> 791</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt id="link-97" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-97', 'max', 'link-19');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">qualities</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;=</tt> <tt class="py-number">62.5</tt><tt class="py-op">:</tt> </tt>
<a name="L792"></a><tt class="py-lineno"> 792</tt>  <tt class="py-line">        <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Data loss - max Solexa quality 62 in Solexa FASTQ"</tt><tt class="py-op">,</tt> </tt>
<a name="L793"></a><tt class="py-lineno"> 793</tt>  <tt class="py-line">                      <tt id="link-98" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-98', 'BiopythonWarning', 'link-15');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L794"></a><tt class="py-lineno"> 794</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt id="link-99" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-99', 'min', 'link-26');">min</a></tt><tt class="py-op">(</tt><tt class="py-number">126</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-100" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.round
Bio.Blast.NCBIStandalone._DescriptionConsumer.round" class="py-name" href="#" onclick="return doclink('link-100', 'round', 'link-31');">round</a></tt><tt class="py-op">(</tt><tt id="link-101" class="py-name"><a title="Bio.SeqIO.QualityIO.solexa_quality_from_phred" class="py-name" href="#" onclick="return doclink('link-101', 'solexa_quality_from_phred', 'link-84');">solexa_quality_from_phred</a></tt><tt class="py-op">(</tt><tt class="py-name">qp</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt id="link-102" class="py-name"><a title="Bio.SeqIO.QualityIO.SOLEXA_SCORE_OFFSET" class="py-name" href="#" onclick="return doclink('link-102', 'SOLEXA_SCORE_OFFSET', 'link-18');">SOLEXA_SCORE_OFFSET</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L795"></a><tt class="py-lineno"> 795</tt>  <tt class="py-line">                   <tt class="py-keyword">for</tt> <tt class="py-name">qp</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qualities</tt><tt class="py-op">)</tt> </tt>
</div><a name="L796"></a><tt class="py-lineno"> 796</tt>  <tt class="py-line"> </tt>
<a name="L797"></a><tt class="py-lineno"> 797</tt>  <tt class="py-line"> </tt>
<a name="L798"></a><tt class="py-lineno"> 798</tt>  <tt class="py-line"><tt class="py-comment"># TODO - Default to nucleotide or even DNA?</tt> </tt>
<a name="FastqGeneralIterator"></a><div id="FastqGeneralIterator-def"><a name="L799"></a><tt class="py-lineno"> 799</tt> <a class="py-toggle" href="#" id="FastqGeneralIterator-toggle" onclick="return toggle('FastqGeneralIterator');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.QualityIO-module.html#FastqGeneralIterator">FastqGeneralIterator</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FastqGeneralIterator-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="FastqGeneralIterator-expanded"><a name="L800"></a><tt class="py-lineno"> 800</tt>  <tt class="py-line">    <tt class="py-docstring">"""Iterate over Fastq records as string tuples (not as SeqRecord objects).</tt> </tt>
<a name="L801"></a><tt class="py-lineno"> 801</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L802"></a><tt class="py-lineno"> 802</tt>  <tt class="py-line"><tt class="py-docstring">    This code does not try to interpret the quality string numerically.  It</tt> </tt>
<a name="L803"></a><tt class="py-lineno"> 803</tt>  <tt class="py-line"><tt class="py-docstring">    just returns tuples of the title, sequence and quality as strings.  For</tt> </tt>
<a name="L804"></a><tt class="py-lineno"> 804</tt>  <tt class="py-line"><tt class="py-docstring">    the sequence and quality, any whitespace (such as new lines) is removed.</tt> </tt>
<a name="L805"></a><tt class="py-lineno"> 805</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L806"></a><tt class="py-lineno"> 806</tt>  <tt class="py-line"><tt class="py-docstring">    Our SeqRecord based FASTQ iterators call this function internally, and then</tt> </tt>
<a name="L807"></a><tt class="py-lineno"> 807</tt>  <tt class="py-line"><tt class="py-docstring">    turn the strings into a SeqRecord objects, mapping the quality string into</tt> </tt>
<a name="L808"></a><tt class="py-lineno"> 808</tt>  <tt class="py-line"><tt class="py-docstring">    a list of numerical scores.  If you want to do a custom quality mapping,</tt> </tt>
<a name="L809"></a><tt class="py-lineno"> 809</tt>  <tt class="py-line"><tt class="py-docstring">    then you might consider calling this function directly.</tt> </tt>
<a name="L810"></a><tt class="py-lineno"> 810</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L811"></a><tt class="py-lineno"> 811</tt>  <tt class="py-line"><tt class="py-docstring">    For parsing FASTQ files, the title string from the "@" line at the start</tt> </tt>
<a name="L812"></a><tt class="py-lineno"> 812</tt>  <tt class="py-line"><tt class="py-docstring">    of each record can optionally be omitted on the "+" lines.  If it is</tt> </tt>
<a name="L813"></a><tt class="py-lineno"> 813</tt>  <tt class="py-line"><tt class="py-docstring">    repeated, it must be identical.</tt> </tt>
<a name="L814"></a><tt class="py-lineno"> 814</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L815"></a><tt class="py-lineno"> 815</tt>  <tt class="py-line"><tt class="py-docstring">    The sequence string and the quality string can optionally be split over</tt> </tt>
<a name="L816"></a><tt class="py-lineno"> 816</tt>  <tt class="py-line"><tt class="py-docstring">    multiple lines, although several sources discourage this.  In comparison,</tt> </tt>
<a name="L817"></a><tt class="py-lineno"> 817</tt>  <tt class="py-line"><tt class="py-docstring">    for the FASTA file format line breaks between 60 and 80 characters are</tt> </tt>
<a name="L818"></a><tt class="py-lineno"> 818</tt>  <tt class="py-line"><tt class="py-docstring">    the norm.</tt> </tt>
<a name="L819"></a><tt class="py-lineno"> 819</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L820"></a><tt class="py-lineno"> 820</tt>  <tt class="py-line"><tt class="py-docstring">    **WARNING** - Because the "@" character can appear in the quality string,</tt> </tt>
<a name="L821"></a><tt class="py-lineno"> 821</tt>  <tt class="py-line"><tt class="py-docstring">    this can cause problems as this is also the marker for the start of</tt> </tt>
<a name="L822"></a><tt class="py-lineno"> 822</tt>  <tt class="py-line"><tt class="py-docstring">    a new sequence.  In fact, the "+" sign can also appear as well.  Some</tt> </tt>
<a name="L823"></a><tt class="py-lineno"> 823</tt>  <tt class="py-line"><tt class="py-docstring">    sources recommended having no line breaks in the  quality to avoid this,</tt> </tt>
<a name="L824"></a><tt class="py-lineno"> 824</tt>  <tt class="py-line"><tt class="py-docstring">    but even that is not enough, consider this example::</tt> </tt>
<a name="L825"></a><tt class="py-lineno"> 825</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L826"></a><tt class="py-lineno"> 826</tt>  <tt class="py-line"><tt class="py-docstring">        @071113_EAS56_0053:1:1:998:236</tt> </tt>
<a name="L827"></a><tt class="py-lineno"> 827</tt>  <tt class="py-line"><tt class="py-docstring">        TTTCTTGCCCCCATAGACTGAGACCTTCCCTAAATA</tt> </tt>
<a name="L828"></a><tt class="py-lineno"> 828</tt>  <tt class="py-line"><tt class="py-docstring">        +071113_EAS56_0053:1:1:998:236</tt> </tt>
<a name="L829"></a><tt class="py-lineno"> 829</tt>  <tt class="py-line"><tt class="py-docstring">        IIIIIIIIIIIIIIIIIIIIIIIIIIIIICII+III</tt> </tt>
<a name="L830"></a><tt class="py-lineno"> 830</tt>  <tt class="py-line"><tt class="py-docstring">        @071113_EAS56_0053:1:1:182:712</tt> </tt>
<a name="L831"></a><tt class="py-lineno"> 831</tt>  <tt class="py-line"><tt class="py-docstring">        ACCCAGCTAATTTTTGTATTTTTGTTAGAGACAGTG</tt> </tt>
<a name="L832"></a><tt class="py-lineno"> 832</tt>  <tt class="py-line"><tt class="py-docstring">        +</tt> </tt>
<a name="L833"></a><tt class="py-lineno"> 833</tt>  <tt class="py-line"><tt class="py-docstring">        @IIIIIIIIIIIIIIICDIIIII&lt;%&lt;6&amp;-*).(*%+</tt> </tt>
<a name="L834"></a><tt class="py-lineno"> 834</tt>  <tt class="py-line"><tt class="py-docstring">        @071113_EAS56_0053:1:1:153:10</tt> </tt>
<a name="L835"></a><tt class="py-lineno"> 835</tt>  <tt class="py-line"><tt class="py-docstring">        TGTTCTGAAGGAAGGTGTGCGTGCGTGTGTGTGTGT</tt> </tt>
<a name="L836"></a><tt class="py-lineno"> 836</tt>  <tt class="py-line"><tt class="py-docstring">        +</tt> </tt>
<a name="L837"></a><tt class="py-lineno"> 837</tt>  <tt class="py-line"><tt class="py-docstring">        IIIIIIIIIIIICIIGIIIII&gt;IAIIIE65I=II:6</tt> </tt>
<a name="L838"></a><tt class="py-lineno"> 838</tt>  <tt class="py-line"><tt class="py-docstring">        @071113_EAS56_0053:1:3:990:501</tt> </tt>
<a name="L839"></a><tt class="py-lineno"> 839</tt>  <tt class="py-line"><tt class="py-docstring">        TGGGAGGTTTTATGTGGA</tt> </tt>
<a name="L840"></a><tt class="py-lineno"> 840</tt>  <tt class="py-line"><tt class="py-docstring">        AAGCAGCAATGTACAAGA</tt> </tt>
<a name="L841"></a><tt class="py-lineno"> 841</tt>  <tt class="py-line"><tt class="py-docstring">        +</tt> </tt>
<a name="L842"></a><tt class="py-lineno"> 842</tt>  <tt class="py-line"><tt class="py-docstring">        IIIIIII.IIIIII1@44</tt> </tt>
<a name="L843"></a><tt class="py-lineno"> 843</tt>  <tt class="py-line"><tt class="py-docstring">        @-7.%&lt;&amp;+/$/%4(++(%</tt> </tt>
<a name="L844"></a><tt class="py-lineno"> 844</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L845"></a><tt class="py-lineno"> 845</tt>  <tt class="py-line"><tt class="py-docstring">    This is four PHRED encoded FASTQ entries originally from an NCBI source</tt> </tt>
<a name="L846"></a><tt class="py-lineno"> 846</tt>  <tt class="py-line"><tt class="py-docstring">    (given the read length of 36, these are probably Solexa Illumina reads where</tt> </tt>
<a name="L847"></a><tt class="py-lineno"> 847</tt>  <tt class="py-line"><tt class="py-docstring">    the quality has been mapped onto the PHRED values).</tt> </tt>
<a name="L848"></a><tt class="py-lineno"> 848</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L849"></a><tt class="py-lineno"> 849</tt>  <tt class="py-line"><tt class="py-docstring">    This example has been edited to illustrate some of the nasty things allowed</tt> </tt>
<a name="L850"></a><tt class="py-lineno"> 850</tt>  <tt class="py-line"><tt class="py-docstring">    in the FASTQ format.  Firstly, on the "+" lines most but not all of the</tt> </tt>
<a name="L851"></a><tt class="py-lineno"> 851</tt>  <tt class="py-line"><tt class="py-docstring">    (redundant) identifiers are omitted.  In real files it is likely that all or</tt> </tt>
<a name="L852"></a><tt class="py-lineno"> 852</tt>  <tt class="py-line"><tt class="py-docstring">    none of these extra identifiers will be present.</tt> </tt>
<a name="L853"></a><tt class="py-lineno"> 853</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L854"></a><tt class="py-lineno"> 854</tt>  <tt class="py-line"><tt class="py-docstring">    Secondly, while the first three sequences have been shown without line</tt> </tt>
<a name="L855"></a><tt class="py-lineno"> 855</tt>  <tt class="py-line"><tt class="py-docstring">    breaks, the last has been split over multiple lines.  In real files any line</tt> </tt>
<a name="L856"></a><tt class="py-lineno"> 856</tt>  <tt class="py-line"><tt class="py-docstring">    breaks are likely to be consistent.</tt> </tt>
<a name="L857"></a><tt class="py-lineno"> 857</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L858"></a><tt class="py-lineno"> 858</tt>  <tt class="py-line"><tt class="py-docstring">    Thirdly, some of the quality string lines start with an "@" character.  For</tt> </tt>
<a name="L859"></a><tt class="py-lineno"> 859</tt>  <tt class="py-line"><tt class="py-docstring">    the second record this is unavoidable.  However for the fourth sequence this</tt> </tt>
<a name="L860"></a><tt class="py-lineno"> 860</tt>  <tt class="py-line"><tt class="py-docstring">    only happens because its quality string is split over two lines.  A naive</tt> </tt>
<a name="L861"></a><tt class="py-lineno"> 861</tt>  <tt class="py-line"><tt class="py-docstring">    parser could wrongly treat any line starting with an "@" as the beginning of</tt> </tt>
<a name="L862"></a><tt class="py-lineno"> 862</tt>  <tt class="py-line"><tt class="py-docstring">    a new sequence!  This code copes with this possible ambiguity by keeping</tt> </tt>
<a name="L863"></a><tt class="py-lineno"> 863</tt>  <tt class="py-line"><tt class="py-docstring">    track of the length of the sequence which gives the expected length of the</tt> </tt>
<a name="L864"></a><tt class="py-lineno"> 864</tt>  <tt class="py-line"><tt class="py-docstring">    quality string.</tt> </tt>
<a name="L865"></a><tt class="py-lineno"> 865</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L866"></a><tt class="py-lineno"> 866</tt>  <tt class="py-line"><tt class="py-docstring">    Using this tricky example file as input, this short bit of code demonstrates</tt> </tt>
<a name="L867"></a><tt class="py-lineno"> 867</tt>  <tt class="py-line"><tt class="py-docstring">    what this parsing function would return:</tt> </tt>
<a name="L868"></a><tt class="py-lineno"> 868</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L869"></a><tt class="py-lineno"> 869</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; with open("Quality/tricky.fastq", "rU") as handle:</tt> </tt>
<a name="L870"></a><tt class="py-lineno"> 870</tt>  <tt class="py-line"><tt class="py-docstring">    ...     for (title, sequence, quality) in FastqGeneralIterator(handle):</tt> </tt>
<a name="L871"></a><tt class="py-lineno"> 871</tt>  <tt class="py-line"><tt class="py-docstring">    ...         print(title)</tt> </tt>
<a name="L872"></a><tt class="py-lineno"> 872</tt>  <tt class="py-line"><tt class="py-docstring">    ...         print("%s %s" % (sequence, quality))</tt> </tt>
<a name="L873"></a><tt class="py-lineno"> 873</tt>  <tt class="py-line"><tt class="py-docstring">    ...</tt> </tt>
<a name="L874"></a><tt class="py-lineno"> 874</tt>  <tt class="py-line"><tt class="py-docstring">    071113_EAS56_0053:1:1:998:236</tt> </tt>
<a name="L875"></a><tt class="py-lineno"> 875</tt>  <tt class="py-line"><tt class="py-docstring">    TTTCTTGCCCCCATAGACTGAGACCTTCCCTAAATA IIIIIIIIIIIIIIIIIIIIIIIIIIIIICII+III</tt> </tt>
<a name="L876"></a><tt class="py-lineno"> 876</tt>  <tt class="py-line"><tt class="py-docstring">    071113_EAS56_0053:1:1:182:712</tt> </tt>
<a name="L877"></a><tt class="py-lineno"> 877</tt>  <tt class="py-line"><tt class="py-docstring">    ACCCAGCTAATTTTTGTATTTTTGTTAGAGACAGTG @IIIIIIIIIIIIIIICDIIIII&lt;%&lt;6&amp;-*).(*%+</tt> </tt>
<a name="L878"></a><tt class="py-lineno"> 878</tt>  <tt class="py-line"><tt class="py-docstring">    071113_EAS56_0053:1:1:153:10</tt> </tt>
<a name="L879"></a><tt class="py-lineno"> 879</tt>  <tt class="py-line"><tt class="py-docstring">    TGTTCTGAAGGAAGGTGTGCGTGCGTGTGTGTGTGT IIIIIIIIIIIICIIGIIIII&gt;IAIIIE65I=II:6</tt> </tt>
<a name="L880"></a><tt class="py-lineno"> 880</tt>  <tt class="py-line"><tt class="py-docstring">    071113_EAS56_0053:1:3:990:501</tt> </tt>
<a name="L881"></a><tt class="py-lineno"> 881</tt>  <tt class="py-line"><tt class="py-docstring">    TGGGAGGTTTTATGTGGAAAGCAGCAATGTACAAGA IIIIIII.IIIIII1@44@-7.%&lt;&amp;+/$/%4(++(%</tt> </tt>
<a name="L882"></a><tt class="py-lineno"> 882</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L883"></a><tt class="py-lineno"> 883</tt>  <tt class="py-line"><tt class="py-docstring">    Finally we note that some sources state that the quality string should</tt> </tt>
<a name="L884"></a><tt class="py-lineno"> 884</tt>  <tt class="py-line"><tt class="py-docstring">    start with "!" (which using the PHRED mapping means the first letter always</tt> </tt>
<a name="L885"></a><tt class="py-lineno"> 885</tt>  <tt class="py-line"><tt class="py-docstring">    has a quality score of zero).  This rather restrictive rule is not widely</tt> </tt>
<a name="L886"></a><tt class="py-lineno"> 886</tt>  <tt class="py-line"><tt class="py-docstring">    observed, so is therefore ignored here.  One plus point about this "!" rule</tt> </tt>
<a name="L887"></a><tt class="py-lineno"> 887</tt>  <tt class="py-line"><tt class="py-docstring">    is that (provided there are no line breaks in the quality sequence) it</tt> </tt>
<a name="L888"></a><tt class="py-lineno"> 888</tt>  <tt class="py-line"><tt class="py-docstring">    would prevent the above problem with the "@" character.</tt> </tt>
<a name="L889"></a><tt class="py-lineno"> 889</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L890"></a><tt class="py-lineno"> 890</tt>  <tt class="py-line">    <tt class="py-comment"># We need to call handle.readline() at least four times per record,</tt> </tt>
<a name="L891"></a><tt class="py-lineno"> 891</tt>  <tt class="py-line">    <tt class="py-comment"># so we'll save a property look up each time:</tt> </tt>
<a name="L892"></a><tt class="py-lineno"> 892</tt>  <tt class="py-line">    <tt class="py-name">handle_readline</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-103" class="py-name" targets="Method Bio.File.UndoHandle.readline()=Bio.File.UndoHandle-class.html#readline,Method Bio.bgzf.BgzfReader.readline()=Bio.bgzf.BgzfReader-class.html#readline"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-103', 'readline', 'link-103');">readline</a></tt> </tt>
<a name="L893"></a><tt class="py-lineno"> 893</tt>  <tt class="py-line"> </tt>
<a name="L894"></a><tt class="py-lineno"> 894</tt>  <tt class="py-line">    <tt class="py-comment"># Skip any text before the first record (e.g. blank lines, comments?)</tt> </tt>
<a name="L895"></a><tt class="py-lineno"> 895</tt>  <tt class="py-line">    <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L896"></a><tt class="py-lineno"> 896</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">handle_readline</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L897"></a><tt class="py-lineno"> 897</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L898"></a><tt class="py-lineno"> 898</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt>  <tt class="py-comment"># Premature end of file, or just empty?</tt> </tt>
<a name="L899"></a><tt class="py-lineno"> 899</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"@"</tt><tt class="py-op">:</tt> </tt>
<a name="L900"></a><tt class="py-lineno"> 900</tt>  <tt class="py-line">            <tt class="py-keyword">break</tt> </tt>
<a name="L901"></a><tt class="py-lineno"> 901</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L902"></a><tt class="py-lineno"> 902</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Is this handle in binary mode not text mode?"</tt><tt class="py-op">)</tt> </tt>
<a name="L903"></a><tt class="py-lineno"> 903</tt>  <tt class="py-line"> </tt>
<a name="L904"></a><tt class="py-lineno"> 904</tt>  <tt class="py-line">    <tt class="py-keyword">while</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L905"></a><tt class="py-lineno"> 905</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-string">"@"</tt><tt class="py-op">:</tt> </tt>
<a name="L906"></a><tt class="py-lineno"> 906</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt> </tt>
<a name="L907"></a><tt class="py-lineno"> 907</tt>  <tt class="py-line">                <tt class="py-string">"Records in Fastq files should start with '@' character"</tt><tt class="py-op">)</tt> </tt>
<a name="L908"></a><tt class="py-lineno"> 908</tt>  <tt class="py-line">        <tt class="py-name">title_line</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-104" class="py-name" targets="Method Bio.Seq.Seq.rstrip()=Bio.Seq.Seq-class.html#rstrip"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-104', 'rstrip', 'link-104');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L909"></a><tt class="py-lineno"> 909</tt>  <tt class="py-line">        <tt class="py-comment"># Will now be at least one line of quality data - in most FASTQ files</tt> </tt>
<a name="L910"></a><tt class="py-lineno"> 910</tt>  <tt class="py-line">        <tt class="py-comment"># just one line! We therefore use string concatenation (if needed)</tt> </tt>
<a name="L911"></a><tt class="py-lineno"> 911</tt>  <tt class="py-line">        <tt class="py-comment"># rather using than the "".join(...) trick just in case it is multiline:</tt> </tt>
<a name="L912"></a><tt class="py-lineno"> 912</tt>  <tt class="py-line">        <tt class="py-name">seq_string</tt> <tt class="py-op">=</tt> <tt class="py-name">handle_readline</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-105" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-105', 'rstrip', 'link-104');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L913"></a><tt class="py-lineno"> 913</tt>  <tt class="py-line">        <tt class="py-comment"># There may now be more sequence lines, or the "+" quality marker line:</tt> </tt>
<a name="L914"></a><tt class="py-lineno"> 914</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L915"></a><tt class="py-lineno"> 915</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">handle_readline</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L916"></a><tt class="py-lineno"> 916</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L917"></a><tt class="py-lineno"> 917</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"End of file without quality information."</tt><tt class="py-op">)</tt> </tt>
<a name="L918"></a><tt class="py-lineno"> 918</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"+"</tt><tt class="py-op">:</tt> </tt>
<a name="L919"></a><tt class="py-lineno"> 919</tt>  <tt class="py-line">                <tt class="py-comment"># The title here is optional, but if present must match!</tt> </tt>
<a name="L920"></a><tt class="py-lineno"> 920</tt>  <tt class="py-line">                <tt class="py-name">second_title</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-106" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-106', 'rstrip', 'link-104');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L921"></a><tt class="py-lineno"> 921</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">second_title</tt> <tt class="py-keyword">and</tt> <tt class="py-name">second_title</tt> <tt class="py-op">!=</tt> <tt class="py-name">title_line</tt><tt class="py-op">:</tt> </tt>
<a name="L922"></a><tt class="py-lineno"> 922</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Sequence and quality captions differ."</tt><tt class="py-op">)</tt> </tt>
<a name="L923"></a><tt class="py-lineno"> 923</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L924"></a><tt class="py-lineno"> 924</tt>  <tt class="py-line">            <tt class="py-name">seq_string</tt> <tt class="py-op">+=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-107" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-107', 'rstrip', 'link-104');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt>  <tt class="py-comment"># removes trailing newlines</tt> </tt>
<a name="L925"></a><tt class="py-lineno"> 925</tt>  <tt class="py-line">        <tt class="py-comment"># This is going to slow things down a little, but assuming</tt> </tt>
<a name="L926"></a><tt class="py-lineno"> 926</tt>  <tt class="py-line">        <tt class="py-comment"># this isn't allowed we should try and catch it here:</tt> </tt>
<a name="L927"></a><tt class="py-lineno"> 927</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">" "</tt> <tt class="py-keyword">in</tt> <tt class="py-name">seq_string</tt> <tt class="py-keyword">or</tt> <tt class="py-string">"\t"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">seq_string</tt><tt class="py-op">:</tt> </tt>
<a name="L928"></a><tt class="py-lineno"> 928</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Whitespace is not allowed in the sequence."</tt><tt class="py-op">)</tt> </tt>
<a name="L929"></a><tt class="py-lineno"> 929</tt>  <tt class="py-line">        <tt class="py-name">seq_len</tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">seq_string</tt><tt class="py-op">)</tt> </tt>
<a name="L930"></a><tt class="py-lineno"> 930</tt>  <tt class="py-line"> </tt>
<a name="L931"></a><tt class="py-lineno"> 931</tt>  <tt class="py-line">        <tt class="py-comment"># Will now be at least one line of quality data...</tt> </tt>
<a name="L932"></a><tt class="py-lineno"> 932</tt>  <tt class="py-line">        <tt class="py-name">quality_string</tt> <tt class="py-op">=</tt> <tt class="py-name">handle_readline</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-108" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-108', 'rstrip', 'link-104');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L933"></a><tt class="py-lineno"> 933</tt>  <tt class="py-line">        <tt class="py-comment"># There may now be more quality data, or another sequence, or EOF</tt> </tt>
<a name="L934"></a><tt class="py-lineno"> 934</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L935"></a><tt class="py-lineno"> 935</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">handle_readline</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L936"></a><tt class="py-lineno"> 936</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L937"></a><tt class="py-lineno"> 937</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt>  <tt class="py-comment"># end of file</tt> </tt>
<a name="L938"></a><tt class="py-lineno"> 938</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"@"</tt><tt class="py-op">:</tt> </tt>
<a name="L939"></a><tt class="py-lineno"> 939</tt>  <tt class="py-line">                <tt class="py-comment"># This COULD be the start of a new sequence. However, it MAY just</tt> </tt>
<a name="L940"></a><tt class="py-lineno"> 940</tt>  <tt class="py-line">                <tt class="py-comment"># be a line of quality data which starts with a "@" character.  We</tt> </tt>
<a name="L941"></a><tt class="py-lineno"> 941</tt>  <tt class="py-line">                <tt class="py-comment"># should be able to check this by looking at the sequence length</tt> </tt>
<a name="L942"></a><tt class="py-lineno"> 942</tt>  <tt class="py-line">                <tt class="py-comment"># and the amount of quality data found so far.</tt> </tt>
<a name="L943"></a><tt class="py-lineno"> 943</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">quality_string</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;=</tt> <tt class="py-name">seq_len</tt><tt class="py-op">:</tt> </tt>
<a name="L944"></a><tt class="py-lineno"> 944</tt>  <tt class="py-line">                    <tt class="py-comment"># We expect it to be equal if this is the start of a new record.</tt> </tt>
<a name="L945"></a><tt class="py-lineno"> 945</tt>  <tt class="py-line">                    <tt class="py-comment"># If the quality data is longer, we'll raise an error below.</tt> </tt>
<a name="L946"></a><tt class="py-lineno"> 946</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L947"></a><tt class="py-lineno"> 947</tt>  <tt class="py-line">                <tt class="py-comment"># Continue - its just some (more) quality data.</tt> </tt>
<a name="L948"></a><tt class="py-lineno"> 948</tt>  <tt class="py-line">            <tt class="py-name">quality_string</tt> <tt class="py-op">+=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-109" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-109', 'rstrip', 'link-104');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L949"></a><tt class="py-lineno"> 949</tt>  <tt class="py-line"> </tt>
<a name="L950"></a><tt class="py-lineno"> 950</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">seq_len</tt> <tt class="py-op">!=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">quality_string</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L951"></a><tt class="py-lineno"> 951</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Lengths of sequence and quality values differs "</tt> </tt>
<a name="L952"></a><tt class="py-lineno"> 952</tt>  <tt class="py-line">                             <tt class="py-string">" for %s (%i and %i)."</tt> </tt>
<a name="L953"></a><tt class="py-lineno"> 953</tt>  <tt class="py-line">                             <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">title_line</tt><tt class="py-op">,</tt> <tt class="py-name">seq_len</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">quality_string</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L954"></a><tt class="py-lineno"> 954</tt>  <tt class="py-line"> </tt>
<a name="L955"></a><tt class="py-lineno"> 955</tt>  <tt class="py-line">        <tt class="py-comment"># Return the record and then continue...</tt> </tt>
<a name="L956"></a><tt class="py-lineno"> 956</tt>  <tt class="py-line">        <tt class="py-keyword">yield</tt> <tt class="py-op">(</tt><tt class="py-name">title_line</tt><tt class="py-op">,</tt> <tt class="py-name">seq_string</tt><tt class="py-op">,</tt> <tt class="py-name">quality_string</tt><tt class="py-op">)</tt> </tt>
<a name="L957"></a><tt class="py-lineno"> 957</tt>  <tt class="py-line">    <tt class="py-keyword">raise</tt> <tt class="py-name">StopIteration</tt> </tt>
</div><a name="L958"></a><tt class="py-lineno"> 958</tt>  <tt class="py-line"> </tt>
<a name="L959"></a><tt class="py-lineno"> 959</tt>  <tt class="py-line"> </tt>
<a name="FastqPhredIterator"></a><div id="FastqPhredIterator-def"><a name="L960"></a><tt class="py-lineno"> 960</tt> <a class="py-toggle" href="#" id="FastqPhredIterator-toggle" onclick="return toggle('FastqPhredIterator');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.QualityIO-module.html#FastqPhredIterator">FastqPhredIterator</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt id="link-110" class="py-name"><a title="Bio.Alphabet.single_letter_alphabet" class="py-name" href="#" onclick="return doclink('link-110', 'single_letter_alphabet', 'link-2');">single_letter_alphabet</a></tt><tt class="py-op">,</tt> <tt class="py-param">title2ids</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FastqPhredIterator-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="FastqPhredIterator-expanded"><a name="L961"></a><tt class="py-lineno"> 961</tt>  <tt class="py-line">    <tt class="py-docstring">"""Generator function to iterate over FASTQ records (as SeqRecord objects).</tt> </tt>
<a name="L962"></a><tt class="py-lineno"> 962</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L963"></a><tt class="py-lineno"> 963</tt>  <tt class="py-line"><tt class="py-docstring">        - handle - input file</tt> </tt>
<a name="L964"></a><tt class="py-lineno"> 964</tt>  <tt class="py-line"><tt class="py-docstring">        - alphabet - optional alphabet</tt> </tt>
<a name="L965"></a><tt class="py-lineno"> 965</tt>  <tt class="py-line"><tt class="py-docstring">        - title2ids - A function that, when given the title line from the FASTQ</tt> </tt>
<a name="L966"></a><tt class="py-lineno"> 966</tt>  <tt class="py-line"><tt class="py-docstring">          file (without the beginning &gt;), will return the id, name and</tt> </tt>
<a name="L967"></a><tt class="py-lineno"> 967</tt>  <tt class="py-line"><tt class="py-docstring">          description (in that order) for the record as a tuple of</tt> </tt>
<a name="L968"></a><tt class="py-lineno"> 968</tt>  <tt class="py-line"><tt class="py-docstring">          strings.  If this is not given, then the entire title line</tt> </tt>
<a name="L969"></a><tt class="py-lineno"> 969</tt>  <tt class="py-line"><tt class="py-docstring">          will be used as the description, and the first word as the</tt> </tt>
<a name="L970"></a><tt class="py-lineno"> 970</tt>  <tt class="py-line"><tt class="py-docstring">          id and name.</tt> </tt>
<a name="L971"></a><tt class="py-lineno"> 971</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L972"></a><tt class="py-lineno"> 972</tt>  <tt class="py-line"><tt class="py-docstring">    Note that use of title2ids matches that of Bio.SeqIO.FastaIO.</tt> </tt>
<a name="L973"></a><tt class="py-lineno"> 973</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L974"></a><tt class="py-lineno"> 974</tt>  <tt class="py-line"><tt class="py-docstring">    For each sequence in a (Sanger style) FASTQ file there is a matching string</tt> </tt>
<a name="L975"></a><tt class="py-lineno"> 975</tt>  <tt class="py-line"><tt class="py-docstring">    encoding the PHRED qualities (integers between 0 and about 90) using ASCII</tt> </tt>
<a name="L976"></a><tt class="py-lineno"> 976</tt>  <tt class="py-line"><tt class="py-docstring">    values with an offset of 33.</tt> </tt>
<a name="L977"></a><tt class="py-lineno"> 977</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L978"></a><tt class="py-lineno"> 978</tt>  <tt class="py-line"><tt class="py-docstring">    For example, consider a file containing three short reads::</tt> </tt>
<a name="L979"></a><tt class="py-lineno"> 979</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L980"></a><tt class="py-lineno"> 980</tt>  <tt class="py-line"><tt class="py-docstring">        @EAS54_6_R1_2_1_413_324</tt> </tt>
<a name="L981"></a><tt class="py-lineno"> 981</tt>  <tt class="py-line"><tt class="py-docstring">        CCCTTCTTGTCTTCAGCGTTTCTCC</tt> </tt>
<a name="L982"></a><tt class="py-lineno"> 982</tt>  <tt class="py-line"><tt class="py-docstring">        +</tt> </tt>
<a name="L983"></a><tt class="py-lineno"> 983</tt>  <tt class="py-line"><tt class="py-docstring">        ;;3;;;;;;;;;;;;7;;;;;;;88</tt> </tt>
<a name="L984"></a><tt class="py-lineno"> 984</tt>  <tt class="py-line"><tt class="py-docstring">        @EAS54_6_R1_2_1_540_792</tt> </tt>
<a name="L985"></a><tt class="py-lineno"> 985</tt>  <tt class="py-line"><tt class="py-docstring">        TTGGCAGGCCAAGGCCGATGGATCA</tt> </tt>
<a name="L986"></a><tt class="py-lineno"> 986</tt>  <tt class="py-line"><tt class="py-docstring">        +</tt> </tt>
<a name="L987"></a><tt class="py-lineno"> 987</tt>  <tt class="py-line"><tt class="py-docstring">        ;;;;;;;;;;;7;;;;;-;;;3;83</tt> </tt>
<a name="L988"></a><tt class="py-lineno"> 988</tt>  <tt class="py-line"><tt class="py-docstring">        @EAS54_6_R1_2_1_443_348</tt> </tt>
<a name="L989"></a><tt class="py-lineno"> 989</tt>  <tt class="py-line"><tt class="py-docstring">        GTTGCTTCTGGCGTGGGTGGGGGGG</tt> </tt>
<a name="L990"></a><tt class="py-lineno"> 990</tt>  <tt class="py-line"><tt class="py-docstring">        +</tt> </tt>
<a name="L991"></a><tt class="py-lineno"> 991</tt>  <tt class="py-line"><tt class="py-docstring">        ;;;;;;;;;;;9;7;;.7;393333</tt> </tt>
<a name="L992"></a><tt class="py-lineno"> 992</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L993"></a><tt class="py-lineno"> 993</tt>  <tt class="py-line"><tt class="py-docstring">    For each sequence (e.g. "CCCTTCTTGTCTTCAGCGTTTCTCC") there is a matching</tt> </tt>
<a name="L994"></a><tt class="py-lineno"> 994</tt>  <tt class="py-line"><tt class="py-docstring">    string encoding the PHRED qualities using a ASCII values with an offset of</tt> </tt>
<a name="L995"></a><tt class="py-lineno"> 995</tt>  <tt class="py-line"><tt class="py-docstring">    33 (e.g. ";;3;;;;;;;;;;;;7;;;;;;;88").</tt> </tt>
<a name="L996"></a><tt class="py-lineno"> 996</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L997"></a><tt class="py-lineno"> 997</tt>  <tt class="py-line"><tt class="py-docstring">    Using this module directly you might run:</tt> </tt>
<a name="L998"></a><tt class="py-lineno"> 998</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L999"></a><tt class="py-lineno"> 999</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; with open("Quality/example.fastq", "rU") as handle:</tt> </tt>
<a name="L1000"></a><tt class="py-lineno">1000</tt>  <tt class="py-line"><tt class="py-docstring">    ...     for record in FastqPhredIterator(handle):</tt> </tt>
<a name="L1001"></a><tt class="py-lineno">1001</tt>  <tt class="py-line"><tt class="py-docstring">    ...         print("%s %s" % (record.id, record.seq))</tt> </tt>
<a name="L1002"></a><tt class="py-lineno">1002</tt>  <tt class="py-line"><tt class="py-docstring">    EAS54_6_R1_2_1_413_324 CCCTTCTTGTCTTCAGCGTTTCTCC</tt> </tt>
<a name="L1003"></a><tt class="py-lineno">1003</tt>  <tt class="py-line"><tt class="py-docstring">    EAS54_6_R1_2_1_540_792 TTGGCAGGCCAAGGCCGATGGATCA</tt> </tt>
<a name="L1004"></a><tt class="py-lineno">1004</tt>  <tt class="py-line"><tt class="py-docstring">    EAS54_6_R1_2_1_443_348 GTTGCTTCTGGCGTGGGTGGGGGGG</tt> </tt>
<a name="L1005"></a><tt class="py-lineno">1005</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1006"></a><tt class="py-lineno">1006</tt>  <tt class="py-line"><tt class="py-docstring">    Typically however, you would call this via Bio.SeqIO instead with "fastq"</tt> </tt>
<a name="L1007"></a><tt class="py-lineno">1007</tt>  <tt class="py-line"><tt class="py-docstring">    (or "fastq-sanger") as the format:</tt> </tt>
<a name="L1008"></a><tt class="py-lineno">1008</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1009"></a><tt class="py-lineno">1009</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L1010"></a><tt class="py-lineno">1010</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; with open("Quality/example.fastq", "rU") as handle:</tt> </tt>
<a name="L1011"></a><tt class="py-lineno">1011</tt>  <tt class="py-line"><tt class="py-docstring">    ...     for record in SeqIO.parse(handle, "fastq"):</tt> </tt>
<a name="L1012"></a><tt class="py-lineno">1012</tt>  <tt class="py-line"><tt class="py-docstring">    ...         print("%s %s" % (record.id, record.seq))</tt> </tt>
<a name="L1013"></a><tt class="py-lineno">1013</tt>  <tt class="py-line"><tt class="py-docstring">    EAS54_6_R1_2_1_413_324 CCCTTCTTGTCTTCAGCGTTTCTCC</tt> </tt>
<a name="L1014"></a><tt class="py-lineno">1014</tt>  <tt class="py-line"><tt class="py-docstring">    EAS54_6_R1_2_1_540_792 TTGGCAGGCCAAGGCCGATGGATCA</tt> </tt>
<a name="L1015"></a><tt class="py-lineno">1015</tt>  <tt class="py-line"><tt class="py-docstring">    EAS54_6_R1_2_1_443_348 GTTGCTTCTGGCGTGGGTGGGGGGG</tt> </tt>
<a name="L1016"></a><tt class="py-lineno">1016</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1017"></a><tt class="py-lineno">1017</tt>  <tt class="py-line"><tt class="py-docstring">    If you want to look at the qualities, they are record in each record's</tt> </tt>
<a name="L1018"></a><tt class="py-lineno">1018</tt>  <tt class="py-line"><tt class="py-docstring">    per-letter-annotation dictionary as a simple list of integers:</tt> </tt>
<a name="L1019"></a><tt class="py-lineno">1019</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1020"></a><tt class="py-lineno">1020</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(record.letter_annotations["phred_quality"])</tt> </tt>
<a name="L1021"></a><tt class="py-lineno">1021</tt>  <tt class="py-line"><tt class="py-docstring">    [26, 26, 26, 26, 26, 26, 26, 26, 26, 26, 26, 24, 26, 22, 26, 26, 13, 22, 26, 18, 24, 18, 18, 18, 18]</tt> </tt>
<a name="L1022"></a><tt class="py-lineno">1022</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1023"></a><tt class="py-lineno">1023</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1024"></a><tt class="py-lineno">1024</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt id="link-111" class="py-name"><a title="Bio.SeqIO.QualityIO.SANGER_SCORE_OFFSET" class="py-name" href="#" onclick="return doclink('link-111', 'SANGER_SCORE_OFFSET', 'link-17');">SANGER_SCORE_OFFSET</a></tt> <tt class="py-op">==</tt> <tt class="py-name">ord</tt><tt class="py-op">(</tt><tt class="py-string">"!"</tt><tt class="py-op">)</tt> </tt>
<a name="L1025"></a><tt class="py-lineno">1025</tt>  <tt class="py-line">    <tt class="py-comment"># Originally, I used a list expression for each record:</tt> </tt>
<a name="L1026"></a><tt class="py-lineno">1026</tt>  <tt class="py-line">    <tt class="py-comment">#</tt> </tt>
<a name="L1027"></a><tt class="py-lineno">1027</tt>  <tt class="py-line">    <tt class="py-comment"># qualities = [ord(letter)-SANGER_SCORE_OFFSET for letter in quality_string]</tt> </tt>
<a name="L1028"></a><tt class="py-lineno">1028</tt>  <tt class="py-line">    <tt class="py-comment">#</tt> </tt>
<a name="L1029"></a><tt class="py-lineno">1029</tt>  <tt class="py-line">    <tt class="py-comment"># Precomputing is faster, perhaps partly by avoiding the subtractions.</tt> </tt>
<a name="L1030"></a><tt class="py-lineno">1030</tt>  <tt class="py-line">    <tt class="py-name">q_mapping</tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1031"></a><tt class="py-lineno">1031</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">letter</tt> <tt class="py-keyword">in</tt> <tt id="link-112" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-112', 'range', 'link-28');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">255</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1032"></a><tt class="py-lineno">1032</tt>  <tt class="py-line">        <tt class="py-name">q_mapping</tt><tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-name">letter</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">letter</tt> <tt class="py-op">-</tt> <tt id="link-113" class="py-name"><a title="Bio.SeqIO.QualityIO.SANGER_SCORE_OFFSET" class="py-name" href="#" onclick="return doclink('link-113', 'SANGER_SCORE_OFFSET', 'link-17');">SANGER_SCORE_OFFSET</a></tt> </tt>
<a name="L1033"></a><tt class="py-lineno">1033</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">title_line</tt><tt class="py-op">,</tt> <tt class="py-name">seq_string</tt><tt class="py-op">,</tt> <tt class="py-name">quality_string</tt> <tt class="py-keyword">in</tt> <tt id="link-114" class="py-name" targets="Function Bio.SeqIO.QualityIO.FastqGeneralIterator()=Bio.SeqIO.QualityIO-module.html#FastqGeneralIterator"><a title="Bio.SeqIO.QualityIO.FastqGeneralIterator" class="py-name" href="#" onclick="return doclink('link-114', 'FastqGeneralIterator', 'link-114');">FastqGeneralIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1034"></a><tt class="py-lineno">1034</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">title2ids</tt><tt class="py-op">:</tt> </tt>
<a name="L1035"></a><tt class="py-lineno">1035</tt>  <tt class="py-line">            <tt id="link-115" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-115', 'id', 'link-24');">id</a></tt><tt class="py-op">,</tt> <tt id="link-116" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.name=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.name=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.name=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#name,Method Bio.Phylo.PhyloXMLIO.Writer.name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#name,Variable Bio.codonalign.codonalphabet.CodonAlphabet.name=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-116', 'name', 'link-116');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">descr</tt> <tt class="py-op">=</tt> <tt class="py-name">title2ids</tt><tt class="py-op">(</tt><tt class="py-name">title_line</tt><tt class="py-op">)</tt> </tt>
<a name="L1036"></a><tt class="py-lineno">1036</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1037"></a><tt class="py-lineno">1037</tt>  <tt class="py-line">            <tt class="py-name">descr</tt> <tt class="py-op">=</tt> <tt class="py-name">title_line</tt> </tt>
<a name="L1038"></a><tt class="py-lineno">1038</tt>  <tt class="py-line">            <tt id="link-117" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-117', 'id', 'link-24');">id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">descr</tt><tt class="py-op">.</tt><tt id="link-118" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-118', 'split', 'link-118');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L1039"></a><tt class="py-lineno">1039</tt>  <tt class="py-line">            <tt id="link-119" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-119', 'name', 'link-116');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-120" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-120', 'id', 'link-24');">id</a></tt> </tt>
<a name="L1040"></a><tt class="py-lineno">1040</tt>  <tt class="py-line">        <tt class="py-name">record</tt> <tt class="py-op">=</tt> <tt id="link-121" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-121', 'SeqRecord', 'link-8');">SeqRecord</a></tt><tt class="py-op">(</tt><tt id="link-122" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-122', 'Seq', 'link-4');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">seq_string</tt><tt class="py-op">,</tt> <tt id="link-123" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.alphabet=Bio.SearchIO._model.hsp.HSP-class.html#alphabet,Variable Bio.SearchIO._model.hsp.HSPFragment.alphabet=Bio.SearchIO._model.hsp.HSPFragment-class.html#alphabet"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-123', 'alphabet', 'link-123');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1041"></a><tt class="py-lineno">1041</tt>  <tt class="py-line">                           <tt id="link-124" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-124', 'id', 'link-24');">id</a></tt><tt class="py-op">=</tt><tt id="link-125" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-125', 'id', 'link-24');">id</a></tt><tt class="py-op">,</tt> <tt id="link-126" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-126', 'name', 'link-116');">name</a></tt><tt class="py-op">=</tt><tt id="link-127" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-127', 'name', 'link-116');">name</a></tt><tt class="py-op">,</tt> <tt id="link-128" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DescriptionConsumer.description()=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#description,Method Bio.Phylo.PhyloXMLIO.Writer.description()=Bio.Phylo.PhyloXMLIO.Writer-class.html#description,Variable Bio.SearchIO._model.hit.Hit.description=Bio.SearchIO._model.hit.Hit-class.html#description,Variable Bio.SearchIO._model.query.QueryResult.description=Bio.SearchIO._model.query.QueryResult-class.html#description"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-128', 'description', 'link-128');">description</a></tt><tt class="py-op">=</tt><tt class="py-name">descr</tt><tt class="py-op">)</tt> </tt>
<a name="L1042"></a><tt class="py-lineno">1042</tt>  <tt class="py-line">        <tt class="py-name">qualities</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">q_mapping</tt><tt class="py-op">[</tt><tt class="py-name">letter</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt class="py-name">letter</tt> <tt class="py-keyword">in</tt> <tt class="py-name">quality_string</tt><tt class="py-op">]</tt> </tt>
<a name="L1043"></a><tt class="py-lineno">1043</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">qualities</tt> <tt class="py-keyword">and</tt> <tt class="py-op">(</tt><tt id="link-129" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-129', 'min', 'link-26');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">qualities</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt> <tt class="py-keyword">or</tt> <tt id="link-130" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-130', 'max', 'link-19');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">qualities</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">93</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1044"></a><tt class="py-lineno">1044</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid character in quality string"</tt><tt class="py-op">)</tt> </tt>
<a name="L1045"></a><tt class="py-lineno">1045</tt>  <tt class="py-line">        <tt class="py-comment"># For speed, will now use a dirty trick to speed up assigning the</tt> </tt>
<a name="L1046"></a><tt class="py-lineno">1046</tt>  <tt class="py-line">        <tt class="py-comment"># qualities. We do this to bypass the length check imposed by the</tt> </tt>
<a name="L1047"></a><tt class="py-lineno">1047</tt>  <tt class="py-line">        <tt class="py-comment"># per-letter-annotations restricted dict (as this has already been</tt> </tt>
<a name="L1048"></a><tt class="py-lineno">1048</tt>  <tt class="py-line">        <tt class="py-comment"># checked by FastqGeneralIterator). This is equivalent to:</tt> </tt>
<a name="L1049"></a><tt class="py-lineno">1049</tt>  <tt class="py-line">        <tt class="py-comment"># record.letter_annotations["phred_quality"] = qualities</tt> </tt>
<a name="L1050"></a><tt class="py-lineno">1050</tt>  <tt class="py-line">        <tt class="py-name">dict</tt><tt class="py-op">.</tt><tt id="link-131" class="py-name" targets="Method Bio.Crystal.Chain.__setitem__()=Bio.Crystal.Chain-class.html#__setitem__,Method Bio.Crystal.Crystal.__setitem__()=Bio.Crystal.Crystal-class.html#__setitem__,Method Bio.Entrez.Parser.StructureElement.__setitem__()=Bio.Entrez.Parser.StructureElement-class.html#__setitem__,Method Bio.File._IndexedSeqFileDict.__setitem__()=Bio.File._IndexedSeqFileDict-class.html#__setitem__,Method Bio.Index._InMemoryIndex.__setitem__()=Bio.Index._InMemoryIndex-class.html#__setitem__,Method Bio.PDB.Entity.DisorderedEntityWrapper.__setitem__()=Bio.PDB.Entity.DisorderedEntityWrapper-class.html#__setitem__,Method Bio.PDB.Vector'.Vector.__setitem__()=Bio.PDB.Vector%27.Vector-class.html#__setitem__,Method Bio.Phylo.PhyloXML.Events.__setitem__()=Bio.Phylo.PhyloXML.Events-class.html#__setitem__,Method Bio.Phylo.TreeConstruction._DistanceMatrix.__setitem__()=Bio.Phylo.TreeConstruction._DistanceMatrix-class.html#__setitem__,Method Bio.Phylo.TreeConstruction._Matrix.__setitem__()=Bio.Phylo.TreeConstruction._Matrix-class.html#__setitem__,Method Bio.SearchIO._model.hit.Hit.__setitem__()=Bio.SearchIO._model.hit.Hit-class.html#__setitem__,Method Bio.SearchIO._model.hsp.HSP.__setitem__()=Bio.SearchIO._model.hsp.HSP-class.html#__setitem__,Method Bio.SearchIO._model.query.QueryResult.__setitem__()=Bio.SearchIO._model.query.QueryResult-class.html#__setitem__,Method Bio.Seq.MutableSeq.__setitem__()=Bio.Seq.MutableSeq-class.html#__setitem__,Method Bio.SeqRecord._RestrictedDict.__setitem__()=Bio.SeqRecord._RestrictedDict-class.html#__setitem__,Method Bio._py3k._ordereddict.OrderedDict.__setitem__()=Bio._py3k._ordereddict.OrderedDict-class.html#__setitem__"><a title="Bio.Crystal.Chain.__setitem__
Bio.Crystal.Crystal.__setitem__
Bio.Entrez.Parser.StructureElement.__setitem__
Bio.File._IndexedSeqFileDict.__setitem__
Bio.Index._InMemoryIndex.__setitem__
Bio.PDB.Entity.DisorderedEntityWrapper.__setitem__
Bio.PDB.Vector'.Vector.__setitem__
Bio.Phylo.PhyloXML.Events.__setitem__
Bio.Phylo.TreeConstruction._DistanceMatrix.__setitem__
Bio.Phylo.TreeConstruction._Matrix.__setitem__
Bio.SearchIO._model.hit.Hit.__setitem__
Bio.SearchIO._model.hsp.HSP.__setitem__
Bio.SearchIO._model.query.QueryResult.__setitem__
Bio.Seq.MutableSeq.__setitem__
Bio.SeqRecord._RestrictedDict.__setitem__
Bio._py3k._ordereddict.OrderedDict.__setitem__" class="py-name" href="#" onclick="return doclink('link-131', '__setitem__', 'link-131');">__setitem__</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">_per_letter_annotations</tt><tt class="py-op">,</tt> </tt>
<a name="L1051"></a><tt class="py-lineno">1051</tt>  <tt class="py-line">                         <tt class="py-string">"phred_quality"</tt><tt class="py-op">,</tt> <tt class="py-name">qualities</tt><tt class="py-op">)</tt> </tt>
<a name="L1052"></a><tt class="py-lineno">1052</tt>  <tt class="py-line">        <tt class="py-keyword">yield</tt> <tt class="py-name">record</tt> </tt>
</div><a name="L1053"></a><tt class="py-lineno">1053</tt>  <tt class="py-line"> </tt>
<a name="L1054"></a><tt class="py-lineno">1054</tt>  <tt class="py-line"> </tt>
<a name="FastqSolexaIterator"></a><div id="FastqSolexaIterator-def"><a name="L1055"></a><tt class="py-lineno">1055</tt> <a class="py-toggle" href="#" id="FastqSolexaIterator-toggle" onclick="return toggle('FastqSolexaIterator');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.QualityIO-module.html#FastqSolexaIterator">FastqSolexaIterator</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt id="link-132" class="py-name"><a title="Bio.Alphabet.single_letter_alphabet" class="py-name" href="#" onclick="return doclink('link-132', 'single_letter_alphabet', 'link-2');">single_letter_alphabet</a></tt><tt class="py-op">,</tt> <tt class="py-param">title2ids</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FastqSolexaIterator-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="FastqSolexaIterator-expanded"><a name="L1056"></a><tt class="py-lineno">1056</tt>  <tt class="py-line">    <tt class="py-docstring">r"""Parsing old Solexa/Illumina FASTQ like files (which differ in the quality mapping).</tt> </tt>
<a name="L1057"></a><tt class="py-lineno">1057</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1058"></a><tt class="py-lineno">1058</tt>  <tt class="py-line"><tt class="py-docstring">    The optional arguments are the same as those for the FastqPhredIterator.</tt> </tt>
<a name="L1059"></a><tt class="py-lineno">1059</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1060"></a><tt class="py-lineno">1060</tt>  <tt class="py-line"><tt class="py-docstring">    For each sequence in Solexa/Illumina FASTQ files there is a matching string</tt> </tt>
<a name="L1061"></a><tt class="py-lineno">1061</tt>  <tt class="py-line"><tt class="py-docstring">    encoding the Solexa integer qualities using ASCII values with an offset</tt> </tt>
<a name="L1062"></a><tt class="py-lineno">1062</tt>  <tt class="py-line"><tt class="py-docstring">    of 64.  Solexa scores are scaled differently to PHRED scores, and Biopython</tt> </tt>
<a name="L1063"></a><tt class="py-lineno">1063</tt>  <tt class="py-line"><tt class="py-docstring">    will NOT perform any automatic conversion when loading.</tt> </tt>
<a name="L1064"></a><tt class="py-lineno">1064</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1065"></a><tt class="py-lineno">1065</tt>  <tt class="py-line"><tt class="py-docstring">    NOTE - This file format is used by the OLD versions of the Solexa/Illumina</tt> </tt>
<a name="L1066"></a><tt class="py-lineno">1066</tt>  <tt class="py-line"><tt class="py-docstring">    pipeline. See also the FastqIlluminaIterator function for the NEW version.</tt> </tt>
<a name="L1067"></a><tt class="py-lineno">1067</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1068"></a><tt class="py-lineno">1068</tt>  <tt class="py-line"><tt class="py-docstring">    For example, consider a file containing these five records::</tt> </tt>
<a name="L1069"></a><tt class="py-lineno">1069</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1070"></a><tt class="py-lineno">1070</tt>  <tt class="py-line"><tt class="py-docstring">        @SLXA-B3_649_FC8437_R1_1_1_610_79</tt> </tt>
<a name="L1071"></a><tt class="py-lineno">1071</tt>  <tt class="py-line"><tt class="py-docstring">        GATGTGCAATACCTTTGTAGAGGAA</tt> </tt>
<a name="L1072"></a><tt class="py-lineno">1072</tt>  <tt class="py-line"><tt class="py-docstring">        +SLXA-B3_649_FC8437_R1_1_1_610_79</tt> </tt>
<a name="L1073"></a><tt class="py-lineno">1073</tt>  <tt class="py-line"><tt class="py-docstring">        YYYYYYYYYYYYYYYYYYWYWYYSU</tt> </tt>
<a name="L1074"></a><tt class="py-lineno">1074</tt>  <tt class="py-line"><tt class="py-docstring">        @SLXA-B3_649_FC8437_R1_1_1_397_389</tt> </tt>
<a name="L1075"></a><tt class="py-lineno">1075</tt>  <tt class="py-line"><tt class="py-docstring">        GGTTTGAGAAAGAGAAATGAGATAA</tt> </tt>
<a name="L1076"></a><tt class="py-lineno">1076</tt>  <tt class="py-line"><tt class="py-docstring">        +SLXA-B3_649_FC8437_R1_1_1_397_389</tt> </tt>
<a name="L1077"></a><tt class="py-lineno">1077</tt>  <tt class="py-line"><tt class="py-docstring">        YYYYYYYYYWYYYYWWYYYWYWYWW</tt> </tt>
<a name="L1078"></a><tt class="py-lineno">1078</tt>  <tt class="py-line"><tt class="py-docstring">        @SLXA-B3_649_FC8437_R1_1_1_850_123</tt> </tt>
<a name="L1079"></a><tt class="py-lineno">1079</tt>  <tt class="py-line"><tt class="py-docstring">        GAGGGTGTTGATCATGATGATGGCG</tt> </tt>
<a name="L1080"></a><tt class="py-lineno">1080</tt>  <tt class="py-line"><tt class="py-docstring">        +SLXA-B3_649_FC8437_R1_1_1_850_123</tt> </tt>
<a name="L1081"></a><tt class="py-lineno">1081</tt>  <tt class="py-line"><tt class="py-docstring">        YYYYYYYYYYYYYWYYWYYSYYYSY</tt> </tt>
<a name="L1082"></a><tt class="py-lineno">1082</tt>  <tt class="py-line"><tt class="py-docstring">        @SLXA-B3_649_FC8437_R1_1_1_362_549</tt> </tt>
<a name="L1083"></a><tt class="py-lineno">1083</tt>  <tt class="py-line"><tt class="py-docstring">        GGAAACAAAGTTTTTCTCAACATAG</tt> </tt>
<a name="L1084"></a><tt class="py-lineno">1084</tt>  <tt class="py-line"><tt class="py-docstring">        +SLXA-B3_649_FC8437_R1_1_1_362_549</tt> </tt>
<a name="L1085"></a><tt class="py-lineno">1085</tt>  <tt class="py-line"><tt class="py-docstring">        YYYYYYYYYYYYYYYYYYWWWWYWY</tt> </tt>
<a name="L1086"></a><tt class="py-lineno">1086</tt>  <tt class="py-line"><tt class="py-docstring">        @SLXA-B3_649_FC8437_R1_1_1_183_714</tt> </tt>
<a name="L1087"></a><tt class="py-lineno">1087</tt>  <tt class="py-line"><tt class="py-docstring">        GTATTATTTAATGGCATACACTCAA</tt> </tt>
<a name="L1088"></a><tt class="py-lineno">1088</tt>  <tt class="py-line"><tt class="py-docstring">        +SLXA-B3_649_FC8437_R1_1_1_183_714</tt> </tt>
<a name="L1089"></a><tt class="py-lineno">1089</tt>  <tt class="py-line"><tt class="py-docstring">        YYYYYYYYYYWYYYYWYWWUWWWQQ</tt> </tt>
<a name="L1090"></a><tt class="py-lineno">1090</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1091"></a><tt class="py-lineno">1091</tt>  <tt class="py-line"><tt class="py-docstring">    Using this module directly you might run:</tt> </tt>
<a name="L1092"></a><tt class="py-lineno">1092</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1093"></a><tt class="py-lineno">1093</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; with open("Quality/solexa_example.fastq", "rU") as handle:</tt> </tt>
<a name="L1094"></a><tt class="py-lineno">1094</tt>  <tt class="py-line"><tt class="py-docstring">    ...     for record in FastqSolexaIterator(handle):</tt> </tt>
<a name="L1095"></a><tt class="py-lineno">1095</tt>  <tt class="py-line"><tt class="py-docstring">    ...         print("%s %s" % (record.id, record.seq))</tt> </tt>
<a name="L1096"></a><tt class="py-lineno">1096</tt>  <tt class="py-line"><tt class="py-docstring">    SLXA-B3_649_FC8437_R1_1_1_610_79 GATGTGCAATACCTTTGTAGAGGAA</tt> </tt>
<a name="L1097"></a><tt class="py-lineno">1097</tt>  <tt class="py-line"><tt class="py-docstring">    SLXA-B3_649_FC8437_R1_1_1_397_389 GGTTTGAGAAAGAGAAATGAGATAA</tt> </tt>
<a name="L1098"></a><tt class="py-lineno">1098</tt>  <tt class="py-line"><tt class="py-docstring">    SLXA-B3_649_FC8437_R1_1_1_850_123 GAGGGTGTTGATCATGATGATGGCG</tt> </tt>
<a name="L1099"></a><tt class="py-lineno">1099</tt>  <tt class="py-line"><tt class="py-docstring">    SLXA-B3_649_FC8437_R1_1_1_362_549 GGAAACAAAGTTTTTCTCAACATAG</tt> </tt>
<a name="L1100"></a><tt class="py-lineno">1100</tt>  <tt class="py-line"><tt class="py-docstring">    SLXA-B3_649_FC8437_R1_1_1_183_714 GTATTATTTAATGGCATACACTCAA</tt> </tt>
<a name="L1101"></a><tt class="py-lineno">1101</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1102"></a><tt class="py-lineno">1102</tt>  <tt class="py-line"><tt class="py-docstring">    Typically however, you would call this via Bio.SeqIO instead with</tt> </tt>
<a name="L1103"></a><tt class="py-lineno">1103</tt>  <tt class="py-line"><tt class="py-docstring">    "fastq-solexa" as the format:</tt> </tt>
<a name="L1104"></a><tt class="py-lineno">1104</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1105"></a><tt class="py-lineno">1105</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L1106"></a><tt class="py-lineno">1106</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; with open("Quality/solexa_example.fastq", "rU") as handle:</tt> </tt>
<a name="L1107"></a><tt class="py-lineno">1107</tt>  <tt class="py-line"><tt class="py-docstring">    ...     for record in SeqIO.parse(handle, "fastq-solexa"):</tt> </tt>
<a name="L1108"></a><tt class="py-lineno">1108</tt>  <tt class="py-line"><tt class="py-docstring">    ...         print("%s %s" % (record.id, record.seq))</tt> </tt>
<a name="L1109"></a><tt class="py-lineno">1109</tt>  <tt class="py-line"><tt class="py-docstring">    SLXA-B3_649_FC8437_R1_1_1_610_79 GATGTGCAATACCTTTGTAGAGGAA</tt> </tt>
<a name="L1110"></a><tt class="py-lineno">1110</tt>  <tt class="py-line"><tt class="py-docstring">    SLXA-B3_649_FC8437_R1_1_1_397_389 GGTTTGAGAAAGAGAAATGAGATAA</tt> </tt>
<a name="L1111"></a><tt class="py-lineno">1111</tt>  <tt class="py-line"><tt class="py-docstring">    SLXA-B3_649_FC8437_R1_1_1_850_123 GAGGGTGTTGATCATGATGATGGCG</tt> </tt>
<a name="L1112"></a><tt class="py-lineno">1112</tt>  <tt class="py-line"><tt class="py-docstring">    SLXA-B3_649_FC8437_R1_1_1_362_549 GGAAACAAAGTTTTTCTCAACATAG</tt> </tt>
<a name="L1113"></a><tt class="py-lineno">1113</tt>  <tt class="py-line"><tt class="py-docstring">    SLXA-B3_649_FC8437_R1_1_1_183_714 GTATTATTTAATGGCATACACTCAA</tt> </tt>
<a name="L1114"></a><tt class="py-lineno">1114</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1115"></a><tt class="py-lineno">1115</tt>  <tt class="py-line"><tt class="py-docstring">    If you want to look at the qualities, they are recorded in each record's</tt> </tt>
<a name="L1116"></a><tt class="py-lineno">1116</tt>  <tt class="py-line"><tt class="py-docstring">    per-letter-annotation dictionary as a simple list of integers:</tt> </tt>
<a name="L1117"></a><tt class="py-lineno">1117</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1118"></a><tt class="py-lineno">1118</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(record.letter_annotations["solexa_quality"])</tt> </tt>
<a name="L1119"></a><tt class="py-lineno">1119</tt>  <tt class="py-line"><tt class="py-docstring">    [25, 25, 25, 25, 25, 25, 25, 25, 25, 25, 23, 25, 25, 25, 25, 23, 25, 23, 23, 21, 23, 23, 23, 17, 17]</tt> </tt>
<a name="L1120"></a><tt class="py-lineno">1120</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1121"></a><tt class="py-lineno">1121</tt>  <tt class="py-line"><tt class="py-docstring">    These scores aren't very good, but they are high enough that they map</tt> </tt>
<a name="L1122"></a><tt class="py-lineno">1122</tt>  <tt class="py-line"><tt class="py-docstring">    almost exactly onto PHRED scores:</tt> </tt>
<a name="L1123"></a><tt class="py-lineno">1123</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1124"></a><tt class="py-lineno">1124</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%0.2f" % phred_quality_from_solexa(25))</tt> </tt>
<a name="L1125"></a><tt class="py-lineno">1125</tt>  <tt class="py-line"><tt class="py-docstring">    25.01</tt> </tt>
<a name="L1126"></a><tt class="py-lineno">1126</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1127"></a><tt class="py-lineno">1127</tt>  <tt class="py-line"><tt class="py-docstring">    Let's look at faked example read which is even worse, where there are</tt> </tt>
<a name="L1128"></a><tt class="py-lineno">1128</tt>  <tt class="py-line"><tt class="py-docstring">    more noticeable differences between the Solexa and PHRED scores::</tt> </tt>
<a name="L1129"></a><tt class="py-lineno">1129</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1130"></a><tt class="py-lineno">1130</tt>  <tt class="py-line"><tt class="py-docstring">         @slxa_0001_1_0001_01</tt> </tt>
<a name="L1131"></a><tt class="py-lineno">1131</tt>  <tt class="py-line"><tt class="py-docstring">         ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTNNNNNN</tt> </tt>
<a name="L1132"></a><tt class="py-lineno">1132</tt>  <tt class="py-line"><tt class="py-docstring">         +slxa_0001_1_0001_01</tt> </tt>
<a name="L1133"></a><tt class="py-lineno">1133</tt>  <tt class="py-line"><tt class="py-docstring">         hgfedcba`_^]\[ZYXWVUTSRQPONMLKJIHGFEDCBA@?&gt;=&lt;;</tt> </tt>
<a name="L1134"></a><tt class="py-lineno">1134</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1135"></a><tt class="py-lineno">1135</tt>  <tt class="py-line"><tt class="py-docstring">    Again, you would typically use Bio.SeqIO to read this file in (rather than</tt> </tt>
<a name="L1136"></a><tt class="py-lineno">1136</tt>  <tt class="py-line"><tt class="py-docstring">    calling the Bio.SeqIO.QualtityIO module directly).  Most FASTQ files will</tt> </tt>
<a name="L1137"></a><tt class="py-lineno">1137</tt>  <tt class="py-line"><tt class="py-docstring">    contain thousands of reads, so you would normally use Bio.SeqIO.parse()</tt> </tt>
<a name="L1138"></a><tt class="py-lineno">1138</tt>  <tt class="py-line"><tt class="py-docstring">    as shown above.  This example has only as one entry, so instead we can</tt> </tt>
<a name="L1139"></a><tt class="py-lineno">1139</tt>  <tt class="py-line"><tt class="py-docstring">    use the Bio.SeqIO.read() function:</tt> </tt>
<a name="L1140"></a><tt class="py-lineno">1140</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1141"></a><tt class="py-lineno">1141</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L1142"></a><tt class="py-lineno">1142</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; with open("Quality/solexa_faked.fastq", "rU") as handle:</tt> </tt>
<a name="L1143"></a><tt class="py-lineno">1143</tt>  <tt class="py-line"><tt class="py-docstring">    ...     record = SeqIO.read(handle, "fastq-solexa")</tt> </tt>
<a name="L1144"></a><tt class="py-lineno">1144</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%s %s" % (record.id, record.seq))</tt> </tt>
<a name="L1145"></a><tt class="py-lineno">1145</tt>  <tt class="py-line"><tt class="py-docstring">    slxa_0001_1_0001_01 ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTNNNNNN</tt> </tt>
<a name="L1146"></a><tt class="py-lineno">1146</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(record.letter_annotations["solexa_quality"])</tt> </tt>
<a name="L1147"></a><tt class="py-lineno">1147</tt>  <tt class="py-line"><tt class="py-docstring">    [40, 39, 38, 37, 36, 35, 34, 33, 32, 31, 30, 29, 28, 27, 26, 25, 24, 23, 22, 21, 20, 19, 18, 17, 16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0, -1, -2, -3, -4, -5]</tt> </tt>
<a name="L1148"></a><tt class="py-lineno">1148</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1149"></a><tt class="py-lineno">1149</tt>  <tt class="py-line"><tt class="py-docstring">    These quality scores are so low that when converted from the Solexa scheme</tt> </tt>
<a name="L1150"></a><tt class="py-lineno">1150</tt>  <tt class="py-line"><tt class="py-docstring">    into PHRED scores they look quite different:</tt> </tt>
<a name="L1151"></a><tt class="py-lineno">1151</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1152"></a><tt class="py-lineno">1152</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%0.2f" % phred_quality_from_solexa(-1))</tt> </tt>
<a name="L1153"></a><tt class="py-lineno">1153</tt>  <tt class="py-line"><tt class="py-docstring">    2.54</tt> </tt>
<a name="L1154"></a><tt class="py-lineno">1154</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%0.2f" % phred_quality_from_solexa(-5))</tt> </tt>
<a name="L1155"></a><tt class="py-lineno">1155</tt>  <tt class="py-line"><tt class="py-docstring">    1.19</tt> </tt>
<a name="L1156"></a><tt class="py-lineno">1156</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1157"></a><tt class="py-lineno">1157</tt>  <tt class="py-line"><tt class="py-docstring">    Note you can use the Bio.SeqIO.write() function or the SeqRecord's format</tt> </tt>
<a name="L1158"></a><tt class="py-lineno">1158</tt>  <tt class="py-line"><tt class="py-docstring">    method to output the record(s):</tt> </tt>
<a name="L1159"></a><tt class="py-lineno">1159</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1160"></a><tt class="py-lineno">1160</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(record.format("fastq-solexa"))</tt> </tt>
<a name="L1161"></a><tt class="py-lineno">1161</tt>  <tt class="py-line"><tt class="py-docstring">    @slxa_0001_1_0001_01</tt> </tt>
<a name="L1162"></a><tt class="py-lineno">1162</tt>  <tt class="py-line"><tt class="py-docstring">    ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTNNNNNN</tt> </tt>
<a name="L1163"></a><tt class="py-lineno">1163</tt>  <tt class="py-line"><tt class="py-docstring">    +</tt> </tt>
<a name="L1164"></a><tt class="py-lineno">1164</tt>  <tt class="py-line"><tt class="py-docstring">    hgfedcba`_^]\[ZYXWVUTSRQPONMLKJIHGFEDCBA@?&gt;=&lt;;</tt> </tt>
<a name="L1165"></a><tt class="py-lineno">1165</tt>  <tt class="py-line"><tt class="py-docstring">    &lt;BLANKLINE&gt;</tt> </tt>
<a name="L1166"></a><tt class="py-lineno">1166</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1167"></a><tt class="py-lineno">1167</tt>  <tt class="py-line"><tt class="py-docstring">    Note this output is slightly different from the input file as Biopython</tt> </tt>
<a name="L1168"></a><tt class="py-lineno">1168</tt>  <tt class="py-line"><tt class="py-docstring">    has left out the optional repetition of the sequence identifier on the "+"</tt> </tt>
<a name="L1169"></a><tt class="py-lineno">1169</tt>  <tt class="py-line"><tt class="py-docstring">    line.  If you want the to use PHRED scores, use "fastq" or "qual" as the</tt> </tt>
<a name="L1170"></a><tt class="py-lineno">1170</tt>  <tt class="py-line"><tt class="py-docstring">    output format instead, and Biopython will do the conversion for you:</tt> </tt>
<a name="L1171"></a><tt class="py-lineno">1171</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1172"></a><tt class="py-lineno">1172</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(record.format("fastq"))</tt> </tt>
<a name="L1173"></a><tt class="py-lineno">1173</tt>  <tt class="py-line"><tt class="py-docstring">    @slxa_0001_1_0001_01</tt> </tt>
<a name="L1174"></a><tt class="py-lineno">1174</tt>  <tt class="py-line"><tt class="py-docstring">    ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTNNNNNN</tt> </tt>
<a name="L1175"></a><tt class="py-lineno">1175</tt>  <tt class="py-line"><tt class="py-docstring">    +</tt> </tt>
<a name="L1176"></a><tt class="py-lineno">1176</tt>  <tt class="py-line"><tt class="py-docstring">    IHGFEDCBA@?&gt;=&lt;;:9876543210/.-,++*)('&amp;&amp;%%$$##""</tt> </tt>
<a name="L1177"></a><tt class="py-lineno">1177</tt>  <tt class="py-line"><tt class="py-docstring">    &lt;BLANKLINE&gt;</tt> </tt>
<a name="L1178"></a><tt class="py-lineno">1178</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1179"></a><tt class="py-lineno">1179</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(record.format("qual"))</tt> </tt>
<a name="L1180"></a><tt class="py-lineno">1180</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;slxa_0001_1_0001_01</tt> </tt>
<a name="L1181"></a><tt class="py-lineno">1181</tt>  <tt class="py-line"><tt class="py-docstring">    40 39 38 37 36 35 34 33 32 31 30 29 28 27 26 25 24 23 22 21</tt> </tt>
<a name="L1182"></a><tt class="py-lineno">1182</tt>  <tt class="py-line"><tt class="py-docstring">    20 19 18 17 16 15 14 13 12 11 10 10 9 8 7 6 5 5 4 4 3 3 2 2</tt> </tt>
<a name="L1183"></a><tt class="py-lineno">1183</tt>  <tt class="py-line"><tt class="py-docstring">    1 1</tt> </tt>
<a name="L1184"></a><tt class="py-lineno">1184</tt>  <tt class="py-line"><tt class="py-docstring">    &lt;BLANKLINE&gt;</tt> </tt>
<a name="L1185"></a><tt class="py-lineno">1185</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1186"></a><tt class="py-lineno">1186</tt>  <tt class="py-line"><tt class="py-docstring">    As shown above, the poor quality Solexa reads have been mapped to the</tt> </tt>
<a name="L1187"></a><tt class="py-lineno">1187</tt>  <tt class="py-line"><tt class="py-docstring">    equivalent PHRED score (e.g. -5 to 1 as shown earlier).</tt> </tt>
<a name="L1188"></a><tt class="py-lineno">1188</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1189"></a><tt class="py-lineno">1189</tt>  <tt class="py-line">    <tt class="py-name">q_mapping</tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1190"></a><tt class="py-lineno">1190</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">letter</tt> <tt class="py-keyword">in</tt> <tt id="link-133" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-133', 'range', 'link-28');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">255</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1191"></a><tt class="py-lineno">1191</tt>  <tt class="py-line">        <tt class="py-name">q_mapping</tt><tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-name">letter</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">letter</tt> <tt class="py-op">-</tt> <tt id="link-134" class="py-name"><a title="Bio.SeqIO.QualityIO.SOLEXA_SCORE_OFFSET" class="py-name" href="#" onclick="return doclink('link-134', 'SOLEXA_SCORE_OFFSET', 'link-18');">SOLEXA_SCORE_OFFSET</a></tt> </tt>
<a name="L1192"></a><tt class="py-lineno">1192</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">title_line</tt><tt class="py-op">,</tt> <tt class="py-name">seq_string</tt><tt class="py-op">,</tt> <tt class="py-name">quality_string</tt> <tt class="py-keyword">in</tt> <tt id="link-135" class="py-name"><a title="Bio.SeqIO.QualityIO.FastqGeneralIterator" class="py-name" href="#" onclick="return doclink('link-135', 'FastqGeneralIterator', 'link-114');">FastqGeneralIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1193"></a><tt class="py-lineno">1193</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">title2ids</tt><tt class="py-op">:</tt> </tt>
<a name="L1194"></a><tt class="py-lineno">1194</tt>  <tt class="py-line">            <tt id="link-136" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-136', 'id', 'link-24');">id</a></tt><tt class="py-op">,</tt> <tt id="link-137" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-137', 'name', 'link-116');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">descr</tt> <tt class="py-op">=</tt> <tt class="py-name">title_line</tt> </tt>
<a name="L1195"></a><tt class="py-lineno">1195</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1196"></a><tt class="py-lineno">1196</tt>  <tt class="py-line">            <tt class="py-name">descr</tt> <tt class="py-op">=</tt> <tt class="py-name">title_line</tt> </tt>
<a name="L1197"></a><tt class="py-lineno">1197</tt>  <tt class="py-line">            <tt id="link-138" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-138', 'id', 'link-24');">id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">descr</tt><tt class="py-op">.</tt><tt id="link-139" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-139', 'split', 'link-118');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L1198"></a><tt class="py-lineno">1198</tt>  <tt class="py-line">            <tt id="link-140" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-140', 'name', 'link-116');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-141" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-141', 'id', 'link-24');">id</a></tt> </tt>
<a name="L1199"></a><tt class="py-lineno">1199</tt>  <tt class="py-line">        <tt class="py-name">record</tt> <tt class="py-op">=</tt> <tt id="link-142" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-142', 'SeqRecord', 'link-8');">SeqRecord</a></tt><tt class="py-op">(</tt><tt id="link-143" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-143', 'Seq', 'link-4');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">seq_string</tt><tt class="py-op">,</tt> <tt id="link-144" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-144', 'alphabet', 'link-123');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1200"></a><tt class="py-lineno">1200</tt>  <tt class="py-line">                           <tt id="link-145" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-145', 'id', 'link-24');">id</a></tt><tt class="py-op">=</tt><tt id="link-146" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-146', 'id', 'link-24');">id</a></tt><tt class="py-op">,</tt> <tt id="link-147" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-147', 'name', 'link-116');">name</a></tt><tt class="py-op">=</tt><tt id="link-148" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-148', 'name', 'link-116');">name</a></tt><tt class="py-op">,</tt> <tt id="link-149" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-149', 'description', 'link-128');">description</a></tt><tt class="py-op">=</tt><tt class="py-name">descr</tt><tt class="py-op">)</tt> </tt>
<a name="L1201"></a><tt class="py-lineno">1201</tt>  <tt class="py-line">        <tt class="py-name">qualities</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">q_mapping</tt><tt class="py-op">[</tt><tt class="py-name">letter</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt class="py-name">letter</tt> <tt class="py-keyword">in</tt> <tt class="py-name">quality_string</tt><tt class="py-op">]</tt> </tt>
<a name="L1202"></a><tt class="py-lineno">1202</tt>  <tt class="py-line">        <tt class="py-comment"># DO NOT convert these into PHRED qualities automatically!</tt> </tt>
<a name="L1203"></a><tt class="py-lineno">1203</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">qualities</tt> <tt class="py-keyword">and</tt> <tt class="py-op">(</tt><tt id="link-150" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-150', 'min', 'link-26');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">qualities</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-op">-</tt><tt class="py-number">5</tt> <tt class="py-keyword">or</tt> <tt id="link-151" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-151', 'max', 'link-19');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">qualities</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">62</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1204"></a><tt class="py-lineno">1204</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid character in quality string"</tt><tt class="py-op">)</tt> </tt>
<a name="L1205"></a><tt class="py-lineno">1205</tt>  <tt class="py-line">        <tt class="py-comment"># Dirty trick to speed up this line:</tt> </tt>
<a name="L1206"></a><tt class="py-lineno">1206</tt>  <tt class="py-line">        <tt class="py-comment"># record.letter_annotations["solexa_quality"] = qualities</tt> </tt>
<a name="L1207"></a><tt class="py-lineno">1207</tt>  <tt class="py-line">        <tt class="py-name">dict</tt><tt class="py-op">.</tt><tt id="link-152" class="py-name"><a title="Bio.Crystal.Chain.__setitem__
Bio.Crystal.Crystal.__setitem__
Bio.Entrez.Parser.StructureElement.__setitem__
Bio.File._IndexedSeqFileDict.__setitem__
Bio.Index._InMemoryIndex.__setitem__
Bio.PDB.Entity.DisorderedEntityWrapper.__setitem__
Bio.PDB.Vector'.Vector.__setitem__
Bio.Phylo.PhyloXML.Events.__setitem__
Bio.Phylo.TreeConstruction._DistanceMatrix.__setitem__
Bio.Phylo.TreeConstruction._Matrix.__setitem__
Bio.SearchIO._model.hit.Hit.__setitem__
Bio.SearchIO._model.hsp.HSP.__setitem__
Bio.SearchIO._model.query.QueryResult.__setitem__
Bio.Seq.MutableSeq.__setitem__
Bio.SeqRecord._RestrictedDict.__setitem__
Bio._py3k._ordereddict.OrderedDict.__setitem__" class="py-name" href="#" onclick="return doclink('link-152', '__setitem__', 'link-131');">__setitem__</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">_per_letter_annotations</tt><tt class="py-op">,</tt> </tt>
<a name="L1208"></a><tt class="py-lineno">1208</tt>  <tt class="py-line">                         <tt class="py-string">"solexa_quality"</tt><tt class="py-op">,</tt> <tt class="py-name">qualities</tt><tt class="py-op">)</tt> </tt>
<a name="L1209"></a><tt class="py-lineno">1209</tt>  <tt class="py-line">        <tt class="py-keyword">yield</tt> <tt class="py-name">record</tt> </tt>
</div><a name="L1210"></a><tt class="py-lineno">1210</tt>  <tt class="py-line"> </tt>
<a name="L1211"></a><tt class="py-lineno">1211</tt>  <tt class="py-line"> </tt>
<a name="FastqIlluminaIterator"></a><div id="FastqIlluminaIterator-def"><a name="L1212"></a><tt class="py-lineno">1212</tt> <a class="py-toggle" href="#" id="FastqIlluminaIterator-toggle" onclick="return toggle('FastqIlluminaIterator');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.QualityIO-module.html#FastqIlluminaIterator">FastqIlluminaIterator</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt id="link-153" class="py-name"><a title="Bio.Alphabet.single_letter_alphabet" class="py-name" href="#" onclick="return doclink('link-153', 'single_letter_alphabet', 'link-2');">single_letter_alphabet</a></tt><tt class="py-op">,</tt> <tt class="py-param">title2ids</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FastqIlluminaIterator-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="FastqIlluminaIterator-expanded"><a name="L1213"></a><tt class="py-lineno">1213</tt>  <tt class="py-line">    <tt class="py-docstring">"""Parse Illumina 1.3 to 1.7 FASTQ like files (which differ in the quality mapping).</tt> </tt>
<a name="L1214"></a><tt class="py-lineno">1214</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1215"></a><tt class="py-lineno">1215</tt>  <tt class="py-line"><tt class="py-docstring">    The optional arguments are the same as those for the FastqPhredIterator.</tt> </tt>
<a name="L1216"></a><tt class="py-lineno">1216</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1217"></a><tt class="py-lineno">1217</tt>  <tt class="py-line"><tt class="py-docstring">    For each sequence in Illumina 1.3+ FASTQ files there is a matching string</tt> </tt>
<a name="L1218"></a><tt class="py-lineno">1218</tt>  <tt class="py-line"><tt class="py-docstring">    encoding PHRED integer qualities using ASCII values with an offset of 64.</tt> </tt>
<a name="L1219"></a><tt class="py-lineno">1219</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1220"></a><tt class="py-lineno">1220</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L1221"></a><tt class="py-lineno">1221</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; record = SeqIO.read("Quality/illumina_faked.fastq", "fastq-illumina")</tt> </tt>
<a name="L1222"></a><tt class="py-lineno">1222</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%s %s" % (record.id, record.seq))</tt> </tt>
<a name="L1223"></a><tt class="py-lineno">1223</tt>  <tt class="py-line"><tt class="py-docstring">    Test ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTN</tt> </tt>
<a name="L1224"></a><tt class="py-lineno">1224</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; max(record.letter_annotations["phred_quality"])</tt> </tt>
<a name="L1225"></a><tt class="py-lineno">1225</tt>  <tt class="py-line"><tt class="py-docstring">    40</tt> </tt>
<a name="L1226"></a><tt class="py-lineno">1226</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; min(record.letter_annotations["phred_quality"])</tt> </tt>
<a name="L1227"></a><tt class="py-lineno">1227</tt>  <tt class="py-line"><tt class="py-docstring">    0</tt> </tt>
<a name="L1228"></a><tt class="py-lineno">1228</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1229"></a><tt class="py-lineno">1229</tt>  <tt class="py-line"><tt class="py-docstring">    NOTE - Older versions of the Solexa/Illumina pipeline encoded Solexa scores</tt> </tt>
<a name="L1230"></a><tt class="py-lineno">1230</tt>  <tt class="py-line"><tt class="py-docstring">    with an ASCII offset of 64. They are approximately equal but only for high</tt> </tt>
<a name="L1231"></a><tt class="py-lineno">1231</tt>  <tt class="py-line"><tt class="py-docstring">    quality reads. If you have an old Solexa/Illumina file with negative</tt> </tt>
<a name="L1232"></a><tt class="py-lineno">1232</tt>  <tt class="py-line"><tt class="py-docstring">    Solexa scores, and try and read this as an Illumina 1.3+ file it will fail:</tt> </tt>
<a name="L1233"></a><tt class="py-lineno">1233</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1234"></a><tt class="py-lineno">1234</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; record2 = SeqIO.read("Quality/solexa_faked.fastq", "fastq-illumina")</tt> </tt>
<a name="L1235"></a><tt class="py-lineno">1235</tt>  <tt class="py-line"><tt class="py-docstring">    Traceback (most recent call last):</tt> </tt>
<a name="L1236"></a><tt class="py-lineno">1236</tt>  <tt class="py-line"><tt class="py-docstring">       ...</tt> </tt>
<a name="L1237"></a><tt class="py-lineno">1237</tt>  <tt class="py-line"><tt class="py-docstring">    ValueError: Invalid character in quality string</tt> </tt>
<a name="L1238"></a><tt class="py-lineno">1238</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1239"></a><tt class="py-lineno">1239</tt>  <tt class="py-line"><tt class="py-docstring">    NOTE - True Sanger style FASTQ files use PHRED scores with an offset of 33.</tt> </tt>
<a name="L1240"></a><tt class="py-lineno">1240</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1241"></a><tt class="py-lineno">1241</tt>  <tt class="py-line">    <tt class="py-name">q_mapping</tt> <tt class="py-op">=</tt> <tt class="py-name">dict</tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1242"></a><tt class="py-lineno">1242</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">letter</tt> <tt class="py-keyword">in</tt> <tt id="link-154" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-154', 'range', 'link-28');">range</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-number">255</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1243"></a><tt class="py-lineno">1243</tt>  <tt class="py-line">        <tt class="py-name">q_mapping</tt><tt class="py-op">[</tt><tt class="py-name">chr</tt><tt class="py-op">(</tt><tt class="py-name">letter</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">letter</tt> <tt class="py-op">-</tt> <tt id="link-155" class="py-name"><a title="Bio.SeqIO.QualityIO.SOLEXA_SCORE_OFFSET" class="py-name" href="#" onclick="return doclink('link-155', 'SOLEXA_SCORE_OFFSET', 'link-18');">SOLEXA_SCORE_OFFSET</a></tt> </tt>
<a name="L1244"></a><tt class="py-lineno">1244</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">title_line</tt><tt class="py-op">,</tt> <tt class="py-name">seq_string</tt><tt class="py-op">,</tt> <tt class="py-name">quality_string</tt> <tt class="py-keyword">in</tt> <tt id="link-156" class="py-name"><a title="Bio.SeqIO.QualityIO.FastqGeneralIterator" class="py-name" href="#" onclick="return doclink('link-156', 'FastqGeneralIterator', 'link-114');">FastqGeneralIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1245"></a><tt class="py-lineno">1245</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">title2ids</tt><tt class="py-op">:</tt> </tt>
<a name="L1246"></a><tt class="py-lineno">1246</tt>  <tt class="py-line">            <tt id="link-157" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-157', 'id', 'link-24');">id</a></tt><tt class="py-op">,</tt> <tt id="link-158" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-158', 'name', 'link-116');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">descr</tt> <tt class="py-op">=</tt> <tt class="py-name">title2ids</tt><tt class="py-op">(</tt><tt class="py-name">title_line</tt><tt class="py-op">)</tt> </tt>
<a name="L1247"></a><tt class="py-lineno">1247</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1248"></a><tt class="py-lineno">1248</tt>  <tt class="py-line">            <tt class="py-name">descr</tt> <tt class="py-op">=</tt> <tt class="py-name">title_line</tt> </tt>
<a name="L1249"></a><tt class="py-lineno">1249</tt>  <tt class="py-line">            <tt id="link-159" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-159', 'id', 'link-24');">id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">descr</tt><tt class="py-op">.</tt><tt id="link-160" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-160', 'split', 'link-118');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L1250"></a><tt class="py-lineno">1250</tt>  <tt class="py-line">            <tt id="link-161" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-161', 'name', 'link-116');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-162" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-162', 'id', 'link-24');">id</a></tt> </tt>
<a name="L1251"></a><tt class="py-lineno">1251</tt>  <tt class="py-line">        <tt class="py-name">record</tt> <tt class="py-op">=</tt> <tt id="link-163" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-163', 'SeqRecord', 'link-8');">SeqRecord</a></tt><tt class="py-op">(</tt><tt id="link-164" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-164', 'Seq', 'link-4');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">seq_string</tt><tt class="py-op">,</tt> <tt id="link-165" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-165', 'alphabet', 'link-123');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1252"></a><tt class="py-lineno">1252</tt>  <tt class="py-line">                           <tt id="link-166" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-166', 'id', 'link-24');">id</a></tt><tt class="py-op">=</tt><tt id="link-167" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-167', 'id', 'link-24');">id</a></tt><tt class="py-op">,</tt> <tt id="link-168" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-168', 'name', 'link-116');">name</a></tt><tt class="py-op">=</tt><tt id="link-169" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-169', 'name', 'link-116');">name</a></tt><tt class="py-op">,</tt> <tt id="link-170" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-170', 'description', 'link-128');">description</a></tt><tt class="py-op">=</tt><tt class="py-name">descr</tt><tt class="py-op">)</tt> </tt>
<a name="L1253"></a><tt class="py-lineno">1253</tt>  <tt class="py-line">        <tt class="py-name">qualities</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">q_mapping</tt><tt class="py-op">[</tt><tt class="py-name">letter</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt class="py-name">letter</tt> <tt class="py-keyword">in</tt> <tt class="py-name">quality_string</tt><tt class="py-op">]</tt> </tt>
<a name="L1254"></a><tt class="py-lineno">1254</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">qualities</tt> <tt class="py-keyword">and</tt> <tt class="py-op">(</tt><tt id="link-171" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-171', 'min', 'link-26');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">qualities</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt> <tt class="py-keyword">or</tt> <tt id="link-172" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-172', 'max', 'link-19');">max</a></tt><tt class="py-op">(</tt><tt class="py-name">qualities</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">62</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1255"></a><tt class="py-lineno">1255</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid character in quality string"</tt><tt class="py-op">)</tt> </tt>
<a name="L1256"></a><tt class="py-lineno">1256</tt>  <tt class="py-line">        <tt class="py-comment"># Dirty trick to speed up this line:</tt> </tt>
<a name="L1257"></a><tt class="py-lineno">1257</tt>  <tt class="py-line">        <tt class="py-comment"># record.letter_annotations["phred_quality"] = qualities</tt> </tt>
<a name="L1258"></a><tt class="py-lineno">1258</tt>  <tt class="py-line">        <tt class="py-name">dict</tt><tt class="py-op">.</tt><tt id="link-173" class="py-name"><a title="Bio.Crystal.Chain.__setitem__
Bio.Crystal.Crystal.__setitem__
Bio.Entrez.Parser.StructureElement.__setitem__
Bio.File._IndexedSeqFileDict.__setitem__
Bio.Index._InMemoryIndex.__setitem__
Bio.PDB.Entity.DisorderedEntityWrapper.__setitem__
Bio.PDB.Vector'.Vector.__setitem__
Bio.Phylo.PhyloXML.Events.__setitem__
Bio.Phylo.TreeConstruction._DistanceMatrix.__setitem__
Bio.Phylo.TreeConstruction._Matrix.__setitem__
Bio.SearchIO._model.hit.Hit.__setitem__
Bio.SearchIO._model.hsp.HSP.__setitem__
Bio.SearchIO._model.query.QueryResult.__setitem__
Bio.Seq.MutableSeq.__setitem__
Bio.SeqRecord._RestrictedDict.__setitem__
Bio._py3k._ordereddict.OrderedDict.__setitem__" class="py-name" href="#" onclick="return doclink('link-173', '__setitem__', 'link-131');">__setitem__</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">_per_letter_annotations</tt><tt class="py-op">,</tt> </tt>
<a name="L1259"></a><tt class="py-lineno">1259</tt>  <tt class="py-line">                         <tt class="py-string">"phred_quality"</tt><tt class="py-op">,</tt> <tt class="py-name">qualities</tt><tt class="py-op">)</tt> </tt>
<a name="L1260"></a><tt class="py-lineno">1260</tt>  <tt class="py-line">        <tt class="py-keyword">yield</tt> <tt class="py-name">record</tt> </tt>
</div><a name="L1261"></a><tt class="py-lineno">1261</tt>  <tt class="py-line"> </tt>
<a name="L1262"></a><tt class="py-lineno">1262</tt>  <tt class="py-line"> </tt>
<a name="QualPhredIterator"></a><div id="QualPhredIterator-def"><a name="L1263"></a><tt class="py-lineno">1263</tt> <a class="py-toggle" href="#" id="QualPhredIterator-toggle" onclick="return toggle('QualPhredIterator');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.QualityIO-module.html#QualPhredIterator">QualPhredIterator</a><tt class="py-op">(</tt><tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt id="link-174" class="py-name"><a title="Bio.Alphabet.single_letter_alphabet" class="py-name" href="#" onclick="return doclink('link-174', 'single_letter_alphabet', 'link-2');">single_letter_alphabet</a></tt><tt class="py-op">,</tt> <tt class="py-param">title2ids</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="QualPhredIterator-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="QualPhredIterator-expanded"><a name="L1264"></a><tt class="py-lineno">1264</tt>  <tt class="py-line">    <tt class="py-docstring">"""For QUAL files which include PHRED quality scores, but no sequence.</tt> </tt>
<a name="L1265"></a><tt class="py-lineno">1265</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1266"></a><tt class="py-lineno">1266</tt>  <tt class="py-line"><tt class="py-docstring">    For example, consider this short QUAL file::</tt> </tt>
<a name="L1267"></a><tt class="py-lineno">1267</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1268"></a><tt class="py-lineno">1268</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;EAS54_6_R1_2_1_413_324</tt> </tt>
<a name="L1269"></a><tt class="py-lineno">1269</tt>  <tt class="py-line"><tt class="py-docstring">        26 26 18 26 26 26 26 26 26 26 26 26 26 26 26 22 26 26 26 26</tt> </tt>
<a name="L1270"></a><tt class="py-lineno">1270</tt>  <tt class="py-line"><tt class="py-docstring">        26 26 26 23 23</tt> </tt>
<a name="L1271"></a><tt class="py-lineno">1271</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;EAS54_6_R1_2_1_540_792</tt> </tt>
<a name="L1272"></a><tt class="py-lineno">1272</tt>  <tt class="py-line"><tt class="py-docstring">        26 26 26 26 26 26 26 26 26 26 26 22 26 26 26 26 26 12 26 26</tt> </tt>
<a name="L1273"></a><tt class="py-lineno">1273</tt>  <tt class="py-line"><tt class="py-docstring">        26 18 26 23 18</tt> </tt>
<a name="L1274"></a><tt class="py-lineno">1274</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;EAS54_6_R1_2_1_443_348</tt> </tt>
<a name="L1275"></a><tt class="py-lineno">1275</tt>  <tt class="py-line"><tt class="py-docstring">        26 26 26 26 26 26 26 26 26 26 26 24 26 22 26 26 13 22 26 18</tt> </tt>
<a name="L1276"></a><tt class="py-lineno">1276</tt>  <tt class="py-line"><tt class="py-docstring">        24 18 18 18 18</tt> </tt>
<a name="L1277"></a><tt class="py-lineno">1277</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1278"></a><tt class="py-lineno">1278</tt>  <tt class="py-line"><tt class="py-docstring">    Using this module directly you might run:</tt> </tt>
<a name="L1279"></a><tt class="py-lineno">1279</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1280"></a><tt class="py-lineno">1280</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; with open("Quality/example.qual", "rU") as handle:</tt> </tt>
<a name="L1281"></a><tt class="py-lineno">1281</tt>  <tt class="py-line"><tt class="py-docstring">    ...     for record in QualPhredIterator(handle):</tt> </tt>
<a name="L1282"></a><tt class="py-lineno">1282</tt>  <tt class="py-line"><tt class="py-docstring">    ...         print("%s %s" % (record.id, record.seq))</tt> </tt>
<a name="L1283"></a><tt class="py-lineno">1283</tt>  <tt class="py-line"><tt class="py-docstring">    EAS54_6_R1_2_1_413_324 ?????????????????????????</tt> </tt>
<a name="L1284"></a><tt class="py-lineno">1284</tt>  <tt class="py-line"><tt class="py-docstring">    EAS54_6_R1_2_1_540_792 ?????????????????????????</tt> </tt>
<a name="L1285"></a><tt class="py-lineno">1285</tt>  <tt class="py-line"><tt class="py-docstring">    EAS54_6_R1_2_1_443_348 ?????????????????????????</tt> </tt>
<a name="L1286"></a><tt class="py-lineno">1286</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1287"></a><tt class="py-lineno">1287</tt>  <tt class="py-line"><tt class="py-docstring">    Typically however, you would call this via Bio.SeqIO instead with "qual"</tt> </tt>
<a name="L1288"></a><tt class="py-lineno">1288</tt>  <tt class="py-line"><tt class="py-docstring">    as the format:</tt> </tt>
<a name="L1289"></a><tt class="py-lineno">1289</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1290"></a><tt class="py-lineno">1290</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L1291"></a><tt class="py-lineno">1291</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; with open("Quality/example.qual", "rU") as handle:</tt> </tt>
<a name="L1292"></a><tt class="py-lineno">1292</tt>  <tt class="py-line"><tt class="py-docstring">    ...     for record in SeqIO.parse(handle, "qual"):</tt> </tt>
<a name="L1293"></a><tt class="py-lineno">1293</tt>  <tt class="py-line"><tt class="py-docstring">    ...         print("%s %s" % (record.id, record.seq))</tt> </tt>
<a name="L1294"></a><tt class="py-lineno">1294</tt>  <tt class="py-line"><tt class="py-docstring">    EAS54_6_R1_2_1_413_324 ?????????????????????????</tt> </tt>
<a name="L1295"></a><tt class="py-lineno">1295</tt>  <tt class="py-line"><tt class="py-docstring">    EAS54_6_R1_2_1_540_792 ?????????????????????????</tt> </tt>
<a name="L1296"></a><tt class="py-lineno">1296</tt>  <tt class="py-line"><tt class="py-docstring">    EAS54_6_R1_2_1_443_348 ?????????????????????????</tt> </tt>
<a name="L1297"></a><tt class="py-lineno">1297</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1298"></a><tt class="py-lineno">1298</tt>  <tt class="py-line"><tt class="py-docstring">    Becase QUAL files don't contain the sequence string itself, the seq</tt> </tt>
<a name="L1299"></a><tt class="py-lineno">1299</tt>  <tt class="py-line"><tt class="py-docstring">    property is set to an UnknownSeq object.  As no alphabet was given, this</tt> </tt>
<a name="L1300"></a><tt class="py-lineno">1300</tt>  <tt class="py-line"><tt class="py-docstring">    has defaulted to a generic single letter alphabet and the character "?"</tt> </tt>
<a name="L1301"></a><tt class="py-lineno">1301</tt>  <tt class="py-line"><tt class="py-docstring">    used.</tt> </tt>
<a name="L1302"></a><tt class="py-lineno">1302</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1303"></a><tt class="py-lineno">1303</tt>  <tt class="py-line"><tt class="py-docstring">    By specifying a nucleotide alphabet, "N" is used instead:</tt> </tt>
<a name="L1304"></a><tt class="py-lineno">1304</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1305"></a><tt class="py-lineno">1305</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L1306"></a><tt class="py-lineno">1306</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio.Alphabet import generic_dna</tt> </tt>
<a name="L1307"></a><tt class="py-lineno">1307</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; with open("Quality/example.qual", "rU") as handle:</tt> </tt>
<a name="L1308"></a><tt class="py-lineno">1308</tt>  <tt class="py-line"><tt class="py-docstring">    ...     for record in SeqIO.parse(handle, "qual", alphabet=generic_dna):</tt> </tt>
<a name="L1309"></a><tt class="py-lineno">1309</tt>  <tt class="py-line"><tt class="py-docstring">    ...         print("%s %s" % (record.id, record.seq))</tt> </tt>
<a name="L1310"></a><tt class="py-lineno">1310</tt>  <tt class="py-line"><tt class="py-docstring">    EAS54_6_R1_2_1_413_324 NNNNNNNNNNNNNNNNNNNNNNNNN</tt> </tt>
<a name="L1311"></a><tt class="py-lineno">1311</tt>  <tt class="py-line"><tt class="py-docstring">    EAS54_6_R1_2_1_540_792 NNNNNNNNNNNNNNNNNNNNNNNNN</tt> </tt>
<a name="L1312"></a><tt class="py-lineno">1312</tt>  <tt class="py-line"><tt class="py-docstring">    EAS54_6_R1_2_1_443_348 NNNNNNNNNNNNNNNNNNNNNNNNN</tt> </tt>
<a name="L1313"></a><tt class="py-lineno">1313</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1314"></a><tt class="py-lineno">1314</tt>  <tt class="py-line"><tt class="py-docstring">    However, the quality scores themselves are available as a list of integers</tt> </tt>
<a name="L1315"></a><tt class="py-lineno">1315</tt>  <tt class="py-line"><tt class="py-docstring">    in each record's per-letter-annotation:</tt> </tt>
<a name="L1316"></a><tt class="py-lineno">1316</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1317"></a><tt class="py-lineno">1317</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(record.letter_annotations["phred_quality"])</tt> </tt>
<a name="L1318"></a><tt class="py-lineno">1318</tt>  <tt class="py-line"><tt class="py-docstring">    [26, 26, 26, 26, 26, 26, 26, 26, 26, 26, 26, 24, 26, 22, 26, 26, 13, 22, 26, 18, 24, 18, 18, 18, 18]</tt> </tt>
<a name="L1319"></a><tt class="py-lineno">1319</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1320"></a><tt class="py-lineno">1320</tt>  <tt class="py-line"><tt class="py-docstring">    You can still slice one of these SeqRecord objects with an UnknownSeq:</tt> </tt>
<a name="L1321"></a><tt class="py-lineno">1321</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1322"></a><tt class="py-lineno">1322</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; sub_record = record[5:10]</tt> </tt>
<a name="L1323"></a><tt class="py-lineno">1323</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print("%s %s" % (sub_record.id, sub_record.letter_annotations["phred_quality"]))</tt> </tt>
<a name="L1324"></a><tt class="py-lineno">1324</tt>  <tt class="py-line"><tt class="py-docstring">    EAS54_6_R1_2_1_443_348 [26, 26, 26, 26, 26]</tt> </tt>
<a name="L1325"></a><tt class="py-lineno">1325</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1326"></a><tt class="py-lineno">1326</tt>  <tt class="py-line"><tt class="py-docstring">    As of Biopython 1.59, this parser will accept files with negatives quality</tt> </tt>
<a name="L1327"></a><tt class="py-lineno">1327</tt>  <tt class="py-line"><tt class="py-docstring">    scores but will replace them with the lowest possible PHRED score of zero.</tt> </tt>
<a name="L1328"></a><tt class="py-lineno">1328</tt>  <tt class="py-line"><tt class="py-docstring">    This will trigger a warning, previously it raised a ValueError exception.</tt> </tt>
<a name="L1329"></a><tt class="py-lineno">1329</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1330"></a><tt class="py-lineno">1330</tt>  <tt class="py-line">    <tt class="py-comment"># Skip any text before the first record (e.g. blank lines, comments)</tt> </tt>
<a name="L1331"></a><tt class="py-lineno">1331</tt>  <tt class="py-line">    <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L1332"></a><tt class="py-lineno">1332</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-175" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-175', 'readline', 'link-103');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1333"></a><tt class="py-lineno">1333</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt> <tt class="py-op">==</tt> <tt class="py-string">""</tt><tt class="py-op">:</tt> </tt>
<a name="L1334"></a><tt class="py-lineno">1334</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt>  <tt class="py-comment"># Premature end of file, or just empty?</tt> </tt>
<a name="L1335"></a><tt class="py-lineno">1335</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"&gt;"</tt><tt class="py-op">:</tt> </tt>
<a name="L1336"></a><tt class="py-lineno">1336</tt>  <tt class="py-line">            <tt class="py-keyword">break</tt> </tt>
<a name="L1337"></a><tt class="py-lineno">1337</tt>  <tt class="py-line"> </tt>
<a name="L1338"></a><tt class="py-lineno">1338</tt>  <tt class="py-line">    <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L1339"></a><tt class="py-lineno">1339</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-string">"&gt;"</tt><tt class="py-op">:</tt> </tt>
<a name="L1340"></a><tt class="py-lineno">1340</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt> </tt>
<a name="L1341"></a><tt class="py-lineno">1341</tt>  <tt class="py-line">                <tt class="py-string">"Records in Fasta files should start with '&gt;' character"</tt><tt class="py-op">)</tt> </tt>
<a name="L1342"></a><tt class="py-lineno">1342</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">title2ids</tt><tt class="py-op">:</tt> </tt>
<a name="L1343"></a><tt class="py-lineno">1343</tt>  <tt class="py-line">            <tt id="link-176" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-176', 'id', 'link-24');">id</a></tt><tt class="py-op">,</tt> <tt id="link-177" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-177', 'name', 'link-116');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">descr</tt> <tt class="py-op">=</tt> <tt class="py-name">title2ids</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-178" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-178', 'rstrip', 'link-104');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1344"></a><tt class="py-lineno">1344</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1345"></a><tt class="py-lineno">1345</tt>  <tt class="py-line">            <tt class="py-name">descr</tt> <tt class="py-op">=</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-179" class="py-name"><a title="Bio.Seq.Seq.rstrip" class="py-name" href="#" onclick="return doclink('link-179', 'rstrip', 'link-104');">rstrip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1346"></a><tt class="py-lineno">1346</tt>  <tt class="py-line">            <tt id="link-180" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-180', 'id', 'link-24');">id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">descr</tt><tt class="py-op">.</tt><tt id="link-181" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-181', 'split', 'link-118');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L1347"></a><tt class="py-lineno">1347</tt>  <tt class="py-line">            <tt id="link-182" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-182', 'name', 'link-116');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-183" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-183', 'id', 'link-24');">id</a></tt> </tt>
<a name="L1348"></a><tt class="py-lineno">1348</tt>  <tt class="py-line"> </tt>
<a name="L1349"></a><tt class="py-lineno">1349</tt>  <tt class="py-line">        <tt class="py-name">qualities</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L1350"></a><tt class="py-lineno">1350</tt>  <tt class="py-line">        <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-184" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-184', 'readline', 'link-103');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1351"></a><tt class="py-lineno">1351</tt>  <tt class="py-line">        <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L1352"></a><tt class="py-lineno">1352</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L1353"></a><tt class="py-lineno">1353</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L1354"></a><tt class="py-lineno">1354</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-string">"&gt;"</tt><tt class="py-op">:</tt> </tt>
<a name="L1355"></a><tt class="py-lineno">1355</tt>  <tt class="py-line">                <tt class="py-keyword">break</tt> </tt>
<a name="L1356"></a><tt class="py-lineno">1356</tt>  <tt class="py-line">            <tt class="py-name">qualities</tt><tt class="py-op">.</tt><tt id="link-185" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.extend()=Bio.Align.MultipleSeqAlignment-class.html#extend,Method Bio.SCOP.Raf.SeqMap.extend()=Bio.SCOP.Raf.SeqMap-class.html#extend,Method Bio.Seq.MutableSeq.extend()=Bio.Seq.MutableSeq-class.html#extend"><a title="Bio.Align.MultipleSeqAlignment.extend
Bio.SCOP.Raf.SeqMap.extend
Bio.Seq.MutableSeq.extend" class="py-name" href="#" onclick="return doclink('link-185', 'extend', 'link-185');">extend</a></tt><tt class="py-op">(</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">word</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">word</tt> <tt class="py-keyword">in</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-186" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-186', 'split', 'link-118');">split</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1357"></a><tt class="py-lineno">1357</tt>  <tt class="py-line">            <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-187" class="py-name"><a title="Bio.File.UndoHandle.readline
Bio.bgzf.BgzfReader.readline" class="py-name" href="#" onclick="return doclink('link-187', 'readline', 'link-103');">readline</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1358"></a><tt class="py-lineno">1358</tt>  <tt class="py-line"> </tt>
<a name="L1359"></a><tt class="py-lineno">1359</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">qualities</tt> <tt class="py-keyword">and</tt> <tt id="link-188" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-188', 'min', 'link-26');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">qualities</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L1360"></a><tt class="py-lineno">1360</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">"Negative quality score %i found, "</tt> <tt class="py-op">+</tt> </tt>
<a name="L1361"></a><tt class="py-lineno">1361</tt>  <tt class="py-line">                           <tt class="py-string">"substituting PHRED zero instead."</tt><tt class="py-op">)</tt> </tt>
<a name="L1362"></a><tt class="py-lineno">1362</tt>  <tt class="py-line">                          <tt class="py-op">%</tt> <tt id="link-189" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.min" class="py-name" href="#" onclick="return doclink('link-189', 'min', 'link-26');">min</a></tt><tt class="py-op">(</tt><tt class="py-name">qualities</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-190" class="py-name"><a title="Bio.BiopythonParserWarning" class="py-name" href="#" onclick="return doclink('link-190', 'BiopythonParserWarning', 'link-16');">BiopythonParserWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1363"></a><tt class="py-lineno">1363</tt>  <tt class="py-line">            <tt class="py-name">qualities</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-191" class="py-name"><a title="Bio.motifs.matrix.PositionSpecificScoringMatrix.max" class="py-name" href="#" onclick="return doclink('link-191', 'max', 'link-19');">max</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">q</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">q</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qualities</tt><tt class="py-op">]</tt> </tt>
<a name="L1364"></a><tt class="py-lineno">1364</tt>  <tt class="py-line"> </tt>
<a name="L1365"></a><tt class="py-lineno">1365</tt>  <tt class="py-line">        <tt class="py-comment"># Return the record and then continue...</tt> </tt>
<a name="L1366"></a><tt class="py-lineno">1366</tt>  <tt class="py-line">        <tt class="py-name">record</tt> <tt class="py-op">=</tt> <tt id="link-192" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-192', 'SeqRecord', 'link-8');">SeqRecord</a></tt><tt class="py-op">(</tt><tt id="link-193" class="py-name"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-193', 'UnknownSeq', 'link-6');">UnknownSeq</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">qualities</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-194" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-194', 'alphabet', 'link-123');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L1367"></a><tt class="py-lineno">1367</tt>  <tt class="py-line">                           <tt id="link-195" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-195', 'id', 'link-24');">id</a></tt><tt class="py-op">=</tt><tt id="link-196" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-196', 'id', 'link-24');">id</a></tt><tt class="py-op">,</tt> <tt id="link-197" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-197', 'name', 'link-116');">name</a></tt><tt class="py-op">=</tt><tt id="link-198" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-198', 'name', 'link-116');">name</a></tt><tt class="py-op">,</tt> <tt id="link-199" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-199', 'description', 'link-128');">description</a></tt><tt class="py-op">=</tt><tt class="py-name">descr</tt><tt class="py-op">)</tt> </tt>
<a name="L1368"></a><tt class="py-lineno">1368</tt>  <tt class="py-line">        <tt class="py-comment"># Dirty trick to speed up this line:</tt> </tt>
<a name="L1369"></a><tt class="py-lineno">1369</tt>  <tt class="py-line">        <tt class="py-comment"># record.letter_annotations["phred_quality"] = qualities</tt> </tt>
<a name="L1370"></a><tt class="py-lineno">1370</tt>  <tt class="py-line">        <tt class="py-name">dict</tt><tt class="py-op">.</tt><tt id="link-200" class="py-name"><a title="Bio.Crystal.Chain.__setitem__
Bio.Crystal.Crystal.__setitem__
Bio.Entrez.Parser.StructureElement.__setitem__
Bio.File._IndexedSeqFileDict.__setitem__
Bio.Index._InMemoryIndex.__setitem__
Bio.PDB.Entity.DisorderedEntityWrapper.__setitem__
Bio.PDB.Vector'.Vector.__setitem__
Bio.Phylo.PhyloXML.Events.__setitem__
Bio.Phylo.TreeConstruction._DistanceMatrix.__setitem__
Bio.Phylo.TreeConstruction._Matrix.__setitem__
Bio.SearchIO._model.hit.Hit.__setitem__
Bio.SearchIO._model.hsp.HSP.__setitem__
Bio.SearchIO._model.query.QueryResult.__setitem__
Bio.Seq.MutableSeq.__setitem__
Bio.SeqRecord._RestrictedDict.__setitem__
Bio._py3k._ordereddict.OrderedDict.__setitem__" class="py-name" href="#" onclick="return doclink('link-200', '__setitem__', 'link-131');">__setitem__</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">_per_letter_annotations</tt><tt class="py-op">,</tt> </tt>
<a name="L1371"></a><tt class="py-lineno">1371</tt>  <tt class="py-line">                         <tt class="py-string">"phred_quality"</tt><tt class="py-op">,</tt> <tt class="py-name">qualities</tt><tt class="py-op">)</tt> </tt>
<a name="L1372"></a><tt class="py-lineno">1372</tt>  <tt class="py-line">        <tt class="py-keyword">yield</tt> <tt class="py-name">record</tt> </tt>
<a name="L1373"></a><tt class="py-lineno">1373</tt>  <tt class="py-line"> </tt>
<a name="L1374"></a><tt class="py-lineno">1374</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">line</tt><tt class="py-op">:</tt> </tt>
<a name="L1375"></a><tt class="py-lineno">1375</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt>  <tt class="py-comment"># StopIteration</tt> </tt>
<a name="L1376"></a><tt class="py-lineno">1376</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">False</tt><tt class="py-op">,</tt> <tt class="py-string">"Should not reach this line"</tt> </tt>
</div><a name="L1377"></a><tt class="py-lineno">1377</tt>  <tt class="py-line"> </tt>
<a name="L1378"></a><tt class="py-lineno">1378</tt>  <tt class="py-line"> </tt>
<a name="FastqPhredWriter"></a><div id="FastqPhredWriter-def"><a name="L1379"></a><tt class="py-lineno">1379</tt> <a class="py-toggle" href="#" id="FastqPhredWriter-toggle" onclick="return toggle('FastqPhredWriter');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.SeqIO.QualityIO.FastqPhredWriter-class.html">FastqPhredWriter</a><tt class="py-op">(</tt><tt class="py-base-class">SequentialSequenceWriter</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FastqPhredWriter-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="FastqPhredWriter-expanded"><a name="L1380"></a><tt class="py-lineno">1380</tt>  <tt class="py-line">    <tt class="py-docstring">"""Class to write standard FASTQ format files (using PHRED quality scores).</tt> </tt>
<a name="L1381"></a><tt class="py-lineno">1381</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1382"></a><tt class="py-lineno">1382</tt>  <tt class="py-line"><tt class="py-docstring">    Although you can use this class directly, you are strongly encouraged</tt> </tt>
<a name="L1383"></a><tt class="py-lineno">1383</tt>  <tt class="py-line"><tt class="py-docstring">    to use the Bio.SeqIO.write() function instead via the format name "fastq"</tt> </tt>
<a name="L1384"></a><tt class="py-lineno">1384</tt>  <tt class="py-line"><tt class="py-docstring">    or the alias "fastq-sanger".  For example, this code reads in a standard</tt> </tt>
<a name="L1385"></a><tt class="py-lineno">1385</tt>  <tt class="py-line"><tt class="py-docstring">    Sanger style FASTQ file (using PHRED scores) and re-saves it as another</tt> </tt>
<a name="L1386"></a><tt class="py-lineno">1386</tt>  <tt class="py-line"><tt class="py-docstring">    Sanger style FASTQ file:</tt> </tt>
<a name="L1387"></a><tt class="py-lineno">1387</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1388"></a><tt class="py-lineno">1388</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L1389"></a><tt class="py-lineno">1389</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; record_iterator = SeqIO.parse("Quality/example.fastq", "fastq")</tt> </tt>
<a name="L1390"></a><tt class="py-lineno">1390</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; with open("Quality/temp.fastq", "w") as out_handle:</tt> </tt>
<a name="L1391"></a><tt class="py-lineno">1391</tt>  <tt class="py-line"><tt class="py-docstring">    ...     SeqIO.write(record_iterator, out_handle, "fastq")</tt> </tt>
<a name="L1392"></a><tt class="py-lineno">1392</tt>  <tt class="py-line"><tt class="py-docstring">    3</tt> </tt>
<a name="L1393"></a><tt class="py-lineno">1393</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1394"></a><tt class="py-lineno">1394</tt>  <tt class="py-line"><tt class="py-docstring">    You might want to do this if the original file included extra line breaks,</tt> </tt>
<a name="L1395"></a><tt class="py-lineno">1395</tt>  <tt class="py-line"><tt class="py-docstring">    which while valid may not be supported by all tools.  The output file from</tt> </tt>
<a name="L1396"></a><tt class="py-lineno">1396</tt>  <tt class="py-line"><tt class="py-docstring">    Biopython will have each sequence on a single line, and each quality</tt> </tt>
<a name="L1397"></a><tt class="py-lineno">1397</tt>  <tt class="py-line"><tt class="py-docstring">    string on a single line (which is considered desirable for maximum</tt> </tt>
<a name="L1398"></a><tt class="py-lineno">1398</tt>  <tt class="py-line"><tt class="py-docstring">    compatibility).</tt> </tt>
<a name="L1399"></a><tt class="py-lineno">1399</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1400"></a><tt class="py-lineno">1400</tt>  <tt class="py-line"><tt class="py-docstring">    In this next example, an old style Solexa/Illumina FASTQ file (using Solexa</tt> </tt>
<a name="L1401"></a><tt class="py-lineno">1401</tt>  <tt class="py-line"><tt class="py-docstring">    quality scores) is converted into a standard Sanger style FASTQ file using</tt> </tt>
<a name="L1402"></a><tt class="py-lineno">1402</tt>  <tt class="py-line"><tt class="py-docstring">    PHRED qualities:</tt> </tt>
<a name="L1403"></a><tt class="py-lineno">1403</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1404"></a><tt class="py-lineno">1404</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L1405"></a><tt class="py-lineno">1405</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; record_iterator = SeqIO.parse("Quality/solexa_example.fastq", "fastq-solexa")</tt> </tt>
<a name="L1406"></a><tt class="py-lineno">1406</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; with open("Quality/temp.fastq", "w") as out_handle:</tt> </tt>
<a name="L1407"></a><tt class="py-lineno">1407</tt>  <tt class="py-line"><tt class="py-docstring">    ...     SeqIO.write(record_iterator, out_handle, "fastq")</tt> </tt>
<a name="L1408"></a><tt class="py-lineno">1408</tt>  <tt class="py-line"><tt class="py-docstring">    5</tt> </tt>
<a name="L1409"></a><tt class="py-lineno">1409</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1410"></a><tt class="py-lineno">1410</tt>  <tt class="py-line"><tt class="py-docstring">    This code is also called if you use the .format("fastq") method of a</tt> </tt>
<a name="L1411"></a><tt class="py-lineno">1411</tt>  <tt class="py-line"><tt class="py-docstring">    SeqRecord, or .format("fastq-sanger") if you prefer that alias.</tt> </tt>
<a name="L1412"></a><tt class="py-lineno">1412</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1413"></a><tt class="py-lineno">1413</tt>  <tt class="py-line"><tt class="py-docstring">    Note that Sanger FASTQ files have an upper limit of PHRED quality 93, which is</tt> </tt>
<a name="L1414"></a><tt class="py-lineno">1414</tt>  <tt class="py-line"><tt class="py-docstring">    encoded as ASCII 126, the tilde. If your quality scores are truncated to fit, a</tt> </tt>
<a name="L1415"></a><tt class="py-lineno">1415</tt>  <tt class="py-line"><tt class="py-docstring">    warning is issued.</tt> </tt>
<a name="L1416"></a><tt class="py-lineno">1416</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1417"></a><tt class="py-lineno">1417</tt>  <tt class="py-line"><tt class="py-docstring">    P.S. To avoid cluttering up your working directory, you can delete this</tt> </tt>
<a name="L1418"></a><tt class="py-lineno">1418</tt>  <tt class="py-line"><tt class="py-docstring">    temporary file now:</tt> </tt>
<a name="L1419"></a><tt class="py-lineno">1419</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1420"></a><tt class="py-lineno">1420</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; import os</tt> </tt>
<a name="L1421"></a><tt class="py-lineno">1421</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; os.remove("Quality/temp.fastq")</tt> </tt>
<a name="L1422"></a><tt class="py-lineno">1422</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1423"></a><tt class="py-lineno">1423</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt id="link-201" class="py-name"><a title="Bio.SeqIO.QualityIO.SANGER_SCORE_OFFSET" class="py-name" href="#" onclick="return doclink('link-201', 'SANGER_SCORE_OFFSET', 'link-17');">SANGER_SCORE_OFFSET</a></tt> <tt class="py-op">==</tt> <tt class="py-name">ord</tt><tt class="py-op">(</tt><tt class="py-string">"!"</tt><tt class="py-op">)</tt> </tt>
<a name="L1424"></a><tt class="py-lineno">1424</tt>  <tt class="py-line"> </tt>
<a name="FastqPhredWriter.write_record"></a><div id="FastqPhredWriter.write_record-def"><a name="L1425"></a><tt class="py-lineno">1425</tt> <a class="py-toggle" href="#" id="FastqPhredWriter.write_record-toggle" onclick="return toggle('FastqPhredWriter.write_record');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.QualityIO.FastqPhredWriter-class.html#write_record">write_record</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FastqPhredWriter.write_record-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="FastqPhredWriter.write_record-expanded"><a name="L1426"></a><tt class="py-lineno">1426</tt>  <tt class="py-line">        <tt class="py-docstring">"""Write a single FASTQ record to the file."""</tt> </tt>
<a name="L1427"></a><tt class="py-lineno">1427</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_header_written</tt> </tt>
<a name="L1428"></a><tt class="py-lineno">1428</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_footer_written</tt> </tt>
<a name="L1429"></a><tt class="py-lineno">1429</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_record_written</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L1430"></a><tt class="py-lineno">1430</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - Is an empty sequence allowed in FASTQ format?</tt> </tt>
<a name="L1431"></a><tt class="py-lineno">1431</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-202" class="py-name" targets="Variable Bio.SeqRecord.SeqRecord.seq=Bio.SeqRecord.SeqRecord-class.html#seq,Variable BioSQL.BioSeq.DBSeqRecord.seq=BioSQL.BioSeq.DBSeqRecord-class.html#seq"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-202', 'seq', 'link-202');">seq</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1432"></a><tt class="py-lineno">1432</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"No sequence for record %s"</tt> <tt class="py-op">%</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-203" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-203', 'id', 'link-24');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1433"></a><tt class="py-lineno">1433</tt>  <tt class="py-line">        <tt class="py-name">seq_str</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-204" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-204', 'seq', 'link-202');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1434"></a><tt class="py-lineno">1434</tt>  <tt class="py-line">        <tt class="py-name">qualities_str</tt> <tt class="py-op">=</tt> <tt id="link-205" class="py-name" targets="Function Bio.SeqIO.QualityIO._get_sanger_quality_str()=Bio.SeqIO.QualityIO-module.html#_get_sanger_quality_str"><a title="Bio.SeqIO.QualityIO._get_sanger_quality_str" class="py-name" href="#" onclick="return doclink('link-205', '_get_sanger_quality_str', 'link-205');">_get_sanger_quality_str</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L1435"></a><tt class="py-lineno">1435</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">qualities_str</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">seq_str</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1436"></a><tt class="py-lineno">1436</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Record %s has sequence length %i but %i quality scores"</tt> </tt>
<a name="L1437"></a><tt class="py-lineno">1437</tt>  <tt class="py-line">                             <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-206" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-206', 'id', 'link-24');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">seq_str</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">qualities_str</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1438"></a><tt class="py-lineno">1438</tt>  <tt class="py-line"> </tt>
<a name="L1439"></a><tt class="py-lineno">1439</tt>  <tt class="py-line">        <tt class="py-comment"># FASTQ files can include a description, just like FASTA files</tt> </tt>
<a name="L1440"></a><tt class="py-lineno">1440</tt>  <tt class="py-line">        <tt class="py-comment"># (at least, this is what the NCBI Short Read Archive does)</tt> </tt>
<a name="L1441"></a><tt class="py-lineno">1441</tt>  <tt class="py-line">        <tt id="link-207" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-207', 'id', 'link-24');">id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-208" class="py-name" targets="Method Bio.AlignIO.Interfaces.AlignmentWriter.clean()=Bio.AlignIO.Interfaces.AlignmentWriter-class.html#clean,Method Bio.SeqIO.Interfaces.SequenceWriter.clean()=Bio.SeqIO.Interfaces.SequenceWriter-class.html#clean"><a title="Bio.AlignIO.Interfaces.AlignmentWriter.clean
Bio.SeqIO.Interfaces.SequenceWriter.clean" class="py-name" href="#" onclick="return doclink('link-208', 'clean', 'link-208');">clean</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-209" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-209', 'id', 'link-24');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1442"></a><tt class="py-lineno">1442</tt>  <tt class="py-line">        <tt id="link-210" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-210', 'description', 'link-128');">description</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-211" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentWriter.clean
Bio.SeqIO.Interfaces.SequenceWriter.clean" class="py-name" href="#" onclick="return doclink('link-211', 'clean', 'link-208');">clean</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-212" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-212', 'description', 'link-128');">description</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1443"></a><tt class="py-lineno">1443</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-213" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-213', 'description', 'link-128');">description</a></tt> <tt class="py-keyword">and</tt> <tt id="link-214" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-214', 'description', 'link-128');">description</a></tt><tt class="py-op">.</tt><tt id="link-215" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-215', 'split', 'link-118');">split</a></tt><tt class="py-op">(</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt id="link-216" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-216', 'id', 'link-24');">id</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1444"></a><tt class="py-lineno">1444</tt>  <tt class="py-line">            <tt class="py-comment"># The description includes the id at the start</tt> </tt>
<a name="L1445"></a><tt class="py-lineno">1445</tt>  <tt class="py-line">            <tt id="link-217" class="py-name" targets="Method Bio.Blast.NCBIStandalone._AlignmentConsumer.title()=Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#title,Method Bio.GenBank._FeatureConsumer.title()=Bio.GenBank._FeatureConsumer-class.html#title,Method Bio.GenBank._RecordConsumer.title()=Bio.GenBank._RecordConsumer-class.html#title"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-217', 'title', 'link-217');">title</a></tt> <tt class="py-op">=</tt> <tt id="link-218" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-218', 'description', 'link-128');">description</a></tt> </tt>
<a name="L1446"></a><tt class="py-lineno">1446</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-219" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-219', 'description', 'link-128');">description</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1447"></a><tt class="py-lineno">1447</tt>  <tt class="py-line">            <tt id="link-220" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-220', 'title', 'link-217');">title</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"%s %s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-221" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-221', 'id', 'link-24');">id</a></tt><tt class="py-op">,</tt> <tt id="link-222" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-222', 'description', 'link-128');">description</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1448"></a><tt class="py-lineno">1448</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1449"></a><tt class="py-lineno">1449</tt>  <tt class="py-line">            <tt id="link-223" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-223', 'title', 'link-217');">title</a></tt> <tt class="py-op">=</tt> <tt id="link-224" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-224', 'id', 'link-24');">id</a></tt> </tt>
<a name="L1450"></a><tt class="py-lineno">1450</tt>  <tt class="py-line"> </tt>
<a name="L1451"></a><tt class="py-lineno">1451</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-225" class="py-name" targets="Function Bio.AlignIO.write()=Bio.AlignIO-module.html#write,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.write()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#write,Method Bio.NeuralNetwork.Gene.Pattern.PatternIO.write()=Bio.NeuralNetwork.Gene.Pattern.PatternIO-class.html#write,Method Bio.Phylo.CDAOIO.Writer.write()=Bio.Phylo.CDAOIO.Writer-class.html#write,Function Bio.Phylo.CDAOIO.write()=Bio.Phylo.CDAOIO-module.html#write,Method Bio.Phylo.NeXMLIO.Writer.write()=Bio.Phylo.NeXMLIO.Writer-class.html#write,Function Bio.Phylo.NeXMLIO.write()=Bio.Phylo.NeXMLIO-module.html#write,Method Bio.Phylo.NewickIO.Writer.write()=Bio.Phylo.NewickIO.Writer-class.html#write,Function Bio.Phylo.NewickIO.write()=Bio.Phylo.NewickIO-module.html#write,Function Bio.Phylo.NexusIO.write()=Bio.Phylo.NexusIO-module.html#write,Method Bio.Phylo.PhyloXMLIO.Writer.write()=Bio.Phylo.PhyloXMLIO.Writer-class.html#write,Function Bio.Phylo.PhyloXMLIO.write()=Bio.Phylo.PhyloXMLIO-module.html#write,Function Bio.Phylo._io.write()=Bio.Phylo._io-module.html#write,Function Bio.SearchIO.write()=Bio.SearchIO-module.html#write,Function Bio.SeqIO.write()=Bio.SeqIO-module.html#write,Method Bio.bgzf.BgzfWriter.write()=Bio.bgzf.BgzfWriter-class.html#write,Function Bio.motifs.jaspar.write()=Bio.motifs.jaspar-module.html#write,Function Bio.motifs.transfac.write()=Bio.motifs.transfac-module.html#write,Function Bio.motifs.write()=Bio.motifs-module.html#write"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-225', 'write', 'link-225');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"@%s\n%s\n+\n%s\n"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-226" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-226', 'title', 'link-217');">title</a></tt><tt class="py-op">,</tt> <tt class="py-name">seq_str</tt><tt class="py-op">,</tt> <tt class="py-name">qualities_str</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1452"></a><tt class="py-lineno">1452</tt>  <tt class="py-line"> </tt>
<a name="L1453"></a><tt class="py-lineno">1453</tt>  <tt class="py-line"> </tt>
<a name="QualPhredWriter"></a><div id="QualPhredWriter-def"><a name="L1454"></a><tt class="py-lineno">1454</tt> <a class="py-toggle" href="#" id="QualPhredWriter-toggle" onclick="return toggle('QualPhredWriter');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.SeqIO.QualityIO.QualPhredWriter-class.html">QualPhredWriter</a><tt class="py-op">(</tt><tt class="py-base-class">SequentialSequenceWriter</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="QualPhredWriter-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="QualPhredWriter-expanded"><a name="L1455"></a><tt class="py-lineno">1455</tt>  <tt class="py-line">    <tt class="py-docstring">"""Class to write QUAL format files (using PHRED quality scores).</tt> </tt>
<a name="L1456"></a><tt class="py-lineno">1456</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1457"></a><tt class="py-lineno">1457</tt>  <tt class="py-line"><tt class="py-docstring">    Although you can use this class directly, you are strongly encouraged</tt> </tt>
<a name="L1458"></a><tt class="py-lineno">1458</tt>  <tt class="py-line"><tt class="py-docstring">    to use the Bio.SeqIO.write() function instead.  For example, this code</tt> </tt>
<a name="L1459"></a><tt class="py-lineno">1459</tt>  <tt class="py-line"><tt class="py-docstring">    reads in a FASTQ file and saves the quality scores into a QUAL file:</tt> </tt>
<a name="L1460"></a><tt class="py-lineno">1460</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1461"></a><tt class="py-lineno">1461</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L1462"></a><tt class="py-lineno">1462</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; record_iterator = SeqIO.parse("Quality/example.fastq", "fastq")</tt> </tt>
<a name="L1463"></a><tt class="py-lineno">1463</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; with open("Quality/temp.qual", "w") as out_handle:</tt> </tt>
<a name="L1464"></a><tt class="py-lineno">1464</tt>  <tt class="py-line"><tt class="py-docstring">    ...     SeqIO.write(record_iterator, out_handle, "qual")</tt> </tt>
<a name="L1465"></a><tt class="py-lineno">1465</tt>  <tt class="py-line"><tt class="py-docstring">    3</tt> </tt>
<a name="L1466"></a><tt class="py-lineno">1466</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1467"></a><tt class="py-lineno">1467</tt>  <tt class="py-line"><tt class="py-docstring">    This code is also called if you use the .format("qual") method of a</tt> </tt>
<a name="L1468"></a><tt class="py-lineno">1468</tt>  <tt class="py-line"><tt class="py-docstring">    SeqRecord.</tt> </tt>
<a name="L1469"></a><tt class="py-lineno">1469</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1470"></a><tt class="py-lineno">1470</tt>  <tt class="py-line"><tt class="py-docstring">    P.S. Don't forget to clean up the temp file if you don't need it anymore:</tt> </tt>
<a name="L1471"></a><tt class="py-lineno">1471</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1472"></a><tt class="py-lineno">1472</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; import os</tt> </tt>
<a name="L1473"></a><tt class="py-lineno">1473</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; os.remove("Quality/temp.qual")</tt> </tt>
<a name="L1474"></a><tt class="py-lineno">1474</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="QualPhredWriter.__init__"></a><div id="QualPhredWriter.__init__-def"><a name="L1475"></a><tt class="py-lineno">1475</tt> <a class="py-toggle" href="#" id="QualPhredWriter.__init__-toggle" onclick="return toggle('QualPhredWriter.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.QualityIO.QualPhredWriter-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">handle</tt><tt class="py-op">,</tt> <tt class="py-param">wrap</tt><tt class="py-op">=</tt><tt class="py-number">60</tt><tt class="py-op">,</tt> <tt class="py-param">record2title</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="QualPhredWriter.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="QualPhredWriter.__init__-expanded"><a name="L1476"></a><tt class="py-lineno">1476</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a QUAL writer.</tt> </tt>
<a name="L1477"></a><tt class="py-lineno">1477</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1478"></a><tt class="py-lineno">1478</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L1479"></a><tt class="py-lineno">1479</tt>  <tt class="py-line"><tt class="py-docstring">         - handle - Handle to an output file, e.g. as returned</tt> </tt>
<a name="L1480"></a><tt class="py-lineno">1480</tt>  <tt class="py-line"><tt class="py-docstring">                    by open(filename, "w")</tt> </tt>
<a name="L1481"></a><tt class="py-lineno">1481</tt>  <tt class="py-line"><tt class="py-docstring">         - wrap   - Optional line length used to wrap sequence lines.</tt> </tt>
<a name="L1482"></a><tt class="py-lineno">1482</tt>  <tt class="py-line"><tt class="py-docstring">                    Defaults to wrapping the sequence at 60 characters</tt> </tt>
<a name="L1483"></a><tt class="py-lineno">1483</tt>  <tt class="py-line"><tt class="py-docstring">                    Use zero (or None) for no wrapping, giving a single</tt> </tt>
<a name="L1484"></a><tt class="py-lineno">1484</tt>  <tt class="py-line"><tt class="py-docstring">                    long line for the sequence.</tt> </tt>
<a name="L1485"></a><tt class="py-lineno">1485</tt>  <tt class="py-line"><tt class="py-docstring">         - record2title - Optional function to return the text to be</tt> </tt>
<a name="L1486"></a><tt class="py-lineno">1486</tt>  <tt class="py-line"><tt class="py-docstring">                    used for the title line of each record.  By default</tt> </tt>
<a name="L1487"></a><tt class="py-lineno">1487</tt>  <tt class="py-line"><tt class="py-docstring">                    a combination of the record.id and record.description</tt> </tt>
<a name="L1488"></a><tt class="py-lineno">1488</tt>  <tt class="py-line"><tt class="py-docstring">                    is used.  If the record.description starts with the</tt> </tt>
<a name="L1489"></a><tt class="py-lineno">1489</tt>  <tt class="py-line"><tt class="py-docstring">                    record.id, then just the record.description is used.</tt> </tt>
<a name="L1490"></a><tt class="py-lineno">1490</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1491"></a><tt class="py-lineno">1491</tt>  <tt class="py-line"><tt class="py-docstring">        The record2title argument is present for consistency with the</tt> </tt>
<a name="L1492"></a><tt class="py-lineno">1492</tt>  <tt class="py-line"><tt class="py-docstring">        Bio.SeqIO.FastaIO writer class.</tt> </tt>
<a name="L1493"></a><tt class="py-lineno">1493</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1494"></a><tt class="py-lineno">1494</tt>  <tt class="py-line">        <tt id="link-227" class="py-name"><a title="Bio.SeqIO.Interfaces.SequentialSequenceWriter" class="py-name" href="#" onclick="return doclink('link-227', 'SequentialSequenceWriter', 'link-13');">SequentialSequenceWriter</a></tt><tt class="py-op">.</tt><tt id="link-228" class="py-name" targets="Method Bio.Affy.CelFile.Record.__init__()=Bio.Affy.CelFile.Record-class.html#__init__,Method Bio.Align.AlignInfo.PSSM.__init__()=Bio.Align.AlignInfo.PSSM-class.html#__init__,Method Bio.Align.AlignInfo.SummaryInfo.__init__()=Bio.Align.AlignInfo.SummaryInfo-class.html#__init__,Method Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline.__init__()=Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline-class.html#__init__,Method Bio.Align.Applications._Clustalw.ClustalwCommandline.__init__()=Bio.Align.Applications._Clustalw.ClustalwCommandline-class.html#__init__,Method Bio.Align.Applications._Dialign.DialignCommandline.__init__()=Bio.Align.Applications._Dialign.DialignCommandline-class.html#__init__,Method Bio.Align.Applications._MSAProbs.MSAProbsCommandline.__init__()=Bio.Align.Applications._MSAProbs.MSAProbsCommandline-class.html#__init__,Method Bio.Align.Applications._Mafft.MafftCommandline.__init__()=Bio.Align.Applications._Mafft.MafftCommandline-class.html#__init__,Method Bio.Align.Applications._Muscle.MuscleCommandline.__init__()=Bio.Align.Applications._Muscle.MuscleCommandline-class.html#__init__,Method Bio.Align.Applications._Prank.PrankCommandline.__init__()=Bio.Align.Applications._Prank.PrankCommandline-class.html#__init__,Method Bio.Align.Applications._Probcons.ProbconsCommandline.__init__()=Bio.Align.Applications._Probcons.ProbconsCommandline-class.html#__init__,Method Bio.Align.Applications._TCoffee.TCoffeeCommandline.__init__()=Bio.Align.Applications._TCoffee.TCoffeeCommandline-class.html#__init__,Method Bio.Align.Generic.Alignment.__init__()=Bio.Align.Generic.Alignment-class.html#__init__,Method Bio.Align.MultipleSeqAlignment.__init__()=Bio.Align.MultipleSeqAlignment-class.html#__init__,Method Bio.AlignIO.Interfaces.AlignmentIterator.__init__()=Bio.AlignIO.Interfaces.AlignmentIterator-class.html#__init__,Method Bio.AlignIO.Interfaces.AlignmentWriter.__init__()=Bio.AlignIO.Interfaces.AlignmentWriter-class.html#__init__,Method Bio.AlignIO.Interfaces.SequentialAlignmentWriter.__init__()=Bio.AlignIO.Interfaces.SequentialAlignmentWriter-class.html#__init__,Method Bio.Alphabet.AlphabetEncoder.__init__()=Bio.Alphabet.AlphabetEncoder-class.html#__init__,Method Bio.Alphabet.Gapped.__init__()=Bio.Alphabet.Gapped-class.html#__init__,Method Bio.Alphabet.HasStopCodon.__init__()=Bio.Alphabet.HasStopCodon-class.html#__init__,Method Bio.Application.AbstractCommandline.__init__()=Bio.Application.AbstractCommandline-class.html#__init__,Method Bio.Application.ApplicationError.__init__()=Bio.Application.ApplicationError-class.html#__init__,Method Bio.Application._AbstractParameter.__init__()=Bio.Application._AbstractParameter-class.html#__init__,Method Bio.Application._Argument.__init__()=Bio.Application._Argument-class.html#__init__,Method Bio.Application._Option.__init__()=Bio.Application._Option-class.html#__init__,Method Bio.Application._StaticArgument.__init__()=Bio.Application._StaticArgument-class.html#__init__,Method Bio.Application._Switch.__init__()=Bio.Application._Switch-class.html#__init__,Method Bio.Blast.Applications.NcbiblastformatterCommandline.__init__()=Bio.Blast.Applications.NcbiblastformatterCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastnCommandline.__init__()=Bio.Blast.Applications.NcbiblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastpCommandline.__init__()=Bio.Blast.Applications.NcbiblastpCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbiblastxCommandline.__init__()=Bio.Blast.Applications.NcbiblastxCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbideltablastCommandline.__init__()=Bio.Blast.Applications.NcbideltablastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbipsiblastCommandline.__init__()=Bio.Blast.Applications.NcbipsiblastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbirpsblastCommandline.__init__()=Bio.Blast.Applications.NcbirpsblastCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbirpstblastnCommandline.__init__()=Bio.Blast.Applications.NcbirpstblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbitblastnCommandline.__init__()=Bio.Blast.Applications.NcbitblastnCommandline-class.html#__init__,Method Bio.Blast.Applications.NcbitblastxCommandline.__init__()=Bio.Blast.Applications.NcbitblastxCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbibaseblastCommandline.__init__()=Bio.Blast.Applications._NcbibaseblastCommandline-class.html#__init__,Method Bio.Blast.Applications._Ncbiblast2SeqCommandline.__init__()=Bio.Blast.Applications._Ncbiblast2SeqCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbiblastCommandline.__init__()=Bio.Blast.Applications._NcbiblastCommandline-class.html#__init__,Method Bio.Blast.Applications._NcbiblastMain2SeqCommandline.__init__()=Bio.Blast.Applications._NcbiblastMain2SeqCommandline-class.html#__init__,Method Bio.Blast.NCBIStandalone.BlastErrorParser.__init__()=Bio.Blast.NCBIStandalone.BlastErrorParser-class.html#__init__,Method Bio.Blast.NCBIStandalone.BlastParser.__init__()=Bio.Blast.NCBIStandalone.BlastParser-class.html#__init__,Method Bio.Blast.NCBIStandalone.Iterator.__init__()=Bio.Blast.NCBIStandalone.Iterator-class.html#__init__,Method Bio.Blast.NCBIStandalone.PSIBlastParser.__init__()=Bio.Blast.NCBIStandalone.PSIBlastParser-class.html#__init__,Method Bio.Blast.NCBIStandalone._BlastConsumer.__init__()=Bio.Blast.NCBIStandalone._BlastConsumer-class.html#__init__,Method Bio.Blast.NCBIStandalone._BlastErrorConsumer.__init__()=Bio.Blast.NCBIStandalone._BlastErrorConsumer-class.html#__init__,Method Bio.Blast.NCBIStandalone._PSIBlastConsumer.__init__()=Bio.Blast.NCBIStandalone._PSIBlastConsumer-class.html#__init__,Method Bio.Blast.NCBIXML.BlastParser.__init__()=Bio.Blast.NCBIXML.BlastParser-class.html#__init__,Method Bio.Blast.NCBIXML._XMLparser.__init__()=Bio.Blast.NCBIXML._XMLparser-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableEntry.__init__()=Bio.Blast.ParseBlastTable.BlastTableEntry-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableReader.__init__()=Bio.Blast.ParseBlastTable.BlastTableReader-class.html#__init__,Method Bio.Blast.ParseBlastTable.BlastTableRec.__init__()=Bio.Blast.ParseBlastTable.BlastTableRec-class.html#__init__,Method Bio.Blast.Record.Alignment.__init__()=Bio.Blast.Record.Alignment-class.html#__init__,Method Bio.Blast.Record.Blast.__init__()=Bio.Blast.Record.Blast-class.html#__init__,Method Bio.Blast.Record.DatabaseReport.__init__()=Bio.Blast.Record.DatabaseReport-class.html#__init__,Method Bio.Blast.Record.Description.__init__()=Bio.Blast.Record.Description-class.html#__init__,Method Bio.Blast.Record.HSP.__init__()=Bio.Blast.Record.HSP-class.html#__init__,Method Bio.Blast.Record.Header.__init__()=Bio.Blast.Record.Header-class.html#__init__,Method Bio.Blast.Record.MultipleAlignment.__init__()=Bio.Blast.Record.MultipleAlignment-class.html#__init__,Method Bio.Blast.Record.PSIBlast.__init__()=Bio.Blast.Record.PSIBlast-class.html#__init__,Method Bio.Blast.Record.Parameters.__init__()=Bio.Blast.Record.Parameters-class.html#__init__,Method Bio.Blast.Record.Round.__init__()=Bio.Blast.Record.Round-class.html#__init__,Method Bio.CAPS.CAPSMap.__init__()=Bio.CAPS.CAPSMap-class.html#__init__,Method Bio.CAPS.DifferentialCutsite.__init__()=Bio.CAPS.DifferentialCutsite-class.html#__init__,Method Bio.Cluster.Record.__init__()=Bio.Cluster.Record-class.html#__init__,Method Bio.Compass.Record.__init__()=Bio.Compass.Record-class.html#__init__,Method Bio.Crystal.Chain.__init__()=Bio.Crystal.Chain-class.html#__init__,Method Bio.Crystal.Crystal.__init__()=Bio.Crystal.Crystal-class.html#__init__,Method Bio.Crystal.Hetero.__init__()=Bio.Crystal.Hetero-class.html#__init__,Method Bio.Data.CodonTable.AmbiguousCodonTable.__init__()=Bio.Data.CodonTable.AmbiguousCodonTable-class.html#__init__,Method Bio.Data.CodonTable.AmbiguousForwardTable.__init__()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#__init__,Method Bio.Data.CodonTable.CodonTable.__init__()=Bio.Data.CodonTable.CodonTable-class.html#__init__,Method Bio.Data.CodonTable.NCBICodonTable.__init__()=Bio.Data.CodonTable.NCBICodonTable-class.html#__init__,Method Bio.DocSQL.Create.__init__()=Bio.DocSQL.Create-class.html#__init__,Method Bio.DocSQL.Insert.__init__()=Bio.DocSQL.Insert-class.html#__init__,Method Bio.DocSQL.IterationCursor.__init__()=Bio.DocSQL.IterationCursor-class.html#__init__,Method Bio.DocSQL.Query.__init__()=Bio.DocSQL.Query-class.html#__init__,Method Bio.DocSQL.QueryAll.__init__()=Bio.DocSQL.QueryAll-class.html#__init__,Method Bio.DocSQL.QueryGeneric.__init__()=Bio.DocSQL.QueryGeneric-class.html#__init__,Method Bio.DocSQL.QueryRow.__init__()=Bio.DocSQL.QueryRow-class.html#__init__,Method Bio.DocSQL.QuerySingle.__init__()=Bio.DocSQL.QuerySingle-class.html#__init__,Method Bio.Emboss.Applications.DiffseqCommandline.__init__()=Bio.Emboss.Applications.DiffseqCommandline-class.html#__init__,Method Bio.Emboss.Applications.EInvertedCommandline.__init__()=Bio.Emboss.Applications.EInvertedCommandline-class.html#__init__,Method Bio.Emboss.Applications.ETandemCommandline.__init__()=Bio.Emboss.Applications.ETandemCommandline-class.html#__init__,Method Bio.Emboss.Applications.Est2GenomeCommandline.__init__()=Bio.Emboss.Applications.Est2GenomeCommandline-class.html#__init__,Method Bio.Emboss.Applications.FConsenseCommandline.__init__()=Bio.Emboss.Applications.FConsenseCommandline-class.html#__init__,Method Bio.Emboss.Applications.FDNADistCommandline.__init__()=Bio.Emboss.Applications.FDNADistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FDNAParsCommandline.__init__()=Bio.Emboss.Applications.FDNAParsCommandline-class.html#__init__,Method Bio.Emboss.Applications.FNeighborCommandline.__init__()=Bio.Emboss.Applications.FNeighborCommandline-class.html#__init__,Method Bio.Emboss.Applications.FProtDistCommandline.__init__()=Bio.Emboss.Applications.FProtDistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FProtParsCommandline.__init__()=Bio.Emboss.Applications.FProtParsCommandline-class.html#__init__,Method Bio.Emboss.Applications.FSeqBootCommandline.__init__()=Bio.Emboss.Applications.FSeqBootCommandline-class.html#__init__,Method Bio.Emboss.Applications.FTreeDistCommandline.__init__()=Bio.Emboss.Applications.FTreeDistCommandline-class.html#__init__,Method Bio.Emboss.Applications.FuzznucCommandline.__init__()=Bio.Emboss.Applications.FuzznucCommandline-class.html#__init__,Method Bio.Emboss.Applications.IepCommandline.__init__()=Bio.Emboss.Applications.IepCommandline-class.html#__init__,Method Bio.Emboss.Applications.NeedleCommandline.__init__()=Bio.Emboss.Applications.NeedleCommandline-class.html#__init__,Method Bio.Emboss.Applications.NeedleallCommandline.__init__()=Bio.Emboss.Applications.NeedleallCommandline-class.html#__init__,Method Bio.Emboss.Applications.PalindromeCommandline.__init__()=Bio.Emboss.Applications.PalindromeCommandline-class.html#__init__,Method Bio.Emboss.Applications.Primer3Commandline.__init__()=Bio.Emboss.Applications.Primer3Commandline-class.html#__init__,Method Bio.Emboss.Applications.PrimerSearchCommandline.__init__()=Bio.Emboss.Applications.PrimerSearchCommandline-class.html#__init__,Method Bio.Emboss.Applications.SeqmatchallCommandline.__init__()=Bio.Emboss.Applications.SeqmatchallCommandline-class.html#__init__,Method Bio.Emboss.Applications.SeqretCommandline.__init__()=Bio.Emboss.Applications.SeqretCommandline-class.html#__init__,Method Bio.Emboss.Applications.StretcherCommandline.__init__()=Bio.Emboss.Applications.StretcherCommandline-class.html#__init__,Method Bio.Emboss.Applications.TranalignCommandline.__init__()=Bio.Emboss.Applications.TranalignCommandline-class.html#__init__,Method Bio.Emboss.Applications.WaterCommandline.__init__()=Bio.Emboss.Applications.WaterCommandline-class.html#__init__,Method Bio.Emboss.Applications._EmbossCommandLine.__init__()=Bio.Emboss.Applications._EmbossCommandLine-class.html#__init__,Method Bio.Emboss.Applications._EmbossMinimalCommandLine.__init__()=Bio.Emboss.Applications._EmbossMinimalCommandLine-class.html#__init__,Method Bio.Emboss.Primer3.Primers.__init__()=Bio.Emboss.Primer3.Primers-class.html#__init__,Method Bio.Emboss.Primer3.Record.__init__()=Bio.Emboss.Primer3.Record-class.html#__init__,Method Bio.Emboss.PrimerSearch.Amplifier.__init__()=Bio.Emboss.PrimerSearch.Amplifier-class.html#__init__,Method Bio.Emboss.PrimerSearch.InputRecord.__init__()=Bio.Emboss.PrimerSearch.InputRecord-class.html#__init__,Method Bio.Emboss.PrimerSearch.OutputRecord.__init__()=Bio.Emboss.PrimerSearch.OutputRecord-class.html#__init__,Method Bio.Entrez.Parser.CorruptedXMLError.__init__()=Bio.Entrez.Parser.CorruptedXMLError-class.html#__init__,Method Bio.Entrez.Parser.DataHandler.__init__()=Bio.Entrez.Parser.DataHandler-class.html#__init__,Method Bio.Entrez.Parser.NotXMLError.__init__()=Bio.Entrez.Parser.NotXMLError-class.html#__init__,Method Bio.Entrez.Parser.StructureElement.__init__()=Bio.Entrez.Parser.StructureElement-class.html#__init__,Method Bio.Entrez.Parser.ValidationError.__init__()=Bio.Entrez.Parser.ValidationError-class.html#__init__,Method Bio.ExPASy.Enzyme.Record.__init__()=Bio.ExPASy.Enzyme.Record-class.html#__init__,Method Bio.ExPASy.Prodoc.Record.__init__()=Bio.ExPASy.Prodoc.Record-class.html#__init__,Method Bio.ExPASy.Prodoc.Reference.__init__()=Bio.ExPASy.Prodoc.Reference-class.html#__init__,Method Bio.ExPASy.Prosite.Record.__init__()=Bio.ExPASy.Prosite.Record-class.html#__init__,Method Bio.ExPASy.ScanProsite.ContentHandler.__init__()=Bio.ExPASy.ScanProsite.ContentHandler-class.html#__init__,Method Bio.ExPASy.ScanProsite.Parser.__init__()=Bio.ExPASy.ScanProsite.Parser-class.html#__init__,Method Bio.ExPASy.ScanProsite.Record.__init__()=Bio.ExPASy.ScanProsite.Record-class.html#__init__,Method Bio.FSSP.FSSPAlignDict.__init__()=Bio.FSSP.FSSPAlignDict-class.html#__init__,Method Bio.FSSP.FSSPAlignRec.__init__()=Bio.FSSP.FSSPAlignRec-class.html#__init__,Method Bio.FSSP.FSSPHeader.__init__()=Bio.FSSP.FSSPHeader-class.html#__init__,Method Bio.FSSP.FSSPSumRec.__init__()=Bio.FSSP.FSSPSumRec-class.html#__init__,Method Bio.FSSP.FSSPTools.FSSPMultAlign.__init__()=Bio.FSSP.FSSPTools.FSSPMultAlign-class.html#__init__,Method Bio.FSSP.PosAlign.__init__()=Bio.FSSP.PosAlign-class.html#__init__,Method Bio.FSSP.fssp_rec.fff_rec.__init__()=Bio.FSSP.fssp_rec.fff_rec-class.html#__init__,Method Bio.File.UndoHandle.__init__()=Bio.File.UndoHandle-class.html#__init__,Method Bio.File._IndexedSeqFileDict.__init__()=Bio.File._IndexedSeqFileDict-class.html#__init__,Method Bio.File._SQLiteManySeqFilesDict.__init__()=Bio.File._SQLiteManySeqFilesDict-class.html#__init__,Method Bio.GA.Crossover.General.SafeFitnessCrossover.__init__()=Bio.GA.Crossover.General.SafeFitnessCrossover-class.html#__init__,Method Bio.GA.Crossover.GeneralPoint.GeneralPointCrossover.__init__()=Bio.GA.Crossover.GeneralPoint.GeneralPointCrossover-class.html#__init__,Method Bio.GA.Crossover.GeneralPoint.InterleaveCrossover.__init__()=Bio.GA.Crossover.GeneralPoint.InterleaveCrossover-class.html#__init__,Method Bio.GA.Crossover.Point.SinglePointCrossover.__init__()=Bio.GA.Crossover.Point.SinglePointCrossover-class.html#__init__,Method Bio.GA.Crossover.TwoPoint.TwoPointCrossover.__init__()=Bio.GA.Crossover.TwoPoint.TwoPointCrossover-class.html#__init__,Method Bio.GA.Crossover.Uniform.UniformCrossover.__init__()=Bio.GA.Crossover.Uniform.UniformCrossover-class.html#__init__,Method Bio.GA.Evolver.GenerationEvolver.__init__()=Bio.GA.Evolver.GenerationEvolver-class.html#__init__,Method Bio.GA.Evolver.SteadyStateEvolver.__init__()=Bio.GA.Evolver.SteadyStateEvolver-class.html#__init__,Method Bio.GA.Mutation.General.SafeFitnessMutation.__init__()=Bio.GA.Mutation.General.SafeFitnessMutation-class.html#__init__,Method Bio.GA.Mutation.Simple.ConversionMutation.__init__()=Bio.GA.Mutation.Simple.ConversionMutation-class.html#__init__,Method Bio.GA.Mutation.Simple.SinglePositionMutation.__init__()=Bio.GA.Mutation.Simple.SinglePositionMutation-class.html#__init__,Method Bio.GA.Organism.Organism.__init__()=Bio.GA.Organism.Organism-class.html#__init__,Method Bio.GA.Repair.Stabilizing.AmbiguousRepair.__init__()=Bio.GA.Repair.Stabilizing.AmbiguousRepair-class.html#__init__,Method Bio.GA.Selection.Abstract.AbstractSelection.__init__()=Bio.GA.Selection.Abstract.AbstractSelection-class.html#__init__,Method Bio.GA.Selection.Diversity.DiversitySelection.__init__()=Bio.GA.Selection.Diversity.DiversitySelection-class.html#__init__,Method Bio.GA.Selection.RouletteWheel.RouletteWheelSelection.__init__()=Bio.GA.Selection.RouletteWheel.RouletteWheelSelection-class.html#__init__,Method Bio.GA.Selection.Tournament.TournamentSelection.__init__()=Bio.GA.Selection.Tournament.TournamentSelection-class.html#__init__,Method Bio.GenBank.FeatureParser.__init__()=Bio.GenBank.FeatureParser-class.html#__init__,Method Bio.GenBank.Iterator.__init__()=Bio.GenBank.Iterator-class.html#__init__,Method Bio.GenBank.Record.Feature.__init__()=Bio.GenBank.Record.Feature-class.html#__init__,Method Bio.GenBank.Record.Qualifier.__init__()=Bio.GenBank.Record.Qualifier-class.html#__init__,Method Bio.GenBank.Record.Record.__init__()=Bio.GenBank.Record.Record-class.html#__init__,Method Bio.GenBank.Record.Reference.__init__()=Bio.GenBank.Record.Reference-class.html#__init__,Method Bio.GenBank.RecordParser.__init__()=Bio.GenBank.RecordParser-class.html#__init__,Method Bio.GenBank.Scanner.InsdcScanner.__init__()=Bio.GenBank.Scanner.InsdcScanner-class.html#__init__,Method Bio.GenBank._BaseGenBankConsumer.__init__()=Bio.GenBank._BaseGenBankConsumer-class.html#__init__,Method Bio.GenBank._FeatureConsumer.__init__()=Bio.GenBank._FeatureConsumer-class.html#__init__,Method Bio.GenBank._RecordConsumer.__init__()=Bio.GenBank._RecordConsumer-class.html#__init__,Method Bio.GenBank.utils.FeatureValueCleaner.__init__()=Bio.GenBank.utils.FeatureValueCleaner-class.html#__init__,Method Bio.Geo.Record.Record.__init__()=Bio.Geo.Record.Record-class.html#__init__,Method Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment.__init__()=Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome.Chromosome.__init__()=Bio.Graphics.BasicChromosome.Chromosome-class.html#__init__,Method Bio.Graphics.BasicChromosome.ChromosomeSegment.__init__()=Bio.Graphics.BasicChromosome.ChromosomeSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome.Organism.__init__()=Bio.Graphics.BasicChromosome.Organism-class.html#__init__,Method Bio.Graphics.BasicChromosome.TelomereSegment.__init__()=Bio.Graphics.BasicChromosome.TelomereSegment-class.html#__init__,Method Bio.Graphics.BasicChromosome._ChromosomeComponent.__init__()=Bio.Graphics.BasicChromosome._ChromosomeComponent-class.html#__init__,Method Bio.Graphics.ColorSpiral.ColorSpiral.__init__()=Bio.Graphics.ColorSpiral.ColorSpiral-class.html#__init__,Method Bio.Graphics.Comparative.ComparativeScatterPlot.__init__()=Bio.Graphics.Comparative.ComparativeScatterPlot-class.html#__init__,Method Bio.Graphics.DisplayRepresentation.ChromosomeCounts.__init__()=Bio.Graphics.DisplayRepresentation.ChromosomeCounts-class.html#__init__,Method Bio.Graphics.Distribution.BarChartDistribution.__init__()=Bio.Graphics.Distribution.BarChartDistribution-class.html#__init__,Method Bio.Graphics.Distribution.DistributionPage.__init__()=Bio.Graphics.Distribution.DistributionPage-class.html#__init__,Method Bio.Graphics.Distribution.LineDistribution.__init__()=Bio.Graphics.Distribution.LineDistribution-class.html#__init__,Method Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.__init__()=Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.__init__()=Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.__init__()=Bio.Graphics.GenomeDiagram._Colors.ColorTranslator-class.html#__init__,Method Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.__init__()=Bio.Graphics.GenomeDiagram._CrossLink.CrossLink-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.__init__()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Feature.Feature.__init__()=Bio.Graphics.GenomeDiagram._Feature.Feature-class.html#__init__,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.__init__()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.__init__()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#__init__,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.__init__()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#__init__,Method Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.__init__()=Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer-class.html#__init__,Method Bio.Graphics.GenomeDiagram._Track.Track.__init__()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#__init__,Method Bio.Graphics.KGML_vis.KGMLCanvas.__init__()=Bio.Graphics.KGML_vis.KGMLCanvas-class.html#__init__,Method Bio.HMM.DynamicProgramming.AbstractDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.AbstractDPAlgorithms-class.html#__init__,Method Bio.HMM.DynamicProgramming.LogDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.LogDPAlgorithms-class.html#__init__,Method Bio.HMM.DynamicProgramming.ScaledDPAlgorithms.__init__()=Bio.HMM.DynamicProgramming.ScaledDPAlgorithms-class.html#__init__,Method Bio.HMM.MarkovModel.HiddenMarkovModel.__init__()=Bio.HMM.MarkovModel.HiddenMarkovModel-class.html#__init__,Method Bio.HMM.MarkovModel.MarkovModelBuilder.__init__()=Bio.HMM.MarkovModel.MarkovModelBuilder-class.html#__init__,Method Bio.HMM.Trainer.AbstractTrainer.__init__()=Bio.HMM.Trainer.AbstractTrainer-class.html#__init__,Method Bio.HMM.Trainer.BaumWelchTrainer.__init__()=Bio.HMM.Trainer.BaumWelchTrainer-class.html#__init__,Method Bio.HMM.Trainer.KnownStateTrainer.__init__()=Bio.HMM.Trainer.KnownStateTrainer-class.html#__init__,Method Bio.HMM.Trainer.TrainingSequence.__init__()=Bio.HMM.Trainer.TrainingSequence-class.html#__init__,Method Bio.Index._InMemoryIndex.__init__()=Bio.Index._InMemoryIndex-class.html#__init__,Method Bio.Index._ShelveIndex.__init__()=Bio.Index._ShelveIndex-class.html#__init__,Method Bio.KDTree.KDTree'.KDTree.__init__()=Bio.KDTree.KDTree%27.KDTree-class.html#__init__,Method Bio.KEGG.Compound.Record.__init__()=Bio.KEGG.Compound.Record-class.html#__init__,Method Bio.KEGG.Enzyme.Record.__init__()=Bio.KEGG.Enzyme.Record-class.html#__init__,Method Bio.KEGG.KGML.KGML_parser.KGMLParser.__init__()=Bio.KEGG.KGML.KGML_parser.KGMLParser-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Component.__init__()=Bio.KEGG.KGML.KGML_pathway.Component-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Entry.__init__()=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Graphics.__init__()=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Pathway.__init__()=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Reaction.__init__()=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#__init__,Method Bio.KEGG.KGML.KGML_pathway.Relation.__init__()=Bio.KEGG.KGML.KGML_pathway.Relation-class.html#__init__,Method Bio.LogisticRegression.LogisticRegression.__init__()=Bio.LogisticRegression.LogisticRegression-class.html#__init__,Method Bio.MarkovModel.MarkovModel.__init__()=Bio.MarkovModel.MarkovModel-class.html#__init__,Method Bio.MaxEntropy.MaxEntropy.__init__()=Bio.MaxEntropy.MaxEntropy-class.html#__init__,Method Bio.Motif.Parsers.AlignAce.Record.__init__()=Bio.Motif.Parsers.AlignAce.Record-class.html#__init__,Method Bio.Motif.Parsers.MAST.Record.__init__()=Bio.Motif.Parsers.MAST.Record-class.html#__init__,Method Bio.Motif.Parsers.MEME.MEMEInstance.__init__()=Bio.Motif.Parsers.MEME.MEMEInstance-class.html#__init__,Method Bio.Motif.Parsers.MEME.MEMEMotif.__init__()=Bio.Motif.Parsers.MEME.MEMEMotif-class.html#__init__,Method Bio.Motif.Parsers.MEME.MEMERecord.__init__()=Bio.Motif.Parsers.MEME.MEMERecord-class.html#__init__,Method Bio.Motif.Thresholds.ScoreDistribution.__init__()=Bio.Motif.Thresholds.ScoreDistribution-class.html#__init__,Method Bio.Motif._Motif.Motif.__init__()=Bio.Motif._Motif.Motif-class.html#__init__,Method Bio.NMR.xpktools.Peaklist.__init__()=Bio.NMR.xpktools.Peaklist-class.html#__init__,Method Bio.NMR.xpktools.XpkEntry.__init__()=Bio.NMR.xpktools.XpkEntry-class.html#__init__,Method Bio.NaiveBayes.NaiveBayes.__init__()=Bio.NaiveBayes.NaiveBayes-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.AbstractLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.AbstractLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.InputLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.__init__()=Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer-class.html#__init__,Method Bio.NeuralNetwork.BackPropagation.Network.BasicNetwork.__init__()=Bio.NeuralNetwork.BackPropagation.Network.BasicNetwork-class.html#__init__,Method Bio.NeuralNetwork.Gene.Motif.MotifCoder.__init__()=Bio.NeuralNetwork.Gene.Motif.MotifCoder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Motif.MotifFinder.__init__()=Bio.NeuralNetwork.Gene.Motif.MotifFinder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Pattern.PatternIO.__init__()=Bio.NeuralNetwork.Gene.Pattern.PatternIO-class.html#__init__,Method Bio.NeuralNetwork.Gene.Pattern.PatternRepository.__init__()=Bio.NeuralNetwork.Gene.Pattern.PatternRepository-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness.__init__()=Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder.__init__()=Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness.__init__()=Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator.__init__()=Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.Schema.__init__()=Bio.NeuralNetwork.Gene.Schema.Schema-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SchemaCoder.__init__()=Bio.NeuralNetwork.Gene.Schema.SchemaCoder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SchemaFactory.__init__()=Bio.NeuralNetwork.Gene.Schema.SchemaFactory-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SchemaFinder.__init__()=Bio.NeuralNetwork.Gene.Schema.SchemaFinder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Schema.SimpleFinisher.__init__()=Bio.NeuralNetwork.Gene.Schema.SimpleFinisher-class.html#__init__,Method Bio.NeuralNetwork.Gene.Signature.SignatureCoder.__init__()=Bio.NeuralNetwork.Gene.Signature.SignatureCoder-class.html#__init__,Method Bio.NeuralNetwork.Gene.Signature.SignatureFinder.__init__()=Bio.NeuralNetwork.Gene.Signature.SignatureFinder-class.html#__init__,Method Bio.NeuralNetwork.StopTraining.ValidationIncreaseStop.__init__()=Bio.NeuralNetwork.StopTraining.ValidationIncreaseStop-class.html#__init__,Method Bio.NeuralNetwork.Training.ExampleManager.__init__()=Bio.NeuralNetwork.Training.ExampleManager-class.html#__init__,Method Bio.NeuralNetwork.Training.TrainingExample.__init__()=Bio.NeuralNetwork.Training.TrainingExample-class.html#__init__,Method Bio.Nexus.Nexus.Block.__init__()=Bio.Nexus.Nexus.Block-class.html#__init__,Method Bio.Nexus.Nexus.CharBuffer.__init__()=Bio.Nexus.Nexus.CharBuffer-class.html#__init__,Method Bio.Nexus.Nexus.Commandline.__init__()=Bio.Nexus.Nexus.Commandline-class.html#__init__,Method Bio.Nexus.Nexus.Nexus.__init__()=Bio.Nexus.Nexus.Nexus-class.html#__init__,Method Bio.Nexus.Nexus.StepMatrix.__init__()=Bio.Nexus.Nexus.StepMatrix-class.html#__init__,Method Bio.Nexus.Nodes.Chain.__init__()=Bio.Nexus.Nodes.Chain-class.html#__init__,Method Bio.Nexus.Nodes.Node.__init__()=Bio.Nexus.Nodes.Node-class.html#__init__,Method Bio.Nexus.Trees.NodeData.__init__()=Bio.Nexus.Trees.NodeData-class.html#__init__,Method Bio.Nexus.Trees.Tree.__init__()=Bio.Nexus.Trees.Tree-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractPropertyMap-class.html#__init__,Method Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap.__init__()=Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap-class.html#__init__,Method Bio.PDB.Atom.Atom.__init__()=Bio.PDB.Atom.Atom-class.html#__init__,Method Bio.PDB.Atom.DisorderedAtom.__init__()=Bio.PDB.Atom.DisorderedAtom-class.html#__init__,Method Bio.PDB.Chain.Chain.__init__()=Bio.PDB.Chain.Chain-class.html#__init__,Method Bio.PDB.DSSP'.DSSP.__init__()=Bio.PDB.DSSP%27.DSSP-class.html#__init__,Method Bio.PDB.Dice.ChainSelector.__init__()=Bio.PDB.Dice.ChainSelector-class.html#__init__,Method Bio.PDB.Entity.DisorderedEntityWrapper.__init__()=Bio.PDB.Entity.DisorderedEntityWrapper-class.html#__init__,Method Bio.PDB.Entity.Entity.__init__()=Bio.PDB.Entity.Entity-class.html#__init__,Method Bio.PDB.FragmentMapper'.Fragment.__init__()=Bio.PDB.FragmentMapper%27.Fragment-class.html#__init__,Method Bio.PDB.FragmentMapper'.FragmentMapper.__init__()=Bio.PDB.FragmentMapper%27.FragmentMapper-class.html#__init__,Method Bio.PDB.HSExposure.ExposureCN.__init__()=Bio.PDB.HSExposure.ExposureCN-class.html#__init__,Method Bio.PDB.HSExposure.HSExposureCA.__init__()=Bio.PDB.HSExposure.HSExposureCA-class.html#__init__,Method Bio.PDB.HSExposure.HSExposureCB.__init__()=Bio.PDB.HSExposure.HSExposureCB-class.html#__init__,Method Bio.PDB.HSExposure._AbstractHSExposure.__init__()=Bio.PDB.HSExposure._AbstractHSExposure-class.html#__init__,Method Bio.PDB.MMCIF2Dict.MMCIF2Dict.__init__()=Bio.PDB.MMCIF2Dict.MMCIF2Dict-class.html#__init__,Method Bio.PDB.MMCIFParser'.MMCIFParser.__init__()=Bio.PDB.MMCIFParser%27.MMCIFParser-class.html#__init__,Method Bio.PDB.Model.Model.__init__()=Bio.PDB.Model.Model-class.html#__init__,Method Bio.PDB.NACCESS.NACCESS.__init__()=Bio.PDB.NACCESS.NACCESS-class.html#__init__,Method Bio.PDB.NACCESS.NACCESS_atomic.__init__()=Bio.PDB.NACCESS.NACCESS_atomic-class.html#__init__,Method Bio.PDB.NeighborSearch'.NeighborSearch.__init__()=Bio.PDB.NeighborSearch%27.NeighborSearch-class.html#__init__,Method Bio.PDB.PDBIO'.PDBIO.__init__()=Bio.PDB.PDBIO%27.PDBIO-class.html#__init__,Method Bio.PDB.PDBList'.PDBList.__init__()=Bio.PDB.PDBList%27.PDBList-class.html#__init__,Method Bio.PDB.PDBParser'.PDBParser.__init__()=Bio.PDB.PDBParser%27.PDBParser-class.html#__init__,Method Bio.PDB.PSEA.PSEA.__init__()=Bio.PDB.PSEA.PSEA-class.html#__init__,Method Bio.PDB.Polypeptide.CaPPBuilder.__init__()=Bio.PDB.Polypeptide.CaPPBuilder-class.html#__init__,Method Bio.PDB.Polypeptide.PPBuilder.__init__()=Bio.PDB.Polypeptide.PPBuilder-class.html#__init__,Method Bio.PDB.Polypeptide._PPBuilder.__init__()=Bio.PDB.Polypeptide._PPBuilder-class.html#__init__,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.__init__()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#__init__,Method Bio.PDB.Residue.DisorderedResidue.__init__()=Bio.PDB.Residue.DisorderedResidue-class.html#__init__,Method Bio.PDB.Residue.Residue.__init__()=Bio.PDB.Residue.Residue-class.html#__init__,Method Bio.PDB.ResidueDepth'.ResidueDepth.__init__()=Bio.PDB.ResidueDepth%27.ResidueDepth-class.html#__init__,Method Bio.PDB.Structure.Structure.__init__()=Bio.PDB.Structure.Structure-class.html#__init__,Method Bio.PDB.StructureAlignment'.StructureAlignment.__init__()=Bio.PDB.StructureAlignment%27.StructureAlignment-class.html#__init__,Method Bio.PDB.StructureBuilder.StructureBuilder.__init__()=Bio.PDB.StructureBuilder.StructureBuilder-class.html#__init__,Method Bio.PDB.Superimposer'.Superimposer.__init__()=Bio.PDB.Superimposer%27.Superimposer-class.html#__init__,Method Bio.PDB.Vector'.Vector.__init__()=Bio.PDB.Vector%27.Vector-class.html#__init__,Method Bio.ParserSupport.EventGenerator.__init__()=Bio.ParserSupport.EventGenerator-class.html#__init__,Method Bio.ParserSupport.TaggingConsumer.__init__()=Bio.ParserSupport.TaggingConsumer-class.html#__init__,Method Bio.Pathway.Network.__init__()=Bio.Pathway.Network-class.html#__init__,Method Bio.Pathway.Reaction.__init__()=Bio.Pathway.Reaction-class.html#__init__,Method Bio.Pathway.Rep.Graph.Graph.__init__()=Bio.Pathway.Rep.Graph.Graph-class.html#__init__,Method Bio.Pathway.Rep.MultiGraph.MultiGraph.__init__()=Bio.Pathway.Rep.MultiGraph.MultiGraph-class.html#__init__,Method Bio.Pathway.System.__init__()=Bio.Pathway.System-class.html#__init__,Method Bio.Phylo.Applications._Fasttree.FastTreeCommandline.__init__()=Bio.Phylo.Applications._Fasttree.FastTreeCommandline-class.html#__init__,Method Bio.Phylo.Applications._Phyml.PhymlCommandline.__init__()=Bio.Phylo.Applications._Phyml.PhymlCommandline-class.html#__init__,Method Bio.Phylo.Applications._Raxml.RaxmlCommandline.__init__()=Bio.Phylo.Applications._Raxml.RaxmlCommandline-class.html#__init__,Method Bio.Phylo.BaseTree.BranchColor.__init__()=Bio.Phylo.BaseTree.BranchColor-class.html#__init__,Method Bio.Phylo.BaseTree.Clade.__init__()=Bio.Phylo.BaseTree.Clade-class.html#__init__,Method Bio.Phylo.BaseTree.Tree.__init__()=Bio.Phylo.BaseTree.Tree-class.html#__init__,Method Bio.Phylo.CDAO.Clade.__init__()=Bio.Phylo.CDAO.Clade-class.html#__init__,Method Bio.Phylo.CDAO.Tree.__init__()=Bio.Phylo.CDAO.Tree-class.html#__init__,Method Bio.Phylo.CDAOIO.Parser.__init__()=Bio.Phylo.CDAOIO.Parser-class.html#__init__,Method Bio.Phylo.CDAOIO.Writer.__init__()=Bio.Phylo.CDAOIO.Writer-class.html#__init__,Method Bio.Phylo.NeXML.Clade.__init__()=Bio.Phylo.NeXML.Clade-class.html#__init__,Method Bio.Phylo.NeXML.Tree.__init__()=Bio.Phylo.NeXML.Tree-class.html#__init__,Method Bio.Phylo.NeXMLIO.Parser.__init__()=Bio.Phylo.NeXMLIO.Parser-class.html#__init__,Method Bio.Phylo.NeXMLIO.Writer.__init__()=Bio.Phylo.NeXMLIO.Writer-class.html#__init__,Method Bio.Phylo.Newick.Clade.__init__()=Bio.Phylo.Newick.Clade-class.html#__init__,Method Bio.Phylo.Newick.Tree.__init__()=Bio.Phylo.Newick.Tree-class.html#__init__,Method Bio.Phylo.NewickIO.Parser.__init__()=Bio.Phylo.NewickIO.Parser-class.html#__init__,Method Bio.Phylo.NewickIO.Writer.__init__()=Bio.Phylo.NewickIO.Writer-class.html#__init__,Method Bio.Phylo.PAML._paml.Paml.__init__()=Bio.Phylo.PAML._paml.Paml-class.html#__init__,Method Bio.Phylo.PAML.baseml.Baseml.__init__()=Bio.Phylo.PAML.baseml.Baseml-class.html#__init__,Method Bio.Phylo.PAML.codeml.Codeml.__init__()=Bio.Phylo.PAML.codeml.Codeml-class.html#__init__,Method Bio.Phylo.PAML.yn00.Yn00.__init__()=Bio.Phylo.PAML.yn00.Yn00-class.html#__init__,Method Bio.Phylo.PhyloXML.Accession.__init__()=Bio.Phylo.PhyloXML.Accession-class.html#__init__,Method Bio.Phylo.PhyloXML.Annotation.__init__()=Bio.Phylo.PhyloXML.Annotation-class.html#__init__,Method Bio.Phylo.PhyloXML.BinaryCharacters.__init__()=Bio.Phylo.PhyloXML.BinaryCharacters-class.html#__init__,Method Bio.Phylo.PhyloXML.BranchColor.__init__()=Bio.Phylo.PhyloXML.BranchColor-class.html#__init__,Method Bio.Phylo.PhyloXML.Clade.__init__()=Bio.Phylo.PhyloXML.Clade-class.html#__init__,Method Bio.Phylo.PhyloXML.CladeRelation.__init__()=Bio.Phylo.PhyloXML.CladeRelation-class.html#__init__,Method Bio.Phylo.PhyloXML.Confidence.__init__()=Bio.Phylo.PhyloXML.Confidence-class.html#__init__,Method Bio.Phylo.PhyloXML.Date.__init__()=Bio.Phylo.PhyloXML.Date-class.html#__init__,Method Bio.Phylo.PhyloXML.Distribution.__init__()=Bio.Phylo.PhyloXML.Distribution-class.html#__init__,Method Bio.Phylo.PhyloXML.DomainArchitecture.__init__()=Bio.Phylo.PhyloXML.DomainArchitecture-class.html#__init__,Method Bio.Phylo.PhyloXML.Events.__init__()=Bio.Phylo.PhyloXML.Events-class.html#__init__,Method Bio.Phylo.PhyloXML.Id.__init__()=Bio.Phylo.PhyloXML.Id-class.html#__init__,Method Bio.Phylo.PhyloXML.MolSeq.__init__()=Bio.Phylo.PhyloXML.MolSeq-class.html#__init__,Method Bio.Phylo.PhyloXML.Other.__init__()=Bio.Phylo.PhyloXML.Other-class.html#__init__,Method Bio.Phylo.PhyloXML.Phylogeny.__init__()=Bio.Phylo.PhyloXML.Phylogeny-class.html#__init__,Method Bio.Phylo.PhyloXML.Phyloxml.__init__()=Bio.Phylo.PhyloXML.Phyloxml-class.html#__init__,Method Bio.Phylo.PhyloXML.Point.__init__()=Bio.Phylo.PhyloXML.Point-class.html#__init__,Method Bio.Phylo.PhyloXML.Polygon.__init__()=Bio.Phylo.PhyloXML.Polygon-class.html#__init__,Method Bio.Phylo.PhyloXML.Property.__init__()=Bio.Phylo.PhyloXML.Property-class.html#__init__,Method Bio.Phylo.PhyloXML.ProteinDomain.__init__()=Bio.Phylo.PhyloXML.ProteinDomain-class.html#__init__,Method Bio.Phylo.PhyloXML.Reference.__init__()=Bio.Phylo.PhyloXML.Reference-class.html#__init__,Method Bio.Phylo.PhyloXML.Sequence.__init__()=Bio.Phylo.PhyloXML.Sequence-class.html#__init__,Method Bio.Phylo.PhyloXML.SequenceRelation.__init__()=Bio.Phylo.PhyloXML.SequenceRelation-class.html#__init__,Method Bio.Phylo.PhyloXML.Taxonomy.__init__()=Bio.Phylo.PhyloXML.Taxonomy-class.html#__init__,Method Bio.Phylo.PhyloXML.Uri.__init__()=Bio.Phylo.PhyloXML.Uri-class.html#__init__,Method Bio.Phylo.PhyloXMLIO.Parser.__init__()=Bio.Phylo.PhyloXMLIO.Parser-class.html#__init__,Method Bio.Phylo.PhyloXMLIO.Writer.__init__()=Bio.Phylo.PhyloXMLIO.Writer-class.html#__init__,Method Bio.Phylo.TreeConstruction.DistanceCalculator.__init__()=Bio.Phylo.TreeConstruction.DistanceCalculator-class.html#__init__,Method Bio.Phylo.TreeConstruction.DistanceTreeConstructor.__init__()=Bio.Phylo.TreeConstruction.DistanceTreeConstructor-class.html#__init__,Method Bio.Phylo.TreeConstruction.NNITreeSearcher.__init__()=Bio.Phylo.TreeConstruction.NNITreeSearcher-class.html#__init__,Method Bio.Phylo.TreeConstruction.ParsimonyScorer.__init__()=Bio.Phylo.TreeConstruction.ParsimonyScorer-class.html#__init__,Method Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor.__init__()=Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor-class.html#__init__,Method Bio.Phylo.TreeConstruction._DistanceMatrix.__init__()=Bio.Phylo.TreeConstruction._DistanceMatrix-class.html#__init__,Method Bio.Phylo.TreeConstruction._Matrix.__init__()=Bio.Phylo.TreeConstruction._Matrix-class.html#__init__,Method Bio.PopGen.Async.Async.__init__()=Bio.PopGen.Async.Async-class.html#__init__,Method Bio.PopGen.Async.DirectoryRetriever.__init__()=Bio.PopGen.Async.DirectoryRetriever-class.html#__init__,Method Bio.PopGen.Async.FileRetriever.__init__()=Bio.PopGen.Async.FileRetriever-class.html#__init__,Method Bio.PopGen.Async.Local.Local.__init__()=Bio.PopGen.Async.Local.Local-class.html#__init__,Method Bio.PopGen.FDist.Async.FDistAsync.__init__()=Bio.PopGen.FDist.Async.FDistAsync-class.html#__init__,Method Bio.PopGen.FDist.Async.SplitFDist.__init__()=Bio.PopGen.FDist.Async.SplitFDist-class.html#__init__,Method Bio.PopGen.FDist.Controller.FDistController.__init__()=Bio.PopGen.FDist.Controller.FDistController-class.html#__init__,Method Bio.PopGen.FDist.Record.__init__()=Bio.PopGen.FDist.Record-class.html#__init__,Method Bio.PopGen.GenePop.Controller.GenePopController.__init__()=Bio.PopGen.GenePop.Controller.GenePopController-class.html#__init__,Method Bio.PopGen.GenePop.Controller._FileIterator.__init__()=Bio.PopGen.GenePop.Controller._FileIterator-class.html#__init__,Method Bio.PopGen.GenePop.Controller._GenePopCommandline.__init__()=Bio.PopGen.GenePop.Controller._GenePopCommandline-class.html#__init__,Method Bio.PopGen.GenePop.EasyController.EasyController.__init__()=Bio.PopGen.GenePop.EasyController.EasyController-class.html#__init__,Method Bio.PopGen.GenePop.FileParser.FileRecord.__init__()=Bio.PopGen.GenePop.FileParser.FileRecord-class.html#__init__,Method Bio.PopGen.GenePop.LargeFileParser.Record.__init__()=Bio.PopGen.GenePop.LargeFileParser.Record-class.html#__init__,Method Bio.PopGen.GenePop.Record.__init__()=Bio.PopGen.GenePop.Record-class.html#__init__,Method Bio.PopGen.SimCoal.Async.SimCoalCache.__init__()=Bio.PopGen.SimCoal.Async.SimCoalCache-class.html#__init__,Method Bio.PopGen.SimCoal.Cache.SimCoalCache.__init__()=Bio.PopGen.SimCoal.Cache.SimCoalCache-class.html#__init__,Method Bio.PopGen.SimCoal.Controller.FastSimCoalController.__init__()=Bio.PopGen.SimCoal.Controller.FastSimCoalController-class.html#__init__,Method Bio.PopGen.SimCoal.Controller.SimCoalController.__init__()=Bio.PopGen.SimCoal.Controller.SimCoalController-class.html#__init__,Method Bio.PopGen.SimCoal.Controller._FastSimCoalCommandLine.__init__()=Bio.PopGen.SimCoal.Controller._FastSimCoalCommandLine-class.html#__init__,Method Bio.Restriction.PrintFormat.PrintFormat.__init__()=Bio.Restriction.PrintFormat.PrintFormat-class.html#__init__,Method Bio.Restriction.Restriction.Analysis.__init__()=Bio.Restriction.Restriction.Analysis-class.html#__init__,Method Bio.Restriction.Restriction.FormattedSeq.__init__()=Bio.Restriction.Restriction.FormattedSeq-class.html#__init__,Method Bio.Restriction.Restriction.RestrictionBatch.__init__()=Bio.Restriction.Restriction.RestrictionBatch-class.html#__init__,Method Bio.Restriction.Restriction.RestrictionType.__init__()=Bio.Restriction.Restriction.RestrictionType-class.html#__init__,Method Bio.SCOP.Astral.__init__()=Bio.SCOP.Astral-class.html#__init__,Method Bio.SCOP.Cla.Index.__init__()=Bio.SCOP.Cla.Index-class.html#__init__,Method Bio.SCOP.Cla.Record.__init__()=Bio.SCOP.Cla.Record-class.html#__init__,Method Bio.SCOP.Des.Record.__init__()=Bio.SCOP.Des.Record-class.html#__init__,Method Bio.SCOP.Dom.Record.__init__()=Bio.SCOP.Dom.Record-class.html#__init__,Method Bio.SCOP.Domain.__init__()=Bio.SCOP.Domain-class.html#__init__,Method Bio.SCOP.Hie.Record.__init__()=Bio.SCOP.Hie.Record-class.html#__init__,Method Bio.SCOP.Node.__init__()=Bio.SCOP.Node-class.html#__init__,Method Bio.SCOP.Raf.Res.__init__()=Bio.SCOP.Raf.Res-class.html#__init__,Method Bio.SCOP.Raf.SeqMap.__init__()=Bio.SCOP.Raf.SeqMap-class.html#__init__,Method Bio.SCOP.Raf.SeqMapIndex.__init__()=Bio.SCOP.Raf.SeqMapIndex-class.html#__init__,Method Bio.SCOP.Residues.Residues.__init__()=Bio.SCOP.Residues.Residues-class.html#__init__,Method Bio.SCOP.Scop.__init__()=Bio.SCOP.Scop-class.html#__init__,Method Bio.SVDSuperimposer.SVDSuperimposer.__init__()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabParser.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.__init__()=Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_text.BlastTextParser.__init__()=Bio.SearchIO.BlastIO.blast_text.BlastTextParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.__init__()=Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter-class.html#__init__,Method Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator.__init__()=Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslIndexer.__init__()=Bio.SearchIO.BlatIO.BlatPslIndexer-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslParser.__init__()=Bio.SearchIO.BlatIO.BlatPslParser-class.html#__init__,Method Bio.SearchIO.BlatIO.BlatPslWriter.__init__()=Bio.SearchIO.BlatIO.BlatPslWriter-class.html#__init__,Method Bio.SearchIO.ExonerateIO._base._BaseExonerateParser.__init__()=Bio.SearchIO.ExonerateIO._base._BaseExonerateParser-class.html#__init__,Method Bio.SearchIO.FastaIO.FastaM10Indexer.__init__()=Bio.SearchIO.FastaIO.FastaM10Indexer-class.html#__init__,Method Bio.SearchIO.FastaIO.FastaM10Parser.__init__()=Bio.SearchIO.FastaIO.FastaM10Parser-class.html#__init__,Method Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer.__init__()=Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser.__init__()=Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.__init__()=Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser.__init__()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.__init__()=Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter-class.html#__init__,Method Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser.__init__()=Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser-class.html#__init__,Method Bio.SearchIO._index.SearchIndexer.__init__()=Bio.SearchIO._index.SearchIndexer-class.html#__init__,Method Bio.SearchIO._model.hit.Hit.__init__()=Bio.SearchIO._model.hit.Hit-class.html#__init__,Method Bio.SearchIO._model.hsp.HSP.__init__()=Bio.SearchIO._model.hsp.HSP-class.html#__init__,Method Bio.SearchIO._model.hsp.HSPFragment.__init__()=Bio.SearchIO._model.hsp.HSPFragment-class.html#__init__,Method Bio.SearchIO._model.query.QueryResult.__init__()=Bio.SearchIO._model.query.QueryResult-class.html#__init__,Method Bio.Seq.MutableSeq.__init__()=Bio.Seq.MutableSeq-class.html#__init__,Method Bio.Seq.Seq.__init__()=Bio.Seq.Seq-class.html#__init__,Method Bio.Seq.UnknownSeq.__init__()=Bio.Seq.UnknownSeq-class.html#__init__,Method Bio.SeqFeature.CompoundLocation.__init__()=Bio.SeqFeature.CompoundLocation-class.html#__init__,Method Bio.SeqFeature.FeatureLocation.__init__()=Bio.SeqFeature.FeatureLocation-class.html#__init__,Method Bio.SeqFeature.PositionGap.__init__()=Bio.SeqFeature.PositionGap-class.html#__init__,Method Bio.SeqFeature.Reference.__init__()=Bio.SeqFeature.Reference-class.html#__init__,Method Bio.SeqFeature.SeqFeature.__init__()=Bio.SeqFeature.SeqFeature-class.html#__init__,Method Bio.SeqIO.FastaIO.FastaWriter.__init__()=Bio.SeqIO.FastaIO.FastaWriter-class.html#__init__,Method Bio.SeqIO.Interfaces.SequenceIterator.__init__()=Bio.SeqIO.Interfaces.SequenceIterator-class.html#__init__,Method Bio.SeqIO.Interfaces.SequenceWriter.__init__()=Bio.SeqIO.Interfaces.SequenceWriter-class.html#__init__,Method Bio.SeqIO.Interfaces.SequentialSequenceWriter.__init__()=Bio.SeqIO.Interfaces.SequentialSequenceWriter-class.html#__init__,Method Bio.SeqIO.PhdIO.PhdWriter.__init__()=Bio.SeqIO.PhdIO.PhdWriter-class.html#__init__,Method Bio.SeqIO.QualityIO.QualPhredWriter.__init__()=Bio.SeqIO.QualityIO.QualPhredWriter-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.SeqXmlIterator.__init__()=Bio.SeqIO.SeqXmlIO.SeqXmlIterator-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.SeqXmlWriter.__init__()=Bio.SeqIO.SeqXmlIO.SeqXmlWriter-class.html#__init__,Method Bio.SeqIO.SeqXmlIO.XMLRecordIterator.__init__()=Bio.SeqIO.SeqXmlIO.XMLRecordIterator-class.html#__init__,Method Bio.SeqIO.SffIO.SffWriter.__init__()=Bio.SeqIO.SffIO.SffWriter-class.html#__init__,Method Bio.SeqIO.SffIO._AddTellHandle.__init__()=Bio.SeqIO.SffIO._AddTellHandle-class.html#__init__,Method Bio.SeqIO.UniprotIO.Parser.__init__()=Bio.SeqIO.UniprotIO.Parser-class.html#__init__,Method Bio.SeqIO._index.IntelliGeneticsRandomAccess.__init__()=Bio.SeqIO._index.IntelliGeneticsRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SeqFileRandomAccess.__init__()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SequentialSeqFileRandomAccess.__init__()=Bio.SeqIO._index.SequentialSeqFileRandomAccess-class.html#__init__,Method Bio.SeqIO._index.SffRandomAccess.__init__()=Bio.SeqIO._index.SffRandomAccess-class.html#__init__,Method Bio.SeqRecord.SeqRecord.__init__()=Bio.SeqRecord.SeqRecord-class.html#__init__,Method Bio.SeqRecord._RestrictedDict.__init__()=Bio.SeqRecord._RestrictedDict-class.html#__init__,Method Bio.SeqUtils.CodonUsage.CodonAdaptationIndex.__init__()=Bio.SeqUtils.CodonUsage.CodonAdaptationIndex-class.html#__init__,Method Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.__init__()=Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint-class.html#__init__,Method Bio.SeqUtils.ProtParam.ProteinAnalysis.__init__()=Bio.SeqUtils.ProtParam.ProteinAnalysis-class.html#__init__,Method Bio.Sequencing.Ace.ACEFileRecord.__init__()=Bio.Sequencing.Ace.ACEFileRecord-class.html#__init__,Method Bio.Sequencing.Ace.Contig.__init__()=Bio.Sequencing.Ace.Contig-class.html#__init__,Method Bio.Sequencing.Ace.Reads.__init__()=Bio.Sequencing.Ace.Reads-class.html#__init__,Method Bio.Sequencing.Ace.af.__init__()=Bio.Sequencing.Ace.af-class.html#__init__,Method Bio.Sequencing.Ace.bs.__init__()=Bio.Sequencing.Ace.bs-class.html#__init__,Method Bio.Sequencing.Ace.ct.__init__()=Bio.Sequencing.Ace.ct-class.html#__init__,Method Bio.Sequencing.Ace.ds.__init__()=Bio.Sequencing.Ace.ds-class.html#__init__,Method Bio.Sequencing.Ace.qa.__init__()=Bio.Sequencing.Ace.qa-class.html#__init__,Method Bio.Sequencing.Ace.rd.__init__()=Bio.Sequencing.Ace.rd-class.html#__init__,Method Bio.Sequencing.Ace.rt.__init__()=Bio.Sequencing.Ace.rt-class.html#__init__,Method Bio.Sequencing.Ace.wa.__init__()=Bio.Sequencing.Ace.wa-class.html#__init__,Method Bio.Sequencing.Ace.wr.__init__()=Bio.Sequencing.Ace.wr-class.html#__init__,Method Bio.Sequencing.Applications._Novoalign.NovoalignCommandline.__init__()=Bio.Sequencing.Applications._Novoalign.NovoalignCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaAlignCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaAlignCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaBwaswCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaBwaswCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaIndexCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaIndexCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaSampeCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaSampeCommandline-class.html#__init__,Method Bio.Sequencing.Applications._bwa.BwaSamseCommandline.__init__()=Bio.Sequencing.Applications._bwa.BwaSamseCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsSortCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsSortCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline-class.html#__init__,Method Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline.__init__()=Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline-class.html#__init__,Method Bio.Sequencing.Phd.Record.__init__()=Bio.Sequencing.Phd.Record-class.html#__init__,Method Bio.SubsMat.FreqTable.FreqTable.__init__()=Bio.SubsMat.FreqTable.FreqTable-class.html#__init__,Method Bio.SubsMat.SeqMat.__init__()=Bio.SubsMat.SeqMat-class.html#__init__,Method Bio.SwissProt.KeyWList.Record.__init__()=Bio.SwissProt.KeyWList.Record-class.html#__init__,Method Bio.SwissProt.Record.__init__()=Bio.SwissProt.Record-class.html#__init__,Method Bio.SwissProt.Reference.__init__()=Bio.SwissProt.Reference-class.html#__init__,Method Bio.UniGene.ProtsimLine.__init__()=Bio.UniGene.ProtsimLine-class.html#__init__,Method Bio.UniGene.Record.__init__()=Bio.UniGene.Record-class.html#__init__,Method Bio.UniGene.STSLine.__init__()=Bio.UniGene.STSLine-class.html#__init__,Method Bio.UniGene.SequenceLine.__init__()=Bio.UniGene.SequenceLine-class.html#__init__,Method Bio.Wise.dnal.Statistics.__init__()=Bio.Wise.dnal.Statistics-class.html#__init__,Method Bio.Wise.psw.AlignmentColumn.__init__()=Bio.Wise.psw.AlignmentColumn-class.html#__init__,Method Bio.Wise.psw.ColumnUnit.__init__()=Bio.Wise.psw.ColumnUnit-class.html#__init__,Method Bio._py3k._ordereddict.OrderedDict.__init__()=Bio._py3k._ordereddict.OrderedDict-class.html#__init__,Method Bio.bgzf.BgzfReader.__init__()=Bio.bgzf.BgzfReader-class.html#__init__,Method Bio.bgzf.BgzfWriter.__init__()=Bio.bgzf.BgzfWriter-class.html#__init__,Method Bio.codonalign.codonalignment.CodonAlignment.__init__()=Bio.codonalign.codonalignment.CodonAlignment-class.html#__init__,Method Bio.codonalign.codonseq.CodonSeq.__init__()=Bio.codonalign.codonseq.CodonSeq-class.html#__init__,Method Bio.kNN.kNN.__init__()=Bio.kNN.kNN-class.html#__init__,Method Bio.motifs.Instances.__init__()=Bio.motifs.Instances-class.html#__init__,Method Bio.motifs.Motif.__init__()=Bio.motifs.Motif-class.html#__init__,Method Bio.motifs.alignace.Record.__init__()=Bio.motifs.alignace.Record-class.html#__init__,Method Bio.motifs.applications._alignace.AlignAceCommandline.__init__()=Bio.motifs.applications._alignace.AlignAceCommandline-class.html#__init__,Method Bio.motifs.applications._alignace.CompareAceCommandline.__init__()=Bio.motifs.applications._alignace.CompareAceCommandline-class.html#__init__,Method Bio.motifs.applications._xxmotif.XXmotifCommandline.__init__()=Bio.motifs.applications._xxmotif.XXmotifCommandline-class.html#__init__,Method Bio.motifs.jaspar.Motif.__init__()=Bio.motifs.jaspar.Motif-class.html#__init__,Method Bio.motifs.jaspar.Record.__init__()=Bio.motifs.jaspar.Record-class.html#__init__,Method Bio.motifs.jaspar.db.JASPAR5.__init__()=Bio.motifs.jaspar.db.JASPAR5-class.html#__init__,Method Bio.motifs.mast.Record.__init__()=Bio.motifs.mast.Record-class.html#__init__,Method Bio.motifs.matrix.GenericPositionMatrix.__init__()=Bio.motifs.matrix.GenericPositionMatrix-class.html#__init__,Method Bio.motifs.matrix.PositionWeightMatrix.__init__()=Bio.motifs.matrix.PositionWeightMatrix-class.html#__init__,Method Bio.motifs.meme.Instance.__init__()=Bio.motifs.meme.Instance-class.html#__init__,Method Bio.motifs.meme.Motif.__init__()=Bio.motifs.meme.Motif-class.html#__init__,Method Bio.motifs.meme.Record.__init__()=Bio.motifs.meme.Record-class.html#__init__,Method Bio.motifs.thresholds.ScoreDistribution.__init__()=Bio.motifs.thresholds.ScoreDistribution-class.html#__init__,Method Bio.motifs.transfac.Record.__init__()=Bio.motifs.transfac.Record-class.html#__init__,Method Bio.pairwise2.affine_penalty.__init__()=Bio.pairwise2.affine_penalty-class.html#__init__,Method Bio.pairwise2.dictionary_match.__init__()=Bio.pairwise2.dictionary_match-class.html#__init__,Method Bio.pairwise2.identity_match.__init__()=Bio.pairwise2.identity_match-class.html#__init__,Method BioSQL.BioSeq.DBSeq.__init__()=BioSQL.BioSeq.DBSeq-class.html#__init__,Method BioSQL.BioSeq.DBSeqRecord.__init__()=BioSQL.BioSeq.DBSeqRecord-class.html#__init__,Method BioSQL.BioSeqDatabase.Adaptor.__init__()=BioSQL.BioSeqDatabase.Adaptor-class.html#__init__,Method BioSQL.BioSeqDatabase.BioSeqDatabase.__init__()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#__init__,Method BioSQL.BioSeqDatabase.DBServer.__init__()=BioSQL.BioSeqDatabase.DBServer-class.html#__init__,Method BioSQL.BioSeqDatabase._CursorWrapper.__init__()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#__init__,Method BioSQL.DBUtils.Generic_dbutils.__init__()=BioSQL.DBUtils.Generic_dbutils-class.html#__init__,Method BioSQL.Loader.DatabaseLoader.__init__()=BioSQL.Loader.DatabaseLoader-class.html#__init__,Method BioSQL.Loader.DatabaseRemover.__init__()=BioSQL.Loader.DatabaseRemover-class.html#__init__"><a title="Bio.Affy.CelFile.Record.__init__
Bio.Align.AlignInfo.PSSM.__init__
Bio.Align.AlignInfo.SummaryInfo.__init__
Bio.Align.Applications._ClustalOmega.ClustalOmegaCommandline.__init__
Bio.Align.Applications._Clustalw.ClustalwCommandline.__init__
Bio.Align.Applications._Dialign.DialignCommandline.__init__
Bio.Align.Applications._MSAProbs.MSAProbsCommandline.__init__
Bio.Align.Applications._Mafft.MafftCommandline.__init__
Bio.Align.Applications._Muscle.MuscleCommandline.__init__
Bio.Align.Applications._Prank.PrankCommandline.__init__
Bio.Align.Applications._Probcons.ProbconsCommandline.__init__
Bio.Align.Applications._TCoffee.TCoffeeCommandline.__init__
Bio.Align.Generic.Alignment.__init__
Bio.Align.MultipleSeqAlignment.__init__
Bio.AlignIO.Interfaces.AlignmentIterator.__init__
Bio.AlignIO.Interfaces.AlignmentWriter.__init__
Bio.AlignIO.Interfaces.SequentialAlignmentWriter.__init__
Bio.Alphabet.AlphabetEncoder.__init__
Bio.Alphabet.Gapped.__init__
Bio.Alphabet.HasStopCodon.__init__
Bio.Application.AbstractCommandline.__init__
Bio.Application.ApplicationError.__init__
Bio.Application._AbstractParameter.__init__
Bio.Application._Argument.__init__
Bio.Application._Option.__init__
Bio.Application._StaticArgument.__init__
Bio.Application._Switch.__init__
Bio.Blast.Applications.NcbiblastformatterCommandline.__init__
Bio.Blast.Applications.NcbiblastnCommandline.__init__
Bio.Blast.Applications.NcbiblastpCommandline.__init__
Bio.Blast.Applications.NcbiblastxCommandline.__init__
Bio.Blast.Applications.NcbideltablastCommandline.__init__
Bio.Blast.Applications.NcbipsiblastCommandline.__init__
Bio.Blast.Applications.NcbirpsblastCommandline.__init__
Bio.Blast.Applications.NcbirpstblastnCommandline.__init__
Bio.Blast.Applications.NcbitblastnCommandline.__init__
Bio.Blast.Applications.NcbitblastxCommandline.__init__
Bio.Blast.Applications._NcbibaseblastCommandline.__init__
Bio.Blast.Applications._Ncbiblast2SeqCommandline.__init__
Bio.Blast.Applications._NcbiblastCommandline.__init__
Bio.Blast.Applications._NcbiblastMain2SeqCommandline.__init__
Bio.Blast.NCBIStandalone.BlastErrorParser.__init__
Bio.Blast.NCBIStandalone.BlastParser.__init__
Bio.Blast.NCBIStandalone.Iterator.__init__
Bio.Blast.NCBIStandalone.PSIBlastParser.__init__
Bio.Blast.NCBIStandalone._BlastConsumer.__init__
Bio.Blast.NCBIStandalone._BlastErrorConsumer.__init__
Bio.Blast.NCBIStandalone._PSIBlastConsumer.__init__
Bio.Blast.NCBIXML.BlastParser.__init__
Bio.Blast.NCBIXML._XMLparser.__init__
Bio.Blast.ParseBlastTable.BlastTableEntry.__init__
Bio.Blast.ParseBlastTable.BlastTableReader.__init__
Bio.Blast.ParseBlastTable.BlastTableRec.__init__
Bio.Blast.Record.Alignment.__init__
Bio.Blast.Record.Blast.__init__
Bio.Blast.Record.DatabaseReport.__init__
Bio.Blast.Record.Description.__init__
Bio.Blast.Record.HSP.__init__
Bio.Blast.Record.Header.__init__
Bio.Blast.Record.MultipleAlignment.__init__
Bio.Blast.Record.PSIBlast.__init__
Bio.Blast.Record.Parameters.__init__
Bio.Blast.Record.Round.__init__
Bio.CAPS.CAPSMap.__init__
Bio.CAPS.DifferentialCutsite.__init__
Bio.Cluster.Record.__init__
Bio.Compass.Record.__init__
Bio.Crystal.Chain.__init__
Bio.Crystal.Crystal.__init__
Bio.Crystal.Hetero.__init__
Bio.Data.CodonTable.AmbiguousCodonTable.__init__
Bio.Data.CodonTable.AmbiguousForwardTable.__init__
Bio.Data.CodonTable.CodonTable.__init__
Bio.Data.CodonTable.NCBICodonTable.__init__
Bio.DocSQL.Create.__init__
Bio.DocSQL.Insert.__init__
Bio.DocSQL.IterationCursor.__init__
Bio.DocSQL.Query.__init__
Bio.DocSQL.QueryAll.__init__
Bio.DocSQL.QueryGeneric.__init__
Bio.DocSQL.QueryRow.__init__
Bio.DocSQL.QuerySingle.__init__
Bio.Emboss.Applications.DiffseqCommandline.__init__
Bio.Emboss.Applications.EInvertedCommandline.__init__
Bio.Emboss.Applications.ETandemCommandline.__init__
Bio.Emboss.Applications.Est2GenomeCommandline.__init__
Bio.Emboss.Applications.FConsenseCommandline.__init__
Bio.Emboss.Applications.FDNADistCommandline.__init__
Bio.Emboss.Applications.FDNAParsCommandline.__init__
Bio.Emboss.Applications.FNeighborCommandline.__init__
Bio.Emboss.Applications.FProtDistCommandline.__init__
Bio.Emboss.Applications.FProtParsCommandline.__init__
Bio.Emboss.Applications.FSeqBootCommandline.__init__
Bio.Emboss.Applications.FTreeDistCommandline.__init__
Bio.Emboss.Applications.FuzznucCommandline.__init__
Bio.Emboss.Applications.IepCommandline.__init__
Bio.Emboss.Applications.NeedleCommandline.__init__
Bio.Emboss.Applications.NeedleallCommandline.__init__
Bio.Emboss.Applications.PalindromeCommandline.__init__
Bio.Emboss.Applications.Primer3Commandline.__init__
Bio.Emboss.Applications.PrimerSearchCommandline.__init__
Bio.Emboss.Applications.SeqmatchallCommandline.__init__
Bio.Emboss.Applications.SeqretCommandline.__init__
Bio.Emboss.Applications.StretcherCommandline.__init__
Bio.Emboss.Applications.TranalignCommandline.__init__
Bio.Emboss.Applications.WaterCommandline.__init__
Bio.Emboss.Applications._EmbossCommandLine.__init__
Bio.Emboss.Applications._EmbossMinimalCommandLine.__init__
Bio.Emboss.Primer3.Primers.__init__
Bio.Emboss.Primer3.Record.__init__
Bio.Emboss.PrimerSearch.Amplifier.__init__
Bio.Emboss.PrimerSearch.InputRecord.__init__
Bio.Emboss.PrimerSearch.OutputRecord.__init__
Bio.Entrez.Parser.CorruptedXMLError.__init__
Bio.Entrez.Parser.DataHandler.__init__
Bio.Entrez.Parser.NotXMLError.__init__
Bio.Entrez.Parser.StructureElement.__init__
Bio.Entrez.Parser.ValidationError.__init__
Bio.ExPASy.Enzyme.Record.__init__
Bio.ExPASy.Prodoc.Record.__init__
Bio.ExPASy.Prodoc.Reference.__init__
Bio.ExPASy.Prosite.Record.__init__
Bio.ExPASy.ScanProsite.ContentHandler.__init__
Bio.ExPASy.ScanProsite.Parser.__init__
Bio.ExPASy.ScanProsite.Record.__init__
Bio.FSSP.FSSPAlignDict.__init__
Bio.FSSP.FSSPAlignRec.__init__
Bio.FSSP.FSSPHeader.__init__
Bio.FSSP.FSSPSumRec.__init__
Bio.FSSP.FSSPTools.FSSPMultAlign.__init__
Bio.FSSP.PosAlign.__init__
Bio.FSSP.fssp_rec.fff_rec.__init__
Bio.File.UndoHandle.__init__
Bio.File._IndexedSeqFileDict.__init__
Bio.File._SQLiteManySeqFilesDict.__init__
Bio.GA.Crossover.General.SafeFitnessCrossover.__init__
Bio.GA.Crossover.GeneralPoint.GeneralPointCrossover.__init__
Bio.GA.Crossover.GeneralPoint.InterleaveCrossover.__init__
Bio.GA.Crossover.Point.SinglePointCrossover.__init__
Bio.GA.Crossover.TwoPoint.TwoPointCrossover.__init__
Bio.GA.Crossover.Uniform.UniformCrossover.__init__
Bio.GA.Evolver.GenerationEvolver.__init__
Bio.GA.Evolver.SteadyStateEvolver.__init__
Bio.GA.Mutation.General.SafeFitnessMutation.__init__
Bio.GA.Mutation.Simple.ConversionMutation.__init__
Bio.GA.Mutation.Simple.SinglePositionMutation.__init__
Bio.GA.Organism.Organism.__init__
Bio.GA.Repair.Stabilizing.AmbiguousRepair.__init__
Bio.GA.Selection.Abstract.AbstractSelection.__init__
Bio.GA.Selection.Diversity.DiversitySelection.__init__
Bio.GA.Selection.RouletteWheel.RouletteWheelSelection.__init__
Bio.GA.Selection.Tournament.TournamentSelection.__init__
Bio.GenBank.FeatureParser.__init__
Bio.GenBank.Iterator.__init__
Bio.GenBank.Record.Feature.__init__
Bio.GenBank.Record.Qualifier.__init__
Bio.GenBank.Record.Record.__init__
Bio.GenBank.Record.Reference.__init__
Bio.GenBank.RecordParser.__init__
Bio.GenBank.Scanner.InsdcScanner.__init__
Bio.GenBank._BaseGenBankConsumer.__init__
Bio.GenBank._FeatureConsumer.__init__
Bio.GenBank._RecordConsumer.__init__
Bio.GenBank.utils.FeatureValueCleaner.__init__
Bio.Geo.Record.Record.__init__
Bio.Graphics.BasicChromosome.AnnotatedChromosomeSegment.__init__
Bio.Graphics.BasicChromosome.Chromosome.__init__
Bio.Graphics.BasicChromosome.ChromosomeSegment.__init__
Bio.Graphics.BasicChromosome.Organism.__init__
Bio.Graphics.BasicChromosome.TelomereSegment.__init__
Bio.Graphics.BasicChromosome._ChromosomeComponent.__init__
Bio.Graphics.ColorSpiral.ColorSpiral.__init__
Bio.Graphics.Comparative.ComparativeScatterPlot.__init__
Bio.Graphics.DisplayRepresentation.ChromosomeCounts.__init__
Bio.Graphics.Distribution.BarChartDistribution.__init__
Bio.Graphics.Distribution.DistributionPage.__init__
Bio.Graphics.Distribution.LineDistribution.__init__
Bio.Graphics.GenomeDiagram._AbstractDrawer.AbstractDrawer.__init__
Bio.Graphics.GenomeDiagram._CircularDrawer.CircularDrawer.__init__
Bio.Graphics.GenomeDiagram._Colors.ColorTranslator.__init__
Bio.Graphics.GenomeDiagram._CrossLink.CrossLink.__init__
Bio.Graphics.GenomeDiagram._Diagram.Diagram.__init__
Bio.Graphics.GenomeDiagram._Feature.Feature.__init__
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.__init__
Bio.Graphics.GenomeDiagram._Graph.GraphData.__init__
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.__init__
Bio.Graphics.GenomeDiagram._LinearDrawer.LinearDrawer.__init__
Bio.Graphics.GenomeDiagram._Track.Track.__init__
Bio.Graphics.KGML_vis.KGMLCanvas.__init__
Bio.HMM.DynamicProgramming.AbstractDPAlgorithms.__init__
Bio.HMM.DynamicProgramming.LogDPAlgorithms.__init__
Bio.HMM.DynamicProgramming.ScaledDPAlgorithms.__init__
Bio.HMM.MarkovModel.HiddenMarkovModel.__init__
Bio.HMM.MarkovModel.MarkovModelBuilder.__init__
Bio.HMM.Trainer.AbstractTrainer.__init__
Bio.HMM.Trainer.BaumWelchTrainer.__init__
Bio.HMM.Trainer.KnownStateTrainer.__init__
Bio.HMM.Trainer.TrainingSequence.__init__
Bio.Index._InMemoryIndex.__init__
Bio.Index._ShelveIndex.__init__
Bio.KDTree.KDTree'.KDTree.__init__
Bio.KEGG.Compound.Record.__init__
Bio.KEGG.Enzyme.Record.__init__
Bio.KEGG.KGML.KGML_parser.KGMLParser.__init__
Bio.KEGG.KGML.KGML_pathway.Component.__init__
Bio.KEGG.KGML.KGML_pathway.Entry.__init__
Bio.KEGG.KGML.KGML_pathway.Graphics.__init__
Bio.KEGG.KGML.KGML_pathway.Pathway.__init__
Bio.KEGG.KGML.KGML_pathway.Reaction.__init__
Bio.KEGG.KGML.KGML_pathway.Relation.__init__
Bio.LogisticRegression.LogisticRegression.__init__
Bio.MarkovModel.MarkovModel.__init__
Bio.MaxEntropy.MaxEntropy.__init__
Bio.Motif.Parsers.AlignAce.Record.__init__
Bio.Motif.Parsers.MAST.Record.__init__
Bio.Motif.Parsers.MEME.MEMEInstance.__init__
Bio.Motif.Parsers.MEME.MEMEMotif.__init__
Bio.Motif.Parsers.MEME.MEMERecord.__init__
Bio.Motif.Thresholds.ScoreDistribution.__init__
Bio.Motif._Motif.Motif.__init__
Bio.NMR.xpktools.Peaklist.__init__
Bio.NMR.xpktools.XpkEntry.__init__
Bio.NaiveBayes.NaiveBayes.__init__
Bio.NeuralNetwork.BackPropagation.Layer.AbstractLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.__init__
Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.__init__
Bio.NeuralNetwork.BackPropagation.Network.BasicNetwork.__init__
Bio.NeuralNetwork.Gene.Motif.MotifCoder.__init__
Bio.NeuralNetwork.Gene.Motif.MotifFinder.__init__
Bio.NeuralNetwork.Gene.Pattern.PatternIO.__init__
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.__init__
Bio.NeuralNetwork.Gene.Schema.DifferentialSchemaFitness.__init__
Bio.NeuralNetwork.Gene.Schema.GeneticAlgorithmFinder.__init__
Bio.NeuralNetwork.Gene.Schema.MostCountSchemaFitness.__init__
Bio.NeuralNetwork.Gene.Schema.RandomMotifGenerator.__init__
Bio.NeuralNetwork.Gene.Schema.Schema.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaCoder.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaFactory.__init__
Bio.NeuralNetwork.Gene.Schema.SchemaFinder.__init__
Bio.NeuralNetwork.Gene.Schema.SimpleFinisher.__init__
Bio.NeuralNetwork.Gene.Signature.SignatureCoder.__init__
Bio.NeuralNetwork.Gene.Signature.SignatureFinder.__init__
Bio.NeuralNetwork.StopTraining.ValidationIncreaseStop.__init__
Bio.NeuralNetwork.Training.ExampleManager.__init__
Bio.NeuralNetwork.Training.TrainingExample.__init__
Bio.Nexus.Nexus.Block.__init__
Bio.Nexus.Nexus.CharBuffer.__init__
Bio.Nexus.Nexus.Commandline.__init__
Bio.Nexus.Nexus.Nexus.__init__
Bio.Nexus.Nexus.StepMatrix.__init__
Bio.Nexus.Nodes.Chain.__init__
Bio.Nexus.Nodes.Node.__init__
Bio.Nexus.Trees.NodeData.__init__
Bio.Nexus.Trees.Tree.__init__
Bio.PDB.AbstractPropertyMap.AbstractAtomPropertyMap.__init__
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.__init__
Bio.PDB.AbstractPropertyMap.AbstractResiduePropertyMap.__init__
Bio.PDB.Atom.Atom.__init__
Bio.PDB.Atom.DisorderedAtom.__init__
Bio.PDB.Chain.Chain.__init__
Bio.PDB.DSSP'.DSSP.__init__
Bio.PDB.Dice.ChainSelector.__init__
Bio.PDB.Entity.DisorderedEntityWrapper.__init__
Bio.PDB.Entity.Entity.__init__
Bio.PDB.FragmentMapper'.Fragment.__init__
Bio.PDB.FragmentMapper'.FragmentMapper.__init__
Bio.PDB.HSExposure.ExposureCN.__init__
Bio.PDB.HSExposure.HSExposureCA.__init__
Bio.PDB.HSExposure.HSExposureCB.__init__
Bio.PDB.HSExposure._AbstractHSExposure.__init__
Bio.PDB.MMCIF2Dict.MMCIF2Dict.__init__
Bio.PDB.MMCIFParser'.MMCIFParser.__init__
Bio.PDB.Model.Model.__init__
Bio.PDB.NACCESS.NACCESS.__init__
Bio.PDB.NACCESS.NACCESS_atomic.__init__
Bio.PDB.NeighborSearch'.NeighborSearch.__init__
Bio.PDB.PDBIO'.PDBIO.__init__
Bio.PDB.PDBList'.PDBList.__init__
Bio.PDB.PDBParser'.PDBParser.__init__
Bio.PDB.PSEA.PSEA.__init__
Bio.PDB.Polypeptide.CaPPBuilder.__init__
Bio.PDB.Polypeptide.PPBuilder.__init__
Bio.PDB.Polypeptide._PPBuilder.__init__
Bio.PDB.QCPSuperimposer.QCPSuperimposer.__init__
Bio.PDB.Residue.DisorderedResidue.__init__
Bio.PDB.Residue.Residue.__init__
Bio.PDB.ResidueDepth'.ResidueDepth.__init__
Bio.PDB.Structure.Structure.__init__
Bio.PDB.StructureAlignment'.StructureAlignment.__init__
Bio.PDB.StructureBuilder.StructureBuilder.__init__
Bio.PDB.Superimposer'.Superimposer.__init__
Bio.PDB.Vector'.Vector.__init__
Bio.ParserSupport.EventGenerator.__init__
Bio.ParserSupport.TaggingConsumer.__init__
Bio.Pathway.Network.__init__
Bio.Pathway.Reaction.__init__
Bio.Pathway.Rep.Graph.Graph.__init__
Bio.Pathway.Rep.MultiGraph.MultiGraph.__init__
Bio.Pathway.System.__init__
Bio.Phylo.Applications._Fasttree.FastTreeCommandline.__init__
Bio.Phylo.Applications._Phyml.PhymlCommandline.__init__
Bio.Phylo.Applications._Raxml.RaxmlCommandline.__init__
Bio.Phylo.BaseTree.BranchColor.__init__
Bio.Phylo.BaseTree.Clade.__init__
Bio.Phylo.BaseTree.Tree.__init__
Bio.Phylo.CDAO.Clade.__init__
Bio.Phylo.CDAO.Tree.__init__
Bio.Phylo.CDAOIO.Parser.__init__
Bio.Phylo.CDAOIO.Writer.__init__
Bio.Phylo.NeXML.Clade.__init__
Bio.Phylo.NeXML.Tree.__init__
Bio.Phylo.NeXMLIO.Parser.__init__
Bio.Phylo.NeXMLIO.Writer.__init__
Bio.Phylo.Newick.Clade.__init__
Bio.Phylo.Newick.Tree.__init__
Bio.Phylo.NewickIO.Parser.__init__
Bio.Phylo.NewickIO.Writer.__init__
Bio.Phylo.PAML._paml.Paml.__init__
Bio.Phylo.PAML.baseml.Baseml.__init__
Bio.Phylo.PAML.codeml.Codeml.__init__
Bio.Phylo.PAML.yn00.Yn00.__init__
Bio.Phylo.PhyloXML.Accession.__init__
Bio.Phylo.PhyloXML.Annotation.__init__
Bio.Phylo.PhyloXML.BinaryCharacters.__init__
Bio.Phylo.PhyloXML.BranchColor.__init__
Bio.Phylo.PhyloXML.Clade.__init__
Bio.Phylo.PhyloXML.CladeRelation.__init__
Bio.Phylo.PhyloXML.Confidence.__init__
Bio.Phylo.PhyloXML.Date.__init__
Bio.Phylo.PhyloXML.Distribution.__init__
Bio.Phylo.PhyloXML.DomainArchitecture.__init__
Bio.Phylo.PhyloXML.Events.__init__
Bio.Phylo.PhyloXML.Id.__init__
Bio.Phylo.PhyloXML.MolSeq.__init__
Bio.Phylo.PhyloXML.Other.__init__
Bio.Phylo.PhyloXML.Phylogeny.__init__
Bio.Phylo.PhyloXML.Phyloxml.__init__
Bio.Phylo.PhyloXML.Point.__init__
Bio.Phylo.PhyloXML.Polygon.__init__
Bio.Phylo.PhyloXML.Property.__init__
Bio.Phylo.PhyloXML.ProteinDomain.__init__
Bio.Phylo.PhyloXML.Reference.__init__
Bio.Phylo.PhyloXML.Sequence.__init__
Bio.Phylo.PhyloXML.SequenceRelation.__init__
Bio.Phylo.PhyloXML.Taxonomy.__init__
Bio.Phylo.PhyloXML.Uri.__init__
Bio.Phylo.PhyloXMLIO.Parser.__init__
Bio.Phylo.PhyloXMLIO.Writer.__init__
Bio.Phylo.TreeConstruction.DistanceCalculator.__init__
Bio.Phylo.TreeConstruction.DistanceTreeConstructor.__init__
Bio.Phylo.TreeConstruction.NNITreeSearcher.__init__
Bio.Phylo.TreeConstruction.ParsimonyScorer.__init__
Bio.Phylo.TreeConstruction.ParsimonyTreeConstructor.__init__
Bio.Phylo.TreeConstruction._DistanceMatrix.__init__
Bio.Phylo.TreeConstruction._Matrix.__init__
Bio.PopGen.Async.Async.__init__
Bio.PopGen.Async.DirectoryRetriever.__init__
Bio.PopGen.Async.FileRetriever.__init__
Bio.PopGen.Async.Local.Local.__init__
Bio.PopGen.FDist.Async.FDistAsync.__init__
Bio.PopGen.FDist.Async.SplitFDist.__init__
Bio.PopGen.FDist.Controller.FDistController.__init__
Bio.PopGen.FDist.Record.__init__
Bio.PopGen.GenePop.Controller.GenePopController.__init__
Bio.PopGen.GenePop.Controller._FileIterator.__init__
Bio.PopGen.GenePop.Controller._GenePopCommandline.__init__
Bio.PopGen.GenePop.EasyController.EasyController.__init__
Bio.PopGen.GenePop.FileParser.FileRecord.__init__
Bio.PopGen.GenePop.LargeFileParser.Record.__init__
Bio.PopGen.GenePop.Record.__init__
Bio.PopGen.SimCoal.Async.SimCoalCache.__init__
Bio.PopGen.SimCoal.Cache.SimCoalCache.__init__
Bio.PopGen.SimCoal.Controller.FastSimCoalController.__init__
Bio.PopGen.SimCoal.Controller.SimCoalController.__init__
Bio.PopGen.SimCoal.Controller._FastSimCoalCommandLine.__init__
Bio.Restriction.PrintFormat.PrintFormat.__init__
Bio.Restriction.Restriction.Analysis.__init__
Bio.Restriction.Restriction.FormattedSeq.__init__
Bio.Restriction.Restriction.RestrictionBatch.__init__
Bio.Restriction.Restriction.RestrictionType.__init__
Bio.SCOP.Astral.__init__
Bio.SCOP.Cla.Index.__init__
Bio.SCOP.Cla.Record.__init__
Bio.SCOP.Des.Record.__init__
Bio.SCOP.Dom.Record.__init__
Bio.SCOP.Domain.__init__
Bio.SCOP.Hie.Record.__init__
Bio.SCOP.Node.__init__
Bio.SCOP.Raf.Res.__init__
Bio.SCOP.Raf.SeqMap.__init__
Bio.SCOP.Raf.SeqMapIndex.__init__
Bio.SCOP.Residues.Residues.__init__
Bio.SCOP.Scop.__init__
Bio.SVDSuperimposer.SVDSuperimposer.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabIndexer.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabParser.__init__
Bio.SearchIO.BlastIO.blast_tab.BlastTabWriter.__init__
Bio.SearchIO.BlastIO.blast_text.BlastTextParser.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlIndexer.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlParser.__init__
Bio.SearchIO.BlastIO.blast_xml.BlastXmlWriter.__init__
Bio.SearchIO.BlastIO.blast_xml._BlastXmlGenerator.__init__
Bio.SearchIO.BlatIO.BlatPslIndexer.__init__
Bio.SearchIO.BlatIO.BlatPslParser.__init__
Bio.SearchIO.BlatIO.BlatPslWriter.__init__
Bio.SearchIO.ExonerateIO._base._BaseExonerateParser.__init__
Bio.SearchIO.FastaIO.FastaM10Indexer.__init__
Bio.SearchIO.FastaIO.FastaM10Parser.__init__
Bio.SearchIO.HmmerIO._base._BaseHmmerTextIndexer.__init__
Bio.SearchIO.HmmerIO.hmmer2_text.Hmmer2TextParser.__init__
Bio.SearchIO.HmmerIO.hmmer3_domtab.Hmmer3DomtabHmmhitWriter.__init__
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabParser.__init__
Bio.SearchIO.HmmerIO.hmmer3_tab.Hmmer3TabWriter.__init__
Bio.SearchIO.HmmerIO.hmmer3_text.Hmmer3TextParser.__init__
Bio.SearchIO._index.SearchIndexer.__init__
Bio.SearchIO._model.hit.Hit.__init__
Bio.SearchIO._model.hsp.HSP.__init__
Bio.SearchIO._model.hsp.HSPFragment.__init__
Bio.SearchIO._model.query.QueryResult.__init__
Bio.Seq.MutableSeq.__init__
Bio.Seq.Seq.__init__
Bio.Seq.UnknownSeq.__init__
Bio.SeqFeature.CompoundLocation.__init__
Bio.SeqFeature.FeatureLocation.__init__
Bio.SeqFeature.PositionGap.__init__
Bio.SeqFeature.Reference.__init__
Bio.SeqFeature.SeqFeature.__init__
Bio.SeqIO.FastaIO.FastaWriter.__init__
Bio.SeqIO.Interfaces.SequenceIterator.__init__
Bio.SeqIO.Interfaces.SequenceWriter.__init__
Bio.SeqIO.Interfaces.SequentialSequenceWriter.__init__
Bio.SeqIO.PhdIO.PhdWriter.__init__
Bio.SeqIO.QualityIO.QualPhredWriter.__init__
Bio.SeqIO.SeqXmlIO.SeqXmlIterator.__init__
Bio.SeqIO.SeqXmlIO.SeqXmlWriter.__init__
Bio.SeqIO.SeqXmlIO.XMLRecordIterator.__init__
Bio.SeqIO.SffIO.SffWriter.__init__
Bio.SeqIO.SffIO._AddTellHandle.__init__
Bio.SeqIO.UniprotIO.Parser.__init__
Bio.SeqIO._index.IntelliGeneticsRandomAccess.__init__
Bio.SeqIO._index.SeqFileRandomAccess.__init__
Bio.SeqIO._index.SequentialSeqFileRandomAccess.__init__
Bio.SeqIO._index.SffRandomAccess.__init__
Bio.SeqRecord.SeqRecord.__init__
Bio.SeqRecord._RestrictedDict.__init__
Bio.SeqUtils.CodonUsage.CodonAdaptationIndex.__init__
Bio.SeqUtils.IsoelectricPoint.IsoelectricPoint.__init__
Bio.SeqUtils.ProtParam.ProteinAnalysis.__init__
Bio.Sequencing.Ace.ACEFileRecord.__init__
Bio.Sequencing.Ace.Contig.__init__
Bio.Sequencing.Ace.Reads.__init__
Bio.Sequencing.Ace.af.__init__
Bio.Sequencing.Ace.bs.__init__
Bio.Sequencing.Ace.ct.__init__
Bio.Sequencing.Ace.ds.__init__
Bio.Sequencing.Ace.qa.__init__
Bio.Sequencing.Ace.rd.__init__
Bio.Sequencing.Ace.rt.__init__
Bio.Sequencing.Ace.wa.__init__
Bio.Sequencing.Ace.wr.__init__
Bio.Sequencing.Applications._Novoalign.NovoalignCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaAlignCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaBwaswCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaIndexCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaSampeCommandline.__init__
Bio.Sequencing.Applications._bwa.BwaSamseCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsCalmdCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsCatCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsFaidxCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsFixmateCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsIdxstatsCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsIndexCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsMergeCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsMpileupCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsPhaseCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsReheaderCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsRmdupCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsSortCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsTargetcutCommandline.__init__
Bio.Sequencing.Applications._samtools.SamtoolsViewCommandline.__init__
Bio.Sequencing.Phd.Record.__init__
Bio.SubsMat.FreqTable.FreqTable.__init__
Bio.SubsMat.SeqMat.__init__
Bio.SwissProt.KeyWList.Record.__init__
Bio.SwissProt.Record.__init__
Bio.SwissProt.Reference.__init__
Bio.UniGene.ProtsimLine.__init__
Bio.UniGene.Record.__init__
Bio.UniGene.STSLine.__init__
Bio.UniGene.SequenceLine.__init__
Bio.Wise.dnal.Statistics.__init__
Bio.Wise.psw.AlignmentColumn.__init__
Bio.Wise.psw.ColumnUnit.__init__
Bio._py3k._ordereddict.OrderedDict.__init__
Bio.bgzf.BgzfReader.__init__
Bio.bgzf.BgzfWriter.__init__
Bio.codonalign.codonalignment.CodonAlignment.__init__
Bio.codonalign.codonseq.CodonSeq.__init__
Bio.kNN.kNN.__init__
Bio.motifs.Instances.__init__
Bio.motifs.Motif.__init__
Bio.motifs.alignace.Record.__init__
Bio.motifs.applications._alignace.AlignAceCommandline.__init__
Bio.motifs.applications._alignace.CompareAceCommandline.__init__
Bio.motifs.applications._xxmotif.XXmotifCommandline.__init__
Bio.motifs.jaspar.Motif.__init__
Bio.motifs.jaspar.Record.__init__
Bio.motifs.jaspar.db.JASPAR5.__init__
Bio.motifs.mast.Record.__init__
Bio.motifs.matrix.GenericPositionMatrix.__init__
Bio.motifs.matrix.PositionWeightMatrix.__init__
Bio.motifs.meme.Instance.__init__
Bio.motifs.meme.Motif.__init__
Bio.motifs.meme.Record.__init__
Bio.motifs.thresholds.ScoreDistribution.__init__
Bio.motifs.transfac.Record.__init__
Bio.pairwise2.affine_penalty.__init__
Bio.pairwise2.dictionary_match.__init__
Bio.pairwise2.identity_match.__init__
BioSQL.BioSeq.DBSeq.__init__
BioSQL.BioSeq.DBSeqRecord.__init__
BioSQL.BioSeqDatabase.Adaptor.__init__
BioSQL.BioSeqDatabase.BioSeqDatabase.__init__
BioSQL.BioSeqDatabase.DBServer.__init__
BioSQL.BioSeqDatabase._CursorWrapper.__init__
BioSQL.DBUtils.Generic_dbutils.__init__
BioSQL.Loader.DatabaseLoader.__init__
BioSQL.Loader.DatabaseRemover.__init__" class="py-name" href="#" onclick="return doclink('link-228', '__init__', 'link-228');">__init__</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L1495"></a><tt class="py-lineno">1495</tt>  <tt class="py-line">        <tt class="py-comment"># self.handle = handle</tt> </tt>
<a name="L1496"></a><tt class="py-lineno">1496</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">wrap</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L1497"></a><tt class="py-lineno">1497</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">wrap</tt><tt class="py-op">:</tt> </tt>
<a name="L1498"></a><tt class="py-lineno">1498</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">wrap</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L1499"></a><tt class="py-lineno">1499</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt> </tt>
<a name="L1500"></a><tt class="py-lineno">1500</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">wrap</tt> <tt class="py-op">=</tt> <tt class="py-name">wrap</tt> </tt>
<a name="L1501"></a><tt class="py-lineno">1501</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">record2title</tt> <tt class="py-op">=</tt> <tt class="py-name">record2title</tt> </tt>
</div><a name="L1502"></a><tt class="py-lineno">1502</tt>  <tt class="py-line"> </tt>
<a name="QualPhredWriter.write_record"></a><div id="QualPhredWriter.write_record-def"><a name="L1503"></a><tt class="py-lineno">1503</tt> <a class="py-toggle" href="#" id="QualPhredWriter.write_record-toggle" onclick="return toggle('QualPhredWriter.write_record');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.QualityIO.QualPhredWriter-class.html#write_record">write_record</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="QualPhredWriter.write_record-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="QualPhredWriter.write_record-expanded"><a name="L1504"></a><tt class="py-lineno">1504</tt>  <tt class="py-line">        <tt class="py-docstring">"""Write a single QUAL record to the file."""</tt> </tt>
<a name="L1505"></a><tt class="py-lineno">1505</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_header_written</tt> </tt>
<a name="L1506"></a><tt class="py-lineno">1506</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_footer_written</tt> </tt>
<a name="L1507"></a><tt class="py-lineno">1507</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_record_written</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L1508"></a><tt class="py-lineno">1508</tt>  <tt class="py-line"> </tt>
<a name="L1509"></a><tt class="py-lineno">1509</tt>  <tt class="py-line">        <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt> </tt>
<a name="L1510"></a><tt class="py-lineno">1510</tt>  <tt class="py-line">        <tt class="py-name">wrap</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">wrap</tt> </tt>
<a name="L1511"></a><tt class="py-lineno">1511</tt>  <tt class="py-line"> </tt>
<a name="L1512"></a><tt class="py-lineno">1512</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">record2title</tt><tt class="py-op">:</tt> </tt>
<a name="L1513"></a><tt class="py-lineno">1513</tt>  <tt class="py-line">            <tt id="link-229" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-229', 'title', 'link-217');">title</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-230" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentWriter.clean
Bio.SeqIO.Interfaces.SequenceWriter.clean" class="py-name" href="#" onclick="return doclink('link-230', 'clean', 'link-208');">clean</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">record2title</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1514"></a><tt class="py-lineno">1514</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1515"></a><tt class="py-lineno">1515</tt>  <tt class="py-line">            <tt id="link-231" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-231', 'id', 'link-24');">id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-232" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentWriter.clean
Bio.SeqIO.Interfaces.SequenceWriter.clean" class="py-name" href="#" onclick="return doclink('link-232', 'clean', 'link-208');">clean</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-233" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-233', 'id', 'link-24');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1516"></a><tt class="py-lineno">1516</tt>  <tt class="py-line">            <tt id="link-234" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-234', 'description', 'link-128');">description</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-235" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentWriter.clean
Bio.SeqIO.Interfaces.SequenceWriter.clean" class="py-name" href="#" onclick="return doclink('link-235', 'clean', 'link-208');">clean</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-236" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-236', 'description', 'link-128');">description</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1517"></a><tt class="py-lineno">1517</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-237" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-237', 'description', 'link-128');">description</a></tt> <tt class="py-keyword">and</tt> <tt id="link-238" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-238', 'description', 'link-128');">description</a></tt><tt class="py-op">.</tt><tt id="link-239" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-239', 'split', 'link-118');">split</a></tt><tt class="py-op">(</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt id="link-240" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-240', 'id', 'link-24');">id</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1518"></a><tt class="py-lineno">1518</tt>  <tt class="py-line">                <tt class="py-comment"># The description includes the id at the start</tt> </tt>
<a name="L1519"></a><tt class="py-lineno">1519</tt>  <tt class="py-line">                <tt id="link-241" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-241', 'title', 'link-217');">title</a></tt> <tt class="py-op">=</tt> <tt id="link-242" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-242', 'description', 'link-128');">description</a></tt> </tt>
<a name="L1520"></a><tt class="py-lineno">1520</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-243" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-243', 'description', 'link-128');">description</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1521"></a><tt class="py-lineno">1521</tt>  <tt class="py-line">                <tt id="link-244" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-244', 'title', 'link-217');">title</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"%s %s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-245" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-245', 'id', 'link-24');">id</a></tt><tt class="py-op">,</tt> <tt id="link-246" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-246', 'description', 'link-128');">description</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1522"></a><tt class="py-lineno">1522</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1523"></a><tt class="py-lineno">1523</tt>  <tt class="py-line">                <tt id="link-247" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-247', 'title', 'link-217');">title</a></tt> <tt class="py-op">=</tt> <tt id="link-248" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-248', 'id', 'link-24');">id</a></tt> </tt>
<a name="L1524"></a><tt class="py-lineno">1524</tt>  <tt class="py-line">        <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-249" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-249', 'write', 'link-225');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"&gt;%s\n"</tt> <tt class="py-op">%</tt> <tt id="link-250" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-250', 'title', 'link-217');">title</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1525"></a><tt class="py-lineno">1525</tt>  <tt class="py-line"> </tt>
<a name="L1526"></a><tt class="py-lineno">1526</tt>  <tt class="py-line">        <tt class="py-name">qualities</tt> <tt class="py-op">=</tt> <tt id="link-251" class="py-name" targets="Function Bio.SeqIO.QualityIO._get_phred_quality()=Bio.SeqIO.QualityIO-module.html#_get_phred_quality"><a title="Bio.SeqIO.QualityIO._get_phred_quality" class="py-name" href="#" onclick="return doclink('link-251', '_get_phred_quality', 'link-251');">_get_phred_quality</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L1527"></a><tt class="py-lineno">1527</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1528"></a><tt class="py-lineno">1528</tt>  <tt class="py-line">            <tt class="py-comment"># This rounds to the nearest integer.</tt> </tt>
<a name="L1529"></a><tt class="py-lineno">1529</tt>  <tt class="py-line">            <tt class="py-comment"># TODO - can we record a float in a qual file?</tt> </tt>
<a name="L1530"></a><tt class="py-lineno">1530</tt>  <tt class="py-line">            <tt class="py-name">qualities_strs</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-string">"%i"</tt> <tt class="py-op">%</tt> <tt id="link-252" class="py-name"><a title="Bio.Blast.NCBIStandalone._BlastConsumer.round
Bio.Blast.NCBIStandalone._DescriptionConsumer.round" class="py-name" href="#" onclick="return doclink('link-252', 'round', 'link-31');">round</a></tt><tt class="py-op">(</tt><tt class="py-name">q</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">q</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qualities</tt><tt class="py-op">]</tt> </tt>
<a name="L1531"></a><tt class="py-lineno">1531</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">TypeError</tt> <tt class="py-keyword">as</tt> <tt class="py-name">e</tt><tt class="py-op">:</tt> </tt>
<a name="L1532"></a><tt class="py-lineno">1532</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">None</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qualities</tt><tt class="py-op">:</tt> </tt>
<a name="L1533"></a><tt class="py-lineno">1533</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"A quality value of None was found"</tt><tt class="py-op">)</tt> </tt>
<a name="L1534"></a><tt class="py-lineno">1534</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1535"></a><tt class="py-lineno">1535</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">e</tt> </tt>
<a name="L1536"></a><tt class="py-lineno">1536</tt>  <tt class="py-line"> </tt>
<a name="L1537"></a><tt class="py-lineno">1537</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">wrap</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">5</tt><tt class="py-op">:</tt> </tt>
<a name="L1538"></a><tt class="py-lineno">1538</tt>  <tt class="py-line">            <tt class="py-comment"># Fast wrapping</tt> </tt>
<a name="L1539"></a><tt class="py-lineno">1539</tt>  <tt class="py-line">            <tt id="link-253" class="py-name" targets="Variable BioSQL.BioSeq.DBSeq.data=BioSQL.BioSeq.DBSeq-class.html#data"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-253', 'data', 'link-253');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-string">" "</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">qualities_strs</tt><tt class="py-op">)</tt> </tt>
<a name="L1540"></a><tt class="py-lineno">1540</tt>  <tt class="py-line">            <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L1541"></a><tt class="py-lineno">1541</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt id="link-254" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-254', 'data', 'link-253');">data</a></tt><tt class="py-op">)</tt> <tt class="py-op">&lt;=</tt> <tt class="py-name">wrap</tt><tt class="py-op">:</tt> </tt>
<a name="L1542"></a><tt class="py-lineno">1542</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-255" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-255', 'write', 'link-225');">write</a></tt><tt class="py-op">(</tt><tt id="link-256" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-256', 'data', 'link-253');">data</a></tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L1543"></a><tt class="py-lineno">1543</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L1544"></a><tt class="py-lineno">1544</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1545"></a><tt class="py-lineno">1545</tt>  <tt class="py-line">                    <tt class="py-comment"># By construction there must be spaces in the first X chars</tt> </tt>
<a name="L1546"></a><tt class="py-lineno">1546</tt>  <tt class="py-line">                    <tt class="py-comment"># (unless we have X digit or higher quality scores!)</tt> </tt>
<a name="L1547"></a><tt class="py-lineno">1547</tt>  <tt class="py-line">                    <tt id="link-257" class="py-name" targets="Variable Bio.PDB.Polypeptide.i=Bio.PDB.Polypeptide-module.html#i,Variable Bio.SeqIO.SffIO.i=Bio.SeqIO.SffIO-module.html#i"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-257', 'i', 'link-257');">i</a></tt> <tt class="py-op">=</tt> <tt id="link-258" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-258', 'data', 'link-253');">data</a></tt><tt class="py-op">.</tt><tt id="link-259" class="py-name" targets="Method Bio.Seq.Seq.rfind()=Bio.Seq.Seq-class.html#rfind"><a title="Bio.Seq.Seq.rfind" class="py-name" href="#" onclick="return doclink('link-259', 'rfind', 'link-259');">rfind</a></tt><tt class="py-op">(</tt><tt class="py-string">" "</tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">wrap</tt><tt class="py-op">)</tt> </tt>
<a name="L1548"></a><tt class="py-lineno">1548</tt>  <tt class="py-line">                    <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-260" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-260', 'write', 'link-225');">write</a></tt><tt class="py-op">(</tt><tt id="link-261" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-261', 'data', 'link-253');">data</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt id="link-262" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-262', 'i', 'link-257');">i</a></tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L1549"></a><tt class="py-lineno">1549</tt>  <tt class="py-line">                    <tt id="link-263" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-263', 'data', 'link-253');">data</a></tt> <tt class="py-op">=</tt> <tt id="link-264" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-264', 'data', 'link-253');">data</a></tt><tt class="py-op">[</tt><tt id="link-265" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-265', 'i', 'link-257');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L1550"></a><tt class="py-lineno">1550</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">wrap</tt><tt class="py-op">:</tt> </tt>
<a name="L1551"></a><tt class="py-lineno">1551</tt>  <tt class="py-line">            <tt class="py-comment"># Safe wrapping</tt> </tt>
<a name="L1552"></a><tt class="py-lineno">1552</tt>  <tt class="py-line">            <tt class="py-keyword">while</tt> <tt class="py-name">qualities_strs</tt><tt class="py-op">:</tt> </tt>
<a name="L1553"></a><tt class="py-lineno">1553</tt>  <tt class="py-line">                <tt class="py-name">line</tt> <tt class="py-op">=</tt> <tt class="py-name">qualities_strs</tt><tt class="py-op">.</tt><tt id="link-266" class="py-name" targets="Method Bio.File._IndexedSeqFileDict.pop()=Bio.File._IndexedSeqFileDict-class.html#pop,Method Bio.SearchIO._model.hit.Hit.pop()=Bio.SearchIO._model.hit.Hit-class.html#pop,Method Bio.SearchIO._model.query.QueryResult.pop()=Bio.SearchIO._model.query.QueryResult-class.html#pop,Method Bio.Seq.MutableSeq.pop()=Bio.Seq.MutableSeq-class.html#pop,Method Bio._py3k._ordereddict.OrderedDict.pop()=Bio._py3k._ordereddict.OrderedDict-class.html#pop"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop
Bio._py3k._ordereddict.OrderedDict.pop" class="py-name" href="#" onclick="return doclink('link-266', 'pop', 'link-266');">pop</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L1554"></a><tt class="py-lineno">1554</tt>  <tt class="py-line">                <tt class="py-keyword">while</tt> <tt class="py-name">qualities_strs</tt> \ </tt>
<a name="L1555"></a><tt class="py-lineno">1555</tt>  <tt class="py-line">                        <tt class="py-keyword">and</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">line</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt> <tt class="py-op">+</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">qualities_strs</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">&lt;</tt> <tt class="py-name">wrap</tt><tt class="py-op">:</tt> </tt>
<a name="L1556"></a><tt class="py-lineno">1556</tt>  <tt class="py-line">                    <tt class="py-name">line</tt> <tt class="py-op">+=</tt> <tt class="py-string">" "</tt> <tt class="py-op">+</tt> <tt class="py-name">qualities_strs</tt><tt class="py-op">.</tt><tt id="link-267" class="py-name"><a title="Bio.File._IndexedSeqFileDict.pop
Bio.SearchIO._model.hit.Hit.pop
Bio.SearchIO._model.query.QueryResult.pop
Bio.Seq.MutableSeq.pop
Bio._py3k._ordereddict.OrderedDict.pop" class="py-name" href="#" onclick="return doclink('link-267', 'pop', 'link-266');">pop</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L1557"></a><tt class="py-lineno">1557</tt>  <tt class="py-line">                <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-268" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-268', 'write', 'link-225');">write</a></tt><tt class="py-op">(</tt><tt class="py-name">line</tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt><tt class="py-op">)</tt> </tt>
<a name="L1558"></a><tt class="py-lineno">1558</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1559"></a><tt class="py-lineno">1559</tt>  <tt class="py-line">            <tt class="py-comment"># No wrapping</tt> </tt>
<a name="L1560"></a><tt class="py-lineno">1560</tt>  <tt class="py-line">            <tt id="link-269" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-269', 'data', 'link-253');">data</a></tt> <tt class="py-op">=</tt> <tt class="py-string">" "</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">qualities_strs</tt><tt class="py-op">)</tt> </tt>
<a name="L1561"></a><tt class="py-lineno">1561</tt>  <tt class="py-line">            <tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-270" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-270', 'write', 'link-225');">write</a></tt><tt class="py-op">(</tt><tt id="link-271" class="py-name"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-271', 'data', 'link-253');">data</a></tt> <tt class="py-op">+</tt> <tt class="py-string">"\n"</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1562"></a><tt class="py-lineno">1562</tt>  <tt class="py-line"> </tt>
<a name="L1563"></a><tt class="py-lineno">1563</tt>  <tt class="py-line"> </tt>
<a name="FastqSolexaWriter"></a><div id="FastqSolexaWriter-def"><a name="L1564"></a><tt class="py-lineno">1564</tt> <a class="py-toggle" href="#" id="FastqSolexaWriter-toggle" onclick="return toggle('FastqSolexaWriter');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.SeqIO.QualityIO.FastqSolexaWriter-class.html">FastqSolexaWriter</a><tt class="py-op">(</tt><tt class="py-base-class">SequentialSequenceWriter</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FastqSolexaWriter-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="FastqSolexaWriter-expanded"><a name="L1565"></a><tt class="py-lineno">1565</tt>  <tt class="py-line">    <tt class="py-docstring">r"""Write old style Solexa/Illumina FASTQ format files (with Solexa qualities).</tt> </tt>
<a name="L1566"></a><tt class="py-lineno">1566</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1567"></a><tt class="py-lineno">1567</tt>  <tt class="py-line"><tt class="py-docstring">    This outputs FASTQ files like those from the early Solexa/Illumina</tt> </tt>
<a name="L1568"></a><tt class="py-lineno">1568</tt>  <tt class="py-line"><tt class="py-docstring">    pipeline, using Solexa scores and an ASCII offset of 64. These are</tt> </tt>
<a name="L1569"></a><tt class="py-lineno">1569</tt>  <tt class="py-line"><tt class="py-docstring">    NOT compatible with the standard Sanger style PHRED FASTQ files.</tt> </tt>
<a name="L1570"></a><tt class="py-lineno">1570</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1571"></a><tt class="py-lineno">1571</tt>  <tt class="py-line"><tt class="py-docstring">    If your records contain a "solexa_quality" entry under letter_annotations,</tt> </tt>
<a name="L1572"></a><tt class="py-lineno">1572</tt>  <tt class="py-line"><tt class="py-docstring">    this is used, otherwise any "phred_quality" entry will be used after</tt> </tt>
<a name="L1573"></a><tt class="py-lineno">1573</tt>  <tt class="py-line"><tt class="py-docstring">    conversion using the solexa_quality_from_phred function. If neither style</tt> </tt>
<a name="L1574"></a><tt class="py-lineno">1574</tt>  <tt class="py-line"><tt class="py-docstring">    of quality scores are present, an exception is raised.</tt> </tt>
<a name="L1575"></a><tt class="py-lineno">1575</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1576"></a><tt class="py-lineno">1576</tt>  <tt class="py-line"><tt class="py-docstring">    Although you can use this class directly, you are strongly encouraged</tt> </tt>
<a name="L1577"></a><tt class="py-lineno">1577</tt>  <tt class="py-line"><tt class="py-docstring">    to use the Bio.SeqIO.write() function instead.  For example, this code</tt> </tt>
<a name="L1578"></a><tt class="py-lineno">1578</tt>  <tt class="py-line"><tt class="py-docstring">    reads in a FASTQ file and re-saves it as another FASTQ file:</tt> </tt>
<a name="L1579"></a><tt class="py-lineno">1579</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1580"></a><tt class="py-lineno">1580</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L1581"></a><tt class="py-lineno">1581</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; record_iterator = SeqIO.parse("Quality/solexa_example.fastq", "fastq-solexa")</tt> </tt>
<a name="L1582"></a><tt class="py-lineno">1582</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; with open("Quality/temp.fastq", "w") as out_handle:</tt> </tt>
<a name="L1583"></a><tt class="py-lineno">1583</tt>  <tt class="py-line"><tt class="py-docstring">    ...     SeqIO.write(record_iterator, out_handle, "fastq-solexa")</tt> </tt>
<a name="L1584"></a><tt class="py-lineno">1584</tt>  <tt class="py-line"><tt class="py-docstring">    5</tt> </tt>
<a name="L1585"></a><tt class="py-lineno">1585</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1586"></a><tt class="py-lineno">1586</tt>  <tt class="py-line"><tt class="py-docstring">    You might want to do this if the original file included extra line breaks,</tt> </tt>
<a name="L1587"></a><tt class="py-lineno">1587</tt>  <tt class="py-line"><tt class="py-docstring">    which (while valid) may not be supported by all tools.  The output file</tt> </tt>
<a name="L1588"></a><tt class="py-lineno">1588</tt>  <tt class="py-line"><tt class="py-docstring">    from Biopython will have each sequence on a single line, and each quality</tt> </tt>
<a name="L1589"></a><tt class="py-lineno">1589</tt>  <tt class="py-line"><tt class="py-docstring">    string on a single line (which is considered desirable for maximum</tt> </tt>
<a name="L1590"></a><tt class="py-lineno">1590</tt>  <tt class="py-line"><tt class="py-docstring">    compatibility).</tt> </tt>
<a name="L1591"></a><tt class="py-lineno">1591</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1592"></a><tt class="py-lineno">1592</tt>  <tt class="py-line"><tt class="py-docstring">    This code is also called if you use the .format("fastq-solexa") method of</tt> </tt>
<a name="L1593"></a><tt class="py-lineno">1593</tt>  <tt class="py-line"><tt class="py-docstring">    a SeqRecord. For example,</tt> </tt>
<a name="L1594"></a><tt class="py-lineno">1594</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1595"></a><tt class="py-lineno">1595</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; record = SeqIO.read("Quality/sanger_faked.fastq", "fastq-sanger")</tt> </tt>
<a name="L1596"></a><tt class="py-lineno">1596</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(record.format("fastq-solexa"))</tt> </tt>
<a name="L1597"></a><tt class="py-lineno">1597</tt>  <tt class="py-line"><tt class="py-docstring">    @Test PHRED qualities from 40 to 0 inclusive</tt> </tt>
<a name="L1598"></a><tt class="py-lineno">1598</tt>  <tt class="py-line"><tt class="py-docstring">    ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTN</tt> </tt>
<a name="L1599"></a><tt class="py-lineno">1599</tt>  <tt class="py-line"><tt class="py-docstring">    +</tt> </tt>
<a name="L1600"></a><tt class="py-lineno">1600</tt>  <tt class="py-line"><tt class="py-docstring">    hgfedcba`_^]\[ZYXWVUTSRQPONMLKJHGFECB@&gt;;;</tt> </tt>
<a name="L1601"></a><tt class="py-lineno">1601</tt>  <tt class="py-line"><tt class="py-docstring">    &lt;BLANKLINE&gt;</tt> </tt>
<a name="L1602"></a><tt class="py-lineno">1602</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1603"></a><tt class="py-lineno">1603</tt>  <tt class="py-line"><tt class="py-docstring">    Note that Solexa FASTQ files have an upper limit of Solexa quality 62, which is</tt> </tt>
<a name="L1604"></a><tt class="py-lineno">1604</tt>  <tt class="py-line"><tt class="py-docstring">    encoded as ASCII 126, the tilde.  If your quality scores must be truncated to fit,</tt> </tt>
<a name="L1605"></a><tt class="py-lineno">1605</tt>  <tt class="py-line"><tt class="py-docstring">    a warning is issued.</tt> </tt>
<a name="L1606"></a><tt class="py-lineno">1606</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1607"></a><tt class="py-lineno">1607</tt>  <tt class="py-line"><tt class="py-docstring">    P.S. Don't forget to delete the temp file if you don't need it anymore:</tt> </tt>
<a name="L1608"></a><tt class="py-lineno">1608</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1609"></a><tt class="py-lineno">1609</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; import os</tt> </tt>
<a name="L1610"></a><tt class="py-lineno">1610</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; os.remove("Quality/temp.fastq")</tt> </tt>
<a name="L1611"></a><tt class="py-lineno">1611</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="FastqSolexaWriter.write_record"></a><div id="FastqSolexaWriter.write_record-def"><a name="L1612"></a><tt class="py-lineno">1612</tt> <a class="py-toggle" href="#" id="FastqSolexaWriter.write_record-toggle" onclick="return toggle('FastqSolexaWriter.write_record');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.QualityIO.FastqSolexaWriter-class.html#write_record">write_record</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FastqSolexaWriter.write_record-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="FastqSolexaWriter.write_record-expanded"><a name="L1613"></a><tt class="py-lineno">1613</tt>  <tt class="py-line">        <tt class="py-docstring">"""Write a single FASTQ record to the file."""</tt> </tt>
<a name="L1614"></a><tt class="py-lineno">1614</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_header_written</tt> </tt>
<a name="L1615"></a><tt class="py-lineno">1615</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_footer_written</tt> </tt>
<a name="L1616"></a><tt class="py-lineno">1616</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_record_written</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L1617"></a><tt class="py-lineno">1617</tt>  <tt class="py-line"> </tt>
<a name="L1618"></a><tt class="py-lineno">1618</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - Is an empty sequence allowed in FASTQ format?</tt> </tt>
<a name="L1619"></a><tt class="py-lineno">1619</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-272" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-272', 'seq', 'link-202');">seq</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1620"></a><tt class="py-lineno">1620</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"No sequence for record %s"</tt> <tt class="py-op">%</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-273" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-273', 'id', 'link-24');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1621"></a><tt class="py-lineno">1621</tt>  <tt class="py-line">        <tt class="py-name">seq_str</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-274" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-274', 'seq', 'link-202');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1622"></a><tt class="py-lineno">1622</tt>  <tt class="py-line">        <tt class="py-name">qualities_str</tt> <tt class="py-op">=</tt> <tt id="link-275" class="py-name" targets="Function Bio.SeqIO.QualityIO._get_solexa_quality_str()=Bio.SeqIO.QualityIO-module.html#_get_solexa_quality_str"><a title="Bio.SeqIO.QualityIO._get_solexa_quality_str" class="py-name" href="#" onclick="return doclink('link-275', '_get_solexa_quality_str', 'link-275');">_get_solexa_quality_str</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L1623"></a><tt class="py-lineno">1623</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">qualities_str</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">seq_str</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1624"></a><tt class="py-lineno">1624</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Record %s has sequence length %i but %i quality scores"</tt> </tt>
<a name="L1625"></a><tt class="py-lineno">1625</tt>  <tt class="py-line">                             <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-276" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-276', 'id', 'link-24');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">seq_str</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">qualities_str</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1626"></a><tt class="py-lineno">1626</tt>  <tt class="py-line"> </tt>
<a name="L1627"></a><tt class="py-lineno">1627</tt>  <tt class="py-line">        <tt class="py-comment"># FASTQ files can include a description, just like FASTA files</tt> </tt>
<a name="L1628"></a><tt class="py-lineno">1628</tt>  <tt class="py-line">        <tt class="py-comment"># (at least, this is what the NCBI Short Read Archive does)</tt> </tt>
<a name="L1629"></a><tt class="py-lineno">1629</tt>  <tt class="py-line">        <tt id="link-277" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-277', 'id', 'link-24');">id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-278" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentWriter.clean
Bio.SeqIO.Interfaces.SequenceWriter.clean" class="py-name" href="#" onclick="return doclink('link-278', 'clean', 'link-208');">clean</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-279" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-279', 'id', 'link-24');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1630"></a><tt class="py-lineno">1630</tt>  <tt class="py-line">        <tt id="link-280" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-280', 'description', 'link-128');">description</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-281" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentWriter.clean
Bio.SeqIO.Interfaces.SequenceWriter.clean" class="py-name" href="#" onclick="return doclink('link-281', 'clean', 'link-208');">clean</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-282" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-282', 'description', 'link-128');">description</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1631"></a><tt class="py-lineno">1631</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-283" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-283', 'description', 'link-128');">description</a></tt> <tt class="py-keyword">and</tt> <tt id="link-284" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-284', 'description', 'link-128');">description</a></tt><tt class="py-op">.</tt><tt id="link-285" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-285', 'split', 'link-118');">split</a></tt><tt class="py-op">(</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt id="link-286" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-286', 'id', 'link-24');">id</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1632"></a><tt class="py-lineno">1632</tt>  <tt class="py-line">            <tt class="py-comment"># The description includes the id at the start</tt> </tt>
<a name="L1633"></a><tt class="py-lineno">1633</tt>  <tt class="py-line">            <tt id="link-287" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-287', 'title', 'link-217');">title</a></tt> <tt class="py-op">=</tt> <tt id="link-288" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-288', 'description', 'link-128');">description</a></tt> </tt>
<a name="L1634"></a><tt class="py-lineno">1634</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-289" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-289', 'description', 'link-128');">description</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1635"></a><tt class="py-lineno">1635</tt>  <tt class="py-line">            <tt id="link-290" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-290', 'title', 'link-217');">title</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"%s %s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-291" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-291', 'id', 'link-24');">id</a></tt><tt class="py-op">,</tt> <tt id="link-292" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-292', 'description', 'link-128');">description</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1636"></a><tt class="py-lineno">1636</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1637"></a><tt class="py-lineno">1637</tt>  <tt class="py-line">            <tt id="link-293" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-293', 'title', 'link-217');">title</a></tt> <tt class="py-op">=</tt> <tt id="link-294" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-294', 'id', 'link-24');">id</a></tt> </tt>
<a name="L1638"></a><tt class="py-lineno">1638</tt>  <tt class="py-line"> </tt>
<a name="L1639"></a><tt class="py-lineno">1639</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-295" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-295', 'write', 'link-225');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"@%s\n%s\n+\n%s\n"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-296" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-296', 'title', 'link-217');">title</a></tt><tt class="py-op">,</tt> <tt class="py-name">seq_str</tt><tt class="py-op">,</tt> <tt class="py-name">qualities_str</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1640"></a><tt class="py-lineno">1640</tt>  <tt class="py-line"> </tt>
<a name="L1641"></a><tt class="py-lineno">1641</tt>  <tt class="py-line"> </tt>
<a name="FastqIlluminaWriter"></a><div id="FastqIlluminaWriter-def"><a name="L1642"></a><tt class="py-lineno">1642</tt> <a class="py-toggle" href="#" id="FastqIlluminaWriter-toggle" onclick="return toggle('FastqIlluminaWriter');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="Bio.SeqIO.QualityIO.FastqIlluminaWriter-class.html">FastqIlluminaWriter</a><tt class="py-op">(</tt><tt class="py-base-class">SequentialSequenceWriter</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FastqIlluminaWriter-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="FastqIlluminaWriter-expanded"><a name="L1643"></a><tt class="py-lineno">1643</tt>  <tt class="py-line">    <tt class="py-docstring">r"""Write Illumina 1.3+ FASTQ format files (with PHRED quality scores).</tt> </tt>
<a name="L1644"></a><tt class="py-lineno">1644</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1645"></a><tt class="py-lineno">1645</tt>  <tt class="py-line"><tt class="py-docstring">    This outputs FASTQ files like those from the Solexa/Illumina 1.3+ pipeline,</tt> </tt>
<a name="L1646"></a><tt class="py-lineno">1646</tt>  <tt class="py-line"><tt class="py-docstring">    using PHRED scores and an ASCII offset of 64. Note these files are NOT</tt> </tt>
<a name="L1647"></a><tt class="py-lineno">1647</tt>  <tt class="py-line"><tt class="py-docstring">    compatible with the standard Sanger style PHRED FASTQ files which use an</tt> </tt>
<a name="L1648"></a><tt class="py-lineno">1648</tt>  <tt class="py-line"><tt class="py-docstring">    ASCII offset of 32.</tt> </tt>
<a name="L1649"></a><tt class="py-lineno">1649</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1650"></a><tt class="py-lineno">1650</tt>  <tt class="py-line"><tt class="py-docstring">    Although you can use this class directly, you are strongly encouraged to</tt> </tt>
<a name="L1651"></a><tt class="py-lineno">1651</tt>  <tt class="py-line"><tt class="py-docstring">    use the Bio.SeqIO.write() function with format name "fastq-illumina"</tt> </tt>
<a name="L1652"></a><tt class="py-lineno">1652</tt>  <tt class="py-line"><tt class="py-docstring">    instead. This code is also called if you use the .format("fastq-illumina")</tt> </tt>
<a name="L1653"></a><tt class="py-lineno">1653</tt>  <tt class="py-line"><tt class="py-docstring">    method of a SeqRecord. For example,</tt> </tt>
<a name="L1654"></a><tt class="py-lineno">1654</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1655"></a><tt class="py-lineno">1655</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L1656"></a><tt class="py-lineno">1656</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; record = SeqIO.read("Quality/sanger_faked.fastq", "fastq-sanger")</tt> </tt>
<a name="L1657"></a><tt class="py-lineno">1657</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(record.format("fastq-illumina"))</tt> </tt>
<a name="L1658"></a><tt class="py-lineno">1658</tt>  <tt class="py-line"><tt class="py-docstring">    @Test PHRED qualities from 40 to 0 inclusive</tt> </tt>
<a name="L1659"></a><tt class="py-lineno">1659</tt>  <tt class="py-line"><tt class="py-docstring">    ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTN</tt> </tt>
<a name="L1660"></a><tt class="py-lineno">1660</tt>  <tt class="py-line"><tt class="py-docstring">    +</tt> </tt>
<a name="L1661"></a><tt class="py-lineno">1661</tt>  <tt class="py-line"><tt class="py-docstring">    hgfedcba`_^]\[ZYXWVUTSRQPONMLKJIHGFEDCBA@</tt> </tt>
<a name="L1662"></a><tt class="py-lineno">1662</tt>  <tt class="py-line"><tt class="py-docstring">    &lt;BLANKLINE&gt;</tt> </tt>
<a name="L1663"></a><tt class="py-lineno">1663</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1664"></a><tt class="py-lineno">1664</tt>  <tt class="py-line"><tt class="py-docstring">    Note that Illumina FASTQ files have an upper limit of PHRED quality 62, which is</tt> </tt>
<a name="L1665"></a><tt class="py-lineno">1665</tt>  <tt class="py-line"><tt class="py-docstring">    encoded as ASCII 126, the tilde. If your quality scores are truncated to fit, a</tt> </tt>
<a name="L1666"></a><tt class="py-lineno">1666</tt>  <tt class="py-line"><tt class="py-docstring">    warning is issued.</tt> </tt>
<a name="L1667"></a><tt class="py-lineno">1667</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="FastqIlluminaWriter.write_record"></a><div id="FastqIlluminaWriter.write_record-def"><a name="L1668"></a><tt class="py-lineno">1668</tt> <a class="py-toggle" href="#" id="FastqIlluminaWriter.write_record-toggle" onclick="return toggle('FastqIlluminaWriter.write_record');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.QualityIO.FastqIlluminaWriter-class.html#write_record">write_record</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="FastqIlluminaWriter.write_record-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="FastqIlluminaWriter.write_record-expanded"><a name="L1669"></a><tt class="py-lineno">1669</tt>  <tt class="py-line">        <tt class="py-docstring">"""Write a single FASTQ record to the file."""</tt> </tt>
<a name="L1670"></a><tt class="py-lineno">1670</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_header_written</tt> </tt>
<a name="L1671"></a><tt class="py-lineno">1671</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-keyword">not</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_footer_written</tt> </tt>
<a name="L1672"></a><tt class="py-lineno">1672</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_record_written</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L1673"></a><tt class="py-lineno">1673</tt>  <tt class="py-line"> </tt>
<a name="L1674"></a><tt class="py-lineno">1674</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - Is an empty sequence allowed in FASTQ format?</tt> </tt>
<a name="L1675"></a><tt class="py-lineno">1675</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-297" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-297', 'seq', 'link-202');">seq</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1676"></a><tt class="py-lineno">1676</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"No sequence for record %s"</tt> <tt class="py-op">%</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-298" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-298', 'id', 'link-24');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1677"></a><tt class="py-lineno">1677</tt>  <tt class="py-line">        <tt class="py-name">seq_str</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-299" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-299', 'seq', 'link-202');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1678"></a><tt class="py-lineno">1678</tt>  <tt class="py-line">        <tt class="py-name">qualities_str</tt> <tt class="py-op">=</tt> <tt id="link-300" class="py-name" targets="Function Bio.SeqIO.QualityIO._get_illumina_quality_str()=Bio.SeqIO.QualityIO-module.html#_get_illumina_quality_str"><a title="Bio.SeqIO.QualityIO._get_illumina_quality_str" class="py-name" href="#" onclick="return doclink('link-300', '_get_illumina_quality_str', 'link-300');">_get_illumina_quality_str</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L1679"></a><tt class="py-lineno">1679</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">qualities_str</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">seq_str</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1680"></a><tt class="py-lineno">1680</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Record %s has sequence length %i but %i quality scores"</tt> </tt>
<a name="L1681"></a><tt class="py-lineno">1681</tt>  <tt class="py-line">                             <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-301" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-301', 'id', 'link-24');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">seq_str</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">qualities_str</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1682"></a><tt class="py-lineno">1682</tt>  <tt class="py-line"> </tt>
<a name="L1683"></a><tt class="py-lineno">1683</tt>  <tt class="py-line">        <tt class="py-comment"># FASTQ files can include a description, just like FASTA files</tt> </tt>
<a name="L1684"></a><tt class="py-lineno">1684</tt>  <tt class="py-line">        <tt class="py-comment"># (at least, this is what the NCBI Short Read Archive does)</tt> </tt>
<a name="L1685"></a><tt class="py-lineno">1685</tt>  <tt class="py-line">        <tt id="link-302" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-302', 'id', 'link-24');">id</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-303" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentWriter.clean
Bio.SeqIO.Interfaces.SequenceWriter.clean" class="py-name" href="#" onclick="return doclink('link-303', 'clean', 'link-208');">clean</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-304" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-304', 'id', 'link-24');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1686"></a><tt class="py-lineno">1686</tt>  <tt class="py-line">        <tt id="link-305" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-305', 'description', 'link-128');">description</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-306" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentWriter.clean
Bio.SeqIO.Interfaces.SequenceWriter.clean" class="py-name" href="#" onclick="return doclink('link-306', 'clean', 'link-208');">clean</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-307" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-307', 'description', 'link-128');">description</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1687"></a><tt class="py-lineno">1687</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-308" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-308', 'description', 'link-128');">description</a></tt> <tt class="py-keyword">and</tt> <tt id="link-309" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-309', 'description', 'link-128');">description</a></tt><tt class="py-op">.</tt><tt id="link-310" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-310', 'split', 'link-118');">split</a></tt><tt class="py-op">(</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt id="link-311" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-311', 'id', 'link-24');">id</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1688"></a><tt class="py-lineno">1688</tt>  <tt class="py-line">            <tt class="py-comment"># The description includes the id at the start</tt> </tt>
<a name="L1689"></a><tt class="py-lineno">1689</tt>  <tt class="py-line">            <tt id="link-312" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-312', 'title', 'link-217');">title</a></tt> <tt class="py-op">=</tt> <tt id="link-313" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-313', 'description', 'link-128');">description</a></tt> </tt>
<a name="L1690"></a><tt class="py-lineno">1690</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-314" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-314', 'description', 'link-128');">description</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1691"></a><tt class="py-lineno">1691</tt>  <tt class="py-line">            <tt id="link-315" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-315', 'title', 'link-217');">title</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"%s %s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-316" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-316', 'id', 'link-24');">id</a></tt><tt class="py-op">,</tt> <tt id="link-317" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-317', 'description', 'link-128');">description</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1692"></a><tt class="py-lineno">1692</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L1693"></a><tt class="py-lineno">1693</tt>  <tt class="py-line">            <tt id="link-318" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-318', 'title', 'link-217');">title</a></tt> <tt class="py-op">=</tt> <tt id="link-319" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-319', 'id', 'link-24');">id</a></tt> </tt>
<a name="L1694"></a><tt class="py-lineno">1694</tt>  <tt class="py-line"> </tt>
<a name="L1695"></a><tt class="py-lineno">1695</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">handle</tt><tt class="py-op">.</tt><tt id="link-320" class="py-name"><a title="Bio.AlignIO.write
Bio.Graphics.GenomeDiagram._Diagram.Diagram.write
Bio.NeuralNetwork.Gene.Pattern.PatternIO.write
Bio.Phylo.CDAOIO.Writer.write
Bio.Phylo.CDAOIO.write
Bio.Phylo.NeXMLIO.Writer.write
Bio.Phylo.NeXMLIO.write
Bio.Phylo.NewickIO.Writer.write
Bio.Phylo.NewickIO.write
Bio.Phylo.NexusIO.write
Bio.Phylo.PhyloXMLIO.Writer.write
Bio.Phylo.PhyloXMLIO.write
Bio.Phylo._io.write
Bio.SearchIO.write
Bio.SeqIO.write
Bio.bgzf.BgzfWriter.write
Bio.motifs.jaspar.write
Bio.motifs.transfac.write
Bio.motifs.write" class="py-name" href="#" onclick="return doclink('link-320', 'write', 'link-225');">write</a></tt><tt class="py-op">(</tt><tt class="py-string">"@%s\n%s\n+\n%s\n"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-321" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-321', 'title', 'link-217');">title</a></tt><tt class="py-op">,</tt> <tt class="py-name">seq_str</tt><tt class="py-op">,</tt> <tt class="py-name">qualities_str</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1696"></a><tt class="py-lineno">1696</tt>  <tt class="py-line"> </tt>
<a name="L1697"></a><tt class="py-lineno">1697</tt>  <tt class="py-line"> </tt>
<a name="PairedFastaQualIterator"></a><div id="PairedFastaQualIterator-def"><a name="L1698"></a><tt class="py-lineno">1698</tt> <a class="py-toggle" href="#" id="PairedFastaQualIterator-toggle" onclick="return toggle('PairedFastaQualIterator');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="Bio.SeqIO.QualityIO-module.html#PairedFastaQualIterator">PairedFastaQualIterator</a><tt class="py-op">(</tt><tt class="py-param">fasta_handle</tt><tt class="py-op">,</tt> <tt class="py-param">qual_handle</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">=</tt><tt id="link-322" class="py-name"><a title="Bio.Alphabet.single_letter_alphabet" class="py-name" href="#" onclick="return doclink('link-322', 'single_letter_alphabet', 'link-2');">single_letter_alphabet</a></tt><tt class="py-op">,</tt> <tt class="py-param">title2ids</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="PairedFastaQualIterator-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="PairedFastaQualIterator-expanded"><a name="L1699"></a><tt class="py-lineno">1699</tt>  <tt class="py-line">    <tt class="py-docstring">"""Iterate over matched FASTA and QUAL files as SeqRecord objects.</tt> </tt>
<a name="L1700"></a><tt class="py-lineno">1700</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1701"></a><tt class="py-lineno">1701</tt>  <tt class="py-line"><tt class="py-docstring">    For example, consider this short QUAL file with PHRED quality scores::</tt> </tt>
<a name="L1702"></a><tt class="py-lineno">1702</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1703"></a><tt class="py-lineno">1703</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;EAS54_6_R1_2_1_413_324</tt> </tt>
<a name="L1704"></a><tt class="py-lineno">1704</tt>  <tt class="py-line"><tt class="py-docstring">        26 26 18 26 26 26 26 26 26 26 26 26 26 26 26 22 26 26 26 26</tt> </tt>
<a name="L1705"></a><tt class="py-lineno">1705</tt>  <tt class="py-line"><tt class="py-docstring">        26 26 26 23 23</tt> </tt>
<a name="L1706"></a><tt class="py-lineno">1706</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;EAS54_6_R1_2_1_540_792</tt> </tt>
<a name="L1707"></a><tt class="py-lineno">1707</tt>  <tt class="py-line"><tt class="py-docstring">        26 26 26 26 26 26 26 26 26 26 26 22 26 26 26 26 26 12 26 26</tt> </tt>
<a name="L1708"></a><tt class="py-lineno">1708</tt>  <tt class="py-line"><tt class="py-docstring">        26 18 26 23 18</tt> </tt>
<a name="L1709"></a><tt class="py-lineno">1709</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;EAS54_6_R1_2_1_443_348</tt> </tt>
<a name="L1710"></a><tt class="py-lineno">1710</tt>  <tt class="py-line"><tt class="py-docstring">        26 26 26 26 26 26 26 26 26 26 26 24 26 22 26 26 13 22 26 18</tt> </tt>
<a name="L1711"></a><tt class="py-lineno">1711</tt>  <tt class="py-line"><tt class="py-docstring">        24 18 18 18 18</tt> </tt>
<a name="L1712"></a><tt class="py-lineno">1712</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1713"></a><tt class="py-lineno">1713</tt>  <tt class="py-line"><tt class="py-docstring">    And a matching FASTA file::</tt> </tt>
<a name="L1714"></a><tt class="py-lineno">1714</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1715"></a><tt class="py-lineno">1715</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;EAS54_6_R1_2_1_413_324</tt> </tt>
<a name="L1716"></a><tt class="py-lineno">1716</tt>  <tt class="py-line"><tt class="py-docstring">        CCCTTCTTGTCTTCAGCGTTTCTCC</tt> </tt>
<a name="L1717"></a><tt class="py-lineno">1717</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;EAS54_6_R1_2_1_540_792</tt> </tt>
<a name="L1718"></a><tt class="py-lineno">1718</tt>  <tt class="py-line"><tt class="py-docstring">        TTGGCAGGCCAAGGCCGATGGATCA</tt> </tt>
<a name="L1719"></a><tt class="py-lineno">1719</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;EAS54_6_R1_2_1_443_348</tt> </tt>
<a name="L1720"></a><tt class="py-lineno">1720</tt>  <tt class="py-line"><tt class="py-docstring">        GTTGCTTCTGGCGTGGGTGGGGGGG</tt> </tt>
<a name="L1721"></a><tt class="py-lineno">1721</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1722"></a><tt class="py-lineno">1722</tt>  <tt class="py-line"><tt class="py-docstring">    You can parse these separately using Bio.SeqIO with the "qual" and</tt> </tt>
<a name="L1723"></a><tt class="py-lineno">1723</tt>  <tt class="py-line"><tt class="py-docstring">    "fasta" formats, but then you'll get a group of SeqRecord objects with</tt> </tt>
<a name="L1724"></a><tt class="py-lineno">1724</tt>  <tt class="py-line"><tt class="py-docstring">    no sequence, and a matching group with the sequence but not the</tt> </tt>
<a name="L1725"></a><tt class="py-lineno">1725</tt>  <tt class="py-line"><tt class="py-docstring">    qualities.  Because it only deals with one input file handle, Bio.SeqIO</tt> </tt>
<a name="L1726"></a><tt class="py-lineno">1726</tt>  <tt class="py-line"><tt class="py-docstring">    can't be used to read the two files together - but this function can!</tt> </tt>
<a name="L1727"></a><tt class="py-lineno">1727</tt>  <tt class="py-line"><tt class="py-docstring">    For example,</tt> </tt>
<a name="L1728"></a><tt class="py-lineno">1728</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1729"></a><tt class="py-lineno">1729</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; with open("Quality/example.fasta", "rU") as f:</tt> </tt>
<a name="L1730"></a><tt class="py-lineno">1730</tt>  <tt class="py-line"><tt class="py-docstring">    ...     with open("Quality/example.qual", "rU") as q:</tt> </tt>
<a name="L1731"></a><tt class="py-lineno">1731</tt>  <tt class="py-line"><tt class="py-docstring">    ...         for record in PairedFastaQualIterator(f, q):</tt> </tt>
<a name="L1732"></a><tt class="py-lineno">1732</tt>  <tt class="py-line"><tt class="py-docstring">    ...             print("%s %s" % (record.id, record.seq))</tt> </tt>
<a name="L1733"></a><tt class="py-lineno">1733</tt>  <tt class="py-line"><tt class="py-docstring">    ...</tt> </tt>
<a name="L1734"></a><tt class="py-lineno">1734</tt>  <tt class="py-line"><tt class="py-docstring">    EAS54_6_R1_2_1_413_324 CCCTTCTTGTCTTCAGCGTTTCTCC</tt> </tt>
<a name="L1735"></a><tt class="py-lineno">1735</tt>  <tt class="py-line"><tt class="py-docstring">    EAS54_6_R1_2_1_540_792 TTGGCAGGCCAAGGCCGATGGATCA</tt> </tt>
<a name="L1736"></a><tt class="py-lineno">1736</tt>  <tt class="py-line"><tt class="py-docstring">    EAS54_6_R1_2_1_443_348 GTTGCTTCTGGCGTGGGTGGGGGGG</tt> </tt>
<a name="L1737"></a><tt class="py-lineno">1737</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1738"></a><tt class="py-lineno">1738</tt>  <tt class="py-line"><tt class="py-docstring">    As with the FASTQ or QUAL parsers, if you want to look at the qualities,</tt> </tt>
<a name="L1739"></a><tt class="py-lineno">1739</tt>  <tt class="py-line"><tt class="py-docstring">    they are in each record's per-letter-annotation dictionary as a simple</tt> </tt>
<a name="L1740"></a><tt class="py-lineno">1740</tt>  <tt class="py-line"><tt class="py-docstring">    list of integers:</tt> </tt>
<a name="L1741"></a><tt class="py-lineno">1741</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1742"></a><tt class="py-lineno">1742</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; print(record.letter_annotations["phred_quality"])</tt> </tt>
<a name="L1743"></a><tt class="py-lineno">1743</tt>  <tt class="py-line"><tt class="py-docstring">    [26, 26, 26, 26, 26, 26, 26, 26, 26, 26, 26, 24, 26, 22, 26, 26, 13, 22, 26, 18, 24, 18, 18, 18, 18]</tt> </tt>
<a name="L1744"></a><tt class="py-lineno">1744</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1745"></a><tt class="py-lineno">1745</tt>  <tt class="py-line"><tt class="py-docstring">    If you have access to data as a FASTQ format file, using that directly</tt> </tt>
<a name="L1746"></a><tt class="py-lineno">1746</tt>  <tt class="py-line"><tt class="py-docstring">    would be simpler and more straight forward.  Note that you can easily use</tt> </tt>
<a name="L1747"></a><tt class="py-lineno">1747</tt>  <tt class="py-line"><tt class="py-docstring">    this function to convert paired FASTA and QUAL files into FASTQ files:</tt> </tt>
<a name="L1748"></a><tt class="py-lineno">1748</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1749"></a><tt class="py-lineno">1749</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; from Bio import SeqIO</tt> </tt>
<a name="L1750"></a><tt class="py-lineno">1750</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; with open("Quality/example.fasta", "rU") as f:</tt> </tt>
<a name="L1751"></a><tt class="py-lineno">1751</tt>  <tt class="py-line"><tt class="py-docstring">    ...     with open("Quality/example.qual", "rU") as q:</tt> </tt>
<a name="L1752"></a><tt class="py-lineno">1752</tt>  <tt class="py-line"><tt class="py-docstring">    ...         SeqIO.write(PairedFastaQualIterator(f, q), "Quality/temp.fastq", "fastq")</tt> </tt>
<a name="L1753"></a><tt class="py-lineno">1753</tt>  <tt class="py-line"><tt class="py-docstring">    ...</tt> </tt>
<a name="L1754"></a><tt class="py-lineno">1754</tt>  <tt class="py-line"><tt class="py-docstring">    3</tt> </tt>
<a name="L1755"></a><tt class="py-lineno">1755</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1756"></a><tt class="py-lineno">1756</tt>  <tt class="py-line"><tt class="py-docstring">    And don't forget to clean up the temp file if you don't need it anymore:</tt> </tt>
<a name="L1757"></a><tt class="py-lineno">1757</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1758"></a><tt class="py-lineno">1758</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; import os</tt> </tt>
<a name="L1759"></a><tt class="py-lineno">1759</tt>  <tt class="py-line"><tt class="py-docstring">    &gt;&gt;&gt; os.remove("Quality/temp.fastq")</tt> </tt>
<a name="L1760"></a><tt class="py-lineno">1760</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1761"></a><tt class="py-lineno">1761</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-323" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-323', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-324" class="py-name"><a title="Bio.SeqIO" class="py-name" href="#" onclick="return doclink('link-324', 'SeqIO', 'link-11');">SeqIO</a></tt><tt class="py-op">.</tt><tt id="link-325" class="py-name" targets="Module Bio.AlignIO.FastaIO=Bio.AlignIO.FastaIO-module.html,Module Bio.SearchIO.FastaIO=Bio.SearchIO.FastaIO-module.html,Module Bio.SeqIO.FastaIO=Bio.SeqIO.FastaIO-module.html"><a title="Bio.AlignIO.FastaIO
Bio.SearchIO.FastaIO
Bio.SeqIO.FastaIO" class="py-name" href="#" onclick="return doclink('link-325', 'FastaIO', 'link-325');">FastaIO</a></tt> <tt class="py-keyword">import</tt> <tt id="link-326" class="py-name" targets="Function Bio.SeqIO.FastaIO.FastaIterator()=Bio.SeqIO.FastaIO-module.html#FastaIterator"><a title="Bio.SeqIO.FastaIO.FastaIterator" class="py-name" href="#" onclick="return doclink('link-326', 'FastaIterator', 'link-326');">FastaIterator</a></tt> </tt>
<a name="L1762"></a><tt class="py-lineno">1762</tt>  <tt class="py-line">    <tt class="py-name">fasta_iter</tt> <tt class="py-op">=</tt> <tt id="link-327" class="py-name"><a title="Bio.SeqIO.FastaIO.FastaIterator" class="py-name" href="#" onclick="return doclink('link-327', 'FastaIterator', 'link-326');">FastaIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">fasta_handle</tt><tt class="py-op">,</tt> <tt id="link-328" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-328', 'alphabet', 'link-123');">alphabet</a></tt><tt class="py-op">=</tt><tt id="link-329" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-329', 'alphabet', 'link-123');">alphabet</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1763"></a><tt class="py-lineno">1763</tt>  <tt class="py-line">                               <tt class="py-name">title2ids</tt><tt class="py-op">=</tt><tt class="py-name">title2ids</tt><tt class="py-op">)</tt> </tt>
<a name="L1764"></a><tt class="py-lineno">1764</tt>  <tt class="py-line">    <tt class="py-name">qual_iter</tt> <tt class="py-op">=</tt> <tt id="link-330" class="py-name" targets="Function Bio.SeqIO.QualityIO.QualPhredIterator()=Bio.SeqIO.QualityIO-module.html#QualPhredIterator"><a title="Bio.SeqIO.QualityIO.QualPhredIterator" class="py-name" href="#" onclick="return doclink('link-330', 'QualPhredIterator', 'link-330');">QualPhredIterator</a></tt><tt class="py-op">(</tt><tt class="py-name">qual_handle</tt><tt class="py-op">,</tt> <tt id="link-331" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-331', 'alphabet', 'link-123');">alphabet</a></tt><tt class="py-op">=</tt><tt id="link-332" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-332', 'alphabet', 'link-123');">alphabet</a></tt><tt class="py-op">,</tt> </tt>
<a name="L1765"></a><tt class="py-lineno">1765</tt>  <tt class="py-line">                                  <tt class="py-name">title2ids</tt><tt class="py-op">=</tt><tt class="py-name">title2ids</tt><tt class="py-op">)</tt> </tt>
<a name="L1766"></a><tt class="py-lineno">1766</tt>  <tt class="py-line"> </tt>
<a name="L1767"></a><tt class="py-lineno">1767</tt>  <tt class="py-line">    <tt class="py-comment"># Using (Python 3 style) zip wouldn't load everything into memory,</tt> </tt>
<a name="L1768"></a><tt class="py-lineno">1768</tt>  <tt class="py-line">    <tt class="py-comment"># but also would not catch any extra records found in only one file.</tt> </tt>
<a name="L1769"></a><tt class="py-lineno">1769</tt>  <tt class="py-line">    <tt class="py-keyword">while</tt> <tt class="py-name">True</tt><tt class="py-op">:</tt> </tt>
<a name="L1770"></a><tt class="py-lineno">1770</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1771"></a><tt class="py-lineno">1771</tt>  <tt class="py-line">            <tt class="py-name">f_rec</tt> <tt class="py-op">=</tt> <tt id="link-333" class="py-name" targets="Method Bio.AlignIO.Interfaces.AlignmentIterator.next()=Bio.AlignIO.Interfaces.AlignmentIterator-class.html#next,Method Bio.Blast.NCBIStandalone.Iterator.next()=Bio.Blast.NCBIStandalone.Iterator-class.html#next,Method Bio.Blast.ParseBlastTable.BlastTableReader.next()=Bio.Blast.ParseBlastTable.BlastTableReader-class.html#next,Method Bio.DocSQL.IterationCursor.next()=Bio.DocSQL.IterationCursor-class.html#next,Method Bio.File.UndoHandle.next()=Bio.File.UndoHandle-class.html#next,Method Bio.GenBank.Iterator.next()=Bio.GenBank.Iterator-class.html#next,Method Bio.Nexus.Nexus.CharBuffer.next()=Bio.Nexus.Nexus.CharBuffer-class.html#next,Method Bio.PopGen.GenePop.Controller._FileIterator.next()=Bio.PopGen.GenePop.Controller._FileIterator-class.html#next,Method Bio.SeqIO.Interfaces.SequenceIterator.next()=Bio.SeqIO.Interfaces.SequenceIterator-class.html#next,Method Bio.bgzf.BgzfReader.next()=Bio.bgzf.BgzfReader-class.html#next"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-333', 'next', 'link-333');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">fasta_iter</tt><tt class="py-op">)</tt> </tt>
<a name="L1772"></a><tt class="py-lineno">1772</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L1773"></a><tt class="py-lineno">1773</tt>  <tt class="py-line">            <tt class="py-name">f_rec</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L1774"></a><tt class="py-lineno">1774</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1775"></a><tt class="py-lineno">1775</tt>  <tt class="py-line">            <tt class="py-name">q_rec</tt> <tt class="py-op">=</tt> <tt id="link-334" class="py-name"><a title="Bio.AlignIO.Interfaces.AlignmentIterator.next
Bio.Blast.NCBIStandalone.Iterator.next
Bio.Blast.ParseBlastTable.BlastTableReader.next
Bio.DocSQL.IterationCursor.next
Bio.File.UndoHandle.next
Bio.GenBank.Iterator.next
Bio.Nexus.Nexus.CharBuffer.next
Bio.PopGen.GenePop.Controller._FileIterator.next
Bio.SeqIO.Interfaces.SequenceIterator.next
Bio.bgzf.BgzfReader.next" class="py-name" href="#" onclick="return doclink('link-334', 'next', 'link-333');">next</a></tt><tt class="py-op">(</tt><tt class="py-name">qual_iter</tt><tt class="py-op">)</tt> </tt>
<a name="L1776"></a><tt class="py-lineno">1776</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">StopIteration</tt><tt class="py-op">:</tt> </tt>
<a name="L1777"></a><tt class="py-lineno">1777</tt>  <tt class="py-line">            <tt class="py-name">q_rec</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L1778"></a><tt class="py-lineno">1778</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">f_rec</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt> <tt class="py-keyword">and</tt> <tt class="py-name">q_rec</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1779"></a><tt class="py-lineno">1779</tt>  <tt class="py-line">            <tt class="py-comment"># End of both files</tt> </tt>
<a name="L1780"></a><tt class="py-lineno">1780</tt>  <tt class="py-line">            <tt class="py-keyword">break</tt> </tt>
<a name="L1781"></a><tt class="py-lineno">1781</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">f_rec</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1782"></a><tt class="py-lineno">1782</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"FASTA file has more entries than the QUAL file."</tt><tt class="py-op">)</tt> </tt>
<a name="L1783"></a><tt class="py-lineno">1783</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">q_rec</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L1784"></a><tt class="py-lineno">1784</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"QUAL file has more entries than the FASTA file."</tt><tt class="py-op">)</tt> </tt>
<a name="L1785"></a><tt class="py-lineno">1785</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">f_rec</tt><tt class="py-op">.</tt><tt id="link-335" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-335', 'id', 'link-24');">id</a></tt> <tt class="py-op">!=</tt> <tt class="py-name">q_rec</tt><tt class="py-op">.</tt><tt id="link-336" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-336', 'id', 'link-24');">id</a></tt><tt class="py-op">:</tt> </tt>
<a name="L1786"></a><tt class="py-lineno">1786</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"FASTA and QUAL entries do not match (%s vs %s)."</tt> </tt>
<a name="L1787"></a><tt class="py-lineno">1787</tt>  <tt class="py-line">                             <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">f_rec</tt><tt class="py-op">.</tt><tt id="link-337" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-337', 'id', 'link-24');">id</a></tt><tt class="py-op">,</tt> <tt class="py-name">q_rec</tt><tt class="py-op">.</tt><tt id="link-338" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-338', 'id', 'link-24');">id</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1788"></a><tt class="py-lineno">1788</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">f_rec</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">q_rec</tt><tt class="py-op">.</tt><tt id="link-339" class="py-name"><a title="Bio.SeqRecord.SeqRecord.letter_annotations" class="py-name" href="#" onclick="return doclink('link-339', 'letter_annotations', 'link-21');">letter_annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"phred_quality"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1789"></a><tt class="py-lineno">1789</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Sequence length and number of quality scores disagree for %s"</tt> </tt>
<a name="L1790"></a><tt class="py-lineno">1790</tt>  <tt class="py-line">                             <tt class="py-op">%</tt> <tt class="py-name">f_rec</tt><tt class="py-op">.</tt><tt id="link-340" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-340', 'id', 'link-24');">id</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1791"></a><tt class="py-lineno">1791</tt>  <tt class="py-line">        <tt class="py-comment"># Merge the data....</tt> </tt>
<a name="L1792"></a><tt class="py-lineno">1792</tt>  <tt class="py-line">        <tt class="py-name">f_rec</tt><tt class="py-op">.</tt><tt id="link-341" class="py-name"><a title="Bio.SeqRecord.SeqRecord.letter_annotations" class="py-name" href="#" onclick="return doclink('link-341', 'letter_annotations', 'link-21');">letter_annotations</a></tt><tt class="py-op">[</tt> </tt>
<a name="L1793"></a><tt class="py-lineno">1793</tt>  <tt class="py-line">            <tt class="py-string">"phred_quality"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">q_rec</tt><tt class="py-op">.</tt><tt id="link-342" class="py-name"><a title="Bio.SeqRecord.SeqRecord.letter_annotations" class="py-name" href="#" onclick="return doclink('link-342', 'letter_annotations', 'link-21');">letter_annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"phred_quality"</tt><tt class="py-op">]</tt> </tt>
<a name="L1794"></a><tt class="py-lineno">1794</tt>  <tt class="py-line">        <tt class="py-keyword">yield</tt> <tt class="py-name">f_rec</tt> </tt>
</div><a name="L1795"></a><tt class="py-lineno">1795</tt>  <tt class="py-line">    <tt class="py-comment"># Done</tt> </tt>
<a name="L1796"></a><tt class="py-lineno">1796</tt>  <tt class="py-line"> </tt>
<a name="L1797"></a><tt class="py-lineno">1797</tt>  <tt class="py-line"> </tt>
<a name="L1798"></a><tt class="py-lineno">1798</tt>  <tt class="py-line"><tt class="py-keyword">if</tt> <tt class="py-name">__name__</tt> <tt class="py-op">==</tt> <tt class="py-string">"__main__"</tt><tt class="py-op">:</tt> </tt>
<a name="L1799"></a><tt class="py-lineno">1799</tt>  <tt class="py-line">    <tt class="py-keyword">from</tt> <tt id="link-343" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-343', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-344" class="py-name" targets="Module Bio.Phylo._utils=Bio.Phylo._utils-module.html,Module Bio.SearchIO._utils=Bio.SearchIO._utils-module.html,Module Bio._utils=Bio._utils-module.html"><a title="Bio.Phylo._utils
Bio.SearchIO._utils
Bio._utils" class="py-name" href="#" onclick="return doclink('link-344', '_utils', 'link-344');">_utils</a></tt> <tt class="py-keyword">import</tt> <tt id="link-345" class="py-name" targets="Function Bio._utils.run_doctest()=Bio._utils-module.html#run_doctest"><a title="Bio._utils.run_doctest" class="py-name" href="#" onclick="return doclink('link-345', 'run_doctest', 'link-345');">run_doctest</a></tt> </tt>
<a name="L1800"></a><tt class="py-lineno">1800</tt>  <tt class="py-line">    <tt id="link-346" class="py-name"><a title="Bio._utils.run_doctest" class="py-name" href="#" onclick="return doclink('link-346', 'run_doctest', 'link-345');">run_doctest</a></tt><tt class="py-op">(</tt><tt class="py-name">verbose</tt><tt class="py-op">=</tt><tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L1801"></a><tt class="py-lineno">1801</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Oct 21 18:04:29 2015
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
