<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>BioSQL.BioSeq</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="BioSQL-module.html">Package&nbsp;BioSQL</a> ::
        Module&nbsp;BioSeq
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="BioSQL.BioSeq-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="BioSQL.BioSeq-module.html">Module BioSQL.BioSeq</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2002 by Andrew Dalke.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># Revisions 2007-2009 copyright by Peter Cock.  All rights reserved.</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># Revisions 2008-2009 copyright by Cymon J. Cox.  All rights reserved.</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-comment"># Note that BioSQL (including the database schema and scripts) is</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-comment"># available and licensed separately.  Please consult www.biosql.org</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-docstring">"""Implementations of Biopython-like Seq objects on top of BioSQL.</tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-docstring">This allows retrival of items stored in a BioSQL database using</tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-docstring">a biopython-like SeqRecord and Seq interface.</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-docstring">Note: Currently we do not support recording per-letter-annotations</tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-docstring">(like quality scores) in BioSQL.</tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-1', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">unicode</tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-2" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-2', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-3" class="py-name" targets="Package Bio.Alphabet=Bio.Alphabet-module.html,Class Bio.Alphabet.Alphabet=Bio.Alphabet.Alphabet-class.html"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-3', 'Alphabet', 'link-3');">Alphabet</a></tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-4" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-4', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-5" class="py-name" targets="Module Bio.Seq=Bio.Seq-module.html,Class Bio.Seq.Seq=Bio.Seq.Seq-class.html"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-5', 'Seq', 'link-5');">Seq</a></tt> <tt class="py-keyword">import</tt> <tt id="link-6" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-6', 'Seq', 'link-5');">Seq</a></tt><tt class="py-op">,</tt> <tt id="link-7" class="py-name" targets="Class Bio.Seq.UnknownSeq=Bio.Seq.UnknownSeq-class.html"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-7', 'UnknownSeq', 'link-7');">UnknownSeq</a></tt> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-8" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-8', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-9" class="py-name" targets="Module Bio.SeqRecord=Bio.SeqRecord-module.html,Class Bio.SeqRecord.SeqRecord=Bio.SeqRecord.SeqRecord-class.html"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-9', 'SeqRecord', 'link-9');">SeqRecord</a></tt> <tt class="py-keyword">import</tt> <tt id="link-10" class="py-name"><a title="Bio.SeqRecord
Bio.SeqRecord.SeqRecord" class="py-name" href="#" onclick="return doclink('link-10', 'SeqRecord', 'link-9');">SeqRecord</a></tt><tt class="py-op">,</tt> <tt id="link-11" class="py-name" targets="Class Bio.SeqRecord._RestrictedDict=Bio.SeqRecord._RestrictedDict-class.html"><a title="Bio.SeqRecord._RestrictedDict" class="py-name" href="#" onclick="return doclink('link-11', '_RestrictedDict', 'link-11');">_RestrictedDict</a></tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-12" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-12', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-13" class="py-name" targets="Module Bio.SeqFeature=Bio.SeqFeature-module.html,Class Bio.SeqFeature.SeqFeature=Bio.SeqFeature.SeqFeature-class.html"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-13', 'SeqFeature', 'link-13');">SeqFeature</a></tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"> </tt>
<a name="DBSeq"></a><div id="DBSeq-def"><a name="L27"></a><tt class="py-lineno"> 27</tt> <a class="py-toggle" href="#" id="DBSeq-toggle" onclick="return toggle('DBSeq');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="BioSQL.BioSeq.DBSeq-class.html">DBSeq</a><tt class="py-op">(</tt><tt class="py-base-class">Seq</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBSeq-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="DBSeq-expanded"><a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line">    <tt class="py-docstring">"""BioSQL equivalent of the Biopython Seq object."""</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"> </tt>
<a name="DBSeq.__init__"></a><div id="DBSeq.__init__-def"><a name="L30"></a><tt class="py-lineno"> 30</tt> <a class="py-toggle" href="#" id="DBSeq.__init__-toggle" onclick="return toggle('DBSeq.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq.DBSeq-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">primary_id</tt><tt class="py-op">,</tt> <tt class="py-param">adaptor</tt><tt class="py-op">,</tt> <tt class="py-param">alphabet</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">,</tt> <tt class="py-param">length</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBSeq.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBSeq.__init__-expanded"><a name="L31"></a><tt class="py-lineno"> 31</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a new DBSeq object referring to a BioSQL entry.</tt> </tt>
<a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-docstring">        You wouldn't normally create a DBSeq object yourself, this is done</tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-docstring">        for you when retreiving a DBSeqRecord object from the database.</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">primary_id</tt> <tt class="py-op">=</tt> <tt class="py-name">primary_id</tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt> <tt class="py-op">=</tt> <tt class="py-name">adaptor</tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-14" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.alphabet=Bio.SearchIO._model.hsp.HSP-class.html#alphabet,Variable Bio.SearchIO._model.hsp.HSPFragment.alphabet=Bio.SearchIO._model.hsp.HSPFragment-class.html#alphabet"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-14', 'alphabet', 'link-14');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-15" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-15', 'alphabet', 'link-14');">alphabet</a></tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-16" class="py-name" targets="Method Bio.Motif.Parsers.MEME.MEMEInstance._length()=Bio.Motif.Parsers.MEME.MEMEInstance-class.html#_length"><a title="Bio.Motif.Parsers.MEME.MEMEInstance._length" class="py-name" href="#" onclick="return doclink('link-16', '_length', 'link-16');">_length</a></tt> <tt class="py-op">=</tt> <tt id="link-17" class="py-name" targets="Method Bio.Blast.NCBIStandalone._AlignmentConsumer.length()=Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#length"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-17', 'length', 'link-17');">length</a></tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-18" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.start=Bio.SeqFeature.CompoundLocation-class.html#start,Variable Bio.SeqFeature.FeatureLocation.start=Bio.SeqFeature.FeatureLocation-class.html#start"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-18', 'start', 'link-18');">start</a></tt> <tt class="py-op">=</tt> <tt id="link-19" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-19', 'start', 'link-18');">start</a></tt> </tt>
</div><a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"> </tt>
<a name="DBSeq.__len__"></a><div id="DBSeq.__len__-def"><a name="L42"></a><tt class="py-lineno"> 42</tt> <a class="py-toggle" href="#" id="DBSeq.__len__-toggle" onclick="return toggle('DBSeq.__len__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq.DBSeq-class.html#__len__">__len__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBSeq.__len__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBSeq.__len__-expanded"><a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-20" class="py-name"><a title="Bio.Motif.Parsers.MEME.MEMEInstance._length" class="py-name" href="#" onclick="return doclink('link-20', '_length', 'link-16');">_length</a></tt> </tt>
</div><a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line"> </tt>
<a name="DBSeq.__getitem__"></a><div id="DBSeq.__getitem__-def"><a name="L45"></a><tt class="py-lineno"> 45</tt> <a class="py-toggle" href="#" id="DBSeq.__getitem__-toggle" onclick="return toggle('DBSeq.__getitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq.DBSeq-class.html#__getitem__">__getitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">index</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># Seq API requirement</tt> </tt>
</div><a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line">        <tt class="py-comment"># Note since Python 2.0, __getslice__ is deprecated</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line">        <tt class="py-comment"># and __getitem__ is used instead.</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line">        <tt class="py-comment"># See http://docs.python.org/ref/sequence-methods.html</tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-21" class="py-name" targets="Method Bio.Crystal.Chain.index()=Bio.Crystal.Chain-class.html#index,Method Bio.SCOP.Raf.SeqMap.index()=Bio.SCOP.Raf.SeqMap-class.html#index,Method Bio.SearchIO._model.hit.Hit.index()=Bio.SearchIO._model.hit.Hit-class.html#index,Method Bio.SearchIO._model.query.QueryResult.index()=Bio.SearchIO._model.query.QueryResult-class.html#index,Function Bio.SearchIO.index()=Bio.SearchIO-module.html#index,Method Bio.Seq.MutableSeq.index()=Bio.Seq.MutableSeq-class.html#index,Function Bio.SeqIO.index()=Bio.SeqIO-module.html#index"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-21', 'index', 'link-21');">index</a></tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line">            <tt class="py-comment"># Return a single letter as a string</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line">            <tt id="link-22" class="py-name" targets="Variable Bio.PDB.Polypeptide.i=Bio.PDB.Polypeptide-module.html#i,Variable Bio.SeqIO.SffIO.i=Bio.SeqIO.SffIO-module.html#i"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-22', 'i', 'link-22');">i</a></tt> <tt class="py-op">=</tt> <tt id="link-23" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-23', 'index', 'link-21');">index</a></tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-24" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-24', 'i', 'link-22');">i</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-op">-</tt><tt id="link-25" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-25', 'i', 'link-22');">i</a></tt> <tt class="py-op">&gt;</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-26" class="py-name"><a title="Bio.Motif.Parsers.MEME.MEMEInstance._length" class="py-name" href="#" onclick="return doclink('link-26', '_length', 'link-16');">_length</a></tt><tt class="py-op">:</tt> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt class="py-name">IndexError</tt><tt class="py-op">(</tt><tt id="link-27" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-27', 'i', 'link-22');">i</a></tt><tt class="py-op">)</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line">                <tt id="link-28" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-28', 'i', 'link-22');">i</a></tt> <tt class="py-op">=</tt> <tt id="link-29" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-29', 'i', 'link-22');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-30" class="py-name"><a title="Bio.Motif.Parsers.MEME.MEMEInstance._length" class="py-name" href="#" onclick="return doclink('link-30', '_length', 'link-16');">_length</a></tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-31" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-31', 'i', 'link-22');">i</a></tt> <tt class="py-op">&gt;=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-32" class="py-name"><a title="Bio.Motif.Parsers.MEME.MEMEInstance._length" class="py-name" href="#" onclick="return doclink('link-32', '_length', 'link-16');">_length</a></tt><tt class="py-op">:</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">IndexError</tt><tt class="py-op">(</tt><tt id="link-33" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-33', 'i', 'link-22');">i</a></tt><tt class="py-op">)</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-34" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.get_subseq_as_string()=BioSQL.BioSeqDatabase.Adaptor-class.html#get_subseq_as_string"><a title="BioSQL.BioSeqDatabase.Adaptor.get_subseq_as_string" class="py-name" href="#" onclick="return doclink('link-34', 'get_subseq_as_string', 'link-34');">get_subseq_as_string</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">primary_id</tt><tt class="py-op">,</tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line">                                                     <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-35" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-35', 'start', 'link-18');">start</a></tt> <tt class="py-op">+</tt> <tt id="link-36" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-36', 'i', 'link-22');">i</a></tt><tt class="py-op">,</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line">                                                     <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-37" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-37', 'start', 'link-18');">start</a></tt> <tt class="py-op">+</tt> <tt id="link-38" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-38', 'i', 'link-22');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-39" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-39', 'index', 'link-21');">index</a></tt><tt class="py-op">,</tt> <tt class="py-name">slice</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Unexpected index type"</tt><tt class="py-op">)</tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line"> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line">        <tt class="py-comment"># Return the (sub)sequence as another DBSeq or Seq object</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line">        <tt class="py-comment"># (see the Seq obect's __getitem__ method)</tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-40" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-40', 'index', 'link-21');">index</a></tt><tt class="py-op">.</tt><tt id="link-41" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-41', 'start', 'link-18');">start</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line">            <tt id="link-42" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-42', 'i', 'link-22');">i</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line">            <tt id="link-43" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-43', 'i', 'link-22');">i</a></tt> <tt class="py-op">=</tt> <tt id="link-44" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-44', 'index', 'link-21');">index</a></tt><tt class="py-op">.</tt><tt id="link-45" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-45', 'start', 'link-18');">start</a></tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-46" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-46', 'i', 'link-22');">i</a></tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line">            <tt class="py-comment"># Map to equavilent positive index</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-op">-</tt><tt id="link-47" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-47', 'i', 'link-22');">i</a></tt> <tt class="py-op">&gt;</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-48" class="py-name"><a title="Bio.Motif.Parsers.MEME.MEMEInstance._length" class="py-name" href="#" onclick="return doclink('link-48', '_length', 'link-16');">_length</a></tt><tt class="py-op">:</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">IndexError</tt><tt class="py-op">(</tt><tt id="link-49" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-49', 'i', 'link-22');">i</a></tt><tt class="py-op">)</tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line">            <tt id="link-50" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-50', 'i', 'link-22');">i</a></tt> <tt class="py-op">=</tt> <tt id="link-51" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-51', 'i', 'link-22');">i</a></tt> <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-52" class="py-name"><a title="Bio.Motif.Parsers.MEME.MEMEInstance._length" class="py-name" href="#" onclick="return doclink('link-52', '_length', 'link-16');">_length</a></tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-53" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-53', 'i', 'link-22');">i</a></tt> <tt class="py-op">&gt;=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-54" class="py-name"><a title="Bio.Motif.Parsers.MEME.MEMEInstance._length" class="py-name" href="#" onclick="return doclink('link-54', '_length', 'link-16');">_length</a></tt><tt class="py-op">:</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line">            <tt class="py-comment"># Trivial case, should return empty string!</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line">            <tt id="link-55" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-55', 'i', 'link-22');">i</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-56" class="py-name"><a title="Bio.Motif.Parsers.MEME.MEMEInstance._length" class="py-name" href="#" onclick="return doclink('link-56', '_length', 'link-16');">_length</a></tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line"> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-57" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-57', 'index', 'link-21');">index</a></tt><tt class="py-op">.</tt><tt class="py-name">stop</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line">            <tt class="py-name">j</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-58" class="py-name"><a title="Bio.Motif.Parsers.MEME.MEMEInstance._length" class="py-name" href="#" onclick="return doclink('link-58', '_length', 'link-16');">_length</a></tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line">            <tt class="py-name">j</tt> <tt class="py-op">=</tt> <tt id="link-59" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-59', 'index', 'link-21');">index</a></tt><tt class="py-op">.</tt><tt class="py-name">stop</tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">j</tt> <tt class="py-op">&lt;</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line">            <tt class="py-comment"># Map to equavilent positive index</tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-op">-</tt><tt class="py-name">j</tt> <tt class="py-op">&gt;</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-60" class="py-name"><a title="Bio.Motif.Parsers.MEME.MEMEInstance._length" class="py-name" href="#" onclick="return doclink('link-60', '_length', 'link-16');">_length</a></tt><tt class="py-op">:</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">IndexError</tt><tt class="py-op">(</tt><tt class="py-name">j</tt><tt class="py-op">)</tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line">            <tt class="py-name">j</tt> <tt class="py-op">=</tt> <tt class="py-name">j</tt> <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-61" class="py-name"><a title="Bio.Motif.Parsers.MEME.MEMEInstance._length" class="py-name" href="#" onclick="return doclink('link-61', '_length', 'link-16');">_length</a></tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">j</tt> <tt class="py-op">&gt;=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-62" class="py-name"><a title="Bio.Motif.Parsers.MEME.MEMEInstance._length" class="py-name" href="#" onclick="return doclink('link-62', '_length', 'link-16');">_length</a></tt><tt class="py-op">:</tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line">            <tt class="py-name">j</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-63" class="py-name"><a title="Bio.Motif.Parsers.MEME.MEMEInstance._length" class="py-name" href="#" onclick="return doclink('link-63', '_length', 'link-16');">_length</a></tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line"> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-64" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-64', 'i', 'link-22');">i</a></tt> <tt class="py-op">&gt;=</tt> <tt class="py-name">j</tt><tt class="py-op">:</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line">            <tt class="py-comment"># Trivial case, empty string.</tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-65" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-65', 'Seq', 'link-5');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-string">""</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-66" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-66', 'alphabet', 'link-14');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-67" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-67', 'index', 'link-21');">index</a></tt><tt class="py-op">.</tt><tt class="py-name">step</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt> <tt class="py-keyword">or</tt> <tt id="link-68" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-68', 'index', 'link-21');">index</a></tt><tt class="py-op">.</tt><tt class="py-name">step</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line">            <tt class="py-comment"># Easy case - can return a DBSeq with the start and end adjusted</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">primary_id</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-69" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-69', 'alphabet', 'link-14');">alphabet</a></tt><tt class="py-op">,</tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line">                                  <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-70" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-70', 'start', 'link-18');">start</a></tt> <tt class="py-op">+</tt> <tt id="link-71" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-71', 'i', 'link-22');">i</a></tt><tt class="py-op">,</tt> <tt class="py-name">j</tt> <tt class="py-op">-</tt> <tt id="link-72" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-72', 'i', 'link-22');">i</a></tt><tt class="py-op">)</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line">            <tt class="py-comment"># Tricky.  Will have to create a Seq object because of the stride</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line">            <tt id="link-73" class="py-name" targets="Method Bio.Restriction.Restriction.Analysis.full()=Bio.Restriction.Restriction.Analysis-class.html#full"><a title="Bio.Restriction.Restriction.Analysis.full" class="py-name" href="#" onclick="return doclink('link-73', 'full', 'link-73');">full</a></tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-74" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.get_subseq_as_string" class="py-name" href="#" onclick="return doclink('link-74', 'get_subseq_as_string', 'link-34');">get_subseq_as_string</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">primary_id</tt><tt class="py-op">,</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line">                                                     <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-75" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-75', 'start', 'link-18');">start</a></tt> <tt class="py-op">+</tt> <tt id="link-76" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-76', 'i', 'link-22');">i</a></tt><tt class="py-op">,</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line">                                                     <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-77" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-77', 'start', 'link-18');">start</a></tt> <tt class="py-op">+</tt> <tt class="py-name">j</tt><tt class="py-op">)</tt> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt id="link-78" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-78', 'Seq', 'link-5');">Seq</a></tt><tt class="py-op">(</tt><tt id="link-79" class="py-name"><a title="Bio.Restriction.Restriction.Analysis.full" class="py-name" href="#" onclick="return doclink('link-79', 'full', 'link-73');">full</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">:</tt><tt id="link-80" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-80', 'index', 'link-21');">index</a></tt><tt class="py-op">.</tt><tt class="py-name">step</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-81" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-81', 'alphabet', 'link-14');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line"> </tt>
<a name="DBSeq.tostring"></a><div id="DBSeq.tostring-def"><a name="L105"></a><tt class="py-lineno">105</tt> <a class="py-toggle" href="#" id="DBSeq.tostring-toggle" onclick="return toggle('DBSeq.tostring');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq.DBSeq-class.html#tostring">tostring</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBSeq.tostring-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBSeq.tostring-expanded"><a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line">        <tt class="py-docstring">"""Returns the full sequence as a python string (DEPRECATED).</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line"><tt class="py-docstring">        You are now encouraged to use str(my_seq) instead of</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line"><tt class="py-docstring">        my_seq.tostring()."""</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line">        <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line">        <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"This method is obsolete; please use str(my_seq) "</tt> </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">                      <tt class="py-string">"instead of my_seq.tostring()."</tt><tt class="py-op">,</tt> </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">                      <tt class="py-name">PendingDeprecationWarning</tt><tt class="py-op">)</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-82" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.get_subseq_as_string" class="py-name" href="#" onclick="return doclink('link-82', 'get_subseq_as_string', 'link-34');">get_subseq_as_string</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">primary_id</tt><tt class="py-op">,</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">                                                 <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-83" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-83', 'start', 'link-18');">start</a></tt><tt class="py-op">,</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">                                                 <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-84" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-84', 'start', 'link-18');">start</a></tt> <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-85" class="py-name"><a title="Bio.Motif.Parsers.MEME.MEMEInstance._length" class="py-name" href="#" onclick="return doclink('link-85', '_length', 'link-16');">_length</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line"> </tt>
<a name="DBSeq.__str__"></a><div id="DBSeq.__str__-def"><a name="L118"></a><tt class="py-lineno">118</tt> <a class="py-toggle" href="#" id="DBSeq.__str__-toggle" onclick="return toggle('DBSeq.__str__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq.DBSeq-class.html#__str__">__str__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBSeq.__str__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBSeq.__str__-expanded"><a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line">        <tt class="py-docstring">"""Returns the full sequence as a python string."""</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-86" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.get_subseq_as_string" class="py-name" href="#" onclick="return doclink('link-86', 'get_subseq_as_string', 'link-34');">get_subseq_as_string</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">primary_id</tt><tt class="py-op">,</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">                                                 <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-87" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-87', 'start', 'link-18');">start</a></tt><tt class="py-op">,</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line">                                                 <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-88" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-88', 'start', 'link-18');">start</a></tt> <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-89" class="py-name"><a title="Bio.Motif.Parsers.MEME.MEMEInstance._length" class="py-name" href="#" onclick="return doclink('link-89', '_length', 'link-16');">_length</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line"> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line">    <tt id="link-90" class="py-name" targets="Variable BioSQL.BioSeq.DBSeq.data=BioSQL.BioSeq.DBSeq-class.html#data"><a title="BioSQL.BioSeq.DBSeq.data" class="py-name" href="#" onclick="return doclink('link-90', 'data', 'link-90');">data</a></tt> <tt class="py-op">=</tt> <tt id="link-91" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.property()=Bio.Phylo.PhyloXMLIO.Parser-class.html#property,Method Bio.Phylo.PhyloXMLIO.Writer.property()=Bio.Phylo.PhyloXMLIO.Writer-class.html#property"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-91', 'property', 'link-91');">property</a></tt><tt class="py-op">(</tt><tt id="link-92" class="py-name" targets="Method Bio.Crystal.Crystal.tostring()=Bio.Crystal.Crystal-class.html#tostring,Method Bio.Seq.MutableSeq.tostring()=Bio.Seq.MutableSeq-class.html#tostring,Method Bio.Seq.Seq.tostring()=Bio.Seq.Seq-class.html#tostring,Method BioSQL.BioSeq.DBSeq.tostring()=BioSQL.BioSeq.DBSeq-class.html#tostring"><a title="Bio.Crystal.Crystal.tostring
Bio.Seq.MutableSeq.tostring
Bio.Seq.Seq.tostring
BioSQL.BioSeq.DBSeq.tostring" class="py-name" href="#" onclick="return doclink('link-92', 'tostring', 'link-92');">tostring</a></tt><tt class="py-op">,</tt> <tt class="py-name">doc</tt><tt class="py-op">=</tt><tt class="py-string">"Sequence as string (DEPRECATED)"</tt><tt class="py-op">)</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line"> </tt>
<a name="DBSeq.toseq"></a><div id="DBSeq.toseq-def"><a name="L126"></a><tt class="py-lineno">126</tt> <a class="py-toggle" href="#" id="DBSeq.toseq-toggle" onclick="return toggle('DBSeq.toseq');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq.DBSeq-class.html#toseq">toseq</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBSeq.toseq-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBSeq.toseq-expanded"><a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line">        <tt class="py-docstring">"""Returns the full sequence as a Seq object."""</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line">        <tt class="py-comment"># Note - the method name copies that of the MutableSeq object</tt> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-93" class="py-name"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-93', 'Seq', 'link-5');">Seq</a></tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-94" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-94', 'alphabet', 'link-14');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line"> </tt>
<a name="DBSeq.__add__"></a><div id="DBSeq.__add__-def"><a name="L131"></a><tt class="py-lineno">131</tt> <a class="py-toggle" href="#" id="DBSeq.__add__-toggle" onclick="return toggle('DBSeq.__add__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq.DBSeq-class.html#__add__">__add__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBSeq.__add__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBSeq.__add__-expanded"><a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line">        <tt class="py-comment"># Let the Seq object deal with the alphabet issues etc</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-95" class="py-name" targets="Method Bio.Seq.MutableSeq.toseq()=Bio.Seq.MutableSeq-class.html#toseq,Method BioSQL.BioSeq.DBSeq.toseq()=BioSQL.BioSeq.DBSeq-class.html#toseq"><a title="Bio.Seq.MutableSeq.toseq
BioSQL.BioSeq.DBSeq.toseq" class="py-name" href="#" onclick="return doclink('link-95', 'toseq', 'link-95');">toseq</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt id="link-96" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Parser.other()=Bio.Phylo.PhyloXMLIO.Parser-class.html#other,Method Bio.Phylo.PhyloXMLIO.Writer.other()=Bio.Phylo.PhyloXMLIO.Writer-class.html#other"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-96', 'other', 'link-96');">other</a></tt> </tt>
</div><a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line"> </tt>
<a name="DBSeq.__radd__"></a><div id="DBSeq.__radd__-def"><a name="L135"></a><tt class="py-lineno">135</tt> <a class="py-toggle" href="#" id="DBSeq.__radd__-toggle" onclick="return toggle('DBSeq.__radd__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq.DBSeq-class.html#__radd__">__radd__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">other</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBSeq.__radd__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBSeq.__radd__-expanded"><a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line">        <tt class="py-comment"># Let the Seq object deal with the alphabet issues etc</tt> </tt>
<a name="L137"></a><tt class="py-lineno">137</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-97" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.other
Bio.Phylo.PhyloXMLIO.Writer.other" class="py-name" href="#" onclick="return doclink('link-97', 'other', 'link-96');">other</a></tt> <tt class="py-op">+</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-98" class="py-name"><a title="Bio.Seq.MutableSeq.toseq
BioSQL.BioSeq.DBSeq.toseq" class="py-name" href="#" onclick="return doclink('link-98', 'toseq', 'link-95');">toseq</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line"> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line"> </tt>
<a name="_retrieve_seq"></a><div id="_retrieve_seq-def"><a name="L140"></a><tt class="py-lineno">140</tt> <a class="py-toggle" href="#" id="_retrieve_seq-toggle" onclick="return toggle('_retrieve_seq');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq-module.html#_retrieve_seq">_retrieve_seq</a><tt class="py-op">(</tt><tt class="py-param">adaptor</tt><tt class="py-op">,</tt> <tt class="py-param">primary_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_retrieve_seq-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_retrieve_seq-expanded"><a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line">    <tt class="py-comment"># The database schema ensures there will be only one matching</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line">    <tt class="py-comment"># row in the table.</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line"> </tt>
<a name="L144"></a><tt class="py-lineno">144</tt>  <tt class="py-line">    <tt class="py-comment"># If an UnknownSeq was recorded, seq will be NULL,</tt> </tt>
<a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line">    <tt class="py-comment"># but length will be populated.  This means length(seq)</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line">    <tt class="py-comment"># will return None.</tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line">    <tt class="py-name">seqs</tt> <tt class="py-op">=</tt> <tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-99" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.execute_and_fetchall()=BioSQL.BioSeqDatabase.Adaptor-class.html#execute_and_fetchall,Method BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetchall()=BioSQL.BioSeqDatabase.MysqlConnectorAdaptor-class.html#execute_and_fetchall"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetchall
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetchall" class="py-name" href="#" onclick="return doclink('link-99', 'execute_and_fetchall', 'link-99');">execute_and_fetchall</a></tt><tt class="py-op">(</tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line">        <tt class="py-string">"SELECT alphabet, length, length(seq) FROM biosequence"</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line">        <tt class="py-string">" WHERE bioentry_id = %s"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">primary_id</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">seqs</tt><tt class="py-op">:</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line">    <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">seqs</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line">    <tt class="py-name">moltype</tt><tt class="py-op">,</tt> <tt class="py-name">given_length</tt><tt class="py-op">,</tt> <tt id="link-100" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-100', 'length', 'link-17');">length</a></tt> <tt class="py-op">=</tt> <tt class="py-name">seqs</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line"> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line">        <tt id="link-101" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-101', 'length', 'link-17');">length</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-102" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-102', 'length', 'link-17');">length</a></tt><tt class="py-op">)</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">        <tt class="py-name">given_length</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-103" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-103', 'length', 'link-17');">length</a></tt><tt class="py-op">)</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt id="link-104" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-104', 'length', 'link-17');">length</a></tt> <tt class="py-op">==</tt> <tt class="py-name">given_length</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">        <tt class="py-name">have_seq</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">TypeError</tt><tt class="py-op">:</tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt id="link-105" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-105', 'length', 'link-17');">length</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">        <tt class="py-name">seqs</tt> <tt class="py-op">=</tt> <tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-106" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetchall
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetchall" class="py-name" href="#" onclick="return doclink('link-106', 'execute_and_fetchall', 'link-99');">execute_and_fetchall</a></tt><tt class="py-op">(</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line">            <tt class="py-string">"SELECT alphabet, length, seq FROM biosequence"</tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">            <tt class="py-string">" WHERE bioentry_id = %s"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">primary_id</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">seqs</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line">        <tt class="py-name">moltype</tt><tt class="py-op">,</tt> <tt class="py-name">given_length</tt><tt class="py-op">,</tt> <tt id="link-107" class="py-name" targets="Variable Bio.SeqRecord.SeqRecord.seq=Bio.SeqRecord.SeqRecord-class.html#seq,Variable BioSQL.BioSeq.DBSeqRecord.seq=BioSQL.BioSeq.DBSeqRecord-class.html#seq"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-107', 'seq', 'link-107');">seq</a></tt> <tt class="py-op">=</tt> <tt class="py-name">seqs</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt id="link-108" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-108', 'seq', 'link-107');">seq</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt> <tt class="py-keyword">or</tt> <tt id="link-109" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-109', 'seq', 'link-107');">seq</a></tt> <tt class="py-op">==</tt> <tt class="py-string">""</tt> </tt>
<a name="L168"></a><tt class="py-lineno">168</tt>  <tt class="py-line">        <tt id="link-110" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-110', 'length', 'link-17');">length</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">given_length</tt><tt class="py-op">)</tt> </tt>
<a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line">        <tt class="py-name">have_seq</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt id="link-111" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-111', 'seq', 'link-107');">seq</a></tt> </tt>
<a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line">    <tt class="py-keyword">del</tt> <tt class="py-name">given_length</tt> </tt>
<a name="L172"></a><tt class="py-lineno">172</tt>  <tt class="py-line"> </tt>
<a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line">    <tt class="py-name">moltype</tt> <tt class="py-op">=</tt> <tt class="py-name">moltype</tt><tt class="py-op">.</tt><tt id="link-112" class="py-name" targets="Method Bio.Seq.Seq.lower()=Bio.Seq.Seq-class.html#lower,Method Bio.Seq.UnknownSeq.lower()=Bio.Seq.UnknownSeq-class.html#lower,Method Bio.SeqRecord.SeqRecord.lower()=Bio.SeqRecord.SeqRecord-class.html#lower"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-112', 'lower', 'link-112');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt>  <tt class="py-comment"># might be upper case in database</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line">    <tt class="py-comment"># We have no way of knowing if these sequences will use IUPAC</tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line">    <tt class="py-comment"># alphabets, and we certainly can't assume they are unambiguous!</tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">moltype</tt> <tt class="py-op">==</tt> <tt class="py-string">"dna"</tt><tt class="py-op">:</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line">        <tt id="link-113" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-113', 'alphabet', 'link-14');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-114" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-114', 'Alphabet', 'link-3');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-115" class="py-name" targets="Variable Bio.Alphabet.generic_dna=Bio.Alphabet-module.html#generic_dna"><a title="Bio.Alphabet.generic_dna" class="py-name" href="#" onclick="return doclink('link-115', 'generic_dna', 'link-115');">generic_dna</a></tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">moltype</tt> <tt class="py-op">==</tt> <tt class="py-string">"rna"</tt><tt class="py-op">:</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line">        <tt id="link-116" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-116', 'alphabet', 'link-14');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-117" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-117', 'Alphabet', 'link-3');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-118" class="py-name" targets="Variable Bio.Alphabet.generic_rna=Bio.Alphabet-module.html#generic_rna"><a title="Bio.Alphabet.generic_rna" class="py-name" href="#" onclick="return doclink('link-118', 'generic_rna', 'link-118');">generic_rna</a></tt> </tt>
<a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">moltype</tt> <tt class="py-op">==</tt> <tt class="py-string">"protein"</tt><tt class="py-op">:</tt> </tt>
<a name="L181"></a><tt class="py-lineno">181</tt>  <tt class="py-line">        <tt id="link-119" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-119', 'alphabet', 'link-14');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-120" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-120', 'Alphabet', 'link-3');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-121" class="py-name" targets="Variable Bio.Alphabet.generic_protein=Bio.Alphabet-module.html#generic_protein"><a title="Bio.Alphabet.generic_protein" class="py-name" href="#" onclick="return doclink('link-121', 'generic_protein', 'link-121');">generic_protein</a></tt> </tt>
<a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">moltype</tt> <tt class="py-op">==</tt> <tt class="py-string">"unknown"</tt><tt class="py-op">:</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">        <tt class="py-comment"># This is used in BioSQL/Loader.py and would happen</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">        <tt class="py-comment"># for any generic or nucleotide alphabets.</tt> </tt>
<a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line">        <tt id="link-122" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-122', 'alphabet', 'link-14');">alphabet</a></tt> <tt class="py-op">=</tt> <tt id="link-123" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-123', 'Alphabet', 'link-3');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-124" class="py-name" targets="Variable Bio.Alphabet.single_letter_alphabet=Bio.Alphabet-module.html#single_letter_alphabet"><a title="Bio.Alphabet.single_letter_alphabet" class="py-name" href="#" onclick="return doclink('link-124', 'single_letter_alphabet', 'link-124');">single_letter_alphabet</a></tt> </tt>
<a name="L186"></a><tt class="py-lineno">186</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">AssertionError</tt><tt class="py-op">(</tt><tt class="py-string">"Unknown moltype: %s"</tt> <tt class="py-op">%</tt> <tt class="py-name">moltype</tt><tt class="py-op">)</tt> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line"> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">have_seq</tt><tt class="py-op">:</tt> </tt>
<a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-125" class="py-name" targets="Class BioSQL.BioSeq.DBSeq=BioSQL.BioSeq.DBSeq-class.html"><a title="BioSQL.BioSeq.DBSeq" class="py-name" href="#" onclick="return doclink('link-125', 'DBSeq', 'link-125');">DBSeq</a></tt><tt class="py-op">(</tt><tt class="py-name">primary_id</tt><tt class="py-op">,</tt> <tt class="py-name">adaptor</tt><tt class="py-op">,</tt> <tt id="link-126" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-126', 'alphabet', 'link-14');">alphabet</a></tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-127" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-127', 'length', 'link-17');">length</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L191"></a><tt class="py-lineno">191</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-128" class="py-name"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-128', 'UnknownSeq', 'link-7');">UnknownSeq</a></tt><tt class="py-op">(</tt><tt id="link-129" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-129', 'length', 'link-17');">length</a></tt><tt class="py-op">,</tt> <tt id="link-130" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-130', 'alphabet', 'link-14');">alphabet</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line"> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line"> </tt>
<a name="_retrieve_dbxrefs"></a><div id="_retrieve_dbxrefs-def"><a name="L195"></a><tt class="py-lineno">195</tt> <a class="py-toggle" href="#" id="_retrieve_dbxrefs-toggle" onclick="return toggle('_retrieve_dbxrefs');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq-module.html#_retrieve_dbxrefs">_retrieve_dbxrefs</a><tt class="py-op">(</tt><tt class="py-param">adaptor</tt><tt class="py-op">,</tt> <tt class="py-param">primary_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_retrieve_dbxrefs-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_retrieve_dbxrefs-expanded"><a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">    <tt class="py-docstring">"""Retrieve the database cross references for the sequence."""</tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line">    <tt class="py-name">_dbxrefs</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L198"></a><tt class="py-lineno">198</tt>  <tt class="py-line">    <tt id="link-131" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.dbxrefs=BioSQL.BioSeq.DBSeqRecord-class.html#dbxrefs"><a title="BioSQL.BioSeq.DBSeqRecord.dbxrefs" class="py-name" href="#" onclick="return doclink('link-131', 'dbxrefs', 'link-131');">dbxrefs</a></tt> <tt class="py-op">=</tt> <tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-132" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetchall
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetchall" class="py-name" href="#" onclick="return doclink('link-132', 'execute_and_fetchall', 'link-99');">execute_and_fetchall</a></tt><tt class="py-op">(</tt> </tt>
<a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line">        <tt class="py-string">"SELECT dbname, accession, version"</tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line">        <tt class="py-string">" FROM bioentry_dbxref join dbxref using (dbxref_id)"</tt> </tt>
<a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line">        <tt class="py-string">" WHERE bioentry_id = %s"</tt> </tt>
<a name="L202"></a><tt class="py-lineno">202</tt>  <tt class="py-line">        <tt class="py-string">" ORDER BY rank"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">primary_id</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">dbname</tt><tt class="py-op">,</tt> <tt id="link-133" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.accession()=Bio.GenBank._FeatureConsumer-class.html#accession,Method Bio.GenBank._RecordConsumer.accession()=Bio.GenBank._RecordConsumer-class.html#accession,Method Bio.Phylo.PhyloXMLIO.Parser.accession()=Bio.Phylo.PhyloXMLIO.Parser-class.html#accession,Method Bio.Phylo.PhyloXMLIO.Writer.accession()=Bio.Phylo.PhyloXMLIO.Writer-class.html#accession"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-133', 'accession', 'link-133');">accession</a></tt><tt class="py-op">,</tt> <tt id="link-134" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HeaderConsumer.version()=Bio.Blast.NCBIStandalone._HeaderConsumer-class.html#version,Function Bio.Cluster.cluster.version()=Bio.Cluster.cluster-module.html#version,Method Bio.GenBank._FeatureConsumer.version()=Bio.GenBank._FeatureConsumer-class.html#version,Method Bio.GenBank._RecordConsumer.version()=Bio.GenBank._RecordConsumer-class.html#version,Variable Bio.motifs.jaspar.Motif.version=Bio.motifs.jaspar.Motif-class.html#version"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-134', 'version', 'link-134');">version</a></tt> <tt class="py-keyword">in</tt> <tt id="link-135" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.dbxrefs" class="py-name" href="#" onclick="return doclink('link-135', 'dbxrefs', 'link-131');">dbxrefs</a></tt><tt class="py-op">:</tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-136" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-136', 'version', 'link-134');">version</a></tt> <tt class="py-keyword">and</tt> <tt id="link-137" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-137', 'version', 'link-134');">version</a></tt> <tt class="py-op">!=</tt> <tt class="py-string">"0"</tt><tt class="py-op">:</tt> </tt>
<a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line">            <tt class="py-name">v</tt> <tt class="py-op">=</tt> <tt class="py-string">"%s.%s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-138" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-138', 'accession', 'link-133');">accession</a></tt><tt class="py-op">,</tt> <tt id="link-139" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-139', 'version', 'link-134');">version</a></tt><tt class="py-op">)</tt> </tt>
<a name="L206"></a><tt class="py-lineno">206</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line">            <tt class="py-name">v</tt> <tt class="py-op">=</tt> <tt id="link-140" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-140', 'accession', 'link-133');">accession</a></tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line">        <tt class="py-name">_dbxrefs</tt><tt class="py-op">.</tt><tt id="link-141" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-141', 'append', 'link-141');">append</a></tt><tt class="py-op">(</tt><tt class="py-string">"%s:%s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">dbname</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">_dbxrefs</tt> </tt>
</div><a name="L210"></a><tt class="py-lineno">210</tt>  <tt class="py-line"> </tt>
<a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line"> </tt>
<a name="_retrieve_features"></a><div id="_retrieve_features-def"><a name="L212"></a><tt class="py-lineno">212</tt> <a class="py-toggle" href="#" id="_retrieve_features-toggle" onclick="return toggle('_retrieve_features');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq-module.html#_retrieve_features">_retrieve_features</a><tt class="py-op">(</tt><tt class="py-param">adaptor</tt><tt class="py-op">,</tt> <tt class="py-param">primary_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_retrieve_features-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_retrieve_features-expanded"><a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line">    <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">"SELECT seqfeature_id, type.name, rank"</tt> \ </tt>
<a name="L214"></a><tt class="py-lineno">214</tt>  <tt class="py-line">          <tt class="py-string">" FROM seqfeature join term type on (type_term_id = type.term_id)"</tt> \ </tt>
<a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line">          <tt class="py-string">" WHERE bioentry_id = %s"</tt> \ </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line">          <tt class="py-string">" ORDER BY rank"</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line">    <tt id="link-142" class="py-name" targets="Variable Bio.Restriction.Restriction.AanI.results=Bio.Restriction.Restriction.AanI-class.html#results,Variable Bio.Restriction.Restriction.AarI.results=Bio.Restriction.Restriction.AarI-class.html#results,Variable Bio.Restriction.Restriction.AasI.results=Bio.Restriction.Restriction.AasI-class.html#results,Variable Bio.Restriction.Restriction.AatII.results=Bio.Restriction.Restriction.AatII-class.html#results,Variable Bio.Restriction.Restriction.AbaSI.results=Bio.Restriction.Restriction.AbaSI-class.html#results,Variable Bio.Restriction.Restriction.AbsI.results=Bio.Restriction.Restriction.AbsI-class.html#results,Variable Bio.Restriction.Restriction.Acc16I.results=Bio.Restriction.Restriction.Acc16I-class.html#results,Variable Bio.Restriction.Restriction.Acc36I.results=Bio.Restriction.Restriction.Acc36I-class.html#results,Variable Bio.Restriction.Restriction.Acc65I.results=Bio.Restriction.Restriction.Acc65I-class.html#results,Variable Bio.Restriction.Restriction.AccB1I.results=Bio.Restriction.Restriction.AccB1I-class.html#results,Variable Bio.Restriction.Restriction.AccB7I.results=Bio.Restriction.Restriction.AccB7I-class.html#results,Variable Bio.Restriction.Restriction.AccBSI.results=Bio.Restriction.Restriction.AccBSI-class.html#results,Variable Bio.Restriction.Restriction.AccI.results=Bio.Restriction.Restriction.AccI-class.html#results,Variable Bio.Restriction.Restriction.AccII.results=Bio.Restriction.Restriction.AccII-class.html#results,Variable Bio.Restriction.Restriction.AccIII.results=Bio.Restriction.Restriction.AccIII-class.html#results,Variable Bio.Restriction.Restriction.AceIII.results=Bio.Restriction.Restriction.AceIII-class.html#results,Variable Bio.Restriction.Restriction.AciI.results=Bio.Restriction.Restriction.AciI-class.html#results,Variable Bio.Restriction.Restriction.AclI.results=Bio.Restriction.Restriction.AclI-class.html#results,Variable Bio.Restriction.Restriction.AclWI.results=Bio.Restriction.Restriction.AclWI-class.html#results,Variable Bio.Restriction.Restriction.AcoI.results=Bio.Restriction.Restriction.AcoI-class.html#results,Variable Bio.Restriction.Restriction.AcsI.results=Bio.Restriction.Restriction.AcsI-class.html#results,Variable Bio.Restriction.Restriction.AcuI.results=Bio.Restriction.Restriction.AcuI-class.html#results,Variable Bio.Restriction.Restriction.AcvI.results=Bio.Restriction.Restriction.AcvI-class.html#results,Variable Bio.Restriction.Restriction.AcyI.results=Bio.Restriction.Restriction.AcyI-class.html#results,Variable Bio.Restriction.Restriction.AdeI.results=Bio.Restriction.Restriction.AdeI-class.html#results,Variable Bio.Restriction.Restriction.AfaI.results=Bio.Restriction.Restriction.AfaI-class.html#results,Variable Bio.Restriction.Restriction.AfeI.results=Bio.Restriction.Restriction.AfeI-class.html#results,Variable Bio.Restriction.Restriction.AfiI.results=Bio.Restriction.Restriction.AfiI-class.html#results,Variable Bio.Restriction.Restriction.AflII.results=Bio.Restriction.Restriction.AflII-class.html#results,Variable Bio.Restriction.Restriction.AflIII.results=Bio.Restriction.Restriction.AflIII-class.html#results,Variable Bio.Restriction.Restriction.AgeI.results=Bio.Restriction.Restriction.AgeI-class.html#results,Variable Bio.Restriction.Restriction.AgsI.results=Bio.Restriction.Restriction.AgsI-class.html#results,Variable Bio.Restriction.Restriction.AhaIII.results=Bio.Restriction.Restriction.AhaIII-class.html#results,Variable Bio.Restriction.Restriction.AhdI.results=Bio.Restriction.Restriction.AhdI-class.html#results,Variable Bio.Restriction.Restriction.AhlI.results=Bio.Restriction.Restriction.AhlI-class.html#results,Variable Bio.Restriction.Restriction.AjiI.results=Bio.Restriction.Restriction.AjiI-class.html#results,Variable Bio.Restriction.Restriction.AjnI.results=Bio.Restriction.Restriction.AjnI-class.html#results,Variable Bio.Restriction.Restriction.AjuI.results=Bio.Restriction.Restriction.AjuI-class.html#results,Variable Bio.Restriction.Restriction.AleI.results=Bio.Restriction.Restriction.AleI-class.html#results,Variable Bio.Restriction.Restriction.AlfI.results=Bio.Restriction.Restriction.AlfI-class.html#results,Variable Bio.Restriction.Restriction.AloI.results=Bio.Restriction.Restriction.AloI-class.html#results,Variable Bio.Restriction.Restriction.AluBI.results=Bio.Restriction.Restriction.AluBI-class.html#results,Variable Bio.Restriction.Restriction.AluI.results=Bio.Restriction.Restriction.AluI-class.html#results,Variable Bio.Restriction.Restriction.Alw21I.results=Bio.Restriction.Restriction.Alw21I-class.html#results,Variable Bio.Restriction.Restriction.Alw26I.results=Bio.Restriction.Restriction.Alw26I-class.html#results,Variable Bio.Restriction.Restriction.Alw44I.results=Bio.Restriction.Restriction.Alw44I-class.html#results,Variable Bio.Restriction.Restriction.AlwFI.results=Bio.Restriction.Restriction.AlwFI-class.html#results,Variable Bio.Restriction.Restriction.AlwI.results=Bio.Restriction.Restriction.AlwI-class.html#results,Variable Bio.Restriction.Restriction.AlwNI.results=Bio.Restriction.Restriction.AlwNI-class.html#results,Variable Bio.Restriction.Restriction.Ama87I.results=Bio.Restriction.Restriction.Ama87I-class.html#results,Variable Bio.Restriction.Restriction.Aor13HI.results=Bio.Restriction.Restriction.Aor13HI-class.html#results,Variable Bio.Restriction.Restriction.Aor51HI.results=Bio.Restriction.Restriction.Aor51HI-class.html#results,Variable Bio.Restriction.Restriction.AoxI.results=Bio.Restriction.Restriction.AoxI-class.html#results,Variable Bio.Restriction.Restriction.ApaBI.results=Bio.Restriction.Restriction.ApaBI-class.html#results,Variable Bio.Restriction.Restriction.ApaI.results=Bio.Restriction.Restriction.ApaI-class.html#results,Variable Bio.Restriction.Restriction.ApaLI.results=Bio.Restriction.Restriction.ApaLI-class.html#results,Variable Bio.Restriction.Restriction.ApeKI.results=Bio.Restriction.Restriction.ApeKI-class.html#results,Variable Bio.Restriction.Restriction.ApoI.results=Bio.Restriction.Restriction.ApoI-class.html#results,Variable Bio.Restriction.Restriction.ApyPI.results=Bio.Restriction.Restriction.ApyPI-class.html#results,Variable Bio.Restriction.Restriction.AquII.results=Bio.Restriction.Restriction.AquII-class.html#results,Variable Bio.Restriction.Restriction.AquIII.results=Bio.Restriction.Restriction.AquIII-class.html#results,Variable Bio.Restriction.Restriction.AquIV.results=Bio.Restriction.Restriction.AquIV-class.html#results,Variable Bio.Restriction.Restriction.ArsI.results=Bio.Restriction.Restriction.ArsI-class.html#results,Variable Bio.Restriction.Restriction.AscI.results=Bio.Restriction.Restriction.AscI-class.html#results,Variable Bio.Restriction.Restriction.AseI.results=Bio.Restriction.Restriction.AseI-class.html#results,Variable Bio.Restriction.Restriction.Asi256I.results=Bio.Restriction.Restriction.Asi256I-class.html#results,Variable Bio.Restriction.Restriction.AsiGI.results=Bio.Restriction.Restriction.AsiGI-class.html#results,Variable Bio.Restriction.Restriction.AsiSI.results=Bio.Restriction.Restriction.AsiSI-class.html#results,Variable Bio.Restriction.Restriction.Asp700I.results=Bio.Restriction.Restriction.Asp700I-class.html#results,Variable Bio.Restriction.Restriction.Asp718I.results=Bio.Restriction.Restriction.Asp718I-class.html#results,Variable Bio.Restriction.Restriction.AspA2I.results=Bio.Restriction.Restriction.AspA2I-class.html#results,Variable Bio.Restriction.Restriction.AspBHI.results=Bio.Restriction.Restriction.AspBHI-class.html#results,Variable Bio.Restriction.Restriction.AspLEI.results=Bio.Restriction.Restriction.AspLEI-class.html#results,Variable Bio.Restriction.Restriction.AspS9I.results=Bio.Restriction.Restriction.AspS9I-class.html#results,Variable Bio.Restriction.Restriction.AssI.results=Bio.Restriction.Restriction.AssI-class.html#results,Variable Bio.Restriction.Restriction.AsuC2I.results=Bio.Restriction.Restriction.AsuC2I-class.html#results,Variable Bio.Restriction.Restriction.AsuHPI.results=Bio.Restriction.Restriction.AsuHPI-class.html#results,Variable Bio.Restriction.Restriction.AsuI.results=Bio.Restriction.Restriction.AsuI-class.html#results,Variable Bio.Restriction.Restriction.AsuII.results=Bio.Restriction.Restriction.AsuII-class.html#results,Variable Bio.Restriction.Restriction.AsuNHI.results=Bio.Restriction.Restriction.AsuNHI-class.html#results,Variable Bio.Restriction.Restriction.AvaI.results=Bio.Restriction.Restriction.AvaI-class.html#results,Variable Bio.Restriction.Restriction.AvaII.results=Bio.Restriction.Restriction.AvaII-class.html#results,Variable Bio.Restriction.Restriction.AvaIII.results=Bio.Restriction.Restriction.AvaIII-class.html#results,Variable Bio.Restriction.Restriction.AvrII.results=Bio.Restriction.Restriction.AvrII-class.html#results,Variable Bio.Restriction.Restriction.AxyI.results=Bio.Restriction.Restriction.AxyI-class.html#results,Variable Bio.Restriction.Restriction.BaeGI.results=Bio.Restriction.Restriction.BaeGI-class.html#results,Variable Bio.Restriction.Restriction.BaeI.results=Bio.Restriction.Restriction.BaeI-class.html#results,Variable Bio.Restriction.Restriction.BalI.results=Bio.Restriction.Restriction.BalI-class.html#results,Variable Bio.Restriction.Restriction.BamHI.results=Bio.Restriction.Restriction.BamHI-class.html#results,Variable Bio.Restriction.Restriction.BanI.results=Bio.Restriction.Restriction.BanI-class.html#results,Variable Bio.Restriction.Restriction.BanII.results=Bio.Restriction.Restriction.BanII-class.html#results,Variable Bio.Restriction.Restriction.BarI.results=Bio.Restriction.Restriction.BarI-class.html#results,Variable Bio.Restriction.Restriction.BasI.results=Bio.Restriction.Restriction.BasI-class.html#results,Variable Bio.Restriction.Restriction.BauI.results=Bio.Restriction.Restriction.BauI-class.html#results,Variable Bio.Restriction.Restriction.Bbr7I.results=Bio.Restriction.Restriction.Bbr7I-class.html#results,Variable Bio.Restriction.Restriction.BbrPI.results=Bio.Restriction.Restriction.BbrPI-class.html#results,Variable Bio.Restriction.Restriction.BbsI.results=Bio.Restriction.Restriction.BbsI-class.html#results,Variable Bio.Restriction.Restriction.Bbv12I.results=Bio.Restriction.Restriction.Bbv12I-class.html#results,Variable Bio.Restriction.Restriction.BbvCI.results=Bio.Restriction.Restriction.BbvCI-class.html#results,Variable Bio.Restriction.Restriction.BbvI.results=Bio.Restriction.Restriction.BbvI-class.html#results,Variable Bio.Restriction.Restriction.BbvII.results=Bio.Restriction.Restriction.BbvII-class.html#results,Variable Bio.Restriction.Restriction.BccI.results=Bio.Restriction.Restriction.BccI-class.html#results,Variable Bio.Restriction.Restriction.Bce83I.results=Bio.Restriction.Restriction.Bce83I-class.html#results,Variable Bio.Restriction.Restriction.BceAI.results=Bio.Restriction.Restriction.BceAI-class.html#results,Variable Bio.Restriction.Restriction.BcefI.results=Bio.Restriction.Restriction.BcefI-class.html#results,Variable Bio.Restriction.Restriction.BcgI.results=Bio.Restriction.Restriction.BcgI-class.html#results,Variable Bio.Restriction.Restriction.BciT130I.results=Bio.Restriction.Restriction.BciT130I-class.html#results,Variable Bio.Restriction.Restriction.BciVI.results=Bio.Restriction.Restriction.BciVI-class.html#results,Variable Bio.Restriction.Restriction.BclI.results=Bio.Restriction.Restriction.BclI-class.html#results,Variable Bio.Restriction.Restriction.BcnI.results=Bio.Restriction.Restriction.BcnI-class.html#results,Variable Bio.Restriction.Restriction.BcoDI.results=Bio.Restriction.Restriction.BcoDI-class.html#results,Variable Bio.Restriction.Restriction.BcuI.results=Bio.Restriction.Restriction.BcuI-class.html#results,Variable Bio.Restriction.Restriction.BdaI.results=Bio.Restriction.Restriction.BdaI-class.html#results,Variable Bio.Restriction.Restriction.BetI.results=Bio.Restriction.Restriction.BetI-class.html#results,Variable Bio.Restriction.Restriction.BfaI.results=Bio.Restriction.Restriction.BfaI-class.html#results,Variable Bio.Restriction.Restriction.BfiI.results=Bio.Restriction.Restriction.BfiI-class.html#results,Variable Bio.Restriction.Restriction.BfmI.results=Bio.Restriction.Restriction.BfmI-class.html#results,Variable Bio.Restriction.Restriction.BfoI.results=Bio.Restriction.Restriction.BfoI-class.html#results,Variable Bio.Restriction.Restriction.BfrI.results=Bio.Restriction.Restriction.BfrI-class.html#results,Variable Bio.Restriction.Restriction.BfuAI.results=Bio.Restriction.Restriction.BfuAI-class.html#results,Variable Bio.Restriction.Restriction.BfuCI.results=Bio.Restriction.Restriction.BfuCI-class.html#results,Variable Bio.Restriction.Restriction.BfuI.results=Bio.Restriction.Restriction.BfuI-class.html#results,Variable Bio.Restriction.Restriction.BglI.results=Bio.Restriction.Restriction.BglI-class.html#results,Variable Bio.Restriction.Restriction.BglII.results=Bio.Restriction.Restriction.BglII-class.html#results,Variable Bio.Restriction.Restriction.BinI.results=Bio.Restriction.Restriction.BinI-class.html#results,Variable Bio.Restriction.Restriction.BisI.results=Bio.Restriction.Restriction.BisI-class.html#results,Variable Bio.Restriction.Restriction.BlnI.results=Bio.Restriction.Restriction.BlnI-class.html#results,Variable Bio.Restriction.Restriction.BlpI.results=Bio.Restriction.Restriction.BlpI-class.html#results,Variable Bio.Restriction.Restriction.BlsI.results=Bio.Restriction.Restriction.BlsI-class.html#results,Variable Bio.Restriction.Restriction.BmcAI.results=Bio.Restriction.Restriction.BmcAI-class.html#results,Variable Bio.Restriction.Restriction.Bme1390I.results=Bio.Restriction.Restriction.Bme1390I-class.html#results,Variable Bio.Restriction.Restriction.Bme18I.results=Bio.Restriction.Restriction.Bme18I-class.html#results,Variable Bio.Restriction.Restriction.BmeDI.results=Bio.Restriction.Restriction.BmeDI-class.html#results,Variable Bio.Restriction.Restriction.BmeRI.results=Bio.Restriction.Restriction.BmeRI-class.html#results,Variable Bio.Restriction.Restriction.BmeT110I.results=Bio.Restriction.Restriction.BmeT110I-class.html#results,Variable Bio.Restriction.Restriction.BmgBI.results=Bio.Restriction.Restriction.BmgBI-class.html#results,Variable Bio.Restriction.Restriction.BmgI.results=Bio.Restriction.Restriction.BmgI-class.html#results,Variable Bio.Restriction.Restriction.BmgT120I.results=Bio.Restriction.Restriction.BmgT120I-class.html#results,Variable Bio.Restriction.Restriction.BmiI.results=Bio.Restriction.Restriction.BmiI-class.html#results,Variable Bio.Restriction.Restriction.BmrFI.results=Bio.Restriction.Restriction.BmrFI-class.html#results,Variable Bio.Restriction.Restriction.BmrI.results=Bio.Restriction.Restriction.BmrI-class.html#results,Variable Bio.Restriction.Restriction.BmsI.results=Bio.Restriction.Restriction.BmsI-class.html#results,Variable Bio.Restriction.Restriction.BmtI.results=Bio.Restriction.Restriction.BmtI-class.html#results,Variable Bio.Restriction.Restriction.BmuI.results=Bio.Restriction.Restriction.BmuI-class.html#results,Variable Bio.Restriction.Restriction.BoxI.results=Bio.Restriction.Restriction.BoxI-class.html#results,Variable Bio.Restriction.Restriction.BpiI.results=Bio.Restriction.Restriction.BpiI-class.html#results,Variable Bio.Restriction.Restriction.BplI.results=Bio.Restriction.Restriction.BplI-class.html#results,Variable Bio.Restriction.Restriction.BpmI.results=Bio.Restriction.Restriction.BpmI-class.html#results,Variable Bio.Restriction.Restriction.Bpu10I.results=Bio.Restriction.Restriction.Bpu10I-class.html#results,Variable Bio.Restriction.Restriction.Bpu1102I.results=Bio.Restriction.Restriction.Bpu1102I-class.html#results,Variable Bio.Restriction.Restriction.Bpu14I.results=Bio.Restriction.Restriction.Bpu14I-class.html#results,Variable Bio.Restriction.Restriction.BpuEI.results=Bio.Restriction.Restriction.BpuEI-class.html#results,Variable Bio.Restriction.Restriction.BpuMI.results=Bio.Restriction.Restriction.BpuMI-class.html#results,Variable Bio.Restriction.Restriction.BpvUI.results=Bio.Restriction.Restriction.BpvUI-class.html#results,Variable Bio.Restriction.Restriction.Bsa29I.results=Bio.Restriction.Restriction.Bsa29I-class.html#results,Variable Bio.Restriction.Restriction.BsaAI.results=Bio.Restriction.Restriction.BsaAI-class.html#results,Variable Bio.Restriction.Restriction.BsaBI.results=Bio.Restriction.Restriction.BsaBI-class.html#results,Variable Bio.Restriction.Restriction.BsaHI.results=Bio.Restriction.Restriction.BsaHI-class.html#results,Variable Bio.Restriction.Restriction.BsaI.results=Bio.Restriction.Restriction.BsaI-class.html#results,Variable Bio.Restriction.Restriction.BsaJI.results=Bio.Restriction.Restriction.BsaJI-class.html#results,Variable Bio.Restriction.Restriction.BsaWI.results=Bio.Restriction.Restriction.BsaWI-class.html#results,Variable Bio.Restriction.Restriction.BsaXI.results=Bio.Restriction.Restriction.BsaXI-class.html#results,Variable Bio.Restriction.Restriction.BsbI.results=Bio.Restriction.Restriction.BsbI-class.html#results,Variable Bio.Restriction.Restriction.Bsc4I.results=Bio.Restriction.Restriction.Bsc4I-class.html#results,Variable Bio.Restriction.Restriction.BscAI.results=Bio.Restriction.Restriction.BscAI-class.html#results,Variable Bio.Restriction.Restriction.BscGI.results=Bio.Restriction.Restriction.BscGI-class.html#results,Variable Bio.Restriction.Restriction.Bse118I.results=Bio.Restriction.Restriction.Bse118I-class.html#results,Variable Bio.Restriction.Restriction.Bse1I.results=Bio.Restriction.Restriction.Bse1I-class.html#results,Variable Bio.Restriction.Restriction.Bse21I.results=Bio.Restriction.Restriction.Bse21I-class.html#results,Variable Bio.Restriction.Restriction.Bse3DI.results=Bio.Restriction.Restriction.Bse3DI-class.html#results,Variable Bio.Restriction.Restriction.Bse8I.results=Bio.Restriction.Restriction.Bse8I-class.html#results,Variable Bio.Restriction.Restriction.BseAI.results=Bio.Restriction.Restriction.BseAI-class.html#results,Variable Bio.Restriction.Restriction.BseBI.results=Bio.Restriction.Restriction.BseBI-class.html#results,Variable Bio.Restriction.Restriction.BseCI.results=Bio.Restriction.Restriction.BseCI-class.html#results,Variable Bio.Restriction.Restriction.BseDI.results=Bio.Restriction.Restriction.BseDI-class.html#results,Variable Bio.Restriction.Restriction.BseGI.results=Bio.Restriction.Restriction.BseGI-class.html#results,Variable Bio.Restriction.Restriction.BseJI.results=Bio.Restriction.Restriction.BseJI-class.html#results,Variable Bio.Restriction.Restriction.BseLI.results=Bio.Restriction.Restriction.BseLI-class.html#results,Variable Bio.Restriction.Restriction.BseMI.results=Bio.Restriction.Restriction.BseMI-class.html#results,Variable Bio.Restriction.Restriction.BseMII.results=Bio.Restriction.Restriction.BseMII-class.html#results,Variable Bio.Restriction.Restriction.BseNI.results=Bio.Restriction.Restriction.BseNI-class.html#results,Variable Bio.Restriction.Restriction.BsePI.results=Bio.Restriction.Restriction.BsePI-class.html#results,Variable Bio.Restriction.Restriction.BseRI.results=Bio.Restriction.Restriction.BseRI-class.html#results,Variable Bio.Restriction.Restriction.BseSI.results=Bio.Restriction.Restriction.BseSI-class.html#results,Variable Bio.Restriction.Restriction.BseX3I.results=Bio.Restriction.Restriction.BseX3I-class.html#results,Variable Bio.Restriction.Restriction.BseXI.results=Bio.Restriction.Restriction.BseXI-class.html#results,Variable Bio.Restriction.Restriction.BseYI.results=Bio.Restriction.Restriction.BseYI-class.html#results,Variable Bio.Restriction.Restriction.BsgI.results=Bio.Restriction.Restriction.BsgI-class.html#results,Variable Bio.Restriction.Restriction.Bsh1236I.results=Bio.Restriction.Restriction.Bsh1236I-class.html#results,Variable Bio.Restriction.Restriction.Bsh1285I.results=Bio.Restriction.Restriction.Bsh1285I-class.html#results,Variable Bio.Restriction.Restriction.BshFI.results=Bio.Restriction.Restriction.BshFI-class.html#results,Variable Bio.Restriction.Restriction.BshNI.results=Bio.Restriction.Restriction.BshNI-class.html#results,Variable Bio.Restriction.Restriction.BshTI.results=Bio.Restriction.Restriction.BshTI-class.html#results,Variable Bio.Restriction.Restriction.BshVI.results=Bio.Restriction.Restriction.BshVI-class.html#results,Variable Bio.Restriction.Restriction.BsiEI.results=Bio.Restriction.Restriction.BsiEI-class.html#results,Variable Bio.Restriction.Restriction.BsiHKAI.results=Bio.Restriction.Restriction.BsiHKAI-class.html#results,Variable Bio.Restriction.Restriction.BsiHKCI.results=Bio.Restriction.Restriction.BsiHKCI-class.html#results,Variable Bio.Restriction.Restriction.BsiI.results=Bio.Restriction.Restriction.BsiI-class.html#results,Variable Bio.Restriction.Restriction.BsiSI.results=Bio.Restriction.Restriction.BsiSI-class.html#results,Variable Bio.Restriction.Restriction.BsiWI.results=Bio.Restriction.Restriction.BsiWI-class.html#results,Variable Bio.Restriction.Restriction.BsiYI.results=Bio.Restriction.Restriction.BsiYI-class.html#results,Variable Bio.Restriction.Restriction.BslFI.results=Bio.Restriction.Restriction.BslFI-class.html#results,Variable Bio.Restriction.Restriction.BslI.results=Bio.Restriction.Restriction.BslI-class.html#results,Variable Bio.Restriction.Restriction.BsmAI.results=Bio.Restriction.Restriction.BsmAI-class.html#results,Variable Bio.Restriction.Restriction.BsmBI.results=Bio.Restriction.Restriction.BsmBI-class.html#results,Variable Bio.Restriction.Restriction.BsmFI.results=Bio.Restriction.Restriction.BsmFI-class.html#results,Variable Bio.Restriction.Restriction.BsmI.results=Bio.Restriction.Restriction.BsmI-class.html#results,Variable Bio.Restriction.Restriction.BsnI.results=Bio.Restriction.Restriction.BsnI-class.html#results,Variable Bio.Restriction.Restriction.Bso31I.results=Bio.Restriction.Restriction.Bso31I-class.html#results,Variable Bio.Restriction.Restriction.BsoBI.results=Bio.Restriction.Restriction.BsoBI-class.html#results,Variable Bio.Restriction.Restriction.Bsp119I.results=Bio.Restriction.Restriction.Bsp119I-class.html#results,Variable Bio.Restriction.Restriction.Bsp120I.results=Bio.Restriction.Restriction.Bsp120I-class.html#results,Variable Bio.Restriction.Restriction.Bsp1286I.results=Bio.Restriction.Restriction.Bsp1286I-class.html#results,Variable Bio.Restriction.Restriction.Bsp13I.results=Bio.Restriction.Restriction.Bsp13I-class.html#results,Variable Bio.Restriction.Restriction.Bsp1407I.results=Bio.Restriction.Restriction.Bsp1407I-class.html#results,Variable Bio.Restriction.Restriction.Bsp143I.results=Bio.Restriction.Restriction.Bsp143I-class.html#results,Variable Bio.Restriction.Restriction.Bsp1720I.results=Bio.Restriction.Restriction.Bsp1720I-class.html#results,Variable Bio.Restriction.Restriction.Bsp19I.results=Bio.Restriction.Restriction.Bsp19I-class.html#results,Variable Bio.Restriction.Restriction.Bsp24I.results=Bio.Restriction.Restriction.Bsp24I-class.html#results,Variable Bio.Restriction.Restriction.Bsp68I.results=Bio.Restriction.Restriction.Bsp68I-class.html#results,Variable Bio.Restriction.Restriction.BspACI.results=Bio.Restriction.Restriction.BspACI-class.html#results,Variable Bio.Restriction.Restriction.BspCNI.results=Bio.Restriction.Restriction.BspCNI-class.html#results,Variable Bio.Restriction.Restriction.BspD6I.results=Bio.Restriction.Restriction.BspD6I-class.html#results,Variable Bio.Restriction.Restriction.BspDI.results=Bio.Restriction.Restriction.BspDI-class.html#results,Variable Bio.Restriction.Restriction.BspEI.results=Bio.Restriction.Restriction.BspEI-class.html#results,Variable Bio.Restriction.Restriction.BspFNI.results=Bio.Restriction.Restriction.BspFNI-class.html#results,Variable Bio.Restriction.Restriction.BspGI.results=Bio.Restriction.Restriction.BspGI-class.html#results,Variable Bio.Restriction.Restriction.BspHI.results=Bio.Restriction.Restriction.BspHI-class.html#results,Variable Bio.Restriction.Restriction.BspLI.results=Bio.Restriction.Restriction.BspLI-class.html#results,Variable Bio.Restriction.Restriction.BspLU11I.results=Bio.Restriction.Restriction.BspLU11I-class.html#results,Variable Bio.Restriction.Restriction.BspMI.results=Bio.Restriction.Restriction.BspMI-class.html#results,Variable Bio.Restriction.Restriction.BspMII.results=Bio.Restriction.Restriction.BspMII-class.html#results,Variable Bio.Restriction.Restriction.BspNCI.results=Bio.Restriction.Restriction.BspNCI-class.html#results,Variable Bio.Restriction.Restriction.BspOI.results=Bio.Restriction.Restriction.BspOI-class.html#results,Variable Bio.Restriction.Restriction.BspPI.results=Bio.Restriction.Restriction.BspPI-class.html#results,Variable Bio.Restriction.Restriction.BspQI.results=Bio.Restriction.Restriction.BspQI-class.html#results,Variable Bio.Restriction.Restriction.BspT104I.results=Bio.Restriction.Restriction.BspT104I-class.html#results,Variable Bio.Restriction.Restriction.BspT107I.results=Bio.Restriction.Restriction.BspT107I-class.html#results,Variable Bio.Restriction.Restriction.BspTI.results=Bio.Restriction.Restriction.BspTI-class.html#results,Variable Bio.Restriction.Restriction.BsrBI.results=Bio.Restriction.Restriction.BsrBI-class.html#results,Variable Bio.Restriction.Restriction.BsrDI.results=Bio.Restriction.Restriction.BsrDI-class.html#results,Variable Bio.Restriction.Restriction.BsrFI.results=Bio.Restriction.Restriction.BsrFI-class.html#results,Variable Bio.Restriction.Restriction.BsrGI.results=Bio.Restriction.Restriction.BsrGI-class.html#results,Variable Bio.Restriction.Restriction.BsrI.results=Bio.Restriction.Restriction.BsrI-class.html#results,Variable Bio.Restriction.Restriction.BsrSI.results=Bio.Restriction.Restriction.BsrSI-class.html#results,Variable Bio.Restriction.Restriction.BssAI.results=Bio.Restriction.Restriction.BssAI-class.html#results,Variable Bio.Restriction.Restriction.BssECI.results=Bio.Restriction.Restriction.BssECI-class.html#results,Variable Bio.Restriction.Restriction.BssHII.results=Bio.Restriction.Restriction.BssHII-class.html#results,Variable Bio.Restriction.Restriction.BssKI.results=Bio.Restriction.Restriction.BssKI-class.html#results,Variable Bio.Restriction.Restriction.BssMI.results=Bio.Restriction.Restriction.BssMI-class.html#results,Variable Bio.Restriction.Restriction.BssNAI.results=Bio.Restriction.Restriction.BssNAI-class.html#results,Variable Bio.Restriction.Restriction.BssNI.results=Bio.Restriction.Restriction.BssNI-class.html#results,Variable Bio.Restriction.Restriction.BssSI.results=Bio.Restriction.Restriction.BssSI-class.html#results,Variable Bio.Restriction.Restriction.BssT1I.results=Bio.Restriction.Restriction.BssT1I-class.html#results,Variable Bio.Restriction.Restriction.Bst1107I.results=Bio.Restriction.Restriction.Bst1107I-class.html#results,Variable Bio.Restriction.Restriction.Bst2BI.results=Bio.Restriction.Restriction.Bst2BI-class.html#results,Variable Bio.Restriction.Restriction.Bst2UI.results=Bio.Restriction.Restriction.Bst2UI-class.html#results,Variable Bio.Restriction.Restriction.Bst4CI.results=Bio.Restriction.Restriction.Bst4CI-class.html#results,Variable Bio.Restriction.Restriction.Bst6I.results=Bio.Restriction.Restriction.Bst6I-class.html#results,Variable Bio.Restriction.Restriction.BstACI.results=Bio.Restriction.Restriction.BstACI-class.html#results,Variable Bio.Restriction.Restriction.BstAFI.results=Bio.Restriction.Restriction.BstAFI-class.html#results,Variable Bio.Restriction.Restriction.BstAPI.results=Bio.Restriction.Restriction.BstAPI-class.html#results,Variable Bio.Restriction.Restriction.BstAUI.results=Bio.Restriction.Restriction.BstAUI-class.html#results,Variable Bio.Restriction.Restriction.BstBAI.results=Bio.Restriction.Restriction.BstBAI-class.html#results,Variable Bio.Restriction.Restriction.BstBI.results=Bio.Restriction.Restriction.BstBI-class.html#results,Variable Bio.Restriction.Restriction.BstC8I.results=Bio.Restriction.Restriction.BstC8I-class.html#results,Variable Bio.Restriction.Restriction.BstDEI.results=Bio.Restriction.Restriction.BstDEI-class.html#results,Variable Bio.Restriction.Restriction.BstDSI.results=Bio.Restriction.Restriction.BstDSI-class.html#results,Variable Bio.Restriction.Restriction.BstEII.results=Bio.Restriction.Restriction.BstEII-class.html#results,Variable Bio.Restriction.Restriction.BstENI.results=Bio.Restriction.Restriction.BstENI-class.html#results,Variable Bio.Restriction.Restriction.BstF5I.results=Bio.Restriction.Restriction.BstF5I-class.html#results,Variable Bio.Restriction.Restriction.BstFNI.results=Bio.Restriction.Restriction.BstFNI-class.html#results,Variable Bio.Restriction.Restriction.BstH2I.results=Bio.Restriction.Restriction.BstH2I-class.html#results,Variable Bio.Restriction.Restriction.BstHHI.results=Bio.Restriction.Restriction.BstHHI-class.html#results,Variable Bio.Restriction.Restriction.BstKTI.results=Bio.Restriction.Restriction.BstKTI-class.html#results,Variable Bio.Restriction.Restriction.BstMAI.results=Bio.Restriction.Restriction.BstMAI-class.html#results,Variable Bio.Restriction.Restriction.BstMBI.results=Bio.Restriction.Restriction.BstMBI-class.html#results,Variable Bio.Restriction.Restriction.BstMCI.results=Bio.Restriction.Restriction.BstMCI-class.html#results,Variable Bio.Restriction.Restriction.BstMWI.results=Bio.Restriction.Restriction.BstMWI-class.html#results,Variable Bio.Restriction.Restriction.BstNI.results=Bio.Restriction.Restriction.BstNI-class.html#results,Variable Bio.Restriction.Restriction.BstNSI.results=Bio.Restriction.Restriction.BstNSI-class.html#results,Variable Bio.Restriction.Restriction.BstOI.results=Bio.Restriction.Restriction.BstOI-class.html#results,Variable Bio.Restriction.Restriction.BstPAI.results=Bio.Restriction.Restriction.BstPAI-class.html#results,Variable Bio.Restriction.Restriction.BstPI.results=Bio.Restriction.Restriction.BstPI-class.html#results,Variable Bio.Restriction.Restriction.BstSCI.results=Bio.Restriction.Restriction.BstSCI-class.html#results,Variable Bio.Restriction.Restriction.BstSFI.results=Bio.Restriction.Restriction.BstSFI-class.html#results,Variable Bio.Restriction.Restriction.BstSLI.results=Bio.Restriction.Restriction.BstSLI-class.html#results,Variable Bio.Restriction.Restriction.BstSNI.results=Bio.Restriction.Restriction.BstSNI-class.html#results,Variable Bio.Restriction.Restriction.BstUI.results=Bio.Restriction.Restriction.BstUI-class.html#results,Variable Bio.Restriction.Restriction.BstV1I.results=Bio.Restriction.Restriction.BstV1I-class.html#results,Variable Bio.Restriction.Restriction.BstV2I.results=Bio.Restriction.Restriction.BstV2I-class.html#results,Variable Bio.Restriction.Restriction.BstX2I.results=Bio.Restriction.Restriction.BstX2I-class.html#results,Variable Bio.Restriction.Restriction.BstXI.results=Bio.Restriction.Restriction.BstXI-class.html#results,Variable Bio.Restriction.Restriction.BstYI.results=Bio.Restriction.Restriction.BstYI-class.html#results,Variable Bio.Restriction.Restriction.BstZ17I.results=Bio.Restriction.Restriction.BstZ17I-class.html#results,Variable Bio.Restriction.Restriction.BstZI.results=Bio.Restriction.Restriction.BstZI-class.html#results,Variable Bio.Restriction.Restriction.Bsu15I.results=Bio.Restriction.Restriction.Bsu15I-class.html#results,Variable Bio.Restriction.Restriction.Bsu36I.results=Bio.Restriction.Restriction.Bsu36I-class.html#results,Variable Bio.Restriction.Restriction.BsuI.results=Bio.Restriction.Restriction.BsuI-class.html#results,Variable Bio.Restriction.Restriction.BsuRI.results=Bio.Restriction.Restriction.BsuRI-class.html#results,Variable Bio.Restriction.Restriction.BtgI.results=Bio.Restriction.Restriction.BtgI-class.html#results,Variable Bio.Restriction.Restriction.BtgZI.results=Bio.Restriction.Restriction.BtgZI-class.html#results,Variable Bio.Restriction.Restriction.BthCI.results=Bio.Restriction.Restriction.BthCI-class.html#results,Variable Bio.Restriction.Restriction.BtrI.results=Bio.Restriction.Restriction.BtrI-class.html#results,Variable Bio.Restriction.Restriction.BtsCI.results=Bio.Restriction.Restriction.BtsCI-class.html#results,Variable Bio.Restriction.Restriction.BtsI.results=Bio.Restriction.Restriction.BtsI-class.html#results,Variable Bio.Restriction.Restriction.BtsIMutI.results=Bio.Restriction.Restriction.BtsIMutI-class.html#results,Variable Bio.Restriction.Restriction.BtuMI.results=Bio.Restriction.Restriction.BtuMI-class.html#results,Variable Bio.Restriction.Restriction.BveI.results=Bio.Restriction.Restriction.BveI-class.html#results,Variable Bio.Restriction.Restriction.Cac8I.results=Bio.Restriction.Restriction.Cac8I-class.html#results,Variable Bio.Restriction.Restriction.CaiI.results=Bio.Restriction.Restriction.CaiI-class.html#results,Variable Bio.Restriction.Restriction.CauII.results=Bio.Restriction.Restriction.CauII-class.html#results,Variable Bio.Restriction.Restriction.CchII.results=Bio.Restriction.Restriction.CchII-class.html#results,Variable Bio.Restriction.Restriction.CchIII.results=Bio.Restriction.Restriction.CchIII-class.html#results,Variable Bio.Restriction.Restriction.CciI.results=Bio.Restriction.Restriction.CciI-class.html#results,Variable Bio.Restriction.Restriction.CciNI.results=Bio.Restriction.Restriction.CciNI-class.html#results,Variable Bio.Restriction.Restriction.Cdi630V.results=Bio.Restriction.Restriction.Cdi630V-class.html#results,Variable Bio.Restriction.Restriction.CdiI.results=Bio.Restriction.Restriction.CdiI-class.html#results,Variable Bio.Restriction.Restriction.CdpI.results=Bio.Restriction.Restriction.CdpI-class.html#results,Variable Bio.Restriction.Restriction.CfoI.results=Bio.Restriction.Restriction.CfoI-class.html#results,Variable Bio.Restriction.Restriction.Cfr10I.results=Bio.Restriction.Restriction.Cfr10I-class.html#results,Variable Bio.Restriction.Restriction.Cfr13I.results=Bio.Restriction.Restriction.Cfr13I-class.html#results,Variable Bio.Restriction.Restriction.Cfr42I.results=Bio.Restriction.Restriction.Cfr42I-class.html#results,Variable Bio.Restriction.Restriction.Cfr9I.results=Bio.Restriction.Restriction.Cfr9I-class.html#results,Variable Bio.Restriction.Restriction.CfrI.results=Bio.Restriction.Restriction.CfrI-class.html#results,Variable Bio.Restriction.Restriction.Cgl13032I.results=Bio.Restriction.Restriction.Cgl13032I-class.html#results,Variable Bio.Restriction.Restriction.Cgl13032II.results=Bio.Restriction.Restriction.Cgl13032II-class.html#results,Variable Bio.Restriction.Restriction.ChaI.results=Bio.Restriction.Restriction.ChaI-class.html#results,Variable Bio.Restriction.Restriction.CjeFIII.results=Bio.Restriction.Restriction.CjeFIII-class.html#results,Variable Bio.Restriction.Restriction.CjeFV.results=Bio.Restriction.Restriction.CjeFV-class.html#results,Variable Bio.Restriction.Restriction.CjeI.results=Bio.Restriction.Restriction.CjeI-class.html#results,Variable Bio.Restriction.Restriction.CjeNII.results=Bio.Restriction.Restriction.CjeNII-class.html#results,Variable Bio.Restriction.Restriction.CjeNIII.results=Bio.Restriction.Restriction.CjeNIII-class.html#results,Variable Bio.Restriction.Restriction.CjeP659IV.results=Bio.Restriction.Restriction.CjeP659IV-class.html#results,Variable Bio.Restriction.Restriction.CjePI.results=Bio.Restriction.Restriction.CjePI-class.html#results,Variable Bio.Restriction.Restriction.CjuI.results=Bio.Restriction.Restriction.CjuI-class.html#results,Variable Bio.Restriction.Restriction.CjuII.results=Bio.Restriction.Restriction.CjuII-class.html#results,Variable Bio.Restriction.Restriction.ClaI.results=Bio.Restriction.Restriction.ClaI-class.html#results,Variable Bio.Restriction.Restriction.CpoI.results=Bio.Restriction.Restriction.CpoI-class.html#results,Variable Bio.Restriction.Restriction.CseI.results=Bio.Restriction.Restriction.CseI-class.html#results,Variable Bio.Restriction.Restriction.CsiI.results=Bio.Restriction.Restriction.CsiI-class.html#results,Variable Bio.Restriction.Restriction.Csp6I.results=Bio.Restriction.Restriction.Csp6I-class.html#results,Variable Bio.Restriction.Restriction.CspAI.results=Bio.Restriction.Restriction.CspAI-class.html#results,Variable Bio.Restriction.Restriction.CspCI.results=Bio.Restriction.Restriction.CspCI-class.html#results,Variable Bio.Restriction.Restriction.CspI.results=Bio.Restriction.Restriction.CspI-class.html#results,Variable Bio.Restriction.Restriction.CstMI.results=Bio.Restriction.Restriction.CstMI-class.html#results,Variable Bio.Restriction.Restriction.CviAII.results=Bio.Restriction.Restriction.CviAII-class.html#results,Variable Bio.Restriction.Restriction.CviJI.results=Bio.Restriction.Restriction.CviJI-class.html#results,Variable Bio.Restriction.Restriction.CviKI_1.results=Bio.Restriction.Restriction.CviKI_1-class.html#results,Variable Bio.Restriction.Restriction.CviQI.results=Bio.Restriction.Restriction.CviQI-class.html#results,Variable Bio.Restriction.Restriction.CviRI.results=Bio.Restriction.Restriction.CviRI-class.html#results,Variable Bio.Restriction.Restriction.DdeI.results=Bio.Restriction.Restriction.DdeI-class.html#results,Variable Bio.Restriction.Restriction.DinI.results=Bio.Restriction.Restriction.DinI-class.html#results,Variable Bio.Restriction.Restriction.DpnI.results=Bio.Restriction.Restriction.DpnI-class.html#results,Variable Bio.Restriction.Restriction.DpnII.results=Bio.Restriction.Restriction.DpnII-class.html#results,Variable Bio.Restriction.Restriction.DraI.results=Bio.Restriction.Restriction.DraI-class.html#results,Variable Bio.Restriction.Restriction.DraII.results=Bio.Restriction.Restriction.DraII-class.html#results,Variable Bio.Restriction.Restriction.DraIII.results=Bio.Restriction.Restriction.DraIII-class.html#results,Variable Bio.Restriction.Restriction.DraRI.results=Bio.Restriction.Restriction.DraRI-class.html#results,Variable Bio.Restriction.Restriction.DrdI.results=Bio.Restriction.Restriction.DrdI-class.html#results,Variable Bio.Restriction.Restriction.DrdII.results=Bio.Restriction.Restriction.DrdII-class.html#results,Variable Bio.Restriction.Restriction.DriI.results=Bio.Restriction.Restriction.DriI-class.html#results,Variable Bio.Restriction.Restriction.DsaI.results=Bio.Restriction.Restriction.DsaI-class.html#results,Variable Bio.Restriction.Restriction.DseDI.results=Bio.Restriction.Restriction.DseDI-class.html#results,Variable Bio.Restriction.Restriction.EaeI.results=Bio.Restriction.Restriction.EaeI-class.html#results,Variable Bio.Restriction.Restriction.EagI.results=Bio.Restriction.Restriction.EagI-class.html#results,Variable Bio.Restriction.Restriction.Eam1104I.results=Bio.Restriction.Restriction.Eam1104I-class.html#results,Variable Bio.Restriction.Restriction.Eam1105I.results=Bio.Restriction.Restriction.Eam1105I-class.html#results,Variable Bio.Restriction.Restriction.EarI.results=Bio.Restriction.Restriction.EarI-class.html#results,Variable Bio.Restriction.Restriction.EciI.results=Bio.Restriction.Restriction.EciI-class.html#results,Variable Bio.Restriction.Restriction.Ecl136II.results=Bio.Restriction.Restriction.Ecl136II-class.html#results,Variable Bio.Restriction.Restriction.EclXI.results=Bio.Restriction.Restriction.EclXI-class.html#results,Variable Bio.Restriction.Restriction.Eco105I.results=Bio.Restriction.Restriction.Eco105I-class.html#results,Variable Bio.Restriction.Restriction.Eco130I.results=Bio.Restriction.Restriction.Eco130I-class.html#results,Variable Bio.Restriction.Restriction.Eco147I.results=Bio.Restriction.Restriction.Eco147I-class.html#results,Variable Bio.Restriction.Restriction.Eco24I.results=Bio.Restriction.Restriction.Eco24I-class.html#results,Variable Bio.Restriction.Restriction.Eco31I.results=Bio.Restriction.Restriction.Eco31I-class.html#results,Variable Bio.Restriction.Restriction.Eco32I.results=Bio.Restriction.Restriction.Eco32I-class.html#results,Variable Bio.Restriction.Restriction.Eco47I.results=Bio.Restriction.Restriction.Eco47I-class.html#results,Variable Bio.Restriction.Restriction.Eco47III.results=Bio.Restriction.Restriction.Eco47III-class.html#results,Variable Bio.Restriction.Restriction.Eco52I.results=Bio.Restriction.Restriction.Eco52I-class.html#results,Variable Bio.Restriction.Restriction.Eco53kI.results=Bio.Restriction.Restriction.Eco53kI-class.html#results,Variable Bio.Restriction.Restriction.Eco57I.results=Bio.Restriction.Restriction.Eco57I-class.html#results,Variable Bio.Restriction.Restriction.Eco57MI.results=Bio.Restriction.Restriction.Eco57MI-class.html#results,Variable Bio.Restriction.Restriction.Eco72I.results=Bio.Restriction.Restriction.Eco72I-class.html#results,Variable Bio.Restriction.Restriction.Eco81I.results=Bio.Restriction.Restriction.Eco81I-class.html#results,Variable Bio.Restriction.Restriction.Eco88I.results=Bio.Restriction.Restriction.Eco88I-class.html#results,Variable Bio.Restriction.Restriction.Eco91I.results=Bio.Restriction.Restriction.Eco91I-class.html#results,Variable Bio.Restriction.Restriction.EcoHI.results=Bio.Restriction.Restriction.EcoHI-class.html#results,Variable Bio.Restriction.Restriction.EcoICRI.results=Bio.Restriction.Restriction.EcoICRI-class.html#results,Variable Bio.Restriction.Restriction.EcoNI.results=Bio.Restriction.Restriction.EcoNI-class.html#results,Variable Bio.Restriction.Restriction.EcoO109I.results=Bio.Restriction.Restriction.EcoO109I-class.html#results,Variable Bio.Restriction.Restriction.EcoO65I.results=Bio.Restriction.Restriction.EcoO65I-class.html#results,Variable Bio.Restriction.Restriction.EcoRI.results=Bio.Restriction.Restriction.EcoRI-class.html#results,Variable Bio.Restriction.Restriction.EcoRII.results=Bio.Restriction.Restriction.EcoRII-class.html#results,Variable Bio.Restriction.Restriction.EcoRV.results=Bio.Restriction.Restriction.EcoRV-class.html#results,Variable Bio.Restriction.Restriction.EcoT14I.results=Bio.Restriction.Restriction.EcoT14I-class.html#results,Variable Bio.Restriction.Restriction.EcoT22I.results=Bio.Restriction.Restriction.EcoT22I-class.html#results,Variable Bio.Restriction.Restriction.EcoT38I.results=Bio.Restriction.Restriction.EcoT38I-class.html#results,Variable Bio.Restriction.Restriction.EgeI.results=Bio.Restriction.Restriction.EgeI-class.html#results,Variable Bio.Restriction.Restriction.EheI.results=Bio.Restriction.Restriction.EheI-class.html#results,Variable Bio.Restriction.Restriction.ErhI.results=Bio.Restriction.Restriction.ErhI-class.html#results,Variable Bio.Restriction.Restriction.EsaBC3I.results=Bio.Restriction.Restriction.EsaBC3I-class.html#results,Variable Bio.Restriction.Restriction.EsaSSI.results=Bio.Restriction.Restriction.EsaSSI-class.html#results,Variable Bio.Restriction.Restriction.Esp3I.results=Bio.Restriction.Restriction.Esp3I-class.html#results,Variable Bio.Restriction.Restriction.EspI.results=Bio.Restriction.Restriction.EspI-class.html#results,Variable Bio.Restriction.Restriction.FaeI.results=Bio.Restriction.Restriction.FaeI-class.html#results,Variable Bio.Restriction.Restriction.FaiI.results=Bio.Restriction.Restriction.FaiI-class.html#results,Variable Bio.Restriction.Restriction.FalI.results=Bio.Restriction.Restriction.FalI-class.html#results,Variable Bio.Restriction.Restriction.FaqI.results=Bio.Restriction.Restriction.FaqI-class.html#results,Variable Bio.Restriction.Restriction.FatI.results=Bio.Restriction.Restriction.FatI-class.html#results,Variable Bio.Restriction.Restriction.FauI.results=Bio.Restriction.Restriction.FauI-class.html#results,Variable Bio.Restriction.Restriction.FauNDI.results=Bio.Restriction.Restriction.FauNDI-class.html#results,Variable Bio.Restriction.Restriction.FbaI.results=Bio.Restriction.Restriction.FbaI-class.html#results,Variable Bio.Restriction.Restriction.FblI.results=Bio.Restriction.Restriction.FblI-class.html#results,Variable Bio.Restriction.Restriction.FinI.results=Bio.Restriction.Restriction.FinI-class.html#results,Variable Bio.Restriction.Restriction.FmuI.results=Bio.Restriction.Restriction.FmuI-class.html#results,Variable Bio.Restriction.Restriction.Fnu4HI.results=Bio.Restriction.Restriction.Fnu4HI-class.html#results,Variable Bio.Restriction.Restriction.FnuDII.results=Bio.Restriction.Restriction.FnuDII-class.html#results,Variable Bio.Restriction.Restriction.FokI.results=Bio.Restriction.Restriction.FokI-class.html#results,Variable Bio.Restriction.Restriction.FriOI.results=Bio.Restriction.Restriction.FriOI-class.html#results,Variable Bio.Restriction.Restriction.FseI.results=Bio.Restriction.Restriction.FseI-class.html#results,Variable Bio.Restriction.Restriction.Fsp4HI.results=Bio.Restriction.Restriction.Fsp4HI-class.html#results,Variable Bio.Restriction.Restriction.FspAI.results=Bio.Restriction.Restriction.FspAI-class.html#results,Variable Bio.Restriction.Restriction.FspBI.results=Bio.Restriction.Restriction.FspBI-class.html#results,Variable Bio.Restriction.Restriction.FspEI.results=Bio.Restriction.Restriction.FspEI-class.html#results,Variable Bio.Restriction.Restriction.FspI.results=Bio.Restriction.Restriction.FspI-class.html#results,Variable Bio.Restriction.Restriction.GauT27I.results=Bio.Restriction.Restriction.GauT27I-class.html#results,Variable Bio.Restriction.Restriction.GdiII.results=Bio.Restriction.Restriction.GdiII-class.html#results,Variable Bio.Restriction.Restriction.GlaI.results=Bio.Restriction.Restriction.GlaI-class.html#results,Variable Bio.Restriction.Restriction.GluI.results=Bio.Restriction.Restriction.GluI-class.html#results,Variable Bio.Restriction.Restriction.GsaI.results=Bio.Restriction.Restriction.GsaI-class.html#results,Variable Bio.Restriction.Restriction.GsuI.results=Bio.Restriction.Restriction.GsuI-class.html#results,Variable Bio.Restriction.Restriction.HaeI.results=Bio.Restriction.Restriction.HaeI-class.html#results,Variable Bio.Restriction.Restriction.HaeII.results=Bio.Restriction.Restriction.HaeII-class.html#results,Variable Bio.Restriction.Restriction.HaeIII.results=Bio.Restriction.Restriction.HaeIII-class.html#results,Variable Bio.Restriction.Restriction.HapII.results=Bio.Restriction.Restriction.HapII-class.html#results,Variable Bio.Restriction.Restriction.HauII.results=Bio.Restriction.Restriction.HauII-class.html#results,Variable Bio.Restriction.Restriction.HgaI.results=Bio.Restriction.Restriction.HgaI-class.html#results,Variable Bio.Restriction.Restriction.HgiAI.results=Bio.Restriction.Restriction.HgiAI-class.html#results,Variable Bio.Restriction.Restriction.HgiCI.results=Bio.Restriction.Restriction.HgiCI-class.html#results,Variable Bio.Restriction.Restriction.HgiEII.results=Bio.Restriction.Restriction.HgiEII-class.html#results,Variable Bio.Restriction.Restriction.HgiJII.results=Bio.Restriction.Restriction.HgiJII-class.html#results,Variable Bio.Restriction.Restriction.HhaI.results=Bio.Restriction.Restriction.HhaI-class.html#results,Variable Bio.Restriction.Restriction.Hin1I.results=Bio.Restriction.Restriction.Hin1I-class.html#results,Variable Bio.Restriction.Restriction.Hin1II.results=Bio.Restriction.Restriction.Hin1II-class.html#results,Variable Bio.Restriction.Restriction.Hin4I.results=Bio.Restriction.Restriction.Hin4I-class.html#results,Variable Bio.Restriction.Restriction.Hin4II.results=Bio.Restriction.Restriction.Hin4II-class.html#results,Variable Bio.Restriction.Restriction.Hin6I.results=Bio.Restriction.Restriction.Hin6I-class.html#results,Variable Bio.Restriction.Restriction.HinP1I.results=Bio.Restriction.Restriction.HinP1I-class.html#results,Variable Bio.Restriction.Restriction.HincII.results=Bio.Restriction.Restriction.HincII-class.html#results,Variable Bio.Restriction.Restriction.HindII.results=Bio.Restriction.Restriction.HindII-class.html#results,Variable Bio.Restriction.Restriction.HindIII.results=Bio.Restriction.Restriction.HindIII-class.html#results,Variable Bio.Restriction.Restriction.HinfI.results=Bio.Restriction.Restriction.HinfI-class.html#results,Variable Bio.Restriction.Restriction.HpaI.results=Bio.Restriction.Restriction.HpaI-class.html#results,Variable Bio.Restriction.Restriction.HpaII.results=Bio.Restriction.Restriction.HpaII-class.html#results,Variable Bio.Restriction.Restriction.HphI.results=Bio.Restriction.Restriction.HphI-class.html#results,Variable Bio.Restriction.Restriction.Hpy166II.results=Bio.Restriction.Restriction.Hpy166II-class.html#results,Variable Bio.Restriction.Restriction.Hpy178III.results=Bio.Restriction.Restriction.Hpy178III-class.html#results,Variable Bio.Restriction.Restriction.Hpy188I.results=Bio.Restriction.Restriction.Hpy188I-class.html#results,Variable Bio.Restriction.Restriction.Hpy188III.results=Bio.Restriction.Restriction.Hpy188III-class.html#results,Variable Bio.Restriction.Restriction.Hpy8I.results=Bio.Restriction.Restriction.Hpy8I-class.html#results,Variable Bio.Restriction.Restriction.Hpy99I.results=Bio.Restriction.Restriction.Hpy99I-class.html#results,Variable Bio.Restriction.Restriction.Hpy99XIII.results=Bio.Restriction.Restriction.Hpy99XIII-class.html#results,Variable Bio.Restriction.Restriction.Hpy99XIV.results=Bio.Restriction.Restriction.Hpy99XIV-class.html#results,Variable Bio.Restriction.Restriction.HpyAV.results=Bio.Restriction.Restriction.HpyAV-class.html#results,Variable Bio.Restriction.Restriction.HpyCH4III.results=Bio.Restriction.Restriction.HpyCH4III-class.html#results,Variable Bio.Restriction.Restriction.HpyCH4IV.results=Bio.Restriction.Restriction.HpyCH4IV-class.html#results,Variable Bio.Restriction.Restriction.HpyCH4V.results=Bio.Restriction.Restriction.HpyCH4V-class.html#results,Variable Bio.Restriction.Restriction.HpyF10VI.results=Bio.Restriction.Restriction.HpyF10VI-class.html#results,Variable Bio.Restriction.Restriction.HpyF3I.results=Bio.Restriction.Restriction.HpyF3I-class.html#results,Variable Bio.Restriction.Restriction.HpySE526I.results=Bio.Restriction.Restriction.HpySE526I-class.html#results,Variable Bio.Restriction.Restriction.Hsp92I.results=Bio.Restriction.Restriction.Hsp92I-class.html#results,Variable Bio.Restriction.Restriction.Hsp92II.results=Bio.Restriction.Restriction.Hsp92II-class.html#results,Variable Bio.Restriction.Restriction.HspAI.results=Bio.Restriction.Restriction.HspAI-class.html#results,Variable Bio.Restriction.Restriction.Jma19592I.results=Bio.Restriction.Restriction.Jma19592I-class.html#results,Variable Bio.Restriction.Restriction.KasI.results=Bio.Restriction.Restriction.KasI-class.html#results,Variable Bio.Restriction.Restriction.KflI.results=Bio.Restriction.Restriction.KflI-class.html#results,Variable Bio.Restriction.Restriction.Kpn2I.results=Bio.Restriction.Restriction.Kpn2I-class.html#results,Variable Bio.Restriction.Restriction.KpnI.results=Bio.Restriction.Restriction.KpnI-class.html#results,Variable Bio.Restriction.Restriction.KroI.results=Bio.Restriction.Restriction.KroI-class.html#results,Variable Bio.Restriction.Restriction.Ksp22I.results=Bio.Restriction.Restriction.Ksp22I-class.html#results,Variable Bio.Restriction.Restriction.Ksp632I.results=Bio.Restriction.Restriction.Ksp632I-class.html#results,Variable Bio.Restriction.Restriction.KspAI.results=Bio.Restriction.Restriction.KspAI-class.html#results,Variable Bio.Restriction.Restriction.KspI.results=Bio.Restriction.Restriction.KspI-class.html#results,Variable Bio.Restriction.Restriction.Kzo9I.results=Bio.Restriction.Restriction.Kzo9I-class.html#results,Variable Bio.Restriction.Restriction.LguI.results=Bio.Restriction.Restriction.LguI-class.html#results,Variable Bio.Restriction.Restriction.LpnI.results=Bio.Restriction.Restriction.LpnI-class.html#results,Variable Bio.Restriction.Restriction.LpnPI.results=Bio.Restriction.Restriction.LpnPI-class.html#results,Variable Bio.Restriction.Restriction.Lsp1109I.results=Bio.Restriction.Restriction.Lsp1109I-class.html#results,Variable Bio.Restriction.Restriction.LweI.results=Bio.Restriction.Restriction.LweI-class.html#results,Variable Bio.Restriction.Restriction.MabI.results=Bio.Restriction.Restriction.MabI-class.html#results,Variable Bio.Restriction.Restriction.MaeI.results=Bio.Restriction.Restriction.MaeI-class.html#results,Variable Bio.Restriction.Restriction.MaeII.results=Bio.Restriction.Restriction.MaeII-class.html#results,Variable Bio.Restriction.Restriction.MaeIII.results=Bio.Restriction.Restriction.MaeIII-class.html#results,Variable Bio.Restriction.Restriction.MalI.results=Bio.Restriction.Restriction.MalI-class.html#results,Variable Bio.Restriction.Restriction.MaqI.results=Bio.Restriction.Restriction.MaqI-class.html#results,Variable Bio.Restriction.Restriction.MauBI.results=Bio.Restriction.Restriction.MauBI-class.html#results,Variable Bio.Restriction.Restriction.MbiI.results=Bio.Restriction.Restriction.MbiI-class.html#results,Variable Bio.Restriction.Restriction.MboI.results=Bio.Restriction.Restriction.MboI-class.html#results,Variable Bio.Restriction.Restriction.MboII.results=Bio.Restriction.Restriction.MboII-class.html#results,Variable Bio.Restriction.Restriction.McaTI.results=Bio.Restriction.Restriction.McaTI-class.html#results,Variable Bio.Restriction.Restriction.McrI.results=Bio.Restriction.Restriction.McrI-class.html#results,Variable Bio.Restriction.Restriction.MfeI.results=Bio.Restriction.Restriction.MfeI-class.html#results,Variable Bio.Restriction.Restriction.MflI.results=Bio.Restriction.Restriction.MflI-class.html#results,Variable Bio.Restriction.Restriction.MhlI.results=Bio.Restriction.Restriction.MhlI-class.html#results,Variable Bio.Restriction.Restriction.MjaIV.results=Bio.Restriction.Restriction.MjaIV-class.html#results,Variable Bio.Restriction.Restriction.MkaDII.results=Bio.Restriction.Restriction.MkaDII-class.html#results,Variable Bio.Restriction.Restriction.MlsI.results=Bio.Restriction.Restriction.MlsI-class.html#results,Variable Bio.Restriction.Restriction.MluCI.results=Bio.Restriction.Restriction.MluCI-class.html#results,Variable Bio.Restriction.Restriction.MluI.results=Bio.Restriction.Restriction.MluI-class.html#results,Variable Bio.Restriction.Restriction.MluNI.results=Bio.Restriction.Restriction.MluNI-class.html#results,Variable Bio.Restriction.Restriction.Mly113I.results=Bio.Restriction.Restriction.Mly113I-class.html#results,Variable Bio.Restriction.Restriction.MlyI.results=Bio.Restriction.Restriction.MlyI-class.html#results,Variable Bio.Restriction.Restriction.MmeI.results=Bio.Restriction.Restriction.MmeI-class.html#results,Variable Bio.Restriction.Restriction.MnlI.results=Bio.Restriction.Restriction.MnlI-class.html#results,Variable Bio.Restriction.Restriction.Mph1103I.results=Bio.Restriction.Restriction.Mph1103I-class.html#results,Variable Bio.Restriction.Restriction.MreI.results=Bio.Restriction.Restriction.MreI-class.html#results,Variable Bio.Restriction.Restriction.MroI.results=Bio.Restriction.Restriction.MroI-class.html#results,Variable Bio.Restriction.Restriction.MroNI.results=Bio.Restriction.Restriction.MroNI-class.html#results,Variable Bio.Restriction.Restriction.MroXI.results=Bio.Restriction.Restriction.MroXI-class.html#results,Variable Bio.Restriction.Restriction.MscI.results=Bio.Restriction.Restriction.MscI-class.html#results,Variable Bio.Restriction.Restriction.MseI.results=Bio.Restriction.Restriction.MseI-class.html#results,Variable Bio.Restriction.Restriction.MslI.results=Bio.Restriction.Restriction.MslI-class.html#results,Variable Bio.Restriction.Restriction.Msp20I.results=Bio.Restriction.Restriction.Msp20I-class.html#results,Variable Bio.Restriction.Restriction.MspA1I.results=Bio.Restriction.Restriction.MspA1I-class.html#results,Variable Bio.Restriction.Restriction.MspCI.results=Bio.Restriction.Restriction.MspCI-class.html#results,Variable Bio.Restriction.Restriction.MspI.results=Bio.Restriction.Restriction.MspI-class.html#results,Variable Bio.Restriction.Restriction.MspJI.results=Bio.Restriction.Restriction.MspJI-class.html#results,Variable Bio.Restriction.Restriction.MspR9I.results=Bio.Restriction.Restriction.MspR9I-class.html#results,Variable Bio.Restriction.Restriction.MssI.results=Bio.Restriction.Restriction.MssI-class.html#results,Variable Bio.Restriction.Restriction.MstI.results=Bio.Restriction.Restriction.MstI-class.html#results,Variable Bio.Restriction.Restriction.MunI.results=Bio.Restriction.Restriction.MunI-class.html#results,Variable Bio.Restriction.Restriction.Mva1269I.results=Bio.Restriction.Restriction.Mva1269I-class.html#results,Variable Bio.Restriction.Restriction.MvaI.results=Bio.Restriction.Restriction.MvaI-class.html#results,Variable Bio.Restriction.Restriction.MvnI.results=Bio.Restriction.Restriction.MvnI-class.html#results,Variable Bio.Restriction.Restriction.MvrI.results=Bio.Restriction.Restriction.MvrI-class.html#results,Variable Bio.Restriction.Restriction.MwoI.results=Bio.Restriction.Restriction.MwoI-class.html#results,Variable Bio.Restriction.Restriction.NaeI.results=Bio.Restriction.Restriction.NaeI-class.html#results,Variable Bio.Restriction.Restriction.NarI.results=Bio.Restriction.Restriction.NarI-class.html#results,Variable Bio.Restriction.Restriction.NciI.results=Bio.Restriction.Restriction.NciI-class.html#results,Variable Bio.Restriction.Restriction.NcoI.results=Bio.Restriction.Restriction.NcoI-class.html#results,Variable Bio.Restriction.Restriction.NdeI.results=Bio.Restriction.Restriction.NdeI-class.html#results,Variable Bio.Restriction.Restriction.NdeII.results=Bio.Restriction.Restriction.NdeII-class.html#results,Variable Bio.Restriction.Restriction.NgoAVIII.results=Bio.Restriction.Restriction.NgoAVIII-class.html#results,Variable Bio.Restriction.Restriction.NgoMIV.results=Bio.Restriction.Restriction.NgoMIV-class.html#results,Variable Bio.Restriction.Restriction.NhaXI.results=Bio.Restriction.Restriction.NhaXI-class.html#results,Variable Bio.Restriction.Restriction.NheI.results=Bio.Restriction.Restriction.NheI-class.html#results,Variable Bio.Restriction.Restriction.NlaCI.results=Bio.Restriction.Restriction.NlaCI-class.html#results,Variable Bio.Restriction.Restriction.NlaIII.results=Bio.Restriction.Restriction.NlaIII-class.html#results,Variable Bio.Restriction.Restriction.NlaIV.results=Bio.Restriction.Restriction.NlaIV-class.html#results,Variable Bio.Restriction.Restriction.Nli3877I.results=Bio.Restriction.Restriction.Nli3877I-class.html#results,Variable Bio.Restriction.Restriction.NmeAIII.results=Bio.Restriction.Restriction.NmeAIII-class.html#results,Variable Bio.Restriction.Restriction.NmeDI.results=Bio.Restriction.Restriction.NmeDI-class.html#results,Variable Bio.Restriction.Restriction.NmuCI.results=Bio.Restriction.Restriction.NmuCI-class.html#results,Variable Bio.Restriction.Restriction.NotI.results=Bio.Restriction.Restriction.NotI-class.html#results,Variable Bio.Restriction.Restriction.NruI.results=Bio.Restriction.Restriction.NruI-class.html#results,Variable Bio.Restriction.Restriction.NsbI.results=Bio.Restriction.Restriction.NsbI-class.html#results,Variable Bio.Restriction.Restriction.NsiI.results=Bio.Restriction.Restriction.NsiI-class.html#results,Variable Bio.Restriction.Restriction.NspBII.results=Bio.Restriction.Restriction.NspBII-class.html#results,Variable Bio.Restriction.Restriction.NspI.results=Bio.Restriction.Restriction.NspI-class.html#results,Variable Bio.Restriction.Restriction.NspV.results=Bio.Restriction.Restriction.NspV-class.html#results,Variable Bio.Restriction.Restriction.OliI.results=Bio.Restriction.Restriction.OliI-class.html#results,Variable Bio.Restriction.Restriction.PabI.results=Bio.Restriction.Restriction.PabI-class.html#results,Variable Bio.Restriction.Restriction.PacI.results=Bio.Restriction.Restriction.PacI-class.html#results,Variable Bio.Restriction.Restriction.PaeI.results=Bio.Restriction.Restriction.PaeI-class.html#results,Variable Bio.Restriction.Restriction.PaeR7I.results=Bio.Restriction.Restriction.PaeR7I-class.html#results,Variable Bio.Restriction.Restriction.PagI.results=Bio.Restriction.Restriction.PagI-class.html#results,Variable Bio.Restriction.Restriction.PalAI.results=Bio.Restriction.Restriction.PalAI-class.html#results,Variable Bio.Restriction.Restriction.PasI.results=Bio.Restriction.Restriction.PasI-class.html#results,Variable Bio.Restriction.Restriction.PauI.results=Bio.Restriction.Restriction.PauI-class.html#results,Variable Bio.Restriction.Restriction.PceI.results=Bio.Restriction.Restriction.PceI-class.html#results,Variable Bio.Restriction.Restriction.PciI.results=Bio.Restriction.Restriction.PciI-class.html#results,Variable Bio.Restriction.Restriction.PciSI.results=Bio.Restriction.Restriction.PciSI-class.html#results,Variable Bio.Restriction.Restriction.PcsI.results=Bio.Restriction.Restriction.PcsI-class.html#results,Variable Bio.Restriction.Restriction.PctI.results=Bio.Restriction.Restriction.PctI-class.html#results,Variable Bio.Restriction.Restriction.PdiI.results=Bio.Restriction.Restriction.PdiI-class.html#results,Variable Bio.Restriction.Restriction.PdmI.results=Bio.Restriction.Restriction.PdmI-class.html#results,Variable Bio.Restriction.Restriction.PenI.results=Bio.Restriction.Restriction.PenI-class.html#results,Variable Bio.Restriction.Restriction.PfeI.results=Bio.Restriction.Restriction.PfeI-class.html#results,Variable Bio.Restriction.Restriction.Pfl1108I.results=Bio.Restriction.Restriction.Pfl1108I-class.html#results,Variable Bio.Restriction.Restriction.Pfl23II.results=Bio.Restriction.Restriction.Pfl23II-class.html#results,Variable Bio.Restriction.Restriction.PflFI.results=Bio.Restriction.Restriction.PflFI-class.html#results,Variable Bio.Restriction.Restriction.PflMI.results=Bio.Restriction.Restriction.PflMI-class.html#results,Variable Bio.Restriction.Restriction.PfoI.results=Bio.Restriction.Restriction.PfoI-class.html#results,Variable Bio.Restriction.Restriction.PinAI.results=Bio.Restriction.Restriction.PinAI-class.html#results,Variable Bio.Restriction.Restriction.PlaDI.results=Bio.Restriction.Restriction.PlaDI-class.html#results,Variable Bio.Restriction.Restriction.Ple19I.results=Bio.Restriction.Restriction.Ple19I-class.html#results,Variable Bio.Restriction.Restriction.PleI.results=Bio.Restriction.Restriction.PleI-class.html#results,Variable Bio.Restriction.Restriction.PluTI.results=Bio.Restriction.Restriction.PluTI-class.html#results,Variable Bio.Restriction.Restriction.PmaCI.results=Bio.Restriction.Restriction.PmaCI-class.html#results,Variable Bio.Restriction.Restriction.PmeI.results=Bio.Restriction.Restriction.PmeI-class.html#results,Variable Bio.Restriction.Restriction.PmlI.results=Bio.Restriction.Restriction.PmlI-class.html#results,Variable Bio.Restriction.Restriction.PpiI.results=Bio.Restriction.Restriction.PpiI-class.html#results,Variable Bio.Restriction.Restriction.PpsI.results=Bio.Restriction.Restriction.PpsI-class.html#results,Variable Bio.Restriction.Restriction.Ppu10I.results=Bio.Restriction.Restriction.Ppu10I-class.html#results,Variable Bio.Restriction.Restriction.Ppu21I.results=Bio.Restriction.Restriction.Ppu21I-class.html#results,Variable Bio.Restriction.Restriction.PpuMI.results=Bio.Restriction.Restriction.PpuMI-class.html#results,Variable Bio.Restriction.Restriction.PscI.results=Bio.Restriction.Restriction.PscI-class.html#results,Variable Bio.Restriction.Restriction.PshAI.results=Bio.Restriction.Restriction.PshAI-class.html#results,Variable Bio.Restriction.Restriction.PshBI.results=Bio.Restriction.Restriction.PshBI-class.html#results,Variable Bio.Restriction.Restriction.PsiI.results=Bio.Restriction.Restriction.PsiI-class.html#results,Variable Bio.Restriction.Restriction.Psp03I.results=Bio.Restriction.Restriction.Psp03I-class.html#results,Variable Bio.Restriction.Restriction.Psp124BI.results=Bio.Restriction.Restriction.Psp124BI-class.html#results,Variable Bio.Restriction.Restriction.Psp1406I.results=Bio.Restriction.Restriction.Psp1406I-class.html#results,Variable Bio.Restriction.Restriction.Psp5II.results=Bio.Restriction.Restriction.Psp5II-class.html#results,Variable Bio.Restriction.Restriction.Psp6I.results=Bio.Restriction.Restriction.Psp6I-class.html#results,Variable Bio.Restriction.Restriction.PspCI.results=Bio.Restriction.Restriction.PspCI-class.html#results,Variable Bio.Restriction.Restriction.PspEI.results=Bio.Restriction.Restriction.PspEI-class.html#results,Variable Bio.Restriction.Restriction.PspGI.results=Bio.Restriction.Restriction.PspGI-class.html#results,Variable Bio.Restriction.Restriction.PspLI.results=Bio.Restriction.Restriction.PspLI-class.html#results,Variable Bio.Restriction.Restriction.PspN4I.results=Bio.Restriction.Restriction.PspN4I-class.html#results,Variable Bio.Restriction.Restriction.PspOMI.results=Bio.Restriction.Restriction.PspOMI-class.html#results,Variable Bio.Restriction.Restriction.PspOMII.results=Bio.Restriction.Restriction.PspOMII-class.html#results,Variable Bio.Restriction.Restriction.PspPI.results=Bio.Restriction.Restriction.PspPI-class.html#results,Variable Bio.Restriction.Restriction.PspPPI.results=Bio.Restriction.Restriction.PspPPI-class.html#results,Variable Bio.Restriction.Restriction.PspPRI.results=Bio.Restriction.Restriction.PspPRI-class.html#results,Variable Bio.Restriction.Restriction.PspXI.results=Bio.Restriction.Restriction.PspXI-class.html#results,Variable Bio.Restriction.Restriction.PsrI.results=Bio.Restriction.Restriction.PsrI-class.html#results,Variable Bio.Restriction.Restriction.PssI.results=Bio.Restriction.Restriction.PssI-class.html#results,Variable Bio.Restriction.Restriction.PstI.results=Bio.Restriction.Restriction.PstI-class.html#results,Variable Bio.Restriction.Restriction.PstNI.results=Bio.Restriction.Restriction.PstNI-class.html#results,Variable Bio.Restriction.Restriction.PsuI.results=Bio.Restriction.Restriction.PsuI-class.html#results,Variable Bio.Restriction.Restriction.PsyI.results=Bio.Restriction.Restriction.PsyI-class.html#results,Variable Bio.Restriction.Restriction.PteI.results=Bio.Restriction.Restriction.PteI-class.html#results,Variable Bio.Restriction.Restriction.PvuI.results=Bio.Restriction.Restriction.PvuI-class.html#results,Variable Bio.Restriction.Restriction.PvuII.results=Bio.Restriction.Restriction.PvuII-class.html#results,Variable Bio.Restriction.Restriction.R2_BceSIV.results=Bio.Restriction.Restriction.R2_BceSIV-class.html#results,Variable Bio.Restriction.Restriction.RceI.results=Bio.Restriction.Restriction.RceI-class.html#results,Variable Bio.Restriction.Restriction.RdeGBI.results=Bio.Restriction.Restriction.RdeGBI-class.html#results,Variable Bio.Restriction.Restriction.RdeGBII.results=Bio.Restriction.Restriction.RdeGBII-class.html#results,Variable Bio.Restriction.Restriction.RdeGBIII.results=Bio.Restriction.Restriction.RdeGBIII-class.html#results,Variable Bio.Restriction.Restriction.RflFIII.results=Bio.Restriction.Restriction.RflFIII-class.html#results,Variable Bio.Restriction.Restriction.RgaI.results=Bio.Restriction.Restriction.RgaI-class.html#results,Variable Bio.Restriction.Restriction.RigI.results=Bio.Restriction.Restriction.RigI-class.html#results,Variable Bio.Restriction.Restriction.RlaI.results=Bio.Restriction.Restriction.RlaI-class.html#results,Variable Bio.Restriction.Restriction.RleAI.results=Bio.Restriction.Restriction.RleAI-class.html#results,Variable Bio.Restriction.Restriction.RpaB5I.results=Bio.Restriction.Restriction.RpaB5I-class.html#results,Variable Bio.Restriction.Restriction.RpaBI.results=Bio.Restriction.Restriction.RpaBI-class.html#results,Variable Bio.Restriction.Restriction.RpaI.results=Bio.Restriction.Restriction.RpaI-class.html#results,Variable Bio.Restriction.Restriction.RpaTI.results=Bio.Restriction.Restriction.RpaTI-class.html#results,Variable Bio.Restriction.Restriction.RruI.results=Bio.Restriction.Restriction.RruI-class.html#results,Variable Bio.Restriction.Restriction.RsaI.results=Bio.Restriction.Restriction.RsaI-class.html#results,Variable Bio.Restriction.Restriction.RsaNI.results=Bio.Restriction.Restriction.RsaNI-class.html#results,Variable Bio.Restriction.Restriction.RseI.results=Bio.Restriction.Restriction.RseI-class.html#results,Variable Bio.Restriction.Restriction.Rsr2I.results=Bio.Restriction.Restriction.Rsr2I-class.html#results,Variable Bio.Restriction.Restriction.RsrII.results=Bio.Restriction.Restriction.RsrII-class.html#results,Variable Bio.Restriction.Restriction.SacI.results=Bio.Restriction.Restriction.SacI-class.html#results,Variable Bio.Restriction.Restriction.SacII.results=Bio.Restriction.Restriction.SacII-class.html#results,Variable Bio.Restriction.Restriction.SalI.results=Bio.Restriction.Restriction.SalI-class.html#results,Variable Bio.Restriction.Restriction.SanDI.results=Bio.Restriction.Restriction.SanDI-class.html#results,Variable Bio.Restriction.Restriction.SapI.results=Bio.Restriction.Restriction.SapI-class.html#results,Variable Bio.Restriction.Restriction.SaqAI.results=Bio.Restriction.Restriction.SaqAI-class.html#results,Variable Bio.Restriction.Restriction.SatI.results=Bio.Restriction.Restriction.SatI-class.html#results,Variable Bio.Restriction.Restriction.Sau3AI.results=Bio.Restriction.Restriction.Sau3AI-class.html#results,Variable Bio.Restriction.Restriction.Sau96I.results=Bio.Restriction.Restriction.Sau96I-class.html#results,Variable Bio.Restriction.Restriction.SauI.results=Bio.Restriction.Restriction.SauI-class.html#results,Variable Bio.Restriction.Restriction.SbfI.results=Bio.Restriction.Restriction.SbfI-class.html#results,Variable Bio.Restriction.Restriction.ScaI.results=Bio.Restriction.Restriction.ScaI-class.html#results,Variable Bio.Restriction.Restriction.SchI.results=Bio.Restriction.Restriction.SchI-class.html#results,Variable Bio.Restriction.Restriction.SciI.results=Bio.Restriction.Restriction.SciI-class.html#results,Variable Bio.Restriction.Restriction.ScrFI.results=Bio.Restriction.Restriction.ScrFI-class.html#results,Variable Bio.Restriction.Restriction.SdaI.results=Bio.Restriction.Restriction.SdaI-class.html#results,Variable Bio.Restriction.Restriction.SdeAI.results=Bio.Restriction.Restriction.SdeAI-class.html#results,Variable Bio.Restriction.Restriction.SdeOSI.results=Bio.Restriction.Restriction.SdeOSI-class.html#results,Variable Bio.Restriction.Restriction.SduI.results=Bio.Restriction.Restriction.SduI-class.html#results,Variable Bio.Restriction.Restriction.SecI.results=Bio.Restriction.Restriction.SecI-class.html#results,Variable Bio.Restriction.Restriction.SelI.results=Bio.Restriction.Restriction.SelI-class.html#results,Variable Bio.Restriction.Restriction.SetI.results=Bio.Restriction.Restriction.SetI-class.html#results,Variable Bio.Restriction.Restriction.SexAI.results=Bio.Restriction.Restriction.SexAI-class.html#results,Variable Bio.Restriction.Restriction.SfaAI.results=Bio.Restriction.Restriction.SfaAI-class.html#results,Variable Bio.Restriction.Restriction.SfaNI.results=Bio.Restriction.Restriction.SfaNI-class.html#results,Variable Bio.Restriction.Restriction.SfcI.results=Bio.Restriction.Restriction.SfcI-class.html#results,Variable Bio.Restriction.Restriction.SfeI.results=Bio.Restriction.Restriction.SfeI-class.html#results,Variable Bio.Restriction.Restriction.SfiI.results=Bio.Restriction.Restriction.SfiI-class.html#results,Variable Bio.Restriction.Restriction.SfoI.results=Bio.Restriction.Restriction.SfoI-class.html#results,Variable Bio.Restriction.Restriction.Sfr274I.results=Bio.Restriction.Restriction.Sfr274I-class.html#results,Variable Bio.Restriction.Restriction.Sfr303I.results=Bio.Restriction.Restriction.Sfr303I-class.html#results,Variable Bio.Restriction.Restriction.SfuI.results=Bio.Restriction.Restriction.SfuI-class.html#results,Variable Bio.Restriction.Restriction.SgeI.results=Bio.Restriction.Restriction.SgeI-class.html#results,Variable Bio.Restriction.Restriction.SgfI.results=Bio.Restriction.Restriction.SgfI-class.html#results,Variable Bio.Restriction.Restriction.SgrAI.results=Bio.Restriction.Restriction.SgrAI-class.html#results,Variable Bio.Restriction.Restriction.SgrBI.results=Bio.Restriction.Restriction.SgrBI-class.html#results,Variable Bio.Restriction.Restriction.SgrDI.results=Bio.Restriction.Restriction.SgrDI-class.html#results,Variable Bio.Restriction.Restriction.SgrTI.results=Bio.Restriction.Restriction.SgrTI-class.html#results,Variable Bio.Restriction.Restriction.SgsI.results=Bio.Restriction.Restriction.SgsI-class.html#results,Variable Bio.Restriction.Restriction.SimI.results=Bio.Restriction.Restriction.SimI-class.html#results,Variable Bio.Restriction.Restriction.SlaI.results=Bio.Restriction.Restriction.SlaI-class.html#results,Variable Bio.Restriction.Restriction.SmaI.results=Bio.Restriction.Restriction.SmaI-class.html#results,Variable Bio.Restriction.Restriction.SmiI.results=Bio.Restriction.Restriction.SmiI-class.html#results,Variable Bio.Restriction.Restriction.SmiMI.results=Bio.Restriction.Restriction.SmiMI-class.html#results,Variable Bio.Restriction.Restriction.SmlI.results=Bio.Restriction.Restriction.SmlI-class.html#results,Variable Bio.Restriction.Restriction.SmoI.results=Bio.Restriction.Restriction.SmoI-class.html#results,Variable Bio.Restriction.Restriction.SnaBI.results=Bio.Restriction.Restriction.SnaBI-class.html#results,Variable Bio.Restriction.Restriction.SnaI.results=Bio.Restriction.Restriction.SnaI-class.html#results,Variable Bio.Restriction.Restriction.Sno506I.results=Bio.Restriction.Restriction.Sno506I-class.html#results,Variable Bio.Restriction.Restriction.SpeI.results=Bio.Restriction.Restriction.SpeI-class.html#results,Variable Bio.Restriction.Restriction.SphI.results=Bio.Restriction.Restriction.SphI-class.html#results,Variable Bio.Restriction.Restriction.SplI.results=Bio.Restriction.Restriction.SplI-class.html#results,Variable Bio.Restriction.Restriction.SpoDI.results=Bio.Restriction.Restriction.SpoDI-class.html#results,Variable Bio.Restriction.Restriction.SrfI.results=Bio.Restriction.Restriction.SrfI-class.html#results,Variable Bio.Restriction.Restriction.Sse232I.results=Bio.Restriction.Restriction.Sse232I-class.html#results,Variable Bio.Restriction.Restriction.Sse8387I.results=Bio.Restriction.Restriction.Sse8387I-class.html#results,Variable Bio.Restriction.Restriction.Sse8647I.results=Bio.Restriction.Restriction.Sse8647I-class.html#results,Variable Bio.Restriction.Restriction.Sse9I.results=Bio.Restriction.Restriction.Sse9I-class.html#results,Variable Bio.Restriction.Restriction.SseBI.results=Bio.Restriction.Restriction.SseBI-class.html#results,Variable Bio.Restriction.Restriction.SsiI.results=Bio.Restriction.Restriction.SsiI-class.html#results,Variable Bio.Restriction.Restriction.SspD5I.results=Bio.Restriction.Restriction.SspD5I-class.html#results,Variable Bio.Restriction.Restriction.SspDI.results=Bio.Restriction.Restriction.SspDI-class.html#results,Variable Bio.Restriction.Restriction.SspI.results=Bio.Restriction.Restriction.SspI-class.html#results,Variable Bio.Restriction.Restriction.SstE37I.results=Bio.Restriction.Restriction.SstE37I-class.html#results,Variable Bio.Restriction.Restriction.SstI.results=Bio.Restriction.Restriction.SstI-class.html#results,Variable Bio.Restriction.Restriction.Sth132I.results=Bio.Restriction.Restriction.Sth132I-class.html#results,Variable Bio.Restriction.Restriction.Sth302II.results=Bio.Restriction.Restriction.Sth302II-class.html#results,Variable Bio.Restriction.Restriction.StrI.results=Bio.Restriction.Restriction.StrI-class.html#results,Variable Bio.Restriction.Restriction.StsI.results=Bio.Restriction.Restriction.StsI-class.html#results,Variable Bio.Restriction.Restriction.StuI.results=Bio.Restriction.Restriction.StuI-class.html#results,Variable Bio.Restriction.Restriction.StyD4I.results=Bio.Restriction.Restriction.StyD4I-class.html#results,Variable Bio.Restriction.Restriction.StyI.results=Bio.Restriction.Restriction.StyI-class.html#results,Variable Bio.Restriction.Restriction.SwaI.results=Bio.Restriction.Restriction.SwaI-class.html#results,Variable Bio.Restriction.Restriction.TaaI.results=Bio.Restriction.Restriction.TaaI-class.html#results,Variable Bio.Restriction.Restriction.TaiI.results=Bio.Restriction.Restriction.TaiI-class.html#results,Variable Bio.Restriction.Restriction.TaqI.results=Bio.Restriction.Restriction.TaqI-class.html#results,Variable Bio.Restriction.Restriction.TaqII.results=Bio.Restriction.Restriction.TaqII-class.html#results,Variable Bio.Restriction.Restriction.TasI.results=Bio.Restriction.Restriction.TasI-class.html#results,Variable Bio.Restriction.Restriction.TatI.results=Bio.Restriction.Restriction.TatI-class.html#results,Variable Bio.Restriction.Restriction.TauI.results=Bio.Restriction.Restriction.TauI-class.html#results,Variable Bio.Restriction.Restriction.TfiI.results=Bio.Restriction.Restriction.TfiI-class.html#results,Variable Bio.Restriction.Restriction.Tru1I.results=Bio.Restriction.Restriction.Tru1I-class.html#results,Variable Bio.Restriction.Restriction.Tru9I.results=Bio.Restriction.Restriction.Tru9I-class.html#results,Variable Bio.Restriction.Restriction.TscAI.results=Bio.Restriction.Restriction.TscAI-class.html#results,Variable Bio.Restriction.Restriction.TseFI.results=Bio.Restriction.Restriction.TseFI-class.html#results,Variable Bio.Restriction.Restriction.TseI.results=Bio.Restriction.Restriction.TseI-class.html#results,Variable Bio.Restriction.Restriction.TsoI.results=Bio.Restriction.Restriction.TsoI-class.html#results,Variable Bio.Restriction.Restriction.Tsp45I.results=Bio.Restriction.Restriction.Tsp45I-class.html#results,Variable Bio.Restriction.Restriction.Tsp4CI.results=Bio.Restriction.Restriction.Tsp4CI-class.html#results,Variable Bio.Restriction.Restriction.TspDTI.results=Bio.Restriction.Restriction.TspDTI-class.html#results,Variable Bio.Restriction.Restriction.TspEI.results=Bio.Restriction.Restriction.TspEI-class.html#results,Variable Bio.Restriction.Restriction.TspGWI.results=Bio.Restriction.Restriction.TspGWI-class.html#results,Variable Bio.Restriction.Restriction.TspMI.results=Bio.Restriction.Restriction.TspMI-class.html#results,Variable Bio.Restriction.Restriction.TspRI.results=Bio.Restriction.Restriction.TspRI-class.html#results,Variable Bio.Restriction.Restriction.TssI.results=Bio.Restriction.Restriction.TssI-class.html#results,Variable Bio.Restriction.Restriction.TstI.results=Bio.Restriction.Restriction.TstI-class.html#results,Variable Bio.Restriction.Restriction.TsuI.results=Bio.Restriction.Restriction.TsuI-class.html#results,Variable Bio.Restriction.Restriction.Tth111I.results=Bio.Restriction.Restriction.Tth111I-class.html#results,Variable Bio.Restriction.Restriction.Tth111II.results=Bio.Restriction.Restriction.Tth111II-class.html#results,Variable Bio.Restriction.Restriction.UbaF11I.results=Bio.Restriction.Restriction.UbaF11I-class.html#results,Variable Bio.Restriction.Restriction.UbaF12I.results=Bio.Restriction.Restriction.UbaF12I-class.html#results,Variable Bio.Restriction.Restriction.UbaF13I.results=Bio.Restriction.Restriction.UbaF13I-class.html#results,Variable Bio.Restriction.Restriction.UbaF14I.results=Bio.Restriction.Restriction.UbaF14I-class.html#results,Variable Bio.Restriction.Restriction.UbaF9I.results=Bio.Restriction.Restriction.UbaF9I-class.html#results,Variable Bio.Restriction.Restriction.UbaPI.results=Bio.Restriction.Restriction.UbaPI-class.html#results,Variable Bio.Restriction.Restriction.UcoMSI.results=Bio.Restriction.Restriction.UcoMSI-class.html#results,Variable Bio.Restriction.Restriction.UnbI.results=Bio.Restriction.Restriction.UnbI-class.html#results,Variable Bio.Restriction.Restriction.Van91I.results=Bio.Restriction.Restriction.Van91I-class.html#results,Variable Bio.Restriction.Restriction.Vha464I.results=Bio.Restriction.Restriction.Vha464I-class.html#results,Variable Bio.Restriction.Restriction.VneI.results=Bio.Restriction.Restriction.VneI-class.html#results,Variable Bio.Restriction.Restriction.VpaK11AI.results=Bio.Restriction.Restriction.VpaK11AI-class.html#results,Variable Bio.Restriction.Restriction.VpaK11BI.results=Bio.Restriction.Restriction.VpaK11BI-class.html#results,Variable Bio.Restriction.Restriction.VspI.results=Bio.Restriction.Restriction.VspI-class.html#results,Variable Bio.Restriction.Restriction.WviI.results=Bio.Restriction.Restriction.WviI-class.html#results,Variable Bio.Restriction.Restriction.XagI.results=Bio.Restriction.Restriction.XagI-class.html#results,Variable Bio.Restriction.Restriction.XapI.results=Bio.Restriction.Restriction.XapI-class.html#results,Variable Bio.Restriction.Restriction.XbaI.results=Bio.Restriction.Restriction.XbaI-class.html#results,Variable Bio.Restriction.Restriction.XceI.results=Bio.Restriction.Restriction.XceI-class.html#results,Variable Bio.Restriction.Restriction.XcmI.results=Bio.Restriction.Restriction.XcmI-class.html#results,Variable Bio.Restriction.Restriction.XhoI.results=Bio.Restriction.Restriction.XhoI-class.html#results,Variable Bio.Restriction.Restriction.XhoII.results=Bio.Restriction.Restriction.XhoII-class.html#results,Variable Bio.Restriction.Restriction.XmaI.results=Bio.Restriction.Restriction.XmaI-class.html#results,Variable Bio.Restriction.Restriction.XmaIII.results=Bio.Restriction.Restriction.XmaIII-class.html#results,Variable Bio.Restriction.Restriction.XmaJI.results=Bio.Restriction.Restriction.XmaJI-class.html#results,Variable Bio.Restriction.Restriction.XmiI.results=Bio.Restriction.Restriction.XmiI-class.html#results,Variable Bio.Restriction.Restriction.XmnI.results=Bio.Restriction.Restriction.XmnI-class.html#results,Variable Bio.Restriction.Restriction.XspI.results=Bio.Restriction.Restriction.XspI-class.html#results,Variable Bio.Restriction.Restriction.YkrI.results=Bio.Restriction.Restriction.YkrI-class.html#results,Variable Bio.Restriction.Restriction.ZraI.results=Bio.Restriction.Restriction.ZraI-class.html#results,Variable Bio.Restriction.Restriction.ZrmI.results=Bio.Restriction.Restriction.ZrmI-class.html#results,Variable Bio.Restriction.Restriction.Zsp2I.results=Bio.Restriction.Restriction.Zsp2I-class.html#results"><a title="Bio.Restriction.Restriction.AanI.results
Bio.Restriction.Restriction.AarI.results
Bio.Restriction.Restriction.AasI.results
Bio.Restriction.Restriction.AatII.results
Bio.Restriction.Restriction.AbaSI.results
Bio.Restriction.Restriction.AbsI.results
Bio.Restriction.Restriction.Acc16I.results
Bio.Restriction.Restriction.Acc36I.results
Bio.Restriction.Restriction.Acc65I.results
Bio.Restriction.Restriction.AccB1I.results
Bio.Restriction.Restriction.AccB7I.results
Bio.Restriction.Restriction.AccBSI.results
Bio.Restriction.Restriction.AccI.results
Bio.Restriction.Restriction.AccII.results
Bio.Restriction.Restriction.AccIII.results
Bio.Restriction.Restriction.AceIII.results
Bio.Restriction.Restriction.AciI.results
Bio.Restriction.Restriction.AclI.results
Bio.Restriction.Restriction.AclWI.results
Bio.Restriction.Restriction.AcoI.results
Bio.Restriction.Restriction.AcsI.results
Bio.Restriction.Restriction.AcuI.results
Bio.Restriction.Restriction.AcvI.results
Bio.Restriction.Restriction.AcyI.results
Bio.Restriction.Restriction.AdeI.results
Bio.Restriction.Restriction.AfaI.results
Bio.Restriction.Restriction.AfeI.results
Bio.Restriction.Restriction.AfiI.results
Bio.Restriction.Restriction.AflII.results
Bio.Restriction.Restriction.AflIII.results
Bio.Restriction.Restriction.AgeI.results
Bio.Restriction.Restriction.AgsI.results
Bio.Restriction.Restriction.AhaIII.results
Bio.Restriction.Restriction.AhdI.results
Bio.Restriction.Restriction.AhlI.results
Bio.Restriction.Restriction.AjiI.results
Bio.Restriction.Restriction.AjnI.results
Bio.Restriction.Restriction.AjuI.results
Bio.Restriction.Restriction.AleI.results
Bio.Restriction.Restriction.AlfI.results
Bio.Restriction.Restriction.AloI.results
Bio.Restriction.Restriction.AluBI.results
Bio.Restriction.Restriction.AluI.results
Bio.Restriction.Restriction.Alw21I.results
Bio.Restriction.Restriction.Alw26I.results
Bio.Restriction.Restriction.Alw44I.results
Bio.Restriction.Restriction.AlwFI.results
Bio.Restriction.Restriction.AlwI.results
Bio.Restriction.Restriction.AlwNI.results
Bio.Restriction.Restriction.Ama87I.results
Bio.Restriction.Restriction.Aor13HI.results
Bio.Restriction.Restriction.Aor51HI.results
Bio.Restriction.Restriction.AoxI.results
Bio.Restriction.Restriction.ApaBI.results
Bio.Restriction.Restriction.ApaI.results
Bio.Restriction.Restriction.ApaLI.results
Bio.Restriction.Restriction.ApeKI.results
Bio.Restriction.Restriction.ApoI.results
Bio.Restriction.Restriction.ApyPI.results
Bio.Restriction.Restriction.AquII.results
Bio.Restriction.Restriction.AquIII.results
Bio.Restriction.Restriction.AquIV.results
Bio.Restriction.Restriction.ArsI.results
Bio.Restriction.Restriction.AscI.results
Bio.Restriction.Restriction.AseI.results
Bio.Restriction.Restriction.Asi256I.results
Bio.Restriction.Restriction.AsiGI.results
Bio.Restriction.Restriction.AsiSI.results
Bio.Restriction.Restriction.Asp700I.results
Bio.Restriction.Restriction.Asp718I.results
Bio.Restriction.Restriction.AspA2I.results
Bio.Restriction.Restriction.AspBHI.results
Bio.Restriction.Restriction.AspLEI.results
Bio.Restriction.Restriction.AspS9I.results
Bio.Restriction.Restriction.AssI.results
Bio.Restriction.Restriction.AsuC2I.results
Bio.Restriction.Restriction.AsuHPI.results
Bio.Restriction.Restriction.AsuI.results
Bio.Restriction.Restriction.AsuII.results
Bio.Restriction.Restriction.AsuNHI.results
Bio.Restriction.Restriction.AvaI.results
Bio.Restriction.Restriction.AvaII.results
Bio.Restriction.Restriction.AvaIII.results
Bio.Restriction.Restriction.AvrII.results
Bio.Restriction.Restriction.AxyI.results
Bio.Restriction.Restriction.BaeGI.results
Bio.Restriction.Restriction.BaeI.results
Bio.Restriction.Restriction.BalI.results
Bio.Restriction.Restriction.BamHI.results
Bio.Restriction.Restriction.BanI.results
Bio.Restriction.Restriction.BanII.results
Bio.Restriction.Restriction.BarI.results
Bio.Restriction.Restriction.BasI.results
Bio.Restriction.Restriction.BauI.results
Bio.Restriction.Restriction.Bbr7I.results
Bio.Restriction.Restriction.BbrPI.results
Bio.Restriction.Restriction.BbsI.results
Bio.Restriction.Restriction.Bbv12I.results
Bio.Restriction.Restriction.BbvCI.results
Bio.Restriction.Restriction.BbvI.results
Bio.Restriction.Restriction.BbvII.results
Bio.Restriction.Restriction.BccI.results
Bio.Restriction.Restriction.Bce83I.results
Bio.Restriction.Restriction.BceAI.results
Bio.Restriction.Restriction.BcefI.results
Bio.Restriction.Restriction.BcgI.results
Bio.Restriction.Restriction.BciT130I.results
Bio.Restriction.Restriction.BciVI.results
Bio.Restriction.Restriction.BclI.results
Bio.Restriction.Restriction.BcnI.results
Bio.Restriction.Restriction.BcoDI.results
Bio.Restriction.Restriction.BcuI.results
Bio.Restriction.Restriction.BdaI.results
Bio.Restriction.Restriction.BetI.results
Bio.Restriction.Restriction.BfaI.results
Bio.Restriction.Restriction.BfiI.results
Bio.Restriction.Restriction.BfmI.results
Bio.Restriction.Restriction.BfoI.results
Bio.Restriction.Restriction.BfrI.results
Bio.Restriction.Restriction.BfuAI.results
Bio.Restriction.Restriction.BfuCI.results
Bio.Restriction.Restriction.BfuI.results
Bio.Restriction.Restriction.BglI.results
Bio.Restriction.Restriction.BglII.results
Bio.Restriction.Restriction.BinI.results
Bio.Restriction.Restriction.BisI.results
Bio.Restriction.Restriction.BlnI.results
Bio.Restriction.Restriction.BlpI.results
Bio.Restriction.Restriction.BlsI.results
Bio.Restriction.Restriction.BmcAI.results
Bio.Restriction.Restriction.Bme1390I.results
Bio.Restriction.Restriction.Bme18I.results
Bio.Restriction.Restriction.BmeDI.results
Bio.Restriction.Restriction.BmeRI.results
Bio.Restriction.Restriction.BmeT110I.results
Bio.Restriction.Restriction.BmgBI.results
Bio.Restriction.Restriction.BmgI.results
Bio.Restriction.Restriction.BmgT120I.results
Bio.Restriction.Restriction.BmiI.results
Bio.Restriction.Restriction.BmrFI.results
Bio.Restriction.Restriction.BmrI.results
Bio.Restriction.Restriction.BmsI.results
Bio.Restriction.Restriction.BmtI.results
Bio.Restriction.Restriction.BmuI.results
Bio.Restriction.Restriction.BoxI.results
Bio.Restriction.Restriction.BpiI.results
Bio.Restriction.Restriction.BplI.results
Bio.Restriction.Restriction.BpmI.results
Bio.Restriction.Restriction.Bpu10I.results
Bio.Restriction.Restriction.Bpu1102I.results
Bio.Restriction.Restriction.Bpu14I.results
Bio.Restriction.Restriction.BpuEI.results
Bio.Restriction.Restriction.BpuMI.results
Bio.Restriction.Restriction.BpvUI.results
Bio.Restriction.Restriction.Bsa29I.results
Bio.Restriction.Restriction.BsaAI.results
Bio.Restriction.Restriction.BsaBI.results
Bio.Restriction.Restriction.BsaHI.results
Bio.Restriction.Restriction.BsaI.results
Bio.Restriction.Restriction.BsaJI.results
Bio.Restriction.Restriction.BsaWI.results
Bio.Restriction.Restriction.BsaXI.results
Bio.Restriction.Restriction.BsbI.results
Bio.Restriction.Restriction.Bsc4I.results
Bio.Restriction.Restriction.BscAI.results
Bio.Restriction.Restriction.BscGI.results
Bio.Restriction.Restriction.Bse118I.results
Bio.Restriction.Restriction.Bse1I.results
Bio.Restriction.Restriction.Bse21I.results
Bio.Restriction.Restriction.Bse3DI.results
Bio.Restriction.Restriction.Bse8I.results
Bio.Restriction.Restriction.BseAI.results
Bio.Restriction.Restriction.BseBI.results
Bio.Restriction.Restriction.BseCI.results
Bio.Restriction.Restriction.BseDI.results
Bio.Restriction.Restriction.BseGI.results
Bio.Restriction.Restriction.BseJI.results
Bio.Restriction.Restriction.BseLI.results
Bio.Restriction.Restriction.BseMI.results
Bio.Restriction.Restriction.BseMII.results
Bio.Restriction.Restriction.BseNI.results
Bio.Restriction.Restriction.BsePI.results
Bio.Restriction.Restriction.BseRI.results
Bio.Restriction.Restriction.BseSI.results
Bio.Restriction.Restriction.BseX3I.results
Bio.Restriction.Restriction.BseXI.results
Bio.Restriction.Restriction.BseYI.results
Bio.Restriction.Restriction.BsgI.results
Bio.Restriction.Restriction.Bsh1236I.results
Bio.Restriction.Restriction.Bsh1285I.results
Bio.Restriction.Restriction.BshFI.results
Bio.Restriction.Restriction.BshNI.results
Bio.Restriction.Restriction.BshTI.results
Bio.Restriction.Restriction.BshVI.results
Bio.Restriction.Restriction.BsiEI.results
Bio.Restriction.Restriction.BsiHKAI.results
Bio.Restriction.Restriction.BsiHKCI.results
Bio.Restriction.Restriction.BsiI.results
Bio.Restriction.Restriction.BsiSI.results
Bio.Restriction.Restriction.BsiWI.results
Bio.Restriction.Restriction.BsiYI.results
Bio.Restriction.Restriction.BslFI.results
Bio.Restriction.Restriction.BslI.results
Bio.Restriction.Restriction.BsmAI.results
Bio.Restriction.Restriction.BsmBI.results
Bio.Restriction.Restriction.BsmFI.results
Bio.Restriction.Restriction.BsmI.results
Bio.Restriction.Restriction.BsnI.results
Bio.Restriction.Restriction.Bso31I.results
Bio.Restriction.Restriction.BsoBI.results
Bio.Restriction.Restriction.Bsp119I.results
Bio.Restriction.Restriction.Bsp120I.results
Bio.Restriction.Restriction.Bsp1286I.results
Bio.Restriction.Restriction.Bsp13I.results
Bio.Restriction.Restriction.Bsp1407I.results
Bio.Restriction.Restriction.Bsp143I.results
Bio.Restriction.Restriction.Bsp1720I.results
Bio.Restriction.Restriction.Bsp19I.results
Bio.Restriction.Restriction.Bsp24I.results
Bio.Restriction.Restriction.Bsp68I.results
Bio.Restriction.Restriction.BspACI.results
Bio.Restriction.Restriction.BspCNI.results
Bio.Restriction.Restriction.BspD6I.results
Bio.Restriction.Restriction.BspDI.results
Bio.Restriction.Restriction.BspEI.results
Bio.Restriction.Restriction.BspFNI.results
Bio.Restriction.Restriction.BspGI.results
Bio.Restriction.Restriction.BspHI.results
Bio.Restriction.Restriction.BspLI.results
Bio.Restriction.Restriction.BspLU11I.results
Bio.Restriction.Restriction.BspMI.results
Bio.Restriction.Restriction.BspMII.results
Bio.Restriction.Restriction.BspNCI.results
Bio.Restriction.Restriction.BspOI.results
Bio.Restriction.Restriction.BspPI.results
Bio.Restriction.Restriction.BspQI.results
Bio.Restriction.Restriction.BspT104I.results
Bio.Restriction.Restriction.BspT107I.results
Bio.Restriction.Restriction.BspTI.results
Bio.Restriction.Restriction.BsrBI.results
Bio.Restriction.Restriction.BsrDI.results
Bio.Restriction.Restriction.BsrFI.results
Bio.Restriction.Restriction.BsrGI.results
Bio.Restriction.Restriction.BsrI.results
Bio.Restriction.Restriction.BsrSI.results
Bio.Restriction.Restriction.BssAI.results
Bio.Restriction.Restriction.BssECI.results
Bio.Restriction.Restriction.BssHII.results
Bio.Restriction.Restriction.BssKI.results
Bio.Restriction.Restriction.BssMI.results
Bio.Restriction.Restriction.BssNAI.results
Bio.Restriction.Restriction.BssNI.results
Bio.Restriction.Restriction.BssSI.results
Bio.Restriction.Restriction.BssT1I.results
Bio.Restriction.Restriction.Bst1107I.results
Bio.Restriction.Restriction.Bst2BI.results
Bio.Restriction.Restriction.Bst2UI.results
Bio.Restriction.Restriction.Bst4CI.results
Bio.Restriction.Restriction.Bst6I.results
Bio.Restriction.Restriction.BstACI.results
Bio.Restriction.Restriction.BstAFI.results
Bio.Restriction.Restriction.BstAPI.results
Bio.Restriction.Restriction.BstAUI.results
Bio.Restriction.Restriction.BstBAI.results
Bio.Restriction.Restriction.BstBI.results
Bio.Restriction.Restriction.BstC8I.results
Bio.Restriction.Restriction.BstDEI.results
Bio.Restriction.Restriction.BstDSI.results
Bio.Restriction.Restriction.BstEII.results
Bio.Restriction.Restriction.BstENI.results
Bio.Restriction.Restriction.BstF5I.results
Bio.Restriction.Restriction.BstFNI.results
Bio.Restriction.Restriction.BstH2I.results
Bio.Restriction.Restriction.BstHHI.results
Bio.Restriction.Restriction.BstKTI.results
Bio.Restriction.Restriction.BstMAI.results
Bio.Restriction.Restriction.BstMBI.results
Bio.Restriction.Restriction.BstMCI.results
Bio.Restriction.Restriction.BstMWI.results
Bio.Restriction.Restriction.BstNI.results
Bio.Restriction.Restriction.BstNSI.results
Bio.Restriction.Restriction.BstOI.results
Bio.Restriction.Restriction.BstPAI.results
Bio.Restriction.Restriction.BstPI.results
Bio.Restriction.Restriction.BstSCI.results
Bio.Restriction.Restriction.BstSFI.results
Bio.Restriction.Restriction.BstSLI.results
Bio.Restriction.Restriction.BstSNI.results
Bio.Restriction.Restriction.BstUI.results
Bio.Restriction.Restriction.BstV1I.results
Bio.Restriction.Restriction.BstV2I.results
Bio.Restriction.Restriction.BstX2I.results
Bio.Restriction.Restriction.BstXI.results
Bio.Restriction.Restriction.BstYI.results
Bio.Restriction.Restriction.BstZ17I.results
Bio.Restriction.Restriction.BstZI.results
Bio.Restriction.Restriction.Bsu15I.results
Bio.Restriction.Restriction.Bsu36I.results
Bio.Restriction.Restriction.BsuI.results
Bio.Restriction.Restriction.BsuRI.results
Bio.Restriction.Restriction.BtgI.results
Bio.Restriction.Restriction.BtgZI.results
Bio.Restriction.Restriction.BthCI.results
Bio.Restriction.Restriction.BtrI.results
Bio.Restriction.Restriction.BtsCI.results
Bio.Restriction.Restriction.BtsI.results
Bio.Restriction.Restriction.BtsIMutI.results
Bio.Restriction.Restriction.BtuMI.results
Bio.Restriction.Restriction.BveI.results
Bio.Restriction.Restriction.Cac8I.results
Bio.Restriction.Restriction.CaiI.results
Bio.Restriction.Restriction.CauII.results
Bio.Restriction.Restriction.CchII.results
Bio.Restriction.Restriction.CchIII.results
Bio.Restriction.Restriction.CciI.results
Bio.Restriction.Restriction.CciNI.results
Bio.Restriction.Restriction.Cdi630V.results
Bio.Restriction.Restriction.CdiI.results
Bio.Restriction.Restriction.CdpI.results
Bio.Restriction.Restriction.CfoI.results
Bio.Restriction.Restriction.Cfr10I.results
Bio.Restriction.Restriction.Cfr13I.results
Bio.Restriction.Restriction.Cfr42I.results
Bio.Restriction.Restriction.Cfr9I.results
Bio.Restriction.Restriction.CfrI.results
Bio.Restriction.Restriction.Cgl13032I.results
Bio.Restriction.Restriction.Cgl13032II.results
Bio.Restriction.Restriction.ChaI.results
Bio.Restriction.Restriction.CjeFIII.results
Bio.Restriction.Restriction.CjeFV.results
Bio.Restriction.Restriction.CjeI.results
Bio.Restriction.Restriction.CjeNII.results
Bio.Restriction.Restriction.CjeNIII.results
Bio.Restriction.Restriction.CjeP659IV.results
Bio.Restriction.Restriction.CjePI.results
Bio.Restriction.Restriction.CjuI.results
Bio.Restriction.Restriction.CjuII.results
Bio.Restriction.Restriction.ClaI.results
Bio.Restriction.Restriction.CpoI.results
Bio.Restriction.Restriction.CseI.results
Bio.Restriction.Restriction.CsiI.results
Bio.Restriction.Restriction.Csp6I.results
Bio.Restriction.Restriction.CspAI.results
Bio.Restriction.Restriction.CspCI.results
Bio.Restriction.Restriction.CspI.results
Bio.Restriction.Restriction.CstMI.results
Bio.Restriction.Restriction.CviAII.results
Bio.Restriction.Restriction.CviJI.results
Bio.Restriction.Restriction.CviKI_1.results
Bio.Restriction.Restriction.CviQI.results
Bio.Restriction.Restriction.CviRI.results
Bio.Restriction.Restriction.DdeI.results
Bio.Restriction.Restriction.DinI.results
Bio.Restriction.Restriction.DpnI.results
Bio.Restriction.Restriction.DpnII.results
Bio.Restriction.Restriction.DraI.results
Bio.Restriction.Restriction.DraII.results
Bio.Restriction.Restriction.DraIII.results
Bio.Restriction.Restriction.DraRI.results
Bio.Restriction.Restriction.DrdI.results
Bio.Restriction.Restriction.DrdII.results
Bio.Restriction.Restriction.DriI.results
Bio.Restriction.Restriction.DsaI.results
Bio.Restriction.Restriction.DseDI.results
Bio.Restriction.Restriction.EaeI.results
Bio.Restriction.Restriction.EagI.results
Bio.Restriction.Restriction.Eam1104I.results
Bio.Restriction.Restriction.Eam1105I.results
Bio.Restriction.Restriction.EarI.results
Bio.Restriction.Restriction.EciI.results
Bio.Restriction.Restriction.Ecl136II.results
Bio.Restriction.Restriction.EclXI.results
Bio.Restriction.Restriction.Eco105I.results
Bio.Restriction.Restriction.Eco130I.results
Bio.Restriction.Restriction.Eco147I.results
Bio.Restriction.Restriction.Eco24I.results
Bio.Restriction.Restriction.Eco31I.results
Bio.Restriction.Restriction.Eco32I.results
Bio.Restriction.Restriction.Eco47I.results
Bio.Restriction.Restriction.Eco47III.results
Bio.Restriction.Restriction.Eco52I.results
Bio.Restriction.Restriction.Eco53kI.results
Bio.Restriction.Restriction.Eco57I.results
Bio.Restriction.Restriction.Eco57MI.results
Bio.Restriction.Restriction.Eco72I.results
Bio.Restriction.Restriction.Eco81I.results
Bio.Restriction.Restriction.Eco88I.results
Bio.Restriction.Restriction.Eco91I.results
Bio.Restriction.Restriction.EcoHI.results
Bio.Restriction.Restriction.EcoICRI.results
Bio.Restriction.Restriction.EcoNI.results
Bio.Restriction.Restriction.EcoO109I.results
Bio.Restriction.Restriction.EcoO65I.results
Bio.Restriction.Restriction.EcoRI.results
Bio.Restriction.Restriction.EcoRII.results
Bio.Restriction.Restriction.EcoRV.results
Bio.Restriction.Restriction.EcoT14I.results
Bio.Restriction.Restriction.EcoT22I.results
Bio.Restriction.Restriction.EcoT38I.results
Bio.Restriction.Restriction.EgeI.results
Bio.Restriction.Restriction.EheI.results
Bio.Restriction.Restriction.ErhI.results
Bio.Restriction.Restriction.EsaBC3I.results
Bio.Restriction.Restriction.EsaSSI.results
Bio.Restriction.Restriction.Esp3I.results
Bio.Restriction.Restriction.EspI.results
Bio.Restriction.Restriction.FaeI.results
Bio.Restriction.Restriction.FaiI.results
Bio.Restriction.Restriction.FalI.results
Bio.Restriction.Restriction.FaqI.results
Bio.Restriction.Restriction.FatI.results
Bio.Restriction.Restriction.FauI.results
Bio.Restriction.Restriction.FauNDI.results
Bio.Restriction.Restriction.FbaI.results
Bio.Restriction.Restriction.FblI.results
Bio.Restriction.Restriction.FinI.results
Bio.Restriction.Restriction.FmuI.results
Bio.Restriction.Restriction.Fnu4HI.results
Bio.Restriction.Restriction.FnuDII.results
Bio.Restriction.Restriction.FokI.results
Bio.Restriction.Restriction.FriOI.results
Bio.Restriction.Restriction.FseI.results
Bio.Restriction.Restriction.Fsp4HI.results
Bio.Restriction.Restriction.FspAI.results
Bio.Restriction.Restriction.FspBI.results
Bio.Restriction.Restriction.FspEI.results
Bio.Restriction.Restriction.FspI.results
Bio.Restriction.Restriction.GauT27I.results
Bio.Restriction.Restriction.GdiII.results
Bio.Restriction.Restriction.GlaI.results
Bio.Restriction.Restriction.GluI.results
Bio.Restriction.Restriction.GsaI.results
Bio.Restriction.Restriction.GsuI.results
Bio.Restriction.Restriction.HaeI.results
Bio.Restriction.Restriction.HaeII.results
Bio.Restriction.Restriction.HaeIII.results
Bio.Restriction.Restriction.HapII.results
Bio.Restriction.Restriction.HauII.results
Bio.Restriction.Restriction.HgaI.results
Bio.Restriction.Restriction.HgiAI.results
Bio.Restriction.Restriction.HgiCI.results
Bio.Restriction.Restriction.HgiEII.results
Bio.Restriction.Restriction.HgiJII.results
Bio.Restriction.Restriction.HhaI.results
Bio.Restriction.Restriction.Hin1I.results
Bio.Restriction.Restriction.Hin1II.results
Bio.Restriction.Restriction.Hin4I.results
Bio.Restriction.Restriction.Hin4II.results
Bio.Restriction.Restriction.Hin6I.results
Bio.Restriction.Restriction.HinP1I.results
Bio.Restriction.Restriction.HincII.results
Bio.Restriction.Restriction.HindII.results
Bio.Restriction.Restriction.HindIII.results
Bio.Restriction.Restriction.HinfI.results
Bio.Restriction.Restriction.HpaI.results
Bio.Restriction.Restriction.HpaII.results
Bio.Restriction.Restriction.HphI.results
Bio.Restriction.Restriction.Hpy166II.results
Bio.Restriction.Restriction.Hpy178III.results
Bio.Restriction.Restriction.Hpy188I.results
Bio.Restriction.Restriction.Hpy188III.results
Bio.Restriction.Restriction.Hpy8I.results
Bio.Restriction.Restriction.Hpy99I.results
Bio.Restriction.Restriction.Hpy99XIII.results
Bio.Restriction.Restriction.Hpy99XIV.results
Bio.Restriction.Restriction.HpyAV.results
Bio.Restriction.Restriction.HpyCH4III.results
Bio.Restriction.Restriction.HpyCH4IV.results
Bio.Restriction.Restriction.HpyCH4V.results
Bio.Restriction.Restriction.HpyF10VI.results
Bio.Restriction.Restriction.HpyF3I.results
Bio.Restriction.Restriction.HpySE526I.results
Bio.Restriction.Restriction.Hsp92I.results
Bio.Restriction.Restriction.Hsp92II.results
Bio.Restriction.Restriction.HspAI.results
Bio.Restriction.Restriction.Jma19592I.results
Bio.Restriction.Restriction.KasI.results
Bio.Restriction.Restriction.KflI.results
Bio.Restriction.Restriction.Kpn2I.results
Bio.Restriction.Restriction.KpnI.results
Bio.Restriction.Restriction.KroI.results
Bio.Restriction.Restriction.Ksp22I.results
Bio.Restriction.Restriction.Ksp632I.results
Bio.Restriction.Restriction.KspAI.results
Bio.Restriction.Restriction.KspI.results
Bio.Restriction.Restriction.Kzo9I.results
Bio.Restriction.Restriction.LguI.results
Bio.Restriction.Restriction.LpnI.results
Bio.Restriction.Restriction.LpnPI.results
Bio.Restriction.Restriction.Lsp1109I.results
Bio.Restriction.Restriction.LweI.results
Bio.Restriction.Restriction.MabI.results
Bio.Restriction.Restriction.MaeI.results
Bio.Restriction.Restriction.MaeII.results
Bio.Restriction.Restriction.MaeIII.results
Bio.Restriction.Restriction.MalI.results
Bio.Restriction.Restriction.MaqI.results
Bio.Restriction.Restriction.MauBI.results
Bio.Restriction.Restriction.MbiI.results
Bio.Restriction.Restriction.MboI.results
Bio.Restriction.Restriction.MboII.results
Bio.Restriction.Restriction.McaTI.results
Bio.Restriction.Restriction.McrI.results
Bio.Restriction.Restriction.MfeI.results
Bio.Restriction.Restriction.MflI.results
Bio.Restriction.Restriction.MhlI.results
Bio.Restriction.Restriction.MjaIV.results
Bio.Restriction.Restriction.MkaDII.results
Bio.Restriction.Restriction.MlsI.results
Bio.Restriction.Restriction.MluCI.results
Bio.Restriction.Restriction.MluI.results
Bio.Restriction.Restriction.MluNI.results
Bio.Restriction.Restriction.Mly113I.results
Bio.Restriction.Restriction.MlyI.results
Bio.Restriction.Restriction.MmeI.results
Bio.Restriction.Restriction.MnlI.results
Bio.Restriction.Restriction.Mph1103I.results
Bio.Restriction.Restriction.MreI.results
Bio.Restriction.Restriction.MroI.results
Bio.Restriction.Restriction.MroNI.results
Bio.Restriction.Restriction.MroXI.results
Bio.Restriction.Restriction.MscI.results
Bio.Restriction.Restriction.MseI.results
Bio.Restriction.Restriction.MslI.results
Bio.Restriction.Restriction.Msp20I.results
Bio.Restriction.Restriction.MspA1I.results
Bio.Restriction.Restriction.MspCI.results
Bio.Restriction.Restriction.MspI.results
Bio.Restriction.Restriction.MspJI.results
Bio.Restriction.Restriction.MspR9I.results
Bio.Restriction.Restriction.MssI.results
Bio.Restriction.Restriction.MstI.results
Bio.Restriction.Restriction.MunI.results
Bio.Restriction.Restriction.Mva1269I.results
Bio.Restriction.Restriction.MvaI.results
Bio.Restriction.Restriction.MvnI.results
Bio.Restriction.Restriction.MvrI.results
Bio.Restriction.Restriction.MwoI.results
Bio.Restriction.Restriction.NaeI.results
Bio.Restriction.Restriction.NarI.results
Bio.Restriction.Restriction.NciI.results
Bio.Restriction.Restriction.NcoI.results
Bio.Restriction.Restriction.NdeI.results
Bio.Restriction.Restriction.NdeII.results
Bio.Restriction.Restriction.NgoAVIII.results
Bio.Restriction.Restriction.NgoMIV.results
Bio.Restriction.Restriction.NhaXI.results
Bio.Restriction.Restriction.NheI.results
Bio.Restriction.Restriction.NlaCI.results
Bio.Restriction.Restriction.NlaIII.results
Bio.Restriction.Restriction.NlaIV.results
Bio.Restriction.Restriction.Nli3877I.results
Bio.Restriction.Restriction.NmeAIII.results
Bio.Restriction.Restriction.NmeDI.results
Bio.Restriction.Restriction.NmuCI.results
Bio.Restriction.Restriction.NotI.results
Bio.Restriction.Restriction.NruI.results
Bio.Restriction.Restriction.NsbI.results
Bio.Restriction.Restriction.NsiI.results
Bio.Restriction.Restriction.NspBII.results
Bio.Restriction.Restriction.NspI.results
Bio.Restriction.Restriction.NspV.results
Bio.Restriction.Restriction.OliI.results
Bio.Restriction.Restriction.PabI.results
Bio.Restriction.Restriction.PacI.results
Bio.Restriction.Restriction.PaeI.results
Bio.Restriction.Restriction.PaeR7I.results
Bio.Restriction.Restriction.PagI.results
Bio.Restriction.Restriction.PalAI.results
Bio.Restriction.Restriction.PasI.results
Bio.Restriction.Restriction.PauI.results
Bio.Restriction.Restriction.PceI.results
Bio.Restriction.Restriction.PciI.results
Bio.Restriction.Restriction.PciSI.results
Bio.Restriction.Restriction.PcsI.results
Bio.Restriction.Restriction.PctI.results
Bio.Restriction.Restriction.PdiI.results
Bio.Restriction.Restriction.PdmI.results
Bio.Restriction.Restriction.PenI.results
Bio.Restriction.Restriction.PfeI.results
Bio.Restriction.Restriction.Pfl1108I.results
Bio.Restriction.Restriction.Pfl23II.results
Bio.Restriction.Restriction.PflFI.results
Bio.Restriction.Restriction.PflMI.results
Bio.Restriction.Restriction.PfoI.results
Bio.Restriction.Restriction.PinAI.results
Bio.Restriction.Restriction.PlaDI.results
Bio.Restriction.Restriction.Ple19I.results
Bio.Restriction.Restriction.PleI.results
Bio.Restriction.Restriction.PluTI.results
Bio.Restriction.Restriction.PmaCI.results
Bio.Restriction.Restriction.PmeI.results
Bio.Restriction.Restriction.PmlI.results
Bio.Restriction.Restriction.PpiI.results
Bio.Restriction.Restriction.PpsI.results
Bio.Restriction.Restriction.Ppu10I.results
Bio.Restriction.Restriction.Ppu21I.results
Bio.Restriction.Restriction.PpuMI.results
Bio.Restriction.Restriction.PscI.results
Bio.Restriction.Restriction.PshAI.results
Bio.Restriction.Restriction.PshBI.results
Bio.Restriction.Restriction.PsiI.results
Bio.Restriction.Restriction.Psp03I.results
Bio.Restriction.Restriction.Psp124BI.results
Bio.Restriction.Restriction.Psp1406I.results
Bio.Restriction.Restriction.Psp5II.results
Bio.Restriction.Restriction.Psp6I.results
Bio.Restriction.Restriction.PspCI.results
Bio.Restriction.Restriction.PspEI.results
Bio.Restriction.Restriction.PspGI.results
Bio.Restriction.Restriction.PspLI.results
Bio.Restriction.Restriction.PspN4I.results
Bio.Restriction.Restriction.PspOMI.results
Bio.Restriction.Restriction.PspOMII.results
Bio.Restriction.Restriction.PspPI.results
Bio.Restriction.Restriction.PspPPI.results
Bio.Restriction.Restriction.PspPRI.results
Bio.Restriction.Restriction.PspXI.results
Bio.Restriction.Restriction.PsrI.results
Bio.Restriction.Restriction.PssI.results
Bio.Restriction.Restriction.PstI.results
Bio.Restriction.Restriction.PstNI.results
Bio.Restriction.Restriction.PsuI.results
Bio.Restriction.Restriction.PsyI.results
Bio.Restriction.Restriction.PteI.results
Bio.Restriction.Restriction.PvuI.results
Bio.Restriction.Restriction.PvuII.results
Bio.Restriction.Restriction.R2_BceSIV.results
Bio.Restriction.Restriction.RceI.results
Bio.Restriction.Restriction.RdeGBI.results
Bio.Restriction.Restriction.RdeGBII.results
Bio.Restriction.Restriction.RdeGBIII.results
Bio.Restriction.Restriction.RflFIII.results
Bio.Restriction.Restriction.RgaI.results
Bio.Restriction.Restriction.RigI.results
Bio.Restriction.Restriction.RlaI.results
Bio.Restriction.Restriction.RleAI.results
Bio.Restriction.Restriction.RpaB5I.results
Bio.Restriction.Restriction.RpaBI.results
Bio.Restriction.Restriction.RpaI.results
Bio.Restriction.Restriction.RpaTI.results
Bio.Restriction.Restriction.RruI.results
Bio.Restriction.Restriction.RsaI.results
Bio.Restriction.Restriction.RsaNI.results
Bio.Restriction.Restriction.RseI.results
Bio.Restriction.Restriction.Rsr2I.results
Bio.Restriction.Restriction.RsrII.results
Bio.Restriction.Restriction.SacI.results
Bio.Restriction.Restriction.SacII.results
Bio.Restriction.Restriction.SalI.results
Bio.Restriction.Restriction.SanDI.results
Bio.Restriction.Restriction.SapI.results
Bio.Restriction.Restriction.SaqAI.results
Bio.Restriction.Restriction.SatI.results
Bio.Restriction.Restriction.Sau3AI.results
Bio.Restriction.Restriction.Sau96I.results
Bio.Restriction.Restriction.SauI.results
Bio.Restriction.Restriction.SbfI.results
Bio.Restriction.Restriction.ScaI.results
Bio.Restriction.Restriction.SchI.results
Bio.Restriction.Restriction.SciI.results
Bio.Restriction.Restriction.ScrFI.results
Bio.Restriction.Restriction.SdaI.results
Bio.Restriction.Restriction.SdeAI.results
Bio.Restriction.Restriction.SdeOSI.results
Bio.Restriction.Restriction.SduI.results
Bio.Restriction.Restriction.SecI.results
Bio.Restriction.Restriction.SelI.results
Bio.Restriction.Restriction.SetI.results
Bio.Restriction.Restriction.SexAI.results
Bio.Restriction.Restriction.SfaAI.results
Bio.Restriction.Restriction.SfaNI.results
Bio.Restriction.Restriction.SfcI.results
Bio.Restriction.Restriction.SfeI.results
Bio.Restriction.Restriction.SfiI.results
Bio.Restriction.Restriction.SfoI.results
Bio.Restriction.Restriction.Sfr274I.results
Bio.Restriction.Restriction.Sfr303I.results
Bio.Restriction.Restriction.SfuI.results
Bio.Restriction.Restriction.SgeI.results
Bio.Restriction.Restriction.SgfI.results
Bio.Restriction.Restriction.SgrAI.results
Bio.Restriction.Restriction.SgrBI.results
Bio.Restriction.Restriction.SgrDI.results
Bio.Restriction.Restriction.SgrTI.results
Bio.Restriction.Restriction.SgsI.results
Bio.Restriction.Restriction.SimI.results
Bio.Restriction.Restriction.SlaI.results
Bio.Restriction.Restriction.SmaI.results
Bio.Restriction.Restriction.SmiI.results
Bio.Restriction.Restriction.SmiMI.results
Bio.Restriction.Restriction.SmlI.results
Bio.Restriction.Restriction.SmoI.results
Bio.Restriction.Restriction.SnaBI.results
Bio.Restriction.Restriction.SnaI.results
Bio.Restriction.Restriction.Sno506I.results
Bio.Restriction.Restriction.SpeI.results
Bio.Restriction.Restriction.SphI.results
Bio.Restriction.Restriction.SplI.results
Bio.Restriction.Restriction.SpoDI.results
Bio.Restriction.Restriction.SrfI.results
Bio.Restriction.Restriction.Sse232I.results
Bio.Restriction.Restriction.Sse8387I.results
Bio.Restriction.Restriction.Sse8647I.results
Bio.Restriction.Restriction.Sse9I.results
Bio.Restriction.Restriction.SseBI.results
Bio.Restriction.Restriction.SsiI.results
Bio.Restriction.Restriction.SspD5I.results
Bio.Restriction.Restriction.SspDI.results
Bio.Restriction.Restriction.SspI.results
Bio.Restriction.Restriction.SstE37I.results
Bio.Restriction.Restriction.SstI.results
Bio.Restriction.Restriction.Sth132I.results
Bio.Restriction.Restriction.Sth302II.results
Bio.Restriction.Restriction.StrI.results
Bio.Restriction.Restriction.StsI.results
Bio.Restriction.Restriction.StuI.results
Bio.Restriction.Restriction.StyD4I.results
Bio.Restriction.Restriction.StyI.results
Bio.Restriction.Restriction.SwaI.results
Bio.Restriction.Restriction.TaaI.results
Bio.Restriction.Restriction.TaiI.results
Bio.Restriction.Restriction.TaqI.results
Bio.Restriction.Restriction.TaqII.results
Bio.Restriction.Restriction.TasI.results
Bio.Restriction.Restriction.TatI.results
Bio.Restriction.Restriction.TauI.results
Bio.Restriction.Restriction.TfiI.results
Bio.Restriction.Restriction.Tru1I.results
Bio.Restriction.Restriction.Tru9I.results
Bio.Restriction.Restriction.TscAI.results
Bio.Restriction.Restriction.TseFI.results
Bio.Restriction.Restriction.TseI.results
Bio.Restriction.Restriction.TsoI.results
Bio.Restriction.Restriction.Tsp45I.results
Bio.Restriction.Restriction.Tsp4CI.results
Bio.Restriction.Restriction.TspDTI.results
Bio.Restriction.Restriction.TspEI.results
Bio.Restriction.Restriction.TspGWI.results
Bio.Restriction.Restriction.TspMI.results
Bio.Restriction.Restriction.TspRI.results
Bio.Restriction.Restriction.TssI.results
Bio.Restriction.Restriction.TstI.results
Bio.Restriction.Restriction.TsuI.results
Bio.Restriction.Restriction.Tth111I.results
Bio.Restriction.Restriction.Tth111II.results
Bio.Restriction.Restriction.UbaF11I.results
Bio.Restriction.Restriction.UbaF12I.results
Bio.Restriction.Restriction.UbaF13I.results
Bio.Restriction.Restriction.UbaF14I.results
Bio.Restriction.Restriction.UbaF9I.results
Bio.Restriction.Restriction.UbaPI.results
Bio.Restriction.Restriction.UcoMSI.results
Bio.Restriction.Restriction.UnbI.results
Bio.Restriction.Restriction.Van91I.results
Bio.Restriction.Restriction.Vha464I.results
Bio.Restriction.Restriction.VneI.results
Bio.Restriction.Restriction.VpaK11AI.results
Bio.Restriction.Restriction.VpaK11BI.results
Bio.Restriction.Restriction.VspI.results
Bio.Restriction.Restriction.WviI.results
Bio.Restriction.Restriction.XagI.results
Bio.Restriction.Restriction.XapI.results
Bio.Restriction.Restriction.XbaI.results
Bio.Restriction.Restriction.XceI.results
Bio.Restriction.Restriction.XcmI.results
Bio.Restriction.Restriction.XhoI.results
Bio.Restriction.Restriction.XhoII.results
Bio.Restriction.Restriction.XmaI.results
Bio.Restriction.Restriction.XmaIII.results
Bio.Restriction.Restriction.XmaJI.results
Bio.Restriction.Restriction.XmiI.results
Bio.Restriction.Restriction.XmnI.results
Bio.Restriction.Restriction.XspI.results
Bio.Restriction.Restriction.YkrI.results
Bio.Restriction.Restriction.ZraI.results
Bio.Restriction.Restriction.ZrmI.results
Bio.Restriction.Restriction.Zsp2I.results" class="py-name" href="#" onclick="return doclink('link-142', 'results', 'link-142');">results</a></tt> <tt class="py-op">=</tt> <tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-143" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetchall
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetchall" class="py-name" href="#" onclick="return doclink('link-143', 'execute_and_fetchall', 'link-99');">execute_and_fetchall</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">primary_id</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line">    <tt class="py-name">seq_feature_list</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L219"></a><tt class="py-lineno">219</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">seqfeature_id</tt><tt class="py-op">,</tt> <tt class="py-name">seqfeature_type</tt><tt class="py-op">,</tt> <tt class="py-name">seqfeature_rank</tt> <tt class="py-keyword">in</tt> <tt id="link-144" class="py-name"><a title="Bio.Restriction.Restriction.AanI.results
Bio.Restriction.Restriction.AarI.results
Bio.Restriction.Restriction.AasI.results
Bio.Restriction.Restriction.AatII.results
Bio.Restriction.Restriction.AbaSI.results
Bio.Restriction.Restriction.AbsI.results
Bio.Restriction.Restriction.Acc16I.results
Bio.Restriction.Restriction.Acc36I.results
Bio.Restriction.Restriction.Acc65I.results
Bio.Restriction.Restriction.AccB1I.results
Bio.Restriction.Restriction.AccB7I.results
Bio.Restriction.Restriction.AccBSI.results
Bio.Restriction.Restriction.AccI.results
Bio.Restriction.Restriction.AccII.results
Bio.Restriction.Restriction.AccIII.results
Bio.Restriction.Restriction.AceIII.results
Bio.Restriction.Restriction.AciI.results
Bio.Restriction.Restriction.AclI.results
Bio.Restriction.Restriction.AclWI.results
Bio.Restriction.Restriction.AcoI.results
Bio.Restriction.Restriction.AcsI.results
Bio.Restriction.Restriction.AcuI.results
Bio.Restriction.Restriction.AcvI.results
Bio.Restriction.Restriction.AcyI.results
Bio.Restriction.Restriction.AdeI.results
Bio.Restriction.Restriction.AfaI.results
Bio.Restriction.Restriction.AfeI.results
Bio.Restriction.Restriction.AfiI.results
Bio.Restriction.Restriction.AflII.results
Bio.Restriction.Restriction.AflIII.results
Bio.Restriction.Restriction.AgeI.results
Bio.Restriction.Restriction.AgsI.results
Bio.Restriction.Restriction.AhaIII.results
Bio.Restriction.Restriction.AhdI.results
Bio.Restriction.Restriction.AhlI.results
Bio.Restriction.Restriction.AjiI.results
Bio.Restriction.Restriction.AjnI.results
Bio.Restriction.Restriction.AjuI.results
Bio.Restriction.Restriction.AleI.results
Bio.Restriction.Restriction.AlfI.results
Bio.Restriction.Restriction.AloI.results
Bio.Restriction.Restriction.AluBI.results
Bio.Restriction.Restriction.AluI.results
Bio.Restriction.Restriction.Alw21I.results
Bio.Restriction.Restriction.Alw26I.results
Bio.Restriction.Restriction.Alw44I.results
Bio.Restriction.Restriction.AlwFI.results
Bio.Restriction.Restriction.AlwI.results
Bio.Restriction.Restriction.AlwNI.results
Bio.Restriction.Restriction.Ama87I.results
Bio.Restriction.Restriction.Aor13HI.results
Bio.Restriction.Restriction.Aor51HI.results
Bio.Restriction.Restriction.AoxI.results
Bio.Restriction.Restriction.ApaBI.results
Bio.Restriction.Restriction.ApaI.results
Bio.Restriction.Restriction.ApaLI.results
Bio.Restriction.Restriction.ApeKI.results
Bio.Restriction.Restriction.ApoI.results
Bio.Restriction.Restriction.ApyPI.results
Bio.Restriction.Restriction.AquII.results
Bio.Restriction.Restriction.AquIII.results
Bio.Restriction.Restriction.AquIV.results
Bio.Restriction.Restriction.ArsI.results
Bio.Restriction.Restriction.AscI.results
Bio.Restriction.Restriction.AseI.results
Bio.Restriction.Restriction.Asi256I.results
Bio.Restriction.Restriction.AsiGI.results
Bio.Restriction.Restriction.AsiSI.results
Bio.Restriction.Restriction.Asp700I.results
Bio.Restriction.Restriction.Asp718I.results
Bio.Restriction.Restriction.AspA2I.results
Bio.Restriction.Restriction.AspBHI.results
Bio.Restriction.Restriction.AspLEI.results
Bio.Restriction.Restriction.AspS9I.results
Bio.Restriction.Restriction.AssI.results
Bio.Restriction.Restriction.AsuC2I.results
Bio.Restriction.Restriction.AsuHPI.results
Bio.Restriction.Restriction.AsuI.results
Bio.Restriction.Restriction.AsuII.results
Bio.Restriction.Restriction.AsuNHI.results
Bio.Restriction.Restriction.AvaI.results
Bio.Restriction.Restriction.AvaII.results
Bio.Restriction.Restriction.AvaIII.results
Bio.Restriction.Restriction.AvrII.results
Bio.Restriction.Restriction.AxyI.results
Bio.Restriction.Restriction.BaeGI.results
Bio.Restriction.Restriction.BaeI.results
Bio.Restriction.Restriction.BalI.results
Bio.Restriction.Restriction.BamHI.results
Bio.Restriction.Restriction.BanI.results
Bio.Restriction.Restriction.BanII.results
Bio.Restriction.Restriction.BarI.results
Bio.Restriction.Restriction.BasI.results
Bio.Restriction.Restriction.BauI.results
Bio.Restriction.Restriction.Bbr7I.results
Bio.Restriction.Restriction.BbrPI.results
Bio.Restriction.Restriction.BbsI.results
Bio.Restriction.Restriction.Bbv12I.results
Bio.Restriction.Restriction.BbvCI.results
Bio.Restriction.Restriction.BbvI.results
Bio.Restriction.Restriction.BbvII.results
Bio.Restriction.Restriction.BccI.results
Bio.Restriction.Restriction.Bce83I.results
Bio.Restriction.Restriction.BceAI.results
Bio.Restriction.Restriction.BcefI.results
Bio.Restriction.Restriction.BcgI.results
Bio.Restriction.Restriction.BciT130I.results
Bio.Restriction.Restriction.BciVI.results
Bio.Restriction.Restriction.BclI.results
Bio.Restriction.Restriction.BcnI.results
Bio.Restriction.Restriction.BcoDI.results
Bio.Restriction.Restriction.BcuI.results
Bio.Restriction.Restriction.BdaI.results
Bio.Restriction.Restriction.BetI.results
Bio.Restriction.Restriction.BfaI.results
Bio.Restriction.Restriction.BfiI.results
Bio.Restriction.Restriction.BfmI.results
Bio.Restriction.Restriction.BfoI.results
Bio.Restriction.Restriction.BfrI.results
Bio.Restriction.Restriction.BfuAI.results
Bio.Restriction.Restriction.BfuCI.results
Bio.Restriction.Restriction.BfuI.results
Bio.Restriction.Restriction.BglI.results
Bio.Restriction.Restriction.BglII.results
Bio.Restriction.Restriction.BinI.results
Bio.Restriction.Restriction.BisI.results
Bio.Restriction.Restriction.BlnI.results
Bio.Restriction.Restriction.BlpI.results
Bio.Restriction.Restriction.BlsI.results
Bio.Restriction.Restriction.BmcAI.results
Bio.Restriction.Restriction.Bme1390I.results
Bio.Restriction.Restriction.Bme18I.results
Bio.Restriction.Restriction.BmeDI.results
Bio.Restriction.Restriction.BmeRI.results
Bio.Restriction.Restriction.BmeT110I.results
Bio.Restriction.Restriction.BmgBI.results
Bio.Restriction.Restriction.BmgI.results
Bio.Restriction.Restriction.BmgT120I.results
Bio.Restriction.Restriction.BmiI.results
Bio.Restriction.Restriction.BmrFI.results
Bio.Restriction.Restriction.BmrI.results
Bio.Restriction.Restriction.BmsI.results
Bio.Restriction.Restriction.BmtI.results
Bio.Restriction.Restriction.BmuI.results
Bio.Restriction.Restriction.BoxI.results
Bio.Restriction.Restriction.BpiI.results
Bio.Restriction.Restriction.BplI.results
Bio.Restriction.Restriction.BpmI.results
Bio.Restriction.Restriction.Bpu10I.results
Bio.Restriction.Restriction.Bpu1102I.results
Bio.Restriction.Restriction.Bpu14I.results
Bio.Restriction.Restriction.BpuEI.results
Bio.Restriction.Restriction.BpuMI.results
Bio.Restriction.Restriction.BpvUI.results
Bio.Restriction.Restriction.Bsa29I.results
Bio.Restriction.Restriction.BsaAI.results
Bio.Restriction.Restriction.BsaBI.results
Bio.Restriction.Restriction.BsaHI.results
Bio.Restriction.Restriction.BsaI.results
Bio.Restriction.Restriction.BsaJI.results
Bio.Restriction.Restriction.BsaWI.results
Bio.Restriction.Restriction.BsaXI.results
Bio.Restriction.Restriction.BsbI.results
Bio.Restriction.Restriction.Bsc4I.results
Bio.Restriction.Restriction.BscAI.results
Bio.Restriction.Restriction.BscGI.results
Bio.Restriction.Restriction.Bse118I.results
Bio.Restriction.Restriction.Bse1I.results
Bio.Restriction.Restriction.Bse21I.results
Bio.Restriction.Restriction.Bse3DI.results
Bio.Restriction.Restriction.Bse8I.results
Bio.Restriction.Restriction.BseAI.results
Bio.Restriction.Restriction.BseBI.results
Bio.Restriction.Restriction.BseCI.results
Bio.Restriction.Restriction.BseDI.results
Bio.Restriction.Restriction.BseGI.results
Bio.Restriction.Restriction.BseJI.results
Bio.Restriction.Restriction.BseLI.results
Bio.Restriction.Restriction.BseMI.results
Bio.Restriction.Restriction.BseMII.results
Bio.Restriction.Restriction.BseNI.results
Bio.Restriction.Restriction.BsePI.results
Bio.Restriction.Restriction.BseRI.results
Bio.Restriction.Restriction.BseSI.results
Bio.Restriction.Restriction.BseX3I.results
Bio.Restriction.Restriction.BseXI.results
Bio.Restriction.Restriction.BseYI.results
Bio.Restriction.Restriction.BsgI.results
Bio.Restriction.Restriction.Bsh1236I.results
Bio.Restriction.Restriction.Bsh1285I.results
Bio.Restriction.Restriction.BshFI.results
Bio.Restriction.Restriction.BshNI.results
Bio.Restriction.Restriction.BshTI.results
Bio.Restriction.Restriction.BshVI.results
Bio.Restriction.Restriction.BsiEI.results
Bio.Restriction.Restriction.BsiHKAI.results
Bio.Restriction.Restriction.BsiHKCI.results
Bio.Restriction.Restriction.BsiI.results
Bio.Restriction.Restriction.BsiSI.results
Bio.Restriction.Restriction.BsiWI.results
Bio.Restriction.Restriction.BsiYI.results
Bio.Restriction.Restriction.BslFI.results
Bio.Restriction.Restriction.BslI.results
Bio.Restriction.Restriction.BsmAI.results
Bio.Restriction.Restriction.BsmBI.results
Bio.Restriction.Restriction.BsmFI.results
Bio.Restriction.Restriction.BsmI.results
Bio.Restriction.Restriction.BsnI.results
Bio.Restriction.Restriction.Bso31I.results
Bio.Restriction.Restriction.BsoBI.results
Bio.Restriction.Restriction.Bsp119I.results
Bio.Restriction.Restriction.Bsp120I.results
Bio.Restriction.Restriction.Bsp1286I.results
Bio.Restriction.Restriction.Bsp13I.results
Bio.Restriction.Restriction.Bsp1407I.results
Bio.Restriction.Restriction.Bsp143I.results
Bio.Restriction.Restriction.Bsp1720I.results
Bio.Restriction.Restriction.Bsp19I.results
Bio.Restriction.Restriction.Bsp24I.results
Bio.Restriction.Restriction.Bsp68I.results
Bio.Restriction.Restriction.BspACI.results
Bio.Restriction.Restriction.BspCNI.results
Bio.Restriction.Restriction.BspD6I.results
Bio.Restriction.Restriction.BspDI.results
Bio.Restriction.Restriction.BspEI.results
Bio.Restriction.Restriction.BspFNI.results
Bio.Restriction.Restriction.BspGI.results
Bio.Restriction.Restriction.BspHI.results
Bio.Restriction.Restriction.BspLI.results
Bio.Restriction.Restriction.BspLU11I.results
Bio.Restriction.Restriction.BspMI.results
Bio.Restriction.Restriction.BspMII.results
Bio.Restriction.Restriction.BspNCI.results
Bio.Restriction.Restriction.BspOI.results
Bio.Restriction.Restriction.BspPI.results
Bio.Restriction.Restriction.BspQI.results
Bio.Restriction.Restriction.BspT104I.results
Bio.Restriction.Restriction.BspT107I.results
Bio.Restriction.Restriction.BspTI.results
Bio.Restriction.Restriction.BsrBI.results
Bio.Restriction.Restriction.BsrDI.results
Bio.Restriction.Restriction.BsrFI.results
Bio.Restriction.Restriction.BsrGI.results
Bio.Restriction.Restriction.BsrI.results
Bio.Restriction.Restriction.BsrSI.results
Bio.Restriction.Restriction.BssAI.results
Bio.Restriction.Restriction.BssECI.results
Bio.Restriction.Restriction.BssHII.results
Bio.Restriction.Restriction.BssKI.results
Bio.Restriction.Restriction.BssMI.results
Bio.Restriction.Restriction.BssNAI.results
Bio.Restriction.Restriction.BssNI.results
Bio.Restriction.Restriction.BssSI.results
Bio.Restriction.Restriction.BssT1I.results
Bio.Restriction.Restriction.Bst1107I.results
Bio.Restriction.Restriction.Bst2BI.results
Bio.Restriction.Restriction.Bst2UI.results
Bio.Restriction.Restriction.Bst4CI.results
Bio.Restriction.Restriction.Bst6I.results
Bio.Restriction.Restriction.BstACI.results
Bio.Restriction.Restriction.BstAFI.results
Bio.Restriction.Restriction.BstAPI.results
Bio.Restriction.Restriction.BstAUI.results
Bio.Restriction.Restriction.BstBAI.results
Bio.Restriction.Restriction.BstBI.results
Bio.Restriction.Restriction.BstC8I.results
Bio.Restriction.Restriction.BstDEI.results
Bio.Restriction.Restriction.BstDSI.results
Bio.Restriction.Restriction.BstEII.results
Bio.Restriction.Restriction.BstENI.results
Bio.Restriction.Restriction.BstF5I.results
Bio.Restriction.Restriction.BstFNI.results
Bio.Restriction.Restriction.BstH2I.results
Bio.Restriction.Restriction.BstHHI.results
Bio.Restriction.Restriction.BstKTI.results
Bio.Restriction.Restriction.BstMAI.results
Bio.Restriction.Restriction.BstMBI.results
Bio.Restriction.Restriction.BstMCI.results
Bio.Restriction.Restriction.BstMWI.results
Bio.Restriction.Restriction.BstNI.results
Bio.Restriction.Restriction.BstNSI.results
Bio.Restriction.Restriction.BstOI.results
Bio.Restriction.Restriction.BstPAI.results
Bio.Restriction.Restriction.BstPI.results
Bio.Restriction.Restriction.BstSCI.results
Bio.Restriction.Restriction.BstSFI.results
Bio.Restriction.Restriction.BstSLI.results
Bio.Restriction.Restriction.BstSNI.results
Bio.Restriction.Restriction.BstUI.results
Bio.Restriction.Restriction.BstV1I.results
Bio.Restriction.Restriction.BstV2I.results
Bio.Restriction.Restriction.BstX2I.results
Bio.Restriction.Restriction.BstXI.results
Bio.Restriction.Restriction.BstYI.results
Bio.Restriction.Restriction.BstZ17I.results
Bio.Restriction.Restriction.BstZI.results
Bio.Restriction.Restriction.Bsu15I.results
Bio.Restriction.Restriction.Bsu36I.results
Bio.Restriction.Restriction.BsuI.results
Bio.Restriction.Restriction.BsuRI.results
Bio.Restriction.Restriction.BtgI.results
Bio.Restriction.Restriction.BtgZI.results
Bio.Restriction.Restriction.BthCI.results
Bio.Restriction.Restriction.BtrI.results
Bio.Restriction.Restriction.BtsCI.results
Bio.Restriction.Restriction.BtsI.results
Bio.Restriction.Restriction.BtsIMutI.results
Bio.Restriction.Restriction.BtuMI.results
Bio.Restriction.Restriction.BveI.results
Bio.Restriction.Restriction.Cac8I.results
Bio.Restriction.Restriction.CaiI.results
Bio.Restriction.Restriction.CauII.results
Bio.Restriction.Restriction.CchII.results
Bio.Restriction.Restriction.CchIII.results
Bio.Restriction.Restriction.CciI.results
Bio.Restriction.Restriction.CciNI.results
Bio.Restriction.Restriction.Cdi630V.results
Bio.Restriction.Restriction.CdiI.results
Bio.Restriction.Restriction.CdpI.results
Bio.Restriction.Restriction.CfoI.results
Bio.Restriction.Restriction.Cfr10I.results
Bio.Restriction.Restriction.Cfr13I.results
Bio.Restriction.Restriction.Cfr42I.results
Bio.Restriction.Restriction.Cfr9I.results
Bio.Restriction.Restriction.CfrI.results
Bio.Restriction.Restriction.Cgl13032I.results
Bio.Restriction.Restriction.Cgl13032II.results
Bio.Restriction.Restriction.ChaI.results
Bio.Restriction.Restriction.CjeFIII.results
Bio.Restriction.Restriction.CjeFV.results
Bio.Restriction.Restriction.CjeI.results
Bio.Restriction.Restriction.CjeNII.results
Bio.Restriction.Restriction.CjeNIII.results
Bio.Restriction.Restriction.CjeP659IV.results
Bio.Restriction.Restriction.CjePI.results
Bio.Restriction.Restriction.CjuI.results
Bio.Restriction.Restriction.CjuII.results
Bio.Restriction.Restriction.ClaI.results
Bio.Restriction.Restriction.CpoI.results
Bio.Restriction.Restriction.CseI.results
Bio.Restriction.Restriction.CsiI.results
Bio.Restriction.Restriction.Csp6I.results
Bio.Restriction.Restriction.CspAI.results
Bio.Restriction.Restriction.CspCI.results
Bio.Restriction.Restriction.CspI.results
Bio.Restriction.Restriction.CstMI.results
Bio.Restriction.Restriction.CviAII.results
Bio.Restriction.Restriction.CviJI.results
Bio.Restriction.Restriction.CviKI_1.results
Bio.Restriction.Restriction.CviQI.results
Bio.Restriction.Restriction.CviRI.results
Bio.Restriction.Restriction.DdeI.results
Bio.Restriction.Restriction.DinI.results
Bio.Restriction.Restriction.DpnI.results
Bio.Restriction.Restriction.DpnII.results
Bio.Restriction.Restriction.DraI.results
Bio.Restriction.Restriction.DraII.results
Bio.Restriction.Restriction.DraIII.results
Bio.Restriction.Restriction.DraRI.results
Bio.Restriction.Restriction.DrdI.results
Bio.Restriction.Restriction.DrdII.results
Bio.Restriction.Restriction.DriI.results
Bio.Restriction.Restriction.DsaI.results
Bio.Restriction.Restriction.DseDI.results
Bio.Restriction.Restriction.EaeI.results
Bio.Restriction.Restriction.EagI.results
Bio.Restriction.Restriction.Eam1104I.results
Bio.Restriction.Restriction.Eam1105I.results
Bio.Restriction.Restriction.EarI.results
Bio.Restriction.Restriction.EciI.results
Bio.Restriction.Restriction.Ecl136II.results
Bio.Restriction.Restriction.EclXI.results
Bio.Restriction.Restriction.Eco105I.results
Bio.Restriction.Restriction.Eco130I.results
Bio.Restriction.Restriction.Eco147I.results
Bio.Restriction.Restriction.Eco24I.results
Bio.Restriction.Restriction.Eco31I.results
Bio.Restriction.Restriction.Eco32I.results
Bio.Restriction.Restriction.Eco47I.results
Bio.Restriction.Restriction.Eco47III.results
Bio.Restriction.Restriction.Eco52I.results
Bio.Restriction.Restriction.Eco53kI.results
Bio.Restriction.Restriction.Eco57I.results
Bio.Restriction.Restriction.Eco57MI.results
Bio.Restriction.Restriction.Eco72I.results
Bio.Restriction.Restriction.Eco81I.results
Bio.Restriction.Restriction.Eco88I.results
Bio.Restriction.Restriction.Eco91I.results
Bio.Restriction.Restriction.EcoHI.results
Bio.Restriction.Restriction.EcoICRI.results
Bio.Restriction.Restriction.EcoNI.results
Bio.Restriction.Restriction.EcoO109I.results
Bio.Restriction.Restriction.EcoO65I.results
Bio.Restriction.Restriction.EcoRI.results
Bio.Restriction.Restriction.EcoRII.results
Bio.Restriction.Restriction.EcoRV.results
Bio.Restriction.Restriction.EcoT14I.results
Bio.Restriction.Restriction.EcoT22I.results
Bio.Restriction.Restriction.EcoT38I.results
Bio.Restriction.Restriction.EgeI.results
Bio.Restriction.Restriction.EheI.results
Bio.Restriction.Restriction.ErhI.results
Bio.Restriction.Restriction.EsaBC3I.results
Bio.Restriction.Restriction.EsaSSI.results
Bio.Restriction.Restriction.Esp3I.results
Bio.Restriction.Restriction.EspI.results
Bio.Restriction.Restriction.FaeI.results
Bio.Restriction.Restriction.FaiI.results
Bio.Restriction.Restriction.FalI.results
Bio.Restriction.Restriction.FaqI.results
Bio.Restriction.Restriction.FatI.results
Bio.Restriction.Restriction.FauI.results
Bio.Restriction.Restriction.FauNDI.results
Bio.Restriction.Restriction.FbaI.results
Bio.Restriction.Restriction.FblI.results
Bio.Restriction.Restriction.FinI.results
Bio.Restriction.Restriction.FmuI.results
Bio.Restriction.Restriction.Fnu4HI.results
Bio.Restriction.Restriction.FnuDII.results
Bio.Restriction.Restriction.FokI.results
Bio.Restriction.Restriction.FriOI.results
Bio.Restriction.Restriction.FseI.results
Bio.Restriction.Restriction.Fsp4HI.results
Bio.Restriction.Restriction.FspAI.results
Bio.Restriction.Restriction.FspBI.results
Bio.Restriction.Restriction.FspEI.results
Bio.Restriction.Restriction.FspI.results
Bio.Restriction.Restriction.GauT27I.results
Bio.Restriction.Restriction.GdiII.results
Bio.Restriction.Restriction.GlaI.results
Bio.Restriction.Restriction.GluI.results
Bio.Restriction.Restriction.GsaI.results
Bio.Restriction.Restriction.GsuI.results
Bio.Restriction.Restriction.HaeI.results
Bio.Restriction.Restriction.HaeII.results
Bio.Restriction.Restriction.HaeIII.results
Bio.Restriction.Restriction.HapII.results
Bio.Restriction.Restriction.HauII.results
Bio.Restriction.Restriction.HgaI.results
Bio.Restriction.Restriction.HgiAI.results
Bio.Restriction.Restriction.HgiCI.results
Bio.Restriction.Restriction.HgiEII.results
Bio.Restriction.Restriction.HgiJII.results
Bio.Restriction.Restriction.HhaI.results
Bio.Restriction.Restriction.Hin1I.results
Bio.Restriction.Restriction.Hin1II.results
Bio.Restriction.Restriction.Hin4I.results
Bio.Restriction.Restriction.Hin4II.results
Bio.Restriction.Restriction.Hin6I.results
Bio.Restriction.Restriction.HinP1I.results
Bio.Restriction.Restriction.HincII.results
Bio.Restriction.Restriction.HindII.results
Bio.Restriction.Restriction.HindIII.results
Bio.Restriction.Restriction.HinfI.results
Bio.Restriction.Restriction.HpaI.results
Bio.Restriction.Restriction.HpaII.results
Bio.Restriction.Restriction.HphI.results
Bio.Restriction.Restriction.Hpy166II.results
Bio.Restriction.Restriction.Hpy178III.results
Bio.Restriction.Restriction.Hpy188I.results
Bio.Restriction.Restriction.Hpy188III.results
Bio.Restriction.Restriction.Hpy8I.results
Bio.Restriction.Restriction.Hpy99I.results
Bio.Restriction.Restriction.Hpy99XIII.results
Bio.Restriction.Restriction.Hpy99XIV.results
Bio.Restriction.Restriction.HpyAV.results
Bio.Restriction.Restriction.HpyCH4III.results
Bio.Restriction.Restriction.HpyCH4IV.results
Bio.Restriction.Restriction.HpyCH4V.results
Bio.Restriction.Restriction.HpyF10VI.results
Bio.Restriction.Restriction.HpyF3I.results
Bio.Restriction.Restriction.HpySE526I.results
Bio.Restriction.Restriction.Hsp92I.results
Bio.Restriction.Restriction.Hsp92II.results
Bio.Restriction.Restriction.HspAI.results
Bio.Restriction.Restriction.Jma19592I.results
Bio.Restriction.Restriction.KasI.results
Bio.Restriction.Restriction.KflI.results
Bio.Restriction.Restriction.Kpn2I.results
Bio.Restriction.Restriction.KpnI.results
Bio.Restriction.Restriction.KroI.results
Bio.Restriction.Restriction.Ksp22I.results
Bio.Restriction.Restriction.Ksp632I.results
Bio.Restriction.Restriction.KspAI.results
Bio.Restriction.Restriction.KspI.results
Bio.Restriction.Restriction.Kzo9I.results
Bio.Restriction.Restriction.LguI.results
Bio.Restriction.Restriction.LpnI.results
Bio.Restriction.Restriction.LpnPI.results
Bio.Restriction.Restriction.Lsp1109I.results
Bio.Restriction.Restriction.LweI.results
Bio.Restriction.Restriction.MabI.results
Bio.Restriction.Restriction.MaeI.results
Bio.Restriction.Restriction.MaeII.results
Bio.Restriction.Restriction.MaeIII.results
Bio.Restriction.Restriction.MalI.results
Bio.Restriction.Restriction.MaqI.results
Bio.Restriction.Restriction.MauBI.results
Bio.Restriction.Restriction.MbiI.results
Bio.Restriction.Restriction.MboI.results
Bio.Restriction.Restriction.MboII.results
Bio.Restriction.Restriction.McaTI.results
Bio.Restriction.Restriction.McrI.results
Bio.Restriction.Restriction.MfeI.results
Bio.Restriction.Restriction.MflI.results
Bio.Restriction.Restriction.MhlI.results
Bio.Restriction.Restriction.MjaIV.results
Bio.Restriction.Restriction.MkaDII.results
Bio.Restriction.Restriction.MlsI.results
Bio.Restriction.Restriction.MluCI.results
Bio.Restriction.Restriction.MluI.results
Bio.Restriction.Restriction.MluNI.results
Bio.Restriction.Restriction.Mly113I.results
Bio.Restriction.Restriction.MlyI.results
Bio.Restriction.Restriction.MmeI.results
Bio.Restriction.Restriction.MnlI.results
Bio.Restriction.Restriction.Mph1103I.results
Bio.Restriction.Restriction.MreI.results
Bio.Restriction.Restriction.MroI.results
Bio.Restriction.Restriction.MroNI.results
Bio.Restriction.Restriction.MroXI.results
Bio.Restriction.Restriction.MscI.results
Bio.Restriction.Restriction.MseI.results
Bio.Restriction.Restriction.MslI.results
Bio.Restriction.Restriction.Msp20I.results
Bio.Restriction.Restriction.MspA1I.results
Bio.Restriction.Restriction.MspCI.results
Bio.Restriction.Restriction.MspI.results
Bio.Restriction.Restriction.MspJI.results
Bio.Restriction.Restriction.MspR9I.results
Bio.Restriction.Restriction.MssI.results
Bio.Restriction.Restriction.MstI.results
Bio.Restriction.Restriction.MunI.results
Bio.Restriction.Restriction.Mva1269I.results
Bio.Restriction.Restriction.MvaI.results
Bio.Restriction.Restriction.MvnI.results
Bio.Restriction.Restriction.MvrI.results
Bio.Restriction.Restriction.MwoI.results
Bio.Restriction.Restriction.NaeI.results
Bio.Restriction.Restriction.NarI.results
Bio.Restriction.Restriction.NciI.results
Bio.Restriction.Restriction.NcoI.results
Bio.Restriction.Restriction.NdeI.results
Bio.Restriction.Restriction.NdeII.results
Bio.Restriction.Restriction.NgoAVIII.results
Bio.Restriction.Restriction.NgoMIV.results
Bio.Restriction.Restriction.NhaXI.results
Bio.Restriction.Restriction.NheI.results
Bio.Restriction.Restriction.NlaCI.results
Bio.Restriction.Restriction.NlaIII.results
Bio.Restriction.Restriction.NlaIV.results
Bio.Restriction.Restriction.Nli3877I.results
Bio.Restriction.Restriction.NmeAIII.results
Bio.Restriction.Restriction.NmeDI.results
Bio.Restriction.Restriction.NmuCI.results
Bio.Restriction.Restriction.NotI.results
Bio.Restriction.Restriction.NruI.results
Bio.Restriction.Restriction.NsbI.results
Bio.Restriction.Restriction.NsiI.results
Bio.Restriction.Restriction.NspBII.results
Bio.Restriction.Restriction.NspI.results
Bio.Restriction.Restriction.NspV.results
Bio.Restriction.Restriction.OliI.results
Bio.Restriction.Restriction.PabI.results
Bio.Restriction.Restriction.PacI.results
Bio.Restriction.Restriction.PaeI.results
Bio.Restriction.Restriction.PaeR7I.results
Bio.Restriction.Restriction.PagI.results
Bio.Restriction.Restriction.PalAI.results
Bio.Restriction.Restriction.PasI.results
Bio.Restriction.Restriction.PauI.results
Bio.Restriction.Restriction.PceI.results
Bio.Restriction.Restriction.PciI.results
Bio.Restriction.Restriction.PciSI.results
Bio.Restriction.Restriction.PcsI.results
Bio.Restriction.Restriction.PctI.results
Bio.Restriction.Restriction.PdiI.results
Bio.Restriction.Restriction.PdmI.results
Bio.Restriction.Restriction.PenI.results
Bio.Restriction.Restriction.PfeI.results
Bio.Restriction.Restriction.Pfl1108I.results
Bio.Restriction.Restriction.Pfl23II.results
Bio.Restriction.Restriction.PflFI.results
Bio.Restriction.Restriction.PflMI.results
Bio.Restriction.Restriction.PfoI.results
Bio.Restriction.Restriction.PinAI.results
Bio.Restriction.Restriction.PlaDI.results
Bio.Restriction.Restriction.Ple19I.results
Bio.Restriction.Restriction.PleI.results
Bio.Restriction.Restriction.PluTI.results
Bio.Restriction.Restriction.PmaCI.results
Bio.Restriction.Restriction.PmeI.results
Bio.Restriction.Restriction.PmlI.results
Bio.Restriction.Restriction.PpiI.results
Bio.Restriction.Restriction.PpsI.results
Bio.Restriction.Restriction.Ppu10I.results
Bio.Restriction.Restriction.Ppu21I.results
Bio.Restriction.Restriction.PpuMI.results
Bio.Restriction.Restriction.PscI.results
Bio.Restriction.Restriction.PshAI.results
Bio.Restriction.Restriction.PshBI.results
Bio.Restriction.Restriction.PsiI.results
Bio.Restriction.Restriction.Psp03I.results
Bio.Restriction.Restriction.Psp124BI.results
Bio.Restriction.Restriction.Psp1406I.results
Bio.Restriction.Restriction.Psp5II.results
Bio.Restriction.Restriction.Psp6I.results
Bio.Restriction.Restriction.PspCI.results
Bio.Restriction.Restriction.PspEI.results
Bio.Restriction.Restriction.PspGI.results
Bio.Restriction.Restriction.PspLI.results
Bio.Restriction.Restriction.PspN4I.results
Bio.Restriction.Restriction.PspOMI.results
Bio.Restriction.Restriction.PspOMII.results
Bio.Restriction.Restriction.PspPI.results
Bio.Restriction.Restriction.PspPPI.results
Bio.Restriction.Restriction.PspPRI.results
Bio.Restriction.Restriction.PspXI.results
Bio.Restriction.Restriction.PsrI.results
Bio.Restriction.Restriction.PssI.results
Bio.Restriction.Restriction.PstI.results
Bio.Restriction.Restriction.PstNI.results
Bio.Restriction.Restriction.PsuI.results
Bio.Restriction.Restriction.PsyI.results
Bio.Restriction.Restriction.PteI.results
Bio.Restriction.Restriction.PvuI.results
Bio.Restriction.Restriction.PvuII.results
Bio.Restriction.Restriction.R2_BceSIV.results
Bio.Restriction.Restriction.RceI.results
Bio.Restriction.Restriction.RdeGBI.results
Bio.Restriction.Restriction.RdeGBII.results
Bio.Restriction.Restriction.RdeGBIII.results
Bio.Restriction.Restriction.RflFIII.results
Bio.Restriction.Restriction.RgaI.results
Bio.Restriction.Restriction.RigI.results
Bio.Restriction.Restriction.RlaI.results
Bio.Restriction.Restriction.RleAI.results
Bio.Restriction.Restriction.RpaB5I.results
Bio.Restriction.Restriction.RpaBI.results
Bio.Restriction.Restriction.RpaI.results
Bio.Restriction.Restriction.RpaTI.results
Bio.Restriction.Restriction.RruI.results
Bio.Restriction.Restriction.RsaI.results
Bio.Restriction.Restriction.RsaNI.results
Bio.Restriction.Restriction.RseI.results
Bio.Restriction.Restriction.Rsr2I.results
Bio.Restriction.Restriction.RsrII.results
Bio.Restriction.Restriction.SacI.results
Bio.Restriction.Restriction.SacII.results
Bio.Restriction.Restriction.SalI.results
Bio.Restriction.Restriction.SanDI.results
Bio.Restriction.Restriction.SapI.results
Bio.Restriction.Restriction.SaqAI.results
Bio.Restriction.Restriction.SatI.results
Bio.Restriction.Restriction.Sau3AI.results
Bio.Restriction.Restriction.Sau96I.results
Bio.Restriction.Restriction.SauI.results
Bio.Restriction.Restriction.SbfI.results
Bio.Restriction.Restriction.ScaI.results
Bio.Restriction.Restriction.SchI.results
Bio.Restriction.Restriction.SciI.results
Bio.Restriction.Restriction.ScrFI.results
Bio.Restriction.Restriction.SdaI.results
Bio.Restriction.Restriction.SdeAI.results
Bio.Restriction.Restriction.SdeOSI.results
Bio.Restriction.Restriction.SduI.results
Bio.Restriction.Restriction.SecI.results
Bio.Restriction.Restriction.SelI.results
Bio.Restriction.Restriction.SetI.results
Bio.Restriction.Restriction.SexAI.results
Bio.Restriction.Restriction.SfaAI.results
Bio.Restriction.Restriction.SfaNI.results
Bio.Restriction.Restriction.SfcI.results
Bio.Restriction.Restriction.SfeI.results
Bio.Restriction.Restriction.SfiI.results
Bio.Restriction.Restriction.SfoI.results
Bio.Restriction.Restriction.Sfr274I.results
Bio.Restriction.Restriction.Sfr303I.results
Bio.Restriction.Restriction.SfuI.results
Bio.Restriction.Restriction.SgeI.results
Bio.Restriction.Restriction.SgfI.results
Bio.Restriction.Restriction.SgrAI.results
Bio.Restriction.Restriction.SgrBI.results
Bio.Restriction.Restriction.SgrDI.results
Bio.Restriction.Restriction.SgrTI.results
Bio.Restriction.Restriction.SgsI.results
Bio.Restriction.Restriction.SimI.results
Bio.Restriction.Restriction.SlaI.results
Bio.Restriction.Restriction.SmaI.results
Bio.Restriction.Restriction.SmiI.results
Bio.Restriction.Restriction.SmiMI.results
Bio.Restriction.Restriction.SmlI.results
Bio.Restriction.Restriction.SmoI.results
Bio.Restriction.Restriction.SnaBI.results
Bio.Restriction.Restriction.SnaI.results
Bio.Restriction.Restriction.Sno506I.results
Bio.Restriction.Restriction.SpeI.results
Bio.Restriction.Restriction.SphI.results
Bio.Restriction.Restriction.SplI.results
Bio.Restriction.Restriction.SpoDI.results
Bio.Restriction.Restriction.SrfI.results
Bio.Restriction.Restriction.Sse232I.results
Bio.Restriction.Restriction.Sse8387I.results
Bio.Restriction.Restriction.Sse8647I.results
Bio.Restriction.Restriction.Sse9I.results
Bio.Restriction.Restriction.SseBI.results
Bio.Restriction.Restriction.SsiI.results
Bio.Restriction.Restriction.SspD5I.results
Bio.Restriction.Restriction.SspDI.results
Bio.Restriction.Restriction.SspI.results
Bio.Restriction.Restriction.SstE37I.results
Bio.Restriction.Restriction.SstI.results
Bio.Restriction.Restriction.Sth132I.results
Bio.Restriction.Restriction.Sth302II.results
Bio.Restriction.Restriction.StrI.results
Bio.Restriction.Restriction.StsI.results
Bio.Restriction.Restriction.StuI.results
Bio.Restriction.Restriction.StyD4I.results
Bio.Restriction.Restriction.StyI.results
Bio.Restriction.Restriction.SwaI.results
Bio.Restriction.Restriction.TaaI.results
Bio.Restriction.Restriction.TaiI.results
Bio.Restriction.Restriction.TaqI.results
Bio.Restriction.Restriction.TaqII.results
Bio.Restriction.Restriction.TasI.results
Bio.Restriction.Restriction.TatI.results
Bio.Restriction.Restriction.TauI.results
Bio.Restriction.Restriction.TfiI.results
Bio.Restriction.Restriction.Tru1I.results
Bio.Restriction.Restriction.Tru9I.results
Bio.Restriction.Restriction.TscAI.results
Bio.Restriction.Restriction.TseFI.results
Bio.Restriction.Restriction.TseI.results
Bio.Restriction.Restriction.TsoI.results
Bio.Restriction.Restriction.Tsp45I.results
Bio.Restriction.Restriction.Tsp4CI.results
Bio.Restriction.Restriction.TspDTI.results
Bio.Restriction.Restriction.TspEI.results
Bio.Restriction.Restriction.TspGWI.results
Bio.Restriction.Restriction.TspMI.results
Bio.Restriction.Restriction.TspRI.results
Bio.Restriction.Restriction.TssI.results
Bio.Restriction.Restriction.TstI.results
Bio.Restriction.Restriction.TsuI.results
Bio.Restriction.Restriction.Tth111I.results
Bio.Restriction.Restriction.Tth111II.results
Bio.Restriction.Restriction.UbaF11I.results
Bio.Restriction.Restriction.UbaF12I.results
Bio.Restriction.Restriction.UbaF13I.results
Bio.Restriction.Restriction.UbaF14I.results
Bio.Restriction.Restriction.UbaF9I.results
Bio.Restriction.Restriction.UbaPI.results
Bio.Restriction.Restriction.UcoMSI.results
Bio.Restriction.Restriction.UnbI.results
Bio.Restriction.Restriction.Van91I.results
Bio.Restriction.Restriction.Vha464I.results
Bio.Restriction.Restriction.VneI.results
Bio.Restriction.Restriction.VpaK11AI.results
Bio.Restriction.Restriction.VpaK11BI.results
Bio.Restriction.Restriction.VspI.results
Bio.Restriction.Restriction.WviI.results
Bio.Restriction.Restriction.XagI.results
Bio.Restriction.Restriction.XapI.results
Bio.Restriction.Restriction.XbaI.results
Bio.Restriction.Restriction.XceI.results
Bio.Restriction.Restriction.XcmI.results
Bio.Restriction.Restriction.XhoI.results
Bio.Restriction.Restriction.XhoII.results
Bio.Restriction.Restriction.XmaI.results
Bio.Restriction.Restriction.XmaIII.results
Bio.Restriction.Restriction.XmaJI.results
Bio.Restriction.Restriction.XmiI.results
Bio.Restriction.Restriction.XmnI.results
Bio.Restriction.Restriction.XspI.results
Bio.Restriction.Restriction.YkrI.results
Bio.Restriction.Restriction.ZraI.results
Bio.Restriction.Restriction.ZrmI.results
Bio.Restriction.Restriction.Zsp2I.results" class="py-name" href="#" onclick="return doclink('link-144', 'results', 'link-142');">results</a></tt><tt class="py-op">:</tt> </tt>
<a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line">        <tt class="py-comment"># Get qualifiers [except for db_xref which is stored separately]</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line">        <tt class="py-name">qvs</tt> <tt class="py-op">=</tt> <tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-145" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetchall
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetchall" class="py-name" href="#" onclick="return doclink('link-145', 'execute_and_fetchall', 'link-99');">execute_and_fetchall</a></tt><tt class="py-op">(</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line">            <tt class="py-string">"SELECT name, value"</tt> </tt>
<a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line">            <tt class="py-string">" FROM seqfeature_qualifier_value  join term using (term_id)"</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line">            <tt class="py-string">" WHERE seqfeature_id = %s"</tt> </tt>
<a name="L225"></a><tt class="py-lineno">225</tt>  <tt class="py-line">            <tt class="py-string">" ORDER BY rank"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">seqfeature_id</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line">        <tt class="py-name">qualifiers</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">qv_name</tt><tt class="py-op">,</tt> <tt class="py-name">qv_value</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qvs</tt><tt class="py-op">:</tt> </tt>
<a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line">            <tt class="py-name">qualifiers</tt><tt class="py-op">.</tt><tt id="link-146" class="py-name" targets="Method Bio.Crystal.Crystal.setdefault()=Bio.Crystal.Crystal-class.html#setdefault,Method Bio._py3k._ordereddict.OrderedDict.setdefault()=Bio._py3k._ordereddict.OrderedDict-class.html#setdefault"><a title="Bio.Crystal.Crystal.setdefault
Bio._py3k._ordereddict.OrderedDict.setdefault" class="py-name" href="#" onclick="return doclink('link-146', 'setdefault', 'link-146');">setdefault</a></tt><tt class="py-op">(</tt><tt class="py-name">qv_name</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-147" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-147', 'append', 'link-141');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">qv_value</tt><tt class="py-op">)</tt> </tt>
<a name="L229"></a><tt class="py-lineno">229</tt>  <tt class="py-line">        <tt class="py-comment"># Get db_xrefs [special case of qualifiers]</tt> </tt>
<a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line">        <tt class="py-name">qvs</tt> <tt class="py-op">=</tt> <tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-148" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetchall
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetchall" class="py-name" href="#" onclick="return doclink('link-148', 'execute_and_fetchall', 'link-99');">execute_and_fetchall</a></tt><tt class="py-op">(</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line">            <tt class="py-string">"SELECT dbxref.dbname, dbxref.accession"</tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line">            <tt class="py-string">" FROM dbxref join seqfeature_dbxref using (dbxref_id)"</tt> </tt>
<a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line">            <tt class="py-string">" WHERE seqfeature_dbxref.seqfeature_id = %s"</tt> </tt>
<a name="L234"></a><tt class="py-lineno">234</tt>  <tt class="py-line">            <tt class="py-string">" ORDER BY rank"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">seqfeature_id</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">qv_name</tt><tt class="py-op">,</tt> <tt class="py-name">qv_value</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qvs</tt><tt class="py-op">:</tt> </tt>
<a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line">            <tt id="link-149" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.value()=Bio.Phylo.PhyloXMLIO.Writer-class.html#value"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-149', 'value', 'link-149');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"%s:%s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">qv_name</tt><tt class="py-op">,</tt> <tt class="py-name">qv_value</tt><tt class="py-op">)</tt> </tt>
<a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line">            <tt class="py-name">qualifiers</tt><tt class="py-op">.</tt><tt id="link-150" class="py-name"><a title="Bio.Crystal.Crystal.setdefault
Bio._py3k._ordereddict.OrderedDict.setdefault" class="py-name" href="#" onclick="return doclink('link-150', 'setdefault', 'link-146');">setdefault</a></tt><tt class="py-op">(</tt><tt class="py-string">"db_xref"</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-151" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-151', 'append', 'link-141');">append</a></tt><tt class="py-op">(</tt><tt id="link-152" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-152', 'value', 'link-149');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line">        <tt class="py-comment"># Get locations</tt> </tt>
<a name="L239"></a><tt class="py-lineno">239</tt>  <tt class="py-line">        <tt id="link-153" class="py-name"><a title="Bio.Restriction.Restriction.AanI.results
Bio.Restriction.Restriction.AarI.results
Bio.Restriction.Restriction.AasI.results
Bio.Restriction.Restriction.AatII.results
Bio.Restriction.Restriction.AbaSI.results
Bio.Restriction.Restriction.AbsI.results
Bio.Restriction.Restriction.Acc16I.results
Bio.Restriction.Restriction.Acc36I.results
Bio.Restriction.Restriction.Acc65I.results
Bio.Restriction.Restriction.AccB1I.results
Bio.Restriction.Restriction.AccB7I.results
Bio.Restriction.Restriction.AccBSI.results
Bio.Restriction.Restriction.AccI.results
Bio.Restriction.Restriction.AccII.results
Bio.Restriction.Restriction.AccIII.results
Bio.Restriction.Restriction.AceIII.results
Bio.Restriction.Restriction.AciI.results
Bio.Restriction.Restriction.AclI.results
Bio.Restriction.Restriction.AclWI.results
Bio.Restriction.Restriction.AcoI.results
Bio.Restriction.Restriction.AcsI.results
Bio.Restriction.Restriction.AcuI.results
Bio.Restriction.Restriction.AcvI.results
Bio.Restriction.Restriction.AcyI.results
Bio.Restriction.Restriction.AdeI.results
Bio.Restriction.Restriction.AfaI.results
Bio.Restriction.Restriction.AfeI.results
Bio.Restriction.Restriction.AfiI.results
Bio.Restriction.Restriction.AflII.results
Bio.Restriction.Restriction.AflIII.results
Bio.Restriction.Restriction.AgeI.results
Bio.Restriction.Restriction.AgsI.results
Bio.Restriction.Restriction.AhaIII.results
Bio.Restriction.Restriction.AhdI.results
Bio.Restriction.Restriction.AhlI.results
Bio.Restriction.Restriction.AjiI.results
Bio.Restriction.Restriction.AjnI.results
Bio.Restriction.Restriction.AjuI.results
Bio.Restriction.Restriction.AleI.results
Bio.Restriction.Restriction.AlfI.results
Bio.Restriction.Restriction.AloI.results
Bio.Restriction.Restriction.AluBI.results
Bio.Restriction.Restriction.AluI.results
Bio.Restriction.Restriction.Alw21I.results
Bio.Restriction.Restriction.Alw26I.results
Bio.Restriction.Restriction.Alw44I.results
Bio.Restriction.Restriction.AlwFI.results
Bio.Restriction.Restriction.AlwI.results
Bio.Restriction.Restriction.AlwNI.results
Bio.Restriction.Restriction.Ama87I.results
Bio.Restriction.Restriction.Aor13HI.results
Bio.Restriction.Restriction.Aor51HI.results
Bio.Restriction.Restriction.AoxI.results
Bio.Restriction.Restriction.ApaBI.results
Bio.Restriction.Restriction.ApaI.results
Bio.Restriction.Restriction.ApaLI.results
Bio.Restriction.Restriction.ApeKI.results
Bio.Restriction.Restriction.ApoI.results
Bio.Restriction.Restriction.ApyPI.results
Bio.Restriction.Restriction.AquII.results
Bio.Restriction.Restriction.AquIII.results
Bio.Restriction.Restriction.AquIV.results
Bio.Restriction.Restriction.ArsI.results
Bio.Restriction.Restriction.AscI.results
Bio.Restriction.Restriction.AseI.results
Bio.Restriction.Restriction.Asi256I.results
Bio.Restriction.Restriction.AsiGI.results
Bio.Restriction.Restriction.AsiSI.results
Bio.Restriction.Restriction.Asp700I.results
Bio.Restriction.Restriction.Asp718I.results
Bio.Restriction.Restriction.AspA2I.results
Bio.Restriction.Restriction.AspBHI.results
Bio.Restriction.Restriction.AspLEI.results
Bio.Restriction.Restriction.AspS9I.results
Bio.Restriction.Restriction.AssI.results
Bio.Restriction.Restriction.AsuC2I.results
Bio.Restriction.Restriction.AsuHPI.results
Bio.Restriction.Restriction.AsuI.results
Bio.Restriction.Restriction.AsuII.results
Bio.Restriction.Restriction.AsuNHI.results
Bio.Restriction.Restriction.AvaI.results
Bio.Restriction.Restriction.AvaII.results
Bio.Restriction.Restriction.AvaIII.results
Bio.Restriction.Restriction.AvrII.results
Bio.Restriction.Restriction.AxyI.results
Bio.Restriction.Restriction.BaeGI.results
Bio.Restriction.Restriction.BaeI.results
Bio.Restriction.Restriction.BalI.results
Bio.Restriction.Restriction.BamHI.results
Bio.Restriction.Restriction.BanI.results
Bio.Restriction.Restriction.BanII.results
Bio.Restriction.Restriction.BarI.results
Bio.Restriction.Restriction.BasI.results
Bio.Restriction.Restriction.BauI.results
Bio.Restriction.Restriction.Bbr7I.results
Bio.Restriction.Restriction.BbrPI.results
Bio.Restriction.Restriction.BbsI.results
Bio.Restriction.Restriction.Bbv12I.results
Bio.Restriction.Restriction.BbvCI.results
Bio.Restriction.Restriction.BbvI.results
Bio.Restriction.Restriction.BbvII.results
Bio.Restriction.Restriction.BccI.results
Bio.Restriction.Restriction.Bce83I.results
Bio.Restriction.Restriction.BceAI.results
Bio.Restriction.Restriction.BcefI.results
Bio.Restriction.Restriction.BcgI.results
Bio.Restriction.Restriction.BciT130I.results
Bio.Restriction.Restriction.BciVI.results
Bio.Restriction.Restriction.BclI.results
Bio.Restriction.Restriction.BcnI.results
Bio.Restriction.Restriction.BcoDI.results
Bio.Restriction.Restriction.BcuI.results
Bio.Restriction.Restriction.BdaI.results
Bio.Restriction.Restriction.BetI.results
Bio.Restriction.Restriction.BfaI.results
Bio.Restriction.Restriction.BfiI.results
Bio.Restriction.Restriction.BfmI.results
Bio.Restriction.Restriction.BfoI.results
Bio.Restriction.Restriction.BfrI.results
Bio.Restriction.Restriction.BfuAI.results
Bio.Restriction.Restriction.BfuCI.results
Bio.Restriction.Restriction.BfuI.results
Bio.Restriction.Restriction.BglI.results
Bio.Restriction.Restriction.BglII.results
Bio.Restriction.Restriction.BinI.results
Bio.Restriction.Restriction.BisI.results
Bio.Restriction.Restriction.BlnI.results
Bio.Restriction.Restriction.BlpI.results
Bio.Restriction.Restriction.BlsI.results
Bio.Restriction.Restriction.BmcAI.results
Bio.Restriction.Restriction.Bme1390I.results
Bio.Restriction.Restriction.Bme18I.results
Bio.Restriction.Restriction.BmeDI.results
Bio.Restriction.Restriction.BmeRI.results
Bio.Restriction.Restriction.BmeT110I.results
Bio.Restriction.Restriction.BmgBI.results
Bio.Restriction.Restriction.BmgI.results
Bio.Restriction.Restriction.BmgT120I.results
Bio.Restriction.Restriction.BmiI.results
Bio.Restriction.Restriction.BmrFI.results
Bio.Restriction.Restriction.BmrI.results
Bio.Restriction.Restriction.BmsI.results
Bio.Restriction.Restriction.BmtI.results
Bio.Restriction.Restriction.BmuI.results
Bio.Restriction.Restriction.BoxI.results
Bio.Restriction.Restriction.BpiI.results
Bio.Restriction.Restriction.BplI.results
Bio.Restriction.Restriction.BpmI.results
Bio.Restriction.Restriction.Bpu10I.results
Bio.Restriction.Restriction.Bpu1102I.results
Bio.Restriction.Restriction.Bpu14I.results
Bio.Restriction.Restriction.BpuEI.results
Bio.Restriction.Restriction.BpuMI.results
Bio.Restriction.Restriction.BpvUI.results
Bio.Restriction.Restriction.Bsa29I.results
Bio.Restriction.Restriction.BsaAI.results
Bio.Restriction.Restriction.BsaBI.results
Bio.Restriction.Restriction.BsaHI.results
Bio.Restriction.Restriction.BsaI.results
Bio.Restriction.Restriction.BsaJI.results
Bio.Restriction.Restriction.BsaWI.results
Bio.Restriction.Restriction.BsaXI.results
Bio.Restriction.Restriction.BsbI.results
Bio.Restriction.Restriction.Bsc4I.results
Bio.Restriction.Restriction.BscAI.results
Bio.Restriction.Restriction.BscGI.results
Bio.Restriction.Restriction.Bse118I.results
Bio.Restriction.Restriction.Bse1I.results
Bio.Restriction.Restriction.Bse21I.results
Bio.Restriction.Restriction.Bse3DI.results
Bio.Restriction.Restriction.Bse8I.results
Bio.Restriction.Restriction.BseAI.results
Bio.Restriction.Restriction.BseBI.results
Bio.Restriction.Restriction.BseCI.results
Bio.Restriction.Restriction.BseDI.results
Bio.Restriction.Restriction.BseGI.results
Bio.Restriction.Restriction.BseJI.results
Bio.Restriction.Restriction.BseLI.results
Bio.Restriction.Restriction.BseMI.results
Bio.Restriction.Restriction.BseMII.results
Bio.Restriction.Restriction.BseNI.results
Bio.Restriction.Restriction.BsePI.results
Bio.Restriction.Restriction.BseRI.results
Bio.Restriction.Restriction.BseSI.results
Bio.Restriction.Restriction.BseX3I.results
Bio.Restriction.Restriction.BseXI.results
Bio.Restriction.Restriction.BseYI.results
Bio.Restriction.Restriction.BsgI.results
Bio.Restriction.Restriction.Bsh1236I.results
Bio.Restriction.Restriction.Bsh1285I.results
Bio.Restriction.Restriction.BshFI.results
Bio.Restriction.Restriction.BshNI.results
Bio.Restriction.Restriction.BshTI.results
Bio.Restriction.Restriction.BshVI.results
Bio.Restriction.Restriction.BsiEI.results
Bio.Restriction.Restriction.BsiHKAI.results
Bio.Restriction.Restriction.BsiHKCI.results
Bio.Restriction.Restriction.BsiI.results
Bio.Restriction.Restriction.BsiSI.results
Bio.Restriction.Restriction.BsiWI.results
Bio.Restriction.Restriction.BsiYI.results
Bio.Restriction.Restriction.BslFI.results
Bio.Restriction.Restriction.BslI.results
Bio.Restriction.Restriction.BsmAI.results
Bio.Restriction.Restriction.BsmBI.results
Bio.Restriction.Restriction.BsmFI.results
Bio.Restriction.Restriction.BsmI.results
Bio.Restriction.Restriction.BsnI.results
Bio.Restriction.Restriction.Bso31I.results
Bio.Restriction.Restriction.BsoBI.results
Bio.Restriction.Restriction.Bsp119I.results
Bio.Restriction.Restriction.Bsp120I.results
Bio.Restriction.Restriction.Bsp1286I.results
Bio.Restriction.Restriction.Bsp13I.results
Bio.Restriction.Restriction.Bsp1407I.results
Bio.Restriction.Restriction.Bsp143I.results
Bio.Restriction.Restriction.Bsp1720I.results
Bio.Restriction.Restriction.Bsp19I.results
Bio.Restriction.Restriction.Bsp24I.results
Bio.Restriction.Restriction.Bsp68I.results
Bio.Restriction.Restriction.BspACI.results
Bio.Restriction.Restriction.BspCNI.results
Bio.Restriction.Restriction.BspD6I.results
Bio.Restriction.Restriction.BspDI.results
Bio.Restriction.Restriction.BspEI.results
Bio.Restriction.Restriction.BspFNI.results
Bio.Restriction.Restriction.BspGI.results
Bio.Restriction.Restriction.BspHI.results
Bio.Restriction.Restriction.BspLI.results
Bio.Restriction.Restriction.BspLU11I.results
Bio.Restriction.Restriction.BspMI.results
Bio.Restriction.Restriction.BspMII.results
Bio.Restriction.Restriction.BspNCI.results
Bio.Restriction.Restriction.BspOI.results
Bio.Restriction.Restriction.BspPI.results
Bio.Restriction.Restriction.BspQI.results
Bio.Restriction.Restriction.BspT104I.results
Bio.Restriction.Restriction.BspT107I.results
Bio.Restriction.Restriction.BspTI.results
Bio.Restriction.Restriction.BsrBI.results
Bio.Restriction.Restriction.BsrDI.results
Bio.Restriction.Restriction.BsrFI.results
Bio.Restriction.Restriction.BsrGI.results
Bio.Restriction.Restriction.BsrI.results
Bio.Restriction.Restriction.BsrSI.results
Bio.Restriction.Restriction.BssAI.results
Bio.Restriction.Restriction.BssECI.results
Bio.Restriction.Restriction.BssHII.results
Bio.Restriction.Restriction.BssKI.results
Bio.Restriction.Restriction.BssMI.results
Bio.Restriction.Restriction.BssNAI.results
Bio.Restriction.Restriction.BssNI.results
Bio.Restriction.Restriction.BssSI.results
Bio.Restriction.Restriction.BssT1I.results
Bio.Restriction.Restriction.Bst1107I.results
Bio.Restriction.Restriction.Bst2BI.results
Bio.Restriction.Restriction.Bst2UI.results
Bio.Restriction.Restriction.Bst4CI.results
Bio.Restriction.Restriction.Bst6I.results
Bio.Restriction.Restriction.BstACI.results
Bio.Restriction.Restriction.BstAFI.results
Bio.Restriction.Restriction.BstAPI.results
Bio.Restriction.Restriction.BstAUI.results
Bio.Restriction.Restriction.BstBAI.results
Bio.Restriction.Restriction.BstBI.results
Bio.Restriction.Restriction.BstC8I.results
Bio.Restriction.Restriction.BstDEI.results
Bio.Restriction.Restriction.BstDSI.results
Bio.Restriction.Restriction.BstEII.results
Bio.Restriction.Restriction.BstENI.results
Bio.Restriction.Restriction.BstF5I.results
Bio.Restriction.Restriction.BstFNI.results
Bio.Restriction.Restriction.BstH2I.results
Bio.Restriction.Restriction.BstHHI.results
Bio.Restriction.Restriction.BstKTI.results
Bio.Restriction.Restriction.BstMAI.results
Bio.Restriction.Restriction.BstMBI.results
Bio.Restriction.Restriction.BstMCI.results
Bio.Restriction.Restriction.BstMWI.results
Bio.Restriction.Restriction.BstNI.results
Bio.Restriction.Restriction.BstNSI.results
Bio.Restriction.Restriction.BstOI.results
Bio.Restriction.Restriction.BstPAI.results
Bio.Restriction.Restriction.BstPI.results
Bio.Restriction.Restriction.BstSCI.results
Bio.Restriction.Restriction.BstSFI.results
Bio.Restriction.Restriction.BstSLI.results
Bio.Restriction.Restriction.BstSNI.results
Bio.Restriction.Restriction.BstUI.results
Bio.Restriction.Restriction.BstV1I.results
Bio.Restriction.Restriction.BstV2I.results
Bio.Restriction.Restriction.BstX2I.results
Bio.Restriction.Restriction.BstXI.results
Bio.Restriction.Restriction.BstYI.results
Bio.Restriction.Restriction.BstZ17I.results
Bio.Restriction.Restriction.BstZI.results
Bio.Restriction.Restriction.Bsu15I.results
Bio.Restriction.Restriction.Bsu36I.results
Bio.Restriction.Restriction.BsuI.results
Bio.Restriction.Restriction.BsuRI.results
Bio.Restriction.Restriction.BtgI.results
Bio.Restriction.Restriction.BtgZI.results
Bio.Restriction.Restriction.BthCI.results
Bio.Restriction.Restriction.BtrI.results
Bio.Restriction.Restriction.BtsCI.results
Bio.Restriction.Restriction.BtsI.results
Bio.Restriction.Restriction.BtsIMutI.results
Bio.Restriction.Restriction.BtuMI.results
Bio.Restriction.Restriction.BveI.results
Bio.Restriction.Restriction.Cac8I.results
Bio.Restriction.Restriction.CaiI.results
Bio.Restriction.Restriction.CauII.results
Bio.Restriction.Restriction.CchII.results
Bio.Restriction.Restriction.CchIII.results
Bio.Restriction.Restriction.CciI.results
Bio.Restriction.Restriction.CciNI.results
Bio.Restriction.Restriction.Cdi630V.results
Bio.Restriction.Restriction.CdiI.results
Bio.Restriction.Restriction.CdpI.results
Bio.Restriction.Restriction.CfoI.results
Bio.Restriction.Restriction.Cfr10I.results
Bio.Restriction.Restriction.Cfr13I.results
Bio.Restriction.Restriction.Cfr42I.results
Bio.Restriction.Restriction.Cfr9I.results
Bio.Restriction.Restriction.CfrI.results
Bio.Restriction.Restriction.Cgl13032I.results
Bio.Restriction.Restriction.Cgl13032II.results
Bio.Restriction.Restriction.ChaI.results
Bio.Restriction.Restriction.CjeFIII.results
Bio.Restriction.Restriction.CjeFV.results
Bio.Restriction.Restriction.CjeI.results
Bio.Restriction.Restriction.CjeNII.results
Bio.Restriction.Restriction.CjeNIII.results
Bio.Restriction.Restriction.CjeP659IV.results
Bio.Restriction.Restriction.CjePI.results
Bio.Restriction.Restriction.CjuI.results
Bio.Restriction.Restriction.CjuII.results
Bio.Restriction.Restriction.ClaI.results
Bio.Restriction.Restriction.CpoI.results
Bio.Restriction.Restriction.CseI.results
Bio.Restriction.Restriction.CsiI.results
Bio.Restriction.Restriction.Csp6I.results
Bio.Restriction.Restriction.CspAI.results
Bio.Restriction.Restriction.CspCI.results
Bio.Restriction.Restriction.CspI.results
Bio.Restriction.Restriction.CstMI.results
Bio.Restriction.Restriction.CviAII.results
Bio.Restriction.Restriction.CviJI.results
Bio.Restriction.Restriction.CviKI_1.results
Bio.Restriction.Restriction.CviQI.results
Bio.Restriction.Restriction.CviRI.results
Bio.Restriction.Restriction.DdeI.results
Bio.Restriction.Restriction.DinI.results
Bio.Restriction.Restriction.DpnI.results
Bio.Restriction.Restriction.DpnII.results
Bio.Restriction.Restriction.DraI.results
Bio.Restriction.Restriction.DraII.results
Bio.Restriction.Restriction.DraIII.results
Bio.Restriction.Restriction.DraRI.results
Bio.Restriction.Restriction.DrdI.results
Bio.Restriction.Restriction.DrdII.results
Bio.Restriction.Restriction.DriI.results
Bio.Restriction.Restriction.DsaI.results
Bio.Restriction.Restriction.DseDI.results
Bio.Restriction.Restriction.EaeI.results
Bio.Restriction.Restriction.EagI.results
Bio.Restriction.Restriction.Eam1104I.results
Bio.Restriction.Restriction.Eam1105I.results
Bio.Restriction.Restriction.EarI.results
Bio.Restriction.Restriction.EciI.results
Bio.Restriction.Restriction.Ecl136II.results
Bio.Restriction.Restriction.EclXI.results
Bio.Restriction.Restriction.Eco105I.results
Bio.Restriction.Restriction.Eco130I.results
Bio.Restriction.Restriction.Eco147I.results
Bio.Restriction.Restriction.Eco24I.results
Bio.Restriction.Restriction.Eco31I.results
Bio.Restriction.Restriction.Eco32I.results
Bio.Restriction.Restriction.Eco47I.results
Bio.Restriction.Restriction.Eco47III.results
Bio.Restriction.Restriction.Eco52I.results
Bio.Restriction.Restriction.Eco53kI.results
Bio.Restriction.Restriction.Eco57I.results
Bio.Restriction.Restriction.Eco57MI.results
Bio.Restriction.Restriction.Eco72I.results
Bio.Restriction.Restriction.Eco81I.results
Bio.Restriction.Restriction.Eco88I.results
Bio.Restriction.Restriction.Eco91I.results
Bio.Restriction.Restriction.EcoHI.results
Bio.Restriction.Restriction.EcoICRI.results
Bio.Restriction.Restriction.EcoNI.results
Bio.Restriction.Restriction.EcoO109I.results
Bio.Restriction.Restriction.EcoO65I.results
Bio.Restriction.Restriction.EcoRI.results
Bio.Restriction.Restriction.EcoRII.results
Bio.Restriction.Restriction.EcoRV.results
Bio.Restriction.Restriction.EcoT14I.results
Bio.Restriction.Restriction.EcoT22I.results
Bio.Restriction.Restriction.EcoT38I.results
Bio.Restriction.Restriction.EgeI.results
Bio.Restriction.Restriction.EheI.results
Bio.Restriction.Restriction.ErhI.results
Bio.Restriction.Restriction.EsaBC3I.results
Bio.Restriction.Restriction.EsaSSI.results
Bio.Restriction.Restriction.Esp3I.results
Bio.Restriction.Restriction.EspI.results
Bio.Restriction.Restriction.FaeI.results
Bio.Restriction.Restriction.FaiI.results
Bio.Restriction.Restriction.FalI.results
Bio.Restriction.Restriction.FaqI.results
Bio.Restriction.Restriction.FatI.results
Bio.Restriction.Restriction.FauI.results
Bio.Restriction.Restriction.FauNDI.results
Bio.Restriction.Restriction.FbaI.results
Bio.Restriction.Restriction.FblI.results
Bio.Restriction.Restriction.FinI.results
Bio.Restriction.Restriction.FmuI.results
Bio.Restriction.Restriction.Fnu4HI.results
Bio.Restriction.Restriction.FnuDII.results
Bio.Restriction.Restriction.FokI.results
Bio.Restriction.Restriction.FriOI.results
Bio.Restriction.Restriction.FseI.results
Bio.Restriction.Restriction.Fsp4HI.results
Bio.Restriction.Restriction.FspAI.results
Bio.Restriction.Restriction.FspBI.results
Bio.Restriction.Restriction.FspEI.results
Bio.Restriction.Restriction.FspI.results
Bio.Restriction.Restriction.GauT27I.results
Bio.Restriction.Restriction.GdiII.results
Bio.Restriction.Restriction.GlaI.results
Bio.Restriction.Restriction.GluI.results
Bio.Restriction.Restriction.GsaI.results
Bio.Restriction.Restriction.GsuI.results
Bio.Restriction.Restriction.HaeI.results
Bio.Restriction.Restriction.HaeII.results
Bio.Restriction.Restriction.HaeIII.results
Bio.Restriction.Restriction.HapII.results
Bio.Restriction.Restriction.HauII.results
Bio.Restriction.Restriction.HgaI.results
Bio.Restriction.Restriction.HgiAI.results
Bio.Restriction.Restriction.HgiCI.results
Bio.Restriction.Restriction.HgiEII.results
Bio.Restriction.Restriction.HgiJII.results
Bio.Restriction.Restriction.HhaI.results
Bio.Restriction.Restriction.Hin1I.results
Bio.Restriction.Restriction.Hin1II.results
Bio.Restriction.Restriction.Hin4I.results
Bio.Restriction.Restriction.Hin4II.results
Bio.Restriction.Restriction.Hin6I.results
Bio.Restriction.Restriction.HinP1I.results
Bio.Restriction.Restriction.HincII.results
Bio.Restriction.Restriction.HindII.results
Bio.Restriction.Restriction.HindIII.results
Bio.Restriction.Restriction.HinfI.results
Bio.Restriction.Restriction.HpaI.results
Bio.Restriction.Restriction.HpaII.results
Bio.Restriction.Restriction.HphI.results
Bio.Restriction.Restriction.Hpy166II.results
Bio.Restriction.Restriction.Hpy178III.results
Bio.Restriction.Restriction.Hpy188I.results
Bio.Restriction.Restriction.Hpy188III.results
Bio.Restriction.Restriction.Hpy8I.results
Bio.Restriction.Restriction.Hpy99I.results
Bio.Restriction.Restriction.Hpy99XIII.results
Bio.Restriction.Restriction.Hpy99XIV.results
Bio.Restriction.Restriction.HpyAV.results
Bio.Restriction.Restriction.HpyCH4III.results
Bio.Restriction.Restriction.HpyCH4IV.results
Bio.Restriction.Restriction.HpyCH4V.results
Bio.Restriction.Restriction.HpyF10VI.results
Bio.Restriction.Restriction.HpyF3I.results
Bio.Restriction.Restriction.HpySE526I.results
Bio.Restriction.Restriction.Hsp92I.results
Bio.Restriction.Restriction.Hsp92II.results
Bio.Restriction.Restriction.HspAI.results
Bio.Restriction.Restriction.Jma19592I.results
Bio.Restriction.Restriction.KasI.results
Bio.Restriction.Restriction.KflI.results
Bio.Restriction.Restriction.Kpn2I.results
Bio.Restriction.Restriction.KpnI.results
Bio.Restriction.Restriction.KroI.results
Bio.Restriction.Restriction.Ksp22I.results
Bio.Restriction.Restriction.Ksp632I.results
Bio.Restriction.Restriction.KspAI.results
Bio.Restriction.Restriction.KspI.results
Bio.Restriction.Restriction.Kzo9I.results
Bio.Restriction.Restriction.LguI.results
Bio.Restriction.Restriction.LpnI.results
Bio.Restriction.Restriction.LpnPI.results
Bio.Restriction.Restriction.Lsp1109I.results
Bio.Restriction.Restriction.LweI.results
Bio.Restriction.Restriction.MabI.results
Bio.Restriction.Restriction.MaeI.results
Bio.Restriction.Restriction.MaeII.results
Bio.Restriction.Restriction.MaeIII.results
Bio.Restriction.Restriction.MalI.results
Bio.Restriction.Restriction.MaqI.results
Bio.Restriction.Restriction.MauBI.results
Bio.Restriction.Restriction.MbiI.results
Bio.Restriction.Restriction.MboI.results
Bio.Restriction.Restriction.MboII.results
Bio.Restriction.Restriction.McaTI.results
Bio.Restriction.Restriction.McrI.results
Bio.Restriction.Restriction.MfeI.results
Bio.Restriction.Restriction.MflI.results
Bio.Restriction.Restriction.MhlI.results
Bio.Restriction.Restriction.MjaIV.results
Bio.Restriction.Restriction.MkaDII.results
Bio.Restriction.Restriction.MlsI.results
Bio.Restriction.Restriction.MluCI.results
Bio.Restriction.Restriction.MluI.results
Bio.Restriction.Restriction.MluNI.results
Bio.Restriction.Restriction.Mly113I.results
Bio.Restriction.Restriction.MlyI.results
Bio.Restriction.Restriction.MmeI.results
Bio.Restriction.Restriction.MnlI.results
Bio.Restriction.Restriction.Mph1103I.results
Bio.Restriction.Restriction.MreI.results
Bio.Restriction.Restriction.MroI.results
Bio.Restriction.Restriction.MroNI.results
Bio.Restriction.Restriction.MroXI.results
Bio.Restriction.Restriction.MscI.results
Bio.Restriction.Restriction.MseI.results
Bio.Restriction.Restriction.MslI.results
Bio.Restriction.Restriction.Msp20I.results
Bio.Restriction.Restriction.MspA1I.results
Bio.Restriction.Restriction.MspCI.results
Bio.Restriction.Restriction.MspI.results
Bio.Restriction.Restriction.MspJI.results
Bio.Restriction.Restriction.MspR9I.results
Bio.Restriction.Restriction.MssI.results
Bio.Restriction.Restriction.MstI.results
Bio.Restriction.Restriction.MunI.results
Bio.Restriction.Restriction.Mva1269I.results
Bio.Restriction.Restriction.MvaI.results
Bio.Restriction.Restriction.MvnI.results
Bio.Restriction.Restriction.MvrI.results
Bio.Restriction.Restriction.MwoI.results
Bio.Restriction.Restriction.NaeI.results
Bio.Restriction.Restriction.NarI.results
Bio.Restriction.Restriction.NciI.results
Bio.Restriction.Restriction.NcoI.results
Bio.Restriction.Restriction.NdeI.results
Bio.Restriction.Restriction.NdeII.results
Bio.Restriction.Restriction.NgoAVIII.results
Bio.Restriction.Restriction.NgoMIV.results
Bio.Restriction.Restriction.NhaXI.results
Bio.Restriction.Restriction.NheI.results
Bio.Restriction.Restriction.NlaCI.results
Bio.Restriction.Restriction.NlaIII.results
Bio.Restriction.Restriction.NlaIV.results
Bio.Restriction.Restriction.Nli3877I.results
Bio.Restriction.Restriction.NmeAIII.results
Bio.Restriction.Restriction.NmeDI.results
Bio.Restriction.Restriction.NmuCI.results
Bio.Restriction.Restriction.NotI.results
Bio.Restriction.Restriction.NruI.results
Bio.Restriction.Restriction.NsbI.results
Bio.Restriction.Restriction.NsiI.results
Bio.Restriction.Restriction.NspBII.results
Bio.Restriction.Restriction.NspI.results
Bio.Restriction.Restriction.NspV.results
Bio.Restriction.Restriction.OliI.results
Bio.Restriction.Restriction.PabI.results
Bio.Restriction.Restriction.PacI.results
Bio.Restriction.Restriction.PaeI.results
Bio.Restriction.Restriction.PaeR7I.results
Bio.Restriction.Restriction.PagI.results
Bio.Restriction.Restriction.PalAI.results
Bio.Restriction.Restriction.PasI.results
Bio.Restriction.Restriction.PauI.results
Bio.Restriction.Restriction.PceI.results
Bio.Restriction.Restriction.PciI.results
Bio.Restriction.Restriction.PciSI.results
Bio.Restriction.Restriction.PcsI.results
Bio.Restriction.Restriction.PctI.results
Bio.Restriction.Restriction.PdiI.results
Bio.Restriction.Restriction.PdmI.results
Bio.Restriction.Restriction.PenI.results
Bio.Restriction.Restriction.PfeI.results
Bio.Restriction.Restriction.Pfl1108I.results
Bio.Restriction.Restriction.Pfl23II.results
Bio.Restriction.Restriction.PflFI.results
Bio.Restriction.Restriction.PflMI.results
Bio.Restriction.Restriction.PfoI.results
Bio.Restriction.Restriction.PinAI.results
Bio.Restriction.Restriction.PlaDI.results
Bio.Restriction.Restriction.Ple19I.results
Bio.Restriction.Restriction.PleI.results
Bio.Restriction.Restriction.PluTI.results
Bio.Restriction.Restriction.PmaCI.results
Bio.Restriction.Restriction.PmeI.results
Bio.Restriction.Restriction.PmlI.results
Bio.Restriction.Restriction.PpiI.results
Bio.Restriction.Restriction.PpsI.results
Bio.Restriction.Restriction.Ppu10I.results
Bio.Restriction.Restriction.Ppu21I.results
Bio.Restriction.Restriction.PpuMI.results
Bio.Restriction.Restriction.PscI.results
Bio.Restriction.Restriction.PshAI.results
Bio.Restriction.Restriction.PshBI.results
Bio.Restriction.Restriction.PsiI.results
Bio.Restriction.Restriction.Psp03I.results
Bio.Restriction.Restriction.Psp124BI.results
Bio.Restriction.Restriction.Psp1406I.results
Bio.Restriction.Restriction.Psp5II.results
Bio.Restriction.Restriction.Psp6I.results
Bio.Restriction.Restriction.PspCI.results
Bio.Restriction.Restriction.PspEI.results
Bio.Restriction.Restriction.PspGI.results
Bio.Restriction.Restriction.PspLI.results
Bio.Restriction.Restriction.PspN4I.results
Bio.Restriction.Restriction.PspOMI.results
Bio.Restriction.Restriction.PspOMII.results
Bio.Restriction.Restriction.PspPI.results
Bio.Restriction.Restriction.PspPPI.results
Bio.Restriction.Restriction.PspPRI.results
Bio.Restriction.Restriction.PspXI.results
Bio.Restriction.Restriction.PsrI.results
Bio.Restriction.Restriction.PssI.results
Bio.Restriction.Restriction.PstI.results
Bio.Restriction.Restriction.PstNI.results
Bio.Restriction.Restriction.PsuI.results
Bio.Restriction.Restriction.PsyI.results
Bio.Restriction.Restriction.PteI.results
Bio.Restriction.Restriction.PvuI.results
Bio.Restriction.Restriction.PvuII.results
Bio.Restriction.Restriction.R2_BceSIV.results
Bio.Restriction.Restriction.RceI.results
Bio.Restriction.Restriction.RdeGBI.results
Bio.Restriction.Restriction.RdeGBII.results
Bio.Restriction.Restriction.RdeGBIII.results
Bio.Restriction.Restriction.RflFIII.results
Bio.Restriction.Restriction.RgaI.results
Bio.Restriction.Restriction.RigI.results
Bio.Restriction.Restriction.RlaI.results
Bio.Restriction.Restriction.RleAI.results
Bio.Restriction.Restriction.RpaB5I.results
Bio.Restriction.Restriction.RpaBI.results
Bio.Restriction.Restriction.RpaI.results
Bio.Restriction.Restriction.RpaTI.results
Bio.Restriction.Restriction.RruI.results
Bio.Restriction.Restriction.RsaI.results
Bio.Restriction.Restriction.RsaNI.results
Bio.Restriction.Restriction.RseI.results
Bio.Restriction.Restriction.Rsr2I.results
Bio.Restriction.Restriction.RsrII.results
Bio.Restriction.Restriction.SacI.results
Bio.Restriction.Restriction.SacII.results
Bio.Restriction.Restriction.SalI.results
Bio.Restriction.Restriction.SanDI.results
Bio.Restriction.Restriction.SapI.results
Bio.Restriction.Restriction.SaqAI.results
Bio.Restriction.Restriction.SatI.results
Bio.Restriction.Restriction.Sau3AI.results
Bio.Restriction.Restriction.Sau96I.results
Bio.Restriction.Restriction.SauI.results
Bio.Restriction.Restriction.SbfI.results
Bio.Restriction.Restriction.ScaI.results
Bio.Restriction.Restriction.SchI.results
Bio.Restriction.Restriction.SciI.results
Bio.Restriction.Restriction.ScrFI.results
Bio.Restriction.Restriction.SdaI.results
Bio.Restriction.Restriction.SdeAI.results
Bio.Restriction.Restriction.SdeOSI.results
Bio.Restriction.Restriction.SduI.results
Bio.Restriction.Restriction.SecI.results
Bio.Restriction.Restriction.SelI.results
Bio.Restriction.Restriction.SetI.results
Bio.Restriction.Restriction.SexAI.results
Bio.Restriction.Restriction.SfaAI.results
Bio.Restriction.Restriction.SfaNI.results
Bio.Restriction.Restriction.SfcI.results
Bio.Restriction.Restriction.SfeI.results
Bio.Restriction.Restriction.SfiI.results
Bio.Restriction.Restriction.SfoI.results
Bio.Restriction.Restriction.Sfr274I.results
Bio.Restriction.Restriction.Sfr303I.results
Bio.Restriction.Restriction.SfuI.results
Bio.Restriction.Restriction.SgeI.results
Bio.Restriction.Restriction.SgfI.results
Bio.Restriction.Restriction.SgrAI.results
Bio.Restriction.Restriction.SgrBI.results
Bio.Restriction.Restriction.SgrDI.results
Bio.Restriction.Restriction.SgrTI.results
Bio.Restriction.Restriction.SgsI.results
Bio.Restriction.Restriction.SimI.results
Bio.Restriction.Restriction.SlaI.results
Bio.Restriction.Restriction.SmaI.results
Bio.Restriction.Restriction.SmiI.results
Bio.Restriction.Restriction.SmiMI.results
Bio.Restriction.Restriction.SmlI.results
Bio.Restriction.Restriction.SmoI.results
Bio.Restriction.Restriction.SnaBI.results
Bio.Restriction.Restriction.SnaI.results
Bio.Restriction.Restriction.Sno506I.results
Bio.Restriction.Restriction.SpeI.results
Bio.Restriction.Restriction.SphI.results
Bio.Restriction.Restriction.SplI.results
Bio.Restriction.Restriction.SpoDI.results
Bio.Restriction.Restriction.SrfI.results
Bio.Restriction.Restriction.Sse232I.results
Bio.Restriction.Restriction.Sse8387I.results
Bio.Restriction.Restriction.Sse8647I.results
Bio.Restriction.Restriction.Sse9I.results
Bio.Restriction.Restriction.SseBI.results
Bio.Restriction.Restriction.SsiI.results
Bio.Restriction.Restriction.SspD5I.results
Bio.Restriction.Restriction.SspDI.results
Bio.Restriction.Restriction.SspI.results
Bio.Restriction.Restriction.SstE37I.results
Bio.Restriction.Restriction.SstI.results
Bio.Restriction.Restriction.Sth132I.results
Bio.Restriction.Restriction.Sth302II.results
Bio.Restriction.Restriction.StrI.results
Bio.Restriction.Restriction.StsI.results
Bio.Restriction.Restriction.StuI.results
Bio.Restriction.Restriction.StyD4I.results
Bio.Restriction.Restriction.StyI.results
Bio.Restriction.Restriction.SwaI.results
Bio.Restriction.Restriction.TaaI.results
Bio.Restriction.Restriction.TaiI.results
Bio.Restriction.Restriction.TaqI.results
Bio.Restriction.Restriction.TaqII.results
Bio.Restriction.Restriction.TasI.results
Bio.Restriction.Restriction.TatI.results
Bio.Restriction.Restriction.TauI.results
Bio.Restriction.Restriction.TfiI.results
Bio.Restriction.Restriction.Tru1I.results
Bio.Restriction.Restriction.Tru9I.results
Bio.Restriction.Restriction.TscAI.results
Bio.Restriction.Restriction.TseFI.results
Bio.Restriction.Restriction.TseI.results
Bio.Restriction.Restriction.TsoI.results
Bio.Restriction.Restriction.Tsp45I.results
Bio.Restriction.Restriction.Tsp4CI.results
Bio.Restriction.Restriction.TspDTI.results
Bio.Restriction.Restriction.TspEI.results
Bio.Restriction.Restriction.TspGWI.results
Bio.Restriction.Restriction.TspMI.results
Bio.Restriction.Restriction.TspRI.results
Bio.Restriction.Restriction.TssI.results
Bio.Restriction.Restriction.TstI.results
Bio.Restriction.Restriction.TsuI.results
Bio.Restriction.Restriction.Tth111I.results
Bio.Restriction.Restriction.Tth111II.results
Bio.Restriction.Restriction.UbaF11I.results
Bio.Restriction.Restriction.UbaF12I.results
Bio.Restriction.Restriction.UbaF13I.results
Bio.Restriction.Restriction.UbaF14I.results
Bio.Restriction.Restriction.UbaF9I.results
Bio.Restriction.Restriction.UbaPI.results
Bio.Restriction.Restriction.UcoMSI.results
Bio.Restriction.Restriction.UnbI.results
Bio.Restriction.Restriction.Van91I.results
Bio.Restriction.Restriction.Vha464I.results
Bio.Restriction.Restriction.VneI.results
Bio.Restriction.Restriction.VpaK11AI.results
Bio.Restriction.Restriction.VpaK11BI.results
Bio.Restriction.Restriction.VspI.results
Bio.Restriction.Restriction.WviI.results
Bio.Restriction.Restriction.XagI.results
Bio.Restriction.Restriction.XapI.results
Bio.Restriction.Restriction.XbaI.results
Bio.Restriction.Restriction.XceI.results
Bio.Restriction.Restriction.XcmI.results
Bio.Restriction.Restriction.XhoI.results
Bio.Restriction.Restriction.XhoII.results
Bio.Restriction.Restriction.XmaI.results
Bio.Restriction.Restriction.XmaIII.results
Bio.Restriction.Restriction.XmaJI.results
Bio.Restriction.Restriction.XmiI.results
Bio.Restriction.Restriction.XmnI.results
Bio.Restriction.Restriction.XspI.results
Bio.Restriction.Restriction.YkrI.results
Bio.Restriction.Restriction.ZraI.results
Bio.Restriction.Restriction.ZrmI.results
Bio.Restriction.Restriction.Zsp2I.results" class="py-name" href="#" onclick="return doclink('link-153', 'results', 'link-142');">results</a></tt> <tt class="py-op">=</tt> <tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-154" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetchall
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetchall" class="py-name" href="#" onclick="return doclink('link-154', 'execute_and_fetchall', 'link-99');">execute_and_fetchall</a></tt><tt class="py-op">(</tt> </tt>
<a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line">            <tt class="py-string">"SELECT location_id, start_pos, end_pos, strand"</tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line">            <tt class="py-string">" FROM location"</tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line">            <tt class="py-string">" WHERE seqfeature_id = %s"</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line">            <tt class="py-string">" ORDER BY rank"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">seqfeature_id</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line">        <tt class="py-name">locations</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line">        <tt class="py-comment"># convert to Python standard form</tt> </tt>
<a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line">        <tt class="py-comment"># Convert strand = 0 to strand = None</tt> </tt>
<a name="L247"></a><tt class="py-lineno">247</tt>  <tt class="py-line">        <tt class="py-comment"># re: comment in Loader.py:</tt> </tt>
<a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line">        <tt class="py-comment"># Biopython uses None when we don't know strand information but</tt> </tt>
<a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line">        <tt class="py-comment"># BioSQL requires something (non null) and sets this as zero</tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line">        <tt class="py-comment"># So we'll use the strand or 0 if Biopython spits out None</tt> </tt>
<a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">location_id</tt><tt class="py-op">,</tt> <tt id="link-155" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-155', 'start', 'link-18');">start</a></tt><tt class="py-op">,</tt> <tt id="link-156" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.end=Bio.SeqFeature.CompoundLocation-class.html#end,Variable Bio.SeqFeature.FeatureLocation.end=Bio.SeqFeature.FeatureLocation-class.html#end"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-156', 'end', 'link-156');">end</a></tt><tt class="py-op">,</tt> <tt id="link-157" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HSPConsumer.strand()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#strand,Variable Bio.SeqFeature.CompoundLocation.strand=Bio.SeqFeature.CompoundLocation-class.html#strand,Variable Bio.SeqFeature.FeatureLocation.strand=Bio.SeqFeature.FeatureLocation-class.html#strand,Variable Bio.SeqFeature.SeqFeature.strand=Bio.SeqFeature.SeqFeature-class.html#strand"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-157', 'strand', 'link-157');">strand</a></tt> <tt class="py-keyword">in</tt> <tt id="link-158" class="py-name"><a title="Bio.Restriction.Restriction.AanI.results
Bio.Restriction.Restriction.AarI.results
Bio.Restriction.Restriction.AasI.results
Bio.Restriction.Restriction.AatII.results
Bio.Restriction.Restriction.AbaSI.results
Bio.Restriction.Restriction.AbsI.results
Bio.Restriction.Restriction.Acc16I.results
Bio.Restriction.Restriction.Acc36I.results
Bio.Restriction.Restriction.Acc65I.results
Bio.Restriction.Restriction.AccB1I.results
Bio.Restriction.Restriction.AccB7I.results
Bio.Restriction.Restriction.AccBSI.results
Bio.Restriction.Restriction.AccI.results
Bio.Restriction.Restriction.AccII.results
Bio.Restriction.Restriction.AccIII.results
Bio.Restriction.Restriction.AceIII.results
Bio.Restriction.Restriction.AciI.results
Bio.Restriction.Restriction.AclI.results
Bio.Restriction.Restriction.AclWI.results
Bio.Restriction.Restriction.AcoI.results
Bio.Restriction.Restriction.AcsI.results
Bio.Restriction.Restriction.AcuI.results
Bio.Restriction.Restriction.AcvI.results
Bio.Restriction.Restriction.AcyI.results
Bio.Restriction.Restriction.AdeI.results
Bio.Restriction.Restriction.AfaI.results
Bio.Restriction.Restriction.AfeI.results
Bio.Restriction.Restriction.AfiI.results
Bio.Restriction.Restriction.AflII.results
Bio.Restriction.Restriction.AflIII.results
Bio.Restriction.Restriction.AgeI.results
Bio.Restriction.Restriction.AgsI.results
Bio.Restriction.Restriction.AhaIII.results
Bio.Restriction.Restriction.AhdI.results
Bio.Restriction.Restriction.AhlI.results
Bio.Restriction.Restriction.AjiI.results
Bio.Restriction.Restriction.AjnI.results
Bio.Restriction.Restriction.AjuI.results
Bio.Restriction.Restriction.AleI.results
Bio.Restriction.Restriction.AlfI.results
Bio.Restriction.Restriction.AloI.results
Bio.Restriction.Restriction.AluBI.results
Bio.Restriction.Restriction.AluI.results
Bio.Restriction.Restriction.Alw21I.results
Bio.Restriction.Restriction.Alw26I.results
Bio.Restriction.Restriction.Alw44I.results
Bio.Restriction.Restriction.AlwFI.results
Bio.Restriction.Restriction.AlwI.results
Bio.Restriction.Restriction.AlwNI.results
Bio.Restriction.Restriction.Ama87I.results
Bio.Restriction.Restriction.Aor13HI.results
Bio.Restriction.Restriction.Aor51HI.results
Bio.Restriction.Restriction.AoxI.results
Bio.Restriction.Restriction.ApaBI.results
Bio.Restriction.Restriction.ApaI.results
Bio.Restriction.Restriction.ApaLI.results
Bio.Restriction.Restriction.ApeKI.results
Bio.Restriction.Restriction.ApoI.results
Bio.Restriction.Restriction.ApyPI.results
Bio.Restriction.Restriction.AquII.results
Bio.Restriction.Restriction.AquIII.results
Bio.Restriction.Restriction.AquIV.results
Bio.Restriction.Restriction.ArsI.results
Bio.Restriction.Restriction.AscI.results
Bio.Restriction.Restriction.AseI.results
Bio.Restriction.Restriction.Asi256I.results
Bio.Restriction.Restriction.AsiGI.results
Bio.Restriction.Restriction.AsiSI.results
Bio.Restriction.Restriction.Asp700I.results
Bio.Restriction.Restriction.Asp718I.results
Bio.Restriction.Restriction.AspA2I.results
Bio.Restriction.Restriction.AspBHI.results
Bio.Restriction.Restriction.AspLEI.results
Bio.Restriction.Restriction.AspS9I.results
Bio.Restriction.Restriction.AssI.results
Bio.Restriction.Restriction.AsuC2I.results
Bio.Restriction.Restriction.AsuHPI.results
Bio.Restriction.Restriction.AsuI.results
Bio.Restriction.Restriction.AsuII.results
Bio.Restriction.Restriction.AsuNHI.results
Bio.Restriction.Restriction.AvaI.results
Bio.Restriction.Restriction.AvaII.results
Bio.Restriction.Restriction.AvaIII.results
Bio.Restriction.Restriction.AvrII.results
Bio.Restriction.Restriction.AxyI.results
Bio.Restriction.Restriction.BaeGI.results
Bio.Restriction.Restriction.BaeI.results
Bio.Restriction.Restriction.BalI.results
Bio.Restriction.Restriction.BamHI.results
Bio.Restriction.Restriction.BanI.results
Bio.Restriction.Restriction.BanII.results
Bio.Restriction.Restriction.BarI.results
Bio.Restriction.Restriction.BasI.results
Bio.Restriction.Restriction.BauI.results
Bio.Restriction.Restriction.Bbr7I.results
Bio.Restriction.Restriction.BbrPI.results
Bio.Restriction.Restriction.BbsI.results
Bio.Restriction.Restriction.Bbv12I.results
Bio.Restriction.Restriction.BbvCI.results
Bio.Restriction.Restriction.BbvI.results
Bio.Restriction.Restriction.BbvII.results
Bio.Restriction.Restriction.BccI.results
Bio.Restriction.Restriction.Bce83I.results
Bio.Restriction.Restriction.BceAI.results
Bio.Restriction.Restriction.BcefI.results
Bio.Restriction.Restriction.BcgI.results
Bio.Restriction.Restriction.BciT130I.results
Bio.Restriction.Restriction.BciVI.results
Bio.Restriction.Restriction.BclI.results
Bio.Restriction.Restriction.BcnI.results
Bio.Restriction.Restriction.BcoDI.results
Bio.Restriction.Restriction.BcuI.results
Bio.Restriction.Restriction.BdaI.results
Bio.Restriction.Restriction.BetI.results
Bio.Restriction.Restriction.BfaI.results
Bio.Restriction.Restriction.BfiI.results
Bio.Restriction.Restriction.BfmI.results
Bio.Restriction.Restriction.BfoI.results
Bio.Restriction.Restriction.BfrI.results
Bio.Restriction.Restriction.BfuAI.results
Bio.Restriction.Restriction.BfuCI.results
Bio.Restriction.Restriction.BfuI.results
Bio.Restriction.Restriction.BglI.results
Bio.Restriction.Restriction.BglII.results
Bio.Restriction.Restriction.BinI.results
Bio.Restriction.Restriction.BisI.results
Bio.Restriction.Restriction.BlnI.results
Bio.Restriction.Restriction.BlpI.results
Bio.Restriction.Restriction.BlsI.results
Bio.Restriction.Restriction.BmcAI.results
Bio.Restriction.Restriction.Bme1390I.results
Bio.Restriction.Restriction.Bme18I.results
Bio.Restriction.Restriction.BmeDI.results
Bio.Restriction.Restriction.BmeRI.results
Bio.Restriction.Restriction.BmeT110I.results
Bio.Restriction.Restriction.BmgBI.results
Bio.Restriction.Restriction.BmgI.results
Bio.Restriction.Restriction.BmgT120I.results
Bio.Restriction.Restriction.BmiI.results
Bio.Restriction.Restriction.BmrFI.results
Bio.Restriction.Restriction.BmrI.results
Bio.Restriction.Restriction.BmsI.results
Bio.Restriction.Restriction.BmtI.results
Bio.Restriction.Restriction.BmuI.results
Bio.Restriction.Restriction.BoxI.results
Bio.Restriction.Restriction.BpiI.results
Bio.Restriction.Restriction.BplI.results
Bio.Restriction.Restriction.BpmI.results
Bio.Restriction.Restriction.Bpu10I.results
Bio.Restriction.Restriction.Bpu1102I.results
Bio.Restriction.Restriction.Bpu14I.results
Bio.Restriction.Restriction.BpuEI.results
Bio.Restriction.Restriction.BpuMI.results
Bio.Restriction.Restriction.BpvUI.results
Bio.Restriction.Restriction.Bsa29I.results
Bio.Restriction.Restriction.BsaAI.results
Bio.Restriction.Restriction.BsaBI.results
Bio.Restriction.Restriction.BsaHI.results
Bio.Restriction.Restriction.BsaI.results
Bio.Restriction.Restriction.BsaJI.results
Bio.Restriction.Restriction.BsaWI.results
Bio.Restriction.Restriction.BsaXI.results
Bio.Restriction.Restriction.BsbI.results
Bio.Restriction.Restriction.Bsc4I.results
Bio.Restriction.Restriction.BscAI.results
Bio.Restriction.Restriction.BscGI.results
Bio.Restriction.Restriction.Bse118I.results
Bio.Restriction.Restriction.Bse1I.results
Bio.Restriction.Restriction.Bse21I.results
Bio.Restriction.Restriction.Bse3DI.results
Bio.Restriction.Restriction.Bse8I.results
Bio.Restriction.Restriction.BseAI.results
Bio.Restriction.Restriction.BseBI.results
Bio.Restriction.Restriction.BseCI.results
Bio.Restriction.Restriction.BseDI.results
Bio.Restriction.Restriction.BseGI.results
Bio.Restriction.Restriction.BseJI.results
Bio.Restriction.Restriction.BseLI.results
Bio.Restriction.Restriction.BseMI.results
Bio.Restriction.Restriction.BseMII.results
Bio.Restriction.Restriction.BseNI.results
Bio.Restriction.Restriction.BsePI.results
Bio.Restriction.Restriction.BseRI.results
Bio.Restriction.Restriction.BseSI.results
Bio.Restriction.Restriction.BseX3I.results
Bio.Restriction.Restriction.BseXI.results
Bio.Restriction.Restriction.BseYI.results
Bio.Restriction.Restriction.BsgI.results
Bio.Restriction.Restriction.Bsh1236I.results
Bio.Restriction.Restriction.Bsh1285I.results
Bio.Restriction.Restriction.BshFI.results
Bio.Restriction.Restriction.BshNI.results
Bio.Restriction.Restriction.BshTI.results
Bio.Restriction.Restriction.BshVI.results
Bio.Restriction.Restriction.BsiEI.results
Bio.Restriction.Restriction.BsiHKAI.results
Bio.Restriction.Restriction.BsiHKCI.results
Bio.Restriction.Restriction.BsiI.results
Bio.Restriction.Restriction.BsiSI.results
Bio.Restriction.Restriction.BsiWI.results
Bio.Restriction.Restriction.BsiYI.results
Bio.Restriction.Restriction.BslFI.results
Bio.Restriction.Restriction.BslI.results
Bio.Restriction.Restriction.BsmAI.results
Bio.Restriction.Restriction.BsmBI.results
Bio.Restriction.Restriction.BsmFI.results
Bio.Restriction.Restriction.BsmI.results
Bio.Restriction.Restriction.BsnI.results
Bio.Restriction.Restriction.Bso31I.results
Bio.Restriction.Restriction.BsoBI.results
Bio.Restriction.Restriction.Bsp119I.results
Bio.Restriction.Restriction.Bsp120I.results
Bio.Restriction.Restriction.Bsp1286I.results
Bio.Restriction.Restriction.Bsp13I.results
Bio.Restriction.Restriction.Bsp1407I.results
Bio.Restriction.Restriction.Bsp143I.results
Bio.Restriction.Restriction.Bsp1720I.results
Bio.Restriction.Restriction.Bsp19I.results
Bio.Restriction.Restriction.Bsp24I.results
Bio.Restriction.Restriction.Bsp68I.results
Bio.Restriction.Restriction.BspACI.results
Bio.Restriction.Restriction.BspCNI.results
Bio.Restriction.Restriction.BspD6I.results
Bio.Restriction.Restriction.BspDI.results
Bio.Restriction.Restriction.BspEI.results
Bio.Restriction.Restriction.BspFNI.results
Bio.Restriction.Restriction.BspGI.results
Bio.Restriction.Restriction.BspHI.results
Bio.Restriction.Restriction.BspLI.results
Bio.Restriction.Restriction.BspLU11I.results
Bio.Restriction.Restriction.BspMI.results
Bio.Restriction.Restriction.BspMII.results
Bio.Restriction.Restriction.BspNCI.results
Bio.Restriction.Restriction.BspOI.results
Bio.Restriction.Restriction.BspPI.results
Bio.Restriction.Restriction.BspQI.results
Bio.Restriction.Restriction.BspT104I.results
Bio.Restriction.Restriction.BspT107I.results
Bio.Restriction.Restriction.BspTI.results
Bio.Restriction.Restriction.BsrBI.results
Bio.Restriction.Restriction.BsrDI.results
Bio.Restriction.Restriction.BsrFI.results
Bio.Restriction.Restriction.BsrGI.results
Bio.Restriction.Restriction.BsrI.results
Bio.Restriction.Restriction.BsrSI.results
Bio.Restriction.Restriction.BssAI.results
Bio.Restriction.Restriction.BssECI.results
Bio.Restriction.Restriction.BssHII.results
Bio.Restriction.Restriction.BssKI.results
Bio.Restriction.Restriction.BssMI.results
Bio.Restriction.Restriction.BssNAI.results
Bio.Restriction.Restriction.BssNI.results
Bio.Restriction.Restriction.BssSI.results
Bio.Restriction.Restriction.BssT1I.results
Bio.Restriction.Restriction.Bst1107I.results
Bio.Restriction.Restriction.Bst2BI.results
Bio.Restriction.Restriction.Bst2UI.results
Bio.Restriction.Restriction.Bst4CI.results
Bio.Restriction.Restriction.Bst6I.results
Bio.Restriction.Restriction.BstACI.results
Bio.Restriction.Restriction.BstAFI.results
Bio.Restriction.Restriction.BstAPI.results
Bio.Restriction.Restriction.BstAUI.results
Bio.Restriction.Restriction.BstBAI.results
Bio.Restriction.Restriction.BstBI.results
Bio.Restriction.Restriction.BstC8I.results
Bio.Restriction.Restriction.BstDEI.results
Bio.Restriction.Restriction.BstDSI.results
Bio.Restriction.Restriction.BstEII.results
Bio.Restriction.Restriction.BstENI.results
Bio.Restriction.Restriction.BstF5I.results
Bio.Restriction.Restriction.BstFNI.results
Bio.Restriction.Restriction.BstH2I.results
Bio.Restriction.Restriction.BstHHI.results
Bio.Restriction.Restriction.BstKTI.results
Bio.Restriction.Restriction.BstMAI.results
Bio.Restriction.Restriction.BstMBI.results
Bio.Restriction.Restriction.BstMCI.results
Bio.Restriction.Restriction.BstMWI.results
Bio.Restriction.Restriction.BstNI.results
Bio.Restriction.Restriction.BstNSI.results
Bio.Restriction.Restriction.BstOI.results
Bio.Restriction.Restriction.BstPAI.results
Bio.Restriction.Restriction.BstPI.results
Bio.Restriction.Restriction.BstSCI.results
Bio.Restriction.Restriction.BstSFI.results
Bio.Restriction.Restriction.BstSLI.results
Bio.Restriction.Restriction.BstSNI.results
Bio.Restriction.Restriction.BstUI.results
Bio.Restriction.Restriction.BstV1I.results
Bio.Restriction.Restriction.BstV2I.results
Bio.Restriction.Restriction.BstX2I.results
Bio.Restriction.Restriction.BstXI.results
Bio.Restriction.Restriction.BstYI.results
Bio.Restriction.Restriction.BstZ17I.results
Bio.Restriction.Restriction.BstZI.results
Bio.Restriction.Restriction.Bsu15I.results
Bio.Restriction.Restriction.Bsu36I.results
Bio.Restriction.Restriction.BsuI.results
Bio.Restriction.Restriction.BsuRI.results
Bio.Restriction.Restriction.BtgI.results
Bio.Restriction.Restriction.BtgZI.results
Bio.Restriction.Restriction.BthCI.results
Bio.Restriction.Restriction.BtrI.results
Bio.Restriction.Restriction.BtsCI.results
Bio.Restriction.Restriction.BtsI.results
Bio.Restriction.Restriction.BtsIMutI.results
Bio.Restriction.Restriction.BtuMI.results
Bio.Restriction.Restriction.BveI.results
Bio.Restriction.Restriction.Cac8I.results
Bio.Restriction.Restriction.CaiI.results
Bio.Restriction.Restriction.CauII.results
Bio.Restriction.Restriction.CchII.results
Bio.Restriction.Restriction.CchIII.results
Bio.Restriction.Restriction.CciI.results
Bio.Restriction.Restriction.CciNI.results
Bio.Restriction.Restriction.Cdi630V.results
Bio.Restriction.Restriction.CdiI.results
Bio.Restriction.Restriction.CdpI.results
Bio.Restriction.Restriction.CfoI.results
Bio.Restriction.Restriction.Cfr10I.results
Bio.Restriction.Restriction.Cfr13I.results
Bio.Restriction.Restriction.Cfr42I.results
Bio.Restriction.Restriction.Cfr9I.results
Bio.Restriction.Restriction.CfrI.results
Bio.Restriction.Restriction.Cgl13032I.results
Bio.Restriction.Restriction.Cgl13032II.results
Bio.Restriction.Restriction.ChaI.results
Bio.Restriction.Restriction.CjeFIII.results
Bio.Restriction.Restriction.CjeFV.results
Bio.Restriction.Restriction.CjeI.results
Bio.Restriction.Restriction.CjeNII.results
Bio.Restriction.Restriction.CjeNIII.results
Bio.Restriction.Restriction.CjeP659IV.results
Bio.Restriction.Restriction.CjePI.results
Bio.Restriction.Restriction.CjuI.results
Bio.Restriction.Restriction.CjuII.results
Bio.Restriction.Restriction.ClaI.results
Bio.Restriction.Restriction.CpoI.results
Bio.Restriction.Restriction.CseI.results
Bio.Restriction.Restriction.CsiI.results
Bio.Restriction.Restriction.Csp6I.results
Bio.Restriction.Restriction.CspAI.results
Bio.Restriction.Restriction.CspCI.results
Bio.Restriction.Restriction.CspI.results
Bio.Restriction.Restriction.CstMI.results
Bio.Restriction.Restriction.CviAII.results
Bio.Restriction.Restriction.CviJI.results
Bio.Restriction.Restriction.CviKI_1.results
Bio.Restriction.Restriction.CviQI.results
Bio.Restriction.Restriction.CviRI.results
Bio.Restriction.Restriction.DdeI.results
Bio.Restriction.Restriction.DinI.results
Bio.Restriction.Restriction.DpnI.results
Bio.Restriction.Restriction.DpnII.results
Bio.Restriction.Restriction.DraI.results
Bio.Restriction.Restriction.DraII.results
Bio.Restriction.Restriction.DraIII.results
Bio.Restriction.Restriction.DraRI.results
Bio.Restriction.Restriction.DrdI.results
Bio.Restriction.Restriction.DrdII.results
Bio.Restriction.Restriction.DriI.results
Bio.Restriction.Restriction.DsaI.results
Bio.Restriction.Restriction.DseDI.results
Bio.Restriction.Restriction.EaeI.results
Bio.Restriction.Restriction.EagI.results
Bio.Restriction.Restriction.Eam1104I.results
Bio.Restriction.Restriction.Eam1105I.results
Bio.Restriction.Restriction.EarI.results
Bio.Restriction.Restriction.EciI.results
Bio.Restriction.Restriction.Ecl136II.results
Bio.Restriction.Restriction.EclXI.results
Bio.Restriction.Restriction.Eco105I.results
Bio.Restriction.Restriction.Eco130I.results
Bio.Restriction.Restriction.Eco147I.results
Bio.Restriction.Restriction.Eco24I.results
Bio.Restriction.Restriction.Eco31I.results
Bio.Restriction.Restriction.Eco32I.results
Bio.Restriction.Restriction.Eco47I.results
Bio.Restriction.Restriction.Eco47III.results
Bio.Restriction.Restriction.Eco52I.results
Bio.Restriction.Restriction.Eco53kI.results
Bio.Restriction.Restriction.Eco57I.results
Bio.Restriction.Restriction.Eco57MI.results
Bio.Restriction.Restriction.Eco72I.results
Bio.Restriction.Restriction.Eco81I.results
Bio.Restriction.Restriction.Eco88I.results
Bio.Restriction.Restriction.Eco91I.results
Bio.Restriction.Restriction.EcoHI.results
Bio.Restriction.Restriction.EcoICRI.results
Bio.Restriction.Restriction.EcoNI.results
Bio.Restriction.Restriction.EcoO109I.results
Bio.Restriction.Restriction.EcoO65I.results
Bio.Restriction.Restriction.EcoRI.results
Bio.Restriction.Restriction.EcoRII.results
Bio.Restriction.Restriction.EcoRV.results
Bio.Restriction.Restriction.EcoT14I.results
Bio.Restriction.Restriction.EcoT22I.results
Bio.Restriction.Restriction.EcoT38I.results
Bio.Restriction.Restriction.EgeI.results
Bio.Restriction.Restriction.EheI.results
Bio.Restriction.Restriction.ErhI.results
Bio.Restriction.Restriction.EsaBC3I.results
Bio.Restriction.Restriction.EsaSSI.results
Bio.Restriction.Restriction.Esp3I.results
Bio.Restriction.Restriction.EspI.results
Bio.Restriction.Restriction.FaeI.results
Bio.Restriction.Restriction.FaiI.results
Bio.Restriction.Restriction.FalI.results
Bio.Restriction.Restriction.FaqI.results
Bio.Restriction.Restriction.FatI.results
Bio.Restriction.Restriction.FauI.results
Bio.Restriction.Restriction.FauNDI.results
Bio.Restriction.Restriction.FbaI.results
Bio.Restriction.Restriction.FblI.results
Bio.Restriction.Restriction.FinI.results
Bio.Restriction.Restriction.FmuI.results
Bio.Restriction.Restriction.Fnu4HI.results
Bio.Restriction.Restriction.FnuDII.results
Bio.Restriction.Restriction.FokI.results
Bio.Restriction.Restriction.FriOI.results
Bio.Restriction.Restriction.FseI.results
Bio.Restriction.Restriction.Fsp4HI.results
Bio.Restriction.Restriction.FspAI.results
Bio.Restriction.Restriction.FspBI.results
Bio.Restriction.Restriction.FspEI.results
Bio.Restriction.Restriction.FspI.results
Bio.Restriction.Restriction.GauT27I.results
Bio.Restriction.Restriction.GdiII.results
Bio.Restriction.Restriction.GlaI.results
Bio.Restriction.Restriction.GluI.results
Bio.Restriction.Restriction.GsaI.results
Bio.Restriction.Restriction.GsuI.results
Bio.Restriction.Restriction.HaeI.results
Bio.Restriction.Restriction.HaeII.results
Bio.Restriction.Restriction.HaeIII.results
Bio.Restriction.Restriction.HapII.results
Bio.Restriction.Restriction.HauII.results
Bio.Restriction.Restriction.HgaI.results
Bio.Restriction.Restriction.HgiAI.results
Bio.Restriction.Restriction.HgiCI.results
Bio.Restriction.Restriction.HgiEII.results
Bio.Restriction.Restriction.HgiJII.results
Bio.Restriction.Restriction.HhaI.results
Bio.Restriction.Restriction.Hin1I.results
Bio.Restriction.Restriction.Hin1II.results
Bio.Restriction.Restriction.Hin4I.results
Bio.Restriction.Restriction.Hin4II.results
Bio.Restriction.Restriction.Hin6I.results
Bio.Restriction.Restriction.HinP1I.results
Bio.Restriction.Restriction.HincII.results
Bio.Restriction.Restriction.HindII.results
Bio.Restriction.Restriction.HindIII.results
Bio.Restriction.Restriction.HinfI.results
Bio.Restriction.Restriction.HpaI.results
Bio.Restriction.Restriction.HpaII.results
Bio.Restriction.Restriction.HphI.results
Bio.Restriction.Restriction.Hpy166II.results
Bio.Restriction.Restriction.Hpy178III.results
Bio.Restriction.Restriction.Hpy188I.results
Bio.Restriction.Restriction.Hpy188III.results
Bio.Restriction.Restriction.Hpy8I.results
Bio.Restriction.Restriction.Hpy99I.results
Bio.Restriction.Restriction.Hpy99XIII.results
Bio.Restriction.Restriction.Hpy99XIV.results
Bio.Restriction.Restriction.HpyAV.results
Bio.Restriction.Restriction.HpyCH4III.results
Bio.Restriction.Restriction.HpyCH4IV.results
Bio.Restriction.Restriction.HpyCH4V.results
Bio.Restriction.Restriction.HpyF10VI.results
Bio.Restriction.Restriction.HpyF3I.results
Bio.Restriction.Restriction.HpySE526I.results
Bio.Restriction.Restriction.Hsp92I.results
Bio.Restriction.Restriction.Hsp92II.results
Bio.Restriction.Restriction.HspAI.results
Bio.Restriction.Restriction.Jma19592I.results
Bio.Restriction.Restriction.KasI.results
Bio.Restriction.Restriction.KflI.results
Bio.Restriction.Restriction.Kpn2I.results
Bio.Restriction.Restriction.KpnI.results
Bio.Restriction.Restriction.KroI.results
Bio.Restriction.Restriction.Ksp22I.results
Bio.Restriction.Restriction.Ksp632I.results
Bio.Restriction.Restriction.KspAI.results
Bio.Restriction.Restriction.KspI.results
Bio.Restriction.Restriction.Kzo9I.results
Bio.Restriction.Restriction.LguI.results
Bio.Restriction.Restriction.LpnI.results
Bio.Restriction.Restriction.LpnPI.results
Bio.Restriction.Restriction.Lsp1109I.results
Bio.Restriction.Restriction.LweI.results
Bio.Restriction.Restriction.MabI.results
Bio.Restriction.Restriction.MaeI.results
Bio.Restriction.Restriction.MaeII.results
Bio.Restriction.Restriction.MaeIII.results
Bio.Restriction.Restriction.MalI.results
Bio.Restriction.Restriction.MaqI.results
Bio.Restriction.Restriction.MauBI.results
Bio.Restriction.Restriction.MbiI.results
Bio.Restriction.Restriction.MboI.results
Bio.Restriction.Restriction.MboII.results
Bio.Restriction.Restriction.McaTI.results
Bio.Restriction.Restriction.McrI.results
Bio.Restriction.Restriction.MfeI.results
Bio.Restriction.Restriction.MflI.results
Bio.Restriction.Restriction.MhlI.results
Bio.Restriction.Restriction.MjaIV.results
Bio.Restriction.Restriction.MkaDII.results
Bio.Restriction.Restriction.MlsI.results
Bio.Restriction.Restriction.MluCI.results
Bio.Restriction.Restriction.MluI.results
Bio.Restriction.Restriction.MluNI.results
Bio.Restriction.Restriction.Mly113I.results
Bio.Restriction.Restriction.MlyI.results
Bio.Restriction.Restriction.MmeI.results
Bio.Restriction.Restriction.MnlI.results
Bio.Restriction.Restriction.Mph1103I.results
Bio.Restriction.Restriction.MreI.results
Bio.Restriction.Restriction.MroI.results
Bio.Restriction.Restriction.MroNI.results
Bio.Restriction.Restriction.MroXI.results
Bio.Restriction.Restriction.MscI.results
Bio.Restriction.Restriction.MseI.results
Bio.Restriction.Restriction.MslI.results
Bio.Restriction.Restriction.Msp20I.results
Bio.Restriction.Restriction.MspA1I.results
Bio.Restriction.Restriction.MspCI.results
Bio.Restriction.Restriction.MspI.results
Bio.Restriction.Restriction.MspJI.results
Bio.Restriction.Restriction.MspR9I.results
Bio.Restriction.Restriction.MssI.results
Bio.Restriction.Restriction.MstI.results
Bio.Restriction.Restriction.MunI.results
Bio.Restriction.Restriction.Mva1269I.results
Bio.Restriction.Restriction.MvaI.results
Bio.Restriction.Restriction.MvnI.results
Bio.Restriction.Restriction.MvrI.results
Bio.Restriction.Restriction.MwoI.results
Bio.Restriction.Restriction.NaeI.results
Bio.Restriction.Restriction.NarI.results
Bio.Restriction.Restriction.NciI.results
Bio.Restriction.Restriction.NcoI.results
Bio.Restriction.Restriction.NdeI.results
Bio.Restriction.Restriction.NdeII.results
Bio.Restriction.Restriction.NgoAVIII.results
Bio.Restriction.Restriction.NgoMIV.results
Bio.Restriction.Restriction.NhaXI.results
Bio.Restriction.Restriction.NheI.results
Bio.Restriction.Restriction.NlaCI.results
Bio.Restriction.Restriction.NlaIII.results
Bio.Restriction.Restriction.NlaIV.results
Bio.Restriction.Restriction.Nli3877I.results
Bio.Restriction.Restriction.NmeAIII.results
Bio.Restriction.Restriction.NmeDI.results
Bio.Restriction.Restriction.NmuCI.results
Bio.Restriction.Restriction.NotI.results
Bio.Restriction.Restriction.NruI.results
Bio.Restriction.Restriction.NsbI.results
Bio.Restriction.Restriction.NsiI.results
Bio.Restriction.Restriction.NspBII.results
Bio.Restriction.Restriction.NspI.results
Bio.Restriction.Restriction.NspV.results
Bio.Restriction.Restriction.OliI.results
Bio.Restriction.Restriction.PabI.results
Bio.Restriction.Restriction.PacI.results
Bio.Restriction.Restriction.PaeI.results
Bio.Restriction.Restriction.PaeR7I.results
Bio.Restriction.Restriction.PagI.results
Bio.Restriction.Restriction.PalAI.results
Bio.Restriction.Restriction.PasI.results
Bio.Restriction.Restriction.PauI.results
Bio.Restriction.Restriction.PceI.results
Bio.Restriction.Restriction.PciI.results
Bio.Restriction.Restriction.PciSI.results
Bio.Restriction.Restriction.PcsI.results
Bio.Restriction.Restriction.PctI.results
Bio.Restriction.Restriction.PdiI.results
Bio.Restriction.Restriction.PdmI.results
Bio.Restriction.Restriction.PenI.results
Bio.Restriction.Restriction.PfeI.results
Bio.Restriction.Restriction.Pfl1108I.results
Bio.Restriction.Restriction.Pfl23II.results
Bio.Restriction.Restriction.PflFI.results
Bio.Restriction.Restriction.PflMI.results
Bio.Restriction.Restriction.PfoI.results
Bio.Restriction.Restriction.PinAI.results
Bio.Restriction.Restriction.PlaDI.results
Bio.Restriction.Restriction.Ple19I.results
Bio.Restriction.Restriction.PleI.results
Bio.Restriction.Restriction.PluTI.results
Bio.Restriction.Restriction.PmaCI.results
Bio.Restriction.Restriction.PmeI.results
Bio.Restriction.Restriction.PmlI.results
Bio.Restriction.Restriction.PpiI.results
Bio.Restriction.Restriction.PpsI.results
Bio.Restriction.Restriction.Ppu10I.results
Bio.Restriction.Restriction.Ppu21I.results
Bio.Restriction.Restriction.PpuMI.results
Bio.Restriction.Restriction.PscI.results
Bio.Restriction.Restriction.PshAI.results
Bio.Restriction.Restriction.PshBI.results
Bio.Restriction.Restriction.PsiI.results
Bio.Restriction.Restriction.Psp03I.results
Bio.Restriction.Restriction.Psp124BI.results
Bio.Restriction.Restriction.Psp1406I.results
Bio.Restriction.Restriction.Psp5II.results
Bio.Restriction.Restriction.Psp6I.results
Bio.Restriction.Restriction.PspCI.results
Bio.Restriction.Restriction.PspEI.results
Bio.Restriction.Restriction.PspGI.results
Bio.Restriction.Restriction.PspLI.results
Bio.Restriction.Restriction.PspN4I.results
Bio.Restriction.Restriction.PspOMI.results
Bio.Restriction.Restriction.PspOMII.results
Bio.Restriction.Restriction.PspPI.results
Bio.Restriction.Restriction.PspPPI.results
Bio.Restriction.Restriction.PspPRI.results
Bio.Restriction.Restriction.PspXI.results
Bio.Restriction.Restriction.PsrI.results
Bio.Restriction.Restriction.PssI.results
Bio.Restriction.Restriction.PstI.results
Bio.Restriction.Restriction.PstNI.results
Bio.Restriction.Restriction.PsuI.results
Bio.Restriction.Restriction.PsyI.results
Bio.Restriction.Restriction.PteI.results
Bio.Restriction.Restriction.PvuI.results
Bio.Restriction.Restriction.PvuII.results
Bio.Restriction.Restriction.R2_BceSIV.results
Bio.Restriction.Restriction.RceI.results
Bio.Restriction.Restriction.RdeGBI.results
Bio.Restriction.Restriction.RdeGBII.results
Bio.Restriction.Restriction.RdeGBIII.results
Bio.Restriction.Restriction.RflFIII.results
Bio.Restriction.Restriction.RgaI.results
Bio.Restriction.Restriction.RigI.results
Bio.Restriction.Restriction.RlaI.results
Bio.Restriction.Restriction.RleAI.results
Bio.Restriction.Restriction.RpaB5I.results
Bio.Restriction.Restriction.RpaBI.results
Bio.Restriction.Restriction.RpaI.results
Bio.Restriction.Restriction.RpaTI.results
Bio.Restriction.Restriction.RruI.results
Bio.Restriction.Restriction.RsaI.results
Bio.Restriction.Restriction.RsaNI.results
Bio.Restriction.Restriction.RseI.results
Bio.Restriction.Restriction.Rsr2I.results
Bio.Restriction.Restriction.RsrII.results
Bio.Restriction.Restriction.SacI.results
Bio.Restriction.Restriction.SacII.results
Bio.Restriction.Restriction.SalI.results
Bio.Restriction.Restriction.SanDI.results
Bio.Restriction.Restriction.SapI.results
Bio.Restriction.Restriction.SaqAI.results
Bio.Restriction.Restriction.SatI.results
Bio.Restriction.Restriction.Sau3AI.results
Bio.Restriction.Restriction.Sau96I.results
Bio.Restriction.Restriction.SauI.results
Bio.Restriction.Restriction.SbfI.results
Bio.Restriction.Restriction.ScaI.results
Bio.Restriction.Restriction.SchI.results
Bio.Restriction.Restriction.SciI.results
Bio.Restriction.Restriction.ScrFI.results
Bio.Restriction.Restriction.SdaI.results
Bio.Restriction.Restriction.SdeAI.results
Bio.Restriction.Restriction.SdeOSI.results
Bio.Restriction.Restriction.SduI.results
Bio.Restriction.Restriction.SecI.results
Bio.Restriction.Restriction.SelI.results
Bio.Restriction.Restriction.SetI.results
Bio.Restriction.Restriction.SexAI.results
Bio.Restriction.Restriction.SfaAI.results
Bio.Restriction.Restriction.SfaNI.results
Bio.Restriction.Restriction.SfcI.results
Bio.Restriction.Restriction.SfeI.results
Bio.Restriction.Restriction.SfiI.results
Bio.Restriction.Restriction.SfoI.results
Bio.Restriction.Restriction.Sfr274I.results
Bio.Restriction.Restriction.Sfr303I.results
Bio.Restriction.Restriction.SfuI.results
Bio.Restriction.Restriction.SgeI.results
Bio.Restriction.Restriction.SgfI.results
Bio.Restriction.Restriction.SgrAI.results
Bio.Restriction.Restriction.SgrBI.results
Bio.Restriction.Restriction.SgrDI.results
Bio.Restriction.Restriction.SgrTI.results
Bio.Restriction.Restriction.SgsI.results
Bio.Restriction.Restriction.SimI.results
Bio.Restriction.Restriction.SlaI.results
Bio.Restriction.Restriction.SmaI.results
Bio.Restriction.Restriction.SmiI.results
Bio.Restriction.Restriction.SmiMI.results
Bio.Restriction.Restriction.SmlI.results
Bio.Restriction.Restriction.SmoI.results
Bio.Restriction.Restriction.SnaBI.results
Bio.Restriction.Restriction.SnaI.results
Bio.Restriction.Restriction.Sno506I.results
Bio.Restriction.Restriction.SpeI.results
Bio.Restriction.Restriction.SphI.results
Bio.Restriction.Restriction.SplI.results
Bio.Restriction.Restriction.SpoDI.results
Bio.Restriction.Restriction.SrfI.results
Bio.Restriction.Restriction.Sse232I.results
Bio.Restriction.Restriction.Sse8387I.results
Bio.Restriction.Restriction.Sse8647I.results
Bio.Restriction.Restriction.Sse9I.results
Bio.Restriction.Restriction.SseBI.results
Bio.Restriction.Restriction.SsiI.results
Bio.Restriction.Restriction.SspD5I.results
Bio.Restriction.Restriction.SspDI.results
Bio.Restriction.Restriction.SspI.results
Bio.Restriction.Restriction.SstE37I.results
Bio.Restriction.Restriction.SstI.results
Bio.Restriction.Restriction.Sth132I.results
Bio.Restriction.Restriction.Sth302II.results
Bio.Restriction.Restriction.StrI.results
Bio.Restriction.Restriction.StsI.results
Bio.Restriction.Restriction.StuI.results
Bio.Restriction.Restriction.StyD4I.results
Bio.Restriction.Restriction.StyI.results
Bio.Restriction.Restriction.SwaI.results
Bio.Restriction.Restriction.TaaI.results
Bio.Restriction.Restriction.TaiI.results
Bio.Restriction.Restriction.TaqI.results
Bio.Restriction.Restriction.TaqII.results
Bio.Restriction.Restriction.TasI.results
Bio.Restriction.Restriction.TatI.results
Bio.Restriction.Restriction.TauI.results
Bio.Restriction.Restriction.TfiI.results
Bio.Restriction.Restriction.Tru1I.results
Bio.Restriction.Restriction.Tru9I.results
Bio.Restriction.Restriction.TscAI.results
Bio.Restriction.Restriction.TseFI.results
Bio.Restriction.Restriction.TseI.results
Bio.Restriction.Restriction.TsoI.results
Bio.Restriction.Restriction.Tsp45I.results
Bio.Restriction.Restriction.Tsp4CI.results
Bio.Restriction.Restriction.TspDTI.results
Bio.Restriction.Restriction.TspEI.results
Bio.Restriction.Restriction.TspGWI.results
Bio.Restriction.Restriction.TspMI.results
Bio.Restriction.Restriction.TspRI.results
Bio.Restriction.Restriction.TssI.results
Bio.Restriction.Restriction.TstI.results
Bio.Restriction.Restriction.TsuI.results
Bio.Restriction.Restriction.Tth111I.results
Bio.Restriction.Restriction.Tth111II.results
Bio.Restriction.Restriction.UbaF11I.results
Bio.Restriction.Restriction.UbaF12I.results
Bio.Restriction.Restriction.UbaF13I.results
Bio.Restriction.Restriction.UbaF14I.results
Bio.Restriction.Restriction.UbaF9I.results
Bio.Restriction.Restriction.UbaPI.results
Bio.Restriction.Restriction.UcoMSI.results
Bio.Restriction.Restriction.UnbI.results
Bio.Restriction.Restriction.Van91I.results
Bio.Restriction.Restriction.Vha464I.results
Bio.Restriction.Restriction.VneI.results
Bio.Restriction.Restriction.VpaK11AI.results
Bio.Restriction.Restriction.VpaK11BI.results
Bio.Restriction.Restriction.VspI.results
Bio.Restriction.Restriction.WviI.results
Bio.Restriction.Restriction.XagI.results
Bio.Restriction.Restriction.XapI.results
Bio.Restriction.Restriction.XbaI.results
Bio.Restriction.Restriction.XceI.results
Bio.Restriction.Restriction.XcmI.results
Bio.Restriction.Restriction.XhoI.results
Bio.Restriction.Restriction.XhoII.results
Bio.Restriction.Restriction.XmaI.results
Bio.Restriction.Restriction.XmaIII.results
Bio.Restriction.Restriction.XmaJI.results
Bio.Restriction.Restriction.XmiI.results
Bio.Restriction.Restriction.XmnI.results
Bio.Restriction.Restriction.XspI.results
Bio.Restriction.Restriction.YkrI.results
Bio.Restriction.Restriction.ZraI.results
Bio.Restriction.Restriction.ZrmI.results
Bio.Restriction.Restriction.Zsp2I.results" class="py-name" href="#" onclick="return doclink('link-158', 'results', 'link-142');">results</a></tt><tt class="py-op">:</tt> </tt>
<a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-159" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-159', 'start', 'link-18');">start</a></tt><tt class="py-op">:</tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line">                <tt id="link-160" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-160', 'start', 'link-18');">start</a></tt> <tt class="py-op">-=</tt> <tt class="py-number">1</tt> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-161" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-161', 'strand', 'link-157');">strand</a></tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line">                <tt id="link-162" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-162', 'strand', 'link-157');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-163" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-163', 'strand', 'link-157');">strand</a></tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-op">(</tt><tt class="py-op">+</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Invalid strand %s found in database for "</tt> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line">                                 <tt class="py-string">"seqfeature_id %s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-164" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-164', 'strand', 'link-157');">strand</a></tt><tt class="py-op">,</tt> <tt class="py-name">seqfeature_id</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-165" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-165', 'end', 'link-156');">end</a></tt> <tt class="py-op">&lt;</tt> <tt id="link-166" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-166', 'start', 'link-18');">start</a></tt><tt class="py-op">:</tt> </tt>
<a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line">                <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line">                <tt class="py-keyword">from</tt> <tt id="link-167" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-167', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-168" class="py-name" targets="Class Bio.BiopythonWarning=Bio.BiopythonWarning-class.html"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-168', 'BiopythonWarning', 'link-168');">BiopythonWarning</a></tt> </tt>
<a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line">                <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Inverted location start/end (%i and %i) for "</tt> </tt>
<a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line">                              <tt class="py-string">"seqfeature_id %s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-169" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-169', 'start', 'link-18');">start</a></tt><tt class="py-op">,</tt> <tt id="link-170" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-170', 'end', 'link-156');">end</a></tt><tt class="py-op">,</tt> <tt class="py-name">seqfeature_id</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line">                              <tt id="link-171" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-171', 'BiopythonWarning', 'link-168');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L265"></a><tt class="py-lineno">265</tt>  <tt class="py-line">            <tt class="py-name">locations</tt><tt class="py-op">.</tt><tt id="link-172" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-172', 'append', 'link-141');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">location_id</tt><tt class="py-op">,</tt> <tt id="link-173" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-173', 'start', 'link-18');">start</a></tt><tt class="py-op">,</tt> <tt id="link-174" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-174', 'end', 'link-156');">end</a></tt><tt class="py-op">,</tt> <tt id="link-175" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-175', 'strand', 'link-157');">strand</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line">        <tt class="py-comment"># Get possible remote reference information</tt> </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line">        <tt class="py-name">remote_results</tt> <tt class="py-op">=</tt> <tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-176" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetchall
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetchall" class="py-name" href="#" onclick="return doclink('link-176', 'execute_and_fetchall', 'link-99');">execute_and_fetchall</a></tt><tt class="py-op">(</tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line">            <tt class="py-string">"SELECT location_id, dbname, accession, version"</tt> </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line">            <tt class="py-string">" FROM location join dbxref using (dbxref_id)"</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line">            <tt class="py-string">" WHERE seqfeature_id = %s"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">seqfeature_id</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line">        <tt id="link-177" class="py-name" targets="Method BioSQL.BioSeqDatabase.BioSeqDatabase.lookup()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#lookup"><a title="BioSQL.BioSeqDatabase.BioSeqDatabase.lookup" class="py-name" href="#" onclick="return doclink('link-177', 'lookup', 'link-177');">lookup</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">location_id</tt><tt class="py-op">,</tt> <tt class="py-name">dbname</tt><tt class="py-op">,</tt> <tt id="link-178" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-178', 'accession', 'link-133');">accession</a></tt><tt class="py-op">,</tt> <tt id="link-179" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-179', 'version', 'link-134');">version</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">remote_results</tt><tt class="py-op">:</tt> </tt>
<a name="L273"></a><tt class="py-lineno">273</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-180" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-180', 'version', 'link-134');">version</a></tt> <tt class="py-keyword">and</tt> <tt id="link-181" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-181', 'version', 'link-134');">version</a></tt> <tt class="py-op">!=</tt> <tt class="py-string">"0"</tt><tt class="py-op">:</tt> </tt>
<a name="L274"></a><tt class="py-lineno">274</tt>  <tt class="py-line">                <tt class="py-name">v</tt> <tt class="py-op">=</tt> <tt class="py-string">"%s.%s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-182" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-182', 'accession', 'link-133');">accession</a></tt><tt class="py-op">,</tt> <tt id="link-183" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-183', 'version', 'link-134');">version</a></tt><tt class="py-op">)</tt> </tt>
<a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line">                <tt class="py-name">v</tt> <tt class="py-op">=</tt> <tt id="link-184" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-184', 'accession', 'link-133');">accession</a></tt> </tt>
<a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line">            <tt class="py-comment"># subfeature remote location db_ref are stored as a empty string when</tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line">            <tt class="py-comment"># not present</tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">dbname</tt> <tt class="py-op">==</tt> <tt class="py-string">""</tt><tt class="py-op">:</tt> </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line">                <tt class="py-name">dbname</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line">            <tt id="link-185" class="py-name"><a title="BioSQL.BioSeqDatabase.BioSeqDatabase.lookup" class="py-name" href="#" onclick="return doclink('link-185', 'lookup', 'link-177');">lookup</a></tt><tt class="py-op">[</tt><tt class="py-name">location_id</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-op">(</tt><tt class="py-name">dbname</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt><tt class="py-op">)</tt> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line"> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line">        <tt class="py-name">feature</tt> <tt class="py-op">=</tt> <tt id="link-186" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-186', 'SeqFeature', 'link-13');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-187" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-187', 'SeqFeature', 'link-13');">SeqFeature</a></tt><tt class="py-op">(</tt><tt id="link-188" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.type()=Bio.Phylo.PhyloXMLIO.Writer-class.html#type"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-188', 'type', 'link-188');">type</a></tt><tt class="py-op">=</tt><tt class="py-name">seqfeature_type</tt><tt class="py-op">)</tt> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line">        <tt class="py-comment"># Store the key as a private property</tt> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line">        <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt class="py-name">_seqfeature_id</tt> <tt class="py-op">=</tt> <tt class="py-name">seqfeature_id</tt> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line">        <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt class="py-name">qualifiers</tt> <tt class="py-op">=</tt> <tt class="py-name">qualifiers</tt> </tt>
<a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">locations</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt><tt class="py-op">:</tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">locations</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line">            <tt class="py-name">location_id</tt><tt class="py-op">,</tt> <tt id="link-189" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-189', 'start', 'link-18');">start</a></tt><tt class="py-op">,</tt> <tt id="link-190" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-190', 'end', 'link-156');">end</a></tt><tt class="py-op">,</tt> <tt id="link-191" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-191', 'strand', 'link-157');">strand</a></tt> <tt class="py-op">=</tt> <tt class="py-name">locations</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line">            <tt class="py-comment"># See Bug 2677, we currently don't record the location_operator</tt> </tt>
<a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line">            <tt class="py-comment"># For consistency with older versions Biopython, default to "".</tt> </tt>
<a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line">            <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-192" class="py-name" targets="Variable Bio.SeqFeature.SeqFeature.location_operator=Bio.SeqFeature.SeqFeature-class.html#location_operator"><a title="Bio.SeqFeature.SeqFeature.location_operator" class="py-name" href="#" onclick="return doclink('link-192', 'location_operator', 'link-192');">location_operator</a></tt> <tt class="py-op">=</tt> \ </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line">                <tt id="link-193" class="py-name" targets="Function BioSQL.BioSeq._retrieve_location_qualifier_value()=BioSQL.BioSeq-module.html#_retrieve_location_qualifier_value"><a title="BioSQL.BioSeq._retrieve_location_qualifier_value" class="py-name" href="#" onclick="return doclink('link-193', '_retrieve_location_qualifier_value', 'link-193');">_retrieve_location_qualifier_value</a></tt><tt class="py-op">(</tt><tt class="py-name">adaptor</tt><tt class="py-op">,</tt> <tt class="py-name">location_id</tt><tt class="py-op">)</tt> </tt>
<a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line">            <tt class="py-name">dbname</tt><tt class="py-op">,</tt> <tt id="link-194" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-194', 'version', 'link-134');">version</a></tt> <tt class="py-op">=</tt> <tt id="link-195" class="py-name"><a title="BioSQL.BioSeqDatabase.BioSeqDatabase.lookup" class="py-name" href="#" onclick="return doclink('link-195', 'lookup', 'link-177');">lookup</a></tt><tt class="py-op">.</tt><tt id="link-196" class="py-name" targets="Method Bio.Crystal.Crystal.get()=Bio.Crystal.Crystal-class.html#get,Method Bio.Data.CodonTable.AmbiguousForwardTable.get()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#get,Method Bio.File._IndexedSeqFileDict.get()=Bio.File._IndexedSeqFileDict-class.html#get,Method Bio.File._IndexedSeqFileProxy.get()=Bio.File._IndexedSeqFileProxy-class.html#get,Method Bio.File._SQLiteManySeqFilesDict.get()=Bio.File._SQLiteManySeqFilesDict-class.html#get,Method Bio.Restriction.Restriction.RestrictionBatch.get()=Bio.Restriction.Restriction.RestrictionBatch-class.html#get,Method Bio.SearchIO._index.SearchIndexer.get()=Bio.SearchIO._index.SearchIndexer-class.html#get,Method Bio.SeqIO._index.SeqFileRandomAccess.get()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#get,Method Bio.SeqIO._index.SffRandomAccess.get()=Bio.SeqIO._index.SffRandomAccess-class.html#get,Method Bio.SeqIO._index.SffTrimedRandomAccess.get()=Bio.SeqIO._index.SffTrimedRandomAccess-class.html#get,Method Bio.SeqIO._index.UniprotRandomAccess.get()=Bio.SeqIO._index.UniprotRandomAccess-class.html#get"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-196', 'get', 'link-196');">get</a></tt><tt class="py-op">(</tt><tt class="py-name">location_id</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line">            <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-197" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.location()=Bio.GenBank._FeatureConsumer-class.html#location,Method Bio.GenBank._RecordConsumer.location()=Bio.GenBank._RecordConsumer-class.html#location,Method Bio.Phylo.PhyloXMLIO.Writer.location()=Bio.Phylo.PhyloXMLIO.Writer-class.html#location"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-197', 'location', 'link-197');">location</a></tt> <tt class="py-op">=</tt> <tt id="link-198" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-198', 'SeqFeature', 'link-13');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-199" class="py-name" targets="Class Bio.SeqFeature.FeatureLocation=Bio.SeqFeature.FeatureLocation-class.html"><a title="Bio.SeqFeature.FeatureLocation" class="py-name" href="#" onclick="return doclink('link-199', 'FeatureLocation', 'link-199');">FeatureLocation</a></tt><tt class="py-op">(</tt><tt id="link-200" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-200', 'start', 'link-18');">start</a></tt><tt class="py-op">,</tt> <tt id="link-201" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-201', 'end', 'link-156');">end</a></tt><tt class="py-op">)</tt> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line">            <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-202" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-202', 'strand', 'link-157');">strand</a></tt> <tt class="py-op">=</tt> <tt id="link-203" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-203', 'strand', 'link-157');">strand</a></tt> </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line">            <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-204" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.ref_db=Bio.SeqFeature.CompoundLocation-class.html#ref_db,Variable Bio.SeqFeature.SeqFeature.ref_db=Bio.SeqFeature.SeqFeature-class.html#ref_db"><a title="Bio.SeqFeature.CompoundLocation.ref_db
Bio.SeqFeature.SeqFeature.ref_db" class="py-name" href="#" onclick="return doclink('link-204', 'ref_db', 'link-204');">ref_db</a></tt> <tt class="py-op">=</tt> <tt class="py-name">dbname</tt> </tt>
<a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line">            <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-205" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.ref=Bio.SeqFeature.CompoundLocation-class.html#ref,Variable Bio.SeqFeature.SeqFeature.ref=Bio.SeqFeature.SeqFeature-class.html#ref"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-205', 'ref', 'link-205');">ref</a></tt> <tt class="py-op">=</tt> <tt id="link-206" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-206', 'version', 'link-134');">version</a></tt> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line">            <tt id="link-207" class="py-name" targets="Variable Bio.SeqFeature.SeqFeature.sub_features=Bio.SeqFeature.SeqFeature-class.html#sub_features"><a title="Bio.SeqFeature.SeqFeature.sub_features" class="py-name" href="#" onclick="return doclink('link-207', 'sub_features', 'link-207');">sub_features</a></tt> <tt class="py-op">=</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-208" class="py-name"><a title="Bio.SeqFeature.SeqFeature.sub_features" class="py-name" href="#" onclick="return doclink('link-208', 'sub_features', 'link-207');">sub_features</a></tt> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt id="link-209" class="py-name"><a title="Bio.SeqFeature.SeqFeature.sub_features" class="py-name" href="#" onclick="return doclink('link-209', 'sub_features', 'link-207');">sub_features</a></tt> <tt class="py-op">==</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt id="link-210" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-210', 'location', 'link-197');">location</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">locations</tt><tt class="py-op">:</tt> </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line">                <tt class="py-name">location_id</tt><tt class="py-op">,</tt> <tt id="link-211" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-211', 'start', 'link-18');">start</a></tt><tt class="py-op">,</tt> <tt id="link-212" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-212', 'end', 'link-156');">end</a></tt><tt class="py-op">,</tt> <tt id="link-213" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-213', 'strand', 'link-157');">strand</a></tt> <tt class="py-op">=</tt> <tt id="link-214" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-214', 'location', 'link-197');">location</a></tt> </tt>
<a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line">                <tt class="py-name">dbname</tt><tt class="py-op">,</tt> <tt id="link-215" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-215', 'version', 'link-134');">version</a></tt> <tt class="py-op">=</tt> <tt id="link-216" class="py-name"><a title="BioSQL.BioSeqDatabase.BioSeqDatabase.lookup" class="py-name" href="#" onclick="return doclink('link-216', 'lookup', 'link-177');">lookup</a></tt><tt class="py-op">.</tt><tt id="link-217" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-217', 'get', 'link-196');">get</a></tt><tt class="py-op">(</tt><tt class="py-name">location_id</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line">                <tt class="py-name">subfeature</tt> <tt class="py-op">=</tt> <tt id="link-218" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-218', 'SeqFeature', 'link-13');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-219" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-219', 'SeqFeature', 'link-13');">SeqFeature</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line">                <tt class="py-name">subfeature</tt><tt class="py-op">.</tt><tt id="link-220" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-220', 'type', 'link-188');">type</a></tt> <tt class="py-op">=</tt> <tt class="py-name">seqfeature_type</tt> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line">                <tt class="py-name">subfeature</tt><tt class="py-op">.</tt><tt id="link-221" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-221', 'location', 'link-197');">location</a></tt> <tt class="py-op">=</tt> <tt id="link-222" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-222', 'SeqFeature', 'link-13');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-223" class="py-name"><a title="Bio.SeqFeature.FeatureLocation" class="py-name" href="#" onclick="return doclink('link-223', 'FeatureLocation', 'link-199');">FeatureLocation</a></tt><tt class="py-op">(</tt><tt id="link-224" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-224', 'start', 'link-18');">start</a></tt><tt class="py-op">,</tt> <tt id="link-225" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-225', 'end', 'link-156');">end</a></tt><tt class="py-op">)</tt> </tt>
<a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line">                <tt class="py-comment"># subfeature.location_operator = \</tt> </tt>
<a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line">                <tt class="py-comment">#    _retrieve_location_qualifier_value(adaptor, location_id)</tt> </tt>
<a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line">                <tt class="py-name">subfeature</tt><tt class="py-op">.</tt><tt id="link-226" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-226', 'strand', 'link-157');">strand</a></tt> <tt class="py-op">=</tt> <tt id="link-227" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-227', 'strand', 'link-157');">strand</a></tt> </tt>
<a name="L312"></a><tt class="py-lineno">312</tt>  <tt class="py-line">                <tt class="py-name">subfeature</tt><tt class="py-op">.</tt><tt id="link-228" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref_db
Bio.SeqFeature.SeqFeature.ref_db" class="py-name" href="#" onclick="return doclink('link-228', 'ref_db', 'link-204');">ref_db</a></tt> <tt class="py-op">=</tt> <tt class="py-name">dbname</tt> </tt>
<a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line">                <tt class="py-name">subfeature</tt><tt class="py-op">.</tt><tt id="link-229" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-229', 'ref', 'link-205');">ref</a></tt> <tt class="py-op">=</tt> <tt id="link-230" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-230', 'version', 'link-134');">version</a></tt> </tt>
<a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line">                <tt id="link-231" class="py-name"><a title="Bio.SeqFeature.SeqFeature.sub_features" class="py-name" href="#" onclick="return doclink('link-231', 'sub_features', 'link-207');">sub_features</a></tt><tt class="py-op">.</tt><tt id="link-232" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-232', 'append', 'link-141');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">subfeature</tt><tt class="py-op">)</tt> </tt>
<a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line">            <tt class="py-comment"># Locations are in order, but because of remote locations for</tt> </tt>
<a name="L316"></a><tt class="py-lineno">316</tt>  <tt class="py-line">            <tt class="py-comment"># sub-features they are not necessarily in numerical order:</tt> </tt>
<a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line">            <tt class="py-name">strands</tt> <tt class="py-op">=</tt> <tt id="link-233" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.set()=Bio.Nexus.Nexus.StepMatrix-class.html#set,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.set()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#set,Method Bio.SVDSuperimposer.SVDSuperimposer.set()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#set"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-233', 'set', 'link-233');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">sf</tt><tt class="py-op">.</tt><tt id="link-234" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-234', 'strand', 'link-157');">strand</a></tt> <tt class="py-keyword">for</tt> <tt class="py-name">sf</tt> <tt class="py-keyword">in</tt> <tt id="link-235" class="py-name"><a title="Bio.SeqFeature.SeqFeature.sub_features" class="py-name" href="#" onclick="return doclink('link-235', 'sub_features', 'link-207');">sub_features</a></tt><tt class="py-op">)</tt> </tt>
<a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">strands</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt> <tt class="py-keyword">and</tt> <tt class="py-op">-</tt><tt class="py-number">1</tt> <tt class="py-keyword">in</tt> <tt class="py-name">strands</tt><tt class="py-op">:</tt> </tt>
<a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line">                <tt class="py-comment"># Evil hack time for backwards compatibility</tt> </tt>
<a name="L320"></a><tt class="py-lineno">320</tt>  <tt class="py-line">                <tt class="py-comment"># TODO - Check if BioPerl and (old) Biopython did the same,</tt> </tt>
<a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line">                <tt class="py-comment"># we may have an existing incompatibility lurking here...</tt> </tt>
<a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line">                <tt class="py-name">locs</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">f</tt><tt class="py-op">.</tt><tt id="link-236" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-236', 'location', 'link-197');">location</a></tt> <tt class="py-keyword">for</tt> <tt class="py-name">f</tt> <tt class="py-keyword">in</tt> <tt id="link-237" class="py-name"><a title="Bio.SeqFeature.SeqFeature.sub_features" class="py-name" href="#" onclick="return doclink('link-237', 'sub_features', 'link-207');">sub_features</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">]</tt> </tt>
<a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line">                <tt class="py-comment"># All forward, or mixed strands</tt> </tt>
<a name="L325"></a><tt class="py-lineno">325</tt>  <tt class="py-line">                <tt class="py-name">locs</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">f</tt><tt class="py-op">.</tt><tt id="link-238" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-238', 'location', 'link-197');">location</a></tt> <tt class="py-keyword">for</tt> <tt class="py-name">f</tt> <tt class="py-keyword">in</tt> <tt id="link-239" class="py-name"><a title="Bio.SeqFeature.SeqFeature.sub_features" class="py-name" href="#" onclick="return doclink('link-239', 'sub_features', 'link-207');">sub_features</a></tt><tt class="py-op">]</tt> </tt>
<a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line">            <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-240" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-240', 'location', 'link-197');">location</a></tt> <tt class="py-op">=</tt> <tt id="link-241" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-241', 'SeqFeature', 'link-13');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-242" class="py-name" targets="Class Bio.SeqFeature.CompoundLocation=Bio.SeqFeature.CompoundLocation-class.html"><a title="Bio.SeqFeature.CompoundLocation" class="py-name" href="#" onclick="return doclink('link-242', 'CompoundLocation', 'link-242');">CompoundLocation</a></tt><tt class="py-op">(</tt> </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line">                <tt class="py-name">locs</tt><tt class="py-op">,</tt> <tt class="py-name">seqfeature_type</tt><tt class="py-op">)</tt> </tt>
<a name="L328"></a><tt class="py-lineno">328</tt>  <tt class="py-line">            <tt class="py-comment"># TODO - See Bug 2677 - we don't yet record location_operator,</tt> </tt>
<a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line">            <tt class="py-comment"># so for consistency with older versions of Biopython default</tt> </tt>
<a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line">            <tt class="py-comment"># to assuming its a join.</tt> </tt>
<a name="L331"></a><tt class="py-lineno">331</tt>  <tt class="py-line">            <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-243" class="py-name"><a title="Bio.SeqFeature.SeqFeature.location_operator" class="py-name" href="#" onclick="return doclink('link-243', 'location_operator', 'link-192');">location_operator</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"join"</tt> </tt>
<a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line">        <tt class="py-name">seq_feature_list</tt><tt class="py-op">.</tt><tt id="link-244" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-244', 'append', 'link-141');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">feature</tt><tt class="py-op">)</tt> </tt>
<a name="L333"></a><tt class="py-lineno">333</tt>  <tt class="py-line"> </tt>
<a name="L334"></a><tt class="py-lineno">334</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">seq_feature_list</tt> </tt>
</div><a name="L335"></a><tt class="py-lineno">335</tt>  <tt class="py-line"> </tt>
<a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line"> </tt>
<a name="_retrieve_location_qualifier_value"></a><div id="_retrieve_location_qualifier_value-def"><a name="L337"></a><tt class="py-lineno">337</tt> <a class="py-toggle" href="#" id="_retrieve_location_qualifier_value-toggle" onclick="return toggle('_retrieve_location_qualifier_value');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq-module.html#_retrieve_location_qualifier_value">_retrieve_location_qualifier_value</a><tt class="py-op">(</tt><tt class="py-param">adaptor</tt><tt class="py-op">,</tt> <tt class="py-param">location_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_retrieve_location_qualifier_value-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_retrieve_location_qualifier_value-expanded"><a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line">    <tt id="link-245" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-245', 'value', 'link-149');">value</a></tt> <tt class="py-op">=</tt> <tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-246" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0()=BioSQL.BioSeqDatabase.Adaptor-class.html#execute_and_fetch_col0,Method BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0()=BioSQL.BioSeqDatabase.MysqlConnectorAdaptor-class.html#execute_and_fetch_col0"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0" class="py-name" href="#" onclick="return doclink('link-246', 'execute_and_fetch_col0', 'link-246');">execute_and_fetch_col0</a></tt><tt class="py-op">(</tt> </tt>
<a name="L339"></a><tt class="py-lineno">339</tt>  <tt class="py-line">        <tt class="py-string">"SELECT value FROM location_qualifier_value"</tt> </tt>
<a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line">        <tt class="py-string">" WHERE location_id = %s"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">location_id</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line">    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-247" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-247', 'value', 'link-149');">value</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L343"></a><tt class="py-lineno">343</tt>  <tt class="py-line">    <tt class="py-keyword">except</tt> <tt class="py-name">IndexError</tt><tt class="py-op">:</tt> </tt>
<a name="L344"></a><tt class="py-lineno">344</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">""</tt> </tt>
</div><a name="L345"></a><tt class="py-lineno">345</tt>  <tt class="py-line"> </tt>
<a name="L346"></a><tt class="py-lineno">346</tt>  <tt class="py-line"> </tt>
<a name="_retrieve_annotations"></a><div id="_retrieve_annotations-def"><a name="L347"></a><tt class="py-lineno">347</tt> <a class="py-toggle" href="#" id="_retrieve_annotations-toggle" onclick="return toggle('_retrieve_annotations');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq-module.html#_retrieve_annotations">_retrieve_annotations</a><tt class="py-op">(</tt><tt class="py-param">adaptor</tt><tt class="py-op">,</tt> <tt class="py-param">primary_id</tt><tt class="py-op">,</tt> <tt class="py-param">taxon_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_retrieve_annotations-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_retrieve_annotations-expanded"><a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line">    <tt id="link-248" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.annotations=BioSQL.BioSeq.DBSeqRecord-class.html#annotations"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-248', 'annotations', 'link-248');">annotations</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L349"></a><tt class="py-lineno">349</tt>  <tt class="py-line">    <tt id="link-249" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-249', 'annotations', 'link-248');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-250" class="py-name" targets="Method Bio.File._IndexedSeqFileDict.update()=Bio.File._IndexedSeqFileDict-class.html#update,Method Bio.Index._InMemoryIndex.update()=Bio.Index._InMemoryIndex-class.html#update,Method Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.update()=Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer-class.html#update,Method Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.update()=Bio.NeuralNetwork.BackPropagation.Layer.InputLayer-class.html#update,Method Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.update()=Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer-class.html#update,Method Bio.SeqRecord._RestrictedDict.update()=Bio.SeqRecord._RestrictedDict-class.html#update,Method Bio._py3k._ordereddict.OrderedDict.update()=Bio._py3k._ordereddict.OrderedDict-class.html#update"><a title="Bio.File._IndexedSeqFileDict.update
Bio.Index._InMemoryIndex.update
Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.update
Bio.SeqRecord._RestrictedDict.update
Bio._py3k._ordereddict.OrderedDict.update" class="py-name" href="#" onclick="return doclink('link-250', 'update', 'link-250');">update</a></tt><tt class="py-op">(</tt><tt id="link-251" class="py-name" targets="Function BioSQL.BioSeq._retrieve_qualifier_value()=BioSQL.BioSeq-module.html#_retrieve_qualifier_value"><a title="BioSQL.BioSeq._retrieve_qualifier_value" class="py-name" href="#" onclick="return doclink('link-251', '_retrieve_qualifier_value', 'link-251');">_retrieve_qualifier_value</a></tt><tt class="py-op">(</tt><tt class="py-name">adaptor</tt><tt class="py-op">,</tt> <tt class="py-name">primary_id</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line">    <tt id="link-252" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-252', 'annotations', 'link-248');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-253" class="py-name"><a title="Bio.File._IndexedSeqFileDict.update
Bio.Index._InMemoryIndex.update
Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.update
Bio.SeqRecord._RestrictedDict.update
Bio._py3k._ordereddict.OrderedDict.update" class="py-name" href="#" onclick="return doclink('link-253', 'update', 'link-250');">update</a></tt><tt class="py-op">(</tt><tt id="link-254" class="py-name" targets="Function BioSQL.BioSeq._retrieve_reference()=BioSQL.BioSeq-module.html#_retrieve_reference"><a title="BioSQL.BioSeq._retrieve_reference" class="py-name" href="#" onclick="return doclink('link-254', '_retrieve_reference', 'link-254');">_retrieve_reference</a></tt><tt class="py-op">(</tt><tt class="py-name">adaptor</tt><tt class="py-op">,</tt> <tt class="py-name">primary_id</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L351"></a><tt class="py-lineno">351</tt>  <tt class="py-line">    <tt id="link-255" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-255', 'annotations', 'link-248');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-256" class="py-name"><a title="Bio.File._IndexedSeqFileDict.update
Bio.Index._InMemoryIndex.update
Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.update
Bio.SeqRecord._RestrictedDict.update
Bio._py3k._ordereddict.OrderedDict.update" class="py-name" href="#" onclick="return doclink('link-256', 'update', 'link-250');">update</a></tt><tt class="py-op">(</tt><tt id="link-257" class="py-name" targets="Function BioSQL.BioSeq._retrieve_taxon()=BioSQL.BioSeq-module.html#_retrieve_taxon"><a title="BioSQL.BioSeq._retrieve_taxon" class="py-name" href="#" onclick="return doclink('link-257', '_retrieve_taxon', 'link-257');">_retrieve_taxon</a></tt><tt class="py-op">(</tt><tt class="py-name">adaptor</tt><tt class="py-op">,</tt> <tt class="py-name">primary_id</tt><tt class="py-op">,</tt> <tt class="py-name">taxon_id</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L352"></a><tt class="py-lineno">352</tt>  <tt class="py-line">    <tt id="link-258" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-258', 'annotations', 'link-248');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-259" class="py-name"><a title="Bio.File._IndexedSeqFileDict.update
Bio.Index._InMemoryIndex.update
Bio.NeuralNetwork.BackPropagation.Layer.HiddenLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.InputLayer.update
Bio.NeuralNetwork.BackPropagation.Layer.OutputLayer.update
Bio.SeqRecord._RestrictedDict.update
Bio._py3k._ordereddict.OrderedDict.update" class="py-name" href="#" onclick="return doclink('link-259', 'update', 'link-250');">update</a></tt><tt class="py-op">(</tt><tt id="link-260" class="py-name" targets="Function BioSQL.BioSeq._retrieve_comment()=BioSQL.BioSeq-module.html#_retrieve_comment"><a title="BioSQL.BioSeq._retrieve_comment" class="py-name" href="#" onclick="return doclink('link-260', '_retrieve_comment', 'link-260');">_retrieve_comment</a></tt><tt class="py-op">(</tt><tt class="py-name">adaptor</tt><tt class="py-op">,</tt> <tt class="py-name">primary_id</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line">    <tt class="py-comment"># Convert values into strings in cases of unicode from the database.</tt> </tt>
<a name="L354"></a><tt class="py-lineno">354</tt>  <tt class="py-line">    <tt class="py-comment"># BioSQL could eventually be expanded to be unicode aware.</tt> </tt>
<a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line">    <tt class="py-name">str_anns</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">val</tt> <tt class="py-keyword">in</tt> <tt id="link-261" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-261', 'annotations', 'link-248');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-262" class="py-name" targets="Method Bio.Crystal.Crystal.items()=Bio.Crystal.Crystal-class.html#items,Method Bio.File._IndexedSeqFileDict.items()=Bio.File._IndexedSeqFileDict-class.html#items,Method Bio.Phylo.PhyloXML.Events.items()=Bio.Phylo.PhyloXML.Events-class.html#items,Variable Bio.SearchIO._model.query.QueryResult.items=Bio.SearchIO._model.query.QueryResult-class.html#items,Method Bio._py3k._ordereddict.OrderedDict.items()=Bio._py3k._ordereddict.OrderedDict-class.html#items,Method BioSQL.BioSeqDatabase.BioSeqDatabase.items()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#items,Method BioSQL.BioSeqDatabase.DBServer.items()=BioSQL.BioSeqDatabase.DBServer-class.html#items"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
Bio._py3k._ordereddict.OrderedDict.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-262', 'items', 'link-262');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">val</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L358"></a><tt class="py-lineno">358</tt>  <tt class="py-line">            <tt class="py-name">val</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-263" class="py-name" targets="Function BioSQL.BioSeq._make_unicode_into_string()=BioSQL.BioSeq-module.html#_make_unicode_into_string"><a title="BioSQL.BioSeq._make_unicode_into_string" class="py-name" href="#" onclick="return doclink('link-263', '_make_unicode_into_string', 'link-263');">_make_unicode_into_string</a></tt><tt class="py-op">(</tt><tt id="link-264" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.x=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#x"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-264', 'x', 'link-264');">x</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt id="link-265" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.x" class="py-name" href="#" onclick="return doclink('link-265', 'x', 'link-264');">x</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">val</tt><tt class="py-op">]</tt> </tt>
<a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">val</tt><tt class="py-op">,</tt> <tt class="py-name">unicode</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line">            <tt class="py-name">val</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">val</tt><tt class="py-op">)</tt> </tt>
<a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line">        <tt class="py-name">str_anns</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">val</tt> </tt>
<a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">str_anns</tt> </tt>
</div><a name="L363"></a><tt class="py-lineno">363</tt>  <tt class="py-line"> </tt>
<a name="L364"></a><tt class="py-lineno">364</tt>  <tt class="py-line"> </tt>
<a name="_make_unicode_into_string"></a><div id="_make_unicode_into_string-def"><a name="L365"></a><tt class="py-lineno">365</tt> <a class="py-toggle" href="#" id="_make_unicode_into_string-toggle" onclick="return toggle('_make_unicode_into_string');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq-module.html#_make_unicode_into_string">_make_unicode_into_string</a><tt class="py-op">(</tt><tt class="py-param">text</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_make_unicode_into_string-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_make_unicode_into_string-expanded"><a name="L366"></a><tt class="py-lineno">366</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">,</tt> <tt class="py-name">unicode</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L367"></a><tt class="py-lineno">367</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">text</tt><tt class="py-op">)</tt> </tt>
<a name="L368"></a><tt class="py-lineno">368</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">text</tt> </tt>
</div><a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line"> </tt>
<a name="L371"></a><tt class="py-lineno">371</tt>  <tt class="py-line"> </tt>
<a name="_retrieve_qualifier_value"></a><div id="_retrieve_qualifier_value-def"><a name="L372"></a><tt class="py-lineno">372</tt> <a class="py-toggle" href="#" id="_retrieve_qualifier_value-toggle" onclick="return toggle('_retrieve_qualifier_value');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq-module.html#_retrieve_qualifier_value">_retrieve_qualifier_value</a><tt class="py-op">(</tt><tt class="py-param">adaptor</tt><tt class="py-op">,</tt> <tt class="py-param">primary_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_retrieve_qualifier_value-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_retrieve_qualifier_value-expanded"><a name="L373"></a><tt class="py-lineno">373</tt>  <tt class="py-line">    <tt class="py-name">qvs</tt> <tt class="py-op">=</tt> <tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-266" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetchall
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetchall" class="py-name" href="#" onclick="return doclink('link-266', 'execute_and_fetchall', 'link-99');">execute_and_fetchall</a></tt><tt class="py-op">(</tt> </tt>
<a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line">        <tt class="py-string">"SELECT name, value"</tt> </tt>
<a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line">        <tt class="py-string">" FROM bioentry_qualifier_value JOIN term USING (term_id)"</tt> </tt>
<a name="L376"></a><tt class="py-lineno">376</tt>  <tt class="py-line">        <tt class="py-string">" WHERE bioentry_id = %s"</tt> </tt>
<a name="L377"></a><tt class="py-lineno">377</tt>  <tt class="py-line">        <tt class="py-string">" ORDER BY rank"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">primary_id</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L378"></a><tt class="py-lineno">378</tt>  <tt class="py-line">    <tt class="py-name">qualifiers</tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L379"></a><tt class="py-lineno">379</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt id="link-267" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.name=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.name=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.name=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#name,Method Bio.Phylo.PhyloXMLIO.Writer.name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#name,Variable Bio.codonalign.codonalphabet.CodonAlphabet.name=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-267', 'name', 'link-267');">name</a></tt><tt class="py-op">,</tt> <tt id="link-268" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-268', 'value', 'link-149');">value</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">qvs</tt><tt class="py-op">:</tt> </tt>
<a name="L380"></a><tt class="py-lineno">380</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-269" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-269', 'name', 'link-267');">name</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"keyword"</tt><tt class="py-op">:</tt> </tt>
<a name="L381"></a><tt class="py-lineno">381</tt>  <tt class="py-line">            <tt id="link-270" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-270', 'name', 'link-267');">name</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"keywords"</tt> </tt>
<a name="L382"></a><tt class="py-lineno">382</tt>  <tt class="py-line">        <tt class="py-comment"># See handling of "date" in Loader.py</tt> </tt>
<a name="L383"></a><tt class="py-lineno">383</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-271" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-271', 'name', 'link-267');">name</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"date_changed"</tt><tt class="py-op">:</tt> </tt>
<a name="L384"></a><tt class="py-lineno">384</tt>  <tt class="py-line">            <tt id="link-272" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-272', 'name', 'link-267');">name</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"date"</tt> </tt>
<a name="L385"></a><tt class="py-lineno">385</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt id="link-273" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-273', 'name', 'link-267');">name</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"secondary_accession"</tt><tt class="py-op">:</tt> </tt>
<a name="L386"></a><tt class="py-lineno">386</tt>  <tt class="py-line">            <tt id="link-274" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-274', 'name', 'link-267');">name</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"accessions"</tt> </tt>
<a name="L387"></a><tt class="py-lineno">387</tt>  <tt class="py-line">        <tt class="py-name">qualifiers</tt><tt class="py-op">.</tt><tt id="link-275" class="py-name"><a title="Bio.Crystal.Crystal.setdefault
Bio._py3k._ordereddict.OrderedDict.setdefault" class="py-name" href="#" onclick="return doclink('link-275', 'setdefault', 'link-146');">setdefault</a></tt><tt class="py-op">(</tt><tt id="link-276" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-276', 'name', 'link-267');">name</a></tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-277" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-277', 'append', 'link-141');">append</a></tt><tt class="py-op">(</tt><tt id="link-278" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-278', 'value', 'link-149');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L388"></a><tt class="py-lineno">388</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">qualifiers</tt> </tt>
</div><a name="L389"></a><tt class="py-lineno">389</tt>  <tt class="py-line"> </tt>
<a name="L390"></a><tt class="py-lineno">390</tt>  <tt class="py-line"> </tt>
<a name="_retrieve_reference"></a><div id="_retrieve_reference-def"><a name="L391"></a><tt class="py-lineno">391</tt> <a class="py-toggle" href="#" id="_retrieve_reference-toggle" onclick="return toggle('_retrieve_reference');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq-module.html#_retrieve_reference">_retrieve_reference</a><tt class="py-op">(</tt><tt class="py-param">adaptor</tt><tt class="py-op">,</tt> <tt class="py-param">primary_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_retrieve_reference-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_retrieve_reference-expanded"><a name="L392"></a><tt class="py-lineno">392</tt>  <tt class="py-line">    <tt class="py-comment"># XXX dbxref_qualifier_value</tt> </tt>
<a name="L393"></a><tt class="py-lineno">393</tt>  <tt class="py-line"> </tt>
<a name="L394"></a><tt class="py-lineno">394</tt>  <tt class="py-line">    <tt class="py-name">refs</tt> <tt class="py-op">=</tt> <tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-279" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetchall
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetchall" class="py-name" href="#" onclick="return doclink('link-279', 'execute_and_fetchall', 'link-99');">execute_and_fetchall</a></tt><tt class="py-op">(</tt> </tt>
<a name="L395"></a><tt class="py-lineno">395</tt>  <tt class="py-line">        <tt class="py-string">"SELECT start_pos, end_pos, "</tt> </tt>
<a name="L396"></a><tt class="py-lineno">396</tt>  <tt class="py-line">        <tt class="py-string">" location, title, authors,"</tt> </tt>
<a name="L397"></a><tt class="py-lineno">397</tt>  <tt class="py-line">        <tt class="py-string">" dbname, accession"</tt> </tt>
<a name="L398"></a><tt class="py-lineno">398</tt>  <tt class="py-line">        <tt class="py-string">" FROM bioentry_reference"</tt> </tt>
<a name="L399"></a><tt class="py-lineno">399</tt>  <tt class="py-line">        <tt class="py-string">" JOIN reference USING (reference_id)"</tt> </tt>
<a name="L400"></a><tt class="py-lineno">400</tt>  <tt class="py-line">        <tt class="py-string">" LEFT JOIN dbxref USING (dbxref_id)"</tt> </tt>
<a name="L401"></a><tt class="py-lineno">401</tt>  <tt class="py-line">        <tt class="py-string">" WHERE bioentry_id = %s"</tt> </tt>
<a name="L402"></a><tt class="py-lineno">402</tt>  <tt class="py-line">        <tt class="py-string">" ORDER BY rank"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">primary_id</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L403"></a><tt class="py-lineno">403</tt>  <tt class="py-line">    <tt class="py-name">references</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L404"></a><tt class="py-lineno">404</tt>  <tt class="py-line">    <tt class="py-keyword">for</tt> <tt id="link-280" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-280', 'start', 'link-18');">start</a></tt><tt class="py-op">,</tt> <tt id="link-281" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-281', 'end', 'link-156');">end</a></tt><tt class="py-op">,</tt> <tt id="link-282" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-282', 'location', 'link-197');">location</a></tt><tt class="py-op">,</tt> <tt id="link-283" class="py-name" targets="Method Bio.Blast.NCBIStandalone._AlignmentConsumer.title()=Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#title,Method Bio.GenBank._FeatureConsumer.title()=Bio.GenBank._FeatureConsumer-class.html#title,Method Bio.GenBank._RecordConsumer.title()=Bio.GenBank._RecordConsumer-class.html#title"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-283', 'title', 'link-283');">title</a></tt><tt class="py-op">,</tt> <tt id="link-284" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.authors()=Bio.GenBank._FeatureConsumer-class.html#authors,Method Bio.GenBank._RecordConsumer.authors()=Bio.GenBank._RecordConsumer-class.html#authors"><a title="Bio.GenBank._FeatureConsumer.authors
Bio.GenBank._RecordConsumer.authors" class="py-name" href="#" onclick="return doclink('link-284', 'authors', 'link-284');">authors</a></tt><tt class="py-op">,</tt> <tt class="py-name">dbname</tt><tt class="py-op">,</tt> <tt id="link-285" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-285', 'accession', 'link-133');">accession</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">refs</tt><tt class="py-op">:</tt> </tt>
<a name="L405"></a><tt class="py-lineno">405</tt>  <tt class="py-line">        <tt id="link-286" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HeaderConsumer.reference()=Bio.Blast.NCBIStandalone._HeaderConsumer-class.html#reference,Method Bio.Phylo.PhyloXMLIO.Parser.reference()=Bio.Phylo.PhyloXMLIO.Parser-class.html#reference,Method Bio.Phylo.PhyloXMLIO.Writer.reference()=Bio.Phylo.PhyloXMLIO.Writer-class.html#reference"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-286', 'reference', 'link-286');">reference</a></tt> <tt class="py-op">=</tt> <tt id="link-287" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-287', 'SeqFeature', 'link-13');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-288" class="py-name" targets="Class Bio.ExPASy.Prodoc.Reference=Bio.ExPASy.Prodoc.Reference-class.html,Class Bio.GenBank.Record.Reference=Bio.GenBank.Record.Reference-class.html,Class Bio.Phylo.PhyloXML.Reference=Bio.Phylo.PhyloXML.Reference-class.html,Class Bio.SeqFeature.Reference=Bio.SeqFeature.Reference-class.html,Class Bio.SwissProt.Reference=Bio.SwissProt.Reference-class.html"><a title="Bio.ExPASy.Prodoc.Reference
Bio.GenBank.Record.Reference
Bio.Phylo.PhyloXML.Reference
Bio.SeqFeature.Reference
Bio.SwissProt.Reference" class="py-name" href="#" onclick="return doclink('link-288', 'Reference', 'link-288');">Reference</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L406"></a><tt class="py-lineno">406</tt>  <tt class="py-line">        <tt class="py-comment"># If the start/end are missing, reference.location is an empty list</tt> </tt>
<a name="L407"></a><tt class="py-lineno">407</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-op">(</tt><tt id="link-289" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-289', 'start', 'link-18');">start</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-op">(</tt><tt id="link-290" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-290', 'end', 'link-156');">end</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L408"></a><tt class="py-lineno">408</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-291" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-291', 'start', 'link-18');">start</a></tt> <tt class="py-keyword">is</tt> <tt class="py-keyword">not</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L409"></a><tt class="py-lineno">409</tt>  <tt class="py-line">                <tt id="link-292" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-292', 'start', 'link-18');">start</a></tt> <tt class="py-op">-=</tt> <tt class="py-number">1</tt>  <tt class="py-comment"># python counting</tt> </tt>
<a name="L410"></a><tt class="py-lineno">410</tt>  <tt class="py-line">            <tt id="link-293" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-293', 'reference', 'link-286');">reference</a></tt><tt class="py-op">.</tt><tt id="link-294" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-294', 'location', 'link-197');">location</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-295" class="py-name"><a title="Bio.SeqFeature
Bio.SeqFeature.SeqFeature" class="py-name" href="#" onclick="return doclink('link-295', 'SeqFeature', 'link-13');">SeqFeature</a></tt><tt class="py-op">.</tt><tt id="link-296" class="py-name"><a title="Bio.SeqFeature.FeatureLocation" class="py-name" href="#" onclick="return doclink('link-296', 'FeatureLocation', 'link-199');">FeatureLocation</a></tt><tt class="py-op">(</tt><tt id="link-297" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-297', 'start', 'link-18');">start</a></tt><tt class="py-op">,</tt> <tt id="link-298" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-298', 'end', 'link-156');">end</a></tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L411"></a><tt class="py-lineno">411</tt>  <tt class="py-line">        <tt class="py-comment"># Don't replace the default "" with None.</tt> </tt>
<a name="L412"></a><tt class="py-lineno">412</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-299" class="py-name"><a title="Bio.GenBank._FeatureConsumer.authors
Bio.GenBank._RecordConsumer.authors" class="py-name" href="#" onclick="return doclink('link-299', 'authors', 'link-284');">authors</a></tt><tt class="py-op">:</tt> </tt>
<a name="L413"></a><tt class="py-lineno">413</tt>  <tt class="py-line">            <tt id="link-300" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-300', 'reference', 'link-286');">reference</a></tt><tt class="py-op">.</tt><tt id="link-301" class="py-name"><a title="Bio.GenBank._FeatureConsumer.authors
Bio.GenBank._RecordConsumer.authors" class="py-name" href="#" onclick="return doclink('link-301', 'authors', 'link-284');">authors</a></tt> <tt class="py-op">=</tt> <tt id="link-302" class="py-name"><a title="Bio.GenBank._FeatureConsumer.authors
Bio.GenBank._RecordConsumer.authors" class="py-name" href="#" onclick="return doclink('link-302', 'authors', 'link-284');">authors</a></tt> </tt>
<a name="L414"></a><tt class="py-lineno">414</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-303" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-303', 'title', 'link-283');">title</a></tt><tt class="py-op">:</tt> </tt>
<a name="L415"></a><tt class="py-lineno">415</tt>  <tt class="py-line">            <tt id="link-304" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-304', 'reference', 'link-286');">reference</a></tt><tt class="py-op">.</tt><tt id="link-305" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-305', 'title', 'link-283');">title</a></tt> <tt class="py-op">=</tt> <tt id="link-306" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-306', 'title', 'link-283');">title</a></tt> </tt>
<a name="L416"></a><tt class="py-lineno">416</tt>  <tt class="py-line">        <tt id="link-307" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-307', 'reference', 'link-286');">reference</a></tt><tt class="py-op">.</tt><tt id="link-308" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.journal()=Bio.GenBank._FeatureConsumer-class.html#journal,Method Bio.GenBank._RecordConsumer.journal()=Bio.GenBank._RecordConsumer-class.html#journal"><a title="Bio.GenBank._FeatureConsumer.journal
Bio.GenBank._RecordConsumer.journal" class="py-name" href="#" onclick="return doclink('link-308', 'journal', 'link-308');">journal</a></tt> <tt class="py-op">=</tt> <tt id="link-309" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-309', 'location', 'link-197');">location</a></tt> </tt>
<a name="L417"></a><tt class="py-lineno">417</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">dbname</tt> <tt class="py-op">==</tt> <tt class="py-string">'PUBMED'</tt><tt class="py-op">:</tt> </tt>
<a name="L418"></a><tt class="py-lineno">418</tt>  <tt class="py-line">            <tt id="link-310" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-310', 'reference', 'link-286');">reference</a></tt><tt class="py-op">.</tt><tt id="link-311" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.pubmed_id()=Bio.GenBank._FeatureConsumer-class.html#pubmed_id,Method Bio.GenBank._RecordConsumer.pubmed_id()=Bio.GenBank._RecordConsumer-class.html#pubmed_id"><a title="Bio.GenBank._FeatureConsumer.pubmed_id
Bio.GenBank._RecordConsumer.pubmed_id" class="py-name" href="#" onclick="return doclink('link-311', 'pubmed_id', 'link-311');">pubmed_id</a></tt> <tt class="py-op">=</tt> <tt id="link-312" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-312', 'accession', 'link-133');">accession</a></tt> </tt>
<a name="L419"></a><tt class="py-lineno">419</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">dbname</tt> <tt class="py-op">==</tt> <tt class="py-string">'MEDLINE'</tt><tt class="py-op">:</tt> </tt>
<a name="L420"></a><tt class="py-lineno">420</tt>  <tt class="py-line">            <tt id="link-313" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-313', 'reference', 'link-286');">reference</a></tt><tt class="py-op">.</tt><tt id="link-314" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.medline_id()=Bio.GenBank._FeatureConsumer-class.html#medline_id,Method Bio.GenBank._RecordConsumer.medline_id()=Bio.GenBank._RecordConsumer-class.html#medline_id"><a title="Bio.GenBank._FeatureConsumer.medline_id
Bio.GenBank._RecordConsumer.medline_id" class="py-name" href="#" onclick="return doclink('link-314', 'medline_id', 'link-314');">medline_id</a></tt> <tt class="py-op">=</tt> <tt id="link-315" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-315', 'accession', 'link-133');">accession</a></tt> </tt>
<a name="L421"></a><tt class="py-lineno">421</tt>  <tt class="py-line">        <tt class="py-name">references</tt><tt class="py-op">.</tt><tt id="link-316" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-316', 'append', 'link-141');">append</a></tt><tt class="py-op">(</tt><tt id="link-317" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-317', 'reference', 'link-286');">reference</a></tt><tt class="py-op">)</tt> </tt>
<a name="L422"></a><tt class="py-lineno">422</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">references</tt><tt class="py-op">:</tt> </tt>
<a name="L423"></a><tt class="py-lineno">423</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">{</tt><tt class="py-string">'references'</tt><tt class="py-op">:</tt> <tt class="py-name">references</tt><tt class="py-op">}</tt> </tt>
<a name="L424"></a><tt class="py-lineno">424</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L425"></a><tt class="py-lineno">425</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
</div><a name="L426"></a><tt class="py-lineno">426</tt>  <tt class="py-line"> </tt>
<a name="L427"></a><tt class="py-lineno">427</tt>  <tt class="py-line"> </tt>
<a name="_retrieve_taxon"></a><div id="_retrieve_taxon-def"><a name="L428"></a><tt class="py-lineno">428</tt> <a class="py-toggle" href="#" id="_retrieve_taxon-toggle" onclick="return toggle('_retrieve_taxon');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq-module.html#_retrieve_taxon">_retrieve_taxon</a><tt class="py-op">(</tt><tt class="py-param">adaptor</tt><tt class="py-op">,</tt> <tt class="py-param">primary_id</tt><tt class="py-op">,</tt> <tt class="py-param">taxon_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_retrieve_taxon-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_retrieve_taxon-expanded"><a name="L429"></a><tt class="py-lineno">429</tt>  <tt class="py-line">    <tt id="link-318" class="py-name" targets="Variable Bio.Graphics.ColorSpiral.ColorSpiral.a=Bio.Graphics.ColorSpiral.ColorSpiral-class.html#a"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-318', 'a', 'link-318');">a</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
<a name="L430"></a><tt class="py-lineno">430</tt>  <tt class="py-line">    <tt class="py-name">common_names</tt> <tt class="py-op">=</tt> <tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-319" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0" class="py-name" href="#" onclick="return doclink('link-319', 'execute_and_fetch_col0', 'link-246');">execute_and_fetch_col0</a></tt><tt class="py-op">(</tt> </tt>
<a name="L431"></a><tt class="py-lineno">431</tt>  <tt class="py-line">        <tt class="py-string">"SELECT name FROM taxon_name WHERE taxon_id = %s"</tt> </tt>
<a name="L432"></a><tt class="py-lineno">432</tt>  <tt class="py-line">        <tt class="py-string">" AND name_class = 'genbank common name'"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">taxon_id</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L433"></a><tt class="py-lineno">433</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">common_names</tt><tt class="py-op">:</tt> </tt>
<a name="L434"></a><tt class="py-lineno">434</tt>  <tt class="py-line">        <tt id="link-320" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-320', 'a', 'link-318');">a</a></tt><tt class="py-op">[</tt><tt class="py-string">'source'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">common_names</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L435"></a><tt class="py-lineno">435</tt>  <tt class="py-line">    <tt class="py-name">scientific_names</tt> <tt class="py-op">=</tt> <tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-321" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0" class="py-name" href="#" onclick="return doclink('link-321', 'execute_and_fetch_col0', 'link-246');">execute_and_fetch_col0</a></tt><tt class="py-op">(</tt> </tt>
<a name="L436"></a><tt class="py-lineno">436</tt>  <tt class="py-line">        <tt class="py-string">"SELECT name FROM taxon_name WHERE taxon_id = %s"</tt> </tt>
<a name="L437"></a><tt class="py-lineno">437</tt>  <tt class="py-line">        <tt class="py-string">" AND name_class = 'scientific name'"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">taxon_id</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L438"></a><tt class="py-lineno">438</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">scientific_names</tt><tt class="py-op">:</tt> </tt>
<a name="L439"></a><tt class="py-lineno">439</tt>  <tt class="py-line">        <tt id="link-322" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-322', 'a', 'link-318');">a</a></tt><tt class="py-op">[</tt><tt class="py-string">'organism'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">scientific_names</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L440"></a><tt class="py-lineno">440</tt>  <tt class="py-line">    <tt class="py-name">ncbi_taxids</tt> <tt class="py-op">=</tt> <tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-323" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0" class="py-name" href="#" onclick="return doclink('link-323', 'execute_and_fetch_col0', 'link-246');">execute_and_fetch_col0</a></tt><tt class="py-op">(</tt> </tt>
<a name="L441"></a><tt class="py-lineno">441</tt>  <tt class="py-line">        <tt class="py-string">"SELECT ncbi_taxon_id FROM taxon WHERE taxon_id = %s"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">taxon_id</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L442"></a><tt class="py-lineno">442</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">ncbi_taxids</tt> <tt class="py-keyword">and</tt> <tt class="py-name">ncbi_taxids</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-keyword">and</tt> <tt class="py-name">ncbi_taxids</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">!=</tt> <tt class="py-string">"0"</tt><tt class="py-op">:</tt> </tt>
<a name="L443"></a><tt class="py-lineno">443</tt>  <tt class="py-line">        <tt id="link-324" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-324', 'a', 'link-318');">a</a></tt><tt class="py-op">[</tt><tt class="py-string">'ncbi_taxid'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">ncbi_taxids</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L444"></a><tt class="py-lineno">444</tt>  <tt class="py-line"> </tt>
<a name="L445"></a><tt class="py-lineno">445</tt>  <tt class="py-line">    <tt class="py-comment"># Old code used the left/right values in the taxon table to get the</tt> </tt>
<a name="L446"></a><tt class="py-lineno">446</tt>  <tt class="py-line">    <tt class="py-comment"># taxonomy lineage in one SQL command.  This was actually very slow,</tt> </tt>
<a name="L447"></a><tt class="py-lineno">447</tt>  <tt class="py-line">    <tt class="py-comment"># and would fail if the (optional) left/right values were missing.</tt> </tt>
<a name="L448"></a><tt class="py-lineno">448</tt>  <tt class="py-line">    <tt class="py-comment">#</tt> </tt>
<a name="L449"></a><tt class="py-lineno">449</tt>  <tt class="py-line">    <tt class="py-comment"># The following code is based on a contribution from Eric Gibert, and</tt> </tt>
<a name="L450"></a><tt class="py-lineno">450</tt>  <tt class="py-line">    <tt class="py-comment"># relies on the taxon table's parent_taxon_id field only (ignoring the</tt> </tt>
<a name="L451"></a><tt class="py-lineno">451</tt>  <tt class="py-line">    <tt class="py-comment"># optional left/right values).  This means that it has to make a</tt> </tt>
<a name="L452"></a><tt class="py-lineno">452</tt>  <tt class="py-line">    <tt class="py-comment"># separate SQL query for each entry in the lineage, but it does still</tt> </tt>
<a name="L453"></a><tt class="py-lineno">453</tt>  <tt class="py-line">    <tt class="py-comment"># appear to be *much* faster.  See Bug 2494.</tt> </tt>
<a name="L454"></a><tt class="py-lineno">454</tt>  <tt class="py-line">    <tt id="link-325" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.taxonomy()=Bio.GenBank._FeatureConsumer-class.html#taxonomy,Method Bio.GenBank._RecordConsumer.taxonomy()=Bio.GenBank._RecordConsumer-class.html#taxonomy,Variable Bio.Phylo.PhyloXML.Clade.taxonomy=Bio.Phylo.PhyloXML.Clade-class.html#taxonomy,Method Bio.Phylo.PhyloXMLIO.Writer.taxonomy()=Bio.Phylo.PhyloXMLIO.Writer-class.html#taxonomy"><a title="Bio.GenBank._FeatureConsumer.taxonomy
Bio.GenBank._RecordConsumer.taxonomy
Bio.Phylo.PhyloXML.Clade.taxonomy
Bio.Phylo.PhyloXMLIO.Writer.taxonomy" class="py-name" href="#" onclick="return doclink('link-325', 'taxonomy', 'link-325');">taxonomy</a></tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L455"></a><tt class="py-lineno">455</tt>  <tt class="py-line">    <tt class="py-keyword">while</tt> <tt class="py-name">taxon_id</tt><tt class="py-op">:</tt> </tt>
<a name="L456"></a><tt class="py-lineno">456</tt>  <tt class="py-line">        <tt id="link-326" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-326', 'name', 'link-267');">name</a></tt><tt class="py-op">,</tt> <tt id="link-327" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.rank()=Bio.Phylo.PhyloXMLIO.Writer-class.html#rank"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-327', 'rank', 'link-327');">rank</a></tt><tt class="py-op">,</tt> <tt class="py-name">parent_taxon_id</tt> <tt class="py-op">=</tt> <tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-328" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.execute_one()=BioSQL.BioSeqDatabase.Adaptor-class.html#execute_one,Method BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_one()=BioSQL.BioSeqDatabase.MysqlConnectorAdaptor-class.html#execute_one"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_one
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_one" class="py-name" href="#" onclick="return doclink('link-328', 'execute_one', 'link-328');">execute_one</a></tt><tt class="py-op">(</tt> </tt>
<a name="L457"></a><tt class="py-lineno">457</tt>  <tt class="py-line">            <tt class="py-string">"SELECT taxon_name.name, taxon.node_rank, taxon.parent_taxon_id"</tt> </tt>
<a name="L458"></a><tt class="py-lineno">458</tt>  <tt class="py-line">            <tt class="py-string">" FROM taxon, taxon_name"</tt> </tt>
<a name="L459"></a><tt class="py-lineno">459</tt>  <tt class="py-line">            <tt class="py-string">" WHERE taxon.taxon_id=taxon_name.taxon_id"</tt> </tt>
<a name="L460"></a><tt class="py-lineno">460</tt>  <tt class="py-line">            <tt class="py-string">" AND taxon_name.name_class='scientific name'"</tt> </tt>
<a name="L461"></a><tt class="py-lineno">461</tt>  <tt class="py-line">            <tt class="py-string">" AND taxon.taxon_id = %s"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">taxon_id</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L462"></a><tt class="py-lineno">462</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">taxon_id</tt> <tt class="py-op">==</tt> <tt class="py-name">parent_taxon_id</tt><tt class="py-op">:</tt> </tt>
<a name="L463"></a><tt class="py-lineno">463</tt>  <tt class="py-line">            <tt class="py-comment"># If the taxon table has been populated by the BioSQL script</tt> </tt>
<a name="L464"></a><tt class="py-lineno">464</tt>  <tt class="py-line">            <tt class="py-comment"># load_ncbi_taxonomy.pl this is how top parent nodes are stored.</tt> </tt>
<a name="L465"></a><tt class="py-lineno">465</tt>  <tt class="py-line">            <tt class="py-comment"># Personally, I would have used a NULL parent_taxon_id here.</tt> </tt>
<a name="L466"></a><tt class="py-lineno">466</tt>  <tt class="py-line">            <tt class="py-keyword">break</tt> </tt>
<a name="L467"></a><tt class="py-lineno">467</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-329" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-329', 'rank', 'link-327');">rank</a></tt> <tt class="py-op">!=</tt> <tt class="py-string">"no rank"</tt><tt class="py-op">:</tt> </tt>
<a name="L468"></a><tt class="py-lineno">468</tt>  <tt class="py-line">            <tt class="py-comment"># For consistency with older versions of Biopython, we are only</tt> </tt>
<a name="L469"></a><tt class="py-lineno">469</tt>  <tt class="py-line">            <tt class="py-comment"># interested in taxonomy entries with a stated rank.</tt> </tt>
<a name="L470"></a><tt class="py-lineno">470</tt>  <tt class="py-line">            <tt class="py-comment"># Add this to the start of the lineage list.</tt> </tt>
<a name="L471"></a><tt class="py-lineno">471</tt>  <tt class="py-line">            <tt id="link-330" class="py-name"><a title="Bio.GenBank._FeatureConsumer.taxonomy
Bio.GenBank._RecordConsumer.taxonomy
Bio.Phylo.PhyloXML.Clade.taxonomy
Bio.Phylo.PhyloXMLIO.Writer.taxonomy" class="py-name" href="#" onclick="return doclink('link-330', 'taxonomy', 'link-325');">taxonomy</a></tt><tt class="py-op">.</tt><tt id="link-331" class="py-name" targets="Method Bio.Crystal.Chain.insert()=Bio.Crystal.Chain-class.html#insert,Method Bio.PDB.Entity.Entity.insert()=Bio.PDB.Entity.Entity-class.html#insert,Method Bio.Phylo.TreeConstruction._Matrix.insert()=Bio.Phylo.TreeConstruction._Matrix-class.html#insert,Method Bio.Seq.MutableSeq.insert()=Bio.Seq.MutableSeq-class.html#insert"><a title="Bio.Crystal.Chain.insert
Bio.PDB.Entity.Entity.insert
Bio.Phylo.TreeConstruction._Matrix.insert
Bio.Seq.MutableSeq.insert" class="py-name" href="#" onclick="return doclink('link-331', 'insert', 'link-331');">insert</a></tt><tt class="py-op">(</tt><tt class="py-number">0</tt><tt class="py-op">,</tt> <tt id="link-332" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-332', 'name', 'link-267');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L472"></a><tt class="py-lineno">472</tt>  <tt class="py-line">        <tt class="py-name">taxon_id</tt> <tt class="py-op">=</tt> <tt class="py-name">parent_taxon_id</tt> </tt>
<a name="L473"></a><tt class="py-lineno">473</tt>  <tt class="py-line"> </tt>
<a name="L474"></a><tt class="py-lineno">474</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt id="link-333" class="py-name"><a title="Bio.GenBank._FeatureConsumer.taxonomy
Bio.GenBank._RecordConsumer.taxonomy
Bio.Phylo.PhyloXML.Clade.taxonomy
Bio.Phylo.PhyloXMLIO.Writer.taxonomy" class="py-name" href="#" onclick="return doclink('link-333', 'taxonomy', 'link-325');">taxonomy</a></tt><tt class="py-op">:</tt> </tt>
<a name="L475"></a><tt class="py-lineno">475</tt>  <tt class="py-line">        <tt id="link-334" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-334', 'a', 'link-318');">a</a></tt><tt class="py-op">[</tt><tt class="py-string">'taxonomy'</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt id="link-335" class="py-name"><a title="Bio.GenBank._FeatureConsumer.taxonomy
Bio.GenBank._RecordConsumer.taxonomy
Bio.Phylo.PhyloXML.Clade.taxonomy
Bio.Phylo.PhyloXMLIO.Writer.taxonomy" class="py-name" href="#" onclick="return doclink('link-335', 'taxonomy', 'link-325');">taxonomy</a></tt> </tt>
<a name="L476"></a><tt class="py-lineno">476</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt id="link-336" class="py-name"><a title="Bio.Graphics.ColorSpiral.ColorSpiral.a" class="py-name" href="#" onclick="return doclink('link-336', 'a', 'link-318');">a</a></tt> </tt>
</div><a name="L477"></a><tt class="py-lineno">477</tt>  <tt class="py-line"> </tt>
<a name="L478"></a><tt class="py-lineno">478</tt>  <tt class="py-line"> </tt>
<a name="_retrieve_comment"></a><div id="_retrieve_comment-def"><a name="L479"></a><tt class="py-lineno">479</tt> <a class="py-toggle" href="#" id="_retrieve_comment-toggle" onclick="return toggle('_retrieve_comment');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq-module.html#_retrieve_comment">_retrieve_comment</a><tt class="py-op">(</tt><tt class="py-param">adaptor</tt><tt class="py-op">,</tt> <tt class="py-param">primary_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_retrieve_comment-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_retrieve_comment-expanded"><a name="L480"></a><tt class="py-lineno">480</tt>  <tt class="py-line">    <tt class="py-name">qvs</tt> <tt class="py-op">=</tt> <tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-337" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetchall
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetchall" class="py-name" href="#" onclick="return doclink('link-337', 'execute_and_fetchall', 'link-99');">execute_and_fetchall</a></tt><tt class="py-op">(</tt> </tt>
<a name="L481"></a><tt class="py-lineno">481</tt>  <tt class="py-line">        <tt class="py-string">"SELECT comment_text FROM comment"</tt> </tt>
<a name="L482"></a><tt class="py-lineno">482</tt>  <tt class="py-line">        <tt class="py-string">" WHERE bioentry_id=%s"</tt> </tt>
<a name="L483"></a><tt class="py-lineno">483</tt>  <tt class="py-line">        <tt class="py-string">" ORDER BY rank"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">primary_id</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L484"></a><tt class="py-lineno">484</tt>  <tt class="py-line">    <tt class="py-name">comments</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">comm</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt class="py-name">comm</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qvs</tt><tt class="py-op">]</tt> </tt>
<a name="L485"></a><tt class="py-lineno">485</tt>  <tt class="py-line">    <tt class="py-comment"># Don't want to add an empty list...</tt> </tt>
<a name="L486"></a><tt class="py-lineno">486</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">comments</tt><tt class="py-op">:</tt> </tt>
<a name="L487"></a><tt class="py-lineno">487</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">{</tt><tt class="py-string">"comment"</tt><tt class="py-op">:</tt> <tt class="py-name">comments</tt><tt class="py-op">}</tt> </tt>
<a name="L488"></a><tt class="py-lineno">488</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L489"></a><tt class="py-lineno">489</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt> </tt>
</div><a name="L490"></a><tt class="py-lineno">490</tt>  <tt class="py-line"> </tt>
<a name="L491"></a><tt class="py-lineno">491</tt>  <tt class="py-line"> </tt>
<a name="DBSeqRecord"></a><div id="DBSeqRecord-def"><a name="L492"></a><tt class="py-lineno">492</tt> <a class="py-toggle" href="#" id="DBSeqRecord-toggle" onclick="return toggle('DBSeqRecord');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="BioSQL.BioSeq.DBSeqRecord-class.html">DBSeqRecord</a><tt class="py-op">(</tt><tt class="py-base-class">SeqRecord</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBSeqRecord-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="DBSeqRecord-expanded"><a name="L493"></a><tt class="py-lineno">493</tt>  <tt class="py-line">    <tt class="py-docstring">"""BioSQL equivalent of the Biopython SeqRecord object."""</tt> </tt>
<a name="L494"></a><tt class="py-lineno">494</tt>  <tt class="py-line"> </tt>
<a name="DBSeqRecord.__init__"></a><div id="DBSeqRecord.__init__-def"><a name="L495"></a><tt class="py-lineno">495</tt> <a class="py-toggle" href="#" id="DBSeqRecord.__init__-toggle" onclick="return toggle('DBSeqRecord.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq.DBSeqRecord-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">adaptor</tt><tt class="py-op">,</tt> <tt class="py-param">primary_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBSeqRecord.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBSeqRecord.__init__-expanded"><a name="L496"></a><tt class="py-lineno">496</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_adaptor</tt> <tt class="py-op">=</tt> <tt class="py-name">adaptor</tt> </tt>
<a name="L497"></a><tt class="py-lineno">497</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_primary_id</tt> <tt class="py-op">=</tt> <tt class="py-name">primary_id</tt> </tt>
<a name="L498"></a><tt class="py-lineno">498</tt>  <tt class="py-line"> </tt>
<a name="L499"></a><tt class="py-lineno">499</tt>  <tt class="py-line">        <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_biodatabase_id</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_taxon_id</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-338" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-338', 'name', 'link-267');">name</a></tt><tt class="py-op">,</tt> </tt>
<a name="L500"></a><tt class="py-lineno">500</tt>  <tt class="py-line">         <tt id="link-339" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-339', 'accession', 'link-133');">accession</a></tt><tt class="py-op">,</tt> <tt id="link-340" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-340', 'version', 'link-134');">version</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_identifier</tt><tt class="py-op">,</tt> </tt>
<a name="L501"></a><tt class="py-lineno">501</tt>  <tt class="py-line">         <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_division</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-341" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DescriptionConsumer.description()=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#description,Method Bio.Phylo.PhyloXMLIO.Writer.description()=Bio.Phylo.PhyloXMLIO.Writer-class.html#description,Variable Bio.SearchIO._model.hit.Hit.description=Bio.SearchIO._model.hit.Hit-class.html#description,Variable Bio.SearchIO._model.query.QueryResult.description=Bio.SearchIO._model.query.QueryResult-class.html#description"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-341', 'description', 'link-341');">description</a></tt><tt class="py-op">)</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_adaptor</tt><tt class="py-op">.</tt><tt id="link-342" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_one
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_one" class="py-name" href="#" onclick="return doclink('link-342', 'execute_one', 'link-328');">execute_one</a></tt><tt class="py-op">(</tt> </tt>
<a name="L502"></a><tt class="py-lineno">502</tt>  <tt class="py-line">            <tt class="py-string">"SELECT biodatabase_id, taxon_id, name, accession, version,"</tt> </tt>
<a name="L503"></a><tt class="py-lineno">503</tt>  <tt class="py-line">            <tt class="py-string">" identifier, division, description"</tt> </tt>
<a name="L504"></a><tt class="py-lineno">504</tt>  <tt class="py-line">            <tt class="py-string">" FROM bioentry"</tt> </tt>
<a name="L505"></a><tt class="py-lineno">505</tt>  <tt class="py-line">            <tt class="py-string">" WHERE bioentry_id = %s"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_primary_id</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L506"></a><tt class="py-lineno">506</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-343" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-343', 'version', 'link-134');">version</a></tt> <tt class="py-keyword">and</tt> <tt id="link-344" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-344', 'version', 'link-134');">version</a></tt> <tt class="py-op">!=</tt> <tt class="py-string">"0"</tt><tt class="py-op">:</tt> </tt>
<a name="L507"></a><tt class="py-lineno">507</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-345" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.id=Bio.KEGG.KGML.KGML_pathway.Component-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Entry.id=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.id=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#id,Method Bio.Phylo.PhyloXMLIO.Parser.id()=Bio.Phylo.PhyloXMLIO.Parser-class.html#id,Method Bio.Phylo.PhyloXMLIO.Writer.id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#id,Variable Bio.SearchIO._model.hit.Hit.id=Bio.SearchIO._model.hit.Hit-class.html#id,Variable Bio.SearchIO._model.query.QueryResult.id=Bio.SearchIO._model.query.QueryResult-class.html#id"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-345', 'id', 'link-345');">id</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"%s.%s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-346" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-346', 'accession', 'link-133');">accession</a></tt><tt class="py-op">,</tt> <tt id="link-347" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-347', 'version', 'link-134');">version</a></tt><tt class="py-op">)</tt> </tt>
<a name="L508"></a><tt class="py-lineno">508</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L509"></a><tt class="py-lineno">509</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-348" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-348', 'id', 'link-345');">id</a></tt> <tt class="py-op">=</tt> <tt id="link-349" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-349', 'accession', 'link-133');">accession</a></tt> </tt>
<a name="L510"></a><tt class="py-lineno">510</tt>  <tt class="py-line">        <tt class="py-comment"># We don't yet record any per-letter-annotations in the</tt> </tt>
<a name="L511"></a><tt class="py-lineno">511</tt>  <tt class="py-line">        <tt class="py-comment"># BioSQL database, but we should set this property up</tt> </tt>
<a name="L512"></a><tt class="py-lineno">512</tt>  <tt class="py-line">        <tt class="py-comment"># for completeness (and the __str__ method).</tt> </tt>
<a name="L513"></a><tt class="py-lineno">513</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L514"></a><tt class="py-lineno">514</tt>  <tt class="py-line">            <tt id="link-350" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-350', 'length', 'link-17');">length</a></tt> <tt class="py-op">=</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-351" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-351', 'seq', 'link-107');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L515"></a><tt class="py-lineno">515</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt><tt class="py-op">:</tt> </tt>
<a name="L516"></a><tt class="py-lineno">516</tt>  <tt class="py-line">            <tt class="py-comment"># Could be no sequence in the database!</tt> </tt>
<a name="L517"></a><tt class="py-lineno">517</tt>  <tt class="py-line">            <tt id="link-352" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-352', 'length', 'link-17');">length</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L518"></a><tt class="py-lineno">518</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_per_letter_annotations</tt> <tt class="py-op">=</tt> <tt id="link-353" class="py-name"><a title="Bio.SeqRecord._RestrictedDict" class="py-name" href="#" onclick="return doclink('link-353', '_RestrictedDict', 'link-11');">_RestrictedDict</a></tt><tt class="py-op">(</tt><tt id="link-354" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-354', 'length', 'link-17');">length</a></tt><tt class="py-op">=</tt><tt id="link-355" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-355', 'length', 'link-17');">length</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L519"></a><tt class="py-lineno">519</tt>  <tt class="py-line"> </tt>
<a name="DBSeqRecord.__get_seq"></a><div id="DBSeqRecord.__get_seq-def"><a name="L520"></a><tt class="py-lineno">520</tt> <a class="py-toggle" href="#" id="DBSeqRecord.__get_seq-toggle" onclick="return toggle('DBSeqRecord.__get_seq');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq.DBSeqRecord-class.html#__get_seq">__get_seq</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBSeqRecord.__get_seq-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBSeqRecord.__get_seq-expanded"><a name="L521"></a><tt class="py-lineno">521</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-string">"_seq"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L522"></a><tt class="py-lineno">522</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq</tt> <tt class="py-op">=</tt> <tt id="link-356" class="py-name" targets="Function BioSQL.BioSeq._retrieve_seq()=BioSQL.BioSeq-module.html#_retrieve_seq"><a title="BioSQL.BioSeq._retrieve_seq" class="py-name" href="#" onclick="return doclink('link-356', '_retrieve_seq', 'link-356');">_retrieve_seq</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_adaptor</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_primary_id</tt><tt class="py-op">)</tt> </tt>
<a name="L523"></a><tt class="py-lineno">523</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq</tt> </tt>
</div><a name="L524"></a><tt class="py-lineno">524</tt>  <tt class="py-line"> </tt>
<a name="DBSeqRecord.__set_seq"></a><div id="DBSeqRecord.__set_seq-def"><a name="L525"></a><tt class="py-lineno">525</tt> <a class="py-toggle" href="#" id="DBSeqRecord.__set_seq-toggle" onclick="return toggle('DBSeqRecord.__set_seq');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq.DBSeqRecord-class.html#__set_seq">__set_seq</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">seq</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBSeqRecord.__set_seq-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBSeqRecord.__set_seq-expanded"><a name="L526"></a><tt class="py-lineno">526</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq</tt> <tt class="py-op">=</tt> <tt id="link-357" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-357', 'seq', 'link-107');">seq</a></tt> </tt>
</div><a name="L527"></a><tt class="py-lineno">527</tt>  <tt class="py-line"> </tt>
<a name="DBSeqRecord.__del_seq"></a><div id="DBSeqRecord.__del_seq-def"><a name="L528"></a><tt class="py-lineno">528</tt> <a class="py-toggle" href="#" id="DBSeqRecord.__del_seq-toggle" onclick="return toggle('DBSeqRecord.__del_seq');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq.DBSeqRecord-class.html#__del_seq">__del_seq</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBSeqRecord.__del_seq-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBSeqRecord.__del_seq-expanded"><a name="L529"></a><tt class="py-lineno">529</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_seq</tt> </tt>
</div><a name="L530"></a><tt class="py-lineno">530</tt>  <tt class="py-line">    <tt id="link-358" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-358', 'seq', 'link-107');">seq</a></tt> <tt class="py-op">=</tt> <tt id="link-359" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-359', 'property', 'link-91');">property</a></tt><tt class="py-op">(</tt><tt id="link-360" class="py-name" targets="Method BioSQL.BioSeq.DBSeqRecord.__get_seq()=BioSQL.BioSeq.DBSeqRecord-class.html#__get_seq"><a title="BioSQL.BioSeq.DBSeqRecord.__get_seq" class="py-name" href="#" onclick="return doclink('link-360', '__get_seq', 'link-360');">__get_seq</a></tt><tt class="py-op">,</tt> <tt id="link-361" class="py-name" targets="Method BioSQL.BioSeq.DBSeqRecord.__set_seq()=BioSQL.BioSeq.DBSeqRecord-class.html#__set_seq"><a title="BioSQL.BioSeq.DBSeqRecord.__set_seq" class="py-name" href="#" onclick="return doclink('link-361', '__set_seq', 'link-361');">__set_seq</a></tt><tt class="py-op">,</tt> <tt id="link-362" class="py-name" targets="Method BioSQL.BioSeq.DBSeqRecord.__del_seq()=BioSQL.BioSeq.DBSeqRecord-class.html#__del_seq"><a title="BioSQL.BioSeq.DBSeqRecord.__del_seq" class="py-name" href="#" onclick="return doclink('link-362', '__del_seq', 'link-362');">__del_seq</a></tt><tt class="py-op">,</tt> <tt class="py-string">"Seq object"</tt><tt class="py-op">)</tt> </tt>
<a name="L531"></a><tt class="py-lineno">531</tt>  <tt class="py-line"> </tt>
<a name="DBSeqRecord.__get_dbxrefs"></a><div id="DBSeqRecord.__get_dbxrefs-def"><a name="L532"></a><tt class="py-lineno">532</tt> <a class="py-toggle" href="#" id="DBSeqRecord.__get_dbxrefs-toggle" onclick="return toggle('DBSeqRecord.__get_dbxrefs');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq.DBSeqRecord-class.html#__get_dbxrefs">__get_dbxrefs</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBSeqRecord.__get_dbxrefs-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBSeqRecord.__get_dbxrefs-expanded"><a name="L533"></a><tt class="py-lineno">533</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-string">"_dbxrefs"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L534"></a><tt class="py-lineno">534</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_dbxrefs</tt> <tt class="py-op">=</tt> <tt id="link-363" class="py-name" targets="Function BioSQL.BioSeq._retrieve_dbxrefs()=BioSQL.BioSeq-module.html#_retrieve_dbxrefs"><a title="BioSQL.BioSeq._retrieve_dbxrefs" class="py-name" href="#" onclick="return doclink('link-363', '_retrieve_dbxrefs', 'link-363');">_retrieve_dbxrefs</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_adaptor</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_primary_id</tt><tt class="py-op">)</tt> </tt>
<a name="L535"></a><tt class="py-lineno">535</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_dbxrefs</tt> </tt>
</div><a name="L536"></a><tt class="py-lineno">536</tt>  <tt class="py-line"> </tt>
<a name="DBSeqRecord.__set_dbxrefs"></a><div id="DBSeqRecord.__set_dbxrefs-def"><a name="L537"></a><tt class="py-lineno">537</tt> <a class="py-toggle" href="#" id="DBSeqRecord.__set_dbxrefs-toggle" onclick="return toggle('DBSeqRecord.__set_dbxrefs');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq.DBSeqRecord-class.html#__set_dbxrefs">__set_dbxrefs</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">dbxrefs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBSeqRecord.__set_dbxrefs-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBSeqRecord.__set_dbxrefs-expanded"><a name="L538"></a><tt class="py-lineno">538</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_dbxrefs</tt> <tt class="py-op">=</tt> <tt id="link-364" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.dbxrefs" class="py-name" href="#" onclick="return doclink('link-364', 'dbxrefs', 'link-131');">dbxrefs</a></tt> </tt>
</div><a name="L539"></a><tt class="py-lineno">539</tt>  <tt class="py-line"> </tt>
<a name="DBSeqRecord.__del_dbxrefs"></a><div id="DBSeqRecord.__del_dbxrefs-def"><a name="L540"></a><tt class="py-lineno">540</tt> <a class="py-toggle" href="#" id="DBSeqRecord.__del_dbxrefs-toggle" onclick="return toggle('DBSeqRecord.__del_dbxrefs');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq.DBSeqRecord-class.html#__del_dbxrefs">__del_dbxrefs</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBSeqRecord.__del_dbxrefs-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBSeqRecord.__del_dbxrefs-expanded"><a name="L541"></a><tt class="py-lineno">541</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_dbxrefs</tt> </tt>
</div><a name="L542"></a><tt class="py-lineno">542</tt>  <tt class="py-line">    <tt id="link-365" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.dbxrefs" class="py-name" href="#" onclick="return doclink('link-365', 'dbxrefs', 'link-131');">dbxrefs</a></tt> <tt class="py-op">=</tt> <tt id="link-366" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-366', 'property', 'link-91');">property</a></tt><tt class="py-op">(</tt><tt id="link-367" class="py-name" targets="Method BioSQL.BioSeq.DBSeqRecord.__get_dbxrefs()=BioSQL.BioSeq.DBSeqRecord-class.html#__get_dbxrefs"><a title="BioSQL.BioSeq.DBSeqRecord.__get_dbxrefs" class="py-name" href="#" onclick="return doclink('link-367', '__get_dbxrefs', 'link-367');">__get_dbxrefs</a></tt><tt class="py-op">,</tt> <tt id="link-368" class="py-name" targets="Method BioSQL.BioSeq.DBSeqRecord.__set_dbxrefs()=BioSQL.BioSeq.DBSeqRecord-class.html#__set_dbxrefs"><a title="BioSQL.BioSeq.DBSeqRecord.__set_dbxrefs" class="py-name" href="#" onclick="return doclink('link-368', '__set_dbxrefs', 'link-368');">__set_dbxrefs</a></tt><tt class="py-op">,</tt> <tt id="link-369" class="py-name" targets="Method BioSQL.BioSeq.DBSeqRecord.__del_dbxrefs()=BioSQL.BioSeq.DBSeqRecord-class.html#__del_dbxrefs"><a title="BioSQL.BioSeq.DBSeqRecord.__del_dbxrefs" class="py-name" href="#" onclick="return doclink('link-369', '__del_dbxrefs', 'link-369');">__del_dbxrefs</a></tt><tt class="py-op">,</tt> </tt>
<a name="L543"></a><tt class="py-lineno">543</tt>  <tt class="py-line">                       <tt class="py-string">"Database cross references"</tt><tt class="py-op">)</tt> </tt>
<a name="L544"></a><tt class="py-lineno">544</tt>  <tt class="py-line"> </tt>
<a name="DBSeqRecord.__get_features"></a><div id="DBSeqRecord.__get_features-def"><a name="L545"></a><tt class="py-lineno">545</tt> <a class="py-toggle" href="#" id="DBSeqRecord.__get_features-toggle" onclick="return toggle('DBSeqRecord.__get_features');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq.DBSeqRecord-class.html#__get_features">__get_features</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBSeqRecord.__get_features-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBSeqRecord.__get_features-expanded"><a name="L546"></a><tt class="py-lineno">546</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-string">"_features"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L547"></a><tt class="py-lineno">547</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_features</tt> <tt class="py-op">=</tt> <tt id="link-370" class="py-name" targets="Function BioSQL.BioSeq._retrieve_features()=BioSQL.BioSeq-module.html#_retrieve_features"><a title="BioSQL.BioSeq._retrieve_features" class="py-name" href="#" onclick="return doclink('link-370', '_retrieve_features', 'link-370');">_retrieve_features</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_adaptor</tt><tt class="py-op">,</tt> </tt>
<a name="L548"></a><tt class="py-lineno">548</tt>  <tt class="py-line">                                                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_primary_id</tt><tt class="py-op">)</tt> </tt>
<a name="L549"></a><tt class="py-lineno">549</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_features</tt> </tt>
</div><a name="L550"></a><tt class="py-lineno">550</tt>  <tt class="py-line"> </tt>
<a name="DBSeqRecord.__set_features"></a><div id="DBSeqRecord.__set_features-def"><a name="L551"></a><tt class="py-lineno">551</tt> <a class="py-toggle" href="#" id="DBSeqRecord.__set_features-toggle" onclick="return toggle('DBSeqRecord.__set_features');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq.DBSeqRecord-class.html#__set_features">__set_features</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">features</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBSeqRecord.__set_features-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBSeqRecord.__set_features-expanded"><a name="L552"></a><tt class="py-lineno">552</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_features</tt> <tt class="py-op">=</tt> <tt id="link-371" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.features=BioSQL.BioSeq.DBSeqRecord-class.html#features"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-371', 'features', 'link-371');">features</a></tt> </tt>
</div><a name="L553"></a><tt class="py-lineno">553</tt>  <tt class="py-line"> </tt>
<a name="DBSeqRecord.__del_features"></a><div id="DBSeqRecord.__del_features-def"><a name="L554"></a><tt class="py-lineno">554</tt> <a class="py-toggle" href="#" id="DBSeqRecord.__del_features-toggle" onclick="return toggle('DBSeqRecord.__del_features');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq.DBSeqRecord-class.html#__del_features">__del_features</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBSeqRecord.__del_features-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBSeqRecord.__del_features-expanded"><a name="L555"></a><tt class="py-lineno">555</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_features</tt> </tt>
</div><a name="L556"></a><tt class="py-lineno">556</tt>  <tt class="py-line">    <tt id="link-372" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-372', 'features', 'link-371');">features</a></tt> <tt class="py-op">=</tt> <tt id="link-373" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-373', 'property', 'link-91');">property</a></tt><tt class="py-op">(</tt><tt id="link-374" class="py-name" targets="Method BioSQL.BioSeq.DBSeqRecord.__get_features()=BioSQL.BioSeq.DBSeqRecord-class.html#__get_features"><a title="BioSQL.BioSeq.DBSeqRecord.__get_features" class="py-name" href="#" onclick="return doclink('link-374', '__get_features', 'link-374');">__get_features</a></tt><tt class="py-op">,</tt> <tt id="link-375" class="py-name" targets="Method BioSQL.BioSeq.DBSeqRecord.__set_features()=BioSQL.BioSeq.DBSeqRecord-class.html#__set_features"><a title="BioSQL.BioSeq.DBSeqRecord.__set_features" class="py-name" href="#" onclick="return doclink('link-375', '__set_features', 'link-375');">__set_features</a></tt><tt class="py-op">,</tt> <tt id="link-376" class="py-name" targets="Method BioSQL.BioSeq.DBSeqRecord.__del_features()=BioSQL.BioSeq.DBSeqRecord-class.html#__del_features"><a title="BioSQL.BioSeq.DBSeqRecord.__del_features" class="py-name" href="#" onclick="return doclink('link-376', '__del_features', 'link-376');">__del_features</a></tt><tt class="py-op">,</tt> </tt>
<a name="L557"></a><tt class="py-lineno">557</tt>  <tt class="py-line">                        <tt class="py-string">"Features"</tt><tt class="py-op">)</tt> </tt>
<a name="L558"></a><tt class="py-lineno">558</tt>  <tt class="py-line"> </tt>
<a name="DBSeqRecord.__get_annotations"></a><div id="DBSeqRecord.__get_annotations-def"><a name="L559"></a><tt class="py-lineno">559</tt> <a class="py-toggle" href="#" id="DBSeqRecord.__get_annotations-toggle" onclick="return toggle('DBSeqRecord.__get_annotations');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq.DBSeqRecord-class.html#__get_annotations">__get_annotations</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBSeqRecord.__get_annotations-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBSeqRecord.__get_annotations-expanded"><a name="L560"></a><tt class="py-lineno">560</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">,</tt> <tt class="py-string">"_annotations"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L561"></a><tt class="py-lineno">561</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_annotations</tt> <tt class="py-op">=</tt> <tt id="link-377" class="py-name" targets="Function BioSQL.BioSeq._retrieve_annotations()=BioSQL.BioSeq-module.html#_retrieve_annotations"><a title="BioSQL.BioSeq._retrieve_annotations" class="py-name" href="#" onclick="return doclink('link-377', '_retrieve_annotations', 'link-377');">_retrieve_annotations</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_adaptor</tt><tt class="py-op">,</tt> </tt>
<a name="L562"></a><tt class="py-lineno">562</tt>  <tt class="py-line">                                                      <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_primary_id</tt><tt class="py-op">,</tt> </tt>
<a name="L563"></a><tt class="py-lineno">563</tt>  <tt class="py-line">                                                      <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_taxon_id</tt><tt class="py-op">)</tt> </tt>
<a name="L564"></a><tt class="py-lineno">564</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_identifier</tt><tt class="py-op">:</tt> </tt>
<a name="L565"></a><tt class="py-lineno">565</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_annotations</tt><tt class="py-op">[</tt><tt class="py-string">"gi"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_identifier</tt> </tt>
<a name="L566"></a><tt class="py-lineno">566</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_division</tt><tt class="py-op">:</tt> </tt>
<a name="L567"></a><tt class="py-lineno">567</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_annotations</tt><tt class="py-op">[</tt><tt class="py-string">"data_file_division"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_division</tt> </tt>
<a name="L568"></a><tt class="py-lineno">568</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_annotations</tt> </tt>
</div><a name="L569"></a><tt class="py-lineno">569</tt>  <tt class="py-line"> </tt>
<a name="DBSeqRecord.__set_annotations"></a><div id="DBSeqRecord.__set_annotations-def"><a name="L570"></a><tt class="py-lineno">570</tt> <a class="py-toggle" href="#" id="DBSeqRecord.__set_annotations-toggle" onclick="return toggle('DBSeqRecord.__set_annotations');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq.DBSeqRecord-class.html#__set_annotations">__set_annotations</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">annotations</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBSeqRecord.__set_annotations-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBSeqRecord.__set_annotations-expanded"><a name="L571"></a><tt class="py-lineno">571</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_annotations</tt> <tt class="py-op">=</tt> <tt id="link-378" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-378', 'annotations', 'link-248');">annotations</a></tt> </tt>
</div><a name="L572"></a><tt class="py-lineno">572</tt>  <tt class="py-line"> </tt>
<a name="DBSeqRecord.__del_annotations"></a><div id="DBSeqRecord.__del_annotations-def"><a name="L573"></a><tt class="py-lineno">573</tt> <a class="py-toggle" href="#" id="DBSeqRecord.__del_annotations-toggle" onclick="return toggle('DBSeqRecord.__del_annotations');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeq.DBSeqRecord-class.html#__del_annotations">__del_annotations</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBSeqRecord.__del_annotations-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBSeqRecord.__del_annotations-expanded"><a name="L574"></a><tt class="py-lineno">574</tt>  <tt class="py-line">        <tt class="py-keyword">del</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">_annotations</tt> </tt>
</div><a name="L575"></a><tt class="py-lineno">575</tt>  <tt class="py-line">    <tt id="link-379" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-379', 'annotations', 'link-248');">annotations</a></tt> <tt class="py-op">=</tt> <tt id="link-380" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Parser.property
Bio.Phylo.PhyloXMLIO.Writer.property" class="py-name" href="#" onclick="return doclink('link-380', 'property', 'link-91');">property</a></tt><tt class="py-op">(</tt><tt id="link-381" class="py-name" targets="Method BioSQL.BioSeq.DBSeqRecord.__get_annotations()=BioSQL.BioSeq.DBSeqRecord-class.html#__get_annotations"><a title="BioSQL.BioSeq.DBSeqRecord.__get_annotations" class="py-name" href="#" onclick="return doclink('link-381', '__get_annotations', 'link-381');">__get_annotations</a></tt><tt class="py-op">,</tt> <tt id="link-382" class="py-name" targets="Method BioSQL.BioSeq.DBSeqRecord.__set_annotations()=BioSQL.BioSeq.DBSeqRecord-class.html#__set_annotations"><a title="BioSQL.BioSeq.DBSeqRecord.__set_annotations" class="py-name" href="#" onclick="return doclink('link-382', '__set_annotations', 'link-382');">__set_annotations</a></tt><tt class="py-op">,</tt> </tt>
<a name="L576"></a><tt class="py-lineno">576</tt>  <tt class="py-line">                           <tt id="link-383" class="py-name" targets="Method BioSQL.BioSeq.DBSeqRecord.__del_annotations()=BioSQL.BioSeq.DBSeqRecord-class.html#__del_annotations"><a title="BioSQL.BioSeq.DBSeqRecord.__del_annotations" class="py-name" href="#" onclick="return doclink('link-383', '__del_annotations', 'link-383');">__del_annotations</a></tt><tt class="py-op">,</tt> <tt class="py-string">"Annotations"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L577"></a><tt class="py-lineno">577</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Oct 21 18:04:27 2015
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
