<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>BioSQL.BioSeqDatabase</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="BioSQL-module.html">Package&nbsp;BioSQL</a> ::
        Module&nbsp;BioSeqDatabase
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="BioSQL.BioSeqDatabase-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="BioSQL.BioSeqDatabase-module.html">Module BioSQL.BioSeqDatabase</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">  1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2002 by Andrew Dalke.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">  2</tt>  <tt class="py-line"><tt class="py-comment"># Revisions 2007-2016 copyright by Peter Cock.  All rights reserved.</tt> </tt>
<a name="L3"></a><tt class="py-lineno">  3</tt>  <tt class="py-line"><tt class="py-comment"># Revisions 2009 copyright by Cymon J. Cox.  All rights reserved.</tt> </tt>
<a name="L4"></a><tt class="py-lineno">  4</tt>  <tt class="py-line"><tt class="py-comment"># Revisions 2013-2014 copyright by Tiago Antao.  All rights reserved.</tt> </tt>
<a name="L5"></a><tt class="py-lineno">  5</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L6"></a><tt class="py-lineno">  6</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L7"></a><tt class="py-lineno">  7</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L8"></a><tt class="py-lineno">  8</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L9"></a><tt class="py-lineno">  9</tt>  <tt class="py-line"><tt class="py-comment"># Note that BioSQL (including the database schema and scripts) is</tt> </tt>
<a name="L10"></a><tt class="py-lineno"> 10</tt>  <tt class="py-line"><tt class="py-comment"># available and licensed separately.  Please consult www.biosql.org</tt> </tt>
<a name="L11"></a><tt class="py-lineno"> 11</tt>  <tt class="py-line"><tt class="py-docstring">"""Connect with a BioSQL database and load Biopython like objects from it.</tt> </tt>
<a name="L12"></a><tt class="py-lineno"> 12</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L13"></a><tt class="py-lineno"> 13</tt>  <tt class="py-line"><tt class="py-docstring">This provides interfaces for loading biological objects from a relational</tt> </tt>
<a name="L14"></a><tt class="py-lineno"> 14</tt>  <tt class="py-line"><tt class="py-docstring">database, and is compatible with the BioSQL standards.</tt> </tt>
<a name="L15"></a><tt class="py-lineno"> 15</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L16"></a><tt class="py-lineno"> 16</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">os</tt> </tt>
<a name="L17"></a><tt class="py-lineno"> 17</tt>  <tt class="py-line"><tt class="py-keyword">import</tt> <tt class="py-name">sys</tt> </tt>
<a name="L18"></a><tt class="py-lineno"> 18</tt>  <tt class="py-line"> </tt>
<a name="L19"></a><tt class="py-lineno"> 19</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-1" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-1', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt id="link-2" class="py-name" targets="Variable Bio._py3k._universal_read_mode=Bio._py3k-module.html#_universal_read_mode"><a title="Bio._py3k._universal_read_mode" class="py-name" href="#" onclick="return doclink('link-2', '_universal_read_mode', 'link-2');">_universal_read_mode</a></tt> </tt>
<a name="L20"></a><tt class="py-lineno"> 20</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-3" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-3', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-4" class="py-name"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-4', '_py3k', 'link-1');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt id="link-5" class="py-name" targets="Function Bio._py3k._bytes_bytearray_to_str()=Bio._py3k-module.html#_bytes_bytearray_to_str"><a title="Bio._py3k._bytes_bytearray_to_str" class="py-name" href="#" onclick="return doclink('link-5', '_bytes_bytearray_to_str', 'link-5');">_bytes_bytearray_to_str</a></tt> <tt class="py-keyword">as</tt> <tt class="py-name">bytearray_to_str</tt> </tt>
<a name="L21"></a><tt class="py-lineno"> 21</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-6" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-6', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-7" class="py-name" targets="Class Bio.BiopythonDeprecationWarning=Bio.BiopythonDeprecationWarning-class.html"><a title="Bio.BiopythonDeprecationWarning" class="py-name" href="#" onclick="return doclink('link-7', 'BiopythonDeprecationWarning', 'link-7');">BiopythonDeprecationWarning</a></tt> </tt>
<a name="L22"></a><tt class="py-lineno"> 22</tt>  <tt class="py-line"> </tt>
<a name="L23"></a><tt class="py-lineno"> 23</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt> <tt class="py-keyword">import</tt> <tt id="link-8" class="py-name" targets="Module BioSQL.BioSeq=BioSQL.BioSeq-module.html"><a title="BioSQL.BioSeq" class="py-name" href="#" onclick="return doclink('link-8', 'BioSeq', 'link-8');">BioSeq</a></tt> </tt>
<a name="L24"></a><tt class="py-lineno"> 24</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt> <tt class="py-keyword">import</tt> <tt id="link-9" class="py-name" targets="Module BioSQL.Loader=BioSQL.Loader-module.html"><a title="BioSQL.Loader" class="py-name" href="#" onclick="return doclink('link-9', 'Loader', 'link-9');">Loader</a></tt> </tt>
<a name="L25"></a><tt class="py-lineno"> 25</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-op">.</tt> <tt class="py-keyword">import</tt> <tt id="link-10" class="py-name" targets="Module BioSQL.DBUtils=BioSQL.DBUtils-module.html"><a title="BioSQL.DBUtils" class="py-name" href="#" onclick="return doclink('link-10', 'DBUtils', 'link-10');">DBUtils</a></tt> </tt>
<a name="L26"></a><tt class="py-lineno"> 26</tt>  <tt class="py-line"> </tt>
<a name="L27"></a><tt class="py-lineno"> 27</tt>  <tt class="py-line"> </tt>
<a name="L28"></a><tt class="py-lineno"> 28</tt>  <tt class="py-line"><tt id="link-11" class="py-name" targets="Variable BioSQL.BioSeqDatabase._POSTGRES_RULES_PRESENT=BioSQL.BioSeqDatabase-module.html#_POSTGRES_RULES_PRESENT"><a title="BioSQL.BioSeqDatabase._POSTGRES_RULES_PRESENT" class="py-name" href="#" onclick="return doclink('link-11', '_POSTGRES_RULES_PRESENT', 'link-11');">_POSTGRES_RULES_PRESENT</a></tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt>  <tt class="py-comment"># Hack for BioSQL Bug 2839</tt> </tt>
<a name="L29"></a><tt class="py-lineno"> 29</tt>  <tt class="py-line"> </tt>
<a name="L30"></a><tt class="py-lineno"> 30</tt>  <tt class="py-line"> </tt>
<a name="open_database"></a><div id="open_database-def"><a name="L31"></a><tt class="py-lineno"> 31</tt> <a class="py-toggle" href="#" id="open_database-toggle" onclick="return toggle('open_database');">-</a><tt class="py-line"><tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase-module.html#open_database">open_database</a><tt class="py-op">(</tt><tt class="py-param">driver</tt><tt class="py-op">=</tt><tt class="py-string">"MySQLdb"</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="open_database-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="open_database-expanded"><a name="L32"></a><tt class="py-lineno"> 32</tt>  <tt class="py-line">    <tt class="py-docstring">"""Load an existing BioSQL-style database.</tt> </tt>
<a name="L33"></a><tt class="py-lineno"> 33</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L34"></a><tt class="py-lineno"> 34</tt>  <tt class="py-line"><tt class="py-docstring">    This function is the easiest way to retrieve a connection to a</tt> </tt>
<a name="L35"></a><tt class="py-lineno"> 35</tt>  <tt class="py-line"><tt class="py-docstring">    database, doing something like:</tt> </tt>
<a name="L36"></a><tt class="py-lineno"> 36</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L37"></a><tt class="py-lineno"> 37</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; from BioSeq import BioSeqDatabase</tt> </tt>
<a name="L38"></a><tt class="py-lineno"> 38</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; server = BioSeqDatabase.open_database(user="root", db="minidb")</tt> </tt>
<a name="L39"></a><tt class="py-lineno"> 39</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L40"></a><tt class="py-lineno"> 40</tt>  <tt class="py-line"><tt class="py-docstring">    Arguments:</tt> </tt>
<a name="L41"></a><tt class="py-lineno"> 41</tt>  <tt class="py-line"><tt class="py-docstring">     - driver - The name of the database driver to use for connecting. The</tt> </tt>
<a name="L42"></a><tt class="py-lineno"> 42</tt>  <tt class="py-line"><tt class="py-docstring">       driver should implement the python DB API. By default, the MySQLdb</tt> </tt>
<a name="L43"></a><tt class="py-lineno"> 43</tt>  <tt class="py-line"><tt class="py-docstring">       driver is used.</tt> </tt>
<a name="L44"></a><tt class="py-lineno"> 44</tt>  <tt class="py-line"><tt class="py-docstring">     - user -the username to connect to the database with.</tt> </tt>
<a name="L45"></a><tt class="py-lineno"> 45</tt>  <tt class="py-line"><tt class="py-docstring">     - password, passwd - the password to connect with</tt> </tt>
<a name="L46"></a><tt class="py-lineno"> 46</tt>  <tt class="py-line"><tt class="py-docstring">     - host - the hostname of the database</tt> </tt>
<a name="L47"></a><tt class="py-lineno"> 47</tt>  <tt class="py-line"><tt class="py-docstring">     - database or db - the name of the database</tt> </tt>
<a name="L48"></a><tt class="py-lineno"> 48</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L49"></a><tt class="py-lineno"> 49</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L50"></a><tt class="py-lineno"> 50</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">driver</tt> <tt class="py-op">==</tt> <tt class="py-string">"psycopg"</tt><tt class="py-op">:</tt> </tt>
<a name="L51"></a><tt class="py-lineno"> 51</tt>  <tt class="py-line">        <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Using BioSQL with psycopg (version one) is no "</tt> </tt>
<a name="L52"></a><tt class="py-lineno"> 52</tt>  <tt class="py-line">                         <tt class="py-string">"longer supported. Use psycopg2 instead."</tt><tt class="py-op">)</tt> </tt>
<a name="L53"></a><tt class="py-lineno"> 53</tt>  <tt class="py-line"> </tt>
<a name="L54"></a><tt class="py-lineno"> 54</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt id="link-12" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.name=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.name=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.name=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#name,Method Bio.Phylo.PhyloXMLIO.Writer.name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#name,Variable Bio.codonalign.codonalphabet.CodonAlphabet.name=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-12', 'name', 'link-12');">name</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"java"</tt><tt class="py-op">:</tt> </tt>
<a name="L55"></a><tt class="py-lineno"> 55</tt>  <tt class="py-line">        <tt class="py-keyword">from</tt> <tt class="py-name">com</tt><tt class="py-op">.</tt><tt class="py-name">ziclix</tt><tt class="py-op">.</tt><tt class="py-name">python</tt><tt class="py-op">.</tt><tt class="py-name">sql</tt> <tt class="py-keyword">import</tt> <tt class="py-name">zxJDBC</tt> </tt>
<a name="L56"></a><tt class="py-lineno"> 56</tt>  <tt class="py-line">        <tt class="py-name">module</tt> <tt class="py-op">=</tt> <tt class="py-name">zxJDBC</tt> </tt>
<a name="L57"></a><tt class="py-lineno"> 57</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">driver</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">"MySQLdb"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L58"></a><tt class="py-lineno"> 58</tt>  <tt class="py-line">            <tt class="py-name">jdbc_driver</tt> <tt class="py-op">=</tt> <tt class="py-string">"com.mysql.jdbc.Driver"</tt> </tt>
<a name="L59"></a><tt class="py-lineno"> 59</tt>  <tt class="py-line">            <tt class="py-name">url_pref</tt> <tt class="py-op">=</tt> <tt class="py-string">"jdbc:mysql://"</tt> <tt class="py-op">+</tt> <tt class="py-name">kwargs</tt><tt class="py-op">[</tt><tt class="py-string">"host"</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-string">"/"</tt> </tt>
<a name="L60"></a><tt class="py-lineno"> 60</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">driver</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">"psycopg2"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L61"></a><tt class="py-lineno"> 61</tt>  <tt class="py-line">            <tt class="py-name">jdbc_driver</tt> <tt class="py-op">=</tt> <tt class="py-string">"org.postgresql.Driver"</tt> </tt>
<a name="L62"></a><tt class="py-lineno"> 62</tt>  <tt class="py-line">            <tt class="py-name">url_pref</tt> <tt class="py-op">=</tt> <tt class="py-string">"jdbc:postgresql://"</tt> <tt class="py-op">+</tt> <tt class="py-name">kwargs</tt><tt class="py-op">[</tt><tt class="py-string">"host"</tt><tt class="py-op">]</tt> <tt class="py-op">+</tt> <tt class="py-string">"/"</tt> </tt>
<a name="L63"></a><tt class="py-lineno"> 63</tt>  <tt class="py-line"> </tt>
<a name="L64"></a><tt class="py-lineno"> 64</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L65"></a><tt class="py-lineno"> 65</tt>  <tt class="py-line">        <tt class="py-name">module</tt> <tt class="py-op">=</tt> <tt class="py-name">__import__</tt><tt class="py-op">(</tt><tt class="py-name">driver</tt><tt class="py-op">,</tt> <tt class="py-name">fromlist</tt><tt class="py-op">=</tt><tt class="py-op">[</tt><tt class="py-string">"connect"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L66"></a><tt class="py-lineno"> 66</tt>  <tt class="py-line">    <tt class="py-name">connect</tt> <tt class="py-op">=</tt> <tt class="py-name">module</tt><tt class="py-op">.</tt><tt class="py-name">connect</tt> </tt>
<a name="L67"></a><tt class="py-lineno"> 67</tt>  <tt class="py-line"> </tt>
<a name="L68"></a><tt class="py-lineno"> 68</tt>  <tt class="py-line">    <tt class="py-comment"># Different drivers use different keywords...</tt> </tt>
<a name="L69"></a><tt class="py-lineno"> 69</tt>  <tt class="py-line">    <tt class="py-name">kw</tt> <tt class="py-op">=</tt> <tt class="py-name">kwargs</tt><tt class="py-op">.</tt><tt id="link-13" class="py-name" targets="Method Bio.Crystal.Crystal.copy()=Bio.Crystal.Crystal-class.html#copy,Method Bio.File._IndexedSeqFileDict.copy()=Bio.File._IndexedSeqFileDict-class.html#copy,Method Bio.GA.Organism.Organism.copy()=Bio.GA.Organism.Organism-class.html#copy,Method Bio.PDB.Atom.Atom.copy()=Bio.PDB.Atom.Atom-class.html#copy,Method Bio.PDB.Entity.Entity.copy()=Bio.PDB.Entity.Entity-class.html#copy,Method Bio.PDB.vectors.Vector.copy()=Bio.PDB.vectors.Vector-class.html#copy"><a title="Bio.Crystal.Crystal.copy
Bio.File._IndexedSeqFileDict.copy
Bio.GA.Organism.Organism.copy
Bio.PDB.Atom.Atom.copy
Bio.PDB.Entity.Entity.copy
Bio.PDB.vectors.Vector.copy" class="py-name" href="#" onclick="return doclink('link-13', 'copy', 'link-13');">copy</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L70"></a><tt class="py-lineno"> 70</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">driver</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">"MySQLdb"</tt><tt class="py-op">,</tt> <tt class="py-string">"mysql.connector"</tt><tt class="py-op">]</tt> <tt class="py-keyword">and</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt id="link-14" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-14', 'name', 'link-12');">name</a></tt> <tt class="py-op">!=</tt> <tt class="py-string">"java"</tt><tt class="py-op">:</tt> </tt>
<a name="L71"></a><tt class="py-lineno"> 71</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"database"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">kw</tt><tt class="py-op">:</tt> </tt>
<a name="L72"></a><tt class="py-lineno"> 72</tt>  <tt class="py-line">            <tt class="py-name">kw</tt><tt class="py-op">[</tt><tt class="py-string">"db"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">kw</tt><tt class="py-op">[</tt><tt class="py-string">"database"</tt><tt class="py-op">]</tt> </tt>
<a name="L73"></a><tt class="py-lineno"> 73</tt>  <tt class="py-line">            <tt class="py-keyword">del</tt> <tt class="py-name">kw</tt><tt class="py-op">[</tt><tt class="py-string">"database"</tt><tt class="py-op">]</tt> </tt>
<a name="L74"></a><tt class="py-lineno"> 74</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"password"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">kw</tt><tt class="py-op">:</tt> </tt>
<a name="L75"></a><tt class="py-lineno"> 75</tt>  <tt class="py-line">            <tt class="py-name">kw</tt><tt class="py-op">[</tt><tt class="py-string">"passwd"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">kw</tt><tt class="py-op">[</tt><tt class="py-string">"password"</tt><tt class="py-op">]</tt> </tt>
<a name="L76"></a><tt class="py-lineno"> 76</tt>  <tt class="py-line">            <tt class="py-keyword">del</tt> <tt class="py-name">kw</tt><tt class="py-op">[</tt><tt class="py-string">"password"</tt><tt class="py-op">]</tt> </tt>
<a name="L77"></a><tt class="py-lineno"> 77</tt>  <tt class="py-line">        <tt class="py-comment"># kw["charset"] = "utf8"</tt> </tt>
<a name="L78"></a><tt class="py-lineno"> 78</tt>  <tt class="py-line">        <tt class="py-comment"># kw["use_unicode"] = True</tt> </tt>
<a name="L79"></a><tt class="py-lineno"> 79</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L80"></a><tt class="py-lineno"> 80</tt>  <tt class="py-line">        <tt class="py-comment"># DB-API recommendations</tt> </tt>
<a name="L81"></a><tt class="py-lineno"> 81</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"db"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">kw</tt><tt class="py-op">:</tt> </tt>
<a name="L82"></a><tt class="py-lineno"> 82</tt>  <tt class="py-line">            <tt class="py-name">kw</tt><tt class="py-op">[</tt><tt class="py-string">"database"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">kw</tt><tt class="py-op">[</tt><tt class="py-string">"db"</tt><tt class="py-op">]</tt> </tt>
<a name="L83"></a><tt class="py-lineno"> 83</tt>  <tt class="py-line">            <tt class="py-keyword">del</tt> <tt class="py-name">kw</tt><tt class="py-op">[</tt><tt class="py-string">"db"</tt><tt class="py-op">]</tt> </tt>
<a name="L84"></a><tt class="py-lineno"> 84</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"passwd"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">kw</tt><tt class="py-op">:</tt> </tt>
<a name="L85"></a><tt class="py-lineno"> 85</tt>  <tt class="py-line">            <tt class="py-name">kw</tt><tt class="py-op">[</tt><tt class="py-string">"password"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">kw</tt><tt class="py-op">[</tt><tt class="py-string">"passwd"</tt><tt class="py-op">]</tt> </tt>
<a name="L86"></a><tt class="py-lineno"> 86</tt>  <tt class="py-line">            <tt class="py-keyword">del</tt> <tt class="py-name">kw</tt><tt class="py-op">[</tt><tt class="py-string">"passwd"</tt><tt class="py-op">]</tt> </tt>
<a name="L87"></a><tt class="py-lineno"> 87</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">driver</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">"psycopg2"</tt><tt class="py-op">,</tt> <tt class="py-string">"pgdb"</tt><tt class="py-op">]</tt> <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt class="py-name">kw</tt><tt class="py-op">.</tt><tt id="link-15" class="py-name" targets="Method Bio.Crystal.Crystal.get()=Bio.Crystal.Crystal-class.html#get,Method Bio.Data.CodonTable.AmbiguousForwardTable.get()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#get,Method Bio.File._IndexedSeqFileDict.get()=Bio.File._IndexedSeqFileDict-class.html#get,Method Bio.File._IndexedSeqFileProxy.get()=Bio.File._IndexedSeqFileProxy-class.html#get,Method Bio.File._SQLiteManySeqFilesDict.get()=Bio.File._SQLiteManySeqFilesDict-class.html#get,Method Bio.Restriction.Restriction.RestrictionBatch.get()=Bio.Restriction.Restriction.RestrictionBatch-class.html#get,Method Bio.SearchIO._index.SearchIndexer.get()=Bio.SearchIO._index.SearchIndexer-class.html#get,Method Bio.SeqIO._index.SeqFileRandomAccess.get()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#get,Method Bio.SeqIO._index.SffRandomAccess.get()=Bio.SeqIO._index.SffRandomAccess-class.html#get,Method Bio.SeqIO._index.SffTrimedRandomAccess.get()=Bio.SeqIO._index.SffTrimedRandomAccess-class.html#get,Method Bio.SeqIO._index.UniprotRandomAccess.get()=Bio.SeqIO._index.UniprotRandomAccess-class.html#get"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-15', 'get', 'link-15');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"database"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L88"></a><tt class="py-lineno"> 88</tt>  <tt class="py-line">        <tt class="py-name">kw</tt><tt class="py-op">[</tt><tt class="py-string">"database"</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">"template1"</tt> </tt>
<a name="L89"></a><tt class="py-lineno"> 89</tt>  <tt class="py-line">    <tt class="py-comment"># SQLite connect takes the database name as input</tt> </tt>
<a name="L90"></a><tt class="py-lineno"> 90</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt id="link-16" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-16', 'name', 'link-12');">name</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"java"</tt><tt class="py-op">:</tt> </tt>
<a name="L91"></a><tt class="py-lineno"> 91</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">driver</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">"MySQLdb"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L92"></a><tt class="py-lineno"> 92</tt>  <tt class="py-line">            <tt class="py-name">conn</tt> <tt class="py-op">=</tt> <tt class="py-name">connect</tt><tt class="py-op">(</tt><tt class="py-name">url_pref</tt> <tt class="py-op">+</tt> <tt class="py-name">kw</tt><tt class="py-op">.</tt><tt id="link-17" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-17', 'get', 'link-15');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"database"</tt><tt class="py-op">,</tt> <tt class="py-string">"mysql"</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L93"></a><tt class="py-lineno"> 93</tt>  <tt class="py-line">                           <tt class="py-name">kw</tt><tt class="py-op">[</tt><tt class="py-string">"user"</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">kw</tt><tt class="py-op">[</tt><tt class="py-string">"password"</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">jdbc_driver</tt><tt class="py-op">)</tt> </tt>
<a name="L94"></a><tt class="py-lineno"> 94</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">driver</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">"psycopg2"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L95"></a><tt class="py-lineno"> 95</tt>  <tt class="py-line">            <tt class="py-name">conn</tt> <tt class="py-op">=</tt> <tt class="py-name">connect</tt><tt class="py-op">(</tt><tt class="py-name">url_pref</tt> <tt class="py-op">+</tt> <tt class="py-name">kw</tt><tt class="py-op">.</tt><tt id="link-18" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-18', 'get', 'link-15');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"database"</tt><tt class="py-op">,</tt> <tt class="py-string">"postgresql"</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> </tt>
<a name="L96"></a><tt class="py-lineno"> 96</tt>  <tt class="py-line">                           <tt class="py-string">"?stringtype=unspecified"</tt><tt class="py-op">,</tt> </tt>
<a name="L97"></a><tt class="py-lineno"> 97</tt>  <tt class="py-line">                           <tt class="py-name">kw</tt><tt class="py-op">[</tt><tt class="py-string">"user"</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">kw</tt><tt class="py-op">[</tt><tt class="py-string">"password"</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">jdbc_driver</tt><tt class="py-op">)</tt> </tt>
<a name="L98"></a><tt class="py-lineno"> 98</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">driver</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">"sqlite3"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L99"></a><tt class="py-lineno"> 99</tt>  <tt class="py-line">        <tt class="py-name">conn</tt> <tt class="py-op">=</tt> <tt class="py-name">connect</tt><tt class="py-op">(</tt><tt class="py-name">kw</tt><tt class="py-op">[</tt><tt class="py-string">"database"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L100"></a><tt class="py-lineno">100</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L101"></a><tt class="py-lineno">101</tt>  <tt class="py-line">        <tt class="py-name">conn</tt> <tt class="py-op">=</tt> <tt class="py-name">connect</tt><tt class="py-op">(</tt><tt class="py-op">**</tt><tt class="py-name">kw</tt><tt class="py-op">)</tt> </tt>
<a name="L102"></a><tt class="py-lineno">102</tt>  <tt class="py-line"> </tt>
<a name="L103"></a><tt class="py-lineno">103</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt id="link-19" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-19', 'name', 'link-12');">name</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"java"</tt><tt class="py-op">:</tt> </tt>
<a name="L104"></a><tt class="py-lineno">104</tt>  <tt class="py-line">        <tt class="py-name">server</tt> <tt class="py-op">=</tt> <tt id="link-20" class="py-name" targets="Class BioSQL.BioSeqDatabase.DBServer=BioSQL.BioSeqDatabase.DBServer-class.html"><a title="BioSQL.BioSeqDatabase.DBServer" class="py-name" href="#" onclick="return doclink('link-20', 'DBServer', 'link-20');">DBServer</a></tt><tt class="py-op">(</tt><tt class="py-name">conn</tt><tt class="py-op">,</tt> <tt class="py-name">module</tt><tt class="py-op">,</tt> <tt class="py-name">driver</tt><tt class="py-op">)</tt> </tt>
<a name="L105"></a><tt class="py-lineno">105</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L106"></a><tt class="py-lineno">106</tt>  <tt class="py-line">        <tt class="py-name">server</tt> <tt class="py-op">=</tt> <tt id="link-21" class="py-name"><a title="BioSQL.BioSeqDatabase.DBServer" class="py-name" href="#" onclick="return doclink('link-21', 'DBServer', 'link-20');">DBServer</a></tt><tt class="py-op">(</tt><tt class="py-name">conn</tt><tt class="py-op">,</tt> <tt class="py-name">module</tt><tt class="py-op">)</tt> </tt>
<a name="L107"></a><tt class="py-lineno">107</tt>  <tt class="py-line"> </tt>
<a name="L108"></a><tt class="py-lineno">108</tt>  <tt class="py-line">    <tt class="py-comment"># TODO - Remove the following once BioSQL Bug 2839 is fixed.</tt> </tt>
<a name="L109"></a><tt class="py-lineno">109</tt>  <tt class="py-line">    <tt class="py-comment"># Test for RULES in PostgreSQL schema, see also Bug 2833.</tt> </tt>
<a name="L110"></a><tt class="py-lineno">110</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">driver</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">"psycopg2"</tt><tt class="py-op">,</tt> <tt class="py-string">"pgdb"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L111"></a><tt class="py-lineno">111</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">"SELECT ev_class FROM pg_rewrite WHERE "</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L112"></a><tt class="py-lineno">112</tt>  <tt class="py-line">              <tt class="py-string">"rulename='rule_bioentry_i1' OR "</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L113"></a><tt class="py-lineno">113</tt>  <tt class="py-line">              <tt class="py-string">"rulename='rule_bioentry_i2';"</tt> </tt>
<a name="L114"></a><tt class="py-lineno">114</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">server</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-22" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.execute_and_fetchall()=BioSQL.BioSeqDatabase.Adaptor-class.html#execute_and_fetchall,Method BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetchall()=BioSQL.BioSeqDatabase.MysqlConnectorAdaptor-class.html#execute_and_fetchall"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetchall
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetchall" class="py-name" href="#" onclick="return doclink('link-22', 'execute_and_fetchall', 'link-22');">execute_and_fetchall</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L115"></a><tt class="py-lineno">115</tt>  <tt class="py-line">            <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L116"></a><tt class="py-lineno">116</tt>  <tt class="py-line">            <tt class="py-keyword">from</tt> <tt id="link-23" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-23', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-24" class="py-name" targets="Class Bio.BiopythonWarning=Bio.BiopythonWarning-class.html"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-24', 'BiopythonWarning', 'link-24');">BiopythonWarning</a></tt> </tt>
<a name="L117"></a><tt class="py-lineno">117</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Your BioSQL PostgreSQL schema includes some rules "</tt> </tt>
<a name="L118"></a><tt class="py-lineno">118</tt>  <tt class="py-line">                          <tt class="py-string">"currently required for bioperl-db but which may"</tt> </tt>
<a name="L119"></a><tt class="py-lineno">119</tt>  <tt class="py-line">                          <tt class="py-string">"cause problems loading data using Biopython (see "</tt> </tt>
<a name="L120"></a><tt class="py-lineno">120</tt>  <tt class="py-line">                          <tt class="py-string">"BioSQL's RedMine Bug 2839 aka GitHub Issue 4 "</tt> </tt>
<a name="L121"></a><tt class="py-lineno">121</tt>  <tt class="py-line">                          <tt class="py-string">"https://github.com/biosql/biosql/issues/4). "</tt> </tt>
<a name="L122"></a><tt class="py-lineno">122</tt>  <tt class="py-line">                          <tt class="py-string">"If you do not use BioPerl, please remove these "</tt> </tt>
<a name="L123"></a><tt class="py-lineno">123</tt>  <tt class="py-line">                          <tt class="py-string">"rules. Biopython should cope with the rules "</tt> </tt>
<a name="L124"></a><tt class="py-lineno">124</tt>  <tt class="py-line">                          <tt class="py-string">"present, but with a performance penalty when "</tt> </tt>
<a name="L125"></a><tt class="py-lineno">125</tt>  <tt class="py-line">                          <tt class="py-string">"loading new records."</tt><tt class="py-op">,</tt> <tt id="link-25" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-25', 'BiopythonWarning', 'link-24');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L126"></a><tt class="py-lineno">126</tt>  <tt class="py-line">            <tt class="py-keyword">global</tt> <tt id="link-26" class="py-name"><a title="BioSQL.BioSeqDatabase._POSTGRES_RULES_PRESENT" class="py-name" href="#" onclick="return doclink('link-26', '_POSTGRES_RULES_PRESENT', 'link-11');">_POSTGRES_RULES_PRESENT</a></tt> </tt>
<a name="L127"></a><tt class="py-lineno">127</tt>  <tt class="py-line">            <tt id="link-27" class="py-name"><a title="BioSQL.BioSeqDatabase._POSTGRES_RULES_PRESENT" class="py-name" href="#" onclick="return doclink('link-27', '_POSTGRES_RULES_PRESENT', 'link-11');">_POSTGRES_RULES_PRESENT</a></tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L128"></a><tt class="py-lineno">128</tt>  <tt class="py-line"> </tt>
<a name="L129"></a><tt class="py-lineno">129</tt>  <tt class="py-line">    <tt class="py-keyword">elif</tt> <tt class="py-name">driver</tt> <tt class="py-op">==</tt> <tt class="py-string">'sqlite3'</tt><tt class="py-op">:</tt> </tt>
<a name="L130"></a><tt class="py-lineno">130</tt>  <tt class="py-line">        <tt class="py-comment"># Tell SQLite that we want to use foreign keys</tt> </tt>
<a name="L131"></a><tt class="py-lineno">131</tt>  <tt class="py-line">        <tt class="py-comment"># https://www.sqlite.org/foreignkeys.html#fk_enable</tt> </tt>
<a name="L132"></a><tt class="py-lineno">132</tt>  <tt class="py-line">        <tt class="py-name">server</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-28" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.execute()=BioSQL.BioSeqDatabase.Adaptor-class.html#execute,Method BioSQL.BioSeqDatabase._CursorWrapper.execute()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#execute,Method BioSQL.DBUtils.Generic_dbutils.execute()=BioSQL.DBUtils.Generic_dbutils-class.html#execute,Method BioSQL.DBUtils.Sqlite_dbutils.execute()=BioSQL.DBUtils.Sqlite_dbutils-class.html#execute"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-28', 'execute', 'link-28');">execute</a></tt><tt class="py-op">(</tt><tt class="py-string">'PRAGMA foreign_keys = ON'</tt><tt class="py-op">)</tt> </tt>
<a name="L133"></a><tt class="py-lineno">133</tt>  <tt class="py-line"> </tt>
<a name="L134"></a><tt class="py-lineno">134</tt>  <tt class="py-line">    <tt class="py-keyword">return</tt> <tt class="py-name">server</tt> </tt>
</div><a name="L135"></a><tt class="py-lineno">135</tt>  <tt class="py-line"> </tt>
<a name="L136"></a><tt class="py-lineno">136</tt>  <tt class="py-line"> </tt>
<a name="DBServer"></a><div id="DBServer-def"><a name="L137"></a><tt class="py-lineno">137</tt> <a class="py-toggle" href="#" id="DBServer-toggle" onclick="return toggle('DBServer');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.DBServer-class.html">DBServer</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBServer-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="DBServer-expanded"><a name="L138"></a><tt class="py-lineno">138</tt>  <tt class="py-line">    <tt class="py-docstring">"""Represents a BioSQL database continaing namespaces (sub-databases).</tt> </tt>
<a name="L139"></a><tt class="py-lineno">139</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L140"></a><tt class="py-lineno">140</tt>  <tt class="py-line"><tt class="py-docstring">    This acts like a Python dictionary, giving access to each namespace</tt> </tt>
<a name="L141"></a><tt class="py-lineno">141</tt>  <tt class="py-line"><tt class="py-docstring">    (defined by a row in the biodatabase table) as a BioSeqDatabase object.</tt> </tt>
<a name="L142"></a><tt class="py-lineno">142</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L143"></a><tt class="py-lineno">143</tt>  <tt class="py-line"> </tt>
<a name="DBServer.__init__"></a><div id="DBServer.__init__-def"><a name="L144"></a><tt class="py-lineno">144</tt> <a class="py-toggle" href="#" id="DBServer.__init__-toggle" onclick="return toggle('DBServer.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.DBServer-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">conn</tt><tt class="py-op">,</tt> <tt class="py-param">module</tt><tt class="py-op">,</tt> <tt class="py-param">module_name</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBServer.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBServer.__init__-expanded"><a name="L145"></a><tt class="py-lineno">145</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a DBServer object.</tt> </tt>
<a name="L146"></a><tt class="py-lineno">146</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L147"></a><tt class="py-lineno">147</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L148"></a><tt class="py-lineno">148</tt>  <tt class="py-line"><tt class="py-docstring">         - conn - A database connection object</tt> </tt>
<a name="L149"></a><tt class="py-lineno">149</tt>  <tt class="py-line"><tt class="py-docstring">         - module - The module used to create the database connection</tt> </tt>
<a name="L150"></a><tt class="py-lineno">150</tt>  <tt class="py-line"><tt class="py-docstring">         - module_name - Optionally, the name of the module. Default: module.__name__</tt> </tt>
<a name="L151"></a><tt class="py-lineno">151</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L152"></a><tt class="py-lineno">152</tt>  <tt class="py-line"><tt class="py-docstring">        Normally you would not want to create a DBServer object yourself.</tt> </tt>
<a name="L153"></a><tt class="py-lineno">153</tt>  <tt class="py-line"><tt class="py-docstring">        Instead use the open_database function, which returns an instance of DBServer.</tt> </tt>
<a name="L154"></a><tt class="py-lineno">154</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L155"></a><tt class="py-lineno">155</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">module</tt> <tt class="py-op">=</tt> <tt class="py-name">module</tt> </tt>
<a name="L156"></a><tt class="py-lineno">156</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">module_name</tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L157"></a><tt class="py-lineno">157</tt>  <tt class="py-line">            <tt class="py-name">module_name</tt> <tt class="py-op">=</tt> <tt class="py-name">module</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt> </tt>
<a name="L158"></a><tt class="py-lineno">158</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">module_name</tt> <tt class="py-op">==</tt> <tt class="py-string">"mysql.connector"</tt> <tt class="py-keyword">and</tt> <tt class="py-name">sys</tt><tt class="py-op">.</tt><tt class="py-name">version_info</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-number">3</tt><tt class="py-op">:</tt> </tt>
<a name="L159"></a><tt class="py-lineno">159</tt>  <tt class="py-line">            <tt class="py-name">wrap_cursor</tt> <tt class="py-op">=</tt> <tt class="py-name">True</tt> </tt>
<a name="L160"></a><tt class="py-lineno">160</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L161"></a><tt class="py-lineno">161</tt>  <tt class="py-line">            <tt class="py-name">wrap_cursor</tt> <tt class="py-op">=</tt> <tt class="py-name">False</tt> </tt>
<a name="L162"></a><tt class="py-lineno">162</tt>  <tt class="py-line">        <tt class="py-comment"># Get module specific Adaptor or the base (general) Adaptor</tt> </tt>
<a name="L163"></a><tt class="py-lineno">163</tt>  <tt class="py-line">        <tt class="py-name">Adapt</tt> <tt class="py-op">=</tt> <tt id="link-29" class="py-name" targets="Variable BioSQL.BioSeqDatabase._interface_specific_adaptors=BioSQL.BioSeqDatabase-module.html#_interface_specific_adaptors"><a title="BioSQL.BioSeqDatabase._interface_specific_adaptors" class="py-name" href="#" onclick="return doclink('link-29', '_interface_specific_adaptors', 'link-29');">_interface_specific_adaptors</a></tt><tt class="py-op">.</tt><tt id="link-30" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-30', 'get', 'link-15');">get</a></tt><tt class="py-op">(</tt><tt class="py-name">module_name</tt><tt class="py-op">,</tt> <tt id="link-31" class="py-name" targets="Class BioSQL.BioSeqDatabase.Adaptor=BioSQL.BioSeqDatabase.Adaptor-class.html"><a title="BioSQL.BioSeqDatabase.Adaptor" class="py-name" href="#" onclick="return doclink('link-31', 'Adaptor', 'link-31');">Adaptor</a></tt><tt class="py-op">)</tt> </tt>
<a name="L164"></a><tt class="py-lineno">164</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt> <tt class="py-op">=</tt> <tt class="py-name">Adapt</tt><tt class="py-op">(</tt><tt class="py-name">conn</tt><tt class="py-op">,</tt> <tt id="link-32" class="py-name"><a title="BioSQL.DBUtils" class="py-name" href="#" onclick="return doclink('link-32', 'DBUtils', 'link-10');">DBUtils</a></tt><tt class="py-op">.</tt><tt id="link-33" class="py-name" targets="Function BioSQL.DBUtils.get_dbutils()=BioSQL.DBUtils-module.html#get_dbutils"><a title="BioSQL.DBUtils.get_dbutils" class="py-name" href="#" onclick="return doclink('link-33', 'get_dbutils', 'link-33');">get_dbutils</a></tt><tt class="py-op">(</tt><tt class="py-name">module_name</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L165"></a><tt class="py-lineno">165</tt>  <tt class="py-line">                             <tt class="py-name">wrap_cursor</tt><tt class="py-op">=</tt><tt class="py-name">wrap_cursor</tt><tt class="py-op">)</tt> </tt>
<a name="L166"></a><tt class="py-lineno">166</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">module_name</tt> <tt class="py-op">=</tt> <tt class="py-name">module_name</tt> </tt>
</div><a name="L167"></a><tt class="py-lineno">167</tt>  <tt class="py-line"> </tt>
<a name="DBServer.__repr__"></a><div id="DBServer.__repr__-def"><a name="L168"></a><tt class="py-lineno">168</tt> <a class="py-toggle" href="#" id="DBServer.__repr__-toggle" onclick="return toggle('DBServer.__repr__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.DBServer-class.html#__repr__">__repr__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBServer.__repr__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBServer.__repr__-expanded"><a name="L169"></a><tt class="py-lineno">169</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a short description of the class name and database connection."""</tt> </tt>
<a name="L170"></a><tt class="py-lineno">170</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">__class__</tt><tt class="py-op">.</tt><tt class="py-name">__name__</tt> <tt class="py-op">+</tt> <tt class="py-string">"(%r)"</tt> <tt class="py-op">%</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt class="py-name">conn</tt> </tt>
</div><a name="L171"></a><tt class="py-lineno">171</tt>  <tt class="py-line"> </tt>
<a name="DBServer.__getitem__"></a><div id="DBServer.__getitem__-def"><a name="L172"></a><tt class="py-lineno">172</tt> <a class="py-toggle" href="#" id="DBServer.__getitem__-toggle" onclick="return toggle('DBServer.__getitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.DBServer-class.html#__getitem__">__getitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBServer.__getitem__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBServer.__getitem__-expanded"><a name="L173"></a><tt class="py-lineno">173</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a BioSeqDatabase object.</tt> </tt>
<a name="L174"></a><tt class="py-lineno">174</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L175"></a><tt class="py-lineno">175</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L176"></a><tt class="py-lineno">176</tt>  <tt class="py-line"><tt class="py-docstring">            - name - The name of the BioSeqDatabase</tt> </tt>
<a name="L177"></a><tt class="py-lineno">177</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L178"></a><tt class="py-lineno">178</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L179"></a><tt class="py-lineno">179</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-34" class="py-name" targets="Module BioSQL.BioSeqDatabase=BioSQL.BioSeqDatabase-module.html,Class BioSQL.BioSeqDatabase.BioSeqDatabase=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html"><a title="BioSQL.BioSeqDatabase
BioSQL.BioSeqDatabase.BioSeqDatabase" class="py-name" href="#" onclick="return doclink('link-34', 'BioSeqDatabase', 'link-34');">BioSeqDatabase</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">,</tt> <tt id="link-35" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-35', 'name', 'link-12');">name</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L180"></a><tt class="py-lineno">180</tt>  <tt class="py-line"> </tt>
<a name="DBServer.__len__"></a><div id="DBServer.__len__-def"><a name="L181"></a><tt class="py-lineno">181</tt> <a class="py-toggle" href="#" id="DBServer.__len__-toggle" onclick="return toggle('DBServer.__len__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.DBServer-class.html#__len__">__len__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBServer.__len__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBServer.__len__-expanded"><a name="L182"></a><tt class="py-lineno">182</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return number of namespaces (sub-databases) in this database."""</tt> </tt>
<a name="L183"></a><tt class="py-lineno">183</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">"SELECT COUNT(name) FROM biodatabase;"</tt> </tt>
<a name="L184"></a><tt class="py-lineno">184</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-36" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0()=BioSQL.BioSeqDatabase.Adaptor-class.html#execute_and_fetch_col0,Method BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0()=BioSQL.BioSeqDatabase.MysqlConnectorAdaptor-class.html#execute_and_fetch_col0"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0" class="py-name" href="#" onclick="return doclink('link-36', 'execute_and_fetch_col0', 'link-36');">execute_and_fetch_col0</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div><a name="L185"></a><tt class="py-lineno">185</tt>  <tt class="py-line"> </tt>
<a name="DBServer.__contains__"></a><div id="DBServer.__contains__-def"><a name="L186"></a><tt class="py-lineno">186</tt> <a class="py-toggle" href="#" id="DBServer.__contains__-toggle" onclick="return toggle('DBServer.__contains__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.DBServer-class.html#__contains__">__contains__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBServer.__contains__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBServer.__contains__-expanded"><a name="L187"></a><tt class="py-lineno">187</tt>  <tt class="py-line">        <tt class="py-docstring">"""Check if a namespace (sub-database) in this database."""</tt> </tt>
<a name="L188"></a><tt class="py-lineno">188</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">"SELECT COUNT(name) FROM biodatabase WHERE name=%s;"</tt> </tt>
<a name="L189"></a><tt class="py-lineno">189</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">bool</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-37" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0" class="py-name" href="#" onclick="return doclink('link-37', 'execute_and_fetch_col0', 'link-36');">execute_and_fetch_col0</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt id="link-38" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.value()=Bio.Phylo.PhyloXMLIO.Writer-class.html#value"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-38', 'value', 'link-38');">value</a></tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div><a name="L190"></a><tt class="py-lineno">190</tt>  <tt class="py-line"> </tt>
<a name="DBServer.__iter__"></a><div id="DBServer.__iter__-def"><a name="L191"></a><tt class="py-lineno">191</tt> <a class="py-toggle" href="#" id="DBServer.__iter__-toggle" onclick="return toggle('DBServer.__iter__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.DBServer-class.html#__iter__">__iter__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBServer.__iter__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBServer.__iter__-expanded"><a name="L192"></a><tt class="py-lineno">192</tt>  <tt class="py-line">        <tt class="py-docstring">"""Iterate over namespaces (sub-databases) in the database."""</tt> </tt>
<a name="L193"></a><tt class="py-lineno">193</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - Iterate over the cursor, much more efficient</tt> </tt>
<a name="L194"></a><tt class="py-lineno">194</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">iter</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-39" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.list_biodatabase_names()=BioSQL.BioSeqDatabase.Adaptor-class.html#list_biodatabase_names"><a title="BioSQL.BioSeqDatabase.Adaptor.list_biodatabase_names" class="py-name" href="#" onclick="return doclink('link-39', 'list_biodatabase_names', 'link-39');">list_biodatabase_names</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L195"></a><tt class="py-lineno">195</tt>  <tt class="py-line"> </tt>
<a name="L196"></a><tt class="py-lineno">196</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">dict</tt><tt class="py-op">,</tt> <tt class="py-string">"iteritems"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L197"></a><tt class="py-lineno">197</tt>  <tt class="py-line">        <tt class="py-comment"># Python 2, use iteritems etc</tt> </tt>
<a name="DBServer.keys"></a><div id="DBServer.keys-def"><a name="L198"></a><tt class="py-lineno">198</tt> <a class="py-toggle" href="#" id="DBServer.keys-toggle" onclick="return toggle('DBServer.keys');">-</a><tt class="py-line">        <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.DBServer-class.html#keys">keys</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBServer.keys-collapsed" style="display:none;" pad="+++" indent="++++++++++++++++"></div><div id="DBServer.keys-expanded"><a name="L199"></a><tt class="py-lineno">199</tt>  <tt class="py-line">            <tt class="py-docstring">"""List of namespaces (sub-databases) in the database."""</tt> </tt>
<a name="L200"></a><tt class="py-lineno">200</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-40" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.list_biodatabase_names" class="py-name" href="#" onclick="return doclink('link-40', 'list_biodatabase_names', 'link-39');">list_biodatabase_names</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L201"></a><tt class="py-lineno">201</tt>  <tt class="py-line"> </tt>
<a name="DBServer.values"></a><div id="DBServer.values-def"><a name="L202"></a><tt class="py-lineno">202</tt> <a class="py-toggle" href="#" id="DBServer.values-toggle" onclick="return toggle('DBServer.values');">-</a><tt class="py-line">        <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.DBServer-class.html#values">values</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBServer.values-collapsed" style="display:none;" pad="+++" indent="++++++++++++++++"></div><div id="DBServer.values-expanded"><a name="L203"></a><tt class="py-lineno">203</tt>  <tt class="py-line">            <tt class="py-docstring">"""List of BioSeqDatabase objects in the database."""</tt> </tt>
<a name="L204"></a><tt class="py-lineno">204</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">]</tt> </tt>
</div><a name="L205"></a><tt class="py-lineno">205</tt>  <tt class="py-line"> </tt>
<a name="DBServer.items"></a><div id="DBServer.items-def"><a name="L206"></a><tt class="py-lineno">206</tt> <a class="py-toggle" href="#" id="DBServer.items-toggle" onclick="return toggle('DBServer.items');">-</a><tt class="py-line">        <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.DBServer-class.html#items">items</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBServer.items-collapsed" style="display:none;" pad="+++" indent="++++++++++++++++"></div><div id="DBServer.items-expanded"><a name="L207"></a><tt class="py-lineno">207</tt>  <tt class="py-line">            <tt class="py-docstring">"""List of (namespace, BioSeqDatabase) for entries in database."""</tt> </tt>
<a name="L208"></a><tt class="py-lineno">208</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">]</tt> </tt>
</div><a name="L209"></a><tt class="py-lineno">209</tt>  <tt class="py-line"> </tt>
<a name="DBServer.iterkeys"></a><div id="DBServer.iterkeys-def"><a name="L210"></a><tt class="py-lineno">210</tt> <a class="py-toggle" href="#" id="DBServer.iterkeys-toggle" onclick="return toggle('DBServer.iterkeys');">-</a><tt class="py-line">        <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.DBServer-class.html#iterkeys">iterkeys</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBServer.iterkeys-collapsed" style="display:none;" pad="+++" indent="++++++++++++++++"></div><div id="DBServer.iterkeys-expanded"><a name="L211"></a><tt class="py-lineno">211</tt>  <tt class="py-line">            <tt class="py-docstring">"""Iterate over namespaces (sub-databases) in the database."""</tt> </tt>
<a name="L212"></a><tt class="py-lineno">212</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">iter</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
</div><a name="L213"></a><tt class="py-lineno">213</tt>  <tt class="py-line"> </tt>
<a name="DBServer.itervalues"></a><div id="DBServer.itervalues-def"><a name="L214"></a><tt class="py-lineno">214</tt> <a class="py-toggle" href="#" id="DBServer.itervalues-toggle" onclick="return toggle('DBServer.itervalues');">-</a><tt class="py-line">        <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.DBServer-class.html#itervalues">itervalues</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBServer.itervalues-collapsed" style="display:none;" pad="+++" indent="++++++++++++++++"></div><div id="DBServer.itervalues-expanded"><a name="L215"></a><tt class="py-lineno">215</tt>  <tt class="py-line">            <tt class="py-docstring">"""Iterate over BioSeqDatabase objects in the database."""</tt> </tt>
<a name="L216"></a><tt class="py-lineno">216</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">:</tt> </tt>
<a name="L217"></a><tt class="py-lineno">217</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt class="py-name">self</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> </tt>
</div><a name="L218"></a><tt class="py-lineno">218</tt>  <tt class="py-line"> </tt>
<a name="DBServer.iteritems"></a><div id="DBServer.iteritems-def"><a name="L219"></a><tt class="py-lineno">219</tt> <a class="py-toggle" href="#" id="DBServer.iteritems-toggle" onclick="return toggle('DBServer.iteritems');">-</a><tt class="py-line">        <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.DBServer-class.html#iteritems">iteritems</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBServer.iteritems-collapsed" style="display:none;" pad="+++" indent="++++++++++++++++"></div><div id="DBServer.iteritems-expanded"><a name="L220"></a><tt class="py-lineno">220</tt>  <tt class="py-line">            <tt class="py-docstring">"""Iterate over (namespace, BioSeqDatabase) in the database."""</tt> </tt>
<a name="L221"></a><tt class="py-lineno">221</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">:</tt> </tt>
<a name="L222"></a><tt class="py-lineno">222</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> </tt>
</div><a name="L223"></a><tt class="py-lineno">223</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L224"></a><tt class="py-lineno">224</tt>  <tt class="py-line">        <tt class="py-comment"># Python 3, items etc are all iterators</tt> </tt>
<a name="DBServer.keys"></a><div id="DBServer.keys-def"><a name="L225"></a><tt class="py-lineno">225</tt> <a class="py-toggle" href="#" id="DBServer.keys-toggle" onclick="return toggle('DBServer.keys');">-</a><tt class="py-line">        <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.DBServer-class.html#keys">keys</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBServer.keys-collapsed" style="display:none;" pad="+++" indent="++++++++++++++++"></div><div id="DBServer.keys-expanded"><a name="L226"></a><tt class="py-lineno">226</tt>  <tt class="py-line">            <tt class="py-docstring">"""Iterate over namespaces (sub-databases) in the database."""</tt> </tt>
<a name="L227"></a><tt class="py-lineno">227</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">iter</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
</div><a name="L228"></a><tt class="py-lineno">228</tt>  <tt class="py-line"> </tt>
<a name="DBServer.values"></a><div id="DBServer.values-def"><a name="L229"></a><tt class="py-lineno">229</tt> <a class="py-toggle" href="#" id="DBServer.values-toggle" onclick="return toggle('DBServer.values');">-</a><tt class="py-line">        <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.DBServer-class.html#values">values</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBServer.values-collapsed" style="display:none;" pad="+++" indent="++++++++++++++++"></div><div id="DBServer.values-expanded"><a name="L230"></a><tt class="py-lineno">230</tt>  <tt class="py-line">            <tt class="py-docstring">"""Iterate over BioSeqDatabase objects in the database."""</tt> </tt>
<a name="L231"></a><tt class="py-lineno">231</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">:</tt> </tt>
<a name="L232"></a><tt class="py-lineno">232</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt class="py-name">self</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> </tt>
</div><a name="L233"></a><tt class="py-lineno">233</tt>  <tt class="py-line"> </tt>
<a name="DBServer.items"></a><div id="DBServer.items-def"><a name="L234"></a><tt class="py-lineno">234</tt> <a class="py-toggle" href="#" id="DBServer.items-toggle" onclick="return toggle('DBServer.items');">-</a><tt class="py-line">        <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.DBServer-class.html#items">items</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBServer.items-collapsed" style="display:none;" pad="+++" indent="++++++++++++++++"></div><div id="DBServer.items-expanded"><a name="L235"></a><tt class="py-lineno">235</tt>  <tt class="py-line">            <tt class="py-docstring">"""Iterate over (namespace, BioSeqDatabase) in the database."""</tt> </tt>
<a name="L236"></a><tt class="py-lineno">236</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">:</tt> </tt>
<a name="L237"></a><tt class="py-lineno">237</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> </tt>
</div><a name="L238"></a><tt class="py-lineno">238</tt>  <tt class="py-line"> </tt>
<a name="DBServer.__delitem__"></a><div id="DBServer.__delitem__-def"><a name="L239"></a><tt class="py-lineno">239</tt> <a class="py-toggle" href="#" id="DBServer.__delitem__-toggle" onclick="return toggle('DBServer.__delitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.DBServer-class.html#__delitem__">__delitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBServer.__delitem__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBServer.__delitem__-expanded"><a name="L240"></a><tt class="py-lineno">240</tt>  <tt class="py-line">        <tt class="py-docstring">"""Remove a namespace and all its entries."""</tt> </tt>
<a name="L241"></a><tt class="py-lineno">241</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-41" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-41', 'name', 'link-12');">name</a></tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">:</tt> </tt>
<a name="L242"></a><tt class="py-lineno">242</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">KeyError</tt><tt class="py-op">(</tt><tt id="link-42" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-42', 'name', 'link-12');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L243"></a><tt class="py-lineno">243</tt>  <tt class="py-line">        <tt class="py-name">db_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-43" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.fetch_dbid_by_dbname()=BioSQL.BioSeqDatabase.Adaptor-class.html#fetch_dbid_by_dbname"><a title="BioSQL.BioSeqDatabase.Adaptor.fetch_dbid_by_dbname" class="py-name" href="#" onclick="return doclink('link-43', 'fetch_dbid_by_dbname', 'link-43');">fetch_dbid_by_dbname</a></tt><tt class="py-op">(</tt><tt id="link-44" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-44', 'name', 'link-12');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L244"></a><tt class="py-lineno">244</tt>  <tt class="py-line">        <tt class="py-name">remover</tt> <tt class="py-op">=</tt> <tt id="link-45" class="py-name"><a title="BioSQL.Loader" class="py-name" href="#" onclick="return doclink('link-45', 'Loader', 'link-9');">Loader</a></tt><tt class="py-op">.</tt><tt id="link-46" class="py-name" targets="Class BioSQL.Loader.DatabaseRemover=BioSQL.Loader.DatabaseRemover-class.html"><a title="BioSQL.Loader.DatabaseRemover" class="py-name" href="#" onclick="return doclink('link-46', 'DatabaseRemover', 'link-46');">DatabaseRemover</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">,</tt> <tt class="py-name">db_id</tt><tt class="py-op">)</tt> </tt>
<a name="L245"></a><tt class="py-lineno">245</tt>  <tt class="py-line">        <tt class="py-name">remover</tt><tt class="py-op">.</tt><tt id="link-47" class="py-name" targets="Method Bio.Crystal.Chain.remove()=Bio.Crystal.Chain-class.html#remove,Method Bio.Graphics.BasicChromosome._ChromosomeComponent.remove()=Bio.Graphics.BasicChromosome._ChromosomeComponent-class.html#remove,Method Bio.Restriction.Restriction.RestrictionBatch.remove()=Bio.Restriction.Restriction.RestrictionBatch-class.html#remove,Method Bio.Seq.MutableSeq.remove()=Bio.Seq.MutableSeq-class.html#remove,Method BioSQL.Loader.DatabaseRemover.remove()=BioSQL.Loader.DatabaseRemover-class.html#remove"><a title="Bio.Crystal.Chain.remove
Bio.Graphics.BasicChromosome._ChromosomeComponent.remove
Bio.Restriction.Restriction.RestrictionBatch.remove
Bio.Seq.MutableSeq.remove
BioSQL.Loader.DatabaseRemover.remove" class="py-name" href="#" onclick="return doclink('link-47', 'remove', 'link-47');">remove</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L246"></a><tt class="py-lineno">246</tt>  <tt class="py-line"> </tt>
<a name="DBServer.remove_database"></a><div id="DBServer.remove_database-def"><a name="L247"></a><tt class="py-lineno">247</tt> <a class="py-toggle" href="#" id="DBServer.remove_database-toggle" onclick="return toggle('DBServer.remove_database');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.DBServer-class.html#remove_database">remove_database</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">db_name</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBServer.remove_database-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBServer.remove_database-expanded"><a name="L248"></a><tt class="py-lineno">248</tt>  <tt class="py-line">        <tt class="py-docstring">"""Remove a namespace and all its entries (OBSOLETE).</tt> </tt>
<a name="L249"></a><tt class="py-lineno">249</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L250"></a><tt class="py-lineno">250</tt>  <tt class="py-line"><tt class="py-docstring">        Try to remove all references to items in a database:</tt> </tt>
<a name="L251"></a><tt class="py-lineno">251</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L252"></a><tt class="py-lineno">252</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; server.remove_database(name)</tt> </tt>
<a name="L253"></a><tt class="py-lineno">253</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L254"></a><tt class="py-lineno">254</tt>  <tt class="py-line"><tt class="py-docstring">        In keeping with the dictionary interface, you can now do this:</tt> </tt>
<a name="L255"></a><tt class="py-lineno">255</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L256"></a><tt class="py-lineno">256</tt>  <tt class="py-line"><tt class="py-docstring">        &gt;&gt;&gt; del server[name]</tt> </tt>
<a name="L257"></a><tt class="py-lineno">257</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L258"></a><tt class="py-lineno">258</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L259"></a><tt class="py-lineno">259</tt>  <tt class="py-line">        <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L260"></a><tt class="py-lineno">260</tt>  <tt class="py-line">        <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"This method is deprecated.  In keeping with the "</tt> </tt>
<a name="L261"></a><tt class="py-lineno">261</tt>  <tt class="py-line">                      <tt class="py-string">"dictionary interface, you can now use 'del "</tt> </tt>
<a name="L262"></a><tt class="py-lineno">262</tt>  <tt class="py-line">                      <tt class="py-string">"server[name]' instead"</tt><tt class="py-op">,</tt> <tt id="link-48" class="py-name"><a title="Bio.BiopythonDeprecationWarning" class="py-name" href="#" onclick="return doclink('link-48', 'BiopythonDeprecationWarning', 'link-7');">BiopythonDeprecationWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L263"></a><tt class="py-lineno">263</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-49" class="py-name" targets="Method Bio.Crystal.Chain.__delitem__()=Bio.Crystal.Chain-class.html#__delitem__,Method Bio.Crystal.Crystal.__delitem__()=Bio.Crystal.Crystal-class.html#__delitem__,Method Bio.Index._InMemoryIndex.__delitem__()=Bio.Index._InMemoryIndex-class.html#__delitem__,Method Bio.PDB.Chain.Chain.__delitem__()=Bio.PDB.Chain.Chain-class.html#__delitem__,Method Bio.PDB.Entity.Entity.__delitem__()=Bio.PDB.Entity.Entity-class.html#__delitem__,Method Bio.Phylo.PhyloXML.Events.__delitem__()=Bio.Phylo.PhyloXML.Events-class.html#__delitem__,Method Bio.Phylo.TreeConstruction._Matrix.__delitem__()=Bio.Phylo.TreeConstruction._Matrix-class.html#__delitem__,Method Bio.SearchIO._model.hit.Hit.__delitem__()=Bio.SearchIO._model.hit.Hit-class.html#__delitem__,Method Bio.SearchIO._model.hsp.HSP.__delitem__()=Bio.SearchIO._model.hsp.HSP-class.html#__delitem__,Method Bio.SearchIO._model.query.QueryResult.__delitem__()=Bio.SearchIO._model.query.QueryResult-class.html#__delitem__,Method Bio.Seq.MutableSeq.__delitem__()=Bio.Seq.MutableSeq-class.html#__delitem__,Method Bio.phenotype.phen_micro.PlateRecord.__delitem__()=Bio.phenotype.phen_micro.PlateRecord-class.html#__delitem__,Method BioSQL.BioSeqDatabase.BioSeqDatabase.__delitem__()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#__delitem__,Method BioSQL.BioSeqDatabase.DBServer.__delitem__()=BioSQL.BioSeqDatabase.DBServer-class.html#__delitem__"><a title="Bio.Crystal.Chain.__delitem__
Bio.Crystal.Crystal.__delitem__
Bio.Index._InMemoryIndex.__delitem__
Bio.PDB.Chain.Chain.__delitem__
Bio.PDB.Entity.Entity.__delitem__
Bio.Phylo.PhyloXML.Events.__delitem__
Bio.Phylo.TreeConstruction._Matrix.__delitem__
Bio.SearchIO._model.hit.Hit.__delitem__
Bio.SearchIO._model.hsp.HSP.__delitem__
Bio.SearchIO._model.query.QueryResult.__delitem__
Bio.Seq.MutableSeq.__delitem__
Bio.phenotype.phen_micro.PlateRecord.__delitem__
BioSQL.BioSeqDatabase.BioSeqDatabase.__delitem__
BioSQL.BioSeqDatabase.DBServer.__delitem__" class="py-name" href="#" onclick="return doclink('link-49', '__delitem__', 'link-49');">__delitem__</a></tt><tt class="py-op">(</tt><tt class="py-name">db_name</tt><tt class="py-op">)</tt> </tt>
</div><a name="L264"></a><tt class="py-lineno">264</tt>  <tt class="py-line"> </tt>
<a name="DBServer.new_database"></a><div id="DBServer.new_database-def"><a name="L265"></a><tt class="py-lineno">265</tt> <a class="py-toggle" href="#" id="DBServer.new_database-toggle" onclick="return toggle('DBServer.new_database');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.DBServer-class.html#new_database">new_database</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">db_name</tt><tt class="py-op">,</tt> <tt class="py-param">authority</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">description</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBServer.new_database-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBServer.new_database-expanded"><a name="L266"></a><tt class="py-lineno">266</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add a new database to the server and return it."""</tt> </tt>
<a name="L267"></a><tt class="py-lineno">267</tt>  <tt class="py-line">        <tt class="py-comment"># make the database</tt> </tt>
<a name="L268"></a><tt class="py-lineno">268</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">r"INSERT INTO biodatabase (name, authority, description)"</tt> \ </tt>
<a name="L269"></a><tt class="py-lineno">269</tt>  <tt class="py-line">              <tt class="py-string">r" VALUES (%s, %s, %s)"</tt> </tt>
<a name="L270"></a><tt class="py-lineno">270</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-50" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-50', 'execute', 'link-28');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">db_name</tt><tt class="py-op">,</tt> <tt class="py-name">authority</tt><tt class="py-op">,</tt> <tt id="link-51" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DescriptionConsumer.description()=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#description,Method Bio.Phylo.PhyloXMLIO.Writer.description()=Bio.Phylo.PhyloXMLIO.Writer-class.html#description,Variable Bio.SearchIO._model.hit.Hit.description=Bio.SearchIO._model.hit.Hit-class.html#description,Variable Bio.SearchIO._model.query.QueryResult.description=Bio.SearchIO._model.query.QueryResult-class.html#description"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-51', 'description', 'link-51');">description</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L271"></a><tt class="py-lineno">271</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-52" class="py-name"><a title="BioSQL.BioSeqDatabase
BioSQL.BioSeqDatabase.BioSeqDatabase" class="py-name" href="#" onclick="return doclink('link-52', 'BioSeqDatabase', 'link-34');">BioSeqDatabase</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">,</tt> <tt class="py-name">db_name</tt><tt class="py-op">)</tt> </tt>
</div><a name="L272"></a><tt class="py-lineno">272</tt>  <tt class="py-line"> </tt>
<a name="DBServer.load_database_sql"></a><div id="DBServer.load_database_sql-def"><a name="L273"></a><tt class="py-lineno">273</tt> <a class="py-toggle" href="#" id="DBServer.load_database_sql-toggle" onclick="return toggle('DBServer.load_database_sql');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.DBServer-class.html#load_database_sql">load_database_sql</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sql_file</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBServer.load_database_sql-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBServer.load_database_sql-expanded"><a name="L274"></a><tt class="py-lineno">274</tt>  <tt class="py-line">        <tt class="py-docstring">"""Load a database schema into the given database.</tt> </tt>
<a name="L275"></a><tt class="py-lineno">275</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L276"></a><tt class="py-lineno">276</tt>  <tt class="py-line"><tt class="py-docstring">        This is used to create tables, etc when a database is first created.</tt> </tt>
<a name="L277"></a><tt class="py-lineno">277</tt>  <tt class="py-line"><tt class="py-docstring">        sql_file should specify the complete path to a file containing</tt> </tt>
<a name="L278"></a><tt class="py-lineno">278</tt>  <tt class="py-line"><tt class="py-docstring">        SQL entries for building the tables.</tt> </tt>
<a name="L279"></a><tt class="py-lineno">279</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L280"></a><tt class="py-lineno">280</tt>  <tt class="py-line">        <tt class="py-comment"># Not sophisticated enough for PG schema. Is it needed by MySQL?</tt> </tt>
<a name="L281"></a><tt class="py-lineno">281</tt>  <tt class="py-line">        <tt class="py-comment"># Looks like we need this more complicated way for both. Leaving it</tt> </tt>
<a name="L282"></a><tt class="py-lineno">282</tt>  <tt class="py-line">        <tt class="py-comment"># the default and removing the simple-minded approach.</tt> </tt>
<a name="L283"></a><tt class="py-lineno">283</tt>  <tt class="py-line"> </tt>
<a name="L284"></a><tt class="py-lineno">284</tt>  <tt class="py-line">        <tt class="py-comment"># read the file with all comment lines removed</tt> </tt>
<a name="L285"></a><tt class="py-lineno">285</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt> </tt>
<a name="L286"></a><tt class="py-lineno">286</tt>  <tt class="py-line">        <tt class="py-keyword">with</tt> <tt id="link-53" class="py-name" targets="Function Bio.bgzf.open()=Bio.bgzf-module.html#open"><a title="Bio.bgzf.open" class="py-name" href="#" onclick="return doclink('link-53', 'open', 'link-53');">open</a></tt><tt class="py-op">(</tt><tt class="py-name">sql_file</tt><tt class="py-op">,</tt> <tt id="link-54" class="py-name"><a title="Bio._py3k._universal_read_mode" class="py-name" href="#" onclick="return doclink('link-54', '_universal_read_mode', 'link-2');">_universal_read_mode</a></tt><tt class="py-op">)</tt> <tt class="py-keyword">as</tt> <tt class="py-name">sql_handle</tt><tt class="py-op">:</tt> </tt>
<a name="L287"></a><tt class="py-lineno">287</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">sql_handle</tt><tt class="py-op">:</tt> </tt>
<a name="L288"></a><tt class="py-lineno">288</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-55" class="py-name" targets="Method Bio.Seq.Seq.startswith()=Bio.Seq.Seq-class.html#startswith"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-55', 'startswith', 'link-55');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"--"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># don't include comment lines</tt> </tt>
<a name="L289"></a><tt class="py-lineno">289</tt>  <tt class="py-line">                    <tt class="py-keyword">pass</tt> </tt>
<a name="L290"></a><tt class="py-lineno">290</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-56" class="py-name"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-56', 'startswith', 'link-55');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"#"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># ditto for MySQL comments</tt> </tt>
<a name="L291"></a><tt class="py-lineno">291</tt>  <tt class="py-line">                    <tt class="py-keyword">pass</tt> </tt>
<a name="L292"></a><tt class="py-lineno">292</tt>  <tt class="py-line">                <tt class="py-keyword">elif</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-57" class="py-name" targets="Method Bio.Seq.Seq.strip()=Bio.Seq.Seq-class.html#strip"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-57', 'strip', 'link-57');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt>  <tt class="py-comment"># only include non-blank lines</tt> </tt>
<a name="L293"></a><tt class="py-lineno">293</tt>  <tt class="py-line">                    <tt class="py-name">sql</tt> <tt class="py-op">+=</tt> <tt class="py-name">line</tt><tt class="py-op">.</tt><tt id="link-58" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-58', 'strip', 'link-57');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-string">" "</tt> </tt>
<a name="L294"></a><tt class="py-lineno">294</tt>  <tt class="py-line"> </tt>
<a name="L295"></a><tt class="py-lineno">295</tt>  <tt class="py-line">        <tt class="py-comment"># two ways to load the SQL</tt> </tt>
<a name="L296"></a><tt class="py-lineno">296</tt>  <tt class="py-line">        <tt class="py-comment"># 1. PostgreSQL can load it all at once and actually needs to</tt> </tt>
<a name="L297"></a><tt class="py-lineno">297</tt>  <tt class="py-line">        <tt class="py-comment"># due to FUNCTION defines at the end of the SQL which mess up</tt> </tt>
<a name="L298"></a><tt class="py-lineno">298</tt>  <tt class="py-line">        <tt class="py-comment"># the splitting by semicolons</tt> </tt>
<a name="L299"></a><tt class="py-lineno">299</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">module_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">"psycopg2"</tt><tt class="py-op">,</tt> <tt class="py-string">"pgdb"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L300"></a><tt class="py-lineno">300</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-59" class="py-name" targets="Method Bio.DocSQL.Query.cursor()=Bio.DocSQL.Query-class.html#cursor,Method Bio.DocSQL.QuerySingle.cursor()=Bio.DocSQL.QuerySingle-class.html#cursor"><a title="Bio.DocSQL.Query.cursor
Bio.DocSQL.QuerySingle.cursor" class="py-name" href="#" onclick="return doclink('link-59', 'cursor', 'link-59');">cursor</a></tt><tt class="py-op">.</tt><tt id="link-60" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-60', 'execute', 'link-28');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">)</tt> </tt>
<a name="L301"></a><tt class="py-lineno">301</tt>  <tt class="py-line">        <tt class="py-comment"># 2. MySQL needs the database loading split up into single lines of</tt> </tt>
<a name="L302"></a><tt class="py-lineno">302</tt>  <tt class="py-line">        <tt class="py-comment"># SQL executed one at a time</tt> </tt>
<a name="L303"></a><tt class="py-lineno">303</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">module_name</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">"mysql.connector"</tt><tt class="py-op">,</tt> <tt class="py-string">"MySQLdb"</tt><tt class="py-op">,</tt> <tt class="py-string">"sqlite3"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L304"></a><tt class="py-lineno">304</tt>  <tt class="py-line">            <tt class="py-name">sql_parts</tt> <tt class="py-op">=</tt> <tt class="py-name">sql</tt><tt class="py-op">.</tt><tt id="link-61" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-61', 'split', 'link-61');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">";"</tt><tt class="py-op">)</tt>  <tt class="py-comment"># one line per sql command</tt> </tt>
<a name="L305"></a><tt class="py-lineno">305</tt>  <tt class="py-line">            <tt class="py-comment"># don't use the last item, it's blank</tt> </tt>
<a name="L306"></a><tt class="py-lineno">306</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">sql_line</tt> <tt class="py-keyword">in</tt> <tt class="py-name">sql_parts</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L307"></a><tt class="py-lineno">307</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-62" class="py-name"><a title="Bio.DocSQL.Query.cursor
Bio.DocSQL.QuerySingle.cursor" class="py-name" href="#" onclick="return doclink('link-62', 'cursor', 'link-59');">cursor</a></tt><tt class="py-op">.</tt><tt id="link-63" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-63', 'execute', 'link-28');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">sql_line</tt><tt class="py-op">)</tt> </tt>
<a name="L308"></a><tt class="py-lineno">308</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L309"></a><tt class="py-lineno">309</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Module %s not supported by the loader."</tt> <tt class="py-op">%</tt> </tt>
<a name="L310"></a><tt class="py-lineno">310</tt>  <tt class="py-line">                             <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">module_name</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L311"></a><tt class="py-lineno">311</tt>  <tt class="py-line"> </tt>
<a name="DBServer.commit"></a><div id="DBServer.commit-def"><a name="L312"></a><tt class="py-lineno">312</tt> <a class="py-toggle" href="#" id="DBServer.commit-toggle" onclick="return toggle('DBServer.commit');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.DBServer-class.html#commit">commit</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBServer.commit-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBServer.commit-expanded"><a name="L313"></a><tt class="py-lineno">313</tt>  <tt class="py-line">        <tt class="py-docstring">"""Commit the current transaction to the database."""</tt> </tt>
<a name="L314"></a><tt class="py-lineno">314</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-64" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.commit()=BioSQL.BioSeqDatabase.Adaptor-class.html#commit,Method BioSQL.BioSeqDatabase.DBServer.commit()=BioSQL.BioSeqDatabase.DBServer-class.html#commit"><a title="BioSQL.BioSeqDatabase.Adaptor.commit
BioSQL.BioSeqDatabase.DBServer.commit" class="py-name" href="#" onclick="return doclink('link-64', 'commit', 'link-64');">commit</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L315"></a><tt class="py-lineno">315</tt>  <tt class="py-line"> </tt>
<a name="DBServer.rollback"></a><div id="DBServer.rollback-def"><a name="L316"></a><tt class="py-lineno">316</tt> <a class="py-toggle" href="#" id="DBServer.rollback-toggle" onclick="return toggle('DBServer.rollback');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.DBServer-class.html#rollback">rollback</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBServer.rollback-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBServer.rollback-expanded"><a name="L317"></a><tt class="py-lineno">317</tt>  <tt class="py-line">        <tt class="py-docstring">"""Roll-back the current transaction."""</tt> </tt>
<a name="L318"></a><tt class="py-lineno">318</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-65" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.rollback()=BioSQL.BioSeqDatabase.Adaptor-class.html#rollback,Method BioSQL.BioSeqDatabase.DBServer.rollback()=BioSQL.BioSeqDatabase.DBServer-class.html#rollback"><a title="BioSQL.BioSeqDatabase.Adaptor.rollback
BioSQL.BioSeqDatabase.DBServer.rollback" class="py-name" href="#" onclick="return doclink('link-65', 'rollback', 'link-65');">rollback</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L319"></a><tt class="py-lineno">319</tt>  <tt class="py-line"> </tt>
<a name="DBServer.close"></a><div id="DBServer.close-def"><a name="L320"></a><tt class="py-lineno">320</tt> <a class="py-toggle" href="#" id="DBServer.close-toggle" onclick="return toggle('DBServer.close');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.DBServer-class.html#close">close</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DBServer.close-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="DBServer.close-expanded"><a name="L321"></a><tt class="py-lineno">321</tt>  <tt class="py-line">        <tt class="py-docstring">"""Close the connection. No further activity possible."""</tt> </tt>
<a name="L322"></a><tt class="py-lineno">322</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-66" class="py-name" targets="Method Bio.File._IndexedSeqFileDict.close()=Bio.File._IndexedSeqFileDict-class.html#close,Method Bio.File._SQLiteManySeqFilesDict.close()=Bio.File._SQLiteManySeqFilesDict-class.html#close,Method Bio.SeqIO.SffIO._AddTellHandle.close()=Bio.SeqIO.SffIO._AddTellHandle-class.html#close,Method Bio.bgzf.BgzfReader.close()=Bio.bgzf.BgzfReader-class.html#close,Method Bio.bgzf.BgzfWriter.close()=Bio.bgzf.BgzfWriter-class.html#close,Method BioSQL.BioSeqDatabase.Adaptor.close()=BioSQL.BioSeqDatabase.Adaptor-class.html#close,Method BioSQL.BioSeqDatabase.DBServer.close()=BioSQL.BioSeqDatabase.DBServer-class.html#close"><a title="Bio.File._IndexedSeqFileDict.close
Bio.File._SQLiteManySeqFilesDict.close
Bio.SeqIO.SffIO._AddTellHandle.close
Bio.bgzf.BgzfReader.close
Bio.bgzf.BgzfWriter.close
BioSQL.BioSeqDatabase.Adaptor.close
BioSQL.BioSeqDatabase.DBServer.close" class="py-name" href="#" onclick="return doclink('link-66', 'close', 'link-66');">close</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L323"></a><tt class="py-lineno">323</tt>  <tt class="py-line"> </tt>
<a name="L324"></a><tt class="py-lineno">324</tt>  <tt class="py-line"> </tt>
<a name="_CursorWrapper"></a><div id="_CursorWrapper-def"><a name="L325"></a><tt class="py-lineno">325</tt> <a class="py-toggle" href="#" id="_CursorWrapper-toggle" onclick="return toggle('_CursorWrapper');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase._CursorWrapper-class.html">_CursorWrapper</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_CursorWrapper-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="_CursorWrapper-expanded"><a name="L326"></a><tt class="py-lineno">326</tt>  <tt class="py-line">    <tt class="py-docstring">"""A wraper for mysql.connector resolving bytestring representations."""</tt> </tt>
<a name="L327"></a><tt class="py-lineno">327</tt>  <tt class="py-line"> </tt>
<a name="_CursorWrapper.__init__"></a><div id="_CursorWrapper.__init__-def"><a name="L328"></a><tt class="py-lineno">328</tt> <a class="py-toggle" href="#" id="_CursorWrapper.__init__-toggle" onclick="return toggle('_CursorWrapper.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase._CursorWrapper-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">real_cursor</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_CursorWrapper.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="_CursorWrapper.__init__-expanded"><a name="L329"></a><tt class="py-lineno">329</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">real_cursor</tt> <tt class="py-op">=</tt> <tt class="py-name">real_cursor</tt> </tt>
</div><a name="L330"></a><tt class="py-lineno">330</tt>  <tt class="py-line"> </tt>
<a name="_CursorWrapper.execute"></a><div id="_CursorWrapper.execute-def"><a name="L331"></a><tt class="py-lineno">331</tt> <a class="py-toggle" href="#" id="_CursorWrapper.execute-toggle" onclick="return toggle('_CursorWrapper.execute');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase._CursorWrapper-class.html#execute">execute</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">operation</tt><tt class="py-op">,</tt> <tt class="py-param">params</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-param">multi</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_CursorWrapper.execute-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="_CursorWrapper.execute-expanded"><a name="L332"></a><tt class="py-lineno">332</tt>  <tt class="py-line">        <tt class="py-docstring">"""Execute a sql statement."""</tt> </tt>
<a name="L333"></a><tt class="py-lineno">333</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">real_cursor</tt><tt class="py-op">.</tt><tt id="link-67" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-67', 'execute', 'link-28');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">operation</tt><tt class="py-op">,</tt> <tt class="py-name">params</tt><tt class="py-op">,</tt> <tt class="py-name">multi</tt><tt class="py-op">)</tt> </tt>
</div><a name="L334"></a><tt class="py-lineno">334</tt>  <tt class="py-line"> </tt>
<a name="_CursorWrapper.executemany"></a><div id="_CursorWrapper.executemany-def"><a name="L335"></a><tt class="py-lineno">335</tt> <a class="py-toggle" href="#" id="_CursorWrapper.executemany-toggle" onclick="return toggle('_CursorWrapper.executemany');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase._CursorWrapper-class.html#executemany">executemany</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">operation</tt><tt class="py-op">,</tt> <tt class="py-param">params</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_CursorWrapper.executemany-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="_CursorWrapper.executemany-expanded"><a name="L336"></a><tt class="py-lineno">336</tt>  <tt class="py-line">        <tt class="py-docstring">"""Execute many sql statements."""</tt> </tt>
<a name="L337"></a><tt class="py-lineno">337</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">real_cursor</tt><tt class="py-op">.</tt><tt id="link-68" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.executemany()=BioSQL.BioSeqDatabase.Adaptor-class.html#executemany,Method BioSQL.BioSeqDatabase._CursorWrapper.executemany()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#executemany,Method BioSQL.DBUtils.Generic_dbutils.executemany()=BioSQL.DBUtils.Generic_dbutils-class.html#executemany,Method BioSQL.DBUtils.Sqlite_dbutils.executemany()=BioSQL.DBUtils.Sqlite_dbutils-class.html#executemany"><a title="BioSQL.BioSeqDatabase.Adaptor.executemany
BioSQL.BioSeqDatabase._CursorWrapper.executemany
BioSQL.DBUtils.Generic_dbutils.executemany
BioSQL.DBUtils.Sqlite_dbutils.executemany" class="py-name" href="#" onclick="return doclink('link-68', 'executemany', 'link-68');">executemany</a></tt><tt class="py-op">(</tt><tt class="py-name">operation</tt><tt class="py-op">,</tt> <tt class="py-name">params</tt><tt class="py-op">)</tt> </tt>
</div><a name="L338"></a><tt class="py-lineno">338</tt>  <tt class="py-line"> </tt>
<a name="_CursorWrapper._convert_tuple"></a><div id="_CursorWrapper._convert_tuple-def"><a name="L339"></a><tt class="py-lineno">339</tt> <a class="py-toggle" href="#" id="_CursorWrapper._convert_tuple-toggle" onclick="return toggle('_CursorWrapper._convert_tuple');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase._CursorWrapper-class.html#_convert_tuple">_convert_tuple</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">tuple_</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_CursorWrapper._convert_tuple-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="_CursorWrapper._convert_tuple-expanded"><a name="L340"></a><tt class="py-lineno">340</tt>  <tt class="py-line">        <tt class="py-docstring">"""Decode any bytestrings present in the row (PRIVATE)."""</tt> </tt>
<a name="L341"></a><tt class="py-lineno">341</tt>  <tt class="py-line">        <tt class="py-name">tuple_list</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">tuple_</tt><tt class="py-op">)</tt> </tt>
<a name="L342"></a><tt class="py-lineno">342</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-69" class="py-name" targets="Variable Bio.PDB.Polypeptide.i=Bio.PDB.Polypeptide-module.html#i,Variable Bio.SeqIO.SffIO.i=Bio.SeqIO.SffIO-module.html#i"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-69', 'i', 'link-69');">i</a></tt><tt class="py-op">,</tt> <tt class="py-name">elem</tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt class="py-name">tuple_list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L343"></a><tt class="py-lineno">343</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-70" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.type()=Bio.Phylo.PhyloXMLIO.Writer-class.html#type"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-70', 'type', 'link-70');">type</a></tt><tt class="py-op">(</tt><tt class="py-name">elem</tt><tt class="py-op">)</tt> <tt class="py-keyword">is</tt> <tt class="py-name">bytes</tt><tt class="py-op">:</tt> </tt>
<a name="L344"></a><tt class="py-lineno">344</tt>  <tt class="py-line">                <tt class="py-name">tuple_list</tt><tt class="py-op">[</tt><tt id="link-71" class="py-name"><a title="Bio.PDB.Polypeptide.i
Bio.SeqIO.SffIO.i" class="py-name" href="#" onclick="return doclink('link-71', 'i', 'link-69');">i</a></tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">elem</tt><tt class="py-op">.</tt><tt class="py-name">decode</tt><tt class="py-op">(</tt><tt class="py-string">"utf-8"</tt><tt class="py-op">)</tt> </tt>
<a name="L345"></a><tt class="py-lineno">345</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">tuple</tt><tt class="py-op">(</tt><tt class="py-name">tuple_list</tt><tt class="py-op">)</tt> </tt>
</div><a name="L346"></a><tt class="py-lineno">346</tt>  <tt class="py-line"> </tt>
<a name="_CursorWrapper._convert_list"></a><div id="_CursorWrapper._convert_list-def"><a name="L347"></a><tt class="py-lineno">347</tt> <a class="py-toggle" href="#" id="_CursorWrapper._convert_list-toggle" onclick="return toggle('_CursorWrapper._convert_list');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase._CursorWrapper-class.html#_convert_list">_convert_list</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">lst</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_CursorWrapper._convert_list-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="_CursorWrapper._convert_list-expanded"><a name="L348"></a><tt class="py-lineno">348</tt>  <tt class="py-line">        <tt class="py-name">ret_lst</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L349"></a><tt class="py-lineno">349</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">tuple_</tt> <tt class="py-keyword">in</tt> <tt class="py-name">lst</tt><tt class="py-op">:</tt> </tt>
<a name="L350"></a><tt class="py-lineno">350</tt>  <tt class="py-line">            <tt class="py-name">new_tuple</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-72" class="py-name" targets="Method BioSQL.BioSeqDatabase._CursorWrapper._convert_tuple()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#_convert_tuple"><a title="BioSQL.BioSeqDatabase._CursorWrapper._convert_tuple" class="py-name" href="#" onclick="return doclink('link-72', '_convert_tuple', 'link-72');">_convert_tuple</a></tt><tt class="py-op">(</tt><tt class="py-name">tuple_</tt><tt class="py-op">)</tt> </tt>
<a name="L351"></a><tt class="py-lineno">351</tt>  <tt class="py-line">            <tt class="py-name">ret_lst</tt><tt class="py-op">.</tt><tt id="link-73" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-73', 'append', 'link-73');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">new_tuple</tt><tt class="py-op">)</tt> </tt>
<a name="L352"></a><tt class="py-lineno">352</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">ret_lst</tt> </tt>
</div><a name="L353"></a><tt class="py-lineno">353</tt>  <tt class="py-line"> </tt>
<a name="_CursorWrapper.fetchall"></a><div id="_CursorWrapper.fetchall-def"><a name="L354"></a><tt class="py-lineno">354</tt> <a class="py-toggle" href="#" id="_CursorWrapper.fetchall-toggle" onclick="return toggle('_CursorWrapper.fetchall');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase._CursorWrapper-class.html#fetchall">fetchall</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_CursorWrapper.fetchall-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="_CursorWrapper.fetchall-expanded"><a name="L355"></a><tt class="py-lineno">355</tt>  <tt class="py-line">        <tt class="py-name">rv</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">real_cursor</tt><tt class="py-op">.</tt><tt id="link-74" class="py-name" targets="Method BioSQL.BioSeqDatabase._CursorWrapper.fetchall()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#fetchall"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchall" class="py-name" href="#" onclick="return doclink('link-74', 'fetchall', 'link-74');">fetchall</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L356"></a><tt class="py-lineno">356</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-75" class="py-name" targets="Method BioSQL.BioSeqDatabase._CursorWrapper._convert_list()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#_convert_list"><a title="BioSQL.BioSeqDatabase._CursorWrapper._convert_list" class="py-name" href="#" onclick="return doclink('link-75', '_convert_list', 'link-75');">_convert_list</a></tt><tt class="py-op">(</tt><tt class="py-name">rv</tt><tt class="py-op">)</tt> </tt>
</div><a name="L357"></a><tt class="py-lineno">357</tt>  <tt class="py-line"> </tt>
<a name="_CursorWrapper.fetchone"></a><div id="_CursorWrapper.fetchone-def"><a name="L358"></a><tt class="py-lineno">358</tt> <a class="py-toggle" href="#" id="_CursorWrapper.fetchone-toggle" onclick="return toggle('_CursorWrapper.fetchone');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase._CursorWrapper-class.html#fetchone">fetchone</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="_CursorWrapper.fetchone-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="_CursorWrapper.fetchone-expanded"><a name="L359"></a><tt class="py-lineno">359</tt>  <tt class="py-line">        <tt class="py-name">tuple_</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">real_cursor</tt><tt class="py-op">.</tt><tt id="link-76" class="py-name" targets="Method BioSQL.BioSeqDatabase._CursorWrapper.fetchone()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#fetchone"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchone" class="py-name" href="#" onclick="return doclink('link-76', 'fetchone', 'link-76');">fetchone</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L360"></a><tt class="py-lineno">360</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-77" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper._convert_tuple" class="py-name" href="#" onclick="return doclink('link-77', '_convert_tuple', 'link-72');">_convert_tuple</a></tt><tt class="py-op">(</tt><tt class="py-name">tuple_</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L361"></a><tt class="py-lineno">361</tt>  <tt class="py-line"> </tt>
<a name="L362"></a><tt class="py-lineno">362</tt>  <tt class="py-line"> </tt>
<a name="Adaptor"></a><div id="Adaptor-def"><a name="L363"></a><tt class="py-lineno">363</tt> <a class="py-toggle" href="#" id="Adaptor-toggle" onclick="return toggle('Adaptor');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.Adaptor-class.html">Adaptor</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Adaptor-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="Adaptor-expanded"><a name="L364"></a><tt class="py-lineno">364</tt>  <tt class="py-line">    <tt class="py-docstring">"""High level wrapper for a database connection and cursor.</tt> </tt>
<a name="L365"></a><tt class="py-lineno">365</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L366"></a><tt class="py-lineno">366</tt>  <tt class="py-line"><tt class="py-docstring">    Most database calls in BioSQL are done indirectly though this adaptor</tt> </tt>
<a name="L367"></a><tt class="py-lineno">367</tt>  <tt class="py-line"><tt class="py-docstring">    class. This provides helper methods for fetching data and executing</tt> </tt>
<a name="L368"></a><tt class="py-lineno">368</tt>  <tt class="py-line"><tt class="py-docstring">    sql.</tt> </tt>
<a name="L369"></a><tt class="py-lineno">369</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L370"></a><tt class="py-lineno">370</tt>  <tt class="py-line"> </tt>
<a name="Adaptor.__init__"></a><div id="Adaptor.__init__-def"><a name="L371"></a><tt class="py-lineno">371</tt> <a class="py-toggle" href="#" id="Adaptor.__init__-toggle" onclick="return toggle('Adaptor.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.Adaptor-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">conn</tt><tt class="py-op">,</tt> <tt class="py-param">dbutils</tt><tt class="py-op">,</tt> <tt class="py-param">wrap_cursor</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Adaptor.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Adaptor.__init__-expanded"><a name="L372"></a><tt class="py-lineno">372</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create an Adaptor object.</tt> </tt>
<a name="L373"></a><tt class="py-lineno">373</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L374"></a><tt class="py-lineno">374</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L375"></a><tt class="py-lineno">375</tt>  <tt class="py-line"><tt class="py-docstring">         - conn - A database connection</tt> </tt>
<a name="L376"></a><tt class="py-lineno">376</tt>  <tt class="py-line"><tt class="py-docstring">         - dbutils - A BioSQL.DBUtils object</tt> </tt>
<a name="L377"></a><tt class="py-lineno">377</tt>  <tt class="py-line"><tt class="py-docstring">         - wrap_cursor - Optional, whether to wrap the cursor object</tt> </tt>
<a name="L378"></a><tt class="py-lineno">378</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L379"></a><tt class="py-lineno">379</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L380"></a><tt class="py-lineno">380</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">conn</tt> <tt class="py-op">=</tt> <tt class="py-name">conn</tt> </tt>
<a name="L381"></a><tt class="py-lineno">381</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">wrap_cursor</tt><tt class="py-op">:</tt> </tt>
<a name="L382"></a><tt class="py-lineno">382</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-78" class="py-name"><a title="Bio.DocSQL.Query.cursor
Bio.DocSQL.QuerySingle.cursor" class="py-name" href="#" onclick="return doclink('link-78', 'cursor', 'link-59');">cursor</a></tt> <tt class="py-op">=</tt> <tt id="link-79" class="py-name" targets="Class BioSQL.BioSeqDatabase._CursorWrapper=BioSQL.BioSeqDatabase._CursorWrapper-class.html"><a title="BioSQL.BioSeqDatabase._CursorWrapper" class="py-name" href="#" onclick="return doclink('link-79', '_CursorWrapper', 'link-79');">_CursorWrapper</a></tt><tt class="py-op">(</tt><tt class="py-name">conn</tt><tt class="py-op">.</tt><tt id="link-80" class="py-name"><a title="Bio.DocSQL.Query.cursor
Bio.DocSQL.QuerySingle.cursor" class="py-name" href="#" onclick="return doclink('link-80', 'cursor', 'link-59');">cursor</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L383"></a><tt class="py-lineno">383</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L384"></a><tt class="py-lineno">384</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-81" class="py-name"><a title="Bio.DocSQL.Query.cursor
Bio.DocSQL.QuerySingle.cursor" class="py-name" href="#" onclick="return doclink('link-81', 'cursor', 'link-59');">cursor</a></tt> <tt class="py-op">=</tt> <tt class="py-name">conn</tt><tt class="py-op">.</tt><tt id="link-82" class="py-name"><a title="Bio.DocSQL.Query.cursor
Bio.DocSQL.QuerySingle.cursor" class="py-name" href="#" onclick="return doclink('link-82', 'cursor', 'link-59');">cursor</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L385"></a><tt class="py-lineno">385</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbutils</tt> <tt class="py-op">=</tt> <tt class="py-name">dbutils</tt> </tt>
</div><a name="L386"></a><tt class="py-lineno">386</tt>  <tt class="py-line"> </tt>
<a name="Adaptor.last_id"></a><div id="Adaptor.last_id-def"><a name="L387"></a><tt class="py-lineno">387</tt> <a class="py-toggle" href="#" id="Adaptor.last_id-toggle" onclick="return toggle('Adaptor.last_id');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.Adaptor-class.html#last_id">last_id</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">table</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Adaptor.last_id-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Adaptor.last_id-expanded"><a name="L388"></a><tt class="py-lineno">388</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the last row id for the selected table."""</tt> </tt>
<a name="L389"></a><tt class="py-lineno">389</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbutils</tt><tt class="py-op">.</tt><tt id="link-83" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.last_id()=BioSQL.BioSeqDatabase.Adaptor-class.html#last_id,Method BioSQL.DBUtils.Generic_dbutils.last_id()=BioSQL.DBUtils.Generic_dbutils-class.html#last_id,Method BioSQL.DBUtils.Mysql_dbutils.last_id()=BioSQL.DBUtils.Mysql_dbutils-class.html#last_id,Method BioSQL.DBUtils._PostgreSQL_dbutils.last_id()=BioSQL.DBUtils._PostgreSQL_dbutils-class.html#last_id"><a title="BioSQL.BioSeqDatabase.Adaptor.last_id
BioSQL.DBUtils.Generic_dbutils.last_id
BioSQL.DBUtils.Mysql_dbutils.last_id
BioSQL.DBUtils._PostgreSQL_dbutils.last_id" class="py-name" href="#" onclick="return doclink('link-83', 'last_id', 'link-83');">last_id</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-84" class="py-name"><a title="Bio.DocSQL.Query.cursor
Bio.DocSQL.QuerySingle.cursor" class="py-name" href="#" onclick="return doclink('link-84', 'cursor', 'link-59');">cursor</a></tt><tt class="py-op">,</tt> <tt class="py-name">table</tt><tt class="py-op">)</tt> </tt>
</div><a name="L390"></a><tt class="py-lineno">390</tt>  <tt class="py-line"> </tt>
<a name="Adaptor.autocommit"></a><div id="Adaptor.autocommit-def"><a name="L391"></a><tt class="py-lineno">391</tt> <a class="py-toggle" href="#" id="Adaptor.autocommit-toggle" onclick="return toggle('Adaptor.autocommit');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.Adaptor-class.html#autocommit">autocommit</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">y</tt><tt class="py-op">=</tt><tt class="py-name">True</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Adaptor.autocommit-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Adaptor.autocommit-expanded"><a name="L392"></a><tt class="py-lineno">392</tt>  <tt class="py-line">        <tt class="py-docstring">"""Set the autocommit mode. True values enable; False value disable."""</tt> </tt>
<a name="L393"></a><tt class="py-lineno">393</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbutils</tt><tt class="py-op">.</tt><tt id="link-85" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.autocommit()=BioSQL.BioSeqDatabase.Adaptor-class.html#autocommit,Method BioSQL.DBUtils.Generic_dbutils.autocommit()=BioSQL.DBUtils.Generic_dbutils-class.html#autocommit,Method BioSQL.DBUtils.Pgdb_dbutils.autocommit()=BioSQL.DBUtils.Pgdb_dbutils-class.html#autocommit,Method BioSQL.DBUtils.Psycopg2_dbutils.autocommit()=BioSQL.DBUtils.Psycopg2_dbutils-class.html#autocommit"><a title="BioSQL.BioSeqDatabase.Adaptor.autocommit
BioSQL.DBUtils.Generic_dbutils.autocommit
BioSQL.DBUtils.Pgdb_dbutils.autocommit
BioSQL.DBUtils.Psycopg2_dbutils.autocommit" class="py-name" href="#" onclick="return doclink('link-85', 'autocommit', 'link-85');">autocommit</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">conn</tt><tt class="py-op">,</tt> <tt id="link-86" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Graphics.y=Bio.KEGG.KGML.KGML_pathway.Graphics-class.html#y"><a title="Bio.KEGG.KGML.KGML_pathway.Graphics.y" class="py-name" href="#" onclick="return doclink('link-86', 'y', 'link-86');">y</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L394"></a><tt class="py-lineno">394</tt>  <tt class="py-line"> </tt>
<a name="Adaptor.commit"></a><div id="Adaptor.commit-def"><a name="L395"></a><tt class="py-lineno">395</tt> <a class="py-toggle" href="#" id="Adaptor.commit-toggle" onclick="return toggle('Adaptor.commit');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.Adaptor-class.html#commit">commit</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Adaptor.commit-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Adaptor.commit-expanded"><a name="L396"></a><tt class="py-lineno">396</tt>  <tt class="py-line">        <tt class="py-docstring">"""Commit the current transaction."""</tt> </tt>
<a name="L397"></a><tt class="py-lineno">397</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">conn</tt><tt class="py-op">.</tt><tt id="link-87" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.commit
BioSQL.BioSeqDatabase.DBServer.commit" class="py-name" href="#" onclick="return doclink('link-87', 'commit', 'link-64');">commit</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L398"></a><tt class="py-lineno">398</tt>  <tt class="py-line"> </tt>
<a name="Adaptor.rollback"></a><div id="Adaptor.rollback-def"><a name="L399"></a><tt class="py-lineno">399</tt> <a class="py-toggle" href="#" id="Adaptor.rollback-toggle" onclick="return toggle('Adaptor.rollback');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.Adaptor-class.html#rollback">rollback</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Adaptor.rollback-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Adaptor.rollback-expanded"><a name="L400"></a><tt class="py-lineno">400</tt>  <tt class="py-line">        <tt class="py-docstring">"""Roll-back the current transaction."""</tt> </tt>
<a name="L401"></a><tt class="py-lineno">401</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">conn</tt><tt class="py-op">.</tt><tt id="link-88" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.rollback
BioSQL.BioSeqDatabase.DBServer.rollback" class="py-name" href="#" onclick="return doclink('link-88', 'rollback', 'link-65');">rollback</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L402"></a><tt class="py-lineno">402</tt>  <tt class="py-line"> </tt>
<a name="Adaptor.close"></a><div id="Adaptor.close-def"><a name="L403"></a><tt class="py-lineno">403</tt> <a class="py-toggle" href="#" id="Adaptor.close-toggle" onclick="return toggle('Adaptor.close');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.Adaptor-class.html#close">close</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Adaptor.close-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Adaptor.close-expanded"><a name="L404"></a><tt class="py-lineno">404</tt>  <tt class="py-line">        <tt class="py-docstring">"""Close the connection. No further activity possible."""</tt> </tt>
<a name="L405"></a><tt class="py-lineno">405</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">conn</tt><tt class="py-op">.</tt><tt id="link-89" class="py-name"><a title="Bio.File._IndexedSeqFileDict.close
Bio.File._SQLiteManySeqFilesDict.close
Bio.SeqIO.SffIO._AddTellHandle.close
Bio.bgzf.BgzfReader.close
Bio.bgzf.BgzfWriter.close
BioSQL.BioSeqDatabase.Adaptor.close
BioSQL.BioSeqDatabase.DBServer.close" class="py-name" href="#" onclick="return doclink('link-89', 'close', 'link-66');">close</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div><a name="L406"></a><tt class="py-lineno">406</tt>  <tt class="py-line"> </tt>
<a name="Adaptor.fetch_dbid_by_dbname"></a><div id="Adaptor.fetch_dbid_by_dbname-def"><a name="L407"></a><tt class="py-lineno">407</tt> <a class="py-toggle" href="#" id="Adaptor.fetch_dbid_by_dbname-toggle" onclick="return toggle('Adaptor.fetch_dbid_by_dbname');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.Adaptor-class.html#fetch_dbid_by_dbname">fetch_dbid_by_dbname</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">dbname</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Adaptor.fetch_dbid_by_dbname-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Adaptor.fetch_dbid_by_dbname-expanded"><a name="L408"></a><tt class="py-lineno">408</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the internal id for the sub-database using its name."""</tt> </tt>
<a name="L409"></a><tt class="py-lineno">409</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-90" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-90', 'execute', 'link-28');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L410"></a><tt class="py-lineno">410</tt>  <tt class="py-line">            <tt class="py-string">r"select biodatabase_id from biodatabase where name = %s"</tt><tt class="py-op">,</tt> </tt>
<a name="L411"></a><tt class="py-lineno">411</tt>  <tt class="py-line">            <tt class="py-op">(</tt><tt class="py-name">dbname</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L412"></a><tt class="py-lineno">412</tt>  <tt class="py-line">        <tt class="py-name">rv</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-91" class="py-name"><a title="Bio.DocSQL.Query.cursor
Bio.DocSQL.QuerySingle.cursor" class="py-name" href="#" onclick="return doclink('link-91', 'cursor', 'link-59');">cursor</a></tt><tt class="py-op">.</tt><tt id="link-92" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchall" class="py-name" href="#" onclick="return doclink('link-92', 'fetchall', 'link-74');">fetchall</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L413"></a><tt class="py-lineno">413</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">rv</tt><tt class="py-op">:</tt> </tt>
<a name="L414"></a><tt class="py-lineno">414</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">KeyError</tt><tt class="py-op">(</tt><tt class="py-string">"Cannot find biodatabase with name %r"</tt> <tt class="py-op">%</tt> <tt class="py-name">dbname</tt><tt class="py-op">)</tt> </tt>
<a name="L415"></a><tt class="py-lineno">415</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">rv</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
</div><a name="L416"></a><tt class="py-lineno">416</tt>  <tt class="py-line"> </tt>
<a name="Adaptor.fetch_seqid_by_display_id"></a><div id="Adaptor.fetch_seqid_by_display_id-def"><a name="L417"></a><tt class="py-lineno">417</tt> <a class="py-toggle" href="#" id="Adaptor.fetch_seqid_by_display_id-toggle" onclick="return toggle('Adaptor.fetch_seqid_by_display_id');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.Adaptor-class.html#fetch_seqid_by_display_id">fetch_seqid_by_display_id</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">dbid</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Adaptor.fetch_seqid_by_display_id-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Adaptor.fetch_seqid_by_display_id-expanded"><a name="L418"></a><tt class="py-lineno">418</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the internal id for a sequence using its display id.</tt> </tt>
<a name="L419"></a><tt class="py-lineno">419</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L420"></a><tt class="py-lineno">420</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L421"></a><tt class="py-lineno">421</tt>  <tt class="py-line"><tt class="py-docstring">         - dbid - the internal id for the sub-database</tt> </tt>
<a name="L422"></a><tt class="py-lineno">422</tt>  <tt class="py-line"><tt class="py-docstring">         - name - the name of the sequence. Corresponds to the</tt> </tt>
<a name="L423"></a><tt class="py-lineno">423</tt>  <tt class="py-line"><tt class="py-docstring">           name column of the bioentry table of the SQL schema</tt> </tt>
<a name="L424"></a><tt class="py-lineno">424</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L425"></a><tt class="py-lineno">425</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L426"></a><tt class="py-lineno">426</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">r"select bioentry_id from bioentry where name = %s"</tt> </tt>
<a name="L427"></a><tt class="py-lineno">427</tt>  <tt class="py-line">        <tt class="py-name">fields</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-93" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-93', 'name', 'link-12');">name</a></tt><tt class="py-op">]</tt> </tt>
<a name="L428"></a><tt class="py-lineno">428</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">dbid</tt><tt class="py-op">:</tt> </tt>
<a name="L429"></a><tt class="py-lineno">429</tt>  <tt class="py-line">            <tt class="py-name">sql</tt> <tt class="py-op">+=</tt> <tt class="py-string">" and biodatabase_id = %s"</tt> </tt>
<a name="L430"></a><tt class="py-lineno">430</tt>  <tt class="py-line">            <tt class="py-name">fields</tt><tt class="py-op">.</tt><tt id="link-94" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-94', 'append', 'link-73');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">dbid</tt><tt class="py-op">)</tt> </tt>
<a name="L431"></a><tt class="py-lineno">431</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-95" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-95', 'execute', 'link-28');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-name">fields</tt><tt class="py-op">)</tt> </tt>
<a name="L432"></a><tt class="py-lineno">432</tt>  <tt class="py-line">        <tt class="py-name">rv</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-96" class="py-name"><a title="Bio.DocSQL.Query.cursor
Bio.DocSQL.QuerySingle.cursor" class="py-name" href="#" onclick="return doclink('link-96', 'cursor', 'link-59');">cursor</a></tt><tt class="py-op">.</tt><tt id="link-97" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchall" class="py-name" href="#" onclick="return doclink('link-97', 'fetchall', 'link-74');">fetchall</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L433"></a><tt class="py-lineno">433</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">rv</tt><tt class="py-op">:</tt> </tt>
<a name="L434"></a><tt class="py-lineno">434</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">IndexError</tt><tt class="py-op">(</tt><tt class="py-string">"Cannot find display id %r"</tt> <tt class="py-op">%</tt> <tt id="link-98" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-98', 'name', 'link-12');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L435"></a><tt class="py-lineno">435</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">rv</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L436"></a><tt class="py-lineno">436</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">IndexError</tt><tt class="py-op">(</tt><tt class="py-string">"More than one entry with display id %r"</tt> <tt class="py-op">%</tt> <tt id="link-99" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-99', 'name', 'link-12');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L437"></a><tt class="py-lineno">437</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">rv</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
</div><a name="L438"></a><tt class="py-lineno">438</tt>  <tt class="py-line"> </tt>
<a name="Adaptor.fetch_seqid_by_accession"></a><div id="Adaptor.fetch_seqid_by_accession-def"><a name="L439"></a><tt class="py-lineno">439</tt> <a class="py-toggle" href="#" id="Adaptor.fetch_seqid_by_accession-toggle" onclick="return toggle('Adaptor.fetch_seqid_by_accession');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.Adaptor-class.html#fetch_seqid_by_accession">fetch_seqid_by_accession</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">dbid</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Adaptor.fetch_seqid_by_accession-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Adaptor.fetch_seqid_by_accession-expanded"><a name="L440"></a><tt class="py-lineno">440</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the internal id for a sequence using its accession.</tt> </tt>
<a name="L441"></a><tt class="py-lineno">441</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L442"></a><tt class="py-lineno">442</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L443"></a><tt class="py-lineno">443</tt>  <tt class="py-line"><tt class="py-docstring">         - dbid - the internal id for the sub-database</tt> </tt>
<a name="L444"></a><tt class="py-lineno">444</tt>  <tt class="py-line"><tt class="py-docstring">         - name - the accession of the sequence. Corresponds to the</tt> </tt>
<a name="L445"></a><tt class="py-lineno">445</tt>  <tt class="py-line"><tt class="py-docstring">           accession column of the bioentry table of the SQL schema</tt> </tt>
<a name="L446"></a><tt class="py-lineno">446</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L447"></a><tt class="py-lineno">447</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L448"></a><tt class="py-lineno">448</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">r"select bioentry_id from bioentry where accession = %s"</tt> </tt>
<a name="L449"></a><tt class="py-lineno">449</tt>  <tt class="py-line">        <tt class="py-name">fields</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-100" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-100', 'name', 'link-12');">name</a></tt><tt class="py-op">]</tt> </tt>
<a name="L450"></a><tt class="py-lineno">450</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">dbid</tt><tt class="py-op">:</tt> </tt>
<a name="L451"></a><tt class="py-lineno">451</tt>  <tt class="py-line">            <tt class="py-name">sql</tt> <tt class="py-op">+=</tt> <tt class="py-string">" and biodatabase_id = %s"</tt> </tt>
<a name="L452"></a><tt class="py-lineno">452</tt>  <tt class="py-line">            <tt class="py-name">fields</tt><tt class="py-op">.</tt><tt id="link-101" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-101', 'append', 'link-73');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">dbid</tt><tt class="py-op">)</tt> </tt>
<a name="L453"></a><tt class="py-lineno">453</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-102" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-102', 'execute', 'link-28');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-name">fields</tt><tt class="py-op">)</tt> </tt>
<a name="L454"></a><tt class="py-lineno">454</tt>  <tt class="py-line">        <tt class="py-name">rv</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-103" class="py-name"><a title="Bio.DocSQL.Query.cursor
Bio.DocSQL.QuerySingle.cursor" class="py-name" href="#" onclick="return doclink('link-103', 'cursor', 'link-59');">cursor</a></tt><tt class="py-op">.</tt><tt id="link-104" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchall" class="py-name" href="#" onclick="return doclink('link-104', 'fetchall', 'link-74');">fetchall</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L455"></a><tt class="py-lineno">455</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">rv</tt><tt class="py-op">:</tt> </tt>
<a name="L456"></a><tt class="py-lineno">456</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">IndexError</tt><tt class="py-op">(</tt><tt class="py-string">"Cannot find accession %r"</tt> <tt class="py-op">%</tt> <tt id="link-105" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-105', 'name', 'link-12');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L457"></a><tt class="py-lineno">457</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">rv</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L458"></a><tt class="py-lineno">458</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">IndexError</tt><tt class="py-op">(</tt><tt class="py-string">"More than one entry with accession %r"</tt> <tt class="py-op">%</tt> <tt id="link-106" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-106', 'name', 'link-12');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L459"></a><tt class="py-lineno">459</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">rv</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
</div><a name="L460"></a><tt class="py-lineno">460</tt>  <tt class="py-line"> </tt>
<a name="Adaptor.fetch_seqids_by_accession"></a><div id="Adaptor.fetch_seqids_by_accession-def"><a name="L461"></a><tt class="py-lineno">461</tt> <a class="py-toggle" href="#" id="Adaptor.fetch_seqids_by_accession-toggle" onclick="return toggle('Adaptor.fetch_seqids_by_accession');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.Adaptor-class.html#fetch_seqids_by_accession">fetch_seqids_by_accession</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">dbid</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Adaptor.fetch_seqids_by_accession-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Adaptor.fetch_seqids_by_accession-expanded"><a name="L462"></a><tt class="py-lineno">462</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a list internal ids using an accession.</tt> </tt>
<a name="L463"></a><tt class="py-lineno">463</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L464"></a><tt class="py-lineno">464</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L465"></a><tt class="py-lineno">465</tt>  <tt class="py-line"><tt class="py-docstring">         - dbid - the internal id for the sub-database</tt> </tt>
<a name="L466"></a><tt class="py-lineno">466</tt>  <tt class="py-line"><tt class="py-docstring">         - name - the accession of the sequence. Corresponds to the</tt> </tt>
<a name="L467"></a><tt class="py-lineno">467</tt>  <tt class="py-line"><tt class="py-docstring">           accession column of the bioentry table of the SQL schema</tt> </tt>
<a name="L468"></a><tt class="py-lineno">468</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L469"></a><tt class="py-lineno">469</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L470"></a><tt class="py-lineno">470</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">r"select bioentry_id from bioentry where accession = %s"</tt> </tt>
<a name="L471"></a><tt class="py-lineno">471</tt>  <tt class="py-line">        <tt class="py-name">fields</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-107" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-107', 'name', 'link-12');">name</a></tt><tt class="py-op">]</tt> </tt>
<a name="L472"></a><tt class="py-lineno">472</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">dbid</tt><tt class="py-op">:</tt> </tt>
<a name="L473"></a><tt class="py-lineno">473</tt>  <tt class="py-line">            <tt class="py-name">sql</tt> <tt class="py-op">+=</tt> <tt class="py-string">" and biodatabase_id = %s"</tt> </tt>
<a name="L474"></a><tt class="py-lineno">474</tt>  <tt class="py-line">            <tt class="py-name">fields</tt><tt class="py-op">.</tt><tt id="link-108" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-108', 'append', 'link-73');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">dbid</tt><tt class="py-op">)</tt> </tt>
<a name="L475"></a><tt class="py-lineno">475</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-109" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0" class="py-name" href="#" onclick="return doclink('link-109', 'execute_and_fetch_col0', 'link-36');">execute_and_fetch_col0</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-name">fields</tt><tt class="py-op">)</tt> </tt>
</div><a name="L476"></a><tt class="py-lineno">476</tt>  <tt class="py-line"> </tt>
<a name="Adaptor.fetch_seqid_by_version"></a><div id="Adaptor.fetch_seqid_by_version-def"><a name="L477"></a><tt class="py-lineno">477</tt> <a class="py-toggle" href="#" id="Adaptor.fetch_seqid_by_version-toggle" onclick="return toggle('Adaptor.fetch_seqid_by_version');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.Adaptor-class.html#fetch_seqid_by_version">fetch_seqid_by_version</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">dbid</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Adaptor.fetch_seqid_by_version-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Adaptor.fetch_seqid_by_version-expanded"><a name="L478"></a><tt class="py-lineno">478</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the internal id for a sequence using its accession and version.</tt> </tt>
<a name="L479"></a><tt class="py-lineno">479</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L480"></a><tt class="py-lineno">480</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L481"></a><tt class="py-lineno">481</tt>  <tt class="py-line"><tt class="py-docstring">         - dbid - the internal id for the sub-database</tt> </tt>
<a name="L482"></a><tt class="py-lineno">482</tt>  <tt class="py-line"><tt class="py-docstring">         - name - the accession of the sequence containing a version number.</tt> </tt>
<a name="L483"></a><tt class="py-lineno">483</tt>  <tt class="py-line"><tt class="py-docstring">           Must correspond to &lt;accession&gt;.&lt;version&gt;</tt> </tt>
<a name="L484"></a><tt class="py-lineno">484</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L485"></a><tt class="py-lineno">485</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L486"></a><tt class="py-lineno">486</tt>  <tt class="py-line">        <tt class="py-name">acc_version</tt> <tt class="py-op">=</tt> <tt id="link-110" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-110', 'name', 'link-12');">name</a></tt><tt class="py-op">.</tt><tt id="link-111" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-111', 'split', 'link-61');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">"."</tt><tt class="py-op">)</tt> </tt>
<a name="L487"></a><tt class="py-lineno">487</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">acc_version</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> </tt>
<a name="L488"></a><tt class="py-lineno">488</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">IndexError</tt><tt class="py-op">(</tt><tt class="py-string">"Bad version %r"</tt> <tt class="py-op">%</tt> <tt id="link-112" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-112', 'name', 'link-12');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L489"></a><tt class="py-lineno">489</tt>  <tt class="py-line">        <tt id="link-113" class="py-name" targets="Variable Bio.FSSP.fssp_rec.align.acc=Bio.FSSP.fssp_rec.align-class.html#acc"><a title="Bio.FSSP.fssp_rec.align.acc" class="py-name" href="#" onclick="return doclink('link-113', 'acc', 'link-113');">acc</a></tt> <tt class="py-op">=</tt> <tt class="py-name">acc_version</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L490"></a><tt class="py-lineno">490</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">acc_version</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">2</tt><tt class="py-op">:</tt> </tt>
<a name="L491"></a><tt class="py-lineno">491</tt>  <tt class="py-line">            <tt id="link-114" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HeaderConsumer.version()=Bio.Blast.NCBIStandalone._HeaderConsumer-class.html#version,Function Bio.Cluster.cluster.version()=Bio.Cluster.cluster-module.html#version,Method Bio.GenBank._FeatureConsumer.version()=Bio.GenBank._FeatureConsumer-class.html#version,Method Bio.GenBank._RecordConsumer.version()=Bio.GenBank._RecordConsumer-class.html#version,Variable Bio.motifs.jaspar.Motif.version=Bio.motifs.jaspar.Motif-class.html#version"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-114', 'version', 'link-114');">version</a></tt> <tt class="py-op">=</tt> <tt class="py-name">acc_version</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L492"></a><tt class="py-lineno">492</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L493"></a><tt class="py-lineno">493</tt>  <tt class="py-line">            <tt id="link-115" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-115', 'version', 'link-114');">version</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"0"</tt> </tt>
<a name="L494"></a><tt class="py-lineno">494</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">r"SELECT bioentry_id FROM bioentry WHERE accession = %s"</tt> \ </tt>
<a name="L495"></a><tt class="py-lineno">495</tt>  <tt class="py-line">              <tt class="py-string">r" AND version = %s"</tt> </tt>
<a name="L496"></a><tt class="py-lineno">496</tt>  <tt class="py-line">        <tt class="py-name">fields</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-116" class="py-name"><a title="Bio.FSSP.fssp_rec.align.acc" class="py-name" href="#" onclick="return doclink('link-116', 'acc', 'link-113');">acc</a></tt><tt class="py-op">,</tt> <tt id="link-117" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-117', 'version', 'link-114');">version</a></tt><tt class="py-op">]</tt> </tt>
<a name="L497"></a><tt class="py-lineno">497</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">dbid</tt><tt class="py-op">:</tt> </tt>
<a name="L498"></a><tt class="py-lineno">498</tt>  <tt class="py-line">            <tt class="py-name">sql</tt> <tt class="py-op">+=</tt> <tt class="py-string">" and biodatabase_id = %s"</tt> </tt>
<a name="L499"></a><tt class="py-lineno">499</tt>  <tt class="py-line">            <tt class="py-name">fields</tt><tt class="py-op">.</tt><tt id="link-118" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-118', 'append', 'link-73');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">dbid</tt><tt class="py-op">)</tt> </tt>
<a name="L500"></a><tt class="py-lineno">500</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-119" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-119', 'execute', 'link-28');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-name">fields</tt><tt class="py-op">)</tt> </tt>
<a name="L501"></a><tt class="py-lineno">501</tt>  <tt class="py-line">        <tt class="py-name">rv</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-120" class="py-name"><a title="Bio.DocSQL.Query.cursor
Bio.DocSQL.QuerySingle.cursor" class="py-name" href="#" onclick="return doclink('link-120', 'cursor', 'link-59');">cursor</a></tt><tt class="py-op">.</tt><tt id="link-121" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchall" class="py-name" href="#" onclick="return doclink('link-121', 'fetchall', 'link-74');">fetchall</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L502"></a><tt class="py-lineno">502</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">rv</tt><tt class="py-op">:</tt> </tt>
<a name="L503"></a><tt class="py-lineno">503</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">IndexError</tt><tt class="py-op">(</tt><tt class="py-string">"Cannot find version %r"</tt> <tt class="py-op">%</tt> <tt id="link-122" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-122', 'name', 'link-12');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L504"></a><tt class="py-lineno">504</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">rv</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L505"></a><tt class="py-lineno">505</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">IndexError</tt><tt class="py-op">(</tt><tt class="py-string">"More than one entry with version %r"</tt> <tt class="py-op">%</tt> <tt id="link-123" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-123', 'name', 'link-12');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L506"></a><tt class="py-lineno">506</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">rv</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
</div><a name="L507"></a><tt class="py-lineno">507</tt>  <tt class="py-line"> </tt>
<a name="Adaptor.fetch_seqid_by_identifier"></a><div id="Adaptor.fetch_seqid_by_identifier-def"><a name="L508"></a><tt class="py-lineno">508</tt> <a class="py-toggle" href="#" id="Adaptor.fetch_seqid_by_identifier-toggle" onclick="return toggle('Adaptor.fetch_seqid_by_identifier');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.Adaptor-class.html#fetch_seqid_by_identifier">fetch_seqid_by_identifier</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">dbid</tt><tt class="py-op">,</tt> <tt class="py-param">identifier</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Adaptor.fetch_seqid_by_identifier-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Adaptor.fetch_seqid_by_identifier-expanded"><a name="L509"></a><tt class="py-lineno">509</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return the internal id for a sequence using its identifier.</tt> </tt>
<a name="L510"></a><tt class="py-lineno">510</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L511"></a><tt class="py-lineno">511</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L512"></a><tt class="py-lineno">512</tt>  <tt class="py-line"><tt class="py-docstring">         - dbid - the internal id for the sub-database</tt> </tt>
<a name="L513"></a><tt class="py-lineno">513</tt>  <tt class="py-line"><tt class="py-docstring">         - identifier - the identifier of the sequence. Corresponds to</tt> </tt>
<a name="L514"></a><tt class="py-lineno">514</tt>  <tt class="py-line"><tt class="py-docstring">           the identifier column of the bioentry table in the SQL schema.</tt> </tt>
<a name="L515"></a><tt class="py-lineno">515</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L516"></a><tt class="py-lineno">516</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L517"></a><tt class="py-lineno">517</tt>  <tt class="py-line">        <tt class="py-comment"># YB: was fetch_seqid_by_seqid</tt> </tt>
<a name="L518"></a><tt class="py-lineno">518</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">"SELECT bioentry_id FROM bioentry WHERE identifier = %s"</tt> </tt>
<a name="L519"></a><tt class="py-lineno">519</tt>  <tt class="py-line">        <tt class="py-name">fields</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">identifier</tt><tt class="py-op">]</tt> </tt>
<a name="L520"></a><tt class="py-lineno">520</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">dbid</tt><tt class="py-op">:</tt> </tt>
<a name="L521"></a><tt class="py-lineno">521</tt>  <tt class="py-line">            <tt class="py-name">sql</tt> <tt class="py-op">+=</tt> <tt class="py-string">" and biodatabase_id = %s"</tt> </tt>
<a name="L522"></a><tt class="py-lineno">522</tt>  <tt class="py-line">            <tt class="py-name">fields</tt><tt class="py-op">.</tt><tt id="link-124" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-124', 'append', 'link-73');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">dbid</tt><tt class="py-op">)</tt> </tt>
<a name="L523"></a><tt class="py-lineno">523</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-125" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-125', 'execute', 'link-28');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-name">fields</tt><tt class="py-op">)</tt> </tt>
<a name="L524"></a><tt class="py-lineno">524</tt>  <tt class="py-line">        <tt class="py-name">rv</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-126" class="py-name"><a title="Bio.DocSQL.Query.cursor
Bio.DocSQL.QuerySingle.cursor" class="py-name" href="#" onclick="return doclink('link-126', 'cursor', 'link-59');">cursor</a></tt><tt class="py-op">.</tt><tt id="link-127" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchall" class="py-name" href="#" onclick="return doclink('link-127', 'fetchall', 'link-74');">fetchall</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L525"></a><tt class="py-lineno">525</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">rv</tt><tt class="py-op">:</tt> </tt>
<a name="L526"></a><tt class="py-lineno">526</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">IndexError</tt><tt class="py-op">(</tt><tt class="py-string">"Cannot find display id %r"</tt> <tt class="py-op">%</tt> <tt class="py-name">identifier</tt><tt class="py-op">)</tt> </tt>
<a name="L527"></a><tt class="py-lineno">527</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">rv</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
</div><a name="L528"></a><tt class="py-lineno">528</tt>  <tt class="py-line"> </tt>
<a name="Adaptor.list_biodatabase_names"></a><div id="Adaptor.list_biodatabase_names-def"><a name="L529"></a><tt class="py-lineno">529</tt> <a class="py-toggle" href="#" id="Adaptor.list_biodatabase_names-toggle" onclick="return toggle('Adaptor.list_biodatabase_names');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.Adaptor-class.html#list_biodatabase_names">list_biodatabase_names</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Adaptor.list_biodatabase_names-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Adaptor.list_biodatabase_names-expanded"><a name="L530"></a><tt class="py-lineno">530</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a list of all of the sub-databases."""</tt> </tt>
<a name="L531"></a><tt class="py-lineno">531</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-128" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0" class="py-name" href="#" onclick="return doclink('link-128', 'execute_and_fetch_col0', 'link-36');">execute_and_fetch_col0</a></tt><tt class="py-op">(</tt> </tt>
<a name="L532"></a><tt class="py-lineno">532</tt>  <tt class="py-line">            <tt class="py-string">"SELECT name FROM biodatabase"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L533"></a><tt class="py-lineno">533</tt>  <tt class="py-line"> </tt>
<a name="Adaptor.list_bioentry_ids"></a><div id="Adaptor.list_bioentry_ids-def"><a name="L534"></a><tt class="py-lineno">534</tt> <a class="py-toggle" href="#" id="Adaptor.list_bioentry_ids-toggle" onclick="return toggle('Adaptor.list_bioentry_ids');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.Adaptor-class.html#list_bioentry_ids">list_bioentry_ids</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">dbid</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Adaptor.list_bioentry_ids-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Adaptor.list_bioentry_ids-expanded"><a name="L535"></a><tt class="py-lineno">535</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a list of internal ids for all of the sequences in a sub-databae.</tt> </tt>
<a name="L536"></a><tt class="py-lineno">536</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L537"></a><tt class="py-lineno">537</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L538"></a><tt class="py-lineno">538</tt>  <tt class="py-line"><tt class="py-docstring">         - dbid - The internal id for a sub-database</tt> </tt>
<a name="L539"></a><tt class="py-lineno">539</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L540"></a><tt class="py-lineno">540</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L541"></a><tt class="py-lineno">541</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-129" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0" class="py-name" href="#" onclick="return doclink('link-129', 'execute_and_fetch_col0', 'link-36');">execute_and_fetch_col0</a></tt><tt class="py-op">(</tt> </tt>
<a name="L542"></a><tt class="py-lineno">542</tt>  <tt class="py-line">            <tt class="py-string">"SELECT bioentry_id FROM bioentry WHERE biodatabase_id = %s"</tt><tt class="py-op">,</tt> </tt>
<a name="L543"></a><tt class="py-lineno">543</tt>  <tt class="py-line">            <tt class="py-op">(</tt><tt class="py-name">dbid</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L544"></a><tt class="py-lineno">544</tt>  <tt class="py-line"> </tt>
<a name="Adaptor.list_bioentry_display_ids"></a><div id="Adaptor.list_bioentry_display_ids-def"><a name="L545"></a><tt class="py-lineno">545</tt> <a class="py-toggle" href="#" id="Adaptor.list_bioentry_display_ids-toggle" onclick="return toggle('Adaptor.list_bioentry_display_ids');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.Adaptor-class.html#list_bioentry_display_ids">list_bioentry_display_ids</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">dbid</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Adaptor.list_bioentry_display_ids-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Adaptor.list_bioentry_display_ids-expanded"><a name="L546"></a><tt class="py-lineno">546</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a list of all sequence names in a sub-databae.</tt> </tt>
<a name="L547"></a><tt class="py-lineno">547</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L548"></a><tt class="py-lineno">548</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L549"></a><tt class="py-lineno">549</tt>  <tt class="py-line"><tt class="py-docstring">         - dbid - The internal id for a sub-database</tt> </tt>
<a name="L550"></a><tt class="py-lineno">550</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L551"></a><tt class="py-lineno">551</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L552"></a><tt class="py-lineno">552</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-130" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0" class="py-name" href="#" onclick="return doclink('link-130', 'execute_and_fetch_col0', 'link-36');">execute_and_fetch_col0</a></tt><tt class="py-op">(</tt> </tt>
<a name="L553"></a><tt class="py-lineno">553</tt>  <tt class="py-line">            <tt class="py-string">"SELECT name FROM bioentry WHERE biodatabase_id = %s"</tt><tt class="py-op">,</tt> </tt>
<a name="L554"></a><tt class="py-lineno">554</tt>  <tt class="py-line">            <tt class="py-op">(</tt><tt class="py-name">dbid</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L555"></a><tt class="py-lineno">555</tt>  <tt class="py-line"> </tt>
<a name="Adaptor.list_any_ids"></a><div id="Adaptor.list_any_ids-def"><a name="L556"></a><tt class="py-lineno">556</tt> <a class="py-toggle" href="#" id="Adaptor.list_any_ids-toggle" onclick="return toggle('Adaptor.list_any_ids');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.Adaptor-class.html#list_any_ids">list_any_ids</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sql</tt><tt class="py-op">,</tt> <tt class="py-param">args</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Adaptor.list_any_ids-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Adaptor.list_any_ids-expanded"><a name="L557"></a><tt class="py-lineno">557</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return ids given a SQL statement to select for them.</tt> </tt>
<a name="L558"></a><tt class="py-lineno">558</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L559"></a><tt class="py-lineno">559</tt>  <tt class="py-line"><tt class="py-docstring">        This assumes that the given SQL does a SELECT statement that</tt> </tt>
<a name="L560"></a><tt class="py-lineno">560</tt>  <tt class="py-line"><tt class="py-docstring">        returns a list of items. This parses them out of the 2D list</tt> </tt>
<a name="L561"></a><tt class="py-lineno">561</tt>  <tt class="py-line"><tt class="py-docstring">        they come as and just returns them in a list.</tt> </tt>
<a name="L562"></a><tt class="py-lineno">562</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L563"></a><tt class="py-lineno">563</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-131" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0" class="py-name" href="#" onclick="return doclink('link-131', 'execute_and_fetch_col0', 'link-36');">execute_and_fetch_col0</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-name">args</tt><tt class="py-op">)</tt> </tt>
</div><a name="L564"></a><tt class="py-lineno">564</tt>  <tt class="py-line"> </tt>
<a name="Adaptor.execute_one"></a><div id="Adaptor.execute_one-def"><a name="L565"></a><tt class="py-lineno">565</tt> <a class="py-toggle" href="#" id="Adaptor.execute_one-toggle" onclick="return toggle('Adaptor.execute_one');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.Adaptor-class.html#execute_one">execute_one</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sql</tt><tt class="py-op">,</tt> <tt class="py-param">args</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Adaptor.execute_one-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Adaptor.execute_one-expanded"><a name="L566"></a><tt class="py-lineno">566</tt>  <tt class="py-line">        <tt class="py-docstring">"""Execute sql that returns 1 record, and return the record."""</tt> </tt>
<a name="L567"></a><tt class="py-lineno">567</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-132" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-132', 'execute', 'link-28');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-name">args</tt> <tt class="py-keyword">or</tt> <tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L568"></a><tt class="py-lineno">568</tt>  <tt class="py-line">        <tt class="py-name">rv</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-133" class="py-name"><a title="Bio.DocSQL.Query.cursor
Bio.DocSQL.QuerySingle.cursor" class="py-name" href="#" onclick="return doclink('link-133', 'cursor', 'link-59');">cursor</a></tt><tt class="py-op">.</tt><tt id="link-134" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchall" class="py-name" href="#" onclick="return doclink('link-134', 'fetchall', 'link-74');">fetchall</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L569"></a><tt class="py-lineno">569</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">rv</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-string">"Expected 1 response, got %d"</tt> <tt class="py-op">%</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">rv</tt><tt class="py-op">)</tt> </tt>
<a name="L570"></a><tt class="py-lineno">570</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">rv</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
</div><a name="L571"></a><tt class="py-lineno">571</tt>  <tt class="py-line"> </tt>
<a name="Adaptor.execute"></a><div id="Adaptor.execute-def"><a name="L572"></a><tt class="py-lineno">572</tt> <a class="py-toggle" href="#" id="Adaptor.execute-toggle" onclick="return toggle('Adaptor.execute');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.Adaptor-class.html#execute">execute</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sql</tt><tt class="py-op">,</tt> <tt class="py-param">args</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Adaptor.execute-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Adaptor.execute-expanded"><a name="L573"></a><tt class="py-lineno">573</tt>  <tt class="py-line">        <tt class="py-docstring">"""Just execute an sql command."""</tt> </tt>
<a name="L574"></a><tt class="py-lineno">574</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt id="link-135" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-135', 'name', 'link-12');">name</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"java"</tt><tt class="py-op">:</tt> </tt>
<a name="L575"></a><tt class="py-lineno">575</tt>  <tt class="py-line">            <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-name">sql</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"%s"</tt><tt class="py-op">,</tt> <tt class="py-string">"?"</tt><tt class="py-op">)</tt> </tt>
<a name="L576"></a><tt class="py-lineno">576</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbutils</tt><tt class="py-op">.</tt><tt id="link-136" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-136', 'execute', 'link-28');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-137" class="py-name"><a title="Bio.DocSQL.Query.cursor
Bio.DocSQL.QuerySingle.cursor" class="py-name" href="#" onclick="return doclink('link-137', 'cursor', 'link-59');">cursor</a></tt><tt class="py-op">,</tt> <tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-name">args</tt><tt class="py-op">)</tt> </tt>
</div><a name="L577"></a><tt class="py-lineno">577</tt>  <tt class="py-line"> </tt>
<a name="Adaptor.executemany"></a><div id="Adaptor.executemany-def"><a name="L578"></a><tt class="py-lineno">578</tt> <a class="py-toggle" href="#" id="Adaptor.executemany-toggle" onclick="return toggle('Adaptor.executemany');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.Adaptor-class.html#executemany">executemany</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sql</tt><tt class="py-op">,</tt> <tt class="py-param">args</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Adaptor.executemany-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Adaptor.executemany-expanded"><a name="L579"></a><tt class="py-lineno">579</tt>  <tt class="py-line">        <tt class="py-docstring">"""Execute many sql commands."""</tt> </tt>
<a name="L580"></a><tt class="py-lineno">580</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">os</tt><tt class="py-op">.</tt><tt id="link-138" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-138', 'name', 'link-12');">name</a></tt> <tt class="py-op">==</tt> <tt class="py-string">"java"</tt><tt class="py-op">:</tt> </tt>
<a name="L581"></a><tt class="py-lineno">581</tt>  <tt class="py-line">            <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-name">sql</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">"%s"</tt><tt class="py-op">,</tt> <tt class="py-string">"?"</tt><tt class="py-op">)</tt> </tt>
<a name="L582"></a><tt class="py-lineno">582</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbutils</tt><tt class="py-op">.</tt><tt id="link-139" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.executemany
BioSQL.BioSeqDatabase._CursorWrapper.executemany
BioSQL.DBUtils.Generic_dbutils.executemany
BioSQL.DBUtils.Sqlite_dbutils.executemany" class="py-name" href="#" onclick="return doclink('link-139', 'executemany', 'link-68');">executemany</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-140" class="py-name"><a title="Bio.DocSQL.Query.cursor
Bio.DocSQL.QuerySingle.cursor" class="py-name" href="#" onclick="return doclink('link-140', 'cursor', 'link-59');">cursor</a></tt><tt class="py-op">,</tt> <tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-name">args</tt><tt class="py-op">)</tt> </tt>
</div><a name="L583"></a><tt class="py-lineno">583</tt>  <tt class="py-line"> </tt>
<a name="Adaptor.get_subseq_as_string"></a><div id="Adaptor.get_subseq_as_string-def"><a name="L584"></a><tt class="py-lineno">584</tt> <a class="py-toggle" href="#" id="Adaptor.get_subseq_as_string-toggle" onclick="return toggle('Adaptor.get_subseq_as_string');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.Adaptor-class.html#get_subseq_as_string">get_subseq_as_string</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">seqid</tt><tt class="py-op">,</tt> <tt class="py-param">start</tt><tt class="py-op">,</tt> <tt class="py-param">end</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Adaptor.get_subseq_as_string-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Adaptor.get_subseq_as_string-expanded"><a name="L585"></a><tt class="py-lineno">585</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a substring of a sequence.</tt> </tt>
<a name="L586"></a><tt class="py-lineno">586</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L587"></a><tt class="py-lineno">587</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L588"></a><tt class="py-lineno">588</tt>  <tt class="py-line"><tt class="py-docstring">         - seqid - The internal id for the sequence</tt> </tt>
<a name="L589"></a><tt class="py-lineno">589</tt>  <tt class="py-line"><tt class="py-docstring">         - start - The start position of the sequence; 0-indexed</tt> </tt>
<a name="L590"></a><tt class="py-lineno">590</tt>  <tt class="py-line"><tt class="py-docstring">         - end - The end position of the sequence</tt> </tt>
<a name="L591"></a><tt class="py-lineno">591</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L592"></a><tt class="py-lineno">592</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L593"></a><tt class="py-lineno">593</tt>  <tt class="py-line">        <tt id="link-141" class="py-name" targets="Method Bio.Blast.NCBIStandalone._AlignmentConsumer.length()=Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#length"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-141', 'length', 'link-141');">length</a></tt> <tt class="py-op">=</tt> <tt id="link-142" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.end=Bio.SeqFeature.CompoundLocation-class.html#end,Variable Bio.SeqFeature.FeatureLocation.end=Bio.SeqFeature.FeatureLocation-class.html#end"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-142', 'end', 'link-142');">end</a></tt> <tt class="py-op">-</tt> <tt id="link-143" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.start=Bio.SeqFeature.CompoundLocation-class.html#start,Variable Bio.SeqFeature.FeatureLocation.start=Bio.SeqFeature.FeatureLocation-class.html#start"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-143', 'start', 'link-143');">start</a></tt> </tt>
<a name="L594"></a><tt class="py-lineno">594</tt>  <tt class="py-line">        <tt class="py-comment"># XXX Check this on MySQL and PostgreSQL. substr should be general,</tt> </tt>
<a name="L595"></a><tt class="py-lineno">595</tt>  <tt class="py-line">        <tt class="py-comment"># does it need dbutils?</tt> </tt>
<a name="L596"></a><tt class="py-lineno">596</tt>  <tt class="py-line">        <tt class="py-comment"># return self.execute_one(</tt> </tt>
<a name="L597"></a><tt class="py-lineno">597</tt>  <tt class="py-line">        <tt class="py-comment">#    """select SUBSTRING(seq FROM %s FOR %s)</tt> </tt>
<a name="L598"></a><tt class="py-lineno">598</tt>  <tt class="py-line">        <tt class="py-comment">#             from biosequence where bioentry_id = %s""",</tt> </tt>
<a name="L599"></a><tt class="py-lineno">599</tt>  <tt class="py-line">        <tt class="py-comment">#    (start+1, length, seqid))[0]</tt> </tt>
<a name="L600"></a><tt class="py-lineno">600</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L601"></a><tt class="py-lineno">601</tt>  <tt class="py-line">        <tt class="py-comment"># Convert to a string on returning for databases that give back</tt> </tt>
<a name="L602"></a><tt class="py-lineno">602</tt>  <tt class="py-line">        <tt class="py-comment"># unicode. Shouldn't need unicode for sequences so this seems safe.</tt> </tt>
<a name="L603"></a><tt class="py-lineno">603</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-144" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.execute_one()=BioSQL.BioSeqDatabase.Adaptor-class.html#execute_one,Method BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_one()=BioSQL.BioSeqDatabase.MysqlConnectorAdaptor-class.html#execute_one"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_one
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_one" class="py-name" href="#" onclick="return doclink('link-144', 'execute_one', 'link-144');">execute_one</a></tt><tt class="py-op">(</tt> </tt>
<a name="L604"></a><tt class="py-lineno">604</tt>  <tt class="py-line">            <tt class="py-string">"""select SUBSTR(seq, %s, %s)</tt> </tt>
<a name="L605"></a><tt class="py-lineno">605</tt>  <tt class="py-line"><tt class="py-string">                     from biosequence where bioentry_id = %s"""</tt><tt class="py-op">,</tt> </tt>
<a name="L606"></a><tt class="py-lineno">606</tt>  <tt class="py-line">            <tt class="py-op">(</tt><tt id="link-145" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-145', 'start', 'link-143');">start</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt id="link-146" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.length" class="py-name" href="#" onclick="return doclink('link-146', 'length', 'link-141');">length</a></tt><tt class="py-op">,</tt> <tt class="py-name">seqid</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div><a name="L607"></a><tt class="py-lineno">607</tt>  <tt class="py-line"> </tt>
<a name="Adaptor.execute_and_fetch_col0"></a><div id="Adaptor.execute_and_fetch_col0-def"><a name="L608"></a><tt class="py-lineno">608</tt> <a class="py-toggle" href="#" id="Adaptor.execute_and_fetch_col0-toggle" onclick="return toggle('Adaptor.execute_and_fetch_col0');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.Adaptor-class.html#execute_and_fetch_col0">execute_and_fetch_col0</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sql</tt><tt class="py-op">,</tt> <tt class="py-param">args</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Adaptor.execute_and_fetch_col0-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Adaptor.execute_and_fetch_col0-expanded"><a name="L609"></a><tt class="py-lineno">609</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a list of values from the first column in the row."""</tt> </tt>
<a name="L610"></a><tt class="py-lineno">610</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-147" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-147', 'execute', 'link-28');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-name">args</tt> <tt class="py-keyword">or</tt> <tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L611"></a><tt class="py-lineno">611</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">field</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt class="py-name">field</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-148" class="py-name"><a title="Bio.DocSQL.Query.cursor
Bio.DocSQL.QuerySingle.cursor" class="py-name" href="#" onclick="return doclink('link-148', 'cursor', 'link-59');">cursor</a></tt><tt class="py-op">.</tt><tt id="link-149" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchall" class="py-name" href="#" onclick="return doclink('link-149', 'fetchall', 'link-74');">fetchall</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
</div><a name="L612"></a><tt class="py-lineno">612</tt>  <tt class="py-line"> </tt>
<a name="Adaptor.execute_and_fetchall"></a><div id="Adaptor.execute_and_fetchall-def"><a name="L613"></a><tt class="py-lineno">613</tt> <a class="py-toggle" href="#" id="Adaptor.execute_and_fetchall-toggle" onclick="return toggle('Adaptor.execute_and_fetchall');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.Adaptor-class.html#execute_and_fetchall">execute_and_fetchall</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sql</tt><tt class="py-op">,</tt> <tt class="py-param">args</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="Adaptor.execute_and_fetchall-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="Adaptor.execute_and_fetchall-expanded"><a name="L614"></a><tt class="py-lineno">614</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a list of tuples of all rows."""</tt> </tt>
<a name="L615"></a><tt class="py-lineno">615</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-150" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-150', 'execute', 'link-28');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-name">args</tt> <tt class="py-keyword">or</tt> <tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L616"></a><tt class="py-lineno">616</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-151" class="py-name"><a title="Bio.DocSQL.Query.cursor
Bio.DocSQL.QuerySingle.cursor" class="py-name" href="#" onclick="return doclink('link-151', 'cursor', 'link-59');">cursor</a></tt><tt class="py-op">.</tt><tt id="link-152" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchall" class="py-name" href="#" onclick="return doclink('link-152', 'fetchall', 'link-74');">fetchall</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L617"></a><tt class="py-lineno">617</tt>  <tt class="py-line"> </tt>
<a name="L618"></a><tt class="py-lineno">618</tt>  <tt class="py-line"> </tt>
<a name="MysqlConnectorAdaptor"></a><div id="MysqlConnectorAdaptor-def"><a name="L619"></a><tt class="py-lineno">619</tt> <a class="py-toggle" href="#" id="MysqlConnectorAdaptor-toggle" onclick="return toggle('MysqlConnectorAdaptor');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.MysqlConnectorAdaptor-class.html">MysqlConnectorAdaptor</a><tt class="py-op">(</tt><tt class="py-base-class">Adaptor</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MysqlConnectorAdaptor-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="MysqlConnectorAdaptor-expanded"><a name="L620"></a><tt class="py-lineno">620</tt>  <tt class="py-line">    <tt class="py-docstring">"""A BioSQL Adaptor class with fixes for the MySQL interface.</tt> </tt>
<a name="L621"></a><tt class="py-lineno">621</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L622"></a><tt class="py-lineno">622</tt>  <tt class="py-line"><tt class="py-docstring">    BioSQL was failing due to returns of bytearray objects from</tt> </tt>
<a name="L623"></a><tt class="py-lineno">623</tt>  <tt class="py-line"><tt class="py-docstring">    the mysql-connector-python database connector. This adaptor</tt> </tt>
<a name="L624"></a><tt class="py-lineno">624</tt>  <tt class="py-line"><tt class="py-docstring">    class scrubs returns of bytearrays and of byte strings converting</tt> </tt>
<a name="L625"></a><tt class="py-lineno">625</tt>  <tt class="py-line"><tt class="py-docstring">    them to string objects instead. This adaptor class was made in</tt> </tt>
<a name="L626"></a><tt class="py-lineno">626</tt>  <tt class="py-line"><tt class="py-docstring">    response to backwards incompatible changes added to</tt> </tt>
<a name="L627"></a><tt class="py-lineno">627</tt>  <tt class="py-line"><tt class="py-docstring">    mysql-connector-python in release 2.0.0 of the package.</tt> </tt>
<a name="L628"></a><tt class="py-lineno">628</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L629"></a><tt class="py-lineno">629</tt>  <tt class="py-line"> </tt>
<a name="MysqlConnectorAdaptor.execute_one"></a><div id="MysqlConnectorAdaptor.execute_one-def"><a name="L630"></a><tt class="py-lineno">630</tt> <a class="py-toggle" href="#" id="MysqlConnectorAdaptor.execute_one-toggle" onclick="return toggle('MysqlConnectorAdaptor.execute_one');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.MysqlConnectorAdaptor-class.html#execute_one">execute_one</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sql</tt><tt class="py-op">,</tt> <tt class="py-param">args</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MysqlConnectorAdaptor.execute_one-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MysqlConnectorAdaptor.execute_one-expanded"><a name="L631"></a><tt class="py-lineno">631</tt>  <tt class="py-line">        <tt class="py-docstring">"""Execute sql that returns 1 record, and return the record."""</tt> </tt>
<a name="L632"></a><tt class="py-lineno">632</tt>  <tt class="py-line">        <tt class="py-name">out</tt> <tt class="py-op">=</tt> <tt class="py-name">super</tt><tt class="py-op">(</tt><tt id="link-153" class="py-name" targets="Class BioSQL.BioSeqDatabase.MysqlConnectorAdaptor=BioSQL.BioSeqDatabase.MysqlConnectorAdaptor-class.html"><a title="BioSQL.BioSeqDatabase.MysqlConnectorAdaptor" class="py-name" href="#" onclick="return doclink('link-153', 'MysqlConnectorAdaptor', 'link-153');">MysqlConnectorAdaptor</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-154" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_one
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_one" class="py-name" href="#" onclick="return doclink('link-154', 'execute_one', 'link-144');">execute_one</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-name">args</tt><tt class="py-op">)</tt> </tt>
<a name="L633"></a><tt class="py-lineno">633</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">tuple</tt><tt class="py-op">(</tt><tt class="py-name">bytearray_to_str</tt><tt class="py-op">(</tt><tt class="py-name">v</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">v</tt> <tt class="py-keyword">in</tt> <tt class="py-name">out</tt><tt class="py-op">)</tt> </tt>
</div><a name="L634"></a><tt class="py-lineno">634</tt>  <tt class="py-line"> </tt>
<a name="MysqlConnectorAdaptor.execute_and_fetch_col0"></a><div id="MysqlConnectorAdaptor.execute_and_fetch_col0-def"><a name="L635"></a><tt class="py-lineno">635</tt> <a class="py-toggle" href="#" id="MysqlConnectorAdaptor.execute_and_fetch_col0-toggle" onclick="return toggle('MysqlConnectorAdaptor.execute_and_fetch_col0');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.MysqlConnectorAdaptor-class.html#execute_and_fetch_col0">execute_and_fetch_col0</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sql</tt><tt class="py-op">,</tt> <tt class="py-param">args</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MysqlConnectorAdaptor.execute_and_fetch_col0-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MysqlConnectorAdaptor.execute_and_fetch_col0-expanded"><a name="L636"></a><tt class="py-lineno">636</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a list of values from the first column in the row."""</tt> </tt>
<a name="L637"></a><tt class="py-lineno">637</tt>  <tt class="py-line">        <tt class="py-name">out</tt> <tt class="py-op">=</tt> <tt class="py-name">super</tt><tt class="py-op">(</tt><tt id="link-155" class="py-name"><a title="BioSQL.BioSeqDatabase.MysqlConnectorAdaptor" class="py-name" href="#" onclick="return doclink('link-155', 'MysqlConnectorAdaptor', 'link-153');">MysqlConnectorAdaptor</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-156" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0" class="py-name" href="#" onclick="return doclink('link-156', 'execute_and_fetch_col0', 'link-36');">execute_and_fetch_col0</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-name">args</tt><tt class="py-op">)</tt> </tt>
<a name="L638"></a><tt class="py-lineno">638</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">bytearray_to_str</tt><tt class="py-op">(</tt><tt class="py-name">column</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">column</tt> <tt class="py-keyword">in</tt> <tt class="py-name">out</tt><tt class="py-op">]</tt> </tt>
</div><a name="L639"></a><tt class="py-lineno">639</tt>  <tt class="py-line"> </tt>
<a name="MysqlConnectorAdaptor.execute_and_fetchall"></a><div id="MysqlConnectorAdaptor.execute_and_fetchall-def"><a name="L640"></a><tt class="py-lineno">640</tt> <a class="py-toggle" href="#" id="MysqlConnectorAdaptor.execute_and_fetchall-toggle" onclick="return toggle('MysqlConnectorAdaptor.execute_and_fetchall');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.MysqlConnectorAdaptor-class.html#execute_and_fetchall">execute_and_fetchall</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">sql</tt><tt class="py-op">,</tt> <tt class="py-param">args</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="MysqlConnectorAdaptor.execute_and_fetchall-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="MysqlConnectorAdaptor.execute_and_fetchall-expanded"><a name="L641"></a><tt class="py-lineno">641</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a list of tuples of all rows."""</tt> </tt>
<a name="L642"></a><tt class="py-lineno">642</tt>  <tt class="py-line">        <tt class="py-name">out</tt> <tt class="py-op">=</tt> <tt class="py-name">super</tt><tt class="py-op">(</tt><tt id="link-157" class="py-name"><a title="BioSQL.BioSeqDatabase.MysqlConnectorAdaptor" class="py-name" href="#" onclick="return doclink('link-157', 'MysqlConnectorAdaptor', 'link-153');">MysqlConnectorAdaptor</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-158" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetchall
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetchall" class="py-name" href="#" onclick="return doclink('link-158', 'execute_and_fetchall', 'link-22');">execute_and_fetchall</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-name">args</tt><tt class="py-op">)</tt> </tt>
<a name="L643"></a><tt class="py-lineno">643</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">tuple</tt><tt class="py-op">(</tt><tt class="py-name">bytearray_to_str</tt><tt class="py-op">(</tt><tt class="py-name">v</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">v</tt> <tt class="py-keyword">in</tt> <tt class="py-name">o</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">o</tt> <tt class="py-keyword">in</tt> <tt class="py-name">out</tt><tt class="py-op">]</tt> </tt>
</div></div><a name="L644"></a><tt class="py-lineno">644</tt>  <tt class="py-line"> </tt>
<a name="L645"></a><tt class="py-lineno">645</tt>  <tt class="py-line"> </tt>
<a name="L646"></a><tt class="py-lineno">646</tt>  <tt class="py-line"><tt id="link-159" class="py-name"><a title="BioSQL.BioSeqDatabase._interface_specific_adaptors" class="py-name" href="#" onclick="return doclink('link-159', '_interface_specific_adaptors', 'link-29');">_interface_specific_adaptors</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L647"></a><tt class="py-lineno">647</tt>  <tt class="py-line">    <tt class="py-comment"># If SQL interfaces require a specific adaptor, use this to map the adaptor</tt> </tt>
<a name="L648"></a><tt class="py-lineno">648</tt>  <tt class="py-line">    <tt class="py-string">"mysql.connector"</tt><tt class="py-op">:</tt> <tt id="link-160" class="py-name"><a title="BioSQL.BioSeqDatabase.MysqlConnectorAdaptor" class="py-name" href="#" onclick="return doclink('link-160', 'MysqlConnectorAdaptor', 'link-153');">MysqlConnectorAdaptor</a></tt><tt class="py-op">,</tt> </tt>
<a name="L649"></a><tt class="py-lineno">649</tt>  <tt class="py-line"><tt class="py-op">}</tt> </tt>
<a name="L650"></a><tt class="py-lineno">650</tt>  <tt class="py-line"> </tt>
<a name="L651"></a><tt class="py-lineno">651</tt>  <tt class="py-line"><tt id="link-161" class="py-name" targets="Variable BioSQL.BioSeqDatabase._allowed_lookups=BioSQL.BioSeqDatabase-module.html#_allowed_lookups"><a title="BioSQL.BioSeqDatabase._allowed_lookups" class="py-name" href="#" onclick="return doclink('link-161', '_allowed_lookups', 'link-161');">_allowed_lookups</a></tt> <tt class="py-op">=</tt> <tt class="py-op">{</tt> </tt>
<a name="L652"></a><tt class="py-lineno">652</tt>  <tt class="py-line">    <tt class="py-comment"># Lookup name / function name to get id, function to list all ids</tt> </tt>
<a name="L653"></a><tt class="py-lineno">653</tt>  <tt class="py-line">    <tt class="py-string">'primary_id'</tt><tt class="py-op">:</tt> <tt class="py-string">"fetch_seqid_by_identifier"</tt><tt class="py-op">,</tt> </tt>
<a name="L654"></a><tt class="py-lineno">654</tt>  <tt class="py-line">    <tt class="py-string">'gi'</tt><tt class="py-op">:</tt> <tt class="py-string">"fetch_seqid_by_identifier"</tt><tt class="py-op">,</tt> </tt>
<a name="L655"></a><tt class="py-lineno">655</tt>  <tt class="py-line">    <tt class="py-string">'display_id'</tt><tt class="py-op">:</tt> <tt class="py-string">"fetch_seqid_by_display_id"</tt><tt class="py-op">,</tt> </tt>
<a name="L656"></a><tt class="py-lineno">656</tt>  <tt class="py-line">    <tt class="py-string">'name'</tt><tt class="py-op">:</tt> <tt class="py-string">"fetch_seqid_by_display_id"</tt><tt class="py-op">,</tt> </tt>
<a name="L657"></a><tt class="py-lineno">657</tt>  <tt class="py-line">    <tt class="py-string">'accession'</tt><tt class="py-op">:</tt> <tt class="py-string">"fetch_seqid_by_accession"</tt><tt class="py-op">,</tt> </tt>
<a name="L658"></a><tt class="py-lineno">658</tt>  <tt class="py-line">    <tt class="py-string">'version'</tt><tt class="py-op">:</tt> <tt class="py-string">"fetch_seqid_by_version"</tt><tt class="py-op">,</tt> </tt>
<a name="L659"></a><tt class="py-lineno">659</tt>  <tt class="py-line"><tt class="py-op">}</tt> </tt>
<a name="L660"></a><tt class="py-lineno">660</tt>  <tt class="py-line"> </tt>
<a name="L661"></a><tt class="py-lineno">661</tt>  <tt class="py-line"> </tt>
<a name="BioSeqDatabase"></a><div id="BioSeqDatabase-def"><a name="L662"></a><tt class="py-lineno">662</tt> <a class="py-toggle" href="#" id="BioSeqDatabase-toggle" onclick="return toggle('BioSeqDatabase');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.BioSeqDatabase-class.html">BioSeqDatabase</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BioSeqDatabase-collapsed" style="display:none;" pad="+++" indent="++++"></div><div id="BioSeqDatabase-expanded"><a name="L663"></a><tt class="py-lineno">663</tt>  <tt class="py-line">    <tt class="py-docstring">"""Represents a namespace (sub-database) within the BioSQL database.</tt> </tt>
<a name="L664"></a><tt class="py-lineno">664</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L665"></a><tt class="py-lineno">665</tt>  <tt class="py-line"><tt class="py-docstring">    i.e. One row in the biodatabase table, and all all rows in the bioentry</tt> </tt>
<a name="L666"></a><tt class="py-lineno">666</tt>  <tt class="py-line"><tt class="py-docstring">    table associated with it.</tt> </tt>
<a name="L667"></a><tt class="py-lineno">667</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L668"></a><tt class="py-lineno">668</tt>  <tt class="py-line"> </tt>
<a name="BioSeqDatabase.__init__"></a><div id="BioSeqDatabase.__init__-def"><a name="L669"></a><tt class="py-lineno">669</tt> <a class="py-toggle" href="#" id="BioSeqDatabase.__init__-toggle" onclick="return toggle('BioSeqDatabase.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">adaptor</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BioSeqDatabase.__init__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BioSeqDatabase.__init__-expanded"><a name="L670"></a><tt class="py-lineno">670</tt>  <tt class="py-line">        <tt class="py-docstring">"""Create a BioDatabase object.</tt> </tt>
<a name="L671"></a><tt class="py-lineno">671</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L672"></a><tt class="py-lineno">672</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L673"></a><tt class="py-lineno">673</tt>  <tt class="py-line"><tt class="py-docstring">         - adaptor - A BioSQL.Adaptor object</tt> </tt>
<a name="L674"></a><tt class="py-lineno">674</tt>  <tt class="py-line"><tt class="py-docstring">         - name - The name of the sub-database (namespace)</tt> </tt>
<a name="L675"></a><tt class="py-lineno">675</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L676"></a><tt class="py-lineno">676</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L677"></a><tt class="py-lineno">677</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt> <tt class="py-op">=</tt> <tt class="py-name">adaptor</tt> </tt>
<a name="L678"></a><tt class="py-lineno">678</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-162" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-162', 'name', 'link-12');">name</a></tt> <tt class="py-op">=</tt> <tt id="link-163" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-163', 'name', 'link-12');">name</a></tt> </tt>
<a name="L679"></a><tt class="py-lineno">679</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbid</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-164" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.fetch_dbid_by_dbname" class="py-name" href="#" onclick="return doclink('link-164', 'fetch_dbid_by_dbname', 'link-43');">fetch_dbid_by_dbname</a></tt><tt class="py-op">(</tt><tt id="link-165" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-165', 'name', 'link-12');">name</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L680"></a><tt class="py-lineno">680</tt>  <tt class="py-line"> </tt>
<a name="BioSeqDatabase.__repr__"></a><div id="BioSeqDatabase.__repr__-def"><a name="L681"></a><tt class="py-lineno">681</tt> <a class="py-toggle" href="#" id="BioSeqDatabase.__repr__-toggle" onclick="return toggle('BioSeqDatabase.__repr__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#__repr__">__repr__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BioSeqDatabase.__repr__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BioSeqDatabase.__repr__-expanded"><a name="L682"></a><tt class="py-lineno">682</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a short summary of the BioSeqDatabase."""</tt> </tt>
<a name="L683"></a><tt class="py-lineno">683</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-string">"BioSeqDatabase(%r, %r)"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-166" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-166', 'name', 'link-12');">name</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L684"></a><tt class="py-lineno">684</tt>  <tt class="py-line"> </tt>
<a name="BioSeqDatabase.get_Seq_by_id"></a><div id="BioSeqDatabase.get_Seq_by_id-def"><a name="L685"></a><tt class="py-lineno">685</tt> <a class="py-toggle" href="#" id="BioSeqDatabase.get_Seq_by_id-toggle" onclick="return toggle('BioSeqDatabase.get_Seq_by_id');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#get_Seq_by_id">get_Seq_by_id</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BioSeqDatabase.get_Seq_by_id-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BioSeqDatabase.get_Seq_by_id-expanded"><a name="L686"></a><tt class="py-lineno">686</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get a DBSeqRecord object by its name.</tt> </tt>
<a name="L687"></a><tt class="py-lineno">687</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L688"></a><tt class="py-lineno">688</tt>  <tt class="py-line"><tt class="py-docstring">        Example: seq_rec = db.get_Seq_by_id('ROA1_HUMAN')</tt> </tt>
<a name="L689"></a><tt class="py-lineno">689</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L690"></a><tt class="py-lineno">690</tt>  <tt class="py-line"><tt class="py-docstring">        The name of this method is misleading since it returns a DBSeqRecord</tt> </tt>
<a name="L691"></a><tt class="py-lineno">691</tt>  <tt class="py-line"><tt class="py-docstring">        rather than a DBSeq ojbect, and presumably was to mirror BioPerl.</tt> </tt>
<a name="L692"></a><tt class="py-lineno">692</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L693"></a><tt class="py-lineno">693</tt>  <tt class="py-line">        <tt class="py-name">seqid</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-167" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.fetch_seqid_by_display_id()=BioSQL.BioSeqDatabase.Adaptor-class.html#fetch_seqid_by_display_id"><a title="BioSQL.BioSeqDatabase.Adaptor.fetch_seqid_by_display_id" class="py-name" href="#" onclick="return doclink('link-167', 'fetch_seqid_by_display_id', 'link-167');">fetch_seqid_by_display_id</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbid</tt><tt class="py-op">,</tt> <tt id="link-168" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-168', 'name', 'link-12');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L694"></a><tt class="py-lineno">694</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-169" class="py-name"><a title="BioSQL.BioSeq" class="py-name" href="#" onclick="return doclink('link-169', 'BioSeq', 'link-8');">BioSeq</a></tt><tt class="py-op">.</tt><tt id="link-170" class="py-name" targets="Class BioSQL.BioSeq.DBSeqRecord=BioSQL.BioSeq.DBSeqRecord-class.html"><a title="BioSQL.BioSeq.DBSeqRecord" class="py-name" href="#" onclick="return doclink('link-170', 'DBSeqRecord', 'link-170');">DBSeqRecord</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">,</tt> <tt class="py-name">seqid</tt><tt class="py-op">)</tt> </tt>
</div><a name="L695"></a><tt class="py-lineno">695</tt>  <tt class="py-line"> </tt>
<a name="BioSeqDatabase.get_Seq_by_acc"></a><div id="BioSeqDatabase.get_Seq_by_acc-def"><a name="L696"></a><tt class="py-lineno">696</tt> <a class="py-toggle" href="#" id="BioSeqDatabase.get_Seq_by_acc-toggle" onclick="return toggle('BioSeqDatabase.get_Seq_by_acc');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#get_Seq_by_acc">get_Seq_by_acc</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BioSeqDatabase.get_Seq_by_acc-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BioSeqDatabase.get_Seq_by_acc-expanded"><a name="L697"></a><tt class="py-lineno">697</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get a DBSeqRecord object by accession number.</tt> </tt>
<a name="L698"></a><tt class="py-lineno">698</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L699"></a><tt class="py-lineno">699</tt>  <tt class="py-line"><tt class="py-docstring">        Example: seq_rec = db.get_Seq_by_acc('X77802')</tt> </tt>
<a name="L700"></a><tt class="py-lineno">700</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L701"></a><tt class="py-lineno">701</tt>  <tt class="py-line"><tt class="py-docstring">        The name of this method is misleading since it returns a DBSeqRecord</tt> </tt>
<a name="L702"></a><tt class="py-lineno">702</tt>  <tt class="py-line"><tt class="py-docstring">        rather than a DBSeq ojbect, and presumably was to mirror BioPerl.</tt> </tt>
<a name="L703"></a><tt class="py-lineno">703</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L704"></a><tt class="py-lineno">704</tt>  <tt class="py-line">        <tt class="py-name">seqid</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-171" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.fetch_seqid_by_accession()=BioSQL.BioSeqDatabase.Adaptor-class.html#fetch_seqid_by_accession"><a title="BioSQL.BioSeqDatabase.Adaptor.fetch_seqid_by_accession" class="py-name" href="#" onclick="return doclink('link-171', 'fetch_seqid_by_accession', 'link-171');">fetch_seqid_by_accession</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbid</tt><tt class="py-op">,</tt> <tt id="link-172" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-172', 'name', 'link-12');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L705"></a><tt class="py-lineno">705</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-173" class="py-name"><a title="BioSQL.BioSeq" class="py-name" href="#" onclick="return doclink('link-173', 'BioSeq', 'link-8');">BioSeq</a></tt><tt class="py-op">.</tt><tt id="link-174" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord" class="py-name" href="#" onclick="return doclink('link-174', 'DBSeqRecord', 'link-170');">DBSeqRecord</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">,</tt> <tt class="py-name">seqid</tt><tt class="py-op">)</tt> </tt>
</div><a name="L706"></a><tt class="py-lineno">706</tt>  <tt class="py-line"> </tt>
<a name="BioSeqDatabase.get_Seq_by_ver"></a><div id="BioSeqDatabase.get_Seq_by_ver-def"><a name="L707"></a><tt class="py-lineno">707</tt> <a class="py-toggle" href="#" id="BioSeqDatabase.get_Seq_by_ver-toggle" onclick="return toggle('BioSeqDatabase.get_Seq_by_ver');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#get_Seq_by_ver">get_Seq_by_ver</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BioSeqDatabase.get_Seq_by_ver-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BioSeqDatabase.get_Seq_by_ver-expanded"><a name="L708"></a><tt class="py-lineno">708</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get a DBSeqRecord object by version number.</tt> </tt>
<a name="L709"></a><tt class="py-lineno">709</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L710"></a><tt class="py-lineno">710</tt>  <tt class="py-line"><tt class="py-docstring">        Example: seq_rec = db.get_Seq_by_ver('X77802.1')</tt> </tt>
<a name="L711"></a><tt class="py-lineno">711</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L712"></a><tt class="py-lineno">712</tt>  <tt class="py-line"><tt class="py-docstring">        The name of this method is misleading since it returns a DBSeqRecord</tt> </tt>
<a name="L713"></a><tt class="py-lineno">713</tt>  <tt class="py-line"><tt class="py-docstring">        rather than a DBSeq ojbect, and presumably was to mirror BioPerl.</tt> </tt>
<a name="L714"></a><tt class="py-lineno">714</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L715"></a><tt class="py-lineno">715</tt>  <tt class="py-line">        <tt class="py-name">seqid</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-175" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.fetch_seqid_by_version()=BioSQL.BioSeqDatabase.Adaptor-class.html#fetch_seqid_by_version"><a title="BioSQL.BioSeqDatabase.Adaptor.fetch_seqid_by_version" class="py-name" href="#" onclick="return doclink('link-175', 'fetch_seqid_by_version', 'link-175');">fetch_seqid_by_version</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbid</tt><tt class="py-op">,</tt> <tt id="link-176" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-176', 'name', 'link-12');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L716"></a><tt class="py-lineno">716</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-177" class="py-name"><a title="BioSQL.BioSeq" class="py-name" href="#" onclick="return doclink('link-177', 'BioSeq', 'link-8');">BioSeq</a></tt><tt class="py-op">.</tt><tt id="link-178" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord" class="py-name" href="#" onclick="return doclink('link-178', 'DBSeqRecord', 'link-170');">DBSeqRecord</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">,</tt> <tt class="py-name">seqid</tt><tt class="py-op">)</tt> </tt>
</div><a name="L717"></a><tt class="py-lineno">717</tt>  <tt class="py-line"> </tt>
<a name="BioSeqDatabase.get_Seqs_by_acc"></a><div id="BioSeqDatabase.get_Seqs_by_acc-def"><a name="L718"></a><tt class="py-lineno">718</tt> <a class="py-toggle" href="#" id="BioSeqDatabase.get_Seqs_by_acc-toggle" onclick="return toggle('BioSeqDatabase.get_Seqs_by_acc');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#get_Seqs_by_acc">get_Seqs_by_acc</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BioSeqDatabase.get_Seqs_by_acc-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BioSeqDatabase.get_Seqs_by_acc-expanded"><a name="L719"></a><tt class="py-lineno">719</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get a list of DBSeqRecord objects by accession number.</tt> </tt>
<a name="L720"></a><tt class="py-lineno">720</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L721"></a><tt class="py-lineno">721</tt>  <tt class="py-line"><tt class="py-docstring">        Example: seq_recs = db.get_Seq_by_acc('X77802')</tt> </tt>
<a name="L722"></a><tt class="py-lineno">722</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L723"></a><tt class="py-lineno">723</tt>  <tt class="py-line"><tt class="py-docstring">        The name of this method is misleading since it returns a list of</tt> </tt>
<a name="L724"></a><tt class="py-lineno">724</tt>  <tt class="py-line"><tt class="py-docstring">        DBSeqRecord objects rather than a list of DBSeq ojbects, and presumably</tt> </tt>
<a name="L725"></a><tt class="py-lineno">725</tt>  <tt class="py-line"><tt class="py-docstring">        was to mirror BioPerl.</tt> </tt>
<a name="L726"></a><tt class="py-lineno">726</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L727"></a><tt class="py-lineno">727</tt>  <tt class="py-line">        <tt class="py-name">seqids</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-179" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.fetch_seqids_by_accession()=BioSQL.BioSeqDatabase.Adaptor-class.html#fetch_seqids_by_accession"><a title="BioSQL.BioSeqDatabase.Adaptor.fetch_seqids_by_accession" class="py-name" href="#" onclick="return doclink('link-179', 'fetch_seqids_by_accession', 'link-179');">fetch_seqids_by_accession</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbid</tt><tt class="py-op">,</tt> <tt id="link-180" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-180', 'name', 'link-12');">name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L728"></a><tt class="py-lineno">728</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt id="link-181" class="py-name"><a title="BioSQL.BioSeq" class="py-name" href="#" onclick="return doclink('link-181', 'BioSeq', 'link-8');">BioSeq</a></tt><tt class="py-op">.</tt><tt id="link-182" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord" class="py-name" href="#" onclick="return doclink('link-182', 'DBSeqRecord', 'link-170');">DBSeqRecord</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">,</tt> <tt class="py-name">seqid</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">seqid</tt> <tt class="py-keyword">in</tt> <tt class="py-name">seqids</tt><tt class="py-op">]</tt> </tt>
</div><a name="L729"></a><tt class="py-lineno">729</tt>  <tt class="py-line"> </tt>
<a name="BioSeqDatabase.get_all_primary_ids"></a><div id="BioSeqDatabase.get_all_primary_ids-def"><a name="L730"></a><tt class="py-lineno">730</tt> <a class="py-toggle" href="#" id="BioSeqDatabase.get_all_primary_ids-toggle" onclick="return toggle('BioSeqDatabase.get_all_primary_ids');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#get_all_primary_ids">get_all_primary_ids</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BioSeqDatabase.get_all_primary_ids-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BioSeqDatabase.get_all_primary_ids-expanded"><a name="L731"></a><tt class="py-lineno">731</tt>  <tt class="py-line">        <tt class="py-docstring">"""All the primary_ids of the sequences in the database (OBSOLETE).</tt> </tt>
<a name="L732"></a><tt class="py-lineno">732</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L733"></a><tt class="py-lineno">733</tt>  <tt class="py-line"><tt class="py-docstring">        These maybe ids (display style) or accession numbers or</tt> </tt>
<a name="L734"></a><tt class="py-lineno">734</tt>  <tt class="py-line"><tt class="py-docstring">        something else completely different - they *are not*</tt> </tt>
<a name="L735"></a><tt class="py-lineno">735</tt>  <tt class="py-line"><tt class="py-docstring">        meaningful outside of this database implementation.</tt> </tt>
<a name="L736"></a><tt class="py-lineno">736</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L737"></a><tt class="py-lineno">737</tt>  <tt class="py-line"><tt class="py-docstring">        Please use .keys() instead of .get_all_primary_ids()</tt> </tt>
<a name="L738"></a><tt class="py-lineno">738</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L739"></a><tt class="py-lineno">739</tt>  <tt class="py-line">        <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L740"></a><tt class="py-lineno">740</tt>  <tt class="py-line">        <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Use bio_seq_database.keys() instead of "</tt> </tt>
<a name="L741"></a><tt class="py-lineno">741</tt>  <tt class="py-line">                      <tt class="py-string">"bio_seq_database.get_all_primary_ids()"</tt><tt class="py-op">,</tt> </tt>
<a name="L742"></a><tt class="py-lineno">742</tt>  <tt class="py-line">                      <tt id="link-183" class="py-name"><a title="Bio.BiopythonDeprecationWarning" class="py-name" href="#" onclick="return doclink('link-183', 'BiopythonDeprecationWarning', 'link-7');">BiopythonDeprecationWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L743"></a><tt class="py-lineno">743</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-184" class="py-name" targets="Method Bio.Crystal.Crystal.keys()=Bio.Crystal.Crystal-class.html#keys,Method Bio.File._IndexedSeqFileDict.keys()=Bio.File._IndexedSeqFileDict-class.html#keys,Method Bio.File._SQLiteManySeqFilesDict.keys()=Bio.File._SQLiteManySeqFilesDict-class.html#keys,Method Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys()=Bio.PDB.AbstractPropertyMap.AbstractPropertyMap-class.html#keys,Method Bio.Phylo.PhyloXML.Events.keys()=Bio.Phylo.PhyloXML.Events-class.html#keys,Method BioSQL.BioSeqDatabase.BioSeqDatabase.keys()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#keys,Method BioSQL.BioSeqDatabase.DBServer.keys()=BioSQL.BioSeqDatabase.DBServer-class.html#keys"><a title="Bio.Crystal.Crystal.keys
Bio.File._IndexedSeqFileDict.keys
Bio.File._SQLiteManySeqFilesDict.keys
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys
Bio.Phylo.PhyloXML.Events.keys
BioSQL.BioSeqDatabase.BioSeqDatabase.keys
BioSQL.BioSeqDatabase.DBServer.keys" class="py-name" href="#" onclick="return doclink('link-184', 'keys', 'link-184');">keys</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L744"></a><tt class="py-lineno">744</tt>  <tt class="py-line"> </tt>
<a name="BioSeqDatabase.__getitem__"></a><div id="BioSeqDatabase.__getitem__-def"><a name="L745"></a><tt class="py-lineno">745</tt> <a class="py-toggle" href="#" id="BioSeqDatabase.__getitem__-toggle" onclick="return toggle('BioSeqDatabase.__getitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#__getitem__">__getitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">key</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BioSeqDatabase.__getitem__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BioSeqDatabase.__getitem__-expanded"><a name="L746"></a><tt class="py-lineno">746</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a DBSeqRecord for one of the sequences in the sub-database.</tt> </tt>
<a name="L747"></a><tt class="py-lineno">747</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L748"></a><tt class="py-lineno">748</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L749"></a><tt class="py-lineno">749</tt>  <tt class="py-line"><tt class="py-docstring">         - key - The internal id for the sequence</tt> </tt>
<a name="L750"></a><tt class="py-lineno">750</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L751"></a><tt class="py-lineno">751</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L752"></a><tt class="py-lineno">752</tt>  <tt class="py-line">        <tt class="py-name">record</tt> <tt class="py-op">=</tt> <tt id="link-185" class="py-name"><a title="BioSQL.BioSeq" class="py-name" href="#" onclick="return doclink('link-185', 'BioSeq', 'link-8');">BioSeq</a></tt><tt class="py-op">.</tt><tt id="link-186" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord" class="py-name" href="#" onclick="return doclink('link-186', 'DBSeqRecord', 'link-170');">DBSeqRecord</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">,</tt> <tt class="py-name">key</tt><tt class="py-op">)</tt> </tt>
<a name="L753"></a><tt class="py-lineno">753</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt class="py-name">_biodatabase_id</tt> <tt class="py-op">!=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbid</tt><tt class="py-op">:</tt> </tt>
<a name="L754"></a><tt class="py-lineno">754</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">KeyError</tt><tt class="py-op">(</tt><tt class="py-string">"Entry %r does exist, but not in current name space"</tt> <tt class="py-op">%</tt> <tt class="py-name">key</tt><tt class="py-op">)</tt> </tt>
<a name="L755"></a><tt class="py-lineno">755</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">record</tt> </tt>
</div><a name="L756"></a><tt class="py-lineno">756</tt>  <tt class="py-line"> </tt>
<a name="BioSeqDatabase.__delitem__"></a><div id="BioSeqDatabase.__delitem__-def"><a name="L757"></a><tt class="py-lineno">757</tt> <a class="py-toggle" href="#" id="BioSeqDatabase.__delitem__-toggle" onclick="return toggle('BioSeqDatabase.__delitem__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#__delitem__">__delitem__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">key</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BioSeqDatabase.__delitem__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BioSeqDatabase.__delitem__-expanded"><a name="L758"></a><tt class="py-lineno">758</tt>  <tt class="py-line">        <tt class="py-docstring">"""Remove an entry and all its annotation."""</tt> </tt>
<a name="L759"></a><tt class="py-lineno">759</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">key</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">:</tt> </tt>
<a name="L760"></a><tt class="py-lineno">760</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">KeyError</tt><tt class="py-op">(</tt><tt class="py-string">"Entry %r cannot be deleted. "</tt> </tt>
<a name="L761"></a><tt class="py-lineno">761</tt>  <tt class="py-line">                           <tt class="py-string">"It was not found or is invalid"</tt> <tt class="py-op">%</tt> <tt class="py-name">key</tt><tt class="py-op">)</tt> </tt>
<a name="L762"></a><tt class="py-lineno">762</tt>  <tt class="py-line">        <tt class="py-comment"># Assuming this will automatically cascade to the other tables...</tt> </tt>
<a name="L763"></a><tt class="py-lineno">763</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">"DELETE FROM bioentry "</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L764"></a><tt class="py-lineno">764</tt>  <tt class="py-line">              <tt class="py-string">"WHERE biodatabase_id=%s AND bioentry_id=%s;"</tt> </tt>
<a name="L765"></a><tt class="py-lineno">765</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-187" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-187', 'execute', 'link-28');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbid</tt><tt class="py-op">,</tt> <tt class="py-name">key</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L766"></a><tt class="py-lineno">766</tt>  <tt class="py-line"> </tt>
<a name="BioSeqDatabase.__len__"></a><div id="BioSeqDatabase.__len__-def"><a name="L767"></a><tt class="py-lineno">767</tt> <a class="py-toggle" href="#" id="BioSeqDatabase.__len__-toggle" onclick="return toggle('BioSeqDatabase.__len__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#__len__">__len__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BioSeqDatabase.__len__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BioSeqDatabase.__len__-expanded"><a name="L768"></a><tt class="py-lineno">768</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return number of records in this namespace (sub database)."""</tt> </tt>
<a name="L769"></a><tt class="py-lineno">769</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">"SELECT COUNT(bioentry_id) FROM bioentry "</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L770"></a><tt class="py-lineno">770</tt>  <tt class="py-line">              <tt class="py-string">"WHERE biodatabase_id=%s;"</tt> </tt>
<a name="L771"></a><tt class="py-lineno">771</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-188" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0" class="py-name" href="#" onclick="return doclink('link-188', 'execute_and_fetch_col0', 'link-36');">execute_and_fetch_col0</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbid</tt><tt class="py-op">,</tt> <tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div><a name="L772"></a><tt class="py-lineno">772</tt>  <tt class="py-line"> </tt>
<a name="BioSeqDatabase.__contains__"></a><div id="BioSeqDatabase.__contains__-def"><a name="L773"></a><tt class="py-lineno">773</tt> <a class="py-toggle" href="#" id="BioSeqDatabase.__contains__-toggle" onclick="return toggle('BioSeqDatabase.__contains__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#__contains__">__contains__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">value</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BioSeqDatabase.__contains__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BioSeqDatabase.__contains__-expanded"><a name="L774"></a><tt class="py-lineno">774</tt>  <tt class="py-line">        <tt class="py-docstring">"""Check if a primary (internal) id is this namespace (sub database)."""</tt> </tt>
<a name="L775"></a><tt class="py-lineno">775</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">"SELECT COUNT(bioentry_id) FROM bioentry "</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L776"></a><tt class="py-lineno">776</tt>  <tt class="py-line">              <tt class="py-string">"WHERE biodatabase_id=%s AND bioentry_id=%s;"</tt> </tt>
<a name="L777"></a><tt class="py-lineno">777</tt>  <tt class="py-line">        <tt class="py-comment"># The bioentry_id field is an integer in the schema.</tt> </tt>
<a name="L778"></a><tt class="py-lineno">778</tt>  <tt class="py-line">        <tt class="py-comment"># PostgreSQL will throw an error if we use a non integer in the query.</tt> </tt>
<a name="L779"></a><tt class="py-lineno">779</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L780"></a><tt class="py-lineno">780</tt>  <tt class="py-line">            <tt class="py-name">bioentry_id</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-189" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-189', 'value', 'link-38');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L781"></a><tt class="py-lineno">781</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L782"></a><tt class="py-lineno">782</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">False</tt> </tt>
<a name="L783"></a><tt class="py-lineno">783</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">bool</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-190" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0" class="py-name" href="#" onclick="return doclink('link-190', 'execute_and_fetch_col0', 'link-36');">execute_and_fetch_col0</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> </tt>
<a name="L784"></a><tt class="py-lineno">784</tt>  <tt class="py-line">                                                        <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbid</tt><tt class="py-op">,</tt> <tt class="py-name">bioentry_id</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
</div><a name="L785"></a><tt class="py-lineno">785</tt>  <tt class="py-line"> </tt>
<a name="BioSeqDatabase.__iter__"></a><div id="BioSeqDatabase.__iter__-def"><a name="L786"></a><tt class="py-lineno">786</tt> <a class="py-toggle" href="#" id="BioSeqDatabase.__iter__-toggle" onclick="return toggle('BioSeqDatabase.__iter__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#__iter__">__iter__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BioSeqDatabase.__iter__-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BioSeqDatabase.__iter__-expanded"><a name="L787"></a><tt class="py-lineno">787</tt>  <tt class="py-line">        <tt class="py-docstring">"""Iterate over ids (which may not be meaningful outside this database)."""</tt> </tt>
<a name="L788"></a><tt class="py-lineno">788</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - Iterate over the cursor, much more efficient</tt> </tt>
<a name="L789"></a><tt class="py-lineno">789</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">iter</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-191" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.list_bioentry_ids()=BioSQL.BioSeqDatabase.Adaptor-class.html#list_bioentry_ids"><a title="BioSQL.BioSeqDatabase.Adaptor.list_bioentry_ids" class="py-name" href="#" onclick="return doclink('link-191', 'list_bioentry_ids', 'link-191');">list_bioentry_ids</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbid</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L790"></a><tt class="py-lineno">790</tt>  <tt class="py-line"> </tt>
<a name="L791"></a><tt class="py-lineno">791</tt>  <tt class="py-line">    <tt class="py-keyword">if</tt> <tt class="py-name">hasattr</tt><tt class="py-op">(</tt><tt class="py-name">dict</tt><tt class="py-op">,</tt> <tt class="py-string">"iteritems"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L792"></a><tt class="py-lineno">792</tt>  <tt class="py-line">        <tt class="py-comment"># Python 2, use iteritems etc</tt> </tt>
<a name="BioSeqDatabase.keys"></a><div id="BioSeqDatabase.keys-def"><a name="L793"></a><tt class="py-lineno">793</tt> <a class="py-toggle" href="#" id="BioSeqDatabase.keys-toggle" onclick="return toggle('BioSeqDatabase.keys');">-</a><tt class="py-line">        <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#keys">keys</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BioSeqDatabase.keys-collapsed" style="display:none;" pad="+++" indent="++++++++++++++++"></div><div id="BioSeqDatabase.keys-expanded"><a name="L794"></a><tt class="py-lineno">794</tt>  <tt class="py-line">            <tt class="py-docstring">"""List of ids which may not be meaningful outside this database."""</tt> </tt>
<a name="L795"></a><tt class="py-lineno">795</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-192" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.list_bioentry_ids" class="py-name" href="#" onclick="return doclink('link-192', 'list_bioentry_ids', 'link-191');">list_bioentry_ids</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbid</tt><tt class="py-op">)</tt> </tt>
</div><a name="L796"></a><tt class="py-lineno">796</tt>  <tt class="py-line"> </tt>
<a name="BioSeqDatabase.values"></a><div id="BioSeqDatabase.values-def"><a name="L797"></a><tt class="py-lineno">797</tt> <a class="py-toggle" href="#" id="BioSeqDatabase.values-toggle" onclick="return toggle('BioSeqDatabase.values');">-</a><tt class="py-line">        <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#values">values</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BioSeqDatabase.values-collapsed" style="display:none;" pad="+++" indent="++++++++++++++++"></div><div id="BioSeqDatabase.values-expanded"><a name="L798"></a><tt class="py-lineno">798</tt>  <tt class="py-line">            <tt class="py-docstring">"""List of DBSeqRecord objects in the namespace (sub database)."""</tt> </tt>
<a name="L799"></a><tt class="py-lineno">799</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-name">self</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">]</tt> </tt>
</div><a name="L800"></a><tt class="py-lineno">800</tt>  <tt class="py-line"> </tt>
<a name="BioSeqDatabase.items"></a><div id="BioSeqDatabase.items-def"><a name="L801"></a><tt class="py-lineno">801</tt> <a class="py-toggle" href="#" id="BioSeqDatabase.items-toggle" onclick="return toggle('BioSeqDatabase.items');">-</a><tt class="py-line">        <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#items">items</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BioSeqDatabase.items-collapsed" style="display:none;" pad="+++" indent="++++++++++++++++"></div><div id="BioSeqDatabase.items-expanded"><a name="L802"></a><tt class="py-lineno">802</tt>  <tt class="py-line">            <tt class="py-docstring">"""List of (id, DBSeqRecord) for the namespace (sub database)."""</tt> </tt>
<a name="L803"></a><tt class="py-lineno">803</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">]</tt> </tt>
</div><a name="L804"></a><tt class="py-lineno">804</tt>  <tt class="py-line"> </tt>
<a name="BioSeqDatabase.iterkeys"></a><div id="BioSeqDatabase.iterkeys-def"><a name="L805"></a><tt class="py-lineno">805</tt> <a class="py-toggle" href="#" id="BioSeqDatabase.iterkeys-toggle" onclick="return toggle('BioSeqDatabase.iterkeys');">-</a><tt class="py-line">        <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#iterkeys">iterkeys</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BioSeqDatabase.iterkeys-collapsed" style="display:none;" pad="+++" indent="++++++++++++++++"></div><div id="BioSeqDatabase.iterkeys-expanded"><a name="L806"></a><tt class="py-lineno">806</tt>  <tt class="py-line">            <tt class="py-docstring">"""Iterate over ids (which may not be meaningful outside this database)."""</tt> </tt>
<a name="L807"></a><tt class="py-lineno">807</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">iter</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
</div><a name="L808"></a><tt class="py-lineno">808</tt>  <tt class="py-line"> </tt>
<a name="BioSeqDatabase.itervalues"></a><div id="BioSeqDatabase.itervalues-def"><a name="L809"></a><tt class="py-lineno">809</tt> <a class="py-toggle" href="#" id="BioSeqDatabase.itervalues-toggle" onclick="return toggle('BioSeqDatabase.itervalues');">-</a><tt class="py-line">        <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#itervalues">itervalues</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BioSeqDatabase.itervalues-collapsed" style="display:none;" pad="+++" indent="++++++++++++++++"></div><div id="BioSeqDatabase.itervalues-expanded"><a name="L810"></a><tt class="py-lineno">810</tt>  <tt class="py-line">            <tt class="py-docstring">"""Iterate over DBSeqRecord objects in the namespace (sub database)."""</tt> </tt>
<a name="L811"></a><tt class="py-lineno">811</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">:</tt> </tt>
<a name="L812"></a><tt class="py-lineno">812</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt class="py-name">self</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> </tt>
</div><a name="L813"></a><tt class="py-lineno">813</tt>  <tt class="py-line"> </tt>
<a name="BioSeqDatabase.iteritems"></a><div id="BioSeqDatabase.iteritems-def"><a name="L814"></a><tt class="py-lineno">814</tt> <a class="py-toggle" href="#" id="BioSeqDatabase.iteritems-toggle" onclick="return toggle('BioSeqDatabase.iteritems');">-</a><tt class="py-line">        <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#iteritems">iteritems</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BioSeqDatabase.iteritems-collapsed" style="display:none;" pad="+++" indent="++++++++++++++++"></div><div id="BioSeqDatabase.iteritems-expanded"><a name="L815"></a><tt class="py-lineno">815</tt>  <tt class="py-line">            <tt class="py-docstring">"""Iterate over (id, DBSeqRecord) for the namespace (sub database)."""</tt> </tt>
<a name="L816"></a><tt class="py-lineno">816</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">:</tt> </tt>
<a name="L817"></a><tt class="py-lineno">817</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> </tt>
</div><a name="L818"></a><tt class="py-lineno">818</tt>  <tt class="py-line">    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L819"></a><tt class="py-lineno">819</tt>  <tt class="py-line">        <tt class="py-comment"># Python 3, items etc are all iterators</tt> </tt>
<a name="BioSeqDatabase.keys"></a><div id="BioSeqDatabase.keys-def"><a name="L820"></a><tt class="py-lineno">820</tt> <a class="py-toggle" href="#" id="BioSeqDatabase.keys-toggle" onclick="return toggle('BioSeqDatabase.keys');">-</a><tt class="py-line">        <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#keys">keys</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BioSeqDatabase.keys-collapsed" style="display:none;" pad="+++" indent="++++++++++++++++"></div><div id="BioSeqDatabase.keys-expanded"><a name="L821"></a><tt class="py-lineno">821</tt>  <tt class="py-line">            <tt class="py-docstring">"""Iterate over ids (which may not be meaningful outside this database)."""</tt> </tt>
<a name="L822"></a><tt class="py-lineno">822</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">iter</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">)</tt> </tt>
</div><a name="L823"></a><tt class="py-lineno">823</tt>  <tt class="py-line"> </tt>
<a name="BioSeqDatabase.values"></a><div id="BioSeqDatabase.values-def"><a name="L824"></a><tt class="py-lineno">824</tt> <a class="py-toggle" href="#" id="BioSeqDatabase.values-toggle" onclick="return toggle('BioSeqDatabase.values');">-</a><tt class="py-line">        <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#values">values</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BioSeqDatabase.values-collapsed" style="display:none;" pad="+++" indent="++++++++++++++++"></div><div id="BioSeqDatabase.values-expanded"><a name="L825"></a><tt class="py-lineno">825</tt>  <tt class="py-line">            <tt class="py-docstring">"""Iterate over DBSeqRecord objects in the namespace (sub database)."""</tt> </tt>
<a name="L826"></a><tt class="py-lineno">826</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">:</tt> </tt>
<a name="L827"></a><tt class="py-lineno">827</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt class="py-name">self</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> </tt>
</div><a name="L828"></a><tt class="py-lineno">828</tt>  <tt class="py-line"> </tt>
<a name="BioSeqDatabase.items"></a><div id="BioSeqDatabase.items-def"><a name="L829"></a><tt class="py-lineno">829</tt> <a class="py-toggle" href="#" id="BioSeqDatabase.items-toggle" onclick="return toggle('BioSeqDatabase.items');">-</a><tt class="py-line">        <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#items">items</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BioSeqDatabase.items-collapsed" style="display:none;" pad="+++" indent="++++++++++++++++"></div><div id="BioSeqDatabase.items-expanded"><a name="L830"></a><tt class="py-lineno">830</tt>  <tt class="py-line">            <tt class="py-docstring">"""Iterate over (id, DBSeqRecord) for the namespace (sub database)."""</tt> </tt>
<a name="L831"></a><tt class="py-lineno">831</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">self</tt><tt class="py-op">:</tt> </tt>
<a name="L832"></a><tt class="py-lineno">832</tt>  <tt class="py-line">                <tt class="py-keyword">yield</tt> <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">[</tt><tt class="py-name">key</tt><tt class="py-op">]</tt> </tt>
</div><a name="L833"></a><tt class="py-lineno">833</tt>  <tt class="py-line"> </tt>
<a name="BioSeqDatabase.lookup"></a><div id="BioSeqDatabase.lookup-def"><a name="L834"></a><tt class="py-lineno">834</tt> <a class="py-toggle" href="#" id="BioSeqDatabase.lookup-toggle" onclick="return toggle('BioSeqDatabase.lookup');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#lookup">lookup</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-op">**</tt><tt class="py-param">kwargs</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BioSeqDatabase.lookup-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BioSeqDatabase.lookup-expanded"><a name="L835"></a><tt class="py-lineno">835</tt>  <tt class="py-line">        <tt class="py-docstring">"""Return a DBSeqRecord using an acceptable identifier.</tt> </tt>
<a name="L836"></a><tt class="py-lineno">836</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L837"></a><tt class="py-lineno">837</tt>  <tt class="py-line"><tt class="py-docstring">        Arguments:</tt> </tt>
<a name="L838"></a><tt class="py-lineno">838</tt>  <tt class="py-line"><tt class="py-docstring">         - kwargs - A single key-value pair where the key is one</tt> </tt>
<a name="L839"></a><tt class="py-lineno">839</tt>  <tt class="py-line"><tt class="py-docstring">           of primary_id, gi, display_id, name, accession, version</tt> </tt>
<a name="L840"></a><tt class="py-lineno">840</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L841"></a><tt class="py-lineno">841</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L842"></a><tt class="py-lineno">842</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">kwargs</tt><tt class="py-op">)</tt> <tt class="py-op">!=</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L843"></a><tt class="py-lineno">843</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"single key/value parameter expected"</tt><tt class="py-op">)</tt> </tt>
<a name="L844"></a><tt class="py-lineno">844</tt>  <tt class="py-line">        <tt class="py-name">k</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt> <tt class="py-op">=</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt class="py-name">kwargs</tt><tt class="py-op">.</tt><tt id="link-193" class="py-name" targets="Method Bio.Crystal.Crystal.items()=Bio.Crystal.Crystal-class.html#items,Method Bio.File._IndexedSeqFileDict.items()=Bio.File._IndexedSeqFileDict-class.html#items,Method Bio.Phylo.PhyloXML.Events.items()=Bio.Phylo.PhyloXML.Events-class.html#items,Variable Bio.SearchIO._model.query.QueryResult.items=Bio.SearchIO._model.query.QueryResult-class.html#items,Method BioSQL.BioSeqDatabase.BioSeqDatabase.items()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#items,Method BioSQL.BioSeqDatabase.DBServer.items()=BioSQL.BioSeqDatabase.DBServer-class.html#items"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-193', 'items', 'link-193');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L845"></a><tt class="py-lineno">845</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">k</tt> <tt class="py-keyword">not</tt> <tt class="py-keyword">in</tt> <tt id="link-194" class="py-name"><a title="BioSQL.BioSeqDatabase._allowed_lookups" class="py-name" href="#" onclick="return doclink('link-194', '_allowed_lookups', 'link-161');">_allowed_lookups</a></tt><tt class="py-op">:</tt> </tt>
<a name="L846"></a><tt class="py-lineno">846</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">TypeError</tt><tt class="py-op">(</tt><tt class="py-string">"lookup() expects one of %r, not %r"</tt> <tt class="py-op">%</tt> </tt>
<a name="L847"></a><tt class="py-lineno">847</tt>  <tt class="py-line">                            <tt class="py-op">(</tt><tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-195" class="py-name"><a title="BioSQL.BioSeqDatabase._allowed_lookups" class="py-name" href="#" onclick="return doclink('link-195', '_allowed_lookups', 'link-161');">_allowed_lookups</a></tt><tt class="py-op">.</tt><tt id="link-196" class="py-name"><a title="Bio.Crystal.Crystal.keys
Bio.File._IndexedSeqFileDict.keys
Bio.File._SQLiteManySeqFilesDict.keys
Bio.PDB.AbstractPropertyMap.AbstractPropertyMap.keys
Bio.Phylo.PhyloXML.Events.keys
BioSQL.BioSeqDatabase.BioSeqDatabase.keys
BioSQL.BioSeqDatabase.DBServer.keys" class="py-name" href="#" onclick="return doclink('link-196', 'keys', 'link-184');">keys</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">k</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L848"></a><tt class="py-lineno">848</tt>  <tt class="py-line">        <tt class="py-name">lookup_name</tt> <tt class="py-op">=</tt> <tt id="link-197" class="py-name"><a title="BioSQL.BioSeqDatabase._allowed_lookups" class="py-name" href="#" onclick="return doclink('link-197', '_allowed_lookups', 'link-161');">_allowed_lookups</a></tt><tt class="py-op">[</tt><tt class="py-name">k</tt><tt class="py-op">]</tt> </tt>
<a name="L849"></a><tt class="py-lineno">849</tt>  <tt class="py-line">        <tt class="py-name">lookup_func</tt> <tt class="py-op">=</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">,</tt> <tt class="py-name">lookup_name</tt><tt class="py-op">)</tt> </tt>
<a name="L850"></a><tt class="py-lineno">850</tt>  <tt class="py-line">        <tt class="py-name">seqid</tt> <tt class="py-op">=</tt> <tt class="py-name">lookup_func</tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbid</tt><tt class="py-op">,</tt> <tt class="py-name">v</tt><tt class="py-op">)</tt> </tt>
<a name="L851"></a><tt class="py-lineno">851</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt id="link-198" class="py-name"><a title="BioSQL.BioSeq" class="py-name" href="#" onclick="return doclink('link-198', 'BioSeq', 'link-8');">BioSeq</a></tt><tt class="py-op">.</tt><tt id="link-199" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord" class="py-name" href="#" onclick="return doclink('link-199', 'DBSeqRecord', 'link-170');">DBSeqRecord</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">,</tt> <tt class="py-name">seqid</tt><tt class="py-op">)</tt> </tt>
</div><a name="L852"></a><tt class="py-lineno">852</tt>  <tt class="py-line"> </tt>
<a name="BioSeqDatabase.get_Seq_by_primary_id"></a><div id="BioSeqDatabase.get_Seq_by_primary_id-def"><a name="L853"></a><tt class="py-lineno">853</tt> <a class="py-toggle" href="#" id="BioSeqDatabase.get_Seq_by_primary_id-toggle" onclick="return toggle('BioSeqDatabase.get_Seq_by_primary_id');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#get_Seq_by_primary_id">get_Seq_by_primary_id</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">seqid</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BioSeqDatabase.get_Seq_by_primary_id-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BioSeqDatabase.get_Seq_by_primary_id-expanded"><a name="L854"></a><tt class="py-lineno">854</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get a DBSeqRecord by the primary (internal) id (OBSOLETE).</tt> </tt>
<a name="L855"></a><tt class="py-lineno">855</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L856"></a><tt class="py-lineno">856</tt>  <tt class="py-line"><tt class="py-docstring">        Rather than db.get_Seq_by_primary_id(my_id) use db[my_id]</tt> </tt>
<a name="L857"></a><tt class="py-lineno">857</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L858"></a><tt class="py-lineno">858</tt>  <tt class="py-line"><tt class="py-docstring">        The name of this method is misleading since it returns a DBSeqRecord</tt> </tt>
<a name="L859"></a><tt class="py-lineno">859</tt>  <tt class="py-line"><tt class="py-docstring">        rather than a DBSeq ojbect, and presumably was to mirror BioPerl.</tt> </tt>
<a name="L860"></a><tt class="py-lineno">860</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L861"></a><tt class="py-lineno">861</tt>  <tt class="py-line">        <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L862"></a><tt class="py-lineno">862</tt>  <tt class="py-line">        <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"Use bio_seq_database[my_id] instead of "</tt> </tt>
<a name="L863"></a><tt class="py-lineno">863</tt>  <tt class="py-line">                      <tt class="py-string">"bio_seq_database.get_Seq_by_primary_id(my_id)"</tt><tt class="py-op">,</tt> </tt>
<a name="L864"></a><tt class="py-lineno">864</tt>  <tt class="py-line">                      <tt id="link-200" class="py-name"><a title="Bio.BiopythonDeprecationWarning" class="py-name" href="#" onclick="return doclink('link-200', 'BiopythonDeprecationWarning', 'link-7');">BiopythonDeprecationWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L865"></a><tt class="py-lineno">865</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">[</tt><tt class="py-name">seqid</tt><tt class="py-op">]</tt> </tt>
</div><a name="L866"></a><tt class="py-lineno">866</tt>  <tt class="py-line"> </tt>
<a name="BioSeqDatabase.load"></a><div id="BioSeqDatabase.load-def"><a name="L867"></a><tt class="py-lineno">867</tt> <a class="py-toggle" href="#" id="BioSeqDatabase.load-toggle" onclick="return toggle('BioSeqDatabase.load');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#load">load</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record_iterator</tt><tt class="py-op">,</tt> <tt class="py-param">fetch_NCBI_taxonomy</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="BioSeqDatabase.load-collapsed" style="display:none;" pad="+++" indent="++++++++"></div><div id="BioSeqDatabase.load-expanded"><a name="L868"></a><tt class="py-lineno">868</tt>  <tt class="py-line">        <tt class="py-docstring">"""Load a set of SeqRecords into the BioSQL database.</tt> </tt>
<a name="L869"></a><tt class="py-lineno">869</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L870"></a><tt class="py-lineno">870</tt>  <tt class="py-line"><tt class="py-docstring">        record_iterator is either a list of SeqRecord objects, or an</tt> </tt>
<a name="L871"></a><tt class="py-lineno">871</tt>  <tt class="py-line"><tt class="py-docstring">        Iterator object that returns SeqRecord objects (such as the</tt> </tt>
<a name="L872"></a><tt class="py-lineno">872</tt>  <tt class="py-line"><tt class="py-docstring">        output from the Bio.SeqIO.parse() function), which will be</tt> </tt>
<a name="L873"></a><tt class="py-lineno">873</tt>  <tt class="py-line"><tt class="py-docstring">        used to populate the database.</tt> </tt>
<a name="L874"></a><tt class="py-lineno">874</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L875"></a><tt class="py-lineno">875</tt>  <tt class="py-line"><tt class="py-docstring">        fetch_NCBI_taxonomy is boolean flag allowing or preventing</tt> </tt>
<a name="L876"></a><tt class="py-lineno">876</tt>  <tt class="py-line"><tt class="py-docstring">        connection to the taxonomic database on the NCBI server</tt> </tt>
<a name="L877"></a><tt class="py-lineno">877</tt>  <tt class="py-line"><tt class="py-docstring">        (via Bio.Entrez) to fetch a detailed taxonomy for each</tt> </tt>
<a name="L878"></a><tt class="py-lineno">878</tt>  <tt class="py-line"><tt class="py-docstring">        SeqRecord.</tt> </tt>
<a name="L879"></a><tt class="py-lineno">879</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L880"></a><tt class="py-lineno">880</tt>  <tt class="py-line"><tt class="py-docstring">        Example::</tt> </tt>
<a name="L881"></a><tt class="py-lineno">881</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L882"></a><tt class="py-lineno">882</tt>  <tt class="py-line"><tt class="py-docstring">            from Bio import SeqIO</tt> </tt>
<a name="L883"></a><tt class="py-lineno">883</tt>  <tt class="py-line"><tt class="py-docstring">            count = db.load(SeqIO.parse(open(filename), format))</tt> </tt>
<a name="L884"></a><tt class="py-lineno">884</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L885"></a><tt class="py-lineno">885</tt>  <tt class="py-line"><tt class="py-docstring">        Returns the number of records loaded.</tt> </tt>
<a name="L886"></a><tt class="py-lineno">886</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L887"></a><tt class="py-lineno">887</tt>  <tt class="py-line">        <tt class="py-name">db_loader</tt> <tt class="py-op">=</tt> <tt id="link-201" class="py-name"><a title="BioSQL.Loader" class="py-name" href="#" onclick="return doclink('link-201', 'Loader', 'link-9');">Loader</a></tt><tt class="py-op">.</tt><tt id="link-202" class="py-name" targets="Class BioSQL.Loader.DatabaseLoader=BioSQL.Loader.DatabaseLoader-class.html"><a title="BioSQL.Loader.DatabaseLoader" class="py-name" href="#" onclick="return doclink('link-202', 'DatabaseLoader', 'link-202');">DatabaseLoader</a></tt><tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbid</tt><tt class="py-op">,</tt> </tt>
<a name="L888"></a><tt class="py-lineno">888</tt>  <tt class="py-line">                                          <tt class="py-name">fetch_NCBI_taxonomy</tt><tt class="py-op">)</tt> </tt>
<a name="L889"></a><tt class="py-lineno">889</tt>  <tt class="py-line">        <tt class="py-name">num_records</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L890"></a><tt class="py-lineno">890</tt>  <tt class="py-line">        <tt class="py-keyword">global</tt> <tt id="link-203" class="py-name"><a title="BioSQL.BioSeqDatabase._POSTGRES_RULES_PRESENT" class="py-name" href="#" onclick="return doclink('link-203', '_POSTGRES_RULES_PRESENT', 'link-11');">_POSTGRES_RULES_PRESENT</a></tt> </tt>
<a name="L891"></a><tt class="py-lineno">891</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">cur_record</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record_iterator</tt><tt class="py-op">:</tt> </tt>
<a name="L892"></a><tt class="py-lineno">892</tt>  <tt class="py-line">            <tt class="py-name">num_records</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L893"></a><tt class="py-lineno">893</tt>  <tt class="py-line">            <tt class="py-comment"># Hack to work around BioSQL Bug 2839 - If using PostgreSQL and</tt> </tt>
<a name="L894"></a><tt class="py-lineno">894</tt>  <tt class="py-line">            <tt class="py-comment"># the RULES are present check for a duplicate record before loading</tt> </tt>
<a name="L895"></a><tt class="py-lineno">895</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-204" class="py-name"><a title="BioSQL.BioSeqDatabase._POSTGRES_RULES_PRESENT" class="py-name" href="#" onclick="return doclink('link-204', '_POSTGRES_RULES_PRESENT', 'link-11');">_POSTGRES_RULES_PRESENT</a></tt><tt class="py-op">:</tt> </tt>
<a name="L896"></a><tt class="py-lineno">896</tt>  <tt class="py-line">                <tt class="py-comment"># Recreate what the Loader's _load_bioentry_table will do:</tt> </tt>
<a name="L897"></a><tt class="py-lineno">897</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">cur_record</tt><tt class="py-op">.</tt><tt id="link-205" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.id=Bio.KEGG.KGML.KGML_pathway.Component-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Entry.id=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.id=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#id,Variable Bio.PDB.Entity.Entity.id=Bio.PDB.Entity.Entity-class.html#id,Method Bio.Phylo.PhyloXMLIO.Parser.id()=Bio.Phylo.PhyloXMLIO.Parser-class.html#id,Method Bio.Phylo.PhyloXMLIO.Writer.id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#id,Variable Bio.SearchIO._model.hit.Hit.id=Bio.SearchIO._model.hit.Hit-class.html#id,Variable Bio.SearchIO._model.query.QueryResult.id=Bio.SearchIO._model.query.QueryResult-class.html#id"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-205', 'id', 'link-205');">id</a></tt><tt class="py-op">.</tt><tt id="link-206" class="py-name" targets="Method Bio.Crystal.Chain.count()=Bio.Crystal.Chain-class.html#count,Method Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count()=Bio.NeuralNetwork.Gene.Pattern.PatternRepository-class.html#count,Method Bio.Seq.MutableSeq.count()=Bio.Seq.MutableSeq-class.html#count,Method Bio.Seq.Seq.count()=Bio.Seq.Seq-class.html#count,Method Bio.Seq.UnknownSeq.count()=Bio.Seq.UnknownSeq-class.html#count,Method Bio.motifs.Instances.count()=Bio.motifs.Instances-class.html#count"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-206', 'count', 'link-206');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">"."</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L898"></a><tt class="py-lineno">898</tt>  <tt class="py-line">                    <tt id="link-207" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.accession()=Bio.GenBank._FeatureConsumer-class.html#accession,Method Bio.GenBank._RecordConsumer.accession()=Bio.GenBank._RecordConsumer-class.html#accession,Method Bio.Phylo.PhyloXMLIO.Parser.accession()=Bio.Phylo.PhyloXMLIO.Parser-class.html#accession,Method Bio.Phylo.PhyloXMLIO.Writer.accession()=Bio.Phylo.PhyloXMLIO.Writer-class.html#accession"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-207', 'accession', 'link-207');">accession</a></tt><tt class="py-op">,</tt> <tt id="link-208" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-208', 'version', 'link-114');">version</a></tt> <tt class="py-op">=</tt> <tt class="py-name">cur_record</tt><tt class="py-op">.</tt><tt id="link-209" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-209', 'id', 'link-205');">id</a></tt><tt class="py-op">.</tt><tt id="link-210" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-210', 'split', 'link-61');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">'.'</tt><tt class="py-op">)</tt> </tt>
<a name="L899"></a><tt class="py-lineno">899</tt>  <tt class="py-line">                    <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L900"></a><tt class="py-lineno">900</tt>  <tt class="py-line">                        <tt id="link-211" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-211', 'version', 'link-114');">version</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-212" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-212', 'version', 'link-114');">version</a></tt><tt class="py-op">)</tt> </tt>
<a name="L901"></a><tt class="py-lineno">901</tt>  <tt class="py-line">                    <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L902"></a><tt class="py-lineno">902</tt>  <tt class="py-line">                        <tt id="link-213" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-213', 'accession', 'link-207');">accession</a></tt> <tt class="py-op">=</tt> <tt class="py-name">cur_record</tt><tt class="py-op">.</tt><tt id="link-214" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-214', 'id', 'link-205');">id</a></tt> </tt>
<a name="L903"></a><tt class="py-lineno">903</tt>  <tt class="py-line">                        <tt id="link-215" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-215', 'version', 'link-114');">version</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L904"></a><tt class="py-lineno">904</tt>  <tt class="py-line">                <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L905"></a><tt class="py-lineno">905</tt>  <tt class="py-line">                    <tt id="link-216" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-216', 'accession', 'link-207');">accession</a></tt> <tt class="py-op">=</tt> <tt class="py-name">cur_record</tt><tt class="py-op">.</tt><tt id="link-217" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.PDB.Entity.Entity.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-217', 'id', 'link-205');">id</a></tt> </tt>
<a name="L906"></a><tt class="py-lineno">906</tt>  <tt class="py-line">                    <tt id="link-218" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-218', 'version', 'link-114');">version</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L907"></a><tt class="py-lineno">907</tt>  <tt class="py-line">                <tt id="link-219" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.gi()=Bio.GenBank._FeatureConsumer-class.html#gi,Method Bio.GenBank._RecordConsumer.gi()=Bio.GenBank._RecordConsumer-class.html#gi"><a title="Bio.GenBank._FeatureConsumer.gi
Bio.GenBank._RecordConsumer.gi" class="py-name" href="#" onclick="return doclink('link-219', 'gi', 'link-219');">gi</a></tt> <tt class="py-op">=</tt> <tt class="py-name">cur_record</tt><tt class="py-op">.</tt><tt id="link-220" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.annotations=BioSQL.BioSeq.DBSeqRecord-class.html#annotations"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-220', 'annotations', 'link-220');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-221" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-221', 'get', 'link-15');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"gi"</tt><tt class="py-op">)</tt> </tt>
<a name="L908"></a><tt class="py-lineno">908</tt>  <tt class="py-line">                <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">"SELECT bioentry_id FROM bioentry WHERE (identifier "</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L909"></a><tt class="py-lineno">909</tt>  <tt class="py-line">                      <tt class="py-string">"= '%s' AND biodatabase_id = '%s') OR (accession = "</tt> <tt class="py-op">+</tt> \ </tt>
<a name="L910"></a><tt class="py-lineno">910</tt>  <tt class="py-line">                      <tt class="py-string">"'%s' AND version = '%s' AND biodatabase_id = '%s')"</tt> </tt>
<a name="L911"></a><tt class="py-lineno">911</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-222" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-222', 'execute', 'link-28');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L912"></a><tt class="py-lineno">912</tt>  <tt class="py-line">                    <tt class="py-name">sql</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt id="link-223" class="py-name"><a title="Bio.GenBank._FeatureConsumer.gi
Bio.GenBank._RecordConsumer.gi" class="py-name" href="#" onclick="return doclink('link-223', 'gi', 'link-219');">gi</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbid</tt><tt class="py-op">,</tt> <tt id="link-224" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-224', 'accession', 'link-207');">accession</a></tt><tt class="py-op">,</tt> <tt id="link-225" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-225', 'version', 'link-114');">version</a></tt><tt class="py-op">,</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbid</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L913"></a><tt class="py-lineno">913</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-226" class="py-name"><a title="Bio.DocSQL.Query.cursor
Bio.DocSQL.QuerySingle.cursor" class="py-name" href="#" onclick="return doclink('link-226', 'cursor', 'link-59');">cursor</a></tt><tt class="py-op">.</tt><tt id="link-227" class="py-name"><a title="BioSQL.BioSeqDatabase._CursorWrapper.fetchone" class="py-name" href="#" onclick="return doclink('link-227', 'fetchone', 'link-76');">fetchone</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L914"></a><tt class="py-lineno">914</tt>  <tt class="py-line">                    <tt class="py-keyword">raise</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt class="py-name">conn</tt><tt class="py-op">.</tt><tt class="py-name">IntegrityError</tt><tt class="py-op">(</tt> </tt>
<a name="L915"></a><tt class="py-lineno">915</tt>  <tt class="py-line">                        <tt class="py-string">"Duplicate record detected: "</tt> </tt>
<a name="L916"></a><tt class="py-lineno">916</tt>  <tt class="py-line">                        <tt class="py-string">"record has not been inserted"</tt><tt class="py-op">)</tt> </tt>
<a name="L917"></a><tt class="py-lineno">917</tt>  <tt class="py-line">            <tt class="py-comment"># End of hack</tt> </tt>
<a name="L918"></a><tt class="py-lineno">918</tt>  <tt class="py-line">            <tt class="py-name">db_loader</tt><tt class="py-op">.</tt><tt id="link-228" class="py-name" targets="Method BioSQL.Loader.DatabaseLoader.load_seqrecord()=BioSQL.Loader.DatabaseLoader-class.html#load_seqrecord"><a title="BioSQL.Loader.DatabaseLoader.load_seqrecord" class="py-name" href="#" onclick="return doclink('link-228', 'load_seqrecord', 'link-228');">load_seqrecord</a></tt><tt class="py-op">(</tt><tt class="py-name">cur_record</tt><tt class="py-op">)</tt> </tt>
<a name="L919"></a><tt class="py-lineno">919</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">num_records</tt> </tt>
</div></div><a name="L920"></a><tt class="py-lineno">920</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Apr  4 10:49:45 2018
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
