<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>BioSQL.Loader</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="BioSQL-module.html">Package&nbsp;BioSQL</a> ::
        Module&nbsp;Loader
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="BioSQL.Loader-pysrc.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<h1 class="epydoc">Source Code for <a href="BioSQL.Loader-module.html">Module BioSQL.Loader</a></h1>
<pre class="py-src">
<a name="L1"></a><tt class="py-lineno">   1</tt>  <tt class="py-line"><tt class="py-comment"># Copyright 2002 by Andrew Dalke.  All rights reserved.</tt> </tt>
<a name="L2"></a><tt class="py-lineno">   2</tt>  <tt class="py-line"><tt class="py-comment"># Revisions 2007-2009 copyright by Peter Cock.  All rights reserved.</tt> </tt>
<a name="L3"></a><tt class="py-lineno">   3</tt>  <tt class="py-line"><tt class="py-comment"># Revisions 2008 copyright by Cymon J. Cox.  All rights reserved.</tt> </tt>
<a name="L4"></a><tt class="py-lineno">   4</tt>  <tt class="py-line"><tt class="py-comment"># This code is part of the Biopython distribution and governed by its</tt> </tt>
<a name="L5"></a><tt class="py-lineno">   5</tt>  <tt class="py-line"><tt class="py-comment"># license.  Please see the LICENSE file that should have been included</tt> </tt>
<a name="L6"></a><tt class="py-lineno">   6</tt>  <tt class="py-line"><tt class="py-comment"># as part of this package.</tt> </tt>
<a name="L7"></a><tt class="py-lineno">   7</tt>  <tt class="py-line"><tt class="py-comment">#</tt> </tt>
<a name="L8"></a><tt class="py-lineno">   8</tt>  <tt class="py-line"><tt class="py-comment"># Note that BioSQL (including the database schema and scripts) is</tt> </tt>
<a name="L9"></a><tt class="py-lineno">   9</tt>  <tt class="py-line"><tt class="py-comment"># available and licensed separately.  Please consult www.biosql.org</tt> </tt>
<a name="L10"></a><tt class="py-lineno">  10</tt>  <tt class="py-line"> </tt>
<a name="L11"></a><tt class="py-lineno">  11</tt>  <tt class="py-line"><tt class="py-docstring">"""Load biopython objects into a BioSQL database for persistent storage.</tt> </tt>
<a name="L12"></a><tt class="py-lineno">  12</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L13"></a><tt class="py-lineno">  13</tt>  <tt class="py-line"><tt class="py-docstring">This code makes it possible to store biopython objects in a relational</tt> </tt>
<a name="L14"></a><tt class="py-lineno">  14</tt>  <tt class="py-line"><tt class="py-docstring">database and then retrieve them back. You shouldn't use any of the</tt> </tt>
<a name="L15"></a><tt class="py-lineno">  15</tt>  <tt class="py-line"><tt class="py-docstring">classes in this module directly. Rather, call the load() method on</tt> </tt>
<a name="L16"></a><tt class="py-lineno">  16</tt>  <tt class="py-line"><tt class="py-docstring">a database object.</tt> </tt>
<a name="L17"></a><tt class="py-lineno">  17</tt>  <tt class="py-line"><tt class="py-docstring">"""</tt> </tt>
<a name="L18"></a><tt class="py-lineno">  18</tt>  <tt class="py-line"><tt class="py-comment"># standard modules</tt> </tt>
<a name="L19"></a><tt class="py-lineno">  19</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">__future__</tt> <tt class="py-keyword">import</tt> <tt class="py-name">print_function</tt> </tt>
<a name="L20"></a><tt class="py-lineno">  20</tt>  <tt class="py-line"> </tt>
<a name="L21"></a><tt class="py-lineno">  21</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt class="py-name">time</tt> <tt class="py-keyword">import</tt> <tt class="py-name">gmtime</tt><tt class="py-op">,</tt> <tt class="py-name">strftime</tt> </tt>
<a name="L22"></a><tt class="py-lineno">  22</tt>  <tt class="py-line"> </tt>
<a name="L23"></a><tt class="py-lineno">  23</tt>  <tt class="py-line"><tt class="py-comment"># biopython</tt> </tt>
<a name="L24"></a><tt class="py-lineno">  24</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-0" class="py-name" targets="Package Bio=Bio-module.html"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-0', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-1" class="py-name" targets="Package Bio.Alphabet=Bio.Alphabet-module.html,Class Bio.Alphabet.Alphabet=Bio.Alphabet.Alphabet-class.html"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-1', 'Alphabet', 'link-1');">Alphabet</a></tt> </tt>
<a name="L25"></a><tt class="py-lineno">  25</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-2" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-2', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-3" class="py-name" targets="Package Bio.SeqUtils=Bio.SeqUtils-module.html"><a title="Bio.SeqUtils" class="py-name" href="#" onclick="return doclink('link-3', 'SeqUtils', 'link-3');">SeqUtils</a></tt><tt class="py-op">.</tt><tt id="link-4" class="py-name" targets="Module Bio.SeqUtils.CheckSum=Bio.SeqUtils.CheckSum-module.html"><a title="Bio.SeqUtils.CheckSum" class="py-name" href="#" onclick="return doclink('link-4', 'CheckSum', 'link-4');">CheckSum</a></tt> <tt class="py-keyword">import</tt> <tt id="link-5" class="py-name" targets="Function Bio.SeqUtils.CheckSum.crc64()=Bio.SeqUtils.CheckSum-module.html#crc64"><a title="Bio.SeqUtils.CheckSum.crc64" class="py-name" href="#" onclick="return doclink('link-5', 'crc64', 'link-5');">crc64</a></tt> </tt>
<a name="L26"></a><tt class="py-lineno">  26</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-6" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-6', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-7" class="py-name" targets="Package Bio.Entrez=Bio.Entrez-module.html"><a title="Bio.Entrez" class="py-name" href="#" onclick="return doclink('link-7', 'Entrez', 'link-7');">Entrez</a></tt> </tt>
<a name="L27"></a><tt class="py-lineno">  27</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-8" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-8', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-9" class="py-name" targets="Module Bio.Seq=Bio.Seq-module.html,Class Bio.Seq.Seq=Bio.Seq.Seq-class.html"><a title="Bio.Seq
Bio.Seq.Seq" class="py-name" href="#" onclick="return doclink('link-9', 'Seq', 'link-9');">Seq</a></tt> <tt class="py-keyword">import</tt> <tt id="link-10" class="py-name" targets="Class Bio.Seq.UnknownSeq=Bio.Seq.UnknownSeq-class.html"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-10', 'UnknownSeq', 'link-10');">UnknownSeq</a></tt> </tt>
<a name="L28"></a><tt class="py-lineno">  28</tt>  <tt class="py-line"> </tt>
<a name="L29"></a><tt class="py-lineno">  29</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-11" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-11', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-12" class="py-name" targets="Package Bio._py3k=Bio._py3k-module.html"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-12', '_py3k', 'link-12');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt id="link-13" class="py-name" targets="Function Bio._py3k._is_int_or_long()=Bio._py3k-module.html#_is_int_or_long"><a title="Bio._py3k._is_int_or_long" class="py-name" href="#" onclick="return doclink('link-13', '_is_int_or_long', 'link-13');">_is_int_or_long</a></tt> </tt>
<a name="L30"></a><tt class="py-lineno">  30</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-14" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-14', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-15" class="py-name"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-15', '_py3k', 'link-12');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt id="link-16" class="py-name" targets="Method Bio.Graphics.GenomeDiagram._Diagram.Diagram.range()=Bio.Graphics.GenomeDiagram._Diagram.Diagram-class.html#range,Method Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range()=Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Graph.GraphData.range()=Bio.Graphics.GenomeDiagram._Graph.GraphData-class.html#range,Method Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range()=Bio.Graphics.GenomeDiagram._GraphSet.GraphSet-class.html#range,Method Bio.Graphics.GenomeDiagram._Track.Track.range()=Bio.Graphics.GenomeDiagram._Track.Track-class.html#range"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-16', 'range', 'link-16');">range</a></tt> </tt>
<a name="L31"></a><tt class="py-lineno">  31</tt>  <tt class="py-line"><tt class="py-keyword">from</tt> <tt id="link-17" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-17', 'Bio', 'link-0');">Bio</a></tt><tt class="py-op">.</tt><tt id="link-18" class="py-name"><a title="Bio._py3k" class="py-name" href="#" onclick="return doclink('link-18', '_py3k', 'link-12');">_py3k</a></tt> <tt class="py-keyword">import</tt> <tt class="py-name">basestring</tt> </tt>
<a name="L32"></a><tt class="py-lineno">  32</tt>  <tt class="py-line"> </tt>
<a name="L33"></a><tt class="py-lineno">  33</tt>  <tt class="py-line"><tt class="py-name">__docformat__</tt> <tt class="py-op">=</tt> <tt class="py-string">"restructuredtext en"</tt> </tt>
<a name="L34"></a><tt class="py-lineno">  34</tt>  <tt class="py-line"> </tt>
<a name="L35"></a><tt class="py-lineno">  35</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader"></a><div id="DatabaseLoader-def"><a name="L36"></a><tt class="py-lineno">  36</tt> <a class="py-toggle" href="#" id="DatabaseLoader-toggle" onclick="return toggle('DatabaseLoader');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html">DatabaseLoader</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="DatabaseLoader-expanded"><a name="L37"></a><tt class="py-lineno">  37</tt>  <tt class="py-line">    <tt class="py-docstring">"""Object used to load SeqRecord objects into a BioSQL database."""</tt> </tt>
<a name="L38"></a><tt class="py-lineno">  38</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader.__init__"></a><div id="DatabaseLoader.__init__-def"><a name="L39"></a><tt class="py-lineno">  39</tt> <a class="py-toggle" href="#" id="DatabaseLoader.__init__-toggle" onclick="return toggle('DatabaseLoader.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">adaptor</tt><tt class="py-op">,</tt> <tt class="py-param">dbid</tt><tt class="py-op">,</tt> <tt class="py-param">fetch_NCBI_taxonomy</tt><tt class="py-op">=</tt><tt class="py-name">False</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader.__init__-expanded"><a name="L40"></a><tt class="py-lineno">  40</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize with connection information for the database.</tt> </tt>
<a name="L41"></a><tt class="py-lineno">  41</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L42"></a><tt class="py-lineno">  42</tt>  <tt class="py-line"><tt class="py-docstring">        Creating a DatabaseLoader object is normally handled via the</tt> </tt>
<a name="L43"></a><tt class="py-lineno">  43</tt>  <tt class="py-line"><tt class="py-docstring">        BioSeqDatabase DBServer object, for example::</tt> </tt>
<a name="L44"></a><tt class="py-lineno">  44</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L45"></a><tt class="py-lineno">  45</tt>  <tt class="py-line"><tt class="py-docstring">            from BioSQL import BioSeqDatabase</tt> </tt>
<a name="L46"></a><tt class="py-lineno">  46</tt>  <tt class="py-line"><tt class="py-docstring">            server = BioSeqDatabase.open_database(driver="MySQLdb", user="gbrowse",</tt> </tt>
<a name="L47"></a><tt class="py-lineno">  47</tt>  <tt class="py-line"><tt class="py-docstring">                             passwd = "biosql", host = "localhost", db="test_biosql")</tt> </tt>
<a name="L48"></a><tt class="py-lineno">  48</tt>  <tt class="py-line"><tt class="py-docstring">            try:</tt> </tt>
<a name="L49"></a><tt class="py-lineno">  49</tt>  <tt class="py-line"><tt class="py-docstring">                db = server["test"]</tt> </tt>
<a name="L50"></a><tt class="py-lineno">  50</tt>  <tt class="py-line"><tt class="py-docstring">            except KeyError:</tt> </tt>
<a name="L51"></a><tt class="py-lineno">  51</tt>  <tt class="py-line"><tt class="py-docstring">                db = server.new_database("test", description="For testing GBrowse")</tt> </tt>
<a name="L52"></a><tt class="py-lineno">  52</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L53"></a><tt class="py-lineno">  53</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt> <tt class="py-op">=</tt> <tt class="py-name">adaptor</tt> </tt>
<a name="L54"></a><tt class="py-lineno">  54</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbid</tt> <tt class="py-op">=</tt> <tt class="py-name">dbid</tt> </tt>
<a name="L55"></a><tt class="py-lineno">  55</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">fetch_NCBI_taxonomy</tt> <tt class="py-op">=</tt> <tt class="py-name">fetch_NCBI_taxonomy</tt> </tt>
</div><a name="L56"></a><tt class="py-lineno">  56</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader.load_seqrecord"></a><div id="DatabaseLoader.load_seqrecord-def"><a name="L57"></a><tt class="py-lineno">  57</tt> <a class="py-toggle" href="#" id="DatabaseLoader.load_seqrecord-toggle" onclick="return toggle('DatabaseLoader.load_seqrecord');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#load_seqrecord">load_seqrecord</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader.load_seqrecord-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader.load_seqrecord-expanded"><a name="L58"></a><tt class="py-lineno">  58</tt>  <tt class="py-line">        <tt class="py-docstring">"""Load a Biopython SeqRecord into the database."""</tt> </tt>
<a name="L59"></a><tt class="py-lineno">  59</tt>  <tt class="py-line">        <tt class="py-name">bioentry_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-19" class="py-name" targets="Method BioSQL.Loader.DatabaseLoader._load_bioentry_table()=BioSQL.Loader.DatabaseLoader-class.html#_load_bioentry_table"><a title="BioSQL.Loader.DatabaseLoader._load_bioentry_table" class="py-name" href="#" onclick="return doclink('link-19', '_load_bioentry_table', 'link-19');">_load_bioentry_table</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L60"></a><tt class="py-lineno">  60</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-20" class="py-name" targets="Method BioSQL.Loader.DatabaseLoader._load_bioentry_date()=BioSQL.Loader.DatabaseLoader-class.html#_load_bioentry_date"><a title="BioSQL.Loader.DatabaseLoader._load_bioentry_date" class="py-name" href="#" onclick="return doclink('link-20', '_load_bioentry_date', 'link-20');">_load_bioentry_date</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">,</tt> <tt class="py-name">bioentry_id</tt><tt class="py-op">)</tt> </tt>
<a name="L61"></a><tt class="py-lineno">  61</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-21" class="py-name" targets="Method BioSQL.Loader.DatabaseLoader._load_biosequence()=BioSQL.Loader.DatabaseLoader-class.html#_load_biosequence"><a title="BioSQL.Loader.DatabaseLoader._load_biosequence" class="py-name" href="#" onclick="return doclink('link-21', '_load_biosequence', 'link-21');">_load_biosequence</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">,</tt> <tt class="py-name">bioentry_id</tt><tt class="py-op">)</tt> </tt>
<a name="L62"></a><tt class="py-lineno">  62</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-22" class="py-name" targets="Method BioSQL.Loader.DatabaseLoader._load_comment()=BioSQL.Loader.DatabaseLoader-class.html#_load_comment"><a title="BioSQL.Loader.DatabaseLoader._load_comment" class="py-name" href="#" onclick="return doclink('link-22', '_load_comment', 'link-22');">_load_comment</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">,</tt> <tt class="py-name">bioentry_id</tt><tt class="py-op">)</tt> </tt>
<a name="L63"></a><tt class="py-lineno">  63</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-23" class="py-name" targets="Method BioSQL.Loader.DatabaseLoader._load_dbxrefs()=BioSQL.Loader.DatabaseLoader-class.html#_load_dbxrefs"><a title="BioSQL.Loader.DatabaseLoader._load_dbxrefs" class="py-name" href="#" onclick="return doclink('link-23', '_load_dbxrefs', 'link-23');">_load_dbxrefs</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">,</tt> <tt class="py-name">bioentry_id</tt><tt class="py-op">)</tt> </tt>
<a name="L64"></a><tt class="py-lineno">  64</tt>  <tt class="py-line">        <tt class="py-name">references</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-24" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.annotations=BioSQL.BioSeq.DBSeqRecord-class.html#annotations"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-24', 'annotations', 'link-24');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-25" class="py-name" targets="Method Bio.Crystal.Crystal.get()=Bio.Crystal.Crystal-class.html#get,Method Bio.Data.CodonTable.AmbiguousForwardTable.get()=Bio.Data.CodonTable.AmbiguousForwardTable-class.html#get,Method Bio.File._IndexedSeqFileDict.get()=Bio.File._IndexedSeqFileDict-class.html#get,Method Bio.File._IndexedSeqFileProxy.get()=Bio.File._IndexedSeqFileProxy-class.html#get,Method Bio.File._SQLiteManySeqFilesDict.get()=Bio.File._SQLiteManySeqFilesDict-class.html#get,Method Bio.Restriction.Restriction.RestrictionBatch.get()=Bio.Restriction.Restriction.RestrictionBatch-class.html#get,Method Bio.SearchIO._index.SearchIndexer.get()=Bio.SearchIO._index.SearchIndexer-class.html#get,Method Bio.SeqIO._index.SeqFileRandomAccess.get()=Bio.SeqIO._index.SeqFileRandomAccess-class.html#get,Method Bio.SeqIO._index.SffRandomAccess.get()=Bio.SeqIO._index.SffRandomAccess-class.html#get,Method Bio.SeqIO._index.SffTrimedRandomAccess.get()=Bio.SeqIO._index.SffTrimedRandomAccess-class.html#get,Method Bio.SeqIO._index.UniprotRandomAccess.get()=Bio.SeqIO._index.UniprotRandomAccess-class.html#get"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-25', 'get', 'link-25');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'references'</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L65"></a><tt class="py-lineno">  65</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-26" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HeaderConsumer.reference()=Bio.Blast.NCBIStandalone._HeaderConsumer-class.html#reference,Method Bio.Phylo.PhyloXMLIO.Parser.reference()=Bio.Phylo.PhyloXMLIO.Parser-class.html#reference,Method Bio.Phylo.PhyloXMLIO.Writer.reference()=Bio.Phylo.PhyloXMLIO.Writer-class.html#reference"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-26', 'reference', 'link-26');">reference</a></tt><tt class="py-op">,</tt> <tt id="link-27" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.rank()=Bio.Phylo.PhyloXMLIO.Writer-class.html#rank"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-27', 'rank', 'link-27');">rank</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">zip</tt><tt class="py-op">(</tt><tt class="py-name">references</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">(</tt><tt id="link-28" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-28', 'range', 'link-16');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">references</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L66"></a><tt class="py-lineno">  66</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-29" class="py-name" targets="Method BioSQL.Loader.DatabaseLoader._load_reference()=BioSQL.Loader.DatabaseLoader-class.html#_load_reference"><a title="BioSQL.Loader.DatabaseLoader._load_reference" class="py-name" href="#" onclick="return doclink('link-29', '_load_reference', 'link-29');">_load_reference</a></tt><tt class="py-op">(</tt><tt id="link-30" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-30', 'reference', 'link-26');">reference</a></tt><tt class="py-op">,</tt> <tt id="link-31" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-31', 'rank', 'link-27');">rank</a></tt><tt class="py-op">,</tt> <tt class="py-name">bioentry_id</tt><tt class="py-op">)</tt> </tt>
<a name="L67"></a><tt class="py-lineno">  67</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-32" class="py-name" targets="Method BioSQL.Loader.DatabaseLoader._load_annotations()=BioSQL.Loader.DatabaseLoader-class.html#_load_annotations"><a title="BioSQL.Loader.DatabaseLoader._load_annotations" class="py-name" href="#" onclick="return doclink('link-32', '_load_annotations', 'link-32');">_load_annotations</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">,</tt> <tt class="py-name">bioentry_id</tt><tt class="py-op">)</tt> </tt>
<a name="L68"></a><tt class="py-lineno">  68</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">seq_feature_num</tt> <tt class="py-keyword">in</tt> <tt id="link-33" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-33', 'range', 'link-16');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-34" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.features=BioSQL.BioSeq.DBSeqRecord-class.html#features"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-34', 'features', 'link-34');">features</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L69"></a><tt class="py-lineno">  69</tt>  <tt class="py-line">            <tt class="py-name">seq_feature</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-35" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-35', 'features', 'link-34');">features</a></tt><tt class="py-op">[</tt><tt class="py-name">seq_feature_num</tt><tt class="py-op">]</tt> </tt>
<a name="L70"></a><tt class="py-lineno">  70</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-36" class="py-name" targets="Method BioSQL.Loader.DatabaseLoader._load_seqfeature()=BioSQL.Loader.DatabaseLoader-class.html#_load_seqfeature"><a title="BioSQL.Loader.DatabaseLoader._load_seqfeature" class="py-name" href="#" onclick="return doclink('link-36', '_load_seqfeature', 'link-36');">_load_seqfeature</a></tt><tt class="py-op">(</tt><tt class="py-name">seq_feature</tt><tt class="py-op">,</tt> <tt class="py-name">seq_feature_num</tt><tt class="py-op">,</tt> <tt class="py-name">bioentry_id</tt><tt class="py-op">)</tt> </tt>
</div><a name="L71"></a><tt class="py-lineno">  71</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader._get_ontology_id"></a><div id="DatabaseLoader._get_ontology_id-def"><a name="L72"></a><tt class="py-lineno">  72</tt> <a class="py-toggle" href="#" id="DatabaseLoader._get_ontology_id-toggle" onclick="return toggle('DatabaseLoader._get_ontology_id');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#_get_ontology_id">_get_ontology_id</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">name</tt><tt class="py-op">,</tt> <tt class="py-param">definition</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader._get_ontology_id-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader._get_ontology_id-expanded"><a name="L73"></a><tt class="py-lineno">  73</tt>  <tt class="py-line">        <tt class="py-docstring">"""Returns the identifier for the named ontology (PRIVATE).</tt> </tt>
<a name="L74"></a><tt class="py-lineno">  74</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L75"></a><tt class="py-lineno">  75</tt>  <tt class="py-line"><tt class="py-docstring">        This looks through the onotology table for a the given entry name.</tt> </tt>
<a name="L76"></a><tt class="py-lineno">  76</tt>  <tt class="py-line"><tt class="py-docstring">        If it is not found, a row is added for this ontology (using the</tt> </tt>
<a name="L77"></a><tt class="py-lineno">  77</tt>  <tt class="py-line"><tt class="py-docstring">        definition if supplied).  In either case, the id corresponding to</tt> </tt>
<a name="L78"></a><tt class="py-lineno">  78</tt>  <tt class="py-line"><tt class="py-docstring">        the provided name is returned, so that you can reference it in</tt> </tt>
<a name="L79"></a><tt class="py-lineno">  79</tt>  <tt class="py-line"><tt class="py-docstring">        another table.</tt> </tt>
<a name="L80"></a><tt class="py-lineno">  80</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L81"></a><tt class="py-lineno">  81</tt>  <tt class="py-line">        <tt class="py-name">oids</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-37" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0()=BioSQL.BioSeqDatabase.Adaptor-class.html#execute_and_fetch_col0,Method BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0()=BioSQL.BioSeqDatabase.MysqlConnectorAdaptor-class.html#execute_and_fetch_col0"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0" class="py-name" href="#" onclick="return doclink('link-37', 'execute_and_fetch_col0', 'link-37');">execute_and_fetch_col0</a></tt><tt class="py-op">(</tt> </tt>
<a name="L82"></a><tt class="py-lineno">  82</tt>  <tt class="py-line">            <tt class="py-string">"SELECT ontology_id FROM ontology WHERE name = %s"</tt><tt class="py-op">,</tt> </tt>
<a name="L83"></a><tt class="py-lineno">  83</tt>  <tt class="py-line">            <tt class="py-op">(</tt><tt id="link-38" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Entry.name=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Pathway.name=Bio.KEGG.KGML.KGML_pathway.Pathway-class.html#name,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.name=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#name,Method Bio.Phylo.PhyloXMLIO.Writer.name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#name,Variable Bio.codonalign.codonalphabet.CodonAlphabet.name=Bio.codonalign.codonalphabet.CodonAlphabet-class.html#name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-38', 'name', 'link-38');">name</a></tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L84"></a><tt class="py-lineno">  84</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">oids</tt><tt class="py-op">:</tt> </tt>
<a name="L85"></a><tt class="py-lineno">  85</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">oids</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L86"></a><tt class="py-lineno">  86</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-39" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.execute()=BioSQL.BioSeqDatabase.Adaptor-class.html#execute,Method BioSQL.BioSeqDatabase._CursorWrapper.execute()=BioSQL.BioSeqDatabase._CursorWrapper-class.html#execute,Method BioSQL.DBUtils.Generic_dbutils.execute()=BioSQL.DBUtils.Generic_dbutils-class.html#execute,Method BioSQL.DBUtils.Sqlite_dbutils.execute()=BioSQL.DBUtils.Sqlite_dbutils-class.html#execute"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-39', 'execute', 'link-39');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L87"></a><tt class="py-lineno">  87</tt>  <tt class="py-line">            <tt class="py-string">"INSERT INTO ontology(name, definition) VALUES (%s, %s)"</tt><tt class="py-op">,</tt> </tt>
<a name="L88"></a><tt class="py-lineno">  88</tt>  <tt class="py-line">            <tt class="py-op">(</tt><tt id="link-40" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-40', 'name', 'link-38');">name</a></tt><tt class="py-op">,</tt> <tt id="link-41" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.definition()=Bio.GenBank._FeatureConsumer-class.html#definition,Method Bio.GenBank._RecordConsumer.definition()=Bio.GenBank._RecordConsumer-class.html#definition"><a title="Bio.GenBank._FeatureConsumer.definition
Bio.GenBank._RecordConsumer.definition" class="py-name" href="#" onclick="return doclink('link-41', 'definition', 'link-41');">definition</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L89"></a><tt class="py-lineno">  89</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-42" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.last_id()=BioSQL.BioSeqDatabase.Adaptor-class.html#last_id,Method BioSQL.DBUtils.Generic_dbutils.last_id()=BioSQL.DBUtils.Generic_dbutils-class.html#last_id,Method BioSQL.DBUtils.Mysql_dbutils.last_id()=BioSQL.DBUtils.Mysql_dbutils-class.html#last_id,Method BioSQL.DBUtils._PostgreSQL_dbutils.last_id()=BioSQL.DBUtils._PostgreSQL_dbutils-class.html#last_id"><a title="BioSQL.BioSeqDatabase.Adaptor.last_id
BioSQL.DBUtils.Generic_dbutils.last_id
BioSQL.DBUtils.Mysql_dbutils.last_id
BioSQL.DBUtils._PostgreSQL_dbutils.last_id" class="py-name" href="#" onclick="return doclink('link-42', 'last_id', 'link-42');">last_id</a></tt><tt class="py-op">(</tt><tt class="py-string">"ontology"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L90"></a><tt class="py-lineno">  90</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader._get_term_id"></a><div id="DatabaseLoader._get_term_id-def"><a name="L91"></a><tt class="py-lineno">  91</tt> <a class="py-toggle" href="#" id="DatabaseLoader._get_term_id-toggle" onclick="return toggle('DatabaseLoader._get_term_id');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#_get_term_id">_get_term_id</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> </tt>
<a name="L92"></a><tt class="py-lineno">  92</tt>  <tt class="py-line">                     <tt class="py-param">name</tt><tt class="py-op">,</tt> </tt>
<a name="L93"></a><tt class="py-lineno">  93</tt>  <tt class="py-line">                     <tt class="py-param">ontology_id</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L94"></a><tt class="py-lineno">  94</tt>  <tt class="py-line">                     <tt class="py-param">definition</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L95"></a><tt class="py-lineno">  95</tt>  <tt class="py-line">                     <tt class="py-param">identifier</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader._get_term_id-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader._get_term_id-expanded"><a name="L96"></a><tt class="py-lineno">  96</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get the id that corresponds to a term (PRIVATE).</tt> </tt>
<a name="L97"></a><tt class="py-lineno">  97</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L98"></a><tt class="py-lineno">  98</tt>  <tt class="py-line"><tt class="py-docstring">        This looks through the term table for a the given term. If it</tt> </tt>
<a name="L99"></a><tt class="py-lineno">  99</tt>  <tt class="py-line"><tt class="py-docstring">        is not found, a new id corresponding to this term is created.</tt> </tt>
<a name="L100"></a><tt class="py-lineno"> 100</tt>  <tt class="py-line"><tt class="py-docstring">        In either case, the id corresponding to that term is returned, so</tt> </tt>
<a name="L101"></a><tt class="py-lineno"> 101</tt>  <tt class="py-line"><tt class="py-docstring">        that you can reference it in another table.</tt> </tt>
<a name="L102"></a><tt class="py-lineno"> 102</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L103"></a><tt class="py-lineno"> 103</tt>  <tt class="py-line"><tt class="py-docstring">        The ontology_id should be used to disambiguate the term.</tt> </tt>
<a name="L104"></a><tt class="py-lineno"> 104</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L105"></a><tt class="py-lineno"> 105</tt>  <tt class="py-line"> </tt>
<a name="L106"></a><tt class="py-lineno"> 106</tt>  <tt class="py-line">        <tt class="py-comment"># try to get the term id</tt> </tt>
<a name="L107"></a><tt class="py-lineno"> 107</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">r"SELECT term_id FROM term "</tt> \ </tt>
<a name="L108"></a><tt class="py-lineno"> 108</tt>  <tt class="py-line">              <tt class="py-string">r"WHERE name = %s"</tt> </tt>
<a name="L109"></a><tt class="py-lineno"> 109</tt>  <tt class="py-line">        <tt class="py-name">fields</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt id="link-43" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-43', 'name', 'link-38');">name</a></tt><tt class="py-op">]</tt> </tt>
<a name="L110"></a><tt class="py-lineno"> 110</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">ontology_id</tt><tt class="py-op">:</tt> </tt>
<a name="L111"></a><tt class="py-lineno"> 111</tt>  <tt class="py-line">            <tt class="py-name">sql</tt> <tt class="py-op">+=</tt> <tt class="py-string">' AND ontology_id = %s'</tt> </tt>
<a name="L112"></a><tt class="py-lineno"> 112</tt>  <tt class="py-line">            <tt class="py-name">fields</tt><tt class="py-op">.</tt><tt id="link-44" class="py-name" targets="Method Bio.Align.MultipleSeqAlignment.append()=Bio.Align.MultipleSeqAlignment-class.html#append,Method Bio.Crystal.Chain.append()=Bio.Crystal.Chain-class.html#append,Method Bio.SCOP.Raf.SeqMap.append()=Bio.SCOP.Raf.SeqMap-class.html#append,Method Bio.SearchIO._model.hit.Hit.append()=Bio.SearchIO._model.hit.Hit-class.html#append,Method Bio.SearchIO._model.query.QueryResult.append()=Bio.SearchIO._model.query.QueryResult-class.html#append,Method Bio.Seq.MutableSeq.append()=Bio.Seq.MutableSeq-class.html#append,Method Bio.Wise.psw.Alignment.append()=Bio.Wise.psw.Alignment-class.html#append,Method Bio.Wise.psw.AlignmentColumn.append()=Bio.Wise.psw.AlignmentColumn-class.html#append"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-44', 'append', 'link-44');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">ontology_id</tt><tt class="py-op">)</tt> </tt>
<a name="L113"></a><tt class="py-lineno"> 113</tt>  <tt class="py-line">        <tt class="py-name">id_results</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-45" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.execute_and_fetchall()=BioSQL.BioSeqDatabase.Adaptor-class.html#execute_and_fetchall,Method BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetchall()=BioSQL.BioSeqDatabase.MysqlConnectorAdaptor-class.html#execute_and_fetchall"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetchall
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetchall" class="py-name" href="#" onclick="return doclink('link-45', 'execute_and_fetchall', 'link-45');">execute_and_fetchall</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-name">fields</tt><tt class="py-op">)</tt> </tt>
<a name="L114"></a><tt class="py-lineno"> 114</tt>  <tt class="py-line">        <tt class="py-comment"># something is wrong</tt> </tt>
<a name="L115"></a><tt class="py-lineno"> 115</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">id_results</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L116"></a><tt class="py-lineno"> 116</tt>  <tt class="py-line">            <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Multiple term ids for %s: %r"</tt> <tt class="py-op">%</tt> </tt>
<a name="L117"></a><tt class="py-lineno"> 117</tt>  <tt class="py-line">                             <tt class="py-op">(</tt><tt id="link-46" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-46', 'name', 'link-38');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">id_results</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L118"></a><tt class="py-lineno"> 118</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">id_results</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L119"></a><tt class="py-lineno"> 119</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">id_results</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L120"></a><tt class="py-lineno"> 120</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L121"></a><tt class="py-lineno"> 121</tt>  <tt class="py-line">            <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">r"INSERT INTO term (name, definition,"</tt> \ </tt>
<a name="L122"></a><tt class="py-lineno"> 122</tt>  <tt class="py-line">                  <tt class="py-string">r" identifier, ontology_id)"</tt> \ </tt>
<a name="L123"></a><tt class="py-lineno"> 123</tt>  <tt class="py-line">                  <tt class="py-string">r" VALUES (%s, %s, %s, %s)"</tt> </tt>
<a name="L124"></a><tt class="py-lineno"> 124</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-47" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-47', 'execute', 'link-39');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt id="link-48" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-48', 'name', 'link-38');">name</a></tt><tt class="py-op">,</tt> <tt id="link-49" class="py-name"><a title="Bio.GenBank._FeatureConsumer.definition
Bio.GenBank._RecordConsumer.definition" class="py-name" href="#" onclick="return doclink('link-49', 'definition', 'link-41');">definition</a></tt><tt class="py-op">,</tt> </tt>
<a name="L125"></a><tt class="py-lineno"> 125</tt>  <tt class="py-line">                                       <tt class="py-name">identifier</tt><tt class="py-op">,</tt> <tt class="py-name">ontology_id</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L126"></a><tt class="py-lineno"> 126</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-50" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.last_id
BioSQL.DBUtils.Generic_dbutils.last_id
BioSQL.DBUtils.Mysql_dbutils.last_id
BioSQL.DBUtils._PostgreSQL_dbutils.last_id" class="py-name" href="#" onclick="return doclink('link-50', 'last_id', 'link-42');">last_id</a></tt><tt class="py-op">(</tt><tt class="py-string">"term"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L127"></a><tt class="py-lineno"> 127</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader._add_dbxref"></a><div id="DatabaseLoader._add_dbxref-def"><a name="L128"></a><tt class="py-lineno"> 128</tt> <a class="py-toggle" href="#" id="DatabaseLoader._add_dbxref-toggle" onclick="return toggle('DatabaseLoader._add_dbxref');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#_add_dbxref">_add_dbxref</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">dbname</tt><tt class="py-op">,</tt> <tt class="py-param">accession</tt><tt class="py-op">,</tt> <tt class="py-param">version</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader._add_dbxref-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader._add_dbxref-expanded"><a name="L129"></a><tt class="py-lineno"> 129</tt>  <tt class="py-line">        <tt class="py-docstring">"""Insert a dbxref and return its id."""</tt> </tt>
<a name="L130"></a><tt class="py-lineno"> 130</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-51" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-51', 'execute', 'link-39');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L131"></a><tt class="py-lineno"> 131</tt>  <tt class="py-line">            <tt class="py-string">"INSERT INTO dbxref(dbname, accession, version)"</tt> </tt>
<a name="L132"></a><tt class="py-lineno"> 132</tt>  <tt class="py-line">            <tt class="py-string">" VALUES (%s, %s, %s)"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">dbname</tt><tt class="py-op">,</tt> <tt id="link-52" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.accession()=Bio.GenBank._FeatureConsumer-class.html#accession,Method Bio.GenBank._RecordConsumer.accession()=Bio.GenBank._RecordConsumer-class.html#accession,Method Bio.Phylo.PhyloXMLIO.Parser.accession()=Bio.Phylo.PhyloXMLIO.Parser-class.html#accession,Method Bio.Phylo.PhyloXMLIO.Writer.accession()=Bio.Phylo.PhyloXMLIO.Writer-class.html#accession"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-52', 'accession', 'link-52');">accession</a></tt><tt class="py-op">,</tt> <tt id="link-53" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HeaderConsumer.version()=Bio.Blast.NCBIStandalone._HeaderConsumer-class.html#version,Function Bio.Cluster.cluster.version()=Bio.Cluster.cluster-module.html#version,Method Bio.GenBank._FeatureConsumer.version()=Bio.GenBank._FeatureConsumer-class.html#version,Method Bio.GenBank._RecordConsumer.version()=Bio.GenBank._RecordConsumer-class.html#version,Variable Bio.motifs.jaspar.Motif.version=Bio.motifs.jaspar.Motif-class.html#version"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-53', 'version', 'link-53');">version</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L133"></a><tt class="py-lineno"> 133</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-54" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.last_id
BioSQL.DBUtils.Generic_dbutils.last_id
BioSQL.DBUtils.Mysql_dbutils.last_id
BioSQL.DBUtils._PostgreSQL_dbutils.last_id" class="py-name" href="#" onclick="return doclink('link-54', 'last_id', 'link-42');">last_id</a></tt><tt class="py-op">(</tt><tt class="py-string">"dbxref"</tt><tt class="py-op">)</tt> </tt>
</div><a name="L134"></a><tt class="py-lineno"> 134</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader._get_taxon_id"></a><div id="DatabaseLoader._get_taxon_id-def"><a name="L135"></a><tt class="py-lineno"> 135</tt> <a class="py-toggle" href="#" id="DatabaseLoader._get_taxon_id-toggle" onclick="return toggle('DatabaseLoader._get_taxon_id');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#_get_taxon_id">_get_taxon_id</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader._get_taxon_id-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader._get_taxon_id-expanded"><a name="L136"></a><tt class="py-lineno"> 136</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get the taxon id for this record (PRIVATE).</tt> </tt>
<a name="L137"></a><tt class="py-lineno"> 137</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L138"></a><tt class="py-lineno"> 138</tt>  <tt class="py-line"><tt class="py-docstring">        record - a SeqRecord object</tt> </tt>
<a name="L139"></a><tt class="py-lineno"> 139</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L140"></a><tt class="py-lineno"> 140</tt>  <tt class="py-line"><tt class="py-docstring">        This searches the taxon/taxon_name tables using the</tt> </tt>
<a name="L141"></a><tt class="py-lineno"> 141</tt>  <tt class="py-line"><tt class="py-docstring">        NCBI taxon ID, scientific name and common name to find</tt> </tt>
<a name="L142"></a><tt class="py-lineno"> 142</tt>  <tt class="py-line"><tt class="py-docstring">        the matching taxon table entry's id.</tt> </tt>
<a name="L143"></a><tt class="py-lineno"> 143</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L144"></a><tt class="py-lineno"> 144</tt>  <tt class="py-line"><tt class="py-docstring">        If the species isn't in the taxon table, and we have at</tt> </tt>
<a name="L145"></a><tt class="py-lineno"> 145</tt>  <tt class="py-line"><tt class="py-docstring">        least the NCBI taxon ID, scientific name or common name,</tt> </tt>
<a name="L146"></a><tt class="py-lineno"> 146</tt>  <tt class="py-line"><tt class="py-docstring">        at least a minimal stub entry is created in the table.</tt> </tt>
<a name="L147"></a><tt class="py-lineno"> 147</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L148"></a><tt class="py-lineno"> 148</tt>  <tt class="py-line"><tt class="py-docstring">        Returns the taxon id (database key for the taxon table,</tt> </tt>
<a name="L149"></a><tt class="py-lineno"> 149</tt>  <tt class="py-line"><tt class="py-docstring">        not an NCBI taxon ID), or None if the taxonomy information</tt> </tt>
<a name="L150"></a><tt class="py-lineno"> 150</tt>  <tt class="py-line"><tt class="py-docstring">        is missing.</tt> </tt>
<a name="L151"></a><tt class="py-lineno"> 151</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L152"></a><tt class="py-lineno"> 152</tt>  <tt class="py-line"><tt class="py-docstring">        See also the BioSQL script load_ncbi_taxonomy.pl which</tt> </tt>
<a name="L153"></a><tt class="py-lineno"> 153</tt>  <tt class="py-line"><tt class="py-docstring">        will populate and update the taxon/taxon_name tables</tt> </tt>
<a name="L154"></a><tt class="py-lineno"> 154</tt>  <tt class="py-line"><tt class="py-docstring">        with the latest information from the NCBI.</tt> </tt>
<a name="L155"></a><tt class="py-lineno"> 155</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L156"></a><tt class="py-lineno"> 156</tt>  <tt class="py-line"> </tt>
<a name="L157"></a><tt class="py-lineno"> 157</tt>  <tt class="py-line">        <tt class="py-comment"># To find the NCBI taxid, first check for a top level annotation</tt> </tt>
<a name="L158"></a><tt class="py-lineno"> 158</tt>  <tt class="py-line">        <tt class="py-name">ncbi_taxon_id</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L159"></a><tt class="py-lineno"> 159</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"ncbi_taxid"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-55" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-55', 'annotations', 'link-24');">annotations</a></tt><tt class="py-op">:</tt> </tt>
<a name="L160"></a><tt class="py-lineno"> 160</tt>  <tt class="py-line">            <tt class="py-comment"># Could be a list of IDs.</tt> </tt>
<a name="L161"></a><tt class="py-lineno"> 161</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-56" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-56', 'annotations', 'link-24');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"ncbi_taxid"</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L162"></a><tt class="py-lineno"> 162</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-57" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-57', 'annotations', 'link-24');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"ncbi_taxid"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L163"></a><tt class="py-lineno"> 163</tt>  <tt class="py-line">                    <tt class="py-name">ncbi_taxon_id</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-58" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-58', 'annotations', 'link-24');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"ncbi_taxid"</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L164"></a><tt class="py-lineno"> 164</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L165"></a><tt class="py-lineno"> 165</tt>  <tt class="py-line">                <tt class="py-name">ncbi_taxon_id</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-59" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-59', 'annotations', 'link-24');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"ncbi_taxid"</tt><tt class="py-op">]</tt> </tt>
<a name="L166"></a><tt class="py-lineno"> 166</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">ncbi_taxon_id</tt><tt class="py-op">:</tt> </tt>
<a name="L167"></a><tt class="py-lineno"> 167</tt>  <tt class="py-line">            <tt class="py-comment"># Secondly, look for a source feature</tt> </tt>
<a name="L168"></a><tt class="py-lineno"> 168</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">f</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-60" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.features" class="py-name" href="#" onclick="return doclink('link-60', 'features', 'link-34');">features</a></tt><tt class="py-op">:</tt> </tt>
<a name="L169"></a><tt class="py-lineno"> 169</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">f</tt><tt class="py-op">.</tt><tt id="link-61" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.type()=Bio.Phylo.PhyloXMLIO.Writer-class.html#type"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-61', 'type', 'link-61');">type</a></tt> <tt class="py-op">==</tt> <tt class="py-string">'source'</tt><tt class="py-op">:</tt> </tt>
<a name="L170"></a><tt class="py-lineno"> 170</tt>  <tt class="py-line">                    <tt class="py-name">quals</tt> <tt class="py-op">=</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">f</tt><tt class="py-op">,</tt> <tt class="py-string">'qualifiers'</tt><tt class="py-op">,</tt> <tt class="py-op">{</tt><tt class="py-op">}</tt><tt class="py-op">)</tt> </tt>
<a name="L171"></a><tt class="py-lineno"> 171</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-string">"db_xref"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">quals</tt><tt class="py-op">:</tt> </tt>
<a name="L172"></a><tt class="py-lineno"> 172</tt>  <tt class="py-line">                        <tt class="py-keyword">for</tt> <tt class="py-name">db_xref</tt> <tt class="py-keyword">in</tt> <tt class="py-name">f</tt><tt class="py-op">.</tt><tt class="py-name">qualifiers</tt><tt class="py-op">[</tt><tt class="py-string">"db_xref"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L173"></a><tt class="py-lineno"> 173</tt>  <tt class="py-line">                            <tt class="py-keyword">if</tt> <tt class="py-name">db_xref</tt><tt class="py-op">.</tt><tt id="link-62" class="py-name" targets="Method Bio.Seq.Seq.startswith()=Bio.Seq.Seq-class.html#startswith"><a title="Bio.Seq.Seq.startswith" class="py-name" href="#" onclick="return doclink('link-62', 'startswith', 'link-62');">startswith</a></tt><tt class="py-op">(</tt><tt class="py-string">"taxon:"</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L174"></a><tt class="py-lineno"> 174</tt>  <tt class="py-line">                                <tt class="py-name">ncbi_taxon_id</tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">db_xref</tt><tt class="py-op">[</tt><tt class="py-number">6</tt><tt class="py-op">:</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L175"></a><tt class="py-lineno"> 175</tt>  <tt class="py-line">                                <tt class="py-keyword">break</tt> </tt>
<a name="L176"></a><tt class="py-lineno"> 176</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-name">ncbi_taxon_id</tt><tt class="py-op">:</tt> </tt>
<a name="L177"></a><tt class="py-lineno"> 177</tt>  <tt class="py-line">                    <tt class="py-keyword">break</tt> </tt>
<a name="L178"></a><tt class="py-lineno"> 178</tt>  <tt class="py-line"> </tt>
<a name="L179"></a><tt class="py-lineno"> 179</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L180"></a><tt class="py-lineno"> 180</tt>  <tt class="py-line">            <tt id="link-63" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.scientific_name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#scientific_name"><a title="Bio.Phylo.PhyloXMLIO.Writer.scientific_name" class="py-name" href="#" onclick="return doclink('link-63', 'scientific_name', 'link-63');">scientific_name</a></tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-64" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-64', 'annotations', 'link-24');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"organism"</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">255</tt><tt class="py-op">]</tt> </tt>
<a name="L181"></a><tt class="py-lineno"> 181</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L182"></a><tt class="py-lineno"> 182</tt>  <tt class="py-line">            <tt id="link-65" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.scientific_name" class="py-name" href="#" onclick="return doclink('link-65', 'scientific_name', 'link-63');">scientific_name</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L183"></a><tt class="py-lineno"> 183</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L184"></a><tt class="py-lineno"> 184</tt>  <tt class="py-line">            <tt id="link-66" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.common_name()=Bio.Phylo.PhyloXMLIO.Writer-class.html#common_name"><a title="Bio.Phylo.PhyloXMLIO.Writer.common_name" class="py-name" href="#" onclick="return doclink('link-66', 'common_name', 'link-66');">common_name</a></tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-67" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-67', 'annotations', 'link-24');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"source"</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">255</tt><tt class="py-op">]</tt> </tt>
<a name="L185"></a><tt class="py-lineno"> 185</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L186"></a><tt class="py-lineno"> 186</tt>  <tt class="py-line">            <tt id="link-68" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.common_name" class="py-name" href="#" onclick="return doclink('link-68', 'common_name', 'link-66');">common_name</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L187"></a><tt class="py-lineno"> 187</tt>  <tt class="py-line">        <tt class="py-comment"># Note: The maximum length for taxon names in the schema is 255.</tt> </tt>
<a name="L188"></a><tt class="py-lineno"> 188</tt>  <tt class="py-line">        <tt class="py-comment"># Cropping it now should help in getting a match when searching,</tt> </tt>
<a name="L189"></a><tt class="py-lineno"> 189</tt>  <tt class="py-line">        <tt class="py-comment"># and avoids an error if we try and add these to the database.</tt> </tt>
<a name="L190"></a><tt class="py-lineno"> 190</tt>  <tt class="py-line"> </tt>
<a name="L191"></a><tt class="py-lineno"> 191</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">ncbi_taxon_id</tt><tt class="py-op">:</tt> </tt>
<a name="L192"></a><tt class="py-lineno"> 192</tt>  <tt class="py-line">            <tt class="py-comment"># Good, we have the NCBI taxon to go on - this is unambiguous :)</tt> </tt>
<a name="L193"></a><tt class="py-lineno"> 193</tt>  <tt class="py-line">            <tt class="py-comment"># Note that the scientific name and common name will only be</tt> </tt>
<a name="L194"></a><tt class="py-lineno"> 194</tt>  <tt class="py-line">            <tt class="py-comment"># used if we have to record a stub entry.</tt> </tt>
<a name="L195"></a><tt class="py-lineno"> 195</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-69" class="py-name" targets="Method BioSQL.Loader.DatabaseLoader._get_taxon_id_from_ncbi_taxon_id()=BioSQL.Loader.DatabaseLoader-class.html#_get_taxon_id_from_ncbi_taxon_id"><a title="BioSQL.Loader.DatabaseLoader._get_taxon_id_from_ncbi_taxon_id" class="py-name" href="#" onclick="return doclink('link-69', '_get_taxon_id_from_ncbi_taxon_id', 'link-69');">_get_taxon_id_from_ncbi_taxon_id</a></tt><tt class="py-op">(</tt><tt class="py-name">ncbi_taxon_id</tt><tt class="py-op">,</tt> </tt>
<a name="L196"></a><tt class="py-lineno"> 196</tt>  <tt class="py-line">                                                         <tt id="link-70" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.scientific_name" class="py-name" href="#" onclick="return doclink('link-70', 'scientific_name', 'link-63');">scientific_name</a></tt><tt class="py-op">,</tt> </tt>
<a name="L197"></a><tt class="py-lineno"> 197</tt>  <tt class="py-line">                                                         <tt id="link-71" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.common_name" class="py-name" href="#" onclick="return doclink('link-71', 'common_name', 'link-66');">common_name</a></tt><tt class="py-op">)</tt> </tt>
<a name="L198"></a><tt class="py-lineno"> 198</tt>  <tt class="py-line"> </tt>
<a name="L199"></a><tt class="py-lineno"> 199</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt id="link-72" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.common_name" class="py-name" href="#" onclick="return doclink('link-72', 'common_name', 'link-66');">common_name</a></tt> <tt class="py-keyword">and</tt> <tt class="py-keyword">not</tt> <tt id="link-73" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.scientific_name" class="py-name" href="#" onclick="return doclink('link-73', 'scientific_name', 'link-63');">scientific_name</a></tt><tt class="py-op">:</tt> </tt>
<a name="L200"></a><tt class="py-lineno"> 200</tt>  <tt class="py-line">            <tt class="py-comment"># Nothing to go on... and there is no point adding</tt> </tt>
<a name="L201"></a><tt class="py-lineno"> 201</tt>  <tt class="py-line">            <tt class="py-comment"># a new entry to the database.  We'll just leave this</tt> </tt>
<a name="L202"></a><tt class="py-lineno"> 202</tt>  <tt class="py-line">            <tt class="py-comment"># sequence's taxon as a NULL in the database.</tt> </tt>
<a name="L203"></a><tt class="py-lineno"> 203</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">None</tt> </tt>
<a name="L204"></a><tt class="py-lineno"> 204</tt>  <tt class="py-line"> </tt>
<a name="L205"></a><tt class="py-lineno"> 205</tt>  <tt class="py-line">        <tt class="py-comment"># Next, we'll try to find a match based on the species name</tt> </tt>
<a name="L206"></a><tt class="py-lineno"> 206</tt>  <tt class="py-line">        <tt class="py-comment"># (stored in GenBank files as the organism and/or the source).</tt> </tt>
<a name="L207"></a><tt class="py-lineno"> 207</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-74" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.scientific_name" class="py-name" href="#" onclick="return doclink('link-74', 'scientific_name', 'link-63');">scientific_name</a></tt><tt class="py-op">:</tt> </tt>
<a name="L208"></a><tt class="py-lineno"> 208</tt>  <tt class="py-line">            <tt class="py-name">taxa</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-75" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0" class="py-name" href="#" onclick="return doclink('link-75', 'execute_and_fetch_col0', 'link-37');">execute_and_fetch_col0</a></tt><tt class="py-op">(</tt> </tt>
<a name="L209"></a><tt class="py-lineno"> 209</tt>  <tt class="py-line">                <tt class="py-string">"SELECT taxon_id FROM taxon_name"</tt> </tt>
<a name="L210"></a><tt class="py-lineno"> 210</tt>  <tt class="py-line">                <tt class="py-string">" WHERE name_class = 'scientific name' AND name = %s"</tt><tt class="py-op">,</tt> </tt>
<a name="L211"></a><tt class="py-lineno"> 211</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt id="link-76" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.scientific_name" class="py-name" href="#" onclick="return doclink('link-76', 'scientific_name', 'link-63');">scientific_name</a></tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L212"></a><tt class="py-lineno"> 212</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">taxa</tt><tt class="py-op">:</tt> </tt>
<a name="L213"></a><tt class="py-lineno"> 213</tt>  <tt class="py-line">                <tt class="py-comment"># Good, mapped the scientific name to a taxon table entry</tt> </tt>
<a name="L214"></a><tt class="py-lineno"> 214</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">taxa</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L215"></a><tt class="py-lineno"> 215</tt>  <tt class="py-line"> </tt>
<a name="L216"></a><tt class="py-lineno"> 216</tt>  <tt class="py-line">        <tt class="py-comment"># Last chance...</tt> </tt>
<a name="L217"></a><tt class="py-lineno"> 217</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-77" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.common_name" class="py-name" href="#" onclick="return doclink('link-77', 'common_name', 'link-66');">common_name</a></tt><tt class="py-op">:</tt> </tt>
<a name="L218"></a><tt class="py-lineno"> 218</tt>  <tt class="py-line">            <tt class="py-name">taxa</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-78" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0" class="py-name" href="#" onclick="return doclink('link-78', 'execute_and_fetch_col0', 'link-37');">execute_and_fetch_col0</a></tt><tt class="py-op">(</tt> </tt>
<a name="L219"></a><tt class="py-lineno"> 219</tt>  <tt class="py-line">                <tt class="py-string">"SELECT DISTINCT taxon_id FROM taxon_name"</tt> </tt>
<a name="L220"></a><tt class="py-lineno"> 220</tt>  <tt class="py-line">                <tt class="py-string">" WHERE name = %s"</tt><tt class="py-op">,</tt> </tt>
<a name="L221"></a><tt class="py-lineno"> 221</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt id="link-79" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.common_name" class="py-name" href="#" onclick="return doclink('link-79', 'common_name', 'link-66');">common_name</a></tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L222"></a><tt class="py-lineno"> 222</tt>  <tt class="py-line">            <tt class="py-comment"># Its natural that several distinct taxa will have the same common</tt> </tt>
<a name="L223"></a><tt class="py-lineno"> 223</tt>  <tt class="py-line">            <tt class="py-comment"># name - in which case we can't resolve the taxon uniquely.</tt> </tt>
<a name="L224"></a><tt class="py-lineno"> 224</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">taxa</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L225"></a><tt class="py-lineno"> 225</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Taxa: %d species have name %r"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt> </tt>
<a name="L226"></a><tt class="py-lineno"> 226</tt>  <tt class="py-line">                    <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">taxa</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L227"></a><tt class="py-lineno"> 227</tt>  <tt class="py-line">                    <tt id="link-80" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.common_name" class="py-name" href="#" onclick="return doclink('link-80', 'common_name', 'link-66');">common_name</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L228"></a><tt class="py-lineno"> 228</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">taxa</tt><tt class="py-op">:</tt> </tt>
<a name="L229"></a><tt class="py-lineno"> 229</tt>  <tt class="py-line">                <tt class="py-comment"># Good, mapped the common name to a taxon table entry</tt> </tt>
<a name="L230"></a><tt class="py-lineno"> 230</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">taxa</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L231"></a><tt class="py-lineno"> 231</tt>  <tt class="py-line"> </tt>
<a name="L232"></a><tt class="py-lineno"> 232</tt>  <tt class="py-line">        <tt class="py-comment"># At this point, as far as we can tell, this species isn't</tt> </tt>
<a name="L233"></a><tt class="py-lineno"> 233</tt>  <tt class="py-line">        <tt class="py-comment"># in the taxon table already.  So we'll have to add it.</tt> </tt>
<a name="L234"></a><tt class="py-lineno"> 234</tt>  <tt class="py-line">        <tt class="py-comment"># We don't have an NCBI taxonomy ID, so if we do record just</tt> </tt>
<a name="L235"></a><tt class="py-lineno"> 235</tt>  <tt class="py-line">        <tt class="py-comment"># a stub entry, there is no simple way to fix this later.</tt> </tt>
<a name="L236"></a><tt class="py-lineno"> 236</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L237"></a><tt class="py-lineno"> 237</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - Should we try searching the NCBI taxonomy using the</tt> </tt>
<a name="L238"></a><tt class="py-lineno"> 238</tt>  <tt class="py-line">        <tt class="py-comment"># species name?</tt> </tt>
<a name="L239"></a><tt class="py-lineno"> 239</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L240"></a><tt class="py-lineno"> 240</tt>  <tt class="py-line">        <tt class="py-comment"># OK, let's try inserting the species.</tt> </tt>
<a name="L241"></a><tt class="py-lineno"> 241</tt>  <tt class="py-line">        <tt class="py-comment"># Chances are we don't have enough information ...</tt> </tt>
<a name="L242"></a><tt class="py-lineno"> 242</tt>  <tt class="py-line">        <tt class="py-comment"># Furthermore, it won't be in the hierarchy.</tt> </tt>
<a name="L243"></a><tt class="py-lineno"> 243</tt>  <tt class="py-line"> </tt>
<a name="L244"></a><tt class="py-lineno"> 244</tt>  <tt class="py-line">        <tt class="py-name">lineage</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L245"></a><tt class="py-lineno"> 245</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">c</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-81" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-81', 'annotations', 'link-24');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-82" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-82', 'get', 'link-25');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"taxonomy"</tt><tt class="py-op">,</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L246"></a><tt class="py-lineno"> 246</tt>  <tt class="py-line">            <tt class="py-name">lineage</tt><tt class="py-op">.</tt><tt id="link-83" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-83', 'append', 'link-44');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-name">c</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L247"></a><tt class="py-lineno"> 247</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">lineage</tt><tt class="py-op">:</tt> </tt>
<a name="L248"></a><tt class="py-lineno"> 248</tt>  <tt class="py-line">            <tt class="py-name">lineage</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-string">"genus"</tt> </tt>
<a name="L249"></a><tt class="py-lineno"> 249</tt>  <tt class="py-line">        <tt class="py-name">lineage</tt><tt class="py-op">.</tt><tt id="link-84" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-84', 'append', 'link-44');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-string">"species"</tt><tt class="py-op">,</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-85" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-85', 'annotations', 'link-24');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"organism"</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L250"></a><tt class="py-lineno"> 250</tt>  <tt class="py-line">        <tt class="py-comment"># XXX do we have them?</tt> </tt>
<a name="L251"></a><tt class="py-lineno"> 251</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"subspecies"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-86" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-86', 'annotations', 'link-24');">annotations</a></tt><tt class="py-op">:</tt> </tt>
<a name="L252"></a><tt class="py-lineno"> 252</tt>  <tt class="py-line">            <tt class="py-name">lineage</tt><tt class="py-op">.</tt><tt id="link-87" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-87', 'append', 'link-44');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-string">"subspecies"</tt><tt class="py-op">,</tt> </tt>
<a name="L253"></a><tt class="py-lineno"> 253</tt>  <tt class="py-line">                            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-88" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-88', 'annotations', 'link-24');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"subspecies"</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L254"></a><tt class="py-lineno"> 254</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"variant"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-89" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-89', 'annotations', 'link-24');">annotations</a></tt><tt class="py-op">:</tt> </tt>
<a name="L255"></a><tt class="py-lineno"> 255</tt>  <tt class="py-line">            <tt class="py-name">lineage</tt><tt class="py-op">.</tt><tt id="link-90" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-90', 'append', 'link-44');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> <tt class="py-string">"varietas"</tt><tt class="py-op">,</tt> </tt>
<a name="L256"></a><tt class="py-lineno"> 256</tt>  <tt class="py-line">                            <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-91" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-91', 'annotations', 'link-24');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"variant"</tt><tt class="py-op">]</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L257"></a><tt class="py-lineno"> 257</tt>  <tt class="py-line">        <tt class="py-name">lineage</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> <tt class="py-op">=</tt> <tt class="py-name">ncbi_taxon_id</tt> </tt>
<a name="L258"></a><tt class="py-lineno"> 258</tt>  <tt class="py-line"> </tt>
<a name="L259"></a><tt class="py-lineno"> 259</tt>  <tt class="py-line">        <tt class="py-name">left_value</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-92" class="py-name" targets="Method BioSQL.BioSeqDatabase.Adaptor.execute_one()=BioSQL.BioSeqDatabase.Adaptor-class.html#execute_one,Method BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_one()=BioSQL.BioSeqDatabase.MysqlConnectorAdaptor-class.html#execute_one"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_one
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_one" class="py-name" href="#" onclick="return doclink('link-92', 'execute_one', 'link-92');">execute_one</a></tt><tt class="py-op">(</tt> </tt>
<a name="L260"></a><tt class="py-lineno"> 260</tt>  <tt class="py-line">            <tt class="py-string">"SELECT MAX(left_value) FROM taxon"</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L261"></a><tt class="py-lineno"> 261</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">left_value</tt><tt class="py-op">:</tt> </tt>
<a name="L262"></a><tt class="py-lineno"> 262</tt>  <tt class="py-line">            <tt class="py-name">left_value</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L263"></a><tt class="py-lineno"> 263</tt>  <tt class="py-line">        <tt class="py-name">left_value</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L264"></a><tt class="py-lineno"> 264</tt>  <tt class="py-line"> </tt>
<a name="L265"></a><tt class="py-lineno"> 265</tt>  <tt class="py-line">        <tt class="py-comment"># XXX -- Brad: Fixing this for now in an ugly way because</tt> </tt>
<a name="L266"></a><tt class="py-lineno"> 266</tt>  <tt class="py-line">        <tt class="py-comment"># I am getting overlaps for right_values. I need to dig into this</tt> </tt>
<a name="L267"></a><tt class="py-lineno"> 267</tt>  <tt class="py-line">        <tt class="py-comment"># more to actually understand how it works. I'm not sure it is</tt> </tt>
<a name="L268"></a><tt class="py-lineno"> 268</tt>  <tt class="py-line">        <tt class="py-comment"># actually working right anyhow.</tt> </tt>
<a name="L269"></a><tt class="py-lineno"> 269</tt>  <tt class="py-line">        <tt class="py-name">right_start_value</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-93" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_one
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_one" class="py-name" href="#" onclick="return doclink('link-93', 'execute_one', 'link-92');">execute_one</a></tt><tt class="py-op">(</tt> </tt>
<a name="L270"></a><tt class="py-lineno"> 270</tt>  <tt class="py-line">            <tt class="py-string">"SELECT MAX(right_value) FROM taxon"</tt><tt class="py-op">)</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L271"></a><tt class="py-lineno"> 271</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">right_start_value</tt><tt class="py-op">:</tt> </tt>
<a name="L272"></a><tt class="py-lineno"> 272</tt>  <tt class="py-line">            <tt class="py-name">right_start_value</tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L273"></a><tt class="py-lineno"> 273</tt>  <tt class="py-line">        <tt class="py-name">right_value</tt> <tt class="py-op">=</tt> <tt class="py-name">right_start_value</tt> <tt class="py-op">+</tt> <tt class="py-number">2</tt> <tt class="py-op">*</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">lineage</tt><tt class="py-op">)</tt> <tt class="py-op">-</tt> <tt class="py-number">1</tt> </tt>
<a name="L274"></a><tt class="py-lineno"> 274</tt>  <tt class="py-line"> </tt>
<a name="L275"></a><tt class="py-lineno"> 275</tt>  <tt class="py-line">        <tt class="py-name">parent_taxon_id</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L276"></a><tt class="py-lineno"> 276</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">taxon</tt> <tt class="py-keyword">in</tt> <tt class="py-name">lineage</tt><tt class="py-op">:</tt> </tt>
<a name="L277"></a><tt class="py-lineno"> 277</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-94" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-94', 'execute', 'link-39');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L278"></a><tt class="py-lineno"> 278</tt>  <tt class="py-line">                <tt class="py-string">"INSERT INTO taxon(parent_taxon_id, ncbi_taxon_id, node_rank,"</tt> </tt>
<a name="L279"></a><tt class="py-lineno"> 279</tt>  <tt class="py-line">                <tt class="py-string">" left_value, right_value)"</tt> </tt>
<a name="L280"></a><tt class="py-lineno"> 280</tt>  <tt class="py-line">                <tt class="py-string">" VALUES (%s, %s, %s, %s, %s)"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">parent_taxon_id</tt><tt class="py-op">,</tt> </tt>
<a name="L281"></a><tt class="py-lineno"> 281</tt>  <tt class="py-line">                                                 <tt class="py-name">taxon</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L282"></a><tt class="py-lineno"> 282</tt>  <tt class="py-line">                                                 <tt class="py-name">taxon</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L283"></a><tt class="py-lineno"> 283</tt>  <tt class="py-line">                                                 <tt class="py-name">left_value</tt><tt class="py-op">,</tt> </tt>
<a name="L284"></a><tt class="py-lineno"> 284</tt>  <tt class="py-line">                                                 <tt class="py-name">right_value</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L285"></a><tt class="py-lineno"> 285</tt>  <tt class="py-line">            <tt class="py-name">taxon_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-95" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.last_id
BioSQL.DBUtils.Generic_dbutils.last_id
BioSQL.DBUtils.Mysql_dbutils.last_id
BioSQL.DBUtils._PostgreSQL_dbutils.last_id" class="py-name" href="#" onclick="return doclink('link-95', 'last_id', 'link-42');">last_id</a></tt><tt class="py-op">(</tt><tt class="py-string">"taxon"</tt><tt class="py-op">)</tt> </tt>
<a name="L286"></a><tt class="py-lineno"> 286</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-96" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-96', 'execute', 'link-39');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L287"></a><tt class="py-lineno"> 287</tt>  <tt class="py-line">                <tt class="py-string">"INSERT INTO taxon_name(taxon_id, name, name_class)"</tt> </tt>
<a name="L288"></a><tt class="py-lineno"> 288</tt>  <tt class="py-line">                <tt class="py-string">"VALUES (%s, %s, 'scientific name')"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">taxon_id</tt><tt class="py-op">,</tt> <tt class="py-name">taxon</tt><tt class="py-op">[</tt><tt class="py-number">2</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">255</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L289"></a><tt class="py-lineno"> 289</tt>  <tt class="py-line">            <tt class="py-comment"># Note the name field is limited to 255, some SwissProt files</tt> </tt>
<a name="L290"></a><tt class="py-lineno"> 290</tt>  <tt class="py-line">            <tt class="py-comment"># have a multi-species name which can be longer.  So truncate this.</tt> </tt>
<a name="L291"></a><tt class="py-lineno"> 291</tt>  <tt class="py-line">            <tt class="py-name">left_value</tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L292"></a><tt class="py-lineno"> 292</tt>  <tt class="py-line">            <tt class="py-name">right_value</tt> <tt class="py-op">-=</tt> <tt class="py-number">1</tt> </tt>
<a name="L293"></a><tt class="py-lineno"> 293</tt>  <tt class="py-line">            <tt class="py-name">parent_taxon_id</tt> <tt class="py-op">=</tt> <tt class="py-name">taxon_id</tt> </tt>
<a name="L294"></a><tt class="py-lineno"> 294</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-97" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.common_name" class="py-name" href="#" onclick="return doclink('link-97', 'common_name', 'link-66');">common_name</a></tt><tt class="py-op">:</tt> </tt>
<a name="L295"></a><tt class="py-lineno"> 295</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-98" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-98', 'execute', 'link-39');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L296"></a><tt class="py-lineno"> 296</tt>  <tt class="py-line">                <tt class="py-string">"INSERT INTO taxon_name(taxon_id, name, name_class)"</tt> </tt>
<a name="L297"></a><tt class="py-lineno"> 297</tt>  <tt class="py-line">                <tt class="py-string">"VALUES (%s, %s, 'common name')"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt> </tt>
<a name="L298"></a><tt class="py-lineno"> 298</tt>  <tt class="py-line">                    <tt class="py-name">taxon_id</tt><tt class="py-op">,</tt> <tt id="link-99" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.common_name" class="py-name" href="#" onclick="return doclink('link-99', 'common_name', 'link-66');">common_name</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L299"></a><tt class="py-lineno"> 299</tt>  <tt class="py-line"> </tt>
<a name="L300"></a><tt class="py-lineno"> 300</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">taxon_id</tt> </tt>
</div><a name="L301"></a><tt class="py-lineno"> 301</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader._fix_name_class"></a><div id="DatabaseLoader._fix_name_class-def"><a name="L302"></a><tt class="py-lineno"> 302</tt> <a class="py-toggle" href="#" id="DatabaseLoader._fix_name_class-toggle" onclick="return toggle('DatabaseLoader._fix_name_class');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#_fix_name_class">_fix_name_class</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">entrez_name</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader._fix_name_class-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader._fix_name_class-expanded"><a name="L303"></a><tt class="py-lineno"> 303</tt>  <tt class="py-line">        <tt class="py-docstring">"""Map Entrez name terms to those used in taxdump (PRIVATE).</tt> </tt>
<a name="L304"></a><tt class="py-lineno"> 304</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L305"></a><tt class="py-lineno"> 305</tt>  <tt class="py-line"><tt class="py-docstring">        We need to make this conversion to match the taxon_name.name_class</tt> </tt>
<a name="L306"></a><tt class="py-lineno"> 306</tt>  <tt class="py-line"><tt class="py-docstring">        values used by the BioSQL load_ncbi_taxonomy.pl script.</tt> </tt>
<a name="L307"></a><tt class="py-lineno"> 307</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L308"></a><tt class="py-lineno"> 308</tt>  <tt class="py-line"><tt class="py-docstring">        e.g.</tt> </tt>
<a name="L309"></a><tt class="py-lineno"> 309</tt>  <tt class="py-line"><tt class="py-docstring">        "ScientificName" -&gt; "scientific name",</tt> </tt>
<a name="L310"></a><tt class="py-lineno"> 310</tt>  <tt class="py-line"><tt class="py-docstring">        "EquivalentName" -&gt; "equivalent name",</tt> </tt>
<a name="L311"></a><tt class="py-lineno"> 311</tt>  <tt class="py-line"><tt class="py-docstring">        "Synonym" -&gt; "synonym",</tt> </tt>
<a name="L312"></a><tt class="py-lineno"> 312</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L313"></a><tt class="py-lineno"> 313</tt>  <tt class="py-line">        <tt class="py-comment"># Add any special cases here:</tt> </tt>
<a name="L314"></a><tt class="py-lineno"> 314</tt>  <tt class="py-line">        <tt class="py-comment">#</tt> </tt>
<a name="L315"></a><tt class="py-lineno"> 315</tt>  <tt class="py-line">        <tt class="py-comment"># known = {}</tt> </tt>
<a name="L316"></a><tt class="py-lineno"> 316</tt>  <tt class="py-line">        <tt class="py-comment"># try:</tt> </tt>
<a name="L317"></a><tt class="py-lineno"> 317</tt>  <tt class="py-line">        <tt class="py-comment">#     return known[entrez_name]</tt> </tt>
<a name="L318"></a><tt class="py-lineno"> 318</tt>  <tt class="py-line">        <tt class="py-comment"># except KeyError:</tt> </tt>
<a name="L319"></a><tt class="py-lineno"> 319</tt>  <tt class="py-line">        <tt class="py-comment">#     pass</tt> </tt>
<a name="L320"></a><tt class="py-lineno"> 320</tt>  <tt class="py-line"> </tt>
<a name="L321"></a><tt class="py-lineno"> 321</tt>  <tt class="py-line">        <tt class="py-comment"># Try automatically by adding spaces before each capital</tt> </tt>
<a name="L322"></a><tt class="py-lineno"> 322</tt>  <tt class="py-line">        <tt class="py-keyword">def</tt> <tt class="py-def-name">add_space</tt><tt class="py-op">(</tt><tt class="py-param">letter</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L323"></a><tt class="py-lineno"> 323</tt>  <tt class="py-line">            <tt class="py-docstring">"""Adds a space before a capital letter."""</tt> </tt>
<a name="L324"></a><tt class="py-lineno"> 324</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">letter</tt><tt class="py-op">.</tt><tt class="py-name">isupper</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L325"></a><tt class="py-lineno"> 325</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-string">" "</tt> <tt class="py-op">+</tt> <tt class="py-name">letter</tt><tt class="py-op">.</tt><tt id="link-100" class="py-name" targets="Method Bio.Seq.Seq.lower()=Bio.Seq.Seq-class.html#lower,Method Bio.Seq.UnknownSeq.lower()=Bio.Seq.UnknownSeq-class.html#lower,Method Bio.SeqRecord.SeqRecord.lower()=Bio.SeqRecord.SeqRecord-class.html#lower"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-100', 'lower', 'link-100');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L326"></a><tt class="py-lineno"> 326</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L327"></a><tt class="py-lineno"> 327</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">letter</tt> </tt>
</div><a name="L328"></a><tt class="py-lineno"> 328</tt>  <tt class="py-line">        <tt class="py-name">answer</tt> <tt class="py-op">=</tt> <tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">add_space</tt><tt class="py-op">(</tt><tt class="py-name">letter</tt><tt class="py-op">)</tt> <tt class="py-keyword">for</tt> <tt class="py-name">letter</tt> <tt class="py-keyword">in</tt> <tt class="py-name">entrez_name</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-101" class="py-name" targets="Method Bio.Seq.Seq.strip()=Bio.Seq.Seq-class.html#strip"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-101', 'strip', 'link-101');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L329"></a><tt class="py-lineno"> 329</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">answer</tt> <tt class="py-op">==</tt> <tt class="py-name">answer</tt><tt class="py-op">.</tt><tt id="link-102" class="py-name"><a title="Bio.Seq.Seq.lower
Bio.Seq.UnknownSeq.lower
Bio.SeqRecord.SeqRecord.lower" class="py-name" href="#" onclick="return doclink('link-102', 'lower', 'link-100');">lower</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L330"></a><tt class="py-lineno"> 330</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">answer</tt> </tt>
</div><a name="L331"></a><tt class="py-lineno"> 331</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader._get_taxon_id_from_ncbi_taxon_id"></a><div id="DatabaseLoader._get_taxon_id_from_ncbi_taxon_id-def"><a name="L332"></a><tt class="py-lineno"> 332</tt> <a class="py-toggle" href="#" id="DatabaseLoader._get_taxon_id_from_ncbi_taxon_id-toggle" onclick="return toggle('DatabaseLoader._get_taxon_id_from_ncbi_taxon_id');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#_get_taxon_id_from_ncbi_taxon_id">_get_taxon_id_from_ncbi_taxon_id</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">ncbi_taxon_id</tt><tt class="py-op">,</tt> </tt>
<a name="L333"></a><tt class="py-lineno"> 333</tt>  <tt class="py-line">                                         <tt class="py-param">scientific_name</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L334"></a><tt class="py-lineno"> 334</tt>  <tt class="py-line">                                         <tt class="py-param">common_name</tt><tt class="py-op">=</tt><tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader._get_taxon_id_from_ncbi_taxon_id-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader._get_taxon_id_from_ncbi_taxon_id-expanded"><a name="L335"></a><tt class="py-lineno"> 335</tt>  <tt class="py-line">        <tt class="py-docstring">"""Get the taxon id for this record from the NCBI taxon ID (PRIVATE).</tt> </tt>
<a name="L336"></a><tt class="py-lineno"> 336</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L337"></a><tt class="py-lineno"> 337</tt>  <tt class="py-line"><tt class="py-docstring">        ncbi_taxon_id - string containing an NCBI taxon id</tt> </tt>
<a name="L338"></a><tt class="py-lineno"> 338</tt>  <tt class="py-line"><tt class="py-docstring">        scientific_name - string, used if a stub entry is recorded</tt> </tt>
<a name="L339"></a><tt class="py-lineno"> 339</tt>  <tt class="py-line"><tt class="py-docstring">        common_name - string, used if a stub entry is recorded</tt> </tt>
<a name="L340"></a><tt class="py-lineno"> 340</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L341"></a><tt class="py-lineno"> 341</tt>  <tt class="py-line"><tt class="py-docstring">        This searches the taxon table using ONLY the NCBI taxon ID</tt> </tt>
<a name="L342"></a><tt class="py-lineno"> 342</tt>  <tt class="py-line"><tt class="py-docstring">        to find the matching taxon table entry's ID (database key).</tt> </tt>
<a name="L343"></a><tt class="py-lineno"> 343</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L344"></a><tt class="py-lineno"> 344</tt>  <tt class="py-line"><tt class="py-docstring">        If the species isn't in the taxon table, and the fetch_NCBI_taxonomy</tt> </tt>
<a name="L345"></a><tt class="py-lineno"> 345</tt>  <tt class="py-line"><tt class="py-docstring">        flag is true, Biopython will attempt to go online using Bio.Entrez</tt> </tt>
<a name="L346"></a><tt class="py-lineno"> 346</tt>  <tt class="py-line"><tt class="py-docstring">        to fetch the official NCBI lineage, recursing up the tree until an</tt> </tt>
<a name="L347"></a><tt class="py-lineno"> 347</tt>  <tt class="py-line"><tt class="py-docstring">        existing entry is found in the database or the full lineage has been</tt> </tt>
<a name="L348"></a><tt class="py-lineno"> 348</tt>  <tt class="py-line"><tt class="py-docstring">        fetched.</tt> </tt>
<a name="L349"></a><tt class="py-lineno"> 349</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L350"></a><tt class="py-lineno"> 350</tt>  <tt class="py-line"><tt class="py-docstring">        Otherwise the NCBI taxon ID, scientific name and common name are</tt> </tt>
<a name="L351"></a><tt class="py-lineno"> 351</tt>  <tt class="py-line"><tt class="py-docstring">        recorded as a minimal stub entry in the taxon and taxon_name tables.</tt> </tt>
<a name="L352"></a><tt class="py-lineno"> 352</tt>  <tt class="py-line"><tt class="py-docstring">        Any partial information about the lineage from the SeqRecord is NOT</tt> </tt>
<a name="L353"></a><tt class="py-lineno"> 353</tt>  <tt class="py-line"><tt class="py-docstring">        recorded.  This should mean that (re)running the BioSQL script</tt> </tt>
<a name="L354"></a><tt class="py-lineno"> 354</tt>  <tt class="py-line"><tt class="py-docstring">        load_ncbi_taxonomy.pl can fill in the taxonomy lineage.</tt> </tt>
<a name="L355"></a><tt class="py-lineno"> 355</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L356"></a><tt class="py-lineno"> 356</tt>  <tt class="py-line"><tt class="py-docstring">        Returns the taxon id (database key for the taxon table, not</tt> </tt>
<a name="L357"></a><tt class="py-lineno"> 357</tt>  <tt class="py-line"><tt class="py-docstring">        an NCBI taxon ID).</tt> </tt>
<a name="L358"></a><tt class="py-lineno"> 358</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L359"></a><tt class="py-lineno"> 359</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">ncbi_taxon_id</tt> </tt>
<a name="L360"></a><tt class="py-lineno"> 360</tt>  <tt class="py-line"> </tt>
<a name="L361"></a><tt class="py-lineno"> 361</tt>  <tt class="py-line">        <tt class="py-name">taxon_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-103" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0" class="py-name" href="#" onclick="return doclink('link-103', 'execute_and_fetch_col0', 'link-37');">execute_and_fetch_col0</a></tt><tt class="py-op">(</tt> </tt>
<a name="L362"></a><tt class="py-lineno"> 362</tt>  <tt class="py-line">            <tt class="py-string">"SELECT taxon_id FROM taxon WHERE ncbi_taxon_id = %s"</tt><tt class="py-op">,</tt> </tt>
<a name="L363"></a><tt class="py-lineno"> 363</tt>  <tt class="py-line">            <tt class="py-op">(</tt><tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">ncbi_taxon_id</tt><tt class="py-op">)</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L364"></a><tt class="py-lineno"> 364</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">taxon_id</tt><tt class="py-op">:</tt> </tt>
<a name="L365"></a><tt class="py-lineno"> 365</tt>  <tt class="py-line">            <tt class="py-comment"># Good, we have mapped the NCBI taxid to a taxon table entry</tt> </tt>
<a name="L366"></a><tt class="py-lineno"> 366</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">taxon_id</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L367"></a><tt class="py-lineno"> 367</tt>  <tt class="py-line"> </tt>
<a name="L368"></a><tt class="py-lineno"> 368</tt>  <tt class="py-line">        <tt class="py-comment"># At this point, as far as we can tell, this species isn't</tt> </tt>
<a name="L369"></a><tt class="py-lineno"> 369</tt>  <tt class="py-line">        <tt class="py-comment"># in the taxon table already.  So we'll have to add it.</tt> </tt>
<a name="L370"></a><tt class="py-lineno"> 370</tt>  <tt class="py-line"> </tt>
<a name="L371"></a><tt class="py-lineno"> 371</tt>  <tt class="py-line">        <tt class="py-name">parent_taxon_id</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L372"></a><tt class="py-lineno"> 372</tt>  <tt class="py-line">        <tt id="link-104" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-104', 'rank', 'link-27');">rank</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"species"</tt> </tt>
<a name="L373"></a><tt class="py-lineno"> 373</tt>  <tt class="py-line">        <tt class="py-name">genetic_code</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L374"></a><tt class="py-lineno"> 374</tt>  <tt class="py-line">        <tt class="py-name">mito_genetic_code</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L375"></a><tt class="py-lineno"> 375</tt>  <tt class="py-line">        <tt class="py-name">species_names</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L376"></a><tt class="py-lineno"> 376</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-105" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.scientific_name" class="py-name" href="#" onclick="return doclink('link-105', 'scientific_name', 'link-63');">scientific_name</a></tt><tt class="py-op">:</tt> </tt>
<a name="L377"></a><tt class="py-lineno"> 377</tt>  <tt class="py-line">            <tt class="py-name">species_names</tt><tt class="py-op">.</tt><tt id="link-106" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-106', 'append', 'link-44');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">"scientific name"</tt><tt class="py-op">,</tt> <tt id="link-107" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.scientific_name" class="py-name" href="#" onclick="return doclink('link-107', 'scientific_name', 'link-63');">scientific_name</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L378"></a><tt class="py-lineno"> 378</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-108" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.common_name" class="py-name" href="#" onclick="return doclink('link-108', 'common_name', 'link-66');">common_name</a></tt><tt class="py-op">:</tt> </tt>
<a name="L379"></a><tt class="py-lineno"> 379</tt>  <tt class="py-line">            <tt class="py-name">species_names</tt><tt class="py-op">.</tt><tt id="link-109" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-109', 'append', 'link-44');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-string">"common name"</tt><tt class="py-op">,</tt> <tt id="link-110" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.common_name" class="py-name" href="#" onclick="return doclink('link-110', 'common_name', 'link-66');">common_name</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L380"></a><tt class="py-lineno"> 380</tt>  <tt class="py-line"> </tt>
<a name="L381"></a><tt class="py-lineno"> 381</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">fetch_NCBI_taxonomy</tt><tt class="py-op">:</tt> </tt>
<a name="L382"></a><tt class="py-lineno"> 382</tt>  <tt class="py-line">            <tt class="py-comment"># Go online to get the parent taxon ID!</tt> </tt>
<a name="L383"></a><tt class="py-lineno"> 383</tt>  <tt class="py-line">            <tt class="py-name">handle</tt> <tt class="py-op">=</tt> <tt id="link-111" class="py-name"><a title="Bio.Entrez" class="py-name" href="#" onclick="return doclink('link-111', 'Entrez', 'link-7');">Entrez</a></tt><tt class="py-op">.</tt><tt id="link-112" class="py-name" targets="Function Bio.Entrez.efetch()=Bio.Entrez-module.html#efetch"><a title="Bio.Entrez.efetch" class="py-name" href="#" onclick="return doclink('link-112', 'efetch', 'link-112');">efetch</a></tt><tt class="py-op">(</tt> </tt>
<a name="L384"></a><tt class="py-lineno"> 384</tt>  <tt class="py-line">                <tt id="link-113" class="py-name" targets="Module Bio.motifs.jaspar.db=Bio.motifs.jaspar.db-module.html"><a title="Bio.motifs.jaspar.db" class="py-name" href="#" onclick="return doclink('link-113', 'db', 'link-113');">db</a></tt><tt class="py-op">=</tt><tt class="py-string">"taxonomy"</tt><tt class="py-op">,</tt> <tt id="link-114" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Component.id=Bio.KEGG.KGML.KGML_pathway.Component-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Entry.id=Bio.KEGG.KGML.KGML_pathway.Entry-class.html#id,Variable Bio.KEGG.KGML.KGML_pathway.Reaction.id=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#id,Method Bio.Phylo.PhyloXMLIO.Parser.id()=Bio.Phylo.PhyloXMLIO.Parser-class.html#id,Method Bio.Phylo.PhyloXMLIO.Writer.id()=Bio.Phylo.PhyloXMLIO.Writer-class.html#id,Variable Bio.SearchIO._model.hit.Hit.id=Bio.SearchIO._model.hit.Hit-class.html#id,Variable Bio.SearchIO._model.query.QueryResult.id=Bio.SearchIO._model.query.QueryResult-class.html#id"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-114', 'id', 'link-114');">id</a></tt><tt class="py-op">=</tt><tt class="py-name">ncbi_taxon_id</tt><tt class="py-op">,</tt> <tt class="py-name">retmode</tt><tt class="py-op">=</tt><tt class="py-string">"XML"</tt><tt class="py-op">)</tt> </tt>
<a name="L385"></a><tt class="py-lineno"> 385</tt>  <tt class="py-line">            <tt class="py-name">taxonomic_record</tt> <tt class="py-op">=</tt> <tt id="link-115" class="py-name"><a title="Bio.Entrez" class="py-name" href="#" onclick="return doclink('link-115', 'Entrez', 'link-7');">Entrez</a></tt><tt class="py-op">.</tt><tt id="link-116" class="py-name" targets="Function Bio.Affy.CelFile.read()=Bio.Affy.CelFile-module.html#read,Function Bio.AlignIO.read()=Bio.AlignIO-module.html#read,Function Bio.Blast.NCBIXML.read()=Bio.Blast.NCBIXML-module.html#read,Function Bio.Cluster.read()=Bio.Cluster-module.html#read,Function Bio.Compass.read()=Bio.Compass-module.html#read,Function Bio.Emboss.Primer3.read()=Bio.Emboss.Primer3-module.html#read,Function Bio.Emboss.PrimerSearch.read()=Bio.Emboss.PrimerSearch-module.html#read,Method Bio.Entrez.Parser.DataHandler.read()=Bio.Entrez.Parser.DataHandler-class.html#read,Function Bio.Entrez.read()=Bio.Entrez-module.html#read,Function Bio.ExPASy.Enzyme.read()=Bio.ExPASy.Enzyme-module.html#read,Function Bio.ExPASy.Prodoc.read()=Bio.ExPASy.Prodoc-module.html#read,Function Bio.ExPASy.Prosite.read()=Bio.ExPASy.Prosite-module.html#read,Function Bio.ExPASy.ScanProsite.read()=Bio.ExPASy.ScanProsite-module.html#read,Method Bio.File.UndoHandle.read()=Bio.File.UndoHandle-class.html#read,Function Bio.KEGG.KGML.KGML_parser.read()=Bio.KEGG.KGML.KGML_parser-module.html#read,Function Bio.Medline.read()=Bio.Medline-module.html#read,Function Bio.Motif.Parsers.AlignAce.read()=Bio.Motif.Parsers.AlignAce-module.html#read,Function Bio.Motif.Parsers.MAST.read()=Bio.Motif.Parsers.MAST-module.html#read,Function Bio.Motif.Parsers.MEME.read()=Bio.Motif.Parsers.MEME-module.html#read,Function Bio.Motif.read()=Bio.Motif-module.html#read,Method Bio.NeuralNetwork.Gene.Pattern.PatternIO.read()=Bio.NeuralNetwork.Gene.Pattern.PatternIO-class.html#read,Method Bio.Nexus.Nexus.Nexus.read()=Bio.Nexus.Nexus.Nexus-class.html#read,Function Bio.Phylo.PAML.baseml.read()=Bio.Phylo.PAML.baseml-module.html#read,Function Bio.Phylo.PAML.codeml.read()=Bio.Phylo.PAML.codeml-module.html#read,Function Bio.Phylo.PAML.yn00.read()=Bio.Phylo.PAML.yn00-module.html#read,Method Bio.Phylo.PhyloXMLIO.Parser.read()=Bio.Phylo.PhyloXMLIO.Parser-class.html#read,Function Bio.Phylo.PhyloXMLIO.read()=Bio.Phylo.PhyloXMLIO-module.html#read,Function Bio.Phylo._io.read()=Bio.Phylo._io-module.html#read,Function Bio.PopGen.GenePop.FileParser.read()=Bio.PopGen.GenePop.FileParser-module.html#read,Function Bio.PopGen.GenePop.LargeFileParser.read()=Bio.PopGen.GenePop.LargeFileParser-module.html#read,Function Bio.PopGen.GenePop.read()=Bio.PopGen.GenePop-module.html#read,Function Bio.SearchIO.read()=Bio.SearchIO-module.html#read,Method Bio.SeqIO.SffIO._AddTellHandle.read()=Bio.SeqIO.SffIO._AddTellHandle-class.html#read,Function Bio.SeqIO.read()=Bio.SeqIO-module.html#read,Function Bio.Sequencing.Ace.read()=Bio.Sequencing.Ace-module.html#read,Function Bio.Sequencing.Phd.read()=Bio.Sequencing.Phd-module.html#read,Function Bio.SwissProt.read()=Bio.SwissProt-module.html#read,Function Bio.UniGene.read()=Bio.UniGene-module.html#read,Method Bio.bgzf.BgzfReader.read()=Bio.bgzf.BgzfReader-class.html#read,Function Bio.motifs.alignace.read()=Bio.motifs.alignace-module.html#read,Function Bio.motifs.jaspar.read()=Bio.motifs.jaspar-module.html#read,Function Bio.motifs.mast.read()=Bio.motifs.mast-module.html#read,Function Bio.motifs.meme.read()=Bio.motifs.meme-module.html#read,Function Bio.motifs.read()=Bio.motifs-module.html#read,Function Bio.motifs.transfac.read()=Bio.motifs.transfac-module.html#read"><a title="Bio.Affy.CelFile.read
Bio.AlignIO.read
Bio.Blast.NCBIXML.read
Bio.Cluster.read
Bio.Compass.read
Bio.Emboss.Primer3.read
Bio.Emboss.PrimerSearch.read
Bio.Entrez.Parser.DataHandler.read
Bio.Entrez.read
Bio.ExPASy.Enzyme.read
Bio.ExPASy.Prodoc.read
Bio.ExPASy.Prosite.read
Bio.ExPASy.ScanProsite.read
Bio.File.UndoHandle.read
Bio.KEGG.KGML.KGML_parser.read
Bio.Medline.read
Bio.Motif.Parsers.AlignAce.read
Bio.Motif.Parsers.MAST.read
Bio.Motif.Parsers.MEME.read
Bio.Motif.read
Bio.NeuralNetwork.Gene.Pattern.PatternIO.read
Bio.Nexus.Nexus.Nexus.read
Bio.Phylo.PAML.baseml.read
Bio.Phylo.PAML.codeml.read
Bio.Phylo.PAML.yn00.read
Bio.Phylo.PhyloXMLIO.Parser.read
Bio.Phylo.PhyloXMLIO.read
Bio.Phylo._io.read
Bio.PopGen.GenePop.FileParser.read
Bio.PopGen.GenePop.LargeFileParser.read
Bio.PopGen.GenePop.read
Bio.SearchIO.read
Bio.SeqIO.SffIO._AddTellHandle.read
Bio.SeqIO.read
Bio.Sequencing.Ace.read
Bio.Sequencing.Phd.read
Bio.SwissProt.read
Bio.UniGene.read
Bio.bgzf.BgzfReader.read
Bio.motifs.alignace.read
Bio.motifs.jaspar.read
Bio.motifs.mast.read
Bio.motifs.meme.read
Bio.motifs.read
Bio.motifs.transfac.read" class="py-name" href="#" onclick="return doclink('link-116', 'read', 'link-116');">read</a></tt><tt class="py-op">(</tt><tt class="py-name">handle</tt><tt class="py-op">)</tt> </tt>
<a name="L386"></a><tt class="py-lineno"> 386</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">taxonomic_record</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L387"></a><tt class="py-lineno"> 387</tt>  <tt class="py-line">                <tt class="py-keyword">assert</tt> <tt class="py-name">taxonomic_record</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-string">"TaxId"</tt><tt class="py-op">]</tt> <tt class="py-op">==</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">ncbi_taxon_id</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> \ </tt>
<a name="L388"></a><tt class="py-lineno"> 388</tt>  <tt class="py-line">                    <tt class="py-string">"%s versus %s"</tt> <tt class="py-op">%</tt> <tt class="py-op">(</tt><tt class="py-name">taxonomic_record</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-string">"TaxId"</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L389"></a><tt class="py-lineno"> 389</tt>  <tt class="py-line">                                      <tt class="py-name">ncbi_taxon_id</tt><tt class="py-op">)</tt> </tt>
<a name="L390"></a><tt class="py-lineno"> 390</tt>  <tt class="py-line">                <tt class="py-name">parent_taxon_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-117" class="py-name" targets="Method BioSQL.Loader.DatabaseLoader._get_taxon_id_from_ncbi_lineage()=BioSQL.Loader.DatabaseLoader-class.html#_get_taxon_id_from_ncbi_lineage"><a title="BioSQL.Loader.DatabaseLoader._get_taxon_id_from_ncbi_lineage" class="py-name" href="#" onclick="return doclink('link-117', '_get_taxon_id_from_ncbi_lineage', 'link-117');">_get_taxon_id_from_ncbi_lineage</a></tt><tt class="py-op">(</tt> </tt>
<a name="L391"></a><tt class="py-lineno"> 391</tt>  <tt class="py-line">                    <tt class="py-name">taxonomic_record</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-string">"LineageEx"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L392"></a><tt class="py-lineno"> 392</tt>  <tt class="py-line">                <tt id="link-118" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-118', 'rank', 'link-27');">rank</a></tt> <tt class="py-op">=</tt> <tt class="py-name">taxonomic_record</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-string">"Rank"</tt><tt class="py-op">]</tt> </tt>
<a name="L393"></a><tt class="py-lineno"> 393</tt>  <tt class="py-line">                <tt class="py-name">genetic_code</tt> <tt class="py-op">=</tt> <tt class="py-name">taxonomic_record</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-string">"GeneticCode"</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-string">"GCId"</tt><tt class="py-op">]</tt> </tt>
<a name="L394"></a><tt class="py-lineno"> 394</tt>  <tt class="py-line">                <tt class="py-name">mito_genetic_code</tt> <tt class="py-op">=</tt> <tt class="py-name">taxonomic_record</tt><tt class="py-op">[</tt> </tt>
<a name="L395"></a><tt class="py-lineno"> 395</tt>  <tt class="py-line">                    <tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-string">"MitoGeneticCode"</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-string">"MGCId"</tt><tt class="py-op">]</tt> </tt>
<a name="L396"></a><tt class="py-lineno"> 396</tt>  <tt class="py-line">                <tt class="py-name">species_names</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">(</tt><tt class="py-string">"scientific name"</tt><tt class="py-op">,</tt> </tt>
<a name="L397"></a><tt class="py-lineno"> 397</tt>  <tt class="py-line">                                  <tt class="py-name">taxonomic_record</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-string">"ScientificName"</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">]</tt> </tt>
<a name="L398"></a><tt class="py-lineno"> 398</tt>  <tt class="py-line">                <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L399"></a><tt class="py-lineno"> 399</tt>  <tt class="py-line">                    <tt class="py-keyword">for</tt> <tt class="py-name">name_class</tt><tt class="py-op">,</tt> <tt class="py-name">names</tt> <tt class="py-keyword">in</tt> <tt class="py-name">taxonomic_record</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-string">"OtherNames"</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-119" class="py-name" targets="Method Bio.Crystal.Crystal.items()=Bio.Crystal.Crystal-class.html#items,Method Bio.File._IndexedSeqFileDict.items()=Bio.File._IndexedSeqFileDict-class.html#items,Method Bio.Phylo.PhyloXML.Events.items()=Bio.Phylo.PhyloXML.Events-class.html#items,Variable Bio.SearchIO._model.query.QueryResult.items=Bio.SearchIO._model.query.QueryResult-class.html#items,Method Bio._py3k._ordereddict.OrderedDict.items()=Bio._py3k._ordereddict.OrderedDict-class.html#items,Method BioSQL.BioSeqDatabase.BioSeqDatabase.items()=BioSQL.BioSeqDatabase.BioSeqDatabase-class.html#items,Method BioSQL.BioSeqDatabase.DBServer.items()=BioSQL.BioSeqDatabase.DBServer-class.html#items"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
Bio._py3k._ordereddict.OrderedDict.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-119', 'items', 'link-119');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L400"></a><tt class="py-lineno"> 400</tt>  <tt class="py-line">                        <tt class="py-name">name_class</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-120" class="py-name" targets="Method BioSQL.Loader.DatabaseLoader._fix_name_class()=BioSQL.Loader.DatabaseLoader-class.html#_fix_name_class"><a title="BioSQL.Loader.DatabaseLoader._fix_name_class" class="py-name" href="#" onclick="return doclink('link-120', '_fix_name_class', 'link-120');">_fix_name_class</a></tt><tt class="py-op">(</tt><tt class="py-name">name_class</tt><tt class="py-op">)</tt> </tt>
<a name="L401"></a><tt class="py-lineno"> 401</tt>  <tt class="py-line">                        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">names</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L402"></a><tt class="py-lineno"> 402</tt>  <tt class="py-line">                            <tt class="py-comment"># The Entrez parser seems to return single entry</tt> </tt>
<a name="L403"></a><tt class="py-lineno"> 403</tt>  <tt class="py-line">                            <tt class="py-comment"># lists as just a string which is annoying.</tt> </tt>
<a name="L404"></a><tt class="py-lineno"> 404</tt>  <tt class="py-line">                            <tt class="py-name">names</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">names</tt><tt class="py-op">]</tt> </tt>
<a name="L405"></a><tt class="py-lineno"> 405</tt>  <tt class="py-line">                        <tt class="py-keyword">for</tt> <tt id="link-121" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-121', 'name', 'link-38');">name</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">names</tt><tt class="py-op">:</tt> </tt>
<a name="L406"></a><tt class="py-lineno"> 406</tt>  <tt class="py-line">                            <tt class="py-comment"># Want to ignore complex things like ClassCDE</tt> </tt>
<a name="L407"></a><tt class="py-lineno"> 407</tt>  <tt class="py-line">                            <tt class="py-comment"># entries</tt> </tt>
<a name="L408"></a><tt class="py-lineno"> 408</tt>  <tt class="py-line">                            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-122" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-122', 'name', 'link-38');">name</a></tt><tt class="py-op">,</tt> <tt class="py-name">basestring</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L409"></a><tt class="py-lineno"> 409</tt>  <tt class="py-line">                                <tt class="py-name">species_names</tt><tt class="py-op">.</tt><tt id="link-123" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-123', 'append', 'link-44');">append</a></tt><tt class="py-op">(</tt><tt class="py-op">(</tt><tt class="py-name">name_class</tt><tt class="py-op">,</tt> <tt id="link-124" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-124', 'name', 'link-38');">name</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L410"></a><tt class="py-lineno"> 410</tt>  <tt class="py-line">                <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L411"></a><tt class="py-lineno"> 411</tt>  <tt class="py-line">                    <tt class="py-comment"># OtherNames isn't always present,</tt> </tt>
<a name="L412"></a><tt class="py-lineno"> 412</tt>  <tt class="py-line">                    <tt class="py-comment"># e.g. NCBI taxon 41205, Bromheadia finlaysoniana</tt> </tt>
<a name="L413"></a><tt class="py-lineno"> 413</tt>  <tt class="py-line">                    <tt class="py-keyword">pass</tt> </tt>
<a name="L414"></a><tt class="py-lineno"> 414</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L415"></a><tt class="py-lineno"> 415</tt>  <tt class="py-line">            <tt class="py-keyword">pass</tt> </tt>
<a name="L416"></a><tt class="py-lineno"> 416</tt>  <tt class="py-line">            <tt class="py-comment"># If we are not allowed to go online, we will record the bare minimum;</tt> </tt>
<a name="L417"></a><tt class="py-lineno"> 417</tt>  <tt class="py-line">            <tt class="py-comment"># as long as the NCBI taxon id is present, then (re)running</tt> </tt>
<a name="L418"></a><tt class="py-lineno"> 418</tt>  <tt class="py-line">            <tt class="py-comment"># load_ncbi_taxonomy.pl should fill in the taxonomomy lineage</tt> </tt>
<a name="L419"></a><tt class="py-lineno"> 419</tt>  <tt class="py-line">            <tt class="py-comment"># (and update the species names).</tt> </tt>
<a name="L420"></a><tt class="py-lineno"> 420</tt>  <tt class="py-line">            <tt class="py-comment">#</tt> </tt>
<a name="L421"></a><tt class="py-lineno"> 421</tt>  <tt class="py-line">            <tt class="py-comment"># I am NOT going to try and record the lineage, even if it</tt> </tt>
<a name="L422"></a><tt class="py-lineno"> 422</tt>  <tt class="py-line">            <tt class="py-comment"># is in the record annotation as a list of names, as we won't</tt> </tt>
<a name="L423"></a><tt class="py-lineno"> 423</tt>  <tt class="py-line">            <tt class="py-comment"># know the NCBI taxon IDs for these parent nodes.</tt> </tt>
<a name="L424"></a><tt class="py-lineno"> 424</tt>  <tt class="py-line"> </tt>
<a name="L425"></a><tt class="py-lineno"> 425</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-125" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-125', 'execute', 'link-39');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L426"></a><tt class="py-lineno"> 426</tt>  <tt class="py-line">            <tt class="py-string">"INSERT INTO taxon(parent_taxon_id, ncbi_taxon_id, node_rank,"</tt> </tt>
<a name="L427"></a><tt class="py-lineno"> 427</tt>  <tt class="py-line">            <tt class="py-string">" genetic_code, mito_genetic_code, left_value, right_value)"</tt> </tt>
<a name="L428"></a><tt class="py-lineno"> 428</tt>  <tt class="py-line">            <tt class="py-string">" VALUES (%s, %s, %s, %s, %s, %s, %s)"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">parent_taxon_id</tt><tt class="py-op">,</tt> </tt>
<a name="L429"></a><tt class="py-lineno"> 429</tt>  <tt class="py-line">                                                     <tt class="py-name">ncbi_taxon_id</tt><tt class="py-op">,</tt> </tt>
<a name="L430"></a><tt class="py-lineno"> 430</tt>  <tt class="py-line">                                                     <tt id="link-126" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-126', 'rank', 'link-27');">rank</a></tt><tt class="py-op">,</tt> </tt>
<a name="L431"></a><tt class="py-lineno"> 431</tt>  <tt class="py-line">                                                     <tt class="py-name">genetic_code</tt><tt class="py-op">,</tt> </tt>
<a name="L432"></a><tt class="py-lineno"> 432</tt>  <tt class="py-line">                                                     <tt class="py-name">mito_genetic_code</tt><tt class="py-op">,</tt> </tt>
<a name="L433"></a><tt class="py-lineno"> 433</tt>  <tt class="py-line">                                                     <tt class="py-name">None</tt><tt class="py-op">,</tt> </tt>
<a name="L434"></a><tt class="py-lineno"> 434</tt>  <tt class="py-line">                                                     <tt class="py-name">None</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L435"></a><tt class="py-lineno"> 435</tt>  <tt class="py-line">        <tt class="py-name">taxon_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-127" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.last_id
BioSQL.DBUtils.Generic_dbutils.last_id
BioSQL.DBUtils.Mysql_dbutils.last_id
BioSQL.DBUtils._PostgreSQL_dbutils.last_id" class="py-name" href="#" onclick="return doclink('link-127', 'last_id', 'link-42');">last_id</a></tt><tt class="py-op">(</tt><tt class="py-string">"taxon"</tt><tt class="py-op">)</tt> </tt>
<a name="L436"></a><tt class="py-lineno"> 436</tt>  <tt class="py-line"> </tt>
<a name="L437"></a><tt class="py-lineno"> 437</tt>  <tt class="py-line">        <tt class="py-comment"># Record the scientific name, common name, etc</tt> </tt>
<a name="L438"></a><tt class="py-lineno"> 438</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">name_class</tt><tt class="py-op">,</tt> <tt id="link-128" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-128', 'name', 'link-38');">name</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">species_names</tt><tt class="py-op">:</tt> </tt>
<a name="L439"></a><tt class="py-lineno"> 439</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-129" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-129', 'execute', 'link-39');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L440"></a><tt class="py-lineno"> 440</tt>  <tt class="py-line">                <tt class="py-string">"INSERT INTO taxon_name(taxon_id, name, name_class)"</tt> </tt>
<a name="L441"></a><tt class="py-lineno"> 441</tt>  <tt class="py-line">                <tt class="py-string">" VALUES (%s, %s, %s)"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">taxon_id</tt><tt class="py-op">,</tt> </tt>
<a name="L442"></a><tt class="py-lineno"> 442</tt>  <tt class="py-line">                                         <tt id="link-130" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-130', 'name', 'link-38');">name</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">255</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L443"></a><tt class="py-lineno"> 443</tt>  <tt class="py-line">                                         <tt class="py-name">name_class</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L444"></a><tt class="py-lineno"> 444</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">taxon_id</tt> </tt>
</div><a name="L445"></a><tt class="py-lineno"> 445</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader._get_taxon_id_from_ncbi_lineage"></a><div id="DatabaseLoader._get_taxon_id_from_ncbi_lineage-def"><a name="L446"></a><tt class="py-lineno"> 446</tt> <a class="py-toggle" href="#" id="DatabaseLoader._get_taxon_id_from_ncbi_lineage-toggle" onclick="return toggle('DatabaseLoader._get_taxon_id_from_ncbi_lineage');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#_get_taxon_id_from_ncbi_lineage">_get_taxon_id_from_ncbi_lineage</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">taxonomic_lineage</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader._get_taxon_id_from_ncbi_lineage-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader._get_taxon_id_from_ncbi_lineage-expanded"><a name="L447"></a><tt class="py-lineno"> 447</tt>  <tt class="py-line">        <tt class="py-docstring">"""This is recursive! (PRIVATE).</tt> </tt>
<a name="L448"></a><tt class="py-lineno"> 448</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L449"></a><tt class="py-lineno"> 449</tt>  <tt class="py-line"><tt class="py-docstring">        taxonomic_lineage - list of taxonomy dictionaries from Bio.Entrez</tt> </tt>
<a name="L450"></a><tt class="py-lineno"> 450</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L451"></a><tt class="py-lineno"> 451</tt>  <tt class="py-line"><tt class="py-docstring">        First dictionary in list is the taxonomy root, highest would be the species.</tt> </tt>
<a name="L452"></a><tt class="py-lineno"> 452</tt>  <tt class="py-line"><tt class="py-docstring">        Each dictionary includes:</tt> </tt>
<a name="L453"></a><tt class="py-lineno"> 453</tt>  <tt class="py-line"><tt class="py-docstring">        - TaxID (string, NCBI taxon id)</tt> </tt>
<a name="L454"></a><tt class="py-lineno"> 454</tt>  <tt class="py-line"><tt class="py-docstring">        - Rank (string, e.g. "species", "genus", ..., "phylum", ...)</tt> </tt>
<a name="L455"></a><tt class="py-lineno"> 455</tt>  <tt class="py-line"><tt class="py-docstring">        - ScientificName (string)</tt> </tt>
<a name="L456"></a><tt class="py-lineno"> 456</tt>  <tt class="py-line"><tt class="py-docstring">        (and that is all at the time of writing)</tt> </tt>
<a name="L457"></a><tt class="py-lineno"> 457</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L458"></a><tt class="py-lineno"> 458</tt>  <tt class="py-line"><tt class="py-docstring">        This method will record all the lineage given, returning the taxon id</tt> </tt>
<a name="L459"></a><tt class="py-lineno"> 459</tt>  <tt class="py-line"><tt class="py-docstring">        (database key, not NCBI taxon id) of the final entry (the species).</tt> </tt>
<a name="L460"></a><tt class="py-lineno"> 460</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L461"></a><tt class="py-lineno"> 461</tt>  <tt class="py-line">        <tt class="py-name">ncbi_taxon_id</tt> <tt class="py-op">=</tt> <tt class="py-name">taxonomic_lineage</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-string">"TaxId"</tt><tt class="py-op">]</tt> </tt>
<a name="L462"></a><tt class="py-lineno"> 462</tt>  <tt class="py-line"> </tt>
<a name="L463"></a><tt class="py-lineno"> 463</tt>  <tt class="py-line">        <tt class="py-comment"># Is this in the database already?  Check the taxon table...</tt> </tt>
<a name="L464"></a><tt class="py-lineno"> 464</tt>  <tt class="py-line">        <tt class="py-name">taxon_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-131" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0" class="py-name" href="#" onclick="return doclink('link-131', 'execute_and_fetch_col0', 'link-37');">execute_and_fetch_col0</a></tt><tt class="py-op">(</tt> </tt>
<a name="L465"></a><tt class="py-lineno"> 465</tt>  <tt class="py-line">            <tt class="py-string">"SELECT taxon_id FROM taxon"</tt> </tt>
<a name="L466"></a><tt class="py-lineno"> 466</tt>  <tt class="py-line">            <tt class="py-string">" WHERE ncbi_taxon_id=%s"</tt> <tt class="py-op">%</tt> <tt class="py-name">ncbi_taxon_id</tt><tt class="py-op">)</tt> </tt>
<a name="L467"></a><tt class="py-lineno"> 467</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">taxon_id</tt><tt class="py-op">:</tt> </tt>
<a name="L468"></a><tt class="py-lineno"> 468</tt>  <tt class="py-line">            <tt class="py-comment"># we could verify that the Scientific Name etc in the database</tt> </tt>
<a name="L469"></a><tt class="py-lineno"> 469</tt>  <tt class="py-line">            <tt class="py-comment"># is the same and update it or print a warning if not...</tt> </tt>
<a name="L470"></a><tt class="py-lineno"> 470</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">taxon_id</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L471"></a><tt class="py-lineno"> 471</tt>  <tt class="py-line">                <tt class="py-keyword">assert</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">taxon_id</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt> </tt>
<a name="L472"></a><tt class="py-lineno"> 472</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">taxon_id</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L473"></a><tt class="py-lineno"> 473</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L474"></a><tt class="py-lineno"> 474</tt>  <tt class="py-line">                <tt class="py-keyword">return</tt> <tt class="py-name">taxon_id</tt> </tt>
<a name="L475"></a><tt class="py-lineno"> 475</tt>  <tt class="py-line"> </tt>
<a name="L476"></a><tt class="py-lineno"> 476</tt>  <tt class="py-line">        <tt class="py-comment"># We have to record this.</tt> </tt>
<a name="L477"></a><tt class="py-lineno"> 477</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">taxonomic_lineage</tt><tt class="py-op">)</tt> <tt class="py-op">&gt;</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt> </tt>
<a name="L478"></a><tt class="py-lineno"> 478</tt>  <tt class="py-line">            <tt class="py-comment"># Use recursion to find out the taxon id (database key) of the</tt> </tt>
<a name="L479"></a><tt class="py-lineno"> 479</tt>  <tt class="py-line">            <tt class="py-comment"># parent.</tt> </tt>
<a name="L480"></a><tt class="py-lineno"> 480</tt>  <tt class="py-line">            <tt class="py-name">parent_taxon_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-132" class="py-name"><a title="BioSQL.Loader.DatabaseLoader._get_taxon_id_from_ncbi_lineage" class="py-name" href="#" onclick="return doclink('link-132', '_get_taxon_id_from_ncbi_lineage', 'link-117');">_get_taxon_id_from_ncbi_lineage</a></tt><tt class="py-op">(</tt> </tt>
<a name="L481"></a><tt class="py-lineno"> 481</tt>  <tt class="py-line">                <tt class="py-name">taxonomic_lineage</tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt> </tt>
<a name="L482"></a><tt class="py-lineno"> 482</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt id="link-133" class="py-name"><a title="Bio._py3k._is_int_or_long" class="py-name" href="#" onclick="return doclink('link-133', '_is_int_or_long', 'link-13');">_is_int_or_long</a></tt><tt class="py-op">(</tt><tt class="py-name">parent_taxon_id</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">parent_taxon_id</tt><tt class="py-op">)</tt> </tt>
<a name="L483"></a><tt class="py-lineno"> 483</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L484"></a><tt class="py-lineno"> 484</tt>  <tt class="py-line">            <tt class="py-name">parent_taxon_id</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L485"></a><tt class="py-lineno"> 485</tt>  <tt class="py-line"> </tt>
<a name="L486"></a><tt class="py-lineno"> 486</tt>  <tt class="py-line">        <tt class="py-comment"># INSERT new taxon</tt> </tt>
<a name="L487"></a><tt class="py-lineno"> 487</tt>  <tt class="py-line">        <tt id="link-134" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-134', 'rank', 'link-27');">rank</a></tt> <tt class="py-op">=</tt> <tt class="py-name">taxonomic_lineage</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-135" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-135', 'get', 'link-25');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"Rank"</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
<a name="L488"></a><tt class="py-lineno"> 488</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-136" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-136', 'execute', 'link-39');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L489"></a><tt class="py-lineno"> 489</tt>  <tt class="py-line">            <tt class="py-string">"INSERT INTO taxon(ncbi_taxon_id, parent_taxon_id, node_rank)"</tt> </tt>
<a name="L490"></a><tt class="py-lineno"> 490</tt>  <tt class="py-line">            <tt class="py-string">" VALUES (%s, %s, %s)"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">ncbi_taxon_id</tt><tt class="py-op">,</tt> <tt class="py-name">parent_taxon_id</tt><tt class="py-op">,</tt> <tt id="link-137" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-137', 'rank', 'link-27');">rank</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L491"></a><tt class="py-lineno"> 491</tt>  <tt class="py-line">        <tt class="py-name">taxon_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-138" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.last_id
BioSQL.DBUtils.Generic_dbutils.last_id
BioSQL.DBUtils.Mysql_dbutils.last_id
BioSQL.DBUtils._PostgreSQL_dbutils.last_id" class="py-name" href="#" onclick="return doclink('link-138', 'last_id', 'link-42');">last_id</a></tt><tt class="py-op">(</tt><tt class="py-string">"taxon"</tt><tt class="py-op">)</tt> </tt>
<a name="L492"></a><tt class="py-lineno"> 492</tt>  <tt class="py-line">        <tt class="py-keyword">assert</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">taxon_id</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">int</tt><tt class="py-op">,</tt> <tt id="link-139" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.long()=Bio.Phylo.PhyloXMLIO.Writer-class.html#long"><a title="Bio.Phylo.PhyloXMLIO.Writer.long" class="py-name" href="#" onclick="return doclink('link-139', 'long', 'link-139');">long</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt class="py-name">repr</tt><tt class="py-op">(</tt><tt class="py-name">taxon_id</tt><tt class="py-op">)</tt> </tt>
<a name="L493"></a><tt class="py-lineno"> 493</tt>  <tt class="py-line">        <tt class="py-comment"># ... and its name in taxon_name</tt> </tt>
<a name="L494"></a><tt class="py-lineno"> 494</tt>  <tt class="py-line">        <tt id="link-140" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.scientific_name" class="py-name" href="#" onclick="return doclink('link-140', 'scientific_name', 'link-63');">scientific_name</a></tt> <tt class="py-op">=</tt> <tt class="py-name">taxonomic_lineage</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-141" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-141', 'get', 'link-25');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"ScientificName"</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
<a name="L495"></a><tt class="py-lineno"> 495</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-142" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.scientific_name" class="py-name" href="#" onclick="return doclink('link-142', 'scientific_name', 'link-63');">scientific_name</a></tt><tt class="py-op">:</tt> </tt>
<a name="L496"></a><tt class="py-lineno"> 496</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-143" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-143', 'execute', 'link-39');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L497"></a><tt class="py-lineno"> 497</tt>  <tt class="py-line">                <tt class="py-string">"INSERT INTO taxon_name(taxon_id, name, name_class)"</tt> </tt>
<a name="L498"></a><tt class="py-lineno"> 498</tt>  <tt class="py-line">                <tt class="py-string">" VALUES (%s, %s, 'scientific name')"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">taxon_id</tt><tt class="py-op">,</tt> </tt>
<a name="L499"></a><tt class="py-lineno"> 499</tt>  <tt class="py-line">                                                        <tt id="link-144" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.scientific_name" class="py-name" href="#" onclick="return doclink('link-144', 'scientific_name', 'link-63');">scientific_name</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-number">255</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L500"></a><tt class="py-lineno"> 500</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">taxon_id</tt> </tt>
</div><a name="L501"></a><tt class="py-lineno"> 501</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader._load_bioentry_table"></a><div id="DatabaseLoader._load_bioentry_table-def"><a name="L502"></a><tt class="py-lineno"> 502</tt> <a class="py-toggle" href="#" id="DatabaseLoader._load_bioentry_table-toggle" onclick="return toggle('DatabaseLoader._load_bioentry_table');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#_load_bioentry_table">_load_bioentry_table</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader._load_bioentry_table-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader._load_bioentry_table-expanded"><a name="L503"></a><tt class="py-lineno"> 503</tt>  <tt class="py-line">        <tt class="py-docstring">"""Fill the bioentry table with sequence information (PRIVATE).</tt> </tt>
<a name="L504"></a><tt class="py-lineno"> 504</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L505"></a><tt class="py-lineno"> 505</tt>  <tt class="py-line"><tt class="py-docstring">        record - SeqRecord object to add to the database.</tt> </tt>
<a name="L506"></a><tt class="py-lineno"> 506</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L507"></a><tt class="py-lineno"> 507</tt>  <tt class="py-line">        <tt class="py-comment"># get the pertinent info and insert it</tt> </tt>
<a name="L508"></a><tt class="py-lineno"> 508</tt>  <tt class="py-line"> </tt>
<a name="L509"></a><tt class="py-lineno"> 509</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-145" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-145', 'id', 'link-114');">id</a></tt><tt class="py-op">.</tt><tt id="link-146" class="py-name" targets="Method Bio.Crystal.Chain.count()=Bio.Crystal.Chain-class.html#count,Method Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count()=Bio.NeuralNetwork.Gene.Pattern.PatternRepository-class.html#count,Method Bio.Seq.MutableSeq.count()=Bio.Seq.MutableSeq-class.html#count,Method Bio.Seq.Seq.count()=Bio.Seq.Seq-class.html#count,Method Bio.Seq.UnknownSeq.count()=Bio.Seq.UnknownSeq-class.html#count,Method Bio.motifs.Instances.count()=Bio.motifs.Instances-class.html#count"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-146', 'count', 'link-146');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">"."</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">1</tt><tt class="py-op">:</tt>  <tt class="py-comment"># try to get a version from the id</tt> </tt>
<a name="L510"></a><tt class="py-lineno"> 510</tt>  <tt class="py-line">            <tt class="py-comment"># This assumes the string is something like "XXXXXXXX.123"</tt> </tt>
<a name="L511"></a><tt class="py-lineno"> 511</tt>  <tt class="py-line">            <tt id="link-147" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-147', 'accession', 'link-52');">accession</a></tt><tt class="py-op">,</tt> <tt id="link-148" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-148', 'version', 'link-53');">version</a></tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-149" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-149', 'id', 'link-114');">id</a></tt><tt class="py-op">.</tt><tt id="link-150" class="py-name" targets="Method Bio.Nexus.Trees.Tree.split()=Bio.Nexus.Trees.Tree-class.html#split,Method Bio.Phylo.BaseTree.TreeMixin.split()=Bio.Phylo.BaseTree.TreeMixin-class.html#split,Method Bio.Restriction.Restriction.RestrictionBatch.split()=Bio.Restriction.Restriction.RestrictionBatch-class.html#split,Method Bio.Seq.Seq.split()=Bio.Seq.Seq-class.html#split"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-150', 'split', 'link-150');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">'.'</tt><tt class="py-op">)</tt> </tt>
<a name="L512"></a><tt class="py-lineno"> 512</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L513"></a><tt class="py-lineno"> 513</tt>  <tt class="py-line">                <tt id="link-151" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-151', 'version', 'link-53');">version</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-152" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-152', 'version', 'link-53');">version</a></tt><tt class="py-op">)</tt> </tt>
<a name="L514"></a><tt class="py-lineno"> 514</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt> <tt class="py-name">ValueError</tt><tt class="py-op">:</tt> </tt>
<a name="L515"></a><tt class="py-lineno"> 515</tt>  <tt class="py-line">                <tt id="link-153" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-153', 'accession', 'link-52');">accession</a></tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-154" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-154', 'id', 'link-114');">id</a></tt> </tt>
<a name="L516"></a><tt class="py-lineno"> 516</tt>  <tt class="py-line">                <tt id="link-155" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-155', 'version', 'link-53');">version</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L517"></a><tt class="py-lineno"> 517</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt>  <tt class="py-comment"># otherwise just use a version of 0</tt> </tt>
<a name="L518"></a><tt class="py-lineno"> 518</tt>  <tt class="py-line">            <tt id="link-156" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-156', 'accession', 'link-52');">accession</a></tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-157" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-157', 'id', 'link-114');">id</a></tt> </tt>
<a name="L519"></a><tt class="py-lineno"> 519</tt>  <tt class="py-line">            <tt id="link-158" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-158', 'version', 'link-53');">version</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L520"></a><tt class="py-lineno"> 520</tt>  <tt class="py-line"> </tt>
<a name="L521"></a><tt class="py-lineno"> 521</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"accessions"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-159" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-159', 'annotations', 'link-24');">annotations</a></tt> \ </tt>
<a name="L522"></a><tt class="py-lineno"> 522</tt>  <tt class="py-line">                <tt class="py-keyword">and</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-160" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-160', 'annotations', 'link-24');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"accessions"</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt> \ </tt>
<a name="L523"></a><tt class="py-lineno"> 523</tt>  <tt class="py-line">                <tt class="py-keyword">and</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-161" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-161', 'annotations', 'link-24');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"accessions"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L524"></a><tt class="py-lineno"> 524</tt>  <tt class="py-line">            <tt class="py-comment"># Take the first accession (one if there is more than one)</tt> </tt>
<a name="L525"></a><tt class="py-lineno"> 525</tt>  <tt class="py-line">            <tt id="link-162" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-162', 'accession', 'link-52');">accession</a></tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-163" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-163', 'annotations', 'link-24');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"accessions"</tt><tt class="py-op">]</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L526"></a><tt class="py-lineno"> 526</tt>  <tt class="py-line"> </tt>
<a name="L527"></a><tt class="py-lineno"> 527</tt>  <tt class="py-line">        <tt class="py-comment"># Find the taxon id (this is not just the NCBI Taxon ID)</tt> </tt>
<a name="L528"></a><tt class="py-lineno"> 528</tt>  <tt class="py-line">        <tt class="py-comment"># NOTE - If the species isn't defined in the taxon table,</tt> </tt>
<a name="L529"></a><tt class="py-lineno"> 529</tt>  <tt class="py-line">        <tt class="py-comment"># a new minimal entry is created.</tt> </tt>
<a name="L530"></a><tt class="py-lineno"> 530</tt>  <tt class="py-line">        <tt class="py-name">taxon_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-164" class="py-name" targets="Method BioSQL.Loader.DatabaseLoader._get_taxon_id()=BioSQL.Loader.DatabaseLoader-class.html#_get_taxon_id"><a title="BioSQL.Loader.DatabaseLoader._get_taxon_id" class="py-name" href="#" onclick="return doclink('link-164', '_get_taxon_id', 'link-164');">_get_taxon_id</a></tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">)</tt> </tt>
<a name="L531"></a><tt class="py-lineno"> 531</tt>  <tt class="py-line"> </tt>
<a name="L532"></a><tt class="py-lineno"> 532</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-string">"gi"</tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-165" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-165', 'annotations', 'link-24');">annotations</a></tt><tt class="py-op">:</tt> </tt>
<a name="L533"></a><tt class="py-lineno"> 533</tt>  <tt class="py-line">            <tt class="py-name">identifier</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-166" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-166', 'annotations', 'link-24');">annotations</a></tt><tt class="py-op">[</tt><tt class="py-string">"gi"</tt><tt class="py-op">]</tt> </tt>
<a name="L534"></a><tt class="py-lineno"> 534</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L535"></a><tt class="py-lineno"> 535</tt>  <tt class="py-line">            <tt class="py-name">identifier</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-167" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Component.id
Bio.KEGG.KGML.KGML_pathway.Entry.id
Bio.KEGG.KGML.KGML_pathway.Reaction.id
Bio.Phylo.PhyloXMLIO.Parser.id
Bio.Phylo.PhyloXMLIO.Writer.id
Bio.SearchIO._model.hit.Hit.id
Bio.SearchIO._model.query.QueryResult.id" class="py-name" href="#" onclick="return doclink('link-167', 'id', 'link-114');">id</a></tt> </tt>
<a name="L536"></a><tt class="py-lineno"> 536</tt>  <tt class="py-line"> </tt>
<a name="L537"></a><tt class="py-lineno"> 537</tt>  <tt class="py-line">        <tt class="py-comment"># Allow description and division to default to NULL as in BioPerl.</tt> </tt>
<a name="L538"></a><tt class="py-lineno"> 538</tt>  <tt class="py-line">        <tt id="link-168" class="py-name" targets="Method Bio.Blast.NCBIStandalone._DescriptionConsumer.description()=Bio.Blast.NCBIStandalone._DescriptionConsumer-class.html#description,Method Bio.Phylo.PhyloXMLIO.Writer.description()=Bio.Phylo.PhyloXMLIO.Writer-class.html#description,Variable Bio.SearchIO._model.hit.Hit.description=Bio.SearchIO._model.hit.Hit-class.html#description,Variable Bio.SearchIO._model.query.QueryResult.description=Bio.SearchIO._model.query.QueryResult-class.html#description"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-168', 'description', 'link-168');">description</a></tt> <tt class="py-op">=</tt> <tt class="py-name">getattr</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">,</tt> <tt class="py-string">'description'</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
<a name="L539"></a><tt class="py-lineno"> 539</tt>  <tt class="py-line">        <tt class="py-name">division</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-169" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-169', 'annotations', 'link-24');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-170" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-170', 'get', 'link-25');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"data_file_division"</tt><tt class="py-op">,</tt> <tt class="py-name">None</tt><tt class="py-op">)</tt> </tt>
<a name="L540"></a><tt class="py-lineno"> 540</tt>  <tt class="py-line"> </tt>
<a name="L541"></a><tt class="py-lineno"> 541</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">"""</tt> </tt>
<a name="L542"></a><tt class="py-lineno"> 542</tt>  <tt class="py-line"><tt class="py-string">        INSERT INTO bioentry (</tt> </tt>
<a name="L543"></a><tt class="py-lineno"> 543</tt>  <tt class="py-line"><tt class="py-string">         biodatabase_id,</tt> </tt>
<a name="L544"></a><tt class="py-lineno"> 544</tt>  <tt class="py-line"><tt class="py-string">         taxon_id,</tt> </tt>
<a name="L545"></a><tt class="py-lineno"> 545</tt>  <tt class="py-line"><tt class="py-string">         name,</tt> </tt>
<a name="L546"></a><tt class="py-lineno"> 546</tt>  <tt class="py-line"><tt class="py-string">         accession,</tt> </tt>
<a name="L547"></a><tt class="py-lineno"> 547</tt>  <tt class="py-line"><tt class="py-string">         identifier,</tt> </tt>
<a name="L548"></a><tt class="py-lineno"> 548</tt>  <tt class="py-line"><tt class="py-string">         division,</tt> </tt>
<a name="L549"></a><tt class="py-lineno"> 549</tt>  <tt class="py-line"><tt class="py-string">         description,</tt> </tt>
<a name="L550"></a><tt class="py-lineno"> 550</tt>  <tt class="py-line"><tt class="py-string">         version)</tt> </tt>
<a name="L551"></a><tt class="py-lineno"> 551</tt>  <tt class="py-line"><tt class="py-string">        VALUES (</tt> </tt>
<a name="L552"></a><tt class="py-lineno"> 552</tt>  <tt class="py-line"><tt class="py-string">         %s,</tt> </tt>
<a name="L553"></a><tt class="py-lineno"> 553</tt>  <tt class="py-line"><tt class="py-string">         %s,</tt> </tt>
<a name="L554"></a><tt class="py-lineno"> 554</tt>  <tt class="py-line"><tt class="py-string">         %s,</tt> </tt>
<a name="L555"></a><tt class="py-lineno"> 555</tt>  <tt class="py-line"><tt class="py-string">         %s,</tt> </tt>
<a name="L556"></a><tt class="py-lineno"> 556</tt>  <tt class="py-line"><tt class="py-string">         %s,</tt> </tt>
<a name="L557"></a><tt class="py-lineno"> 557</tt>  <tt class="py-line"><tt class="py-string">         %s,</tt> </tt>
<a name="L558"></a><tt class="py-lineno"> 558</tt>  <tt class="py-line"><tt class="py-string">         %s,</tt> </tt>
<a name="L559"></a><tt class="py-lineno"> 559</tt>  <tt class="py-line"><tt class="py-string">         %s)"""</tt> </tt>
<a name="L560"></a><tt class="py-lineno"> 560</tt>  <tt class="py-line">        <tt class="py-comment"># print self.dbid, taxon_id, record.name, accession, identifier, \</tt> </tt>
<a name="L561"></a><tt class="py-lineno"> 561</tt>  <tt class="py-line">        <tt class="py-comment">#        division, description, version</tt> </tt>
<a name="L562"></a><tt class="py-lineno"> 562</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-171" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-171', 'execute', 'link-39');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbid</tt><tt class="py-op">,</tt> </tt>
<a name="L563"></a><tt class="py-lineno"> 563</tt>  <tt class="py-line">                                   <tt class="py-name">taxon_id</tt><tt class="py-op">,</tt> </tt>
<a name="L564"></a><tt class="py-lineno"> 564</tt>  <tt class="py-line">                                   <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-172" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Entry.name
Bio.KEGG.KGML.KGML_pathway.Pathway.name
Bio.KEGG.KGML.KGML_pathway.Reaction.name
Bio.Phylo.PhyloXMLIO.Writer.name
Bio.codonalign.codonalphabet.CodonAlphabet.name" class="py-name" href="#" onclick="return doclink('link-172', 'name', 'link-38');">name</a></tt><tt class="py-op">,</tt> </tt>
<a name="L565"></a><tt class="py-lineno"> 565</tt>  <tt class="py-line">                                   <tt id="link-173" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-173', 'accession', 'link-52');">accession</a></tt><tt class="py-op">,</tt> </tt>
<a name="L566"></a><tt class="py-lineno"> 566</tt>  <tt class="py-line">                                   <tt class="py-name">identifier</tt><tt class="py-op">,</tt> </tt>
<a name="L567"></a><tt class="py-lineno"> 567</tt>  <tt class="py-line">                                   <tt class="py-name">division</tt><tt class="py-op">,</tt> </tt>
<a name="L568"></a><tt class="py-lineno"> 568</tt>  <tt class="py-line">                                   <tt id="link-174" class="py-name"><a title="Bio.Blast.NCBIStandalone._DescriptionConsumer.description
Bio.Phylo.PhyloXMLIO.Writer.description
Bio.SearchIO._model.hit.Hit.description
Bio.SearchIO._model.query.QueryResult.description" class="py-name" href="#" onclick="return doclink('link-174', 'description', 'link-168');">description</a></tt><tt class="py-op">,</tt> </tt>
<a name="L569"></a><tt class="py-lineno"> 569</tt>  <tt class="py-line">                                   <tt id="link-175" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.version
Bio.Cluster.cluster.version
Bio.GenBank._FeatureConsumer.version
Bio.GenBank._RecordConsumer.version
Bio.motifs.jaspar.Motif.version" class="py-name" href="#" onclick="return doclink('link-175', 'version', 'link-53');">version</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L570"></a><tt class="py-lineno"> 570</tt>  <tt class="py-line">        <tt class="py-comment"># now retrieve the id for the bioentry</tt> </tt>
<a name="L571"></a><tt class="py-lineno"> 571</tt>  <tt class="py-line">        <tt class="py-name">bioentry_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-176" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.last_id
BioSQL.DBUtils.Generic_dbutils.last_id
BioSQL.DBUtils.Mysql_dbutils.last_id
BioSQL.DBUtils._PostgreSQL_dbutils.last_id" class="py-name" href="#" onclick="return doclink('link-176', 'last_id', 'link-42');">last_id</a></tt><tt class="py-op">(</tt><tt class="py-string">'bioentry'</tt><tt class="py-op">)</tt> </tt>
<a name="L572"></a><tt class="py-lineno"> 572</tt>  <tt class="py-line"> </tt>
<a name="L573"></a><tt class="py-lineno"> 573</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">bioentry_id</tt> </tt>
</div><a name="L574"></a><tt class="py-lineno"> 574</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader._load_bioentry_date"></a><div id="DatabaseLoader._load_bioentry_date-def"><a name="L575"></a><tt class="py-lineno"> 575</tt> <a class="py-toggle" href="#" id="DatabaseLoader._load_bioentry_date-toggle" onclick="return toggle('DatabaseLoader._load_bioentry_date');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#_load_bioentry_date">_load_bioentry_date</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">,</tt> <tt class="py-param">bioentry_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader._load_bioentry_date-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader._load_bioentry_date-expanded"><a name="L576"></a><tt class="py-lineno"> 576</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add the effective date of the entry into the database.</tt> </tt>
<a name="L577"></a><tt class="py-lineno"> 577</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L578"></a><tt class="py-lineno"> 578</tt>  <tt class="py-line"><tt class="py-docstring">        record - a SeqRecord object with an annotated date</tt> </tt>
<a name="L579"></a><tt class="py-lineno"> 579</tt>  <tt class="py-line"><tt class="py-docstring">        bioentry_id - corresponding database identifier</tt> </tt>
<a name="L580"></a><tt class="py-lineno"> 580</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L581"></a><tt class="py-lineno"> 581</tt>  <tt class="py-line">        <tt class="py-comment"># dates are GenBank style, like:</tt> </tt>
<a name="L582"></a><tt class="py-lineno"> 582</tt>  <tt class="py-line">        <tt class="py-comment"># 14-SEP-2000</tt> </tt>
<a name="L583"></a><tt class="py-lineno"> 583</tt>  <tt class="py-line">        <tt id="link-177" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.date()=Bio.GenBank._FeatureConsumer-class.html#date,Method Bio.GenBank._RecordConsumer.date()=Bio.GenBank._RecordConsumer-class.html#date,Method Bio.Phylo.PhyloXMLIO.Parser.date()=Bio.Phylo.PhyloXMLIO.Parser-class.html#date,Method Bio.Phylo.PhyloXMLIO.Writer.date()=Bio.Phylo.PhyloXMLIO.Writer-class.html#date"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-177', 'date', 'link-177');">date</a></tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-178" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-178', 'annotations', 'link-24');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-179" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-179', 'get', 'link-25');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">"date"</tt><tt class="py-op">,</tt> </tt>
<a name="L584"></a><tt class="py-lineno"> 584</tt>  <tt class="py-line">                                      <tt class="py-name">strftime</tt><tt class="py-op">(</tt><tt class="py-string">"%d-%b-%Y"</tt><tt class="py-op">,</tt> <tt class="py-name">gmtime</tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-180" class="py-name" targets="Method Bio.Seq.Seq.upper()=Bio.Seq.Seq-class.html#upper,Method Bio.Seq.UnknownSeq.upper()=Bio.Seq.UnknownSeq-class.html#upper,Method Bio.SeqRecord.SeqRecord.upper()=Bio.SeqRecord.SeqRecord-class.html#upper"><a title="Bio.Seq.Seq.upper
Bio.Seq.UnknownSeq.upper
Bio.SeqRecord.SeqRecord.upper" class="py-name" href="#" onclick="return doclink('link-180', 'upper', 'link-180');">upper</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L585"></a><tt class="py-lineno"> 585</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-181" class="py-name"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-181', 'date', 'link-177');">date</a></tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L586"></a><tt class="py-lineno"> 586</tt>  <tt class="py-line">            <tt id="link-182" class="py-name"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-182', 'date', 'link-177');">date</a></tt> <tt class="py-op">=</tt> <tt id="link-183" class="py-name"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-183', 'date', 'link-177');">date</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L587"></a><tt class="py-lineno"> 587</tt>  <tt class="py-line">        <tt class="py-name">annotation_tags_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-184" class="py-name" targets="Method BioSQL.Loader.DatabaseLoader._get_ontology_id()=BioSQL.Loader.DatabaseLoader-class.html#_get_ontology_id"><a title="BioSQL.Loader.DatabaseLoader._get_ontology_id" class="py-name" href="#" onclick="return doclink('link-184', '_get_ontology_id', 'link-184');">_get_ontology_id</a></tt><tt class="py-op">(</tt><tt class="py-string">"Annotation Tags"</tt><tt class="py-op">)</tt> </tt>
<a name="L588"></a><tt class="py-lineno"> 588</tt>  <tt class="py-line">        <tt class="py-name">date_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-185" class="py-name" targets="Method BioSQL.Loader.DatabaseLoader._get_term_id()=BioSQL.Loader.DatabaseLoader-class.html#_get_term_id"><a title="BioSQL.Loader.DatabaseLoader._get_term_id" class="py-name" href="#" onclick="return doclink('link-185', '_get_term_id', 'link-185');">_get_term_id</a></tt><tt class="py-op">(</tt><tt class="py-string">"date_changed"</tt><tt class="py-op">,</tt> <tt class="py-name">annotation_tags_id</tt><tt class="py-op">)</tt> </tt>
<a name="L589"></a><tt class="py-lineno"> 589</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">r"INSERT INTO bioentry_qualifier_value"</tt> \ </tt>
<a name="L590"></a><tt class="py-lineno"> 590</tt>  <tt class="py-line">              <tt class="py-string">r" (bioentry_id, term_id, value, rank)"</tt> \ </tt>
<a name="L591"></a><tt class="py-lineno"> 591</tt>  <tt class="py-line">              <tt class="py-string">r" VALUES (%s, %s, %s, 1)"</tt> </tt>
<a name="L592"></a><tt class="py-lineno"> 592</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-186" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-186', 'execute', 'link-39');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">bioentry_id</tt><tt class="py-op">,</tt> <tt class="py-name">date_id</tt><tt class="py-op">,</tt> <tt id="link-187" class="py-name"><a title="Bio.GenBank._FeatureConsumer.date
Bio.GenBank._RecordConsumer.date
Bio.Phylo.PhyloXMLIO.Parser.date
Bio.Phylo.PhyloXMLIO.Writer.date" class="py-name" href="#" onclick="return doclink('link-187', 'date', 'link-177');">date</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L593"></a><tt class="py-lineno"> 593</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader._load_biosequence"></a><div id="DatabaseLoader._load_biosequence-def"><a name="L594"></a><tt class="py-lineno"> 594</tt> <a class="py-toggle" href="#" id="DatabaseLoader._load_biosequence-toggle" onclick="return toggle('DatabaseLoader._load_biosequence');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#_load_biosequence">_load_biosequence</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">,</tt> <tt class="py-param">bioentry_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader._load_biosequence-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader._load_biosequence-expanded"><a name="L595"></a><tt class="py-lineno"> 595</tt>  <tt class="py-line">        <tt class="py-docstring">"""Record a SeqRecord's sequence and alphabet in the database (PRIVATE).</tt> </tt>
<a name="L596"></a><tt class="py-lineno"> 596</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L597"></a><tt class="py-lineno"> 597</tt>  <tt class="py-line"><tt class="py-docstring">        record - a SeqRecord object with a seq property</tt> </tt>
<a name="L598"></a><tt class="py-lineno"> 598</tt>  <tt class="py-line"><tt class="py-docstring">        bioentry_id - corresponding database identifier</tt> </tt>
<a name="L599"></a><tt class="py-lineno"> 599</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L600"></a><tt class="py-lineno"> 600</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-188" class="py-name" targets="Variable Bio.SeqRecord.SeqRecord.seq=Bio.SeqRecord.SeqRecord-class.html#seq,Variable BioSQL.BioSeq.DBSeqRecord.seq=BioSQL.BioSeq.DBSeqRecord-class.html#seq"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-188', 'seq', 'link-188');">seq</a></tt> <tt class="py-keyword">is</tt> <tt class="py-name">None</tt><tt class="py-op">:</tt> </tt>
<a name="L601"></a><tt class="py-lineno"> 601</tt>  <tt class="py-line">            <tt class="py-comment"># The biosequence table entry is optional, so if we haven't</tt> </tt>
<a name="L602"></a><tt class="py-lineno"> 602</tt>  <tt class="py-line">            <tt class="py-comment"># got a sequence, we don't need to write to the table.</tt> </tt>
<a name="L603"></a><tt class="py-lineno"> 603</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L604"></a><tt class="py-lineno"> 604</tt>  <tt class="py-line"> </tt>
<a name="L605"></a><tt class="py-lineno"> 605</tt>  <tt class="py-line">        <tt class="py-comment"># determine the string representation of the alphabet</tt> </tt>
<a name="L606"></a><tt class="py-lineno"> 606</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-189" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-189', 'seq', 'link-188');">seq</a></tt><tt class="py-op">.</tt><tt id="link-190" class="py-name" targets="Variable Bio.SearchIO._model.hsp.HSP.alphabet=Bio.SearchIO._model.hsp.HSP-class.html#alphabet,Variable Bio.SearchIO._model.hsp.HSPFragment.alphabet=Bio.SearchIO._model.hsp.HSPFragment-class.html#alphabet"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-190', 'alphabet', 'link-190');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-191" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-191', 'Alphabet', 'link-1');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-192" class="py-name" targets="Class Bio.Alphabet.DNAAlphabet=Bio.Alphabet.DNAAlphabet-class.html"><a title="Bio.Alphabet.DNAAlphabet" class="py-name" href="#" onclick="return doclink('link-192', 'DNAAlphabet', 'link-192');">DNAAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L607"></a><tt class="py-lineno"> 607</tt>  <tt class="py-line">            <tt id="link-193" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-193', 'alphabet', 'link-190');">alphabet</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"dna"</tt> </tt>
<a name="L608"></a><tt class="py-lineno"> 608</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-194" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-194', 'seq', 'link-188');">seq</a></tt><tt class="py-op">.</tt><tt id="link-195" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-195', 'alphabet', 'link-190');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-196" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-196', 'Alphabet', 'link-1');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-197" class="py-name" targets="Class Bio.Alphabet.RNAAlphabet=Bio.Alphabet.RNAAlphabet-class.html"><a title="Bio.Alphabet.RNAAlphabet" class="py-name" href="#" onclick="return doclink('link-197', 'RNAAlphabet', 'link-197');">RNAAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L609"></a><tt class="py-lineno"> 609</tt>  <tt class="py-line">            <tt id="link-198" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-198', 'alphabet', 'link-190');">alphabet</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"rna"</tt> </tt>
<a name="L610"></a><tt class="py-lineno"> 610</tt>  <tt class="py-line">        <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-199" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-199', 'seq', 'link-188');">seq</a></tt><tt class="py-op">.</tt><tt id="link-200" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-200', 'alphabet', 'link-190');">alphabet</a></tt><tt class="py-op">,</tt> <tt id="link-201" class="py-name"><a title="Bio.Alphabet
Bio.Alphabet.Alphabet" class="py-name" href="#" onclick="return doclink('link-201', 'Alphabet', 'link-1');">Alphabet</a></tt><tt class="py-op">.</tt><tt id="link-202" class="py-name" targets="Class Bio.Alphabet.ProteinAlphabet=Bio.Alphabet.ProteinAlphabet-class.html"><a title="Bio.Alphabet.ProteinAlphabet" class="py-name" href="#" onclick="return doclink('link-202', 'ProteinAlphabet', 'link-202');">ProteinAlphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L611"></a><tt class="py-lineno"> 611</tt>  <tt class="py-line">            <tt id="link-203" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-203', 'alphabet', 'link-190');">alphabet</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"protein"</tt> </tt>
<a name="L612"></a><tt class="py-lineno"> 612</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L613"></a><tt class="py-lineno"> 613</tt>  <tt class="py-line">            <tt id="link-204" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-204', 'alphabet', 'link-190');">alphabet</a></tt> <tt class="py-op">=</tt> <tt class="py-string">"unknown"</tt> </tt>
<a name="L614"></a><tt class="py-lineno"> 614</tt>  <tt class="py-line"> </tt>
<a name="L615"></a><tt class="py-lineno"> 615</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-205" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-205', 'seq', 'link-188');">seq</a></tt><tt class="py-op">,</tt> <tt id="link-206" class="py-name"><a title="Bio.Seq.UnknownSeq" class="py-name" href="#" onclick="return doclink('link-206', 'UnknownSeq', 'link-10');">UnknownSeq</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L616"></a><tt class="py-lineno"> 616</tt>  <tt class="py-line">            <tt class="py-name">seq_str</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L617"></a><tt class="py-lineno"> 617</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L618"></a><tt class="py-lineno"> 618</tt>  <tt class="py-line">            <tt class="py-name">seq_str</tt> <tt class="py-op">=</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-207" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-207', 'seq', 'link-188');">seq</a></tt><tt class="py-op">)</tt> </tt>
<a name="L619"></a><tt class="py-lineno"> 619</tt>  <tt class="py-line"> </tt>
<a name="L620"></a><tt class="py-lineno"> 620</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">r"INSERT INTO biosequence (bioentry_id, version, "</tt> \ </tt>
<a name="L621"></a><tt class="py-lineno"> 621</tt>  <tt class="py-line">              <tt class="py-string">r"length, seq, alphabet) "</tt> \ </tt>
<a name="L622"></a><tt class="py-lineno"> 622</tt>  <tt class="py-line">              <tt class="py-string">r"VALUES (%s, 0, %s, %s, %s)"</tt> </tt>
<a name="L623"></a><tt class="py-lineno"> 623</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-208" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-208', 'execute', 'link-39');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">bioentry_id</tt><tt class="py-op">,</tt> </tt>
<a name="L624"></a><tt class="py-lineno"> 624</tt>  <tt class="py-line">                                   <tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-209" class="py-name"><a title="Bio.SeqRecord.SeqRecord.seq
BioSQL.BioSeq.DBSeqRecord.seq" class="py-name" href="#" onclick="return doclink('link-209', 'seq', 'link-188');">seq</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> </tt>
<a name="L625"></a><tt class="py-lineno"> 625</tt>  <tt class="py-line">                                   <tt class="py-name">seq_str</tt><tt class="py-op">,</tt> </tt>
<a name="L626"></a><tt class="py-lineno"> 626</tt>  <tt class="py-line">                                   <tt id="link-210" class="py-name"><a title="Bio.SearchIO._model.hsp.HSP.alphabet
Bio.SearchIO._model.hsp.HSPFragment.alphabet" class="py-name" href="#" onclick="return doclink('link-210', 'alphabet', 'link-190');">alphabet</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L627"></a><tt class="py-lineno"> 627</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader._load_comment"></a><div id="DatabaseLoader._load_comment-def"><a name="L628"></a><tt class="py-lineno"> 628</tt> <a class="py-toggle" href="#" id="DatabaseLoader._load_comment-toggle" onclick="return toggle('DatabaseLoader._load_comment');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#_load_comment">_load_comment</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">,</tt> <tt class="py-param">bioentry_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader._load_comment-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader._load_comment-expanded"><a name="L629"></a><tt class="py-lineno"> 629</tt>  <tt class="py-line">        <tt class="py-docstring">"""Record a SeqRecord's annotated comment in the database (PRIVATE).</tt> </tt>
<a name="L630"></a><tt class="py-lineno"> 630</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L631"></a><tt class="py-lineno"> 631</tt>  <tt class="py-line"><tt class="py-docstring">        record - a SeqRecord object with an annotated comment</tt> </tt>
<a name="L632"></a><tt class="py-lineno"> 632</tt>  <tt class="py-line"><tt class="py-docstring">        bioentry_id - corresponding database identifier</tt> </tt>
<a name="L633"></a><tt class="py-lineno"> 633</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L634"></a><tt class="py-lineno"> 634</tt>  <tt class="py-line">        <tt class="py-name">comments</tt> <tt class="py-op">=</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-211" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-211', 'annotations', 'link-24');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-212" class="py-name"><a title="Bio.Crystal.Crystal.get
Bio.Data.CodonTable.AmbiguousForwardTable.get
Bio.File._IndexedSeqFileDict.get
Bio.File._IndexedSeqFileProxy.get
Bio.File._SQLiteManySeqFilesDict.get
Bio.Restriction.Restriction.RestrictionBatch.get
Bio.SearchIO._index.SearchIndexer.get
Bio.SeqIO._index.SeqFileRandomAccess.get
Bio.SeqIO._index.SffRandomAccess.get
Bio.SeqIO._index.SffTrimedRandomAccess.get
Bio.SeqIO._index.UniprotRandomAccess.get" class="py-name" href="#" onclick="return doclink('link-212', 'get', 'link-25');">get</a></tt><tt class="py-op">(</tt><tt class="py-string">'comment'</tt><tt class="py-op">)</tt> </tt>
<a name="L635"></a><tt class="py-lineno"> 635</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">comments</tt><tt class="py-op">:</tt> </tt>
<a name="L636"></a><tt class="py-lineno"> 636</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> </tt>
<a name="L637"></a><tt class="py-lineno"> 637</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">comments</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L638"></a><tt class="py-lineno"> 638</tt>  <tt class="py-line">            <tt class="py-comment"># It should be a string then...</tt> </tt>
<a name="L639"></a><tt class="py-lineno"> 639</tt>  <tt class="py-line">            <tt class="py-name">comments</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">comments</tt><tt class="py-op">]</tt> </tt>
<a name="L640"></a><tt class="py-lineno"> 640</tt>  <tt class="py-line"> </tt>
<a name="L641"></a><tt class="py-lineno"> 641</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-213" class="py-name" targets="Method Bio.Crystal.Chain.index()=Bio.Crystal.Chain-class.html#index,Method Bio.SCOP.Raf.SeqMap.index()=Bio.SCOP.Raf.SeqMap-class.html#index,Method Bio.SearchIO._model.hit.Hit.index()=Bio.SearchIO._model.hit.Hit-class.html#index,Method Bio.SearchIO._model.query.QueryResult.index()=Bio.SearchIO._model.query.QueryResult-class.html#index,Function Bio.SearchIO.index()=Bio.SearchIO-module.html#index,Method Bio.Seq.MutableSeq.index()=Bio.Seq.MutableSeq-class.html#index,Function Bio.SeqIO.index()=Bio.SeqIO-module.html#index"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-213', 'index', 'link-213');">index</a></tt><tt class="py-op">,</tt> <tt id="link-214" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.comment()=Bio.GenBank._FeatureConsumer-class.html#comment,Method Bio.GenBank._RecordConsumer.comment()=Bio.GenBank._RecordConsumer-class.html#comment"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-214', 'comment', 'link-214');">comment</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt class="py-name">comments</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L642"></a><tt class="py-lineno"> 642</tt>  <tt class="py-line">            <tt id="link-215" class="py-name"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-215', 'comment', 'link-214');">comment</a></tt> <tt class="py-op">=</tt> <tt id="link-216" class="py-name"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-216', 'comment', 'link-214');">comment</a></tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'\n'</tt><tt class="py-op">,</tt> <tt class="py-string">' '</tt><tt class="py-op">)</tt> </tt>
<a name="L643"></a><tt class="py-lineno"> 643</tt>  <tt class="py-line">            <tt class="py-comment"># TODO - Store each line as a separate entry?  This would preserve</tt> </tt>
<a name="L644"></a><tt class="py-lineno"> 644</tt>  <tt class="py-line">            <tt class="py-comment"># the newlines, but we should check BioPerl etc to be consistent.</tt> </tt>
<a name="L645"></a><tt class="py-lineno"> 645</tt>  <tt class="py-line">            <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">"INSERT INTO comment (bioentry_id, comment_text, rank)"</tt> \ </tt>
<a name="L646"></a><tt class="py-lineno"> 646</tt>  <tt class="py-line">                  <tt class="py-string">" VALUES (%s, %s, %s)"</tt> </tt>
<a name="L647"></a><tt class="py-lineno"> 647</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-217" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-217', 'execute', 'link-39');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">bioentry_id</tt><tt class="py-op">,</tt> <tt id="link-218" class="py-name"><a title="Bio.GenBank._FeatureConsumer.comment
Bio.GenBank._RecordConsumer.comment" class="py-name" href="#" onclick="return doclink('link-218', 'comment', 'link-214');">comment</a></tt><tt class="py-op">,</tt> <tt id="link-219" class="py-name"><a title="Bio.Crystal.Chain.index
Bio.SCOP.Raf.SeqMap.index
Bio.SearchIO._model.hit.Hit.index
Bio.SearchIO._model.query.QueryResult.index
Bio.SearchIO.index
Bio.Seq.MutableSeq.index
Bio.SeqIO.index" class="py-name" href="#" onclick="return doclink('link-219', 'index', 'link-213');">index</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L648"></a><tt class="py-lineno"> 648</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader._load_annotations"></a><div id="DatabaseLoader._load_annotations-def"><a name="L649"></a><tt class="py-lineno"> 649</tt> <a class="py-toggle" href="#" id="DatabaseLoader._load_annotations-toggle" onclick="return toggle('DatabaseLoader._load_annotations');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#_load_annotations">_load_annotations</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">,</tt> <tt class="py-param">bioentry_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader._load_annotations-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader._load_annotations-expanded"><a name="L650"></a><tt class="py-lineno"> 650</tt>  <tt class="py-line">        <tt class="py-docstring">"""Record a SeqRecord's misc annotations in the database (PRIVATE).</tt> </tt>
<a name="L651"></a><tt class="py-lineno"> 651</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L652"></a><tt class="py-lineno"> 652</tt>  <tt class="py-line"><tt class="py-docstring">        The annotation strings are recorded in the bioentry_qualifier_value</tt> </tt>
<a name="L653"></a><tt class="py-lineno"> 653</tt>  <tt class="py-line"><tt class="py-docstring">        table, except for special cases like the reference, comment and</tt> </tt>
<a name="L654"></a><tt class="py-lineno"> 654</tt>  <tt class="py-line"><tt class="py-docstring">        taxonomy which are handled with their own tables.</tt> </tt>
<a name="L655"></a><tt class="py-lineno"> 655</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L656"></a><tt class="py-lineno"> 656</tt>  <tt class="py-line"><tt class="py-docstring">        record - a SeqRecord object with an annotations dictionary</tt> </tt>
<a name="L657"></a><tt class="py-lineno"> 657</tt>  <tt class="py-line"><tt class="py-docstring">        bioentry_id - corresponding database identifier</tt> </tt>
<a name="L658"></a><tt class="py-lineno"> 658</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L659"></a><tt class="py-lineno"> 659</tt>  <tt class="py-line">        <tt class="py-name">mono_sql</tt> <tt class="py-op">=</tt> <tt class="py-string">"INSERT INTO bioentry_qualifier_value"</tt> \ </tt>
<a name="L660"></a><tt class="py-lineno"> 660</tt>  <tt class="py-line">                   <tt class="py-string">"(bioentry_id, term_id, value)"</tt> \ </tt>
<a name="L661"></a><tt class="py-lineno"> 661</tt>  <tt class="py-line">                   <tt class="py-string">" VALUES (%s, %s, %s)"</tt> </tt>
<a name="L662"></a><tt class="py-lineno"> 662</tt>  <tt class="py-line">        <tt class="py-name">many_sql</tt> <tt class="py-op">=</tt> <tt class="py-string">"INSERT INTO bioentry_qualifier_value"</tt> \ </tt>
<a name="L663"></a><tt class="py-lineno"> 663</tt>  <tt class="py-line">                   <tt class="py-string">"(bioentry_id, term_id, value, rank)"</tt> \ </tt>
<a name="L664"></a><tt class="py-lineno"> 664</tt>  <tt class="py-line">                   <tt class="py-string">" VALUES (%s, %s, %s, %s)"</tt> </tt>
<a name="L665"></a><tt class="py-lineno"> 665</tt>  <tt class="py-line">        <tt class="py-name">tag_ontology_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-220" class="py-name"><a title="BioSQL.Loader.DatabaseLoader._get_ontology_id" class="py-name" href="#" onclick="return doclink('link-220', '_get_ontology_id', 'link-184');">_get_ontology_id</a></tt><tt class="py-op">(</tt><tt class="py-string">'Annotation Tags'</tt><tt class="py-op">)</tt> </tt>
<a name="L666"></a><tt class="py-lineno"> 666</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">key</tt><tt class="py-op">,</tt> <tt id="link-221" class="py-name" targets="Method Bio.Phylo.PhyloXMLIO.Writer.value()=Bio.Phylo.PhyloXMLIO.Writer-class.html#value"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-221', 'value', 'link-221');">value</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-222" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.annotations" class="py-name" href="#" onclick="return doclink('link-222', 'annotations', 'link-24');">annotations</a></tt><tt class="py-op">.</tt><tt id="link-223" class="py-name"><a title="Bio.Crystal.Crystal.items
Bio.File._IndexedSeqFileDict.items
Bio.Phylo.PhyloXML.Events.items
Bio.SearchIO._model.query.QueryResult.items
Bio._py3k._ordereddict.OrderedDict.items
BioSQL.BioSeqDatabase.BioSeqDatabase.items
BioSQL.BioSeqDatabase.DBServer.items" class="py-name" href="#" onclick="return doclink('link-223', 'items', 'link-119');">items</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L667"></a><tt class="py-lineno"> 667</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">key</tt> <tt class="py-keyword">in</tt> <tt class="py-op">[</tt><tt class="py-string">"references"</tt><tt class="py-op">,</tt> <tt class="py-string">"comment"</tt><tt class="py-op">,</tt> <tt class="py-string">"ncbi_taxid"</tt><tt class="py-op">,</tt> <tt class="py-string">"date"</tt><tt class="py-op">]</tt><tt class="py-op">:</tt> </tt>
<a name="L668"></a><tt class="py-lineno"> 668</tt>  <tt class="py-line">                <tt class="py-comment"># Handled separately</tt> </tt>
<a name="L669"></a><tt class="py-lineno"> 669</tt>  <tt class="py-line">                <tt class="py-keyword">continue</tt> </tt>
<a name="L670"></a><tt class="py-lineno"> 670</tt>  <tt class="py-line">            <tt class="py-name">term_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-224" class="py-name"><a title="BioSQL.Loader.DatabaseLoader._get_term_id" class="py-name" href="#" onclick="return doclink('link-224', '_get_term_id', 'link-185');">_get_term_id</a></tt><tt class="py-op">(</tt><tt class="py-name">key</tt><tt class="py-op">,</tt> <tt class="py-name">ontology_id</tt><tt class="py-op">=</tt><tt class="py-name">tag_ontology_id</tt><tt class="py-op">)</tt> </tt>
<a name="L671"></a><tt class="py-lineno"> 671</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-225" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-225', 'value', 'link-221');">value</a></tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-226" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-226', 'value', 'link-221');">value</a></tt><tt class="py-op">,</tt> <tt class="py-name">tuple</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L672"></a><tt class="py-lineno"> 672</tt>  <tt class="py-line">                <tt id="link-227" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-227', 'rank', 'link-27');">rank</a></tt> <tt class="py-op">=</tt> <tt class="py-number">0</tt> </tt>
<a name="L673"></a><tt class="py-lineno"> 673</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt id="link-228" class="py-name" targets="Variable Bio.KEGG.KGML.KGML_pathway.Reaction.entry=Bio.KEGG.KGML.KGML_pathway.Reaction-class.html#entry,Function Bio.TogoWS.entry()=Bio.TogoWS-module.html#entry"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-228', 'entry', 'link-228');">entry</a></tt> <tt class="py-keyword">in</tt> <tt id="link-229" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-229', 'value', 'link-221');">value</a></tt><tt class="py-op">:</tt> </tt>
<a name="L674"></a><tt class="py-lineno"> 674</tt>  <tt class="py-line">                    <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-230" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-230', 'entry', 'link-228');">entry</a></tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-231" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-231', 'entry', 'link-228');">entry</a></tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L675"></a><tt class="py-lineno"> 675</tt>  <tt class="py-line">                        <tt class="py-comment"># Easy case</tt> </tt>
<a name="L676"></a><tt class="py-lineno"> 676</tt>  <tt class="py-line">                        <tt id="link-232" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-232', 'rank', 'link-27');">rank</a></tt> <tt class="py-op">+=</tt> <tt class="py-number">1</tt> </tt>
<a name="L677"></a><tt class="py-lineno"> 677</tt>  <tt class="py-line">                        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-233" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-233', 'execute', 'link-39');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">many_sql</tt><tt class="py-op">,</tt> </tt>
<a name="L678"></a><tt class="py-lineno"> 678</tt>  <tt class="py-line">                                             <tt class="py-op">(</tt><tt class="py-name">bioentry_id</tt><tt class="py-op">,</tt> <tt class="py-name">term_id</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-234" class="py-name"><a title="Bio.KEGG.KGML.KGML_pathway.Reaction.entry
Bio.TogoWS.entry" class="py-name" href="#" onclick="return doclink('link-234', 'entry', 'link-228');">entry</a></tt><tt class="py-op">)</tt><tt class="py-op">,</tt> <tt id="link-235" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-235', 'rank', 'link-27');">rank</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L679"></a><tt class="py-lineno"> 679</tt>  <tt class="py-line">                    <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L680"></a><tt class="py-lineno"> 680</tt>  <tt class="py-line">                        <tt class="py-keyword">pass</tt> </tt>
<a name="L681"></a><tt class="py-lineno"> 681</tt>  <tt class="py-line">                        <tt class="py-comment"># print "Ignoring annotation '%s' sub-entry of type '%s'" \</tt> </tt>
<a name="L682"></a><tt class="py-lineno"> 682</tt>  <tt class="py-line">                        <tt class="py-comment">#      % (key, str(type(entry)))</tt> </tt>
<a name="L683"></a><tt class="py-lineno"> 683</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-236" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-236', 'value', 'link-221');">value</a></tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">)</tt> <tt class="py-keyword">or</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-237" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-237', 'value', 'link-221');">value</a></tt><tt class="py-op">,</tt> <tt class="py-name">int</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L684"></a><tt class="py-lineno"> 684</tt>  <tt class="py-line">                <tt class="py-comment"># Have a simple single entry, leave rank as the DB default</tt> </tt>
<a name="L685"></a><tt class="py-lineno"> 685</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-238" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-238', 'execute', 'link-39');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">mono_sql</tt><tt class="py-op">,</tt> </tt>
<a name="L686"></a><tt class="py-lineno"> 686</tt>  <tt class="py-line">                                     <tt class="py-op">(</tt><tt class="py-name">bioentry_id</tt><tt class="py-op">,</tt> <tt class="py-name">term_id</tt><tt class="py-op">,</tt> <tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-239" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-239', 'value', 'link-221');">value</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L687"></a><tt class="py-lineno"> 687</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L688"></a><tt class="py-lineno"> 688</tt>  <tt class="py-line">                <tt class="py-keyword">pass</tt> </tt>
</div><a name="L689"></a><tt class="py-lineno"> 689</tt>  <tt class="py-line">                <tt class="py-comment"># print "Ignoring annotation '%s' entry of type '%s'" \</tt> </tt>
<a name="L690"></a><tt class="py-lineno"> 690</tt>  <tt class="py-line">                <tt class="py-comment">#      % (key, type(value))</tt> </tt>
<a name="L691"></a><tt class="py-lineno"> 691</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader._load_reference"></a><div id="DatabaseLoader._load_reference-def"><a name="L692"></a><tt class="py-lineno"> 692</tt> <a class="py-toggle" href="#" id="DatabaseLoader._load_reference-toggle" onclick="return toggle('DatabaseLoader._load_reference');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#_load_reference">_load_reference</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">reference</tt><tt class="py-op">,</tt> <tt class="py-param">rank</tt><tt class="py-op">,</tt> <tt class="py-param">bioentry_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader._load_reference-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader._load_reference-expanded"><a name="L693"></a><tt class="py-lineno"> 693</tt>  <tt class="py-line">        <tt class="py-docstring">"""Record a SeqRecord's annotated references in the database (PRIVATE).</tt> </tt>
<a name="L694"></a><tt class="py-lineno"> 694</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L695"></a><tt class="py-lineno"> 695</tt>  <tt class="py-line"><tt class="py-docstring">        record - a SeqRecord object with annotated references</tt> </tt>
<a name="L696"></a><tt class="py-lineno"> 696</tt>  <tt class="py-line"><tt class="py-docstring">        bioentry_id - corresponding database identifier</tt> </tt>
<a name="L697"></a><tt class="py-lineno"> 697</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L698"></a><tt class="py-lineno"> 698</tt>  <tt class="py-line"> </tt>
<a name="L699"></a><tt class="py-lineno"> 699</tt>  <tt class="py-line">        <tt class="py-name">refs</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L700"></a><tt class="py-lineno"> 700</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-240" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-240', 'reference', 'link-26');">reference</a></tt><tt class="py-op">.</tt><tt id="link-241" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.medline_id()=Bio.GenBank._FeatureConsumer-class.html#medline_id,Method Bio.GenBank._RecordConsumer.medline_id()=Bio.GenBank._RecordConsumer-class.html#medline_id"><a title="Bio.GenBank._FeatureConsumer.medline_id
Bio.GenBank._RecordConsumer.medline_id" class="py-name" href="#" onclick="return doclink('link-241', 'medline_id', 'link-241');">medline_id</a></tt><tt class="py-op">:</tt> </tt>
<a name="L701"></a><tt class="py-lineno"> 701</tt>  <tt class="py-line">            <tt class="py-name">refs</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-242" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0" class="py-name" href="#" onclick="return doclink('link-242', 'execute_and_fetch_col0', 'link-37');">execute_and_fetch_col0</a></tt><tt class="py-op">(</tt> </tt>
<a name="L702"></a><tt class="py-lineno"> 702</tt>  <tt class="py-line">                <tt class="py-string">"SELECT reference_id"</tt> </tt>
<a name="L703"></a><tt class="py-lineno"> 703</tt>  <tt class="py-line">                <tt class="py-string">"  FROM reference JOIN dbxref USING (dbxref_id)"</tt> </tt>
<a name="L704"></a><tt class="py-lineno"> 704</tt>  <tt class="py-line">                <tt class="py-string">" WHERE dbname = 'MEDLINE' AND accession = %s"</tt><tt class="py-op">,</tt> </tt>
<a name="L705"></a><tt class="py-lineno"> 705</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt id="link-243" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-243', 'reference', 'link-26');">reference</a></tt><tt class="py-op">.</tt><tt id="link-244" class="py-name"><a title="Bio.GenBank._FeatureConsumer.medline_id
Bio.GenBank._RecordConsumer.medline_id" class="py-name" href="#" onclick="return doclink('link-244', 'medline_id', 'link-241');">medline_id</a></tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L706"></a><tt class="py-lineno"> 706</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">refs</tt> <tt class="py-keyword">and</tt> <tt id="link-245" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-245', 'reference', 'link-26');">reference</a></tt><tt class="py-op">.</tt><tt id="link-246" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.pubmed_id()=Bio.GenBank._FeatureConsumer-class.html#pubmed_id,Method Bio.GenBank._RecordConsumer.pubmed_id()=Bio.GenBank._RecordConsumer-class.html#pubmed_id"><a title="Bio.GenBank._FeatureConsumer.pubmed_id
Bio.GenBank._RecordConsumer.pubmed_id" class="py-name" href="#" onclick="return doclink('link-246', 'pubmed_id', 'link-246');">pubmed_id</a></tt><tt class="py-op">:</tt> </tt>
<a name="L707"></a><tt class="py-lineno"> 707</tt>  <tt class="py-line">            <tt class="py-name">refs</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-247" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0" class="py-name" href="#" onclick="return doclink('link-247', 'execute_and_fetch_col0', 'link-37');">execute_and_fetch_col0</a></tt><tt class="py-op">(</tt> </tt>
<a name="L708"></a><tt class="py-lineno"> 708</tt>  <tt class="py-line">                <tt class="py-string">"SELECT reference_id"</tt> </tt>
<a name="L709"></a><tt class="py-lineno"> 709</tt>  <tt class="py-line">                <tt class="py-string">"  FROM reference JOIN dbxref USING (dbxref_id)"</tt> </tt>
<a name="L710"></a><tt class="py-lineno"> 710</tt>  <tt class="py-line">                <tt class="py-string">" WHERE dbname = 'PUBMED' AND accession = %s"</tt><tt class="py-op">,</tt> </tt>
<a name="L711"></a><tt class="py-lineno"> 711</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt id="link-248" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-248', 'reference', 'link-26');">reference</a></tt><tt class="py-op">.</tt><tt id="link-249" class="py-name"><a title="Bio.GenBank._FeatureConsumer.pubmed_id
Bio.GenBank._RecordConsumer.pubmed_id" class="py-name" href="#" onclick="return doclink('link-249', 'pubmed_id', 'link-246');">pubmed_id</a></tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L712"></a><tt class="py-lineno"> 712</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">refs</tt><tt class="py-op">:</tt> </tt>
<a name="L713"></a><tt class="py-lineno"> 713</tt>  <tt class="py-line">            <tt class="py-name">s</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-op">]</tt> </tt>
<a name="L714"></a><tt class="py-lineno"> 714</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt class="py-name">f</tt> <tt class="py-keyword">in</tt> <tt id="link-250" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-250', 'reference', 'link-26');">reference</a></tt><tt class="py-op">.</tt><tt id="link-251" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.authors()=Bio.GenBank._FeatureConsumer-class.html#authors,Method Bio.GenBank._RecordConsumer.authors()=Bio.GenBank._RecordConsumer-class.html#authors"><a title="Bio.GenBank._FeatureConsumer.authors
Bio.GenBank._RecordConsumer.authors" class="py-name" href="#" onclick="return doclink('link-251', 'authors', 'link-251');">authors</a></tt><tt class="py-op">,</tt> <tt id="link-252" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-252', 'reference', 'link-26');">reference</a></tt><tt class="py-op">.</tt><tt id="link-253" class="py-name" targets="Method Bio.Blast.NCBIStandalone._AlignmentConsumer.title()=Bio.Blast.NCBIStandalone._AlignmentConsumer-class.html#title,Method Bio.GenBank._FeatureConsumer.title()=Bio.GenBank._FeatureConsumer-class.html#title,Method Bio.GenBank._RecordConsumer.title()=Bio.GenBank._RecordConsumer-class.html#title"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-253', 'title', 'link-253');">title</a></tt><tt class="py-op">,</tt> <tt id="link-254" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-254', 'reference', 'link-26');">reference</a></tt><tt class="py-op">.</tt><tt id="link-255" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.journal()=Bio.GenBank._FeatureConsumer-class.html#journal,Method Bio.GenBank._RecordConsumer.journal()=Bio.GenBank._RecordConsumer-class.html#journal"><a title="Bio.GenBank._FeatureConsumer.journal
Bio.GenBank._RecordConsumer.journal" class="py-name" href="#" onclick="return doclink('link-255', 'journal', 'link-255');">journal</a></tt><tt class="py-op">:</tt> </tt>
<a name="L715"></a><tt class="py-lineno"> 715</tt>  <tt class="py-line">                <tt class="py-name">s</tt><tt class="py-op">.</tt><tt id="link-256" class="py-name"><a title="Bio.Align.MultipleSeqAlignment.append
Bio.Crystal.Chain.append
Bio.SCOP.Raf.SeqMap.append
Bio.SearchIO._model.hit.Hit.append
Bio.SearchIO._model.query.QueryResult.append
Bio.Seq.MutableSeq.append
Bio.Wise.psw.Alignment.append
Bio.Wise.psw.AlignmentColumn.append" class="py-name" href="#" onclick="return doclink('link-256', 'append', 'link-44');">append</a></tt><tt class="py-op">(</tt><tt class="py-name">f</tt> <tt class="py-keyword">or</tt> <tt class="py-string">"&lt;undef&gt;"</tt><tt class="py-op">)</tt> </tt>
<a name="L716"></a><tt class="py-lineno"> 716</tt>  <tt class="py-line">            <tt class="py-name">crc</tt> <tt class="py-op">=</tt> <tt id="link-257" class="py-name"><a title="Bio.SeqUtils.CheckSum.crc64" class="py-name" href="#" onclick="return doclink('link-257', 'crc64', 'link-5');">crc64</a></tt><tt class="py-op">(</tt><tt class="py-string">""</tt><tt class="py-op">.</tt><tt class="py-name">join</tt><tt class="py-op">(</tt><tt class="py-name">s</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L717"></a><tt class="py-lineno"> 717</tt>  <tt class="py-line">            <tt class="py-name">refs</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-258" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0" class="py-name" href="#" onclick="return doclink('link-258', 'execute_and_fetch_col0', 'link-37');">execute_and_fetch_col0</a></tt><tt class="py-op">(</tt> </tt>
<a name="L718"></a><tt class="py-lineno"> 718</tt>  <tt class="py-line">                <tt class="py-string">"SELECT reference_id FROM reference"</tt> </tt>
<a name="L719"></a><tt class="py-lineno"> 719</tt>  <tt class="py-line">                <tt class="py-string">r" WHERE crc = %s"</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">crc</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L720"></a><tt class="py-lineno"> 720</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">refs</tt><tt class="py-op">:</tt> </tt>
<a name="L721"></a><tt class="py-lineno"> 721</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt id="link-259" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-259', 'reference', 'link-26');">reference</a></tt><tt class="py-op">.</tt><tt id="link-260" class="py-name"><a title="Bio.GenBank._FeatureConsumer.medline_id
Bio.GenBank._RecordConsumer.medline_id" class="py-name" href="#" onclick="return doclink('link-260', 'medline_id', 'link-241');">medline_id</a></tt><tt class="py-op">:</tt> </tt>
<a name="L722"></a><tt class="py-lineno"> 722</tt>  <tt class="py-line">                <tt class="py-name">dbxref_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-261" class="py-name" targets="Method BioSQL.Loader.DatabaseLoader._add_dbxref()=BioSQL.Loader.DatabaseLoader-class.html#_add_dbxref"><a title="BioSQL.Loader.DatabaseLoader._add_dbxref" class="py-name" href="#" onclick="return doclink('link-261', '_add_dbxref', 'link-261');">_add_dbxref</a></tt><tt class="py-op">(</tt><tt class="py-string">"MEDLINE"</tt><tt class="py-op">,</tt> </tt>
<a name="L723"></a><tt class="py-lineno"> 723</tt>  <tt class="py-line">                                             <tt id="link-262" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-262', 'reference', 'link-26');">reference</a></tt><tt class="py-op">.</tt><tt id="link-263" class="py-name"><a title="Bio.GenBank._FeatureConsumer.medline_id
Bio.GenBank._RecordConsumer.medline_id" class="py-name" href="#" onclick="return doclink('link-263', 'medline_id', 'link-241');">medline_id</a></tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L724"></a><tt class="py-lineno"> 724</tt>  <tt class="py-line">            <tt class="py-keyword">elif</tt> <tt id="link-264" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-264', 'reference', 'link-26');">reference</a></tt><tt class="py-op">.</tt><tt id="link-265" class="py-name"><a title="Bio.GenBank._FeatureConsumer.pubmed_id
Bio.GenBank._RecordConsumer.pubmed_id" class="py-name" href="#" onclick="return doclink('link-265', 'pubmed_id', 'link-246');">pubmed_id</a></tt><tt class="py-op">:</tt> </tt>
<a name="L725"></a><tt class="py-lineno"> 725</tt>  <tt class="py-line">                <tt class="py-name">dbxref_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-266" class="py-name"><a title="BioSQL.Loader.DatabaseLoader._add_dbxref" class="py-name" href="#" onclick="return doclink('link-266', '_add_dbxref', 'link-261');">_add_dbxref</a></tt><tt class="py-op">(</tt><tt class="py-string">"PUBMED"</tt><tt class="py-op">,</tt> </tt>
<a name="L726"></a><tt class="py-lineno"> 726</tt>  <tt class="py-line">                                             <tt id="link-267" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-267', 'reference', 'link-26');">reference</a></tt><tt class="py-op">.</tt><tt id="link-268" class="py-name"><a title="Bio.GenBank._FeatureConsumer.pubmed_id
Bio.GenBank._RecordConsumer.pubmed_id" class="py-name" href="#" onclick="return doclink('link-268', 'pubmed_id', 'link-246');">pubmed_id</a></tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
<a name="L727"></a><tt class="py-lineno"> 727</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L728"></a><tt class="py-lineno"> 728</tt>  <tt class="py-line">                <tt class="py-name">dbxref_id</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L729"></a><tt class="py-lineno"> 729</tt>  <tt class="py-line">            <tt id="link-269" class="py-name"><a title="Bio.GenBank._FeatureConsumer.authors
Bio.GenBank._RecordConsumer.authors" class="py-name" href="#" onclick="return doclink('link-269', 'authors', 'link-251');">authors</a></tt> <tt class="py-op">=</tt> <tt id="link-270" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-270', 'reference', 'link-26');">reference</a></tt><tt class="py-op">.</tt><tt id="link-271" class="py-name"><a title="Bio.GenBank._FeatureConsumer.authors
Bio.GenBank._RecordConsumer.authors" class="py-name" href="#" onclick="return doclink('link-271', 'authors', 'link-251');">authors</a></tt> <tt class="py-keyword">or</tt> <tt class="py-name">None</tt> </tt>
<a name="L730"></a><tt class="py-lineno"> 730</tt>  <tt class="py-line">            <tt id="link-272" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-272', 'title', 'link-253');">title</a></tt> <tt class="py-op">=</tt> <tt id="link-273" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-273', 'reference', 'link-26');">reference</a></tt><tt class="py-op">.</tt><tt id="link-274" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-274', 'title', 'link-253');">title</a></tt> <tt class="py-keyword">or</tt> <tt class="py-name">None</tt> </tt>
<a name="L731"></a><tt class="py-lineno"> 731</tt>  <tt class="py-line">            <tt class="py-comment"># The location/journal field cannot be Null, so default</tt> </tt>
<a name="L732"></a><tt class="py-lineno"> 732</tt>  <tt class="py-line">            <tt class="py-comment"># to an empty string rather than None:</tt> </tt>
<a name="L733"></a><tt class="py-lineno"> 733</tt>  <tt class="py-line">            <tt id="link-275" class="py-name"><a title="Bio.GenBank._FeatureConsumer.journal
Bio.GenBank._RecordConsumer.journal" class="py-name" href="#" onclick="return doclink('link-275', 'journal', 'link-255');">journal</a></tt> <tt class="py-op">=</tt> <tt id="link-276" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-276', 'reference', 'link-26');">reference</a></tt><tt class="py-op">.</tt><tt id="link-277" class="py-name"><a title="Bio.GenBank._FeatureConsumer.journal
Bio.GenBank._RecordConsumer.journal" class="py-name" href="#" onclick="return doclink('link-277', 'journal', 'link-255');">journal</a></tt> <tt class="py-keyword">or</tt> <tt class="py-string">""</tt> </tt>
<a name="L734"></a><tt class="py-lineno"> 734</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-278" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-278', 'execute', 'link-39');">execute</a></tt><tt class="py-op">(</tt> </tt>
<a name="L735"></a><tt class="py-lineno"> 735</tt>  <tt class="py-line">                <tt class="py-string">"INSERT INTO reference (dbxref_id, location,"</tt> </tt>
<a name="L736"></a><tt class="py-lineno"> 736</tt>  <tt class="py-line">                <tt class="py-string">" title, authors, crc)"</tt> </tt>
<a name="L737"></a><tt class="py-lineno"> 737</tt>  <tt class="py-line">                <tt class="py-string">" VALUES (%s, %s, %s, %s, %s)"</tt><tt class="py-op">,</tt> </tt>
<a name="L738"></a><tt class="py-lineno"> 738</tt>  <tt class="py-line">                <tt class="py-op">(</tt><tt class="py-name">dbxref_id</tt><tt class="py-op">,</tt> <tt id="link-279" class="py-name"><a title="Bio.GenBank._FeatureConsumer.journal
Bio.GenBank._RecordConsumer.journal" class="py-name" href="#" onclick="return doclink('link-279', 'journal', 'link-255');">journal</a></tt><tt class="py-op">,</tt> <tt id="link-280" class="py-name"><a title="Bio.Blast.NCBIStandalone._AlignmentConsumer.title
Bio.GenBank._FeatureConsumer.title
Bio.GenBank._RecordConsumer.title" class="py-name" href="#" onclick="return doclink('link-280', 'title', 'link-253');">title</a></tt><tt class="py-op">,</tt> </tt>
<a name="L739"></a><tt class="py-lineno"> 739</tt>  <tt class="py-line">                 <tt id="link-281" class="py-name"><a title="Bio.GenBank._FeatureConsumer.authors
Bio.GenBank._RecordConsumer.authors" class="py-name" href="#" onclick="return doclink('link-281', 'authors', 'link-251');">authors</a></tt><tt class="py-op">,</tt> <tt class="py-name">crc</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L740"></a><tt class="py-lineno"> 740</tt>  <tt class="py-line">            <tt class="py-name">reference_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-282" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.last_id
BioSQL.DBUtils.Generic_dbutils.last_id
BioSQL.DBUtils.Mysql_dbutils.last_id
BioSQL.DBUtils._PostgreSQL_dbutils.last_id" class="py-name" href="#" onclick="return doclink('link-282', 'last_id', 'link-42');">last_id</a></tt><tt class="py-op">(</tt><tt class="py-string">"reference"</tt><tt class="py-op">)</tt> </tt>
<a name="L741"></a><tt class="py-lineno"> 741</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L742"></a><tt class="py-lineno"> 742</tt>  <tt class="py-line">            <tt class="py-name">reference_id</tt> <tt class="py-op">=</tt> <tt class="py-name">refs</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L743"></a><tt class="py-lineno"> 743</tt>  <tt class="py-line"> </tt>
<a name="L744"></a><tt class="py-lineno"> 744</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-283" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-283', 'reference', 'link-26');">reference</a></tt><tt class="py-op">.</tt><tt id="link-284" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.location()=Bio.GenBank._FeatureConsumer-class.html#location,Method Bio.GenBank._RecordConsumer.location()=Bio.GenBank._RecordConsumer-class.html#location,Method Bio.Phylo.PhyloXMLIO.Writer.location()=Bio.Phylo.PhyloXMLIO.Writer-class.html#location"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-284', 'location', 'link-284');">location</a></tt><tt class="py-op">:</tt> </tt>
<a name="L745"></a><tt class="py-lineno"> 745</tt>  <tt class="py-line">            <tt id="link-285" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.start=Bio.SeqFeature.CompoundLocation-class.html#start,Variable Bio.SeqFeature.FeatureLocation.start=Bio.SeqFeature.FeatureLocation-class.html#start"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-285', 'start', 'link-285');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-number">1</tt> <tt class="py-op">+</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-286" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-286', 'reference', 'link-26');">reference</a></tt><tt class="py-op">.</tt><tt id="link-287" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-287', 'location', 'link-284');">location</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-288" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-288', 'start', 'link-285');">start</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L746"></a><tt class="py-lineno"> 746</tt>  <tt class="py-line">            <tt id="link-289" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.end=Bio.SeqFeature.CompoundLocation-class.html#end,Variable Bio.SeqFeature.FeatureLocation.end=Bio.SeqFeature.FeatureLocation-class.html#end"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-289', 'end', 'link-289');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt class="py-name">str</tt><tt class="py-op">(</tt><tt id="link-290" class="py-name"><a title="Bio.Blast.NCBIStandalone._HeaderConsumer.reference
Bio.Phylo.PhyloXMLIO.Parser.reference
Bio.Phylo.PhyloXMLIO.Writer.reference" class="py-name" href="#" onclick="return doclink('link-290', 'reference', 'link-26');">reference</a></tt><tt class="py-op">.</tt><tt id="link-291" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-291', 'location', 'link-284');">location</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt><tt class="py-op">.</tt><tt id="link-292" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-292', 'end', 'link-289');">end</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L747"></a><tt class="py-lineno"> 747</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L748"></a><tt class="py-lineno"> 748</tt>  <tt class="py-line">            <tt id="link-293" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-293', 'start', 'link-285');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L749"></a><tt class="py-lineno"> 749</tt>  <tt class="py-line">            <tt id="link-294" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-294', 'end', 'link-289');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L750"></a><tt class="py-lineno"> 750</tt>  <tt class="py-line"> </tt>
<a name="L751"></a><tt class="py-lineno"> 751</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">"INSERT INTO bioentry_reference (bioentry_id, reference_id,"</tt> \ </tt>
<a name="L752"></a><tt class="py-lineno"> 752</tt>  <tt class="py-line">              <tt class="py-string">" start_pos, end_pos, rank)"</tt> \ </tt>
<a name="L753"></a><tt class="py-lineno"> 753</tt>  <tt class="py-line">              <tt class="py-string">" VALUES (%s, %s, %s, %s, %s)"</tt> </tt>
<a name="L754"></a><tt class="py-lineno"> 754</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-295" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-295', 'execute', 'link-39');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">bioentry_id</tt><tt class="py-op">,</tt> <tt class="py-name">reference_id</tt><tt class="py-op">,</tt> </tt>
<a name="L755"></a><tt class="py-lineno"> 755</tt>  <tt class="py-line">                                   <tt id="link-296" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-296', 'start', 'link-285');">start</a></tt><tt class="py-op">,</tt> <tt id="link-297" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-297', 'end', 'link-289');">end</a></tt><tt class="py-op">,</tt> <tt id="link-298" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-298', 'rank', 'link-27');">rank</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div><a name="L756"></a><tt class="py-lineno"> 756</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader._load_seqfeature"></a><div id="DatabaseLoader._load_seqfeature-def"><a name="L757"></a><tt class="py-lineno"> 757</tt> <a class="py-toggle" href="#" id="DatabaseLoader._load_seqfeature-toggle" onclick="return toggle('DatabaseLoader._load_seqfeature');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#_load_seqfeature">_load_seqfeature</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">feature</tt><tt class="py-op">,</tt> <tt class="py-param">feature_rank</tt><tt class="py-op">,</tt> <tt class="py-param">bioentry_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader._load_seqfeature-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader._load_seqfeature-expanded"><a name="L758"></a><tt class="py-lineno"> 758</tt>  <tt class="py-line">        <tt class="py-docstring">"""Load a biopython SeqFeature into the database (PRIVATE).</tt> </tt>
<a name="L759"></a><tt class="py-lineno"> 759</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L760"></a><tt class="py-lineno"> 760</tt>  <tt class="py-line">        <tt class="py-comment"># records loaded from a gff file using BCBio.GFF will contain the value</tt> </tt>
<a name="L761"></a><tt class="py-lineno"> 761</tt>  <tt class="py-line">        <tt class="py-comment"># of the 2nd column of the gff as a feature qualifier. The BioSQL wiki</tt> </tt>
<a name="L762"></a><tt class="py-lineno"> 762</tt>  <tt class="py-line">        <tt class="py-comment"># suggests that the source should not go in with the other feature</tt> </tt>
<a name="L763"></a><tt class="py-lineno"> 763</tt>  <tt class="py-line">        <tt class="py-comment"># mappings but instead be put in the term table</tt> </tt>
<a name="L764"></a><tt class="py-lineno"> 764</tt>  <tt class="py-line">        <tt class="py-comment"># (http://www.biosql.org/wiki/Annotation_Mapping)</tt> </tt>
<a name="L765"></a><tt class="py-lineno"> 765</tt>  <tt class="py-line">        <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L766"></a><tt class="py-lineno"> 766</tt>  <tt class="py-line">            <tt id="link-299" class="py-name" targets="Method Bio.GenBank._FeatureConsumer.source()=Bio.GenBank._FeatureConsumer-class.html#source,Method Bio.GenBank._RecordConsumer.source()=Bio.GenBank._RecordConsumer-class.html#source,Method Bio.Pathway.Network.source()=Bio.Pathway.Network-class.html#source"><a title="Bio.GenBank._FeatureConsumer.source
Bio.GenBank._RecordConsumer.source
Bio.Pathway.Network.source" class="py-name" href="#" onclick="return doclink('link-299', 'source', 'link-299');">source</a></tt> <tt class="py-op">=</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt class="py-name">qualifiers</tt><tt class="py-op">[</tt><tt class="py-string">'source'</tt><tt class="py-op">]</tt> </tt>
<a name="L767"></a><tt class="py-lineno"> 767</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt id="link-300" class="py-name"><a title="Bio.GenBank._FeatureConsumer.source
Bio.GenBank._RecordConsumer.source
Bio.Pathway.Network.source" class="py-name" href="#" onclick="return doclink('link-300', 'source', 'link-299');">source</a></tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L768"></a><tt class="py-lineno"> 768</tt>  <tt class="py-line">                <tt id="link-301" class="py-name"><a title="Bio.GenBank._FeatureConsumer.source
Bio.GenBank._RecordConsumer.source
Bio.Pathway.Network.source" class="py-name" href="#" onclick="return doclink('link-301', 'source', 'link-299');">source</a></tt> <tt class="py-op">=</tt> <tt id="link-302" class="py-name"><a title="Bio.GenBank._FeatureConsumer.source
Bio.GenBank._RecordConsumer.source
Bio.Pathway.Network.source" class="py-name" href="#" onclick="return doclink('link-302', 'source', 'link-299');">source</a></tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L769"></a><tt class="py-lineno"> 769</tt>  <tt class="py-line">            <tt class="py-name">seqfeature_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-303" class="py-name" targets="Method BioSQL.Loader.DatabaseLoader._load_seqfeature_basic()=BioSQL.Loader.DatabaseLoader-class.html#_load_seqfeature_basic"><a title="BioSQL.Loader.DatabaseLoader._load_seqfeature_basic" class="py-name" href="#" onclick="return doclink('link-303', '_load_seqfeature_basic', 'link-303');">_load_seqfeature_basic</a></tt><tt class="py-op">(</tt><tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-304" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-304', 'type', 'link-61');">type</a></tt><tt class="py-op">,</tt> <tt class="py-name">feature_rank</tt><tt class="py-op">,</tt> </tt>
<a name="L770"></a><tt class="py-lineno"> 770</tt>  <tt class="py-line">                                                <tt class="py-name">bioentry_id</tt><tt class="py-op">,</tt> <tt id="link-305" class="py-name"><a title="Bio.GenBank._FeatureConsumer.source
Bio.GenBank._RecordConsumer.source
Bio.Pathway.Network.source" class="py-name" href="#" onclick="return doclink('link-305', 'source', 'link-299');">source</a></tt><tt class="py-op">=</tt><tt id="link-306" class="py-name"><a title="Bio.GenBank._FeatureConsumer.source
Bio.GenBank._RecordConsumer.source
Bio.Pathway.Network.source" class="py-name" href="#" onclick="return doclink('link-306', 'source', 'link-299');">source</a></tt><tt class="py-op">)</tt> </tt>
<a name="L771"></a><tt class="py-lineno"> 771</tt>  <tt class="py-line">        <tt class="py-keyword">except</tt> <tt class="py-name">KeyError</tt><tt class="py-op">:</tt> </tt>
<a name="L772"></a><tt class="py-lineno"> 772</tt>  <tt class="py-line">            <tt class="py-name">seqfeature_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-307" class="py-name"><a title="BioSQL.Loader.DatabaseLoader._load_seqfeature_basic" class="py-name" href="#" onclick="return doclink('link-307', '_load_seqfeature_basic', 'link-303');">_load_seqfeature_basic</a></tt><tt class="py-op">(</tt><tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-308" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.type" class="py-name" href="#" onclick="return doclink('link-308', 'type', 'link-61');">type</a></tt><tt class="py-op">,</tt> <tt class="py-name">feature_rank</tt><tt class="py-op">,</tt> </tt>
<a name="L773"></a><tt class="py-lineno"> 773</tt>  <tt class="py-line">                                                <tt class="py-name">bioentry_id</tt><tt class="py-op">)</tt> </tt>
<a name="L774"></a><tt class="py-lineno"> 774</tt>  <tt class="py-line"> </tt>
<a name="L775"></a><tt class="py-lineno"> 775</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-309" class="py-name" targets="Method BioSQL.Loader.DatabaseLoader._load_seqfeature_locations()=BioSQL.Loader.DatabaseLoader-class.html#_load_seqfeature_locations"><a title="BioSQL.Loader.DatabaseLoader._load_seqfeature_locations" class="py-name" href="#" onclick="return doclink('link-309', '_load_seqfeature_locations', 'link-309');">_load_seqfeature_locations</a></tt><tt class="py-op">(</tt><tt class="py-name">feature</tt><tt class="py-op">,</tt> <tt class="py-name">seqfeature_id</tt><tt class="py-op">)</tt> </tt>
<a name="L776"></a><tt class="py-lineno"> 776</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-310" class="py-name" targets="Method BioSQL.Loader.DatabaseLoader._load_seqfeature_qualifiers()=BioSQL.Loader.DatabaseLoader-class.html#_load_seqfeature_qualifiers"><a title="BioSQL.Loader.DatabaseLoader._load_seqfeature_qualifiers" class="py-name" href="#" onclick="return doclink('link-310', '_load_seqfeature_qualifiers', 'link-310');">_load_seqfeature_qualifiers</a></tt><tt class="py-op">(</tt><tt class="py-name">feature</tt><tt class="py-op">.</tt><tt class="py-name">qualifiers</tt><tt class="py-op">,</tt> <tt class="py-name">seqfeature_id</tt><tt class="py-op">)</tt> </tt>
</div><a name="L777"></a><tt class="py-lineno"> 777</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader._load_seqfeature_basic"></a><div id="DatabaseLoader._load_seqfeature_basic-def"><a name="L778"></a><tt class="py-lineno"> 778</tt> <a class="py-toggle" href="#" id="DatabaseLoader._load_seqfeature_basic-toggle" onclick="return toggle('DatabaseLoader._load_seqfeature_basic');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#_load_seqfeature_basic">_load_seqfeature_basic</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">feature_type</tt><tt class="py-op">,</tt> <tt class="py-param">feature_rank</tt><tt class="py-op">,</tt> <tt class="py-param">bioentry_id</tt><tt class="py-op">,</tt> <tt class="py-param">source</tt><tt class="py-op">=</tt><tt class="py-string">'EMBL/GenBank/SwissProt'</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader._load_seqfeature_basic-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader._load_seqfeature_basic-expanded"><a name="L779"></a><tt class="py-lineno"> 779</tt>  <tt class="py-line">        <tt class="py-docstring">"""Load the first tables of a seqfeature and returns the id (PRIVATE).</tt> </tt>
<a name="L780"></a><tt class="py-lineno"> 780</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L781"></a><tt class="py-lineno"> 781</tt>  <tt class="py-line"><tt class="py-docstring">        This loads the "key" of the seqfeature (ie. CDS, gene) and</tt> </tt>
<a name="L782"></a><tt class="py-lineno"> 782</tt>  <tt class="py-line"><tt class="py-docstring">        the basic seqfeature table itself.</tt> </tt>
<a name="L783"></a><tt class="py-lineno"> 783</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L784"></a><tt class="py-lineno"> 784</tt>  <tt class="py-line">        <tt class="py-name">ontology_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-311" class="py-name"><a title="BioSQL.Loader.DatabaseLoader._get_ontology_id" class="py-name" href="#" onclick="return doclink('link-311', '_get_ontology_id', 'link-184');">_get_ontology_id</a></tt><tt class="py-op">(</tt><tt class="py-string">'SeqFeature Keys'</tt><tt class="py-op">)</tt> </tt>
<a name="L785"></a><tt class="py-lineno"> 785</tt>  <tt class="py-line">        <tt class="py-name">seqfeature_key_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-312" class="py-name"><a title="BioSQL.Loader.DatabaseLoader._get_term_id" class="py-name" href="#" onclick="return doclink('link-312', '_get_term_id', 'link-185');">_get_term_id</a></tt><tt class="py-op">(</tt><tt class="py-name">feature_type</tt><tt class="py-op">,</tt> </tt>
<a name="L786"></a><tt class="py-lineno"> 786</tt>  <tt class="py-line">                                              <tt class="py-name">ontology_id</tt><tt class="py-op">=</tt><tt class="py-name">ontology_id</tt><tt class="py-op">)</tt> </tt>
<a name="L787"></a><tt class="py-lineno"> 787</tt>  <tt class="py-line">        <tt class="py-name">source_cat_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-313" class="py-name"><a title="BioSQL.Loader.DatabaseLoader._get_ontology_id" class="py-name" href="#" onclick="return doclink('link-313', '_get_ontology_id', 'link-184');">_get_ontology_id</a></tt><tt class="py-op">(</tt><tt class="py-string">'SeqFeature Sources'</tt><tt class="py-op">)</tt> </tt>
<a name="L788"></a><tt class="py-lineno"> 788</tt>  <tt class="py-line">        <tt class="py-name">source_term_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-314" class="py-name"><a title="BioSQL.Loader.DatabaseLoader._get_term_id" class="py-name" href="#" onclick="return doclink('link-314', '_get_term_id', 'link-185');">_get_term_id</a></tt><tt class="py-op">(</tt><tt id="link-315" class="py-name"><a title="Bio.GenBank._FeatureConsumer.source
Bio.GenBank._RecordConsumer.source
Bio.Pathway.Network.source" class="py-name" href="#" onclick="return doclink('link-315', 'source', 'link-299');">source</a></tt><tt class="py-op">,</tt> </tt>
<a name="L789"></a><tt class="py-lineno"> 789</tt>  <tt class="py-line">                                           <tt class="py-name">ontology_id</tt><tt class="py-op">=</tt><tt class="py-name">source_cat_id</tt><tt class="py-op">)</tt> </tt>
<a name="L790"></a><tt class="py-lineno"> 790</tt>  <tt class="py-line"> </tt>
<a name="L791"></a><tt class="py-lineno"> 791</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">r"INSERT INTO seqfeature (bioentry_id, type_term_id, "</tt> \ </tt>
<a name="L792"></a><tt class="py-lineno"> 792</tt>  <tt class="py-line">              <tt class="py-string">r"source_term_id, rank) VALUES (%s, %s, %s, %s)"</tt> </tt>
<a name="L793"></a><tt class="py-lineno"> 793</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-316" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-316', 'execute', 'link-39');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">bioentry_id</tt><tt class="py-op">,</tt> <tt class="py-name">seqfeature_key_id</tt><tt class="py-op">,</tt> </tt>
<a name="L794"></a><tt class="py-lineno"> 794</tt>  <tt class="py-line">                                   <tt class="py-name">source_term_id</tt><tt class="py-op">,</tt> <tt class="py-name">feature_rank</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L795"></a><tt class="py-lineno"> 795</tt>  <tt class="py-line">        <tt class="py-name">seqfeature_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-317" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.last_id
BioSQL.DBUtils.Generic_dbutils.last_id
BioSQL.DBUtils.Mysql_dbutils.last_id
BioSQL.DBUtils._PostgreSQL_dbutils.last_id" class="py-name" href="#" onclick="return doclink('link-317', 'last_id', 'link-42');">last_id</a></tt><tt class="py-op">(</tt><tt class="py-string">'seqfeature'</tt><tt class="py-op">)</tt> </tt>
<a name="L796"></a><tt class="py-lineno"> 796</tt>  <tt class="py-line"> </tt>
<a name="L797"></a><tt class="py-lineno"> 797</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">seqfeature_id</tt> </tt>
</div><a name="L798"></a><tt class="py-lineno"> 798</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader._load_seqfeature_locations"></a><div id="DatabaseLoader._load_seqfeature_locations-def"><a name="L799"></a><tt class="py-lineno"> 799</tt> <a class="py-toggle" href="#" id="DatabaseLoader._load_seqfeature_locations-toggle" onclick="return toggle('DatabaseLoader._load_seqfeature_locations');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#_load_seqfeature_locations">_load_seqfeature_locations</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">feature</tt><tt class="py-op">,</tt> <tt class="py-param">seqfeature_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader._load_seqfeature_locations-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader._load_seqfeature_locations-expanded"><a name="L800"></a><tt class="py-lineno"> 800</tt>  <tt class="py-line">        <tt class="py-docstring">"""Load all of the locations for a SeqFeature into tables (PRIVATE).</tt> </tt>
<a name="L801"></a><tt class="py-lineno"> 801</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L802"></a><tt class="py-lineno"> 802</tt>  <tt class="py-line"><tt class="py-docstring">        This adds the locations related to the SeqFeature into the</tt> </tt>
<a name="L803"></a><tt class="py-lineno"> 803</tt>  <tt class="py-line"><tt class="py-docstring">        seqfeature_location table. Fuzzies are not handled right now.</tt> </tt>
<a name="L804"></a><tt class="py-lineno"> 804</tt>  <tt class="py-line"><tt class="py-docstring">        For a simple location, ie (1..2), we have a single table row</tt> </tt>
<a name="L805"></a><tt class="py-lineno"> 805</tt>  <tt class="py-line"><tt class="py-docstring">        with seq_start = 1, seq_end = 2, location_rank = 1.</tt> </tt>
<a name="L806"></a><tt class="py-lineno"> 806</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L807"></a><tt class="py-lineno"> 807</tt>  <tt class="py-line"><tt class="py-docstring">        For split locations, ie (1..2, 3..4, 5..6) we would have three</tt> </tt>
<a name="L808"></a><tt class="py-lineno"> 808</tt>  <tt class="py-line"><tt class="py-docstring">        row tables with::</tt> </tt>
<a name="L809"></a><tt class="py-lineno"> 809</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L810"></a><tt class="py-lineno"> 810</tt>  <tt class="py-line"><tt class="py-docstring">            start = 1, end = 2, rank = 1</tt> </tt>
<a name="L811"></a><tt class="py-lineno"> 811</tt>  <tt class="py-line"><tt class="py-docstring">            start = 3, end = 4, rank = 2</tt> </tt>
<a name="L812"></a><tt class="py-lineno"> 812</tt>  <tt class="py-line"><tt class="py-docstring">            start = 5, end = 6, rank = 3</tt> </tt>
<a name="L813"></a><tt class="py-lineno"> 813</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L814"></a><tt class="py-lineno"> 814</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - Record an ontology for the locations (using location.term_id)</tt> </tt>
<a name="L815"></a><tt class="py-lineno"> 815</tt>  <tt class="py-line">        <tt class="py-comment"># which for now as in BioPerl we leave defaulting to NULL.</tt> </tt>
<a name="L816"></a><tt class="py-lineno"> 816</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-318" class="py-name" targets="Variable Bio.SeqFeature.SeqFeature.location_operator=Bio.SeqFeature.SeqFeature-class.html#location_operator"><a title="Bio.SeqFeature.SeqFeature.location_operator" class="py-name" href="#" onclick="return doclink('link-318', 'location_operator', 'link-318');">location_operator</a></tt> <tt class="py-keyword">and</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-319" class="py-name"><a title="Bio.SeqFeature.SeqFeature.location_operator" class="py-name" href="#" onclick="return doclink('link-319', 'location_operator', 'link-318');">location_operator</a></tt> <tt class="py-op">!=</tt> <tt class="py-string">"join"</tt><tt class="py-op">:</tt> </tt>
<a name="L817"></a><tt class="py-lineno"> 817</tt>  <tt class="py-line">            <tt class="py-comment"># e.g. order locations... we don't record "order" so it</tt> </tt>
<a name="L818"></a><tt class="py-lineno"> 818</tt>  <tt class="py-line">            <tt class="py-comment"># will become a "join" on reloading. What does BioPerl do?</tt> </tt>
<a name="L819"></a><tt class="py-lineno"> 819</tt>  <tt class="py-line">            <tt class="py-keyword">import</tt> <tt class="py-name">warnings</tt> </tt>
<a name="L820"></a><tt class="py-lineno"> 820</tt>  <tt class="py-line">            <tt class="py-keyword">from</tt> <tt id="link-320" class="py-name"><a title="Bio" class="py-name" href="#" onclick="return doclink('link-320', 'Bio', 'link-0');">Bio</a></tt> <tt class="py-keyword">import</tt> <tt id="link-321" class="py-name" targets="Class Bio.BiopythonWarning=Bio.BiopythonWarning-class.html"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-321', 'BiopythonWarning', 'link-321');">BiopythonWarning</a></tt> </tt>
<a name="L821"></a><tt class="py-lineno"> 821</tt>  <tt class="py-line">            <tt class="py-name">warnings</tt><tt class="py-op">.</tt><tt class="py-name">warn</tt><tt class="py-op">(</tt><tt class="py-string">"%s location operators are not fully supported"</tt> </tt>
<a name="L822"></a><tt class="py-lineno"> 822</tt>  <tt class="py-line">                          <tt class="py-op">%</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-322" class="py-name"><a title="Bio.SeqFeature.SeqFeature.location_operator" class="py-name" href="#" onclick="return doclink('link-322', 'location_operator', 'link-318');">location_operator</a></tt><tt class="py-op">,</tt> <tt id="link-323" class="py-name"><a title="Bio.BiopythonWarning" class="py-name" href="#" onclick="return doclink('link-323', 'BiopythonWarning', 'link-321');">BiopythonWarning</a></tt><tt class="py-op">)</tt> </tt>
<a name="L823"></a><tt class="py-lineno"> 823</tt>  <tt class="py-line">        <tt class="py-comment"># This will be a list of length one for simple FeatureLocation:</tt> </tt>
<a name="L824"></a><tt class="py-lineno"> 824</tt>  <tt class="py-line">        <tt id="link-324" class="py-name" targets="Variable Bio.SeqFeature.FeatureLocation.parts=Bio.SeqFeature.FeatureLocation-class.html#parts"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-324', 'parts', 'link-324');">parts</a></tt> <tt class="py-op">=</tt> <tt class="py-name">feature</tt><tt class="py-op">.</tt><tt id="link-325" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-325', 'location', 'link-284');">location</a></tt><tt class="py-op">.</tt><tt id="link-326" class="py-name"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-326', 'parts', 'link-324');">parts</a></tt> </tt>
<a name="L825"></a><tt class="py-lineno"> 825</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-327" class="py-name"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-327', 'parts', 'link-324');">parts</a></tt> <tt class="py-keyword">and</tt> <tt id="link-328" class="py-name" targets="Method Bio.Nexus.Nexus.StepMatrix.set()=Bio.Nexus.Nexus.StepMatrix-class.html#set,Method Bio.PDB.QCPSuperimposer.QCPSuperimposer.set()=Bio.PDB.QCPSuperimposer.QCPSuperimposer-class.html#set,Method Bio.SVDSuperimposer.SVDSuperimposer.set()=Bio.SVDSuperimposer.SVDSuperimposer-class.html#set"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-328', 'set', 'link-328');">set</a></tt><tt class="py-op">(</tt><tt class="py-name">loc</tt><tt class="py-op">.</tt><tt id="link-329" class="py-name" targets="Method Bio.Blast.NCBIStandalone._HSPConsumer.strand()=Bio.Blast.NCBIStandalone._HSPConsumer-class.html#strand,Variable Bio.SeqFeature.CompoundLocation.strand=Bio.SeqFeature.CompoundLocation-class.html#strand,Variable Bio.SeqFeature.FeatureLocation.strand=Bio.SeqFeature.FeatureLocation-class.html#strand,Variable Bio.SeqFeature.SeqFeature.strand=Bio.SeqFeature.SeqFeature-class.html#strand"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-329', 'strand', 'link-329');">strand</a></tt> <tt class="py-keyword">for</tt> <tt class="py-name">loc</tt> <tt class="py-keyword">in</tt> <tt id="link-330" class="py-name"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-330', 'parts', 'link-324');">parts</a></tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt id="link-331" class="py-name"><a title="Bio.Nexus.Nexus.StepMatrix.set
Bio.PDB.QCPSuperimposer.QCPSuperimposer.set
Bio.SVDSuperimposer.SVDSuperimposer.set" class="py-name" href="#" onclick="return doclink('link-331', 'set', 'link-328');">set</a></tt><tt class="py-op">(</tt><tt class="py-op">[</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L826"></a><tt class="py-lineno"> 826</tt>  <tt class="py-line">            <tt class="py-comment"># To mimic prior behaviour of Biopython+BioSQL, reverse order</tt> </tt>
<a name="L827"></a><tt class="py-lineno"> 827</tt>  <tt class="py-line">            <tt id="link-332" class="py-name"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-332', 'parts', 'link-324');">parts</a></tt> <tt class="py-op">=</tt> <tt id="link-333" class="py-name"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-333', 'parts', 'link-324');">parts</a></tt><tt class="py-op">[</tt><tt class="py-op">:</tt><tt class="py-op">:</tt><tt class="py-op">-</tt><tt class="py-number">1</tt><tt class="py-op">]</tt> </tt>
<a name="L828"></a><tt class="py-lineno"> 828</tt>  <tt class="py-line">            <tt class="py-comment"># TODO - Check what BioPerl does; see also BioSeq.py code</tt> </tt>
<a name="L829"></a><tt class="py-lineno"> 829</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-334" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-334', 'rank', 'link-27');">rank</a></tt><tt class="py-op">,</tt> <tt class="py-name">loc</tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt id="link-335" class="py-name"><a title="Bio.SeqFeature.FeatureLocation.parts" class="py-name" href="#" onclick="return doclink('link-335', 'parts', 'link-324');">parts</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L830"></a><tt class="py-lineno"> 830</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-336" class="py-name" targets="Method BioSQL.Loader.DatabaseLoader._insert_location()=BioSQL.Loader.DatabaseLoader-class.html#_insert_location"><a title="BioSQL.Loader.DatabaseLoader._insert_location" class="py-name" href="#" onclick="return doclink('link-336', '_insert_location', 'link-336');">_insert_location</a></tt><tt class="py-op">(</tt><tt class="py-name">loc</tt><tt class="py-op">,</tt> <tt id="link-337" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-337', 'rank', 'link-27');">rank</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> <tt class="py-name">seqfeature_id</tt><tt class="py-op">)</tt> </tt>
</div><a name="L831"></a><tt class="py-lineno"> 831</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader._insert_location"></a><div id="DatabaseLoader._insert_location-def"><a name="L832"></a><tt class="py-lineno"> 832</tt> <a class="py-toggle" href="#" id="DatabaseLoader._insert_location-toggle" onclick="return toggle('DatabaseLoader._insert_location');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#_insert_location">_insert_location</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">location</tt><tt class="py-op">,</tt> <tt class="py-param">rank</tt><tt class="py-op">,</tt> <tt class="py-param">seqfeature_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader._insert_location-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader._insert_location-expanded"><a name="L833"></a><tt class="py-lineno"> 833</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add a location of a SeqFeature to the seqfeature_location table (PRIVATE).</tt> </tt>
<a name="L834"></a><tt class="py-lineno"> 834</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L835"></a><tt class="py-lineno"> 835</tt>  <tt class="py-line"><tt class="py-docstring">        TODO - Add location operator to location_qualifier_value?</tt> </tt>
<a name="L836"></a><tt class="py-lineno"> 836</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L837"></a><tt class="py-lineno"> 837</tt>  <tt class="py-line">        <tt class="py-comment"># convert biopython locations to the 1-based location system</tt> </tt>
<a name="L838"></a><tt class="py-lineno"> 838</tt>  <tt class="py-line">        <tt class="py-comment"># used in bioSQL</tt> </tt>
<a name="L839"></a><tt class="py-lineno"> 839</tt>  <tt class="py-line">        <tt class="py-comment"># XXX This could also handle fuzzies</tt> </tt>
<a name="L840"></a><tt class="py-lineno"> 840</tt>  <tt class="py-line">        <tt id="link-338" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-338', 'start', 'link-285');">start</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-339" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-339', 'location', 'link-284');">location</a></tt><tt class="py-op">.</tt><tt id="link-340" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-340', 'start', 'link-285');">start</a></tt><tt class="py-op">)</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt> </tt>
<a name="L841"></a><tt class="py-lineno"> 841</tt>  <tt class="py-line">        <tt id="link-341" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-341', 'end', 'link-289');">end</a></tt> <tt class="py-op">=</tt> <tt class="py-name">int</tt><tt class="py-op">(</tt><tt id="link-342" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-342', 'location', 'link-284');">location</a></tt><tt class="py-op">.</tt><tt id="link-343" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-343', 'end', 'link-289');">end</a></tt><tt class="py-op">)</tt> </tt>
<a name="L842"></a><tt class="py-lineno"> 842</tt>  <tt class="py-line"> </tt>
<a name="L843"></a><tt class="py-lineno"> 843</tt>  <tt class="py-line">        <tt class="py-comment"># Biopython uses None when we don't know strand information but</tt> </tt>
<a name="L844"></a><tt class="py-lineno"> 844</tt>  <tt class="py-line">        <tt class="py-comment"># BioSQL requires something (non null) and sets this as zero</tt> </tt>
<a name="L845"></a><tt class="py-lineno"> 845</tt>  <tt class="py-line">        <tt class="py-comment"># So we'll use the strand or 0 if Biopython spits out None</tt> </tt>
<a name="L846"></a><tt class="py-lineno"> 846</tt>  <tt class="py-line">        <tt id="link-344" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-344', 'strand', 'link-329');">strand</a></tt> <tt class="py-op">=</tt> <tt id="link-345" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-345', 'location', 'link-284');">location</a></tt><tt class="py-op">.</tt><tt id="link-346" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-346', 'strand', 'link-329');">strand</a></tt> <tt class="py-keyword">or</tt> <tt class="py-number">0</tt> </tt>
<a name="L847"></a><tt class="py-lineno"> 847</tt>  <tt class="py-line"> </tt>
<a name="L848"></a><tt class="py-lineno"> 848</tt>  <tt class="py-line">        <tt class="py-comment"># TODO - Record an ontology term for the location (location.term_id)</tt> </tt>
<a name="L849"></a><tt class="py-lineno"> 849</tt>  <tt class="py-line">        <tt class="py-comment"># which for now like BioPerl we'll leave as NULL.</tt> </tt>
<a name="L850"></a><tt class="py-lineno"> 850</tt>  <tt class="py-line">        <tt class="py-comment"># This might allow us to record "between" positions properly, but I</tt> </tt>
<a name="L851"></a><tt class="py-lineno"> 851</tt>  <tt class="py-line">        <tt class="py-comment"># doesn't really see how it could work for before/after fuzzy positions</tt> </tt>
<a name="L852"></a><tt class="py-lineno"> 852</tt>  <tt class="py-line">        <tt class="py-name">loc_term_id</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L853"></a><tt class="py-lineno"> 853</tt>  <tt class="py-line"> </tt>
<a name="L854"></a><tt class="py-lineno"> 854</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt id="link-347" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-347', 'location', 'link-284');">location</a></tt><tt class="py-op">.</tt><tt id="link-348" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.ref=Bio.SeqFeature.CompoundLocation-class.html#ref,Variable Bio.SeqFeature.SeqFeature.ref=Bio.SeqFeature.SeqFeature-class.html#ref"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-348', 'ref', 'link-348');">ref</a></tt><tt class="py-op">:</tt> </tt>
<a name="L855"></a><tt class="py-lineno"> 855</tt>  <tt class="py-line">            <tt class="py-comment"># sub_feature remote locations when they are in the same db as the current</tt> </tt>
<a name="L856"></a><tt class="py-lineno"> 856</tt>  <tt class="py-line">            <tt class="py-comment"># record do not have a value for ref_db, which the SeqFeature object</tt> </tt>
<a name="L857"></a><tt class="py-lineno"> 857</tt>  <tt class="py-line">            <tt class="py-comment"># stores as None. BioSQL schema requires a varchar and is not NULL</tt> </tt>
<a name="L858"></a><tt class="py-lineno"> 858</tt>  <tt class="py-line">            <tt class="py-name">dbxref_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-349" class="py-name" targets="Method BioSQL.Loader.DatabaseLoader._get_dbxref_id()=BioSQL.Loader.DatabaseLoader-class.html#_get_dbxref_id"><a title="BioSQL.Loader.DatabaseLoader._get_dbxref_id" class="py-name" href="#" onclick="return doclink('link-349', '_get_dbxref_id', 'link-349');">_get_dbxref_id</a></tt><tt class="py-op">(</tt> </tt>
<a name="L859"></a><tt class="py-lineno"> 859</tt>  <tt class="py-line">                <tt id="link-350" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-350', 'location', 'link-284');">location</a></tt><tt class="py-op">.</tt><tt id="link-351" class="py-name" targets="Variable Bio.SeqFeature.CompoundLocation.ref_db=Bio.SeqFeature.CompoundLocation-class.html#ref_db,Variable Bio.SeqFeature.SeqFeature.ref_db=Bio.SeqFeature.SeqFeature-class.html#ref_db"><a title="Bio.SeqFeature.CompoundLocation.ref_db
Bio.SeqFeature.SeqFeature.ref_db" class="py-name" href="#" onclick="return doclink('link-351', 'ref_db', 'link-351');">ref_db</a></tt> <tt class="py-keyword">or</tt> <tt class="py-string">""</tt><tt class="py-op">,</tt> <tt id="link-352" class="py-name"><a title="Bio.GenBank._FeatureConsumer.location
Bio.GenBank._RecordConsumer.location
Bio.Phylo.PhyloXMLIO.Writer.location" class="py-name" href="#" onclick="return doclink('link-352', 'location', 'link-284');">location</a></tt><tt class="py-op">.</tt><tt id="link-353" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.ref
Bio.SeqFeature.SeqFeature.ref" class="py-name" href="#" onclick="return doclink('link-353', 'ref', 'link-348');">ref</a></tt><tt class="py-op">)</tt> </tt>
<a name="L860"></a><tt class="py-lineno"> 860</tt>  <tt class="py-line">        <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L861"></a><tt class="py-lineno"> 861</tt>  <tt class="py-line">            <tt class="py-name">dbxref_id</tt> <tt class="py-op">=</tt> <tt class="py-name">None</tt> </tt>
<a name="L862"></a><tt class="py-lineno"> 862</tt>  <tt class="py-line"> </tt>
<a name="L863"></a><tt class="py-lineno"> 863</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">r"INSERT INTO location (seqfeature_id, dbxref_id, term_id,"</tt> \ </tt>
<a name="L864"></a><tt class="py-lineno"> 864</tt>  <tt class="py-line">              <tt class="py-string">r"start_pos, end_pos, strand, rank) "</tt> \ </tt>
<a name="L865"></a><tt class="py-lineno"> 865</tt>  <tt class="py-line">              <tt class="py-string">r"VALUES (%s, %s, %s, %s, %s, %s, %s)"</tt> </tt>
<a name="L866"></a><tt class="py-lineno"> 866</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-354" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-354', 'execute', 'link-39');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">seqfeature_id</tt><tt class="py-op">,</tt> <tt class="py-name">dbxref_id</tt><tt class="py-op">,</tt> <tt class="py-name">loc_term_id</tt><tt class="py-op">,</tt> </tt>
<a name="L867"></a><tt class="py-lineno"> 867</tt>  <tt class="py-line">                                   <tt id="link-355" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.start
Bio.SeqFeature.FeatureLocation.start" class="py-name" href="#" onclick="return doclink('link-355', 'start', 'link-285');">start</a></tt><tt class="py-op">,</tt> <tt id="link-356" class="py-name"><a title="Bio.SeqFeature.CompoundLocation.end
Bio.SeqFeature.FeatureLocation.end" class="py-name" href="#" onclick="return doclink('link-356', 'end', 'link-289');">end</a></tt><tt class="py-op">,</tt> <tt id="link-357" class="py-name"><a title="Bio.Blast.NCBIStandalone._HSPConsumer.strand
Bio.SeqFeature.CompoundLocation.strand
Bio.SeqFeature.FeatureLocation.strand
Bio.SeqFeature.SeqFeature.strand" class="py-name" href="#" onclick="return doclink('link-357', 'strand', 'link-329');">strand</a></tt><tt class="py-op">,</tt> <tt id="link-358" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-358', 'rank', 'link-27');">rank</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L868"></a><tt class="py-lineno"> 868</tt>  <tt class="py-line"> </tt>
<a name="L869"></a><tt class="py-lineno"> 869</tt>  <tt class="py-line">        <tt class="py-string">"""</tt> </tt>
<a name="L870"></a><tt class="py-lineno"> 870</tt>  <tt class="py-line"><tt class="py-string">        # See Bug 2677</tt> </tt>
<a name="L871"></a><tt class="py-lineno"> 871</tt>  <tt class="py-line"><tt class="py-string">        # TODO - Record the location_operator (e.g. "join" or "order")</tt> </tt>
<a name="L872"></a><tt class="py-lineno"> 872</tt>  <tt class="py-line"><tt class="py-string">        # using the location_qualifier_value table (which we and BioPerl</tt> </tt>
<a name="L873"></a><tt class="py-lineno"> 873</tt>  <tt class="py-line"><tt class="py-string">        # have historically left empty).</tt> </tt>
<a name="L874"></a><tt class="py-lineno"> 874</tt>  <tt class="py-line"><tt class="py-string">        # Note this will need an ontology term for the location qualifer</tt> </tt>
<a name="L875"></a><tt class="py-lineno"> 875</tt>  <tt class="py-line"><tt class="py-string">        # (location_qualifier_value.term_id) for which oddly the schema</tt> </tt>
<a name="L876"></a><tt class="py-lineno"> 876</tt>  <tt class="py-line"><tt class="py-string">        # does not allow NULL.</tt> </tt>
<a name="L877"></a><tt class="py-lineno"> 877</tt>  <tt class="py-line"><tt class="py-string">        if feature.location_operator:</tt> </tt>
<a name="L878"></a><tt class="py-lineno"> 878</tt>  <tt class="py-line"><tt class="py-string">            #e.g. "join" (common),</tt> </tt>
<a name="L879"></a><tt class="py-lineno"> 879</tt>  <tt class="py-line"><tt class="py-string">            #or "order" (see Tests/GenBank/protein_refseq2.gb)</tt> </tt>
<a name="L880"></a><tt class="py-lineno"> 880</tt>  <tt class="py-line"><tt class="py-string">            location_id = self.adaptor.last_id('location')</tt> </tt>
<a name="L881"></a><tt class="py-lineno"> 881</tt>  <tt class="py-line"><tt class="py-string">            loc_qual_term_id = None # Not allowed in BioSQL v1.0.1</tt> </tt>
<a name="L882"></a><tt class="py-lineno"> 882</tt>  <tt class="py-line"><tt class="py-string">            sql = r"INSERT INTO location_qualifier_value" \</tt> </tt>
<a name="L883"></a><tt class="py-lineno"> 883</tt>  <tt class="py-line"><tt class="py-string">                  r"(location_id, term_id, value)" \</tt> </tt>
<a name="L884"></a><tt class="py-lineno"> 884</tt>  <tt class="py-line"><tt class="py-string">                  r"VALUES (%s, %s, %s)"</tt> </tt>
<a name="L885"></a><tt class="py-lineno"> 885</tt>  <tt class="py-line"><tt class="py-string">            self.adaptor.execute(sql, (location_id, loc_qual_term_id,</tt> </tt>
<a name="L886"></a><tt class="py-lineno"> 886</tt>  <tt class="py-line"><tt class="py-string">                                       feature.location_operator))</tt> </tt>
<a name="L887"></a><tt class="py-lineno"> 887</tt>  <tt class="py-line"><tt class="py-string">        """</tt> </tt>
</div><a name="L888"></a><tt class="py-lineno"> 888</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader._load_seqfeature_qualifiers"></a><div id="DatabaseLoader._load_seqfeature_qualifiers-def"><a name="L889"></a><tt class="py-lineno"> 889</tt> <a class="py-toggle" href="#" id="DatabaseLoader._load_seqfeature_qualifiers-toggle" onclick="return toggle('DatabaseLoader._load_seqfeature_qualifiers');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#_load_seqfeature_qualifiers">_load_seqfeature_qualifiers</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">qualifiers</tt><tt class="py-op">,</tt> <tt class="py-param">seqfeature_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader._load_seqfeature_qualifiers-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader._load_seqfeature_qualifiers-expanded"><a name="L890"></a><tt class="py-lineno"> 890</tt>  <tt class="py-line">        <tt class="py-docstring">"""Insert the (key, value) pair qualifiers relating to a feature (PRIVATE).</tt> </tt>
<a name="L891"></a><tt class="py-lineno"> 891</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L892"></a><tt class="py-lineno"> 892</tt>  <tt class="py-line"><tt class="py-docstring">        Qualifiers should be a dictionary of the form:</tt> </tt>
<a name="L893"></a><tt class="py-lineno"> 893</tt>  <tt class="py-line"><tt class="py-docstring">            {key : [value1, value2]}</tt> </tt>
<a name="L894"></a><tt class="py-lineno"> 894</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L895"></a><tt class="py-lineno"> 895</tt>  <tt class="py-line">        <tt class="py-name">tag_ontology_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-359" class="py-name"><a title="BioSQL.Loader.DatabaseLoader._get_ontology_id" class="py-name" href="#" onclick="return doclink('link-359', '_get_ontology_id', 'link-184');">_get_ontology_id</a></tt><tt class="py-op">(</tt><tt class="py-string">'Annotation Tags'</tt><tt class="py-op">)</tt> </tt>
<a name="L896"></a><tt class="py-lineno"> 896</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt class="py-name">qualifier_key</tt> <tt class="py-keyword">in</tt> <tt class="py-name">qualifiers</tt><tt class="py-op">:</tt> </tt>
<a name="L897"></a><tt class="py-lineno"> 897</tt>  <tt class="py-line">            <tt class="py-comment"># Treat db_xref qualifiers differently to sequence annotation</tt> </tt>
<a name="L898"></a><tt class="py-lineno"> 898</tt>  <tt class="py-line">            <tt class="py-comment"># qualifiers by populating the seqfeature_dbxref and dbxref</tt> </tt>
<a name="L899"></a><tt class="py-lineno"> 899</tt>  <tt class="py-line">            <tt class="py-comment"># tables.  Other qualifiers go into the seqfeature_qualifier_value</tt> </tt>
<a name="L900"></a><tt class="py-lineno"> 900</tt>  <tt class="py-line">            <tt class="py-comment"># and (if new) term tables.</tt> </tt>
<a name="L901"></a><tt class="py-lineno"> 901</tt>  <tt class="py-line">            <tt class="py-keyword">if</tt> <tt class="py-name">qualifier_key</tt> <tt class="py-op">!=</tt> <tt class="py-string">'db_xref'</tt><tt class="py-op">:</tt> </tt>
<a name="L902"></a><tt class="py-lineno"> 902</tt>  <tt class="py-line">                <tt class="py-name">qualifier_key_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-360" class="py-name"><a title="BioSQL.Loader.DatabaseLoader._get_term_id" class="py-name" href="#" onclick="return doclink('link-360', '_get_term_id', 'link-185');">_get_term_id</a></tt><tt class="py-op">(</tt><tt class="py-name">qualifier_key</tt><tt class="py-op">,</tt> </tt>
<a name="L903"></a><tt class="py-lineno"> 903</tt>  <tt class="py-line">                                                     <tt class="py-name">ontology_id</tt><tt class="py-op">=</tt><tt class="py-name">tag_ontology_id</tt><tt class="py-op">)</tt> </tt>
<a name="L904"></a><tt class="py-lineno"> 904</tt>  <tt class="py-line">                <tt class="py-comment"># now add all of the values to their table</tt> </tt>
<a name="L905"></a><tt class="py-lineno"> 905</tt>  <tt class="py-line">                <tt class="py-name">entries</tt> <tt class="py-op">=</tt> <tt class="py-name">qualifiers</tt><tt class="py-op">[</tt><tt class="py-name">qualifier_key</tt><tt class="py-op">]</tt> </tt>
<a name="L906"></a><tt class="py-lineno"> 906</tt>  <tt class="py-line">                <tt class="py-keyword">if</tt> <tt class="py-keyword">not</tt> <tt class="py-name">isinstance</tt><tt class="py-op">(</tt><tt class="py-name">entries</tt><tt class="py-op">,</tt> <tt class="py-name">list</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L907"></a><tt class="py-lineno"> 907</tt>  <tt class="py-line">                    <tt class="py-comment"># Could be a plain string, or an int or a float.</tt> </tt>
<a name="L908"></a><tt class="py-lineno"> 908</tt>  <tt class="py-line">                    <tt class="py-comment"># However, we exect a list of strings here.</tt> </tt>
<a name="L909"></a><tt class="py-lineno"> 909</tt>  <tt class="py-line">                    <tt class="py-name">entries</tt> <tt class="py-op">=</tt> <tt class="py-op">[</tt><tt class="py-name">entries</tt><tt class="py-op">]</tt> </tt>
<a name="L910"></a><tt class="py-lineno"> 910</tt>  <tt class="py-line">                <tt class="py-keyword">for</tt> <tt class="py-name">qual_value_rank</tt> <tt class="py-keyword">in</tt> <tt id="link-361" class="py-name"><a title="Bio.Graphics.GenomeDiagram._Diagram.Diagram.range
Bio.Graphics.GenomeDiagram._FeatureSet.FeatureSet.range
Bio.Graphics.GenomeDiagram._Graph.GraphData.range
Bio.Graphics.GenomeDiagram._GraphSet.GraphSet.range
Bio.Graphics.GenomeDiagram._Track.Track.range" class="py-name" href="#" onclick="return doclink('link-361', 'range', 'link-16');">range</a></tt><tt class="py-op">(</tt><tt class="py-name">len</tt><tt class="py-op">(</tt><tt class="py-name">entries</tt><tt class="py-op">)</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L911"></a><tt class="py-lineno"> 911</tt>  <tt class="py-line">                    <tt class="py-name">qualifier_value</tt> <tt class="py-op">=</tt> <tt class="py-name">entries</tt><tt class="py-op">[</tt><tt class="py-name">qual_value_rank</tt><tt class="py-op">]</tt> </tt>
<a name="L912"></a><tt class="py-lineno"> 912</tt>  <tt class="py-line">                    <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">r"INSERT INTO seqfeature_qualifier_value "</tt>\ </tt>
<a name="L913"></a><tt class="py-lineno"> 913</tt>  <tt class="py-line">                          <tt class="py-string">r" (seqfeature_id, term_id, rank, value) VALUES"</tt>\ </tt>
<a name="L914"></a><tt class="py-lineno"> 914</tt>  <tt class="py-line">                          <tt class="py-string">r" (%s, %s, %s, %s)"</tt> </tt>
<a name="L915"></a><tt class="py-lineno"> 915</tt>  <tt class="py-line">                    <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-362" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-362', 'execute', 'link-39');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">seqfeature_id</tt><tt class="py-op">,</tt> </tt>
<a name="L916"></a><tt class="py-lineno"> 916</tt>  <tt class="py-line">                                               <tt class="py-name">qualifier_key_id</tt><tt class="py-op">,</tt> </tt>
<a name="L917"></a><tt class="py-lineno"> 917</tt>  <tt class="py-line">                                               <tt class="py-name">qual_value_rank</tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">,</tt> </tt>
<a name="L918"></a><tt class="py-lineno"> 918</tt>  <tt class="py-line">                                               <tt class="py-name">qualifier_value</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L919"></a><tt class="py-lineno"> 919</tt>  <tt class="py-line">            <tt class="py-keyword">else</tt><tt class="py-op">:</tt> </tt>
<a name="L920"></a><tt class="py-lineno"> 920</tt>  <tt class="py-line">                <tt class="py-comment"># The dbxref_id qualifier/value sets go into the dbxref table</tt> </tt>
<a name="L921"></a><tt class="py-lineno"> 921</tt>  <tt class="py-line">                <tt class="py-comment"># as dbname, accession, version tuples, with dbxref.dbxref_id</tt> </tt>
<a name="L922"></a><tt class="py-lineno"> 922</tt>  <tt class="py-line">                <tt class="py-comment"># being automatically assigned, and into the seqfeature_dbxref</tt> </tt>
<a name="L923"></a><tt class="py-lineno"> 923</tt>  <tt class="py-line">                <tt class="py-comment"># table as seqfeature_id, dbxref_id, and rank tuples</tt> </tt>
<a name="L924"></a><tt class="py-lineno"> 924</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-363" class="py-name" targets="Method BioSQL.Loader.DatabaseLoader._load_seqfeature_dbxref()=BioSQL.Loader.DatabaseLoader-class.html#_load_seqfeature_dbxref"><a title="BioSQL.Loader.DatabaseLoader._load_seqfeature_dbxref" class="py-name" href="#" onclick="return doclink('link-363', '_load_seqfeature_dbxref', 'link-363');">_load_seqfeature_dbxref</a></tt><tt class="py-op">(</tt><tt class="py-name">qualifiers</tt><tt class="py-op">[</tt><tt class="py-name">qualifier_key</tt><tt class="py-op">]</tt><tt class="py-op">,</tt> </tt>
<a name="L925"></a><tt class="py-lineno"> 925</tt>  <tt class="py-line">                                             <tt class="py-name">seqfeature_id</tt><tt class="py-op">)</tt> </tt>
</div><a name="L926"></a><tt class="py-lineno"> 926</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader._load_seqfeature_dbxref"></a><div id="DatabaseLoader._load_seqfeature_dbxref-def"><a name="L927"></a><tt class="py-lineno"> 927</tt> <a class="py-toggle" href="#" id="DatabaseLoader._load_seqfeature_dbxref-toggle" onclick="return toggle('DatabaseLoader._load_seqfeature_dbxref');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#_load_seqfeature_dbxref">_load_seqfeature_dbxref</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">dbxrefs</tt><tt class="py-op">,</tt> <tt class="py-param">seqfeature_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader._load_seqfeature_dbxref-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader._load_seqfeature_dbxref-expanded"><a name="L928"></a><tt class="py-lineno"> 928</tt>  <tt class="py-line">        <tt class="py-docstring">"""Add database crossreferences of a SeqFeature to the database (PRIVATE).</tt> </tt>
<a name="L929"></a><tt class="py-lineno"> 929</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L930"></a><tt class="py-lineno"> 930</tt>  <tt class="py-line"><tt class="py-docstring">            o dbxrefs           List, dbxref data from the source file in the</tt> </tt>
<a name="L931"></a><tt class="py-lineno"> 931</tt>  <tt class="py-line"><tt class="py-docstring">                                format &lt;database&gt;:&lt;accession&gt;</tt> </tt>
<a name="L932"></a><tt class="py-lineno"> 932</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L933"></a><tt class="py-lineno"> 933</tt>  <tt class="py-line"><tt class="py-docstring">            o seqfeature_id     Int, the identifier for the seqfeature in the</tt> </tt>
<a name="L934"></a><tt class="py-lineno"> 934</tt>  <tt class="py-line"><tt class="py-docstring">                                seqfeature table</tt> </tt>
<a name="L935"></a><tt class="py-lineno"> 935</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L936"></a><tt class="py-lineno"> 936</tt>  <tt class="py-line"><tt class="py-docstring">            Insert dbxref qualifier data for a seqfeature into the</tt> </tt>
<a name="L937"></a><tt class="py-lineno"> 937</tt>  <tt class="py-line"><tt class="py-docstring">            seqfeature_dbxref and, if required, dbxref tables.</tt> </tt>
<a name="L938"></a><tt class="py-lineno"> 938</tt>  <tt class="py-line"><tt class="py-docstring">            The dbxref_id qualifier/value sets go into the dbxref table</tt> </tt>
<a name="L939"></a><tt class="py-lineno"> 939</tt>  <tt class="py-line"><tt class="py-docstring">            as dbname, accession, version tuples, with dbxref.dbxref_id</tt> </tt>
<a name="L940"></a><tt class="py-lineno"> 940</tt>  <tt class="py-line"><tt class="py-docstring">            being automatically assigned, and into the seqfeature_dbxref</tt> </tt>
<a name="L941"></a><tt class="py-lineno"> 941</tt>  <tt class="py-line"><tt class="py-docstring">            table as seqfeature_id, dbxref_id, and rank tuples</tt> </tt>
<a name="L942"></a><tt class="py-lineno"> 942</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L943"></a><tt class="py-lineno"> 943</tt>  <tt class="py-line">        <tt class="py-comment"># NOTE - In older versions of Biopython, we would map the GenBank</tt> </tt>
<a name="L944"></a><tt class="py-lineno"> 944</tt>  <tt class="py-line">        <tt class="py-comment"># db_xref "name", for example "GI" to "GeneIndex", and give a warning</tt> </tt>
<a name="L945"></a><tt class="py-lineno"> 945</tt>  <tt class="py-line">        <tt class="py-comment"># for any unknown terms.  This was a long term maintainance problem,</tt> </tt>
<a name="L946"></a><tt class="py-lineno"> 946</tt>  <tt class="py-line">        <tt class="py-comment"># and differed from BioPerl and BioJava's implementation.  See bug 2405</tt> </tt>
<a name="L947"></a><tt class="py-lineno"> 947</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-364" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-364', 'rank', 'link-27');">rank</a></tt><tt class="py-op">,</tt> <tt id="link-365" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-365', 'value', 'link-221');">value</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt id="link-366" class="py-name" targets="Variable BioSQL.BioSeq.DBSeqRecord.dbxrefs=BioSQL.BioSeq.DBSeqRecord-class.html#dbxrefs"><a title="BioSQL.BioSeq.DBSeqRecord.dbxrefs" class="py-name" href="#" onclick="return doclink('link-366', 'dbxrefs', 'link-366');">dbxrefs</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L948"></a><tt class="py-lineno"> 948</tt>  <tt class="py-line">            <tt class="py-comment"># Split the DB:accession format string at colons.  We have to</tt> </tt>
<a name="L949"></a><tt class="py-lineno"> 949</tt>  <tt class="py-line">            <tt class="py-comment"># account for multiple-line and multiple-accession entries</tt> </tt>
<a name="L950"></a><tt class="py-lineno"> 950</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L951"></a><tt class="py-lineno"> 951</tt>  <tt class="py-line">                <tt class="py-name">dbxref_data</tt> <tt class="py-op">=</tt> <tt id="link-367" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-367', 'value', 'link-221');">value</a></tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">' '</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt class="py-name">replace</tt><tt class="py-op">(</tt><tt class="py-string">'\n'</tt><tt class="py-op">,</tt> <tt class="py-string">''</tt><tt class="py-op">)</tt><tt class="py-op">.</tt><tt id="link-368" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-368', 'split', 'link-150');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">':'</tt><tt class="py-op">)</tt> </tt>
<a name="L952"></a><tt class="py-lineno"> 952</tt>  <tt class="py-line">                <tt id="link-369" class="py-name"><a title="Bio.motifs.jaspar.db" class="py-name" href="#" onclick="return doclink('link-369', 'db', 'link-113');">db</a></tt> <tt class="py-op">=</tt> <tt class="py-name">dbxref_data</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L953"></a><tt class="py-lineno"> 953</tt>  <tt class="py-line">                <tt class="py-name">accessions</tt> <tt class="py-op">=</tt> <tt class="py-name">dbxref_data</tt><tt class="py-op">[</tt><tt class="py-number">1</tt><tt class="py-op">:</tt><tt class="py-op">]</tt> </tt>
<a name="L954"></a><tt class="py-lineno"> 954</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt><tt class="py-op">:</tt> </tt>
<a name="L955"></a><tt class="py-lineno"> 955</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt><tt class="py-string">"Parsing of db_xref failed: '%s'"</tt> <tt class="py-op">%</tt> <tt id="link-370" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-370', 'value', 'link-221');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L956"></a><tt class="py-lineno"> 956</tt>  <tt class="py-line">            <tt class="py-comment"># Loop over all the grabbed accessions, and attempt to fill the</tt> </tt>
<a name="L957"></a><tt class="py-lineno"> 957</tt>  <tt class="py-line">            <tt class="py-comment"># table</tt> </tt>
<a name="L958"></a><tt class="py-lineno"> 958</tt>  <tt class="py-line">            <tt class="py-keyword">for</tt> <tt id="link-371" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-371', 'accession', 'link-52');">accession</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">accessions</tt><tt class="py-op">:</tt> </tt>
<a name="L959"></a><tt class="py-lineno"> 959</tt>  <tt class="py-line">                <tt class="py-comment"># Get the dbxref_id value for the dbxref data</tt> </tt>
<a name="L960"></a><tt class="py-lineno"> 960</tt>  <tt class="py-line">                <tt class="py-name">dbxref_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-372" class="py-name"><a title="BioSQL.Loader.DatabaseLoader._get_dbxref_id" class="py-name" href="#" onclick="return doclink('link-372', '_get_dbxref_id', 'link-349');">_get_dbxref_id</a></tt><tt class="py-op">(</tt><tt id="link-373" class="py-name"><a title="Bio.motifs.jaspar.db" class="py-name" href="#" onclick="return doclink('link-373', 'db', 'link-113');">db</a></tt><tt class="py-op">,</tt> <tt id="link-374" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-374', 'accession', 'link-52');">accession</a></tt><tt class="py-op">)</tt> </tt>
<a name="L961"></a><tt class="py-lineno"> 961</tt>  <tt class="py-line">                <tt class="py-comment"># Insert the seqfeature_dbxref data</tt> </tt>
<a name="L962"></a><tt class="py-lineno"> 962</tt>  <tt class="py-line">                <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-375" class="py-name" targets="Method BioSQL.Loader.DatabaseLoader._get_seqfeature_dbxref()=BioSQL.Loader.DatabaseLoader-class.html#_get_seqfeature_dbxref"><a title="BioSQL.Loader.DatabaseLoader._get_seqfeature_dbxref" class="py-name" href="#" onclick="return doclink('link-375', '_get_seqfeature_dbxref', 'link-375');">_get_seqfeature_dbxref</a></tt><tt class="py-op">(</tt><tt class="py-name">seqfeature_id</tt><tt class="py-op">,</tt> <tt class="py-name">dbxref_id</tt><tt class="py-op">,</tt> <tt id="link-376" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-376', 'rank', 'link-27');">rank</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
</div><a name="L963"></a><tt class="py-lineno"> 963</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader._get_dbxref_id"></a><div id="DatabaseLoader._get_dbxref_id-def"><a name="L964"></a><tt class="py-lineno"> 964</tt> <a class="py-toggle" href="#" id="DatabaseLoader._get_dbxref_id-toggle" onclick="return toggle('DatabaseLoader._get_dbxref_id');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#_get_dbxref_id">_get_dbxref_id</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">db</tt><tt class="py-op">,</tt> <tt class="py-param">accession</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader._get_dbxref_id-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader._get_dbxref_id-expanded"><a name="L965"></a><tt class="py-lineno"> 965</tt>  <tt class="py-line">        <tt class="py-docstring">""" _get_dbxref_id(self, db, accession) -&gt; Int</tt> </tt>
<a name="L966"></a><tt class="py-lineno"> 966</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L967"></a><tt class="py-lineno"> 967</tt>  <tt class="py-line"><tt class="py-docstring">            o db          String, the name of the external database containing</tt> </tt>
<a name="L968"></a><tt class="py-lineno"> 968</tt>  <tt class="py-line"><tt class="py-docstring">                          the accession number</tt> </tt>
<a name="L969"></a><tt class="py-lineno"> 969</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L970"></a><tt class="py-lineno"> 970</tt>  <tt class="py-line"><tt class="py-docstring">            o accession   String, the accession of the dbxref data</tt> </tt>
<a name="L971"></a><tt class="py-lineno"> 971</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L972"></a><tt class="py-lineno"> 972</tt>  <tt class="py-line"><tt class="py-docstring">            Finds and returns the dbxref_id for the passed data.  The method</tt> </tt>
<a name="L973"></a><tt class="py-lineno"> 973</tt>  <tt class="py-line"><tt class="py-docstring">            attempts to find an existing record first, and inserts the data</tt> </tt>
<a name="L974"></a><tt class="py-lineno"> 974</tt>  <tt class="py-line"><tt class="py-docstring">            if there is no record.</tt> </tt>
<a name="L975"></a><tt class="py-lineno"> 975</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L976"></a><tt class="py-lineno"> 976</tt>  <tt class="py-line">        <tt class="py-comment"># Check for an existing record</tt> </tt>
<a name="L977"></a><tt class="py-lineno"> 977</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">r'SELECT dbxref_id FROM dbxref WHERE dbname = %s '</tt> \ </tt>
<a name="L978"></a><tt class="py-lineno"> 978</tt>  <tt class="py-line">              <tt class="py-string">r'AND accession = %s'</tt> </tt>
<a name="L979"></a><tt class="py-lineno"> 979</tt>  <tt class="py-line">        <tt class="py-name">dbxref_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-377" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0" class="py-name" href="#" onclick="return doclink('link-377', 'execute_and_fetch_col0', 'link-37');">execute_and_fetch_col0</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt id="link-378" class="py-name"><a title="Bio.motifs.jaspar.db" class="py-name" href="#" onclick="return doclink('link-378', 'db', 'link-113');">db</a></tt><tt class="py-op">,</tt> <tt id="link-379" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-379', 'accession', 'link-52');">accession</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L980"></a><tt class="py-lineno"> 980</tt>  <tt class="py-line">        <tt class="py-comment"># If there was a record, return the dbxref_id, else create the</tt> </tt>
<a name="L981"></a><tt class="py-lineno"> 981</tt>  <tt class="py-line">        <tt class="py-comment"># record and return the created dbxref_id</tt> </tt>
<a name="L982"></a><tt class="py-lineno"> 982</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">dbxref_id</tt><tt class="py-op">:</tt> </tt>
<a name="L983"></a><tt class="py-lineno"> 983</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">dbxref_id</tt><tt class="py-op">[</tt><tt class="py-number">0</tt><tt class="py-op">]</tt> </tt>
<a name="L984"></a><tt class="py-lineno"> 984</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-380" class="py-name"><a title="BioSQL.Loader.DatabaseLoader._add_dbxref" class="py-name" href="#" onclick="return doclink('link-380', '_add_dbxref', 'link-261');">_add_dbxref</a></tt><tt class="py-op">(</tt><tt id="link-381" class="py-name"><a title="Bio.motifs.jaspar.db" class="py-name" href="#" onclick="return doclink('link-381', 'db', 'link-113');">db</a></tt><tt class="py-op">,</tt> <tt id="link-382" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-382', 'accession', 'link-52');">accession</a></tt><tt class="py-op">,</tt> <tt class="py-number">0</tt><tt class="py-op">)</tt> </tt>
</div><a name="L985"></a><tt class="py-lineno"> 985</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader._get_seqfeature_dbxref"></a><div id="DatabaseLoader._get_seqfeature_dbxref-def"><a name="L986"></a><tt class="py-lineno"> 986</tt> <a class="py-toggle" href="#" id="DatabaseLoader._get_seqfeature_dbxref-toggle" onclick="return toggle('DatabaseLoader._get_seqfeature_dbxref');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#_get_seqfeature_dbxref">_get_seqfeature_dbxref</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">seqfeature_id</tt><tt class="py-op">,</tt> <tt class="py-param">dbxref_id</tt><tt class="py-op">,</tt> <tt class="py-param">rank</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader._get_seqfeature_dbxref-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader._get_seqfeature_dbxref-expanded"><a name="L987"></a><tt class="py-lineno"> 987</tt>  <tt class="py-line">        <tt class="py-docstring">""" Check for a pre-existing seqfeature_dbxref entry with the passed</tt> </tt>
<a name="L988"></a><tt class="py-lineno"> 988</tt>  <tt class="py-line"><tt class="py-docstring">            seqfeature_id and dbxref_id.  If one does not exist, insert new</tt> </tt>
<a name="L989"></a><tt class="py-lineno"> 989</tt>  <tt class="py-line"><tt class="py-docstring">            data</tt> </tt>
<a name="L990"></a><tt class="py-lineno"> 990</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L991"></a><tt class="py-lineno"> 991</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L992"></a><tt class="py-lineno"> 992</tt>  <tt class="py-line">        <tt class="py-comment"># Check for an existing record</tt> </tt>
<a name="L993"></a><tt class="py-lineno"> 993</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">r"SELECT seqfeature_id, dbxref_id FROM seqfeature_dbxref "</tt> \ </tt>
<a name="L994"></a><tt class="py-lineno"> 994</tt>  <tt class="py-line">              <tt class="py-string">r"WHERE seqfeature_id = %s AND dbxref_id = %s"</tt> </tt>
<a name="L995"></a><tt class="py-lineno"> 995</tt>  <tt class="py-line">        <tt class="py-name">result</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-383" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0" class="py-name" href="#" onclick="return doclink('link-383', 'execute_and_fetch_col0', 'link-37');">execute_and_fetch_col0</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">seqfeature_id</tt><tt class="py-op">,</tt> </tt>
<a name="L996"></a><tt class="py-lineno"> 996</tt>  <tt class="py-line">                                                           <tt class="py-name">dbxref_id</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L997"></a><tt class="py-lineno"> 997</tt>  <tt class="py-line">        <tt class="py-comment"># If there was a record, return without executing anything, else create</tt> </tt>
<a name="L998"></a><tt class="py-lineno"> 998</tt>  <tt class="py-line">        <tt class="py-comment"># the record and return</tt> </tt>
<a name="L999"></a><tt class="py-lineno"> 999</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">result</tt><tt class="py-op">:</tt> </tt>
<a name="L1000"></a><tt class="py-lineno">1000</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">result</tt> </tt>
<a name="L1001"></a><tt class="py-lineno">1001</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-384" class="py-name" targets="Method BioSQL.Loader.DatabaseLoader._add_seqfeature_dbxref()=BioSQL.Loader.DatabaseLoader-class.html#_add_seqfeature_dbxref"><a title="BioSQL.Loader.DatabaseLoader._add_seqfeature_dbxref" class="py-name" href="#" onclick="return doclink('link-384', '_add_seqfeature_dbxref', 'link-384');">_add_seqfeature_dbxref</a></tt><tt class="py-op">(</tt><tt class="py-name">seqfeature_id</tt><tt class="py-op">,</tt> <tt class="py-name">dbxref_id</tt><tt class="py-op">,</tt> <tt id="link-385" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-385', 'rank', 'link-27');">rank</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1002"></a><tt class="py-lineno">1002</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader._add_seqfeature_dbxref"></a><div id="DatabaseLoader._add_seqfeature_dbxref-def"><a name="L1003"></a><tt class="py-lineno">1003</tt> <a class="py-toggle" href="#" id="DatabaseLoader._add_seqfeature_dbxref-toggle" onclick="return toggle('DatabaseLoader._add_seqfeature_dbxref');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#_add_seqfeature_dbxref">_add_seqfeature_dbxref</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">seqfeature_id</tt><tt class="py-op">,</tt> <tt class="py-param">dbxref_id</tt><tt class="py-op">,</tt> <tt class="py-param">rank</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader._add_seqfeature_dbxref-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader._add_seqfeature_dbxref-expanded"><a name="L1004"></a><tt class="py-lineno">1004</tt>  <tt class="py-line">        <tt class="py-docstring">""" Insert a seqfeature_dbxref row and return the seqfeature_id and</tt> </tt>
<a name="L1005"></a><tt class="py-lineno">1005</tt>  <tt class="py-line"><tt class="py-docstring">            dbxref_id</tt> </tt>
<a name="L1006"></a><tt class="py-lineno">1006</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1007"></a><tt class="py-lineno">1007</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">r'INSERT INTO seqfeature_dbxref '</tt> \ </tt>
<a name="L1008"></a><tt class="py-lineno">1008</tt>  <tt class="py-line">              <tt class="py-string">'(seqfeature_id, dbxref_id, rank) VALUES'</tt> \ </tt>
<a name="L1009"></a><tt class="py-lineno">1009</tt>  <tt class="py-line">              <tt class="py-string">r'(%s, %s, %s)'</tt> </tt>
<a name="L1010"></a><tt class="py-lineno">1010</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-386" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-386', 'execute', 'link-39');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">seqfeature_id</tt><tt class="py-op">,</tt> <tt class="py-name">dbxref_id</tt><tt class="py-op">,</tt> <tt id="link-387" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-387', 'rank', 'link-27');">rank</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1011"></a><tt class="py-lineno">1011</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-name">seqfeature_id</tt><tt class="py-op">,</tt> <tt class="py-name">dbxref_id</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1012"></a><tt class="py-lineno">1012</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader._load_dbxrefs"></a><div id="DatabaseLoader._load_dbxrefs-def"><a name="L1013"></a><tt class="py-lineno">1013</tt> <a class="py-toggle" href="#" id="DatabaseLoader._load_dbxrefs-toggle" onclick="return toggle('DatabaseLoader._load_dbxrefs');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#_load_dbxrefs">_load_dbxrefs</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">record</tt><tt class="py-op">,</tt> <tt class="py-param">bioentry_id</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader._load_dbxrefs-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader._load_dbxrefs-expanded"><a name="L1014"></a><tt class="py-lineno">1014</tt>  <tt class="py-line">        <tt class="py-docstring">"""Load any sequence level cross references into the database (PRIVATE).</tt> </tt>
<a name="L1015"></a><tt class="py-lineno">1015</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1016"></a><tt class="py-lineno">1016</tt>  <tt class="py-line"><tt class="py-docstring">        See table bioentry_dbxref."""</tt> </tt>
<a name="L1017"></a><tt class="py-lineno">1017</tt>  <tt class="py-line">        <tt class="py-keyword">for</tt> <tt id="link-388" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-388', 'rank', 'link-27');">rank</a></tt><tt class="py-op">,</tt> <tt id="link-389" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-389', 'value', 'link-221');">value</a></tt> <tt class="py-keyword">in</tt> <tt class="py-name">enumerate</tt><tt class="py-op">(</tt><tt class="py-name">record</tt><tt class="py-op">.</tt><tt id="link-390" class="py-name"><a title="BioSQL.BioSeq.DBSeqRecord.dbxrefs" class="py-name" href="#" onclick="return doclink('link-390', 'dbxrefs', 'link-366');">dbxrefs</a></tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
<a name="L1018"></a><tt class="py-lineno">1018</tt>  <tt class="py-line">            <tt class="py-comment"># Split the DB:accession string at first colon.</tt> </tt>
<a name="L1019"></a><tt class="py-lineno">1019</tt>  <tt class="py-line">            <tt class="py-comment"># We have to cope with things like:</tt> </tt>
<a name="L1020"></a><tt class="py-lineno">1020</tt>  <tt class="py-line">            <tt class="py-comment"># "MGD:MGI:892" (db="MGD", accession="MGI:892")</tt> </tt>
<a name="L1021"></a><tt class="py-lineno">1021</tt>  <tt class="py-line">            <tt class="py-comment"># "GO:GO:123" (db="GO", accession="GO:123")</tt> </tt>
<a name="L1022"></a><tt class="py-lineno">1022</tt>  <tt class="py-line">            <tt class="py-comment">#</tt> </tt>
<a name="L1023"></a><tt class="py-lineno">1023</tt>  <tt class="py-line">            <tt class="py-comment"># Annoyingly I have seen the NCBI use both the style</tt> </tt>
<a name="L1024"></a><tt class="py-lineno">1024</tt>  <tt class="py-line">            <tt class="py-comment"># "GO:GO:123" and "GO:123" in different vintages.</tt> </tt>
<a name="L1025"></a><tt class="py-lineno">1025</tt>  <tt class="py-line">            <tt class="py-keyword">assert</tt> <tt id="link-391" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-391', 'value', 'link-221');">value</a></tt><tt class="py-op">.</tt><tt id="link-392" class="py-name"><a title="Bio.Crystal.Chain.count
Bio.NeuralNetwork.Gene.Pattern.PatternRepository.count
Bio.Seq.MutableSeq.count
Bio.Seq.Seq.count
Bio.Seq.UnknownSeq.count
Bio.motifs.Instances.count" class="py-name" href="#" onclick="return doclink('link-392', 'count', 'link-146');">count</a></tt><tt class="py-op">(</tt><tt class="py-string">"\n"</tt><tt class="py-op">)</tt> <tt class="py-op">==</tt> <tt class="py-number">0</tt> </tt>
<a name="L1026"></a><tt class="py-lineno">1026</tt>  <tt class="py-line">            <tt class="py-keyword">try</tt><tt class="py-op">:</tt> </tt>
<a name="L1027"></a><tt class="py-lineno">1027</tt>  <tt class="py-line">                <tt id="link-393" class="py-name"><a title="Bio.motifs.jaspar.db" class="py-name" href="#" onclick="return doclink('link-393', 'db', 'link-113');">db</a></tt><tt class="py-op">,</tt> <tt id="link-394" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-394', 'accession', 'link-52');">accession</a></tt> <tt class="py-op">=</tt> <tt id="link-395" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-395', 'value', 'link-221');">value</a></tt><tt class="py-op">.</tt><tt id="link-396" class="py-name"><a title="Bio.Nexus.Trees.Tree.split
Bio.Phylo.BaseTree.TreeMixin.split
Bio.Restriction.Restriction.RestrictionBatch.split
Bio.Seq.Seq.split" class="py-name" href="#" onclick="return doclink('link-396', 'split', 'link-150');">split</a></tt><tt class="py-op">(</tt><tt class="py-string">':'</tt><tt class="py-op">,</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
<a name="L1028"></a><tt class="py-lineno">1028</tt>  <tt class="py-line">                <tt id="link-397" class="py-name"><a title="Bio.motifs.jaspar.db" class="py-name" href="#" onclick="return doclink('link-397', 'db', 'link-113');">db</a></tt> <tt class="py-op">=</tt> <tt id="link-398" class="py-name"><a title="Bio.motifs.jaspar.db" class="py-name" href="#" onclick="return doclink('link-398', 'db', 'link-113');">db</a></tt><tt class="py-op">.</tt><tt id="link-399" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-399', 'strip', 'link-101');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1029"></a><tt class="py-lineno">1029</tt>  <tt class="py-line">                <tt id="link-400" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-400', 'accession', 'link-52');">accession</a></tt> <tt class="py-op">=</tt> <tt id="link-401" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-401', 'accession', 'link-52');">accession</a></tt><tt class="py-op">.</tt><tt id="link-402" class="py-name"><a title="Bio.Seq.Seq.strip" class="py-name" href="#" onclick="return doclink('link-402', 'strip', 'link-101');">strip</a></tt><tt class="py-op">(</tt><tt class="py-op">)</tt> </tt>
<a name="L1030"></a><tt class="py-lineno">1030</tt>  <tt class="py-line">            <tt class="py-keyword">except</tt><tt class="py-op">:</tt> </tt>
<a name="L1031"></a><tt class="py-lineno">1031</tt>  <tt class="py-line">                <tt class="py-keyword">raise</tt> <tt class="py-name">ValueError</tt><tt class="py-op">(</tt> </tt>
<a name="L1032"></a><tt class="py-lineno">1032</tt>  <tt class="py-line">                    <tt class="py-string">"Parsing of dbxrefs list failed: '%s'"</tt> <tt class="py-op">%</tt> <tt id="link-403" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.value" class="py-name" href="#" onclick="return doclink('link-403', 'value', 'link-221');">value</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1033"></a><tt class="py-lineno">1033</tt>  <tt class="py-line">            <tt class="py-comment"># Get the dbxref_id value for the dbxref data</tt> </tt>
<a name="L1034"></a><tt class="py-lineno">1034</tt>  <tt class="py-line">            <tt class="py-name">dbxref_id</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-404" class="py-name"><a title="BioSQL.Loader.DatabaseLoader._get_dbxref_id" class="py-name" href="#" onclick="return doclink('link-404', '_get_dbxref_id', 'link-349');">_get_dbxref_id</a></tt><tt class="py-op">(</tt><tt id="link-405" class="py-name"><a title="Bio.motifs.jaspar.db" class="py-name" href="#" onclick="return doclink('link-405', 'db', 'link-113');">db</a></tt><tt class="py-op">,</tt> <tt id="link-406" class="py-name"><a title="Bio.GenBank._FeatureConsumer.accession
Bio.GenBank._RecordConsumer.accession
Bio.Phylo.PhyloXMLIO.Parser.accession
Bio.Phylo.PhyloXMLIO.Writer.accession" class="py-name" href="#" onclick="return doclink('link-406', 'accession', 'link-52');">accession</a></tt><tt class="py-op">)</tt> </tt>
<a name="L1035"></a><tt class="py-lineno">1035</tt>  <tt class="py-line">            <tt class="py-comment"># Insert the bioentry_dbxref  data</tt> </tt>
<a name="L1036"></a><tt class="py-lineno">1036</tt>  <tt class="py-line">            <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-407" class="py-name" targets="Method BioSQL.Loader.DatabaseLoader._get_bioentry_dbxref()=BioSQL.Loader.DatabaseLoader-class.html#_get_bioentry_dbxref"><a title="BioSQL.Loader.DatabaseLoader._get_bioentry_dbxref" class="py-name" href="#" onclick="return doclink('link-407', '_get_bioentry_dbxref', 'link-407');">_get_bioentry_dbxref</a></tt><tt class="py-op">(</tt><tt class="py-name">bioentry_id</tt><tt class="py-op">,</tt> <tt class="py-name">dbxref_id</tt><tt class="py-op">,</tt> <tt id="link-408" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-408', 'rank', 'link-27');">rank</a></tt> <tt class="py-op">+</tt> <tt class="py-number">1</tt><tt class="py-op">)</tt> </tt>
</div><a name="L1037"></a><tt class="py-lineno">1037</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader._get_bioentry_dbxref"></a><div id="DatabaseLoader._get_bioentry_dbxref-def"><a name="L1038"></a><tt class="py-lineno">1038</tt> <a class="py-toggle" href="#" id="DatabaseLoader._get_bioentry_dbxref-toggle" onclick="return toggle('DatabaseLoader._get_bioentry_dbxref');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#_get_bioentry_dbxref">_get_bioentry_dbxref</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">bioentry_id</tt><tt class="py-op">,</tt> <tt class="py-param">dbxref_id</tt><tt class="py-op">,</tt> <tt class="py-param">rank</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader._get_bioentry_dbxref-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader._get_bioentry_dbxref-expanded"><a name="L1039"></a><tt class="py-lineno">1039</tt>  <tt class="py-line">        <tt class="py-docstring">""" Check for a pre-existing bioentry_dbxref entry with the passed</tt> </tt>
<a name="L1040"></a><tt class="py-lineno">1040</tt>  <tt class="py-line"><tt class="py-docstring">            seqfeature_id and dbxref_id.  If one does not exist, insert new</tt> </tt>
<a name="L1041"></a><tt class="py-lineno">1041</tt>  <tt class="py-line"><tt class="py-docstring">            data</tt> </tt>
<a name="L1042"></a><tt class="py-lineno">1042</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1043"></a><tt class="py-lineno">1043</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1044"></a><tt class="py-lineno">1044</tt>  <tt class="py-line">        <tt class="py-comment"># Check for an existing record</tt> </tt>
<a name="L1045"></a><tt class="py-lineno">1045</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">r"SELECT bioentry_id, dbxref_id FROM bioentry_dbxref "</tt> \ </tt>
<a name="L1046"></a><tt class="py-lineno">1046</tt>  <tt class="py-line">              <tt class="py-string">r"WHERE bioentry_id = %s AND dbxref_id = %s"</tt> </tt>
<a name="L1047"></a><tt class="py-lineno">1047</tt>  <tt class="py-line">        <tt class="py-name">result</tt> <tt class="py-op">=</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-409" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute_and_fetch_col0
BioSQL.BioSeqDatabase.MysqlConnectorAdaptor.execute_and_fetch_col0" class="py-name" href="#" onclick="return doclink('link-409', 'execute_and_fetch_col0', 'link-37');">execute_and_fetch_col0</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">bioentry_id</tt><tt class="py-op">,</tt> </tt>
<a name="L1048"></a><tt class="py-lineno">1048</tt>  <tt class="py-line">                                                           <tt class="py-name">dbxref_id</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1049"></a><tt class="py-lineno">1049</tt>  <tt class="py-line">        <tt class="py-comment"># If there was a record, return without executing anything, else create</tt> </tt>
<a name="L1050"></a><tt class="py-lineno">1050</tt>  <tt class="py-line">        <tt class="py-comment"># the record and return</tt> </tt>
<a name="L1051"></a><tt class="py-lineno">1051</tt>  <tt class="py-line">        <tt class="py-keyword">if</tt> <tt class="py-name">result</tt><tt class="py-op">:</tt> </tt>
<a name="L1052"></a><tt class="py-lineno">1052</tt>  <tt class="py-line">            <tt class="py-keyword">return</tt> <tt class="py-name">result</tt> </tt>
<a name="L1053"></a><tt class="py-lineno">1053</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-name">self</tt><tt class="py-op">.</tt><tt id="link-410" class="py-name" targets="Method BioSQL.Loader.DatabaseLoader._add_bioentry_dbxref()=BioSQL.Loader.DatabaseLoader-class.html#_add_bioentry_dbxref"><a title="BioSQL.Loader.DatabaseLoader._add_bioentry_dbxref" class="py-name" href="#" onclick="return doclink('link-410', '_add_bioentry_dbxref', 'link-410');">_add_bioentry_dbxref</a></tt><tt class="py-op">(</tt><tt class="py-name">bioentry_id</tt><tt class="py-op">,</tt> <tt class="py-name">dbxref_id</tt><tt class="py-op">,</tt> <tt id="link-411" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-411', 'rank', 'link-27');">rank</a></tt><tt class="py-op">)</tt> </tt>
</div><a name="L1054"></a><tt class="py-lineno">1054</tt>  <tt class="py-line"> </tt>
<a name="DatabaseLoader._add_bioentry_dbxref"></a><div id="DatabaseLoader._add_bioentry_dbxref-def"><a name="L1055"></a><tt class="py-lineno">1055</tt> <a class="py-toggle" href="#" id="DatabaseLoader._add_bioentry_dbxref-toggle" onclick="return toggle('DatabaseLoader._add_bioentry_dbxref');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseLoader-class.html#_add_bioentry_dbxref">_add_bioentry_dbxref</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">bioentry_id</tt><tt class="py-op">,</tt> <tt class="py-param">dbxref_id</tt><tt class="py-op">,</tt> <tt class="py-param">rank</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseLoader._add_bioentry_dbxref-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseLoader._add_bioentry_dbxref-expanded"><a name="L1056"></a><tt class="py-lineno">1056</tt>  <tt class="py-line">        <tt class="py-docstring">""" Insert a bioentry_dbxref row and return the seqfeature_id and</tt> </tt>
<a name="L1057"></a><tt class="py-lineno">1057</tt>  <tt class="py-line"><tt class="py-docstring">            dbxref_id</tt> </tt>
<a name="L1058"></a><tt class="py-lineno">1058</tt>  <tt class="py-line"><tt class="py-docstring">        """</tt> </tt>
<a name="L1059"></a><tt class="py-lineno">1059</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">r'INSERT INTO bioentry_dbxref '</tt> \ </tt>
<a name="L1060"></a><tt class="py-lineno">1060</tt>  <tt class="py-line">              <tt class="py-string">'(bioentry_id,dbxref_id,rank) VALUES '</tt> \ </tt>
<a name="L1061"></a><tt class="py-lineno">1061</tt>  <tt class="py-line">              <tt class="py-string">'(%s, %s, %s)'</tt> </tt>
<a name="L1062"></a><tt class="py-lineno">1062</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-412" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-412', 'execute', 'link-39');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">bioentry_id</tt><tt class="py-op">,</tt> <tt class="py-name">dbxref_id</tt><tt class="py-op">,</tt> <tt id="link-413" class="py-name"><a title="Bio.Phylo.PhyloXMLIO.Writer.rank" class="py-name" href="#" onclick="return doclink('link-413', 'rank', 'link-27');">rank</a></tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1063"></a><tt class="py-lineno">1063</tt>  <tt class="py-line">        <tt class="py-keyword">return</tt> <tt class="py-op">(</tt><tt class="py-name">bioentry_id</tt><tt class="py-op">,</tt> <tt class="py-name">dbxref_id</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1064"></a><tt class="py-lineno">1064</tt>  <tt class="py-line"> </tt>
<a name="L1065"></a><tt class="py-lineno">1065</tt>  <tt class="py-line"> </tt>
<a name="DatabaseRemover"></a><div id="DatabaseRemover-def"><a name="L1066"></a><tt class="py-lineno">1066</tt> <a class="py-toggle" href="#" id="DatabaseRemover-toggle" onclick="return toggle('DatabaseRemover');">-</a><tt class="py-line"><tt class="py-keyword">class</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseRemover-class.html">DatabaseRemover</a><tt class="py-op">(</tt><tt class="py-base-class">object</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseRemover-collapsed" style="display:none;" pad="++++" indent="++++"></div><div id="DatabaseRemover-expanded"><a name="L1067"></a><tt class="py-lineno">1067</tt>  <tt class="py-line">    <tt class="py-docstring">"""Complement the Loader functionality by fully removing a database.</tt> </tt>
<a name="L1068"></a><tt class="py-lineno">1068</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1069"></a><tt class="py-lineno">1069</tt>  <tt class="py-line"><tt class="py-docstring">    This probably isn't really useful for normal purposes, since you</tt> </tt>
<a name="L1070"></a><tt class="py-lineno">1070</tt>  <tt class="py-line"><tt class="py-docstring">    can just do a::</tt> </tt>
<a name="L1071"></a><tt class="py-lineno">1071</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1072"></a><tt class="py-lineno">1072</tt>  <tt class="py-line"><tt class="py-docstring">        DROP DATABASE db_name</tt> </tt>
<a name="L1073"></a><tt class="py-lineno">1073</tt>  <tt class="py-line"><tt class="py-docstring"></tt> </tt>
<a name="L1074"></a><tt class="py-lineno">1074</tt>  <tt class="py-line"><tt class="py-docstring">    and then recreate the database. But, it's really useful for testing</tt> </tt>
<a name="L1075"></a><tt class="py-lineno">1075</tt>  <tt class="py-line"><tt class="py-docstring">    purposes.</tt> </tt>
<a name="L1076"></a><tt class="py-lineno">1076</tt>  <tt class="py-line"><tt class="py-docstring">    """</tt> </tt>
<a name="L1077"></a><tt class="py-lineno">1077</tt>  <tt class="py-line"> </tt>
<a name="DatabaseRemover.__init__"></a><div id="DatabaseRemover.__init__-def"><a name="L1078"></a><tt class="py-lineno">1078</tt> <a class="py-toggle" href="#" id="DatabaseRemover.__init__-toggle" onclick="return toggle('DatabaseRemover.__init__');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseRemover-class.html#__init__">__init__</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">,</tt> <tt class="py-param">adaptor</tt><tt class="py-op">,</tt> <tt class="py-param">dbid</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseRemover.__init__-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseRemover.__init__-expanded"><a name="L1079"></a><tt class="py-lineno">1079</tt>  <tt class="py-line">        <tt class="py-docstring">"""Initialize with a database id and adaptor connection."""</tt> </tt>
<a name="L1080"></a><tt class="py-lineno">1080</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt> <tt class="py-op">=</tt> <tt class="py-name">adaptor</tt> </tt>
<a name="L1081"></a><tt class="py-lineno">1081</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbid</tt> <tt class="py-op">=</tt> <tt class="py-name">dbid</tt> </tt>
</div><a name="L1082"></a><tt class="py-lineno">1082</tt>  <tt class="py-line"> </tt>
<a name="DatabaseRemover.remove"></a><div id="DatabaseRemover.remove-def"><a name="L1083"></a><tt class="py-lineno">1083</tt> <a class="py-toggle" href="#" id="DatabaseRemover.remove-toggle" onclick="return toggle('DatabaseRemover.remove');">-</a><tt class="py-line">    <tt class="py-keyword">def</tt> <a class="py-def-name" href="BioSQL.Loader.DatabaseRemover-class.html#remove">remove</a><tt class="py-op">(</tt><tt class="py-param">self</tt><tt class="py-op">)</tt><tt class="py-op">:</tt> </tt>
</div><div id="DatabaseRemover.remove-collapsed" style="display:none;" pad="++++" indent="++++++++"></div><div id="DatabaseRemover.remove-expanded"><a name="L1084"></a><tt class="py-lineno">1084</tt>  <tt class="py-line">        <tt class="py-docstring">"""Remove everything related to the given database id."""</tt> </tt>
<a name="L1085"></a><tt class="py-lineno">1085</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">r"DELETE FROM bioentry WHERE biodatabase_id = %s"</tt> </tt>
<a name="L1086"></a><tt class="py-lineno">1086</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-414" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-414', 'execute', 'link-39');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbid</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
<a name="L1087"></a><tt class="py-lineno">1087</tt>  <tt class="py-line">        <tt class="py-name">sql</tt> <tt class="py-op">=</tt> <tt class="py-string">r"DELETE FROM biodatabase WHERE biodatabase_id = %s"</tt> </tt>
<a name="L1088"></a><tt class="py-lineno">1088</tt>  <tt class="py-line">        <tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">adaptor</tt><tt class="py-op">.</tt><tt id="link-415" class="py-name"><a title="BioSQL.BioSeqDatabase.Adaptor.execute
BioSQL.BioSeqDatabase._CursorWrapper.execute
BioSQL.DBUtils.Generic_dbutils.execute
BioSQL.DBUtils.Sqlite_dbutils.execute" class="py-name" href="#" onclick="return doclink('link-415', 'execute', 'link-39');">execute</a></tt><tt class="py-op">(</tt><tt class="py-name">sql</tt><tt class="py-op">,</tt> <tt class="py-op">(</tt><tt class="py-name">self</tt><tt class="py-op">.</tt><tt class="py-name">dbid</tt><tt class="py-op">,</tt><tt class="py-op">)</tt><tt class="py-op">)</tt> </tt>
</div></div><a name="L1089"></a><tt class="py-lineno">1089</tt>  <tt class="py-line"> </tt><script type="text/javascript">
<!--
expandto(location.href);
// -->
</script>
</pre>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Wed Oct 21 18:04:33 2015
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
