<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>Module Hierarchy</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th bgcolor="#70b0f0" class="navbar-select"
          >&nbsp;&nbsp;&nbsp;Trees&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">&nbsp;</td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="module-tree.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<center><b>
 [ <a href="module-tree.html">Module Hierarchy</a>
 | <a href="class-tree.html">Class Hierarchy</a> ]
</b></center><br />
<h1 class="epydoc">Module Hierarchy</h1>
<ul class="nomargin-top">
    <li> <strong class="uidlink"><a href="Bio-module.html">Bio</a></strong>: <em class="summary">Collection of modules for dealing with biological data in Python.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.Affy-module.html">Bio.Affy</a></strong>: <em class="summary">Deal with Affymetrix related data such as cel files.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.Affy.CelFile-module.html">Bio.Affy.CelFile</a></strong>: <em class="summary">Reading information from Affymetrix CEL files version 3 and 4.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.Align-module.html">Bio.Align</a></strong>: <em class="summary">Code for dealing with sequence alignments.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.Align.AlignInfo-module.html">Bio.Align.AlignInfo</a></strong>: <em class="summary">Extract information from alignment objects.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Align.Applications-module.html">Bio.Align.Applications</a></strong>: <em class="summary">Alignment command line tool wrappers.</em>
    <ul>
    <li class="private"> <strong class="uidlink"><a href="Bio.Align.Applications._ClustalOmega-module.html" onclick="show_private();">Bio.Align.Applications._ClustalOmega</a></strong>: <em class="summary">Command line wrapper for the multiple alignment program Clustal Omega.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Align.Applications._Clustalw-module.html" onclick="show_private();">Bio.Align.Applications._Clustalw</a></strong>: <em class="summary">Command line wrapper for the multiple alignment program Clustal W.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Align.Applications._Dialign-module.html" onclick="show_private();">Bio.Align.Applications._Dialign</a></strong>: <em class="summary">Command line wrapper for the multiple alignment program DIALIGN2-2.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Align.Applications._MSAProbs-module.html" onclick="show_private();">Bio.Align.Applications._MSAProbs</a></strong>: <em class="summary">Command line wrapper for the multiple sequence alignment program MSAProbs.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Align.Applications._Mafft-module.html" onclick="show_private();">Bio.Align.Applications._Mafft</a></strong>: <em class="summary">Command line wrapper for the multiple alignment programme MAFFT.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Align.Applications._Muscle-module.html" onclick="show_private();">Bio.Align.Applications._Muscle</a></strong>: <em class="summary">Command line wrapper for the multiple alignment program MUSCLE.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Align.Applications._Prank-module.html" onclick="show_private();">Bio.Align.Applications._Prank</a></strong>: <em class="summary">Command line wrapper for the multiple alignment program PRANK.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Align.Applications._Probcons-module.html" onclick="show_private();">Bio.Align.Applications._Probcons</a></strong>: <em class="summary">Command line wrapper for the multiple alignment program PROBCONS.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Align.Applications._TCoffee-module.html" onclick="show_private();">Bio.Align.Applications._TCoffee</a></strong>: <em class="summary">Command line wrapper for the multiple alignment program TCOFFEE.</em>    </li>
    </ul>
    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Align._aligners-module.html" onclick="show_private();">Bio.Align._aligners</a></strong>: <em class="summary">C extension module implementing pairwise alignment algorithms</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.AlignIO-module.html">Bio.AlignIO</a></strong>: <em class="summary">Multiple sequence alignment input/output as alignment objects.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.AlignIO.ClustalIO-module.html">Bio.AlignIO.ClustalIO</a></strong>: <em class="summary">Bio.AlignIO support for &quot;clustal&quot; output from CLUSTAL W and other tools.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.AlignIO.EmbossIO-module.html">Bio.AlignIO.EmbossIO</a></strong>: <em class="summary">Bio.AlignIO support for &quot;emboss&quot; alignment output from EMBOSS tools.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.AlignIO.FastaIO-module.html">Bio.AlignIO.FastaIO</a></strong>: <em class="summary">Bio.AlignIO support for &quot;fasta-m10&quot; output from Bill Pearson's FASTA tools.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.AlignIO.Interfaces-module.html">Bio.AlignIO.Interfaces</a></strong>: <em class="summary">AlignIO support module (not for general use).</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.AlignIO.MafIO-module.html">Bio.AlignIO.MafIO</a></strong>: <em class="summary">Bio.AlignIO support for the &quot;maf&quot; multiple alignment format.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.AlignIO.MauveIO-module.html">Bio.AlignIO.MauveIO</a></strong>: <em class="summary">Bio.AlignIO support for &quot;xmfa&quot; output from Mauve/ProgressiveMauve.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.AlignIO.NexusIO-module.html">Bio.AlignIO.NexusIO</a></strong>: <em class="summary">Bio.AlignIO support for the &quot;nexus&quot; file format.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.AlignIO.PhylipIO-module.html">Bio.AlignIO.PhylipIO</a></strong>: <em class="summary">AlignIO support for &quot;phylip&quot; format from Joe Felsenstein's PHYLIP tools.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.AlignIO.StockholmIO-module.html">Bio.AlignIO.StockholmIO</a></strong>: <em class="summary">Bio.AlignIO support for &quot;stockholm&quot; format (used in the PFAM database).</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.Alphabet-module.html">Bio.Alphabet</a></strong>: <em class="summary">Alphabets used in Seq objects etc to declare sequence type and letters.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.Alphabet.IUPAC-module.html">Bio.Alphabet.IUPAC</a></strong>: <em class="summary">Standard nucleotide and protein alphabets defined by IUPAC.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Alphabet.Reduced-module.html">Bio.Alphabet.Reduced</a></strong>: <em class="summary">Reduced alphabets which lump together several amino-acids into one letter.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.Application-module.html">Bio.Application</a></strong>: <em class="summary">General mechanisms to access applications in Biopython.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Blast-module.html">Bio.Blast</a></strong>: <em class="summary">Code for dealing with BLAST programs and output.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.Blast.Applications-module.html">Bio.Blast.Applications</a></strong>: <em class="summary">Definitions for interacting with BLAST related applications.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Blast.NCBIStandalone-module.html">Bio.Blast.NCBIStandalone</a></strong>: <em class="summary">Code for calling standalone BLAST and parsing plain text output (DEPRECATED).</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Blast.NCBIWWW-module.html">Bio.Blast.NCBIWWW</a></strong>: <em class="summary">Code to invoke the NCBI BLAST server over the internet.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Blast.NCBIXML-module.html">Bio.Blast.NCBIXML</a></strong>: <em class="summary">Code to work with the BLAST XML output.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Blast.ParseBlastTable-module.html">Bio.Blast.ParseBlastTable</a></strong>: <em class="summary">A parser for the NCBI blastpgp version 2.2.5 output format.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Blast.Record-module.html">Bio.Blast.Record</a></strong>: <em class="summary">Record classes to hold BLAST output.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.CAPS-module.html">Bio.CAPS</a></strong>: <em class="summary">Cleaved amplified polymorphic sequence (CAPS) markers.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Cluster-module.html">Bio.Cluster</a></strong>: <em class="summary">Cluster Analysis.</em>
    <ul>
    <li class="private"> <strong class="uidlink"><a href="Bio.Cluster._cluster-module.html" onclick="show_private();">Bio.Cluster._cluster</a></strong>: <em class="summary">C Clustering Library</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.Compass-module.html">Bio.Compass</a></strong>: <em class="summary">Code to deal with COMPASS output, a program for profile/profile comparison.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Crystal-module.html">Bio.Crystal</a></strong>: <em class="summary">Represent the NDB Atlas structure (a minimal subset of PDB format).</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Data-module.html">Bio.Data</a></strong>: <em class="summary">Collections of various bits of useful biological data.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.Data.CodonTable-module.html">Bio.Data.CodonTable</a></strong>: <em class="summary">Codon tables based on those from the NCBI.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Data.IUPACData-module.html">Bio.Data.IUPACData</a></strong>: <em class="summary">Information about the IUPAC alphabets.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Data.SCOPData-module.html">Bio.Data.SCOPData</a></strong>: <em class="summary">Additional protein alphabets used in the SCOP database and PDB files.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.Emboss-module.html">Bio.Emboss</a></strong>: <em class="summary">Code to interact with the ever-so-useful EMBOSS programs.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.Emboss.Applications-module.html">Bio.Emboss.Applications</a></strong>: <em class="summary">Code to interact with and run various EMBOSS programs.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Emboss.Primer3-module.html">Bio.Emboss.Primer3</a></strong>: <em class="summary">Code to parse output from the EMBOSS eprimer3 program.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Emboss.PrimerSearch-module.html">Bio.Emboss.PrimerSearch</a></strong>: <em class="summary">Code to interact with the primersearch program from EMBOSS.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.Entrez-module.html">Bio.Entrez</a></strong>: <em class="summary">Provides code to access NCBI over the WWW.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.Entrez.Parser-module.html">Bio.Entrez.Parser</a></strong>: <em class="summary">Parser for XML results returned by NCBI's Entrez Utilities.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.ExPASy-module.html">Bio.ExPASy</a></strong>: <em class="summary">Code to access resources at ExPASy over the WWW.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.ExPASy.Enzyme-module.html">Bio.ExPASy.Enzyme</a></strong>: <em class="summary">Parse the enzyme.dat file from Enzyme at ExPASy.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.ExPASy.Prodoc-module.html">Bio.ExPASy.Prodoc</a></strong>: <em class="summary">Code to work with the prosite.doc file from Prosite.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.ExPASy.Prosite-module.html">Bio.ExPASy.Prosite</a></strong>: <em class="summary">Parser for the prosite dat file from Prosite at ExPASy.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.ExPASy.ScanProsite-module.html">Bio.ExPASy.ScanProsite</a></strong>    </li>
    <li> <strong class="uidlink"><a href="Bio.ExPASy.cellosaurus-module.html">Bio.ExPASy.cellosaurus</a></strong>: <em class="summary">Parser for the cellosaurus.txt file from ExPASy.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.FSSP-module.html">Bio.FSSP</a></strong>: <em class="summary">Parser for FSSP files, used in a database of protein fold classifications.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.FSSP.FSSPTools-module.html">Bio.FSSP.FSSPTools</a></strong>: <em class="summary">Several routines used to extract information from FSSP sections.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.FSSP.fssp_rec-module.html">Bio.FSSP.fssp_rec</a></strong>: <em class="summary">A superclass for reading [f]ixed-column type [f]lat-[f]ile records.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.File-module.html">Bio.File</a></strong>: <em class="summary">Code for more fancy file handles.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.GA-module.html">Bio.GA</a></strong>: <em class="summary">Genetic Algorithm library (DEPRECATED).</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.GA.Crossover-module.html">Bio.GA.Crossover</a></strong>: <em class="summary">Support for crossovers in the Genetic Algorithms module.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.GA.Crossover.General-module.html">Bio.GA.Crossover.General</a></strong>: <em class="summary">General functionality for crossover that doesn't apply.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.GA.Crossover.GeneralPoint-module.html">Bio.GA.Crossover.GeneralPoint</a></strong>: <em class="summary">Generalized N-Point Crossover.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.GA.Crossover.Point-module.html">Bio.GA.Crossover.Point</a></strong>: <em class="summary">Perform two-point crossovers between the genomes of two organisms.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.GA.Crossover.TwoPoint-module.html">Bio.GA.Crossover.TwoPoint</a></strong>: <em class="summary">Perform two-point crossovers between the genomes of two organisms.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.GA.Crossover.Uniform-module.html">Bio.GA.Crossover.Uniform</a></strong>: <em class="summary">Perform uniform crossovers between the genomes of two organisms.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.GA.Evolver-module.html">Bio.GA.Evolver</a></strong>: <em class="summary">Evolution Strategies for a Population.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.GA.Mutation-module.html">Bio.GA.Mutation</a></strong>: <em class="summary">Support for mutations in the Genetic Algorithms module.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.GA.Mutation.General-module.html">Bio.GA.Mutation.General</a></strong>: <em class="summary">General functionality for mutations.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.GA.Mutation.Simple-module.html">Bio.GA.Mutation.Simple</a></strong>: <em class="summary">Perform Simple mutations on an organism's genome.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.GA.Organism-module.html">Bio.GA.Organism</a></strong>: <em class="summary">Deal with an Organism in a Genetic Algorithm population.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.GA.Repair-module.html">Bio.GA.Repair</a></strong>: <em class="summary">Methods for performing repairs that will Stabilize genomes.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.GA.Repair.Stabilizing-module.html">Bio.GA.Repair.Stabilizing</a></strong>: <em class="summary">Methods for performing repairs that will Stabilize genomes.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.GA.Selection-module.html">Bio.GA.Selection</a></strong>: <em class="summary">Support for selections in the Genetic Algorithms module.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.GA.Selection.Abstract-module.html">Bio.GA.Selection.Abstract</a></strong>: <em class="summary">Base selection class from which all Selectors should derive.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.GA.Selection.Diversity-module.html">Bio.GA.Selection.Diversity</a></strong>: <em class="summary">Select individuals into a new population trying to maintain diversity.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.GA.Selection.RouletteWheel-module.html">Bio.GA.Selection.RouletteWheel</a></strong>: <em class="summary">Implement Roulette Wheel selection on a population.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.GA.Selection.Tournament-module.html">Bio.GA.Selection.Tournament</a></strong>: <em class="summary">Provide Tournament style selection.</em>    </li>
    </ul>
    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.GenBank-module.html">Bio.GenBank</a></strong>: <em class="summary">Code to work with GenBank formatted files.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.GenBank.Record-module.html">Bio.GenBank.Record</a></strong>: <em class="summary">Hold GenBank data in a straightforward format.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.GenBank.Scanner-module.html">Bio.GenBank.Scanner</a></strong>: <em class="summary">Internal code for parsing GenBank and EMBL files (PRIVATE).</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.GenBank.utils-module.html">Bio.GenBank.utils</a></strong>: <em class="summary">Useful utilities for helping in parsing GenBank files.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.Geo-module.html">Bio.Geo</a></strong>: <em class="summary">Parser for files from NCBI's Gene Expression Omnibus (GEO).</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.Geo.Record-module.html">Bio.Geo.Record</a></strong>: <em class="summary">Hold GEO data in a straightforward format.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.Graphics-module.html">Bio.Graphics</a></strong>: <em class="summary">Bio.Graphics offers several graphical outputs, all using ReportLab.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.Graphics.BasicChromosome-module.html">Bio.Graphics.BasicChromosome</a></strong>: <em class="summary">Draw representations of organism chromosomes with added information.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Graphics.ColorSpiral-module.html">Bio.Graphics.ColorSpiral</a></strong>: <em class="summary">Generate RGB colours suitable for distinguishing categorical data.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Graphics.Comparative-module.html">Bio.Graphics.Comparative</a></strong>: <em class="summary">Plots to compare information between different sources.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Graphics.DisplayRepresentation-module.html">Bio.Graphics.DisplayRepresentation</a></strong>: <em class="summary">Represent information for graphical display.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Graphics.Distribution-module.html">Bio.Graphics.Distribution</a></strong>: <em class="summary">Display information distributed across a Chromosome-like object.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Graphics.GenomeDiagram-module.html">Bio.Graphics.GenomeDiagram</a></strong>: <em class="summary">GenomeDiagram module integrated into Biopython.</em>
    <ul>
    <li class="private"> <strong class="uidlink"><a href="Bio.Graphics.GenomeDiagram._AbstractDrawer-module.html" onclick="show_private();">Bio.Graphics.GenomeDiagram._AbstractDrawer</a></strong>: <em class="summary">AbstractDrawer module (considered to be a private module, the API may change!).</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Graphics.GenomeDiagram._CircularDrawer-module.html" onclick="show_private();">Bio.Graphics.GenomeDiagram._CircularDrawer</a></strong>: <em class="summary">CircularDrawer module for GenomeDiagram.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Graphics.GenomeDiagram._Colors-module.html" onclick="show_private();">Bio.Graphics.GenomeDiagram._Colors</a></strong>: <em class="summary">Colors module.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Graphics.GenomeDiagram._CrossLink-module.html" onclick="show_private();">Bio.Graphics.GenomeDiagram._CrossLink</a></strong>: <em class="summary">Genome Diagram Feature cross-link module.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Graphics.GenomeDiagram._Diagram-module.html" onclick="show_private();">Bio.Graphics.GenomeDiagram._Diagram</a></strong>: <em class="summary">Provides a container for information concerning the tracks to be drawn in a diagram.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Graphics.GenomeDiagram._Feature-module.html" onclick="show_private();">Bio.Graphics.GenomeDiagram._Feature</a></strong>: <em class="summary">Feature module.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Graphics.GenomeDiagram._FeatureSet-module.html" onclick="show_private();">Bio.Graphics.GenomeDiagram._FeatureSet</a></strong>: <em class="summary">FeatureSet module.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Graphics.GenomeDiagram._Graph-module.html" onclick="show_private();">Bio.Graphics.GenomeDiagram._Graph</a></strong>: <em class="summary">Graph module.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Graphics.GenomeDiagram._GraphSet-module.html" onclick="show_private();">Bio.Graphics.GenomeDiagram._GraphSet</a></strong>: <em class="summary">GraphSet module.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Graphics.GenomeDiagram._LinearDrawer-module.html" onclick="show_private();">Bio.Graphics.GenomeDiagram._LinearDrawer</a></strong>: <em class="summary">Linear Drawer module.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Graphics.GenomeDiagram._Track-module.html" onclick="show_private();">Bio.Graphics.GenomeDiagram._Track</a></strong>: <em class="summary">Track module.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.Graphics.KGML_vis-module.html">Bio.Graphics.KGML_vis</a></strong>: <em class="summary">Classes and functions to visualise a KGML Pathway Map.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.HMM-module.html">Bio.HMM</a></strong>: <em class="summary">A selection of Hidden Markov Model code.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.HMM.DynamicProgramming-module.html">Bio.HMM.DynamicProgramming</a></strong>: <em class="summary">Dynamic Programming algorithms for general usage.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.HMM.MarkovModel-module.html">Bio.HMM.MarkovModel</a></strong>: <em class="summary">Deal with representations of Markov Models.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.HMM.Trainer-module.html">Bio.HMM.Trainer</a></strong>: <em class="summary">Provide trainers which estimate parameters based on training sequences.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.HMM.Utilities-module.html">Bio.HMM.Utilities</a></strong>: <em class="summary">Generic functions which are useful for working with HMMs.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.Index-module.html">Bio.Index</a></strong>: <em class="summary">Index text files.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.KDTree-module.html">Bio.KDTree</a></strong>: <em class="summary">KD tree data structure for searching N-dimensional vectors (OBSOLETE).</em>
    <ul>
    <li> <strong class="uidlink">Bio.KDTree.KDTree</strong>    </li>
    <li> <strong class="uidlink"><a href="Bio.KDTree.KDTree%27-module.html">Bio.KDTree.KDTree'</a></strong>: <em class="summary">KD tree data structure for searching N-dimensional vectors.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.KDTree._CKDTree-module.html" onclick="show_private();">Bio.KDTree._CKDTree</a></strong>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.KEGG-module.html">Bio.KEGG</a></strong>: <em class="summary">Code to work with data from the KEGG database.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.KEGG.Compound-module.html">Bio.KEGG.Compound</a></strong>: <em class="summary">Code to work with the KEGG Ligand/Compound database.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.KEGG.Enzyme-module.html">Bio.KEGG.Enzyme</a></strong>: <em class="summary">Code to work with the KEGG Enzyme database.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.KEGG.Gene-module.html">Bio.KEGG.Gene</a></strong>: <em class="summary">Code to work with the KEGG Gene database.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.KEGG.KGML-module.html">Bio.KEGG.KGML</a></strong>: <em class="summary">Code to work with data from the KEGG database.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.KEGG.KGML.KGML_parser-module.html">Bio.KEGG.KGML.KGML_parser</a></strong>: <em class="summary">Classes and functions to parse a KGML pathway map.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.KEGG.KGML.KGML_pathway-module.html">Bio.KEGG.KGML.KGML_pathway</a></strong>: <em class="summary">Classes to represent a KGML Pathway Map.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.KEGG.Map-module.html">Bio.KEGG.Map</a></strong>: <em class="summary">Load KEGG Pathway maps for use with the Biopython Pathway module.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.KEGG.REST-module.html">Bio.KEGG.REST</a></strong>: <em class="summary">Provides code to access the REST-style KEGG online API.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.LogisticRegression-module.html">Bio.LogisticRegression</a></strong>: <em class="summary">Code for doing logistic regressions.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.MarkovModel-module.html">Bio.MarkovModel</a></strong>: <em class="summary">A state-emitting MarkovModel.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.MaxEntropy-module.html">Bio.MaxEntropy</a></strong>: <em class="summary">Maximum Entropy code.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Medline-module.html">Bio.Medline</a></strong>: <em class="summary">Code to work with Medline from the NCBI.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.NMR-module.html">Bio.NMR</a></strong>: <em class="summary">Code for working with NMR data.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.NMR.NOEtools-module.html">Bio.NMR.NOEtools</a></strong>: <em class="summary">NOEtools: For predicting NOE coordinates from assignment data.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.NMR.xpktools-module.html">Bio.NMR.xpktools</a></strong>: <em class="summary">Tools to manipulate data from nmrview .xpk peaklist files.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.NaiveBayes-module.html">Bio.NaiveBayes</a></strong>: <em class="summary">General Naive Bayes learner.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.NeuralNetwork-module.html">Bio.NeuralNetwork</a></strong>: <em class="summary">Neural Network library (DEPRECATED).</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.NeuralNetwork.BackPropagation-module.html">Bio.NeuralNetwork.BackPropagation</a></strong>: <em class="summary">Back-propagation for neural networks.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.NeuralNetwork.BackPropagation.Layer-module.html">Bio.NeuralNetwork.BackPropagation.Layer</a></strong>: <em class="summary">Model a single layer in a nueral network.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.NeuralNetwork.BackPropagation.Network-module.html">Bio.NeuralNetwork.BackPropagation.Network</a></strong>: <em class="summary">Represent Neural Networks.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.NeuralNetwork.Gene-module.html">Bio.NeuralNetwork.Gene</a></strong>: <em class="summary">Module for using neural networks to model genetic data.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.NeuralNetwork.Gene.Motif-module.html">Bio.NeuralNetwork.Gene.Motif</a></strong>: <em class="summary">Find and deal with motifs in biological sequence data.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.NeuralNetwork.Gene.Pattern-module.html">Bio.NeuralNetwork.Gene.Pattern</a></strong>: <em class="summary">Generic functionality useful for all gene representations.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.NeuralNetwork.Gene.Schema-module.html">Bio.NeuralNetwork.Gene.Schema</a></strong>: <em class="summary">Deal with Motifs or Signatures allowing ambiguity in the sequences.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.NeuralNetwork.Gene.Signature-module.html">Bio.NeuralNetwork.Gene.Signature</a></strong>: <em class="summary">Find and deal with signatures in biological sequence data.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.NeuralNetwork.StopTraining-module.html">Bio.NeuralNetwork.StopTraining</a></strong>: <em class="summary">Classes to help deal with stopping training a neural network.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.NeuralNetwork.Training-module.html">Bio.NeuralNetwork.Training</a></strong>: <em class="summary">Provide classes for dealing with Training Neural Networks.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.Nexus-module.html">Bio.Nexus</a></strong>: <em class="summary">The Bio.Nexus contains a NEXUS file parser and objects to model this data.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.Nexus.Nexus-module.html">Bio.Nexus.Nexus</a></strong>: <em class="summary">Nexus class. Parse the contents of a NEXUS file.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Nexus.Nodes-module.html">Bio.Nexus.Nodes</a></strong>: <em class="summary">Linked list functionality for use in Bio.Nexus.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Nexus.StandardData-module.html">Bio.Nexus.StandardData</a></strong>: <em class="summary">Objects to represent NEXUS standard data type matrix coding.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Nexus.Trees-module.html">Bio.Nexus.Trees</a></strong>: <em class="summary">Tree class to handle phylogenetic trees.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Nexus.cnexus-module.html">Bio.Nexus.cnexus</a></strong>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB-module.html">Bio.PDB</a></strong>: <em class="summary">Classes that deal with macromolecular crystal structures.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.PDB.AbstractPropertyMap-module.html">Bio.PDB.AbstractPropertyMap</a></strong>: <em class="summary">Class that maps (chain_id, residue_id) to a residue property.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.Atom-module.html">Bio.PDB.Atom</a></strong>: <em class="summary">Atom class, used in Structure objects.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.Chain-module.html">Bio.PDB.Chain</a></strong>: <em class="summary">Chain class, used in Structure objects.</em>    </li>
    <li> <strong class="uidlink">Bio.PDB.DSSP</strong>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.DSSP%27-module.html">Bio.PDB.DSSP'</a></strong>: <em class="summary">Use the DSSP program to calculate secondary structure and accessibility.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.Dice-module.html">Bio.PDB.Dice</a></strong>: <em class="summary">Code for chopping up (dicing) a structure.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.Entity-module.html">Bio.PDB.Entity</a></strong>: <em class="summary">Base class for Residue, Chain, Model and Structure classes.</em>    </li>
    <li> <strong class="uidlink">Bio.PDB.FragmentMapper</strong>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.FragmentMapper%27-module.html">Bio.PDB.FragmentMapper'</a></strong>: <em class="summary">Classify protein backbone structure with Kolodny et al's fragment libraries.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.HSExposure-module.html">Bio.PDB.HSExposure</a></strong>: <em class="summary">Half-sphere exposure and coordination number calculation.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.MMCIF2Dict-module.html">Bio.PDB.MMCIF2Dict</a></strong>: <em class="summary">Turn an mmCIF file into a dictionary.</em>    </li>
    <li> <strong class="uidlink">Bio.PDB.MMCIFParser</strong>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.MMCIFParser%27-module.html">Bio.PDB.MMCIFParser'</a></strong>: <em class="summary">mmCIF parsers.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.Model-module.html">Bio.PDB.Model</a></strong>: <em class="summary">Model class, used in Structure objects.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.NACCESS-module.html">Bio.PDB.NACCESS</a></strong>: <em class="summary">Interface for the program NACCESS.</em>    </li>
    <li> <strong class="uidlink">Bio.PDB.NeighborSearch</strong>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.NeighborSearch%27-module.html">Bio.PDB.NeighborSearch'</a></strong>: <em class="summary">Fast atom neighbor lookup using a KD tree (implemented in C).</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.PDBExceptions-module.html">Bio.PDB.PDBExceptions</a></strong>: <em class="summary">Some Bio.PDB-specific exceptions.</em>    </li>
    <li> <strong class="uidlink">Bio.PDB.PDBIO</strong>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.PDBIO%27-module.html">Bio.PDB.PDBIO'</a></strong>: <em class="summary">Output of PDB files.</em>    </li>
    <li> <strong class="uidlink">Bio.PDB.PDBList</strong>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.PDBList%27-module.html">Bio.PDB.PDBList'</a></strong>: <em class="summary">Access the PDB over the internet (e.g. to download structures).</em>    </li>
    <li> <strong class="uidlink">Bio.PDB.PDBParser</strong>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.PDBParser%27-module.html">Bio.PDB.PDBParser'</a></strong>: <em class="summary">Parser for PDB files.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.PSEA-module.html">Bio.PDB.PSEA</a></strong>: <em class="summary">Wrappers for PSEA, a program for secondary structure assignment.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.Polypeptide-module.html">Bio.PDB.Polypeptide</a></strong>: <em class="summary">Polypeptide-related classes (construction and representation).</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.QCPSuperimposer-module.html">Bio.PDB.QCPSuperimposer</a></strong>: <em class="summary">Structural alignment using Quaternion Characteristic Polynomial (QCP).</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.PDB.QCPSuperimposer.qcprotmodule-module.html">Bio.PDB.QCPSuperimposer.qcprotmodule</a></strong>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.Residue-module.html">Bio.PDB.Residue</a></strong>: <em class="summary">Residue class, used by Structure objects.</em>    </li>
    <li> <strong class="uidlink">Bio.PDB.ResidueDepth</strong>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.ResidueDepth%27-module.html">Bio.PDB.ResidueDepth'</a></strong>: <em class="summary">Calculation of residue depth using command line tool MSMS.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.Selection-module.html">Bio.PDB.Selection</a></strong>: <em class="summary">Selection of atoms, residues, etc.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.Structure-module.html">Bio.PDB.Structure</a></strong>: <em class="summary">The structure class, representing a macromolecular structure.</em>    </li>
    <li> <strong class="uidlink">Bio.PDB.StructureAlignment</strong>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.StructureAlignment%27-module.html">Bio.PDB.StructureAlignment'</a></strong>: <em class="summary">Map residues of two structures to each other based on a FASTA alignment.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.StructureBuilder-module.html">Bio.PDB.StructureBuilder</a></strong>: <em class="summary">Consumer class that builds a Structure object.</em>    </li>
    <li> <strong class="uidlink">Bio.PDB.Superimposer</strong>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.Superimposer%27-module.html">Bio.PDB.Superimposer'</a></strong>: <em class="summary">Superimpose two structures.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.Vector-module.html">Bio.PDB.Vector</a></strong>: <em class="summary">Deprecated in favor of Bio.PDB.vectors to solve name collision.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.kdtrees-module.html">Bio.PDB.kdtrees</a></strong>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.mmcifio-module.html">Bio.PDB.mmcifio</a></strong>: <em class="summary">Write an mmCIF file.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.mmtf-module.html">Bio.PDB.mmtf</a></strong>: <em class="summary">Support for loading 3D structures stored in MMTF files.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.PDB.mmtf.DefaultParser-module.html">Bio.PDB.mmtf.DefaultParser</a></strong>: <em class="summary">Code handle loading mmtf-python into Biopython's structures.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink">Bio.PDB.parse_pdb_header</strong>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.parse_pdb_header%27-module.html">Bio.PDB.parse_pdb_header'</a></strong>: <em class="summary">Parse the header of a PDB file.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.PDB.vectors-module.html">Bio.PDB.vectors</a></strong>: <em class="summary">Vector class, including rotation-related functions.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.ParserSupport-module.html">Bio.ParserSupport</a></strong>: <em class="summary">Code to support writing parsers (DEPRECATED).</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Pathway-module.html">Bio.Pathway</a></strong>: <em class="summary">BioPython Pathway module.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.Pathway.Rep-module.html">Bio.Pathway.Rep</a></strong>: <em class="summary">BioPython Pathway support module.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.Pathway.Rep.Graph-module.html">Bio.Pathway.Rep.Graph</a></strong>    </li>
    <li> <strong class="uidlink"><a href="Bio.Pathway.Rep.MultiGraph-module.html">Bio.Pathway.Rep.MultiGraph</a></strong>    </li>
    </ul>
    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.Phylo-module.html">Bio.Phylo</a></strong>: <em class="summary">Package for working with phylogenetic trees.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.Phylo.Applications-module.html">Bio.Phylo.Applications</a></strong>: <em class="summary">Phylogenetics command line tool wrappers.</em>
    <ul>
    <li class="private"> <strong class="uidlink"><a href="Bio.Phylo.Applications._Fasttree-module.html" onclick="show_private();">Bio.Phylo.Applications._Fasttree</a></strong>: <em class="summary">Command-line wrapper for tree inference program Fasttree.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Phylo.Applications._Phyml-module.html" onclick="show_private();">Bio.Phylo.Applications._Phyml</a></strong>: <em class="summary">Command-line wrapper for the tree inference program PhyML.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Phylo.Applications._Raxml-module.html" onclick="show_private();">Bio.Phylo.Applications._Raxml</a></strong>: <em class="summary">Command-line wrapper for the tree inference program RAxML.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.Phylo.BaseTree-module.html">Bio.Phylo.BaseTree</a></strong>: <em class="summary">Base classes for Bio.Phylo objects.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Phylo.CDAO-module.html">Bio.Phylo.CDAO</a></strong>: <em class="summary">Classes corresponding to CDAO trees.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Phylo.CDAOIO-module.html">Bio.Phylo.CDAOIO</a></strong>: <em class="summary">I/O function wrappers for the RDF/CDAO file format.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Phylo.Consensus-module.html">Bio.Phylo.Consensus</a></strong>: <em class="summary">Classes and methods for finding consensus trees.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Phylo.NeXML-module.html">Bio.Phylo.NeXML</a></strong>: <em class="summary">Classes corresponding to NeXML trees.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Phylo.NeXMLIO-module.html">Bio.Phylo.NeXMLIO</a></strong>: <em class="summary">I/O function wrappers for the NeXML file format.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Phylo.Newick-module.html">Bio.Phylo.Newick</a></strong>: <em class="summary">Classes corresponding to Newick trees, also used for Nexus trees.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Phylo.NewickIO-module.html">Bio.Phylo.NewickIO</a></strong>: <em class="summary">I/O function wrappers for the Newick file format.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Phylo.NexusIO-module.html">Bio.Phylo.NexusIO</a></strong>: <em class="summary">I/O function wrappers for <a href="Bio.Nexus-module.html" class="link">Bio.Nexus</a> trees.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Phylo.PAML-module.html">Bio.Phylo.PAML</a></strong>: <em class="summary">Support for PAML.</em>
    <ul>
    <li class="private"> <strong class="uidlink"><a href="Bio.Phylo.PAML._paml-module.html">Bio.Phylo.PAML._paml</a></strong>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Phylo.PAML._parse_baseml-module.html">Bio.Phylo.PAML._parse_baseml</a></strong>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Phylo.PAML._parse_codeml-module.html">Bio.Phylo.PAML._parse_codeml</a></strong>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Phylo.PAML._parse_yn00-module.html">Bio.Phylo.PAML._parse_yn00</a></strong>    </li>
    <li> <strong class="uidlink"><a href="Bio.Phylo.PAML.baseml-module.html">Bio.Phylo.PAML.baseml</a></strong>    </li>
    <li> <strong class="uidlink"><a href="Bio.Phylo.PAML.chi2-module.html">Bio.Phylo.PAML.chi2</a></strong>    </li>
    <li> <strong class="uidlink"><a href="Bio.Phylo.PAML.codeml-module.html">Bio.Phylo.PAML.codeml</a></strong>    </li>
    <li> <strong class="uidlink"><a href="Bio.Phylo.PAML.yn00-module.html">Bio.Phylo.PAML.yn00</a></strong>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.Phylo.PhyloXML-module.html">Bio.Phylo.PhyloXML</a></strong>: <em class="summary">Classes corresponding to phyloXML elements.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Phylo.PhyloXMLIO-module.html">Bio.Phylo.PhyloXMLIO</a></strong>: <em class="summary">PhyloXML reader/parser, writer, and associated functions.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Phylo.TreeConstruction-module.html">Bio.Phylo.TreeConstruction</a></strong>: <em class="summary">Classes and methods for tree construction.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Phylo._cdao_owl-module.html" onclick="show_private();">Bio.Phylo._cdao_owl</a></strong>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Phylo._io-module.html" onclick="show_private();">Bio.Phylo._io</a></strong>: <em class="summary">I/O function wrappers for phylogenetic tree formats.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Phylo._utils-module.html" onclick="show_private();">Bio.Phylo._utils</a></strong>: <em class="summary">Utilities for handling, displaying and exporting Phylo trees.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.PopGen-module.html">Bio.PopGen</a></strong>: <em class="summary">PopGen: Population Genetics and Genomics library in Python.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.PopGen.GenePop-module.html">Bio.PopGen.GenePop</a></strong>: <em class="summary">Code to work with GenePop.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.PopGen.GenePop.Controller-module.html">Bio.PopGen.GenePop.Controller</a></strong>: <em class="summary">Module to control GenePop.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.PopGen.GenePop.EasyController-module.html">Bio.PopGen.GenePop.EasyController</a></strong>: <em class="summary">Control GenePop through an easier interface.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.PopGen.GenePop.FileParser-module.html">Bio.PopGen.GenePop.FileParser</a></strong>: <em class="summary">Code to parse BIG GenePop files.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.PopGen.GenePop.LargeFileParser-module.html">Bio.PopGen.GenePop.LargeFileParser</a></strong>: <em class="summary">Large file parsing of Genepop files.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.PopGen.GenePop.Utils-module.html">Bio.PopGen.GenePop.Utils</a></strong>    </li>
    </ul>
    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.Restriction-module.html">Bio.Restriction</a></strong>: <em class="summary">Restriction Digest Enzymes.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.Restriction.PrintFormat-module.html">Bio.Restriction.PrintFormat</a></strong>: <em class="summary">Print the results of restriction enzyme analysis.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Restriction.RanaConfig-module.html">Bio.Restriction.RanaConfig</a></strong>: <em class="summary">Configuration of the console.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Restriction.Restriction-module.html">Bio.Restriction.Restriction</a></strong>: <em class="summary">Restriction Enzyme classes.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Restriction.Restriction_Dictionary-module.html">Bio.Restriction.Restriction_Dictionary</a></strong>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.SCOP-module.html">Bio.SCOP</a></strong>: <em class="summary">SCOP: Structural Classification of Proteins.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.SCOP.Cla-module.html">Bio.SCOP.Cla</a></strong>: <em class="summary">Handle the SCOP CLAssification file, which describes SCOP domains.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SCOP.Des-module.html">Bio.SCOP.Des</a></strong>: <em class="summary">Handle the SCOP DEScription file.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SCOP.Dom-module.html">Bio.SCOP.Dom</a></strong>: <em class="summary">Handle the SCOP DOMain file.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SCOP.Hie-module.html">Bio.SCOP.Hie</a></strong>: <em class="summary">Handle the SCOP HIErarchy files.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SCOP.Raf-module.html">Bio.SCOP.Raf</a></strong>: <em class="summary">ASTRAL RAF (Rapid Access Format) Sequence Maps.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SCOP.Residues-module.html">Bio.SCOP.Residues</a></strong>: <em class="summary">A collection of residues from a PDB structure.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.SVDSuperimposer-module.html">Bio.SVDSuperimposer</a></strong>: <em class="summary">Align on protein structure onto another using SVD alignment.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SearchIO-module.html">Bio.SearchIO</a></strong>: <em class="summary">Biopython interface for sequence search program outputs.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.SearchIO.BlastIO-module.html">Bio.SearchIO.BlastIO</a></strong>: <em class="summary">Bio.SearchIO support for BLAST+ output formats.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.SearchIO.BlastIO.blast_tab-module.html">Bio.SearchIO.BlastIO.blast_tab</a></strong>: <em class="summary">Bio.SearchIO parser for BLAST+ tab output format, with or without comments.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SearchIO.BlastIO.blast_text-module.html">Bio.SearchIO.BlastIO.blast_text</a></strong>: <em class="summary">Bio.SearchIO parser for BLAST+ plain text output formats.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SearchIO.BlastIO.blast_xml-module.html">Bio.SearchIO.BlastIO.blast_xml</a></strong>: <em class="summary">Bio.SearchIO parser for BLAST+ XML output formats.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.SearchIO.BlatIO-module.html">Bio.SearchIO.BlatIO</a></strong>: <em class="summary">Bio.SearchIO parser for BLAT output formats.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SearchIO.ExonerateIO-module.html">Bio.SearchIO.ExonerateIO</a></strong>: <em class="summary">Bio.SearchIO support for Exonerate output formats.</em>
    <ul>
    <li class="private"> <strong class="uidlink"><a href="Bio.SearchIO.ExonerateIO._base-module.html" onclick="show_private();">Bio.SearchIO.ExonerateIO._base</a></strong>: <em class="summary">Bio.SearchIO abstract base parser for Exonerate standard output format.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SearchIO.ExonerateIO.exonerate_cigar-module.html">Bio.SearchIO.ExonerateIO.exonerate_cigar</a></strong>: <em class="summary">Bio.SearchIO parser for Exonerate cigar output format.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SearchIO.ExonerateIO.exonerate_text-module.html">Bio.SearchIO.ExonerateIO.exonerate_text</a></strong>: <em class="summary">Bio.SearchIO parser for Exonerate plain text output format.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SearchIO.ExonerateIO.exonerate_vulgar-module.html">Bio.SearchIO.ExonerateIO.exonerate_vulgar</a></strong>: <em class="summary">Bio.SearchIO parser for Exonerate vulgar output format.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.SearchIO.FastaIO-module.html">Bio.SearchIO.FastaIO</a></strong>: <em class="summary">Bio.SearchIO support for Bill Pearson's FASTA tools.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SearchIO.HmmerIO-module.html">Bio.SearchIO.HmmerIO</a></strong>: <em class="summary">Bio.SearchIO support for HMMER output formats.</em>
    <ul>
    <li class="private"> <strong class="uidlink"><a href="Bio.SearchIO.HmmerIO._base-module.html" onclick="show_private();">Bio.SearchIO.HmmerIO._base</a></strong>: <em class="summary">Bio.SearchIO base classes for HMMER-related code.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SearchIO.HmmerIO.hmmer2_text-module.html">Bio.SearchIO.HmmerIO.hmmer2_text</a></strong>: <em class="summary">Bio.SearchIO parser for HMMER 2 text output.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SearchIO.HmmerIO.hmmer3_domtab-module.html">Bio.SearchIO.HmmerIO.hmmer3_domtab</a></strong>: <em class="summary">Bio.SearchIO parser for HMMER domain table output format.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SearchIO.HmmerIO.hmmer3_tab-module.html">Bio.SearchIO.HmmerIO.hmmer3_tab</a></strong>: <em class="summary">Bio.SearchIO parser for HMMER table output format.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SearchIO.HmmerIO.hmmer3_text-module.html">Bio.SearchIO.HmmerIO.hmmer3_text</a></strong>: <em class="summary">Bio.SearchIO parser for HMMER plain text output format.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.SearchIO.InterproscanIO-module.html">Bio.SearchIO.InterproscanIO</a></strong>: <em class="summary">Bio.SearchIO support for InterProScan output formats.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.SearchIO.InterproscanIO.interproscan_xml-module.html">Bio.SearchIO.InterproscanIO.interproscan_xml</a></strong>: <em class="summary">Bio.SearchIO parser for InterProScan XML output formats.</em>    </li>
    </ul>
    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.SearchIO._index-module.html">Bio.SearchIO._index</a></strong>: <em class="summary">Custom indexing for Bio.SearchIO objects.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.SearchIO._model-module.html" onclick="show_private();">Bio.SearchIO._model</a></strong>: <em class="summary">Bio.SearchIO objects to model similarity search program outputs.</em>
    <ul class="private">
    <li class="private"> <strong class="uidlink"><a href="Bio.SearchIO._model._base-module.html" onclick="show_private();">Bio.SearchIO._model._base</a></strong>: <em class="summary">Abstract base classes for the SearchIO object model.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.SearchIO._model.hit-module.html" onclick="show_private();">Bio.SearchIO._model.hit</a></strong>: <em class="summary">Bio.SearchIO object to model a single database hit.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.SearchIO._model.hsp-module.html" onclick="show_private();">Bio.SearchIO._model.hsp</a></strong>: <em class="summary">Bio.SearchIO objects to model high scoring regions between query and hit.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.SearchIO._model.query-module.html" onclick="show_private();">Bio.SearchIO._model.query</a></strong>: <em class="summary">Bio.SearchIO object to model search results from a single query.</em>    </li>
    </ul>
    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.SearchIO._utils-module.html" onclick="show_private();">Bio.SearchIO._utils</a></strong>: <em class="summary">Common SearchIO utility functions.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.Seq-module.html">Bio.Seq</a></strong>: <em class="summary">Provide objects to represent biological sequences with alphabets.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqFeature-module.html">Bio.SeqFeature</a></strong>: <em class="summary">Represent a Sequence Feature holding info about a part of a sequence.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqIO-module.html">Bio.SeqIO</a></strong>: <em class="summary">Sequence input/output as SeqRecord objects.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.SeqIO.AbiIO-module.html">Bio.SeqIO.AbiIO</a></strong>: <em class="summary">Bio.SeqIO parser for the ABI format.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqIO.AceIO-module.html">Bio.SeqIO.AceIO</a></strong>: <em class="summary">Bio.SeqIO support for the &quot;ace&quot; file format.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqIO.FastaIO-module.html">Bio.SeqIO.FastaIO</a></strong>: <em class="summary">Bio.SeqIO support for the &quot;fasta&quot; (aka FastA or Pearson) file format.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqIO.IgIO-module.html">Bio.SeqIO.IgIO</a></strong>: <em class="summary">Bio.SeqIO support for the &quot;ig&quot; (IntelliGenetics or MASE) file format.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqIO.InsdcIO-module.html">Bio.SeqIO.InsdcIO</a></strong>: <em class="summary">Bio.SeqIO support for the &quot;genbank&quot; and &quot;embl&quot; file formats.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqIO.Interfaces-module.html">Bio.SeqIO.Interfaces</a></strong>: <em class="summary">Bio.SeqIO support module (not for general use).</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqIO.PdbIO-module.html">Bio.SeqIO.PdbIO</a></strong>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqIO.PhdIO-module.html">Bio.SeqIO.PhdIO</a></strong>: <em class="summary">Bio.SeqIO support for the &quot;phd&quot; file format.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqIO.PirIO-module.html">Bio.SeqIO.PirIO</a></strong>: <em class="summary">Bio.SeqIO support for the &quot;pir&quot; (aka PIR or NBRF) file format.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqIO.QualityIO-module.html">Bio.SeqIO.QualityIO</a></strong>: <em class="summary">Bio.SeqIO support for the FASTQ and QUAL file formats.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqIO.SeqXmlIO-module.html">Bio.SeqIO.SeqXmlIO</a></strong>: <em class="summary">Bio.SeqIO support for the &quot;seqxml&quot; file format, SeqXML.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqIO.SffIO-module.html">Bio.SeqIO.SffIO</a></strong>: <em class="summary">Bio.SeqIO support for the binary Standard Flowgram Format (SFF) file format.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqIO.SwissIO-module.html">Bio.SeqIO.SwissIO</a></strong>: <em class="summary">Bio.SeqIO support for the &quot;swiss&quot; (aka SwissProt/UniProt) file format.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqIO.TabIO-module.html">Bio.SeqIO.TabIO</a></strong>: <em class="summary">Bio.SeqIO support for the &quot;tab&quot; (simple tab separated) file format.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqIO.UniprotIO-module.html">Bio.SeqIO.UniprotIO</a></strong>: <em class="summary">Bio.SeqIO support for the &quot;uniprot-xml&quot; file format.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.SeqIO._convert-module.html">Bio.SeqIO._convert</a></strong>: <em class="summary">Optimised sequence conversion code (PRIVATE).</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.SeqIO._index-module.html">Bio.SeqIO._index</a></strong>: <em class="summary">Dictionary like indexing of sequence files (PRIVATE).</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqRecord-module.html">Bio.SeqRecord</a></strong>: <em class="summary">Represent a Sequence Record, a sequence with annotation.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqUtils-module.html">Bio.SeqUtils</a></strong>: <em class="summary">Miscellaneous functions for dealing with sequences.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.SeqUtils.CheckSum-module.html">Bio.SeqUtils.CheckSum</a></strong>: <em class="summary">Functions to calculate assorted sequence checksums.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqUtils.CodonUsage-module.html">Bio.SeqUtils.CodonUsage</a></strong>: <em class="summary">Methods for codon usage calculations.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqUtils.CodonUsageIndices-module.html">Bio.SeqUtils.CodonUsageIndices</a></strong>: <em class="summary">Codon adaption indxes, including Sharp and Li (1987) E. coli index.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqUtils.IsoelectricPoint-module.html">Bio.SeqUtils.IsoelectricPoint</a></strong>: <em class="summary">Calculate isoelectric points of polypeptides using methods of Bjellqvist.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqUtils.MeltingTemp-module.html">Bio.SeqUtils.MeltingTemp</a></strong>: <em class="summary">Calculate the melting temperature of nucleotide sequences.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqUtils.ProtParam-module.html">Bio.SeqUtils.ProtParam</a></strong>: <em class="summary">Simple protein analysis.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqUtils.ProtParamData-module.html">Bio.SeqUtils.ProtParamData</a></strong>: <em class="summary">Indices to be used with ProtParam.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SeqUtils.lcc-module.html">Bio.SeqUtils.lcc</a></strong>: <em class="summary">Local Composition Complexity.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.Sequencing-module.html">Bio.Sequencing</a></strong>: <em class="summary">Code to deal with various programs for sequencing and assembly.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.Sequencing.Ace-module.html">Bio.Sequencing.Ace</a></strong>: <em class="summary">Parser for ACE files output by PHRAP.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Sequencing.Applications-module.html">Bio.Sequencing.Applications</a></strong>: <em class="summary">Main entry point for sequencing related command line application wrappers.</em>
    <ul>
    <li class="private"> <strong class="uidlink"><a href="Bio.Sequencing.Applications._Novoalign-module.html" onclick="show_private();">Bio.Sequencing.Applications._Novoalign</a></strong>: <em class="summary">Command line wrapper for the short read aligner Novoalign by Novocraft.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Sequencing.Applications._bwa-module.html" onclick="show_private();">Bio.Sequencing.Applications._bwa</a></strong>: <em class="summary">Command line wrapper for bwa.</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio.Sequencing.Applications._samtools-module.html" onclick="show_private();">Bio.Sequencing.Applications._samtools</a></strong>: <em class="summary">Command line wrapper for samtools.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.Sequencing.Phd-module.html">Bio.Sequencing.Phd</a></strong>: <em class="summary">Parser for PHD files output by PHRED and used by PHRAP and CONSED.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.Statistics-module.html">Bio.Statistics</a></strong>: <em class="summary">Basic statistics module.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.Statistics.lowess-module.html">Bio.Statistics.lowess</a></strong>: <em class="summary">Implements the Lowess function for nonparametric regression.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.SubsMat-module.html">Bio.SubsMat</a></strong>: <em class="summary">Substitution matrices, log odds matrices, and operations on them.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.SubsMat.FreqTable-module.html">Bio.SubsMat.FreqTable</a></strong>: <em class="summary">A class to handle frequency tables or letter count files.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.SubsMat.MatrixInfo-module.html">Bio.SubsMat.MatrixInfo</a></strong>: <em class="summary">Substitution matrices for use in alignments, etc.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.SwissProt-module.html">Bio.SwissProt</a></strong>: <em class="summary">Code to work with the sprotXX.dat file from SwissProt.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.SwissProt.KeyWList-module.html">Bio.SwissProt.KeyWList</a></strong>: <em class="summary">Code to parse the keywlist.txt file from SwissProt/UniProt.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.TogoWS-module.html">Bio.TogoWS</a></strong>: <em class="summary">Provides code to access the TogoWS integrated websevices of DBCLS, Japan.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.UniGene-module.html">Bio.UniGene</a></strong>: <em class="summary">Parse Unigene flat file format files such as the Hs.data file.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.UniProt-module.html">Bio.UniProt</a></strong>: <em class="summary">Code for dealing with assorted UniProt file formats.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.UniProt.GOA-module.html">Bio.UniProt.GOA</a></strong>: <em class="summary">Parsers for the GAF, GPA and GPI formats from UniProt-GOA.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.Wise-module.html">Bio.Wise</a></strong>: <em class="summary">Run and process output from the Wise2 package tools.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.Wise.dnal-module.html">Bio.Wise.dnal</a></strong>: <em class="summary">Run and process output from the Wise2 package tool dnal.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.Wise.psw-module.html">Bio.Wise.psw</a></strong>: <em class="summary">Run and process output from the Wise2 package tool psw.</em>    </li>
    </ul>
    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio._py3k-module.html">Bio._py3k</a></strong>: <em class="summary">Python 3 compatibility tools (PRIVATE).</em>    </li>
    <li class="private"> <strong class="uidlink"><a href="Bio._utils-module.html">Bio._utils</a></strong>: <em class="summary">Common utility functions for various Bio submodules.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.bgzf-module.html">Bio.bgzf</a></strong>: <em class="summary">Read and write BGZF compressed files (the GZIP variant used in BAM).</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.codonalign-module.html">Bio.codonalign</a></strong>: <em class="summary">Code for dealing with Codon Alignments.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.codonalign.chisq-module.html">Bio.codonalign.chisq</a></strong>: <em class="summary">Python implementation of chisqprob, to avoid SciPy dependency.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.codonalign.codonalignment-module.html">Bio.codonalign.codonalignment</a></strong>: <em class="summary">Code for dealing with Codon Alignment.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.codonalign.codonalphabet-module.html">Bio.codonalign.codonalphabet</a></strong>: <em class="summary">Code for Codon Alphabet.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.codonalign.codonseq-module.html">Bio.codonalign.codonseq</a></strong>: <em class="summary">Code for dealing with coding sequence.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.cpairwise2-module.html">Bio.cpairwise2</a></strong>: <em class="summary">Optimized C routines that complement pairwise2.py. These are called from within pairwise2.py.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.kNN-module.html">Bio.kNN</a></strong>: <em class="summary">Code for doing k-nearest-neighbors classification.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.motifs-module.html">Bio.motifs</a></strong>: <em class="summary">Tools for sequence motif analysis.</em>
    <ul>
    <li class="private"> <strong class="uidlink"><a href="Bio.motifs._pwm-module.html">Bio.motifs._pwm</a></strong>: <em class="summary">Fast calculations involving position-weight matrices</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.motifs.alignace-module.html">Bio.motifs.alignace</a></strong>: <em class="summary">Parsing AlignACE output files.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.motifs.applications-module.html">Bio.motifs.applications</a></strong>: <em class="summary">Motif command line tool wrappers.</em>
    <ul>
    <li class="private"> <strong class="uidlink"><a href="Bio.motifs.applications._xxmotif-module.html" onclick="show_private();">Bio.motifs.applications._xxmotif</a></strong>: <em class="summary">Command line wrapper for the motif finding program XXmotif.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.motifs.jaspar-module.html">Bio.motifs.jaspar</a></strong>: <em class="summary">JASPAR2014 module.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.motifs.jaspar.db-module.html">Bio.motifs.jaspar.db</a></strong>: <em class="summary">Provides read access to a JASPAR5 formatted database.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.motifs.mast-module.html">Bio.motifs.mast</a></strong>    </li>
    <li> <strong class="uidlink"><a href="Bio.motifs.matrix-module.html">Bio.motifs.matrix</a></strong>: <em class="summary">Support for various forms of sequence motif matrices.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.motifs.meme-module.html">Bio.motifs.meme</a></strong>    </li>
    <li> <strong class="uidlink"><a href="Bio.motifs.thresholds-module.html">Bio.motifs.thresholds</a></strong>: <em class="summary">Approximate calculation of appropriate thresholds for motif finding.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.motifs.transfac-module.html">Bio.motifs.transfac</a></strong>: <em class="summary">Parsing TRANSFAC files.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.pairwise2-module.html">Bio.pairwise2</a></strong>: <em class="summary">Pairwise sequence alignment using a dynamic programming algorithm.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.phenotype-module.html">Bio.phenotype</a></strong>: <em class="summary">phenotype data input/output.</em>
    <ul>
    <li> <strong class="uidlink"><a href="Bio.phenotype.phen_micro-module.html">Bio.phenotype.phen_micro</a></strong>: <em class="summary">Classes to work with Phenotype Microarray data.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.phenotype.pm_fitting-module.html">Bio.phenotype.pm_fitting</a></strong>: <em class="summary">Growth curves fitting and parameters extraction for phenotype data.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="Bio.trie-module.html">Bio.trie</a></strong>: <em class="summary">This module implements a trie data structure.  This allows an O(M)
lookup of a string in a dictionary, where M is the length of the
string.  It also supports approximate matches.</em>    </li>
    <li> <strong class="uidlink"><a href="Bio.triefind-module.html">Bio.triefind</a></strong>: <em class="summary">Given a trie, find all occurrences of a word in the trie in a string.</em>    </li>
    </ul>
    </li>
    <li> <strong class="uidlink"><a href="BioSQL-module.html">BioSQL</a></strong>: <em class="summary">Storing and retrieve biological sequences in a BioSQL relational database.</em>
    <ul>
    <li> <strong class="uidlink"><a href="BioSQL.BioSeq-module.html">BioSQL.BioSeq</a></strong>: <em class="summary">Implementations of Biopython-like Seq objects on top of BioSQL.</em>    </li>
    <li> <strong class="uidlink"><a href="BioSQL.BioSeqDatabase-module.html">BioSQL.BioSeqDatabase</a></strong>: <em class="summary">Connect with a BioSQL database and load Biopython like objects from it.</em>    </li>
    <li> <strong class="uidlink"><a href="BioSQL.DBUtils-module.html">BioSQL.DBUtils</a></strong>: <em class="summary">Helper code for Biopython's BioSQL code (for internal use).</em>    </li>
    <li> <strong class="uidlink"><a href="BioSQL.Loader-module.html">BioSQL.Loader</a></strong>: <em class="summary">Load biopython objects into a BioSQL database for persistent storage.</em>    </li>
    </ul>
    </li>
</ul>
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">

  <!-- Tree link -->
      <th bgcolor="#70b0f0" class="navbar-select"
          >&nbsp;&nbsp;&nbsp;Trees&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://biopython.org">Biopython</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Fri Jun 22 16:34:07 2018
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
